*** gnome-shell/gnome-shell.css	2012-09-30 19:20:50.000000000 +0900
--- gnome-shell/gnome-shell_new.css	2012-10-21 11:01:18.340835467 +0900
***************
*** 414,419 ****
--- 414,437 ----
  }
  
  /* Panel */
+ #panelActivities {
+    border: none;
+    background-image: url("start1.png");
+    /*background-position: 5px 4px;*/
+    width: 63px;
+    height: 24px;
+    color: rgba(0,0,0,0.0);
+ }
+ 
+ #panelActivities:hover {
+    transition-duration: 300;
+    background-image:  url("start2.png");
+ 
+ }
+ #panelActivities:active,
+ #panelActivities:overview {
+    background-image:  url("start2.png");
+ }
  
  #panel {
      background-color: rgba(0,0,0,@panel-opacity);
***************
*** 767,778 ****
  
  .icon-grid {
      spacing: 36px;
!     -shell-grid-horizontal-item-size: 118px;
!     -shell-grid-vertical-item-size: 118px;
  }
  
  .icon-grid .overview-icon {
!     icon-size: 84px;
  }
  
  .all-app {
--- 785,796 ----
  
  .icon-grid {
      spacing: 36px;
!     -shell-grid-horizontal-item-size: 84px;
!     -shell-grid-vertical-item-size: 84px;
  }
  
  .icon-grid .overview-icon {
!     icon-size: 64px;
  }
  
  .all-app {
