*** gnome-shell/gnome-shell.css	2012-07-16 14:38:38.000000000 +0900
--- gnome-shell/gnome-shell_new.css	2012-07-16 20:44:34.917582165 +0900
***************
*** 320,325 ****
--- 320,343 ----
  
  /* Panel */
  
+ #panelActivities {
+    border: none;
+    background-image: url("start1.png");
+    background-position: center center;
+    width: 58px;
+    height: 2.1em;
+    color: rgba(0,0,0,0.0);
+ }
+ 
+ #panelActivities:hover {
+    transition-duration: 300;
+    background-image:  url("start2.png");
+ 
+ }
+ #panelActivities:active,
+ #panelActivities:overview {
+    background-image:  url("start1.png");
+ }
  #panel {
      background-color: rgba(0,0,0,@panel-opacity);
      border: 0;
***************
*** 694,706 ****
  /* Application Launchers and Grid */
  
  .icon-grid {
!     spacing: 48px;
!     -shell-grid-horizontal-item-size: 118px;
!     -shell-grid-vertical-item-size: 118px;
  }
  
  .icon-grid .overview-icon {
!     icon-size: 84px;
  }
  
  .all-app {
--- 712,724 ----
  /* Application Launchers and Grid */
  
  .icon-grid {
!     spacing: 36px;
!     -shell-grid-horizontal-item-size: 70px;
!     -shell-grid-vertical-item-size: 70px;
  }
  
  .icon-grid .overview-icon {
!     icon-size: 48px;
  }
  
  .all-app {
***************
*** 813,820 ****
  }
  
  .contact-content {
!     color: #000;
!     background-color: #fff;
      border-radius: 3px;
      padding: 8px;
      width: 218px;
--- 831,840 ----
  }
  
  .contact-content {
!     color: white;
!     background-gradient-direction: vertical;
!     background-gradient-start: rgba(48,48,48,0.9);
!     background-gradient-end: rgba(32,32,32,0.9);
      border-radius: 3px;
      padding: 8px;
      width: 218px;
***************
*** 1176,1182 ****
  #message-tray {
      background-color: rgba(0,0,0,@panel-opacity);
      border: 0;
!     height: 2.1em;
  }
  
  #notification {
--- 1196,1202 ----
  #message-tray {
      background-color: rgba(0,0,0,@panel-opacity);
      border: 0;
!     height: 36px;
  }
  
  #notification {
***************
*** 1417,1422 ****
--- 1437,1443 ----
   */
  #summary-mode {
      padding: 2px 0 0 4px;
+ 	height: 36px;
  }
  
  #summary-mode:rtl {
