*** gnome-shell-theme-elegance-colors/gnome-shell/gnome-shell.css	2012-11-07 05:04:19.000000000 +0900
--- gnome-shell-theme-elegance-colors/gnome-shell/gnome-shell_new.css	2012-11-07 09:14:21.629245240 +0900
***************
*** 431,436 ****
--- 431,454 ----
  }
  
  /* Panel */
+ #panelActivities {
+    border: none;
+    background-image: url("start1.png");
+    /*background-position: 5px 4px;*/
+    width: 63px;
+    height: 24px;
+    color: rgba(0,0,0,0.0);
+ }
+ 
+ #panelActivities:hover {
+    transition-duration: 300;
+    background-image:  url("start2.png");
+ 
+ }
+ #panelActivities:active,
+ #panelActivities:overview {
+    background-image:  url("start2.png");
+ }
  
  #panel {
      background-gradient-direction: vertical;
***************
*** 791,802 ****
  
  .icon-grid {
      spacing: 36px;
!     -shell-grid-horizontal-item-size: 118px;
!     -shell-grid-vertical-item-size: 118px;
  }
  
  .icon-grid .overview-icon {
!     icon-size: 84px;
  }
  
  .all-app {
--- 809,820 ----
  
  .icon-grid {
      spacing: 36px;
!     -shell-grid-horizontal-item-size: 84px;
!     -shell-grid-vertical-item-size: 84px;
  }
  
  .icon-grid .overview-icon {
!     icon-size: 64px;
  }
  
  .all-app {
***************
*** 2390,2393 ****
  .screen-shield-notifications-box .summary-notification-stack-scrollview {
      padding-top: 0;
      padding-bottom: 0;
! }
\ No newline at end of file
--- 2408,2411 ----
  .screen-shield-notifications-box .summary-notification-stack-scrollview {
      padding-top: 0;
      padding-bottom: 0;
! }
