*** gnome-shell/gnome-shell.css	2012-07-13 12:45:56.000000000 +0900
--- gnome-shell/gnome-shell_new.css	2012-07-14 18:23:13.951186993 +0900
***************
*** 320,325 ****
--- 320,344 ----
  
  /* Panel */
  
+ #panelActivities {
+    border: none;
+    background-image: url("start1.png");
+    background-position: center center;
+    width: 58px;
+    height: 2.1em;
+    color: rgba(0,0,0,0.0);
+ }
+ 
+ #panelActivities:hover {
+    transition-duration: 300;
+    background-image:  url("start2.png");
+ 
+ }
+ #panelActivities:active,
+ #panelActivities:overview {
+    background-image:  url("start1.png");
+ }
+ 
  #panel {
      background-color: rgba(0,0,0,@panel-opacity);
      border: 0;
***************
*** 694,706 ****
  /* Application Launchers and Grid */
  
  .icon-grid {
!     spacing: 48px;
!     -shell-grid-horizontal-item-size: 118px;
!     -shell-grid-vertical-item-size: 118px;
  }
  
  .icon-grid .overview-icon {
!     icon-size: 84px;
  }
  
  .all-app {
--- 713,725 ----
  /* Application Launchers and Grid */
  
  .icon-grid {
!     spacing: 36px;
!     -shell-grid-horizontal-item-size: 70px;
!     -shell-grid-vertical-item-size: 70px;
  }
  
  .icon-grid .overview-icon {
!     icon-size: 48px;
  }
  
  .all-app {
***************
*** 1176,1182 ****
  #message-tray {
      background-color: rgba(0,0,0,@panel-opacity);
      border: 0;
!     height: 2.1em;
  }
  
  #notification {
--- 1195,1201 ----
  #message-tray {
      background-color: rgba(0,0,0,@panel-opacity);
      border: 0;
!     height: 36px;
  }
  
  #notification {
