*** gnome-shell/gnome-shell.css	Sat Jan 12 10:48:34 2013
--- gnome-shell/gnome-shell_new.css	Sun Jan 13 09:32:18 2013
***************
*** 498,503 ****
--- 498,521 ----
  }
  
  /* Panel */
+ #panelActivities {
+    border: none;
+    background-image: url("start1.png");
+    /*background-position: 5px 4px;*/
+    width: 63px;
+    height: 24px;
+    color: rgba(0,0,0,0.0);
+ }
+ 
+ #panelActivities:hover {
+    transition-duration: 300;
+    background-image:  url("start2.png");
+ 
+ }
+ #panelActivities:active,
+ #panelActivities:overview {
+    background-image:  url("start2.png");
+ }
  
  #panel {
      color: @panel-color;
