--- a/XnViewMP/XnView.desktop	2011-09-05 10:09:16.000000000 +0200
+++ b/XnViewMP/XnView.desktop	2011-09-08 14:49:06.000000000 +0200
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Encoding=UTF-8
 Terminal=0
-Exec=/home/pierre/Desktop/XnViewMP/xnview.sh
-Icon=/home/pierre/Desktop/XnViewMP/xnview.png
+Exec=/usr/bin/xnviewmp
+Icon=/usr/share/xnviewmp/xnview.png
 Type=Application
 Categories=Graphics;
 StartupNotify=true
