diff -Naur xsys-2.1.0/parse.c xsys-2.1.0.new/parse.c
--- xsys-2.1.0/parse.c	2006-06-03 02:14:39.000000000 +0200
+++ xsys-2.1.0.new/parse.c	2007-04-12 07:32:16.000000000 +0200
@@ -420,6 +420,8 @@
 		fgets(buffer, bsize, fp);
 	else if((fp = fopen("/etc/turbolinux-release", "r")) != NULL)
 		fgets(buffer, bsize, fp);
+	else if((fp = fopen("/etc/arch-release", "r")) != NULL)
+		snprintf(buffer, bsize, "ArchLinux");
 	else
 		snprintf(buffer, bsize, "Unknown Distro");
 	if(fp != NULL) fclose(fp);
