diff -ruN appearance/AppRun appearance_a/AppRun
--- appearance/AppRun	2012-05-08 23:22:48.000000000 -0400
+++ appearance_a/AppRun	2012-05-08 23:28:02.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import findrox; findrox.version(1, 9, 13)
 
 import sys, os
diff -ruN appearance/makedist.sh appearance_a/makedist.sh
--- appearance/makedist.sh	2012-05-08 23:22:48.000000000 -0400
+++ appearance_a/makedist.sh	2012-05-08 23:28:45.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 SCRIPT=`echo "$0" | sed 's|^\./||'`
 echo "$SCRIPT" | grep -q '^/' || SCRIPT=`pwd`/"$SCRIPT"
diff -ruN appearance/Messages/dist appearance_a/Messages/dist
--- appearance/Messages/dist	2012-05-08 23:22:48.000000000 -0400
+++ appearance_a/Messages/dist	2012-05-08 23:29:02.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 echo Making all .gmo files...
 echo
diff -ruN appearance/Messages/update-po appearance_a/Messages/update-po
--- appearance/Messages/update-po	2012-05-08 23:22:48.000000000 -0400
+++ appearance_a/Messages/update-po	2012-05-08 23:29:34.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 echo Extracting messages from source files...
 echo
