diff -r -u -N lightsquid-1.8/lightsquid.cfg lightsquid/lightsquid.cfg
--- lightsquid-1.8/lightsquid.cfg	2009-07-03 00:29:10.000000000 +0300
+++ lightsquid/lightsquid.cfg	2010-05-25 01:18:42.744291470 +0300
@@ -12,17 +12,17 @@
 # -------------------- GLOBAL VARIABLES  ---------------------------
 
 #path to additional `cfg` files
-$cfgpath             ="/var/www/html/lightsquid";
+$cfgpath             ="/usr/share/webapps/lightsquid";
 #path to `tpl` folder
-$tplpath             ="/var/www/html/lightsquid/tpl";
+$tplpath             ="/usr/share/webapps/lightsquid/tpl";
 #path to `lang` folder
-$langpath            ="/var/www/html/lightsquid/lang";
+$langpath            ="/usr/share/webapps/lightsquid/lang";
 #path to `report` folder
-$reportpath          ="/var/www/html/lightsquid/report";
+$reportpath          ="/usr/share/webapps/lightsquid/report";
 #path to access.log
 $logpath             ="/var/log/squid";
 #path to `ip2name` folder
-$ip2namepath         ="/var/www/html/lightsquid/ip2name";
+$ip2namepath         ="/usr/share/webapps/lightsquid/ip2name";
 
 #path to `lockfile` ;-)
 $lockpath            =$reportpath;
