diff -Naur a/Alacarte/MenuEditor.py b/Alacarte/MenuEditor.py
--- a/Alacarte/MenuEditor.py	2012-10-17 16:25:30.000000000 +0000
+++ b/Alacarte/MenuEditor.py	2012-10-24 09:38:46.000000000 +0000
@@ -23,8 +23,11 @@
 from Alacarte import util
 
 class MenuEditor(object):
-    def __init__(self, name=os.environ['XDG_MENU_PREFIX'] + 'applications.menu'):
-        self.name = name
+    def __init__(self, name='applications.menu'):
+        if os.environ.has_key('XDG_MENU_PREFIX'):
+            self.name = os.environ['XDG_MENU_PREFIX'] + name
+        else:
+            self.name = name
 
         self.tree = GMenu.Tree.new(name, GMenu.TreeFlags.SHOW_EMPTY|GMenu.TreeFlags.INCLUDE_EXCLUDED|GMenu.TreeFlags.INCLUDE_NODISPLAY|GMenu.TreeFlags.SHOW_ALL_SEPARATORS|GMenu.TreeFlags.SORT_DISPLAY_NAME)
         self.tree.connect('changed', self.menuChanged)
