Description: MenuEditor: Put back the top-level "Applications" menu
Author: Jasper St. Pierre <jstpierre@mecheye.net>
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=677420
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=676136
Origin: upstream, commit:eeb19696e948c61f11405564fa9d2d91a1fe31cf

---
Index: alacarte-0.13.4/Alacarte/MenuEditor.py
===================================================================
--- alacarte-0.13.4.orig/Alacarte/MenuEditor.py	2012-06-05 22:36:08.265703366 +0100
+++ alacarte-0.13.4/Alacarte/MenuEditor.py	2012-06-05 22:36:08.277703365 +0100
@@ -107,9 +107,10 @@
             pass
         self.save()
 
-    def getMenus(self, parent=None):
+    def getMenus(self, parent):
         if parent is None:
-            parent = self.applications.tree.get_root_directory()
+            yield (self.applications.tree.get_root_directory(), True)
+            return
 
         item_iter = parent.iter()
         item_type = item_iter.next()
