--- tarfile-ops-nt.cc	2009-07-28 00:17:37.000000000 -0500
+++ tarfile-ops-nt.cc	2009-07-28 00:17:04.000000000 -0500
@@ -74,7 +74,7 @@
 		return index;
 	}
 
-	int close_compressed(int fd)
+	int close_compressed(long fd)
 	{
 		unsigned int ufd = fd;
 		assert( ufd < gzArraySize );
@@ -83,14 +83,14 @@
 		return ret;
 	}
 
-	ssize_t read_compressed(int fd, void *buf, size_t size)
+	ssize_t read_compressed(long fd, void *buf, size_t size)
 	{
 		unsigned int ufd = fd;
 		assert( ufd < gzArraySize );
 		return gzread(gzHandles[ufd], buf, size);
 	}
 
-	ssize_t write_compressed(int fd, const void *buf, size_t size)
+	ssize_t write_compressed(long fd, const void *buf, size_t size)
 	{
 		unsigned int ufd = fd;
 		assert( ufd < gzArraySize );
