--- a/globalmenu.cpp	2010-08-16 19:20:11.180114454 +0300
+++ b/globalmenu.cpp	2010-08-16 19:17:19.000000000 +0300
@@ -94,7 +94,7 @@
     rootNode = doc.firstChild();
     for (QDomElement elem = rootNode.firstChildElement(); !elem.isNull(); elem = elem.nextSiblingElement()) {
         if (elem.attribute("visible") != "0") {
-            result << elem.attribute("label").replace("(_","(&");
+            result << elem.attribute("label").replace("(_","(&").replace("_","&");
         }
     }
     return result;
@@ -275,7 +275,7 @@
         }
 
         if (!itemElem.firstChildElement("menu").isNull()) {
-            KMenu *childMenu = new KMenu(itemElem.attribute("label").replace("(_","(&"),topMenu);
+            KMenu *childMenu = new KMenu(itemElem.attribute("label").replace("(_","(&").replace("_","&"),topMenu);
             buildChildMenu(childMenu,itemElem.firstChildElement("menu"),
                     QString("%1/%2").arg(prefix).arg(itemElem.attribute("id")));
             topMenu->addMenu(childMenu);
@@ -283,7 +283,7 @@
         }
 
         if (itemElem.attribute("type") == "i") {
-            action = new KAction(itemElem.attribute("label").replace("(_","(&"),0);
+            action = new KAction(itemElem.attribute("label").replace("(_","(&").replace("_","&"),0);
             mMapper->setMapping(action,QString("%1/%2").arg(prefix).arg(itemElem.attribute("id")));
             connect(action,SIGNAL(triggered()),mMapper,SLOT(map()));
             if (itemElem.attribute("sensible") == "0") {
@@ -292,7 +292,7 @@
             topMenu->addAction(action);
         }
         if (itemElem.attribute("type") == "c") {
-            action = new KAction(itemElem.attribute("label").replace("(_","(&"),0);
+            action = new KAction(itemElem.attribute("label").replace("(_","(&").replace("_","&"),0);
             mMapper->setMapping(action,QString("%1/%2").arg(prefix).arg(itemElem.attribute("id")));
             connect(action,SIGNAL(triggered()),mMapper,SLOT(map()));
             action->setCheckable(true);
@@ -303,7 +303,7 @@
             topMenu->addAction(action);
         }
         if (itemElem.attribute("type") == "r") {
-            action = new KSelectAction(itemElem.attribute("label").replace("(_","(&"),0);
+            action = new KSelectAction(itemElem.attribute("label").replace("(_","(&").replace("_","&"),0);
             mMapper->setMapping(action,QString("%1/%2").arg(prefix).arg(itemElem.attribute("id")));
             connect(action,SIGNAL(triggered()),mMapper,SLOT(map()));
             action->setCheckable(true);
