--- ./src/ncbi_cxx--12_0_0/src/app/cn3d/cn3d_png.cpp.orig	2007-05-04 19:18:18.000000000 +0200
+++ ./src/ncbi_cxx--12_0_0/src/app/cn3d/cn3d_png.cpp	2014-05-17 14:16:57.896544052 +0200
@@ -56,6 +56,7 @@
 #endif
 
 #include <png.h>
+#include <zlib.h>
 
 #include "remove_header_conflicts.hpp"
 
@@ -668,7 +669,7 @@
         if (!png_ptr) throw "can't create PNG write structure";
         info_ptr = png_create_info_struct(png_ptr);
         if (!info_ptr) throw "can't create PNG info structure";
-        if (setjmp(png_ptr->jmpbuf)) throw "setjmp failed";
+        if (setjmp(png_jmpbuf(png_ptr))) throw "setjmp failed";
         png_init_io(png_ptr, out);
 
         // sets callback that's called by PNG after each written row
