--- ./src/ncbi_cxx--9_0_0/scripts/projects/cn3d.lst.orig	2012-12-17 19:14:02.154549102 +0100
+++ ./src/ncbi_cxx--9_0_0/scripts/projects/cn3d.lst	2012-12-17 19:13:42.567645690 +0100
@@ -35,4 +35,15 @@
 algo/structure/struct_dp$
 algo/structure/struct_util$
 algo/structure/threader$
+objects/seqsplit$
+objects/entrez2$
+objects/submit$
+objects/seqedit$
+objects/genomecoll$
+algo/structure/wx_tools$
+algo/dustmask$
+algo/winmask$
+objmgr$
+objmgr/util$
+util/xregexp$
 app/cn3d
