diff -ru orig/src/main.C new/src/main.C
--- orig/src/main.C	2009-05-30 16:48:11.000000000 +0800
+++ new/src/main.C	2010-01-26 16:10:14.000000000 +0800
@@ -514,7 +514,7 @@
   unsigned int w, h;
   unsigned int max_width, max_height;
 
-  szHint.flags = PMinSize | PResizeInc | PBaseSize | PWinGravity;
+  szHint.flags = PMinSize | PWinGravity;
   szHint.win_gravity = NorthWestGravity;
   /* szHint.min_aspect.x = szHint.min_aspect.y = 1; */
 
@@ -629,6 +629,8 @@
 
   ncol = width  / fwidth;
   nrow = height / fheight;
+  width = ncol * fwidth;
+  height = nrow * fheight;
 }
 
 /*----------------------------------------------------------------------*/
@@ -916,14 +918,6 @@
 
   bool set_hint = !HOOK_INVOKE ((this, HOOK_RESIZE_ALL_WINDOWS, DT_INT, newwidth, DT_INT, newheight, DT_END));
 
-  // to avoid races between us and the wm, we clear the incremental size hints around the xresizewindow
-  if (set_hint)
-    {
-      szHint.flags &= ~(PBaseSize | PResizeInc);
-      XSetWMNormalHints (dpy, parent[0], &szHint);
-      szHint.flags |= PBaseSize | PResizeInc;
-    }
-
   if (!ignoreparent)
     {
 #ifdef SMART_RESIZE
