--- Breathless/buildset.orig	2007-06-17 06:21:03.000000000 +0200
+++ Breathless/buildset	2007-06-20 08:27:10.000000000 +0200
@@ -97,7 +97,7 @@
 echo -ne "\n\t(You can take a look at the apps folder to choose an icon or type kmenu for default kmenu icon) "
 echo -ne "\n\tkmenu (case sensitive - dont put (.png) at the end of your selection): "
 
-read KMENU_ICON
+read KMENU_ICON < kmenuicon
 echo -ne "\nUsing the $KMENU_ICON.png icon as your kmenu icon.\n"
 
 # Add your distributions kmenu specific icon name below. 
@@ -111,7 +111,7 @@
 		
 #	Mandrake Specific Kmenu Icon to create
 
-		cp -f 128x128/apps/$KMENU_ICON.png 128x128/apps/menuk-mdk.png
+#		cp -f 128x128/apps/$KMENU_ICON.png 128x128/apps/menuk-mdk.png
 
 #	Distribution Specific Kmenu Icon to create
 
