diff -ur conky-1.8.1/src/conky.c conky-1.8.1-patched//src/conky.c
--- conky-1.8.1/src/conky.c	2010-10-05 17:29:36.000000000 -0400
+++ conky-1.8.1-patched//src/conky.c	2011-05-06 13:18:17.165605291 -0400
@@ -2389,7 +2389,7 @@
 					|| obj->type == OBJ_lua || obj->type == OBJ_lua_parse
 #endif /* HAVE_LUA */
 					) {
-				substitute_newlines(p, a - 2);
+				//substitute_newlines(p, a - 2);
 			}
 			p += a;
 			p_max_size -= a;
