diff -rupN ies4linux-2.99.0.1.orig//lib/functions.sh ies4linux-2.99.0.1/lib/functions.sh
--- ies4linux-2.99.0.1.orig//lib/functions.sh	2011-02-17 15:54:06.403629608 -0500
+++ ies4linux-2.99.0.1/lib/functions.sh	2011-02-17 15:54:46.903629608 -0500
@@ -33,7 +33,7 @@ function init_variables {
 # Find where wine is
 function find_wine {
 	which wine &> /dev/null || error $MSG_ERROR_INSTALL_WINE
-	wine --version 2>&1  | grep -q "0.9." || warning $MSG_WARNING_OLDWINE
+	wine --version 2>&1  | grep -q "0.9.\|\-1." || warning $MSG_WARNING_OLDWINE
 }
 
 # check for cabextract
