*** filesystem_traits.hpp	2011-02-05 21:47:56.386666668 +0100
--- filesystem_traits.hpp	2011-02-05 21:48:11.026666669 +0100
***************
*** 1155,1161 ****
  
      static file_handle_type invalid_file_handle_value()
      {
!         return NULL;
      }
  
      static file_handle_type open_file(char_type const* fileName, int oflag, int pmode)
--- 1155,1161 ----
  
      static file_handle_type invalid_file_handle_value()
      {
!         return 0;
      }
  
      static file_handle_type open_file(char_type const* fileName, int oflag, int pmode)
