diff -r 936db3ed3d8b liteidex/src/plugins/golangdoc/golangdoc.pro
--- a/liteidex/src/plugins/golangdoc/golangdoc.pro	Thu Nov 15 16:01:52 2012 +0800
+++ b/liteidex/src/plugins/golangdoc/golangdoc.pro	Mon Nov 26 13:36:58 2012 -0600
@@ -11,9 +11,9 @@
 include(../../utils/htmlutil/htmlutil.pri)
 include(../../utils/golangapi/golangapi.pri)
 include(../../3rdparty/qtc_editutil/qtc_editutil.pri)
-include(../../3rdparty/qjson/qjson.pri)
 
 DEFINES += GOLANGDOC_LIBRARY
+LIBS += -lqjson
 
 SOURCES += golangdocplugin.cpp \
     golangdoc.cpp \
