--- ./buildozer/targets/android.py	2014-06-11 01:00:11.255771595 -0700
+++ ./buildozer/targets/android.py	2014-06-11 01:00:27.875771155 -0700
@@ -126,7 +126,7 @@
         self.buildozer.environ['PATH'] = ':'.join(path)
         checkbin = self.buildozer.checkbin
         checkbin('Git (git)', 'git')
-        checkbin('Cython (cython)', 'cython')
+        checkbin('Cython2 (cython2)', 'cython2')
         checkbin('Java compiler (javac)', self.javac_cmd)
         checkbin('Java keytool (keytool)', self.keytool_cmd)
 
