--- src/pwsafe-0.95.1BETA/src/os/linux/rand.cpp.original	2015-05-27 12:07:00.403011774 +0200
+++ src/pwsafe-0.95.1BETA/src/os/linux/rand.cpp	2015-05-27 12:07:20.307959892 +0200
@@ -35,7 +35,7 @@
   ifstream is("/dev/urandom");
   if (!is)
     return false;
-  return is.read(static_cast<char *>(p), len);
+  return !is.read(static_cast<char *>(p), len).fail();
 }
 
 static void get_failsafe_rnd(char * &p, unsigned &slen)
