--- a/tunctl/tunctl.c	2007-08-15 17:10:03.000000000 +0200
+++ b/tunctl/tunctl.c	2009-06-13 12:26:33.000000000 +0200
@@ -38,7 +38,7 @@
   struct group *gr;
   uid_t owner = -1;
   gid_t group = -1;
-  int tap_fd, opt, delete = 0, brief = 0;
+  int tap_fd, opt, delete = 0, brief = 0, tun_mode = 0;
   char *tun = "", *file = "/dev/net/tun", *name = argv[0], *end;
 
   while((opt = getopt(argc, argv, "bd:f:t:u:g:")) > 0){
@@ -82,6 +82,7 @@
 
       case 't':
         tun = optarg;
+	if (!strncmp(tun,"tun",3)) tun_mode = 1;
         break;
       case 'h':
       default:
@@ -103,7 +104,7 @@
 
   memset(&ifr, 0, sizeof(ifr));
 
-  ifr.ifr_flags = IFF_TAP | IFF_NO_PI;
+  ifr.ifr_flags = (tun_mode ? IFF_TUN : IFF_TAP) | IFF_NO_PI;
   strncpy(ifr.ifr_name, tun, sizeof(ifr.ifr_name) - 1);
   if(ioctl(tap_fd, TUNSETIFF, (void *) &ifr) < 0){
     perror("TUNSETIFF");
