diff --git a/src/bitmessageqt/__init__.py b/src/bitmessageqt/__init__.py
index ae5ab8a..2dbec36 100644
--- a/src/bitmessageqt/__init__.py
+++ b/src/bitmessageqt/__init__.py
@@ -2955,7 +2955,7 @@ def run():
     translator = QtCore.QTranslator()
 
     try:
-        translator.load("translations/bitmessage_" + str(locale.getlocale()[0]))
+        translator.load("/usr/share/pybitmessage/translations/bitmessage_" + str(locale.getlocale()[0]))
     except:
         # The above is not compatible with all versions of OSX.
         translator.load("translations/bitmessage_en_US") # Default to english.
