--- OMakefile.orig	2008-03-30 17:12:39.000000000 +0200
+++ OMakefile	2008-07-08 13:02:05.000000000 +0200
@@ -2,12 +2,12 @@
 # Include path for the caml headers <caml/mlvalues.h> and friends.  
 # You might need something this if you have installed OCaml manually or via
 # something like macports:
-#INCLUDES += /opt/local/lib/ocaml
+INCLUDES += /usr/lib/ocaml
 
 # Uncomment one of the following if you want support for extended attributes,
 # depending on your OS:
 #CFLAGS = -DHAVE_OSX_XATTR
-#CFLAGS = -DHAVE_LINUX_XATTR
+CFLAGS = -DHAVE_LINUX_XATTR
 
 # This might be needed if you are using -DHAVE_LINUX_XATTR and are using an
 # older version of glibc, to link against libattr1.
