--- tengine/auto/install~	2011-12-08 01:13:12.437692298 +0800
+++ tengine/auto/install	2011-12-08 01:21:30.015094232 +0800
@@ -124,7 +124,7 @@
 
 	test -f '\$(DESTDIR)$NGX_CONF_PATH' \
 		|| cp conf/nginx.conf '\$(DESTDIR)$NGX_CONF_PATH'
-	cp conf/nginx.conf '\$(DESTDIR)$NGX_CONF_PREFIX/nginx.conf.default'
+	cp conf/nginx.conf '\$(DESTDIR)$NGX_CONF_PATH.default'
 
 	test -d '\$(DESTDIR)`dirname "$NGX_PID_PATH"`' \
 		|| mkdir -p '\$(DESTDIR)`dirname "$NGX_PID_PATH"`'

--- tengine/src/mail/ngx_mail_ssl_module.h~	2011-12-08 00:43:40.170279402 +0800
+++ tengine/src/mail/ngx_mail_ssl_module.h	2011-12-08 00:43:40.183612666 +0800
@@ -33,6 +33,7 @@
 
     ngx_str_t        certificate;
     ngx_str_t        certificate_key;
+    ngx_str_t        pass_phrase_dialog;
     ngx_str_t        dhparam;
     ngx_str_t        ecdh_curve;
 
--- tengine/src/mail/ngx_mail_ssl_module.c~	2011-12-01 21:47:40.000000000 +0800
+++ tengine/src/mail/ngx_mail_ssl_module.c	2011-12-10 01:10:35.254359912 +0800
@@ -41,6 +41,9 @@
 };
 
 
+static ngx_str_t ngx_mail_ssl_unknown_server_name = ngx_string("unknown");
+
+
 static ngx_command_t  ngx_mail_ssl_commands[] = {
 
     { ngx_string("ssl"),
@@ -50,6 +53,13 @@
       offsetof(ngx_mail_ssl_conf_t, enable),
       NULL },
 
+    { ngx_string("ssl_pass_phrase_dialog"),
+      NGX_MAIL_MAIN_CONF|NGX_MAIL_SRV_CONF|NGX_CONF_TAKE1,
+      ngx_conf_set_str_slot,
+      NGX_MAIL_SRV_CONF_OFFSET,
+      offsetof(ngx_mail_ssl_conf_t, pass_phrase_dialog),
+      NULL },
+
     { ngx_string("starttls"),
       NGX_MAIL_MAIN_CONF|NGX_MAIL_SRV_CONF|NGX_CONF_TAKE1,
       ngx_mail_ssl_starttls,
@@ -194,6 +204,8 @@
 
     char                *mode;
     ngx_pool_cleanup_t  *cln;
+    ngx_mail_core_srv_conf_t           *cscf;
+    ngx_http_ssl_pphrase_dialog_conf_t  dialog;
 
     ngx_conf_merge_value(conf->enable, prev->enable, 0);
     ngx_conf_merge_uint_value(conf->starttls, prev->starttls,
@@ -218,6 +230,8 @@
 
     ngx_conf_merge_str_value(conf->ciphers, prev->ciphers, NGX_DEFAULT_CIPHERS);
 
+    ngx_conf_merge_str_value(conf->pass_phrase_dialog, prev->pass_phrase_dialog,
+                         "builtin");
 
     conf->ssl.log = cf->log;
 
@@ -276,8 +290,17 @@
     cln->handler = ngx_ssl_cleanup_ctx;
     cln->data = &conf->ssl;
 
+    cscf = ngx_mail_conf_get_module_srv_conf(cf, ngx_mail_core_module);
+    dialog.ssl = &conf->ssl;
+    dialog.type = &conf->pass_phrase_dialog;
+    if (cscf->server_name.len != 0) {
+        dialog.server_name = &cscf->server_name;
+    } else {
+        dialog.server_name = &ngx_mail_ssl_unknown_server_name;
+    }
+
     if (ngx_ssl_certificate(cf, &conf->ssl, &conf->certificate,
-                            &conf->certificate_key)
+                            &conf->certificate_key, &dialog)
         != NGX_OK)
     {
         return NGX_CONF_ERROR;
