diff -uNr rlocate-0.5.5/src/Makefile rlocate-0.5.5-new/src/Makefile
--- rlocate-0.5.5/src/Makefile	2007-01-12 22:04:54.000000000 +0000
+++ rlocate-0.5.5-new/src/Makefile	2007-01-12 22:07:23.000000000 +0000
@@ -203,7 +203,7 @@
 		  rlocate.h rlocate.c cmds.c cmds.h conf.c conf.h utils.c \
 	   	  utils.h
 
-SUBDIRS = rlocate-daemon rlocate-module rlocate-scripts
+SUBDIRS = rlocate-daemon rlocate-scripts
 EXTRA_DIST = rlocate.cron rlocate-scripts install-cron.sh.in
 AM_CFLAGS = 
 all: all-recursive
@@ -570,8 +570,6 @@
 
 install-exec-hook:
 	$(INSTALL) -d $(RLOCATE_DB_DIR)
-	chown $(root_uid):$(rlocate_group) $(DESTDIR)$(bindir)/$(rlocate)
-	chown $(root_uid):$(rlocate_group) $(RLOCATE_DB_DIR)
 	chmod 2755 $(DESTDIR)$(bindir)/$(rlocate)
 	chmod 0750 $(RLOCATE_DB_DIR)
 	-$(LN_S) -f $(rlocate) $(DESTDIR)$(bindir)/$(updatedb)
diff -uNr rlocate-0.5.5/src/rlocate-scripts/Makefile rlocate-0.5.5-new/src/rlocate-scripts/Makefile
--- rlocate-0.5.5/src/rlocate-scripts/Makefile	2007-01-12 22:04:55.000000000 +0000
+++ rlocate-0.5.5-new/src/rlocate-scripts/Makefile	2007-01-12 22:05:57.000000000 +0000
@@ -352,8 +352,6 @@
 	rm $(srcdir)/rlocate-checkpoint.temp
 
 install-exec-hook:
-	-/usr/sbin/groupadd $(rlocate_group)
-	chown $(root_uid):$(rlocate_group) $(DESTDIR)$(bindir)/rlocate-checkpoint
 	chmod 2755 $(DESTDIR)$(bindir)/rlocate-checkpoint
 
 uninstall-hook:
