--- indicator-session-12.10.4/src/session-menu-mgr.c	2014-06-12 05:52:33.221138172 +0900
+++ indicator-session-12.10.4/src/session-menu-mgr.c	2014-06-12 20:51:24.872881685 +0900
@@ -48,9 +48,9 @@
  #define CMD_LOGOUT   LIBEXECDIR"/gtk-logout-helper --logout"
  #define CMD_SHUTDOWN LIBEXECDIR"/gtk-logout-helper --shutdown"
 #else
- #define CMD_RESTART  "gnome-session-quit --reboot"
- #define CMD_LOGOUT   "gnome-session-quit --logout"
- #define CMD_SHUTDOWN "gnome-session-quit --power-off"
+ #define CMD_RESTART  "systemctl-session-manager reboot"
+ #define CMD_LOGOUT   "systemctl-session-manager exit"
+ #define CMD_SHUTDOWN "systemctl-session-manager poweroff"
 #endif
 
 /**
