diff -Naur a/metadata/expo.xml.in b/metadata/expo.xml.in
--- a/metadata/expo.xml.in	2011-04-21 20:20:09.000000000 +1000
+++ b/metadata/expo.xml.in	2013-11-29 18:37:27.101274546 +1100
@@ -75,7 +75,7 @@
 				<option name="zoom_time" type="float">
 					<_short>Zoom time</_short>
 					<_long>Duration of the zoomout animation</_long>
-					<default>0.5</default>
+					<default>0.3</default>
 					<min>0.1</min>
 					<max>5.0</max>
 					<precision>0.1</precision>
@@ -207,7 +207,7 @@
 					<option name="reflection" type="bool">
 						<_short>Reflection</_short>
 						<_long>Show a reflection of the viewports on the ground</_long>
-						<default>true</default>
+						<default>false</default>
 					</option>
 					<option name="ground_color1" type="color">
 						<_short>Ground color(near)</_short>
