--- biblioteq.pro.orig	2012-06-07 15:09:56.000000000 -0400
+++ biblioteq.pro	2012-06-07 15:10:52.000000000 -0400
@@ -88,9 +88,9 @@
 
 biblioteq.path		= /usr/local/biblioteq
 biblioteq.files		= BiblioteQ
-conf.path		= /usr/local/share
+conf.path		= /usr/local/biblioteq
 conf.files		= biblioteq.conf
-icons.path		= /usr/local/biblioteq/icons.d
+icons.path		= /usr/local/biblioteq
 icons.files		= icons.d
 postinstall.path	= /usr/local/biblioteq
 postinstall.extra	= find /usr/local/biblioteq -name .svn -exec rm -rf {} \\; 2>/dev/null; echo
