--- mosaic-ck/src/readPNG.h	2012-03-25 12:31:22.094307023 +0200
+++ mosaic-ck/src/readPNG.h	2012-03-25 12:32:30.055276613 +0200
@@ -54,7 +54,7 @@
 
 /* Author: DXP */
 
-#include "png.h"
+#include "libpng12/png.h"
 
 unsigned char *
 ReadPNG(FILE *infile,int *width, int *height, XColor *colrs);
--- mosaic-ck/makefiles/Makefile.linux	2012-03-25 12:31:21.517632129 +0200
+++ mosaic-ck/makefiles/Makefile.linux	2012-03-25 12:53:16.655081005 +0200
@@ -138,7 +138,7 @@
 pngdir = /usr
 pnglibdir = $(pngdir)/lib
 pngincludedir = $(pngdir)/include
-pnglibs = $(pnglibdir)/libpng.a $(pnglibdir)/libz.a /usr/lib/libm.a
+pnglibs = $(pnglibdir)/libpng12.a $(pnglibdir)/libz.a /usr/lib/libm.a
 pngflags =  -I$(pngincludedir) -DHAVE_PNG
 
 #### JPEG SUPPORT
