--- kde-workspace-4.9.3/kdm/kfrontend/CMakeLists.txt	2012-08-13 15:53:06.000000000 +0700
+++ kde-workspace-4.9.3m/kdm/kfrontend/CMakeLists.txt	2012-12-03 16:57:46.159761651 +0700
@@ -7,6 +7,7 @@
 include_directories(
 	${CMAKE_CURRENT_SOURCE_DIR}/../kcm/background
 	${KDEBASE_WORKSPACE_SOURCE_DIR}/libs/kdm
+	${KDEBASE_WORKSPACE_SOURCE_DIR}/kcontrol/keyboard
 	${QIMAGEBLITZ_INCLUDES}
 )
 
@@ -30,6 +31,10 @@
 	themer/parse.cpp
 	themer/parse.h
 )
+set(x11helper_SRCS
+   ${KDEBASE_WORKSPACE_SOURCE_DIR}/kcontrol/keyboard/x11_helper.cpp
+   ${KDEBASE_WORKSPACE_SOURCE_DIR}/kcontrol/keyboard/x11_helper.h
+)
 set(kdm_greet_SRCS
 	kdm_greet.c
 	kdm_greet.h
@@ -52,6 +57,7 @@
 	utils.cpp
 	utils.h
 	${kdmthemer_SRCS}
+	${x11helper_SRCS}
 )
 
 if (XDMCP)
