diff -aur krb5-auth-dialog-0.17.old/src/ka-applet.c krb5-auth-dialog-0.17.new/src/ka-applet.c
--- krb5-auth-dialog-0.17.old/src/ka-applet.c	2011-03-17 11:41:35.643931317 +0100
+++ krb5-auth-dialog-0.17.new/src/ka-applet.c	2011-03-17 11:42:15.197660184 +0100
@@ -489,10 +489,9 @@
 	notify_icon = icon ? icon : "krb-valid-ticket";
 
 	applet->priv->notification = \
-		notify_notification_new_with_status_icon(summary,
-							 message,
-							 notify_icon,
-							 applet->priv->tray_icon);
+		notify_notification_new(summary,
+					 message,
+					 notify_icon);
 
 	notify_notification_set_urgency (applet->priv->notification, NOTIFY_URGENCY_NORMAL);
 	notify_notification_add_action (applet->priv->notification,
