diff -Naur xfce4-places-plugin-1.2.0.orig/configure xfce4-places-plugin-1.2.0/configure
--- xfce4-places-plugin-1.2.0.orig/configure	2009-07-31 08:21:08.000000000 +0200
+++ xfce4-places-plugin-1.2.0/configure	2011-04-29 22:42:47.113361266 +0200
@@ -21268,62 +21268,6 @@
   fi
 
 
-  { echo "$as_me:$LINENO: checking for thunar-vfs-1 >= 0.3.2" >&5
-echo $ECHO_N "checking for thunar-vfs-1 >= 0.3.2... $ECHO_C" >&6; }
-  if $PKG_CONFIG "--atleast-version=0.3.2" "thunar-vfs-1" >/dev/null 2>&1; then
-    THUNAR_VFS_VERSION=`$PKG_CONFIG --modversion "thunar-vfs-1"`
-    { echo "$as_me:$LINENO: result: $THUNAR_VFS_VERSION" >&5
-echo "${ECHO_T}$THUNAR_VFS_VERSION" >&6; }
-
-    { echo "$as_me:$LINENO: checking THUNAR_VFS_CFLAGS" >&5
-echo $ECHO_N "checking THUNAR_VFS_CFLAGS... $ECHO_C" >&6; }
-    THUNAR_VFS_CFLAGS=`$PKG_CONFIG --cflags "thunar-vfs-1"`
-    { echo "$as_me:$LINENO: result: $THUNAR_VFS_CFLAGS" >&5
-echo "${ECHO_T}$THUNAR_VFS_CFLAGS" >&6; }
-
-    { echo "$as_me:$LINENO: checking THUNAR_VFS_LIBS" >&5
-echo $ECHO_N "checking THUNAR_VFS_LIBS... $ECHO_C" >&6; }
-    THUNAR_VFS_LIBS=`$PKG_CONFIG --libs "thunar-vfs-1"`
-    { echo "$as_me:$LINENO: result: $THUNAR_VFS_LIBS" >&5
-echo "${ECHO_T}$THUNAR_VFS_LIBS" >&6; }
-
-    THUNAR_VFS_REQUIRED_VERSION=0.3.2
-
-
-
-
-
-
-
-  elif $PKG_CONFIG --exists "thunar-vfs-1" >/dev/null 2>&1; then
-    xdt_cv_version=`$PKG_CONFIG --modversion "thunar-vfs-1"`
-    { echo "$as_me:$LINENO: result: found, but $xdt_cv_version" >&5
-echo "${ECHO_T}found, but $xdt_cv_version" >&6; }
-
-
-      echo "*** The required package thunar-vfs-1 was found on your system,"
-      echo "*** but the installed version ($xdt_cv_version) is too old."
-      echo "*** Please upgrade thunar-vfs-1 to atleast version 0.3.2, or adjust"
-      echo "*** the PKG_CONFIG_PATH environment variable if you installed"
-      echo "*** the new version of the package in a nonstandard prefix so"
-      echo "*** pkg-config is able to find it."
-      exit 1
-
-  else
-    { echo "$as_me:$LINENO: result: not found" >&5
-echo "${ECHO_T}not found" >&6; }
-
-
-      echo "*** The required package thunar-vfs-1 was not found on your system."
-      echo "*** Please install thunar-vfs-1 (atleast version 0.3.2) or adjust"
-      echo "*** the PKG_CONFIG_PATH environment variable if you"
-      echo "*** installed the package in a nonstandard prefix so that"
-      echo "*** pkg-config is able to find it."
-      exit 1
-
-  fi
-
-
 
   # minimum supported version of pkg-config
   xdt_cv_PKG_CONFIG_MIN_VERSION=0.9.0
@@ -21406,26 +21350,26 @@
   fi
 
 
-  { echo "$as_me:$LINENO: checking for exo-0.3 >= 0.3.1.1" >&5
-echo $ECHO_N "checking for exo-0.3 >= 0.3.1.1... $ECHO_C" >&6; }
-  if $PKG_CONFIG "--atleast-version=0.3.1.1" "exo-0.3" >/dev/null 2>&1; then
-    EXO_VERSION=`$PKG_CONFIG --modversion "exo-0.3"`
+  { echo "$as_me:$LINENO: checking for exo-1 >= 0.5.0" >&5
+echo $ECHO_N "checking for exo-1 >= 0.5.0... $ECHO_C" >&6; }
+  if $PKG_CONFIG "--atleast-version=0.5.0" "exo-1" >/dev/null 2>&1; then
+    EXO_VERSION=`$PKG_CONFIG --modversion "exo-1"`
     { echo "$as_me:$LINENO: result: $EXO_VERSION" >&5
 echo "${ECHO_T}$EXO_VERSION" >&6; }
 
     { echo "$as_me:$LINENO: checking EXO_CFLAGS" >&5
 echo $ECHO_N "checking EXO_CFLAGS... $ECHO_C" >&6; }
-    EXO_CFLAGS=`$PKG_CONFIG --cflags "exo-0.3"`
+    EXO_CFLAGS=`$PKG_CONFIG --cflags "exo-1"`
     { echo "$as_me:$LINENO: result: $EXO_CFLAGS" >&5
 echo "${ECHO_T}$EXO_CFLAGS" >&6; }
 
     { echo "$as_me:$LINENO: checking EXO_LIBS" >&5
 echo $ECHO_N "checking EXO_LIBS... $ECHO_C" >&6; }
-    EXO_LIBS=`$PKG_CONFIG --libs "exo-0.3"`
+    EXO_LIBS=`$PKG_CONFIG --libs "exo-1"`
     { echo "$as_me:$LINENO: result: $EXO_LIBS" >&5
 echo "${ECHO_T}$EXO_LIBS" >&6; }
 
-    EXO_REQUIRED_VERSION=0.3.1.1
+    EXO_REQUIRED_VERSION=0.5.0
 
 
 
@@ -21433,15 +21377,15 @@
 
 
 
-  elif $PKG_CONFIG --exists "exo-0.3" >/dev/null 2>&1; then
-    xdt_cv_version=`$PKG_CONFIG --modversion "exo-0.3"`
+  elif $PKG_CONFIG --exists "exo-1" >/dev/null 2>&1; then
+    xdt_cv_version=`$PKG_CONFIG --modversion "exo-1"`
     { echo "$as_me:$LINENO: result: found, but $xdt_cv_version" >&5
 echo "${ECHO_T}found, but $xdt_cv_version" >&6; }
 
 
-      echo "*** The required package exo-0.3 was found on your system,"
+      echo "*** The required package exo-1 was found on your system,"
       echo "*** but the installed version ($xdt_cv_version) is too old."
-      echo "*** Please upgrade exo-0.3 to atleast version 0.3.1.1, or adjust"
+      echo "*** Please upgrade exo-1 to atleast version 0.5.0, or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you installed"
       echo "*** the new version of the package in a nonstandard prefix so"
       echo "*** pkg-config is able to find it."
@@ -21452,8 +21396,8 @@
 echo "${ECHO_T}not found" >&6; }
 
 
-      echo "*** The required package exo-0.3 was not found on your system."
-      echo "*** Please install exo-0.3 (atleast version 0.3.1.1) or adjust"
+      echo "*** The required package exo-1 was not found on your system."
+      echo "*** Please install exo-1 (atleast version 0.5.0) or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you"
       echo "*** installed the package in a nonstandard prefix so that"
       echo "*** pkg-config is able to find it."
