diff -Naur xfce4-places-plugin-1.2.0.orig/configure.in xfce4-places-plugin-1.2.0/configure.in
--- xfce4-places-plugin-1.2.0.orig/configure.in	2009-07-31 08:21:02.000000000 +0200
+++ xfce4-places-plugin-1.2.0/configure.in	2011-03-24 11:30:01.410713907 +0100
@@ -25,7 +25,7 @@
 XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.3.90.2])
 XDT_CHECK_PACKAGE([LIBXFCE4PANEL], [libxfce4panel-1.0], [4.3.90.2])
 XDT_CHECK_PACKAGE([THUNAR_VFS], [thunar-vfs-1], [0.3.2])
-XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.1.1])
+XDT_CHECK_PACKAGE([EXO], [exo-1], [0.5.0])
 
 dnl check for optional packages/versions
 XDT_CHECK_OPTIONAL_PACKAGE([LIBXFCE4PANEL_46], [libxfce4panel-1.0], [4.5.92], [libxfce4panel46], [Take advantage of Xfce 4.6 panel changes])
diff -Naur xfce4-places-plugin-1.2.0.orig/panel-plugin/button.c xfce4-places-plugin-1.2.0/panel-plugin/button.c
--- xfce4-places-plugin-1.2.0.orig/panel-plugin/button.c	2009-07-31 07:10:48.000000000 +0200
+++ xfce4-places-plugin-1.2.0/panel-plugin/button.c	2011-03-24 11:31:15.310413668 +0100
@@ -50,8 +50,7 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
-#include <libxfce4panel/xfce-hvbox.h>
+#include <libxfce4panel/libxfce4panel.h>
 #include <string.h>
 
 #define EXO_API_SUBJECT_TO_CHANGE
diff -Naur xfce4-places-plugin-1.2.0.orig/panel-plugin/cfg.c xfce4-places-plugin-1.2.0/panel-plugin/cfg.c
--- xfce4-places-plugin-1.2.0.orig/panel-plugin/cfg.c	2009-05-28 05:05:53.000000000 +0200
+++ xfce4-places-plugin-1.2.0/panel-plugin/cfg.c	2011-03-24 11:31:27.124503667 +0100
@@ -28,8 +28,7 @@
 #include <gtk/gtk.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
-#include <libxfce4panel/xfce-panel-convenience.h>
+#include <libxfce4panel/libxfce4panel.h>
 #include <libxfcegui4/libxfcegui4.h>
 
 #include "cfg.h"
diff -Naur xfce4-places-plugin-1.2.0.orig/panel-plugin/cfg.h xfce4-places-plugin-1.2.0/panel-plugin/cfg.h
--- xfce4-places-plugin-1.2.0.orig/panel-plugin/cfg.h	2008-01-09 02:20:42.000000000 +0100
+++ xfce4-places-plugin-1.2.0/panel-plugin/cfg.h	2011-03-24 11:31:39.458333668 +0100
@@ -23,7 +23,7 @@
 #define _XFCE_PANEL_PLACES_CFG_H
 
 #include <glib.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 #include "view.h"
 
 typedef struct
diff -Naur xfce4-places-plugin-1.2.0.orig/panel-plugin/places.c xfce4-places-plugin-1.2.0/panel-plugin/places.c
--- xfce4-places-plugin-1.2.0.orig/panel-plugin/places.c	2008-01-09 02:20:42.000000000 +0100
+++ xfce4-places-plugin-1.2.0/panel-plugin/places.c	2011-03-24 11:32:07.524293668 +0100
@@ -26,7 +26,7 @@
 #include <glib.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #include "places.h"
 #include "view.h"
diff -Naur xfce4-places-plugin-1.2.0.orig/panel-plugin/support.c xfce4-places-plugin-1.2.0/panel-plugin/support.c
--- xfce4-places-plugin-1.2.0.orig/panel-plugin/support.c	2009-05-28 04:46:14.000000000 +0200
+++ xfce4-places-plugin-1.2.0/panel-plugin/support.c	2011-03-24 11:33:48.803628667 +0100
@@ -52,10 +52,9 @@
 
     if(path != NULL && *path != '\0'){
 
-        DBG("exo_url_show(%s)", path);
-        exo_success = exo_url_show(path, NULL, NULL);
+    GError         *error = NULL;
 
-        if(!exo_success){
+        if ( !gtk_show_uri (NULL, path, 0, &error)){
             gchar *cmd = g_strconcat("thunar \"", path, "\"", NULL);
             DBG("exec: %s", cmd);
             places_gui_exec(cmd);
@@ -114,8 +113,9 @@
 void
 places_load_file(const gchar *path)
 {
+    GError         *error = NULL;
     if(path != NULL && *path != '\0')
-        exo_url_show(path, NULL, NULL);
+        gtk_show_uri (NULL , path, 0, &error);
 }
 
 /**
diff -Naur xfce4-places-plugin-1.2.0.orig/panel-plugin/view.c xfce4-places-plugin-1.2.0/panel-plugin/view.c
--- xfce4-places-plugin-1.2.0.orig/panel-plugin/view.c	2009-07-31 07:10:48.000000000 +0200
+++ xfce4-places-plugin-1.2.0/panel-plugin/view.c	2011-03-24 11:34:16.969538668 +0100
@@ -49,9 +49,7 @@
 #define USE_GTK_TOOLTIP_API     GTK_CHECK_VERSION(2,12,0)
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
-#include <libxfce4panel/xfce-panel-convenience.h>
-#include <libxfce4panel/xfce-hvbox.h>
+#include <libxfce4panel/libxfce4panel.h>
 #include <libxfcegui4/libxfcegui4.h>
 
 #define EXO_API_SUBJECT_TO_CHANGE
diff -Naur xfce4-places-plugin-1.2.0.orig/panel-plugin/view.h xfce4-places-plugin-1.2.0/panel-plugin/view.h
--- xfce4-places-plugin-1.2.0.orig/panel-plugin/view.h	2008-01-09 02:20:42.000000000 +0100
+++ xfce4-places-plugin-1.2.0/panel-plugin/view.h	2011-03-24 11:34:28.090642002 +0100
@@ -25,7 +25,7 @@
 
 #include <glib.h>
 #include <gtk/gtk.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #define USE_RECENT_DOCUMENTS    GTK_CHECK_VERSION(2,10,0)
 
