diff -Naur xfce4-places-plugin-1.3.0.orig/panel-plugin/support.c xfce4-places-plugin-1.3.0/panel-plugin/support.c
--- xfce4-places-plugin-1.3.0.orig/panel-plugin/support.c	2012-04-02 21:36:16.000000000 +0200
+++ xfce4-places-plugin-1.3.0/panel-plugin/support.c	2012-04-21 20:27:07.996631461 +0200
@@ -48,11 +48,10 @@
 void
 places_load_file_browser(const gchar *path)
 {
-    GError *error = NULL;
-
     if(path != NULL && *path != '\0'){
 
-        if(!gtk_show_uri(NULL, path, 0, &error)){
+        if(!exo_execute_preferred_application("FileManager", path,
+                                              NULL, NULL, NULL)) { 
             gchar *cmd = g_strconcat("thunar \"", path, "\"", NULL);
             DBG("exec: %s", cmd);
             places_gui_exec(cmd);
