diff --git a/panel-plugin/support.c b/panel-plugin/support.c
--- a/panel-plugin/support.c
+++ b/panel-plugin/support.c
@@ -48,11 +48,10 @@
 void
 places_load_file_browser(const gchar *path)
 {
-    GError *error = NULL;
-
     if(path != NULL && *path != '\0'){

-        if(!gtk_show_uri(NULL, path, 0, &error)){
+        if(!exo_execute_preferred_application("FileManager", path,
+                                              NULL, NULL, NULL)) {
             gchar *cmd = g_strconcat("thunar \"", path, "\"", NULL);
             DBG("exec: %s", cmd);
             places_gui_exec(cmd);
