From 35d69169916f15772fa117335492b4e34fb6c140 Mon Sep 17 00:00:00 2001
From: Andrzej <ndrwrdck@gmail.com>
Date: Mon, 07 May 2012 16:03:14 +0000
Subject: Remove some compilation and runtime warnings

---
diff --git a/panel-plugin/button.c b/panel-plugin/button.c
index bf0e3e9..4e3e157 100644
--- a/panel-plugin/button.c
+++ b/panel-plugin/button.c
@@ -443,8 +443,9 @@ places_button_resize(PlacesButton *self)
                  - MAX(button_width, button_height);
     if (show_label) {
         xfce_panel_plugin_set_small(self->plugin, FALSE);
-        gtk_label_set_angle (GTK_LABEL (self->label),
-            (xfce_panel_plugin_get_mode(self->plugin) == GTK_ORIENTATION_VERTICAL) ? -90 : 0);
+        if (self->label != NULL)
+          gtk_label_set_angle (GTK_LABEL (self->label),
+                               (xfce_panel_plugin_get_mode(self->plugin) == XFCE_PANEL_PLUGIN_MODE_VERTICAL) ? -90 : 0);
     } else {
         xfce_panel_plugin_set_small(self->plugin, TRUE);
         new_size /= xfce_panel_plugin_get_nrows(self->plugin);
diff --git a/panel-plugin/cfg.c b/panel-plugin/cfg.c
index 482074f..378bfa9 100644
--- a/panel-plugin/cfg.c
+++ b/panel-plugin/cfg.c
@@ -506,7 +506,7 @@ places_cfg_open_dialog(PlacesCfg *cfg)
     gtk_widget_show(tmp_label);
     gtk_box_pack_start(GTK_BOX(tmp_box), tmp_label, FALSE, FALSE, 0);

-    adj = gtk_adjustment_new(cfg->show_recent_number, 1, 25, 1, 5, 5);
+    adj = gtk_adjustment_new(cfg->show_recent_number, 1, 25, 1, 5, 0);

     tmp_widget = gtk_spin_button_new(GTK_ADJUSTMENT(adj), 1, 0);
     gtk_label_set_mnemonic_widget(GTK_LABEL(tmp_label), tmp_widget);
diff --git a/panel-plugin/model_system.c b/panel-plugin/model_system.c
index aed425f..f0186bd 100644
--- a/panel-plugin/model_system.c
+++ b/panel-plugin/model_system.c
@@ -32,6 +32,7 @@
 #include <glib.h>
 #include <gio/gio.h>

+#include <exo/exo.h>
 #include <libxfce4util/libxfce4util.h>

 #define TRASH          1
