--- vncinstall	2013-03-04 13:51:13.000000000 +0100
+++ vncinstall.arch	2013-06-07 16:39:38.575319421 +0200
@@ -8,71 +8,13 @@
 # vncserver configuration files and a default xstartup.
 #
 
-# Set up defaults and platform-specific overrides
-dst=/usr/local/bin
-lib=/usr/local/lib
-
-SYSNAME=`uname -s`
-case "$SYSNAME" in
-    Linux)
-        dst=/usr/bin
-        lib=/usr/lib
-        ;;
-    SunOS)
-        ;;
-    HP-UX)
-        ;;
-    AIX)
-        ;;
-    *)
-        echo "Unknown platform"
-        exit 1
-        ;;
-esac
-
-if [ $# -gt 4 ]; then 
-    echo "usage: $0 [<installation-directory> [<man-page-directory> [<cups-directory] ] ]"
-    exit 1
-fi
-
-# Installation directory
-if [ $# -gt 0 ]; then
-    dst=$1
-    shift
-fi
+pkgdir="$1"
 
-# Check installation directory exists
-if [ ! -d $dst ]; then
-  echo "Installation directory $dst does not exist."
-  exit 1
-fi
-
-# Man page directories
-if [ $# -gt 0 ]; then
-    mandst="$1/man1"
-    shift
-else
-    if [ "`basename $dst`" = bin ]; then
-        mandst="`dirname $dst`/man/man1"
-        if [ ! -d "$mandst" -a "$dst" = /usr/bin ]; then
-            mandst=/usr/share/man/man1
-        fi
-    fi
-fi
-if [ "$mandst" != "" ]; then
-    if [ ! -d "$mandst" -o ! -w "$mandst" ]; then
-        echo "Can't install manual pages to $mandst"
-        mandst=""
-    fi
-fi
-
-# CUPS directory
-if [ $# -gt 0 ]; then
-    cupsdst=$1
-    shift
-else
-    cupsdst=/usr/lib/cups
-fi
+# Set up defaults and platform-specific overrides
+dst=$pkgdir/usr/bin
+lib=$pkgdir/usr/lib
+mandst=$pkgdir/usr/share/man/man1
+cupsdst=$pkgdir/usr/lib/cups
 
 # Copy files
 for f in vncviewer vncaddrbook vncserver-x11 vncserver-x11-core Xvnc Xvnc-core \
@@ -139,21 +81,21 @@
 done
 
 # Create fonts directory if it doesn't exist
-if [ ! -d /usr/share/vnc ]; then
-  mkdir -p /usr/share/vnc
-  chmod 0755 /usr/share/vnc
+if [ ! -d $pkgdir/usr/share/vnc ]; then
+  mkdir -p $pkgdir/usr/share/vnc
+  chmod 0755 $pkgdir/usr/share/vnc
 fi
 
-if [ ! -d /usr/share/vnc/fonts ]; then
-  mkdir -p /usr/share/vnc/fonts
-  chmod 0755 /usr/share/vnc/fonts
+if [ ! -d $pkgdir/usr/share/vnc/fonts ]; then
+  mkdir -p $pkgdir/usr/share/vnc/fonts
+  chmod 0755 $pkgdir/usr/share/vnc/fonts
 fi
 
-cp -f fonts/* /usr/share/vnc/fonts
-chmod 0444 /usr/share/vnc/fonts/*
+cp -f fonts/* $pkgdir/usr/share/vnc/fonts
+chmod 0444 $pkgdir/usr/share/vnc/fonts/*
 
-cp -f rgb.txt /usr/share/vnc
-chmod 0444 /usr/share/vnc/rgb.txt
+cp -f rgb.txt $pkgdir/usr/share/vnc
+chmod 0444 $pkgdir/usr/share/vnc/rgb.txt
 
 setuid="Xvnc vncserver-x11"
 for i in $setuid; do
@@ -167,4 +109,4 @@
 chown root $cups_vnc
 chmod 0700 $cups_vnc
 
-$dst/vncinitconfig --install-defaults
+#$dst/vncinitconfig --install-defaults
