--- vncinitconfig	2013-12-06 15:44:59.000000000 +0100
+++ vncinitconfig.arch	2014-01-11 15:24:53.304628934 +0100
@@ -223,16 +223,16 @@
 account required pam_unix.so"
 username=\$USER
 ### dirs and utils:
-etc=/etc
+etc=$1/etc
 etcvnc=$etc/vnc
-bin=/usr/local/bin
-libvnc=/usr/local/lib/vnc
-sharevnc=/usr/share/vnc
+bin=$1/usr/bin
+libvnc=$1/usr/lib/vnc
+sharevnc=$1/usr/share/vnc
 awk=awk
 which=which
 ### CUPS:
-cups_backend_dir=/usr/lib/cups/backend
-cups64_backend_dir=/usr/lib64/cups/backend
+cups_backend_dir=$1/usr/lib/cups/backend
+cups64_backend_dir=$1/usr/lib64/cups/backend
 sun_cups_dir=
 link_cups_lib64=0
 ### moduledirname is used for network installs
@@ -249,63 +249,10 @@
 }
 
 SYSNAME=`uname -s`
-case "$SYSNAME" in
-    Linux)
-        link_cups_lib64=1
-        libvnc=/usr/lib/vnc
-        bin=/usr/bin
-        ;;
-    SunOS)
-        if [ -f /usr/lib/security/pam_unix.so.1 ]; then
-            pamconfig="auth required pam_unix.so.1"
-        elif [ -f /usr/lib/security/pam_unix_auth.so.1 ] &&\
-             [ -f /usr/lib/security/pam_authtok_get.so.1 ]; then
-            pamconfig="auth required pam_authtok_get.so.1
-auth required pam_unix_auth.so.1"
-            if [ -f /usr/lib/security/pam_unix_acct.so.1 ]; then
-                pamconfig="$pamconfig
-account required pam_unix_acct.so.1"
-            elif [ -f /usr/lib/security/pam_unix_account.so.1 ]; then
-                pamconfig="$pamconfig
-account required pam_unix_account.so.1"
-            fi
-        else
-            echo "Can't find required PAM modules"
-            exit 1
-        fi
-        sun_cups_dir=/opt/sfw/cups/lib/cups
-        username=\$LOGNAME
-        # We modify the path in order to make sure we find xset:
-        PATH=$PATH:/usr/openwin/bin
-        awk=nawk
-        which() { ksh whence -p "$@"; }
-        which="ksh whence -p"
-        ;;
-    HP-UX)
-        pamconfig="auth required libpam_unix.1"
-        username=\$LOGNAME
-        # We modify the path in order to make sure we find xset:
-        PATH=$PATH:/usr/bin/X11
-        which() { command -v "$@"; }
-        which="command -v"
-        ;;
-    AIX)
-        if [ "$init_pam" = "1" ]; then
-            echo "Automatic PAM configuration not supported on this platform"
-            init_pam=0
-        fi
-        ;;
-    FreeBSD)
-        etcvnc=/usr/local/etc/vnc
-        libvnc=/usr/local/libexec/vnc
-        sharevnc=/usr/local/share/vnc
-        cups_backend_dir=/usr/local/libexec/cups/backend
-        ;;
-    *)
-        echo "Unknown platform"
-        exit 1
-        ;;
-esac
+link_cups_lib64=1
+libvnc=$1/usr/lib/vnc
+bin=$1/usr/bin
+
 
 
 
