diff -ur a/src/agnxlib.c b/src/agnxlib.c
--- a/src/agnxlib.c	2011-01-08 04:18:29.000000000 +0100
+++ b/src/agnxlib.c	2011-05-30 16:23:07.000000000 +0200
@@ -738,7 +738,7 @@
                   "of the main Agena library folder by adding the following line to your profile\n"
                   "file:\n\n"
 #if defined(__unix__) || defined(__APPLE__) || defined(__HAIKU__)
-                  "   export AGENAPATH=/usr/agena/lib\n\n"
+                  "   export AGENAPATH=/usr/lib/agena\n\n"
                   "in UNIX.\n\n"
 #else
                   "   set AGENAPATH=<drive letter>:/agena/lib\n\n"
@@ -977,10 +977,10 @@
   apath = NULL;
 #if defined(__unix__) || defined(__APPLE__)
   /* set default path if path could not be determined before */
-  if (access("/usr/agena/lib", 00|04) != -1) {
-    lua_pushstring(L, "/usr/agena/lib");
+  if (access("/usr/lib/agena", 00|04) != -1) {
+    lua_pushstring(L, "/usr/lib/agena");
     lua_setglobal(L, "libname");
-    debuginfo(debug, "   libname set to default path /usr/agena/lib.\n");
+    debuginfo(debug, "   libname set to default path /usr/lib/agena.\n");
     return;
   }
 #elif defined(__HAIKU__)
diff -ur a/src/config.c b/src/config.c
--- a/src/config.c	2011-01-01 21:31:52.000000000 +0100
+++ b/src/config.c	2011-05-30 16:21:01.000000000 +0200
@@ -136,7 +136,7 @@
 #if defined(__DJGPP__) || defined(__APPLE__) || defined(__HAIKU__)
   fprintf(fp, "#define ACTUAL_SIZE_OF_C_LONG	%ld\n", sizeof(long));
 #else
-  fprintf(fp, "#define ACTUAL_SIZE_OF_C_LONG	%d\n", sizeof(long));
+  fprintf(fp, "#define ACTUAL_SIZE_OF_C_LONG	%ld\n", sizeof(long));
 #endif
   fprintf(fp, "#ifdef NECESSARY_SIZE_OF_C_LONG\n"
               "#  undef NECESSARY_SIZE_OF_C_LONG\n"
