--- src/agnhlps.h	2010-06-14 05:22:40.000000000 +0200
+++ ../../src_patched_working/agena-1.0.1/src/agnhlps.h	2010-10-07 14:09:02.000000000 +0200
@@ -38,6 +38,10 @@
   #endif
 #endif
 
+#ifndef uint64_t
+	#define uint64_t  unsigned long long
+#endif
+
 /* misc functions */
 
 double tools_intpow (double x, int n);
--- src/config.c	2010-06-28 19:03:58.000000000 +0200
+++ ../../src_patched_working/agena-1.0.1/src/config.c	2010-10-07 14:09:35.000000000 +0200
@@ -136,7 +136,7 @@
 #if defined(__DJGPP__) || defined(__APPLE__) || defined(__HAIKU__)
   fprintf(fp, "#define ACTUAL_SIZE_OF_C_LONG	%ld\n", sizeof(long));
 #else
-  fprintf(fp, "#define ACTUAL_SIZE_OF_C_LONG	%d\n", sizeof(long));
+  fprintf(fp, "#define ACTUAL_SIZE_OF_C_LONG	%ld\n", sizeof(long));
 #endif
   fprintf(fp, "#ifdef NECESSARY_SIZE_OF_C_LONG\n"
               "#  undef NECESSARY_SIZE_OF_C_LONG\n"
--- src/makefile.old	2010-10-13 12:12:17.126666673 +0200
+++ src/makefile	2010-10-13 12:13:08.650000006 +0200
@@ -12,7 +12,7 @@
 PLAT= none
 
 CC= gcc
-CFLAGS= -Wall $(MYCFLAGS)
+CFLAGS= -Wall $(MYCFLAGS) -fPIC
 AR= ar rcu
 RANLIB= ranlib
 RM= rm -f
--- src/agnxlib.c.old	2010-10-13 12:51:09.093333338 +0200
+++ src/agnxlib.c	2010-10-13 12:51:41.600000008 +0200
@@ -719,7 +719,7 @@
                   "of the main Agena library folder by adding the following line to your profile\n"
                   "file:\n\n"
 #if defined(__unix__) || defined(__APPLE__) || defined(__HAIKU__)
-                  "   export AGENAPATH=/usr/agena/lib\n\n"
+                  "   export AGENAPATH=/usr/lib/agena\n\n"
                   "in UNIX.\n\n"
 #else
                   "   set AGENAPATH=<drive letter>:/agena/lib\n\n"
@@ -897,8 +897,8 @@
   }
 #if defined(__unix__) || defined(__APPLE__)
   /* set default path if path could not be determined before */
-  if (access("/usr/agena/lib", 00|04) != -1) {
-    lua_pushstring(L, "/usr/agena/lib");
+  if (access("/usr/lib/agena", 00|04) != -1) {
+    lua_pushstring(L, "/usr/lib/agena");
     lua_setglobal(L, "libname");
     return 0;
   }
