diff -urB enemylines3-1.2/c3_tpl.h enemylinepatched/c3_tpl.h
--- enemylines3-1.2/c3_tpl.h	2006-03-30 18:16:47.000000000 +0200
+++ enemylinepatched/c3_tpl.h	2008-06-27 17:34:32.000000000 +0200
@@ -1,6 +1,7 @@
 #include <iostream>
 #include <sstream>
 #include <utility>
+#include <limits>
 #include <cmath>
 
 
diff -urB enemylines3-1.2/font_ttf.h enemylinepatched/font_ttf.h
--- enemylines3-1.2/font_ttf.h	2006-03-30 18:16:47.000000000 +0200
+++ enemylinepatched/font_ttf.h	2008-06-27 17:34:38.000000000 +0200
@@ -7,7 +7,7 @@
 
 class Font_ttf{
 public:
-	static unsigned int Font_ttf::gen_dl(std::string fontname,unsigned int size,C3 col=C3(255,255,255));
+	static unsigned int gen_dl(std::string fontname,unsigned int size,C3 col=C3(255,255,255));
 	static int dx();
 	static int dy();
 };
diff -urB enemylines3-1.2/map.h enemylinepatched/map.h
--- enemylines3-1.2/map.h	2006-03-30 18:16:47.000000000 +0200
+++ enemylinepatched/map.h	2008-06-27 17:34:49.000000000 +0200
@@ -24,7 +24,7 @@
 	unsigned int dz();
 
 
-	bool Map::inside(C3 c);
+	bool inside(C3 c);
 	
 	e_tiletype get(C3);
 	void set(C3,e_tiletype);
diff -urB enemylines3-1.2/random.cc enemylinepatched/random.cc
--- enemylines3-1.2/random.cc	2006-03-30 18:16:47.000000000 +0200
+++ enemylinepatched/random.cc	2008-06-27 17:35:03.000000000 +0200
@@ -1,5 +1,6 @@
 #include "random.h"
 
+#include <cstdlib>
 #include <iostream>
 
 
