Index: pcre-ocaml/src/pcre-ocaml-release-6.1.1/lib/Makefile
===================================================================
--- pcre-ocaml.orig/src/pcre-ocaml-release-6.1.1/lib/Makefile
+++ pcre-ocaml/src/pcre-ocaml-release-6.1.1/lib/Makefile
@@ -3,7 +3,7 @@ OCAMLMAKEFILE = ../OCamlMakefile
 SOURCES := pcre.mli pcre.ml pcre_stubs.c
 CFLAGS := -O2 $(shell pcre-config --cflags)
 PCRE_FLAGS := $(shell pcre-config --libs)
-OCAMLLDFLAGS := -cclib "$(PCRE_FLAGS)"
+OCAMLLDFLAGS := -cclib -lpcre_stubs -cclib "$(PCRE_FLAGS)"
 OCAMLMKLIB_FLAGS := $(PCRE_FLAGS)
 RESULT := pcre
