diff -aur a/usr/share/read-me/read-me.py b/usr/share/read-me/read-me.py 
--- a/usr/share/read-me/read-me.py
+++ b/usr/share/read-me/read-me.py
@@ -27,8 +27,8 @@
 from gi.repository import Gdk
 from gi.repository import WebKit
 from gi.repository import GObject
-from gi.repository import Unity
-from gi.repository import Dbusmenu
+# from gi.repository import Unity
+# from gi.repository import Dbusmenu
 
 from simplejson import dumps as to_json
 from simplejson import loads as from_json