diff --git a/src/terminal.c b/src/terminal.c
index 06de533..d1fe806 100644
--- a/src/terminal.c
+++ b/src/terminal.c
@@ -495,7 +495,8 @@ main (int argc, char **argv)
   GError *error = NULL;
   char *working_directory;
   int ret = EXIT_SUCCESS;
-
+  GtkSettings *settings;
+  
   setlocale (LC_ALL, "");
 
   bindtextdomain (GETTEXT_PACKAGE, TERM_LOCALEDIR);
@@ -550,6 +551,11 @@ main (int argc, char **argv)
 
   g_set_application_name (_("Terminal"));
   
+  settings = gtk_settings_get_default ();
+  g_object_set (G_OBJECT (settings),
+                "gtk-application-prefer-dark-theme", TRUE,
+                NULL);
+  
   /* Unset the these env variables, so they doesn't end up
    * in the factory's env and thus in the terminals' envs.
    */

