diff -Naur par2cmdline-0.4-tbb-20100203/commandline.cpp par2cmdline-0.4-tbb-20100203.new/commandline.cpp
--- par2cmdline-0.4-tbb-20100203/commandline.cpp	2009-02-03 06:13:53.000000000 +0100
+++ par2cmdline-0.4-tbb-20100203.new/commandline.cpp	2010-02-13 18:49:22.165024376 +0100
@@ -24,6 +24,7 @@
 //  Copyright (c) 2007 Intel Corp.
 
 #include "par2cmdline.h"
+#include <memory>
 
 #if __APPLE__
   #include <sys/types.h>
diff -Naur par2cmdline-0.4-tbb-20100203/par2cmdline.cpp par2cmdline-0.4-tbb-20100203.new/par2cmdline.cpp
--- par2cmdline-0.4-tbb-20100203/par2cmdline.cpp	2009-02-03 06:14:52.000000000 +0100
+++ par2cmdline-0.4-tbb-20100203.new/par2cmdline.cpp	2010-02-13 18:49:14.501692109 +0100
@@ -24,6 +24,7 @@
 //  Copyright (c) 2007 Intel Corp.
 
 #include "par2cmdline.h"
+#include <memory>
 
 #ifdef _MSC_VER
 #ifdef _DEBUG
diff -Naur par2cmdline-0.4-tbb-20100203/par2creator.cpp par2cmdline-0.4-tbb-20100203.new/par2creator.cpp
--- par2cmdline-0.4-tbb-20100203/par2creator.cpp	2008-10-09 15:37:56.000000000 +0200
+++ par2cmdline-0.4-tbb-20100203.new/par2creator.cpp	2010-02-13 18:49:57.585021162 +0100
@@ -30,6 +30,7 @@
 //  par2cmdline-0.4-tbb is available at http://chuchusoft.com/par2_tbb
 
 #include "par2cmdline.h"
+#include <memory>
 
 #if WANT_CONCURRENT
   #if CONCURRENT_PIPELINE
diff -Naur par2cmdline-0.4-tbb-20100203/par2repairer.cpp par2cmdline-0.4-tbb-20100203.new/par2repairer.cpp
--- par2cmdline-0.4-tbb-20100203/par2repairer.cpp	2009-02-03 06:14:42.000000000 +0100
+++ par2cmdline-0.4-tbb-20100203.new/par2repairer.cpp	2010-02-13 18:50:04.215020124 +0100
@@ -30,6 +30,7 @@
 //  par2cmdline-0.4-tbb is available at http://chuchusoft.com/par2_tbb
 
 #include "par2cmdline.h"
+#include <memory>
 
 //static unsigned gti;
 //static tbb::atomic<unsigned> gti;
