--- a/reedsolomon-i386-scalar-posix.s	
+++ a/reedsolomon-i386-scalar-posix.s	
@@ -31,6 +31,10 @@ 
 #
 # void rs_process_i386_scalar(void* dst, const void* src, size_t size, const u32* LH);
 #
+
+#if defined(__linux__) && defined(__ELF__)
+.section        .note.GNU-stack, "", %progbits
+#endif
 	.globl rs_process_i386_scalar
 	.text
 rs_process_i386_scalar:
--- a/reedsolomon-i686-mmx-posix.s	
+++ a/reedsolomon-i686-mmx-posix.s	
@@ -31,6 +31,10 @@ 
 #
 # void rs_process_i686_mmx(void* dst, const void* src, size_t size, unsigned* LH);
 #
+
+#if defined(__linux__) && defined(__ELF__)
+.section        .note.GNU-stack, "", %progbits
+#endif
 	.globl rs_process_i686_mmx
 	.text
 rs_process_i686_mmx:
--- a/reedsolomon-x86_64-mmx-posix.s	
+++ a/reedsolomon-x86_64-mmx-posix.s	
@@ -31,6 +31,10 @@ 
 #
 # void rs_process_x86_64_mmx(void* dst, const void* src, size_t size, unsigned* LH);
 #
+
+#if defined(__linux__) && defined(__ELF__)
+.section        .note.GNU-stack, "", %progbits
+#endif
 	.globl rs_process_x86_64_mmx
 	.text
 rs_process_x86_64_mmx:
--- a/reedsolomon-x86_64-scalar-posix.s	
+++ a/reedsolomon-x86_64-scalar-posix.s	
@@ -29,6 +29,10 @@ 
 #
 # void rs_process_x86_64_scalar(void* dst, const void* src, size_t size, const u32* LH);
 #
+
+#if defined(__linux__) && defined(__ELF__)
+.section        .note.GNU-stack, "", %progbits
+#endif
 	.globl rs_process_x86_64_scalar
 	.text
 rs_process_x86_64_scalar:
