--- OpenCV-2.1.0/src/highgui/grfmt_png.cpp	2010-04-05 21:24:44.000000000 -0400
+++ src/OpenCV-2.1.0/src/highgui/grfmt_png.cpp	2012-03-13 19:50:10.000000000 -0400
@@ -51,11 +51,7 @@
     and png2bmp sample from libpng distribution (Copyright (C) 1999-2001 MIYASAKA Masaru)
 \****************************************************************************************/
 
-#ifdef HAVE_LIBPNG_PNG_H
-#include <libpng/png.h>
-#else
-#include <png.h>
-#endif
+#include <libpng14/png.h>
 #include "grfmt_png.h"
 
 namespace cv
@@ -223,7 +219,7 @@
                 png_set_palette_to_rgb( png_ptr );
 
             if( m_color_type == PNG_COLOR_TYPE_GRAY && m_bit_depth < 8 )
-                png_set_gray_1_2_4_to_8( png_ptr );
+                png_set_expand_gray_1_2_4_to_8( png_ptr );
 
             if( CV_MAT_CN(m_type) > 1 && color )
                 png_set_bgr( png_ptr ); // convert RGB to BGR
