--- Makefile.old	2013-11-04 11:48:45.175597297 +0100
+++ Makefile	2013-11-04 11:59:59.315023209 +0100
@@ -32,6 +32,9 @@
 # set to '1' for profiling
 PROFILE=0
 
+# Destination directory for "make install"
+DESTDIR="/usr/local"
+
 ####### END: BUILD CONFIGURATION #######
 
 ifeq ($(USE_CLANG),1)
@@ -54,7 +57,7 @@
     CXXFLAGS += -fcatch-undefined-behavior
   endif
 else
-  CXXFLAGS += -O3 -march=native
+  CXXFLAGS += -O3
 endif
 
 ifeq ($(PROFILE),1)
@@ -90,7 +93,8 @@
 	pdflatex manual.tex || exit 1; cp manual.log manual.prv; done
 
 install: intersynth
-	install intersynth /usr/local/bin/
+	mkdir -p "$(DESTDIR)/bin"
+	install intersynth "$(DESTDIR)/bin/"
 
 clean:
 	rm -f intersynth *.o *.d
