--- violetland/CMakeLists.txt	2012-03-26 12:54:23.668219985 +0200
+++ violetland_fs3/CMakeLists.txt	2012-09-24 00:12:42.088711963 +0200
@@ -8,7 +8,7 @@
 find_package(SDL_mixer REQUIRED)
 find_package(OpenGL REQUIRED)
 find_package(Boost COMPONENTS filesystem system REQUIRED)
-add_definitions(-DBOOST_FILESYSTEM_VERSION=2)
+add_definitions(-DBOOST_FILESYSTEM_VERSION=3)
 
 if(${MINGW})
 	set(INTL_LIBRARY intl)
--- violetland/src/system/utility/FileUtility.cpp	2012-03-26 12:54:11.328219703 +0200
+++ violetland_fs3/src/system/utility/FileUtility.cpp	2012-09-24 00:28:43.312216545 +0200
@@ -20,7 +20,7 @@
 
 	while (dir_it != boost::filesystem::directory_iterator()) {
 		if (boost::filesystem::is_regular_file(*dir_it))
-			files.push_back(dir_it->filename());
+			files.push_back(dir_it->path().filename().string());
 		++dir_it;
 	}
 	return files;
@@ -32,8 +32,8 @@
 
 	while (dir_it != boost::filesystem::directory_iterator()) {
 		if (boost::filesystem::is_directory(*dir_it))
-			if (dir_it->path().filename()[0] != '.')
-				subDirs.push_back(dir_it->path().filename());
+			if (dir_it->path().string()[0] != '.')
+				subDirs.push_back(dir_it->path().filename().string());
 		++dir_it;
 	}
 	return subDirs;
@@ -45,7 +45,7 @@
 
 	while (dir_it != boost::filesystem::directory_iterator()) {
 		if (boost::filesystem::is_directory(*dir_it))
-			if (dir_it->path().filename()[0] != '.')
+			if (dir_it->path().string()[0] != '.')
 				++count;
 		++dir_it;
 	}
