--- RosBE-Builder.sh	2011-05-26 14:02:33.832258692 +0300
+++ RosBE-Builder.sh.no_interact	2011-05-26 14:35:48.454646641 +0300
@@ -170,7 +170,7 @@
 
 	echo "Ready to build and install the ReactOS Build Environment."
 	echo "Press Return to continue or Ctrl+C to exit."
-	read
+
 else
 	installdir=`eval echo $1`
 
@@ -227,16 +227,14 @@
 rs_mkdir_if_not_exists "$rs_prefixdir/$rs_target"
 rs_mkdir_if_not_exists "$rs_supportprefixdir"
 
-# Use -march=native if the host compiler supports it
-echo -n "Checking if the host compiler supports -march=native... "
+# Load the -march setting from makepkg.conf
+source /etc/makepkg.conf
+for i in `echo $CFLAGS`; do
+	if `echo "$i" | grep -q "\-march="`; then
+		rs_host_cflags+=" $i"
+	fi
+done 
 
-if `gcc -march=native -o "$rs_workdir/dummy" "$rs_scriptdir/tools/dummy.c" >& /dev/null`; then
-	echo "yes"
-	rs_host_cflags+=" -march=native"
-	rm "$rs_workdir/dummy"
-else
-	echo "no"
-fi
 
 rs_extract_module "mingw_runtime_dev" "$rs_prefixdir/$rs_target"
 rs_extract_module "w32api" "$rs_prefixdir/$rs_target"
@@ -249,7 +247,17 @@
 	rs_do_command gcc -s -o "$installdir/bin/cpucount" "$rs_scriptdir/tools/cpucount.c"
 fi
 
-rs_cpucount=`$installdir/bin/cpucount -x1`
+# Load the job thread count from makepkg.conf
+# if set, otherwice use cpucount
+if [[ "$MAKEFLAGS" == "" ]]; then
+	rs_cpucount=`$installdir/bin/cpucount -x1`
+else
+	for i in `echo $MAKEFLAGS`; do
+		if `echo "$i" | grep -q "\-j"`; then
+			rs_cpucount=`echo "$i" | sed 's|-j||g'`
+		fi
+	done
+fi
 
 if $rs_process_getincludes; then
 	rs_do_command gcc -s -o "$installdir/bin/getincludes" "$rs_scriptdir/tools/getincludes.c"
@@ -356,15 +364,4 @@
 
 # Finish
 rs_boldmsg "Finished successfully!"
-echo "To create a shortcut to the Build Environment on the Desktop, please switch back to your"
-echo "normal User Account (I assume you ran this script as \"root\")."
-echo "Then execute the following command:"
-echo
-echo "  $installdir/createshortcut.sh"
-echo
-echo "If you just want to start the Build Environment without using a shortcut, execute the"
-echo "following command:"
-echo
-echo "  $installdir/RosBE.sh [source directory] [color code] [architecture]"
-echo
-echo "All parameters for that script are optional."
+
