diff -aur pitz-git.pristine/pitz/cmdline/__init__.py pitz-git.patched/pitz/cmdline/__init__.py
--- pitz-git.pristine/pitz/cmdline/__init__.py	2012-03-03 22:25:22.000000000 +0200
+++ pitz-git.patched/pitz/cmdline/__init__.py	2012-03-04 12:56:25.962736912 +0200
@@ -535,8 +535,17 @@
     ns['send_through_pager'] = clepy.send_through_pager
     ns['edit_with_editor'] = clepy.edit_with_editor
 
-    from IPython.Shell import IPShellEmbed
-    s = IPShellEmbed(['-colors', 'Linux'])
+    try:
+        # ipython >= 0.11
+        from IPython.frontend.terminal.embed import InteractiveShellEmbed
+        from IPython.config.loader import Config
+        cfg = Config()
+        cfg.InteractiveShellEmbed.colors = 'Linux'
+        s = InteractiveShellEmbed(config=cfg)
+    except ImportError:
+        # ipython < 0.11
+        from IPython.Shell import IPShellEmbed
+        s = IPShellEmbed(['-colors', 'Linux'])
     s(local_ns=ns)
 
     # This stuff happens when you close the IPython session.
