--- xen-4.0.0.orig/config/x86_32.mk	2010-04-07 18:12:04.000000000 +0200
+++ xen-4.0.0/config/x86_32.mk	2010-05-16 21:38:33.360521446 +0200
@@ -7,7 +7,7 @@
 CONFIG_XCUTILS := y
 CONFIG_IOEMU := y
 
-CFLAGS += -m32 -march=i686
+CFLAGS += -m32 -march=i686 -I/opt/lib32/usr/include
 
 # Use only if calling $(LD) directly.
 LDFLAGS_DIRECT_OpenBSD = _obsd

--- xen-4.0.0.orig/Config.mk	2010-04-07 18:12:04.000000000 +0200
+++ xen-4.0.0/Config.mk	2010-05-16 21:38:33.360521446 +0200
@@ -173,4 +173,4 @@
 CONFIG_MINITERM    ?= n
 CONFIG_LOMOUNT     ?= n
 
--include $(XEN_ROOT)/.config
+#-include $(XEN_ROOT)/.config
