=== modified file 'opensand-rt/src/BlockManager.cpp'
--- opensand-rt/src/BlockManager.cpp	2014-04-22 09:41:18 +0000
+++ opensand-rt/src/BlockManager.cpp	2014-04-30 18:27:29 +0000
@@ -39,6 +39,7 @@
 
 #include <opensand_output/Output.h>
 
+#include <unistd.h>
 #include <signal.h>
 #include <cstdio>
 #include <cstring>

=== modified file 'opensand-rt/src/MessageEvent.cpp'
--- opensand-rt/src/MessageEvent.cpp	2014-04-22 09:41:18 +0000
+++ opensand-rt/src/MessageEvent.cpp	2014-04-30 18:56:25 +0000
@@ -39,6 +39,7 @@
 
 #include <cstring>
 #include <errno.h>
+#include <unistd.h>
 
 
 MessageEvent::MessageEvent(RtFifo *const fifo,

=== modified file 'opensand-rt/src/RtEvent.cpp'
--- opensand-rt/src/RtEvent.cpp	2014-04-22 09:41:18 +0000
+++ opensand-rt/src/RtEvent.cpp	2014-04-30 18:55:14 +0000
@@ -39,6 +39,7 @@
 #include <opensand_output/Output.h>
 
 #include <cstdlib>
+#include <unistd.h>
 
 RtEvent::RtEvent(event_type_t type, const string &name, int32_t fd, uint8_t priority):
 	type(type),

=== modified file 'opensand-rt/src/test/TestBlock.cpp'
--- opensand-rt/src/test/TestBlock.cpp	2014-04-22 09:41:18 +0000
+++ opensand-rt/src/test/TestBlock.cpp	2014-04-30 18:58:14 +0000
@@ -67,6 +67,7 @@
 #include <cstring>
 #include <cassert>
 #include <signal.h>
+#include <unistd.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>

=== modified file 'opensand-rt/src/test/TestMultiBlocks.cpp'
--- opensand-rt/src/test/TestMultiBlocks.cpp	2014-04-22 09:41:18 +0000
+++ opensand-rt/src/test/TestMultiBlocks.cpp	2014-04-30 18:59:44 +0000
@@ -77,6 +77,7 @@
 #include <signal.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <unistd.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>

