--- rdesktop.orig/disk.c	2008-02-16 03:13:25 +0300
+++ rdesktop/disk.c	2008-07-04 14:13:52 +0400
@@ -1201,8 +1201,8 @@
 			out_uint32_le(out, filestat.st_size);	/* filesize low */
 			out_uint32_le(out, 0);	/* filesize high */
 			out_uint32_le(out, file_attributes);
-			out_uint8(out, 2 * strlen(pdirent->d_name) + 2);	/* unicode length */
-			out_uint8s(out, 7);	/* pad? */
+			out_uint16(out, 2 * strlen(pdirent->d_name) + 2);	/* unicode length */
+			out_uint8s(out, 6);	/* pad? */
 			out_uint8(out, 0);	/* 8.3 file length */
 			out_uint8s(out, 2 * 12);	/* 8.3 unicode length */
 			rdp_out_unistr(out, pdirent->d_name, 2 * strlen(pdirent->d_name));
