diff -aur snapscreenshot-1.0.14.2.orig/snapscreenshot.cc snapscreenshot-1.0.14.2.new/snapscreenshot.cc
--- snapscreenshot-1.0.14.2.orig/snapscreenshot.cc	2008-12-28 18:53:02.000000000 +0100
+++ snapscreenshot-1.0.14.2.new/snapscreenshot.cc	2008-12-28 18:58:06.000000000 +0100
@@ -7,6 +7,8 @@
 #include <unistd.h>
 #include <vector>
 #include <string>
+#include <cstring>
+#include <cstdlib>
 
 using namespace std;
 
