diff -ruN Trash/AppRun Trash_a/AppRun
--- Trash/AppRun	2012-05-08 22:18:17.000000000 -0400
+++ Trash_a/AppRun	2012-05-08 22:21:11.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 # Copyright (c) 2003 Peter A. Geer
 # This program is distributed under the GNU General Public License.
diff -ruN Trash/choices.sh Trash_a/choices.sh
--- Trash/choices.sh	2012-05-08 22:18:17.000000000 -0400
+++ Trash_a/choices.sh	2012-05-08 22:21:38.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 # choices.sh - prints out the path to the user's Choices directory for
 #              an application.
diff -ruN Trash/getname.sh Trash_a/getname.sh
--- Trash/getname.sh	2012-05-08 22:18:17.000000000 -0400
+++ Trash_a/getname.sh	2012-05-08 22:21:59.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 # Takes 2 arguments: a target path and a file name.
 
diff -ruN Trash/Help/libtrash.sh Trash_a/Help/libtrash.sh
--- Trash/Help/libtrash.sh	2012-05-08 22:18:17.000000000 -0400
+++ Trash_a/Help/libtrash.sh	2012-05-08 22:23:16.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 export LD_PRELOAD=/usr/lib/libtrash.so
 CONF_FILE=$HOME/.libtrash.conf
 if [ ! -e $CONF_FILE ] ; then
diff -ruN Trash/parser.sh Trash_a/parser.sh
--- Trash/parser.sh	2012-05-08 22:18:17.000000000 -0400
+++ Trash_a/parser.sh	2012-05-08 22:22:16.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 # Extracts fields from PyMessage output by name or number.
 # Copyright (c) 2003 Peter A. Geer
diff -ruN Trash/simple-msg.sh Trash_a/simple-msg.sh
--- Trash/simple-msg.sh	2012-05-08 22:18:17.000000000 -0400
+++ Trash_a/simple-msg.sh	2012-05-08 22:22:32.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 if [ -n "$MSG_FILE" ] ; then
 	source "$MSG_FILE"
diff -ruN Trash/text/C-text.sh Trash_a/text/C-text.sh
--- Trash/text/C-text.sh	2012-05-08 22:18:17.000000000 -0400
+++ Trash_a/text/C-text.sh	2012-05-08 22:23:37.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 # Source this script to provide message text for Trash.
 TRASH_NOT_SET="Trash can directory does not exist.
 Is libtrash set up correctly?"
diff -ruN Trash/text/es_ES-text.sh Trash_a/text/es_ES-text.sh
--- Trash/text/es_ES-text.sh	2012-05-08 22:18:17.000000000 -0400
+++ Trash_a/text/es_ES-text.sh	2012-05-08 22:23:54.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 # Source this script to provide message text for Trash.
 TRASH_NOT_SET="El directorio de la Papelera no existe.
 ¿Está libtrash configurado correctamente?"
diff -ruN Trash/text/fr_FR-text.sh Trash_a/text/fr_FR-text.sh
--- Trash/text/fr_FR-text.sh	2012-05-08 22:18:17.000000000 -0400
+++ Trash_a/text/fr_FR-text.sh	2012-05-08 22:24:12.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 # Source this script to provide message text for Trash.
 TRASH_NOT_SET="Le dossier Trash n'existe pas.
 Est-ce que la librairie libtrash est configuré correctement ?"
