--- src/wisent-0.6/grammar.py.orig	2010-05-27 01:38:30.208235239 +0300
+++ src/wisent-0.6/grammar.py	2010-05-27 01:40:24.634859929 +0300
@@ -192,6 +192,20 @@
         """Return the `label` given at object construction."""
         return self.label
 
+    def __lt__(self, other):
+        if isinstance(other, str):
+            return self.label < other
+        if isinstance(other, Unique):
+            return self.label < other.label
+        return NotImplemented()
+
+    def __gt__(self, other):
+        if isinstance(other, str):
+            return self.label > other
+        if isinstance(other, Unique):
+            return self.label > other.label
+        return NotImplemented()
+
 class Grammar(object):
 
     """Represent a context free grammar."""
