--- makefile.unices	2007-07-20 18:45:48.000000000 +0530
+++ makefile.unices	2008-08-03 14:42:45.000000000 +0530
@@ -26,18 +26,18 @@
 #*/
 
 # MUST EXIST
-prefix = /usr/local
+prefix = /usr
 exec_prefix = $(prefix)
 # where to put the executables
 bindir = $(exec_prefix)/bin
-mandir = $(prefix)/man
+mandir = $(prefix)/share/man
 # MUST EXIST
-docdir = $(prefix)/doc/MELTING
+docdir = $(prefix)/doc/melting
 srcdir = SRC
 
 # where to find the files containing the sets of nearest_neighbor parameters. 
 # This one is to initialise the default value of melting. 
-NNDIR = $(prefix)/share/MELTING/NNFILES
+NNDIR = $(prefix)/share/melting/NNFILES
 
 #INSTALL = /usr/bin/install -c
 VPATH = $(srcdir)
@@ -60,12 +60,12 @@
 calcul.o : calcul.c calcul.h
 
 install :
-	cp melting $(bindir)
-	cp BIN/tkmelting.pl $(bindir)
-	cp DOC/melting.1 $(mandir)/man1/melting.1
-	if test -e $(prefix)/share/MELTING ; then echo '$(prefix)/share/MELTING is already present'; else mkdir $(prefix)/share/MELTING; fi
-	if test -e $(prefix)/share/MELTING/NNFILES; then echo '$(prefix)/share/MELTING/NNFILES is already present'; else mkdir $(prefix)/share/MELTING/NNFILES; fi
-	cp NNFILES/*.nn $(NNDIR)/
+	install -D -m755 melting $(DESTDIR)/$(bindir)/melting
+	install -D -m755 BIN/tkmelting.pl $(DESTDIR)/$(bindir)/tkmelting.pl
+	install -D -m644 DOC/melting.1 $(DESTDIR)/$(mandir)/man1/melting.1
+	if test -e $(DESTDIR)/$(prefix)/share/melting ; then echo '$(prefix)/share/MELTING is already present'; else mkdir $(DESTDIR)/$(prefix)/share/melting; fi
+	if test -e $(DESTDIR)/$(prefix)/share/melting/NNFILES; then echo '$(prefix)/share/melting/NNFILES is already present'; else mkdir $(DESTDIR)/$(prefix)/share/melting/NNFILES; fi
+	cp NNFILES/*.nn $(DESTDIR)/$(NNDIR)/
 
 .PHONY : clean
 clean :
