Index: src/xfwm4/src/events.c
===================================================================
--- xfwm4-4.10.1.orig/src/events.c
+++ xfwm4-4.10.1/src/events.c
@@ -890,6 +890,13 @@ handleButtonPress (DisplayInfo *display_
     {
         state = ev->state & MODIFIER_MASK;
         win = ev->subwindow;
+        if (!win) {
+	    Window root_ret;
+	    int rx,ry,wx,wy;
+	    unsigned int mask_ret;
+	    XQueryPointer(display_info->dpy, ev->window,
+	        &root_ret, &win, &rx, &ry, &wx, &wy, &mask_ret);
+        }		
         screen_info = c->screen_info;
 
         if ((ev->button == Button1) && (state) && (state == screen_info->params->easy_click))
