diff -Nurp liferea-1.6.0.orig/src/db.c liferea-1.6.0/src/db.c
--- liferea-1.6.0.orig/src/db.c	2009-10-21 18:15:30.409704157 -0300
+++ liferea-1.6.0/src/db.c	2009-10-21 18:16:36.898701497 -0300
@@ -171,7 +171,18 @@ open:
 	g_free (filename);
 	
 	sqlite3_extended_result_codes (db, TRUE);
-	
+
+	char *sync = getenv ("LIFEREA_SYNCHRONOUS");
+	if (sync)
+	{
+		gchar *err;
+		gchar *sql = sqlite3_mprintf ("pragma synchronous = %d;", atoi(sync));
+		if (SQLITE_OK != sqlite3_exec(db, sql, NULL, NULL, &err))
+			debug1 (DEBUG_DB, "setting pragma synchronous failed: %s", err);
+		sqlite3_free (sql);
+		sqlite3_free (err);
+	}
+
 	/* create info table/check versioning info */				   
 	schemaVersion = db_get_schema_version ();
 	debug1 (DEBUG_DB, "current DB schema version: %d", schemaVersion);
