diff -Naur xournal/autogen.sh xournal_autotools/autogen.sh
--- xournal/autogen.sh	2013-03-04 15:47:48.000845524 +0100
+++ xournal_autotools/autogen.sh	2013-03-04 15:47:16.090845412 +0100
@@ -14,7 +14,7 @@
 	export LD_LIBRARY_PATH
 fi
 
-(test -f $srcdir/configure.in) || {
+(test -f $srcdir/configure.ac) || {
     echo -n "**Error**: Directory "\`$srcdir\'" does not look like the"
     echo " top-level package directory"
     exit 1
@@ -28,7 +28,7 @@
   DIE=1
 }
 
-(grep "^AC_PROG_INTLTOOL" $srcdir/configure.in >/dev/null) && {
+(grep "^AC_PROG_INTLTOOL" $srcdir/configure.ac >/dev/null) && {
   (intltoolize --version) < /dev/null > /dev/null 2>&1 || {
     echo 
     echo "**Error**: You must have \`intltool' installed."
@@ -38,7 +38,7 @@
   }
 }
 
-(grep "^AM_PROG_XML_I18N_TOOLS" $srcdir/configure.in >/dev/null) && {
+(grep "^AM_PROG_XML_I18N_TOOLS" $srcdir/configure.ac >/dev/null) && {
   (xml-i18n-toolize --version) < /dev/null > /dev/null 2>&1 || {
     echo
     echo "**Error**: You must have \`xml-i18n-toolize' installed."
@@ -48,7 +48,7 @@
   }
 }
 
-(grep "^AM_PROG_LIBTOOL" $srcdir/configure.in >/dev/null) && {
+(grep "^AM_PROG_LIBTOOL" $srcdir/configure.ac >/dev/null) && {
   (libtool --version) < /dev/null > /dev/null 2>&1 || {
     echo
     echo "**Error**: You must have \`libtool' installed."
@@ -57,8 +57,8 @@
   }
 }
 
-(grep "^AM_GLIB_GNU_GETTEXT" $srcdir/configure.in >/dev/null) && {
-  (grep "sed.*POTFILES" $srcdir/configure.in) > /dev/null || \
+(grep "^AM_GLIB_GNU_GETTEXT" $srcdir/configure.ac >/dev/null) && {
+  (grep "sed.*POTFILES" $srcdir/configure.ac) > /dev/null || \
   (glib-gettextize --version) < /dev/null > /dev/null 2>&1 || {
     echo
     echo "**Error**: You must have \`glib' installed."
@@ -101,7 +101,7 @@
   am_opt=--include-deps;;
 esac
 
-for coin in `find $srcdir -path $srcdir/CVS -prune -o -name configure.in -print`
+for coin in `find $srcdir -path $srcdir/CVS -prune -o -name configure.ac -print`
 do 
   dr=`dirname $coin`
   if test -f $dr/NO-AUTO-GEN; then
@@ -112,7 +112,7 @@
 
       aclocalinclude="$ACLOCAL_FLAGS"
 
-      if grep "^AM_GLIB_GNU_GETTEXT" configure.in >/dev/null; then
+      if grep "^AM_GLIB_GNU_GETTEXT" configure.ac >/dev/null; then
 	echo "Creating $dr/aclocal.m4 ..."
 	test -r $dr/aclocal.m4 || touch $dr/aclocal.m4
 	echo "Running glib-gettextize...  Ignore non-fatal messages."
@@ -120,15 +120,15 @@
 	echo "Making $dr/aclocal.m4 writable ..."
 	test -r $dr/aclocal.m4 && chmod u+w $dr/aclocal.m4
       fi
-      if grep "^AC_PROG_INTLTOOL" configure.in >/dev/null; then
+      if grep "^AC_PROG_INTLTOOL" configure.ac >/dev/null; then
         echo "Running intltoolize..."
 	intltoolize --copy --force --automake
       fi
-      if grep "^AM_PROG_XML_I18N_TOOLS" configure.in >/dev/null; then
+      if grep "^AM_PROG_XML_I18N_TOOLS" configure.ac >/dev/null; then
         echo "Running xml-i18n-toolize..."
 	xml-i18n-toolize --copy --force --automake
       fi
-      if grep "^AM_PROG_LIBTOOL" configure.in >/dev/null; then
+      if grep "^AM_PROG_LIBTOOL" configure.ac >/dev/null; then
 	if test -z "$NO_LIBTOOLIZE" ; then 
 	  echo "Running libtoolize..."
 	  libtoolize --force --copy
@@ -136,7 +136,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AM_CONFIG_HEADERS" configure.ac >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -Naur xournal/config.h.in xournal_autotools/config.h.in
--- xournal/config.h.in	2013-03-04 15:47:48.000845524 +0100
+++ xournal_autotools/config.h.in	2013-03-04 15:47:16.090845412 +0100
@@ -1,4 +1,4 @@
-/* config.h.in.  Generated from configure.in by autoheader.  */
+/* config.h.in.  Generated from configure.ac by autoheader.  */
 
 /* always defined to indicate that i18n is enabled */
 #undef ENABLE_NLS
diff -Naur xournal/configure.ac xournal_autotools/configure.ac
--- xournal/configure.ac	1970-01-01 01:00:00.000000000 +0100
+++ xournal_autotools/configure.ac	2013-03-04 15:47:16.090845412 +0100
@@ -0,0 +1,48 @@
+dnl Process this file with autoconf to produce a configure script.
+
+AC_INIT([xournal], [0.4.7])
+AC_CONFIG_SRCDIR([configure.ac])
+AC_CONFIG_HEADERS(config.h)
+AM_INIT_AUTOMAKE
+AM_MAINTAINER_MODE
+
+AC_SEARCH_LIBS([strerror],[cposix])
+AC_PROG_CC
+AC_PROG_RANLIB
+AC_HEADER_STDC
+
+LDFLAGS="$LDFLAGS -lz -lm"
+
+pkg_modules="gtk+-2.0 >= 2.10.0 libgnomecanvas-2.0 >= 2.4.0 poppler-glib >= 0.5.4 pangoft2 >= 1.0"
+PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
+AC_SUBST(PACKAGE_CFLAGS)
+AC_SUBST(PACKAGE_LIBS)
+
+dnl detect Windows environment...
+AC_CANONICAL_HOST
+AC_MSG_CHECKING([for Windows environment])
+case "$host" in
+ *-*-mingw*)
+   os_win32=yes;;
+ *)
+   os_win32=no;;
+esac
+AC_MSG_RESULT([$os_win32])
+AM_CONDITIONAL(WIN32, test "$os_win32" = "yes")
+
+GETTEXT_PACKAGE=xournal
+AC_SUBST(GETTEXT_PACKAGE)
+AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [Gettext package.])
+
+dnl Add the languages which your application supports here.
+ALL_LINGUAS="`grep -v '^#' "$srcdir/po/LINGUAS" | tr '\n' ' '`"
+AM_GLIB_GNU_GETTEXT
+
+AC_CONFIG_FILES([
+Makefile
+src/Makefile
+src/ttsubset/Makefile
+po/Makefile.in
+])
+AC_OUTPUT
+
diff -Naur xournal/configure.in xournal_autotools/configure.in
--- xournal/configure.in	2013-03-04 15:47:48.000845524 +0100
+++ xournal_autotools/configure.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,47 +0,0 @@
-dnl Process this file with autoconf to produce a configure script.
-
-AC_INIT(configure.in)
-AM_INIT_AUTOMAKE(xournal, 0.4.7)
-AM_CONFIG_HEADER(config.h)
-AM_MAINTAINER_MODE
-
-AC_ISC_POSIX
-AC_PROG_CC
-AM_PROG_CC_STDC
-AC_PROG_RANLIB
-AC_HEADER_STDC
-
-LDFLAGS="$LDFLAGS -lz -lm"
-
-pkg_modules="gtk+-2.0 >= 2.10.0 libgnomecanvas-2.0 >= 2.4.0 poppler-glib >= 0.5.4 pangoft2 >= 1.0"
-PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
-AC_SUBST(PACKAGE_CFLAGS)
-AC_SUBST(PACKAGE_LIBS)
-
-dnl detect Windows environment...
-AC_CANONICAL_HOST
-AC_MSG_CHECKING([for Windows environment])
-case "$host" in
- *-*-mingw*)
-   os_win32=yes;;
- *)
-   os_win32=no;;
-esac
-AC_MSG_RESULT([$os_win32])
-AM_CONDITIONAL(WIN32, test "$os_win32" = "yes")
-
-GETTEXT_PACKAGE=xournal
-AC_SUBST(GETTEXT_PACKAGE)
-AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [Gettext package.])
-
-dnl Add the languages which your application supports here.
-ALL_LINGUAS="`grep -v '^#' "$srcdir/po/LINGUAS" | tr '\n' ' '`"
-AM_GLIB_GNU_GETTEXT
-
-AC_OUTPUT([
-Makefile
-src/Makefile
-src/ttsubset/Makefile
-po/Makefile.in
-])
-
diff -Naur xournal/src/Makefile.am xournal_autotools/src/Makefile.am
--- xournal/src/Makefile.am	2013-03-04 15:47:48.000845524 +0100
+++ xournal_autotools/src/Makefile.am	2013-03-04 15:47:16.090845412 +0100
@@ -6,7 +6,7 @@
   SUBDIRS = ttsubset
 endif
 
-INCLUDES = \
+AM_CPPFLAGS = \
 	-DPACKAGE_DATA_DIR=\""$(datadir)"\" \
 	-DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
 	@PACKAGE_CFLAGS@
diff -Naur xournal/src/ttsubset/Makefile.am xournal_autotools/src/ttsubset/Makefile.am
--- xournal/src/ttsubset/Makefile.am	2013-03-04 15:47:48.000845524 +0100
+++ xournal_autotools/src/ttsubset/Makefile.am	2013-03-04 15:47:16.090845412 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = \
+AM_CPPFLAGS = \
 	@PACKAGE_CFLAGS@ -DNO_MAPPERS -DNO_TYPE3 -DNO_TYPE42
 
 noinst_LIBRARIES = libttsubset.a
