diff --git a/deps/openssl/Makefile.openssl b/deps/openssl/Makefile.openssl
index cb7a3fc..83827f6 100644
--- a/deps/openssl/Makefile.openssl
+++ b/deps/openssl/Makefile.openssl
@@ -1,39 +1,39 @@
 OS_NAME=$(shell uname -s)
 MH_NAME=$(shell uname -m)
 
-CFLAGS += -Iopenssl
-CFLAGS += -Iopenssl-configs/realized
-CFLAGS += -Iopenssl/include
-CFLAGS += -Iopenssl/crypto
-CFLAGS += -Iopenssl/crypto/asn1
-CFLAGS += -Iopenssl/crypto/evp
-CFLAGS += -Iopenssl-configs
+CPPFLAGS += -Iopenssl
+CPPFLAGS += -Iopenssl-configs/realized
+CPPFLAGS += -Iopenssl/include
+CPPFLAGS += -Iopenssl/crypto
+CPPFLAGS += -Iopenssl/crypto/asn1
+CPPFLAGS += -Iopenssl/crypto/evp
+CPPFLAGS += -Iopenssl-configs
 ifeq (${MH_NAME},x86_64)
-CFLAGS += -Iopenssl-configs/x64
+CPPFLAGS += -Iopenssl-configs/x64
 else
-CFLAGS += -Iopenssl-configs/ia32
+CPPFLAGS += -Iopenssl-configs/ia32
 endif
 
-CFLAGS += -DL_ENDIAN
-CFLAGS += -DOPENSSL_THREADS
-CFLAGS += -DPURIFY
-CFLAGS += -D_REENTRANT
-CFLAGS += -DOPENSSL_NO_ASM
-CFLAGS += -DOPENSSL_NO_INLINE_ASM
-CFLAGS += -DOPENSSL_NO_RC2
-CFLAGS += -DOPENSSL_NO_RC5
-CFLAGS += -DOPENSSL_NO_MD4
-CFLAGS += -DOPENSSL_NO_HW
-CFLAGS += -DOPENSSL_NO_GOST
-CFLAGS += -DOPENSSL_NO_CAMELLIA
-CFLAGS += -DOPENSSL_NO_CAPIENG
-CFLAGS += -DOPENSSL_NO_CMS
-CFLAGS += -DOPENSSL_NO_FIPS
-CFLAGS += -DOPENSSL_NO_IDEA
-CFLAGS += -DOPENSSL_NO_MDC2
-CFLAGS += -DOPENSSL_NO_MD2
-CFLAGS += -DOPENSSL_NO_SEED
-CFLAGS += -DOPENSSL_NO_SOCK
+CPPFLAGS += -DL_ENDIAN
+CPPFLAGS += -DOPENSSL_THREADS
+CPPFLAGS += -DPURIFY
+CPPFLAGS += -D_REENTRANT
+CPPFLAGS += -DOPENSSL_NO_ASM
+CPPFLAGS += -DOPENSSL_NO_INLINE_ASM
+CPPFLAGS += -DOPENSSL_NO_RC2
+CPPFLAGS += -DOPENSSL_NO_RC5
+CPPFLAGS += -DOPENSSL_NO_MD4
+CPPFLAGS += -DOPENSSL_NO_HW
+CPPFLAGS += -DOPENSSL_NO_GOST
+CPPFLAGS += -DOPENSSL_NO_CAMELLIA
+CPPFLAGS += -DOPENSSL_NO_CAPIENG
+CPPFLAGS += -DOPENSSL_NO_CMS
+CPPFLAGS += -DOPENSSL_NO_FIPS
+CPPFLAGS += -DOPENSSL_NO_IDEA
+CPPFLAGS += -DOPENSSL_NO_MDC2
+CPPFLAGS += -DOPENSSL_NO_MD2
+CPPFLAGS += -DOPENSSL_NO_SEED
+CPPFLAGS += -DOPENSSL_NO_SOCK
 
 SOURCES=openssl/crypto/aes/aes_cbc.c \
         openssl/crypto/aes/aes_cfb.c \
@@ -610,5 +610,5 @@ libopenssl.a: ${OBJECTS}
 	ar rcs libopenssl.a ${OBJECTS}
 
 build/%.o: build
-	$(CC) ${CFLAGS} -g -Wall -c $< -o $@
+	$(CC) ${CPPFLAGS} ${CFLAGS} -g -Wall -c $< -o $@
 
