--- avxsynth/builtinfunctions/src/filters/focus.cpp	2012-04-24 01:22:14.797459529 +0200
+++ avxsynth/builtinfunctions/src/filters/focus.fix.cpp	2012-04-24 01:22:31.637600039 +0200
@@ -519,9 +519,9 @@
 			vu   = p[x*2+1];
 			p[x*2+1] = w;
 		}
-		uc y     = ScaledPixelClip(p[x*2+0] * center_weight + (yy + p[x*2+2]) * outer_weight);
+		uc y_     = ScaledPixelClip(p[x*2+0] * center_weight + (yy + p[x*2+2]) * outer_weight);
 		yy       = p[x*2+0];
-		p[x*2+0] = y;
+		p[x*2+0] = y_;
 		p[x*2+1] = ScaledPixelClip(p[x*2+1] * center_weight + (uv + p[x*2+1]) * outer_weight);
 		p[x*2+2] = ScaledPixelClip(p[x*2+2] * center_weight + (yy + p[x*2+2]) * outer_weight);
 		p[x*2+3] = ScaledPixelClip(p[x*2+3] * center_weight + (vu + p[x*2+3]) * outer_weight);
