--- avxsynth/builtinfunctions/src/sources/source.fix.cpp	2012-04-24 19:50:59.709185760 +0200
+++ avxsynth/builtinfunctions/src/sources/source.cpp	2012-04-24 01:37:12.571616896 +0200
@@ -393,7 +393,7 @@
 		}
 	}
 	else if (vi.IsYUY2()) {
+		static const unsigned int top_two_thirds[] =
-		static const int top_two_thirds[] =
 //                LtGrey      Yellow        Cyan       Green     Magenta         Red        Blue
 			{ 0x80b480b4, 0x8ea22ca2, 0x2c839c83, 0x3a704870, 0xc654b854, 0xd4416441, 0x7223d423 }; //VYUY
 		w >>= 1;
@@ -406,7 +406,7 @@
 			p += pitch;
 		}
 
+		static const unsigned int two_thirds_to_three_quarters[] =
-		static const int two_thirds_to_three_quarters[] =
 //                 Blue        Black     Magenta       Black        Cyan       Black      LtGrey
 			{ 0x7223d423, 0x80108010, 0xc654b854, 0x80108010, 0x2c839c83, 0x80108010, 0x80b480b4 }; //VYUY
 		for (; y*4 < h*3; ++y) {
@@ -418,7 +418,7 @@
 			p += pitch;
 		}
 
+		static const unsigned int bottom_quarter[] =
-		static const int bottom_quarter[] =
 //                    -I       white          +Q       Black       -4ire       Black       +4ire       Black
 			{ 0x5f109e10, 0x80eb80eb, 0x9510ae10, 0x80108010, 0x80078007, 0x80108010, 0x80198019, 0x80108010 }; //VYUY
 		for (; y < h; ++y) {
