--- indextemplate2.html	2008-11-17 00:00:00.000000000 +0300
+++ indextemplate2.html.p	2014-02-02 22:46:07.675807101 +0400
@@ -2,7 +2,7 @@
 
 <HTML>
   <HEAD>
-    <META http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
+    <META http-equiv="Content-Type" content="text/html; charset=utf-8">
     <META name="generator" content="igal <!--#IGAL-OPTIONS#-->">
     <TITLE>Index of Pictures</TITLE>
     <LINK rel="stylesheet" type="text/css" href="<!--#IGAL-CSS#-->">
