--- slidetemplate2.html	2012-05-28 23:33:38.000000000 +0400
+++ slidetemplate2.html.p	2014-02-02 22:46:17.401876908 +0400
@@ -2,7 +2,7 @@
 
 <HTML>
   <HEAD>
-    <META http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
+    <META http-equiv="Content-Type" content="text/html; charset=utf-8">
     <META name="generator" content="igal 1.4 freshmeat.net/projects/igal">
     <TITLE><!--SLIDE-TITLE--></TITLE>
     <LINK rel="stylesheet" type="text/css" href="igal2.css">
