diff --git a/mk/rt.mk b/mk/rt.mk
index 14aa288..b187343 100644
--- a/mk/rt.mk
+++ b/mk/rt.mk
@@ -178,7 +178,7 @@ export PYTHONPATH := $(PYTHONPATH):$$(S)src/gyp/pylib
 
 $$(LIBUV_MAKEFILE_$(1)_$(2)):
 	(cd $(S)src/libuv/ && \
-	 ./gyp_uv -f make -Dtarget_arch=$$(LIBUV_ARCH_$(1)) -D ninja \
+	 python2 gyp_uv -f make -Dtarget_arch=$$(LIBUV_ARCH_$(1)) -D ninja \
 	   -Goutput_dir=$$(@D) --generator-output $$(@D))
 
 # XXX: Shouldn't need platform-specific conditions here
