diff --git a/build-libs.sh b/build-libs.sh
index 329ffc6..8aa57d0 100755
--- a/build-libs.sh
+++ b/build-libs.sh
@@ -87,11 +87,6 @@ function build-lib() {
 	echo '==============================='
 	echo '============ E-LIB ============'
 	echo '==============================='
-	if [ ! -d "${ESDK}/tools/e-gnu/bin" ]; then
-		echo "In order to build the E-LIB the e-gcc compiler is required. Please"
-		echo "install the Epiphany GNU tools suite first at ${ESDK}/tools/e-gnu!"
-		exit
-	fi
 	cd src/e-lib/Release
 #	make clean
 	make all
diff --git a/src/e-lib/Debug/makefile b/src/e-lib/Debug/makefile
index 1a831df..f2c8f1e 100644
--- a/src/e-lib/Debug/makefile
+++ b/src/e-lib/Debug/makefile
@@ -29,7 +29,7 @@ all: libe-lib.a
 libe-lib.a: $(OBJS) $(USER_OBJS)
 	@echo 'Building target: $@'
 	@echo 'Invoking: Epiphany archiver'
-	e-ar -r  "libe-lib.a" $(OBJS) $(USER_OBJS) $(LIBS)
+	epiphany-elf-ar -r  "libe-lib.a" $(OBJS) $(USER_OBJS) $(LIBS)
 	@echo 'Finished building target: $@'
 	@echo ' '
 
diff --git a/src/e-lib/Debug/src/subdir.mk b/src/e-lib/Debug/src/subdir.mk
index 19589fd..a0b819f 100644
--- a/src/e-lib/Debug/src/subdir.mk
+++ b/src/e-lib/Debug/src/subdir.mk
@@ -109,14 +109,14 @@ C_DEPS += \
 src/%.o: ../src/%.c
 	@echo 'Building file: $<'
 	@echo 'Invoking: Epiphany compiler'
-	e-gcc -I../include -O0 -falign-loops=8 -g3 -Wall -c -fmessage-length=0 -ffp-contract=fast -mlong-calls -mfp-mode=round-nearest -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o "$@" "$<"
+	epiphany-elf-gcc -I../include -O0 -falign-loops=8 -g3 -Wall -c -fmessage-length=0 -ffp-contract=fast -mlong-calls -mfp-mode=round-nearest -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o "$@" "$<"
 	@echo 'Finished building: $<'
 	@echo ' '
 
 src/%.o: ../src/%.s
 	@echo 'Building file: $<'
 	@echo 'Invoking: Epiphany assembler'
-	e-as -g -o "$@" "$<"
+	epiphany-elf-as -g -o "$@" "$<"
 	@echo 'Finished building: $<'
 	@echo ' '
 
diff --git a/src/e-lib/Release/makefile b/src/e-lib/Release/makefile
index 1a831df..f2c8f1e 100644
--- a/src/e-lib/Release/makefile
+++ b/src/e-lib/Release/makefile
@@ -29,7 +29,7 @@ all: libe-lib.a
 libe-lib.a: $(OBJS) $(USER_OBJS)
 	@echo 'Building target: $@'
 	@echo 'Invoking: Epiphany archiver'
-	e-ar -r  "libe-lib.a" $(OBJS) $(USER_OBJS) $(LIBS)
+	epiphany-elf-ar -r  "libe-lib.a" $(OBJS) $(USER_OBJS) $(LIBS)
 	@echo 'Finished building target: $@'
 	@echo ' '
 
diff --git a/src/e-lib/Release/src/subdir.mk b/src/e-lib/Release/src/subdir.mk
index b982cca..4e6a550 100644
--- a/src/e-lib/Release/src/subdir.mk
+++ b/src/e-lib/Release/src/subdir.mk
@@ -109,14 +109,14 @@ C_DEPS += \
 src/%.o: ../src/%.c
 	@echo 'Building file: $<'
 	@echo 'Invoking: Epiphany compiler'
-	e-gcc -I../include -O3 -falign-loops=8 -Wall -c -fmessage-length=0 -ffp-contract=fast -mlong-calls -mfp-mode=round-nearest -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o "$@" "$<"
+	epiphany-elf-gcc -I../include -O3 -falign-loops=8 -Wall -c -fmessage-length=0 -ffp-contract=fast -mlong-calls -mfp-mode=round-nearest -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o "$@" "$<"
 	@echo 'Finished building: $<'
 	@echo ' '
 
 src/%.o: ../src/%.s
 	@echo 'Building file: $<'
 	@echo 'Invoking: Epiphany assembler'
-	e-as  -o "$@" "$<"
+	epiphany-elf-as  -o "$@" "$<"
 	@echo 'Finished building: $<'
 	@echo ' '
 
