diff -aur yaourt-1.3.orig/lib/util.sh.in yaourt-1.3.new/lib/util.sh.in
--- yaourt-1.3.orig/lib/util.sh.in	2013-10-31 03:48:33.615937413 +0330
+++ yaourt-1.3.new/lib/util.sh.in	2013-10-31 03:49:06.999270859 +0330
@@ -164,7 +164,7 @@
 pacman_parse() { LC_ALL=C pacman --color never "${PACMAN_C_ARG[@]}" "$@"; }
 pacman_out()   { $PACMAN "${PACMAN_C_ARG[@]}" "${PACMAN_O_ARG[@]}" "$@"; }
 pkgquery()     { package-query "${PKGQUERY_C_ARG[@]}" "$@"; }
-curl_fetch()   { curl "${CURL_C_ARG[@]}" "$@"; }
+curl_fetch()   { curl -3 "${CURL_C_ARG[@]}" "$@"; }
 
 # Run editor
 # Usage: run_editor ($file, $default_answer)
