diff -aur gbtpd-0.2.1.orig/Makefile.in gbtpd-0.2.1/Makefile.in
--- gbtpd-0.2.1.orig/Makefile.in	2007-09-21 05:47:20.000000000 -0500
+++ gbtpd-0.2.1/Makefile.in	2010-08-05 22:56:22.000000000 -0500
@@ -227,7 +227,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = src po
-gbtpddocdir = ${prefix}/doc/gbtpd
+gbtpddocdir = ${prefix}/share/doc/gbtpd
 gbtpddoc_DATA = \
 	README\
 	COPYING\
diff -aur gbtpd-0.2.1.orig/src/socket.cpp gbtpd-0.2.1/src/socket.cpp
--- gbtpd-0.2.1.orig/src/socket.cpp	2007-09-02 07:24:59.000000000 -0500
+++ gbtpd-0.2.1/src/socket.cpp	2010-08-05 22:51:35.000000000 -0500
@@ -16,6 +16,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <stdint.h>
 #include "socket.hpp"
 
 // Constructor
