diff -Naur wmnetload-1.3/src/wmnetload.c wmnetload-1.3-tn2/src/wmnetload.c
--- wmnetload-1.3/src/wmnetload.c	Wed Jun 11 18:15:36 2003
+++ wmnetload-1.3-tn2/src/wmnetload.c	Wed Jun 11 18:04:55 2003
@@ -140,7 +140,7 @@
 static unsigned long getblendedcolor(const char *, int);
 
 static Pixmap backlight_on, backlight_off, backlight_err, backlight_down;
-static Pixmap parts, font;
+static Pixmap backlight_down_on, backlight_down_off, parts, font;
 
 static char *desc = "\nNetwork interface usage monitor.\n";
 static char *vers = "wmnetload "VERSION" by meem@gnu.org -- compiled "__DATE__;
@@ -468,13 +468,20 @@
 			background = backlight_err;
 		} else {
 			dispflags |= WN_DISP_WARN;
-			background = backlight_down;
+			background = backlight_down_off;
+			if (dispflags & WN_DISP_BACKLIT)
+				background = backlight_down_on;
 		}
 		break;
 
 	case IF_DOWN:
 		dispflags |= WN_DISP_WARN;
 		background = backlight_down;
+		if (options[OPT_KEEP].used) {
+			background = backlight_down_off;
+			if (dispflags & WN_DISP_BACKLIT)
+				background = backlight_down_on;
+		}
 		break;
 
 	case IF_UP:
@@ -559,7 +566,7 @@
 	if (dispflags & WN_DISP_BACKLIT)
 		syoff = WN_FONT_YOFF + WN_FONT_HEIGHT;
 
-	if (dispflags & WN_DISP_WARN)
+	if ((!options[OPT_KEEP].used) && (dispflags & WN_DISP_WARN))
 		syoff = WN_FONT_YOFF + (2 * WN_FONT_HEIGHT);
 
 	/*
@@ -1028,6 +1035,8 @@
 		{ backlight_down_xpm,	&backlight_down	},
 		{ parts_xpm,		&parts		},
 		{ font_xpm,		&font 		},
+		{ backlight_down_on_xpm, &backlight_down_on},
+		{ backlight_down_off_xpm, &backlight_down_off},
 		{ NULL }
 	};
 	unsigned int h, w, i;
diff -Naur wmnetload-1.3/xpm/current/backlight_down_off.xpm wmnetload-1.3-tn2/xpm/current/backlight_down_off.xpm
--- wmnetload-1.3/xpm/current/backlight_down_off.xpm	Thu Jan  1 01:00:00 1970
+++ wmnetload-1.3-tn2/xpm/current/backlight_down_off.xpm	Wed Jun 11 18:08:59 2003
@@ -0,0 +1,70 @@
+/* XPM */
+static char * backlight_down_off_xpm[] = {
+"58 58 9 1",
+" 	c None",
+".	c #020202",
+"+	c #0A0A0A",
+"@	c #484C46",
+"#	c #DEDEDE",
+"$	c #7C827C",
+"%	c #8E968A",
+"&	c #767C6F",
+"*	c #3E463E",
+"  ......................................................  ",
+" .++++++++++++++++++++++++++++++++++++++++++++++++++++++. ",
+".++@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@.#",
+".+@@$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$.#",
+".+@$$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%%%%%&&&&&&%%%%%%%%&&&&&&%%%%%%%%&&&&&&%%%%%%%%%%%%.#",
+".+@$%%%%&%%&&&&%%&%%%%&%%&&&&%%.%%%%&%%&&&&%%&%%%&&%&&%%.#",
+".+@$%%%%&&%%%%%%&&%%%%&&%%%%%%..%%%%&&%%%%%%&&%%%&&%&&%%.#",
+".+@$%%%%&&%%%%%%&&%%%%&&%%%%%%..%%%%&&%%%%%%&&%%%&&&%%%%.#",
+".+@$%%%%&&%%%%%%&&%%%%&&%%%%%%..%%%%&&%%%%%%&&%%%&&%&&%%.#",
+".+@$%%%%&&%%%%%%&&%%%%&&%%%%%%..%%%%&&%%%%%%&&%%%&&%&&%%.#",
+".+@$%%%%&&%%%%%%&&%%%%&&%%%%%%..%%%%&&%%%%%%&&%%%%%%%%%%.#",
+".+@$%%%%&&%%%%%%&&%%%%&&%%%%%%..%%%%&&%%%%%%&&%%%%%%%%%%.#",
+".+@$%%%%&&%%%%%%&&%%%%&&%%%%%%..%%%%&&%%%%%%&&%%%&&%&&%%.#",
+".+@$%%%%&%%&&&&%%&%%%%&%%....%%.%%%%&%%....%%&%%%&&%&&%%.#",
+".+@$%%%%%%&&&&&&%%%%%%%%......%%%%%%%%......%%%%%&&&&&%%.#",
+".+@$%%%%&%%%%%%%%&%%%%.%%%%%%%%.%%%%.%%%%%%%%.%%%&%&%&%%.#",
+".+@$%%%%&&%%%%%%&&%%%%..%%%%%%..%%%%..%%%%%%..%%%&%%%&%%.#",
+".+@$%%%%&&%%%%%%&&%%%%..%%%%%%..%%%%..%%%%%%..%%%%%%%%%%.#",
+".+@$%%%%&&%%%%%%&&%%%%..%%%%%%..%%%%..%%%%%%..%%%%%%%%%%.#",
+".+@$%%%%&&%%%%%%&&%%%%..%%%%%%..%%%%..%%%%%%..%%%&&&&&%%.#",
+".+@$%%%%&&%%%%%%&&%%%%..%%%%%%..%%%%..%%%%%%..%%%&&%%%%%.#",
+".+@$%%%%&&%%%%%%&&%%%%..%%%%%%..%%%%..%%%%%%..%%%&&%&&%%.#",
+".+@$%&&%&%%&&&&%%&%&&%.%%....%%.%&&%.%%&&&&%%.%%%&&%&&%%.#",
+".+@$%&&%%%&&&&&&%%%&&%%%......%%%&&%%%&&&&&&%%%%%&&&&&%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%*************************************************%%.#",
+".+@$%*************************************************&%.#",
+".+@$%%&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%.#",
+".+@$%%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%&&%%%.#",
+".+@$%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%..#",
+" .......................................................# ",
+"  ######################################################  "};
diff -Naur wmnetload-1.3/xpm/current/backlight_down_on.xpm wmnetload-1.3-tn2/xpm/current/backlight_down_on.xpm
--- wmnetload-1.3/xpm/current/backlight_down_on.xpm	Thu Jan  1 01:00:00 1970
+++ wmnetload-1.3-tn2/xpm/current/backlight_down_on.xpm	Wed Jun 11 18:06:47 2003
@@ -0,0 +1,72 @@
+/* XPM */
+static char *backlight_down_on_xpm[] = {
+/* width height num_colors chars_per_pixel */
+"    58    58        7            1",
+/* colors */
+". c None",
+"# c #020202",
+"a c #6ec63b s Back0",
+"b c #dedede",
+"c c #6cb237 s Back1",
+"d c #3e463e",
+"e c #020202",
+/* pixels */
+"..######################################################..",
+".#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa#.",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaaaaaccccccaaaaaaaaccccccaaaaaaaaccccccaaaaaaaaaaaaab",
+"#aaaaaaacaaccccaacaaaacaaccccaaeaaaacaaccccaacaaaccaccaaab",
+"#aaaaaaaccaaaaaaccaaaaccaaaaaaeeaaaaccaaaaaaccaaaccaccaaab",
+"#aaaaaaaccaaaaaaccaaaaccaaaaaaeeaaaaccaaaaaaccaaacccaaaaab",
+"#aaaaaaaccaaaaaaccaaaaccaaaaaaeeaaaaccaaaaaaccaaaccaccaaab",
+"#aaaaaaaccaaaaaaccaaaaccaaaaaaeeaaaaccaaaaaaccaaaccaccaaab",
+"#aaaaaaaccaaaaaaccaaaaccaaaaaaeeaaaaccaaaaaaccaaaaaaaaaaab",
+"#aaaaaaaccaaaaaaccaaaaccaaaaaaeeaaaaccaaaaaaccaaaaaaaaaaab",
+"#aaaaaaaccaaaaaaccaaaaccaaaaaaeeaaaaccaaaaaaccaaaccaccaaab",
+"#aaaaaaacaaccccaacaaaacaaeeeeaaeaaaacaaeeeeaacaaaccaccaaab",
+"#aaaaaaaaaccccccaaaaaaaaeeeeeeaaaaaaaaeeeeeeaaaaacccccaaab",
+"#aaaaaaacaaaaaaaacaaaaeaaaaaaaaeaaaaeaaaaaaaaeaaacacacaaab",
+"#aaaaaaaccaaaaaaccaaaaeeaaaaaaeeaaaaeeaaaaaaeeaaacaaacaaab",
+"#aaaaaaaccaaaaaaccaaaaeeaaaaaaeeaaaaeeaaaaaaeeaaaaaaaaaaab",
+"#aaaaaaaccaaaaaaccaaaaeeaaaaaaeeaaaaeeaaaaaaeeaaaaaaaaaaab",
+"#aaaaaaaccaaaaaaccaaaaeeaaaaaaeeaaaaeeaaaaaaeeaaacccccaaab",
+"#aaaaaaaccaaaaaaccaaaaeeaaaaaaeeaaaaeeaaaaaaeeaaaccaaaaaab",
+"#aaaaaaaccaaaaaaccaaaaeeaaaaaaeeaaaaeeaaaaaaeeaaaccaccaaab",
+"#aaaaccacaaccccaacaccaeaaeeeeaaeaccaeaaccccaaeaaaccaccaaab",
+"#aaaaccaaaccccccaaaccaaaeeeeeeaaaccaaaccccccaaaaacccccaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaadddddddddddddddddddddddddddddddddddddddddddddddddaaab",
+"#aaaadddddddddddddddddddddddddddddddddddddddddddddddddcaab",
+"#aaaaacccccccccccccccccccccccccccccccccccccccccccccccccaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+"#aaaaaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaccaaaab",
+"#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab",
+".#aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab.",
+"..bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb.."
+};
diff -Naur wmnetload-1.3/xpm/current/pixmaps.h wmnetload-1.3-tn2/xpm/current/pixmaps.h
--- wmnetload-1.3/xpm/current/pixmaps.h	Sun Oct 27 06:11:49 2002
+++ wmnetload-1.3-tn2/xpm/current/pixmaps.h	Wed Jun 11 17:59:32 2003
@@ -29,6 +29,8 @@
 #include "../common/pixmaps.h"
 #include "./parts.xpm"
 #include "./backlight_down.xpm"
+#include "./backlight_down_on.xpm"
+#include "./backlight_down_off.xpm"
 #include "./backlight_err.xpm"
 #include "./backlight_on.xpm"
 #include "./backlight_off.xpm"
