diff -rupN ajenti-1.2.17.2/ajenti-panel ajenti-1.2.17.2p/ajenti-panel
--- ajenti-1.2.17.2/ajenti-panel	2014-02-17 11:19:41.000000000 +0100
+++ ajenti-1.2.17.2p/ajenti-panel	2014-02-27 21:25:46.929343578 +0100
@@ -82,6 +82,7 @@ if __name__ == '__main__':
             pidfile=PidFile('/var/run/ajenti.pid'),
             stdout=open('/dev/null', 'w+'),
             stderr=open('/dev/null', 'w+'),
+            detach_process=True
         )
         with context:
             ajenti.log.init_log_rotation()
