Index: kabikaboo-1.7.new/kabikaboo
===================================================================
--- kabikaboo-1.7.orig/kabikaboo
+++ kabikaboo-1.7.new/kabikaboo
@@ -1,2 +1,2 @@
-#!/bin/bash
-python /usr/share/kabikaboo/src/kabikaboo.py
+#!/bin/sh
+exec python2 /usr/share/kabikaboo/src/kabikaboo.py
Index: kabikaboo-1.7.new/src/kabikaboo.py
===================================================================
--- kabikaboo-1.7.orig/src/kabikaboo.py
+++ kabikaboo-1.7.new/src/kabikaboo.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 #
 # Kabikaboo
@@ -2961,11 +2961,8 @@ Kabikaboo. If not, see http://www.gnu.or
         result = False
         # WARNING: LINUX ONLY CODE
         output = commands.getoutput('ps -ax')
-        # check for 2 or more regular instances
-        if output.count('python src/kabikaboo.py') >= 2:
-            result = True
-        # check for instance and development instance
-        if output.count('python src/kabikaboo.py') >= 1 and output.count('python -W ignore::DeprecationWarning src/kabikaboo.py') >= 1:
+        # check for 2 or more instances
+        if output.count('src/kabikaboo.py') >= 2:
             result = True
         if result:
             print 'Duplicate instance of Kabikaboo already running.'
