diff -Nru unity-greeter-13.04.0.orig/configure.ac unity-greeter-13.04.0/configure.ac
--- unity-greeter-13.04.0.orig/configure.ac	2013-01-08 05:31:42.971306181 -0500
+++ unity-greeter-13.04.0/configure.ac	2013-01-08 05:31:59.164554678 -0500
@@ -7,7 +7,7 @@
 AM_INIT_AUTOMAKE
 AM_PROG_CC_C_O
 UG_PROG_VALAC([valac-0.18 valac-0.16 valac], [0.15.1])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_MAINTAINER_MODE
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES(yes)])
 
