--- src/textile/syntax/textile.vim	2013-06-30 17:03:28.342420303 -0400
+++ textile.vim	2013-06-30 17:00:24.261187629 -0400
@@ -13,14 +13,13 @@
 "   tim.theenchanter.com
 " @(#) $Id$
 
-if version < 600
-    syntax clear
-elseif exists("b:current_syntax")
+if exists("b:current_syntax") && !exists("g:syntax_debug")
     finish
 endif
 
 " Textile commands like "h1" are case sensitive, AFAIK.
 syn case match
+syntax sync fromstart
 
 " Textile syntax: <http://textism.com/tools/textile/>
 
@@ -84,4 +83,6 @@
     delcommand HiLink
 endif
 
+let b:current_syntax = "textile"
+
 " vim: set ai et sw=4 :
