--- src/misc/setting.cpp.orig	2011-05-08 16:39:42.186082022 +0200
+++ src/misc/setting.cpp	2011-05-08 16:41:15.586952792 +0200
@@ -1940,12 +1940,6 @@
     switch(type) {
     case NAME:
       break;
-    case PUBLIC_DATA_DIRECTORY:
-    case PRIVATE_DATA_DIRECTORY:
-
-      Translator::dir_scan();
-      ::translator.load();
-      break;
     case LANGUAGE:
       ::translator.load();
       break;
