--- goldendict.pro1	2010-01-18 20:58:47.000000000 +0300
+++ goldendict.pro	2010-01-18 21:03:10.492722785 +0300
@@ -41,10 +41,10 @@ unix { 
     LIBS += -lX11 \
         -lXtst
     PREFIX = $$(PREFIX)
-    isEmpty( PREFIX ):PREFIX = /usr/local
-    DEFINES += PROGRAM_DATA_DIR=\\\"$$PREFIX/share/apps/goldendict/\\\"
+    isEmpty( PREFIX ):PREFIX = /usr/
+    DEFINES += PROGRAM_DATA_DIR=\\\"$$PREFIX/share/goldendict/\\\"
     target.path = $$PREFIX/bin/
-    locale.path = $$PREFIX/share/apps/goldendict/locale/
+    locale.path = $$PREFIX/share/goldendict/locale/
     locale.files = locale/*.qm
     INSTALLS += target \
         locale
