--- old/ocrorast/ocr-reading-order.cc	Wed Feb 29 20:01:08 2012 +0100
+++ new/ocrorast/ocr-reading-order.cc	Fri Jun 08 14:15:31 2012 +0200
@@ -175,7 +175,7 @@
 
     // Make dummy text-lines from horizontal rulings to use them as
     // separating elements for reading order.
-    for(int i=0,l=hor_rulings.length(); i<l; i++){
+    for(int i=0,b=hor_rulings.length(); i<b; i++){
         line l;
         rectangle r = hor_rulings[i];
         l.c = r.ycenter();
--- old/ocrorast/setup.py	2012-06-08 14:18:18.351516465 +0200
+++ new/ocrorast/setup.py	2012-06-13 11:27:23.788810080 +0200
@@ -6,6 +6,9 @@
 
 include_dirs = ['/usr/local/include'] + get_numpy_include_dirs()
 swig_opts = ["-c++"] + ["-I" + d for d in include_dirs]
+env_swig_opts = os.getenv('SWIG_OPTS')
+if env_swig_opts is not None:
+    swig_opts.append(env_swig_opts)
 
 ocrorast = Extension('_ocrorast',
                      library_dirs = ['.'],
