From: Debian CLI Applications Team <pkg-cli-apps-team@lists.alioth.debian.org>
Date: Sun, 10 Apr 2011 15:46:08 +0200
Subject: 04_no_sgen

---
 KeePass/KeePass.csproj |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/KeePass/KeePass.csproj b/KeePass/KeePass.csproj
index 6fe95b3..f6db9c6 100644
--- a/KeePass/KeePass.csproj
+++ b/KeePass/KeePass.csproj
@@ -1186,7 +1186,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-  <PropertyGroup>
-    <PostBuildEvent>"$(FrameworkSDKDir)bin\sgen.exe" /assembly:"$(TargetPath)" /force /nologo /compiler:/keycontainer:VS_KEY_E6D8E2B3F89DA59B /compiler:/delaysign-</PostBuildEvent>
-  </PropertyGroup>
-</Project>
\ No newline at end of file
+</Project>
-- 
