--- src/enemylines5-1.0/c3_tpl.h	2006-01-13 09:45:40.000000000 +0100
+++ c3_tpl-new.h	2009-03-01 18:27:28.000000000 +0100
@@ -2,6 +2,7 @@
 #include <sstream>
 #include <utility>
 #include <cmath>
+#include <limits>
 
 
 using namespace std::rel_ops;
--- src/enemylines5-1.0/random.cc	2006-01-13 09:45:40.000000000 +0100
+++ random-new.cc	2009-03-01 18:26:46.000000000 +0100
@@ -1,5 +1,5 @@
 #include "random.h"
-
+#include <cstdlib>
 #include <iostream>
 
 
--- src/enemylines5-1.0/Makefile	2006-01-13 09:45:42.000000000 +0100
+++ Makefile-new	2009-03-01 18:58:43.000000000 +0100
@@ -23,9 +23,8 @@
 	rm *.o $(TARGET)
 
 install: 
-	mkdir /usr/local/share/enemylines5/
-	cp enemylines5 /usr/local/bin/
-	cp data/* /usr/local/share/enemylines5/
+	cp enemylines5 $(DESTDIR)/usr/bin/
+	cp data/* $(DESTDIR)/usr/share/enemylines5/
 
 installpkg:
 	mkdir /usr/share/enemylines5/
