--- opt/brother/Printers/DCP1510/inf/setupPrintcap.old
+++ opt/brother/Printers/DCP1510/inf/setupPrintcap
@@ -19,7 +19,7 @@
 #
 
 BRPRINTER_NAME=$1
-PRINTCAP_NAME=/etc/printcap.local
+PRINTCAP_NAME=/etc/printcap
 SPOOLER_NAME=/var/spool/lpd/$BRPRINTER_NAME
 DEVICE_IF=/dev/lp0
 
@@ -32,7 +32,7 @@
 	if [ ! -d $SPOOLER_NAME ]; then
 	        mkdir $SPOOLER_NAME
 	fi
-	chown lp $SPOOLER_NAME
+	chown root $SPOOLER_NAME
 	chgrp lp $SPOOLER_NAME
 	chmod 700 $SPOOLER_NAME
 
