diff -rup povray-3.7.0.RC2/Makefile.am povray-3.7.0.RC2.new//Makefile.am
--- povray-3.7.0.RC2/Makefile.am	2011-01-06 23:40:19.000000000 +0100
+++ povray-3.7.0.RC2.new//Makefile.am	2011-01-08 10:54:21.027908186 +0100
@@ -107,19 +107,8 @@ install-data-local:
 	for f in $$filelist ; do \
 	  $(INSTALL_DATA) $(top_srcdir)/doc/$$f $(DESTDIR)$(povdocdir)/$$f && echo "$(DESTDIR)$(povdocdir)/$$f" >> $(povinstall); \
 	done
-	@echo "Creating user directories..."; \
-	for p in $(povuser) $(povconfuser) ; do \
-	  $(mkdir_p) $$p && chown $(povowner) $$p && chgrp $(povgroup) $$p && printf "%s\n" "$$p" "`cat $(povinstall)`" > $(povinstall); \
-	done
-	@echo "Copying user configuration and INI files..."; \
-	for f in povray.conf povray.ini ; do \
-	  if test -f $(povconfuser)/$$f; then \
-	    echo "Creating backup of $(povconfuser)/$$f"; \
-	    mv -f $(povconfuser)/$$f $(povconfuser)/$$f.bak; \
-	  fi; \
-	done; \
-	$(INSTALL_DATA) $(top_srcdir)/povray.conf $(povconfuser)/povray.conf && chown $(povowner) $(povconfuser)/povray.conf && chgrp $(povgroup) $(povconfuser)/povray.conf  && echo "$(povconfuser)/povray.conf" >> $(povinstall); \
-	$(INSTALL_DATA) $(top_builddir)/povray.ini $(povconfuser)/povray.ini && chown $(povowner) $(povconfuser)/povray.ini && chgrp $(povgroup) $(povconfuser)/povray.ini  && echo "$(povconfuser)/povray.ini" >> $(povinstall)
+#	$(INSTALL_DATA) $(top_srcdir)/povray.conf $(povconfuser)/povray.conf && chown $(povowner) $(povconfuser)/povray.conf && chgrp $(povgroup) $(povconfuser)/povray.conf  && echo "$(povconfuser)/povray.conf" >> $(povinstall); \
+#	$(INSTALL_DATA) $(top_builddir)/povray.ini $(povconfuser)/povray.ini && chown $(povowner) $(povconfuser)/povray.ini && chgrp $(povgroup) $(povconfuser)/povray.ini  && echo "$(povconfuser)/povray.ini" >> $(povinstall)
 
 # Remove data, config, and empty folders for 'make uninstall'.
 # Use 'hook' instead of 'local' so as to properly remove *empty* folders (e.g. scripts).
diff -rup povray-3.7.0.RC2/Makefile.in povray-3.7.0.RC2.new//Makefile.in
--- povray-3.7.0.RC2/Makefile.in	2011-01-06 23:40:24.000000000 +0100
+++ povray-3.7.0.RC2.new//Makefile.in	2011-01-08 10:53:59.658482647 +0100
@@ -1038,19 +1038,8 @@ install-data-local:
 	for f in $$filelist ; do \
 	  $(INSTALL_DATA) $(top_srcdir)/doc/$$f $(DESTDIR)$(povdocdir)/$$f && echo "$(DESTDIR)$(povdocdir)/$$f" >> $(povinstall); \
 	done
-	@echo "Creating user directories..."; \
-	for p in $(povuser) $(povconfuser) ; do \
-	  $(mkdir_p) $$p && chown $(povowner) $$p && chgrp $(povgroup) $$p && printf "%s\n" "$$p" "`cat $(povinstall)`" > $(povinstall); \
-	done
-	@echo "Copying user configuration and INI files..."; \
-	for f in povray.conf povray.ini ; do \
-	  if test -f $(povconfuser)/$$f; then \
-	    echo "Creating backup of $(povconfuser)/$$f"; \
-	    mv -f $(povconfuser)/$$f $(povconfuser)/$$f.bak; \
-	  fi; \
-	done; \
-	$(INSTALL_DATA) $(top_srcdir)/povray.conf $(povconfuser)/povray.conf && chown $(povowner) $(povconfuser)/povray.conf && chgrp $(povgroup) $(povconfuser)/povray.conf  && echo "$(povconfuser)/povray.conf" >> $(povinstall); \
-	$(INSTALL_DATA) $(top_builddir)/povray.ini $(povconfuser)/povray.ini && chown $(povowner) $(povconfuser)/povray.ini && chgrp $(povgroup) $(povconfuser)/povray.ini  && echo "$(povconfuser)/povray.ini" >> $(povinstall)
+#	$(INSTALL_DATA) $(top_srcdir)/povray.conf $(povconfuser)/povray.conf && chown $(povowner) $(povconfuser)/povray.conf && chgrp $(povgroup) $(povconfuser)/povray.conf  && echo "$(povconfuser)/povray.conf" >> $(povinstall); \
+#	$(INSTALL_DATA) $(top_builddir)/povray.ini $(povconfuser)/povray.ini && chown $(povowner) $(povconfuser)/povray.ini && chgrp $(povgroup) $(povconfuser)/povray.ini  && echo "$(povconfuser)/povray.ini" >> $(povinstall)
 
 # Remove data, config, and empty folders for 'make uninstall'.
 # Use 'hook' instead of 'local' so as to properly remove *empty* folders (e.g. scripts).
 
