diff -Naur data.org/usr/local/Brother/Printer/mfc885cw/lpd/filtermfc885cw data/usr/local/Brother/Printer/mfc885cw/lpd/filtermfc885cw
--- data.org/usr/local/Brother/Printer/mfc885cw/lpd/filtermfc885cw	2008-11-08 13:26:37.000000000 +0100
+++ data/usr/local/Brother/Printer/mfc885cw/lpd/filtermfc885cw	2008-11-08 13:46:24.000000000 +0100
@@ -21,7 +21,7 @@
 PRINTER="mfc885cw"
 PRINTER_TMP=$(echo $PRINTER | sed -e 's/ //g' -e 's/-//g')
 
-BR_PRT_PATH=/usr/local/Brother/Printer/${PRINTER}
+BR_PRT_PATH=/usr/share/brother/printer/${PRINTER}
 RCFILE=`eval echo $BR_PRT_PATH/inf/brPRINTERrc | eval sed 's/PRINTER/"$PRINTER_TMP"/'`
 PAPER_INF=$BR_PRT_PATH/inf/paperinfij2
 PSCONV=$BR_PRT_PATH/lpd/psconvertij2
