diff -r bd9e0af3a6d5 libpam/Makefile
--- a/libpam/Makefile	Tue Dec 07 11:49:07 2010 -0800
+++ b/libpam/Makefile	Fri Feb 11 19:10:34 2011 +0100
@@ -35,10 +35,10 @@
 	$(RM) *.o *.so google-authenticator pam_google_authenticator_unittest
 
 google-authenticator: google-authenticator.o base32.o hmac.o sha1.o
-	$(CC) -g $(LDFLAGS) -ldl -o $@ $+
+	$(CC) -g $(LDFLAGS) -o $@ $+ -ldl
 
 pam_google_authenticator_unittest: pam_google_authenticator_unittest.o base32.o
-	$(CC) -g $(LDFLAGS) -rdynamic -ldl -lc -o $@ $+
+	$(CC) -g $(LDFLAGS) -rdynamic -lc -o $@ $+ -ldl
 
 pam_google_authenticator.so: base32.o hmac.o sha1.o
 pam_google_authenticator_testing.so: base32.o hmac.o sha1.o
