diff -aur cmunipack-2.0.4/src/muniwin/progress_dlg.h cmunipack-2.0.4-new/src/muniwin/progress_dlg.h
--- cmunipack-2.0.4/src/muniwin/progress_dlg.h	2012-08-12 19:00:12.000000000 +0200
+++ cmunipack-2.0.4-new/src/muniwin/progress_dlg.h	2012-10-13 12:58:30.713397816 +0200
@@ -24,7 +24,7 @@
 #define CMPACK_PROGRESS_DLG_H
 
 #include <gtk/gtk.h>
-#include <glib/gthread.h>
+#include <glib.h>
 
 #include "helper_classes.h"
 
Nur in cmunipack-2.0.4-new/src/muniwin: progress_dlg.h~.
diff -aur cmunipack-2.0.4/src/muniwin/utils.cpp cmunipack-2.0.4-new/src/muniwin/utils.cpp
--- cmunipack-2.0.4/src/muniwin/utils.cpp	2012-10-02 17:36:36.000000000 +0200
+++ cmunipack-2.0.4-new/src/muniwin/utils.cpp	2012-10-13 12:58:13.800064183 +0200
@@ -905,7 +905,7 @@
 		g_free(lock_path);
 		g_free(basedir);
 		g_free(name);
-		m_File = open(p, O_RDWR | O_CREAT);
+		m_File = open(p, O_RDWR | O_CREAT, S_IRWXU);
 		if (m_File != -1) {
 			fl.l_type = F_WRLCK;
 			fl.l_whence = SEEK_SET;
Nur in cmunipack-2.0.4-new/src/muniwin: utils.cpp~.
