From f037809907bcc51295a471ad66b14e90f22d8caa Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Wed, 13 Feb 2013 19:49:41 +0100
Subject: [PATCH] help: Fix markup of heading "USB options" so it appears in -help

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Reviewed-by: Stefan Hajnoczi <stefanha@redhat.com>
Message-id: 1360781383-28635-6-git-send-email-armbru@redhat.com
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
---
 qemu-options.hx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/qemu-options.hx b/qemu-options.hx
index 8c435cd..ce9602f 100644
--- qemu-options.hx
+++ qemu-options.hx
@@ -538,9 +538,10 @@ ETEXI
 STEXI
 @end table
 ETEXI
+DEFHEADING()
 
+DEFHEADING(USB options:)
 STEXI
-USB options:
 @table @option
 ETEXI
 
-- 
1.7.0.4

