--- src/readers/FbxConverter.h.orig	2014-02-14 15:34:50.000000000 +0100
+++ src/readers/FbxConverter.h	2014-02-14 15:34:45.000000000 +0100
@@ -577,7 +577,7 @@
 			Animation *animation = new Animation();
 			model->animations.push_back(animation);
 			animation->id = animStack->GetName();
-			animStack->GetScene()->GetEvaluator()->SetContext(animStack);
+			animStack->GetScene()->SetCurrentAnimationStack(animStack);
 
 			// Add the NodeAnimations to the Animation
 			for (std::map<FbxNode *, AnimInfo>::const_iterator itr = affectedNodes.begin(); itr != affectedNodes.end(); itr++) {
