Index: cblrepo/ListLike-4.1.0/ListLike.cabal
===================================================================
--- cblrepo.orig/ListLike-4.1.0/ListLike.cabal
+++ cblrepo/ListLike-4.1.0/ListLike.cabal
@@ -48,10 +48,10 @@ Library
                 ,containers >= 0.3   && < 0.6
                 ,bytestring >= 0.9.1 && < 0.11
                 ,array      >= 0.3   && < 0.6
-                ,text       >= 0.11  && < 1.2
+                ,text       >= 0.11
                 ,vector     >= 0.5   && < 0.11
                 ,dlist      >= 0.7   && < 0.9
-                ,fmlist     >= 0.8   && < 0.9
+                ,fmlist     >= 0.8
   
 Test-suite listlike-tests
   Hs-source-dirs: src testsrc
