diff -aur gmtp.pristine/gMTP/Makefile gmtp.new/gMTP/Makefile
--- gmtp.pristine/gMTP/Makefile	2012-02-19 22:06:39.000000000 -0600
+++ gmtp.new/gMTP/Makefile	2012-04-25 19:34:25.423664794 -0500
@@ -46,8 +46,8 @@
 GTK_LDFLAGS = `pkg-config --libs gtk+-2.0 gconf-2.0 libmtp id3tag flac vorbisfile`
 
 ifeq ($(MAKECMDGOALS),gtk3)
-GTK_CFLAGS = `pkg-config --cflags gtk+-3.0 gio-2.0 libmtp id3tag flac vorbisfile`
-GTK_LDFLAGS = `pkg-config --libs gtk+-3.0 gio-2.0 libmtp id3tag flac vorbisfile`
+GTK_CFLAGS = `pkg-config --cflags gtk+-3.0 gio-2.0 libmtp id3tag flac vorbisfile gthread-2.0`
+GTK_LDFLAGS = `pkg-config --libs gtk+-3.0 gio-2.0 libmtp id3tag flac vorbisfile gthread-2.0`
 CFLAGS += -DGMTP_USE_GTK3
 endif
 
