Author: SpepS <dreamspepser at yahoo dot it>
Description: Gcc 4.7 fixes
diff -aur din-3.0/include/box.h din-3.0.patched/include/box.h
--- din-3.0/include/box.h	2012-02-02 21:57:56.000000000 +0100
+++ din-3.0.patched/include/box.h	2012-04-18 02:35:22.670111777 +0200
@@ -21,6 +21,8 @@
 #ifndef __box
 #define __box
 
+#include <algorithm>
+
 template <typename T> struct box {
   T left, bottom;
   T right, top;
@@ -47,8 +49,8 @@
   }
 
   inline void calc () {
-    if (left > right) swap (left, right);
-    if (bottom > top) swap (bottom, top);
+    if (left > right) std::swap (left, right);
+    if (bottom > top) std::swap (bottom, top);
     width = right - left;
     height = top - bottom;
     midx = left + width / 2;
diff -aur din-3.0/include/main.h din-3.0.patched/include/main.h
--- din-3.0/include/main.h	2011-10-31 07:48:33.000000000 +0100
+++ din-3.0.patched/include/main.h	2012-04-18 02:30:44.302473555 +0200
@@ -2,6 +2,7 @@
 #define __main
 
 #include <string>
+#include <unistd.h>
 
 void accept_keypress (int accept);
 void warp_pointer (int dx, int dy);
diff -aur din-3.0/src/tcl_interp.cc din-3.0.patched/src/tcl_interp.cc
--- din-3.0/src/tcl_interp.cc	2012-02-09 10:17:24.000000000 +0100
+++ din-3.0.patched/src/tcl_interp.cc	2012-04-18 02:41:38.052583294 +0200
@@ -25,6 +25,7 @@
 #include "ansi_color_codes.h"
 #include <iostream>
 #include <stdlib.h>
+#include <stdint.h>
 
 using namespace std;
 
