--- dwarftherapist/src/gamedatareader.cpp	2010-06-13 09:54:30.543009571 +0200
+++ dwarftherapist-build/src/gamedatareader.cpp	2010-06-13 10:43:53.896984624 +0200
@@ -34,7 +34,7 @@
     : QObject(parent)
 {
     QDir working_dir = QDir::current();
-    QString filename = working_dir.absoluteFilePath("etc/game_data.ini");
+    QString filename = "/usr/share/dwarftherapist/game_data.ini";
     m_data_settings = new QSettings(filename, QSettings::IniFormat);
 
     QStringList labor_names;
--- dwarftherapist/src/dfinstance.cpp	2010-06-28 07:51:20.425550489 +0200
+++ dwarftherapist-build/src/dfinstance.cpp	2010-06-28 07:53:39.235120590 +0200
@@ -72,7 +72,7 @@
 #endif
 #endif
 #endif
-    search_paths << QString("etc/memory_layouts/%1").arg(subdir);
+    search_paths << QString("/usr/share/dwarftherapist/memory_layouts/%1").arg(subdir);
 
     TRACE << "Searching for MemoryLayout ini files in the following directories";
     foreach(QString path, search_paths) {
--- dwarftherapist/src/dwarftherapist.cpp	2010-07-31 22:04:58.291687394 +0200
+++ dwarftherapist-build/src/dwarftherapist.cpp	2010-07-31 22:06:35.031276564 +0200
@@ -91,7 +91,7 @@
 
     //setup logging
     m_log_mgr = new LogManager(this);
-    TruncatingFileLogger *log = m_log_mgr->add_logger("log/run.log");
+    TruncatingFileLogger *log = m_log_mgr->add_logger(QDir::tempPath() + "/dwarftherapist.log");
     if (log) {
         LogAppender *app = m_log_mgr->add_appender("core", log, LL_TRACE);
         if (app) {
