=== modified file 'NuxImage/NPng.cpp'
--- NuxImage/NPng.cpp	2010-12-06 06:09:44 +0000
+++ NuxImage/NPng.cpp	2010-12-09 20:41:42 +0000
@@ -253,7 +253,7 @@
     png_read_info (png_ptr, info_ptr);

     // get width, height, bit-depth and color-type
-    unsigned long width, height;
+    png_uint_32 width, height;
     png_get_IHDR (png_ptr, info_ptr, &width, &height, &bit_depth, &color_type, 0, 0, 0);

     // expand images of all color-type and bit-depth to 3x8 bit RGB images
@@ -463,7 +463,7 @@
     png_read_info (png_ptr, info_ptr);

     // get width, height, bit-depth and color-type
-    unsigned long w, h;
+    png_uint_32 w, h;
     png_get_IHDR (png_ptr, info_ptr, &w, &h, &bit_depth, &color_type, 0, 0, 0);

     // expand images of all color-type and bit-depth to 3x8 bit RGB images
