diff -urN librfsctl-0.0.0-orig/Makefile librfsctl-0.0.0/Makefile
--- librfsctl-0.0.0-orig/Makefile	2008-11-13 13:53:03.000000000 +0100
+++ librfsctl-0.0.0/Makefile	2009-02-21 14:56:07.944167967 +0100
@@ -26,13 +26,13 @@
 		-o $(LIB_NAME).so $(LIB_OBJS)
 
 install: $(LIB_NAME).a $(LIB_NAME).so
-	cp $(HDR_NAME) $(HDR_DIR)/$(HDR_NAME)
-	cp $(LIB_NAME).a $(LIB_DIR)/$(LIB_NAME).a
-	cp $(LIB_NAME).so $(LIB_DIR)/$(LIB_NAME).so.$(VMAR).$(VMIN).$(VREL)
+	cp $(HDR_NAME) $(DESTDIR)$(HDR_DIR)/$(HDR_NAME)
+	cp $(LIB_NAME).a $(DESTDIR)$(LIB_DIR)/$(LIB_NAME).a
+	cp $(LIB_NAME).so $(DESTDIR)$(LIB_DIR)/$(LIB_NAME).so.$(VMAR).$(VMIN).$(VREL)
 	ln -sf $(LIB_DIR)/$(LIB_NAME).so.$(VMAR).$(VMIN).$(VREL) \
-		$(LIB_DIR)/$(LIB_NAME).so.$(VMAR)
+		$(DESTDIR)$(LIB_DIR)/$(LIB_NAME).so.$(VMAR)
 	ln -sf $(LIB_DIR)/$(LIB_NAME).so.$(VMAR) \
-		$(LIB_DIR)/$(LIB_NAME).so
+		$(DESTDIR)$(LIB_DIR)/$(LIB_NAME).so
 
 uninstall:
 	$(RM) $(HDR_DIR)/$(HDR_NAME)
