--- usr/lib64/OpenCL/vendors/intel/iocgui64.sh	2012-03-21 21:59:56.000000000 +0100
+++ usr/lib64/OpenCL/vendors/intel/iocgui64.sh~	2012-05-24 00:59:15.410289185 +0200
@@ -53,13 +53,13 @@
 then
     # Get architecture supported on this machine
 	grep -q -e "^flags.*avx" /proc/cpuinfo
-	retval = $?
+	retval=$?
 	if [ $retval -eq 0 ];
 	then
 	    architecture="avx"
 	else
 	    grep -q -e "^flags.*sse4_2" /proc/cpuinfo
-		retval = $?
+		retval=$?
 	    if [ $retval -eq 0 ];
 		then
 		    architecture="sse4_2"
@@ -68,7 +68,7 @@
 
     # Add 32 bit support when available
     java -jar "$jarpath" x64 $architecture
-	retval = $?
+	retval=$?
     if [ $retval -ne 0 ];
     then
         reporterror "could not find java runtime environment"
