--- kwave-0.8.5.orig/plugins/sonagram/SonagramDialog.cpp
+++ kwave-0.8.5/plugins/sonagram/SonagramDialog.cpp
@@ -105,7 +105,7 @@ void SonagramDialog::parameters(QStringL
     list.clear();
 
     // parameter #0: number of fft points
-    param = pointbox ? pointbox->currentText() : QString(0);
+    param = pointbox ? pointbox->currentText() : QString();
     list.append(param);
 
     // parameter #1: index of the window function
--- kwave-0.8.5.orig/libgui/MenuItem.cpp
+++ kwave-0.8.5/libgui/MenuItem.cpp
@@ -38,7 +38,7 @@ MenuItem::MenuItem(MenuNode *parent,
                    const QKeySequence &shortcut,
                    const QString &uid)
     :MenuNode(parent, name, command, shortcut, uid),
-     m_exclusive_group(0), m_action(0)
+     m_action(0)
 {
     Q_ASSERT(parent);
     if (!parent) return;
--- kwave-0.8.5.orig/libgui/KwaveFileDialog.cpp
+++ kwave-0.8.5/libgui/KwaveFileDialog.cpp
@@ -34,7 +34,7 @@ KwaveFileDialog::KwaveFileDialog(const Q
     const QString &filter, QWidget *parent, bool modal,
     const QString last_url, const QString last_ext)
     :KFileDialog(startDir, filter, parent),
-     m_config_group(0), m_last_url(last_url), m_last_ext(last_ext)
+     m_last_url(last_url), m_last_ext(last_ext)
 {
     setModal(modal);
 
--- kwave-0.8.5.orig/libgui/KwaveFileDialog.h
+++ kwave-0.8.5/libgui/KwaveFileDialog.h
@@ -43,8 +43,8 @@ public:
      */
     KwaveFileDialog(const QString& startDir, const QString& filter,
                     QWidget *parent, bool modal,
-                    const QString last_url = 0,
-                    const QString last_ext = 0);
+                    const QString last_url = QString(),
+                    const QString last_ext = QString());
 
     /** Destructor */
     virtual ~KwaveFileDialog()
