--- setup.py	2013-04-26 11:57:31.162512453 +0100
+++ setup.py.fixed	2013-04-26 11:57:48.819319006 +0100
@@ -45,7 +45,7 @@
       author_email="inform@tiker.net",
       license = "MIT",
 
-      packages=["codepy", "codepy.cgen"],
+      packages=["codepy"],
       install_requires=[
           "pytools>=8",
           "cgen",

--- codepy/libraries.py	2009-12-23 01:47:42.000000000 +0000
+++ codepy/libraries.py.fixed	2013-04-26 11:22:48.146739824 +0100
@@ -95,7 +95,7 @@
     try:
         return aksetup["BOOST_%s_LIBNAME" % basename.upper()]
     except KeyError:
-        return ["boost_%s-%s-mt" % (basename, get_boost_compiler(aksetup))]
+        return ["boost_%s" % basename]

