--- Makefile	2006-08-25 01:29:30.000000000 +1000
+++ Makefile.arch	2009-02-07 18:26:15.000000000 +1100
@@ -15,14 +15,10 @@
 	pod2man odf2html.in > doc/odf2html.1
 
 install: 
-	mkdir -p $(PREFIX)/share/$(PACKAGE)
-	cp odfread  $(PREFIX)/bin/
-	cp odf2html $(PREFIX)/bin/
-	cp elinks.conf $(PREFIX)/share/$(PACKAGE)/
-	cp odt2html.xsl $(PREFIX)/share/$(PACKAGE)/
-	cp odt2html-fast.xsl $(PREFIX)/share/$(PACKAGE)/
-	cp doc/odfread.1 $(PREFIX)/share/man/man1/
-	cp doc/odf2html.1 $(PREFIX)/share/man/man1/
+	install -m 755 -d $(PREFIX)/bin $(PREFIX)/share/$(PACKAGE) $(PREFIX)/share/man/man1
+	install -m 755 -t $(PREFIX)/bin odfread odf2html
+	install -m 644 -t $(PREFIX)/share/$(PACKAGE) elinks.conf odt2html.xsl odt2html-fast.xsl
+	install -m 644 -t $(PREFIX)/share/man/man1/ doc/odfread.1 doc/odf2html.1
 
 clean:
 	rm odfread odf2html elinks.conf
