From 4db567738e76473ad7248134cb632dddf04dcb78 Mon Sep 17 00:00:00 2001
From: Jukka Rissanen <jukka.rissanen@linux.intel.com>
Date: Fri, 26 Aug 2011 15:10:51 +0300
Subject: [PATCH] service: Check NULL pointer when setting ipconfig

Fixes BMC#22766
---
 src/service.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/service.c b/src/service.c
index befbe37..6a68346 100644
--- a/src/service.c
+++ b/src/service.c
@@ -2732,6 +2732,9 @@ static int set_ipconfig(struct connman_service *service,
 	enum connman_ipconfig_type type;
 	int err;
 
+	if (ipconfig == NULL)
+		return -EINVAL;
+
 	old_method = __connman_ipconfig_get_method(ipconfig);
 
 	if (is_connecting_state(service, state) ||
-- 
1.7.6.1

