From 192b4476be18e482f0f1b2f454e1bf542660558c Mon Sep 17 00:00:00 2001
From: Patrik Flykt <patrik.flykt@linux.intel.com>
Date: Fri, 20 Apr 2012 13:47:18 +0300
Subject: [PATCH 07/11] main: Set default values if main.conf is not found

---
 src/main.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/main.c b/src/main.c
index 61e35cb..c0e87b3 100644
--- a/src/main.c
+++ b/src/main.c
@@ -148,8 +148,11 @@ static void parse_config(GKeyFile *config)
 		"cellular",
 	};
 
-	if (config == NULL)
+	if (config == NULL) {
+		connman_settings.auto_connect =
+			parse_service_types(default_auto_connect, 3);
 		return;
+	}
 
 	DBG("parsing main.conf");
 
@@ -462,10 +465,9 @@ int main(int argc, char *argv[])
 	__connman_dbus_init(conn);
 
 	config = load_config(CONFIGDIR "/main.conf");
-	if (config != NULL) {
-		parse_config(config);
+	parse_config(config);
+	if (config != NULL)
 		g_key_file_free(config);
-	}
 
 	__connman_storage_migrate();
 	__connman_technology_init();
-- 
1.7.10

