From 749e99d462f9a3deed7558ae99d472e865f4c7dc Mon Sep 17 00:00:00 2001
From: Patrik Flykt <patrik.flykt@linux.intel.com>
Date: Fri, 20 Apr 2012 14:54:27 +0300
Subject: [PATCH 10/11] main: Mark end of array with NULL in order not to
 crash

---
 src/main.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main.c b/src/main.c
index c0e87b3..70d5bb7 100644
--- a/src/main.c
+++ b/src/main.c
@@ -146,6 +146,7 @@ static void parse_config(GKeyFile *config)
 		"wifi",
 		"ethernet",
 		"cellular",
+		NULL
 	};
 
 	if (config == NULL) {
-- 
1.7.10

