--- pcapy-0.10.5.orig/pcapobj.cc	2008-10-12 19:00:52.000000000 +0100
+++ pcapy-0.10.5/pcapobj.cc	2008-10-12 18:59:31.000000000 +0100
@@ -209,8 +209,11 @@
 
   if(!buf)
     {
-      PyErr_SetString(PcapError, pcap_geterr(pp->pcap));
-      return NULL;
+      // According to Pcap manpage, there is no way to determine whether
+      // an error occured or not when calling pcap_next, so just returns
+      // None
+      Py_INCREF(Py_None);
+      Py_None;
     }
 
   PyObject *pkthdr = new_pcap_pkthdr(&hdr);
