--- lxmusic-notify.c    2011-08-10 22:56:34.645110148 +0200
+++ lxmusic-notify.c2   2011-08-10 22:30:42.250164660 +0200
@@ -57,11 +57,9 @@

	lxmusic_clear_notify (n);

-	notify = notify_notification_new (summary, message,
-	                                  "lxmusic", NULL);
+	notify = notify_notification_new (summary, message, "lxmusic");
	n->notification = notify;

-	notify_notification_attach_to_status_icon (notify, n->status_icon);
	notify_notification_set_urgency (notify, NOTIFY_URGENCY_NORMAL);
	notify_notification_set_timeout (notify, NOTIFY_EXPIRES_DEFAULT);
