diff -ruN -b Skype4Py/api/posix_dbus.py Skype4Py-new/api/posix_dbus.py
--- Skype4Py/api/posix_dbus.py	2009-09-26 09:01:09.000000000 -0300
+++ Skype4Py-new/api/posix_dbus.py	2011-06-20 15:46:21.352060620 -0300
@@ -176,7 +176,7 @@
             import os
             if os.fork() == 0: # we're child
                 os.setsid()
-                os.execlp('skype')
+                os.execlp('skype', 'skype')
 
     def shutdown(self):
         import os
