diff -Naur awahlig-skype4py-3d8ff59~/Skype4Py/api/posix_dbus.py awahlig-skype4py-3d8ff59/Skype4Py/api/posix_dbus.py
--- awahlig-skype4py-3d8ff59~/Skype4Py/api/posix_dbus.py	2012-12-30 14:01:27.731636010 -0600
+++ awahlig-skype4py-3d8ff59/Skype4Py/api/posix_dbus.py	2012-12-30 14:01:45.191288926 -0600
@@ -176,7 +176,7 @@
             import os
             if os.fork() == 0: # we're child
                 os.setsid()
-                os.execlp('skype')
+                os.execlp('skype', 'skype')
 
     def shutdown(self):
         import os
