diff -aur nautilus-2.32.1/src/file-manager/fm-directory-view.c nautilus-2.32.1.new/src/file-manager/fm-directory-view.c
--- nautilus-2.32.1/src/file-manager/fm-directory-view.c	2010-11-13 12:59:01.000000000 +0100
+++ nautilus-2.32.1.new/src/file-manager/fm-directory-view.c	2010-11-27 01:35:35.000000000 +0100
@@ -10854,10 +10854,6 @@
 			      G_TYPE_BOOLEAN, 0);
 	
 	binding_set = gtk_binding_set_by_class (klass);
-	gtk_binding_entry_add_signal (binding_set, GDK_Delete, 0,
-				      "trash", 0);
-	gtk_binding_entry_add_signal (binding_set, GDK_KP_Delete, 0,
-				      "trash", 0);
 	gtk_binding_entry_add_signal (binding_set, GDK_KP_Delete, GDK_SHIFT_MASK,
 				      "delete", 0);
 
