--- Makefile.PL	2010-12-13 14:18:26.000000000 +0100
+++ Makefile.PL.new	2011-02-27 19:58:14.000000000 +0100
@@ -11,7 +11,7 @@
 
 # GENERAL INSTALLATION INFORMATIONS
 my $ALLOW_SYMBOLIC_LINK;
-my $INSTALLDIRECTORY;
+my $INSTALLDIRECTORY = $ENV{'pkgdir'};
 my $BINDIRECTORY;
 my $ETCDIRECTORY;
 my $MANDIRECTORY;
@@ -549,8 +549,8 @@
 	# Unix compatible platform
 	$ALLOW_SYMBOLIC_LINK = 1;
 
-	my $PREFIX = substShellPatterns($options{'prefix'}) || File::Spec->catfile('','usr','local');
-	my $end = File::Spec->catfile('usr','local');
+	my $PREFIX = substShellPatterns($options{'prefix'}) || File::Spec->catfile('','usr');
+	my $end = File::Spec->catfile('usr');
 	if ($PREFIX =~ /\Q$end\E$/) {
 		$end = File::Spec->catfile("$PREFIX",'..','..','etc');
 	}
