--- systemd-36/src/fsck.c.orig	2011-09-23 18:17:15.305822578 +0300
+++ systemd-36/src/fsck.c	2011-09-24 04:46:19.497293848 +0300
@@ -127,12 +127,10 @@
                         arg_skip = true;
                 else if (startswith(w, "fsck.mode"))
                         log_warning("Invalid fsck.mode= parameter. Ignoring.");
-#if defined(TARGET_FEDORA) || defined(TARGET_MANDRIVA)
                 else if (strneq(w, "fastboot", l))
                         arg_skip = true;
                 else if (strneq(w, "forcefsck", l))
                         arg_force = true;
-#endif
         }
 
         free(line);
@@ -146,7 +144,7 @@
         if (access("/forcefsck", F_OK) >= 0)
                 arg_force = true;
 
-        if (access("/run/systemd/show-status", F_OK) >= 0 || plymouth_running())
+        if (access("/run/systemd/show-status", F_OK) >= 0)
                 arg_show_progress = true;
 }
 
