diff -aur nethack-3.4.3-patched/include/config.h nethack-3.4.3-modified/include/config.h
--- nethack-3.4.3-patched/include/config.h	2012-10-27 15:51:37.679413245 -0200
+++ nethack-3.4.3-modified/include/config.h	2012-10-27 16:20:51.893883098 -0200
@@ -182,7 +182,7 @@
 /* #define COMPRESS "/usr/bin/compress" */	/* Lempel-Ziv compression */
 /* #define COMPRESS_EXTENSION ".Z" */		/* compress's extension */
 /* An example of one alternative you might want to use: */
-#define COMPRESS "/bin/gzip"	/* FSF gzip compression */
+#define COMPRESS "/usr/bin/gzip"	/* FSF gzip compression */
 #define COMPRESS_EXTENSION ".gz"		/* normal gzip extension */
 #endif
 
@@ -214,7 +214,7 @@
  * otherwise it will be the current directory.
  */
 # ifndef HACKDIR
-#  define HACKDIR "/nh343"
+#  define HACKDIR "/usr/lib/nethack"
 # endif
 
 /*
@@ -380,7 +380,7 @@
 #define AUTOPICKUP_EXCEPTIONS  /* exceptions to autopickup */
 
 #define DUMP_LOG        /* Dump game end information to a file */
-#define DUMP_FN "/dgldir/userdata/%N/%n/dumplog/%t.nh343.txt"      /* Fixed dumpfile name, if you want
+#define DUMP_FN "/var/games/nethack/userdata/%N/%n/dumplog/%t.nh343.txt"      /* Fixed dumpfile name, if you want
                                    * to prevent definition by users */
 #define DUMPMSGS 20     /* Number of latest messages in the dump file  */
 
diff -aur nethack-3.4.3-patched/include/unixconf.h nethack-3.4.3-modified/include/unixconf.h
--- nethack-3.4.3-patched/include/unixconf.h	2012-10-27 15:51:37.686079911 -0200
+++ nethack-3.4.3-modified/include/unixconf.h	2012-10-27 16:21:44.900449147 -0200
@@ -102,7 +102,7 @@
  * If you want the static parts of your playground on a read-only file
  * system, define VAR_PLAYGROUND to be where the variable parts are kept.
  */
-#define VAR_PLAYGROUND "/nh343/var"
+#define VAR_PLAYGROUND "/var/games/nethack"
 
 
 /*
diff -aur nethack-3.4.3-patched/src/cmd.c nethack-3.4.3-modified/src/cmd.c
--- nethack-3.4.3-patched/src/cmd.c	2012-10-27 15:51:37.696079912 -0200
+++ nethack-3.4.3-modified/src/cmd.c	2012-10-27 19:49:36.614041649 -0200
@@ -540,7 +540,7 @@
 STATIC_PTR int
 wiz_mk_mapglyphdump()
 {
-    mk_mapdump("/dgldir/userdata/%N/%n/nh343.mapdump");
+    mk_mapdump("/var/games/nethack/userdata/%N/%n/nh343.mapdump");
     return 0;
 }
 
diff -aur nethack-3.4.3-patched/src/end.c nethack-3.4.3-modified/src/end.c
--- nethack-3.4.3-patched/src/end.c	2012-10-27 15:51:37.712746579 -0200
+++ nethack-3.4.3-modified/src/end.c	2012-10-27 19:49:25.487404021 -0200
@@ -196,7 +196,7 @@
     mode_t eimode = S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH;
 #endif
     char new_fn[512];
-    Sprintf(new_fn, "%s", dump_format_str("/dgldir/extrainfo-nh343/%n.extrainfo"));
+    Sprintf(new_fn, "%s", dump_format_str("/var/games/nethack/extrainfo/%n.extrainfo"));
 
     extrai = fopen(new_fn, "w");
     if (!extrai) {
diff -aur nethack-3.4.3-patched/src/topten.c nethack-3.4.3-modified/src/topten.c
--- nethack-3.4.3-patched/src/topten.c	2012-10-27 15:51:37.769413249 -0200
+++ nethack-3.4.3-modified/src/topten.c	2012-10-27 19:49:17.160759077 -0200
@@ -444,7 +444,7 @@
 void
 write_HUP_file()
 {
-    mk_HUPfile("/dgldir/userdata/%N/%n/hanguplist.txt");
+    mk_HUPfile("/var/games/nethack/userdata/%N/%n/hanguplist.txt");
 }
 
 
diff -aur nethack-3.4.3-patched/sys/unix/Makefile.src nethack-3.4.3-modified/sys/unix/Makefile.src
--- nethack-3.4.3-patched/sys/unix/Makefile.src	2012-10-27 15:51:37.782746581 -0200
+++ nethack-3.4.3-modified/sys/unix/Makefile.src	2012-10-27 17:41:19.617015915 -0200
@@ -151,7 +151,7 @@
 # flags for debugging:
 # CFLAGS = -g -I../include
 
-CFLAGS = -g -O -I../include
+CFLAGS = -O -I../include
 LFLAGS =
 
 # The Qt and Be window systems are written in C++, while the rest of
@@ -293,7 +293,7 @@
 LIBS =
 
 # make NetHack
-GAME     = nethack.343-nao
+GAME     = nethack
 # GAME     = nethack.prg
 
 # if you defined RANDOM in unixconf.h/tosconf.h since your system did not come
diff -aur nethack-3.4.3-patched/sys/unix/Makefile.top nethack-3.4.3-modified/sys/unix/Makefile.top
--- nethack-3.4.3-patched/sys/unix/Makefile.top	2012-10-27 15:51:37.782746581 -0200
+++ nethack-3.4.3-modified/sys/unix/Makefile.top	2012-10-27 19:48:20.640906764 -0200
@@ -15,14 +15,14 @@
 
 # make NetHack
 PREFIX	 = /opt/nethack/nethack.alt.org
-GAME     = nethack.343-nao
+GAME     = nethack
 # GAME     = nethack.prg
-GAMEUID  = games
-GAMEGRP  = games
+GAMEUID  = root
+GAMEGRP  = root
 
 # Permissions - some places use setgid instead of setuid, for instance
 # See also the option "SECURE" in include/config.h
-GAMEPERM = 0755
+GAMEPERM = 2755
 FILEPERM = 0644
 EXEPERM  = 0755
 DIRPERM  = 0755
@@ -35,9 +35,9 @@
 # therefore there should not be anything in GAMEDIR that you want to keep
 # (if there is, you'll have to do the installation by hand or modify the
 # instructions)
-GAMEDIR  = $(PREFIX)/nh343
-VARDIR  = $(GAMEDIR)/var
-#SHELLDIR = $(PREFIX)/games
+GAMEDIR  = $(PREFIX)/usr/lib/$(GAME)
+VARDIR   = $(PREFIX)/var/games/nethack
+SHELLDIR = $(PREFIX)/usr/bin
 
 # per discussion in Install.X11 and Install.Qt
 VARDATND = 
@@ -185,19 +185,19 @@
 	$(MAKE) dofiles-$${target-nodlb}
 	cp src/$(GAME) $(GAMEDIR)
 	cp util/recover $(GAMEDIR)
-#	-rm -f $(SHELLDIR)/$(GAME)
-#	sed -e 's;/usr/games/lib/nethackdir;$(GAMEDIR);' \
-#		-e 's;HACKDIR/nethack;HACKDIR/$(GAME);' \
-#		< sys/unix/nethack.sh \
-#		> $(SHELLDIR)/$(GAME)
+	-rm -f $(SHELLDIR)/$(GAME)
+	sed -e 's;/usr/games/lib/nethackdir;$(GAMEDIR);' \
+		-e 's;HACKDIR/nethack;HACKDIR/$(GAME);' \
+		< sys/unix/nethack.sh \
+		> $(SHELLDIR)/$(GAME)
 # set up their permissions
 	-( cd $(GAMEDIR) ; $(CHOWN) $(GAMEUID) $(GAME) recover ; \
 			$(CHGRP) $(GAMEGRP) $(GAME) recover )
 	chmod $(GAMEPERM) $(GAMEDIR)/$(GAME)
 	chmod $(EXEPERM) $(GAMEDIR)/recover
-#	-$(CHOWN) $(GAMEUID) $(SHELLDIR)/$(GAME)
-#	$(CHGRP) $(GAMEGRP) $(SHELLDIR)/$(GAME)
-#	chmod $(EXEPERM) $(SHELLDIR)/$(GAME)
+	-$(CHOWN) $(GAMEUID) $(SHELLDIR)/$(GAME)
+	$(CHGRP) $(GAMEGRP) $(SHELLDIR)/$(GAME)
+	chmod $(EXEPERM) $(SHELLDIR)/$(GAME)
 
 dofiles-dlb: check-dlb
 	( cd dat ; cp nhdat $(DATNODLB) $(GAMEDIR) )
@@ -233,13 +233,13 @@
 install: $(GAME) recover $(VARDAT) dungeon spec_levs
 # set up the directories
 # not all mkdirs have -p; those that don't will create a -p directory
-#	-mkdir -p $(SHELLDIR)
+	-mkdir -p $(SHELLDIR)
 	-rm -rf $(GAMEDIR) $(VARDIR)
-	-mkdir -p $(GAMEDIR) $(VARDIR) $(VARDIR)/save
+	-mkdir -p $(GAMEDIR) $(VARDIR)/{,save,extrainfo,userdata}
 	-rmdir ./-p
-	-$(CHOWN) $(GAMEUID) $(GAMEDIR) $(VARDIR) $(VARDIR)/save
-	$(CHGRP) $(GAMEGRP) $(GAMEDIR) $(VARDIR) $(VARDIR)/save
-	chmod $(DIRPERM) $(GAMEDIR) $(VARDIR) $(VARDIR)/save
+	-$(CHOWN) $(GAMEUID) $(GAMEDIR) $(VARDIR)/{,save,extrainfo,userdata}
+	$(CHGRP) $(GAMEGRP) $(GAMEDIR) $(VARDIR)/{,save,extrainfo,userdata}
+	chmod $(DIRPERM) $(GAMEDIR) $(VARDIR)/{,save,extrainfo,userdata}
 # set up the game files
 	( $(MAKE) dofiles )
 # set up some additional files
