*** old/src/sdl/debugger.cpp	2009-08-23 15:11:06.306726603 +0300
--- new/src/sdl/debugger.cpp	2009-08-23 15:13:43.740026449 +0300
***************
*** 941,947 ****
      u32 address = 0;
      u32 value = 0;
      int type = 0;
!     const char *s = args[1];
      char c = *s;
      if(strchr(s, ':')) {
        const char *name = s;
--- 941,947 ----
      u32 address = 0;
      u32 value = 0;
      int type = 0;
!     char *s = args[1];
      char c = *s;
      if(strchr(s, ':')) {
        const char *name = s;
