--- unity-2d-0.1/launcher/UnityApplications/CMakeLists.txt.orig	2011-01-14 19:40:10.000000000 +0100
+++ unity-2d-0.1/launcher/UnityApplications/CMakeLists.txt	2011-01-19 12:39:14.085966085 +0100
@@ -4,6 +4,7 @@
 pkg_check_modules(QTDEE REQUIRED libqtdee)
 pkg_check_modules(GDK REQUIRED gdk-2.0)
 pkg_check_modules(GIO REQUIRED gio-2.0)
+pkg_check_modules(GIO_UNIX REQUIRED gio-unix-2.0)
 pkg_check_modules(WNCK REQUIRED libwnck-1.0)
 
 # Sources
@@ -57,6 +58,7 @@
     ${QTDEE_INCLUDE_DIRS}
     ${GDK_INCLUDE_DIRS}
     ${GIO_INCLUDE_DIRS}
+    ${GIO_UNIX_INCLUDE_DIRS}
     ${WNCK_INCLUDE_DIRS}
     )
 
@@ -69,6 +71,7 @@
     ${QTDEE_LDFLAGS}
     ${GDK_LDFLAGS}
     ${GIO_LDFLAGS}
+    ${GIO_UNIX_LDFLAGS}
     ${WNCK_LDFLAGS}
     )
 
