=== modified file 'panel/applets/legacytray/legacytrayapplet.cpp'
--- panel/applets/legacytray/legacytrayapplet.cpp	2011-08-31 15:45:56 +0000
+++ panel/applets/legacytray/legacytrayapplet.cpp	2011-10-14 21:58:57 +0000
@@ -54,6 +54,7 @@
         SLOT(slotTaskCreated(SystemTray::Task*)));
 
     m_whitelist = m_dconfPanel->property("systrayWhitelist").toStringList();
+    m_whitelistAll = m_whitelist.contains("all", Qt::CaseInsensitive);
 }
 
 LegacyTrayApplet::~LegacyTrayApplet()
@@ -69,7 +70,7 @@
        classes to allow in the Panel's systray implementation." but here we only
        support matching on WM_CLASS.
     */
-    if (!m_whitelist.contains(task->name())) {
+    if (!m_whitelistAll && !m_whitelist.contains(task->name(), Qt::CaseInsensitive)) {
         return;
     }
 

=== modified file 'panel/applets/legacytray/legacytrayapplet.h'
--- panel/applets/legacytray/legacytrayapplet.h	2011-08-31 15:45:56 +0000
+++ panel/applets/legacytray/legacytrayapplet.h	2011-10-14 21:58:57 +0000
@@ -50,6 +50,7 @@
     SystemTray::FdoSelectionManager* m_selectionManager;
     QConf* m_dconfPanel;
     QStringList m_whitelist;
+    bool m_whitelistAll;
 };
 
 #endif /* LEGACYTRAYAPPLET_H */

