diff -Naur hybserv.orig/Makefile.in hybserv/Makefile.in
--- hybserv.orig/Makefile.in	2006-09-16 19:11:43.000000000 +0200
+++ hybserv/Makefile.in	2012-12-09 00:29:59.708083818 +0100
@@ -3,24 +3,24 @@
 
 # This is installation prefix. It can be overrided by passing PREFIX
 # argument to make, as in making a fake tree for binary package.
-PREFIX = @prefix@
+PREFIX = $(DESTDIR)/@prefix@
 
 # Set this to the directory to install the hybserv binary
-BINDIR = $(PREFIX)/hybserv
+BINDIR = $(PREFIX)/sbin
 
 # Set this to the directory to install hybserv.conf, motd.dcc, etc
-CONFDIR = $(PREFIX)/hybserv
+CONFDIR = $(DESTDIR)/@sysconfdir@/hybserv
 
 # Set this to the directory to install help files 
 # (should match HelpPath in settings.conf)
-HELPDIR = $(PREFIX)/hybserv/help
+HELPDIR = $(PREFIX)/share/doc/ircd-hybrid-serv/help
 
 # Set this to the directory to install document files 
-DOCDIR = $(PREFIX)/hybserv/doc
+DOCDIR = $(PREFIX)/share/doc/ircd-hybrid-serv
 
 # Set this to the directory for logs repository
 # (should match LogPath in settings.conf)
-LOGDIR = $(PREFIX)/hybserv/logs
+LOGDIR = $(DESTDIR)/@localstatedir@/ircd
 
 ############ Don't change anything below here ############
 
@@ -103,15 +103,15 @@
 	@echo "*** Installing binaries ***"
 	@if test ! -d $(BINDIR); then \
 		echo "** Creating $(BINDIR) **"; \
-		$(MKDIR) $(BINDIR) 2>/dev/null; \
+		$(MKDIR) -p $(BINDIR) 2>/dev/null; \
 	fi
 	@if test ! -d $(CONFDIR); then \
 		echo "** Creating $(CONFDIR) **"; \
-		$(MKDIR) $(CONFDIR) 2>/dev/null; \
+		$(MKDIR) -p $(CONFDIR) 2>/dev/null; \
 	fi
 	@if test ! -d $(LOGDIR); then \
 		echo "** Creating $(LOGDIR) **"; \
-		$(MKDIR) $(LOGDIR) 2>/dev/null; \
+		$(MKDIR) -p $(LOGDIR) 2>/dev/null; \
 	fi
 	@echo "** Installing Hybserv2 in $(BINDIR) **"
 	@$(INSTALL_BIN) "bin/hybserv" "$(BINDIR)/hybserv"
@@ -142,11 +142,11 @@
 	@echo "*** Installing help files ***"
 	@if test ! -d $(HELPDIR); then \
 		echo "** Creating $(HELPDIR) **"; \
-		$(MKDIR) $(HELPDIR) 2>/dev/null; \
+		$(MKDIR) -p $(HELPDIR) 2>/dev/null; \
 	else \
 		echo "** Removing old $(HELPDIR) **"; \
 		$(RMR) $(HELPDIR); \
-		$(MKDIR) $(HELPDIR) 2>/dev/null; \
+		$(MKDIR) -p $(HELPDIR) 2>/dev/null; \
 	fi
 	@echo "** Installing help files in $(HELPDIR) **"
 	@$(CPR) help/* $(HELPDIR)
@@ -156,7 +156,7 @@
 	@echo "*** Installing document files ***"
 	@if test ! -d $(DOCDIR); then \
 		echo "** Creating $(DOCDIR) **"; \
-		$(MKDIR) $(DOCDIR) 2>/dev/null; \
+		$(MKDIR) -p $(DOCDIR) 2>/dev/null; \
 	fi
 	@echo "** Installing doc files in $(DOCDIR) **";
 	@for file in $(DOCFILES1); do \
