diff -Naur hybserv.orig/include/config.h.in hybserv/include/config.h.in
--- hybserv.orig/include/config.h.in	2007-02-03 13:27:50.000000000 +0100
+++ hybserv/include/config.h.in	2012-12-09 02:09:07.289337184 +0100
@@ -14,7 +14,7 @@
  * This will automagically transform into correct SETPATH. Don't change
  * it.
  */
-#define   SETPATH   "@prefix@/hybserv/settings.conf"
+#define   SETPATH   "/etc/ircd-hybrid/hybserv/settings.conf"
 
 /*
  * Define this to enable NickServ (Nickname Services)
