--- src/recpt1/driver/pt1_pci.c	2014-08-01 13:32:48.371057026 +0900
+++ src/recpt1/driver/pt1_pci.c	2013-06-25 05:24:25.000000000 +0900
@@ -43,10 +43,10 @@
 #include	"pt1_i2c.h"
 #include	"pt1_tuner_data.h"
 #include	"pt1_ioctl.h"
+#include	"version.h"
 
 /* These identify the driver base version and may not be removed. */
-static char version[] __devinitdata =
-DRV_NAME ".c: " DRV_VERSION " " DRV_RELDATE " \n";
+static char version[] = DRV_NAME ".c: " DRV_VERSION " " DRV_RELDATE " \n";
 
 MODULE_AUTHOR("Tomoaki Ishikawa tomy@users.sourceforge.jp and Yoshiki Yazawa yaz@honeyplanet.jp");
 #define	DRIVER_DESC		"PCI earthsoft PT1/2 driver"
@@ -694,7 +694,7 @@
 	}
 	return 0 ;
 }
-static int __devinit pt1_pci_init_one (struct pci_dev *pdev,
+static int pt1_pci_init_one (struct pci_dev *pdev,
 				     const struct pci_device_id *ent)
 {
 	int			rc ;
@@ -920,7 +920,7 @@
 
 }
 
-static void __devexit pt1_pci_remove_one(struct pci_dev *pdev)
+static void pt1_pci_remove_one(struct pci_dev *pdev)
 {
 
 	int		lp ;
@@ -987,7 +987,7 @@
 static struct pci_driver pt1_driver = {
 	.name		= DRV_NAME,
 	.probe		= pt1_pci_init_one,
-	.remove		= __devexit_p(pt1_pci_remove_one),
+	.remove		= pt1_pci_remove_one,
 	.id_table	= pt1_pci_tbl,
 #ifdef CONFIG_PM
 	.suspend	= pt1_pci_suspend,
