From 2ad77f04a9b5ca8945d753a1fd6b69f9b19a0c24 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?J=C3=B6rg=20H=C3=B6hle?= <hoehle@users.sourceforge.net>
Date: Fri, 14 Sep 2012 15:06:44 +0200
Subject: [PATCH 08/25] mciseq: Pause causes an ABORT notification.

---
 dlls/mciseq/mcimidi.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/dlls/mciseq/mcimidi.c b/dlls/mciseq/mcimidi.c
index 04eaaf1..5e72ef1 100644
--- a/dlls/mciseq/mcimidi.c
+++ b/dlls/mciseq/mcimidi.c
@@ -1207,12 +1207,16 @@ static DWORD MIDI_mciPlay(WINE_MCIMIDI* wmm, DWORD dwFlags, LPMCI_PLAY_PARMS lpP
  */
 static DWORD MIDI_mciPause(WINE_MCIMIDI* wmm, DWORD dwFlags, LPMCI_GENERIC_PARMS lpParms)
 {
+    HANDLE oldcb = NULL;
+
     TRACE("(%d, %08X, %p);\n", wmm->wDevID, dwFlags, lpParms);
 
     if (wmm->dwStatus == MCI_MODE_PLAY) {
 	wmm->dwStatus = MCI_MODE_PAUSE;
+	oldcb = InterlockedExchangePointer(&wmm->hCallback, NULL);
 	ResetEvent(wmm->hEvent);
     }
+    if (oldcb) mciDriverNotify(oldcb, wmm->wDevID, MCI_NOTIFY_ABORTED);
 
     if ((dwFlags & MCI_NOTIFY) && lpParms)
 	MIDI_mciNotify(lpParms->dwCallback, wmm, MCI_NOTIFY_SUCCESSFUL);
-- 
1.5.6.3

