From 0dcda04c5796dab7f340706f1485d938cc5159b5 Mon Sep 17 00:00:00 2001
From: Ulrich Czekalla <ulrich@localhost.localdomain>
Date: Tue, 28 Oct 2008 04:07:11 -0400
Subject: [PATCH] Fix bug linking sync object into linked list

---
 dlls/ntdll/sync.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/dlls/ntdll/sync.c b/dlls/ntdll/sync.c
index 374d1dc..a74821b 100644
--- a/dlls/ntdll/sync.c
+++ b/dlls/ntdll/sync.c
@@ -319,9 +319,10 @@ NTSTATUS init_pipe_object(LPPIPEOBJ pobj, WORD type, int signaled)
     if (SUCCEEDED(ret))
     {
         RtlEnterCriticalSection(&csPipeObjs);
+        pobj->prev = 0;
         pobj->next = pipeobjs;
         if (pipeobjs)
-           pobj->prev = pipeobjs->prev;
+           pipeobjs->prev = pobj;
         pipeobjs = pobj;
         pipe_count++;
         RtlLeaveCriticalSection(&csPipeObjs);
-- 
1.5.4.1

