From 35d6c12177e7307a0fed09041f271c9274a54db5 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?J=C3=B6rg=20H=C3=B6hle?= <hoehle@users.sourceforge.net>
Date: Wed, 3 Oct 2012 18:41:51 +0200
Subject: [PATCH 20/25] mciseq: Check stop condition after sleeping.

---
 dlls/mciseq/mcimidi.c |   19 ++++++++++---------
 1 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/dlls/mciseq/mcimidi.c b/dlls/mciseq/mcimidi.c
index 8711f03..e20ab70 100644
--- a/dlls/mciseq/mcimidi.c
+++ b/dlls/mciseq/mcimidi.c
@@ -879,16 +879,14 @@ static DWORD MIDI_player(WINE_MCIMIDI* wmm, DWORD dwFlags)
 
     wmm->dwPulse = 0;
     wmm->dwTempo = 500000;
+    /* FIXME: the externally visible position should remain within [from..to] */
     wmm->dwPositionMS = 0;
 
-    while (wmm->dwStatus != MCI_MODE_STOP && wmm->dwStatus != MCI_MODE_NOT_READY) {
+    while ((mmt = MIDI_mciFindNextEvent(wmm, &hiPulse))) {
 
 	doPlay = (wmm->dwPositionMS >= dwStartMS && wmm->dwPositionMS <= wmm->dwEndMS);
-
-	TRACE("wmm->dwStatus=%d, doPlay=%c\n", wmm->dwStatus, doPlay ? 'T' : 'F');
-
-	if ((mmt = MIDI_mciFindNextEvent(wmm, &hiPulse)) == NULL)
-	    break;  /* no more event on tracks */
+	/* FIXME: position + hiPulse >= start */
+	TRACE("Mode=%u Event %08X Note=%c\n", wmm->dwStatus, mmt->dwEventData, doPlay ? 'Y' : 'N');
 
 	/* if starting playing, then set StartTicks to the value it would have had
 	 * if play had started at position 0
@@ -905,9 +903,8 @@ static DWORD MIDI_player(WINE_MCIMIDI* wmm, DWORD dwFlags)
 		DWORD	togo = wmm->dwStartTicks + wmm->dwPositionMS;
 		DWORD	tc = GetTickCount();
 
-		TRACE("Pulses hi=0x%08x <> cur=0x%08x\n", hiPulse, wmm->dwPulse);
-		TRACE("Wait until %u => %u ms\n",
-		      tc - wmm->dwStartTicks, togo - wmm->dwStartTicks);
+		TRACE("Delay from %u to %u, at %u to %u ms\n", wmm->dwPulse, hiPulse,
+		      tc - wmm->dwStartTicks,  wmm->dwPositionMS);
 		if (tc < togo)
 		    Sleep(togo - tc);
 	    }
@@ -925,6 +922,10 @@ static DWORD MIDI_player(WINE_MCIMIDI* wmm, DWORD dwFlags)
 	    startedPlaying = FALSE;
 	    continue; /* after sleeping, check conditions again */
 	}
+	if (wmm->dwStatus == MCI_MODE_STOP || wmm->dwStatus == MCI_MODE_NOT_READY)
+	    break;
+	else if (wmm->dwStatus != MCI_MODE_PLAY)
+	    continue;
 
 	switch (LOBYTE(LOWORD(mmt->dwEventData))) {
 	case 0xF0:
-- 
1.5.6.3

