--- wine-1.3.24/dlls/kernel32/volume.c.orig	2011-07-27 11:48:12.883525967 -0700
+++ wine-1.3.24/dlls/kernel32/volume.c	2011-07-27 11:49:44.284650916 -0700
@@ -666,7 +666,18 @@
         if (flags) *flags = FILE_CASE_PRESERVED_NAMES;  /* FIXME */
         break;
     default:
-        if (fsname) lstrcpynW( fsname, ntfsW, fsname_len );
+        if (fsname)
+        {
+            const char* env = getenv("MUSE_DEFAULT_WINE_FILESYSTEM");
+            if (env && *env != '\0')
+            {
+                wine_utf8_mbstowcs( 0, env, strlen(env) + 1, fsname, fsname_len );
+                TRACE("Honoring MUSE_DEFAULT_WINE_FILESYSTEM: %s\n", debugstr_w(fsname));
+            }
+            else
+                lstrcpynW( fsname, ntfsW, fsname_len );
+        }
+
         if (filename_len) *filename_len = 255;
         if (flags) *flags = FILE_CASE_PRESERVED_NAMES | FILE_PERSISTENT_ACLS;
         break;
