--- wine-1.3.24/dlls/kernel32/volume.c.orig	2011-07-27 11:51:52.856526010 -0700
+++ wine-1.3.24/dlls/kernel32/volume.c	2011-07-27 11:55:33.964526395 -0700
@@ -287,6 +287,46 @@
     return ret;
 }
 
+/* fetch the type of a drive from the registry */
+static BOOL get_registry_label( const WCHAR *device, WCHAR *label, DWORD len )
+{
+    static const WCHAR drive_types_keyW[] = {'M','a','c','h','i','n','e','\\',
+                                             'S','o','f','t','w','a','r','e','\\',
+                                             'W','i','n','e','\\',
+                                             'D','r','i','v','e','L','a','b','e','l','s',0 };
+    OBJECT_ATTRIBUTES attr;
+    UNICODE_STRING nameW;
+    HANDLE hkey;
+    DWORD dummy;
+    BOOL ret = FALSE;
+    char tmp[128 + sizeof(KEY_VALUE_PARTIAL_INFORMATION)];
+    WCHAR driveW[] = {'A',':',0};
+
+    attr.Length = sizeof(attr);
+    attr.RootDirectory = 0;
+    attr.ObjectName = &nameW;
+    attr.Attributes = 0;
+    attr.SecurityDescriptor = NULL;
+    attr.SecurityQualityOfService = NULL;
+    RtlInitUnicodeString( &nameW, drive_types_keyW );
+    /* @@ Wine registry key: HKLM\Software\Wine\Drives */
+    if (NtOpenKey( &hkey, KEY_ALL_ACCESS, &attr ) != STATUS_SUCCESS) return DRIVE_UNKNOWN;
+
+    driveW[0] = device[4];
+    RtlInitUnicodeString( &nameW, driveW );
+    if (!NtQueryValueKey( hkey, &nameW, KeyValuePartialInformation, tmp, sizeof(tmp), &dummy ))
+    {
+        WCHAR *data = (WCHAR *)((KEY_VALUE_PARTIAL_INFORMATION *)tmp)->Data;
+        if (data && *data)
+        {
+            lstrcpynW( label, data, len);
+            TRACE("%s -> %sd\n", debugstr_w(device), debugstr_w(label), len );
+            ret = TRUE;
+        }
+    }
+    NtClose( hkey );
+    return ret;
+}
 
 /******************************************************************
  *		VOLUME_FindCdRomDataBestVoldesc
@@ -422,6 +462,7 @@
         break;
     case FS_UNKNOWN:
         get_filesystem_label( device, label, len );
+	if (label[0] == '\0') get_registry_label( device->Buffer, label, len );
         return;
     case FS_FAT1216:
         label_ptr = superblock + 0x2b;
@@ -646,7 +687,11 @@
     }
     if (info.DeviceType == FILE_DEVICE_CD_ROM_FILE_SYSTEM) type = FS_ISO9660;
 
-    if (label && label_len) get_filesystem_label( &nt_name, label, label_len );
+    if (label && label_len) 
+    {
+        get_filesystem_label( &nt_name, label, label_len );
+        if ( label[0] == '\0' ) get_registry_label( nt_name.Buffer, label, label_len );
+    }
     if (serial) *serial = get_filesystem_serial( &nt_name );
 
 fill_fs_info:  /* now fill in the information that depends on the file system type */
