--- wine-1.3.24/dlls/ole32/ole2.c.orig	2011-12-08 14:44:37.811271736 -0800
+++ wine-1.3.24/dlls/ole32/ole2.c	2011-12-08 14:45:46.934191902 -0800
@@ -2180,7 +2180,16 @@ static LRESULT WINAPI OLEDD_DragTrackerW
     case WM_TIMER:
     case WM_MOUSEMOVE:
     {
-      OLEDD_TrackMouseMove((TrackerWindowInfo*)GetWindowLongPtrA(hwnd, 0));
+      TrackerWindowInfo* p = (TrackerWindowInfo*)GetWindowLongA(hwnd, 0); 
+      if (!p->trackingDone)
+      {
+       p->dwKeyState = OLEDD_GetButtonState();
+       if (p->dwKeyState & MK_LBUTTON)
+           OLEDD_TrackMouseMove(p);
+       else
+           OLEDD_TrackStateChange(p);
+      }
+
       break;
     }
     case WM_LBUTTONUP:
