--- wine-1.3.24/tools/winegcc/Makefile.in.orig	2011-12-21 10:12:35.082528892 -0800
+++ wine-1.3.24/tools/winegcc/Makefile.in	2011-12-21 10:13:12.675449915 -0800
@@ -4,9 +4,9 @@
 	-DLIBDIR="\"$(libdir)\"" \
 	-DDLLFLAGS="\"@DLLFLAGS@\"" \
 	-DLDDLLFLAGS="\"@LDDLLFLAGS@\"" \
-	-DCC="\"$(CC)\"" \
-	-DCPP="\"@CPPBIN@\"" \
-	-DCXX="\"@CXX@\"" \
+	-DCC="\"gcc\"" \
+	-DCPP="\"cpp\"" \
+	-DCXX="\"g++\"" \
 	-DPRELINK="\"$(PRELINK)\""
 
 PROGRAMS = \
