--- wine-1.3.24/dlls/user32/cursoricon.c.orig	2011-12-28 12:22:07.621056775 -0800
+++ wine-1.3.24/dlls/user32/cursoricon.c	2011-12-28 13:01:47.757181527 -0800
@@ -895,8 +895,13 @@
                        0, 0, bmi_copy->bmiHeader.biWidth, bmi_copy->bmiHeader.biHeight,
                        color_bits, bmi_copy, DIB_RGB_COLORS, SRCCOPY );
 
-        if (bmi_has_alpha( bmi_copy, color_bits ))
-            alpha = create_alpha_bitmap( color, mask, bmi_copy, color_bits );
+        if (bmi_has_alpha( bmi_copy, color_bits )) {
+            const char* env = getenv("MUSE_DISABLE_ALPHA_FOR_GDB");
+            if (env && *env != '0')
+                TRACE("MUSE_DISABLE_ALPHA_FOR_GDB - Skipping alpha for GDB\n");
+            else
+                alpha = create_alpha_bitmap( color, mask, bmi_copy, color_bits );
+        }
 
         /* convert info to monochrome to copy the mask */
         bmi_copy->bmiHeader.biBitCount = 1;
