--- wine/dlls/advapi32/service.c.old	2013-12-07 19:36:29.675558032 -0500
+++ wine/dlls/advapi32/service.c	2013-12-07 19:35:24.404554664 -0500
@@ -544,7 +544,7 @@
                             (LPBYTE)&spi, sizeof(spi), &i );
                     if (res)
                     {
-                        FIXME("service should be able to delay shutdown\n");
+                        // FIXME("service should be able to delay shutdown\n");
                         timeout += spi.dwPreshutdownTimeout;
                         ret = service_handle_control( services[i], SERVICE_CONTROL_PRESHUTDOWN );
                         wait_handles[n++] = services[i]->thread;
