--- wine-1.5.31/server/fd.c~	2013-05-26 12:30:40.876435809 -0400
+++ wine-1.5.31/server/fd.c	2013-05-26 13:26:46.717609450 -0400
@@ -1479,7 +1479,7 @@
     /* only a few access bits are meaningful wrt sharing */
-    const unsigned int read_access = FILE_READ_DATA | FILE_EXECUTE;
+    /*const unsigned int read_access = FILE_READ_DATA | FILE_EXECUTE;*/
-    const unsigned int write_access = FILE_WRITE_DATA | FILE_APPEND_DATA;
+    /*const unsigned int write_access = FILE_WRITE_DATA | FILE_APPEND_DATA;*/
-    const unsigned int all_access = read_access | write_access | DELETE;
+    /*const unsigned int all_access = read_access | write_access | DELETE;*/
 
     unsigned int existing_sharing = FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE;
     unsigned int existing_access = 0;

 
