--- wine-1.5.31/dlls/ntoskrnl.exe/ntoskrnl.c.orig	2013-05-29 03:21:11.461106861 -0400
+++ wine-1.5.31/dlls/ntoskrnl.exe/ntoskrnl.c	2013-05-29 03:20:13.378103865 -0400
@@ -751,7 +751,7 @@
  */
 void WINAPI IoAcquireCancelSpinLock( PKIRQL Irql )
 {
-    FIXME( "stub: %p\n", irql );
+    FIXME( "stub: %p\n", Irql );
 }
 
 /***********************************************************************
@@ -759,7 +759,7 @@
  */
 void WINAPI IoReleaseCancelSpinLock( KIRQL Irql )
 {
-    FIXME( "stub: %u\n", irql );
+    FIXME( "stub: %u\n", Irql );
 }
 
 /***********************************************************************
