--- wine-1.3.24/dlls/ntdll/file.c.orig	2011-07-29 14:17:26.415459732 -0700
+++ wine-1.3.24/dlls/ntdll/file.c	2011-07-29 14:18:58.620601783 -0700
@@ -163,6 +163,14 @@
             return io->u.Status;
         }
 
+        /* todo - hack for Ivory Library Installer - force GENERIC_xxx access on */
+        if ((access & STANDARD_RIGHTS_ALL) == STANDARD_RIGHTS_ALL) {
+            if ((access & (GENERIC_READ|GENERIC_WRITE)) == 0) {
+                FIXME("L GENERIC_xxx_ACCESS\n");
+                access |= (GENERIC_READ|GENERIC_WRITE);
+            }
+        }
+
         SERVER_START_REQ( create_file )
         {
             req->access     = access;
