--- wine-1.3.24/dlls/user32/dialog.c.orig	2011-07-29 12:01:21.599459656 -0700
+++ wine-1.5.22/dlls/user32/dialog.c	2011-07-29 12:09:34.896459932 -0700
@@ -724,6 +724,14 @@ HWND WINAPI CreateDialogParamA
     HRSRC hrsrc;
     LPCDLGTEMPLATEA ptr;
 
+    /* L bug #2602 - defeat dialog box */
+    const char* env = getenv("L_DEFEAT_DIALOGBOX");
+    if (env && isdigit(*env)) {
+        WINE_MESSAGE("DialogBoxParamA: Honoring L_DEFEAT_DIALOGBOX(%s)\n", env);
+        return(HWND)(atoi(env));
+    }
+    /* end L bug #2602 */
+
     if (!(hrsrc = FindResourceA( hInst, name, (LPSTR)RT_DIALOG ))) return 0;
     if (!(ptr = LoadResource(hInst, hrsrc))) return 0;
     return CreateDialogIndirectParamA( hInst, ptr, owner, dlgProc, param );
@@ -739,6 +747,14 @@ HWND WINAPI CreateDialogParamW( HINSTANCE hInst, LPCWSTR name, HWND owner,
     HRSRC hrsrc;
     LPCDLGTEMPLATEA ptr;
 
+    /* L bug #2602 - defeat dialog box */
+    const char* env = getenv("L_DEFEAT_DIALOGBOX");
+    if (env && isdigit(*env)) {
+        WINE_MESSAGE("DialogBoxParamW: Honoring L_DEFEAT_DIALOGBOX(%s)\n", env);
+        return(HWND)(atoi(env));
+       }
+    /* end L bug #2602 */
+
     if (!(hrsrc = FindResourceW( hInst, name, (LPWSTR)RT_DIALOG ))) return 0;
     if (!(ptr = LoadResource(hInst, hrsrc))) return 0;
     return CreateDialogIndirectParamW( hInst, ptr, owner, dlgProc, param );
@@ -876,11 +894,21 @@
 /***********************************************************************
  *		DialogBoxIndirectParamAorW (USER32.@)
  */
+int g_LDefeatDialogBoxValue;
 INT_PTR WINAPI DialogBoxIndirectParamAorW( HINSTANCE hInstance, LPCVOID template,
                                            HWND owner, DLGPROC dlgProc,
                                            LPARAM param, DWORD flags )
 {
-    HWND hwnd = DIALOG_CreateIndirect( hInstance, template, owner, dlgProc, param, !flags, TRUE );
+    /* L bug #2602 - defeat dialog box */
+    HWND hwnd;
+    const char* env = getenv("L_DEFEAT_DIALOGBOX");
+    if (env && isdigit(*env)) {
+        WINE_MESSAGE("DialogBoxIndirectParamAorW: Honoring L_DEFEAT_DIALOGBOX(%s)\n", env);
+        return(atoi(env));
+       }
+    /* end L bug #2602 */
+
+    hwnd = DIALOG_CreateIndirect( hInstance, template, owner, dlgProc, param, !flags, TRUE );
     if (hwnd) return DIALOG_DoDialogBox( hwnd, owner );
     return -1;
 }
