--- wine-1.6-rc4/dlls/mciseq/mcimidi.c.orig	2013-07-15 12:12:05.650083696 -0400
+++ wine-1.6-rc4/dlls/mciseq/mcimidi.c	2013-07-15 15:04:15.976616630 -0400
@@ -44,6 +44,8 @@
 #define MIDI_NOTEOFF             0x80
 #define MIDI_NOTEON              0x90
 
+#define L_RT_THREADS 1
+
 typedef struct {
     DWORD		dwFirst;		/* offset in file of track */
     DWORD		dwLast;			/* number of bytes in file of track */
@@ -1173,7 +1175,7 @@
 	} else {
 	    HANDLE events[2] = {wmm->hEvent, wmm->hThread};
 
-	    SetThreadPriority(wmm->hThread, THREAD_PRIORITY_TIME_CRITICAL);
+	    SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL);
 	    dwRet = 0;
 	    /* no dead-lock since either midiOutOpen returns an error or hEvent is set
 	     * (or the thread crashes). */
