--- wine-1.7.14/dlls/ole32/ole2.c.orig	2011-12-08 14:44:37.811271736 -0800
+++ wine-1.7.14/dlls/ole32/ole2.c	2011-12-08 14:45:46.934191902 -0800
@@ -2198,6 +2199,19 @@
     }
     case WM_TIMER:
     case WM_MOUSEMOVE:
+    {
+      TrackerWindowInfo* p = (TrackerWindowInfo*)GetWindowLongPtrA(hwnd, 0); 
+      if (!p->trackingDone)
+      {
+       p->dwKeyState = OLEDD_GetButtonState();
+       if (p->dwKeyState & MK_LBUTTON)
+           OLEDD_TrackStateChange(p);
+       else
+           OLEDD_TrackStateChange(p);
+      }
+
+      break;
+    }
     case WM_LBUTTONUP:
     case WM_MBUTTONUP:
     case WM_RBUTTONUP:
