--- wine-1.6-rc5/dlls/winepulse.drv/mmdevdrv.c.orig	2013-07-16 11:33:39.652422038 -0400
+++ wine-1.6-rc5/dlls/winepulse.drv/mmdevdrv.c	2013-07-16 11:36:40.257431355 -0400
@@ -59,6 +59,8 @@
 
 #include "wine/list.h"
 
+#define L_RT_THREADS 1 
+
 #define NULL_PTR_ERR MAKE_HRESULT(SEVERITY_ERROR, FACILITY_WIN32, RPC_X_NULL_REF_POINTER)
 
 WINE_DEFAULT_DEBUG_CHANNEL(pulse);
@@ -113,7 +115,7 @@
             pthread_mutex_init(&pulse_lock, NULL);
     } else if (reason == DLL_PROCESS_DETACH) {
         if (pulse_thread)
-           SetThreadPriority(pulse_thread, THREAD_PRIORITY_TIME_CRITICAL);
+           //SetThreadPriority(pulse_thread, THREAD_PRIORITY_TIME_CRITICAL);
         if (pulse_ctx) {
            pa_context_disconnect(pulse_ctx);
            pa_context_unref(pulse_ctx);
@@ -419,7 +421,7 @@
             ERR("Failed to create mainloop thread.");
             return E_FAIL;
         }
-        SetThreadPriority(pulse_thread, THREAD_PRIORITY_TIME_CRITICAL);
+        //SetThreadPriority(pulse_thread, THREAD_PRIORITY_TIME_CRITICAL);
         pthread_cond_wait(&pulse_cond, &pulse_lock);
     }
  
