--- a/dlls/kernel32/thread.c	
+++ a/dlls/kernel32/thread.c	
@@ -411,13 +411,16 @@ DWORD WINAPI SetThreadIdealProcessor(
     HANDLE hThread,          /* [in] Specifies the thread of interest */
     DWORD dwIdealProcessor)  /* [in] Specifies the new preferred processor */
 {
-    FIXME("(%p): stub\n",hThread);
+    TRACE("(%p), %u\n", hThread, dwIdealProcessor);
     if (dwIdealProcessor > MAXIMUM_PROCESSORS)
     {
         SetLastError(ERROR_INVALID_PARAMETER);
         return ~0u;
     }
-    return 0;
+    if (SetThreadAffinityMask( hThread, (1 << dwIdealProcessor) ))
+        return 0;
+    else
+        return ~0u;
 }
