--- wine-1.5.31/dlls/ntdll/signal_x86_64.c.orig	2013-06-05 14:41:23.561413465 -0400
+++ wine-1.5.31/dlls/ntdll/signal_x86_64.c	2013-06-05 14:41:34.366414023 -0400
@@ -2437,8 +2439,11 @@
 NTSTATUS signal_alloc_thread( TEB **teb )
 {
     static size_t sigstack_zero_bits;
+    struct ntdll_thread_data *thread_data;
+    struct ntdll_thread_data *parent_data = NULL;
     SIZE_T size;
     NTSTATUS status;
+    void *addr = NULL;
 
     if (!sigstack_zero_bits)
     {
