--- wine-l-pa-1.7.20/dlls/kernel32/thread.c.orig	2014-06-25 16:55:58.384382152 -0400
+++ wine-l-pa-1.7.20/dlls/kernel32/thread.c	2014-06-25 16:56:20.404383288 -0400
@@ -43,6 +43,7 @@
     #include <sched.h>
     #include <pthread.h>
     #include <errno.h>
+    #include <stdint.h>
 #endif
 
 #include "kernel_private.h"
--- wine-l-pa-1.7.20/dlls/ntdll/thread.c.orig	2014-06-25 15:20:50.331087678 -0400
+++ wine-l-pa-1.7.20/dlls/ntdll/thread.c	2014-06-25 16:54:58.800379078 -0400
@@ -45,6 +45,12 @@
 #include "ntdll_misc.h"
 #include "ddk/wdm.h"
 #include "wine/exception.h"
+#if L_RT_THREADS
+    #include <sched.h>
+    #include <pthread.h>
+    #include <errno.h>
+    #include <stdint.h>
+#endif
 
 WINE_DEFAULT_DEBUG_CHANNEL(thread);
 WINE_DECLARE_DEBUG_CHANNEL(relay);
