From 6e53e047b361996046c9717b4895852c4390e003 Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <maarten.lankhorst@canonical.com>
Date: Sat, 31 Aug 2013 08:35:59 +0200
Subject: [PATCH 02/41] valgrind prevent crash hack

---
 libs/wine/loader.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

Index: wine1.6-1.7.1-actually1.6/libs/wine/loader.c
===================================================================
--- wine1.6-1.7.1-actually1.6.orig/libs/wine/loader.c	2013-09-13 16:19:05.000000000 -0700
+++ wine1.6-1.7.1-actually1.6/libs/wine/loader.c	2013-09-13 16:20:06.000000000 -0700
@@ -59,6 +59,10 @@
 #include "winbase.h"
 #include "wine/library.h"
 
+#ifdef HAVE_VALGRIND_MEMCHECK_H
+#include <valgrind/memcheck.h>
+#endif
+
 /* argc/argv for the Windows application */
 int __wine_main_argc = 0;
 char **__wine_main_argv = NULL;
@@ -644,7 +648,6 @@
     return ret;
 }
 
-
 /***********************************************************************
  *           set_max_limit
  *
@@ -655,6 +658,11 @@
 #ifdef HAVE_SETRLIMIT
     struct rlimit rlimit;
 
+#if defined(RLIMIT_NOFILE) && defined(RUNNING_ON_VALGRIND)
+    if (limit == RLIMIT_NOFILE && RUNNING_ON_VALGRIND)
+        return;
+#endif
+
     if (!getrlimit( limit, &rlimit ))
     {
         rlimit.rlim_cur = rlimit.rlim_max;
