@@ -, +, @@ 
---
 dlls/msi/table.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)
--- a/dlls/msi/table.c	
+++ a/dlls/msi/table.c	
@@ -671,7 +671,9 @@
     /* Note: _Columns table doesn't have non-persistent data */
 
     /* if maxcount is non-zero, assume it's exactly right for this table */
-    if (colinfo) memset( colinfo, 0, maxcount * sizeof(*colinfo) );
+    if (maxcount) {
+        memset( colinfo, 0, maxcount * sizeof(*colinfo) );
+    }
     count = table->row_count;
     for (i = 0; i < count; i++)
     {
-- 
