--- wine-1.6-rc4/dlls/winmm/time.c.orig	2013-07-12 14:37:04.000000000 -0400
+++ wine-1.6-rc4/dlls/winmm/time.c	2013-07-15 14:57:05.959594445 -0400
@@ -39,6 +39,8 @@
 #include <sys/poll.h>
 #endif
 
+#define L_RT_THREADS 1
+
 #include "windef.h"
 #include "winbase.h"
 #include "mmsystem.h"
@@ -258,7 +260,7 @@
         HMODULE mod;
         GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS, (LPCWSTR)TIME_MMSysTimeThread, &mod);
         TIME_hMMTimer = CreateThread(NULL, 0, TIME_MMSysTimeThread, mod, 0, NULL);
-        SetThreadPriority(TIME_hMMTimer, THREAD_PRIORITY_TIME_CRITICAL);
+        SetThreadPriority(TIME_hMMTimer, THREAD_PRIORITY_TIME_CRITICAL);
     }
 }
 
