--- wine-1.6-rc4/dlls/winmm/winmm.c.orig	2013-07-12 14:37:04.000000000 -0400
+++ wine-1.6-rc4/dlls/winmm/winmm.c	2013-07-15 14:51:31.074577169 -0400
@@ -53,6 +53,8 @@
 #include "winternl.h"
 #include "winemm.h"
 
+#define L_RT_THREADS 1
+
 #include "wine/debug.h"
 
 WINE_DEFAULT_DEBUG_CHANNEL(winmm);
@@ -1301,7 +1303,7 @@
 	midiStreamClose((HMIDISTRM)hMidiOut);
 	return MMSYSERR_NOMEM;
     }
-    SetThreadPriority(lpMidiStrm->hThread, THREAD_PRIORITY_TIME_CRITICAL);
+    SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL);
 
     /* wait for thread to have started, and for its queue to be created */
     WaitForSingleObject(lpMidiStrm->hEvent, INFINITE);
@@ -1522,7 +1524,7 @@
         if (hEvent) CloseHandle(hEvent);
         return TASKERR_OUTOFMEMORY;
     }
-    SetThreadPriority(hThread, THREAD_PRIORITY_TIME_CRITICAL);
+    SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL);
     if (ph) *ph = hEvent;
     CloseHandle(hThread);
     return 0;
