--- a/web/includes/dream.php	2011-11-28 15:24:12.027403373 +0000
+++ b/web/includes/dream.php	2011-11-28 15:24:22.430866699 +0000
@@ -260,7 +260,7 @@
 		curl_setopt($curl, CURLOPT_URL, $uri);
 		curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
 		curl_setopt($curl, CURLOPT_TIMEOUT, $this->timeout);
-		curl_setopt($curl, CURLOPT_FOLLOWLOCATION, 1);
+		//curl_setopt($curl, CURLOPT_FOLLOWLOCATION, 1);
 		curl_setopt($curl, CURLOPT_MAXREDIRS, 10);
 		curl_setopt($curl, CURLOPT_CUSTOMREQUEST, $verb);
 		curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false);
--- a/web/deki/core/http_plug.php	2011-12-01 13:00:31.997512174 +0000
+++ b/web/deki/core/http_plug.php	2011-12-01 13:01:48.081802116 +0000
@@ -484,7 +484,7 @@
 		//curl_setopt($curl, CURLOPT_PROXY, 'X.X.X.X:8888');
 		
 		curl_setopt($curl, CURLOPT_TIMEOUT, $this->timeout);
-		curl_setopt($curl, CURLOPT_FOLLOWLOCATION, 1);
+		//curl_setopt($curl, CURLOPT_FOLLOWLOCATION, 1);
 		curl_setopt($curl, CURLOPT_MAXREDIRS, 10);
 		curl_setopt($curl, CURLOPT_CUSTOMREQUEST, $verb);
 		curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, false);
