diff -ru gnome15-0.6.3/configure gnome15-0.6.3-patched/configure
--- gnome15-0.6.3/configure	2011-05-09 01:51:40.000000000 +0200
+++ gnome15-0.6.3-patched/configure	2011-05-10 00:23:24.000000000 +0200
@@ -3143,8 +3143,8 @@
     PYTHON_NAME=`basename $PYTHON`
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking $PYTHON_NAME module: opencv" >&5
 $as_echo_n "checking $PYTHON_NAME module: opencv... " >&6; }
-	$PYTHON -c "import opencv" 2>/dev/null
-	if test $? -eq 0;
+	$PYTHON -c "import pyopencv" 2>/dev/null
+	if test $? -ne 0;
 	then
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
diff -ru gnome15-0.6.3/src/plugins/webcam/webcam.py gnome15-0.6.3-patched/src/plugins/webcam/webcam.py
--- gnome15-0.6.3/src/plugins/webcam/webcam.py	2011-03-15 00:47:20.000000000 +0100
+++ gnome15-0.6.3-patched/src/plugins/webcam/webcam.py	2011-05-10 00:25:15.000000000 +0200
@@ -23,10 +23,10 @@
 import gnome15.g15_util as g15util
 import gnome15.g15_screen as g15screen
 import gtk
-import opencv
+import pyopencv
 import os
 #this is important for capturing/displaying images
-from opencv import highgui
+from pyopencv import highgui
 from threading import RLock
 import logging
 logger = logging.getLogger("webcam")
@@ -105,8 +105,8 @@
         try:
             im = highgui.cvQueryFrame(self._camera)
             if im:
-                im = opencv.cvGetMat(im)
-                return opencv.adaptors.Ipl2PIL(im)
+                im = pyopencv.cvGetMat(im)
+                return pyopencv.adaptors.Ipl2PIL(im)
         finally:
             self._lock.release()
     
