diff --git a/src/gnome15/util/g15os.py b/src/gnome15/util/g15os.py
index 6573552..6622a62 100644
--- a/src/gnome15/util/g15os.py
+++ b/src/gnome15/util/g15os.py
@@ -42,7 +42,7 @@ def run_script(script, args = None, background = True):
             a += "\"%s\"" % arg
     p = os.path.realpath(os.path.join(g15globals.scripts_dir,script))
     logger.info("Running '%s'" % p)
-    return os.system("python \"%s\" %s %s" % ( p, a, " &" if background else "" ))
+    return os.system("python2 \"%s\" %s %s" % ( p, a, " &" if background else "" ))
 
 def get_command_output(cmd):
     """
