--- src/blitz-0.9/blitz/mathfunc.h~	2008-08-09 22:30:23.559474609 +0200
+++ src/blitz-0.9/blitz/mathfunc.h	2008-08-09 22:32:14.462798142 +0200
@@ -1,5 +1,6 @@
 // Generated: genmathfunc.cpp Jan 28 2005 12:04:32
 
+#include <cstdlib>
 #ifndef BZ_MATHFUNC_H
 #define BZ_MATHFUNC_H
 
--- src/blitz-0.9/blitz/funcs.h~	2008-08-09 22:30:41.184227089 +0200
+++ src/blitz-0.9/blitz/funcs.h	2008-08-09 22:31:57.980323025 +0200
@@ -27,6 +27,7 @@
 #ifndef BZ_FUNCS_H
 #define BZ_FUNCS_H
 
+#include <cstdlib>
 #include <blitz/blitz.h>
 #include <blitz/promote.h>
 #include <blitz/prettyprint.h>
