diff --git a/gvc-applet.c b/gvc-applet.c
index a73a845..f1b07a5 100755
--- a/gvc-applet.c
+++ b/gvc-applet.c
@@ -41,18 +41,18 @@
 #define SCALE_SIZE 128
 
 static const char *output_icon_names[] = {
-        "audio-volume-muted-symbolic",
-        "audio-volume-low-symbolic",
-        "audio-volume-medium-symbolic",
-        "audio-volume-high-symbolic",
+        "audio-volume-muted",
+        "audio-volume-low",
+        "audio-volume-medium",
+        "audio-volume-high",
         NULL
 };
 
 static const char *input_icon_names[] = {
-        "audio-input-microphone-muted-symbolic",
-        "audio-input-microphone-low-symbolic",
-        "audio-input-microphone-medium-symbolic",
-        "audio-input-microphone-high-symbolic",
+        "audio-input-microphone-muted",
+        "audio-input-microphone-low",
+        "audio-input-microphone-medium",
+        "audio-input-microphone-high",
         NULL
 };
 
