=== modified file 'configure.ac'
--- configure.ac	2012-01-08 18:26:35 +0000
+++ configure.ac	2012-01-08 18:50:09 +0000
@@ -142,8 +142,15 @@
         file_beagle=$enableval,
         enable_file=auto)
 
-if test "x$enable_file" != "xno"; then 
-        PKG_CHECK_MODULES([DB4O], [db4o], HAVE_DB4O=yes, HAVE_DB4O=no)
+if test "x$enable_file" != "xno"; then
+        PKG_CHECK_MODULES([DB4O], [db4o >= 8.0], [HAVE_DB4O_8=yes], [HAVE_DB4O_8=no])
+        AM_CONDITIONAL([HAVE_DB4O_8], [test "$HAVE_DB4O_8" = "yes"])
+
+        if test "$HAVE_DB4O_8" = "yes"; then
+				HAVE_DB4O=yes
+        else
+				PKG_CHECK_MODULES([DB4O], [db4o], HAVE_DB4O=yes, HAVE_DB4O=no)
+        fi
 
         if test "x$enable_file" = "xyes" -a "x$HAVE_DB4O" = "xno"; then
 				AC_MSG_ERROR([File support explicitly requested, but dependencies not met])

=== modified file 'plugins/File/Makefile.am'
--- plugins/File/Makefile.am	2009-08-15 17:26:24 +0000
+++ plugins/File/Makefile.am	2012-01-08 18:50:09 +0000
@@ -1,6 +1,11 @@
 # Simple component buildsystem
 if HAVE_DB4O
 include $(top_srcdir)/build.plugins.rules.mk
+
+if HAVE_DB4O_8
+MCS_FLAGS += -d:HAVE_DB4O_8
+endif
+
 else
 EXTRA_DIST = $(FILES)
 endif

=== modified file 'plugins/File/src/FilePlugin.cs'
--- plugins/File/src/FilePlugin.cs	2009-08-08 05:25:34 +0000
+++ plugins/File/src/FilePlugin.cs	2012-01-08 18:50:09 +0000
@@ -127,7 +127,11 @@
                 odb = Db4oFactory.OpenFile (Path.Combine (Daemon.ConfigDirectory, "tracks.db"));
             }
 
+#if HAVE_DB4O_8
+            IObjectSet result = odb.QueryByExample (typeof (Track));
+#else
             IObjectSet result = odb.Get (typeof (Track));
+#endif
             log.DebugFormat ("{0} songs in database", result.Count);
             
             foreach (Track song in result) {
@@ -261,7 +265,11 @@
                 if (UpdateTrack (track, file)) {
                     db.AddTrack (track);   
                     trackHash[file] = track;
+#if HAVE_DB4O_8
+                    odb.Store (track);
+#else
                     odb.Set (track);
+#endif
                 }
             } catch {
             }

