=== modified file 'configure.ac'
--- configure.ac	2012-01-06 21:33:51 +0000
+++ configure.ac	2012-01-08 18:26:35 +0000
@@ -104,11 +104,11 @@
         PKG_CHECK_MODULES(GLADE_SHARP_20, glade-sharp-2.0)
         AC_SUBST(GLADE_SHARP_LIBS_20)
 
-        PKG_CHECK_MODULES(NDESK_DBUS_10, ndesk-dbus-1.0)
-        AC_SUBST([NDESK_DBUS_10_LIBS])
+        PKG_CHECK_MODULES(DBUS_10, dbus-sharp-1.0)
+        AC_SUBST([DBUS_10_LIBS])
 
-        PKG_CHECK_MODULES(NDESK_DBUS_GLIB_10, ndesk-dbus-glib-1.0)
-        AC_SUBST([NDESK_DBUS_GLIB_10_LIBS])
+        PKG_CHECK_MODULES(DBUS_GLIB_10, dbus-sharp-glib-1.0)
+        AC_SUBST([DBUS_GLIB_10_LIBS])
 
         PKG_CHECK_MODULES(X_SESSION, sm)
         AC_SUBST([X_SESSION_LIBS])

=== modified file 'plugins/Banshee/Makefile.am'
--- plugins/Banshee/Makefile.am	2009-08-15 17:26:24 +0000
+++ plugins/Banshee/Makefile.am	2012-01-08 18:26:35 +0000
@@ -17,8 +17,8 @@
 REFERENCES = \
 	$(BANSHEE_INDEXER_LIBS) \
 	$(LOG_4_NET_LIBS) \
-	$(NDESK_DBUS_10_LIBS) \
-	$(NDESK_DBUS_GLIB_10_LIBS) \
+	$(DBUS_10_LIBS) \
+	$(DBUS_GLIB_10_LIBS) \
 	$(NINI_LIBS) \
 	Mono.Posix \
 	System

=== modified file 'plugins/Banshee/src/BansheePlugin.cs'
--- plugins/Banshee/src/BansheePlugin.cs	2009-08-08 00:46:02 +0000
+++ plugins/Banshee/src/BansheePlugin.cs	2012-01-08 18:26:35 +0000
@@ -7,7 +7,7 @@
 using Banshee.Collection.Indexer.RemoteHelper;
 
 using DAAP;
-using NDesk.DBus;
+using DBus;
 using Nini;
 using log4net;
 

