diff --git a/trunk/indra/llimagej2coj/llimagej2coj.cpp b/trunk/indra/llimagej2coj/llimagej2coj.cpp
index be6b6c8..88a27f9 100644
--- a/trunk/indra/llimagej2coj/llimagej2coj.cpp
+++ b/trunk/indra/llimagej2coj/llimagej2coj.cpp
@@ -174,6 +174,8 @@ BOOL LLImageJ2COJ::decodeImpl(LLImageJ2C &base, LLImageRaw &raw_image, F32 decod
 			opj_image_destroy(image);
 		}
 
+		base.decodeFailed();
+
 		return TRUE; // done
 	}
 
@@ -184,7 +186,7 @@ BOOL LLImageJ2COJ::decodeImpl(LLImageJ2C &base, LLImageRaw &raw_image, F32 decod
 		{
 			// if we didn't get the discard level we're expecting, fail
 			opj_image_destroy(image);
-			base.mDecoding = FALSE;
+			base.decodeFailed();
 			return TRUE;
 		}
 	}
@@ -196,7 +198,8 @@ BOOL LLImageJ2COJ::decodeImpl(LLImageJ2C &base, LLImageRaw &raw_image, F32 decod
 		{
 			opj_image_destroy(image);
 		}
-			
+
+		base.decodeFailed();
 		return TRUE;
 	}
 
