--- indra/llimagej2coj/llimagej2coj.cpp	2011-09-21 08:19:12.000000000 +0800
+++ indra/llimagej2coj/llimagej2coj.fix.cpp	2011-09-21 08:29:36.000000000 +0800
@@ -42,7 +42,7 @@
 const char* fallbackEngineInfoLLImageJ2CImpl()
 {
 	static std::string version_string =
-		std::string("OpenJPEG: " OPENJPEG_VERSION ", Runtime: ")
+		std::string("OpenJPEG Runtime: ")
 		+ opj_version();
 	return version_string.c_str();
 }
