*** gnome-raw-thumbnailer-2.0.1/gnome-raw-thumbnailer.schemas	2005-02-18 19:35:40.000000000 +0000
--- gnome-raw-thumbnailer-2.0.1/gnome-raw-thumbnailer.schemas	2010-07-18 15:10:29.472501172 +0100
***************
*** 22,26 ****
--- 23,396 ----
                  <long></long>
              </locale>
          </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-adobe-dng/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-adobe-dng/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-adobe-dng/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-adobe-dng/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-canon-crw/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-canon-crw/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-canon-crw/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-canon-crw/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-canon-cr2/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-canon-cr2/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-canon-cr2/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-canon-cr2/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-fuji-raf/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-fuji-raf/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-fuji-raf/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-fuji-raf/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-kodak-dcr/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-kodak-dcr/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-kodak-dcr/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-kodak-dcr/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-kodak-k25/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-kodak-k25/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-kodak-k25/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-kodak-k25/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-kodak-kdc/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-kodak-kdc/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-kodak-kdc/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-kodak-kdc/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-minolta-mrw/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-minolta-mrw/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-minolta-mrw/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-minolta-mrw/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-nikon-nef/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-nikon-nef/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-nikon-nef/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-nikon-nef/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-olympus-orf/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-olympus-orf/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-olympus-orf/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-olympus-orf/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-panasonic-raw/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-panasonic-raw/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-panasonic-raw/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-panasonic-raw/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-pentax-pef/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-pentax-pef/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-pentax-pef/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-pentax-pef/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-sigma-x3f/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-sigma-x3f/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-sigma-x3f/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-sigma-x3f/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-sony-srf/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-sony-srf/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-sony-srf/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-sony-srf/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-sony-sr2/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-sony-sr2/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-sony-sr2/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-sony-sr2/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-sony-arw/enable</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-sony-arw/enable</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>bool</type>
+             <default>true</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+         <schema>
+             <key>/schemas/desktop/gnome/thumbnailers/image@x-sony-arw/command</key>
+             <applyto>/desktop/gnome/thumbnailers/image@x-sony-arw/command</applyto>
+             <owner>gnome-raw-thumbnailer</owner>
+             <type>string</type>
+             <default>gnome-raw-thumbnailer -s %s %u %o</default>
+             <locale name="C">
+                 <short></short>
+                 <long></long>
+             </locale>
+         </schema>
+ 
      </schemalist>
  </gconfschemafile>
