Index: cblrepo/haskell-nano-hmac/src/nano-hmac-0.2.0/nano-hmac.cabal
===================================================================
--- cblrepo.orig/haskell-nano-hmac/src/nano-hmac-0.2.0/nano-hmac.cabal
+++ cblrepo/haskell-nano-hmac/src/nano-hmac-0.2.0/nano-hmac.cabal
@@ -20,7 +20,7 @@ library
   else
     build-depends:     base < 3
   exposed-modules:     Data.Digest.OpenSSL.HMAC
-  ghc-options:         -Wall -Werror -O2 -fvia-C
+  ghc-options:         -Wall -O2
   extensions:          ForeignFunctionInterface, BangPatterns, CPP
   includes:            openssl/hmac.h
   extra-libraries:     crypto ssl
