diff --git a/ROX-Filer/src/pinboard.c b/ROX-Filer/src/pinboard.c
index 466930b..fcfde98 100644
--- a/ROX-Filer/src/pinboard.c
+++ b/ROX-Filer/src/pinboard.c
@@ -1511,11 +1511,11 @@ static void backdrop_from_xml(xmlNode *node)
 	if (style)
 	{
 		current_pinboard->backdrop_style =
-		  g_strcasecmp(style, "Tiled") == 0 ? BACKDROP_TILE :
-		  g_strcasecmp(style, "Scaled") == 0 ? BACKDROP_SCALE :
+		  g_strcasecmp(style, "Tile") == 0 ? BACKDROP_TILE :
+		  g_strcasecmp(style, "Scale") == 0 ? BACKDROP_SCALE :
+		  g_strcasecmp(style, "Stretch") == 0 ? BACKDROP_STRETCH :
+		  g_strcasecmp(style, "Centre") == 0 ? BACKDROP_CENTRE :
 		  g_strcasecmp(style, "Fit") == 0 ? BACKDROP_FIT :
-		  g_strcasecmp(style, "Stretched") == 0 ? BACKDROP_STRETCH :
-		  g_strcasecmp(style, "Centred") == 0 ? BACKDROP_CENTRE :
 		  g_strcasecmp(style, "Program") == 0 ? BACKDROP_PROGRAM :
 							     BACKDROP_NONE;
 		g_free(style);
@@ -1660,11 +1660,11 @@ static void pinboard_save(void)
 		tree = xmlNewTextChild(root, NULL, "backdrop",
 				current_pinboard->backdrop);
 		xmlSetProp(tree, "style",
-			style == BACKDROP_TILE   ? "Tiled" :
-			style == BACKDROP_CENTRE ? "Centred" :
-			style == BACKDROP_SCALE  ? "Scaled" :
+			style == BACKDROP_TILE   ? "Tile" :
+			style == BACKDROP_SCALE  ? "Scale" :
+			style == BACKDROP_STRETCH  ? "Stretch" :
+			style == BACKDROP_CENTRE ? "Centre" :
 			style == BACKDROP_FIT    ? "Fit" :
-			style == BACKDROP_STRETCH  ? "Stretched" :
 						   "Program");
 	}
 
diff --git a/ROX-Filer/src/remote.c b/ROX-Filer/src/remote.c
index b71a06a..abe1503 100644
--- a/ROX-Filer/src/remote.c
+++ b/ROX-Filer/src/remote.c
@@ -885,6 +885,7 @@ static xmlNodePtr rpc_SetBackdrop(GList *args)
 	    !g_strcasecmp(style, "Scale") ? BACKDROP_SCALE :
 	    !g_strcasecmp(style, "Stretch") ? BACKDROP_STRETCH :
 	    !g_strcasecmp(style, "Centre") ? BACKDROP_CENTRE :
+	    !g_strcasecmp(style, "Fit") ? BACKDROP_FIT :
 					     BACKDROP_NONE;
 
 	if (s == BACKDROP_NONE)
