Index: pidgin-2.5.1/libpurple/protocols/msn/switchboard.c
===================================================================
--- pidgin-2.5.1.orig/libpurple/protocols/msn/switchboard.c
+++ pidgin-2.5.1/libpurple/protocols/msn/switchboard.c
@@ -874,6 +874,17 @@ plain_msg(MsnCmdProc *cmdproc, MsnMessag
 	}
 #endif
 
+	/** For MSN Group. Deals with the "P4-Context" field */
+	if ((value = msn_message_get_attr(msg, "P4-Context")) != NULL)
+	{
+		char *body_enc_post;
+
+		body_enc_post = g_strdup_printf("<b>**%s**</b> %s", value, body_enc ? body_enc : "");
+
+		g_free(body_enc);
+		body_enc = body_enc_post;
+	}
+
 	if ((value = msn_message_get_attr(msg, "X-MMS-IM-Format")) != NULL)
 	{
 		char *pre, *post;
Index: pidgin-2.5.1/libpurple/protocols/msnp9/switchboard.c
===================================================================
--- pidgin-2.5.1.orig/libpurple/protocols/msnp9/switchboard.c
+++ pidgin-2.5.1/libpurple/protocols/msnp9/switchboard.c
@@ -864,6 +864,17 @@ plain_msg(MsnCmdProc *cmdproc, MsnMessag
 	}
 #endif
 
+	/** For MSN Group. Deals with "P4-Context" field. */
+	if ((value = msn_message_get_attr(msg, "P4-Context")) != NULL)
+	{
+		char *body_enc_post;
+
+		body_enc_post = g_strdup_printf("<b>**%s**</b> %s", value, body_enc ? body_enc : "");
+
+		g_feee(body_enc);
+		body_enc = body_enc_post;
+	}
+
 	if ((value = msn_message_get_attr(msg, "X-MMS-IM-Format")) != NULL)
 	{
 		char *pre, *post;
