--- domino/domino.cpp.orig	2007-03-14 12:42:21.000000000 +0100
+++ domino/domino.cpp	2007-03-23 18:41:26.000000000 +0100
@@ -4715,14 +4717,25 @@
 					
 				}
 				
+				QPixmap outlinePixmap = 0;
+				bool drawOutlinePixmap = false;
+				if(flags &QStyle::Style_HasFocus && focusIndicatorSettings.indicateFocus && !pixmap.isNull()) {
+					outlinePixmap = renderOutlineIcon(&pixmap);
+					drawOutlinePixmap = true;
+				}
 				
 				if (tb->iconTextMode() == KToolBar::IconOnly) // icon only
 				{
 					if( !pixmap.isNull())
 					{
+						
 						dx = ( tb->width() - pixmap.width() ) / 2;
 						dy = ( tb->height()-1 - pixmap.height() ) / 2;
 						
+						if(drawOutlinePixmap) {
+							p->drawPixmap( dx-1, dy-1, outlinePixmap );
+						}
+						
 						p->drawPixmap( dx, dy, pixmap );
 					}
 				}
@@ -4733,7 +4746,11 @@
 					{
 						dx = 4;
 						dy = ( tb->height()-1 - pixmap.height() ) / 2;
-	
+						
+						if(drawOutlinePixmap) {
+							p->drawPixmap( dx-1, dy-1, outlinePixmap );
+						}
+						
 						p->drawPixmap( dx, dy, pixmap );
 					}
 					if (!tb->textLabel().isNull())
@@ -4765,7 +4782,11 @@
 					{
 						dx = (tb->width() - pixmap.width()) / 2;
 						dy = (tb->height()-1 - fm.lineSpacing() - pixmap.height()) / 2;
-	
+						
+						if(drawOutlinePixmap) {
+							p->drawPixmap( dx-1, dy-1, outlinePixmap );
+						}
+						
 						p->drawPixmap( dx, dy, pixmap );
 					}
 	
@@ -4846,6 +4867,13 @@
 						else if(_highlightToolBtnIcon && toolbutton->hasMouse() && toolbutton->isEnabled())
 							pm = KImageEffect::fade(img, 0.25, white);
 						
+						QPixmap outlinePixmap = 0;
+						bool drawOutlinePixmap = false;
+						if(flags &QStyle::Style_HasFocus && focusIndicatorSettings.indicateFocus) {
+							outlinePixmap = renderOutlineIcon(&pm);
+							drawOutlinePixmap = true;
+						}
+						
 						if ( toolbutton->usesTextLabel() ) {
 							p->setFont( toolbutton->font() );
 							QRect pr = rect, tr = rect;
@@ -4857,18 +4885,28 @@
 								int fh = p->fontMetrics().height();
 								pr.addCoords( 0, 1, 0, -fh-3 );
 								tr.addCoords( 0, pr.bottom(), 0, -3 );
+								
+								if(drawOutlinePixmap)
+									drawItem( p, pr, AlignCenter, cg, TRUE, &outlinePixmap, QString::null );
+								
 								drawItem( p, pr, AlignCenter, cg, TRUE, &pm, QString::null );
 								alignment |= AlignCenter;
 							} else {
 								pr.setWidth( pm.width() + 8 );
 								tr.addCoords( pr.right(), 0, 0, 0 );
 								
+								if(drawOutlinePixmap)
+									drawItem( p, pr, AlignCenter, cg, TRUE, &outlinePixmap, QString::null );
+								
 								drawItem( p, pr, AlignCenter, cg, TRUE, &pm, QString::null );
 								alignment |= AlignLeft | AlignVCenter;
 							}
 							dominoDrawItem( p, tr, alignment, cg, enabled, 0, toolbutton->textLabel(), toolbutton->textLabel().length(), &btext, textEffectSettings.mode > 0 ? 1 : 0);
 						}
 						else {
+							if(drawOutlinePixmap)
+								drawItem( p, rect, AlignCenter, cg, TRUE, &outlinePixmap, QString::null );
+							
 							drawItem( p, rect, AlignCenter, cg, TRUE, &pm, QString::null );
 						}
 					}
