diff -Naur domino-0.4.orig/domino/eventfilter.cpp domino-0.4/domino/eventfilter.cpp
--- domino-0.4.orig/domino/eventfilter.cpp	2007-08-22 04:19:07.000000000 +0200
+++ domino-0.4/domino/eventfilter.cpp	2007-02-16 22:17:36.000000000 +0100
@@ -444,7 +444,7 @@
 			case 6/*QEvent::KeyPress*/: {
 				// it will be reseted in QLineEdit::mouseMoveEvent()
 				if(isLineEdit && !khtmlMode && isEnabled && !isReadOnly && !isDateTimeEditor)
-					static_cast<QWidget*>(obj)->setCursor(Qt::blankCursor);
+					static_cast<QWidget*>(obj)->setCursor(Qt::ibeamCursor);
 				return false;
 			}
 			case QEvent::ApplicationPaletteChange:
@@ -1687,7 +1687,7 @@
 			case 6/*QEvent::KeyPress*/: {
 				QTextEdit* te = dynamic_cast<QTextEdit*>(obj->parent());
 				if(te->isEnabled() && !te->isReadOnly())
-					static_cast<QWidget*>(obj)->setCursor(Qt::blankCursor);
+					static_cast<QWidget*>(obj)->setCursor(Qt::ibeamCursor);
 				return false;
 			}
 			default:
