--- domino/domino.cpp.orig	2007-03-14 12:42:21.000000000 +0100
+++ domino/domino.cpp	2007-03-14 12:42:44.000000000 +0100
@@ -538,8 +538,10 @@
 		minTabLength = 0;
 	
 	if(!strcmp(qApp->className(), "KApplication")) {
+		QString oldgroup = KGlobal::config()->group();
 		KGlobal::config()->setGroup("KDE");
 		macStyleBar = KGlobal::config()->readBoolEntry("macStyle", false);
+		KGlobal::config()->setGroup(oldgroup); // reset the group, do not interfere with the application
 		KApplicationMode = true;
 	}
 	else {
