diff -Naur vice-2.3-orig/src/gfxoutputdrv/ffmpegdrv.c vice-2.3/src/gfxoutputdrv/ffmpegdrv.c
--- vice-2.3-orig/src/gfxoutputdrv/ffmpegdrv.c	2011-11-28 22:41:20.204389694 -0500
+++ vice-2.3/src/gfxoutputdrv/ffmpegdrv.c	2011-11-28 22:47:48.396209074 -0500
@@ -342,7 +342,7 @@
 
     c = st->codec;
     c->codec_id = ffmpegdrv_fmt->audio_codec;
-    c->codec_type = CODEC_TYPE_AUDIO;
+    c->codec_type = AVMEDIA_TYPE_AUDIO;
     c->sample_fmt = SAMPLE_FMT_S16;
 
     /* put sample parameters */
@@ -370,7 +370,7 @@
         pkt.size = (*ffmpeglib.p_avcodec_encode_audio)(c, 
                         audio_outbuf, audio_outbuf_size, audio_in->buffer);
         pkt.pts = c->coded_frame->pts;
-        pkt.flags |= PKT_FLAG_KEY;
+        pkt.flags |= AV_PKT_FLAG_KEY;
         pkt.stream_index = audio_st->index;
         pkt.data = audio_outbuf;
 
@@ -559,7 +559,7 @@
 
     c = st->codec;
     c->codec_id = ffmpegdrv_fmt->video_codec;
-    c->codec_type = CODEC_TYPE_VIDEO;
+    c->codec_type = AVMEDIA_TYPE_VIDEO;
 
     /* put sample parameters */
     c->bit_rate = video_bitrate;
@@ -819,7 +819,7 @@
     if (ffmpegdrv_oc->oformat->flags & AVFMT_RAWPICTURE) {
         AVPacket pkt;
         (*ffmpeglib.p_av_init_packet)(&pkt);
-        pkt.flags |= PKT_FLAG_KEY;
+        pkt.flags |= AV_PKT_FLAG_KEY;
         pkt.stream_index = video_st->index;
         pkt.data = (uint8_t*)picture;
         pkt.size = sizeof(AVPicture);
@@ -840,7 +840,7 @@
             (*ffmpeglib.p_av_init_packet)(&pkt);
             pkt.pts = c->coded_frame->pts;
             if (c->coded_frame->key_frame)
-                pkt.flags |= PKT_FLAG_KEY;
+                pkt.flags |= AV_PKT_FLAG_KEY;
             pkt.stream_index = video_st->index;
             pkt.data = video_outbuf;
             pkt.size = out_size;