--- etc/config.py
+++ etc/config.py
@@ -3,8 +3,8 @@
 from caw import Caw
 from caw.widgets.spacer import Spacer
 from caw.widgets.text import Text
-from caw.widgets.battery import Battery
-from caw.widgets.volume import Volume
+#from caw.widgets.battery import Battery # new path: /sys/class/power_supply/
+#from caw.widgets.volume import Volume # looks for /dev/mixer --> OSS legacy stuff
 from caw.widgets.clock import Clock
 from caw.widgets.cpu import CPU
 from caw.widgets.systray import Systray
@@ -91,15 +91,15 @@
 
 widgets.append(Text(" :: "))
 
-widgets.append(Battery(normal_fg=0x9f78ff))
+#widgets.append(Battery(normal_fg=0x9f78ff))
 widgets.append(Text(" :: "))
 
-vol = Volume(device='PCM')
-widgets.append(vol)
-widgets.append(Text("% ", 0xaaaaaa, button5=vol.button5, button4=vol.button4))
-vol = Volume()
-widgets.append(vol)
-widgets.append(Text("% ", 0xaaaaaa, button5=vol.button5, button4=vol.button4))
+#vol = Volume(device='PCM')
+#widgets.append(vol)
+#widgets.append(Text("% ", 0xaaaaaa, button5=vol.button5, button4=vol.button4))
+#vol = Volume()
+#widgets.append(vol)
+#widgets.append(Text("% ", 0xaaaaaa, button5=vol.button5, button4=vol.button4))
 
 widgets.append(Text(":: "))
 
