--- README
+++ README
@@ -8,7 +8,7 @@ Dependencies
 libxcb 1.11
 xcb-proto 1.11
 xcb-util 0.3.4
-xpyb 1.3.1 OR xcffib 0.2.2
+xpyb 1.3.1 OR xcffib 0.2.3
 cairo 1.8.8
 pango 1.24.5
 
--- caw/caw.py
+++ caw/caw.py
@@ -358,9 +358,9 @@ class Caw:
             try:
                 event = self.connection.poll_for_event()
                 #print "Event:", type(event)
-                #print "OpCode:", event.type
+                #print "OpCode:", event.response_type
                 #print "Window:", getattr(event, 'window', None)
-                if event.type == 161:
+                if event.response_type == 161:
                     event = xproto.ClientMessageEvent(event)
                 for func in self.events[type(event)]:
                     func(event)
