diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 3422394..e1e983a 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -20,7 +20,7 @@
 project(btbb-python)
 
 # Check for python installation
-find_package(PythonInterp)
+find_package(PythonInterp 2)
 if(NOT PYTHONINTERP_FOUND)
 	return()
 else()
diff --git a/python/pcaptools/Makefile b/python/pcaptools/Makefile
index f76569e..df077db 100644
--- a/python/pcaptools/Makefile
+++ b/python/pcaptools/Makefile
@@ -18,7 +18,7 @@
 # Boston, MA 02110-1301, USA.
 
 INSTALL ?= /usr/bin/install
-PYTHON ?= /usr/bin/env python
+PYTHON ?= /usr/bin/env python2
 
 INSTALL_DIR ?= /usr/bin
 
diff --git a/python/pcaptools/btaptap b/python/pcaptools/btaptap
index 3b9eb72..dcb87c3 100755
--- a/python/pcaptools/btaptap
+++ b/python/pcaptools/btaptap
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2009 Joshua Wright, Michael Ossmann
 # 
diff --git a/python/pcaptools/pcapdump/pcapdump.py b/python/pcaptools/pcapdump/pcapdump.py
index 647fc60..4e164c4 100644
--- a/python/pcaptools/pcapdump/pcapdump.py
+++ b/python/pcaptools/pcapdump/pcapdump.py
@@ -17,6 +17,8 @@
 # the Free Software Foundation, Inc., 51 Franklin Street,
 # Boston, MA 02110-1301, USA.
 
+#!/usr/bin/env python2
+
 import struct
 import time
 
diff --git a/python/pcaptools/setup.py b/python/pcaptools/setup.py
index bd7a928..365c0a7 100755
--- a/python/pcaptools/setup.py
+++ b/python/pcaptools/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 pcapdump setup
 
diff --git a/python/pcaptools/setup.py.in b/python/pcaptools/setup.py.in
index 79196d0..414b9e5 100755
--- a/python/pcaptools/setup.py.in
+++ b/python/pcaptools/setup.py.in
@@ -1,4 +1,5 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
+
 """
 pcapdump setup
 
diff --git a/python/utils/encode_sw.py b/python/utils/encode_sw.py
index 6583abf..115887d 100755
--- a/python/utils/encode_sw.py
+++ b/python/utils/encode_sw.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # produce a sync word for a given LAP
 
diff --git a/python/utils/gen_barker_correct.py b/python/utils/gen_barker_correct.py
index f428def..40fe5a3 100755
--- a/python/utils/gen_barker_correct.py
+++ b/python/utils/gen_barker_correct.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Reverse the binary of thesebarker codes for the host code
 barkers = (0x0d, 0x72)
diff --git a/python/utils/gen_check_tables.py b/python/utils/gen_check_tables.py
index 0ae41df..10e9c26 100755
--- a/python/utils/gen_check_tables.py
+++ b/python/utils/gen_check_tables.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # (64,30) linear block code stuff
 
diff --git a/python/utils/le_whitening.py b/python/utils/le_whitening.py
index 6ce9a9b..172b220 100755
--- a/python/utils/le_whitening.py
+++ b/python/utils/le_whitening.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 num_channels = 40
 bits = []
diff --git a/wireshark/plugins/btatt/tools/make-dissector-reg.py b/wireshark/plugins/btatt/tools/make-dissector-reg.py
index 060460c..ba98105 100755
--- a/wireshark/plugins/btatt/tools/make-dissector-reg.py
+++ b/wireshark/plugins/btatt/tools/make-dissector-reg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Looks for registration routines in the protocol dissectors,
 # and assembles C code to call all the routines.
diff --git a/wireshark/plugins/btbb/tools/make-dissector-reg.py b/wireshark/plugins/btbb/tools/make-dissector-reg.py
index 060460c..ba98105 100755
--- a/wireshark/plugins/btbb/tools/make-dissector-reg.py
+++ b/wireshark/plugins/btbb/tools/make-dissector-reg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Looks for registration routines in the protocol dissectors,
 # and assembles C code to call all the routines.
diff --git a/wireshark/plugins/btle/tools/make-dissector-reg.py b/wireshark/plugins/btle/tools/make-dissector-reg.py
index 060460c..ba98105 100755
--- a/wireshark/plugins/btle/tools/make-dissector-reg.py
+++ b/wireshark/plugins/btle/tools/make-dissector-reg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Looks for registration routines in the protocol dissectors,
 # and assembles C code to call all the routines.
diff --git a/wireshark/plugins/btsm/tools/make-dissector-reg.py b/wireshark/plugins/btsm/tools/make-dissector-reg.py
index 060460c..ba98105 100755
--- a/wireshark/plugins/btsm/tools/make-dissector-reg.py
+++ b/wireshark/plugins/btsm/tools/make-dissector-reg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Looks for registration routines in the protocol dissectors,
 # and assembles C code to call all the routines.
