diff -urN greenbone-security-assistant-2.0.1.orig//CMakeLists.txt greenbone-security-assistant-2.0.1/CMakeLists.txt
--- greenbone-security-assistant-2.0.1.orig//CMakeLists.txt	2011-03-02 12:43:39.000000000 +0100
+++ greenbone-security-assistant-2.0.1/CMakeLists.txt	2011-05-15 10:46:09.893684513 +0200
@@ -353,7 +353,8 @@
 set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -Wl,-z,relro -Wl,-z,now")
 
 set (CMAKE_C_FLAGS_DEBUG        "${CMAKE_C_FLAGS_DEBUG}")
-set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -Werror")
+# set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -Werror")
+set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall")
 
 add_subdirectory (src)
 
