diff -ru lunatic-python-1.0 lunatic-python-1.0-lua5.1Only in lunatic-python-1.0-lua5.1: build
diff -ru lunatic-python-1.0/python.lua lunatic-python-1.0-lua5.1/python.lua
--- lunatic-python-1.0/python.lua	2003-12-12 20:37:57.000000000 -0800
+++ lunatic-python-1.0-lua5.1/python.lua	2008-06-26 22:48:34.000000000 -0700
@@ -1,6 +1,6 @@
 local path = os.getenv("LUA_SOPATH")
 if path then
-	func = loadlib(path.."/lua-python.so", "luaopen_python")
+	func = package.loadlib(path.."/lua-python.so", "luaopen_python")
 	if func then
 		func()
 		return
@@ -10,7 +10,7 @@
 local loaded = false
 for i = 10, 2, -1 do
 	for j = 10, 2, -1 do
-		func = loadlib(string.format(modmask, i, j), "luaopen_python")
+		func = package.loadlib(string.format(modmask, i, j), "luaopen_python")
 		if func then
 			loaded = true
 			func()
