--- libs/evoral/evoral/TimeConverter.hpp.old	2011-05-05 15:23:30.476045899 -0400
+++ libs/evoral/evoral/TimeConverter.hpp	2011-05-05 15:23:42.745648168 -0400
@@ -54,8 +54,10 @@
 /** A stub TimeConverter that simple statically casts between types. */
 template<typename A, typename B>
 class IdentityConverter : public TimeConverter<A,B> {
-	B to(A a)   const { return static_cast<B>(a); }
-	A from(B b) const { return static_cast<A>(b); }
+    public:
+        IdentityConverter(){}
+        B to(A a)   const { return static_cast<B>(a); }
+        A from(B b) const { return static_cast<A>(b); }
 };
 
 