--- modplugxmms-2.05/modplugxmms/archive/arch_gzip.cpp 2003-10-16 06:19:24.000000000 +0200
+++ modplugxmms-2.05.new/modplugxmms/archive/arch_gzip.cpp 2008-04-03 11:24:42.000000000 +0200
@@ -10,6 +10,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <stdio.h>
+#include <string.h>

 #include "arch_gzip.h"

--- modplugxmms-2.05/modplugxmms/archive/arch_rar.cpp 2003-10-16 06:24:06.000000000 +0200
+++ modplugxmms-2.05.new/modplugxmms/archive/arch_rar.cpp 2008-04-03 11:25:29.000000000 +0200
@@ -10,6 +10,8 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>

 #include "arch_rar.h"
 #include <iostream>
