diff -rup Evergreen-ILS-2.0.6/Open-ILS//src/apachemods/Makefile.am src/Evergreen-ILS-2.0.6/Open-ILS/src/apachemods/Makefile.am
--- Evergreen-ILS-2.0.6/Open-ILS//src/apachemods/Makefile.am	2009-10-29 01:00:26.000000000 +0000
+++ src/Evergreen-ILS-2.0.6/Open-ILS/src/apachemods/Makefile.am	2011-06-07 19:11:54.015880557 +0100
@@ -6,6 +6,7 @@
 
 AM_CFLAGS = -D_LARGEFILE64_SOURCE -Wall -I@abs_top_srcdir@/Open-ILS/include/ -I$(LIBXML2_HEADERS) -I$(APACHE2_HEADERS) -I$(APR_HEADERS) -I$(OPENSRF_HEADERS)
 AM_LDFLAGS = -L$(LIBDIR) -L$(OPENSRF_LIBS)
+AP_LIBEXECDIR = `$(APXS2) -q LIBEXECDIR`
 
 if BUILDAPACHEMODS
 OILSAPACHEINST = apachemods
@@ -14,10 +15,11 @@ endif
 install-exec-local: $(OILSAPACHEINST)
 
 apachemods:
+	$(MKDIR_P) $(DESTDIR)$(AP_LIBEXECDIR)
 	$(APXS2) -c $(AM_LDFLAGS) -lxml2 -lopensrf -lxslt -lexpat $(AM_CFLAGS) @srcdir@/mod_xmlent.c
-	$(APXS2) -i -a @srcdir@/mod_xmlent.la
+	$(APXS2) -i -S LIBEXECDIR=$(DESTDIR)$(AP_LIBEXECDIR) -a @srcdir@/mod_xmlent.la
 	$(APXS2) -c $(AM_LDFLAGS) -lxml2 -lopensrf -lxslt -lexpat $(AM_CFLAGS) @srcdir@/mod_idlchunk.c
-	$(APXS2) -i -a @srcdir@/mod_idlchunk.la
+	$(APXS2) -i -S LIBEXECDIR=$(DESTDIR)$(AP_LIBEXECDIR) -a @srcdir@/mod_idlchunk.la
 
 clean-local:
 	rm -f @srcdir@/mod_xmlent.la @srcdir@/mod_xmlent.lo @srcdir@/mod_xmlent.slo
diff -rup Evergreen-ILS-2.0.6/Open-ILS//src/Makefile.am src/Evergreen-ILS-2.0.6/Open-ILS/src/Makefile.am
--- Evergreen-ILS-2.0.6/Open-ILS//src/Makefile.am	2010-10-20 14:26:37.000000000 +0100
+++ src/Evergreen-ILS-2.0.6/Open-ILS/src/Makefile.am	2011-06-07 19:27:23.404065099 +0100
@@ -149,7 +149,7 @@ EXTRA_DIST = @srcdir@/perlmods @srcdir@/
 
 # Install header files
 
-oilsincludedir = $(DESTDIR)@includedir@/openils
+oilsincludedir = @includedir@/openils
 headsdir = @top_srcdir@/Open-ILS/include/openils
 oilsinclude_HEADERS = $(headsdir)/idl_fieldmapper.h $(headsdir)/oils_constants.h $(headsdir)/oils_event.h $(headsdir)/oils_idl.h $(headsdir)/oils_utils.h
 
@@ -168,17 +168,17 @@ uninstall-hook:
 ilscore-install:
 	@echo $@
 	@echo "Installing Perl modules"
-	$(MKDIR_P) $(perldir)
-	$(MKDIR_P) $(TEMPLATEDIR)
+	$(MKDIR_P) $(DESTDIR)$(perldir)
+	$(MKDIR_P) $(DESTDIR)$(TEMPLATEDIR)
-	cp -r @srcdir@/perlmods/* $(perldir)
-	cp -r @srcdir@/templates/marc $(TEMPLATEDIR)
-	cp -r @srcdir@/templates/password-reset $(TEMPLATEDIR)
+	cp -r @srcdir@/perlmods/* $(DESTDIR)$(perldir)
+	cp -r @srcdir@/templates/marc $(DESTDIR)$(TEMPLATEDIR)
+	cp -r @srcdir@/templates/password-reset $(DESTDIR)$(TEMPLATEDIR)
 	sed -i 's|SYSCONFDIR|@sysconfdir@|g' '$(DESTDIR)@libdir@/perl5/OpenILS/WWW/Web.pm'
 	sed -i 's|SYSCONFDIR|@sysconfdir@|g' '$(DESTDIR)@libdir@/perl5/OpenILS/WWW/Method.pm'
-	@echo "Installing string templates to $(TEMPLATEDIR)"
-	$(MKDIR_P) $(TEMPLATEDIR)
-	$(MKDIR_P) $(datadir)/overdue/
+	@echo "Installing string templates to $(DESTDIR)$(TEMPLATEDIR)"
+	$(MKDIR_P) $(DESTDIR)$(TEMPLATEDIR)
+	$(MKDIR_P) $(DESTDIR)$(datadir)/overdue/
-	cp -r @srcdir@/templates/strings $(TEMPLATEDIR)
+	cp -r @srcdir@/templates/strings $(DESTDIR)$(TEMPLATEDIR)
 	sed -i 's|LOCALSTATEDIR|@localstatedir@|g' '$(DESTDIR)@sysconfdir@/oils_sip.xml.example'
 	sed -i 's|SYSCONFDIR|@sysconfdir@|g' '$(DESTDIR)@sysconfdir@/oils_sip.xml.example'
 	sed -i 's|LOCALSTATEDIR|@localstatedir@|g' '$(DESTDIR)@sysconfdir@/opensrf_core.xml.example'
diff -rup Evergreen-ILS-2.0.6/Open-ILS//web/Makefile.am src/Evergreen-ILS-2.0.6/Open-ILS/web/Makefile.am
--- Evergreen-ILS-2.0.6/Open-ILS//web/Makefile.am	2010-09-13 22:18:45.000000000 +0100
+++ src/Evergreen-ILS-2.0.6/Open-ILS/web/Makefile.am	2011-06-07 19:31:59.562170024 +0100
@@ -11,1 +11,1 @@
-reportsdir = $(DESTDIR)$(WEBDIR)/reports/
+reportsdir = $(WEBDIR)/reports/
@@ -62,7 +62,7 @@ offline-install:
 	@echo "Installing offline CGIs to $(CGIDIR)/offline";
 	$(MKDIR_P) $(TMP)
 	$(MKDIR_P) $(DESTDIR)$(CGIDIR)/offline;
-	$(MKDIR_P) $(datadir)/offline;
+	$(MKDIR_P) $(DESTDIR)$(datadir)/offline;
 	perl -pe "s{##CONFIG##}{@sysconfdir@}" < @top_srcdir@/Open-ILS/src/offline/offline.pl > $(TMP)/offline.pl;
 	cp $(TMP)/offline.pl $(DESTDIR)$(CGIDIR)/offline/
 	chmod +x $(DESTDIR)$(CGIDIR)/offline/offline.pl
diff -rup Evergreen-ILS-2.0.6/Open-ILS//xul/staff_client/Makefile.am src/Evergreen-ILS-2.0.6/Open-ILS/xul/staff_client/Makefile.am
--- Evergreen-ILS-2.0.6/Open-ILS//xul/staff_client/Makefile.am	2010-03-24 19:43:49.000000000 +0000
+++ src/Evergreen-ILS-2.0.6/Open-ILS/xul/staff_client/Makefile.am	2011-06-07 19:49:14.484247782 +0100
@@ -109,9 +109,9 @@ server-xul: build
-	mkdir -p $(WEBDIR)
-	mkdir -p $(WEBDIR)/xul/
+	mkdir -p $(DESTDIR)$(WEBDIR)
+	mkdir -p $(DESTDIR)$(WEBDIR)/xul/
 	@echo "BUILD_ID = $(STAFF_CLIENT_BUILD_ID)"
-	@echo "Copying xul into $(WEBDIR)/xul/$(STAFF_CLIENT_BUILD_ID)"
-	mkdir -p "$(WEBDIR)/xul/$(STAFF_CLIENT_BUILD_ID)"
-	cp -R @top_srcdir@/Open-ILS/xul/staff_client/build/server "${WEBDIR}/xul/${STAFF_CLIENT_BUILD_ID}/"
+	@echo "Copying xul into $(DESTDIR)$(WEBDIR)/xul/$(STAFF_CLIENT_BUILD_ID)"
+	mkdir -p "$(DESTDIR)$(WEBDIR)/xul/$(STAFF_CLIENT_BUILD_ID)"
+	cp -R @top_srcdir@/Open-ILS/xul/staff_client/build/server "$(DESTDIR)${WEBDIR}/xul/${STAFF_CLIENT_BUILD_ID}/"
 
 compress-javascript: build
 	@echo "Size of build/ before compression = " `du -sh build/`
