--- wiLink-2.3.3.orig/src/qnetio/CMakeLists.txt	2012-05-31 16:39:59.000000000 +0200
+++ wiLink-2.3.3/src/qnetio/CMakeLists.txt	2012-06-02 15:01:25.204627061 +0200
@@ -2,7 +2,7 @@
 include(FindPkgConfig)
 find_package(Qt4 REQUIRED)
 
-set(PLUGIN_INSTALL_DIR lib/qt4/plugins CACHE STRING "Plugins installation directory.")
+set(PLUGIN_INSTALL_DIR lib/qt/plugins CACHE STRING "Plugins installation directory.")
 
 add_subdirectory(src)
 if(NOT QNETIO_DONT_USE_TESTS)
