--- wiLink-2.3.3.orig/CMakeLists.txt	2012-05-31 16:39:59.000000000 +0200
+++ wiLink-2.3.3/CMakeLists.txt	2012-06-02 14:24:25.343630768 +0200
@@ -18,7 +18,7 @@
 pkg_check_modules(QXMPP qxmpp REQUIRED)
 
 # Extra runtime dependencies
-set(WILINK_IMPORTS_DIR lib/qt4/imports)
+set(WILINK_IMPORTS_DIR lib/qt/imports)
 if(APPLE)
     set(WILINK_EXTRA_FRAMEWORKS Growl)
     set(WILINK_EXTRA_IMPORTS QtWebKit)
