--- configure.ac	2013-05-17 01:58:15.635997393 +0900
+++ configure.ac	2013-05-17 02:00:07.016021418 +0900
@@ -5,7 +5,7 @@
 AC_CANONICAL_HOST
 AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_DEFINE_UNQUOTED(PLATFORM, "$ac_cv_host",
 		   [the type of system on which the package will run.])
@@ -16,7 +16,7 @@
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 AC_MINIX
-AM_C_PROTOTYPES
+AC_C_PROTOTYPES
 AC_SYS_LARGEFILE
 dnl Workaround for Zsh bug? on Cygwin:
 dnl
