--- tresor-patch-3.8.2_aesni	2014-08-11 20:46:22.000000000 +0100
+++ tresor-patch-3.14.16_aesni	2014-08-13 11:04:10.688477672 +0100
@@ -22,11 +22,11 @@
 +++ b/Makefile
 @@ -1,7 +1,7 @@
  VERSION = 3
- PATCHLEVEL = 8
- SUBLEVEL = 2
+ PATCHLEVEL = 14
+ SUBLEVEL = 16
 -EXTRAVERSION =
 +EXTRAVERSION = -tresor0.5
- NAME = Unicycling Gorilla
+ NAME = Remembering Coco
  
  # *DOCUMENTATION*
 diff --git a/arch/x86/crypto/Makefile b/arch/x86/crypto/Makefile
@@ -41,10 +41,11 @@
  obj-$(CONFIG_CRYPTO_GHASH_CLMUL_NI_INTEL) += ghash-clmulni-intel.o
  
  obj-$(CONFIG_CRYPTO_CRC32C_INTEL) += crc32c-intel.o
-@@ -48,6 +49,7 @@ serpent-sse2-x86_64-y := serpent-sse2-x86_64-asm_64.o serpent_sse2_glue.o
- serpent-avx-x86_64-y := serpent-avx-x86_64-asm_64.o serpent_avx_glue.o
+@@ -48,7 +49,8 @@ endif
+ endif
  
  aesni-intel-y := aesni-intel_asm.o aesni-intel_glue.o fpu.o
+ aesni-intel-$(CONFIG_64BIT) += aesni-intel_avx-x86_64.o
 +tresor-y := tresor_asm.o tresor_glue.o tresor_key.o
  ghash-clmulni-intel-y := ghash-clmulni-intel_asm.o ghash-clmulni-intel_glue.o
  sha1-ssse3-y := sha1_ssse3_asm.o sha1_ssse3_glue.o
