*** guake-0.4.4/data/guake.schemas.old	2012-10-01 13:18:59.319556631 -0500
--- guake-0.4.4/data/guake.schemas	2012-10-01 13:17:49.052889426 -0500
***************
*** 450,455 ****
--- 450,492 ----
              </locale>
          </schema>
  
+         <schema>
+             <key>/schemas/apps/guake/keybindings/local/fontsize_decrease</key>
+             <applyto>/apps/guake/keybindings/local/fontsize_decrease</applyto>
+             <owner>guake</owner>
+             <type>string</type>
+             <default>&lt;Control&gt;minus</default>
+             <locale name="C">
+                 <short>Decrease fontsize.</short>
+                 <long>Decrease fontsize by 2 units.</long>
+             </locale>
+         </schema>
+ 
+        <schema>
+             <key>/schemas/apps/guake/keybindings/local/fontsize_increase</key>
+             <applyto>/apps/guake/keybindings/local/fontsize_increase</applyto>
+             <owner>guake</owner>
+             <type>string</type>
+             <default>&lt;Control&gt;plus</default>
+             <locale name="C">
+                 <short>Increase fontsize.</short>
+                 <long>Increase fontsize by 2 units.</long>
+             </locale>
+         </schema>
+ 
+ 		<schema>
+             <key>/schemas/apps/guake/keybindings/local/fontsize_reset</key>
+             <applyto>/apps/guake/keybindings/local/fontsize_reset</applyto>
+             <owner>guake</owner>
+             <type>string</type>
+             <default>&lt;Control&gt;0</default>
+             <locale name="C">
+                 <short>Set the default fontsize.</short>
+                 <long>Set the default fontsize.</long>
+             </locale>
+         </schema>
+ 
+ 
          <!-- These options are not shown in guake-prefs window to
               avoid bloating that pretty ui =) -->
  
***************
*** 478,484 ****
              <default>true</default>
              <locale name="C">
                  <short>If guake must show a control to make resizing more handful</short>
!                 <long>When true, guake shows a resizer on its bottom to allow user 
                  resize guake.</long>
              </locale>
          </schema>
--- 515,521 ----
              <default>true</default>
              <locale name="C">
                  <short>If guake must show a control to make resizing more handful</short>
!                 <long>When true, guake shows a resizer on its bottom to allow user
                  resize guake.</long>
              </locale>
          </schema>
