*** guake-0.4.4/src/prefs.py.old	2012-10-01 13:23:04.262891846 -0500
--- guake-0.4.4/src/prefs.py	2012-10-01 13:20:48.772890808 -0500
***************
*** 84,90 ****
                 'label': 'Copy text to clipboard'},
                {'key': LKEY('clipboard_paste'),
                 'label': 'Paste text from clipboard'},
!               ]}
      ]
  
  PALETTES = [
--- 84,100 ----
                 'label': 'Copy text to clipboard'},
                {'key': LKEY('clipboard_paste'),
                 'label': 'Paste text from clipboard'},
!               ]},
! 
!     {'label': _('Font size'),
! 		'keys': [{'key': LKEY('fontsize_increase'),
! 				'label': _('Increase fontsize')},
! 				{'key': LKEY('fontsize_decrease'),
! 				'label': _('Decrease fontsize')},
! 				{'key': LKEY('fontsize_reset'),
! 				'label': _('Reset default fontsize')},
!                ]}
! 
      ]
  
  PALETTES = [
***************
*** 392,399 ****
          palette_index = combo.get_active()
          if palette_index == 4:
              return
!         self.client.set_string(KEY('/style/font/palette'), 
!             PALETTES[palette_index]) 
          self.set_palette_colors(PALETTES[palette_index])
  
      def on_palette_color_set(self, btn):
--- 402,409 ----
          palette_index = combo.get_active()
          if palette_index == 4:
              return
!         self.client.set_string(KEY('/style/font/palette'),
!             PALETTES[palette_index])
          self.set_palette_colors(PALETTES[palette_index])
  
      def on_palette_color_set(self, btn):
***************
*** 415,421 ****
          for i in range(len(PALETTES)):
              if palette == PALETTES[i]:
                  self.get_widget('palette_name').set_active(i)
!     
      def set_palette_colors(self, palette):
          """Updates the color buttons with the given palette
          """
--- 425,431 ----
          for i in range(len(PALETTES)):
              if palette == PALETTES[i]:
                  self.get_widget('palette_name').set_active(i)
! 
      def set_palette_colors(self, palette):
          """Updates the color buttons with the given palette
          """
***************
*** 532,538 ****
          value = self.client.get_string(KEY('/style/font/palette'))
          self.set_palette_name(value)
          self.set_palette_colors(value)
!     
          # background image
          value = self.client.get_string(KEY('/style/background/image'))
          if os.path.isfile(value or ''):
--- 542,548 ----
          value = self.client.get_string(KEY('/style/font/palette'))
          self.set_palette_name(value)
          self.set_palette_colors(value)
! 
          # background image
          value = self.client.get_string(KEY('/style/background/image'))
          if os.path.isfile(value or ''):
