Index: gtk+3.0-3.1.16/gtk/gtk.symbols
===================================================================
--- gtk+3.0-3.1.16.orig/gtk/gtk.symbols	2011-09-05 08:52:11.049554717 +0200
+++ gtk+3.0-3.1.16/gtk/gtk.symbols	2011-09-05 08:52:13.489554730 +0200
@@ -3859,6 +3859,7 @@
 gtk_window_unstick
 gtk_wrap_mode_get_type
 ubuntu_gtk_menu_shell_activate_mnemonic
+ubuntu_gtk_widget_set_has_grab
 ubuntu_menu_proxy_get_type
 ubuntu_menu_proxy_get
 ubuntu_menu_proxy_insert
Index: gtk+3.0-3.1.16/gtk/gtkwidget.c
===================================================================
--- gtk+3.0-3.1.16.orig/gtk/gtkwidget.c	2011-09-05 08:52:02.109554655 +0200
+++ gtk+3.0-3.1.16/gtk/gtkwidget.c	2011-09-05 08:52:13.499554731 +0200
@@ -14539,3 +14539,11 @@
 
   return widget->priv->context;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: gtk+3.0-3.1.16/gtk/gtkwidget.h
===================================================================
--- gtk+3.0-3.1.16.orig/gtk/gtkwidget.h	2011-09-05 08:52:02.129554657 +0200
+++ gtk+3.0-3.1.16/gtk/gtkwidget.h	2011-09-05 08:52:13.509554732 +0200
@@ -950,6 +950,9 @@
 GtkWidgetPath *   gtk_widget_get_path (GtkWidget *widget);
 
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
