diff --git a/Makefile b/Makefile
index f99e768..160940c 100755
--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,7 @@ DEPDIR = ./deps
 
 
 # Set the install directory for the executable
-INSTALLDIR = /usr/local/bin
+INSTALLDIR = ${DESTDIR}
 
 
 # Set the compilation flags and such
diff --git a/src/OBPager.h b/src/OBPager.h
index 733eda9..02ed7b9 100755
--- a/src/OBPager.h
+++ b/src/OBPager.h
@@ -30,11 +30,11 @@
 
 // And include the Xlib stuff we need
 
-#include <Xlib.h>
-#include <Xutil.h>
-#include <Xos.h>
-#include <Xatom.h>
-#include <shape.h>
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+#include <X11/Xos.h>
+#include <X11/Xatom.h>
+#include <X11/extensions/shape.h>
 
 
 // Also include the custom Xlib "helper" classes we use to manage ownership of X resources, etc.
@@ -42,6 +42,10 @@
 #include "XHelperClasses.h"
 
 
+// missing libs
+#include <stdlib.h>
+#include <memory>
+
 // During development, it's nice to generate a lot of debug output
 
 //#define __DEBUG__
diff --git a/src/XHelperClasses.h b/src/XHelperClasses.h
index 3a46866..71074e5 100755
--- a/src/XHelperClasses.h
+++ b/src/XHelperClasses.h
@@ -23,11 +23,11 @@
 #include <iostream>
 #include <string>
 
-#include <Xlib.h>
-#include <Xutil.h>
-#include <Xos.h>
-#include <Xatom.h>
-#include <shape.h>
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+#include <X11/Xos.h>
+#include <X11/Xatom.h>
+#include <X11/extensions/shape.h>
 
 
 //#define __DEBUG__
diff --git a/src/main.cc b/src/main.cc
index 5725b6d..e5ca07f 100755
--- a/src/main.cc
+++ b/src/main.cc
@@ -27,6 +27,9 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+// missing lib
+#include <errno.h>
+
 
 // Include the headers for the pager class and also the spiffy custom exception class
 
