diff -Naur synapse-0.2.8.2.orig/src/plugins/system-management.vala synapse-0.2.8.2/src/plugins/system-management.vala
--- synapse-0.2.8.2.orig/src/plugins/system-management.vala	2011-12-20 23:33:28.726533259 +0100
+++ synapse-0.2.8.2/src/plugins/system-management.vala	2011-12-20 23:42:00.399853310 +0100
@@ -129,7 +129,7 @@
           {
             yield dbus_interface.about_to_sleep ();
           }
-          catch (DBus.Error not_there_error) { }
+          catch (Error not_there_error) { }
           // yea kinda nasty
           GnomeScreenSaverPlugin.lock_screen ();
           // wait 2 seconds
@@ -199,7 +199,7 @@
           {
             yield dbus_interface.about_to_sleep ();
           }
-          catch (DBus.Error not_there_error) { }
+          catch (Error not_there_error) { }
           // yea kinda nasty
           GnomeScreenSaverPlugin.lock_screen ();
           // wait 2 seconds
