--- include/allegrogl/GLext/gl_ext_api.h	2007-11-10 20:14:21.000000000 +0100
+++ include/allegrogl/GLext/gl_ext_api.h	2009-01-08 22:17:46.000000000 +0100
@@ -1822,11 +1822,11 @@
 AGL_API(void, BindBufferOffsetNV,(GLenum, GLuint, GLuint, GLintptr))
 AGL_API(void, BindBufferBaseNV,  (GLenum, GLuint, GLuint))
 AGL_API(void, TransformFeedbackAttribsNV, (GLsizei, const GLint *, GLenum))
 AGL_API(void, TransformFeedbackVaryingsNV,(GLuint, GLsizei, const GLint *, GLenum))
 AGL_API(void, BeginTransformFeedbackNV,   (GLenum))
-AGL_API(void, EndTransformFeedbackNV,     (GLvoid))
+AGL_API(void, EndTransformFeedbackNV,     (void))
 AGL_API(GLint, GetVaryingLocationNV, (GLuint, const GLchar *))
 AGL_API(void, GetActiveVaryingNV,    (GLuint, GLuint, GLsizei, GLsizei *, GLsizei *, GLenum *, GLchar *))
 AGL_API(void, ActiveVaryingNV,       (GLuint, const GLchar *))
 AGL_API(void, GetTransformFeedbackVaryingNV, (GLuint, GLuint, GLint *))
 #if !defined AGL_EXT_draw_buffers2
