--- make/makefile.unx	2007-11-10 20:33:06.000000000 +0100
+++ make/makefile.unx	2009-01-08 23:08:43.000000000 +0100
@@ -43,10 +43,6 @@
 LIB_PATH_U = @LIB_PATH_U@
 LIB_LINKS_U = @LIB_LINKS_U@
 
-define LDCONFIG
-   ldconfig    # works only if you are root
-endef
-
 HEADERS += include/alleggl_config.h
 PLATFORM_OBJECT = $(addprefix $(OBJ_DIR)/, $(PLATFORM_SOURCE:.c=.o))
 GENERAL_LIB_OBJS = $(addprefix $(OBJ_DIR)/, $(notdir $(C_SOURCE:.c=.o)))
