diff -Naur awn-extras-build.orig/applets/maintained/notification-daemon/daemon.c awn-extras-build/applets/maintained/notification-daemon/daemon.c
--- awn-extras-build.orig/applets/maintained/notification-daemon/daemon.c	2011-03-17 19:09:06.000000000 +0100
+++ awn-extras-build/applets/maintained/notification-daemon/daemon.c	2011-03-17 19:10:34.236666679 +0100
@@ -1487,7 +1487,15 @@
   if (fork() == 0)
   {
     notify_init("notify-send");
-    notify = notify_notification_new(summary, body, icon_str, NULL);
+    #ifdef NOTIFY_CHECK_VERSION
+    #if NOTIFY_CHECK_VERSION (0, 7, 0)
+        notify = notify_notification_new(summary, body, icon_str);
+    #else
+        notify = notify_notification_new(summary, body, icon_str, NULL);
+    #endif
+    #else
+        notify = notify_notification_new(summary, body, icon_str, NULL);
+    #endif
     notify_notification_set_category(notify, type);
     notify_notification_set_urgency(notify, urgency);
     notify_notification_set_timeout(notify, expire_timeout);
