diff -r -u a/app/Makefile.am b/app/Makefile.am
--- a/app/Makefile.am	2014-01-13 10:37:07.000000000 +0100
+++ b/app/Makefile.am	2014-01-16 13:20:29.797864311 +0100
@@ -21,7 +21,7 @@
 seafile_LDADD =  @CCNET_LIBS@ \
 	$(top_builddir)/lib/libseafile.la \
 	$(top_builddir)/lib/libseafile_common.la \
-	@SEARPC_LIBS@ \
+	@SEARPC_LIBS@ @JANSSON_LIBS@ \
 	@GLIB2_LIBS@  @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3
 
 seafile_LDFALGS = @STATIC_COMPILE@ @CONSOLE@
@@ -29,7 +29,7 @@
 if COMPILE_SERVER
 seafserv_tool_SOURCES = seafserv-tool.c
 seafserv_tool_LDADD =  @CCNET_LIBS@ \
-	$(top_builddir)/lib/libseafile.la @SEARPC_LIBS@ \
+	$(top_builddir)/lib/libseafile.la @SEARPC_LIBS@ @JANSSON_LIBS@ \
 	@GLIB2_LIBS@  @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3
 
 seafserv_tool_LDFLAGS = @STATIC_COMPILE@ @CONSOLE@ @SERVER_PKG_RPATH@
diff -r -u a/configure.ac b/configure.ac
--- a/configure.ac	2014-01-13 10:37:07.000000000 +0100
+++ b/configure.ac	2014-01-16 13:20:29.797864311 +0100
@@ -233,6 +233,7 @@
 GLIB_REQUIRED=2.16.0
 CCNET_REQUIRED=0.9.3
 SEARPC_REQUIRED=1.0
+JANSSON_REQUIRED=2.2.1
 ZDB_REQUIRED=2.10
 #LIBNAUTILUS_EXTENSION_REQUIRED=2.30.1
 CURL_REQUIRED=7.17
@@ -254,6 +255,11 @@
 AC_SUBST(SEARPC_CFLAGS)
 AC_SUBST(SEARPC_LIBS)
 
+PKG_CHECK_MODULES(JANSSON, [jansson >= $JANSSON_REQUIRED])
+AC_SUBST(JANSSON_CFLAGS)
+AC_SUBST(JANSSON_LIBS)
+
+
 PKG_CHECK_MODULES(LIBEVENT, [libevent >= $LIBEVENT_REQUIRED])
 AC_SUBST(LIBEVENT_CFLAGS)
 AC_SUBST(LIBEVENT_LIBS)
diff -r -u a/daemon/Makefile.am b/daemon/Makefile.am
--- a/daemon/Makefile.am	2014-01-13 10:37:07.000000000 +0100
+++ b/daemon/Makefile.am	2014-01-16 13:23:24.359982579 +0100
@@ -139,7 +139,7 @@
 	@GLIB2_LIBS@  @GOBJECT_LIBS@ -lssl @LIB_RT@ @LIB_UUID@ -lsqlite3 -levent \
 	$(top_builddir)/common/cdc/libcdc.la \
 	$(top_builddir)/common/index/libindex.la ${LIB_WS32} \
-	@SEARPC_LIBS@ @CCNET_LIBS@ @GNOME_KEYRING_LIBS@
+	@SEARPC_LIBS@ @CCNET_LIBS@ @GNOME_KEYRING_LIBS@ @JANSSON_LIBS@
 
 seaf_daemon_LDFLAGS = @STATIC_COMPILE@ @CONSOLE@
 
