--- libopencm3_orig/lib/cm3/vector.c	2012-11-28 10:25:53.533216539 +0100
+++ libopencm3/lib/cm3/vector.c	2012-11-28 10:27:39.162372288 +0100
@@ -25,7 +25,8 @@
 /* load the weak symbols for IRQ_HANDLERS */
 #include "../dispatch/vector_nvic.c"
 
-#define WEAK __attribute__ ((weak))
+#define WEAK  __attribute__ ((weak))
+#define NAKED __attribute__ ((naked))
 
 /* Symbols exported by the linker script(s): */
 extern unsigned _data_loadaddr, _data, _edata, _ebss, _stack;
@@ -63,7 +64,7 @@
 	}
 };
 
-void WEAK reset_handler(void)
+void WEAK NAKED reset_handler(void)
 {
 	volatile unsigned *src, *dest;
 
