--- data/lxdm.conf.orig	2012-05-04 01:42:30.103771861 -0300
+++ data/lxdm.conf	2012-05-04 01:43:00.303772821 -0300
@@ -31,10 +31,10 @@
 
 [display]
 ## gtk theme used by greeter
-gtk_theme=Clearlooks
+gtk_theme=Industrial
 
 ## background of the greeter
-bg=/usr/share/backgrounds/default.png
+#bg=/usr/share/backgrounds/default.png
 
 ## if show bottom pane
 bottom_pane=1
@@ -46,7 +46,7 @@
 keyboard=0
 
 ## the theme of greeter
-theme=Industrial
+theme=Archlinux
 
 [input]
 
