commit 1505f94ea75fe0e571bf55db7aaca3a533a9a475
Author: Sergio Correia <sergio.correia@openbossa.org>
Date:   Wed Oct 23 02:00:36 2013 -0300

    Use isASCIISpace instead of ctype's isspace in gtest

diff --git a/Source/ThirdParty/gtest/include/gtest/internal/gtest-internal.h b/Source/ThirdParty/gtest/include/gtest/internal/gtest-internal.h
index 31a66e9..8f69a95 100644
--- a/Source/ThirdParty/gtest/include/gtest/internal/gtest-internal.h
+++ b/Source/ThirdParty/gtest/include/gtest/internal/gtest-internal.h
@@ -46,7 +46,7 @@
 #include <unistd.h>
 #endif  // GTEST_OS_LINUX
 
-#include <ctype.h>
+#include <wtf/ASCIICType.h>
 #include <string.h>
 #include <iomanip>
 #include <limits>
@@ -647,7 +647,7 @@ inline const char* SkipComma(const char* str) {
   if (comma == NULL) {
     return NULL;
   }
-  while (isspace(*(++comma))) {}
+  while (isASCIISpace(*(++comma))) {}
   return comma;
 }

