diff -Nur mozilla-release.orig/toolkit/modules/CertUtils.jsm mozilla-release/toolkit/modules/CertUtils.jsm
--- mozilla-release.orig/toolkit/modules/CertUtils.jsm	2014-10-11 07:06:48.000000000 -0200
+++ mozilla-release/toolkit/modules/CertUtils.jsm	2014-10-16 03:44:30.226448002 -0200
@@ -174,7 +174,9 @@
 }
 
 function isBuiltinToken(tokenName) {
-  return tokenName == "Builtin Object Token";
+  return tokenName == "Builtin Object Token" ||
+         tokenName == "Default Trust" ||
+         tokenName == "System Trust";
 }
 
 /**