--- iceweasel-9.0.1/browser/app/Makefile.in~	2011-12-25 16:19:29.000000000 -0500
+++ iceweasel-9.0.1/browser/app/Makefile.in	2011-12-25 16:19:55.000000000 -0500
@@ -257,7 +257,7 @@
 else
 ifdef LIBXUL_SDK
 libs::
-	cp $(LIBXUL_DIST)/bin/$(XULRUNNER_STUB_NAME)$(BIN_SUFFIX) $(DIST)/bin/firefox$(BIN_SUFFIX)
+	cp $(LIBXUL_DIST)/bin/$(XULRUNNER_STUB_NAME)$(BIN_SUFFIX) $(DIST)/bin/iceweasel$(BIN_SUFFIX)
 endif
 endif