--- nanomsg.orig/src/core/global.c	2014-03-30 16:05:30.927159640 +0200
+++ nanomsg/src/core/global.c	2014-03-30 16:05:47.547228934 +0200
@@ -75,6 +75,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdlib.h>
+#include <pthread.h>
 #include <time.h>
 
 #if defined NN_HAVE_WINDOWS
@@ -864,11 +865,7 @@ static void nn_global_submit_counter (in
     if (self.statistics_socket >= 0) {
         /*  TODO(tailhook) add HAVE_GMTIME_R ifdef  */
         time(&numtime);
-#ifdef NN_HAVE_WINDOWS
-        gmtime_s (&strtime, &numtime);
-#else
         gmtime_r (&numtime, &strtime);
-#endif
         strftime (timebuf, 20, "%Y-%m-%dT%H:%M:%S", &strtime);
         if(*s->socket_name) {
             len = sprintf (buf, "ESTP:%s:%s:socket.%s:%s: %sZ 10 %llu:c",
@@ -905,11 +902,7 @@ static void nn_global_submit_level (int
     if (self.statistics_socket >= 0) {
         /*  TODO(tailhook) add HAVE_GMTIME_R ifdef  */
         time(&numtime);
-#ifdef NN_HAVE_WINDOWS
-        gmtime_s (&strtime, &numtime);
-#else
         gmtime_r (&numtime, &strtime);
-#endif
         strftime (timebuf, 20, "%Y-%m-%dT%H:%M:%S", &strtime);
         if(*s->socket_name) {
             len = sprintf (buf, "ESTP:%s:%s:socket.%s:%s: %sZ 10 %d",
@@ -942,11 +935,7 @@ static void nn_global_submit_errors (int
     if (self.statistics_socket >= 0) {
         /*  TODO(tailhook) add HAVE_GMTIME_R ifdef  */
         time(&numtime);
-#ifdef NN_HAVE_WINDOWS
-        gmtime_s (&strtime, &numtime);
-#else
         gmtime_r (&numtime, &strtime);
-#endif
         strftime (timebuf, 20, "%Y-%m-%dT%H:%M:%S", &strtime);
         if(*s->socket_name) {
             len = sprintf (buf, "ESTP:%s:%s:socket.%s:%s: %sZ 10 %d\n",
