diff -ur xscrabble-2.12.orig/build xscrabble-2.12/build
--- xscrabble-2.12.orig/build	2012-08-24 15:01:41.729918950 +0100
+++ xscrabble-2.12/build	2012-08-24 15:02:48.387616457 +0100
@@ -1,9 +1,9 @@
 #!/bin/sh
 
 # Install defaults
-APPDEFAULTS=/etc/X11/app-defaults
-BINDIR=/usr/games
-LIBDIR=/usr/share/games/scrabble
+APPDEFAULTS=/usr/share/X11/app-defaults
+BINDIR=/usr/bin
+LIBDIR=/usr/share/xscrabble
 
 # Version (don't modify without knowing!!)
 MAJOR=2
diff -ur xscrabble-2.12.orig/XScrabble xscrabble-2.12/XScrabble
--- xscrabble-2.12.orig/XScrabble	2012-08-24 15:01:41.729918950 +0100
+++ xscrabble-2.12/XScrabble	2012-08-24 15:02:48.394283240 +0100
@@ -1,9 +1,9 @@
 ! *** XScrabble_en (English) resource file ***
 
 ! edit the following four pathnames as required:
-*dictfile:			/usr/share/games/scrabble/en/OSPD3
-*scorefile:			/usr/share/games/scrabble/en/scrabble_scores
-*rulesfile:			/usr/share/games/scrabble/en/scrabble_rules
+*dictfile:			/usr/share/xscrabble/en/OSPD3
+*scorefile:			/usr/share/xscrabble/en/scrabble_scores
+*rulesfile:			/usr/share/xscrabble/en/scrabble_rules
 
 ! version string, to make sure correct resource file is loaded
 *version:			212
