diff -crB ../cgames-2.2/cblocks/answers.c .//cblocks/answers.c
*** ../cgames-2.2/cblocks/answers.c	2000-09-17 16:12:51.000000000 +0000
--- .//cblocks/answers.c	2010-03-29 14:04:10.775154616 +0000
***************
*** 14,19 ****
--- 14,20 ----
  #include	"fileread.h"
  #include	"play.h"
  #include	"answers.h"
+ #include    <string.h>
  
  /* The directory containing the user's solution files.
   */
***************
*** 155,161 ****
  	return TRUE;
  
      for (;;) {
! 	n = getline(fp, buf, sizeof buf);
  	if (n < 0)
  	    return FALSE;
  	if (*buf != '\n' && *buf != ';')
--- 156,162 ----
  	return TRUE;
  
      for (;;) {
! 	n = ggetline(fp, buf, sizeof buf);
  	if (n < 0)
  	    return FALSE;
  	if (*buf != '\n' && *buf != ';')
diff -crB ../cgames-2.2/cblocks/fileread.c .//cblocks/fileread.c
*** ../cgames-2.2/cblocks/fileread.c	2000-09-17 16:11:45.000000000 +0000
--- .//cblocks/fileread.c	2010-03-29 14:04:00.139318389 +0000
***************
*** 30,36 ****
  /* Read one full line from fp and store the first len characters in
   * buf.
   */
! int getline(FILE *fp, char *buf, int len)
  {
      int	ch, n;
  
--- 30,36 ----
  /* Read one full line from fp and store the first len characters in
   * buf.
   */
! int ggetline(FILE *fp, char *buf, int len)
  {
      int	ch, n;
  
***************
*** 59,65 ****
      for (;;) {
  	ch = fgetc(series->mapfp);
  	if (ch == ';') {
! 	    getline(series->mapfp, buf, sizeof buf);
  	    if (*buf == ';') {
  		for (n = 1 ; isspace(buf[n]) ; ++n) ;
  		strncpy(series->name, buf + n, sizeof series->name - 1);
--- 59,65 ----
      for (;;) {
  	ch = fgetc(series->mapfp);
  	if (ch == ';') {
! 	    ggetline(series->mapfp, buf, sizeof buf);
  	    if (*buf == ';') {
  		for (n = 1 ; isspace(buf[n]) ; ++n) ;
  		strncpy(series->name, buf + n, sizeof series->name - 1);
diff -crB ../cgames-2.2/cblocks/fileread.h .//cblocks/fileread.h
*** ../cgames-2.2/cblocks/fileread.h	2000-09-17 16:12:37.000000000 +0000
--- .//cblocks/fileread.h	2010-03-29 14:04:00.145152009 +0000
***************
*** 53,59 ****
   * buf are lost. The return value is the length of the string stored
   * in buf, or -1 if fgets() returned NULL.
   */
! extern int getline(FILE *fp, char *buf, int len);
  
  /* Find the puzzle files and allocate an array of gameseries
   * structures for them.
--- 53,59 ----
   * buf are lost. The return value is the length of the string stored
   * in buf, or -1 if fgets() returned NULL.
   */
! extern int ggetline(FILE *fp, char *buf, int len);
  
  /* Find the puzzle files and allocate an array of gameseries
   * structures for them.
diff -crB ../cgames-2.2/cblocks/movelist.c .//cblocks/movelist.c
*** ../cgames-2.2/cblocks/movelist.c	2000-09-07 21:13:01.000000000 +0000
--- .//cblocks/movelist.c	2010-03-29 14:04:10.775154616 +0000
***************
*** 7,12 ****
--- 7,13 ----
  #include	<stdlib.h>
  #include	"gen.h"
  #include	"movelist.h"
+ #include    <string.h>
  
  /* Initialize or reinitialize list as empty.
   */
diff -crB ../cgames-2.2/cblocks/parse.c .//cblocks/parse.c
*** ../cgames-2.2/cblocks/parse.c	2000-09-10 03:47:09.000000000 +0000
--- .//cblocks/parse.c	2010-03-29 14:04:00.145152009 +0000
***************
*** 50,56 ****
      memset(map, ' ', fmi->ysize * fmi->xsize);
      p = map;
      for (y = 0 ; y < fmi->ysize ; ++y) {
! 	n = getline(fp, buf, sizeof buf);
  	if (n < 0 || *buf == ';')
  	    break;
  	if (!ok)
--- 50,56 ----
      memset(map, ' ', fmi->ysize * fmi->xsize);
      p = map;
      for (y = 0 ; y < fmi->ysize ; ++y) {
! 	n = ggetline(fp, buf, sizeof buf);
  	if (n < 0 || *buf == ';')
  	    break;
  	if (!ok)
***************
*** 96,102 ****
      int		n;
  
      for (;;) {
! 	n = getline(fp, buf, sizeof buf);
  	if (n < 0)
  	    return NULL;
  	if (!n || *buf == ';')
--- 96,102 ----
      int		n;
  
      for (;;) {
! 	n = ggetline(fp, buf, sizeof buf);
  	if (n < 0)
  	    return NULL;
  	if (!n || *buf == ';')
diff -crB ../cgames-2.2/cblocks/play.c .//cblocks/play.c
*** ../cgames-2.2/cblocks/play.c	2000-11-11 13:13:17.000000000 +0000
--- .//cblocks/play.c	2010-03-29 14:04:10.775154616 +0000
***************
*** 9,14 ****
--- 9,15 ----
  #include	"cblocks.h"
  #include	"userio.h"
  #include	"play.h"
+ #include    <string.h>
  
  /* The bits of a cell that move with a block.
   */
diff -crB ../cgames-2.2/csokoban/answers.c .//csokoban/answers.c
*** ../cgames-2.2/csokoban/answers.c	2000-11-05 00:30:18.000000000 +0000
--- .//csokoban/answers.c	2010-03-29 14:04:00.145152009 +0000
***************
*** 115,121 ****
  	return TRUE;
  
      for (;;) {
! 	n = getline(fp, buf, sizeof buf);
  	if (n < 0)
  	    return FALSE;
  	if (*buf != '\n' && *buf != ';')
--- 115,121 ----
  	return TRUE;
  
      for (;;) {
! 	n = ggetline(fp, buf, sizeof buf);
  	if (n < 0)
  	    return FALSE;
  	if (*buf != '\n' && *buf != ';')
***************
*** 133,139 ****
      game->movebestcount = n;
      game->movebestpushcount = m;
  
!     if (getline(fp, buf, sizeof buf) <= 0
  			|| sscanf(buf, "%d moves, %d pushes", &n, &m) < 2
  			|| !readanswer(fp, &game->pushanswer, n)) {
  	copymovelist(&game->pushanswer, &game->moveanswer);
--- 133,139 ----
      game->movebestcount = n;
      game->movebestpushcount = m;
  
!     if (ggetline(fp, buf, sizeof buf) <= 0
  			|| sscanf(buf, "%d moves, %d pushes", &n, &m) < 2
  			|| !readanswer(fp, &game->pushanswer, n)) {
  	copymovelist(&game->pushanswer, &game->moveanswer);
diff -crB ../cgames-2.2/csokoban/fileread.c .//csokoban/fileread.c
*** ../cgames-2.2/csokoban/fileread.c	2000-09-17 16:19:59.000000000 +0000
--- .//csokoban/fileread.c	2010-03-29 14:04:00.145152009 +0000
***************
*** 38,44 ****
  /* Read one full line from fp and store the first len characters in
   * buf.
   */
! int getline(FILE *fp, char *buf, int len)
  {
      int	ch, n;
  
--- 38,44 ----
  /* Read one full line from fp and store the first len characters in
   * buf.
   */
! int ggetline(FILE *fp, char *buf, int len)
  {
      int	ch, n;
  
***************
*** 162,168 ****
  	    --y;
  	    if (ch == '\n')
  		continue;
! 	    n = getline(fp, buf, sizeof buf);
  	    if (n < 0)
  		return FALSE;
  	    if (ch == ';' && *buf == ';') {
--- 162,168 ----
  	    --y;
  	    if (ch == '\n')
  		continue;
! 	    n = ggetline(fp, buf, sizeof buf);
  	    if (n < 0)
  		return FALSE;
  	    if (ch == ';' && *buf == ';') {
***************
*** 177,183 ****
  	    continue;
  	}
  	buf[0] = ch;
! 	getline(fp, buf + 1, sizeof buf - 1);
  	if (badmap)
  	    continue;
  	x = 1;
--- 177,183 ----
  	    continue;
  	}
  	buf[0] = ch;
! 	ggetline(fp, buf + 1, sizeof buf - 1);
  	if (badmap)
  	    continue;
  	x = 1;
***************
*** 239,245 ****
      for (;;) {
  	ch = fgetc(series->mapfp);
  	if (ch == ';') {
! 	    getline(series->mapfp, buf, sizeof buf);
  	    if (*buf == ';') {
  		for (n = 1 ; isspace(buf[n]) ; ++n) ;
  		strncpy(series->name, buf + n, sizeof series->name - 1);
--- 239,245 ----
      for (;;) {
  	ch = fgetc(series->mapfp);
  	if (ch == ';') {
! 	    ggetline(series->mapfp, buf, sizeof buf);
  	    if (*buf == ';') {
  		for (n = 1 ; isspace(buf[n]) ; ++n) ;
  		strncpy(series->name, buf + n, sizeof series->name - 1);
diff -crB ../cgames-2.2/csokoban/fileread.h .//csokoban/fileread.h
*** ../cgames-2.2/csokoban/fileread.h	2000-09-17 16:19:32.000000000 +0000
--- .//csokoban/fileread.h	2010-03-29 14:04:00.145152009 +0000
***************
*** 56,62 ****
   * buf are lost. The return value is the length of the string stored
   * in buf, or -1 if fgets() returned NULL.
   */
! extern int getline(FILE *fp, char *buf, int len);
  
  /* Find the puzzle files and allocate an array of gameseries
   * structures for them.
--- 56,62 ----
   * buf are lost. The return value is the length of the string stored
   * in buf, or -1 if fgets() returned NULL.
   */
! extern int ggetline(FILE *fp, char *buf, int len);
  
  /* Find the puzzle files and allocate an array of gameseries
   * structures for them.
diff -crB ../cgames-2.2/csokoban/movelist.c .//csokoban/movelist.c
*** ../cgames-2.2/csokoban/movelist.c	2000-09-07 21:13:16.000000000 +0000
--- .//csokoban/movelist.c	2010-03-29 14:04:10.775154616 +0000
***************
*** 7,12 ****
--- 7,13 ----
  #include	<stdlib.h>
  #include	"gen.h"
  #include	"movelist.h"
+ #include    <string.h>
  
  /* Initialize or reinitialize list as empty.
   */
diff -crB ../cgames-2.2/csokoban/play.c .//csokoban/play.c
*** ../cgames-2.2/csokoban/play.c	2000-11-05 00:40:13.000000000 +0000
--- .//csokoban/play.c	2010-03-29 14:04:10.775154616 +0000
***************
*** 9,14 ****
--- 9,15 ----
  #include	"csokoban.h"
  #include	"userio.h"
  #include	"play.h"
+ #include    <string.h>
  
  /* One entry on the saved-state stack.
   */
