diff -Naur 3dfsb-1.0-rc1-orig/compile.sh 3dfsb-1.0-rc1/compile.sh
--- 3dfsb-1.0-rc1-orig/compile.sh	2015-01-07 12:38:24.438918027 -0500
+++ 3dfsb-1.0-rc1/compile.sh	2015-01-07 12:38:39.269035515 -0500
@@ -50,7 +50,7 @@
 	#link=$(pkg-config --cflags gstreamer-1.0)" "$(pkg-config --cflags gstreamer-gl-1.0)" "$(pkg-config --cflags gstreamer-video-1.0)" "$(pkg-config --cflags gdk-pixbuf-2.0)
 	#gccopt="-static -static-libgcc"
 
-    gcc $warnings $gccopt -v $ARGS1 $ARGS2 -I/usr/local/include -I/usr/include/ -I/usr/X11R6/include $includes -L/usr/lib/ -L/usr/local/lib/ -L/usr/X11R6/lib -D_THREAD_SAFE -g -x c $link -lSDL_image -lglut -lXmu -lXi -lXext -lX11 -lm -lsmpeg -o 3dfsb 3dfsb.c
+    gcc $warnings $gccopt -v $ARGS1 $ARGS2 -I/usr/local/include -I/usr/include/ -I/usr/X11R6/include $includes -L/usr/lib/ -L/usr/local/lib/ -L/usr/X11R6/lib -D_THREAD_SAFE -g -x c $link -lSDL_image -lglut -lXmu -lXi -lXext -lX11 -lm -lsmpeg -lmagic -o 3dfsb 3dfsb.c
 
 elif uname -s | grep -i -c "BEOS" > /dev/null; then 
     echo "BeOS detected.";
