diff -rup xvba-video.orig/src/Makefile.am xvba-video/src/Makefile.am
--- xvba-video.orig/src/Makefile.am	2011-09-08 03:17:28.000000000 +0200
+++ xvba-video/src/Makefile.am	2011-09-09 17:47:41.283017833 +0200
@@ -72,11 +72,11 @@ xvba_drv_video_la_LDFLAGS	= $(LDADD)
 
 noinst_HEADERS = $(source_h)
 
-install-data-hook:
-	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
-	for drv in $(DRIVERS); do				\
-	    ln -s xvba_drv_video.so $${drv}_drv_video.so;	\
-	done
+#install-data-hook:
+#	cd $(DESTDIR)$(LIBVA_DRIVERS_PATH) ;			\
+#	for drv in $(DRIVERS); do				\
+#	    ln -s xvba_drv_video.so $${drv}_drv_video.so;	\
+#	done
 
 # Extra clean files so that maintainer-clean removes *everything*
 MAINTAINERCLEANFILES = Makefile.in config.h.in
