diff -Naur a/Makefile b/Makefile
--- a/Makefile	2013-03-23 07:41:13.000000000 +0000
+++ b/Makefile	2013-03-26 16:33:28.000000000 +0000
@@ -1,7 +1,6 @@
-
-export prefix=/usr/local
+export prefix ?= /usr/local
 export milena_dir=$(prefix)/share/milena
-export doc_dir=$(prefix)/share/doc/milena
+export doc_dir=$(DESTDIR)$(prefix)/share/doc/milena
 export speechd_dir=$(shell ./find_speechd)
 export distro=$(shell uname -o)
 export mbrola=$(shell if [ -f ./config.dat ] ; then cat ./config.dat | awk -F= '/mbrola=/ {print $$2}'; fi)
