diff -Naur a/Makefile b/Makefile
--- a/Makefile	2012-07-04 10:49:35.000000000 +0000
+++ b/Makefile	2012-07-04 10:56:49.000000000 +0000
@@ -29,7 +29,6 @@
 		
 install:
 	make -C src install
-	ldconfig
 	mkdir -p $(doc_dir)
 	install -m 644 README* $(doc_dir)
 	make -C utils install
@@ -40,7 +39,6 @@
 	for i in README*;do rm -f $(doc_dir)/"$$i";done
 	if [ -d $(doc_dir) ] ; then rmdir --ignore-fail-on-non-empty $(doc_dir);fi
 	make -C src uninstall
-	ldconfig
 	make -C utils uninstall
 	make -C milena_ws uninstall
 	make -C subplayer uninstall
diff -Naur a/src/Makefile b/src/Makefile
--- a/src/Makefile	2012-03-18 10:43:41.000000000 +0000
+++ b/src/Makefile	2012-07-04 10:58:12.000000000 +0000
@@ -59,10 +59,13 @@
 	rm -f *~ *.o *.so milena milena.h milena.pc milena_cyrillic.h
 
 install:	milena
+	install -dm 0755 $(DESTDIR)$(prefix)/bin
+	install -dm 0755 $(DESTDIR)$(LIBDIR)/pkgconfig
+	install -dm 0755 $(DESTDIR)$(prefix)/include
 	install -m 0755 milena $(DESTDIR)$(prefix)/bin
 	install -m 0755 libmilena.so libmilena_mbrola.so $(DESTDIR)$(LIBDIR)
 	install -m 0644 milena.h milena_mbrola.h $(DESTDIR)$(prefix)/include/
-	if [ -d $(LIBDIR)/pkgconfig ] ; then install -m 0644 milena.pc $(DESTDIR)$(LIBDIR)/pkgconfig; fi
+	if [ -d $(LIBDIR)/pkgconfig ] ; then install -m 0644 milena.pc $(DESTDIR)$(LIBDIR)/pkgconfig/milena.pc; fi
 
 uninstall:
 	rm -f $(DESTDIR)$(prefix)/bin/milena
