--- Makefile.orig	2011-11-17 13:15:45.000000000 +0100
+++ Makefile	2011-11-17 13:18:58.000000000 +0100
@@ -1,7 +1,6 @@
-
-export prefix=/usr/local
+export prefix ?= /usr/local
 export milena_dir=$(prefix)/share/milena
-export doc_dir=$(prefix)/share/doc/milena
+export doc_dir=$(DESTDIR)$(prefix)/share/doc/milena
 export speechd_dir=$(shell ./find_speechd)
 export distro=$(shell uname -o)
 export mbrola=$(shell if [ -f ./config.dat ] ; then cat ./config.dat | awk -F= '/mbrola=/ {print $$2}'; fi)
@@ -11,7 +10,6 @@
 
 all: config.dat
 	make -C src all
-	make -C data all
 	make -C utils all
 	make -C speechd all
 	make -C milena_ws all
@@ -24,7 +22,6 @@
 clean:
 	rm -f config.dat
 	make -C src clean
-	make -C data clean
 	make -C utils clean
 	make -C speechd clean
 	make -C milena_ws clean
@@ -32,10 +29,8 @@
 		
 install:
 	make -C src install
-	ldconfig
 	mkdir -p $(doc_dir)
 	install -m 644 README* $(doc_dir)
-	make -C data install
 	make -C utils install
 	make -C milena_ws install
 	make -C subplayer install
@@ -44,8 +39,6 @@
 	for i in README*;do rm -f $(doc_dir)/"$$i";done
 	if [ -d $(doc_dir) ] ; then rmdir --ignore-fail-on-non-empty $(doc_dir);fi
 	make -C src uninstall
-	ldconfig
-	make -C data uninstall
 	make -C utils uninstall
 	make -C milena_ws uninstall
 	make -C subplayer uninstall
--- src/Makefile.orig	2011-11-17 13:29:29.000000000 +0100
+++ src/Makefile	2011-11-17 13:32:29.000000000 +0100
@@ -57,10 +57,13 @@
 	rm -f *~ *.o *.so milena milena.h milena.pc milena_cyrillic.h
 
 install:	milena
+	install -dm 0755 $(DESTDIR)$(prefix)/bin
+	install -dm 0755 $(DESTDIR)$(LIBDIR)/pkgconfig
+	install -dm 0755 $(DESTDIR)$(prefix)/include
 	install -m 0755 milena $(DESTDIR)$(prefix)/bin
 	install -m 0755 libmilena.so libmilena_mbrola.so $(DESTDIR)$(LIBDIR)
 	install -m 0644 milena.h milena_mbrola.h $(DESTDIR)$(prefix)/include/
-	if [ -d $(LIBDIR)/pkgconfig ] ; then install -m 0644 milena.pc $(DESTDIR)$(LIBDIR)/pkgconfig; fi
+	if [ -d $(LIBDIR)/pkgconfig ] ; then install -m 0644 milena.pc $(DESTDIR)$(LIBDIR)/pkgconfig/milena.pc; fi
 
 uninstall:
 	rm -f $(DESTDIR)$(prefix)/bin/milena
