diff -ru unico-old//configure.ac unico/configure.ac
--- unico-old//configure.ac	2011-06-23 18:56:22.416622139 +0200
+++ unico/configure.ac	2011-06-23 18:56:36.659955118 +0200
@@ -35,7 +35,7 @@
 PKG_PROG_PKG_CONFIG
 
 AC_SUBST(glib_req, 2.26.0)
-AC_SUBST(gtk_req, 3.1.6)
+AC_SUBST(gtk_req, 3.0.11)
 AC_SUBST(cairo_req, 1.10)
 
 PKG_CHECK_MODULES(DEPS, glib-2.0 >= $glib_req gtk+-3.0 >= $gtk_req cairo >= $cairo_req)
