--- XGngeo/setup.py	2012-01-09 16:49:55.712992948 -0200
+++ XGngeo-new/setup.py	2012-01-09 17:02:16.018982367 -0200
@@ -48,26 +48,28 @@
 		#ROM info.
 		(os.path.join("share", "xgngeo"),glob.glob(os.path.join("data", "rominfos.*"))),
 		#License text.
-		(os.path.join("share", "xgngeo"),['LICENSE.txt']),
+		#    This license, GPL, is one of common licenses and therefore is part of
+		#    'licenses' package - No need to install this file again.
+		#(os.path.join("share", "xgngeo"),['LICENSE.txt']),
 		#Plain text documentation.
-		(os.path.join("share", "xgngeo","doc"),[os.path.join("doc", "xgngeo-doc.txt")]),
+		(os.path.join("share", "doc", "xgngeo"),[os.path.join("doc", "xgngeo-doc.txt")]),
 		# *.desktop (menu entry)
 		(os.path.join("share", "applications"), [os.path.join("data", "misc", "xgngeo.desktop")]),
 		#Localization files.
 		#Spanish
-		(os.path.join("share", "xgngeo", "locale", "es", "LC_MESSAGES"),
+		(os.path.join("share", "locale", "es", "LC_MESSAGES"),
 			[os.path.join("data", "locale", "es", "LC_MESSAGES", "xgngeo.mo")]),
 		#German
-		(os.path.join("share", "xgngeo", "locale", "de", "LC_MESSAGES"),
+		(os.path.join("share", "locale", "de", "LC_MESSAGES"),
 			[os.path.join("data", "locale", "de", "LC_MESSAGES", "xgngeo.mo")]),
 		#French
-		(os.path.join("share", "xgngeo", "locale", "fr", "LC_MESSAGES"),
+		(os.path.join("share", "locale", "fr", "LC_MESSAGES"),
 			[os.path.join("data", "locale", "fr", "LC_MESSAGES", "xgngeo.mo")]),
 		#Polish
-		(os.path.join("share", "xgngeo", "locale", "pl", "LC_MESSAGES"),
+		(os.path.join("share", "locale", "pl", "LC_MESSAGES"),
 			[os.path.join("data", "locale", "pl", "LC_MESSAGES", "xgngeo.mo")]),
 		#Portuguese of Brazil
-		(os.path.join("share", "xgngeo", "locale", "pt_BR", "LC_MESSAGES"),
+		(os.path.join("share", "locale", "pt_BR", "LC_MESSAGES"),
 			[os.path.join("data", "locale", "pt_BR", "LC_MESSAGES", "xgngeo.mo")])
 		]
 	)
