From 01b266eb1067c47dff97e7ae08fa7b9d36ee7389 Mon Sep 17 00:00:00 2001
From: Simon Josefsson <simon@josefsson.org>
Date: Tue, 26 Apr 2011 14:30:41 +0000
Subject: Also attempt to replay a TOTP.

---
diff --git a/liboath/tests/tst_usersfile.c b/liboath/tests/tst_usersfile.c
index 13993d2..c195274 100644
--- a/liboath/tests/tst_usersfile.c
+++ b/liboath/tests/tst_usersfile.c
@@ -125,6 +125,14 @@ main (void)
       return 1;
     }
 
+  rc = oath_authenticate_usersfile (CREDS,
+				    "eve", "299833", 1, NULL, &last_otp);
+  if (rc != OATH_REPLAYED_OTP)
+    {
+      printf ("oath_authenticate_usersfile: %d\n", rc);
+      return 1;
+    }
+
   rc = oath_done ();
   if (rc != OATH_OK)
     {
--
cgit v0.8.3.4
