diff -ru orig/ide/preferences.ml coq-8.4/ide/preferences.ml
--- orig/ide/preferences.ml	2012-08-08 20:54:37.000000000 +0200
+++ coq-8.4/ide/preferences.ml	2012-09-16 20:47:50.228590841 +0200
@@ -36,6 +36,7 @@
     | `CONTROL -> "<Control>"
     | `SHIFT -> "<Shift>"
     |  `BUTTON1| `BUTTON2| `BUTTON3| `BUTTON4| `BUTTON5| `LOCK -> ""
+    | _ -> ""
 
 let mod_list_to_str l = List.fold_left (fun s m -> (mod_to_str m)^s) "" l
 
diff -ru orig/ide/utils/okey.ml coq-8.4/ide/utils/okey.ml
--- orig/ide/utils/okey.ml	2006-03-08 12:44:47.000000000 +0100
+++ coq-8.4/ide/utils/okey.ml	2012-09-16 20:46:35.627568953 +0200
@@ -47,6 +47,7 @@
   | `BUTTON3 -> 1024
   | `BUTTON4 -> 2048
   | `BUTTON5 -> 4096
+  | _ -> 0
 
 let print_modifier l =
   List.iter
