diff -r 9a463162f3e0 -r e6ce24fa06bd imms/clients/audacious/audplugin.cc
--- a/imms/clients/audacious/audplugin.cc	Sun May 08 13:31:00 2011 +0000
+++ b/imms/clients/audacious/audplugin.cc	Sun May 08 19:02:56 2011 +0200
@@ -49,28 +49,6 @@
 
 string cur_path = "", last_path = "";
 
-extern "C" {
-void init(void);
-void about(void);
-void configure(void);
-void cleanup(void);
-}
-
-static GeneralPlugin imms_gp =
-{
-    0,
-    0,
-    PACKAGE_STRING,
-    init,
-    cleanup,
-    about,
-    configure
-};
-
-GeneralPlugin *gp_plugin_list[] = { &imms_gp, NULL };
-
-SIMPLE_GENERAL_PLUGIN(imms, gp_plugin_list);
-
 // Wrapper that frees memory
 string imms_get_playlist_item(int at)
 {
@@ -82,7 +60,7 @@
     string result = tmp;
     free(tmp);
     gchar *realfn = g_filename_from_uri(result.c_str(), NULL, NULL);
-    tmp = filename_to_utf8(realfn ? realfn : result.c_str());
+    tmp = g_filename_to_utf8(realfn ? realfn : result.c_str(), -1, NULL, NULL, NULL);
     result = tmp;
     free(tmp);
     free(realfn);
diff -r 9a463162f3e0 -r e6ce24fa06bd imms/clients/audacious/audplugindef.c
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/imms/clients/audacious/audplugindef.c	Sun May 08 19:02:56 2011 +0200
@@ -0,0 +1,25 @@
+#include <audacious/plugin.h>
+#include <audacious/audctrl.h>
+#include <audacious/drct.h>
+#include <libaudcore/audstrings.h>
+
+#include "immsconf.h"
+
+void init(void);
+void about(void);
+void configure(void);
+void cleanup(void);
+
+static GeneralPlugin imms_gp =
+{
+    PACKAGE_STRING,
+    init,
+    cleanup,
+    about,
+    configure,
+    NULL
+};
+
+GeneralPlugin *gp_plugin_list[] = { &imms_gp, NULL };
+
+SIMPLE_GENERAL_PLUGIN(imms, gp_plugin_list);
diff -r 9a463162f3e0 -r e6ce24fa06bd imms/clients/audacious/rules.mk
--- a/imms/clients/audacious/rules.mk	Sun May 08 13:31:00 2011 +0000
+++ b/imms/clients/audacious/rules.mk	Sun May 08 19:02:56 2011 +0200
@@ -16,10 +16,11 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 #
-libaudaciousimms.so: audplugin.o audaciousinterface.o clientstubbase.o libimmscore.a 
+libaudaciousimms.so: audplugindef.o audplugin.o audaciousinterface.o clientstubbase.o libimmscore.a 
 libaudaciousimms-LIBS = $(AUDACIOUSLDFLAGS)
 
 audaciousinterface-CPPFLAGS=$(AUDACIOUSCPPFLAGS)
+audplugindef-CPPFLAGS=$(AUDACIOUSCPPFLAGS)
 audplugin-CPPFLAGS=$(AUDACIOUSCPPFLAGS)
 
 audaciousinterface.o: bmpinterface.c
diff -r 9a463162f3e0 -r e6ce24fa06bd imms/clients/bmp/bmpinterface.c
--- a/imms/clients/bmp/bmpinterface.c	Sun May 08 13:31:00 2011 +0000
+++ b/imms/clients/bmp/bmpinterface.c	Sun May 08 19:02:56 2011 +0200
@@ -62,12 +62,13 @@
     }
 }
 
-void init(void)
+gboolean init(void)
 {
     imms_init();
     read_config();
     imms_setup(use_xidle);
     poll_tag = gtk_timeout_add(200, poll_func, NULL);
+    return TRUE;
 }
 
 void cleanup(void)
diff -r 9a463162f3e0 -r e6ce24fa06bd imms/configure.ac
--- a/imms/configure.ac	Sun May 08 13:31:00 2011 +0000
+++ b/imms/configure.ac	Sun May 08 19:02:56 2011 +0200
@@ -364,20 +364,21 @@
 AC_CHECK_HEADERS(audacious/plugin.h,, [with_audacious=no])
 
 if test "$with_audacious" != "no"; then
+    AC_LANG_PUSH(C)
     AC_CHECK_HEADERS(audacious/drct.h,, [with_audacious=no])
     AC_MSG_CHECKING([for aud_drct_pq_add in -laudacious])
-    AC_TRY_LINK([extern "C" {
-                 #include <audacious/drct.h>
+    AC_TRY_LINK([#include <audacious/drct.h>
                  #include <audacious/plugin.h>
                  #include <audacious/dbus.h>
-                }
+
                   SIMPLE_GENERAL_PLUGIN(0, 0);],
                 [aud_drct_pq_add(0)],
                 [AC_MSG_RESULT(yes)],
                 [with_audacious="no"
                  AC_MSG_RESULT(no)]);
-    AC_DEFINE(AUDACIOUS, 1, [Define for Audacious 2.4 support])
+    AC_DEFINE(AUDACIOUS, 1, [Define for Audacious 2.5 support])
     audacious_CFLAGS="$audacious_CFLAGS -DAUDACIOUS"
+    AC_LANG_POP(C)
 fi
 
 if test "$with_audacious" = "no"; then
