Index: clients/audacious/audplugin.cc
===================================================================
--- clients/audacious/audplugin.cc	(revision 262)
+++ clients/audacious/audplugin.cc	(working copy)
@@ -82,7 +82,7 @@
     string result = tmp;
     free(tmp);
     gchar *realfn = g_filename_from_uri(result.c_str(), NULL, NULL);
-    tmp = filename_to_utf8(realfn ? realfn : result.c_str());
+    tmp = g_filename_to_utf8(realfn ? realfn : result.c_str(), -1, NULL, NULL, NULL);
     result = tmp;
     free(tmp);
     free(realfn);
