diff --git a/src/gpm-manager.c b/src/gpm-manager.c
index 5cff047..7561b22 100644
--- a/src/gpm-manager.c
+++ b/src/gpm-manager.c
@@ -1004,7 +1004,7 @@ struct GpmManagerPrivate
 	/* save in local cache */
 	manager->priv->on_battery = on_battery;
 
-	/* ConsoleKit says we are not on active console */
+	/* ConsoleKit/systemd say we are not on active console */
 	if (!LOGIND_RUNNING() && !egg_console_kit_is_active (manager->priv->console)) {
 		egg_debug ("ignoring as not on active console");
 		return;
