--- drivers/gpu/drm/i915/intel_opregion.c.orig	2013-07-01 02:13:29.000000000 +0400
+++ drivers/gpu/drm/i915/intel_opregion.c	2013-09-03 03:24:26.556353050 +0400
@@ -150,6 +150,7 @@
 #ifdef CONFIG_ACPI
 static u32 asle_set_backlight(struct drm_device *dev, u32 bclp)
 {
+	/*
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	struct opregion_asle __iomem *asle = dev_priv->opregion.asle;
 	u32 max;
@@ -166,6 +167,7 @@
 	max = intel_panel_get_max_backlight(dev);
 	intel_panel_set_backlight(dev, bclp * max / 255);
 	iowrite32((bclp*0x64)/0xff | ASLE_CBLV_VALID, &asle->cblv);
+	*/
 
 	return 0;
 }
