--- ShotgunDebuggerSource/sdb.h	2010-02-04 20:29:34.000000000 +0100
+++ sdb.h	2010-06-01 13:21:19.000000000 +0200
@@ -87,18 +87,8 @@
 
 #define PACKAGE_RELEASE_DATE "June 15, 2005"
 
-#ifndef PKGDATADIR
-#ifdef _WIN32
-#define PKGDATADIR "./"
+#define PKGDATADIR "/usr/share/games/shotgun-debugger/Resources/"
 #define LOCALSTATEDIR "/.sdb/"
-#elif __MACOSX__
-#define PKGDATADIR "./Shotgun Debugger.app/Contents/Resources/"
-#define LOCALSTATEDIR "/Library/Application Support/Shotgun Debugger/" // relative to home dir
-#else
-#define PKGDATADIR "/usr/local/share/shotgun-debugger"
-#define LOCALSTATEDIR "/.sdb/"
-#endif
-#endif
 
 #define CONFIG_FILE "preferences.txt"
 #define PLAYER_DATA_FILE "player.dat"
