--- YACReader-7.1.0-src/YACReaderLibrary.desktop	2014-07-01 21:21:59.000000000 +0200
+++ YACReader-7.1.0-src/YACReaderLibrary.desktop	2014-07-06 16:37:25.893461695 +0200
@@ -8,5 +8,5 @@
 Type=Application
 StartupNotify=true
 Categories=Graphics;Viewer;
-MimeType=application/x-cbz;application/x-cbr;application/x-cbt;application/x-cb7;
+MimeType=
 X-Desktop-File-Install-Version=0.22
--- YACReader-7.1.0-src/YACReader.desktop	2014-07-01 21:21:59.000000000 +0200
+++ YACReader-7.1.0-src/YACReader.desktop	2014-07-06 16:37:25.893461695 +0200
@@ -8,5 +8,5 @@
 Type=Application
 StartupNotify=true
 Categories=Graphics;Viewer;
-MimeType=application/x-cbz;application/x-cbr;application/x-cbt;application/x-cb7;
+MimeType=application/x-cbz;application/x-cbr;application/x-cbt;application/x-cb7;application/x-pdf;application/x-zip;application/x-rar;application/x-7z;inode/directory;
 X-Desktop-File-Install-Version=0.22
