--- delay.c	2002-02-09 07:26:39.000000000 -0800
+++ delay.c	2008-03-04 03:07:07.835575808 -0800
@@ -17,6 +17,7 @@
 */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <time.h>
 #include <sys/time.h>
 #include <unistd.h>
@@ -341,7 +342,7 @@
 		exit(-1);
 	}
 
-	if (!strcmp(argv[optind], "until")) {
+	if (!strcmp(argv[optind], "until") || !strcmp(argv[optind], "til")) {
 		dtime = parsetime(argc-optind, &argv[optind+1]);
 		if (!dtime) exit(-1);
 		dtime -= time(NULL);
