--- cmake/BuildParameters.cmake	2011-03-13 03:54:46.366934378 -0300
+++ cmake.tmp/BuildParameters.cmake	2011-03-13 03:54:10.145081157 -0300
@@ -151,7 +151,7 @@
 #-------------------------------------------------------------------------------
 # Select library system vs 3rdparty
 #-------------------------------------------------------------------------------
-if(NOT DEFINED L10N_PORTABLE)
-    set(L10N_PORTABLE TRUE)
-    message(STATUS "Install localization file in bin/Langs by default")
-endif(NOT DEFINED L10N_PORTABLE)
+#if(NOT DEFINED L10N_PORTABLE)
+#    set(L10N_PORTABLE TRUE)
+#    message(STATUS "Install localization file in bin/Langs by default")
+#endif(NOT DEFINED L10N_PORTABLE)
