--- sflphone-0.9.12/astylerc.bak	2011-02-14 22:55:04.171946616 +0100
+++ sflphone-0.9.12/astylerc	2011-02-14 22:55:26.491441451 +0100
@@ -11,6 +11,7 @@
 indent-switches         # Indent 'switch' blocks so that the 'case X:' statements are indented in the switch block
 break-blocks            # Pad empty lines around header blocks (e.g. 'if', 'while'...).
 brackets=linux
-unpad=paren
+unpad-paren
 formatted
 -d
+-n
