diff -aur sflphone-1.1.0.orig/daemon/libs/iax2/iax-client.h sflphone-1.1.0/daemon/libs/iax2/iax-client.h
--- sflphone-1.1.0.orig/daemon/libs/iax2/iax-client.h	2012-05-17 23:16:25.000000000 +0200
+++ sflphone-1.1.0/daemon/libs/iax2/iax-client.h	2012-05-18 21:59:30.377474421 +0200
@@ -26,6 +26,7 @@
 #include "frame.h"
 #include "iax2.h"
 #include "iax2-parser.h"
+#include "sys/socket.h"
 
 #define MAXSTRLEN 80
 
diff -aur sflphone-1.1.0.orig/daemon/src/fileutils.cpp sflphone-1.1.0/daemon/src/fileutils.cpp
--- sflphone-1.1.0.orig/daemon/src/fileutils.cpp	2012-05-17 23:16:25.000000000 +0200
+++ sflphone-1.1.0/daemon/src/fileutils.cpp	2012-05-18 21:58:08.334137574 +0200
@@ -37,6 +37,7 @@
 #include <string>
 #include <sstream>
 #include <iostream>
+#include <unistd.h>
 #include "fileutils.h"
 
 namespace fileutils {
