diff --git a/sflphone-client-gnome/src/sflnotify.c b/sflphone-client-gnome/src/sflnotify.c
index a51b73d..d29cf0f 100644
--- a/sflphone-client-gnome/src/sflnotify.c
+++ b/sflphone-client-gnome/src/sflnotify.c
@@ -44,12 +44,9 @@ void create_new_gnome_notification (gchar *title, gchar *body, NotifyUrgency urg
         notify_init ("SFLphone");
 
         // Set struct fields
-        _notif->notification = notify_notification_new (title, body, NULL, NULL);
+        _notif->notification = notify_notification_new (title, body, NULL);
         //_notif->icon = gdk_pixbuf_new_from_file_at_size (LOGO, 120, 120, NULL);
         _notif->icon = gdk_pixbuf_new_from_file (LOGO_SMALL, NULL);
-#if GTK_CHECK_VERSION(2,10,0)
-        notify_notification_attach_to_status_icon (_notif->notification , get_status_icon());
-#endif
 
         notify_notification_set_urgency (_notif->notification, urgency);
