diff --git a/dlls/d3d9/device.c b/dlls/d3d9/device.c
index 80349cd..23f50f5 100644
--- a/dlls/d3d9/device.c
+++ b/dlls/d3d9/device.c
@@ -2815,7 +2815,7 @@ static HRESULT WINAPI d3d9_device_PresentEx(IDirect3DDevice9Ex *iface,
     FIXME("iface %p, src_rect %p, dst_rect %p, dst_window_override %p, dirty_region %p, flags %#x stub!\n",
             iface, src_rect, dst_rect, dst_window_override, dirty_region, flags);
 
-    return E_NOTIMPL;
+    return d3d9_device_Present(iface, src_rect, dst_rect, dst_window_override, dirty_region);
 }
 
 static HRESULT WINAPI d3d9_device_GetGPUThreadPriority(IDirect3DDevice9Ex *iface, INT *priority)
