diff -u flactag-1.1/CommandLine.cc flactag-1.1.new/CommandLine.cc
--- flactag-1.1/CommandLine.cc	2008-12-16 06:20:56.000000000 -0500
+++ flactag-1.1.new/CommandLine.cc	2011-05-01 13:29:17.541043280 -0400
@@ -29,6 +29,7 @@
 #include <sstream>
 
 #include <getopt.h>
+#include <cstdio>
 
 #include "ErrorLog.h"
 
diff -u flactag-1.1/ConfigFile.cc flactag-1.1.new/ConfigFile.cc
--- flactag-1.1/ConfigFile.cc	2008-12-16 06:20:56.000000000 -0500
+++ flactag-1.1.new/ConfigFile.cc	2011-05-01 13:29:17.540043292 -0400
@@ -25,6 +25,7 @@
 ----------------------------------------------------------------------------*/
 
 #include "ConfigFile.h"
+#include <cstdio>
 
 CConfigFile::CConfigFile()
 {
diff -u flactag-1.1/ErrorLog.cc flactag-1.1.new/ErrorLog.cc
--- flactag-1.1/ErrorLog.cc	2008-12-16 06:20:56.000000000 -0500
+++ flactag-1.1.new/ErrorLog.cc	2011-05-01 13:29:17.539043303 -0400
@@ -25,6 +25,7 @@
 ----------------------------------------------------------------------------*/
 
 #include "ErrorLog.h"
+#include <cstdio>
 
 const std::vector<std::string>::size_type CErrorLog::m_cMaxSize=100;
 	
diff -u flactag-1.1/FileNameBuilder.cc flactag-1.1.new/FileNameBuilder.cc
--- flactag-1.1/FileNameBuilder.cc	2008-12-16 06:20:56.000000000 -0500
+++ flactag-1.1.new/FileNameBuilder.cc	2011-05-01 13:29:17.540043291 -0400
@@ -27,6 +27,7 @@
 #include "FileNameBuilder.h"
 
 #include <stdlib.h>
+#include <stdio.h>
 
 #include <unac.h>
 
diff -u flactag-1.1/flactag.cc flactag-1.1.new/flactag.cc
--- flactag-1.1/flactag.cc	2008-12-16 06:20:56.000000000 -0500
+++ flactag-1.1.new/flactag.cc	2011-05-01 13:29:17.539043303 -0400
@@ -156,7 +156,7 @@
 								AlbumNum=count;
 								MatchFound=true;
 								
-								printf("%s: Album ID in album %d is a match\n",m_FlacFile.c_str(),count);
+								printf("%s: Album ID in album %d is a match\n",m_FlacFile.c_str(),(int)count);
 							}
 						}
 						
 flactag.1.xml: flactag.1.txt Makefile
 	asciidoc -d manpage -b docbook flactag.1.txt
diff -u flactag-1.1/UTF8Tag.cc flactag-1.1.new/UTF8Tag.cc
--- flactag-1.1/UTF8Tag.cc	2008-12-16 06:20:56.000000000 -0500
+++ flactag-1.1.new/UTF8Tag.cc	2011-05-01 13:29:17.540043292 -0400
@@ -32,6 +32,7 @@
 #include <errno.h>
 #include <string.h>
 #include <stdlib.h>
+#include <stdio.h>
 
 #include <iconv.h>
 
