diff -urN waf-1.6.3-old/waflib/Tools/fc.py waf-1.6.3/waflib/Tools/fc.py
--- waf-1.6.3-old/waflib/Tools/fc.py	2011-03-03 13:54:29.000000000 +0100
+++ waf-1.6.3/waflib/Tools/fc.py	2011-03-03 13:54:38.000000000 +0100
@@ -144,6 +144,7 @@
 	color = 'YELLOW'
 	run_str = '${FC} ${FCLNK_SRC_F}${SRC} ${FCLNK_TGT_F}${TGT} ${FCSTLIB_MARKER} ${FCSTLIBPATH_ST:STLIBPATH} ${FCSTLIB_ST:STLIB} ${FCSHLIB_MARKER} ${FCLIBPATH_ST:LIBPATH} ${FCLIB_ST:LIB} ${LINKFLAGS}'
 	inst_to = '${BINDIR}'
+	chmod   = Utils.O755
 
 class fcshlib(fcprogram):
 	"""Link fortran libraries"""
