diff -ur tigervnc-1.2.0.orig/CMakeLists.txt tigervnc-1.2.0/CMakeLists.txt
--- tigervnc-1.2.0.orig/CMakeLists.txt	2012-03-09 21:34:29.000000000 +0100
+++ tigervnc-1.2.0/CMakeLists.txt	2012-03-26 11:12:02.454978705 +0200
@@ -316,7 +316,7 @@
   if(APPLE)
     set(FLTK_LIBRARIES "-framework Carbon -framework Cocoa -framework ApplicationServices")
   elseif(NOT WIN32)
-    set(FLTK_LIBRARIES "-ldl")
+    set(FLTK_LIBRARIES "-ldl -lfontconfig")
   endif()
   message(STATUS "Using included FLTK library")
 endif()
@@ -411,8 +411,6 @@
 endif()
 add_subdirectory(vncviewer)
 
-include(cmake/BuildPackages.cmake)
-
 # uninstall
 configure_file("${CMAKE_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
   "cmake_uninstall.cmake" IMMEDIATE @ONLY)
diff -ur tigervnc-1.2.0.orig/unix/CMakeLists.txt tigervnc-1.2.0/unix/CMakeLists.txt
--- tigervnc-1.2.0.orig/unix/CMakeLists.txt	2011-11-03 20:01:18.000000000 +0100
+++ tigervnc-1.2.0/unix/CMakeLists.txt	2012-03-26 11:10:42.303980733 +0200
@@ -5,4 +5,4 @@
 add_subdirectory(x0vncserver)
 
 install(PROGRAMS vncserver DESTINATION bin)
-install(FILES vncserver.man DESTINATION man/man1 RENAME vncserver.1)
+install(FILES vncserver.man DESTINATION share/man/man1 RENAME vncserver.1)
diff -ur tigervnc-1.2.0.orig/unix/vncconfig/CMakeLists.txt tigervnc-1.2.0/unix/vncconfig/CMakeLists.txt
--- tigervnc-1.2.0.orig/unix/vncconfig/CMakeLists.txt	2011-06-25 09:32:34.000000000 +0200
+++ tigervnc-1.2.0/unix/vncconfig/CMakeLists.txt	2012-03-26 11:10:42.303980733 +0200
@@ -12,4 +12,4 @@
 target_link_libraries(vncconfig tx rfb network rdr ${X11_LIBRARIES})
 
 install(TARGETS vncconfig DESTINATION bin)
-install(FILES vncconfig.man DESTINATION man/man1 RENAME vncconfig.1)
+install(FILES vncconfig.man DESTINATION share/man/man1 RENAME vncconfig.1)
diff -ur tigervnc-1.2.0.orig/unix/vncpasswd/CMakeLists.txt tigervnc-1.2.0/unix/vncpasswd/CMakeLists.txt
--- tigervnc-1.2.0.orig/unix/vncpasswd/CMakeLists.txt	2011-06-25 09:32:34.000000000 +0200
+++ tigervnc-1.2.0/unix/vncpasswd/CMakeLists.txt	2012-03-26 11:10:42.303980733 +0200
@@ -6,4 +6,4 @@
 target_link_libraries(vncpasswd tx rfb rdr os)
 
 install(TARGETS vncpasswd DESTINATION bin)
-install(FILES vncpasswd.man DESTINATION man/man1 RENAME vncpasswd.1)
+install(FILES vncpasswd.man DESTINATION share/man/man1 RENAME vncpasswd.1)
diff -ur tigervnc-1.2.0.orig/unix/x0vncserver/CMakeLists.txt tigervnc-1.2.0/unix/x0vncserver/CMakeLists.txt
--- tigervnc-1.2.0.orig/unix/x0vncserver/CMakeLists.txt	2011-08-19 18:41:06.000000000 +0200
+++ tigervnc-1.2.0/unix/x0vncserver/CMakeLists.txt	2012-03-26 11:10:42.303980733 +0200
@@ -24,4 +24,4 @@
 endif()
 
 install(TARGETS x0vncserver DESTINATION bin)
-install(FILES x0vncserver.man DESTINATION man/man1 RENAME x0vncserver.1)
+install(FILES x0vncserver.man DESTINATION share/man/man1 RENAME x0vncserver.1)
diff -ur tigervnc-1.2.0.orig/vncviewer/CMakeLists.txt tigervnc-1.2.0/vncviewer/CMakeLists.txt
--- tigervnc-1.2.0.orig/vncviewer/CMakeLists.txt	2011-10-05 11:32:28.000000000 +0200
+++ tigervnc-1.2.0/vncviewer/CMakeLists.txt	2012-03-26 11:11:42.870223550 +0200
@@ -55,5 +55,5 @@
   install(TARGETS vncviewer DESTINATION bin)
 endif()
 if(UNIX)
-  install(FILES vncviewer.man DESTINATION man/man1 RENAME vncviewer.1)
+  install(FILES vncviewer.man DESTINATION share/man/man1 RENAME vncviewer.1)
 endif()
