diff -aur scgraph-0.13/server/control_loop.cc src/scgraph-0.13/server/control_loop.cc
--- scgraph-0.13/server/control_loop.cc	2008-07-05 23:52:16.000000000 +0200
+++ src/scgraph-0.13/server/control_loop.cc	2011-08-31 13:58:46.000000000 +0200
@@ -17,8 +17,7 @@
 	_quit (false),
 	_freq (Options::get_instance()->_control_rate)
 {
-	int ret;
-	ret = pthread_mutex_init (&_mutex, 0);
+	pthread_mutex_init (&_mutex, 0);
 
 }
 
@@ -33,8 +32,7 @@
 {
 	Options *options = Options::get_instance ();
 
-	int ret;
-	ret = pthread_create (&_thread, 0, c_control_thread_func, this);
+	pthread_create (&_thread, 0, c_control_thread_func, this);
 
 	if (options->_verbose >= 2)
 		std::cout << "[ControlLoop]: Running!" << std::endl;
@@ -42,21 +40,19 @@
 
 void ControlLoop::stop ()
 {
-	int ret;
-
-	ret = pthread_mutex_lock (&_mutex);
+	pthread_mutex_lock (&_mutex);
 	_quit = true;
-	ret = pthread_mutex_unlock (&_mutex);
+	pthread_mutex_unlock (&_mutex);
 
-	ret = pthread_join (_thread, 0);
+	pthread_join (_thread, 0);
 }
 
 
 void ControlLoop::set_rate (int rate)
 {
-	int ret = pthread_mutex_lock (&_mutex);
+	pthread_mutex_lock (&_mutex);
 	_freq = rate;
-	ret = pthread_mutex_unlock (&_mutex);
+	pthread_mutex_unlock (&_mutex);
 }
 
 void *c_control_thread_func (void *arg)
diff -aur scgraph-0.13/server/jack_client.cc src/scgraph-0.13/server/jack_client.cc
--- scgraph-0.13/server/jack_client.cc	2008-07-05 23:52:16.000000000 +0200
+++ src/scgraph-0.13/server/jack_client.cc	2011-08-31 13:58:46.000000000 +0200
@@ -27,13 +27,16 @@
 {
 	Options *options = Options::get_instance ();
 
+	jack_status_t jack_status;
+	jack_options_t  jack_options = JackNullOption;
+
 	if (options->_verbose >= 1)
 		std::cout << "[JackClient]: Initializing " << num_of_ports << " ports" << std::endl;
 
 	if (num_of_ports == 0)
 		return;
 
-	_client = jack_client_new (_jack_name.c_str ());
+	_client = jack_client_open (_jack_name.c_str (), jack_options, &jack_status);
 	if (!_client)
 		throw ("[JackClient]: Error: couldn't register as jack client");
 
diff -aur scgraph-0.13/server/Makefile src/scgraph-0.13/server/Makefile
--- scgraph-0.13/server/Makefile	2009-01-22 12:03:14.000000000 +0100
+++ src/scgraph-0.13/server/Makefile	2011-08-31 13:58:46.000000000 +0200
@@ -73,8 +73,8 @@
   COMPILE_FLAGS +=  -DSC_DARWIN
 endif
 
-#LINK_FLAGS = -lpthread $(ADDITIONAL_LINK_FLAGS) $(QT_LINK_FLAGS) -lm `pkg-config sndfile GraphicsMagick++ --cflags --libs` 
-LINK_FLAGS = -lpthread $(ADDITIONAL_LINK_FLAGS) $(QT_LINK_FLAGS) -lm `pkg-config sndfile --cflags --static` /Users/fschmidt/local/opt/GraphicsMagick-1.3.1/lib/libGraphicsMagick++.a  /Users/fschmidt/local/opt/GraphicsMagick-1.3.1/lib/libGraphicsMagick.a 
+LINK_FLAGS = -lpthread $(ADDITIONAL_LINK_FLAGS) $(QT_LINK_FLAGS) -lm `pkg-config sndfile GraphicsMagick++ --cflags --libs` 
+#LINK_FLAGS = -lpthread $(ADDITIONAL_LINK_FLAGS) $(QT_LINK_FLAGS) -lm `pkg-config sndfile --cflags --static` /usr/lib/libGraphicsMagick++.a  /usr/lib/libGraphicsMagick.a 
 
 ifeq ($(HAVE_JACK),1)
   LINK_FLAGS += `pkg-config jack --libs fftw3f` -DHAVE_JACK
diff -aur scgraph-0.13/server/plugin_pool.cc src/scgraph-0.13/server/plugin_pool.cc
--- scgraph-0.13/server/plugin_pool.cc	2009-01-22 14:54:41.000000000 +0100
+++ src/scgraph-0.13/server/plugin_pool.cc	2011-08-31 13:59:52.000000000 +0200
@@ -4,7 +4,9 @@
 
 #include "options.h"
 
+#ifndef  _XOPEN_SOURCE
 #define _XOPEN_SOURCE 600
+#endif
 #include <stdlib.h>
 
 #include <stdlib.h>
diff -aur scgraph-0.13/server/plugins/gl_renderer.cc src/scgraph-0.13/server/plugins/gl_renderer.cc
--- scgraph-0.13/server/plugins/gl_renderer.cc	2008-12-10 17:41:49.000000000 +0100
+++ src/scgraph-0.13/server/plugins/gl_renderer.cc	2011-08-31 13:58:46.000000000 +0200
@@ -5,7 +5,7 @@
 #include "../texture_pool.h"
 
 
-
+#include <stdio.h>
 #include <pthread.h>
 #include <unistd.h>
 
