--- vrep.sh	2014-02-03 18:38:42.969461494 +0100
+++ vrep.sh	2014-02-03 18:38:27.979279568 +0100
@@ -19,6 +19,28 @@
 		dirname=$PWD/$dirname
 	fi
 	LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$dirname
+
+	# Create temporary files
+	mkdir -p /tmp/v-rep/system
+	mkdir -p /tmp/v-rep/models
+	touch /tmp/v-rep/updateCheck.dat
+	touch /tmp/v-rep/debugLog.txt
+	cp $dirname/remoteApiConnections.txt.default /tmp/v-rep/remoteApiConnections.txt
+	touch /tmp/v-rep/system/settings.dat
+	cp $dirname/system/usrset.txt.default /tmp/v-rep/system/usrset.txt
+ 	chmod a+rw /tmp/v-rep/updateCheck.dat
+ 	chmod a+rw /tmp/v-rep/debugLog.txt
+ 	chmod a+rw /tmp/v-rep/remoteApiConnections.txt
+ 	chmod a+rw /tmp/v-rep/system/settings.dat
+ 	chmod a+rw /tmp/v-rep/system/usrset.txt
+ 	# Thumbnails
+ 	find $dirname -type f -iname "v_repModelThumbnails.ttmt.default" | while read file
+ 	do
+ 	 	mkdir -p "/tmp/v-rep/${file}"
+ 	 	cp "$file" "/tmp/v-rep/${file%.ttmt.default}.ttmt"
+ 	 	chmod a+rw "/tmp/v-rep/${file%.ttmt.default}.ttmt"
+ 	done
+
 	export LD_LIBRARY_PATH
 	$dirname/$appname "$@"
 fi
