--- config_default.sh	2014-10-31 01:49:17.000000000 +0800
+++ config_default.sh.bak	2015-01-13 10:04:12.482482913 +0800
@@ -32,7 +32,7 @@
     -c
     -Wall
     -O2
-    $(wx-config --static=no --debug=no --cflags)
+    $(wx-config-2.8 --static=no --debug=no --cflags)
     "
 
 RC=""
@@ -41,7 +41,7 @@
 LD="g++"
 LDFLAGS="-s"
 LDLIBS="
-    $(wx-config --static=no --debug=no --libs)
+    $(wx-config-2.8 --static=no --debug=no --libs)
     -lcurl
     -lcrypto
     "
