Index: ecore/ecore.c_ecore_exe.pxi
===================================================================
--- ecore/ecore.c_ecore_exe.pxi	(revision 55991)
+++ ecore/ecore.c_ecore_exe.pxi	(working copy)
@@ -91,7 +91,7 @@
 
 
 cdef class ExeEventFilter:
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self.exe = NULL
         self.handler = NULL
         self.owner = None
@@ -280,7 +280,7 @@
     One may delete handlers explicitly with L{EventHandler.delete()}
     method.
     """
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self.exe = NULL
         self.__data = None
         self.__callbacks = {}
Index: ecore/evas/ecore.evas.c_ecore_evas_base.pxi
===================================================================
--- ecore/evas/ecore.evas.c_ecore_evas_base.pxi	(revision 55991)
+++ ecore/evas/ecore.evas.c_ecore_evas_base.pxi	(working copy)
@@ -203,7 +203,7 @@
      callback_pre_render, callback_resize, callback_show, callback_sticky,
      callback_unsticky
     """
-    def __new__(self, *args, **kargs):
+    def __cinit__(self, *args, **kargs):
         self.obj = NULL
         self.data = dict()
 
Index: ecore/win32/ecore.win32.c_ecore_win32_window.pxi
===================================================================
--- ecore/win32/ecore.win32.c_ecore_win32_window.pxi	(revision 55991)
+++ ecore/win32/ecore.win32.c_ecore_win32_window.pxi	(working copy)
@@ -20,7 +20,7 @@
 cdef class Window:
     cdef Ecore_Win32_Window *obj
 
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self.obj = NULL
 
     def __init__(self, Window parent=None, int x=0, int y=0, int w=1, int h=1,
