diff -Naur liferea-1.8.0.orig/src/ui/ui_indicator.c liferea-1.8.0/src/ui/ui_indicator.c
--- liferea-1.8.0.orig/src/ui/ui_indicator.c	2011-11-28 22:50:58.000000000 +0100
+++ liferea-1.8.0/src/ui/ui_indicator.c	2011-12-12 03:37:52.348847077 +0100
@@ -179,7 +179,7 @@
 	pixbuf = gdk_pixbuf_new_from_file (node->iconFile, NULL);
 
 	/* display favicon */
-	indicate_indicator_set_property_icon (indicator, "icon", pixbuf);
+	indicate_gtk_indicator_set_property_icon (indicator, "icon", pixbuf);
 	gdk_pixbuf_unref (pixbuf);
 
 	sprintf (count, "%u", node->unreadCount);
diff -Naur liferea-1.8.0.orig/src/ui/ui_indicator.h liferea-1.8.0/src/ui/ui_indicator.h
--- liferea-1.8.0.orig/src/ui/ui_indicator.h	2011-11-28 22:50:58.000000000 +0100
+++ liferea-1.8.0/src/ui/ui_indicator.h	2011-12-12 03:36:46.006276952 +0100
@@ -21,6 +21,10 @@
 #ifndef _UI_INDICATOR_H
 #define _UI_INDICATOR_H
 
+#ifdef HAVE_CONFIG_H
+	#include <config.h>
+#endif
+
 #include <glib.h>
 
 /**
