--- cairo-dock-plug-ins/CMakeLists.txt	2011-06-20 05:06:14.814428000 +0600
+++ cairo-dock-plug-ins-build/CMakeLists.txt	2011-06-20 05:14:19.940465986 +0600
@@ -197,7 +197,7 @@
 ############# DBUS #################
 message (STATUS "> DBus:")
 message (STATUS " * Python:")
-find_program (PYTHON_EXECUTABLE python)
+find_program (PYTHON_EXECUTABLE python2)
 EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "import sys; print sys.version" OUTPUT_VARIABLE PYTHON_VERSION)
 if ("${PYTHON_VERSION}" STREQUAL "")
        message (STATUS "Could not find Python, won't install Python interface.")
