--- src/beat-box/wscript	2011-05-23 17:15:33.283011000 +0200
+++ src/beat-box/wscript	2011-05-23 17:19:15.763113972 +0200
@@ -54,37 +54,37 @@
 def build(bld):
 	#install basic desktop file
 	bld.install_files('${PREFIX}/share/applications', '/data/' + APPNAME + '.desktop')
-	
+
 	#install icons
 	bld.install_files('${PREFIX}/share/icons/hicolor/scalable/apps', '/images/icons/128x128/apps/beatbox.svg');
-   	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/apps', '/images/icons/16x16/apps/beatbox.svg');
-   	bld.install_files('${PREFIX}/share/icons/hicolor/22x22/apps', '/images/icons/22x22/apps/beatbox.svg');
-   	bld.install_files('${PREFIX}/share/icons/hicolor/24x24/apps', '/images/icons/24x24/apps/beatbox.svg');
-   	bld.install_files('${PREFIX}/share/icons/hicolor/32x32/apps', '/images/icons/32x32/apps/beatbox.svg');
-   	bld.install_files('${PREFIX}/share/icons/hicolor/48x48/apps', '/images/icons/48x48/apps/beatbox.svg');
-   	bld.install_files('${PREFIX}/share/icons/hicolor/64x64/apps', '/images/icons/64x64/apps/beatbox.svg');
-   	bld.install_files('${PREFIX}/share/icons/hicolor/128x128/apps', '/images/icons/128x128/apps/beatbox.svg');
-   	
-   	#shuffle and repeat
-   	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/status', '/images/icons/16x16/status/media-playlist-repeat-active-symbolic.svg');
+	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/apps', '/images/icons/16x16/apps/beatbox.svg');
+	bld.install_files('${PREFIX}/share/icons/hicolor/22x22/apps', '/images/icons/22x22/apps/beatbox.svg');
+	bld.install_files('${PREFIX}/share/icons/hicolor/24x24/apps', '/images/icons/24x24/apps/beatbox.svg');
+	bld.install_files('${PREFIX}/share/icons/hicolor/32x32/apps', '/images/icons/32x32/apps/beatbox.svg');
+	bld.install_files('${PREFIX}/share/icons/hicolor/48x48/apps', '/images/icons/48x48/apps/beatbox.svg');
+	bld.install_files('${PREFIX}/share/icons/hicolor/64x64/apps', '/images/icons/64x64/apps/beatbox.svg');
+	bld.install_files('${PREFIX}/share/icons/hicolor/128x128/apps', '/images/icons/128x128/apps/beatbox.svg');
+
+	#shuffle and repeat
+	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/status', '/images/icons/16x16/status/media-playlist-repeat-active-symbolic.svg');
 	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/status', '/images/icons/16x16/status/media-playlist-repeat-symbolic.svg');
 	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/status', '/images/icons/16x16/status/media-playlist-shuffle-active-symbolic.svg');
 	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/status', '/images/icons/16x16/status/media-playlist-shuffle-symbolic.svg');
-	
+
 	#star, not star
 	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/status', '/images/icons/16x16/status/starred.svg');
 	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/status', '/images/icons/16x16/status/not-starred.svg');
-	
+
 	#playlist icons
 	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/status', '/images/icons/16x16/mimes/playlist.svg');
 	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/status', '/images/icons/16x16/mimes/playlist-automatic.svg');
 	bld.install_files('${PREFIX}/share/icons/hicolor/22x22/status', '/images/icons/22x22/mimes/playlist.svg');
 	bld.install_files('${PREFIX}/share/icons/hicolor/22x22/status', '/images/icons/22x22/mimes/playlist-automatic.svg');
-	
+
 	#last fm
 	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/actions', '/images/icons/16x16/actions/lastfm-love.svg');
 	bld.install_files('${PREFIX}/share/icons/hicolor/16x16/actions', '/images/icons/16x16/actions/lastfm-ban.svg');
-	
+
 	bld.program(
 		packages      = 'gtk+-2.0 gee-1.0 gstreamer-0.10 taglib_c gio-2.0 sqlheavy-0.1 libxml-2.0 gconf-2.0 libnotify gnet-2.0 unique-1.0 libsoup-2.4 json-glib-1.0',
 		target        = APPNAME,
