diff --git a/PKLAEntry.h b/PKLAEntry.h
--- a/PKLAEntry.h
+++ b/PKLAEntry.h
@@ -24,6 +24,7 @@ class PKLAEntry {
     QString resultAny;
     QString resultInactive;
     QString resultActive;
+    QString filePath;
 
     int filePriority;
     int fileOrder;
@@ -33,6 +34,7 @@ class PKLAEntry {
     static QString textFromImpl(PolkitQt1::ActionDescription::ImplicitAuthorization implicit);
 };
 Q_DECLARE_METATYPE(PKLAEntry)
+Q_DECLARE_METATYPE(QList<PKLAEntry>);
 
 typedef QList<PKLAEntry> PKLAEntryList;
 
diff --git a/PKLAEntry.cpp b/PKLAEntry.cpp
--- a/PKLAEntry.cpp
+++ b/PKLAEntry.cpp
@@ -14,7 +14,7 @@
 QDBusArgument& operator<<(QDBusArgument& argument, const PKLAEntry& entry)
 {
     argument.beginStructure();
-    argument << entry.title << entry.identity << entry.action << entry.resultAny << entry.resultInactive << entry.resultActive
+    argument << entry.title << entry.identity << entry.action << entry.resultAny << entry.resultInactive << entry.resultActive << entry.filePath
              << entry.filePriority << entry.fileOrder;
     argument.endStructure();
     return argument;
@@ -23,7 +23,7 @@ QDBusArgument& operator<<(QDBusArgument& argument, const PKLAEntry& entry)
 const QDBusArgument& operator>>(const QDBusArgument& argument, PKLAEntry& entry)
 {
     argument.beginStructure();
-    argument >> entry.title >> entry.identity >> entry.action >> entry.resultAny >> entry.resultInactive >> entry.resultActive
+    argument >> entry.title >> entry.identity >> entry.action >> entry.resultAny >> entry.resultInactive >> entry.resultActive >> entry.filePath
              >> entry.filePriority >> entry.fileOrder;
     argument.endStructure();
     return argument;
diff --git a/helper/org.kde.polkitkde1.helper.xml b/helper/org.kde.polkitkde1.helper.xml
--- a/helper/org.kde.polkitkde1.helper.xml
+++ b/helper/org.kde.polkitkde1.helper.xml
@@ -12,8 +12,8 @@
            <arg direction="out" type="av" />
        </method>
        <method name="writePolicy" >
-           <annotation name="com.trolltech.QtDBus.QtTypeName.In0" value="QVariantList"/>
-           <arg direction="in" type="av" />
+           <annotation name="com.trolltech.QtDBus.QtTypeName.In0" value="QList&lt;PKLAEntry&gt;"/>
+           <arg direction="in" type="a(sssssssii)" />
        </method>
 
    </interface>
diff --git a/helper/polkitkde1helper.h b/helper/polkitkde1helper.h
--- a/helper/polkitkde1helper.h
+++ b/helper/polkitkde1helper.h
@@ -11,6 +11,7 @@
 #ifndef POLKITKDE1HELPER_H
 #define POLKITKDE1HELPER_H
 
+#include <QDir>
 #include <QtCore/QObject>
 #include <QtDBus/QDBusContext>
 #include <QtDBus/qdbusargument.h>
@@ -28,11 +29,12 @@ class PolkitKde1Helper : public QObject, protected QDBusContext
     public Q_SLOTS:
         void saveGlobalConfiguration(const QString &adminIdentities, int systemPriority, int policiesPriority);
         QVariantList retrievePolicies();
-        void writePolicy(const QVariantList &policy);
+        void writePolicy(const QList<PKLAEntry> &policy);
 
     private:
         QVariantList entriesFromFile(int filePriority, const QString &fileContents);
         QString formatPKLAEntry(const PKLAEntry &entry);
+        QFileInfoList oldNestedList;
 };
 
 #endif // POLKITKDE1HELPER_H
diff --git a/helper/polkitkde1helper.cpp b/helper/polkitkde1helper.cpp
--- a/helper/polkitkde1helper.cpp
+++ b/helper/polkitkde1helper.cpp
@@ -29,6 +29,7 @@ PolkitKde1Helper::PolkitKde1Helper(QObject* parent)
     : QObject(parent)
 {
     qDBusRegisterMetaType<PKLAEntry>();
+    qDBusRegisterMetaType<QList<PKLAEntry> >();
     (void) new HelperAdaptor(this);
     // Register the DBus service
     if (!QDBusConnection::systemBus().registerService("org.kde.polkitkde1.helper")) {
@@ -119,6 +120,9 @@ QVariantList PolkitKde1Helper::retrievePolicies()
             qDebug() << "Parsing file " << nestedInfo.absoluteFilePath();
             retlist.append(entriesFromFile(filePriority, nestedInfo.absoluteFilePath()));
         }
+
+        // Save all the fileinfo's, for a later delete if we choose to save
+        oldNestedList << nestedList;
     }
 
     return retlist;
@@ -141,6 +145,7 @@ QVariantList PolkitKde1Helper::entriesFromFile(int filePriority, const QString&
         if (line.startsWith('[')) {
             // Ok, that's a key entry
             PKLAEntry entry;
+            entry.filePath = filePath;
             entry.title = line.split('[').last().split(']').first();
             entry.filePriority = filePriority;
             entry.fileOrder = priority;
@@ -165,7 +170,7 @@ QVariantList PolkitKde1Helper::entriesFromFile(int filePriority, const QString&
 
                 // Did we parse it all?
                 if (!entry.identity.isEmpty() && !entry.action.isEmpty() && !entry.resultActive.isEmpty() &&
-                    !entry.resultAny.isEmpty() && !entry.resultInactive.isEmpty()) {
+                    !entry.resultAny.isEmpty() && !entry.resultInactive.isEmpty() && !entry.filePath.isEmpty()) {
                     // Awesome, add and wave goodbye. And also increase the priority
                     ++priority;
                     qDebug() << "PKLA Parsed:" << entry.title << entry.action << entry.identity << entry.resultAny
@@ -180,36 +185,60 @@ QVariantList PolkitKde1Helper::entriesFromFile(int filePriority, const QString&
     return retlist;
 }
 
-void PolkitKde1Helper::writePolicy(const QVariantList& policy)
+void PolkitKde1Helper::writePolicy(const QList<PKLAEntry>& policy)
 {
-    PKLAEntryList entries;
-    foreach (const QVariant &variant, policy) {
-        entries.append(variant.value<PKLAEntry>());
+    QList<PKLAEntry> entries = policy;
+
+    // First delete all the old files, we do not need them anymore
+    foreach(const QFileInfo &nestedInfo, oldNestedList) {
+        QFile::remove(nestedInfo.absoluteFilePath());
     }
+    oldNestedList.clear();
+
     qSort(entries.begin(), entries.end(), orderByPriorityLessThan);
 
     QSettings kdesettings("/etc/polkit-1/polkit-kde-1.conf", QSettings::IniFormat);
     kdesettings.beginGroup("General");
 
-    QString path = QString("/var/lib/polkit-1/localauthority/%1-polkitkde/%2.conf")
-                          .arg(kdesettings.value("PoliciesPriority",75).toInt()).arg(entries.first().action);
+    QString pathName = QString("/var/lib/polkit-1/localauthority/%1-polkitkde.d/")
+                            .arg(kdesettings.value("PoliciesPriority",75).toInt());
 
-    if (QFile::exists(path)) {
-        QFile::remove(path);
-    }
+    foreach(const PKLAEntry &entry, entries) {
+        QString fullPath;
+
+        // First check if it already has a filePath,
+        // else generate fullPath from the policy action name
+        if (!entry.filePath.isEmpty()) {
+            fullPath = entry.filePath;
+        } else {
+            QStringList dotSplittedFileName = entry.action.split(".");
 
-    QString contents;
+            // Skip the action name
+            dotSplittedFileName.removeLast();
+            dotSplittedFileName << "pkla";
+
+            // Check if the polkitkde.d dir exists. If not, create it.
+            QDir path(pathName);
+
+            if (!path.exists()) {
+                if (!path.mkpath(path.absolutePath())) {
+                    return;
+                }
+            }
 
-    foreach (const PKLAEntry &entry, entries) {
+            fullPath = pathName + dotSplittedFileName.join(".");
+        }
+
+        QString contents;
         contents.append(formatPKLAEntry(entry));
         contents.append('\n');
-    }
 
-    QFile wfile(path);
-    wfile.open(QIODevice::WriteOnly | QIODevice::Truncate | QIODevice::Text);
-    wfile.write(contents.toUtf8());
-    wfile.flush();
-    wfile.close();
+        QFile wfile(fullPath);
+        wfile.open(QIODevice::Append | QIODevice::Truncate | QIODevice::Text);
+        wfile.write(contents.toUtf8());
+        wfile.flush();
+        wfile.close();
+    }
 }
 
 QString PolkitKde1Helper::formatPKLAEntry(const PKLAEntry& entry)
diff --git a/polkitactions/PolkitActionsKCM.cpp b/polkitactions/PolkitActionsKCM.cpp
--- a/polkitactions/PolkitActionsKCM.cpp
+++ b/polkitactions/PolkitActionsKCM.cpp
@@ -23,6 +23,7 @@
 #include "PolicyItem.h"
 #include <QDBusMessage>
 #include <QDBusConnection>
+#include <QDBusMetaType>
 #include <qdbuspendingcall.h>
 
 K_PLUGIN_FACTORY(KCMPolkitActionsFactory,
@@ -46,6 +47,9 @@ PolkitActionsKCM::PolkitActionsKCM(QWidget* parent, const QVariantList& args)
     setAboutData(about);
 
     qRegisterMetaType<PolkitQt1::ActionDescription>();
+    qRegisterMetaType<PKLAEntry>("PKLAEntry");
+    qDBusRegisterMetaType<PKLAEntry>();
+    qDBusRegisterMetaType<QList<PKLAEntry> >();
 
     // Build the UI
     m_ui->setupUi(this);
@@ -90,11 +94,11 @@ void PolkitActionsKCM::save()
                                                           "org.kde.polkitkde1.helper",
                                                           QLatin1String("writePolicy"));
     QList<QVariant> argumentList;
-    QVariantList policies;
+    QList<PKLAEntry> policies;
     foreach (const PKLAEntry &entry, m_actionWidget.data()->entries()) {
-        policies << QVariant::fromValue(entry);
+        policies << entry;
     }
-    argumentList << policies;
+    argumentList << QVariant::fromValue(policies);
 
     message.setArguments(argumentList);
 
-- 
1.7.6

