--- geogebra-old	2011-08-27 19:13:37.000000000 +0200
+++ geogebra	2011-08-28 08:29:13.350440149 +0200
@@ -4,23 +4,6 @@
 #---------------------------------------------
 
 #---------------------------------------------
-# Used environment variables:
-#
-# GG_SCRIPTNAME=<name of originally called script to start GeoGebra> # If unset, name of this script will be used
-#
-# GG_PATH=<path of directory containing geogebra.jar> # If unset, path of this script will be used.
-# In this case if the path of script does not contain the geogebra.jar file, /usr/share/geogebra will be used.
-#
-# JAVACMD=<Java command> # If unset, java will be used
-
-#---------------------------------------------
-# If GG_SCRIPTNAME not set, use name of this script
-
-if [ -z "$GG_SCRIPTNAME" ]; then
-	GG_SCRIPTNAME=$(basename $0)
-fi
-
-#---------------------------------------------
 # Define usage function
 
 func_usage()
@@ -57,13 +40,6 @@
 }
 
 #---------------------------------------------
-# If JAVACMD not set, use java
-
-if [ -z "$JAVACMD" ]; then
-	JAVACMD='java'
-fi
-
-#---------------------------------------------
 # Check for option --help and pass memory options to Java, others to GeoGebra
 
 for i in "$@"; do
@@ -101,24 +77,6 @@
 fi
 
 #---------------------------------------------
-# If GG_PATH not set, use path of this script
-
-if [ -z "$GG_PATH" ]; then
-	GG_PATH="${BASH_SOURCE[0]}"
-	if [ -h "${GG_PATH}" ]; then
-		while [ -h "${GG_PATH}" ]; do
-		GG_PATH=`readlink "${GG_PATH}"`
-		done
-	fi
-	pushd . > /dev/null
-	cd `dirname ${GG_PATH}` > /dev/null
-	GG_PATH=`pwd`
-	popd > /dev/null
-	
-	test -r "${GG_PATH}/geogebra.jar" || GG_PATH=/usr/share/geogebra
-fi
-
-#---------------------------------------------
 # Run
 
-exec "$JAVACMD" $JAVA_OPTS -jar "$GG_PATH/geogebra.jar" $GG_OPTS "$@"
+exec java $JAVA_OPTS -jar "/usr/share/java/geogebra42-beta/geogebra.jar" $GG_OPTS "$@"
