diff --git a/src/icalendar/prop.py b/src/icalendar/prop.py
index ea412e9..3527b5f 100644
--- a/src/icalendar/prop.py
+++ b/src/icalendar/prop.py
@@ -586,7 +586,7 @@ class vDDDTypes:
                 wrong_type_used = 0
         if wrong_type_used:
             raise ValueError ('You must use datetime, date or timedelta')
-        if isinstance(dt, date):
+        if isinstance(dt, date) and not isinstance(dt, datetime):
             self.params = Parameters(dict(value='DATE'))
 
         self.dt = dt
