--- Make.old/makedefs	2008-12-19 04:22:30.027021800 -0500
+++ Make/makedefs	2008-12-19 04:22:10.927067616 -0500
@@ -82,8 +82,8 @@
 BININST    = bin
 LIBINST    = lib
 
-INST_LOCATION  = /usr/local
-INST_LIBS      = $(INST_LOCATION)/lib$(ARCHINST)
+INST_LOCATION  = /usr
+INST_LIBS      = $(INST_LOCATION)/lib
 INST_INCLUDE   = $(INST_LOCATION)/include
 INST_SHARE     = $(INST_LOCATION)/share
 INST_EXAMPLES  = $(INST_LOCATION)/bin
