--- strhandle.cc.org	2000-06-20 21:36:58.000000000 +0200
+++ strhandle.cc	2010-01-14 11:33:54.695507689 +0100
@@ -24,7 +24,7 @@
 /* Like str_cutat, but the beginning of str is copied to buffer at left */
 char *str_splitat(char const * const str, char *left, char cut)
 {
-    char *pos = strchr(str,cut);
+    char *pos = (char *)strchr(str,cut);
     if (!pos) {
         strcpy(left,str);
         return NULL;
