diff -ur a/twitux-0.69/src/twitux-app.c b/twitux-0.69/src/twitux-app.c
--- a/twitux-0.69/src/twitux-app.c	2009-01-25 03:24:00.000000000 +0200
+++ b/twitux-0.69/src/twitux-app.c	2010-05-02 10:54:37.000000000 +0300
@@ -1470,7 +1470,7 @@
 	GdkPixbuf	 *pixbuf;
 	GError		 *error = NULL;
 
-	pixbuf = gdk_pixbuf_new_from_file (file, &error);
+	pixbuf = gdk_pixbuf_new_from_file_at_size (file, 48, 48, &error);
 
 	if (!pixbuf){
 		twitux_debug (DEBUG_DOMAIN_SETUP, "Image error: %s: %s",
diff -ur a/twitux-0.69/src/twitux-network.c b/twitux-0.69/src/twitux-network.c
--- a/twitux-0.69/src/twitux-network.c	2009-01-23 04:04:50.000000000 +0200
+++ b/twitux-0.69/src/twitux-network.c	2010-05-02 10:51:50.000000000 +0300
@@ -379,7 +379,7 @@
 
 /* Get an image from servers */
 void
-twitux_network_get_image (const gchar  *url_image,
+twitux_network_get_image (const TwituxUser	*user,
 						  GtkTreeIter   iter)
 {
 	gchar	*image_file;
@@ -387,13 +387,16 @@
 
 	TwituxImage *image;
 
-	/* save using the filename */
-	image_name = strrchr (url_image, '/');
-	if (image_name && image_name[1] != '\0') {
-		image_name++;
-	} else {
-		image_name = "twitux_unknown_image";
+	/* save using the username as the filename */
+        /* Bugfix #412390 */
+	if (user->screen_name)
+	{
+            image_name = user->screen_name;
 	}
+        else
+        {
+            image_name = "twitux_unknown_image";        	
+        }
 
 	image_file = g_build_filename (g_get_home_dir(), ".gnome2",
 								   TWITUX_CACHE_IMAGES,
@@ -414,7 +417,7 @@
 	g_free (image_file);
 
 	/* Note: 'image' will be freed in 'network_cb_on_image' */
-	network_get_data (url_image, network_cb_on_image, image);
+	network_get_data (user->image_url, network_cb_on_image, image);
 }
 
 
diff -ur a/twitux-0.69/src/twitux-network.h b/twitux-0.69/src/twitux-network.h
--- a/twitux-0.69/src/twitux-network.h	2009-01-19 05:00:26.000000000 +0200
+++ b/twitux-0.69/src/twitux-network.h	2010-05-02 10:12:14.000000000 +0300
@@ -72,7 +72,7 @@
 GList *twitux_network_get_followers	(void);
 
 /* Get an image from servers */
-void twitux_network_get_image (const gchar *url_image,
+void twitux_network_get_image (const TwituxUser *user,
 							   GtkTreeIter iter);
 
 /* Add a user to follow */
diff -ur a/twitux-0.69/src/twitux-parser.c b/twitux-0.69/src/twitux-parser.c
--- a/twitux-0.69/src/twitux-parser.c	2009-01-25 04:25:56.000000000 +0200
+++ b/twitux-0.69/src/twitux-parser.c	2010-05-02 10:13:16.000000000 +0300
@@ -278,8 +278,7 @@
 			g_free (tweet);
 
 			/* Get Image */
-			twitux_network_get_image (status->user->image_url,
-									  iter);
+			twitux_network_get_image (status->user, iter);
 
 			/* Free struct */
 			parser_free_user (status->user);
