diff --git a/libpurple/protocols/oscar/clientlogin.c b/libpurple/protocols/oscar/clientlogin.c
index a6e37fc..06b3315 100644
--- a/libpurple/protocols/oscar/clientlogin.c
+++ b/libpurple/protocols/oscar/clientlogin.c
@@ -74,20 +74,18 @@ static const gchar *get_start_oscar_session_url(OscarData *od)
 }
 
 /*
- * Using clientLogin requires a developer ID.  This key is for libpurple.
- * It is the default key for all libpurple-based clients.  AOL encourages
- * UIs (especially ones with lots of users) to override this with their
- * own key.  This key is owned by the AIM account "markdoliner"
+ * Using clientLogin requires a developer ID.  This key is stolen from
+ * the official AIR client.  AOL encourages UIs (especially ones with
+ * lots of users) to override this with their own key, and then blocks
+ * them.
  *
  * Keys can be managed at http://developer.aim.com/manageKeys.jsp
  */
-#define DEFAULT_CLIENT_KEY "ma15d7JTxbmVG-RP"
+#define DEFAULT_CLIENT_KEY "ic1-IIcaJnnNV5xA"
 
 static const char *get_client_key(OscarData *od)
 {
-	return oscar_get_ui_info_string(
-			od->icq ? "prpl-icq-clientkey" : "prpl-aim-clientkey",
-			DEFAULT_CLIENT_KEY);
+	return DEFAULT_CLIENT_KEY;
 }
 
 static gchar *generate_error_message(xmlnode *resp, const char *url)
