diff -Naur src/jsonrpc.h src/jsonrpc.h
--- src/jsonrpc.h 2012-02-02 11:32:34.000000000 -0600
+++ src/jsonrpc.h 2013-02-21 15:43:51.886277000 -0600
@@ -42,7 +42,7 @@
 #define JSONRPC_CPP_VERSION_STRING "0.4.0"
 
 /* include from external jsoncpp lib */
-#include <json/json.h>
+#include <jsoncpp/json/json.h>
 
 /* include all headers from JsonRpc-Cpp lib */
 #include "jsonrpc_common.h"
diff -Naur src/jsonrpc_handler.h src/jsonrpc_handler.h
--- src/jsonrpc_handler.h 2011-05-06 11:36:10.000000000 -0500
+++ src/jsonrpc_handler.h 2013-02-21 15:44:02.686029000 -0600
@@ -30,7 +30,7 @@
 #include <string>
 #include <list>
 
-#include <json/json.h>
+#include <jsoncpp/json/json.h>
 
 #include "jsonrpc_common.h"

diff -Naur examples/test-rpc.h examples/test-rpc.h
--- examples/test-rpc.h 2011-05-06 11:40:41.000000000 -0500
+++ examples/test-rpc.h 2013-02-21 15:44:43.351753000 -0600
@@ -25,7 +25,7 @@
 #ifndef TEST_RPC_H
 #define TEST_RPC_H 
 
-#include <json/json.h>
+#include <jsoncpp/json/json.h>
 
 /**
  * \class TestRpc
