--- plugins/output-pngfile/output-pngfile.c	2013-04-23 06:00:36.304800268 +0200
+++ plugins/output-pngfile/output-pngfile.c	2013-04-23 06:01:32.274801731 +0200
@@ -198,7 +198,7 @@
 		const RSIccProfile *profile = rs_color_space_get_icc_profile(pngfile->color_space, pngfile->save16bit);
 		rs_icc_profile_get_data(profile, &data, &data_length);
 
-		png_set_iCCP(png_ptr, info_ptr, (png_charp)rs_color_space_get_name(pngfile->color_space), PNG_COMPRESSION_TYPE_BASE, (png_charp)data, data_length);
+		png_set_iCCP(png_ptr, info_ptr, (png_charp)rs_color_space_get_name(pngfile->color_space), PNG_COMPRESSION_TYPE_BASE, (png_const_bytep)data, data_length);
 		if (pngfile->save16bit)
 			png_set_gAMA(png_ptr, info_ptr, 1.0);
 	}
