--- rawstudio-org/profiles/Makefile.am	2013-02-10 17:11:50.211256000 +0100
+++ rawstudio-build/profiles/Makefile.am	2013-04-13 01:08:29.584911645 +0200
@@ -1019,7 +1019,7 @@
 	simple-src/Leica-D-LUX-4-simple.xml\
 	simple-src/SONY_SLTA99V-simple.xml\
 	simple-src/SONY_NEX-6-simple.xml\
-	simple-src/SONY_SLTA37-simple.xml\	
+	simple-src/SONY_SLTA37-simple.xml\
 	simple-src/PENTAX_K-30-simple.xml\
 	simple-src/PENTAX_Q10-simple.xml\
 	simple-src/PENTAX_K-5_II_s-simple.xml\
