diff -crB rtorrent/src/rtorrent-0.8.6/src/ui/download_list.cc rtorrent-vi/src/rtorrent-0.8.6/src/ui/download_list.cc
*** rtorrent/src/rtorrent-0.8.6/src/ui/download_list.cc	2009-11-12 03:03:48.000000000 -0500
--- rtorrent-vi/src/rtorrent-0.8.6/src/ui/download_list.cc	2010-03-19 00:32:57.000000000 -0400
***************
*** 345,353 ****
      m_uiArray[DISPLAY_LOG]->bindings()['B' - '@'] =
      m_uiArray[DISPLAY_LOG]->bindings()[' '] = sigc::bind(sigc::mem_fun(*this, &DownloadList::activate_display), DISPLAY_DOWNLOAD_LIST);
  
    m_uiArray[DISPLAY_DOWNLOAD_LIST]->bindings()[KEY_RIGHT] =
      m_uiArray[DISPLAY_DOWNLOAD_LIST]->bindings()['F' - '@'] = sigc::bind(sigc::mem_fun(*this, &DownloadList::activate_display), DISPLAY_DOWNLOAD);
!   m_uiArray[DISPLAY_DOWNLOAD_LIST]->bindings()['l']       = sigc::bind(sigc::mem_fun(*this, &DownloadList::activate_display), DISPLAY_LOG);
  }
  
  }
--- 345,360 ----
      m_uiArray[DISPLAY_LOG]->bindings()['B' - '@'] =
      m_uiArray[DISPLAY_LOG]->bindings()[' '] = sigc::bind(sigc::mem_fun(*this, &DownloadList::activate_display), DISPLAY_DOWNLOAD_LIST);
  
+   m_uiArray[DISPLAY_LOG]->bindings()['h'] =
+     m_uiArray[DISPLAY_LOG]->bindings()['B' - '@'] =
+     m_uiArray[DISPLAY_LOG]->bindings()[' '] = sigc::bind(sigc::mem_fun(*this, &DownloadList::activate_display), DISPLAY_DOWNLOAD_LIST);
+ 
    m_uiArray[DISPLAY_DOWNLOAD_LIST]->bindings()[KEY_RIGHT] =
      m_uiArray[DISPLAY_DOWNLOAD_LIST]->bindings()['F' - '@'] = sigc::bind(sigc::mem_fun(*this, &DownloadList::activate_display), DISPLAY_DOWNLOAD);
! 
!   m_uiArray[DISPLAY_DOWNLOAD_LIST]->bindings()['l'] =
!     m_uiArray[DISPLAY_DOWNLOAD_LIST]->bindings()['F' - '@'] = sigc::bind(sigc::mem_fun(*this, &DownloadList::activate_display), DISPLAY_DOWNLOAD);
!   m_uiArray[DISPLAY_DOWNLOAD_LIST]->bindings()['L']       = sigc::bind(sigc::mem_fun(*this, &DownloadList::activate_display), DISPLAY_LOG);
  }
  
  }
diff -crB rtorrent/src/rtorrent-0.8.6/src/ui/element_chunks_seen.cc rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_chunks_seen.cc
*** rtorrent/src/rtorrent-0.8.6/src/ui/element_chunks_seen.cc	2009-11-12 03:03:48.000000000 -0500
--- rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_chunks_seen.cc	2010-03-19 00:32:57.000000000 -0400
***************
*** 54,61 ****
--- 54,65 ----
  
    m_bindings[KEY_LEFT] = m_bindings['B' - '@'] = sigc::mem_fun(&m_slotExit, &slot_type::operator());
  
+   m_bindings['h'] = m_bindings['B' - '@'] = sigc::mem_fun(&m_slotExit, &slot_type::operator());
+ 
    m_bindings[KEY_DOWN]  = m_bindings['N' - '@'] = sigc::mem_fun(*this, &ElementChunksSeen::receive_next);
+   m_bindings['j']  = m_bindings['N' - '@'] = sigc::mem_fun(*this, &ElementChunksSeen::receive_next);
    m_bindings[KEY_UP]    = m_bindings['P' - '@'] = sigc::mem_fun(*this, &ElementChunksSeen::receive_prev);
+   m_bindings['k']    = m_bindings['P' - '@'] = sigc::mem_fun(*this, &ElementChunksSeen::receive_prev);
    m_bindings[KEY_NPAGE] = sigc::mem_fun(*this, &ElementChunksSeen::receive_pagenext);
    m_bindings[KEY_PPAGE] = sigc::mem_fun(*this, &ElementChunksSeen::receive_pageprev);
  }
diff -crB rtorrent/src/rtorrent-0.8.6/src/ui/element_download_list.cc rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_download_list.cc
*** rtorrent/src/rtorrent-0.8.6/src/ui/element_download_list.cc	2009-11-12 03:03:48.000000000 -0500
--- rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_download_list.cc	2010-03-19 00:32:57.000000000 -0400
***************
*** 96,102 ****
--- 96,104 ----
    m_bindings['9']           = sigc::bind(sigc::mem_fun(*this, &ElementDownloadList::receive_change_view), "active");
  
    m_bindings[KEY_UP]   = m_bindings['P' - '@'] = sigc::mem_fun(*this, &ElementDownloadList::receive_prev);
+   m_bindings['k']   = m_bindings['P' - '@'] = sigc::mem_fun(*this, &ElementDownloadList::receive_prev);
    m_bindings[KEY_DOWN] = m_bindings['N' - '@'] = sigc::mem_fun(*this, &ElementDownloadList::receive_next);
+   m_bindings['j'] = m_bindings['N' - '@'] = sigc::mem_fun(*this, &ElementDownloadList::receive_next);
  }
  
  void
diff -crB rtorrent/src/rtorrent-0.8.6/src/ui/element_file_list.cc rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_file_list.cc
*** rtorrent/src/rtorrent-0.8.6/src/ui/element_file_list.cc	2009-11-12 03:03:48.000000000 -0500
--- rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_file_list.cc	2010-03-19 00:32:57.000000000 -0400
***************
*** 65,71 ****
--- 65,73 ----
    m_collapsed(false) {
  
    m_bindings[KEY_LEFT] = m_bindings['B' - '@']  = sigc::mem_fun(&m_slotExit, &slot_type::operator());
+   m_bindings['h'] = m_bindings['B' - '@']  = sigc::mem_fun(&m_slotExit, &slot_type::operator());
    m_bindings[KEY_RIGHT] = m_bindings['F' - '@'] = sigc::mem_fun(*this, &ElementFileList::receive_select);
+   m_bindings['l'] = m_bindings['F' - '@'] = sigc::mem_fun(*this, &ElementFileList::receive_select);
  
    m_bindings[' '] = sigc::mem_fun(*this, &ElementFileList::receive_priority);
    m_bindings['*'] = sigc::mem_fun(*this, &ElementFileList::receive_change_all);
***************
*** 74,80 ****
--- 76,84 ----
    m_bindings[KEY_PPAGE] = sigc::mem_fun(*this, &ElementFileList::receive_pageprev);
  
    m_bindings[KEY_DOWN] = m_bindings['N' - '@'] = sigc::mem_fun(*this, &ElementFileList::receive_next);
+   m_bindings['j'] = m_bindings['N' - '@'] = sigc::mem_fun(*this, &ElementFileList::receive_next);
    m_bindings[KEY_UP]   = m_bindings['P' - '@'] = sigc::mem_fun(*this, &ElementFileList::receive_prev);
+   m_bindings['k']   = m_bindings['P' - '@'] = sigc::mem_fun(*this, &ElementFileList::receive_prev);
  }
  
  inline ElementText*
diff -crB rtorrent/src/rtorrent-0.8.6/src/ui/element_menu.cc rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_menu.cc
*** rtorrent/src/rtorrent-0.8.6/src/ui/element_menu.cc	2009-11-12 03:03:48.000000000 -0500
--- rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_menu.cc	2010-03-19 00:32:57.000000000 -0400
***************
*** 72,82 ****
    m_entry(entry_invalid) {
  
    // Move bindings into a function that defines default bindings.
!   m_bindings[KEY_LEFT] = m_bindings['B' - '@']  = sigc::mem_fun(&m_slotExit, &slot_type::operator());  
    m_bindings[KEY_RIGHT] = m_bindings['F' - '@'] = sigc::mem_fun(this, &ElementMenu::entry_select);
  
    m_bindings[KEY_UP]   = m_bindings['P' - '@'] = sigc::mem_fun(this, &ElementMenu::entry_prev);
    m_bindings[KEY_DOWN] = m_bindings['N' - '@'] = sigc::mem_fun(this, &ElementMenu::entry_next);
  }
  
  ElementMenu::~ElementMenu() {
--- 72,86 ----
    m_entry(entry_invalid) {
  
    // Move bindings into a function that defines default bindings.
!   m_bindings[KEY_LEFT] = m_bindings['B' - '@']  = sigc::mem_fun(&m_slotExit, &slot_type::operator());
!   m_bindings['h'] = m_bindings['B' - '@']  = sigc::mem_fun(&m_slotExit, &slot_type::operator());  
    m_bindings[KEY_RIGHT] = m_bindings['F' - '@'] = sigc::mem_fun(this, &ElementMenu::entry_select);
+   m_bindings['l'] = m_bindings['F' - '@'] = sigc::mem_fun(this, &ElementMenu::entry_select);
  
    m_bindings[KEY_UP]   = m_bindings['P' - '@'] = sigc::mem_fun(this, &ElementMenu::entry_prev);
+   m_bindings['k']   = m_bindings['P' - '@'] = sigc::mem_fun(this, &ElementMenu::entry_prev);
    m_bindings[KEY_DOWN] = m_bindings['N' - '@'] = sigc::mem_fun(this, &ElementMenu::entry_next);
+   m_bindings['j'] = m_bindings['N' - '@'] = sigc::mem_fun(this, &ElementMenu::entry_next);
  }
  
  ElementMenu::~ElementMenu() {
diff -crB rtorrent/src/rtorrent-0.8.6/src/ui/element_peer_list.cc rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_peer_list.cc
*** rtorrent/src/rtorrent-0.8.6/src/ui/element_peer_list.cc	2009-11-12 03:03:48.000000000 -0500
--- rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_peer_list.cc	2010-03-19 00:32:57.000000000 -0400
***************
*** 73,86 ****
  
    m_elementInfo->slot_exit(sigc::bind(sigc::mem_fun(this, &ElementPeerList::activate_display), DISPLAY_LIST));
  
!   m_bindings['k']       = sigc::mem_fun(this, &ElementPeerList::receive_disconnect_peer);
    m_bindings['*']       = sigc::mem_fun(this, &ElementPeerList::receive_snub_peer);
    m_bindings['B']       = sigc::mem_fun(this, &ElementPeerList::receive_ban_peer);
!   m_bindings[KEY_LEFT] = m_bindings['B' - '@']  = sigc::mem_fun(&m_slotExit, &slot_type::operator());  
    m_bindings[KEY_RIGHT] = m_bindings['F' - '@'] = sigc::bind(sigc::mem_fun(this, &ElementPeerList::activate_display), DISPLAY_INFO);
  
    m_bindings[KEY_UP]   = m_bindings['P' - '@'] = sigc::mem_fun(this, &ElementPeerList::receive_prev);
    m_bindings[KEY_DOWN] = m_bindings['N' - '@'] = sigc::mem_fun(this, &ElementPeerList::receive_next);
  }
  
  ElementPeerList::~ElementPeerList() {
--- 73,90 ----
  
    m_elementInfo->slot_exit(sigc::bind(sigc::mem_fun(this, &ElementPeerList::activate_display), DISPLAY_LIST));
  
!   m_bindings['K']       = sigc::mem_fun(this, &ElementPeerList::receive_disconnect_peer);
    m_bindings['*']       = sigc::mem_fun(this, &ElementPeerList::receive_snub_peer);
    m_bindings['B']       = sigc::mem_fun(this, &ElementPeerList::receive_ban_peer);
!   m_bindings[KEY_LEFT] = m_bindings['B' - '@']  = sigc::mem_fun(&m_slotExit, &slot_type::operator());
!   m_bindings['h'] = m_bindings['B' - '@']  = sigc::mem_fun(&m_slotExit, &slot_type::operator());  
    m_bindings[KEY_RIGHT] = m_bindings['F' - '@'] = sigc::bind(sigc::mem_fun(this, &ElementPeerList::activate_display), DISPLAY_INFO);
+   m_bindings['l'] = m_bindings['F' - '@'] = sigc::bind(sigc::mem_fun(this, &ElementPeerList::activate_display), DISPLAY_INFO);
  
    m_bindings[KEY_UP]   = m_bindings['P' - '@'] = sigc::mem_fun(this, &ElementPeerList::receive_prev);
+   m_bindings['k']   = m_bindings['P' - '@'] = sigc::mem_fun(this, &ElementPeerList::receive_prev);
    m_bindings[KEY_DOWN] = m_bindings['N' - '@'] = sigc::mem_fun(this, &ElementPeerList::receive_next);
+   m_bindings['j'] = m_bindings['N' - '@'] = sigc::mem_fun(this, &ElementPeerList::receive_next);
  }
  
  ElementPeerList::~ElementPeerList() {
diff -crB rtorrent/src/rtorrent-0.8.6/src/ui/element_text.cc rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_text.cc
*** rtorrent/src/rtorrent-0.8.6/src/ui/element_text.cc	2009-11-12 03:03:48.000000000 -0500
--- rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_text.cc	2010-03-19 00:32:57.000000000 -0400
***************
*** 54,60 ****
    m_columnWidth(0) {
  
    // Move bindings into a function that defines default bindings.
!   m_bindings[KEY_LEFT] = m_bindings['B' - '@'] = sigc::mem_fun(&m_slotExit, &slot_type::operator());  
  
  //   m_bindings[KEY_UP]    = sigc::mem_fun(this, &ElementText::entry_prev);
  //   m_bindings[KEY_DOWN]  = sigc::mem_fun(this, &ElementText::entry_next);
--- 54,62 ----
    m_columnWidth(0) {
  
    // Move bindings into a function that defines default bindings.
!   m_bindings[KEY_LEFT] = m_bindings['B' - '@'] = sigc::mem_fun(&m_slotExit, &slot_type::operator());
! 
!   m_bindings['h'] = m_bindings['B' - '@'] = sigc::mem_fun(&m_slotExit, &slot_type::operator());  
  
  //   m_bindings[KEY_UP]    = sigc::mem_fun(this, &ElementText::entry_prev);
  //   m_bindings[KEY_DOWN]  = sigc::mem_fun(this, &ElementText::entry_next);
diff -crB rtorrent/src/rtorrent-0.8.6/src/ui/element_tracker_list.cc rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_tracker_list.cc
*** rtorrent/src/rtorrent-0.8.6/src/ui/element_tracker_list.cc	2009-11-12 03:03:48.000000000 -0500
--- rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_tracker_list.cc	2010-03-19 00:32:57.000000000 -0400
***************
*** 55,66 ****
--- 55,69 ----
    m_focus(0) {
  
    m_bindings[KEY_LEFT] = m_bindings['B' - '@'] = sigc::mem_fun(&m_slotExit, &slot_type::operator());
+   m_bindings['h'] = m_bindings['B' - '@'] = sigc::mem_fun(&m_slotExit, &slot_type::operator());
  
    m_bindings[' ']      = sigc::mem_fun(*this, &ElementTrackerList::receive_cycle_group);
    m_bindings['*']      = sigc::mem_fun(*this, &ElementTrackerList::receive_disable);
  
    m_bindings[KEY_DOWN] = m_bindings['N' - '@'] = sigc::mem_fun(*this, &ElementTrackerList::receive_next);
+   m_bindings['j'] = m_bindings['N' - '@'] = sigc::mem_fun(*this, &ElementTrackerList::receive_next);
    m_bindings[KEY_UP]   = m_bindings['P' - '@'] = sigc::mem_fun(*this, &ElementTrackerList::receive_prev);
+   m_bindings['k']   = m_bindings['P' - '@'] = sigc::mem_fun(*this, &ElementTrackerList::receive_prev);
  }
  
  void
diff -crB rtorrent/src/rtorrent-0.8.6/src/ui/element_transfer_list.cc rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_transfer_list.cc
*** rtorrent/src/rtorrent-0.8.6/src/ui/element_transfer_list.cc	2009-11-12 03:03:48.000000000 -0500
--- rtorrent-vi/src/rtorrent-0.8.6/src/ui/element_transfer_list.cc	2010-03-19 00:32:57.000000000 -0400
***************
*** 54,61 ****
--- 54,65 ----
  
    m_bindings[KEY_LEFT] = m_bindings['B' - '@'] = sigc::mem_fun(&m_slotExit, &slot_type::operator());
  
+   m_bindings['h'] = m_bindings['B' - '@'] = sigc::mem_fun(&m_slotExit, &slot_type::operator());
+ 
    m_bindings[KEY_DOWN]  = sigc::mem_fun(*this, &ElementTransferList::receive_next);
+   m_bindings['j']  = sigc::mem_fun(*this, &ElementTransferList::receive_next);
    m_bindings[KEY_UP]    = sigc::mem_fun(*this, &ElementTransferList::receive_prev);
+   m_bindings['k']    = sigc::mem_fun(*this, &ElementTransferList::receive_prev);
    m_bindings[KEY_NPAGE] = sigc::mem_fun(*this, &ElementTransferList::receive_pagenext);
    m_bindings[KEY_PPAGE] = sigc::mem_fun(*this, &ElementTransferList::receive_pageprev);
  }
