--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -710,8 +710,10 @@
 	/* Walk through all bpp values. Luckily they're all nicely spaced with 2
 	 * bpc in between. */
 	bpp = pipe_config->pipe_bpp;
-	if (is_edp(intel_dp) && dev_priv->vbt.edp_bpp)
-		bpp = min_t(int, bpp, dev_priv->vbt.edp_bpp);
+	if (is_edp(intel_dp) && dev_priv->vbt.edp_bpp && (dev_priv->quirks & QUIRK_IGNORE_VBT_BPP) == 0) {
+		DRM_DEBUG_KMS("clamping bpp for eDP panel to BIOS-provided %i\n", dev_priv->vbt.edp_bpp);
+ 		bpp = min_t(int, bpp, dev_priv->vbt.edp_bpp);
+	}

 	for (; bpp >= 6*3; bpp -= 2*3) {
 		mode_rate = intel_dp_link_required(adjusted_mode->clock, bpp);
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -9426,6 +9426,15 @@
 	DRM_INFO("applying no-PCH_PWM_ENABLE quirk\n");
 }

+/* Some machines (ux31a) advertise the panel should use 18bpp, but it lies.
+ */
+static void quirk_ignore_vbt_bpp(struct drm_device *dev)
+{
+	struct drm_i915_private *dev_priv = dev->dev_private;
+	dev_priv->quirks |= QUIRK_IGNORE_VBT_BPP;
+	DRM_INFO("applying IGNORE_VBT_BPP quirk\n");
+}
+
 struct intel_quirk {
 	int device;
 	int subsystem_vendor;
@@ -9500,6 +9509,9 @@
 	{ 0x0116, 0x1028, 0x052e, quirk_no_pcm_pwm_enable },
 	/* Dell XPS13 HD and XPS13 FHD Ivy Bridge */
 	{ 0x0166, 0x1028, 0x058b, quirk_no_pcm_pwm_enable },
+
+	/* Asus Zenbook UX31A Ivybridge eDP */
+	{ 0x0166, 0x1043, 0x1517, quirk_ignore_vbt_bpp },
 };

 static void intel_init_quirks(struct drm_device *dev)
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -556,6 +556,7 @@
 #define QUIRK_LVDS_SSC_DISABLE (1<<1)
 #define QUIRK_INVERT_BRIGHTNESS (1<<2)
 #define QUIRK_NO_PCH_PWM_ENABLE (1<<3)
+#define QUIRK_IGNORE_VBT_BPP	(1<<4)

 struct intel_fbdev;
 struct intel_fbc_work;
