diff -ur a/mm/vmscan.c b/mm/vmscan.c
--- a/mm/vmscan.c	2015-02-09 05:54:22.000000000 +0300
+++ b/mm/vmscan.c	2015-02-28 02:00:24.000000000 +0300
@@ -2519,6 +2519,11 @@
 	unsigned long writeback_threshold;
 	bool zones_reclaimable;
 
+#ifdef CONFIG_FREEZER
+	if (unlikely(pm_freezing && !sc->hibernation_mode))
+		return 0;
+#endif
+
 	delayacct_freepages_start();
 
 	if (global_reclaim(sc))
diff -ur a/fs/super.c b/fs/super.c
--- a/fs/super.c	2015-02-28 02:10:07.000000000 +0300
+++ b/fs/super.c	2015-02-28 02:22:53.492933254 +0300
@@ -37,8 +37,10 @@
 
 
 LIST_HEAD(super_blocks);
+EXPORT_SYMBOL_GPL(super_blocks);
 DEFINE_SPINLOCK(sb_lock);
 
+
 static char *sb_writers_name[SB_FREEZE_LEVELS] = {
 	"sb_writers",
 	"sb_pagefaults",
@@ -462,7 +464,7 @@
 			return ERR_PTR(-ENOMEM);
 		goto retry;
 	}
-		
+
 	err = set(s, data);
 	if (err) {
 		spin_unlock(&sb_lock);
@@ -563,7 +565,7 @@
 /**
  *	get_super - get the superblock of a device
  *	@bdev: device to get the superblock for
- *	
+ *
  *	Scans the superblock list and finds the superblock of the file system
  *	mounted on the device given. %NULL is returned if no match is found.
  */
@@ -653,7 +655,7 @@
 	spin_unlock(&sb_lock);
 	return NULL;
 }
- 
+
 struct super_block *user_get_super(dev_t dev)
 {
 	struct super_block *sb;
diff -ur a/include/linux/mm.h b/include/linux/mm.h
--- a/include/linux/mm.h	2015-02-28 02:10:07.000000000 +0300
+++ b/include/linux/mm.h	2015-02-28 09:19:44.989007129 +0300
@@ -241,7 +241,7 @@
 /*
  * These are the virtual MM functions - opening of an area, closing and
  * unmapping it (needed to keep files on disk up-to-date etc), pointer
- * to the functions called when a no-page or a wp-page exception occurs. 
+ * to the functions called when a no-page or a wp-page exception occurs.
  */
 struct vm_operations_struct {
 	void (*open)(struct vm_area_struct * area);
@@ -2110,6 +2110,9 @@
 					void __user *, size_t *, loff_t *);
 #endif
 
+void drop_pagecache(void);
+void drop_slab_node(int nid);
+
 unsigned long shrink_node_slabs(gfp_t gfp_mask, int nid,
 				unsigned long nr_scanned,
 				unsigned long nr_eligible);
