diff -ur trunk.bak/src/sdcv.cpp trunk/src/sdcv.cpp
--- trunk.bak/src/sdcv.cpp	2011-09-18 22:13:30.551808751 -0500
+++ trunk/src/sdcv.cpp	2011-09-19 02:14:42.341920018 -0500
@@ -50,6 +50,21 @@
 	}
 };
 
+struct GetFilename {
+	GetFilename(std::string bookname_, std::string& filename_) :
+		bookname(bookname_), the_filename(filename_)
+    {}
+	void operator()(const std::string& filename, bool) {
+		DictInfo dict_info;
+        if (dict_info.load_from_ifo_file(filename, false) &&
+            dict_info.bookname.compare(bookname) == 0)
+            the_filename = filename;
+	}
+private:
+    std::string bookname;
+    std::string& the_filename;
+};
+
 struct CreateDisableList {
 	CreateDisableList(gchar **use_dist_list, strlist_t& disable_list_) :
 		disable_list(disable_list_)
@@ -176,6 +191,7 @@
 	}
 
 	strlist_t disable_list;
+    strlist_t order_list;
 	//DictInfoList  dict_info_list;
   
 	if (use_dict_list) {
@@ -183,6 +199,18 @@
 		CreateDisableList create_disable_list(get_impl(use_dict_list), disable_list);
 		for_each_file(dicts_dir_list, ".ifo", empty_list, 
 			      empty_list, create_disable_list);
+        gchar **p = get_impl(use_dict_list);
+        while (*p) {
+            std::string filename;
+            GetFilename get_file_name(*p, filename);
+            for_each_file(dicts_dir_list, ".ifo", empty_list,
+                          empty_list, get_file_name);
+            if (!filename.empty() &&
+                std::find(order_list.begin(), order_list.end(), 
+			      filename)==order_list.end())
+                order_list.push_back(filename);
+            ++p;
+        }
 	}
 
     
@@ -192,8 +220,7 @@
 
   
 	Library lib(utf8_input, utf8_output);
-	strlist_t empty_list;
-	lib.load(dicts_dir_list, empty_list, disable_list);
+	lib.load(dicts_dir_list, order_list, disable_list);
 
 
 	std::auto_ptr<read_line> io(create_readline_object());
