diff -ruPN ori/modules/python/Makefile.am new/modules/python/Makefile.am
--- ori/modules/python/Makefile.am	2010-05-01 11:49:09.433690402 +0200
+++ new/modules/python/Makefile.am	2010-05-02 18:22:23.909410423 +0200
@@ -12,9 +12,9 @@
 install-exec-am: all
 	cp build/*/dionaea/*.so $(DESTDIR)$(pkglibdir)/python.so
 	rm -rf $(DESTDIR)$(pkglibdir)/python/dionaea/core.so
-	ln -s $(DESTDIR)$(pkglibdir)/python.so $(DESTDIR)$(pkglibdir)/python/dionaea/core.so
+	ln -s ../../python.so $(DESTDIR)$(pkglibdir)/python/dionaea/core.so
 	rm -rf $(DESTDIR)$(pkglibdir)/python/dionaea/pyev.so
-	ln -s $(DESTDIR)$(pkglibdir)/python.so $(DESTDIR)$(pkglibdir)/python/dionaea/pyev.so
+	ln -s ../../python.so $(DESTDIR)$(pkglibdir)/python/dionaea/pyev.so
 
 clean:
 	python setup.py clean

