--- dionaea/modules/python/Makefile.am	2011-05-31 11:22:54.629809144 +0200
+++ dionaea-build/modules/python/Makefile.am	2011-05-31 11:53:31.439763330 +0200
@@ -7,14 +7,14 @@
 all: binding.c
 
 binding.c: setup.py setup.py.in binding.pyx module.c module.h ../../config.h ../../include/protocol.h ../../include/dionaea.h
-	$(PYTHON) setup.py build
+	 python2 setup.py build
 
 install-exec-am: all
 	cp build/*/dionaea/*.so $(DESTDIR)$(pkglibdir)/python.so
 	rm -rf $(DESTDIR)$(pkglibdir)/python/dionaea/core.so
-	ln -s $(DESTDIR)$(pkglibdir)/python.so $(DESTDIR)$(pkglibdir)/python/dionaea/core.so
+	ln -s ../../python.so $(DESTDIR)$(pkglibdir)/python/dionaea/core.so
 	rm -rf $(DESTDIR)$(pkglibdir)/python/dionaea/pyev.so
-	ln -s $(DESTDIR)$(pkglibdir)/python.so $(DESTDIR)$(pkglibdir)/python/dionaea/pyev.so
+	ln -s ../../python.so $(DESTDIR)$(pkglibdir)/python/dionaea/pyev.so
 
 clean:
 	$(PYTHON) setup.py clean
