--- ./configure.orig	2013-12-14 15:42:47.367732350 +0100
+++ ./configure	2013-12-14 15:39:44.622049795 +0100
@@ -220,9 +220,12 @@
 	chmod 755 \$(PREFIX)/include/board
 	chmod 644 \$(PREFIX)/include/Board.h
 	chmod 644 \$(PREFIX)/include/board/*.h
+	chmod 644 \$(PREFIX)/include/board/*.ih
 	install -d \$(PREFIX)/share/libboard/examples
 	\$(CP) examples/*.cpp \$(PREFIX)/share/libboard/examples
 	\$(CP) examples/README \$(PREFIX)/share/libboard/examples
+	chmod 644 \$(PREFIX)/share/libboard/examples/*.cpp
+	chmod 644 \$(PREFIX)/share/libboard/examples/README
 	\$(CP) ${LICENSE} \$(PREFIX)/share/libboard/
 
 doc: doc/html/index.html
