Index: src/arguments.c
===================================================================
--- src/arguments.c	(revision 105)
+++ src/arguments.c	(local copy)
@@ -78,6 +78,7 @@
     { "atmega32u2",     tar_atmega32u2,     adc_AVR,   0x2FF0, 0x03eb, 0x08000, 0x1000, true,  128, true, false,  128, 0x0400 },
     { "atmega16u4",     tar_atmega16u4,     adc_AVR,   0x2FF3, 0x03eb, 0x04000, 0x1000, true,  128, true,  false, 128, 0x0200 },
     { "atmega8u2",      tar_atmega8u2,      adc_AVR,   0x2FEE, 0x03eb, 0x02000, 0x1000, true,  128, true,  false, 128, 0x0200 },
+    { "atmega16u2",     tar_atmega16u2,     adc_AVR,   0x2FEF, 0x03eb, 0x04000, 0x1000, true,  128, true,  false, 128, 0x0200 },
     { "at32uc3a0128",   tar_at32uc3a0128,   adc_AVR32, 0x2FF8, 0x03eb, 0x20000, 0x2000, false, 512, false, true,  0,   0      },
     { "at32uc3a1128",   tar_at32uc3a1128,   adc_AVR32, 0x2FF8, 0x03eb, 0x20000, 0x2000, false, 512, false, true,  0,   0      },
     { "at32uc3a0256",   tar_at32uc3a0256,   adc_AVR32, 0x2FF8, 0x03eb, 0x40000, 0x2000, false, 512, false, true,  0,   0      },
Index: src/arguments.h
===================================================================
--- src/arguments.h	(revision 105)
+++ src/arguments.h	(local copy)
@@ -56,6 +56,7 @@
                     tar_atmega32u2,
                     tar_atmega16u4,
                     tar_atmega8u2,
+                    tar_atmega16u2,
                     tar_at32uc3b064,
                     tar_at32uc3b164,
                     tar_at32uc3b0128,
