--- ./dvdrip-queue1	2011-03-01 19:31:14.000000000 -0500
+++ ./dvdrip-queue2	2011-03-01 19:38:54.416640063 -0500
@@ -200,7 +200,7 @@
 			version = "0.0.0"
 			
 		
-		if (version == '0.98.6') or (version == '0.98.7') or (version == '0.98.8'):
+		if (version == '0.98.6') or (version == '0.98.7') or (version == '0.98.8') or (version == '0.98.9') or (version == '0.98.10') or (version == '0.98.11'):
 			exit
 		else:
 			self.wTree.get_widget("versionError").show()
