diff -rupN eclim_1.7.3.orig/ant/build.gant eclim_1.7.3/ant/build.gant
--- eclim_1.7.3.orig/ant/build.gant	2012-03-01 16:41:46.000000000 +0100
+++ eclim_1.7.3/ant/build.gant	2012-03-19 16:21:40.962794407 +0100
@@ -145,7 +145,7 @@ target(name: 'build.vim'){
 
 target(name: 'deploy'){
   depends('deploy.eclipse', 'deploy.vim')
-  mkdir(dir: '${user.home}/.eclim/resources/ext')
+  mkdir(dir: '${vim.files}/eclim/resources/ext')
 }
 
 target(name: 'deploy.eclipse'){
