diff -rupN eclim_1.7.2.orig/src/doc/extension/eclim/sphinx/builder.py eclim_1.7.2/src/doc/extension/eclim/sphinx/builder.py
--- eclim_1.7.2.orig/src/doc/extension/eclim/sphinx/builder.py	2011-02-17 16:55:14.000000000 +0100
+++ eclim_1.7.2/src/doc/extension/eclim/sphinx/builder.py	2012-01-24 22:04:13.473570968 +0100
@@ -125,7 +125,7 @@ class EclimBuilder(StandaloneHTMLBuilder
 # EV: end main toc
 
     # local TOC and global TOC tree
-    toc = self.render_partial(self.env.get_toc_for(docname))['fragment']
+    toc = self.render_partial(self.env.get_toc_for(docname,0))['fragment']
 
     return dict(
       parents = parents,
