diff -Naur original/plugins/place/place.xml.in modified/plugins/place/place.xml.in
--- original/plugins/place/place.xml.in	2014-06-10 21:24:10.445049000 +0100
+++ modified/plugins/place/place.xml.in	2014-06-10 21:47:52.089133150 +0100
@@ -20,8 +20,8 @@
 	    <option name="mode" type="int">
 		<_short>Placement Mode</_short>
 		<_long>Algorithm to use for window placement</_long>
-		<default>0</default>
-		<min>2</min>
+		<default>2</default>
+		<min>0</min>
 		<max>5</max>
 		<desc>
 		    <value>0</value>
