diff --git a/qt5/immodule/quimplatforminputcontextplugin.pro.in b/qt5/immodule/quimplatforminputcontextplugin.pro.in
index 4696d12..6bdc888 100644
--- a/qt5/immodule/quimplatforminputcontextplugin.pro.in
+++ b/qt5/immodule/quimplatforminputcontextplugin.pro.in
@@ -38,5 +38,5 @@ SOURCES += @srcdir@/quimplatforminputcontext.cpp \
 OTHER_FILES += uim.json
 
 TARGET = uimplatforminputcontextplugin
-
+DESTDIR = 
 target.path += @DESTDIR@$$[QT_INSTALL_PLUGINS]/platforminputcontexts
