--- giftui-0.4.1.orig/src/ui_transfer_cb.c
+++ giftui-0.4.1/src/ui_transfer_cb.c
@@ -18,12 +18,12 @@
  */
 
 #include "main.h"
+#include "gtkcellrendererprogress.h"
 
 #include <string.h>
 #include <libgift/libgift.h>
 #include <gtk/gtk.h>
 
-#include "gtkcellrendererprogress.h"
 
 #include "io.h"
 #include "event.h"
--- giftui-0.4.1.orig/src/ui_transfer.c
+++ giftui-0.4.1/src/ui_transfer.c
@@ -18,13 +18,13 @@
  */
 
 #include "main.h"
+#include "gtkcellrendererprogress.h"
 
 #include <string.h>
 #include <libgift/libgift.h>
 #include <gdk/gdkkeysyms.h>
 #include <gtk/gtk.h>
 
-#include "gtkcellrendererprogress.h"
 
 #include "io.h"
 #include "event.h"
--- giftui-0.4.1.orig/src/main.c
+++ giftui-0.4.1/src/main.c
@@ -197,6 +197,8 @@
 	bindtextdomain (PACKAGE, LOCALEDIR);
 	textdomain (PACKAGE);
 #endif	
+	/* init GObject system */
+	g_type_init();
 	
 	/* configuration init & load */
 	giftui_config_init ();
--- giftui-0.4.1.orig/src/ui_about.c
+++ giftui-0.4.1/src/ui_about.c
@@ -153,7 +153,7 @@
 	gtk_container_add (GTK_CONTAINER (scrolled), about->list);
 	
 	/* create columns */
-	for (i = 0; i < ABOUT_ICON ; i++)
+	for (i = 0; i < ABOUT_VERSION ; i++)
 	{
 		column = gtk_tree_view_column_new ();
 		gtk_tree_view_column_set_title (column, column_headers[i]);
