diff --git a/src/vogleditor/vogleditor_qlaunchtracerdialog.cpp b/src/vogleditor/vogleditor_qlaunchtracerdialog.cpp
index 9dc2478..23d3381 100644
--- a/src/vogleditor/vogleditor_qlaunchtracerdialog.cpp
+++ b/src/vogleditor/vogleditor_qlaunchtracerdialog.cpp
@@ -24,7 +24,7 @@ QString vogleditor_QLaunchTracerDialog::get_command_line()
     QString executable;
     if (ui->steamLauncherCheckBox->isChecked())
     {
-        executable = appDirectory.absoluteFilePath("../../bin/steamlauncher.sh");
+        executable = appDirectory.absoluteFilePath("/usr/bin/steamlauncher");
 
         if (ui->amd64CheckBox->isChecked())
         {
@@ -106,14 +106,18 @@ QProcessEnvironment vogleditor_QLaunchTracerDialog::get_process_environment()
         }
 
         QString LD_PRELOAD = "";
+#ifdef __amd64__
         if (ui->amd64CheckBox->isChecked())
         {
-            LD_PRELOAD += appDirectory.absoluteFilePath("../../vogl_build/bin/libvogltrace64.so");
+            LD_PRELOAD += appDirectory.absoluteFilePath("/usr/lib/libvogltrace.so");
         }
         else
         {
-            LD_PRELOAD += appDirectory.absoluteFilePath("../../vogl_build/bin/libvogltrace32.so");
+            LD_PRELOAD += appDirectory.absoluteFilePath("/usr/lib32/libvogltrace.so");
         }
+#else
+        LD_PRELOAD += appDirectory.absoluteFilePath("/usr/lib/libvogltrace.so");
+#endif
 
         if (getenv("LD_PRELOAD"))
         {
