diff -aur toxic-0.5.2.orig/src/toxic.c toxic-0.5.2/src/toxic.c
--- toxic-0.5.2.orig/src/toxic.c	2014-10-14 17:30:03.035850257 +0200
+++ toxic-0.5.2/src/toxic.c	2014-10-14 17:30:51.095662823 +0200
@@ -650,7 +650,7 @@
             exit_toxic_err("failed in load_data", FATALERR_FILEOP);
         }
 
-        bool is_encrypted = tox_is_data_encrypted((uint8_t *) buf);
+        bool is_encrypted = tox_is_save_encrypted((uint8_t *) buf);
 
         /* attempt to encrypt an already encrypted data file */
         if (arg_opts.encrypt_data && is_encrypted)
