--- server/trunk/vendor/plugins/project_management/lib/zip/zip.rb	2009/07/20 10:57:03	1315
+++ server/trunk/vendor/plugins/project_management/lib/zip/zip.rb	2010/06/27 15:52:16	1965
@@ -1,7 +1,7 @@
 require 'delegate'
 require 'singleton'
 require 'tempfile'
-require 'ftools'
+#require 'ftools'
 require 'stringio'
 require 'zlib'
 require 'zip/stdrubyext'
@@ -1566,7 +1566,7 @@
       tmpFilename = tmpfile.path
       tmpfile.close
       if yield tmpFilename
-	File.move(tmpFilename, name)
+	FileUtils.mv(tmpFilename, name)
       end
     end

