diff -r -u4 a/kernel/nv-linux.h b/kernel/nv-linux.h
--- a/kernel/nv-linux.h	2012-08-04 12:55:46.838997064 -0700
+++ b/kernel/nv-linux.h	2012-08-04 12:56:00.928021702 -0700
@@ -785,9 +785,9 @@
 #elif (NV_KMEM_CACHE_CREATE_ARGUMENT_COUNT == 5)
 #define NV_KMEM_CACHE_CREATE(kmem_cache, name, type)            \
     {                                                           \
         kmem_cache = kmem_cache_create(name, sizeof(type),      \
-                        0, 0, NULL);                            \
+                        0, SLAB_USERCOPY, NULL);                \
     }
 #else
 #error "NV_KMEM_CACHE_CREATE_ARGUMENT_COUNT value unrecognized!"
 #endif
