Index: kopete/libkopete/private/kopeteviewmanager.cpp
===================================================================
--- a/kopete/libkopete/private/kopeteviewmanager.cpp	(révision 1277741)
+++ b/kopete/libkopete/private/kopeteviewmanager.cpp	(copie de travail)
@@ -308,13 +308,16 @@
 			if ( msg.from() && d->eventList.isEmpty() ) // may happen for internal messages
 				showNotification = true;
 
-			event = new Kopete::MessageEvent( msg, session );
-			d->eventList.append( event );
+			if ( !isActiveWindow )
+			{
+				event = new Kopete::MessageEvent( msg, session );
+				d->eventList.append( event );
 
-			// Don't call readMessages twice. We call it later in this method. Fixes bug 168978.
-			if ( d->useQueue )
-				connect( event, SIGNAL(done(Kopete::MessageEvent*)),
-				         this, SLOT(slotEventDeleted(Kopete::MessageEvent*)) );
+				// Don't call readMessages twice. We call it later in this method. Fixes bug 168978.
+				if ( d->useQueue )
+					connect( event, SIGNAL(done(Kopete::MessageEvent*)),
+					         this, SLOT(slotEventDeleted(Kopete::MessageEvent*)) );
+			}
 		}
 	}
 
