--- ../../NewWave4OB/openbox-3/themerc	2009-04-26 12:41:10.000000000 -0400
+++ ./themerc	2009-06-27 11:19:56.000000000 -0400
@@ -107,9 +107,9 @@
 menu.items.disabled.text.color: #999999
 
 menu.items.active.bg: flat gradient vertical border
-menu.items.active.bg.border.color: #FFE8B7
-menu.items.active.bg.color: #FFC16F
-menu.items.active.bg.colorTo: #FF926F
+menu.items.active.bg.border.color: #E5E5E5
+menu.items.active.bg.color: #CDCDCD
+menu.items.active.bg.colorTo: #A1A1A1
 menu.items.active.text.color: #000000
 menu.items.active.disabled.text.color: #4E4E4E
 
@@ -123,9 +123,9 @@
 osd.bg.border.color: #00ff00
 
 osd.hilight.bg: flat gradient vertical
-osd.hilight.bg.border.color: #A76215
-osd.hilight.bg.color: #F5BB7E
-osd.hilight.bg.colorTo: #F5987E
+osd.hilight.bg.border.color: #646464
+osd.hilight.bg.color: #CDCDCD
+osd.hilight.bg.colorTo: #DCDCDC
 
 osd.unhilight.bg: flat gradient vertical
 osd.unhilight.bg.color: #d3d3d3
