--- setup.py.old	2013-05-23 09:56:28.358395900 +0200
+++ setup.py	2013-05-23 09:56:43.318395953 +0200
@@ -24,7 +24,7 @@
 
 def version_tagged_in_git():
     try:
-        return str(subprocess.check_output(["git", "describe", "--always", "--tags", "--match", "*.*.*.*"]), 'utf8').strip()
+        return str(subprocess.check_output(["git", "describe", "--always", "--tags", "--match", "*.*.*.*", "--abbrev=0"]), 'utf8').strip()
     except:
         return _version
 
