--- src/CMakeLists.txt	2012-10-17 15:40:18.000000000 +0200
+++ src/CMakeLists.txt	2012-12-17 17:09:49.599126787 +0100
@@ -16,11 +16,11 @@
 
 
 # include files (.h files) directories
-include_directories($ENV{ROOTSYS}/include
+include_directories($ENV{ROOTSYS}/include/root
                                         ${TRUEE_BINARY_DIR}/include)
 
 # library directories
-link_directories($ENV{ROOTSYS}/lib)
+link_directories($ENV{ROOTSYS}/lib/root)
 
 # Create a library called "TrueeCore"
 # which includes the source files "*.cxx"
