--- ptrtd.c.orig	2008-08-08 13:38:06.000000000 +0200
+++ ptrtd.c	2008-08-08 13:38:34.000000000 +0200
@@ -201,13 +201,13 @@
 	if( do_config )
 	{
 		printf( "Tunnel: %s\n", ifname );
-		sprintf( cmd, "/sbin/ip link set %s up", ifname );
+		sprintf( cmd, "/usr/sbin/ip link set %s up", ifname );
 		printf( "command: %s\n", cmd );
 		system( cmd );
-		sprintf( cmd, "/sbin/ip addr add fe80::1/64 dev %s", ifname );
+		sprintf( cmd, "/usr/sbin/ip addr add fe80::1/64 dev %s", ifname );
 		printf( "command: %s\n", cmd );
 		system( cmd );
-		sprintf( cmd, "/sbin/ip route add %s/%d dev %s via fe80::5",
+		sprintf( cmd, "/usr/sbin/ip route add %s/%d dev %s via fe80::5",
 				prefix, plen, ifname );
 		printf( "command: %s\n", cmd );
 		system( cmd );
