diff -c tpserver/frame.cpp /home/jm/builds/tpserver-cpp-0.5.0/tpserver/frame.cpp
*** tpserver/frame.cpp	2007-12-05 12:52:07.000000000 +0800
--- /home/jm/builds/tpserver-cpp-0.5.0/tpserver/frame.cpp	2008-05-05 11:08:08.000000000 +0800
***************
*** 18,23 ****
--- 18,24 ----
   *
   */
  
+ #include <cstdlib>
  #include <string.h>
  #include <netinet/in.h>
  #include <string>
diff -c tpserver/logging.cpp /home/jm/builds/tpserver-cpp-0.5.0/tpserver/logging.cpp
*** tpserver/logging.cpp	2007-05-30 13:52:13.000000000 +0800
--- /home/jm/builds/tpserver-cpp-0.5.0/tpserver/logging.cpp	2008-05-05 11:08:35.000000000 +0800
***************
*** 18,23 ****
--- 18,24 ----
   *
   */
  
+ #include <cstdlib>
  #include <iostream>
  #include <stdio.h>
  #include <stdarg.h>
diff -c tpserver/net.h /home/jm/builds/tpserver-cpp-0.5.0/tpserver/net.h
*** tpserver/net.h	2008-02-26 20:46:19.000000000 +0800
--- /home/jm/builds/tpserver-cpp-0.5.0/tpserver/net.h	2008-05-05 11:06:14.000000000 +0800
***************
*** 21,26 ****
--- 21,27 ----
   */
  
  //#include <pthread.h>
+ #include <algorithm>
  #include <map>
  #include <queue>
  #include <functional>
