--- configure.bup	2012-02-03 10:42:19.425380720 +0100
+++ configure	2012-02-03 10:43:58.231512108 +0100
@@ -22455,7 +22455,7 @@
 
 		{ { echo "$as_me:$LINENO: error: GL/gl.h or OpenGL/gl.h is needed, please specify its location with --with-gl-inc.  If this still fails, please contact henryj@paradise.net.nz, include the string FTGL somewhere in the subject line and provide a copy of the config.log file that was left behind." >&5
 echo "$as_me: error: GL/gl.h or OpenGL/gl.h is needed, please specify its location with --with-gl-inc.  If this still fails, please contact henryj@paradise.net.nz, include the string FTGL somewhere in the subject line and provide a copy of the config.log file that was left behind." >&2;}
-   { (exit 1); exit 1; }; }
+   { (echo "ignoring missing GL library."); true; }; }
 
 fi
 
@@ -22645,7 +22645,7 @@
 echo "${ECHO_T}no" >&6; }
     { { echo "$as_me:$LINENO: error: GL library could not be found, please specify its location with --with-gl-lib.  If this still fails, please contact henryj@paradise.net.nz, include the string FTGL somewhere in the subject line and provide a copy of the config.log file that was left behind." >&5
 echo "$as_me: error: GL library could not be found, please specify its location with --with-gl-lib.  If this still fails, please contact henryj@paradise.net.nz, include the string FTGL somewhere in the subject line and provide a copy of the config.log file that was left behind." >&2;}
-   { (exit 1); exit 1; }; }
+   { (echo "ignoring missing GL library."); true; }; }
 fi
 
 if test "${ac_cv_header_GL_glu_h+set}" = set; then
@@ -22926,7 +22926,7 @@
 
 		{ { echo "$as_me:$LINENO: error: GL/glu.h or OpenGL/glu.h is needed, please specify its location with --with-gl-inc.  If this still fails, please contact henryj@paradise.net.nz, include the string FTGL somewhere in the subject line and provide a copy of the config.log file that was left behind." >&5
 echo "$as_me: error: GL/glu.h or OpenGL/glu.h is needed, please specify its location with --with-gl-inc.  If this still fails, please contact henryj@paradise.net.nz, include the string FTGL somewhere in the subject line and provide a copy of the config.log file that was left behind." >&2;}
-   { (exit 1); exit 1; }; }
+   { (echo "ignoring missing GLU library."); true; }; }
 
 fi
 
@@ -23115,7 +23115,7 @@
 echo "${ECHO_T}no" >&6; }
     { { echo "$as_me:$LINENO: error: GLU library could not be found, please specify its location with --with-gl-lib.  If this still fails, please contact henryj@paradise.net.nz, include the string FTGL somewhere in the subject line and provide a copy of the config.log file that was left behind." >&5
 echo "$as_me: error: GLU library could not be found, please specify its location with --with-gl-lib.  If this still fails, please contact henryj@paradise.net.nz, include the string FTGL somewhere in the subject line and provide a copy of the config.log file that was left behind." >&2;}
-   { (exit 1); exit 1; }; }
+   { (echo "ignoring missing GLU library."); true; }; }
 fi
 
 fi
