diff -aur ingen/src/client/wscript ingen-build//src/client/wscript
--- ingen/src/client/wscript	2011-06-14 20:26:12.000000000 +0200
+++ ingen-build//src/client/wscript	2011-06-14 20:53:07.000000000 +0200
@@ -33,4 +33,4 @@
     obj.target          = 'ingen_client'
     obj.install_path    = '${LIBDIR}'
     obj.use             = 'libingen_shared'
-    autowaf.use_lib(bld, obj, 'GLIBMM LV2CORE LILV SUIL RAUL SORD SOUP SIGCPP LIBLO SOUP')
+    autowaf.use_lib(bld, obj, 'GLIBMM LV2CORE LILV SUIL RAUL SORD SOUP SIGCPP LIBLO SOUP RASQAL')
diff -aur ingen/wscript ingen-build//wscript
--- ingen/wscript	2011-06-14 20:26:14.000000000 +0200
+++ ingen-build//wscript	2011-06-14 20:52:23.000000000 +0200
@@ -56,6 +56,8 @@
                       atleast_version='0.8.0', mandatory=True)
     autowaf.check_pkg(conf, 'sord-0', uselib_store='SORD',
                       atleast_version='0.3.0', mandatory=False)
+    autowaf.check_pkg(conf, 'rasqal', uselib_store='RASQAL',
+                      atleast_version='0.9.0', mandatory=True)
     if not Options.options.no_gui:
         autowaf.check_pkg(conf, 'gtkmm-2.4', uselib_store='GTKMM',
                           atleast_version='2.12.0', mandatory=False)
