--- src/zm_signal.cpp.orig	2012-08-30 13:24:16.479872021 -0700
+++ src/zm_signal.cpp	2012-08-30 13:27:35.909871996 -0700
@@ -82,13 +82,13 @@
 	int trace_size = 0;
 
 #if HAVE_STRUCT_SIGCONTEXT_EIP
-	Error( "Signal address is %p, from %p", (void *)context.cr2, (void *)context.eip );
+	Error( "Signal address is %p, from %p", (void *)context.fault_address, (void *)context.eip );
 
 	trace_size = backtrace( trace, TRACE_SIZE );
 	// overwrite sigaction with caller's address
 	trace[1] = (void *)context.eip;
 #elif HAVE_STRUCT_SIGCONTEXT
-	Error( "Signal address is %p, no eip", (void *)context.cr2 );
+	Error( "Signal address is %p, no eip", (void *)context.fault_address );
 
 	trace_size = backtrace( trace, TRACE_SIZE );
 #else // HAVE_STRUCT_SIGCONTEXT
