--- zm/scripts/zmupdate.pl.in	2009-06-12 15:45:00.000000000 -0700
+++ zm-build/scripts/zmupdate.pl.in	2009-06-22 17:55:54.000000000 -0700
@@ -403,7 +403,7 @@
                 $command .= " -p".$db_pass;
             }
         }
-        $command .= " ".ZM_DB_NAME." < ".ZM_PATH_BUILD."/db/zm_update-".$version.".sql";
+        $command .= " ".ZM_DB_NAME." < ".ZM_PATH_UPDATE."/zm_update-".$version.".sql";
 
         print( "Executing '$command'\n" ) if ( DBG_LEVEL > 0 );
         my $output = qx($command);

