--- ZoneMinder-1.25.0/src/zm_stream.cpp	2009-05-08 09:47:27.000000000 -0400
+++ ZoneMinder-1.25.0.ajs/src/zm_stream.cpp	2013-01-14 16:52:18.000000000 -0500
@@ -262,7 +262,7 @@ bool StreamBase::sendTextFrame( const ch
         fprintf( stdout, "--ZoneMinderFrame\r\n" );
         fprintf( stdout, "Content-Length: %d\r\n", n_bytes );
         fprintf( stdout, "Content-Type: image/jpeg\r\n\r\n" );
-        if ( fwrite( buffer, n_bytes, 1, stdout ) )
+        if ( fwrite( buffer, n_bytes, 1, stdout ) != 1 )
         {
             Error( "Unable to send stream text frame: %s", strerror(errno) );
             return( false );

