From fb9612ed33729460d209d4080cf5750271a02092 Mon Sep 17 00:00:00 2001
From: Dave Reisner <dreisner@archlinux.org>
Date: Fri, 8 Jul 2011 21:10:14 -0400
Subject: [PATCH] use given instead of deprecated switch

---
 wlcsv2c.pl |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/wlcsv2c.pl b/wlcsv2c.pl
index 8575371..704d6e9 100644
--- a/wlcsv2c.pl
+++ b/wlcsv2c.pl
@@ -2,21 +2,21 @@
 # Rodolfo Garcia kix_at_kix.es
 # suspend database management v1.0
 
-use Switch;
+use feature qw(switch);
 
-switch ($#ARGV) {
-  case 0 {
+given ($#ARGV) {
+  when (0) {
     my $fin = $ARGV[0];
     open(INF, "<$fin") or die "Could not open input file\n";
     open(OUF, ">&STDOUT") or die "Could not open output file\n";
   }
-  case 1 {
+  when (1) {
     my $fin = $ARGV[0];
     my $fout = $ARGV[1];
     open(INF, "<$fin") or die "Could not open input file\n";
     open(OUF, ">$fout") or die "Could not open output file\n";
   }
-  else {
+  default {
     my $msg = $0 . " <input-file> [output-file]";
     die $msg;
   }
-- 
1.7.6


