diff --git a/system-monitor@paradoxxx.zero.gmail.com/extension.js b/system-monitor@paradoxxx.zero.gmail.com/extension.js
index 010481a..b1681eb 100644
--- a/system-monitor@paradoxxx.zero.gmail.com/extension.js
+++ b/system-monitor@paradoxxx.zero.gmail.com/extension.js
@@ -400,7 +400,7 @@ const smMountsMonitor = new Lang.Class({
             this.manager = this._volumeMonitor;
             this.mount_added_id = this.manager.connect('mount-added', Lang.bind(this, this.refresh));
             this.mount_removed_id = this.manager.connect('mount-removed', Lang.bind(this, this.refresh));
-            //need to add the other signals here 
+            //need to add the other signals here
             this.connected = true;
         }
         catch (e) {
@@ -465,7 +465,7 @@ const Bar = new Lang.Class({
         this.actor.set_height(this.mounts.length * (3 * this.thickness) / 2 );
         let [width, height] = this.actor.get_surface_size();
         let cr = this.actor.get_context();
- 
+
         let x0 = width/8;
         let y0 = this.thickness/2;
         cr.setLineWidth(this.thickness);
@@ -857,51 +857,6 @@ const Battery = new Lang.Class({
         // callback function for when battery stats updated.
         let battery_found = false;
         let isBattery = false;
-        this._proxy.GetDevicesRemote(Lang.bind(this, function(devices, error) {
-            if (error) {
-
-                log("SM: Power proxy error: " + error)
-                this.actor.hide();
-                this.menu_item.actor.hide();
-                return;
-            }
-
-            let [result] = devices;
-            for (let i = 0; i < result.length; i++) {
-                let [device_id, device_type, icon, percentage, state, seconds] = result[i];
-
-                if (Compat.versionCompare(shell_Version, "3.9"))
-                   isBattery = (device_type == Power.UPower.DeviceKind.BATTERY);
-                else
-                   isBattery = (device_type == Power.UPDeviceType.BATTERY);
-
-                if (isBattery && !battery_found) {
-                    battery_found = true;
-                    //grab data
-                    if (seconds > 60){
-                        let time = Math.round(seconds / 60);
-                        let minutes = time % 60;
-                        let hours = Math.floor(time / 60);
-                        this.timeString = C_("battery time remaining","%d:%02d").format(hours,minutes);
-                    } else {
-                        this.timeString = '-- ';
-                    }
-                    this.percentage = Math.ceil(percentage);
-                    this.gicon = Gio.icon_new_for_string(icon);
-
-                    if (Schema.get_boolean(this.elt + '-display'))
-                        this.actor.show()
-                    if (Schema.get_boolean(this.elt + '-show-menu'))
-                        this.menu_item.actor.show();
-                }
-            }
-            if (!battery_found) {
-                log("SM: No battery found")
-                this.actor.hide();
-
-                this.menu_item.actor.hide();
-            }
-        }));
     },
     hide_system_icon: function(override) {
         let value = Schema.get_boolean(this.elt + '-hidesystem');
@@ -999,7 +954,7 @@ function createCpus()
             numcores = 1;
         }
     }
-    
+
     // there are several cores to display,
     // instantiate each cpu
     if (numcores > 1) {
@@ -1178,8 +1133,8 @@ const Cpu = new Lang.Class({
 
 const Disk = new Lang.Class({
     Name: 'SystemMonitor.Disk',
-    Extends: ElementBase, 
-    
+    Extends: ElementBase,
+
     elt: 'disk',
     color_name: ['read', 'write'],
     _init: function() {
@@ -1430,7 +1385,7 @@ const Net = new Lang.Class({
         while (str.length < length) {
             str = '0' + str;
         }
-        return str;   
+        return str;
     },
 
     _apply: function() {
@@ -1723,10 +1678,10 @@ var enable = function () {
         Main.__sm.elts.push(new Thermal());
         Main.__sm.elts.push(new Fan());
         Main.__sm.elts.push(new Battery());
-        
+
         let tray = Main.__sm.tray;
-        
-        
+
+
         if (Schema.get_boolean("move-clock")) {
             let dateMenu;
             if (Compat.versionCompare(shell_Version, "3.5.90")){
@@ -1752,7 +1707,7 @@ var enable = function () {
         } else {
             Main.panel._addToPanelBox('system-monitor', tray, 1, panel);
         }
-        
+
         let box = new St.BoxLayout();
         tray.actor.add_actor(box);
         box.add_actor(Main.__sm.icon.actor);
@@ -1770,7 +1725,7 @@ var enable = function () {
         let bar_item = Main.__sm.bar;
         bar_item.create_menu_item();
         tray.menu.addMenuItem(bar_item.menu_item);
-        
+
         change_usage();
         Schema.connect('changed::' + 'disk-usage-style', change_usage);
 
@@ -1853,7 +1808,7 @@ var disable = function () {
     for (let eltName in Main.__sm.elts) {
         Main.__sm.elts[eltName].destroy();
     }
-    
+
     if (!Compat.versionCompare(shell_Version,"3.5")){
         Main.__sm.tray.destroy();
         StatusArea.systemMonitor = null;
diff --git a/system-monitor@paradoxxx.zero.gmail.com/metadata.json b/system-monitor@paradoxxx.zero.gmail.com/metadata.json
index 4fae73a..3e48a9e 100644
--- a/system-monitor@paradoxxx.zero.gmail.com/metadata.json
+++ b/system-monitor@paradoxxx.zero.gmail.com/metadata.json
@@ -1,5 +1,5 @@
 {
-    "shell-version": ["3.4","3.6","3.8","3.10"],
+    "shell-version": ["3.4","3.6","3.8","3.10","3.12"],
     "uuid": "system-monitor@paradoxxx.zero.gmail.com",
     "name": "system-monitor",
     "url": "https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet",
