--- numpy-1.7.0b2/numpy/distutils/exec_command.py.orig	2012-08-13 11:53:16.000000000 +0800
+++ numpy-1.7.0b2/numpy/distutils/exec_command.py	2012-10-23 14:34:01.875934505 +0800
@@ -100,6 +100,9 @@
     if exe.startswith('"'):
         exe = exe[1:-1]
 
+    parts = exe.split(' -')
+    exe = parts[0].strip()
+
     suffixes = ['']
     if os.name in ['nt','dos','os2']:
         fn,ext = os.path.splitext(exe)
@@ -119,6 +122,7 @@
             if not os.path.islink(f_ext):
                 f_ext = realpath(f_ext)
             if os.path.isfile(f_ext) and os.access(f_ext, os.X_OK):
+                f_ext = ' -'.join([f_ext] + parts[1:])
                 log.good('Found executable %s' % f_ext)
                 _cache[key] = f_ext
                 return f_ext
