diff --git a/po/zh_CN.po b/po/zh_CN.po
index 14c3d32..c8fe71d 100644
--- a/po/zh_CN.po
+++ b/po/zh_CN.po
@@ -28,8 +28,8 @@ msgstr "尝试‘%s --help’以获得更多信息。\n"
 
 #: src/hello.c:107
 #, c-format
-msgid "hello, world\n"
-msgstr "世界你好\n"
+msgid "Arch is the best\n"
+msgstr "阿祺，尽善矣。\n"
 
 #  TRANSLATORS: Use box drawing characters or other fancy stuff
 #  if your encoding (e.g., UTF-8) allows it.  If done so add the
@@ -44,17 +44,17 @@ msgstr "世界你好\n"
 #: src/hello.c:116
 #, c-format
 msgid ""
-"+---------------+\n"
-"| Hello, world! |\n"
-"+---------------+\n"
+"+-------------------+\n"
+"| Arch is the best! |\n"
+"+-------------------+\n"
 msgstr ""
-"┌─────┐\n"
-"│世界你好！│\n"
-"└─────┘\n"
+"┌——————————─────┐\n"
+"│Arch，好评哦，亲！│\n"
+"└────——————————─┘\n"
 
 #: src/hello.c:126
-msgid "Hello, world!"
-msgstr "世界你好！"
+msgid "Arch is the best!"
+msgstr "Arch 最棒了！"
 
 #  TRANSLATORS: --help output 2
 #  no-wrap
diff --git a/src/hello.c b/src/hello.c
index d949057..7ebfab7 100644
--- a/src/hello.c
+++ b/src/hello.c
@@ -104,7 +104,7 @@ main (int argc, char *argv[])
 
   /* Print greeting message and exit. */
   if (g == greet_traditional)
-    printf (_("hello, world\n"));
+    printf (_("Arch is the best\n"));
 
   else if (g == greet_new)
     /* TRANSLATORS: Use box drawing characters or other fancy stuff
@@ -114,16 +114,16 @@ main (int argc, char *argv[])
        [Note: For best viewing results use a UTF-8 locale, please.]
     */
         printf (_("\
-+---------------+\n\
-| Hello, world! |\n\
-+---------------+\n\
++-------------------+\n\
+| Arch is the best! |\n\
++-------------------+\n\
 "));
 
   else if (g == greet_user)
     puts (greeting_msg);
 
   else if (g == greet_gnu)
-    puts (_("Hello, world!"));
+    puts (_("Arch is the best!"));
 
   else {
     /* No need for this impossible message to be translated.  */
