diff -aur stalonetray-0.8.1.old/src/settings.c stalonetray-0.8.1.new/src/settings.c
--- stalonetray-0.8.1.old/src/settings.c	2010-10-13 09:51:12.000000000 +0200
+++ stalonetray-0.8.1.new/src/settings.c	2013-05-01 21:41:53.112431373 +0200
@@ -76,6 +76,7 @@
 	settings.scrollbars_inc		= -1;
 	settings.wm_strut_mode		= WM_STRUT_AUTO;
 	settings.kludge_flags		= 0;
+	settings.lower_on_start     = 0;
 	settings.remote_click_name  = NULL;
 	settings.remote_click_btn   = REMOTE_CLICK_BTN_DEFAULT;
 	settings.remote_click_cnt   = REMOTE_CLICK_CNT_DEFAULT;
@@ -431,6 +432,7 @@
 	{ "-i", "--icon-size", "icon_size", {&settings.icon_size}, (param_parser_t) &parse_int, 1, 1, 0, NULL}, 
 	{"-h", "--help", NULL, {&settings.need_help}, (param_parser_t) &parse_bool, 0, 0, 0, "true" },
 	{NULL, "--kludges", "kludges", {&settings.kludge_flags}, (param_parser_t) &parse_kludges, 1, 1, 0, NULL},
+	{NULL, "--lower-on-start", "lower_on_start", {&settings.lower_on_start}, (param_parser_t) &parse_bool, 1, 1, 1, "true"},
 	{NULL, "--max-geometry", "max_geometry", {&settings.max_geometry_str}, (param_parser_t) &parse_copystr, 1, 1, 0, NULL},
 	{NULL, "--no-shrink", "no_shrink", {&settings.shrink_back_mode}, (param_parser_t) &parse_bool_rev, 1, 1, 1, "true"},
 	{"-p", "--parent-bg", "parent_bg", {&settings.parent_bg}, (param_parser_t) &parse_bool, 1, 1, 1, "true"},
@@ -505,6 +507,7 @@
 			"                                 - fix_window_pos (fix window position),\n"
 			"                                 - force_icons_size (ignore icon resizes),\n"
 			"                                 - use_icons_hints (use icon size hints)\n"
+			"    --lower-on-start            lower tray window on startup\n"
 			"    --max-geometry <geometry>   set tray maximal width and height; 0 indicates\n"
 			"                                no limit in respective direction\n"
 			"    --no-shrink                 do not shrink window back after icon removal\n"
diff -aur stalonetray-0.8.1.old/src/settings.h stalonetray-0.8.1.new/src/settings.h
--- stalonetray-0.8.1.old/src/settings.h	2009-10-17 12:15:28.000000000 +0200
+++ stalonetray-0.8.1.new/src/settings.h	2013-05-01 21:41:11.222379347 +0200
@@ -33,6 +33,7 @@
     int shrink_back_mode;      /* Keep tray's window size minimal */
 	int dockapp_mode;          /* Activate dockapp mode */
 	int kludge_flags;          /* What kludges to activate */
+	int lower_on_start;        /* Lower tray on startup */
     
     int need_help;             /* Print usage and exit */
 
diff -aur stalonetray-0.8.1.old/src/tray.c stalonetray-0.8.1.new/src/tray.c
--- stalonetray-0.8.1.old/src/tray.c	2013-05-01 21:18:46.578159000 +0200
+++ stalonetray-0.8.1.new/src/tray.c	2013-05-01 22:51:12.668937674 +0200
@@ -691,5 +691,7 @@
 	/* TODO: check if this is really necessary */
 	tray_set_wm_hints();
 	tray_update_window_props();
+	if (settings.lower_on_start)
+		XLowerWindow(tray_data.dpy, tray_data.tray);
 }
 
diff -aur stalonetray-0.8.1.old/stalonetray.1 stalonetray-0.8.1.new/stalonetray.1
--- stalonetray-0.8.1.old/stalonetray.1	2010-11-06 15:21:36.000000000 +0100
+++ stalonetray-0.8.1.new/stalonetray.1	2013-05-01 22:55:34.412391216 +0200
@@ -218,6 +218,12 @@
 .RE
 .RE
 .PP
+\fB\-\-lower\-on\-start\fR, \fBlower_on_start\fR [\fIbool\fR]
+.RS 4
+Lower tray window on startup\&. Default value:
+\fBfalse\fR\&.
+.RE
+.PP
 \fB\-\-max\-geometry\fR \fIgeometry_spec\fR, \fBmax_geometry\fR \fIgeometry_spec\fR
 .RS 4
 Set tray`s maximal geometry to
