--- src/pyvlfeat-0.1.1a3/setup.py	2011-05-04 19:48:14.000000000 +0400
+++ setup.py	2015-05-02 21:31:09.410537983 +0300
@@ -47,7 +47,7 @@
 Libraries = None
 BuildExtension = build_ext
 CompileArgs = ['-msse2', '-O2', '-fPIC', '-w']
-LinkArgs = ['-msse', '-shared', '-lboost_python-mt-py26']
+LinkArgs = ['-msse', '-shared', '-lboost_python']
 
 def mkExtension(name):
     modname = '_' + name.lower()
