--- 40-nw-fermi.rules	2011-07-22 00:57:40.000000000 +0200
+++ 40-nw-fermi.rules.new	2011-11-29 12:15:48.519610005 +0100
@@ -4,8 +4,8 @@
 # KERNEL!="event[0-9]*", GOTO="nw-fermi-end"
 
 KERNEL=="nwfermi?", SYMLINK+="nw-fermi", RUN+="/usr/sbin/nwfermi_daemon /daemon /instanceId $number"
-KERNEL=="hidraw*", SUBSYSTEM=="hidraw", SYSFS{idVendor}=="1926", MODE="0666", SYMLINK+="nw-fermi-device"
-KERNEL=="hidraw*", SUBSYSTEM=="hidraw", SYSFS{idVendor}=="0b8c", SYSFS{idProduct}=="0069", MODE="0666", SYMLINK+="nw-fermi-device"
+KERNEL=="hidraw*", SUBSYSTEM=="hidraw", ATTRS{idVendor}=="1926", MODE="0666", SYMLINK+="nw-fermi-device"
+KERNEL=="hidraw*", SUBSYSTEM=="hidraw", ATTRS{idVendor}=="0b8c", ATTRS{idProduct}=="0069", MODE="0666", SYMLINK+="nw-fermi-device"
 ACTION=="add|change", SUBSYSTEM=="input", ATTRS{name}=="Nextwindow Fermi Touchscreen", ENV{NAME}="nextwindow", ENV{ID_INPUT.tags}="nextwindow"
 
-LABEL="nw-fermi-end"
+LABEL="nw-fermi-end" 
