--- quassel/src/qtui/indicatornotificationbackend.cpp	2010-04-16 16:44:13.000000000 +0100
+++ quassel/src/qtui/indicatornotificationbackend.cpp	2010-08-02 10:31:34.415696001 +0100
@@ -42,7 +42,7 @@
   : AbstractNotificationBackend(parent)
 {
   NotificationSettings notificationSettings;
-  _enabled = notificationSettings.value("Indicator/Enabled", false).toBool();
+  _enabled = notificationSettings.value("Indicator/Enabled", true).toBool();
 
   notificationSettings.notify("Indicator/Enabled", this, SLOT(enabledChanged(const QVariant &)));
 
@@ -181,7 +181,7 @@
 
 void IndicatorNotificationBackend::ConfigWidget::load() {
   NotificationSettings s;
-  enabled = s.value("Indicator/Enabled", false).toBool();
+  enabled = s.value("Indicator/Enabled", true).toBool();
 
   ui.enabled->setChecked(enabled);
   setChangedState(false);
--- quassel/src/qtui/settingspages/appearancesettingspage.ui	2010-04-16 16:44:13.000000000 +0100
+++ quassel/src/qtui/settingspages/appearancesettingspage.ui	2010-08-02 10:38:34.019696000 +0100
@@ -150,13 +150,13 @@
       <string>Show system tray icon</string>
      </property>
      <property name="checked">
-      <bool>true</bool>
+      <bool>false</bool>
      </property>
      <property name="settingsKey" stdset="0">
       <string notr="true">UseSystemTrayIcon</string>
      </property>
      <property name="defaultValue" stdset="0">
-      <bool>true</bool>
+      <bool>false</bool>
      </property>
     </widget>
    </item>
--- quassel/src/qtui/qtui.cpp	2010-04-16 16:44:13.000000000 +0100
+++ quassel/src/qtui/qtui.cpp	2010-08-02 10:38:36.035696000 +0100
@@ -75,7 +75,7 @@
 void QtUi::init() {
   _mainWin->init();
   QtUiSettings uiSettings;
-  uiSettings.initAndNotify("UseSystemTrayIcon", this, SLOT(useSystemTrayChanged(QVariant)), true);
+  uiSettings.initAndNotify("UseSystemTrayIcon", this, SLOT(useSystemTrayChanged(QVariant)), false);
 
   GraphicalUi::init(); // needs to be called after the mainWin is initialized
 }
