--- a/src/wl/sys/wl_linux.c	2012-01-05 17:54:16.360237702 +0100
+++ b/src/wl/sys/wl_linux.c	2012-01-05 18:00:42.054353682 +0100
@@ -386,7 +386,11 @@ static const struct net_device_ops wl_ne
 #endif
 	.ndo_get_stats = wl_get_stats,
 	.ndo_set_mac_address = wl_set_mac_address,
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(3, 2, 0))
 	.ndo_set_multicast_list = wl_set_multicast_list,
+#else
+	.ndo_set_rx_mode = wl_set_multicast_list,
+#endif
 	.ndo_do_ioctl = wl_ioctl
 };
 
