=== modified file 'gnome-activity-journal/src/histogram.py'
--- gnome-activity-journal/src/histogram.py	2011-03-16 16:36:25 +0000
+++ gnome-activity-journal/src/histogram.py	2011-10-03 18:10:10 +0000
@@ -157,7 +157,7 @@
         """
         The major drawing method that the expose event calls directly
         """
-        widget.style.set_background(widget.window, gtk.STATE_NORMAL)
+        #widget.style.set_background(widget.window, gtk.STATE_NORMAL)
         context = widget.window.cairo_create()
         self.expose(widget, event, context)
 

