--- xlander.h.old	2012-02-05 21:39:37.991663067 +0600
+++ xlander.h	2012-02-05 21:39:43.978329770 +0600
@@ -110,7 +110,7 @@
 DATABASE *DBInit (), *DBInitFromData (), *LoadDataBase ();
 void DBInsert (), DBFinish (), DBPlot (), SwapBuffers (), exit ();
 #ifndef _AIX
-char *malloc ();
+//char *malloc ();
 #endif
 double atof ();
 #ifdef sun
--- xlander.c.old	2012-02-05 21:39:17.218329625 +0600
+++ xlander.c	2012-02-05 21:40:20.078329970 +0600
@@ -155,5 +155,6 @@
 #else
       mask = sigblock (sigmask (SIGINT));
 #endif
+      usleep(50000);
    }
 }
