--- ./mercurial_keyring.py
+++ ./mercurial_keyring.py
@@ -370,7 +370,7 @@ def try_smtp_login(ui, smtp_obj, usernam
                  (username))
         smtp_obj.login(username, password)
         return True
-    except smtplib.SMTPException, inst:
+    except smtplib.SMTPException as inst:
         if inst.smtp_code == 535:
             ui.status(_("SMTP login failed: %s\n\n") % inst.smtp_error)
             return False
@@ -421,10 +421,10 @@ def keyring_supported_smtp(ui, username)
     def send(sender, recipients, msg):
         try:
             return s.sendmail(sender, recipients, msg)
-        except smtplib.SMTPRecipientsRefused, inst:
+        except smtplib.SMTPRecipientsRefused as inst:
             recipients = [r[1] for r in inst.recipients.values()]
             raise util.Abort('\n' + '\n'.join(recipients))
-        except smtplib.SMTPException, inst:
+        except smtplib.SMTPException as inst:
             raise util.Abort(inst)
 
     return send
