diff --git a/CMakeLists.txt b/CMakeLists.txt
index aba9032..4de64c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,8 +11,10 @@ set(QTKEYCHAIN_SOVERSION 0)
 set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" "${PROJECT_SOURCE_DIR}/cmake/Modules")
 include(GNUInstallDirs)
 
-# try Qt5 first, and prefer that if found
-find_package(Qt5Core QUIET)
+# try Qt5 first, and prefer that if found, unless BUILD_WITH_QT4 is set
+if (NOT BUILD_WITH_QT4)
+  find_package(Qt5Core QUIET)
+endif()
 if (Qt5Core_FOUND)
   if(UNIX AND NOT APPLE)
     find_package(Qt5DBus REQUIRED)
