diff -aur berrybots.pristine/guizipper.cpp berrybots.new/guizipper.cpp
--- berrybots.pristine/guizipper.cpp	2013-10-07 17:18:54.000000000 +1300
+++ berrybots.new/guizipper.cpp	2013-10-19 15:15:55.439331548 +1300
@@ -21,8 +21,8 @@
 #include <fcntl.h>
 #include <sstream>
 #include <sys/stat.h>
-#include <libarchive/archive.h>
-#include <libarchive/archive_entry.h>
+#include <archive.h>
+#include <archive_entry.h>
 #include <platformstl/filesystem/filesystem_traits.hpp>
 #include "filemanager.h"
 #include "guizipper.h"
Only in berrybots.new: guizipper.cpp~
diff -aur berrybots.pristine/guizipper.h berrybots.new/guizipper.h
--- berrybots.pristine/guizipper.h	2013-10-07 17:18:54.000000000 +1300
+++ berrybots.new/guizipper.h	2013-10-19 15:12:38.938610586 +1300
@@ -21,7 +21,7 @@
 #ifndef GUI_ZIPPER_H
 #define GUI_ZIPPER_H
 
-#include <libarchive/archive.h>
+#include <archive.h>
 #include "filemanager.h"
 #include "zipper.h"
 
Only in berrybots.new: guizipper.h~
diff -aur berrybots.pristine/Makefile berrybots.new/Makefile
--- berrybots.pristine/Makefile	2013-10-07 18:13:29.000000000 +1300
+++ berrybots.new/Makefile	2013-10-19 16:44:03.355133534 +1300
@@ -94,20 +94,17 @@
 # Sources and flags for building on Linux / GTK
 LINUX_EXTRA_SOURCES =  relativebasedir.cpp relativerespath.cpp
 LINUX_EXTRA_SOURCES += ./luajit/src/libluajit.a
-LINUX_EXTRA_SOURCES += ${LIBARCHIVE_PATH}/.libs/libarchive.a
+LINUX_EXTRA_SOURCES += /usr/lib/libarchive.a
 
-LINUX_CFLAGS =  -I./luajit/src -I./stlsoft-1.9.116/include -I${LIBARCHIVE_PATH}
-LINUX_CFLAGS += -I${SFML_PATH}/include
-LINUX_CFLAGS += `${WXWIDGETS_PATH}/wx-config --cflags`
+LINUX_CFLAGS =  -I./luajit/src -I./stlsoft-1.9.116/include
+LINUX_CFLAGS += `/usr/bin/wx-config-2.9 --cflags`
 
-LINUX_LDFLAGS =  -L${SFML_BUILD_PATH}/lib `${WXWIDGETS_PATH}/wx-config --libs`
-LINUX_LDFLAGS += -lsfml-graphics -lsfml-window -lsfml-system -lz -ldl
+LINUX_LDFLAGS =  `/usr/bin/wx-config-2.9 --libs`
+LINUX_LDFLAGS += -lacl -lsfml-graphics -lsfml-window -lsfml-system -lz -ldl
 
 LINUXCLI_CFLAGS =  -I./luajit/src -I./stlsoft-1.9.116/include
-LINUXCLI_CFLAGS += -I${LIBARCHIVE_PATH} -I${SFML_PATH}/include
 
-LINUXCLI_LDFLAGS =  -L${SFML_BUILD_PATH}/lib
-LINUXCLI_LDFLAGS += -lsfml-graphics -lsfml-window -lsfml-system -lz -ldl
+LINUXCLI_LDFLAGS = -lacl -lsfml-graphics -lsfml-window -lsfml-system -lz -ldl
 ##############################################################################
 
 
@@ -181,17 +178,11 @@
 linux:
 	$(MAKE_LUAJIT)
 	$(CC) ${SOURCES} ${LINUX_EXTRA_SOURCES} ${LINUX_CFLAGS} ${LINUX_LDFLAGS} -o bbgui
-	cp -r ${SFML_BUILD_PATH}/lib ./sfml-lib
-	cp ./scripts/bb_gui_linux.sh ./berrybots.sh
-	chmod 755 ./berrybots.sh
 	@echo "==== Successfully built BerryBots $(VERSION) ===="
 
 linuxcli:
 	$(MAKE_LUAJIT)
 	$(CC) ${CLI_SOURCES} ${LINUX_EXTRA_SOURCES} ${LINUXCLI_CFLAGS} ${LINUXCLI_LDFLAGS} -o bbgui
-	cp -r ${SFML_BUILD_PATH}/lib ./sfml-lib
-	cp ./scripts/bb_gui_linux.sh ./berrybots.sh
-	chmod 755 ./berrybots.sh
 	@echo "==== Successfully built BerryBots $(VERSION) ===="
 
 clean:
Only in berrybots.new: Makefile~
