diff -aur berrybots.pristine/Makefile.in berrybots.new/Makefile.in
--- berrybots.pristine/Makefile.in	2013-11-11 12:47:10.000000000 +1300
+++ berrybots.new/Makefile.in	2013-11-14 13:32:28.701041184 +1300
@@ -223,7 +223,7 @@
 osx:
 	$(MAKE_LUAJIT)
 ifeq ($(SFML_HEADER_PATH), )
-	$(CC) ${SOURCES} ${OSX_EXTRA_SOURCES} ${OSX_CFLAGS} ${OSX_LDFLAGS} -o berrybots
+	$(CC) ${SOURCES} ${OSX_EXTRA_SOURCES} ${OSX_CFLAGS} ${OSX_LDFLAGS} -o bbgui
 else
 	./scripts/osx_copy_fix_dylib_paths.sh "${SFML_LIB_PATH}" "sfml-lib"
 	$(CC) ${SOURCES} ${OSX_EXTRA_SOURCES} ${OSX_CFLAGS} ${OSX_LDFLAGS} -o bbgui
@@ -249,7 +249,7 @@
 linux:
 	$(MAKE_LUAJIT)
 ifeq ($(SFML_HEADER_PATH), )
-	$(CC) ${SOURCES} ${LINUX_EXTRA_SOURCES} ${LINUX_CFLAGS} ${LINUX_LDFLAGS} -o berrybots
+	$(CC) ${SOURCES} ${LINUX_EXTRA_SOURCES} ${LINUX_CFLAGS} ${LINUX_LDFLAGS} -o bbgui
 else
   ifeq ($(LOCAL_SFML), 1)
 	$(MAKE_SFML)
@@ -264,9 +264,9 @@
 	cp scripts/bb_gui_linux.sh berrybots
 	rm -rf sfml-lib
 	cp -r ${SFML_LIB_PATH} sfml-lib
-	chmod 755 bbgui
-endif
 	chmod 755 berrybots
+endif
+	chmod 755 bbgui
 	@echo "==== Successfully built BerryBots $(VERSION) ===="
 	@echo "==== Launch BerryBots with: ./berrybots"
 
@@ -298,9 +298,12 @@
 	cp -r stages $(DESTDIR)$(datadir)/berrybots
 	cp -r runners $(DESTDIR)$(datadir)/berrybots
 	cp -r apidoc $(DESTDIR)$(datadir)/berrybots
+ifneq ($(SFML_HEADER_PATH), )
 	mkdir -p $(DESTDIR)$(libdir)
 	rm -rf $(DESTDIR)$(libdir)/berrybots
 	cp -r sfml-lib $(DESTDIR)$(libdir)/berrybots
+endif
+
 ifeq ($(.DEFAULT_GOAL), osx)
 	./scripts/osx_fix_dylib_paths.sh $(DESTDIR)$(bindir)/berrybots $(DESTDIR)$(libdir)/berrybots
 endif
