Only in openexr_viewers-1.0.2/config: OpenEXR_ViewersConfig.h
diff -rup openexr_viewers-1.0.2/exrdisplay/applyCtl.cpp openexr_viewers-1.0.2.new/exrdisplay/applyCtl.cpp
--- openexr_viewers-1.0.2/exrdisplay/applyCtl.cpp	2007-07-26 04:34:32.000000000 +0200
+++ openexr_viewers-1.0.2.new/exrdisplay/applyCtl.cpp	2012-07-03 16:46:30.212022027 +0200
@@ -39,6 +39,8 @@
 //-----------------------------------------------------------------------------
 
 #include <applyCtl.h>
+#include <cstdlib>
+#include <cstdio>
 
 #if HAVE_CTL_INTERPRETER
 
diff -rup openexr_viewers-1.0.2/playexr/ctlToLut.cpp openexr_viewers-1.0.2.new/playexr/ctlToLut.cpp
--- openexr_viewers-1.0.2/playexr/ctlToLut.cpp	2007-07-26 04:34:32.000000000 +0200
+++ openexr_viewers-1.0.2.new/playexr/ctlToLut.cpp	2012-07-03 16:46:30.212022027 +0200
@@ -39,6 +39,8 @@
 //-----------------------------------------------------------------------------
 
 #include <ctlToLut.h>
+#include <cstdlib>
+#include <cstdio>
 
 #if HAVE_CTL_INTERPRETER
 
diff -rup openexr_viewers-1.0.2/playexr/main.cpp openexr_viewers-1.0.2.new/playexr/main.cpp
--- openexr_viewers-1.0.2/playexr/main.cpp	2007-07-26 02:37:05.000000000 +0200
+++ openexr_viewers-1.0.2.new/playexr/main.cpp	2012-07-03 16:49:54.085723541 +0200
@@ -48,6 +48,8 @@
 #include <vector>
 #include <string>
 #include <stdlib.h>
+#include <cstring>
+#include <unistd.h>
 
 using namespace std;
 
