diff -Naur sizzle-0.0.31.orig/libsizzle/posix.c sizzle-0.0.31.new/libsizzle/posix.c
--- sizzle-0.0.31.orig/libsizzle/posix.c	2012-04-09 01:34:53.000000000 +0200
+++ sizzle-0.0.31.new/libsizzle/posix.c	2012-04-09 01:35:49.000000000 +0200
@@ -2975,7 +2975,6 @@
   static char * s_func_name = FUNC_NAME;
 
   CHECK_TYPE (1, dir_stream_p (stream), _("directory stream"));
-  dir_stream_open_p (stream) = 0;
   closedir (dir_stream_dir (stream));
   *result = zzz_unspecified;
   return RESULT_SUCCESS;
