diff --git a/src/include/plibc.h b/src/include/plibc.h
index 7525fb4..a6794c8 100644
--- a/src/include/plibc.h
+++ b/src/include/plibc.h
@@ -502,7 +502,7 @@ size_t strnlen (const char *str, size_t maxlen);
 #endif
 char *stpcpy(char *dest, const char *src);
 char *strcasestr(const char *haystack_start, const char *needle_start);
-#ifndef __MINGW64__
+#if !(defined __MINGW64__ || defined __MINGW32__)
 #define strcasecmp(a, b) stricmp(a, b)
 #define strncasecmp(a, b, c) strnicmp(a, b, c)
 #endif
