--- usr/share/java/tda/bin/tda.sh.bak	2010-02-06 18:01:44.000000000 -0600
+++ usr/share/java/tda/bin/tda.sh	2011-06-30 16:23:37.462621525 -0500
@@ -2,4 +2,6 @@
 # Unix shell script for starting tda. If you have big log files
 # you might need to adjzst Xmx setting.
 
-cd ..; java -Xmx256m -jar ./tda.jar 
+TDA_HOME=/usr/share/java/tda
+
+nohup $JAVA_HOME/bin/java -Xmx256m -jar $TDA_HOME/tda.jar > /dev/null 2>&1 &
