From bd07c954ef5b9c08491bcad6089da29d824240ab Mon Sep 17 00:00:00 2001
From: Robin McCorkell <rmccorkell@karoshi.org.uk>
Date: Sat, 31 May 2014 17:33:35 +0100
Subject: [PATCH] Allow symlinking paths.php

---
 paths.php        | 3 ++-
 public/index.php | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/paths.php b/paths.php
index 37c5b1e..522b7e5 100644
--- a/paths.php
+++ b/paths.php
@@ -94,7 +94,8 @@ $paths['public'] = 'public';
 // --------------------------------------------------------------
 // Change to the current working directory.
 // --------------------------------------------------------------
-chdir(__DIR__);
+// Do chdir in public/index.php
+//chdir(__DIR__);
 
 // --------------------------------------------------------------
 // Define the directory separator for the environment.
diff --git a/public/index.php b/public/index.php
index 5da356f..4a468bb 100644
--- a/public/index.php
+++ b/public/index.php
@@ -21,6 +21,7 @@ $web = true;
 // --------------------------------------------------------------
 // Set the core Laravel path constants.
 // --------------------------------------------------------------
-require '../paths.php';
+chdir(__DIR__."/..");
+require 'paths.php';
 
 // --------------------------------------------------------------
-- 
1.9.3
