--- acm-5.0-ico-20100109.orig/src/manifest.h	2009-01-28 17:15:06.000000000 +0500
+++ acm-5.0-ico-20100109/src/manifest.h	2012-05-28 01:46:40.456338076 +0600
@@ -157,7 +157,7 @@
  */
 
 #ifndef ACM_LIBRARY
-#define ACM_LIBRARY		"/usr/local/games/acm/"
+#define ACM_LIBRARY		"/usr/share/acm-5.0-ico"
 #endif
 
 /*
--- acm-5.0-ico-20100109.orig/acm.tk	2008-08-10 00:02:51.000000000 +0600
+++ acm-5.0-ico-20100109/acm.tk	2012-05-28 01:57:47.119701636 +0600
@@ -422,8 +422,8 @@
 	set longitude ""
 	set objects ""
 	set plane "C-172"
-	if { [file exists "src/acm"] } {
-		set program "src/acm"
+	if { [file exists "/usr/bin/acm-5.0-ico"] } {
+		set program "/usr/bin/acm-5.0-ico -objects /usr/share/acm-5.0-ico"
 	} else {
 		set program "?"
 	}
--- acm-5.0-ico-20100109.orig/configure	2006-01-05 18:58:54.000000000 +0500
+++ acm-5.0-ico-20100109/configure	2012-05-28 01:44:55.162999969 +0600
@@ -527,11 +527,7 @@
 
 
 
-if test -d /usr/lib/games ; then
-    OBVDIR="/usr/lib/games/acm"
-else
-    OBVDIR=\$\(prefix\)/games/acm
-fi
+OBVDIR="/usr/share/acm-5.0-ico"
 
 
 rand=0
