=== modified file 'SConstruct'
--- SConstruct	2013-01-21 17:23:55 +0000
+++ SConstruct	2013-02-22 16:08:05 +0000
@@ -188,6 +188,7 @@
 # static linking have beed addressed
 #
 env.Append(LINKFLAGS = link_arch)
+env.Append(LINKFLAGS = '-lboost_system')

 #
 # Check required headers and libraries (autoconf functionality)

=== modified file 'garb/SConscript'
--- garb/SConscript	2012-05-15 16:20:10 +0000
+++ garb/SConscript	2013-02-22 15:29:36 +0000
@@ -42,6 +42,7 @@
     # as a last resort try shared library
     env.Append(LIBS=['boost_program_options'])

+env.Append(LIBS=['boost_system'])
 # special environment for garb_config.cpp
 conf_env = env.Clone()
 Import('GALERA_VER', 'GALERA_REV')

=== modified file 'gcs/src/SConscript'
--- gcs/src/SConscript	2012-03-17 00:19:40 +0000
+++ gcs/src/SConscript	2013-02-22 15:33:39 +0000
@@ -41,6 +41,7 @@
 gcs_test_env.Prepend(LIBS = File('#/gcomm/src/libgcomm.a'))
 gcs_test_env.Prepend(LIBS = File('#/gcache/src/libgcache.a'))
 gcs_test_env.Prepend(LIBS = File('#/gcs/src/libgcs.a'))
+gcs_test_env.Append(LIBS=['boost_system'])

 gcs_test_env.Program(target = 'gcs_test',
                      source = 'gcs_test.c',
