--- pylotro/PyLotROLauncher/PyLotROUtils.py	2011-06-02 15:38:53.000000000 +0100
+++ pylotro/PyLotROLauncher/PyLotROUtils.py	2012-08-29 12:57:16.122911445 +0100
@@ -33,6 +33,7 @@
 import glob
 from PyQt4 import QtCore
 import xml.dom.minidom
+from xml.sax.saxutils import escape as escape_xml
 
 # If Python 3.0 is in use use http otherwise httplib
 if sys.version_info[:2] < (3, 0):
@@ -464,7 +465,7 @@
 <soap:Body><LoginAccount xmlns=\"http://www.turbine.com/SE/GLS\"><username>%s</username>\
 <password>%s</password><additionalInfo></additionalInfo></LoginAccount></soap:Body></soap:Envelope>"
 
-		SoapMessage = SM_TEMPLATE%(name, password)
+		SoapMessage = SM_TEMPLATE % tuple(map(escape_xml, (name, password)))
 
 		webresp = None
 
