diff --git a/theme.cfg b/theme.cfg
index ddefefa..6aca275 100644
--- a/theme.cfg
+++ b/theme.cfg
@@ -1,5 +1,5 @@
 [default]
 steps = 7
-dir = /usr/share/icons/oxygen/22x22/status
+dir = /usr/share/fdpowermon/icons
 charging = 0:battery-charging-low.png, 10:battery-charging-low.png, 30:battery-charging-caution.png, 50:battery-charging-040.png, 70:battery-charging-060.png, 90:battery-charging-080.png,99:battery-charging.png
 discharging = 2:battery-missing.png:battery-low.png,10:battery-low.png,30:battery-caution.png,50:battery-040.png,70:battery-060.png,90:battery-080.png,100:battery-100.png
