diff -r -U 1 Net-SinFP-2.06.orig//bin/sinfp.pl Net-SinFP-2.06//bin/sinfp.pl
--- Net-SinFP-2.06.orig//bin/sinfp.pl	2010-08-12 00:36:14.170516733 +0530
+++ Net-SinFP-2.06//bin/sinfp.pl	2010-08-12 00:44:19.740367135 +0530
@@ -7,3 +7,2 @@
 use FindBin qw($Bin);
-use lib "$Bin/../lib";
 
@@ -71,3 +70,3 @@
 
-my $dbFile;
+my $dbFile = "/usr/share/sinfp/sinfp.db";
 if ($opts{s}) {
@@ -75,8 +74,3 @@
 }
-else {
-   for ("$Bin/../db/", "$Bin/") {
-      $dbFile = $_.'sinfp.db';
-      last if -f $dbFile;
-   }
-}
+
 print "DEBUG: using db: $dbFile\n" if $opts{v};
