diff --git a/stiler.py b/stiler.py
index 35b7328..1a444f4 100755
--- a/stiler.py
+++ b/stiler.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ############################################################################
 # Copyright (c) 2009   unohu <unohu0@gmail.com> and Contributors           #
diff --git a/xbindkeysrc b/xbindkeysrc
index 0605867..eacad5a 100644
--- a/xbindkeysrc
+++ b/xbindkeysrc
@@ -7,52 +7,52 @@
 # stiler key bindings #
 
 #Tile Left
-"stiler.py left"
+"stiler left"
     m:0x1c + c:83
     Control+Alt+Mod2 + KP_Left 
 
 #Tile Right
-"stiler.py right"
+"stiler right"
     m:0x1c + c:85
     Control+Alt+Mod2 + KP_Right 
 
 #Tile Swap
-"stiler.py swap"
+"stiler swap"
     m:0x1c + c:91
     Control+Alt+Mod2 + KP_Delete 
 
 #Tile Top
-"stiler.py top"
+"stiler top"
     m:0x1c + c:80
     Control+Alt+Mod2 + KP_Up 
 
 #Tile Top_Left
-"stiler.py top_left"
+"stiler top_left"
     m:0x1c + c:79
     Control+Alt+Mod2 + KP_Home 
 
 #Tile Top_Right
-"stiler.py top_right"
+"stiler top_right"
     m:0x1c + c:81
     Control+Alt+Mod2 + KP_Prior 
 
 #Tile Middle
-"stiler.py middle"
+"stiler middle"
     m:0x1c + c:84
     Control+Alt+Mod2 + KP_Begin 
 
 #Tile Bottom_Left
-"stiler.py bottom_left"
+"stiler bottom_left"
     m:0x1c + c:87
     Control+Alt+Mod2 + KP_End 
 
 #Tile Bottom
-"stiler.py bottom"
+"stiler bottom"
     m:0x1c + c:88
     Control+Alt+Mod2 + KP_Down 
 
 #Tile Bottom_Right
-"stiler.py bottom_right"
+"stiler bottom_right"
     m:0x1c + c:89
     Control+Alt+Mod2 + KP_Next 
 
