diff -u tulip/software/crash_handling/BfdWrapper.h src/tulip/software/crash_handling/BfdWrapper.h
--- tulip/software/crash_handling/BfdWrapper.h	2013-06-10 14:09:21.000000000 +0200
+++ src/tulip/software/crash_handling/BfdWrapper.h	2013-06-16 12:54:07.711916113 +0200
@@ -23,11 +23,11 @@
 #ifdef __MINGW32__
 #include <windows.h>
 #include <stdint.h>
-// workaround for latest versions of MinGW binutils
-#define PACKAGE
-#define PACKAGE_VERSION
 #endif
 
+#define PACKAGE	 "Tulip"
+#define PACKAGE_VERSION 4.3.0
+
 #include <cstdlib>
 #include <cstdio>
 #include <cstdarg>
diff -u tulip/software/crash_handling/CMakeLists.txt src/tulip/software/crash_handling/CMakeLists.txt
--- tulip/software/crash_handling/CMakeLists.txt	2013-06-10 14:09:21.000000000 +0200
+++ src/tulip/software/crash_handling/CMakeLists.txt	2013-06-16 12:55:06.352155902 +0200
@@ -7,8 +7,6 @@
 UNSET(HAVE_BFD CACHE)
 IF(LINUX OR MINGW)
 INCLUDE(CheckIncludeFile)
-# workaround for latest versions of MinGW binutils
-SET(CMAKE_REQUIRED_DEFINITIONS "-DPACKAGE -DPACKAGE_VERSION")
 CHECK_INCLUDE_FILE(bfd.h HAVE_BFD)
 
 IF(HAVE_BFD)
@@ -30,7 +28,7 @@
 ADD_LIBRARY(crash_handling STATIC ${LIB_SRCS})
 
 IF(UNIX)
-	TARGET_LINK_LIBRARIES(crash_handling dl)
+	TARGET_LINK_LIBRARIES(crash_handling dl z)
 ENDIF(UNIX)
 
 IF(HAVE_BFD)
