*** tulip-config.in-old	2011-07-12 09:02:35.000000000 +0200
--- tulip-config.in	2011-07-19 22:46:42.914630912 +0200
***************
*** 4,10 ****
  if test "$thisdir" = "." ; then
    thisdir=$PWD
  fi
! prefix=@prefix@
  exec_prefix=@exec_prefix@
  includedir=@includedir@
  libdir=@libdir@
--- 4,10 ----
  if test "$thisdir" = "." ; then
    thisdir=$PWD
  fi
! prefix=/opt/tulip
  exec_prefix=@exec_prefix@
  includedir=@includedir@
  libdir=@libdir@
