--- Otools/generate-launcher	2010-06-25 23:42:35.000000000 -0600
+++ tools/generate-launcher	2010-06-25 23:40:57.000000000 -0600
@@ -30,6 +30,8 @@
 fi
 XDG_DATA_DIRS="\$EXAILE_PREFIX/share/:\$XDG_DATA_DIRS"
 export XDG_DATA_DIRS
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/xulrunner-1.9.2
+export MOZILLA_FIVE_HOME=/usr/lib/xulrunner-1.9.2
 exec python \$EXAILE_PREFIX$LIBDIR/exaile/exaile.py --datadir=\$EXAILE_PREFIX/share/exaile/data --startgui "\$@"
 exit \$?
 
