diff -Nur midori-0.4.5.orig/extensions/wscript_build midori-0.4.5/extensions/wscript_build
--- midori-0.4.5.orig/extensions/wscript_build	2012-04-15 15:51:37.644397846 +0200
+++ midori-0.4.5/extensions/wscript_build	2012-04-15 15:53:26.333836562 +0200
@@ -35,9 +35,9 @@
     obj.target = target
     obj.includes = '..'
     obj.source = source
-    obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML HILDON'
+    obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML HILDON GNOME-KEYRING'
     obj.vapi_dirs = '../midori ../katze'
-    obj.packages = 'glib-2.0 gio-2.0 libsoup-2.4 midori katze'
+    obj.packages = 'glib-2.0 gio-2.0 libsoup-2.4 midori katze gnome-keyring-1'
     if bld.env['HAVE_GTK3']:
         obj.packages += ' gtk+-3.0 webkitgtk-3.0'
     else:
diff -Nur midori-0.4.5.orig/wscript midori-0.4.5/wscript
--- midori-0.4.5.orig/wscript	2012-04-15 15:51:37.654397794 +0200
+++ midori-0.4.5/wscript	2012-04-15 15:54:01.000323855 +0200
@@ -267,6 +267,7 @@
         conf.define ('LIBSOUP_2_37_1', 1)
     check_pkg ('libxml-2.0', '2.6')
     check_pkg ('sqlite3', '3.0', True, var='SQLITE')
+    check_pkg ('gnome-keyring-1', '3.0.0', True, var='GNOME-KEYRING')
 
     if option_enabled ('hildon'):
         if check_pkg ('hildon-1', mandatory=False, var='HILDON'):
