diff -Nur midori-0.4.3.orig/extensions/wscript_build midori-0.4.3/extensions/wscript_build
--- midori-0.4.3.orig/extensions/wscript_build	2011-12-19 15:56:40.047607030 +0100
+++ midori-0.4.3/extensions/wscript_build	2011-12-19 15:57:10.230682456 +0100
@@ -32,9 +32,9 @@
     obj.target = target
     obj.includes = '..'
     obj.source = source
-    obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML HILDON'
+    obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML HILDON GNOME-KEYRING'
     obj.vapi_dirs = '../midori'
-    obj.packages = 'glib-2.0 gio-2.0 libsoup-2.4 midori'
+    obj.packages = 'glib-2.0 gio-2.0 libsoup-2.4 midori gnome-keyring-1'
     if bld.env['HAVE_GTK3']:
         obj.packages += ' gtk+-3.0 webkitgtk-3.0'
     else:
diff -Nur midori-0.4.3.orig/wscript midori-0.4.3/wscript
--- midori-0.4.3.orig/wscript	2011-12-19 15:56:40.057606945 +0100
+++ midori-0.4.3/wscript	2011-12-19 15:57:10.230682456 +0100
@@ -247,6 +247,7 @@
     conf.define ('LIBSOUP_VERSION', conf.check_cfg (modversion='libsoup-2.4'))
     check_pkg ('libxml-2.0', '2.6')
     check_pkg ('sqlite3', '3.0', True, var='SQLITE')
+    check_pkg ('gnome-keyring-1', '3.0.0', True, var='GNOME-KEYRING')
 
     if option_enabled ('hildon'):
         if check_pkg ('hildon-1', mandatory=False, var='HILDON'):
