diff -Nur midori-0.4.8.orig/extensions/wscript_build midori-0.4.8/extensions/wscript_build
--- midori-0.4.8.orig/extensions/wscript_build	2013-02-06 00:28:05.000000000 +0100
+++ midori-0.4.8/extensions/wscript_build	2013-02-06 14:44:22.895322521 +0100
@@ -30,11 +30,11 @@
     obj.target = target
     obj.includes = '.. ../katze ../midori'
     obj.source = source
-    obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML CLUTTER GRANITE'
+    obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML CLUTTER GRANITE GNOME-KEYRING'
     if 'vala' in source:
         obj.env.append_value ('CCFLAGS', '-w')
         obj.vapi_dirs = '../midori ../katze'
-        obj.packages = 'glib-2.0 gio-2.0 libsoup-2.4 webkitgtk-3.0 sqlite3 midori midori-core katze'
+        obj.packages = 'glib-2.0 gio-2.0 libsoup-2.4 webkitgtk-3.0 sqlite3 midori midori-core katze gnome-keyring-1'
         if bld.env['HAVE_GTK3']:
             obj.packages += ' gtk+-3.0'
         else:
diff -Nur midori-0.4.8.orig/wscript midori-0.4.8/wscript
--- midori-0.4.8.orig/wscript	2013-02-06 00:28:05.000000000 +0100
+++ midori-0.4.8/wscript	2013-02-06 14:42:26.591978664 +0100
@@ -305,6 +305,7 @@
     check_pkg ('libxml-2.0', '2.6')
     conf.undefine ('LIBXML_VERSION') # Defined in xmlversion.h
     check_pkg ('sqlite3', '3.6.19', var='SQLITE')
+    check_pkg ('gnome-keyring-1', '3.0.0', True, var='GNOME-KEYRING')
 
     # Store options in env, since 'Options' is not persistent
     if 'CC' in os.environ: conf.env['CC'] = os.environ['CC'].split()
