diff -Nur midori-0.4.1.orig/extensions/wscript_build midori-0.4.1/extensions/wscript_build
--- midori-0.4.1.orig/extensions/wscript_build	2011-10-10 00:41:47.000000000 +0200
+++ midori-0.4.1/extensions/wscript_build	2011-10-17 23:11:38.348715874 +0200
@@ -32,9 +32,9 @@
     obj.target = target
     obj.includes = '..'
     obj.source = source
-    obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML HILDON'
+    obj.uselib = 'UNIQUE LIBSOUP GIO GTK SQLITE WEBKIT LIBXML HILDON GNOME-KEYRING'
     obj.vapi_dirs = '../midori'
-    obj.packages = 'glib-2.0 gio-2.0 libsoup-2.4 midori'
+    obj.packages = 'glib-2.0 gio-2.0 libsoup-2.4 midori gnome-keyring-1'
     if bld.env['HAVE_GTK3']:
         obj.packages += ' gtk+-3.0 webkitgtk-3.0'
     else:
diff -Nur midori-0.4.1.orig/wscript midori-0.4.1/wscript
--- midori-0.4.1.orig/wscript	2011-10-10 00:41:47.000000000 +0200
+++ midori-0.4.1/wscript	2011-10-17 23:08:40.816163596 +0200
@@ -218,6 +218,7 @@
     conf.define ('LIBSOUP_VERSION', conf.check_cfg (modversion='libsoup-2.4'))
     check_pkg ('libxml-2.0', '2.6')
     check_pkg ('sqlite3', '3.0', True, var='SQLITE')
+    check_pkg ('gnome-keyring-1', '3.0.0', True, var='GNOME-KEYRING')
 
     if conf.env['HAVE_LIBSOUP_2_27_90']:
        idn = 'yes'
