diff -Nur ksquirrel-0.8.0.orig//ksquirrel/fmt_filters.cpp ksquirrel-0.8.0/ksquirrel/fmt_filters.cpp
--- ksquirrel-0.8.0.orig//ksquirrel/fmt_filters.cpp	2007-11-29 21:41:29.000000000 +0100
+++ ksquirrel-0.8.0/ksquirrel/fmt_filters.cpp	2008-05-24 12:19:46.000000000 +0200
@@ -74,6 +74,7 @@
 
 #include <cmath>
 #include <algorithm>
+#include <cstring>
 
 namespace fmt_filters
 {
diff -Nur ksquirrel-0.8.0.orig//ksquirrel/imageedit/sq_converter.cpp ksquirrel-0.8.0/ksquirrel/imageedit/sq_converter.cpp
--- ksquirrel-0.8.0.orig//ksquirrel/imageedit/sq_converter.cpp	2007-11-22 00:04:03.000000000 +0100
+++ ksquirrel-0.8.0/ksquirrel/imageedit/sq_converter.cpp	2008-05-24 12:14:33.000000000 +0200
@@ -19,6 +19,8 @@
 #include "config.h"
 #endif
 
+#include <cstdlib>
+
 #include <qapplication.h>
 #include <qfile.h>
 
diff -Nur ksquirrel-0.8.0.orig//ksquirrel/ksquirrelpart/fmt_filters.cpp ksquirrel-0.8.0/ksquirrel/ksquirrelpart/fmt_filters.cpp
--- ksquirrel-0.8.0.orig//ksquirrel/ksquirrelpart/fmt_filters.cpp	2007-11-29 21:41:29.000000000 +0100
+++ ksquirrel-0.8.0/ksquirrel/ksquirrelpart/fmt_filters.cpp	2008-05-24 12:17:54.000000000 +0200
@@ -74,6 +74,7 @@
 
 #include <cmath>
 #include <algorithm>
+#include <cstring>
 
 namespace fmt_filters
 {
diff -Nur ksquirrel-0.8.0.orig//ksquirrel/sq_imageloader.cpp ksquirrel-0.8.0/ksquirrel/sq_imageloader.cpp
--- ksquirrel-0.8.0.orig//ksquirrel/sq_imageloader.cpp	2007-11-29 17:59:17.000000000 +0100
+++ ksquirrel-0.8.0/ksquirrel/sq_imageloader.cpp	2008-05-24 12:22:16.000000000 +0200
@@ -19,6 +19,8 @@
 #include "config.h"
 #endif
 
+#include <cstdlib>
+
 #include <qfile.h>
 
 #include "sq_imageloader.h"
