--- a/tf/CMakeLists.txt	2013-05-23 11:49:54.455479173 -0400
+++ b/tf/CMakeLists.txt	2013-05-23 12:12:17.068830263 -0400
@@ -11,7 +11,7 @@
 include_directories(SYSTEM ${Boost_INCLUDE_DIR}
                            ${catkin_INCLUDE_DIRS}
 )
-include_directories(include)
+include_directories(include ${CATKIN_DEVEL_PREFIX}/include)
 link_directories(${catkin_LIBRARY_DIRS})
 
 add_message_files(DIRECTORY msg FILES tfMessage.msg)
