From c9db74abcfc5cafb34d6430ae536202aaffb0979 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A1s=20Czig=C3=A1ny?= <aczigany@opswat.com>
Date: Wed, 10 Dec 2014 15:56:04 +0100
Subject: [PATCH] as: link to libluajit-5.1

---
 as/src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/as/src/Makefile b/as/src/Makefile
index e7b1838..fff1d0d 100644
--- a/as/src/Makefile
+++ b/as/src/Makefile
@@ -77,7 +77,7 @@ ifeq ($(USE_LUAJIT),1)
   ifeq ($(LD_LUAJIT),static)
     AS_LIBRARIES += $(LUAJIT)/src/libluajit.a
   else
-    LIBRARIES += -L$(LUAJIT)/src -lluajit
+    LIBRARIES += -L$(LUAJIT)/src -lluajit-5.1
   endif
 else
   ifeq ($(LD_LUA),static)
-- 
2.1.3

