--- a/compartment.c	2002-08-24 19:49:26.000000000 +0200
+++ b/compartment.c	2007-02-03 18:36:00.000000000 +0100
@@ -83,18 +83,7 @@
 void help() {
     fprintf(stderr, "%s %s %s %s %s\n\n", PROGRAM_NAME, VERSION, AUTHOR, AUTHOR2, POINTER);
     fprintf(stderr, "Syntax: %s [options] /full/path/to/program\n", prg);
-    fprintf(stderr, "Options:
-\t --chroot path\t chroot to path
-\t --user user\t change uid to this user
-\t --group group\t change gid to this group
-\t --init program\t execute this program/script before doing anything
-\t --cap capset\t set capset name. This option can be used several times.
-\t --verbose\t be verbose
-\t --quiet\t do no logging (to syslog)
-\t --fork\t\t fork (if everything is fine)
-\nHints: always try to chroot; use --user&group if possible; chroot and chown all
-files to another user than root if you use capabilties. Read the README file!
-\nKnown capset names: none");
+    fprintf(stderr, "Options:\n\t --chroot path\t chroot to path\n\t --user user\t change uid to this user\n\t --group group\t change gid to this group\n\t --init program\t execute this program/script before doing anything\n\t --cap capset\t set capset name. This option can be used several times.\n\t --verbose\t be verbose\n\t --quiet\t do no logging (to syslog)\n\t --fork\t\t fork (if everything is fine)\n\nHints: always try to chroot; use --user&group if possible; chroot and chown all\nfiles to another user than root if you use capabilties. Read the README file!\n\nKnown capset names: none");
     tmp = 0;
     while(strlen(cap_set_names[tmp])>0)
         fprintf(stderr," %s",cap_set_names[tmp++]);
@@ -189,11 +178,11 @@
     my_secure();
     openlog("Compartment", LOG_PID, LOG_DAEMON);
     if (sizeof(uid_t) == 4) {
-        (unsigned long int) uidrange = 65535;
+        uidrange = 65535;
     } else { if (sizeof(uid_t) == 8) {
-        (unsigned long int) uidrange = (unsigned long int) 2147483646; //4294967295;
+        uidrange = (unsigned long int) 2147483646; //4294967295;
      } else
-        fprintf(stderr, "Warning: weird uid size: %d\n", sizeof(uid_t));
+        fprintf(stderr, "Warning: weird uid size: %ld\n", sizeof(uid_t));
     }
 
     program_params = 1;
@@ -442,11 +431,7 @@
 	    if (access(_argv[0], X_OK) < 0)
 		print_msg("Execute bit missing, or no permissions to execute %s\n", _argv[0]);
 	    else
-	        print_msg("Could not properly execute %s - the chroot environment might not be
-set up correctly:
-Create the directories /etc and /lib in chroot_dir and run \"ldd %s\"
-to see which libraries are needed. Copy these to chroot_dir/lib, then chdir to
-chroot_dir and execute \"ldconfig -X -r .\"\n", _argv[0], _argv[0]);
+	        print_msg("Could not properly execute %s - the chroot environment might not be\nset up correctly:\nCreate the directories /etc and /lib in chroot_dir and run \"ldd %s\"\nto see which libraries are needed. Copy these to chroot_dir/lib, then chdir to\nchroot_dir and execute \"ldconfig -X -r .\"\n", _argv[0], _argv[0]);
     }
 
     return 1;
