--- trunk/src/PythonStuff.cpp	2014-08-02 12:12:07.140318788 +0200
+++ PythonStuff.cpp	2014-08-02 12:16:18.796985657 +0200
@@ -239,7 +239,7 @@
 					#endif
 				#endif
 
-				Execute(wxString(_T("python \"")) + path + wxString(_T("backplot.py\" \"")) + m_program->m_machine.reader + wxString(_T("\" \"")) + m_filename + wxString(_T("\"")) );
+				Execute(wxString(_T("python2 \"")) + path + wxString(_T("backplot.py\" \"")) + m_program->m_machine.reader + wxString(_T("\" \"")) + m_filename + wxString(_T("\"")) );
 			#endif
 		} // End if - else
 	}
@@ -310,7 +310,7 @@
         Execute(wxString(_T("\"")) + theApp.GetDllFolder() + wxString(_T("\\post.bat\" \"")) + path.GetFullPath() + wxString(_T("\"")));
 #else
 
-        wxString post_path = wxString(_T("python ")) + path.GetFullPath();
+        wxString post_path = wxString(_T("python2 ")) + path.GetFullPath();
 		Execute(post_path);
 #endif
 	}
