diff -ruN atol-0.7.3.orig/plugins/src/ArjLib/ArjLib.cpp atol-0.7.3.new/plugins/src/ArjLib/ArjLib.cpp
--- atol-0.7.3.orig/plugins/src/ArjLib/ArjLib.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/ArjLib/ArjLib.cpp	2008-06-19 23:29:03.000000000 +0200
@@ -9,13 +9,14 @@
 #include "../plugin_int.h"
 #include "ArchiveInfo.h"
 #include <string>
+#include <cstring>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <time.h>
-
-#ifndef _WIN32
- #include <ctype.h>
-#endif
+
+#ifndef _WIN32
+ #include <ctype.h>
+#endif
 
 //global data
 COpenArchivesList	g_ArchList;
diff -ruN atol-0.7.3.orig/plugins/src/Bz2Lib/Bz2Lib.cpp atol-0.7.3.new/plugins/src/Bz2Lib/Bz2Lib.cpp
--- atol-0.7.3.orig/plugins/src/Bz2Lib/Bz2Lib.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/Bz2Lib/Bz2Lib.cpp	2008-06-19 23:33:38.000000000 +0200
@@ -3,6 +3,7 @@
 
 #include "ArchiveInfo.h"
 #include <string>
+#include <cstring>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <time.h>
diff -ruN atol-0.7.3.orig/plugins/src/Bz2Lib/bz2.cpp atol-0.7.3.new/plugins/src/Bz2Lib/bz2.cpp
--- atol-0.7.3.orig/plugins/src/Bz2Lib/bz2.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/Bz2Lib/bz2.cpp	2008-06-19 23:35:20.000000000 +0200
@@ -7,6 +7,7 @@
 ////////////////////////////////////////////////////////////////////////////
 
 #include "bz2.h"
+#include <cstring>
 
 CBZ2::CBZ2()
 {
diff -ruN atol-0.7.3.orig/plugins/src/GzLib/GzLib.cpp atol-0.7.3.new/plugins/src/GzLib/GzLib.cpp
--- atol-0.7.3.orig/plugins/src/GzLib/GzLib.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/GzLib/GzLib.cpp	2008-06-19 23:37:56.000000000 +0200
@@ -9,6 +9,7 @@
 #include "../plugin_int.h"
 #include "ArchiveInfo.h"
 #include <string>
+#include <cstring>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <time.h>
diff -ruN atol-0.7.3.orig/plugins/src/GzLib/gzip.cpp atol-0.7.3.new/plugins/src/GzLib/gzip.cpp
--- atol-0.7.3.orig/plugins/src/GzLib/gzip.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/GzLib/gzip.cpp	2008-06-19 23:39:37.000000000 +0200
@@ -38,6 +38,7 @@
 #include "gzip.h"
 #include <limits.h>
 #include <time.h>
+#include <cstring>
 
 //////////////////////////////////////////////////////////////////////
 // Construction/Destruction
diff -ruN atol-0.7.3.orig/plugins/src/LstLib/EntryTree.cpp atol-0.7.3.new/plugins/src/LstLib/EntryTree.cpp
--- atol-0.7.3.orig/plugins/src/LstLib/EntryTree.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/LstLib/EntryTree.cpp	2008-06-19 23:42:41.000000000 +0200
@@ -1,3 +1,4 @@
+#include <cstring>
 ////////////////////////////////////////////////////////////////////////////
 // Atol file manager project <http://atol.sf.net>
 //
diff -ruN atol-0.7.3.orig/plugins/src/LstLib/EntryTree.h atol-0.7.3.new/plugins/src/LstLib/EntryTree.h
--- atol-0.7.3.orig/plugins/src/LstLib/EntryTree.h	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/LstLib/EntryTree.h	2008-06-19 23:44:26.000000000 +0200
@@ -14,6 +14,7 @@
 #endif // _MSC_VER > 1000
 
 #include <string>
+#include <cstring>
 #include <vector>
 #include <algorithm>
 
diff -ruN atol-0.7.3.orig/plugins/src/TarLib/LibTar.cpp atol-0.7.3.new/plugins/src/TarLib/LibTar.cpp
--- atol-0.7.3.orig/plugins/src/TarLib/LibTar.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/TarLib/LibTar.cpp	2008-06-19 23:47:12.000000000 +0200
@@ -6,6 +6,7 @@
 // File:
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "LibTar.h"
 #ifdef _WIN32
  #include <time.h>
diff -ruN atol-0.7.3.orig/plugins/src/TarLib/TarLib.cpp atol-0.7.3.new/plugins/src/TarLib/TarLib.cpp
--- atol-0.7.3.orig/plugins/src/TarLib/TarLib.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/TarLib/TarLib.cpp	2008-06-19 23:45:32.000000000 +0200
@@ -9,6 +9,7 @@
 #include "../plugin_int.h"
 #include "ArchiveInfo.h"
 #include <string>
+#include <cstring>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <time.h>
diff -ruN atol-0.7.3.orig/plugins/src/ZLib/ZLib.cpp atol-0.7.3.new/plugins/src/ZLib/ZLib.cpp
--- atol-0.7.3.orig/plugins/src/ZLib/ZLib.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/ZLib/ZLib.cpp	2008-06-20 00:02:49.000000000 +0200
@@ -3,6 +3,7 @@
 
 #include "ArchiveInfo.h"
 #include <string>
+#include <cstring>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <time.h>
diff -ruN atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipArchive.cpp atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipArchive.cpp
--- atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipArchive.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipArchive.cpp	2008-06-19 23:54:46.000000000 +0200
@@ -10,7 +10,8 @@
 // For the licensing details see the file License.txt
 ////////////////////////////////////////////////////////////////////////////////
 
-
+#include <cstring>
+#include <climits>
 #include "stdafx.h"
 #include "ZipArchive.h"
 // #include "ZipPathComponent.h"
diff -ruN atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipCentralDir.cpp atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipCentralDir.cpp
--- atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipCentralDir.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipCentralDir.cpp	2008-06-19 23:55:53.000000000 +0200
@@ -10,7 +10,7 @@
 // For the licensing details see the file License.txt
 ////////////////////////////////////////////////////////////////////////////////
 
-
+#include <cstring>
 #include "stdafx.h"
 #include "ZipCentralDir.h"
 #include "ZipArchive.h"
diff -ruN atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipCompatibility.cpp atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipCompatibility.cpp
--- atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipCompatibility.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipCompatibility.cpp	2008-06-19 23:57:05.000000000 +0200
@@ -10,6 +10,7 @@
 // For the licensing details see the file License.txt
 ////////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "stdafx.h"
 #include "ZipCompatibility.h"
 #include "ZipPlatform.h"
diff -ruN atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipException.cpp atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipException.cpp
--- atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipException.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipException.cpp	2008-06-19 23:58:11.000000000 +0200
@@ -10,6 +10,7 @@
 // For the licensing details see the file License.txt
 ////////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "stdafx.h"
 #include "ZipException.h"
 #include <errno.h>
diff -ruN atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipFileHeader.cpp atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipFileHeader.cpp
--- atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/ZipFileHeader.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/ZipFileHeader.cpp	2008-06-20 00:01:27.000000000 +0200
@@ -10,7 +10,7 @@
 // For the licensing details see the file License.txt
 ////////////////////////////////////////////////////////////////////////////////
 
-
+#include <climits>
 #include "stdafx.h"
 #include "ZipFileHeader.h"
 #include "ZipAutoBuffer.h"
diff -ruN atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/stl/ZipString.h atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/stl/ZipString.h
--- atol-0.7.3.orig/plugins/src/ZipLib/ZipArchive/stl/ZipString.h	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/plugins/src/ZipLib/ZipArchive/stl/ZipString.h	2008-06-20 00:00:15.000000000 +0200
@@ -10,6 +10,7 @@
 // For the licensing details see the file License.txt
 ////////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #ifndef __ZIPSTRING_H__
 #define __ZIPSTRING_H__
 #include "stdafx.h"
diff -ruN atol-0.7.3.orig/src/FileSearchDlg.cpp atol-0.7.3.new/src/FileSearchDlg.cpp
--- atol-0.7.3.orig/src/FileSearchDlg.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/FileSearchDlg.cpp	2008-06-20 00:17:49.000000000 +0200
@@ -6,6 +6,7 @@
 // File: Dialog to define options for a file search operation
 //////////////////////////////////////////////////////////////////////////// 
 
+#include <cstdlib>
 #include "support.h"
 #include "FileSearchDlg.h"
 #include "core/PathName.h"
diff -ruN atol-0.7.3.orig/src/PackFilesDlg.cpp atol-0.7.3.new/src/PackFilesDlg.cpp
--- atol-0.7.3.orig/src/PackFilesDlg.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/PackFilesDlg.cpp	2008-06-20 00:19:20.000000000 +0200
@@ -6,6 +6,7 @@
 // File: Start dialog for pack operation, defines archive type, name and additional properties
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "PackFilesDlg.h"
 #include "support.h"
 #include "core/String.h"
diff -ruN atol-0.7.3.orig/src/SiteManagerDlg.cpp atol-0.7.3.new/src/SiteManagerDlg.cpp
--- atol-0.7.3.orig/src/SiteManagerDlg.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/SiteManagerDlg.cpp	2008-06-20 00:26:00.000000000 +0200
@@ -6,6 +6,7 @@
 // File: Dialog to define and start remote connections (FTP, SFTP)
 //////////////////////////////////////////////////////////////////////////// 
 
+#include <cstdlib>
 #include "SiteManagerDlg.h"
 #include "support.h"
 #include "core/opcodes.h"
diff -ruN atol-0.7.3.orig/src/core/ConnectionInfo.h atol-0.7.3.new/src/core/ConnectionInfo.h
--- atol-0.7.3.orig/src/core/ConnectionInfo.h	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/ConnectionInfo.h	2008-06-20 00:04:48.000000000 +0200
@@ -6,6 +6,7 @@
 // File: TOFIX
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #ifndef CONNECTIONINFO_H
 #define CONNECTIONINFO_H
 
diff -ruN atol-0.7.3.orig/src/core/GuiLanguage.cpp atol-0.7.3.new/src/core/GuiLanguage.cpp
--- atol-0.7.3.orig/src/core/GuiLanguage.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/GuiLanguage.cpp	2008-06-20 00:07:05.000000000 +0200
@@ -7,6 +7,7 @@
 ////////////////////////////////////////////////////////////////////////////
 
 #include "GuiLanguage.h"
+#include <cstring>
 #include <algorithm>
 #include <gtk/gtk.h>
 #ifdef ENABLE_NLS
diff -ruN atol-0.7.3.orig/src/core/PathName.cpp atol-0.7.3.new/src/core/PathName.cpp
--- atol-0.7.3.orig/src/core/PathName.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/PathName.cpp	2008-06-20 00:08:49.000000000 +0200
@@ -6,6 +6,7 @@
 // File: PathName class implementation
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "PathName.h"
 #include "System.h"
 #include <sys/stat.h>
diff -ruN atol-0.7.3.orig/src/core/PluginManager.cpp atol-0.7.3.new/src/core/PluginManager.cpp
--- atol-0.7.3.orig/src/core/PluginManager.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/PluginManager.cpp	2008-06-20 00:15:51.000000000 +0200
@@ -6,6 +6,7 @@
 // File: Handles working with archiver plugin .DLL/.so objects
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "PluginManager.h"
 #include "PathName.h"
 #include "System.h"
diff -ruN atol-0.7.3.orig/src/core/ProxyFtp.cpp atol-0.7.3.new/src/core/ProxyFtp.cpp
--- atol-0.7.3.orig/src/core/ProxyFtp.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/ProxyFtp.cpp	2008-06-20 00:32:20.000000000 +0200
@@ -6,6 +6,7 @@
 // File: TOFIX
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "ProxyFtp.h"
 #include "_ftp/xSocket.h"
 
diff -ruN atol-0.7.3.orig/src/core/ProxySocks.cpp atol-0.7.3.new/src/core/ProxySocks.cpp
--- atol-0.7.3.orig/src/core/ProxySocks.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/ProxySocks.cpp	2008-06-20 00:34:01.000000000 +0200
@@ -6,6 +6,7 @@
 // File: TOFIX
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "ProxySocks.h"
 #include "_ftp/xSocket.h"
 #ifdef _WIN32
diff -ruN atol-0.7.3.orig/src/core/String.cpp atol-0.7.3.new/src/core/String.cpp
--- atol-0.7.3.orig/src/core/String.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/String.cpp	2008-06-20 00:10:21.000000000 +0200
@@ -6,6 +6,7 @@
 // File: Implements portable string class
 //////////////////////////////////////////////////////////////////////////// 
 
+#include <cstring>
 #include "String.h"
 #include <stdio.h>
 #include <stdarg.h>
diff -ruN atol-0.7.3.orig/src/core/System.cpp atol-0.7.3.new/src/core/System.cpp
--- atol-0.7.3.orig/src/core/System.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/System.cpp	2008-06-20 00:21:29.000000000 +0200
@@ -6,6 +6,7 @@
 // File: Platform independent part of System utility class implementation.
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "System.h"
 #include "util.h"
 #include "debug.h"
diff -ruN atol-0.7.3.orig/src/core/VfsListing.h atol-0.7.3.new/src/core/VfsListing.h
--- atol-0.7.3.orig/src/core/VfsListing.h	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/VfsListing.h	2008-06-20 00:12:02.000000000 +0200
@@ -6,6 +6,7 @@
 // File: VfsListing is an collection of virtual file system items (VfsItem)
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #ifndef VFSLISTING_H
 #define VFSLISTING_H
 
diff -ruN atol-0.7.3.orig/src/core/_ftp/FtpListParser.cpp atol-0.7.3.new/src/core/_ftp/FtpListParser.cpp
--- atol-0.7.3.orig/src/core/_ftp/FtpListParser.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/_ftp/FtpListParser.cpp	2008-06-20 00:27:55.000000000 +0200
@@ -6,6 +6,7 @@
 // File: TOFIX
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstdlib>
 #include "FtpListParser.h"
 #include <time.h>
 #include <ctype.h>
diff -ruN atol-0.7.3.orig/src/core/_ftp/xSocket.cpp atol-0.7.3.new/src/core/_ftp/xSocket.cpp
--- atol-0.7.3.orig/src/core/_ftp/xSocket.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/_ftp/xSocket.cpp	2008-06-20 00:30:42.000000000 +0200
@@ -6,6 +6,7 @@
 // File: xSocket class handles TCP/IP socket operations
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "xSocket.h"
 #include "../debug.h"
 
diff -ruN atol-0.7.3.orig/src/core/_unx/SystemImpl.cpp atol-0.7.3.new/src/core/_unx/SystemImpl.cpp
--- atol-0.7.3.orig/src/core/_unx/SystemImpl.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/_unx/SystemImpl.cpp	2008-06-20 00:24:12.000000000 +0200
@@ -6,10 +6,11 @@
 // File: System utility class - Unix/Gnome implementation.
 ////////////////////////////////////////////////////////////////////////////
 
-#include <vector>
-#include <string>
+#include <vector>
+#include <string>
+#include <cstring>
 #include <unistd.h>
-#include <dirent.h>
+#include <dirent.h>
 #include <limits.h>
 #include <errno.h>
 #include <sys/types.h>
diff -ruN atol-0.7.3.orig/src/core/_util/Base64.cpp atol-0.7.3.new/src/core/_util/Base64.cpp
--- atol-0.7.3.orig/src/core/_util/Base64.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/_util/Base64.cpp	2008-06-20 00:35:53.000000000 +0200
@@ -9,6 +9,7 @@
 //    to the 4byte border (when encoding).
 //////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "Base64.h"
 #include "../debug.h"
 
diff -ruN atol-0.7.3.orig/src/core/util.cpp atol-0.7.3.new/src/core/util.cpp
--- atol-0.7.3.orig/src/core/util.cpp	2008-06-19 23:27:47.000000000 +0200
+++ atol-0.7.3.new/src/core/util.cpp	2008-06-20 00:13:48.000000000 +0200
@@ -6,6 +6,7 @@
 // File: wxutil implementation
 ////////////////////////////////////////////////////////////////////////////
 
+#include <cstring>
 #include "util.h"
 #include <sys/stat.h>
 #include "PathName.h"
