Index: octopi-0.4.0/notifier/octopi-notifier/mainwindow.cpp
===================================================================
--- octopi-0.4.0.orig/notifier/octopi-notifier/mainwindow.cpp
+++ octopi-0.4.0/notifier/octopi-notifier/mainwindow.cpp
@@ -12,6 +12,8 @@
 #include <QProcess>
 #include <QMessageBox>
 
+#include <KStatusNotifierItem>
+
 /*
  * This is Octopi Notifier slim interface code :-)
  */
@@ -34,14 +36,19 @@ void MainWindow::initSystemTrayIcon()
 {
   m_commandExecuting = ectn_NONE;
   m_outdatedPackageList = new QStringList();
-  m_systemTrayIcon = new QSystemTrayIcon(this);
-  m_systemTrayIcon->setObjectName("systemTrayIcon");
+//  m_systemTrayIcon = new QSystemTrayIcon(this);
+  m_systemTrayIcon = new KStatusNotifierItem(this);
+//  m_systemTrayIcon->setObjectName("systemTrayIcon");
+  m_systemTrayIcon->setTitle("Octopi Notifier");
 
   m_icon = IconHelper::getIconOctopiTransparent();
-  m_systemTrayIcon->setIcon(m_icon); 
+//  m_systemTrayIcon->setIcon(m_icon); 
+  m_systemTrayIcon->setIconByPixmap(m_icon); 
+  m_systemTrayIcon->setToolTipIconByPixmap(m_icon);
   setWindowIcon(m_icon);
-  m_systemTrayIcon->show();
-  m_systemTrayIcon->setToolTip(StrConstants::getSyncDatabases());
+//  m_systemTrayIcon->show();
+  m_systemTrayIcon->setToolTipSubTitle(StrConstants::getSyncDatabases());
+  m_systemTrayIcon->setToolTipTitle("Octopi Notifier");
   qApp->processEvents();
 
   m_actionExit = new QAction(IconHelper::getIconExit(), tr("Exit"), this);
@@ -69,6 +76,9 @@ void MainWindow::initSystemTrayIcon()
   m_systemTrayIconMenu->addAction(m_actionExit);
   m_systemTrayIcon->setContextMenu(m_systemTrayIconMenu);
 
+  // disable "standard" actions (restore & quit)
+  m_systemTrayIcon->setStandardActionsEnabled(false);
+
   connect ( m_systemTrayIcon , SIGNAL( activated( QSystemTrayIcon::ActivationReason ) ),
             this, SLOT( execSystemTrayActivated ( QSystemTrayIcon::ActivationReason ) ) );
 
@@ -320,12 +330,14 @@ void MainWindow::pacmanHelperTimerTimeou
   }
 
   m_icon = IconHelper::getIconOctopiTransparent();
-  m_systemTrayIcon->setIcon(m_icon);
-  m_systemTrayIcon->setToolTip(StrConstants::getSyncDatabases());
+//  m_systemTrayIcon->setIcon(m_icon);
+  m_systemTrayIcon->setIconByPixmap(m_icon);
+  m_systemTrayIcon->setToolTipIconByPixmap(m_icon);
+  m_systemTrayIcon->setToolTipSubTitle(StrConstants::getSyncDatabases());
   qApp->processEvents();
 
   m_systemTrayIconMenu->close();
-  m_systemTrayIcon->setContextMenu(0);
+//  m_systemTrayIcon->setContextMenu(0);
   m_commandExecuting = ectn_SYNC_DATABASE;
   m_pacmanHelperClient->syncdb();
 }
@@ -336,8 +348,8 @@ void MainWindow::pacmanHelperTimerTimeou
 void MainWindow::afterPacmanHelperSyncDatabase()
 {
   m_actionOctopi->setEnabled(true);
-  m_systemTrayIcon->setContextMenu(m_systemTrayIconMenu);
-  m_systemTrayIconMenu->close();
+//  m_systemTrayIcon->setContextMenu(m_systemTrayIconMenu);
+//  m_systemTrayIconMenu->close();
   m_commandExecuting = ectn_NONE;
 
   disconnect(m_pacmanDatabaseSystemWatcher,
@@ -355,14 +367,16 @@ void MainWindow::afterPacmanHelperSyncDa
       if (m_numberOfOutdatedPackages == 1)
       {
         notification = StrConstants::getOneNewUpdate();
-        m_systemTrayIcon->setToolTip(notification);
-        if (!UnixCommand::isAppRunning("spun", true)) sendNotification(notification);
+        m_systemTrayIcon->setToolTipSubTitle(notification);
+//        if (!UnixCommand::isAppRunning("spun", true)) sendNotification(notification);
+	m_systemTrayIcon->showMessage("Octopi Notifier", notification, m_systemTrayIcon->iconName());
       }
       else if (m_numberOfOutdatedPackages > 1)
       {
         notification = StrConstants::getNewUpdates().arg(m_numberOfOutdatedPackages);
-        m_systemTrayIcon->setToolTip(notification);
-        if (!UnixCommand::isAppRunning("spun", true)) sendNotification(notification);
+        m_systemTrayIcon->setToolTipSubTitle(notification);
+//        if (!UnixCommand::isAppRunning("spun", true)) sendNotification(notification);
+	m_systemTrayIcon->showMessage("Octopi Notifier", notification, m_systemTrayIcon->iconName());
       }
     }
   }
@@ -373,14 +387,16 @@ void MainWindow::afterPacmanHelperSyncDa
     if (numberOfOutdatedPackages == 1)
     {
       notification = StrConstants::getOneNewUpdate();
-      m_systemTrayIcon->setToolTip(notification);
-      if (!UnixCommand::isAppRunning("spun", true)) sendNotification(notification);
+      m_systemTrayIcon->setToolTipSubTitle(notification);
+//      if (!UnixCommand::isAppRunning("spun", true)) sendNotification(notification);
+      m_systemTrayIcon->showMessage("Octopi Notifier", notification, m_systemTrayIcon->iconName());
     }
     else if (numberOfOutdatedPackages > 1)
     {
       notification = StrConstants::getNewUpdates().arg(numberOfOutdatedPackages);
-      m_systemTrayIcon->setToolTip(notification);
-      if (!UnixCommand::isAppRunning("spun", true)) sendNotification(notification);
+      m_systemTrayIcon->setToolTipSubTitle(notification);
+//      if (!UnixCommand::isAppRunning("spun", true)) sendNotification(notification);
+      m_systemTrayIcon->showMessage("Octopi Notifier", notification, m_systemTrayIcon->iconName());
     }
   }
 
@@ -424,28 +440,28 @@ void MainWindow::refreshAppIcon()
 
   if (m_numberOfOutdatedPackages == 0 && m_numberOfOutdatedAURPackages == 0)
   {
-    m_systemTrayIcon->setToolTip("");
+    m_systemTrayIcon->setToolTipSubTitle("");
   }
   else if (m_numberOfOutdatedPackages > 0)
   {
     if (m_numberOfOutdatedPackages == 1)
     {
-      m_systemTrayIcon->setToolTip(StrConstants::getOneNewUpdate());
+      m_systemTrayIcon->setToolTipSubTitle(StrConstants::getOneNewUpdate());
     }
     else if (m_numberOfOutdatedPackages > 1)
     {
-      m_systemTrayIcon->setToolTip(StrConstants::getNewUpdates().arg(m_numberOfOutdatedPackages));
+      m_systemTrayIcon->setToolTipSubTitle(StrConstants::getNewUpdates().arg(m_numberOfOutdatedPackages));
     }
   }
   else if (m_numberOfOutdatedAURPackages > 0)
   {
     if (m_numberOfOutdatedAURPackages == 1)
     {
-      m_systemTrayIcon->setToolTip(StrConstants::getOneNewUpdate());
+      m_systemTrayIcon->setToolTipSubTitle(StrConstants::getOneNewUpdate());
     }
     else if (m_numberOfOutdatedAURPackages > 1)
     {
-      m_systemTrayIcon->setToolTip(StrConstants::getNewUpdates().arg(m_numberOfOutdatedAURPackages));
+      m_systemTrayIcon->setToolTipSubTitle(StrConstants::getNewUpdates().arg(m_numberOfOutdatedAURPackages));
     }
   }
 
@@ -458,20 +474,27 @@ void MainWindow::refreshAppIcon()
     }
 
     m_icon = IconHelper::getIconOctopiRed();
+    m_systemTrayIcon->setAttentionIconByPixmap(m_icon);
+    m_systemTrayIcon->setStatus(KStatusNotifierItem::NeedsAttention);
   }
   else if(m_outdatedAURPackageList->count() > 0) //YELLOW ICON!
   {
     m_actionSystemUpgrade->setVisible(false);
     m_icon = IconHelper::getIconOctopiYellow();
+    m_systemTrayIcon->setAttentionIconByPixmap(m_icon);
+    m_systemTrayIcon->setStatus(KStatusNotifierItem::NeedsAttention);
   }
   else //YEAHHH... GREEN ICON!
   {
     m_actionSystemUpgrade->setVisible(false);
     m_icon = IconHelper::getIconOctopiGreen();
+    m_systemTrayIcon->setStatus(KStatusNotifierItem::Passive);
   }
 
   setWindowIcon(m_icon);
-  m_systemTrayIcon->setIcon(m_icon);
+//  m_systemTrayIcon->setIcon(m_icon);
+  m_systemTrayIcon->setIconByPixmap(m_icon);
+  m_systemTrayIcon->setToolTipIconByPixmap(m_icon);
 }
 
 /*
Index: octopi-0.4.0/notifier/octopi-notifier/mainwindow.h
===================================================================
--- octopi-0.4.0.orig/notifier/octopi-notifier/mainwindow.h
+++ octopi-0.4.0/notifier/octopi-notifier/mainwindow.h
@@ -14,6 +14,8 @@ class QAction;
 class QFileSystemWatcher;
 class PacmanHelperClient;
 
+class KStatusNotifierItem;
+
 enum ExecOpt { ectn_NORMAL_EXEC_OPT, ectn_SYSUPGRADE_EXEC_OPT, ectn_SYSUPGRADE_NOCONFIRM_EXEC_OPT };
 
 class MainWindow : public QMainWindow
@@ -58,7 +60,8 @@ private:
   QStringList *m_outdatedPackageList;
   QStringList *m_outdatedAURPackageList;
   QTimer *m_pacmanHelperTimer;
-  QSystemTrayIcon *m_systemTrayIcon;
+//  QSystemTrayIcon *m_systemTrayIcon;
+  KStatusNotifierItem * m_systemTrayIcon;
   QMenu *m_systemTrayIconMenu;
   QFileSystemWatcher *m_pacmanDatabaseSystemWatcher;
   PacmanHelperClient *m_pacmanHelperClient;
Index: octopi-0.4.0/notifier/octopi-notifier/octopi-notifier.pro
===================================================================
--- octopi-0.4.0.orig/notifier/octopi-notifier/octopi-notifier.pro
+++ octopi-0.4.0/notifier/octopi-notifier/octopi-notifier.pro
@@ -4,7 +4,9 @@
 #
 #-------------------------------------------------
 
-QT += core xml gui network dbus
+include(/usr/share/qt/mkspecs/modules/qt_KNotifications.pri)
+
+QT += core xml gui network dbus KNotifications
 
 greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
 
