diff -Naur sonic-pi-2.5.0/app/gui/qt/SonicPi.pro sonic-pi-2.5.0.new/app/gui/qt/SonicPi.pro
--- sonic-pi-2.5.0/app/gui/qt/SonicPi.pro	2015-04-13 21:22:07.000000000 +0200
+++ sonic-pi-2.5.0.new/app/gui/qt/SonicPi.pro	2015-05-10 20:08:50.477523305 +0200
@@ -97,7 +97,7 @@
 RC_FILE = SonicPi.rc
 
 ICON = images/app.icns
-LIBS         += -lqscintilla2
+LIBS         += -lqt5scintilla2
 
 win32 {
 	install_qsci.files = $$[QT_INSTALL_LIBS]\qscintilla2.dll
diff -Naur sonic-pi-2.5.0/app/server/bin/compile-extensions.rb sonic-pi-2.5.0.new/app/server/bin/compile-extensions.rb
--- sonic-pi-2.5.0/app/server/bin/compile-extensions.rb	2015-04-13 21:22:07.000000000 +0200
+++ sonic-pi-2.5.0.new/app/server/bin/compile-extensions.rb	2015-05-10 20:03:57.040160287 +0200
@@ -40,7 +40,6 @@
   File.expand_path(File.dirname(__FILE__) + '/../vendor/ruby-prof/ext/ruby_prof/'),
 
   File.expand_path(File.dirname(__FILE__) + '/../vendor/interception/ext/'),
-  File.expand_path(File.dirname(__FILE__) + '/../vendor/did_you_mean/ext/did_you_mean')
 ]
 
 if os == :osx
