unchanged:
--- swac-explore-0.4.1/src/sqlite3++.cc	2009-02-06 19:01:29.000000000 +0800
+++ swac-explore-0.4.1.new/src/sqlite3++.cc	2010-03-03 00:32:09.680404192 +0800
@@ -18,6 +18,7 @@
  */
 
 #include <iostream>
+#include <cstdio>
 
 #include "sqlite3++.hh"
 
unchanged:
--- swac-get-0.5/src/basic.cc	2008-09-30 04:53:36.000000000 +0800
+++ swac-get-0.5.new/src/basic.cc	2010-03-03 00:44:33.243671263 +0800
@@ -22,6 +22,7 @@
 
 #include <sstream>
 #include <iostream>
+#include <cstdio>
 #include "basic.hh"
 #include "global.hh"
 
unchanged:
--- swac-get-0.5/src/posix.cc	2009-02-02 08:08:57.000000000 +0800
+++ swac-get-0.5.new/src/posix.cc	2010-03-03 01:32:14.456718510 +0800
@@ -27,6 +27,7 @@
 #include <cstring>
 #include <list>
 #include <iostream>
+#include <cstdio>
 
 #include "posix.hh"
 
unchanged:
--- swac-get-0.5/src/sqlite3++.cc	2009-02-02 08:03:42.000000000 +0800
+++ swac-get-0.5.new/src/sqlite3++.cc	2010-03-03 00:32:24.537494325 +0800
@@ -19,6 +19,7 @@
  */
 
 #include <iostream>
+#include <cstdio>
 #include <string>
 
 #include "sqlite3++.hh"
unchanged:
--- swac-play-0.2/src/sqlite3++.cc	2009-02-05 18:57:39.000000000 +0800
+++ swac-play-0.2.new/src/sqlite3++.cc	2010-03-03 00:32:01.727079753 +0800
@@ -18,6 +18,7 @@
  */
 
 #include <iostream>
+#include <cstdio>
 
 #include "sqlite3++.hh"
 
only in patch2:
unchanged:
--- swac-scan-0.2/src/speex.cc	2009-02-05 18:58:23.000000000 +0800
+++ swac-scan-0.2.new/src/speex.cc	2010-03-03 01:32:49.093798745 +0800
@@ -19,6 +19,7 @@
 
 #include <ogg/ogg.h>
 #include <iostream>
+#include <cstdio>
 #include "speex.hh"
 
 