*** sam2p-svn/rule.hpp	2013-07-28 12:12:53.080419179 +0200
--- sam2p-svn/rule.hpp.new	2013-07-28 12:14:37.048005793 +0200
***************
*** 88,94 ****
        PR_PNGAuto=15,
        PR_PNGAutoMaybe=25,
        PR_PNGAutoBadUnsigned=45,
!       PR_PNGAutoBadSigned=55,
      END_STATIC_ENUM()
      pr_t Predictor;
  
--- 88,94 ----
        PR_PNGAuto=15,
        PR_PNGAutoMaybe=25,
        PR_PNGAutoBadUnsigned=45,
!       PR_PNGAutoBadSigned=55
      END_STATIC_ENUM()
      pr_t Predictor;
  
