diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d52601..55c4da1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -103,7 +103,8 @@ macro(qtc_check_x11)
     set(QTC_ENABLE_X11 Off)
     return()
   endif()
-  pkg_check_modules(QTC_X11 xcb xcb-image x11-xcb)
+  unset(__pkg_config_checked_QTC_X11 CACHE)
+  pkg_check_modules(QTC_X11 xcb x11-xcb)
   if(NOT QTC_X11_FOUND)
     message(WARNING
       "Cannot find required modules for x11, disable x11 support.")
@@ -132,6 +133,7 @@ endif()
 
 if(ENABLE_GTK2)
   find_package(PkgConfig REQUIRED)
+  unset(__pkg_config_checked_GTK2 CACHE)
   pkg_check_modules(GTK2 REQUIRED gtk+-2.0)
   if(GTK2_THEME_PREFIX)
     set(GTK2_PREIX "${GTK2_THEME_PREFIX}")
diff --git a/qtcurve-utils/x11helpers.c b/qtcurve-utils/x11helpers.c
index bcd5a6c..be7aa36 100644
--- a/qtcurve-utils/x11helpers.c
+++ b/qtcurve-utils/x11helpers.c
@@ -24,6 +24,5 @@
 #include "x11icccm.h"
 #include "log.h"
-#include <xcb/xcb_image.h>
 #include <X11/Xlib.h>
 #include <X11/Xatom.h>
 #include <shadow0-png.h>
diff --git a/qtcurve-utils/x11utils.c b/qtcurve-utils/x11utils.c
index ed1ea70..c3c58b1 100644
--- a/qtcurve-utils/x11utils.c
+++ b/qtcurve-utils/x11utils.c
@@ -24,7 +24,6 @@
 #include "x11shadow_p.h"
 #include "log.h"
 
-#include <xcb/xcb_image.h>
 #include <X11/Xlib-xcb.h>
 
 static Display *qtc_disp = NULL;
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 3615487..c1397f9 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -27,6 +27,7 @@ elseif(ENABLE_QT4)
   set(gen_image_header_DEFS)
 else()
   find_package(PkgConfig REQUIRED)
+  unset(__pkg_config_checked_GEN_IMG_GTK CACHE)
   pkg_check_modules(GEN_IMG_GTK REQUIRED cairo-png)
   set(gen_image_header_SRCS
     gen_image_header_cairo.c)
-- 
1.8.4

