Description: Always disable valgrind tests
Origin: vendor
Bug-Debian: http://bugs.debian.org/690968
Forwarded: not-needed
Last-Update: 2012-10-22

--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -59,7 +59,7 @@
 TEST_F =
 TEST_T =
 else # if not cross-compiling:
-TEST = srcdir=$(srcdir) $(PERL) $(PERLFLAGS) $(srcdir)/runtests.pl
+TEST = srcdir=$(srcdir) $(PERL) $(PERLFLAGS) $(srcdir)/runtests.pl -n
 TEST_Q = -a -s
 TEST_F = -a -p -r
 TEST_T = -a -t
