diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -2642,18 +2642,18 @@ LIBS=$ac_check_lib_save_LIBS
 fi
 echo "$as_me:$LINENO: result: $ac_cv_lib_ssl_RSA_new" >&5
 echo "${ECHO_T}$ac_cv_lib_ssl_RSA_new" >&6
-if test $ac_cv_lib_ssl_RSA_new = yes; then
+#if test $ac_cv_lib_ssl_RSA_new = yes; then
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBSSL 1
 _ACEOF
 
   LIBS="-lssl $LIBS"
 
-else
+#else
 
-		    echo "Error! You need libssl."
-		    exit -1
-fi
+#		    echo "Error! You need libssl."
+#		    exit -1
+#fi
 
 
 
diff --git a/Makefile b/Makefile.in
--- a/raop_play/Makefile.in
+++ b/raop_play/Makefile.in
@@ -20,7 +20,7 @@ pcm_stream.o flac_stream.o
 all: $(TARGET)
 
 raop_play: $(OBJS)
-	$(CC) -o $@  -lssl -lsamplerate -lid3tag $^
+	$(CC) -o $@  -lssl -lsamplerate -lid3tag -lcrypto $^
 
 install:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)/
