--- opencity-0.0.6.4stable.orig/opencity.desktop	2011-07-03 22:37:04.000000000 +0600
+++ opencity-0.0.6.4stable/opencity.desktop	2012-05-21 19:16:00.852389675 +0600
@@ -1,12 +1,11 @@
 [Desktop Entry]
 Name=OpenCity
-Encoding=UTF-8
 Comment=A 3D city simulator
 Comment[fr]=Jeu de simulation de Ville en 3D
 Categories=Game;StrategyGame;
 Exec=opencity
 GenericName=City simulator
 GenericName[fr]=Simulateur de ville
-Icon=opencity.png
+Icon=opencity
 Terminal=false
-Type=Application
\ No newline at end of file
+Type=Application
