diff -urN qpdf-2.2.3/libqpdf/QPDF_linearization.cc qpdf-2.2.3-patched//libqpdf/QPDF_linearization.cc
--- qpdf-2.2.3/libqpdf/QPDF_linearization.cc	2011-01-31 15:59:42.000000000 +0100
+++ qpdf-2.2.3-patched//libqpdf/QPDF_linearization.cc	2011-05-15 13:25:40.000000000 +0200
@@ -94,7 +94,6 @@
 
     static PCRE lindict_re("(?s:(\\d+)\\s+0\\s+obj\\s*<<)");
 
-    off_t offset = -1;
     int lindict_obj = -1;
     char* p = buf;
     while (lindict_obj == -1)
@@ -102,7 +101,6 @@
 	PCRE::Match m(lindict_re.match(p));
 	if (m)
 	{
-	    offset = m.getOffset(0) + (p - buf);
 	    lindict_obj = atoi(m.getMatch(1).c_str());
 	    if (m.getMatch(0).find('\n') != std::string::npos)
 	    {
