--- qpdf-2.3.1-orig/examples/pdf-mod-info.cc	2011-12-28 23:19:40.000000000 +0100
+++ qpdf-2.3.1/examples/pdf-mod-info.cc	2012-05-17 23:03:28.675581199 +0200
@@ -8,6 +8,7 @@
 #include <iostream>
 #include <string.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 static char const* version = "1.1";
 static char const* whoami = 0;
--- qpdf-2.3.1-orig/include/qpdf/QPDFExc.hh	2011-12-28 23:19:40.000000000 +0100
+++ qpdf-2.3.1/include/qpdf/QPDFExc.hh	2012-05-17 23:04:12.055583814 +0200
@@ -11,6 +11,7 @@
 #include <qpdf/DLL.h>
 #include <qpdf/Constants.h>
 #include <stdexcept>
+#include <sys/types.h>
 
 class QPDFExc: public std::runtime_error
 {
--- qpdf-2.3.1-orig/include/qpdf/QPDFObjectHandle.hh	2011-12-28 23:19:40.000000000 +0100
+++ qpdf-2.3.1/include/qpdf/QPDFObjectHandle.hh	2012-05-17 23:03:50.635580174 +0200
@@ -12,6 +12,7 @@
 #include <vector>
 #include <set>
 #include <map>
+#include <sys/types.h>
 
 #include <qpdf/DLL.h>
 
