--- configure.in.orig	2005-07-08 11:17:02.000000000 +0100
+++ configure.in	2005-07-08 11:14:08.000000000 +0100
@@ -19,6 +19,8 @@
 AC_SUBST(PACKAGE)
 AC_SUBST(VERSION)
 
+AC_PROG_INSTALL
+
 AC_PREREQ(2.0)
 
 case "`uname -s`" in
@@ -52,30 +54,46 @@
 SC_LOAD_TKCONFIG
 
 #--------------------------------------------------------------------
-# See if --with-itcl was specified to give the [incr Tcl] and [incr Tk]
-# source code location.  If not, then we'll assume that the Iwidgets
-# package is being installed directly beneath the toplevel Itcl directory.
+# See if --with-itcl was specified to give the [incr Tcl] source code
+# location.  If not, then we'll assume that the Iwidgets package is being
+# installed as a sibling directory to itcl and we attempt to guess the
+# correct path by the last item matched with ../itcl*.
 #--------------------------------------------------------------------
 
-AC_ARG_WITH(itcl, [  --with-itcl=DIR         use Itcl 3.0 binaries from DIR],
-        itcl_search=$withval, itcl_search=`cd ${IWIDGETS_SRC_DIR}/..; pwd`)
+AC_ARG_WITH(itcl, [  --with-itcl=DIR         use Itcl 3.3 binaries from DIR],
+        itcl_search=$withval, itcl_search="/usr/lib/itcl3.4")
 
 ITCL_LIB_DIR=""
-if test -r $itcl_search/itcl/library/itcl.tcl; then
-    if test -r $itcl_search/itk/library/itk.tcl; then
-        if test -r $itcl_search/config/mkinstalldirs; then
-            ITCL_LIB_DIR=$itcl_search/itcl/library
-            ITCL_SRC_DIR=$itcl_search/itcl
-            ITK_LIB_DIR=$itcl_search/itk/library
-            ITK_SRC_DIR=$itcl_search/itk
-        fi
-    fi
+if test -r $itcl_search/itcl.tcl; then
+    ITCL_LIB_DIR=$itcl_search
+    ITCL_SRC_DIR=$itcl_search
 fi
 
 if test -z "$ITCL_LIB_DIR"; then
-    AC_MSG_ERROR(Can't find Itcl source.  Use --with-itcl to specify the the toplevel [incr Tcl] directory on your system.  (This directory should contain itcl/library/itcl.tcl, itk/library/itk.tcl, and config/mkinstalldirs.))
+    AC_MSG_ERROR(Can't find Itcl source.  Use --with-itcl to specify the the toplevel [incr Tcl] directory on your system.  (This directory should contain itcl.tcl.))
 fi
 
+#--------------------------------------------------------------------
+# See if --with-itk was specified to give the [incr Tk] source code
+# location.  If not, then we'll assume that the Iwidgets package is being
+# installed as a sibling directory to itcl and we attempt to guess the
+# correct path by the last item matched with ../itk*.
+#--------------------------------------------------------------------
+
+AC_ARG_WITH(itk, [  --with-itk=DIR         use Itk 3.3 binaries from DIR],
+        itk_search=$withval, itk_search="/usr/lib/itk3.3")
+
+ITK_LIB_DIR=""
+if test -r $itk_search/itk.tcl; then
+    ITK_LIB_DIR=$itk_search
+    ITK_SRC_DIR=$itk_search
+fi
+
+if test -z "$ITK_LIB_DIR"; then
+    AC_MSG_ERROR(Can't find Itk source.  Use --with-itk to specify the the toplevel [incr Tk] directory on your system.  (This directory should contain itk.tcl.))
+fi
+
+
 
 #--------------------------------------------------------------------
 #   Needed for running the test suite
--- Makefile.in.orig	2005-07-08 11:17:12.000000000 +0100
+++ Makefile.in	2005-07-08 11:01:20.000000000 +0100
@@ -78,11 +78,11 @@
 # modify any of this stuff by hand.
 #----------------------------------------------------------------
 
-INSTALL =		$(TCLSH_PROG) `$(CYGPATH) $(ITCL_SRC_DIR)/../config/installFile.tcl` -c
-INSTALL_PROGRAM =	$(INSTALL)
-INSTALL_DATA =		$(INSTALL) -m 644
-INSTALL_SCRIPT =	$(INSTALL) -m 555
-MKINSTALLDIRS =		$(ITCL_SRC_DIR)/../config/mkinstalldirs
+INSTALL =		@INSTALL@
+INSTALL_PROGRAM =	@INSTALL_PROGRAM@
+INSTALL_DATA =		@INSTALL_DATA@
+INSTALL_SCRIPT =	@INSTALL_SCRIPT@
+MKINSTALLDIRS =		mkdir -p
 RANLIB =		@RANLIB@
 LN_S =			ln -s
 TOP_DIR =		$(srcdir)
--- install-sh.orig	2005-07-08 11:38:48.000000000 +0100
+++ install-sh	2005-07-08 11:14:05.000000000 +0100
@@ -0,0 +1,119 @@
+#!/bin/sh
+
+#
+# install - install a program, script, or datafile
+# This comes from X11R5; it is not part of GNU.
+#
+# $XConsortium: install.sh,v 1.2 89/12/18 14:47:22 jim Exp $
+#
+# This script is compatible with the BSD install script, but was written
+# from scratch.
+#
+
+
+# set DOITPROG to echo to test this script
+
+# Don't use :- since 4.3BSD and earlier shells don't like it.
+doit="${DOITPROG-}"
+
+
+# put in absolute paths if you don't have them in your path; or use env. vars.
+
+mvprog="${MVPROG-mv}"
+cpprog="${CPPROG-cp}"
+chmodprog="${CHMODPROG-chmod}"
+chownprog="${CHOWNPROG-chown}"
+chgrpprog="${CHGRPPROG-chgrp}"
+stripprog="${STRIPPROG-strip}"
+rmprog="${RMPROG-rm}"
+
+instcmd="$mvprog"
+chmodcmd=""
+chowncmd=""
+chgrpcmd=""
+stripcmd=""
+rmcmd="$rmprog -f"
+mvcmd="$mvprog"
+src=""
+dst=""
+
+while [ x"$1" != x ]; do
+    case $1 in
+	-c) instcmd="$cpprog"
+	    shift
+	    continue;;
+
+	-m) chmodcmd="$chmodprog $2"
+	    shift
+	    shift
+	    continue;;
+
+	-o) chowncmd="$chownprog $2"
+	    shift
+	    shift
+	    continue;;
+
+	-g) chgrpcmd="$chgrpprog $2"
+	    shift
+	    shift
+	    continue;;
+
+	-s) stripcmd="$stripprog"
+	    shift
+	    continue;;
+
+	*)  if [ x"$src" = x ]
+	    then
+		src=$1
+	    else
+		dst=$1
+	    fi
+	    shift
+	    continue;;
+    esac
+done
+
+if [ x"$src" = x ]
+then
+	echo "install:  no input file specified"
+	exit 1
+fi
+
+if [ x"$dst" = x ]
+then
+	echo "install:  no destination specified"
+	exit 1
+fi
+
+
+# If destination is a directory, append the input filename; if your system
+# does not like double slashes in filenames, you may need to add some logic
+
+if [ -d $dst ]
+then
+	dst="$dst"/`basename $src`
+fi
+
+# Make a temp file name in the proper directory.
+
+dstdir=`dirname $dst`
+dsttmp=$dstdir/#inst.$$#
+
+# Move or copy the file name to the temp name
+
+$doit $instcmd $src $dsttmp
+
+# and set any options; do chmod last to preserve setuid bits
+
+if [ x"$chowncmd" != x ]; then $doit $chowncmd $dsttmp; fi
+if [ x"$chgrpcmd" != x ]; then $doit $chgrpcmd $dsttmp; fi
+if [ x"$stripcmd" != x ]; then $doit $stripcmd $dsttmp; fi
+if [ x"$chmodcmd" != x ]; then $doit $chmodcmd $dsttmp; fi
+
+# Now rename the file to the real destination.
+
+$doit $rmcmd $dst
+$doit $mvcmd $dsttmp $dst
+
+
+exit 0
