--- /home/testuser/tmp/dcp-j525w/opt/brother/Printers/dcpj525w/cupswrapper/cupswrapperdcpj525w (copy)
+++ /home/testuser/tmp/dcp-j525w/opt/brother/Printers/dcpj525w/cupswrapper/cupswrapperdcpj525w
@@ -47,21 +47,14 @@
   rm -f /usr/lib64/cups/filter/brother_lpdwrapper_${printer_model}
   rm -f /opt/brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1
 #  rm -f /usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1
-if [  -e /etc/init.d/cups ]; then
-   /etc/init.d/cups restart
-elif [  -e /etc/init.d/cupsys ]; then
-   /etc/init.d/cupsys restart
-fi
+systemctl restart cups.service
 #  /etc/init.d/cups restart
   exit 0
 fi
 if [ "$1" = "-r" ]; then
   lpadmin -x ${printer_name}
-if [  -e /etc/init.d/cups ]; then
-   /etc/init.d/cups restart
-elif [  -e /etc/init.d/cupsys ]; then
-   /etc/init.d/cupsys restart
-fi
+  
+systemctl restart cups.service
 #  /etc/init.d/cups restart
   exit 0
 fi
@@ -288,19 +281,7 @@
 
 #chmod a+w /opt/brother/${device_model}/${printer_model}/inf/br${printer_model}rc
 #chmod a+w /opt/brother/${device_model}/${printer_model}/inf
-if [ -e /etc/init.d/lpd ]; then
-   /etc/init.d/lpd stop
-fi
-if [  -e /etc/init.d/lprng ]; then
-   /etc/init.d/lprng stop
-fi
-
-
-if [  -e /etc/init.d/cups ]; then
-   /etc/init.d/cups restart
-elif [  -e /etc/init.d/cupsys ]; then
-   /etc/init.d/cupsys restart
-fi
+systemctl restart cups.service
 
 sleep 2s
 
