diff --git a/local_gridfile.h b/local_gridfile.h
index 2353a1a..70e48b0 100644
--- a/local_gridfile.h
+++ b/local_gridfile.h
@@ -1,6 +1,7 @@
 #ifndef _LOCAL_GRIDFILE_H
 #define _LOCAL_GRIDFILE_H
 
+#include "sys/types.h"
 #include <vector>
 #include <cstring>
 #include <iostream>
diff --git a/Makefile b/Makefile
index e73c523..adbca2b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,13 @@
 CC=g++
 CCOPTS=-g -D_FILE_OFFSET_BITS=64 -I. -I/usr/local/include
-LDOPTS=-L/usr/local/lib -L. -lmongoclient -lfuse_ino64 -lboost_thread-mt -lboost_filesystem-mt -lboost_system-mt
+
+MACHINE= $(shell uname -s)
+
+ifeq ($(MACHINE),Darwin)
+	LDOPTS=-L/usr/local/lib -L. -lmongoclient -lfuse_ino64 -lboost_thread-mt -lboost_filesystem-mt -lboost_system-mt
+else
+	LDOPTS=-L/usr/local/lib -L. -lmongoclient -lfuse -lboost_thread -lboost_filesystem -lboost_system -lpthread
+endif
 
 mount_gridfs : operations.o local_gridfile.o main.o options.o
 	$(CC) -o mount_gridfs main.o operations.o options.o local_gridfile.o $(LDOPTS)
diff --git a/local_gridfile.h b/local_gridfile.h
index 2353a1a..182f2fd 100644
--- a/local_gridfile.h
+++ b/local_gridfile.h
@@ -5,6 +5,10 @@
 #include <cstring>
 #include <iostream>
 
+#ifdef __linux__
+  #include "sys/types.h"
+#endif
+
 const unsigned int DEFAULT_CHUNK_SIZE = 256 * 1024;
 
 class LocalGridFile {
