=== modified file 'core/luban/ui/actions/__init__.py'
--- core/luban/ui/actions/__init__.py	2012-07-05 15:40:57 +0000
+++ core/luban/ui/actions/__init__.py	2012-10-23 04:58:24 +0000
@@ -127,7 +127,7 @@
 def importAllElementActions():
     modules = action_modules
     for name in modules:
-        __import__(name, fromlist=['.'], globals=globals())
+        __import__(name, fromlist=['.'], globals=globals(), level=1)
         continue
     return
 importAllElementActions()

=== modified file 'core/luban/ui/elements/__init__.py'
--- core/luban/ui/elements/__init__.py	2011-12-08 17:29:33 +0000
+++ core/luban/ui/elements/__init__.py	2012-10-23 04:58:24 +0000
@@ -44,7 +44,7 @@
 def importAllElements():
     modules = elementtypes
     for name in modules:
-        __import__(name, fromlist=['.'], globals=globals())
+        __import__(name, fromlist=['.'], globals=globals(), level=1)
         continue
     return
 importAllElements()

