=== modified file 'timber/luban.timber/actions/__init__.py'
--- timber/luban.timber/actions/__init__.py	2011-11-14 01:59:59 +0000
+++ timber/luban.timber/actions/__init__.py	2012-10-24 03:48:03 +0000
@@ -50,7 +50,7 @@
 def importAllActions():
     modules = action_modules
     for name in modules:
-        __import__(name, fromlist=['.'], globals=globals())
+        __import__(name, fromlist=['.'], globals=globals(), level=1)
         continue
     return
 importAllActions()

=== modified file 'timber/luban.timber/elements/__init__.py'
--- timber/luban.timber/elements/__init__.py	2012-06-17 21:57:08 +0000
+++ timber/luban.timber/elements/__init__.py	2012-10-24 03:48:03 +0000
@@ -61,7 +61,7 @@
 def importAllElements():
     modules = elementtypes
     for name in modules:
-        __import__(name, fromlist=['.'], globals=globals())
+        __import__(name, fromlist=['.'], globals=globals(), level=1)
         continue
     return
 importAllElements()

