diff -rupN omnibus-backend.org/CMakeLists.txt omnibus-backend/CMakeLists.txt
--- omnibus-backend.org/CMakeLists.txt	2012-04-09 16:53:23.932853258 +0200
+++ omnibus-backend/CMakeLists.txt	2012-04-09 16:54:12.849254920 +0200
@@ -4,6 +4,7 @@ project(omnibus-filetransfer)
 ### LIBS
 ## BOOST
 find_package(Boost REQUIRED COMPONENTS system filesystem date_time)
+find_package(Threads)
 
 ## Ariba
 # BRANCH: Ariba-Prefix manually set
@@ -61,7 +62,7 @@ add_executable(omnibus-bin
 
 
 ### LINKER
-target_link_libraries(omnibus-bin ${Boost_LIBRARIES} ${ARIBA_LIBRARY})
+target_link_libraries(omnibus-bin ${Boost_LIBRARIES} ${ARIBA_LIBRARY} ${CMAKE_THREAD_LIBS_INIT})
 
 
 ### INSTALL
Binary files omnibus-backend.org/.svn/wc.db and omnibus-backend/.svn/wc.db differ
