Index: lxdm-svn/src/lxdm.c
===================================================================
--- lxdm-svn/src/lxdm.c	(revision 2334)
+++ lxdm-svn/src/lxdm.c	(working copy)
@@ -212,6 +212,15 @@
     exit(0);
 }
 
+void log_clear(void)
+{
+    FILE *log;
+    log=fopen("/var/log/lxdm.log","w");
+    if(!log)
+        return;
+    fclose(log);
+}
+
 void log_print(char *fmt, ...)
 {
     static FILE *log;
@@ -229,6 +238,11 @@
         if( !log )
             return;
     }
+    fclose(log);
+    log=freopen("/var/log/lxdm.log","a",stdout);
+    setvbuf(stdout, NULL, _IOLBF, BUFSIZ);
+    freopen("/var/log/lxdm.log","a",stderr);
+    setvbuf(stderr, NULL, _IONBF, BUFSIZ);
     va_start(ap, fmt);
     vfprintf(log, fmt, ap);
     va_end(ap);
@@ -594,6 +608,7 @@
     if( !getenv("DISPLAY") )
         putenv("DISPLAY=:0");
 
+    log_print("%s\n","start X");
     create_server_auth();
 
     arg = g_key_file_get_string(config, "server", "arg", 0);
@@ -607,6 +622,8 @@
     {
     case 0:
         setpgid( 0, getpid() );
+	// required to catch output
+	log_print("%s\n", "lxdm forked");
         execvp(args[0], args);
         break;
     case -1:
@@ -1040,6 +1057,7 @@
 
     set_signal();
     lxdm_get_tty();
+    log_clear();
     startx();
 
     for( tmp = 0; tmp < 200; tmp++ )
