diff -aur xrmap-2.34.orig/Imakefile xrmap-2.34.new/Imakefile
--- xrmap-2.34.orig/Imakefile	2009-01-11 13:26:33.000000000 +0100
+++ xrmap-2.34.new/Imakefile	2009-01-11 13:59:49.000000000 +0100
@@ -1,29 +1,25 @@
 XCOMM Install to /usr or /usr/local  (or something else) ??
-DESTDIR=/usr
+DESTDIR=/usr/
 
 XCOMM Name of man subdirectory of DESTDIR:
-MANDIR=/X11R6/man/man1
+MANDIR=share/man/man1
 
 XCOMM Name of bin subdirectory of DESTDIR: /X11R6/bin or maybe just  /bin
-BINDIR=/X11R6/bin
+BINDIR=bin
 
 XCOMM Name of share subdirectory of xrmap:
-SHAREDIR=/share/xrmap
+SHAREDIR=share/xrmap
 
 XCOMM Name of library subdirectory of DESTDIR:
-EMXBINDIR=$(SHAREDIR)/bin
+EMXBINDIR=bin
 
 XCOMM Name of DOCDIR's subdirectories of DESTDIR:
-DOCDIR=/X11R6/lib/X11/doc 
-DOCHTMLDIR=/X11R6/lib/X11/doc/html
+DOCDIR=doc 
+DOCHTMLDIR=$(DOCDIR)/html
 
 XCOMM Do you want to use the libxpm bug work around ? Comment out if not
 XPMWORKAROUND=-DXPMBUG
 
-XCOMM Do you want to use GPS tracking
-#define DOGPS=-DDOGPS
-
-
 XCOMM Normally -- except for new architectures -- you should be done there.
 
 XCOMM Compile options:
@@ -45,19 +41,12 @@
 
 XCOMM Solaris, Linux
 ZLIBDIR=-L/usr/lib 
-CCOPTIONS=-O $(XPMWORKAROUND) -DZLIB -DSHAREDIR=\"$(DESTDIR)$(SHAREDIR)\" -DEMXBINDIR=\"$(DESTDIR)$(EMXBINDIR)\" $(DOGPS)
+CCOPTIONS=-O $(XPMWORKAROUND) -DZLIB -DSHAREDIR=\"$(DESTDIR)$(SHAREDIR)\" -DEMXBINDIR=\"$(DESTDIR)$(EMXBINDIR)\" 
 #endif
 
-#ifdef DOGPS
-SRCS=gpstrack.c explorer.c image.c xrmap.c
-OBJS=gpstrack.o explorer.o image.o xrmap.o
-GPSLIB=-lgps
-#else
-SRCS=explorer.c image.c xrmap.c
-OBJS=explorer.o image.o xrmap.o
-#endif
 
-LOCAL_LIBRARIES=$(XLIB) $(XPMLIB) $(GPSLIB) $(ZLIBDIR) -ljpeg -lpng -lz -lm 
+
+LOCAL_LIBRARIES=$(XLIB) $(XPMLIB) $(ZLIBDIR) -ljpeg -lpng -lz -lm
 DEPLIBS=		# I don't know what I'm doing, and I hate imake!
 
 numdefs.h:
diff -aur xrmap-2.34.orig/Makefile.noimake xrmap-2.34.new/Makefile.noimake
--- xrmap-2.34.orig/Makefile.noimake	2009-01-11 13:26:33.000000000 +0100
+++ xrmap-2.34.new/Makefile.noimake	2009-01-11 15:16:06.000000000 +0100
@@ -1,42 +1,30 @@
-DESTDIR=/usr
+DESTDIR=/usr/
 X11DIR=$(DESTDIR)/X11R6
 SHAREDIR=/share/xrmap
-EMXBINDIR=$(SHAREDIR)/bin
+EMXBINDIR=bin
 
 PS_VIEWER="gv -noantialias"
 IM_VIEWER="display -background gray84"
-HTML_VIEWER="dillo"
+HTML_VIEWER="firefox"
 PRINT_CMD="lpr -Plocal"
-EDITOR=$(DESTDIR)$(SHAREDIR)"/bin/emx"
+EDITOR=$(DESTDIR)/bin/emx
 MIDIPLAYER="timidity -ig"
 SVG_VIEWER="inkscape"
 SVG_CONVERT="inkscape %s --export-height=%d --export-png=%s"
 
-# Do you want to use GPS tracking
-# #define DOGPS=-DDOGPS
-
-#ifdef DOGPS
-  GPSLIB=-lgps
-  GPSTRACK=gpstrack.o
-#endif
 
 CCOPTIONS=-O6 -g -Wall -DZLIB -DXPMBUG -DSHAREDIR=\"$(DESTDIR)$(SHAREDIR)\" \
   -DPS_VIEWER=\"$(PS_VIEWER)\" -DIM_VIEWER=\"$(IM_VIEWER)\" \
   -DSVG_VIEWER=\"$(SVG_VIEWER)\" -DSVG_CONVERT=\"$(SVG_CONVERT)\" \
   -DHTML_VIEWER=\"$(HTML_VIEWER)\" -DPRINT_CMD=\"$(PRINT_CMD)\" \
   -DEDITOR=\"$(EDITOR)\" -DMIDIPLAYER=\"$(MIDIPLAYER)\" \
-  -DEMXBINDIR=\"$(DESTDIR)$(EMXBINDIR)\" $(DOGPS)
+  -DEMXBINDIR=\"$(DESTDIR)$(EMXBINDIR)\" 
   
 LDOPTIONS=-O6 -g -Wall  -lXpm -lX11 $(GPSLIB) -ljpeg -lz -lpng -lm -L/usr/X11R6/lib
 ## LDOPTIONS=-O6 -g -Wall libxpm/libXpm.a -lX11 $(GPSLIB) -ljpeg -lz -lpng -lm -L/usr/X11R6/lib
 
 all: $(GPSTRACK) explorer.o image.o xrmap.o
-	gcc $(LDOPTIONS) -o xrmap $(GPSTRACK) explorer.o image.o xrmap.o
-	cd editkit ; make -f Makefile.kit emx \
-	    DESTDIR=$(DESTDIR) \
-	    BINDIR=$(EMXBINDIR) \
-	    MANDIR=$(SHAREDIR)/editkit/man \
-	    SHAREDIR=$(SHAREDIR)/editkit/rc	
+	gcc $(LDOPTIONS) -o xrmap explorer.o image.o xrmap.o
 	cd earthview ; make DESTDIR=$(DESTDIR)	
 	@cat EXTRADATA
 	
@@ -49,14 +37,11 @@
 image.o: image.c
 	gcc $(CCOPTIONS) -c image.c
 
-gpstrack.o: gpstrack.h gpstrack.c
-	gcc $(CCOPTIONS) -c gpstrack.c
-	
 xrmap.o: xrmap.h version.h bitmaps.h numdefs.h menudefs.h gpstrack.h xrmap.c
 	gcc $(CCOPTIONS) -c xrmap.c
 
 install:
-	install -c -s xrmap $(X11DIR)/bin
+	install -c -s xrmap $(DESTDIR)/bin
 	mkdir -p $(DESTDIR)$(SHAREDIR)
 	cp -f Xrmaprc Locations $(DESTDIR)$(SHAREDIR)
 	./tools/extprogs >> $(DESTDIR)$(SHAREDIR)/Xrmaprc	
@@ -78,15 +63,7 @@
 	cp -f earthview/README $(DESTDIR)$(SHAREDIR)/doc/README.earthview
 	cd rc ; rm -f *~
 	cp -fR rc $(DESTDIR)$(SHAREDIR)
-	cp -f xrmap.man $(X11DIR)/man/man1/xrmap.1x
-	if [ -r $(X11DIR)/xrmap.1x.gz ] ; \
-	   then gzip -f $(DESTDIR)$(MANDIR)/xrmap.1x ; fi
-	cd editkit ; make -f Makefile.kit emx \
-	    DESTDIR=$(DESTDIR) \
-	    BINDIR=$(EMXBINDIR) \
-	    MANDIR=$(SHAREDIR)/editkit/man \
-	    SHAREDIR=$(SHAREDIR)/editkit/rc ; \
-	make -f Makefile.kit install DESTDIR=$(DESTDIR)
+	cp -f xrmap.man $(DESTDIR)/share/man/man1x/xrmap.1x
 	cd earthview ; make install DESTDIR=$(DESTDIR)	
 	@cat EXTRADATA
 
diff -aur xrmap-2.34.orig/earthview/Makefile xrmap-2.34.new/earthview/Makefile
--- xrmap-2.34.orig/earthview/Makefile	2009-01-11 13:26:33.000000000 +0100
+++ xrmap-2.34.new/earthview/Makefile	2009-01-11 15:00:23.000000000 +0100
@@ -2,7 +2,7 @@
 DESTDIR=/usr
 #fi
 
-BINDIR=$(DESTDIR)/X11R6/bin
+BINDIR=$(DESTDIR)/bin
 DATADIR=$(DESTDIR)/share/xrmap/earthdata
 
 all: earthview
@@ -13,7 +13,7 @@
 install:
 	mkdir -p $(BINDIR)
 	install -c -s earthview $(BINDIR)
-	cp -f README $(DESTDIR)/share/xrmap/doc/README.earthview
+	cp -f README $(DESTDIR)/share/doc/xrmap/README.earthview
 
 help: earthview
 	./earthview -help
diff -aur xrmap-2.34.orig/editkit/Imakefile xrmap-2.34.new/editkit/Imakefile
--- xrmap-2.34.orig/editkit/Imakefile	2009-01-11 13:26:33.000000000 +0100
+++ xrmap-2.34.new/editkit/Imakefile	2009-01-11 14:06:07.000000000 +0100
@@ -5,25 +5,25 @@
 
 XCOMM Name of man subdirectory of DESTDIR:
   #ifndef MANDIR
-    MANDIR=/X11R6/man/man1
+    MANDIR=/usr/share/man/man1
   #endif
 
 XCOMM Name of bin subdirectory of DESTDIR: /X11R6/bin or maybe just  /bin
   #ifndef BINDIR
-    BINDIR=/X11R6/bin
+    BINDIR=/bin
   #endif
 
 XCOMM Name of share subdirectory of DESTDIR_XRMAP:
   #ifndef SHAREDIR
-    SHAREDIR=$(DESTDIR)/share/editkit
+    SHAREDIR=$(DESTDIR)/usr/share/xrmap/editkit
   #endif
 
 XCOMM Name of DOCDIR's subdirectories of DESTDIR:
   #ifndef DOCDIR
-    DOCDIR=/X11R6/lib/X11/doc
+    DOCDIR=$(DESTDIR)/usr/share/doc/xrmap
   #endif
   #ifndef DOCHTMLDIR
-    DOCHTMLDIR=/X11R6/lib/X11/doc/html
+    DOCHTMLDIR=$(DESTDIR)/usr/share/doc/xrmap/html
   #endif
 
 XCOMM
diff -aur xrmap-2.34.orig/editkit/Makefile.kit xrmap-2.34.new/editkit/Makefile.kit
--- xrmap-2.34.orig/editkit/Makefile.kit	2009-01-11 13:26:33.000000000 +0100
+++ xrmap-2.34.new/editkit/Makefile.kit	2009-01-11 14:52:25.000000000 +0100
@@ -14,7 +14,7 @@
 # TLL for Terry Loveall personal settings
 
 #ifndef SHAREDIR
-SHAREDIR=/usr/share/editkit
+SHAREDIR=/usr/share/xrmap/editkit
 #endif
 
 CFLAGS = -O2 -fomit-frame-pointer -pipe -Wall -DTHREED -DGREEK -DJUSTIFY -DSHAREDIR="\"$(SHAREDIR)\""
@@ -60,21 +60,7 @@
 
 emx.o: Makefile version.h messages.def x11.c edit.h edit.c x11_bind_em.c
 	gcc $(CFLAGS) -DEMACS -o emx.o -c x11.c
-
-install:
-	if [ -x edt ]; then cp -f edt $(BINDIR); fi
-	if [ -x emt ]; then cp -f emt $(BINDIR); fi
-	if [ -x edx ]; then cp -f edx $(BINDIR); fi
-	if [ -x emx ]; then cp -f emx $(BINDIR); fi
-	mkdir -p $(SHAREDIR)
-	cp -f rc.example $(SHAREDIR)/rc.common
-	cp -f README $(SHAREDIR)
-	if [ -x edt ] || [ -x edx ]; then cp -f MANUAL.wo*ar $(SHAREDIR); fi
-	if [ -x emt ] || [ -x emx ]; then cp -f MANUAL.emacs $(SHAREDIR); fi
-	if [ -x edt ]; then cd $(SHAREDIR); ln -sf rc.common edtrc; fi 
-	if [ -x emt ]; then cd $(SHAREDIR); ln -sf rc.common emtrc; fi 
-	if [ -x edx ]; then cd $(SHAREDIR); ln -sf rc.common edxrc; fi 
-	if [ -x emx ]; then cd $(SHAREDIR); ln -sf rc.common emxrc; fi 
+	ln -sf rc.common emxrc
 
 clean:
 	-rm -f *.o *~ edt emt edx emx
diff -aur xrmap-2.34.orig/version.h xrmap-2.34.new/version.h
--- xrmap-2.34.orig/version.h	2009-01-11 13:26:33.000000000 +0100
+++ xrmap-2.34.new/version.h	2009-01-11 15:22:19.000000000 +0100
@@ -2,5 +2,5 @@
 
 #define PACKAGE "xrmap"
 #define VERSION "2.34"
-#define DATEPKG "March 13, 2005"
+#define DATEPKG "August 21, 2008"
 #define COPYRIGHT "(C) Jean-Pierre Demailly, licensed under the GPL"
