diff -ru iptraf-3.0.1-orig/src/hostmon.c iptraf-3.0.1/src/hostmon.c
--- iptraf-3.0.1-orig/src/hostmon.c	2013-05-10 14:48:59.120643355 +1000
+++ iptraf-3.0.1/src/hostmon.c	2013-05-10 14:49:04.560383687 +1000
@@ -32,7 +32,7 @@
 #include <linux/if_ether.h>
 #include <linux/netdevice.h>
 #include <linux/if_fddi.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <net/if_arp.h>
 #include <stdlib.h>
 #include <time.h>
diff -ru iptraf-3.0.1-orig/src/othptab.c iptraf-3.0.1/src/othptab.c
--- iptraf-3.0.1-orig/src/othptab.c	2013-05-10 14:48:59.120643355 +1000
+++ iptraf-3.0.1/src/othptab.c	2013-05-10 14:49:04.580382734 +1000
@@ -20,7 +20,7 @@
 #include <asm/types.h>
 #include <sys/socket.h>
 #include <linux/if_ether.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <linux/netdevice.h>
 #include <linux/if_fddi.h>
 #include <winops.h>
diff -ru iptraf-3.0.1-orig/src/packet.c iptraf-3.0.1/src/packet.c
--- iptraf-3.0.1-orig/src/packet.c	2013-05-10 14:48:59.120643355 +1000
+++ iptraf-3.0.1/src/packet.c	2013-05-10 14:49:04.580382734 +1000
@@ -36,7 +36,7 @@
 #include <linux/if_ether.h>
 #include <linux/netdevice.h>
 #include <linux/if_fddi.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <linux/isdn.h>
 #include <linux/sockios.h>
 #include <msgboxes.h>
diff -ru iptraf-3.0.1-orig/src/tcptable.h iptraf-3.0.1/src/tcptable.h
--- iptraf-3.0.1-orig/src/tcptable.h	2013-05-10 14:48:59.120643355 +1000
+++ iptraf-3.0.1/src/tcptable.h	2013-05-10 14:49:04.590382256 +1000
@@ -24,7 +24,7 @@
 #include <linux/if_ether.h>
 #include <linux/netdevice.h>
 #include <linux/if_fddi.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 // #include <net/if.h>
 #include <netinet/ip.h>
 #include <netinet/udp.h>
diff -ru iptraf-3.0.1-orig/src/tr.c iptraf-3.0.1/src/tr.c
--- iptraf-3.0.1-orig/src/tr.c	2013-05-10 14:48:59.120643355 +1000
+++ iptraf-3.0.1/src/tr.c	2013-05-10 14:49:04.590382256 +1000
@@ -7,7 +7,7 @@
  */
 
 #include <asm/types.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <netinet/in.h>
 
 unsigned int get_tr_ip_offset(unsigned char *pkt)
