diff --git a/gcc/d/patches/patch-gcc-4.8.x b/gcc/d/patches/patch-gcc-4.8.x
index 0968e32..0787e19 100644
--- a/gcc/d/patches/patch-gcc-4.8.x
+++ b/gcc/d/patches/patch-gcc-4.8.x
@@ -1,5 +1,5 @@
---- gcc-4.8-20120617/gcc/config/i386/i386.c	2012-06-13 18:46:59.000000000 +0100
-+++ gcc-4.8/gcc/config/i386/i386.c	2012-06-18 20:40:41.882589307 +0100
+--- gcc/config/i386/i386.c	2012-06-13 18:46:59.000000000 +0100
++++ gcc/config/i386/i386.c	2012-06-18 20:40:41.882589307 +0100
 @@ -5441,6 +5441,14 @@ ix86_return_pops_args (tree fundecl, tre
  
    return 0;
@@ -78,8 +78,8 @@
    /* End element.  */
    { NULL,        0, 0, false, false, false, NULL, false }
  };
---- gcc-4.8-20120617/gcc/config/rs6000/rs6000.c	2012-06-17 22:08:39.000000000 +0100
-+++ gcc-4.8/gcc/config/rs6000/rs6000.c	2012-06-18 20:40:42.126589298 +0100
+--- gcc/config/rs6000/rs6000.c	2012-06-17 22:08:39.000000000 +0100
++++ gcc/config/rs6000/rs6000.c	2012-06-18 20:40:42.126589298 +0100
 @@ -21296,7 +21296,8 @@ rs6000_output_function_epilogue (FILE *f
  	 either, so for now use 0.  */
        if (! strcmp (language_string, "GNU C")
@@ -90,8 +90,8 @@
  	i = 0;
        else if (! strcmp (language_string, "GNU F77")
  	       || ! strcmp (language_string, "GNU Fortran"))
---- gcc-4.8-20120617/gcc/doc/frontends.texi	2011-01-03 20:52:22.000000000 +0000
-+++ gcc-4.8/gcc/doc/frontends.texi	2012-04-22 17:09:57.525883721 +0100
+--- gcc/doc/frontends.texi	2011-01-03 20:52:22.000000000 +0000
++++ gcc/doc/frontends.texi	2012-04-22 17:09:57.525883721 +0100
 @@ -10,6 +10,7 @@
  @cindex GNU Compiler Collection
  @cindex GNU C Compiler
@@ -109,8 +109,8 @@
  
  The abbreviation @dfn{GCC} has multiple meanings in common use.  The
  current official meaning is ``GNU Compiler Collection'', which refers
---- gcc-4.8-20120617/gcc/doc/install.texi	2012-05-29 15:14:06.000000000 +0100
-+++ gcc-4.8/gcc/doc/install.texi	2012-06-18 20:39:45.058591380 +0100
+--- gcc/doc/install.texi	2012-05-29 15:14:06.000000000 +0100
++++ gcc/doc/install.texi	2012-06-18 20:39:45.058591380 +0100
 @@ -1360,12 +1360,12 @@ their runtime libraries should be built.
  grep language= */config-lang.in
  @end smallexample
@@ -126,8 +126,8 @@
  
  @item --enable-stage1-languages=@var{lang1},@var{lang2},@dots{}
  Specify that a particular subset of compilers and their runtime
---- gcc-4.8-20120617/gcc/doc/invoke.texi	2012-06-15 10:22:00.000000000 +0100
-+++ gcc-4.8/gcc/doc/invoke.texi	2012-06-18 20:40:42.598589280 +0100
+--- gcc/doc/invoke.texi	2012-06-15 10:22:00.000000000 +0100
++++ gcc/doc/invoke.texi	2012-06-18 20:40:42.598589280 +0100
 @@ -1137,6 +1137,15 @@ called @dfn{specs}.
  Ada source code file containing a library unit body (a subprogram or
  package body).  Such files are also called @dfn{bodies}.
@@ -152,8 +152,8 @@
  f77  f77-cpp-input f95  f95-cpp-input
  go
  java
---- gcc-4.8-20120617/gcc/doc/sourcebuild.texi	2012-03-15 12:25:47.000000000 +0000
-+++ gcc-4.8/gcc/doc/sourcebuild.texi	2012-04-22 17:25:20.189850056 +0100
+--- gcc/doc/sourcebuild.texi	2012-03-15 12:25:47.000000000 +0000
++++ gcc/doc/sourcebuild.texi	2012-04-22 17:25:20.189850056 +0100
 @@ -104,6 +104,9 @@ dereferencing operations.
  @item libobjc
  The Objective-C and Objective-C++ runtime library.
@@ -164,8 +164,8 @@
  @item libssp
  The Stack protector runtime library.
  
---- gcc-4.8-20120617/gcc/doc/standards.texi	2011-12-21 17:53:58.000000000 +0000
-+++ gcc-4.8/gcc/doc/standards.texi	2012-04-22 17:11:38.553880036 +0100
+--- gcc/doc/standards.texi	2011-12-21 17:53:58.000000000 +0000
++++ gcc/doc/standards.texi	2012-04-22 17:11:38.553880036 +0100
 @@ -289,6 +289,16 @@ a specific version.  In general GCC trac
  closely, and any given release will support the language as of the
  date that the release was frozen.
@@ -183,8 +183,8 @@
  @section References for other languages
  
  @xref{Top, GNAT Reference Manual, About This Guide, gnat_rm,
---- gcc-4.8-20120617/gcc/dwarf2out.c	2012-06-17 22:08:39.000000000 +0100
-+++ gcc-4.8/gcc/dwarf2out.c	2012-06-18 20:40:42.726589276 +0100
+--- gcc/dwarf2out.c	2012-06-17 22:08:39.000000000 +0100
++++ gcc/dwarf2out.c	2012-06-18 20:40:42.726589276 +0100
 @@ -18105,6 +18105,8 @@ gen_compile_unit_die (const char *filena
    language = DW_LANG_C89;
    if (strcmp (language_string, "GNU C++") == 0)
@@ -194,8 +194,8 @@
    else if (strcmp (language_string, "GNU F77") == 0)
      language = DW_LANG_Fortran77;
    else if (strcmp (language_string, "GNU Pascal") == 0)
---- gcc-4.8-20120617/gcc/gcc.c	2012-06-01 08:55:39.000000000 +0100
-+++ gcc-4.8/gcc/gcc.c	2012-06-18 20:39:46.586591323 +0100
+--- gcc/gcc.c	2012-06-01 08:55:39.000000000 +0100
++++ gcc/gcc.c	2012-06-18 20:39:46.586591323 +0100
 @@ -935,6 +935,7 @@ static const struct compiler default_com
    {".java", "#Java", 0, 0, 0}, {".class", "#Java", 0, 0, 0},
    {".zip", "#Java", 0, 0, 0}, {".jar", "#Java", 0, 0, 0},
