diff --git a/Makefile b/Makefile
index 53c3c07..b70556f 100644
--- a/Makefile
+++ b/Makefile
@@ -3,6 +3,7 @@ obj-m += sony-laptop.o
 obj-m += test.o
 
 KDIR := /lib/modules/$(shell uname -r)
+KDIR_CLEAN := lib/modules/$(shell uname -r)
 ifdef SYSSRC
  KERNEL_SOURCES := $(SYSSRC)
 else
@@ -26,16 +27,12 @@ clean:
 	@rm -f *~
 
 install:
-	mkdir -p $(KDIR)/updates/
-	cp sony-laptop.ko $(KDIR)/updates/
-	#cp nvidia_bl.ko $(KDIR)/updates/
-	depmod -a
+	mkdir -p $(KDIR_CLEAN)/updates/
+	cp sony-laptop.ko $(KDIR_CLEAN)/updates/
 
 uninstall:
 	rm $(KDIR)/updates/sony-laptop.ko
-	#rm $(KDIR)/updates/nvidia_bl.ko
 	rmmod sony-laptop
-	#rmmod nvidia_bl
 	depmod -a
 	modprobe sony-laptop 
 
