diff -Npur src.orig//tree.hh src/tree.hh
--- src.orig//tree.hh	2011-04-25 19:29:48.000000000 +0000
+++ src/tree.hh	2011-04-25 19:29:56.000000000 +0000
@@ -30,7 +30,7 @@
 #include <set>
 #include <queue>
 #include <algorithm>
-
+#include <cstddef>
 
 /// A node in the tree, combining links to other nodes as well as the actual data.
 template<class T>
diff -Npur src.orig//youngtab.hh src/youngtab.hh
--- src.orig//youngtab.hh	2011-04-25 19:35:02.000000000 +0000
+++ src/youngtab.hh	2011-04-25 19:35:12.000000000 +0000
@@ -33,7 +33,7 @@
 #include <list>
 #include <gmpxx.h>
 #include "combinatorics.hh"
-
+#include <cstddef>
 typedef mpz_class yngint_t;
 typedef mpq_class yngrat_t;
 
