--- src/unstable/src/Drawing/PanelRenderer.vala	2015-01-17 02:29:36.817351000 +0900
+++ src/unstable/src/Drawing/PanelRenderer.vala.new	2015-01-17 02:30:03.879653878 +0900
@@ -67,7 +67,7 @@
 
         }
 
-        ~DockRenderer ()
+        ~PanelRenderer ()
         {
             manager.hide_manager.notify["Hidden"].disconnect (hidden_changed);
             manager.settings.notify["slim-panel-edge"].disconnect (manager.window.queue_draw);
@@ -204,4 +204,4 @@
         }
 
     }
-}
\ No newline at end of file
+}
