--- src/main.h
+++ src/main.h
@@ -7,12 +7,15 @@
 #include "prozilla.h"
 
 /* Gettext */
+#if !defined COMMON_H // prozilla/src/common.h contains definitions below. This is included in
+                      // netrc.h. netrc.h is included in prozilla.h.
 #include <libintl.h>
 #define _(String) dgettext (PACKAGE,String)
 #define gettext_noop(String) (String)
 #ifndef HAVE_GNOME
 #define N_(String) gettext_noop (String)
 #endif
+#endif
 /* Gettext */
 
 
