--- configure.in
+++ configure.in
@@ -6,7 +6,6 @@
 
 AM_CONFIG_HEADER(config.h)
 
-AC_LANG_CPLUSPLUS
 AC_ISC_POSIX     
 
 AC_CONFIG_SUBDIRS(libprozilla)
@@ -38,6 +37,7 @@
 dnl -------------
 ALL_LINGUAS="pt_BR nl ro it fr"
 AM_GNU_GETTEXT
+AM_GNU_GETTEXT_VERSION(0.12.1)
   
 
 dnl Checks for typedefs, structures, and compiler characteristics.
@@ -153,5 +153,5 @@
 fi
 			
 
-CXXFLAGS="-Wall -ggdb -D_REENTRANT"
 AC_OUTPUT(Makefile src/Makefile man/Makefile intl/Makefile po/Makefile.in)
+CXXFLAGS="-Wall -ggdb -D_REENTRANT $CXXFLAGS"
--- libprozilla/configure.in
+++ libprozilla/configure.in
@@ -51,6 +51,7 @@
 ALL_LINGUAS="pt_BR ro nl it fr"
 
 AM_GNU_GETTEXT
+AM_GNU_GETTEXT_VERSION(0.12.1)
 
 dnl ---------------------------------------------------------------------------
 dnl Checks for libraries.
--- libprozilla/src/Makefile.am
+++ libprozilla/src/Makefile.am
@@ -36,5 +36,5 @@
 
 includedir = $(prefix)/include
 include_HEADERS = prozilla.h netrc.h
-CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\" -ggdb 
+AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\"
 LIBS            = @LIBS@ $(THREAD_LIBS)
--- src/Makefile.am
+++ src/Makefile.am
@@ -4,12 +4,11 @@
 
 bin_PROGRAMS    = prozgui
 
-CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\"
+AM_CPPFLAGS     = -DLOCALEDIR=\"$(datadir)/locale\"
 
 prozgui_SOURCES = main.cpp download_win.cpp  Fl_Progress.cxx we.cxx init.cpp options.cpp dl_win.cxx prefs.cpp ftps_win.cxx  ftpsearch_win.cpp getopt.c savefile.cpp
 INCLUDES        = -I. -I..  -I../libprozilla/src -I@includedir@ 
-CFLAGS          = @CFLAGS@ -D_REENTRANT -Wall -ggdb
-CFLAGS         += -DGLOBAL_CONF_FILE="\"@sysconfdir@/prozilla.conf\"" -DLOCALEDIR=\"$(datadir)/locale\"
-LDFLAGS		= @LDFLAGS@ 
+AM_CFLAGS       = -D_REENTRANT -Wall
+AM_CFLAGS       += -DGLOBAL_CONF_FILE="\"@sysconfdir@/prozilla.conf\"" -DLOCALEDIR=\"$(datadir)/locale\"
 LDADD           = ../libprozilla/src/libprozilla.la  -L@libdir@  -L/usr/X11R6/lib  -lfltk 
 LIBS            = @LIBS@ $(THREAD_LIBS) 
