diff --git a/infrastructure/bin/comp.sh b/infrastructure/bin/comp.sh
index 5f65c2f..63378d4 100755
--- a/infrastructure/bin/comp.sh
+++ b/infrastructure/bin/comp.sh
@@ -183,7 +183,7 @@ echo "copying files..."
 cp net.appjet.ajstdlib/streaming-client.js build/net/appjet/ajstdlib/
 if [ $OBFUSC ] ; then
     echo obfuscating...
-    scala -classpath $CP:. net.appjet.bodylock.compressor \
+    $SCALA -classpath $CP:. net.appjet.bodylock.compressor \
 	build/net/appjet/ajstdlib/streaming-client.js
 fi
 
@@ -191,7 +191,7 @@ cp net.appjet.ajstdlib/streaming-iframe.html build/net/appjet/ajstdlib/
 mkdir -p build/net/appjet/ajstdlib/modules
 
 echo "building javascript classfiles..."
-scala -classpath $CP net.appjet.bodylock.Compiler \
+$SCALA -classpath $CP net.appjet.bodylock.Compiler \
     -destination=build/net/appjet/ajstdlib/ \
     -cutPrefix=framework-src \
     `find framework-src -name '*.js'`
diff --git a/infrastructure/ace/bin/make b/infrastructure/ace/bin/make
index 98a48f4..c667fb4 100755
--- a/infrastructure/ace/bin/make
+++ b/infrastructure/ace/bin/make
@@ -1,7 +1,7 @@
 #!/bin/sh
 mkdir -p ../../etherpad/src/etherpad/collab/ace
 mkdir -p ../../etherpad/src/static/js
-exec scala -classpath lib/yuicompressor-2.4-appjet.jar:lib/rhino-js-1.7r1.jar $0 $@
+exec $SCALA -classpath lib/yuicompressor-2.4-appjet.jar:lib/rhino-js-1.7r1.jar $0 $@
 !#
 
 import java.io._;
diff --git a/infrastructure/ace/bin/serve b/infrastructure/ace/bin/serve
index e02e042..59d4fc1 100755
--- a/infrastructure/ace/bin/serve
+++ b/infrastructure/ace/bin/serve
@@ -1,5 +1,5 @@
 #!/bin/bash
-scala -nocompdaemon -Dlog4j.mortbay.loglevel=WARN -classpath lib/jetty-6.1.7.jar:lib/jetty-util-6.1.7.jar:lib/servlet-api-2.5-6.1.3.jar $0 $@ &
+$SCALA -nocompdaemon -Dlog4j.mortbay.loglevel=WARN -classpath lib/jetty-6.1.7.jar:lib/jetty-util-6.1.7.jar:lib/servlet-api-2.5-6.1.3.jar $0 $@ &
 exit
 !#
 
