diff --git a/Collisions/src/OgreBulletCollisionsRay.cpp b/Collisions/src/OgreBulletCollisionsRay.cpp
index 73213d3..6d34e8d 100644
--- a/Collisions/src/OgreBulletCollisionsRay.cpp
+++ b/Collisions/src/OgreBulletCollisionsRay.cpp
@@ -84,7 +84,11 @@ namespace OgreBulletCollisions
     // -------------------------------------------------------------------------
     Object  *CollisionClosestRayResultCallback::getCollidedObject () const
     {        
+    #if BT_BULLET_VERSION>=281
+        return mWorld->findObject(const_cast<btCollisionObject*>(static_cast<btCollisionWorld::ClosestRayResultCallback *> (mRayResultCallback)->m_collisionObject));
+    #else
         return mWorld->findObject(static_cast<btCollisionWorld::ClosestRayResultCallback *> (mRayResultCallback)->m_collisionObject);
+    #endif
 	}
     // -------------------------------------------------------------------------
 	CollisionClosestRayResultCallback::CollisionClosestRayResultCallback(const Ogre::Ray &ray, CollisionsWorld *world, Ogre::Real max_distance) :
diff --git a/Collisions/src/OgreBulletCollisionsWorld.cpp b/Collisions/src/OgreBulletCollisionsWorld.cpp
index e16a73b..a41fe2e 100644
--- a/Collisions/src/OgreBulletCollisionsWorld.cpp
+++ b/Collisions/src/OgreBulletCollisionsWorld.cpp
@@ -223,8 +223,8 @@ void CollisionsWorld::discreteCollide()
 	{
 		btPersistentManifold* contactManifold = mWorld->getDispatcher()->getManifoldByIndexInternal(i);
 
-		btCollisionObject* obA = static_cast<btCollisionObject*>(contactManifold->getBody0());
-		btCollisionObject* obB = static_cast<btCollisionObject*>(contactManifold->getBody1());
+        btCollisionObject* obA = const_cast<btCollisionObject*>(contactManifold->getBody0());
+        btCollisionObject* obB = const_cast<btCollisionObject*>(contactManifold->getBody1());
 
 		contactManifold->refreshContactPoints(obA->getWorldTransform(),obB->getWorldTransform());
 
diff --git a/Dynamics/src/OgreBulletDynamicsWorld.cpp b/Dynamics/src/OgreBulletDynamicsWorld.cpp
index eb55069..4b5d9a9 100644
--- a/Dynamics/src/OgreBulletDynamicsWorld.cpp
+++ b/Dynamics/src/OgreBulletDynamicsWorld.cpp
@@ -108,8 +108,13 @@ namespace OgreBulletDynamics
 			{
 				btPersistentManifold* contactManifold = mWorld->getDispatcher()->getManifoldByIndexInternal(i);
 
-				btCollisionObject* obA = static_cast<btCollisionObject*>(contactManifold->getBody0());
-				btCollisionObject* obB = static_cast<btCollisionObject*>(contactManifold->getBody1());
+                #if BT_BULLET_VERSION>=281
+                    btCollisionObject* obA = const_cast<btCollisionObject*>(contactManifold->getBody0());; // For newer Bullet versions
+                    btCollisionObject* obB = const_cast<btCollisionObject*>(contactManifold->getBody1()); // For newer Bullet versions
+                #else
+                    btCollisionObject* obA = static_cast<btCollisionObject*>(contactManifold->getBody0()); // For older Bullet versions (original code)
+                    btCollisionObject* obB = static_cast<btCollisionObject*>(contactManifold->getBody1()); // For older Bullet versions (original code)
+                #endif
 
 				contactManifold->refreshContactPoints(obA->getWorldTransform(),obB->getWorldTransform());
 
