# Makes the build work even if _JAVA_OPTIONS exists
--- a/build.gradle	Sat Nov 01 22:54:08 2014 +0300
+++ b/build.gradle	Tue Nov 11 23:58:28 2014 +0000
@@ -726,6 +726,9 @@
 try {
     if (inStream.readLine() != null) {
         String v = inStream.readLine();
+        while (!v.contains(" Runtime")) {
+            v = inStream.readLine();
+        }
         if (v != null) {
             int ib = v.indexOf(" (build ");
             if (ib != -1) {
