--- pdf_image_new.vala	2010-10-25 10:59:08.634657461 +0200
+++ classes/pdf_image.vala	2010-02-18 20:29:47.000000000 +0100
@@ -247,8 +247,8 @@
         protected Pixmap render_page( int page_number ) {
             var background_pixmap = new Pixmap( null, this.scaled_width, this.scaled_height, 24 );
             var gc = new GC( background_pixmap );
-            Gdk.Color white;
-            Gdk.Color.parse( "white", out white );
+            Color white;
+            Color.parse( "white", out white );
             gc.set_rgb_fg_color( white );
             background_pixmap.draw_rectangle( gc, true, 0, 0, this.scaled_width, this.scaled_height );
 
