--- configure.o	2012-01-12 15:59:03.891292010 +0100
+++ configure	2012-01-12 15:59:59.914628238 +0100
@@ -16699,7 +16699,7 @@
     dir="$1"
     if test -z "$dir" -o "$dir" = "/"; then
         dir=/
-	lib=/lib
+	lib=/usr/lib
 	inc=/usr/include
     else
         lib="$dir/lib${LIBSUFFIX}"
@@ -16799,7 +16799,7 @@
     dir="$1"
     if test -z "$dir" -o "$dir" = "/"; then
         dir="/"
-	lib=/lib
+	lib=/usr/lib
 	inc=/usr/include
     else
         lib="$dir/lib${LIBSUFFIX}"
@@ -16891,7 +16891,7 @@
     dir="$1"
     if test -z "$dir" -o "$dir" = "/"; then
         dir="/"
-	lib=/lib
+	lib=/usr/lib
 	inc=/usr/include
     else
         lib="$dir/lib${LIBSUFFIX}"
@@ -16972,7 +16972,7 @@
     dir="$1"
     if test -z "$dir" -o "$dir" = "/"; then
         dir="/"
-	lib=/lib
+	lib=/usr/lib
 	inc=/usr/include
     else
         lib="$dir/lib${LIBSUFFIX}"
