--- QoreLib.cpp.o	2012-01-12 16:08:17.401320607 +0100
+++ QoreLib.cpp	2012-01-12 16:08:28.037987825 +0100
@@ -46,7 +46,7 @@
 int qore_version_major               = VERSION_MAJOR;
 int qore_version_minor               = VERSION_MINOR;
 int qore_version_sub                 = VERSION_SUB;
-int qore_build_number                = BUILD;
+int qore_build_number                = 1;
 int qore_target_bits                 = TARGET_BITS;
 const char *qore_target_os           = TARGET_OS;
 const char *qore_target_arch         = TARGET_ARCH;
