diff -uNr boinc/Makefile.am boinc-build/Makefile.am
--- boinc/Makefile.am	2011-09-16 16:58:37.000000000 +0900
+++ boinc-build/Makefile.am	2011-09-16 17:51:21.000000000 +0900
@@ -22,9 +22,6 @@
 
 if ENABLE_MANAGER
    CLIENTGUI_SUBDIRS = clientgui clientgui/res locale
-if BUILD_X11_SCREENSAVER
-   CLIENTGUI_SUBDIRS += clientscr
-endif
 endif
 
 # ORDER MATTERS below.  One must build dependencies FIRST, then things
diff -uNr boinc/configure.ac boinc-build/configure.ac
--- boinc/configure.ac	2011-09-16 16:58:37.000000000 +0900
+++ boinc-build/configure.ac	2011-09-16 17:50:36.000000000 +0900
@@ -901,7 +901,6 @@
                  apps/Makefile
                  clientgui/Makefile
                  clientgui/res/Makefile
-                 clientscr/Makefile
                  client/Makefile
                  client/win/boinc_path_config.py:py/boinc_path_config.py.in
                  client/scripts/Makefile
