--- boinc_core_release_7_0_28/configure.ac	2012-11-04 18:01:29.750741150 +0900
+++ boinc_core_release_7_0_28-build/configure.ac	2012-11-04 19:51:14.501794193 +0900
@@ -550,11 +550,7 @@
 AM_CONDITIONAL(BUILD_GRAPHICS_API, [ test "$have_glut" = yes ])
 
 dnl check for X screen saver lib
-AC_CHECK_LIB([Xss], [XScreenSaverAllocInfo], [have_Xss="yes"], [have_Xss="no"])
-AC_CHECK_HEADER([X11/extensions/scrnsaver.h], [have_Xss="yes"], [have_Xss="no"])
-if test "$have_Xss" == no; then
-    AC_MSG_WARN([libxss missing, disabling X ScreenSaver user idle detection])
-fi
+have_Xss="no"
 
 dnl --------------------------------------------------------------------------------
 dnl put double-inclusion protection into config.h
@@ -901,7 +897,6 @@
                  apps/Makefile
                  clientgui/Makefile
                  clientgui/res/Makefile
-                 clientscr/Makefile
                  client/Makefile
                  client/win/boinc_path_config.py:py/boinc_path_config.py.in
                  client/scripts/Makefile
