diff -aur old/Makefile new/Makefile
--- old/Makefile	2011-11-28 19:38:11.000000000 +0100
+++ new/Makefile	2012-12-27 13:37:39.422906389 +0100
@@ -1,9 +1,5 @@
 all:
 	$(MAKE) -C src all
-	$(MAKE) -C test all
-
-test:
-	$(MAKE) -C test all
 
 install:
 	$(MAKE) -C src install
@@ -21,10 +17,8 @@
 
 clean:
 	$(MAKE) -C src clean
-	$(MAKE) -C test clean
 
 distclean:
 	$(MAKE) -C src distclean
-	$(MAKE) -C test distclean
 
-.PHONY: all test install uninstall clean distclean
+.PHONY: all install uninstall clean distclean
diff -aur old/src/Makefile new/src/Makefile
--- old/src/Makefile	2012-01-11 21:27:41.000000000 +0100
+++ new/src/Makefile	2012-12-27 13:38:49.773576667 +0100
@@ -52,8 +52,8 @@
 
 PREFIX = /usr/local
 
-CFLAGS += -std=gnu99 -Wall -DFAKE_STAT -DFAKE_INTERNAL_CALLS -fPIC -DPOSIX_REALTIME -DLIMITEDFAKING -DSPAWNSUPPORT
-LDFLAGS += -shared -ldl -lm -lpthread
+CFLAGS += -std=gnu99 -Wall -DFAKE_STAT -DFAKE_INTERNAL_CALLS -fPIC -DPOSIX_REALTIME -DLIMITEDFAKING -DSPAWNSUPPORT -m32
+LDFLAGS += -shared -ldl -lm -lpthread -m32
 
 SRC = faketime.c
 OBJ = faketime.o faketimeMT.o
@@ -79,14 +79,14 @@
 
 install: ${LIBS}
 	@echo 
-	@echo "Copying the faketime libraries to ${DESTDIR}${PREFIX}/lib/faketime and the faketime wrapper script to ${DESTDIR}${PREFIX}/bin ..."
-	$(INSTALL) -dm0755 "${DESTDIR}${PREFIX}/lib/faketime/"
-	$(INSTALL) -m0644 ${LIBS} "${DESTDIR}${PREFIX}/lib/faketime/"
+	@echo "Copying the faketime libraries to ${DESTDIR}${PREFIX}/lib32/faketime and the faketime wrapper script to ${DESTDIR}${PREFIX}/bin ..."
+	$(INSTALL) -dm0755 "${DESTDIR}${PREFIX}/lib32/faketime/"
+	$(INSTALL) -m0644 ${LIBS} "${DESTDIR}${PREFIX}/lib32/faketime/"
 	$(INSTALL) -Dm0755 faketime "${DESTDIR}${PREFIX}/bin/faketime"
 
 uninstall:
-	for f in ${LIBS}; do rm -f "${DESTDIR}${PREFIX}/lib/faketime/$$f"; done
-	rmdir "${DESTDIR}${PREFIX}/lib/faketime"
+	for f in ${LIBS}; do rm -f "${DESTDIR}${PREFIX}/lib32/faketime/$$f"; done
+	rmdir "${DESTDIR}${PREFIX}/lib32/faketime"
 	rm -f "${DESTDIR}${PREFIX}/bin/faketime"
 
 .PHONY: all clean distclean install uninstall
