--- ./src/ircptray_main.c.bak	2010-07-07 11:16:10.000000000 -0300
+++ ./src/ircptray_main.c	2010-12-16 20:33:56.143333345 -0300
@@ -533,8 +533,9 @@
 		y = 1;
 		while ( labels[y] ) {
 			ck1 = gtk_check_button_new();
-			GTK_WIDGET_UNSET_FLAGS (ck1, GTK_CAN_FOCUS);
+			gtk_widget_set_can_focus (ck1, FALSE);
 			gtk_widget_set_sensitive (ck1, FALSE);
+			
 
 			if ( (y <= 8) && (dev->hints[0] & 1<<(y-1)) )
 				gtk_toggle_button_set_active( GTK_TOGGLE_BUTTON(ck1), 1 );
@@ -560,7 +561,7 @@
 	close = gtk_button_new_from_stock ("gtk-close");
 	gtk_widget_show (close);
 	gtk_dialog_add_action_widget (GTK_DIALOG (info), close, GTK_RESPONSE_CLOSE);
-	GTK_WIDGET_SET_FLAGS (close, GTK_CAN_DEFAULT);
+	/*GTK_WIDGET_SET_FLAGS (close, GTK_CAN_DEFAULT);*/
 
 	g_signal_connect (GTK_OBJECT (info), "response",
 			    G_CALLBACK (on_dialog_response),
