--- ./src/ircptray_main.c	2011-05-30 13:00:38.000000000 -0300
+++ ircptray_main.c	2011-06-11 17:18:47.000000000 -0300
@@ -248,14 +248,13 @@
 
 	notification = notify_notification_new (g_get_application_name(),
                                             notify_msg,
-                                            icon,
-                                            NULL);
+                                            icon);
     if (notification == NULL) {
         g_warning("failed to setup notification");
         return;
     }
 
-	notify_notification_attach_to_status_icon (notification, AppTrayIcon);
+	/*notify_notification_attach_to_status_icon (notification, AppTrayIcon);*/
 
 	if (!notify_notification_show (notification, NULL))
 	    g_warning("failed to send notification");
@@ -277,11 +276,10 @@
 
 	notification = notify_notification_new (g_get_application_name(),
                                        notify_msg,
-                                       icon,
-                                       NULL);
+                                       icon);
 
 	if (notification != NULL) {
-	    notify_notification_attach_to_status_icon (notification, AppTrayIcon);
+	  /*  notify_notification_attach_to_status_icon (notification, AppTrayIcon);*/
 
 		if (!notify_notification_show (notification, NULL))
 			g_warning("failed to send notification\n");
