--- ./src/recvfile.c	2011-05-30 13:00:38.000000000 -0300
+++ recvfile.c	2011-06-11 17:17:34.000000000 -0300
@@ -255,8 +255,7 @@
 
 				n = notify_notification_new (title,
                                        msg,
-                                       MAIN_ICON,
-                                       GTK_WIDGET(AppTrayIcon));
+                                       MAIN_ICON);
 
 				if (n != NULL) {
 					notify_notification_set_urgency(n, NOTIFY_URGENCY_NORMAL);
