--- ntp-dev-4.2.7p261/scripts/html2man.in	2011-07-17 02:34:49.000000000 -0400
+++ ntp-dev-4.2.7p261.new/scripts/html2man.in	2012-02-28 13:33:04.620147049 -0500
@@ -49,8 +49,9 @@
     "This file was automatically generated from HTML source.\n";
 
 mkdir $MANDIR, 0777;
-mkdir "$MANDIR/man8", 0777;
-mkdir "$MANDIR/man5", 0777;
+foreach $section (keys %{{map { $_->[1] => 1 } values(%manfiles) }}) {
+    mkdir "$MANDIR/man$section", 0777;
+}
 
 # Do the actual processing
 foreach $file (keys %manfiles) {
@@ -67,23 +68,23 @@
 
     $p = HTML::TokeParser->new("$filename.html") || die "Can't open $filename.html: $!";
     $fileout = "$MANDIR/man$fileinfo->[1]/$fileinfo->[0].$fileinfo->[1]";
-    open(MANOUT, ">$fileout")
-	|| die "Can't open: $!";
+    open(my $manout, ">", $fileout)
+	    or die "Can't open $fileout: $!";
 
     $p->get_tag("title");
     $name = $p->get_text("/title");
     $p->get_tag("hr");		# Skip past image and quote, hopefully
 
     # Setup man header
-    print MANOUT ".TH " . $fileinfo->[0] . " " . $fileinfo->[1] .  "\n";
-    print MANOUT ".SH NAME\n";
+    print $manout ".TH " . $fileinfo->[0] . " " . $fileinfo->[1] .  "\n";
+    print $manout ".SH NAME\n";
     $pat = $fileinfo->[0];
     if ($name =~ /$pat/) {
     } else {
 	# Add the manpage name, if not in the HTML title already
-	print MANOUT "$fileinfo->[0] - ";
+	print $manout "$fileinfo->[0] - ";
     }
-    print MANOUT "$name\n.SH \\ \n\n";
+    print $manout "$name\n.SH \\ \n\n";
 
     @fontstack = ();
     $deflevel = 0;
@@ -110,7 +111,7 @@
 		$text =~ s/&amp\;/&/g;
 		$text =~ s/^\./\\[char46]/;
 	    }
-	    print MANOUT "$text";
+	    print $manout "$text";
 	    $tag = 0;
 	}
 	if($token->[0] eq "S") {
@@ -120,57 +121,57 @@
 		if ($filename eq "ntpd" &&
 		    ($text eq "CONFIGURATION OPTIONS")) {
 			$ignore = 1;
-			close(MANOUT);
-			open(MANOUT, ">/dev/null");
+			close($manout);
+			open($manout, ">", "/dev/null");
 		} elsif ($ignore) {
 		    $ignore = 0;
-		    close(MANOUT);
-		    open(MANOUT, ">>$fileout");
+		    close($manout);
+		    open($manout, ">>", $fileout);
 		}
-		print MANOUT "\n\n.SH $text\n";
+		print $manout "\n\n.SH $text\n";
 	    }
 	    if($token->[1] eq "tt") {
 		push @fontstack, "tt";
-		print MANOUT "\\fB";
+		print $manout "\\fB";
 	    }
 	    if($token->[1] eq "i") {
 		push @fontstack, "i";
-		print MANOUT "\\fI";
+		print $manout "\\fI";
 	    }
 	    if($token->[1] eq "address") {
 		my $text = $p->get_trimmed_text("/address");
-		print MANOUT "\n.SH AUTHOR\n$text\n";
+		print $manout "\n.SH AUTHOR\n$text\n";
 	    }
 	    if($token->[1] eq "dt" || $token->[1] eq "br" && $deflevel > 0) {
-		print MANOUT "\n.TP 8\n";
+		print $manout "\n.TP 8\n";
 		$tag = 1;
 	    }
 	    if($token->[1] eq "dd") {
-		print MANOUT "\n";
+		print $manout "\n";
 		$tag = 1;
 	    }
 	    if($token->[1] eq "dl") {
 		$deflevel+=1;
 		if ($deflevel > 0) {
-		    print MANOUT "\n.RS ", $deflevel > 1 ? 8 : 0;
+		    print $manout "\n.RS ", $deflevel > 1 ? 8 : 0;
 		}
 	    }
 	    if($token->[1] eq "p") {
-		print MANOUT "\n";
+		print $manout "\n";
 	    }
 	    if($token->[1] eq "pre") {
-		print MANOUT "\n.nf";
+		print $manout "\n.nf";
 		$pre = 1;
 	    }
 	    if($token->[1] eq "table") {
-		print MANOUT "\n.TS\n";
-		print MANOUT "expand allbox tab(%);\n";
-		print MANOUT $table_headers{$filename};
-		print MANOUT "\n";
+		print $manout "\n.TS\n";
+		print $manout "expand allbox tab(%);\n";
+		print $manout $table_headers{$filename};
+		print $manout "\n";
 	    }
 	    if($token->[1] eq "td") {
 		if ($first_td == 0) {
-		    print MANOUT " % ";
+		    print $manout " % ";
 		}
 		$first_td = 0;
 	    }
@@ -187,7 +188,7 @@
 		if ($#fontstack < 0) { $fontswitch = "\\fR"; }
 		elsif ($fontstack[$#fontstack] eq "tt") { $fontswitch = "\\fB"; }
 		else { $fontswitch = "\\fI"; }
-		print MANOUT "$fontswitch";
+		print $manout "$fontswitch";
 	    }
 	    if($token->[1] eq "i") {
 		$f = pop @fontstack;
@@ -197,38 +198,38 @@
 		if ($#fontstack < 0) { $fontswitch = "\\fR"; }
 		elsif ($fontstack[$#fontstack] eq "tt") { $fontswitch = "\\fB"; }
 		else { $fontswitch = "\\fI"; }
-		print MANOUT "$fontswitch";
+		print $manout "$fontswitch";
 	    }
 	    if($token->[1] eq "dl") {
 		if ($deflevel > 0) {
-		    print MANOUT "\n.RE";
+		    print $manout "\n.RE";
 		}
-		print MANOUT "\n";
+		print $manout "\n";
 		$deflevel-=1;
 	    }
 	    if($token->[1] eq "p") {
-		print MANOUT "\n";
+		print $manout "\n";
 		$tag = 1;
 	    }
 	    if($token->[1] eq "pre") {
-		print MANOUT "\n.fi";
+		print $manout "\n.fi";
 		$pre = 0;
 	    }
 	    if($token->[1] eq "table") {
-		print MANOUT ".TE\n";
+		print $manout ".TE\n";
 	    }
 	    if($token->[1] eq "tr") {
-		print MANOUT "\n";
+		print $manout "\n";
 		$first_td = 1;
 	    }
 	}
     }
     if ($ignore) {
-	close(MANOUT);
-	open(MANOUT, ">>$fileout");
+	close($manout);
+	open($manout, ">>", $fileout);
     }
-    print MANOUT "\n.SH SEE ALSO\n\n";
-    print MANOUT "$fileinfo->[2]\n\n";
-    print MANOUT "$seealso_disclaimer\n";
-    close(MANOUT);
+    print $manout "\n.SH SEE ALSO\n\n";
+    print $manout "$fileinfo->[2]\n\n";
+    print $manout "$seealso_disclaimer\n";
+    close($manout);
 }
