diff -Nur ginkgocadx-3.1.0.498.20.orig/src/cadxcore/commands/comandoexportacion.cpp ginkgocadx-3.1.0.498.20/src/cadxcore/commands/comandoexportacion.cpp
--- ginkgocadx-3.1.0.498.20.orig/src/cadxcore/commands/comandoexportacion.cpp	2012-12-14 00:23:16.000000000 +1100
+++ ginkgocadx-3.1.0.498.20/src/cadxcore/commands/comandoexportacion.cpp	2012-12-27 16:41:35.939160807 +1100
@@ -41,7 +41,6 @@
 
 #include <itkExceptionObject.h>
 #include <itkImage.h>
-#include <itkOrientedImage.h>
 #include <itkRGBPixel.h>
 #include <itkGDCMImageIO.h>
 #include <itkJPEGImageIO.h>
diff -Nur ginkgocadx-3.1.0.498.20.orig/src/cadxcore/commands/thumbnailscommand.cpp ginkgocadx-3.1.0.498.20/src/cadxcore/commands/thumbnailscommand.cpp
--- ginkgocadx-3.1.0.498.20.orig/src/cadxcore/commands/thumbnailscommand.cpp	2012-12-14 00:23:16.000000000 +1100
+++ ginkgocadx-3.1.0.498.20/src/cadxcore/commands/thumbnailscommand.cpp	2012-12-27 16:41:27.471471788 +1100
@@ -51,7 +51,6 @@
 
 #include <itkExceptionObject.h>
 #include <itkImage.h>
-#include <itkOrientedImage.h>
 #include <itkRGBPixel.h>
 #include <itkGDCMImageIO.h>
 #include <itkImageFileReader.h>
