--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -53,7 +53,7 @@ ENDIF()
 
 IF(NOT USE_CUSTOM_WX)
 MESSAGE("Using system wxWidgets")
-FIND_PACKAGE(wxWidgets 3.0.0 COMPONENTS core base gl net xml aui adv richtext xrc REQUIRED)
+FIND_PACKAGE(wxWidgets 3.0.0 COMPONENTS core base gl net xml aui adv html propgrid richtext xrc REQUIRED)
 ELSE()
 MESSAGE("Using custom wxWidgets")
 SET_CUSTOM_wxWidgets_PACKAGE()
--- a/src/cadxcore/VTKInria3D/wxVTK/wxVTKRenderWindowInteractor.cpp
+++ b/src/cadxcore/VTKInria3D/wxVTK/wxVTKRenderWindowInteractor.cpp
@@ -150,7 +150,6 @@ wxWindow* wxGetTopLevelParent(wxWindow *win)
 #ifdef __WXGTK20__
 #if  wxCHECK_VERSION(2, 9, 0)
 #include <gtk/gtk.h>
-#include <wx/gtk/private/win_gtk.h>
 #else
 #include <wx/gtk/win_gtk.h>
 #endif
--- a/src/cadxcore/main/gui/configuration/localdatabaseconfigurationpanel.cpp
+++ b/src/cadxcore/main/gui/configuration/localdatabaseconfigurationpanel.cpp
@@ -220,7 +220,7 @@ namespace GNC {
 			int tmp = -1;
 			if (m_pChoiceDeleteOlder->IsEnabled()) {
 				for (std::map<int, std::string>::const_iterator it = m_mapTimePeriods.begin(); it != m_mapTimePeriods.end(); ++it) {
-					if (wxString::FromUTF8((*it).second.c_str()).CompareTo(m_pChoiceDeleteOlder->GetString(m_pChoiceDeleteOlder->GetSelection())) == 0) {
+					if (wxString::FromUTF8((*it).second.c_str()).CompareTo(m_pChoiceDeleteOlder->GetString(m_pChoiceDeleteOlder->GetSelection()).wx_str()) == 0) {
 						tmp = (*it).first;
 						break;
 					}
--- a/src/cadxcore/main/gui/configuration/panelconfiguracionpermisos.cpp
+++ b/src/cadxcore/main/gui/configuration/panelconfiguracionpermisos.cpp
@@ -521,7 +521,7 @@ namespace GNC {
 				m_pModel->GetValueByRow(value, i, PERM_COL_NAME_IDX);
 				profile["Name"] = value.GetString().ToUTF8();
 				m_pModel->GetValueByRow(value, i, PERM_COL_ACTIVE_IDX);				
-				profile.insertBool("Apply", value.GetString().CompareTo(_("Yes")) == 0);
+				profile.insertBool("Apply", value.GetString().CompareTo(wxT("Yes")) == 0);
 				m_pModel->GetValueByRow(value, i, PERM_COL_XML);			
 				profile["XML"] = value.GetString().ToUTF8();
 				profiles.push_back(profile);
--- a/src/cadxcore/wx/maskedtextctrl/maskedtextctrl.cpp
+++ b/src/cadxcore/wx/maskedtextctrl/maskedtextctrl.cpp
@@ -953,7 +953,7 @@ int wxMaskedTextCtrl::GetPreviousInputLocation(int nSelectionStart)
 	int nInitialInputLocation=nNextInputLocation;
 
 	wxMaskData* pobjData = NULL;
-	for(wxNode* node = m_listData.Item(nNextInputLocation);node;nNextInputLocation--)
+	for(wxList::compatibility_iterator node = m_listData.Item(nNextInputLocation);node;nNextInputLocation--)
 	{
 		pobjData = (wxMaskData *) node->GetData();
 		if(pobjData->IsInputData())
