diff -Nur ginkgocadx-3.1.0.498.20.orig/src/CMakeLists.txt ginkgocadx-3.1.0.498.20/src/CMakeLists.txt
--- ginkgocadx-3.1.0.498.20.orig/src/CMakeLists.txt	2012-12-14 00:49:42.000000000 +1100
+++ ginkgocadx-3.1.0.498.20/src/CMakeLists.txt	2012-12-27 16:48:06.140283406 +1100
@@ -47,7 +47,7 @@
 
 IF(NOT USE_CUSTOM_WX)
 MESSAGE("Using system wxWidgets")
-FIND_PACKAGE(wxWidgets 2.8.11 COMPONENTS core base gl net xml aui adv richtext xrc REQUIRED)
+FIND_PACKAGE(wxWidgets 2.8.11 COMPONENTS core base gl net xml aui adv html richtext xrc REQUIRED)
 ELSE()
 MESSAGE("Using custom wxWidgets")
 SET_CUSTOM_wxWidgets_PACKAGE()
