diff -Naur multimon-1.0/gen.c multimon-sox-arg/gen.c
--- multimon-1.0/gen.c	2005-07-26 21:09:38.000000000 +0200
+++ multimon-sox-arg/gen.c	2009-07-08 13:18:34.000000000 +0200
@@ -320,7 +320,7 @@
 				perror("dup2");
 			close(pipedes[0]); /* close reading pipe end */
 			execlp("sox", "sox", 
-			       "-t", "raw",  "-s", "-w", "-r", srate, "-",
+			       "-t", "raw",  "-s", "-2", "-r", srate, "-",
 			       "-t", type, fname,
 			       NULL);
 			perror("execlp");
diff -Naur multimon-1.0/unixinput.c multimon-sox-arg/unixinput.c
--- multimon-1.0/unixinput.c	2002-12-04 03:49:16.000000000 +0100
+++ multimon-sox-arg/unixinput.c	2009-07-08 13:17:36.000000000 +0200
@@ -324,7 +324,7 @@
 			close(pipedes[1]); /* close writing pipe end */
 			execlp("sox", "sox", 
 			       "-t", type, fname,
-			       "-t", "raw", "-s", "-w", "-r", srate, "-",
+			       "-t", "raw", "-s", "-2", "-r", srate, "-",
 			       NULL);
 			perror("execlp");
 			exit(10);
