--- acdcontrol.cpp.orig	2007-04-05 16:41:50.000000000 +0200
+++ acdcontrol.cpp	2012-01-07 20:08:21.895279144 +0100
@@ -57,8 +57,9 @@
 const int CINEMA_DISPLAY_23_NEW           = 0x9218;
 const int CINEMA_DISPLAY_20_OLD           = 0x9219;
 const int CINEMA_DISPLAY_20_NEW           = 0x9219;
-const int CINEMA_DISPLAY_24               = 0x921e;
+const int CINEMA_DISPLAY_24               = 0x9236;
 const int CINEMA_DISPLAY_30               = 0x9232;
+const int CINEMA_DISPLAY_27               = 0x9226;
 
 const int S1                              = 0x8002;
 
@@ -540,6 +541,8 @@
                                      "Apple Cinema Display 24\"" ));
   supportedDevices.insert( DeviceId( APPLE, CINEMA_DISPLAY_30,
                                      "Apple Cinema HD Display 30\"" ));
+  supportedDevices.insert( DeviceId( APPLE, CINEMA_DISPLAY_27,
+                                     "Apple Cinema HD Display 27\"" ));
 
   supportedDevices.insert( DeviceId( SAMSUNG, S1,
                                      "Samsung SyncMaster 757NF" ));
