diff --git a/indra/newview/llcommandlineparser.cpp b/indra/newview/llcommandlineparser.cpp
index c74406b..db51107 100644
--- a/indra/newview/llcommandlineparser.cpp
+++ b/indra/newview/llcommandlineparser.cpp
@@ -162,6 +162,12 @@ public:
         return mIsComposing;
     }
 
+    // FIXME: the error occurred in boost 1.42.0-1 in AUR package dirty fix
+    virtual bool is_required() const 
+    {
+        return false;
+    }
+
     virtual bool apply_default(boost::any& value_store) const
     {
         return false; // No defaults.
