diff --git a/indra/llmedia/llmediaimplgstreamervidplug.cpp b/indra/llmedia/llmediaimplgstreamervidplug.cpp
index 5e96666..7378fd3 100644
--- a/indra/llmedia/llmediaimplgstreamervidplug.cpp
+++ b/indra/llmedia/llmediaimplgstreamervidplug.cpp
@@ -441,10 +441,9 @@ plugin_init (GstPlugin * plugin)
    some g++ versions buggily avoid __attribute__((constructor)) functions -
    so we provide an explicit plugin init function.
  */
-void gst_slvideo_init_class (void)
-{
+
 #define PACKAGE "packagehack"
-	static GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
+	GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
 				  GST_VERSION_MINOR,
 				  "private-slvideoplugin", 
 				  "SL Video sink plugin",
@@ -452,6 +451,9 @@ void gst_slvideo_init_class (void)
 				  "Second Life",
 				  "http://www.secondlife.com/");
 #undef PACKAGE
+
+void gst_slvideo_init_class (void)
+{
 	ll_gst_plugin_register_static (&gst_plugin_desc);
 	//fprintf(stderr, "\n\n\nCLASS INIT\n\n\n");
 }
