--- a/indra/llimage/llpngwrapper.cpp
+++ b/ndra/llimage/llpngwrapper.cpp
@@ -216,7 +216,7 @@ void LLPngWrapper::normalizeImage()
 	}
     if (mColorType == PNG_COLOR_TYPE_GRAY && mBitDepth < 8)
 	{
-		png_set_gray_1_2_4_to_8(mReadPngPtr);
+		png_set_expand_gray_1_2_4_to_8(mReadPngPtr);
 	}
 	if (mColorType == PNG_COLOR_TYPE_GRAY
 		|| mColorType == PNG_COLOR_TYPE_GRAY_ALPHA)
@@ -364,7 +364,7 @@ void LLPngWrapper::releaseResources()
 {
 	if (mReadPngPtr || mReadInfoPtr)
 	{
-		png_destroy_read_struct(&mReadPngPtr, &mReadInfoPtr, png_infopp_NULL);
+		png_destroy_read_struct(&mReadPngPtr, &mReadInfoPtr, NULL);
 		mReadPngPtr = NULL;
 		mReadInfoPtr = NULL;
 	}
--- a/indra/llimage/llpngwrapper.h
+++ b/indra/llimage/llpngwrapper.h
@@ -32,7 +32,7 @@
 #ifndef LL_LLPNGWRAPPER_H
 #define LL_LLPNGWRAPPER_H
 
-#include "libpng12/png.h"
+#include "libpng14/png.h"
 #include "llimage.h"
 
 class LLPngWrapper
