From 10425c099f6026136ff700d2f471b22952a92428 Mon Sep 17 00:00:00 2001
From: rakib <rakib.mullick@gmail.com>
Date: Thu, 5 Jun 2014 18:31:50 +0600
Subject: [PATCH]  Separate RT_GROUP_SCHED and FAIR_GROUP_SCHED.

---
 kernel/sched/bld.h | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/kernel/sched/bld.h b/kernel/sched/bld.h
index 965ec1d..e14dc21 100644
--- a/kernel/sched/bld.h
+++ b/kernel/sched/bld.h
@@ -5,11 +5,6 @@ static LIST_HEAD(rt_rq_head);
 static LIST_HEAD(cfs_rq_head);
 static DEFINE_RWLOCK(cfs_list_lock);
 #ifdef	CONFIG_FAIR_GROUP_SCHED
-static inline struct rq *rq_of_rt(struct rt_rq *rt_rq)
-{
-	return rt_rq->rq;
-}
-
 static inline struct rq *rq_of_cfs(struct cfs_rq *cfs_rq)
 {
 	return cfs_rq->rq;
@@ -19,7 +14,14 @@ static inline struct rq *rq_of_cfs(struct cfs_rq *cfs_rq)
 {
 	return container_of(cfs_rq, struct rq, cfs);
 }
+#endif
 
+#ifdef	CONFIG_RT_GROUP_SCHED
+static inline struct rq *rq_of_rt(struct rt_rq *rt_rq)
+{
+	return rt_rq->rq;
+}
+#else
 static inline struct rq *rq_of_rt(struct rt_rq *rt_rq)
 {
 	return container_of(rt_rq, struct rq, rt);
-- 
1.9.3

