--- Filesys-SmbClient-3.99_51/Makefile.PL.old	2012-06-21 20:32:16.485717561 +0200
+++ Filesys-SmbClient-3.99_51/Makefile.PL	2012-06-21 20:34:51.056362697 +0200
@@ -7,19 +7,19 @@
 
 # path libsmbclient.h
 my $include = try_to_find("libsmbclient.h");
-$include = prompt("Where can I find libsmbclient.h ?", $include);
+#$include = prompt("Where can I find libsmbclient.h ?", $include);
 warn_user("$include/libsmbclient.h") if (!-r "$include/libsmbclient.h");
 
 # path libsmbclient.so
 my $lib = try_to_find("libsmbclient.so");
-$lib = prompt("Where can I find libsmbclient.so ?",$lib);
+#$lib = prompt("Where can I find libsmbclient.so ?",$lib);
 warn_user("libsmbclient.so") if (!-r "$lib/libsmbclient.so");
 
 # tests demande ?
-my $ans = 
-  prompt("Do you want make tests ?(you will be prompted for server / \n".
-	   "worgroup / share / user / password to make read write access.",
-	   "no");
+#my $ans = 
+#  prompt("Do you want make tests ?(you will be prompted for server / \n".
+#	   "worgroup / share / user / password to make read write access.",
+#	   "no");
 if ($ans =~ /^y(es)?$/i) 
   {
     my $server = prompt("Server ?","localhost");
@@ -33,7 +33,7 @@
   }
 
 # Trace for debug
-$ans = prompt("Compile module with trace to STDERR ?", "no");
+#$ans = prompt("Compile module with trace to STDERR ?", "no");
 if ($ans =~ /^y(es)?$/i) { $define = "-DVERBOSE"; }
 
 # Create config.h for alloca via configure
