--- OLD/src/Makefile.am
+++ NEW/src/Makefile.am
@@ -1,19 +1,25 @@
 ## Process this file with automake to produce Makefile.in
 SUBDIRS= 
 
-bin_PROGRAMS = stardict-editor
-
-noinst_PROGRAMS = pydict2dic olddic2newdic oxford2dic directory2dic dictd2dic \
-	wquick2dic ec50 directory2treedic treedict2dir jdictionary mova xmlinout \
-	soothill kanjidic2 powerword kdic 21tech 21shiji buddhist tabfile dsl2dict \
-	cedict edict duden stardict-dict-update degb2utf frgb2utf jpgb2utf gmx2utf \
-	rucn kingsoft wikipedia wikipediaImage babylon stardict2txt stardict-verify \
-	fest2dict i2e2dict downloadwiki testutf8 ooo2dict myspell2dic exc2i2e \
-	dictbuilder tabfile2sql KangXi Unihan xiaoxuetang-ja wubi bgl2txt ydp2dict \
-	wordnet lingvosound2resdb resdatabase2dir dir2resdatabase stardict-index \
+bin_PROGRAMS = stardict-editor \
+	stardict-pydict2dic stardict-olddic2newdic stardict-oxford2dic \
+	stardict-directory2dic stardict-dictd2dic stardict-wquick2dic stardict-ec50 \
+	stardict-directory2treedic stardict-treedict2dir stardict-jdictionary \
+	stardict-mova stardict-xmlinout stardict-soothill stardict-kanjidic2 \
+	stardict-powerword stardict-kdic stardict-21tech stardict-21shiji \
+	stardict-buddhist stardict-tabfile stardict-dsl2dict stardict-cedict \
+	stardict-edict stardict-duden stardict-dict-update stardict-degb2utf \
+	stardict-frgb2utf stardict-jpgb2utf stardict-gmx2utf stardict-rucn \
+	stardict-kingsoft stardict-wikipedia stardict-wikipediaImage stardict-babylon \
+	stardict-stardict2txt stardict-verify stardict-fest2dict stardict-i2e2dict \
+	stardict-downloadwiki stardict-testutf8 stardict-ooo2dict stardict-myspell2dic \
+	stardict-exc2i2e stardict-dictbuilder stardict-tabfile2sql stardict-KangXi \
+	stardict-Unihan stardict-xiaoxuetang-ja stardict-wubi stardict-bgl2txt \
+	stardict-ydp2dict stardict-wordnet stardict-lingvosound2resdb \
+	stardict-resdatabase2dir stardict-dir2resdatabase stardict-index \
 	stardict-text2bin stardict-bin2text
 
-noinst_LIBRARIES = libbgl2txt.a
+lib_LIBRARIES = libbgl2txt.a
 
 AM_CPPFLAGS = $(STARDICT_CFLAGS) -I$(top_builddir) -I$(top_srcdir)
 
@@ -22,230 +28,230 @@
 	bgl_babylonreader.cpp bgl_stardictbuilder.h bgl_stardictbuilder.cpp \
 	resourcewrap.hpp libcommon.cpp libcommon.h
 
-lingvosound2resdb_LDFLAGS = 
-lingvosound2resdb_LDADD = $(STARDICT_LIBS)
-lingvosound2resdb_SOURCES = lingvosound2resdb.cpp
-
-dir2resdatabase_LDFLAGS = 
-dir2resdatabase_LDADD = $(STARDICT_LIBS) -lz
-dir2resdatabase_SOURCES = dir2resdatabase.cpp resourcewrap.hpp \
+stardict_lingvosound2resdb_LDFLAGS = 
+stardict_lingvosound2resdb_LDADD = $(STARDICT_LIBS)
+stardict_lingvosound2resdb_SOURCES = lingvosound2resdb.cpp
+
+stardict_dir2resdatabase_LDFLAGS = 
+stardict_dir2resdatabase_LDADD = $(STARDICT_LIBS) -lz
+stardict_dir2resdatabase_SOURCES = dir2resdatabase.cpp resourcewrap.hpp \
 	libcommon.cpp libcommon.h
 
-resdatabase2dir_LDFLAGS = 
-resdatabase2dir_LDADD = $(STARDICT_LIBS) -lz
-resdatabase2dir_SOURCES = resdatabase2dir.cpp resourcewrap.hpp \
+stardict_resdatabase2dir_LDFLAGS = 
+stardict_resdatabase2dir_LDADD = $(STARDICT_LIBS) -lz
+stardict_resdatabase2dir_SOURCES = resdatabase2dir.cpp resourcewrap.hpp \
 	libcommon.cpp libcommon.h
 
-pydict2dic_LDFLAGS = 
-pydict2dic_LDADD = $(STARDICT_LIBS)
-pydict2dic_SOURCES = pydict2dic.c
-
-olddic2newdic_LDFLAGS = 
-olddic2newdic_LDADD = $(STARDICT_LIBS)
-olddic2newdic_SOURCES = olddic2newdic.cpp
-
-oxford2dic_LDFLAGS = 
-oxford2dic_LDADD = $(STARDICT_LIBS)
-oxford2dic_SOURCES = oxford2dic.cpp
-
-directory2dic_LDFLAGS = 
-directory2dic_LDADD = $(STARDICT_LIBS)
-directory2dic_SOURCES = directory2dic.cpp
-
-dictd2dic_LDFLAGS = 
-dictd2dic_LDADD = $(STARDICT_LIBS)
-dictd2dic_SOURCES = dictd2dic.cpp
-
-wquick2dic_LDFLAGS = 
-wquick2dic_LDADD = $(STARDICT_LIBS)
-wquick2dic_SOURCES = wquick2dic.cpp
-
-ec50_LDFLAGS = 
-ec50_LDADD = $(STARDICT_LIBS)
-ec50_SOURCES = ec50.cpp
-
-directory2treedic_LDFLAGS = 
-directory2treedic_LDADD = $(STARDICT_LIBS)
-directory2treedic_SOURCES = directory2treedic.cpp
-
-treedict2dir_LDFLAGS =
-treedict2dir_LDADD = $(STARDICT_LIBS)
-treedict2dir_SOURCES = treedict2dir.cpp
-
-jdictionary_LDFLAGS = 
-jdictionary_LDADD = $(STARDICT_LIBS)
-jdictionary_SOURCES = jdictionary.cpp
-
-mova_LDFLAGS = 
-mova_LDADD = $(STARDICT_LIBS)
-mova_SOURCES = mova.cpp
-
-xmlinout_LDFLAGS =
-xmlinout_LDADD = $(STARDICT_LIBS)
-xmlinout_SOURCES = xmlinout.cpp
-
-soothill_LDFLAGS =
-soothill_LDADD = $(STARDICT_LIBS)
-soothill_SOURCES = soothill.cpp
-
-kanjidic2_LDFLAGS =
-kanjidic2_LDADD = $(STARDICT_LIBS)
-kanjidic2_SOURCES = kanjidic2.cpp
-
-powerword_LDFLAGS =
-powerword_LDADD = $(STARDICT_LIBS)
-powerword_SOURCES = powerword.cpp
-
-kdic_LDFLAGS =
-kdic_LDADD = $(STARDICT_LIBS)
-kdic_SOURCES = kdic.cpp
-
-21tech_LDFLAGS =
-21tech_LDADD = $(STARDICT_LIBS)
-21tech_SOURCES = 21tech.cpp
-
-21shiji_LDFLAGS =
-21shiji_LDADD = $(STARDICT_LIBS)
-21shiji_SOURCES = 21shiji.cpp
-
-buddhist_LDFLAGS =
-buddhist_LDADD = $(STARDICT_LIBS)
-buddhist_SOURCES = buddhist.cpp
-
-tabfile_LDFLAGS =
-tabfile_LDADD = $(STARDICT_LIBS) -lz
-tabfile_SOURCES = tabfile.cpp libtabfile.cpp libtabfile.h libcommon.cpp \
+stardict_pydict2dic_LDFLAGS = 
+stardict_pydict2dic_LDADD = $(STARDICT_LIBS)
+stardict_pydict2dic_SOURCES = pydict2dic.c
+
+stardict_olddic2newdic_LDFLAGS = 
+stardict_olddic2newdic_LDADD = $(STARDICT_LIBS)
+stardict_olddic2newdic_SOURCES = olddic2newdic.cpp
+
+stardict_oxford2dic_LDFLAGS = 
+stardict_oxford2dic_LDADD = $(STARDICT_LIBS)
+stardict_oxford2dic_SOURCES = oxford2dic.cpp
+
+stardict_directory2dic_LDFLAGS = 
+stardict_directory2dic_LDADD = $(STARDICT_LIBS)
+stardict_directory2dic_SOURCES = directory2dic.cpp
+
+stardict_dictd2dic_LDFLAGS = 
+stardict_dictd2dic_LDADD = $(STARDICT_LIBS)
+stardict_dictd2dic_SOURCES = dictd2dic.cpp
+
+stardict_wquick2dic_LDFLAGS = 
+stardict_wquick2dic_LDADD = $(STARDICT_LIBS)
+stardict_wquick2dic_SOURCES = wquick2dic.cpp
+
+stardict_ec50_LDFLAGS = 
+stardict_ec50_LDADD = $(STARDICT_LIBS)
+stardict_ec50_SOURCES = ec50.cpp
+
+stardict_directory2treedic_LDFLAGS = 
+stardict_directory2treedic_LDADD = $(STARDICT_LIBS)
+stardict_directory2treedic_SOURCES = directory2treedic.cpp
+
+stardict_treedict2dir_LDFLAGS =
+stardict_treedict2dir_LDADD = $(STARDICT_LIBS)
+stardict_treedict2dir_SOURCES = treedict2dir.cpp
+
+stardict_jdictionary_LDFLAGS = 
+stardict_jdictionary_LDADD = $(STARDICT_LIBS)
+stardict_jdictionary_SOURCES = jdictionary.cpp
+
+stardict_mova_LDFLAGS = 
+stardict_mova_LDADD = $(STARDICT_LIBS)
+stardict_mova_SOURCES = mova.cpp
+
+stardict_xmlinout_LDFLAGS =
+stardict_xmlinout_LDADD = $(STARDICT_LIBS)
+stardict_xmlinout_SOURCES = xmlinout.cpp
+
+stardict_soothill_LDFLAGS =
+stardict_soothill_LDADD = $(STARDICT_LIBS)
+stardict_soothill_SOURCES = soothill.cpp
+
+stardict_kanjidic2_LDFLAGS =
+stardict_kanjidic2_LDADD = $(STARDICT_LIBS)
+stardict_kanjidic2_SOURCES = kanjidic2.cpp
+
+stardict_powerword_LDFLAGS =
+stardict_powerword_LDADD = $(STARDICT_LIBS)
+stardict_powerword_SOURCES = powerword.cpp
+
+stardict_kdic_LDFLAGS =
+stardict_kdic_LDADD = $(STARDICT_LIBS)
+stardict_kdic_SOURCES = kdic.cpp
+
+stardict_21tech_LDFLAGS =
+stardict_21tech_LDADD = $(STARDICT_LIBS)
+stardict_21tech_SOURCES = 21tech.cpp
+
+stardict_21shiji_LDFLAGS =
+stardict_21shiji_LDADD = $(STARDICT_LIBS)
+stardict_21shiji_SOURCES = 21shiji.cpp
+
+stardict_buddhist_LDFLAGS =
+stardict_buddhist_LDADD = $(STARDICT_LIBS)
+stardict_buddhist_SOURCES = buddhist.cpp
+
+stardict_tabfile_LDFLAGS =
+stardict_tabfile_LDADD = $(STARDICT_LIBS) -lz
+stardict_tabfile_SOURCES = tabfile.cpp libtabfile.cpp libtabfile.h libcommon.cpp \
 	libcommon.h resourcewrap.hpp
 
-cedict_LDFLAGS =
-cedict_LDADD = $(STARDICT_LIBS)
-cedict_SOURCES = cedict.cpp
-
-edict_LDFLAGS =
-edict_LDADD = $(STARDICT_LIBS)
-edict_SOURCES = edict.cpp
-
-dsl2dict_LDFLAGS =
-dsl2dict_LDADD = $(STARDICT_LIBS) $(DSL2DICT_LIBS)
-dsl2dict_SOURCES = dsl2dict.c
-
-duden_LDFLAGS =
-duden_LDADD = $(STARDICT_LIBS)
-duden_SOURCES = duden.cpp
-
-degb2utf_LDFLAGS =
-degb2utf_LDADD = $(STARDICT_LIBS)
-degb2utf_SOURCES = degb2utf.cpp
-
-frgb2utf_LDFLAGS =
-frgb2utf_LDADD = $(STARDICT_LIBS)
-frgb2utf_SOURCES = frgb2utf.cpp
-
-jpgb2utf_LDFLAGS =
-jpgb2utf_LDADD = $(STARDICT_LIBS)
-jpgb2utf_SOURCES = jpgb2utf.cpp
-
-gmx2utf_LDFLAGS =
-gmx2utf_LDADD = $(STARDICT_LIBS)
-gmx2utf_SOURCES = gmx2utf.cpp
-
-rucn_LDFLAGS =
-rucn_LDADD = $(STARDICT_LIBS)
-rucn_SOURCES = rucn.cpp
-
-kingsoft_LDFLAGS =
-kingsoft_LDADD = $(STARDICT_LIBS)
-kingsoft_SOURCES = kingsoft.cpp
-
-wikipedia_CPPFLAGS = $(AM_CPPFLAGS) $(LFS_CFLAGS)
-wikipedia_LDFLAGS = $(LFS_LDFLAGS)
-wikipedia_LDADD = $(STARDICT_LIBS) $(LFS_LIBS)
-wikipedia_SOURCES = wikipedia.cpp
-
-wikipediaImage_CPPFLAGS = $(AM_CPPFLAGS) $(MYSQL_CFLAGS)
-wikipediaImage_LDFLAGS = $(MYSQL_LDFLAGS)
-wikipediaImage_LDADD = $(STARDICT_LIBS) $(MYSQL_LIBS)
-wikipediaImage_SOURCES = wikipediaImage.cpp
-
-tabfile2sql_CPPFLAGS = $(AM_CPPFLAGS) $(MYSQL_CFLAGS)
-tabfile2sql_LDFLAGS = $(MYSQL_LDFLAGS)
-tabfile2sql_LDADD = $(STARDICT_LIBS) $(MYSQL_LIBS)
-tabfile2sql_SOURCES = tabfile2sql.cpp
-
-babylon_LDFLAGS =
-babylon_LDADD = $(STARDICT_LIBS) -lz
-babylon_SOURCES = babylon.cpp libbabylonfile.cpp libbabylonfile.h \
+stardict_cedict_LDFLAGS =
+stardict_cedict_LDADD = $(STARDICT_LIBS)
+stardict_cedict_SOURCES = cedict.cpp
+
+stardict_edict_LDFLAGS =
+stardict_edict_LDADD = $(STARDICT_LIBS)
+stardict_edict_SOURCES = edict.cpp
+
+stardict_dsl2dict_LDFLAGS =
+stardict_dsl2dict_LDADD = $(STARDICT_LIBS) $(DSL2DICT_LIBS)
+stardict_dsl2dict_SOURCES = dsl2dict.c
+
+stardict_duden_LDFLAGS =
+stardict_duden_LDADD = $(STARDICT_LIBS)
+stardict_duden_SOURCES = duden.cpp
+
+stardict_degb2utf_LDFLAGS =
+stardict_degb2utf_LDADD = $(STARDICT_LIBS)
+stardict_degb2utf_SOURCES = degb2utf.cpp
+
+stardict_frgb2utf_LDFLAGS =
+stardict_frgb2utf_LDADD = $(STARDICT_LIBS)
+stardict_frgb2utf_SOURCES = frgb2utf.cpp
+
+stardict_jpgb2utf_LDFLAGS =
+stardict_jpgb2utf_LDADD = $(STARDICT_LIBS)
+stardict_jpgb2utf_SOURCES = jpgb2utf.cpp
+
+stardict_gmx2utf_LDFLAGS =
+stardict_gmx2utf_LDADD = $(STARDICT_LIBS)
+stardict_gmx2utf_SOURCES = gmx2utf.cpp
+
+stardict_rucn_LDFLAGS =
+stardict_rucn_LDADD = $(STARDICT_LIBS)
+stardict_rucn_SOURCES = rucn.cpp
+
+stardict_kingsoft_LDFLAGS =
+stardict_kingsoft_LDADD = $(STARDICT_LIBS)
+stardict_kingsoft_SOURCES = kingsoft.cpp
+
+stardict_wikipedia_CPPFLAGS = $(AM_CPPFLAGS) $(LFS_CFLAGS)
+stardict_wikipedia_LDFLAGS = $(LFS_LDFLAGS)
+stardict_wikipedia_LDADD = $(STARDICT_LIBS) $(LFS_LIBS)
+stardict_wikipedia_SOURCES = wikipedia.cpp
+
+stardict_wikipediaImage_CPPFLAGS = $(AM_CPPFLAGS) $(MYSQL_CFLAGS)
+stardict_wikipediaImage_LDFLAGS = $(MYSQL_LDFLAGS)
+stardict_wikipediaImage_LDADD = $(STARDICT_LIBS) $(MYSQL_LIBS)
+stardict_wikipediaImage_SOURCES = wikipediaImage.cpp
+
+stardict_tabfile2sql_CPPFLAGS = $(AM_CPPFLAGS) $(MYSQL_CFLAGS)
+stardict_tabfile2sql_LDFLAGS = $(MYSQL_LDFLAGS)
+stardict_tabfile2sql_LDADD = $(STARDICT_LIBS) $(MYSQL_LIBS)
+stardict_tabfile2sql_SOURCES = tabfile2sql.cpp
+
+stardict_babylon_LDFLAGS =
+stardict_babylon_LDADD = $(STARDICT_LIBS) -lz
+stardict_babylon_SOURCES = babylon.cpp libbabylonfile.cpp libbabylonfile.h \
 	libcommon.cpp libcommon.h resourcewrap.hpp
 
-stardict2txt_LDFLAGS =
-stardict2txt_LDADD = $(STARDICT_LIBS) -lz
-stardict2txt_SOURCES = stardict2txt.cpp libstardict2txt.cpp libstardict2txt.h \
+stardict_stardict2txt_LDFLAGS =
+stardict_stardict2txt_LDADD = $(STARDICT_LIBS) -lz
+stardict_stardict2txt_SOURCES = stardict2txt.cpp libstardict2txt.cpp libstardict2txt.h \
 	libcommon.cpp libcommon.h resourcewrap.hpp
 
 stardict_dict_update_LDFLAGS = 
 stardict_dict_update_LDADD = $(STARDICT_LIBS) -lz
 stardict_dict_update_SOURCES = stardict_dict_update.cpp
 
-fest2dict_LDFLAGS = 
-fest2dict_LDADD = 
-fest2dict_SOURCES = fest2dict.c
-
-testutf8_LDFLAGS = 
-testutf8_LDADD = $(STARDICT_LIBS)
-testutf8_SOURCES = testutf8.c
-
-i2e2dict_LDFLAGS = 
-i2e2dict_LDADD = $(STARDICT_LIBS)
-i2e2dict_SOURCES = i2e2dict.c
-
-ooo2dict_LDFLAGS = 
-ooo2dict_LDADD = $(STARDICT_LIBS)
-ooo2dict_SOURCES = ooo2dict.c
-
-myspell2dic_LDFLAGS = 
-myspell2dic_LDADD = $(STARDICT_LIBS)
-myspell2dic_SOURCES = myspell2dic.c
-
-exc2i2e_LDFLAGS = 
-exc2i2e_LDADD = $(STARDICT_LIBS)
-exc2i2e_SOURCES = exc2i2e.c
-
-dictbuilder_LDFLAGS = 
-dictbuilder_LDADD = $(STARDICT_LIBS)
-dictbuilder_SOURCES = dictbuilder.cpp dictbuilder-tree.h
-
-downloadwiki_LDFLAGS = 
-downloadwiki_LDADD = $(STARDICT_LIBS)
-downloadwiki_SOURCES = downloadwiki.cpp
-
-KangXi_LDFLAGS =
-KangXi_LDADD = $(STARDICT_LIBS)
-KangXi_SOURCES = KangXi.cpp
-
-Unihan_LDFLAGS =
-Unihan_LDADD = $(STARDICT_LIBS)
-Unihan_SOURCES = Unihan.cpp
-
-xiaoxuetang_ja_LDFLAGS =
-xiaoxuetang_ja_LDADD = $(STARDICT_LIBS)
-xiaoxuetang_ja_SOURCES = xiaoxuetang-ja.cpp
-
-wubi_LDFLAGS =
-wubi_LDADD = $(STARDICT_LIBS)
-wubi_SOURCES = wubi.cpp
-
-ydp2dict_LDFLAGS =
-ydp2dict_LDADD = $(STARDICT_LIBS)
-ydp2dict_SOURCES = ydp2dict.c
-
-wordnet_LDFLAGS =
-wordnet_LDADD = $(STARDICT_LIBS)
-wordnet_SOURCES = wordnet.cpp
-
-bgl2txt_DEPENDENCIES = libbgl2txt.a
-bgl2txt_LDFLAGS =
-bgl2txt_LDADD = $(STARDICT_LIBS) libbgl2txt.a -lz
-bgl2txt_SOURCES = bgl2txt.cpp libcommon.h libcommon.cpp
+stardict_fest2dict_LDFLAGS = 
+stardict_fest2dict_LDADD = 
+stardict_fest2dict_SOURCES = fest2dict.c
+
+stardict_testutf8_LDFLAGS = 
+stardict_testutf8_LDADD = $(STARDICT_LIBS)
+stardict_testutf8_SOURCES = testutf8.c
+
+stardict_i2e2dict_LDFLAGS = 
+stardict_i2e2dict_LDADD = $(STARDICT_LIBS)
+stardict_i2e2dict_SOURCES = i2e2dict.c
+
+stardict_ooo2dict_LDFLAGS = 
+stardict_ooo2dict_LDADD = $(STARDICT_LIBS)
+stardict_ooo2dict_SOURCES = ooo2dict.c
+
+stardict_myspell2dic_LDFLAGS = 
+stardict_myspell2dic_LDADD = $(STARDICT_LIBS)
+stardict_myspell2dic_SOURCES = myspell2dic.c
+
+stardict_exc2i2e_LDFLAGS = 
+stardict_exc2i2e_LDADD = $(STARDICT_LIBS)
+stardict_exc2i2e_SOURCES = exc2i2e.c
+
+stardict_dictbuilder_LDFLAGS = 
+stardict_dictbuilder_LDADD = $(STARDICT_LIBS)
+stardict_dictbuilder_SOURCES = dictbuilder.cpp dictbuilder-tree.h
+
+stardict_downloadwiki_LDFLAGS = 
+stardict_downloadwiki_LDADD = $(STARDICT_LIBS)
+stardict_downloadwiki_SOURCES = downloadwiki.cpp
+
+stardict_KangXi_LDFLAGS =
+stardict_KangXi_LDADD = $(STARDICT_LIBS)
+stardict_KangXi_SOURCES = KangXi.cpp
+
+stardict_Unihan_LDFLAGS =
+stardict_Unihan_LDADD = $(STARDICT_LIBS)
+stardict_Unihan_SOURCES = Unihan.cpp
+
+stardict_xiaoxuetang_ja_LDFLAGS =
+stardict_xiaoxuetang_ja_LDADD = $(STARDICT_LIBS)
+stardict_xiaoxuetang_ja_SOURCES = xiaoxuetang-ja.cpp
+
+stardict_wubi_LDFLAGS =
+stardict_wubi_LDADD = $(STARDICT_LIBS)
+stardict_wubi_SOURCES = wubi.cpp
+
+stardict_ydp2dict_LDFLAGS =
+stardict_ydp2dict_LDADD = $(STARDICT_LIBS)
+stardict_ydp2dict_SOURCES = ydp2dict.c
+
+stardict_wordnet_LDFLAGS =
+stardict_wordnet_LDADD = $(STARDICT_LIBS)
+stardict_wordnet_SOURCES = wordnet.cpp
+
+stardict_bgl2txt_DEPENDENCIES = libbgl2txt.a
+stardict_bgl2txt_LDFLAGS =
+stardict_bgl2txt_LDADD = $(STARDICT_LIBS) libbgl2txt.a -lz
+stardict_bgl2txt_SOURCES = bgl2txt.cpp libcommon.h libcommon.cpp
 
 stardict_index_LDFLAGS =
 stardict_index_LDADD = $(STARDICT_LIBS)
