*** configure.in.orig	2013-07-25 01:17:19.087699388 +0200
--- configure.in	2013-07-25 01:19:24.621559439 +0200
***************
*** 108,121 ****
  
  
  dnl Check for MMDB.
! AM_PATH_MMDB(, [echo Error locating necessary mmdb; exit 1],,)
  
! AM_WITH_MMDBSSM
  AM_WITH_CCP4SRS
  
  dnl Check for Clipper.  we exit if clipper is not found.
  #
! AM_PATH_CLIPPER(, [echo Error locating Clipper - a CNS-enabled version of Clipper \(2.1-090520-ac or later\) is required now; exit 1],AC_MSG_ERROR([Cannot find Clipper]))
  
  # we set default coot_guile to true
  coot_guile=true
--- 108,136 ----
  
  
  dnl Check for MMDB.
! # AM_PATH_MMDB(, [echo Error locating necessary mmdb; exit 1],,)
  
! # AM_WITH_MMDBSSM
  AM_WITH_CCP4SRS
  
+ PKG_CHECK_MODULES(MMDB, mmdb >= 1.12)
+ MMDB_CXXFLAGS="-DHAVE_MMDB_IGNORE_HASH -DHAVE_MMDB_WITH_CISPEP $MMDB_CFLAGS"
+ AC_SUBST(MMDB_CXXFLAGS)
+ AC_SUBST(MMDB_LIBS)
+ 
+ PKG_CHECK_MODULES(MMDBSSM, ssm >= 0.1)
+ MMDBSSM_CXXFLAGS="-DHAVE_SSMLIB $MMDBSSM_CFLAGS"
+ AC_SUBST(MMDBSSM_CXXFLAGS)
+ AC_SUBST(MMDBSSM_LIBS)
+ 
  dnl Check for Clipper.  we exit if clipper is not found.
  #
! # AM_PATH_CLIPPER(, [echo Error locating Clipper - a CNS-enabled version of Clipper \(2.1-090520-ac or later\) is required now; exit 1],AC_MSG_ERROR([Cannot find Clipper]))
! 
! PKG_CHECK_MODULES(CLIPPER, clipper >= 2.0)
! CLIPPER_CXXFLAGS=$CLIPPER_CFLAGS
! AC_SUBST(CLIPPER_CXXFLAGS)
! AC_SUBST(CLIPPER_LIBS)
  
  # we set default coot_guile to true
  coot_guile=true
