From f3c4a4943c5e765596ef0497d7784447185941c0 Mon Sep 17 00:00:00 2001
From: Luka Perkov <luka@openwrt.org>
Date: Sun, 26 Jan 2014 13:49:40 +0100
Subject: [PATCH] [ubus] lua: fix whitespaces and typo

Signed-off-by: Luka Perkov <luka@openwrt.org>
---
 lua/ubus.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lua/ubus.c b/lua/ubus.c
index 84ee06b..45ea3bb 100644
--- a/lua/ubus.c
+++ b/lua/ubus.c
@@ -19,8 +19,8 @@
 #include <lauxlib.h>
 #include <lua.h>
 
-#define MODNAME        "ubus"
-#define METANAME       MODNAME ".meta"
+#define MODNAME		"ubus"
+#define METANAME	MODNAME ".meta"
 
 static lua_State *state;
 
@@ -417,7 +417,7 @@ static struct ubus_object* ubus_lua_load_object(lua_State *L)
 	obj->o.type->id = 0;
 	obj->o.type->methods = obj->o.methods;
 
-	/* create the he callback lookup table */
+	/* create the callback lookup table */
 	lua_createtable(L, 1, 0);
 	lua_getglobal(L, "__ubus_cb");
 	lua_pushvalue(L, -2);
-- 
1.8.5.3

