--- qtbase-opensource-src-5.1.1/src/plugins/platforms/windows/qwindowstheme.cpp.orig	2013-08-29 22:38:22.849226696 +0200
+++ qtbase-opensource-src-5.1.1/src/plugins/platforms/windows/qwindowstheme.cpp	2013-08-29 22:47:41.852203317 +0200
@@ -39,6 +39,10 @@
 **
 ****************************************************************************/
 
+/* The structure SHSTOCKICONINFO which is used in
+ * this file is only available as of Windows Vista */
+#define _WIN32_WINNT 0x0600
+
 #include "qwindowstheme.h"
 #include "qwindowsdialoghelpers.h"
 #include "qwindowscontext.h"
