diff -Naur orig/premake4.lua new/premake4.lua
--- orig/premake4.lua	2014-05-19 10:42:14.000000000 +0200
+++ new/premake4.lua	2014-10-17 16:25:20.688127783 +0200
@@ -30,7 +30,7 @@
 		}
 	else
 		includedirs {
-			"/opt/SDL-2.0/include/SDL2",
+			"/usr/include/SDL2",
 			"/usr/include/GL",
 		}
 	end
diff -Naur orig/src/core_lua.c new/src/core_lua.c
--- orig/src/core_lua.c	2014-07-04 17:12:50.000000000 +0200
+++ new/src/core_lua.c	2014-10-17 16:27:20.714791315 +0200
@@ -622,10 +622,15 @@
 	SDL_RWops *src = PHYSFSRWOPS_openRead(name);
 	if (!src)
 	{
-		return luaL_error(L, "could not load font: %s (%d)", name, size);
+		char *sysfont = NULL;
+		asprintf(&sysfont, "/usr/share/fonts/TTF/%s", basename(name));
+		*f = TTF_OpenFont(sysfont, size);
+		free(sysfont);
+	}
+	else 
+	{
+		*f = TTF_OpenFontRW(src, TRUE, size);
 	}
-
-	*f = TTF_OpenFontRW(src, TRUE, size);
 
 	if (!*f)
 	{
