diff -Naur t-engine4-src-1.0.0beta40/premake4.lua t-engine4-src-1.0.0beta40-fixed/premake4.lua
--- t-engine4-src-1.0.0beta40/premake4.lua	2012-02-03 14:53:35.000000000 +0100
+++ t-engine4-src-1.0.0beta40-fixed/premake4.lua	2012-05-10 23:06:33.254603667 +0200
@@ -18,7 +18,7 @@
 		"src/physfs",
 		"src/physfs/zlib123",
 		"src/bzip2",
-		"/opt/SDL-2.0/include/SDL2/",
+		"/usr/include/SDL2",
 		"/usr/include/GL",
 	}
 	if _OPTIONS.lua == "default" then includedirs{"src/lua"}
