--- ../dummy/src/profile.c	2011-11-23 21:00:15.605263295 -0200
+++ src/profile.c	2011-11-23 21:00:51.724729177 -0200
@@ -207,7 +207,7 @@
 	profile->lock_oqueue = SDL_CreateMutex();
 	profile->wait_oqueue = SDL_CreateSemaphore(0);
 
-	thread = SDL_CreateThread(thread_profile, profile);
+	thread = SDL_CreateThread(thread_profile, "t_profile", profile);
 	if (thread == NULL) {
 		printf("Unable to create profile thread: %s\n", SDL_GetError());
 		return -1;
