The jit2 Lua engine is not supported for x86_64.

diff -Naur t-engine4-src-1.0.0beta40/build/options.lua t-engine4-src-1.0.0beta40-fixed/build/options.lua
--- t-engine4-src-1.0.0beta40/build/options.lua	2012-05-08 14:18:34.000000000 +0200
+++ t-engine4-src-1.0.0beta40-fixed/build/options.lua	2012-05-10 23:28:49.592462042 +0200
@@ -52,4 +52,4 @@
 	description = "Doesnt remove jit2 build files",
 }
 
-_OPTIONS.lua = _OPTIONS.lua or "jit2"
+_OPTIONS.lua = _OPTIONS.lua or "default"
