diff -u -8 -p -r1.98 cmd-set-option.c
--- cmd-set-option.c	2 Jul 2010 02:45:52 -0000	1.98
+++ cmd-set-option.c	5 Aug 2010 05:47:19 -0000
@@ -153,19 +153,28 @@ const struct set_option_entry set_window
 	{ "mode-fg", SET_OPTION_COLOUR, 0, 0, NULL },
 	{ "mode-keys", SET_OPTION_CHOICE, 0, 0, set_option_mode_keys_list },
 	{ "mode-mouse", SET_OPTION_FLAG, 0, 0, NULL },
 	{ "monitor-activity", SET_OPTION_FLAG, 0, 0, NULL },
 	{ "monitor-content", SET_OPTION_STRING, 0, 0, NULL },
 	{ "remain-on-exit", SET_OPTION_FLAG, 0, 0, NULL },
 	{ "synchronize-panes", SET_OPTION_FLAG, 0, 0, NULL },
 	{ "utf8", SET_OPTION_FLAG, 0, 0, NULL },
-	{ "window-status-alert-attr", SET_OPTION_ATTRIBUTES, 0, 0, NULL },
-	{ "window-status-alert-bg", SET_OPTION_COLOUR, 0, 0, NULL },
-	{ "window-status-alert-fg", SET_OPTION_COLOUR, 0, 0, NULL },
+	/* v window status patch */
+	{ "window-status-activity-attr", SET_OPTION_ATTRIBUTES, 0, 0, NULL },
+	{ "window-status-activity-bg", SET_OPTION_COLOUR, 0, 0, NULL },
+	{ "window-status-activity-fg", SET_OPTION_COLOUR, 0, 0, NULL },
+	{ "window-status-content-attr", SET_OPTION_ATTRIBUTES, 0, 0, NULL },
+	{ "window-status-content-bg", SET_OPTION_COLOUR, 0, 0, NULL },
+	{ "window-status-content-fg", SET_OPTION_COLOUR, 0, 0, NULL },
+	{ "window-status-bell-attr", SET_OPTION_ATTRIBUTES, 0, 0, NULL },
+	{ "window-status-bell-bg", SET_OPTION_COLOUR, 0, 0, NULL },
+	{ "window-status-bell-fg", SET_OPTION_COLOUR, 0, 0, NULL },
+	/* ^ window status patch */
+	{ "window-status-bell-attr", SET_OPTION_ATTRIBUTES, 0, 0, NULL },
 	{ "window-status-attr", SET_OPTION_ATTRIBUTES, 0, 0, NULL },
 	{ "window-status-bg", SET_OPTION_COLOUR, 0, 0, NULL },
 	{ "window-status-current-attr", SET_OPTION_ATTRIBUTES, 0, 0, NULL },
 	{ "window-status-current-bg", SET_OPTION_COLOUR, 0, 0, NULL },
 	{ "window-status-current-fg", SET_OPTION_COLOUR, 0, 0, NULL },
 	{ "window-status-current-format", SET_OPTION_STRING, 0, 0, NULL },
 	{ "window-status-fg", SET_OPTION_COLOUR, 0, 0, NULL },
 	{ "window-status-format", SET_OPTION_STRING, 0, 0, NULL },
diff -u -8 -p -r1.149 status.c
--- status.c	22 Jun 2010 23:26:18 -0000	1.149
+++ status.c	5 Aug 2010 05:47:23 -0000
@@ -578,27 +578,54 @@ status_print(
 		if (bg != 8)
 			colour_set_bg(gc, bg);
 		attr = options_get_number(oo, "window-status-current-attr");
 		if (attr != 0)
 			gc->attr = attr;
 		fmt = options_get_string(oo, "window-status-current-format");
 	}
 
-	if (wl->flags & WINLINK_ALERTFLAGS) {
-		fg = options_get_number(oo, "window-status-alert-fg");
+	/* v window status patch */
+	/* Bell is highest priority, followed by content, followed by activity.
+	 * _I_ think this is the sanest thing to do, but maybe the priority should
+	 * be configurable? */
+	if (wl->flags & WINLINK_BELL) {
+		fg = options_get_number(oo, "window-status-bell-fg");
 		if (fg != 8)
 			colour_set_fg(gc, fg);
-		bg = options_get_number(oo, "window-status-alert-bg");
+		bg = options_get_number(oo, "window-status-bell-bg");
 		if (bg != 8)
 			colour_set_bg(gc, bg);
-		attr = options_get_number(oo, "window-status-alert-attr");
+		attr = options_get_number(oo, "window-status-bell-attr");
 		if (attr != 0)
 			gc->attr = attr;
 	}
+	else if (wl->flags & WINLINK_CONTENT) {
+		fg = options_get_number(oo, "window-status-content-fg");
+		if (fg != 8)
+			colour_set_fg(gc, fg);
+		bg = options_get_number(oo, "window-status-content-bg");
+		if (bg != 8)
+			colour_set_bg(gc, bg);
+		attr = options_get_number(oo, "window-status-content-attr");
+		if (attr != 0)
+			gc->attr = attr;
+	}
+	else if (wl->flags & WINLINK_ACTIVITY) {
+		fg = options_get_number(oo, "window-status-activity-fg");
+		if (fg != 8)
+			colour_set_fg(gc, fg);
+		bg = options_get_number(oo, "window-status-activity-bg");
+		if (bg != 8)
+			colour_set_bg(gc, bg);
+		attr = options_get_number(oo, "window-status-activity-attr");
+		if (attr != 0)
+			gc->attr = attr;
+	}
+	/* ^ window status patch */
 
 	text = status_replace(c, wl, fmt, t, 1);
 	return (text);
 }
 
 /* Set a status line message. */
 void printflike2
 status_message_set(struct client *c, const char *fmt, ...)
diff -u -8 -p -r1.264 tmux.1
--- tmux.1	17 Jul 2010 14:39:01 -0000	1.264
+++ tmux.1	5 Aug 2010 05:47:24 -0000
@@ -2188,26 +2188,45 @@ Set status line foreground colour for a 
 .It Ic window-status-format Ar string
 Set the format in which the window is displayed in the status line window list.
 See the
 .Ar status-left
 option for details of special character sequences available.
 The default is
 .Ql #I:#W#F .
 .Pp
-.It Ic window-status-alert-attr Ar attributes
-Set status line attributes for windows which have an alert (bell, activity
-or content).
+.\" v window status patch "/.
+.It Ic window-status-activity-attr Ar attributes
+Set status line attributes for windows which have an activity alert.
 .Pp
-.It Ic window-status-alert-bg Ar colour
-Set status line background colour for windows with an alert.
+.It Ic window-status-activity-bg Ar colour
+Set status line background colour for windows with an activity alert.
 .Pp
-.It Ic window-status-alert-fg Ar colour
-Set status line foreground colour for windows with an alert.
+.It Ic window-status-activity-fg Ar colour
+Set status line foreground colour for windows with an activity alert.
 .Pp
+.It Ic window-status-content-attr Ar attributes
+Set status line attributes for windows which have an content alert.
+.Pp
+.It Ic window-status-content-bg Ar colour
+Set status line background colour for windows with an content alert.
+.Pp
+.It Ic window-status-content-fg Ar colour
+Set status line foreground colour for windows with an content alert.
+.Pp
+.It Ic window-status-bell-attr Ar attributes
+Set status line attributes for windows which have a bell alert.
+.Pp
+.It Ic window-status-bell-bg Ar colour
+Set status line background colour for windows with a bell alert.
+.Pp
+.It Ic window-status-bell-fg Ar colour
+Set status line foreground colour for windows with a bell alert.
+.Pp
+.\" ^ window status patch "/.
 .It Ic window-status-current-attr Ar attributes
 Set status line attributes for the currently active window.
 .Pp
 .It Ic window-status-current-bg Ar colour
 Set status line background colour for the currently active window.
 .Pp
 .It Ic window-status-current-fg Ar colour
 Set status line foreground colour for the currently active window.
diff -u -8 -p -r1.214 tmux.c
--- tmux.c	17 Jul 2010 14:36:41 -0000	1.214
+++ tmux.c	5 Aug 2010 05:47:24 -0000
@@ -415,19 +415,27 @@ main(int argc, char **argv)
 	options_set_number(wo, "monitor-activity", 0);
 	options_set_string(wo, "monitor-content", "%s", "");
 	options_set_number(wo, "window-status-attr", 0);
 	options_set_number(wo, "window-status-bg", 8);
 	options_set_number(wo, "window-status-current-attr", 0);
 	options_set_number(wo, "window-status-current-bg", 8);
 	options_set_number(wo, "window-status-current-fg", 8);
 	options_set_number(wo, "window-status-fg", 8);
-	options_set_number(wo, "window-status-alert-attr", GRID_ATTR_REVERSE);
-	options_set_number(wo, "window-status-alert-bg", 8);
-	options_set_number(wo, "window-status-alert-fg", 8);
+	/* v window status patch */
+	options_set_number(wo, "window-status-activity-attr", GRID_ATTR_REVERSE);
+	options_set_number(wo, "window-status-activity-bg", 8);
+	options_set_number(wo, "window-status-activity-fg", 8);
+	options_set_number(wo, "window-status-content-attr", GRID_ATTR_REVERSE);
+	options_set_number(wo, "window-status-content-bg", 8);
+	options_set_number(wo, "window-status-content-fg", 8);
+	options_set_number(wo, "window-status-bell-attr", GRID_ATTR_REVERSE);
+	options_set_number(wo, "window-status-bell-bg", 8);
+	options_set_number(wo, "window-status-bell-fg", 8);
+	/* ^ window status patch */
 	options_set_string(wo, "window-status-format", "#I:#W#F");
 	options_set_string(wo, "window-status-current-format", "#I:#W#F");
 	options_set_string(wo, "word-separators", " -_@");
 	options_set_number(wo, "xterm-keys", 0);
 	options_set_number(wo, "remain-on-exit", 0);
 	options_set_number(wo, "synchronize-panes", 0);
 
 	if (flags & IDENTIFY_UTF8) {
diff -u -8 -p -r1.51 tmux.vim
--- examples/tmux.vim	27 Jul 2010 18:29:07 -0000	1.51
+++ examples/tmux.vim	5 Aug 2010 05:47:28 -0000
@@ -67,18 +67,24 @@ syn keyword tmuxOptsSetw monitor-activit
 syn keyword tmuxOptsSetw force-height remain-on-exit uft8 mode-fg mode-bg
 syn keyword tmuxOptsSetw mode-keys clock-mode-colour clock-mode-style
 syn keyword tmuxOptsSetw xterm-keys mode-attr window-status-attr
 syn keyword tmuxOptsSetw window-status-bg window-status-fg automatic-rename
 syn keyword tmuxOptsSetw main-pane-width main-pane-height monitor-content
 syn keyword tmuxOptsSetw window-status-current-attr window-status-current-bg
 syn keyword tmuxOptsSetw window-status-current-fg mode-mouse synchronize-panes
 syn keyword tmuxOptsSetw window-status-format window-status-current-format
-syn keyword tmuxOptsSetw word-separators window-status-alert-alert
-syn keyword tmuxOptsSetw window-status-alert-bg window-status-alert-fg
+" v window status patch "
+syn keyword tmuxOptsSetw word-separators window-status-activity-attr
+syn keyword tmuxOptsSetw window-status-activity-bg window-status-activity-fg
+syn keyword tmuxOptsSetw word-separators window-status-content-attr
+syn keyword tmuxOptsSetw window-status-content-bg window-status-content-fg
+syn keyword tmuxOptsSetw word-separators window-status-bell-attr
+syn keyword tmuxOptsSetw window-status-bell-bg window-status-bell-fg
+" ^ window status patch "
 
 syn keyword tmuxTodo FIXME NOTE TODO XXX contained
 
 syn match tmuxKey		/\(C-\|M-\|\^\)\p/	display
 syn match tmuxNumber 		/\d\+/			display
 syn match tmuxOptions		/\s-\a\+/		display
 syn match tmuxVariable		/\w\+=/			display
 syn match tmuxVariableExpansion	/\${\=\w\+}\=/		display
