*** Makefile	2010-06-30 17:09:29.000000000 +0200
--- Makefile.new	2010-07-02 17:52:22.304219150 +0200
*************** ELISP_FILES		+= scala-mode-feature
*** 32,37 ****
--- 32,38 ----
  ELISP_FILES		+= scala-mode-feature-electric
  ELISP_FILES		+= scala-mode-feature-speedbar
  ELISP_FILES		+= scala-mode-feature-tags
+ ELISP_FILES		+= scala-mode-variables
  
  ELISP_SOURCES		+= $(ELISP_FILES:%=$(SOURCE_DIR)/%.el)
  
