diff --git a/BuildTools/SCons/SConstruct b/BuildTools/SCons/SConstruct
index cd7a25a..6441553 100644
--- a/BuildTools/SCons/SConstruct
+++ b/BuildTools/SCons/SConstruct
@@ -131,7 +131,7 @@ conf.Finish()
 # Boost
 boost_conf_env = conf_env.Clone()
 boost_flags = {}
-boost_flags["CPPDEFINES"] = [("BOOST_FILESYSTEM_VERSION", "2")]
+boost_flags["CPPDEFINES"] = [("BOOST_FILESYSTEM_VERSION", "3")]
 if env.get("boost_libdir", None) :
 	boost_flags["LIBPATH"] = [env["boost_libdir"]]
 if env.get("boost_includedir", None) :
diff --git a/Swiften/Base/sleep.cpp b/Swiften/Base/sleep.cpp
index 7161217..4c977a1 100644
--- a/Swiften/Base/sleep.cpp
+++ b/Swiften/Base/sleep.cpp
@@ -12,7 +12,7 @@ namespace Swift {
 
 void sleep(unsigned int msecs) {
 	boost::xtime xt;
-	boost::xtime_get(&xt, boost::TIME_UTC);
+	boost::xtime_get(&xt, boost::TIME_UTC_);
 	xt.nsec += msecs*1000000;
 	boost::thread::sleep(xt);
 }
diff --git a/Swiften/FileTransfer/FileTransferManagerImpl.cpp b/Swiften/FileTransfer/FileTransferManagerImpl.cpp
index 853738c..4137ddd 100644
--- a/Swiften/FileTransfer/FileTransferManagerImpl.cpp
+++ b/Swiften/FileTransfer/FileTransferManagerImpl.cpp
@@ -103,7 +103,7 @@ boost::optional<JID> FileTransferManagerImpl::highestPriorityJIDSupportingFileTr
 }
 
 OutgoingFileTransfer::ref FileTransferManagerImpl::createOutgoingFileTransfer(const JID& to, const boost::filesystem::path& filepath, const std::string& description, boost::shared_ptr<ReadBytestream> bytestream) {
-	std::string filename = filepath.filename();
+	std::string filename = filepath.filename().string();
 	boost::uintmax_t sizeInBytes = boost::filesystem::file_size(filepath);
 	boost::posix_time::ptime lastModified = boost::posix_time::from_time_t(boost::filesystem::last_write_time(filepath));
 	return createOutgoingFileTransfer(to, filename, description, sizeInBytes, lastModified, bytestream);
