diff --git a/common/profile-sync-daemon.in b/common/profile-sync-daemon.in
index ecc8758..37955a6 100644
--- a/common/profile-sync-daemon.in
+++ b/common/profile-sync-daemon.in
@@ -52,6 +52,7 @@ opera-beta
 opera-next
 otter-browser
 palemoon
+pycharm
 qupzilla
 rekonq
 seamonkey
@@ -216,6 +217,9 @@ config_check() {
 			palemoon)
 				return
 				;;
+			pycharm)
+				return
+				;;
 			vivaldi)
 				return
 				;;
@@ -357,6 +361,10 @@ set_which() {
 			DIRArr[0]="$homedir/.config/$browser"
 			PSNAME="$browser"
 			;;
+		pycharm)
+			DIRArr[0]="$homedir/.PyCharm40"
+			PSNAME="$browser"
+			;;
 		vivaldi)
 			DIRArr[0]="$homedir/.config/$browser"
 			PSNAME="$browser"-bin
diff --git a/common/psd.conf b/common/psd.conf
index da0281a..767f54c 100644
--- a/common/psd.conf
+++ b/common/psd.conf
@@ -40,6 +40,7 @@ USERS=""
 #  otter-browser
 #  qupzilla
 #  palemoon
+#  pycharm
 #  rekonq
 #  seamonkey
 #  vivaldi
