diff -urN ./opt/kde/share/applnk/Utilities/K-EncFS-v2.desktop ../new/opt/kde/share/applnk/Utilities/K-EncFS-v2.desktop
--- ./opt/kde/share/applnk/Utilities/K-EncFS-v2.desktop	2007-03-20 22:44:26.000000000 +0300
+++ ../new/opt/kde/share/applnk/Utilities/K-EncFS-v2.desktop	2008-04-24 09:53:07.000000000 +0400
@@ -5,6 +5,6 @@
 Type=Application
 Terminal=false
 Encoding=UTF-8
-Icon=/usr/local/lib/kencfs2/file_locked.png
+Icon=/opt/kde/lib/kencfs2/file_locked.png
 Comment=Encryption utility for KDE
 Exec=kencfs2
diff -urN ./usr/local/lib/kencfs2/help.html ../new/usr/local/lib/kencfs2/help.html
--- ./usr/local/lib/kencfs2/help.html	2007-12-16 20:50:21.000000000 +0300
+++ ../new/usr/local/lib/kencfs2/help.html	2008-04-24 09:54:48.000000000 +0400
@@ -104,7 +104,7 @@
 <P CLASS="western">It will be trayed after exiting the main window.
 (See the lock icon!) Left-clicking on the tray icon, the main window
 will appear. 
-<br><br>If you choose the <FONT COLOR="#008000">'Login during KDE startup'</FONT> option, K-EncFS asks your password after KDE login if the file system has not been mounted yet. This function requires editing the 'startkde' script. You can find more information about this in the file /usr/local/lib/kencfs2/startkde/readme.startkde.</P>
+<br><br>If you choose the <FONT COLOR="#008000">'Login during KDE startup'</FONT> option, K-EncFS asks your password after KDE login if the file system has not been mounted yet. This function requires editing the 'startkde' script. You can find more information about this in the file /opt/kde/lib/kencfs2/startkde/readme.startkde.</P>
 <P CLASS="western"><BR>
 </P>
 <H2><B>Options and usage</B></H2>
diff -urN ./usr/local/lib/kencfs2/helpconnect.py ../new/usr/local/lib/kencfs2/helpconnect.py
--- ./usr/local/lib/kencfs2/helpconnect.py	2007-03-20 21:44:40.000000000 +0300
+++ ../new/usr/local/lib/kencfs2/helpconnect.py	2008-04-24 09:53:51.000000000 +0400
@@ -25,7 +25,7 @@
     def __init__(self, parent=None):
         Help.__init__(self, parent)
 
-        helpdata=self.load("/usr/local/lib/kencfs2/help.html")
+        helpdata=self.load("/opt/kde/lib/kencfs2/help.html")
         self.textEdit1.setText(helpdata)
 
         self.setFixedSize(646, 570)
diff -urN ./usr/local/lib/kencfs2/kencfs.py ../new/usr/local/lib/kencfs2/kencfs.py
--- ./usr/local/lib/kencfs2/kencfs.py	2007-12-16 21:04:51.000000000 +0300
+++ ../new/usr/local/lib/kencfs2/kencfs.py	2008-04-24 09:54:38.000000000 +0400
@@ -326,7 +326,7 @@
 
     def slotShow(self):
         if self.checkHideaddress.isChecked():
-            os.system("konqueror --profile /usr/local/lib/kencfs2/encryptedprofil " + self.user_encrypted +" &")
+            os.system("konqueror --profile /opt/kde/lib/kencfs2/encryptedprofil " + self.user_encrypted +" &")
         else:
             os.system("konqueror " + self.user_encrypted +" &")
 
@@ -401,7 +401,7 @@
             output = os.popen("cat $(which startkde) | grep kencfspass.py", "r").readline()
             print output
             if output=="":
-                QMessageBox.warning(self, "K-EncFS: Warning!", "The file 'startkde' has not been patched yet!\n\nPlease edit this script file manually according to\nthe instructions can be found in the following file:\n/usr/local/lib/kencfs2/startkde/readme.startkde!\n")
+                QMessageBox.warning(self, "K-EncFS: Warning!", "The file 'startkde' has not been patched yet!\n\nPlease edit this script file manually according to\nthe instructions can be found in the following file:\n/opt/kde/lib/kencfs2/startkde/readme.startkde!\n")
 
         else:
             try:
@@ -544,7 +544,7 @@
 #        self.systray.changeTitle(1, QString("text"))
 #        self.systray.ctxMenuHiding()
 
-        self.systray.setPixmap (icons.loadIcon("/usr/local/lib/kencfs2/file_locked.png", KIcon.Toolbar))
+        self.systray.setPixmap (icons.loadIcon("/opt/kde/lib/kencfs2/file_locked.png", KIcon.Toolbar))
 # other versions:
 #        self.systray.setPixmap (icons.loadIcon("/usr/local/lib/kencfs2/file_locked.png", KIcon.MainToolbar))
 #        self.systray.changeTitle(1, QString("text"))
@@ -597,7 +597,7 @@
 desc = 'A frontend to encfs'
 ver = 'Encrypted FS FE 2.1'
 
-sys.argv.extend(['--icon', '/usr/local/lib/kencfs2/file_locked.png', '--caption', ver])
+sys.argv.extend(['--icon', '/opt/kde/lib/kencfs2/file_locked.png', '--caption', ver])
 
 about = KAboutData(appName, ver, desc)
 KCmdLineArgs.init(sys.argv, about)
diff -urN ./usr/local/lib/kencfs2/startkde/readme.startkde ../new/usr/local/lib/kencfs2/startkde/readme.startkde
--- ./usr/local/lib/kencfs2/startkde/readme.startkde	2007-12-16 14:57:23.000000000 +0300
+++ ../new/usr/local/lib/kencfs2/startkde/readme.startkde	2008-04-24 09:53:34.000000000 +0400
@@ -12,9 +12,9 @@
 .
 .
 else # inserted line 1
-if [ -e /usr/local/lib/kencfs2/kencfspass_ui.py ] && [ -e ~/.kencfs2/start_with_kde.flag ] # inserted line 2
+if [ -e /opt/kde/lib/kencfs2/kencfspass_ui.py ] && [ -e ~/.kencfs2/start_with_kde.flag ] # inserted line 2
 then # inserted line 3
-python /usr/local/lib/kencfs2/kencfspass.py # inserted line 4
+python /opt/kde/lib/kencfs2/kencfspass.py # inserted line 4
 fi # inserted line 5
 
 fi # last fi
