--- gst-libs/gst/vaapi/gstvaapidecoder_ffmpeg.c	2011-07-08 01:18:06.000000000 +0300
+++ gst-libs/gst/vaapi/gstvaapidecoder_ffmpeg.c.new	2011-07-08 01:53:13.170179549 +0300
@@ -1,4 +1,4 @@
-/*
+/**
  *  gstvaapidecoder_ffmpeg.c - FFmpeg-based decoder
  *
  *  gstreamer-vaapi (C) 2010-2011 Splitted-Desktop Systems
@@ -480,11 +480,18 @@
     int bytes_read, got_picture = 0;
 
     GST_VAAPI_DISPLAY_LOCK(display);
-    bytes_read = avcodec_decode_video(
+
+     AVPacket packet;
+     av_init_packet(&packet);
+     packet.data = buf;
+     packet.size = buf_size;
+
+
+    bytes_read = avcodec_decode_video2(
         priv->avctx,
         priv->frame,
         &got_picture,
-        buf, buf_size
+	&packet
     );
     GST_VAAPI_DISPLAY_UNLOCK(display);
     if (!got_picture)
@@ -530,12 +537,12 @@
 
     if (priv->pctx) {
         do {
-            int parsed_size = av_parser_parse(
+            int parsed_size = av_parser_parse2(
                 priv->pctx,
                 priv->avctx,
                 &outbuf, &outbuf_size,
                 GST_BUFFER_DATA(buffer) + inbuf_ofs, inbuf_size,
-                inbuf_ts, inbuf_ts
+                inbuf_ts, inbuf_ts, AV_NOPTS_VALUE
             );
             got_frame = outbuf && outbuf_size > 0;
 
