--- a/exfat_super.c	2013-09-07 13:19:29.181343133 +0200
+++ b/exfat_super.c	2013-09-07 13:16:44.442345059 +0200
@@ -276,9 +276,15 @@
 #endif
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,11,0)
+static int exfat_readdir(struct file *filp, struct dir_context *ctx)
+{
+	struct inode *inode = file_inode(filp);
+#else
 static int exfat_readdir(struct file *filp, void *dirent, filldir_t filldir)
 {
 	struct inode *inode = filp->f_path.dentry->d_inode;
+#endif
 	struct super_block *sb = inode->i_sb;
 	struct exfat_sb_info *sbi = EXFAT_SB(sb);
 	FS_INFO_T *p_fs = &(sbi->fs_info);
@@ -289,8 +295,11 @@
 	int err = 0;
 
 	__lock_super(sb);
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,11,0)
+	cpos = ctx->pos;
+#else
 	cpos = filp->f_pos;
+#endif
 	if ((p_fs->vol_type == EXFAT) || (inode->i_ino == EXFAT_ROOT_INO)) {
 		while (cpos < 2) {
 			if (inode->i_ino == EXFAT_ROOT_INO)
@@ -299,12 +308,19 @@
 				inum = inode->i_ino;
 			else
 				inum = parent_ino(filp->f_path.dentry);
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,11,0)
+			if (!dir_emit_dots(filp, ctx))
+				goto out;
+			cpos++;
+			ctx->pos++;
+		}
+#else
 			if (filldir(dirent, "..", cpos+1, cpos, inum, DT_DIR) < 0)
 				goto out;
 			cpos++;
 			filp->f_pos++;
 		}
+#endif
 		if (cpos == 2) {
 			cpos = 0;
 		}
@@ -350,14 +366,20 @@
 			inum = iunique(sb, EXFAT_ROOT_INO);
 		}
 	}
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,11,0)
+	if (!dir_emit(ctx, de.Name, strlen(de.Name), inum,
+			(de.Attr & ATTR_SUBDIR) ? DT_DIR : DT_REG))
+		goto out;
 
+	ctx->pos = cpos;
+#else
 	if (filldir(dirent, de.Name, strlen(de.Name), cpos-1, inum,
 				(de.Attr & ATTR_SUBDIR) ? DT_DIR : DT_REG) < 0)
 		goto out;
 
 	filp->f_pos = cpos;
+#endif
 	goto get_new;
-
 end_of_dir:
 	filp->f_pos = cpos;
 out:
@@ -439,7 +461,11 @@
 const struct file_operations exfat_dir_operations = {
 	.llseek     = generic_file_llseek,
 	.read       = generic_read_dir,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,11,0)
+	.iterate    = exfat_readdir,
+#else
 	.readdir    = exfat_readdir,
+#endif
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,36)
 	.ioctl      = exfat_generic_ioctl,
 	.fsync      = exfat_file_fsync,
