diff -aur tmda-cgi-0.16.4/Authenticate.py tmda-cgi-0.16.4-fixed/Authenticate.py
--- tmda-cgi-0.16.4/Authenticate.py	2004-02-02 10:08:35.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/Authenticate.py	2010-12-08 04:36:57.494301441 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/CgiUtil.py tmda-cgi-0.16.4-fixed/CgiUtil.py
--- tmda-cgi-0.16.4/CgiUtil.py	2006-10-02 17:42:20.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/CgiUtil.py	2010-12-08 04:36:57.460966651 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/CharSetAlias.py tmda-cgi-0.16.4-fixed/CharSetAlias.py
--- tmda-cgi-0.16.4/CharSetAlias.py	2003-05-12 12:59:23.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/CharSetAlias.py	2010-12-08 04:36:58.654352051 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Jim Ramsay <i.am@jimramsay.com>
 #
diff -aur tmda-cgi-0.16.4/configure tmda-cgi-0.16.4-fixed/configure
--- tmda-cgi-0.16.4/configure	2006-09-21 13:49:51.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/configure	2010-12-08 04:36:55.460879394 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/EditFilter.py tmda-cgi-0.16.4-fixed/EditFilter.py
--- tmda-cgi-0.16.4/EditFilter.py	2004-03-18 12:54:17.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/EditFilter.py	2010-12-08 04:36:58.681019882 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/EditList.py tmda-cgi-0.16.4-fixed/EditList.py
--- tmda-cgi-0.16.4/EditList.py	2005-03-24 09:25:29.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/EditList.py	2010-12-08 04:36:55.237535891 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/GenAddr.py tmda-cgi-0.16.4-fixed/GenAddr.py
--- tmda-cgi-0.16.4/GenAddr.py	2004-03-18 12:54:17.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/GenAddr.py	2010-12-08 04:36:55.487547225 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/genpass.py tmda-cgi-0.16.4-fixed/genpass.py
--- tmda-cgi-0.16.4/genpass.py	2003-03-29 17:46:27.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/genpass.py	2010-12-08 04:36:57.480967526 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/GlobalConfig.py tmda-cgi-0.16.4-fixed/GlobalConfig.py
--- tmda-cgi-0.16.4/GlobalConfig.py	2004-03-18 12:54:17.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/GlobalConfig.py	2010-12-08 04:36:55.250870454 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/ht2html/calcroot.py tmda-cgi-0.16.4-fixed/ht2html/calcroot.py
--- tmda-cgi-0.16.4/ht2html/calcroot.py	2003-04-13 11:58:08.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/ht2html/calcroot.py	2010-12-08 04:36:58.634351177 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 #
 # simple script to calculate HTROOT from ROOT_OFFSET.  Pass ROOT_OFFSET in on
 # the command line
diff -aur tmda-cgi-0.16.4/ht2html/ht2html.py tmda-cgi-0.16.4-fixed/ht2html/ht2html.py
--- tmda-cgi-0.16.4/ht2html/ht2html.py	2003-04-13 11:58:08.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/ht2html/ht2html.py	2010-12-08 04:36:58.631017700 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 
 """Generate a .html file from a .ht template.
 
diff -aur tmda-cgi-0.16.4/Info.py tmda-cgi-0.16.4-fixed/Info.py
--- tmda-cgi-0.16.4/Info.py	2004-03-18 12:54:17.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/Info.py	2010-12-08 04:36:55.454212436 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/Install.py tmda-cgi-0.16.4-fixed/Install.py
--- tmda-cgi-0.16.4/Install.py	2004-03-19 17:08:16.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/Install.py	2010-12-08 04:36:58.661019006 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/LocalConfig.py tmda-cgi-0.16.4-fixed/LocalConfig.py
--- tmda-cgi-0.16.4/LocalConfig.py	2005-03-24 10:49:14.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/LocalConfig.py	2010-12-08 04:36:55.494214183 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/Login.py tmda-cgi-0.16.4-fixed/Login.py
--- tmda-cgi-0.16.4/Login.py	2003-05-04 16:26:25.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/Login.py	2010-12-08 04:36:58.674352926 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/override.py tmda-cgi-0.16.4-fixed/override.py
--- tmda-cgi-0.16.4/override.py	2003-05-03 23:26:48.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/override.py	2010-12-08 04:36:58.427675494 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/PendList.py tmda-cgi-0.16.4-fixed/PendList.py
--- tmda-cgi-0.16.4/PendList.py	2007-03-29 14:48:14.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/PendList.py	2010-12-08 04:36:57.347628377 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/Release.py tmda-cgi-0.16.4-fixed/Release.py
--- tmda-cgi-0.16.4/Release.py	2006-10-02 16:30:11.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/Release.py	2010-12-08 04:36:55.230868424 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/reporter.py tmda-cgi-0.16.4-fixed/reporter.py
--- tmda-cgi-0.16.4/reporter.py	2004-04-08 11:05:05.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/reporter.py	2010-12-08 04:36:55.540882886 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python -O
+#!/usr/bin/env python2 -O
 
 """SpamCop Reporting Program. 
 
diff -aur tmda-cgi-0.16.4/Session.py tmda-cgi-0.16.4-fixed/Session.py
--- tmda-cgi-0.16.4/Session.py	2007-04-16 12:29:01.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/Session.py	2010-12-08 04:36:55.557550276 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/stubs/qmailuserassign.py tmda-cgi-0.16.4-fixed/stubs/qmailuserassign.py
--- tmda-cgi-0.16.4/stubs/qmailuserassign.py	2003-05-06 19:19:38.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/stubs/qmailuserassign.py	2010-12-08 04:36:55.514215056 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/stubs/vmailmgr1.py tmda-cgi-0.16.4-fixed/stubs/vmailmgr1.py
--- tmda-cgi-0.16.4/stubs/vmailmgr1.py	2007-03-29 12:34:09.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/stubs/vmailmgr1.py	2010-12-08 04:36:55.534215925 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 John Maslanik <maz@mlx.net>
 #
diff -aur tmda-cgi-0.16.4/stubs/vmailmgrinfo.py tmda-cgi-0.16.4-fixed/stubs/vmailmgrinfo.py
--- tmda-cgi-0.16.4/stubs/vmailmgrinfo.py	2004-02-02 15:02:06.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/stubs/vmailmgrinfo.py	2010-12-08 04:36:55.507548095 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 John Maslanik <maz@mlx.net>
 #
diff -aur tmda-cgi-0.16.4/stubs/vpopmail1.py tmda-cgi-0.16.4-fixed/stubs/vpopmail1.py
--- tmda-cgi-0.16.4/stubs/vpopmail1.py	2003-12-09 11:41:15.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/stubs/vpopmail1.py	2010-12-08 04:36:55.520882015 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/stubs/xams1.py tmda-cgi-0.16.4-fixed/stubs/xams1.py
--- tmda-cgi-0.16.4/stubs/xams1.py	2003-10-04 17:58:26.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/stubs/xams1.py	2010-12-08 04:36:55.527548968 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/Templates.py tmda-cgi-0.16.4-fixed/Templates.py
--- tmda-cgi-0.16.4/Templates.py	2004-03-18 12:54:17.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/Templates.py	2010-12-08 04:36:55.437545041 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/TestAddr.py tmda-cgi-0.16.4-fixed/TestAddr.py
--- tmda-cgi-0.16.4/TestAddr.py	2004-03-18 12:54:17.000000000 -0500
+++ tmda-cgi-0.16.4-fixed/TestAddr.py	2010-12-08 04:36:55.547549845 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/Theme.py tmda-cgi-0.16.4-fixed/Theme.py
--- tmda-cgi-0.16.4/Theme.py	2006-10-02 16:01:16.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/Theme.py	2010-12-08 04:36:57.504301876 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2003 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/tmda-cgi.py tmda-cgi-0.16.4-fixed/tmda-cgi.py
--- tmda-cgi-0.16.4/tmda-cgi.py	2010-12-08 04:36:08.302154794 -0500
+++ tmda-cgi-0.16.4-fixed/tmda-cgi.py	2010-12-08 04:36:57.524302750 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/Unicode.py tmda-cgi-0.16.4-fixed/Unicode.py
--- tmda-cgi-0.16.4/Unicode.py	2004-05-04 11:45:18.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/Unicode.py	2010-12-08 04:36:55.467546354 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
diff -aur tmda-cgi-0.16.4/View.py tmda-cgi-0.16.4-fixed/View.py
--- tmda-cgi-0.16.4/View.py	2006-10-11 12:53:24.000000000 -0400
+++ tmda-cgi-0.16.4-fixed/View.py	2010-12-08 04:36:55.447545481 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright (C) 2002 Gre7g Luterman <gre7g@wolfhome.com>
 #
