--- icu/source/configure.in.orig	2012-02-10 22:49:57.096591442 +0100
+++ icu/source/configure.in	2012-02-10 22:50:53.066621409 +0100
@@ -629,6 +629,12 @@
 ia64-*-hpux*)
     GENCCODE_ASSEMBLY="-a aCC-ia64"
     ;;
+i686-*-mingw32)
+    GENCCODE_ASSEMBLY="-a gcc-mingw-x86"
+    ;;
+x86_64-*-mingw32)
+    GENCCODE_ASSEMBLY="-a gcc-mingw-x64"
+    ;;
 esac
 AC_SUBST(GENCCODE_ASSEMBLY)
 
--- icu/source/config/mh-mingw.orig	2012-02-11 00:25:43.637578250 +0100
+++ icu/source/config/mh-mingw	2012-02-11 00:29:08.051465305 +0100
@@ -132,9 +132,6 @@
 # Environment variable to set a runtime search path
 LDLIBRARYPATH_ENVVAR = PATH
 
-# The type of assembly to write for generating an object file
-GENCCODE_ASSEMBLY=-a gcc-cygwin
-
 # These are needed to allow the pkgdata GNU make files to work
 PKGDATA_DEFS = -DU_MAKE=\"$(MAKE)\"
 
