--- a/kiwi
+++ b/kiwi
@@ -1,11 +1,6 @@
 #!/bin/sh
-basedir=`dirname "$0"`
 
-case `uname` in
-    *CYGWIN*) basedir=`cygpath -w "$basedir"`;;
-esac
-
-node $basedir/server/server.js "$@"
+node /usr/lib/kiwiirc/server/server.js "$@"
 ret=$?
 
 exit $ret

--- a/config.example.js
+++ b/config.example.js
@@ -6,7 +6,7 @@
 
 
 // Log file location
-conf.log = "kiwi.log";
+conf.log = "/var/log/kiwiirc.log";
 
 
 
