--- src/ephy-session.c~	2006-02-14 07:12:33.000000000 +0800
+++ src/ephy-session.c	2006-02-25 02:04:42.000000000 +0800
@@ -563,6 +563,19 @@
 	crashed_session = g_file_test (saved_session, G_FILE_TEST_EXISTS);
 	g_free (saved_session);
 
+	if (crashed_session)
+        {
+		ephy_session_queue_command (session,
+				  	    EPHY_SESSION_CMD_LOAD_SESSION,
+					    SESSION_CRASHED, NULL,
+					    user_time, TRUE);
+       	        ephy_session_queue_command (session,
+				            EPHY_SESSION_CMD_MAYBE_OPEN_WINDOW,
+				            NULL, NULL, user_time, FALSE);
+                return;
+        }
+
+
 	if (crashed_session == FALSE ||
 	    priv->windows != NULL ||
 	    priv->tool_windows != NULL)
