diff --git a/gdal/frmts/gif/biggifdataset.cpp b/gdal/frmts/gif/biggifdataset.cpp
index 69e7d76..21f4bb3 100644
--- a/gdal/frmts/gif/biggifdataset.cpp
+++ b/gdal/frmts/gif/biggifdataset.cpp
@@ -339,7 +339,7 @@ CPLErr BIGGIFDataset::ReOpen()
 /*      If the file is already open, close it so we can restart.        */
 /* -------------------------------------------------------------------- */
     if( hGifFile != NULL )
-        DGifCloseFile( hGifFile );
+        GIFAbstractDataset::myDGifCloseFile( hGifFile );
 
 /* -------------------------------------------------------------------- */
 /*      If we are actually reopening, then we assume that access to     */
@@ -373,12 +373,7 @@ CPLErr BIGGIFDataset::ReOpen()
     VSIFSeekL( fp, 0, SEEK_SET );
 
     nLastLineRead = -1;
-#if defined(GIFLIB_MAJOR) && GIFLIB_MAJOR >= 5
-    int nError;
-    hGifFile = DGifOpen( fp, VSIGIFReadFunc, &nError );
-#else
-    hGifFile = DGifOpen( fp, VSIGIFReadFunc );
-#endif
+    hGifFile = GIFAbstractDataset::myDGifOpen( fp, VSIGIFReadFunc );
     if( hGifFile == NULL )
     {
         CPLError( CE_Failure, CPLE_OpenFailed, 
@@ -413,7 +408,7 @@ CPLErr BIGGIFDataset::ReOpen()
 
     if( RecordType != IMAGE_DESC_RECORD_TYPE )
     {
-        DGifCloseFile( hGifFile );
+        GIFAbstractDataset::myDGifCloseFile( hGifFile );
         hGifFile = NULL;
 
         CPLError( CE_Failure, CPLE_OpenFailed, 
@@ -423,7 +418,7 @@ CPLErr BIGGIFDataset::ReOpen()
     
     if (DGifGetImageDesc(hGifFile) == GIF_ERROR)
     {
-        DGifCloseFile( hGifFile );
+        GIFAbstractDataset::myDGifCloseFile( hGifFile );
         hGifFile = NULL;
 
         CPLError( CE_Failure, CPLE_OpenFailed, 
diff --git a/gdal/frmts/gif/gifabstractdataset.cpp b/gdal/frmts/gif/gifabstractdataset.cpp
index 250a8fe..e348f31 100644
--- a/gdal/frmts/gif/gifabstractdataset.cpp
+++ b/gdal/frmts/gif/gifabstractdataset.cpp
@@ -82,7 +82,7 @@ GIFAbstractDataset::~GIFAbstractDataset()
     }
 
     if( hGifFile )
-        DGifCloseFile( hGifFile );
+        myDGifCloseFile( hGifFile );
 
     if( fp != NULL )
         VSIFCloseL( fp );
@@ -336,3 +336,45 @@ void GIFAbstractDataset::DetectGeoreferencing( GDALOpenInfo * poOpenInfo )
                                adfGeoTransform, poOpenInfo->GetSiblingFiles(), NULL );
     }
 }
+
+/************************************************************************/
+/*                            myDGifOpen()                              */
+/************************************************************************/
+
+GifFileType* GIFAbstractDataset::myDGifOpen( void *userPtr, InputFunc readFunc )
+{
+#if defined(GIFLIB_MAJOR) && GIFLIB_MAJOR >= 5
+    int nErrorCode;
+    return DGifOpen( userPtr, readFunc, &nErrorCode );
+#else
+    return DGifOpen( userPtr, readFunc );
+#endif
+}
+
+/************************************************************************/
+/*                          myDGifCloseFile()                           */
+/************************************************************************/
+
+int GIFAbstractDataset::myDGifCloseFile( GifFileType *hGifFile )
+{
+#if defined(GIFLIB_MAJOR) && ((GIFLIB_MAJOR == 5 && GIFLIB_MINOR >= 1) || GIFLIB_MAJOR > 5)
+    int nErrorCode;
+    return DGifCloseFile( hGifFile, &nErrorCode );
+#else
+    return DGifCloseFile( hGifFile );
+#endif
+}
+
+/************************************************************************/
+/*                          myEGifCloseFile()                           */
+/************************************************************************/
+
+int GIFAbstractDataset::myEGifCloseFile( GifFileType *hGifFile )
+{
+#if defined(GIFLIB_MAJOR) && ((GIFLIB_MAJOR == 5 && GIFLIB_MINOR >= 1) || GIFLIB_MAJOR > 5)
+    int nErrorCode;
+    return EGifCloseFile( hGifFile, &nErrorCode );
+#else
+    return EGifCloseFile( hGifFile );
+#endif
+}
diff --git a/gdal/frmts/gif/gifabstractdataset.h b/gdal/frmts/gif/gifabstractdataset.h
index c70b0b0..282b881 100644
--- a/gdal/frmts/gif/gifabstractdataset.h
+++ b/gdal/frmts/gif/gifabstractdataset.h
@@ -75,6 +75,10 @@ class GIFAbstractDataset : public GDALPamDataset
     virtual char  **GetMetadata( const char * pszDomain = "" );
 
     static int          Identify( GDALOpenInfo * );
+
+    static GifFileType* myDGifOpen( void *userPtr, InputFunc readFunc );
+    static int          myDGifCloseFile( GifFileType *hGifFile );
+    static int          myEGifCloseFile( GifFileType *hGifFile );
 };
 
 
diff --git a/gdal/frmts/gif/gifdataset.cpp b/gdal/frmts/gif/gifdataset.cpp
index 1523371..ead9d8e 100644
--- a/gdal/frmts/gif/gifdataset.cpp
+++ b/gdal/frmts/gif/gifdataset.cpp
@@ -331,12 +331,7 @@ GDALDataset *GIFDataset::Open( GDALOpenInfo * poOpenInfo )
     fp = poOpenInfo->fpL;
     poOpenInfo->fpL = NULL;
 
-#if defined(GIFLIB_MAJOR) && GIFLIB_MAJOR >= 5
-    int nError;
-    hGifFile = DGifOpen( fp, VSIGIFReadFunc, &nError );
-#else
-    hGifFile = DGifOpen( fp, VSIGIFReadFunc );
-#endif
+    hGifFile = GIFAbstractDataset::myDGifOpen( fp, VSIGIFReadFunc );
     if( hGifFile == NULL )
     {
         VSIFCloseL( fp );
@@ -385,7 +380,7 @@ GDALDataset *GIFDataset::Open( GDALOpenInfo * poOpenInfo )
             CPLDebug( "GIF",
                       "Due to limitations of the GDAL GIF driver we deliberately avoid\n"
                       "opening large GIF files (larger than 100 megapixels).");
-            DGifCloseFile( hGifFile );
+            GIFAbstractDataset::myDGifCloseFile( hGifFile );
             VSIFCloseL( fp );
             return NULL;
     }
@@ -393,15 +388,11 @@ GDALDataset *GIFDataset::Open( GDALOpenInfo * poOpenInfo )
         }
     }
 
-    DGifCloseFile( hGifFile );
+    GIFAbstractDataset::myDGifCloseFile( hGifFile );
 
     VSIFSeekL( fp, 0, SEEK_SET);
 
-#if defined(GIFLIB_MAJOR) && GIFLIB_MAJOR >= 5
-    hGifFile = DGifOpen( fp, VSIGIFReadFunc, &nError );
-#else
-    hGifFile = DGifOpen( fp, VSIGIFReadFunc );
-#endif
+    hGifFile = GIFAbstractDataset::myDGifOpen( fp, VSIGIFReadFunc );
     if( hGifFile == NULL )
     {
         VSIFCloseL( fp );
@@ -418,7 +409,7 @@ GDALDataset *GIFDataset::Open( GDALOpenInfo * poOpenInfo )
     if( nGifErr != GIF_OK || hGifFile->SavedImages == NULL )
     {
         VSIFCloseL( fp );
-        DGifCloseFile(hGifFile);
+        GIFAbstractDataset::myDGifCloseFile(hGifFile);
 
         if( nGifErr == D_GIF_ERR_DATA_TOO_BIG )
         {
@@ -651,7 +642,7 @@ GIFDataset::CreateCopy( const char * pszFilename, GDALDataset *poSrcDS,
     {
         GifFreeMapObject(psGifCT);
         GDALPrintGifError(hGifFile, "Error writing gif file.");
-        EGifCloseFile(hGifFile);
+        GIFAbstractDataset::myEGifCloseFile(hGifFile);
         VSIFCloseL( fp );
         return NULL;
     }
@@ -675,7 +666,7 @@ GIFDataset::CreateCopy( const char * pszFilename, GDALDataset *poSrcDS,
     if (EGifPutImageDesc(hGifFile, 0, 0, nXSize, nYSize, bInterlace, NULL) == GIF_ERROR )
     {
         GDALPrintGifError(hGifFile, "Error writing gif file.");
-        EGifCloseFile(hGifFile);
+        GIFAbstractDataset::myEGifCloseFile(hGifFile);
         VSIFCloseL( fp );
         return NULL;
     }
@@ -758,7 +749,7 @@ GIFDataset::CreateCopy( const char * pszFilename, GDALDataset *poSrcDS,
 /* -------------------------------------------------------------------- */
 /*      cleanup                                                         */
 /* -------------------------------------------------------------------- */
-    if (EGifCloseFile(hGifFile) == GIF_ERROR)
+    if (GIFAbstractDataset::myEGifCloseFile(hGifFile) == GIF_ERROR)
     {
         CPLError( CE_Failure, CPLE_AppDefined, 
                   "EGifCloseFile() failed.\n" );
@@ -809,7 +800,7 @@ GIFDataset::CreateCopy( const char * pszFilename, GDALDataset *poSrcDS,
 
 error:
     if (hGifFile)
-        EGifCloseFile(hGifFile);
+        GIFAbstractDataset::myEGifCloseFile(hGifFile);
     if (fp)
         VSIFCloseL( fp );
     if (pabyScanline)
