diff -aur gazebo-1.0.1.orig/cmake/SearchForStuff.cmake gazebo-1.0.1/cmake/SearchForStuff.cmake
--- gazebo-1.0.1.orig/cmake/SearchForStuff.cmake	2012-05-15 23:35:14.000000000 +0600
+++ gazebo-1.0.1/cmake/SearchForStuff.cmake	2012-05-18 19:03:00.053364097 +0600
@@ -127,14 +127,14 @@
 
       message(STATUS "tinyxml has no passed in paths, try to auto detect manually.")
 
-      find_path(tinyxml_include_dir tinyxml/tinyxml.hpp ${tinyxml_include_dirs} ENV CPATH)
+      find_path(tinyxml_include_dir tinyxml.h ${tinyxml_include_dirs} ENV CPATH)
       
       if (NOT tinyxml_include_dir)
-        message (STATUS "Looking for tinyxml/tinyxml.hpp - not found.")
+        message (STATUS "Looking for tinyxml.h - not found.")
         set (tinyxml_include_dirs /usr/include CACHE STRING
           "tinyxml include paths. Use this to override automatic detection.")
       else (NOT tinyxml_include_dir)
-        message (STATUS "Looking for tinyxml/tinyxml.hpp - found")
+        message (STATUS "Looking for tinyxml.h - found")
         set (assim_include_dirs ${tinyxml_include_dir} CACHE STRING
           "tinyxml include paths. Use this to override automatic detection.")
       endif (NOT tinyxml_include_dir)
@@ -166,10 +166,10 @@
 
   #################################################
   # Find TBB
-  pkg_check_modules(TBB tbb)
-  IF (NOT TBB_FOUND)
-    BUILD_ERROR ("Missing: TBB - Threading Building Blocks")
-  ENDIF (NOT TBB_FOUND)
+#  pkg_check_modules(TBB tbb)
+#  IF (NOT TBB_FOUND)
+#    BUILD_ERROR ("Missing: TBB - Threading Building Blocks")
+#  ENDIF (NOT TBB_FOUND)
 
   #################################################
   # Find OGRE 
diff -aur gazebo-1.0.1.orig/src/CMakeLists.txt gazebo-1.0.1/src/CMakeLists.txt
--- gazebo-1.0.1.orig/src/CMakeLists.txt	2012-05-15 23:35:21.000000000 +0600
+++ gazebo-1.0.1/src/CMakeLists.txt	2012-05-18 19:05:18.796698215 +0600
@@ -47,6 +47,7 @@
                                gazebo_transport
                                gazebo_sdf_interface
                                gazebo_msgs
+                               pthread
                                ${boost_libraries}
                                ${freeimage_library}
                                ${PROTOBUF_LIBRARY})
diff -aur gazebo-1.0.1.orig/tools/CMakeLists.txt gazebo-1.0.1/tools/CMakeLists.txt
--- gazebo-1.0.1.orig/tools/CMakeLists.txt	2012-05-15 23:35:14.000000000 +0600
+++ gazebo-1.0.1/tools/CMakeLists.txt	2012-05-18 19:03:00.060030762 +0600
@@ -10,22 +10,22 @@
 add_definitions(${tinyxml_cflags})
 
 add_executable( gzsdf gzsdf.cc )
-target_link_libraries(gzsdf ${tinyxml_libraries} gazebo_sdf_interface)
+target_link_libraries(gzsdf ${tinyxml_libraries} gazebo_sdf_interface pthread)
 INSTALL_EXECUTABLE( gzsdf )
 
 add_executable( gztopic gztopic.cc )
-target_link_libraries(gztopic gazebo_msgs gazebo_common gazebo_transport )
+target_link_libraries(gztopic gazebo_msgs gazebo_common gazebo_transport pthread)
 INSTALL_EXECUTABLE( gztopic )
 
 add_executable( gzstats gzstats.cc )
-target_link_libraries(gzstats gazebo_msgs gazebo_common gazebo_transport )
+target_link_libraries(gzstats gazebo_msgs gazebo_common gazebo_transport pthread)
 INSTALL_EXECUTABLE( gzstats )
 
 add_executable( sdf2pov sdf2pov.cc )
-target_link_libraries(sdf2pov gazebo_common gazebo_sdf_interface)
+target_link_libraries(sdf2pov gazebo_common gazebo_sdf_interface pthread)
 INSTALL_EXECUTABLE( sdf2pov )
 
 add_executable( gzfactory gzfactory.cc )
-target_link_libraries(gzfactory gazebo_common gazebo_transport boost_program_options)
+target_link_libraries(gzfactory gazebo_common gazebo_transport boost_program_options pthread)
 INSTALL_EXECUTABLE( gzfactory )
 
