--- gazebo-1.0.0.orig/cmake/SearchForStuff.cmake	2012-04-17 21:32:54.000000000 +0600
+++ gazebo-1.0.0/cmake/SearchForStuff.cmake	2012-05-08 15:43:42.909572935 +0600
@@ -125,14 +125,14 @@
 
       message(STATUS "tinyxml has no passed in paths, try to auto detect manually.")
 
-      find_path(tinyxml_include_dir tinyxml/tinyxml.hpp ${tinyxml_include_dirs} ENV CPATH)
+      find_path(tinyxml_include_dir tinyxml.h ${tinyxml_include_dirs} ENV CPATH)
       
       if (NOT tinyxml_include_dir)
-        message (STATUS "Looking for tinyxml/tinyxml.hpp - not found.")
+        message (STATUS "Looking for tinyxml.h - not found.")
         set (tinyxml_include_dirs /usr/include CACHE STRING
           "tinyxml include paths. Use this to override automatic detection.")
       else (NOT tinyxml_include_dir)
-        message (STATUS "Looking for tinyxml/tinyxml.hpp - found")
+        message (STATUS "Looking for tinyxml.h - found")
         set (assim_include_dirs ${tinyxml_include_dir} CACHE STRING
           "tinyxml include paths. Use this to override automatic detection.")
       endif (NOT tinyxml_include_dir)
@@ -164,10 +164,10 @@
 
   #################################################
   # Find TBB
-  pkg_check_modules(TBB tbb)
-  IF (NOT TBB_FOUND)
-    BUILD_ERROR ("Missing: TBB - Threading Building Blocks")
-  ENDIF (NOT TBB_FOUND)
+#  pkg_check_modules(TBB tbb)
+#  IF (NOT TBB_FOUND)
+#    BUILD_ERROR ("Missing: TBB - Threading Building Blocks")
+#  ENDIF (NOT TBB_FOUND)
 
   #################################################
   # Find OGRE 
@@ -199,7 +199,7 @@
 
   # Bug in libogre-dev ubuntu install package which cause ogre_library_dirs to 
   # be empty
-  if (${ogre_library_dirs} STREQUAL "")
+  if ("${ogre_library_dirs}" STREQUAL "")
     set (ogre_library_dirs "/usr/lib")
   endif()
 
--- gazebo-1.0.0.orig/src/CMakeLists.txt	2012-04-17 21:32:57.000000000 +0600
+++ gazebo-1.0.0/src/CMakeLists.txt	2012-05-08 16:39:40.496258580 +0600
@@ -42,6 +42,7 @@
                                 gazebo_transport
                                 gazebo_sdf_interface
                                 gazebo_msgs
+                                pthread
                                 ${boost_libraries}
                                 ${freeimage_library}
                                 ${PROTOBUF_LIBRARY})
