--- configure   2010-06-30 11:47:36.792557234 +0200
+++ configure   2010-06-30 11:47:58.832138313 +0200
@@ -20767,7 +20767,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#define __STDC_CONSTANT_MACROS
 extern "C" {
 #include <libavformat/avformat.h>
 #include <libavcodec/avcodec.h>

--- libffms/include/ffms.h	2010-11-17 20:08:01.140001822 +0100
+++ libffms/include/ffms.h	2010-11-17 20:10:44.380001822 +0100
@@ -221,7 +221,7 @@
 } FFMS_VideoProperties;
 
 typedef struct {
-	int SampleFormat;
+	int AVSampleFormat;
 	int SampleRate;
 	int BitsPerSample;
 	int Channels;
--- libffms/src/core/utils.cpp	2010-11-17 20:08:01.140001822 +0100
+++ libffms/src/core/utils.cpp	2010-11-17 20:11:00.066668488 +0100
@@ -218,7 +218,7 @@
 
 void FillAP(FFMS_AudioProperties &AP, AVCodecContext *CTX, FFMS_Track &Frames) {
 	AP.SampleFormat = static_cast<FFMS_SampleFormat>(CTX->sample_fmt);
-	AP.BitsPerSample = av_get_bits_per_sample_format(CTX->sample_fmt);
+	AP.BitsPerSample = av_get_bits_per_sample_fmt(CTX->sample_fmt);
 	if (CTX->sample_fmt == SAMPLE_FMT_S32 && CTX->bits_per_raw_sample)
 		AP.BitsPerSample = CTX->bits_per_raw_sample;
 
