--- rc.in	2015-01-05 20:32:29.000000000 +0100
+++ rc.busybox.in	2015-02-26 20:00:29.552189011 +0100
@@ -123,13 +123,14 @@
 	rc.motd
 }
 
-rc.halt() {
-	case "$action" in
-		poweroff|shutdown) echo 'o' > /proc/sysrq-trigger;;
-		halt) :;;
-		reboot|*) echo 'b' > /proc/sysrq-trigger;;
-	esac
-}
+# This is handled by busybox
+# rc.halt() {
+# 	case "$action" in
+# 		poweroff|shutdown) echo 'o' > /proc/sysrq-trigger;;
+# 		halt) :;;
+# 		reboot|*) echo 'b' > /proc/sysrq-trigger;;
+# 	esac
+# }
 
 rc.shutdown() {
 	rc.services_stop
@@ -139,7 +140,7 @@
 
 	echo "Halt complete."
 	
-	rc.halt
+	#rc.halt
 }
 
 rc.hostname() {
