--- gnupg-1.4.10/g10/gpg.c.old	2009-08-03 19:37:12.000000000 +0300
+++ gnupg-1.4.10/g10/gpg.c	2009-09-03 16:55:40.000000000 +0300
@@ -1892,7 +1892,7 @@
     opt.cert_digest_algo = 0;
     opt.compress_algo = -1; /* defaults to DEFAULT_COMPRESS_ALGO */
     opt.s2k_mode = 3; /* iterated+salted */
-    opt.s2k_count = 96; /* 65536 iterations */
+    opt.s2k_count = 208; /* 8388608 byte count */
 #ifdef USE_CAST5
     opt.s2k_cipher_algo = CIPHER_ALGO_CAST5;
 #else
