diff -aur kluppe-0.6.14/Makefile kluppe-0.6.14.patched/Makefile
--- kluppe-0.6.14/Makefile	2009-04-15 16:01:00.000000000 +0200
+++ kluppe-0.6.14.patched/Makefile	2012-09-17 04:54:30.750650405 +0200
@@ -1,6 +1,6 @@
 CC = gcc
 VERSION = `cat VERSION`
-INSTALL_PREFIX = /usr/local
+INSTALL_PREFIX = /usr
 BIN_DIR = $(INSTALL_PREFIX)/bin
 PIXMAPS_DIR = $(INSTALL_PREFIX)/share/pixmaps
 TEMP_ROOT = /tmp
diff -aur kluppe-0.6.14/src/common/alphatrack.h kluppe-0.6.14.patched/src/common/alphatrack.h
--- kluppe-0.6.14/src/common/alphatrack.h	2008-07-06 13:35:13.000000000 +0200
+++ kluppe-0.6.14.patched/src/common/alphatrack.h	2012-09-17 04:52:18.628607013 +0200
@@ -9,6 +9,7 @@
 
 #include <pthread.h>
 #include <usb.h>
+#include <stdint.h>
 
 #define ALPHATRACK_VENDORID  0x165b
 #define ALPHATRACK_PRODUCTID 0xFAD1
diff -aur kluppe-0.6.14/src/frontend/kluppe/Makefile kluppe-0.6.14.patched/src/frontend/kluppe/Makefile
--- kluppe-0.6.14/src/frontend/kluppe/Makefile	2008-07-05 10:55:06.000000000 +0200
+++ kluppe-0.6.14.patched/src/frontend/kluppe/Makefile	2012-09-17 04:51:44.254741982 +0200
@@ -1,11 +1,11 @@
 BIN_DIR = $(INSTALL_PREFIX)/bin
 PIXMAPS_DIR = $(INSTALL_PREFIX)/share/pixmaps
-MAN_DIR = $(INSTALL_PREFIX)/man/man1
+MAN_DIR = $(INSTALL_PREFIX)/share/man/man1
 SOURCES = sfpreview.c configfile.c griddialog.c threshdialog.c graindialog.c customcurve.c looper.c gtkmeter.c gtkloopview.c kluppe.c alphatrack_callback.c
 TARGETS = $(SOURCES:.c=.o)
 
 kluppe: $(TARGETS) 
-	$(CC) $(CFLAGS) -DPIXMAPS_DIR=\"$(PIXMAPS_DIR)\" *.o ../../common/*.o -o kluppe `pkg-config gtk+-2.0 gthread-2.0 libusb alsa jack sndfile liblo libxml-2.0 --libs gthread-2.0`
+	$(CC) $(CFLAGS) -DPIXMAPS_DIR=\"$(PIXMAPS_DIR)\" *.o ../../common/*.o -o kluppe `pkg-config gtk+-2.0 gthread-2.0 libusb alsa jack sndfile liblo libxml-2.0 --libs gthread-2.0` -lm
 
 .c.o: 
 	$(CC) $(CFLAGS) -DPIXMAPS_DIR=\"$(PIXMAPS_DIR)\" -c -o $@ $*.c `pkg-config gtk+-2.0 --cflags` `xml2-config --cflags`
