--- xorgedit/guiframe_old.cpp	2015-05-31 14:13:40.981744335 -0400
+++ xorgedit/guiframe.cpp	2015-05-31 14:06:41.960219162 -0400
@@ -143,7 +143,7 @@
 // Class Declaration
 // **************************
 
-GuiFrame::GuiFrame(const wxString& title) : wxFrame(NULL, -1, title, wxDefaultPosition, wxDefaultSize, wxDEFAULT_FRAME_STYLE & ~ (wxRESIZE_BORDER | wxRESIZE_BOX | wxMAXIMIZE_BOX))
+GuiFrame::GuiFrame(const wxString& title) : wxFrame(NULL, -1, title, wxDefaultPosition, wxDefaultSize, wxDEFAULT_FRAME_STYLE & ~ (wxRESIZE_BORDER | wxMAXIMIZE_BOX))
 {
   MyString funcName("GuiFrame::GuiFrame");
 
@@ -2713,7 +2713,7 @@
   }
 
   // file dialog for opening files
-  wxFileDialog openFileDialog(this, _("Choose a file"), wxString::FromAscii(settings.GetCurrentDir().data()), wxEmptyString, _T("*"), wxOPEN | wxCHANGE_DIR);
+  wxFileDialog openFileDialog(this, _("Choose a file"), wxString::FromAscii(settings.GetCurrentDir().data()), wxEmptyString, _T("*"), wxFD_OPEN | wxFD_CHANGE_DIR);
 
   if (openFileDialog.ShowModal() == wxID_OK)     // file has been chosen
   {
@@ -2826,7 +2826,7 @@
   }
 
   // file dialog for opening files
-  wxFileDialog saveFileDialog(this, _("Choose a file"), wxEmptyString, wxEmptyString, wxEmptyString,  wxSAVE | wxOVERWRITE_PROMPT | wxCHANGE_DIR );
+  wxFileDialog saveFileDialog(this, _("Choose a file"), wxEmptyString, wxEmptyString, wxEmptyString,  wxFD_SAVE | wxFD_OVERWRITE_PROMPT | wxFD_CHANGE_DIR );
 
   if (saveFileDialog.ShowModal() == wxID_OK)     // file has been chosen
   {
