--- wakfu/transition.conf	2014-02-03 08:33:06.000000000 -0500
+++ wakfu/transition.conf.new	2014-07-06 14:30:38.970527001 -0400
@@ -26,5 +26,8 @@
 else if equals(sys.osname, "mac")
     application_path = "${root}/transition.app"
 
-update.projects = ["transition", "game"]
+update.projects = ["game"]
+
+# Use the DBus service, which seems to work better, instead of sudo
+plugins.enable *= "dbusservice"
 
