diff --git a/quodlibet/gdist/icons.py b/quodlibet/gdist/icons.py
--- a/quodlibet/gdist/icons.py
+++ b/quodlibet/gdist/icons.py
@@ -25,7 +25,7 @@ class build_icon_cache(Command):
 
     def run(self):
         self.spawn(
-            ['gtk-update-icon-cache-3.0', '-f', 'quodlibet/images/hicolor'])
+            ['gtk-update-icon-cache', '-f', 'quodlibet/images/hicolor'])
 
 
 class install_icons(Command):
@@ -60,7 +60,7 @@ class install_icons(Command):
         self.copy_tree(png, png_dst)
 
         # this fails during packaging.. so ignore the outcome
-        subprocess.call(['gtk-update-icon-cache-3.0', basepath])
+        subprocess.call(['gtk-update-icon-cache', basepath])
 
         # install png versions to /usr/share/pixmaps
         basepath = os.path.join(self.prefix, 'share', 'pixmaps')
