--- BemusedServerDlg.h.orig	2007-05-11 20:41:13.000000000 +0200
+++ BemusedServerDlg.h	2007-05-11 20:41:37.000000000 +0200
@@ -63,11 +63,11 @@
 	void HandleBluetooth();
 
 	bool CheckWinamp(const char* aWindowName);
-  bool CBemusedServerDlg::ReadFile(int handle,char* buffer,int maxbytes,long unsigned int* bytesread,int* bla);
-  bool CBemusedServerDlg::ReadFile(int handle,unsigned char* buffer,int maxbytes,long unsigned int* bytesread,int* bla);
-  bool CBemusedServerDlg::WriteFile(int handle, char* buffer,int maxbytes, long unsigned int* byteswritten,int* bla);
-  bool CBemusedServerDlg::WriteFile(int handle,unsigned char* buffer,int maxbytes, long unsigned int* byteswritten,int* bla);
-  char* CBemusedServerDlg::GetLastError();
+  bool ReadFile(int handle,char* buffer,int maxbytes,long unsigned int* bytesread,int* bla);
+  bool ReadFile(int handle,unsigned char* buffer,int maxbytes,long unsigned int* bytesread,int* bla);
+  bool WriteFile(int handle, char* buffer,int maxbytes, long unsigned int* byteswritten,int* bla);
+  bool WriteFile(int handle,unsigned char* buffer,int maxbytes, long unsigned int* byteswritten,int* bla);
+  char* GetLastError();
 
 	void WriteListToPhone();
 	void WriteDirectoryListToPhone();
@@ -99,7 +99,7 @@
 	void FadeOut();
 	void ReadIniFile();
 
-	void CBemusedServerDlg::AddBookmarks(CDirTreeNode* aNode, CDirTreeNode* pNode, bool aRecursive);
+	void AddBookmarks(CDirTreeNode* aNode, CDirTreeNode* pNode, bool aRecursive);
 	void GenerateFileList(const char* aRoot, bool aRecursive);
 	void ReadDir(char* aDir, CDirTreeNode* aNode, bool aRecursive);
 	bool WriteNodeToBuffer(CDirTreeNode* aNode, unsigned char*& aBufferPtr, int& aSizeRemaining);
--- Makefile.orig	2007-05-11 21:03:41.000000000 +0200
+++ Makefile	2007-05-11 21:10:26.000000000 +0200
@@ -2,14 +2,16 @@
 SHELL = /bin/sh
 LIB = -lxmms -lbluetooth
 INCL = -I/usr/include/xmms -I./
+PREFIX = /usr
+DESTDIR =
 
 bemusedlinuxserver: 	main.cpp BemusedServerDlg.cpp
 			$(CXX) -o bemusedlinuxserver $(INCL) $(LIB) `gtk-config --libs --cflags` main.cpp BemusedServerDlg.cpp
 			
 
 install:		bemusedlinuxserver
-			cp -u ./bemusedlinuxserver /usr/bin/bemusedlinuxserver
-			cp -i --reply=no ./bemused.conf /etc/bemused.conf
+			cp ./bemusedlinuxserver $(DESTDIR)$(PREFIX)/bin/bemusedlinuxserver
+			cp ./bemused.conf $(DESTDIR)/etc/bemused.conf
 
 
 				 
