--- src/qesteidutil-build/qtsingleapplication/src/qtlocalpeer.cpp	2012-05-07 02:21:42.513867455 +0300
+++ src/qesteidutil/qtsingleapplication/src/qtlocalpeer.cpp	2012-05-07 02:23:54.103154575 +0300
@@ -56,7 +56,9 @@
 static PProcessIdToSessionId pProcessIdToSessionId = 0;
 #endif
 #if defined(Q_OS_UNIX)
+#include <sys/types.h>
 #include <time.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
