diff --git a/configure.ac b/configure.ac
index 7e5841d..79412af 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,6 +43,10 @@ AC_PROG_CC
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
+# automake 1.12 seems to require this, but automake 1.11 doesn't recognize it
+m4_pattern_allow([AM_PROG_AR])
+AM_PROG_AR
+
 LIBCURL_CHECK_CONFIG([yes], [], [], [
   AC_MSG_ERROR([[Libcurl header files not found, install libcurl-dev.]])])
 
