--- trunk-build/src/best_move.cpp	2013-09-18 10:26:35.358204856 +0200
+++ trunk-build/src/best_move.cpp	2013-09-18 10:27:20.018653221 +0200
@@ -77,7 +77,7 @@
 {
 	util::ZStream::setZipFileSuffixes(util::ZStream::Strings(1, "pgn"));
 
-	char const* ecoPath = SHAREDIR "/data/eco.bin";
+	char const* ecoPath = "/usr/share/scidb-beta" "/data/eco.bin";
 	mstl::ifstream stream(ecoPath, mstl::ios_base::in | mstl::ios_base::binary);
 	db::EcoTable::specimen().load(stream);
 
