--- trunk-build/configure	2013-09-17 15:01:24.476962328 +0200
+++ trunk-build/configure 15:00:55.954530139 +0200
@@ -1903,7 +1903,7 @@
             # don't use /usr/local/ on Fedora systems
             set prefix /usr/local
         } else {
-            set prefix /usr
+            set prefix  $configure(prefix)
         }
         if {![file isdirectory $prefix/share/fonts]} {
             set option(FONTDIR) "~/.fonts"
