--- taggo.old	2012-11-19 21:47:04.951294619 +0100
+++ taggo	2012-11-19 21:47:27.077845033 +0100
@@ -12,8 +12,8 @@
     tags_folder = ''
 
     def __init__(self):
-        self.mydir = os.path.dirname(os.path.abspath(__file__))
-        self.config_file = '%s/taggo.cfg' % self.mydir
+        self.mydir = os.environ['HOME']
+        self.config_file = '%s/.taggo.cfg' % self.mydir
         if not os.path.isfile(self.config_file):
             print 'Writing default config to %s\n' % self.config_file
             self.write_config()
