--- package.orig/SVDLIBC/Makefile	2009-11-17 02:34:01.098240515 -0700
+++ package.new/SVDLIBC/Makefile	2009-11-17 02:36:58.131511152 -0700
@@ -1,12 +1,4 @@
-# Linux or Windows:
-CC = gcc -Wall -O4 -march=i486
-# CC = icc -w1 -O3 -march=i486
-
-# Macintosh:
-ifeq ($(HOSTTYPE),powerpc)
-  CC = cc -pipe -O3 -Wall -fno-common -arch ppc
-endif
-
+HOSTTYPE=${CHOST}
 LIBS=-lm
 OBJ=svdlib.o svdutil.o las2.o
 
