diff -u rdesktop-1.5.0.orig/rdesktop.c rdesktop-1.5.0/rdesktop.c
--- rdesktop-1.5.0.orig/rdesktop.c	2006-08-07 19:45:43.000000000 +0800
+++ rdesktop-1.5.0/rdesktop.c	2006-10-10 09:39:12.000000000 +0800
@@ -91,6 +91,7 @@
 RD_BOOL g_owncolmap = False;
 RD_BOOL g_ownbackstore = True;	/* We can't rely on external BackingStore */
 RD_BOOL g_seamless_rdp = False;
+RD_BOOL g_raw_keyboard = False;
 uint32 g_embed_wnd;
 uint32 g_rdp5_performanceflags =
 	RDP5_NO_WALLPAPER | RDP5_NO_FULLWINDOWDRAG | RDP5_NO_MENUANIMATIONS;
@@ -191,6 +192,7 @@
 	fprintf(stderr, "   -0: attach to console\n");
 	fprintf(stderr, "   -4: use RDP version 4\n");
 	fprintf(stderr, "   -5: use RDP version 5 (default)\n");
+	fprintf(stderr, "   -y: use raw keyboard (default no)\n");
 }
 
 static void
@@ -427,7 +429,7 @@
 #endif
 
 	while ((c = getopt(argc, argv,
-			   VNCOPT "Au:L:d:s:c:p:n:k:g:fbBeEmzCDKS:T:NX:a:x:Pr:045h?")) != -1)
+			   VNCOPT "Au:L:d:s:c:p:n:k:g:fbBeEmzCDKS:T:NX:a:x:Pr:045yh?")) != -1)
 	{
 		switch (c)
 		{
@@ -746,7 +748,9 @@
 			case '5':
 				g_use_rdp5 = True;
 				break;
-
+			case 'y':
+                                g_raw_keyboard = True;
+                                break;
 			case 'h':
 			case '?':
 			default:
