diff -u rdesktop-1.5.0.orig/xwin.c rdesktop-1.5.0/xwin.c
--- rdesktop-1.5.0.orig/xwin.c	2006-08-07 19:45:44.000000000 +0800
+++ rdesktop-1.5.0/xwin.c	2006-10-10 09:59:02.000000000 +0800
@@ -27,6 +27,8 @@
 #include <strings.h>
 #include "rdesktop.h"
 #include "xproto.h"
+#include "X11/XKBlib.h"
+#include "X11/keysym.h"
 
 /* We can't include Xproto.h because of conflicting defines for BOOL */
 #define X_ConfigureWindow              12
@@ -45,6 +47,7 @@
    As of RDP 5.1, it may be 8, 15, 16 or 24. */
 extern int g_server_depth;
 extern int g_win_button_size;
+extern RD_BOOL g_raw_keyboard;
 
 Display *g_display;
 Time g_last_gesturetime;
@@ -129,6 +132,7 @@
     so its endianess doesn't matter)
  */
 static RD_BOOL g_no_translate_image = False;
+static RD_BOOL g_modeswitch_down = False;
 
 /* endianness */
 static RD_BOOL g_host_be;
@@ -1956,6 +1960,7 @@
 	}
 }
 
+extern void vk_handle_key( XKeyEvent *xkey);
 
 /* Process events in Xlib queue
    Returns 0 after user quit, 1 otherwise */
@@ -1980,6 +1985,12 @@
 			continue;
 		}
 
+                if( g_raw_keyboard && ((xevent.type == KeyPress) || (xevent.type == KeyRelease))) {
+                        g_last_gesturetime = xevent.xkey.time;
+                        vk_handle_key( (XKeyEvent*)&xevent);
+                        continue;
+                }
+
 		switch (xevent.type)
 		{
 			case VisibilityNotify:
@@ -2018,6 +2029,10 @@
 					XLookupString((XKeyEvent *) & xevent,
 						      str, sizeof(str), &keysym, NULL);
 				}
+				
+				// Mode_switch during XGrabKeyboard fix: Ungrab Keyboard during Mode_switch
+				if ( keysym == XK_Mode_switch ) g_modeswitch_down = True;
+				if ( g_focused && g_modeswitch_down ) XUngrabKeyboard(g_display, CurrentTime);
 
 				DEBUG_KBD(("KeyPress for keysym (0x%lx, %s)\n", keysym,
 					   get_ksname(keysym)));
@@ -2034,6 +2049,10 @@
 				g_last_gesturetime = xevent.xkey.time;
 				XLookupString((XKeyEvent *) & xevent, str,
 					      sizeof(str), &keysym, NULL);
+				// Mode_switch during XGrabKeyboard fix: Regrab Keyboard after Mode_switch
+				if ( keysym == XK_Mode_switch ) g_modeswitch_down = False;
+				if ( g_focused && !g_modeswitch_down ) XGrabKeyboard(g_display, g_wnd, True, GrabModeAsync, GrabModeAsync, CurrentTime);
+
 
 				DEBUG_KBD(("\nKeyRelease for keysym (0x%lx, %s)\n", keysym,
 					   get_ksname(keysym)));
