diff -ru a/sinamicroblog.cpp b/sinamicroblog.cpp
--- a/sinamicroblog.cpp	2011-12-19 19:30:38.095438146 +0800
+++ b/sinamicroblog.cpp	2011-12-19 19:29:57.888989297 +0800
@@ -469,17 +468,17 @@
         return 0;
 }
 
-QString SinaMicroBlog::postUrl( Choqok::Account* account, const QString& username, const QString& postId ) const
+QString SinaMicroBlog::postUrl( Choqok::Account* account, const QString& userId, const QString& postId ) const
 {
     Q_UNUSED(account)
-    Q_UNUSED(username)
-    return QString( "http://weibo.com/status/%2" ).arg( postId );
+    return QString();
+    // return QString( "http://weibo.com/%1/status/%2" ).arg( userId ).arg( postId );
 }
 
 QString SinaMicroBlog::profileUrl( Choqok::Account* account, const QString& username ) const
 {
     Q_UNUSED(account)
-    return QString( "http://weibo.com/%1" ).arg( username );
+    return QString( "http://weibo.com/n/%1" ).arg( username );
 }
 
 void SinaMicroBlog::retweetPost( Choqok::Account* theAccount, Choqok::Post* post )
