diff -uNr zarafa-7.1.7.old/caldav/CalDAV.cpp zarafa-7.1.7/caldav/CalDAV.cpp
--- zarafa-7.1.7.old/caldav/CalDAV.cpp	2013-09-27 16:58:27.000000000 -0600
+++ zarafa-7.1.7/caldav/CalDAV.cpp	2014-03-19 14:46:09.360454187 -0600
@@ -197,7 +197,7 @@
     struct sigaction act = {{0}};
 
 	// Configuration
-	char opt = '\0';
+	int opt = '\0';
 	const char *lpszCfg = ECConfig::GetDefaultPath("ical.cfg");
 	const configsetting_t lpDefaults[] = {
 		{ "run_as_user", "" },
diff -uNr zarafa-7.1.7.old/common/Trace.cpp zarafa-7.1.7/common/Trace.cpp
--- zarafa-7.1.7.old/common/Trace.cpp	2013-09-27 16:58:27.000000000 -0600
+++ zarafa-7.1.7/common/Trace.cpp	2014-03-19 16:51:06.370457761 -0600
@@ -126,7 +126,7 @@
 
 	len = pos + 3;
 
-	if (format && va) {
+	if (format) {
 		va_copy(va_lentest, va);
 		len += _vsnprintf(NULL, 0, format, va_lentest);
 		va_end(va_lentest);
@@ -139,7 +139,7 @@
 
 	memcpy(buffer, debug, pos);
 
-	if (format && va)
+	if (format)
 		pos = _vsnprintf(buffer+pos, len-pos, format, va);
 
 	if(pos == -1) {
diff -uNr zarafa-7.1.7.old/ECtools/zarafa-search/zarafa-search.cpp zarafa-7.1.7/ECtools/zarafa-search/zarafa-search.cpp
--- zarafa-7.1.7.old/ECtools/zarafa-search/zarafa-search.cpp	2013-09-27 16:58:27.000000000 -0600
+++ zarafa-7.1.7/ECtools/zarafa-search/zarafa-search.cpp	2014-03-19 14:48:28.075454253 -0600
@@ -551,7 +551,7 @@
 	};
 
 	while (true) {
-		char c = my_getopt_long_permissive(argc, argv, "c:h:iuFV", long_options, NULL);
+		int c = my_getopt_long_permissive(argc, argv, "c:h:iuFV", long_options, NULL);
 		
 		if (c == -1)
 			break;
