diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index 62e0b56..41dc52e 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -8,9 +8,9 @@ FILE(GLOB man8 RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
     "*.8")
 
 install(FILES ${man1}
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/man/man1)
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man1)
 install(FILES ${man8}
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/man/man8)
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man8)
 
 ####################################
 ## Convert wiki pages to TXT/HTML ##
@@ -57,11 +57,11 @@ ADD_CUSTOM_TARGET(docs-html DEPENDS ${htmlFiles})
 ADD_CUSTOM_TARGET(docs-txt DEPENDS ${txtFiles})
 
 install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/doc/ulatencyd
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/doc/ulatencyd
         OPTIONAL)
 
 install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/txt
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/doc/ulatencyd
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/doc/ulatencyd
         OPTIONAL)
 
 
@@ -80,7 +80,7 @@ add_custom_target(docs-api
                   DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/api)
 
 install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/api
-        DESTINATION ${CMAKE_INSTALL_PREFIX}/doc/ulatencyd
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/share/doc/ulatencyd
         OPTIONAL)
 
 
