--- pkg/sdk/shared/src/shrUtils.cpp.orig	2011-04-07 10:36:20.013234725 +0200
+++ pkg/sdk/shared/src/shrUtils.cpp	2011-04-07 10:41:00.025799423 +0200
@@ -596,6 +596,10 @@
         "../data/",                                 // up 1 in tree, "/data/" subdir 
         "../src/",                                  // up 1 in tree, "/src/" subdir 
         "../inc/",                                  // up 1 in tree, "/inc/" subdir 
+        "../src/<executable_name>/",                // up 1 in tree, "/src/<executable_name>/" subdir 
+        "../src/<executable_name>/data/",           // up 1 in tree, "/src/<executable_name>/data/" subdir 
+        "../src/<executable_name>/src/",            // up 1 in tree, "/src/<executable_name>/src/" subdir 
+        "../OpenCL/src/<executable_name>/inc/",     // up 1 in tree, "/OpenCL/src/<executable_name>/inc/" subdir 
         "../OpenCL/src/<executable_name>/",         // up 1 in tree, "/OpenCL/src/<executable_name>/" subdir 
         "../OpenCL/src/<executable_name>/data/",    // up 1 in tree, "/OpenCL/src/<executable_name>/data/" subdir 
         "../OpenCL/src/<executable_name>/src/",     // up 1 in tree, "/OpenCL/src/<executable_name>/src/" subdir 
--- pkg/sdk/CUDALibraries/src/boxFilterNPP/boxFilterNPP.cpp.orig	2011-05-23 06:57:49.000000000 +0200
+++ pkg/sdk/CUDALibraries/src/boxFilterNPP/boxFilterNPP.cpp	2011-05-31 13:47:40.561169789 +0200
@@ -37,7 +37,7 @@
     if (filename == 0)
         return 0;
     size_t filename_len = strlen(filename);
-    const char data_folder[] = "../../data/";
+    const char data_folder[] = "data/";
     size_t data_folder_len = strlen(data_folder);
     char* file_path = 
       (char*) malloc( sizeof(char) * (data_folder_len + filename_len + 1));
--- pkg/sdk/CUDALibraries/src/freeImageInteropNPP/freeImageInteropNPP.cpp.orig	2011-05-23 06:57:49.000000000 +0200
+++ pkg/sdk/CUDALibraries/src/freeImageInteropNPP/freeImageInteropNPP.cpp	2011-05-31 13:47:40.547835889 +0200
@@ -66,7 +66,7 @@
     if (filename == 0)
         return 0;
     size_t filename_len = strlen(filename);
-    const char data_folder[] = "../../data/";
+    const char data_folder[] = "data/";
     size_t data_folder_len = strlen(data_folder);
     char* file_path = 
       (char*) malloc( sizeof(char) * (data_folder_len + filename_len + 1));
--- pkg/sdk/CUDALibraries/src/histEqualizationNPP/histEqualizationNPP.cpp.orig	2011-05-23 06:57:49.000000000 +0200
+++ pkg/sdk/CUDALibraries/src/histEqualizationNPP/histEqualizationNPP.cpp	2011-05-31 13:47:40.554502839 +0200
@@ -38,7 +38,7 @@
     if (filename == 0)
         return 0;
     size_t filename_len = strlen(filename);
-    const char data_folder[] = "../../data/";
+    const char data_folder[] = "data/";
     size_t data_folder_len = strlen(data_folder);
     char* file_path = 
       (char*) malloc( sizeof(char) * (data_folder_len + filename_len + 1));
--- pkg/sdk/CUDALibraries/src/imageSegmentationNPP/imageSegmentationNPP.cpp.orig	2011-05-23 06:57:49.000000000 +0200
+++ pkg/sdk/CUDALibraries/src/imageSegmentationNPP/imageSegmentationNPP.cpp	2011-05-31 13:47:40.541168939 +0200
@@ -51,7 +51,7 @@
     if (filename == 0)
         return 0;
     size_t filename_len = strlen(filename);
-    const char data_folder[] = "../../data/";
+    const char data_folder[] = "data/";
     size_t data_folder_len = strlen(data_folder);
     char* file_path = 
       (char*) malloc( sizeof(char) * (data_folder_len + filename_len + 1));
