--- pkg/sdk/C/common/common.mk.orig	2011-04-07 10:22:48.898804324 +0200
+++ pkg/sdk/C/common/common.mk	2011-04-07 10:24:03.351932604 +0200
@@ -23,7 +23,7 @@
 # Add new SM Versions here as devices with new Compute Capability are released
 SM_VERSIONS   := 10 11 12 13 20 21
 
-CUDA_INSTALL_PATH ?= /usr/local/cuda
+CUDA_INSTALL_PATH ?= /usr
 
 ifdef cuda-install
 	CUDA_INSTALL_PATH := $(cuda-install)
@@ -61,7 +61,7 @@
 LINK       := g++ -fPIC
 
 # Includes
-INCLUDES  += -I. -I$(CUDA_INSTALL_PATH)/include -I$(COMMONDIR)/inc -I$(SHAREDDIR)/inc
+INCLUDES  += -I. -I$(CUDA_INSTALL_PATH)/include/cuda -I$(COMMONDIR)/inc -I$(SHAREDDIR)/inc
 
 # Warning flags
 CXXWARN_FLAGS := \
@@ -89,7 +89,7 @@
 # architecture flag for nvcc and gcc compilers build
 CUBIN_ARCH_FLAG :=
 CXX_ARCH_FLAGS  :=
-NVCCFLAGS       :=
+NVCCFLAGS       := --compiler-bindir=/opt/gcc-4.4
 LIB_ARCH        := $(OSARCH)
 
 # Determining the necessary Cross-Compilation Flags
--- pkg/sdk/CUDALibraries/common/UtilNPP/defines.mk.orig	2011-04-07 10:25:38.788336773 +0200
+++ pkg/sdk/CUDALibraries/common/UtilNPP/defines.mk	2011-04-07 10:26:14.034906592 +0200
@@ -38,7 +38,7 @@
 # Add new SM Versions here as devices with new Compute Capability are released
 SM_VERSIONS := sm_10 sm_11 sm_12 sm_13 sm_20 sm_21
 
-CUDA_INSTALL_PATH ?= /usr/local/cuda
+CUDA_INSTALL_PATH ?= /usr
 NPP_INSTALL_PATH  ?= ../../../npp
 NPP_INCLUDE_PATH  ?= ../../common/npp/include
 
@@ -82,7 +82,7 @@
 LINK       := g++ -fPIC
 
 # Includes
-INCLUDES  += -I. -I$(CUDA_INSTALL_PATH)/include -I$(ROOTDIR)/inc -I$(NPP_INCLUDE_PATH) -I../FreeImage/include
+INCLUDES  += -I. -I$(CUDA_INSTALL_PATH)/include/cuda -I$(ROOTDIR)/inc -I$(NPP_INCLUDE_PATH) -I../FreeImage/include
 
 ifneq ($(DARWIN),)
     CXXWARN_FLAGS := 
--- pkg/sdk/CUDALibraries/common/common.mk.orig	2011-04-07 10:25:38.791670097 +0200
+++ pkg/sdk/CUDALibraries/common/common.mk	2011-04-07 10:27:59.771282822 +0200
@@ -23,7 +23,7 @@
 # Add new SM Versions here as devices with new Compute Capability are released
 SM_VERSIONS   := 10 11 12 13 20 21
 
-CUDA_INSTALL_PATH ?= /usr/local/cuda
+CUDA_INSTALL_PATH ?= /usr
 
 ifdef cuda-install
 	CUDA_INSTALL_PATH := $(cuda-install)
@@ -61,7 +61,7 @@
 LINK       := g++ -fPIC
 
 # Includes
-INCLUDES  += -I. -I$(CUDA_INSTALL_PATH)/include -I$(COMMONDIR)/inc -I$(SHAREDDIR)/inc
+INCLUDES  += -I. -I$(CUDA_INSTALL_PATH)/include/cuda -I$(COMMONDIR)/inc -I$(SHAREDDIR)/inc
 
 # Warning flags
 CXXWARN_FLAGS := \
@@ -89,7 +89,7 @@
 # architecture flag for nvcc and gcc compilers build
 CUBIN_ARCH_FLAG :=
 CXX_ARCH_FLAGS  :=
-NVCCFLAGS       :=
+NVCCFLAGS       := --compiler-bindir=/opt/gcc-4.4
 LIB_ARCH        := $(OSARCH)
 
 # Determining the necessary Cross-Compilation Flags
--- pkg/sdk/CUDALibraries/src/imageSegmentationNPP/Makefile.orig	2011-04-07 10:25:38.791670097 +0200
+++ pkg/sdk/CUDALibraries/src/imageSegmentationNPP/Makefile	2011-04-07 10:29:11.924416209 +0200
@@ -29,7 +29,7 @@
 #
 ################################################################################
 
-CUDA_INSTALL_PATH ?= /usr/local/cuda
+CUDA_INSTALL_PATH ?= /usr
 
 ifdef cuda-install
 	CUDA_INSTALL_PATH := $(cuda-install)
@@ -107,7 +107,7 @@
 endif
 
 CXX := g++ -fPIC $(CXX_ARCH_FLAGS)
-INC := -I. -I../../common/UtilNPP/ -I../../../shared/inc/ -I$(CUDA_INSTALL_PATH)/include/
+INC := -I. -I../../common/UtilNPP/ -I../../../shared/inc/ -I$(CUDA_INSTALL_PATH)/include/cuda
 
 LIB := -L$(SDK_INSTALL_PATH)/common/lib/ -lnpp -L$(CUDA_INSTALL_PATH)/lib$(CUDALIBARCH)/ -L$(CUDA_INSTALL_PATH)/lib/ -lcudart 
 LIB += -L$(SDK_INSTALL_PATH)/common/FreeImage/lib/$(OSLOWER)
--- pkg/sdk/CUDALibraries/src/freeImageInteropNPP/Makefile.orig	2011-04-07 10:25:38.791670097 +0200
+++ pkg/sdk/CUDALibraries/src/freeImageInteropNPP/Makefile	2011-04-07 10:29:04.237770832 +0200
@@ -29,7 +29,7 @@
 #
 ################################################################################
 
-CUDA_INSTALL_PATH ?= /usr/local/cuda
+CUDA_INSTALL_PATH ?= /usr
 
 ifdef cuda-install
 	CUDA_INSTALL_PATH := $(cuda-install)
@@ -103,7 +103,7 @@
 endif
 
 CXX := g++ -fPIC $(CXX_ARCH_FLAGS)
-INC := -I. -I../../common/UtilNPP/ -I../../../shared/inc -I../../common/FreeImage/include -I$(CUDA_INSTALL_PATH)/include/
+INC := -I. -I../../common/UtilNPP/ -I../../../shared/inc -I../../common/FreeImage/include -I$(CUDA_INSTALL_PATH)/include/cuda
 
 LIB := -L$(SDK_INSTALL_PATH)/common/lib -L$(CUDA_INSTALL_PATH)/lib$(CUDALIBARCH) -lcudart -lnpp
 LIB += -L$(SDK_INSTALL_PATH)/common/FreeImage/lib/$(OSLOWER) 
--- pkg/sdk/CUDALibraries/src/histEqualizationNPP/Makefile.orig	2011-04-07 10:25:38.795003421 +0200
+++ pkg/sdk/CUDALibraries/src/histEqualizationNPP/Makefile	2011-04-07 10:28:46.857818980 +0200
@@ -29,7 +29,7 @@
 #
 ################################################################################
 
-CUDA_INSTALL_PATH ?= /usr/local/cuda
+CUDA_INSTALL_PATH ?= /usr
 
 ifdef cuda-install
 	CUDA_INSTALL_PATH := $(cuda-install)
@@ -103,7 +103,7 @@
 endif
 
 CXX := g++ -fPIC $(CXX_ARCH_FLAGS) 
-INC := -I. -I../../common/UtilNPP/ -I../../../shared/inc -I$(CUDA_INSTALL_PATH)/include/
+INC := -I. -I../../common/UtilNPP/ -I../../../shared/inc -I$(CUDA_INSTALL_PATH)/include/cuda
 
 LIB := -L$(SDK_INSTALL_PATH)/common/lib/ -lnpp -L$(CUDA_INSTALL_PATH)/lib$(CUDALIBARCH)/ -L$(CUDA_INSTALL_PATH)/lib/ -lcudart
 LIB += -L$(SDK_INSTALL_PATH)/common/FreeImage/lib/$(OSLOWER)
--- pkg/sdk/CUDALibraries/src/boxFilterNPP/Makefile.orig	2011-04-07 10:25:38.795003421 +0200
+++ pkg/sdk/CUDALibraries/src/boxFilterNPP/Makefile	2011-04-07 10:28:31.921193700 +0200
@@ -29,7 +29,7 @@
 #
 ################################################################################
 
-CUDA_INSTALL_PATH ?= /usr/local/cuda
+CUDA_INSTALL_PATH ?= /usr
 
 ifdef cuda-install
 	CUDA_INSTALL_PATH := $(cuda-install)
@@ -104,7 +104,7 @@
 endif
 
 CXX := g++ -fPIC $(CXX_ARCH_FLAGS)
-INC := -I. -I../../common/UtilNPP/ -I../../../shared/inc -I$(CUDA_INSTALL_PATH)/include/
+INC := -I. -I../../common/UtilNPP/ -I../../../shared/inc -I$(CUDA_INSTALL_PATH)/include/cuda
 
 LIB := -L$(SDK_INSTALL_PATH)/common/lib -L$(CUDA_INSTALL_PATH)/lib$(CUDALIBARCH) -lnpp -lcudart 
 LIB += -L$(SDK_INSTALL_PATH)/common/FreeImage/lib/$(OSLOWER)
