=== modified file 'src/menubarapplet.cpp'
--- src/menubarapplet.cpp	2011-01-14 20:07:40 +0000
+++ src/menubarapplet.cpp	2011-02-22 12:03:29 +0000
@@ -248,6 +248,12 @@
     mMenuWidget = new MenuWidget(this);
     mMenuWidget->setMenu(menu);
     mLayout->addItem(mMenuWidget);
+    
+    QPropertyAnimation *a = new QPropertyAnimation(mMenuWidget, "opacity");
+    a->setStartValue(0);
+    a->setEndValue(1);
+    a->start(QAbstractAnimation::DeleteWhenStopped);
+    
 }
 
 void MenuBarApplet::createConfigurationInterface(KConfigDialog* parent)

