--- src/menuwidget.cpp.old	2011-02-14 00:48:12.920000021 +0200
+++ src/menuwidget.cpp	2011-02-14 00:47:00.143333356 +0200
@@ -18,11 +18,13 @@
 #include <QMenu>
 #include <QTimer>
 #include <QToolButton>
+#include <QGraphicsDropShadowEffect>
 
 // KDE
 #include <KDebug>
 #include <Plasma/Containment>
 #include <Plasma/Corona>
+#include <Plasma/Theme>
 
 MenuWidget::MenuWidget(Plasma::Applet* applet)
 : QGraphicsWidget(applet)
@@ -124,6 +126,12 @@
     connect(button, SIGNAL(clicked()), SLOT(slotButtonClicked()));
     mLayout->addItem(button);
 
+    QGraphicsDropShadowEffect* shadow = new QGraphicsDropShadowEffect();
+    shadow->setBlurRadius(5);
+    shadow->setOffset(QPointF(0, 1));
+    shadow->setColor(Plasma::Theme::defaultTheme()->color(Plasma::Theme::BackgroundColor));
+    setGraphicsEffect(shadow);
+
     button->setText(text);
 
     disconnect(menu, 0, this, 0); // Ensure we are only connected to aboutToHide() once
