diff --git bin/pyxis bin/pyxis
index 9b447fb..0350278 100755
--- bin/pyxis
+++ bin/pyxis
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #Pyxis and Original Sipie: Sirius Command Line Player
 #Copyright (C) Corey Ling, Eli Criffield
 #
diff --git pyxis/Config.py pyxis/Config.py
index d4d6d29..a36778e 100644
--- pyxis/Config.py
+++ pyxis/Config.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #Pyxis and Original Sipie: Sirius Command Line Player
 #Copyright (C) Corey Ling, Eli Criffield
 #
diff --git pyxis/Debug.py pyxis/Debug.py
index d4af851..bf1bd6e 100644
--- pyxis/Debug.py
+++ pyxis/Debug.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #Pyxis
 #Copyright (C) Corey Ling
 #
diff --git pyxis/Exceptions.py pyxis/Exceptions.py
index a1b5f8d..0270a02 100644
--- pyxis/Exceptions.py
+++ pyxis/Exceptions.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #Pyxis and Original Sipie: Sirius Command Line Player
 #Copyright (C) Corey Ling, Eli Criffield
 #
diff --git pyxis/Interface.py pyxis/Interface.py
index 3c60a7c..1b0c05c 100644
--- pyxis/Interface.py
+++ pyxis/Interface.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #Pyxis and Original Sipie: Sirius Command Line Player
 #Copyright (C) Corey Ling, Eli Criffield
 #
diff --git pyxis/Player.py pyxis/Player.py
index 8e945ba..c467acd 100644
--- pyxis/Player.py
+++ pyxis/Player.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #Pyxis and Original Sipie: Sirius Command Line Player
 #Copyright (C) Corey Ling, Eli Criffield
 #
diff --git pyxis/ProviderCanada.py pyxis/ProviderCanada.py
index 4d84952..9ff2aa6 100644
--- pyxis/ProviderCanada.py
+++ pyxis/ProviderCanada.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #Pyxis and Original Sipie: Sirius Command Line Player
 #Copyright (C) Corey Ling, Eli Criffield
 #
diff --git pyxis/ProviderUSA.py pyxis/ProviderUSA.py
index 7c30eaf..b1fb950 100644
--- pyxis/ProviderUSA.py
+++ pyxis/ProviderUSA.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #Pyxis and Original Sipie: Sirius Command Line Player
 #Copyright (C) Corey Ling, Eli Criffield
 #
diff --git pyxis/Sirius.py pyxis/Sirius.py
index 30a25d6..75f1fb6 100644
--- pyxis/Sirius.py
+++ pyxis/Sirius.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #Pyxis and Original Sipie: Sirius Command Line Player
 #Copyright (C) Corey Ling, Eli Criffield
 #
