Index: cblrepo/Agda-2.4.2.2/Agda.cabal
===================================================================
--- cblrepo.orig/Agda-2.4.2.2/Agda.cabal
+++ cblrepo/Agda-2.4.2.2/Agda.cabal
@@ -98,7 +98,7 @@ library
     -- fixed in hashable 1.2.1.0.
     -- https://github.com/tibbe/hashable/issues/57.
     , hashable >= 1.1.2.3 && < 1.2 || >= 1.2.1.0 && < 1.3
-    , hashtables >= 1.0 && < 1.2
+    , hashtables >= 1.0 && < 1.3
     , haskeline >= 0.7 && < 0.8
     , haskell-src-exts >= 1.9.6 && < 1.17
     -- mtl-2.1 contains a severe bug.
