diff --git a/dlls/kernel32/sync.c b/dlls/kernel32/sync.c
index 2b7f4b0..12ffdc4 100644
--- a/dlls/kernel32/sync.c
+++ b/dlls/kernel32/sync.c
@@ -456,7 +456,7 @@ HANDLE WINAPI DECLSPEC_HOTPATCH CreateEventExA( SECURITY_ATTRIBUTES *sa, LPCSTR
 /***********************************************************************
  *           CreateEventExW    (KERNEL32.@)
  */
-HANDLE WINAPI DECLSPEC_HOTPATCH CreateEventExW( SECURITY_ATTRIBUTES *sa, LPCWSTR name, DWORD flags, DWORD access )
+HANDLE WINAPI __attribute__((optimize("-O0"))) CreateEventExW( SECURITY_ATTRIBUTES *sa, LPCWSTR name, DWORD flags, DWORD access )
 {
     HANDLE ret;
     UNICODE_STRING nameW; 
