diff --git a/bin/ptxdist b/bin/ptxdist
index a47d968..9981582 100755
--- a/bin/ptxdist
+++ b/bin/ptxdist
@@ -2,6 +2,7 @@
 
 LANG=C
 LC_ALL=POSIX
+export PATH=$(dirname $(readlink -f /usr/bin/ptxdist)):$PATH
 
 declare -a PTXDIST_ARGS_FULL
 declare -a PTXDIST_ARGS_SECOND PTX_MAKE_ARGS
diff --git a/scripts/ptxdist_version.sh b/scripts/ptxdist_version.sh
index 78736d0..759a061 100644
--- a/scripts/ptxdist_version.sh
+++ b/scripts/ptxdist_version.sh
@@ -6,6 +6,7 @@
 _ptxd_get_version()
 {
     PTXDIST_VERSION_FULL="$("${PTXDIST_TOPDIR:=.}/scripts/kernel/setlocalversion" "${PTXDIST_TOPDIR}/.tarball-version")"
+    PTXDIST_VERSION_FULL=$(echo ${PTXDIST_VERSION_FULL} | sed -e 's/-dirty//g')
 
     local orig_IFS="${IFS}"
     local IFS="."
