--- dhcpdump.c.bak	2008-06-24 05:26:52.000000000 +0200
+++ dhcpdump.c	2013-10-03 11:24:39.895037895 +0200
@@ -163,10 +163,10 @@
 	strcpy(ip_origin, (u_char *)inet_ntoa(ip->ip_src));
 	strcpy(ip_destination, (u_char *)inet_ntoa(ip->ip_dst));
 
-	if (hmask && check_ch((u_char *)(sp + offset), ntohs(udp->uh_ulen)))
+	if (hmask && check_ch((u_char *)(sp + offset), ntohs(udp->len)))
 		return;
 
-	printdata((u_char *)(sp + offset), ntohs(udp->uh_ulen));
+	printdata((u_char *)(sp + offset), ntohs(udp->len));
 }
 
 // check for matching CHADDR (Peter Apian-Bennewitz <apian@ise.fhg.de>)
