--- StepMania-3.9-src-orig/src/ModelTypes.h	2004-08-19 10:29:16.000000000 +0200
+++ StepMania-3.9-src/src/ModelTypes.h.new	2008-04-04 03:48:01.000000000 +0200
@@ -6,6 +6,7 @@
 #define MS_MAX_NAME             32
 
 #include "RageTypes.h"
+#include <cstring>
 
 struct msTriangle
 {
--- StepMania-3.9-src-orig/src/crypto51/filters.h	2004-05-13 08:02:27.000000000 +0200
+++ StepMania-3.9-src/src/crypto51/filters.h.new	2008-04-04 03:50:33.000000000 +0200
@@ -7,6 +7,7 @@
 #include "smartptr.h"
 #include "queue.h"
 #include "algparam.h"
+#include <memory>
 
 NAMESPACE_BEGIN(CryptoPP)
 
--- StepMania-3.9-src-orig/src/crypto/CryptSH512.h	2004-06-08 07:30:52.000000000 +0200
+++ StepMania-3.9-src/src/crypto/CryptSH512.h.new	2008-04-04 03:50:24.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef SSH_SHA512_H
 #define SSH_SHA512_H
 
+#include <cstring>
+
 typedef struct
 {
 	unsigned long hi, lo;
--- StepMania-3.9-src-orig/src/crypto/CryptSHA.h	2004-06-08 07:30:52.000000000 +0200
+++ StepMania-3.9-src/src/crypto/CryptSHA.h.new	2008-04-04 03:53:07.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef SSHSHA_H
 #define SSHSHA_H
 
+#include <cstring>
+
 typedef struct {
 	uint32_t h[5];
 	unsigned char block[64];
--- StepMania-3.9-src-orig/src/crypto/CryptRand.h	2004-06-08 07:30:52.000000000 +0200
+++ StepMania-3.9-src/src/crypto/CryptRand.h.new	2008-04-04 03:52:58.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef SSH_RAND_H
 #define SSH_RAND_H
 
+#include <cstring>
+
 void random_init();
 void random_add_noise( const CString &noise );
 unsigned char random_byte();
--- StepMania-3.9-src-orig/src/crypto/CryptBn.h	2004-06-08 07:30:52.000000000 +0200
+++ StepMania-3.9-src/src/crypto/CryptBn.h.new	2008-04-04 03:52:51.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef SSHBN_H
 #define SSHBN_H
 
+#include <cstring>
+
 #if defined __GNUC__ && defined __i386__
 typedef unsigned long BignumInt;
 typedef unsigned long long BignumDblInt;
--- StepMania-3.9-src-orig/src/NetworkSyncServer.h	2008-04-04 03:46:16.000000000 +0200
+++ StepMania-3.9-src/src/NetworkSyncServer.h.new	2008-04-04 03:52:43.000000000 +0200
@@ -7,6 +7,8 @@
 #include "ezsockets.h"
 #define NETMAXBUFFERSIZE 1020
 
+#include <cstring>
+
 class LanPlayer
 {
 public:
--- StepMania-3.9-src-orig/src/RageSurface.h	2004-09-06 01:37:04.000000000 +0200
+++ StepMania-3.9-src/src/RageSurface.h.new	2008-04-04 03:52:33.000000000 +0200
@@ -3,6 +3,8 @@
 #ifndef RAGE_SURFACE_H
 #define RAGE_SURFACE_H
 
+#include <cstring>
+
 /* XXX remove? */
 struct RageSurfaceColor 
 {
--- StepMania-3.9-src-orig/src/RageSoundReader.h	2004-08-21 09:01:54.000000000 +0200
+++ StepMania-3.9-src/src/RageSoundReader.h.new	2008-04-04 03:52:25.000000000 +0200
@@ -5,6 +5,8 @@
 #ifndef RAGE_SOUND_READER_H
 #define RAGE_SOUND_READER_H
 
+#include <cstring>
+
 class SoundReader
 {
 	mutable string error;
--- StepMania-3.9-src-orig/src/RageFile.h	2004-07-24 05:01:23.000000000 +0200
+++ StepMania-3.9-src/src/RageFile.h.new	2008-04-04 03:52:18.000000000 +0200
@@ -5,6 +5,8 @@
 
 #include "RageFileBasic.h"
 
+#include <cstring>
+
 /* This is the high-level interface, which interfaces with RageFileObj implementations
  * and RageFileManager. */
 class RageFile: public RageFileBasic
--- StepMania-3.9-src-orig/src/archutils/Unix/LinuxThreadHelpers.h	2004-09-08 06:45:16.000000000 +0200
+++ StepMania-3.9-src/src/archutils/Unix/LinuxThreadHelpers.h.new	2008-04-04 03:52:08.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef PID_THREAD_HELPERS_H
 #define PID_THREAD_HELPERS_H
 
+#include <cstring>
+
 RString ThreadsVersion();
 
 /* Get the current thread's ThreadID. */
--- StepMania-3.9-src-orig/src/archutils/Unix/AssertionHandler.cpp	2004-06-16 05:32:46.000000000 +0200
+++ StepMania-3.9-src/src/archutils/Unix/AssertionHandler.cpp.new	2008-04-04 03:52:00.000000000 +0200
@@ -4,6 +4,7 @@
 #include "archutils/Unix/EmergencyShutdown.h"
 #include <unistd.h>
 #include <assert.h>
+#include <cstring>
 
 /* We can define this symbol to catch failed assert() calls.  This is only used
  * for library code that uses assert(); internally we always use ASSERT, which
--- StepMania-3.9-src-orig/src/arch/Sound/RageSoundDriver.h	2004-09-09 10:51:07.000000000 +0200
+++ StepMania-3.9-src/src/arch/Sound/RageSoundDriver.h.new	2008-04-04 03:51:53.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef RAGE_SOUND_DRIVER
 #define RAGE_SOUND_DRIVER
 
+#include <cstring>
+
 class RageSoundBase;
 class RageSoundDriver
 {
--- StepMania-3.9-src-orig/src/PlayerOptions.h	2004-08-29 23:26:51.000000000 +0200
+++ StepMania-3.9-src/src/PlayerOptions.h.new	2008-04-04 03:51:45.000000000 +0200
@@ -3,6 +3,8 @@
 #ifndef PLAYER_OPTIONS_H
 #define PLAYER_OPTIONS_H
 
+#include <cstring>
+
 class Song;
 class Steps;
 class Course;
--- StepMania-3.9-src-orig/src/NoteDataUtil.h	2004-08-29 23:26:51.000000000 +0200
+++ StepMania-3.9-src/src/NoteDataUtil.h.new	2008-04-04 03:51:37.000000000 +0200
@@ -6,5 +6,6 @@
 #include "GameConstantsAndTypes.h"	// for RadarCategory
 #include "NoteTypes.h"
+#include <cstring>
 
 struct PlayerOptions;
 struct RadarValues;
--- StepMania-3.9-src-orig/src/Game.h	2004-09-06 04:48:04.000000000 +0200
+++ StepMania-3.9-src/src/Game.h.new	2008-04-04 03:51:29.000000000 +0200
@@ -7,6 +7,7 @@
 #include "MenuInput.h"
 #include "GameConstantsAndTypes.h"
 #include "RageInputDevice.h"
+#include <cstring>
 
 struct lua_State;
 
--- StepMania-3.9-src-orig/src/CodeDetector.h	2004-06-01 00:42:11.000000000 +0200
+++ StepMania-3.9-src/src/CodeDetector.h.new	2008-04-04 03:51:09.000000000 +0200
@@ -4,6 +4,7 @@
 #define CODEDETECTOR_H
 
 #include "GameInput.h"
+#include <cstring>
 
 struct CodeItem
 {
--- StepMania-3.9-src-orig/src/ScreenMiniMenu.h	2004-08-11 09:30:41.000000000 +0200
+++ StepMania-3.9-src/src/ScreenMiniMenu.h.new	2008-04-04 03:51:01.000000000 +0200
@@ -5,6 +5,7 @@
 
 #include "ScreenOptions.h"
 #include "GameConstantsAndTypes.h"
+#include <cstring>
 
 typedef bool (*MenuRowUpdateEnabled)();
 
--- StepMania-3.9-src-orig/src/HighScore.h	2004-08-22 19:23:07.000000000 +0200
+++ StepMania-3.9-src/src/HighScore.h.new	2008-04-04 03:50:50.000000000 +0200
@@ -8,6 +8,7 @@
 #include "RadarValues.h"
 #include "DateTime.h"
 #include "RageUtil_AutoPtr.h"
+#include <cstring>
 
 class XNode;
 
--- StepMania-3.9-src-orig/src/RageUtil.h	2008-04-04 03:46:16.000000000 +0200
+++ StepMania-3.9-src/src/RageUtil.h.new	2008-04-04 03:50:41.000000000 +0200
@@ -4,6 +4,8 @@
 #define RAGEUTIL_H
 
 #include <map>
+#include <climits>
+#include <cstring>
 #include <vector>
 #include "Foreach.h"
 class RageFileDriver;
--- StepMania-3.9-src-orig/src/crypto/CryptMD5.h	2004-06-08 07:30:52.000000000 +0200
+++ StepMania-3.9-src/src/crypto/CryptMD5.h.new	2008-04-04 03:50:16.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef SSHMD5_H
 #define SSHMD5_H
 
+#include <cstring>
+
 typedef struct {
 	uint32_t h[4];
 } MD5_Core_State;
--- StepMania-3.9-src-orig/src/ezsockets.h	2005-10-01 05:32:45.000000000 +0200
+++ StepMania-3.9-src/src/ezsockets.h.new	2008-04-04 03:50:08.000000000 +0200
@@ -18,6 +18,7 @@
 #include <vector>
 #include <fcntl.h>
 #include <ctype.h>
+#include <cstring>
 
 #if defined(_XBOX)
 // Summary : WinsockX is bad, XTL is good.
--- StepMania-3.9-src-orig/src/RageSound.h	2004-08-21 09:01:54.000000000 +0200
+++ StepMania-3.9-src/src/RageSound.h.new	2008-04-04 03:49:44.000000000 +0200
@@ -6,6 +6,7 @@
 #include "RageTimer.h"
 #include "RageUtil_CircularBuffer.h"
 #include "RageSoundPosMap.h"
+#include <cstring>
 
 class SoundReader;
 
--- StepMania-3.9-src-orig/src/archutils/Unix/CrashHandler.h	2004-09-08 09:34:38.000000000 +0200
+++ StepMania-3.9-src/src/archutils/Unix/CrashHandler.h.new	2008-04-04 03:49:38.000000000 +0200
@@ -3,6 +3,8 @@
 
 #include <csignal>
 #include <ucontext.h>
+#include <cstring>
+
 struct BacktraceContext;
 
 namespace CrashHandler
--- StepMania-3.9-src-orig/src/archutils/Unix/SignalHandler.h	2004-05-15 10:27:42.000000000 +0200
+++ StepMania-3.9-src/src/archutils/Unix/SignalHandler.h.new	2008-04-04 03:49:30.000000000 +0200
@@ -3,6 +3,7 @@
 
 #include <csignal>
 #include <ucontext.h>
+#include <cstring>
        
 class SaveSignals
 {
--- StepMania-3.9-src-orig/src/arch/Lights/LightsDriver.h	2004-07-07 22:11:30.000000000 +0200
+++ StepMania-3.9-src/src/arch/Lights/LightsDriver.h.new	2008-04-04 03:49:22.000000000 +0200
@@ -4,6 +4,7 @@
 #define LightsDriver_H
 
 #include "LightsManager.h"
+#include <cstring>
 
 struct LightsState;
 
--- StepMania-3.9-src-orig/src/NotesLoader.h	2004-09-13 10:02:16.000000000 +0200
+++ StepMania-3.9-src/src/NotesLoader.h.new	2008-04-04 03:49:05.000000000 +0200
@@ -3,6 +3,7 @@
 
 #include "RageUtil.h"
 #include <set>
+#include <cstring>
 
 class Song;
 
--- StepMania-3.9-src-orig/src/LyricsLoader.h	2004-06-01 00:42:12.000000000 +0200
+++ StepMania-3.9-src/src/LyricsLoader.h.new	2008-04-04 03:48:52.000000000 +0200
@@ -3,6 +3,8 @@
 #ifndef LYRICS_LOADER_H
 #define LYRICS_LOADER_H
 
+#include <cstring>
+
 class Song;
 
 class LyricsLoader
--- StepMania-3.9-src-orig/src/DateTime.h	2004-10-13 04:43:19.000000000 +0200
+++ StepMania-3.9-src/src/DateTime.h.new	2008-04-04 03:48:41.000000000 +0200
@@ -2,6 +2,7 @@
 #define DATE_TIME_H
 
 #include <ctime>
+#include <cstring>
 
 const int NUM_LAST_DAYS = 7;
 const int NUM_LAST_WEEKS = 52;
--- StepMania-3.9-src-orig/src/InputFilter.h	2004-09-10 00:14:33.000000000 +0200
+++ StepMania-3.9-src/src/InputFilter.h.new	2008-04-04 04:23:14.000000000 +0200
@@ -4,6 +4,7 @@
 #define INPUTFILTER_H
 
 #include "RageInputDevice.h"
+#include <cstring>
 
 enum InputEventType
 {
--- StepMania-3.9-src-orig/src/EnumHelper.h	2004-09-10 00:14:33.000000000 +0200
+++ StepMania-3.9-src/src/EnumHelper.h.new	2008-04-04 04:23:14.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef EnumHelper_H
 #define EnumHelper_H
 
+#include <memory>
+
 /*
  * Safely add an integer to an enum.
  *
--- StepMania-3.9-src-orig/src/LuaBinding.h	2004-09-10 00:14:33.000000000 +0200
+++ StepMania-3.9-src/src/LuaBinding.h.new	2008-04-04 04:23:14.000000000 +0200
@@ -4,6 +4,7 @@
 #define LuaBinding_H
 
 #include "LuaManager.h"
+#include <cstring>
 class LuaReference;
 
 namespace LuaBinding
--- StepMania-3.9-src-orig/src/archutils/Unix/CrashHandlerInternal.cpp	2004-09-10 00:14:33.000000000 +0200
+++ StepMania-3.9-src/src/archutils/Unix/CrashHandlerInternal.cpp.new	2008-04-04 04:23:14.000000000 +0200
@@ -2,6 +2,7 @@
 #include "CrashHandlerInternal.h"
 
 #include <csignal>
+#include <cstring>
 
 const char *itoa( unsigned n )
 {
