--- StepMania-3.9-src-orig/src/RoomWheel.h	2004-08-29 23:35:14.000000000 -0500
+++ StepMania-3.9-src/src/RoomWheel.h	2006-11-18 13:36:18.000000000 -0600
@@ -10,7 +10,7 @@
 
 struct RoomWheelData : public WheelItemBaseData
 {
-	RoomWheelData() { WheelItemBaseData::WheelItemBaseData(); }
+	RoomWheelData() : WheelItemBaseData() { }
 	RoomWheelData( WheelItemType wit, RString title, RString SubTitle, RageColor color );
 
 	RString			m_sDesc;
