--- StepMania-3.9-src-orig/src/arch/InputHandler/InputHandler_SDL.cpp	2004-09-09 21:12:40.000000000 -0500
+++ StepMania-3.9-src/src/arch/InputHandler/InputHandler_SDL.cpp	2006-06-18 00:09:53.000000000 -0500
@@ -123,7 +123,7 @@
 	SDL_KEYDOWN, SDL_KEYUP, SDL_JOYBUTTONDOWN, SDL_JOYBUTTONUP,
 	SDL_JOYAXISMOTION, SDL_JOYHATMOTION, -1
 };
-static int SDL_EventMask;
+static int SDL_EventMask2;
 
 InputHandler_SDL::InputHandler_SDL()
 {
@@ -173,7 +173,7 @@
 	for(i = 0; Handled_SDL_Events[i] != -1; ++i)
 	{
 		mySDL_EventState(Handled_SDL_Events[i], SDL_ENABLE);
-		SDL_EventMask |= SDL_EVENTMASK(Handled_SDL_Events[i]);
+		SDL_EventMask2 |= SDL_EVENTMASK(Handled_SDL_Events[i]);
 	}
 }
 
@@ -192,7 +192,7 @@
 void InputHandler_SDL::Update(float fDeltaTime)
 {
 	SDL_Event event;
-	while(SDL_GetEvent(event, SDL_EventMask))
+	while(SDL_GetEvent(event, SDL_EventMask2))
 	{
 		switch(event.type)
 		{
