Index: src/arch/MemoryCard/MemoryCardDriverThreaded_Linux.cpp
===================================================================
--- src/arch/MemoryCard/MemoryCardDriverThreaded_Linux.cpp	(revision 28364)
+++ src/arch/MemoryCard/MemoryCardDriverThreaded_Linux.cpp	(working copy)
@@ -6,6 +6,7 @@
 
 #include <cerrno>
 #include <fcntl.h>
+#include <sys/stat.h>
 #include <dirent.h>
 
 bool MemoryCardDriverThreaded_Linux::TestWrite( UsbStorageDevice* pDevice )
Index: src/arch/InputHandler/InputHandler_Linux_Joystick.cpp
===================================================================
--- src/arch/InputHandler/InputHandler_Linux_Joystick.cpp	(revision 28364)
+++ src/arch/InputHandler/InputHandler_Linux_Joystick.cpp	(working copy)
@@ -7,6 +7,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <sys/stat.h>
 
 #include <errno.h>
 #include <sys/types.h>
