=== modified file 'src/image.c'
--- src/image.c	2014-06-04 21:50:06 +0000
+++ src/image.c	2014-06-18 22:42:07 +0000
@@ -7410,7 +7410,11 @@
   if (!check_image_size (f, gif->SWidth, gif->SHeight))
     {
       image_error ("Invalid image size (see `max-image-size')", Qnil, Qnil);
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1 
+      fn_DGifCloseFile (gif, NULL);
+#else
       fn_DGifCloseFile (gif);
+#endif
       return 0;
     }
 
@@ -7419,7 +7423,11 @@
   if (rc == GIF_ERROR || gif->ImageCount <= 0)
     {
       image_error ("Error reading `%s'", img->spec, Qnil);
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+      fn_DGifCloseFile (gif, NULL);
+#else
       fn_DGifCloseFile (gif);
+#endif
       return 0;
     }
 
@@ -7431,7 +7439,11 @@
       {
 	image_error ("Invalid image number `%s' in image `%s'",
 		     image_number, img->spec);
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+	fn_DGifCloseFile (gif, NULL);
+#else
 	fn_DGifCloseFile (gif);
+#endif
 	return 0;
       }
   }
@@ -7449,7 +7461,11 @@
   if (!check_image_size (f, width, height))
     {
       image_error ("Invalid image size (see `max-image-size')", Qnil, Qnil);
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+      fn_DGifCloseFile (gif, NULL);
+#else
       fn_DGifCloseFile (gif);
+#endif
       return 0;
     }
 
@@ -7467,7 +7483,11 @@
 	     && 0 <= subimg_left && subimg_left <= width - subimg_width))
 	{
 	  image_error ("Subimage does not fit in image", Qnil, Qnil);
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+	  fn_DGifCloseFile (gif, NULL);
+#else
 	  fn_DGifCloseFile (gif);
+#endif
 	  return 0;
 	}
     }
@@ -7475,7 +7495,11 @@
   /* Create the X image and pixmap.  */
   if (!image_create_x_image_and_pixmap (f, img, width, height, 0, &ximg, 0))
     {
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+      fn_DGifCloseFile (gif, NULL);
+#else
       fn_DGifCloseFile (gif);
+#endif
       return 0;
     }
 
@@ -7645,9 +7669,11 @@
     img->lisp_data = Fcons (Qcount,
 			    Fcons (make_number (gif->ImageCount),
 				   img->lisp_data));
-
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+  fn_DGifCloseFile (gif, NULL);
+#else
   fn_DGifCloseFile (gif);
-
+#endif
   /* Maybe fill in the background field while we have ximg handy. */
   if (NILP (image_spec_value (img->spec, QCbackground, NULL)))
     /* Casting avoids a GCC warning.  */

