--- makefile.orig	2006-02-20 03:20:38.000000000 -0500
+++ makefile	2006-02-20 03:20:59.000000000 -0500
@@ -39,22 +39,22 @@
 # Makefile commands:---------------------------------------------------------
 
 all:	bobobot
-	@echo "... Making sure data directory ($(DATA)/...) exists"
-	- mkdir $(DATA)
-	chmod 755 $(DATA)
+	@echo "... Making sure data directory ($(DESTDIR)/$(DATA)/...) exists"
+	- mkdir -p $(DESTDIR)/$(DATA)
+	chmod 755 $(DESTDIR)/$(DATA)
 	@echo "... Making sure image, sound and level directories exist"
-	- mkdir $(DATA)/sounds $(DATA)/images $(DATA)/levels $(DATA)/mods
-	chmod 755 $(DATA)/sounds $(DATA)/images $(DATA)/levels $(DATA)/mods
-	@echo "... Copying data into data directories ($(DATA)/...)"
-	- cp sounds/* $(DATA)/sounds 2> /dev/null
-	- cp images/* $(DATA)/images 2> /dev/null
-	- cp levels/* $(DATA)/levels 2> /dev/null
-	- cp mods/* $(DATA)/mods 2> /dev/null
-	chmod 644 $(DATA)/sounds/* $(DATA)/images/* $(DATA)/levels/* \
-		$(DATA)/mods/*
+	- mkdir -p $(DESTDIR)/$(DATA)/sounds $(DESTDIR)/$(DATA)/images $(DESTDIR)/$(DATA)/levels $(DESTDIR)/$(DATA)/mods
+	chmod 755 $(DESTDIR)/$(DATA)/sounds $(DESTDIR)/$(DATA)/images $(DESTDIR)/$(DATA)/levels $(DESTDIR)/$(DATA)/mods
+	@echo "... Copying data into data directories ($(DESTDIR)/$(DATA)/...)"
+	- cp sounds/* $(DESTDIR)/$(DATA)/sounds 2> /dev/null
+	- cp images/* $(DESTDIR)/$(DATA)/images 2> /dev/null
+	- cp levels/* $(DESTDIR)/$(DATA)/levels 2> /dev/null
+	- cp mods/* $(DESTDIR)/$(DATA)/mods 2> /dev/null
+	chmod 644 $(DESTDIR)/$(DATA)/sounds/* $(DESTDIR)/$(DATA)/images/* $(DESTDIR)/$(DATA)/levels/* \
+		$(DESTDIR)/$(DATA)/mods/*
 	@echo "... Copying executable into installation location \
-		($(INSTALLROOT))"
-	- cp bobobot $(INSTALLROOT) 2> /dev/null
+		($(DESTDIR)/$(INSTALLROOT))"
+	- cp bobobot $(DESTDIR)/$(INSTALLROOT) 2> /dev/null
 
 clean:
 	- rm obj/*.o bobobot
