--- jss-4.2.6.orig/security/jss/org/mozilla/jss/ssl/SSLServerSocket.c	2010-11-17 18:54:56.000000000 -0500
+++ jss-4.2.6/security/jss/org/mozilla/jss/ssl/SSLServerSocket.c	2010-11-18 09:46:34.000000000 -0500
@@ -124,20 +124,16 @@
             /* Clean up after PR_interrupt. */
             PR_NT_CancelIo(sock->fd);
 #endif
-            JSSL_throwSSLSocketException(env, 
-                "Accept operation interrupted with error code " + err);
+            JSSL_throwSSLSocketException(env, "Accept operation interrupted");
         } else if( err == PR_IO_TIMEOUT_ERROR ) {
 #ifdef WINNT
             PR_NT_CancelIo(sock->fd);
 #endif
-            JSSL_throwSSLSocketException(env, 
-                "Accept operation timed out with error code " + err);
+            JSSL_throwSSLSocketException(env, "Accept operation timed out");
         } else if( err == PR_IO_ERROR ) {
-            JSSL_throwSSLSocketException(env, 
-                "Accept operation received IO error with error code " + err);
+            JSSL_throwSSLSocketException(env, "Accept operation received IO error");
         } else {
-            JSSL_throwSSLSocketException(env, 
-                "Accept operation failed with error code " + err);
+            JSSL_throwSSLSocketException(env, "Accept operation failed");
         }
         goto finish;
     }
