diff -ur ../sources.orig/configure ./configure
--- ../sources.orig/configure	2011-12-09 10:55:11.761537125 +0100
+++ ./configure	2011-12-09 12:17:40.918145411 +0100
@@ -17549,13 +17549,13 @@
 
 
 # Boost.Regex
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lboost_regex-mt" >&5
-$as_echo_n "checking for main in -lboost_regex-mt... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lboost_regex" >&5
+$as_echo_n "checking for main in -lboost_regex... " >&6; }
 if test "${ac_cv_lib_boost_regex_mt_main+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lboost_regex-mt  $LIBS"
+LIBS="-lboost_regex  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -17580,7 +17580,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_boost_regex_mt_main" >&5
 $as_echo "$ac_cv_lib_boost_regex_mt_main" >&6; }
 if test "x$ac_cv_lib_boost_regex_mt_main" = x""yes; then :
-  REGEX_LIBS='-lboost_regex-mt'
+  REGEX_LIBS='-lboost_regex'
 else
   as_fn_error $? "Boost.Regex not found" "$LINENO" 5
 fi
@@ -17588,13 +17588,13 @@
 
 
 # Boost.Filesystem
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lboost_filesystem-mt" >&5
-$as_echo_n "checking for main in -lboost_filesystem-mt... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lboost_filesystem" >&5
+$as_echo_n "checking for main in -lboost_filesystem... " >&6; }
 if test "${ac_cv_lib_boost_filesystem_mt_main+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lboost_filesystem-mt  $LIBS"
+LIBS="-lboost_filesystem  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -17619,7 +17619,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_boost_filesystem_mt_main" >&5
 $as_echo "$ac_cv_lib_boost_filesystem_mt_main" >&6; }
 if test "x$ac_cv_lib_boost_filesystem_mt_main" = x""yes; then :
-  BOOST_FILESYSTEM_LIBS='-lboost_filesystem-mt'
+  BOOST_FILESYSTEM_LIBS='-lboost_filesystem'
 else
   as_fn_error $? "Boost.Filesystem not found" "$LINENO" 5
 fi
@@ -17627,13 +17627,13 @@
 
 
 # Boost.System
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lboost_system-mt" >&5
-$as_echo_n "checking for main in -lboost_system-mt... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lboost_system" >&5
+$as_echo_n "checking for main in -lboost_system... " >&6; }
 if test "${ac_cv_lib_boost_system_mt_main+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lboost_system-mt  $LIBS"
+LIBS="-lboost_system  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -17658,7 +17658,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_boost_system_mt_main" >&5
 $as_echo "$ac_cv_lib_boost_system_mt_main" >&6; }
 if test "x$ac_cv_lib_boost_system_mt_main" = x""yes; then :
-  BOOST_SYSTEM_LIBS='-lboost_system-mt'
+  BOOST_SYSTEM_LIBS='-lboost_system'
 else
   as_fn_error $? "Boost.System not found" "$LINENO" 5
 fi
@@ -17761,17 +17761,17 @@
 fi
 
 
-    # Also check boost_python-mt since apparently it seems to randomly change on
+    # Also check boost_python since apparently it seems to randomly change on
     # different Ubuntu distributions.
     if test -z "$BOOST_PYTHON_LIBS"
     then
-        { $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lboost_python-mt" >&5
-$as_echo_n "checking for main in -lboost_python-mt... " >&6; }
+        { $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lboost_python" >&5
+$as_echo_n "checking for main in -lboost_python... " >&6; }
 if test "${ac_cv_lib_boost_python_mt_main+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lboost_python-mt  $LIBS"
+LIBS="-lboost_python  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -17796,7 +17796,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_boost_python_mt_main" >&5
 $as_echo "$ac_cv_lib_boost_python_mt_main" >&6; }
 if test "x$ac_cv_lib_boost_python_mt_main" = x""yes; then :
-  BOOST_PYTHON_LIBS='-lboost_python-mt'
+  BOOST_PYTHON_LIBS='-lboost_python'
 else
   script_module=''
 fi
diff -ur ../sources.orig/configure.ac ./configure.ac
--- ../sources.orig/configure.ac	2011-12-09 10:55:11.708205642 +0100
+++ ./configure.ac	2011-12-09 12:17:40.918145411 +0100
@@ -100,20 +100,20 @@
 AC_SUBST([GLU_LIBS])
 
 # Boost.Regex
-AC_CHECK_LIB([boost_regex-mt], [main],
-             [REGEX_LIBS='-lboost_regex-mt'],
+AC_CHECK_LIB([boost_regex], [main],
+             [REGEX_LIBS='-lboost_regex'],
              [AC_MSG_ERROR([Boost.Regex not found])])
 AC_SUBST([REGEX_LIBS])
 
 # Boost.Filesystem
-AC_CHECK_LIB([boost_filesystem-mt], [main],
-             [BOOST_FILESYSTEM_LIBS='-lboost_filesystem-mt'],
+AC_CHECK_LIB([boost_filesystem], [main],
+             [BOOST_FILESYSTEM_LIBS='-lboost_filesystem'],
              [AC_MSG_ERROR([Boost.Filesystem not found])])
 AC_SUBST([BOOST_FILESYSTEM_LIBS])
 
 # Boost.System
-AC_CHECK_LIB([boost_system-mt], [main],
-             [BOOST_SYSTEM_LIBS='-lboost_system-mt'],
+AC_CHECK_LIB([boost_system], [main],
+             [BOOST_SYSTEM_LIBS='-lboost_system'],
              [AC_MSG_ERROR([Boost.System not found])])
 AC_SUBST([BOOST_SYSTEM_LIBS])
 
@@ -136,12 +136,12 @@
                  [BOOST_PYTHON_LIBS='-lboost_python'],
                  [])
 
-    # Also check boost_python-mt since apparently it seems to randomly change on
+    # Also check boost_python since apparently it seems to randomly change on
     # different Ubuntu distributions.
     if test -z "$BOOST_PYTHON_LIBS"
     then
-        AC_CHECK_LIB([boost_python-mt], [main],
-                     [BOOST_PYTHON_LIBS='-lboost_python-mt'],
+        AC_CHECK_LIB([boost_python], [main],
+                     [BOOST_PYTHON_LIBS='-lboost_python'],
                      [script_module=''])
     fi
 
diff -ur ../sources.orig/plugins/dm.difficulty/DifficultyEditor.cpp ./plugins/dm.difficulty/DifficultyEditor.cpp
--- ../sources.orig/plugins/dm.difficulty/DifficultyEditor.cpp	2011-12-09 10:55:11.714872080 +0100
+++ ./plugins/dm.difficulty/DifficultyEditor.cpp	2011-12-09 12:17:41.018140872 +0100
@@ -184,7 +184,7 @@
 	Gtk::Label* argumentLabel = Gtk::manage(new gtkutil::LeftAlignedLabel(_("Argument:")));
 
 	// The appType chooser
-	_appTypeCombo = Gtk::manage(new Gtk::ComboBox(difficulty::Setting::getAppTypeStore()));
+	_appTypeCombo = Gtk::manage(new Gtk::ComboBox(Glib::RefPtr<Gtk::TreeModel>(difficulty::Setting::getAppTypeStore())));
 	_appTypeCombo->signal_changed().connect(sigc::mem_fun(*this, &DifficultyEditor::onAppTypeChange));
 
 	// Add the cellrenderer for the apptype text
diff -ur ../sources.orig/plugins/dm.stimresponse/ClassEditor.cpp ./plugins/dm.stimresponse/ClassEditor.cpp
--- ../sources.orig/plugins/dm.stimresponse/ClassEditor.cpp	2011-12-09 10:55:11.718205298 +0100
+++ ./plugins/dm.stimresponse/ClassEditor.cpp	2011-12-09 12:17:41.018140872 +0100
@@ -153,7 +153,7 @@
 	widgets.label = Gtk::manage(new gtkutil::LeftAlignedLabel(_("Type:")));
 
 	// Cast the helper class onto a ListStore and create a new treeview
-	widgets.list = Gtk::manage(new Gtk::ComboBox(_stimTypes.getListStore()));
+	widgets.list = Gtk::manage(new Gtk::ComboBox(Glib::RefPtr<Gtk::TreeModel>(_stimTypes.getListStore())));
 	widgets.list->set_size_request(-1, -1);
 
 	// Add the cellrenderer for the name
diff -ur ../sources.orig/plugins/dm.stimresponse/EffectArgumentItem.cpp ./plugins/dm.stimresponse/EffectArgumentItem.cpp
--- ../sources.orig/plugins/dm.stimresponse/EffectArgumentItem.cpp	2011-12-09 10:55:11.718205298 +0100
+++ ./plugins/dm.stimresponse/EffectArgumentItem.cpp	2011-12-09 12:17:41.018140872 +0100
@@ -139,7 +139,7 @@
 	_stimTypes(stimTypes)
 {
 	// Cast the helper class onto a ListStore and create a new treeview
-	_comboBox = Gtk::manage(new Gtk::ComboBox(_stimTypes.getListStore()));
+	_comboBox = Gtk::manage(new Gtk::ComboBox(Glib::RefPtr<Gtk::TreeModel>(_stimTypes.getListStore())));
 
 	// Add the cellrenderer for the name
 	Gtk::CellRendererText* nameRenderer = Gtk::manage(new Gtk::CellRendererText);
diff -ur ../sources.orig/plugins/particles/ParticlesManager.cpp ./plugins/particles/ParticlesManager.cpp
--- ../sources.orig/plugins/particles/ParticlesManager.cpp	2011-12-09 10:55:11.721538515 +0100
+++ ./plugins/particles/ParticlesManager.cpp	2011-12-09 12:17:47.434518598 +0100
@@ -289,11 +289,7 @@
 	fs::path tempFile = targetFile;
 	
 	tempFile.remove_filename();
-#if BOOST_VERSION < 104600
 	tempFile /= "_" + targetFile.filename();
-#else
-	tempFile /= "_" + targetFile.filename().string();
-#endif
 
 	std::ofstream tempStream(tempFile.string().c_str());
 
diff -ur ../sources.orig/radiant/map/AutoSaver.cpp ./radiant/map/AutoSaver.cpp
--- ../sources.orig/radiant/map/AutoSaver.cpp	2011-12-09 10:55:11.691539559 +0100
+++ ./radiant/map/AutoSaver.cpp	2011-12-09 12:17:47.434518598 +0100
@@ -115,11 +115,11 @@
 	snapshotPath /= GlobalRegistry().get(RKEY_AUTOSAVE_SNAPSHOTS_FOLDER);
 
 	// Retrieve the mapname
-#if BOOST_VERSION < 104600
+// #if BOOST_VERSION < 104600
 	std::string mapName = fullPath.filename();
-#else
-	std::string mapName = fullPath.filename().string();
-#endif
+// #else
+// 	std::string mapName = fullPath.filename().string();
+// #endif
 
 	// Check if the folder exists and create it if necessary
 	if (boost::filesystem::exists(snapshotPath) || os::makeDirectory(snapshotPath.string()))
