--- mem_linux.c	2003-03-17 13:23:05.000000000 +0000
+++ mem_linux.c1	2006-09-14 00:01:21.000000000 +0000
@@ -34,10 +34,22 @@
   perror("/proc/meminfo");
   exit(1);
  }
- while(fgetc(file)!='\n'){}
- fscanf(file, "%*s %Ld %Ld %Ld %Ld %Ld %Ld",
-        &mem_total, &mem_used, &mem_free, &mem_shared, &mem_buffers, &mem_cached);
- fscanf(file, "%*s %Ld %Ld %Ld",
-        &swp_total, &swp_used, &swp_free);
+ mem_shared = 0;
+ fscanf(file,"%*s %Ld %*s",&mem_total);
+ fscanf(file,"%*s %Ld %*s",&mem_free);
+ fscanf(file,"%*s %Ld %*s",&mem_buffers);
+ fscanf(file,"%*s %Ld %*s",&mem_cached);
+ fscanf(file,"%*s %*d %*s");
+ fscanf(file,"%*s %Ld %*s",&mem_used);
+ fscanf(file,"%*s %*d %*s");
+ fscanf(file,"%*s %*d %*s");
+ fscanf(file,"%*s %*d %*s");
+ fscanf(file,"%*s %*d %*s");
+ fscanf(file,"%*s %*d %*s");
+ fscanf(file,"%*s %Ld %*s",&swp_total);
+ fscanf(file,"%*s %Ld %*s",&swp_free);
+ fscanf(file,"%*s %Ld %*s");
+ fscanf(file,"%*s %Ld %*s");
+ fscanf(file,"%*s %Ld %*s",&swp_used);
  fclose(file);
 }
