diff -ur a/eclipse.desktop b/eclipse.desktop
--- eclipse.desktop	2011-08-07 08:38:07.253771166 +0200
+++ b/eclipse.desktop	2011-08-07 08:47:03.400451335 +0200
@@ -1,8 +1,8 @@
 [Desktop Entry]
-Name=Eclipse
-Comment=A Java Development Environment
-Exec=eclipse
-Icon=eclipse
+Name=Eclipse-jee
+Comment=Eclipse IDE for Java EE Developers
+Exec=eclipse-jee
+Icon=eclipse-jee
 Terminal=false
 Type=Application
 Categories=Java;Development;IDE;
