diff -ru suspend-utils-9a5329f.orig/s2ram-main.c suspend-utils-9a5329f/s2ram-main.c
--- suspend-utils-9a5329f.orig/s2ram-main.c	2009-09-22 23:17:44.000000000 +0200
+++ suspend-utils-9a5329f/s2ram-main.c		2010-03-28 15:09:08.203379624 +0200
@@ -84,10 +84,11 @@
 	if (ret)
 		goto out;
 
-	/* switch to console 1 first, since we might be in X */
+	/* switch to console 63 first, since we might be in X,
+	 * but don't want to see the console first */
 	active_console = fgconsole();
-	printf("Switching from vt%d to vt1", active_console);
-	if (chvt(1))
+	printf("Switching from vt%d to vt63", active_console);
+	if (chvt(63))
 		printf("... succeeded\n");
 	else
 		printf("... failed\n");
