diff -Naur a/backend/configure.in b/backend/configure.in
--- a/backend/configure.in	2012-09-03 08:41:01.000000000 +0200
+++ b/backend/configure.in	2013-04-27 17:51:41.656416291 +0200
@@ -1,8 +1,6 @@
 AC_INIT(src/cnij_backend_common.c)
 AM_INIT_AUTOMAKE(backend, 3.90-1)
 
-AM_CONFIG_HEADER(config.h)
-
 AC_ARG_ENABLE(progpath)
 if test "${enable_progpath}" = "" ; then
    enable_progpath="${ac_default_prefix}/bin"
diff -Naur a/backendnet/configure.in b/backendnet/configure.in
--- a/backendnet/configure.in	2012-09-03 08:41:01.000000000 +0200
+++ b/backendnet/configure.in	2013-04-27 17:51:41.659749624 +0200
@@ -1,8 +1,6 @@
 AC_INIT(backend/cnijnetlm.c)
 AM_INIT_AUTOMAKE(backendnet, 3.90-1)
 
-AM_CONFIG_HEADER(config.h)
-
 AC_ARG_ENABLE(progpath)
 if test "${enable_progpath}" = "" ; then
    enable_progpath="${ac_default_prefix}/bin"
diff -Naur a/cngpij/configure.in b/cngpij/configure.in
--- a/cngpij/configure.in	2012-09-03 08:41:01.000000000 +0200
+++ b/cngpij/configure.in	2013-04-27 17:51:41.659749624 +0200
@@ -1,8 +1,6 @@
 AC_INIT(cngpij/bjcups.c)
 AM_INIT_AUTOMAKE(cngpij, 3.90-1)
 
-AM_CONFIG_HEADER(config.h)
-
 AC_ARG_ENABLE(progpath)
 if test "${enable_progpath}" = "" ; then
    enable_progpath="${ac_default_prefix}/bin"
diff -Naur a/cngpijmnt/configure.in b/cngpijmnt/configure.in
--- a/cngpijmnt/configure.in	2012-09-03 08:41:00.000000000 +0200
+++ b/cngpijmnt/configure.in	2013-04-27 17:51:41.656416291 +0200
@@ -1,8 +1,6 @@
 AC_INIT(src/main.c)
 AM_INIT_AUTOMAKE(cngpijmnt, 3.90-1)
 
-AM_CONFIG_HEADER(config.h)
-
 AC_ARG_ENABLE(progpath)
 if test "${enable_progpath}" = "" ; then
    enable_progpath="${ac_default_prefix}/bin"
diff -Naur a/cngpijmon/cnijnpr/config.h.in b/cngpijmon/cnijnpr/config.h.in
--- a/cngpijmon/cnijnpr/config.h.in	1970-01-01 01:00:00.000000000 +0100
+++ b/cngpijmon/cnijnpr/config.h.in	2013-04-27 18:36:24.176410521 +0200
@@ -0,0 +1 @@
+/* config.h.in.  Generated from configure.in by autoheader.  */
diff -Naur a/cngpijmon/cnijnpr/configure.in b/cngpijmon/cnijnpr/configure.in
--- a/cngpijmon/cnijnpr/configure.in	2012-09-07 04:41:39.000000000 +0200
+++ b/cngpijmon/cnijnpr/configure.in	2013-04-27 18:08:50.059747412 +0200
@@ -1,8 +1,7 @@
 AC_INIT(configure.in)
 AM_INIT_AUTOMAKE(cnijnpr, 3.80-1)
 
-AM_CONFIG_HEADER(config.h)
-
+AC_CONFIG_HEADERS(config.h)
 
 AC_ARG_ENABLE(libpath)
 if test "${enable_libpath}" = "" ; then
diff -Naur a/cngpijmon/configure.in b/cngpijmon/configure.in
--- a/cngpijmon/configure.in	2012-09-07 04:41:38.000000000 +0200
+++ b/cngpijmon/configure.in	2013-04-27 17:51:41.666416292 +0200
@@ -18,11 +18,10 @@
 
 
 AM_INIT_AUTOMAKE(cngpijmon${program_suffix}, 3.90-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(src/config.h)
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 
 AM_PATH_GTK_2_0(2.0.0, ,
diff -Naur a/cngpijmon/src/config.h.in b/cngpijmon/src/config.h.in
--- a/cngpijmon/src/config.h.in	1970-01-01 01:00:00.000000000 +0100
+++ b/cngpijmon/src/config.h.in	2013-04-27 18:36:24.176410521 +0200
@@ -0,0 +1 @@
+/* src/config.h.in.  Generated from configure.in by autoheader.  */
diff -Naur a/cnijfilter/configure.in b/cnijfilter/configure.in
--- a/cnijfilter/configure.in	2012-09-06 02:25:49.000000000 +0200
+++ b/cnijfilter/configure.in	2013-04-27 17:51:41.646416291 +0200
@@ -25,7 +25,7 @@
 
 
 AM_INIT_AUTOMAKE(cif${program_suffix}, 3.80-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(src/config.h)
 
 dnl Checks for programs.
 AC_ISC_POSIX
@@ -45,7 +45,6 @@
 AC_SUBST(ARC)
 
 AC_PROG_CC
-AM_PROG_CC_STDC
 
 dnl Checks for libraries.
 
diff -Naur a/lgmon/configure.in b/lgmon/configure.in
--- a/lgmon/configure.in	2012-10-18 03:20:13.000000000 +0200
+++ b/lgmon/configure.in	2013-04-27 17:51:41.656416291 +0200
@@ -9,7 +9,7 @@
 fi
 
 AM_INIT_AUTOMAKE(lgmon{program_suffix}, 3.90-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(src/config.h)
 dnl AM_MAINTAINER_MODE
 
 AC_DEFINE_UNQUOTED(PACKAGE_PRINTER_MODEL, "${program_suffix}")
@@ -39,7 +39,6 @@
 AC_SUBST(ARC)
 
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AC_PROG_RANLIB
 
diff -Naur a/lgmon/src/config.h.in b/lgmon/src/config.h.in
--- a/lgmon/src/config.h.in	1970-01-01 01:00:00.000000000 +0100
+++ b/lgmon/src/config.h.in	2013-04-27 18:36:24.173077187 +0200
@@ -0,0 +1 @@
+/* src/config.h.in.  Generated from configure.in by autoheader.  */
diff -Naur a/libs/configure.in b/libs/configure.in
--- a/libs/configure.in	2012-09-03 08:41:02.000000000 +0200
+++ b/libs/configure.in	2013-04-27 18:23:58.356412125 +0200
@@ -1,8 +1,6 @@
 AC_INIT(paramlist/paramlist.h)
 AM_INIT_AUTOMAKE(buftool, 1.0.0)
 
-AM_CONFIG_HEADER(config.h)
-
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
diff -Naur a/maintenance/configure.in b/maintenance/configure.in
--- a/maintenance/configure.in	2012-09-04 11:35:36.000000000 +0200
+++ b/maintenance/configure.in	2013-04-27 17:51:41.653082958 +0200
@@ -9,7 +9,7 @@
 fi
 
 AM_INIT_AUTOMAKE(maintenance${program_suffix}, 3.90-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(src/config.h)
 AM_MAINTAINER_MODE
 
 AC_DEFINE_UNQUOTED(PACKAGE_PRINTER_MODEL, "${program_suffix}")
@@ -43,7 +43,6 @@
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 
 AM_PATH_GTK_2_0(2.6.0, ,
diff -Naur a/maintenance/src/config.h.in b/maintenance/src/config.h.in
--- a/maintenance/src/config.h.in	1970-01-01 01:00:00.000000000 +0100
+++ b/maintenance/src/config.h.in	2013-04-27 18:36:24.173077187 +0200
@@ -0,0 +1 @@
+/* src/config.h.in.  Generated from configure.in by autoheader.  */
diff -Naur a/pstocanonij/configure.in b/pstocanonij/configure.in
--- a/pstocanonij/configure.in	2012-09-03 08:41:00.000000000 +0200
+++ b/pstocanonij/configure.in	2013-04-27 17:51:41.649749624 +0200
@@ -1,8 +1,6 @@
 AC_INIT(filter/pstocanonij.c)
 AM_INIT_AUTOMAKE(pstocanonij, 1.0.0)
 
-AM_CONFIG_HEADER(config.h)
-
 AC_ARG_ENABLE(progpath)
 if test "${enable_progpath}" = "" ; then
    enable_progpath="${ac_default_prefix}/bin"
