--- win/configure.in.orig	2011-11-04 13:47:58.000000000 +0100
+++ win/configure.in	2012-01-16 22:29:19.438943290 +0100
@@ -56,9 +56,9 @@
 AC_C_INLINE
 AC_HEADER_STDC
 
-AC_CHECK_TOOL(AR, ar)
-AC_CHECK_TOOL(RANLIB, ranlib)
-AC_CHECK_TOOL(RC, windres)
+AC_CHECK_TOOL(AR, i686-w64-mingw32-ar)
+AC_CHECK_TOOL(RANLIB, i686-w64-mingw32-ranlib)
+AC_CHECK_TOOL(RC, i686-w64-mingw32-windres)
 
 #--------------------------------------------------------------------
 # Checks to see if the make program sets the $MAKE variable.
