--- pan/general/file-util.h
+++ pan/general/file-util.h
@@ -22,6 +22,7 @@
 #define _FileUtil_h_
 
 #include <string>
+#include <cstdlib>
 extern "C" {
   #include <stddef.h>
   #include <stdio.h>
--- pan/general/file-util.h	2007-12-10 17:38:10.000000000 +0100
+++ pan/general/file-util.h	2007-12-14 11:20:32.000000000 +0100
@@ -23,6 +23,8 @@

 #include <string>
 #include <cstdlib>
+#include <cstdio>
+#include <bits/stl_algo.h>
 extern "C" {
   #include <stddef.h>
   #include <stdio.h>
--- pan//usenet-utils/gnksa.h	2007-08-01 18:59:59.000000000 +0200
+++ pan/usenet-utils/gnksa.h	2007-12-14 11:27:47.000000000 +0100
@@ -22,6 +22,7 @@

 #include <string>
 #include <ctime>
+#include <climits>

 namespace pan
 {
--- pan//usenet-utils/numbers.h	2007-08-01 18:59:59.000000000 +0200
+++ pan/usenet-utils/numbers.h	2007-12-14 11:23:42.000000000 +0100
@@ -22,6 +22,7 @@

 #include <vector>
 #include <pan/general/string-view.h>
+#include <bits/stl_algo.h>

 class string;

--- pan/gui/prefs.h	2007-08-01 19:00:00.000000000 +0200
+++ pan/gui/prefs.h	2007-12-14 11:32:47.000000000 +0100
@@ -23,6 +23,8 @@
 #include <map>
 #include <set>
 #include <string>
+#include <cstdlib>
+#include <cstring>
 #include <vector>
 #include <pan/general/string-view.h>
 #include <gtk/gtk.h>
--- pan/tasks/socket.h	2007-08-01 19:00:00.000000000 +0200
+++ pan/tasks/socket.h	2007-12-14 11:30:23.000000000 +0100
@@ -21,6 +21,7 @@
 #define __Socket_h__

 #include <string>
+#include <bits/stl_algo.h>

 namespace pan
 {
--- pan/general/string-view.h
+++ pan/general/string-view.h
@@ -22,6 +22,7 @@

 #include <iosfwd>
 #include <string>
+#include <cstring>

 namespace pan
 {
