--- Makefile	2010-03-10 20:05:04.000000000 +0100
+++ Makefile_new	2011-12-10 10:06:29.789474620 +0100
@@ -26,7 +26,7 @@
 
 
 LOOPBACK_O =	loopback.o
-alsa-loopback:	LDLIBS += -lclthreads -lclalsadrv -lpthread -lrt
+alsa-loopback:	LDLIBS += -lclthreads -lclalsadrv -lasound -lpthread -lrt
 alsa-loopback:	$(LOOPBACK_O)
 	g++ $(LDFLAGS) -o $@ $(LOOPBACK_O) $(LDLIBS)
 $(LOOPBACK_O):
@@ -34,7 +34,7 @@
 
 
 LATENCY_O =	latency.o
-alsa-latency:	LDLIBS += -lclthreads -lclalsadrv -lpthread -lrt
+alsa-latency:	LDLIBS += -lclthreads -lclalsadrv -lasound -lpthread -lrt
 alsa-latency:	$(LATENCY_O)
 	g++ $(LDFLAGS) -o $@ $(LATENCY_O) $(LDLIBS)
 $(LATENCY_O):
