--- Makefile.orig	2012-08-25 23:01:26.161416204 +0400
+++ Makefile	2012-08-25 23:05:03.910236564 +0400
@@ -25,4 +25,3 @@
 	install -c -m 0644 ifd-ASEDriveIIIe-USB.bundle/Contents/Resources/English.lproj/InfoPlist.strings "${DRIVER_DIR}/Contents/Resources/English.lproj"
 	install -c -d "${DRIVER_DIR}/Contents/Linux"
 	install -c -m 0755 libASEDriveIIIe-USB.so "${DRIVER_DIR}/Contents/Linux"
-	install -c -m 0644 50-pcscd-asedriveiiie.rules /etc/udev/rules.d/.
