diff -rupN sparkleshare-0.8.4-orig/SparkleShare/Makefile sparkleshare-0.8.4/SparkleShare/Makefile
--- sparkleshare-0.8.4-orig/SparkleShare/Makefile	2012-03-19 11:25:19.704395999 -0400
+++ sparkleshare-0.8.4/SparkleShare/Makefile	2012-03-19 11:27:29.696612143 -0400
@@ -786,7 +786,6 @@ $(ASSEMBLY_FILE): $(SOURCES_BUILD) $(RES
 	fi;
 
 install-data-hook:
-	update-desktop-database $(datadir)/applications
 	echo "Installed protocol handler sparkleshare://"
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
