diff -rupN sparkleshare-0.2-rc1/SparkleShare/SparkleIntro.cs sparkleshare-0.2-rc1/SparkleShare/SparkleIntro.cs
--- sparkleshare-0.2-rc1/SparkleShare/SparkleIntro.cs	2011-04-17 03:36:14.000000000 -0400
+++ sparkleshare-0.2-rc1/SparkleShare/SparkleIntro.cs	2011-04-17 03:37:12.000000000 -0400
@@ -81,6 +81,7 @@ namespace SparkleShare {
 				};
 
 					string full_name  = new UnixUserInfo (UnixEnvironment.UserName).RealName;
+					if (string.IsNullOrEmpty(full_name)) full_name = "";
                     full_name.TrimEnd (",".ToCharArray());
 
 					Label name_label = new Label ("<b>" + _("Full Name:") + "</b>") {
