diff -urp a/src/drmmode_display.c b/src/drmmode_display.c
--- a/src/drmmode_display.c	2011-08-10 15:44:37.000000000 +0000
+++ b/src/drmmode_display.c	2011-09-29 06:52:38.000000000 +0000
@@ -1726,7 +1726,8 @@ Bool drmmode_pre_init(ScrnInfoPtr pScrn,
 	if (!drmmode->mode_res)
 		return FALSE;
 
-	xf86CrtcSetSizeRange(pScrn, 320, 200, drmmode->mode_res->max_width, drmmode->mode_res->max_height);
+	xf86CrtcSetSizeRange(pScrn, 160, 100, drmmode->mode_res->max_width, drmmode->mode_res->max_height);
+	xf86DrvMsg(pScrn->scrnIndex, X_INFO, "arcade-drm: range min set (160x100)\n");
 	for (i = 0; i < drmmode->mode_res->count_crtcs; i++)
 		if (!xf86IsEntityShared(pScrn->entityList[0]) || pScrn->confScreen->device->screen == i)
 			drmmode_crtc_init(pScrn, drmmode, i);
