diff -ur QGLViewer/VRender/TopologicalSortMethod.cpp QGLViewer.new/VRender/TopologicalSortMethod.cpp
--- QGLViewer/VRender/TopologicalSortMethod.cpp	2007-08-28 23:38:59.000000000 +0200
+++ QGLViewer.new/VRender/TopologicalSortMethod.cpp	2008-05-09 09:36:25.000000000 +0200
@@ -45,6 +45,7 @@
 
 *****************************************************************************/
 
+#include <limits.h>
 #include <assert.h>
 #include "VRender.h"
 #include "Primitive.h"
diff -ur QGLViewer/VRender/VRender.cpp QGLViewer.new/VRender/VRender.cpp
--- QGLViewer/VRender/VRender.cpp	2007-08-28 23:38:59.000000000 +0200
+++ QGLViewer.new/VRender/VRender.cpp	2008-05-09 09:36:25.000000000 +0200
@@ -55,7 +55,9 @@
 # include <GL/gl.h>
 #endif
 
+#include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 #include <vector>
 #include "VRender.h"
 #include "ParserGL.h"
diff -ur QGLViewer/manipulatedFrame.cpp QGLViewer.new/manipulatedFrame.cpp
--- QGLViewer/manipulatedFrame.cpp	2007-08-28 23:38:58.000000000 +0200
+++ QGLViewer.new/manipulatedFrame.cpp	2008-05-09 09:38:29.000000000 +0200
@@ -23,6 +23,7 @@
 
 *****************************************************************************/
 
+#include "stdlib.h"
 #include "domUtils.h"
 #include "manipulatedFrame.h"
 #include "qglviewer.h"
