diff -Naur emgd-dkms/emgd/drm/emgd_crtc.c emgd-1.10.0.2209.PATCHED/emgd/drm/emgd_crtc.c
--- emgd-dkms/emgd/drm/emgd_crtc.c	2011-11-15 11:19:42.000000000 +0000
+++ emgd-1.10.0.2209.PATCHED/emgd/drm/emgd_crtc.c	2012-05-13 18:33:48.763764480 +0100
@@ -345,11 +345,11 @@
 
 	plane_fb_info->width          = fb->width;
 	plane_fb_info->height         = fb->height;
-	plane_fb_info->screen_pitch   = fb->pitch;
+	plane_fb_info->screen_pitch   = fb->pitches[0];
 	plane_fb_info->flags          = 0;
 	plane_fb_info->allocated      = 1;
 	plane_fb_info->fb_base_offset = emgd_fb->gtt_offset;
-	plane_fb_info->visible_offset = (y * fb->pitch) +
+	plane_fb_info->visible_offset = (y * fb->pitches[0]) +
 		(x * (fb->bits_per_pixel / 8));
 
 
@@ -835,7 +835,7 @@
 		/* Rendering complete; program the plane registers */
 		igd_surface.flags        = IGD_SURFACE_DISPLAY;
 		igd_surface.offset       = crtc->newfb->gtt_offset;
-		igd_surface.pitch        = crtc->newfb->base.pitch;
+		igd_surface.pitch        = crtc->newfb->base.pitches[0];
 		igd_surface.width        = crtc->newfb->base.width;
 		igd_surface.height       = crtc->newfb->base.height;
 		igd_surface.pixel_format = IGD_PF_ARGB32;
diff -Naur emgd-dkms/emgd/drm/emgd_drv.c emgd-1.10.0.2209.PATCHED/emgd/drm/emgd_drv.c
--- emgd-dkms/emgd/drm/emgd_drv.c	2011-11-15 11:19:42.000000000 +0000
+++ emgd-1.10.0.2209.PATCHED/emgd/drm/emgd_drv.c	2012-05-13 18:41:14.929651662 +0100
@@ -38,6 +38,7 @@
 #include <drm/drm_crtc_helper.h>
 #include <linux/version.h>
 #include <linux/device.h>
+#include <linux/module.h>
 #include <drm/drm_pciids.h>
 #include <intelpci.h>
 #include "drm_emgd_private.h"
@@ -2277,6 +2278,18 @@
 static struct pci_driver emgd_pci_driver = EMGD_PCI_DRIVER;
 #endif
 
+
+static const struct file_operations emgd_drm_fops = {
+        .owner   = THIS_MODULE,
+        .open    = drm_open,
+        .release = drm_release,
+        .unlocked_ioctl   = drm_ioctl,
+        .mmap    = emgd_mmap,
+        .poll    = drm_poll,
+        .fasync  = drm_fasync,
+        .read    = drm_read,
+};
+
 /**
  * DRM Sub driver entry points
  */
@@ -2304,20 +2317,7 @@
 	.get_reg_ofs        = drm_core_get_reg_ofs,
 #endif
 	.ioctls             = emgd_ioctl,
-	.fops = {
-		.owner   = THIS_MODULE,
-		.open    = drm_open,
-		.release = drm_release,
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,32)
-		.unlocked_ioctl   = drm_ioctl,
-#else
-		.ioctl   = drm_ioctl,
-#endif
-		.mmap    = emgd_mmap,
-		.poll    = drm_poll,
-		.fasync  = drm_fasync,
-		.read    = drm_read,
-	},
+	.fops               = &emgd_drm_fops,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,38)
 	.pci_driver          = EMGD_PCI_DRIVER,
 #endif
diff -Naur emgd-dkms/emgd/drm/emgd_fbcon.c emgd-1.10.0.2209.PATCHED/emgd/drm/emgd_fbcon.c
--- emgd-dkms/emgd/drm/emgd_fbcon.c	2011-11-15 11:19:42.000000000 +0000
+++ emgd-1.10.0.2209.PATCHED/emgd/drm/emgd_fbcon.c	2012-05-13 18:33:05.778134873 +0100
@@ -36,7 +36,7 @@
 #include <drmP.h>
 #include <drm_crtc_helper.h>
 #include <linux/vga_switcheroo.h>
-
+#include <linux/module.h>
 
 #include "user_config.h"
 #include "drm_emgd_private.h"
@@ -124,7 +124,7 @@
 	info->fix.ywrapstep   = 0;
 	info->fix.accel       = FB_ACCEL_NONE;
 	info->fix.type_aux    = 0;
-	info->fix.line_length = fb->pitch;
+	info->fix.line_length = fb->pitches[0];
 }
 
 
