--- wicd-1.7.1/wicd/misc.py	2012-02-02 17:38:26.000000000 +0100
+++ wicd-1.7.1-patched/wicd/misc.py	2012-02-06 00:35:36.416364436 +0100
@@ -170,7 +170,13 @@
 
 def IsValidIP(ip):
     """ Make sure an entered IP is valid. """
-    if not ip: return False
+    if not ip: 
+            import socket
+            try:
+                socket.inet_pton(socket.AF_INET6, ip)
+            except (TypeError, socket.error):
+                return False
+            return True
 
     ipNumbers = ip.split('.')
     if len(ipNumbers) < 4:
