--- wicd-1.7.0/wicd/misc.py	2010-12-12 03:54:41.000000000 +0800
+++ wicd-1.7.0.patched/wicd/misc.py	2011-01-11 16:39:53.000000000 +0800
@@ -151,7 +151,12 @@ def IsValidIP(ip):
                 if not number.isdigit() or int(number) > 255:
                     return False
             return ipNumbers
-    return False
+    import socket
+    try:
+        socket.inet_pton(socket.AF_INET6, ip)
+    except (TypeError, socket.error):
+        return False
+    return True
 
 def PromptToStartDaemon():
     """ Prompt the user to start the daemon """
