diff -urN mumbles-0.4.old/bin/mumbles mumbles-0.4/bin/mumbles
--- mumbles-0.4.old/bin/mumbles	2007-08-27 05:30:20.000000000 +0700
+++ mumbles-0.4/bin/mumbles	2008-10-09 01:43:07.000000000 +0700
@@ -8,7 +8,7 @@
 #------------------------------------------------------------------------
 import os, sys
 
-sys.path.append("/usr/share/python-support/mumbles/src/")
+sys.path.append("/usr/lib/python2.5/site-packages/mumbles")
 os.environ.setdefault("MUMBLES_PATH", "/usr/share/mumbles/")
 
 from MumblesGlobals import *
diff -urN mumbles-0.4.old/bin/mumbles-send mumbles-0.4/bin/mumbles-send
--- mumbles-0.4.old/bin/mumbles-send	2007-08-29 10:17:51.000000000 +0700
+++ mumbles-0.4/bin/mumbles-send	2008-10-09 01:43:17.000000000 +0700
@@ -22,7 +22,7 @@
 from socket import AF_INET, SOCK_DGRAM, socket
 from getpass import getpass
 
-sys.path.append("/usr/share/python-support/mumbles/src/")
+sys.path.append("/usr/lib/python2.5/site-packages/mumbles")
 os.environ.setdefault("MUMBLES_PATH", "/usr/share/mumbles/")
 
 from MumblesGlobals import *
