--- ../synaptics-0.14.6-old/synaptics.c	2006-07-15 17:54:29.000000000 +0200
+++ synaptics.c	2006-12-09 20:46:36.000000000 +0100
@@ -1626,12 +1626,12 @@
     double_click = FALSE;
     if (!para->updown_button_scrolling) {
 	if (hw->down) {		/* map down button to middle button */
-	    hw->middle = TRUE;
+	    hw->middle = FALSE;
 	}
 
 	if (hw->up) {		/* up button generates double click */
 	    if (!priv->prev_up)
-		double_click = TRUE;
+		double_click = FALSE;
 	}
 	priv->prev_up = hw->up;
 
