diff -aur libkml-1.2.0.orig/configure libkml-1.2.0/configure
--- libkml-1.2.0.orig/configure
+++ libkml-1.2.0/configure
@@ -16934,11 +16934,15 @@
 
 	if test "x$SWIG" != x; then
 		SWIG_VERSION=`swig -version 2>&1|grep Version|cut -d" " -f3`
-		if test ${SWIG_VERSION:4} -lt 35; then
+		SWIG_VERSION_MAJOR=${SWIG_VERSION:0:1}
+		SWIG_VERSION_MINOR=${SWIG_VERSION:2:1}
+		SWIG_VERSION_LAST=${SWIG_VERSION:4}
+		if test ${SWIG_VERSION_MAJOR} -le 1 && test ${SWIG_VERSION_MINOR} -le 3 && ${SWIG_VERSION_LAST} -lt 35; then
 			{ $as_echo "$as_me:$LINENO: WARNING: You have SWIG $SWIG_VERSION installed, but libkml requires at least SWIG 1.3.35. The bindings will not be built." >&5
 $as_echo "$as_me: WARNING: You have SWIG $SWIG_VERSION installed, but libkml requires at least SWIG 1.3.35. The bindings will not be built." >&2;}
 			SWIG=
 		fi
+		unset SWIG_VERSION_MAJOR SWIG_VERSION_MINOR SWIG_VERSION_LAST
 	fi
 fi
  if test "x$SWIG" != x; then
