--- defaultlist.orig	2010-09-21 16:34:18.000000000 +0200
+++ defaultlist	2010-09-21 16:38:23.000000000 +0200
@@ -5,6 +5,7 @@ am
 am_ET
 ang
 ar
+ara
 ar_AE
 ar_BH
 ar_DZ
@@ -24,10 +25,13 @@ ar_SY
 ar_TN
 ar_YE
 as
+ast
 az
 az_AZ
 az_IR
 be
+be@latin
+be-tarask
 be_BY
 bg
 bg_BG
@@ -39,8 +43,10 @@ bs
 bs_BA
 byn
 ca
+ca@valencia
 ca_ES
 ca_ES@euro
+crh
 cs
 cs_CZ
 cy
@@ -59,6 +65,7 @@ de_DE@euro
 de_LU
 de_LU@euro
 dv
+dz
 el
 el_GR
 el_GR.UTF-8
@@ -67,6 +74,7 @@ en
 en@IPA
 en@boldquot
 en@quot
+en@shaw
 en_AU
 en_BW
 en_CA
@@ -162,6 +170,7 @@ hy_AM
 ia
 id
 id_ID
+io
 is
 is_IS
 it
@@ -182,6 +191,7 @@ ka_GE
 kk
 kl
 kl_GL
+km
 kn
 ko
 ko_KR
@@ -192,11 +202,15 @@ ku
 kw
 kw_GB
 ky
+lg
 li
+lo
 lt
 lt_LT
 lv
 lv_LV
+mai
+mg
 mi
 mi_NZ
 mk
@@ -210,7 +224,9 @@ ms
 ms_MY
 mt
 mt_MT
+my
 my_MM
+nds
 nb
 nb_NO
 ne
@@ -247,6 +263,7 @@ ru_RU.UTF-8
 ru_UA
 rw
 sa
+si
 sid
 sk
 sk_SK
@@ -259,6 +276,9 @@ sq_AL
 sr
 sr@Latn
 sr@ije
+sr@ijekavian
+sr@ijekavianlatin
+sr@latin
 sr_YU
 sr_YU@cyrillic
 sv
@@ -293,6 +313,7 @@ ur
 ur_PK
 uz
 uz@Latn
+uz@cyrillic
 uz_UZ
 ve
 vi
