--- getdics.pl	2006-10-30 02:17:36.000000000 +0900
+++ ../../../getdics-updurl.pl	2011-07-24 15:00:34.000000000 +0900
@@ -80,7 +80,7 @@
 
 my @dics = (
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/edict.gz",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/edict.gz",
 		"package"     => "edict.gz",
 		"format"      => "gzip",
 		"encoding"    => "EUC-JP",
@@ -90,7 +90,7 @@
 		"section"     => "basic",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/kanjidic.gz",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/kanjidic.gz",
 		"package"     => "kanjidic.gz",
 		"format"      => "gzip",
 		"encoding"    => "EUC-JP",
@@ -102,7 +102,7 @@
 
 # EXTRA
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/enamdict.gz",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/enamdict.gz",
 		"package"     => "enamdict.gz",
 		"format"      => "gzip",
 		"encoding"    => "EUC-JP",
@@ -112,7 +112,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/j_places.gz",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/j_places.gz",
 		"package"     => "j_places.gz",
 		"format"      => "gzip",
 		"encoding"    => "EUC-JP",
@@ -122,7 +122,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/lifscdic.Z",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/lifscdic.Z",
 		"package"     => "lifscdic.Z",
 		"format"      => "gzip",
 		"encoding"    => "EUC-JP",
@@ -132,7 +132,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/engscidic.gz",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/engscidic.gz",
 		"package"     => "engscidic.gz",
 		"format"      => "gzip",
 		"encoding"    => "EUC-JP",
@@ -142,7 +142,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/compdic.gz",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/compdic.gz",
 		"package"     => "compdic.gz",
 		"format"      => "gzip",
 		"encoding"    => "EUC-JP",
@@ -152,7 +152,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/geodic.gz",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/geodic.gz",
 		"package"     => "geodic.gz",
 		"format"      => "gzip",
 		"encoding"    => "EUC-JP",
@@ -162,7 +162,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/compverb.gz",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/compverb.gz",
 		"package"     => "compverb.gz",
 		"format"      => "gzip",
 		"encoding"    => "EUC-JP",
@@ -172,7 +172,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/riverwater.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/riverwater.zip",
 		"package"     => "riverwater.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -182,7 +182,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/4jword3_edict.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/4jword3_edict.zip",
 		"package"     => "4jword3_edict.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -192,7 +192,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/classical.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/classical.zip",
 		"package"     => "classical.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -202,7 +202,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/lingdic.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/lingdic.zip",
 		"package"     => "lingdic.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -212,7 +212,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/forsdic.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/forsdic.zip",
 		"package"     => "forsdic.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -222,7 +222,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/aviation.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/aviation.zip",
 		"package"     => "aviation.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -232,7 +232,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/concrete.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/concrete.zip",
 		"package"     => "concrete.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -242,7 +242,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/findic.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/findic.zip",
 		"package"     => "findic.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -252,7 +252,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/mktdic.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/mktdic.zip",
 		"package"     => "mktdic.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -262,7 +262,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/stardict.zip",
+		"url"         => "http://ftp.monash.edu.au/pub/nihongo/stardict.zip",
 		"package"     => "stardict.zip",
 		"format"      => "zip",
 		"encoding"    => "EUC-JP",
@@ -272,7 +272,7 @@
 		"section"     => "extra",
 	    },
 	    { 
-		"url"         => "http://ftp.cc.monash.edu.au/pub/nihongo/buddhdic.zip",
+		"url"         => "http://japanologie.arts.kuleuven.be/mirrors/monash/Buddhdic.zip",
 		"package"     => "buddhdic.zip",
 		"format"      => "zip",
 		"encoding"    => "UTF-8",
@@ -328,7 +328,7 @@
     next if (($BASICSETUP == 1) && ($dic->{section} eq "extra"));
 
 # DOWNLOAD    
-#wget -c http://ftp.cc.monash.edu.au/pub/nihongo/edict.gz -O dl/edict.gz
+#wget -c http://ftp.monash.edu.au/pub/nihongo/edict.gz -O dl/edict.gz
     if (! -f "$DLDIR/$dic->{package}") {
 	print "Downloading \"$dic->{name}\" from $dic->{url}\n";
 	my @wget = ("$WGET -q -c $dic->{url} -O $DLDIR/$dic->{package}");
