diff -urN separate+-0.5.9/Makefile separate+-0.5.9-new/Makefile
--- separate+-0.5.9/Makefile	2010-08-17 05:14:32.000000000 +0800
+++ separate+-0.5.9-new/Makefile	2015-01-10 18:29:37.246239120 +0800
@@ -21,6 +21,7 @@
 LCMS_LIB     = `pkg-config $(LCMS_PKG) --libs`
 TIFF_LIB     = -ltiff
 JPEG_LIB     = -ljpeg
+LIBM_LIB     = -lm
 
 GETTEXT_PACKAGE = gimp20-separate
 
@@ -37,7 +38,7 @@
 NLS     = -DENABLE_NLS -DGETTEXT_PACKAGE="\"$(GETTEXT_PACKAGE)\""
 CFLAGS += $(GIMP_INCLUDE) $(GTK_INCLUDE) $(LCMS_INCLUDE) $(CM) $(NLS) \
           -DUSE_ICC_BUTTON
-LIBS    = $(GIMP_LIB) $(GTK_LIB) $(LCMS_LIB) $(TIFF_LIB) $(JPEG_LIB)
+LIBS    = $(GIMP_LIB) $(GTK_LIB) $(LCMS_LIB) $(TIFF_LIB) $(JPEG_LIB) $(LIBM_LIB)
 
 SOURCES = $(SEPARATE_SOURCES) $(IMPORT_SOURCES) \
           $(ICC_COLORSPACE_SOURCES) $(EXTRA_SOURCES)
