diff -Naur base/SConstruct base-fixed/SConstruct
--- base/SConstruct	2013-04-07 08:30:59.130000911 -0800
+++ base-fixed/SConstruct	2013-04-09 18:45:46.385001898 -0800
@@ -218,7 +218,7 @@
 Requires.private: glib-2.0 @PRIVATE_PC@
 Libs: -lbase-${bslot}
 Libs.private: @BASE_LIBS@ @LIBS@
-Cflags: -I${includedir}/base-${bslot} -Wno-c++11-extensions
+Cflags: -I${includedir}/base-${bslot}
 """)])
 
 env.Substfile('libbase-%s.so' % BASE_VER,
