--- src/gagx.c.org	2012-08-08 11:24:53.238275267 +0200
+++ src/gagx.c	2012-08-08 11:25:14.104261086 +0200
@@ -3327,7 +3327,7 @@
       cmax2 = cmax2 + 2.0*cint2;
     }
   }
-  snprintf(pout,511,"%g %g %g %g \n",cmin1,cmax1,cmin2,cmax2);
+  snprintf(pout,255,"%g %g %g %g \n",cmin1,cmax1,cmin2,cmax2);
   gaprnt(2,pout);
   gxscal (pcm->xsiz1,pcm->xsiz2,pcm->ysiz1,pcm->ysiz2,
           cmin1,cmax1,cmin2,cmax2);
@@ -3903,7 +3903,7 @@
     /* call routine in gxcntr.c to write out contour line vertices and values */
     rc = gxshplin(sfid,dbfid,dbanch);
     if (rc>0) {
-      snprintf(pout,511,"%d contours written to shapefile %s\n",rc,fnroot);
+      snprintf(pout,255,"%d contours written to shapefile %s\n",rc,fnroot);
       gaprnt(2,pout);
     }
     else if (rc==-1) {
@@ -3928,7 +3928,7 @@
     /* call routine in gxshad2.c to write out polygon vertices and values */
     rc = s2shpwrt(sfid,dbfid,dbanch);
     if (rc>0) {
-      snprintf(pout,511,"%d polygons written to shapefile %s\n",rc,fnroot);
+      snprintf(pout,255,"%d polygons written to shapefile %s\n",rc,fnroot);
       gaprnt(2,pout);
     }
     else if (rc==-1) {
@@ -4140,9 +4140,9 @@
     rc = gxclvert(kmlfp);
     if (rc>0) {
       if (pcm->kmlname)
-	snprintf(pout,511,"%d contours written to KML file %s\n",rc,pcm->kmlname);
+	snprintf(pout,255,"%d contours written to KML file %s\n",rc,pcm->kmlname);
       else
-	snprintf(pout,511,"%d contours written to KML file grads.kml\n",rc);
+	snprintf(pout,255,"%d contours written to KML file grads.kml\n",rc);
       gaprnt(2,pout);
     }
     else err=1;
@@ -4186,9 +4186,9 @@
     rc = s2polyvert(kmlfp);
     if (rc>0) {
       if (pcm->kmlname)
-	snprintf(pout,511,"%d polygons written to KML file %s\n",rc,pcm->kmlname);
+	snprintf(pout,255,"%d polygons written to KML file %s\n",rc,pcm->kmlname);
       else
-	snprintf(pout,511,"%d polygons written to KML file grads.kml\n",rc);
+	snprintf(pout,255,"%d polygons written to KML file grads.kml\n",rc);
       gaprnt(2,pout);
     }
     else err=1;
