diff --git a/src/wave.c b/src/wave.c
index 2bc37a8..837ad60 100644
--- a/src/wave.c
+++ b/src/wave.c
@@ -607,7 +607,7 @@ load_waveform_sample(int wavenum, int num_cycles, double octaves, sample_t scale
 			fclose(rawfile);
 
 			/* normalize sample to [-1,1] */
-			pos_max = neg_max = 0;
+			offset = pos_max = neg_max = 0;
 			for (sample = 0; sample < input_len; sample++) {
 				if (in_buf[sample] > pos_max) {
 					pos_max = in_buf[sample];
@@ -615,9 +615,12 @@ load_waveform_sample(int wavenum, int num_cycles, double octaves, sample_t scale
 				if (in_buf[sample] < neg_max) {
 					neg_max = in_buf[sample];
 				}
+				offset -= in_buf[sample];
 			}
-			offset = (pos_max + neg_max) / -2.0;
-			scalar = 1.0 / (float)((pos_max - neg_max) / 2.0);
+			offset /= (float)input_len;
+			if ((scalar = pos_max + offset) < -(offset + neg_max))
+			  scalar = -(offset + neg_max);
+			scalar = 1.0/scalar;
 			for (sample = 0; sample < input_len; sample++) {
 				in_buf[sample] = (in_buf[sample] + offset) * scalar;
 			}

