--- setup.py    2008-03-31 20:15:20.000000000 +0100
+++ setup.py.new    2009-09-14 23:08:33.183411774 +0100
@@ -132,7 +132,6 @@
        description      = DESC,
        long_description = LONG_DESC,
        license          = 'GPL2',
-       url              = 'http://code.google.com/p/morphin/',
        download_url     = 'http://code.google.com/p/morphin/downloads/list',
        packages         = PACKAGES,
        package_dir      = {'morphin' : '.'},