diff -Naur linux-3.2.orig/drivers/i2c/busses/i2c-isch.c linux-3.2.i2c/drivers/i2c/busses/i2c-isch.c
--- linux-3.2.orig/drivers/i2c/busses/i2c-isch.c	2012-02-13 15:00:48.829411263 +0000
+++ linux-3.2.i2c/drivers/i2c/busses/i2c-isch.c	2012-02-16 17:31:36.736477851 +0000
@@ -258,7 +258,7 @@
 	if (!res)
 		return -EBUSY;
 
-	if (!request_region(res->start, resource_size(res), dev->name)) {
+	if (!request_region(res->start, 8, dev->name)) {
 		dev_err(&dev->dev, "SMBus region 0x%x already in use!\n",
 			sch_smba);
 		return -EBUSY;
@@ -290,7 +290,7 @@
 	if (sch_smba) {
 		i2c_del_adapter(&sch_adapter);
 		res = platform_get_resource(pdev, IORESOURCE_IO, 0);
-		release_region(res->start, resource_size(res));
+		release_region(res->start, 8);
 		sch_smba = 0;
 	}
 
