diff -Naur1 power_manager/common.mk power_manager-stack/common.mk
--- power_manager/common.mk	2013-10-18 06:22:47.710000934 +0000
+++ power_manager-stack/common.mk	2013-10-18 06:23:48.125000963 +0000
@@ -278,5 +278,4 @@
 # The same goes for CFLAGS.
-COMMON_CFLAGS-gcc := -fstack-protector-strong -fvisibility=internal -ggdb3 \
-  -Wa,--noexecstack
-COMMON_CFLAGS-clang := -fstack-protector-all -fvisibility=hidden -ggdb
+COMMON_CFLAGS-gcc := -fvisibility=internal -ggdb3 -Wa,--noexecstack
+COMMON_CFLAGS-clang := -fvisibility=hidden -ggdb
 COMMON_CFLAGS := -Wall -Werror -fno-strict-aliasing -O1 -Wformat=2
