diff -Naur1 power_manager/common/util_dbus.cc power_manager-utmp/common/util_dbus.cc
--- power_manager/common/util_dbus.cc	2013-10-18 06:19:25.780000837 +0000
+++ power_manager-utmp/common/util_dbus.cc	2013-10-18 06:21:11.800000940 +0000
@@ -10,2 +10,4 @@
 
+#include <utmp.h>
+
 #include <climits>
@@ -24,24 +26,7 @@
   DCHECK(state);
-  chromeos::dbus::Proxy proxy(chromeos::dbus::GetSystemBusConnection(),
-                              login_manager::kSessionManagerServiceName,
-                              login_manager::kSessionManagerServicePath,
-                              login_manager::kSessionManagerInterface);
-
-  GError* error = NULL;
-  gchar* state_arg = NULL;
-  if (dbus_g_proxy_call(proxy.gproxy(),
-                        login_manager::kSessionManagerRetrieveSessionState,
-                        &error,
-                        G_TYPE_INVALID,
-                        G_TYPE_STRING,
-                        &state_arg,
-                        G_TYPE_INVALID)) {
-    *state = state_arg;
-    g_free(state_arg);
-    return true;
-  }
-  LOG(ERROR) << "Unable to retrieve session state from session manager: "
-             << error->message;
-  g_error_free(error);
-  return false;
+
+  setutent();
+  *state = (getutent() == NULL)?"stopped":"started";
+
+  return true;
 }
