Subject: Fix compilation on 32bit systems
Origin: http://sourceforge.net/p/fakerootng/code/347/
Forwarded: upstream-origin
--- a/arch/platform.h
+++ b/arch/platform.h
@@ -1,6 +1,8 @@
 #ifndef PLATFORM_H
 #define PLATFORM_H
 
+#include <sys/types.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
--- a/daemon.cpp
+++ b/daemon.cpp
@@ -83,7 +83,8 @@
         }
 
         if( static_cast<unsigned>(num_read)<sizeof(T) ) {
-            dlog("Session %d produced short read (expected %lu, got %ld)", fd, sizeof(T), num_read);
+            dlog("Session %d produced short read (expected %lu, got %ld)", fd, (long unsigned int) sizeof(T),
+                    (long int) num_read);
             throw daemonCtrl::short_msg_exception();
         }
 
@@ -115,7 +116,8 @@
             throw errno_exception("Send failed");
 
         if( static_cast<unsigned>(num_written)<sizeof(T) ) {
-            dlog("Session %d produced short send (expected %lu, got %ld)", fd, sizeof(T), num_written);
+            dlog("Session %d produced short send (expected %lu, got %ld)", fd, (long unsigned int) sizeof(T),
+                    (long int) num_written);
             throw daemonCtrl::short_msg_exception();
         }
     }
