diff -urN libz80/Makefile libz80.fixed/Makefile
--- libz80/Makefile	2002-04-21 18:30:00.000000000 -0500
+++ libz80.fixed/Makefile	2009-11-04 10:47:16.418113709 -0600
@@ -1,20 +1,22 @@
-OBJS = z80.c
-FLAGS = -fPIC -Wall -strict -ANSI -g
+OBJS = z80.o
+FLAGS = -fPIC -Wall -ansi -g
 
-force: clean all
+all: codegen libz80.so
 
-all: libz80.so
+codegen: clean
+	cd codegen && make opcodes
 
 libz80.so: z80.h $(OBJS)
-	cd codegen && make opcodes
 	gcc -shared -o libz80.so $(OBJS)
-	
+
 .c.o:
 	gcc $(FLAGS) -c $<
 
 install:
-	install -m 666 libz80.so /usr/lib
-	install -m 666 z80.h /usr/include
+	install -d ${DESTDIR}/usr/lib
+	install -d ${DESTDIR}/usr/include
+	install -m 666 libz80.so ${DESTDIR}/usr/lib
+	install -m 666 z80.h ${DESTDIR}/usr/include
 
 clean:
 	rm -f *.o *.so core
diff -urN libz80/z80.c libz80.fixed/z80.c
--- libz80/z80.c	2002-04-22 11:01:02.000000000 -0500
+++ libz80.fixed/z80.c	2009-11-04 10:40:16.984523276 -0600
@@ -236,7 +236,7 @@
 }
 
 
-// Adjust flags after AND, OR, XOR
+/* Adjust flags after AND, OR, XOR */
 static void adjustLogicFlag (Z80Context *ctx, int flagH)
 {
     VALFLAG(F_S, (BR.A & 0x80) != 0);
