--- a/translate/gcc/Makefile.in
+++ b/translate/gcc/Makefile.in
@@ -175,13 +175,13 @@
 	echo "   Install_Prefix : constant String :=" >> tmp-dpathes.ads
 	echo "     \"$(exec_prefix)\";" >> tmp-dpathes.ads
 	echo "   Compiler_Gcc   : constant String :=" >> tmp-dpathes.ads
-	echo "     \"libexec/gcc/$(target_noncanonical)/$(version)/ghdl1$(exeext)\";" >> tmp-dpathes.ads
+	echo "     \"/usr/lib/gcc/$(target_noncanonical)/$(version)/ghdl1$(exeext)\";" >> tmp-dpathes.ads
 	echo "   Compiler_Debug : constant String := \"\";" >> tmp-dpathes.ads
 	echo "   Compiler_Mcode : constant String := \"\";" >> tmp-dpathes.ads
 	echo "   Compiler_Llvm  : constant String := \"\";" >> tmp-dpathes.ads
 	echo "   Post_Processor : constant String := \"\";" >> tmp-dpathes.ads
 	echo "   Lib_Prefix     : constant String :=">> tmp-dpathes.ads
-	echo "     \"lib/gcc/$(target_noncanonical)/$(version)/vhdl/lib/\";" >> tmp-dpathes.ads
+	echo "     \"/usr/lib/gcc/$(target_noncanonical)/$(version)/vhdl/lib/\";" >> tmp-dpathes.ads
 	echo "end Default_Pathes;" >> tmp-dpathes.ads
 	$(srcdir)/../../move-if-change tmp-dpathes.ads $@

