--- src/Makefile.orig	2003-05-07 17:04:27 +0400
+++ src/Makefile	2004-07-02 22:16:58 +0400
@@ -8,13 +8,13 @@
 
 # Configuration options
 # Uncomment for debugging output
-DEBUG += -g
+#DEBUG += -g
 
 # For profiling
 #DEBUG += -pg -a -fprofile-arcs
 
 # Disable for debugging
-#OPT = -O2
+OPT = -O2
 
 CFLAGS += -Wall -Wstrict-prototypes $(OPT) $(DEBUG) $(KERNELDIR)
 OBJECTS = main.o daemon.o process.o utils.o config.o capture.c
