diff -urN filebench-1.4.4/bin/filebench filebench-1.4.4.port/bin/filebench
--- filebench-1.4.4/bin/filebench	2008-10-31 14:38:00.000000000 -0400
+++ filebench-1.4.4.port/bin/filebench	2009-04-02 17:28:48.000000000 -0400
@@ -31,8 +31,8 @@
 my $QUIT = 0;
 my $USE_XANADU = 0;
 my $TIMEOUT = 60;
-my $FILEBENCH = "/opt/filebench";
-my $PROG = "/opt/filebench/bin/go_filebench";
+my $FILEBENCH = "/usr/share/filebench";
+my $PROG = "/usr/bin/go_filebench";
 my $FSCRIPT;
 my $SCRIPT_NO;
 my @CONFLIST;
diff -urN filebench-1.4.4/configure.in filebench-1.4.4.port/configure.in
--- filebench-1.4.4/configure.in	2009-03-17 10:13:16.000000000 -0400
+++ filebench-1.4.4.port/configure.in	2009-04-02 17:22:50.000000000 -0400
@@ -80,7 +80,7 @@
 	##DEFINES="-DLINUX_PORT -D_XOPEN_SOURCE=500 -D_LARGEFILE_SOURCE64 -D_FILE_OFFSET_BITS=64 -D__USE_LARGEFILE64 -D__USE_UNIX98"
 	DEFINES="-DLINUX_PORT -D_GNU_SOURCE -D__USE_LARGEFILE64"
 	CFLAGS="${CFLAGS=} -O -I/usr/include/nptl -DYYDEBUG=1"
-	LDLIBS="-lm  -lm -lpthread -laio -lfl -lrt -ltecla -lsocket -lnsl"
+	LDLIBS="-lm  -lm -lpthread -laio -lfl -lrt -ltecla -lnsl"
 	LFLAGS="-t"
 	YFLAGS="-d"
         AC_DEFINE(HAVE_OFF64_T, 1,
diff -urN filebench-1.4.4/filebench/fb_avl.h filebench-1.4.4.port/filebench/fb_avl.h
--- filebench-1.4.4/filebench/fb_avl.h	2009-03-16 15:53:31.000000000 -0400
+++ filebench-1.4.4.port/filebench/fb_avl.h	2009-04-02 17:24:28.000000000 -0400
@@ -155,6 +155,8 @@
 #define	AVL_INDEX2CHILD(x)	((x) & 1)
 #define	AVL_MKINDEX(n, c)	((avl_index_t)(n) | (c))
 
+#define ulong_t unsigned long
+#define uint_t unsigned int
 
 /*
  * The tree structure. The fields avl_root, avl_compar, and avl_offset come
diff -urN filebench-1.4.4/filebench/multi_client_sync.c filebench-1.4.4.port/filebench/multi_client_sync.c
--- filebench-1.4.4/filebench/multi_client_sync.c	2009-03-16 15:53:21.000000000 -0400
+++ filebench-1.4.4.port/filebench/multi_client_sync.c	2009-04-02 17:24:41.000000000 -0400
@@ -65,7 +65,7 @@
 		return (FILEBENCH_ERROR);
 	}
 
-	if (gethostbyname_r(master_name, &master_info, buffer, MCS_MSGLENGTH,
+	if (gethostbyname_r(master_name, &master_info, buffer, MCS_MSGLENGTH,NULL,
 	    &error_num) == NULL) {
 		filebench_log(LOG_ERROR, "could not locate sync master");
 		return (FILEBENCH_ERROR);
diff -urN filebench-1.4.4/filebench/multi_client_sync.h filebench-1.4.4.port/filebench/multi_client_sync.h
--- filebench-1.4.4/filebench/multi_client_sync.h	2009-03-16 15:53:32.000000000 -0400
+++ filebench-1.4.4.port/filebench/multi_client_sync.h	2009-04-02 17:24:41.000000000 -0400
@@ -28,7 +28,7 @@
 
 #include <sys/types.h>
 #include <sys/socket.h>
-#include <inet/ip.h>
+#include <netinet/ip.h>
 
 int mc_sync_open_sock(char *master_name, int master_port, char *client_name);
 int mc_sync_synchronize(int synch_point);
