diff -ur oblige_357_src.orig/gui/hdr_lua.h oblige_357_src/gui/hdr_lua.h
--- oblige_357_src.orig/gui/hdr_lua.h	2011-11-05 18:12:42.000000000 +0100
+++ oblige_357_src/gui/hdr_lua.h	2011-11-05 18:12:57.000000000 +0100
@@ -23,9 +23,11 @@
 
 /* LUA Scripting Language */
 
+extern "C" {
 #include "lua.h"
 #include "lauxlib.h"
 #include "lualib.h"
+}
 
 #endif // __OB_HDR_LUA__
 
diff -ur oblige_357_src.orig/gui/q_bsp.cc oblige_357_src/gui/q_bsp.cc
--- oblige_357_src.orig/gui/q_bsp.cc	2011-11-05 18:12:42.000000000 +0100
+++ oblige_357_src/gui/q_bsp.cc	2011-11-05 18:12:57.000000000 +0100
@@ -107,7 +107,7 @@
   // convert each newline into CR/LF pair
   while (*str)
   {
-    char *next = strchr(str, '\n');
+    const char *next = strchr(str, '\n');
 
     Append(str, next ? (next - str) : strlen(str));
 
diff -ur oblige_357_src.orig/gui/ui_window.cc oblige_357_src/gui/ui_window.cc
--- oblige_357_src.orig/gui/ui_window.cc	2011-11-05 18:12:42.000000000 +0100
+++ oblige_357_src/gui/ui_window.cc	2011-11-05 18:20:44.000000000 +0100
@@ -26,9 +26,9 @@
 #include <unistd.h>
 #endif
 
-#if (FL_MAJOR_VERSION != 1 ||  \
-     FL_MINOR_VERSION != 1 ||  \
-     FL_PATCH_VERSION < 7)
+#if (FL_MAJOR_VERSION < 1 ||  \
+     (FL_MAJOR_VERSION == 1 && FL_MINOR_VERSION < 1) ||  \
+     (FL_MAJOR_VERSION == 1 && FL_MINOR_VERSION == 1 && FL_PATCH_VERSION < 7))
 #error "Require FLTK version 1.1.7 or later"
 #endif
 
