--- linux/drivers/media/dvb-core/dvbdev.c
+++ linux/drivers/media/dvb-core/dvbdev.c
@@ -68,7 +68,7 @@ static int dvb_device_open(struct inode *inode, struct file *file)
 {
 	struct dvb_device *dvbdev;
 
-	mutex_lock(&dvbdev_mutex);
+	//mutex_lock(&dvbdev_mutex);
 	down_read(&minor_rwsem);
 	dvbdev = dvb_minors[iminor(inode)];
 
@@ -84,12 +84,12 @@ static int dvb_device_open(struct inode *inode, struct file *file)
 		if (file->f_op->open)
 			err = file->f_op->open(inode,file);
 		up_read(&minor_rwsem);
-		mutex_unlock(&dvbdev_mutex);
+		//mutex_unlock(&dvbdev_mutex);
 		return err;
 	}
 fail:
 	up_read(&minor_rwsem);
-	mutex_unlock(&dvbdev_mutex);
+	//mutex_unlock(&dvbdev_mutex);
 	return -ENODEV;
 }
 
@@ -431,6 +431,7 @@ int dvb_usercopy(struct file *file,
 	kfree(mbuf);
 	return err;
 }
+EXPORT_SYMBOL(dvb_usercopy);
 
 static int dvb_uevent(struct device *dev, struct kobj_uevent_env *env)
 {
