diff -urN 2.0/.bzr/branch/branch.conf 2.0b/.bzr/branch/branch.conf
--- 2.0/.bzr/branch/branch.conf	2011-01-05 16:40:32.813333366 -0700
+++ 2.0b/.bzr/branch/branch.conf	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-parent_location = http://bazaar.launchpad.net/~pbxt-core/pbxt/2.0/
diff -urN 2.0/.bzr/branch/format 2.0b/.bzr/branch/format
--- 2.0/.bzr/branch/format	2011-01-05 16:40:32.416666699 -0700
+++ 2.0b/.bzr/branch/format	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-Bazaar Branch Format 6 (bzr 0.15)
diff -urN 2.0/.bzr/branch/last-revision 2.0b/.bzr/branch/last-revision
--- 2.0/.bzr/branch/last-revision	2011-01-05 16:40:32.426666699 -0700
+++ 2.0b/.bzr/branch/last-revision	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-926 paul.mccullagh@primebase.org-20101210134955-a3t0v0ddx2fi8vy5
diff -urN 2.0/.bzr/branch-format 2.0b/.bzr/branch-format
--- 2.0/.bzr/branch-format	2011-01-05 16:38:48.970000031 -0700
+++ 2.0b/.bzr/branch-format	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-Bazaar-NG meta directory, format 1
diff -urN 2.0/.bzr/checkout/conflicts 2.0b/.bzr/checkout/conflicts
--- 2.0/.bzr/checkout/conflicts	2011-01-05 16:40:38.143333365 -0700
+++ 2.0b/.bzr/checkout/conflicts	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-BZR conflict list format 1
Binary files 2.0/.bzr/checkout/dirstate and 2.0b/.bzr/checkout/dirstate differ
diff -urN 2.0/.bzr/checkout/format 2.0b/.bzr/checkout/format
--- 2.0/.bzr/checkout/format	2011-01-05 16:40:32.820000032 -0700
+++ 2.0b/.bzr/checkout/format	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-Bazaar Working Tree Format 4 (bzr 0.15)
diff -urN 2.0/.bzr/README 2.0b/.bzr/README
--- 2.0/.bzr/README	2011-01-05 16:38:48.970000031 -0700
+++ 2.0b/.bzr/README	1969-12-31 17:00:00.000000000 -0700
@@ -1,3 +0,0 @@
-This is a Bazaar control directory.
-Do not change any files in this directory.
-See http://bazaar-vcs.org/ for more information about Bazaar.
diff -urN 2.0/.bzr/repository/format 2.0b/.bzr/repository/format
--- 2.0/.bzr/repository/format	2011-01-05 16:38:49.003333365 -0700
+++ 2.0b/.bzr/repository/format	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-Bazaar pack repository format 1 (needs bzr 0.92)
Binary files 2.0/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.iix and 2.0b/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.iix differ
Binary files 2.0/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.rix and 2.0b/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.rix differ
diff -urN 2.0/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.six 2.0b/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.six
--- 2.0/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.six	2011-01-05 16:40:32.403333365 -0700
+++ 2.0b/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.six	1969-12-31 17:00:00.000000000 -0700
@@ -1,5 +0,0 @@
-Bazaar Graph Index 1
-node_ref_lists=0
-key_elements=1
-len=0
-
Binary files 2.0/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.tix and 2.0b/.bzr/repository/indices/fb2c3778292e6860570041cf9ad22eb8.tix differ
Binary files 2.0/.bzr/repository/pack-names and 2.0b/.bzr/repository/pack-names differ
Binary files 2.0/.bzr/repository/packs/fb2c3778292e6860570041cf9ad22eb8.pack and 2.0b/.bzr/repository/packs/fb2c3778292e6860570041cf9ad22eb8.pack differ
diff -urN 2.0/src/ha_pbxt.cc 2.0b/src/ha_pbxt.cc
--- 2.0/src/ha_pbxt.cc	2011-01-05 16:40:36.407820000 -0700
+++ 2.0b/src/ha_pbxt.cc	2011-01-11 00:50:08.646666893 -0700
@@ -2025,7 +2025,7 @@
 			xt_ha_open_database_of_table(self, (XTPathStrPtr) NULL);
 		}
 
-		err = myxt_statistics_fill_table(self, thd, tables, cond, system_charset_info);
+		err = myxt_statistics_fill_table(self, thd, tables, cond, (void*)system_charset_info);
 	}
 	catch_(a) {
 		err = xt_ha_pbxt_thread_error_for_mysql(thd, self, FALSE);
@@ -5855,7 +5855,7 @@
 	MYSQL_PLUGIN_VAR_HEADER;
 };
 
-static void pbxt_record_cache_size_func(THD *XT_UNUSED(thd), struct st_mysql_sys_var *var, void *tgt, const void *save)
+static void pbxt_record_cache_size_func(THD *XT_UNUSED(thd), struct st_mysql_sys_var *var, void *tgt,const  void *save)
 {
 	xtInt8	record_cache_size;
 
diff -urN 2.0/src/myxt_xt.cc 2.0b/src/myxt_xt.cc
--- 2.0/src/myxt_xt.cc	2011-01-05 16:40:36.407820000 -0700
+++ 2.0b/src/myxt_xt.cc	2011-01-11 00:47:49.106666893 -0700
@@ -2300,7 +2300,7 @@
 		seg->length = key_part->length;
 		seg->bit_start = seg->bit_end = 0;
 		seg->bit_length = seg->bit_pos = 0;
-		seg->charset = field->charset();
+		seg->charset = (struct charset_info_st*)field->charset();
 
 		if (field->null_ptr) {
 			key_length++;
