--- contrib/python/cjdnslog.orig	2012-11-02 03:32:18.294677190 -0400
+++ contrib/python/cjdnslog	2012-11-02 03:32:28.191514196 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # You may redistribute this program and/or modify it under the terms of
 # the GNU General Public License as published by the Free Software Foundation,
 # either version 3 of the License, or (at your option) any later version.
