diff -Naur original/data/Makefile.am modified/data/Makefile.am
--- original/data/Makefile.am	2014-08-12 14:00:21.000000000 +0100
+++ modified/data/Makefile.am	2014-08-26 17:44:41.717021515 +0100
@@ -32,7 +32,7 @@
 
 sessiondir = $(datadir)/gnome-session/sessions
 session_in_files = gnome-flashback.session.desktop.in gnome-flashback-compiz.session.desktop.in
-session_DATA = gnome-flashback.session
+session_DATA = $(session_in_files:.session.desktop.in=.session)
 
 xsessiondir = $(datadir)/xsessions
 xsession_in_in_files = gnome-fallback.desktop.in.in gnome-flashback-compiz.desktop.in.in
