diff --git a/Rakefile b/Rakefile
index 8ba64b4..1b94d1d 100644
--- a/Rakefile
+++ b/Rakefile
@@ -1,7 +1,7 @@
 task :default => [:check_erl_version, :check_previous_install, :build, :test]
 task :build => [:scanner, :parser, :reia, :ebin, :clean]
 
-PREFIX="/usr/local"
+PREFIX="../../pkg/usr"
 
 def erlang_version
   version = `erl -version 2>&1`.strip.match(/\d\.\d\.\d$/)
@@ -14,7 +14,7 @@ def erlang_version
 end
 
 def erl_lib_dir
-  `erl -noshell -eval "io:format(code:lib_dir())" -s init stop`
+  "../../pkg/usr/lib/erlang/lib"
 end
 
 def reia_install_dir
@@ -150,4 +150,4 @@ task :distclean => :clean do
 end
 
 task :ci => %w[distclean test]
-task :cruise => :ci
\ No newline at end of file
+task :cruise => :ci
