diff --git a/libparted/Makefile.am b/libparted/Makefile.am
index 4bc1939..85b70ae 100644
--- a/libparted/Makefile.am
+++ b/libparted/Makefile.am
@@ -18,7 +18,7 @@ AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
 SUBDIRS       = labels fs . $(SUBDIRS_CHECK)
 
 
-partedincludedir      = -I$(top_srcdir)/lib -I$(top_srcdir)/include
+partedincludedir      = -I$(top_srcdir)/lib -I$(top_builddir)/lib -I$(top_srcdir)/include -I$(top_builddir)/include
 lib_LTLIBRARIES	      =	libparted.la
 
 # Set the shared library version, per Libtool's guidelines.
diff --git a/libparted/fs/Makefile.am b/libparted/fs/Makefile.am
index cb09ffb..e728c3b 100644
--- a/libparted/fs/Makefile.am
+++ b/libparted/fs/Makefile.am
@@ -6,7 +6,7 @@
 SUBDIRS       = amiga ext2 ufs fat ntfs hfs linux_swap xfs jfs reiserfs \
 		nilfs2 # bfs
 
-partedincludedir      =	-I$(top_srcdir)/include
+partedincludedir      =	-I$(top_srcdir)/include -I$(top_builddir)/include
 noinst_LTLIBRARIES    =	libfs.la
 libfs_la_LDFLAGS  = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 			-release $(LT_RELEASE)
diff --git a/libparted/fs/amiga/Makefile.am b/libparted/fs/amiga/Makefile.am
index 788ecac..a9ec421 100644
--- a/libparted/fs/amiga/Makefile.am
+++ b/libparted/fs/amiga/Makefile.am
@@ -1,5 +1,5 @@
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-partedincludedir      =	-I$(top_srcdir)/include
+partedincludedir      =	-I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES    =	libamigafs.la
 libamigafs_la_SOURCES   =	amiga.h		\
diff --git a/libparted/fs/ext2/Makefile.am b/libparted/fs/ext2/Makefile.am
index 0acd080..15eac2a 100644
--- a/libparted/fs/ext2/Makefile.am
+++ b/libparted/fs/ext2/Makefile.am
@@ -1,5 +1,5 @@
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-partedincludedir      =	-I$(top_srcdir)/include
+partedincludedir      =	-I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES    =	libext2.la
 libext2_la_SOURCES    =	ext2.h \
diff --git a/libparted/fs/fat/Makefile.am b/libparted/fs/fat/Makefile.am
index 9c975d7..41fb8de 100644
--- a/libparted/fs/fat/Makefile.am
+++ b/libparted/fs/fat/Makefile.am
@@ -1,4 +1,4 @@
-partedincludedir      =	-I$(top_srcdir)/include
+partedincludedir      =	-I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES    =	libfat.la
 libfat_la_SOURCES     = bootsector.c	\
diff --git a/libparted/fs/hfs/Makefile.am b/libparted/fs/hfs/Makefile.am
index f85493c..5691912 100644
--- a/libparted/fs/hfs/Makefile.am
+++ b/libparted/fs/hfs/Makefile.am
@@ -1,5 +1,5 @@
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-partedincludedir	= -I$(top_srcdir)/include
+partedincludedir	= -I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES	= libhfs.la
 libhfs_la_SOURCES	= hfs.c hfs.h \
diff --git a/libparted/fs/jfs/Makefile.am b/libparted/fs/jfs/Makefile.am
index a477874..f55b60d 100644
--- a/libparted/fs/jfs/Makefile.am
+++ b/libparted/fs/jfs/Makefile.am
@@ -1,5 +1,5 @@
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-partedincludedir	= -I$(top_srcdir)/include
+partedincludedir	= -I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES	= libjfs.la
 libjfs_la_SOURCES	= jfs.c jfs_superblock.h jfs_types.h
diff --git a/libparted/fs/linux_swap/Makefile.am b/libparted/fs/linux_swap/Makefile.am
index a366fc6..4a9260a 100644
--- a/libparted/fs/linux_swap/Makefile.am
+++ b/libparted/fs/linux_swap/Makefile.am
@@ -1,4 +1,4 @@
-partedincludedir	= -I$(top_srcdir)/include
+partedincludedir	= -I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES	= liblinuxswap.la
 liblinuxswap_la_SOURCES	= linux_swap.c
diff --git a/libparted/fs/nilfs2/Makefile.am b/libparted/fs/nilfs2/Makefile.am
index c7cfd68..4c2ed84 100644
--- a/libparted/fs/nilfs2/Makefile.am
+++ b/libparted/fs/nilfs2/Makefile.am
@@ -1,4 +1,4 @@
-partedincludedir	= -I$(top_srcdir)/include
+partedincludedir	= -I$(top_srcdir)/include -I$(top_builddir)/include
 
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
 
diff --git a/libparted/fs/ntfs/Makefile.am b/libparted/fs/ntfs/Makefile.am
index 732692a..d9a370c 100644
--- a/libparted/fs/ntfs/Makefile.am
+++ b/libparted/fs/ntfs/Makefile.am
@@ -1,5 +1,5 @@
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-partedincludedir	= -I$(top_srcdir)/include
+partedincludedir	= -I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES	= libntfs.la
 libntfs_la_SOURCES	= ntfs.c
diff --git a/libparted/fs/reiserfs/Makefile.am b/libparted/fs/reiserfs/Makefile.am
index 7df005a..16cebe0 100644
--- a/libparted/fs/reiserfs/Makefile.am
+++ b/libparted/fs/reiserfs/Makefile.am
@@ -1,5 +1,5 @@
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-partedincludedir	= -I$(top_srcdir)/include
+partedincludedir	= -I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES	= libreiserfs.la
 
diff --git a/libparted/fs/ufs/Makefile.am b/libparted/fs/ufs/Makefile.am
index c63e8d2..895fa55 100644
--- a/libparted/fs/ufs/Makefile.am
+++ b/libparted/fs/ufs/Makefile.am
@@ -1,5 +1,5 @@
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-partedincludedir	= -I$(top_srcdir)/include
+partedincludedir	= -I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES	= libufs.la
 libufs_la_SOURCES	= ufs.c
diff --git a/libparted/fs/xfs/Makefile.am b/libparted/fs/xfs/Makefile.am
index 8e55052..7c7ed7f 100644
--- a/libparted/fs/xfs/Makefile.am
+++ b/libparted/fs/xfs/Makefile.am
@@ -1,5 +1,5 @@
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-partedincludedir	= -I$(top_srcdir)/include
+partedincludedir	= -I$(top_srcdir)/include -I$(top_builddir)/include
 
 noinst_LTLIBRARIES	= libxfs.la
 libxfs_la_SOURCES	= xfs.c xfs_sb.h xfs_types.h platform_defs.h
diff --git a/libparted/labels/Makefile.am b/libparted/labels/Makefile.am
index 967e537..62877b7 100644
--- a/libparted/labels/Makefile.am
+++ b/libparted/labels/Makefile.am
@@ -12,7 +12,7 @@ endif
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
 
 partedincludedir = \
-  -I$(top_srcdir)/lib -I$(top_srcdir)/include -I$(top_srcdir)/libparted
+  -I$(top_srcdir)/lib -I$(top_builddir)/lib -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir)/libparted
 noinst_LTLIBRARIES    =	liblabels.la
 
 liblabels_la_SOURCES = \
diff --git a/libparted/tests/Makefile.am b/libparted/tests/Makefile.am
index de8d03e..6453c07 100644
--- a/libparted/tests/Makefile.am
+++ b/libparted/tests/Makefile.am
@@ -14,8 +14,8 @@ LDADD = \
 
 AM_CPPFLAGS = \
   $(CHECK_CFLAGS) \
-  -I$(top_srcdir)/lib \
-  -I$(top_srcdir)/include
+  -I$(top_srcdir)/lib -I$(top_builddir)/lib \
+  -I$(top_srcdir)/include -I$(top_builddir)/include
 
 label_SOURCES = common.h common.c label.c
 disk_SOURCES  = common.h common.c disk.c
diff --git a/parted/Makefile.am b/parted/Makefile.am
index 7b093c6..55be5ae 100644
--- a/parted/Makefile.am
+++ b/parted/Makefile.am
@@ -3,7 +3,7 @@ BUILT_SOURCES =
 sbin_PROGRAMS = parted
 
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
-partedincludedir      = -I$(top_srcdir)/lib -I$(top_srcdir)/include
+partedincludedir      = -I$(top_srcdir)/lib -I$(top_builddir)/lib -I$(top_srcdir)/include -I$(top_builddir)/include
 
 parted_SOURCES = command.c	\
 		 command.h	\
diff --git a/partprobe/Makefile.am b/partprobe/Makefile.am
index f140991..90d8856 100644
--- a/partprobe/Makefile.am
+++ b/partprobe/Makefile.am
@@ -1,4 +1,4 @@
-partedincludedir      = -I$(top_srcdir)/lib -I$(top_srcdir)/include
+partedincludedir      = -I$(top_srcdir)/lib -I$(top_builddir)/lib -I$(top_srcdir)/include -I$(top_builddir)/include
 
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
 sbin_PROGRAMS = partprobe
diff --git a/tests/Makefile.am b/tests/Makefile.am
index f442204..4544bda 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -73,8 +73,8 @@ check_PROGRAMS = print-align print-max dup-clobber duplicate
 LDADD = \
   $(top_builddir)/libparted/libparted.la
 AM_CPPFLAGS = \
-  -I$(top_srcdir)/lib \
-  -I$(top_srcdir)/include
+  -I$(top_srcdir)/lib -I$(top_builddir)/lib \
+  -I$(top_srcdir)/include -I$(top_builddir)/include
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
 
 parted_dir = $(abs_top_builddir)/parted
