diff -Naur a/cado.h b/cado.h
--- a/cado.h	2014-11-15 11:39:41.539366085 +0100
+++ b/cado.h	2014-11-15 11:39:53.679076445 +0100
@@ -61,6 +61,7 @@
  * _POSIX_C_SOURCE==2 ?) fileno */
 #define _XOPEN_SOURCE   700     /* posix_memalign lrand48 strndup */
 #define _BSD_SOURCE     /* M_LN2 gethostname strdup random */
+#define _DEFAULT_SOURCE
 #define _ISOC99_SOURCE  /* Sometimes there's link trickery which causes fscanf to be linked in *only* when this is defined */
 #ifndef __cplusplus
 #define _GNU_SOURCE         /* asprintf vasprintf */
