--- audacity-svn13387/src/FFmpeg.h	2014-10-05 01:26:05.491172703 +0200
+++ audacity-svn/src/FFmpeg.h	2014-10-05 01:28:56.454499855 +0200
@@ -676,7 +676,7 @@
    );
    FFMPEG_FUNCTION_NO_RETURN(
       avformat_close_input,
-      (AVFormatContext **s),
+      (const AVFormatContext **s),
       (s)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
@@ -755,7 +755,7 @@
    FFMPEG_FUNCTION_WITH_RETURN(
       int,
       av_fifo_size,
-      (AVFifoBuffer *f),
+      (const AVFifoBuffer *f),
       (f)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
@@ -801,7 +801,7 @@
    FFMPEG_FUNCTION_WITH_RETURN(
       AVDictionaryEntry *,
       av_dict_get,
-      (AVDictionary *m, const char *key, const AVDictionaryEntry *prev, int flags),
+      (FF_CONST_AVUTIL53 AVDictionary *m, const char *key, const AVDictionaryEntry *prev, int flags),
       (m, key, prev, flags)
    );
    FFMPEG_FUNCTION_WITH_RETURN(

@@ -676,7 +676,7 @@
    );
    FFMPEG_FUNCTION_NO_RETURN(
       avformat_close_input,
-      (const AVFormatContext **s),
+      (AVFormatContext **s),
       (s)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
@@ -688,7 +688,7 @@
    FFMPEG_FUNCTION_WITH_RETURN(
       AVOutputFormat*,
       av_oformat_next,
-      (AVOutputFormat *f),
+      (const AVOutputFormat *f),
       (f)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
--- audacity-svn13387/src/FFmpeg.h	2014-10-05 01:56:04.001107671 +0200
+++ audacity-svn/src/FFmpeg.h	2014-10-05 01:58:12.924436343 +0200
@@ -676,7 +676,7 @@
    );
    FFMPEG_FUNCTION_NO_RETURN(
       avformat_close_input,
-      (const AVFormatContext **s),
+      (AVFormatContext **s),
       (s)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
@@ -688,7 +688,7 @@
    FFMPEG_FUNCTION_WITH_RETURN(
       AVOutputFormat*,
       av_oformat_next,
-      (AVOutputFormat *f),
+      (const AVOutputFormat *f),
       (f)
    );
    FFMPEG_FUNCTION_WITH_RETURN(
