--- ../vixn/install.sh
+++ install.sh
@@ -35,7 +35,7 @@
 install --mode=0644 vixn/*.py "$libdir/vixn"
 echo "Compiling python code"
 cd "$libdir/vixn"
-python -O - <<SCRIPT
+python2 -O - <<SCRIPT
 import compileall
 compileall.compile_dir('.')
 SCRIPT
@@ -44,7 +44,7 @@
 relative()
 {
     # Prints path $1 converted from absolute to relative to $2
-    python -c "
+    python2 -c "
 import os
 p1 = '$1'.split(os.path.sep)
 p2 = '$2'.split(os.path.sep)
@@ -83,7 +83,7 @@
     set -e
 fi
 echo "Building HTML docs"
-python makedocs.py
+python2 makedocs.py
 echo "Installing docs"
 [ -f ChangeLog ] && install --mode=0644 ChangeLog "$docdir"
 install --mode=0644 docs/COPYING docs/README "$docdir"
