diff -Naur vixn-0.7.15.orig/install.sh vixn-0.7.15.new/install.sh
--- vixn-0.7.15.orig/install.sh	2012-07-06 08:27:09.000000000 +0200
+++ vixn-0.7.15.new/install.sh	2012-07-06 08:28:11.000000000 +0200
@@ -35,7 +35,7 @@
 install --mode=0644 vixn/*.py "$libdir/vixn"
 echo "Compiling python code"
 cd "$libdir/vixn"
-python -O - <<SCRIPT
+python2 -O - <<SCRIPT
 import compileall
 compileall.compile_dir('.')
 SCRIPT
@@ -44,7 +44,7 @@
 relative()
 {
     # Prints path $1 converted from absolute to relative to $2
-    python -c "
+    python2 -c "
 import os
 p1 = '$1'.split(os.path.sep)
 p2 = '$2'.split(os.path.sep)
@@ -85,7 +85,7 @@
     set -e
 fi
 echo "Building HTML docs"
-python makedocs.py
+python2 makedocs.py
 echo "Installing docs"
 [ -f ChangeLog ] && install --mode=0644 ChangeLog "$docdir"
 install --mode=0644 docs/COPYING docs/README "$docdir"
diff -Naur vixn-0.7.15.orig/install.sh.orig vixn-0.7.15.new/install.sh.orig
--- vixn-0.7.15.orig/install.sh.orig	1970-01-01 01:00:00.000000000 +0100
+++ vixn-0.7.15.new/install.sh.orig	2012-07-06 08:27:16.000000000 +0200
@@ -0,0 +1,94 @@
+#!/bin/sh
+
+APPNAME=vixn
+
+cd "`dirname "$0"`"
+srcdir="`pwd`"
+
+if [ -n "$1" ] ; then
+    mkdir -p "$1"
+    cd "$1"
+    prefix="`pwd`"
+else
+    prefix=/usr/local
+fi
+
+libdir="$prefix/lib/$APPNAME"
+datadir="$prefix/share"
+confdir="$datadir/$APPNAME/config"
+docdir="$prefix/share/doc/$APPNAME/vixn"
+bindir="$prefix/bin"
+icondir="$datadir/icons/hicolor/scalable/apps"
+desktopdir="$datadir/applications"
+dbusdir="$datadir/dbus-1/services"
+
+cd "$srcdir"
+
+echo "Creating/ensuring directories:"
+echo $libdir $confdir $bindir
+echo $icondir $desktopdir $dbusdir
+echo $docdir
+install -d "$libdir/vixn" "$confdir" "$docdir/html" "$bindir" "$icondir" "$desktopdir" "$dbusdir"
+
+echo "Installing python code"
+install vixn.py vixgrep "$libdir"
+install --mode=0644 vixn/*.py "$libdir/vixn"
+echo "Compiling python code"
+cd "$libdir/vixn"
+python -O - <<SCRIPT
+import compileall
+compileall.compile_dir('.')
+SCRIPT
+cd "$srcdir"
+
+relative()
+{
+    # Prints path $1 converted from absolute to relative to $2
+    python -c "
+import os
+p1 = '$1'.split(os.path.sep)
+p2 = '$2'.split(os.path.sep)
+if os.path.isdir('$2'):
+    p2.append(p1[-1])
+for n in range(min(len(p1), len(p2))):
+    if p1[n] != p2[n]:
+        break
+p3 = ['..'] * (len(p2) - n - 1) + p1[n:]
+print os.path.sep.join(p3)
+"
+}
+
+echo "Linking executables in $bindir"
+#ln -nfs "$libdir/vixn.py" "$bindir/vixn"
+#ln -nfs "$libdir/vixgrep" "$bindir/vixgrep"
+ln -nfs `relative "$libdir/vixn.py" "$bindir/vixn"` "$bindir/vixn"
+ln -nfs `relative "$libdir/vixgrep" "$bindir/vixgrep"` "$bindir/vixgrep"
+
+echo "Installing data"
+install --mode=0755 -d "$confdir/languages"
+install --mode=0644 -t "$confdir/languages" config/languages/*
+install --mode=0644 config/menus "$confdir"
+install --mode=0644 vixn_logo.svg "$datadir/$APPNAME"
+install --mode=0644 vixn_logo.svg "$icondir/vixn.svg"
+install --mode=0644 vixn.desktop "$desktopdir"
+if [ "$prefix" = "/usr/local" ] ; then
+    install --mode=0644 org.vixn.service "$dbusdir"
+else
+    sed "s|/usr/local|${prefix}|" < org.vixn.service > "$dbusdir/org.vixn.service"
+fi
+
+
+if [ ! -f ChangeLog ] ; then
+    echo "Creating ChangeLog with git"
+    set +e
+    git log > ChangeLog && ln -s ../ChangeLog docs/ChangeLog
+    set -e
+fi
+echo "Building HTML docs"
+python makedocs.py
+echo "Installing docs"
+[ -f ChangeLog ] && install --mode=0644 ChangeLog "$docdir"
+install --mode=0644 docs/COPYING docs/README "$docdir"
+install --mode=0644 docs/html/*.html docs/html/*.txt "$docdir/html"
+install --mode=0755 docs/html/list_filetypes "$docdir/html"
+
diff -Naur vixn-0.7.15.orig/vixgrep vixn-0.7.15.new/vixgrep
--- vixn-0.7.15.orig/vixgrep	2012-07-06 08:27:09.000000000 +0200
+++ vixn-0.7.15.new/vixgrep	2012-07-06 08:29:31.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import os
 import subprocess
