Description: Bump libindicate version check to 0.7.
Author: Felix Geyer <debfx-pkg@fobos.de>

--- libindicate-qt-0.2.5.91.orig/CMakeLists.txt
+++ libindicate-qt-0.2.5.91/CMakeLists.txt
@@ -17,7 +17,7 @@ find_package(PkgConfig REQUIRED)
 
 add_definitions(-Wall)
 
-pkg_check_modules(INDICATE REQUIRED indicate-0.5)
+pkg_check_modules(INDICATE REQUIRED indicate-0.7)
 
 include_directories(
     ${QT_INCLUDE_DIR}
