--- Makefile.old	2011-04-12 20:38:35.812601439 +0200
+++ Makefile	2011-04-12 20:13:49.189320715 +0200
@@ -4,12 +4,17 @@
 CFLAGS = -Wall -c -I/usr/include/SDL $(DEBUG)
 LFLAGS = -Wall -L/usr/lib -lSDL -lpthread -lSDL_image -lSDL_gfx -lSDL_ttf -lSDL_mixer -lstdc++ $(DEBUG)
 
+
+ifndef DESTDIR
+DESTDIR=/usr/local
+endif
+
 ifndef GAMES_BINDIR
-GAMES_BINDIR=/usr/games/bin
+GAMES_BINDIR=$(DESTDIR)/bin
 endif
 
 ifndef GAMES_DATADIR
-GAMES_DATADIR=/usr/share/games
+GAMES_DATADIR=$(DESTDIR)/share
 endif
 
 CFLAGS+=-DBINDIR="\"$(GAMES_BINDIR)\""
@@ -44,10 +49,10 @@
 
 
 install:
-	mkdir -p $(DESTDIR)/$(GAMES_BINDIR)/
-	install -g games -m 750 ./snake $(DESTDIR)/$(GAMES_BINDIR)/
-	mkdir -p $(DESTDIR)/$(GAMES_DATADIR)/snake/
-	install -g games -m 640 ./src/res/* $(DESTDIR)/$(GAMES_DATADIR)/snake/
+	mkdir -p /$(GAMES_BINDIR)/
+	install $(BIN_INSTALL_PARAMS) ./snake /$(GAMES_BINDIR)/
+	mkdir -p /$(GAMES_DATADIR)/snake/
+	install $(DATA_INSTALL_PARAMS) ./src/res/* /$(GAMES_DATADIR)/snake/
 
 
 uninstall:
