diff -Naur qcairo-1.8.7.1-git74d6b5/configure.ac qcairo-1.8.7.1-git74d6b5.new/configure.ac
--- qcairo-1.8.7.1-git74d6b5/configure.ac	2009-01-27 18:06:58.000000000 -0300
+++ qcairo-1.8.7.1-git74d6b5.new/configure.ac	2010-11-22 17:51:44.357687550 -0300
@@ -161,7 +161,7 @@
   AC_ARG_VAR([png_REQUIRES], [module name for libpng to search for using pkg-config])
   if test "x$png_REQUIRES" = x; then
     # libpng13 is GnuWin32's libpng-1.2.8 :-(
-    for l in libpng12 libpng13 libpng10 ; do
+    for l in libpng12 libpng13 libpng10 libpng; do
       if $PKG_CONFIG --exists $l ; then
         png_REQUIRES=$l
         use_png=yes
