--- qtbase-opensource-src-5.1.0/src/plugins/platforms/offscreen/qoffscreenintegration.cpp.orig	2013-07-10 20:40:52.350023574 +0200
+++ qtbase-opensource-src-5.1.0/src/plugins/platforms/offscreen/qoffscreenintegration.cpp	2013-07-10 21:03:31.496725681 +0200
@@ -103,8 +103,12 @@
 #endif
 #elif defined(Q_OS_WIN)
     m_eventDispatcher = new QOffscreenEventDispatcher<QEventDispatcherWin32>();
+#if defined (QT_NO_FREETYPE)
+    m_fontDatabase.reset(new QPlatformFontDatabase());
+#else
     m_fontDatabase.reset(new QBasicFontDatabase());
 #endif
+#endif
 
 #ifndef QT_NO_DRAGANDDROP
     m_drag.reset(new QOffscreenDrag);
