diff --git a/pspirkeyb/Makefile b/pspirkeyb/Makefile
index 0d4af2c..c422694 100644
--- a/pspirkeyb/Makefile
+++ b/pspirkeyb/Makefile
@@ -32,13 +32,13 @@ PSPSDK=$(shell psp-config --pspsdk-path)
 
 install: all
 	$(MAKE) -C libpspirkeyb install
-	mkdir -p $(PSPSDK)/samples/irkeyb
-	mkdir -p $(PSPSDK)/samples/irkeyb/keymap
-	cp libpspirkeyb/keymap/*.ini $(PSPSDK)/samples/irkeyb/keymap
-	cp libpspirkeyb/keymap/README.txt $(PSPSDK)/samples/irkeyb/keymap
-	cp libpspirkeyb/config/pspirkeyb.ini $(PSPSDK)/samples/irkeyb
-	cp asciidemo/main.c $(PSPSDK)/samples/irkeyb
-	cp asciidemo/Makefile.sdk $(PSPSDK)/samples/irkeyb/Makefile
+	install -d $(DESTDIR)$(PSPSDK)/samples/irkeyb
+	install -d $(DESTDIR)$(PSPSDK)/samples/irkeyb/keymap
+	cp libpspirkeyb/keymap/*.ini $(DESTDIR)$(PSPSDK)/samples/irkeyb/keymap
+	cp libpspirkeyb/keymap/README.txt $(DESTDIR)$(PSPSDK)/samples/irkeyb/keymap
+	cp libpspirkeyb/config/pspirkeyb.ini $(DESTDIR)$(PSPSDK)/samples/irkeyb
+	cp asciidemo/main.c $(DESTDIR)$(PSPSDK)/samples/irkeyb
+	cp asciidemo/Makefile.sdk $(DESTDIR)$(PSPSDK)/samples/irkeyb/Makefile
 
 uninstall:
 	$(MAKE) -C libpspirkeyb install
diff --git a/pspirkeyb/libpspirkeyb/Makefile b/pspirkeyb/libpspirkeyb/Makefile
index a1d5dd9..53375c8 100644
--- a/pspirkeyb/libpspirkeyb/Makefile
+++ b/pspirkeyb/libpspirkeyb/Makefile
@@ -13,11 +13,13 @@ PSPSDK=$(shell psp-config --pspsdk-path)
 include $(PSPSDK)/lib/build.mak
 
 install: all
-	cp libpspirkeyb.a $(PSPSDK)/lib
-	cp include/pspirkeyb.h $(PSPSDK)/include
-	cp include/pspirkeyb_rawkeys.h $(PSPSDK)/include
+	install -d $(DESTDIR)$(PSPSDK)/lib
+	cp libpspirkeyb.a $(DESTDIR)$(PSPSDK)/lib
+	install -d $(DESTDIR)$(PSPSDK)/include
+	cp include/pspirkeyb.h $(DESTDIR)$(PSPSDK)/include
+	cp include/pspirkeyb_rawkeys.h $(DESTDIR)$(PSPSDK)/include
 
 uninstall:
-	rm -f $(PSPSDK)/lib/libpspirkeyb.a
-	rm -f $(PSPSDK)/include/pspirkeyb.h
-	rm -f $(PSPSDK)/include/pspirkeyb_rawkeys.h
+	rm -f $(DESTDIR)$(PSPSDK)/lib/libpspirkeyb.a
+	rm -f $(DESTDIR)$(PSPSDK)/include/pspirkeyb.h
+	rm -f $(DESTDIR)$(PSPSDK)/include/pspirkeyb_rawkeys.h
