--- pdfsandwich.ml	2011-09-25 13:20:10.123429858 +0200
+++ pdfsandwich.ml.new	2011-09-25 13:21:17.506759118 +0200
@@ -90,13 +90,13 @@
 		pr ("\nParallel processing with " ^ (string_of_int nthreads) ^ " threads started.\nProcessing page order may differ from original page order.\n");
 	
 	let process_page (curr_page, pdfname) =
-		let tmppicfile = Filename.temp_file "pdfsandwich" ".bmp" in
+		let tmppicfile = Filename.temp_file "pdfsandwich" ".tiff" in
 		let tmpocrfile = Filename.temp_file "pdfsandwich" ".hOCR" in
 		let tmpcolfigfile = Filename.temp_file "pdfsandwich" "_col.bmp" in
 		if not !quiet then
 			Printf.printf "Processing page %i.\n" curr_page;
 		flush_all ();
-		run (!convert ^ " " ^ convertopts ^ " -type Bilevel  -density " ^ resolution ^ " \"" ^ infile ^ "[" ^ (string_of_int (curr_page-1)) ^ "]\" BMP3:" ^ tmppicfile);
+		run (!convert ^ " " ^ convertopts ^ " -type Bilevel  -density " ^ resolution ^ " \"" ^ infile ^ "[" ^ (string_of_int (curr_page-1)) ^ "]\" TIFF:" ^ tmppicfile);
 		let cunout = if not !verbose then ">/dev/null 2>&1" else "" in
 		run (!cuneiform ^ " " ^ cuniopts ^ " -l " ^ language ^ " -f hocr -o " ^ tmpocrfile ^  " " ^ tmppicfile ^ cunout);
 		let picfile = 
@@ -134,7 +134,7 @@
 	let pdffilenamelist = List.map snd (Array.to_list tmppdf_arr) in
 	let pdfliststring = String.concat " " pdffilenamelist in
 	pr ("OCR done. Writing \"" ^ outfile ^ "\"");
-	run (!gs ^ " -q -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -sOutputFile=\"" ^ outfile ^ "\" " ^ pdfliststring);
+	run (!gs ^ " -q -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dPDFSETTINGS=/prepress -sOutputFile=\"" ^ outfile ^ "\" " ^ pdfliststring);
 	List.iter Sys.remove pdffilenamelist;
 ;;
 
