--- boost_1_43_0/libs/thread/src/win32/tss_pe.cpp	2008-10-13 22:30:13.000000000 +0200
+++ boost_1_43_0-new/libs/thread/src/win32/tss_pe.cpp	2010-07-04 00:28:50.352555432 +0200
@@ -11,7 +11,14 @@
 
 #if defined(BOOST_HAS_WINTHREADS) && defined(BOOST_THREAD_BUILD_LIB) 
 
-#if (defined(__MINGW32__) && !defined(_WIN64)) || defined(__MINGW64__)
+#if !defined(_WIN64) && defined(__MINGW32__)
+
+#if __MINGW32_MAJOR_VERSION > 3 || __MINGW32_MAJOR_VERSION == 3 && __MINGW32_MINOR_VERSION >= 18
+
+// MinGW32 3.18 and later has his own TLS implementation
+extern "C" void tss_cleanup_implemented(void) {}
+
+#else
 
 #include <boost/thread/detail/tss_hooks.hpp>
 
@@ -77,6 +84,9 @@
 };
 
 
+#endif // __MINGW32_MAJOR_VERSION > 3 || __MINGW32_MAJOR_VERSION == 3 && __MINGW32_MINOR_VERSION >= 18
+
+
 #elif  defined(_MSC_VER) && !defined(UNDER_CE)
 
     #include <boost/thread/detail/tss_hooks.hpp>
