Index: src/arch/ArchHooks/ArchHooks_Unix.cpp
===================================================================
--- src/arch/ArchHooks/ArchHooks_Unix.cpp	(revision 27475)
+++ src/arch/ArchHooks/ArchHooks_Unix.cpp	(working copy)
@@ -266,6 +266,7 @@
 	FILEMAN->Mount( "dir", sUserDataPath + "/Logs", "/Logs" );
 	FILEMAN->Mount( "dir", sUserDataPath + "/Save", "/Save" );
 	FILEMAN->Mount( "dir", sUserDataPath + "/Screenshots", "/Screenshots" );
+	FILEMAN->Mount( "dir", (RString)szHome + "/" + PRODUCT_ID, "/" );
 
 	RString Root;
 	struct stat st;
