Index: src/global.h
===================================================================
--- src/global.h	(revision 27475)
+++ src/global.h	(working copy)
@@ -45,9 +45,11 @@
 
 /* Make sure everyone has min and max: */
 #include <algorithm>
+#include <memory>
 
 /* Everything will need string for one reason or another: */
 #include <string>
+#include <cstring>
 
 /* And vector: */
 #include <vector>
Index: src/ScreenOptionsMasterPrefs.cpp
===================================================================
--- src/ScreenOptionsMasterPrefs.cpp	(revision 27475)
+++ src/ScreenOptionsMasterPrefs.cpp	(working copy)
@@ -127,7 +127,7 @@
 }
 
 template <>
-static void MovePref<bool>( int &iSel, bool bToSel, const ConfOption *pConfOption )
+void MovePref<bool>( int &iSel, bool bToSel, const ConfOption *pConfOption )
 {
 	IPreference *pPref = IPreference::GetPreferenceByName( pConfOption->m_sPrefName );
 	ASSERT_M( pPref != NULL, pConfOption->m_sPrefName );
