diff -aur cnijfilter-common-2.70.pristine/ppd/canonmp510.ppd cnijfilter-common-2.70.new/ppd/canonmp510.ppd
--- cnijfilter-common-2.70.pristine/ppd/canonmp510.ppd	2007-02-01 01:46:04.000000000 +0100
+++ cnijfilter-common-2.70.new/ppd/canonmp510.ppd	2009-11-18 21:23:27.929777804 +0100
@@ -61,7 +61,11 @@
 
 *OpenUI *Resolution/Output Resolution: PickOne
 *DefaultResolution: 600
+*Resolution 150/150 dpi: "<</HWResolution[150 150]>>setpagedevice"
+*Resolution 300/300 dpi: "<</HWResolution[300 300]>>setpagedevice"
 *Resolution 600/600 dpi: "<</HWResolution[600 600]>>setpagedevice"
+*Resolution 1200/1200 dpi: "<</HWResolution[1200 1200]>>setpagedevice"
+*Resolution 2400/2400 dpi: "<</HWResolution[2400 2400]>>setpagedevice"
 *CloseUI: *Resolution
 
 *OpenUI *ColorModel/Color Model: PickOne
@@ -140,6 +144,20 @@
 *InputSlot frontplain/Front for Plain Paper: "<</MediaPosition 4>>setpagedevice"
 *CloseUI: *InputSlot
 
+*OpenUI *CNQuality/Quality: PickOne
+*DefaultCNQuality: 3
+*CNQuality 2/High: "2"
+*CNQuality 3/Normal: "3"
+*CNQuality 4/Standard: "4"
+*CNQuality 5/Economy: "5"
+*CloseUI: *CNQuality
+
+*OpenUI *CNGrayscale/Grayscale: PickOne
+*DefaultCNGrayscale: false
+*CNGrayscale false/Off: "false"
+*CNGrayscale true/On: "true"
+*CloseUI: *CNGrayscale
+
 *DefaultImageableArea: A4
 *ImageableArea Letter: "18.14 14.17 594.14 783.50"
 *ImageableArea Legal: "18.14 14.17 594.14 999.50"
