--- a/configure.ac	2013-07-24 09:38:59.241386193 -0430
+++ b/configure.ac	2013-07-23 15:05:44.136713469 -0430
@@ -98,6 +98,9 @@
 AC_FUNC_REALLOC
 AC_CHECK_FUNCS([getcwd memset mkdir strchr strcspn strdup strrchr strstr strtol], [], [AC_MSG_ERROR([Cannot find required function.])])
 
+# Support for large files in 32bit systems
+AC_SYS_LARGEFILE
+
 # Output
 # Required for gphoto-ptp2
 AX_NEED_STDINT_H([src/_stdint.h])
