diff -aur lio-utils/scripts/rc.target lio-utils.new/scripts/rc.target
--- lio-utils/scripts/rc.target	2011-05-06 23:16:58.000000000 +0200
+++ lio-utils.new/scripts/rc.target	2012-02-27 16:14:25.000000000 +0100
@@ -112,11 +112,22 @@
 			continue
 		fi
 
-		if [[ $i =~ ".rpmsave" ]]; then
+		# files from arch package manager
+		if [[ $i =~ ".pacsave" ]]; then
+			continue
+		fi
+		if [[ $i =~ ".pacnew" ]]; then
+			continue
+		fi
+		if [[ $i =~ ".pacorig" ]]; then
 			continue
 		fi
 
-		if [[ $i =~ ".dpkg-dist" ]]; then
+		# files from the free branch
+		if [[ $i =~ ".json" ]]; then
+			continue
+		fi
+		if [[ $i =~ ".backup" ]]; then
 			continue
 		fi
 
