From 2915682ecc79d3dbfa003d7546b336bb91feec31 Mon Sep 17 00:00:00 2001
From: Paul Gideon Dann <pdgiddie@gmail.com>
Date: Tue, 15 Feb 2011 21:19:52 +0000
Subject: [PATCH 1/2] Fixing bug that causes monthly jobs to be skipped

---
 database.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/database.c b/database.c
index 955875f..b797f65 100644
--- a/database.c
+++ b/database.c
@@ -410,8 +410,10 @@ SynchronizeFile(const char *dpath, const char *fileName, const char *userName)
 								break;
 							case 5:
 								line.cl_Freq = MONTHLY_FREQ;
+								break;
 							case 6:
 								line.cl_Freq = YEARLY_FREQ;
+								break;
 							/* else line.cl_Freq will remain 0 */
 						}
 					}
-- 
1.7.4.1

