--- Makefile	2013-02-23 11:12:41.000000000 -0500
+++ Makefile2	2013-05-30 15:43:38.003756093 -0400
@@ -29,7 +29,7 @@
 #===============================================================================
 
 MPI_DIR   = /opt/mpich/3.0.2-$(COMPILER)
-HDF5_DIR  = /opt/hdf5/1.8.10-$(COMPILER)
+HDF5_DIR  = /opt/hdf5/1.8.10-patch1-$(COMPILER)
 PETSC_DIR = /opt/petsc/3.3-p6-$(COMPILER)
 
 #===============================================================================
@@ -233,18 +233,32 @@
 	$(F90) $(objects) $(templates) $(xml_fort) $(LDFLAGS) -o $@
 install:
 	@install -D $(program) $(DESTDIR)$(prefix)/bin/$(program)
+	@install -D utils/convert_xsdir.py $(DESTDIR)$(prefix)/bin/convert_xsdir
+	@install -D utils/convert_xsdata.py $(DESTDIR)$(prefix)/bin/convert_xsdata
 	@install -D utils/statepoint_cmp.py $(DESTDIR)$(prefix)/bin/statepoint_cmp
+	@install -D utils/statepoint.py $(DESTDIR)$(prefix)/lib/python2.7/$(program)/statepoint.py
 	@install -D utils/statepoint_histogram.py $(DESTDIR)$(prefix)/bin/statepoint_histogram
 	@install -D utils/statepoint_meshplot.py $(DESTDIR)$(prefix)/bin/statepoint_meshplot
 	@install -D ../man/man1/openmc.1 $(DESTDIR)$(prefix)/share/man/man1/openmc.1
-	@install -D ../LICENSE $(DESTDIR)$(prefix)/share/doc/$(program)/copyright
+	@install -D ../LICENSE $(DESTDIR)$(prefix)/share/licenses/$(program)/LICENSE
+	@install -D ../cross_sections.xml $(DESTDIR)/etc/$(program)/cross_sections.xml
+	@install -D ../cross_sections_ascii.xml $(DESTDIR)/etc/$(program)/cross_sections_ascii.xml
+	@install -D ../cross_sections_serpent.xml $(DESTDIR)/etc/$(program)/cross_sections_serpent.xml
+
 uninstall:
 	@rm $(DESTDIR)$(prefix)/bin/$(program)
 	@rm $(DESTDIR)$(prefix)/bin/statepoint_cmp
 	@rm $(DESTDIR)$(prefix)/bin/statepoint_histogram
 	@rm $(DESTDIR)$(prefix)/bin/statepoint_meshplot
+	@rm $(DESTDIR)$(prefix)/lib/python2.7/$(program)/statepoint.py
+	@rm $(DESTDIR)$(prefix)/bin/convert_xsdir
+	@rm $(DESTDIR)$(prefix)/bin/convert_xsdata
 	@rm $(DESTDIR)$(prefix)/share/man/man1/openmc.1
-	@rm $(DESTDIR)$(prefix)/share/doc/$(program)/copyright
+	@rm $(DESTDIR)$(prefix)/share/licenses/$(program)/LICENSE
+	@rm $(DESTDIR)/etc/$(program)/cross_sections.xml
+	@rm $(DESTDIR)/etc/$(program)/cross_sections_ascii.xml
+	@rm $(DESTDIR)/etc/$(program)/cross_sections_serpent.xml
+
 distclean: clean
 	cd xml-fortran; make clean
 	cd templates; make clean
