diff --git a/Opcodes/CMakeLists.txt b/Opcodes/CMakeLists.txt
index 436b068..ab1d6bb 100644
--- a/Opcodes/CMakeLists.txt
+++ b/Opcodes/CMakeLists.txt
@@ -23,7 +23,8 @@ check_include_file(lo/lo.h OSC_HEADER)
 find_library(LIBLO_LIBRARY lo)
 find_package(PNG)
 find_package(PythonLibs)
-find_package(Lua51)
+find_package(PkgConfig)
+pkg_check_modules(LUAJIT luajit)
 check_include_file(wiiuse.h WIIUSE_H)
 find_library(WIIUSE_LIB wiiuse)
 
@@ -127,10 +128,10 @@ if(BUILD_PYTHON_OPCODES)
     make_plugin(py py/pythonopcodes.c ${PYTHON_LIBRARIES})
 endif()
 
-check_deps(BUILD_LUA_OPCODES LUA51_FOUND OPENMP_FOUND)
+check_deps(BUILD_LUA_OPCODES LUAJIT_FOUND OPENMP_FOUND)
 if(BUILD_LUA_OPCODES)
-    include_directories(${LUA_INCLUDE_DIR})
-    make_plugin(LuaCsound LuaCsound.cpp ${LUA_LIBRARIES})
+    include_directories(BEFORE ${LUAJIT_INCLUDEDIR})
+    make_plugin(LuaCsound LuaCsound.cpp ${LUAJIT_LIBRARIES})
     add_compiler_flags(${OpenMP_CXX_FLAGS} TARGETS LuaCsound)
     add_compiler_flags(${OpenMP_CXX_FLAGS} TARGETS LuaCsound LINKER)
 endif()
diff --git a/frontends/CsoundAC/CMakeLists.txt b/frontends/CsoundAC/CMakeLists.txt
index cd648a4..0d9d691 100644
--- a/frontends/CsoundAC/CMakeLists.txt
+++ b/frontends/CsoundAC/CMakeLists.txt
@@ -13,11 +13,12 @@ find_package(FLTK)
 ENDIF(APPLE)
 find_package(SWIG)
 find_package(PythonLibs)
-find_package(Lua51)
+find_package(PkgConfig)
+pkg_check_modules(LUAJIT luajit)
 
 check_deps(BUILD_CSOUND_AC Boost_FOUND BUILD_CXX_INTERFACE FLTK_FOUND LIBSNDFILE_LIBRARY)
 check_deps(BUILD_CSOUND_AC_PYTHON_INTERFACE SWIG_FOUND PYTHONLIBS_FOUND)
-check_deps(BUILD_CSOUND_AC_LUA_INTERFACE SWIG_FOUND LUA51_FOUND)
+check_deps(BUILD_CSOUND_AC_LUA_INTERFACE SWIG_FOUND LUAJIT_FOUND)
 
 if(BUILD_CSOUND_AC)
     include_directories(${CMAKE_HOME_DIRECTORY}/interfaces)
@@ -116,17 +117,18 @@ if(BUILD_CSOUND_AC)
 
     if(BUILD_CSOUND_AC_LUA_INTERFACE)
         message(STATUS "Building CsoundAC Lua interface")
-        find_package(Lua51)
-        include_directories(${LUA_INCLUDE_DIR})
+        find_package(PkgConfig)
+        pkg_check_modules(LUAJIT luajit)
+        include_directories(BEFORE ${LUAJIT_INCLUDEDIR})
 
         SET_SOURCE_FILES_PROPERTIES(luaCsoundAC.i
             PROPERTIES CPLUSPLUS ON
             SWIG_MODULE_NAME luaCsoundAC)
         SWIG_ADD_MODULE(luaCsoundAC lua luaCsoundAC.i)
-        SWIG_LINK_LIBRARIES(luaCsoundAC libCsoundAC libcsnd ${CSOUNDLIB} ${FLTK_LIBRARIES} ${LIBSNDFILE_LIBRARY} ${LUA_LIBRARIES})
+        SWIG_LINK_LIBRARIES(luaCsoundAC libCsoundAC libcsnd ${CSOUNDLIB} ${FLTK_LIBRARIES} ${LIBSNDFILE_LIBRARY} ${LUAJIT_LIBRARIES})
     
         if(MUSICXML_LIBRARY) 
-            SWIG_LINK_LIBRARIES(luaCsoundAC libCsoundAC libcsnd ${CSOUNDLIB} ${FLTK_LIBRARIES} ${LIBSNDFILE_LIBRARY} ${MUSICXML_LIBRARY} ${LUA_LIBRARIES})
+            SWIG_LINK_LIBRARIES(luaCsoundAC libCsoundAC libcsnd ${CSOUNDLIB} ${FLTK_LIBRARIES} ${LIBSNDFILE_LIBRARY} ${MUSICXML_LIBRARY} ${LUAJIT_LIBRARIES})
         endif()
 
         set_target_properties(${SWIG_MODULE_luaCsoundAC_REAL_NAME} PROPERTIES
diff --git a/interfaces/CMakeLists.txt b/interfaces/CMakeLists.txt
index e652732..6352583 100644
--- a/interfaces/CMakeLists.txt
+++ b/interfaces/CMakeLists.txt
@@ -46,11 +46,12 @@ FIND_PACKAGE(SWIG 2.0)
 FIND_PACKAGE(PythonLibs)
 FIND_PACKAGE(JNI)
 FIND_PACKAGE(Java)
-FIND_PACKAGE(Lua51)
+FIND_PACKAGE(PkgConfig)
+pkg_check_modules(LUAJIT luajit)
 
 check_deps(BUILD_PYTHON_INTERFACE SWIG_FOUND BUILD_CXX_INTERFACE PYTHONLIBS_FOUND)
 check_deps(BUILD_JAVA_INTERFACE SWIG_FOUND BUILD_CXX_INTERFACE JNI_FOUND JAVA_FOUND)
-check_deps(BUILD_LUA_INTERFACE SWIG_FOUND BUILD_CXX_INTERFACE LUA51_FOUND)
+check_deps(BUILD_LUA_INTERFACE SWIG_FOUND BUILD_CXX_INTERFACE LUAJIT_FOUND)
 
 if(BUILD_PYTHON_INTERFACE OR BUILD_JAVA_INTERFACE OR BUILD_LUA_INTERFACE)
 
@@ -139,9 +140,9 @@ if(BUILD_PYTHON_INTERFACE OR BUILD_JAVA_INTERFACE OR BUILD_LUA_INTERFACE)
     endif()
 
     if(BUILD_LUA_INTERFACE)
-        INCLUDE_DIRECTORIES(${LUA_INCLUDE_DIR})
+        INCLUDE_DIRECTORIES(BEFORE ${LUAJIT_INCLUDEDIR})
 
-        set(swiglua_LIBS ${LUA_LIBRARIES} ${CSOUNDLIB} libcsnd)
+        set(swiglua_LIBS ${LUAJIT_LIBRARIES} ${CSOUNDLIB} libcsnd)
     
         SET_SOURCE_FILES_PROPERTIES(lua_interface.i
             PROPERTIES CPLUSPLUS ON
