diff -ur guikachu-1.5.6-new/data/Makefile.in guikachu-1.5.6/data/Makefile.in
--- guikachu-1.5.6/data/Makefile.in	2005-11-03 14:51:48.000000000 -0800
+++ guikachu-1.5.6-new/data/Makefile.in	2005-12-29 10:37:11.000000000 -0800
@@ -133,7 +133,7 @@
 PO_IN_DATADIR_TRUE = @PO_IN_DATADIR_TRUE@
 RANLIB = @RANLIB@
 STRIP = @STRIP@
-UPDATE_MIME = @UPDATE_MIME@
+
 USE_NLS = @USE_NLS@
 VERSION = @VERSION@
 XSLTPROC = @XSLTPROC@
@@ -421,16 +421,13 @@
 %.xml: %.xml.in $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po)
 	$(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -x -u -c $(top_builddir)/po/.intltool-merge-cache
 
-install-mime: $(mime_package_files)
-@HAVE_UPDATE_MIME_TRUE@	$(UPDATE_MIME) $(mimedir)
-
 %.schemas: %.schemas.in $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po)
 	$(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -s -u -c $(top_builddir)/po/.intltool-merge-cache
 
 install-schemas: $(schemas_DATA)
 	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $?
 
-install-data-local: install-schemas install-mime
+install-data-local: install-schemas
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
