diff -crB istanbul-0.2.2//bin/istanbul.in istanbul-0.2.2-python2//bin/istanbul.in
*** istanbul-0.2.2//bin/istanbul.in	2007-02-23 14:01:12.000000000 +0300
--- istanbul-0.2.2-python2//bin/istanbul.in	2011-07-24 11:40:28.494861557 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  # -*- Mode: Python -*-
  # vi:si:et:sw=4:sts=4:ts=4
  #
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- Mode: Python -*-
  # vi:si:et:sw=4:sts=4:ts=4
  #
