From 89c3c12dc587a3528d654f0720c5d05b8a7f7def Mon Sep 17 00:00:00 2001
From: Jekyll Wu <adaptee@gmail.com>
Date: Sat, 5 May 2012 09:01:40 +0800
Subject: [PATCH] fix import

---
 config.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.py b/config.py
index ab63f0b..09887e6 100644
--- a/config.py
+++ b/config.py
@@ -11,7 +11,7 @@ from md5sum import md5sum
 import sys
 import os
 import exceptions
-import bsddb
+import bsddb3 as bsddb
 from ConfigParser import ConfigParser
 home=os.environ['HOME']
 cfghome=os.path.join(home,'.pychmviewer')
-- 
1.7.10.1

