--- vinagre-0.5.0.orig/src/vinagre-tab.h	2008-05-09 16:25:27.000000000 +0200
+++ vinagre-0.5.0/src/vinagre-tab.h		2008-05-09 16:33:24.000000000 +0200
@@ -87,6 +87,8 @@
 void		  vinagre_tab_paste_text	(VinagreTab *tab,
 						 const gchar *text);
 
+void              vinagre_tab_send_cad          (VinagreTab *tab);
+
 gboolean	  vinagre_tab_set_scaling	(VinagreTab *tab, gboolean active);
 gboolean	  vinagre_tab_get_scaling	(VinagreTab *tab);
 

--- vinagre-0.5.0.orig/src/vinagre-tab.c	2008-05-09 16:25:27.000000000 +0200
+++ vinagre-0.5.0/src/vinagre-tab.c		2008-05-09 16:36:53.000000000 +0200
@@ -27,6 +27,8 @@
 #include <gnome-keyring.h>
 #include <vncdisplay.h>
 
+#include <gdk/gdkkeysyms.h>
+
 #include "vinagre-notebook.h"
 #include "vinagre-tab.h"
 #include "vinagre-utils.h"
@@ -822,4 +824,13 @@
   return tab->priv->state;
 }
 
+void
+vinagre_tab_send_cad(VinagreTab *tab)
+{
+  g_return_if_fail (VINAGRE_IS_TAB (tab));
+   
+  guint keys[] = { GDK_Control_L, GDK_Alt_L, GDK_Delete };
+  vnc_display_send_keys(VNC_DISPLAY(tab->priv->vnc), keys, sizeof(keys)/sizeof(keys[0]));
+}
+
 /* vim: ts=8 */

--- vinagre-0.5.0.orig/src/vinagre-window.c	2008-05-09 16:25:27.000000000 +0200
+++ vinagre-0.5.0/src/vinagre-window.c	2008-05-09 16:40:38.000000000 +0200
@@ -187,6 +187,19 @@
 	return FALSE;
     }
 
+  switch (event->keyval)
+    {
+    case GDK_F11:
+      if (window->priv->active_tab)
+        vinagre_window_toggle_fullscreen (window);
+      return FALSE;
+    case GDK_Insert:
+      if (window->priv->active_tab) {
+        if ((event->state & GDK_CONTROL_MASK) && (event->state & GDK_MOD1_MASK))
+          vinagre_tab_send_cad(VINAGRE_TAB(window->priv->active_tab));
+        }
+      }
+
   return GTK_WIDGET_CLASS (vinagre_window_parent_class)->key_press_event (widget, event);
 }
 
