diff --git a/djbdns-1.05/dnscache-conf.c b/djbdns-1.05/dnscache-conf.c
index e8c12be..c7972e3 100644
--- a/djbdns-1.05/dnscache-conf.c
+++ b/djbdns-1.05/dnscache-conf.c
@@ -121,13 +121,13 @@ int main(int argc,char **argv)
   seed_addtime(); start("env/DATALIMIT"); outs("3000000\n"); finish();
   seed_addtime(); perm(0644);
   seed_addtime(); start("run");
-  outs("#!/bin/sh\nexec 2>&1\nexec <seed\nexec envdir ./env sh -c '\n  exec envuidgid "); outs(user);
-  outs(" softlimit -o250 -d \"$DATALIMIT\" ");
+  outs("#!/bin/sh\nexec 2>&1\nexec <seed\nexec chpst -e ./env sh -c '\n  exec chpst -U "); outs(user);
+  outs(" -o250 -d \"$DATALIMIT\" ");
   outs(auto_home); outs("/bin/dnscache\n'\n"); finish();
   seed_addtime(); perm(0755);
   seed_addtime(); start("log/run");
-  outs("#!/bin/sh\nexec setuidgid "); outs(loguser);
-  outs(" multilog t ./main\n"); finish();
+  outs("#!/bin/sh\nexec chpst -u "); outs(loguser);
+  outs(" svlogd t ./main\n"); finish();
   seed_addtime(); perm(0755);
   seed_addtime(); makedir("root");
   seed_addtime(); perm(02755);
diff --git a/djbdns-1.05/generic-conf.c b/djbdns-1.05/generic-conf.c
index db4aa1d..6c864b4 100644
--- a/djbdns-1.05/generic-conf.c
+++ b/djbdns-1.05/generic-conf.c
@@ -92,8 +92,8 @@ void makelog(const char *user,int uid,int gid)
 
   start("log/run");
   outs("#!/bin/sh\nexec");
-  outs(" setuidgid "); outs(user);
-  outs(" multilog t ./main\n");
+  outs(" chpst -u "); outs(user);
+  outs(" svlogd t ./main\n");
   finish();
   perm(0755);
 }
diff --git a/djbdns-1.05/tinydns-conf.c b/djbdns-1.05/tinydns-conf.c
index db83f11..44702cf 100644
--- a/djbdns-1.05/tinydns-conf.c
+++ b/djbdns-1.05/tinydns-conf.c
@@ -45,8 +45,8 @@ int main(int argc,char **argv)
   perm(0644);
 
   start("run");
-  outs("#!/bin/sh\nexec 2>&1\nexec envuidgid "); outs(user);
-  outs(" envdir ./env softlimit -d300000 ");
+  outs("#!/bin/sh\nexec 2>&1\nexec chpst -U "); outs(user);
+  outs(" -e ./env -d300000 ");
   outs(auto_home); outs("/bin/tinydns\n");
   finish();
   perm(0755);
diff --git a/djbdns-1.05/axfrdns-conf.c b/djbdns-1.05/axfrdns-conf.c
index 4dc8657..56a7637 100644
--- a/djbdns-1.05/axfrdns-conf.c
+++ b/djbdns-1.05/axfrdns-conf.c
@@ -49,8 +49,8 @@ int main(int argc,char **argv)
   perm(0644);
 
   start("run");
-  outs("#!/bin/sh\nexec 2>&1\nexec envdir ./env sh -c '\n  exec envuidgid "); outs(user);
-  outs(" softlimit -d300000 tcpserver -vDRHl0 -x tcp.cdb -- \"$IP\" 53 ");
+  outs("#!/bin/sh\nexec 2>&1\nexec chpst -e ./env sh -c '\n  exec chpst -U "); outs(user);
+  outs(" -d300000 tcpsvd -v -l $IP -i ./peers \"$IP\" 53 ");
   outs(auto_home); outs("/bin/axfrdns\n'\n");
   finish();
   perm(0755);
diff --git a/djbdns-1.05/pickdns-conf.c b/djbdns-1.05/pickdns-conf.c
index 9edd184..6c00fa8 100644
--- a/djbdns-1.05/pickdns-conf.c
+++ b/djbdns-1.05/pickdns-conf.c
@@ -45,8 +45,8 @@ int main(int argc,char **argv)
   perm(0644);
 
   start("run");
-  outs("#!/bin/sh\nexec 2>&1\nexec envuidgid "); outs(user);
-  outs(" envdir ./env softlimit -d250000 ");
+  outs("#!/bin/sh\nexec 2>&1\nexec chpst -U "); outs(user);
+  outs(" -e ./env -d250000 ");
   outs(auto_home); outs("/bin/pickdns\n");
   finish();
   perm(0755);
diff --git a/djbdns-1.05/rbldns-conf.c b/djbdns-1.05/rbldns-conf.c
index 79d446f..4a2bd40 100644
--- a/djbdns-1.05/rbldns-conf.c
+++ b/djbdns-1.05/rbldns-conf.c
@@ -50,8 +50,8 @@ int main(int argc,char **argv)
   perm(0644);
 
   start("run");
-  outs("#!/bin/sh\nexec 2>&1\nexec envuidgid "); outs(user);
-  outs(" envdir ./env softlimit -d250000 ");
+  outs("#!/bin/sh\nexec 2>&1\nexec chpst -U "); outs(user);
+  outs(" -e ./env -d250000 ");
   outs(auto_home); outs("/bin/rbldns\n");
   finish();
   perm(0755);
diff --git a/djbdns-1.05/walldns-conf.c b/djbdns-1.05/walldns-conf.c
index b46f19a..e0ed3d0 100644
--- a/djbdns-1.05/walldns-conf.c
+++ b/djbdns-1.05/walldns-conf.c
@@ -45,8 +45,8 @@ int main(int argc,char **argv)
   perm(0644);
 
   start("run");
-  outs("#!/bin/sh\nexec 2>&1\nexec envuidgid "); outs(user);
-  outs(" envdir ./env softlimit -d250000 ");
+  outs("#!/bin/sh\nexec 2>&1\nexec chpst -U "); outs(user);
+  outs(" -e ./env -d250000 ");
   outs(auto_home); outs("/bin/walldns\n");
   finish();
   perm(0755);
