diff --git a/configure.ac b/configure.ac
index 5fc3929..a700df4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4,7 +4,12 @@ AC_INIT([GtkTerm], [1.0], [https://fedorahosted.org/gtkterm])
 AM_INIT_AUTOMAKE([foreign -Wall -Werror])
 AC_PROG_CC
 AC_CONFIG_HEADERS([config.h])
-AC_CONFIG_FILES([Makefile src/Makefile])
+
+GETTEXT_PACKAGE="gtkterm"
+AC_SUBST(GETTEXT_PACKAGE)
+AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [GETTEXT package name])
+IT_PROG_INTLTOOL([0.40.0])
+AM_GLIB_GNU_GETTEXT
 
 # Use AM_SILENT_RULES if present
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
@@ -21,5 +26,6 @@ AC_SUBST(GNUCFLAGS)
 # Linux is required for arbitrary baud rates
 AC_CHECK_HEADERS_ONCE([linux/serial.h])
 
+AC_CONFIG_FILES([Makefile src/Makefile po/Makefile.in])
 
 AC_OUTPUT
