diff -U 3 -H -d -p -r -N -- gtkterm-orig/src/serie.c gtkterm-src/src/serie.c
--- gtkterm-orig/src/serie.c	2012-09-12 13:59:23.822657028 +0200
+++ gtkterm-src/src/serie.c	2012-09-12 13:56:30.929321396 +0200
@@ -437,16 +437,38 @@ gint create_lockfile(char *port)
     int res;
     uid_t real_uid;
     gid_t real_gid;
+    int lockDirOk = 0;
 
     real_uid = getuid();
     real_gid = getgid();
     username = (getpwuid(real_uid))->pw_name;
 
     /* First see if the lock file directory is present. */
-    if(P_LOCK[0] && stat(P_LOCK, &stt) == 0)
-	snprintf(lockfile, sizeof(lockfile), "%s/LCK..%s", P_LOCK, mbasename(port, buf, sizeof(buf)));
+    if(P_LOCK[0])
+    {
+        if (stat(P_LOCK, &stt))
+        {
+            if (mkdir(P_LOCK, 0777) == 0)
+            {
+                lockDirOk = 1;
+            }
+        }
+        else if (S_ISDIR(stt.st_mode))
+        {
+            lockDirOk = 1;
+        }
+    }
+
+    if(lockDirOk)
+    {
+        snprintf(lockfile, sizeof(lockfile), "%s/LCK..%s", P_LOCK, mbasename(port, buf, sizeof(buf)));
+    }
     else
-	lockfile[0] = 0;
+    {
+        i18n_fprintf(stderr, _("Cannot create lockfile. Sorry.\n"));
+        lockfile[0] = 0;
+	return -1;
+    }
 
     if(lockfile[0] && (fd = open(lockfile, O_RDONLY)) >= 0)
     {
diff -U 3 -H -d -p -r -N -- gtkterm-orig/src/serie.h gtkterm-src/src/serie.h
--- gtkterm-orig/src/serie.h	2012-09-12 13:59:23.822657028 +0200
+++ gtkterm-src/src/serie.h	2012-09-12 13:23:11.815961517 +0200
@@ -33,7 +33,7 @@ gchar* get_port_string(void);
 #define BUFFER_EMISSION 4096
 #define LINE_FEED 0x0A
 #define POLL_DELAY 100               /* in ms (for control signals) */
-#define P_LOCK "/var/lock"           /* lock file location */
+#define P_LOCK "/var/lock/gtkterm"           /* lock file location */
 
 
 #endif
