diff -ru a/lib/php/src/ext/thrift_protocol/php_thrift_protocol.cpp b/lib/php/src/ext/thrift_protocol/php_thrift_protocol.cpp
--- a/lib/php/src/ext/thrift_protocol/php_thrift_protocol.cpp	2012-03-21 08:17:30.000000000 +0800
+++ b/lib/php/src/ext/thrift_protocol/php_thrift_protocol.cpp	2012-03-21 08:18:29.000000000 +0800
@@ -92,7 +92,7 @@
 #include "zend_exceptions.h"
 #include "php_thrift_protocol.h"
 
-static function_entry thrift_protocol_functions[] = {
+static zend_function_entry thrift_protocol_functions[] = {
   PHP_FE(thrift_protocol_write_binary, NULL)
   PHP_FE(thrift_protocol_read_binary, NULL)
   {NULL, NULL, NULL}
