--- patch-2.6.39-ck2.orig	2011-07-09 18:36:33.930129931 +0200
+++ patch-2.6.39-ck2	2011-07-09 18:37:03.227763663 +0200
@@ -9219,15 +9219,15 @@ Index: linux-2.6.39-ck2/init/calibrate.c
 ===================================================================
 --- linux-2.6.39-ck2.orig/init/calibrate.c	2011-05-19 16:29:17.000000000 +1000
 +++ linux-2.6.39-ck2/init/calibrate.c	2011-06-04 09:10:50.271373487 +1000
-@@ -208,7 +208,7 @@ void __cpuinit calibrate_delay(void)
+@@ -209,7 +209,7 @@ void __cpuinit calibrate_delay(void)
  	if (!printed)
  		pr_cont("%lu.%02lu BogoMIPS (lpj=%lu)\n",
- 			loops_per_jiffy/(500000/HZ),
--			(loops_per_jiffy/(5000/HZ)) % 100, loops_per_jiffy);
-+			(loops_per_jiffy * 10 /(50000/HZ)) % 100, loops_per_jiffy);
+ 			lpj/(500000/HZ),
+-			(lpj/(5000/HZ)) % 100, lpj);
++			(lpj * 10 / (50000/HZ)) % 100, lpj);
  
+ 	loops_per_jiffy = lpj;
  	printed = true;
- }
 Index: linux-2.6.39-ck2/kernel/Kconfig.preempt
 ===================================================================
 --- linux-2.6.39-ck2.orig/kernel/Kconfig.preempt	2009-06-10 13:05:27.000000000 +1000
