--- Makefile.orig	2008-09-23 11:49:43.000000000 -0400
+++ Makefile	2009-04-17 20:50:07.640427365 -0400
@@ -17,7 +17,7 @@
 ##  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 ##
 
-APP = jomar3D
+APP = jomar3d
 APPVERSION=1.1b
 
 EXEC = $(APP)
@@ -32,20 +32,20 @@
 VERSION=$(APPVERSION)$(SVNREV)
 
 DESTDIR = 
-BASEDIR = $(DESTDIR)/usr/local
-BINDIR  = $(BASEDIR)/games
-DATADIR = $(BASEDIR)/share/games/$(APP)
+BASEDIR = $(DESTDIR)/usr
+BINDIR  = $(BASEDIR)/bin
+DATADIR = $(BASEDIR)/share/$(APP)
 #DATADIR = .
 
 SOUNDDIR = $(DATADIR)/sound
 MUSICDIR = $(DATADIR)/music
 SKINDIR  = $(DATADIR)/skin
 LANGDIR  = $(DATADIR)/lang
-SCOREDIR = $(DESTDIR)/var/games
+SCOREDIR = ${pkgdir}/var/games
 #SCOREDIR = .
 
-DESKTOPDIR = $(BASEDIR)/share/applications
-ICONDIR    = $(BASEDIR)/share/pixmaps
+DESKTOPDIR = ${pkgdir}/$(BASEDIR)/share/applications
+ICONDIR    = ${pkgdir}/$(BASEDIR)/share/pixmaps
 
 SCOREFILE   = $(SCOREDIR)/$(APP)-scores
 DESKTOPFILE = $(DESKTOPDIR)/$(APP).desktop
@@ -130,48 +130,48 @@
 run: $(EXEC)
 	./"$(EXEC)"
 
-install: install-bin install-data install-skins install-score verif_path #list-files
+install: install-bin install-icon verif_path #list-files
 
 install-bin:
 	install -d "$(BINDIR)"
 	install -m 2755 -g games "$(EXEC)" "$(BINDIR)"
 
-install-data:
-	install -d "$(DATADIR)"
-	install -d "$(MUSICDIR)"
-	install -m 644 music/* "$(MUSICDIR)"
-	install -d "$(SOUNDDIR)"
-	install -m 644 sound/* "$(SOUNDDIR)"
-	install -d "$(LANGDIR)"
-	install -m 644 lang/*.jlg "$(LANGDIR)"
+#install-data:
+#	install -d "$(DATADIR)"
+#	install -d "$(MUSICDIR)"
+#	install -m 644 music/* "$(MUSICDIR)"
+#	install -d "$(SOUNDDIR)"
+#	install -m 644 sound/* "$(SOUNDDIR)"
+#	install -d "$(LANGDIR)"
+#	install -m 644 lang/*.jlg "$(LANGDIR)"
 
-install-desktop: install-icon
-	TEMPFILE=`tempfile`; \
-	cat "$(APP).desktop" \
-		| sed 's#$${EXEC}#$(BINDIR)/$(EXEC)#g' \
-		| sed 's#$${DATADIR}#$(DATADIR)#g' \
-		> $$TEMPFILE; \
-	install -d "$(DESKTOPDIR)"; \
-	install -m 644 $$TEMPFILE "$(DESKTOPFILE)"; \
-	rm -f $$TEMPFILE
+#install-desktop: install-icon
+#	TEMPFILE=`tempfile`; \
+#	cat "$(APP).desktop" \
+#		| sed 's#$${EXEC}#$(BINDIR)/$(EXEC)#g' \
+#		| sed 's#$${DATADIR}#$(DATADIR)#g' \
+#		> $$TEMPFILE; \
+#	install -d "$(DESKTOPDIR)"; \
+#	install -m 644 $$TEMPFILE "$(DESKTOPFILE)"; \
+#	rm -f $$TEMPFILE
 
 install-icon:
 	install -d "$(ICONDIR)"
 	install -m 644 "$(APP).svg" "$(ICONFILE)"
 
-install-skins: install-data
-	install -d "$(SKINDIR)"
-	cd skin && \
-	for skin in *.jsk; do \
-	  install -d "$(SKINDIR)"/$$skin; \
-	  install -m 644 $$skin/* "$(SKINDIR)"/$$skin; \
-	done
+#install-skins: install-data
+#	install -d "$(SKINDIR)"
+#	cd skin && \
+#	for skin in *.jsk; do \
+#	  install -d "$(SKINDIR)"/$$skin; \
+#	  install -m 644 $$skin/* "$(SKINDIR)"/$$skin; \
+#	done
 
-install-score:
-	install -d "$(SCOREDIR)"
-	touch "$(SCOREFILE)" && \
-	  chgrp games "$(SCOREFILE)" && \
-	  chmod 664 "$(SCOREFILE)"
+#install-score:
+#	install -d "$(SCOREDIR)"
+#	touch "$(SCOREFILE)" && \
+#	  chgrp games "$(SCOREFILE)" && \
+#	  chmod 664 "$(SCOREFILE)"
 
 # check if the istallation directory is in the user's PATH
 verif_path:
