--- Dofus/transition.conf	2014-06-30 11:34:19.000000000 -0400
+++ Dofus/transition.conf	2014-07-06 14:11:00.442349419 -0400
@@ -3,7 +3,7 @@
 game_name = "Dofus"
 update.game.id = "dofus2"
 
-update.projects = ["transition", "game"]
+update.projects = ["game"]
 
 if equals(sys.osname, "windows") {
      application_path = "${root}/Dofus.exe"
@@ -19,3 +19,15 @@
      include "Dofus.app/Contents/Resources/dofus.ici"
 }
 
+# Use the DBus service, which seems to work better, instead of sudo
+plugins.enable *= "dbusservice"
+plugins.enable *= "vlc"
+
+# Use adl instead of Air binaries on Arch Linux
+bypass_air_installation = true
+
+dofus.reg.path = "${root}/share/reg/bin/air-generic-launcher.sh"
+
+launcher.command = """
+"${root}/bin/air-generic-launcher.sh" --lang=${i18n.lang} --update-server-port=${service_port} --updater_version=v2
+"""
