--- src/Saaghar/saaghar.pro
+++ saaghar.pro
@@ -151,9 +151,9 @@
 
 unix:!macx {
 TARGET = saaghar
-LOG = $$system(mkdir ~/.Pojh)
-LOG += $$system(mkdir ~/.Pojh/Saaghar)
-LOG  += $$system(cp -n $${SOURCE_DIR}/utilities/* ~/.Pojh/Saaghar/)
+#LOG = $$system(mkdir ~/.Pojh)
+#LOG += $$system(mkdir ~/.Pojh/Saaghar)
+#LOG  += $$system(cp -n $${SOURCE_DIR}/utilities/* ~/.Pojh/Saaghar/)
 
 PREFIX = /usr
 DESKTOPDIR = $${PREFIX}/share/applications
