--- src/saaghar/SaagharWidget.cpp
+++ SaagharWidget.fix.cpp
@@ -451,11 +451,13 @@
 		QString verseData = QString::number(verses.at(i)->_PoemID)+"|"+QString::number(verses.at(i)->_Order)+"|"+QString::number((int)verses.at(i)->_Position);
 		mesraItem->setData(Qt::UserRole, "VerseData=|"+verseData);
 
+		bool spacerColumnIsPresent = false;
 		if (centeredView)
 		{
 			switch (verses.at(i)->_Position)
 			{
 				case Right :
+					spacerColumnIsPresent = true;
 					if (!flagEmptyThirdColumn)
 					{
 						firstEmptyThirdColumn = row;
@@ -473,6 +475,7 @@
 					break;
 
 				case Left :
+					spacerColumnIsPresent = true;
 					if (!flagEmptyThirdColumn)
 					{
 						firstEmptyThirdColumn = row;
@@ -531,6 +534,12 @@
 				default:
 					break;
 			}
+			if (spacerColumnIsPresent)
+			{
+				QTableWidgetItem *tmp = new QTableWidgetItem("");
+				tmp->setFlags(Qt::NoItemFlags);
+				tableViewWidget->setItem(row, 2, tmp);			
+			}
 		}// end of centeredView
 
 		QString simplifiedText = currentVerseText;
@@ -617,7 +626,7 @@
 	}
 	/////////////////////////////////////
 
-	setFlagOfEmptyColumn(tableViewWidget, firstEmptyThirdColumn);
+	//setFlagOfEmptyColumn(tableViewWidget, firstEmptyThirdColumn);
 	resizeTable(tableViewWidget);
 
 	QApplication::restoreOverrideCursor();
