diff --git a/calise/sun.py b/calise/sun.py
index cdc6dea..1f092fe 100644
--- a/calise/sun.py
+++ b/calise/sun.py
@@ -83,12 +83,17 @@ def getSun(latitude, longitude, curTime=None):
         try:
             set_dur = obs.next_setting(sun).datetime()
             set_dur = (
-                set_epoch - int(set_dur.strftime('%s')) +
+                set_epoch - int(set_dur.strftime('%s')) + 
                 time.timezone + dst)
             rise_dur = obs.next_rising(sun).datetime()
             rise_dur = (
                 int(rise_dur.strftime('%s')) - rise_epoch -
                 time.timezone - dst)
+            # TODO: the 4 rows below are a crappy workaround, need to be fixed
+            if set_dur < 0:
+                set_dur += 24 * 60 * 60
+            if rise_dur < 0:
+                rise_dur += 24 * 60 * 60
         except ephem.NeverUpError:
             set_dur = set_epoch - rise_epoch
             rise_dur = set_epoch - rise_epoch
