--- everborn.orig/Makefile	2012-02-22 12:31:42.967191244 +0600
+++ everborn/Makefile	2012-02-22 12:32:15.647191430 +0600
@@ -4,10 +4,10 @@
 
 OUT = ./everborn
 DUMP_DIR = DUMP/
-CFLAGS = -g `sdl-config --cflags` -Wno-write-strings
+CFLAGS += `sdl-config --cflags` -Wno-write-strings
 #LIBMIXER = /usr/lib/libSDL_mixer.a -lvorbisfile -lmikmod -lsmpeg
 LIBMIXER = -lSDL_mixer
-LIBS = -lGL -lm -lpng `sdl-config --libs` $(LIBMIXER)
+LIBS = -lGL -lm -lpng -lz `sdl-config --libs` $(LIBMIXER)
 
 default: flags $(OUT)
 
--- everborn.orig/data.cpp	2012-02-22 12:41:32.327194577 +0600
+++ everborn/data.cpp	2012-02-22 12:41:47.533861328 +0600
@@ -339,7 +339,7 @@
 }
 
 static envC envUserDir("userdir", "Everborn", dataCheckDirs);
-static envC envBaseDir("basedir", "/usr/share/Everborn", dataCheckDirs);
+static envC envBaseDir("basedir", "/usr/share/everborn", dataCheckDirs);
 
 
 
--- everborn.orig/pngio.cpp	2012-02-22 12:31:42.977191244 +0600
+++ everborn/pngio.cpp	2012-02-22 12:32:32.460524856 +0600
@@ -18,6 +18,7 @@
  the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  Boston, MA 02111-1307, USA.
 */
+#include <zlib.h>
 
 #include <png.h>
 #include <stdio.h>
