diff --git a/src/gifcreator.h b/src/gifcreator.h
index 3e1720e..9b9f659 100644
--- a/src/gifcreator.h
+++ b/src/gifcreator.h
@@ -31,6 +31,17 @@ using namespace std;
 typedef vector<GifByteType> Frame;
 typedef unsigned char Byte;
 
+static void
+PrintGifError()
+{
+	char *Err = GifErrorString();
+
+	if (Err != NULL)
+		fprintf(stderr, "\nGIF-LIB error: %s.\n", Err);
+	else
+		fprintf(stderr, "\nGIF-LIB undefined error %d.\n", GifError());
+}
+
 class GifCreator
 {
 
