diff -rupN libhl2-2.4.0/HLLib/BSPFile.cpp libhl2-2.4.0_mod/HLLib/BSPFile.cpp
--- libhl2-2.4.0/HLLib/BSPFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/BSPFile.cpp	2010-11-07 19:52:41.587193054 +0100
@@ -12,6 +12,7 @@
 #include "HLLib.h"
 #include "BSPFile.h"
 #include "Streams.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/DirectoryFile.cpp libhl2-2.4.0_mod/HLLib/DirectoryFile.cpp
--- libhl2-2.4.0/HLLib/DirectoryFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/DirectoryFile.cpp	2010-11-07 19:52:41.587193054 +0100
@@ -15,6 +15,7 @@
 #include "Streams.h"
 #include "Package.h"
 #include "Utility.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/DirectoryFolder.cpp libhl2-2.4.0_mod/HLLib/DirectoryFolder.cpp
--- libhl2-2.4.0/HLLib/DirectoryFolder.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/DirectoryFolder.cpp	2010-11-07 19:52:41.587193054 +0100
@@ -15,6 +15,7 @@
 #include "Utility.h"
 
 #include <algorithm>
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/DirectoryItem.cpp libhl2-2.4.0_mod/HLLib/DirectoryItem.cpp
--- libhl2-2.4.0/HLLib/DirectoryItem.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/DirectoryItem.cpp	2010-11-07 19:52:41.588192383 +0100
@@ -12,6 +12,7 @@
 #include "DirectoryItem.h"
 #include "DirectoryFile.h"
 #include "DirectoryFolder.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/Error.cpp libhl2-2.4.0_mod/HLLib/Error.cpp
--- libhl2-2.4.0/HLLib/Error.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/Error.cpp	2010-11-07 19:52:41.588192383 +0100
@@ -10,6 +10,9 @@
  */
 
 #include "Error.h"
+#include <cstring>
+#include <cctype>
+
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/FileMapping.cpp libhl2-2.4.0_mod/HLLib/FileMapping.cpp
--- libhl2-2.4.0/HLLib/FileMapping.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/FileMapping.cpp	2010-11-07 19:52:41.588192383 +0100
@@ -11,6 +11,7 @@
 
 #include "HLLib.h"
 #include "FileMapping.h"
+#include <cstring>
 
 using namespace HLLib;
 using namespace HLLib::Mapping;
diff -rupN libhl2-2.4.0/HLLib/FileStream.cpp libhl2-2.4.0_mod/HLLib/FileStream.cpp
--- libhl2-2.4.0/HLLib/FileStream.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/FileStream.cpp	2010-11-07 19:52:41.589192482 +0100
@@ -11,6 +11,7 @@
 
 #include "HLLib.h"
 #include "FileStream.h"
+#include <cstring>
 
 using namespace HLLib;
 using namespace HLLib::Streams;
diff -rupN libhl2-2.4.0/HLLib/GCFFile.cpp libhl2-2.4.0_mod/HLLib/GCFFile.cpp
--- libhl2-2.4.0/HLLib/GCFFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/GCFFile.cpp	2010-11-07 19:52:41.589192482 +0100
@@ -13,6 +13,7 @@
 #include "GCFFile.h"
 #include "Streams.h"
 #include "Checksum.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/GCFStream.cpp libhl2-2.4.0_mod/HLLib/GCFStream.cpp
--- libhl2-2.4.0/HLLib/GCFStream.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/GCFStream.cpp	2010-11-07 19:52:41.589192482 +0100
@@ -11,6 +11,7 @@
 
 #include "HLLib.h"
 #include "GCFStream.h"
+#include <cstring>
 
 using namespace HLLib;
 using namespace HLLib::Streams;
diff -rupN libhl2-2.4.0/HLLib/HLLib.cpp libhl2-2.4.0_mod/HLLib/HLLib.cpp
--- libhl2-2.4.0/HLLib/HLLib.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/HLLib.cpp	2010-11-07 19:52:41.589192482 +0100
@@ -10,6 +10,7 @@
  */
 
 #include "HLLib.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/MappingStream.cpp libhl2-2.4.0_mod/HLLib/MappingStream.cpp
--- libhl2-2.4.0/HLLib/MappingStream.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/MappingStream.cpp	2010-11-07 19:52:41.589192482 +0100
@@ -11,6 +11,7 @@
 
 #include "HLLib.h"
 #include "MappingStream.h"
+#include <cstring>
 
 using namespace HLLib;
 using namespace HLLib::Streams;
diff -rupN libhl2-2.4.0/HLLib/MemoryStream.cpp libhl2-2.4.0_mod/HLLib/MemoryStream.cpp
--- libhl2-2.4.0/HLLib/MemoryStream.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/MemoryStream.cpp	2010-11-07 19:52:41.590191815 +0100
@@ -11,6 +11,7 @@
 
 #include "HLLib.h"
 #include "MemoryStream.h"
+#include <cstring>
 
 using namespace HLLib;
 using namespace HLLib::Streams;
diff -rupN libhl2-2.4.0/HLLib/NCFFile.cpp libhl2-2.4.0_mod/HLLib/NCFFile.cpp
--- libhl2-2.4.0/HLLib/NCFFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/NCFFile.cpp	2010-11-07 19:52:41.590191815 +0100
@@ -14,6 +14,7 @@
 #include "Streams.h"
 #include "Checksum.h"
 #include "Utility.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/PAKFile.cpp libhl2-2.4.0_mod/HLLib/PAKFile.cpp
--- libhl2-2.4.0/HLLib/PAKFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/PAKFile.cpp	2010-11-07 19:52:41.590191815 +0100
@@ -12,6 +12,7 @@
 #include "HLLib.h"
 #include "PAKFile.h"
 #include "Streams.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/Utility.cpp libhl2-2.4.0_mod/HLLib/Utility.cpp
--- libhl2-2.4.0/HLLib/Utility.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/Utility.cpp	2010-11-07 19:52:41.590191815 +0100
@@ -10,6 +10,7 @@
  */
 
 #include "Utility.h"
+#include <cstring>
 
 hlBool HLLib::GetFileExists(const hlChar *lpPath)
 {
diff -rupN libhl2-2.4.0/HLLib/VBSPFile.cpp libhl2-2.4.0_mod/HLLib/VBSPFile.cpp
--- libhl2-2.4.0/HLLib/VBSPFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/VBSPFile.cpp	2010-11-07 19:52:41.590191815 +0100
@@ -13,6 +13,7 @@
 #include "VBSPFile.h"
 #include "Streams.h"
 #include "Checksum.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/VPKFile.cpp libhl2-2.4.0_mod/HLLib/VPKFile.cpp
--- libhl2-2.4.0/HLLib/VPKFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/VPKFile.cpp	2010-11-07 19:52:41.591191563 +0100
@@ -14,6 +14,7 @@
 #include "Mappings.h"
 #include "Streams.h"
 #include "Checksum.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/WADFile.cpp libhl2-2.4.0_mod/HLLib/WADFile.cpp
--- libhl2-2.4.0/HLLib/WADFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/WADFile.cpp	2010-11-07 19:52:41.591191563 +0100
@@ -12,6 +12,7 @@
 #include "HLLib.h"
 #include "WADFile.h"
 #include "Streams.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/Wrapper.cpp libhl2-2.4.0_mod/HLLib/Wrapper.cpp
--- libhl2-2.4.0/HLLib/Wrapper.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/Wrapper.cpp	2010-11-07 19:52:41.591191563 +0100
@@ -15,6 +15,7 @@
 #include "Streams.h"
 #include "Packages.h"
 #include "Wrapper.h"
+#include <cstring>
 
 using namespace HLLib;
 using namespace HLLib::Streams;
diff -rupN libhl2-2.4.0/HLLib/XZPFile.cpp libhl2-2.4.0_mod/HLLib/XZPFile.cpp
--- libhl2-2.4.0/HLLib/XZPFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/XZPFile.cpp	2010-11-07 19:52:41.591191563 +0100
@@ -12,6 +12,7 @@
 #include "HLLib.h"
 #include "XZPFile.h"
 #include "Streams.h"
+#include <cstring>
 
 using namespace HLLib;
 
diff -rupN libhl2-2.4.0/HLLib/ZIPFile.cpp libhl2-2.4.0_mod/HLLib/ZIPFile.cpp
--- libhl2-2.4.0/HLLib/ZIPFile.cpp	2010-11-07 19:42:53.000000000 +0100
+++ libhl2-2.4.0_mod/HLLib/ZIPFile.cpp	2010-11-07 19:52:41.591191563 +0100
@@ -13,6 +13,7 @@
 #include "ZIPFile.h"
 #include "Streams.h"
 #include "Checksum.h"
+#include <cstring>
 
 using namespace HLLib;
 
