---
 bsect.c |   21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

Index: lilo-22.7/bsect.c
===================================================================
--- lilo-22.7.orig/bsect.c
+++ lilo-22.7/bsect.c
@@ -1058,16 +1058,19 @@ if (image) { /* long section specific to
       "read-write")) strcat(options,"rw ");
     if ((root = cfg_get_strg(cf_kernel,"root")) || (root = cfg_get_strg(
       cf_options,"root")))  {
-	if (!strcasecmp(root,"current")) {
-	    if (stat("/",&st) < 0) pdie("stat /");
-	    sprintf(strchr(options,0),"root=%x ",(unsigned int) st.st_dev);
-	}
-	else if (strlen(root)>6 && !strncmp(root,"LABEL=",6)) {
+	    /* root= is a property of either:
+	     * prepare_namespace()
+	     * or /init in initramfs
+	     * or /linuxrc in a loopmounted initrd
+	     * Its not a property of the bootloader or anything else
+	     * related to loading the kernel/initrd into memory (or configuring
+	     * that event)
+	     *
+	     * Pass root= via /proc/cmdline as configured in the config file
+	     * If no root= is configured, initrd has a hardcoded value
+	     * based on the '/' entry in /etc/fstab
+	     */
 	    sprintf(strchr(options,0),"root=%s ", root);
-	}
-	else {
-	    sprintf(strchr(options,0),"root=%x ",dev_number(root));
-	}
       }	
     if ((ram_disk = cfg_get_strg(cf_kernel,"ramdisk")) || (ram_disk =
       cfg_get_strg(cf_options,"ramdisk")))
