--- lilo-22.7/crt.S
+++ lilo-22.7/crt.S
@@ -689,9 +689,9 @@
 	.ascii	"LILO"
 #ifdef LCF_VERSION
 	.ascii	" "	
-	.ascii	S(VERSION_MAJOR)
+	.ascii	SA(VERSION_MAJOR)
 	.ascii	"."
-	.ascii	S(VERSION_MINOR)
+	.ascii	SA(VERSION_MINOR)
 	.ascii	VERSION_EDIT
 #endif
 	.ascii	"  Boot Menu"
--- lilo-22.7/lilo.c
+++ lilo-22.7/lilo.c
@@ -214,7 +214,7 @@
 #if __MSDOS__
 	    printf("%s\n", name
 #else /* !__MSDOS__ */
-	    fprintf(stdout,"%s%-" S(MAX_IMAGE_NAME) "s %s%s",verbose > 0 ? "  " : "",name,
+	    fprintf(stdout,"%s%-" SA(MAX_IMAGE_NAME) "s %s%s",verbose > 0 ? "  " : "",name,
 	      image ? "" : "*",
 #ifdef LCF_VIRTUAL
 	      descrs.d.descr[image].flags & FLAG_VMDEFAULT ? "@" :
--- lilo-22.7/lilo.h
+++ lilo-22.7/lilo.h
@@ -70,7 +70,7 @@
 
 #define nelem(a) (sizeof(a)/sizeof((a)[0]))
 #define S2(x) #x
-#define S(x) S2(x)
+#define SA(x) S2(x)
 
 #include "version.h"
 #define VERSION 256*VERSION_MINOR+VERSION_MAJOR
--- lilo-22.7/probe.c
+++ lilo-22.7/probe.c
@@ -264,7 +264,7 @@
 		needed==4 ? "22.0" :
 		needed==5 ? "22.5.1" :
 		needed==6 ? "22.5.7" :
-		S(VERSION_MAJOR) "." S(VERSION_MINOR) VERSION_EDIT  );
+		SA(VERSION_MAJOR) "." SA(VERSION_MINOR) VERSION_EDIT  );
 	return 1;
     }
     return 0;
--- lilo-22.7/second.S
+++ lilo-22.7/second.S
@@ -3922,9 +3922,9 @@
 #endif
 #ifdef LCF_VERSION
 	.ascii	" "	
-	.ascii	S(VERSION_MAJOR)
+	.ascii	SA(VERSION_MAJOR)
 	.ascii	"."
-	.ascii	S(VERSION_MINOR)
+	.ascii	SA(VERSION_MINOR)
 	.ascii	VERSION_EDIT
 #endif
 	.byte	32,0
