diff -urNp lilo-22.7.orig/QuickInst lilo-22.7/QuickInst
--- lilo-22.7.orig/QuickInst	2003-12-15 05:49:37.000000000 +0100
+++ lilo-22.7/QuickInst	2005-10-05 16:11:35.000000000 +0200
@@ -816,11 +816,15 @@ Type  !cat $CFG_FILE  if you want to che
 EOF
 if yesno "Test LILO installation now ?" yes; then
     echo
-    outfile=${TMPDIR:-/tmp}/out$$
-    errflag=${TMPDIR:-/tmp}/err$$
-    if [ -f $errflag ]; then
-	rm $errflag
-    fi
+    tmpbase=${TMPDIR:-/tmp}/lilo_quickinstall
+    rm -rf $tmpbase
+    mkdir $tmpbase || {
+	echo "Could not create temporary directory."
+	exit 1
+    }
+
+    outfile=$tmpbase/out$$
+    errflag=$tmpbase/err$$
     (
 	if
 	    $SBIN_DIR/lilo -t 2>&1
@@ -859,7 +863,7 @@ EOF
 	grep '^Warning: ' <$outfile
 	echo
     fi
-    rm $outfile
+    rm -rf $tmpbase
     echo
     if yesno "Install LILO now ?" yes; then
 	cat <<EOF
