Author: Andreas Beckmann <anbe@debian.org>
Description: make silent commands in NVIDIA's Makefile.kbuild verbose

--- a/nvidia-modules-common.mk
+++ b/nvidia-modules-common.mk
@@ -133,10 +133,10 @@
 #
 
 $(obj)/$(CORE_OBJS):
-	@cp $(src)/$(CORE_OBJS) $(obj)/$(CORE_OBJS)
+	$(NVQ)cp $(src)/$(CORE_OBJS) $(obj)/$(CORE_OBJS)
 
 $(obj)/$(VERSION_HEADER):
-	@echo \#define NV_COMPILER \"`$(CC) -v 2>&1 | tail -n 1`\" > $@
+	$(NVQ)echo \#define NV_COMPILER \"`$(CC) -v 2>&1 | tail -n 1`\" > $@
 
 CONFTEST_COMPILE_TEST_HEADERS := $(obj)/conftest/macros.h \
 $(obj)/conftest/functions.h $(obj)/conftest/symbols.h $(obj)/conftest/types.h \
@@ -148,10 +148,10 @@
 $(obj)/conftest.h $(CONFTEST_COMPILE_TEST_HEADERS): conftest-compile-test
 
 conftest-compile-test: $(src)/conftest.sh
-	@if ! $(CONFTEST) compile_tests $(COMPILE_TESTS); then exit 1; fi
+	$(NVQ)$(CONFTEST) compile_tests $(COMPILE_TESTS)
 
 $(obj)/conftest/patches.h:
-	@if ! $(CONFTEST) patch_check; then exit 1; fi
+	$(NVQ)$(CONFTEST) patch_check
 
 MODULE_GLUE_TARGETS = $(addprefix $(obj)/,$(MODULE_GLUE_OBJS))
 
@@ -180,10 +180,10 @@
 #
 
 suser-sanity-check:
-	@if ! $(CONFTEST) suser_sanity_check; then exit 1; fi
+	$(NVQ)$(CONFTEST) suser_sanity_check
 
 rmmod-sanity-check:
-	@if ! $(CONFTEST) rmmod_sanity_check; then exit 1; fi
+	$(NVQ)$(CONFTEST) rmmod_sanity_check
 
 BUILD_SANITY_CHECKS = \
 	cc_version_check \
@@ -194,7 +194,7 @@
 	preempt_rt_sanity_check
 
 build-sanity-checks:
-	@for SANITY_CHECK in $(BUILD_SANITY_CHECKS); do \
+	$(NVQ)for SANITY_CHECK in $(BUILD_SANITY_CHECKS); do \
 	 if ! $(CONFTEST) $$SANITY_CHECK full_output; then \
 	 exit 1; \
 	 fi; \
@@ -215,10 +215,10 @@
 define BUILD_MODULE_RULE
  module: $(1)
  $(1): build-sanity-checks $(3)
-	@echo "NVIDIA: calling KBUILD..."; \
-	$$(MAKE) "CC=$$(CC)" NV_MODULE_SUFFIX=$$(strip $(2)) $$(KBUILD_PARAMS) modules; \
-	echo "NVIDIA: left KBUILD."; \
-	if ! [ -f $(1) ]; then \
+	@echo "NVIDIA: calling KBUILD..."
+	$$(NVQ)$$(MAKE) "CC=$$(CC)" NV_MODULE_SUFFIX=$$(strip $(2)) $$(KBUILD_PARAMS) modules
+	@echo "NVIDIA: left KBUILD."
+	$$(NVQ)if ! [ -f $(1) ]; then \
 	 echo "$(1) failed to build!"; \
 	 exit 1; \
 	elif [ -f split-object-file.sh ]; then \
@@ -233,7 +233,7 @@
 #
 
 module-sign: $(MODULE_OBJECT)
-	@if [ -z "$(MODSECKEY)" ] || [ -z "$(MODPUBKEY)" ]; then \
+	$(NVQ)if [ -z "$(MODSECKEY)" ] || [ -z "$(MODPUBKEY)" ]; then \
 	 echo "ERROR: MODSECKEY and MODPUBKEY must be set to sign modules!"; \
 	 exit 1; \
 	elif ! [ -f "$(MODSECKEY)" ]; then \
@@ -265,7 +265,7 @@
 #
 
 module-install: suser-sanity-check module
-	@mkdir -p $(MODULE_ROOT)/video; \
+	$(NVQ)mkdir -p $(MODULE_ROOT)/video; \
 	install -m 0664 -o root -g root $(MODULE_OBJECT) $(MODULE_ROOT)/video; \
 	PATH="$(PATH):/bin:/sbin" depmod -a
 
@@ -275,7 +275,7 @@
 #
 
 package-install: module-install rmmod-sanity-check
-	@PATH="$(PATH):/bin:/sbin" modprobe $(MODULE_NAME) && \
+	$(NVQ)PATH="$(PATH):/bin:/sbin" modprobe $(MODULE_NAME) && \
 	echo "$(MODULE_OBJECT) installed successfully.";
 
 #
@@ -300,20 +300,20 @@
 #
 
 $(KERNEL_GLUE_NAME): $(MODULE_OBJECT)
-	@$(LD) $(EXTRA_LDFLAGS) $(MODULE_COMMON_SCRIPT) -r -o $(KERNEL_GLUE_NAME) $(KERNEL_GLUE_OBJS)
+	$(NVQ)$(LD) $(EXTRA_LDFLAGS) $(MODULE_COMMON_SCRIPT) -r -o $(KERNEL_GLUE_NAME) $(KERNEL_GLUE_OBJS)
 
 #
 # Support hack, KBUILD isn't prepared to clean up after external modules.
 #
 
 clean:
-	@$(RM) -f $(MODULE_GLUE_OBJS) $(KERNEL_GLUE_OBJS)
-	@$(RM) -f build-in.o nv-linux*.o *.d .*.cmd .*.flags
-	@$(RM) -f $(MODULE_NAME)*.o $(MODULE_NAME)*.ko*
-	@$(RM) -f $(MODULE_NAME)*.mod* $(VERSION_HEADER) *~
-	@$(RM) -f conftest*.c conftest.h
-	@$(RM) -rf conftest
-	@$(RM) -rf Module*.symvers .tmp_versions modules.order
+	$(NVQ)$(RM) -f $(MODULE_GLUE_OBJS) $(KERNEL_GLUE_OBJS)
+	$(NVQ)$(RM) -f build-in.o nv-linux*.o *.d .*.cmd .*.flags
+	$(NVQ)$(RM) -f $(MODULE_NAME)*.o $(MODULE_NAME)*.ko*
+	$(NVQ)$(RM) -f $(MODULE_NAME)*.mod* $(VERSION_HEADER) *~
+	$(NVQ)$(RM) -f conftest*.c conftest.h
+	$(NVQ)$(RM) -rf conftest
+	$(NVQ)$(RM) -rf Module*.symvers .tmp_versions modules.order
 
 #
 # This target just prints the kernel module filename (for use by the
