diff -Naur a/nv-linux.h b/nv-linux.h
--- a/nv-linux.h	2013-12-03 23:24:48.484495874 +0100
+++ b/nv-linux.h	2013-12-03 23:27:44.684030888 +0100
@@ -392,8 +392,13 @@
 #endif
 
 #if defined(NVCPU_X86) || defined(NVCPU_X86_64)
+#if 0
 #define CACHE_FLUSH()  asm volatile("wbinvd":::"memory")
 #define WRITE_COMBINE_FLUSH() asm volatile("sfence":::"memory")
+#else
+#define CACHE_FLUSH()
+#define WRITE_COMBINE_FLUSH() asm volatile("sfence":::"memory")
+#endif
 #elif defined(NVCPU_ARM)
 #define CACHE_FLUSH() cpu_cache.flush_kern_all()
 #define WRITE_COMBINE_FLUSH()   \
diff -Naur a/nv-pat.c b/nv-pat.c
--- a/nv-pat.c	2013-12-03 23:24:33.987007640 +0100
+++ b/nv-pat.c	2013-12-03 23:26:57.615744800 +0100
@@ -34,7 +34,9 @@
 {
     unsigned long cr0 = read_cr0();
     write_cr0(((cr0 & (0xdfffffff)) | 0x40000000));
+#if 0
     wbinvd();
+#endif
     *cr4 = read_cr4();
     if (*cr4 & 0x80) write_cr4(*cr4 & ~0x80);
     __flush_tlb();
@@ -43,7 +45,9 @@
 static inline void nv_enable_caches(unsigned long cr4)
 {
     unsigned long cr0 = read_cr0();
+#if 0
     wbinvd();
+#endif
     __flush_tlb();
     write_cr0((cr0 & 0x9fffffff));
     if (cr4 & 0x80) write_cr4(cr4);
