From 51038e7e8da0f48c8a0340f8cea1effd565398bb Mon Sep 17 00:00:00 2001
From: Sven Schober <sven.schober@uni-ulm.de>
Date: Wed, 25 May 2011 10:38:46 +0200
Subject: [PATCH] fix loader location

---
 wuala |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/wuala b/wuala
index 7eacf1e..e2f89cb 100755
--- a/wuala
+++ b/wuala
@@ -9,7 +9,7 @@ function launchWuala() {
   fi
   echo $JAVA
   pushd "$(dirname $0)" > /dev/null
-  $JAVA -ea -Xmx384m -Djava.net.preferIPv4Stack=true -jar loader3.jar -installed "$@"
+  $JAVA -ea -Xmx384m -Djava.net.preferIPv4Stack=true -jar /usr/share/java/wuala/loader3.jar -installed "$@"
   popd > /dev/null
 }
 
-- 
1.7.5.2

