diff -ru pylms-1.0.orig/bin/pylms pylms-1.0/bin/pylms
--- pylms-1.0.orig/bin/pylms	2012-03-17 12:23:10.000000000 +0100
+++ pylms-1.0/bin/pylms	2012-06-23 15:41:15.618470810 +0200
@@ -16,7 +16,7 @@
 VERSION = "1.00"
 
 logger = logging.getLogger(__name__)
-LOG_HELP = ','.join(["%d=%s" % (4-x, logging.getLevelName((x+1)*10)) for x in xrange(5)])
+LOG_HELP = ','.join(["%d=%s" % (4-x, logging.getLevelName((x+1)*10)) for x in range(5)])
 LOG_FORMAT_CONS = '%(levelname)s: %(message)s'
 LOG_FORMAT_FILE = '%(asctime)s %(name)s[%(process)d] %(levelname)10s %(message)s'
 LOGLEVEL_DICT = { 1 : 50, 2:40, 3:20, 4:10, 5:1 }
@@ -59,7 +59,7 @@
 else:
     logfilename = os.path.normpath(options.logfile)
     logging.basicConfig(level=verbosity, format=LOG_FORMAT_FILE, filename=logfilename, filemode='a')
-    print >> sys.stderr, "Logging to %s" % logfilename
+    print("Logging to %s" % logfilename, file=sys.stderr)
 
 # Main
 
@@ -124,7 +124,7 @@
             result = func(*params)
             if options.raw:
                 if result is not None:
-                    print result
+                    print(result)
             else:
                 if result is None:
                     result = "No Output"
@@ -145,7 +145,7 @@
                     result = func(*params)
                     if options.raw:
                         if result is not None:
-                            print result
+                            print(result)
                     else:
                         if result is None:
                             result = "No Output"
diff -ru pylms-1.0.orig/pylms/tests.py pylms-1.0/pylms/tests.py
--- pylms-1.0.orig/pylms/tests.py	2012-03-17 12:25:49.000000000 +0100
+++ pylms-1.0/pylms/tests.py	2012-06-23 15:40:23.354843481 +0200
@@ -3,13 +3,13 @@
 s = server.Server("10.0.2.10")
 s.connect()
 
-print s.get_players()
+print(s.get_players())
 
 p = s.get_player("Lounge")
 p.set_volume(10)
 
 r = s.request("songinfo 0 100 track_id:94")
-print r
+print(r)
 
 r = s.request("trackstat getrating 1019")
-print r
+print(r)
