diff --git a/npiet.c b/npiet.c
index 3b2affa..dd4f527 100644
--- a/npiet.c
+++ b/npiet.c
@@ -1203,6 +1203,19 @@ read_png (char *fname)
 
 #include <gif_lib.h>
 
+/* Compatibility with giflib >= 5.1 (they like to break the API...) */
+int _DGifCloseFile (struct GifFileType * gif)
+{
+  int ret;
+#if defined GIFLIB_MAJOR && GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+  int errorcode;
+  ret = DGifCloseFile(gif, &errorcode);
+#else
+  ret = DGifCloseFile(gif);
+#endif
+  return ret;
+}
+
 #if defined GIFLIB_MAJOR && GIFLIB_MAJOR >= 5
 static void PrintGifErrorStr(char *str)
 {
@@ -1247,14 +1260,14 @@ read_gif (char *fname)
 #else
     PrintGifError ();
 #endif
-    DGifCloseFile (gif);
+    _DGifCloseFile (gif);
     return -1;
   }
 
   if (rtype != IMAGE_DESC_RECORD_TYPE 
       || DGifGetImageDesc (gif) == GIF_ERROR) {
     fprintf (stderr, "error: unknown gif format - exiting\n");
-    DGifCloseFile (gif);
+    _DGifCloseFile (gif);
     exit (-1);
   }
 
@@ -1320,7 +1333,7 @@ read_gif (char *fname)
     }
   }
 
-  DGifCloseFile (gif);
+  _DGifCloseFile (gif);
 
   return 0;
 }
