--- aegisub/Makefile.target.orig	2012-01-02 15:45:24.896646968 +0100
+++ aegisub/Makefile.target	2012-01-02 16:14:39.693373687 +0100
@@ -66,30 +66,30 @@
 
 install-program:
 ifdef PROGRAM_INSTALL
-	@$(BIN_MKDIR) -p $(P_BINDIR)
-	@$(BIN_INSTALL) $(PROGRAM) $(P_BINDIR)/$(PROGRAM)
+	@$(BIN_MKDIR) -p $(DESTDIR)$(P_BINDIR)
+	@$(BIN_INSTALL) $(PROGRAM) $(DESTDIR)$(P_BINDIR)/$(PROGRAM)
 endif
 
 install-lib:
 ifdef LIB_SHARED_INSTALL
-	$(BIN_MKDIR) -p $(P_LIB)
-	$(BIN_INSTALL) $(LIB_SHARED_FULL) $(P_LIB)/$(LIB_SHARED_FULL)
-	$(BIN_LN) -sf $(LIB_SHARED_FULL) $(P_LIB)/$(LIB_SHARED)
+	$(BIN_MKDIR) -p $(DESTDIR)$(P_LIB)
+	$(BIN_INSTALL) $(LIB_SHARED_FULL) $(DESTDIR)$(P_LIB)/$(LIB_SHARED_FULL)
+	$(BIN_LN) -sf $(LIB_SHARED_FULL) $(DESTDIR)$(P_LIB)/$(LIB_SHARED)
 endif
 
 install-share:
 ifdef DATA_SHARE
-	@$(BIN_MKDIR) -p $(addprefix $(P_DATA)/, $(dir $(DATA_SHARE)))
+	@$(BIN_MKDIR) -p $(addprefix $(DESTDIR)$(P_DATA)/, $(dir $(DATA_SHARE)))
 	@for i in $(DATA_SHARE); do \
-		$(BIN_ECHO) "$(BIN_INSTALL) $$i $(P_DATA)/$$i"; \
-		$(BIN_INSTALL) $$i $(P_DATA)/$$i; \
+		$(BIN_ECHO) "$(BIN_INSTALL) $$i $(DESTDIR)$(P_DATA)/$$i"; \
+		$(BIN_INSTALL) $$i $(DESTDIR)$(P_DATA)/$$i; \
 	done
 endif
 ifdef DATA_SHARE_DOC
-	@$(BIN_MKDIR) -p $(addprefix $(P_DOC)/, $(dir $(DATA_SHARE_DOC)))
+	@$(BIN_MKDIR) -p $(addprefix $(DESTDIR)$(P_DOC)/, $(dir $(DATA_SHARE_DOC)))
 	@for i in $(DATA_SHARE_DOC); do \
-		$(BIN_ECHO) "$(BIN_INSTALL) $$i $(P_DOC)/$$i"; \
-		$(BIN_INSTALL) $$i $(P_DOC)/$$i; \
+		$(BIN_ECHO) "$(BIN_INSTALL) $$i $(DESTDIR)$(P_DOC)/$$i"; \
+		$(BIN_INSTALL) $$i $(DESTDIR)$(P_DOC)/$$i; \
 	done
 endif
 
