--- yafray.old/linux-settings.py	2006-07-14 12:14:52.000000000 +0200
+++ yafray/linux-settings.py	2006-07-20 14:38:26.000000000 +0200
@@ -5,13 +5,15 @@
 
 srcroot = os.getcwd()
 prefix = ''
+libdif = ''
 
 def init(args): 
-	global prefix
+	global prefix, libdir
 	prefix = args.get('prefix','/usr/local')
+        libdir = args.get('libdir','/lib')
 
-def get_libpath(args): return prefix+"/lib"
-def get_pluginpath(args): return prefix+"/lib/yafray"
+def get_libpath(args): return prefix+libdir
+def get_pluginpath(args): return prefix+libdir+"/yafray"
 def get_binpath(args): return prefix+"/bin"
 def get_confpath(args): return prefix+"/etc"
 def get_cxxflags(args):
@@ -40,7 +42,7 @@
 			return False
 		
 	def get_include(args): return [ exr.PATH + "/include/OpenEXR" ]
-	def get_libpath(args): return [ exr.PATH + "/lib" ]
+	def get_libpath(args): return [ exr.PATH + libdir ]
 	def get_libs(args): return ['IlmImf', 'Imath', 'Iex', 'Half']
 
 class jpeg(globalinfo.library):
