fix bug #137603

--- linux-settings.py.orig	2008-01-01 15:36:57.000000000 +0100
+++ linux-settings.py	2008-01-01 15:37:10.000000000 +0100
@@ -15,7 +15,7 @@
 def get_libpath(args): return prefix+libdir
 def get_pluginpath(args): return prefix+libdir+"/yafray"
 def get_binpath(args): return prefix+"/bin"
-def get_confpath(args): return prefix+"/etc"
+def get_confpath(args): return "/etc"
 def get_cxxflags(args):
 	debug=args.get('debug',0)
 	arch=args.get('arch','')
