--- wscript	2012-09-06 16:50:37.542366041 +0800
+++ wscript-new	2012-09-06 16:50:33.465699329 +0800
@@ -122,10 +122,10 @@
 	bld(rule='${ADOC} -a icons=true -a stylesheet=${SRC[1].abspath()} -a iconsdir=. -a toc -d book -o ${TGT} ${SRC[0].abspath()}',
 		source='waf.txt waf.css', target='single.html', scan=ascii_doc_scan)
 
-	bld(rule='${A2X} -L -a toc --icons-dir=. --icons -D ${gen.path.get_bld().abspath()} -d book -f pdf ${SRC}',
+	bld(rule='${A2X} -L -a toc --icons-dir=. --icons -D ${gen.path.get_bld().abspath()} -d book -f pdf --dblatex-opts "-s ${SRC[0].parent.abspath()}/asciidoc-dblatex.sty" --dblatex-opts "-p ${SRC[0].parent.abspath()}/asciidoc-dblatex.xsl" ${SRC}',
 		source='waf.txt', target='waf.pdf', scan=ascii_doc_scan)
 
-	bld(rule='${A2X} -L -a toc --icons-dir=. --icons -D ${gen.path.get_bld().abspath()} -d article -f pdf ${SRC}',
+	bld(rule='${A2X} -L -a toc --icons-dir=. --icons -D ${gen.path.get_bld().abspath()} -d article -f pdf --dblatex-opts "-s ${SRC[0].parent.abspath()}/asciidoc-dblatex.sty" --dblatex-opts "-p ${SRC[0].parent.abspath()}/asciidoc-dblatex.xsl" ${SRC}',
 		source='intro_waf_1.6.txt', target='intro_waf_1.6.pdf', scan=ascii_doc_scan)
 
 	bld(rule='ln -sf single.html index.html', shell=True)
