--- conf.py	2012-10-16 23:46:22.608811721 +0800
+++ conf-new.py	2012-10-16 23:46:11.355476878 +0800
@@ -401,7 +401,7 @@
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
-#html_theme_options = {}
+html_theme_options = {'collapsiblesidebar': True}
 
 # Add any paths that contain custom themes here, relative to this directory.
 #html_theme_path = []
