--- esteidpkcs11loader-3.7.2.1043/chrome/content/pkcs11-loader.js	2013-08-15 16:52:27.000000000 +0300
+++ esteidpkcs11loader-3.7.2.1043.1/chrome/content/pkcs11-loader.js	2013-10-22 13:35:11.395521022 +0300
@@ -27,7 +27,7 @@
 const nsLocalFile = "@mozilla.org/file/local;1";
 const nsPKCS11 = "@mozilla.org/security/pkcs11;1";
 
-const dllName = "onepin-opensc-pkcs11";
+const dllName = "opensc-pkcs11";
 
 var Ci = Components.interfaces;
 var Cc = Components.classes;
@@ -68,7 +68,7 @@
   if (platform == "Macintosh") {
     moduleDll = "/Library/OpenSC/lib/" + dllName + ".so";
   } else if (platform == "X11") {
-    var libDir = (bits == 64) ? "/usr/lib64/" : "/usr/lib/";
+    var libDir = "/usr/lib/";
     moduleDll = libDir + dllName + ".so";
   } else if (platform == "Windows") {
     moduleDll = dllName + ".dll";
@@ -87,7 +87,7 @@
     if (module) {
       if (module.libName == moduleDll && module.name == EstEidModName) {
         needToLoad = false;
-      } else if (module.name == EstEidModName || /(opensc-pkcs11|esteid-pkcs11)\.(so|dll)$/.test(module.libName)) {
+      } else if (module.name == EstEidModName || /(onepin-opensc-pkcs11|esteid-pkcs11)\.(so|dll)$/.test(module.libName)) {
         removeModule(module)
       }
     }
