diff -rupN lldb-orig/lib/Makefile lldb-new/lib/Makefile
--- lldb-orig/lib/Makefile	2013-07-11 14:19:52.506941589 +0200
+++ lldb-new/lib/Makefile	2013-07-11 14:20:38.020906061 +0200
@@ -17,7 +17,7 @@ NO_BUILD_ARCHIVE = 1
 LINK_LIBS_IN_SHARED = 1
 SHARED_LIBRARY = 1
 
-PYTHON_BUILD_FLAGS = $(shell python-config --ldflags)
+PYTHON_BUILD_FLAGS = $(shell python2-config --ldflags)
 
 # Include all archives in liblldb.so file
 USEDLIBS = lldbAPI.a \
diff -rupN lldb-orig/Makefile lldb-new/Makefile
--- lldb-orig/Makefile	2013-07-11 14:19:52.563609041 +0200
+++ lldb-new/Makefile	2013-07-11 14:20:27.110754740 +0200
@@ -29,7 +29,7 @@ LEVEL := $(LLDB_LEVEL)/../..
 include $(LEVEL)/Makefile.common
 
 # Set Python include directory
-PYTHON_INC_DIR = $(shell python-config --includes)
+PYTHON_INC_DIR = $(shell python2-config --includes)
 # Set common LLDB build flags.
 CPP.Flags += -I$(PROJ_SRC_DIR)/$(LLDB_LEVEL)/include 
 CPP.Flags += -I$(PROJ_OBJ_DIR)/$(LLDB_LEVEL)/include
