--- php-gtk-build/Makefile
+++ Makefile
@@ -12,7 +12,7 @@
 RE2C = exit 0;
 AWK = gawk
 AWK = gawk
-PHP = /usr/bin/php
+PHP = /usr/bin/php -d 'open_basedir='
 CFLAGS_CLEAN = $(CFLAGS)
 GTK_LIBVERSION = 2.22
 PHP_GTK2_SHARED_LIBADD = -pthread -lgobject-2.0 -lgthread-2.0 -lrt -lglib-2.0 -pthread -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lm -lcairo -lpng14 -lpango-1.0 -lfreetype -lfontconfig -lgobject-2.0 -lgmodule-2.0 -lgthread-2.0 -lrt -lglib-2.0 -pthread -latk-1.0 -lgobject-2.0 -lgthread-2.0 -lrt -lglib-2.0 -pthread -lpango-1.0 -lgobject-2.0 -lgmodule-2.0 -lgthread-2.0 -lrt -lglib-2.0 -lcairo -pthread -lglade-2.0 -lgtk-x11-2.0 -lxml2 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lm -lcairo -lpng14 -lpango-1.0 -lfreetype -lfontconfig -lgobject-2.0 -lgmodule-2.0 -lgthread-2.0 -lrt -lglib-2.0

