--- knetworkmanager.conf	2010-09-09 22:29:59.000000000 +0200
+++ knetworkmanager.conf.patched	2012-04-13 02:29:48.726606532 +0200
@@ -29,19 +29,6 @@
 		<deny send_destination="org.freedesktop.NetworkManagerUserSettings"
 		      send_interface="org.freedesktop.NetworkManagerSettings.Connection.Secrets"/>
 	</policy>
-	<policy group="netdev">
-		<allow own="org.freedesktop.NetworkManagerUserSettings"/>
-
-		<allow send_destination="org.freedesktop.NetworkManagerUserSettings"
-		       send_interface="org.freedesktop.NetworkManagerSettings"/>
-
-		<allow send_destination="org.freedesktop.NetworkManagerUserSettings"
-		       send_interface="org.freedesktop.NetworkManagerSettings.Connection"/>
-
-		<!-- Only root can get secrets -->
-		<deny send_destination="org.freedesktop.NetworkManagerUserSettings"
-		      send_interface="org.freedesktop.NetworkManagerSettings.Connection.Secrets"/>
-	</policy>
 	<policy group="users">
 		<allow own="org.freedesktop.NetworkManagerInfo"/>
 
