--- httpbin/setup.py	2012-11-05 00:09:21.423337195 -0700
+++ httpbin-work/setup.py	2012-11-05 00:10:23.328324958 -0700
@@ -16,12 +16,12 @@
 
 
 deps = [
-    'Flask==0.8',
-    'Flask-Script==0.3.1',
-    'Jinja2==2.5.5',
-    'Werkzeug==0.8.1',
-    'distribute==0.6.15',
-    'decorator==3.3.2',
+    'Flask>=0.8',
+    'Flask-Script>=0.3.1',
+    'Jinja2>=2.5.5',
+    'Werkzeug>=0.8.1',
+    'distribute>=0.6.15',
+    'decorator>=3.3.2',
     'gevent'
 ]
 
