From a561fa1aebcbaca27fa2081fdec0a6a4c7c43237 Mon Sep 17 00:00:00 2001
From: graysky <graysky@archlinux.us>
Date: Mon, 12 Jan 2015 17:22:41 -0500
Subject: [PATCH] allow INSTALL_PREFIX to work

---
 Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index 7c886bd..9e958f3 100644
--- a/Makefile
+++ b/Makefile
@@ -152,17 +152,17 @@ strip:
 	$(CROSS_COMPILE)strip $(MODULE_NAME).ko --strip-unneeded
 
 install:
-	install -d $(DESTDIR)$(MODDESTDIR)
-	install -m644 $(MODULE_NAME).ko  $(DESTDIR)$(MODDESTDIR)
-	install -d $(DESTDIR)/lib/firmware/rtlwifi
-	install -m644 rtl8192dufw.bin $(DESTDIR)/lib/firmware/rtlwifi
-	install -m644 rtl8192dufw_wol.bin $(DESTDIR)/lib/firmware/rtlwifi
+	install -d $(DESTDIR)$(INSTALL_PREFIX)$(MODDESTDIR)
+	install -m644 $(MODULE_NAME).ko  $(DESTDIR)$(INSTALL_PREFIX)$(MODDESTDIR)
+	install -d $(DESTDIR)$(INSTALL_PREFIX)/lib/firmware/rtlwifi
+	install -m644 rtl8192dufw.bin $(DESTDIR)$(INSTALL_PREFIX)/lib/firmware/rtlwifi
+	install -m644 rtl8192dufw_wol.bin $(DESTDIR)$(INSTALL_PREFIX)/lib/firmware/rtlwifi
 	depmod -a
 
 uninstall:
-	rm -f $(DESTDIR)$(MODDESTDIR)/$(MODULE_NAME).ko
-	rm -f $(DESTDIR)/lib/firmware/rtlwifi/rtl8192dufw.bin
-	rm -f $(DESTDIR)/lib/firmware/rtlwifi/rtl8192dufw_wol.bin
+	rm -f $(DESTDIR)$(INSTALL_PREFIX)$(MODDESTDIR)/$(MODULE_NAME).ko
+	rm -f $(DESTDIR)$(INSTALL_PREFIX)/lib/firmware/rtlwifi/rtl8192dufw.bin
+	rm -f $(DESTDIR)$(INSTALL_PREFIX)/lib/firmware/rtlwifi/rtl8192dufw_wol.bin
 	depmod -a
 
 config_r:
