From e517391cf4b17db5e88fc80be4fa1bd4500241dd Mon Sep 17 00:00:00 2001
From: William J Bowman <bluephoenix47@gmail.com>
Date: Sat, 10 Mar 2012 02:59:04 -0500
Subject: [PATCH] Fixed a declaration difference

---
 gnuwrapper.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnuwrapper.cpp b/gnuwrapper.cpp
index cc8314f..f2fa398 100644
--- a/gnuwrapper.cpp
+++ b/gnuwrapper.cpp
@@ -39,7 +39,7 @@ extern "C" {
   static void *(*old_realloc_hook)(void *ptr, size_t size, const void *caller);
   static void *(*old_memalign_hook)(size_t alignment, size_t size, const void *caller);
 
-  void (*__malloc_initialize_hook) (void) = my_init_hook;
+  void (* volatile __malloc_initialize_hook) (void) = my_init_hook;
 
   static void my_init_hook (void) {
     // Store the old hooks.
-- 
1.7.9.3

