diff -ru pida.orig/pida/editors/emacs/emacs.py pida/pida/editors/emacs/emacs.py
--- pida.orig/pida/editors/emacs/emacs.py	2012-03-10 02:57:31.000000000 -0800
+++ pida/pida/editors/emacs/emacs.py	2012-03-10 03:00:18.000000000 -0800
@@ -294,13 +294,6 @@
     @classmethod
     def get_sanity_errors(cls):
         errors = []
-        from pida.core.pdbus import has_dbus
-        if not has_dbus:
-            errors = [
-                'dbus python disfunctional',
-                'please repair the python dbus bindings',
-                '(note that it won\'t work for root)'
-            ]
 
         try:
             import subprocess
diff -ru pida.orig/pida/ui/completer.py pida/pida/ui/completer.py
--- pida.orig/pida/ui/completer.py	2012-03-10 02:57:31.000000000 -0800
+++ pida/pida/ui/completer.py	2012-03-10 03:01:02.000000000 -0800
@@ -69,7 +69,7 @@
    
     gsignal('user-abort', str, flags=gobject.SIGNAL_RUN_LAST)
     gsignal('user-accept', str, flags=gobject.SIGNAL_RUN_LAST)
-    gsignal('user-abort', str, flags=gobject.SIGNAL_RUN_LAST)
+    gsignal('suggestion-selected', str, flags=gobject.SIGNAL_RUN_LAST)
 
 
     def __init__(self, show_input=True, show_icons=True,
