--- configure.ac.orig	2008-11-16 20:34:03.218250704 +0800
+++ configure.ac	2008-11-16 20:34:15.141583676 +0800
@@ -25,7 +25,7 @@
       AC_HELP_STRING([--disable-alsa], [Don't use alsa sound system]),
       enable_alsa=$enableval, enable_alsa=yes)
 if test x$enable_alsa != "xno" ; then
-    AM_PATH_ALSA(0.9.0, AC_DEFINE(HAVE_ALSA,1,[Define if ALSA is available]))
+    true
 fi
 
 # Checks for header files.
