--- agent.py	2009-12-11 14:01:02.000000000 +0100
+++ agent_new.py	2009-12-22 19:44:10.000000000 +0100
@@ -41,7 +41,7 @@
 	path = os.path.dirname(path)
 	
 	config = ConfigParser.ConfigParser()
-	config.read(path + '/config.cfg')
+	config.read('/etc/sd-agent/config.cfg')
 	
 	# Core config
 	agentConfig['sdUrl'] = config.get('Main', 'sd_url')
@@ -348,4 +348,4 @@
 		
 	else:
 		print 'usage: %s start|stop|restart|status|update' % sys.argv[0]
-		sys.exit(2)
\ No newline at end of file
+		sys.exit(2)
