--- qtdso-0.3.1/src/dsowid.h	2006-11-11 09:14:58.000000000 +0100
+++ ../qtdso-0.3.1/src/dsowid.h	2008-06-27 23:28:40.000000000 +0200
@@ -1,4 +1,4 @@
-//======================================================================
+ //======================================================================
 // File:		dsowid.h
 // Author:	Matthias Toussaint
 // Created:	Sun Jun  9 22:27:14 CEST 2002
@@ -24,7 +24,7 @@
 #include <qwidget.h>
 #include <qpixmap.h>
 #include <type.h>
-
+ #include <stdlib.h>
 #include <fftw3.h>
 
 #include <dso.h>
