--- libvdpau-0.5/configure	2012-09-04 14:26:38.000000000 -0300
+++ libvdpau-0.5/configure.new	2012-09-09 03:39:38.733475434 -0300
@@ -15295,42 +15295,8 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for XEXT" >&5
 $as_echo_n "checking for XEXT... " >&6; }
 
-if test -n "$XEXT_CFLAGS"; then
-    pkg_cv_XEXT_CFLAGS="$XEXT_CFLAGS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"xext\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "xext") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_XEXT_CFLAGS=`$PKG_CONFIG --cflags "xext" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-if test -n "$XEXT_LIBS"; then
-    pkg_cv_XEXT_LIBS="$XEXT_LIBS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"xext\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "xext") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_XEXT_LIBS=`$PKG_CONFIG --libs "xext" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-
-
+# Disabled XEXT in lib32-libvdpau to follow libvdpau official package.
+pkg_failed=no
 
 if test $pkg_failed = yes; then
    	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
