$NetBSD: patch-aa,v 1.2 2009/01/06 18:54:51 kefren Exp $
--- SConstruct.orig	2007-10-26 12:11:04.000000000 +0300
+++ SConstruct	2008-04-24 21:34:37.000000000 +0300
@@ -12,7 +12,7 @@
 opts.Add(PathOption('LIBDIR', 'System library path', '$PREFIX/lib'));
 opts.Add(PathOption('BINDIR', 'System executable path', '$PREFIX/bin'));
 opts.Add(PathOption('SHAREDIR', 'System shared data path', '$PREFIX/share'));
-opts.Add(PathOption('SYSCONFDIR', 'System configuration path', '/etc'));
+opts.Add(PathOption('SYSCONFDIR', 'System configuration path', '$PREFIX/share'));
 opts.Add(PathOption('DESTDIR', 'System base installation path', '/'));
 
 opts.Add(BoolOption('DEBUG', 'Compile with debug information', 0));
@@ -179,7 +179,7 @@
 	env['RESOURCE'] = Split('')
 	env['CSCFLAGS'] = Split('')
 
-env = Environment(options = opts)
+env = Environment(ENV = os.environ, options = opts)
 conf = Configure(env, custom_tests = { 'CheckPKGConfig' : CheckPKGConfig,
                                        'CheckPKG' : CheckPKG })
 Help(opts.GenerateHelpText(env))
$NetBSD: patch-ab,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/General/VisibleColumnsWindow.cs.orig	2008-04-24 12:24:49.000000000 +0300
+++ gtk/src/General/VisibleColumnsWindow.cs	2007-10-26 12:11:04.000000000 +0300
@@ -69,12 +69,12 @@
 			CellRendererToggle toggleRenderer = new CellRendererToggle();
 			toggleRenderer.Toggled += OnColumnToggled;
 			column = treeView.AppendColumn(null, toggleRenderer);
-			column.SetCellDataFunc(toggleRenderer, SetToggleColumn);
+			column.SetCellDataFunc(toggleRenderer, (Gtk.TreeCellDataFunc)SetToggleColumn);
 
 			CellRendererText textRenderer = new CellRendererText();
 			column = treeView.AppendColumn(Catalog.GetString("Name"),
 				textRenderer);
-			column.SetCellDataFunc(textRenderer, SetNameColumn);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)SetNameColumn);
 
 			// Model
 			store = new ListStore(typeof(TreeViewColumn));
@@ -90,7 +90,7 @@
 
 		#region Methods
 
-		public static void ConnectToWidget(Widget widget, Action action,
+		public static void ConnectToWidget(Widget widget, Gtk.Action action,
 			TreeViewColumn[] columns, params TreeViewColumn[] inactiveColumns)
 		{
 			if (widget == null)
$NetBSD: patch-ac,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/HubWindow.cs.orig	2008-04-24 12:12:20.000000000 +0300
+++ gtk/src/HubWindow.cs	2007-10-26 12:11:04.000000000 +0300
@@ -122,12 +122,12 @@
 
 			CellRendererPixbuf pixbufRenderer = new CellRendererPixbuf();
 			column.PackStart(pixbufRenderer , false);
-			column.SetCellDataFunc(pixbufRenderer, SetPixbuf);
+			column.SetCellDataFunc(pixbufRenderer, (Gtk.TreeCellDataFunc)SetPixbuf);
 
 			CellRendererText textRenderer = new CellRendererText();
 			textRenderer.Ellipsize = Pango.EllipsizeMode.End;
 			column.PackStart(textRenderer , true);
-			column.SetCellDataFunc(textRenderer, SetName);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)SetName);
 
 			// Events
 			addHubButton.Clicked += OnAddHubButtonClicked;
$NetBSD: patch-ad,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/MainWindow.cs.old	2008-04-24 12:17:45.000000000 +0300
+++ gtk/src/MainWindow.cs	2007-10-26 12:11:04.000000000 +0300
@@ -45,11 +45,11 @@
 		private Page currentPage;
 		private ActionGroup pageActions;
 		private uint mergeId;
-		private Dictionary<MenuItem, Action> itemToAction;
+		private Dictionary<MenuItem, Gtk.Action> itemToAction;
 
 		private UIManager uim;
 		private ActionGroup actionGroup;
-		private Action closeAction;
+		private Gtk.Action closeAction;
 		private ToggleAction toolbarVisibleAction;
 		private RadioAction layoutTabsAction;
 
@@ -85,7 +85,7 @@
 			}
 			this.favoriteHubManager = favoriteHubManager;
 
-			itemToAction = new Dictionary<MenuItem, Action>();
+			itemToAction = new Dictionary<MenuItem, Gtk.Action>();
 
 			Window.Title = Util.AppName;
 
@@ -399,7 +399,7 @@
 
 		private void OnMenuItemSelected(object obj, EventArgs args)
 		{
-			Action action;
+			Gtk.Action action;
 			if (itemToAction.TryGetValue((MenuItem)obj, out action) &&
 				action.Tooltip != null)
 			{
$NetBSD: patch-ae,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/Page/DownloadPage.cs.old	2008-04-24 11:22:07.000000000 +0300
+++ gtk/src/Page/DownloadPage.cs	2007-10-26 12:11:04.000000000 +0300
@@ -36,10 +36,10 @@
 		private MessageArea messageArea;
 
 		private ActionGroup actionGroup;
-		private Action startAction;
-		private Action stopAction;
-		private Action removeAction;
-		private Action clearAction;
+		private Gtk.Action startAction;
+		private Gtk.Action stopAction;
+		private Gtk.Action removeAction;
+		private Gtk.Action clearAction;
 
 		private DownloadStore downloadStore;
 		private TreePath virtualRoot;
$NetBSD: patch-af,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/Page/PageManagerTree.cs.old	2008-04-24 12:23:26.000000000 +0300
+++ gtk/src/Page/PageManagerTree.cs	2007-10-26 12:11:04.000000000 +0300
@@ -117,12 +117,12 @@
 
 			pixbufRenderer = new CellRendererPixbuf();
 			column.PackStart(pixbufRenderer , false);
-			column.SetCellDataFunc(pixbufRenderer, PixbufDataHandler);
+			column.SetCellDataFunc(pixbufRenderer, (Gtk.TreeCellDataFunc)PixbufDataHandler);
 
 			textRenderer = new CellRendererText();
 			textRenderer.Ellipsize = Pango.EllipsizeMode.End;
 			column.PackStart(textRenderer , true);
-			column.SetCellDataFunc(textRenderer, TextDataHandler);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)TextDataHandler);
 
 			// Events
 			favoriteHubManager.HubsSorted += OnFavoriteHubsSorted;
$NetBSD: patch-ag,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/Page/SearchPage.cs.old	2008-04-24 12:26:20.000000000 +0300
+++ gtk/src/Page/SearchPage.cs	2007-10-26 12:11:04.000000000 +0300
@@ -150,7 +150,7 @@
 				{
 					if (renderer is CellRendererText)
 					{
-						column.SetCellDataFunc(renderer, SetTextStyle);
+						column.SetCellDataFunc(renderer, (Gtk.TreeCellDataFunc)SetTextStyle);
 					}
 				}
 			}
@@ -468,7 +468,7 @@
 			item.Show();
 
 			// Search by TTH
-			Action searchAction = actionGroup["SearchTTH"];
+			Gtk.Action searchAction = actionGroup["SearchTTH"];
 			searchAction.Sensitive = searchButton.Sensitive && GetSelectedTTH() != null;
 			menu.Append(searchAction.CreateMenuItem());
 		}
$NetBSD: patch-ah,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/Page/UploadPage.cs.old	2008-04-24 11:22:21.000000000 +0300
+++ gtk/src/Page/UploadPage.cs	2007-10-26 12:11:04.000000000 +0300
@@ -40,9 +40,9 @@
 		private uint timeout;
 
 		private ActionGroup actionGroup;
-		private Action stopAction;
-		private Action removeAction;
-		private Action clearAction;
+		private Gtk.Action stopAction;
+		private Gtk.Action removeAction;
+		private Gtk.Action clearAction;
 
 		[Widget]
 		private ScrolledWindow scrolledWindow;
$NetBSD: patch-ai,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/View/DownloadView.cs.old	2008-04-24 12:22:29.000000000 +0300
+++ gtk/src/View/DownloadView.cs	2007-10-26 12:11:04.000000000 +0300
@@ -76,7 +76,7 @@
 			column = GetColumn("Progress");
 			progressRenderer = (CellRendererProgress)column.CellRenderers[0];
 
-			column.SetCellDataFunc(progressRenderer, SetProgressVisible);
+			column.SetCellDataFunc(progressRenderer, (Gtk.TreeCellDataFunc)SetProgressVisible);
 		}
 
 		#region Methods
$NetBSD: patch-aj,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/View/FileListView.cs	2008-04-24 12:17:17.000000000 +0300
+++ gtk/src/View/FileListView.cs	2007-10-26 12:11:04.000000000 +0300
@@ -68,7 +68,7 @@
 
 			pixbufRenderer = new CellRendererPixbuf();
 			column.PackStart(pixbufRenderer , false);
-			column.SetCellDataFunc(pixbufRenderer, SetPixbuf);
+			column.SetCellDataFunc(pixbufRenderer, (Gtk.TreeCellDataFunc)SetPixbuf);
 
 			textRenderer = new CellRendererText();
 			column.PackStart(textRenderer , true);
$NetBSD: patch-ak,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/View/SearchResultView.cs.old	2008-04-24 12:20:56.000000000 +0300
+++ gtk/src/View/SearchResultView.cs	2007-10-26 12:11:04.000000000 +0300
@@ -30,10 +30,10 @@
 	public class SearchResultView : ExtendedTreeView
 	{
 		private ActionGroup actionGroup;
-		private Action downloadAction;
-		private Action saveAsAction;
-		private Action fileListAction;
-		private Action matchQueueAction;
+		private Gtk.Action downloadAction;
+		private Gtk.Action saveAsAction;
+		private Gtk.Action fileListAction;
+		private Gtk.Action matchQueueAction;
 
 		#region Constructors
 
@@ -86,7 +86,7 @@
 
 			pixbufRenderer = new CellRendererPixbuf();
 			column.PackStart(pixbufRenderer , false);
-			column.SetCellDataFunc(pixbufRenderer, SetPixbuf);
+			column.SetCellDataFunc(pixbufRenderer, (Gtk.TreeCellDataFunc)SetPixbuf);
 
 			textRenderer = new CellRendererText();
 			textRenderer.Ellipsize = Pango.EllipsizeMode.End;
$NetBSD: patch-al,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/View/TransferView.cs.old	2008-04-24 12:18:11.000000000 +0300
+++ gtk/src/View/TransferView.cs	2007-10-26 12:11:04.000000000 +0300
@@ -61,14 +61,14 @@
 
 			pixbufRenderer = new CellRendererPixbuf();
 			column.PackStart(pixbufRenderer, false);
-			column.SetCellDataFunc(pixbufRenderer, SetPixbuf);
+			column.SetCellDataFunc(pixbufRenderer, (Gtk.TreeCellDataFunc)SetPixbuf);
 
 			textRenderer = new CellRendererText();
 			textRenderer.Ellipsize = Pango.EllipsizeMode.End;
 			column.PackStart(textRenderer, true);
 			column.AddAttribute(textRenderer, "text",
 				(int)TransferStoreColumn.Name);
-			column.SetCellDataFunc(textRenderer, SetTextStyle);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)SetTextStyle);
 
 			ExpanderColumn = column;
 
@@ -77,7 +77,7 @@
 			column = AppendColumn(Catalog.GetString("Size"), textRenderer,
 				"text", (int)TransferStoreColumn.SizeText);
 			column.SortColumnId = (int)TransferStoreColumn.Size;
-			column.SetCellDataFunc(textRenderer, SetTextStyle);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)SetTextStyle);
 
 			// Status
 			textRenderer = new CellRendererText();
@@ -85,7 +85,7 @@
 			column = AppendColumn(Catalog.GetString("Status"), textRenderer);
 			column.Resizable = true;
 			column.MinWidth = 100;
-			column.SetCellDataFunc(textRenderer, SetStatusText);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)SetStatusText);
 
 			// Progress
 			progressRenderer = new CellRendererProgress();
@@ -102,7 +102,7 @@
 			column = AppendColumn(Catalog.GetString("Time Left"), textRenderer,
 				"text", (int)TransferStoreColumn.TimeLeft);
 			column.SortColumnId = (int)TransferStoreColumn.TimeLeft;
-			column.SetCellDataFunc(textRenderer, SetTextStyle);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)SetTextStyle);
 		}
 
 		#region Methods
$NetBSD: patch-am,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/View/UploadView.cs.old	2008-04-24 12:15:50.000000000 +0300
+++ gtk/src/View/UploadView.cs	2007-10-26 12:11:04.000000000 +0300
@@ -39,7 +39,7 @@
 				"text", (int)UploadStore.Column.Started);
 			column = GetColumn(0);
 			column.SortColumnId = (int)UploadStore.Column.Started;
-			column.SetCellDataFunc(textRenderer, SetTextStyle);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)SetTextStyle);
 
 			// Directory
 			textRenderer = new CellRendererText();
@@ -50,7 +50,7 @@
 			column.Expand = true;
 			column.Resizable = true;
 			column.SortColumnId = (int)TransferStoreColumn.Directory;
-			column.SetCellDataFunc(textRenderer, SetTextStyle);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)SetTextStyle);
 
 			// User
 			textRenderer = new CellRendererText();
@@ -59,7 +59,7 @@
 			column = GetColumn(3);
 			column.Resizable = true;
 			column.SortColumnId = (int)UploadStore.Column.User;
-			column.SetCellDataFunc(textRenderer, SetTextStyle);
+			column.SetCellDataFunc(textRenderer, (Gtk.TreeCellDataFunc)SetTextStyle);
 		}
 
 		#region Methods
$NetBSD: patch-an,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/src/View/UserView.cs.old	2008-04-24 12:21:54.000000000 +0300
+++ gtk/src/View/UserView.cs	2007-10-26 12:11:04.000000000 +0300
@@ -68,7 +68,7 @@
 
 			pixbufRenderer = new CellRendererPixbuf();
 			column.PackStart(pixbufRenderer , false);
-			column.SetCellDataFunc(pixbufRenderer, RenderUserPixbuf);
+			column.SetCellDataFunc(pixbufRenderer, (Gtk.TreeCellDataFunc)RenderUserPixbuf);
 
 			textRenderer = new CellRendererText();
 			textRenderer.Ellipsize = Pango.EllipsizeMode.End;
@@ -140,9 +140,9 @@
 
 		protected override void OnPopulatePopup(Menu menu)
 		{
-			Action fileListAction = actionGroup["FileList"];
-			Action messageAction = actionGroup["PrivateMessage"];
-			Action infoAction = actionGroup["Information"];
+			Gtk.Action fileListAction = actionGroup["FileList"];
+			Gtk.Action messageAction = actionGroup["PrivateMessage"];
+			Gtk.Action infoAction = actionGroup["Information"];
 
 			TreeIter iter;
 			if (Selection.GetSelected(out iter))
$NetBSD: patch-ao,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- gtk/SConscript.orig	2008-04-25 09:23:52.000000000 +0300
+++ gtk/SConscript	2008-04-25 09:25:00.000000000 +0300
@@ -100,8 +100,8 @@
 
 env.Alias("install", env.InstallProgram(lib_dir, '#bin/DCSharp.exe'))
 env.Alias("install", env.InstallData(lib_dir, '#bin/DCSharp.exe.config'))
-env.Alias("install", env.InstallLibrary(lib_dir, '#lib/NDesk.DBus.dll'))
-env.Alias("install", env.InstallLibrary(lib_dir, '#lib/NDesk.DBus.GLib.dll'))
-env.Alias("install", env.InstallData(lib_dir, '#lib/NDesk.DBus.GLib.dll.config'))
+# env.Alias("install", env.InstallLibrary(lib_dir, '#lib/NDesk.DBus.dll'))
+# env.Alias("install", env.InstallLibrary(lib_dir, '#lib/NDesk.DBus.GLib.dll'))
+# env.Alias("install", env.InstallData(lib_dir, '#lib/NDesk.DBus.GLib.dll.config'))
 
 SConscript('po/SConscript')
$NetBSD: patch-ap,v 1.1.1.1 2008/04/25 16:10:27 kefren Exp $
--- bin/dcsharp.in.orig	2009-06-01 22:02:09.808221668 +0200
+++ bin/dcsharp.in	2009-06-01 22:02:18.054711697 +0200
@@ -7,4 +7,4 @@
 export MONO_PATH="$MONO_PATH:$LIB_DIR/dcsharp"
 export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$LIB_DIR/dcsharp"
 
-exec mono "$LIB_DIR/dcsharp/DCSharp.exe" "$@"
+exec dbus-launch mono "$LIB_DIR/dcsharp/DCSharp.exe" "$@"
