From de51e19bcb88672251b69baeff2e0ad62e17918e Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Tue, 30 Aug 2011 19:44:20 +0000
Subject: Fix gstreamer initialization

---
diff --git a/LongoMatch/Common/GStreamer.cs b/LongoMatch/Common/GStreamer.cs
index 49ad1ce..999ea3f 100644
--- a/LongoMatch/Common/GStreamer.cs
+++ b/LongoMatch/Common/GStreamer.cs
@@ -27,7 +27,7 @@ namespace LongoMatch.Common
 	{
 		
 		[DllImport("libgstreamer-0.10.dll") /* willfully unmapped */ ]
-		static extern void gst_init (string argv);
+		static extern void gst_init (int argc, string argv);
 		[DllImport("libgstreamer-0.10.dll") /* willfully unmapped */ ]
 		static extern IntPtr gst_registry_get_default ();
 		[DllImport("libgstreamer-0.10.dll") /* willfully unmapped */ ]
@@ -40,7 +40,7 @@ namespace LongoMatch.Common
 		
 		public static void Init() {
 			SetUpEnvironment ();
-			gst_init ("");
+			gst_init (0, "");
 		}
 		
 		public static bool CheckInstallation () {
--
cgit v0.9.0.2
