diff --git a/src/habak.c b/src/habak.c
index 69d1b6d..4c6b391 100644
--- a/src/habak.c
+++ b/src/habak.c
@@ -111,6 +111,10 @@ Active by default.\n\
 Implies \"-mC\".\n\
 Inactive by default.\n\
 \n\
+-mW -- Habak will be proportionally stretched to fill the screen horizontally.\n\
+Implies \"-mC\".\n\
+Inactive by default.\n\
+\n\
 -ms -- Habak will be non-proportionally stretched to fill the screen.\n\
 Overrides and can be overriden by \"-mp\".\n\
 Inactive by default.\n\
@@ -206,6 +210,10 @@ for (i=1;i<argc;i++) {
 		d("parse scale");
 		template_habak.scale_art=MOD_SCALE;
 	}
+	else if (!strcmp(argv[i],"-mW")) {
+		d("parse width scale");
+		template_habak.scale_art=MOD_SCALE_WIDTH;
+	}
 	else if (!strcmp(argv[i],"-ms")) {
 		d("parse uglyscale");
 		template_habak.scale_art=MOD_SCALE_NOPROP;
@@ -404,6 +412,12 @@ switch (current_list_item->data.type) {
 		current_list_item->data.height=current_list_item->data.orig_height*yco;
 		current_list_item->data.x = (scr->width-current_list_item->data.width)/2;
 		current_list_item->data.y = (scr->height-current_list_item->data.height)/2;
+		} else if (current_list_item->data.scale_art==MOD_SCALE_WIDTH) {
+		xco=((double)scr->width/(double)current_list_item->data.orig_width);
+		yco=xco;
+		current_list_item->data.width=current_list_item->data.orig_width*xco;
+		current_list_item->data.height=current_list_item->data.orig_height*yco;
+		current_list_item->data.x = (scr->width-current_list_item->data.width)/2;
 		} else {
 		/* brzydkie skalowanie 
 		 * W sumie to mog³em je robiæ wcze¶niej, ale dorzucam je tutaj,
diff --git a/src/types.h b/src/types.h
index 07c69af..e152b3c 100644
--- a/src/types.h
+++ b/src/types.h
@@ -10,6 +10,7 @@
 #define MOD_NOCENTER 0
 #define MOD_NOSCALE 0
 #define MOD_SCALE 'S'
+#define MOD_SCALE_WIDTH 'W'
 #define MOD_SCALE_NOPROP 's'
 #define MOD_PAINT 'c'
 #define MOD_POSITION 'p'
