diff --git a/neo/framework/Licensee.h b/neo/framework/Licensee.h
index fe15c76..ecbcba2 100644
--- a/neo/framework/Licensee.h
+++ b/neo/framework/Licensee.h
@@ -95,9 +95,9 @@ If you have questions concerning this license or the applicable additional terms
 
 // Linux info
 #ifdef ID_DEMO_BUILD
-	#define LINUX_DEFAULT_PATH			"/usr/local/games/dhewm3-demo"
+	#define LINUX_DEFAULT_PATH			"/usr/share/doom3-demo"
 #else
-	#define LINUX_DEFAULT_PATH			"/usr/local/games/dhewm3"
+	#define LINUX_DEFAULT_PATH			"/usr/share/doom3"
 #endif
 
 // CD Key file info
