From fc721e66beb7d2000f1b6f8bdb81bf1f90aabf35 Mon Sep 17 00:00:00 2001
From: August Z. Flatby <august@shortcut.no>
Date: Mon, 22 Jun 2009 17:15:32 +0200
Subject: [PATCH] Fix for ruby 1.9: Not all exception classes support '+' in 1.9. This causes problems with Errno::EPIPE in reconnect scenarios.

---
 lib/stomp/connection.rb |   20 ++++++++++----------
 1 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/lib/stomp/connection.rb b/lib/stomp/connection.rb
index ddfd4c3..0b537c6 100644
--- a/lib/stomp/connection.rb
+++ b/lib/stomp/connection.rb
@@ -55,18 +55,18 @@ module Stomp
           @failure = nil
           begin
             s = TCPSocket.open @host, @port
-	    headers = @connect_headers.clone
-	    headers[:login] = @login
-	    headers[:passcode] = @passcode
+            headers = @connect_headers.clone
+            headers[:login] = @login
+            headers[:passcode] = @passcode
             _transmit(s, "CONNECT", headers)
             @connect = _receive(s)
             # replay any subscriptions.
             @subscriptions.each { |k,v| _transmit(s, "SUBSCRIBE", v) }
           rescue
-            @failure = $!;
-            s=nil;
+            @failure = $!
+            s = nil
             raise unless @reliable
-            $stderr.print "connect failed: " + $! +" will retry in #{@reconnect_delay}\n";
+            $stderr.print "connect failed: #{$!}. will retry in #{@reconnect_delay}\n";
             sleep(@reconnect_delay);
           end
         end
@@ -165,9 +165,9 @@ module Stomp
           s = socket
           return _receive(s)
         rescue
-          @failure = $!;
+          @failure = $!
           raise unless @reliable
-          $stderr.print "receive failed: " + $!;
+          $stderr.print "receive failed: #{$!}"
         end
       end
     end
@@ -231,9 +231,9 @@ module Stomp
             _transmit(s, command, headers, body)
             return
           rescue
-            @failure = $!;
+            @failure = $!
             raise unless @reliable
-            $stderr.print "transmit failed: " + $!+"\n";
+            $stderr.print "transmit failed: #{$i}\n"
           end
         end
       end
-- 
1.6.1

