--- open-dis-3.0.orig//Compile/premake/premake_gnumake_linux.sh	2008-02-22 21:39:54.000000000 +0100
+++ open-dis-3.0//Compile/premake/premake_gnumake_linux.sh	2010-08-29 05:42:10.306668344 +0200
@@ -4,6 +4,6 @@ pushd Projects
 #export HAWKNL_INC_DIR="where the headers for HawkNL exist."
 #export HAWKNL_LIB_DIR="where the library for HawkNL exists."
 premake --file DIS.lua --target gnu
-premake --file DIS.lua --target vs2003
-premake --file DIS.lua --target vs2005
+#premake --file DIS.lua --target vs2003
+#premake --file DIS.lua --target vs2005
 popd
