--- shutdown-dialog.py.orig	2010-08-31 14:17:02.000000000 -0700
+++ shutdown-dialog.py	2010-09-02 22:40:32.000000000 -0700
@@ -147,7 +147,7 @@
                 subprocess.Popen("kill -9 " + str(proc.pid), shell = True)
 
         if status != 0:
-            subprocess.Popen("gksudo -- " + cmd, shell = True)
+            subprocess.Popen("gksu -- " + cmd, shell = True)
 
         self.terminate()
 
