--- Makefile.orig	2013-07-27 14:07:34.098426303 +0200
+++ Makefile	2013-07-27 14:06:41.828943776 +0200
@@ -98,7 +98,7 @@
 	install --directory $(DESTDIR)/$(GPR_DIR)
 	sed template_for_installed_project \
           $(foreach var,LIB_NAME SRC_DIR ALI_DIR LDLIBS LIB_DIR, \
-          -e 's/$$($(var))/$(subst $(space),"$(comma) ",$($(var)))/g') \
+          -e 's#$$($(var))#$(subst $(space),"$(comma) ",$($(var)))#g') \
           > $(DESTDIR)/$(GPR_DIR)/$(LIB_NAME).gpr
 	chmod 644 $(DESTDIR)/$(GPR_DIR)/$(LIB_NAME).gpr
 	install --directory $(DESTDIR)/$(ALI_DIR)/$(LIB_NAME)
