Common subdirectories: sks-1.1.1/bdb and sks-1.1.1-patched/bdb
diff -u sks-1.1.1/Makefile.local.unused sks-1.1.1-patched/Makefile.local.unused
--- sks-1.1.1/Makefile.local.unused	2009-03-25 20:18:03.000000000 -0500
+++ sks-1.1.1-patched/Makefile.local.unused	2009-10-13 16:54:17.835163112 -0500
@@ -1,7 +1,7 @@
 BDBLIB=-L/usr/lib
 BDBINCLUDE=-I/usr/include
 PREFIX=/usr/local
-LIBDB=-ldb-4.6
+LIBDB=-ldb-5.3
 MANDIR=/usr/share/man
 export BDBLIB
 export BDBINCLUDE
