diff -aur old/backend/src/cnij_backend_common.c new/backend/src/cnij_backend_common.c
--- old/backend/src/cnij_backend_common.c	2011-02-14 08:27:08.000000000 +0100
+++ new/backend/src/cnij_backend_common.c	2012-03-06 18:31:15.873614611 +0100
@@ -38,6 +38,7 @@
 // CUPS Header
 #include <cups/cups.h>
 #include <cups/ipp.h>
+#include <cups/ppd.h>
 
 // Header file for CANON
 #include "cnij_backend_common.h"
Only in new/backend/src: cnij_backend_common.c.rej
diff -aur old/cngpijmon/src/bjcupsmon_cups.c new/cngpijmon/src/bjcupsmon_cups.c
--- old/cngpijmon/src/bjcupsmon_cups.c	2011-03-01 06:52:18.000000000 +0100
+++ new/cngpijmon/src/bjcupsmon_cups.c	2012-03-06 18:32:03.766950732 +0100
@@ -20,6 +20,7 @@
 /*** Includes ***/
 #include <cups/cups.h>
 #include <cups/language.h>
+#include <cups/ppd.h>
 #include <sys/types.h>
 #include <unistd.h>
 #include <pwd.h>
