diff -crB notification-daemon-0.4.0-orig/src/daemon//daemon.c notification-daemon-0.4.0/src/daemon//daemon.c
*** notification-daemon-0.4.0-orig/src/daemon//daemon.c	2008-09-26 03:27:07.000000000 +0200
--- notification-daemon-0.4.0/src/daemon//daemon.c	2011-03-14 13:21:09.542176893 +0100
***************
*** 1077,1082 ****
--- 1077,1087 ----
  	gchar *sound_file = NULL;
  	gboolean sound_enabled;
  	gint i;
+     gint offset_x, offset_y;
+ 	GConfClient *client = get_gconf_client();
+ 
+ 	offset_x = gconf_client_get_int(client, GCONF_KEY_OFFSET_X, NULL);
+ 	offset_y = gconf_client_get_int(client, GCONF_KEY_OFFSET_Y, NULL);
  
  	if (id > 0)
  	{
***************
*** 1285,1291 ****
  		 * that position.
  		 */
  		theme_set_notification_arrow(nw, TRUE, x, y);
! 		theme_move_notification(nw, x, y);
  	}
  	else
  	{
--- 1290,1297 ----
  		 * that position.
  		 */
  		theme_set_notification_arrow(nw, TRUE, x, y);
! 		//theme_move_notification(nw, x, y);
!         theme_move_notification(nw, x + offset_x, y + offset_y);
  	}
  	else
  	{
***************
*** 1321,1326 ****
--- 1327,1333 ----
  		monitor_notification_source_windows(daemon, nt, window_xid);
  		sync_notification_position(daemon, nw, window_xid);
  	}
+     theme_move_notification(nw, x + offset_x, y + offset_y);
  
  	if (!screensaver_active(GTK_WIDGET(nw)) &&
  		!fullscreen_window_exists(GTK_WIDGET(nw)))
***************
*** 1329,1335 ****
  		if (sound_file != NULL)
  			sound_play(sound_file);
  	}
! 
  	g_free(sound_file);
  
  #if CHECK_DBUS_VERSION(0, 60)
--- 1336,1342 ----
  		if (sound_file != NULL)
  			sound_play(sound_file);
  	}
!     
  	g_free(sound_file);
  
  #if CHECK_DBUS_VERSION(0, 60)
diff -crB notification-daemon-0.4.0-orig/src/daemon//daemon.h notification-daemon-0.4.0/src/daemon//daemon.h
*** notification-daemon-0.4.0-orig/src/daemon//daemon.h	2008-03-15 00:11:43.000000000 +0100
--- notification-daemon-0.4.0/src/daemon//daemon.h	2011-03-14 10:01:40.903876099 +0100
***************
*** 34,39 ****
--- 34,41 ----
  #define GCONF_KEY_POPUP_LOCATION GCONF_KEY_DAEMON "/popup_location"
  #define GCONF_KEY_SOUND_ENABLED  GCONF_KEY_DAEMON "/sound_enabled"
  #define GCONF_KEY_DEFAULT_SOUND  GCONF_KEY_DAEMON "/default_sound"
+ #define GCONF_KEY_OFFSET_X       GCONF_KEY_DAEMON "/offset_x"
+ #define GCONF_KEY_OFFSET_Y       GCONF_KEY_DAEMON "/offset_y"
  
  #define NOTIFY_TYPE_DAEMON (notify_daemon_get_type())
  #define NOTIFY_DAEMON(obj) \
diff -crB notification-daemon-0.4.0-orig/src/daemon//stack.c notification-daemon-0.4.0/src/daemon//stack.c
*** notification-daemon-0.4.0-orig/src/daemon//stack.c	2008-03-15 00:11:43.000000000 +0100
--- notification-daemon-0.4.0/src/daemon//stack.c	2011-03-14 13:29:21.395114537 +0100
***************
*** 208,213 ****
--- 208,217 ----
  	GdkRectangle workarea;
  	GSList *l;
  	gint x, y, shiftx = 0, shifty = 0, index = 1;
+     GConfClient *client = get_gconf_client();
+     gint offset_x, offset_y;
+     offset_x = gconf_client_get_int(client, GCONF_KEY_OFFSET_X, NULL);
+     offset_y = gconf_client_get_int(client, GCONF_KEY_OFFSET_Y, NULL);
  
  	get_work_area(GTK_WIDGET(nw), &workarea);
  	get_origin_coordinates(stack->location, &workarea, &x, &y,
***************
*** 231,237 ****
  			translate_coordinates(stack->location, &workarea, &x, &y,
  								  &shiftx, &shifty, req.width, req.height,
  								  index++);
! 			theme_move_notification(nw2, x, y);
  		}
  		else if (nw_l != NULL)
  		{
--- 235,241 ----
  			translate_coordinates(stack->location, &workarea, &x, &y,
  								  &shiftx, &shifty, req.width, req.height,
  								  index++);
! 			theme_move_notification(nw2, x + offset_x, y + offset_y);
  		}
  		else if (nw_l != NULL)
  		{
