diff --git a/util/kdevplatform_shell_environment.sh b/util/kdevplatform_shell_environment.sh
index 83ccba5..a227a62 100755
--- a/util/kdevplatform_shell_environment.sh
+++ b/util/kdevplatform_shell_environment.sh
@@ -60,7 +60,7 @@ checkToolsInPath() {
 }
 
 # Check if all required tools are there (on the host machine)
-checkToolsInPath sed qdbus ls cut dirname mktemp basename readlink hostname
+checkToolsInPath sed qdbus-qt4 ls cut dirname mktemp basename readlink hostname
 
 if [ -n "$KDEV_SSH_FORWARD_CHAIN" ]; then
     # Check for additional utilities that are required on the client machine
@@ -69,11 +69,11 @@ fi
 
 # Queries the session name from the running application instance
 getSessionName() {
-    qdbus "$KDEV_DBUS_ID" /kdevelop/SessionController org.kdevelop.kdevelop.KDevelop.SessionController.sessionName
+    qdbus-qt4 "$KDEV_DBUS_ID" /kdevelop/SessionController org.kdevelop.kdevelop.KDevelop.SessionController.sessionName
 }
 
 getSessionDir() {
-    qdbus "$KDEV_DBUS_ID" /kdevelop/SessionController org.kdevelop.kdevelop.KDevelop.SessionController.sessionDir
+    qdbus-qt4 "$KDEV_DBUS_ID" /kdevelop/SessionController org.kdevelop.kdevelop.KDevelop.SessionController.sessionDir
 }
 
 getCurrentShellEnvPath() {
@@ -260,7 +260,7 @@ shev! () {
 
 # Opens a document in internally in the application
 openDocument () {
-    RESULT=$(qdbus $KDEV_DBUS_ID /org/kdevelop/DocumentController org.kdevelop.DocumentController.openDocumentSimple $1)
+    RESULT=$(qdbus-qt4 $KDEV_DBUS_ID /org/kdevelop/DocumentController org.kdevelop.DocumentController.openDocumentSimple $1)
     if ! [ "X$RESULT" == "Xtrue" ]; then
         echo "Failed to open $1"
     fi
@@ -268,7 +268,7 @@ openDocument () {
 
 # Opens a document in internally in the application
 openDocuments () {
-    RESULT=$(qdbus $KDEV_DBUS_ID /org/kdevelop/DocumentController org.kdevelop.DocumentController.openDocumentsSimple "(" $1 ")")
+    RESULT=$(qdbus-qt4 $KDEV_DBUS_ID /org/kdevelop/DocumentController org.kdevelop.DocumentController.openDocumentsSimple "(" $1 ")")
     if ! [ "X$RESULT" == "Xtrue" ]; then
         echo "Failed to open $1"
     fi
@@ -282,7 +282,7 @@ executeInApp () {
     if [ -z "$WD" ]; then
         WD=$(pwd)
     fi
-    RESULT=$(qdbus $KDEV_DBUS_ID /org/kdevelop/ExternalScriptPlugin org.kdevelop.ExternalScriptPlugin.executeCommand "$CMD" "$WD")
+    RESULT=$(qdbus-qt4 $KDEV_DBUS_ID /org/kdevelop/ExternalScriptPlugin org.kdevelop.ExternalScriptPlugin.executeCommand "$CMD" "$WD")
     if [ "X$RESULT" == "Xtrue" ]; then
         echo "Execution failed"
     fi
@@ -296,26 +296,26 @@ executeInAppSync () {
     if [ -z "$WD" ]; then
         WD=$(pwd)
     fi
-    RESULT=$(qdbus $KDEV_DBUS_ID /org/kdevelop/ExternalScriptPlugin org.kdevelop.ExternalScriptPlugin.executeCommandSync "$CMD" "$WD")
+    RESULT=$(qdbus-qt4 $KDEV_DBUS_ID /org/kdevelop/ExternalScriptPlugin org.kdevelop.ExternalScriptPlugin.executeCommandSync "$CMD" "$WD")
     echo "$RESULT"
 }
 
 # Getter functions:
 
 getActiveDocument () {
-    qdbus "$KDEV_DBUS_ID" /org/kdevelop/DocumentController org.kdevelop.DocumentController.activeDocumentPath "$@"
+    qdbus-qt4 "$KDEV_DBUS_ID" /org/kdevelop/DocumentController org.kdevelop.DocumentController.activeDocumentPath "$@"
 }
 
 getOpenDocuments () {
-    qdbus "$KDEV_DBUS_ID" /org/kdevelop/DocumentController org.kdevelop.DocumentController.activeDocumentPaths
+    qdbus-qt4 "$KDEV_DBUS_ID" /org/kdevelop/DocumentController org.kdevelop.DocumentController.activeDocumentPaths
 }
 
 raise! () {
-    qdbus "$KDEV_DBUS_ID" /kdevelop/MainWindow org.kdevelop.MainWindow.ensureVisible
+    qdbus-qt4 "$KDEV_DBUS_ID" /kdevelop/MainWindow org.kdevelop.MainWindow.ensureVisible
 }
 
 bdir! () {
-    TARG=$(qdbus "$KDEV_DBUS_ID" /org/kdevelop/ProjectController org.kdevelop.ProjectController.mapSourceBuild "$(pwd)" false)
+    TARG=$(qdbus-qt4 "$KDEV_DBUS_ID" /org/kdevelop/ProjectController org.kdevelop.ProjectController.mapSourceBuild "$(pwd)" false)
     if [ -n "$TARG" ]; then
         cd -- "$TARG"
     else
@@ -324,7 +324,7 @@ bdir! () {
 }
 
 project! () {
-    TARG=$(qdbus "$KDEV_DBUS_ID" /org/kdevelop/ProjectController org.kdevelop.ProjectController.mapSourceBuild "$(pwd)" true)
+    TARG=$(qdbus-qt4 "$KDEV_DBUS_ID" /org/kdevelop/ProjectController org.kdevelop.ProjectController.mapSourceBuild "$(pwd)" true)
     if [ -n "$TARG" ]; then
         cd -- "$TARG"
     else
@@ -336,7 +336,7 @@ project! () {
 # Main functions:
 
 raise!() {
-    qdbus "$KDEV_DBUS_ID" /kdevelop/MainWindow org.kdevelop.MainWindow.ensureVisible
+    qdbus-qt4 "$KDEV_DBUS_ID" /kdevelop/MainWindow org.kdevelop.MainWindow.ensureVisible
 }
 
 sync!() {
@@ -563,7 +563,7 @@ search!() {
         LOCATION="$LOCATION;$(mapFileToClient "$LOC")"
     done
 
-    qdbus "$KDEV_DBUS_ID" /org/kdevelop/GrepViewPlugin org.kdevelop.kdevelop.GrepViewPlugin.startSearch "$PATTERN" "$LOCATION" true
+    qdbus-qt4 "$KDEV_DBUS_ID" /org/kdevelop/GrepViewPlugin org.kdevelop.kdevelop.GrepViewPlugin.startSearch "$PATTERN" "$LOCATION" true
 }
 
 dsearch!() {
@@ -592,7 +592,7 @@ dsearch!() {
         LOCATION="$LOCATION;$(mapFileToClient "$LOC")"
     done
 
-    qdbus "$KDEV_DBUS_ID" /org/kdevelop/GrepViewPlugin org.kdevelop.kdevelop.GrepViewPlugin.startSearch "$PATTERN" "$LOCATION" false
+    qdbus-qt4 "$KDEV_DBUS_ID" /org/kdevelop/GrepViewPlugin org.kdevelop.kdevelop.GrepViewPlugin.startSearch "$PATTERN" "$LOCATION" false
 }
 
 ##### SSH DBUS FORWARDING --------------------------------------------------------------------------------------------------------------------
