diff --git a/language/duchain/duchainregister.h b/language/duchain/duchainregister.h
index 805b387..e8418cd 100644
--- a/language/duchain/duchainregister.h
+++ b/language/duchain/duchainregister.h
@@ -160,10 +160,10 @@ class KDEVPLATFORMLANGUAGE_EXPORT DUChainItemSystem {
 template<class T, class Data>
 struct DUChainItemRegistrator {
   DUChainItemRegistrator() {
-    DUChainItemSystem::self().registerTypeClass<T, Data>();
+    DUChainItemSystem::self().template registerTypeClass<T, Data>();
   }
   ~DUChainItemRegistrator() {
-    DUChainItemSystem::self().unregisterTypeClass<T, Data>();
+    DUChainItemSystem::self().template unregisterTypeClass<T, Data>();
   }
 };
 
diff --git a/language/duchain/types/typeregister.h b/language/duchain/types/typeregister.h
index 1ef9e44..8eb95a2 100644
--- a/language/duchain/types/typeregister.h
+++ b/language/duchain/types/typeregister.h
@@ -195,10 +195,10 @@ class KDEVPLATFORMLANGUAGE_EXPORT TypeSystem {
 template<class T, class Data>
 struct TypeSystemRegistrator {
   TypeSystemRegistrator() {
-    TypeSystem::self().registerTypeClass<T, Data>();
+    TypeSystem::self().template registerTypeClass<T, Data>();
   }
   ~TypeSystemRegistrator() {
-    TypeSystem::self().unregisterTypeClass<T, Data>();
+    TypeSystem::self().template unregisterTypeClass<T, Data>();
   }
 };
 
