--- Fragile/metadata.desktop.orig	2010-06-10 09:41:04.000000000 +0200
+++ Fragile/metadata.desktop	2011-08-02 21:41:36.573541002 +0200
@@ -1,12 +1,19 @@
 [Desktop Entry]
 Name=Fragile
-
+Name[sr]=Крхко
+Name[sr@ijekavian]=Крхко
+Name[sr@ijekavianlatin]=Krhko
+Name[sr@latin]=Krhko
 Comment=Shiny!
+Comment[sr]=Стакленаста сјајна тема за Плазму
+Comment[sr@ijekavian]=Стакленаста сјајна тема за Плазму
+Comment[sr@ijekavianlatin]=Staklenasta sjajna tema za Plasmu
+Comment[sr@latin]=Staklenasta sjajna tema za Plasmu
 
 X-KDE-PluginInfo-Author=GarthJohnson
 X-KDE-PluginInfo-Email=garth83@gmail.com
 X-KDE-PluginInfo-Name=Fragile
-X-KDE-PluginInfo-Version=1.3
+X-KDE-PluginInfo-Version=1.4
 X-KDE-PluginInfo-Website=http://www.kde-look.org
 X-KDE-PluginInfo-Category=Plasma Theme
 X-KDE-PluginInfo-Depends=KDE4
