diff --git a/TelepathyQt/account.cpp b/TelepathyQt/account.cpp
index c35759a..a5b12ad 100644
--- a/TelepathyQt/account.cpp
+++ b/TelepathyQt/account.cpp
@@ -52,7 +52,6 @@
 #include <QRegExp>
 #include <QSharedPointer>
 #include <QTimer>
-#include <QWeakPointer>
 
 #include <string.h>
 
@@ -634,7 +633,7 @@ struct Account::Private::DispatcherContext
     Client::ChannelDispatcherInterface *iface;
 
     bool introspected, supportsHints;
-    QWeakPointer<PendingVariant> introspectOp;
+    QPointer<PendingVariant> introspectOp;
 
 private:
     DispatcherContext(const DispatcherContext &);
diff --git a/TelepathyQt/utils.cpp b/TelepathyQt/utils.cpp
index 5b922d9..e4614c2 100644
--- a/TelepathyQt/utils.cpp
+++ b/TelepathyQt/utils.cpp
@@ -204,7 +204,7 @@ QVariant parseValueWithDBusSignature(const QString &value,
         case QVariant::StringList:
             {
                 QStringList list;
-                QByteArray rawValue = value.toAscii();
+                QByteArray rawValue = value.toLatin1();
                 if (KeyFile::unescapeStringList(rawValue, 0, rawValue.size(), list)) {
                     return QVariant(list);
                 } else {
diff --git a/cmake/modules/FindQt5.cmake b/cmake/modules/FindQt5.cmake
index fb30054..8bf4fc5 100644
--- a/cmake/modules/FindQt5.cmake
+++ b/cmake/modules/FindQt5.cmake
@@ -80,7 +80,7 @@ ENDIF(NOT QTVERSION MATCHES "5.*")
 FIND_PACKAGE(PkgConfig REQUIRED)
 
 IF(NOT Qt5_FIND_COMPONENTS)
-  SET(_COMPONENTS QtCore QtDBus QtGui QtNetwork QtTest QtWidgets QtXml)
+  SET(_COMPONENTS Qt5Core Qt5DBus Qt5Gui Qt5Network Qt5Test Qt5Widgets Qt5Xml)
 ELSE(NOT Qt5_FIND_COMPONENTS)
   SET(_COMPONENTS ${Qt5_FIND_COMPONENTS})
 ENDIF(NOT Qt5_FIND_COMPONENTS)
@@ -153,7 +153,7 @@ MARK_AS_ADVANCED(QT_INCLUDES QT_INCLUDE_DIR
                  QT_QMAKE_EXECUTABLE_FINDQT QT_QMAKE_EXECUTABLE QT_MOC_EXECUTABLE)
 
 # Invokes pkgconfig, cleans up the result and sets variables
-EXECUTE_PROCESS(COMMAND ${PKG_CONFIG_EXECUTABLE} --variable qt_config QtCore
+EXECUTE_PROCESS(COMMAND ${PKG_CONFIG_EXECUTABLE} --variable qt_config Qt5Core
     OUTPUT_VARIABLE _pkgconfig_flags
     RESULT_VARIABLE _pkgconfig_failed)
 STRING(REPLACE " " ";" QT_CONFIG_FLAGS "${_pkgconfig_flags}")
