diff --git a/cpp/src/IceUtil/FileUtil.cpp b/cpp/src/IceUtil/FileUtil.cpp
index 1c1b974..4f2f9c5 100644
--- a/cpp/src/IceUtil/FileUtil.cpp
+++ b/cpp/src/IceUtil/FileUtil.cpp
@@ -14,6 +14,11 @@
 #include <climits>
 #include <string.h>
 
+#ifdef __linux
+#  include <sys/types.h>
+#  include <unistd.h>
+#endif
+
 #ifdef _WIN32
 #  include <process.h>
 #endif
diff --git a/cpp/src/Slice/FileTracker.cpp b/cpp/src/Slice/FileTracker.cpp
index af009fe..4ebf1ab 100644
--- a/cpp/src/Slice/FileTracker.cpp
+++ b/cpp/src/Slice/FileTracker.cpp
@@ -9,7 +9,7 @@
 
 #include <Slice/FileTracker.h>
 
-#ifdef __sun
+#if defined(__sun) || defined(__linux)
 #   include <unistd.h>
 #endif
 
