--- src/ideviceinstaller.c	2011-08-07 14:04:11.718124077 +0800
+++ src/ideviceinstaller.c	2011-08-07 14:38:15.283719798 +0800
@@ -163,7 +163,7 @@
 	}
 
 	*buffer = malloc(zs.size);
-	if (zip_fread(zfile, *buffer, zs.size) != zs.size) {
+	if (zip_fread(zfile, *buffer, zs.size) != (int) zs.size) {
 		fprintf(stderr, "ERROR: zip_fread %" PRIu64 " bytes from '%s'\n", (uint64_t)zs.size, filename);
 		free(*buffer);
 		*buffer = NULL;
@@ -562,6 +562,12 @@
 		plist_t bname = plist_dict_get_item(info, "CFBundleName");
 		if (bname) {
 			plist_get_string_val(bname, &bundlename);
+		} else {
+			/* no CFBundleName, use CFBundleDisplayName */
+			bname = plist_dict_get_item(info, "CFBundleDisplayName");
+			if (bname) {
+				plist_get_string_val(bname, &bundlename);
+			}
 		}
 		plist_free(info);
 
