--- src/ideviceinstaller.c	2012-09-13 10:12:12.000000000 +0800
+++ src/ideviceinstaller.c.new	2012-09-13 10:14:06.000000000 +0800
@@ -698,6 +698,12 @@
 			plist_t bname = plist_dict_get_item(info, "CFBundleName");
 			if (bname) {
 				plist_get_string_val(bname, &bundlename);
+			} else {
+				/* no CFBundleName, use CFBundleDisplayName */
+				bname = plist_dict_get_item(info, "CFBundleDisplayName");
+				if (bname) {
+					plist_get_string_val(bname, &bundlename);
+				}
 			}
 			plist_free(info);
 
