Index: Makefile
===================================================================
--- Makefile	(revision 461)
+++ Makefile	(working copy)
@@ -6,8 +6,8 @@
 rpn64: rpn.c
 	$(CC) -o rpn64 $(CFLAGS) -DRPN64 rpn.c
 
-iomem: LDLIBS += -lreadline -ltermcap
-teletype: LDLIBS += -lreadline -ltermcap
+iomem: LDLIBS += -lreadline
+teletype: LDLIBS += -lreadline
 
 install: all
 	install indt.sh /usr/local/bin/indt
