diff -ru libvpx-1.2.0/build/make/configure.sh patched/build/make/configure.sh
--- libvpx-1.2.0/build/make/configure.sh	2013-05-10 16:03:44.000000000 +0200
+++ patched/build/make/configure.sh	2013-06-16 20:02:41.366195598 +0200
@@ -1070,10 +1070,12 @@
             win32)
                 add_asflags -f win32
                 enabled debug && add_asflags -g cv8
+                add_ldflags "-Wl,-no-undefined"
             ;;
             win64)
                 add_asflags -f x64
                 enabled debug && add_asflags -g cv8
+                add_ldflags "-Wl,-no-undefined"
             ;;
             linux*|solaris*)
                 add_asflags -f elf${bits}
Only in patched/build/make: configure.sh~
diff -ru libvpx-1.2.0/build/make/Makefile patched/build/make/Makefile
--- libvpx-1.2.0/build/make/Makefile	2013-05-10 16:03:44.000000000 +0200
+++ patched/build/make/Makefile	2013-06-16 21:46:57.467777466 +0200
@@ -253,6 +253,7 @@
 	$(if $(quiet),@echo "    [LD] $$@")
 	$(qexec)$$(LD) -shared $$(LDFLAGS) \
             -Wl,--no-undefined -Wl,-soname,$$(SONAME) \
+            -Wl,-out-implib,$$@.implib \
             -Wl,--version-script,$$(SO_VERSION_SCRIPT) -o $$@ \
             $$(filter %.o,$$?) $$(extralibs)
 endef
Only in patched/build/make: Makefile~
diff -ru libvpx-1.2.0/configure patched/configure
--- libvpx-1.2.0/configure	2013-05-10 16:03:44.000000000 +0200
+++ patched/configure	2013-06-16 21:00:03.113323364 +0200
@@ -452,18 +452,18 @@
 }
 
 process_detect() {
-    if enabled shared; then
+    #if enabled shared; then
         # Can only build shared libs on a subset of platforms. Doing this check
         # here rather than at option parse time because the target auto-detect
         # magic happens after the command line has been parsed.
-        if ! enabled linux; then
-            if enabled gnu; then
-                echo "--enable-shared is only supported on ELF; assuming this is OK"
-            else
-                die "--enable-shared only supported on ELF for now"
-            fi
-        fi
-    fi
+        #if ! enabled linux; then
+        #    if enabled gnu; then
+        #        echo "--enable-shared is only supported on ELF; assuming this is OK"
+        #    else
+        #        die "--enable-shared only supported on ELF for now"
+        #    fi
+        #fi
+    #fi
     if [ -z "$CC" ]; then
         echo "Bypassing toolchain for environment detection."
         enable external_build
Only in patched: configure~
diff -ru libvpx-1.2.0/libs.mk patched/libs.mk
--- libvpx-1.2.0/libs.mk	2013-05-10 16:03:44.000000000 +0200
+++ patched/libs.mk	2013-06-16 21:44:08.108635963 +0200
@@ -214,6 +214,7 @@
 
 INSTALL-LIBS-$(CONFIG_SHARED) += $(LIBVPX_SO_SYMLINKS)
 INSTALL-LIBS-$(CONFIG_SHARED) += $(LIBSUBDIR)/$(LIBVPX_SO)
+INSTALL-LIBS-$(CONFIG_SHARED) += $(LIBSUBDIR)/$(LIBVPX_SO).implib
 
 LIBS-$(BUILD_LIBVPX) += vpx.pc
 vpx.pc: config.mk libs.mk
Only in patched: libs.mk~
