--- cmdline/cmdline.cpp~	2006-12-29 23:06:53.000000000 +0200
+++ cmdline/cmdline.cpp	2006-12-29 23:14:20.000000000 +0200
@@ -172,11 +172,11 @@
 	SLtt_get_terminfo();
 	SLtt_get_screen_size();
 	if (SLkp_init() == -1) {
-		SLang_doerror("SLkp_init() failed.");
+		SLang_verror(SL_INTRINSIC_ERROR, "SLkp_init() failed.");
 		exit(-1);
 	}
 	if (SLang_init_tty(-1, 0, 0) == -1) {
-		SLang_doerror("SLang_init_tty() failed.");
+		SLang_verror(SL_INTRINSIC_ERROR, "SLang_init_tty() failed.");
 		exit(-1);
 	}
 	SLang_set_abort_signal(NULL);
@@ -196,7 +196,7 @@
 	//SLsmg_normal_video();
 
 	if (SLsmg_init_smg() == -1) {
-		SLang_doerror("SLsmg_init_smg() failed.");
+		SLang_verror(SL_INTRINSIC_ERROR, "SLsmg_init_smg() failed.");
 		SLang_reset_tty();
 		exit(-1);
 	}
