diff -ur src_a/net/netservice.h src/net/netservice.h
--- src_a/net/netservice.h	2007-02-04 15:07:58.000000000 +0100
+++ src/net/netservice.h	2011-10-06 15:39:37.460284179 +0200
@@ -50,7 +50,7 @@
   int m_retries;
 };
 
-class NetService : protected QThread
+class NetService : public QThread
 {
   Q_OBJECT
  
diff -ur src_a/net/warpcopydisk.h src/net/warpcopydisk.h
--- src_a/net/warpcopydisk.h	2007-02-04 14:55:24.000000000 +0100
+++ src/net/warpcopydisk.h	2011-10-06 15:42:59.533623287 +0200
@@ -5,7 +5,7 @@
 #include "warpcopy.h"
 
 //! a tool class to copy a whole disk via warp copy service
-class WarpCopyDisk : protected QThread
+class WarpCopyDisk : public QThread
 {
   Q_OBJECT
  
