diff -ru LabCurves.orig//mm extern LabCurves.py LabCurves/mm extern LabCurves.py
--- LabCurves.orig//mm extern LabCurves.py	2010-06-18 19:51:59.000000000 +0200
+++ LabCurves/mm extern LabCurves.py	2010-07-21 13:21:01.961355985 +0200
@@ -63,8 +63,7 @@
   pdb.file_tiff_save2(tempimage, tempdrawable, tempfilename, tempfilename, 0, 1)
 
   # Command line
-  path = sys.path[0]
-  command = path + "/LabCurves/LabCurves " + "\"" + tempfilename + "\""
+  command = "/usr/bin/LabCurves " + "\"" + tempfilename + "\""
 
   # Invoke external command
   pdb.gimp_progress_set_text ("run LabCurves...")
