diff -Naur indicator-applet-0.5.0.orig/src/tomboykeybinder.h indicator-applet-0.5.0/src/tomboykeybinder.h
--- indicator-applet-0.5.0.orig/src/tomboykeybinder.h	2011-07-07 17:43:26.000000000 +0200
+++ indicator-applet-0.5.0/src/tomboykeybinder.h	2012-05-21 04:46:53.343551314 +0200
@@ -18,7 +18,7 @@
 #ifndef __TOMBOY_KEY_BINDER_H__
 #define __TOMBOY_KEY_BINDER_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
