--- configure	2011-02-13 13:11:56.000000000 -0500
+++ ../../configure	2012-03-06 01:04:10.678740450 -0500
@@ -13,7 +13,16 @@
 # tclsh program with a version number in the file name is found, the
 # default program to execute is just tclsh.
 # The backslashes at the end of these lines are needed: \
-tclsh=tclsh8.5;
+TCL_VERSION_SCRIPT='
+puts $tcl_version
+exit 0
+'
+
+tcl_version=`tclsh<<EOF
+$TCL_VERSION_SCRIPT
+EOF`
+
+tclsh=tclsh$tcl_version;
 
 # This shell script code did not work for me in Linux so it is
 # commented out for now; the configure script will just run in
