diff --git a/src/command_peer.cc b/src/command_peer.cc
index c17bed2..4bedc43 100644
--- a/src/command_peer.cc
+++ b/src/command_peer.cc
@@ -146,6 +146,11 @@ initialize_command_peer() {
 
   ADD_CP_VALUE_UNI("completed_percent",   std::ptr_fun(&retrieve_p_completed_percent));
 
+  ADD_CP_VALUE_MEM_UNI("last_connection", &torrent::Peer::peer_info, &torrent::PeerInfo::last_connection);
+  ADD_CP_VALUE_MEM_UNI("last_transfer",   &torrent::Peer::peer_info, &torrent::PeerInfo::last_transfer);
+  ADD_CP_VALUE_MEM_UNI("last_transfer_up", &torrent::Peer::peer_info, &torrent::PeerInfo::last_transfer_up);
+  ADD_CP_VALUE_MEM_UNI("last_transfer_down", &torrent::Peer::peer_info, &torrent::PeerInfo::last_transfer_down);
+
   ADD_CP_VALUE_MEM_UNI("up_rate",         &torrent::Peer::up_rate, &torrent::Rate::rate);
   ADD_CP_VALUE_MEM_UNI("up_total",        &torrent::Peer::up_rate, &torrent::Rate::total);
   ADD_CP_VALUE_MEM_UNI("down_rate",       &torrent::Peer::down_rate, &torrent::Rate::rate);
diff --git a/src/command_ui.cc b/src/command_ui.cc
index 9d39360..915c2f0 100644
--- a/src/command_ui.cc
+++ b/src/command_ui.cc
@@ -274,6 +274,9 @@ apply_to_time(int flags, const torrent::Object& rawArgs) {
 
 torrent::Object
 apply_to_elapsed_time(const torrent::Object& rawArgs) {
+  if (rawArgs.as_value() == 0)
+    return "--:--:--";
+
   uint64_t arg = cachedTime.seconds() - rawArgs.as_value();
 
   char buffer[48];
diff --git a/src/display/text_element_value.cc b/src/display/text_element_value.cc
index 403046c..61cf217 100644
--- a/src/display/text_element_value.cc
+++ b/src/display/text_element_value.cc
@@ -68,7 +68,9 @@ TextElementValueBase::print(char* first, char* last, Canvas::attributes_list* at
   int64_t val = value(target.second);
 
   // Transform the value if needed.
-  if (m_flags & flag_elapsed)
+  if (m_flags & (flag_elapsed | flag_remaining) && val == 0)
+    val = -1;
+  else if (m_flags & flag_elapsed)
     val = cachedTime.seconds() - val;
   else if (m_flags & flag_remaining)
     val = val - cachedTime.seconds();
@@ -100,7 +102,7 @@ TextElementValueBase::print(char* first, char* last, Canvas::attributes_list* at
       first += std::min<ptrdiff_t>(std::max(snprintf(first, last - first + 1, "%5.1f TB", (double)val / (int64_t(1) << 40)), 0), last - first + 1);
 
   } else if (m_flags & flag_timer) {
-    if (val == 0)
+    if (val < 0)
       first += std::min<ptrdiff_t>(std::max(snprintf(first, last - first + 1, "--:--:--"), 0), last - first + 1);
     else
       first += std::min<ptrdiff_t>(std::max(snprintf(first, last - first + 1, "%2d:%02d:%02d", (int)(val / 3600), (int)((val / 60) % 60), (int)(val % 60)), 0), last - first + 1);
diff --git a/src/ui/element_peer_list.cc b/src/ui/element_peer_list.cc
index c5ad738..da08e15 100644
--- a/src/ui/element_peer_list.cc
+++ b/src/ui/element_peer_list.cc
@@ -109,6 +109,7 @@ ElementPeerList::create_info() {
   element->push_column("Options:",   te_command("p.get_options_str="));
   element->push_column("Connected:", te_command("if=$p.is_incoming=,incoming,outgoing"));
   element->push_column("Encrypted:", te_command("if=$p.is_encrypted=,yes,$p.is_obfuscated=,handshake,no"));
+  element->push_column("Time:",      te_command("cat=Connected\\ ,$to_elapsed_time=$p.get_last_connection=,\\ \\ Idle\\ ,$to_elapsed_time=$p.get_last_transfer_up=,\\ up\\ \\ ,$to_elapsed_time=$p.get_last_transfer_down=,\\ down"));
 
   element->push_back("");
   element->push_column("Snubbed:",   te_command("if=$p.is_snubbed=,yes,no"));
