--- Makefile.orig	2012-02-22 11:40:02.113840380 +0600
+++ Makefile	2012-02-22 11:43:35.623841589 +0600
@@ -10,11 +10,11 @@
 #					xtrojka version (do not change!)
 VERSION=123
 #					where to put the binary
-TARGET_DIR=/usr/games
+TARGET_DIR=/usr/bin
 #					where to put the manpage
-MANDIR=/usr/local/man/man6
+MANDIR=/usr/share/man/man6
 #					the location of the highscorefile
-HSFILE=/fill_in_your_scorefile
+HSFILE=/var/lib/xtrojka/xtrojka.score
 #					your favourite packer
 COMPRESS=gzip -v9
 #					your favourite c-compiler
--- slist.c.orig	2012-02-22 11:46:39.337175963 +0600
+++ slist.c	2012-02-22 11:47:07.660509454 +0600
@@ -17,6 +17,7 @@
 #include <time.h>
 #include <signal.h>
 #include <pwd.h>
+#include <errno.h>
 
 #include <X11/Intrinsic.h>
 #include <X11/Xlib.h>
--- xtrojka.man.orig	2012-02-22 11:55:57.620512450 +0600
+++ xtrojka.man	2012-02-22 11:56:53.813846101 +0600
@@ -94,7 +94,7 @@
 
 .SH FILES
 .TP 15
- __SCOREDIR__
+ /var/lib/xtrojka/xtrojka.score
 The 
 .IR xtrojka
 highscore file
