commit ce795ae2667dfba2ab8f4e8d5be78596a433fb79
Author: Benjamin Chrétien <chretien@lirmm.fr>
Date:   Sun Mar 9 20:51:58 2014 +0100

    Add Boost patch (remove -mt suffix).

diff --git a/Makefile.hrpsys-base b/Makefile.hrpsys-base
index 47620ae..e4f5f79 100644
--- a/Makefile.hrpsys-base
+++ b/Makefile.hrpsys-base
@@ -1,10 +1,10 @@
 INSTALL_DIR ?= $(CURDIR)
-SVN_DIR     = build/hrpsys-base-source
 PATCH_DIR   = $(CURDIR)
+SVN_DIR     = build/hrpsys-base-source
 #SVN_URL    = https://hrpsys-base.googlecode.com/svn/trunk
 SVN_URL     = http://hrpsys-base.googlecode.com/svn/tags/315.1.8
 #SVN_REVISION=-r@REVISION@
-SVN_PATCH   = ${PATCH_DIR}/patch/qhull.patch
+SVN_PATCH   = ${PATCH_DIR}/patch/qhull.patch ${PATCH_DIR}/patch/boost.patch
 SVN_CMDLINE = svn --non-interactive --trust-server-cert
 MK_DIR      = $(shell rospack find mk)
 NUM_OF_CPUS = $(shell grep -c '^processor' /proc/cpuinfo)
diff --git a/catkin.cmake b/catkin.cmake
index 0bdf829..767060a 100644
--- a/catkin.cmake
+++ b/catkin.cmake
@@ -20,6 +20,7 @@ if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/installed)
     INSTALL_DIR=${CATKIN_DEVEL_PREFIX}
     OPENRTM_DIR='' # keep blank so that FindOpenRTM.cmake uses rtm-config
     MK_DIR=${mk_PREFIX}/share/mk
+    PATCH_DIR=${PROJECT_SOURCE_DIR}
     PKG_CONFIG_PATH=${openhrp3_PREFIX}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH} # for openrtm3.1.pc
     installed
     RESULT_VARIABLE _make_failed)
diff --git a/patch/boost.patch b/patch/boost.patch
new file mode 100644
index 0000000..553e0ae
--- /dev/null
+++ b/patch/boost.patch
@@ -0,0 +1,28 @@
+Index: lib/util/CMakeLists.txt
+===================================================================
+--- lib/util/CMakeLists.txt
++++ lib/util/CMakeLists.txt
+@@ -53,8 +53,8 @@
+   ${GLUT_LIBRARIES} 
+   ${QHULL_LIBRARIES}
+   GLEW
+-  boost_thread-mt
+-  boost_system-mt
++  boost_thread
++  boost_system
+   )
+ 
+ set(target hrpsysUtil)
+Index: lib/util/CMakeLists.txt
+===================================================================
+--- util/simulator/CMakeLists.txt
++++ util/simulator/CMakeLists.txt
+@@ -40,7 +40,7 @@
+      )
+ else()
+    target_link_libraries(hrpsysext 
+-     boost_python-mt
++     boost_python
+      hrpsysUtil
+      ${PYTHON_LIBRARIES}
+      )
