index 3c22019..baa807f 100644
--- a/yakuake/app/terminal.cpp
+++ b/yakuake/app/terminal.cpp
@@ -24,6 +24,7 @@
 
 #include <KActionCollection>
 #include <KApplication>
+#include <KAction>
 #include <KColorScheme>
 #include <kde_terminal_interface.h>
 #include <KLocalizedString>
@@ -183,6 +184,11 @@ void Terminal::displayKPartLoadError()
     layout->setStretchFactor(label,5);
 }
 
+void Terminal::paste()
+{
+    QApplication::postEvent(m_terminalWidget, new QKeyEvent(QEvent::KeyPress, Qt::Key_V, Qt::ShiftModifier|Qt::ControlModifier));
+}
+
 void Terminal::disableOffendingPartActions()
 {
     // This is an unwelcome stop-gap that will be removed once we can
@@ -220,6 +226,10 @@ void Terminal::disableOffendingPartActions()
 
         action = actionCollection->action("shrink-font");
         if (action) action->setEnabled(false);
+	
+        KAction *newAct = actionCollection->addAction(KStandardAction::New, this, SLOT(paste()));
+        newAct->setShortcut(QKeySequence(Qt::ALT + Qt::Key_V));
+        newAct->setEnabled(true);
     }
 }
 
index 23e70a0..9d8bc00 100644
--- a/yakuake/app/terminal.h
+++ b/yakuake/app/terminal.h
@@ -81,6 +81,7 @@ class Terminal : public QObject
         void overrideShortcut(QKeyEvent* event, bool& override);
         void silenceDetected();
         void activityDetected();
+        void paste();
 
 
     private:
