--- build_visit2_8_0	2014-10-15 17:22:04.652329055 +0200
+++ build_visit2_8_0.new	2014-10-15 18:26:02.891820496 +0200
@@ -3956,6 +3956,8 @@
     cd $VISIT_DIR
     #cp $START_DIR/$(hostname).cmake config-site
 
+    sed -i 's/#ifdef\ __linux__/#ifdef\ __linux__\n#define\ GLX_GLXEXT_LEGACY/g' vtkqt/vtkQtRenderWindow.C
+
     #
     # Call cmake
     # 
@@ -15943,6 +15945,10 @@
                  "failing harmlessly on a second application."
         fi
     fi
+
+    sed -i 's/\/\/#define\ GLX_GLXEXT_LEGACY/#define\ GLX_GLXEXT_LEGACY/g' \
+      Rendering/OpenGL/vtkXOpenGLRenderWindow.cxx
+
     # move back up to the start dir 
     cd "$START_DIR"
 
