*** usr/share/gnome-shell/extensions/enhanced-search@linuxdeepin.com/extension.js	2012-07-05 17:33:00.000000000 +0900
--- usr/share/gnome-shell/extensions/enhanced-search@linuxdeepin.com/extension_new.js	2012-08-03 20:49:24.336733155 +0900
***************
*** 94,100 ****
      Gettext.bindtextdomain('enhanced-search', localePath);
      _ = Gettext.domain('enhanced-search').gettext;
  
!     let google = {'uri': "http://www.google.com.hk/search?ie=UTF-8&q=",
                    'title': _("Google"),
                    'icon': "google.png"
                   };
--- 94,100 ----
      Gettext.bindtextdomain('enhanced-search', localePath);
      _ = Gettext.domain('enhanced-search').gettext;
  
!     let google = {'uri': "http://www.google.co.jp/search?ie=UTF-8&q=",
                    'title': _("Google"),
                    'icon': "google.png"
                   };
***************
*** 102,115 ****
                   'title': _("Baidu"),
                   'icon': "baidu.png"
                  };
!     let taobao = {'uri': "http://s.taobao.com/search?q=",
!                   'title': _("Taobao"),
!                   'icon': "taobao.png"
!                  };
      let yahoo = {'uri': "http://cn.search.yahoo.com/search?ie=UTF-8&p=",
                   'title': _("Yahoo!"),
                   'icon': "yahoo.png"
                  };
      let bing = {'uri': "http://cn.bing.com/search?q=",
                  'title': _("Bing"),
                  'icon': "bing.png"
--- 102,131 ----
                   'title': _("Baidu"),
                   'icon': "baidu.png"
                  };
! 	let dictcn = {'uri': "http://dict.cn/",
!                  'title': _("dictcn"),
!                  'icon': "dictcn.png"
!                 };
      let yahoo = {'uri': "http://cn.search.yahoo.com/search?ie=UTF-8&p=",
                   'title': _("Yahoo!"),
                   'icon': "yahoo.png"
                  };
+     let PKG_NAME = {'uri': "https://www.archlinux.de/?page=Packages&searchfield=name&packager=0&search=",
+                 'title': _("PKG NAME"),
+                 'icon': "pkg.png"
+                };
+     let PKG_DESC = {'uri': "https://www.archlinux.de/?page=Packages&searchfield=description&packager=0&search=",
+                 'title': _("PKG DESC"),
+                 'icon': "pkg.png"
+                };
+     let PKG_FILE = {'uri': "https://www.archlinux.de/?page=Packages&searchfield=file&packager=0&search=",
+                 'title': _("PKG FILE"),
+                 'icon': "pkg.png"
+                };
+     let PKG_AUR = {'uri': "https://aur.archlinux.org/packages.php?O=0&K=",
+                   'title': _("PKG AUR"),
+                   'icon': "pkg.png"
+                  };
      let bing = {'uri': "http://cn.bing.com/search?q=",
                  'title': _("Bing"),
                  'icon': "bing.png"
***************
*** 118,123 ****
--- 134,144 ----
                    'title': _("Flickr"),
                    'icon': "flickr.png"
                   };
+ 
+     let verycd = {'uri': "http://www.verycd.com/search/entries/",
+                   'title': _("verycd"),
+                   'icon': "verycd.png"
+                  };
      let wikipedia = {'uri': "http://zh.wikipedia.org/wiki/",
                       'title': _("Wikipedia"),
                       'icon': "wikipedia.png"
***************
*** 130,136 ****
                     'title': _("YouTube"),
                     'icon': "youtube.png"
                    };
!     searchEngines = eval(_("[google, bing, yahoo, flickr, wikipedia, twitter, youtube]"));
  
      searchProvidersBox = Main.overview._viewSelector._searchTab._searchResults._searchProvidersBox;
      currentPath = extensionMeta.path;
--- 151,157 ----
                     'title': _("YouTube"),
                     'icon': "youtube.png"
                    };
! 	searchEngines = eval(_("[google, baidu, dictcn, PKG_NAME, PKG_DESC, PKG_FILE, PKG_AUR, verycd, wikipedia, youtube]"));
  
      searchProvidersBox = Main.overview._viewSelector._searchTab._searchResults._searchProvidersBox;
      currentPath = extensionMeta.path;
