diff -rupN ndiswrapper-1.58-orig/driver/loader.c ndiswrapper-1.58/driver/loader.c
--- ndiswrapper-1.58-orig/driver/loader.c	2013-02-19 19:00:37.000000000 +0100
+++ ndiswrapper-1.58/driver/loader.c	2013-05-13 08:58:39.707145000 +0200
@@ -575,7 +575,7 @@ static int load_user_space_driver(struct
 	} else {
 		printk(KERN_INFO "%s: driver %s (%s) loaded\n",
 		       DRIVER_NAME, wrap_driver->name, wrap_driver->version);
-		add_taint(TAINT_PROPRIETARY_MODULE);
+		add_taint(TAINT_PROPRIETARY_MODULE, LOCKDEP_STILL_OK);
 		EXIT1(return 0);
 	}
 }
#diff -rupN ndiswrapper-1.58-orig/driver/Makefile ndiswrapper-1.58/driver/Makefile
#--- ndiswrapper-1.58-orig/driver/Makefile	2013-02-19 19:00:37.000000000 +0100
#+++ ndiswrapper-1.58/driver/Makefile	2013-05-13 10:02:06.507185180 +0200
#@@ -185,7 +185,7 @@ clean:
# 	rm -rf .tmp_versions
# 
# install: config_check $(MODULE)
#-	@/sbin/modinfo $(MODULE) | grep -q "^vermagic: *$(KVERS) " || \
#+	@/usr/bin/kmod $(MODULE) | grep -q "^vermagic: *$(KVERS) " || \
# 		{ echo "$(MODULE)" is not for Linux $(KVERS); exit 1; }
# 	mkdir -p -m 755 $(DESTDIR)$(INST_DIR)
# 	install -m 0644 $(MODULE) $(DESTDIR)$(INST_DIR)
diff -rupN ndiswrapper-1.58-orig/driver/wrapper.c ndiswrapper-1.58/driver/wrapper.c
--- ndiswrapper-1.58-orig/driver/wrapper.c	2013-02-19 19:00:37.000000000 +0100
+++ ndiswrapper-1.58/driver/wrapper.c	2013-05-13 09:02:07.486148000 +0200
@@ -72,7 +72,7 @@ static void module_cleanup(void)
 static int __init wrapper_init(void)
 {
 #ifdef TAINT_OOT_MODULE
-	add_taint(TAINT_OOT_MODULE);
+	add_taint(TAINT_OOT_MODULE, LOCKDEP_STILL_OK);
 #endif
 	printk(KERN_INFO "%s version %s loaded (smp=%s, preempt=%s)\n",
 	       DRIVER_NAME, DRIVER_VERSION,
