diff -rupN Sockets-2.3.9.2.orig/TcpSocket.cpp Sockets-2.3.9.2.openssl-1_0_0-fix/TcpSocket.cpp
--- Sockets-2.3.9.2.orig/TcpSocket.cpp	2010-02-14 09:48:25.000000000 +0100
+++ Sockets-2.3.9.2.openssl-1_0_0-fix/TcpSocket.cpp	2010-05-07 11:13:14.634372823 +0200
@@ -1301,7 +1301,11 @@ void TcpSocket::InitializeContext(const
 	/* Create our context*/
 	if (m_server_contexts.find(context) == m_server_contexts.end())
 	{
-		SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : SSLv3_method();
+                #if OPENSSL_VERSION_NUMBER >= 0x10000000L
+                        const SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : SSLv3_method();
+                #else
+                        SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : SSLv3_method();
+                #endif
 		m_ssl_ctx = m_server_contexts[context] = SSL_CTX_new(meth);
 		SSL_CTX_set_mode(m_ssl_ctx, SSL_MODE_AUTO_RETRY|SSL_MODE_ENABLE_PARTIAL_WRITE);
 		// session id
@@ -1337,7 +1341,11 @@ void TcpSocket::InitializeContext(const
 	/* Create our context*/
 	if (m_server_contexts.find(context) == m_server_contexts.end())
 	{
-		SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : SSLv3_method();
+                #if OPENSSL_VERSION_NUMBER >= 0x10000000L
+                        const SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : SSLv3_method();
+                #else
+                        SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : SSLv3_method();
+                #endif
 		m_ssl_ctx = m_server_contexts[context] = SSL_CTX_new(meth);
 		SSL_CTX_set_mode(m_ssl_ctx, SSL_MODE_AUTO_RETRY|SSL_MODE_ENABLE_PARTIAL_WRITE);
 		// session id
