Index: cblrepo/diagrams-lib-1.2.0.7/diagrams-lib.cabal
===================================================================
--- cblrepo.orig/diagrams-lib-1.2.0.7/diagrams-lib.cabal
+++ cblrepo/diagrams-lib-1.2.0.7/diagrams-lib.cabal
@@ -98,14 +98,14 @@ Library
                        dual-tree >= 0.2 && < 0.3,
                        diagrams-core >= 1.2 && < 1.3,
                        active >= 0.1 && < 0.2,
-                       vector-space >= 0.7.7 && < 0.9,
+                       vector-space >= 0.7.7,
                        vector-space-points >= 0.1.2 && < 0.3,
                        MemoTrie >= 0.6 && < 0.7,
                        colour >= 2.3.2 && < 2.4,
                        data-default-class < 0.1,
                        fingertree >= 0.1 && < 0.2,
                        intervals >= 0.7 && < 0.8,
-                       lens >= 4.0 && < 4.7,
+                       lens >= 4.0,
                        tagged >= 0.7,
                        optparse-applicative >= 0.11 && < 0.12,
                        filepath,
