Index: cblrepo/diagrams-lib-1.2.0.7/diagrams-lib.cabal
===================================================================
--- cblrepo.orig/diagrams-lib-1.2.0.7/diagrams-lib.cabal
+++ cblrepo/diagrams-lib-1.2.0.7/diagrams-lib.cabal
@@ -105,7 +105,7 @@ Library
                        data-default-class < 0.1,
                        fingertree >= 0.1 && < 0.2,
                        intervals >= 0.7 && < 0.8,
-                       lens >= 4.0 && < 4.7,
+                       lens >= 4.0,
                        tagged >= 0.7,
                        optparse-applicative >= 0.11 && < 0.12,
                        filepath,
