--- oprofile-analyze.py.orig	2012-01-15 12:51:40.000000000 +0100
+++ oprofile-analyze.py		2012-01-15 13:09:09.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python -E
+#!/usr/bin/python2 -E
 
 import os
 import re
@@ -6,7 +6,7 @@
 import string
 import getopt
 
-script_path = os.path.realpath(os.path.dirname(sys.argv[0]))
+script_path = "/usr"
 
 ##### DEFAULT SETTINGS #####
 
@@ -24,7 +24,7 @@
 output_dir = '/tmp/oprofile/'
 
 # -f, --fs
-root_fs = script_path + "/root/"
+root_fs = os.path.expanduser("~/tegra")
 
 # -p, --pull
 need_to_pull_binary = 1
@@ -33,7 +33,7 @@
 oprofile_binaries_path = script_path + "/bin/"
 
 # -a, --abi
-oprofile_abi_path = script_path + "/abi/arm_abi"
+oprofile_abi_path = script_path + "/share/tegra/abi/arm_abi"
 
 # inactive now, sorry
 do_backtrace = 0
@@ -260,7 +260,7 @@
             os.system(oprofile_binaries_path + "/opreport " + " --session-dir=." + " -cl" + " --root=" + root_fs + " --symbols " + binary + ' | ' + 'python ' + oprofile_binaries_path + '/gprof2dot.py -f oprofile' + ' | ' + 'dot -Tps -o ' + output_dir + '/callgraph.ps')
             os.system("convert +antialias " + output_dir + '/callgraph.ps' + " " + output_dir + '/callgraph.bmp > /dev/null 2>&1')
         elif current_os == 'Linux':	
-            os.system(oprofile_binaries_path + "/opreport " + " --session-dir=." + " -cl" + " --root=" + root_fs + " --symbols " + binary + ' | ' + 'python ' + oprofile_binaries_path + '/gprof2dot.py -f oprofile' + ' | ' + 'dot -Tpng -o ' + output_dir + '/callgraph.png')
+            os.system(oprofile_binaries_path + "/opreport " + " --session-dir=." + " -cl" + " --root=" + root_fs + " --symbols " + binary + ' | ' + 'gprof2dot -f oprofile' + ' | ' + 'dot -Tpng -o ' + output_dir + '/callgraph.png')
         if current_os == 'Cygwin':
             os.system(image_viewer + ' ' + '`' + 'cygpath -w ' + output_dir + '/callgraph.bmp' + '`')
         elif current_os == 'Linux':
