diff -rupN _master/lib/screenshooter-imgur.c _master-patched/lib/screenshooter-imgur.c
--- _master/lib/screenshooter-imgur.c	2014-12-17 11:21:04.776664000 +0100
+++ _master-patched/lib/screenshooter-imgur.c	2014-12-17 11:13:26.584580000 +0100
@@ -170,7 +170,9 @@ void screenshooter_upload_to_imgur   (co
                                           G_TYPE_STRING, title);
 
   g_signal_connect (job, "ask", G_CALLBACK (cb_ask_for_information), NULL);
-  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded), NULL);
+/*  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded), NULL);
+  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded_copy_to_clipboard), new_last_user);*/
+  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded_copy_to_clipboard), NULL);
   g_signal_connect (job, "error", G_CALLBACK (cb_error), NULL);
   g_signal_connect (job, "finished", G_CALLBACK (cb_finished), dialog);
   g_signal_connect (job, "info-message", G_CALLBACK (cb_update_info), label);
diff -rupN _master/lib/screenshooter-job-callbacks.c _master-patched/lib/screenshooter-job-callbacks.c
--- _master/lib/screenshooter-job-callbacks.c	2014-12-17 11:21:04.776664000 +0100
+++ _master-patched/lib/screenshooter-job-callbacks.c	2014-12-17 12:30:12.930806325 +0100
@@ -95,7 +95,7 @@ void cb_finished (ExoJob *job, GtkWidget
   g_signal_handlers_disconnect_matched (job,
                                         G_SIGNAL_MATCH_FUNC,
                                         0, 0, NULL,
-                                        cb_image_uploaded,
+                                        cb_image_uploaded_copy_to_clipboard,
                                         NULL);
 
   g_signal_handlers_disconnect_matched (job,
@@ -543,3 +543,41 @@ void cb_image_uploaded (ScreenshooterJob
   g_object_unref (bb_buffer);
 }
 
+
+void cb_image_uploaded_copy_to_clipboard (ScreenshooterJob *job,
+						 gchar		  *upload_name,
+						 gchar		 **last_user)
+{
+  const gchar *image_url;
+  gchar *job_type, *title;
+  gchar *last_user_temp;
+
+
+  GtkClipboard *clipboard;
+
+  g_return_if_fail (upload_name != NULL);
+/*  g_return_if_fail (last_user == NULL || *last_user == NULL);*/
+  job_type = g_object_get_data(G_OBJECT (job), "jobtype");
+  if (!strcmp(job_type, "imgur")) {
+    image_url = g_strdup_printf ("http://i.imgur.com/%s.png", upload_name);
+  } else {
+/*    g_return_if_fail (upload_name != NULL);*/
+    g_return_if_fail (last_user == NULL || *last_user == NULL);
+    image_url = g_strdup_printf ("http://www.zimagez.com/zimage/%s.php", upload_name);
+    last_user_temp = g_object_get_data (G_OBJECT (job), "user");
+  } 
+
+  TRACE ("Adding the image URL to the clipboard...");
+
+  clipboard = gtk_clipboard_get_for_display (gdk_display_get_default(), GDK_SELECTION_CLIPBOARD);
+
+  gtk_clipboard_set_text (clipboard, image_url, -1);
+
+  if (last_user) {
+    if (last_user_temp == NULL)
+      last_user_temp = g_strdup ("");
+
+    *last_user = g_strdup (last_user_temp);
+  }
+}
+
diff -rupN _master/lib/screenshooter-job-callbacks.h _master-patched/lib/screenshooter-job-callbacks.h
--- _master/lib/screenshooter-job-callbacks.h	2014-12-17 11:21:04.776664000 +0100
+++ _master-patched/lib/screenshooter-job-callbacks.h	2014-12-17 10:39:12.978859000 +0100
@@ -51,7 +51,8 @@ cb_update_info                     (ExoJ
                                     gchar             *message,
                                     GtkWidget         *label);
 void
-cb_image_uploaded                  (ScreenshooterJob  *job,
+cb_image_uploaded_copy_to_clipboard
+                                   (ScreenshooterJob  *job,
                                     gchar             *upload_name,
                                     gchar            **last_user);
 void
diff -rupN _master/lib/screenshooter-zimagez.c _master-patched/lib/screenshooter-zimagez.c
--- _master/lib/screenshooter-zimagez.c	2014-12-17 11:21:04.776664000 +0100
+++ _master-patched/lib/screenshooter-zimagez.c	2014-12-17 11:18:24.297929000 +0100
@@ -650,7 +650,8 @@ void screenshooter_upload_to_zimagez (co
                                           G_TYPE_STRING, title);
 
   g_signal_connect (job, "ask", G_CALLBACK (cb_ask_for_information), NULL);
-  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded), new_last_user);
+/*  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded), new_last_user);*/
+  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded_copy_to_clipboard), new_last_user);
   g_signal_connect (job, "error", G_CALLBACK (cb_error), NULL);
   g_signal_connect (job, "finished", G_CALLBACK (cb_finished), dialog);
   g_signal_connect (job, "info-message", G_CALLBACK (cb_update_info), label);