diff -rupN master/lib/screenshooter-imgur.c master-cb/lib/screenshooter-imgur.c
--- master/lib/screenshooter-imgur.c	2014-12-16 21:23:09.603909485 +0100
+++ master-cb/lib/screenshooter-imgur.c	2014-12-16 21:20:39.485500251 +0100
@@ -38,6 +38,10 @@ static ScreenshooterJob *imgur_upload_to
 static gboolean          imgur_upload_job          (ScreenshooterJob  *job,
                                                     GValueArray       *param_values,
                                                     GError           **error);
+static void              cb_image_uploaded_copy_to_clipboard         
+						   (ScreenshooterJob  *job,
+                                                    gchar             *upload_name,
+                                                    gchar            **last_user);
 static void              cb_image_uploaded         (ScreenshooterJob  *job,
                                                     gchar             *upload_name,
                                                     gchar            **last_user);
@@ -177,6 +181,35 @@ static ScreenshooterJob
 
 
 
+static void cb_image_uploaded_copy_to_clipboard (ScreenshooterJob *job,
+						 gchar		  *upload_name,
+						 gchar		 **last_user)
+{
+  gchar *last_user_temp;
+  const gchar *image_url;
+
+  GtkClipboard *clipboard;
+
+  g_return_if_fail (upload_name != NULL);
+  g_return_if_fail (last_user == NULL || *last_user == NULL);
+  image_url = g_strdup_printf ("http://i.imgur.com/%s.png", upload_name);
+
+  TRACE ("Adding the image URL to the clipboard...");
+
+  clipboard = gtk_clipboard_get_for_display (gdk_display_get_default(), GDK_SELECTION_CLIPBOARD);
+
+  gtk_clipboard_set_text (clipboard, image_url, -1);
+
+  last_user_temp = g_object_get_data (G_OBJECT (job), "user");
+  
+  if (last_user_temp == NULL)
+    last_user_temp = g_strdup ("");
+
+  *last_user = g_strdup (last_user_temp);
+}
+
+
+
 static void cb_image_uploaded (ScreenshooterJob  *job,
                                gchar             *upload_name,
                                gchar            **last_user)
@@ -378,7 +411,7 @@ static void cb_finished (ExoJob *job, Gt
   g_signal_handlers_disconnect_matched (job,
                                         G_SIGNAL_MATCH_FUNC,
                                         0, 0, NULL,
-                                        cb_image_uploaded,
+                                        cb_image_uploaded_copy_to_clipboard,
                                         NULL);
 
   g_signal_handlers_disconnect_matched (job,
@@ -499,7 +532,7 @@ void screenshooter_upload_to_imgur   (co
 
   job = imgur_upload_to_imgur (image_path, last_user, title);
 
-  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded), new_last_user);
+  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded_copy_to_clipboard), new_last_user);
   g_signal_connect (job, "error", G_CALLBACK (cb_error), NULL);
   g_signal_connect (job, "finished", G_CALLBACK (cb_finished), dialog);
   g_signal_connect (job, "info-message", G_CALLBACK (cb_update_info), label);
diff -rupN master/lib/screenshooter-zimagez.c master-cb/lib/screenshooter-zimagez.c
--- master/lib/screenshooter-zimagez.c	2014-12-16 21:23:09.607242937 +0100
+++ master-cb/lib/screenshooter-zimagez.c	2014-12-16 21:21:33.250647548 +0100
@@ -68,6 +68,10 @@ static void              cb_ask_for_info
                                                     GtkListStore      *liststore,
                                                     const gchar       *message,
                                                     gpointer           unused);
+static void              cb_image_uploaded_copy_to_clipboard         
+						   (ScreenshooterJob  *job,
+                                                    gchar             *upload_name,
+                                                    gchar            **last_user);
 static void              cb_image_uploaded         (ScreenshooterJob  *job,
                                                     gchar             *upload_name,
                                                     gchar            **last_user);
@@ -868,6 +872,35 @@ cb_ask_for_information (ScreenshooterJob
 
 
 
+static void cb_image_uploaded_copy_to_clipboard (ScreenshooterJob *job,
+						 gchar		  *upload_name,
+						 gchar		 **last_user)
+{
+  gchar *last_user_temp;
+  const gchar *image_url;
+
+  GtkClipboard *clipboard;
+
+  g_return_if_fail (upload_name != NULL);
+  g_return_if_fail (last_user == NULL || *last_user == NULL);
+  image_url = g_strdup_printf ("http://www.zimagez.com/zimage/%s.php", upload_name);
+
+  TRACE ("Adding the image URL to the clipboard...");
+
+  clipboard = gtk_clipboard_get_for_display (gdk_display_get_default(), GDK_SELECTION_CLIPBOARD);
+
+  gtk_clipboard_set_text (clipboard, image_url, -1);
+
+  last_user_temp = g_object_get_data (G_OBJECT (job), "user");
+  
+  if (last_user_temp == NULL)
+    last_user_temp = g_strdup ("");
+
+  *last_user = g_strdup (last_user_temp);
+}
+
+
+
 static void cb_image_uploaded (ScreenshooterJob  *job,
                                gchar             *upload_name,
                                gchar            **last_user)
@@ -1071,7 +1104,7 @@ static void cb_finished (ExoJob *job, Gt
   g_signal_handlers_disconnect_matched (job,
                                         G_SIGNAL_MATCH_FUNC,
                                         0, 0, NULL,
-                                        cb_image_uploaded,
+                                        cb_image_uploaded_copy_to_clipboard,
                                         NULL);
 
   g_signal_handlers_disconnect_matched (job,
@@ -1199,7 +1232,7 @@ void screenshooter_upload_to_zimagez (co
   job = zimagez_upload_to_zimagez (image_path, last_user, title);
 
   g_signal_connect (job, "ask", G_CALLBACK (cb_ask_for_information), NULL);
-  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded), new_last_user);
+  g_signal_connect (job, "image-uploaded", G_CALLBACK (cb_image_uploaded_copy_to_clipboard), new_last_user);
   g_signal_connect (job, "error", G_CALLBACK (cb_error), NULL);
   g_signal_connect (job, "finished", G_CALLBACK (cb_finished), dialog);
   g_signal_connect (job, "info-message", G_CALLBACK (cb_update_info), label);
