--- DefaultRules.mk	2008-08-07 17:58:34.000000000 +0200
+++ DefaultRules.mk.new	2009-12-10 20:32:15.000000000 +0100
@@ -73,7 +73,7 @@
 $(OBJDIR)/%$(OBJ_SUFFIX): %.cpp
 	@echo using BD = $$BD
 	$(CC) $(CCFLAGS) $(CCLOCALFLAGS) $(COMPONLY_OPTION) $(INCL) \
-	$(INC_OPTION)$(OBJDIR) $(INC_OPTION).						\
+	$(INC_OPTION)$(OBJDIR) $(INCLUDE_EXTRA_QT4) $(INC_OPTION).						\
 	 $(OBJ_OPTION) $@ $< $($(PROJ)SODEF)
 $(OBJDIR)/%$(OBJ_SUFFIX): %.mm
 	@echo using BD = $$BD
@@ -163,7 +163,7 @@
 	@-rm -f $@
 	@echo '# Module dependencies' > $@
 	@$(CC) $(DEPEND_OPTION) $< $(CCFLAGS) $(CCLOCALFLAGS) $(INCL) 	\
-	 $(INC_OPTION)$(OBJDIR) $(INC_OPTION). 							\
+	 $(INC_OPTION)$(OBJDIR) $(INC_OPTION). $(INCLUDE_EXTRA_QT4)							\
 	 | $(SED) -e 's/^\([^:]*:\)/$(OBJDIR)\/\1/1' 					\
 	 		  -e 's/\/usr\/include\/[^ ]*//g'						\
 	 		  -e 's/\/usr\/Software\/[^ ]*//g'						\
@@ -386,10 +386,10 @@
 	$(MOC) $(call cnvUnix2Win,$<) -i -DOSG_WITH_QT -o $@
 else
 $(OBJDIR)/%_qt_moc.cpp: %_qt.h
-	$(MOC) $< -i -DOSG_WITH_QT -I$(INCLUDE_QT4)/QtDesigner -o $@
+	$(MOC) $< -DOSG_WITH_QT -I$(INCLUDE_QT4)/QtDesigner -o $@
 
 $(OBJDIR)/%_qt_moc.cpp: %_qt.cpp
-	$(MOC) $< -i -DOSG_WITH_QT -I$(INCLUDE_QT4)/QtDesigner -o $@
+	$(MOC) $< -DOSG_WITH_QT -I$(INCLUDE_QT4)/QtDesigner $(INCLUDE_EXTRA_QT4) -o $@
 endif
 else
 ifeq ($(OS_BASE), cygwin)
