diff --git a/Client/Makefile b/Client/Makefile
index 5805d7b..0d91f77 100644
--- beaker/Client/Makefile
+++ beaker/Client/Makefile.fixed
@@ -12,7 +12,7 @@ include ../py_rules.mk
 .PHONY: build
 build: $(_SETUP_PY)
 	env PYTHONPATH=../Common$${PYTHONPATH:+:$$PYTHONPATH} \
-	    python $(SETUP_PY) build \
+	    python2 $(SETUP_PY) build \
 	    | tee build.out
 	@if grep -q ImportError build.out ; then \
 	    echo "ImportErrors in build output, did Sphinx autodoc fail?" ; \
diff --git a/py_rules.mk b/py_rules.mk
index c84d279..6bffc20 100644
--- beaker/py_rules.mk
+++ beaker/py_rules.mk.fixed
@@ -24,16 +24,16 @@ $(_SETUP_PY):	$(_SETUP_PY).in
 	sed -e 's/@VERSION@/$(PKGVERSION)/g' < $< > $@
 
 py-build-ext:	$(_SETUP_PY) $(PY_SOURCES)
-	python $(SETUP_PY) build_ext -i
+	python2 $(SETUP_PY) build_ext -i
 
 py-build:   $(_SETUP_PY) $(PY_SOURCES)
-	python $(SETUP_PY) build
+	python2 $(SETUP_PY) build
 
 py-install:	$(_SETUP_PY)
-	python $(SETUP_PY) install -O1 --skip-build --root $(DESTDIR)
+	python2 $(SETUP_PY) install -O1 --skip-build --root $(DESTDIR)
 
 py-clean:	$(_SETUP_PY)
-	python $(SETUP_PY) clean; \
+	python2 $(SETUP_PY) clean; \
 	rm -f $(SETUP_PY); \
 	rm -rf build
 
