Index: configure.ac
===================================================================
RCS file: /cvsroot/minidlna/minidlna/configure.ac,v
retrieving revision 1.3
diff -u -r1.3 configure.ac
--- configure.ac	21 Sep 2011 21:55:56 -0000	1.3
+++ configure.ac	14 Jan 2013 14:33:50 -0000
@@ -419,17 +419,17 @@
     if test -n "$dir"; then
         LDFLAGS="$LDFLAGS -L$dir/lib"
     fi
-    AC_CHECK_LIB([avformat], [av_open_input_file], [LIBAVFORMAT_LIBS="-lavformat"], [unset ac_cv_lib_avformat_av_open_input_file; LDFLAGS="$LDFLAGS_SAVE"; continue])
+    AC_CHECK_LIB([avformat], [avformat_open_input], [LIBAVFORMAT_LIBS="-lavformat"], [unset ac_cv_lib_avformat_avformat_open_input; LDFLAGS="$LDFLAGS_SAVE"; continue])
     AC_SUBST(LIBJPEG_LIBS)
     break
 done
-test x"$ac_cv_lib_avformat_av_open_input_file" = x"yes" || AC_MSG_ERROR([Could not find libavformat - part of ffmpeg])
+test x"$ac_cv_lib_avformat_avformat_open_input" = x"yes" || AC_MSG_ERROR([Could not find libavformat - part of ffmpeg])
 AC_SUBST(LIBAVFORMAT_LIBS)
 
 AC_CHECK_LIB(avutil ,[av_rescale_q], [LIBAVUTIL_LIBS="-lavutil"], [AC_MSG_ERROR([Could not find libavutil - part of ffmpeg])])
 AC_SUBST(LIBAVUTIL_LIBS)
 
-AC_CHECK_LIB(avcodec ,[avcodec_init], [LIBAVCODEC_LIBS="-lavcodec"], [AC_MSG_ERROR([Could not find libavcodec - part of ffmpeg])])
+AC_CHECK_LIB(avcodec ,[av_parser_init], [LIBAVCODEC_LIBS="-lavcodec"], [AC_MSG_ERROR([Could not find libavcodec - part of ffmpeg])])
 AC_SUBST(LIBAVCODEC_LIBS)
 
 AC_CHECK_LIB(pthread, pthread_create)
Index: metadata.c
===================================================================
RCS file: /cvsroot/minidlna/minidlna/metadata.c,v
retrieving revision 1.82
diff -u -r1.82 metadata.c
--- metadata.c	16 Sep 2011 23:39:57 -0000	1.82
+++ metadata.c	14 Jan 2013 14:33:54 -0000
@@ -75,6 +75,10 @@
 #include <avformat.h>
 #endif
 
+#define AVMetadataTag AVDictionaryEntry
+#define av_metadata_get av_dict_get
+#define AV_METADATA_IGNORE_SUFFIX AV_DICT_IGNORE_SUFFIX
+
 #include "upnpglobalvars.h"
 #include "upnpreplyparse.h"
 #include "metadata.h"
