diff -Naur a/scangearmp/configure.in b/scangearmp/configure.in
--- a/scangearmp/configure.in	2008-11-19 00:32:24.000000000 -0500
+++ b/scangearmp/configure.in	2013-04-23 18:42:11.843793004 -0400
@@ -13,13 +13,12 @@
 AC_SUBST(enable_libpath)
 
 AM_INIT_AUTOMAKE(scangearmp, 1.30-1)
-AM_CONFIG_HEADER(config.h)
+AM_PROG_LIBTOOL
 AC_PROG_LIBTOOL
 AM_MAINTAINER_MODE
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AC_PROG_RANLIB

