--- Makefile	2004-07-14 17:48:52.000000000 -0400
+++ Makefile.ok	2009-04-27 12:24:36.842238070 -0400
@@ -2,13 +2,13 @@
 # Allan Crossman
 
 # Where the graphics and sound files go...
-DATAPATH=/usr/local/share/komidata/
+DATAPATH=/usr/share/komidata/
 
 # Where to put the binary...
-BINPATH=/usr/local/bin/
+BINPATH=/usr/bin/
 
 # Where to put the manpage...
-MANPATH=/usr/local/man/man6/
+MANPATH=/usr/man/man6/
 
 MIXER=-lSDL_mixer
 SDL_LIB=$(shell sdl-config --static-libs) $(MIXER)
