diff -ur a/twcw-2.0/twcw.desktop b/twcw-2.0/twcw.desktop
--- a/twcw-2.0/twcw.desktop	2008-07-07 20:28:54.000000000 +0300
+++ b/twcw-2.0/twcw.desktop	2010-03-09 23:56:40.000000000 +0200
@@ -1,10 +1,8 @@
 [Desktop Entry]
 Name=twcw
-Name[en_US]=twcw
 Comment=Send and Practice Morse Code from the keyboard
-Comment[en_US]=Send and Practice Morse Code from the keyboard
 Exec=twcw
-Icon=twcw.png
+Icon=twcw
 Terminal=false
 Type=Application
-Categories=HamRadio
+Categories=HamRadio;
