diff --git a/src/game.cpp b/src/game.cpp
index 9a512ce..0161f90 100644
--- a/src/game.cpp
+++ b/src/game.cpp
@@ -593,7 +593,9 @@ void end_game(int err)
 bool initialize_vehicletypes()
 {
    CMarkup vtfile;
-   if(!vtfile.Load("art/vehicles.xml"))
+   if(!vtfile.Load("art/vehicles.xml")
+		   && !vtfile.Load("/usr/share/lcs/art/vehicles.xml")
+		   && !vtfile.Load("/usr/local/share/lcs/art/vehicles.xml"))
    { //File is missing or not valid XML.
       addstr("FAILED to load vehicles.xml!");
       getch();
@@ -617,7 +619,9 @@ bool initialize_vehicletypes()
 bool initialize_cliptypes()
 {
    CMarkup ctfile;
-   if(!ctfile.Load("art/clips.xml"))
+  if(!ctfile.Load("art/clips.xml")
+		   && !ctfile.Load("/usr/share/lcs/art/clips.xml")
+		   && !ctfile.Load("/usr/local/share/lcs/art/clips.xml"))
    { //File is missing or not valid XML.
       addstr("FAILED to load clips.xml!");
       getch();
@@ -637,7 +641,9 @@ bool initialize_cliptypes()
 bool initialize_weapontypes()
 {
    CMarkup wtfile;
-   if(!wtfile.Load("art/weapons.xml"))
+   if(!wtfile.Load("art/weapons.xml")
+		   && !wtfile.Load("/usr/share/lcs/art/weapons.xml")
+		   && !wtfile.Load("/usr/local/share/lcs/art/weapons.xml"))
    { //File is missing or not valid XML.
       addstr("FAILED to load weapons.xml!");
       getch();
@@ -658,7 +664,9 @@ bool initialize_armortypes()
 {
    //Armors
    CMarkup atfile;
-   if(!atfile.Load("art/armors.xml"))
+   if(!atfile.Load("art/armors.xml")
+		   && !atfile.Load("/usr/share/lcs/art/armors.xml")
+		   && !atfile.Load("/usr/local/share/lcs/art/armors.xml"))
    { //File is missing or not valid XML.
       addstr("FAILED to load armors.xml!");
       getch();
@@ -674,7 +682,9 @@ bool initialize_armortypes()
    }
    
    //Masks
-   if(!atfile.Load("art/masks.xml"))
+   if(!atfile.Load("art/masks.xml")
+		   && !atfile.Load("/usr/share/lcs/art/masks.xml")
+		   && !atfile.Load("/usr/local/share/lcs/art/masks.xml"))
    { //File is missing or not valid XML.
       addstr("FAILED to load masks.xml!");
       getch();
@@ -714,7 +724,9 @@ bool initialize_armortypes()
 bool initialize_loottypes()
 {
    CMarkup ltfile;
-   if(!ltfile.Load("art/loot.xml"))
+   if(!ltfile.Load("art/loot.xml")
+		   && !ltfile.Load("/usr/share/lcs/art/loot.xml")
+		   && !ltfile.Load("/usr/local/share/lcs/art/loot.xml"))
    { //File is missing or not valid XML.
       addstr("FAILED to load loot.xml!");
       getch();
