--- gsdjvu-1.3/build-gsdjvu.orig	2007-09-22 03:10:30.000000000 -0300
+++ gsdjvu-1.3/build-gsdjvu	2009-04-19 14:36:20.732572523 -0300
@@ -193,17 +193,23 @@
 rm -f 2>/dev/null gsdjvu
 cp $gsfiles/gsdjvu gsdjvu
 
-rm -f 2>/dev/null src/gdevdjvu.c
-cp $gsfiles/gdevdjvu.c src/gdevdjvu.c
+rm -f 2>/dev/null base/gdevdjvu.c
+cp $gsfiles/gdevdjvu.c base/gdevdjvu.c
 
 rm -f 2>/dev/null lib/ps2utf8.ps
 cp $gsfiles/ps2utf8.ps lib/ps2utf8.ps
 
-test -r src/contrib.mak.gsdjvu && \
-  mv src/contrib.mak.gsdjvu src/contrib.mak
-cp src/contrib.mak src/contrib.mak.gsdjvu
-grep -q djvusep src/contrib.mak || \
-  cat $gsfiles/gsdjvu.mak >> src/contrib.mak
+test -r base/contrib.mak.gsdjvu && \
+  mv base/contrib.mak.gsdjvu base/contrib.mak
+cp base/contrib.mak base/contrib.mak.gsdjvu
+grep -q djvusep base/contrib.mak || \
+  cat $gsfiles/gsdjvu.mak >> base/contrib.mak
+
+sed -i -e '/$(EXTRA_INIT_FILES)/ a\ps2utf8.ps \\' "base/unixinst.mak" || exit 1
+
+for p in ../*.patch; do
+    patch -Np1 -i "$p" || exit 1
+done
 
 for lib in jpeg libpng zlib ; do
  for libn in $lib-* ; do
@@ -291,7 +297,7 @@
     #read tmp
     
     DESTDIR="${DESTDIR-}"
-    INSTALL="${INSTALL-./src/instcopy}"
+    INSTALL="${INSTALL-./base/instcopy}"
     MKDIRP="${MKDIRP-mkdir -p}"
     
     gsdir="$builddir/INST/gsdjvu"
@@ -320,9 +326,9 @@
         for n in `find Resource -type f -print` ; do
             $INSTALL -m 644 $n "$DESTDIR$gsdir/$n"
         done
-        test -r lib/gs_res.ps.gsdjvu || \
-            cp lib/gs_res.ps lib/gs_res.ps.gsdjvu
-        sed < lib/gs_res.ps.gsdjvu > lib/gs_res.ps \
+        test -r Resource/Init/gs_res.ps.gsdjvu || \
+            cp Resource/Init/gs_res.ps Resource/Init/gs_res.ps.gsdjvu
+        sed < Resource/Init/gs_res.ps.gsdjvu > Resource/Init/gs_res.ps \
          -e 's:(/\(Resource/[a-zA-Z/]*\)):(\1) findlibfile {pop} {pop &} ifelse:'
     fi
     
