--- toshset-1.76/Makefile.in~	2010-10-29 11:31:34.743334921 +0200
+++ toshset-1.76/Makefile.in	2010-10-29 11:33:09.416668169 +0200
@@ -304,11 +304,9 @@
 install: all
 	@mkdir -p $(DESTDIR)$(BINDESTDIR)
 	@mkdir -p $(DESTDIR)$(MANDESTDIR)
-	$(DESTDIR)/pm/sleep.d/
 	for program in $(PROGRAMS); do \
 		$(BININSTALL) $$program $(DESTDIR)/$(BINDESTDIR)/$$program;\
 	done
-	$(BININSTALL) novatel_3g_suspend $(DESTDIR)/pm/sleep.d/
 	(cd ..;\
 	for manpage in $(MANPAGES); do \
 		$(MANINSTALL) $$manpage $(DESTDIR)/$(MANDESTDIR)/$$manpage;\
