diff -aur toshset-1.76.pristine/cdsString.cc toshset-1.76.new/cdsString.cc
--- toshset-1.76.pristine/cdsString.cc	2010-02-12 18:05:53.000000000 +0100
+++ toshset-1.76.new/cdsString.cc	2013-03-18 20:46:54.159308582 +0100
@@ -503,7 +503,7 @@
   // read in up to (not including) white space
   //
 {
- long flags = s.flags();
+ std::ios_base::fmtflags flags = s.flags();
  if (flags & ios::skipws) s >> ws;
  s.unsetf(ios::skipws);
  x = "";
