diff -urN dosbox.orig/src/gui/sdlmain.cpp dosbox/src/gui/sdlmain.cpp
--- dosbox.orig/src/gui/sdlmain.cpp	2014-02-20 20:13:16.465532013 -0500
+++ dosbox/src/gui/sdlmain.cpp	2014-02-20 20:38:10.422747196 -0500
@@ -240,6 +240,7 @@
 	}
 
 	if(paused) strcat(title," PAUSED");
+	if(sdl.mouse.locked) strcat(title," (Press Ctrl-F10 to release mouse)");
 	SDL_WM_SetCaption(title,VERSION);
 }
 
@@ -385,6 +386,7 @@
 		(sdl.draw.callback)( GFX_CallBackReset );
 	GFX_Start();
 	CPU_Reset_AutoAdjust();
+	GFX_SetTitle(-1,-1,false);
 }
 
 void GFX_ForceFullscreenExit(void) {
@@ -718,6 +720,7 @@
 		if (sdl.mouse.autoenable || !sdl.mouse.autolock) SDL_ShowCursor(SDL_ENABLE);
 	}
         mouselocked=sdl.mouse.locked;
+        	GFX_SetTitle(-1,-1,false); 
 }
 
 void GFX_UpdateSDLCaptureState(void) {
