--- rtsx.c	2011-12-05 04:41:04.000000000 +0100
+++ rtsx.c	2012-05-19 09:34:59.497714741 +0200
@@ -492,14 +492,18 @@
 		 * the maximum known LUN
 		 */
 		else if (chip->srb->device->id) {
-			printk(KERN_ERR "Bad target number (%d:%d)\n",
-				  chip->srb->device->id, chip->srb->device->lun);
+			/* remove Bad target kernel messages
+			 * printk(KERN_ERR "Bad target number (%d:%d)\n",
+			 *	  chip->srb->device->id, chip->srb->device->lun);
+			 */
 			chip->srb->result = DID_BAD_TARGET << 16;
 		}
 
 		else if (chip->srb->device->lun > chip->max_lun) {
-			printk(KERN_ERR "Bad LUN (%d:%d)\n",
-				  chip->srb->device->id, chip->srb->device->lun);
+			/* remove Bad LUN kernel messages
+			 * printk(KERN_ERR "Bad LUN (%d:%d)\n",
+			 * 	  chip->srb->device->id, chip->srb->device->lun);
+			 */
 			chip->srb->result = DID_BAD_TARGET << 16;
 		}
 
