--- rtsx.c	2011-12-05 04:41:04.000000000 +0100
+++ rtsx.c	2012-06-18 20:43:47.713726220 +0200
@@ -995,6 +995,11 @@
 		goto errout;
 	}
 
+        /* set the supported max_lun and max_id for the scsi host
+         * NOTE: the minimal value of max_id is 1 */
+        host->max_id = 1;
+        host->max_lun = dev->chip->max_lun;
+
 	err = scsi_add_host(host, &pci->dev);
 	if (err) {
 		printk(KERN_ERR "Unable to add the scsi host\n");
