diff -BrauN repo-old/repo repo/repo
--- repo-old/repo	2011-02-10 20:21:39.690000130 +0100
+++ repo/repo	2011-02-10 20:21:46.773333463 +0100
@@ -20,7 +20,7 @@
 # limitations under the License.
 
 magic='--calling-python-from-/bin/sh--'
-"""exec" python -E "$0" "$@" """#$magic"
+"""exec" python2 -E "$0" "$@" """#$magic"
 if __name__ == '__main__':
   import sys
   if sys.argv[-1] == '#%s' % magic:
@@ -579,7 +579,8 @@
     main = my_main
 
   ver_str = '.'.join(map(lambda x: str(x), VERSION))
-  me = [main,
+  me = ["/usr/bin/python2",
+  		main,
         '--repo-dir=%s' % dir,
         '--wrapper-version=%s' % ver_str,
         '--wrapper-path=%s' % wrapper_path,
@@ -587,7 +588,7 @@
   me.extend(orig_args)
   me.extend(extra_args)
   try:
-    os.execv(main, me)
+    os.execv("/usr/bin/python2", me)
   except OSError, e:
     print >>sys.stderr, "fatal: unable to start %s" % main
     print >>sys.stderr, "fatal: %s" % e
