diff --git a/repo b/repo
index 70f4157..5a6c2d8 100755
--- a/repo
+++ b/repo
@@ -20,7 +20,7 @@ REPO_REV='stable'
 # limitations under the License.
 
 magic='--calling-python-from-/bin/sh--'
-"""exec" python -E "$0" "$@" """#$magic"
+"""exec" python2 -E "$0" "$@" """#$magic"
 if __name__ == '__main__':
   import sys
   if sys.argv[-1] == '#%s' % magic:
@@ -722,7 +722,8 @@ def main(orig_args):
     repo_main = my_main
 
   ver_str = '.'.join(map(lambda x: str(x), VERSION))
-  me = [repo_main,
+  me = [ "/usr/bin/python2",
+        repo_main,
         '--repo-dir=%s' % rel_repo_dir,
         '--wrapper-version=%s' % ver_str,
         '--wrapper-path=%s' % wrapper_path,
@@ -730,7 +731,7 @@ def main(orig_args):
   me.extend(orig_args)
   me.extend(extra_args)
   try:
-    os.execv(repo_main, me)
+    os.execv("/usr/bin/python2", me)
   except OSError as e:
     print >>sys.stderr, "fatal: unable to start %s" % repo_main
     print >>sys.stderr, "fatal: %s" % e
