--- a/sharetray.cpp	2012-02-22 16:19:21.000000000 +0100
+++ b/sharetray.cpp	2012-12-09 20:01:25.140573000 +0100
@@ -33,6 +33,7 @@
 #include <QDateTime>
 #include <grp.h>
 #include <QProcess>
+#include <unistd.h>
 #define STAT_ACT_COUNT 10
 
 #define VERSION "3.0.1.5"
@@ -175,6 +176,7 @@
 
 
 	actStop->setEnabled ( false );
+	slotStartSharing();
 
 	// unix signals (TERM, INT) are piped into a unix socket and will raise Qt events
 	if (::socketpair(AF_UNIX, SOCK_STREAM, 0, sigkeybintFd))
@@ -322,7 +324,7 @@
 	if ( serverSocket->listen ( socketFname ) )
 	{
 
-		chown ( socketFname.toAscii(),getuid(),getgrnam ( "x2godesktopsharing" )->gr_gid );
+		chown ( socketFname.toAscii(),getuid(),getgrnam ( "x2godesktopshare" )->gr_gid );
 		QFile::setPermissions ( socketFname,
 		                        QFile::ReadOwner|QFile::WriteOwner|QFile::ReadGroup|QFile::WriteGroup );
 		connect ( serverSocket,SIGNAL ( newConnection() ),
