diff -rupN src/RtAudio/RtAudio.cpp src-new/RtAudio/RtAudio.cpp
--- src/RtAudio/RtAudio.cpp	2012-09-15 15:12:04.000000000 +0200
+++ src-new/RtAudio/RtAudio.cpp	2013-10-28 18:25:37.016006307 +0100
@@ -5657,7 +5657,8 @@ bool RtApiAlsa :: probeDeviceOpen( unsig
         if ( result < 0 ) break;
         if ( subdevice < 0 ) break;
         if ( nDevices == device ) {
-          sprintf( name, "hw:%d,%d", card, subdevice );
+          //sprintf( name, "hw:%d,%d", card, subdevice );
+	  sprintf( name, "pulse");
           snd_ctl_close( chandle );
           goto foundDevice;
         }
@@ -5696,6 +5697,7 @@ bool RtApiAlsa :: probeDeviceOpen( unsig
 
   snd_pcm_t *phandle;
   int openMode = SND_PCM_ASYNC;
+  printf( "pcm name %s\n", name);
   result = snd_pcm_open( &phandle, name, stream, openMode );
   if ( result < 0 ) {
     if ( mode == OUTPUT )
