--- src/makefile	2011-06-18 18:45:25.942821000 +0000
+++ src/makefile	2011-06-18 18:48:36.902821002 +0000
@@ -18,6 +18,8 @@
 INSTALL_SOCKET_LIB=$(INSTALL_TOP_LIB)/socket
 INSTALL_MIME_SHARE=$(INSTALL_TOP_SHARE)/mime
 INSTALL_MIME_LIB=$(INSTALL_TOP_LIB)/mime
+#
+INSTALL_UNIX_LIB=$(INSTALL_SOCKET_LIB)
 
 #------
 # Supported platforms
@@ -173,7 +175,7 @@
 	@echo "where PLATFORM is one of these:"
 	@echo "   $(PLATS)"
 
-all: $(SOCKET_SO) $(MIME_SO)
+all: $(SOCKET_SO) $(MIME_SO) $(UNIX_SO)
 
 $(SOCKET_SO): $(SOCKET_OBJS)
 	$(LD) $(SOCKET_OBJS) $(LDFLAGS)$@ 
@@ -193,6 +195,8 @@
 	$(INSTALL_EXEC) $(SOCKET_SO) $(INSTALL_SOCKET_LIB)/core.$(SO)
 	mkdir -p $(INSTALL_MIME_LIB)
 	$(INSTALL_EXEC) $(MIME_SO) $(INSTALL_MIME_LIB)/core.$(SO)
+	mkdir -p $(INSTALL_UNIX_LIB)
+	$(INSTALL_EXEC) $(UNIX_SO) $(INSTALL_UNIX_LIB)/$(UNIX_SO)
 
 local:
 	$(MAKE) install INSTALL_TOP_LIB=.. INSTALL_TOP_SHARE=..
