diff -aur vinterm-0.5.0/Makefile vinterm-0.5.0.patched/Makefile
--- vinterm-0.5.0/Makefile	2013-08-19 18:34:06.000000000 +0200
+++ vinterm-0.5.0.patched/Makefile	2013-08-19 23:20:55.953442448 +0200
@@ -73,27 +73,26 @@
 
 install: all
 	@echo installing executable file to ${DESTDIR}${PREFIX}/bin
-	@mkdir -p ${DESTDIR}${PREFIX}/bin
-	@cp -f vinterm ${DESTDIR}${PREFIX}/bin
-	@chmod 755 ${DESTDIR}${PREFIX}/bin/vinterm
+	@install -d ${DESTDIR}${PREFIX}/bin
+	@install -Dm755 vinterm ${DESTDIR}${PREFIX}/bin
 	@echo installing data to ${DESTDIR}${VINTERMPREFIX}
-	@mkdir -p ${DESTDIR}${VINTERMPREFIX}
-	@cp -f ${DATA} ${DESTDIR}${VINTERMPREFIX}
+	@install -d ${DESTDIR}${VINTERMPREFIX}
+	@install -Dm644 ${DATA} ${DESTDIR}${VINTERMPREFIX}
 	@echo installing manual page to ${DESTDIR}${MANPREFIX}/man1
-	@mkdir -p ${DESTDIR}${MANPREFIX}/man1
+	@install -d ${DESTDIR}${MANPREFIX}/man1
 	@sed "s/VERSION/${VERSION}/g" < vinterm.1 > ${DESTDIR}${MANPREFIX}/man1/vinterm.1
 	@chmod 644 ${DESTDIR}${MANPREFIX}/man1/vinterm.1
 	@echo compiling terminfo data
 	@tic -s data/vinterm.info
 	@echo installing icons
-	@cp -f icon/icon_16.png  /usr/share/icons/hicolor/16x16/apps/vinterm.png
-	@cp -f icon/icon_22.png  /usr/share/icons/hicolor/22x22/apps/vinterm.png
-	@cp -f icon/icon_32.png  /usr/share/icons/hicolor/32x32/apps/vinterm.png
-	@cp -f icon/icon_48.png  /usr/share/icons/hicolor/48x48/apps/vinterm.png
-	@cp -f icon/icon_64.png  /usr/share/icons/hicolor/64x64/apps/vinterm.png
-	@cp -f icon/icon_128.png /usr/share/icons/hicolor/128x128/apps/vinterm.png
-	@cp -f icon/icon.svg     /usr/share/icons/hicolor/scalable/apps/vinterm.svg
-	@cp -f data/vinterm.desktop /usr/share/applications
+	@install -Dm644 icon/icon_16.png  $(DESTDIR)$(PREFIX)/share/icons/hicolor/16x16/apps/vinterm.png
+	@install -Dm644 icon/icon_22.png  $(DESTDIR)$(PREFIX)/share/icons/hicolor/22x22/apps/vinterm.png
+	@install -Dm644 icon/icon_32.png  $(DESTDIR)$(PREFIX)/share/icons/hicolor/32x32/apps/vinterm.png
+	@install -Dm644 icon/icon_48.png  $(DESTDIR)$(PREFIX)/share/icons/hicolor/48x48/apps/vinterm.png
+	@install -Dm644 icon/icon_64.png  $(DESTDIR)$(PREFIX)/share/icons/hicolor/64x64/apps/vinterm.png
+	@install -Dm644 icon/icon_128.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/128x128/apps/vinterm.png
+	@install -Dm644 icon/icon.svg     $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/apps/vinterm.svg
+	@install -Dm644 data/vinterm.desktop $(DESTDIR)$(PREFIX)/share/applications/vinterm.desktop
 
 # TODO - check data
 uninstall:
@@ -105,13 +104,13 @@
 	@echo removing manual page from ${DESTDIR}${MANPREFIX}/man1
 	@rm -f ${DESTDIR}${MANPREFIX}/man1/vinterm.1
 	@echo removing icons
-	@rm -f /usr/share/icons/hicolor/16x16/apps/vinterm.png
-	@rm -f /usr/share/icons/hicolor/22x22/apps/vinterm.png
-	@rm -f /usr/share/icons/hicolor/32x32/apps/vinterm.png
-	@rm -f /usr/share/icons/hicolor/48x48/apps/vinterm.png
-	@rm -f /usr/share/icons/hicolor/64x64/apps/vinterm.png
-	@rm -f /usr/share/icons/hicolor/128x128/apps/vinterm.png
-	@rm -f /usr/share/icons/hicolor/scalable/apps/vinterm.svg
-	@rm -f /usr/share/applications/vinterm.desktop
+	@rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/16x16/apps/vinterm.png
+	@rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/22x22/apps/vinterm.png
+	@rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/32x32/apps/vinterm.png
+	@rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/48x48/apps/vinterm.png
+	@rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/64x64/apps/vinterm.png
+	@rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/128x128/apps/vinterm.png
+	@rm -f $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/apps/vinterm.svg
+	@rm -f $(DESTDIR)$(PREFIX)/share/applications/vinterm.desktop
 
 .PHONY: all options clean dist install uninstall
