diff -par -u2 idleize-2.0-ori/Makefile idleize-2.0/Makefile
--- idleize-2.0-ori/Makefile	2002-11-11 01:43:12.000000000 +0100
+++ idleize-2.0/Makefile	2012-07-15 15:02:32.415434237 +0200
@@ -29,14 +29,14 @@ MANEXT = 1
 
 ### Linux
-#OWNER = root
-#GROUP = root
-#MODES = 111
-
-### MacOS X
-LIBS = -framework IOKit -framework CoreFoundation
 OWNER = root
-GROUP = wheel
+GROUP = root
 MODES = 111
 
+### MacOS X
+#LIBS = -framework IOKit -framework CoreFoundation
+#OWNER = root
+#GROUP = wheel
+#MODES = 111
+
 ### Tru64
 #OWNER = bin
Nur in idleize-2.0: README.
Nur in idleize-2.0: idleize.
Nur in idleize-2.0: idleize.o.
Nur in idleize-2.0: install.sh.
