diff -rNu rt3090-2.4.0.4/include/rt_config.h rt3090-2.4.0.4-new/include/rt_config.h
--- rt3090-2.4.0.4/include/rt_config.h	2011-02-03 20:10:16.000000000 +0800
+++ rt3090-2.4.0.4-new/include/rt_config.h	2013-06-13 11:22:48.000000000 +0800
@@ -28,6 +28,11 @@
 #ifndef	__RT_CONFIG_H__
 #define	__RT_CONFIG_H__
 
+#define __devinit
+#define __devexit
+#define __devexit_p(x) (x)
+#define __devinitdata
+
 #define VENDOR_FEATURE1_SUPPORT
 //#define VENDOR_FEATURE2_SUPPORT
 //#define VENDOR_FEATURE3_SUPPORT
diff -rNu rt3090-2.4.0.4/os/linux/cfg80211.c rt3090-2.4.0.4-new/os/linux/cfg80211.c
--- rt3090-2.4.0.4/os/linux/cfg80211.c	2011-05-05 21:41:31.000000000 +0800
+++ rt3090-2.4.0.4-new/os/linux/cfg80211.c	2013-06-13 11:24:29.000000000 +0800
@@ -1352,7 +1352,7 @@
 #endif // RFKILL_HW_SUPPORT //
 
 static struct cfg80211_ops CFG80211_Ops = {
-	.set_channel			= CFG80211_OpsSetChannel,
+	.set_monitor_channel	= CFG80211_OpsSetChannel,
 	.change_virtual_intf	= CFG80211_OpsChgVirtualInf,
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,30))
@@ -2059,7 +2059,7 @@
 		pChan->band = IEEE80211_BAND_2GHZ;
 	/* End of if */
 
-	pChan->center_freq = ieee80211_channel_to_frequency(ChanId);
+	pChan->center_freq = ieee80211_channel_to_frequency(ChanId, pChan->band);
 
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,32))
 	if (pAd->CommonCfg.PhyMode >= PHY_11ABGN_MIXED)
@@ -2620,7 +2620,7 @@
 	for(IdLoop=0; IdLoop<NumOfChan; IdLoop++)
 	{
 		pChannels[IdLoop].center_freq = \
-					ieee80211_channel_to_frequency(Cfg80211_Chan[IdLoop]);
+					ieee80211_channel_to_frequency(Cfg80211_Chan[IdLoop], pChannels[IdLoop].band);
 		pChannels[IdLoop].hw_value = IdLoop;
 
 		if (IdLoop < CFG80211_NUM_OF_CHAN_2GHZ)
