--- Makefile	2011-09-23 11:46:06.383671859 -0400
+++ Makefile.new	2011-09-23 11:46:52.210338124 -0400
@@ -5,9 +5,9 @@
 CXX     = g++
 CC      = gcc
 LD      = g++
-LDFLAGS = -shared -rdynamic `python-config --ldflags` -lboost_python
-LIBS    = -lstdc++ `python-config --libs`
-CFLAGS  = -Wall -I/usr/include `python-config --includes` -I./ -I./kbool/include -g -fPIC
+LDFLAGS = -shared -rdynamic `python2-config --ldflags` -lboost_python
+LIBS    = -lstdc++ `python2-config --libs`
+CFLAGS  = -Wall -I/usr/include `python2-config --includes` -I./ -I./kbool/include -g -fPIC
 
 LIBNAME	= area
 LIBOBJS	= Arc.o Area.o AreaBoolean.o AreaDxf.o AreaOrderer.o AreaPocket.o booleng.o Circle.o Construction.o Curve.o dxf.o Finite.o graph.o graphlst.o instonly.o kurve.o line.o link.o lpoint.o Matrix.o node.o offset.o PythonStuff.o record.o scanbeam.o
@@ -28,11 +28,11 @@
 install: $(LIBOUT)
 	strip $^
 	chmod 644 $^
-	mkdir -p $(DESTDIR)`python -c "from distutils.sysconfig import get_python_lib; print get_python_lib(plat_specific=1, standard_lib=0, prefix='$(PREFIX)')"`
-	install $^ $(DESTDIR)`python -c "from distutils.sysconfig import get_python_lib; print get_python_lib(plat_specific=1, standard_lib=0, prefix='$(PREFIX)')"`
+	mkdir -p $(DESTDIR)`python2 -c "from distutils.sysconfig import get_python_lib; print get_python_lib(plat_specific=1, standard_lib=0, prefix='$(PREFIX)')"`
+	install $^ $(DESTDIR)`python2 -c "from distutils.sysconfig import get_python_lib; print get_python_lib(plat_specific=1, standard_lib=0, prefix='$(PREFIX)')"`
 
 test:
-	python test.py
+	python2 test.py
 
 Arc.o: Arc.cpp
 	$(CC) -c $? ${CFLAGS} -o $@
