diff -ur qt3.orig/qt-x11-free-3.3.8-b/include/qobject.h qt3.new/qt-x11-free-3.3.8-b/include/qobject.h
--- qt3.orig/qt-x11-free-3.3.8-b/include/qobject.h	2008-01-15 13:09:13.000000000 -0600
+++ qt3.new/qt-x11-free-3.3.8-b/include/qobject.h	2011-01-01 18:33:19.715656496 -0600
@@ -101,8 +101,11 @@
 
     QObject           *child( const char *objName, const char *inheritsClass = 0, bool recursiveSearch = TRUE ); //### const in 4.0
     const QObjectList *children() const { return childObjects; }
+    QObjectList childrenListObject();
+    const QObjectList childrenListObject() const;
 
     static const QObjectList *objectTrees();
+    static const QObjectList objectTreesListObject();
 
     QObjectList	      *queryList( const char *inheritsClass = 0,
 				  const char *objName = 0,
diff -ur qt3.orig/qt-x11-free-3.3.8-b/src/kernel/qobject.cpp qt3.new/qt-x11-free-3.3.8-b/src/kernel/qobject.cpp
--- qt3.orig/qt-x11-free-3.3.8-b/src/kernel/qobject.cpp	2008-01-15 13:09:13.000000000 -0600
+++ qt3.new/qt-x11-free-3.3.8-b/src/kernel/qobject.cpp	2011-01-01 18:28:16.191270264 -0600
@@ -360,6 +360,30 @@
     }
 }
 
+/*! \internal
+    TQt compatibility function
+*/
+QObjectList QObject::childrenListObject() {
+	if (children()) return *(children());
+	else return QObjectList();
+}
+
+/*! \internal
+    TQt compatibility function
+*/
+const QObjectList QObject::childrenListObject() const {
+	if (children()) return *(children());
+	else return QObjectList();
+}
+
+/*! \internal
+    TQt compatibility function
+*/
+const QObjectList QObject::objectTreesListObject() {
+	if (objectTrees()) return *(objectTrees());
+	else return QObjectList();
+}
+
 
 /*****************************************************************************
   QObject member functions
diff -ur qt3.orig/qt-x11-free-3.3.8-b/src/kernel/qobject.h qt3.new/qt-x11-free-3.3.8-b/src/kernel/qobject.h
--- qt3.orig/qt-x11-free-3.3.8-b/src/kernel/qobject.h	2008-01-15 13:09:13.000000000 -0600
+++ qt3.new/qt-x11-free-3.3.8-b/src/kernel/qobject.h	2011-01-01 18:33:19.715656496 -0600
@@ -101,8 +101,11 @@
 
     QObject           *child( const char *objName, const char *inheritsClass = 0, bool recursiveSearch = TRUE ); //### const in 4.0
     const QObjectList *children() const { return childObjects; }
+    QObjectList childrenListObject();
+    const QObjectList childrenListObject() const;
 
     static const QObjectList *objectTrees();
+    static const QObjectList objectTreesListObject();
 
     QObjectList	      *queryList( const char *inheritsClass = 0,
 				  const char *objName = 0,
