diff -Naur original/src/easytag.c modified/src/easytag.c
--- original/src/easytag.c	2014-06-22 20:46:38.000000000 +0100
+++ modified/src/easytag.c	2014-09-20 12:38:48.320954921 +0100
@@ -277,7 +277,7 @@
 
     /* Tag */
     TagArea = Create_Tag_Area();
-    gtk_box_pack_start(GTK_BOX(VBox),TagArea,FALSE,FALSE,0);
+    gtk_box_pack_start(GTK_BOX(VBox),TagArea,TRUE,TRUE,0);
 
     /* Vertical pane for Browser Area + FileArea + TagArea */
     MainWindowVPaned = gtk_paned_new(GTK_ORIENTATION_VERTICAL);
@@ -726,9 +726,15 @@
     TagFrame = gtk_frame_new(_("Tag"));
     gtk_container_set_border_width(GTK_CONTAINER(TagFrame),2);
 
+    /* ScrolledWindow */
+    ScrolledWindow = gtk_scrolled_window_new(NULL, NULL);
+    gtk_container_add(GTK_CONTAINER(TagFrame),ScrolledWindow);
+    gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(ScrolledWindow),
+                                   GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
+
     /* Box for the notebook (only for setting a border) */
     VBox = gtk_box_new(GTK_ORIENTATION_VERTICAL,0);
-    gtk_container_add(GTK_CONTAINER(TagFrame),VBox);
+    gtk_scrolled_window_add_with_viewport (ScrolledWindow, VBox);
     gtk_container_set_border_width(GTK_CONTAINER(VBox),2);
 
     /*
@@ -736,7 +742,7 @@
      */
     TagNoteBook = gtk_notebook_new();
     gtk_notebook_popup_enable(GTK_NOTEBOOK(TagNoteBook));
-    //gtk_container_add(GTK_CONTAINER(TagFrame),TagNoteBook);
+    // gtk_container_add(GTK_CONTAINER(ScrolledWindow),TagNoteBook);
     gtk_box_pack_start(GTK_BOX(VBox),TagNoteBook,TRUE,TRUE,0);
     gtk_notebook_set_tab_pos(GTK_NOTEBOOK(TagNoteBook),GTK_POS_TOP);
     gtk_notebook_set_show_border(GTK_NOTEBOOK(TagNoteBook),FALSE);
diff -Naur original/src/easytag.h modified/src/easytag.h
--- original/src/easytag.h	2014-06-22 20:46:38.000000000 +0100
+++ modified/src/easytag.h	2014-09-20 12:36:16.641986812 +0100
@@ -62,6 +62,7 @@
 
 /* TAG Area */
 GtkWidget    *TagFrame;
+GtkWidget    *ScrolledWindow;
 GtkWidget    *TagNoteBook;
 GtkWidget    *TitleEntry;
 GtkWidget    *ArtistEntry;
