--- Makefile.orig	2013-03-29 12:30:15.212435226 +0000
+++ Makefile	2013-03-29 12:36:37.260360082 +0000
@@ -10,7 +10,7 @@
 DOCDIR     ?= $(INSTALLDIR)/share/uzbl/docs
 RUN_PREFIX ?= $(PREFIX)
 
-ENABLE_WEBKIT2 ?= auto
+ENABLE_WEBKIT2 ?= no
 ENABLE_GTK3    ?= auto
 
 PYTHON=python3
@@ -189,7 +189,7 @@
 	install -m755 uzbl-core $(INSTALLDIR)/bin/uzbl-core
 
 install-event-manager: install-dirs
-	$(PYTHON) setup.py install --prefix=$(PREFIX) --install-scripts=$(INSTALLDIR)/bin $(PYINSTALL_EXTRA)
+	$(PYTHON) setup.py install --prefix=$(INSTALLDIR) --install-scripts=$(INSTALLDIR)/bin $(PYINSTALL_EXTRA)
 
 install-uzbl-browser: install-dirs install-uzbl-core install-event-manager
 	sed 's#^PREFIX=.*#PREFIX=$(RUN_PREFIX)#' < bin/uzbl-browser > $(INSTALLDIR)/bin/uzbl-browser
