--- a/chrome-wrapper	2011-03-20 08:46:37.000000000 +0000
+++ b/chrome-wrapper	2011-03-20 08:50:14.000000000 +0000
@@ -7,7 +7,7 @@
 # Running Chromium via this script makes it possible to set Chromium as the
 # default browser directly out of a compile, without needing to package it.
 
-DESKTOP="chromium-devel"
+DESKTOP="chromium"
 TITLE="Chromium"
 
 usage() {
@@ -55,20 +55,46 @@
     mkdir -p "$apps"
     cat > "$apps/$DESKTOP.desktop" << EOF
 [Desktop Entry]
-Version=1.0
 Encoding=UTF-8
+Version=1.0
+Categories=Application;Network;WebBrowser;
+
+Comment=Browse the World Wide Web with Chromium
+Comment[de]=Navigieren im World Wide Web mit Chrom
+Comment[es]=Navegar por la World Wide Web con cromo
+Comment[fr]=Naviguer sur le World Wide Web avec Chrome
+Comment[hi]=ब्राउज़ करें वर्ल्ड वाइड वेब क्रोमियम के साथ
+Comment[it]=Sfoglia il World Wide Web con cromo
+Comment[ja]=閲覧は World Wide Web クロムと
+Comment[ko]=브라 우즈는 World Wide Web 크롬과
+Comment[ru]=Обзор World Wide Web с хромом
+Comment[zh]=浏览万维网的铬
+GenericName=Web Browser
+GenericName[de]=Web-Browser
+GenericName[es]=Navegador Web
+GenericName[fr]=Navigateur Web
+GenericName[hi]=वेब ब्राउज़र
+GenericName[it]=Web Browser
+GenericName[ja]=Webブラウザ
+GenericName[ko]=웹 브라우저
+GenericName[ru]=Веб-браузер
+GenericName[zh]=Web浏览器
 Name=$TITLE
-Exec=$CHROME_WRAPPER %U
+
+TryExec=chromium
+Exec=chromium %U
+Icon=chromium.png
+MimeType=text/html;text/xml;application/xhtml+xml;application/xml;application/vnd.mozilla.xul+xml;application/rss+xml;application/rdf+xml;x-directory/webdav;x-directory/webdav-prefer-directory;image/gif;image/jpeg;image/png;
+StartupNotification=false
+StartupWMClass=Chromium
 Terminal=false
-Icon=$HERE/product_logo_48.png
 Type=Application
-Categories=Application;Network;WebBrowser;
-MimeType=text/html;text/xml;application/xhtml_xml;
+X-MultipleArgs=false
 EOF
 }
 
 # Let the wrapped binary know that it has been run through the wrapper.
-export CHROME_WRAPPER="`readlink -f "$0"`"
+export CHROME_WRAPPER="/usr/bin/$DESKTOP"
 export CHROME_DESKTOP="$DESKTOP.desktop"
 
 HERE="`dirname "$CHROME_WRAPPER"`"
