diff -Naur blam-1.8.9.orig/src/ChannelCollection.cs blam-1.8.9/src/ChannelCollection.cs
--- blam-1.8.9.orig/src/ChannelCollection.cs	2011-07-31 14:21:26.000000000 +0200
+++ blam-1.8.9/src/ChannelCollection.cs	2012-03-07 23:47:09.097208190 +0100
@@ -153,6 +153,10 @@
 
 		// Move the file to the real one
 		try {
+		    if (!File.Exists(this.FileName)) {
+			File.Create(this.FileName);
+		    }
+
 		    File.Replace (tmpFile, this.FileName, this.FileName + ".bk");
 		} catch (Exception e) {
 		    Console.Error.WriteLine ("File replace error: " + e.Message);
