--- asusoled-clock.orig	2008-12-28 22:28:15.000000000 +0200
+++ asusoled-clock	2008-12-28 22:28:49.000000000 +0200
@@ -24,7 +24,7 @@
 test -x $MOGRIFY || exit 1
 
 # We initialize the ASUS Oled screen
-$ASUSOLED -e
+$ASUSOLED -e > /dev/null
 
 TEMPORARY_PIXMAP=`/bin/tempfile`
 
@@ -33,7 +33,7 @@
   /bin/cp $CLOCK_PIXMAP $TEMPORARY_PIXMAP
   $MOGRIFY -fill white -pointsize 15 -annotate 0x0+2+15 `date $DATE_FORMAT` $TEMPORARY_PIXMAP
   $MOGRIFY -fill white -pointsize 15 -annotate 0x0+2+31 `date $TIME_FORMAT` $TEMPORARY_PIXMAP
-  $ASUSOLED -s $TEMPORARY_PIXMAP
+  $ASUSOLED -s $TEMPORARY_PIXMAP > /dev/null
   /bin/sleep 1
 done
 
