diff --git a/pam_yubico.c b/pam_yubico.c
index 36ec4b4..282b7e9 100644
--- a/pam_yubico.c
+++ b/pam_yubico.c
@@ -581,7 +581,7 @@ do_challenge_response(pam_handle_t *pamh, struct cfg *cfg, const char *username)
 
   /* There is a bug that makes the YubiKey 2.2 send the same response for all challenges
      unless HMAC_LT64 is set, check for that here */
-  if (memcmp(buf, state.response response_len) == 0) {
+  if (memcmp(buf, state.response, response_len) == 0) {
     D(("Same response for second challenge, YubiKey should be reconfigured with the option HMAC_LT64"));
   }
 
