diff --git a/src/ConvertPackage/ConvertFile.py b/src/ConvertPackage/ConvertFile.py
index 81b10d0..b025934 100644
--- a/src/ConvertPackage/ConvertFile.py
+++ b/src/ConvertPackage/ConvertFile.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import os, zipfile
 import imghdr
diff --git a/src/ConvertPackage/book.py b/src/ConvertPackage/book.py
index d19c5c6..5233858 100755
--- a/src/ConvertPackage/book.py
+++ b/src/ConvertPackage/book.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (C) 2010  Alex Yatskov
 #
diff --git a/src/ConvertPackage/convert.py b/src/ConvertPackage/convert.py
index 3cee49d..b828d95 100755
--- a/src/ConvertPackage/convert.py
+++ b/src/ConvertPackage/convert.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (C) 2010  Alex Yatskov
 #
diff --git a/src/ConvertPackage/image.py b/src/ConvertPackage/image.py
index 05562a4..3142f2c 100755
--- a/src/ConvertPackage/image.py
+++ b/src/ConvertPackage/image.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (C) 2010  Alex Yatskov
 #
diff --git a/src/manga.py b/src/manga.py
index 33502cd..cf403e8 100755
--- a/src/manga.py
+++ b/src/manga.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (C) 2010  
 #
@@ -188,9 +188,9 @@ def main():
 	if (options.convert_Directory):	
 		options.inputDir = os.path.abspath(options.inputDir)
 	
-	# Changes the working directory to the script location
-	if (os.path.dirname(sys.argv[0]) != ""):
-		os.chdir(os.path.dirname(sys.argv[0]))
+	# Dont change the working directory to the script location
+	# if (os.path.dirname(sys.argv[0]) != ""):
+	#	os.chdir(os.path.dirname(sys.argv[0]))
 
 	options.outputMgr = progressBarManager()
 	options.outputMgr.start()
diff --git a/src/outputManager/base.py b/src/outputManager/base.py
index 416b685..b09214a 100755
--- a/src/outputManager/base.py
+++ b/src/outputManager/base.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # The outputManager synchronizes the output display for all the various threads
 
diff --git a/src/outputManager/progressBarManager.py b/src/outputManager/progressBarManager.py
index daff498..cf455a5 100644
--- a/src/outputManager/progressBarManager.py
+++ b/src/outputManager/progressBarManager.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # The progressBarManager synchronizes the progress bars for the program
 
diff --git a/src/outputManager/progressbar.py b/src/outputManager/progressbar.py
index b668647..3d91334 100644
--- a/src/outputManager/progressbar.py
+++ b/src/outputManager/progressbar.py
@@ -1,7 +1,7 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 #
-# progressbar  - Text progressbar library for python.
+# progressbar  - Text progressbar library for python2.
 # Copyright (c) 2005 Nilton Volpato
 #
 # This library is free software; you can redistribute it and/or
@@ -19,7 +19,7 @@
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
 
-"""Text progressbar library for python.
+"""Text progressbar library for python2.
 
 This library provides a text mode progressbar. This is typically used
 to display the progress of a long running operation, providing a
diff --git a/src/parsers/animea.py b/src/parsers/animea.py
index 0b1d2ac..9b9cc96 100644
--- a/src/parsers/animea.py
+++ b/src/parsers/animea.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ####################################################################
 # For more detailed comments look at MangaFoxParser
diff --git a/src/parsers/base.py b/src/parsers/base.py
index 7878e28..0b3d373 100755
--- a/src/parsers/base.py
+++ b/src/parsers/base.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #####################
 
diff --git a/src/parsers/factory.py b/src/parsers/factory.py
index 6887b52..2afa99c 100644
--- a/src/parsers/factory.py
+++ b/src/parsers/factory.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #####################
 
diff --git a/src/parsers/mangafox.py b/src/parsers/mangafox.py
index b05c237..2ca9485 100644
--- a/src/parsers/mangafox.py
+++ b/src/parsers/mangafox.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ####################
 
@@ -100,7 +100,7 @@ class MangaFox(SiteParserBase):
 					if (self.lastDownloaded == self.chapters[i][1]):
 						lowerRange = i + 1
 
-			# this might need to be len(self.chapters) + 1, I'm unsure as to whether python adds +1 to i after the loop or not
+			# this might need to be len(self.chapters) + 1, I'm unsure as to whether python2 adds +1 to i after the loop or not
 			upperRange = len(self.chapters)
 			
 			# which ones do we want?
diff --git a/src/parsers/mangareader.py b/src/parsers/mangareader.py
index 6f14408..74a0f1d 100644
--- a/src/parsers/mangareader.py
+++ b/src/parsers/mangareader.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ####################################################################
 # For more detailed comments look at MangaFoxParser
@@ -51,7 +51,7 @@ class MangaReader(SiteParserBase):
 				if (self.lastDownloaded == self.chapters[i][1]):
 					lowerRange = i + 1
 		
-		# this might need to be len(self.chapters) + 1, I'm unsure as to whether python adds +1 to i after the loop or not
+		# this might need to be len(self.chapters) + 1, I'm unsure as to whether python2 adds +1 to i after the loop or not
 		upperRange = len(self.chapters)
 		self.isPrependMangaName = False				
 		if (not self.auto):
diff --git a/src/parsers/otakuworks.py b/src/parsers/otakuworks.py
index 088f0c9..4ca06b5 100644
--- a/src/parsers/otakuworks.py
+++ b/src/parsers/otakuworks.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ####################################################################
 # For more detailed comments look at MangaFoxParser
@@ -54,7 +54,7 @@ class OtakuWorks(SiteParserBase):
 				if (self.lastDownloaded == self.chapters[i][1]):
 					lowerRange = i + 1
 		
-		# this might need to be len(self.chapters) + 1, I'm unsure as to whether python adds +1 to i after the loop or not
+		# this might need to be len(self.chapters) + 1, I'm unsure as to whether python2 adds +1 to i after the loop or not
 		upperRange = len(self.chapters)	
 	
 		if (not self.auto):
diff --git a/src/parsers/thread.py b/src/parsers/thread.py
index ed574a8..74c45c6 100644
--- a/src/parsers/thread.py
+++ b/src/parsers/thread.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #####################
 
diff --git a/src/setup.py b/src/setup.py
new file mode 100644
index 0000000..e9460c8
--- /dev/null
+++ b/src/setup.py
@@ -0,0 +1,15 @@
+#!/usr/bin/env python2
+
+# Simple setup script for manga_downloader
+# https://github.com/jiaweihli/manga_downloader
+
+from distutils.core import setup
+
+setup(name='manga_downloader',
+      version='0.9',
+      description='Download manga from varios sites',
+      author='Jiawei Li',
+      url='https://github.com/jiaweihli/manga_downloader',
+      py_modules=['util','xmlparser'],
+      packages=['outputManager', 'parsers', 'ConvertPackage'],
+      )
diff --git a/src/util.py b/src/util.py
index eb561d9..452e2d6 100755
--- a/src/util.py
+++ b/src/util.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ####################
 
diff --git a/src/xmlparser.py b/src/xmlparser.py
index 746f615..7d230c0 100755
--- a/src/xmlparser.py
+++ b/src/xmlparser.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ######################
 
