--- include/rtmp_def.h.orig	2012-05-08 07:27:26.000000000 +0200
+++ include/rtmp_def.h	2012-05-29 14:46:26.000000000 +0200
@@ -1490,8 +1490,8 @@
 #define INF_MAIN_DEV_NAME		"wlan"
 #define INF_MBSSID_DEV_NAME		"wlan"
 #else
-#define INF_MAIN_DEV_NAME		"ra"
-#define INF_MBSSID_DEV_NAME		"ra"
+#define INF_MAIN_DEV_NAME		"wlan"
+#define INF_MBSSID_DEV_NAME		"wlan"
 #endif /* ANDROID_SUPPORT */
 #define INF_WDS_DEV_NAME		"wds"
 #define INF_APCLI_DEV_NAME		"apcli"
