--- ctpp2-2.6.9/CMakeLists.txt	2011-01-28 19:03:54.000000000 +0300
+++ ctpp2-2.6.9/CMakeLists.txt.new	2011-02-20 20:30:10.604063998 +0300
@@ -546,46 +546,58 @@
 ADD_TEST(Code_stack_test                    VMCodeStackTest)
 ADD_TEST(Create_executable_file             CTPP2VMTest selftest.ct2)
 ADD_TEST(VM_self_test                       ctpp2vm selftest.ct2)
+SET_TESTS_PROPERTIES(VM_self_test PROPERTIES DEPENDS Create_executable_file)
 
 
 ADD_TEST(Output_variables_C                 ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/output_variables.tmpl Output_variables.ct2)
 ADD_TEST(Output_variables_R                 ctpp2vm Output_variables.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/testdata.json)
+SET_TESTS_PROPERTIES(Output_variables_R PROPERTIES DEPENDS Output_variables_C)
 
 ADD_TEST(Comparisons_C                      ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/comparisons.tmpl Comparisons.ct2)
 ADD_TEST(Comparisons_R                      ctpp2vm Comparisons.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/testdata.json)
+SET_TESTS_PROPERTIES(Comparisons_R PROPERTIES DEPENDS Comparisons_C)
 
 ADD_TEST(Arith_ops_C                        ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/arith_ops.tmpl Arith_ops.ct2)
 ADD_TEST(Arith_ops_R                        ctpp2vm Arith_ops.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/testdata.json)
+SET_TESTS_PROPERTIES(Arith_ops_R PROPERTIES DEPENDS Arith_ops_C)
 
 ADD_TEST(Formulas_C                         ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/formulas.tmpl Formulas.ct2)
 ADD_TEST(Formulas_R                         ctpp2vm Formulas.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/testdata.json)
+SET_TESTS_PROPERTIES(Formulas_R PROPERTIES DEPENDS Formulas_C)
 
 ADD_TEST(Functions_C                        ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/functions.tmpl Functions.ct2)
 ADD_TEST(Functions_R                        ctpp2vm Functions.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/testdata.json)
+SET_TESTS_PROPERTIES(Functions_R PROPERTIES DEPENDS Functions_C)
 
 IF (${MD5_SUPPORT} MATCHES "ON")
     ADD_TEST(Function_md5_C                 ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/function_md5.tmpl Function_md5.ct2)
     ADD_TEST(Function_md5_R                 ctpp2vm Function_md5.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/testdata.json)
+    SET_TESTS_PROPERTIES(Function_md5_R PROPERTIES DEPENDS Function_md5_C)
 ENDIF (${MD5_SUPPORT} MATCHES "ON")
 
 IF (${ICONV_SUPPORT} MATCHES "ON")
     ADD_TEST(Function_iconv_C               ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/function_iconv.tmpl Function_iconv.ct2)
     ADD_TEST(Function_iconv_R               ctpp2vm Function_iconv.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/testdata.json)
+    SET_TESTS_PROPERTIES(Function_iconv_R PROPERTIES DEPENDS Function_iconv_C)
 ENDIF (${ICONV_SUPPORT} MATCHES "ON")
 
 IF (${GETTEXT_SUPPORT} MATCHES "ON")
     ADD_TEST(Function_gettext_C             ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/function_gettext.tmpl Function_gettext.ct2)
     ADD_TEST(Function_gettext_R             ctpp2vm Function_gettext.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/testdata.json)
+    SET_TESTS_PROPERTIES(Function_gettext_R PROPERTIES DEPENDS Function_gettext_C)
 ENDIF (${GETTEXT_SUPPORT} MATCHES "ON")
 
 ADD_TEST(Loops_C                            ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/loops.tmpl Loops.ct2)
 ADD_TEST(Loops_R                            ctpp2vm Loops.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/tests/testdata/testdata.json)
+SET_TESTS_PROPERTIES(Loops_R PROPERTIES DEPENDS Loops_C)
 
 ADD_TEST(Lebowski_bench_loop_C              ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/examples/lebowski-bench-loop.tmpl lebowski-bench-loop.ct2)
 ADD_TEST(Lebowski_bench_loop_R              ctpp2vm lebowski-bench-loop.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/examples/lebowski-bench.json)
+SET_TESTS_PROPERTIES(Lebowski_bench_loop_R PROPERTIES DEPENDS Lebowski_bench_loop_C)
 
 ADD_TEST(Lebowski_bench_foreach_C           ctpp2c ${CMAKE_CURRENT_SOURCE_DIR}/examples/lebowski-bench-foreach.tmpl lebowski-bench-foreach.ct2)
 ADD_TEST(Lebowski_bench_foreach_R           ctpp2vm lebowski-bench-foreach.ct2 ${CMAKE_CURRENT_SOURCE_DIR}/examples/lebowski-bench.json)
+SET_TESTS_PROPERTIES(Lebowski_bench_foreach_R PROPERTIES DEPENDS Lebowski_bench_foreach_C)
 
 ADD_TEST(DTOA                               CTPP2DTOATest)
 
