=== modified file 'src/indicator-printers-service.c'
--- src/indicator-printers-service.c	2012-03-20 18:50:39 +0000
+++ src/indicator-printers-service.c	2012-09-07 17:07:36 +0000
@@ -56,7 +56,7 @@
 
     attr = ippFindAttribute (resp, "notify-subscription-id", IPP_TAG_INTEGER);
     if (attr)
-        id = attr->values[0].integer;
+        id = ippGetInteger (attr, 0);
     else
         g_warning ("ipp-create-printer-subscription response doesn't contain "
                    "subscription id.\n");

