--- a/Dawawin/daw_customs.py    2013-05-03 08:27:28.000000000 +0200
+++ b/Dawawin/daw_customs.py    2013-05-03 22:52:01.586845884 +0200
@@ -17,7 +17,10 @@
 my_return = 1
 
 #a--------------------------------------------------
-APP_DIR      = dirname(dirname(realpath(__file__)))
+if exists('/usr/share/dawawin'):
+    APP_DIR = '/usr/share/dawawin'
+elif exists('/usr/local/share/dawawin'):
+    APP_DIR = '/usr/local/share/dawawin'
 if exists('/usr/share/dawawin/dawawin-data/Dawawin.db'):
     DATA_DIR = '/usr/share/dawawin/dawawin-data'
 elif exists('/usr/local/share/dawawin/dawawin-data/Dawawin.db'):
