diff -Naur eid-mw-read-only-orig/common/src/socket/socket.h eid-mw-read-only/common/src/socket/socket.h
--- eid-mw-read-only-orig/common/src/socket/socket.h	2012-04-26 13:43:30.023417198 +0200
+++ eid-mw-read-only/common/src/socket/socket.h	2012-04-26 13:43:21.650083392 +0200
@@ -29,6 +29,7 @@
 	typedef UINT_PTR  SOCKET;
 	#define INVALID_SOCKET  (SOCKET)(~0)
 	#include "sys/socket.h"
+	#include <unistd.h>
 #endif
 
 #include <string>
diff -Naur eid-mw-read-only-orig/common/src/socket/socketserver.h eid-mw-read-only/common/src/socket/socketserver.h
--- eid-mw-read-only-orig/common/src/socket/socketserver.h	2012-04-26 13:43:30.026750531 +0200
+++ eid-mw-read-only/common/src/socket/socketserver.h	2012-04-26 13:43:21.650083392 +0200
@@ -22,6 +22,7 @@
 #include "../export.h"
 
 #include "socket.h"
+#include <unistd.h>
 
 namespace eIDMW
 {
diff -Naur eid-mw-read-only-orig/common/src/util.h eid-mw-read-only/common/src/util.h
--- eid-mw-read-only-orig/common/src/util.h	2012-04-26 13:38:59.580068647 +0200
+++ eid-mw-read-only/common/src/util.h	2012-04-26 13:44:15.540086425 +0200
@@ -30,6 +30,7 @@
 #include <string.h>
 #include <iostream>
 #include "mw_util.h"
+#include <unistd.h>
 
 namespace eIDMW
 {
