diff -Nur gnome-clock-screensaver-1.0.1-orig/Makefile gnome-clock-screensaver-1.0.1/Makefile
--- gnome-clock-screensaver-1.0.1-orig/Makefile	2011-01-15 12:32:49.000000000 +0200
+++ gnome-clock-screensaver-1.0.1/Makefile	2011-08-28 17:12:55.156334577 +0300
@@ -25,7 +25,7 @@
 clean:
 	rm -f $(OBJECTS) $(TARGET) ${DESKTOP} core* *.o *~
 install:
-	${INSTALL} -s -m 0755 ${TARGET}  ${GS_LIBEXEC_DIR}
-	${INSTALL}    -m 0644 ${DESKTOP} ${GS_DESKTOP_DIR}
+	${INSTALL} -m755 ${TARGET}  ${DESTDIR}${GS_LIBEXEC_DIR}
+	${INSTALL} -m644 ${DESKTOP} ${DESTDIR}${GS_DESKTOP_DIR}
 uninstall:
 	-rm ${GS_LIBEXEC_DIR}/${TARGET} ${GS_DESKTOP_DIR}/${DESKTOP}
