diff -wbBur ziproxy-3.3.0/src/image.c ziproxy-3.3.0.q/src/image.c
--- ziproxy-3.3.0/src/image.c	2013-01-04 20:48:50.000000000 +0400
+++ ziproxy-3.3.0.q/src/image.c	2013-05-19 15:36:09.447475372 +0400
@@ -93,6 +93,10 @@
 #define MIN_INSIZE_TO_JPEG 600
 #define MIN_INSIZE_TO_JP2K 800
 
+
+#define false 0
+#define true !!0
+
 //Forwards. There are more utility functions, but they're used only once.
 static raw_bitmap *new_raw_bitmap();
 
