--- cclive-git/bootstrap.sh	2014-12-08 14:40:23.666225785 +0100
+++ cclive-git.new/bootstrap.sh	2014-12-08 14:43:48.239290579 +0100
@@ -60,4 +60,4 @@ do
 done
 
 echo "Generate configuration files..."
-autoreconf -if && echo "You can now run 'configure'"
+autoreconf -ifv && echo "You can now run 'configure'"
diff -rupN cclive-git/configure.ac cclive-git.new/configure.ac
--- cclive-git/configure.ac	2014-12-08 14:40:23.666225785 +0100
+++ cclive-git.new/configure.ac	2014-12-08 14:47:30.482100153 +0100
@@ -17,8 +17,8 @@ AC_USE_SYSTEM_EXTENSIONS
 AC_DEFINE_UNQUOTED([CANONICAL_TARGET], "$target",
   [Define to canonical target])
 
-AM_INIT_AUTOMAKE([1.11.1 -Wall -Werror dist-xz no-dist-gzip tar-ustar])
-AM_SILENT_RULES([yes])
+AM_INIT_AUTOMAKE([1.11.1 -Wall -Werror dist-xz no-dist-gzip tar-ustar subdir-objects])
+AM_SILENT_RULES([no])
 
 # GNU Automake 1.12 requires this macro. Earlier versions do not
 # recognize this macro. Work around this.
