diff -ru exmap-0.10.orig/jutil/TestRunner.cpp exmap-0.10/jutil/TestRunner.cpp
--- exmap-0.10.orig/jutil/TestRunner.cpp	2009-12-29 22:34:03.284721825 -0500
+++ exmap-0.10/jutil/TestRunner.cpp	2009-12-29 22:35:11.588042553 -0500
@@ -1,5 +1,7 @@
 #include "TestRunner.hpp"
 #include "Pcre.hpp"
+#include <cstdlib>
+#include <cstdio>
 
 using namespace std;
 
diff -ru exmap-0.10.orig/jutil/jutil.cpp exmap-0.10/jutil/jutil.cpp
--- exmap-0.10.orig/jutil/jutil.cpp	2009-12-29 22:34:03.284721825 -0500
+++ exmap-0.10/jutil/jutil.cpp	2009-12-29 22:34:46.488033452 -0500
@@ -6,6 +6,8 @@
 #include <limits.h>
 #include <dirent.h>
 
+#include <cstring>
+
 using namespace std;
 using namespace jutil;
 
diff -ru exmap-0.10.orig/jutil/jutil.hpp exmap-0.10/jutil/jutil.hpp
--- exmap-0.10.orig/jutil/jutil.hpp	2009-12-29 22:34:03.284721825 -0500
+++ exmap-0.10/jutil/jutil.hpp	2009-12-29 22:34:46.491382761 -0500
@@ -7,6 +7,7 @@
 #include <map>
 
 #include <stdio.h>
+#include <cstdlib>
 
 // For getuid/setuid
 #include <unistd.h>
diff -ru exmap-0.10.orig/kernel/exmap.c exmap-0.10/kernel/exmap.c
--- exmap-0.10.orig/kernel/exmap.c	2009-12-29 22:34:03.284721825 -0500
+++ exmap-0.10/kernel/exmap.c	2009-12-29 22:34:46.491382761 -0500
@@ -507,7 +507,7 @@
 							NULL);
 
 	if (exmap_proc_file == NULL) {
-		remove_proc_entry (PROCFS_NAME, &proc_root);
+		remove_proc_entry (PROCFS_NAME, NULL);
 		printk (KERN_ALERT "/proc/%s: could not initialize\n",
 			PROCFS_NAME);
 		return -ENOMEM;
@@ -532,5 +532,5 @@
 void cleanup_module ()
 {
 	printk (KERN_INFO "/proc/%s: remove\n", PROCFS_NAME);
-	remove_proc_entry (PROCFS_NAME, &proc_root);
+	remove_proc_entry (PROCFS_NAME, NULL);
 }
diff -ru exmap-0.10.orig/src/Elf.cpp exmap-0.10/src/Elf.cpp
--- exmap-0.10.orig/src/Elf.cpp	2009-12-29 22:34:03.332205805 -0500
+++ exmap-0.10/src/Elf.cpp	2009-12-29 22:34:46.494716705 -0500
@@ -6,6 +6,8 @@
 #include <sstream>
 #include <unistd.h> // getpagesize()
 
+#include <cstring>
+
 using namespace std;
 using namespace jutil;
 using namespace Elf;
diff -ru exmap-0.10.orig/src/exmtool.cpp exmap-0.10/src/exmtool.cpp
--- exmap-0.10.orig/src/exmtool.cpp	2009-12-29 22:34:03.334725121 -0500
+++ exmap-0.10/src/exmtool.cpp	2009-12-29 22:34:46.494716705 -0500
@@ -6,6 +6,7 @@
 #include <sstream>
 #include <iostream>
 #include <vector>
+#include <cstring>
 
 using namespace std;
 using namespace Exmap;
@@ -18,9 +19,9 @@
 
 struct command
 {
-    char *command;
+    const char *command;
     Handler handler;
-    char *usage;
+    const char *usage;
 } cmd_handles[] = {
     { "procs",
       do_procs,
