*** setup.py    2010-12-23 00:41:56.000000000 +0100
--- setup2.py   2011-01-04 20:10:30.196424474 +0100
***************
*** 1,7 ****
  #!/usr/bin/env python
  
  import os
- import py_compile
  import sys
  
  from glob import glob
--- 1,6 ----
***************
*** 9,14 ****
--- 8,14 ----
  from distutils.core import setup
  from distutils.command.install_data import install_data
  from distutils.dist import DistributionMetadata
+ from distutils.util import byte_compile
  
  # Patch distutils if it can't cope with the "classifiers" or
  # "download_url" keywords
***************
*** 25,35 ****
      (os.path.join("share", "doc", "arista"), [
          "README.md", "LICENSE", "AUTHORS"
      ]),
!     (os.path.join("lib", "nautilus", "extensions-2.0", "python"), ["arista-nautilus.py"]),
  ]
  
! for (prefix, path) in [("arista", "presets"), 
!                         ("arista", "ui"), 
                          ("", "locale")]:
      for root, dirs, files in os.walk(path):
          to_add = []
--- 25,35 ----
      (os.path.join("share", "doc", "arista"), [
          "README.md", "LICENSE", "AUTHORS"
      ]),
!     (os.path.join("share", "nautilus-python", "extensions"), ["arista-nautilus.py"]),
  ]
  
! for (prefix, path) in [("arista", "presets"),
!                         ("arista", "ui"),
                          ("", "locale")]:
      for root, dirs, files in os.walk(path):
          to_add = []
***************
*** 50,58 ****
          for path, fnames in data_files:
              for fname in fnames:
                  if fname.endswith(".py"):
!                     full = os.path.join(sys.prefix, path, fname)
                      print "byte-compiling %s" % full
!                     py_compile.compile(full)
  
  setup(
      name = "arista",
--- 50,58 ----
          for path, fnames in data_files:
              for fname in fnames:
                  if fname.endswith(".py"):
!                     full = os.path.join(self.root + sys.prefix, path, fname)
                      print "byte-compiling %s" % full
!                     byte_compile([full], prefix=self.root, base_dir=sys.prefix)
  
  setup(
      name = "arista",
