From 3c5ad49a12639d58fb4fe6102de625c0e51f2792 Mon Sep 17 00:00:00 2001
From: Christian Hesse <eworm@leda.eworm.de>
Date: Sun, 4 Aug 2013 12:44:59 +0200
Subject: [PATCH 1/1] fix includes

---
 src/common/tuklib_cpucores.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/tuklib_cpucores.c b/src/common/tuklib_cpucores.c
index 93413a3..7574bc9 100644
--- a/src/common/tuklib_cpucores.c
+++ b/src/common/tuklib_cpucores.c
@@ -18,7 +18,7 @@
 #	endif
 #	include <windows.h>
 
-#if defined(TUKLIB_CPUCORES_SYSCTL)
+#elif defined(TUKLIB_CPUCORES_SYSCTL)
 #	ifdef HAVE_SYS_PARAM_H
 #		include <sys/param.h>
 #	endif
-- 
1.8.3.4

