--- functions.orig	2008-03-16 11:12:29.000000000 +0500
+++ functions	2008-03-16 11:14:10.000000000 +0500
@@ -15,8 +15,8 @@
 # Some handy definitions
 VZCTL=/usr/sbin/vzctl
 VZLOCKDIR=/vz/lock
-VECFGDIR=/etc/sysconfig/vz-scripts/
-VZCFG=/etc/sysconfig/vz
+VECFGDIR=/etc/vz/conf
+VZCFG=/etc/vz/vz.conf
 VZLIB_SCRIPTDIR=/usr/lib/vzctl/scripts
 YUM=/usr/share/vzyum/bin/yum
 ARCHES="x86 i386 x86_64 ia64"
