--- src/trooper.h~	2009-11-05 17:03:01.000000000 +0500
+++ src/trooper.h	2009-11-05 17:05:36.690781726 +0500
@@ -58,7 +58,7 @@
 };
 
 bool four_troopers(trooper* landed_troopers[]);
-int drops_on_trooper(SDL_Surface *screen, trooper* trooper, trooper* landed_troopers[]);
+int drops_on_trooper(SDL_Surface *screen, trooper* _trooper, trooper* landed_troopers[]);
 void concatenate(int index, trooper* landed_troopers[]);
 int move_troopers(linkedList2 &troopLList, SDL_Surface *screen, trooper* landed_troopers[]);
 
--- src/trooper.cpp~	2009-11-05 17:05:40.000000000 +0500
+++ src/trooper.cpp	2009-11-05 17:06:20.824109375 +0500
@@ -207,17 +207,17 @@
 }
 
 // checks if the trooper "trooper" falls on top of a landed trooper and handles the changes it requires if so
-int drops_on_trooper(SDL_Surface *screen, trooper* trooper, trooper* landed_troopers[])
+int drops_on_trooper(SDL_Surface *screen, trooper* _trooper, trooper* landed_troopers[])
 {
 	int points = 0;
-	int trooper_x = (trooper->getDst2Rect()).x;
-	int trooper2_x = trooper_x + (trooper->getDst2Rect()).w;
+	int trooper_x = (_trooper->getDst2Rect()).x;
+	int trooper2_x = trooper_x + (_trooper->getDst2Rect()).w;
 	for (int i=0;i<20 && landed_troopers[i] != NULL;i++)
 	{
 		if (((trooper_x >= ((landed_troopers[i])->getDst2Rect()).x) && (trooper_x <= ((landed_troopers[i])->getDst2Rect()).x + ((landed_troopers[i])->getDst2Rect()).w))
 		|| ((trooper2_x >= ((landed_troopers[i])->getDst2Rect()).x) && (trooper2_x <= ((landed_troopers[i])->getDst2Rect()).x + ((landed_troopers[i])->getDst2Rect()).w)))
 		{
-			points += trooper->getPoints();
+			points += _trooper->getPoints();
 			SDL_Rect black1 = (landed_troopers[i])->getBlack(), black2 = (landed_troopers[i])->getBlack2();;
 			//SDL_FillRect(screen, &black1, SDL_MapRGB(screen->format, 0, 0, 0));
 			//SDL_FillRect(screen, &black2, SDL_MapRGB(screen->format, 0, 0, 0));
