--- install.sh	2012-10-10 14:49:25.474802021 -0400
+++ install.sh.new	2012-10-10 16:05:02.505393378 -0400
@@ -1857,11 +1857,12 @@
 
 # --- Set ownership or remove OpenNebula directories ---
 
 if [ "$UNINSTALL" = "no" ] ; then
     for d in $CHOWN_DIRS; do
-        chown -R $ONEADMIN_USER:$ONEADMIN_GROUP $DESTDIR$d
+        :
+#        chown -R $ONEADMIN_USER:$ONEADMIN_GROUP $DESTDIR$d
     done
 else
     for d in `echo $DELETE_DIRS | awk '{for (i=NF;i>=1;i--) printf $i" "}'`; do
         rmdir $d
     done
