diff --git a/src/tsocks.c b/src/tsocks.c
index 02e21fc..9339a27 100644
--- a/src/tsocks.c
+++ b/src/tsocks.c
@@ -235,10 +235,10 @@ void tsocks_init(void)
     #ifdef SUPPORT_RES_API
     lib = dlopen(LIBRESOLV, RTLD_LAZY);
     realres_init = dlsym(lib, "res_init");
-    realresquery = dlsym(lib, "res_query");
-    realressend = dlsym(lib, "res_send");
-    realresquerydomain = dlsym(lib, "res_querydomain");
-    realressearch = dlsym(lib, "res_search");
+    real__res_query = dlsym(lib, "res_query");
+    real__res_send = dlsym(lib, "res_send");
+    real__res_querydomain = dlsym(lib, "res_querydomain");
+    real__res_search = dlsym(lib, "res_search");
     dlclose(lib);
     #endif
 #endif
