diff --git a/libbirdfont/BirdFont.vala b/libbirdfont/BirdFont.vala
index 86a8668..0b570e0 100644
--- a/libbirdfont/BirdFont.vala
+++ b/libbirdfont/BirdFont.vala
@@ -381,19 +381,13 @@ public class BirdFont {
 			DirUtils.create ((!) home.get_path (), 0xFFFFFF);
 		}
 #else
-		home_path = Environment.get_home_dir ();
-	
-		if (is_null (home_path)) {
-			warning ("No home directory set.");
-			home_path = ".";
-		}
-		
+		home_path = Environment.get_user_config_dir ();
 		home = File.new_for_path (home_path);
 #endif
-		settings = home.get_child (".birdfont");
+		settings = home.get_child ("birdfont");
 				
 		if (!settings.query_exists ()) {
-			DirUtils.create ((!) settings.get_path (), 0xFFFFFF);
+			DirUtils.create ((!) settings.get_path (), 0755);
 		}
 			
 		return settings;
@@ -404,7 +398,7 @@ public class BirdFont {
 		File backup = settings.get_child ("backup");
 		
 		if (!backup.query_exists ()) {
-			DirUtils.create ((!) backup.get_path (), 0xFFFFFF);
+			DirUtils.create ((!) backup.get_path (), 0755);;
 		}
 			
 		return backup;
