--- src/piklab-0.15.12/cmake/Piklab.cmake	2012-03-10 21:46:58.000000000 -0500
+++ Piklab.cmake	2012-06-03 14:47:17.863066079 -0400
@@ -53,8 +53,8 @@
   find_package(LIBUSB REQUIRED)
   include_directories(${LIBUSB_INCLUDE_DIRS})
 
-  find_package(Readline REQUIRED)
-  include_directories(${LIBREADLINE_INCLUDE_DIRS})
+  #find_package(Readline REQUIRED)
+  #include_directories(${LIBREADLINE_INCLUDE_DIRS})
 
   find_package(ParallelPortSupport)
 endif(WIN32)
