diff -Nur Synkron-1.6.2-src.orig/synkron.desktop Synkron-1.6.2-src/synkron.desktop
--- Synkron-1.6.2-src.orig/synkron.desktop	1970-01-01 01:00:00.000000000 +0100
+++ Synkron-1.6.2-src/synkron.desktop	2011-10-14 10:55:14.007818864 +0200
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Type=Application
+Icon=Synkron128
+Name=Synkron
+Comment=A folder synchronization tool
+Comment[ru]=Средство для синхронизации папок
+Comment[uk]=Засіб синхронізації тек
+Categories=Utility;
+Exec=synkron
+StartupNotify=false
+Terminal=false
diff -Nur Synkron-1.6.2-src.orig/Synkron.pro Synkron-1.6.2-src/Synkron.pro
--- Synkron-1.6.2-src.orig/Synkron.pro	2011-01-25 13:07:36.000000000 +0100
+++ Synkron-1.6.2-src/Synkron.pro	2011-10-14 10:55:17.831152231 +0200
@@ -43,6 +43,21 @@
 unix:!macx {
      DEFINES += USE_UNIX_TOUCH_COMMAND
      TARGET = synkron
+
+     isEmpty(prefix):prefix = /usr/local
+     isEmpty(PREFIX):PREFIX = $$prefix
+     BINDIR = $$PREFIX/bin
+     DATADIR = $$PREFIX/share
+
+     INSTALLS = target \
+                desktop \
+                pixmaps \
+
+     target.path = $$BINDIR
+     desktop.path = $$DATADIR/applications/
+     desktop.files += $${TARGET}.desktop
+     pixmaps.path = $$DATADIR/pixmaps/
+     pixmaps.files += images/Synkron128.png
 }
 macx {
      # Comment the following line to use the NSFileManager class
