diff -Nurd yaph-0.91/configure.in yaph-0.91.1/configure.in
--- yaph-0.91/configure.in	2003-02-09 07:37:15.000000000 +0200
+++ yaph-0.91.1/configure.in	2012-06-11 21:48:50.513932506 +0300
@@ -96,6 +96,4 @@
 AC_OUTPUT( \
 ./Makefile \
 yaph/Makefile \
-yaph/docs/Makefile \
-yaph/docs/en/Makefile \
 )
diff -Nurd yaph-0.91/yaph/init.c yaph-0.91.1/yaph/init.c
--- yaph-0.91/yaph/init.c	2003-02-09 08:33:04.000000000 +0200
+++ yaph-0.91.1/yaph/init.c	2012-06-11 21:48:17.740776721 +0300
@@ -200,7 +200,7 @@
                 }
             }
         }
-	no_file:
+	no_file:;
 }
 
 // ------   open files
diff -Nurd yaph-0.91/yaph/Makefile.am yaph-0.91.1/yaph/Makefile.am
--- yaph-0.91/yaph/Makefile.am	2003-02-09 08:02:37.000000000 +0200
+++ yaph-0.91.1/yaph/Makefile.am	2012-06-11 22:08:07.554330947 +0300
@@ -1,6 +1,7 @@
 ####### kdevelop will overwrite this part!!! (begin)##########
 bin_PROGRAMS = yaph
 
+sysconf_DATA = yaph.conf
 ## INCLUDES were found outside kdevelop specific part
 
 yaph_SOURCES = threads.c tcp_utils.c log.c init.c content_utils.c main.c 
@@ -9,13 +10,6 @@
 
 EXTRA_DIST = main.c content_utils.c definitions.h functions.h globals.h includes.h init.c log.c tcp_utils.c threads.c yaph.conf yaph.h 
 
-install-data-local:
-	$(mkinstalldirs) /etc/
-	$(INSTALL_DATA) $(srcdir)/yaph.conf /etc/yaph.conf
-
-uninstall-local:
-	-rm -f /etc/yaph.conf
-
 ####### kdevelop will overwrite this part!!! (end)############
 # set the include path found by configure
 INCLUDES= $(all_includes)
diff -Nurd yaph-0.91/yaph/tcp_utils.c yaph-0.91.1/yaph/tcp_utils.c
--- yaph-0.91/yaph/tcp_utils.c	2003-02-09 07:54:38.000000000 +0200
+++ yaph-0.91.1/yaph/tcp_utils.c	2012-06-11 21:47:50.240925700 +0300
@@ -27,7 +27,7 @@
     goto error;
   return sock;
 error:
-  return -1;
+close(sock);  return -1;
 }
 
 int read_line(int fd, char *buff, size_t size)
