diff -crB yaph-0.91/yaph/init.c yaph-0.91-patched/yaph/init.c
*** yaph-0.91/yaph/init.c	2003-02-09 07:33:04.000000000 +0100
--- yaph-0.91/yaph/init.c	2010-04-12 17:04:30.000000000 +0200
***************
*** 126,134 ****
  {/// config file routine
  FILE *cnf=NULL;
  char buff[1024];
! 	if(!(cnf=fopen("./yaph.conf","rb")))
! 		if(!(cnf=fopen("/etc/yaph.conf","rb")))
! 			goto no_file;
  	while(fgets(buff,sizeof(buff),cnf))
  	{
              if(buff[strspn(buff," ")]!='#')
--- 126,133 ----
  {/// config file routine
  FILE *cnf=NULL;
  char buff[1024];
!       if((cnf=fopen("./yaph.conf","rb"))||(cnf=fopen("/etc/yaph.conf","rb")))
!       {
  	while(fgets(buff,sizeof(buff),cnf))
  	{
              if(buff[strspn(buff," ")]!='#')
***************
*** 200,206 ****
                  }
              }
          }
! 	no_file:
  }
  
  // ------   open files
--- 199,205 ----
                  }
              }
          }
!       }
  }
  
  // ------   open files
