diff -uNr monodevelop/main/Main.sln monodevelop.new/main/Main.sln
--- monodevelop/main/Main.sln	2014-07-18 17:19:08.707649804 +0200
+++ monodevelop.new/main/Main.sln	2014-07-18 17:20:54.370985278 +0200
@@ -1411,11 +1411,13 @@
 		{ADDC9D20-83C5-4245-9B46-2C5F6642CFBC}.ReleaseWin32|Any CPU.ActiveCfg = Release|Any CPU
 		{ADDC9D20-83C5-4245-9B46-2C5F6642CFBC}.ReleaseWin32|Any CPU.Build.0 = Release|Any CPU
 		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.DebugGnome|Any CPU.ActiveCfg = Debug|Any CPU
 		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.DebugGnome|Any CPU.Build.0 = Debug|Any CPU
 		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.DebugMac|Any CPU.ActiveCfg = Debug|Any CPU
 		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.DebugWin32|Any CPU.ActiveCfg = Debug|Any CPU
 		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.Release|Any CPU.Build.0 = Release|Any CPU
 		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.ReleaseGnome|Any CPU.ActiveCfg = Release|Any CPU
 		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.ReleaseGnome|Any CPU.Build.0 = Release|Any CPU
 		{AE84B4A7-E6B1-4565-8904-16284DCE41B6}.ReleaseMac|Any CPU.ActiveCfg = Release|Any CPU
