--- server/drivers/imonlcd.c.orig	2009-06-20 15:48:34.000000000 +0200
+++ server/drivers/imonlcd.c	2010-06-04 13:36:41.508573513 +0200
@@ -1184,6 +1184,10 @@
 
 	if (err <= 0)
 		printf("%s: error writing to file descriptor: %d", "imon", err);
+	/* If this is the VFD device, slow down communication to prevent overloading 
+	 * the controller 
+	 */
+	if (p->protocol == PROTOCOL_0038) usleep(5000);
 }
 
 
