--- lualdap.orig	2013-04-17 03:02:47.090796277 +0100
+++ lualdap.c	2013-04-17 03:06:54.337098964 +0100
@@ -380,7 +380,7 @@
 			}
 		}
 		array[n] = NULL;
-	} else 
+	} else
 		return luaL_error (L, LUALDAP_PREFIX"bad argument #%d (table or string expected, got %s)", tab, lua_typename (L, lua_type (L, tab)));
 	return 0;
 }
@@ -872,7 +872,7 @@
 ** Create a metatable.
 */
 static int lualdap_createmeta (lua_State *L) {
-	const luaL_reg methods[] = {
+	const luaL_Reg methods[] = {
 		{"close", lualdap_close},
 		{"add", lualdap_add},
 		{"compare", lualdap_compare},
@@ -987,7 +987,7 @@
 ** Create ldap table and register the open method.
 */
 int luaopen_lualdap (lua_State *L) {
-	struct luaL_reg lualdap[] = {
+	struct luaL_Reg lualdap[] = {
 		{"open_simple", lualdap_open_simple},
 		{NULL, NULL},
 	};
