diff --git a/initialize.py b/initialize.py
old mode 100644
new mode 100755
index c25f1ba..91c1d2f
--- a/initialize.py
+++ b/initialize.py
@@ -1,3 +1,4 @@
+#!/usr/bin/env python2
 """
 Creates the database used by Django
 """
@@ -15,7 +16,8 @@ def setup():
     if not path.exists(settings.DATA_DIR):
         os.makedirs(settings.DATA_DIR)
     print "Creating database %s" % database_file
-    subprocess.call([executable, 'mathics/manage.py', 'syncdb', '--noinput'])
+    dn = os.path.dirname(os.path.realpath(__file__))
+    subprocess.call([dn+'/mathics/manage.py', 'syncdb', '--noinput'])
     os.chmod(database_file, 0766)
     print ""
     print "Mathics initialized successfully."
diff --git a/mathics/manage.py b/mathics/manage.py
old mode 100644
new mode 100755
index 762d925..a4b7d94
--- a/mathics/manage.py
+++ b/mathics/manage.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """from sage.libs.pari.gen import pari
     
diff --git a/setup.py b/setup.py
old mode 100644
new mode 100755
index 200b755..688c0bd
--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from distribute_setup import use_setuptools
 use_setuptools()
 
