diff -crB Palaver-build/setup Palaver-ArchPatches/setup
*** Palaver-build/setup	2013-04-17 23:13:09.467839700 -0500
--- Palaver-ArchPatches/setup	2013-04-17 23:18:18.947821869 -0500
***************
*** 2,14 ****
  
  # Setup speech recognition for the person.
  
- if [ "$(whoami)" != "root" ]; then
- 	echo "Sorry, Please run as root."
- 	exit 1
- fi
- 
- sudo apt-get install sox python2-argparse wget espeak xautomation xvkbd -y
- 
  DIR="$(pwd)"
  
  echo Assuming all files are in "'$DIR'"
--- 2,7 ----
