--- yadex-1.7.0.orig/src/wadlist.cc	Fri Mar 28 18:37:32 2003
+++ yadex-1.7.0/src/wadlist.cc	Tue Jul  3 18:35:21 2007
@@ -175,7 +175,8 @@
   priv->iter = priv->list.erase (i);
   if (priv->iter == priv->list.begin ())
   {
-    priv->iter = 0;			// Catch bugs
+    list_t::iterator empty;
+    priv->iter = empty;			// Catch bugs
     priv->rewound = true;
   }
 }
--- yadex-1.7.0.orig/src/yadex.h	Sat Jan  1 14:23:40 2005
+++ yadex-1.7.0/src/yadex.h	Sat Jan  1 14:23:50 2005
@@ -98,13 +98,13 @@
 #define F_I16_D "hd"
 #define F_I16_H "hX"
 
-typedef unsigned long  u32;
-#define F_U32_D "lu"
-#define F_U32_H "lX"
+typedef unsigned int  u32;
+#define F_U32_D "u"
+#define F_U32_H "X"
 
-typedef signed   long  i32;
-#define F_I32_D "ld"
-#define F_I32_H "lX"
+typedef signed   int  i32;
+#define F_I32_D "d"
+#define F_I32_H "X"
 
 
 /*
--- yadex-1.7.0.orig/src/gfx.cc	2009-05-24 02:01:55.921088470 +0200
+++ yadex-1.7.0/src/gfx.cc	2009-05-24 02:07:44.478136129 +0200
@@ -310,7 +310,7 @@
     win_vis_id    = vis_info->visualid;
 #if defined _cplusplus || defined __cplusplus
     win_vis_class = vis_info->c_class;
-#elif
+#else
     win_vis_class = vis_info->class;
 #endif
     win_ncolours  = vis_info->colormap_size;
