Index: code/missionui/redalert.cpp
===================================================================
--- code/missionui/redalert.cpp	(revision 9720)
+++ code/missionui/redalert.cpp	(working copy)
@@ -714,7 +714,7 @@
 		}
 
 		int found_match = 0;
-		int ship_state = 0;
+		int ship_state = RED_ALERT_DESTROYED_SHIP_CLASS;
 
 		for ( i = 0; i < (int)Red_alert_wingman_status.size(); i++ ) {
 			ras = &Red_alert_wingman_status[i];
@@ -742,7 +742,6 @@
 
 			} else if ( !stricmp(ras->name.c_str(), shipp->ship_name) && ( (ras->ship_class == RED_ALERT_DESTROYED_SHIP_CLASS) || (ras->ship_class == RED_ALERT_PLAYER_DEL_SHIP_CLASS) ) ) {
 				ship_state = ras->ship_class;
-				continue;
 			}
 		}
 
