=== modified file 'configure.ac'
--- old/configure.ac	2011-09-09 08:45:39 +0000
+++ new/configure.ac	2012-05-25 20:04:30 +0000
@@ -70,7 +70,7 @@
 AC_SUBST(GCC_FLAGS)
 
 PKG_CHECK_MODULES(DEPS,
-                  indicate-0.6 >= 0.5.90
+                  indicate-0.7 >= 0.6.90
                   $EVOLUTION_PLUGIN
                   dbus-1
                   dbus-glib-1
@@ -78,6 +78,9 @@
                   libcanberra
                   $GTKHTML_EDITOR
                   unity
+                  libebackend-1.2
+                  libsoup-2.4
+                  gconf-2.0
                   )
 AC_SUBST(DEPS_CFLAGS)
 AC_SUBST(DEPS_LIBS)

=== modified file 'po/Makefile.in.in'
--- old/po/Makefile.in.in	2009-11-19 18:51:08 +0000
+++ new/po/Makefile.in.in	2012-05-25 20:27:00 +0000
@@ -49,8 +49,8 @@
 XGETTEXT = @XGETTEXT@
 INTLTOOL_UPDATE = @INTLTOOL_UPDATE@
 INTLTOOL_EXTRACT = @INTLTOOL_EXTRACT@
-MSGMERGE = INTLTOOL_EXTRACT=$(INTLTOOL_EXTRACT) srcdir=$(srcdir) $(INTLTOOL_UPDATE) --gettext-package $(GETTEXT_PACKAGE) --dist
-GENPOT   = INTLTOOL_EXTRACT=$(INTLTOOL_EXTRACT) srcdir=$(srcdir) $(INTLTOOL_UPDATE) --gettext-package $(GETTEXT_PACKAGE) --pot
+MSGMERGE = INTLTOOL_EXTRACT="$(INTLTOOL_EXTRACT)" XGETTEXT="$(XGETTEXT)" srcdir=$(srcdir) $(INTLTOOL_UPDATE) --gettext-package $(GETTEXT_PACKAGE) --dist
+GENPOT   = INTLTOOL_EXTRACT="$(INTLTOOL_EXTRACT)" XGETTEXT="$(XGETTEXT)" srcdir=$(srcdir) $(INTLTOOL_UPDATE) --gettext-package $(GETTEXT_PACKAGE) --pot
 
 ALL_LINGUAS = @ALL_LINGUAS@
 
@@ -73,15 +73,20 @@
 .SUFFIXES:
 .SUFFIXES: .po .pox .gmo .mo .msg .cat
 
+AM_DEFAULT_VERBOSITY = @AM_DEFAULT_VERBOSITY@
+INTLTOOL_V_MSGFMT = $(INTLTOOL__v_MSGFMT_$(V))
+INTLTOOL__v_MSGFMT_= $(INTLTOOL__v_MSGFMT_$(AM_DEFAULT_VERBOSITY))
+INTLTOOL__v_MSGFMT_0 = @echo "  MSGFMT" $@;
+
 .po.pox:
 	$(MAKE) $(GETTEXT_PACKAGE).pot
 	$(MSGMERGE) $< $(GETTEXT_PACKAGE).pot -o $*.pox
 
 .po.mo:
-	$(MSGFMT) -o $@ $<
+	$(INTLTOOL_V_MSGFMT)$(MSGFMT) -o $@ $<
 
 .po.gmo:
-	file=`echo $* | sed 's,.*/,,'`.gmo \
+	$(INTLTOOL_V_MSGFMT)file=`echo $* | sed 's,.*/,,'`.gmo \
 	  && rm -f $$file && $(GMSGFMT) -o $$file $<
 
 .po.cat:

=== modified file 'src/evolution-indicator.c'
--- old/src/evolution-indicator.c	2011-06-28 22:10:50 +0000
+++ new/src/evolution-indicator.c	2012-05-25 20:47:06 +0000
@@ -42,6 +42,8 @@
 #include <shell/es-event.h>
 #include <shell/e-shell-view.h>
 
+#include <camel/camel.h>
+
 #include <libindicate/server.h>
 #include <libindicate/indicator.h>
 #include <libindicate/indicator-messages.h>
@@ -142,7 +144,6 @@
   static GdkWindow *root = NULL;
   gboolean          res = FALSE;
   Window            xwindow;
-  GList *list;
   EShell *evo_shell;
   EShellWindow *shell_window;
   
@@ -180,16 +181,9 @@
   }
 
   evo_shell = e_shell_get_default ();
-  list = e_shell_get_watched_windows (evo_shell);
-
-  /* Find the first EShellWindow in the list. */
-  while (list != NULL && !E_IS_SHELL_WINDOW (list->data))
-    list = g_list_next (list);
-
-  g_return_if_fail (list != NULL);
 
   /* Get the shell window. */
-  shell_window = E_SHELL_WINDOW (list->data);
+  shell_window = E_SHELL_WINDOW (e_shell_get_active_window (evo_shell));
 
   if (GTK_IS_WINDOW (shell_window))
   {
@@ -228,10 +222,13 @@
 org_gnome_mail_new_notify (EPlugin *ep, EMEventTargetFolder *t)
 {
   gchar *url;
+  CamelURL *service_url;
 
   g_return_if_fail (t != NULL);
 
-  url = camel_service_get_url (CAMEL_SERVICE (t->store));
+  service_url = camel_service_new_camel_url (CAMEL_SERVICE (t->store));
+  url = camel_url_to_string (service_url, 0);
+  camel_url_free (service_url);
 
   if (!t->new)
 		return;
@@ -1008,22 +1005,13 @@
   if (shell)
   {
     GSList    *i;
-    GList     *list;
-    GtkWindow *mail_window = NULL;
     EShellWindow *shell_window;
     EShellView *shell_view;
     GtkAction *action;
 
-    list = e_shell_get_watched_windows (shell);
-
-    /* Find the first EShellWindow in the list. */
-    while (list != NULL && !E_IS_SHELL_WINDOW (list->data))
-        list = g_list_next (list);
-
-    g_return_if_fail (list != NULL);
 
     /* Present the shell window. */
-    shell_window = E_SHELL_WINDOW (list->data);
+    shell_window = E_SHELL_WINDOW (e_shell_get_active_window (shell));
     if (!evolution_is_focused ())
     {
         gtk_window_present_with_time (GTK_WINDOW (shell_window), timestamp);

