diff -Naur evolution-indicator-0.2.10.orig/configure.ac evolution-indicator-0.2.10/configure.ac
--- evolution-indicator-0.2.10.orig/configure.ac	2010-06-22 15:08:29.000000000 +0200
+++ evolution-indicator-0.2.10/configure.ac	2011-03-15 15:22:17.751123572 +0100
@@ -55,7 +55,7 @@
                   evolution-plugin
                   dbus-1
                   dbus-glib-1
-                  libnotify
+                  libnotify >= 0.7.0
                   libcanberra
                   gtkhtml-editor
                   )
diff -Naur evolution-indicator-0.2.10.orig/src/evolution-indicator.c evolution-indicator-0.2.10/src/evolution-indicator.c
--- evolution-indicator-0.2.10.orig/src/evolution-indicator.c	2010-06-22 15:06:04.000000000 +0200
+++ evolution-indicator-0.2.10/src/evolution-indicator.c	2011-03-15 15:21:45.453452868 +0100
@@ -289,7 +289,7 @@
 
     if (notification == NULL)
     {
-      notification = notify_notification_new (" ", " ", "mail-unread", NULL);
+      notification = notify_notification_new (" ", " ", "mail-unread");
     }
 
     trans = g_dngettext (PACKAGE, 
