diff -Naur gnushogi-1.3.2/gnushogi/Makefile.in gnushogi-1.3.2.new/gnushogi/Makefile.in
--- gnushogi-1.3.2/gnushogi/Makefile.in	2004-07-08 02:16:35.000000000 +0200
+++ gnushogi-1.3.2.new/gnushogi/Makefile.in	2009-07-06 19:30:29.000000000 +0200
@@ -58,7 +58,7 @@
 LIBS    = -lm
 
 # Display routines.
-LCURSES =   -lcurses -ltermcap
+LCURSES =   -lcurses
 
 
 #
diff -Naur gnushogi-1.3.2/gnushogi/Makefile.profile.in gnushogi-1.3.2.new/gnushogi/Makefile.profile.in
--- gnushogi-1.3.2/gnushogi/Makefile.profile.in	2004-07-08 02:16:35.000000000 +0200
+++ gnushogi-1.3.2.new/gnushogi/Makefile.profile.in	2009-07-06 19:30:40.000000000 +0200
@@ -58,7 +58,7 @@
 LIBS    = -lm
 
 # Display routines.
-LCURSES =   -lcurses -ltermcap
+LCURSES =   -lcurses
 
 
 #
