Index: gnome-session-2.28.0/data/gnome-wm
===================================================================
--- gnome-session-2.28.0.orig/data/gnome-wm	2009-07-29 02:36:07.000000000 +0200
+++ gnome-session-2.28.0/data/gnome-wm	2009-09-22 22:58:55.000000000 +0200
@@ -40,11 +40,6 @@
   esac
 done
 
-# WINDOW_MANAGER overrides all
-if [ -z "$WINDOW_MANAGER" ] ; then
-    WINDOW_MANAGER=`gconftool-2 --get /desktop/gnome/session/required_components/windowmanager 2> /dev/null`
-fi
-
 # Migrate compiz to compiz-manager if possible and needed
 if [ "x$WINDOW_MANAGER" = "xcompiz" -o "x$DEFWM" = "xcompiz" ]; then
   which compiz-manager > /dev/null 2>&1
@@ -64,45 +59,30 @@
   WINDOW_MANAGER=""
 fi
 
-if [ -z "$WINDOW_MANAGER" ] ; then
-  # Create a list of window manager we can handle, trying to only use the
-  # compositing ones when it makes sense
-
-  xdpyinfo 2> /dev/null | grep -q "^ *Composite$" 2> /dev/null
-  IS_X_COMPOSITED=$?
+# If compiz-gtk is installed, we always use it; it will fall back to 
+# metacity if it cannot find decent hardware.
 
-  KNOWN_WM="sawfish sawmill enlightenment icewm wmaker fvwm2 qvwm fvwm twm kwm"
-  if [ $IS_X_COMPOSITED -eq 0 ] ; then
-    KNOWN_WM="mutter compiz-manager compiz beryl $KNOWN_WM"
-  fi
-  # metacity is still the default wm in GNOME
-  KNOWN_WM="metacity $KNOWN_WM"
+if [ -z "$WINDOW_MANAGER" ] && [ -x /usr/bin/gtk-window-decorator ] ; then
+  WINDOW_MANAGER=compiz
+fi
 
-  OLDIFS=$IFS
-  if [ -z "$DEFWM" -o "x$DEFWM" = "xgnome-wm" ]; then
+# If compiz is not installed (which is the default), use metacity instead.
 
-    for wm in $KNOWN_WM ; do
-      IFS=":"
-      for dir in $PATH ; do
-        if [ -x "$dir/$wm" ] ; then
-          WINDOW_MANAGER="$dir/$wm"
-          break 2
-        fi
-      done
-      IFS=$OLDIFS
-    done
+if [ -z "$WINDOW_MANAGER" ]; then
+  KNOWN_WM="metacity sawfish mutter"
 
-  else
-    WINDOW_MANAGER=$DEFWM
-  fi
-  IFS=$OLDIFS
+  for wm in $KNOWN_WM; do
+    if [ -x /usr/bin/"$wm" ]; then
+      WINDOW_MANAGER=/usr/bin/"$wm"
+      break
+    fi
+  done
 fi
 
-# If no window manager can be found, we default to xterm
+# Look for the default window manager on the system
 
 if [ -z "$WINDOW_MANAGER" ] ; then
-  echo "WARNING: No window manager can be found."
-  WINDOW_MANAGER=xterm
+  WINDOW_MANAGER=$(readlink /etc/alternatives/x-window-manager 2>/dev/null)
 fi
 
 # Now create options OPT1, OPT2 and OPT3 based on the windowmanager used
@@ -127,7 +107,7 @@
       OPT1=-s
       OPT2=$SMID
       ;;
-    fvwm)
+    fvwm|fvwm2)
       OPT1=-i
       OPT2=$SMID
       ;;
@@ -145,16 +125,15 @@
 
 case `basename $WINDOW_MANAGER` in
   compiz)
-    export LIBGL_ALWAYS_INDIRECT=1
     gtk-window-decorator &
-    OPT3=glib
-    OPT4=gconf
+    #OPT3=glib
+    #OPT4=gconf
     ;;
   beryl)
     emerald &
     ;;
 esac
 
-exec $WINDOW_MANAGER $OPT1 $OPT2 $OPT3 $OPT4
+exec "$WINDOW_MANAGER" $OPT1 $OPT2 $OPT3 $OPT4
 
 echo "ERROR: No window manager could run!"
