Index: gnome-session-2.30.0/gnome-session/gsm-xsmp-client.c
===================================================================
--- gnome-session-2.30.0.orig/gnome-session/gsm-xsmp-client.c	2010-06-04 14:02:15.986733685 +1000
+++ gnome-session-2.30.0/gnome-session/gsm-xsmp-client.c	2010-06-04 14:02:38.024237334 +1000
@@ -803,10 +803,12 @@
 xsmp_get_app_name (GsmClient *client)
 {
         SmProp *prop;
-        char   *name;
+        char   *name = NULL;
 
         prop = find_property (GSM_XSMP_CLIENT (client), SmProgram, NULL);
-        name = prop_to_command (prop);
+        if (prop) {
+                name = prop_to_command (prop);
+        }
 
         return name;
 }
