--- start.sh.orig	2010-10-20 23:47:29.000000000 +0400
+++ start.sh	2010-10-20 23:47:43.000000000 +0400
@@ -15,4 +15,4 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
 
-python prefix/share/decibel-audio-player/src/decibel-audio-player.py "$@" &
+python2 prefix/share/decibel-audio-player/src/decibel-audio-player.py "$@" &
