--- extconf.rb.orig	2009-04-09 09:05:44.000000000 +0700
+++ extconf.rb	2009-10-02 14:09:40.000000000 +0700
@@ -1,6 +1,6 @@
 version_components = RUBY_VERSION.split('.').map { |c| c.to_i }
 
-need_fastthread = ( !defined? RUBY_ENGINE )
+need_fastthread = ( !defined? RUBY_ENGINE ) || (RUBY_ENGINE == 'ruby')
 need_fastthread &= ( RUBY_PLATFORM != 'java' )
 need_fastthread &= version_components[0..1] == [1, 8]
 
