--- OpenStudio.desktop	2015-02-06 01:28:57.413940858 -0300
+++ OpenStudio2.desktop	2015-02-06 01:36:00.413956656 -0300
@@ -3,7 +3,7 @@
 Version=1.0
 Name=OpenStudio
 Comment=Supports whole building energy modeling using EnergyPlus
-Exec=/usr/local/bin/OpenStudio %f
+Exec=LD_LIBRARY_PATH=/usr/local/lib/links:$LD_LIBRARY_PATH  /usr/local/bin/OpenStudio %f
 Terminal=false
 Type=Application
 Icon=OpenStudio
