diff --git a/kded/CMakeLists.txt b/kded/CMakeLists.txt
index 6283135..e4b1974 100644
--- a/kded/CMakeLists.txt
+++ b/kded/CMakeLists.txt
@@ -84,7 +84,7 @@ install(TARGETS kbuildsycoca5 ${INSTALL_TARGETS_DEFAULT_ARGS} )
 if (WIN32)
 install( FILES applications.menu  DESTINATION  ${SHARE_INSTALL_PREFIX}/xdg/menus )
 else (WIN32)
-install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus )
+install( FILES kde-applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus )
 endif (WIN32)
 install( FILES kdedmodule.desktop DESTINATION  ${SERVICETYPES_INSTALL_DIR} )
 
diff --git a/kded/kbuildsycoca.cpp b/kded/kbuildsycoca.cpp
index 8cc0925..ef41db3 100644
--- a/kded/kbuildsycoca.cpp
+++ b/kded/kbuildsycoca.cpp
@@ -312,7 +312,7 @@ bool KBuildSycoca::build()
      if (!m_trackId.isEmpty())
         g_vfolder->setTrackId(m_trackId);
 
-     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("applications.menu");
+     VFolderMenu::SubMenu *kdeMenu = g_vfolder->parseMenu("kde-applications.menu");
 
      KServiceGroup::Ptr entry = g_buildServiceGroupFactory->addNew("/", kdeMenu->directoryFile, KServiceGroup::Ptr(), false);
      entry->setLayoutInfo(kdeMenu->layoutList);
