diff -U10 -r battiold-0.3.8/src/Battery.py batti-0.3.8/src/Battery.py
--- battiold-0.3.8/src/Battery.py	2011-01-15 18:18:11.000000000 +0100
+++ batti-0.3.8/src/Battery.py	2011-11-18 18:41:00.000000000 +0100
@@ -85,70 +85,78 @@
             tooltip = _("Battery level: %s%%") % (info._percentage)
             notification = tooltip
         else:
             time_str = self._str_time(info._time)
             tooltip = _('Battery level: %(level)s%%\nProviding power for approximately %(time)s') % {'level': info._percentage, 'time': time_str}
             notification = _('Power for approximately <b>%s</b> remaining') % time_str
         
         if info._state == STATE_CHARGING:
             self.__systray.set_blinking(False)
             self.__systray.set_tooltip(_('Charging battery\nBattery level: %s%%') % info._percentage)
-            if info._percentage > 80:
-                icon = 'battery-full-charging'
-            elif info._percentage > 40:
-                icon = 'battery-good-charging'
-            elif info._percentage > 20:
-                icon = 'battery-low-charging'
+            if info._percentage > 90:
+                icon = 'batti-charging-100'
+            elif info._percentage > 70:
+                icon = 'batti-charging-080'
+            elif info._percentage > 50:
+                icon = 'batti-charging-060'
+            elif info._percentage > 50:
+                icon = 'batti-charging-040'
+            elif info._percentage > 30:
+                icon = 'batti-charging-020'
             else:
-                icon = 'battery-caution-charging'
+                icon = 'batti-charging-000'
             self.__set_tray_icon(icon)
             
             self.__shown_bat_charged = False
             if not self.__shown_charging:
                 self.__shown_bat_critical = False
                 self.__shown_bat_low = False
                 self.__shown_charging = True
                 self.__shown_on_bat = False
                 self._notify(False, icon, _('Charging battery'), notification)
                 
         elif info._state == STATE_DISCHARGING:
             self.__systray.set_tooltip(tooltip)
-            if info._percentage > 80:
-                icon = 'battery-full'
-            elif info._percentage > 40:
-                icon = 'battery-good'
+            if info._percentage > 90:
+                icon = 'batti-100'
+            elif info._percentage > 70:
+                icon = 'batti-080'
+            elif info._percentage > 50:
+                icon = 'batti-060'
+            elif info._percentage > 30:
+                icon = 'batti-040'
             elif info._percentage > 20:
-                icon = 'battery-low'
-            elif info._percentage > 5:
-                icon = 'battery-caution'
+                icon = 'batti-020'
+            elif info._percentage > 7:
+                icon = 'batti-000'
                 if not self.__shown_bat_low:
                     self._notify(True, icon, _('Low battery level'), notification)
                     self.__shown_bat_low = True
             else:
-                icon = 'battery-empty'
+                icon = 'batti-empty'
                 self.__systray.set_blinking(True)
                 if not self.__shown_bat_critical:
                     self._notify(True, icon, _('Critical battery level'), notification)
                     self.__shown_bat_critical = True
             self.__set_tray_icon(icon)
             self.__shown_bat_charged = False
             if not self.__shown_on_bat:
                 self._notify(False, icon, _("Discharging battery"), notification)
                 self.__shown_charging = False
                 self.__shown_on_bat = True
                     
         elif info._state == STATE_CHARGED:
             self.__systray.set_tooltip(_('Battery charged\n%s') % tooltip)
-            self.__set_tray_icon('battery-full')
+            self.__set_tray_icon('batti-charged')
             self.__systray.set_blinking(False)
             if not self.__shown_bat_charged:
-                self._notify(False, 'battery-full-charging', _('Battery charged'), notification)
+                self._notify(False, 'batti-100', _('Battery charged'), notification)
                 self.__shown_bat_charged = True
                      
     
     def __set_tray_icon(self, icon_name):
         self.__systray.set_from_icon_name(icon_name)
     
     
     def _set_icon_visibility(self, visible):
         self.__systray.set_visible(visible)
     
