diff -aur page-speed-sdk.pristine/third_party/instaweb/src/net/instaweb/htmlparse/html_parse.cc page-speed-sdk.new/third_party/instaweb/src/net/instaweb/htmlparse/html_parse.cc
--- page-speed-sdk.pristine/third_party/instaweb/src/net/instaweb/htmlparse/html_parse.cc	2010-09-23 15:51:26.000000000 +0400
+++ page-speed-sdk.new/third_party/instaweb/src/net/instaweb/htmlparse/html_parse.cc	2011-12-06 12:55:35.972147047 +0400
@@ -269,7 +269,7 @@
     // sure the elements have the expected parent based on context, when
     // we can figure out what the expected parent is.
     HtmlElement* start_element = event->GetStartElement();
-    HtmlElement* actual_parent = NULL;
+    //HtmlElement* actual_parent = NULL;
     if (start_element != NULL) {
       CheckEventParent(event, expect_parent, start_element->parent());
       CHECK(start_element->begin() == current_);
@@ -290,7 +290,7 @@
           CHECK(element_stack.back() == end_element);
           element_stack.pop_back();
         }
-        actual_parent = end_element->parent();
+        //actual_parent = end_element->parent();
         expect_parent = element_stack.empty() ? NULL : element_stack.back();
         CheckEventParent(event, expect_parent, end_element->parent());
       } else {
