diff -ru eglaysher-rlvm-bee99d3/vendor/luabind/include/luabind/detail/call_function.hpp eglaysher-rlvm-bee99d3-patched/vendor/luabind/include/luabind/detail/call_function.hpp
--- eglaysher-rlvm-bee99d3/vendor/luabind/include/luabind/detail/call_function.hpp	2011-05-15 21:44:42.000000000 +0200
+++ eglaysher-rlvm-bee99d3-patched/vendor/luabind/include/luabind/detail/call_function.hpp	2012-12-14 18:14:24.736834205 +0100
@@ -323,7 +323,8 @@
 
 #endif // LUABIND_CALL_FUNCTION_HPP_INCLUDED
 
-#elif BOOST_PP_ITERATION_FLAGS() == 1
+#else
+#if BOOST_PP_ITERATION_FLAGS() == 1
 
 #define LUABIND_TUPLE_PARAMS(z, n, data) const A##n *
 #define LUABIND_OPERATOR_PARAMS(z, n, data) const A##n & a##n
@@ -440,4 +441,5 @@
 
 
 #endif
+#endif
 
diff -ru eglaysher-rlvm-bee99d3/vendor/luabind/include/luabind/detail/call_member.hpp eglaysher-rlvm-bee99d3-patched/vendor/luabind/include/luabind/detail/call_member.hpp
--- eglaysher-rlvm-bee99d3/vendor/luabind/include/luabind/detail/call_member.hpp	2011-05-15 21:44:42.000000000 +0200
+++ eglaysher-rlvm-bee99d3-patched/vendor/luabind/include/luabind/detail/call_member.hpp	2012-12-14 18:12:38.508140617 +0100
@@ -316,7 +316,8 @@
 
 #endif // LUABIND_CALL_MEMBER_HPP_INCLUDED
 
-#elif BOOST_PP_ITERATION_FLAGS() == 1
+#else
+#if BOOST_PP_ITERATION_FLAGS() == 1
 
 #define LUABIND_TUPLE_PARAMS(z, n, data) const A##n *
 #define LUABIND_OPERATOR_PARAMS(z, n, data) const A##n & a##n
@@ -360,4 +361,5 @@
 #undef LUABIND_TUPLE_PARAMS
 
 #endif
+#endif
 
diff -ru eglaysher-rlvm-bee99d3/vendor/luabind/include/luabind/wrapper_base.hpp eglaysher-rlvm-bee99d3-patched/vendor/luabind/include/luabind/wrapper_base.hpp
--- eglaysher-rlvm-bee99d3/vendor/luabind/include/luabind/wrapper_base.hpp	2011-05-15 21:44:42.000000000 +0200
+++ eglaysher-rlvm-bee99d3-patched/vendor/luabind/include/luabind/wrapper_base.hpp	2012-12-14 18:13:48.440613715 +0100
@@ -89,7 +89,8 @@
 
 #endif // LUABIND_WRAPPER_BASE_HPP_INCLUDED
 
-#elif BOOST_PP_ITERATION_FLAGS() == 1
+#else
+#if BOOST_PP_ITERATION_FLAGS() == 1
 
 #define LUABIND_TUPLE_PARAMS(z, n, data) const A##n *
 #define LUABIND_OPERATOR_PARAMS(z, n, data) const A##n & a##n
@@ -188,3 +189,4 @@
 #undef N
 
 #endif
+#endif
