diff -ru eglaysher-rlvm-bee99d3/src/Systems/Base/TextPage.hpp eglaysher-rlvm-bee99d3-patched/src/Systems/Base/TextPage.hpp
--- eglaysher-rlvm-bee99d3/src/Systems/Base/TextPage.hpp	2013-02-08 20:55:54.501748530 +0100
+++ eglaysher-rlvm-bee99d3-patched/src/Systems/Base/TextPage.hpp	2013-02-08 20:56:17.388142606 +0100
@@ -30,6 +30,7 @@
 
 #include <boost/ptr_container/ptr_vector.hpp>
 #include <boost/function.hpp>
+#include <boost/noncopyable.hpp>
 
 #include <algorithm>
 #include <string>
