diff -ru gmvault-1.7-beta.orig/etc/scripts/gmvault gmvault-1.7-beta/etc/scripts/gmvault
--- gmvault-1.7-beta.orig/etc/scripts/gmvault	2012-08-15 21:08:22.000000000 +0200
+++ gmvault-1.7-beta/etc/scripts/gmvault	2012-08-19 20:32:19.010912023 +0200
@@ -25,9 +25,9 @@
     #set pythonhome but it should not be necessary
     export PYTHONHOME="$GMVAULT_HOME/lib/python-lib"
 
-elif [ -f "$GMVAULT_HOME/bin/python" ]; then
+elif [ -f "$GMVAULT_HOME/bin/python2" ]; then
     #installed from src distribution (python bin is in the same dir as gmvault script)
-    PYTHON_BIN="$GMVAULT_HOME/bin/python"
+    PYTHON_BIN="$GMVAULT_HOME/bin/python2"
 
 else
     #look for python2.7 first otherwise try python2.6
