--- gmvault-1.5-beta.orig/etc/scripts/gmvault	2012-06-01 23:42:52.843762547 +0200
+++ gmvault-1.5-beta/etc/scripts/gmvault	2012-06-01 23:44:07.960434292 +0200
@@ -21,9 +21,9 @@
     #set pythonhome but it should not be necessary
     export PYTHONHOME=$GMVAULT_HOME/lib/python-lib
 
-elif [ -f "$GMVAULT_HOME/bin/python" ]; then
+elif [ -f "$GMVAULT_HOME/bin/python2" ]; then
     #installed from src distribution (python bin is in the same dir as gmvault script)
-    PYTHON_BIN=$GMVAULT_HOME/bin/python
+    PYTHON_BIN=$GMVAULT_HOME/bin/python2
 
 else
     PYTHON_BIN=`which python2.7 2>/dev/null`
