diff -rupN tora-2.1.2/src/tooracleconnection.cpp tora-2.1.2.new/src/tooracleconnection.cpp
--- tora-2.1.2/src/tooracleconnection.cpp	2010-03-21 21:09:32.000000000 +0100
+++ tora-2.1.2.new/src/tooracleconnection.cpp	2010-05-08 22:24:01.924767681 +0200
@@ -898,7 +898,7 @@ class oracleConnection : public toConnec
             {
                 // Ignore any errors here
             }
-            return QString::QString();
+            return QString();
         }
 
         virtual toQuery::queryImpl *createQuery(toQuery *query, toConnectionSub *sub)
