--- a/src/preview.c	2012-11-14 01:21:52.487228725 -0500
+++ b/src/preview.c	2012-11-14 01:26:06.503557617 -0500
@@ -327,28 +327,28 @@
             switch (*layout) {
             case 'D':
                 a = focus ?
-                    theme->btn_desk->a_focused_unpressed :
-                    theme->btn_desk->a_unfocused_unpressed;
+                    theme->a_focused_unpressed_desk :
+                    theme->a_unfocused_unpressed_desk;
                 break;
             case 'S':
                 a = focus ?
-                    theme->btn_shade->a_focused_unpressed :
-                    theme->btn_shade->a_unfocused_unpressed;
+                    theme->a_focused_unpressed_shade :
+                    theme->a_unfocused_unpressed_shade;
                 break;
             case 'I':
                 a = focus ?
-                    theme->btn_iconify->a_focused_unpressed :
-                    theme->btn_iconify->a_unfocused_unpressed;
+                    theme->a_focused_unpressed_iconify :
+                    theme->a_unfocused_unpressed_iconify;
                 break;
             case 'M':
                 a = focus ?
-                    theme->btn_max->a_focused_unpressed :
-                    theme->btn_max->a_unfocused_unpressed;
+                    theme->a_focused_unpressed_max :
+                    theme->a_unfocused_unpressed_max;
                 break;
             case 'C':
                 a = focus ?
-                    theme->btn_close->a_focused_unpressed :
-                    theme->btn_close->a_unfocused_unpressed;
+                    theme->a_focused_unpressed_close :
+                    theme->a_unfocused_unpressed_close;
                 break;
             default:
                 continue;
