--- /tmp/kompose-0.5.4/src/komposetaskprefswidget.cpp	2005-07-06 00:32:41.000000000 +0200
+++ src/komposetaskprefswidget.cpp	2006-03-06 11:27:43.000000000 +0100
@@ -28,7 +28,7 @@
 KomposeTaskPrefsWidget::KomposeTaskPrefsWidget(KomposeTaskWidget *parent, const char *name)
     : QDockArea(Qt::Horizontal , QDockArea::Normal, parent, name)
 {
-  KToolBar* pToolBar = new KToolBar( this, i18n("Task Actions") );
+  KToolBar* pToolBar = new KToolBar( this, i18n("Task Actions").latin1() );
   pToolBar->setLabel(i18n("Task"));
   pToolBar->setResizeEnabled ( false );
   pToolBar->setMovingEnabled ( false );
