--- a/src/QtGain.cpp
+++ b/src/QtGain.cpp
@@ -269,16 +269,13 @@ void QtGain::processFiles_TrackGain(){
         return;
     }
 
-    // Check output of replay gain tools
-    processOutput();
-
-
     // ******************************************
-    // New code for skipping FLAC files 1/3
+    // New code for skipping FLAC files 1/4
     if (this->m_lastFileSkipped) {
         addSkippedFiles();
         if (isLogSkippedFiles()) addToErrorlog(tr("<font color='green'>File already gained -> skipped</font> :: %1<br>").arg(currentFileName()));
     } else {
+        // Check output of replay gain tools
         processOutput();
     }
     this->m_lastFileSkipped = false;
@@ -318,12 +315,9 @@ void QtGain::processFiles_TrackGain(){
         // FLAC
         else if (allFiles[finishedFiles()].endsWith(".flac", Qt::CaseInsensitive)){
             ui.FileFormatIcon->setPixmap(QPixmap(":/Icons/flac.png"));
-            Parameter << "--add-replay-gain" << allFiles[finishedFiles()];
-            TrackGainProcess->start(FlacGainBinaryPath(), Parameter);
-
 
             // ******************************************
-            // New code for skipping FLAC files 2/3
+            // New code for skipping FLAC files 2/4
             // Check if flac file is already gained, metaflac does not do this by itself
             QProcess checkProcess;
             checkProcess.start(FlacGainBinaryPath(), QStringList() << "--show-tag" << "REPLAYGAIN_TRACK_GAIN" << allFiles[finishedFiles()]);
@@ -379,7 +373,7 @@ void QtGain::processFiles_TrackGain(){
         addFinishedFiles();
 
         // ******************************************
-        // New code for skipping FLAC files 3/3
+        // New code for skipping FLAC files 3/4
         if (this->m_lastFileSkipped) QMetaObject::invokeMethod(this, "processFiles_TrackGain", Qt::QueuedConnection);
         //
     }
@@ -534,6 +528,10 @@ void QtGain::processFoundFiles(){
         }
 
         setGUI_To(BUSY);
+        // ******************************************
+        // New code for skipping FLAC files 4/4
+        this->m_lastFileSkipped = false;
+        //
         isTrackGain() ? processFiles_TrackGain() : processFiles_AlbumGain();
     } else {
         setGUI_To(IDLE, tr("No media files found..."));
