--- src/OpenFOAM-2.0.0/src/parallel/decompose/metisDecomp/Make/options	2011-06-17 11:45:29.020013982 -0500
+++ src/OpenFOAM-2.0.0/src/parallel/decompose/metisDecomp/Make/options	2011-06-17 12:02:53.292574934 -0500
@@ -1,7 +1,7 @@
 EXE_INC = \
     -I$(WM_THIRD_PARTY_DIR)/metis-5.0pre2/include \
-    -I../../dummyThirdParty/metisDecomp  /* location of dummy metis.h */ \
+    -I/usr/include/metis \
     -I../decompositionMethods/lnInclude
 
 LIB_LIBS = \
-    -L$(FOAM_EXT_LIBBIN) -lmetis -lGKlib
+    -L$(FOAM_EXT_LIBBIN) -lparmetis
