diff -aur mse-2.0.0/data/yugioh.mse-game/auto_replace mse-2.0.0.mod/data/yugioh.mse-game/auto_replace
--- mse-2.0.0/data/yugioh.mse-game/auto_replace	2011-02-05 14:12:00.000000000 -0800
+++ mse-2.0.0.mod/data/yugioh.mse-game/auto_replace	2011-10-20 09:44:30.000000000 -0700
@@ -13,4 +13,4 @@
 	whole word: false
 auto replace:
 	match: ~
-	replace: CARDNAME
\ No newline at end of file
+	replace: CARDNAME
diff -aur mse-2.0.0/data/yugioh.mse-game/card_fields mse-2.0.0.mod/data/yugioh.mse-game/card_fields
--- mse-2.0.0/data/yugioh.mse-game/card_fields	2011-02-05 14:12:00.000000000 -0800
+++ mse-2.0.0.mod/data/yugioh.mse-game/card_fields	2011-10-20 09:44:30.000000000 -0700
@@ -10,6 +10,7 @@
 	choice: synchro monster
 	choice: dark synchro monster
 	choice: fusion monster
+	choice: xyz monster
 	choice: token monster
 	choice: obelisk
 	choice: slifer
@@ -25,6 +26,7 @@
 		synchro monster:   rgb(26,26,26)
 		dark synchro monster:   rgb(26,26,26)
 		fusion monster:	   rgb(26,26,26)
+		xyz monster:	   rgb(26,26,26)
 		token monster:	   rgb(120,18,0)
 		obelisk:	   rgb(26,26,26)
 		slifer:		   rgb(26,26,26)
diff -aur mse-2.0.0/data/yugioh.mse-game/script mse-2.0.0.mod/data/yugioh.mse-game/script
--- mse-2.0.0/data/yugioh.mse-game/script	2011-02-05 14:12:00.000000000 -0800
+++ mse-2.0.0.mod/data/yugioh.mse-game/script	2011-10-20 09:44:30.000000000 -0700
@@ -76,6 +76,7 @@
 		else if card.type_1	== "<word-list-monster>Divine-Beast</word-list-monster>"		then "obelisk"
 		else if card.type_1	== "<word-list-monster>Legendary Dragon</word-list-monster>"		then "legendary dragons"
 		else if	contains(card.monster_type, match:"Fusion")						then "fusion monster"
+		else if	contains(card.monster_type, match:"Xyz")						then "xyz monster"
 		else if	contains(card.monster_type, match:"Ritual")						then "ritual monster"
 		else if	contains(card.monster_type, match:"Dark Synchro")					then "dark synchro monster"
 		else if	contains(card.monster_type, match:"Dark Tuner")						then "effect monster"
@@ -122,6 +123,7 @@
 		card.card_type == "dark synchro monster" or
 	        card.card_type == "synchro monster" or
 	        card.card_type == "fusion monster" or
+	        card.card_type == "xyz monster" or
 	        card.card_type == "token monster" or
 		card.card_type == "obelisk"        or
 		card.card_type == "slifer"	   or
@@ -159,13 +161,14 @@
 		else if chosen(choice:"ritual monster", type) then "E"
 		else if chosen(choice:"obelisk", type) then "F"
 		else if chosen(choice:"fusion monster", type) then "G"
-		else if chosen(choice:"synchro monster", type) then "H"
-		else if chosen(choice:"dark synchro monster", type) then "I"
-		else if chosen(choice:"token monster", type) then "J"
-		else if chosen(choice:"legendary dragons", type) then "K"
-		else if chosen(choice:"spell card", type) then "L"
-		else if chosen(choice:"trap card", type) then "M"
-		else "N"
+		else if chosen(choice:"xyz monster", type) then "H"
+		else if chosen(choice:"synchro monster", type) then "I"
+		else if chosen(choice:"dark synchro monster", type) then "J"
+		else if chosen(choice:"token monster", type) then "K"
+		else if chosen(choice:"legendary dragons", type) then "L"
+		else if chosen(choice:"spell card", type) then "M"
+		else if chosen(choice:"trap card", type) then "N"
+		else "O"
 	}
 	
 	pos_of_card := {
Only in mse-2.0.0.mod/data/yugioh.mse-game: script.orig
diff -aur mse-2.0.0/data/yugioh.mse-game/word_lists mse-2.0.0.mod/data/yugioh.mse-game/word_lists
--- mse-2.0.0/data/yugioh.mse-game/word_lists	2011-02-05 14:12:00.000000000 -0800
+++ mse-2.0.0.mod/data/yugioh.mse-game/word_lists	2011-10-20 09:55:58.000000000 -0700
@@ -36,6 +36,7 @@
 	word: Ritual
 	word: Spirit
 	word: Synchro
+	word: Xyz
 	word: Toon
 	word: Tuner
 	word: Union
diff -aur mse-2.0.0/data/yugioh-standard-levels.mse-symbol-font/symbol-font mse-2.0.0.mod/data/yugioh-standard-levels.mse-symbol-font/symbol-font
--- mse-2.0.0/data/yugioh-standard-levels.mse-symbol-font/symbol-font	2011-02-05 14:12:00.000000000 -0800
+++ mse-2.0.0.mod/data/yugioh-standard-levels.mse-symbol-font/symbol-font	2011-10-20 09:54:31.000000000 -0700
@@ -36,6 +36,11 @@
 	image font size: 28
 symbol:
 	code: *
+	image: xstar.png
+	enabled: {card.card_type=="xyz monster"}
+	image font size: 20
+symbol:
+	code: *
 	image: dstar.png
 	enabled: {card.card_type=="dark synchro monster"}
 	image font size: 20
diff -aur mse-2.0.0/data/yugioh-standard.mse-style/style mse-2.0.0.mod/data/yugioh-standard.mse-style/style
--- mse-2.0.0/data/yugioh-standard.mse-style/style	2011-02-05 14:12:00.000000000 -0800
+++ mse-2.0.0.mod/data/yugioh-standard.mse-style/style	2011-10-20 10:32:52.193957061 -0700
@@ -56,6 +56,7 @@
 			ritual monster: card-ritual.jpg
 			synchro monster: card-synchro.jpg
 			dark synchro monster: card-dsynchro.jpg
+			xyz monster: card-xyz.jpg
 			fusion monster: card-fusion.jpg
 			token monster: card-token.jpg
 			obelisk: card-obelisk.jpg
@@ -100,7 +101,7 @@
 				else if card.rarity=="ultimate rare" then rgb(60,0,0)
 				else if card.rarity=="parallel rare" then rgb(216,199,53)
 				else if card.rarity=="gold tech" then rgb(216,199,53)
-				else if card.card_type=="dark synchro monster" then rgb(255,255,255)
+				else if card.card_type=="dark synchro monster" or card.card_type=="xyz monster" then rgb(255,255,255)
 				else if is_nmonster() then rgb(255,255,255)
 				else rgb(0,0,0)
 	attribute:
@@ -127,7 +128,7 @@
 		height: 28
 		width: {if is_nmonster() then 275 else 280 }
 		z index: 2
-		alignment: { if card.card_type=="dark synchro monster" then "middle left" else "middle right" }
+		alignment: { if card.card_type=="dark synchro monster" or card.card_type=="xyz monster" then "middle left" else "middle right" }
 		font:
 			name: MatrixBoldSmallCaps
 			italic name: Matrix
@@ -174,8 +175,8 @@
 		font:
 			name: Palatino Linotype
 			size: 8
-			color: 
-				script: if card.card_type=="dark synchro monster" then rgb(255,255,255)
+			color:
+				script: if card.card_type=="dark synchro monster" or card.card_type=="xyz monster" then rgb(255,255,255)
 				else rgb(0,0,0)
 	edition:
 		left: 43
@@ -186,8 +187,8 @@
 		font:
 			name: { if card.edition=="DUEL TERMINAL" then "BankGothic Md BT" else "Palatino Linotype Bold" }
 			size: { if card.edition=="DUEL TERMINAL" then "10" else "8" }
-			color: 
-				script: if card.card_type=="dark synchro monster" then rgb(255,255,255)
+			color:
+				script: if card.card_type=="dark synchro monster" or card.card_type=="xyz monster" then rgb(255,255,255)
 				else rgb(0,0,0)
 	############################# Text box
 	rule text:
@@ -265,8 +266,8 @@
 		font:
 			name: Palatino Linotype
 			size: 8
-			color: 
-				script: if card.card_type=="dark synchro monster" then rgb(255,255,255)
+			color:
+				script: if card.card_type=="dark synchro monster" or card.card_type=="xyz monster" then rgb(255,255,255)
 				else rgb(0,0,0)
 	rarity:
 		left: 332
@@ -306,8 +307,8 @@
 		font:
 			name: Palatino Linotype
 			size: 7
-			color: 
-				script: if card.card_type=="dark synchro monster" then rgb(255,255,255)
+			color:
+				script: if card.card_type=="dark synchro monster" or card.card_type=="xyz monster" then rgb(255,255,255)
 				else rgb(0,0,0)
 ############################################extra card field:
 extra card field:
