diff -rupN a/usr/share/tunesviewer/firstsetup.py b/usr/share/tunesviewer/firstsetup.py
--- a/usr/share/tunesviewer/firstsetup.py	2011-05-19 01:54:50.000000000 +0700
+++ b/usr/share/tunesviewer/firstsetup.py	2011-08-07 14:15:25.155887749 +0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #Initial setup - sets up browser integration.
 #This is also called by configbox.
 
diff -rupN a/usr/share/tunesviewer/TunesViewer.py b/usr/share/tunesviewer/TunesViewer.py
--- a/usr/share/tunesviewer/TunesViewer.py	2011-07-14 23:36:06.000000000 +0700
+++ b/usr/share/tunesviewer/TunesViewer.py	2011-08-07 14:15:18.595923287 +0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # TunesViewer
 # A small, easy-to-use tool to access iTunesU and podcast media.
