diff -cr tunesviewer-1.0/usr/share/tunesviewer/firstsetup.py tunesviewer-1.0-mod/usr/share/tunesviewer/firstsetup.py
*** tunesviewer-1.0/usr/share/tunesviewer/firstsetup.py	2010-03-17 13:20:46.000000000 +0700
--- tunesviewer-1.0-mod/usr/share/tunesviewer/firstsetup.py	2011-02-11 20:35:28.670000180 +0700
***************
*** 1,4 ****
! #!/usr/bin/env python
  #Initial setup - set up browser.
  #This needs to be run as user, not root.
  
--- 1,4 ----
! #!/usr/bin/env python2
  #Initial setup - set up browser.
  #This needs to be run as user, not root.
  
diff -cr tunesviewer-1.0/usr/share/tunesviewer/TunesViewer.py tunesviewer-1.0-mod/usr/share/tunesviewer/TunesViewer.py
*** tunesviewer-1.0/usr/share/tunesviewer/TunesViewer.py	2011-02-10 01:45:58.000000000 +0700
--- tunesviewer-1.0-mod/usr/share/tunesviewer/TunesViewer.py	2011-02-11 20:35:38.256666841 +0700
***************
*** 1,4 ****
! #!/usr/bin/env python
  
  # TunesViewer
  # Small, easy-to-use tool to access iTunesU and podcast media.
--- 1,4 ----
! #!/usr/bin/env python2
  
  # TunesViewer
  # Small, easy-to-use tool to access iTunesU and podcast media.
