diff -Naur ./cinnamon-control-center.orig/panels/keyboard/keyboard-shortcuts.c ./cinnamon-control-center/panels/keyboard/keyboard-shortcuts.c
--- ./cinnamon-control-center.orig/panels/keyboard/keyboard-shortcuts.c	2013-01-26 20:26:16.003595190 -0500
+++ ./cinnamon-control-center/panels/keyboard/keyboard-shortcuts.c	2013-01-26 20:27:09.293601269 -0500
@@ -1626,7 +1626,7 @@
   gtk_tree_view_column_set_sort_column_id (column, DETAIL_DESCRIPTION_COLUMN);
 
   renderer = (GtkCellRenderer *) g_object_new (GTK_TYPE_CELL_RENDERER_ACCEL,
-                                               "accel-mode", GTK_CELL_RENDERER_ACCEL_MODE_MODIFIER_TAP,
+                                               "accel-mode", GTK_CELL_RENDERER_ACCEL_MODE_OTHER,
                                                NULL);
 
   g_signal_connect (renderer, "accel_edited",
