diff -aurN memwrite-1.0.3/src/memrw/Types.cpp fixed/src/memrw/Types.cpp
--- memwrite-1.0.3/src/memrw/Types.cpp	2012-12-03 17:38:58.000000000 +0200
+++ fixed/src/memrw/Types.cpp	2014-06-07 21:46:29.000000000 +0300
@@ -217,10 +217,12 @@
             }
             break;
         case MEM_TYPE_STRING:
-            char tmpStr[m_size + 1];
-            memcpy(tmpStr, m_data, m_size);
-            tmpStr[m_size] = '\0';
-            return tmpStr;
+            {
+                char tmpStr[m_size + 1];
+                memcpy(tmpStr, m_data, m_size);
+                tmpStr[m_size] = '\0';
+                return tmpStr;
+            }
             break;
         MACROS_ALL_TYPE_BLOC(MACROS_GET_VALUE_STR_BLOC)
         default:
