diff -aurN memwrite-1.0.3/src/findFile.cpp fixed/src/findFile.cpp
--- memwrite-1.0.3/src/findFile.cpp	2012-12-03 17:38:58.000000000 +0200
+++ fixed/src/findFile.cpp	2014-02-17 04:51:12.000000000 +0200
@@ -37,14 +37,14 @@
     }
 
     #if defined(__UNIX__)
-    testPath = wxString(wxStandardPaths().GetInstallPrefix() + _T("/share/") + _T(PACKAGE) + _T("/") + str);
+    testPath = wxString(wxStandardPaths::Get().GetInstallPrefix() + _T("/share/") + _T(PACKAGE) + _T("/") + str);
     if(wxFileExists(testPath) || wxDirExists(testPath)) // Si c'est dans le dossier /usr/.../share/nomprog
     {
         return testPath;
     }
     #endif // defined(__UNIX__)
 
-    testPath = wxString(wxStandardPaths().GetDataDir() + _T("/") + str);
+    testPath = wxString(wxStandardPaths::Get().GetDataDir() + _T("/") + str);
     if(wxFileExists(testPath) || wxDirExists(testPath)) // Si c'est dans le dossier /usr/.../share/nomPaquet
     {
         return testPath;
diff -aurN memwrite-1.0.3/src/launch-memwrite fixed/src/launch-memwrite
--- memwrite-1.0.3/src/launch-memwrite	1970-01-01 03:00:00.000000000 +0300
+++ fixed/src/launch-memwrite	2014-02-17 05:38:35.770882075 +0200
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Config
+	# Le script quitte en cas de variables non déclarée
+set -o nounset
+
+scriptPath=$(readlink -f $0)
+scriptDir=$(dirname ${scriptPath})
+
+CFG_FILE='/proc/sys/kernel/yama/ptrace_scope'
+NORMAL_USER="$USER"
+EXEC='memwrite'
+PTRACE_OK='0'
+
+if [ ! -f "$CFG_FILE" ]; then # Pas de fichier de conf => On peut rien faire
+	"$EXEC"
+	exit 0
+fi
+
+if [ "$#" -eq 1 ] && [ "$USER" = 'root' ]; then
+	PTRACE_PREVIOUS=$(cat "$CFG_FILE")
+	echo "$PTRACE_OK" > "$CFG_FILE"
+	su -c "$EXEC" "$1" # Launch memwrite
+	echo "$PTRACE_PREVIOUS" > "$CFG_FILE"
+else
+	# Verification du comportement de ptrace
+	if [ ! "$(cat "$CFG_FILE")" = "$PTRACE_OK" ]; then
+		gksudo --description 'Memwrite' "$scriptPath" "$USER"
+	else # Comportement déjà ok
+		"$EXEC"
+	fi
+fi
+
+exit 0
diff -aurN memwrite-1.0.3/src/linux-menu/memwrite.desktop fixed/src/linux-menu/memwrite.desktop
--- memwrite-1.0.3/src/linux-menu/memwrite.desktop	2012-12-03 17:38:58.000000000 +0200
+++ fixed/src/linux-menu/memwrite.desktop	2014-02-17 05:37:41.500548978 +0200
@@ -2,7 +2,7 @@
 [Desktop Entry]
 Version=1.0
 Encoding=UTF-8
-Exec=lunch-memwrite
+Exec=launch-memwrite
 Icon=memwrite-icon.png
 Terminal=false
 X-MultipleArgs=false
diff -aurN memwrite-1.0.3/src/lunch-memwrite fixed/src/lunch-memwrite
--- memwrite-1.0.3/src/lunch-memwrite	2012-12-03 17:38:58.000000000 +0200
+++ fixed/src/lunch-memwrite	1970-01-01 03:00:00.000000000 +0300
@@ -1,34 +0,0 @@
-#!/bin/bash
-
-# Config
-	# Le script quitte en cas de variables non déclarée
-set -o nounset
-
-scriptPath=$(readlink -f $0)
-scriptDir=$(dirname ${scriptPath})
-
-CFG_FILE='/proc/sys/kernel/yama/ptrace_scope'
-NORMAL_USER="$USER"
-EXEC='memwrite'
-PTRACE_OK='0'
-
-if [ ! -f "$CFG_FILE" ]; then # Pas de fichier de conf => On peut rien faire
-	"$EXEC"
-	exit 0
-fi
-
-if [ "$#" -eq 1 ] && [ "$USER" = 'root' ]; then
-	PTRACE_PREVIOUS=$(cat "$CFG_FILE")
-	echo "$PTRACE_OK" > "$CFG_FILE"
-	su -c "$EXEC" "$1" # Lunch memwrite
-	echo "$PTRACE_PREVIOUS" > "$CFG_FILE"
-else
-	# Verification du comportement de ptrace
-	if [ ! "$(cat "$CFG_FILE")" = "$PTRACE_OK" ]; then
-		gksudo --description 'Memwrite' "$scriptPath" "$USER"
-	else # Comportement déjà ok
-		"$EXEC"
-	fi
-fi
-
-exit 0
diff -aurN memwrite-1.0.3/src/Makefile.am fixed/src/Makefile.am
--- memwrite-1.0.3/src/Makefile.am	2012-12-03 17:38:58.000000000 +0200
+++ fixed/src/Makefile.am	2014-02-17 05:37:30.683815921 +0200
@@ -16,7 +16,7 @@
 	DialogNumber.hpp DialogNumber.cpp DialogTypesList.hpp \
 	DialogTypesList.cpp
 	
-bin_SCRIPTS = lunch-memwrite
+bin_SCRIPTS = launch-memwrite
 
 
 ## --------------- Resources windows ---------------
@@ -38,4 +38,4 @@
 	"$(srcdir)/installDll.sh" "$(prefix)" "$(host)"
 endif
 
-EXTRA_DIST = $(srcdir)/*.cbp $(man_MANS) $(srcdir)/rc.rc $(srcdir)/lunch-memwrite $(srcdir)/installDll.sh
+EXTRA_DIST = $(srcdir)/*.cbp $(man_MANS) $(srcdir)/rc.rc $(srcdir)/launch-memwrite $(srcdir)/installDll.sh
diff -aurN memwrite-1.0.3/src/Makefile.in fixed/src/Makefile.in
--- memwrite-1.0.3/src/Makefile.in	2012-12-03 17:38:58.000000000 +0200
+++ fixed/src/Makefile.in	2014-02-17 05:37:04.206986746 +0200
@@ -315,11 +315,11 @@
 	DialogNumber.hpp DialogNumber.cpp DialogTypesList.hpp \
 	DialogTypesList.cpp
 
-bin_SCRIPTS = lunch-memwrite
+bin_SCRIPTS = launch-memwrite
 @HAVE_WINDRES_FALSE@memwrite_SOURCES = $(memwriteSrcBase)
 @HAVE_WINDRES_TRUE@memwrite_SOURCES = $(memwriteSrcBase) rc.rc
 man_MANS = memwrite.1
-EXTRA_DIST = $(srcdir)/*.cbp $(man_MANS) $(srcdir)/rc.rc $(srcdir)/lunch-memwrite $(srcdir)/installDll.sh
+EXTRA_DIST = $(srcdir)/*.cbp $(man_MANS) $(srcdir)/rc.rc $(srcdir)/launch-memwrite $(srcdir)/installDll.sh
 all: config.hpp
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
