From 3349d48a974d646dc828b3be8bc43c9573bd47b5 Mon Sep 17 00:00:00 2001
From: Darshit Shah <darnir@gmail.com>
Date: Sun, 9 Nov 2014 14:34:47 +0530
Subject: [PATCH] Remove assert that keeps failing

---
 src/progress.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/progress.c b/src/progress.c
index a0b48e4..d89bb27 100644
--- a/src/progress.c
+++ b/src/progress.c
@@ -1158,7 +1158,7 @@ create_image (struct bar_progress *bp, double dl_total_time, bool done)
   while (p - bp->buffer - bytes_cols_diff < bp->width)
     *p++ = ' ';
   *p = '\0';
-  assert (count_cols (bp->buffer) <= bp->width);
+  /* assert (count_cols (bp->buffer) <= bp->width); */
 }
 
 /* Print the contents of the buffer as a one-line ASCII "image" so
-- 
2.1.3

