diff -aur distribute-0.6.14.old/distribute_setup.py distribute-0.6.14.new/distribute_setup.py
--- distribute-0.6.14.old/distribute_setup.py	2010-11-21 23:27:39.254299041 -0800
+++ distribute-0.6.14.new/distribute_setup.py	2010-11-21 23:32:07.153614815 -0800
@@ -299,8 +299,7 @@
         log.warn('Could not find the install location')
         return
     pyver = '%s.%s' % (sys.version_info[0], sys.version_info[1])
-    setuptools_file = 'setuptools-%s-py%s.egg-info' % \
-            (SETUPTOOLS_FAKED_VERSION, pyver)
+    setuptools_file = 'setuptools-%s.egg-info' % SETUPTOOLS_FAKED_VERSION
     pkg_info = os.path.join(placeholder, setuptools_file)
     if os.path.exists(pkg_info):
         log.warn('%s already exists', pkg_info)
diff -aur distribute-0.6.14.old/setup.py distribute-0.6.14.new/setup.py
--- distribute-0.6.14.old/setup.py	2010-11-21 23:27:39.254299041 -0800
+++ distribute-0.6.14.new/setup.py	2010-11-21 23:31:00.953399890 -0800
@@ -182,11 +182,8 @@
             "dependency_links.txt = setuptools.command.egg_info:overwrite_arg",
         ],
 
-        "console_scripts": [
-             "easy_install = setuptools.command.easy_install:main",
-             "easy_install-%s = setuptools.command.easy_install:main"
-                % sys.version[:3]
-        ],
+        "console_scripts":
+            ["easy_install-%s = setuptools.command.easy_install:main" % sys.version[:3]],
 
         "setuptools.file_finders":
             ["svn_cvs = setuptools.command.sdist:_default_revctrl"],
