Index: file-roller-3.4.1/src/fr-command-7z.c
===================================================================
--- file-roller-3.4.1.orig/src/fr-command-7z.c	2011-09-01 19:10:30.000000000 +0800
+++ file-roller-3.4.1/src/fr-command-7z.c	2012-07-12 12:53:51.000000000 +0800
@@ -194,7 +194,11 @@
 static void
 fr_command_7z_begin_command (FrCommand *comm)
 {
-	if (is_program_in_path ("7z"))
+	if (is_mime_type(comm->mime_type, "application/zip")
+        || is_mime_type(comm->mime_type, "application/x-cbz")
+        && is_program_in_path("7za"))
+        fr_process_begin_command (comm->process, "7za");
+    else if (is_program_in_path("7z"))
 		fr_process_begin_command (comm->process, "7z");
 	else if (is_program_in_path ("7za"))
 		fr_process_begin_command (comm->process, "7za");
