diff -rupN notify-osd.orig//src/defaults.c notify-osd/src/defaults.c
--- notify-osd.orig//src/defaults.c	2010-03-09 12:33:56.391464000 +0100
+++ notify-osd/src/defaults.c	2010-06-04 14:12:05.608277667 +0200
@@ -147,7 +147,7 @@ enum
  * the visual guide for jaunty-notifications */
 #define DEFAULT_FADE_IN_TIMEOUT      250
 #define DEFAULT_FADE_OUT_TIMEOUT     1000
-#define DEFAULT_ON_SCREEN_TIMEOUT    10000
+#define DEFAULT_ON_SCREEN_TIMEOUT    3000
 
 /* GConf-keys to watch */
 #define GCONF_UI_FONT_NAME        "/desktop/gnome/interface/font_name"
