diff --git a/gmailfs.py b/gmailfs.py
index 39c8b64..0da0f5e 100755
--- a/gmailfs.py
+++ b/gmailfs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
 #@+leo-ver=4
 #@+node:@file gmailfs.py
 #@@first
@@ -1701,7 +1701,7 @@ class Gmailfs(Fuse):
 	# messages with the given label
 	#self.imap.debug = 4
 	trash_all = 0
-	if "IMAPFS_TRASH_ALL" in os.environ:
+	if os.environ.get("IMAPFS_TRASH_ALL"):
 		trash_all = 1
 	if trash_all:
 		print("deleting existing messages...")
