diff -rupN lunatic-python-1.0/python.lua lunatic-python-1.0-new/python.lua
--- lunatic-python-1.0/python.lua	2003-12-13 02:37:57.000000000 -0200
+++ lunatic-python-1.0-new/python.lua	2013-06-08 22:17:24.000000000 -0300
@@ -1,6 +1,6 @@
 local path = os.getenv("LUA_SOPATH")
 if path then
-	func = loadlib(path.."/lua-python.so", "luaopen_python")
+	func = package.loadlib(path.."/lua-python.so", "luaopen_python")
 	if func then
 		func()
 		return
@@ -10,7 +10,7 @@ local modmask = "/usr/lib/python%d.%d/si
 local loaded = false
 for i = 10, 2, -1 do
 	for j = 10, 2, -1 do
-		func = loadlib(string.format(modmask, i, j), "luaopen_python")
+		func = package.loadlib(string.format(modmask, i, j), "luaopen_python")
 		if func then
 			loaded = true
 			func()
diff -rupN lunatic-python-1.0/setup.py lunatic-python-1.0-new/setup.py
--- lunatic-python-1.0/setup.py	2005-10-18 21:10:07.000000000 -0200
+++ lunatic-python-1.0-new/setup.py	2013-06-08 22:17:24.000000000 -0300
@@ -9,7 +9,7 @@ if os.path.isfile("MANIFEST"):
 # You may have to change these
 PYLIBS = ["python"+get_python_version(), "pthread", "util"]
 PYLIBDIR = [get_python_lib(standard_lib=True)+"/config"]
-LUALIBS = ["lua", "lualib"]
+LUALIBS = ["lua"]
 LUALIBDIR = []
 
 setup(name="lunatic-python",
diff -rupN lunatic-python-1.0/src/luainpython.c lunatic-python-1.0-new/src/luainpython.c
--- lunatic-python-1.0/src/luainpython.c	2005-10-18 21:07:02.000000000 -0200
+++ lunatic-python-1.0-new/src/luainpython.c	2013-06-08 22:19:59.000000000 -0300
@@ -22,9 +22,9 @@
 */
 #include <Python.h>
 
-#include <lua.h>
-#include <lauxlib.h>
-#include <lualib.h>
+#include <lua5.1/lua.h>
+#include <lua5.1/lauxlib.h>
+#include <lua5.1/lualib.h>
 
 #include "pythoninlua.h"
 #include "luainpython.h"
@@ -488,12 +488,12 @@ initlua(void)
 
 	if (!L) {
 		L = lua_open();
-		luaopen_base(L);
-		luaopen_table(L);
-		luaopen_io(L);
-		luaopen_string(L);
-		luaopen_debug(L);
-		luaopen_loadlib(L);
+
+		/* loading each lib separately has some deep conflict
+		 * with python's readline module so we obey the holy
+		 * docs by lua people and use the magic loader.
+		 */
+		luaL_openlibs(L);
 		luaopen_python(L);
 		lua_settop(L, 0);
 	}
diff -rupN lunatic-python-1.0/src/pythoninlua.c lunatic-python-1.0-new/src/pythoninlua.c
--- lunatic-python-1.0/src/pythoninlua.c	2005-10-18 21:07:07.000000000 -0200
+++ lunatic-python-1.0-new/src/pythoninlua.c	2013-06-08 22:19:53.000000000 -0300
@@ -22,8 +22,8 @@
 */
 #include <Python.h>
 
-#include <lua.h>
-#include <lauxlib.h>
+#include <lua5.1/lua.h>
+#include <lua5.1/lauxlib.h>
 
 #include "pythoninlua.h"
 #include "luainpython.h"
@@ -340,7 +340,7 @@ static int py_object_tostring(lua_State
 	return 1;
 }
 
-static const luaL_reg py_object_lib[] = {
+static const luaL_Reg py_object_lib[] = {
 	{"__call",	py_object_call},
 	{"__index",	py_object_index},
 	{"__newindex",	py_object_newindex},
@@ -542,7 +542,7 @@ static int py_import(lua_State *L)
 	return ret;
 }
 
-static const luaL_reg py_lib[] = {
+static const luaL_Reg py_lib[] = {
 	{"execute",	py_execute},
 	{"eval",	py_eval},
 	{"asindx",	py_asindx},
