diff --git a/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp b/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp
index 2c4ed21..e669b3c 100644
--- a/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp
+++ b/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp
@@ -276,7 +276,12 @@ disassemble(const void* func, llvm::raw_ostream & Out)
 #endif
    OwningPtr<TargetMachine> TM(T->createTargetMachine(Triple, sys::getHostCPUName(), "", options));
 
-#if HAVE_LLVM >= 0x0306
+#if HAVE_LLVM >= 0x0307
+   const TargetInstrInfo *TII = 0;
+   Out << "error: disabled with llvm >= 3.7 \n";
+   Out.flush();
+   return 0;
+#elif HAVE_LLVM >= 0x0306
    const TargetInstrInfo *TII = TM->getSubtargetImpl()->getInstrInfo();
 #else
    const TargetInstrInfo *TII = TM->getInstrInfo();
