--- old/src/gd_config.cpp	2014-03-26 22:40:45.000000000 +0100
+++ new/src/gd_config.cpp	2014-05-17 00:45:24.659416116 +0200
@@ -548,8 +548,6 @@
 		G_Conf.midiSystem = RtMidi::UNIX_JACK;
 	else if (!strcmp("Multimedia MIDI", system->text(system->value())))
 		G_Conf.midiSystem = RtMidi::WINDOWS_MM;
-	else if (!strcmp("Kernel Streaming MIDI", system->text(system->value())))
-		G_Conf.midiSystem = RtMidi::WINDOWS_KS;
 	else if (!strcmp("OSX Core MIDI", system->text(system->value())))
 		G_Conf.midiSystem = RtMidi::MACOSX_CORE;
 
@@ -581,8 +579,6 @@
 
 	if (kernelMidi::hasAPI(RtMidi::WINDOWS_MM))
 		system->add("Multimedia MIDI");
-	if (kernelMidi::hasAPI(RtMidi::WINDOWS_KS))
-		system->add("Kernel Streaming MIDI");
 
 #elif defined (__APPLE__)
 
@@ -594,7 +590,6 @@
 		case RtMidi::LINUX_ALSA:  system->show("ALSA"); break;
 		case RtMidi::UNIX_JACK:   system->show("Jack"); break;
 		case RtMidi::WINDOWS_MM:  system->show("Multimedia MIDI"); break;
-		case RtMidi::WINDOWS_KS:  system->show("Kernel Streaming MIDI"); break;
 		case RtMidi::MACOSX_CORE: system->show("OSX Core MIDI"); break;
 		default: system->value(0); break;
 	}
