diff -aur src.orig/src/gd_mainWindow.cpp src/src/gd_mainWindow.cpp
--- src.orig/src/gd_mainWindow.cpp	2012-04-20 00:21:13.000000000 +0200
+++ src/src/gd_mainWindow.cpp	2012-04-23 17:23:58.189906375 +0200
@@ -315,8 +315,8 @@
 	
 	
 	const Fl_Menu_Item *m = menu->popup(
-		mainWin->x() + mainWin->menu_file->x(), 
-		mainWin->y() + mainWin->menu_file->h(), 
+		mainWin->menu_file->x(), 
+		mainWin->menu_file->h(), 
 		0, 0, b);
 	if (!m) return;
 	
@@ -372,8 +372,8 @@
 	b->color(COLOR_BG_0); 
 	
 	const Fl_Menu_Item *m = menu->popup(
-		mainWin->x() + mainWin->menu_edit->x(), 
-		mainWin->y() + mainWin->menu_edit->h(), 
+		mainWin->menu_edit->x(), 
+		mainWin->menu_edit->h(), 
 		0, 0, b);
 	if (!m) return;
 	
diff -aur src.orig/src/ge_waveform.cpp src/src/ge_waveform.cpp
--- src.orig/src/ge_waveform.cpp	2012-04-20 00:21:13.000000000 +0200
+++ src/src/ge_waveform.cpp	2012-04-23 17:24:15.393505759 +0200
@@ -523,7 +523,7 @@
 	b->textcolor(COLOR_TEXT_0);
 	b->color(COLOR_BG_0); 
 	
-	const Fl_Menu_Item *m = menu->popup(Fl::event_x_root(), Fl::event_y_root(), 0, 0, b);
+	const Fl_Menu_Item *m = menu->popup(Fl::event_x(), Fl::event_y(), 0, 0, b);
 	if (!m) {
 		menuOpen = false;
 		return;
diff -aur src.orig/src/gg_keyboard.cpp src/src/gg_keyboard.cpp
--- src.orig/src/gg_keyboard.cpp	2012-04-20 00:21:13.000000000 +0200
+++ src/src/gg_keyboard.cpp	2012-04-23 17:23:58.189906375 +0200
@@ -133,7 +133,7 @@
 	b->textcolor(COLOR_TEXT_0);
 	b->color(COLOR_BG_0); 
 
-	const Fl_Menu_Item *m = rclick_menu->popup(Fl::event_x_root(), Fl::event_y_root(), 0, 0, b);
+	const Fl_Menu_Item *m = rclick_menu->popup(sampleButton[chan]->x(), sampleButton[chan]->h() + sampleButton[chan]->y(), 0, 0, b);
 	if (!m) return;
 
 	if (strcmp(m->label(), "Load new sample...") == 0) {
