--- cbind/CMakeLists.txt	2010-02-24 07:48:29.000000000 -0600
+++ cbind/CMakeLists.txt	2010-05-04 10:10:46.937211933 -0500
@@ -5,7 +5,7 @@
    $ENV{NETCDF_DIR}/libsrc
 )	
 
-find_path( NETCDF_LIBRARY libnetcdf.a
+find_path( NETCDF_LIBRARY libnetcdf.so
    $ENV{ACCESS}/lib
    $ENV{NETCDF_DIR}/lib
    $ENV{NETCDF_DIR}/libsrc/.libs
@@ -224,7 +224,7 @@
 
 ADD_LIBRARY(exoIIv2c ${cbind_SRCS})
 
-TARGET_LINK_LIBRARIES(exoIIv2c ${NETCDF_LIBRARY}/libnetcdf.a)
+TARGET_LINK_LIBRARIES(exoIIv2c ${NETCDF_LIBRARY}/libnetcdf.so)
 
 IF(VTK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(exoIIv2c PROPERTIES ${VTK_LIBRARY_PROPERTIES})
