From 1ed2c0d662f09c927c750abdb874e86b4de750e0 Mon Sep 17 00:00:00 2001
From: Daniel Wallace <daniel.wallace@gatech.edu>
Date: Sat, 4 Aug 2012 01:21:00 -0400
Subject: [PATCH 4/4] remove lboost_thread-mt

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 2c26469..15b2e96 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -127,7 +127,7 @@ DIRS= $(BIN_DIRS) $(EC_DIRS) $(XBMCTEX_DIRS) $(DVDPCODECS_DIRS) $(PAPCODECS_DIRS
 	$(LIB_DIRS) $(SS_DIRS) $(VIS_DIRS) $(SKIN_DIRS) $(LIVE_DIRS)
 
 LIBS=@LIBS@
-LIBS+=-lboost_system -lboost_filesystem -lboost_thread -lboost_thread-mt -lboost_iostreams -lboost_program_options -lboost_regex -lboost_signals
+LIBS+=-lboost_system -lboost_filesystem -lboost_thread -lboost_iostreams -lboost_program_options -lboost_regex -lboost_signals
 
 CFLAGS=@CFLAGS@
 CXXFLAGS=@CXXFLAGS@
-- 
1.7.11.4

