diff -u laika-build/Makefile laika-build/Makefile
--- laika-build/Makefile	2012-03-10 23:00:57.652626704 -0500
+++ laika-build/Makefile	2012-03-10 23:01:17.726256195 -0500
@@ -127,7 +127,7 @@
 	$(LIB_DIRS) $(SS_DIRS) $(VIS_DIRS) $(SKIN_DIRS) $(LIVE_DIRS)
 
 LIBS=-lva -lva-glx -lmicrohttpd -lSDL_image -lrt -ltiff -ljasper -lresolv -ldl -lSDL_mixer -lsmbclient -lfaad -lmysqlclient -lmodplug -lmpeg2convert -lmpeg2 -lwavpack -lcrypto -lssl -lz -llzo2 -lpthread -lbz2 -lass -lGLU -lGLEW -lGL  -lmad   -lfontconfig   -lfribidi   -lsqlite3   -lpng15   -lpcre   -lpcre -lpcrecpp   -lcdio -lm   -lsamplerate   -lfreetype   -logg   -lvorbis -lm -logg   -lvorbisenc -lvorbis -lm -logg   -L/usr/lib -lmysqlclient -lpthread -lz -lm -lrt -lssl -lcrypto -ldl -lasound   -lenca -lm   -lXt -lX11   -lXtst   -lXext   -lXmu -lXt -lX11   -lXinerama   -lcurl   -ldbus-1 -lpthread -lrt   -lSDL -lpthread  
-LIBS+=-lboost_system -lboost_filesystem -lboost_thread -lboost_thread-mt -lboost_iostreams -lboost_program_options -lboost_regex -lboost_signals
+LIBS+=-lboost_system -lboost_filesystem -lboost_thread -lboost_iostreams -lboost_program_options -lboost_regex -lboost_signals
 
 CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fstack-protector --param=ssp-buffer-size=4 -D_FORTIFY_SOURCE=2 -fPIC -DPIC -D_REENTRANT -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -g -D_DEBUG -Wall -O2
 CXXFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fstack-protector --param=ssp-buffer-size=4 -D_FORTIFY_SOURCE=2 -fPIC -DPIC -D_REENTRANT -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -g -D_DEBUG -Wall -O2
