--- editorsave.cpp.orig	2013-08-02 19:30:21.255995070 +0200
+++ editorsave.cpp	2013-08-02 19:27:39.982667083 +0200
@@ -635,15 +635,14 @@
             file << "<bye>"<< StripXML(board->bye) <<"</bye>\n";
         }
         saveColorStuff(file, board);
-    
-        int i,j;
+  
         file << "\t<ground>\n";
         resetSavedPosition();
-        for (i=0;i<XYE_HORZ;i++) for (j=0;j<XYE_VERT;j++) saveGroundObject(file,editor::board->objects[i][j],i,XYE_VERT-j-1);
+        for (int i=0;i<XYE_HORZ;i++) for (int j=0;j<XYE_VERT;j++) saveGroundObject(file,editor::board->objects[i][j],i,XYE_VERT-j-1);
         file << "\t</ground>\n";
         file << "\t<objects>\n";
         resetSavedPosition();
-        for (i=0;i<XYE_HORZ;i++) for (j=0;j<XYE_VERT;j++)
+        for (int i=0;i<XYE_HORZ;i++) for (int j=0;j<XYE_VERT;j++)
         {
             saveNormalObject(file,editor::board->objects[i][j],i,XYE_VERT-j-1);
         }
