--- ./setup.py.orig	2011-04-23 00:02:41.202202000 +0000
+++ ./setup.py	2011-04-23 00:09:37.384855310 +0000
@@ -61,60 +61,15 @@
 
 # - screenlets-subdirs and all their data go into PREFIX/<name>/...
 all_dirlist		= scan_dir_list(['src'])
-all_files_list	= make_file_list(all_dirlist, 'share/screenlets/screenlets-pack-all', 
+all_files_list	= make_file_list(all_dirlist, 'share/screenlets/',
 	strip='src/')	# to strip this string from filenames
 
-basic_screenlets = [
-	# we don't add random screenlets, less is better
-	# we should add just one screenlet for each function (if possible)
-	# the one which is most usable for general public
-	# we should remove screenlets which do not work
-	#
-	# if there are too many basic screenlets, it's annoying for the new user
-	#
-	# usual desktop stuff
-	'src/Ruler', # pixel ruler
-	'src/Lipik', # sticky notes
-	'src/Clock', # do we need also digital or text based?
-	'src/Trash', # Trashbin
-	'src/Calc', # Calculator
-	'src/ClearCalendar', # calendar with some ics ability
-	'src/FolderView', # it may be not too usable, but still basic
-	
-	# music/audio related stuff
-	'src/NowPlaying', # Shows info for song currently playing
-	'src/Lyrics', # Querys and shows lyrics for songs while playing
-	
-	# just visual candy, there has to be some
-	'src/Flower', # some digital nature
-	'src/Sticker', # we are for the Linux (Tux Screenlet would be fun too)
-	
-	# data from web
-	'src/MailCheck', # IMAP and gmail
-	'src/Slideshow', # Media RSS, Flickr, folder on disk
-	'src/ClearRss', # Text RSS
-	
-	# weather, by now two options
-	'src/ClearWeather', # source: weather.com
-	'src/FreemeteoWeather', # source: freemeteo.com
-	
-	# system sensors
-	'src/Meter', # bar meter
-	'src/Sensors', # vertical graph and time graph
-	'src/Sysmonitor', # various indicators
-	
-#	'src/',
-	]
-	
-basic_dirlist		= scan_dir_list(basic_screenlets, addself=True)
-
-basic_files_list	= make_file_list(basic_dirlist, 'share/screenlets/screenlets-pack-basic', 
-	strip='src/')	# to strip this string from filenames
 
 # Install translation files
 buildcmd = "msgfmt -o build/locale/%s/LC_MESSAGES/%s.mo %s/%s.po"
 mopath = "build/locale/%s/LC_MESSAGES/%s.mo"
-destpath = "share/screenlets/%s/%s/mo/%s/LC_MESSAGES"
+destpath = "share/screenlets/%s/mo/%s/LC_MESSAGES"
+destpath = "share/screenlets/%s/mo/%s/LC_MESSAGES"
 for podir in os.listdir("src"):
 	full_po_path = "src/"+podir+"/po"
 	if os.path.exists(full_po_path):
@@ -127,9 +82,7 @@
 					if not os.path.isdir ("build/locale/%s/LC_MESSAGES" % dname):
 						os.makedirs ("build/locale/%s/LC_MESSAGES" % dname)
 					os.system (buildcmd % (dname, name, full_po_path, dname))
-					all_files_list.append ((destpath % ("screenlets-pack-all", podir, dname), [mopath % (dname,name)]))
-					if "src/"+podir in basic_screenlets:
-						basic_files_list.append ((destpath % ("screenlets-pack-basic", podir, dname), [mopath % (dname,name)]))
+					all_files_list.append ((destpath % (podir, dname), [mopath % (dname,name)]))
 
 setup(name = 'screenlets-pack-all',
 	version			= VERSION,
@@ -141,13 +94,3 @@
 	data_files		= all_files_list
 	)
 
-setup(name = 'screenlets-pack-basic',
-	version			= VERSION,
-	author			= 'Guido Tabbernuk',
-	author_email	= 'boamaod@gmail.com',
-	url				= 'http://www.screenlets.org',
-	license			= 'GPL v3',
-	description		= 'Package containing only basic screenlets.',
-	data_files		= basic_files_list
-	)
-
