--- ./setup.py.orig	2011-01-25 19:19:01.806666725 -0500
+++ ./setup.py	2011-01-25 19:17:18.893333391 -0500
@@ -61,29 +61,13 @@
 
 # - screenlets-subdirs and all their data go into PREFIX/<name>/...
 all_dirlist		= scan_dir_list(['src'])
-all_files_list	= make_file_list(all_dirlist, 'share/screenlets/screenlets-pack-all', 
-	strip='src/')	# to strip this string from filenames
-
-basic_screenlets = [
-	'src/Ruler',
-	'src/Lipik',
-	'src/Clock',
-	'src/Flower',
-	'src/Slideshow',
-	'src/Trash',
-	'src/Calc',
-#	'src/',
-	]
-	
-basic_dirlist		= scan_dir_list(basic_screenlets, addself=True)
-
-basic_files_list	= make_file_list(basic_dirlist, 'share/screenlets/screenlets-pack-basic', 
+all_files_list	= make_file_list(all_dirlist, 'share/screenlets/', 
 	strip='src/')	# to strip this string from filenames
 
 # Install translation files
 buildcmd = "msgfmt -o build/locale/%s/LC_MESSAGES/%s.mo %s/%s.po"
 mopath = "build/locale/%s/LC_MESSAGES/%s.mo"
-destpath = "share/screenlets/%s/%s/mo/%s/LC_MESSAGES"
+destpath = "share/screenlets/%s/mo/%s/LC_MESSAGES"
 for podir in os.listdir("src"):
 	full_po_path = "src/"+podir+"/po"
 	if os.path.exists(full_po_path):
@@ -96,9 +80,7 @@
 					if not os.path.isdir ("build/locale/%s/LC_MESSAGES" % dname):
 						os.makedirs ("build/locale/%s/LC_MESSAGES" % dname)
 					os.system (buildcmd % (dname, name, full_po_path, dname))
-					all_files_list.append ((destpath % ("screenlets-pack-all", podir, dname), [mopath % (dname,name)]))
-					if "src/"+podir in basic_screenlets:
-						basic_files_list.append ((destpath % ("screenlets-pack-basic", podir, dname), [mopath % (dname,name)]))
+					all_files_list.append ((destpath % (podir, dname), [mopath % (dname,name)]))
 
 setup(name = 'screenlets-pack-all',
 	version			= VERSION,
@@ -110,13 +92,3 @@
 	data_files		= all_files_list
 	)
 
-setup(name = 'screenlets-pack-basic',
-	version			= VERSION,
-	author			= 'Guido Tabbernuk',
-	author_email	= 'boamaod@gmail.com',
-	url				= 'http://www.screenlets.org',
-	license			= 'GPL v3',
-	description		= 'Package containing only basic screenlets.',
-	data_files		= basic_files_list
-	)
-
