--- Makefile.orig	2010-12-18 10:38:29.000000000 +0100
+++ Makefile	2010-12-19 16:52:41.736666670 +0100
@@ -129,8 +129,8 @@
 
 # Flags to enable native support of webcams, using the OpenCV library.
 OPENCV_CFLAGS = -Dcimg_use_opencv -I$(USR)/include/opencv
-OPENCV_LDFLAGS = -lcv -lhighgui
-# OPENCV_LDFLAGS = -lopencv_core -lopencv_highgui   # Use this for OpenCV 2.2.0 !
+# OPENCV_LDFLAGS = -lcv -lhighgui
+OPENCV_LDFLAGS = -lopencv_core -lopencv_highgui   # Use this for OpenCV 2.2.0 !
 
 # Flags to enable native support of most classical image file formats, using the GraphicsMagick++ library.
 MAGICK_CFLAGS = -Dcimg_use_magick -I$(USR)/include/GraphicsMagick
