diff -Nur rosinstall-0.6.9-orig/src/rosinstall/setupfiles.py rosinstall-0.6.9/src/rosinstall/setupfiles.py
--- rosinstall-0.6.9-orig/src/rosinstall/setupfiles.py  2012-03-20 05:52:38.000000000 +0100
+++ rosinstall-0.6.9/src/rosinstall/setupfiles.py       2012-03-26 00:37:03.335827848 +0200
@@ -96,7 +96,7 @@
 # use python script to read ros_package_path and setup-file elements
   text += """
 # python script to read .rosinstall even when rosnistall is not installed
-export _PARSED_CONFIG=`/usr/bin/env python << EOPYTHON
+export _PARSED_CONFIG=`/usr/bin/env python2 << EOPYTHON
 import sys, os, yaml;
 filename = '.rosinstall'
 if 'ROS_WORKSPACE' in os.environ:
@@ -150,7 +150,7 @@
 unset _ROS_PACKAGE_PATH_ROSINSTALL
 
 # using ROS_ROOT now being in ROS_PACKAGE_PATH
-export _ROS_ROOT_ROSINSTALL=`/usr/bin/env python << EOPYTHON
+export _ROS_ROOT_ROSINSTALL=`/usr/bin/env python2 << EOPYTHON
 import sys, os;
 if 'ROS_PACKAGE_PATH' in os.environ:
   pkg_path = os.environ['ROS_PACKAGE_PATH']
