diff --git a/README b/README
index 73425ca..5fc809e 100644
--- a/README
+++ b/README
@@ -8,8 +8,7 @@
 contained in  email messages.  It  is a replacement for  the "urlview"
 program.
 
-  urlscan  requires Python and  the python-urwid  library, as  well as
-  sensible-browser from the debianutils package.
+  urlscan  requires Python and  the python-urwid  library.
 
 1) Features
 
@@ -38,9 +37,7 @@ macro attach,compose \cb "<pipe-entry> urlscan<Enter>" "call urlscan to extract
   Once  this  is done,  Control-b  while  reading  mail in  mutt  will
 automatically invoke urlscan on the message.
 
-  urlscan uses  sensible-browser to invoke the default  Web browser of
-the  current environment.   To choose  a particular  browser,  set the
-environment variable BROWSER; e.g.,
+  To choose  a particular  browser,  set the environment variable BROWSER; e.g.,
 
 export BROWSER=/usr/bin/epiphany
 
diff --git a/modules/urlscan/browser.py b/modules/urlscan/browser.py
index a86a1ca..249ae62 100644
--- a/modules/urlscan/browser.py
+++ b/modules/urlscan/browser.py
@@ -17,11 +17,11 @@
 
 '''Contains logic to invoke the default system browser.'''
 
-import os
+import os, subprocess
 
 def browseto(url, background = False):
     if background:
-        mode = os.P_NOWAIT
+        cmd = subprocess.call
     else:
-        mode = os.P_WAIT
-    os.spawnl(mode, '/usr/bin/sensible-browser', '/usr/bin/sensible-browser', url)
+        cmd = subprocess.Popen
+    cmd([os.environ["BROWSER"], url])
diff --git a/setup.py b/setup.py
index b927174..d39afc4 100755
--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from distutils.core import setup
 
diff --git a/urlscan b/urlscan
index 28e8820..c8e4027 100755
--- a/urlscan
+++ b/urlscan
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # A simple urlview replacement that handles things like quoted-printable properly.
 # aka "urlview minus teh suck"
