diff -rud hydra-7.6/dpl4hydra.sh hydra-7.6-archlinux/dpl4hydra.sh
--- hydra-7.6/dpl4hydra.sh	2013-11-01 10:42:36.000000000 +0000
+++ hydra-7.6-archlinux/dpl4hydra.sh	2014-01-08 00:13:34.968156934 +0000
@@ -8,8 +8,7 @@
 # Credits:  Thanks to van Hauser for support and fixing portability issues.
 #           "The universe is an intelligence test." -Timothy Leary (R.I.P.)
 
-INSTALLDIR=/usr/local
-LOCATION=etc
+LOCATION=/etc/hydra
 
 usage ()
 {
@@ -156,7 +155,7 @@
 LC_ALL=C
 export LC_ALL
 DPLPATH="."
-test -r "$DPLPATH/dpl4hydra_full.csv" || DPLPATH="$INSTALLDIR/$LOCATION"
+test -r "$DPLPATH/dpl4hydra_full.csv" || DPLPATH="$LOCATION"
 FULLFILE="$DPLPATH/dpl4hydra_full.csv"
 OLDFILE="$DPLPATH/dpl4hydra_full.old"
 LOCALFILE="$DPLPATH/dpl4hydra_local.csv"
diff -rud hydra-7.6/Makefile.am hydra-7.6-archlinux/Makefile.am
--- hydra-7.6/Makefile.am	2013-12-25 07:06:44.000000000 +0000
+++ hydra-7.6-archlinux/Makefile.am	2014-01-08 00:13:57.501488889 +0000
@@ -60,7 +60,7 @@
 	-mkdir -p $(PREFIX)$(DIR)
 	cp -f hydra-wizard.sh $(BINS) $(PREFIX)$(DIR) && cd $(PREFIX)$(DIR) && chmod 755 hydra-wizard.sh $(BINS)
 	-echo OK > /dev/null && test -x xhydra && cp xhydra $(PREFIX)$(DIR) && cd $(PREFIX)$(DIR) && chmod 755 xhydra || echo OK > /dev/null
-	-sed -e "s|^INSTALLDIR=.*|INSTALLDIR="$(PREFIX)"|" dpl4hydra.sh > $(PREFIX)/bin/dpl4hydra.sh
+	-cp -f dpl4hydra.sh > $(PREFIX)/bin/dpl4hydra.sh
 	-chmod 755 $(PREFIX)/bin/dpl4hydra.sh
 	-mkdir -p $(PREFIX)/etc/
 	-cp -f *.csv $(PREFIX)/etc/
