diff -Naur yzis-1.0-alpha1.old/CMakeLists.txt yzis-1.0-alpha1/CMakeLists.txt
--- yzis-1.0-alpha1.old/CMakeLists.txt	2008-12-25 03:09:55.000000000 -0600
+++ yzis-1.0-alpha1/CMakeLists.txt	2008-12-25 03:09:58.000000000 -0600
@@ -141,7 +141,7 @@
 # define some variables
 #
 if (NOT APPLNK_INSTALL_DIR)
-	set(APPLNK_INSTALL_DIR share/applnk)
+	set(APPLNK_INSTALL_DIR share/applications)
 endif (NOT APPLNK_INSTALL_DIR)
 
 if (NOT ICON_INSTALL_DIR)
diff -Naur yzis-1.0-alpha1.old/libyzis/debug.cpp yzis-1.0-alpha1/libyzis/debug.cpp
--- yzis-1.0-alpha1.old/libyzis/debug.cpp	2008-12-25 03:09:55.000000000 -0600
+++ yzis-1.0-alpha1/libyzis/debug.cpp	2008-12-25 03:09:58.000000000 -0600
@@ -112,14 +112,17 @@
 #ifdef DEBUG
     _level = YZ_DEBUG_LEVEL;
 #else
-    _level = YZ_WARNING_LEVEL;
+    _level = YZ_FATAL_LEVEL;
 #endif
 
+    /*
 #ifndef YZIS_WIN32_GCC
     setDebugOutput( "/tmp/yzisdebug-" + QString(getpwuid(geteuid())->pw_name) + ".log" );
 #else
     setDebugOutput( "/tmp/yzisdebug.log" );
 #endif
+    */
+    setDebugOutput( "stderr" );
 
     // our message handler does not manage to display all messages. So,
     // it is better left off disabled at the moment. The last one gets lost
diff -Naur yzis-1.0-alpha1.old/qyzis/CMakeLists.txt yzis-1.0-alpha1/qyzis/CMakeLists.txt
--- yzis-1.0-alpha1.old/qyzis/CMakeLists.txt	2008-12-25 03:09:55.000000000 -0600
+++ yzis-1.0-alpha1/qyzis/CMakeLists.txt	2008-12-25 03:09:58.000000000 -0600
@@ -70,5 +70,5 @@
 target_link_libraries(qyzis ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} yzis )
 
 ########### install files ###############
-install( FILES qyzis.desktop DESTINATION ${APPLNK_INSTALL_DIR}/Editors )
+install( FILES qyzis.desktop DESTINATION ${APPLNK_INSTALL_DIR} )
 
diff -Naur yzis-1.0-alpha1.old/qyzis/pics/CMakeLists.txt yzis-1.0-alpha1/qyzis/pics/CMakeLists.txt
--- yzis-1.0-alpha1.old/qyzis/pics/CMakeLists.txt	2008-12-25 03:09:55.000000000 -0600
+++ yzis-1.0-alpha1/qyzis/pics/CMakeLists.txt	2008-12-25 03:09:58.000000000 -0600
@@ -5,6 +5,7 @@
 	foreach(dimension 16 32 64 128)
 		install(FILES cr${dimension}-app-qyzis.png DESTINATION ${ICON_INSTALL_DIR}/hicolor/${dimension}x${dimension}/apps RENAME qyzis.png)
 	endforeach(dimension)
+	install(FILES cr64-app-qyzis.png DESTINATION /usr/share/pixmaps RENAME qyzis.png)
 endif (KDE4_FOUND)
 
 if (APPLE)
diff -Naur yzis-1.0-alpha1.old/qyzis/qyzis.desktop yzis-1.0-alpha1/qyzis/qyzis.desktop
--- yzis-1.0-alpha1.old/qyzis/qyzis.desktop	2008-12-25 03:09:55.000000000 -0600
+++ yzis-1.0-alpha1/qyzis/qyzis.desktop	2008-12-25 03:10:23.000000000 -0600
@@ -1,11 +1,11 @@
 [Desktop Entry]
-Encoding=UTF-8
+Version=1.0
 Name=Qyzis
 Exec=qyzis %u
 Icon=qyzis
 Type=Application
 DocPath=qyzis/qyzis.html
-GenericName=Yzis Frontend for Qt
-Terminal=0
-Categories=Qt;TextEditor
+GenericName=A Qt interface to yzis
+Terminal=false
+Categories=Qt;Utility;TextEditor;
 MimeType=text/plain;text/x-makefile;text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-moc;text/x-pascal;text/x-tcl;text/x-tex;application/x-shellscript;text/rdf;
