diff -ur gmail-notify.orig/gmailatom.py gmail-notify/gmailatom.py
--- gmail-notify.orig/gmailatom.py	2012-08-25 18:21:14.461040561 +0100
+++ gmail-notify/gmailatom.py	2012-08-25 18:22:07.825199097 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 # gmailatom 0.0.1
diff -ur gmail-notify.orig/GmailConfig.py gmail-notify/GmailConfig.py
--- gmail-notify.orig/GmailConfig.py	2012-08-25 18:21:14.461040561 +0100
+++ gmail-notify/GmailConfig.py	2012-08-25 18:22:07.808532160 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 import pygtk
 pygtk.require('2.0')
@@ -9,6 +9,7 @@
 import xmllangs
 import sys
 
+sys.path[0] = "/usr/share/gmail-notify"
 LANGSXML_PATH=sys.path[0]+"/langs.xml"
 ICON_PATH=sys.path[0]+"/gmail-notify-icon.png"
 
diff -ur gmail-notify.orig/GmailPopupMenu.py gmail-notify/GmailPopupMenu.py
--- gmail-notify.orig/GmailPopupMenu.py	2012-08-25 18:21:14.461040561 +0100
+++ gmail-notify/GmailPopupMenu.py	2012-08-25 18:22:07.838532635 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 # Uploaded by juan_grande 2005/02/14 18:38 UTC
 import pygtk
diff -ur gmail-notify.orig/notifier.py gmail-notify/notifier.py
--- gmail-notify.orig/notifier.py	2012-08-25 18:21:14.461040561 +0100
+++ gmail-notify/notifier.py	2012-08-25 18:22:07.775198309 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 # Uploaded by juan_grande 2005/02/24 18:38 UTC
 import pygtk
@@ -15,6 +15,7 @@
 import GmailPopupMenu
 import gmailatom
 
+sys.path[0] = "/usr/share/gmail-notify"
 BKG_PATH=sys.path[0]+"/background.jpg"
 ICON_PATH=sys.path[0]+"/icon.png"
 ICON2_PATH=sys.path[0]+"/icon2.png"
diff -ur gmail-notify.orig/xmllangs.py gmail-notify/xmllangs.py
--- gmail-notify.orig/xmllangs.py	2012-08-25 18:21:14.461040561 +0100
+++ gmail-notify/xmllangs.py	2012-08-25 18:22:07.791865235 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 # xmllangs 0.0.1
