diff -Nur GotoBLAS2.orig/exports/Makefile GotoBLAS2/exports/Makefile
--- GotoBLAS2.orig/exports/Makefile	2011-10-20 19:26:38.865550738 +0200
+++ GotoBLAS2/exports/Makefile	2011-10-20 19:27:10.985349115 +0200
@@ -22,6 +22,12 @@
 endif
 endif
 
+ifeq ($(OSNAME), Linux)
+ifeq ($(F_COMPILER), GFORTRAN)
+EXTRALIB += -lgfortran
+endif
+endif
+
 all::
 
 libs::
diff -Nur GotoBLAS2.orig/f_check GotoBLAS2/f_check
--- GotoBLAS2.orig/f_check	2011-10-20 19:26:38.845550863 +0200
+++ GotoBLAS2/f_check	2011-10-20 19:27:31.595219262 +0200
@@ -274,6 +274,7 @@
 	    && ($flags !~ /kernel32/)
 	    && ($flags !~ /advapi32/)
 	    && ($flags !~ /shell32/)
+          && ($flags !~ /-l$/)
 	    ) {
 	    $linker_l .= $flags . " ";
 	}
