--- run-agent-echoey.sh	2010-04-03 15:20:48.000000000 +0200
+++ run-agent-echoey.sh	2010-04-03 15:22:24.000000000 +0200
@@ -3,6 +3,11 @@
 # This script will start the Echoey the FedOne wave agent.
 #
 
+SCRIPT_PATH=`readlink -f $0`
+SCRIPT_DIR=`dirname $SCRIPT_PATH`
+
+cd $SCRIPT_DIR
+
 if [ -f run-config.sh ] ; then
   . run-config.sh
 else
--- run-client-console.sh	2010-04-03 15:22:35.000000000 +0200
+++ run-client-console.sh	2010-04-03 15:22:58.000000000 +0200
@@ -3,6 +3,11 @@
 # This script will start the FedOne wave client.
 #
 
+SCRIPT_PATH=`readlink -f $0`
+SCRIPT_DIR=`dirname $SCRIPT_PATH`
+
+cd $SCRIPT_DIR
+
 if [ -f run-config.sh ] ; then
   . run-config.sh
 else
--- run-server.sh	2010-04-03 15:25:18.000000000 +0200
+++ run-server.sh	2010-04-03 15:25:30.000000000 +0200
@@ -3,6 +3,11 @@
 # This script will start the FedOne wave server.
 #
 
+SCRIPT_PATH=`readlink -f $0`
+SCRIPT_DIR=`dirname $SCRIPT_PATH`
+
+cd $SCRIPT_DIR
+
 if [ -f run-config.sh ] ; then
   . run-config.sh
 else
--- run-agent-probey.sh	2010-04-03 15:34:40.000000000 +0200
+++ run-agent-probey.sh	2010-04-03 15:35:05.000000000 +0200
@@ -3,6 +3,11 @@
 # This script will start the Echoey the FedOne wave agent.
 #
 
+SCRIPT_PATH=`readlink -f $0`
+SCRIPT_DIR=`dirname $SCRIPT_PATH`
+
+cd $SCRIPT_DIR
+
 if [ -f run-config.sh ] ; then
   . run-config.sh
 else
