diff --git a/compile.sh b/compile.sh
index 165a0a5..a2b0c77 100755
--- a/compile.sh
+++ b/compile.sh
@@ -1,5 +1,5 @@
 #!/bin/sh
-GAMBCLIBDIR="`gsc -e \"(display (path-expand \\\"~~lib\\\"))\"`"
-echo "bh.scm:" && gsc -link -l "$GAMBCLIBDIR/_gambcgsc" bh.scm && gsc -obj bh_.c bh.c && gsc -exe -ld-options "$GAMBCLIBDIR/libgambcgsc.a" bh_.o bh.o
+GAMBCLIBDIR="`gambitc -e \"(display (path-expand \\\"~~lib\\\"))\"`"
+echo "bh.scm:" && gambitc -link -l "$GAMBCLIBDIR/_gambcgsc" bh.scm && gambitc -obj bh_.c bh.c && gambitc -exe -ld-options "$GAMBCLIBDIR/libgambcgsc.a" bh_.o bh.o
 rm -f bh_.o bh_.c bh.c bh.o
 
diff --git a/src/blackhole.scm b/src/blackhole.scm
index 88897b8..c819b1a 100755
--- a/src/blackhole.scm
+++ b/src/blackhole.scm
@@ -1,6 +1,6 @@
 (##namespace ("bh#"))
 
-(##include "~~/lib/gambit#.scm")
+(##include "~~/lib/gambit-c/gambit#.scm")
 
 (declare (standard-bindings)
 	 (extended-bindings)
diff --git a/src/compile-load.scm b/src/compile-load.scm
index 1fdd6cc..979a4ce 100644
--- a/src/compile-load.scm
+++ b/src/compile-load.scm
@@ -227,7 +227,7 @@
           (set! prev-hook c#expand-source)
           (set! c#expand-source hook))
         (lambda ()
-          (compile-file-to-c "/dev/null"
+          (compile-file-to-target "/dev/null"
                              output: fn
                              options: options))
         (lambda ()
