diff --git a/configure.ac b/configure.ac
index 324ada6..480dd12 100644
--- a/configure.ac
+++ b/configure.ac
@@ -15,13 +15,13 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [The gettext package])
 AM_GLIB_GNU_GETTEXT
 IT_PROG_INTLTOOL([0.35.0])
 
-AM_PATH_PYTHON(2.6)
+AM_PATH_PYTHON(2.7, 2.6)
 AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(Could not find Python headers)])
 
 AC_ENABLE_SHARED(yes)
 AC_ENABLE_STATIC(no)
 
-PYTHON_LIBS=`python-config --libs`
+PYTHON_LIBS=`${PYTHON}-config --libs`
 AC_SUBST(PYTHON_LIBS)
 
 AC_ARG_VAR([PYREXC], [Pyrex compiler])
