This patch is required for docutils <= 0.8

--- nautilus-dropbox-0.6.9/Makefile.in	2011-08-12 19:39:47.000000000 -0300
+++ nautilus-dropbox-0.6.9-new/Makefile.in	2011-08-18 13:21:01.705943759 -0300
@@ -846,7 +846,7 @@
 
 dropbox.1: dropbox.txt.in dropbox docgen.py
 	python docgen.py $(PACKAGE_VERSION) < dropbox.txt.in > dropbox.txt
-	python rst2man.py dropbox.txt > dropbox.1
+	rst2man.py dropbox.txt dropbox.1
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
