diff -aur qgo2/src/board/boardhandler.cpp qgo2-build/src/board/boardhandler.cpp
--- qgo2/src/board/boardhandler.cpp	2009-09-04 21:58:02.000000000 +0200
+++ qgo2-build/src/board/boardhandler.cpp	2010-10-03 22:10:37.460009490 +0200
@@ -672,22 +672,22 @@
 			case markKoMarker:
 			case markSquare:
 				modified = true;
-				board->setMark(x, y, markSquare, false);
+				board->setMark(x, y, markSquare, false, QString::null);
 				break;
 				
 			case markCircle:
 				modified = true;
-				board->setMark(x, y, markCircle, false);
+				board->setMark(x, y, markCircle, false, QString::null);
 				break;
 				
 			case markTriangle:
 				modified = true;
-				board->setMark(x, y, markTriangle, false);
+				board->setMark(x, y, markTriangle, false, QString::null);
 				break;
 				
 			case markCross:
 				modified = true;
-				board->setMark(x, y, markCross, false);
+				board->setMark(x, y, markCross, false, QString::null);
 				break;
 				
 			case markText:
@@ -702,14 +702,14 @@
 				
 			case markTerrBlack:
 				modified = true;
-				board->setMark(x, y, markTerrBlack, false);
+				board->setMark(x, y, markTerrBlack, false, QString::null);
 				if(m->getStoneAt(x,y) == stoneWhite)	//awkward but may have been ghosted
 					m->invalidateStone(x, y);
 				break;
 				
 			case markTerrWhite:
 				modified = true;
-				board->setMark(x, y, markTerrWhite, false);
+				board->setMark(x, y, markTerrWhite, false, QString::null);
 				if(m->getStoneAt(x,y) == stoneBlack)	//awkward but may have been ghosted
 					m->invalidateStone(x, y);
 				break;
diff -aur qgo2/src/board/boardwindow.cpp qgo2-build/src/board/boardwindow.cpp
--- qgo2/src/board/boardwindow.cpp	2010-07-26 17:03:41.000000000 +0200
+++ qgo2-build/src/board/boardwindow.cpp	2010-10-03 22:12:17.686676158 +0200
@@ -84,7 +84,7 @@
 	
 	//Loads the sgf file if any
 	if (! gameData->fileName.isEmpty())
-		loadSGF(gameData->fileName);
+		loadSGF(gameData->fileName, QString::null);
 
 	//creates the board interface (or proxy) that will handle the moves an command requests
 	switch (gameData->gameMode)
diff -aur qgo2/src/network/playergamelistings.h qgo2-build/src/network/playergamelistings.h
--- qgo2/src/network/playergamelistings.h	2010-02-23 21:59:45.000000000 +0100
+++ qgo2-build/src/network/playergamelistings.h	2010-10-03 22:05:11.233342825 +0200
@@ -66,21 +66,21 @@
 	bool hidden;
 	PlayerListing() : id(0), 
 	online(0),
-	name(0),
-	notnickname(0),
-	info(0),
-	idletime(0),
+	name(QString::null),
+	notnickname(QString::null),
+	info(QString::null),
+	idletime(QString::null),
 	seconds_idle(0),
-	rank(0),
+	rank(QString::null),
 	rank_score(0),
-	country(0),
+	country(QString::null),
 	wins(0), 
 	losses(0),
 	rated_games(0),
 	observing(0),
 	playing(0), 
-	extInfo(0),
-	email_address(0),
+	extInfo(QString::null),
+	email_address(QString::null),
 	nmatch(0),
 	nmatch_handicapMin(0),
 	specialbyte(0),
@@ -189,7 +189,7 @@
 	_white_rank_score(0), 
 	_black_rank_score(0), 
 	observers(0), 
-	result(0),
+	result(QString::null),
 	flags(IN_PROGRESS),
 	rated(0), 
 	owner_id(0), 
diff -aur qgo2/src/sgf/sgfparser.cpp qgo2-build/src/sgf/sgfparser.cpp
--- qgo2/src/sgf/sgfparser.cpp	2010-07-26 17:05:12.000000000 +0200
+++ qgo2-build/src/sgf/sgfparser.cpp	2010-10-03 22:19:00.150009490 +0200
@@ -262,7 +262,7 @@
 	}
 	*/
 	if (!initGame(toParse, fileName))
-		return corruptSgf();
+		return corruptSgf(0, QString::null);
 	return doParse(toParse);
 }
 /*
@@ -538,7 +538,7 @@
 			if (pos != posVarBegin)
 			{
 				delete toParse;
-				return corruptSgf(pos);
+				return corruptSgf(pos, QString::null);
 			}
 			
 			// qDebug("Var BEGIN at %d, moves = %d", pos, moves);
@@ -554,7 +554,7 @@
 			if (pos != posVarEnd)
 			{
 				delete toParse;
-				return corruptSgf(pos);
+				return corruptSgf(pos, QString::null);
 			}
 			
 			// qDebug("VAR END");
@@ -589,7 +589,7 @@
 			if (pos != posNode)
 			{
 				delete toParse;
-				return corruptSgf(pos);
+				return corruptSgf(pos, QString::null);
 			}
 			
 			// qDebug("Node at %d", pos);
@@ -794,7 +794,7 @@
 					if (toParse->at(pos) != '[' && prop != unknownProp)
 					{
 						delete toParse;
-						return corruptSgf(pos);
+						return corruptSgf(pos, QString::null);
 					}
 					
 					// Empty type
@@ -1114,7 +1114,7 @@
 								if (toParse->at(pos) != ':')
 								{
 									delete toParse;
-									return corruptSgf(pos);
+									return corruptSgf(pos, QString::null);
 								}
 								moveStr = "";
 								while (toParse->at(++pos) != ']' &&
@@ -1268,7 +1268,7 @@
 	
 	default:
 		delete toParse;
-		return corruptSgf(pointer);
+		return corruptSgf(pointer, QString::null);
 	}
 	
 	} while (pointer < strLength && pos >= 0);
@@ -1324,11 +1324,11 @@
 	
 	pos += 2;
 	if (toParse[pos] != '[')
-		return  corruptSgf(pos);
+		return  corruptSgf(pos, QString::null);
 	while (toParse[++pos] != ']' && pos < strLength)
 		result.append(toParse[pos]);
 	if (pos > strLength)
-		return  corruptSgf(pos);
+		return  corruptSgf(pos, QString::null);
 	
 	return true;
 }
