diff -aur kde-runtime-4.9.5/kioslave/smb/kio_smb.h kde-runtime-4.9.5.fix/kioslave/smb/kio_smb.h
--- kde-runtime-4.9.5/kioslave/smb/kio_smb.h	2012-11-30 00:26:48.000000000 -0700
+++ kde-runtime-4.9.5.fix/kioslave/smb/kio_smb.h	2013-01-07 03:36:39.388823590 -0700
@@ -75,7 +75,7 @@
 //---------------------------
 #include "kio_smb_internal.h"
 
-#define MAX_XFER_BUF_SIZE           16348
+#define MAX_XFER_BUF_SIZE           262144
 #define KIO_SMB                     7106
 
 using namespace KIO;
