--- liferea-1.6.6b/configure.ac.orig	2011-11-14 12:47:52.000000000 +0100
+++ liferea-1.6.6b/configure.ac	2011-11-14 12:53:40.288925846 +0100
@@ -79,7 +79,7 @@
 dnl *********
 
 if test "x$enable_libnotify" = "xyes"; then
-   PKG_CHECK_MODULES([LIBNOTIFY], [libnotify >= 0.3.2, libnotify <= 0.5],enable_libnotify=yes,enable_libnotify=no)
+   PKG_CHECK_MODULES([LIBNOTIFY], libnotify >= 0.7.0,enable_libnotify=yes,enable_libnotify=no)
    AC_SUBST(LIBNOTIFY_CFLAGS)
    AC_SUBST(LIBNOTIFY_LIBS)
 else
