=== modified file 'src/Makefile.am'
Index: fix-for-bug-732848/src/Makefile.am
===================================================================
--- fix-for-bug-732848.orig/src/Makefile.am	2011-03-12 12:50:17.735382000 +0100
+++ fix-for-bug-732848/src/Makefile.am	2011-03-12 12:50:30.401905830 +0100
@@ -97,3 +97,9 @@
 liferea_LDADD += $(LIBINDICATE_LIBS)
 
 endif
+
+if WITH_LIBUNITY
+
+liferea_LDADD += $(LIBUNITY_LIBS)
+
+endif
Index: fix-for-bug-732848/src/ui/Makefile.am
===================================================================
--- fix-for-bug-732848.orig/src/ui/Makefile.am	2011-03-12 12:50:17.735382000 +0100
+++ fix-for-bug-732848/src/ui/Makefile.am	2011-03-12 12:50:30.401905830 +0100
@@ -8,7 +8,7 @@
 
 noinst_LIBRARIES = libliui.a
 
-libliui_a_CFLAGS = $(PACKAGE_CFLAGS) $(DBUS_CFLAGS)	$(LIBINDICATE_CFLAGS)
+libliui_a_CFLAGS = $(PACKAGE_CFLAGS) $(DBUS_CFLAGS)	$(LIBINDICATE_CFLAGS) $(LIBUNITY_CFLAGS)
 libliui_a_SOURCES = \
 	auth_dialog.c auth_dialog.h \
 	browser_tabs.c browser_tabs.h \
@@ -35,4 +35,5 @@
 	ui_session.c ui_session.h \
 	ui_subscription.c ui_subscription.h \
 	ui_tray.c ui_tray.h \
+	ui_unity.c ui_unity.h \
 	ui_update.c ui_update.h 
Index: fix-for-bug-732848/src/ui/ui_tray.c
===================================================================
--- fix-for-bug-732848.orig/src/ui/ui_tray.c	2011-03-12 12:50:17.735382000 +0100
+++ fix-for-bug-732848/src/ui/ui_tray.c	2011-03-12 12:50:30.405905830 +0100
@@ -188,6 +188,7 @@
 	gchar	*msg, *tmp;
 
 	ui_indicator_update ();
+	ui_unity_update();
 
 	if (!trayIcon_priv)
 		return;
Index: fix-for-bug-732848/src/ui/ui_unity.c
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ fix-for-bug-732848/src/ui/ui_unity.c	2011-03-12 12:50:52.949905875 +0100
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2011 Jean-Philippe Orsini <jeanfi@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor Boston, MA 02110-1301,  USA
+ */
+
+#include "ui_unity.h"
+
+#ifdef HAVE_LIBUNITY
+
+#include <feedlist.h>
+
+#include <unity.h>
+
+static UnityLauncherEntry *entry;
+
+void ui_unity_update (void)
+{
+	int count;
+
+	if (!entry)
+		entry = unity_launcher_entry_get_for_desktop_file
+			("liferea.desktop");
+
+	if (entry) {
+		count = feedlist_get_unread_item_count ();
+
+		if (count) {
+			unity_launcher_entry_set_count (entry, count);
+			unity_launcher_entry_set_count_visible (entry, TRUE);
+		} else {
+			unity_launcher_entry_set_count_visible (entry, FALSE);
+		}
+	}
+}
+
+void ui_unity_destroy (void)
+{
+	if (entry) {
+		unity_launcher_entry_set_count_visible (entry, FALSE);
+		g_main_context_iteration(NULL, FALSE);
+	}
+}
+
+#else
+
+void ui_unity_update (void) {}
+void ui_unity_destroy (void) {}
+
+#endif  /* HAVE_LIBUNITY */
Index: fix-for-bug-732848/src/ui/ui_unity.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ fix-for-bug-732848/src/ui/ui_unity.h	2011-03-12 12:50:30.405905830 +0100
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2011 Jean-Philippe Orsini <jeanfi@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor Boston, MA 02110-1301,  USA
+ */
+
+#ifndef _UI_UNITY_H
+#define _UI_UNITY_H
+
+#ifdef HAVE_CONFIG_H
+	#include <config.h>
+#endif
+
+void ui_unity_update ();
+void ui_unity_destroy ();
+
+#endif  /* _UI_UNITY_H */
Index: fix-for-bug-732848/configure.ac
===================================================================
--- fix-for-bug-732848.orig/configure.ac	2011-03-12 12:50:17.735382000 +0100
+++ fix-for-bug-732848/configure.ac	2011-03-12 12:50:30.405905830 +0100
@@ -29,6 +29,7 @@
 AC_ARG_ENABLE(nm,        AS_HELP_STRING([--disable-nm],[compile without NetworkManager support]),,enable_nm=yes)
 AC_ARG_ENABLE(libnotify, AS_HELP_STRING([--disable-libnotify],[don't compile the libnotify plugin]),,enable_libnotify=yes)
 AC_ARG_ENABLE(libindicate, AS_HELP_STRING([--disable-libindicate],[compile without libindicate support]),,enable_libindicate=yes)
+AC_ARG_ENABLE(libunity, AS_HELP_STRING([--disable-libunity],[compile without libunity support]),,enable_libunity=yes)
 AC_ARG_ENABLE(lua,       AS_HELP_STRING([--disable-lua],[don't compile with LUA scripting support]),,enable_lua=yes)
 dnl AC_ARG_ENABLE(avahi,     AS_HELP_STRING([--disable-avahi],[don't compile with AVAHI support]),,enable_avahi=yes)
 
@@ -119,6 +120,24 @@
 
 AM_CONDITIONAL(WITH_LIBINDICATE, test "x$enable_libindicate" = "xyes")
 
+dnl *********
+dnl libunity
+dnl *********
+
+if test "x$enable_libunity" = "xyes"; then
+   PKG_CHECK_MODULES([LIBUNITY], unity >= 3.4.2,enable_libunity=yes,enable_libunity=no)
+   AC_SUBST(LIBUNITY_CFLAGS)
+   AC_SUBST(LIBUNITY_LIBS)
+else
+   enable_libunity=no
+fi
+
+if test "x$enable_libunity" = "xyes"; then
+  AC_DEFINE(HAVE_LIBUNITY, 1, [Define if libunity support is enabled])
+fi
+
+AM_CONDITIONAL(WITH_LIBUNITY, test "x$enable_libunity" = "xyes")
+
 dnl *****
 dnl AVAHI
 dnl *****
@@ -260,6 +279,7 @@
 echo "Use DBUS........................ : $enable_dbus"
 echo "Use NetworkManager.............. : $enable_nm"
 echo "Use libindicate................. : $enable_libindicate"
+echo "Use libunity.................... : $enable_libunity"
 dnl echo "AVAHI Support................... : $enable_avahi"
 echo
 eval eval echo Liferea will be installed in $bindir.
Index: fix-for-bug-732848/src/ui/liferea_shell.c
===================================================================
--- fix-for-bug-732848.orig/src/ui/liferea_shell.c	2011-03-12 12:50:17.735382000 +0100
+++ fix-for-bug-732848/src/ui/liferea_shell.c	2011-03-12 12:50:30.409905830 +0100
@@ -1333,6 +1333,7 @@
 	ui_feedlist_select (NULL);
 	liferea_shell_save_position ();
 	ui_indicator_destroy ();
+	ui_unity_destroy ();
 	ui_tray_enable (FALSE);
 	notification_enable (FALSE);
 	g_object_unref (shell->priv->tabs);
