diff --git a/src/mount_image.c b/src/mount_image.c
index a05a822..9e4824a 100644
--- a/src/mount_image.c
+++ b/src/mount_image.c
@@ -1404,7 +1404,7 @@ static int execute_fusermount(const char *dir)
 	}
 	if (pid == 0) {
 		/* Child */
-		execlp("fusermount", "fusermount", "-u", dir, NULL);
+		execlp("fusermount", "fusermount", "-u", "-z", dir, NULL);
 		ERROR_WITH_ERRNO("Failed to execute `fusermount'");
 		exit(WIMLIB_ERR_FUSERMOUNT);
 	}
