--- src/opencv-2.4.10/unix-install/opencv.pc	2014-10-18 22:03:52.783919873 +0200
+++ opencv.pc	2014-10-19 16:43:08.775103845 +0200
@@ -3,11 +3,12 @@
 prefix=/usr
 exec_prefix=${prefix}
 libdir=${prefix}/lib
+cudalib=/opt/cuda/lib
 includedir_old=${prefix}/include/opencv
 includedir_new=${prefix}/include
 
 Name: OpenCV
 Description: Open Source Computer Vision Library
 Version: 2.4.10
-Libs: -L${libdir} -lopencv_calib3d -lopencv_contrib -lopencv_core -lopencv_features2d -lopencv_flann -lopencv_gpu -lopencv_highgui -lopencv_imgproc -lopencv_legacy -lopencv_ml -lopencv_nonfree -lopencv_objdetect -lopencv_ocl -lopencv_photo -lopencv_stitching -lopencv_superres -lopencv_ts -lopencv_video -lopencv_videostab -lcufft -lnpps -lnppi -lnppc -lcudart -ltbb /lib64/libXext.so /lib64/libX11.so /lib64/libICE.so /lib64/libSM.so /lib64/libGL.so /lib64/libGLU.so -lrt -lpthread -lm -ldl
+Libs: -L${libdir} -L${cudalib} -lopencv_calib3d -lopencv_contrib -lopencv_core -lopencv_features2d -lopencv_flann -lopencv_gpu -lopencv_highgui -lopencv_imgproc -lopencv_legacy -lopencv_ml -lopencv_nonfree -lopencv_objdetect -lopencv_ocl -lopencv_photo -lopencv_stitching -lopencv_superres -lopencv_ts -lopencv_video -lopencv_videostab -lcufft -lnpps -lnppi -lnppc -lcudart -ltbb /lib64/libXext.so /lib64/libX11.so /lib64/libICE.so /lib64/libSM.so /lib64/libGL.so /lib64/libGLU.so -lrt -lpthread -lm -ldl
 Cflags: -I${includedir_old} -I${includedir_new}
