diff -Nur linux-2.6.37.orig/arch/x86/kernel/cpu/transmeta.c linux-2.6.37/arch/x86/kernel/cpu/transmeta.c
--- linux-2.6.37.orig/arch/x86/kernel/cpu/transmeta.c	2011-01-04 19:50:19.000000000 -0500
+++ linux-2.6.37/arch/x86/kernel/cpu/transmeta.c	2011-03-23 09:51:32.000000000 -0400
@@ -96,6 +96,7 @@
 	 */
 	randomize_va_space = 0;
 #endif
+boot_cpu_data.x86 = 6;
 }
 
 static const struct cpu_dev __cpuinitconst transmeta_cpu_dev = {
diff -Nur linux-2.6.37.orig/arch/x86/kernel/entry_32.S linux-2.6.37/arch/x86/kernel/entry_32.S
--- linux-2.6.37.orig/arch/x86/kernel/entry_32.S	2011-01-04 19:50:19.000000000 -0500
+++ linux-2.6.37/arch/x86/kernel/entry_32.S	2011-03-23 09:51:32.000000000 -0400
@@ -931,7 +931,7 @@
 ENTRY(invalid_op)
 	RING0_INT_FRAME
 	pushl_cfi $0
-	pushl_cfi $do_invalid_op
+	pushl_cfi $do_nopl_emu
 	jmp error_code
 	CFI_ENDPROC
 END(invalid_op)
diff -Nur linux-2.6.37.orig/arch/x86/kernel/Makefile linux-2.6.37/arch/x86/kernel/Makefile
--- linux-2.6.37.orig/arch/x86/kernel/Makefile	2011-01-04 19:50:19.000000000 -0500
+++ linux-2.6.37/arch/x86/kernel/Makefile	2011-03-23 09:51:32.000000000 -0400
@@ -91,6 +91,7 @@
 obj-$(CONFIG_APB_TIMER)		+= apb_timer.o
 
 obj-$(CONFIG_AMD_NB)		+= amd_nb.o
+obj-$(CONFIG_MCRUSOE)		+= nopl_emu.o
 obj-$(CONFIG_DEBUG_RODATA_TEST)	+= test_rodata.o
 obj-$(CONFIG_DEBUG_NX_TEST)	+= test_nx.o
 
diff -Nur linux-2.6.37.orig/arch/x86/kernel/nopl_emu.c linux-2.6.37/arch/x86/kernel/nopl_emu.c
--- linux-2.6.37.orig/arch/x86/kernel/nopl_emu.c	1969-12-31 19:00:00.000000000 -0500
+++ linux-2.6.37/arch/x86/kernel/nopl_emu.c	2011-03-23 09:51:32.000000000 -0400
@@ -0,0 +1,111 @@
+/*
+ *  linux/arch/x86/kernel/nopl_emu.c
+ *
+ *  Copyright (C) 2002  Willy Tarreau
+ *  Copyright (C) 2010  Matteo Croce
+ */
+
+#include <asm/processor.h>
+#include <asm/traps.h>
+
+/* 
+ *  This code can be used to allow the AMD Geode to hopefully correctly execute
+ *  some code which was originally compiled for an i686, by emulating NOPL,
+ *  the only missing i686 instruction in the CPU
+ *
+ *  Willy Tarreau <willy@meta-x.org>
+ *  Matteo Croce <technoboy85@gmail.com>
+ */
+
+static inline int do_1f(u8 *ip)
+{
+       int length = 3;
+       switch (*ip) {
+       case 0x84:
+               if (!ip[5])
+                       length++;
+               else
+                       return 0;
+       case 0x80:
+               if (!ip[4] && !ip[3])
+                       length += 2;
+               else
+                       return 0;
+       case 0x44:
+               if (!ip[2])
+                       length++;
+               else
+                       return 0;
+       case 0x40:
+               if (!ip[1])
+                       length++;
+               else
+                       return 0;
+       case 0x00:
+               return length;
+       }
+       return 0;
+}
+
+static inline int do_0f(u8 *ip)
+{
+       if (*ip == 0x1f)
+               return do_1f(ip + 1);
+       return 0;
+}
+
+static inline int do_66(u8 *ip)
+{
+       if (*ip == 0x90)
+               return 2;
+       if (*ip == 0x0f) {
+               int res = do_0f(ip + 1);
+               if (res)
+                       return res + 1;
+               else
+                       return 0;
+       }
+       return 0;
+}
+
+static inline int do_start(u8 *ip)
+{
+       if (*ip == 0x0f)
+               return do_0f(ip + 1);
+       if (*ip == 0x66)
+               return do_66(ip + 1);
+       return 0;
+}
+
+/* [do_nopl_emu] is called by exception 6 after an invalid opcode has been
+ * encountered. It will try to emulate it by doing nothing,
+ * and will send a SIGILL or SIGSEGV to the process if not possible.
+ * the NOPL can have variable length opcodes:
+
+bytes number   opcode
+       2       66 90
+       3       0f 1f 00
+       4       0f 1f 40 00
+       5       0f 1f 44 00 00
+       6       66 0f 1f 44 00 00
+       7       0f 1f 80 00 00 00 00
+       8       0f 1f 84 00 00 00 00 00
+       9       66 0f 1f 84 00 00 00 00 00
+*/
+
+void do_nopl_emu(struct pt_regs *regs, long error_code)
+{
+       u8 *ip = (u8 *)instruction_pointer(regs);
+       int res = do_start(ip);
+
+       if (res) {
+               int i = 0;
+               do {
+                       ip += res;
+                       i++;
+                       res = do_start(ip);
+               } while(res);
+               regs->ip = (typeof(regs->ip))ip;
+       } else
+               do_invalid_op(regs, error_code);
+}
diff -Nur linux-2.6.37.orig/arch/x86/Makefile_32.cpu linux-2.6.37/arch/x86/Makefile_32.cpu
--- linux-2.6.37.orig/arch/x86/Makefile_32.cpu	2011-01-04 19:50:19.000000000 -0500
+++ linux-2.6.37/arch/x86/Makefile_32.cpu	2011-03-23 09:51:32.000000000 -0400
@@ -25,7 +25,7 @@
 # They make zero difference whatsosever to performance at this time.
 cflags-$(CONFIG_MK7)		+= -march=athlon
 cflags-$(CONFIG_MK8)		+= $(call cc-option,-march=k8,-march=athlon)
-cflags-$(CONFIG_MCRUSOE)	+= -march=i686 $(align)-functions=0 $(align)-jumps=0 $(align)-loops=0
+cflags-$(CONFIG_MCRUSOE)	+= -march=i686
 cflags-$(CONFIG_MEFFICEON)	+= -march=i686 $(call tune,pentium3) $(align)-functions=0 $(align)-jumps=0 $(align)-loops=0
 cflags-$(CONFIG_MWINCHIPC6)	+= $(call cc-option,-march=winchip-c6,-march=i586)
 cflags-$(CONFIG_MWINCHIP3D)	+= $(call cc-option,-march=winchip2,-march=i586)
diff -Nur linux-2.6.37.orig/Makefile linux-2.6.37/Makefile
--- linux-2.6.37.orig/Makefile	2011-01-04 19:50:19.000000000 -0500
+++ linux-2.6.37/Makefile	2011-03-23 09:52:15.000000000 -0400
@@ -161,11 +161,12 @@
 # then ARCH is assigned, getting whatever value it gets normally, and 
 # SUBARCH is subsequently ignored.
 
-SUBARCH := $(shell uname -m | sed -e s/i.86/i386/ -e s/sun4u/sparc64/ \
-				  -e s/arm.*/arm/ -e s/sa110/arm/ \
-				  -e s/s390x/s390/ -e s/parisc64/parisc/ \
-				  -e s/ppc.*/powerpc/ -e s/mips.*/mips/ \
-				  -e s/sh[234].*/sh/ )
+#SUBARCH := $(shell uname -m | sed -e s/i.86/i386/ -e s/sun4u/sparc64/ \
+#				  -e s/arm.*/arm/ -e s/sa110/arm/ \
+#				  -e s/s390x/s390/ -e s/parisc64/parisc/ \
+#				  -e s/ppc.*/powerpc/ -e s/mips.*/mips/ \
+#				  -e s/sh[234].*/sh/ )
+SUBARCH := x86
 
 # Cross compiling and selecting different set of gcc/bin-utils
 # ---------------------------------------------------------------------------
