diff --git a/bochs.h b/bochs.h
index c2d6c6b..1deddeb 100644
--- a/bochs.h
+++ b/bochs.h
@@ -452,6 +452,8 @@ BOCHSAPI extern Bit32u apic_id_mask;
 extern bx_bool bx_gui_sighandler;
 #endif
 
+extern int jitter;
+
 // This value controls how often each I/O device's timer handler
 // gets called.  The timer is set up in iodev/devices.cc.
 #define BX_IODEV_HANDLER_PERIOD 1000   // microseconds
diff --git a/iodev/pit82c54.cc b/iodev/pit82c54.cc
index 09dcd8e..31da42d 100644
--- a/iodev/pit82c54.cc
+++ b/iodev/pit82c54.cc
@@ -49,6 +49,7 @@
 
 #include "iodev.h"
 #include "pit82c54.h"
+#include <stdint.h>
 #define LOG_THIS this->
 
 
@@ -410,7 +411,13 @@ void BX_CPP_AttrRegparmN(1) pit_82C54::clock(Bit8u cnum)
       case 2:
         if (thisctr.count_written) {
           if (thisctr.triggerGATE || thisctr.first_pass) {
-            set_count(thisctr, thisctr.inlatch);
+            unsigned n = thisctr.inlatch;
+            if (jitter && n > 5) {
+                n *= (double) rand() / RAND_MAX;
+                if (n < 5)
+                    n = 5;
+            }
+            set_count(thisctr, n);
             thisctr.next_change_time=(thisctr.count_binary-1) & 0xFFFF;
             thisctr.null_count=0;
             if (thisctr.inlatch==1) {
diff --git a/main.cc b/main.cc
index 0f11e31..a81798f 100644
--- a/main.cc
+++ b/main.cc
@@ -101,6 +101,7 @@ BOCHSAPI BX_CPU_C bx_cpu;
 BOCHSAPI BX_MEM_C bx_mem;
 
 char *bochsrc_filename = NULL;
+int jitter = 0;
 
 void bx_print_header()
 {
@@ -629,6 +630,16 @@ int bx_init_main(int argc, char *argv[])
     else if (!strcmp("-q", argv[arg])) {
       SIM->get_param_enum(BXPN_BOCHS_START)->set(BX_QUICK_START);
     }
+    else if (!strcmp("-j", argv[arg])) {
+        if (++arg >= argc) BX_PANIC(("-j must be followed by a number"));
+        else {
+            jitter = 1;
+            char *endptr = NULL;
+            long seed = strtol(argv[arg], &endptr, 10);
+            if (endptr != '\0') BX_PANIC(("-j must be followed by a number"));
+            srand(seed);            
+        }
+    }
     else if (!strcmp("-log", argv[arg])) {
       if (++arg >= argc) BX_PANIC(("-log must be followed by a filename"));
       else SIM->get_param_string(BXPN_LOG_FILENAME)->set(argv[arg]);
