From 9da205cd69693596a7871975bdb0f6da12b955ab Mon Sep 17 00:00:00 2001
From: Alfredo Palhares <masterkorp@masterkorp.net>
Date: Thu, 9 May 2013 02:52:18 +0100
Subject: [PATCH] Do not treat warnings as errors.

---
 core/core.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/core.pro b/core/core.pro
index c6c8671..2b0a578 100644
--- a/core/core.pro
+++ b/core/core.pro
@@ -31,7 +31,7 @@ win32-g++ {
         DEFINES += GIT_HASH=$$system(\"c:/program files/git/bin/git.exe\" log -n 1 --pretty=format:%H)
 }
 unix {
-	QMAKE_CXXFLAGS += -Werror
+	#QMAKE_CXXFLAGS += -Werror
 	target.path = /usr/bin
 	dashboard.path = /usr/share/emstudio/dashboards
 	dashboard.files += src/gauges.qml
-- 
1.8.2.2

