--- aoa360.c.orig	2012-02-16 19:48:33.904966741 +0600
+++ aoa360.c	2012-02-16 19:48:40.951633445 +0600
@@ -58,7 +58,7 @@
 void init_aoa_data()
 {
 
- read_file("naca0012.dat", naca0012_cl, naca0012_cd, naca0012_cm);
+ read_file("/usr/share/vertigo/naca0012.dat", naca0012_cl, naca0012_cd, naca0012_cm);
 
 }
 
--- astro.c.orig	2012-02-16 19:48:29.211633379 +0600
+++ astro.c	2012-02-16 19:48:55.394966860 +0600
@@ -29,7 +29,7 @@
  FILE *fp;
  int i;
 
- if ((fp=fopen("ybstars.dat","r")) == NULL)
+ if ((fp=fopen("/usr/share/vertigo/ybstars.dat","r")) == NULL)
         errormessage("Cannot open file: ybstars.dat\n\r");
 
  if(!feof(fp))
--- config.c.orig	2012-02-16 19:57:25.374969736 +0600
+++ config.c	2012-02-16 20:01:48.644971232 +0600
@@ -93,7 +93,11 @@
 {
  FILE *fp;
 
- if((fp=fopen(cnf_file,"wb"))==NULL)
+ char cnf_path[40];
+ strcpy(cnf_path, getenv("HOME"));
+ strcat(cnf_path, "/.vertigo.cfg");
+
+ if((fp=fopen(cnf_path,"wb"))==NULL)
    errormessage("Cannot create configuration file");
  fwrite(&stick_type        ,sizeof(short),1,fp);
  fwrite(&throttle_type     ,sizeof(short),1,fp);
@@ -156,7 +161,11 @@
 {
  FILE *fp;
 
- if(!(fp=fopen(cnf_file,"rb"))) {
+ char cnf_path[40];
+ strcpy(cnf_path, getenv("HOME"));
+ strcat(cnf_path, "/.vertigo.cfg");
+
+ if(!(fp=fopen(cnf_path,"rb"))) {
           errormessage("Cannot open configuration file.");
        }
  if(fread(&stick_type,sizeof(short),1,fp)!=1)
--- sound.c.orig	2012-02-16 19:42:26.788298002 +0600
+++ sound.c	2012-02-16 19:43:16.441631611 +0600
@@ -35,69 +35,69 @@
 
  digicard =  1;
  n_sounds = 62;
- strcpy(filenames[ 0],"sound/cannon.wav");
- strcpy(filenames[ 1],"sound/screetch.wav");
- strcpy(filenames[ 2],"sound/gear_up.wav");
- strcpy(filenames[ 3],"sound/prop.wav");
- strcpy(filenames[ 4],"sound/jet_ext.wav");
- strcpy(filenames[ 5],"sound/crash.wav");
- strcpy(filenames[ 6],"sound/flaps.wav");
- strcpy(filenames[ 7],"sound/talk/call_bal.wav");
- strcpy(filenames[ 8],"sound/talk/vir_ball.wav");
- strcpy(filenames[ 9],"sound/talk/rog_ball.wav");
- strcpy(filenames[10],"sound/talk/lit_low.wav");
- strcpy(filenames[11],"sound/talk/low.wav");
- strcpy(filenames[12],"sound/talk/lit_high.wav");
- strcpy(filenames[13],"sound/talk/high.wav");
- strcpy(filenames[14],"sound/talk/lit_slow.wav");
- strcpy(filenames[15],"sound/talk/slow.wav");
- strcpy(filenames[16],"sound/talk/lit_fast.wav");
- strcpy(filenames[17],"sound/talk/fast.wav");
- strcpy(filenames[18],"sound/talk/lit_righ.wav");
- strcpy(filenames[19],"sound/talk/lin_righ.wav");
- strcpy(filenames[20],"sound/talk/lit_left.wav");
- strcpy(filenames[21],"sound/talk/lin_left.wav");
- strcpy(filenames[22],"sound/talk/eagle2.wav");
- strcpy(filenames[22],"sound/talk/wind_axi.wav");
- strcpy(filenames[23],"sound/talk/power.wav");
- strcpy(filenames[24],"sound/talk/waveoff.wav");
- strcpy(filenames[25],"sound/talk/bolter.wav");
- strcpy(filenames[26],"sound/talk/low_gear.wav");
- strcpy(filenames[27],"sound/talk/low_hook.wav");
- strcpy(filenames[28],"sound/talk/attitude.wav");
- strcpy(filenames[29],"sound/talk/keep_com.wav");
- strcpy(filenames[30],"sound/stall.wav");
- strcpy(filenames[31],"sound/warning.wav");
- strcpy(filenames[32],"sound/cat_lnch.wav");
- strcpy(filenames[33],"sound/rumble.wav");
- strcpy(filenames[34],"sound/ping.wav");
- strcpy(filenames[35],"sound/explode.wav");
- strcpy(filenames[36],"sound/ab_rmbl.wav");
- strcpy(filenames[37],"sound/ab_ignit.wav");
- strcpy(filenames[38],"sound/talk/n0.wav");
- strcpy(filenames[39],"sound/talk/n1.wav");
- strcpy(filenames[40],"sound/talk/n2.wav");
- strcpy(filenames[41],"sound/talk/n3.wav");
- strcpy(filenames[42],"sound/talk/n4.wav");
- strcpy(filenames[43],"sound/talk/n5.wav");
- strcpy(filenames[44],"sound/talk/n6.wav");
- strcpy(filenames[45],"sound/talk/n7.wav");
- strcpy(filenames[46],"sound/talk/n8.wav");
- strcpy(filenames[47],"sound/talk/n9.wav");
- strcpy(filenames[48],"sound/talk/point.wav");
- strcpy(filenames[49],"sound/talk/tom_ball.wav");
- strcpy(filenames[50],"sound/talk/cm_left.wav");
- strcpy(filenames[51],"sound/talk/cm_right.wav");
- strcpy(filenames[52],"sound/rocket.wav");
- strcpy(filenames[53],"sound/lm/eagle.wav");
- strcpy(filenames[54],"sound/lm/700.wav");
- strcpy(filenames[55],"sound/lm/540.wav");
- strcpy(filenames[56],"sound/lm/200.wav");
- strcpy(filenames[57],"sound/lm/100.wav");
- strcpy(filenames[58],"sound/lm/70.wav");
- strcpy(filenames[59],"sound/lm/50.wav");
- strcpy(filenames[60],"sound/lm/contact.wav");
- strcpy(filenames[61],"sound/lm/engstop.wav");
+ strcpy(filenames[ 0],"/usr/share/vertigo/sound/cannon.wav");
+ strcpy(filenames[ 1],"/usr/share/vertigo/sound/screetch.wav");
+ strcpy(filenames[ 2],"/usr/share/vertigo/sound/gear_up.wav");
+ strcpy(filenames[ 3],"/usr/share/vertigo/sound/prop.wav");
+ strcpy(filenames[ 4],"/usr/share/vertigo/sound/jet_ext.wav");
+ strcpy(filenames[ 5],"/usr/share/vertigo/sound/crash.wav");
+ strcpy(filenames[ 6],"/usr/share/vertigo/sound/flaps.wav");
+ strcpy(filenames[ 7],"/usr/share/vertigo/sound/talk/call_bal.wav");
+ strcpy(filenames[ 8],"/usr/share/vertigo/sound/talk/vir_ball.wav");
+ strcpy(filenames[ 9],"/usr/share/vertigo/sound/talk/rog_ball.wav");
+ strcpy(filenames[10],"/usr/share/vertigo/sound/talk/lit_low.wav");
+ strcpy(filenames[11],"/usr/share/vertigo/sound/talk/low.wav");
+ strcpy(filenames[12],"/usr/share/vertigo/sound/talk/lit_high.wav");
+ strcpy(filenames[13],"/usr/share/vertigo/sound/talk/high.wav");
+ strcpy(filenames[14],"/usr/share/vertigo/sound/talk/lit_slow.wav");
+ strcpy(filenames[15],"/usr/share/vertigo/sound/talk/slow.wav");
+ strcpy(filenames[16],"/usr/share/vertigo/sound/talk/lit_fast.wav");
+ strcpy(filenames[17],"/usr/share/vertigo/sound/talk/fast.wav");
+ strcpy(filenames[18],"/usr/share/vertigo/sound/talk/lit_righ.wav");
+ strcpy(filenames[19],"/usr/share/vertigo/sound/talk/lin_righ.wav");
+ strcpy(filenames[20],"/usr/share/vertigo/sound/talk/lit_left.wav");
+ strcpy(filenames[21],"/usr/share/vertigo/sound/talk/lin_left.wav");
+ strcpy(filenames[22],"/usr/share/vertigo/sound/talk/eagle2.wav");
+ strcpy(filenames[22],"/usr/share/vertigo/sound/talk/wind_axi.wav");
+ strcpy(filenames[23],"/usr/share/vertigo/sound/talk/power.wav");
+ strcpy(filenames[24],"/usr/share/vertigo/sound/talk/waveoff.wav");
+ strcpy(filenames[25],"/usr/share/vertigo/sound/talk/bolter.wav");
+ strcpy(filenames[26],"/usr/share/vertigo/sound/talk/low_gear.wav");
+ strcpy(filenames[27],"/usr/share/vertigo/sound/talk/low_hook.wav");
+ strcpy(filenames[28],"/usr/share/vertigo/sound/talk/attitude.wav");
+ strcpy(filenames[29],"/usr/share/vertigo/sound/talk/keep_com.wav");
+ strcpy(filenames[30],"/usr/share/vertigo/sound/stall.wav");
+ strcpy(filenames[31],"/usr/share/vertigo/sound/warning.wav");
+ strcpy(filenames[32],"/usr/share/vertigo/sound/cat_lnch.wav");
+ strcpy(filenames[33],"/usr/share/vertigo/sound/rumble.wav");
+ strcpy(filenames[34],"/usr/share/vertigo/sound/ping.wav");
+ strcpy(filenames[35],"/usr/share/vertigo/sound/explode.wav");
+ strcpy(filenames[36],"/usr/share/vertigo/sound/ab_rmbl.wav");
+ strcpy(filenames[37],"/usr/share/vertigo/sound/ab_ignit.wav");
+ strcpy(filenames[38],"/usr/share/vertigo/sound/talk/n0.wav");
+ strcpy(filenames[39],"/usr/share/vertigo/sound/talk/n1.wav");
+ strcpy(filenames[40],"/usr/share/vertigo/sound/talk/n2.wav");
+ strcpy(filenames[41],"/usr/share/vertigo/sound/talk/n3.wav");
+ strcpy(filenames[42],"/usr/share/vertigo/sound/talk/n4.wav");
+ strcpy(filenames[43],"/usr/share/vertigo/sound/talk/n5.wav");
+ strcpy(filenames[44],"/usr/share/vertigo/sound/talk/n6.wav");
+ strcpy(filenames[45],"/usr/share/vertigo/sound/talk/n7.wav");
+ strcpy(filenames[46],"/usr/share/vertigo/sound/talk/n8.wav");
+ strcpy(filenames[47],"/usr/share/vertigo/sound/talk/n9.wav");
+ strcpy(filenames[48],"/usr/share/vertigo/sound/talk/point.wav");
+ strcpy(filenames[49],"/usr/share/vertigo/sound/talk/tom_ball.wav");
+ strcpy(filenames[50],"/usr/share/vertigo/sound/talk/cm_left.wav");
+ strcpy(filenames[51],"/usr/share/vertigo/sound/talk/cm_right.wav");
+ strcpy(filenames[52],"/usr/share/vertigo/sound/rocket.wav");
+ strcpy(filenames[53],"/usr/share/vertigo/sound/lm/eagle.wav");
+ strcpy(filenames[54],"/usr/share/vertigo/sound/lm/700.wav");
+ strcpy(filenames[55],"/usr/share/vertigo/sound/lm/540.wav");
+ strcpy(filenames[56],"/usr/share/vertigo/sound/lm/200.wav");
+ strcpy(filenames[57],"/usr/share/vertigo/sound/lm/100.wav");
+ strcpy(filenames[58],"/usr/share/vertigo/sound/lm/70.wav");
+ strcpy(filenames[59],"/usr/share/vertigo/sound/lm/50.wav");
+ strcpy(filenames[60],"/usr/share/vertigo/sound/lm/contact.wav");
+ strcpy(filenames[61],"/usr/share/vertigo/sound/lm/engstop.wav");
 
  /* Only defined for samples where needed. Assumes 1000 ticks/sec */
  for (i=0; i<MAXNSOUNDS; i++)
--- vertigo.c.orig	2012-02-16 20:03:02.318304980 +0600
+++ vertigo.c	2012-02-16 20:04:56.241638959 +0600
@@ -644,7 +644,7 @@
  int i;
 
 
- if(!(fp=fopen("objects/objects.dat","rb"))) {
+ if(!(fp=fopen("/usr/share/vertigo/objects/objects.dat","rb"))) {
           errormessage("Cannot read OBJECTS/OBJECTS.DAT");
        }
 
@@ -677,7 +677,7 @@
  FILE *fp;
  char string[80];
 
- if ((fp=fopen("objects/objects.pos","r")) == NULL)
+ if ((fp=fopen("/usr/share/vertigo/objects/objects.pos","r")) == NULL)
         errormessage("Cannot open file: OBJECTS/OBJECTS.POS\n\r");
 
  fscanf(fp,"%s", string);
@@ -756,7 +756,7 @@
 
  init_sound();
 
- data = load_datafile("fonts.dat");
+ data = load_datafile("/usr/share/vertigo/fonts.dat");
  if (!data) {
       set_gfx_mode(GFX_TEXT, 0, 0, 0, 0);
       allegro_message("Error loading fonts\n");
@@ -768,7 +768,11 @@
    int_cos[i]=cos(degtorad(i));
  }
 
- if(access(cnf_file,0)) { /* VER_CNF.* NOT present */
+ char cnf_path[40];
+ strcpy(cnf_path, getenv("HOME"));
+ strcat(cnf_path, "/.vertigo.cfg");
+
+ if(access(cnf_path,0)) { /* VER_CNF.* NOT present */
 
 #ifdef LINUX
 printf("%s missing, so we start configuring.\n", cnf_file);
--- cgui/cgmenus.c.orig	2012-02-16 17:19:46.541582938 +0600
+++ cgui/cgmenus.c	2012-02-16 17:21:13.068250103 +0600
@@ -912,7 +912,7 @@
   static int winid;
 
   winid = MkDialogue(ADAPTIVE, "Keys reference", W_FLOATING);
-  id = AddTextBox(DOWNLEFT, "keys.txt", 670, 54, TB_TEXTFILE|TB_FRAMESINK|TB_PREFORMAT|3); 
+  id = AddTextBox(DOWNLEFT, "/usr/share/doc/vertigo/keys.txt", 670, 54, TB_TEXTFILE|TB_FRAMESINK|TB_PREFORMAT|3); 
   MakeStretchable(id, NULL, NULL, 0);
   AddButton(DOWNLEFT, "~Close", close_show_keys, &winid);
   DisplayWin();
@@ -933,7 +934,7 @@
   static int winid;
 
   winid = MkDialogue(ADAPTIVE, "Vertigo manual", W_FLOATING);
- id = AddTextBox(DOWNLEFT, "vertigo.txt", 670, 54, TB_TEXTFILE|TB_FRAMESINK|TB_PREFORMAT|3); 
+ id = AddTextBox(DOWNLEFT, "/usr/share/doc/vertigo/vertigo.txt", 670, 54, TB_TEXTFILE|TB_FRAMESINK|TB_PREFORMAT|3); 
   MakeStretchable(id, NULL, NULL, 0);
   AddButton(DOWNLEFT, "~Close", close_show_manual, &winid);
   DisplayWin();
