--- makedefs.orig	2012-02-16 20:05:54.891639289 +0600
+++ makedefs	2012-02-16 20:08:00.071640000 +0600
@@ -11,8 +11,8 @@
 #GRAPHTYPE = CONSOLE
 
 #Select static or shared libraries (Linux only):
-LIBTYPE = STATIC
-#LIBTYPE = SHARED
+#LIBTYPE = STATIC
+LIBTYPE = SHARED
 
 ifndef DJGPP
  LINUX  = TRUE
@@ -32,7 +32,7 @@
 CC     = gcc
 
 #High performance 
-OPTFLAGS = -mcpu=pentium -O2 -ffast-math -fomit-frame-pointer
+OPTFLAGS = -O2 -ffast-math
 #Development
 #OPTFLAGS = 
 
@@ -44,7 +44,7 @@
 #CFLAGS = -g
 
 #Normal stripped
-LINKFLAGS = -s
+#LINKFLAGS = -s
 #For debugging
 #LINKFLAGS =
 
@@ -103,13 +103,13 @@
    SYS_LIBS    = -lmgui $(ALLEGRO_LIB)
  endif
  ifeq ($(GUI),CGUI)
-   SYS_LIBS    = $(ALLEGRO)/lib/unix/libaldat.a -lcgui $(ALLEGRO_LIB)
+   SYS_LIBS    = $(ALLEGRO_LIB) -lcgui
    SYS_OBJECTS = cgmenus.o
    SYS_HEADERS = cgui/cgmenus.h
  endif
 
  ifeq ($(GRAPHTYPE),X11)
-  LIBG   = $(CGUIPATH)/obj/djgpp/tools/datedit.o $(SYS_LIBS) -lm -L/usr/X11R6/lib -lX11
+  LIBG   = $(SYS_LIBS) -lm -lX11
  else
   LIBG   = /x/home/norup/vertigo/cgui/obj/djgpp/tools/datedit.o -lm -lvga $(SYS_LIBS)
  endif
