Index: edje/edje.c_edje_object.pxi
===================================================================
--- edje/edje.c_edje_object.pxi	(revision 55981)
+++ edje/edje.c_edje_object.pxi	(working copy)
@@ -139,7 +139,7 @@
         ...
         >>> my_edje.message_handler_set(msg_dbg)
     """
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self._signal_callbacks = {}
 
     def __init__(self, c_evas.Canvas canvas not None, **kargs):
