--- a/fab.rb	2013-02-22 15:36:57.000000000 +0100
+++ b/fab.rb	2014-10-15 21:53:51.054301814 +0200
@@ -29,7 +29,7 @@
 
 require "rubygems"
 require 'getoptlong'
-require "./cewl_lib.rb"
+require "cewl/cewl_lib.rb"
 
 opts = GetoptLong.new(
 	[ '--help', '-h', GetoptLong::NO_ARGUMENT ],
@@ -37,14 +37,11 @@
 )
 
 def usage
-	puts"xx
-
-Usage: xx [OPTION] ... filename/list
+	puts "Usage: xx [OPTION] ... filename/list
 	-h, --help: show help
 	-v: verbose
 	
 	filename/list: the file or list of files to check
-
 "
 	exit
 end
