--- 30-cjk-aliases.conf	2013-04-23 22:53:00.987375178 +0800
+++ 30-cjk-aliases.conf	2013-04-15 00:06:37.000000000 +0800
@@ -4,15 +4,75 @@
 <!-- Aliases for Korean fonts -->
     <alias>
         <family>Batang</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
         <family>바탕</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
         <family>BatangChe</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
         <family>바탕체</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
 	<family>Myeongjo</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
         <family>명조</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
         <family>MyeongjoChe</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
         <family>명조체</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
         <family>AR MingtiM KSC</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
         <family>Adobe 명조 Std M</family>
+        <accept>
+	    <family>NanumMyeongjo</family>
+            <family>UnBatang</family>
+        </accept>
+    </alias>
+    <alias>
         <family>Adobe Myeongjo Std M</family>
         <accept>
 	    <family>NanumMyeongjo</family>
@@ -21,8 +81,26 @@
     </alias>
     <alias>
         <family>Gungsuh</family>
+        <accept>
+            <family>UnGungseo</family> 
+	    <family>NanumMyeongjo</family>
+        </accept>
+    </alias>
+    <alias>
         <family>궁서</family>
+        <accept>
+            <family>UnGungseo</family> 
+	    <family>NanumMyeongjo</family>
+        </accept>
+    </alias>
+    <alias>
         <family>GungsuhChe</family>
+        <accept>
+            <family>UnGungseo</family> 
+	    <family>NanumMyeongjo</family>
+        </accept>
+    </alias>
+    <alias>
         <family>궁서체</family>
         <accept>
             <family>UnGungseo</family> 
@@ -31,14 +109,68 @@
     </alias>
     <alias>
         <family>Dotum</family>
+        <accept>
+	    <family>NanumGothic</family>
+            <family>UnDotum</family>
+        </accept>
+    </alias>
+    <alias>
         <family>돋움</family>
+        <accept>
+	    <family>NanumGothic</family>
+            <family>UnDotum</family>
+        </accept>
+    </alias>
+    <alias>
         <family>Gothic</family>
+        <accept>
+	    <family>NanumGothic</family>
+            <family>UnDotum</family>
+        </accept>
+    </alias>
+    <alias>
         <family>고딕</family>
+        <accept>
+	    <family>NanumGothic</family>
+            <family>UnDotum</family>
+        </accept>
+    </alias>
+    <alias>
 	<family>Malgun Gothic</family>
+        <accept>
+	    <family>NanumGothic</family>
+            <family>UnDotum</family>
+        </accept>
+    </alias>
+    <alias>
         <family>맑은 고딕</family>
+        <accept>
+	    <family>NanumGothic</family>
+            <family>UnDotum</family>
+        </accept>
+    </alias>
+    <alias>
 	<family>Gulim</family>
+        <accept>
+	    <family>NanumGothic</family>
+            <family>UnDotum</family>
+        </accept>
+    </alias>
+    <alias>
         <family>굴림</family>
+        <accept>
+	    <family>NanumGothic</family>
+            <family>UnDotum</family>
+        </accept>
+    </alias>
+    <alias>
 	<family>AppleGothic</family>
+        <accept>
+	    <family>NanumGothic</family>
+            <family>UnDotum</family>
+        </accept>
+    </alias>
+    <alias>
         <family>애플고딕</family>
         <accept>
 	    <family>NanumGothic</family>
@@ -47,10 +179,40 @@
     </alias>
     <alias>
         <family>DotumChe</family>
+        <accept>
+	    <family>NanumGothicCoding</family>
+	    <family>NanumGothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>돋움체</family>
+        <accept>
+	    <family>NanumGothicCoding</family>
+	    <family>NanumGothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>GothicChe</family>
+        <accept>
+	    <family>NanumGothicCoding</family>
+	    <family>NanumGothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>고딕체</family>
+        <accept>
+	    <family>NanumGothicCoding</family>
+	    <family>NanumGothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>GulimChe</family>
+        <accept>
+	    <family>NanumGothicCoding</family>
+	    <family>NanumGothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>굴림체</family>
         <accept>
 	    <family>NanumGothicCoding</family>
@@ -60,6 +222,16 @@
 <!-- Aliases for Japanese Windows fonts -->
     <alias>
         <family>MS Gothic</family>
+        <accept>
+            <family>TakaoGothic</family>
+            <family>IPAGothic</family>
+            <family>IPAMonaGothic</family>
+            <family>VL Gothic</family>
+            <family>Sazanami Gothic</family>
+            <family>Kochi Gothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>ＭＳ ゴシック</family>
         <accept>
             <family>TakaoGothic</family>
@@ -72,8 +244,38 @@
     </alias>
     <alias>
         <family>MS PGothic</family>
+        <accept>
+            <family>IPAMonaPGothic</family>
+            <family>TakaoPGothic</family>
+            <family>IPAPGothic</family>
+            <family>VL PGothic</family>
+            <family>Sazanami Gothic</family>
+            <family>Kochi Gothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>ＭＳ Ｐゴシック</family>
+        <accept>
+            <family>IPAMonaPGothic</family>
+            <family>TakaoPGothic</family>
+            <family>IPAPGothic</family>
+            <family>VL PGothic</family>
+            <family>Sazanami Gothic</family>
+            <family>Kochi Gothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>MS UIGothic</family>
+        <accept>
+            <family>IPAMonaPGothic</family>
+            <family>TakaoPGothic</family>
+            <family>IPAPGothic</family>
+            <family>VL PGothic</family>
+            <family>Sazanami Gothic</family>
+            <family>Kochi Gothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>Meiryo UI</family>
         <accept>
             <family>IPAMonaPGothic</family>
@@ -86,7 +288,25 @@
     </alias>
     <alias>
         <family>MS Mincho</family>
+        <accept>
+            <family>TakaoMincho</family>
+            <family>IPAMincho</family>
+            <family>IPAMonaMincho</family>
+            <family>Sazanami Mincho</family>
+            <family>Kochi Mincho</family>
+        </accept>
+    </alias>
+    <alias>
         <family>ＭＳ 明朝</family>
+        <accept>
+            <family>TakaoMincho</family>
+            <family>IPAMincho</family>
+            <family>IPAMonaMincho</family>
+            <family>Sazanami Mincho</family>
+            <family>Kochi Mincho</family>
+        </accept>
+    </alias>
+    <alias>
         <family>AR MinchoL JIS</family>
         <accept>
             <family>TakaoMincho</family>
@@ -98,6 +318,15 @@
     </alias>
     <alias>
         <family>MS PMincho</family>
+        <accept>
+            <family>IPAMonaPMincho</family>
+            <family>TakaoPMincho</family>
+            <family>IPAPMincho</family>
+            <family>Sazanami Mincho</family>
+            <family>Kochi Mincho</family>
+        </accept>
+    </alias>
+    <alias>
         <family>ＭＳ Ｐ明朝</family>
         <accept>
             <family>IPAMonaPMincho</family>
@@ -109,6 +338,11 @@
     </alias>
     <alias>
         <family>Meiryo</family>
+        <accept>
+            <family>IPAexGothic</family>
+        </accept>
+    </alias>
+    <alias>
         <family>メイリオ</family>
         <accept>
             <family>IPAexGothic</family>
@@ -117,11 +351,47 @@
 <!-- Aliases for Simplified Chinese Windows fonts -->
     <alias>
         <family>SimSun</family>
+        <accept>
+            <family>HYSong</family>
+            <family>AR PL UMing CN</family>
+        </accept>
+    </alias>
+    <alias>
         <family>NSimSun</family>
+        <accept>
+            <family>HYSong</family>
+            <family>AR PL UMing CN</family>
+        </accept>
+    </alias>
+    <alias>
         <family>SimSun-18030</family>
+        <accept>
+            <family>HYSong</family>
+            <family>AR PL UMing CN</family>
+        </accept>
+    </alias>
+    <alias>
         <family>NSimSun-18030</family>
+        <accept>
+            <family>HYSong</family>
+            <family>AR PL UMing CN</family>
+        </accept>
+    </alias>
+    <alias>
         <family>宋体</family>
+        <accept>
+            <family>HYSong</family>
+            <family>AR PL UMing CN</family>
+        </accept>
+    </alias>
+    <alias>
         <family>新宋体</family>
+        <accept>
+            <family>HYSong</family>
+            <family>AR PL UMing CN</family>
+        </accept>
+    </alias>
+    <alias>
         <family>AR MingtiM GB</family>
         <accept>
             <family>HYSong</family>
@@ -130,6 +400,12 @@
     </alias>
     <alias>
         <family>KaiTi</family>
+        <accept>
+            <family>AR PL UKai CN</family>
+            <family>AR PL ZenKai Uni</family>
+        </accept>
+    </alias>
+    <alias>
         <family>楷体</family>
         <accept>
             <family>AR PL UKai CN</family>
@@ -138,6 +414,12 @@
     </alias>
     <alias>
         <family>Microsoft YaHei</family>
+        <accept>
+            <family>WenQuanYi Micro Hei</family>
+            <family>WenQuanYi Zen Hei</family>
+        </accept>
+    </alias>
+    <alias>
         <family>微软雅黑</family>
         <accept>
             <family>WenQuanYi Micro Hei</family>
@@ -147,9 +429,29 @@
 <!-- Aliases for Traditional Chinese Windows fonts -->
     <alias>
         <family>MingLiU</family>
+        <accept>
+            <family>AR PL UMing TW</family>
+        </accept>
+    </alias>
+    <alias>
         <family>細明體</family>
+        <accept>
+            <family>AR PL UMing TW</family>
+        </accept>
+    </alias>
+    <alias>
         <family>PMingLiU</family>
+        <accept>
+            <family>AR PL UMing TW</family>
+        </accept>
+    </alias>
+    <alias>
         <family>新細明體</family>
+        <accept>
+            <family>AR PL UMing TW</family>
+        </accept>
+    </alias>
+    <alias>
         <family>AR MingtiM BIG-5</family>
         <accept>
             <family>AR PL UMing TW</family>
@@ -157,6 +459,12 @@
     </alias>
     <alias>
         <family>DFKai\-SB</family>
+        <accept>
+            <family>AR PL UKai TW</family>
+            <family>AR PL ZenKai Uni</family>
+        </accept>
+    </alias>
+    <alias>
         <family>標楷體</family>
         <accept>
             <family>AR PL UKai TW</family>
@@ -165,6 +473,12 @@
     </alias>
     <alias>
         <family>Microsoft JhengHei</family>
+        <accept>
+            <family>WenQuanYi Micro Hei</family>
+            <family>WenQuanYi Zen Hei</family>
+        </accept>
+    </alias>
+    <alias>
         <family>微軟正黑體</family>
         <accept>
             <family>WenQuanYi Micro Hei</family>
@@ -174,7 +488,17 @@
 <!-- Alias for HKSCS -->
     <alias>
         <family>Ming (for ISO10646)</family>
+        <accept>
+            <family>AR PL UMing HK</family>
+        </accept>
+    </alias>
+    <alias>
         <family>MingLiU_HKSCS</family>
+        <accept>
+            <family>AR PL UMing HK</family>
+        </accept>
+    </alias>
+    <alias>
         <family>細明體_HKSCS</family>
         <accept>
             <family>AR PL UMing HK</family>
--- 41-ttf-arphic-ukai.conf	2013-04-23 22:53:01.004042310 +0800
+++ 41-ttf-arphic-ukai.conf	2013-04-10 03:16:22.000000000 +0800
@@ -7,10 +7,25 @@
  -->
 	<alias>
 		<family>AR PL ZenKai Uni</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL ZenKai Uni MBE</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UKai CN</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UKai HK</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UKai TW</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UKai TW MBE</family>
 		<default><family>sans-serif</family></default>
 	</alias>
--- 41-ttf-arphic-uming.conf	2013-04-23 22:53:01.004042310 +0800
+++ 41-ttf-arphic-uming.conf	2013-04-10 03:13:26.000000000 +0800
@@ -7,10 +7,25 @@
  -->
 	<alias>
 		<family>AR PL ShanHeiSun Uni</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL ShanHeiSun Uni MBE</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing CN</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing HK</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW</family>
+		<default><family>serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW MBE</family>
 		<default><family>serif</family></default>
 	</alias>
@@ -19,10 +34,25 @@
  -->
 	<alias>
 		<family>AR PL ShanHeiSun Uni</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL ShanHeiSun Uni MBE</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing CN</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing HK</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW MBE</family>
 		<default><family>sans-serif</family></default>
 	</alias>
@@ -31,10 +61,25 @@
  -->
  	<alias>
 		<family>AR PL ShanHeiSun Uni</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL ShanHeiSun Uni MBE</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing CN</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing HK</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>AR PL UMing TW MBE</family>
 		<default><family>monospace</family></default>
 	</alias>
--- 44-wqy-zenhei.conf	2013-04-23 22:53:01.014042589 +0800
+++ 44-wqy-zenhei.conf	2013-04-10 02:55:44.000000000 +0800
@@ -6,7 +6,11 @@
   <match target="font">
         <test qual="any" name="family">
                 <string>WenQuanYi Zen Hei</string>
+        </test> 
+        <test qual="any" name="family">
                 <string>文泉驿正黑</string>
+        </test> 
+        <test qual="any" name="family">
                 <string>文泉驛正黑</string>
         </test> 
         <edit name="globaladvance"><bool>false</bool></edit>
--- 99-language-selector-zh.conf	2013-04-23 22:53:00.994042031 +0800
+++ 99-language-selector-zh.conf	2013-04-10 02:56:29.000000000 +0800
@@ -5,8 +5,14 @@
 	<match target="font" >
 		<test name="family" compare="contains" >
 			<string>Song</string>
+		</test> 
+		<test name="family" compare="contains" >
 			<string>Sun</string>
+		</test> 
+		<test name="family" compare="contains" >
 			<string>Kai</string>
+		</test> 
+		<test name="family" compare="contains" >
 			<string>Ming</string>
 		</test> 
                 <!-- check to see if the font is just regular -->
