--- Crypt-Cracklib-1.5/Cracklib.pm	2008-11-08 15:20:04.000000000 -0800
+++ Crypt-Cracklib-patched/Cracklib.pm	2010-09-28 08:59:19.000000000 -0700
@@ -12,23 +12,8 @@
 @ISA = qw(Exporter DynaLoader);
 @EXPORT = qw(fascist_check check);
 
-$DEFAULT_DICT = "";
-
-{
-	for my $path (qw(
-		/usr/share/pw_dict
-		/var/cache/cracklib/cracklib_dict
-		/usr/lib/cracklib_dict
-		/usr/lib64/cracklib_dict
-		/usr/share/dict/cracklib_words)) {
-
-		if (-f "$path.pwd") {
-
-			$DEFAULT_DICT = $path;
-			last;
-		}
-	}
-}
+# Patched for ArchLinux...
+$DEFAULT_DICT = '/usr/share/cracklib/pw_dict';
 
 # Wrapper.
 sub fascist_check {
