diff -Naur Crypt-Cracklib-1.7/Cracklib.pm Crypt-Cracklib-1.7-patched//Cracklib.pm
--- Crypt-Cracklib-1.7/Cracklib.pm	2010-12-29 18:30:12.000000000 -0500
+++ Crypt-Cracklib-1.7-patched//Cracklib.pm	2011-06-01 16:10:48.000000000 -0400
@@ -10,23 +10,8 @@
 @ISA = qw(Exporter DynaLoader);
 @EXPORT = qw(fascist_check check bad_pass);
 
-$DEFAULT_DICT = "";
-
-{
-  for my $path (qw(
-    /usr/share/pw_dict
-    /var/cache/cracklib/cracklib_dict
-    /usr/lib/cracklib_dict
-    /usr/lib64/cracklib_dict
-    /usr/share/dict/cracklib_words)) {
-
-    if (-f "$path.pwd") {
-
-      $DEFAULT_DICT = $path;
-      last;
-    }
-  }
-}
+# Patched for ArchLinux...
+$DEFAULT_DICT = '/usr/share/cracklib/pw_dict';
 
 # Wrapper.
 sub fascist_check {
