--- origsrc/gettext-0.18.1.1/gettext-tools/misc/convert-archive.in	2010-06-06 08:49:57 -0400
+++ src/gettext-0.18.1.1/gettext-tools/misc/convert-archive.in	2011-10-21 11:35:14 -0400
@@ -270,7 +270,7 @@ case "$to" in
 
     # Set a nonstandard variable, for a good-looking cvs history.
     cvsuser=bruno
-    gcc -shared -fPIC -O cvsuser.c -o cvsuser.so
+    gcc -shared -fPIC -O /c/MinGW-src/gettext/gettext-0.18.1.1-1/src/gettext-0.18.1.1/gettext-tools/misc/cvsuser.c -o cvsuser.so
     cvsuser_hack=`pwd`/cvsuser.so
 
     # Need to pass -d "$CVSROOT", because there may be a CVS directory in the
@@ -292,13 +292,15 @@ case "$to" in
         && cvs -d "$CVSROOT" checkout -r"$cvsver" archive > /dev/null \
         && cvs -d "$CVSROOT" admin -ko `find . -type f -print | sed -e 's,^\./,,' | grep -v '^CVS/' | grep -v '/CVS/'` > /dev/null
        ) 2> cvs.err || { cat cvs.err 1>&2; func_fatal_error "cvs checkout or admin failed"; }
-       rm -rf tmpcheckout
+       rm -rf tmpcheckout || rm -rf tmpcheckout || true
+       echo "Finished importing version $version"
+       sleep 2
      done
     )
     (cd "$cvs_dir" && $TAR cf - $TAR_OPTIONS archive) \
       | gzip -c -9 > "$tofile" \
       || func_fatal_error "archive creation failed"
-    rm -rf "$cvs_dir" cvsuser.so
+    rm -rf "$cvs_dir" cvsuser.so || rm -rf "$cvs_dir" cvsuser.so || true
     ;;
 
   git)
