diff --git a/data/Makefile.am b/data/Makefile.am
index e43f23e..5b7f49e 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -13,8 +13,40 @@ apps_DATA = gnome-music.desktop
 
 @GSETTINGS_RULES@
 
+# icons
+icon16dir = $(datadir)/icons/hicolor/16x16/apps
+icon16_DATA = icons/hicolor/16x16/apps/gnome-music.png
+icon22dir = $(datadir)/icons/hicolor/22x22/apps
+icon22_DATA = icons/hicolor/22x22/apps/gnome-music.png
+icon32dir = $(datadir)/icons/hicolor/32x32/apps
+icon32_DATA = icons/hicolor/32x32/apps/gnome-music.png
+icon48dir = $(datadir)/icons/hicolor/48x48/apps
+icon48_DATA = icons/hicolor/48x48/apps/gnome-music.png
+icon256dir = $(datadir)/icons/hicolor/256x256/apps
+icon256_DATA = icons/hicolor/256x256/apps/gnome-music.png
+icon_files = \
+	$(icon16_DATA) \
+	$(icon22_DATA) \
+	$(icon32_DATA) \
+	$(icon48_DATA) \
+	$(icon256_DATA)
+
+
+gtk_update_icon_cache = gtk-update-icon-cache -f -t $(datadir)/icons/hicolor
+
+install-data-hook: update-icon-cache
+uninstall-hook: update-icon-cache
+update-icon-cache:
+	@-if test -z "$(DESTDIR)"; then \
+		echo "Updating Gtk icon cache."; \
+		$(gtk_update_icon_cache); \
+	else \
+		echo "*** Icon cache not updated.  After (un)install, run this:"; \
+		echo "***   $(gtk_update_icon_cache)"; \
+	fi
+
 EXTRA_DIST = \
-	CREDITS \
+	$(icon_files) \
 	gnome-music.desktop.in \
 	gnome-music.gresource.xml \
 	$(resource_files) \
diff --git a/data/gnome-music.desktop.in b/data/gnome-music.desktop.in
index 3843c39..6693485 100644
--- a/data/gnome-music.desktop.in
+++ b/data/gnome-music.desktop.in
@@ -2,7 +2,7 @@
 _Name=Music
 _GenericName=Music Player
 _Comment=Play and organize your music collection
-Icon=audio-x-generic
+Icon=gnome-music
 Exec=gnome-music
 Terminal=false
 Type=Application
