--- install.pl	2011-02-05 05:02:38.845785881 -0700
+++ install2.pl	2011-02-05 05:03:21.350360881 -0700
@@ -120,10 +120,6 @@
 ### make sure the system binaries are where we think they are.
 &check_commands();
 
-### check to make sure we are running as root
-$< == 0 && $> == 0 or die "You need to be root (or equivalent UID 0",
-    " account) to install/uninstall fwsnort!\n";
-
 if ($uninstall) {
     &uninstall();
 } else {
--- install.pl	2011-01-02 18:38:51.000000000 -0700
+++ install2.pl	2011-02-05 04:59:01.653910881 -0700
@@ -471,8 +471,11 @@
 sub check_commands() {
     my @path = qw(
         /bin
+        /sbin
         /usr/bin
+        /usr/sbin
         /usr/local/bin
+        /usr/local/sbin
     );
     CMD: for my $cmd (keys %cmds) {
         unless (-x $cmds{$cmd}) {
