--- ./example.env	2013-12-03 14:03:52.635403422 -0800
+++ ./example.env	2013-12-03 15:42:03.631914331 -0800
@@ -2,14 +2,14 @@
 export PORT=8070
 export URL=http://localhost:8070/
 export NODE_ENV=development
-export MONGO_URI=mongodb://localhost/
+export MONGO_URI=mongodb://localhost:27017
 export TESTNET=1
 
 # requires bitcoind and https://github.com/shesek/bitcoin-webapi
 export TESTNET_API=http://localhost:9999/
 
 # build target for static assets
-#export TARGET=./build/
+export TARGET=./build/
 
 # enable if behind a reverse proxy
 #export PROXIED=1
