--- ./bin/start.sh	2013-12-03 16:52:50.871802035 -0800
+++ ./bin/start.sh	2013-12-03 16:53:28.455134374 -0800
@@ -1,5 +1,5 @@
 #!/bin/bash
-[ -f .env ]  && source .env
+[ -f /opt/bitrated/.env ]  && source /opt/bitrated/.env
 
 # Kill child process on exit
 trap 'kill $(jobs -p)' EXIT
@@ -7,8 +7,7 @@
 # Re-run server on crashes
 while :
 do
-  ./node_modules/.bin/coffee ./server/app.coffee "$@"
+  /opt/bitrated/node_modules/.bin/coffee /opt/bitrated/server/app.coffee "$@"
   echo "Server crashed with status $?"
   sleep 1
 done
-
