diff -Naur src.ori//gnome-gmail-1.7.2/evolution src//gnome-gmail-1.7.2/evolution
--- src.ori//gnome-gmail-1.7.2/evolution	2011-02-13 02:11:25.000000000 +0100
+++ src//gnome-gmail-1.7.2/evolution	2011-05-20 08:31:33.096062183 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Copyright 2011 David Steele <daves@users.sourceforge.net>
 # This file is part of gnome-gmail
 # Available under the terms of the GNU General Public License version 2 or later
diff -Naur src.ori//gnome-gmail-1.7.2/gnome-gmail src//gnome-gmail-1.7.2/gnome-gmail
--- src.ori//gnome-gmail-1.7.2/gnome-gmail	2011-02-13 02:11:25.000000000 +0100
+++ src//gnome-gmail-1.7.2/gnome-gmail	2011-05-20 08:31:33.099393850 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!/usr/bin/python2 -tt
 #
 # Copyright 2011 David Steele <daves@users.sourceforge.net>
 # This file is part of gnome-gmail
diff -Naur src.ori//gnome-gmail-1.7.2/gnome-gmail.desktop.in src//gnome-gmail-1.7.2/gnome-gmail.desktop.in
--- src.ori//gnome-gmail-1.7.2/gnome-gmail.desktop.in	2011-02-06 20:48:29.000000000 +0100
+++ src//gnome-gmail-1.7.2/gnome-gmail.desktop.in	2011-05-20 08:32:09.924306604 +0200
@@ -2,6 +2,7 @@
 _Name=Gnome Gmail
 GenericName=GMail Web Integration
 _Comment=Integrate GMail with your desktop
+MimeType=application/mbox;message/rfc822;x-scheme-handler/mailto
 Exec=gnome-gmail %U
 Icon=gnome-gmail
 Terminal=false
diff -Naur src.ori//gnome-gmail-1.7.2/setOOmailer src//gnome-gmail-1.7.2/setOOmailer
--- src.ori//gnome-gmail-1.7.2/setOOmailer	2011-02-13 02:11:25.000000000 +0100
+++ src//gnome-gmail-1.7.2/setOOmailer	2011-05-20 08:31:33.096062183 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright 2011 David Steele <daves@users.sourceforge.net>
 # This file is part of gnome-gmail
