diff -Naur src.ori/gnome-gmail-1.8.2/evolution src/gnome-gmail-1.8.2/evolution
--- src.ori/gnome-gmail-1.8.2/evolution	2011-10-28 05:03:05.000000000 +0200
+++ src/gnome-gmail-1.8.2/evolution	2011-11-20 16:59:45.444122989 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Copyright 2011 David Steele <daves@users.sourceforge.net>
 # This file is part of gnome-gmail
 # Available under the terms of the GNU General Public License version 2 or later
diff -Naur src.ori/gnome-gmail-1.8.2/gnome-gmail src/gnome-gmail-1.8.2/gnome-gmail
--- src.ori/gnome-gmail-1.8.2/gnome-gmail	2011-10-28 02:45:56.000000000 +0200
+++ src/gnome-gmail-1.8.2/gnome-gmail	2011-11-20 17:00:04.360921779 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!/usr/bin/python2 -tt
 #
 # Copyright 2011 David Steele <daves@users.sourceforge.net>
 # This file is part of gnome-gmail
diff -Naur src.ori/gnome-gmail-1.8.2/setOOmailer src/gnome-gmail-1.8.2/setOOmailer
--- src.ori/gnome-gmail-1.8.2/setOOmailer	2011-10-28 02:45:56.000000000 +0200
+++ src/gnome-gmail-1.8.2/setOOmailer	2011-11-20 17:13:41.066625928 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright 2011 David Steele <daves@users.sourceforge.net>
 # This file is part of gnome-gmail
@@ -223,7 +223,7 @@
         pass
 
     try:
-        Lbm = LibreMailerCfg( xml_path="/.libreoffice/3/user/registrymodifications.xcu" )
+        Lbm = LibreMailerCfg( xml_path="/.config/libreoffice/4/user/registrymodifications.xcu" )
 
         Lbm.set_external_mailer( external_mailer )
     except:
