diff -Naur src.ori/gnome-gmail-1.8.1/evolution src/gnome-gmail-1.8.1/evolution
--- src.ori/gnome-gmail-1.8.1/evolution	2011-08-17 04:49:22.000000000 +0200
+++ src/gnome-gmail-1.8.1/evolution	2011-09-15 18:12:39.085560950 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Copyright 2011 David Steele <daves@users.sourceforge.net>
 # This file is part of gnome-gmail
 # Available under the terms of the GNU General Public License version 2 or later
diff -Naur src.ori/gnome-gmail-1.8.1/gnome-gmail src/gnome-gmail-1.8.1/gnome-gmail
--- src.ori/gnome-gmail-1.8.1/gnome-gmail	2011-08-30 02:26:01.000000000 +0200
+++ src/gnome-gmail-1.8.1/gnome-gmail	2011-09-15 18:12:47.970000966 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!/usr/bin/python2 -tt
 #
 # Copyright 2011 David Steele <daves@users.sourceforge.net>
 # This file is part of gnome-gmail
diff -Naur src.ori/gnome-gmail-1.8.1/setOOmailer src/gnome-gmail-1.8.1/setOOmailer
--- src.ori/gnome-gmail-1.8.1/setOOmailer	2011-08-19 22:02:46.000000000 +0200
+++ src/gnome-gmail-1.8.1/setOOmailer	2011-09-15 18:14:42.550596167 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright 2011 David Steele <daves@users.sourceforge.net>
 # This file is part of gnome-gmail
@@ -213,7 +213,7 @@
         pass
 
     try:
-        Lbm = LibreMailerCfg( xml_path="/.libreoffice/3/user/registrymodifications.xcu" )
+        Lbm = LibreMailerCfg( xml_path="/.config/.libreoffice/3/user/registrymodifications.xcu" )
 
         Lbm.set_external_mailer( external_mailer )
     except:
