--- QScintilla-gpl-2.8.2/Python/configure.py~	2013-03-07 08:09:00.678704222 +0000
+++ QScintilla-gpl-2.8.2/Python/configure.py	2013-03-07 08:11:48.860880852 +0000
@@ -31,6 +31,8 @@
 import glob
 import optparse
 
+src_dir = os.path.dirname(os.path.abspath(__file__))
+
 try:
     import sysconfig
 except ImportError:
@@ -276,10 +278,10 @@
         qmake = dict(CONFIG='qscintilla2')
 
         if target_configuration.qsci_inc_dir is not None:
-            qmake['INCLUDEPATH'] = quote(target_config.qsci_inc_dir)
+            qmake['INCLUDEPATH'] = quote(target_configuration.qsci_inc_dir)
 
         if target_configuration.qsci_lib_dir is not None:
-            qmake['LIBS'] = '-L%s' % quote(target_config.qsci_inc_dir)
+            qmake['LIBS'] = '-L%s' % quote(target_configuration.qsci_lib_dir)
 
         if target_configuration.qsci_is_dll:
             qmake['DEFINES'] = 'QSCINTILLA_DLL'
@@ -459,7 +461,7 @@
                 last_name = None
                 continue
 
-            parts = l.split('=', maxsplit=1)
+            parts = l.split('=', 1)
             if len(parts) == 2:
                 name = parts[0].strip()
                 value = parts[1].strip()
@@ -504,11 +506,11 @@
                 return default
 
         # Handle any extrapolations.
-        parts = value.split('%(', maxsplit=1)
+        parts = value.split('%(', 1)
         while len(parts) == 2:
             prefix, tail = parts
 
-            parts = tail.split(')', maxsplit=1)
+            parts = tail.split(')', 1)
             if len(parts) != 2:
                 error(
                         "Configuration file contains unterminated "
@@ -527,7 +529,7 @@
 
             value = prefix + xtra_value + suffix
 
-            parts = value.split('%(', maxsplit=1)
+            parts = value.split('%(', 1)
 
         return value
 
@@ -1311,6 +1313,7 @@
     includepath = qmake_config.get('INCLUDEPATH')
     if includepath:
         pro.write('INCLUDEPATH += %s\n' % includepath)
+        pro.write('INCLUDEPATH += %s\n' % '/usr/include/qt/QtWidgets /usr/include/qt/QtPrintSupport')
 
     # Make sure the SIP include directory is searched before the Python include
     # directory if they are different.
