diff -Nur inform7-5Z71.orig/install-inform7.sh inform7-5Z71/install-inform7.sh
--- inform7-5Z71.orig/install-inform7.sh	2009-04-21 08:35:39.000000000 +1000
+++ inform7-5Z71/install-inform7.sh	2009-04-25 09:54:03.353936892 +1000
@@ -82,10 +82,8 @@
 tar xzf ${D}/${COMPDEP} || exit 4
 tar xzf ${D}/${TERPDEP} || exit 4
 
-if [ "$PREFIX" != "/usr/local" ]; then
-    cmd="s|/usr/local|$PREFIX|;"
-    perl -p -i -e "$cmd" bin/i7 || exit 6
-fi
+cmd="s|/usr/local|/usr|;"
+perl -p -i -e "$cmd" bin/i7 || exit 6
 
 cd ${D}
 exit 0
