--- indicator-cpufreq-0.1.4.orig/indicator_cpufreq/com.ubuntu.indicatorcpufreq.policy.in	2014-05-13 23:13:55.621383307 +0900
+++ indicator-cpufreq-0.1.4/indicator_cpufreq/com.ubuntu.indicatorcpufreq.policy.in	2014-05-15 18:40:01.305106732 +0900
@@ -22,4 +22,15 @@
       <allow_active>auth_admin_keep</allow_active>
     </defaults>
   </action>
+
+  <action id="com.ubuntu.indicatorcpufreqselector.setthrottle">
+    <_description>Change device powersaving settings</_description>
+    <_message>System policy prevents you from changing device powersaving.</_message>
+
+    <defaults>
+      <allow_inactive>no</allow_inactive>
+      <allow_active>auth_admin_keep</allow_active>
+    </defaults>
+  </action>
+
 </policyconfig>
