--- indicator-cpufreq/bin/indicator-cpufreq-selector	2014-12-18 14:43:40.368921000 +0900
+++ indicator-cpufreq/bin/indicator-cpufreq-selector	2014-12-18 15:07:46.037052843 +0900
@@ -21,6 +21,8 @@
 import dbus.service
 import dbus.mainloop.glib
 
+import subprocess
+
 from indicator_cpufreq import cpufreq
 
 class PermissionDeniedByPolicy(dbus.DBusException):
@@ -52,6 +54,24 @@
             error = cpufreq.set_frequency(cpu, frequency)
 
     @dbus.service.method(dbus_interface='com.ubuntu.IndicatorCpufreqSelector',
+        in_signature='as',
+        out_signature='',
+        sender_keyword='sender',
+        connection_keyword='conn')
+    def SetThrottle(self, throttle, sender=None, conn=None):
+        self._check_polkit_privilege(sender, conn, 'com.ubuntu.indicatorcpufreqselector.setthrottle')
+        error = subprocess.Popen(["/bin/sh", "throttle", throttle])
+
+    @dbus.service.method(dbus_interface='com.ubuntu.IndicatorCpufreqSelector',
+        in_signature='as',
+        out_signature='',
+        sender_keyword='sender',
+        connection_keyword='conn')
+    def SetHyperthreads(self, hyperthreads, sender=None, conn=None):
+        self._check_polkit_privilege(sender, conn, 'com.ubuntu.indicatorcpufreqselector.sethyperthreads')
+        error = subprocess.Popen(["/bin/sh", "throttle", hyperthreads])
+
+    @dbus.service.method(dbus_interface='com.ubuntu.IndicatorCpufreqSelector',
         in_signature='aus',
         out_signature='',
         sender_keyword='sender',
--- indicator-cpufreq/indicator_cpufreq/com.ubuntu.indicatorcpufreq.policy.in	2014-12-18 14:43:40.368921000 +0900
+++ indicator-cpufreq/indicator_cpufreq/com.ubuntu.indicatorcpufreq.policy.in	2014-12-18 14:20:33.225557677 +0900
@@ -22,4 +22,25 @@
       <allow_active>auth_admin_keep</allow_active>
     </defaults>
   </action>
+
+  <action id="com.ubuntu.indicatorcpufreqselector.setthrottle">
+    <_description>Change device powersaving settings</_description>
+    <_message>System policy prevents you from changing device powersaving.</_message>
+
+    <defaults>
+      <allow_inactive>no</allow_inactive>
+      <allow_active>auth_admin_keep</allow_active>
+    </defaults>
+  </action>
+
+  <action id="com.ubuntu.indicatorcpufreqselector.sethyperthreads">
+    <_description>Change hyperthreading settings</_description>
+    <_message>System policy prevents you from changing hyperthreading.</_message>
+
+    <defaults>
+      <allow_inactive>no</allow_inactive>
+      <allow_active>auth_admin_keep</allow_active>
+    </defaults>
+  </action>
+
 </policyconfig>
--- indicator-cpufreq/indicator_cpufreq/indicator.py	2014-12-18 14:43:40.368921000 +0900
+++ indicator-cpufreq/indicator_cpufreq/indicator.py	2014-12-18 14:39:42.519331153 +0900
@@ -50,6 +50,12 @@
     else:
         return g
 
+def readable_throttle(t):
+        return t
+
+def readable_hyperthreads(h):
+        return h
+
 class MyIndicator(object):
     def __init__(self, show_frequency=False):
         self.show_frequency = show_frequency
@@ -74,25 +80,65 @@
         # frequency menu items
         #freqs = cpufreq.get_available_frequencies(self.cpus[0])
         freqs = reversed(sorted(set(cpufreq.get_available_frequencies(self.cpus[0]))))
+        freqmenu = Gtk.Menu()
+        freqm = Gtk.MenuItem("CPU Frequency")
+        freqm.set_submenu(freqmenu)
         for freq in freqs:
             menu_item = Gtk.RadioMenuItem.new_with_label(group, readable_frequency(freq))
             group = menu_item.get_group()
-            menu.append(menu_item)
+            freqmenu.append(menu_item)
             menu_item.connect("activate", self.select_activated, 'frequency', freq)
             self.select_items[freq] = menu_item
+        if freqs != "":
+            menu.append(freqm)
 
         menu.append(Gtk.SeparatorMenuItem())
 
         # governor menu items
         governors = cpufreq.get_available_governors(self.cpus[0])
+        govmenu = Gtk.Menu()
+        govm = Gtk.MenuItem("CPU Governor")
+        govm.set_submenu(govmenu)
         for governor in governors:
             if governor == 'userspace':
                 continue
             menu_item = Gtk.RadioMenuItem.new_with_label(group, readable_governor(governor))
             group = menu_item.get_group()
-            menu.append(menu_item)
+            govmenu.append(menu_item)
             menu_item.connect('activate', self.select_activated, 'governor', governor)
             self.select_items[governor] = menu_item
+        if governors != "":
+            menu.append(govm)
+
+        menu.append(Gtk.SeparatorMenuItem())
+
+        # throttle menu items
+        throttles = ['full', 'cut']
+        throtmenu = Gtk.Menu()
+        throtm = Gtk.MenuItem("Devices Throttle")
+        throtm.set_submenu(throtmenu)
+        for toggle in throttles:
+            menu_item = Gtk.RadioMenuItem.new_with_label(group, readable_throttle(toggle))
+            group = menu_item.get_group()
+            throtmenu.append(menu_item)
+            menu_item.connect('activate', self.select_activated, 'throttle', toggle)
+            self.select_items[toggle] = menu_item
+        menu.append(throtm)
+
+        menu.append(Gtk.SeparatorMenuItem())
+
+        # hyperthreading menu items
+        hyperthreads = ['on', 'off']
+        htmenu = Gtk.Menu()
+        htm = Gtk.MenuItem("Hyperthreads")
+        htm.set_submenu(htmenu)
+        for toggle in hyperthreads:
+            menu_item = Gtk.RadioMenuItem.new_with_label(group, readable_hyperthreads(toggle))
+            group = menu_item.get_group()
+            htmenu.append(menu_item)
+            menu_item.connect('activate', self.select_activated, 'hyperthreads', toggle)
+            self.select_items[toggle] = menu_item
+        menu.append(htm)
 
         menu.show_all()        
         self.ind.set_menu(menu)
@@ -137,6 +181,12 @@
             if select == 'frequency':
                 proxy.SetFrequency(cpus, dbus.UInt32(value),
                     dbus_interface='com.ubuntu.IndicatorCpufreqSelector')
+            elif select == 'throttle':
+                proxy.SetThrottle(value,
+                    dbus_interface='com.ubuntu.IndicatorCpufreqSelector')
+            elif select == 'hyperthreads':
+                proxy.SetHyperthreads("ht-" + value,
+                    dbus_interface='com.ubuntu.IndicatorCpufreqSelector')
             else:
                 proxy.SetGovernor(cpus, value,
                     dbus_interface='com.ubuntu.IndicatorCpufreqSelector')
