diff --git a/Makefile b/Makefile
index 1e3133f..0a5e35a 100644
--- a/Makefile
+++ b/Makefile
@@ -250,10 +250,18 @@ lib-post-uninstall:
 
 # 'make' or 'make all': compile the WFDB applications without installing them
 all:		config.cache
-	$(MAKE) WFDBROOT=`pwd`/build install check
+	cd lib;	     $(MAKE) all
+	cd app;      $(MAKE) all
+	cd convert;  $(MAKE) all
+	cd data;     $(MAKE) all
+	cd psd;      $(MAKE) all
+	-( cd wave;  $(MAKE) all )
+	cd waverc;   $(MAKE) all
+	-( cd xml;   $(MAKE) all )
+	test -d doc && ( cd doc; $(MAKE) all )
 
 # 'make install': compile and install the WFDB software package
-install:	config.cache
+install:	
 	cd lib;	     $(MAKE) install
 	cd app;      $(MAKE) install
 	cd convert;  $(MAKE) install
diff --git a/app/Makefile b/app/Makefile
index ab25ac7..7dcb826 100644
--- a/app/Makefile
+++ b/app/Makefile
@@ -138,11 +138,11 @@ MFLAGS = -m32
 
 # CFLAGS is the set of C compiler options.  CFLAGS should always include
 # CCDEFS.
-CFLAGS = $(MFLAGS) -g -O $(CCDEFS) $(LC) -I$(INCDIR)
+CFLAGS = $(MFLAGS) -g -O $(CCDEFS) $(LC) -I$(INCDIR) -I..
 
 # LDFLAGS is appended to the C compiler command line to specify loading the
 # WFDB library.
-LDFLAGS = -L$(LIBDIR) -lwfdb $(LL)
+LDFLAGS = -L$(LIBDIR) -lwfdb $(LL) -L../lib
 
 # WFDBLIB is the name of the standard WFDB library.  In order to access it via
 # `-lwfdb', WFDBLIB should be `libwfdb.a'.
@@ -269,30 +269,22 @@ MFILES = Makefile
 
 # `make all': build applications
 all:	$(XFILES)
-	$(STRIP) $(XFILES)
+	sed -i s+/usr/local/database+$(DBDIR)+g $(SCRIPTS)
 
 # `make' or `make install':  build and install applications
-install:	all $(BINDIR) $(PSPDIR) scripts
-	rm -f pschart psfd pschart.exe psfd.exe
-	$(MAKE) pschart psfd	# be sure compiled-in paths are up-to-date
-	$(STRIP) pschart psfd
+install:	$(BINDIR) $(PSPDIR)
 	$(SETXPERMISSIONS) $(XFILES)
 	../install.sh $(BINDIR) $(XFILES)
 	cp $(PSFILES) $(PSPDIR)
 	cd $(PSPDIR); $(SETPERMISSIONS) $(PSFILES)
+	cp $(SCRIPTS) $(BINDIR)
+	cd $(BINDIR); $(SETPERMISSIONS) *setwfdb; $(SETXPERMISSIONS) pnwlogin
 
 # 'make collect': retrieve the installed applications
 collect:
 	../conf/collect.sh $(BINDIR) $(XFILES) $(SCRIPTS)
 	../conf/collect.sh $(PSPDIR) $(PSFILES)
 
-# `make scripts': install customized scripts for setting WFDB path
-scripts:
-	sed s+/usr/local/database+$(DBDIR)+g <setwfdb >$(BINDIR)/setwfdb
-	sed s+/usr/local/database+$(DBDIR)+g <cshsetwfdb >$(BINDIR)/cshsetwfdb
-	sed s+/usr/local/database+$(DBDIR)+g <pnwlogin >$(BINDIR)/pnwlogin
-	cd $(BINDIR); $(SETPERMISSIONS) *setwfdb; $(SETXPERMISSIONS) pnwlogin
-
 uninstall:
 	../uninstall.sh $(PSPDIR) $(PSFILES)
 	../uninstall.sh $(BINDIR) $(XFILES) $(SCRIPTS)
diff --git a/convert/Makefile b/convert/Makefile
index 3f60eac..8d44631 100644
--- a/convert/Makefile
+++ b/convert/Makefile
@@ -138,11 +138,11 @@ MFLAGS = -m32
 
 # CFLAGS is the set of C compiler options.  CFLAGS should always include
 # CCDEFS.
-CFLAGS = $(MFLAGS) -g -O $(CCDEFS) $(LC) -I$(INCDIR)
+CFLAGS = $(MFLAGS) -g -O $(CCDEFS) $(LC) -I$(INCDIR) -I..
 
 # LDFLAGS is appended to the C compiler command line to specify loading the
 # WFDB library.
-LDFLAGS = -L$(LIBDIR) -lwfdb $(LL)
+LDFLAGS = -L$(LIBDIR) -lwfdb $(LL) -L../lib
 
 # WFDBLIB is the name of the standard WFDB library.  In order to access it via
 # `-lwfdb', WFDBLIB should be `libwfdb.a'.
@@ -259,7 +259,7 @@ all:	$(XFILES)
 	$(STRIP) $(XFILES)
 
 # `make' or `make install':  build and install applications
-install:	$(BINDIR) all $(SCRIPTS)
+install:	$(BINDIR) $(SCRIPTS)
 	$(SETXPERMISSIONS) $(XFILES) $(SCRIPTS)
 	../install.sh $(BINDIR) $(XFILES) $(SCRIPTS)
 
diff --git a/lib/Makefile b/lib/Makefile
index c0c2a56..e3d1aae 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -217,11 +217,8 @@ all:	setup $(OFILES)
 
 # `make install':  install the WFDB library and headers
 install:	$(INCLUDES) $(LIBDIR)
-	$(MAKE) clean	    # force recompilation since config may have changed
-	$(MAKE) all
 	cp $(WFDBLIB) $(LIBDIR) 
 	$(SETLPERMISSIONS) $(LIBDIR)/$(WFDBLIB)
-	$(MAKE) lib-post-install 2>/dev/null
 
 # 'make collect':  retrieve the installed WFDB library and headers
 collect:
@@ -273,4 +270,4 @@ signal.o:	wfdb.h wfdblib.h signal.c
 calib.o:	wfdb.h wfdblib.h calib.c
 wfdbio.o:	wfdb.h wfdblib.h wfdbio.c
 	$(CC) $(CFLAGS) -DVERSION='"$(VERSION)"' -DCFLAGS='"-I$(INCDIR)"' \
-	  -DLDFLAGS='"-lwfdb"' -c wfdbio.c
\ No newline at end of file
+	  -DLDFLAGS='"-lwfdb"' -c wfdbio.c
diff --git a/psd/Makefile b/psd/Makefile
index adf8982..dfe3e47 100644
--- a/psd/Makefile
+++ b/psd/Makefile
@@ -248,25 +248,19 @@ SCRIPTS = hrfft hrlomb hrmem hrplot plot2d plot3d
 # `make all': build applications
 all:	$(XFILES)
 	$(STRIP) $(XFILES)
+	sed -i s+BINDIR+$(BINDIR)+g $(SCRIPTS)
 
 # `make' or `make install':  build and install applications
-install:	$(BINDIR) all scripts
+install:	$(BINDIR)
 	$(SETXPERMISSIONS) $(XFILES)
 	../install.sh $(BINDIR) $(XFILES)
+	cp $(SCRIPTS) $(BINDIR)
+	cd $(BINDIR); $(SETXPERMISSIONS) $(SCRIPTS)
 
 # 'make collect': retrieve the installed applications
 collect:
 	../conf/collect.sh $(BINDIR) $(XFILES) $(SCRIPTS)
 
-# `make scripts': customize and install scripts
-scripts:
-	sed s+BINDIR+$(BINDIR)+g <hrfft >$(BINDIR)/hrfft
-	sed s+BINDIR+$(BINDIR)+g <hrlomb >$(BINDIR)/hrlomb
-	sed s+BINDIR+$(BINDIR)+g <hrmem >$(BINDIR)/hrmem
-	sed s+BINDIR+$(BINDIR)+g <hrplot >$(BINDIR)/hrplot
-	cp plot2d plot3d $(BINDIR)
-	cd $(BINDIR); $(SETXPERMISSIONS) $(SCRIPTS)
-
 uninstall:
 	../uninstall.sh $(BINDIR) $(XFILES) $(SCRIPTS)
 
diff --git a/xml/Makefile b/xml/Makefile
index f0017d6..ce73fe9 100644
--- a/xml/Makefile
+++ b/xml/Makefile
@@ -253,7 +253,7 @@ all:	$(XFILES)
 	$(STRIP) $(XFILES)
 
 # `make' or `make install':  build and install applications
-install:	all $(BINDIR)
+install:	$(BINDIR)
 	$(SETXPERMISSIONS) $(XFILES)
 	../install.sh $(BINDIR) $(XFILES)
 
