diff -ur a/kernel/nv.c b/kernel/nv.c
--- a/kernel/nv.c	2013-08-01 05:19:26.000000000 +0430
+++ b/kernel/nv.c	2013-09-26 12:07:28.995029687 +0330
@@ -754,7 +754,7 @@
     NV_SPIN_LOCK_INIT(&km_lock);
 #endif
 
-    NV_KMEM_CACHE_CREATE(nv_stack_t_cache, "nv_stack_t", nv_stack_t);
+    NV_KMEM_CACHE_CREATE(nv_stack_t_cache, "nv_stack_t", nv_stack_t, SLAB_USERCOPY);
     if (nv_stack_t_cache == NULL)
     {
         nv_printf(NV_DBG_ERRORS, "NVRM: stack cache allocation failed!\n");
@@ -870,7 +870,7 @@
     nv->os_state = (void *) &nv_ctl_device;
     nv_lock_init_locks(nv);
 
-    NV_KMEM_CACHE_CREATE(nv_pte_t_cache, "nv_pte_t", nv_pte_t);
+    NV_KMEM_CACHE_CREATE(nv_pte_t_cache, "nv_pte_t", nv_pte_t, 0);
     if (nv_pte_t_cache == NULL)
     {
         rc = -ENOMEM;
@@ -879,7 +879,7 @@
     }
 
     NV_KMEM_CACHE_CREATE(nvidia_p2p_page_t_cache, "nvidia_p2p_page_t",
-            nvidia_p2p_page_t);
+            nvidia_p2p_page_t, 0);
     if (nvidia_p2p_page_t_cache == NULL)
     {
         rc = -ENOMEM;
diff -ur a/kernel/nv-linux.h b/kernel/nv-linux.h
--- a/kernel/nv-linux.h	2013-08-01 05:19:26.000000000 +0430
+++ b/kernel/nv-linux.h	2013-09-26 12:12:06.001020585 +0330
@@ -12,6 +12,15 @@
 #define _NV_LINUX_H_
 
 #include "nv.h"
+
+#ifndef SLAB_USERCOPY
+#ifdef CONFIG_PAX_USERCOPY_SLABS
+#define SLAB_USERCOPY		0x00000200UL
+#else
+#define SLAB_USERCOPY		0x00000000UL
+#endif
+#endif
+
 #include "conftest.h"
 
 #if !defined(NV_VMWARE)
@@ -759,16 +768,16 @@
 
 #if defined(NV_KMEM_CACHE_CREATE_PRESENT)
 #if (NV_KMEM_CACHE_CREATE_ARGUMENT_COUNT == 6)
-#define NV_KMEM_CACHE_CREATE(kmem_cache, name, type)            \
+#define NV_KMEM_CACHE_CREATE(kmem_cache, name, type, flags)            \
     {                                                           \
         kmem_cache = kmem_cache_create(name, sizeof(type),      \
-                        0, 0, NULL, NULL);                      \
+                        0, flags, NULL, NULL);                      \
     }
 #elif (NV_KMEM_CACHE_CREATE_ARGUMENT_COUNT == 5)
-#define NV_KMEM_CACHE_CREATE(kmem_cache, name, type)            \
+#define NV_KMEM_CACHE_CREATE(kmem_cache, name, type, flags)            \
     {                                                           \
         kmem_cache = kmem_cache_create(name, sizeof(type),      \
-                        0, 0, NULL);                            \
+                        0, flags, NULL);                            \
     }
 #else
 #error "NV_KMEM_CACHE_CREATE_ARGUMENT_COUNT value unrecognized!"
@@ -958,7 +967,7 @@
 #endif
 
 #if !defined(NV_VMWARE)
-#define NV_NUM_PHYSPAGES                num_physpages
+#define NV_NUM_PHYSPAGES                totalram_pages
 #define NV_GET_CURRENT_PROCESS()        current->tgid
 #define NV_IN_ATOMIC()                  in_atomic()
 #define NV_LOCAL_BH_DISABLE()           local_bh_disable()
