--- fuego-0.4/smartgame/SgSystem.h.old	2009-06-14 17:02:10.977134354 +0200
+++ fuego-0.4/smartgame/SgSystem.h	2009-06-14 17:06:47.753743909 +0200
@@ -57,6 +57,11 @@
 #define SG_ATTR_FLATTEN
 #endif
 
+#if defined(__GNUC__) && ! defined(__ICC) && \
+    (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 3))
+#include <cstdio>
+#endif
+
 //----------------------------------------------------------------------------
 
 #include <sys/types.h>
