diff --git a/src/include/etherfabric/base.h b/src/include/etherfabric/base.h
--- a/src/include/etherfabric/base.h
+++ b/src/include/etherfabric/base.h
@@ -25,6 +25,10 @@ extern "C" {
 #define CI_DEFAULT_NIC 0
 
 
+#define EF_VI_NIC_PAGE_SHIFT 12
+#define EF_VI_NIC_PAGE_SIZE  (1<<EF_VI_NIC_PAGE_SHIFT)
+
+
   /*! \i_ef_base An [ef_driver_handle] is needed to allocate resources. */
 #ifdef __KERNEL__
 typedef struct efhw_nic*   ef_driver_handle;
diff --git a/src/include/etherfabric/memreg.h b/src/include/etherfabric/memreg.h
--- a/src/include/etherfabric/memreg.h
+++ b/src/include/etherfabric/memreg.h
@@ -25,7 +25,6 @@
 #ifndef __EFAB_MEMREG_H__
 #define __EFAB_MEMREG_H__
 
-#include <ci/efhw/common.h>
 #include <etherfabric/base.h>
 
 #ifdef __cplusplus
@@ -63,8 +62,8 @@ extern int ef_memreg_free(ef_memreg*, ef
    */
 ef_vi_inline ef_addr ef_memreg_dma_addr(ef_memreg* mr, int offset)
 {
-	return mr->mr_dma_addrs[offset >> EFHW_NIC_PAGE_SHIFT] |
-		(offset & (EFHW_NIC_PAGE_SIZE - 1));
+	return mr->mr_dma_addrs[offset >> EF_VI_NIC_PAGE_SHIFT] |
+		(offset & (EF_VI_NIC_PAGE_SIZE - 1));
 }
 
 #ifdef __cplusplus
