diff -aur src/chrome/content/raismth/rai.js src.patched/chrome/content/raismth/rai.js
--- src/chrome/content/raismth/rai.js	2012-07-30 22:28:30.000000000 +0200
+++ src.patched/chrome/content/raismth/rai.js	2013-04-01 22:09:19.921885200 +0200
@@ -2,7 +2,7 @@
 myDump("[RAISMTH] rai module initialization\n");
 
 theApp.rai = {
-	injectSMTH : function(doc, indice) {		
+	injectSMTH : function(doc, indice) {
 		let ancora = doc.getElementById('raismthLogo');
 		ancora.removeChild(ancora.childNodes[1]);
 		let play = null;
@@ -98,6 +98,7 @@
 				ancora.appendChild(div);
 			}
 		}
+		
 	},
 
 	unInjectReplayClipControl : function(indice) {
@@ -477,7 +478,12 @@
 				}
 			}
 			*/
-		}	
+		}
+		var Silver = doc.getElementById('SilverlightControl') || doc.getElementById('silverlightControl');
+//		alert('Silver');
+		if(Silver && Silver.parentNode) {
+			Silver.parentNode.removeChild(Silver);
+		}
         },
 
         injectSilverlightPlayer : function(doc){
