--- ../pkg/usr/share/mullvad/client.conf.linux	2012-03-26 09:39:40.175823464 +0200
+++ usr/share/mullvad/client.conf.linux	2012-03-26 09:59:27.752723140 +0200
@@ -31,8 +31,8 @@
 script-security 2
 
 # Parses DHCP options from openvpn to update resolv.conf
-up /etc/openvpn/update-resolv-conf
-down /etc/openvpn/update-resolv-conf
+up /etc/openvpn/mullvad-up-resolv-conf
+down /etc/openvpn/mullvad-down-resolv-conf
 
 ping 10
 
