--- v4l2loopback-0.8.0/v4l2loopback.c	2013-12-10 18:18:27.000000000 +0100
+++ v4l2loopback-0.8.0.new/v4l2loopback.c	2015-01-16 09:16:18.133850241 +0100
@@ -483,9 +483,14 @@
 	struct v4l2_loopback_device *dev = NULL;
 	unsigned long curr = 0;
 
-	if (strict_strtoul(buf, 0, &curr))
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,18,0)
+	if (kstrtoul(buf, 0, &curr))
 		return -EINVAL;
+#else
 
+	if (strict_strtoul(buf, 0, &curr))
+		return -EINVAL;
+#endif
 	dev = v4l2loopback_cd2dev(cd);
 
 	if (dev->max_openers == curr)
