Index: src/main.C
===================================================================
RCS file: /schmorpforge/rxvt-unicode/src/main.C,v
retrieving revision 1.303
diff -u -r1.303 main.C
--- src/main.C	27 Jan 2008 22:48:33 -0000	1.303
+++ src/main.C	1 Feb 2008 16:27:59 -0000
@@ -644,6 +644,8 @@
 
   ncol = width / fwidth;
   nrow = height / fheight;
+  width = ncol * fwidth;
+  height = nrow * fheight;
 }
 
 /*----------------------------------------------------------------------*/
