*** Languages/LanguageKit/CodeGen/CodeGenLexicalScope.cpp	2009-10-15 13:59:31.000000000 +0300
--- Languages/LanguageKit/CodeGen/CodeGenLexicalScope.cpp	2009-10-18 23:17:56.000000000 +0300
***************
*** 2,8 ****
  #include "CodeGenBlock.h"
  #include "CodeGenModule.h"
  #include <llvm/Module.h>
! 
  
  static Function *getSmallIntModuleFunction(CodeGenModule *CGM, string name)
  {
--- 2,8 ----
  #include "CodeGenBlock.h"
  #include "CodeGenModule.h"
  #include <llvm/Module.h>
! #include <stdio.h>
  
  static Function *getSmallIntModuleFunction(CodeGenModule *CGM, string name)
  {
