*** Frameworks/EtoileFoundation/GNUmakefile	2009-10-20 19:46:41.000000000 +0300
--- Frameworks/EtoileFoundation/GNUmakefile.new	2009-10-20 19:47:15.000000000 +0300
***************
*** 86,92 ****
--- 86,94 ----
  EtoileFoundation_HEADER_FILES += NSFileManager+NameForTempFile.h
  endif
  
+ ifeq ($(GNUSTEP_TARGET_CPU), ix86)
  ADDITIONAL_OBJCFLAGS += -march=i586
+ endif
  
  include $(GNUSTEP_MAKEFILES)/aggregate.make
  -include ../../etoile.make
