--- a/Frameworks/MediaKit/MKMediaFile.h	2009-03-22 07:21:55.000000000 +0900
+++ b/Frameworks/MediaKit/MKMediaFile.h	2009-03-22 05:23:28.000000000 +0900
@@ -2,8 +2,8 @@
 // NOTE: Ugly hack to get rid of deprecation warnings on Ubuntu
 // See https://bugs.launchpad.net/ubuntu/+source/ffmpeg/+bug/122266
 #define attribute_deprecated
-#include <avcodec.h>
-#include <avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 
 @class NSURL;
 /**
