--- get_device.c	2007-03-26 13:51:31.000000000 +0200
+++ get_device.c	2008-06-22 13:16:13.000000000 +0200
@@ -459,7 +459,8 @@
 	xdevice_list = XListInputDevices(display, &nr_devices);
 
 	for(i = 0; i < nr_devices; i++) {
-		if (xdevice_list[i].use == IsXExtensionDevice) {
+		if (xdevice_list[i].use == IsXExtensionDevice ||
+            xdevice_list[i].use == IsXExtensionPointer || xdevice_list[i].use == IsXExtensionKeyboard) {
 			len = strlen(xdevice_list[i].name);
 			snprintf(read_buffer, MAXBUFFER, "%s", xdevice_list[i]
 									.name);
