From fba5b9ae7f5b81faeb6aa4a4a860ab48ea512c90 Mon Sep 17 00:00:00 2001
From: Kamal Mostafa <kamal@canonical.com>
Date: Wed, 10 Oct 2012 13:14:17 -0700
Subject: [PATCH] UBUNTU: SAUCE: input: Cypress PS/2 Trackpad list additional
 contributors

Signed-off-by: Kamal Mostafa <kamal@canonical.com>
Acked-by: Herton Krzesinski <herton.krzesinski@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/input/mouse/cypress_ps2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/input/mouse/cypress_ps2.c b/drivers/input/mouse/cypress_ps2.c
index 0ba679e..317aac9 100644
--- a/drivers/input/mouse/cypress_ps2.c
+++ b/drivers/input/mouse/cypress_ps2.c
@@ -3,6 +3,10 @@
  *
  * Copyright (c) 2012 Cypress Semiconductor Corporation.
  *
+ * Additional contributors include:
+ *   Kamal Mostafa <kamal@canonical.com>
+ *   Kyle Fazzari <git@status.e4ward.com>
+ *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License version 2 as published by
  * the Free Software Foundation.
-- 
1.8.0

