diff --git a/src/dfinstance.cpp b/src/dfinstance.cpp
--- a/src/dfinstance.cpp
+++ b/src/dfinstance.cpp
@@ -92,7 +92,7 @@
     search_paths << working_dir.path();
 
     QString subdir = LAYOUT_SUBDIR;
-    search_paths << QString("etc/memory_layouts/%1").arg(subdir);
+    search_paths << QString("/usr/share/splintermind-attributes/memory_layouts/%1").arg(subdir);
 
     TRACE << "Searching for MemoryLayout ini files in the following directories";
     foreach(QString path, search_paths) {
diff --git a/src/dwarftherapist.cpp b/src/dwarftherapist.cpp
--- a/src/dwarftherapist.cpp
+++ b/src/dwarftherapist.cpp
@@ -95,7 +95,7 @@
 
     //setup logging
     m_log_mgr = new LogManager(this);
-    TruncatingFileLogger *log = m_log_mgr->add_logger("log/run.log");
+    TruncatingFileLogger *log = m_log_mgr->add_logger(QDir::tempPath() + "/splintermind-attributes.log");
     if (log) {
         LogAppender *app = m_log_mgr->add_appender("core", log, LL_TRACE);
         if (app) {
diff --git a/src/gamedatareader.cpp b/src/gamedatareader.cpp
--- a/src/gamedatareader.cpp
+++ b/src/gamedatareader.cpp
@@ -37,8 +37,7 @@
 GameDataReader::GameDataReader(QObject *parent)
     : QObject(parent)
 {
-    QDir working_dir = QDir::current();
-    QString filename = working_dir.absoluteFilePath("etc/game_data.ini");
+    QString filename = "/usr/share/splintermind-attributes/game_data.ini";
     m_data_settings = new QSettings(filename, QSettings::IniFormat);
 
     QStringList labor_names;
