diff --git a/src/driver/0.20.x/Makefile.am b/src/driver/0.20.x/Makefile.am
index a1e7416..65cb006 100644
--- a/src/driver/0.20.x/Makefile.am
+++ b/src/driver/0.20.x/Makefile.am
@@ -8,6 +8,7 @@ AM_CFLAGS =  -g \
 	-DPACKAGE_ICONDIR=\"$(datadir)/valadoc/icons/\" \
 	-I ../../libvaladoc/ \
 	$(GLIB_CFLAGS) \
+	$(LIBGVC_CFLAGS) \
 	$(LIBGEE_CFLAGS) \
 	$(LIBVALA_0_20_X_CFLAGS) \
 	$(NULL)
diff --git a/src/driver/0.22.x/Makefile.am b/src/driver/0.22.x/Makefile.am
index 81fa37d..7659034 100644
--- a/src/driver/0.22.x/Makefile.am
+++ b/src/driver/0.22.x/Makefile.am
@@ -8,6 +8,7 @@ AM_CFLAGS =  -g \
 	-DPACKAGE_ICONDIR=\"$(datadir)/valadoc/icons/\" \
 	-I ../../libvaladoc/ \
 	$(GLIB_CFLAGS) \
+	$(LIBGVC_CFLAGS) \
 	$(LIBGEE_CFLAGS) \
 	$(LIBVALA_0_22_X_CFLAGS) \
 	$(NULL)
