--- ./install.sh 
+++ ./install-patched.sh 
@@ -20,31 +20,32 @@
 WEB_GID=`id -g daemon`
 RC_FILE=/etc/zce.rc
 ZEND_USER=zend
-if [ "$1" = '-g' ];then
-    PREFIX=/usr/local
-else
-    echo "Welcome to $PRODUCT_NAME installation script!"
-    echo "Please specify an installation path [/usr/local]:"
-    read -e PREFIX
-    if [ -z $PREFIX ];then
-        PREFIX="/usr/local"
-    fi
-fi
+#if [ "$1" = '-g' ];then
+#    PREFIX=/usr/local
+#else
+#    echo "Welcome to $PRODUCT_NAME installation script!"
+#    echo "Please specify an installation path [/usr/local]:"
+#    read -e PREFIX
+#    if [ -z $PREFIX ];then
+#        PREFIX="/usr/local"
+#    fi
+#fi
+PREFIX="/opt"
 if [ ! -d ${PREFIX} ];then
     mkdir -p ${PREFIX}
 fi
 ZCE_PREFIX=`readlink -f ${PREFIX}`/${TARGET_DIR}
-if [ -d ${ZCE_PREFIX}/apache2/htdocs ];then mv -f ${ZCE_PREFIX}/apache2/htdocs ${ZCE_PREFIX}/apache2/htdocs.bak-`date +%m%d%y`;fi
-if [ -d ${ZCE_PREFIX}/etc ];then mv -f ${ZCE_PREFIX}/etc ${ZCE_PREFIX}/etc.bak-`date +%m%d%y`;fi
-echo "Extracting files to ${ZCE_PREFIX}..."
-`dirname $0`/7z x -o$PREFIX -y `dirname $0`/$TARGET_DIR.7z 1>/dev/null
-if [ $? -ne 0 ];then 
-    echo "Failed to extract files.. "
-    exit 1
-fi
+#if [ -d ${ZCE_PREFIX}/apache2/htdocs ];then mv -f ${ZCE_PREFIX}/apache2/htdocs ${ZCE_PREFIX}/apache2/htdocs.bak-`date +%m%d%y`;fi
+#if [ -d ${ZCE_PREFIX}/etc ];then mv -f ${ZCE_PREFIX}/etc ${ZCE_PREFIX}/etc.bak-`date +%m%d%y`;fi
+#echo "Extracting files to ${ZCE_PREFIX}..."
+#`dirname $0`/7z x -o$PREFIX -y `dirname $0`/$TARGET_DIR.7z 1>/dev/null
+#if [ $? -ne 0 ];then 
+#    echo "Failed to extract files.. "
+#    exit 1
+#fi
 . ${ZCE_PREFIX}/bin/shell_functions.rc
 groupadd $ZEND_USER
-useradd -d ${ZCE_PREFIX}/gui/lighttpd -s /sbin/nologin -g $ZEND_USER $ZEND_USER 2>/dev/null
+useradd -r -m -d ${ZCE_PREFIX}/gui/lighttpd -s /sbin/nologin -g $ZEND_USER $ZEND_USER 2>/dev/null
 if ! id -g $ZEND_USER > /dev/null 2>&1 ;then
 	echo "Could not create the zend group. This is mandatory for $PRODUCT_NAME to function properly. Aborting installation.."
 	exit 1
@@ -135,13 +136,13 @@
 ln -sf ${ZCE_PREFIX}/apache2/logs/access_log ${ZCE_PREFIX}/var/log/access.log
 # in the event ld scandir is available
 if [ -d /etc/ld.so.conf.d ];then
-	echo "/usr/local/zend/lib" > /etc/ld.so.conf.d/zend_server.conf
+	echo "/opt/zend/lib" > /etc/ld.so.conf.d/zend_server.conf
 	if which ldconfig > /dev/null 2>&1;then
 		ldconfig
 	fi
 fi
-${ZCE_PREFIX}/bin/create_cert.sh
-${ZCE_PREFIX}/bin/zendctl.sh start
+#${ZCE_PREFIX}/bin/create_cert.sh
+#${ZCE_PREFIX}/bin/zendctl.sh start
 $ECHO_CMD "${OK_COLOR}========================== INSTALLATION SUMMARY ===================================================\n${T_RESET}"
 $ECHO_CMD "     $PRODUCT_NAME was installed to ${ZCE_PREFIX}"
 $ECHO_CMD "     The End User License Agreement [EULA] can be viewed under $ZCE_PREFIX/doc/EULA.txt"

