diff -ruN camlspotter-camlimages-e4e7c50d80c2.orig/src/gifread.c camlspotter-camlimages-e4e7c50d80c2/src/gifread.c
--- camlspotter-camlimages-e4e7c50d80c2.orig/src/gifread.c	2013-09-29 12:16:02.000000000 +0200
+++ camlspotter-camlimages-e4e7c50d80c2/src/gifread.c	2014-06-17 18:45:46.558007895 +0200
@@ -140,7 +140,8 @@
   GifFileType *GifFile;
   int i;
 
-  if((GifFile = DGifOpenFileName( String_val(name) )) == NULL){
+  int error;
+  if((GifFile = DGifOpenFileName( String_val(name), &error )) == NULL){
     failwith("DGifOpenFileName");
   }
 
@@ -161,7 +162,8 @@
      segmentation faults */
   ((GifFileType *)hdl)->Image.ColorMap = NULL; 
 
-  DGifCloseFile( (GifFileType *) hdl );
+  int error;
+  DGifCloseFile( (GifFileType *) hdl, &error );
   CAMLreturn0;
 }
 
diff -ruN camlspotter-camlimages-e4e7c50d80c2.orig/src/gifwrite.c camlspotter-camlimages-e4e7c50d80c2/src/gifwrite.c
--- camlspotter-camlimages-e4e7c50d80c2.orig/src/gifwrite.c	2013-09-29 12:16:02.000000000 +0200
+++ camlspotter-camlimages-e4e7c50d80c2/src/gifwrite.c	2014-06-17 18:46:33.379218078 +0200
@@ -51,7 +51,7 @@
 fprintf(stderr, "Creating map with length = %d ...\n", len);
 fflush(stderr);
 */
-  cmapobj = MakeMapObject( len, NULL );
+  cmapobj = GifMakeMapObject( len, NULL );
   for(i=0; i< len; i++){
     cmapobj->Colors[i].Red   = Int_val(Field(Field(cmap,i),0));
     cmapobj->Colors[i].Green = Int_val(Field(Field(cmap,i),1));
@@ -67,7 +67,8 @@
 
   GifFileType *GifFileOut;
 
-  if ((GifFileOut = EGifOpenFileName( String_val( name ), 0) )== NULL) {
+  int error;
+  if ((GifFileOut = EGifOpenFileName( String_val( name ), 0, &error) )== NULL) {
     failwith("EGifOpenFileName");
   }
   /* gcc -fwritable-strings is required to compile libungif */
@@ -87,7 +88,8 @@
      segmentation faults */
   ((GifFileType *)hdl)->Image.ColorMap = NULL; 
 
-  EGifCloseFile( (GifFileType *) hdl );
+  int error;
+  EGifCloseFile( (GifFileType *) hdl, &error );
   CAMLreturn0;
 }
 
