diff -ur slade.orig/Makefile slade/Makefile
--- slade.orig/Makefile	2010-06-23 15:53:29.000000000 +0200
+++ slade/Makefile	2010-06-23 15:55:39.000000000 +0200
@@ -2,8 +2,8 @@
 CXX=g++
 CXXFLAGS=-O2
 LDFLAGS=-lfreeimage
-CXXFLAGS+=`wx-config --version=2.9 --debug=no --cxxflags std,aui,gl,stc`
-LDFLAGS+=`wx-config --version=2.9 --debug=no --libs std,aui,gl,stc`
+CXXFLAGS+=`wx-config-2.9 --version=2.9 --debug=no --cxxflags std,aui,gl,stc`
+LDFLAGS+=`wx-config-2.9 --version=2.9 --debug=no --libs std,aui,gl,stc`
 EXE?=./dist/slade
 SRC=$(wildcard src/*.cpp)
 OBJ=$(patsubst %.cpp,%.o,${SRC})
