diff -ur slade.orig/src/CMakeLists.txt slade/src/CMakeLists.txt
--- slade.orig/src/CMakeLists.txt	2011-08-16 08:38:51.000000000 +0200
+++ slade/src/CMakeLists.txt	2011-08-16 09:19:11.000000000 +0200
@@ -1,4 +1,4 @@
-find_package(wxWidgets COMPONENTS aui gl stc richtext REQUIRED)
+find_package(wxWidgets COMPONENTS aui gl stc richtext adv html REQUIRED)
 include(${wxWidgets_USE_FILE})
 
 set(SFML_FIND_COMPONENTS system audio)
