diff -Nur awn-mintmenu.orig/mintMenuAwn.py awn-mintmenu/mintMenuAwn.py
--- awn-mintmenu.orig/mintMenuAwn.py	2012-04-15 22:08:48.358537402 -0400
+++ awn-mintmenu/mintMenuAwn.py	2012-04-15 22:09:34.125203267 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Copyright (C) 2010 neelance <mail@richard-musiol.de>
 #
 # This program is free software: you can redistribute it and/or modify
@@ -15,7 +15,7 @@
 
 from mintMenu import MenuWin
 from awn.extras import awnlib
-import gnomeapplet
+import mateapplet
 import subprocess
 import gtk
 
@@ -37,13 +37,13 @@
     def get_orient(self):
         orientation = self.applet.get_pos_type()
         if orientation == gtk.POS_BOTTOM:
-            return gnomeapplet.ORIENT_UP
+            return mateapplet.ORIENT_UP
         elif orientation == gtk.POS_TOP:
-            return gnomeapplet.ORIENT_DOWN
+            return mateapplet.ORIENT_DOWN
         elif orientation == gtk.POS_RIGHT:
-            return gnomeapplet.ORIENT_LEFT
+            return mateapplet.ORIENT_LEFT
         elif orientation == gtk.POS_LEFT:
-            return gnomeapplet.ORIENT_RIGHT
+            return mateapplet.ORIENT_RIGHT
         
     def set_size_request(self, width, height):
         pass
@@ -84,7 +84,7 @@
             self.applet.get_icon().set_is_active(False)
 
     def show_menu_editor_cb(self, widget):
-        menu_editor_apps = ("alacarte", "gmenu-simple-editor")
+        menu_editor_apps = ("mozo", "alacarte", "gmenu-simple-editor")
         for command in menu_editor_apps:
             try:
                 subprocess.Popen(command)
