diff -aur package.pristine/src/nemo-icon-view-container.c package.new/src/nemo-icon-view-container.c
--- package.pristine/src/nemo-icon-view-container.c	2014-09-07 22:44:15.539878405 +0200
+++ package.new/src/nemo-icon-view-container.c	2014-09-08 17:15:10.241981375 +0200
@@ -36,6 +36,7 @@
 
 #define ICON_TEXT_ATTRIBUTES_NUM_ITEMS		3
 #define ICON_TEXT_ATTRIBUTES_DEFAULT_TOKENS	"size,date_modified,type"
+#define DESKTOP_MAX_ICON_TEXT 			20
 
 G_DEFINE_TYPE (NemoIconViewContainer, nemo_icon_view_container, NEMO_TYPE_ICON_CONTAINER);
 
@@ -361,6 +362,20 @@
 		*editable_text = nemo_file_get_display_name (file);
 	}
 
+	/* If icon text is too long, strip it and add "..." */
+	if (strlen(*editable_text) > DESKTOP_MAX_ICON_TEXT)
+        {
+                char * new_str ;
+                if((new_str = malloc(DESKTOP_MAX_ICON_TEXT+3+1)) != NULL){
+                        new_str[0] = '\0';   // ensures the memory is an empty string
+                        strcat(new_str,strndup(*editable_text,DESKTOP_MAX_ICON_TEXT));
+                        strcat(new_str,"...");
+                        *editable_text = new_str;
+                }
+		/* TODO: Add tooltip with complete text */
+        }
+
+
 	if (!use_additional) {
 		return;
 	}
