diff -aur voipmonitor-2.0.ori//Makefile voipmonitor-2.0/Makefile
--- voipmonitor-2.0.ori//Makefile	2010-12-20 17:08:33.000000000 +0100
+++ voipmonitor-2.0/Makefile	2010-12-20 17:25:10.000000000 +0100
@@ -39,5 +39,5 @@
 	rm -f $(objects) voipmonitor gzstream/*.o libgzstream.a
 
 install: 
-	install voipmonitor /usr/local/sbin/
-
+	mkdir -p $(DESTDIR)/usr/local/sbin/
+	install voipmonitor $(DESTDIR)/usr/local/sbin/
