--- src/voipmonitor-10.1.8-src/Makefile.in	2014-10-29 22:15:53.996928352 +0200
+++ Makefile.in	2014-10-29 22:14:33.844470117 +0200
@@ -36,27 +36,27 @@
 
 #LDFLAGS := $(LDFLAGS) -static -static-libgcc
 
-shared: LIBS=-lpthread -lpcap -lz -lvorbis -lvorbisenc -logg -lodbc -lmysqlclient -lrt -lsnappy -lcurl -lssl -lcrypto -ljson -lssh -lssh_threads -lxml2 -lrrd
+shared: LIBS=-lpthread -lpcap -lz -lvorbis -lvorbisenc -logg -lodbc -lmysqlclient -lrt -lsnappy -lcurl -lssl -lcrypto -ljson-c -lssh -lssh_threads -lxml2 -lrrd
 shared: CFLAGS+= -DQUEUE_NONBLOCK2 -I/usr/local/include -I/usr/include/mysql/ -I/usr/local/mysql/include -Wall -I jitterbuffer/ -L/usr/local/lib/mysql -L/usr/local/mysql/lib -L/usr/lib/mysql -L/usr/local/lib/ -L/usr/lib64/mysql -fPIC -g3 -O2 
 shared: args = -g3 -Wall  -march=$(GCCARCH) -mtune=${GCCARCH} -O2 
 
-static: LIBS=-static -L/usr/local/mysql/lib -L/usr/local/lib/ -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -lodbc -lltdl -ldl -lrt -lz -lssh -lssh_threads -lcrypt -lm -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lpthread -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson -lrrd -lxml2 -lglib-2.0 -lz
+static: LIBS=-static -L/usr/local/mysql/lib -L/usr/local/lib/ -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -lodbc -lltdl -ldl -lrt -lz -lssh -lssh_threads -lcrypt -lm -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lpthread -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson-c -lrrd -lxml2 -lglib-2.0 -lz
 static: CFLAGS+= -DQUEUE_NONBLOCK2 -I/usr/local/include -I/usr/local/mysql/include/ -I/usr/include/mysql/ -Wall -I jitterbuffer/  -L/usr/local/lib/ -fPIC -O2
 static: args = -O2 -Wall  -march=$(GCCARCH) -mtune=${GCCARCH} 
 
-staticlibc: LIBS=-static -L/opt/libc/lib/ -L/usr/local/mysql/lib -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -lc_nonshared -lodbc -lltdl -ldl -lrt -lz -lssh -lssh_threads -lcurl -lcrypt -lm -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lpthread -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson  -lrrd -lxml2 -lglib-2.0 -lz
+staticlibc: LIBS=-static -L/opt/libc/lib/ -L/usr/local/mysql/lib -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -lc_nonshared -lodbc -lltdl -ldl -lrt -lz -lssh -lssh_threads -lcurl -lcrypt -lm -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lpthread -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson-c  -lrrd -lxml2 -lglib-2.0 -lz
 staticlibc: CFLAGS+= -DQUEUE_NONBLOCK2 -I/opt/libc/include -L/opt/libc/lib/ -I/usr/local/include -I /usr/local/mysql/include -I /usr/include/mysql/ -Wall -I jitterbuffer/  -L/usr/local/lib/ -fPIC -O2
 staticlibc: args = -O2 -Wall  -march=$(GCCARCH) -mtune=${GCCARCH}
 
-armv5shared: LIBS=-lpthread -lpcap -lz -lcurl -lssl -lcrypto -lvorbis -lvorbisenc -logg -lodbc -lmysqlclient -lrt -lsnappy -ljson -lssh -lssh_threads
+armv5shared: LIBS=-lpthread -lpcap -lz -lcurl -lssl -lcrypto -lvorbis -lvorbisenc -logg -lodbc -lmysqlclient -lrt -lsnappy -ljson-c -lssh -lssh_threads
 armv5shared: CFLAGS+= -DQUEUE_NONBLOCK2 -DPACKED -I /usr/local/mysql/include -I /usr/include/mysql/ -g3 -Wall -I jitterbuffer/  -L/usr/local/lib/ -L/usr/local/mysql/lib -L/usr/lib64/mysql -fPIC
 armv5shared: args = -g3 -Wall  
 
-armv5static: LIBS=-static -L /usr/local/mysql/lib -L/usr/lib/mysql -lodbc -lltdl -ldl -lrt -lz -lcrypt -lm -lssh -lssh_threads -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -L/usr/lib64/mysql/ -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson  -lrrd -lxml2 -lglib-2.0 -lz
+armv5static: LIBS=-static -L /usr/local/mysql/lib -L/usr/lib/mysql -lodbc -lltdl -ldl -lrt -lz -lcrypt -lm -lssh -lssh_threads -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -L/usr/lib64/mysql/ -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson-c  -lrrd -lxml2 -lglib-2.0 -lz
 armv5static: CFLAGS+= -DQUEUE_NONBLOCK2 -DPACKED -I /usrl/local/mysql/include -I /usr/include/mysql/ -g3 -Wall -I jitterbuffer/  -L/usr/local/lib/ -L/usr/lib64/mysql -fPIC
 armv5static: args = -g3 -Wall 
 
-core2: LIBS=-static -L/usr/local/lib -L/usr/local/mysql/lib -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -lodbc -lltdl -ldl -lrt -lz -lssh -lssh_threads -lcrypt -lm -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lpthread -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson  -lrrd -lxml2 -lglib-2.0 -lz
+core2: LIBS=-static -L/usr/local/lib -L/usr/local/mysql/lib -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -lodbc -lltdl -ldl -lrt -lz -lssh -lssh_threads -lcrypt -lm -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lpthread -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson-c  -lrrd -lxml2 -lglib-2.0 -lz
 core2: CFLAGS+= -DQUEUE_NONBLOCK2 -I /usr/local/mysql/include -I /usr/include/mysql/ -Wall -I jitterbuffer/  -L/usr/local/lib/ -fPIC -O2 -g3
 core2: args = -O2 -Wall  -march=core2 -mtune=core2 -g3
 
@@ -65,11 +65,11 @@
 core2cust: STARTFILES = $(GLIBCDIR)/crt1.o $(GLIBCDIR)/crti.o $(shell gcc --print-file-name=crtbegin.o)
 core2cust: ENDFILES = $(shell gcc --print-file-name=crtend.o) $(GLIBCDIR)/crtn.o
 core2cust: LIBGROUP = -Wl,--start-group $(GLIBCDIR)/libc.a -lgcc -lgcc_eh -Wl,--end-group
-core2cust: LIBS = -L/opt/libc/lib -L/usr/local/mysql/lib -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -L/usr/local/lib -lodbc -lltdl -ldl -lrt -lz -lssh -lssh_threads -lcrypt -lm -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson -lrrd -lxml2 -lglib-2.0 -lz
+core2cust: LIBS = -L/opt/libc/lib -L/usr/local/mysql/lib -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -L/usr/local/lib -lodbc -lltdl -ldl -lrt -lz -lssh -lssh_threads -lcrypt -lm -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson-c -lrrd -lxml2 -lglib-2.0 -lz
 core2cust: CFLAGS+= -DQUEUE_NONBLOCK2 -I /opt/libc/include -I /usr/local/mysql/include/ -I /usr/include/mysql/ -Wall -I jitterbuffer/  -L/usr/local/lib/ -fPIC -O2 -g3
 core2cust: args = -O2 -Wall  -march=core2 -mtune=core2 -g3
 
-core2debug: LIBS=-static -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -lodbc -lltdl -ldl -lrt -lz -lcrypt -lm -lssh -lssh_threads -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lpthread -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson  -lrrd -lxml2 -lglib-2.0 -lz
+core2debug: LIBS=-static -L/usr/lib/mysql -L/usr/lib64/mysql -L/usr/lib/mysql -lodbc -lltdl -ldl -lrt -lz -lcrypt -lm -lssh -lssh_threads -lcurl -lssl -lcrypto -static-libstdc++ -static-libgcc -lpcap -L/usr/lib/mysql -lpthread -lmysqlclient -lpthread -lz -lc -lvorbis -lvorbisenc -logg -lrt -lsnappy -ljson-c  -lrrd -lxml2 -lglib-2.0 -lz
 core2debug: CFLAGS+= -DQUEUE_NONBLOCK2 -I /usr/include/mysql/ -Wall -I jitterbuffer/  -L/usr/local/lib/ -fPIC -O2 -g3
 core2debug: args = -O2 -Wall  -march=core2 -mtune=core2 -g3
 
