diff -rupN a/main.cpp b/main.cpp
--- a/main.cpp	2006-10-05 18:24:10.000000000 -0500
+++ b/main.cpp	2015-02-10 16:53:52.690455319 -0600
@@ -64,7 +64,7 @@ int main(int argc, char** argv)
   cerr << "loading pos_models";
   for (int i = 0; i < 16; i++) {
     char buf[1000];
-    sprintf(buf, "./models_medline/model.bidir.%d", i);
+    sprintf(buf, "/usr/share/geniatagger/models_medline/model.bidir.%d", i);
     vme[i].load_from_file(buf);
     cerr << ".";
   }
@@ -74,7 +74,7 @@ int main(int argc, char** argv)
   vector<ME_Model> vme_chunking(16);
   for (int i = 0; i < 8; i +=2 ) {
     char buf[1000];
-    sprintf(buf, "./models_chunking/model.bidir.%d", i);
+    sprintf(buf, "/usr/share/geniatagger/models_chunking/model.bidir.%d", i);
     vme_chunking[i].load_from_file(buf);
     cerr << ".";
   }
diff -rupN a/morph.cpp b/morph.cpp
--- a/morph.cpp	2006-02-20 05:07:25.000000000 -0600
+++ b/morph.cpp	2015-02-10 16:54:35.496271333 -0600
@@ -4,6 +4,7 @@
 #include <sstream>
 #include <map>
 #include <set>
+#include <stdlib.h>
 
 using namespace std;
 
@@ -19,14 +20,14 @@ struct MorphDic
   MorphDic() {}
   void Init() {
     cerr << "loading morphdic...";
-    LoadEx("./morphdic/noun.exc", nounex);
-    LoadEx("./morphdic/verb.exc", verbex);
-    LoadEx("./morphdic/adj.exc", adjex);
-    LoadEx("./morphdic/adv.exc", advex);
-    LoadIdx("./morphdic/noun.dic", noundic);
-    LoadIdx("./morphdic/verb.dic", verbdic);
-    LoadIdx("./morphdic/adj.dic", adjdic);
-    cerr << "done." << endl;
+    LoadEx("/usr/share/geniatagger/morphdic/noun.exc", nounex);
+    LoadEx("/usr/share/geniatagger/morphdic/verb.exc", verbex);
+    LoadEx("/usr/share/geniatagger/morphdic/adj.exc", adjex);
+    LoadEx("/usr/share/geniatagger/morphdic/adv.exc", advex);
+    LoadIdx("/usr/share/geniatagger/morphdic/noun.dic", noundic);
+    LoadIdx("/usr/share/geniatagger/morphdic/verb.dic", verbdic);
+    LoadIdx("/usr/share/geniatagger/morphdic/adj.dic", adjdic);
+    cerr << "done" << endl;
   }
   void LoadEx(const string & filename, map<string, string> & exmap) {
     ifstream ifile(filename.c_str());
diff -rupN a/namedentity.cpp b/namedentity.cpp
--- a/namedentity.cpp	2007-04-16 07:30:08.000000000 -0500
+++ b/namedentity.cpp	2015-02-10 16:55:06.488988868 -0600
@@ -384,8 +384,8 @@ ME_Model ne_model;
 
 void load_ne_models()
 {
-  string model_file = "./models_named_entity/model001";
-  string wordinfo_file = "./models_named_entity/word_info";
+  string model_file = "/usr/share/geniatagger/models_named_entity/model001";
+  string wordinfo_file = "/usr/share/geniatagger/models_named_entity/word_info";
 
   cerr << "loading named_entity_models.";
   ne_model.load_from_file(model_file);
