diff -Naur syasokoban-2.0.1/src/djDesktop.cpp syasokoban-2.0.1-new/src/djDesktop.cpp
--- syasokoban-2.0.1/src/djDesktop.cpp	2007-07-21 22:22:10.000000000 +0200
+++ syasokoban-2.0.1-new/src/djDesktop.cpp	2008-05-26 13:49:39.000000000 +0200
@@ -15,7 +15,7 @@
 
 std::string GetDataPath()
 {
-	return std::string("data/");
+	return std::string("/usr/share/syasokoban/");
 }
 
 CdjDesktop::CdjDesktop()
diff -Naur syasokoban-2.0.1/src/djSettings.cpp syasokoban-2.0.1-new/src/djSettings.cpp
--- syasokoban-2.0.1/src/djSettings.cpp	2007-07-21 22:22:10.000000000 +0200
+++ syasokoban-2.0.1-new/src/djSettings.cpp	2008-05-26 13:50:54.000000000 +0200
@@ -7,6 +7,8 @@
 #include "djSettings.h"
 #include "djUtils.h"
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 
 char g_szConfigFile[2048] = { 0 };
 
