diff -aur qgnokii-0.4.2.firstpatch/main.cc qgnokii-0.4.2.new/main.cc
--- qgnokii-0.4.2.firstpatch/main.cc	2009-09-25 20:24:19.000000000 +0200
+++ qgnokii-0.4.2.new/main.cc	2009-09-25 20:33:17.000000000 +0200
@@ -23,11 +23,13 @@
 #include "mainform.h"
 #include <qmessagebox.h>
 #include <qtextedit.h>
-#include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <stdio.h>
+#include <cstdlib>
+
 int main(int argc, char *argv[])
 {
 	QApplication app(argc, argv);
diff -aur qgnokii-0.4.2.firstpatch/qgnokiid.c qgnokii-0.4.2.new/qgnokiid.c
--- qgnokii-0.4.2.firstpatch/qgnokiid.c	2009-09-25 20:24:54.000000000 +0200
+++ qgnokii-0.4.2.new/qgnokiid.c	2009-09-25 20:24:08.000000000 +0200
@@ -24,7 +24,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <gnokii.h>
-#include <string.h>
+#include <strings.h>
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <limits.h>
diff -aur qgnokii-0.4.2.firstpatch/settingsform.ui.h qgnokii-0.4.2.new/settingsform.ui.h
--- qgnokii-0.4.2.firstpatch/settingsform.ui.h	2009-09-25 20:24:19.000000000 +0200
+++ qgnokii-0.4.2.new/settingsform.ui.h	2009-09-25 20:34:21.000000000 +0200
@@ -18,6 +18,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <qfile.h>
+#include <cstdlib>
 
 void settingsForm::init()
 {
