--- Makefile.orig	2005-11-17 00:08:30.000000000 +0100
+++ Makefile	2005-11-17 00:10:24.000000000 +0100
@@ -16,12 +16,12 @@
 	tar zcvf netenv-$(VERSION).tar.gz $(FILES) $(DOC_FILES)
 
 install:
-	install -p -o root -g root -m 744 netenv /usr/sbin
-	install -p -o root -g root -m 755 trpnc /usr/bin
-	install -p -o root -g root -m 755 -d /etc/netenv
-	install -p -o root -g root -m 755 -d /usr/share/doc/packages/netenv
-	install -p -o root -g root -m 644 doc/*.html doc/NEWS doc/lulu* doc/*png /usr/share/doc/packages/netenv
-	install -p -o root -g root -m 744 doc/netenv_setup* doc/edit* /usr/share/doc/packages/netenv
+	install -p -o root -g root -m 744 netenv $(DESTDIR)/usr/sbin
+	install -p -o root -g root -m 755 trpnc $(DESTDIR)/usr/bin
+	install -p -o root -g root -m 755 -d $(DESTDIR)/etc/netenv
+	install -p -o root -g root -m 755 -d $(DESTDIR)/usr/share/doc/packages/netenv
+	install -p -o root -g root -m 644 doc/*.html doc/NEWS doc/lulu* doc/*png $(DESTDIR)/usr/share/doc/packages/netenv
+	install -p -o root -g root -m 744 doc/netenv_setup* doc/edit* $(DESTDIR)/usr/share/doc/packages/netenv
 
 config:
 	@./config_netenv
