--- common.c.orig	2008-01-16 19:08:00.000000000 +0100
+++ common.c	2008-10-19 22:59:55.000000000 +0200
@@ -1265,6 +1265,7 @@
 
 char    com_cachedir[MAX_OSPATH];
 char    com_gamedir[MAX_OSPATH];
+char	com_basedir[MAX_OSPATH];
 
 typedef struct searchpath_s
 {
@@ -1707,25 +1708,30 @@
 COM_AddGameDirectory -- johnfitz -- modified based on topaz's tutorial
 =================
 */
-void COM_AddGameDirectory (char *dir)
+//void COM_AddGameDirectory (char *dir)
+static void
+COM_AddGameDirectory (char *base, char *dir)
 {
 	int i;
 	searchpath_t *search;
 	pack_t *pak;
 	char pakfile[MAX_OSPATH];
 
-	strcpy (com_gamedir, dir);
+	if (!base)
+		return;
+
+	strcpy(com_gamedir, va("%s/%s", base, dir));
 
 	// add the directory to the search path
 	search = Z_Malloc(sizeof(searchpath_t));
-	strcpy (search->filename, dir);
+	strcpy (search->filename, com_gamedir);
 	search->next = com_searchpaths;
 	com_searchpaths = search;
 
 	// add any pak files in the format pak0.pak pak1.pak, ...
 	for (i = 0; ; i++)
 	{
-		sprintf (pakfile, "%s/pak%i.pak", dir, i);
+		sprintf (pakfile, "%s/pak%i.pak", com_gamedir, i);
 		pak = COM_LoadPackFile (pakfile);
 		if (!pak)
 			break;
@@ -1743,21 +1749,23 @@
 */
 void COM_InitFilesystem () //johnfitz -- modified based on topaz's tutorial
 {
+	char *home;
 	int i, j;
-	char basedir[MAX_OSPATH];
 	searchpath_t *search;
 
+	home = getenv("HOME");
+
 	i = COM_CheckParm ("-basedir");
 	if (i && i < com_argc-1)
-		strcpy (basedir, com_argv[i+1]);
+		strcpy (com_basedir, com_argv[i+1]);
 	else
-		strcpy (basedir, host_parms.basedir);
+		strcpy (com_basedir, host_parms.basedir);
 
-	j = strlen (basedir);
+	j = strlen (com_basedir);
 	if (j > 0)
 	{
-		if ((basedir[j-1] == '\\') || (basedir[j-1] == '/'))
-		basedir[j-1] = 0;
+		if ((com_basedir[j-1] == '\\') || (com_basedir[j-1] == '/'))
+			com_basedir[j-1] = 0;
 	}
 
 	i = COM_CheckParm ("-cachedir");
@@ -1774,19 +1782,30 @@
 		com_cachedir[0] = 0;
 
 	// start up with GAMENAME by default (id1)
-	COM_AddGameDirectory (va("%s/"GAMENAME, basedir) );
-	strcpy (com_gamedir, va("%s/"GAMENAME, basedir));
+	COM_AddGameDirectory(com_basedir, GAMENAME);
+	COM_AddGameDirectory(home, ".fitzquake/" GAMENAME);
 
-	if (COM_CheckParm ("-rogue"))
-		COM_AddGameDirectory (va("%s/rogue", basedir) );
-	if (COM_CheckParm ("-hipnotic"))
-		COM_AddGameDirectory (va("%s/hipnotic", basedir) );
+	if (COM_CheckParm ("-rogue")) {
+		COM_AddGameDirectory(com_basedir, "rogue");
+		COM_AddGameDirectory(home, ".fitzquake/rogue");
+	}
+	if (COM_CheckParm ("-hipnotic")) {
+		COM_AddGameDirectory(com_basedir, "hipnotic");
+		COM_AddGameDirectory(home, ".fitzquake/hipnotic");
+	}
 
 	i = COM_CheckParm ("-game");
 	if (i && i < com_argc-1)
 	{
 		com_modified = true;
-		COM_AddGameDirectory (va("%s/%s", basedir, com_argv[i+1]));
+		COM_AddGameDirectory(com_basedir, com_argv[i + 1]);
+		COM_AddGameDirectory(home, va(".fitzquake/%s", com_argv[i + 1]));
+	}
+
+	/* If home is available, create the game directory */
+	if (home) {
+		COM_CreatePath(com_gamedir);
+		Sys_mkdir(com_gamedir);
 	}
 
 	i = COM_CheckParm ("-path");
--- common.h.orig	2008-01-16 19:08:00.000000000 +0100
+++ common.h	2008-10-19 22:49:45.000000000 +0200
@@ -167,6 +167,7 @@
 extern int com_filesize;
 struct cache_user_s;
 
+extern	char	com_basedir[MAX_OSPATH];
 extern	char	com_gamedir[MAX_OSPATH];
 
 void COM_WriteFile (char *filename, void *data, int len);
