--- a/Makefile	2011-10-22 18:56:55.000000000 +0200
+++ b/Makefile	2013-04-22 16:38:31.388576172 +0200
@@ -120,13 +120,20 @@
 EXTRA_CFLAGS       += -I$(src)/src/wl/sys -I$(src)/src/wl/phy
 #EXTRA_CFLAGS       += -DBCMDBG_ASSERT
 
+ifeq ($(KVER),)
+	KVER = $(shell uname -r)
+endif
+
+KDIR     = /lib/modules/$(KVER)/build
+PWD      = $(shell pwd)
+
 EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped
 
 all:
-	KBUILD_NOPEDANTIC=1 make -C /lib/modules/`uname -r`/build M=`pwd`
+	KBUILD_NOPEDANTIC=1 make -C $(KDIR) M=$(PWD)
 
 clean:
-	KBUILD_NOPEDANTIC=1 make -C /lib/modules/`uname -r`/build M=`pwd` clean
+	KBUILD_NOPEDANTIC=1 make -C $(KDIR) M=$(PWD) clean
 
 install:
-	install -D -m 755 wl.ko /lib/modules/`uname -r`/kernel/drivers/net/wireless/wl.ko
+	install -D -m 755 wl.ko /lib/modules/$(KVER)/kernel/drivers/net/wireless/wl.ko
--- a/src/wl/sys/wl_linux.c	2011-10-22 18:56:55.000000000 +0200
+++ b/src/wl/sys/wl_linux.c	2013-04-22 16:38:24.891526375 +0200
@@ -185,6 +185,8 @@
 static void wl_report_radio_state(wl_info_t *wl);
 #endif
 
+MODULE_LICENSE("MIXED/Proprietary");
+
 static struct pci_device_id wl_id_table[] = {
 	{ PCI_VENDOR_ID_BROADCOM, 0x4311, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, 
 	{ PCI_VENDOR_ID_BROADCOM, 0x4312, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 }, 
@@ -256,7 +258,7 @@
 #define quote_str(s) to_str(s)
 
 #ifndef BRCM_WLAN_IFNAME
-#define BRCM_WLAN_IFNAME eth%d
+#define BRCM_WLAN_IFNAME wlan%d
 #endif
 
 static char name[IFNAMSIZ] = quote_str(BRCM_WLAN_IFNAME);
