diff --git a/Makefile b/Makefile
index c6d317c..0ef23d2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
-CFLAGS += `pkg-config bluez --cflags --libs` -Wall -g $(shell dpkg-buildflags --get CFLAGS)
-LDFLAGS += `pkg-config bluez --libs` -g $(shell dpkg-buildflags --get LDFLAGS)
+CFLAGS += `pkg-config bluez --cflags --libs` -Wall -g
+LDFLAGS += `pkg-config bluez --libs` -g
 
 all:
 	$(CC) $(CFLAGS) $(LDFLAGS) novena-disable-ssp.c -o novena-disable-ssp
diff --git a/swap.h b/swap.h
index cce3cd8..4e5a9ab 100644
--- a/swap.h
+++ b/swap.h
@@ -82,6 +82,7 @@ do {						\
 #define newa(t, n) ((t*) alloca(sizeof(t)*(n)))
 #define malloc0(n) (calloc((n), 1))
 
+#if 0
 static inline void bswap_128(const void *src, void *dst)
 {
 	const uint8_t *s = src;
@@ -91,6 +92,7 @@ static inline void bswap_128(const void *src, void *dst)
 	for (i = 0; i < 16; i++)
 		d[15 - i] = s[i];
 }
+#endif
 
 static inline uint16_t get_le16(const void *ptr)
 {
