--- a/index.theme	2014-11-11 05:07:06.000000000 -0600
+++ b/index.theme	2015-02-01 13:42:13.280404967 -0600
@@ -1,7 +1,7 @@
 [Icon Theme]
-Name=Ultra-Flat-Icons
-Comment=An icon theme for Linux desktops, the set is inspired by the latest flat design trend.
-Inherits=
+Name=Ultra Flat Icons Orange Bright Dark
+Comment=A dark icon theme for Linux desktops, the set is inspired by the latest flat design trend.
+Inherits=UltraFlatIconsOrangeBright
 
 Example=x-directory-normal
 
