diff a/InstallFromWorkspace.cobra b/InstallFromWorkspace.cobra
--- a/InstallFromWorkspace.cobra
+++ b/InstallFromWorkspace.cobra
@@ -158,9 +158,9 @@ class InstallFromWorkspace
 		.verifyNewlyBuiltCobra
 		.copyFilesToInstallDirectory
 		.verifyNewlyInstalledCobra
-		.cleanUpWorkspace
-		.installInPath
-		.changeDirectoryInstruction
+		# .cleanUpWorkspace
+		# .installInPath
+		# .changeDirectoryInstruction
 		print 'Visit http://cobra-language.com/ for discussion, wiki, samples, irc and more.'
 		print
 		print 'Success!'
@@ -701,7 +701,7 @@ class InstallFromWorkspace
 
 	def installLibrary(name as String, flags as String, files as String)
 		.buildLibrary(name, flags, files as String)
-		.installLibraryToGAC(name)
+		# .installLibraryToGAC(name)
 		.verifyGacInstallation(name)
 
 	def buildLibrary(name as String, flags as String, files as String)
