diff -Naur openwbem-3.2.2/openwbem.m4 openwbem-3.2.2.new/openwbem.m4
--- openwbem-3.2.2/openwbem.m4	2004-10-21 16:49:24.000000000 +0200
+++ openwbem-3.2.2.new/openwbem.m4	2007-08-20 16:15:22.000000000 +0200
@@ -35,7 +35,7 @@
 dnl   Usage:   CHECK_OPENWBEM([REQUIRED-VERSION
 dnl                      [,ACTION-IF-FOUND[,ACTION-IF-NOT-FOUND]]])
 dnl
-AC_DEFUN(CHECK_OPENWBEM,
+AC_DEFUN([CHECK_OPENWBEM],
 [
 OPENWBEM_CLIENT_LIBS="-lopenwbem -lowclient -lowxml -lowhttpcommon -lowhttpclient"
 OPENWBEM_PROVIDER_LIBS="-lopenwbem -lowprovider -lowcppprovifc"
