--- mplayer-tools-7.8.20110210/Makefile
+++ mplayer-tools-7.8.20110210/Makefile
@@ -4,94 +4,96 @@
 default:
 	@echo Type \`make install\' to install the package.

+DESTDIR=
+
 install:
-	install lib/MPlayerTools.xpm /usr/share/icons
-	install lib/MPlayerTools.xpm /usr/share/pixmaps
-	install lib/MPlayerTools.desktop /usr/share/applications
-	rm -f /etc/profile.d/mplayer-tools.sh || true
-	install lib/mplayer-tools.sh /etc
-	mv -f /etc/profile /etc/profile~ && grep -v '. /etc/mplayer-tools.sh' /etc/profile~ >/etc/profile
-	echo . /etc/mplayer-tools.sh >>/etc/profile
-	install bin/mcatcher /usr/bin
-	install bin/mlist /usr/bin
-	install bin/mlists /usr/bin
-	install bin/mlistd /usr/bin
-	install bin/mstart /usr/bin
-	install bin/mstrip /usr/bin
-	install bin/msave /usr/bin
-	install bin/mcancel /usr/bin
-	install bin/mswitch /usr/bin
-	install bin/mecho /usr/bin
-	install bin/mloop /usr/bin
-	install bin/mhelp /usr/bin
-	install bin/mgui /usr/bin
-	install bin/mguicmd /usr/bin
-	install bin/msum /usr/bin
-	install bin/msynclog /usr/bin
-	install bin/msize /usr/bin
-	install bin/rsize /usr/bin
-	install bin/mcounter /usr/bin
-	install bin/srtadjust /usr/bin
-	install bin/srtshift /usr/bin
-	install bin/sub2srt /usr/bin
-	install bin/tv /usr/bin
-	install bin/tvcancel /usr/bin
-	install bin/tvchannels.cgi /usr/bin
-	install bin/tvconf /usr/bin
-	install bin/tvrecord /usr/bin
-	install bin/tvserver /usr/bin
-	install bin/tvtimeshift /usr/bin
-	install bin/tvtsinfo /usr/bin
-	install bin/tvvolume /usr/bin
-	mkdir -p /usr/lib/mplayer-tools || true
-	install -m 0644 lib/autoloader.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/beautify.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/cancelproc.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/childprocess.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/codecName.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/dcop.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/displaytime.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/dumpobject.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/dvdVolumeID.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/existingDir.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/extractseries.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/findSub.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/getconffilename.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/getFileType.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/getLocalUUID.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/getModes.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/getMPlayer.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/gettermsize.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/guiCommand.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/isLiveStream.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/isMediaFile.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/isPlayable.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/makefullfilename.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/marksaving.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/movedown.png /usr/lib/mplayer-tools
-	install -m 0644 lib/moveup.png /usr/lib/mplayer-tools
-	install -m 0644 lib/mplayer-tools.sh /usr/lib/mplayer-tools
-	install -m 0644 lib/MPlayerTools.desktop /usr/lib/mplayer-tools
-	install -m 0644 lib/MPlayerTools.xpm /usr/lib/mplayer-tools
-	install -m 0644 lib/MPlayerToolsListInput.pm /usr/lib/mplayer-tools
-	install -m 0644 lib/MPlayerToolsOSD.pm /usr/lib/mplayer-tools
-	install -m 0644 lib/MPlayerToolsPlaylist.pm /usr/lib/mplayer-tools
-	install -m 0644 lib/MPlayerToolsSearch.pm /usr/lib/mplayer-tools
-	install -m 0644 lib/MPlayerToolsServer.pm /usr/lib/mplayer-tools
-	install -m 0644 lib/MPlayerToolsTextInput.pm /usr/lib/mplayer-tools
-	install -m 0644 lib/MPlayerToolsTV.desktop /usr/lib/mplayer-tools
-	install -m 0644 lib/overrideName.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/parseCUE.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/PlaylistFormats.pm /usr/lib/mplayer-tools
-	install -m 0644 lib/printplayerinfo.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/readcounters.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/readmconffile.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/readmovieconffile.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/removecmfromlist.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/setStatus.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/setupGUIConnection.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/undefobject.pl /usr/lib/mplayer-tools
-	install -m 0644 lib/writemovieconffile.pl /usr/lib/mplayer-tools
+	install lib/MPlayerTools.xpm ${DESTDIR}/usr/share/icons
+	install lib/MPlayerTools.xpm ${DESTDIR}/usr/share/pixmaps/MPlayerTools.xpm
+	install lib/MPlayerTools.desktop ${DESTDIR}/usr/share/applications/MPlayerTools.desktop
+	#rm -f /etc/profile.d/mplayer-tools.sh || true
+	install lib/mplayer-tools.sh ${DESTDIR}/etc/profile.d
+	#grep -v '. /etc/mplayer-tools.sh' ${DESTDIR}/etc/profile~ >${DESTDIR}/etc/profile
+	#echo . ${DESTDIR}/etc/mplayer-tools.sh >>${DESTDIR}/etc/profile
+	install bin/mcatcher ${DESTDIR}/usr/bin
+	install bin/mlist ${DESTDIR}/usr/bin
+	install bin/mlists ${DESTDIR}/usr/bin
+	install bin/mlistd ${DESTDIR}/usr/bin
+	install bin/mstart ${DESTDIR}/usr/bin
+	install bin/mstrip ${DESTDIR}/usr/bin
+	install bin/msave ${DESTDIR}/usr/bin
+	install bin/mcancel ${DESTDIR}/usr/bin
+	install bin/mswitch ${DESTDIR}/usr/bin
+	install bin/mecho ${DESTDIR}/usr/bin
+	install bin/mloop ${DESTDIR}/usr/bin
+	install bin/mhelp ${DESTDIR}/usr/bin
+	install bin/mgui ${DESTDIR}/usr/bin
+	install bin/mguicmd ${DESTDIR}/usr/bin
+	install bin/msum ${DESTDIR}/usr/bin
+	install bin/msynclog ${DESTDIR}/usr/bin
+	install bin/msize ${DESTDIR}/usr/bin
+	install bin/rsize ${DESTDIR}/usr/bin
+	install bin/mcounter ${DESTDIR}/usr/bin
+	install bin/srtadjust ${DESTDIR}/usr/bin
+	install bin/srtshift ${DESTDIR}/usr/bin
+	install bin/sub2srt ${DESTDIR}/usr/bin
+	install bin/tv ${DESTDIR}/usr/bin
+	install bin/tvcancel ${DESTDIR}/usr/bin
+	install bin/tvchannels.cgi ${DESTDIR}/usr/bin
+	install bin/tvconf ${DESTDIR}/usr/bin
+	install bin/tvrecord ${DESTDIR}/usr/bin
+	install bin/tvserver ${DESTDIR}/usr/bin
+	install bin/tvtimeshift ${DESTDIR}/usr/bin
+	install bin/tvtsinfo ${DESTDIR}/usr/bin
+	install bin/tvvolume ${DESTDIR}/usr/bin
+	mkdir -p ${DESTDIR}/usr/lib/mplayer-tools || true
+	install -m 0644 lib/autoloader.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/beautify.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/cancelproc.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/childprocess.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/codecName.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/dcop.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/displaytime.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/dumpobject.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/dvdVolumeID.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/existingDir.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/extractseries.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/findSub.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/getconffilename.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/getFileType.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/getLocalUUID.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/getModes.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/getMPlayer.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/gettermsize.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/guiCommand.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/isLiveStream.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/isMediaFile.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/isPlayable.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/makefullfilename.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/marksaving.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/movedown.png ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/moveup.png ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/mplayer-tools.sh ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/MPlayerTools.desktop ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/MPlayerTools.xpm ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/MPlayerToolsListInput.pm ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/MPlayerToolsOSD.pm ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/MPlayerToolsPlaylist.pm ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/MPlayerToolsSearch.pm ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/MPlayerToolsServer.pm ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/MPlayerToolsTextInput.pm ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/MPlayerToolsTV.desktop ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/overrideName.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/parseCUE.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/PlaylistFormats.pm ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/printplayerinfo.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/readcounters.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/readmconffile.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/readmovieconffile.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/removecmfromlist.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/setStatus.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/setupGUIConnection.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/undefobject.pl ${DESTDIR}/usr/lib/mplayer-tools
+	install -m 0644 lib/writemovieconffile.pl ${DESTDIR}/usr/lib/mplayer-tools
 	@echo
 	@echo Remember to log out completely before you start using the software.
