--- sigrok-cli.c.orig	2013-12-07 23:40:33.824813569 +0200
+++ sigrok-cli.c	2013-12-07 23:40:49.251836738 +0200
@@ -17,7 +17,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <sigrokdecode.h> /* First, so we avoid a _POSIX_C_SOURCE warning. */
+#include <libsigrokdecode/libsigrokdecode.h> /* First, so we avoid a _POSIX_C_SOURCE warning. */
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
