diff --git a/host/python/CMakeLists.txt b/host/python/CMakeLists.txt
index 7b9f5ce..5f5a8eb 100644
--- a/host/python/CMakeLists.txt
+++ b/host/python/CMakeLists.txt
@@ -20,7 +20,7 @@
 project(ubertooth-python)
 
 # Check for python installation
-find_package(PythonInterp)
+find_package(PythonInterp 2)
 if(NOT PYTHONINTERP_FOUND)
 	return()
 else()
diff --git a/host/python/extcap/btle-extcap.py b/host/python/extcap/btle-extcap.py
index b41d7c3..1b8cfee 100755
--- a/host/python/extcap/btle-extcap.py
+++ b/host/python/extcap/btle-extcap.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright 2013 Mike Ryan
 #
diff --git a/host/python/specan_ui/CMakeLists.txt b/host/python/specan_ui/CMakeLists.txt
index c365bd5..6b4e476 100644
--- a/host/python/specan_ui/CMakeLists.txt
+++ b/host/python/specan_ui/CMakeLists.txt
@@ -31,7 +31,7 @@ add_custom_command(OUTPUT ${OUTPUT}/timestamp
                    DEPENDS ${DEPS})
 
 add_custom_target(specan ALL DEPENDS ${OUTPUT}/timestamp)
-install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} build -b ${OUTPUT} install)")
+install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} build -b ${OUTPUT} install --prefix=${CMAKE_INSTALL_PREFIX})")
 
 add_custom_target(ubertooth-specan-ui ALL DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/ubertooth-specan-ui)
 install(PROGRAMS ubertooth-specan-ui DESTINATION ${INSTALL_DEFAULT_BINDIR})
diff --git a/host/python/specan_ui/build/lib.linux-x86_64-2.7/specan/Ubertooth.py b/host/python/specan_ui/build/lib.linux-x86_64-2.7/specan/Ubertooth.py
index 7a18173..daaf808 100644
--- a/host/python/specan_ui/build/lib.linux-x86_64-2.7/specan/Ubertooth.py
+++ b/host/python/specan_ui/build/lib.linux-x86_64-2.7/specan/Ubertooth.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2011 Jared Boone
 #
diff --git a/host/python/specan_ui/setup.py b/host/python/specan_ui/setup.py
index f93f0a2..e731a21 100644
--- a/host/python/specan_ui/setup.py
+++ b/host/python/specan_ui/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 dfu setup
 
diff --git a/host/python/specan_ui/setup.py.in b/host/python/specan_ui/setup.py.in
index 7051944..78c1e5f 100644
--- a/host/python/specan_ui/setup.py.in
+++ b/host/python/specan_ui/setup.py.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 dfu setup
 
diff --git a/host/python/specan_ui/specan/Ubertooth.py b/host/python/specan_ui/specan/Ubertooth.py
index 7a18173..daaf808 100644
--- a/host/python/specan_ui/specan/Ubertooth.py
+++ b/host/python/specan_ui/specan/Ubertooth.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2011 Jared Boone
 #
diff --git a/host/python/specan_ui/ubertooth-specan-ui b/host/python/specan_ui/ubertooth-specan-ui
index c3c351f..ceab3b3 100755
--- a/host/python/specan_ui/ubertooth-specan-ui
+++ b/host/python/specan_ui/ubertooth-specan-ui
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2011 Jared Boone
 #
diff --git a/host/python/usb_dfu/CMakeLists.txt b/host/python/usb_dfu/CMakeLists.txt
index 229d7bc..913abdc 100644
--- a/host/python/usb_dfu/CMakeLists.txt
+++ b/host/python/usb_dfu/CMakeLists.txt
@@ -31,7 +31,7 @@ add_custom_command(OUTPUT ${OUTPUT}/timestamp
                    DEPENDS ${DEPS})
 
 add_custom_target(dfu ALL DEPENDS ${OUTPUT}/timestamp)
-install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} build -b ${OUTPUT} install)")
+install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} build -b ${OUTPUT} install --prefix=${CMAKE_INSTALL_PREFIX})")
 
 add_custom_target(ubertooth-dfu ALL DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/ubertooth-dfu)
 install(PROGRAMS ubertooth-dfu DESTINATION ${INSTALL_DEFAULT_BINDIR})
diff --git a/host/python/usb_dfu/Makefile b/host/python/usb_dfu/Makefile
index 019b89a..e14af17 100644
--- a/host/python/usb_dfu/Makefile
+++ b/host/python/usb_dfu/Makefile
@@ -18,7 +18,7 @@
 # Boston, MA 02110-1301, USA.
 
 INSTALL ?= /usr/bin/install
-PYTHON ?= /usr/bin/env python
+PYTHON ?= /usr/bin/env python2
 
 INSTALL_DIR ?= /usr/bin
 
diff --git a/host/python/usb_dfu/dfu/dfu.py b/host/python/usb_dfu/dfu/dfu.py
index 19bdbff..0a6c75a 100644
--- a/host/python/usb_dfu/dfu/dfu.py
+++ b/host/python/usb_dfu/dfu/dfu.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2010-2012 Michael Ossmann, Jared Boone, Dominic Spill
 # Copyright 2010 TheSeven
@@ -325,4 +325,4 @@ class DfuSuffix(object):
         crc = self.crc32(firmware)
         firmware += struct.pack('<I', crc)
     
-        return firmware
\ No newline at end of file
+        return firmware
diff --git a/host/python/usb_dfu/setup.py b/host/python/usb_dfu/setup.py
index 311cd55..5e2bacf 100755
--- a/host/python/usb_dfu/setup.py
+++ b/host/python/usb_dfu/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 dfu setup
 
diff --git a/host/python/usb_dfu/setup.py.in b/host/python/usb_dfu/setup.py.in
index af4b73a..1254d30 100755
--- a/host/python/usb_dfu/setup.py.in
+++ b/host/python/usb_dfu/setup.py.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 dfu setup
 
diff --git a/host/python/usb_dfu/ubertooth-dfu b/host/python/usb_dfu/ubertooth-dfu
index 5df7a47..263ae81 100755
--- a/host/python/usb_dfu/ubertooth-dfu
+++ b/host/python/usb_dfu/ubertooth-dfu
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2010, 2011 Michael Ossmann
 #
