--- ./src/shared/sleep-config.c.orig	2013-09-18 01:55:37.241977274 +0400
+++ ./src/shared/sleep-config.c	2013-11-21 02:27:57.608209252 +0400
@@ -206,7 +206,7 @@
                 if (!d)
                         return -ENOMEM;
 
-                if (!streq(type, "partition")) {
+                if (!streq(type, "partition") && !streq(type, "file")) {
                         log_debug("Partition %s has type %s, ignoring.", d, type);
                         continue;
                 }
