--- suri_reloader.old	2015-02-09 23:09:57.413333333 +0100
+++ suri_reloader	2015-02-09 23:06:26.059999999 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 """
 Copyright(C) 2014, Stamus Networks
 Written by Eric Leblond <eleblond@stamus-networks.com>
@@ -55,12 +55,12 @@
 
 def SuriReload(reload = False):
     if reload:
-        if subprocess.call(['service', 'suricata', 'reload']):
+        if subprocess.call(['systemctl', 'reload', 'suricata']):
             logging.error("Unable to reload suricata")
         else:
             logging.info("Reloaded suricata")
     else:
-        if subprocess.call(['service', 'suricata', 'restart']):
+        if subprocess.call(['systemctl', 'restart', 'suricata']):
             logging.error("Unable to restart suricata")
         else:
             logging.info("Restarted suricata")
