From df9373dd4da502356548015d7cba4d0c6897ca1a Mon Sep 17 00:00:00 2001
From: Matthias Larisch <mail@matthias-larisch.de>
Date: Mon, 24 Mar 2014 15:57:08 +0100
Subject: [PATCH 1/2] Fix Makefile to support automake DESTDIR

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 9d92efb..88ab836 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-INSTALL_PLUGINS_DIR	=	/usr/local/lib/ladspa/
+INSTALL_PLUGINS_DIR	=	$(DESTDIR)/usr/lib/ladspa/
 
 CC		=	gcc
 LD		= ld
-- 
1.9.0

