From e4f08689198fbc8d3ca79459b01aeb4eca7937aa Mon Sep 17 00:00:00 2001
From: Matthias Larisch <mail@matthias-larisch.de>
Date: Mon, 24 Mar 2014 16:01:54 +0100
Subject: [PATCH 2/2] fix folder generation

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 88ab836..beea9bc 100644
--- a/Makefile
+++ b/Makefile
@@ -26,7 +26,7 @@ all: $(PLUGINS)
 	$(LD) $(LDFLAGS) -o $@ $<
 
 install: targets
-	test -d $(INSTALL_PLUGINS_DIR) || mkdir $(INSTALL_PLUGINS_DIR)
+	test -d $(INSTALL_PLUGINS_DIR) || mkdir -p $(INSTALL_PLUGINS_DIR)
 	cp *.so $(INSTALL_PLUGINS_DIR)
 
 targets:	$(PLUGINS)
-- 
1.9.0

