diff --git a/src/scripting/lua/core.c b/src/scripting/lua/core.c
index 00303f9..a0a9982 100644
--- a/src/scripting/lua/core.c
+++ b/src/scripting/lua/core.c
@@ -245,7 +245,7 @@ l_execute(LS)
 static int
 l_tmpname(LS)
 {
-	unsigned char *fn = tempnam(NULL, "elinks");
+	int *fn = mkstemp("elinks");
 
 	if (fn) {
 		lua_pushstring(S, fn);
diff --git a/src/session/download.c b/src/session/download.c
index 837f14b..12d8575 100644
--- a/src/session/download.c
+++ b/src/session/download.c
@@ -1005,7 +1005,7 @@ get_temp_name(struct uri *uri)
 	 * We use tempnam() here, which is unsafe (race condition), for now.
 	 * This should be changed at some time, but it needs an in-depth work
 	 * of whole download code. --Zas */
-	unsigned char *nm = tempnam(NULL, ELINKS_TEMPNAME_PREFIX);
+	int *nm = mkstemp(ELINKS_TEMPNAME_PREFIX);
 
 	if (!nm) return NULL;
 
