diff -aur menumaker-0.99.7/MenuMaker/__init__.py menumaker-0.99.7-2/MenuMaker/__init__.py
--- menumaker-0.99.7/MenuMaker/__init__.py	2005-12-13 01:01:42.000000000 -0800
+++ menumaker-0.99.7-2/MenuMaker/__init__.py	2010-10-13 18:14:07.000000000 -0700
@@ -36,7 +36,9 @@
 	Kw("Deskmenu")		: KwS("Deskmenu"),
 	Kw("Xfce4")			: KwS("Xfce", "Xfce4"),
 	Kw("OpenBox3")		: KwS("OpenBox", "OpenBox3"),
-	Kw("WindowMaker")	: KwS("WindowMaker", "WMaker")
+	Kw("WindowMaker")	: KwS("WindowMaker", "WMaker"),
+	Kw("Compiz")		: KwS("Compiz", "Compiz-deskmenu"),
+	Kw("MyGTKMenu")		: KwS("MyGTKMenu")
 }
 
 
@@ -47,12 +49,13 @@
 terms = [
 	(T.xterm, KwS("Xterm")),
 	(T._xfterm, KwS("Xfterm")),
-	(T.xvt, KwS("Rxvt", "Xvt")),
+	(T.xvt, KwS("Rxvt", "Xvt", "Urxvt")),
 	(T.aterm, KwS("Aterm")),
 	(T.wterm, KwS("Wterm")),
 	(T.Eterm, KwS("Eterm")),
 	(T._gterm, KwS("GNOME-terminal", "Gterm")),
-	(T._konsole, KwS("Konsole"))
+	(T._konsole, KwS("Konsole")),
+	(T.roxterm, KwS("ROXterm"))
 ]
 
 
diff -aur menumaker-0.99.7/Prophet/Legacy/Shell.py menumaker-0.99.7-2/Prophet/Legacy/Shell.py
--- menumaker-0.99.7/Prophet/Legacy/Shell.py	2005-12-13 00:50:13.000000000 -0800
+++ menumaker-0.99.7-2/Prophet/Legacy/Shell.py	2010-10-13 18:31:31.000000000 -0700
@@ -128,7 +128,10 @@
 	comment = "RISCOS-like file manager"
 	keywords = KwS(FileManager, Shell)
 
-
+class roxterm(_Terminal) :
+	name = "ROXterm"
+	comment = "X terminal emulator for ROX"
+	keywords = KwS(TerminalEmulator)
 
 
 class tuxcmd(_App, _X11App) :
@@ -200,7 +203,7 @@
 class xvt(_Terminal) : # FIXME : do we need a separate entry for rxvt?
 	name = "Rxvt"
 	comment = "X terminal emulator"
-	exes = ["rxvt", "xvt"]
+	exes = ["rxvt", "xvt", "urxvt"]
 	keywords = KwS(TerminalEmulator)
