diff -Naur trn-4.0-test77/Speller.SH trn-4.0-test77-aur/Speller.SH
--- trn-4.0-test77/Speller.SH	2000-05-24 22:11:41.000000000 -0700
+++ trn-4.0-test77-aur/Speller.SH	2010-09-22 18:10:34.000000000 -0700
@@ -16,7 +16,7 @@
 # what pager you use--if you have kernal paging use cat
 pager="\${PAGER-$pager}"
 # either the ispell program or "none"
-ispell=$ispell_prg
+ispell="$ispell_prg"
 ispell_options=$ispell_options
 test=${test-test}
 sed=${sed-sed}
