Index: src/plugins/metadata_libavformat.c
===================================================================
--- src/plugins/metadata_libavformat.c	(revision 692)
+++ src/plugins/metadata_libavformat.c	(working copy)
@@ -76,8 +76,8 @@
     return -1;
   }
 
-	if(av_find_stream_info(*ctx) < 0) {
-		av_close_input_file(*ctx);
+	if(avformat_find_stream_info(*ctx, NULL) < 0) {
+		avformat_close_input(ctx);
 	  return -1;
 	}
 	
@@ -173,7 +173,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "title", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "title", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "title", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: TITLE: %s\n", tag->value);
     	set_value(metadata->title, sizeof(metadata->title), tag->value);      
@@ -184,7 +184,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "genre", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "genre", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "genre", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: GENRE: %s\n", tag->value);
     	set_value(metadata->genre, sizeof(metadata->genre), tag->value);      
@@ -194,7 +194,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "track", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "track", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "track", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: TRACK: %s\n", tag->value);
     	//set_value(&metadata->genre, tag->value);      
@@ -204,7 +204,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "comment", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "comment", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "comment", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: comment: %s\n", tag->value);
     	set_value(metadata->description, sizeof(metadata->description), tag->value);      
@@ -214,7 +214,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "composer", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "composer", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "composer", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: composer: %s\n", tag->value);
     	set_value(metadata->composer, sizeof(metadata->composer), tag->value);      
@@ -224,7 +224,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "date", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "date", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "date", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: date: %s\n", tag->value);
     	set_value(metadata->date, sizeof(metadata->date), tag->value);      
@@ -234,7 +234,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "language", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "language", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "language", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: language: %s\n", tag->value);
     	set_value(metadata->language, sizeof(metadata->language), tag->value);      
@@ -244,7 +244,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "performer", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "performer", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "performer", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: performer: %s\n", tag->value);
     	//set_value(&metadata->language, tag->value);      
@@ -254,7 +254,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "publisher", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "publisher", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "publisher", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: publisher: %s\n", tag->value);
     	set_value(metadata->publisher, sizeof(metadata->publisher), tag->value);      
@@ -264,7 +264,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "show", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "show", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "show", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: show: %s\n", tag->value);
     	set_value(metadata->series_title, sizeof(metadata->series_title), tag->value);      
@@ -274,7 +274,7 @@
     #if LIBAVFORMAT_VERSION_MAJOR < 53
     if((tag = av_metadata_get(ctx->metadata, "episode_id", NULL, AV_METADATA_IGNORE_SUFFIX))) {
     #elif LIBAVFORMAT_VERSION_MAJOR >= 53
-    if((tag = av_dict_get(ctx->metadata, "episode_id", NULL, AV_METADATA_IGNORE_SUFFIX))) {
+    if((tag = av_dict_get(ctx->metadata, "episode_id", NULL, AV_DICT_IGNORE_SUFFIX))) {
     #endif
       printf("libavformat metadata: episode_id: %s\n", tag->value);
     	set_value(metadata->program_title, sizeof(metadata->program_title), tag->value);      
@@ -289,7 +289,7 @@
 
 void fuppes_metadata_file_close(plugin_info* plugin)
 {
-	av_close_input_file(plugin->user_data);
+	avformat_close_input((AVFormatContext**)&plugin->user_data);
 	plugin->user_data = NULL;
 }
 
