--- massmol-1.5.2.orig/src/linux-menu/massmol.desktop	2011-05-05 01:35:01.000000000 +0600
+++ massmol-1.5.2/src/linux-menu/massmol.desktop	2012-05-09 09:05:38.763825660 +0600
@@ -1,14 +1,13 @@
 [Desktop Entry]
 Version=1.0
-Encoding=UTF-8
 Name=Molar molecular mass
 Name[fr]=Masse molaire moléculaire
 Comment=Gives the molar mass and other information about a given molecule.
 Comment[fr]=Donne la masse molaire et d'autres informations sur une molécule donnée.
 Exec=massmol
-Icon=massmol-icon.png
+Icon=massmol-icon
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Categories=Application;Education;Science;
+Categories=Education;Science;
 StartupNotify=true
