diff -aur package.pristine/konsole-4.9.0/src/MainWindow.cpp package.new/konsole-4.9.0/src/MainWindow.cpp
--- package.pristine/konsole-4.9.0/src/MainWindow.cpp	2012-08-17 00:04:58.006724734 +0300
+++ package.new/konsole-4.9.0/src/MainWindow.cpp	2012-08-17 00:07:38.002524629 +0300
@@ -56,6 +56,12 @@
 #include "settings/GeneralSettings.h"
 #include "settings/TabBarSettings.h"
 
+#ifdef Q_WS_X11
+#include <QX11Info>
+#include <X11/Xlib.h>
+#include <X11/Xatom.h>
+#endif
+
 using namespace Konsole;
 
 MainWindow::MainWindow()
@@ -70,6 +76,12 @@
     // But it is mostly annoying to have translucent menubar and tabbar
     setAttribute(Qt::WA_NoSystemBackground, false);
 
+#ifdef Q_WS_X11
+    Atom net_wm_blur_region = XInternAtom(QX11Info::display(), "_KDE_NET_WM_BLUR_BEHIND_REGION", False);
+    unsigned long region[] = { 0 };
+    XChangeProperty(QX11Info::display(), winId(), net_wm_blur_region, XA_CARDINAL, 32, PropModeReplace, reinterpret_cast<const unsigned char*>(&region), 1);
+#endif
+
     // create actions for menus
     setupActions();
 
