diff -Nurp a/src/synaptics.c b/src/synaptics.c
--- a/src/synaptics.c	2011-03-15 02:20:32.000000000 +0100
+++ b/src/synaptics.c	2013-02-20 15:27:54.185771002 +0100
@@ -696,7 +696,7 @@ static void set_input_params(struct inpu
 
 	if (SYN_CAP_ADV_GESTURE(priv->ext_cap_0c)) {
 		__set_bit(INPUT_PROP_SEMI_MT, dev->propbit);
-		input_mt_init_slots(dev, 2);
+		input_mt_init_slots(dev, 2, 0);
 		input_set_abs_params(dev, ABS_MT_POSITION_X, XMIN_NOMINAL,
 				     priv->x_max ?: XMAX_NOMINAL, 0, 0);
 		input_set_abs_params(dev, ABS_MT_POSITION_Y, YMIN_NOMINAL,
