--- usr/bin/ccnet-web.sh	2013-02-22 14:38:16.359074067 +0100
+++ usr/bin/ccnet-web.sh	2013-02-22 14:38:39.796355366 +0100
@@ -10,7 +10,7 @@
     if [ -f /tmp/ccnet-web.pid ]; then
         kill -9 `cat /tmp/ccnet-web.pid`
     fi
-    /usr/bin/python main.py 127.0.0.1:13420 &
+    /usr/bin/python2 main.py 127.0.0.1:13420 &
     echo $! > /tmp/ccnet-web.pid
 elif [ $1 == "stop" ]; then
     kill -9 `cat /tmp/ccnet-web.pid`
