diff --git a/src/bdsim.cc b/src/bdsim.cc
index c36b297..ab7044b 100644
--- src/bdsim.cc
+++ src/bdsim.cc
@@ -59,6 +59,7 @@ const int DEBUG = 0;
 
 #include "parser/gmad.h"  // GMAD parser
 
+#include <ctime>
 
 
 
@@ -325,7 +326,6 @@ int main(int argc,char** argv) {
 
   // get the seed from options if positive, else
   // user time as a seed
-#include <ctime>
   if(BDSGlobals->GetRandomSeed()>=0)
     seed = BDSGlobals->GetRandomSeed();
   else
