--- pthreads-w32-20110511/tests/GNUmakefile.orig	2011-05-12 11:43:08.000000000 +0300
+++ pthreads-w32-20110511/tests/GNUmakefile	2011-05-12 11:43:36.000000000 +0300
@@ -379,20 +379,20 @@ valid2.pass: valid1.pass
 
 sizes.pass: sizes.exe
 	@ $(ECHO) Running $*
-	./$< > SIZES.$(TEST)
+	wine $< > SIZES.$(TEST)
 	@ $(CAT) SIZES.$(TEST)
 	@ $(ECHO) Passed
 	@ $(TOUCH) $@
 
 %.pass: %.exe
 	@ $(ECHO) Running $*
-	./$*
+	wine $*
 	@ $(ECHO) Passed
 	@ $(TOUCH) $@
 
 %.bench: $(LIB) $(DLL) $(HDR) $(QAPC) $(XXLIBS) %.exe
 	@ $(ECHO) Running $*
-	./$*
+	wine $*
 	@ $(ECHO) Done
 	@ $(TOUCH) $@
 
