--- openboxlogout/openboxlogout.py.old	2009-04-20 16:36:47.000000000 +0300
+++ openboxlogout/openboxlogout.py	2009-04-20 16:45:59.000000000 +0300
@@ -302,24 +302,22 @@
         if (data == 'cancel'):
             self.quit()
         elif (data == 'logout'):
-            os.system('openbox --exit')
+            os.system('gnome-session-save --logout')
+            self.quit()
         elif (data == 'restart'):
             self.dbus_powermanagement.Reboot()
-            #os.system('gdm-control --reboot && openbox --exit')
+            self.quit()
         elif (data == 'shutdown'):
             self.dbus_powermanagement.Shutdown()
-            #os.system('gdm-control --shutdown && openbox --exit')
+            self.quit()
         elif (data == 'suspend'):
             self.dbus_powermanagement.Suspend(0)
-            #os.system('dbus-send --system --print-reply --dest=org.freedesktop.Hal /org/freedesktop/Hal/devices/computer org.freedesktop.Hal.Device.SystemPowerManagement.Suspend int32:0')
             self.quit()
         elif (data == 'hibernate'):
             self.dbus_powermanagement.Hiberate()
-            #os.system('dbus-send --system --print-reply --dest=org.freedesktop.Hal /org/freedesktop/Hal/devices/computer org.freedesktop.Hal.Device.SystemPowerManagement.Hibernate')         
             self.quit()
         elif (data == 'safesuspend'):
             self.dbus_powermanagement.SuspendHybrid(0)
-            #os.system('dbus-send --system --print-reply --dest=org.freedesktop.Hal /org/freedesktop/Hal/devices/computer org.freedesktop.Hal.Device.SystemPowerManagement.SuspendHybrid int32:0')         
             self.quit()
         elif (data == 'lock'):
             os.system('gnome-screensaver-command -l')
