diff --git a/CMakeTests/FindOculusSDK.cmake b/CMakeTests/FindOculusSDK.cmake
index afd00d2..3e7dcca 100644
--- a/CMakeTests/FindOculusSDK.cmake
+++ b/CMakeTests/FindOculusSDK.cmake
@@ -9,17 +9,17 @@ if (NOT OCULUSSDK_FOUND)
 
 	find_path(OCULUSSDK_INCLUDE_DIR NAMES OVR.h
 		PATHS
-		/usr/include/ovr-0.4.3/LibOVR/Include
+		/usr/include/ovr-0.4.4/LibOVR/Include
 		/usr/include
-		/usr/local/include/ovr-0.4.3/LibOVR/Include
+		/usr/local/include/ovr-0.4.4/LibOVR/Include
 		/usr/local/include
 	)
 
 	find_path(OCULUSSDK_SRC_DIR NAMES OVR_CAPI_GL.h
 		PATHS
-		/usr/include/ovr-0.4.3/LibOVR/Src
+		/usr/include/ovr-0.4.4/LibOVR/Src
 		/usr/include
-		/usr/local/include/ovr-0.4.3/LibOVR/Src
+		/usr/local/include/ovr-0.4.4/LibOVR/Src
 		/usr/local/include
 	)
 
diff --git a/Source/Core/VideoBackends/OGL/FramebufferManager.cpp b/Source/Core/VideoBackends/OGL/FramebufferManager.cpp
index 9ca3b1b..cc7378d 100644
--- a/Source/Core/VideoBackends/OGL/FramebufferManager.cpp
+++ b/Source/Core/VideoBackends/OGL/FramebufferManager.cpp
@@ -6,6 +6,8 @@
 #include "VideoBackends/OGL/GLInterface/WGL.h"
 
 #include "VideoCommon/VR920.h"
+#else
+#include "VideoBackends/OGL/GLInterface/GLX.h"
 #endif
 
 #include "Common/CommonFuncs.h"
@@ -617,7 +619,7 @@ void FramebufferManager::ConfigureRift()
 #endif
 #else
 	cfg.OGL.Disp = glXGetCurrentDisplay();
-	cfg.OGL.Win = glXGetCurrentDrawable();
+	//cfg.OGL.Win = glXGetCurrentDrawable();
 #endif
 	int caps = 0;
 	if (g_Config.bChromatic)
