--- vim-peaksea	2011-01-11 08:27:00.385775617 -0500
+++ vim-peaksea-new	2011-01-11 08:50:46.555779033 -0500
@@ -201,7 +201,7 @@
     hi StatusLineNC	ctermfg=fg	ctermbg=250	cterm=NONE
     hi VertSplit	ctermfg=fg	ctermbg=250	cterm=NONE
     hi Underlined	ctermfg=62	ctermbg=NONE	cterm=underline
-    hi Ignore		ctermfg=bg	ctermbg=NONE
+    "hi Ignore		ctermfg=bg	ctermbg=NONE
     " NOTE THIS IS IN THE WARM SECTION
     if v:version >= 700
       if has('spell')
