--- configure_2010-05-31_203051	2010-05-31 20:30:51.000000000 +0200
+++ configure	2010-05-31 20:36:32.000000000 +0200
@@ -4711,14 +4711,15 @@
 
 
 if test "x$WANTS_GTK" = "xyes"; then
-        AM_PATH_GTK_2_0(2.6.0, CFLAGS="$CFLAGS $GTK_CFLAGS"
-                           LIBS="$LIBS $GTK_LIBS -lm",
-                           { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Building without GTK2 display support" >&5
-$as_echo "$as_me: WARNING: Building without GTK2 display support" >&2;}
-
-$as_echo "#define NO_GTK 1" >>confdefs.h
-
-                           GTK_OBJ="")
+  true
+#        AM_PATH_GTK_2_0(2.6.0, CFLAGS="$CFLAGS $GTK_CFLAGS"
+#                           LIBS="$LIBS $GTK_LIBS -lm",
+#                           { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Building without GTK2 display support" >&5
+#$as_echo "$as_me: WARNING: Building without GTK2 display support" >&2;}
+#
+#$as_echo "#define NO_GTK 1" >>confdefs.h
+#
+#                           GTK_OBJ="")
 else
 	$as_echo "#define NO_GTK 1" >>confdefs.h
 
