diff -cr trayfreq-0.2.x-dev2/src/bat_tray.c trayfreq-0.2.x-dev2-fr/src/bat_tray.c
*** trayfreq-0.2.x-dev2/src/bat_tray.c	2009-12-25 11:55:35.000000000 +0100
--- trayfreq-0.2.x-dev2-fr/src/bat_tray.c	2010-07-28 10:38:39.439312830 +0200
***************
*** 50,56 ****
      gchar time[50];
      memset(time, '\0', 50);
      gb_discharge_time(time);
!     sprintf(msg, "Percent Remaining: %i\%\nTime Remaining: %s", gb_percent(), time);
    }
  
    else if(gb_charging())
--- 50,56 ----
      gchar time[50];
      memset(time, '\0', 50);
      gb_discharge_time(time);
!     sprintf(msg, "Niveau en pourcentage: %i\%\nTemps restant : %s", gb_percent(), time);
    }
  
    else if(gb_charging())
***************
*** 58,68 ****
      gchar time[50];
      memset(time, '\0', 50);
      gb_charge_time(time);
!     sprintf(msg, "Percent Charged: %i\%\nTime Remaining: %s", gb_percent(), time);
    }
  
    else if(gb_charged())
!     sprintf(msg, "Fully Charged\nAC Plugged In");
  
    else
      sprintf(msg, "Error");
--- 58,68 ----
      gchar time[50];
      memset(time, '\0', 50);
      gb_charge_time(time);
!     sprintf(msg, "Niveau en pourcentage: %i\%\nTemps restant : %s", gb_percent(), time);
    }
  
    else if(gb_charged())
!     sprintf(msg, "Complètement chargée\nAlimentation connectée");
  
    else
      sprintf(msg, "Error");
diff -cr trayfreq-0.2.x-dev2/src/tray.c trayfreq-0.2.x-dev2-fr/src/tray.c
*** trayfreq-0.2.x-dev2/src/tray.c	2009-12-25 12:07:06.000000000 +0100
--- trayfreq-0.2.x-dev2-fr/src/tray.c	2010-07-28 10:47:34.516760537 +0200
***************
*** 205,211 ****
  
  
    gg_current(0, current_governer, 20);
!   sprintf(msg, "%sGovernor: %s\n", msg, current_governer);
   
    for(i = 0; i < gc_number(); ++i)
      {
--- 205,211 ----
  
  
    gg_current(0, current_governer, 20);
!   sprintf(msg, "%sProfil : %s\n", msg, current_governer);
   
    for(i = 0; i < gc_number(); ++i)
      {
diff -cr trayfreq-0.2.x-dev2/src/trayfreq_set.c trayfreq-0.2.x-dev2-fr/src/trayfreq_set.c
*** trayfreq-0.2.x-dev2/src/trayfreq_set.c	2009-12-25 11:40:12.000000000 +0100
--- trayfreq-0.2.x-dev2-fr/src/trayfreq_set.c	2010-07-28 10:52:40.929262336 +0200
***************
*** 73,127 ****
    gf_init();
  
    if(!argv[1])
!     printf("Use -g to set the governor or -f to set the frequency.\n");
  
    else if(strcmp(argv[1], "-g") == 0)
    {
      if(!argv[2])
!       printf("Pass the governor to set after the -g.\n");
      else
      {
        if(!argv[3])
!         printf("Use -c to set the core.\n");
  
        else if(strcmp(argv[3], "-c") == 0)
        {
          if(!argv[4])
!           printf("Pass the core to set after the -c.\n");
          else
            set_gov(argv[2], argv[4]);
        }
  
        else
!         printf("Use -c to set the core.\n");  
      }
    }
  
    else if(strcmp(argv[1], "-f") == 0)
    {
      if(!argv[2])
!       printf("Pass the frequency to set after the -f.\n");
      else
      {
        if(!argv[3])
!         printf("Use -c to set the core.\n");
  
        else if(strcmp(argv[3], "-c") == 0)
        {
          if(!argv[4])
!           printf("Pass the core to set after the -c.\n");
          else
            set_freq(argv[2], argv[4]);
        }
  
        else
!         printf("Use -c to set the core.\n");  
      }
    }
  
    else
    {
!     printf("Use -g to set the governor or -f to set the frequency.\n");
    }
  
    return 0;
--- 73,127 ----
    gf_init();
  
    if(!argv[1])
!     printf("Utiliser -g pour activer les profils or -f pour activer les fréquences.\n");
  
    else if(strcmp(argv[1], "-g") == 0)
    {
      if(!argv[2])
!       printf("Indiquer le profil après le -g.\n");
      else
      {
        if(!argv[3])
!         printf("Utiliser -c pour activer le core.\n");
  
        else if(strcmp(argv[3], "-c") == 0)
        {
          if(!argv[4])
!           printf("Indiquer le core après le -c.\n");
          else
            set_gov(argv[2], argv[4]);
        }
  
        else
!         printf("Utiliser -c pour activer le core.\n");  
      }
    }
  
    else if(strcmp(argv[1], "-f") == 0)
    {
      if(!argv[2])
!       printf("Indiquer la fréquence après le -f.\n");
      else
      {
        if(!argv[3])
!         printf("Utiliser -c pour activer le core.\n");
  
        else if(strcmp(argv[3], "-c") == 0)
        {
          if(!argv[4])
!           printf("Indiquer le core après le -c.\n");
          else
            set_freq(argv[2], argv[4]);
        }
  
        else
!         printf("Utiliser -c pour activer le core.\n");  
      }
    }
  
    else
    {
!     printf("Utiliser -g pour activer les profils or -f pour activer les fréquences.\n");
    }
  
    return 0;
