diff -Naur transkode~/src/common/commondefs.cpp transkode/src/common/commondefs.cpp
--- transkode~/src/common/commondefs.cpp	2010-01-01 22:36:40.000000000 -0600
+++ transkode/src/common/commondefs.cpp	2010-01-01 22:36:50.000000000 -0600
@@ -38,6 +38,8 @@
 	#include <unistd.h>
 #endif
 
+#include <cstdlib>
+
 /**
  * GeneralOpt
 */
diff -Naur transkode~/src/configdialog/streamsettings.ui.h transkode/src/configdialog/streamsettings.ui.h
--- transkode~/src/configdialog/streamsettings.ui.h	2010-01-01 22:36:40.000000000 -0600
+++ transkode/src/configdialog/streamsettings.ui.h	2010-01-01 22:36:50.000000000 -0600
@@ -42,6 +42,8 @@
 #include <kmessagebox.h>
 #include <klocale.h>
 
+#include <cstdlib>
+
 void StreamSettings::cleanUpRippedStreams()
 {
 	if ( KMessageBox::warningContinueCancel(
diff -Naur transkode~/src/metadata/audible/audibleproperties.cpp transkode/src/metadata/audible/audibleproperties.cpp
--- transkode~/src/metadata/audible/audibleproperties.cpp	2010-01-01 22:36:40.000000000 -0600
+++ transkode/src/metadata/audible/audibleproperties.cpp	2010-01-01 22:36:53.000000000 -0600
@@ -30,6 +30,7 @@
 #include "taglib_audiblefile.h"
 
 #include <netinet/in.h> // ntohl
+#include <stdio.h>
 
 using namespace TagLib;
 
diff -Naur transkode~/src/metadata/audible/audibletag.cpp transkode/src/metadata/audible/audibletag.cpp
--- transkode~/src/metadata/audible/audibletag.cpp	2010-01-01 22:36:40.000000000 -0600
+++ transkode/src/metadata/audible/audibletag.cpp	2010-01-01 22:36:53.000000000 -0600
@@ -28,6 +28,7 @@
 #include <taglib/tag.h>
 
 #include <netinet/in.h> // ntohl
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
diff -Naur transkode~/src/metadata/audible/taglib_audiblefile.cpp transkode/src/metadata/audible/taglib_audiblefile.cpp
--- transkode~/src/metadata/audible/taglib_audiblefile.cpp	2010-01-01 22:36:40.000000000 -0600
+++ transkode/src/metadata/audible/taglib_audiblefile.cpp	2010-01-01 22:36:53.000000000 -0600
@@ -29,6 +29,8 @@
 #include <taglib/tfile.h>
 #include <taglib/audioproperties.h>
 
+#include <stdio.h>
+
 namespace TagLib {
 ////////////////////////////////////////////////////////////////////////////////
 // public members
diff -Naur transkode~/src/metadata/audible/taglib_audiblefiletyperesolver.cpp transkode/src/metadata/audible/taglib_audiblefiletyperesolver.cpp
--- transkode~/src/metadata/audible/taglib_audiblefiletyperesolver.cpp	2010-01-01 22:36:40.000000000 -0600
+++ transkode/src/metadata/audible/taglib_audiblefiletyperesolver.cpp	2010-01-01 22:36:53.000000000 -0600
@@ -22,6 +22,7 @@
 #include "taglib_audiblefiletyperesolver.h"
 #include "taglib_audiblefile.h"
 
+#include <stdio.h>
 #include <string.h>
 
 TagLib::File *AudibleFileTypeResolver::createFile(const char *fileName,
diff -Naur transkode~/src/plugins/processhandler.cpp transkode/src/plugins/processhandler.cpp
--- transkode~/src/plugins/processhandler.cpp	2010-01-01 22:36:40.000000000 -0600
+++ transkode/src/plugins/processhandler.cpp	2010-01-01 22:36:50.000000000 -0600
@@ -28,6 +28,8 @@
 #include <qtimer.h>
 #include <qregexp.h>
 
+#include <cstdlib>
+
 ProcessHandler::ProcessHandler( int priority, int comm, QObject* parent, const char* name ):
 	QProcess( parent, name ),
 	m_priority( priority ),
