diff -aur nrn-7.2.pristine/src/gnu/RNG.h nrn-7.2.new/src/gnu/RNG.h
--- nrn-7.2.pristine/src/gnu/RNG.h	2011-11-22 22:00:11.000000000 +0000
+++ nrn-7.2.new/src/gnu/RNG.h	2013-01-16 13:26:53.466592972 +0000
@@ -36,10 +36,13 @@
 #include <inttypes.h>
 #endif
 #endif
-typedef int32_t _G_int32_t;
-typedef u_int32_t _G_uint32_t;
 #endif
 
+typedef int int32_t;
+typedef unsigned int u_int32_t;
+typedef int32_t _G_int32_t;
+typedef u_int32_t _G_uint32_t; 
+
 #include <assert.h>
 #include <math.h>
 
