--- a/usr/bin/adduser	2013-01-16 09:57:07.211616878 -0700
+++ b/usr/bin/adduser	2013-01-16 10:03:50.748027787 -0700
@@ -96,6 +96,9 @@
 #       * Created
 #######################################################################
 
+source /etc/default/useradd
+DEFAULT_GROUP=$(gawk -F: "/:${GROUP}:/ {print \$1}" /etc/group)
+
 # Path to files
 pfile=/etc/passwd
 gfile=/etc/group
@@ -108,10 +111,10 @@
 chmod=/bin/chmod
 
 # Defaults
-defhome=/home
-defshell=/bin/bash
+defhome=${HOME:-/home}
+defshell=${SHELL:-/bin/bash}
 defchmod=711 # home dir permissions - may be preferable to use 701, however.
-defgroup=users 
+defgroup=${DEFAULT_GROUP:-users}
 AGID="audio cdrom floppy plugdev video power netdev lp scanner" # additional groups for desktop users
 
 # Determine what the minimum UID is (for UID recycling)
