--- src/include/fst/extensions/far/sttable.h.orig	2010-12-01 02:38:56.000000000 +0100
+++ src/include/fst/extensions/far/sttable.h	2010-12-16 16:57:34.000000000 +0100
@@ -70,7 +70,7 @@
       LOG(FATAL) << "STTableWriter::Add: key disorder: " << key;
     last_key_ = key;
     positions_.push_back(stream_.tellp());
-    WriteType(stream_, key);
+    WriteType(stream_, static_cast<int64>(positions_.size()));
     entry_writer_(stream_, t);
   }
 
