Index: pino-0.2.10/src/account_action.vala
===================================================================
--- pino-0.2.10.orig/src/account_action.vala
+++ pino-0.2.10/src/account_action.vala
@@ -38,7 +38,8 @@ public class AccountAction : Action {
 	weak Thread thread_1;
 	
 	public AccountAction() {
-		name = "AccountAct";
+		GLib.Object (name: "AccountAct");
+
 		label = _("Accounts");
 		tooltip = null;
 		stock_id = null;
Index: pino-0.2.10/src/more_window.vala
===================================================================
--- pino-0.2.10.orig/src/more_window.vala
+++ pino-0.2.10/src/more_window.vala
@@ -28,7 +28,7 @@ public class MoreWindow : Window {
 	public signal void moar_event();
 	
 	public MoreWindow() {
-		type = WindowType.POPUP;
+		GLib.Object (type: WindowType.POPUP);
 		
 		//set_position(WindowPosition.MOUSE);
 		
