diff -Naur full-pack/Makefile full-pack-new/Makefile
--- full-pack/Makefile	2006-04-20 18:24:23.000000000 +0300
+++ full-pack-new/Makefile	2011-01-12 00:25:06.000000000 +0200
@@ -4,15 +4,6 @@
 #
 
 
-DATA_DIR1:=$(shell kbgoffice --dump-data-dir 2> /dev/null)
-DATA_DIR2:=$(shell gbgoffice --dump-data-dir 2> /dev/null)
-DATA_DIR3:=$(shell cbgoffice --dump-data-dir 2> /dev/null)
-
-DATA_DIR:=
-DATA_DIR:=$(if ${DATA_DIR},${DATA_DIR},${DATA_DIR1})
-DATA_DIR:=$(if ${DATA_DIR},${DATA_DIR},${DATA_DIR2})
-DATA_DIR:=$(if ${DATA_DIR},${DATA_DIR},${DATA_DIR3})
-
 DATA_DIR:=$(shell echo ${DATA_DIR} | sed 's/\/$$//')
 
 
@@ -43,21 +34,8 @@
 
 check:
 	@echo ""
-	@if (test -z "${DATA_DIR1}"); then \
-		if (test -z "${DATA_DIR2}"); then \
-			if (test -z "${DATA_DIR3}"); then \
-				echo "Application 'kbgoffice' (or 'gbgoffice', or 'cbgoffice') is not installed."; \
-				echo "You need to install at least one of them first."; \
-				echo "For more info visit http://bgoffice.sourceforge.net/"; \
-				echo ""; \
-				exit 1; \
-			else \
-				echo "Found cbgoffice."; \
-			fi \
-		else \
-			echo "Found gbgoffice."; \
-		fi \
-	else \
-		echo "Found kbgoffice."; \
+	@if (test -z "${DATA_DIR}"); then \
+		echo "You must set DATA_DIR"; \
+		exit 1; \
 	fi
 	@echo ""
