diff -Naur alacarte.orig/configure.ac alacarte/configure.ac
--- alacarte.orig/configure.ac	2011-09-30 01:51:22.000000000 +0200
+++ alacarte/configure.ac	2011-09-30 01:57:47.440809579 +0200
@@ -9,7 +9,6 @@
 AM_MAINTAINER_MODE
 
 m4_define(required_pygobject_version, 2.15.1)
-m4_define(required_pygtk_version, 2.13.0)
 
 dnl put the ACLOCAL flags in the makefile
 ACLOCAL="$ACLOCAL $ACLOCAL_FLAGS"
@@ -28,7 +27,6 @@
 dnl Check for correctly installed pygobject
 AC_MSG_CHECKING(for pygobject required_pygobject_version installed for python required_python_abi)
 prog="
-import pygtk; pygtk.require('2.0')
 import gobject
 assert gobject.pygobject_version >= tuple(map(int, 'required_pygobject_version'.split('.')))
 "
@@ -40,34 +38,6 @@
     AC_MSG_ERROR(required pygobject version not found)
 fi
 
-dnl Check for correctly installed pygtk
-AC_MSG_CHECKING(for pygtk required_pygtk_version installed for python ${PYTHON_VERSION})
-prog="
-import pygtk; pygtk.require('2.0')
-import gobject
-assert gobject.pygtk_version >= tuple(map(int, 'required_pygtk_version'.split('.')))
-"
-
-if $PYTHON -c "$prog" 1>&AC_FD_CC 2>&AC_FD_CC; then
-    AC_MSG_RESULT(found)
-else
-    AC_MSG_RESULT(not found)
-    AC_MSG_ERROR(required pygtk version not found)
-fi
-
-dnl Check for correctly installed python-glade
-dnl AC_MSG_CHECKING(for pyglade required_pygtk_version installed for python ${PYTHON_VERSION})
-dnl prog="
-dnl import pygtk; pygtk.require('2.0')
-dnl import gtk.glade
-dnl "
-dnl if $PYTHON -c "$prog" 1>&AC_FD_CC 2>&AC_FD_CC; then
-dnl    AC_MSG_RESULT(found)
-dnl else
-dnl    AC_MSG_RESULT(not found)
-dnl    AC_MSG_ERROR(required pyglade version not found)
-dnl fi
-
 AC_CONFIG_FILES([
         Makefile
         po/Makefile.in
