Index: applet/yawp.cpp
===================================================================
--- applet/yawp.cpp	(revision 483)
+++ applet/yawp.cpp	(revision 486)
@@ -326,7 +326,6 @@
 	loadConfig();
 	loadCustomTheme();
 	initAppletPainter();
-	resize( m_pAppletPainter->getSize(QSize()) );
 
 	//--- just 4 debug ---
 /*	CityWeather dummyCity1;
@@ -347,25 +346,29 @@
 	dummyCity2.setTimeZone("Asia/Shanghai");
 	m_pWeatherModel->addCity(dummyCity2);
 	//--- end of debug section ---
-*/	
+*/
 	updateCitySubMenu();
 	setCityIndex( m_configData.iCityIndex );
 
 	const bool bHasCities = (m_pWeatherModel->rowCount() > 0);
-	m_pManualUpdate->setEnabled( bHasCities );
-	m_pCitySubMenu->setEnabled( bHasCities);
-	
+
 	if (bHasCities)
 		m_iIdPendingEngineConnection = startTimer( m_configData.iStartDelay*60*1000 );
 	else
+	{
+		resize( m_pAppletPainter->getSize(QSize()) );
 		QTimer::singleShot(0, this, SLOT(slotConfigurationRequired()) );
-	
+	}
+
+	m_pManualUpdate->setEnabled( bHasCities );
+	m_pCitySubMenu->setEnabled( bHasCities);
+
 	startTraverseLocationTimeout();
-	
+
 	// Every 5 days, 15 minutes after start of applet, cache directory will be checked to remove old files.
 	if (QDate::currentDate().dayOfYear() % 5 == 0)
 		QTimer::singleShot(15*60*1000, this, SLOT(slotStartCacheCleanUp()) );
-	
+
 	dEndFunct();
 }
 
