--- ada_module/core/src/ada_semantic_tree-generics.adb.orig	2012-05-26 10:47:42.835917491 +0200
+++ ada_module/core/src/ada_semantic_tree-generics.adb	2012-05-26 11:36:07.100183736 +0200
@@ -389,14 +389,12 @@
       Cached : Cache_Access := Get_Cache (Info);
    begin
       if Cached /= null then
-         Result := new Declaration_View_Record'
-           (Entity          =>
-              To_Entity_Access
-                (Instanciated_Package (Cached.all).Generic_Package),
-            Generic_Context =>
-              To_Active
-                (Instanciated_Package (Cached.all).Generic_Context),
-            others => <>);
+         Result := new Declaration_View_Record;
+         Result.Entity := To_Entity_Access
+           (Instanciated_Package (Cached.all).Generic_Package);
+         Declaration_View_Record (Result.all).Generic_Context :=
+           To_Active
+             (Instanciated_Package (Cached.all).Generic_Context);
 
          Ref (Declaration_View_Record (Result.all).Generic_Context);
 
