--- shared.gpr.orig	2011-10-15 22:51:47.916599465 +0300
+++ shared.gpr	2011-10-15 22:51:47.914599210 +0300
@@ -35,7 +35,7 @@
                "-gnatwaCJe", "-gnat05");
 
             for Default_Switches ("C") use
+              ("-g", "-O1") & GnatColl_Shared.Gtk_Include;
-              ("-g", "-O1", "-fdump-xref") & GnatColl_Shared.Gtk_Include;
 
          when "Production" =>
             for Default_Switches ("Ada") use
