diff -Naur xsd-3.3.0+dep.orig/libcult/cult/mm/evptr.hxx xsd-3.3.0+dep/libcult/cult/mm/evptr.hxx
--- xsd-3.3.0+dep.orig/libcult/cult/mm/evptr.hxx	2010-01-01 12:12:06.000000000 +0100
+++ xsd-3.3.0+dep/libcult/cult/mm/evptr.hxx	2012-01-19 06:48:45.713095213 +0100
@@ -70,7 +70,7 @@
       Evptr&
       operator= (Evptr const& ep)
       {
-        assign (ep.cp_ ? ep.cp_ : ep.p_, ep.c_);
+        this->assign (ep.cp_ ? ep.cp_ : ep.p_, ep.c_);
 
         return *this;
       }
diff -Naur xsd-3.3.0+dep.orig/libcult/cult/mm/shptr.hxx xsd-3.3.0+dep/libcult/cult/mm/shptr.hxx
--- xsd-3.3.0+dep.orig/libcult/cult/mm/shptr.hxx	2010-01-01 12:12:06.000000000 +0100
+++ xsd-3.3.0+dep/libcult/cult/mm/shptr.hxx	2012-01-19 07:06:45.656894436 +0100
@@ -53,7 +53,7 @@
       Shptr&
       operator= (Shptr<Y> const& ap)
       {
-        assign (ap);
+        this->assign (ap);
         return *this;
       }

