--- configure	2009-03-16 19:15:41.000000000 +0100
+++ configure2	2012-07-20 12:02:20.658234678 +0200
@@ -180,7 +180,7 @@
 	INSTALL_DAEMON='';
 	DAEMON='';
 	SNMP="enable";
-	MODULE_NESSUS="enable";
+	MODULE_NESSUS="disable";
 	TRAYICON="enable";
 	GNOME="enable";
 	SOUND="enable";
@@ -1279,9 +1279,8 @@
 	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/usr/share/doc/autoscan-network" >> Makefile
 	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/usr/share/pixmaps/autoscan-network" >> Makefile
 	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/usr/share/sounds/autoscan-network" >> Makefile
-	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/usr/lib/menu" >> Makefile
 	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/usr/share/applications" >> Makefile
-	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/usr/share/icons" >> Makefile
+	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/usr/share/icons/hicolor/22x22/apps" >> Makefile
 	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/usr/bin" >> Makefile
 	echo "" >> Makefile
 
@@ -1290,8 +1289,7 @@
 if [ $BUILD_GUI == "yes" ]; then
 	echo "	@echo '[01;34m*** Installing icon...[00m'" >> Makefile
 	echo "	\$(INSTALL) -c -m 644 usr/share/applications/autoscan-network.desktop \$(DESTDIR)/usr/share/applications/" >> Makefile
-	echo "	\$(INSTALL) -c -m 644 usr/share/icons/autoscan-network.png \$(DESTDIR)/usr/share/icons/" >> Makefile
-	echo "	\$(INSTALL) -c -m 644 usr/lib/menu/autoscan-network \$(DESTDIR)/usr/lib/menu/" >> Makefile
+	echo "	\$(INSTALL) -c -m 644 usr/share/icons/autoscan-network.png \$(DESTDIR)/usr/share/icons/hicolor/22x22/apps/" >> Makefile
 	echo "" >> Makefile
 fi
 
@@ -1308,7 +1306,7 @@
 
 if [ $BUILD_GUI == "yes" ] && [ $SYSTEM != "Gentoo" ]; then
 	echo "	@echo '[01;34m***  Gconf config...[00m'" >> Makefile
-	echo "	export GCONF_CONFIG_SOURCE=\$(GCONF);gconftool-2 --makefile-install-rule /usr/share/apps/autoscan-network/autoscan-network.schemas &>/dev/null" >> Makefile
+	echo "	export GCONF_CONFIG_SOURCE=\$(GCONF);gconftool-2 --makefile-install-rule \$(DESTDIR)/usr/share/apps/autoscan-network/autoscan-network.schemas &>/dev/null" >> Makefile
 fi
 
 echo "	@echo '[01;34m*** Installing executable...[00m'" >> Makefile
@@ -1318,17 +1316,8 @@
 
 if [ $BUILD_AGENT == "yes" ]; then
 	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/usr/sbin" >> Makefile
-	echo "	\$(SHTOOL) mkdir -f -p -m 755 \$(DESTDIR)/etc/init.d" >> Makefile
 	echo "	\$(INSTALL) -c -m 755 -o root -g root bin/autoscan-network-daemon \$(DESTDIR)/usr/sbin/" >> Makefile
-	echo "	@instance=\`ps aux  | grep \"[0-9] /usr/sbin/autoscan-network-daemon\" | wc -l\`; echo \"Instance (autoscan-network-daemon) : \$\$instance\"; if [ \"\$\$instance\" -eq 2 ]; then echo \"kill Agent\"; /usr/sbin/autoscan-network-daemon --stop; fi" >> Makefile
-	echo "	\$(INSTALL) -c -m 755 -o root -g root $DAEMON" >> Makefile
-fi
-echo "" >> Makefile
-
-echo "	@echo '[01;34m*** Installing Script...[00m'" >> Makefile
-if [ $BUILD_AGENT == "yes" ] && [ $SYSTEM != "Gentoo" ]; then
-	echo "	$INSTALL_DAEMON" >> Makefile
-	echo "	/etc/init.d/autoscan-network start" >> Makefile
+	echo "	@instance=\`ps aux  | grep \"[0-9] /usr/sbin/autoscan-network-daemon\" | wc -l\`; echo \"Instance (autoscan-network-daemon) : \$\$instance\"; if [ \"\$\$instance\" -eq 2 ]; then echo \"kill Agent\"; rc.d stop autoscan-network; fi" >> Makefile
 fi
 echo "" >> Makefile
 
