--- usr/share/applications/autoscan-network.desktop	2008-07-22 17:02:14.000000000 +0200
+++ usr/share/applications/autoscan-network2.desktop	2012-07-20 11:48:07.358270366 +0200
@@ -3,10 +3,11 @@
 Encoding=UTF-8
 Name=AutoScan Network
 Comment=AutoScan is a utility for network exploration
+Comment[fr_FR]=AutoScan est un utilitaire d'exploration réseau
 Exec=autoscan-network
 Terminal=false
 MultipleArgs=false
 Type=Application
-Icon=/usr/share/pixmaps/autoscan-network/AutoScan.png
-Categories=Application;Network
+Icon=autoscan-network
+Categories=Network
 Name[fr_FR]=AutoScan Network
