diff -urN xf86-input-wacom-0.10.99.1-orig/src/wcmUSB.c xf86-input-wacom-0.10.99.1/src/wcmUSB.c
--- xf86-input-wacom-0.10.99.1-orig/src/wcmUSB.c	2011-04-03 11:16:06.000000000 +0200
+++ xf86-input-wacom-0.10.99.1/src/wcmUSB.c	2011-04-03 11:16:38.000000000 +0200
@@ -1053,7 +1053,7 @@
 {
 	int mask;
 
-	if (btn >= sizeof(int))
+	if (btn >= sizeof(int)*8)
 	{
 		xf86Msg(X_ERROR, "%s: Invalid button number %d. Insufficient "
 				"storage\n", __func__, btn);
