diff -ur xf86-input-wacom-0.10.10.orig/src/wcmXCommand.c xf86-input-wacom-0.10.10/src/wcmXCommand.c
--- xf86-input-wacom-0.10.10.orig/src/wcmXCommand.c	2011-01-14 12:14:27.000000000 +0100
+++ xf86-input-wacom-0.10.10/src/wcmXCommand.c	2011-01-15 16:46:16.000000000 +0100
@@ -322,13 +322,14 @@
 
 	for (i = 0; i < prop->size; i++)
 	{
+		memset(keys[i+1], 0, sizeof(keys[i]));
+
 		if (!values[i])
 			continue;
 
 		XIGetDeviceProperty(dev, values[i], &val);
 
 		/* keys is one based array to align with X buttons */
-		memset(keys[i+1], 0, sizeof(keys[i]));
 		for (j = 0; j < val->size; j++)
 			keys[i+1][j] = ((unsigned int*)val->data)[j];
 	}
diff -ur xf86-input-wacom-0.10.10.orig/tools/xsetwacom.c xf86-input-wacom-0.10.10/tools/xsetwacom.c
--- xf86-input-wacom-0.10.10.orig/tools/xsetwacom.c	2011-01-14 12:11:05.000000000 +0100
+++ xf86-input-wacom-0.10.10/tools/xsetwacom.c	2011-01-15 16:44:51.000000000 +0100
@@ -1294,7 +1294,7 @@
 static void special_map_property(Display *dpy, XDevice *dev, Atom btnact_prop, int offset, int argc, char **argv)
 {
 	unsigned long *data, *btnact_data;
-	Atom type, prop;
+	Atom type, prop = 0;
 	int format;
 	unsigned long btnact_nitems, nitems, bytes_after;
 	int need_update = 0;
@@ -1324,7 +1324,10 @@
 	}
 
 	if (argc == 0) /* unset property */
+	{
+		prop = btnact_data[offset];
 		btnact_data[offset] = 0;
+	}
 	else if (btnact_data[offset]) /* some atom already assigned, modify that */
 		prop = btnact_data[offset];
 	else
@@ -1371,6 +1374,10 @@
 				PropModeReplace,
 				(unsigned char*)btnact_data,
 				btnact_nitems);
+
+	if (argc == 0 && prop)
+	      XDeleteDeviceProperty(dpy, dev, prop);
+
 	XFlush(dpy);
 }
 
@@ -1436,7 +1443,7 @@
 		return;
 
 	nmap = XGetDeviceButtonMapping(dpy, dev, map, nmap);
-	if (btn_no >= nmap)
+	if (btn_no > nmap)
 	{
 		fprintf(stderr, "Button number does not exist on device.\n");
 		return;
