diff -urN xf86-input-wacom-0.10.10.orig/src/wcmCommon.c xf86-input-wacom-0.10.10/src/wcmCommon.c
--- xf86-input-wacom-0.10.10.orig/src/wcmCommon.c	2010-12-08 15:50:12.000000000 +0100
+++ xf86-input-wacom-0.10.10/src/wcmCommon.c	2010-12-08 15:51:03.000000000 +0100
@@ -953,14 +953,12 @@
 
 	/* don't move the cursor if in gesture mode */
 	if (common->wcmGestureMode)
-		goto ret;
+		return;
 
 	/* For touch, only first finger moves the cursor */
 	if ((ds.device_type == TOUCH_ID && common->wcmTouch && !channel) ||
 	    (ds.device_type != TOUCH_ID))
 		commonDispatchDevice(common,channel,pChannel, suppress);
-ret:
-	wcmResetSampleCounter(pChannel);
 }
 
 static int idtotype(int id)
