diff -urN xf86-input-wacom-0.10.99.1-orig/src/wcmUSB.c xf86-input-wacom-0.10.99.1/src/wcmUSB.c
--- xf86-input-wacom-0.10.99.1-orig/src/wcmUSB.c	2011-04-03 11:17:25.000000000 +0200
+++ xf86-input-wacom-0.10.99.1/src/wcmUSB.c	2011-04-03 11:21:37.000000000 +0200
@@ -1412,11 +1412,12 @@
 
 	DBG(6, common, "%d events received\n", private->wcmEventCnt);
 
-	if (private->wcmPenTouch)
-	{
+	if (private->wcmUseMT)
 		private->wcmDeviceType = usbInitToolType(private->wcmEvents,
 						private->wcmEventCnt);
 
+	if (private->wcmPenTouch)
+	{
 		/* We get both pen and touch data from the kernel when they
 		 * both are in/down. So, if we were (hence the need of dslast)
 		 * processing pen events, we should ignore touch events.
