diff -urN xf86-input-wacom-0.10.10.orig/src/wcmValidateDevice.c xf86-input-wacom-0.10.10/src/wcmValidateDevice.c
--- xf86-input-wacom-0.10.10.orig/src/wcmValidateDevice.c	2010-11-22 00:35:22.000000000 +0100
+++ xf86-input-wacom-0.10.10/src/wcmValidateDevice.c	2011-01-02 12:49:19.000000000 +0100
@@ -229,6 +229,9 @@
 		case 0xD1:  /* Bamboo with 2FGT */
 		case 0xD2:  /* Bamboo with 2FGT */
 		case 0xD3:  /* Bamboo with 2FGT */
+		case 0xD8:  /* Bamboo with 2FGT */
+		case 0xDA:  /* Bamboo with 2FGT */
+		case 0xDB:  /* Bamboo with 2FGT */
 			priv->common->tablet_type |= WCM_2FGT;
 			break;
 
