diff -Naur emane-0.6.4/agents/gpsdlocation/gpsdlocationagent.cc emane-0.6.4-patched/agents/gpsdlocation/gpsdlocationagent.cc
--- emane-0.6.4/agents/gpsdlocation/gpsdlocationagent.cc	2010-09-22 09:38:02.000000000 -0400
+++ emane-0.6.4-patched/agents/gpsdlocation/gpsdlocationagent.cc	2011-03-03 13:08:30.013086717 -0500
@@ -36,10 +36,11 @@
 #include "emaneutils/parameterconvert.h"
 #include <sstream>
 
-#include <ace/LSOCK_Connector.h>
-#include <ace/SOCK_Connector.h>
 #include <ace/DEV_Connector.h>
 #include <ace/FILE_Connector.h>
+#include <ace/LSOCK_Connector.h>
+#include <ace/OS_NS_time.h>
+#include <ace/SOCK_Connector.h>
 #include <ace/TTY_IO.h>
 
 #ifdef __APPLE__
@@ -213,7 +214,7 @@
     }
 
   ACE_TTY_IO slaveTTY;
-  ACE_DEV_CONNECTOR devConnector;
+  ACE_DEV_Connector devConnector;
   ACE_TTY_IO::Serial_Params serial_params;
 
   if(devConnector.connect(slaveTTY,ACE_DEV_Addr(&pzName[1])) == -1)
diff -Naur emane-0.6.4/configure.ac emane-0.6.4-patched/configure.ac
--- emane-0.6.4/configure.ac	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/configure.ac	2011-03-03 13:08:30.016420740 -0500
@@ -78,7 +78,7 @@
    LDFLAGS="${ACE_LIBS} ${LDFLAGS}"
 fi
 
-AC_CHECK_HEADER(ace/OS.h,
+AC_CHECK_HEADER(ace/ACE.h,
  [AC_CHECK_LIB([ACE],main,[ACE_LIBS+=' -lACE'],[AC_MSG_ERROR("ACE not installed")])],
  [AC_MSG_ERROR("Missing ACE development headers")])
 
diff -Naur emane-0.6.4/generators/emulationscript/emulationscripteventgenerator.cc emane-0.6.4-patched/generators/emulationscript/emulationscripteventgenerator.cc
--- emane-0.6.4/generators/emulationscript/emulationscripteventgenerator.cc	2010-09-22 09:38:05.000000000 -0400
+++ emane-0.6.4-patched/generators/emulationscript/emulationscripteventgenerator.cc	2011-03-03 13:08:30.006418671 -0500
@@ -39,8 +39,6 @@
 
 #include <sstream>
 
-#include <ace/OS.h>
-
 namespace
 {
   /* ConfigurationDefinition to be filled-in here */
diff -Naur emane-0.6.4/generators/mitremobility/mobilitymodeleventgenerator.cc emane-0.6.4-patched/generators/mitremobility/mobilitymodeleventgenerator.cc
--- emane-0.6.4/generators/mitremobility/mobilitymodeleventgenerator.cc	2010-09-22 09:38:05.000000000 -0400
+++ emane-0.6.4-patched/generators/mitremobility/mobilitymodeleventgenerator.cc	2011-03-03 13:08:30.009752694 -0500
@@ -38,8 +38,6 @@
 
 #include <sstream>
 
-#include <ace/OS.h>
-
 namespace
 {
   EMANE::ConfigurationDefinition defs[] =
diff -Naur emane-0.6.4/generators/mitremobility/mobilitymodelloader.cc emane-0.6.4-patched/generators/mitremobility/mobilitymodelloader.cc
--- emane-0.6.4/generators/mitremobility/mobilitymodelloader.cc	2010-09-22 09:38:05.000000000 -0400
+++ emane-0.6.4-patched/generators/mitremobility/mobilitymodelloader.cc	2011-03-03 13:08:30.013086717 -0500
@@ -34,7 +34,7 @@
 #include "mobilitymodelloader.h"
 #include "emaneutils/spawnmemberfunc.h"
 
-#include <ace/OS.h>
+#include <cmath>
 
 #define NEM_TO_INDEX(x) ((x) - 1)
 
diff -Naur emane-0.6.4/include/emaneevents/locationevent.h emane-0.6.4-patched/include/emaneevents/locationevent.h
--- emane-0.6.4/include/emaneevents/locationevent.h	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/include/emaneevents/locationevent.h	2011-03-03 13:08:29.956408321 -0500
@@ -37,7 +37,6 @@
 #include "emane/emaneevent.h"
 #include "emaneevents/events.h"
 
-#include <ace/OS.h>
 #include <ace/Basic_Types.h>
 
 /**
diff -Naur emane-0.6.4/include/emaneutils/bitpool.inl emane-0.6.4-patched/include/emaneutils/bitpool.inl
--- emane-0.6.4/include/emaneutils/bitpool.inl	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/include/emaneutils/bitpool.inl	2011-03-03 13:08:29.959742344 -0500
@@ -33,6 +33,8 @@
 
 #include "bitpool.h"
 
+#include <cmath>
+
 const ACE_UINT64 USEC_PER_SEC = 1000000;
 
 
diff -Naur emane-0.6.4/include/emaneutils/netutils.h emane-0.6.4-patched/include/emaneutils/netutils.h
--- emane-0.6.4/include/emaneutils/netutils.h	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/include/emaneutils/netutils.h	2011-03-03 13:08:29.959742344 -0500
@@ -34,7 +34,6 @@
 #ifndef EMANEUTILSNETUTILS_HEADER_
 #define EMANEUTILSNETUTILS_HEADER_
 
-#include <ace/OS.h>
 #include <ace/INET_Addr.h>
 
 namespace EMANEUtils
diff -Naur emane-0.6.4/include/emaneutils/recvcancelable.h emane-0.6.4-patched/include/emaneutils/recvcancelable.h
--- emane-0.6.4/include/emaneutils/recvcancelable.h	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/include/emaneutils/recvcancelable.h	2011-03-03 13:08:29.959742344 -0500
@@ -34,7 +34,7 @@
 #ifndef EMANEUTILSRECVCANCELABLE_HEADER_
 #define EMANEUTILSRECVCANCELABLE_HEADER_
 
-#include <ace/OS.h>
+#include <ace/Addr.h>
 
 namespace EMANEUtils
 {
diff -Naur emane-0.6.4/libs/emanelogservice/logservice.cc emane-0.6.4-patched/libs/emanelogservice/logservice.cc
--- emane-0.6.4/libs/emanelogservice/logservice.cc	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/libs/emanelogservice/logservice.cc	2011-03-03 13:08:29.946406252 -0500
@@ -41,7 +41,7 @@
 #include <cstdarg>
 #include <cstdio>
 
-#include <ace/OS.h>
+#include <ace/OS_NS_time.h>
 #include <ace/Log_Msg.h>
 #include <ace/streams.h>
 
@@ -198,7 +198,7 @@
 
 void EMANE::LogService::redirectLogsToRemoteLogger(const ACE_TCHAR *program, const char* addr)
 {
-  if(ACE_LOG_MSG->open(program, ACE_Log_Msg::LOGGER,ACE_LIB_TEXT (addr)) == -1)
+  if(ACE_LOG_MSG->open(program, ACE_Log_Msg::LOGGER,ACE_TEXT (addr)) == -1)
   {
     ACE_DEBUG((LM_ERROR, ACE_TEXT ("RemoteLogging Failed: %s\n"),ACE_OS::strerror(errno)));
   }
diff -Naur emane-0.6.4/libs/emanexmlparser/layerconfiguration.h emane-0.6.4-patched/libs/emanexmlparser/layerconfiguration.h
--- emane-0.6.4/libs/emanexmlparser/layerconfiguration.h	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/libs/emanexmlparser/layerconfiguration.h	2011-03-03 13:08:29.943072229 -0500
@@ -42,7 +42,7 @@
 #include <string>
 #include <map>
 
-#include <ace/OS.h>
+#include <ace/Basic_Types.h>
 
 #include <libxml/tree.h>
 
diff -Naur emane-0.6.4/libs/emanexmlparser/nemconfiguration.h emane-0.6.4-patched/libs/emanexmlparser/nemconfiguration.h
--- emane-0.6.4/libs/emanexmlparser/nemconfiguration.h	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/libs/emanexmlparser/nemconfiguration.h	2011-03-03 13:08:29.943072229 -0500
@@ -40,7 +40,7 @@
 #include <string>
 #include <list>
 
-#include <ace/OS.h>
+#include <ace/Basic_Types.h>
 
 namespace EMANE
 {
diff -Naur emane-0.6.4/libs/emanexmlparser/platformconfiguration.cc emane-0.6.4-patched/libs/emanexmlparser/platformconfiguration.cc
--- emane-0.6.4/libs/emanexmlparser/platformconfiguration.cc	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/libs/emanexmlparser/platformconfiguration.cc	2011-03-03 13:08:29.943072229 -0500
@@ -36,8 +36,6 @@
 #include <iostream>
 #include <sstream>
 
-#include <ace/OS.h>
-
 #include <libxml/tree.h>
 
 EMANE::PlatformConfiguration::PlatformConfiguration(EMANE::ConfigurationParser *pParser,
diff -Naur emane-0.6.4/libs/emanexmlparser/platformconfiguration.h emane-0.6.4-patched/libs/emanexmlparser/platformconfiguration.h
--- emane-0.6.4/libs/emanexmlparser/platformconfiguration.h	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/libs/emanexmlparser/platformconfiguration.h	2011-03-03 13:08:29.943072229 -0500
@@ -41,7 +41,7 @@
 #include <string>
 #include <list>
 
-#include <ace/OS.h>
+#include <ace/Basic_Types.h>
 
 namespace EMANE
 {
diff -Naur emane-0.6.4/models/ieee80211abg/maclayer/modetimingparameters.cc emane-0.6.4-patched/models/ieee80211abg/maclayer/modetimingparameters.cc
--- emane-0.6.4/models/ieee80211abg/maclayer/modetimingparameters.cc	2010-09-22 09:38:05.000000000 -0400
+++ emane-0.6.4-patched/models/ieee80211abg/maclayer/modetimingparameters.cc	2011-03-03 13:08:29.973078439 -0500
@@ -40,7 +40,6 @@
 #include "ieee80211abgphysotmessage.h"
 #include "ieee80211abgmaclayer.h"
 
-#include <ace/OS.h>
 #include <ace/Task.h>
 
 #include <math.h>
diff -Naur emane-0.6.4/models/ieee80211abg/maclayer/modetimingparameters.h emane-0.6.4-patched/models/ieee80211abg/maclayer/modetimingparameters.h
--- emane-0.6.4/models/ieee80211abg/maclayer/modetimingparameters.h	2010-09-22 09:38:05.000000000 -0400
+++ emane-0.6.4-patched/models/ieee80211abg/maclayer/modetimingparameters.h	2011-03-03 13:08:29.976412462 -0500
@@ -36,7 +36,6 @@
 
 #include "macconfig.h"
 
-#include <ace/OS.h>
 #include <ace/Basic_Types.h>
 #include <ace/Time_Value.h>
 #include <ace/Singleton.h>
diff -Naur emane-0.6.4/models/ieee80211abg/maclayer/transmissiontxstate.cc emane-0.6.4-patched/models/ieee80211abg/maclayer/transmissiontxstate.cc
--- emane-0.6.4/models/ieee80211abg/maclayer/transmissiontxstate.cc	2010-09-22 09:38:05.000000000 -0400
+++ emane-0.6.4-patched/models/ieee80211abg/maclayer/transmissiontxstate.cc	2011-03-03 13:08:29.973078439 -0500
@@ -35,8 +35,6 @@
 #include "ieee80211abgmaclayer.h"
 #include "macstatistics.h"
 
-#include <ace/OS.h>
-
 TransmissionTxState::TransmissionTxState(){}
 
 TransmissionTxState::~TransmissionTxState(){}
diff -Naur emane-0.6.4/models/ieee80211abg/phylayer/ieee80211abgbiterrorrate.cc emane-0.6.4-patched/models/ieee80211abg/phylayer/ieee80211abgbiterrorrate.cc
--- emane-0.6.4/models/ieee80211abg/phylayer/ieee80211abgbiterrorrate.cc	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/models/ieee80211abg/phylayer/ieee80211abgbiterrorrate.cc	2011-03-03 13:08:29.996416600 -0500
@@ -39,8 +39,6 @@
 #include <utility>
 #include <sstream>
 
-#include <ace/OS.h>
-
 #include <libxml/parser.h>
 #include <libxml/tree.h>
 #include <libxml/xpath.h>
diff -Naur emane-0.6.4/models/ieee80211abg/phylayer/ieee80211abgnoiseestimation.cc emane-0.6.4-patched/models/ieee80211abg/phylayer/ieee80211abgnoiseestimation.cc
--- emane-0.6.4/models/ieee80211abg/phylayer/ieee80211abgnoiseestimation.cc	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/models/ieee80211abg/phylayer/ieee80211abgnoiseestimation.cc	2011-03-03 13:08:29.999750624 -0500
@@ -37,7 +37,7 @@
 #include <math.h>
 #include <sstream>
 
-#include <ace/OS.h>
+#include <ace/OS_NS_time.h>
 
 namespace
 {
diff -Naur emane-0.6.4/models/ieee80211abg/phylayer/ieee80211abgphylayer.cc emane-0.6.4-patched/models/ieee80211abg/phylayer/ieee80211abgphylayer.cc
--- emane-0.6.4/models/ieee80211abg/phylayer/ieee80211abgphylayer.cc	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/models/ieee80211abg/phylayer/ieee80211abgphylayer.cc	2011-03-03 13:08:29.999750624 -0500
@@ -51,7 +51,7 @@
 #include <sstream>
 
 #include <ace/Time_Value.h>
-#include <ace/OS.h>
+#include <ace/OS_NS_time.h>
 
 #define USEC_PER_SEC 1000000
 
diff -Naur emane-0.6.4/models/ieee80211abg/phylayer/rxbusystate.cc emane-0.6.4-patched/models/ieee80211abg/phylayer/rxbusystate.cc
--- emane-0.6.4/models/ieee80211abg/phylayer/rxbusystate.cc	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/models/ieee80211abg/phylayer/rxbusystate.cc	2011-03-03 13:08:29.999750624 -0500
@@ -35,7 +35,7 @@
 #include "idlestate.h"
 #include "ieee80211abgphylayer.h"
 
-#include <ace/OS.h>
+#include <ace/OS_NS_time.h>
 
 RxBusyState::RxBusyState() 
 {
diff -Naur emane-0.6.4/models/ieee80211abg/phylayer/rxpendingstate.cc emane-0.6.4-patched/models/ieee80211abg/phylayer/rxpendingstate.cc
--- emane-0.6.4/models/ieee80211abg/phylayer/rxpendingstate.cc	2010-09-22 09:38:06.000000000 -0400
+++ emane-0.6.4-patched/models/ieee80211abg/phylayer/rxpendingstate.cc	2011-03-03 13:08:29.999750624 -0500
@@ -35,7 +35,7 @@
 #include "rxbusystate.h"
 #include "ieee80211abgphylayer.h"
 
-#include <ace/OS.h>
+#include <ace/OS_NS_time.h>
 
 RxPendingState::RxPendingState() 
 {
diff -Naur emane-0.6.4/models/rfpipe/phylayer/rfpipephylayer.cc emane-0.6.4-patched/models/rfpipe/phylayer/rfpipephylayer.cc
--- emane-0.6.4/models/rfpipe/phylayer/rfpipephylayer.cc	2010-09-22 09:38:05.000000000 -0400
+++ emane-0.6.4-patched/models/rfpipe/phylayer/rfpipephylayer.cc	2011-03-03 13:08:29.959742344 -0500
@@ -51,6 +51,9 @@
 #include <sstream>
 #include <iomanip>
 
+#include <ace/OS_NS_time.h>
+#include <ace/OS_NS_sys_select.h>
+
 namespace
 {
   EMANE::ConfigurationDefinition defs[] =
diff -Naur emane-0.6.4/services/eventserver/emaneeventserver.cc emane-0.6.4-patched/services/eventserver/emaneeventserver.cc
--- emane-0.6.4/services/eventserver/emaneeventserver.cc	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/services/eventserver/emaneeventserver.cc	2011-03-03 13:08:29.953074298 -0500
@@ -46,8 +46,8 @@
 #include <sstream>
 #include <iostream>
 
-#include <ace/OS.h>
 #include <ace/Get_Opt.h>
+#include <ace/OS_NS_time.h>
 #include <ace/Reactor.h>
 #include <ace/Sched_Params.h>
 
diff -Naur emane-0.6.4/services/eventserver/eventgeneratorconfiguration.h emane-0.6.4-patched/services/eventserver/eventgeneratorconfiguration.h
--- emane-0.6.4/services/eventserver/eventgeneratorconfiguration.h	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/services/eventserver/eventgeneratorconfiguration.h	2011-03-03 13:08:29.956408321 -0500
@@ -40,8 +40,6 @@
 
 #include <string>
 
-#include <ace/OS.h>
-
 namespace EMANE
 {
   /**
diff -Naur emane-0.6.4/services/eventserver/eventgeneratorfactory.h emane-0.6.4-patched/services/eventserver/eventgeneratorfactory.h
--- emane-0.6.4/services/eventserver/eventgeneratorfactory.h	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/services/eventserver/eventgeneratorfactory.h	2011-03-03 13:08:29.953074298 -0500
@@ -39,7 +39,7 @@
 #include "emaneutils/factoryexception.h"
 #include "emaneutils/componenttypes.h"
 
-#include <ace/OS.h>
+#include <ace/OS_NS_dlfcn.h>
 
 namespace EMANE
 {
diff -Naur emane-0.6.4/services/eventserver/eventserviceconfiguration.cc emane-0.6.4-patched/services/eventserver/eventserviceconfiguration.cc
--- emane-0.6.4/services/eventserver/eventserviceconfiguration.cc	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/services/eventserver/eventserviceconfiguration.cc	2011-03-03 13:08:29.956408321 -0500
@@ -43,9 +43,6 @@
 #include <sstream>
 #include <iostream>
 
-#include <ace/OS.h>
-
-
 EMANE::EventServiceConfiguration::EventServiceConfiguration(const std::string &sFile)
   throw(ParseException,ValidateException)
   : LayerConfiguration(EMANE::ConfigurationParserSingleton::instance())
diff -Naur emane-0.6.4/shims/timinganalysis/src/timinganalysisshim.cc emane-0.6.4-patched/shims/timinganalysis/src/timinganalysisshim.cc
--- emane-0.6.4/shims/timinganalysis/src/timinganalysisshim.cc	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/shims/timinganalysis/src/timinganalysisshim.cc	2011-03-03 13:08:30.033090858 -0500
@@ -37,7 +37,7 @@
 
 #include <sstream>
 
-#include <ace/OS.h>
+#include <ace/OS_NS_time.h>
 
 namespace
 {
diff -Naur emane-0.6.4/src/emaneeventd/eventagentconfiguration.h emane-0.6.4-patched/src/emaneeventd/eventagentconfiguration.h
--- emane-0.6.4/src/emaneeventd/eventagentconfiguration.h	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/src/emaneeventd/eventagentconfiguration.h	2011-03-03 13:08:30.029756834 -0500
@@ -41,8 +41,6 @@
 #include <string>
 #include <list>
 
-#include <ace/OS.h>
-
 namespace EMANE
 {
   /**
diff -Naur emane-0.6.4/src/emaneeventd/eventagentfactory.h emane-0.6.4-patched/src/emaneeventd/eventagentfactory.h
--- emane-0.6.4/src/emaneeventd/eventagentfactory.h	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/src/emaneeventd/eventagentfactory.h	2011-03-03 13:08:30.033090858 -0500
@@ -38,7 +38,7 @@
 
 #include "emaneutils/factoryexception.h"
 
-#include <ace/OS.h>
+#include <ace/OS_NS_dlfcn.h>
 
 namespace EMANE
 {
diff -Naur emane-0.6.4/src/emaneeventd/eventdaemonconfiguration.cc emane-0.6.4-patched/src/emaneeventd/eventdaemonconfiguration.cc
--- emane-0.6.4/src/emaneeventd/eventdaemonconfiguration.cc	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/src/emaneeventd/eventdaemonconfiguration.cc	2011-03-03 13:08:30.029756834 -0500
@@ -39,7 +39,6 @@
 #include "configurationparser.h"
 #include "logservice.h"
 
-#include <ace/OS.h>
 #include <utility>
 
 EMANE::EventDaemonConfiguration::EventDaemonConfiguration(const std::string &sFile)
diff -Naur emane-0.6.4/src/emanetransportd/transportdaemonconfiguration.cc emane-0.6.4-patched/src/emanetransportd/transportdaemonconfiguration.cc
--- emane-0.6.4/src/emanetransportd/transportdaemonconfiguration.cc	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/src/emanetransportd/transportdaemonconfiguration.cc	2011-03-03 13:08:30.023088786 -0500
@@ -39,8 +39,6 @@
 
 #include "logservice.h"
 
-#include <ace/OS.h>
-
 /*
  * Constructor
  */
diff -Naur emane-0.6.4/src/emanetransportd/transportfactory.h emane-0.6.4-patched/src/emanetransportd/transportfactory.h
--- emane-0.6.4/src/emanetransportd/transportfactory.h	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/src/emanetransportd/transportfactory.h	2011-03-03 13:08:30.019754763 -0500
@@ -38,7 +38,7 @@
 
 #include "emaneutils/factoryexception.h"
 
-#include <ace/OS.h>
+#include <ace/OS_NS_dlfcn.h>
 
 namespace EMANE
 {
diff -Naur emane-0.6.4/src/emanetransportd/transportinstanceconfiguration.h emane-0.6.4-patched/src/emanetransportd/transportinstanceconfiguration.h
--- emane-0.6.4/src/emanetransportd/transportinstanceconfiguration.h	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/src/emanetransportd/transportinstanceconfiguration.h	2011-03-03 13:08:30.019754763 -0500
@@ -41,8 +41,6 @@
 #include <string>
 #include <list>
 
-#include <ace/OS.h>
-
 namespace EMANE
 {
   /**
diff -Naur emane-0.6.4/src/layerfactory.h emane-0.6.4-patched/src/layerfactory.h
--- emane-0.6.4/src/layerfactory.h	2010-09-22 09:38:03.000000000 -0400
+++ emane-0.6.4-patched/src/layerfactory.h	2011-03-03 13:08:30.026422810 -0500
@@ -37,7 +37,7 @@
 #include "emaneutils/factoryexception.h"
 #include "emaneutils/componenttypes.h"
 
-#include <ace/OS.h>
+#include <ace/OS_NS_dlfcn.h>
 
 namespace EMANE
 {
diff -Naur emane-0.6.4/transports/raw/rawtransport.cc emane-0.6.4-patched/transports/raw/rawtransport.cc
--- emane-0.6.4/transports/raw/rawtransport.cc	2010-09-22 09:38:02.000000000 -0400
+++ emane-0.6.4-patched/transports/raw/rawtransport.cc	2011-03-03 13:08:29.933070160 -0500
@@ -31,8 +31,6 @@
  *
  */
 
-#include <ace/OS.h>
-
 #ifdef WIN32
 #include <winsock2.h>
 #include <iphlpapi.h>
diff -Naur emane-0.6.4/transports/virtual/tuntap.cc emane-0.6.4-patched/transports/virtual/tuntap.cc
--- emane-0.6.4/transports/virtual/tuntap.cc	2010-09-22 09:38:02.000000000 -0400
+++ emane-0.6.4-patched/transports/virtual/tuntap.cc	2011-03-03 13:08:29.939738206 -0500
@@ -34,7 +34,6 @@
 #include <ace/SOCK_Dgram.h>
 #include <ace/Handle_Set.h>
 #include <ace/INET_Addr.h>
-#include <ace/OS.h>
 
 
 #include "tuntap.h"
