--- dahdi-linux-2.2.0.2.old/build_tools/genmodconf	2008-07-02 04:55:52.000000000 +0800
+++ dahdi-linux-2.2.0.2.new/build_tools/genmodconf	2009-08-14 15:04:28.286078867 +0800
@@ -32,7 +32,7 @@
     cmdopts=\$CMDLINE_OPTS
 fi
 if [ -d ${1}/etc/modprobe.d ]; then
-    target=${1}/etc/modprobe.d/dahdi
+    target=${1}/etc/modprobe.d/dahdi.conf
 elif [ -d ${1}/etc/modules.d ]; then
     target=${1}/etc/modules.d/dahdi
 elif [ -f ${1}/etc/modprobe.conf ]; then
