diff -rupN ./tinyos-tools-1.4.0/tools/configure.ac ../src_orig/tinyos-tools-1.4.0/tools/configure.ac
--- ./tinyos-tools-1.4.0/tools/configure.ac	2011-02-20 16:00:25.154647378 +0100
+++ ../src_orig/tinyos-tools-1.4.0/tools/configure.ac	2008-12-28 06:19:00.000000000 +0100
@@ -26,7 +26,7 @@ if test -z "$pathperl" ; then
   AC_MSG_ERROR(I can't find perl); 
 fi
 
-AC_PATH_PROG(pathpython, python)
+AC_PATH_PROG(pathpython, python2)
 if test -z "$pathpython" ; then
   AC_MSG_ERROR(I can't find python); 
 fi
diff -rupN ./tinyos-tools-1.4.0/tools/platforms/intelmote2/openocd/imote2-ocd-program.py ../src_orig/tinyos-tools-1.4.0/tools/platforms/intelmote2/openocd/imote2-ocd-program.py
--- ./tinyos-tools-1.4.0/tools/platforms/intelmote2/openocd/imote2-ocd-program.py	2011-02-20 15:57:13.994647378 +0100
+++ ../src_orig/tinyos-tools-1.4.0/tools/platforms/intelmote2/openocd/imote2-ocd-program.py	2008-05-29 22:23:47.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # imote2-ocd-program.py
 # 
diff -rupN ./tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/elf.py ../src_orig/tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/elf.py
--- ./tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/elf.py	2011-02-20 15:55:20.854647379 +0100
+++ ../src_orig/tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/elf.py	2006-07-12 19:00:39.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import struct
 
 # ELF object file reader
diff -rupN ./tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/serial/__init__.py ../src_orig/tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/serial/__init__.py
--- ./tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/serial/__init__.py	2011-02-20 15:56:14.434647378 +0100
+++ ../src_orig/tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/serial/__init__.py	2008-12-15 02:46:14.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python2 
 #portable serial port access with python
 #this is a wrapper module for different platform implementations
 #
diff -rupN ./tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/serial/serialposix.py ../src_orig/tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/serial/serialposix.py
--- ./tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/serial/serialposix.py	2011-02-20 15:56:28.164647377 +0100
+++ ../src_orig/tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/serial/serialposix.py	2008-12-15 02:46:14.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #module for serial IO for POSIX compatible systems, like Linux
 #see __init__.py
 #
diff -rupN ./tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/tos-bsl.in ../src_orig/tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/tos-bsl.in
--- ./tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/tos-bsl.in	2011-02-20 15:55:29.487980712 +0100
+++ ../src_orig/tinyos-tools-1.4.0/tools/platforms/msp430/pybsl/tos-bsl.in	2006-07-12 19:00:40.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Serial Bootstrap Loader software for the MSP430 embedded proccessor.
 #
 # (C) 2001-2003 Chris Liechti <cliechti@gmx.net>
diff -rupN ./tinyos-tools-1.4.0/tools/tinyos/misc/tos-dump.py ../src_orig/tinyos-tools-1.4.0/tools/tinyos/misc/tos-dump.py
--- ./tinyos-tools-1.4.0/tools/tinyos/misc/tos-dump.py	2011-02-20 15:57:55.864647379 +0100
+++ ../src_orig/tinyos-tools-1.4.0/tools/tinyos/misc/tos-dump.py	2008-12-15 01:50:37.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import sys
 import tos
diff -rupN ./tinyos-tools-1.4.0/tools/tinyos/tosthreads/tosthreads_standard_api.py ../src_orig/tinyos-tools-1.4.0/tools/tinyos/tosthreads/tosthreads_standard_api.py
--- ./tinyos-tools-1.4.0/tools/tinyos/tosthreads/tosthreads_standard_api.py	2011-02-20 15:58:45.487980711 +0100
+++ ../src_orig/tinyos-tools-1.4.0/tools/tinyos/tosthreads/tosthreads_standard_api.py	2009-09-10 21:36:33.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Copyright (c) 2008 Johns Hopkins University.
 # All rights reserved.
