--- mud-waiting.script.old	2012-10-31 08:45:46.902204280 +0100
+++ mud-waiting.script	2012-10-31 08:49:54.267377646 +0100
@@ -73,9 +73,10 @@
    inc = progress;
    time=Math.Int(duration);
    percent=Math.Int(progress * 100);
-   progress_status = "Ich sitze hier seit " + time + " Sekunden und bin erst bei " + percent + " %";
+#   progress_status = "Ich sitze hier seit " + time + " Sekunden und bin erst bei " + percent + " %";
 #   progress_status = "Seit " + time + " Sekunden gebe ich mein Bestes und bin erst bei " + percent + " %";
-#   progress_status = "I've waited " + time + "seconds and are now on "+ percent + "% thoug";
+   progress_status = "I've waited " + time + "seconds and are now on "+ percent + "% thoug";
+#   progress_status = time + " másodperce várok és most " + precent + "%-nál tartok";
    progress_image = Image.Text(progress_status, 1.0, 0.5, 0.25);
    progress_message.SetImage(progress_image);
    progress_message.SetPosition(Window.GetX() + (win_width - progress_image.GetWidth()) / 2, Window.GetY() + win_height * 0.93, 10);
