--- mkv-extractor-gui.sh	2011-11-04 00:18:43.000000000 +0100
+++ mkv-extractor-gui2.sh	2011-11-04 02:26:25.000000000 +0100
@@ -1013,12 +1013,12 @@
 MimeType=video/x-matroska
 Actions=ShowCover
 Encoding=UTF-8
-Icon=/usr/share/mkv-extractor-gui/mkv-extractor-gui.png
+Icon=/opt/mkv-extractor-gui/mkv-extractor-gui.png
 
 [Desktop Action ShowCover]
-Icon=/usr/share/mkv-extractor-gui/mkv-extractor-gui.png
+Icon=/opt/mkv-extractor-gui/mkv-extractor-gui.png
 Name=MKV Extractor Gui
-Exec=mkv-extractor-gui "%U" ' > "${adresse}/mkv_extractor_gui.desktop"
+Exec=/opt/mkv-extractor-gui/go_mkv-extractor-gui.sh
 
 		# Mode executable pour le service menu
 		chmod +x "${adresse}mkv_extractor_gui.desktop"
