--- mkv-extractor-gui.sh	2011-11-04 00:18:43.000000000 +0100
+++ mkv-extractor-gui_patched.sh	2011-11-06 03:16:29.000000000 +0100
@@ -44,7 +44,11 @@
 
 function _encapsulage { encapsulage=${1}; }
 
-function _fichier_tmp { fichier_tmp=${1}; }
+function _fichier_tmp 
+{
+fichier_tmp=${@}
+sed -i "s@^fichier_tmp=.*@fichier_tmp=\"${fichier_tmp}\"@" "${config}"
+}
 
 ##################################################
 ##### ETAPE 1 : SELECTION DU FICHIER/DOSSIER #####
@@ -447,7 +451,7 @@
 fichiers="rm ${fichiers#*\"*\"}"
 
 # Suppression des pistes temporaires si la variable contient des noms de fichiers...
-if [[ ${fichiers} != "rm " && ${fichier_tmp} == True ]]
+if [[ ${encapsulage} == True && ${fichiers} != "rm " && ${fichier_tmp} == True ]]
 then
 	eval ${fichiers}
 	echo "TEXT@@ENDTAG@@_textedebug@@$(eval_gettext $'<blue>Deleting</blue> temporary files.'\\n)"
@@ -1018,7 +1022,7 @@
 [Desktop Action ShowCover]
 Icon=/usr/share/mkv-extractor-gui/mkv-extractor-gui.png
 Name=MKV Extractor Gui
-Exec=mkv-extractor-gui "%U" ' > "${adresse}/mkv_extractor_gui.desktop"
+Exec=/opt/mkv-extractor-gui/go_mkv-extractor-gui.sh ' > "${adresse}/mkv_extractor_gui.desktop"
 
 		# Mode executable pour le service menu
 		chmod +x "${adresse}mkv_extractor_gui.desktop"
