diff -r -u kdelibs/admin/debianrules src/kdelibs/admin/debianrules
--- kdelibs/admin/debianrules	2011-08-21 06:15:36.000000000 +0200
+++ src/kdelibs/admin/debianrules	2012-01-04 15:28:20.000000000 +0100
@@ -16,7 +16,7 @@
 
 $kde_cgidir	=	"$kde_prefix/lib/cgi-bin";
 $kde_confdir	=	"$sysconfdir/trinity";
-$kde_htmldir	=	"$kde_prefix/share/doc/kde/HTML";
+$kde_htmldir	=	"$kde_prefix/share/doc/HTML";
 
 if (defined $ENV{DEB_BUILD_OPTIONS} &&
     $ENV{DEB_BUILD_OPTIONS} =~ /\bnostrip\b/) {
diff -r -u kdelibs/cmake/modules/TDESetupPaths.cmake src/kdelibs/cmake/modules/TDESetupPaths.cmake
--- kdelibs/cmake/modules/TDESetupPaths.cmake	2011-08-21 06:15:37.000000000 +0200
+++ src/kdelibs/cmake/modules/TDESetupPaths.cmake	2012-01-04 15:28:49.000000000 +0100
@@ -41,7 +41,7 @@
   _tde_internal_setup_path( PLUGIN_INSTALL_DIR        "${LIB_INSTALL_DIR}/trinity"                     "The subdirectory relative to the install prefix where plugins will be installed (default is ${LIB_INSTALL_DIR}/trinity)" )
   _tde_internal_setup_path( CONFIG_INSTALL_DIR        "${SHARE_INSTALL_PREFIX}/config"              "The config file install dir" )
   _tde_internal_setup_path( DATA_INSTALL_DIR          "${SHARE_INSTALL_PREFIX}/apps"                "The parent directory where applications can install their data" )
-  _tde_internal_setup_path( HTML_INSTALL_DIR          "${SHARE_INSTALL_PREFIX}/doc/kde/HTML"        "The HTML install dir for documentation" )
+  _tde_internal_setup_path( HTML_INSTALL_DIR          "${SHARE_INSTALL_PREFIX}/doc/HTML"        "The HTML install dir for documentation" )
   _tde_internal_setup_path( ICON_INSTALL_DIR          "${SHARE_INSTALL_PREFIX}/icons"               "The icon install dir (default ${SHARE_INSTALL_PREFIX}/share/icons/)" )
   _tde_internal_setup_path( KCFG_INSTALL_DIR          "${SHARE_INSTALL_PREFIX}/config.kcfg"         "The install dir for kconfig files" )
   _tde_internal_setup_path( LOCALE_INSTALL_DIR        "${SHARE_INSTALL_PREFIX}/locale"              "The install dir for translations" )
diff -r -u kdelibs/kdecore/kstandarddirs.cpp src/kdelibs/kdecore/kstandarddirs.cpp
--- kdelibs/kdecore/kstandarddirs.cpp	2011-08-21 06:15:32.000000000 +0200
+++ src/kdelibs/kdecore/kstandarddirs.cpp	2012-01-04 15:30:24.000000000 +0100
@@ -1030,7 +1030,7 @@
     if (!strcmp(type, "html-bundle"))
 	return "share/doc-bundle/HTML/";
     if (!strcmp(type, "html"))
-	return "share/doc/kde/HTML/";
+	return "share/doc/HTML/";
     if (!strcmp(type, "icon"))
 	return "share/icons/";
     if (!strcmp(type, "config"))
