--- common.mk.orig	2010-01-16 13:27:29.395745938 +0200
+++ common.mk	2010-01-16 13:36:30.409719894 +0200
@@ -40,7 +40,7 @@
 # Add new SM Versions here as devices with new Compute Capability are released
 SM_VERSIONS := sm_10 sm_11 sm_12 sm_13
 
-CUDA_INSTALL_PATH ?= /opt/cuda
+CUDA_INSTALL_PATH ?= /usr
 
 ifdef cuda-install
 	CUDA_INSTALL_PATH := $(cuda-install)
@@ -61,7 +61,7 @@
 ROOTOBJDIR ?= obj
 LIBDIR     := $(ROOTDIR)/../lib
 COMMONDIR  := $(ROOTDIR)/../common
-NVIDIA_SDK_PATH := /opt/cuda/NVIDIA_CUDA_SDK
+NVIDIA_SDK_PATH := /usr/share/cuda-sdk
 
 # Compilers
 NVCC       := $(CUDA_INSTALL_PATH)/bin/nvcc 
@@ -99,7 +99,7 @@
 	-Wmain \
 
 # Compiler-specific flags
-NVCCFLAGS := 
+NVCCFLAGS := --compiler-options "-fpermissive -fno-inline"
 CXXFLAGS  := $(CXXWARN_FLAGS)
 CFLAGS    := $(CWARN_FLAGS)
 
