--- Makefile.orig	2010-08-05 15:34:59.768935507 +0200
+++ Makefile	2010-08-05 15:39:37.026057688 +0200
@@ -1,5 +1,8 @@
-export prefix=/usr/local
+DESTDIR ?=
+export prefix=$(DESTDIR)/usr
 export milena_dir=$(prefix)/share/milena
 
+default:
 install:
-	zcat pl_basewords.dat.gz > $(milena_dir)-words/pl_basewords.dat
+	install -dm 755 "$(milena_dir)-words/"
+	zcat pl_basewords.dat.gz > "$(milena_dir)-words/pl_basewords.dat"
