Index: NetworkEditor/src/widgets/EmbededWidgets.hxx
===================================================================
--- NetworkEditor/src/widgets/EmbededWidgets.hxx	(revision 14241)
+++ NetworkEditor/src/widgets/EmbededWidgets.hxx	(working copy)
@@ -9,8 +9,9 @@
 
 class EmbededWidgetCreatorBase
 {
+protected:
+	typedef std::string Key;
 private:
-	typedef std::string Key;
 	typedef std::map<Key, EmbededWidgetCreatorBase*> Creators;
 	static Creators & creators();
 protected:
