--- clang_complete/autoload/getopts/gcc.vim	2012-03-29 18:02:17.970787408 +0200
+++ clang_complete/autoload/getopts/gcc.vim_new	2012-03-29 18:26:45.180142707 +0200
@@ -1,8 +1,7 @@
 " clang_complete gcc's include paths finder
 " Author: xaizek
 
-let s:scr = expand('<sfile>')
-let s:cache_path = fnamemodify(s:scr, ':p:h')
+let s:cache_path = eval('$HOME') . '/.vim' 
 
 function! getopts#gcc#getopts()
   call s:DetemineFileType()
