--- clang_complete/autoload/getopts/cache.vim	2012-10-28 14:48:28.368253172 +0100
+++ clang_complete/autoload/getopts/cache.vim_new	2012-10-28 14:55:32.488301453 +0100
@@ -1,8 +1,7 @@
 " clang_complete include paths caching helper
 " Author: xaizek, leszekswirski
 
-let s:scr = expand('<sfile>')
-let s:cache_path = fnamemodify(s:scr, ':p:h')
+let s:cache_path = eval('$HOME') . '/.vim'
 
 function! getopts#cache#getcachedopts(ext, optfunc)
   let s:cache = s:cache_path . '/' . &filetype . '.' . a:ext . '.cache'
