--- clang_complete/plugin/clang_complete.vim	2011-12-14 20:56:54.000000000 +0100
+++ clang_complete/plugin/clang_complete.vim_new	2011-12-14 20:57:18.000000000 +0100
@@ -54,6 +54,10 @@
     let g:clang_user_options = ''
   endif
 
+  if !exists('g:clang_library_path')
+    let g:clang_library_path = '/usr/lib/llvm'
+  endif
+
   " Only use libclang if the user clearly show intent to do so for now
   if !exists('g:clang_use_library')
     let g:clang_use_library = (has('python') && exists('g:clang_library_path'))
