Index: epgsearch-0.9.24/epgsearch.c
===================================================================
--- epgsearch-0.9.24.orig/epgsearch.c
+++ epgsearch-0.9.24/epgsearch.c
@@ -66,6 +66,10 @@ The project's page is at http://winni.vd
 #include "menu_announcelist.h"
 #include "confdloader.h"
 
+#if APIVERSNUM >= 10503
+#include <langinfo.h>
+#endif
+
 static const char VERSION[]        = "0.9.24";
 static const char DESCRIPTION[]    =  trNOOP("search the EPG for repeats and more");
 
