diff -ruN epgsearch-1.0.1.beta5/uservars.h epgsearch-1.0.1.beta5.edit/uservars.h
--- epgsearch-1.0.1.beta5/uservars.h	2013-03-25 20:30:59.000000000 +0100
+++ epgsearch-1.0.1.beta5.edit/uservars.h	2013-10-19 13:55:15.458605893 +0200
@@ -538,7 +538,11 @@
     cVideodirVar() : cInternalVar("videodir") {}
     string Evaluate(const cEvent*, bool escapeStrings = false)
 	{
+#if APIVERSNUM > 20101
+	    return cVideoDirectory::Name();
+#else
 	    return VideoDirectory;
+#endif
 	}
 };
 
