--- fusiondirectory-1.0.7.3/contrib/apache/fusiondirectory-apache.conf	2014-03-28 15:37:34.000000000 +0100
+++ fusiondirectory-1.0.7.3.patch/contrib/apache/fusiondirectory-apache.conf	2014-04-16 13:50:25.791308676 +0200
@@ -1,5 +1,5 @@
 # Include FusionDirectory to your web service
-Alias /fusiondirectory /var/www/fusiondirectory/html
+Alias /fusiondirectory /usr/share/webapps/fusiondirectory/html
 
 <IfModule mod_php5.c>
 <Location /fusiondirectory>
