diff -Naur HSFFIG-1.1.3.orig/HSFFIG.cabal HSFFIG-1.1.3/HSFFIG.cabal
--- HSFFIG-1.1.3.orig/HSFFIG.cabal	2010-07-15 04:04:11.000000000 +0200
+++ HSFFIG-1.1.3/HSFFIG.cabal	2012-03-04 19:28:08.686550273 +0100
@@ -45,6 +45,7 @@
 Main-is:	ffipkg.hs
 Hs-source-dirs:	programs
 Ghc-options:    -threaded
+Extensions:     BangPatterns
 Other-modules:	C_Lexer, ProcHdr, TokenOps, C_BNF, SplitBounds, 
                 WriteHsc, Template, HsfUtils,
                 HGmain, SPmain, Makefile Setupfile SetupfileNew
@@ -52,6 +53,7 @@
 Executable:	hsffig
 Main-is:	hsffig.hs
 Ghc-options:    -threaded
+Extensions:     BangPatterns
 Hs-source-dirs:	programs
 Other-modules:	HGmain,
                 C_Lexer, ProcHdr, TokenOps, C_BNF, SplitBounds, WriteHsc, Template, HsfUtils
@@ -59,6 +61,7 @@
 Executable:     testparser
 Main-is:        testparser.hs
 Ghc-options:    -threaded
+Extensions:     BangPatterns
 Hs-source-dirs: programs
 Other-modules:  C_Lexer, ProcHdr, TokenOps, C_BNF, SplitBounds
 
