diff -ur libselinux-2.1.13.orig/include/selinux/selinux.h libselinux-2.1.13/include/selinux/selinux.h
--- libselinux-2.1.13.orig/include/selinux/selinux.h	2013-04-25 21:10:55.088063028 +0200
+++ libselinux-2.1.13/include/selinux/selinux.h	2013-04-25 21:24:33.761081941 +0200
@@ -498,6 +498,7 @@
 
 /* These functions return the paths to specific files under the 
    policy root directory. */
+extern const char *selinux_current_policy_path(void);
 extern const char *selinux_binary_policy_path(void);
 extern const char *selinux_failsafe_context_path(void);
 extern const char *selinux_removable_context_path(void);
diff -ur libselinux-2.1.13.orig/man/man3/selinux_binary_policy_path.3 libselinux-2.1.13/man/man3/selinux_binary_policy_path.3
--- libselinux-2.1.13.orig/man/man3/selinux_binary_policy_path.3	2013-04-25 21:10:55.077063028 +0200
+++ libselinux-2.1.13/man/man3/selinux_binary_policy_path.3	2013-04-25 21:13:45.562066967 +0200
@@ -1,6 +1,6 @@
 .TH "selinux_binary_policy_path" "3" "15 November 2004" "dwalsh@redhat.com" "SELinux API Documentation"
 .SH "NAME"
-selinux_path, selinux_policy_root, selinux_binary_policy_path,
+selinux_path, selinux_policy_root, selinux_binary_policy_path, selinux_current_policy_path,
 selinux_failsafe_context_path, selinux_removable_context_path,
 selinux_default_context_path, selinux_user_contexts_path,
 selinux_file_context_path, selinux_media_context_path,
@@ -17,6 +17,8 @@
 .sp
 .B const char *selinux_binary_policy_path(void);
 .sp
+.B const char *selinux_current_policy_path(void);
+.sp
 .B const char *selinux_failsafe_context_path(void);
 .sp
 .B const char *selinux_removable_context_path(void);
@@ -55,6 +57,9 @@
 .BR selinux_binary_policy_path ()
 returns the binary policy file loaded into kernel.
 .sp
+.BR selinux_current_policy_path ()
+returns the currently loaded policy file from the kernel.
+.sp
 .BR selinux_default_type_path ()
 returns the context file mapping roles to default types.
 .sp
Only in libselinux-2.1.13/src: .selinux_config.c.swp
diff -ur libselinux-2.1.13.orig/src/selinux_config.c libselinux-2.1.13/src/selinux_config.c
--- libselinux-2.1.13.orig/src/selinux_config.c	2013-04-25 21:10:55.086063028 +0200
+++ libselinux-2.1.13/src/selinux_config.c	2013-04-25 21:33:24.130094193 +0200
@@ -8,6 +8,7 @@
 #include <limits.h>
 #include <unistd.h>
 #include <pthread.h>
+#include "policy.h"
 #include "selinux_internal.h"
 #include "get_default_type_internal.h"
 
@@ -303,6 +304,28 @@
 
 hidden_def(selinux_binary_policy_path)
 
+const char *selinux_current_policy_path(void)
+{
+	int rc = 0;
+	int vers = 0;
+	static char policy_path[PATH_MAX];
+
+	snprintf(policy_path, sizeof(policy_path), "%s/policy", selinux_mnt);
+	if (access(policy_path, F_OK) == 0) {
+		vers = security_policyvers();
+		do {
+			/* Check prior versions to see if old policy is available */
+			snprintf(policy_path, sizeof(policy_path), "%s.%d",
+				selinux_binary_policy_path(), vers);
+		} while ((rc = access(policy_path, F_OK)) && --vers > 0);
+
+		if (rc) return NULL;
+	}
+	return policy_path;
+}
+
+hidden_def(selinux_current_policy_path)
+
 const char *selinux_file_context_path(void)
 {
 	return get_path(FILE_CONTEXTS);
diff -ur libselinux-2.1.13.orig/src/selinux_internal.h libselinux-2.1.13/src/selinux_internal.h
--- libselinux-2.1.13.orig/src/selinux_internal.h	2013-04-25 21:10:55.086063028 +0200
+++ libselinux-2.1.13/src/selinux_internal.h	2013-04-25 21:18:55.196074119 +0200
@@ -60,6 +60,7 @@
     hidden_proto(security_setenforce)
     hidden_proto(security_deny_unknown)
     hidden_proto(selinux_boolean_sub)
+    hidden_proto(selinux_current_policy_path)
     hidden_proto(selinux_binary_policy_path)
     hidden_proto(selinux_booleans_subs_path)
     hidden_proto(selinux_default_context_path)
