--- desktop-art/DesktopControl.py	2015-04-06 10:38:04.541310787 +0100
+++ desktop-art/DesktopControl.py	2015-04-06 11:02:19.273729727 +0100
@@ -287,24 +287,7 @@
         ccmask = cairo.Context(surface)
         roundedrec(ccmask, int(x_trans), 0, cover_area_size, cover_area_size, self.conf['roundness'])
         ccmask.fill()
-        # bug in python-cairo, upstream is patched, but only for python3-cairo
-        # https://bugs.launchpad.net/ubuntu/+source/pygobject/+bug/1028115
-        # https://bugzilla.gnome.org/show_bug.cgi?id=667959
-        # https://bugs.freedesktop.org/show_bug.cgi?id=44336
-        # for the moment wrap this into a try except so that it fails gracefully
-        # need to test this on a distro patched and only running python3-cairo
-        
-        #21/04/14 - crash observed on 14.04 - currently presume this is a bug so
-        #lets work around this by a test with the pygobject-version 3.12
-        
-        try:
-            print (desktop_rb3compat.pygobject_version())
-            if desktop_rb3compat.pygobject_version() < 3.12:
-                region = Gdk.cairo_region_create_from_surface(surface)
-                self.get_parent().input_shape_combine_region(region)
-        except:
-            pass
-        
+
         # Draw border
         if self.draw_border:
             print("drawborder")
