Index: qxmlrpc-0.0.svn6/xmlrpc/stable.h
===================================================================
--- qxmlrpc-0.0.svn6.orig/xmlrpc/stable.h	2010-06-04 18:23:58.000000000 +0400
+++ qxmlrpc-0.0.svn6/xmlrpc/stable.h	2010-06-04 18:24:20.000000000 +0400
@@ -2,7 +2,6 @@
 #ifndef STABLE_H
 #define STABLE_H
 	#if defined(__cplusplus) and not defined(__OBJC__)		
-		#include <QtGui>
 		#include <QtXml>
 		#include <QtCore>
 		#include <QtNetwork>
Index: qxmlrpc-0.0.svn6/xmlrpc/xmlrpc.pro
===================================================================
--- qxmlrpc-0.0.svn6.orig/xmlrpc/xmlrpc.pro	2010-06-04 18:23:54.000000000 +0400
+++ qxmlrpc-0.0.svn6/xmlrpc/xmlrpc.pro	2010-06-04 18:24:12.000000000 +0400
@@ -10,6 +10,7 @@
 DEPENDPATH += .
 INCLUDEPATH += . ..
 
+QT -= gui
 QT += xml
 QT += network 
 
