diff -aur k2pdfopt_v2.14.old/CMakeLists.txt k2pdfopt_v2.14.new/CMakeLists.txt
--- k2pdfopt_v2.14.old/CMakeLists.txt	2013-12-31 13:08:46.000000000 -0300
+++ k2pdfopt_v2.14.new/CMakeLists.txt	2014-03-14 23:34:22.974795840 -0300
@@ -52,6 +52,7 @@
 include(FindJasper)
 if(JASPER_FOUND)
   set(HAVE_JASPER_LIB 1)
+  set(K2PDFOPT_LIB ${K2PDFOPT_LIB} -ljasper)
 endif(JASPER_FOUND)
 
 # paths from willuslib/wgs.c
@@ -73,6 +74,7 @@
 # libfreetype6 (>= 2.3.9), libjbig2dec0, libjpeg8 (>= 8c), libx11-6, libxext6, zlib1g (>= 1:1.2.0)
 # MUPDF_STATIC_LDFLAGS misses mupdf-js-none, and doubles libs ...
 
+SET( ENV{PKG_CONFIG_PATH} $ENV{PWD}/mupdf )
 pkg_check_modules(MUPDF mupdf)
 if(MUPDF_FOUND)
   set(HAVE_MUPDF_LIB 1) 
@@ -80,7 +82,7 @@
   include_directories(SYSTEM ${MUPDF_INCLUDEDIR})
   message(STATUS "mupdf libraries: ${MUPDF_LDFLAGS}")
   set(K2PDFOPT_LIB ${K2PDFOPT_LIB} ${MUPDF_LDFLAGS} 
-    -lmupdf-js-none -lopenjpeg -ljbig2dec -ljpeg -lfreetype
+    -lmupdf-js-none -lopenjp2 -ljbig2dec -ljpeg -lfreetype
   )
 endif(MUPDF_FOUND)
 
diff -aur k2pdfopt_v2.14.old/willuslib/bmp.c k2pdfopt_v2.14.new/willuslib/bmp.c
--- k2pdfopt_v2.14.old/willuslib/bmp.c	2013-09-16 13:17:59.000000000 -0300
+++ k2pdfopt_v2.14.new/willuslib/bmp.c	2014-03-14 22:07:30.465240580 -0300
@@ -37,7 +37,7 @@
 #include <jpeglib.h>
 #endif
 #ifdef HAVE_JASPER_LIB
-#include <jasper.h>
+#include <jasper/jasper.h>
 #endif
 
 #define BOUND(x,xmin,xmax)  if ((x)<(xmin)) (x)=(xmin); else { if ((x)>(xmax)) (x)=(xmax); }
