diff -aur k2pdfopt_v2.14.old/CMakeLists.txt k2pdfopt_v2.14.new/CMakeLists.txt
--- k2pdfopt_v2.14.old/CMakeLists.txt	2013-12-31 13:08:46.000000000 -0300
+++ k2pdfopt_v2.14.new/CMakeLists.txt	2014-03-14 17:44:19.688587876 -0300
@@ -52,6 +52,7 @@
 include(FindJasper)
 if(JASPER_FOUND)
   set(HAVE_JASPER_LIB 1)
+  set(K2PDFOPT_LIB ${K2PDFOPT_LIB} -ljasper)
 endif(JASPER_FOUND)
 
 # paths from willuslib/wgs.c
diff -aur k2pdfopt_v2.14.old/willuslib/bmp.c k2pdfopt_v2.14.new/willuslib/bmp.c
--- k2pdfopt_v2.14.old/willuslib/bmp.c	2013-09-16 13:17:59.000000000 -0300
+++ k2pdfopt_v2.14.new/willuslib/bmp.c	2014-03-14 17:45:24.896582312 -0300
@@ -37,7 +37,7 @@
 #include <jpeglib.h>
 #endif
 #ifdef HAVE_JASPER_LIB
-#include <jasper.h>
+#include <jasper/jasper.h>
 #endif
 
 #define BOUND(x,xmin,xmax)  if ((x)<(xmin)) (x)=(xmin); else { if ((x)>(xmax)) (x)=(xmax); }
