--- a/usr/share/applications/wine-browsedrive.desktop	2011-09-16 05:48:57.000000000 +0100
+++ b/usr/share/applications/wine-browsedrive.desktop	2011-09-27 16:04:38.046846572 +0100
@@ -33,7 +33,8 @@
 Comment[hu]=A virtuális C: meghajtó tallózása
 Comment[it]=Esplora l'unità virtuale C:
 Comment[nl]=Bekijk uw virtuele C: schijf
-Exec=xdg-open .wine/dosdevices/c:
+Exec=wine winebrowser c:\\\\
+#Exec=xdg-open ~/.wine/dosdevices/c:
 Terminal=false
 Type=Application
 Icon=folder-wine
