diff -r e43ada2bd7fc Makefile.am
--- a/Makefile.am	Thu Nov 26 17:31:21 2009 +0200
+++ b/Makefile.am	Thu Nov 26 17:32:13 2009 +0200
@@ -3,7 +3,7 @@
 
 @SET_MAKE@
 
-export LIBDIR=$(libdir)/pptpd
+export LIBDIR=$(DESTDIR)/$(libdir)/pptpd
 INCLUDES = -I.
 ## Change this if you don't have gcc
 ## -pedantic removed for now (OpenBSD header files).
diff -r e43ada2bd7fc plugins/Makefile
--- a/plugins/Makefile	Thu Nov 26 17:31:21 2009 +0200
+++ b/plugins/Makefile	Thu Nov 26 17:32:13 2009 +0200
@@ -18,7 +18,7 @@
 %.so: %.c
 	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) $^ $(LDADD)
 
-LIBDIR	?= $(DESTDIR)$(prefix)/lib/pptpd
+LIBDIR	= $(DESTDIR)$(prefix)/lib/pptpd
 
 install: $(PLUGINS)
 	$(INSTALL) -d $(LIBDIR)
