Add error messages inline.

diff -r 3aa368e1a120 bitten/build/ctools.py
--- a/bitten/build/ctools.py	Mon Jan 18 14:40:12 2038 +0009
+++ b/bitten/build/ctools.py	Fri Nov 06 14:04:38 2009 +0100
@@ -245,7 +245,7 @@
                             continue
                         suiteName = suite.children ('SUITE_NAME').next().gettext()
                         info ("%s [%s]" % ("*" * (57 - len (suiteName)), suiteName))
-                        testResults = set()
+                        testResults = {}
                         for record in suite.children ('CUNIT_RUN_TEST_RECORD'):
                             for result in record.children():
                                 if result.name not in ('CUNIT_RUN_TEST_SUCCESS', 'CUNIT_RUN_TEST_FAILURE'):
@@ -253,27 +253,33 @@
                                 testName = result.children ('TEST_NAME').next().gettext()
                                 if testName in testResults:
                                     if result.name == 'CUNIT_RUN_TEST_FAILURE':
-                                        error ("%s(%d): %s"
+                                        test = testResults[testName]
+                                        message = ("%s(%d): %s"
                                                    % (result.children ('FILE_NAME').next().gettext(),
                                                       int (result.children ('LINE_NUMBER').next().gettext()),
                                                       result.children ('CONDITION').next().gettext()))
+                                        error (message)
+                                        test.append (xmlio.Element('traceback')[message])
+                                        test.attr['status'] = 'failure'
                                     continue
-                                testResults.add (testName)
                                 info ("Running %s..." % testName);
                                 test = xmlio.Element('test')
                                 test.attr['fixture'] = suiteName
                                 test.attr['name'] = testName
                                 if result.name == 'CUNIT_RUN_TEST_FAILURE':
-                                    error ("%s(%d): %s"
+                                    message = ("%s(%d): %s"
                                                % (result.children ('FILE_NAME').next().gettext(),
                                                   int (result.children ('LINE_NUMBER').next().gettext()),
                                                   result.children ('CONDITION').next().gettext()))
+                                    error (message)
+                                    test.append (xmlio.Element('traceback')[message])
                                     test.attr['status'] = 'failure'
                                     failed += 1
                                 else:
                                     test.attr['status'] = 'success'
 
                                 results.append(test)
+                                testResults[testName] = test
                                 total += 1
 
             if failed:
