http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=415097

--- mc-4.6.1-old/src/screen.c	2007-03-18 03:00:48.000000000 +0200
+++ mc-4.6.1/src/screen.c	2007-03-18 02:58:32.000000000 +0200
@@ -658,7 +658,7 @@ format_file (WPanel *panel, int file_ind
 			SLsmg_write_nwchars (((wchar_t *) buffer)
 					     + txtlen - n2, n2);
 		    } else
-			SLsmg_write_nwchars ((wchar_t *) buffer, len);
+			SLsmg_write_nwchars (((wchar_t *) buffer) + still, len);
 		} else {
 		    printw ("%*s", still, "");
 		    SLsmg_write_nwchars ((wchar_t *) buffer, txtlen);
