diff -Naur mc-4.6.1.orig/lib/mc.ext.in mc-4.6.1/lib/mc.ext.in
--- mc-4.6.1.orig/lib/mc.ext.in	2005-07-23 18:51:15.000000000 +0200
+++ mc-4.6.1/lib/mc.ext.in	2005-09-11 12:34:41.000000000 +0200
@@ -252,8 +252,8 @@
 	View=%view{ascii} file %f && nm %f
 
 regex/(([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])|\.man)$
-	Open=case %d/%f in */log/*|*/logs/*) cat %f ;; *) nroff @MAN_FLAGS@ @MANDOC@ %f ;; esac | %var{PAGER:more}
-	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) cat %f ;; *) nroff @MAN_FLAGS@ @MANDOC@ %f ;; esac
+	Open=case %d/%f in */log/*|*/logs/*) cat %f ;; *) groff-utf8 -P-c -Tutf8 @MANDOC@ %f ;; esac | %var{PAGER:more}
+	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) cat %f ;; *) groff-utf8 -P-c -Tutf8 @MANDOC@ %f ;; esac
 
 # Troff with me macros.
 # Exception - "read.me" is not a nroff file.
@@ -262,26 +262,26 @@
 	View=
 
 shell/.me
-	Open=nroff @MAN_FLAGS@ -me %f | %var{PAGER:more}
-	View=%view{ascii,nroff} nroff @MAN_FLAGS@ -me %f
+	Open=groff-utf8 -P-c -Tutf8 -me %f | %var{PAGER:more}
+	View=%view{ascii,nroff} groff-utf8 -P-c -Tutf8 -me %f
 
 # Troff with ms macros.
 shell/.ms
-	Open=nroff @MAN_FLAGS@ -ms %f | %var{PAGER:more}
-	View=%view{ascii,nroff} nroff @MAN_FLAGS@ -ms %f
+	Open=groff-utf8 -P-c -Tutf8 -ms %f | %var{PAGER:more}
+	View=%view{ascii,nroff} groff-utf8 -P-c -Tutf8 -ms %f
 
 # Manual page - compressed
 regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])\.g?[Zz]$
-	Open=case %d/%f in */log/*|*/logs/*) gzip -dc %f ;; *) gzip -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac | %var{PAGER:more}
-	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) gzip -dc %f ;; *) gzip -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac
+	Open=case %d/%f in */log/*|*/logs/*) gzip -dc %f ;; *) gzip -dc %f | groff-utf8 -P-c -Tutf8 @MANDOC@ ;; esac | %var{PAGER:more}
+	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) gzip -dc %f ;; *) gzip -dc %f | groff-utf8 -P-c -Tutf8 @MANDOC@ ;; esac
 
 regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])\.bz$
-	Open=case %d/%f in */log/*|*/logs/*) bzip -dc %f ;; *) bzip -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac | %var{PAGER:more}
-	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) bzip -dc %f ;; *) bzip -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac
+	Open=case %d/%f in */log/*|*/logs/*) bzip -dc %f ;; *) bzip -dc %f | groff-utf8 -P-c -Tutf8 @MANDOC@ ;; esac | %var{PAGER:more}
+	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) bzip -dc %f ;; *) bzip -dc %f | groff-utf8 -P-c -Tutf8 @MANDOC@ ;; esac
 
 regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])\.bz2$
-	Open=case %d/%f in */log/*|*/logs/*) bzip2 -dc %f ;; *) bzip2 -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac | %var{PAGER:more}
-	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) bzip2 -dc %f ;; *) bzip2 -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac
+	Open=case %d/%f in */log/*|*/logs/*) bzip2 -dc %f ;; *) bzip2 -dc %f | groff-utf8 -P-c -Tutf8 @MANDOC@ ;; esac | %var{PAGER:more}
+	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) bzip2 -dc %f ;; *) bzip2 -dc %f | groff-utf8 -P-c -Tutf8 @MANDOC@ ;; esac
 
 
 ### Images ###
diff -Naur mc-4.6.1.orig/src/view.c mc-4.6.1/src/view.c
--- mc-4.6.1.orig/src/view.c	2005-09-11 11:35:16.000000000 +0200
+++ mc-4.6.1/src/view.c	2005-09-11 12:29:55.000000000 +0200
@@ -1125,17 +1125,52 @@
                     mblen = 1;
                 }
 
+                offset_type from_save = from;
                 from += mblen - 1;
 
             if (view->viewer_nroff_flag && wc == '\b') {
 #endif /* UTF8 */
+#ifndef UTF8
 		int c_prev;
 		int c_next;
+#else /* UTF8 */
+		wchar_t c_prev;
+		wchar_t c_next;
+#endif /* UTF8 */
 
+#ifndef UTF8
 		if (from + 1 < view->last_byte
 		    && is_printable ((c_next = get_byte (view, from + 1)))
 		    && from > view->first
 		    && is_printable ((c_prev = get_byte (view, from - 1)))
+#else /* UTF8 */
+		/* get next character */
+		c_next = 0;
+		mblen = 0;
+		while (++mblen < MB_LEN_MAX) {
+			if (from + mblen >= view->last_byte) break;
+			mbbuf[mblen - 1] = get_byte (view, from + mblen);
+			int res;
+			memset (&mbs, 0, sizeof (mbs));
+			res = mbrtowc (&c_next, mbbuf, mblen, &mbs);
+			if (res == mblen)
+			    break;
+		}
+		/* get previous character */
+		c_prev = 0;
+		mblen = 0;
+		while (++mblen < MB_LEN_MAX) {
+			if (from_save - mblen <= view->first) break;
+			mbbuf[MB_LEN_MAX - mblen] = get_byte (view, from_save - mblen);
+			int res;
+			memset (&mbs, 0, sizeof (mbs));
+			res = mbrtowc (&c_prev, mbbuf + MB_LEN_MAX - mblen, mblen, &mbs);
+			if (res == mblen)
+			    break;
+		}
+		if (c_next != 0 && is_printable (c_next)
+		    && c_prev != 0 && is_printable (c_prev)
+#endif /* UTF8 */
 		    && (c_prev == c_next || c_prev == '_'
 		        || (c_prev == '+' && c_next == 'o'))) {
 		    if (col <= frame_shift) {
