diff -Naur mc-4.6.1.orig/src/screen.c mc-4.6.1/src/screen.c
--- mc-4.6.1.orig/src/screen.c	2005-05-27 16:19:18.000000000 +0200
+++ mc-4.6.1/src/screen.c	2005-09-05 17:47:45.000000000 +0200
@@ -198,7 +198,7 @@
 
     /* Don't ever show size of ".." since we don't calculate it */
     if (!strcmp (fe->fname, "..")) {
-	return _("UP--DIR");
+	return "       ";
     }
 
 #ifdef HAVE_STRUCT_STAT_ST_RDEV
