diff -Naur mc-4.6.1.orig/src/x11conn.c mc-4.6.1/src/x11conn.c
--- mc-4.6.1.orig/src/x11conn.c	2005-06-07 15:48:47.000000000 +0200
+++ mc-4.6.1/src/x11conn.c	2007-05-03 11:07:54.000000000 +0200
@@ -178,7 +178,8 @@
 {
     Display *retval;
 
-    if (x11_available()) {
+    char *term = getenv("TERM");
+    if (strcmp(term, "linux") && x11_available()) {
 	if (setjmp(x11_exception) == 0) {
 	    longjmp_allowed = TRUE;
 	    retval = func_XOpenDisplay (displayname);
