diff -Naur mc-4.6.1.orig/lib/mc.lib mc-4.6.1/lib/mc.lib
--- mc-4.6.1.orig/lib/mc.lib	2004-09-10 11:23:44.000000000 +0200
+++ mc-4.6.1/lib/mc.lib	2007-08-30 16:06:19.000000000 +0200
@@ -32,16 +32,14 @@
 
 [terminal:linux]
 insert=\e[2~
-f11=\e[23~
-f12=\e[24~
-f13=\e[25~
-f14=\e[26~
-f15=\e[28~
-f16=\e[29~
-f17=\e[31~
-f18=\e[32~
-f19=\e[33~
-f20=\e[34~
+f11=\e[25~
+f12=\e[26~
+f13=\e[28~
+f14=\e[29~
+f15=\e[31~
+f16=\e[32~
+f17=\e[33~
+f18=\e[34~
 kpleft=\eOt
 kpright=\eOv
 kpup=\eOx
@@ -82,6 +80,8 @@
 kpplus=\eOk
 kpminus=\eOm
 kpasterix=\eOj
+kpenter=\eOM
+kpslash=\eOo
 delete=\e[3~
 bs=
 home=\e[1~
@@ -107,6 +107,18 @@
 f19=\e[20;2~
 f20=\e[21;2~
 
+# Vte >= 0.15.
+f11=\eO1;2P
+f12=\eO1;2Q
+f13=\eO1;2R
+f14=\eO1;2S
+
+# Vte patched to send Shift+PgDn, Ctrl+Shift+PgDn, Shift+PgUp, Ctrl+Shift+PgUp.
+pgdn=\e[6;2~
+pgdn=\e[6;6~
+pgup=\e[5;2~
+pgup=\e[5;6~
+
 # Arrows for both keypad modes (application and normal).
 up=\e[A
 up=\eOA
