diff -Naurdp mc-4.6.1.orig/lib/mc.ext.in mc-4.6.1/lib/mc.ext.in
--- mc-4.6.1.orig/lib/mc.ext.in	2005-07-23 18:51:15.000000000 +0200
+++ mc-4.6.1/lib/mc.ext.in	2005-09-05 19:04:31.193103274 +0200
@@ -144,6 +144,13 @@ regex/\.a(rj|[0-9][0-9])$
 	Open=%cd %p#uarj
 	View=%view{ascii} unarj l %f
 
+# esp
+regex/\.e(sp|[0-9][0-9])$
+	Open=%cd %p#uesp
+	View=%view{ascii} unesp v %f
+	Extract=unesp x %f
+	Icon=compressed.xpm
+
 # ha
 regex/\.([Hh][Aa])$
 	Open=%cd %p#uha
@@ -195,6 +202,9 @@ regex/\.(rpm|spm)$
 regex/\.u?deb$
 	Open=%cd %p#deb
 	View=%view{ascii} dpkg-deb -c %f
+# for dpkg-less systems:
+#	Open=%cd %p#uar
+#	View=%view{ascii} ar p %f control.tar.gz | tar xzO control ./control 2>/dev/null ; echo "" ; echo "Filelist:" ; ar p %f data.tar.gz | tar tzv
 
 # ISO9660
 regex/\.iso$
diff -Naurdp mc-4.6.1.orig/syntax/bapc.syntax mc-4.6.1/syntax/bapc.syntax
--- mc-4.6.1.orig/syntax/bapc.syntax	1970-01-01 01:00:00.000000000 +0100
+++ mc-4.6.1/syntax/bapc.syntax	2005-09-05 19:04:31.194103171 +0200
@@ -0,0 +1,209 @@
+# BAssPasC syntax definitions v1.0 for Midnight Commander 4.1.35 and 4.5.x
+# (C) 1999 by A'rpi/ESP-team  <arpi@esp-team.scene.hu>
+# BAPC is a programming language, visit http://esp-team.scene.hu for details!
+
+context default brightcyan
+
+# main BAPC commands:
+    keyword whole VAR white
+    keyword whole CONST white
+
+    keyword whole USES white
+    keyword whole INDIRECT white
+    keyword whole INDIRECTALL white
+    keyword whole INCLUDEPROC white
+    keyword whole INCLUDE white
+    keyword wholeleft INCLUDED white
+    keyword whole ALIGN white
+
+    keyword whole SMOV white
+    keyword whole RMOV white
+    keyword whole LDDM white
+    keyword whole SDDM white
+    keyword whole LDD white
+    keyword whole SDD white
+
+# IF-EQU
+    keyword whole if white
+    keyword whole ifn white
+    keyword whole else white
+    keyword whole endif white
+    
+# BAPC conditionals:
+    keyword whole IF white
+    keyword whole IFS white
+    keyword whole IFN white
+    keyword whole IFNS white
+    keyword whole DO white
+    keyword whole DOCMD white
+    keyword whole THEN white
+    keyword whole THENCMD white
+    keyword whole ELSE white
+    keyword whole ENDIF white
+    keyword whole WHILE white
+    keyword whole ENDWHILE white
+    keyword whole REPEAT white
+    keyword whole UNTIL white
+    keyword whole WITH white
+    keyword whole ENDWITH white
+    keyword whole CASE white
+    keyword whole END white
+    keyword whole ENDCASE white
+    keyword whole ENDM white
+# FOR is too complex, include only the most importants:
+    keyword whole FOR white
+    keyword whole FORL(+) white
+    keyword whole FORLP(+) white
+    keyword whole TO white
+    keyword whole STEP white
+    keyword whole NEXT white
+
+    keyword whole FLAG white
+    keyword whole CXZ white
+    keyword wholeright \s+:S white
+    keyword wholeright \s+:NS white
+    keyword whole AND white
+    keyword whole OR white
+    keyword whole NOT white
+    keyword whole IN[*] white
+    keyword whole NIN[*] white
+
+# one-line commands:
+    keyword wholeleft REP\*+\s white
+    keyword whole REP(+) white
+    keyword whole REP32 white
+    keyword whole PUSH(*) white
+    keyword whole PUSH white
+    keyword whole POP(*) white
+    keyword whole POP white
+
+# BAPC types:
+    keyword whole OFS white
+    keyword whole STROFS white
+    keyword whole OFSSTR white
+    keyword whole BPT white
+    keyword whole WPT white
+    keyword whole DPT white
+    keyword whole QPT white
+    keyword whole TPT white
+    keyword whole DB white
+    keyword whole DW white
+    keyword whole DD white
+    keyword whole DQ white
+    keyword whole DT white
+    keyword whole DT white
+    keyword whole PROC white
+    keyword whole NEAR white
+    keyword whole FAR white
+    keyword whole SHORT white
+
+# STDBAPC instructions
+    keyword whole WRITE yellow
+    keyword whole WRITELN yellow
+    keyword whole WRITESTR yellow
+    keyword whole GOTOXY yellow
+    keyword whole WHEREXY yellow
+    keyword whole UPCASE yellow
+    keyword whole ALDIV yellow
+    keyword whole DOSEXIT yellow
+    keyword whole EXIT yellow
+
+    keyword whole WAITKEY yellow
+    keyword whole READKEY yellow
+    keyword whole READSHIFT yellow
+    keyword wholeleft GETKEY yellow
+    keyword whole KEYPRESSED yellow
+    keyword whole KEYPRESSEDP yellow
+    
+    keyword whole FILEOPEN yellow
+    keyword whole FILEREAD yellow
+    keyword whole FILEWRITE yellow
+    keyword whole FILECLOSE yellow
+    keyword whole FILEPOS yellow
+    keyword whole FILEAPPEND yellow
+
+# BAPC library instructions
+# TOO MANY TO INCLUDE HERE :(
+
+# C-style comments:  /* ... */    
+    keyword /\* green
+    keyword \*/ green
+# BAPC1-style comments:					! fix it !
+    keyword whole linestart .COMMENT green
+    keyword whole linestart .ENDCOMM green
+    keyword whole linestart .ENDCOMMENT green
+    keyword whole linestart .comment green
+    keyword whole linestart .endcomm green
+    keyword whole linestart .endcomment green
+
+# BAPC command separator:
+    keyword // brightmagenta
+
+    keyword {+\s brightmagenta
+    keyword {+\n brightmagenta
+    keyword } brightmagenta
+
+# BAPC operators:
+    keyword := white
+    keyword equ white
+    keyword EQU white
+    keyword -> white
+    keyword => white
+    keyword ?= white
+    keyword >> white
+    keyword << white
+    keyword >0> white
+    keyword <0< white
+    keyword >C> white
+    keyword <C< white
+    keyword \+= white
+    keyword -= white
+    keyword \+\+ white
+    keyword -- white
+    keyword ! yellow
+
+# strings:
+    keyword '*' brightgreen
+
+# addressing:
+    keyword [*] cyan
+    
+# other operators:    
+    keyword \+ yellow
+    keyword > yellow
+    keyword < yellow
+    keyword \* yellow
+#    keyword / yellow
+    keyword % yellow
+    keyword != yellow
+    keyword == yellow
+
+    keyword ( white
+    keyword ) white
+
+    keyword = white
+    keyword - yellow
+    
+    keyword , white
+    keyword : white
+    
+# one-line comment:
+    keyword ;*\n lightgray
+
+context exclusive /\* \*/ lightgray
+
+# These rules don't work :(  but why?????
+# context exclusive linestart .COMMENT .ENDCOMM lightgray
+# context exclusive linestart .comment .endcomm lightgray
+
+# Compiler control:
+context linestart . \n brightred
+    keyword ;*\n lightgray
+    keyword \s+\n white
+    keyword \s+\s white
+
+# macro def.:
+context linestart # \n brightred
+
+
+###############################################################################
diff -Naurdp mc-4.6.1.orig/syntax/html.syntax mc-4.6.1/syntax/html.syntax
--- mc-4.6.1.orig/syntax/html.syntax	2002-09-29 22:22:25.000000000 +0200
+++ mc-4.6.1/syntax/html.syntax	2005-09-05 19:04:31.207101835 +0200
@@ -1,17 +1,34 @@
-context default
-	keyword &\[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ\]; brightgreen
-	keyword &#\{xX\}\[0123456789abcdefABCDEF\]; brightgreen
-	keyword &#\[0123456789\]; brightgreen
-	spellcheck
-
-context <!-- --> brown
-	spellcheck
+# modified by A'rpi/ESP-team  <arpi@esp-team.scene.hu>
 
-context <! > brightred/orange
-	spellcheck
+context default
+    keyword &\[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ\]; brightgreen
+    keyword &#\{xX\}\[0123456789abcdefABCDEF\]; brightgreen
+    keyword &#\[0123456789\]; brightgreen
+#    keyword whole &*; brightgreen
+    keyword <BR> yellow
+    keyword <P> yellow
+    keyword <P\s*> yellow
+    keyword </P> yellow
+    spellcheck
 
-context </ > brightcyan
+context <!-- --> white
+# context <B> </B> white
+# context <I> </I> white
+context <A\s > brightmagenta
 
 context < > brightcyan
-	keyword "*" green
-	keyword = yellow
+    keyword "*.gif" brightred
+    keyword "*.jpg" brightred
+    keyword "*.png" brightred
+    keyword "http:*" brightmagenta
+    keyword "ftp:*" brightmagenta
+    keyword "mailto:*" brightmagenta
+    keyword "gopher:*" brightmagenta
+    keyword "telnet:*" brightmagenta
+    keyword "file:*" brightmagenta
+    keyword HREF="*" brightmagenta
+    keyword href="*" brightmagenta
+    keyword whole "*" cyan
+    keyword "*" cyan
+
+###############################################################################
diff -Naurdp mc-4.6.1.orig/syntax/jasm.syntax mc-4.6.1/syntax/jasm.syntax
--- mc-4.6.1.orig/syntax/jasm.syntax	1970-01-01 01:00:00.000000000 +0100
+++ mc-4.6.1/syntax/jasm.syntax	2005-09-05 19:04:31.208101733 +0200
@@ -0,0 +1,180 @@
+#    Java assembly -- JDisAsm syntax -- (C) 2000. A'rpi/ESP-team
+#  Syntax hilight definition file for Midnight Commander / CoolEdit
+
+context default
+#    keyword whole interface yellow/24
+    keyword whole invoke yellow/24
+#    keyword whole special yellow/24
+#    keyword whole static yellow/24
+#    keyword whole virtual yellow/24
+    keyword whole array yellow/24
+    keyword wholeright [idx] yellow/24
+    keyword whole const yellow/24
+#    keyword whole field yellow/24
+    keyword whole null yellow/24
+
+# mnemonics:
+    keyword whole push yellow/24
+    keyword whole push2 yellow/24
+    keyword whole pop yellow/24
+    keyword whole pop2 yellow/24
+    keyword whole dup yellow/24
+    keyword whole dup2 yellow/24
+    keyword whole goto yellow/24
+    keyword whole return yellow/24
+    keyword whole new yellow/24
+    keyword whole new yellow/24
+    keyword whole nop yellow/24
+    keyword whole swap yellow/24
+    keyword whole iadd yellow/24
+    keyword whole ladd yellow/24
+    keyword whole fadd yellow/24
+    keyword whole dadd yellow/24
+    keyword whole isub yellow/24
+    keyword whole lsub yellow/24
+    keyword whole fsub yellow/24
+    keyword whole dsub yellow/24
+    keyword whole imul yellow/24
+    keyword whole lmul yellow/24
+    keyword whole fmul yellow/24
+    keyword whole dmul yellow/24
+    keyword whole idiv yellow/24
+    keyword whole ldiv yellow/24
+    keyword whole fdiv yellow/24
+    keyword whole ddiv yellow/24
+    keyword whole irem yellow/24
+    keyword whole lrem yellow/24
+    keyword whole frem yellow/24
+    keyword whole drem yellow/24
+    keyword whole ineg yellow/24
+    keyword whole lneg yellow/24
+    keyword whole fneg yellow/24
+    keyword whole dneg yellow/24
+    keyword whole ishl yellow/24
+    keyword whole lshl yellow/24
+    keyword whole ishr yellow/24
+    keyword whole lshr yellow/24
+    keyword whole iushr yellow/24
+    keyword whole lushr yellow/24
+    keyword whole iand yellow/24
+    keyword whole land yellow/24
+    keyword whole ior yellow/24
+    keyword whole lor yellow/24
+    keyword whole ixor yellow/24
+    keyword whole lxor yellow/24
+    keyword whole iinc yellow/24
+    keyword whole i2l yellow/24
+    keyword whole i2f yellow/24
+    keyword whole i2d yellow/24
+    keyword whole l2i yellow/24
+    keyword whole l2f yellow/24
+    keyword whole l2d yellow/24
+    keyword whole f2i yellow/24
+    keyword whole f2l yellow/24
+    keyword whole f2d yellow/24
+    keyword whole d2i yellow/24
+    keyword whole d2l yellow/24
+    keyword whole d2f yellow/24
+    keyword whole i2b yellow/24
+    keyword whole i2c yellow/24
+    keyword whole i2s yellow/24
+    keyword whole cmp yellow/24
+    keyword whole call yellow/24
+    keyword whole ret yellow/24
+    keyword whole arraylength yellow/24
+    keyword whole athrow yellow/24
+    keyword whole checkcast yellow/24
+    keyword whole instanceof yellow/24
+    keyword whole monitor yellow/24
+    keyword whole wide yellow/24
+    keyword whole new yellow/24
+    keyword whole breakpoint yellow/24
+
+
+    keyword whole if\s\s\s\s*\sgoto yellow/24
+    keyword whole if\s*\sgoto yellow/24
+
+    keyword whole int white/24
+    keyword whole float white/24
+    keyword whole double white/24
+    keyword whole byte white/24
+    keyword whole char white/24
+    keyword whole long white/24
+    keyword whole short white/24
+    keyword whole objref white/24
+    keyword whole void white/24
+    keyword whole boolean white/24
+
+    keyword ST[*] brightmagenta/19
+
+#    keyword whole (*) white/14
+    keyword wholeright (*) brightcyan/14
+
+    keyword interface\s*\s brightcyan/14
+    keyword virtual\s*\s brightcyan/14
+    keyword static\s*\s brightcyan/14
+    keyword special\s*\s brightcyan/14
+    keyword field\s*\s brightcyan/14
+    keyword Field\:\s*\s brightcyan/14
+
+    keyword \`\- brightgreen/16
+    keyword \,\- brightgreen/16
+    keyword \|\- brightgreen/16
+    keyword \|\s brightgreen/16
+
+    keyword \{*\} brightred/24
+
+#    keyword \s*\s\{ brightcyan/14
+
+    keyword LOC_*\s brightgreen/16
+    keyword LOC_*\n brightgreen/16
+
+    keyword F\s\#*\: white/24
+    keyword F\s\#\s*\: white/24
+    keyword F\s\#\s\s*\: white/24
+    keyword M\s\#*\: white/24
+    keyword M\s\#\s*\: white/24
+    keyword M\s\#\s\s*\: white/24
+    keyword A\s\#*\: white/24
+    keyword A\s\#\s*\: white/24
+    keyword A\s\#\s\s*\: white/24
+
+
+#    keyword invoke*\( brightcyan/14
+# context \{ \} brightred/24
+#    keyword /\* brown/22
+#    keyword \*/ brown/22
+
+# context if goto yellow/24
+
+
+context " " green/6
+    keyword \\" brightgreen/16
+    keyword %% brightgreen/16
+    keyword %\[#0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[L\]e brightgreen/16
+    keyword %\[#0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[L\]E brightgreen/16
+    keyword %\[#0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[L\]f brightgreen/16
+    keyword %\[#0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[L\]g brightgreen/16
+    keyword %\[#0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[L\]G brightgreen/16
+    keyword %\[0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[hl\]d brightgreen/16
+    keyword %\[0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[hl\]i brightgreen/16
+    keyword %\[#0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[hl\]o brightgreen/16
+    keyword %\[0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[hl\]u brightgreen/16
+    keyword %\[#0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[hl\]x brightgreen/16
+    keyword %\[#0\s-\+,\]\[0123456789\]\[.\]\[0123456789\]\[hl\]X brightgreen/16
+    keyword %\[hl\]n brightgreen/16
+    keyword %\[.\]\[0123456789\]s brightgreen/16
+    keyword %[*] brightgreen/16
+    keyword %c brightgreen/16
+    keyword \\\\ brightgreen/16
+    keyword \\' brightgreen/16
+    keyword \\a brightgreen/16
+    keyword \\b brightgreen/16
+    keyword \\t brightgreen/16
+    keyword \\n brightgreen/16
+    keyword \\v brightgreen/16
+    keyword \\f brightgreen/16
+    keyword \\r brightgreen/16
+    keyword \\0 brightgreen/16
+
+
diff -Naurdp mc-4.6.1.orig/syntax/lsm.syntax mc-4.6.1/syntax/lsm.syntax
--- mc-4.6.1.orig/syntax/lsm.syntax	2002-09-29 22:22:25.000000000 +0200
+++ mc-4.6.1/syntax/lsm.syntax	2005-09-05 19:04:31.219100602 +0200
@@ -1,33 +1,34 @@
+# modified by A'rpi/ESP-team  <arpi@esp-team.scene.hu>
 
 context default
-    spellcheck
-    keyword linestart Begin3		brightmagenta
-    keyword linestart Title:\s\s\s\s\s\s\s\s\s\s	red  yellow
-    keyword linestart Version:\s\s\s\s\s\s\s\s	red  yellow
-    keyword linestart Entered-date:\s\s\s	red  yellow
-    keyword linestart Description:\s\s\s\s	red  yellow
-    keyword linestart Keywords:\s\s\s\s\s\s\s	red  yellow
-    keyword linestart Alternate-site:\s	red  yellow
-    keyword linestart Primary-site:\s\s\s	red  yellow
-    keyword linestart Original-site:\s\s	red  yellow
-    keyword linestart Platforms:\s\s\s\s\s\s	red  yellow
-    keyword linestart Copying-policy:\s	red  yellow
-    keyword linestart End			brightmagenta
+    keyword linestart Begin3		brightmagenta/20
+    keyword linestart Title:		brightred/9
+    keyword linestart Version:		brightred/9
+    keyword linestart Entered-date:	brightred/9
+    keyword linestart Description:	brightred/9
+    keyword linestart Keywords:		brightred/9
+    keyword linestart Author:		brightred/9
+    keyword linestart Maintained-by:	brightred/9
+    keyword linestart Alternate-site:	brightred/9
+    keyword linestart Primary-site:	brightred/9
+    keyword linestart Original-site:	brightred/9
+    keyword linestart Platforms:	brightred/9
+    keyword linestart Copying-policy:	brightred/9
+    keyword linestart End		brightmagenta/20
 
-    keyword linestart \t\t				white yellow
-    keyword linestart \s\s\s\s\s\s\s\s\s\s\s\s\s\s\s\s	white yellow
-    keyword whole GPL	green
-    keyword whole BSD	green
-    keyword whole Shareware	green
-    keyword whole sunsite.unc.edu	green
-    keyword wholeright \s*.tar.gz	green
-    keyword wholeright \s*.lsm	green
+    keyword whole GPL			green/6
+    keyword whole BSD			green/6
+    keyword whole Shareware		green/6
 
-context linestart Author:\s\s\s\s\s\s\s\s\s \n brightred
-    keyword whole \s*@*\s(*)		cyan
+    keyword whole sunsite.unc.edu	green/6
+    keyword wholeright \s+tar+gz	brightgreen/6
+    keyword wholeright \t+tar+gz	brightgreen/6
 
-context linestart Maintained-by:\s\s \n brightred
-    keyword whole \s*@*\s(*)		cyan
+    keyword whole \s*@*\s(*)	cyan/16
+    keyword whole <*@*>		cyan/16
 
-###############################################################################
+context exclusive linestart Title: \n	white/26
+#brightred/9
+#    keyword \s+\n		yellow/26
 
+###############################################################################
\ No newline at end of file
diff -Naurdp mc-4.6.1.orig/syntax/Makefile.in mc-4.6.1/syntax/Makefile.in
--- mc-4.6.1.orig/syntax/Makefile.in	2005-07-23 18:53:15.000000000 +0200
+++ mc-4.6.1/syntax/Makefile.in	2005-09-05 19:10:48.000372050 +0200
@@ -255,7 +255,10 @@ SYNTAXFILES = \
 	tcl.syntax		\
 	texinfo.syntax		\
 	unknown.syntax		\
-	xml.syntax
+	xml.syntax		\
+	bapc.syntax		\
+	mhtml.syntax		\
+	jasm.syntax
 
 @USE_EDIT_TRUE@syntaxdir = $(pkgdatadir)/syntax
 @USE_EDIT_TRUE@syntax_DATA = $(SYNTAXFILES)
diff -Naurdp mc-4.6.1.orig/syntax/makefile.syntax mc-4.6.1/syntax/makefile.syntax
--- mc-4.6.1.orig/syntax/makefile.syntax	2005-05-31 08:03:45.000000000 +0200
+++ mc-4.6.1/syntax/makefile.syntax	2005-09-05 19:08:53.400152713 +0200
@@ -1,3 +1,4 @@
+# modified by A'rpi/ESP-team  <arpi@esp-team.scene.hu>
 context default
     keyword $$ brightcyan
     keyword $(*) yellow
@@ -12,21 +13,20 @@ context default
     keyword whole linestart ifeq magenta
     keyword whole linestart ifneq magenta
     keyword whole linestart else magenta
-    keyword linestart \t lightgray red
+#    keyword linestart \t lightgray red
     keyword whole .PHONY white
     keyword whole .NOEXPORT white
     keyword = white
     keyword : yellow
     keyword \\\n yellow
 # this handles strange cases like @something@@somethingelse@ properly
-    keyword whole @+@ brightmagenta black
-    keyword @+@ brightmagenta black
+    keyword whole @+@ brightmagenta
+    keyword @+@ brightmagenta
 
 # comment
 context linestart # \n brown
-    spellcheck
-    keyword whole @+@ brightmagenta black
-    keyword @+@ brightmagenta black
+    keyword whole @+@ brightmagenta
+    keyword @+@ brightmagenta
 
 # right part of an assignment
 context exclusive = \n brightcyan
@@ -34,9 +34,9 @@ context exclusive = \n brightcyan
     keyword $$ brightcyan
     keyword $(*) yellow
     keyword ${*} brightgreen
-    keyword linestart \t lightgray red
-    keyword whole @+@ brightmagenta black
-    keyword @+@ brightmagenta black
+#    keyword linestart \t lightgray red
+    keyword whole @+@ brightmagenta
+    keyword @+@ brightmagenta
 
 # shell code
 context exclusive linestart \t \n
@@ -44,6 +44,6 @@ context exclusive linestart \t \n
     keyword $$ brightcyan
     keyword $(*) yellow
     keyword ${*} brightgreen
-    keyword linestart \t lightgray red
-    keyword whole @+@ brightmagenta black
-    keyword @+@ brightmagenta black
+#    keyword linestart \t lightgray red
+    keyword whole @+@ brightmagenta
+    keyword @+@ brightmagenta
diff -Naurdp mc-4.6.1.orig/syntax/mhtml.syntax mc-4.6.1/syntax/mhtml.syntax
--- mc-4.6.1.orig/syntax/mhtml.syntax	1970-01-01 01:00:00.000000000 +0100
+++ mc-4.6.1/syntax/mhtml.syntax	2005-09-05 19:04:31.238098650 +0200
@@ -0,0 +1,52 @@
+# modified by A'rpi/ESP-team  <arpi@esp-team.scene.hu>
+
+context default
+# brightcyan/17
+    keyword whole &*; brightgreen/16
+    keyword $+(*) yellow/18
+    keyword $+\s yellow/18
+    keyword $+\n yellow/18
+    keyword $+$ yellow/18
+    keyword $B yellow/18
+    keyword $I yellow/18
+    keyword $BI yellow/18
+    keyword $U yellow/18
+    keyword $R yellow/18
+    keyword ( yellow/18
+    keyword ) yellow/18
+    keyword #+= brightred/11
+    keyword # brightred/11
+
+    keyword %1 yellow/26
+    keyword %2 yellow/26
+    keyword %3 yellow/26
+    keyword %4 yellow/26
+    keyword %5 yellow/26
+
+context <!-- --> brightmagenta/26
+context ` ` brightmagenta/26
+
+context exclusive $+( ) white/26
+    keyword whole &*; brightgreen/16
+    keyword %1 yellow/26
+    keyword %2 yellow/26
+    keyword %3 yellow/26
+    keyword %4 yellow/26
+    keyword %5 yellow/26
+
+#context $+( ) yellow/18
+#    keyword (*) white/26
+#    keyword whole &*; brightgreen/16
+
+#context exclusive $I( ) white/26
+#context exclusive $U( ) white/26
+#context exclusive $BI( ) white/26
+
+#context #+= # brightgreen/16
+#    keyword #+= white/26
+
+context < > brightcyan/17
+    keyword whole "*" cyan/5
+    keyword "*" cyan/5
+
+###############################################################################
diff -Naurdp mc-4.6.1.orig/syntax/Syntax mc-4.6.1/syntax/Syntax
--- mc-4.6.1.orig/syntax/Syntax	2005-05-26 09:21:34.000000000 +0200
+++ mc-4.6.1/syntax/Syntax	2005-09-05 19:04:31.243098136 +0200
@@ -84,6 +84,15 @@ include swig.syntax
 file ..\*\\.(php|PHP)[0-9]?$ PHP\sProgram
 include php.syntax
 
+file ..\*\\.(bap|bp2|bp3|bpp|bpe|lib|l32)$ BAssPasC\sProgram
+include bapc.syntax
+
+file ..\*\\.(mhtml|mac)$ Macro-HTML\sSource
+include mhtml.syntax
+
+file ..\*\\.(jasm)$ JVM\sAssembly
+include jasm.syntax
+
 file ..\*\\.(java|JAVA|Java|jav)$ Java\sProgram
 include java.syntax
 
diff -Naurdp mc-4.6.1.orig/vfs/extfs/extfs.ini mc-4.6.1/vfs/extfs/extfs.ini
--- mc-4.6.1.orig/vfs/extfs/extfs.ini	2004-01-24 01:28:13.000000000 +0100
+++ mc-4.6.1/vfs/extfs/extfs.ini	2005-09-05 19:04:31.252097211 +0200
@@ -7,6 +7,7 @@ uzoo
 ulha
 urar
 uha
+uesp
 # For arj usage you need a special patch to unarj (see unarj.diff)
 uarj
 
diff -Naurdp mc-4.6.1.orig/vfs/extfs/Makefile.in mc-4.6.1/vfs/extfs/Makefile.in
--- mc-4.6.1.orig/vfs/extfs/Makefile.in	2005-07-23 18:53:15.000000000 +0200
+++ mc-4.6.1/vfs/extfs/Makefile.in	2005-09-05 19:09:22.575153674 +0200
@@ -229,7 +229,7 @@ extfsdir = $(pkgdatadir)/extfs
 EXTFS_MISC = README extfs.ini sfs.ini
 
 # Scripts hat don't need adaptation to the local system
-EXTFS_CONST = bpp rpm trpm
+EXTFS_CONST = bpp rpm trpm uesp
 
 # Scripts that need adaptation to the local system - source files
 EXTFS_IN = \
diff -Naurdp mc-4.6.1.orig/vfs/extfs/uesp mc-4.6.1/vfs/extfs/uesp
--- mc-4.6.1.orig/vfs/extfs/uesp	1970-01-01 01:00:00.000000000 +0100
+++ mc-4.6.1/vfs/extfs/uesp	2005-09-05 19:04:31.257096697 +0200
@@ -0,0 +1,15 @@
+#! /bin/sh
+#
+# EXTfs script for Midnight Commander 4.x.x
+# read details in README.MC
+#
+# Written by A'rpi/ESP-team  1999
+#
+
+case "$1" in
+  list) unesp vm $2; exit 0;;
+  copyout)
+    unesp eym $2 $4 /$3 >2 /dev/null
+    exit 0;;
+esac
+exit 1
