diff -Naurdp mc-4.6.1.orig/src/screen.c mc-4.6.1/src/screen.c
--- mc-4.6.1.orig/src/screen.c	2005-05-27 16:19:18.000000000 +0200
+++ mc-4.6.1/src/screen.c	2005-09-05 18:19:38.000000000 +0200
@@ -203,9 +203,18 @@ string_file_size (file_entry *fe, int le
 
 #ifdef HAVE_STRUCT_STAT_ST_RDEV
     if (S_ISBLK (fe->st.st_mode) || S_ISCHR (fe->st.st_mode))
-        g_snprintf (buffer, sizeof (buffer), "%3d,%3d",
-		    (int) ((fe->st.st_rdev >> 8) & 0xff),
-		    (int) (fe->st.st_rdev & 0xff));
+    {
+        if (major(fe->st.st_rdev) < 1000) {
+            g_snprintf (buffer, sizeof (buffer), "%3d", major(fe->st.st_rdev));
+        } else {
+            g_snprintf (buffer, sizeof (buffer), "...");
+        }
+        if (minor(fe->st.st_rdev) < 1000) {
+            g_snprintf (buffer + 3, sizeof (buffer) - 3, ",%3d", minor(fe->st.st_rdev));
+        } else {
+            g_snprintf (buffer + 3, sizeof (buffer) - 3, ",...");
+        }
+    }
     else
 #endif
     {
