https://savannah.gnu.org/bugs/?17822

diff -Naur mc-4.6.1.orig/src/key.c mc-4.6.1/src/key.c
--- mc-4.6.1.orig/src/key.c	2005-06-08 14:27:19.000000000 +0200
+++ mc-4.6.1/src/key.c	2006-10-10 18:01:04.000000000 +0200
@@ -36,6 +36,7 @@
 #include "tty.h"
 #include "mouse.h"
 #include "key.h"
+#include "layout.h"	/* winch_flag */
 #include "main.h"
 #include "win.h"
 #include "cons.saver.h"
@@ -1050,7 +1051,7 @@
 		time_addr = NULL;
 	}
 
-	if (!block) {
+	if (!block || winch_flag) {
 	    time_addr = &timeout;
 	    timeout.tv_sec = 0;
 	    timeout.tv_usec = 0;
@@ -1067,7 +1068,7 @@
 	if (flag == 0) {
 	    if (redo_event)
 		return EV_MOUSE;
-	    if (!block)
+	    if (!block || winch_flag)
 		return EV_NONE;
 	    vfs_timeout_handler ();
 	}
diff -Naur mc-4.6.1.orig/src/layout.c mc-4.6.1/src/layout.c
--- mc-4.6.1.orig/src/layout.c	2005-05-27 16:19:18.000000000 +0200
+++ mc-4.6.1/src/layout.c	2006-10-10 16:41:36.000000000 +0200
@@ -760,6 +760,7 @@
 void
 change_screen_size (void)
 {
+    winch_flag = 0;
 #if defined(HAVE_SLANG) || NCURSES_VERSION_MAJOR >= 4
 #if defined TIOCGWINSZ
 
@@ -792,7 +793,6 @@
     touchwin (stdscr);
 #endif				/* TIOCGWINSZ */
 #endif				/* defined(HAVE_SLANG) || NCURSES_VERSION_MAJOR >= 4 */
-    winch_flag = 0;
 }
 
 static int ok_to_refresh = 1;
