diff -Naur mc-4.6.1.orig/lib/mc.ext.in mc-4.6.1/lib/mc.ext.in
--- mc-4.6.1.orig/lib/mc.ext.in	2005-07-23 18:51:15.000000000 +0200
+++ mc-4.6.1/lib/mc.ext.in	2005-09-09 20:00:55.006072544 +0200
@@ -199,7 +199,7 @@
 # ISO9660
 regex/\.iso$
 	Open=%cd %p#iso9660
-	View=%view{ascii} isoinfo -l -i %f
+	View=%view{ascii} /usr/share/mc/extfs/iso9660 view %f
 
 # 7zip archives (they are not man pages)
 shell/.7z
diff -Naur mc-4.6.1.orig/vfs/extfs/iso9660.in mc-4.6.1/vfs/extfs/iso9660.in
--- mc-4.6.1.orig/vfs/extfs/iso9660.in	2004-10-29 11:14:38.000000000 +0200
+++ mc-4.6.1/vfs/extfs/iso9660.in	2005-09-09 19:59:26.451178245 +0200
@@ -65,6 +65,10 @@
 	$ISOINFO -i "$1" -x "/$2" > "$3" 
 }
 
+mcisofs_view () {
+	$ISOINFO -l -i "$1"
+}
+
 save_ctype=`locale | grep LC_CTYPE`
 LC_ALL=C
 
@@ -80,5 +84,9 @@
     test_iso "$@";
     mcisofs_copyout "$@";
     exit 0;;
+  view)
+    test_iso "$@";
+    mcisofs_view "$@";
+    exit 0;;
 esac
 exit 1
