diff -Naurd a/syntax/Syntax b/syntax/Syntax
--- a/syntax/Syntax	2004-11-08 12:21:58.114914376 +0100
+++ b/syntax/Syntax	2004-11-08 12:22:21.452386585 +0100
@@ -20,7 +20,7 @@
 file [Ss]yntax$ Syntax\sHighlighting\sdefinitions ^#\ssyntax\srules\sversion\s
 include syntax.syntax
 
-file ..\*\\.(diff|rej|patch)$ Diff\sOutput ^(diff|Index:)\s
+file ..\*\\.(diff?|rej|patch)$ Diff\sOutput ^(diff|Index:)\s
 include diff.syntax
 
 file ..\*\\.lsm$ LSM\sFile
