diff -urN mc-4.6.1.orig/lib/mc.ext.in mc-4.6.1/lib/mc.ext.in
--- mc-4.6.1.orig/lib/mc.ext.in	2005-07-23 18:51:15.000000000 +0200
+++ mc-4.6.1/lib/mc.ext.in	2006-05-11 16:41:25.000000000 +0200
@@ -119,6 +119,10 @@
 	Open=%cd %p#utar
 	View=%view{ascii} bzip2 -dc %f 2>/dev/null | tar tvvf -
 
+regex/\.(fpm|fdb)$
+    Open=%cd %p#utar
+    View=%view{ascii} tar tvvf %f
+
 # .tar.F - used in QNX
 regex/\.tar\.F$
 	# Open=%cd %p#utar
