diff -urN mc-4.6.1.orig/syntax/php.syntax mc-4.6.1/syntax/php.syntax
--- mc-4.6.1.orig/syntax/php.syntax	2005-04-17 01:56:50.000000000 +0200
+++ mc-4.6.1/syntax/php.syntax	2005-02-17 14:08:39.000000000 +0100
@@ -3654,6 +1027,10 @@
 
     keyword wholeright $+ brightgreen
 
+    keyword " green
+
+    keyword '*' brightgreen
+
     keyword /\* brown
     keyword \*/ brown
     keyword // brown
@@ -3679,7 +1056,6 @@
     keyword & white
     keyword | white
     keyword :: white
-    keyword -> white
     keyword { brightcyan
     keyword } brightcyan
     keyword ( brightcyan
@@ -3709,7 +1085,7 @@
 context linestart # \n brown
     spellcheck
 
-context " " green
+context exclusive " " green
     spellcheck
     keyword \\" brightgreen
     keyword \\\n brightgreen
@@ -3724,7 +1100,3 @@
     keyword \\\\ brightgreen
     keyword \\' brightgreen
     keyword \\\{abtnvfr\} brightgreen
-
-context ' ' brightgreen
-    spellcheck
-    keyword \\' brightcyan
