diff -Naurd a/syntax/Syntax b/syntax/Syntax
--- a/syntax/Syntax	2007-10-17 14:40:20.000000000 +0200
+++ b/syntax/Syntax	2007-10-16 16:09:34.000000000 +0200
@@ -72,7 +72,7 @@
 file ..\*\.(texi|texinfo|TEXI|TEXINFO)$ Texinfo\sDocument
 include texinfo.syntax
 
-file ..\*\\.([chC]|CC|cxx|cc|cpp|CPP|CXX|hxx|h\.in)$ C/C\+\+\sProgram
+file ..\*\\.([chC]|CC|cxx|cc|cpp|CPP|CXX|hh|hpp|hxx|h\.in)$ C/C\+\+\sProgram
 include c.syntax
 
 file ..\*\\.[fF]$ Fortran\sProgram
diff -Naurd b/lib/mc.ext.in c/lib/mc.ext.in
--- b/lib/mc.ext.in	2007-10-17 14:40:20.000000000 +0200
+++ c/lib/mc.ext.in	2007-10-16 16:07:46.000000000 +0200
@@ -230,7 +230,7 @@
 	Open=%var{EDITOR:vi} %f
 	
 # Header
-regex/\.(h|hpp)$
+regex/\.(h|hh|hpp)$
 	Open=%var{EDITOR:vi} %f
 
 # Object
