diff -Naurd a/lib/mc.ext.in b/lib/mc.ext.in
--- a/lib/mc.ext.in	2005-07-23 18:51:15.000000000 +0200
+++ b/lib/mc.ext.in	2008-01-23 14:56:01.000000000 +0100
@@ -473,6 +473,10 @@
 
 ### Miscellaneous ###
 
+# PKGBUILD
+regex/[Pp][Kk][Gg][Bb][Uu][Ii][Ll][Dd]
+	Open=makepkg --asroot
+
 # Makefile
 regex/[Mm]akefile$
 	Open=make -f %f %{Enter parameters}
diff -Naurd a/syntax/Makefile.am b/syntax/Makefile.am
--- a/syntax/Makefile.am	2005-03-19 18:39:06.000000000 +0100
+++ b/syntax/Makefile.am	2008-01-23 14:56:01.000000000 +0100
@@ -27,6 +27,7 @@
 	pascal.syntax		\
 	perl.syntax		\
 	php.syntax		\
+	pkgbuild.syntax		\
 	po.syntax		\
 	povray.syntax		\
 	python.syntax		\
diff -Naurd a/syntax/Makefile.in b/syntax/Makefile.in
--- a/syntax/Makefile.in	2005-07-23 18:53:15.000000000 +0200
+++ b/syntax/Makefile.in	2008-01-23 14:56:01.000000000 +0100
@@ -241,6 +241,7 @@
 	pascal.syntax		\
 	perl.syntax		\
 	php.syntax		\
+	pkgbuild.syntax		\
 	po.syntax		\
 	povray.syntax		\
 	python.syntax		\
diff -Naurd a/syntax/Syntax b/syntax/Syntax
--- a/syntax/Syntax	2005-05-26 09:21:34.000000000 +0200
+++ b/syntax/Syntax	2008-01-23 14:56:01.000000000 +0100
@@ -144,5 +144,8 @@
 file Don_t_match_me Mail\sfolder ^From\s
 include mail.syntax
 
+file (PKGBUILD|..\*\\.install)$ ArchPKGBUILD
+include pkgbuild.syntax
+
 file .\* unknown
 include unknown.syntax
diff -Naurd a/syntax/pkgbuild.syntax b/syntax/pkgbuild.syntax
--- a/syntax/pkgbuild.syntax	1970-01-01 01:00:00.000000000 +0100
+++ b/syntax/pkgbuild.syntax	2008-01-23 14:56:51.000000000 +0100
@@ -0,0 +1,380 @@
+
+context default
+    keyword ;; brightred
+    keyword \\@ brightred
+    keyword \\$ brightred
+    keyword \\\\ brightred
+    keyword \\" brightred
+    keyword \\' brightred
+    keyword \\` brightred
+    keyword ` brightred
+	keyword ' brightgreen
+	keyword ; brightcyan
+    keyword $(*) brightgreen
+    keyword ${*} brightgreen
+    keyword { brightcyan
+    keyword } brightcyan
+
+    keyword whole linestart #!\[\s\]*\n brightcyan black
+
+    keyword $\* brightred
+    keyword $@ brightred
+    keyword $# brightred
+    keyword $? brightred
+    keyword $- brightred
+    keyword $$ brightred
+    keyword $! brightred
+    keyword $_ brightred
+
+    keyword wholeright $\[0123456789\] brightred
+
+    keyword wholeright $+ brightgreen
+
+    keyword $ brightgreen
+
+    keyword wholeleft linestart function*() brightmagenta
+    keyword wholeleft linestart function\[\s\]+ brightmagenta
+    keyword wholeright +() brightmagenta
+
+wholechars abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._
+
+    keyword whole alias yellow
+    keyword whole bg yellow
+    keyword whole break yellow
+    keyword whole case yellow
+    keyword whole continue yellow
+    keyword whole done yellow
+    keyword whole do yellow
+    keyword whole elif yellow
+    keyword whole else yellow
+    keyword whole esac yellow
+    keyword whole eval yellow
+    keyword whole exec yellow
+    keyword whole exit yellow
+    keyword whole export yellow
+    keyword whole fg yellow
+    keyword whole fi yellow
+    keyword whole for yellow
+    keyword whole if yellow
+    keyword whole in yellow
+    keyword whole read yellow
+    keyword whole return yellow
+    keyword whole select yellow
+    keyword whole set yellow
+    keyword whole shift yellow
+    keyword whole then yellow
+    keyword whole trap yellow
+    keyword whole umask yellow
+    keyword whole unset yellow
+    keyword whole until yellow
+    keyword whole wait yellow
+    keyword whole while yellow
+	
+	keyword whole pkgname brightcyan
+	keyword whole _pkgname brightcyan
+	keyword whole pkgver brightcyan
+	keyword whole _pkgver brightcyan
+	keyword whole pkgrel brightcyan
+	keyword whole _pkgrel brightcyan
+	keyword whole pkgdesc brightcyan
+	keyword whole arch brightcyan
+	keyword whole i686 green
+	keyword whole x86_64 green
+	keyword whole license brightcyan
+	keyword whole url brightcyan
+	keyword whole backup brightcyan
+	keyword whole makedepends brightcyan
+	keyword whole depends brightcyan
+	keyword whole install brightcyan
+	keyword whole conflicts brightcyan
+	keyword whole provides brightcyan
+	keyword whole groups brightcyan
+	keyword whole source brightcyan
+	keyword whole replaces brightcyan
+	keyword whole md5sums brightcyan black
+	keyword whole sha1sums brightcyan black
+	keyword whole sha256sums brightcyan black
+	keyword whole sha512sums brightcyan black
+	keyword whole options brightcyan
+	keyword whole strip green black
+	keyword whole !strip brightred black
+	keyword whole docs green black
+	keyword whole !docs brightred black
+	keyword whole libtool green black
+	keyword whole !libtool brightred black
+	keyword whole emptydirs green black
+	keyword whole !emptydirs brightred black
+	keyword whole ccache green black
+	keyword whole !ccahce brightred black
+	keyword whole distcc green black
+	keyword whole !distcc brightred black
+	keyword whole makeflags green black
+	keyword whole !makeflags brightred black
+	keyword whole force green black
+	keyword whole !force brightred black
+
+    keyword whole ash cyan
+    keyword whole awk cyan
+    keyword whole basename cyan
+    keyword whole bash cyan
+    keyword whole bg_backup cyan
+    keyword whole bg_restore cyan
+    keyword whole bsh cyan
+    keyword whole cat cyan
+    keyword whole cd cyan
+    keyword whole chgrp cyan
+    keyword whole chmod cyan
+    keyword whole chown cyan
+    keyword whole cp cyan
+    keyword whole cpio cyan
+    keyword whole csh cyan
+    keyword whole date cyan
+    keyword whole dd cyan
+    keyword whole df cyan
+    keyword whole dmesg cyan
+    keyword whole dnsdomainname cyan
+    keyword whole doexec cyan
+    keyword whole domainname cyan
+    keyword whole echo cyan
+    keyword whole ed cyan
+    keyword whole egrep cyan
+    keyword whole ex cyan
+    keyword whole false cyan
+    keyword whole fgrep cyan
+    keyword whole find cyan
+    keyword whole fsconf cyan
+    keyword whole gawk cyan
+    keyword whole grep cyan
+    keyword whole gunzip cyan
+    keyword whole gzip cyan
+    keyword whole hostname cyan
+    keyword whole igawk cyan
+    keyword whole ipcalc cyan
+    keyword whole kill cyan
+    keyword whole ksh cyan
+    keyword whole linuxconf cyan
+    keyword whole ln cyan
+    keyword whole login cyan
+    keyword whole lpdconf cyan
+    keyword whole ls cyan
+    keyword whole mail cyan
+    keyword whole mkdir cyan
+    keyword whole mknod cyan
+    keyword whole mktemp cyan
+    keyword whole more cyan
+    keyword whole mount cyan
+    keyword whole mt cyan
+    keyword whole mv cyan
+    keyword whole netconf cyan
+    keyword whole netstat cyan
+    keyword whole nice cyan
+    keyword whole nisdomainname cyan
+    keyword whole ping cyan
+    keyword whole ps cyan
+    keyword whole pwd cyan
+    keyword whole red cyan
+    keyword whole remadmin cyan
+    keyword whole rm cyan
+    keyword whole rmdir cyan
+    keyword whole rpm cyan
+    keyword whole sed cyan
+    keyword whole set cyan
+    keyword whole setserial cyan
+    keyword whole sh cyan
+    keyword whole sleep cyan
+    keyword whole sort cyan
+    keyword whole stty cyan
+    keyword whole su cyan
+    keyword whole sync cyan
+    keyword whole taper cyan
+    keyword whole tar cyan
+    keyword whole tcsh cyan
+    keyword whole test cyan
+    keyword whole time cyan
+    keyword whole touch cyan
+    keyword whole true cyan
+    keyword whole umount cyan
+    keyword whole uname cyan
+    keyword whole userconf cyan
+    keyword whole usleep cyan
+    keyword whole vi cyan
+    keyword whole view cyan
+    keyword whole vim cyan
+    keyword whole xconf cyan
+    keyword whole ypdomainname cyan
+    keyword whole zcat cyan
+    keyword whole zsh cyan
+
+wholechars abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_
+
+context # \n brown
+    spellcheck
+
+context exclusive whole <<\[\s\]\[-\]\[\s\]\[\\\]EOF EOF green
+context exclusive whole <<\[\s\]\[-\]\[\s\]\[\\\]END END green
+
+context ' ' green
+
+context " " green
+    keyword \\* brightgreen
+    keyword \\@ brightgreen
+    keyword \\$ brightgreen
+    keyword \\\\ brightgreen
+    keyword \\` brightgreen
+    keyword \\" brightgreen
+    keyword $(*) brightgreen
+    keyword ${*} brightgreen
+    keyword $\* brightred
+    keyword $@ brightred
+    keyword $# brightred
+    keyword $? brightred
+    keyword $- brightred
+    keyword $$ brightred
+    keyword $! brightred
+    keyword $_ brightred
+    keyword wholeright $\[0123456789\] brightred
+
+    keyword wholeright $+ brightgreen
+
+    keyword $ brightgreen
+
+context exclusive ` ` lightgray black
+    keyword '*' green
+    keyword " green
+    keyword \\` green
+    keyword ; brightcyan	
+    keyword $(*) brightgreen
+    keyword ${*} brightgreen
+    keyword { brightcyan
+    keyword } brightcyan
+
+    keyword $\* brightred
+    keyword $@ brightred
+    keyword $# brightred
+    keyword $? brightred
+    keyword $- brightred
+    keyword $$ brightred
+    keyword $! brightred
+    keyword $_ brightred
+
+    keyword wholeright $\[0123456789\] brightred
+
+    keyword wholeright $+ brightgreen
+
+    keyword $ brightgreen
+
+wholechars abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._
+
+    keyword whole break yellow
+    keyword whole case yellow
+    keyword whole continue yellow
+    keyword whole done yellow
+    keyword whole do yellow
+    keyword whole elif yellow
+    keyword whole else yellow
+    keyword whole esac yellow
+    keyword whole exit yellow
+    keyword whole fi yellow
+    keyword whole for yellow
+    keyword whole if yellow
+    keyword whole in yellow
+    keyword whole return yellow
+    keyword whole select yellow
+    keyword whole shift yellow
+    keyword whole then yellow
+    keyword whole trap yellow
+    keyword whole until yellow
+    keyword whole while yellow
+
+    keyword whole arch cyan
+    keyword whole ash cyan
+    keyword whole awk cyan
+    keyword whole basename cyan
+    keyword whole bash cyan
+    keyword whole bg_backup cyan
+    keyword whole bg_restore cyan
+    keyword whole bsh cyan
+    keyword whole cat cyan
+    keyword whole cd cyan
+    keyword whole chgrp cyan
+    keyword whole chmod cyan
+    keyword whole chown cyan
+    keyword whole cp cyan
+    keyword whole cpio cyan
+    keyword whole csh cyan
+    keyword whole date cyan
+    keyword whole dd cyan
+    keyword whole df cyan
+    keyword whole dmesg cyan
+    keyword whole dnsdomainname cyan
+    keyword whole doexec cyan
+    keyword whole domainname cyan
+    keyword whole echo cyan
+    keyword whole ed cyan
+    keyword whole egrep cyan
+    keyword whole ex cyan
+    keyword whole false cyan
+    keyword whole fgrep cyan
+    keyword whole find cyan
+    keyword whole fsconf cyan
+    keyword whole gawk cyan
+    keyword whole grep cyan
+    keyword whole gunzip cyan
+    keyword whole gzip cyan
+    keyword whole hostname cyan
+    keyword whole igawk cyan
+    keyword whole ipcalc cyan
+    keyword whole kill cyan
+    keyword whole ksh cyan
+    keyword whole linuxconf cyan
+    keyword whole ln cyan
+    keyword whole login cyan
+    keyword whole lpdconf cyan
+    keyword whole ls cyan
+    keyword whole mail cyan
+    keyword whole mkdir cyan
+    keyword whole mknod cyan
+    keyword whole mktemp cyan
+    keyword whole more cyan
+    keyword whole mount cyan
+    keyword whole mt cyan
+    keyword whole mv cyan
+    keyword whole netconf cyan
+    keyword whole netstat cyan
+    keyword whole nice cyan
+    keyword whole nisdomainname cyan
+    keyword whole ping cyan
+    keyword whole ps cyan
+    keyword whole pwd cyan
+    keyword whole red cyan
+    keyword whole remadmin cyan
+    keyword whole rm cyan
+    keyword whole rmdir cyan
+    keyword whole rpm cyan
+    keyword whole sed cyan
+    keyword whole set cyan
+    keyword whole setserial cyan
+    keyword whole sh cyan
+    keyword whole sleep cyan
+    keyword whole sort cyan
+    keyword whole stty cyan
+    keyword whole su cyan
+    keyword whole sync cyan
+    keyword whole taper cyan
+    keyword whole tar cyan
+    keyword whole tcsh cyan
+    keyword whole test cyan
+    keyword whole time cyan
+    keyword whole touch cyan
+    keyword whole true cyan
+    keyword whole umount cyan
+    keyword whole uname cyan
+    keyword whole userconf cyan
+    keyword whole usleep cyan
+    keyword whole vi cyan
+    keyword whole view cyan
+    keyword whole vim cyan
+    keyword whole xconf cyan
+    keyword whole ypdomainname cyan
+    keyword whole zcat cyan
+    keyword whole zsh cyan
