--- src/QxDao/QxSqlDatabase.cpp.orig	2013-02-14 10:08:17.589472872 +0100
+++ src/QxDao/QxSqlDatabase.cpp	2013-02-14 10:08:56.743221704 +0100
@@ -102,7 +102,7 @@
 
    if (bError) { QSqlDatabase::removeDatabase(sDbKeyNew); return QSqlDatabase(); }
    m_lstDbByThread.insert(lCurrThreadId, sDbKeyNew);
-   qDebug("[QxOrm] qx::QxSqlDatabase : create new database connection in thread '%ld' with key '%s'", (long)(lCurrThreadId), qPrintable(sDbKeyNew));
+   qDebug("[QxOrm] qx::QxSqlDatabase : create new database connection in thread '%ld' with key '%s'", static_cast<long>(reinterpret_cast<long long>(lCurrThreadId)), qPrintable(sDbKeyNew));
    return QSqlDatabase::database(sDbKeyNew);
 }
 
