diff -ru subtitlecomposer-0.4.3.old/src//core/sstring.cpp subtitlecomposer-0.4.3/src//core/sstring.cpp
--- subtitlecomposer-0.4.3.old/src//core/sstring.cpp	2008-09-04 08:51:34.000000000 +0200
+++ subtitlecomposer-0.4.3/src//core/sstring.cpp	2010-06-29 11:19:07.063811074 +0200
@@ -920,7 +920,7 @@
 	append( str );
 }
 
-SStringList::SStringList( const SStringList::SStringList& list ):
+SStringList::SStringList( const SStringList& list ):
 	QValueList<SString>( list )
 {
 }
