diff -rupN dockbarx_0.90.3/dockbarx/iconfactory.py dockbarx_0.90.3_new/dockbarx/iconfactory.py
--- dockbarx_0.90.3/dockbarx/iconfactory.py	2012-10-24 17:42:21.000000000 +0200
+++ dockbarx_0.90.3_new/dockbarx/iconfactory.py	2013-10-01 07:44:14.797403664 +0200
@@ -22,12 +22,12 @@ pygtk.require("2.0")
 import gtk
 import gc
 gc.enable()
-import Image
 import array
 import cairo
 import gio
 import os
 import weakref
+from PIL import Image
 from cStringIO import StringIO
 from math import pi, cos, sin
 
diff -rupN dockbarx_0.90.3/dockbarx/theme.py dockbarx_0.90.3_new/dockbarx/theme.py
--- dockbarx_0.90.3/dockbarx/theme.py	2012-10-24 17:42:21.000000000 +0200
+++ dockbarx_0.90.3_new/dockbarx/theme.py	2013-10-01 07:44:39.681415545 +0200
@@ -24,8 +24,8 @@ import gtk
 import gobject
 import cairo
 import os
-import Image
 import array
+from PIL import Image
 from common import ODict
 from common import Globals
 from log import logger
