--- webcam.c	2004-06-26 02:51:29.000000000 +0300
+++ webcam.c	2012-05-13 15:00:35.811165688 +0300
@@ -24,12 +24,11 @@
 #include <X11/Xlib.h>
 #include <Imlib2.h>
 #include <giblib.h>
-#include <curl/types.h>
 #include <curl/curl.h>
 #include <curl/easy.h>
 
 #include <sys/types.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 
 #include <net/if.h>
 
@@ -987,8 +986,6 @@
     /* shut up completely */
     if (ftp_debug)
       curl_easy_setopt(curl_handle, CURLOPT_VERBOSE, 1);
-    else
-      curl_easy_setopt(curl_handle, CURLOPT_MUTE, 1);
 
     curl_easy_setopt(curl_handle, CURLOPT_UPLOAD, 1);
 
