--- cinepaint-0.25.0/plug-ins/pnm/pnm.c	2008-06-16 09:07:51.000000000 +0200
+++ plug-ins/pnm/pnm.c	2009-01-06 08:55:44.000000000 +0100
@@ -660,7 +660,8 @@
   {
     if (mem_n > fsize-fpos)
     {
-      g_message ("%s:%d  storage size of %s is too small: %d",__FILE__,__LINE__,filename , (int)mem_n-fsize-fpos);
+      g_message ("%s:%d  storage size of %s is too small: %d",
+                 __FILE__,__LINE__,filename , (int)(mem_n - fsize - fpos));
       m_free( data )
       return FALSE;
     }
@@ -1018,17 +1019,17 @@
       } while (--len); pt = 0;
 
         if(byteps == 1)
-          snprintf( bytes, 84, "255" );
+          snprintf( bytes, 48, "255" );
         else
         if(byteps == 2)
-          snprintf( bytes, 84, "65535" );
+          snprintf( bytes, 48, "65535" );
         else
         if (byteps == 4)
         {
           if(G_BYTE_ORDER == G_LITTLE_ENDIAN)
-            snprintf( bytes, 84, "-1.0" );
+            snprintf( bytes, 48, "-1.0" );
           else
-            snprintf( bytes, 84, "1.0" );
+            snprintf( bytes, 48, "1.0" );
         }
         else
           g_message ("Error: byteps: %d", byteps);
