--- cinepaint-0.25.0/plug-ins/collect/collect.cpp	2007-11-26 21:15:16.000000000 +0100
+++ plug-ins/collect/collect.cpp	2010-04-29 08:27:59.076692683 +0200
@@ -262,7 +262,7 @@
 do_collect ()
 {
 	gint32		*layers;
-	gint32		 image_ID;
+	gint32		 image_ID = -1;
 	gint		 nlayers;
 
 	nlayers = -1;
@@ -292,7 +292,7 @@
 
           // if solo, renaming to the first selected image
           if(count > 1)
-            gimp_image_set_filename (image_ID, _("Collect"));
+            gimp_image_set_filename (image_ID, (char*)_("Collect"));
           else
             gimp_image_set_filename (image_ID, (char*)fc->value(1));
           layers = gimp_image_get_layers (load_image_ID, &nlayers);
@@ -301,7 +301,7 @@
             return -1;
 
           // renaming the layer to the original filename
-          gimp_layer_set_name (layers[0], strrchr(fc->value(1),'/')+1);
+          gimp_layer_set_name (layers[0], strrchr((char*) fc->value(1),'/')+1);
           GPrecisionType image_base_prec = gimp_drawable_precision (layers[0]);
           int base_gray = gimp_drawable_gray (layers[0]);
 
@@ -341,7 +341,7 @@
                     #endif
                     gimp_image_add_layer (image_ID, layers[0], 0); DBG
                     // set layer name to filename
-                    gimp_layer_set_name(layers[0], strrchr(fc->value(i),'/')+1);
+                    gimp_layer_set_name(layers[0], strrchr((char*) fc->value(i),'/')+1);
 
                     // searching max dimensions
                     if (gimp_image_width(load_image_ID) > max_w)
