diff --git a/app/cms.c b/app/cms.c
index 712e30a..5a176ea 100644
--- a/app/cms.c
+++ b/app/cms.c
@@ -1248,22 +1248,21 @@ cms_get_profile_from_file(char *file_name)
 #ifdef HAVE_OY
     if ( profile == NULL )
     {
+      char *pp_name = NULL;
       /* add path to non pathnamed file names */
       if(file_name && !strchr(file_name, OY_SLASH_C))
       {
-        char *pp_name = oyGetPathFromProfileName( file_name, my_alloc_func );
+        pp_name = oyGetPathFromProfileName( file_name, my_alloc_func );
 
         if(pp_name && strlen(pp_name))
         {
           fullFileName = (char*) calloc (MAX_PATH, sizeof(char));;
           sprintf(fullFileName, "%s%s%s", pp_name, OY_SLASH, file_name);;
         }
-        if(pp_name) free(pp_name);
       } else
       /* catch non correct paths */
       if(file_name && strchr(file_name, OY_SLASH_C))
       {
-        char *pp_name = NULL;
         char *ptr = NULL;
 
         ptr = strrchr(file_name, OY_SLASH_C);
@@ -1277,12 +1276,13 @@ cms_get_profile_from_file(char *file_name)
             fullFileName = (char*) calloc (MAX_PATH, sizeof(char));;
             sprintf(fullFileName, "%s%s%s", pp_name, OY_SLASH, ptr);;
           }
-          if(pp_name) free(pp_name);
         }
       }
 
       /* give lcms */
-      profile = cmsOpenProfileFromFile (fullFileName, "r");
+      if(pp_name)
+        profile = cmsOpenProfileFromFile (fullFileName, "r");
+      if(pp_name) free(pp_name);
     }
 #endif
 
