diff --git a/app/cms.c b/app/cms.c
index 73a9baa..c69b087 100644
--- a/app/cms.c
+++ b/app/cms.c
@@ -491,7 +491,7 @@ void cms_init_oyranos()
 
       /* obtain the net-color spec ICC device profile */
       error = oyOptions_SetFromText( &options,
-                               "//"OY_TYPE_STD"/config/net_color_region_target",
+                               "//"OY_TYPE_STD"/config/x_color_region_target",
                                      "yes", OY_CREATE_NEW );
       error = oyDeviceGetProfile( device, options, &p );
       oyOptions_Release( &options );
@@ -991,8 +991,8 @@ cms_read_standard_profile_dirs(icColorSpaceSignature space)
       for( i = 0; i < size; ++i)
       {
         temp_prof = oyProfiles_Get( iccs, i );
-#  ifdef DEBUG_
-        printf("%s %d: \"%s\" %s\n", i == current ? "*":" ", i,
+#  ifdef DEBUG
+        printf("%d: \"%s\" %s\n", i,
                oyProfile_GetText( temp_prof, oyNAME_DESCRIPTION ),
                oyProfile_GetFileName(temp_prof, 0));
 #  endif
diff --git a/plug-ins/rawphoto/rawphoto.c b/plug-ins/rawphoto/rawphoto.c
index 27378cd..f6b2b8f 100644
--- a/plug-ins/rawphoto/rawphoto.c
+++ b/plug-ins/rawphoto/rawphoto.c
@@ -108,12 +108,22 @@ static void query (void)
 
   dependency_error = system(command);
 
-  free(command);
-
   if(!(dependency_error > 0x200))
     fprintf(stderr, "The camera RAW converter UFRaw (ufraw-cinepaint) is installed. Disabling distributed rawphoto plug-in.\n");
-  else
-    fprintf(stderr, "UFRaw for CinePaint (ufraw-cinepaint) is not found. Enabling distributed rawphoto plug-in.\n");
+
+  if(!(dependency_error > 0x200))
+  {
+    snprintf( command, 1024, "export PATH=~/.cinepaint/plug-ins:%s:$PATH; ufraw-cinepaint", text ? text : "");
+
+    dependency_error = system(command);
+    if(!(dependency_error > 0x200))
+      fprintf(stderr, "The camera RAW converter lraw (is installed. Disabling distributed rawphoto plug-in.\n");
+  }
+
+  free(command);
+
+  if((dependency_error > 0x200))
+    fprintf(stderr, "UFRaw for CinePaint (ufraw-cinepaint) and lraw are not found. Enabling distributed rawphoto plug-in.\n");
   #endif
 
 
