--- plug-ins/tiff/tiff.c	2010-07-25 17:47:41.535636690 +0200
+++ plug-ins/tiff/tiff.c	2010-03-29 12:06:45.720703798 +0200
@@ -985,7 +985,7 @@
               size_t size = 0;
 #ifdef OYRANOS_H
 # if OYRANOS_API > 12
-              if( !oyCheckProfile (profile_name, "Cmyk") )
+              if( !oyCheckProfile (profile_name, 0) )
                 prof_mem = (char*)oyGetProfileBlock( profile_name, &size, myAllocFunc );
 # endif
 #endif 
