--- a/configure	2012-11-07 10:10:07.811288000 +0100
+++ b/configure	2012-11-07 10:21:23.467006000 +0100
@@ -18619,9 +18619,9 @@
     if test "x$FLTK_CONFIG" != "xno"; then
       FLTK_CONFIG="$FLTK_CONFIG"
         FLTK_LDFLAGS="`$FLTK_CONFIG --use-images --ldflags | sed \"$STRIPOPT\"`"
-        FLTK_LIBS_PURE="`echo $FLTK_LDFLAGS | sed  's/-O.// ; s/-isysroot /-L/ ; s/-arch ppc// ; s/-arch i386//'`"
+        FLTK_LIBS_PURE="`echo $FLTK_LDFLAGS | sed  's/ -O. // ; s/,-O[0-9]// ; s/-isysroot /-L/ ; s/-arch ppc// ; s/-arch i386//'`"
         FLTK_LIBS="\$(top_builddir)/lib/fl_i18n/libcinepaint_fl_i18n.la $FLTK_LIBS_PURE"
-        FLTK_CFLAGS=`$FLTK_CONFIG --cxxflags | sed  's/-O.// ; s/-isysroot /-L/ ; s/-arch ppc// ; s/-arch i386//'`
+        FLTK_CFLAGS=`$FLTK_CONFIG --cxxflags | sed  's/ -O. // ; s/,-O[0-9]// ; s/-isysroot /-L/ ; s/-arch ppc// ; s/-arch i386//'`
     else
       { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING:
 *** Check for fltk-config failed. Depending plug-ins may not compile.
