--- a/configure.in	2012-11-05 15:27:33.000000000 +0100
+++ b/configure.in	2012-11-07 07:18:30.000000000 +0100
@@ -513,9 +513,9 @@
     if test "x$FLTK_CONFIG" != "xno"; then
       FLTK_CONFIG="$FLTK_CONFIG"
         FLTK_LDFLAGS="`$FLTK_CONFIG --use-images --ldflags | sed \"$STRIPOPT\"`"
-        FLTK_LIBS_PURE="`echo $FLTK_LDFLAGS | sed  's/-O.// ; s/-isysroot /-L/ ; s/-arch ppc// ; s/-arch i386//'`"
+        FLTK_LIBS_PURE="`echo $FLTK_LDFLAGS | sed  's/ -O. // ; s/,-O[0-9]// ; s/-isysroot /-L/ ; s/-arch ppc// ; s/-arch i386//'`"
         FLTK_LIBS="\$(top_builddir)/lib/fl_i18n/libcinepaint_fl_i18n.la $FLTK_LIBS_PURE"
-        FLTK_CFLAGS=`$FLTK_CONFIG --cxxflags | sed  's/-O.// ; s/-isysroot /-L/ ; s/-arch ppc// ; s/-arch i386//'`
+        FLTK_CFLAGS=`$FLTK_CONFIG --cxxflags | sed 's/ -O. // ; s/,-O[0-9]// ; s/-isysroot /-L/ ; s/-arch ppc// ; s/-arch i386//'`
     else
       AC_MSG_WARN([
 *** Check for fltk-config failed. Depending plug-ins may not compile.
