*** usr/share/gnome-shell/extensions/dock@linuxdeepin.com/extension.js	2012-07-05 17:33:00.000000000 +0900
--- usr/share/gnome-shell/extensions/dock@linuxdeepin.com/extension_new.js	2012-07-21 13:40:43.491603526 +0900
***************
*** 27,33 ****
  
  const Gettext = imports.gettext;
  
! const THUMBNAIL_DEFAULT_WIDTH = 250;
  const THUMBNAIL_DISAPPEAR_TIMEOUT = 100; // milliseconds
  const TOOLTIP_DISAPPEAR_TIMEOUT = 100; // milliseconds
  const TOOLTIP_AUTO_DISAPPEAR_TIMEOUT = 3000; // milliseconds
--- 27,33 ----
  
  const Gettext = imports.gettext;
  
! const THUMBNAIL_DEFAULT_WIDTH = 400;
  const THUMBNAIL_DISAPPEAR_TIMEOUT = 100; // milliseconds
  const TOOLTIP_DISAPPEAR_TIMEOUT = 100; // milliseconds
  const TOOLTIP_AUTO_DISAPPEAR_TIMEOUT = 3000; // milliseconds
***************
*** 484,491 ****
              this._onActivate(Clutter.get_current_event());
          } else if (button == 2) {
              // Last workspace is always empty
!             let launchWorkspace = global.screen.get_workspace_by_index(global.screen.n_workspaces - 1);
!             launchWorkspace.activate(global.get_current_time());
              this.emit('launching');
              this.app.open_new_window(-1);
          }
--- 484,491 ----
              this._onActivate(Clutter.get_current_event());
          } else if (button == 2) {
              // Last workspace is always empty
!             //let launchWorkspace = global.screen.get_workspace_by_index(global.screen.n_workspaces - 1);
!             //launchWorkspace.activate(global.get_current_time());
              this.emit('launching');
              this.app.open_new_window(-1);
          }
***************
*** 1319,1336 ****
  
  
  function enable() {
!     // Move clock.
!     Main.panel._centerBox.remove_actor(dateMenu.actor);
  
!     dateMenu.actor.remove_actor(label);
!     box = new MyBox(label);
!     dateMenu.actor.add_actor(box.actor);
  
!     let children = Main.panel._rightBox.get_children();
!     Main.panel._rightBox.insert_child_at_index(dateMenu.actor, children.length-1);
  
!     // Extend left box.
!     panelConnectId = panel.actor.connect('allocate', allocate);
  
      // Remove application menu.
      Main.panel._leftBox.remove_actor(appMenu.actor);
--- 1319,1336 ----
  
  
  function enable() {
!     //// Move clock.
!     //Main.panel._centerBox.remove_actor(dateMenu.actor);
  
!     //dateMenu.actor.remove_actor(label);
!     //box = new MyBox(label);
!     //dateMenu.actor.add_actor(box.actor);
  
!     //let children = Main.panel._rightBox.get_children();
!     //Main.panel._rightBox.insert_child_at_index(dateMenu.actor, children.length-1);
  
!     //// Extend left box.
!     //panelConnectId = panel.actor.connect('allocate', allocate);
  
      // Remove application menu.
      Main.panel._leftBox.remove_actor(appMenu.actor);
