--- vnr-prefs.c	2012-02-26 22:06:15.000000000 +0400
+++ vnr-prefs_feh_support.c	2013-08-13 04:59:01.456017000 +0400
@@ -276,6 +276,7 @@
     gtk_combo_box_append_text(desktop_env, "XFCE");
     gtk_combo_box_append_text(desktop_env, "LXDE");
     gtk_combo_box_append_text(desktop_env, "FluxBox");
+    gtk_combo_box_append_text(desktop_env, "Feh");
     gtk_combo_box_set_active(desktop_env, prefs->desktop);
 
     gtk_box_pack_end (desktop_box, GTK_WIDGET(desktop_env), FALSE, FALSE, 0);
--- vnr-prefs.h	2012-02-26 22:05:10.000000000 +0400
+++ vnr-prefs_feh_support.h	2013-08-13 04:58:15.936017000 +0400
@@ -50,6 +50,7 @@
     VNR_PREFS_DESKTOP_XFCE,
     VNR_PREFS_DESKTOP_LXDE,
     VNR_PREFS_DESKTOP_FLUXBOX,
+    VNR_PREFS_FEH,
 } VnrPrefsDesktop;
 
 typedef enum{
--- vnr-window.c	2012-02-26 23:30:41.000000000 +0400
+++ vnr-window_feh_support.c	2013-08-13 04:57:27.866016000 +0400
@@ -1316,6 +1316,11 @@
 						"-f", VNR_FILE(win->file_list->data)->path, 
 						NULL);
 				break;
+			case VNR_PREFS_FEH:
+				execlp("feh", "feh",
+						"--bg-fill", VNR_FILE(win->file_list->data)->path,
+						NULL);
+				break;
 			default:
 				_exit(0);	
 		}
