diff -Nur util-vserver-0.30.216-pre2841.orig/scripts/vserver-build.rpm util-vserver-0.30.216-pre2841/scripts/vserver-build.rpm
--- util-vserver-0.30.216-pre2841.orig/scripts/vserver-build.rpm	2009-06-26 10:31:17.000000000 +1000
+++ util-vserver-0.30.216-pre2841/scripts/vserver-build.rpm	2009-08-24 23:34:07.000000000 +1000
@@ -87,7 +87,7 @@
 # before doing the operations.
 tmpdir=$($_MKTEMPDIR -p /var/tmp vserver-build.rpm.XXXXXX)
 manifest=$($_MKTEMP vserver-build.rpm.manifest.XXXXXX)
-trap "rm -rf $tmpdir $manifest" EXIT
+trap "rm -rf $tmpdir $manifest; base.__cleanup" EXIT
 cd "$tmpdir"			
 
 ######  do the real work
