--- Makefile	2005-07-30 17:26:38.000000000 +0200
+++ Makefile.arch	2009-04-27 12:17:03.000000000 +0200
@@ -16,11 +16,11 @@
 	-rm $(PROGNAME)
 
 install: $(PROGNAME)
-	mkdir -p "$(BINDIR)" &&\
-	install -m 755 $< "$(BINDIR)/"
+	mkdir -p "$(DESTDIR)$(BINDIR)" &&\
+	install -m 755 $< "$(DESTDIR)$(BINDIR)/"
 
 uninstall:
-	-rm "$(BINDIR)/$(PROGNAME)"
+	-rm "$(DESTDIR)$(BINDIR)/$(PROGNAME)"
 
 tarball:
 	@dir="TkPasMan-$(VERSION)"		&&\
