diff -Naur xen/Config.mk xen-a/Config.mk
--- xen/Config.mk	2013-08-20 16:01:32.175332341 -0600
+++ xen-a/Config.mk	2013-08-22 10:09:42.675332341 -0600
@@ -208,12 +208,12 @@
 SEABIOS_UPSTREAM_URL ?= http://xenbits.xen.org/git-http/seabios.git
 else
 OVMF_UPSTREAM_URL ?= git://xenbits.xen.org/ovmf.git
-QEMU_UPSTREAM_URL ?= git://xenbits.xen.org/qemu-upstream-unstable.git
-SEABIOS_UPSTREAM_URL ?= git://xenbits.xen.org/seabios.git
+QEMU_UPSTREAM_URL ?= git://git.qemu.org/qemu.git
+SEABIOS_UPSTREAM_URL ?= git://git.seabios.org/seabios.git
 endif
 OVMF_UPSTREAM_REVISION ?= b0855f925c6e2e0b21fbb03fab4b5fb5b6876871
-QEMU_UPSTREAM_REVISION ?= 1c514a7734b7f98625a0d18d5e8ee7581f26e50c
-SEABIOS_UPSTREAM_TAG ?= 3a28511b46f0c2af5fae1b6ed2b0c19d7913cee3
+QEMU_UPSTREAM_REVISION ?= master
+SEABIOS_UPSTREAM_TAG ?= master
 # Wed Jun 26 16:30:45 2013 +0100
 # xen: Don't perform SMP setup.
 
diff -Naur xen/tools/Makefile xen-a/tools/Makefile
--- xen/tools/Makefile	2013-08-22 10:03:16.508332341 -0600
+++ xen-a/tools/Makefile	2013-08-22 10:09:42.677332341 -0600
@@ -180,7 +180,7 @@
 		source=.; \
 	fi; \
 	cd qemu-xen-dir; \
-	$$source/configure --enable-xen --target-list=i386-softmmu \
+	$$source/configure --enable-xen --target-list=i386-softmmu,x86_64-softmmu,arm-softmmu \
 		--prefix=$(PREFIX) \
 		--source-path=$$source \
 		--extra-cflags="-I$(XEN_ROOT)/tools/include \
@@ -192,7 +192,24 @@
 		-L$(XEN_ROOT)/tools/xenstore" \
 		--bindir=$(LIBEXEC) \
 		--datadir=$(SHAREDIR)/qemu-xen \
-		--disable-kvm \
+		--disable-werror \
+                --enable-spice \
+                --enable-usb-redir \
+	 	--enable-libusb \
+                --enable-guest-agent \
+                --enable-sdl \
+		--enable-xen-pci-passthrough \
+		--enable-libusb \
+		--enable-tpm \
+		--enable-bluez \
+		--enable-vhost-net \
+		--enable-vde \
+                --enable-linux-aio \
+                --enable-smartcard-nss \
+                --enable-libiscsi \
+                --audio-drv-list=alsa,sdl \
+                --disable-docs \
+                --enable-mixemu \
 		--disable-docs \
 		--python=$(PYTHON) \
 		$(IOEMU_CONFIGURE_CROSS); \
diff -Naur xen/tools/libxl/libxl_dm.c xen-a/tools/libxl/libxl_dm.c
--- xen/tools/libxl/libxl_dm.c	2013-08-22 10:03:16.590332341 -0600
+++ xen-a/tools/libxl/libxl_dm.c	2013-08-22 10:10:52.508332341 -0600
@@ -38,7 +38,7 @@
 #ifdef QEMU_XEN_PATH
     return QEMU_XEN_PATH;
 #else
-    return libxl__abs_path(gc, "qemu-system-i386", libxl__libexec_path());
+    return libxl__abs_path(gc, "qemu-system-x86_64", libxl__libexec_path());
 #endif
 }
 
