--- pt1-c44e16dbb0e2/arib25/src/Makefile.org	2009-11-18 13:12:44.000000000 +0900
+++ pt1-c44e16dbb0e2/arib25/src/Makefile	2012-09-19 19:48:06.446004743 +0900
@@ -42,7 +42,6 @@
 	install -m755 $(TARGET_LIB) $(PREFIX)/lib/$(TARGET_LIB).$(VER)
 	ln -sf $(PREFIX)/lib/$(TARGET_LIB).$(VER) $(PREFIX)/lib/$(TARGET_LIB).$(MAJOR)
 	ln -sf $(PREFIX)/lib/$(TARGET_LIB).$(MAJOR) $(PREFIX)/lib/$(TARGET_LIB)
-	ldconfig
 
 install-headers:
 	mkdir -p $(DEST_HEADER)
