Index: src/gl/data/gl_data.cpp
===================================================================
--- src/gl/data/gl_data.cpp	(revision 1494)
+++ src/gl/data/gl_data.cpp	(revision 1495)
@@ -336,7 +336,7 @@
 	opt->skyrotatevector2.MakeUnit();
 }
 
-boolean IsLightmodeValid()
+bool IsLightmodeValid()
 {
 	return (glset.map_lightmode >= 0 && glset.map_lightmode <= 4) || glset.map_lightmode == 8;
 }
