diff --git a/docs/rql/build.mk b/docs/rql/build.mk
index e10f6e2..1329f3c 100644
--- a/docs/rql/build.mk
+++ b/docs/rql/build.mk
@@ -13,9 +13,11 @@ DOCS_HEADER := $(RQL_DOCS_DIR)/header.json
 .PHONY: docs
 docs: $(DOCS_JSON_OUT)
 
-$(DOCS_JSON_OUT): $(DOCS_JSON_CONVERT) $(DOCS_YAML_FILES) $(DOCS_HEADER) | $(WEB_ASSETS_BUILD_DIR)/js/.
+ifneq (1,$(USE_PRECOMPILED_WEB_ASSETS))
+  $(DOCS_JSON_OUT): $(DOCS_JSON_CONVERT) $(DOCS_YAML_FILES) $(DOCS_HEADER) | $(WEB_ASSETS_BUILD_DIR)/js/.
 	$P JSON_CONVERT
 	python $(DOCS_JSON_CONVERT) $(DOCS_YAML_DIR) $@ $(DOCS_HEADER)
+endif
 
 .PHONY: docs-validate
 docs-validate: $(DOCS_YAML_FILES) | $(RQL_DOCS_DIR)/build/.
