diff -Naur sdricohcs/sdricoh_cs/sdricoh_cs.c sdricohcs_write/sdricoh_cs/sdricoh_cs.c
--- sdricohcs/sdricoh_cs/sdricoh_cs.c	2007-05-10 14:06:18.000000000 +0200
+++ sdricohcs_write/sdricoh_cs/sdricoh_cs.c	2007-05-10 14:07:13.000000000 +0200
@@ -42,7 +42,7 @@
 #define DRIVER_VERSION "0.1"
 
 static unsigned int debug = 0;
-static unsigned int write = 0;
+static unsigned int write = 1;
 static unsigned int switchlocked = 0;
 
 //#define DEBUG
