--- hyperdex-build/Makefile.am	2013-02-21 18:18:57.375205311 -0800
+++ hyperdex-python2/Makefile.am	2013-02-21 18:22:40.825198721 -0800
@@ -522,10 +522,10 @@
 # upon a PHONY rule.
 if ENABLE_DOCS
 doc/man/hyperdex-daemon.1: doc/man/hyperdex-daemon.rst
-	sphinx-build -W -d ${abs_top_builddir}/doc/_build -b man doc doc/man
+	sphinx-build2 -W -d ${abs_top_builddir}/doc/_build -b man doc doc/man
 
 doc/HyperDex-$(VERSION).pdf:
-	sphinx-build -W -d ${abs_top_builddir}/doc/_build -b latex ${abs_top_srcdir}/doc ${abs_top_builddir}/doc/_build/latex
+	sphinx-build2 -W -d ${abs_top_builddir}/doc/_build -b latex ${abs_top_srcdir}/doc ${abs_top_builddir}/doc/_build/latex
 	make -C ${abs_top_builddir}/doc/_build/latex all-pdf
 	cp ${abs_top_builddir}/doc/_build/latex/HyperDex.pdf doc/HyperDex-$(VERSION).pdf
 endif
--- hyperdex-build/Makefile.am	2013-02-21 18:24:56.671861388 -0800
+++ hyperdex-python2/Makefile.am	2013-02-21 18:22:40.825198721 -0800
@@ -423,7 +423,7 @@
 pyx_verbose_0 = @echo "  PYX   " $@;
 
 client/python/hyperclient.c: client/python/hyperclient.pyx
-	$(pyx_verbose)cython client/python/hyperclient.pyx
+	$(pyx_verbose)cython2 client/python/hyperclient.pyx
 
 client_python_hyperclient_la_SOURCES = client/python/hyperclient.c
 client_python_hyperclient_la_CPPFLAGS = $(PYTHON_CPPFLAGS) $(CPPFLAGS)
