diff --git a/neo/framework/Licensee.h b/neo/framework/Licensee.h
index faecacb..9ef8322 100644
--- a/neo/framework/Licensee.h
+++ b/neo/framework/Licensee.h
@@ -63,7 +63,7 @@ If you have questions concerning this license or the applicable additional terms
 
 // RB begin
 // Linux info
-#define LINUX_DEFAULT_PATH				"/usr/local/games/rbdoom3-bfg"
+#define LINUX_DEFAULT_PATH				"/usr/share/games/doom3bfg"
 // RB end
 
 
