--- sapi/fpm/php-fpm.conf.in	2012-08-15 06:26:05.000000000 +0200
+++ sapi/fpm/php-fpm.conf.in	2012-08-16 14:17:41.809663595 +0200
@@ -12,1 +12,1 @@
-;include=etc/fpm.d/*.conf
+;include=etc/php/fpm.d/*.conf
@@ -148,1 +148,2 @@
-listen = 127.0.0.1:9000
+;listen = 127.0.0.1:9000
+listen = /run/php-fpm/php-fpm.sock
@@ -159,3 +160,3 @@
-;listen.owner = @php_fpm_user@
-;listen.group = @php_fpm_group@
-;listen.mode = 0660
+listen.owner = @php_fpm_user@
+listen.group = @php_fpm_group@
+listen.mode = 0660
@@ -458,1 +459,1 @@
-;chdir = /var/www
+;chdir = /srv/http
