--- src/MnUserTransformation.cpp~	2009-09-30 01:45:52.019587543 +0600
+++ src/MnUserTransformation.cpp	2009-09-30 01:46:46.618751517 +0600
@@ -2,6 +2,7 @@
 #include "Minuit/MnUserCovariance.h"
 
 #include <algorithm>
+#include <cstdio>
 
 class MnParStr {
 
@@ -27,7 +28,7 @@
   p[0] = 'p';
   p[4] = '\0';
   for(unsigned int i = 0; i < par.size(); i++) {
-    std::sprintf(p+1,"%i",i);
+    sprintf(p+1,"%i",i);
     add(p, par[i], err[i]);
   }
 }
