diff -rupN servicemenu-encfs_0.3-kde4/bin/encfs_kmenu.sh servicemenu-encfs_0.3-kde4-patch/bin/encfs_kmenu.sh
--- servicemenu-encfs_0.3-kde4/bin/encfs_kmenu.sh	2010-08-17 19:04:34.000000000 +0200
+++ servicemenu-encfs_0.3-kde4-patch/bin/encfs_kmenu.sh	2011-08-19 21:42:39.558849971 +0200
@@ -76,6 +76,49 @@ load_language_en () {
 
 }
 
+load_language_fr () {
+	msg_checkFuseModule_title="Erreur du module fuse"
+	msg_checkFuseModule_text="Vérifiez que fuse est chargé."
+
+	msg_checkFusePermission_title="Erreur du groupe fuse"
+	msg_checkFusePermission_text="Vérifiez si vous êtres dans le groupe fusep. Dans une console, vous pouvez taper : \"sudo adduser $USER fuse\""
+
+	msg_checkDestDir_title="Attention !"
+	msg_checkDestDir_text="Désolé, le répertoire de destination \"${destDir}\" existe déjà !"
+
+	msg_mkDestDir_title="Erreur de création de répertoire !"
+	msg_mkDestDir_text="Le répertoire de destination \"${destDir}\" ne peut pas être créé."
+
+	msg_askpassword_title="encFS : entrez votre phrase de passe ..."
+	msg_askpassword_text="Entrez votre phrase de passe pour le répertoire \"${encryptedDirName}\""
+
+	msg_mount_mountError_title="Erreur de montage !"
+	msg_mount_mountError_text="Le répertoire \"${destDir}\" ne peut pas être monté"
+
+	msg_mount_finish_title="Montage effectué"
+	msg_mount_finish_text="Répertoire \"${encryptedDirName}\" monté dans \"${destDir}\""
+
+	msg_umount_finish_title="Démontage effectué"
+	msg_umount_finish_text="Répertoire \"${destDir}\" démonté et supprimé"
+
+	msg_preCreate_warning_title="Attention !"
+	msg_preCreate_warning_text="Le répertoire semble non chiffré, voulez-vous continuer ?"
+	msg_preCreate_cancel_title="Opération annulée !"
+	msg_preCreate_cancel_text="La création du répertoire chiffré a été annulée."
+
+	msg_create_error_alreadyMounted="Répertoire de destination déjà monté !"
+	msg_create_error_alreadyExist="Le répertoire de destination existe déjà !"
+
+	msg_create_finish_title="Création effectuée"
+	msg_create_finish_text="Création et montage du répertoire chiffré effectuées."
+
+	msg_change_konsole_title="Changement du mot de passe de \"${encryptedDirName}\""
+
+	msg_info_error="Problème de récupération des informations de \"${encryptedDirName}\""
+	msg_info_title="Informations de \"${encryptedDirName}\""
+
+}
+
 load_language_it () {
 	msg_checkFuseModule_title="Errore modulo fuse"
 	msg_checkFuseModule_text="Controllare che il modulo fuse sia caricato"
@@ -130,7 +173,7 @@ checkFuseModule () {
 }
 
 checkFusePermission () {
-	if [ "$(groups | grep fuse)" = "" ]; then
+	if [ "$(groups | grep fuse)" = "" ] && [ -e "/etc/arch_release" ]; then
 		#echo "${msg_checkFusePermission_text}" #DEBUG
 		"${KDIALOG}" --title "${msg_checkFusePermission_title}" --caption "${msg_checkFusePermission_title}" --icon dialog-cancel --sorry "${msg_checkFusePermission_text}"
 		exit
@@ -210,7 +253,7 @@ change () {
 
 info () {
 	info=`encfsctl info "${encryptedDir}"` || info="${msg_info_error}"
-	kdialog --title "${msg_info_title}" --caption "${msg_info_title}" --icon help-about --msgbox "${info}"
+	${KDIALOG} --title "${msg_info_title}" --caption "${msg_info_title}" --icon help-about --msgbox "${info}"
 }
 
 ################################################ main
diff -rupN servicemenu-encfs_0.3-kde4/desktop/encfs.desktop servicemenu-encfs_0.3-kde4-patch/desktop/encfs.desktop
--- servicemenu-encfs_0.3-kde4/desktop/encfs.desktop	2010-08-17 18:54:57.000000000 +0200
+++ servicemenu-encfs_0.3-kde4-patch/desktop/encfs.desktop	2011-08-19 21:42:16.408975384 +0200
@@ -29,8 +29,10 @@ X-KDE-Submenu=&encFS
 [Desktop Action mount_umount_create]
 Icon=preferences-desktop-cryptography
 Name=Mount, unmount or encrypt
+Name[fr]=Monter, démonter ou chiffrer
 Name[it]=Monta, smonta o cifra
 Exec=encfs_kmenu.sh en encMountUmountCreate "%f"
+Exec[fr]=encfs_kmenu.sh fr encMountUmountCreate "%f"
 Exec[it]=encfs_kmenu.sh it encMountUmountCreate "%f"
 
 ################################################################################
@@ -38,13 +40,17 @@ Exec[it]=encfs_kmenu.sh it encMountUmoun
 [Desktop Action change]
 Icon=dialog-password
 Name=Change password
+Name[fr]=Changer le mot de passe
 Name[it]=Cambia password
 Exec=encfs_kmenu.sh en change "%f"
+Exec[fr]=encfs_kmenu.sh fr change "%f"
 Exec[it]=encfs_kmenu.sh it change "%f"
 
 [Desktop Action info]
 Icon=help-about
 Name=Informations
+Name[fr]=Informations
 Name[it]=Informazioni
 Exec=encfs_kmenu.sh en info "%f"
+Exec[fr]=encfs_kmenu.sh fr info "%f"
 Exec[it]=encfs_kmenu.sh it info "%f"
