diff -rupN src.orig/kde-service-menu-encfs_0.5/bin/kde-service-menu-encfs.sh src/kde-service-menu-encfs_0.5/bin/kde-service-menu-encfs.sh
--- src.orig/kde-service-menu-encfs_0.5/bin/kde-service-menu-encfs.sh	2012-01-25 09:53:23.765610104 +0100
+++ src/kde-service-menu-encfs_0.5/bin/kde-service-menu-encfs.sh	2012-01-25 09:54:47.062274516 +0100
@@ -101,34 +101,34 @@ load_language_en () {
 }
 
 load_language_fr () {
-	msg_error_title="encFS erreur!"
-	msg_warning_title="encFS avertissement!"
+	msg_error_title="encFS erreur !"
+	msg_warning_title="encFS avertissement !"
 	msg_info_title="encFS information"
-	msg_config_title="encFS: configuration"
+	msg_config_title="encFS : configuration"
 	msg_checkFuseModule="Vérifiez que fuse est chargé."
-	msg_checkFusePermission="Vérifiez si vous êtres dans le groupe fusep. Dans une console, vous pouvez taper : \"sudo adduser $USER fuse\""
+	msg_checkFusePermission="Vérifiez si vous appartenez au groupe fuse. Dans une console, vous pouvez taper : \"sudo adduser $USER fuse\""
 	msg_checkDestDir="Désolé, le répertoire de destination \"${destDir}\" existe déjà !"
 	msg_mkDestDir="Le répertoire de destination \"${destDir}\" ne peut pas être créé."
 	msg_askPassword_title="encFS : entrez votre phrase de passe ..."
 	msg_askPassword_text="Entrez votre phrase de passe pour le répertoire \"${sourceDirName}\""
-	msg_mount_abort="opération annulée"
+	msg_mount_abort="Opération annulée"
 	msg_mountError="Le répertoire \"${destDir}\" ne peut pas être monté"
 	msg_mount_finish="Répertoire \"${sourceDirName}\" monté dans \"${destDir}\""
 	msg_umount_finish="Répertoire \"${destDir}\" démonté et supprimé"
 	msg_preCreate="Le répertoire semble non chiffré, voulez-vous continuer?"
 	msg_preCreate_cancel="La création du répertoire chiffré a été annulée."
-	msg_change_password_error="Il ya des problèmes changer le mot de passe \"${sourceDirName}\"!\nIl n'est pas encFS répertoire crypté."
+	msg_change_password_error="Problème lors du changement de mot de passe de \"${sourceDirName}\"!\nCe n'est pas un répertoire chiffré avec encFS."
 	msg_change_password_konsole_title="Changer le mot de passe \"${sourceDirName}\""
 	msg_get_information_title="Informations de \"${sourceDirName}\""
-	msg_get_informations_problem="Problème de récupération des informations de \"${sourceDirName}\""
-	msg_config_cryptMode="Mode Crypt préféré:"
+	msg_get_informations_problem="Problème lors de la récupération des informations de \"${sourceDirName}\""
+	msg_config_cryptMode="Mode Crypt préféré : "
 	msg_config_cryptMode_standard="Standard, sécurisé et compatible."
 	msg_config_cryptMode_paranoia="Paranoïa, plus sûr mais moins compatible."
-	msg_config_autoUnmount="Automatiquement répertoire démonter?"
-	msg_config_autoUnmount_fm="Oui, après le gestionnaire de fichiers fermés."
-	msg_config_autoUnmount_no="Non, je le faire manuellement."
+	msg_config_autoUnmount="Démonter automatiquement le répertoire ?"
+	msg_config_autoUnmount_fm="Oui, après la fermeture du gestionnaire de fichiers."
+	msg_config_autoUnmount_no="Non, je le ferai manuellement."
 	msg_config_saved="Paramètres enregistrés."
-	msg_save_config_failed="Réglages pas sauvé!"
+	msg_save_config_failed="Réglages non enregistrés !"
 }
 
 load_language_it () {
diff -rupN src.orig/kde-service-menu-encfs_0.5/desktop/encfs.desktop src/kde-service-menu-encfs_0.5/desktop/encfs.desktop
--- src.orig/kde-service-menu-encfs_0.5/desktop/encfs.desktop	2012-01-25 09:53:23.762276803 +0100
+++ src/kde-service-menu-encfs_0.5/desktop/encfs.desktop	2012-01-25 09:53:46.278942674 +0100
@@ -72,7 +72,7 @@ Exec[ru]=kde-service-menu-encfs.sh ru en
 Icon=configure
 Name=Configure
 Name[de]=einrichten
-Name[fr]=Configurè
+Name[fr]=Configuration
 Name[it]=Configura
 Name[ru]=конфигурировать
 Exec=kde-service-menu-encfs.sh en encConfigure
