diff --git a/cpp/src/Freeze/MapI.cpp b/cpp/src/Freeze/MapI.cpp
index 820cef8..b717394 100644
--- a/cpp/src/Freeze/MapI.cpp
+++ b/cpp/src/Freeze/MapI.cpp
@@ -1497,14 +1497,7 @@ Freeze::MapHelperI::size() const
 
     try
     {
-#if DB_VERSION_MAJOR != 4
-#error Freeze requires DB 4.x
-#endif
-#if DB_VERSION_MINOR < 3
-        _db->stat(&s, 0);
-#else
         _db->stat(_connection->dbTxn(), &s, 0);
-#endif
     }
     catch(const ::DbException& dx)
     {
