--- gen_src/libdesktop-agnostic/config-client.c.orig	2011-03-13 19:30:13.000000000 +0100
+++ gen_src/libdesktop-agnostic/config-client.c		2013-10-27 13:52:56.860921752 +0100
@@ -318,9 +318,30 @@
 	GValue _tmp2_;
 	GValue single_instance;
 	GError * _inner_error_ = NULL;
+
 	g_return_val_if_fail (self != NULL, FALSE);
 	g_return_val_if_fail (instance_id != NULL, FALSE);
-	single_instance = (_tmp2_ = G_IS_VALUE (_tmp0_ = desktop_agnostic_config_schema_get_metadata_option (self->priv->_schema, "single_instance", &_inner_error_)) ? (g_value_init (&_tmp1_, G_VALUE_TYPE (_tmp0_ = desktop_agnostic_config_schema_get_metadata_option (self->priv->_schema, "single_instance", &_inner_error_))), g_value_copy (_tmp0_ = desktop_agnostic_config_schema_get_metadata_option (self->priv->_schema, "single_instance", &_inner_error_), &_tmp1_), _tmp1_) : (*(_tmp0_ = desktop_agnostic_config_schema_get_metadata_option (self->priv->_schema, "single_instance", &_inner_error_))), __vala_GValue_free0 (_tmp0_), _tmp2_);
+	_tmp0_ = desktop_agnostic_config_schema_get_metadata_option (self->priv->_schema, "single_instance", &_inner_error_) ;
+	if ( G_IS_VALUE (_tmp0_) ) {
+	    g_value_init (&_tmp1_, G_VALUE_TYPE ( _tmp0_ ));
+	    g_value_copy (_tmp0_, &_tmp1_);
+	} else {
+	    __vala_GValue_free0 (_tmp0_);
+	}
+	single_instance = _tmp1_;
+	
+	/* // I'm just quessing what this unholly mess was supposed to do. In any case, ^^this^^ fixed awn.
+	single_instance = (
+	    _tmp2_ = G_IS_VALUE (_tmp0_ = desktop_agnostic_config_schema_get_metadata_option (self->priv->_schema, "single_instance", &_inner_error_)) 
+		? 
+	    (g_value_init (&_tmp1_, G_VALUE_TYPE (_tmp0_ = desktop_agnostic_config_schema_get_metadata_option (self->priv->_schema, "single_instance", 
+	    &_inner_error_))), g_value_copy (_tmp0_ = desktop_agnostic_config_schema_get_metadata_option 
+	    (self->priv->_schema, "single_instance", &_inner_error_), &_tmp1_), _tmp1_)
+	     :
+	     (*(_tmp0_ = desktop_agnostic_config_schema_get_metadata_option (self->priv->_schema, "single_instance", &_inner_error_))), 
+	     __vala_GValue_free0 (_tmp0_), _tmp2_);
+	*/
+	
 	if (_inner_error_ != NULL) {
 		g_propagate_error (error, _inner_error_);
 		return FALSE;
