diff -ur gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.2/gccxml_builtins.h gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.2/gccxml_builtins.h
--- gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.2/gccxml_builtins.h	2010-09-07 21:03:58.000000000 +0200
+++ gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.2/gccxml_builtins.h	2012-07-20 18:44:41.228661183 +0200
@@ -130,7 +130,5 @@
 double      __complex__ __builtin_cpow(double __complex__, double __complex__);
 long double __complex__ __builtin_cpowl(long double __complex__, long double __complex__);
 
-#ifdef __llvm__
 extern unsigned int  __builtin_bswap32(unsigned int _data);
 extern unsigned long __builtin_bswap64(unsigned long _data);
-#endif
diff -ur gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.3/gccxml_builtins.h gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.3/gccxml_builtins.h
--- gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.3/gccxml_builtins.h	2010-09-07 21:04:00.000000000 +0200
+++ gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.3/gccxml_builtins.h	2012-07-20 18:44:41.228661183 +0200
@@ -182,7 +182,5 @@
 #define __is_empty(x) false
 #define __has_trivial_destructor(x) false
 
-#ifdef __llvm__
 extern unsigned int  __builtin_bswap32(unsigned int _data);
 extern unsigned long __builtin_bswap64(unsigned long _data);
-#endif
diff -ur gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.4/gccxml_builtins.h gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.4/gccxml_builtins.h
--- gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.4/gccxml_builtins.h	2010-09-07 21:04:00.000000000 +0200
+++ gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.4/gccxml_builtins.h	2012-07-20 18:44:41.234661405 +0200
@@ -152,7 +152,5 @@
 #define __is_empty(x) false
 #define __has_trivial_destructor(x) false
 
-#ifdef __llvm__
 extern unsigned int  __builtin_bswap32(unsigned int _data);
 extern unsigned long __builtin_bswap64(unsigned long _data);
-#endif
diff -ur gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.5/gccxml_builtins.h gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.5/gccxml_builtins.h
--- gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.5/gccxml_builtins.h	2010-09-07 21:04:01.000000000 +0200
+++ gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.5/gccxml_builtins.h	2012-07-20 18:44:41.235661440 +0200
@@ -153,7 +153,5 @@
 #define __has_trivial_destructor(x) false
 #define __has_trivial_constructor(x) false
 
-#ifdef __llvm__
 extern unsigned int  __builtin_bswap32(unsigned int _data);
 extern unsigned long __builtin_bswap64(unsigned long _data);
-#endif
diff -ur gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.6/gccxml_builtins.h gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.6/gccxml_builtins.h
--- gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.6/gccxml_builtins.h	2011-02-10 14:59:45.000000000 +0100
+++ gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.6/gccxml_builtins.h	2012-07-20 18:44:41.235661440 +0200
@@ -154,7 +154,5 @@
 #define __has_trivial_destructor(x) false
 #define __has_trivial_constructor(x) false
 
-#ifdef __llvm__
 extern unsigned int  __builtin_bswap32(unsigned int _data);
 extern unsigned long __builtin_bswap64(unsigned long _data);
-#endif
diff -ur gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.7/gccxml_builtins.h gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.7/gccxml_builtins.h
--- gccxml-0.9.0-20120719.orig/GCC_XML/Support/GCC/4.7/gccxml_builtins.h	2012-03-09 14:36:33.000000000 +0100
+++ gccxml-0.9.0-20120719/GCC_XML/Support/GCC/4.7/gccxml_builtins.h	2012-07-20 18:44:41.236661476 +0200
@@ -154,7 +154,5 @@
 #define __has_trivial_destructor(x) false
 #define __has_trivial_constructor(x) false
 
-#ifdef __llvm__
 extern unsigned int  __builtin_bswap32(unsigned int _data);
 extern unsigned long __builtin_bswap64(unsigned long _data);
-#endif
