--- craftcc35.orig/dial.h	1997-02-21 16:50:13.000000000 +0500
+++ craftcc35/dial.h	2012-02-28 13:23:38.931309890 +0600
@@ -35,7 +35,7 @@
 
      int  val_dx;
 
-     getline *g_value;
+     Getline *g_value;
      button  *up;
      button  *down;
      button  *up2;
--- craftcc35.orig/dial.hc	1997-02-21 16:50:13.000000000 +0500
+++ craftcc35/dial.hc	2012-02-28 13:23:51.297976629 +0600
@@ -58,7 +58,7 @@
      }.
 
 .  show_buttons
-     {g_value = new getline ("dial", w, s_value,
+     {g_value = new Getline ("dial", w, s_value,
                              x + title_dx, y,
                              val_dx, val_dy); 
       up   = new button (w, "/dial.up", 
--- craftcc35.orig/file_selector.h	1997-02-21 16:50:14.000000000 +0500
+++ craftcc35/file_selector.h	2012-02-28 13:24:31.791310188 +0600
@@ -16,8 +16,8 @@
    win      *w;
 
    selector *sel;
-   getline  *get_name;
-   getline  *get_pattern;
+   Getline  *get_name;
+   Getline  *get_pattern;
 
    int      num_files;
    char     file_list [max_selector_cases][128];
--- craftcc35.orig/file_selector.hc	1997-02-21 16:50:14.000000000 +0500
+++ craftcc35/file_selector.hc	2012-02-28 13:25:10.701310411 +0600
@@ -60,10 +60,10 @@
      }.
 
 .  open_getlines
-     {get_name    = new getline ("get_name", w, file_name,
+     {get_name    = new Getline ("get_name", w, file_name,
                                  name_x, name_y, name_dx, name_dy,
                                  true);
-      get_pattern = new getline ("get_pattern", w, file_pattern,
+      get_pattern = new Getline ("get_pattern", w, file_pattern,
                                  pattern_x, pattern_y, pattern_dx, pattern_dy,
                                  true);
      }.
@@ -115,10 +115,10 @@
      }.
 
 .  open_getlines
-     {get_name    = new getline ("get_name", w, file_name,
+     {get_name    = new Getline ("get_name", w, file_name,
                                  name_x, name_y, name_dx, name_dy,
                                  true);
-      get_pattern = new getline ("get_pattern", w, file_pattern,
+      get_pattern = new Getline ("get_pattern", w, file_pattern,
                                  pattern_x, pattern_y, pattern_dx, pattern_dy,
                                  true);
      }.
--- craftcc35.orig/getline.h	1997-02-21 16:50:14.000000000 +0500
+++ craftcc35/getline.h	2012-02-28 13:28:10.764644758 +0600
@@ -6,7 +6,7 @@
 #include "buttons.h"
 #include "bool.h"
 
-class getline
+class Getline
   {public:
 
    win     *w;
@@ -37,7 +37,7 @@
    bool    was_deactive;
    bool    is_single_char;
    
-   getline      (char name [],
+   Getline      (char name [],
                  win  *w,
                  char string [],
                  int  x,
@@ -48,7 +48,7 @@
                  bool show_histroy_top = false,
                  bool single_char      = false);
 
-   ~getline     ();
+   ~Getline     ();
 
    bool on               ();
    void check_activation ();
--- craftcc35.orig/getline.hc	1997-02-21 16:50:14.000000000 +0500
+++ craftcc35/getline.hc	2012-02-28 13:29:32.634645221 +0600
@@ -4,7 +4,7 @@
 
 #define hist_button_name         "/hist.open"
 
-getline::getline (char v_name   [],
+Getline::Getline (char v_name   [],
                   win  *v_w,        
                   char v_string [], 
                   int  v_x,         
@@ -87,14 +87,14 @@
 
   }
 
-getline::~getline ()
+Getline::~Getline ()
   {if (with_history)
       {delete (hist_button);
        delete (hist);
       };
   }
 
-void getline::refresh ()
+void Getline::refresh ()
   {int cx;
    int dl;
    int d;
@@ -148,7 +148,7 @@
 
   }
 
-void getline::check_activation ()
+void Getline::check_activation ()
   {int d;
 
    w->tick ();
@@ -156,7 +156,7 @@
       active (on ());
   }
 
-bool getline::on ()
+bool Getline::on ()
   {int xm;
    int ym;
    int button;
@@ -171,12 +171,12 @@
 
   }
 
-void getline::active (bool mode)
+void Getline::active (bool mode)
   {was_deactive = (is_active && ! mode);
    is_active    = mode;
   }
 
-bool getline::eval ()
+bool Getline::eval ()
   {check_activation ();
    check_hist_button;
    if (was_deactive)
@@ -341,7 +341,7 @@
 
   }
 
-bool getline::get ()
+bool Getline::get ()
   {bool any_edit;
 
    eval ();
--- craftcc35.orig/masks.hc	1997-02-21 16:52:06.000000000 +0500
+++ craftcc35/masks.hc	2012-02-28 13:31:41.197979285 +0600
@@ -330,7 +330,7 @@
                bool with_history)
  
   {win     *w;
-   getline *gl;
+   Getline *gl;
    button  *cancel;
    bool    is_cancel;
 
@@ -342,7 +342,7 @@
 .  open_win
      {w      = new win (name, "", 100, 100, dx + 40, 80);
       background (w);
-      gl     = new getline (name, w, s, 20, 20, dx, 20, with_history);
+      gl     = new Getline (name, w, s, 20, 20, dx, 20, with_history);
       cancel = new button (w, "cancel", (dx+40) / 2 - 60, 50);
      }.
 
--- craftcc35.orig/option_menu.hc	1997-03-21 17:20:47.000000000 +0500
+++ craftcc35/option_menu.hc	2012-02-28 13:32:46.834646321 +0600
@@ -185,7 +185,7 @@
   {dial    *d_is_self;
    dial    *d_is_suny;
    dial    *d_water_world;
-   getline *seed;
+   Getline *seed;
    char    seed_s [128];
 
    init;
@@ -217,7 +217,7 @@
       d_is_suny     = new dial (w, "", 0,200, 140,is_suny);
       d_water_world = new dial (w, "", 0,200, 180,is_water_world);
       strcpy (seed_s, "");
-      seed = new getline ("", w, seed_s, 200, 250, 90, 24);
+      seed = new Getline ("", w, seed_s, 200, 250, 90, 24);
      }.
 
 .  finish
@@ -270,8 +270,8 @@
   }
 
 void player_menu (int act_menu, int &tic)
-  {getline  *g_host;
-   getline  *g_name;
+  {Getline  *g_host;
+   Getline  *g_name;
 
    init;
    session;
@@ -298,8 +298,8 @@
      }.
 
 .  show_buttons
-     {g_host  = new getline  ("",w, host [pno],60,160,180, 20);
-      g_name  = new getline  ("",w, name [pno],60,230,180, 20);
+     {g_host  = new Getline  ("",w, host [pno],60,160,180, 20);
+      g_name  = new Getline  ("",w, name [pno],60,230,180, 20);
      }.
 
 .  finish
--- craftcc35.orig/player.h	1997-02-21 16:52:14.000000000 +0500
+++ craftcc35/player.h	2012-02-28 13:25:37.071310555 +0600
@@ -24,7 +24,7 @@
 
    color_trans *ct [max_players];
 
-   getline *g_msg;
+   Getline *g_msg;
  
    int    wdx;
    int    wdy;
--- craftcc35.orig/player.hc	1997-03-06 20:53:58.000000000 +0500
+++ craftcc35/player.hc	2012-02-28 13:26:05.281310717 +0600
@@ -68,7 +68,7 @@
 .  open_w_msg
      {w_msg = new win     (w_craft, "msg", host, 0,
                            w_craft_dy - w_msg_dy, w_msg_dx, w_msg_dy);
-      g_msg = new getline ("msg", w_msg, message, 1, 1,
+      g_msg = new Getline ("msg", w_msg, message, 1, 1,
                            w_msg_dx-2, w_msg_dy-2, false, false, true);
      }.
 
