diff -rupN minutor.orig/Makefile minutor/Makefile
--- minutor.orig/Makefile	2011-11-18 21:04:53.000000000 -0500
+++ minutor/Makefile	2012-02-28 14:58:45.228745366 -0500
@@ -16,7 +16,7 @@ EXEC=minutor
 all: $(EXEC)
 
 $(EXEC): $(OBJS) maplib
-	$(CC) -Wl,--as-needed -o $(EXEC) $(OBJS) $(LIBS) MinutorMap/MinutorMap.a
+	$(CC) -Wl,--as-needed -o $(EXEC) $(OBJS) $(LIBS) MinutorMap/MinutorMap.a -lz
 
 install: $(EXEC)
 	install -d $(BIN) $(DESKTOP) $(PIXMAP)
