--- src/multifast-v1.4.2/ahocorasick/Makefile	2013-11-10 15:54:48.000000000 -0500
+++ newMakefile	2014-09-23 19:20:39.036515941 -0400
@@ -1,21 +1,25 @@
 LIBNAME := libahocorasick.a
-CFLAGS := -Wall
+SHARED_LIBNAME := libahocorasick.so.0
+CFLAGS := -Wall -fPIC -O2
 HEADERS := actypes.h ahocorasick.h node.h
 BUILD_DIRECTORY := build
 
-all: $(LIBNAME)
- 
-$(LIBNAME): mkbuilddir ahocorasick.o node.o
-	ar -cvq $(BUILD_DIRECTORY)/$(LIBNAME) $(BUILD_DIRECTORY)/ahocorasick.o $(BUILD_DIRECTORY)/node.o
+all: $(BUILD_DIRECTORY)/$(LIBNAME) $(BUILD_DIRECTORY)/$(SHARED_LIBNAME)
 
-ahocorasick.o: ahocorasick.c $(HEADERS)
-	cc -o $(BUILD_DIRECTORY)/ahocorasick.o -c ahocorasick.c $(CFLAGS)
+$(BUILD_DIRECTORY)/$(LIBNAME): $(BUILD_DIRECTORY)/ahocorasick.o $(BUILD_DIRECTORY)/node.o
+	mkdir -p $(dir $@)
+	ar -cvq $@ $^
 
-node.o: node.c $(HEADERS)
-	cc -o $(BUILD_DIRECTORY)/node.o -c node.c $(CFLAGS)
+$(BUILD_DIRECTORY)/$(SHARED_LIBNAME): $(BUILD_DIRECTORY)/ahocorasick.o $(BUILD_DIRECTORY)/node.o
+	ld -shared -soname=$(SHARED_LIBNAME) $^ -lc -o $@
 
-mkbuilddir:
-	mkdir -p $(BUILD_DIRECTORY)
+$(BUILD_DIRECTORY)/ahocorasick.o: ahocorasick.c $(HEADERS)
+	mkdir -p $(dir $@)
+	cc -o $@ -c $< $(CFLAGS)
+
+$(BUILD_DIRECTORY)/node.o: node.c $(HEADERS)
+	mkdir -p $(dir $@)
+	cc -o $@ -c $< $(CFLAGS)
 
 clean:
 	rm -rf $(BUILD_DIRECTORY)
