diff -Nurd totem-pps-0.0.19.6//totem/plugin/ppslist2.py totem-pps-0.0.19.6.new//totem/plugin/ppslist2.py
--- totem-pps-0.0.19.6//totem/plugin/ppslist2.py	2010-11-16 03:22:10.000000000 +0200
+++ totem-pps-0.0.19.6.new//totem/plugin/ppslist2.py	2011-02-26 16:31:06.760000334 +0200
@@ -1,4 +1,4 @@
-#!/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 # -*- python -*-
 # Author: Luo Jinghua
diff -Nurd totem-pps-0.0.19.6//totem/plugin/ppslist.py totem-pps-0.0.19.6.new//totem/plugin/ppslist.py
--- totem-pps-0.0.19.6//totem/plugin/ppslist.py	2010-11-12 07:41:17.000000000 +0200
+++ totem-pps-0.0.19.6.new//totem/plugin/ppslist.py	2011-02-26 16:31:15.646667001 +0200
@@ -1,4 +1,4 @@
-#!/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 # -*- python -*-
 # Author: Luo Jinghua
diff -Nurd totem-pps-0.0.19.6//totem/plugin/urllib2cache.py totem-pps-0.0.19.6.new//totem/plugin/urllib2cache.py
--- totem-pps-0.0.19.6//totem/plugin/urllib2cache.py	2010-11-14 05:23:25.000000000 +0200
+++ totem-pps-0.0.19.6.new//totem/plugin/urllib2cache.py	2011-02-26 16:30:47.320000334 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 urllib2 caching handler
 Modified from http://code.activestate.com/recipes/491261/ by dbr
