diff -ur jTsSplitter.orig/src/jtssplitter/Mpeg2TSPacket.java jTsSplitter/src/jtssplitter/Mpeg2TSPacket.java
--- jTsSplitter.orig/src/jtssplitter/Mpeg2TSPacket.java	2011-08-13 14:54:13.836784313 +0900
+++ jTsSplitter/src/jtssplitter/Mpeg2TSPacket.java	2011-08-13 16:18:57.504525736 +0900
@@ -142,7 +142,7 @@
         int tableid;
         int sectionlength;
         calc cal = new calc();
-        ArrayList program_number = new ArrayList();
+        ArrayList<PATData> program_number = new ArrayList<PATData>();
         tableid = cal.byte2int(payload_temp, 0, 8);
         sectionlength = cal.byte2int(payload_temp, 12, 12);//-40-32;
         int patnum = sectionlength * 8 - 72;
@@ -563,7 +563,7 @@
         return pat_list_all;
     }
 
-    public void setPAT(ArrayList pat) {
+    public void setPAT(ArrayList<PATData> pat) {
         pat_list_now = pat;
         pat_list_all = pat;
     }
@@ -572,7 +572,7 @@
         return pmt_list;
     }
 
-    public void setPMT(ArrayList pmt) {
+    public void setPMT(ArrayList<PMTData> pmt) {
         pmt_list = pmt;
     }
 
