--- ATLAS/tune/blas/level1/SET/set_x86.c.orig	2012-08-04 09:16:03.604525291 +0200
+++ ATLAS/tune/blas/level1/SET/set_x86.c	2012-08-04 09:16:14.047858083 +0200
@@ -83,8 +83,8 @@
 #else
    #error "This kernel requires a gas x86 assembler!"
 #endif
-#ifndef ATL_SSE2
-   #error "This routine requires SSE2"
+#ifndef ATL_SSE1
+   #error "This routine requires SSE1"
 #endif
 
 #                %edi         %xmm0       %rsi
--- ATLAS/tune/blas/gemm/CASES/ATL_smm6x1x80_sse.c.orig	2012-08-04 20:21:27.342455572 +0200
+++ ATLAS/tune/blas/gemm/CASES/ATL_smm6x1x80_sse.c	2012-08-04 20:21:38.089121227 +0200
@@ -33,8 +33,8 @@
 #if !defined(ATL_GAS_x8632) && !defined(ATL_GAS_x8664)
    #error "This kernel requires x86 gas 32 or 64 bit assembler!"
 #endif
-#ifndef ATL_SSE2
-   #error "This routine requires SSE2!"
+#ifndef ATL_SSE1
+   #error "This routine requires SSE1!"
 #endif
 
 #ifdef SCPLX