--- SRC/Makefile.old	2011-02-08 17:39:23.966666767 +0100
+++ SRC/Makefile	2011-02-08 17:40:52.990000016 +0100
@@ -146,7 +146,7 @@
    sgeequb.o ssyequb.o spoequb.o sgbequb.o \
    sbbcsd.o slapmr.o sorbdb.o sorcsd.o
 
-DSLASRC = spotrs.o sgetrs.o spotrf.o sgetrf.o 
+DSLASRC = spotrs.o sgetrs.o sgetrf.o 
 
 SXLASRC = sgesvxx.o sgerfsx.o sla_gerfsx_extended.o sla_geamv.o		\
    sla_gercond.o sla_rpvgrw.o ssysvxx.o ssyrfsx.o			\
@@ -227,7 +227,7 @@
    cla_hercond_c.o cla_hercond_x.o cla_herpvgrw.o \
    cla_lin_berr.o clarscl2.o clascl2.o cla_wwaddw.o
 
-ZCLASRC = cpotrs.o cgetrs.o cpotrf.o cgetrf.o 
+ZCLASRC = 
 
 DLASRC = \
    dgbbrd.o dgbcon.o dgbequ.o dgbrfs.o dgbsv.o  \
