diff -crB old/Makefile new/Makefile
*** old/Makefile	2011-12-12 02:11:38.000000000 -0700
--- new/Makefile	2012-09-07 13:31:01.051548783 -0600
***************
*** 20,40 ****
  
  ifneq ($(KERNELRELEASE),)
  
!   LINUXVER_GOODFOR_CFG80211:=$(strip $(shell \
!     if [ "$(VERSION)" -ge "2" -a "$(PATCHLEVEL)" -ge "6" -a "$(SUBLEVEL)" -ge "32" -o "$(VERSION)" -ge "3" ]; then \
!       echo TRUE; \
!     else \
!       echo FALSE; \
!     fi \
!   ))
  
!     LINUXVER_WEXT_ONLY:=$(strip $(shell \
!     if [ "$(VERSION)" -ge "2" -a "$(PATCHLEVEL)" -ge "6" -a "$(SUBLEVEL)" -ge "17" ]; then \
!       echo FALSE; \
!     else \
!       echo TRUE; \
!     fi \
!   ))
  
    ifneq ($(API),)
      ifeq ($(API), CFG80211)
--- 20,28 ----
  
  ifneq ($(KERNELRELEASE),)
  
!   LINUXVER_GOODFOR_CFG80211:=TRUE
  
!     LINUXVER_WEXT_ONLY:=FALSE
  
    ifneq ($(API),)
      ifeq ($(API), CFG80211)
diff -crB old/src/wl/sys/wl_iw.h new/src/wl/sys/wl_iw.h
*** old/src/wl/sys/wl_iw.h	2011-12-12 02:11:38.000000000 -0700
--- new/src/wl/sys/wl_iw.h	2012-09-07 13:36:22.532948658 -0600
***************
*** 21,26 ****
--- 21,27 ----
  #ifndef _wl_iw_h_
  #define _wl_iw_h_
  
+ #include <linux/semaphore.h>
  #include <linux/wireless.h>
  
  #include <typedefs.h>
diff -crB old/src/wl/sys/wl_linux.c new/src/wl/sys/wl_linux.c
*** old/src/wl/sys/wl_linux.c	2012-09-06 21:03:28.000000000 -0600
--- new/src/wl/sys/wl_linux.c	2012-09-07 14:02:50.460116450 -0600
***************
*** 47,53 ****
  #include <linux/pci_ids.h>
  #define WLC_MAXBSSCFG		1	
  
- #include <asm/system.h>
  #include <asm/io.h>
  #include <asm/irq.h>
  #include <asm/pgtable.h>
--- 47,52 ----
***************
*** 200,205 ****
--- 199,205 ----
  
  static int nompc = 0;
  module_param(nompc, int, 0);
+ MODULE_LICENSE("Mixed/Proprietary");
  
  #ifdef quote_str
  #undef quote_str
