diff -urN xf86-input-wacom-0.11.0.orig/src/wcmValidateDevice.c xf86-input-wacom-0.11.0/src/wcmValidateDevice.c
--- xf86-input-wacom-0.11.0.orig/src/wcmValidateDevice.c	2011-05-23 08:39:16.000000000 +0200
+++ xf86-input-wacom-0.11.0/src/wcmValidateDevice.c	2011-05-23 08:40:13.000000000 +0200
@@ -254,6 +254,7 @@
 
 		case 0xE2: /* TPC with 2FGT */
 		case 0xE3: /* TPC with 2FGT */
+		case 0xE6: /* TPC with 2FGT */
 			priv->common->tablet_type = WCM_TPC;
 			priv->common->tablet_type |= WCM_LCD;
 			break;
