From f5fcbd8b4a74d1087b7efb4129a1240e86eaf2b5 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Mon, 20 Jun 2011 13:49:43 +1000
Subject: [PATCH] xsetwacom: add missing linebreak in error message.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Jason Gerecke <killertofu@gmail.com>
---
 tools/xsetwacom.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tools/xsetwacom.c b/tools/xsetwacom.c
index 048cf8d..9b5b12f 100644
--- a/tools/xsetwacom.c
+++ b/tools/xsetwacom.c
@@ -1937,7 +1937,7 @@ static void _set_matrix_prop(Display *dpy, XDevice *dev, const float fmatrix[9])
 
 	if (!matrix_prop)
 	{
-		fprintf(stderr, "Server does not support transformation");
+		fprintf(stderr, "Server does not support transformation\n");
 		return;
 	}
 
-- 
1.7.0.1

