diff -ur ufs-tools-0.1/include/ufs/fs.h ufs-tools-0.1b//include/ufs/fs.h
--- ufs-tools-0.1/include/ufs/fs.h	2004-01-12 05:41:23.000000000 -0700
+++ ufs-tools-0.1b//include/ufs/fs.h	2011-07-05 12:06:18.943562839 -0700
@@ -36,6 +36,7 @@
 #ifndef _UFS_FFS_FS_H_
 #define _UFS_FFS_FS_H_
 
+
 /*
  * Each disk drive contains some number of filesystems.
  * A filesystem consists of a number of cylinder groups.
@@ -310,7 +311,7 @@
 	int8_t   fs_old_flags;		/* old FS_ flags */
 	__u8	 fs_fsmnt[MAXMNTLEN];	/* name mounted on */
 	__u8	 fs_volname[MAXVOLLEN];	/* volume name */
-	__u64 fs_swuid;		/* system-wide uid */
+	int64_t  fs_swuid;		/* system-wide uid */
 	int32_t  fs_pad;		/* due to alignment of fs_swuid */
 /* these fields retain the current block allocation info */
 	int32_t	 fs_cgrotor;		/* last cg searched */
@@ -339,7 +340,7 @@
 	int32_t	 fs_contigsumsize;	/* size of cluster summary array */ 
 	int32_t	 fs_maxsymlinklen;	/* max length of an internal symlink */
 	int32_t	 fs_old_inodefmt;	/* format of on-disk inodes */
-	__u64 fs_maxfilesize;	/* maximum representable file size */
+	int64_t  fs_maxfilesize;	/* maximum representable file size */
 	int64_t	 fs_qbmask;		/* ~fs_bmask for use with 64-bit size */
 	int64_t	 fs_qfmask;		/* ~fs_fmask for use with 64-bit size */
 	int32_t	 fs_state;		/* validate fs_clean field */
Only in ufs-tools-0.1b//include/ufs: fs.h-orig
diff -ur ufs-tools-0.1/include/ufs/types.h ufs-tools-0.1b//include/ufs/types.h
--- ufs-tools-0.1/include/ufs/types.h	2004-01-12 02:45:13.000000000 -0700
+++ ufs-tools-0.1b//include/ufs/types.h	2011-07-05 11:58:28.033550979 -0700
@@ -32,10 +32,11 @@
 typedef	int32_t	__s32;
 typedef	uint32_t __u32;
 
+/*
 typedef uint64_t  __s64;
 typedef int64_t   __u64;
 
-/*
+
 typedef int32_t ufs1_daddr_t;
 typedef int64_t ufs2_daddr_t;
 typedef int64_t ufs_time_t;
Only in ufs-tools-0.1b//libufs: block.o
Only in ufs-tools-0.1b//libufs: cgroup.o
Only in ufs-tools-0.1b//libufs: inode.o
Only in ufs-tools-0.1b//libufs: libufs.so
diff -ur ufs-tools-0.1/libufs/Makefile ufs-tools-0.1b//libufs/Makefile
--- ufs-tools-0.1/libufs/Makefile	2004-01-12 05:30:15.000000000 -0700
+++ ufs-tools-0.1b//libufs/Makefile	2011-07-05 14:06:08.243743898 -0700
@@ -4,16 +4,16 @@
 LIBUFS="_LIBUFS"
 
 all : block cgroup inode type sblock
-	cc  -shared -o libufs.so  block.o cgroup.o inode.o type.o sblock.o
+	cc -shared -Wl,-soname,libufs.so.1 -o libufs.so.1.0  block.o cgroup.o inode.o type.o sblock.o 
 block : block.c
-	cc -c block.c  -I${INCLUDE} -D${LIBUFS}
+	cc -c block.c  -I /usr/include -I${INCLUDE} -D${LIBUFS} -fPIC
 cgroup : cgroup.c
-	cc -c cgroup.c  -I${INCLUDE} -D${LIBUFS}
+	cc -c cgroup.c -I /usr/include -I${INCLUDE} -D${LIBUFS} -fPIC
 inode : inode.c
-	cc -c inode.c  -I${INCLUDE} -D${LIBUFS}
+	cc -c inode.c -I /usr/include -I${INCLUDE} -D${LIBUFS} -fPIC
 type : type.c
-	cc -c type.c  -I${INCLUDE} -D${LIBUFS}
+	cc -c type.c -I /usr/include -I${INCLUDE} -D${LIBUFS} -fPIC
 sblock : sblock.c
-	cc -c sblock.c  -I${INCLUDE} -D${LIBUFS}
+	cc -c sblock.c -I /usr/include -I${INCLUDE} -D${LIBUFS} -fPIC
 clean :
 	rm *.o *.so
Only in ufs-tools-0.1b//libufs: Makefile.orig
Only in ufs-tools-0.1b//libufs: sblock.o
Only in ufs-tools-0.1b//libufs: type.o
diff -ur ufs-tools-0.1/Makefile ufs-tools-0.1b//Makefile
--- ufs-tools-0.1/Makefile	2004-01-12 02:45:13.000000000 -0700
+++ ufs-tools-0.1b//Makefile	2011-07-05 14:06:01.100410384 -0700
@@ -4,10 +4,10 @@
 CFLGS=-g
 
 all : mkufs mkfs
-	cc  -o mkufs  mkufs.o mkfs.o  libufs/libufs.so
+	cc  -o mkufs  mkufs.o mkfs.o  libufs/libufs.so.1.0 -I /usr/include/libbsd -lbsd
 mkufs : mkufs.c
-	cc -c mkufs.c  -I${INCLUDE} ${CFLGS}
+	cc -c mkufs.c  -I${INCLUDE} ${CFLGS} -I /usr/include/libbsd -lbsd
 mkfs : mkfs.c
-	cc -c mkfs.c  -I${INCLUDE} ${CFLGS}
+	cc -c mkfs.c  -I${INCLUDE} ${CFLGS} -I /usr/include/libbsd -lbsd
 clean:
 	rm *.o mkufs
diff -ur ufs-tools-0.1/mkfs.c ufs-tools-0.1b//mkfs.c
--- ufs-tools-0.1/mkfs.c	2004-01-12 02:45:13.000000000 -0700
+++ ufs-tools-0.1b//mkfs.c	2011-07-05 12:05:09.513561090 -0700
@@ -1053,8 +1053,9 @@
         return (arc4random());
 }
 
+/*
 u_int32_t arc4random()
 {
     return (rand());
 }
-
+*/
Only in ufs-tools-0.1b/: mkfs.o
Only in ufs-tools-0.1b/: mkufs
diff -ur ufs-tools-0.1/mkufs.c ufs-tools-0.1b//mkufs.c
--- ufs-tools-0.1/mkufs.c	2004-01-12 22:36:35.000000000 -0700
+++ ufs-tools-0.1b//mkufs.c	2011-07-05 12:03:05.936891311 -0700
@@ -24,6 +24,11 @@
 #include<linux/hdreg.h>      /* for ioctl  */
 #include<stdarg.h>
 
+#include <stdlib.h>
+#include <err.h>
+#include <stdarg.h>
+
+
 #define MAXBSIZE        65536   /* must be power of 2 */
 
 #include"ufs/types.h"
@@ -102,7 +107,7 @@
 static struct disklabel *getdisklabel(int ,char *s);
 static void rewritelabel(char *s, struct disklabel *lp);
 static void usage(void);
-static void errx(int n,char *fmt, ...);
+/* static void errx(int n,char *fmt, ...); */
                                                                                 
 
 /*---------------------------------------------------------*/
@@ -548,6 +553,7 @@
 /*----------------------------------------------------
 simulation of BSD functions 
 ----------------------------------------------------*/
+/*
 void errx(int n,char *fmt, ...)
 {
         va_list pvar;
@@ -556,7 +562,9 @@
         va_end(pvar);
         exit(n);
 }
+*/
 /*----------------------------------------------------*/
+/*
 void err(int n,char *fmt, ...)
 {
         va_list pvar;
@@ -564,5 +572,6 @@
         vfprintf(stderr,fmt,pvar);
         va_end(pvar);
 }
+*/
 
 /*------------------------------------------------------*/
Only in ufs-tools-0.1b/: mkufs.o
Only in ufs-tools-0.1b/: README.txt
