From 25bca060eb42b58876d98f115c619ca15e66f655 Mon Sep 17 00:00:00 2001
From: Pale Moon <git-repo@palemoon.org>
Date: Sun, 22 Feb 2015 10:35:47 +0100
Subject: [PATCH] Fix crash reporter leftover build error on some Linux
 versions.

---
 content/media/plugins/MediaPluginHost.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/content/media/plugins/MediaPluginHost.cpp b/content/media/plugins/MediaPluginHost.cpp
index 785ba58..2067c7d 100644
--- a/content/media/plugins/MediaPluginHost.cpp
+++ b/content/media/plugins/MediaPluginHost.cpp
@@ -114,8 +114,6 @@ static bool IsOmxSupported()
     return false;
   }
 
-  ScopedGfxFeatureReporter reporter("Stagefright", forceEnabled);
-
   if (!forceEnabled) {
     nsCOMPtr<nsIGfxInfo> gfxInfo = do_GetService("@mozilla.org/gfx/info;1");
     if (gfxInfo) {
@@ -129,7 +127,6 @@ static bool IsOmxSupported()
     }
   }
 
-  reporter.SetSuccessful();
   return true;
 }
 
