--- oocairo-1.4/doc/lua-oocairo-fontface.pod.orig       2013-05-26 13:54:26.545314741 -0500
+++ oocairo-1.4/doc/lua-oocairo-fontface.pod    2013-05-26 13:55:02.658452432 -0500
@@ -1,3 +1,5 @@
+=encoding UTF-8
+
 =head1 Name
 
 lua-oocairo-fontface - Unscaled font face object
--- oocairo-1.4/doc/lua-oocairo-userfont.pod.orig       2013-05-26 13:55:44.181560816 -0500
+++ oocairo-1.4/doc/lua-oocairo-userfont.pod    2013-05-26 13:56:13.561401652 -0500
@@ -1,3 +1,5 @@
+=encoding UTF-8
+
 =head1 Name
 
 lua-oocairo - Lua API for the Cairo vector graphics library