diff -c -r ivy_generator_linux_qt4_1.3/src/Ivy.cpp ivy_generator_linux_qt4_1.3-new/src/Ivy.cpp
*** ivy_generator_linux_qt4_1.3/src/Ivy.cpp	2007-03-26 22:07:32.000000000 +0200
--- ivy_generator_linux_qt4_1.3-new/src/Ivy.cpp	2008-09-02 17:31:57.000000000 +0200
***************
*** 395,401 ****
  
  
  	//set data path
! 	path = "../textures/";
  
  
  	//create material for leafs
--- 395,401 ----
  
  
  	//set data path
! 	path = "/usr/share/ivy/textures/";
  
  
  	//create material for leafs
