diff -rup gemrb-0.7.1/CMakeLists.txt gemrb-0.7.1-new/CMakeLists.txt
--- gemrb-0.7.1/CMakeLists.txt	2012-07-15 21:07:51.000000000 +0200
+++ gemrb-0.7.1-new/CMakeLists.txt	2012-09-03 14:54:36.311547904 +0200
@@ -157,7 +157,7 @@ ENDIF()
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/")
 
 # Check for all the required and optional dependencies
-INCLUDE(FindPythonLibs)
+FIND_PACKAGE(PythonLibs 2.7)
 IF(PYTHONLIBS_FOUND)
 	MESSAGE(STATUS "Looking for Python libraries and headers: found")
 ELSE()
