diff -u usr/src/emgd-1.16.3196/emgd/drm/emgd_connector.c /usr/src/emgd-1.16.3196/emgd/drm/emgd_connector.c
--- usr/src/emgd-1.16.3196/emgd/drm/emgd_connector.c	2012-09-24 07:57:11.000000000 +0300
+++ /usr/src/emgd-1.16.3196/emgd/drm/emgd_connector.c	2013-03-18 21:34:34.284773347 +0200
@@ -299,7 +299,7 @@
 
 	/* Set the property value to the new one.  This doesn't actually change
      * anything on the HW. */
-	ret = drm_connector_property_set_value(connector, property, value);
+	ret = drm_object_property_set_value(&connector->base, property, value);
 	if (ret) {
 		return ret;
 	}
diff -u usr/src/emgd-1.16.3196/emgd/drm/emgd_drv.c /usr/src/emgd-1.16.3196/emgd/drm/emgd_drv.c
--- usr/src/emgd-1.16.3196/emgd/drm/emgd_drv.c	2013-03-18 21:47:50.712118707 +0200
+++ /usr/src/emgd-1.16.3196/emgd/drm/emgd_drv.c	2013-03-18 21:38:38.483450409 +0200
@@ -2365,7 +2365,7 @@
 } /* emgd_driver_irq_handler() */
 
 
-static int __devinit emgd_pci_probe(struct pci_dev *pdev,
+static int emgd_pci_probe(struct pci_dev *pdev,
 		const struct pci_device_id *ent)
 {
 	if (PCI_FUNC(pdev->devfn)) {
diff -u usr/src/emgd-1.16.3196/emgd/drm/emgd_fb.c /usr/src/emgd-1.16.3196/emgd/drm/emgd_fb.c
--- usr/src/emgd-1.16.3196/emgd/drm/emgd_fb.c	2012-09-24 07:57:11.000000000 +0300
+++ /usr/src/emgd-1.16.3196/emgd/drm/emgd_fb.c	2013-03-18 21:28:31.930069724 +0200
@@ -539,7 +539,7 @@
 				continue;
 		}
 
-		drm_connector_attach_property(drm_connector, new_prop, current_value);
+		drm_object_attach_property(&drm_connector->base, new_prop, current_value);
 		emgd_connector->properties[num_of_properties++] = new_prop;
 	}
 
