--- usr/src/emgd-1.16.3196/emgd/drm/emgd_mmap.c	2012-09-24 07:57:11.000000000 +0300
+++ usr/src/emgd-1.16.3196/emgd/drm/emgd_mmap.c	2013-01-17 19:09:01.067123360 +0200
@@ -101,7 +101,7 @@
 	 */
 	vma->vm_ops = &emgd_vm_ops;
 	vma->vm_private_data = chunk;
-	vma->vm_flags |= VM_RESERVED | VM_IO | VM_MIXEDMAP | VM_DONTEXPAND;
+	vma->vm_flags |= VM_IO | VM_MIXEDMAP | VM_DONTEXPAND | VM_DONTDUMP;
 	pgprot_val(vma->vm_page_prot) =
 		pgprot_val(vma->vm_page_prot) | _PAGE_CACHE_UC_MINUS;
 
--- usr/src/emgd-1.16.3196/Makefile	2012-09-24 07:57:11.000000000 +0300
+++ usr/src/emgd/1.16.3196/Makefile	2013-01-17 18:47:39.291313418 +0200
@@ -51,6 +51,7 @@
 	   -I$(EMGD_MOD_DIR)/emgd/pal/ch7036 \
 	   -I$(EMGD_MOD_DIR)/emgd/drm \
 	   -I$(KERNELDIR)/include/drm \
+	   -I$(KERNELDIR)/include/uapi/drm \
 	   -I/usr/src/linux-headers-2.6.32-5-common/include/drm \
 	   -I$(EMGD_MOD_DIR)/pvr/include4 \
 	   -I$(EMGD_MOD_DIR)/pvr/services4/include \
--- usr/src/emgd-1.16.3196/pvr/services4/srvkm/env/linux/mmap.c	2012-09-24 07:57:11.000000000 +0300
+++ usr/src/emgd-1.16.3196/pvr/services4/srvkm/env/linux/mmap.c	2013-01-17 19:11:12.622898655 +0200
@@ -639,7 +639,8 @@
     PVR_DPF((PVR_DBG_MESSAGE, "%s: Mapped psLinuxMemArea 0x%p\n",
          __FUNCTION__, psOffsetStruct->psLinuxMemArea));
 #endif
-    ps_vma->vm_flags |= VM_RESERVED;
+    ps_vma->vm_flags |= VM_DONTDUMP;
+    
     ps_vma->vm_flags |= VM_IO;
 
 
--- usr/src/emgd-1.16.3196/pvr/services4/srvkm/env/linux/mm.h	2012-09-24 07:57:11.000000000 +0300
+++ usr/src/emgd-1.16.3196/pvr/services4/srvkm/env/linux/mm.h	2013-01-17 19:08:09.090212231 +0200
@@ -30,6 +30,10 @@
 
 #include <asm/io.h>
 
+#ifndef VM_DONTDUMP
+#define VM_DONTDUMP VM_RESERVED
+#endif
+
 #define	PHYS_TO_PFN(phys) ((phys) >> PAGE_SHIFT)
 #define PFN_TO_PHYS(pfn) ((pfn) << PAGE_SHIFT)
 
--- usr/src/emgd-1.16.3196/pvr/services4/srvkm/env/linux/osfunc.c	2012-09-24 07:57:11.000000000 +0300
+++ usr/src/emgd-1.16.3196/pvr/services4/srvkm/env/linux/osfunc.c	2013-01-17 19:06:38.249369370 +0200
@@ -2381,8 +2381,7 @@
         goto error_release_mmap_sem;
     }
 
-
-    if ((psVMArea->vm_flags & (VM_IO | VM_RESERVED)) != (VM_IO | VM_RESERVED))
+    if ((psVMArea->vm_flags & (VM_IO | VM_DONTDUMP)) != (VM_IO | VM_DONTDUMP))
     {
         PVR_DPF((PVR_DBG_ERROR,
             "OSAcquirePhysPageAddr: Memory region does not represent memory mapped I/O (VMA flags: 0x%lx)", psVMArea->vm_flags));
--- usr/src/emgd-1.16.3196/emgd/drm/emgd_mmap.c	2013-01-17 19:28:51.000000000 +0200
+++ usr/src/emgd-1.16.3196/emgd/drm/emgd_mmap.c	2013-01-17 22:53:42.779456397 +0200
@@ -38,6 +38,9 @@
 #include <memory.h>
 #include <memlist.h>
 #include <io.h>
+#include "img_types.h"
+#include "services_headers.h"
+#include "mm.h"
 
 /*
  * Bottom 256MB reserved for display
