--- usr/src/emgd-1.16.3196/emgd/drm/emgd_drv.c	2012-09-24 07:57:11.000000000 +0300
+++ usr/src/emgd-1.16.3196/emgd/drm/emgd_drv.c	2013-01-16 17:17:54.000000000 +0200
@@ -2496,7 +2496,9 @@
 	.irq_postinstall    = emgd_driver_irq_postinstall,
 	.irq_uninstall      = emgd_driver_irq_uninstall,
 	.irq_handler        = emgd_driver_irq_handler,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,6,0)
 	.reclaim_buffers    = drm_core_reclaim_buffers,
+#endif
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,37)
 	.get_map_ofs        = drm_core_get_map_ofs,
 	.get_reg_ofs        = drm_core_get_reg_ofs,
