diff -r 6e3a8fd590de src/arch/MovieTexture/MovieTexture_FFMpeg.cpp
--- a/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp	Thu Jul 07 21:54:14 2011 -0400
+++ b/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp	Fri Jul 08 21:59:58 2011 +0300
@@ -519,7 +519,7 @@
 
 		++m_iFrameNumber;
 
-		if( m_Frame.pict_type == FF_B_TYPE )
+		if( m_Frame.pict_type == avcodec::FF_B_TYPE )
 			m_bHadBframes = true;
 
 		if( m_iFrameNumber == 0 )
