--- sm-ssc/Utils/build.sh.orig	2011-08-13 23:59:46.353081910 +0300
+++ sm-ssc/Utils/build.sh	2011-08-13 23:59:25.519566995 +0300
@@ -67,7 +67,6 @@
 	echo "CONFIGURE_OPTIONS are passed to StepMania's configure."
 	echo ''
 	echo '  -d,  --download             stop after ffmpeg download.'
-	echo '  -f,  --ffmpeg               stop after building ffmpeg.'
 	echo '  -c,  --configure            stop after configuring StepMania.'
 	echo '  -s,  --stepmania            stop after building StepMania'
 	echo '                               (do not copy binaries).'
@@ -98,7 +97,6 @@
 fi
 
 s_download=
-s_ffmpeg=
 s_configure=
 s_stepmania=
 num_jobs=1
@@ -107,7 +105,6 @@
 	shift
 	case $arg in
 		-d|--download)	s_download=yes		;;
-		-f|--ffmpeg)	s_ffmpeg=yes		;;
 		-c|--configure)	s_configure=yes		;;
 		-s|--stepmania)	s_stepmania=yes		;;
 		-h|--help)	usage 0			;;
@@ -118,71 +115,12 @@
 		*)		usage 1			;;
 	esac
 done
-version=0.6.1
-ffmpeg=ffmpeg-$version
-directory=http://ffmpeg.org/releases
-if [ ! -d $ffmpeg ]; then
-	message 'Downloading ffmpeg'
-	if which bzip2 &>/dev/null; then
-		zipcommand=jxf
-		ffmarc=ffmpeg-$version.tar.bz2
-	elif which gzip &>/dev/null; then
-		zipcommand=zxf
-		ffmarc=ffmpeg-$version.tar.gz	
-	else
-		failure 'Install either bzip2 or gzip.'	
-	fi
-	if which curl &>/dev/null; then
-		get='curl -O'
-	elif which wget &>/dev/null; then
-		get=wget
-	else
-		failure 'Install either curl or wget.'
-	fi
-	call $get $directory/$ffmarc
-	message 'Extracting ffmpeg'
-	call tar -$zipcommand $ffmarc
-	message 'Cleaning up temporary files'
-	call rm $ffmarc
-fi
 
-if [ -n "$s_download" ]; then exit 0; fi
-args='--enable-static --disable-debug
---enable-memalign-hack --disable-network --enable-small
---disable-encoders --disable-ffserver
---enable-demuxer=avi
---enable-demuxer=h261 --enable-demuxer=h263 --enable-demuxer=h264
---enable-demuxer=m4v --enable-demuxer=mjpeg --enable-demuxer=mov
---enable-demuxer=mpegps --enable-demuxer=mpegts
---enable-demuxer=mpegvideo --enable-demuxer=ogg
---enable-demuxer=rawvideo --enable-demuxer=yuv4mpegpipe
---enable-decoder=h261 --enable-decoder=h263
---enable-decoder=h263i --enable-decoder=h264 --enable-decoder=huffyuv
---enable-decoder=mjpeg --enable-decoder=mjpegb
---enable-decoder=mpeg_xvmc --enable-decoder=mpeg1video
---enable-decoder=mpeg2video --enable-decoder=mpeg4
---enable-decoder=mpegvideo --enable-decoder=msmpeg4v1
---enable-decoder=msmpeg4v2 --enable-decoder=msmpeg4v3
---enable-decoder=rawvideo --enable-decoder=theora
---enable-parser=h261 --enable-parser=h263 --enable-parser=h264
---enable-parser=mjpeg --enable-parser=mpeg4video
---enable-parser=mpegaudio --enable-parser=mpegvideo
---enable-parser=ac3
---extra-cflags=-Dattribute_deprecated='
-if [ ! -f $ffmpeg/_inst/lib/libavcodec.a ]; then
-	cd $ffmpeg
-	message 'Configuring ffmpeg'
-	call ./configure --prefix="`pwd`/_inst" $args
-	message 'Building ffmpeg'
-	call make --jobs="$num_jobs" install-libs install-headers
-	cd ..
-fi
-if [ -n "$s_ffmpeg" ]; then exit 0; fi
 if [ ! -f _build/src/config.h ]; then
 	message 'Configuring StepMania'
 	mkdir -p _build
 	cd _build
-	call ../configure --with-ffmpeg=../$ffmpeg/_inst "$@"
+	call ../configure --with-jpeg "$@"
 	cd ..
 fi
 if [ -n "$s_configure" ]; then exit 0; fi
