--- StepMania-3.9-src-orig/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp	2007-11-26 18:25:16.000000000 +0100
+++ StepMania-3.9-src/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp	2007-11-26 19:19:17.000000000 +0100
@@ -721,10 +721,18 @@
 	m_iTextureWidth = power_of_two(m_iImageWidth);
 	m_iTextureHeight = power_of_two(m_iImageHeight);
 
-    RageDisplay::PixelFormat pixfmt;
+	RageDisplay::PixelFormat pixfmt;
 	bool PreferHighColor = (TEXTUREMAN->GetPrefs().m_iMovieColorDepth == 32);
-	m_AVTexfmt = FindCompatibleAVFormat( pixfmt, PreferHighColor );
+	m_AVTexfmt = 1; //Force RGB24 since OpenGL likes it
 
+	pixfmt = DISPLAY->FindPixelFormat(	AVPixelFormats[m_AVTexfmt].bpp,
+					  			AVPixelFormats[m_AVTexfmt].masks[0],
+       								AVPixelFormats[m_AVTexfmt].masks[1],
+       								AVPixelFormats[m_AVTexfmt].masks[2],
+       								AVPixelFormats[m_AVTexfmt].masks[3],
+ 								true /* realtime */ );
+
+#if 0
 	if( m_AVTexfmt == -1 )
 		m_AVTexfmt = FindCompatibleAVFormat( pixfmt, !PreferHighColor );
 
@@ -761,7 +769,8 @@
 			break;
 		}
 	}
-	
+#endif
+
 	if( !m_img )
 	{
 		const AVPixelFormat_t *pfd = &AVPixelFormats[m_AVTexfmt];
