diff -Naur StepMania-3.9-src-orig/autoconf/m4/alsa.m4 StepMania-3.9-src/autoconf/m4/alsa.m4
--- StepMania-3.9-src-orig/autoconf/m4/alsa.m4	2003-11-03 00:22:57.000000000 +0100
+++ StepMania-3.9-src/autoconf/m4/alsa.m4	2008-01-22 23:17:53.000000000 +0100
@@ -8,7 +8,7 @@
 dnl enables arguments --with-alsa-prefix=
 dnl                   --with-alsa-enc-prefix=
 dnl
-AC_DEFUN(AM_PATH_ALSA,
+AC_DEFUN([AM_PATH_ALSA],
 [dnl Save the original CFLAGS, LDFLAGS, and LIBS
 alsa_save_CFLAGS="$CFLAGS"
 alsa_save_LDFLAGS="$LDFLAGS"
diff -Naur StepMania-3.9-src-orig/autoconf/m4/audio.m4 StepMania-3.9-src/autoconf/m4/audio.m4
--- StepMania-3.9-src-orig/autoconf/m4/audio.m4	2004-03-19 03:05:55.000000000 +0100
+++ StepMania-3.9-src/autoconf/m4/audio.m4	2008-01-22 23:17:32.000000000 +0100
@@ -1,4 +1,4 @@
-AC_DEFUN(SM_AUDIO, [
+AC_DEFUN([SM_AUDIO], [
 
 AC_ARG_WITH(vorbis, AC_HELP_STRING([--without-vorbis], [Disable Vorbis support]), with_vorbis=$enableval, with_vorbis=yes)
 
diff -Naur StepMania-3.9-src-orig/autoconf/m4/backtrace.m4 StepMania-3.9-src/autoconf/m4/backtrace.m4
--- StepMania-3.9-src-orig/autoconf/m4/backtrace.m4	2004-10-23 00:18:30.000000000 +0200
+++ StepMania-3.9-src/autoconf/m4/backtrace.m4	2008-01-22 23:19:03.000000000 +0100
@@ -1,5 +1,5 @@
 # See if we have a working backtrace_symbols.
-AC_DEFUN(SM_FUNC_BACKTRACE_SYMBOLS,
+AC_DEFUN([SM_FUNC_BACKTRACE_SYMBOLS],
 [
     AC_MSG_CHECKING(for working backtrace_symbols())
     AC_TRY_RUN(
@@ -15,7 +15,7 @@
     AC_MSG_RESULT($have_backtrace_symbols)
 ])
 
-AC_DEFUN(SM_FUNC_CXA_DEMANGLE,
+AC_DEFUN([SM_FUNC_CXA_DEMANGLE],
 [
     # Check for abi::__cxa_demangle (gcc 3.1+)
     AC_MSG_CHECKING(for working cxa_demangle)
@@ -45,7 +45,7 @@
 	
 # To support backtraces in the crash handler, we need two things: a call to
 # get backtrace pointers, and a way to convert them to symbols.
-AC_DEFUN(SM_CHECK_CRASH_HANDLER,
+AC_DEFUN([SM_CHECK_CRASH_HANDLER],
 [
     # See if we have a custom backtrace():
     case $host in
diff -Naur StepMania-3.9-src-orig/autoconf/m4/gtk-2.0.m4 StepMania-3.9-src/autoconf/m4/gtk-2.0.m4
--- StepMania-3.9-src-orig/autoconf/m4/gtk-2.0.m4	2003-07-28 10:57:13.000000000 +0200
+++ StepMania-3.9-src/autoconf/m4/gtk-2.0.m4	2008-01-22 23:15:17.000000000 +0100
@@ -5,7 +5,7 @@
 dnl Test for GTK+, and define GTK_CFLAGS and GTK_LIBS, if gthread is specified in MODULES, 
 dnl pass to pkg-config
 dnl
-AC_DEFUN(AM_PATH_GTK_2_0,
+AC_DEFUN([AM_PATH_GTK_2_0],
 [dnl 
 dnl Get the cflags and libraries from pkg-config
 dnl

diff -Naur StepMania-3.9-src-orig/autoconf/m4/opengl.m4 StepMania-3.9-src/autoconf/m4/opengl.m4
--- StepMania-3.9-src-orig/autoconf/m4/opengl.m4	2004-06-04 22:31:45.000000000 +0200
+++ StepMania-3.9-src/autoconf/m4/opengl.m4	2008-01-22 23:13:56.000000000 +0100
@@ -1,4 +1,4 @@
-AC_DEFUN(SM_X_WITH_OPENGL,
+AC_DEFUN([SM_X_WITH_OPENGL],
 [
     AC_PATH_X
     
diff -Naur StepMania-3.9-src-orig/autoconf/m4/sdl.m4 StepMania-3.9-src/autoconf/m4/sdl.m4
--- StepMania-3.9-src-orig/autoconf/m4/sdl.m4	2003-09-16 21:24:16.000000000 +0200
+++ StepMania-3.9-src/autoconf/m4/sdl.m4	2008-01-22 23:13:01.000000000 +0100
@@ -8,7 +8,7 @@
 dnl AM_PATH_SDL([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for SDL, and define SDL_CFLAGS and SDL_LIBS
 dnl
-AC_DEFUN(AM_PATH_SDL,
+AC_DEFUN([AM_PATH_SDL],
 [dnl 
 dnl Get the cflags and libraries from the sdl-config script
 dnl
diff -Naur StepMania-3.9-src-orig/autoconf/m4/tls.m4 StepMania-3.9-src/autoconf/m4/tls.m4
--- StepMania-3.9-src-orig/autoconf/m4/tls.m4	2004-12-17 03:24:16.000000000 +0100
+++ StepMania-3.9-src/autoconf/m4/tls.m4	2008-01-22 23:12:32.000000000 +0100
@@ -2,7 +2,7 @@
 # the resulting program actually runs, not whether the resulting TLS variables
 # work properly; that check is done at runtime, since we can run binaries
 # compiled with __thread on systems without TLS.
-AC_DEFUN(SM_TLS,
+AC_DEFUN([SM_TLS],
 [
     AC_MSG_CHECKING(for TLS)
     AC_TRY_RUN( [ static __thread int val; int main() { return 0; } ], have_tls=yes,have_tls=no,have_tls=no )
diff -Naur StepMania-3.9-src-orig/autoconf/m4/zlib.m4 StepMania-3.9-src/autoconf/m4/zlib.m4
--- StepMania-3.9-src-orig/autoconf/m4/zlib.m4	2003-12-10 09:07:54.000000000 +0100
+++ StepMania-3.9-src/autoconf/m4/zlib.m4	2008-01-22 23:11:55.000000000 +0100
@@ -1,4 +1,4 @@
-AC_DEFUN(SM_ZLIB,
+AC_DEFUN([SM_ZLIB],
 [
 	AC_CHECK_LIB(z, inflate, have_libz=yes, have_libz=no)
 	AC_CHECK_HEADER(zlib.h, have_libz_header=yes, have_libz_header=no)
