--- StepMania-3.9-src-orig/src/GameState.h	2004-08-29 23:35:14.000000000 -0500
+++ StepMania-3.9-src/src/GameState.h	2006-11-18 13:36:18.000000000 -0600
@@ -85,8 +85,8 @@
 	bool AnyPlayersAreCpu() const;
 
 	void GetCharacters( vector<Character*> &apCharactersOut );
-	Character* GameState::GetRandomCharacter();
-	Character* GameState::GetDefaultCharacter();
+	Character* GetRandomCharacter();
+	Character* GetDefaultCharacter();
 
 	PlayerController	m_PlayerController[NUM_PLAYERS];
 	

--- StepMania-3.9-src-orig/src/StdString.h	2005-09-30 03:09:01.000000000 -0500
+++ StepMania-3.9-src/src/StdString.h	2006-11-18 13:36:18.000000000 -0600
@@ -1087,7 +1087,7 @@
 
 		while ( (nIdx=this->find(szOld, nIdx)) != MYBASE::npos )
 		{
-			replace(this->begin()+nIdx, this->begin()+nIdx+nOldLen, szRealNew);
+			MYBASE::replace(this->begin()+nIdx, this->begin()+nIdx+nOldLen, szRealNew);
 			nReplaced++;
 			nIdx += nNewLen;
 		}

--- StepMania-3.9-src-orig/src/NetworkSyncServer.h	2005-10-16 21:31:11.000000000 -0500
+++ StepMania-3.9-src/src/NetworkSyncServer.h	2006-11-18 13:36:18.000000000 -0600
@@ -117,7 +117,7 @@
 	void GameOver(PacketFunctions& Packet, const unsigned int clientNum);
 	void ScreenNetMusicSelectStatus(PacketFunctions& Packet, const unsigned int clientNum);
 	void AnalizeChat(PacketFunctions &Packet, const unsigned int clientNum);
-	CString StepManiaLanServer::ListPlayers();
+	CString ListPlayers();
 	void Kick(CString &name);
 	void Ban(CString &name);
 	bool IsBanned(CString &ip);

--- StepMania-3.9-src-orig/src/RageUtil_FileDB.h	2004-09-04 23:12:42.000000000 -0500
+++ StepMania-3.9-src/src/RageUtil_FileDB.h	2006-11-18 13:36:18.000000000 -0600
@@ -86,9 +86,9 @@
 	virtual void PopulateFileSet( FileSet &fs, const CString &sPath ) { }
 
 public:
-	FilenameDB::FilenameDB():
+	FilenameDB():
 		ExpireSeconds( -1 ) { }
-	virtual FilenameDB::~FilenameDB() { FlushDirCache(); }
+	virtual ~FilenameDB() { FlushDirCache(); }
 
 	void AddFile( const CString &sPath, int size, int hash, void *priv=NULL );
 	void DelFile( const CString &sPath );
