--- StepMania-3.9-src-orig/src/archutils/Unix/LinuxThreadHelpers.cpp	2004-10-22 17:23:38.000000000 -0500
+++ StepMania-3.9-src/src/archutils/Unix/LinuxThreadHelpers.cpp	2006-11-18 13:36:18.000000000 -0600
@@ -15,7 +15,7 @@
 #include <sys/wait.h>
 #include <sys/ptrace.h>
 #include <sys/stat.h>
-#include <linux/unistd.h>
+#include <sys/syscall.h>
 #define _LINUX_PTRACE_H // hack to prevent broken linux/ptrace.h from conflicting with sys/ptrace.h
 #include <sys/user.h>
 
@@ -29,7 +29,7 @@
 
 static bool g_bUsingNPTL = false;
 
-static _syscall0(pid_t,gettid)
+#define gettid() syscall(SYS_gettid)
 
 #ifndef _CS_GNU_LIBPTHREAD_VERSION
 #define _CS_GNU_LIBPTHREAD_VERSION 3
