--- StepMania-3.9-src-old/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp	2008-07-24 21:21:15.000000000 +0200
+++ StepMania-3.9-src/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp	2008-07-24 21:43:45.000000000 +0200
@@ -25,14 +25,14 @@
 #else
 extern "C"
 {
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 }
 #endif
 
 #if !defined(HAVE_IMG_CONVERT)
 extern "C"
 {
-#include <ffmpeg/swscale.h>
+#include <libswscale/swscale.h>
 }
 #endif // HAVE_IMG_CONVERT
     void img_convert__(AVPicture *dst, avcodec::PixelFormat dst_pix_fmt,
