--- mememaker.sh	2012-03-23 15:11:48.833214990 -0400
+++ mememaker2.sh	2012-03-23 15:12:00.949834801 -0400
@@ -94,13 +94,17 @@
 		echo -n Bottom label: 
 		read bottom
 	fi
+	top="$(echo $top|tr '[:lower:]' '[:upper:]')"
+	bottom="$(echo $bottom |tr '[:lower:]' '[:upper:]')"
 	width=`identify -format %w /tmp/meme.jpg`
-	convert -background '#0008' -fill white -gravity center -size ${width}x30 \
-			  label:"$bottom" \
-			  /tmp/meme.jpg +swap -gravity south -composite  /tmp/out.png
-	convert -background '#0008' -fill white -gravity center -size ${width}x30 \
-			  label:"$top" \
-			  /tmp/out.png +swap -gravity north -composite  /tmp/out.png
+	convert -font /usr/share/fonts/TTF/impact.ttf -pointsize 35 \
+		-background '#0000' -fill white -stroke black -strokewidth 1 \
+		-gravity center -size ${width}x70 label:"$bottom" /tmp/meme.jpg +swap \
+		-gravity south -composite  /tmp/out.png
+	convert -font /usr/share/fonts/TTF/impact.ttf -pointsize 35 \
+		-background '#0000' -fill white -stroke black -strokewidth 1 \
+		-gravity center -size ${width}x70 label:"$top" /tmp/out.png +swap \
+		-gravity north -composite  /tmp/out.png
 	imgur /tmp/out.png
 	rm /tmp/out.png
 	rm /tmp/meme.jpg
