--- a/usr/lib/python3.3/site-packages/system76driver/daemon.py	2013-07-15 18:24:02.000000000 -0400
+++ b/usr/lib/python3.3/site-packages/system76driver/daemon.py	2013-08-18 00:03:10.963872682 -0400
@@ -47,7 +47,7 @@
 # Products in this dict need the brightness hack
 DEFAULT_BRIGHTNESS = {
     'bonx7': ('acpi_video0', 64),
-    'galu1': ('acpi_video0', 70),
+    'galu1': ('intel_backlight', 3692),
     'gazp9': ('intel_backlight', 690),
     'sabc1': ('acpi_video0', 82),
 }
