--- a/usr/lib/python3.4/site-packages/system76driver/daemon.py	2013-12-02 23:32:52.407521841 -0500
+++ b/usr/lib/python3.4/site-packages/system76driver/daemon.py	2013-12-02 23:33:23.777348405 -0500
@@ -47,7 +47,7 @@
 DEFAULT_BRIGHTNESS = {
     'bonx7': ('acpi_video0', 64),
     'daru4': ('intel_backlight', 569),
-    'galu1': ('acpi_video0', 70),
+    'galu1': ('intel_backlight', 3692),
     'gazp9': ('intel_backlight', 690),
     'kudp1': ('acpi_video0', 64),
     'sabc1': ('acpi_video0', 82),
