--- /usr/lib/python3.4/site-packages/system76driver/daemon.py.bak	2014-05-10 01:05:07.962280385 -0400
+++ /usr/lib/python3.4/site-packages/system76driver/daemon.py	2014-05-10 01:06:16.242218815 -0400
@@ -73,7 +73,7 @@
 FIX_CAPS = (
     #('cap_net_raw+p', '/bin/ping6'),
     #('cap_net_raw+p', '/bin/ping'),
-    #('cap_net_raw+p', '/usr/bin/arping'),
+    ('cap_net_raw+p', '/usr/bin/arping'),
     ('cap_ipc_lock+ep', '/usr/bin/gnome-keyring-daemon'),
 )
 MARKER_CAPS_FIXED = '/var/lib/system76-driver/caps-fixed'
