diff -aur renpy-6.14.1-source.new/module/ffdecode.c renpy-6.14.1-source/module/ffdecode.c
--- renpy-6.14.1-source.new/module/ffdecode.c	2012-08-28 16:49:27.217523406 +1100
+++ renpy-6.14.1-source/module/ffdecode.c	2012-08-26 15:28:30.000000000 +1100
@@ -1152,6 +1152,7 @@
     AVFormatContext *ic;
     int err, i, ret, video_index, audio_index;
     AVPacket pkt1, *pkt = &pkt1;
+    AVFormatParameters params, *ap = &params;
     int codecs_locked = 0;
     
     // url_set_interrupt_cb(decode_interrupt_cb);
@@ -1161,6 +1162,8 @@
     is->video_stream = -1;
     is->audio_stream = -1;
 
+    memset(ap, 0, sizeof(*ap));
+
     is->io_context = rwops_open(is->rwops);
 
     codecs_locked = 1;
