diff --git a/module/ffdecode.c b/module/ffdecode.c
index d75bf67..70cb24f 100644
--- a/module/ffdecode.c
+++ b/module/ffdecode.c
@@ -1334,7 +1334,11 @@ static int video_thread(void *arg)
         /* NOTE: ipts is the PTS of the _first_ picture beginning in
            this packet, if any */
         is->video_st->codec->reordered_opaque= pkt->pts;
+#if LIBAVCODEC_VERSION_MAJOR >= 53
+	len1 = avcodec_decode_video2(is->video_st->codec,
+#else
         len1 = avcodec_decode_video(is->video_st->codec,
+#endif
                                     frame, &got_picture,
                                     pkt->data, pkt->size);
 
@@ -1521,7 +1525,11 @@ static int audio_decode_frame(VideoState *is, double *pts_ptr)
         /* NOTE: the audio packet can contain several frames */
         while (is->audio_pkt_size > 0) {
             data_size = sizeof(is->audio_buf1);
+#if LIBAVCODEC_VERSION_MAJOR >= 53
+	    len1 = avcodec_decode_audio3(dec,
+#else
             len1 = avcodec_decode_audio2(dec,
+#endif
                                         (int16_t *)is->audio_buf1, &data_size,
                                         is->audio_pkt_data, is->audio_pkt_size);
             if (len1 < 0) {
