--- src/poppler-0.14.0//Makefile.am	2010-03-15 15:25:51.000000000 -0400
+++ patch/Makefile.am	2010-06-23 22:07:12.000000000 -0400
@@ -34,7 +34,7 @@
 utils_subdir = utils
 endif
 
-SUBDIRS = goo fofi $(splash_subdir) poppler $(utils_subdir) $(glib_subdir) $(qt_subdir) test $(qt4_subdir) $(cpp_subdir)
+SUBDIRS = $(glib_subdir) $(qt_subdir) $(qt4_subdir)
 
 EXTRA_DIST =					\
 	README-XPDF				\
@@ -46,13 +46,9 @@
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA =				\
-	poppler.pc				\
-	$(cairo_pc_file)			\
-	$(splash_pc_file)			\
 	$(glib_pc_file)				\
 	$(qt_pc_file)                           \
-	$(qt4_pc_file)				\
-	$(cpp_pc_file)
+	$(qt4_pc_file)
 
 # Add CMake buildsystem files here so they get added on make dist
 EXTRA_DIST +=							\
--- src/poppler-0.14.0//glib/Makefile.am	2010-06-08 15:18:44.000000000 -0400
+++ patch/glib_Makefile.am	2010-06-23 22:01:39.000000000 -0400
@@ -67,7 +67,7 @@
 	poppler-private.h
 
 libpoppler_glib_la_LIBADD =				\
-	$(top_builddir)/poppler/libpoppler.la		\
+	-lpoppler                                       \
 	$(top_builddir)/poppler/libpoppler-cairo.la	\
 	$(POPPLER_GLIB_LIBS)				\
 	$(GDK_LIBS)					\
@@ -83,7 +83,7 @@
        test-poppler-glib.cc
 
 test_poppler_glib_LDADD =			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                               \
 	libpoppler-glib.la			\
 	$(POPPLER_GLIB_LIBS)			\
 	$(GDK_LIBS)				\
--- src/poppler-0.14.0//glib/demo/Makefile.am	2010-02-16 17:11:06.000000000 -0500
+++ patch/glib_demo_Makefile.am	2010-06-23 21:52:33.000000000 -0400
@@ -49,5 +49,5 @@
 
 LDADD =						\
 	$(top_builddir)/glib/libpoppler-glib.la	\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                               \
 	$(GTK_TEST_LIBS)
--- src/poppler-0.14.0//glib/reference/Makefile.am	2010-01-16 18:43:07.000000000 -0500
+++ patch/glib_reference_Makefile.am	2010-06-23 21:53:36.000000000 -0400
@@ -73,7 +73,7 @@
 	$(GDK_CFLAGS)					\
 	$(FREETYPE_CFLAGS)
 
-GTKDOC_LIBS=$(top_builddir)/poppler/libpoppler.la	\
+GTKDOC_LIBS=-lpoppler                               	\
 	$(top_builddir)/glib/libpoppler-glib.la		\
 	$(POPPLER_GLIB_LIBS)				\
 	$(FREETYPE_LIBS)				\
--- src/poppler-0.14.0//qt/Makefile.am	2010-03-27 09:08:46.000000000 -0400
+++ patch/qt_Makefile.am	2010-06-23 21:54:50.000000000 -0400
@@ -25,7 +25,7 @@
 	poppler-private.h
 
 libpoppler_qt_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                         	\
 	$(POPPLER_QT_LIBS)			\
 	$(FREETYPE_LIBS)			\
 	$(FONTCONFIG_LIBS)
@@ -42,7 +42,7 @@
        test-poppler-qt.cpp
 
 test_poppler_qt_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                           	\
 	libpoppler-qt.la			\
 	$(libqt3_LIBS)				\
 	$(FREETYPE_LIBS)
--- src/poppler-0.14.0//qt4/demos/Makefile.am	2010-01-16 18:43:07.000000000 -0500
+++ patch/qt4_demos_Makefile.am	2010-06-23 21:55:38.000000000 -0400
@@ -7,7 +7,7 @@
 	$(POPPLER_QT4_CFLAGS)
 
 LDADDS =					\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                           	\
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
--- src/poppler-0.14.0//qt4/src/Makefile.am	2010-06-08 15:28:43.000000000 -0400
+++ patch/qt4_src_Makefile.am	2010-06-23 21:56:14.000000000 -0400
@@ -52,7 +52,7 @@
 	poppler-page-transition-private.h
 
 libpoppler_qt4_la_LIBADD = 			\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                          	\
 	$(top_builddir)/poppler/libpoppler-arthur.la	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
--- src/poppler-0.14.0//qt4/tests/Makefile.am	2010-01-16 18:43:07.000000000 -0500
+++ patch/qt4_tests_Makefile.am	2010-06-23 21:57:00.000000000 -0400
@@ -7,7 +7,7 @@
 	$(POPPLER_QT4_CFLAGS)
 
 LDADDS =					\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                           	\
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
--- src/poppler-0.14.0//test/Makefile.am	2010-01-16 18:43:07.000000000 -0500
+++ patch/test_Makefile.am	2010-06-23 21:58:59.000000000 -0400
@@ -53,14 +53,14 @@
        gtk-splash-test.cc
 
 gtk_splash_test_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                          	\
 	$(GTK_TEST_LIBS)
 
 gtk_cairo_test_SOURCES =			\
        gtk-cairo-test.cc
 
 gtk_cairo_test_LDADD =					\
-	$(top_builddir)/poppler/libpoppler.la		\
+	-lpoppler                         		\
 	$(top_builddir)/glib/libpoppler-glib.la		\
 	$(CAIRO_LIBS)					\
 	$(GTK_TEST_LIBS)				\
@@ -71,7 +71,7 @@
 
 pdf_inspector_LDADD =					\
 	$(top_builddir)/poppler/libpoppler-cairo.la	\
-	$(top_builddir)/poppler/libpoppler.la		\
+	-lpoppler                             		\
 	$(CAIRO_LIBS)					\
 	$(FREETYPE_LIBS)				\
 	$(GTK_TEST_LIBS)				\
@@ -82,7 +82,7 @@
        perf-test-preview-dummy.cc
 
 perf_test_LDADD =				\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                         	\
 	$(FREETYPE_LIBS)					\
 	$(X_EXTRA_LIBS)
 
@@ -90,7 +90,7 @@
 	pdf-fullrewrite.cc
 
 pdf_fullrewrite_LDADD = \
-	$(top_builddir)/poppler/libpoppler.la
+	-lpoppler
 
 EXTRA_DIST =					\
 	pdf-operators.c				\
--- src/poppler-0.14.0//utils/Makefile.am	2010-01-16 18:43:07.000000000 -0500
+++ patch/utils_Makefile.am	2010-06-23 21:59:28.000000000 -0400
@@ -20,7 +20,7 @@
 	$(ABIWORD_CFLAGS)
 
 LDADD =						\
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler                         	\
 	$(UTILS_LIBS)				\
 	$(FONTCONFIG_LIBS)
 
