Index: src/squeezeplay/src/ui/jive_framework.c
===================================================================
--- src/squeezeplay/src/ui/jive_framework.c	(wersja 9563)
+++ src/squeezeplay/src/ui/jive_framework.c	(kopia robocza)
@@ -84,13 +84,7 @@
 	{ SDLK_RETURN,		JIVE_KEY_GO },
 	{ SDLK_LEFT,		JIVE_KEY_BACK },
 	{ SDLK_HOME,		JIVE_KEY_HOME },
-	{ SDLK_AudioPlay,	JIVE_KEY_PLAY },
-	{ SDLK_AudioPause,	JIVE_KEY_PAUSE },
 	{ SDLK_KP_PLUS,		JIVE_KEY_ADD },
-	{ SDLK_AudioPrev,	JIVE_KEY_REW },
-	{ SDLK_AudioNext,	JIVE_KEY_FWD },
-	{ SDLK_AudioRaiseVolume,JIVE_KEY_VOLUME_UP },
-	{ SDLK_AudioLowerVolume,JIVE_KEY_VOLUME_DOWN },
 	{ SDLK_PAGEUP,		JIVE_KEY_PAGE_UP },
 	{ SDLK_PAGEDOWN,	JIVE_KEY_PAGE_DOWN },
 	{ SDLK_PRINT,		JIVE_KEY_PRINT },
@@ -101,9 +95,7 @@
 	{ SDLK_F4,              JIVE_KEY_PRESET_4 },
 	{ SDLK_F5,              JIVE_KEY_PRESET_5 },
 	{ SDLK_F6,              JIVE_KEY_PRESET_6 },
-	{ SDLK_AudioMute,       JIVE_KEY_MUTE },
 	{ SDLK_POWER,           JIVE_KEY_POWER },
-	{ SDLK_Sleep,           JIVE_KEY_ALARM },
 	{ SDLK_UNKNOWN,		JIVE_KEY_NONE },
 };
 
@@ -368,13 +360,6 @@
 		jive_sdlevent_pump(L);
 	}
 
-	/* check queue size */
-	if (perfwarn.queue) {
-		if (SDL_EventQueueLength() > perfwarn.queue) {
-			printf("SDL_event_queue > %2d : %3d\n", perfwarn.queue, SDL_EventQueueLength());
-		}
-	}
-
 	/* process events */
 	process_timers(L);
 	while (SDL_PeepEvents(&event, 1, SDL_GETEVENT, SDL_ALLEVENTS) > 0 ) {
