diff -ru xfce4-clipman-plugin-1.2.2.orig/daemon/gsd-clipboard-manager.c xfce4-clipman-plugin-1.2.2/daemon/gsd-clipboard-manager.c
--- xfce4-clipman-plugin-1.2.2.orig/daemon/gsd-clipboard-manager.c	2012-02-08 18:00:11.000000000 -0800
+++ xfce4-clipman-plugin-1.2.2/daemon/gsd-clipboard-manager.c	2012-02-08 18:01:46.000000000 -0800
@@ -292,7 +292,6 @@
         timestamp = GDK_CURRENT_TIME;
 
         XSelectInput (display, window, PropertyChangeMask);
-        XSetSelectionOwner (display, XA_CLIPBOARD_MANAGER, window, timestamp);
 
         g_signal_connect_swapped (manager->priv->default_clipboard, "owner-change",
                                   G_CALLBACK (default_clipboard_owner_change), manager);
diff -ru xfce4-clipman-plugin-1.2.2.orig/panel-plugin/main-panel-plugin.c xfce4-clipman-plugin-1.2.2/panel-plugin/main-panel-plugin.c
--- xfce4-clipman-plugin-1.2.2.orig/panel-plugin/main-panel-plugin.c	2012-02-08 18:00:11.000000000 -0800
+++ xfce4-clipman-plugin-1.2.2/panel-plugin/main-panel-plugin.c	2012-02-08 18:04:47.000000000 -0800
@@ -24,6 +24,9 @@
 #include <locale.h>
 #endif
 
+#include <X11/Xlib.h> 
+#include <X11/Xatom.h>
+
 #include <glib/gstdio.h>
 #include <gtk/gtk.h>
 #include <libxfce4ui/libxfce4ui.h>
@@ -39,6 +42,7 @@
 static void             panel_plugin_register           (XfcePanelPlugin *panel_plugin);
 XFCE_PANEL_PLUGIN_REGISTER (panel_plugin_register);
 
+static gboolean		cb_remember_item		(MyPlugin *plugin);
 static gboolean         cb_button_pressed               (GtkButton *button,
                                                          GdkEventButton *event,
                                                          MyPlugin *plugin);
@@ -98,11 +102,45 @@
                             G_CALLBACK (plugin_free), plugin);
   g_signal_connect (plugin->menu, "deactivate",
                     G_CALLBACK (cb_menu_deactivate), plugin);
-
+  
+  plugin->remembering = FALSE;
+  g_timeout_add(250, (GtkFunction)cb_remember_item, plugin);
+  
   gtk_widget_show_all (GTK_WIDGET (panel_plugin));
 }
 
 static gboolean
+cb_remember_item (MyPlugin *plugin)
+{
+  Display *display = GDK_DISPLAY ();
+  Atom XA_CLIPBOARD = XInternAtom(display, "CLIPBOARD", 0);
+  Window owner;
+  
+  ClipmanHistoryItem *item;
+  GtkClipboard *clipboard;
+  gchar *text;
+  
+  if (plugin->remembering) {
+    return TRUE;
+  } else {
+    plugin->remembering = TRUE;
+    item = clipman_history_get_item_to_restore(plugin->history);
+    if (item != NULL) {
+      if (item->type == CLIPMAN_HISTORY_TYPE_TEXT) {
+        text = item->content.text;
+        clipboard = gtk_clipboard_get(GDK_SELECTION_CLIPBOARD);
+	owner = XGetSelectionOwner(display, XA_CLIPBOARD);
+	if (!owner) {
+	  gtk_clipboard_set_text(clipboard, text, -1);
+	}
+      }
+    }
+    plugin->remembering = FALSE;
+  }
+  return TRUE;
+}
+
+static gboolean
 cb_button_pressed (GtkButton *button,
                    GdkEventButton *event,
                    MyPlugin *plugin)
diff -ru xfce4-clipman-plugin-1.2.2.orig/panel-plugin/plugin.h xfce4-clipman-plugin-1.2.2/panel-plugin/plugin.h
--- xfce4-clipman-plugin-1.2.2.orig/panel-plugin/plugin.h	2012-02-08 18:00:11.000000000 -0800
+++ xfce4-clipman-plugin-1.2.2/panel-plugin/plugin.h	2012-02-08 18:05:17.000000000 -0800
@@ -58,6 +58,7 @@
   GtkMenuPositionFunc   menu_position_func;
   GtkWidget            *popup_menu;
   gulong                popup_menu_id;
+  gboolean		remembering;
 };
 
 /*
