--- gcc-4.8.2.original/gcc/var-tracking.c    2013-02-08 00:19:27.000000000 +0100
+++ gcc-4.8.2/gcc/var-tracking.c        2014-02-19 14:57:06.397110272 +0100
@@ -9798,7 +9798,9 @@
       val = cselib_lookup_from_insn (reg, GET_MODE (reg), 1,
                                     VOIDmode, get_insns ());
       preserve_value (val);
-      cselib_preserve_cfa_base_value (val, REGNO (reg));
+
+      if (reg != hard_frame_pointer_rtx && fixed_regs[REGNO (reg)])
+       cselib_preserve_cfa_base_value (val, REGNO (reg));
       expr = plus_constant (GET_MODE (stack_pointer_rtx),
                            stack_pointer_rtx, -ofst);
       cselib_add_permanent_equiv (val, expr, get_insns ());
