diff -urN nesicide/compiler/makefile nesicide.new/compiler/makefile
--- nesicide/compiler/makefile	2011-04-22 01:08:22.387111349 -0300
+++ nesicide.new/compiler/makefile	2011-04-22 01:25:31.399764852 -0300
@@ -1,17 +1,17 @@
 
 ifeq ("$(OSTYPE)", "cygwin")
+	CC := gcc-3
+	CFLAGS_LIB       := -mno-cygwin 
+	LIB_PREFIX   := lib
+	LIB_POSTFIX  := .a
+	EXEC_POSTFIX := .exe
+	LIB_CREATE   := ar crv
+else
 	CC := gcc
 	CFLAGS_LIB  := -g
 	LIB_PREFIX  := lib
 	LIB_POSTFIX := .a
 	LIB_CREATE  := ar crvs
-else
-	CC := gcc-3
-	CFLAGS_LIB       := -mno-cygwin 
-   LIB_PREFIX   := lib
-   LIB_POSTFIX  := .a
-	EXEC_POSTFIX := .exe
-	LIB_CREATE   := ar crv
 endif
 
 ASM=asm
