diff --git a/stepmake/aclocal.m4 b/stepmake/aclocal.m4
index 9cb66d7..b688191 100644
--- a/stepmake/aclocal.m4
+++ b/stepmake/aclocal.m4
@@ -972,7 +972,7 @@ AC_DEFUN(STEPMAKE_PYTHON_DEVEL, [
 	fi
     ])
     
-    AC_CHECK_PROGS(PYTHON_CONFIG, python-config, no)
+    AC_CHECK_PROGS(PYTHON_CONFIG, python2-config python-config, no)
 
     if test -z "$PYTHON_CFLAGS" -a "$PYTHON_CONFIG" != "no"; then
         # Clean out junk: http://bugs.python.org/issue3290
