diff -Naur gip-1.7.0-1.old/build.sh gip-1.7.0-1/build.sh
--- gip-1.7.0-1.old/build.sh	2009-01-07 15:51:38.000000000 -0600
+++ gip-1.7.0-1/build.sh	2009-01-07 15:51:42.000000000 -0600
@@ -17,8 +17,9 @@
 # Installation paths. (Plugins will be installed in the LIBDIR)
 INST_IN_LIBDIR=`find . -name "*.glade"`
 INST_IN_PIXMAPDIR=`find src/ -name "*.png"`
-INST_IN_DOCDIR=''
+INST_IN_DOCDIR='ChangeLog'
 INST_IN_GNOMEDOCDIR=`find doc/ -maxdepth 1 -name "*.xml" -o -path "doc/figures*"`
+INST_IN_APPDIR='gip.desktop'
 
 ###############################################################################
 #
@@ -150,11 +151,11 @@
   esac
 done
 
-INST_BINDIR="$INST_PREFIX/bin/"
-INST_LIBDIR="$INST_PREFIX/lib/$EXECUTABLE"
+INST_BINDIR="$INST_PREFIX/bin"
+INST_LIBDIR="$INST_PREFIX/share/$EXECUTABLE/data"
 INST_LOCALEDIR="$INST_PREFIX/share/locale"
-INST_PIXMAPDIR="$INST_PREFIX/lib/$EXECUTABLE"
-INST_DOCDIR="$INST_PREFIX/doc/$EXECUTABLE"
+INST_PIXMAPDIR="$INST_PREFIX/share/pixmaps/$EXECUTABLE"
+INST_DOCDIR="$INST_PREFIX/share/doc/$EXECUTABLE"
 INST_GNOMEDOCDIR="$INST_PREFIX/share/gnome/help/$EXECUTABLE/C"
 INST_MIME_XML="$INST_PREFIX/share/mime/packages/"
 INST_MIME_KEY="$INST_PREFIX/share/mime-info/"
@@ -280,10 +281,7 @@
 echo "# Installation in $INST_PREFIX";
 echo "####################################################################";
 if [ ! -d "$INST_PREFIX" ]; then
-  echo "Installation prefix directory $INST_PREFIX does not exist!" >&2
-  echo "Use \"--prefix prefix\" to specify an existing directory." >&2
-  echo "Installation aborted." >&2
-  exit 1
+  make_installdirs.sh "$INST_PREFIX" || exit 1
 fi
 
 echo Creating directories...
@@ -294,26 +292,31 @@
 make_installdirs.sh $CREATEDIRS                       || exit 1
 
 echo Installing $EXECUTABLE...
-cp $EXECUTABLE $INST_BINDIR                           || exit 1
+install -Dm755 $EXECUTABLE $INST_BINDIR               || exit 1
 
 echo Installing libraries in $INST_LIBDIR...
 if [ "$INST_IN_LIBDIR" != "" ]; then
-  cp $INST_IN_LIBDIR $INST_LIBDIR                     || exit 1
+  install -Dm644 $INST_IN_LIBDIR $INST_LIBDIR         || exit 1
 fi
 
 echo Installing pixmaps in $INST_PIXMAPDIR...
 if [ "$INST_IN_PIXMAPDIR" != "" ]; then
-  cp $INST_IN_PIXMAPDIR $INST_PIXMAPDIR               || exit 1
+  install -Dm644 $INST_IN_PIXMAPDIR $INST_PIXMAPDIR    || exit 1
+  ln -sf gip/calculator.png $INST_PIXMAPDIR/../gip.png || exit 1
 fi
 
 echo Installing documentation in $INST_DOCDIR...
 if [ "$INST_IN_DOCDIR" != "" ]; then
-  cp $INST_IN_DOCDIR $INST_DOCDIR                     || exit 1
+  install -Dm644 $INST_IN_DOCDIR $INST_DOCDIR         || exit 1
 fi
 
 echo Installing GNOME documentation in $INST_GNOMEDOCDIR...
 if [ -d "$INST_GNOMEDOCDIR" -a -w "$INST_GNOMEDOCDIR" ]; then
-  cp -r "$INST_IN_GNOMEDOCDIR" "$INST_GNOMEDOCDIR"
+  if [ -n "$INST_IN_GNOMEDOCDIR" ]; then
+    cp -r "$INST_IN_GNOMEDOCDIR" "$INST_GNOMEDOCDIR"
+    find "$INST_GNOMEDOCDIR" -type d -exec chmod 755 '{}' \;
+    find "$INST_GNOMEDOCDIR" -type f -exec chmod 644 '{}' \;
+  fi
 else
   echo "WARNING: GNOME documentation directory not found or it is not"  >&2
   echo "         writeable! Skipping GNOME documentation installation." >&2
@@ -326,12 +329,12 @@
 fi
 
 echo Installing application icons...
-cp src/pixmaps/calc32.png $INST_ICON/32x32/apps/calc.png
-cp src/pixmaps/calc48.png $INST_ICON/48x48/apps/calc.png
+install -Dm644 src/pixmaps/calc32.png $INST_ICON/32x32/apps/calc.png
+install -Dm644 src/pixmaps/calc48.png $INST_ICON/48x48/apps/calc.png
 
 echo Creating menu entry and registering MIME extension...
-cp $EXECUTABLE.desktop $INST_MENU_XDG     || exit 1
-cp $EXECUTABLE.xml     $INST_MIME_XML     || exit 1
+install -Dm644 $EXECUTABLE.desktop $INST_MENU_XDG     || exit 1
+install -Dm644 $EXECUTABLE.xml     $INST_MIME_XML     || exit 1
 
 echo
 echo $PROGRAMNAME has been installed successfully!
diff -Naur gip-1.7.0-1.old/gip.desktop.in gip-1.7.0-1/gip.desktop.in
--- gip-1.7.0-1.old/gip.desktop.in	2009-01-07 15:51:38.000000000 -0600
+++ gip-1.7.0-1/gip.desktop.in	2009-01-07 15:51:51.000000000 -0600
@@ -1,15 +1,15 @@
 [Desktop Entry]
-Encoding=UTF-8
-Name=%APP_NAME% IP Address Calculator
-Name[de]=%APP_NAME% IP-Adress-Rechner
-GenericName=IP Adress Calculator
+Version=1.0
+Name=%APP_NAME%
+Name[de]=%APP_NAME%
+GenericName=IP Address Calculator
 GenericName[de]=IP-Adress-Rechner
 Comment=Calculate IP addresses and networks
 Comment[de]=IP-Adressen und Netzbereiche berechnen
-Exec=%INSTALL_BIN%%APP_SYSNAME% %U
+Exec=%APP_SYSNAME% %U
 StartupNotify=true
 Terminal=false
-Icon=%INSTALL_LIB%/calculator.png
+Icon=%APP_SYSNAME%
 Type=Application
-Categories=GNOME;GTK;Utility;
+Categories=GNOME;GTK;Network;
 MimeType=application/x-%APP_SYSNAME%;
diff -Naur gip-1.7.0-1.old/installer/install_locales.sh gip-1.7.0-1/installer/install_locales.sh
--- gip-1.7.0-1.old/installer/install_locales.sh	2009-01-07 15:51:38.000000000 -0600
+++ gip-1.7.0-1/installer/install_locales.sh	2009-01-07 15:51:42.000000000 -0600
@@ -6,8 +6,8 @@
   LOCALEFILES="$@"
   for FILE in $LOCALEFILES; do
     LOCALE=`echo $FILE | sed 's/.*\///; s/\.mo//'`
-    mkdir -p $POTDIR/$LOCALE/LC_MESSAGES/
-    cp $FILE $POTDIR/$LOCALE/LC_MESSAGES/$DOMAIN.mo || return 1
+    install -dm755 $POTDIR/$LOCALE/LC_MESSAGES/
+    install -Dm644 $FILE $POTDIR/$LOCALE/LC_MESSAGES/$DOMAIN.mo || return 1
   done
   return 0
 }
diff -Naur gip-1.7.0-1.old/installer/make_installdirs.sh gip-1.7.0-1/installer/make_installdirs.sh
--- gip-1.7.0-1.old/installer/make_installdirs.sh	2009-01-07 15:51:38.000000000 -0600
+++ gip-1.7.0-1/installer/make_installdirs.sh	2009-01-07 15:51:42.000000000 -0600
@@ -6,7 +6,7 @@
 for DIR in "$@"; do
   if [ ! -e "$DIR" ]; then
     echo "Creating directory $DIR... "
-    mkdir -p $DIR
+    install -dm755 $DIR
     if [ "$?" -ne "0" ]; then
       echo "Failed to create $DIR."
       exit 1
