diff -ru src.orig/suspend-utils-1.0/fbsplash_funcs.c src//suspend-utils-1.0/fbsplash_funcs.c
--- src.orig/suspend-utils-1.0/fbsplash_funcs.c	2011-03-07 00:24:15.000000000 +0100
+++ src//suspend-utils-1.0/fbsplash_funcs.c	2011-03-09 18:24:57.585624052 +0100
@@ -70,7 +70,7 @@
 	fbsplashr_message_set(theme,
 		(mode==SPL_RESUME) ? "Resuming..." : "Suspending..." );
 
-	fbsplashr_render_screen(theme, true, false, was_silent ? FBSPL_EFF_NONE : FBSPL_EFF_FADEIN);
+	fbsplashr_render_screen(theme, true, false, FBSPL_EFF_NONE);
 
 	have_input = !fbsplashr_input_init();
 
@@ -104,7 +104,7 @@
 	}
 
 	if (fbsplash_is_silent()) {
-		fbsplashr_render_screen(theme, true, false, was_silent ? FBSPL_EFF_NONE : FBSPL_EFF_FADEOUT);
+		fbsplashr_render_screen(theme, true, false, FBSPL_EFF_NONE);
 	}
 
 	fbsplashr_tty_silent_cleanup();
