diff -ru src.orig/suspend-utils-1.0/suspend.c src//suspend-utils-1.0/suspend.c
--- src.orig/suspend-utils-1.0/suspend.c	2011-03-07 00:24:15.000000000 +0100
+++ src//suspend-utils-1.0/suspend.c	2011-03-09 18:20:28.935624105 +0100
@@ -1708,7 +1708,7 @@
 	 * drop to 'verbose mode' after the freeze() call.
 	 */
 	splash.switch_to();
-	splash.progress(15);
+	// splash.progress(15);
 
 	if (error) {
 		suspend_error("Freeze failed.");
@@ -2499,7 +2499,7 @@
 		goto Restore_console;
 	}
 
-	splash.progress(5);
+	// splash.progress(5);
 
 #ifdef CONFIG_BOTH
 	/* If s2ram_hacks returns != 0, better not try to suspend to RAM */
@@ -2521,7 +2521,7 @@
 
 	sync();
 
-	splash.progress(10);
+	// splash.progress(10);
 
 	rlim.rlim_cur = 0;
 	rlim.rlim_max = 0;
