diff -ru src.orig/suspend-utils-1.0/fbsplash_funcs.c src//suspend-utils-1.0/fbsplash_funcs.c
--- src.orig/suspend-utils-1.0/fbsplash_funcs.c	2011-03-07 00:24:15.000000000 +0100
+++ src//suspend-utils-1.0/fbsplash_funcs.c	2011-03-09 18:08:08.685624046 +0100
@@ -109,8 +109,10 @@

 	fbsplashr_tty_silent_cleanup();

+/* avoid switching to ugly tty1
 	if (!was_silent && fbsplash_is_silent())
 		fbsplash_set_verbose(0);
+*/

 	fbsplashr_theme_free(theme);
 	theme = NULL;
diff -ru src.orig/suspend-utils-1.0/suspend.c src//suspend-utils-1.0/suspend.c
--- src.orig/suspend-utils-1.0/suspend.c	2011-03-07 00:24:15.000000000 +0100
+++ src//suspend-utils-1.0/suspend.c	2011-03-09 18:14:27.518957363 +0100
@@ -1839,11 +1839,13 @@
 	if (fd < 0)
 		return fd;

+/* avoid switching to ugly tty1 - allways switch to unused vt and clear
 	tiocl[0] = TIOCL_GETKMSGREDIRECT;
 	if (!ioctl(fd, TIOCLINUX, tiocl)) {
 		if (tiocl[0] > 0)
 			vt = tiocl[0];
 	}
+*/

 	clear_vt = 0;
 	error = ioctl(fd, VT_GETSTATE, &vtstat);
