diff -ru src.orig//suspend-utils-1.0/suspend.c src//suspend-utils-1.0/suspend.c
--- src.orig//suspend-utils-1.0/suspend.c	2011-08-04 15:11:21.987197928 +0200
+++ src//suspend-utils-1.0/suspend.c	2011-08-04 15:12:29.263868275 +0200
@@ -1866,6 +1866,9 @@
 	fd = open(vt_name, O_RDWR);
 	if (fd < 0)
 		return fd;
+// If already on default Fbsplash VT, don't change away from it
+if (*orig_vc != 16)
+{
 	error = ioctl(fd, VT_ACTIVATE, vt);
 	if (error) {
 		suspend_error("Could not activate the VT %d.", vt);
@@ -1878,6 +1881,7 @@
 		fflush(stderr);
 		goto Close_fd;
 	}
+}
 
 	if (clear_vt) {
 		char *msg = "\33[H\33[J";
