diff -ru src.orig/suspend-utils-1.0/splash.c src/suspend-utils-1.0/splash.c
--- src.orig/suspend-utils-1.0/splash.c	2011-03-07 00:24:15.000000000 +0100
+++ src/suspend-utils-1.0/splash.c	2011-08-04 11:12:08.188474130 +0200
@@ -205,7 +205,7 @@
 	if (!mode)
 		return;
 
-	printf("Looking for splash system... ");
+//	printf("Looking for splash system... ");
 
 	if (!bootsplash_open()) {
 		splash->finish      = bootsplash_finish;
@@ -236,12 +236,12 @@
 	} else if (0) {
 		/* add another splash system here */
 	} else {
-		printf("none\n");
+		printf("No splash system found!\n");
 		if (!open_input_fd())
 			splash->key_pressed = key_pressed;
 		return;
 	}
-	printf("found\n");
+//	printf("found\n");
 
 	splash->progress(0);
 }
