diff -u suspend-0.8.orig//fbsplash_funcs.c suspend-0.8//fbsplash_funcs.c
--- suspend-0.8.orig//fbsplash_funcs.c	2007-12-31 19:50:12.000000000 +0100
+++ suspend-0.8//fbsplash_funcs.c	2010-07-05 18:00:47.557210733 +0200
@@ -109,8 +109,10 @@

 	fbsplashr_tty_silent_cleanup();

+/* avoid switching to ugly tty1
 	if (!was_silent && fbsplash_is_silent())
 		fbsplash_set_verbose(0);
+*/

 	fbsplashr_theme_free(theme);
 	theme = NULL;
diff -u suspend-0.8.orig//suspend.c suspend-0.8//suspend.c
--- suspend-0.8.orig//suspend.c	2007-12-31 19:50:12.000000000 +0100
+++ suspend-0.8//suspend.c	2010-07-05 18:00:09.417602260 +0200
@@ -958,11 +958,13 @@
 	if (fd < 0)
 		return fd;

+/* avoid switching to ugly tty1 - allways switch to unused vt and clear
 	tiocl[0] = TIOCL_GETKMSGREDIRECT;
 	if (!ioctl(fd, TIOCLINUX, tiocl)) {
 		if (tiocl[0] > 0)
 			vt = tiocl[0];
 	}
+*/

 	clear_vt = 0;
 	error = ioctl(fd, VT_GETSTATE, &vtstat);
