diff -ru src.orig/suspend-utils-1.0/conf/suspend.conf src//suspend-utils-1.0/conf/suspend.conf
--- src.orig/suspend-utils-1.0/conf/suspend.conf	2011-03-07 00:24:15.000000000 +0100
+++ src//suspend-utils-1.0/conf/suspend.conf	2011-03-09 18:29:17.568957366 +0100
@@ -1,9 +1,14 @@
 snapshot device = /dev/snapshot
-resume device = <path_to_resume_device_file>
+## Swap-partition or partition containing the swapfile
+resume device = /dev/sda99
+## Output from swap-offset <swapfile>
+#resume offset = 123456
+## Try to limit the size of the image created on disk
 #image size = 350000000
+## Misc. options
 #suspend loglevel = 2
-#compute checksum = y
+compute checksum = y
 #compress = y
 #encrypt = y
 #early writeout = y
-#splash = y
+splash = y
