diff -ur a/Fl_MoxGen-0.05/write_pdf.c b/Fl_MoxGen-0.05/write_pdf.c
--- a/Fl_MoxGen-0.05/write_pdf.c	2009-12-22 04:55:01.000000000 +0200
+++ b/Fl_MoxGen-0.05/write_pdf.c	2010-12-11 22:08:23.000000000 +0200
@@ -25,7 +25,7 @@
 /*                                                       */
 /*********************************************************/
 
-#define PAGE_SIZE HPDF_PAGE_SIZE_LETTER
+#define PAGE_SIZE HPDF_PAGE_SIZE_A4
 /* #define PAGE_SIZE HPDF_PAGE_SIZE_LEGAL */
 /* #define PAGE_SIZE HPDF_PAGE_SIZE_A3 */
 /* #define PAGE_SIZE HPDF_PAGE_SIZE_A4 */
