diff --git a/src/plugins/imageformats/mng/mng.pro b/src/plugins/imageformats/mng/mng.pro
index 17e364f..50006de 100644
--- a/src/plugins/imageformats/mng/mng.pro
+++ b/src/plugins/imageformats/mng/mng.pro
@@ -14,6 +14,5 @@ config_libmng {
 
 SOURCES += main.cpp
 
-DESTDIR = $$QT.gui.plugins/imageformats
 target.path += $$[QT_INSTALL_PLUGINS]/imageformats
 INSTALLS += target
diff --git a/src/plugins/imageformats/tga/tga.pro b/src/plugins/imageformats/tga/tga.pro
index 63b6f9c..d2b7308 100644
--- a/src/plugins/imageformats/tga/tga.pro
+++ b/src/plugins/imageformats/tga/tga.pro
@@ -9,7 +9,6 @@ OTHER_FILES += tga.json
 
 SOURCES += main.cpp
 
-DESTDIR = $$QT.gui.plugins/imageformats
 target.path += $$[QT_INSTALL_PLUGINS]/imageformats
 INSTALLS += target
 
diff --git a/src/plugins/imageformats/tiff/tiff.pro b/src/plugins/imageformats/tiff/tiff.pro
index 6d306b7..77f782e 100644
--- a/src/plugins/imageformats/tiff/tiff.pro
+++ b/src/plugins/imageformats/tiff/tiff.pro
@@ -12,6 +12,5 @@ config_libtiff {
     include($$PWD/../../../3rdparty/libtiff.pri)
 }
 
-DESTDIR = $$QT.gui.plugins/imageformats
 target.path += $$[QT_INSTALL_PLUGINS]/imageformats
 INSTALLS += target
diff --git a/src/plugins/imageformats/wbmp/wbmp.pro b/src/plugins/imageformats/wbmp/wbmp.pro
index 6ec358f..5a920c1 100644
--- a/src/plugins/imageformats/wbmp/wbmp.pro
+++ b/src/plugins/imageformats/wbmp/wbmp.pro
@@ -7,6 +7,5 @@ OTHER_FILES += wbmp.json
 
 SOURCES += main.cpp
 
-DESTDIR = $$QT.gui.plugins/imageformats
 target.path += $$[QT_INSTALL_PLUGINS]/imageformats
 INSTALLS += target
