Index: elementary/elementary.c_elementary_genlist.pxi
===================================================================
--- elementary/elementary.c_elementary_genlist.pxi	(revision 55980)
+++ elementary/elementary.c_elementary_genlist.pxi	(working copy)
@@ -126,7 +126,7 @@
     cdef readonly object _state_get_func
     cdef readonly object _del_func
 
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self._item_style = "default"
         self._label_get_func = None
         self._icon_get_func = None
Index: elementary/elementary.c_elementary_list.pxi
===================================================================
--- elementary/elementary.c_elementary_list.pxi	(revision 55980)
+++ elementary/elementary.c_elementary_list.pxi	(working copy)
@@ -49,7 +49,7 @@
     cdef Elm_List_Item *item
     cdef object cbt
 
-    def __new__(self):
+    def __cinit__(self):
         self.item = NULL
 
     def __del_cb(self):
Index: elementary/elementary.c_elementary_gengrid.pxi
===================================================================
--- elementary/elementary.c_elementary_gengrid.pxi	(revision 55980)
+++ elementary/elementary.c_elementary_gengrid.pxi	(working copy)
@@ -126,7 +126,7 @@
     cdef readonly object _state_get_func
     cdef readonly object _del_func
 
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self._item_style = "default"
         self._label_get_func = None
         self._icon_get_func = None
