--- config	2012-11-29 18:02:38.228035842 +0000
+++ config.new	2012-11-29 18:37:10.640235901 +0000
@@ -369,6 +369,9 @@
 #
 actionsfile match-all.action # Actions that are applied to all sites and maybe overruled later on.
 actionsfile default.action   # Main actions file
+actionsfile https_everywhere.action
+actionsfile easyprivacy.action
+#actionsfile easylist.action # probably quite buggy
 actionsfile user.action      # User customizations
 #
 #
