*** Makefile.orig	2003-01-28 15:51:32.000000000 +0100
--- Makefile	2009-02-12 10:48:48.000000000 +0100
***************
*** 1,2 ****
--- 1,3 ----
  CFLAGS := -W -Wall -O2 -D__UNIX__
+ LDFLAGS := -lm
  CC	:= gcc
***************
*** 43,45 ****
  vstrip :	 file_io.o in_buffer.o main.o parse_ifo.o vstrip.o
! 	$(CC)	-g  file_io.o in_buffer.o main.o parse_ifo.o vstrip.o -o vstrip
  file_io.o : s_types.h file_io.h file_io.c
--- 44,46 ----
  vstrip :	 file_io.o in_buffer.o main.o parse_ifo.o vstrip.o
! 	$(CC) $(CFLAGS) $(LDFLAGS) file_io.o in_buffer.o main.o parse_ifo.o vstrip.o -o $@
  file_io.o : s_types.h file_io.h file_io.c
