--- src/luainpython.c.orig	2011-07-26 12:00:11.039211810 +0600
+++ src/luainpython.c	2011-07-26 12:01:58.323596282 +0600
@@ -226,7 +226,12 @@
 	}
 	rc = py_convert(L, attr, 0);
 	if (rc) {
-		rc = py_convert(L, value, 0);
+		if (value != NULL) {
+			rc = py_convert(L, value, 0);
+		} else {
+			/* We're deleting an attribute. */
+			lua_pushnil(L);
+		}
 		if (rc) {
 			lua_settable(L, -3);
 			ret = 0;
