diff -aur rkhunter.orig/files/i18n/en rkhunter/files/i18n/en
--- rkhunter.orig/files/i18n/en	2012-01-29 19:30:07.000000000 +0100
+++ rkhunter/files/i18n/en	2012-11-01 09:07:33.000000000 +0100
@@ -502,6 +502,8 @@
 SYSTEM_CONFIGS_SSH_PROTO_DIFF2:Rkhunter configuration option 'ALLOW_SSH_PROT_V1': $1
 SYSTEM_CONFIGS_SSH_PROTO_NOTFOUND:The SSH configuration option 'Protocol' has not been set.
 SYSTEM_CONFIGS_SSH_PROTO_NOTFOUND:The default value may be '2,1', to allow the use of protocol version 1.
+SYSTEM_CONFIGS_SYSLOG_SYSTEMD_JOURNAL:Checking for running systemd journal daemon
+SYSTEM_CONFIGS_SYSLOG_SYSTEMD_JOURNAL_NO_FILE:The systemd journal daemon is running, but no configuration file can be found.
 SYSTEM_CONFIGS_SYSLOG:Checking for running syslog daemon
 SYSTEM_CONFIGS_SYSLOG_NOT_RUNNING:The syslog daemon is not running.
 SYSTEM_CONFIGS_SYSLOG_METALOG_RUNNING:The syslog daemon is not running, but a metalog daemon has been found.
diff -aur rkhunter.orig/files/rkhunter rkhunter/files/rkhunter
--- rkhunter.orig/files/rkhunter	2012-09-12 23:24:37.000000000 +0200
+++ rkhunter/files/rkhunter	2012-11-01 09:12:21.000000000 +0100
@@ -15065,12 +15065,22 @@
 
 	SYSLOG_SEEN=0
 	METALOG_SEEN=0; SOCKLOG_SEEN=0
+	SYSTEMD_JOURNAL_SEEN=0
 
 	if [ -n "${PS_CMD}" ]; then
 		PS_ARGS="ax"
 
 		test $SUNOS -eq 1 -o $IRIXOS -eq 1 && PS_ARGS="-ef"
 
+		RKHTMPVAR=`${PS_CMD} ${PS_ARGS} | egrep 'systemd-journald( |$)' | grep -v 'egrep'`
+		if [ -n "${RKHTMPVAR}" ]; then
+			SYSTEMD_JOURNAL_SEEN=1
+
+			display --to SCREEN+LOG --type PLAIN --result FOUND --color GREEN --log-indent 2 --screen-indent 4 SYSTEM_CONFIGS_SYSLOG_SYSTEMD_JOURNAL
+		else
+			display --to SCREEN+LOG --type PLAIN --result NOT_FOUND --color GREEN --log-indent 2 --screen-indent 4 SYSTEM_CONFIGS_SYSLOG_SYSTEMD_JOURNAL
+		fi
+
 		RKHTMPVAR=`${PS_CMD} ${PS_ARGS} | egrep '(syslogd|syslog-ng)( |$)' | grep -v 'egrep'`
 
 		if [ -n "${RKHTMPVAR}" ]; then
@@ -15094,8 +15104,12 @@
 					display --to SCREEN+LOG --type PLAIN --result NOT_FOUND --color GREEN --log-indent 2 --screen-indent 4 SYSTEM_CONFIGS_SYSLOG
 					display --to LOG --type INFO --log-indent 2 SYSTEM_CONFIGS_SYSLOG_SOCKLOG_RUNNING
 				else
-					display --to SCREEN+LOG --type PLAIN --result WARNING --color RED --log-indent 2 --screen-indent 4 SYSTEM_CONFIGS_SYSLOG
-					display --to LOG --type WARNING SYSTEM_CONFIGS_SYSLOG_NOT_RUNNING
+					if [ $SYSTEMD_JOURNAL_SEEN -eq 1 ]; then
+						display --to SCREEN+LOG --type PLAIN --result NOT_FOUND --color GREEN --log-indent 2 --screen-indent 4 SYSTEM_CONFIGS_SYSLOG
+					else
+						display --to SCREEN+LOG --type PLAIN --result WARNING --color RED --log-indent 2 --screen-indent 4 SYSTEM_CONFIGS_SYSLOG
+						display --to LOG --type WARNING SYSTEM_CONFIGS_SYSLOG_NOT_RUNNING
+					fi
 				fi
 			fi
 		fi
@@ -15104,6 +15118,18 @@
 		display --to LOG --type INFO NOT_FOUND_CMD "ps"
 	fi
 
+	#
+	# Check if the systemd journal configuration exists.
+	#
+
+	if [ $SYSTEMD_JOURNAL_SEEN -eq 1 ]; then
+		if [ -f "/etc/systemd/journald.conf" ]; then
+			display --to SCREEN+LOG --type PLAIN --result FOUND --color GREEN --log-indent 2 --screen-indent 4 SYSTEM_CONFIGS_FILE "systemd journal"
+		else
+			display --to SCREEN+LOG --type PLAIN --result WARNING --color RED --log-indent 2 --screen-indent 4 SYSTEM_CONFIGS_FILE "systemd journal"
+			display --to LOG --type WARNING SYSTEM_CONFIGS_SYSLOG_SYSTEMD_JOURNAL_NO_FILE
+		fi
+	fi
 
 	#
 	# Now loop through the syslog configuration files.
