diff -ur rkhunter-1.4.0.bu/files/rkhunter rkhunter-1.4.0/files/rkhunter
--- rkhunter-1.4.0.bu/files/rkhunter	2013-05-28 20:51:49.082767810 +0200
+++ rkhunter-1.4.0/files/rkhunter	2013-05-28 20:53:54.967183921 +0200
@@ -13953,7 +13953,7 @@
 		display --to LOG --type INFO STARTING_TEST promisc
 
 
-		if [ -n "${IFCONFIG_CMD}" ]; then
+		if [ -n "${IFCONFIG_CMD}" -o -n "${IP_CMD}" ]; then
 			PROMISCSCAN1=""; PROMISCSCAN2=""
 
 			IFWLIST=`get_option 2 multi ALLOWPROMISCIF` || exit 1
