--- src/xylib-0.9/configure.ac	2012-05-20 15:53:30.000000000 -0500
+++ nsrc/xylib-0.9/configure.ac	2012-07-24 14:33:43.714645883 -0500
@@ -50,7 +50,7 @@
                  [AC_MSG_ERROR([Could not find necessary C++ libs headers])])
 AC_CHECK_HEADERS([boost/cstdint.hpp boost/detail/endian.hpp], [],
                  [AC_MSG_ERROR([Could not find necessary Boost headers])])
-AC_CHECK_HEADERS([boost/spirit/core.hpp], [],
+AC_CHECK_HEADERS([boost/spirit/version.hpp], [],
                  [AC_MSG_ERROR([Boost::Spirit headers were not found.])])
 AC_CHECK_HEADERS([boost/tokenizer.hpp], [],
                  [AC_MSG_ERROR([Boost Tokenizer header not found.])])
--- src/xylib-0.9/configure	2012-07-24 14:46:47.339120052 -0500
+++ nsrc/xylib-0.9/configure	2012-07-24 14:47:11.522892858 -0500
@@ -15504,10 +15504,10 @@
 
 done
 
-for ac_header in boost/spirit/core.hpp
+for ac_header in boost/spirit/version.hpp
 do :
-  ac_fn_cxx_check_header_mongrel "$LINENO" "boost/spirit/core.hpp" "ac_cv_header_boost_spirit_core_hpp" "$ac_includes_default"
-if test "x$ac_cv_header_boost_spirit_core_hpp" = xyes; then :
+  ac_fn_cxx_check_header_mongrel "$LINENO" "boost/spirit/version.hpp" "ac_cv_header_boost_spirit_version_hpp" "$ac_includes_default"
+if test "x$ac_cv_header_boost_spirit_version_hpp" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_BOOST_SPIRIT_CORE_HPP 1
 _ACEOF
