diff -Naur musescore-git.orig/mscore/musescore.cpp musescore-git.new/mscore/musescore.cpp
--- musescore-git.orig/mscore/musescore.cpp	2011-10-29 12:49:32.000000000 +0200
+++ musescore-git.new/mscore/musescore.cpp	2011-10-29 12:51:07.000000000 +0200
@@ -1059,15 +1059,15 @@
       if (MScore::debugMode)
             qDebug("open handbook for language <%s>", qPrintable(lang));
 
-      QFileInfo mscoreHelp(mscoreGlobalShare + QString("man/MuseScore-") + lang + QString(".pdf"));
+      QFileInfo mscoreHelp(mscoreGlobalShare + QString("../doc/musescore-git/MuseScore-") + lang + QString(".pdf"));
       if (!mscoreHelp.isReadable()) {
             if (MScore::debugMode) {
                   qDebug("cannot open doc <%s>", qPrintable(mscoreHelp.filePath()));
                   }
             lang = lang.left(2);
-            mscoreHelp.setFile(mscoreGlobalShare + QString("man/MuseScore-") + lang + QString(".pdf"));
+            mscoreHelp.setFile(mscoreGlobalShare + QString("../doc/musescore-git/MuseScore-") + lang + QString(".pdf"));
             if(!mscoreHelp.isReadable()){
-                mscoreHelp.setFile(mscoreGlobalShare + QString("man/MuseScore-en.pdf"));
+                mscoreHelp.setFile(mscoreGlobalShare + QString("../doc/musescore-git/MuseScore-en.pdf"));
                 if (!mscoreHelp.isReadable()) {
                       QString info(tr("MuseScore handbook not found at: \n"));
                       info += mscoreHelp.filePath();
