diff -Naur musescore-git.orig//mscore/CMakeLists.txt musescore-git.new//mscore/CMakeLists.txt
--- musescore-git.orig//CMakeLists.txt	2010-06-16 23:10:50.000000000 +0200
+++ musescore-git.new//CMakeLists.txt	2010-06-16 23:13:58.000000000 +0200
@@ -116,7 +116,7 @@
       SET(Mscore_SHARE_NAME    "mscore.app/")
   endif(MINGW)
 else (MINGW OR APPLE)
-      SET(Mscore_INSTALL_NAME  "mscore-${Mscore_VERSION}/")
+      SET(Mscore_INSTALL_NAME  "musescore/")
       SET(Mscore_SHARE_NAME    "share/")
 endif (MINGW OR APPLE)
