diff -ur /tmp/old/bookreader-0.2/src/bookmarksdlg.h /tmp/new/bookreader-0.2/src/bookmarksdlg.h
--- /tmp/old/bookreader-0.2/src/bookmarksdlg.h 2009-10-26 14:53:59.366497355 +0300
+++ /tmp/new/bookreader-0.2/src/bookmarksdlg.h 2009-10-26 14:54:23.179827565 +0300
@@ -22,6 +22,7 @@

#include <qvariant.h>
#include <qdialog.h>
+#include <algorithm>
#include "bookmark.h"

class QVBoxLayout;
diff -ur /tmp/old/bookreader-0.2/src/renderer.h /tmp/new/bookreader-0.2/src/renderer.h
--- /tmp/old/bookreader-0.2/src/renderer.h 2009-10-26 14:46:51.099873685 +0300
+++ /tmp/new/bookreader-0.2/src/renderer.h 2009-10-26 14:52:27.611149272 +0300
@@ -27,6 +27,8 @@
#include <qsize.h>
#include <qtimer.h>
#include <vector>
+#include <memory>
+#include <algorithm>
class QStringList;

class TextLine
