--- main.c.0	2009-04-03 22:35:07.235260321 -0500
+++ main.c	2009-04-03 23:04:29.755652593 -0500
@@ -76,6 +76,7 @@
 	{"passwd-on-stdin", 0, 0, '5'},
 	{"no-passwd", 0, 0, '6'},
 	{"reconnect-timeout", 1, 0, '7'},
+  {"on-connect", 1, 0, '8'},
 	{NULL, 0, 0, 0},
 };
 
@@ -134,8 +135,26 @@
 		*c = 0;
 }
 
+void run_on_connect_app (const char *on_connect_app, const char *ifname, const char *vpn_addr)
+{
+  char cmd_text [512];
+  const int cmd_size = sizeof (cmd_text)-1;
+  if (NULL == on_connect_app)
+    return;
+
+  snprintf (cmd_text, cmd_size, "%s %s %s", on_connect_app, ifname, vpn_addr);
+  cmd_text [cmd_size] = '\0';
+
+  if (0 == fork ())
+  {
+    setsid ();
+    exit (system (cmd_text));
+  }
+}
+
 int main(int argc, char **argv)
 {
+  char *on_connect_app = NULL;
 	struct openconnect_info *vpninfo;
 	struct utsname utsbuf;
 	int cookieonly = 0;
@@ -205,6 +224,9 @@
 		case '7':
 			vpninfo->reconnect_timeout = atoi(optarg);
 			break;
+    case '8':
+      on_connect_app = optarg;
+      break;
 		case 'C':
 			vpninfo->cookie = optarg;
 			break;
@@ -353,6 +375,8 @@
 			  vpninfo->vpn_addr,
 			  (vpninfo->dtls_fd==-1)?(vpninfo->deflate?"SSL + deflate":"SSL"):"DTLS");
 
+  run_on_connect_app (on_connect_app, vpninfo->ifname, vpninfo->vpn_addr);
+
 	vpn_mainloop(vpninfo);
 	exit(1);
 }
