--- configure.orig	2012-11-09 23:14:38.635414585 +0100
+++ configure	2012-11-09 23:41:48.183518371 +0100
@@ -4606,7 +4606,7 @@
     fi
     if [ "$CFG_OPENGL" = "auto" ] || [ "$CFG_OPENGL" = "yes" ]; then
         if compileTest x11/opengl "OpenGL"; then
-            CFG_OPENGL=desktop
+            CFG_OPENGL=angle
         elif compileTest unix/opengles2 "OpenGL ES 2.x"; then
             CFG_OPENGL=es2
         else
@@ -5327,6 +5327,11 @@
     QT_CONFIG="$QT_CONFIG opengles2"
 fi
 
+if [ "$CFG_OPENGL" = "angle" ]; then
+    QCONFIG_FLAGS="$QCONFIG_FLAGS QT_OPENGL_ES QT_OPENGL_ES_2"
+    QT_CONFIG="$QT_CONFIG opengles2 egl angle"
+fi
+
 # build up the variables for output
 if [ "$CFG_DEBUG" = "yes" ]; then
     QMAKE_OUTDIR="${QMAKE_OUTDIR}debug"
@@ -6271,6 +6275,8 @@
     echo "OpenGL support ......... yes (Desktop OpenGL)"
 elif [ "$CFG_OPENGL" = "es2" ]; then
     echo "OpenGL support ......... yes (OpenGL ES 2.x)"
+elif [ "$CFG_OPENGL" = "angle" ]; then
+    echo "OpenGL support ......... yes (Angle)"
 else
     echo "OpenGL support ......... no"
 fi
--- src/plugins/platforms/windows/qwindowsglcontext.cpp.orig	2012-12-21 01:45:08.022424992 +0100
+++ src/plugins/platforms/windows/qwindowsglcontext.cpp	2012-12-21 01:52:35.773119490 +0100
@@ -50,7 +50,9 @@
 #include <qpa/qplatformnativeinterface.h>
 
 #include <wingdi.h>
+#ifndef QT_OPENGL_ES_2
 #include <GL/gl.h>
+#endif
 
 // #define DEBUG_GL
 
