diff -aur Gebabbel-0.4/src/MyMainWindow.cpp Gebabbel-0.4-patched/src/MyMainWindow.cpp
--- Gebabbel-0.4/src/MyMainWindow.cpp	2007-09-18 22:26:49.000000000 +0200
+++ Gebabbel-0.4-patched/src/MyMainWindow.cpp	2010-05-05 21:49:03.000000000 +0200
@@ -219,7 +219,7 @@
 
 		// Creating the toolbar
 		MainToolBar = addToolBar( tr( "Main Toolbar" ) );
-		MainToolBar->setIconSize( QSize::QSize( 16, 16 ) );
+		MainToolBar->setIconSize( QSize( 16, 16 ) );
 		MainToolBar->setAllowedAreas( Qt::TopToolBarArea );
 		MainToolBar->setMovable( false );
 
