From 557cea7201d1e9797b0542b7b91463f16a12719f Mon Sep 17 00:00:00 2001
From: Taegil Bae <esrevinu@gmail.com>
Date: Wed, 21 May 2014 19:37:02 +0900
Subject: [PATCH 7/7] fix compile error related to libevdev

---
 configure.ac    | 4 ++++
 src/Makefile.am | 4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 15b7700..e75d91f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -57,6 +57,10 @@ if test "x$HAVE_XI22" = xyes && test "x$HAVE_MTDEV" = xyes; then
         AC_DEFINE(MULTITOUCH, 1, [XI2.2 available])
 fi
 
+# Synaptics options
+AC_DEFINE(BUILD_EVENTCOMM, 1, [Optional backend eventcomm enabled])
+PKG_CHECK_MODULES(LIBEVDEV, [libevdev >= 0.4])
+
 # Define a configure option for an alternate input module directory
 AC_ARG_WITH(xorg-module-dir,
             AC_HELP_STRING([--with-xorg-module-dir=DIR],
diff --git a/src/Makefile.am b/src/Makefile.am
index c589de2..9d2379f 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -25,12 +25,12 @@
 # _ladir passes a dummy rpath to libtool so the thing will actually link
 # TODO: -nostdlib/-Bstatic/-lgcc platform magic, not installing the .a, etc.
 
-AM_CFLAGS = $(XORG_CFLAGS) $(CWARNFLAGS)
+AM_CFLAGS = $(XORG_CFLAGS) $(CWARNFLAGS) $(LIBEVDEV_CFLAGS)
 AM_CPPFLAGS =-I$(top_srcdir)/include
 
 @DRIVER_NAME@_drv_la_LTLIBRARIES = @DRIVER_NAME@_drv.la
 @DRIVER_NAME@_drv_la_LDFLAGS = -module -avoid-version
-@DRIVER_NAME@_drv_la_LIBADD = $(MTDEV_LIBS) $(UDEV_LIBS)
+@DRIVER_NAME@_drv_la_LIBADD = $(MTDEV_LIBS) $(UDEV_LIBS) $(LIBEVDEV_LIBS)
 @DRIVER_NAME@_drv_ladir = @inputdir@
 
 @DRIVER_NAME@_drv_la_SOURCES = @DRIVER_NAME@.c \
-- 
1.9.3

