From a53945741308182aa384bed5c35f20d7a608d700 Mon Sep 17 00:00:00 2001
From: Taegil Bae <esrevinu@gmail.com>
Date: Sat, 22 Feb 2014 22:08:17 +0900
Subject: [PATCH 10/10] add synatics' files into Makefile.am

---
 src/Makefile.am | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index da76540..c589de2 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -40,5 +40,12 @@ AM_CPPFLAGS =-I$(top_srcdir)/include
                                emuWheel.c \
                                draglock.c \
                                apple.c \
-                               axis_labels.h
+                               axis_labels.h \
+                               eventcomm.c \
+                               eventcomm.h \
+                               properties.c \
+                               synaptics.c \
+                               synapticsstr.h \
+                               synproto.c \
+                               synproto.h
 
-- 
1.9.0

