--- nitrotasks/NitrotasksWindow.py	2012-08-13 21:06:48.000000000 +0200
+++ nitrotasks/NitrotasksWindow.py	2012-08-25 07:03:41.268412469 +0200
@@ -18,7 +18,7 @@
 gettext.textdomain('nitrotasks')
 
 import webbrowser, os, sys, re, pickle, urllib, urllib2, threading
-from gi.repository import Gtk, WebKit, Unity, GObject # pylint: disable=E0611
+from gi.repository import Gtk, WebKit, GObject # pylint: disable=E0611
 import logging
 logger = logging.getLogger('nitrotasks')
 
@@ -86,8 +86,6 @@
         s = self.webview.get_settings()
         s.set_property("enable-universal-access-from-file-uris", True)
 
-        launcher = Unity.LauncherEntry.get_for_desktop_id ("nitrotasks.desktop")
-
         def external(this, widget, data = None):
             self.webview.execute_script('cmd("' + this.get_action_name() + '")')
 
@@ -153,10 +151,6 @@
                     else:
                         self.toolbar.hide()
 
-                elif title[0] == 'count':
-                    launcher.set_property("count", int(title[1]))
-                    launcher.set_property("count_visible", True)
-
                 elif title[0] == 'deleteOld':
                     os.remove(home + '/.nitrodata.pkl')
 
@@ -197,4 +191,4 @@
         self.mnu_contents.connect ("activate", menuexternal, None)
         self.mnu_donate.connect ("activate", menuexternal, None)
         self.mnu_donors.connect ("activate", menuexternal, None)
-        self.mnu_about.connect ("activate", menuexternal, None)
\ No newline at end of file
+        self.mnu_about.connect ("activate", menuexternal, None)
