diff -uNr mozc-1.1.773.102.orig//dictionary/gen_zip_code_seed.py mozc-1.1.773.102/dictionary/gen_zip_code_seed.py
--- mozc-1.1.773.102.orig//dictionary/gen_zip_code_seed.py	2011-07-19 14:42:18.000000000 +0900
+++ mozc-1.1.773.102/dictionary/gen_zip_code_seed.py	2011-07-25 04:41:16.000000000 +0900
@@ -78,7 +78,7 @@
     zip_code = self.FormatZip(self.zip_code)
     address = unicodedata.normalize('NFKC', self.address)
     line = '\t'.join([zip_code, '0', '0', '0', address,
-                      address, '*', ZIP_CODE_LABEL])
+                      ZIP_CODE_LABEL])
     print line.encode('utf-8')
 
 
