--- HotPotatoOnlineSources-v1.2.0.orig/src/glfont2.h	2006-02-03 22:29:18.000000000 +0800
+++ HotPotatoOnlineSources-v1.2.0/src/glfont2.h	2006-09-27 21:25:25.000000000 +0800
@@ -142,7 +142,7 @@
     float width;
 
     //Height is the same for now...might change in future
-    size->second = (int)(header.chars[header.start_char]->dy *
+    size->second = (int)(header.chars[header.start_char].dy *
       header.tex_height);
 
     //Calculate width of string
