--- HotPotatoOnlineSources-v1.2.0.orig/src/CMessageBox.cpp	2006-02-04 22:49:14.000000000 +0800
+++ HotPotatoOnlineSources-v1.2.0/src/CMessageBox.cpp	2006-09-27 21:25:25.000000000 +0800
@@ -36,7 +36,7 @@
 #endif
 
 #include "ComDefines.h"
-#include <GL/glpng.h>
+#include "include/GL/glpng.h"
 
 #include "CMessageBox.h"
 
--- HotPotatoOnlineSources-v1.2.0.orig/src/CPngTexture.cpp	2006-02-04 22:49:14.000000000 +0800
+++ HotPotatoOnlineSources-v1.2.0/src/CPngTexture.cpp	2006-09-27 21:26:09.000000000 +0800
@@ -33,7 +33,7 @@
 #else
 #include <GL/glut.h>
 #endif
-#include <GL/glpng.h>
+#include "include/GL/glpng.h"
 
 //#define WITH_DATA_CHANGE
 
--- HotPotatoOnlineSources-v1.2.0.orig/src/CStaticImage.cpp	2006-02-04 22:49:14.000000000 +0800
+++ HotPotatoOnlineSources-v1.2.0/src/CStaticImage.cpp	2006-09-27 21:27:00.000000000 +0800
@@ -34,7 +34,7 @@
 #else
 #include <GL/glut.h>
 #endif
-#include <GL/glpng.h>
+#include "include/GL/glpng.h"
 
 CStaticImage::CStaticImage(int PosX, int PosY, int Width, int Height, bool Visible)
 {
--- HotPotatoOnlineSources-v1.2.0.orig/src/CTextbox.cpp	2006-02-04 22:49:14.000000000 +0800
+++ HotPotatoOnlineSources-v1.2.0/src/CTextbox.cpp	2006-09-27 21:27:13.000000000 +0800
@@ -39,7 +39,7 @@
 #else
 #include <GL/glut.h>
 #endif
-#include <GL/glpng.h>
+#include "include/GL/glpng.h"
 
 //#define PLACEMENT_HACK
 
