--- src/CMakeLists.txt	2010-11-22 04:31:15.000000000 +0100
+++ src2/CMakeLists.txt	2012-05-20 04:24:20.981056947 +0200
@@ -123,7 +123,10 @@
     ${QT_QTDBUS_LIBRARY}
     ${QT_QTNETWORK_LIBRARY}
     ${CWIID_LIBRARY}
-  )
+rt
+dl
+pthread  
+)
 
   install(TARGETS wiimotedev-daemon DESTINATION ${WIIMOTEDEV_SBIN_DIR})
 endif(BUILD_DAEMON)
@@ -136,7 +139,10 @@
     ${QT_QTCORE_LIBRARY}
     ${QT_QTDBUS_LIBRARY}
     ${QT_QTNETWORK_LIBRARY}
-  )
+rt
+dl
+pthread  
+)
 
   install(TARGETS wiimotedev-client DESTINATION ${WIIMOTEDEV_SBIN_DIR})
 endif(BUILD_CLIENT)
@@ -148,6 +154,9 @@
   target_link_libraries(wiimotedev-uinput
     ${QT_QTCORE_LIBRARY}
     ${QT_QTDBUS_LIBRARY}
+rt
+dl
+pthread
   )
 
   install(TARGETS wiimotedev-uinput DESTINATION ${WIIMOTEDEV_SBIN_DIR})
diff -Naur src/devices/classicgamepad.cpp src2/devices/classicgamepad.cpp
--- src/devices/classicgamepad.cpp	2010-11-22 04:31:15.000000000 +0100
+++ src2/devices/classicgamepad.cpp	2012-05-20 04:24:20.994390228 +0200
@@ -18,7 +18,7 @@
  **********************************************************************************/
 
 #include "devices/classicgamepad.h"
-
+#include <unistd.h>
 ClassicGamepadDevice::ClassicGamepadDevice(QString deviceName) :
   UInputObject(),
   deviceName(deviceName)
diff -Naur src/devices/eventdevice.cpp src2/devices/eventdevice.cpp
--- src/devices/eventdevice.cpp	2010-11-22 04:31:15.000000000 +0100
+++ src2/devices/eventdevice.cpp	2012-05-20 04:24:20.994390228 +0200
@@ -18,7 +18,7 @@
  **********************************************************************************/
 
 #include "devices/eventdevice.h"
-
+#include <unistd.h>
 bool UInputEvent::uinput_open(bool replay){
   if (alreadyOpened) uinput_close();
   if (!(uinput_fd = open(uinputFile.toAscii().constData(), O_WRONLY | O_NDELAY))) {
diff -Naur src/devices/general.cpp src2/devices/general.cpp
--- src/devices/general.cpp	2010-11-22 04:31:15.000000000 +0100
+++ src2/devices/general.cpp	2012-05-20 04:24:20.994390228 +0200
@@ -18,7 +18,7 @@
  **********************************************************************************/
 
 #include "devices/general.h"
-
+#include <unistd.h>
 #include <QStringList>
 #include <QFile>
 
diff -Naur src/devices/mouse.cpp src2/devices/mouse.cpp
--- src/devices/mouse.cpp	2010-11-22 04:31:15.000000000 +0100
+++ src2/devices/mouse.cpp	2012-05-20 04:24:20.994390228 +0200
@@ -18,7 +18,7 @@
  **********************************************************************************/
 
 #include "devices/mouse.h"
-
+#include <unistd.h>
 bool UInputMouse::uinput_open(QRect absRect, bool abs){
   if (alreadyOpened) uinput_close();
   if (!(uinput_fd = open(uinputFile.toAscii().constData(), O_WRONLY | O_NDELAY))) {
diff -Naur src/devices/wiimotegamepad.cpp src2/devices/wiimotegamepad.cpp
--- src/devices/wiimotegamepad.cpp	2010-11-22 04:31:15.000000000 +0100
+++ src2/devices/wiimotegamepad.cpp	2012-05-20 04:24:20.994390228 +0200
@@ -18,7 +18,7 @@
  **********************************************************************************/
 
 #include "devices/wiimotegamepad.h"
-
+#include <unistd.h>
 WiimoteGamepadDevice::WiimoteGamepadDevice(QString deviceName) :
   UInputObject(),
   deviceName(deviceName)
--- src/config.h.in	2010-11-22 04:31:15.000000000 +0100
+++ src2/config.h.in	2012-05-20 04:24:20.981056947 +0200
@@ -21,7 +21,7 @@
 #define CONFIG_H_IN
 
 #define CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}"
-
+#include <unistd.h>
 #cmakedefine USE_SYSTEM_CWIID
 
 #define WIIMOTEDEV_VERSION_MAJOR ${WIIMOTEDEV_VERSION_MAJOR}
