--- a/plasma/contents/code/veromixcommon/LADSPAEffects.py	2012-09-15 18:45:59.634986771 +0200
+++ b/plasma/contents/code/veromixcommon/LADSPAEffects.py	2012-09-15 19:32:52.371188055 +0200
@@ -217,7 +217,7 @@
             name = line[0:-1]
             filename =  os.path.basename(name)
             try:
-                status,out = getstatusoutput("analyseplugin " + filename)
+                status,out = getstatusoutput("analyseplugin " + name)
                 if status != 0:
                     print("Veromix LADSPA: command 'analyseplugin' returend an error:")
                     print(out)
