diff -Naur astk-1.8.0.orig//ASTK_CLIENT/bin/astk astk-1.8.0/ASTK_CLIENT/bin/astk
--- astk-1.8.0.orig//ASTK_CLIENT/bin/astk	2010-06-02 00:09:07.027430361 +0200
+++ astk-1.8.0/ASTK_CLIENT/bin/astk	2010-06-02 15:10:50.019478367 +0200
@@ -17,6 +17,12 @@
 #    1 AVENUE DU GENERAL DE GAULLE, 92141 CLAMART CEDEX, FRANCE.
 # ==============================================================================
 
+# to solve buffering problem with gfortran
+export GFORTRAN_UNBUFFERED_ALL=y
+
+# to solve encoding problems with editor and xterm
+# export LANG=fr_FR.iso88591
+
 set_prefix() {
    local this=`readlink -n -f $1`
    local bin=`dirname $this`
diff -Naur astk-1.8.0.orig//ASTK_CLIENT/bin/bsf astk-1.8.0/ASTK_CLIENT/bin/bsf
--- astk-1.8.0.orig//ASTK_CLIENT/bin/bsf	2010-06-02 00:09:07.027430361 +0200
+++ astk-1.8.0/ASTK_CLIENT/bin/bsf	2010-06-02 00:09:28.311039150 +0200
@@ -32,6 +32,9 @@
 if [ -f $prefix/etc/codeaster/profile.sh ]; then
    . $prefix/etc/codeaster/profile.sh
 fi
+if [ -f ?ASTER_ROOT?/etc/codeaster/profile.sh ]; then
+   . ?ASTER_ROOT?/etc/codeaster/profile.sh
+fi
 
 # start tcl/tk interpreter
 if [ -z "$WISHEXECUTABLE" ]; then
diff -Naur astk-1.8.0.orig//ASTK_CLIENT/etc/astkrc/config_serveurs astk-1.8.0/ASTK_CLIENT/etc/astkrc/config_serveurs
--- astk-1.8.0.orig//ASTK_CLIENT/etc/astkrc/config_serveurs	2010-06-02 00:09:06.881787574 +0200
+++ astk-1.8.0/ASTK_CLIENT/etc/astkrc/config_serveurs	2010-06-02 15:16:44.898969850 +0200
@@ -10,4 +10,20 @@
 xterm : ?TERMINAL? -sb -si -geometry 90x32
 editeur : ?EDITOR?
 islocal : oui
+plate-forme : ?IFDEF?
+batch : non
+batch_tpsmax : 259200
+batch_memmax : 12000
+batch_nbpmax : 16
+batch_mpi_nbpmax : _VIDE
+batch_queue_group : _VIDE
+interactif : oui
+interactif_tpsmax : 356400
+interactif_memmax : 2048
+interactif_nbpmax : 16
+interactif_mpi_nbpmax : _VIDE
+vers : ?ASTER_VERSION?
+noeud : ?SERVER_NAME?
+mach_ref : non
+islocal : oui
 #
diff -Naur astk-1.8.0.orig//ASTK_CLIENT/etc/astkrc/outils astk-1.8.0/ASTK_CLIENT/etc/astkrc/outils
--- astk-1.8.0.orig//ASTK_CLIENT/etc/astkrc/outils	2010-06-02 00:09:06.881787574 +0200
+++ astk-1.8.0/ASTK_CLIENT/etc/astkrc/outils	2010-06-02 10:38:03.963975667 +0200
@@ -4,7 +4,12 @@
 astkrc_version : 1.1
 #
 nom : Eficas
-cmde : ?TOOLS_DIR?/eficas (@F)
+cmde : ?TOOLS_DIR?/eficasQt (@F)
+ext : comm com0 com1 com2
+dist : on
+#
+nom : EficasTk
+cmde : ?TOOLS_DIR?/eficasTk (@F)
 ext : comm com0 com1 com2
 dist : on
 #
@@ -23,3 +28,8 @@
 ext : dat
 dist : on
 #
+nom : Stanley
+cmde : ?TOOLS_DIR?/../../?vaster?/bibpyt/Stanley/stanley.py
+ext : _VIDE
+dist : off
+#
\ Manca newline alla fine del file
diff -Naur astk-1.8.0.orig//ASTK_CLIENT/etc/astkrc/prefs astk-1.8.0/ASTK_CLIENT/etc/astkrc/prefs
--- astk-1.8.0.orig//ASTK_CLIENT/etc/astkrc/prefs	2010-06-02 00:09:06.881787574 +0200
+++ astk-1.8.0/ASTK_CLIENT/etc/astkrc/prefs	2010-06-02 15:21:02.622277274 +0200
@@ -9,3 +9,5 @@
 freq_actu : 3
 nb_ligne : 20
 nom_domaine : ?DOMAIN_NAME?
+remote_copy_protocol : SCP
+remote_shell_protocol : SSH
diff -Naur astk-1.8.0.orig//ASTK_SERV/asrun/config.py astk-1.8.0/ASTK_SERV/asrun/config.py
--- astk-1.8.0.orig//ASTK_SERV/asrun/config.py	2010-06-02 00:09:07.107636186 +0200
+++ astk-1.8.0/ASTK_SERV/asrun/config.py	2010-06-02 00:09:28.385230557 +0200
@@ -68,8 +68,8 @@
          'SRCHIST'         : ['histor'],
          # name of "binaries" (as results of a make)
          'MAKE'            : ['debug nodebug'],
-         'BIN_NODBG'       : ['asteru.exe'],
-         'BIN_DBG'         : ['asterd.exe'],
+         'BIN_NODBG'       : ['asteru'],
+         'BIN_DBG'         : ['asterd'],
          'BINCMDE'         : ['commande'],
          'BINELE'          : ['elements'],
          'BINPICKLED'      : ['cata_ele.pickled'],
diff -Naur astk-1.8.0.orig//ASTK_SERV/bin/as_run astk-1.8.0/ASTK_SERV/bin/as_run
--- astk-1.8.0.orig//ASTK_SERV/bin/as_run	2010-06-02 00:09:07.107636186 +0200
+++ astk-1.8.0/ASTK_SERV/bin/as_run	2010-06-02 00:09:28.451577559 +0200
@@ -32,6 +32,9 @@
 if [ -f $prefix/etc/codeaster/profile.sh ]; then
    . $prefix/etc/codeaster/profile.sh
 fi
+if [ -f ?ASTER_ROOT?/etc/codeaster/profile.sh ]; then
+   . ?ASTER_ROOT?/etc/codeaster/profile.sh
+fi
 
 # start Python interpreter
 if [ -z "$PYTHONEXECUTABLE" ]; then
diff -Naur astk-1.8.0.orig//ASTK_SERV/bin/parallel_cp astk-1.8.0/ASTK_SERV/bin/parallel_cp
--- astk-1.8.0.orig//ASTK_SERV/bin/parallel_cp	2010-06-02 00:09:07.107636186 +0200
+++ astk-1.8.0/ASTK_SERV/bin/parallel_cp	2010-06-02 00:09:28.485395593 +0200
@@ -32,6 +32,9 @@
 if [ -f $prefix/etc/codeaster/profile.sh ]; then
    . $prefix/etc/codeaster/profile.sh
 fi
+if [ -f ?ASTER_ROOT?/etc/codeaster/profile.sh ]; then
+   . ?ASTER_ROOT?/etc/codeaster/profile.sh
+fi
 
 # start Python interpreter
 if [ -z "$PYTHONEXECUTABLE" ]; then
diff -Naur astk-1.8.0.orig//ASTK_SERV/etc/asrun astk-1.8.0/ASTK_SERV/etc/asrun
--- astk-1.8.0.orig//ASTK_SERV/etc/asrun	2010-06-02 00:09:07.027430361 +0200
+++ astk-1.8.0/ASTK_SERV/etc/asrun	2010-06-02 00:09:28.577043299 +0200
@@ -38,6 +38,10 @@
 # (used by mpi executions). For example : /export/tmp, /home/tmp...
 shared_tmp : /tmp
 
+# directories for tests
+rep_test : /tmp
+resu_test : /tmp
+
 # shared folders = for which remote copy is never necessary (comma separated)
 shared_folders :
 
@@ -108,12 +112,12 @@
 # batch_job : to display information about jobs
 # batch_kil : to delete jobs
 # --- Sun Grid Engine example
-batch_nom : SunGE
-batch_ini : /opt/SunGE6.0/default/common/settings.sh
-batch_sub : /opt/SunGE6.0/bin/lx24-x86/qsub
-batch_jid : JOB_ID
-batch_job : /opt/SunGE6.0/bin/lx24-x86/qstat
-batch_kil : /opt/SunGE6.0/bin/lx24-x86/qdel
+# batch_nom : SunGE
+# batch_ini : /opt/SunGE6.0/default/common/settings.sh
+# batch_sub : /opt/SunGE6.0/bin/lx24-x86/qsub
+# batch_jid : JOB_ID
+# batch_job : /opt/SunGE6.0/bin/lx24-x86/qstat
+# batch_kil : /opt/SunGE6.0/bin/lx24-x86/qdel
 
 # --- LSF example
 #batch_nom : LSF
@@ -136,16 +140,16 @@
 # mpirun
 #    available arguments are : mpi_hostfile, mpi_nbnoeud, mpi_nbcpu
 #    (use Python string formatting style)
-mpirun_cmd : ?MPIRUN? -np %(mpi_nbcpu)s --hostfile %(mpi_hostfile)s %(program)s
+mpirun_cmd : ?MPIRUN? -np %(mpi nbcpu)s %(program)s
 
 # file which contains list of hosts (REQUIRED even if it is not used in mpirun_cmd)
 mpi_hostfile : ?ASTER_CONFDIR?/aster-mpihosts
 
 # command called to initialize MPI environment (for example for mpich2)
-#mpi_ini : mpdboot --totalnum=%(mpi_nbnoeud)s --file=%(mpi_hostfile)s ; sleep 10
+mpi_ini : mpdboot --totalnum=%(mpi_nbnoeud)s --file=%(mpi_hostfile)s ; sleep 10
 
 # command called to close the MPI session (for example for mpich2)
-#mpi_end : mpdallexit
+mpi_end : mpdallexit
 
 # shell command to get processor id
 # LAM/MPI : echo $LAMRANK
diff -Naur astk-1.8.0.orig//ASTK_SERV/etc/aster astk-1.8.0/ASTK_SERV/etc/aster
--- astk-1.8.0.orig//ASTK_SERV/etc/aster	2010-06-02 00:09:07.027430361 +0200
+++ astk-1.8.0/ASTK_SERV/etc/aster	2010-06-02 00:09:28.610679954 +0200
@@ -6,5 +6,6 @@
 # available versions
 # DO NOT EDIT FOLLOWING LINE !
 #?vers : VVV?
-
+#vers : ?vaster?
+#vers : ?parvaster?
 
diff -Naur astk-1.8.0.orig//ASTK_SERV/etc/profile.csh astk-1.8.0/ASTK_SERV/etc/profile.csh
--- astk-1.8.0.orig//ASTK_SERV/etc/profile.csh	2010-06-02 00:09:07.027430361 +0200
+++ astk-1.8.0/ASTK_SERV/etc/profile.csh	2010-06-02 00:09:28.830698882 +0200
@@ -38,6 +38,11 @@
 # this should not be usefull...
 #setenv PYTHONHOME ?HOME_PYTHON?
 
+# Might be useful to fix errors on 64 bits platforms
+# See <http://www.code-aster.org/forum/read.php?f=11&i=1603&t=1574>
+setenv HOME_MED ?HOME_MED?
+setenv HOME_HDF ?HOME_HDF?
+
 setenv WISHEXECUTABLE ?WISH_EXE?
 
 
diff -Naur astk-1.8.0.orig//ASTK_SERV/etc/profile.sh astk-1.8.0/ASTK_SERV/etc/profile.sh
--- astk-1.8.0.orig//ASTK_SERV/etc/profile.sh	2010-06-02 00:09:07.027430361 +0200
+++ astk-1.8.0/ASTK_SERV/etc/profile.sh	2010-06-02 16:35:12.271834269 +0200
@@ -26,7 +26,7 @@
    export LD_LIBRARY_PATH=?HOME_PYTHON?/lib:$LD_LIBRARY_PATH
 fi
 
-if [ -z "$PATH" ]; then
+if [ -z "$PYTHONPATH" ]; then
    export PYTHONPATH=?ASRUN_SITE_PKG?
 else
    export PYTHONPATH=?ASRUN_SITE_PKG?:$PYTHONPATH
@@ -37,6 +37,13 @@
 # this should not be usefull...
 #export PYTHONHOME=?HOME_PYTHON?
 
+# Might be useful to fix errors on 64 bits platforms
+# See <http://www.code-aster.org/forum/read.php?f=11&i=1603&t=1574>
+HOME_MED=?HOME_MED?
+export HOME_MED
+HOME_HDF=?HOME_HDF?
+export HOME_HDF
+
 export WISHEXECUTABLE=?WISH_EXE?
 
 
