diff -crB ubuntuone-client-1.7.2/bin/u1sdtool ubuntuone-client-1.7.2-python2/bin/u1sdtool
*** ubuntuone-client-1.7.2/bin/u1sdtool	2011-08-09 01:32:50.000000000 +0400
--- ubuntuone-client-1.7.2-python2/bin/u1sdtool	2011-08-29 00:10:45.421676003 +0400
***************
*** 1,4 ****
! #! /usr/bin/env python
  
  # u1sdtool - command line utility for controlling ubuntuone-syncdaemon
  #
--- 1,4 ----
! #!/usr/bin/env python2
  
  # u1sdtool - command line utility for controlling ubuntuone-syncdaemon
  #
diff -crB ubuntuone-client-1.7.2/bin/ubuntuone-launch ubuntuone-client-1.7.2-python2/bin/ubuntuone-launch
*** ubuntuone-client-1.7.2/bin/ubuntuone-launch	2011-08-09 01:32:50.000000000 +0400
--- ubuntuone-client-1.7.2-python2/bin/ubuntuone-launch	2011-08-29 00:10:59.741648359 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  
  # ubuntuone-launch - Ubuntu One storage synchronization daemon startup helper
  #
--- 1,4 ----
! #!/usr/bin/env python2
  
  # ubuntuone-launch - Ubuntu One storage synchronization daemon startup helper
  #
diff -crB ubuntuone-client-1.7.2/bin/ubuntuone-login ubuntuone-client-1.7.2-python2/bin/ubuntuone-login
*** ubuntuone-client-1.7.2/bin/ubuntuone-login	2011-08-24 23:24:49.000000000 +0400
--- ubuntuone-client-1.7.2-python2/bin/ubuntuone-login	2011-08-29 00:11:11.238292835 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  # -*- coding: utf-8 -*-
  #
  # Author: Natalia B. Bidart <natalia.bidart@canonical.com>
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  #
  # Author: Natalia B. Bidart <natalia.bidart@canonical.com>
diff -crB ubuntuone-client-1.7.2/bin/ubuntuone-syncdaemon ubuntuone-client-1.7.2-python2/bin/ubuntuone-syncdaemon
*** ubuntuone-client-1.7.2/bin/ubuntuone-syncdaemon	2011-08-25 23:58:26.000000000 +0400
--- ubuntuone-client-1.7.2-python2/bin/ubuntuone-syncdaemon	2011-08-29 00:11:22.841603779 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  
  # ubuntuone-syncdaemon - Ubuntu One storage synchronization daemon
  #
--- 1,4 ----
! #!/usr/bin/env python2
  
  # ubuntuone-syncdaemon - Ubuntu One storage synchronization daemon
  #
diff -crB ubuntuone-client-1.7.2/contrib/dbus-docs ubuntuone-client-1.7.2-python2/contrib/dbus-docs
*** ubuntuone-client-1.7.2/contrib/dbus-docs	2011-08-09 01:32:50.000000000 +0400
--- ubuntuone-client-1.7.2-python2/contrib/dbus-docs	2011-08-29 00:12:28.251477636 +0400
***************
*** 1,4 ****
! #!/usr/bin/env python
  #
  # Author: Guillermo Gonzalez <guillermo.gonzalez@canonical.com>
  #
--- 1,4 ----
! #!/usr/bin/env python2
  #
  # Author: Guillermo Gonzalez <guillermo.gonzalez@canonical.com>
  #
diff -crB ubuntuone-client-1.7.2/contrib/dump_metadata.py ubuntuone-client-1.7.2-python2/contrib/dump_metadata.py
*** ubuntuone-client-1.7.2/contrib/dump_metadata.py	2011-08-09 01:32:50.000000000 +0400
--- ubuntuone-client-1.7.2-python2/contrib/dump_metadata.py	2011-08-29 00:12:58.804752093 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  # Copyright 2009 Canonical Ltd.  All rights reserved.
  
  """
--- 1,4 ----
! #!/usr/bin/env python2
  # Copyright 2009 Canonical Ltd.  All rights reserved.
  
  """
diff -crB ubuntuone-client-1.7.2/ubuntuone/platform/windows/pyinotify.py ubuntuone-client-1.7.2-python2/ubuntuone/platform/windows/pyinotify.py
*** ubuntuone-client-1.7.2/ubuntuone/platform/windows/pyinotify.py	2011-08-09 01:32:50.000000000 +0400
--- ubuntuone-client-1.7.2-python2/ubuntuone/platform/windows/pyinotify.py	2011-08-29 00:12:00.191531735 +0400
***************
*** 1,4 ****
! #!/usr/bin/env python
  
  # pyinotify.py - python interface to inotify
  # Copyright (c) 2010 Sebastien Martini <seb@dbzteam.org>
--- 1,4 ----
! #!/usr/bin/env python2
  
  # pyinotify.py - python interface to inotify
  # Copyright (c) 2010 Sebastien Martini <seb@dbzteam.org>
