--- ./src/facron.mk	2013-06-04 09:54:44.257515981 -0700
+++ ./src/facron.mk	2013-06-04 09:54:53.004182950 -0700
@@ -15,11 +15,11 @@
 # You should have received a copy of the GNU General Public License
 # along with facron.  If not, see <http://www.gnu.org/licenses/>.
 
-sbin_PROGRAMS += \
-	sbin/facron \
+bin_PROGRAMS += \
+	bin/facron \
 	$(NULL)
 
-sbin_facron_SOURCES = \
+bin_facron_SOURCES = \
 	src/facron/facron.c \
 	src/facron/facron-conf.h \
 	src/facron/facron-conf.c \
@@ -31,9 +31,9 @@
 	src/facron/facron-parser.c \
 	$(NULL)
 
-sbin_facron_CFLAGS = \
+bin_facron_CFLAGS = \
 	$(AM_CFLAGS) \
 	$(NULL)
 
-sbin_facron_LDADD = \
+bin_facron_LDADD = \
 	$(NULL)
