--- src/wizmainwindow.cpp	2014-05-13 11:25:18.889914639 +0800
+++ src/wizmainwindow.cpp	2014-05-11 20:19:45.089037493 +0800
@@ -1,15 +1,8 @@
 #include "wizmainwindow.h"
 
-#include <QToolBar>
-#include <QMenuBar>
-#include <QApplication>
-#include <QDesktopWidget>
-#include <QMessageBox>
+#include <QtWidgets>
 #include <QUndoStack>
 #include <QEvent>
-#include <QHBoxLayout>
-#include <QLabel>
-#include <QFileDialog>
 
 #ifdef Q_OS_MAC
 #include <Carbon/Carbon.h>
