diff -uNr www/curvecpkeys.dynhtml www2/curvecpkeys.dynhtml
--- www/curvecpkeys.dynhtml	2011-11-08 10:34:04.988311359 +0100
+++ www2/curvecpkeys.dynhtml	2011-11-08 13:33:40.141197434 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110913
 # Jan Mojzis
 # Public domain.
diff -uNr www/data.dynhtml www2/data.dynhtml
--- www/data.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/data.dynhtml	2011-11-08 13:33:40.141197434 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110927
 # Jan Mojzis
 # Public domain.
diff -uNr www/dnscurvekeys.dynhtml www2/dnscurvekeys.dynhtml
--- www/dnscurvekeys.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/dnscurvekeys.dynhtml	2011-11-08 13:33:40.141197434 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110913
 # Jan Mojzis
 # Public domain.
diff -uNr www/dns_dnscacheconfig.dynhtml www2/dns_dnscacheconfig.dynhtml
--- www/dns_dnscacheconfig.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/dns_dnscacheconfig.dynhtml	2011-11-08 13:33:40.141197434 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110927
 # Jan Mojzis
 # Public domain.
diff -uNr www/dns_dnslocalconfig.dynhtml www2/dns_dnslocalconfig.dynhtml
--- www/dns_dnslocalconfig.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/dns_dnslocalconfig.dynhtml	2011-11-08 13:33:40.141197434 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 201108015
 # Jan Mojzis
 # Public domain.
diff -uNr www/dns_dnsrootconfig.dynhtml www2/dns_dnsrootconfig.dynhtml
--- www/dns_dnsrootconfig.dynhtml	2011-11-08 10:34:04.988311359 +0100
+++ www2/dns_dnsrootconfig.dynhtml	2011-11-08 13:33:40.141197434 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 201109027
 # Jan Mojzis
 # Public domain.
diff -uNr www/dns.dynhtml www2/dns.dynhtml
--- www/dns.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/dns.dynhtml	2011-11-08 13:33:40.141197434 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110926
 # Jan Mojzis
 # Public domain.
diff -uNr www/dns_stats.dynhtml www2/dns_stats.dynhtml
--- www/dns_stats.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/dns_stats.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import sys, os, time, header, footer, lib, htmltemplate
 
diff -uNr www/forwarder_config.dynhtml www2/forwarder_config.dynhtml
--- www/forwarder_config.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/forwarder_config.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110927
 # Jan Mojzis
 # Public domain.
diff -uNr www/forwarder.dynhtml www2/forwarder.dynhtml
--- www/forwarder.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/forwarder.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110927
 # Jan Mojzis
 # Public domain.
diff -uNr www/httpproxy_stats.dynhtml www2/httpproxy_stats.dynhtml
--- www/httpproxy_stats.dynhtml	2011-11-08 10:34:04.988311359 +0100
+++ www2/httpproxy_stats.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import sys, os, time, header, lib, htmltemplate
 
diff -uNr www/index.dynhtml www2/index.dynhtml
--- www/index.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/index.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110802
 # Jan Mojzis
 # Public domain.
diff -uNr www/keys_config.dynhtml www2/keys_config.dynhtml
--- www/keys_config.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/keys_config.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110927
 # Jan Mojzis
 # Public domain.
diff -uNr www/log.dynhtml www2/log.dynhtml
--- www/log.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/log.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110927
 # Jan Mojzis
 # Public domain.
diff -uNr www/proxy_config.dynhtml www2/proxy_config.dynhtml
--- www/proxy_config.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/proxy_config.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110823
 # Jan Mojzis
 # Public domain.
diff -uNr www/proxy.dynhtml www2/proxy.dynhtml
--- www/proxy.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/proxy.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110823
 # Jan Mojzis
 # Public domain.
diff -uNr www/troubleshooting.dynhtml www2/troubleshooting.dynhtml
--- www/troubleshooting.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/troubleshooting.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110823
 # Jan Mojzis
 # Public domain.
diff -uNr www/vpn_config.dynhtml www2/vpn_config.dynhtml
--- www/vpn_config.dynhtml	2011-11-08 10:34:04.988311359 +0100
+++ www2/vpn_config.dynhtml	2011-11-08 13:33:40.144530770 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110926
 # Jan Mojzis
 # Public domain.
diff -uNr www/vpn.dynhtml www2/vpn.dynhtml
--- www/vpn.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/vpn.dynhtml	2011-11-08 13:33:40.147864105 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20110823
 # Jan Mojzis
 # Public domain.
diff -uNr www/vpn_aclconfig.dynhtml www2/vpn_aclconfig.dynhtml
--- www/vpn_aclconfig.dynhtml	2011-11-08 10:34:04.984978025 +0100
+++ www2/vpn_aclconfig.dynhtml	2011-11-08 13:33:40.147864105 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # 20111108
 # Jan Mojzis
 # Public domain.


