--- drivers/gpu/drm/i915/intel_opregion.c.orig	2013-09-03 00:46:10.000000000 +0400
+++ drivers/gpu/drm/i915/intel_opregion.c	2013-09-03 18:50:33.563873398 +0400
@@ -160,6 +160,7 @@
 #ifdef CONFIG_ACPI
 static u32 asle_set_backlight(struct drm_device *dev, u32 bclp)
 {
+	/*
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	struct opregion_asle __iomem *asle = dev_priv->opregion.asle;
 
@@ -174,6 +175,7 @@
 
 	intel_panel_set_backlight(dev, bclp, 255);
 	iowrite32((bclp*0x64)/0xff | ASLE_CBLV_VALID, &asle->cblv);
+	*/
 
 	return 0;
 }
