diff -ur zdoom.orig/src/gameconfigfile.cpp zdoom/src/gameconfigfile.cpp
--- zdoom.orig/src/gameconfigfile.cpp	2009-12-27 23:00:35.000000000 +0100
+++ zdoom/src/gameconfigfile.cpp	2009-12-27 23:12:57.000000000 +0100
@@ -138,6 +138,7 @@
 #else
 		SetValueForKey ("Path", "~/" GAME_DIR, true);
 		SetValueForKey ("Path", SHARE_DIR, true);
+		SetValueForKey ("Path", "/usr/share/doom", true);
 #endif
 	}
 
diff -ur zdoom.orig/src/sdl/i_system.h zdoom/src/sdl/i_system.h
--- zdoom.orig/src/sdl/i_system.h	2009-12-27 22:59:27.000000000 +0100
+++ zdoom/src/sdl/i_system.h	2009-12-27 23:11:42.000000000 +0100
@@ -32,7 +32,7 @@
 struct WadStuff;
 
 #ifndef SHARE_DIR
-#define SHARE_DIR "/usr/local/share/"
+#define SHARE_DIR "/usr/share/games/zdoom/"
 #endif
 
 // Index values into the LanguageIDs array
