--- jsonrpc/proxy.py.orig	2011-11-16 11:44:06.145643756 +0200
+++ jsonrpc/proxy.py	2011-11-16 11:45:27.198537988 +0200
@@ -41,7 +41,7 @@
          postdata = dumps({"method": self.__serviceName, 'params': args, 'id':'jsonrpc'})
          respdata = urllib.urlopen(self.__serviceURL, postdata).read()
          resp = loads(respdata)
-         if resp['error'] != None:
+         if 'error' in resp and resp['error'] != None:
              raise JSONRPCException(resp['error'])
          else:
              return resp['result']
