--- /home/aris/source/arch-packages/fuzzylite/src/original/qtfuzzylite/CMakeLists.txt
+++ /home/aris/source/arch-packages/fuzzylite/src/new/qtfuzzylite/CMakeLists.txt
@@ -50,13 +50,13 @@
 if(FL_DEBUG)
    set(CMAKE_BUILD_TYPE Debug)
    set(FL_CONF -dbg)
-   find_library(fl-shared-dbg fuzzylite-dbg 
-        HINTS "${CMAKE_SOURCE_DIR}/../fuzzylite/bin")
+   find_library(fl-shared-dbg fuzzylite-dbg
+   HINTS "${CMAKE_SOURCE_DIR}/../fuzzylite/build/lib")
 else(FL_DEBUG)
    set(CMAKE_BUILD_TYPE Release)
    set(FL_CONF)
-   find_library(fl-shared NAMES fuzzylite 
-        HINTS "${CMAKE_SOURCE_DIR}/../fuzzylite/bin")
+   find_library(fl-shared NAMES fuzzylite
+   HINTS "${CMAKE_SOURCE_DIR}/../fuzzylite/build/lib")
 endif(FL_DEBUG)
 
 
