diff -Naur fuzzylite-3.1.old/qtfuzzylite/CMakeLists.txt fuzzylite-3.1/qtfuzzylite/CMakeLists.txt
--- fuzzylite-3.1.old/qtfuzzylite/CMakeLists.txt	2013-12-03 20:25:49.913379218 -0200
+++ fuzzylite-3.1/qtfuzzylite/CMakeLists.txt	2013-12-03 20:26:08.190351513 -0200
@@ -116,9 +116,9 @@
     set(qtfl-sources ${qtfl-sources} ui/qtfuzzylite.rc)
 endif()
 
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 if(FL_USE_QT5)
-set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
 find_package(Qt5Widgets REQUIRED)
 #find_package(Qt5Core REQUIRED)
