diff -Naur smuxi-0.8.orig/lib/Makefile.am smuxi-0.8/lib/Makefile.am
--- smuxi-0.8.orig/lib/Makefile.am	2010-09-02 02:48:15.000000000 +0200
+++ smuxi-0.8/lib/Makefile.am	2011-12-13 07:28:00.851893336 +0100
@@ -3,7 +3,7 @@
 OUTPUT_DIR = $(top_builddir)/bin/$(PROFILE)
 OBJECT_DIR = $(OUTPUT_DIR)/obj
 SOURCE_PATTERNS = *.cs */*.cs */*/*.cs */*/*/*.cs
-XBUILD_FLAGS = /p:Configuration=Debug /p:WarningLevel=0 /p:OutputPath=$(abspath $(OUTPUT_DIR)) /p:BaseIntermediateOutputPath=$(abspath $(OBJECT_DIR))/ /p:DocumentationFile=
+XBUILD_FLAGS = /p:Configuration=Release /p:SignAssembly=false /p:WarningLevel=0 /p:OutputPath=$(abspath $(OUTPUT_DIR)) /p:BaseIntermediateOutputPath=$(abspath $(OBJECT_DIR))/ /p:DocumentationFile=
 
 JSON_SUBDIR = Newtonsoft.Json
 JSON_SRCDIR = $(srcdir)/$(JSON_SUBDIR)/Src/Newtonsoft.Json
diff -Naur smuxi-0.8.orig/lib/Makefile.in smuxi-0.8/lib/Makefile.in
--- smuxi-0.8.orig/lib/Makefile.in	2010-09-02 21:03:45.000000000 +0200
+++ smuxi-0.8/lib/Makefile.in	2011-12-13 07:28:00.858559613 +0100
@@ -285,7 +285,7 @@
 OUTPUT_DIR = $(top_builddir)/bin/$(PROFILE)
 OBJECT_DIR = $(OUTPUT_DIR)/obj
 SOURCE_PATTERNS = *.cs */*.cs */*/*.cs */*/*/*.cs
-XBUILD_FLAGS = /p:Configuration=Debug /p:WarningLevel=0 /p:OutputPath=$(abspath $(OUTPUT_DIR)) /p:BaseIntermediateOutputPath=$(abspath $(OBJECT_DIR))/ /p:DocumentationFile=
+XBUILD_FLAGS = /p:Configuration=Release /p:SignAssembly=false /p:WarningLevel=0 /p:OutputPath=$(abspath $(OUTPUT_DIR)) /p:BaseIntermediateOutputPath=$(abspath $(OBJECT_DIR))/ /p:DocumentationFile=
 JSON_SUBDIR = Newtonsoft.Json
 JSON_SRCDIR = $(srcdir)/$(JSON_SUBDIR)/Src/Newtonsoft.Json
 JSON_SOURCE_FILES = $(foreach pattern, $(SOURCE_PATTERNS), $(wildcard $(JSON_SRCDIR)/$(pattern)))
diff -Naur smuxi-0.8.orig/lib/Twitterizer/Twitterizer2/Methods/User/TwitterUserCollection.cs smuxi-0.8/lib/Twitterizer/Twitterizer2/Methods/User/TwitterUserCollection.cs
--- smuxi-0.8.orig/lib/Twitterizer/Twitterizer2/Methods/User/TwitterUserCollection.cs	2010-09-01 19:25:17.000000000 +0200
+++ smuxi-0.8/lib/Twitterizer/Twitterizer2/Methods/User/TwitterUserCollection.cs	2011-12-13 07:25:10.595739753 +0100
@@ -59,7 +59,7 @@
         /// Gets or sets information about the user's rate usage.
         /// </summary>
         /// <value>The rate limiting object.</value>
-        public new RateLimiting RateLimiting { get; internal set; }
+        public new RateLimiting RateLimiting { get; set; }
 
         /// <summary>
         /// Gets or sets the paged command.
