diff --git lib/qtsingleapplication/src/qtlocalpeer.cpp lib/qtsingleapplication/src/qtlocalpeer.cpp
index 382d182..506c142 100644
--- lib/qtsingleapplication/src/qtlocalpeer.cpp
+++ lib/qtsingleapplication/src/qtlocalpeer.cpp
@@ -50,6 +50,7 @@ static PProcessIdToSessionId pProcessIdToSessionId = 0;
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
