--- src/CslListCtrlPlayer.cpp	2010-09-12 18:47:22.000000000 +0200
+++ src/CslListCtrlPlayer.cpp	2010-07-20 15:52:32.000000000 +0200
@@ -110,9 +110,13 @@
         s+=info->GameMode+_(" on ");
     if (!info->Map.IsEmpty())
         s+=info->Map+wxT(" ");
-    if (info->TimeRemain>0)
-        s+=wxString::Format(wxT("(< %d %s)"),info->TimeRemain,info->TimeRemain==1 ?
-                            _("Minute"):_("Minutes"))+wxT(" ");
+
+	if (info->TimeRemain>0)
+		if (info->Protocol >= 258) 	
+			s+=wxString::Format(wxT("%s"), FormatSeconds(info->TimeRemain).c_str())+wxT(" ");
+		else
+			s+=wxString::Format(wxT("(< %d %s)"),info->TimeRemain,info->TimeRemain==1 ?
+								_("Minute"):_("Minutes"))+wxT(" ");
     s.Replace(wxT("&"),wxT("&&"));
 
     return s;
--- src/engine/CslExtendedInfo.h	2010-09-12 18:47:20.000000000 +0200
+++ src/engine/CslExtendedInfo.h	2010-07-20 09:53:16.000000000 +0200
@@ -26,7 +26,7 @@
 */
 
 #define CSL_EX_VERSION_MIN      102
-#define CSL_EX_VERSION_MAX      104
+#define CSL_EX_VERSION_MAX      105
 // commands
 #define CSL_EX_PING_UPTIME      0
 #define CSL_EX_PING_PLAYERSTATS 1
--- src/engine/CslGameSauerbraten.cpp	2010-09-12 18:47:20.000000000 +0200
+++ src/engine/CslGameSauerbraten.cpp	2010-07-20 15:42:38.000000000 +0200
@@ -53,7 +53,7 @@
 {
     static const wxChar* versions[] =
     {
-        wxT("Trooper"),wxT("CTF"),wxT("Assassin"),wxT("Summer"),
+        wxT("Justice"),wxT("Trooper"),wxT("CTF"),wxT("Assassin"),wxT("Summer"),
         wxT("Spring"),wxT("Gui"),wxT("Water"),wxT("Normalmap"),
         wxT("Sp"),wxT("Occlusion"),wxT("Shader"),wxT("Physics"),
         wxT("Mp"),wxT(""),wxT("Agc"),wxT("Quakecon"),wxT("Independence")
@@ -92,7 +92,7 @@
             wxT("ffa"),wxT("coop edit"),wxT("teamplay"),wxT("instagib"),wxT("instagib team"),
             wxT("efficiency"),wxT("efficiency team"),wxT("tactics"),wxT("tactics team"),
             wxT("capture"),wxT("regen capture"),wxT("ctf"),wxT("insta ctf"),
-            wxT("protect"),wxT("insta protect")
+            wxT("protect"),wxT("insta protect"),wxT("hold"),wxT("insta hold"),wxT("efficiency ctf"),wxT("efficiency protect"),wxT("efficiency hold")
         };
         return (mode>=0 && (size_t)mode<sizeof(modes)/sizeof(modes[0])) ?
                modes[mode] : T2C(_("unknown"));
@@ -188,6 +188,12 @@
     if (numattr>=3)
     {
         info.TimeRemain=attr[2];
+		if (info.Protocol >= 258) {
+			int time = info.TimeRemain;
+			info.TimeRemain = info.TimeRemain/60;
+			if (time > 0)
+					info.TimeRemain++;
+		}
         if (info.Protocol<254)
             info.TimeRemain++;
     }
--- src/engine/CslGameSauerbraten.h	2010-09-12 18:47:20.000000000 +0200
+++ src/engine/CslGameSauerbraten.h	2010-07-20 09:58:04.000000000 +0200
@@ -26,7 +26,7 @@
 */
 
 #define CSL_DEFAULT_NAME_SB           wxT("Sauerbraten")
-#define CSL_LAST_PROTOCOL_SB          257
+#define CSL_LAST_PROTOCOL_SB          258
 
 #define CSL_DEFAULT_PORT_SB           28785
 
