--- /common.h
+++ /common.h
@@ -818,11 +818,11 @@
     
     /* TODO: Evil casting... */
     char* find(char c) const
-     { return strchr(mArray.begin(), c); }
+     { return const_cast<char*>(strchr(mArray.begin(), c)); }
     char* findr(char c) const
-     { return strrchr(mArray.begin(), c); }
+     { return const_cast<char*>(strrchr(mArray.begin(), c)); }
     char* find(const char* s) const 
-     { return strstr(mArray.begin(), s); }
+     { return const_cast<char*>(strstr(mArray.begin(), s)); }
     
     String padLeft(char padChar, size_t len) const;
     String padCenter(char padChar, size_t len) const;
