diff -Naur epiphany-extensions-2.22.0orig/configure.ac epiphany-extensions-2.22.0/configure.ac
--- epiphany-extensions-2.22.0orig/configure.ac	2008-05-05 15:38:25.000000000 +0200
+++ epiphany-extensions-2.22.0/configure.ac	2008-05-05 15:41:13.000000000 +0200
@@ -120,7 +120,8 @@
 		  gmodule-2.0 \
 		  gtk+-2.0 >= $LIBGTK_REQUIRED \
                   gconf-2.0 \
-		  epiphany-$EPIPHANY_API_VERSION >= $EPIPHANY_REQUIRED])
+		  epiphany-$EPIPHANY_API_VERSION >= $EPIPHANY_REQUIRED] \
+		  gnome-vfs-2.0)
 AC_SUBST([EPIPHANY_DEPENDENCY_CFLAGS])
 AC_SUBST([EPIPHANY_DEPENDENCY_LIBS])
 
@@ -161,9 +162,9 @@
 
 AC_MSG_CHECKING([which extensions to build])
 
-ALL_EXTENSIONS="actions adblock auto-reload auto-scroller certificates error-viewer extensions-manager-ui gestures greasemonkey java-console livehttpheaders page-info permissions push-scroller rss sample sample-mozilla select-stylesheet sidebar smart-bookmarks tab-groups tab-states"
+ALL_EXTENSIONS="actions adblock auto-reload auto-scroller certificates error-viewer extensions-manager-ui gestures greasemonkey java-console livehttpheaders page-info permissions push-scroller rss sample sample-mozilla select-stylesheet sidebar smart-bookmarks tab-groups tab-states sessionsaver"
 USEFUL_EXTENSIONS="actions adblock auto-reload auto-scroller certificates error-viewer extensions-manager-ui gestures java-console page-info push-scroller select-stylesheet sidebar smart-bookmarks tab-groups tab-states"
-DEFAULT_EXTENSIONS="actions adblock auto-scroller certificates error-viewer extensions-manager-ui gestures java-console page-info push-scroller select-stylesheet sidebar smart-bookmarks tab-groups tab-states"
+DEFAULT_EXTENSIONS="actions adblock auto-scroller certificates error-viewer extensions-manager-ui gestures java-console page-info push-scroller select-stylesheet sidebar smart-bookmarks tab-groups tab-states sessionsaver"
 
 PYTHON_ALL_EXTENSIONS="python-console sample-python favicon cc-license-viewer epilicious"
 PYTHON_USEFUL_EXTENSIONS="python-console favicon cc-license-viewer"
@@ -411,6 +412,7 @@
 extensions/tab-groups/Makefile
 extensions/java-console/Makefile
 extensions/java-console/mozilla/Makefile
+extensions/sessionsaver/Makefile
 help/Makefile
 po/Makefile.in
 ])
