diff -aur jq-1.1/gen_utf8_tables.py jq-1.1.new/gen_utf8_tables.py
--- jq-1.1/gen_utf8_tables.py	2012-10-22 15:30:25.404658245 +0100
+++ jq-1.1.new/gen_utf8_tables.py	2012-10-22 16:15:32.005382363 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
 
 mask = lambda n: (1 << n) - 1
 
diff -aur jq-1.1/Makefile jq-1.1.new/Makefile
--- jq-1.1/Makefile	2012-10-22 16:13:20.879178256 +0100
+++ jq-1.1.new/Makefile	2012-10-22 16:15:56.257395972 +0100
@@ -19,7 +19,7 @@
 parser.gen.h: parser.gen.c
 
 jv_utf8_tables.gen.h: gen_utf8_tables.py
-	python $^ > $@
+	python2 $^ > $@
 jv_unicode.c: jv_utf8_tables.gen.h
 
 version.gen.h: VERSION
@@ -51,8 +51,8 @@
 tarball: docs/content/2.download/source/jq.tgz
 
 install: jq
-	install -d -m 0755 $(prefix)/bin
-	install -m 0755 jq $(prefix)/bin
+	install -d -m 0755 $(DESTDIR)$(prefix)/bin
+	install -m 0755 jq $(DESTDIR)$(prefix)/bin
 
 uninstall:
 	test -d $(prefix)/bin && \
