diff -Naur tagplop~/Makefile tagplop/Makefile
--- tagplop~/Makefile	2009-01-18 13:53:13.000000000 -0600
+++ tagplop/Makefile	2009-01-18 13:58:44.000000000 -0600
@@ -1,10 +1,10 @@
-
+DESTDIR=
 
 CC=gcc
-INSTDIR=/usr/share/tagplop
+INSTDIR=/usr
 # -DFILENAME=\"$(INSTDIR)/tagplop.glade\"
-CFLAGS=-g `pkg-config --cflags gtk+-2.0 libglade-2.0 taglib` `xml2-config --cflags` -DFILENAME=\"$(INSTDIR)/tagplop.glade\"
-LIBS=`pkg-config --libs gtk+-2.0 libglade-2.0` `xml2-config --libs` -ltag_c -ltunepimp -lcurl
+CFLAGS+=-g `pkg-config --cflags gtk+-2.0 libglade-2.0 taglib` `xml2-config --cflags` -DFILENAME=\"$(INSTDIR)/share/tagplop/tagplop.glade\"
+LIBS+=`pkg-config --libs gtk+-2.0 libglade-2.0` `xml2-config --libs` -ltag_c -ltunepimp -lcurl
 
 TARGET=tagplop
 OBJECTS=main.o functions.o network.o xml.o gtkchosentrack.o
@@ -37,10 +37,13 @@
 	rm -f *.o *~ $(TARGET) *.bak
 
 install:
-	mkdir -p $(INSTDIR) && \
-	cp tagplop.glade $(INSTDIR)/. && \
-	cp tagplop /usr/bin/.
+	install -D -m 755 $(TARGET) $(DESTDIR)$(INSTDIR)/bin/$(TARGET)
+	install -D -m 644 $(TARGET).glade $(DESTDIR)$(INSTDIR)/share/$(TARGET)/$(TARGET).glade
+	install -D -m 644 $(TARGET).xpm $(DESTDIR)$(INSTDIR)/share/pixmaps/$(TARGET).xpm
+	install -D -m 644 $(TARGET).desktop $(DESTDIR)$(INSTDIR)/share/applications/$(TARGET).desktop
 
 uninstall:
-	rm /usr/bin/tagplop && rm -rf $(INSTDIR)
-
+	rm -rf $(DESTDIR)$(INSTDIR)/share/$(TARGET)
+	rm -f $(DESTDIR)$(INSTDIR)/bin/$(TARGET)
+	rm -f $(DESTDIR)$(INSTDIR)/share/pixmaps/$(TARGET).xpm
+	rm -f $(DESTDIR)$(INSTDIR)/share/applications/$(TARGET).desktop
diff -Naur tagplop~/tagplop.desktop tagplop/tagplop.desktop
--- tagplop~/tagplop.desktop	1969-12-31 18:00:00.000000000 -0600
+++ tagplop/tagplop.desktop	2009-01-18 13:53:18.000000000 -0600
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=Tagplop
+Comment=A simple music file tagger
+TryExec=/usr/bin/tagplop
+Exec=tagplop
+Icon=/usr/share/pixmaps/tagplop.xpm
+Categories=Audio;AudioVideoEditing;
diff -Naur tagplop~/tagplop.xpm tagplop/tagplop.xpm
--- tagplop~/tagplop.xpm	1969-12-31 18:00:00.000000000 -0600
+++ tagplop/tagplop.xpm	2009-01-18 13:53:18.000000000 -0600
@@ -0,0 +1,73 @@
+/* XPM */
+static const char *tagplop_xpm[] = {
+"48 48 22 1",
+" 	c None",
+".	c #8D2E2E",
+"+	c #8C2E2E",
+"@	c #903030",
+"#	c #933030",
+"$	c #923131",
+"%	c #892D2D",
+"&	c #882D2D",
+"*	c #943131",
+"=	c #923030",
+"-	c #8D2F2F",
+";	c #892E2E",
+">	c #8F2F2F",
+",	c #933131",
+"'	c #913030",
+")	c #943030",
+"!	c #8A2D2D",
+"~	c #8E2F2F",
+"{	c #872D2D",
+"]	c #8C2F2F",
+"^	c #8B2E2E",
+"/	c #902F2F",
+"                                                ",
+"                                                ",
+"                                                ",
+"                                                ",
+"                                                ",
+"                                                ",
+"                                                ",
+"                                .+              ",
+"                                @@              ",
+"                                ##              ",
+"                                $#%             ",
+"                                &*=             ",
+"                               - $*;            ",
+"                               >  ,,            ",
+"                               '  -*=           ",
+"                               >   ')!          ",
+"                              >.    ==          ",
+"                              ,      #~         ",
+"                              *      {*         ",
+"                             +#       ''        ",
+"                             '#        #        ",
+"                             #=        '        ",
+"                             #@        ]        ",
+"                            ^*+        ^        ",
+"                            >*         ~        ",
+"                            =*         /        ",
+"                            *,                  ",
+"                           %*'                  ",
+"                           /*/                  ",
+"                           **^                  ",
+"                           **                   ",
+"                          ~**                   ",
+"                &+]%      =),                   ",
+"             @********=   =*=                   ",
+"           @**)******)**@!#*@                   ",
+"          ,*),*********#*##*~                   ",
+"         #*,****************^                   ",
+"         *#****************#%                   ",
+"        +*****************)#                    ",
+"         *)**************#*=                    ",
+"         @*#************#*#                     ",
+"          '*#**********,*#                      ",
+"           -##********)#@                       ",
+"             %~'=,,='>^                         ",
+"                                                ",
+"                                                ",
+"                                                ",
+"                                                "};
