diff -aur stereograph-0.33b.orig/gfxio.c stereograph-0.33b/gfxio.c
--- stereograph-0.33b.orig/gfxio.c	2003-11-17 15:32:56.000000000 -0600
+++ stereograph-0.33b/gfxio.c	2012-05-22 22:45:17.000000000 -0500
@@ -648,7 +648,7 @@
         ptr += 1;
     else
         ptr = file_name;
-    sprintf(cmd, "gcc -c -I. -I./include -I%s %s -o /tmp/stereo-%s.o ; cd /tmp ; "
+    sprintf(cmd, "gcc -fPIC -c -I. -I./include -I%s %s -o /tmp/stereo-%s.o ; cd /tmp ; "
                  "gcc -shared -Wl,-soname,stereo-%s.so stereo-%s.o -o stereo-%s.so\n",
 	         ((stereograph_include_dir)?
 		  stereograph_include_dir:"."),
@@ -872,12 +872,14 @@
 		}
 		
 		/* libpng error handling */
+      #if 0 /* error: dereferencing pointer to incomplete type */
 		if (setjmp(png_ptr->jmpbuf)) {
 			png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);
         		if(stereograph_verbose) printf("FAILED;\n"); else fprintf(stderr, "reading gfx data...FAILED\n");
 	        	fprintf(stderr, "libpng reported an error!\n");
 			return GFX_ERROR_LIBPNG;
 		}
+      #endif
 		
 
 		/* setting up file pointer */
@@ -1050,12 +1052,14 @@
 	}
 		
 	/* libpng error handling */
+   #if 0 /* error: dereferencing pointer to incomplete type */
 	if (setjmp(png_ptr->jmpbuf)) {
 		png_destroy_write_struct(&png_ptr, &info_ptr);
        		if(stereograph_verbose) printf("FAILED;\n"); else fprintf(stderr, "writing gfx data...FAILED\n");
         	fprintf(stderr, "libpng reported an error!\n");
 		return GFX_ERROR_LIBPNG;
 	}
+   #endif
 		
 	/* setting up file pointer */
 	png_init_io(png_ptr, ofile);
diff -aur stereograph-0.33b.orig/Makefile stereograph-0.33b/Makefile
--- stereograph-0.33b.orig/Makefile	2003-11-17 15:34:44.000000000 -0600
+++ stereograph-0.33b/Makefile	2012-05-22 22:44:25.000000000 -0500
@@ -12,7 +12,8 @@
 #
 
 #PREFIX=/usr/local
-PREFIX=`grep -i "prefix =" ../Makefile | grep -i -v exec | cut -d"=" -f 2`
+#PREFIX=`grep -i "prefix =" ../Makefile | grep -i -v exec | cut -d"=" -f 2`
+PREFIX=/usr
 
 # Linux
 CC  = gcc
diff -aur stereograph-0.33b.orig/Makefile.in stereograph-0.33b/Makefile.in
--- stereograph-0.33b.orig/Makefile.in	2003-11-09 09:33:17.000000000 -0600
+++ stereograph-0.33b/Makefile.in	2012-05-22 22:44:25.000000000 -0500
@@ -14,8 +14,8 @@
 
 
 install:
-	cp stereograph $(PREFIX)/bin/stereograph
-	cp stereograph.1 $(PREFIX)/man/man1/stereograph.1
+	cp stereograph $(DESTDIR)$(PREFIX)/bin/stereograph
+	cp stereograph.1 $(DESTDIR)$(PREFIX)/share/man/man1/stereograph.1
 
 clean:
 	rm -f $(OBJ0) *~ stereograph core
@@ -23,5 +23,5 @@
 distclean: clean	
 
 un-install:
-	rm $(PREFIX)/bin/stereograph
-	rm $(PREFIX)/man/man1/stereograph.1
+	rm $(DESTDIR)$(PREFIX)/bin/stereograph
+	rm $(DESTDIR)$(PREFIX)/man/man1/stereograph.1
