From e4212be1773ff44ed5cf1b339baad96b78e6d0a2 Mon Sep 17 00:00:00 2001
From: Jon Nordby <jononor@gmail.com>
Date: Fri, 3 Dec 2010 10:59:47 +0100
Subject: [PATCH] Build: Accept libtool 2.4 as well.

---
 autogen.sh |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index 4711ef3..6c3aa86 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -22,6 +22,9 @@ if libtoolize --version < /dev/null > /dev/null 2>&1 ; then
         2.2*)
         have_libtool=true
         ;;
+        2.4*)
+        have_libtool=true
+        ;;
     esac
 fi
 if $have_libtool ; then : ; else
-- 
1.7.3.2

