--- factory.orig/GNUmakefile.in	2010-05-07 19:26:37.000000000 +0200
+++ factory/GNUmakefile.in	2010-06-20 17:54:06.509063447 +0200
@@ -430,15 +430,15 @@
 #
 
 # how to create library objets
-%.o: %.cc config.h
+%.o: %.cc factoryconf.h config.h
 	$(CXX) -c $< $(LIBCXXFLAGS) -o $@
-%.o: %.c config.h
+%.o: %.c factoryconf.h config.h
 	$(CC) -c $< $(LIBCFLAGS) -o $@
 # the debug version:
-%.og: %.cc config.h
+%.og: %.cc factoryconf.h config.h
 	$(CXX) -g -c $< $(LIBCGGFLAGS) -o $@
 # the profiling version:
-%.op: %.cc config.h
+%.op: %.cc factoryconf.h config.h
 	$(CXX) -O -pg -c $< $(LIBCGGFLAGS) -o $@
 
 # how to create dependency files.  To avoid broken dependency files
--- libfac.orig/Makefile.in	2010-01-29 10:10:49.000000000 +0100
+++ libfac/Makefile.in	2010-06-20 17:56:37.511131412 +0200
@@ -86,6 +86,11 @@
 OPTOBJ:
 	${MKINSTALLDIRS} OPTOBJ
 
+${OPTFACTOROBJ}: OPTOBJ
+${OPTCHARSETOBJ}: OPTOBJ
+${DEBOBJ}: OPTOBJ
+${PROFOBJ}: OPTOBJ
+
 OPTOBJ/%.o: %.cc 
 	$(CXX) $(CXXFLAGS) $(CXXTEMPLFLAGS) $(CPPFLAGS) $(DEFS) -c $< -o $@
 
