--- src/hyperdex-0.2b6/configure~	2012-02-17 14:48:06.000000000 +0100
+++ src/hyperdex-0.2b6/configure	2012-02-25 12:20:23.967519548 +0100
@@ -17257,7 +17257,7 @@
 $as_echo_n "checking for a version of Python >= 2.6... " >&6; }
 		ac_supports_python_ver=`$PYTHON -c "import sys; \
 			ver = sys.version.split ()[0]; \
-			print (ver >= 2.6)"`
+			print (ver >= \"2.6\")"`
 		if test "$ac_supports_python_ver" = "True"; then
 	   	   { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
