diff -Naur musescore-svn.orig/mscore/CMakeLists.txt musescore-svn.new/mscore/CMakeLists.txt
--- musescore-svn.orig/mscore/CMakeLists.txt	2011-10-29 12:50:28.000000000 +0200
+++ musescore-svn.new/mscore/CMakeLists.txt	2011-11-04 18:56:52.000000000 +0100
@@ -34,7 +34,7 @@
 set(HAS_AUDIOFILE TRUE)             # requires libsndfile
 
 set(STATIC_SCRIPT_BINDINGS TRUE)
-set(BUILD_SCRIPTGEN   TRUE)         # Generate Qt script bindings. (collides with qtscriptgenerator project on several distrib)
+set(BUILD_SCRIPTGEN FALSE)         # Generate Qt script bindings. (collides with qtscriptgenerator project on several distrib)
 set(USE_SSE           FALSE)
 set(SOUNDFONT3        TRUE)         # enable ogg vorbis compressed fonts, require ogg & vorbis
 set(AEOLUS            TRUE)         # pipe organ synthesizer
