--- scripts/Dpkg/Arch.pm.old	2012-04-05 16:27:02.510959023 +0200
+++ scripts/Dpkg/Arch.pm	2012-04-05 16:27:33.480959203 +0200
@@ -59,8 +59,8 @@
 	# dpkg-architecture itself, by avoiding computing the DEB_BUILD_
 	# variables when they are not requested.
 
-	my $build_arch = `dpkg --print-architecture`;
-	syserr("dpkg --print-architecture failed") if $? >> 8;
+	my $build_arch = `uname -m`;
+	syserr("uname -m failed") if $? >> 8;
 
 	chomp $build_arch;
 	return $build_arch;
