Index: src/core/org/apache/hadoop/fs/kfs/KFSOutputStream.java
===================================================================
--- src/core/org/apache/hadoop/fs/kfs/KFSOutputStream.java	(revision 764287)
+++ src/core/org/apache/hadoop/fs/kfs/KFSOutputStream.java	(working copy)
@@ -73,7 +73,9 @@
         }
 
         // touch the progress before going into KFS since the call can block
-        progressReporter.progress();
+        if (progressReporter != null) {
+            progressReporter.progress();
+        }
         kfsChannel.write(ByteBuffer.wrap(b, off, len));
     }
 
@@ -82,7 +84,9 @@
             throw new IOException("File closed");
         }
         // touch the progress before going into KFS since the call can block
-        progressReporter.progress();
+        if (progressReporter != null) {
+            progressReporter.progress();
+        }
         kfsChannel.sync();
     }
 
