+++ configure	2011-06-22 04:39:56.189168395 +0200
--- configure.orig	2011-06-22 04:39:14.805423984 +0200
@@ -162,7 +162,7 @@
     if [ "`uname -s`" = "SunOS" ] ; then
         default_install=/usr/ucb/install
     else
+        default_install=/bin/install
-        default_install=/usr/bin/install
     fi
     finalize install "$default_install"
     echo "$install"
@@ -279,7 +279,7 @@
 #######################################################################
 
 # Workaround for Solaris "Bad string" issue when LOCALE is undefined
+tr="/bin/tr"
-tr="/usr/bin/tr"
 test -f /usr/xpg4/bin/tr && tr="/usr/xpg4/bin/tr"
 
 . ./config.inc
