diff --git a/config.def.h b/config.def.h
index a221c86..c3aef5e 100644
--- a/config.def.h
+++ b/config.def.h
@@ -103,3 +103,4 @@ static Key keys[] = {
     { MODKEY|GDK_SHIFT_MASK,GDK_g,      togglegeolocation, { 0 } },
 };
 
+static char *searchengine   = "https://duckduckgo.com/?q=";
diff --git a/surf.c b/surf.c
index cebd469..a5c0030 100644
--- a/surf.c
+++ b/surf.c
@@ -626,6 +626,8 @@ loaduri(Client *c, const Arg *arg) {
 		rp = realpath(uri, NULL);
 		u = g_strdup_printf("file://%s", rp);
 		free(rp);
+	} else if (*uri == ' ') {
+		u = g_strdup_printf("%s%s", searchengine, uri+1);
 	} else {
 		u = g_strrstr(uri, "://") ? g_strdup(uri)
 			: g_strdup_printf("http://%s", uri);
