--- Makefile	2005-08-24 08:17:13.000000000 +0200
+++ Makefile.ok	2007-05-17 16:05:14.000000000 +0200
@@ -58,7 +58,7 @@
 # low overhead version (internal parser)
 lo_xcalib: xcalib.c
 	$(CC) $(CFLAGS) -c xcalib.c -I$(XINCLUDEDIR) -DXCALIB_VERSION=\"$(XCALIB_VERSION)\"
-	$(CC) $(CFLAGS) -L$(XLIBDIR) -lm -o xcalib xcalib.o -lX11 -lXxf86vm -lXext
+	$(CC) $(LDFLAGS) -L$(XLIBDIR) -o xcalib xcalib.o -lm -lX11 -lXxf86vm -lXext
 
 fglrx_xcalib: xcalib.c
 	$(CC) $(CFLAGS) -c xcalib.c -I$(XINCLUDEDIR) -DXCALIB_VERSION=\"$(XCALIB_VERSION)\" -I$(FGLRXINCLUDEDIR) -DFGLRX
