--- vapor-2.1.0.orig/options.mk	2011-07-28 23:22:45.000000000 +0600
+++ vapor-2.1.0/options.mk	2012-05-12 22:47:46.055249905 +0600
@@ -12,10 +12,10 @@
 # in $(INSTALL_PREFIX_DIR)/bin, libraries in $(INSTALL_PREFIX_DIR)/lib
 # etc,.
 #
-INSTALL_PREFIX_DIR=/usr/local
+INSTALL_PREFIX_DIR=/usr
 
 # Set RELEASE to 1 to compile with optimizations and without debug info.
-RELEASE=0
+RELEASE=1
 
 # Set DEBUG to 1 if you want diagnostic messages turned on
 DEBUG=0
@@ -37,14 +37,14 @@
 # linker normally checks. The linker will search the directories
 # in the order specified.
 #
-LIB_SEARCH_DIRS = 
+LIB_SEARCH_DIRS = /usr/lib
 
 # Set INC_SEARCH_DIRS to a list of directories containing include files
 # not on the default search path for the compiler. Typically 3rd party 
 # dependencies (e.g. netCDF, IDL) are not installed in a location where 
 # compiler normally checks. The compiler will search the directories 
 # in the order specified.  
-INC_SEARCH_DIRS =
+INC_SEARCH_DIRS = /usr/include
 
 # Set NETCDF_LIBS to the name of your netCDF library and any library
 # dependencies that netcdf has. The default is simply 'netcdf'. Starting
@@ -65,7 +65,7 @@
 # http://www.trolltech.com, and when possible, from the vapor
 # web site: www.vapor.ucar.edu. Qt version 4.6.1 or higher is required.
 #
-QTDIR =
+QTDIR = /usr
 
 # If BUILD_GUI is set to 1 **and** this is a Mac system, set HAVE_QT_FRAMEWORK
 # to 1 if your Qt libraries are built as Mac Frameworks, or to 0 if 
@@ -91,7 +91,7 @@
 # Python directory where the sub directories 'lib', and 'include' 
 # may be found. Python version 2.6.5 or higher is required. 
 #
-PYTHONDIR =
+PYTHONDIR = /usr
 
 # If BUILD_PYTHON is set to 1, set PYTHONVERSION to the version 
 # number (both major and minor, e.g. "2.6") of Python.
@@ -99,7 +99,7 @@
 # under $PYTHONDIR/lib/python${PYTHONVERSION} and 
 # $PYTHONDIR/include/python${PYTHONVERSION}, respectively.
 #
-PYTHONVERSION = 2.6
+PYTHONVERSION = 2.7
 
 
 # Set to 1 if you want to add support for Adaptive Mesh Refinement grids
