--- vapor-2.1.0.orig/apps/vaporgui/main/vizwinmgr.cpp	2012-01-13 00:33:17.000000000 +0600
+++ vapor-2.1.0/apps/vaporgui/main/vizwinmgr.cpp	2012-05-12 23:18:22.921926955 +0600
@@ -43,6 +43,7 @@
 #include <iostream>
 #include <fstream>
 #include <sstream>
+#include <typeinfo>
 #include "GL/glew.h"
 #include "glwindow.h"
 #include "histo.h"
--- vapor-2.1.0.orig/lib/vdf/WaveCodecIO.cpp	2011-09-01 00:04:41.000000000 +0600
+++ vapor-2.1.0/lib/vdf/WaveCodecIO.cpp	2012-05-12 23:01:27.655254550 +0600
@@ -5,6 +5,7 @@
 #include <netcdf.h>
 #include <sys/stat.h>
 #include <cstdlib>
+#include <unistd.h>
 #include <vapor/WaveCodecIO.h>
 #ifdef WIN32
 #include <windows.h>
--- vapor-2.1.0.orig/make/config/base.mk	2011-08-31 04:00:50.000000000 +0600
+++ vapor-2.1.0/make/config/base.mk	2012-05-13 12:41:19.378111797 +0600
@@ -385,7 +385,7 @@
 LD_SEARCH_FLAGS += $(addprefix -L, $(LIB_SEARCH_DIRS))
 
 ifdef	RPATHFLAG
-RPATHS += $(addprefix $(RPATHFLAG), $(DSO_DIR))
+#RPATHS += $(addprefix $(RPATHFLAG), $(DSO_DIR))
 ifdef	QT_LIB_DIRS
 RPATHS += $(addprefix $(RPATHFLAG), $(QT_LIB_DIRS))
 endif
