--- a/Makefile
+++ b/Makefile
@@ -3,8 +3,8 @@
 CXX = `$(WXCONFIG) --cxx`
 CFLAGS = -O2
 CXXFLAGS = -O2
-WXCXXFLAGS= `$(WXCONFIG) --cxxflags` -Iudis86 -Imhash/include -MMD -O2 -c ${OPTFLAGS}
-WXLDFLAGS = `$(WXCONFIG) --libs std,aui`
+WXCXXFLAGS= `$(WXCONFIG) --cxxflags` -I/usr/include/udis86 -Imhash/include -MMD -O2 -c ${OPTFLAGS}
+WXLDFLAGS = `$(WXCONFIG) --libs std,aui` -ludis86
 RC = `$(WXCONFIG) --rescomp`
 #RC = x86_64-w64-mingw32-windres --define WX_CPU_AMD64
 RCFLAGS = `$(WXCONFIG) --cxxflags | sed s/' '-m.*//g;`
@@ -23,7 +23,7 @@
 			src/HexEditorCtrl/wxHexCtrl/Tag.cpp\
 			src/HexEditorCtrl/HexEditorCtrlGui.cpp\
 			src/HexEditorFrame.cpp
-LIBS = udis86/libudis86/.libs/libudis86.a mhash/lib/.libs/libmhash.a
+LIBS = mhash/lib/.libs/libmhash.a
 OBJECTS=$(SOURCES:.cpp=.o)
 DEPENDS=$(OBJECTS:.o=.d)
 LANGUAGEDIRS=tr ru
@@ -61,11 +61,6 @@
 %.mo : %.po
 	$(MSGFMT) $< -o $@
 
-udis86/libudis86/.libs/libudis86.a:
-	cd udis86;./autogen.sh
-	cd udis86;./configure --host=$(HOST) CC=$(CC) CXX=$(CXX) CFLAGS="${CFLAGS}" CXXFLAGS="${CXXFLAGS}"
-	cd udis86/libudis86; $(MAKE) $(MFLAGS)
-
 mhash/lib/.libs/libmhash.a:
 	cd mhash; ./configure --host=$(HOST) CC=$(CC) CXX=$(CXX) CFLAGS="${CFLAGS}" CXXFLAGS="${CXXFLAGS}"
 	cd mhash; $(MAKE) $(MFLAGS)
@@ -168,7 +163,6 @@
 	rm -rf $(EXECUTABLE_DIR_MAC)
 	rm -f locale/*/wxHexEditor.mo
 distclean: clean
-	cd udis86;$(MAKE) distclean
 	cd mhash;$(MAKE) distclean
 
 # include the auto-generated dependency files
--- a/src/HexPanels.h
+++ b/src/HexPanels.h
@@ -27,7 +27,7 @@
 #include "HexEditorFrame.h"
 #include "HexEditorCtrl/HexEditorCtrl.h"
 #include "HexEditorCtrl/wxHexCtrl/wxHexCtrl.h"
-#include "../udis86/udis86.h"
+#include <udis86.h>
 
 #ifdef WX_GCH
 #include <wx_pch.h>
