diff -Nru wxHexEditor.orig/makefile wxHexEditor/makefile
--- wxHexEditor.orig/makefile	2012-03-19 08:06:33.800880224 -0400
+++ wxHexEditor/makefile	2012-03-19 08:07:05.694213671 -0400
@@ -1,8 +1,8 @@
 WXCONFIG = wx-config
 CC = `$(WXCONFIG) --cc`
 CPP = `$(WXCONFIG) --cxx`
-CXXFLAGS= `$(WXCONFIG) --cxxflags` -Iudis86 -Imhash/include -MMD -O2 -c ${OPTFLAGS}
-LDFLAGS = `$(WXCONFIG) --libs`
+CXXFLAGS= `$(WXCONFIG) --cxxflags` -I/usr/include/libudis86 -Imhash/include -MMD -O2 -c ${OPTFLAGS}
+LDFLAGS = `$(WXCONFIG) --libs` -ludis86
 RC = `$(WXCONFIG) --rescomp`
 #RC = x86_64-w64-mingw32-windres --define WX_CPU_AMD64
 RCFLAGS = `$(WXCONFIG) --cxxflags | sed s/' '-m.*//g;`
@@ -19,7 +19,7 @@
 			src/HexEditorCtrl/wxHexCtrl/Tag.cpp\
 			src/HexEditorCtrl/HexEditorCtrlGui.cpp\
 			src/HexEditorFrame.cpp
-LIBS = udis86/libudis86/.libs/libudis86.a mhash/lib/.libs/libmhash.a
+LIBS = mhash/lib/.libs/libmhash.a
 OBJECTS=$(SOURCES:.cpp=.o)
 DEPENDS=$(OBJECTS:.o=.d)
 RESOURCES= resources/resource.rc
@@ -48,11 +48,6 @@
 %.o : %.rc
 	$(RC) $(RCFLAGS) $< -o $@
 
-udis86/libudis86/.libs/libudis86.a:
-	cd udis86;./autogen.sh
-	cd udis86;./configure --host=$(HOST)
-	cd udis86/libudis86; $(MAKE) $(MFLAGS)
-
 mhash/lib/.libs/libmhash.a:
 	cd mhash; ./configure --host=$(HOST)
 	cd mhash; $(MAKE) $(MFLAGS)
@@ -143,7 +138,6 @@
 	rm -f $(EXECUTABLE_WIN)
 	rm -rf $(EXECUTABLE_DIR_MAC)
 distclean: clean
-	cd udis86;$(MAKE) distclean
 	cd mhash;$(MAKE) distclean
 
 # include the auto-generated dependency files
diff -Nru wxHexEditor.orig/src/HexPanels.h wxHexEditor/src/HexPanels.h
--- wxHexEditor.orig/src/HexPanels.h	2012-03-19 08:06:33.790880223 -0400
+++ wxHexEditor/src/HexPanels.h	2012-03-19 08:06:46.487546943 -0400
@@ -27,7 +27,7 @@
 #include "HexEditorFrame.h"
 #include "HexEditorCtrl/HexEditorCtrl.h"
 #include "HexEditorCtrl/wxHexCtrl/wxHexCtrl.h"
-#include "../udis86/udis86.h"
+#include <udis86.h>
 
 #ifdef WX_GCH
 #include <wx_pch.h>
