--- src/HexEditor.cpp~	2010-04-21 17:35:38.000000000 -0500
+++ src/HexEditor.cpp	2010-07-02 11:36:43.000000000 -0500
@@ -869,20 +869,20 @@
 	}
 
 void HexEditor::FindDialog( void ){
-	class FindDialog *myfind = new FindDialog::FindDialog( this, myfile );
+	::FindDialog *myfind = new ::FindDialog( this, myfile );
 	myfind->ShowModal();
 	myfind->Destroy();
 	}
 
 void HexEditor::ReplaceDialog( void ){
-	class ReplaceDialog *myfind = new ReplaceDialog::ReplaceDialog( this, myfile );
+	::ReplaceDialog *myfind = new ::ReplaceDialog( this, myfile );
 	myfind->ShowModal();
 	myfind->Destroy();
 	}
 
 void HexEditor::GotoDialog( void ){
 	uint64_t newoffset;
-	class GotoDialog *mygoto = new GotoDialog::GotoDialog( this, newoffset, CursorOffset(), FileLength(), myDialogVector );
+	::GotoDialog *mygoto = new ::GotoDialog( this, newoffset, CursorOffset(), FileLength(), myDialogVector );
 	if( mygoto->ShowModal() == wxID_OK ){
 		Goto( newoffset );
 		}
