--- extern_libs.lua	2011-09-14 14:02:56.215407986 +0200
+++ extern_libs.lua.old	2011-09-14 14:02:26.422074990 +0200
@@ -77,7 +77,7 @@
 -- * defines: a table of symbols to define
 extern_lib_defs = {
 	boost = {
-		unix_names = { "boost_signals-mt", "boost_filesystem-mt", "boost_system-mt" },
+		unix_names = { "boost_signals", "boost_filesystem", "boost_system" },
 		osx_names = { "boost_signals-mt", "boost_filesystem-mt", "boost_system-mt" }
 	},
 	cryptopp = {
