--- extern_libs4.lua.old	2012-03-07 10:11:44.950507362 +0100
+++ extern_libs4.lua	2012-03-07 10:12:58.190507217 +0100
@@ -218,7 +218,7 @@
 			end
 			add_default_links({
 				android_names = { "boost_filesystem-gcc-mt", "boost_system-gcc-mt" },
-				unix_names = { "boost_filesystem-mt", "boost_system-mt" },
+				unix_names = { "boost_filesystem", "boost_system" },
 				bsd_names = { "boost_filesystem", "boost_system" },
 			})
 		end,
@@ -227,7 +227,7 @@
 		link_settings = function()
 			add_default_links({
 				android_names = { "boost_signals-gcc-mt" },
-				unix_names = { "boost_signals-mt" },
+				unix_names = { "boost_signals" },
 				bsd_names = { "boost_signals" },
 			})
 		end,
