diff --git a/etc/buildsys/config.mk b/etc/buildsys/config.mk
index b0447c8..f010389 100644
--- a/etc/buildsys/config.mk
+++ b/etc/buildsys/config.mk
@@ -75,6 +75,7 @@ endif
 BUILD_TYPE ?= fawkes
 ARCH=$(shell uname -m)
 OS=$(shell uname -s)
+DISTRO=$(shell sed -n "/ID/ s/ID=//p" /etc/os-release)
 
 
 ### Directories
diff --git a/etc/buildsys/lua.mk b/etc/buildsys/lua.mk
index 5222f0c..7c66972 100644
--- a/etc/buildsys/lua.mk
+++ b/etc/buildsys/lua.mk
@@ -67,6 +67,15 @@ ifeq ($(HAVE_LUA),1)
       _HAVE_TOLUA_BIN=1
       TOLUAPP=tolua++$(LUA_VERSION)
     endif
+    # Arch Linux
+    ifneq ($(wildcard $(SYSROOT)/usr/lib/libtolua++.a),)
+      _HAVE_TOLUA_LIB=1
+      TOLUA_LIBS=tolua++ stdc++
+    endif
+    ifneq ($(wildcard $(SYSROOT)/usr/bin/tolua++),)
+      _HAVE_TOLUA_BIN=1
+      TOLUAPP=tolua++
+    endif
     ifeq ($(_HAVE_TOLUA_LIB)$(_HAVE_TOLUA_BIN),11)
       HAVE_TOLUA = 1
     endif
diff --git a/src/plugins/flite/flite.mk b/src/plugins/flite/flite.mk
index d7eb713..96cf532 100644
--- a/src/plugins/flite/flite.mk
+++ b/src/plugins/flite/flite.mk
@@ -17,6 +17,13 @@ ifneq ($(wildcard $(SYSROOT)/usr/include/flite/flite.h $(SYSROOT)/usr/local/incl
   HAVE_FLITE=1
 endif
 
+ifeq ($(HAVE_FLITE),1)
+  ifeq ($(DISTRO),arch)
+    CFLAGS_FLITE += -lflite_cmu_us_kal -lflite_usenglish -lflite_cmulex -lflite
+    LDFLAGS_FLITE += -lflite_cmu_us_kal -lflite_usenglish -lflite_cmulex -lflite
+  endif
+endif
+
 ifneq ($(PKGCONFIG),)
   HAVE_ALSA = $(if $(shell $(PKGCONFIG) --exists 'alsa'; echo $${?/1/}),1,0)
   CFLAGS_ALSA  = $(shell $(PKGCONFIG) --cflags 'alsa')
