=== modified file 'src/bindings/org/gnome/glib/Plumbing.java'
--- src/bindings/org/gnome/glib/Plumbing.java	2011-06-19 04:33:24 +0000
+++ src/bindings/org/gnome/glib/Plumbing.java	2011-12-15 21:09:40 +0000
@@ -153,7 +153,7 @@
      * Retrieve an array of appropriate Java Boxeds for the given array of
      * pointers.
      */
-    protected static Boxed[] boxedArrayFor(final Class<?> type, final long[] pointers, Boxed[] boxeds) {
+    protected static Boxed[] boxedArrayFor(final Class<? extends Boxed> type, final long[] pointers, Boxed[] boxeds) {
         if (pointers == null) {
             return null;
         }
@@ -170,7 +170,7 @@
      * @throw ClassCastException if the GType pointed by given pointer is not
      *        a GBoxed.
      */
-    protected static Boxed boxedFor(Class<?> type, final long pointer) {
+    protected static Boxed boxedFor(Class<? extends Boxed> type, final long pointer) {
         Boxed proxy;
 
         if (pointer == 0L) {
@@ -202,7 +202,7 @@
      * 
      * @see #boxedFor(Class, long)
      */
-    protected static void fillBoxedArray(Class<?> type, Boxed[] boxeds, long[] pointers) {
+    protected static void fillBoxedArray(Class<? extends Boxed> type, Boxed[] boxeds, long[] pointers) {
         if (pointers == null) {
             return;
         }

