diff -aur adeskbar-0.4.3/src/adesk/core.py src/adeskbar-0.4.3/src/adesk/core.py
--- adeskbar-0.4.3/src/adesk/core.py	2010-10-11 22:16:02.000000000 +0200
+++ src/adeskbar-0.4.3/src/adesk/core.py	2011-08-27 00:31:32.000000000 +0200
@@ -10,6 +10,8 @@
 import gmenu
 from xdg import BaseDirectory, DesktopEntry
 
+import cgi
+
 # Only for debugging : False / True
 DEBUG = 0
 
@@ -124,7 +126,7 @@
     if not label:
         label = ''
         
-    bt_label.set_markup(label)
+    bt_label.set_markup(cgi.escape(label))
     bt_label.set_alignment(0.2, 1)
     align = gtk.Alignment(1, 0, 0.5, 0.5)
     align.show()
