diff -Naur a/drivers/net/wireless/rtlwifi/Makefile b/linux-3.6/drivers/net/wireless/rtlwifi/Makefile
--- a/drivers/net/wireless/rtlwifi/Makefile	2012-11-27 15:40:33.071768000 +0100
+++ b/drivers/net/wireless/rtlwifi/Makefile	2012-11-27 15:57:10.068608424 +0100
@@ -24,6 +24,8 @@
 obj-$(CONFIG_RTL8192CU)		+= rtl8192cu/
 obj-$(CONFIG_RTL8192SE)		+= rtl8192se/
 obj-$(CONFIG_RTL8192DE)		+= rtl8192de/
+obj-$(CONFIG_RTL8192DE)		+= rtl8723e/
+
 
 
 ccflags-y += -D__CHECK_ENDIAN__

diff -Naur a/drivers/net/wireless/rtlwifi/rtl8723e/Makefile b/linux-3.6/drivers/net/wireless/rtlwifi/rtl8723e/Makefile
--- a/drivers/net/wireless/rtlwifi/rtl8723e/Makefile	2012-11-27 15:10:51.458151000 +0100
+++ b/drivers/net/wireless/rtlwifi/rtl8723e/Makefile	2012-11-27 16:00:43.501977892 +0100
@@ -1,46 +1,19 @@
+rtl8723e-objs :=		\
+		hw.o		\
+		table.o		\
+		sw.o		\
+		trx.o		\
+		led.o		\
+		fw.o		\
+		phy.o		\
+		f.o		\
+		dm.o		\
+		pwrseq.o	\
+		pwrseqcmd.o	\
+		hal_btc.o	\
+		hal_bt_coexist.o
 
-CC = gcc
-KVER  := $(shell uname -r)
-KSRC := /lib/modules/$(KVER)/build
-MODDESTDIR := /lib/modules/$(KVER)/kernel/drivers/net/wireless/
-PWD := $(shell pwd)
-
-EXTRA_CFLAGS += -O2
-#EXTRA_CFLAGS += -DRTK_FW_HDR
-
-####################################################
 obj-m := rtl8723e.o
 
-
-PCI_8723E_HAL_OBJS:=	\
-	hw.o		\
-	table.o		\
-	sw.o		\
-	trx.o		\
-	led.o		\
-	fw.o		\
-	phy.o		\
-	rf.o		\
-	dm.o		\
-	pwrseq.o	\
-	pwrseqcmd.o	\
-	hal_btc.o	\
-	hal_bt_coexist.o	\
-
-####################################################
-
-rtl8723e-objs += $(PCI_8723E_HAL_OBJS)
-
-default:
-	        $(MAKE) -C $(KSRC) M=$(PWD) modules
-
-.PHONY:clean
-clean:
-	rm -fr *.mod.c *.mod *.o .*.cmd *.ko *~
-	rm -fr .tmp_versions
-	rm -fr Modules.symvers
-	rm -fr Module.symvers
-	rm -fr Module.markers
-	rm -fr modules.order
-	rm -fr tags
+ccflags-y += -D__CHECK_ENDIAN__
 
