--- gmediarender-0.0.6-patch/configure.in	2012-09-13 16:05:28.109891146 +0200
+++ gmediarender-0.0.6/configure.in	2012-09-13 16:05:47.077117797 +0200
@@ -28,7 +28,7 @@
 if test -n "$GCC"; then
    EXTRA_GCC_DEBUG_CFLAGS="$CFLAGS"
    EXTRA_GCC_DEBUG_CXXFLAGS="$CXXFLAGS"
-   CFLAGS="-Wall -Wpointer-arith -Wcast-align -Wmissing-prototypes -Wmissing-declarations"
+   CFLAGS="-Wall -Wpointer-arith -Wcast-align -Wmissing-prototypes -Wmissing-declarations /usr/lib/libixml.so.2"
    CXXFLAGS="-Wall -Wpointer-arith -Wcast-align"
 fi  
 
