diff --git a/config/environment.rb b/config/environment.rb
index d4e34ed..7f52674 100644
--- a/config/environment.rb
+++ b/config/environment.rb
@@ -2,7 +2,7 @@
 
 # Uncomment below to force Rails into production mode when
 # you don't control web/app server and can't set it the proper way
-# ENV['RAILS_ENV'] ||= 'production'
+ENV['RAILS_ENV'] ||= 'production'
 
 # Specifies gem version of Rails to use when vendor/rails is not present
 RAILS_GEM_VERSION = '2.3.5' unless defined? RAILS_GEM_VERSION
diff --git a/config/environments/production.rb b/config/environments/production.rb
index ed9b828..e2e0181 100644
--- a/config/environments/production.rb
+++ b/config/environments/production.rb
@@ -1,5 +1,8 @@
 # Settings specified here will take precedence over those in config/environment.rb
 
+# Disable SSL to simplify deployment
+SslRequirement.disable_ssl_check = true
+
 # The production environment is meant for finished, "live" apps.
 # Code is not reloaded between requests
 config.cache_classes = true
