diff --git a/lib/gitorious/diagnostics.rb b/lib/gitorious/diagnostics.rb
index d847dc5..78b7e4a 100644
--- a/lib/gitorious/diagnostics.rb
+++ b/lib/gitorious/diagnostics.rb
@@ -182,7 +182,7 @@ module Gitorious
     end
 
     def ultrasphinx_up?
-      atleast_one_process_name_matching("searchd")
+      atleast_one_process_name_matching("sphinx-searchd")
       # TODO + does it respond on configured port, expected by rails app?
       # TODO check ps -o for pid, we got the pidfile for it
     end
