diff --git a/vendor/plugins/ultrasphinx/tasks/ultrasphinx.rake b/vendor/plugins/ultrasphinx/tasks/ultrasphinx.rake
index 9181802..ac5bc19 100644
--- a/vendor/plugins/ultrasphinx/tasks/ultrasphinx.rake
+++ b/vendor/plugins/ultrasphinx/tasks/ultrasphinx.rake
@@ -48,7 +48,7 @@ namespace :ultrasphinx do
     task :start => [:_environment] do
       FileUtils.mkdir_p File.dirname(Ultrasphinx::DAEMON_SETTINGS["log"]) rescue nil
       raise Ultrasphinx::DaemonError, "Already running" if ultrasphinx_daemon_running?
-      system "searchd --config '#{Ultrasphinx::CONF_PATH}'"
+      system "sphinx-searchd --config '#{Ultrasphinx::CONF_PATH}'"
       sleep(4) # give daemon a chance to write the pid file
       if ultrasphinx_daemon_running?
         say "started successfully"
@@ -152,7 +152,7 @@ def ultrasphinx_index(index)
   rotate = ultrasphinx_daemon_running?
   ultrasphinx_create_index_path
   
-  cmd = "indexer --config '#{Ultrasphinx::CONF_PATH}'"
+  cmd = "sphinx-indexer --config '#{Ultrasphinx::CONF_PATH}'"
   cmd << " #{ENV['OPTS']} " if ENV['OPTS']
   cmd << " --rotate" if rotate
   cmd << " #{index}"
@@ -171,7 +171,7 @@ def ultrasphinx_merge
     raise "#{index} index is missing" unless File.exist? "#{Ultrasphinx::INDEX_SETTINGS['path']}/sphinx_index_#{index}.spa"
   end
   
-  cmd = "indexer --config '#{Ultrasphinx::CONF_PATH}'"
+  cmd = "sphinx-indexer --config '#{Ultrasphinx::CONF_PATH}'"
   cmd << " #{ENV['OPTS']} " if ENV['OPTS']
   cmd << " --rotate" if rotate
   cmd << " --merge #{indexes.join(' ')}"
@@ -187,7 +187,7 @@ def ultrasphinx_check_rotate
   failed = Dir[Ultrasphinx::INDEX_SETTINGS['path'] + "/*.new.*"]
   if failed.any?
     say "warning; index failed to rotate! Deleting new indexes"
-    say "try 'killall searchd' and then 'rake ultrasphinx:daemon:start'"
+    say "try 'killall sphinx-searchd' and then 'rake ultrasphinx:daemon:start'"
     failed.each {|f| File.delete f }
   else
     say "index rotated ok"
@@ -203,4 +203,4 @@ end
 def say msg
   Ultrasphinx.say msg
 end
-  
\ No newline at end of file
+  
diff --git a/vendor/plugins/ultrasphinx/vendor/riddle/spec/sphinx_helper.rb b/vendor/plugins/ultrasphinx/vendor/riddle/spec/sphinx_helper.rb
index 7d476a5..327d91b 100644
--- a/vendor/plugins/ultrasphinx/vendor/riddle/spec/sphinx_helper.rb
+++ b/vendor/plugins/ultrasphinx/vendor/riddle/spec/sphinx_helper.rb
@@ -55,7 +55,7 @@ class SphinxHelper
   end
   
   def index
-    cmd = "indexer --config #{@path}/fixtures/sphinx/spec.conf --all"
+    cmd = "sphinx-indexer --config #{@path}/fixtures/sphinx/spec.conf --all"
     cmd << " --rotate" if running?
     `#{cmd}`
   end
@@ -63,7 +63,7 @@ class SphinxHelper
   def start
     return if running?
 
-    cmd = "searchd --config #{@path}/fixtures/sphinx/spec.conf"
+    cmd = "sphinx-searchd --config #{@path}/fixtures/sphinx/spec.conf"
     `#{cmd}`
 
     sleep(1)
@@ -89,4 +89,4 @@ class SphinxHelper
   def running?
     pid && `ps #{pid} | wc -l`.to_i > 1
   end
-end
\ No newline at end of file
+end
