--- a/lib/hypnodata.cc
+++ b/lib/hypnodata.cc
@@ -1,5 +1,6 @@
 #include <stdlib.h>
 #include <string.h>
+#include <cstdio>
 #include <napkin/exception.h>
 #include <napkin/util.h>
 #include <napkin/types.h>
--- a/lib/st-decode.cc
+++ b/lib/st-decode.cc
@@ -2,6 +2,7 @@
 #include <string.h>
 #include <stdexcept>
 #include <numeric>
+#include <stdint.h>
 #include <napkin/exception.h>
 #include <napkin/st/decode.h>
