diff -rupN vc-dwim-1.5.orig/tests/symlinked-changelog vc-dwim-1.5/tests/symlinked-changelog
--- vc-dwim-1.5.orig/tests/symlinked-changelog	2011-05-04 11:44:36.000000000 -0400
+++ vc-dwim-1.5/tests/symlinked-changelog	2011-05-23 10:38:18.319418289 -0400
@@ -18,6 +18,8 @@ EOF
 HOME=`pwd`
 export HOME
 bzr whoami 'U. Ser <user@example.com>'
+git config --global user.name "U. Ser"
+git config --global user.email "<user@example.com>"
 
 (cd m && bzr init && bzr add . && bzr ci -m m . ) || framework_failure_
 ln -s ../d/ChangeLog m
