diff -aur MTPdude-0.0.6/MTPdude.desktop.in MTPdude-0.0.6-diff/MTPdude.desktop.in
--- MTPdude-0.0.6/MTPdude.desktop.in	2006-03-04 03:14:51.000000000 +0100
+++ MTPdude-0.0.6-diff/MTPdude.desktop.in	2006-05-21 16:02:41.000000000 +0200
@@ -1,9 +1,11 @@
 
 [Desktop Entry]
 Name=MTPdude
-Comment=MTPdude Version 0.1
-Exec=mtpdude
-Icon=@PACKAGE_PIXMAPS_DIR@/ 
+Comment=MTPdude Version 0.0.6
+Exec=mtpdude 
 Terminal=false
-MultipleArgs=false
+X-MultipleArgs=false
 Type=Application
+Categories=Application;AudioVideo;
+Encoding=UTF-8
+
diff -aur MTPdude-0.0.6/src/functions.c MTPdude-0.0.6-diff/src/functions.c
--- MTPdude-0.0.6/src/functions.c	2006-04-16 18:02:48.000000000 +0200
+++ MTPdude-0.0.6-diff/src/functions.c	2006-05-18 01:45:16.000000000 +0200
@@ -605,7 +605,7 @@
 	g_message("LIBMTP_Send_Track_From_File(daten->device,daten->touploadfilename,track,NULL,NULL);\n");
 	g_message("endpath:%s\n",fullpath);
 	#endif
-	ret = LIBMTP_Send_Track_From_File(daten->device,fullpath,track,(LIBMTP_progressfunc_t const * const)progress,daten);
+	ret = LIBMTP_Send_Track_From_File(daten->device,fullpath,track,(LIBMTP_progressfunc_t const * const)progress,daten,0);
 	if(ret!=0)
 	{
 		gtk_label_set_text(GTK_LABEL(daten->lblstatusbar),_("There was an error during file transfer"));
diff -aur MTPdude-0.0.6/src/mp3file.c MTPdude-0.0.6-diff/src/mp3file.c
--- MTPdude-0.0.6/src/mp3file.c	2006-04-13 12:15:49.000000000 +0200
+++ MTPdude-0.0.6-diff/src/mp3file.c	2006-05-18 01:45:37.000000000 +0200
@@ -18,6 +18,7 @@
 
 #include <sys/stat.h>
 #include "common.h"
+#include <id3.h>
 
 
 
