*** geogebra-old	2011-05-12 19:02:10.000000000 +0200
--- geogebra	2011-05-17 00:01:54.225904234 +0200
***************
*** 26,36 ****
  _USAGE
  }
  
- # if Java command not set, use java
- if [ -z "$JAVACMD" ]; then
- 	JAVACMD='java'
- fi
- 
  # check for option --help and pass memory options to Java, others to GeoGebra
  for i in "$@"; do
  	case "$i" in
--- 26,31 ----
***************
*** 64,82 ****
  	JAVA_OPTS="$JAVA_OPTS -Xms32m"
  fi
  
- # if path of geogebra.jar not set, use path of this script
- if [ -z "$GG_PATH" ]; then
- 	GG_PATH="${BASH_SOURCE[0]}"
- 	if [ -h "${GG_PATH}" ]; then
- 		while [ -h "${GG_PATH}" ]; do
- 		GG_PATH=`readlink "${GG_PATH}"`
- 		done
- 	fi
- 	pushd . > /dev/null
- 	cd `dirname ${GG_PATH}` > /dev/null
- 	GG_PATH=`pwd`
- 	popd > /dev/null
- fi
- 
  # run
! exec $JAVACMD $JAVA_OPTS -jar $GG_PATH/geogebra.jar $GG_OPTS "$@"
--- 59,63 ----
  	JAVA_OPTS="$JAVA_OPTS -Xms32m"
  fi
  
  # run
! exec java $JAVA_OPTS -jar /usr/share/java/geogebra/geogebra.jar --CAS=Maxima $GG_OPTS "$@"
