From: David Paleino <dapal@debian.org>
Subject: improve random seed
Origin: vendor
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=547595
Forwarded: Alvaro Lopez Ortega <alvaro@alobbs.com> <20090921114216.1978b3d6@pingu>
---
 src/main.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

--- macchanger.orig/src/main.c
+++ macchanger/src/main.c
@@ -27,6 +27,8 @@
 #include <config.h>
 #endif
 
+#include <sys/time.h>
+#include <sys/types.h>
 #include <time.h>
 #include <stdio.h>
 #include <getopt.h>
@@ -175,7 +177,12 @@ main (int argc, char *argv[])
 	device_name = argv[optind];
 	
 	
-	srandom(time(NULL));
+	struct timeval tv;
+	if (gettimeofday(&tv, NULL) == 0) {
+		srandom(tv.tv_sec * tv.tv_usec * getpid());
+	} else {
+		srandom(time(NULL));
+	}
 
         /* Read the MAC */
 	if ((net = mc_net_info_new(device_name)) == NULL) exit(1);
