diff -aur adeskmenu/adesk/__init__.py adeskmenu.diff/adesk/__init__.py
--- adeskmenu/adesk/__init__.py	2010-06-22 20:02:55.000000000 +0200
+++ adeskmenu.diff/adesk/__init__.py	2010-09-26 02:54:46.000000000 +0200
@@ -265,7 +265,8 @@
     bt.set_border_width(0)
     bt.set_focus_on_click(False)
     bt.set_property('can-focus', False)
-
+    if not label:
+        label = "Unknown"
 
     if image == None:
         bt_image = gtk.Image()
