diff -ur --exclude transport_config.mk a/xmlrpc-c-1.16.44/lib/curl_transport/curlmulti.c b/xmlrpc-c-1.16.44/lib/curl_transport/curlmulti.c
--- a/xmlrpc-c-1.16.44/lib/curl_transport/curlmulti.c	2011-03-25 18:58:35.000000000 +0200
+++ b/xmlrpc-c-1.16.44/lib/curl_transport/curlmulti.c	2012-12-04 00:53:33.000000000 +0200
@@ -17,7 +17,6 @@
 #endif
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 #include <curl/multi.h>
 
diff -ur --exclude transport_config.mk a/xmlrpc-c-1.16.44/lib/curl_transport/curltransaction.c b/xmlrpc-c-1.16.44/lib/curl_transport/curltransaction.c
--- a/xmlrpc-c-1.16.44/lib/curl_transport/curltransaction.c	2011-08-11 04:28:08.000000000 +0300
+++ b/xmlrpc-c-1.16.44/lib/curl_transport/curltransaction.c	2012-12-04 00:53:58.000000000 +0200
@@ -15,7 +15,6 @@
 #include "version.h"
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include "curlversion.h"
diff -ur --exclude transport_config.mk a/xmlrpc-c-1.16.44/lib/curl_transport/xmlrpc_curl_transport.c b/xmlrpc-c-1.16.44/lib/curl_transport/xmlrpc_curl_transport.c
--- a/xmlrpc-c-1.16.44/lib/curl_transport/xmlrpc_curl_transport.c	2011-03-25 18:58:35.000000000 +0200
+++ b/xmlrpc-c-1.16.44/lib/curl_transport/xmlrpc_curl_transport.c	2012-12-04 00:54:29.000000000 +0200
@@ -83,7 +83,6 @@
 #include "xmlrpc-c/time_int.h"
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 #include <curl/multi.h>
 
