*** trayconf.h	2010-12-28 15:42:53.773333735 +0700
--- trayconf.h	2010-12-28 15:43:45.670000402 +0700
***************
*** 17,23 ****
  #ifndef trayconfPLUGIN_H
  #define trayconfPLUGIN_H
  
! #include "qutim/plugininterface.h"
  
  #include "trayconfSettings.h"
  #include "tray.h"
--- 17,23 ----
  #ifndef trayconfPLUGIN_H
  #define trayconfPLUGIN_H
  
! #include <../../../sdk02/include/qutim/plugininterface.h>
  
  #include "trayconfSettings.h"
  #include "tray.h"
