--- plasma-wifi-0.5/CMakeLists.txt~	2012-07-18 09:18:01.482513076 +0200
+++ plasma-wifi-0.5/CMakeLists.txt	2012-07-18 09:18:15.919101532 +0200
@@ -6,7 +6,6 @@
 # Find the required Libaries
 find_package(KDE4 REQUIRED)
 include(KDE4Defaults)
-find_package(Plasma REQUIRED)
  
 add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 include_directories(
@@ -22,7 +21,7 @@
 # Now make sure all files get to the right place
 kde4_add_plugin(plasma_applet_wifi_signal ${plasma_wifi_signal_SRCS})
 target_link_libraries(plasma_applet_wifi_signal 
-                      ${PLASMA_LIBS} ${KDE4_KDEUI_LIBS})
+                      ${KDE4_PLASMA_LIBS} ${KDE4_KDEUI_LIBS})
 
 # Build the tests
 #kde4_add_unit_test( test_plasma_wifi test_plasma-wifi-signal.cpp )
