diff -r 906dd8087f12 docs/manual/illustrations/CMakeLists.txt
--- a/docs/manual/illustrations/CMakeLists.txt	Tue Jul 09 23:15:37 2013 +0200
+++ b/docs/manual/illustrations/CMakeLists.txt	Thu Jul 11 21:41:42 2013 +0100
@@ -29,7 +29,7 @@
 
 set_target_properties(OgreProceduralIllustrations PROPERTIES DEBUG_POSTFIX _d)
 
-target_link_libraries(OgreProceduralIllustrations ${OGRE_LIBRARIES} ${OIS_LIBRARIES} ${OgreProcedural_LIBRARIES})
+target_link_libraries(OgreProceduralIllustrations ${OGRE_LIBRARIES} ${OIS_LIBRARIES} ${OgreProcedural_LIBRARIES} ${Boost_LIBRARIES})
 
 procedural_create_vcproj_userfile(OgreProceduralIllustrations)
 
diff -r 906dd8087f12 samples/extrusion/CMakeLists.txt
--- a/samples/extrusion/CMakeLists.txt	Tue Jul 09 23:15:37 2013 +0200
+++ b/samples/extrusion/CMakeLists.txt	Thu Jul 11 21:41:42 2013 +0100
@@ -22,11 +22,11 @@
 
 set_target_properties(Sample_Extrusion PROPERTIES DEBUG_POSTFIX _d)
 
-target_link_libraries(Sample_Extrusion ${OgreProcedural_LIBRARIES} ${OGRE_LIBRARIES} ${OIS_LIBRARIES} )
+target_link_libraries(Sample_Extrusion ${OgreProcedural_LIBRARIES} ${OGRE_LIBRARIES} ${OIS_LIBRARIES} ${Boost_LIBRARIES} )
 
 if ((${OGRE_VERSION} VERSION_EQUAL "1.9.0") OR (${OGRE_VERSION} VERSION_GREATER "1.9.0"))
 	include_directories(${OGRE_Overlay_INCLUDE_DIR})
 	target_link_libraries(Sample_Extrusion ${OGRE_Overlay_LIBRARIES})
 endif()
 
-procedural_create_vcproj_userfile(Sample_Extrusion)
\ No newline at end of file
+procedural_create_vcproj_userfile(Sample_Extrusion)
diff -r 906dd8087f12 samples/material/CMakeLists.txt
--- a/samples/material/CMakeLists.txt	Tue Jul 09 23:15:37 2013 +0200
+++ b/samples/material/CMakeLists.txt	Thu Jul 11 21:41:42 2013 +0100
@@ -22,11 +22,11 @@
 
 set_target_properties(Sample_Material PROPERTIES DEBUG_POSTFIX _d)
 
-target_link_libraries(Sample_Material ${OgreProcedural_LIBRARIES} ${OGRE_LIBRARIES} ${OIS_LIBRARIES} ${OGRE_RTShaderSystem_LIBRARY})
+target_link_libraries(Sample_Material ${OgreProcedural_LIBRARIES} ${OGRE_LIBRARIES} ${OIS_LIBRARIES} ${OGRE_RTShaderSystem_LIBRARY} ${Boost_LIBRARIES})
 
 if ((${OGRE_VERSION} VERSION_EQUAL "1.9.0") OR (${OGRE_VERSION} VERSION_GREATER "1.9.0"))
 	include_directories(${OGRE_Overlay_INCLUDE_DIR})
 	target_link_libraries(Sample_Material ${OGRE_Overlay_LIBRARIES})
 endif()
 
-procedural_create_vcproj_userfile(Sample_Material)
\ No newline at end of file
+procedural_create_vcproj_userfile(Sample_Material)
diff -r 906dd8087f12 samples/primitives/CMakeLists.txt
--- a/samples/primitives/CMakeLists.txt	Tue Jul 09 23:15:37 2013 +0200
+++ b/samples/primitives/CMakeLists.txt	Thu Jul 11 21:41:42 2013 +0100
@@ -22,11 +22,11 @@
 
 set_target_properties(Sample_Primitives PROPERTIES DEBUG_POSTFIX _d)
 
-target_link_libraries(Sample_Primitives ${OgreProcedural_LIBRARIES} ${OGRE_LIBRARIES} ${OIS_LIBRARIES})
+target_link_libraries(Sample_Primitives ${OgreProcedural_LIBRARIES} ${OGRE_LIBRARIES} ${OIS_LIBRARIES} ${Boost_LIBRARIES})
 
 if ((${OGRE_VERSION} VERSION_EQUAL "1.9.0") OR (${OGRE_VERSION} VERSION_GREATER "1.9.0"))
 	include_directories(${OGRE_Overlay_INCLUDE_DIR})
 	target_link_libraries(Sample_Primitives ${OGRE_Overlay_LIBRARIES})
 endif()
 
-procedural_create_vcproj_userfile(Sample_Primitives)
\ No newline at end of file
+procedural_create_vcproj_userfile(Sample_Primitives)
diff -r 906dd8087f12 samples/svg/CMakeLists.txt
--- a/samples/svg/CMakeLists.txt	Tue Jul 09 23:15:37 2013 +0200
+++ b/samples/svg/CMakeLists.txt	Thu Jul 11 21:41:42 2013 +0100
@@ -22,11 +22,11 @@
 
 set_target_properties(Sample_SVG PROPERTIES DEBUG_POSTFIX _d)
 
-target_link_libraries(Sample_SVG ${OgreProcedural_LIBRARIES} ${OGRE_LIBRARIES} ${OIS_LIBRARIES})
+target_link_libraries(Sample_SVG ${OgreProcedural_LIBRARIES} ${OGRE_LIBRARIES} ${OIS_LIBRARIES} ${Boost_LIBRARIES})
 
 if ((${OGRE_VERSION} VERSION_EQUAL "1.9.0") OR (${OGRE_VERSION} VERSION_GREATER "1.9.0"))
 	include_directories(${OGRE_Overlay_INCLUDE_DIR})
 	target_link_libraries(Sample_SVG ${OGRE_Overlay_LIBRARIES})
 endif()
 
-procedural_create_vcproj_userfile(Sample_SVG)
\ No newline at end of file
+procedural_create_vcproj_userfile(Sample_SVG)
