--- gst-plugins-good-0.10.31/gst/deinterlace/tvtime/x86-64_macros.inc	2011-12-11 13:48:11.000000000 -0500
+++ gst-plugins-good-0.10.31/gst/deinterlace/tvtime/x86-64_macros.inc.new	2013-04-13 15:42:17.175088182 -0400
@@ -35,8 +35,6 @@
 
 #ifndef XAX
 
-#if defined (HAVE_CPU_I386) && !defined(HAVE_CPU_X86_64)
-
 #define XAX   "eax"
 #define XBX   "ebx"
 #define XCX   "ecx"
@@ -55,28 +53,4 @@
 #define SHRX  "shrl"
 #define SUBX  "subl"
 
-#elif defined (HAVE_CPU_X86_64)
-
-#define XAX   "rax"
-#define XBX   "rbx"
-#define XCX   "rcx"
-#define XDX   "rdx"
-#define XSI   "rsi"
-#define XDI   "rdi"
-#define XSP   "rsp"
-#define MOVX  "movq"
-#define LEAX  "leaq"
-#define DECX  "decq"
-#define PUSHX "pushq"
-#define POPX  "popq"
-#define CMPX  "cmpq"
-#define ADDX  "addq"
-#define SHLX  "shlq"
-#define SHRX  "shrq"
-#define SUBX  "subq"
-
-#else
-#error Undefined architecture. Define either ARCH_X86 or ARCH_X86_64.
-#endif
-
 #endif
