Index: autogen.sh
===================================================================
--- autogen.sh	(revision 588)
+++ autogen.sh	(working copy)
@@ -1,5 +1,5 @@
 #! /bin/sh
-AM_VERSION=-1.11
+AM_VERSION=-1.12
 AC_VERSION=
 
 set -x
Index: src/core/cfontconfig.cpp
===================================================================
--- src/core/cfontconfig.cpp	(revision 588)
+++ src/core/cfontconfig.cpp	(working copy)
@@ -43,7 +43,8 @@
     FcDefaultSubstitute(pattern);
 
     FcFontSet* font_set = FcFontSetCreate();
-    FcFontSet* patterns = FcFontSort(0, pattern, FcTrue, 0, NULL);
+    FcResult fresult;
+    FcFontSet* patterns = FcFontSort(0, pattern, FcTrue, 0, &fresult);
 
     for (int i = 0; i < patterns->nfont; i++) {
 
