--- save.c.orig	2009-01-26 14:02:54.630783832 +0100
+++ save.c	2009-01-26 14:12:20.944908215 +0100
@@ -252,7 +252,7 @@
 }
 
 void SaveSettings() {
-	FILE *f=fopen("system/settings.txt","w");
+	FILE *f=fopen("./tuxboy.cfg","w");
 	fprintf(f,"%d\n",scale);
 	fprintf(f,"%d\n",CustomW);
 	fprintf(f,"%d\n",CustomH);
@@ -268,9 +268,11 @@
 }
 
 void LoadSettings() {
-	FILE *f=fopen("system/settings.txt","r");
+	FILE *f=fopen("./tuxboy.cfg","r");
 
 	if(!f) {
+		printf("Unable to open ./tuxboy.cfg, trying /etc/tuxboy/tuxboy.cfg\n");
+		FILE *f=fopen("/etc/tuxboy/tuxboy.cfg","r");
 		SaveSettings();
 		return;
 	}
