--- Makefile.am.orig	2006-12-03 01:32:07.000000000 +0100
+++ Makefile.am	2006-12-03 01:32:21.000000000 +0100
@@ -85,24 +85,6 @@
 localedir = $(game_datadir)/locale
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 
-install-data-local:
-	@$(NORMAL_INSTALL)
-	if test -d $(srcdir)/image; then \
-	  $(mkinstalldirs) $(DESTDIR)$(game_datadir); \
-	  $(mkinstalldirs) $(DESTDIR)$(game_datadir)/image; \
-	  cp -r $(srcdir)/image/* $(DESTDIR)$(game_datadir)/image;\
-	fi; \
-	if test -d $(srcdir)/level; then \
-	  $(mkinstalldirs) $(DESTDIR)$(game_datadir)/ ; \
-	  $(mkinstalldirs) $(DESTDIR)$(game_datadir)/level; \
-	  cp -r $(srcdir)/level/* $(DESTDIR)$(game_datadir)/level;\
-	fi; \
-	if test -d $(srcdir)/sounds; then \
-	  $(mkinstalldirs) $(DESTDIR)$(game_datadir)/ ; \
-	  $(mkinstalldirs) $(DESTDIR)$(game_datadir)/sounds; \
-	  cp -r $(srcdir)/sounds/* $(DESTDIR)$(game_datadir)/sounds;\
-	fi;
-
 indent:
 	 indent -npro -gnu -ts4 -i4 -br -lp -sob -l100 -ss -ncs *.[ch]
 
