*** src/zip/zpng.c	2012-05-29 18:44:36.888974112 +0200
--- src/zip/zpng.c	2012-05-29 18:59:19.292452834 +0200
***************
*** 80,89 ****
  	
  	/*set a lot of image info (code adapted from libpng documentation!)*/
  	png_set_IHDR(png_ptr, info_ptr, width, height,
! 		8, PNG_COLOR_TYPE_RGB_ALPHA, PNG_INTERLACE_NONE,
  		PNG_COMPRESSION_TYPE_DEFAULT, PNG_FILTER_TYPE_DEFAULT);
  	
- 	info_ptr->color_type=PNG_COLOR_TYPE_RGB_ALPHA;
  
  	/*Allocate an array of scanline pointers*/
  	row_pointers=(png_bytep*)malloc(height*sizeof(png_bytep));
--- 80,88 ----
  	
  	/*set a lot of image info (code adapted from libpng documentation!)*/
  	png_set_IHDR(png_ptr, info_ptr, width, height,
! 		8, PNG_COLOR_TYPE_RGB, PNG_INTERLACE_NONE,
  		PNG_COMPRESSION_TYPE_DEFAULT, PNG_FILTER_TYPE_DEFAULT);
  	
  
  	/*Allocate an array of scanline pointers*/
  	row_pointers=(png_bytep*)malloc(height*sizeof(png_bytep));
