diff -rupN src_bak//Makefile src/Makefile
--- src_bak//Makefile	2010-12-16 17:05:52.146666670 -0800
+++ src/Makefile	2010-12-16 17:00:19.400000004 -0800
@@ -6,7 +6,7 @@ APPSHAREDIR = $(SHAREDIR)/$(APPNAME)
 
 #default build
 FFS_CPPFLAGS = -Wall -pipe -DNDEBUG -DwxUSE_UNICODE `wx-config --cxxflags --debug=no --unicode=yes` `pkg-config --cflags gtk+-2.0` -DFFS_LINUX -DTIXML_USE_STL -O3 -pthread
-LINKFLAGS    = `wx-config --libs --debug=no --unicode=yes` -lboost_thread -O3 -pthread
+LINKFLAGS    = `wx-config --libs --debug=no --unicode=yes` -lboost_thread-mt -O3 -pthread
 
 #static build used for precompiled release
 ifeq ($(BUILD),release)
