From: Ken VanDine <ken.vandine@ubuntu.com>
Description: Grab handles default shoud be configurable 
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=654557
Bug-Ubuntu: https://launchpad.net/bugs/809919
Author: Ted Gould <ted@ubuntu.com>

=== modified file 'gtk/gtkwindow.c'
--- old/gtk/gtkwindow.c	2011-07-05 20:08:57 +0000
+++ new/gtk/gtkwindow.c	2011-07-12 20:55:58 +0000
@@ -945,6 +945,12 @@
                                                              P_("Height of resize grip"),
                                                              0, G_MAXINT, 16, GTK_PARAM_READWRITE));
 
+  gtk_widget_class_install_style_property (widget_class,
+                                           g_param_spec_boolean ("resize-grip-default",
+                                                                 P_("Whether by default windows have resize grips"),
+                                                                 P_("Adjusts the default value of resize grips to match the theme.  Applications can still override and it's effected by whether the window is resizable or maximized."),
+                                                                 TRUE, GTK_PARAM_READWRITE));
+
 
   /**
    * GtkWindow:application:
@@ -1113,6 +1119,14 @@
   priv->has_resize_grip = TRUE;
   priv->mnemonics_visible = TRUE;
 
+  /* Check the style property to see if resize grips should be on
+     or off by default */
+  gboolean resize_default = TRUE;
+  gtk_widget_style_get (window,
+                        "resize-grip-default", &resize_default,
+                        NULL);
+  priv->has_resize_grip = resize_default;
+
   g_object_ref_sink (window);
   priv->has_user_ref_count = TRUE;
   toplevel_list = g_slist_prepend (toplevel_list, window);

