--- b/src/synaptics.c	2012-06-12 03:23:34.000000000 +0200
+++ a/src/synaptics.c	2012-10-08 02:15:49.867461023 +0200
@@ -3152,11 +3152,12 @@
     Bool restart_touches = FALSE;
     int i;
 
-    if (para->click_action[F3_CLICK1] || para->tap_action[F3_TAP])
+    /* if (para->click_action[F3_CLICK1] || para->tap_action[F3_TAP])
         min_touches = 4;
-    else if (para->click_action[F2_CLICK1] || para->tap_action[F2_TAP] ||
+    else  if (para->click_action[F2_CLICK1] || para->tap_action[F2_TAP] ||
              para->scroll_twofinger_vert || para->scroll_twofinger_horiz)
         min_touches = 3;
+    */
 
     /* Count new number of active touches */
     for (i = 0; i < hw->num_mt_mask; i++) {