--- a/dlls/winex11.drv/mouse.c    2014-08-10 18:07:21.000000000 +0200
+++ b/dlls/winex11.drv/mouse.c     2014-08-28 19:38:48.884645392 +0200
@@ -363,9 +363,6 @@
     HWND msg_hwnd = 0;
     POINT pos;
 
-    if (GetWindowThreadProcessId( GetDesktopWindow(), NULL ) == GetCurrentThreadId())
-        return TRUE;  /* don't clip in the desktop process */
-
     if (!data) return FALSE;
     if (!(clip_window = init_clip_window())) return TRUE;
 
@@ -1387,6 +1384,9 @@
 
     if (!clip) clip = &virtual_rect;
 
+    if (GetWindowThreadProcessId( GetDesktopWindow(), NULL ) == GetCurrentThreadId())
+        return TRUE;  /* don't clip in the desktop process */
+
     if (grab_pointer)
     {
         HWND foreground = GetForegroundWindow();
