diff --git a/CMakeLists.txt b/CMakeLists.txt
index 407ddb0..af0c3d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,7 +29,9 @@ install(FILES COPYING README.md ChangeLog.md
 set(EAR_LIB_FILE ${CMAKE_SHARED_LIBRARY_PREFIX}ear${CMAKE_SHARED_LIBRARY_SUFFIX})
 set(DEFAULT_PRELOAD_FILE ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/${EAR_LIB_FILE})
 set(DEFAULT_OUTPUT_FILE compile_commands.json)
-set(DEFAULT_CONFIG_FILE ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_SYSCONFDIR}/bear.conf)
+set(DEFAULT_CONFIG_FILE
+  ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_SYSCONFDIR}/bear.conf CACHE STRING
+  "default configuration file path")
 
 add_subdirectory(src)
 add_subdirectory(test)
