--- hint_set.c.bak	2011-09-30 15:56:36.891626679 -0400
+++ hint_set.c	2011-09-30 15:56:40.991507664 -0400
@@ -49,7 +49,6 @@
   va_list ap;
   unsigned int index;
   unsigned int count;
-  const char * arg;
 
   /* iterate varargs and compute name/value pairs */
 
@@ -58,7 +57,7 @@
   count = 0;
   while (va_arg(ap, char *) != NULL)
   {
-    arg = va_arg(ap, const char *);
+    va_arg(ap, const char *);
     count++;
   }
 
