From 9a6a34f46869480ea7a5440165fa8acce0028d01 Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <icq@gnome.org>
Date: Sun, 11 Nov 2012 13:02:23 +0000
Subject: build: Bump minimum GLib version to 2.35.1

g_type_init is not needed anymore
---
diff --git a/configure.ac b/configure.ac
index ce19948..3fbffd2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -27,7 +27,7 @@ GLIB_GSETTINGS
 CLUTTER_GTK_MIN_VERSION=1.3.2
 CLUTTER_MIN_VERSION=1.11
 EXEMPI_MIN_VERSION=1.99.5
-GLIB_MIN_VERSION=2.28
+GLIB_MIN_VERSION=2.35.1
 LIBEXIF_MIN_VERSION=0.6.14
 LIBRSVG_MIN_VERSION=2.26.0
 
diff --git a/src/photos-application.c b/src/photos-application.c
index ad43551..7db7327 100644
--- a/src/photos-application.c
+++ b/src/photos-application.c
@@ -228,7 +228,6 @@ photos_application_class_init (PhotosApplicationClass *class)
 GtkApplication *
 photos_application_new (void)
 {
-  g_type_init ();
   return g_object_new (PHOTOS_TYPE_APPLICATION,
                        "application-id", "org.gnome." PACKAGE_NAME,
                        "flags", G_APPLICATION_FLAGS_NONE,
--
cgit v0.9.0.2
