diff -urN openvas-manager-2.0.3.orig//CMakeLists.txt openvas-manager-2.0.3/CMakeLists.txt
--- openvas-manager-2.0.3.orig//CMakeLists.txt	2011-04-15 15:30:03.000000000 +0200
+++ openvas-manager-2.0.3/CMakeLists.txt	2011-05-15 10:28:58.364398937 +0200
@@ -243,7 +243,8 @@
 set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -Wl,-z,relro -Wl,-z,now")
 
 set (CMAKE_C_FLAGS_DEBUG        "${CMAKE_C_FLAGS_DEBUG}")
-set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -Werror -D_BSD_SOURCE -D_ISOC99_SOURCE -D_SVID_SOURCE")
+# set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -Werror -D_BSD_SOURCE -D_ISOC99_SOURCE -D_SVID_SOURCE")
+set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -D_BSD_SOURCE -D_ISOC99_SOURCE -D_SVID_SOURCE")
 
 add_subdirectory (src)
 
@@ -257,7 +258,7 @@
 
 ## Install
 
-install (CODE "file (MAKE_DIRECTORY ${OPENVAS_STATE_DIR}/openvasmd/report_formats/)")
+install (CODE "file (MAKE_DIRECTORY \$ENV{DESTDIR}/${OPENVAS_STATE_DIR}/openvasmd/report_formats/)")
 
 install (FILES ${CMAKE_BINARY_DIR}/src/openvasmd_log.conf
          DESTINATION ${OPENVAS_SYSCONF_DIR})
