diff -wbBur 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/MODULE/common/rtmp_init.c 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/MODULE/common/rtmp_init.c
--- 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/MODULE/common/rtmp_init.c	2010-01-28 08:23:46.000000000 +0300
+++ 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/MODULE/common/rtmp_init.c	2011-01-02 00:47:29.596666668 +0300
@@ -5239,7 +5239,7 @@
 #ifdef OS_ABL_SUPPORT
 
 #ifdef RTMP_MAC_USB
-MODULE_LICENSE("RALINK");
+MODULE_LICENSE("GPL");
 #endif // RTMP_MAC_USB //
 
 EXPORT_SYMBOL(NICInitializeAdapter);
diff -wbBur 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/MODULE/include/os/rt_linux.h 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/MODULE/include/os/rt_linux.h
--- 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/MODULE/include/os/rt_linux.h	2010-02-03 08:59:36.000000000 +0300
+++ 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/MODULE/include/os/rt_linux.h	2011-01-02 00:47:14.260000002 +0300
@@ -1082,8 +1082,8 @@
 #define RT28XX_PUT_DEVICE													usb_put_dev
 #define RTUSB_ALLOC_URB(iso)												usb_alloc_urb(iso, GFP_ATOMIC)
 #define RTUSB_SUBMIT_URB(pUrb)												usb_submit_urb(pUrb, GFP_ATOMIC)
-#define RTUSB_URB_ALLOC_BUFFER(pUsb_Dev, BufSize, pDma_addr)				usb_buffer_alloc(pUsb_Dev, BufSize, GFP_ATOMIC, pDma_addr)
-#define RTUSB_URB_FREE_BUFFER(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)	usb_buffer_free(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)
+#define RTUSB_URB_ALLOC_BUFFER(pUsb_Dev, BufSize, pDma_addr)				usb_alloc_coherent(pUsb_Dev, BufSize, GFP_ATOMIC, pDma_addr)
+#define RTUSB_URB_FREE_BUFFER(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)	usb_free_coherent(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)
 #else
 
 #define RT28XX_PUT_DEVICE													rausb_put_dev
diff -wbBur 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/MODULE/Makefile 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/MODULE/Makefile
--- 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/MODULE/Makefile	2010-01-28 08:23:50.000000000 +0300
+++ 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/MODULE/Makefile	2011-01-02 00:47:14.260000002 +0300
@@ -217,15 +217,15 @@
 	$(MAKE) -C $(RT28xx_DIR)/os/linux/
 
 ifeq ($(RT28xx_MODE),AP)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)ap.o /tftpboot
+	-cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)ap.o /tftpboot
 ifeq ($(PLATFORM),INF_AMAZON_SE)
-	cp -f /tftpboot/rt2870ap.o /backup/ifx/build/root_filesystem/lib/modules/2.4.31-Amazon_SE-3.6.2.2-R0416_Ralink/kernel/drivers/net
+	-cp -f /tftpboot/rt2870ap.o /backup/ifx/build/root_filesystem/lib/modules/2.4.31-Amazon_SE-3.6.2.2-R0416_Ralink/kernel/drivers/net
 endif
 else	
 ifeq ($(RT28xx_MODE),APSTA)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)apsta.o /tftpboot
+	-cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)apsta.o /tftpboot
 else
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)sta.o /tftpboot
+	-cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)sta.o /tftpboot
 endif	
 endif	
 else
@@ -242,14 +242,14 @@
 endif
 
 ifeq ($(RT28xx_MODE),AP)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)ap.ko /tftpboot
+	-cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)ap.ko /tftpboot
 #	rm -f os/linux/rt2860ap.ko.lzma
 #	/root/bin/lzma e os/linux/rt$(CHIPSET)ap.ko os/linux/rt$(CHIPSET)ap.ko.lzma
 else	
 ifeq ($(RT28xx_MODE),APSTA)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)apsta.ko /tftpboot
+	-cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)apsta.ko /tftpboot
 else
-	cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)sta.ko /tftpboot
+	-cp -f $(RT28xx_DIR)/os/linux/rt$(CHIPSET)sta.ko /tftpboot
 endif
 endif	
 endif
diff -wbBur 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/MODULE/os/linux/modules.order 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/MODULE/os/linux/modules.order
--- 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/MODULE/os/linux/modules.order	2010-01-28 08:23:44.000000000 +0300
+++ 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/MODULE/os/linux/modules.order	2011-01-02 00:48:16.516666668 +0300
@@ -1 +1 @@
-kernel//home/Ralink/SWRelease_RT3070STA/RT_WIFI_20091120_RT3070STA_release/os/linux/rtnet3070sta.ko
+kernel//home/sergej/aur.i/rt3070ap/src/2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/MODULE/os/linux/rt3070ap.ko
diff -wbBur 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/NETIF/include/os/rt_linux.h 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/NETIF/include/os/rt_linux.h
--- 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/NETIF/include/os/rt_linux.h	2010-02-03 09:00:16.000000000 +0300
+++ 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/NETIF/include/os/rt_linux.h	2011-01-02 00:47:14.260000002 +0300
@@ -1082,8 +1082,8 @@
 #define RT28XX_PUT_DEVICE													usb_put_dev
 #define RTUSB_ALLOC_URB(iso)												usb_alloc_urb(iso, GFP_ATOMIC)
 #define RTUSB_SUBMIT_URB(pUrb)												usb_submit_urb(pUrb, GFP_ATOMIC)
-#define RTUSB_URB_ALLOC_BUFFER(pUsb_Dev, BufSize, pDma_addr)				usb_buffer_alloc(pUsb_Dev, BufSize, GFP_ATOMIC, pDma_addr)
-#define RTUSB_URB_FREE_BUFFER(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)	usb_buffer_free(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)
+#define RTUSB_URB_ALLOC_BUFFER(pUsb_Dev, BufSize, pDma_addr)				usb_alloc_coherent(pUsb_Dev, BufSize, GFP_ATOMIC, pDma_addr)
+#define RTUSB_URB_FREE_BUFFER(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)	usb_free_coherent(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)
 #else
 
 #define RT28XX_PUT_DEVICE													rausb_put_dev
diff -wbBur 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/NETIF/os/linux/rt_main_dev.c 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/NETIF/os/linux/rt_main_dev.c
--- 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/NETIF/os/linux/rt_main_dev.c	2010-01-28 08:23:52.000000000 +0300
+++ 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/NETIF/os/linux/rt_main_dev.c	2011-01-02 00:47:14.260000002 +0300
@@ -47,9 +47,9 @@
 MODULE_PARM_DESC (mac, "rt28xx: wireless mac addr");
 
 #ifdef OS_ABL_SUPPORT
-#ifdef RTMP_MAC_USB
+//#ifdef RTMP_MAC_USB
 MODULE_LICENSE("GPL");
-#endif // RTMP_MAC_USB //
+//#endif // RTMP_MAC_USB //
 
 UCHAR ZERO_MAC_ADDR[MAC_ADDR_LEN]  = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
 #endif // OS_ABL_SUPPORT //
diff -wbBur 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/UTIL/include/os/rt_linux.h 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/UTIL/include/os/rt_linux.h
--- 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/UTIL/include/os/rt_linux.h	2010-02-03 08:59:58.000000000 +0300
+++ 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/UTIL/include/os/rt_linux.h	2011-01-02 00:47:14.260000002 +0300
@@ -1082,8 +1082,8 @@
 #define RT28XX_PUT_DEVICE													usb_put_dev
 #define RTUSB_ALLOC_URB(iso)												usb_alloc_urb(iso, GFP_ATOMIC)
 #define RTUSB_SUBMIT_URB(pUrb)												usb_submit_urb(pUrb, GFP_ATOMIC)
-#define RTUSB_URB_ALLOC_BUFFER(pUsb_Dev, BufSize, pDma_addr)				usb_buffer_alloc(pUsb_Dev, BufSize, GFP_ATOMIC, pDma_addr)
-#define RTUSB_URB_FREE_BUFFER(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)	usb_buffer_free(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)
+#define RTUSB_URB_ALLOC_BUFFER(pUsb_Dev, BufSize, pDma_addr)				usb_alloc_coherent(pUsb_Dev, BufSize, GFP_ATOMIC, pDma_addr)
+#define RTUSB_URB_FREE_BUFFER(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)	usb_free_coherent(pUsb_Dev, BufSize, pTransferBuf, Dma_addr)
 #else
 
 #define RT28XX_PUT_DEVICE													rausb_put_dev
diff -wbBur 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/UTIL/os/linux/rt_linux.c 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/UTIL/os/linux/rt_linux.c
--- 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/UTIL/os/linux/rt_linux.c	2010-01-28 08:23:50.000000000 +0300
+++ 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/UTIL/os/linux/rt_linux.c	2011-01-02 00:47:14.263333335 +0300
@@ -2,11 +2,11 @@
 
 ULONG	RTDebugLevel = RT_DEBUG_ERROR;
 
-#ifdef RTMP_MAC_USB
-#ifdef OS_ABL_SUPPORT
+//#ifdef RTMP_MAC_USB
+//#ifdef OS_ABL_SUPPORT
 MODULE_LICENSE("GPL");
-#endif // OS_ABL_SUPPORT //
-#endif // RTMP_MAC_USB //
+//#endif // OS_ABL_SUPPORT //
+//#endif // RTMP_MAC_USB //
 
 #ifdef SYSTEM_LOG_SUPPORT
 // for wireless system event message
diff -wbBur 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/UTIL/os/linux/rt_usb_util.c 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/UTIL/os/linux/rt_usb_util.c
--- 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA/UTIL/os/linux/rt_usb_util.c	2010-01-28 08:23:50.000000000 +0300
+++ 2010_0203_RT3070_SoftAP_v2.4.0.1_DPA.my/UTIL/os/linux/rt_usb_util.c	2011-01-02 00:47:14.263333335 +0300
@@ -123,7 +123,7 @@
 							gfp_t mem_flags,
 							dma_addr_t *dma)
 {
-	return usb_buffer_alloc(dev, size, mem_flags, dma);
+	return usb_alloc_coherent(dev, size, mem_flags, dma);
 }
 EXPORT_SYMBOL(rausb_buffer_alloc);
 
@@ -133,7 +133,7 @@
 							void *addr,
 							dma_addr_t dma)
 {
-	usb_buffer_free(dev, size, addr, dma);
+	usb_free_coherent(dev, size, addr, dma);
 }
 EXPORT_SYMBOL(rausb_buffer_free);
 #endif // LINUX_VERSION_CODE //
