diff -ruN kanatest-0.4.8/src/about.c kanatest-0.4.8.1//src/about.c
--- kanatest-0.4.8/src/about.c	2009-04-13 01:10:45.000000000 +0600
+++ kanatest-0.4.8.1//src/about.c	2011-01-09 23:27:15.000000000 +0500
@@ -430,7 +430,7 @@
     g_signal_connect (G_OBJECT (close_button), "clicked",
                         G_CALLBACK (about_close_button_cb), appGUI);
     gtk_container_add (GTK_CONTAINER (hbuttonbox), close_button);
-    GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+    gtk_widget_set_can_default (close_button, TRUE);
 
     gtk_widget_grab_default (close_button);
 
diff -ruN kanatest-0.4.8/src/chart.c kanatest-0.4.8.1//src/chart.c
--- kanatest-0.4.8/src/chart.c	2009-04-13 01:10:45.000000000 +0600
+++ kanatest-0.4.8.1//src/chart.c	2011-01-09 23:26:02.000000000 +0500
@@ -260,7 +260,7 @@
     g_signal_connect (G_OBJECT (close_button), "clicked",
                         G_CALLBACK (chart_close_button_cb), appGUI);
     gtk_container_add (GTK_CONTAINER (hbuttonbox), close_button);
-    GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+    gtk_widget_set_can_default (close_button, TRUE);
 
     chart_display_kanas(HIRAGANA, appGUI);
 
diff -ruN kanatest-0.4.8/src/gui.c kanatest-0.4.8.1//src/gui.c
--- kanatest-0.4.8/src/gui.c	2009-04-13 01:10:45.000000000 +0600
+++ kanatest-0.4.8.1//src/gui.c	2011-01-10 00:24:18.000000000 +0500
@@ -561,7 +561,7 @@
     gtk_box_pack_start (GTK_BOX (hbox2), appGUI->timer_label, TRUE, TRUE, 0);
 
     appGUI->stat_button = gui_stock_label_button(NULL, KANATEST_STOCK_BUTTON_STATISTICS);
-    GTK_WIDGET_UNSET_FLAGS (appGUI->stat_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (appGUI->stat_button, FALSE);
     g_signal_connect (G_OBJECT (appGUI->stat_button), "clicked",
                         G_CALLBACK (show_statistics_window_cb), appGUI);
     gtk_widget_show (appGUI->stat_button);
@@ -570,7 +572,7 @@
     gtk_widget_set_tooltip_text (appGUI->stat_button, _("Statistics"));
 
     appGUI->chart_button = gui_stock_label_button(NULL, KANATEST_STOCK_BUTTON_CHART);
-    GTK_WIDGET_UNSET_FLAGS (appGUI->chart_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (appGUI->chart_button, FALSE);
     g_signal_connect (G_OBJECT (appGUI->chart_button), "clicked",
                         G_CALLBACK (show_chart_window_cb), appGUI);
     gtk_widget_show (appGUI->chart_button);
@@ -579,7 +582,7 @@
     gtk_widget_set_tooltip_text (appGUI->chart_button, _("Kana chart"));
 
     appGUI->prefs_button = gui_stock_label_button(NULL, KANATEST_STOCK_BUTTON_OPTIONS);
-    GTK_WIDGET_UNSET_FLAGS (appGUI->prefs_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (appGUI->prefs_button, FALSE);
     g_signal_connect (G_OBJECT (appGUI->prefs_button), "clicked",
                         G_CALLBACK (show_options_window_cb), appGUI);
     gtk_widget_show (appGUI->prefs_button);
@@ -588,7 +591,7 @@
     gtk_widget_set_tooltip_text (appGUI->prefs_button, _("Options"));
 
     appGUI->about_button = gui_stock_label_button(NULL, KANATEST_STOCK_BUTTON_ABOUT);
-    GTK_WIDGET_UNSET_FLAGS (appGUI->about_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (appGUI->about_button, FALSE);
     g_signal_connect (G_OBJECT (appGUI->about_button), "clicked",
                         G_CALLBACK (show_about_window_cb), appGUI);
     gtk_widget_show (appGUI->about_button);
@@ -660,33 +663,33 @@
     gtk_box_set_spacing (GTK_BOX (hbuttonbox), 5);
 
     appGUI->start_button = gui_stock_label_button(_("Start"), GTK_STOCK_EXECUTE);
-    GTK_WIDGET_UNSET_FLAGS (appGUI->start_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (appGUI->start_button, FALSE);
     g_signal_connect (G_OBJECT (appGUI->start_button), "clicked",
                         G_CALLBACK (start_test_cb), appGUI);
     gtk_widget_show (appGUI->start_button);
     gtk_container_add (GTK_CONTAINER (hbuttonbox), appGUI->start_button);
     gtk_container_set_border_width (GTK_CONTAINER (appGUI->start_button), 4);
-    GTK_WIDGET_SET_FLAGS (appGUI->start_button, GTK_CAN_DEFAULT);
+    gtk_widget_set_can_default (appGUI->start_button, TRUE);
     gtk_widget_set_tooltip_text (appGUI->start_button, _("Press to begin testing..."));
 
     appGUI->stop_button = gui_stock_label_button(_("Stop"), GTK_STOCK_STOP);
-    GTK_WIDGET_UNSET_FLAGS (appGUI->stop_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (appGUI->stop_button, FALSE);
     g_signal_connect (G_OBJECT (appGUI->stop_button), "clicked",
                         G_CALLBACK (stop_test_cb), appGUI);
     gtk_widget_show (appGUI->stop_button);
     gtk_container_add (GTK_CONTAINER (hbuttonbox), appGUI->stop_button);
     gtk_container_set_border_width (GTK_CONTAINER (appGUI->stop_button), 4);
-    GTK_WIDGET_SET_FLAGS (appGUI->stop_button, GTK_CAN_DEFAULT);
+    gtk_widget_set_can_default (appGUI->stop_button, TRUE);
     gtk_widget_set_tooltip_text (appGUI->stop_button, _("Press to stop testing..."));
 
     appGUI->quit_button = gui_stock_label_button(_("Quit"), GTK_STOCK_QUIT);
-    GTK_WIDGET_UNSET_FLAGS (appGUI->quit_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (appGUI->quit_button, FALSE);
     g_signal_connect (G_OBJECT (appGUI->quit_button), "clicked",
                         G_CALLBACK (gui_close_window_cb), appGUI);
     gtk_widget_show (appGUI->quit_button);
     gtk_container_add (GTK_CONTAINER (hbuttonbox), appGUI->quit_button);
     gtk_container_set_border_width (GTK_CONTAINER (appGUI->quit_button), 4);
-    GTK_WIDGET_SET_FLAGS (appGUI->quit_button, GTK_CAN_DEFAULT);
+    gtk_widget_set_can_default (appGUI->quit_button, TRUE);
     gtk_widget_set_tooltip_text (appGUI->quit_button, _("Exit!"));
 
     gui_disable_test (appGUI);
diff -ruN kanatest-0.4.8/src/options.c kanatest-0.4.8.1//src/options.c
--- kanatest-0.4.8/src/options.c	2009-04-13 01:10:45.000000000 +0600
+++ kanatest-0.4.8.1//src/options.c	2011-01-10 00:22:36.000000000 +0500
@@ -521,7 +521,7 @@
 
     appGUI->opt->hiragana_radiobutton = gtk_radio_button_new_with_mnemonic (NULL, _("Hiragana"));
     gtk_widget_show (appGUI->opt->hiragana_radiobutton);
-    GTK_WIDGET_UNSET_FLAGS (appGUI->opt->hiragana_radiobutton, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (appGUI->opt->hiragana_radiobutton, FALSE);
     gtk_box_pack_start (GTK_BOX (hbox1), appGUI->opt->hiragana_radiobutton, FALSE, FALSE, 0);
     gtk_radio_button_set_group (GTK_RADIO_BUTTON (appGUI->opt->hiragana_radiobutton), radiobutton_group);
     radiobutton_group = gtk_radio_button_get_group (GTK_RADIO_BUTTON (appGUI->opt->hiragana_radiobutton));
@@ -530,7 +530,7 @@
 
     katakana_radiobutton = gtk_radio_button_new_with_mnemonic (NULL, _("Katakana"));
     gtk_widget_show (katakana_radiobutton);
-    GTK_WIDGET_UNSET_FLAGS (katakana_radiobutton, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (katakana_radiobutton, FALSE);
     gtk_box_pack_start (GTK_BOX (hbox1), katakana_radiobutton, FALSE, FALSE, 0);
     gtk_radio_button_set_group (GTK_RADIO_BUTTON (katakana_radiobutton), radiobutton_group);
     radiobutton_group = gtk_radio_button_get_group (GTK_RADIO_BUTTON (katakana_radiobutton));
@@ -669,14 +669,14 @@
     g_signal_connect (G_OBJECT (cancel_button), "clicked",
                         G_CALLBACK (auto_select_close_button_cb), appGUI);
     gtk_container_add (GTK_CONTAINER (hbuttonbox), cancel_button);
-    GTK_WIDGET_SET_FLAGS (cancel_button, GTK_CAN_DEFAULT);
+    gtk_widget_set_can_default (cancel_button, TRUE);
 
     appGUI->opt->select_button = gui_stock_label_button(_("Select"), GTK_STOCK_INDEX);
     gtk_widget_show (appGUI->opt->select_button);
     g_signal_connect (G_OBJECT (appGUI->opt->select_button), "clicked",
                         G_CALLBACK (auto_select_select_button_cb), appGUI);
     gtk_container_add (GTK_CONTAINER (hbuttonbox), appGUI->opt->select_button);
-    GTK_WIDGET_SET_FLAGS (appGUI->opt->select_button, GTK_CAN_DEFAULT);
+    gtk_widget_set_can_default(appGUI->opt->select_button, TRUE);
 
     if (config.hiragana_mode == TRUE) {
         gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(appGUI->opt->hiragana_radiobutton), TRUE);
@@ -746,7 +748,7 @@
     gtk_container_add (GTK_CONTAINER (appGUI->opt->options_window), vbox1);
 
     appGUI->opt->notebook = gtk_notebook_new();
-    GTK_WIDGET_UNSET_FLAGS (appGUI->opt->notebook, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (appGUI->opt->notebook, FALSE);
     gtk_widget_show (appGUI->opt->notebook);
     gtk_notebook_set_tab_pos (GTK_NOTEBOOK(appGUI->opt->notebook), GTK_POS_TOP);
     gtk_box_pack_start (GTK_BOX (vbox1), appGUI->opt->notebook, TRUE, TRUE, 0);
@@ -823,7 +825,7 @@
                     (GtkAttachOptions) (0), 0, 0);
 
     combobox_ca_timeout = gtk_combo_box_new_text ();
-    GTK_WIDGET_UNSET_FLAGS (combobox_ca_timeout, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (combobox_ca_timeout, FALSE);
     gtk_widget_show (combobox_ca_timeout);
     g_signal_connect (G_OBJECT (combobox_ca_timeout), "changed",
                       G_CALLBACK (options_combobox_ca_timeout_handler_cb), NULL);
@@ -964,7 +966,7 @@
                 g_signal_connect (G_OBJECT (appGUI->opt->check_buttons[pos]), "toggled",
                                   G_CALLBACK (kana_selected_cb), &msg[pos]);
 
-                GTK_WIDGET_UNSET_FLAGS (appGUI->opt->check_buttons[pos], GTK_CAN_FOCUS);
+                gtk_widget_set_can_focus (appGUI->opt->check_buttons[pos], FALSE);
                 gtk_container_add (GTK_CONTAINER (frames[pos]), appGUI->opt->check_buttons[pos]);
 
                 if(config.user_defined_lesson[pos] == '+') {
@@ -980,7 +982,7 @@
 
         appGUI->opt->row_button_s[j] = gui_stock_label_button (NULL, GTK_STOCK_APPLY);
         gtk_button_set_relief (GTK_BUTTON (appGUI->opt->row_button_s[j]), GTK_RELIEF_NONE);
-        GTK_WIDGET_UNSET_FLAGS (appGUI->opt->row_button_s[j], GTK_CAN_FOCUS);
+        gtk_widget_set_can_focus (appGUI->opt->row_button_s[j], FALSE);
         gtk_widget_set_tooltip_text (appGUI->opt->row_button_s[j], _("Select all kanas in row"));
         gtk_widget_show (appGUI->opt->row_button_s[j]);
         gtk_table_attach (GTK_TABLE (table), appGUI->opt->row_button_s[j], i, i+1, j, j+1,
@@ -996,7 +998,7 @@
 
         appGUI->opt->row_button_c[j] = gui_stock_label_button (NULL, GTK_STOCK_CLEAR);
         gtk_button_set_relief (GTK_BUTTON (appGUI->opt->row_button_c[j]), GTK_RELIEF_NONE);
-        GTK_WIDGET_UNSET_FLAGS (appGUI->opt->row_button_c[j], GTK_CAN_FOCUS);
+        gtk_widget_set_can_focus (appGUI->opt->row_button_c[j], FALSE);
         gtk_widget_set_tooltip_text (appGUI->opt->row_button_c[j], _("Deselect all kanas in row"));
         gtk_widget_show (appGUI->opt->row_button_c[j]);
         gtk_table_attach (GTK_TABLE (table), appGUI->opt->row_button_c[j], i, i+1, j, j+1,
@@ -1026,7 +1028,7 @@
     gtk_misc_set_padding (GTK_MISC (appGUI->opt->number_of_kanas_label), 4, 0);
 
     auto_select_button = gtk_button_new_with_label (_("Statistics based selection"));
-    GTK_WIDGET_UNSET_FLAGS (auto_select_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (auto_select_button, FALSE);
     gtk_widget_show (auto_select_button);
     gtk_box_pack_end (GTK_BOX (hbox), auto_select_button, FALSE, FALSE, 0);
     g_signal_connect (G_OBJECT (auto_select_button), "clicked",
@@ -1044,7 +1046,7 @@
     gtk_misc_set_padding (GTK_MISC (label), 4, 0);
 
     combobox_dtf = gtk_combo_box_new_text ();
-    GTK_WIDGET_UNSET_FLAGS (combobox_dtf, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (combobox_dtf, FALSE);
     gtk_widget_show (combobox_dtf);
     gtk_box_pack_start (GTK_BOX (hbox), combobox_dtf, FALSE, FALSE, 0);
     g_signal_connect (G_OBJECT (combobox_dtf), "changed",
@@ -1068,21 +1070,21 @@
     gtk_misc_set_padding (GTK_MISC (label), 4, 0);
 
     select_all_button = gtk_button_new_with_label (_("All"));
-    GTK_WIDGET_UNSET_FLAGS (select_all_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (select_all_button, FALSE);
     gtk_widget_show (select_all_button);
     gtk_container_add (GTK_CONTAINER (hbuttonbox_s), select_all_button);
     g_signal_connect (G_OBJECT (select_all_button), "clicked",
                       G_CALLBACK (select_all_action_cb), appGUI);
 
     select_none_button = gtk_button_new_with_label (_("None"));
-    GTK_WIDGET_UNSET_FLAGS (select_none_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (select_none_button, FALSE);
     gtk_widget_show (select_none_button);
     gtk_container_add (GTK_CONTAINER (hbuttonbox_s), select_none_button);
     g_signal_connect (G_OBJECT (select_none_button), "clicked",
                       G_CALLBACK (select_none_action_cb), appGUI);
 
     invert_selection_button = gtk_button_new_with_label (_("Invert"));
-    GTK_WIDGET_UNSET_FLAGS (invert_selection_button, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus (invert_selection_button, FALSE);
     gtk_widget_show (select_none_button);
     gtk_widget_show (invert_selection_button);
     gtk_container_add (GTK_CONTAINER (hbuttonbox_s), invert_selection_button);
@@ -1103,7 +1105,7 @@
     g_signal_connect (G_OBJECT (close_button), "clicked",
                       G_CALLBACK (options_close_button_cb), appGUI);
     gtk_container_add (GTK_CONTAINER (hbuttonbox), close_button);
-    GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+    gtk_widget_set_can_default (close_button, TRUE);
 
     switch(config.repeat_mode) {
         case REPEAT_ALL:
diff -ruN kanatest-0.4.8/src/stats.c kanatest-0.4.8.1//src/stats.c
--- kanatest-0.4.8/src/stats.c	2009-04-13 01:10:45.000000000 +0600
+++ kanatest-0.4.8.1//src/stats.c	2011-01-10 00:24:16.000000000 +0500
@@ -405,7 +405,7 @@
     gtk_container_add (GTK_CONTAINER (appGUI->sts->stat_window), vbox1);
 
     appGUI->sts->notebook = gtk_notebook_new ();
-    GTK_WIDGET_UNSET_FLAGS(appGUI->sts->notebook, GTK_CAN_FOCUS);
+    gtk_widget_set_can_focus(appGUI->sts->notebook, FALSE);
     gtk_widget_show (appGUI->sts->notebook);
     gtk_box_pack_start (GTK_BOX (vbox1), appGUI->sts->notebook, TRUE, TRUE, 4);
 
@@ -638,7 +638,7 @@
     g_signal_connect (G_OBJECT (close_button), "clicked",
                         G_CALLBACK (stats_window_close_cb), appGUI);
     gtk_container_add (GTK_CONTAINER (hbuttonbox), close_button);
-    GTK_WIDGET_SET_FLAGS (close_button, GTK_CAN_DEFAULT);
+    gtk_widget_set_can_default (close_button, TRUE);
 
     if (appGUI->sts->active_tab != -1) {
         gtk_notebook_set_current_page (GTK_NOTEBOOK (appGUI->sts->notebook), appGUI->sts->active_tab);
