diff --git a/build.ninja b/build.ninja
index f9d6d02..1244004 100644
--- a/build.ninja
+++ b/build.ninja
@@ -1,7 +1,7 @@
 cc = gcc
 include .config
 ocamlflags = -warn-error +a -w +a -g -safe-string
-libs = -lmupdf $mujs -lpthread -L$mupdf/build/$buildtype
+libs = -lmupdf -lmujs -lpthread -L$mupdf/build/$buildtype
 
 cflags = $cflags -Wall -Werror -I$mupdf/include -I$freetype -g $
        -std=c99 -pedantic-errors -Wunused-parameter -Wsign-compare -Wshadow
diff --git a/configure.sh b/configure.sh
index 255d7d4..632b9d2 100644
--- a/configure.sh
+++ b/configure.sh
@@ -50,7 +50,7 @@ expr >/dev/null "$0" : "/.*" && {
 }
 builddir=$(cd $builddir >/dev/null $builddir && pwd -P)
 
-libs="$(pkg-config --libs $pkgs) -ljpeg -ljbig2dec -lopenjpeg"
+libs="$(pkg-config --libs $pkgs) -ljpeg -ljbig2dec -lopenjp2"
 
 test $ocamlfind && {
     lablgldir="$(ocamlfind query lablgl)" || exit 1
