diff --git a/build.ninja b/build.ninja
index 18efd63..f2e8955 100644
--- a/build.ninja
+++ b/build.ninja
@@ -31,7 +31,8 @@ build $builddir/link.o: cc $srcdir/link.c
 build $builddir/help.cmo: ocamlc $builddir/help.ml
 build $builddir/utils.cmo: ocamlc $srcdir/utils.ml
 build $builddir/bo.cmo: ocamlc $srcdir/le/bo.ml
-build $builddir/wsi.cmi: ocamlc $srcdir/wsi.mli
+build $builddir/wsi.cmi: ocamlc $srcdir/wsi.mli $
+      | $builddir/utils.cmo
 build $builddir/wsi.cmo: ocamlc $srcdir/wsi.ml $
       | $builddir/bo.cmo $builddir/utils.cmo $builddir/wsi.cmi
 build $builddir/parser.cmo: ocamlc $srcdir/parser.ml
