--- CMakeLists.txt.orig	2010-10-15 15:35:00.000000000 +0200
+++ CMakeLists.txt	2010-10-19 14:16:40.792071840 +0200
@@ -221,14 +221,14 @@
       set_config_option(HAVE_BLAS "Blas(IntelMKL)")
       set_config_option(HAVE_LAPACK "Lapack(IntelMKL)")
     else(LAPACK_LIBRARIES)
-      # on Linux also try to find ATLAS without a Fortran compiler,
+      # on Linux also try to find lapack without a Fortran compiler,
       # because cmake ships with a buggy FindBLAS e.g. on Ubuntu Lucid
       # Lynx
-      set(ATLAS_LIBS_REQUIRED lapack f77blas cblas atlas)
-      find_all_libraries(LAPACK_LIBRARIES ATLAS_LIBS_REQUIRED "" "")
+      set(DIST_LIBS_REQUIRED lapack blas dl)
+      find_all_libraries(LAPACK_LIBRARIES DIST_LIBS_REQUIRED "" "")
       if(LAPACK_LIBRARIES)
-        set_config_option(HAVE_BLAS "Blas(ATLAS)")
-        set_config_option(HAVE_LAPACK "Lapack(ATLAS)")
+         set_config_option(HAVE_BLAS "Blas(DIST)")
+         set_config_option(HAVE_LAPACK "Lapack(DIST)")
       endif(LAPACK_LIBRARIES)
     endif(LAPACK_LIBRARIES)
   elseif(${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
