--- Geo/OCC_Connect.cpp.orig	2011-03-16 22:20:30.765010597 +0100
+++ Geo/OCC_Connect.cpp	2011-03-16 22:20:39.819118690 +0100
@@ -597,7 +597,7 @@
             double tolerance=std::max(BRep_Tool::Tolerance(edge),
                                       BRep_Tool::Tolerance(vertex));
             for(int i=1;i<=distance.NbExt();i++) {
-                if(distance.Value(i)<tolerance) {
+                if(distance.SquareDistance(i)<tolerance) {
                     try {
                         // No idea why this can fail
                         splitter1.Add(vertex,distance.Parameter(i),edge);
@@ -616,7 +616,7 @@
                             double d2=BRep_Tool::Pnt(v2).Distance(
                                 BRep_Tool::Pnt(vertex));
                             cout << "Adding " << i << " to edge " << e
-                                << " distance=" << distance.Value(i)
+                                << " distance=" << distance.SquareDistance(i)
                                 << " parameter=" << distance.Parameter(i)
                                 << " point=" << distance.Point(i)
                                 << " dv1=" << d1
@@ -649,7 +649,7 @@
             double tolerance=std::max(BRep_Tool::Tolerance(edge),
                                       BRep_Tool::Tolerance(vertex));
             for(int i=1;i<=distance.NbExt();i++) {
-                if(distance.Value(i)<tolerance) {
+                if(distance.SquareDistance(i)<tolerance) {
                     try {
                         splitter2.Add(vertex,distance.Parameter(i),edge);
                     }
@@ -667,7 +667,7 @@
                             double d2=BRep_Tool::Pnt(v2).Distance(
                                 BRep_Tool::Pnt(vertex));
                             cout << "Adding " << i << " to edge " << e
-                                << " distance=" << distance.Value(i)
+                                << " distance=" << distance.SquareDistance(i)
                                 << " parameter=" << distance.Parameter(i)
                                 << " point=" << distance.Point(i)
                                 << " dv1=" << d1
