--- /home/schmitte/CGoGN/SCHNApps/src/window.cpp	2013-04-02 13:25:58.906370207 +0200
+++ windows2.cpp	2013-04-02 15:06:24.206104942 +0200
@@ -90,7 +90,7 @@
 	connect(actionManagePlugins, SIGNAL(triggered()), this, SLOT(cb_showPluginsDialog()));
 	connect(actionManageMaps, SIGNAL(triggered()), this, SLOT(cb_showMapsDialog()));
 
-	registerPluginsDirectory(m_appPath + QString("/../lib"));
+	registerPluginsDirectory(QString("/usr/lib/SCHNApps"));
 }
 
 Window::~Window()
