diff -ruNa nixnote2/communication/communicationmanager.cpp nixnote2-qt5/communication/communicationmanager.cpp
--- nixnote2/communication/communicationmanager.cpp	2014-12-27 19:31:07.957392031 +0800
+++ nixnote2-qt5/communication/communicationmanager.cpp	2014-12-31 10:23:04.000000000 +0800
@@ -40,7 +40,8 @@
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <execinfo.h>
-
+//fix
+#include <QUrlQuery>
 extern Global global;
 
 
@@ -53,7 +54,8 @@
     clientName = "NixNote/Linux";
     inkNoteList = new QList< QPair<QString, QImage*>* >();
     thumbnailList = new QList< QPair<QString, QImage*>* >();
-    postData = new QUrl();
+    //postData = new QUrl();
+    postData = new QUrlQuery();
     tagGuidMap = new QHash<QString,QString>;
     initComplete = false;
     noteStore = NULL;
@@ -887,7 +889,9 @@
     size.setWidth(r->width);
     postData->clear();
     postData->addQueryItem("auth", authToken);
-
+    
+    
+    
     QEventLoop loop;
     QObject::connect(networkAccessManager, SIGNAL(finished(QNetworkReply*)), &loop, SLOT(quit()));
 
@@ -899,8 +903,8 @@
         QNetworkRequest request(url);
 
         request.setHeader(QNetworkRequest::ContentTypeHeader, "application/x-www-form-urlencoded");
-        QNetworkReply *reply = networkAccessManager->post(request,postData->encodedQuery());
-
+        QNetworkReply *reply = networkAccessManager->post(request,postData->query(QUrl::FullyEncoded).toUtf8());
+        
         // Execute the event loop here, now we will wait here until readyRead() signal is emitted
         // which in turn will trigger event loop quit.
         loop.exec();
diff -ruNa nixnote2/communication/communicationmanager.h nixnote2-qt5/communication/communicationmanager.h
--- nixnote2/communication/communicationmanager.h	2014-12-27 19:31:07.957392031 +0800
+++ nixnote2-qt5/communication/communicationmanager.h	2014-12-31 10:21:07.000000000 +0800
@@ -68,7 +68,8 @@
     bool initComplete;
     QSqlDatabase *db;
     bool initNoteStore();
-    QUrl *postData;
+    //QUrl *postData;
+    QUrlQuery *postData;
 
     QString userStorePath;
     QString noteStorePath;
diff -ruNa nixnote2/dialog/logviewer.cpp nixnote2-qt5/dialog/logviewer.cpp
--- nixnote2/dialog/logviewer.cpp	2014-12-27 19:31:07.977392031 +0800
+++ nixnote2-qt5/dialog/logviewer.cpp	2014-12-31 11:03:23.000000000 +0800
@@ -63,7 +63,7 @@
     fd.setFileMode(QFileDialog::AnyFile);
     fd.setConfirmOverwrite(true);
     fd.setWindowTitle(tr("Save Log"));
-    fd.setFilter(tr("NixNote Log (*.log);;All Files (*.*)"));
+    fd.setNameFilter(tr("NixNote Log (*.log);;All Files (*.*)"));
     fd.setAcceptMode(QFileDialog::AcceptSave);
         fd.setDirectory(QDir::homePath());
     if (fd.exec() == 0 || fd.selectedFiles().size() == 0) {
diff -ruNa nixnote2/dialog/screencapture.cpp nixnote2-qt5/dialog/screencapture.cpp
--- nixnote2/dialog/screencapture.cpp	2014-12-27 19:31:07.978392031 +0800
+++ nixnote2-qt5/dialog/screencapture.cpp	2014-12-31 10:44:48.000000000 +0800
@@ -22,8 +22,8 @@
 
 #include "screencapture.h"
 
-#include <QtGui/QDesktopWidget>
-#include <QtGui/QApplication>
+#include <QDesktopWidget>
+#include <QApplication>
 
 
 ScreenCapture::ScreenCapture(QWidget *parent) :
@@ -108,7 +108,9 @@
     QRect txtRect = QApplication::desktop()->screenGeometry(QApplication::desktop()->primaryScreen());
     QString txtTip = QApplication::tr("Use your mouse to draw a rectangle to screenshot or exit pressing\nany key or using the right or middle mouse buttons.");
 
-    txtRect.setHeight(qRound(txtRect.height() / 10)); // rounded val of text rect height
+    //txtRect.setHeight(qRound(txtRect.height() / 10)); // rounded val of text rect height
+    //fix,remove qround
+    txtRect.setHeight(txtRect.height() / 10);
 
     painter.setPen(QPen(Qt::red)); // ste message rect border color
     painter.setBrush(QBrush(QColor(255, 255, 255, 180), Qt::SolidPattern));
diff -ruNa nixnote2/dialog/screencapture.h nixnote2-qt5/dialog/screencapture.h
--- nixnote2/dialog/screencapture.h	2014-12-27 19:31:07.978392031 +0800
+++ nixnote2-qt5/dialog/screencapture.h	2014-12-29 10:36:49.749052889 +0800
@@ -24,7 +24,7 @@
 #define SCREENCAPTURE_H
 
 #include <QDialog>
-#include <QtGui/QDialog>
+#include <QtWidgets/QDialog>
 #include <QtGui/QMouseEvent>
 #include <QtGui/QPainter>
 #include <QtGui/QPixmap>
diff -ruNa nixnote2/gui/favoritesview.cpp nixnote2-qt5/gui/favoritesview.cpp
--- nixnote2/gui/favoritesview.cpp	2014-12-27 19:31:07.983392030 +0800
+++ nixnote2-qt5/gui/favoritesview.cpp	2014-12-31 11:04:48.000000000 +0800
@@ -34,6 +34,10 @@
 #include "sql/sharednotebooktable.h"
 #include <QMessageBox>
 
+//fix
+#include <QDrag>
+#include <QDragEnterEvent>
+
 extern Global global;
 
 #define NAME_POSITION 0
diff -ruNa nixnote2/gui/nbrowserwindow.cpp nixnote2-qt5/gui/nbrowserwindow.cpp
--- nixnote2/gui/nbrowserwindow.cpp	2014-12-27 19:31:08.001392030 +0800
+++ nixnote2-qt5/gui/nbrowserwindow.cpp	2014-12-31 09:37:01.000000000 +0800
@@ -61,6 +61,9 @@
 #include <QPrinterInfo>
 #include <iostream>
 #include <istream>
+//fix
+#include <QtWebKitWidgets>
+#include <QWebPage>
 
 extern Global global;
 
diff -ruNa nixnote2/gui/nnotebookview.cpp nixnote2-qt5/gui/nnotebookview.cpp
--- nixnote2/gui/nnotebookview.cpp	2014-12-27 19:31:08.002392030 +0800
+++ nixnote2-qt5/gui/nnotebookview.cpp	2014-12-31 09:21:37.000000000 +0800
@@ -28,11 +28,12 @@
 #include <QMessageBox>
 #include <QTextDocument>
 #include <QFontMetrics>
-
-#include <QWindowsStyle>
-#include <QMotifStyle>
-#include <QPlastiqueStyle>
-#include <QCleanlooksStyle>
+//Clean up removed style
+//#include <QWindowsStyle>
+#include <QStyle>
+//#include <QMotifStyle>
+//#include <QPlastiqueStyle>
+//#include <QCleanlooksStyle>
 
 #include "sql/notebooktable.h"
 #include "sql/linkednotebooktable.h"
@@ -42,6 +43,10 @@
 #include "gui/nnotebookviewdelegate.h"
 #include "sql/nsqlquery.h"
 
+//fix
+#include <QDrag>
+#include <QDragEnterEvent>
+
 #define NAME_POSITION 0
 
 extern Global global;
diff -ruNa nixnote2/gui/nsearchview.cpp nixnote2-qt5/gui/nsearchview.cpp
--- nixnote2/gui/nsearchview.cpp	2014-12-27 19:31:08.002392030 +0800
+++ nixnote2-qt5/gui/nsearchview.cpp	2014-12-31 09:07:31.000000000 +0800
@@ -32,6 +32,10 @@
 #include <QMessageBox>
 #include <QPainter>
 
+//fix
+#include <QDrag>
+#include <QDragEnterEvent>
+
 #define NAME_POSITION 0
 
 extern Global global;
diff -ruNa nixnote2/gui/ntableview.cpp nixnote2-qt5/gui/ntableview.cpp
--- nixnote2/gui/ntableview.cpp	2014-12-27 19:31:08.003392029 +0800
+++ nixnote2-qt5/gui/ntableview.cpp	2014-12-31 12:51:33.000000000 +0800
@@ -34,6 +34,10 @@
 #include "sql/notetable.h"
 #include "sql/notebooktable.h"
 
+//fix
+#include <QDrag>
+#include <QDragEnterEvent>
+
 //*****************************************************************
 //* This class overrides QTableView and is used to provide a
 //* list of notes to the user
@@ -54,7 +58,8 @@
 
     tableViewHeader = new NTableViewHeader(Qt::Horizontal, this);
     this->setHorizontalHeader(tableViewHeader);
-    this->horizontalHeader()->setMovable(true);
+    //this->horizontalHeader()->setMovable(true);
+    this->horizontalHeader()->setSectionsMovable(true);
 
     QLOG_TRACE() << "Setting up edit triggers";
     this->setEditTriggers(QAbstractItemView::NoEditTriggers);
diff -ruNa nixnote2/gui/ntableviewheader.cpp nixnote2-qt5/gui/ntableviewheader.cpp
--- nixnote2/gui/ntableviewheader.cpp	2014-12-27 19:31:08.003392029 +0800
+++ nixnote2-qt5/gui/ntableviewheader.cpp	2014-12-31 12:53:09.000000000 +0800
@@ -30,7 +30,8 @@
     QHeaderView(orientation, parent)
 {
 
-    setClickable(true);
+    //setClickable(true);
+    setSectionsClickable(true);
     setSortIndicatorShown(true);
     setContextMenuPolicy(Qt::ActionsContextMenu);
 
diff -ruNa nixnote2/gui/ntagview.cpp nixnote2-qt5/gui/ntagview.cpp
--- nixnote2/gui/ntagview.cpp	2014-12-27 19:31:08.003392029 +0800
+++ nixnote2-qt5/gui/ntagview.cpp	2014-12-31 09:08:10.000000000 +0800
@@ -36,6 +36,10 @@
 #include <QMessageBox>
 #include <QPainter>
 
+//fix
+#include <QDrag>
+#include <QDragEnterEvent>
+
 #define NAME_POSITION 0
 
 extern Global global;
diff -ruNa nixnote2/gui/plugins/pluginfactory.h nixnote2-qt5/gui/plugins/pluginfactory.h
--- nixnote2/gui/plugins/pluginfactory.h	2014-12-27 19:31:08.004392029 +0800
+++ nixnote2-qt5/gui/plugins/pluginfactory.h	2014-12-31 11:19:49.000000000 +0800
@@ -22,7 +22,7 @@
 
 #include <QWebPluginFactory>
 #include <QNetworkAccessManager>
-#include <poppler-qt4.h>
+#include <poppler-qt5.h>
 #include "gui/plugins/popplerviewer.h"
 #include <QWebPage>
 
diff -ruNa nixnote2/gui/plugins/popplerviewer.cpp nixnote2-qt5/gui/plugins/popplerviewer.cpp
--- nixnote2/gui/plugins/popplerviewer.cpp	2014-12-27 19:31:08.004392029 +0800
+++ nixnote2-qt5/gui/plugins/popplerviewer.cpp	2014-12-31 11:30:33.000000000 +0800
@@ -21,7 +21,7 @@
 #include <QGridLayout>
 #include <QHBoxLayout>
 #include <QLabel>
-#include <poppler-qt4.h>
+#include <poppler-qt5.h>
 #include <QGraphicsPixmapItem>
 #include <QImage>
 #include <QPushButton>
diff -ruNa nixnote2/gui/plugins/popplerviewer.h nixnote2-qt5/gui/plugins/popplerviewer.h
--- nixnote2/gui/plugins/popplerviewer.h	2014-12-27 19:31:08.004392029 +0800
+++ nixnote2-qt5/gui/plugins/popplerviewer.h	2014-12-31 11:20:05.000000000 +0800
@@ -28,7 +28,7 @@
 #include <QLabel>
 #include <QGraphicsScene>
 #include <QGraphicsView>
-#include <poppler-qt4.h>
+#include <poppler-qt5.h>
 
 #include "gui/plugins/popplergraphicsview.h"
 
diff -ruNa nixnote2/html/enmlformatter.cpp nixnote2-qt5/html/enmlformatter.cpp
--- nixnote2/html/enmlformatter.cpp	2014-12-27 19:31:08.008392029 +0800
+++ nixnote2-qt5/html/enmlformatter.cpp	2014-12-31 09:38:08.000000000 +0800
@@ -26,6 +26,9 @@
 #include <QFileIconProvider>
 #include <QIcon>
 #include <QMessageBox>
+//fix
+#include <QtWebKitWidgets>
+#include <QWebPage>
 
 #include <iostream>
 using namespace std;
diff -ruNa nixnote2/html/noteformatter.cpp nixnote2-qt5/html/noteformatter.cpp
--- nixnote2/html/noteformatter.cpp	2014-12-27 19:31:08.008392029 +0800
+++ nixnote2-qt5/html/noteformatter.cpp	2014-12-31 11:23:54.000000000 +0800
@@ -29,12 +29,14 @@
 
 #include <QFileSystemModel>
 #include <QFileIconProvider>
-#include <poppler-qt4.h>
+#include <poppler-qt5.h>
 #include <QIcon>
 #include <boost/property_tree/xml_parser.hpp>
 #include <boost/property_tree/ptree.hpp>
 #include <QList>
-
+//fix
+#include <QtWebKitWidgets>
+#include <QWebPage>
 
 #include <iostream>
 using namespace std;
diff -ruNa nixnote2/html/thumbnailer.cpp nixnote2-qt5/html/thumbnailer.cpp
--- nixnote2/html/thumbnailer.cpp	2014-12-27 19:31:08.009392029 +0800
+++ nixnote2-qt5/html/thumbnailer.cpp	2014-12-31 09:34:57.000000000 +0800
@@ -18,12 +18,14 @@
 ***********************************************************************************/
 
 #include "thumbnailer.h"
-#include <QtWebKit>
+//#include <QtWebKit>
+#include <QtWebKitWidgets>
+#include <QWebPage>
+
 #include <QtSql>
 #include <QTextDocument>
 #include "global.h"
 #include "sql/notetable.h"
-
 extern Global global;
 
 /* Generic constructor. */
diff -ruNa nixnote2/main.cpp nixnote2-qt5/main.cpp
--- nixnote2/main.cpp	2014-12-27 19:31:08.033392028 +0800
+++ nixnote2-qt5/main.cpp	2014-12-29 10:22:54.459075965 +0800
@@ -18,7 +18,8 @@
 ***********************************************************************************/
 
 
-#include <QtGui/QApplication>
+//#include <QtGui/QApplication>
+#include <QtWidgets/QApplication>
 #include "nixnote.h"
 #include "global.h"
 #include "settings/startupconfig.h"
diff -ruNa nixnote2/NixNote2.pro nixnote2-qt5/NixNote2.pro
--- nixnote2/NixNote2.pro	2014-12-27 19:31:07.957392031 +0800
+++ nixnote2-qt5/NixNote2.pro	2014-12-31 12:21:31.000000000 +0800
@@ -4,7 +4,7 @@
 #
 #-------------------------------------------------
 
-QT       += core gui webkit sql network xml
+QT       += core gui webkit sql network xml webkitwidgets printsupport
 
 TARGET = nixnote2
 TEMPLATE = app
@@ -341,16 +341,17 @@
 
 #INCLUDEPATH += /usr/local/include/thrift \
 #            /usr/include/thrift \
-INCLUDEPATH += /usr/include/poppler/qt4
+INCLUDEPATH += /usr/include/poppler/qt5
 
 #LIBS +=    -Wl,-L./lib -lthrift \
 #LIBS +=    -lthrift \
 LIBS +=    -lopencv_core -lopencv_highgui -lopencv_imgproc \
            -lhunspell \
-           -lpthread -L/usr/lib -lpoppler-qt4 -g -rdynamic
+           -lpthread -L/usr/lib -lpoppler-qt5 -g -rdynamic
 #           -Wl,-rpath=/usr/lib/nixnote2
 
 #QMAKE_CXXFLAGS += `dpkg-buildflags --get CFLAGS`
 QMAKE_CXXFLAGS +=-g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security
 #QMAKE_LFLAGS += `dpkg-buildflags --get LDFLAGS`
 QMAKE_LFLAGS += -Wl,-Bsymbolic-functions -Wl,-z,relro
+#Release:QMAKE_POST_LINK += strip --strip-unneeded $(DESTDIR_TARGET)
diff -ruNa nixnote2/nixnote.cpp nixnote2-qt5/nixnote.cpp
--- nixnote2/nixnote.cpp	2014-12-27 19:31:08.034392028 +0800
+++ nixnote2-qt5/nixnote.cpp	2014-12-31 12:48:10.000000000 +0800
@@ -101,7 +101,8 @@
     global.settings->endGroup();
 
 
-    QTextCodec::setCodecForCStrings(QTextCodec::codecForName("UTF-8"));
+    //QTextCodec::setCodecForCStrings(QTextCodec::codecForName("UTF-8"));
+    QTextCodec::setCodecForLocale(QTextCodec::codecForName("UTF-8"));
     this->setDebugLevel();
 
     QTranslator *nixnoteTranslator = new QTranslator();
@@ -1394,7 +1395,7 @@
         fd.setWindowTitle(tr("Backup Database"));
     else
         fd.setWindowTitle(tr("Export Notes"));
-    fd.setFilter(tr("NixNote Export (*.nnex);;All Files (*.*)"));
+    fd.setNameFilter(tr("NixNote Export (*.nnex);;All Files (*.*)"));
     fd.setAcceptMode(QFileDialog::AcceptSave);
     if (saveLastPath == "")
         fd.setDirectory(QDir::homePath());
@@ -1484,9 +1485,9 @@
         fd.setWindowTitle(tr("Import Notes"));
 
     if (fullRestore) {
-        fd.setFilter(tr("NixNote Export (*.nnex);;All Files (*.*)"));
+        fd.setNameFilter(tr("NixNote Export (*.nnex);;All Files (*.*)"));
     } else {
-        fd.setFilter(tr("NixNote Export (*.nnex);;Evernote Export (*.enex);;All Files (*.*)"));
+        fd.setNameFilter(tr("NixNote Export (*.nnex);;Evernote Export (*.enex);;All Files (*.*)"));
     }
     fd.setAcceptMode(QFileDialog::AcceptOpen);
     if (saveLastPath == "")
@@ -1635,7 +1636,8 @@
     NotebookTable notebookTable(global.db);
     n.content = newNoteBody;
     n.title = "Untitled note";
-    QString uuid = QUuid::createUuid();
+    //QString uuid = QUuid::createUuid();
+    QString uuid = QUuid::createUuid().toString();
     uuid = uuid.mid(1);
     uuid.chop(1);
     n.guid = uuid;
@@ -1675,7 +1677,8 @@
     NotebookTable notebookTable(global.db);
     n.content = newNoteBody;
     n.title = "Untitled note";
-    QString uuid = QUuid::createUuid();
+    //QString uuid = QUuid::createUuid();
+    QString uuid = QUuid::createUuid().toString();
     uuid = uuid.mid(1);
     uuid.chop(1);
     n.guid = uuid;
diff -ruNa nixnote2/nixnote.h nixnote2-qt5/nixnote.h
--- nixnote2/nixnote.h	2014-12-27 19:31:08.035392028 +0800
+++ nixnote2-qt5/nixnote.h	2014-12-29 10:23:36.859074760 +0800
@@ -20,7 +20,8 @@
 #ifndef NIXNOTE_H
 #define NIXNOTE_H
 
-#include <QtGui/QMainWindow>
+//#include <QtGui/QMainWindow>
+#include <QtWidgets/QMainWindow>
 #include <QVBoxLayout>
 #include <QHBoxLayout>
 #include <QSplitter>
diff -ruNa nixnote2/oauth/oauthwindow.cpp nixnote2-qt5/oauth/oauthwindow.cpp
--- nixnote2/oauth/oauthwindow.cpp	2014-12-27 19:31:08.035392028 +0800
+++ nixnote2-qt5/oauth/oauthwindow.cpp	2014-12-31 09:39:11.000000000 +0800
@@ -87,7 +87,7 @@
 
     // Turn on TLS (sometimes it isn't on by default)
     QSslConfiguration config = QSslConfiguration::defaultConfiguration();
-    config.setProtocol(QSsl::TlsV1);
+    config.setProtocol(QSsl::TlsV1_0);
     config.setProtocol(QSsl::SslV3);
     QSslConfiguration::setDefaultConfiguration(config);
 
diff -ruNa nixnote2/sql/notetable.cpp nixnote2-qt5/sql/notetable.cpp
--- nixnote2/sql/notetable.cpp	2014-12-27 19:31:08.075392026 +0800
+++ nixnote2-qt5/sql/notetable.cpp	2014-12-31 09:23:55.000000000 +0800
@@ -175,7 +175,7 @@
         query.bindValue(":key", NOTE_CONTENT);
         QByteArray b;
         QString content = t.content;
-        b.append(content.toAscii());
+        b.append(content.toLatin1());
         query.bindValue(":data", b);
         query.exec();
     }
diff -ruNa nixnote2/sql/usertable.h nixnote2-qt5/sql/usertable.h
--- nixnote2/sql/usertable.h	2014-12-27 19:31:08.077392026 +0800
+++ nixnote2-qt5/sql/usertable.h	2014-12-29 10:46:07.979037466 +0800
@@ -19,7 +19,8 @@
 
 #ifndef USERTABLE_H
 #define USERTABLE_H
-#include <Q_LONG>
+//#include <Q_LONG>
+#include "qglobal.h"
 #include <QSqlDatabase>
 #include <qevercloud/include/QEverCloud.h>
 
diff -ruNa nixnote2/threads/indexrunner.cpp nixnote2-qt5/threads/indexrunner.cpp
--- nixnote2/threads/indexrunner.cpp	2014-12-27 19:31:08.078392026 +0800
+++ nixnote2-qt5/threads/indexrunner.cpp	2014-12-31 11:27:37.000000000 +0800
@@ -24,7 +24,7 @@
 #include "sql/resourcetable.h"
 #include <QTextDocument>
 #include <QtXml>
-#include <poppler-qt4.h>
+#include <poppler-qt5.h>
 
 extern Global global;
 using namespace Poppler;
diff -ruNa nixnote2/xml/exportdata.cpp nixnote2-qt5/xml/exportdata.cpp
--- nixnote2/xml/exportdata.cpp	2014-12-27 19:31:08.100392025 +0800
+++ nixnote2-qt5/xml/exportdata.cpp	2014-12-31 10:54:43.000000000 +0800
@@ -651,14 +651,14 @@
 
 
 
-void ExportData::createNode(QString nodeName, QBool value) {
+/*void ExportData::createNode(QString nodeName, bool value) {
     if (value)
         writer->writeTextElement(nodeName, "true");
     else
         writer->writeTextElement(nodeName, "false");
     return;
 }
-
+*/
 
 
 
diff -ruNa nixnote2/xml/exportdata.h nixnote2-qt5/xml/exportdata.h
--- nixnote2/xml/exportdata.h	2014-12-27 19:31:08.100392025 +0800
+++ nixnote2-qt5/xml/exportdata.h	2014-12-29 10:59:40.892348362 +0800
@@ -48,7 +48,7 @@
     void createBinaryNode(QString nodeName, string value);
     void createTimestampNode(QString nodeName, Timestamp value);
     void createNode(QString nodeName, bool value);
-    void createNode(QString nodeName, QBool value);
+//    void createNode(QString nodeName, QBool value);
     void createNode(QString nodeName, double value);
     void createNode(QString nodename, int64_t value);
     void writeTags();
diff -ruNa nixnote2/xml/xmlhighlighter.cpp nixnote2-qt5/xml/xmlhighlighter.cpp
--- nixnote2/xml/xmlhighlighter.cpp	2014-12-27 19:31:08.101392025 +0800
+++ nixnote2-qt5/xml/xmlhighlighter.cpp	2014-12-31 10:25:37.000000000 +0800
@@ -136,7 +136,7 @@
 
     for (; i < text.length(); i++)
     {
-        switch (text.at(i).toAscii())
+        switch (text.at(i).toLatin1())
         {
         case '<':
             brackets++;
diff -ruNa nixnote2/xml/xmlhighlighter.h nixnote2-qt5/xml/xmlhighlighter.h
--- nixnote2/xml/xmlhighlighter.h	2014-12-27 19:31:08.101392025 +0800
+++ nixnote2-qt5/xml/xmlhighlighter.h	2014-12-31 10:38:37.000000000 +0800
@@ -19,6 +19,8 @@
 #include <QSyntaxHighlighter>
 #include <QTextCharFormat>
 #include <QColor>
+//fix
+#include <QTextEdit>
 
 class XmlHighlighter : public QSyntaxHighlighter
 {
