--- app/fingerprinter/LAV_Source.cpp	2013-10-10 23:05:06.357827166 +0200
+++ app/fingerprinter/LAV_Source.cpp	2013-10-10 23:05:37.918679186 +0200
@@ -48,10 +48,11 @@
 /*
    Don't change these values!
    The fingerprinter expects signed 16 bit pcm w/ 1 or 2 channels
-*/
+*/ 
 const AVSampleFormat outSampleFmt = AV_SAMPLE_FMT_S16;
 const int outSampleSize           = av_get_bytes_per_sample(outSampleFmt);
 const int outMaxChannels          = 2;
+const int AVCODEC_MAX_AUDIO_FRAME_SIZE = 192000;
 
 
 class LAV_SourcePrivate
