diff -wruiN libe-0.2.7.old/benchmarks/lockfree_hash_map.cc libe-0.2.7/benchmarks/lockfree_hash_map.cc
--- libe-0.2.7.old/benchmarks/lockfree_hash_map.cc	2012-08-14 11:14:53.349971316 +0200
+++ libe-0.2.7/benchmarks/lockfree_hash_map.cc	2012-08-14 11:15:10.589971117 +0200
@@ -30,6 +30,7 @@
 // C includes
 #include <cstdlib>
 #include <stdint.h>
+#include <unistd.h>
 
 // C++
 #include <iostream>
diff -wruiN libe-0.2.7.old/benchmarks/lockfree_hash_set.cc libe-0.2.7/benchmarks/lockfree_hash_set.cc
--- libe-0.2.7.old/benchmarks/lockfree_hash_set.cc	2012-08-14 11:14:53.349971316 +0200
+++ libe-0.2.7/benchmarks/lockfree_hash_set.cc	2012-08-14 11:15:18.323304363 +0200
@@ -30,6 +30,7 @@
 // C includes
 #include <cstdlib>
 #include <stdint.h>
+#include <unistd.h>
 
 // C++
 #include <iostream>
