--- pkg/sdk/CUDALibraries/src/imageSegmentationNPP/imageSegmentationNPP.cpp.orig	2011-03-23 10:56:12.230494279 +0100
+++ pkg/sdk/CUDALibraries/src/imageSegmentationNPP/imageSegmentationNPP.cpp	2011-03-23 10:57:42.932181885 +0100
@@ -141,7 +141,7 @@
         // if more than one command line arg, use the first arg as the filename,
         // otherwise assume the filename included with the sample
         std::string sFilename[2] = { "../../data/person.txt",
-                                     "../../../data/person.txt" };
+                                     "data/person.txt" };
 
 		parseCommandLineArguments(argc, argv);
 		if (g_bQATest == false && argc > 1) {
--- pkg/sdk/CUDALibraries/src/freeImageInteropNPP/freeImageInteropNPP.cpp.orig	2011-03-23 10:56:12.227160638 +0100
+++ pkg/sdk/CUDALibraries/src/freeImageInteropNPP/freeImageInteropNPP.cpp	2011-03-23 10:56:48.140467584 +0100
@@ -124,7 +124,7 @@
         // if more than one command line arg, use the first arg as the filename,
         // otherwise assume the filename included with the sample
         std::string sFilename[2] = { "../../data/Lena.pgm",
-                                     "../../../data/Lena.pgm" };
+                                     "data/Lena.pgm" };
 		parseCommandLineArguments(argc, argv);
 		if (g_bQATest == false && argc > 1) {
 			sFilename[0] = argv[1];
--- pkg/sdk/CUDALibraries/src/histEqualizationNPP/histEqualizationNPP.cpp.orig	2011-03-23 10:56:12.230494279 +0100
+++ pkg/sdk/CUDALibraries/src/histEqualizationNPP/histEqualizationNPP.cpp	2011-03-23 10:57:53.016444527 +0100
@@ -40,7 +40,7 @@
         // if more than one command line arg, use the first arg as the filename,
         // otherwise assume the filename included with the sample
         std::string sFilename[2] = { "../../data/Lena.pgm",
-                                     "../../../data/Lena.pgm" };
+                                     "data/Lena.pgm" };
 		parseCommandLineArguments(argc, argv);
 		if (g_bQATest == false && argc > 1) {
 			sFilename[0] = argv[1];
--- pkg/sdk/CUDALibraries/src/boxFilterNPP/boxFilterNPP.cpp.orig	2011-03-23 10:56:12.227160638 +0100
+++ pkg/sdk/CUDALibraries/src/boxFilterNPP/boxFilterNPP.cpp	2011-03-23 10:56:31.565607863 +0100
@@ -39,7 +39,7 @@
         // if more than one command line arg, use the first arg as the filename,
         // otherwise assume the filename included with the sample
         std::string sFilename[2] = { "../../data/Lena.pgm",
-                                     "../../../data/Lena.pgm" };
+                                     "data/Lena.pgm" };
 		parseCommandLineArguments(argc, argv);
 		if (g_bQATest == false && argc > 1) {
 			sFilename[0] = argv[1];
--- pkg/sdk/shared/src/shrUtils.cpp.orig	2011-03-23 11:13:29.189765267 +0100
+++ pkg/sdk/shared/src/shrUtils.cpp	2011-03-23 11:14:14.263955003 +0100
@@ -588,6 +588,10 @@
         "../data/",                                 // up 1 in tree, "/data/" subdir 
         "../src/",                                  // up 1 in tree, "/src/" subdir 
         "../inc/",                                  // up 1 in tree, "/inc/" subdir 
+        "../src/<executable_name>/",                // up 1 in tree, "/src/<executable_name>/" subdir 
+        "../src/<executable_name>/data/",           // up 1 in tree, "/src/<executable_name>/data/" subdir 
+        "../src/<executable_name>/src/",            // up 1 in tree, "/src/<executable_name>/src/" subdir 
+        "../OpenCL/src/<executable_name>/inc/",     // up 1 in tree, "/OpenCL/src/<executable_name>/inc/" subdir 
         "../OpenCL/src/<executable_name>/",         // up 1 in tree, "/OpenCL/src/<executable_name>/" subdir 
         "../OpenCL/src/<executable_name>/data/",    // up 1 in tree, "/OpenCL/src/<executable_name>/data/" subdir 
         "../OpenCL/src/<executable_name>/src/",     // up 1 in tree, "/OpenCL/src/<executable_name>/src/" subdir 
