diff -aur monopd-0.9.3.orig/src/cardgroup.cpp monopd-0.9.3.new/src/cardgroup.cpp
--- monopd-0.9.3.orig/src/cardgroup.cpp	2008-12-28 22:24:55.000000000 +0100
+++ monopd-0.9.3.new/src/cardgroup.cpp	2008-12-28 22:37:36.000000000 +0100
@@ -14,7 +14,7 @@
 // the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 // Boston, MA 02111-1307, USA.
 
-#include <algo.h>
+#include <algorithm>
 
 #include "card.h"
 #include "cardgroup.h"
diff -aur monopd-0.9.3.orig/src/estategroup.cpp monopd-0.9.3.new/src/estategroup.cpp
--- monopd-0.9.3.orig/src/estategroup.cpp	2008-12-28 22:24:55.000000000 +0100
+++ monopd-0.9.3.new/src/estategroup.cpp	2008-12-28 22:38:37.000000000 +0100
@@ -15,6 +15,7 @@
 // Boston, MA 02111-1307, USA.
 
 #include <iostream>
+#include <cstdlib>
 
 #include <math++/nodes.h>
 
diff -aur monopd-0.9.3.orig/src/game.cpp monopd-0.9.3.new/src/game.cpp
--- monopd-0.9.3.orig/src/game.cpp	2008-12-28 22:24:55.000000000 +0100
+++ monopd-0.9.3.new/src/game.cpp	2008-12-28 22:40:01.000000000 +0100
@@ -20,7 +20,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <syslog.h>
-#include <algo.h> // libstdc++ from the gcc 2.95 has no #include <algo> yet :(
+#include <algorithm> 
+#include <cstring>
 
 #include <map>
 #include <string>
diff -aur monopd-0.9.3.orig/src/main.cpp monopd-0.9.3.new/src/main.cpp
--- monopd-0.9.3.orig/src/main.cpp	2008-12-28 22:24:55.000000000 +0100
+++ monopd-0.9.3.new/src/main.cpp	2008-12-28 22:41:53.000000000 +0100
@@ -21,6 +21,7 @@
 
 #include <iostream>
 #include <string>
+#include <cstdlib>
 
 #include <libcapsinetwork/listener.h>
 #include <libcapsinetwork/socket.h>
diff -aur monopd-0.9.3.orig/src/player.cpp monopd-0.9.3.new/src/player.cpp
--- monopd-0.9.3.orig/src/player.cpp	2008-12-28 22:24:55.000000000 +0100
+++ monopd-0.9.3.new/src/player.cpp	2008-12-28 22:44:05.000000000 +0100
@@ -19,7 +19,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#include <string>
+#include <cstring>
 
 #include <libcapsinetwork/socket.h>
 
diff -aur monopd-0.9.3.orig/src/server.cpp monopd-0.9.3.new/src/server.cpp
--- monopd-0.9.3.orig/src/server.cpp	2008-12-28 22:24:55.000000000 +0100
+++ monopd-0.9.3.new/src/server.cpp	2008-12-28 22:50:46.000000000 +0100
@@ -27,6 +27,8 @@
 #include <unistd.h>
 
 #include <string>
+#include <cstring>
+#include <cstdlib>
 
 #include <libcapsinetwork/socket.h>
 
