--- Makefile.old	2009-07-15 23:49:08.000000000 +0400
+++ Makefile	2009-07-16 00:47:24.000000000 +0400
@@ -35,25 +35,29 @@
 	cd src && make clean
 
 webinstall: 
-	$(INSTALL) -d -m 755 $(PATH_TO_WWW)/cgi-bin/admin $(PATH_TO_WWW)/stat $(PATH_TO_WWW)/images 
-	$(INSTALL) -m 755 cgi-bin/*.cgi cgi-bin/*.pl cgi-bin/*.tmpl cgi-bin/*.res $(PATH_TO_WWW)/cgi-bin
-	$(INSTALL) -m 755 cgi-bin/admin/*.cgi cgi-bin/admin/*.tmpl cgi-bin/admin/*.res $(PATH_TO_WWW)/cgi-bin/admin
-	$(INSTALL) -m 644 cgi-bin/admin/.htaccess $(PATH_TO_WWW)/cgi-bin/admin
-	$(INSTALL) -m 644 cgi-bin/images/*.gif cgi-bin/images/*.jpg $(PATH_TO_WWW)/images
-	$(INSTALL) -m 644 addon/index.html $(PATH_TO_WWW)
-	$(INSTALL) -m 644 addon/netams-apache $(PATH_TO_ETC)/netams-apache
+	mkdir $(DESTDIR)$(PATH_TO_ETC)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(PATH_TO_WWW)/cgi-bin/admin $(DESTDIR)$(PATH_TO_WWW)/stat $(DESTDIR)$(PATH_TO_WWW)/images 
+	$(INSTALL) -m 755 cgi-bin/*.cgi cgi-bin/*.pl cgi-bin/*.tmpl cgi-bin/*.res $(DESTDIR)$(PATH_TO_WWW)/cgi-bin
+	$(INSTALL) -m 755 cgi-bin/admin/*.cgi cgi-bin/admin/*.tmpl cgi-bin/admin/*.res $(DESTDIR)$(PATH_TO_WWW)/cgi-bin/admin
+	$(INSTALL) -m 644 cgi-bin/admin/.htaccess $(DESTDIR)$(PATH_TO_WWW)/cgi-bin/admin
+	$(INSTALL) -m 644 cgi-bin/images/*.gif cgi-bin/images/*.jpg $(DESTDIR)$(PATH_TO_WWW)/images
+	$(INSTALL) -m 644 addon/index.html $(DESTDIR)$(PATH_TO_WWW)
+	$(INSTALL) -m 644 addon/netams-apache $(DESTDIR)$(PATH_TO_ETC)/netams-apache
 
 install: all webinstall
-	$(INSTALL) src/netams $(DAEMONDIR)
-	cd src ; $(INSTALL) $(PROG) $(SBINDIR) ; cd ..
-	$(INSTALL) -d -m 644 $(MANDIR)/man8
-	$(INSTALL) -m 644 doc/*.8 $(MANDIR)/man8
-	$(INSTALL) -m 644 addon/netams.conf $(PATH_TO_ETC)/netams.conf.sample
-	$(INSTALL) -m 644 addon/.netamsctl.rc $(PATH_TO_ETC)/.netamsctl.rc
-	$(INSTALL) -m 644 addon/ru-networks.txt $(PATH_TO_ETC)/ru-networks.txt.sample
-	$(INSTALL) -m 644 addon/ua-networks.txt $(PATH_TO_ETC)/ua-networks.txt.sample
+	mkdir $(DESTDIR)/usr
+	mkdir $(DESTDIR)$(DAEMONDIR)
+	mkdir $(DESTDIR)$(PATH_TO_SCRIPTS)
+	$(INSTALL) src/netams $(DESTDIR)$(DAEMONDIR)
+	cd src ; $(INSTALL) $(PROG) $(DESTDIR)$(SBINDIR) ; cd ..
+	$(INSTALL) -d -m 644 $(DESTDIR)$(MANDIR)/man8
+	$(INSTALL) -m 644 doc/*.8 $(DESTDIR)$(MANDIR)/man8
+	$(INSTALL) -m 644 addon/netams.conf $(DESTDIR)$(PATH_TO_ETC)/netams.conf.sample
+	$(INSTALL) -m 644 addon/.netamsctl.rc $(DESTDIR)$(PATH_TO_ETC)/.netamsctl.rc
+	$(INSTALL) -m 644 addon/ru-networks.txt $(DESTDIR)$(PATH_TO_ETC)/ru-networks.txt.sample
+	$(INSTALL) -m 644 addon/ua-networks.txt $(DESTDIR)$(PATH_TO_ETC)/ua-networks.txt.sample
 	sed -e "s#DAEMONDIR#$(DAEMONDIR)#g" -e "s#PATH_TO_ETC#$(PATH_TO_ETC)#g" addon/$(STARTUP_SCRIPT) > /tmp/netams-startup.sh 
-	$(INSTALL) -m 755 /tmp/netams-startup.sh $(PATH_TO_SCRIPTS)/$(STARTUP_SCRIPT)
+	$(INSTALL) -m 755 /tmp/netams-startup.sh $(DESTDIR)$(PATH_TO_SCRIPTS)/$(STARTUP_SCRIPT)
 	rm -f /tmp/netams-startup.sh
 	@echo
 	@echo "PLEASE READ THE DOCUMENTATION FIRST!"
