--- Makefile	2013-03-14 18:48:25.850863941 -0300
+++ Makefile	2013-03-14 18:49:56.984058255 -0300
@@ -73,7 +73,10 @@
 	$(HEXAGONSRCDIR)/Global/Factory.cpp \
 	$(HEXAGONSRCDIR)/Online/Online.cpp \
 	$(HEXAGONSRCDIR)/Utils/MD5.cpp \
-	$(HEXAGONSRCDIR)/Utils/Utils.cpp
+	$(HEXAGONSRCDIR)/Utils/Utils.cpp \
+	$(HEXAGONSRCDIR)/Utils/Base64.cpp \
+	$(HEXAGONSRCDIR)/Compatibility/Compatibility.cpp \
+	$(HEXAGONSRCDIR)/Utils/FPSWatcher.cpp
 STARTSRC = \
 	$(STARTSRCDIR)/Assets/AssetManager.cpp \
 	$(STARTSRCDIR)/Assets/AssetFolder.cpp \
