diff --git a/src/libscotch/bgraph_bipart_gg.c b/src/libscotch/bgraph_bipart_gg.c
index a568f8d..e11f547 100644
--- a/src/libscotch/bgraph_bipart_gg.c
+++ b/src/libscotch/bgraph_bipart_gg.c
@@ -71,6 +71,8 @@
 #include "bgraph.h"
 #include "bgraph_bipart_gg.h"
 
+#include <stdint.h>
+
 /*
 **  The static variables.
 */
diff --git a/src/libscotch/dgraph_fold_dup.c b/src/libscotch/dgraph_fold_dup.c
index 25d294c..753ee46 100644
--- a/src/libscotch/dgraph_fold_dup.c
+++ b/src/libscotch/dgraph_fold_dup.c
@@ -55,6 +55,8 @@
 #include "dgraph.h"
 #include "dgraph_fold_dup.h"
 
+#include <stdint.h>
+
 /******************************/
 /*                            */
 /* This routine handles       */
diff --git a/src/libscotch/dgraph_halo.c b/src/libscotch/dgraph_halo.c
index 4c3cdd6..24af460 100644
--- a/src/libscotch/dgraph_halo.c
+++ b/src/libscotch/dgraph_halo.c
@@ -65,6 +65,8 @@
 #include "dgraph.h"
 #include "dgraph_halo.h"
 
+#include <stdint.h>
+
 /* These routines fill the send arrays used by
 ** all of the halo routines.
 ** They return:
diff --git a/src/libscotch/hdgraph_order_nd.c b/src/libscotch/hdgraph_order_nd.c
index d26afa0..e75b6e7 100644
--- a/src/libscotch/hdgraph_order_nd.c
+++ b/src/libscotch/hdgraph_order_nd.c
@@ -67,6 +67,8 @@
 #include "vdgraph.h"
 #include "vdgraph_separate_st.h"
 
+#include <stdint.h>
+
 /*****************************/
 /*                           */
 /* This is the main routine. */
diff --git a/src/libscotch/kdgraph_map_rb_part.c b/src/libscotch/kdgraph_map_rb_part.c
index 7c936b8..ca09075 100644
--- a/src/libscotch/kdgraph_map_rb_part.c
+++ b/src/libscotch/kdgraph_map_rb_part.c
@@ -73,6 +73,8 @@
 #include "kdgraph_map_rb_part.h"
 #include "kdgraph_map_st.h"
 
+#include <stdint.h>
+
 /*****************************/
 /*                           */
 /* This is the main routine. */
diff --git a/src/libscotch/vgraph_separate_gg.c b/src/libscotch/vgraph_separate_gg.c
index c746c7f..b2c147f 100644
--- a/src/libscotch/vgraph_separate_gg.c
+++ b/src/libscotch/vgraph_separate_gg.c
@@ -65,6 +65,8 @@
 #include "vgraph.h"
 #include "vgraph_separate_gg.h"
 
+#include <stdint.h>
+
 /*
 **  The static variables.
 */
