--- broken/mumps/Makefile	2010-04-29 14:00:01.555112436 +0200
+++ fixed/mumps/Makefile	2010-04-29 14:00:12.461777164 +0200
@@ -7,12 +7,12 @@
 # the arguments given to './configure' script for building SDPA
 ###################################################################
 
-CC          = ${shell grep "^CC ="        ${SDPA_MAKEFILE} | cut -f2 -d=}
-CFLAGS      = ${shell grep "^CFLAGS = "   ${SDPA_MAKEFILE} | cut -f2 -d=}
-CXX         = ${shell grep "^CXX ="       ${SDPA_MAKEFILE} | cut -f2 -d=}
-CXXFLAGS    = ${shell grep "^CXXFLAGS = " ${SDPA_MAKEFILE} | cut -f2 -d=}
-F77         = ${shell grep "^F77 ="       ${SDPA_MAKEFILE} | cut -f2 -d=}
-FFLAGS      = ${shell grep "^FFLAGS = "   ${SDPA_MAKEFILE} | cut -f2 -d=}
+CC          = ${shell grep "^CC ="        ${SDPA_MAKEFILE} | cut -f2- -d=}
+CFLAGS      = ${shell grep "^CFLAGS = "   ${SDPA_MAKEFILE} | cut -f2- -d=}
+CXX         = ${shell grep "^CXX ="       ${SDPA_MAKEFILE} | cut -f2- -d=}
+CXXFLAGS    = ${shell grep "^CXXFLAGS = " ${SDPA_MAKEFILE} | cut -f2- -d=}
+F77         = ${shell grep "^F77 ="       ${SDPA_MAKEFILE} | cut -f2- -d=}
+FFLAGS      = ${shell grep "^FFLAGS = "   ${SDPA_MAKEFILE} | cut -f2- -d=}
 ###################################################################
 
 all: build/lib/libdmumps.a
