--- tapioca-glib-0.14.0.1.orig//tapioca/core/Makefile.am	2007-05-23 21:05:05.000000000 +0200
+++ tapioca-glib-0.14.0.1//tapioca/core/Makefile.am	2010-06-29 13:49:59.641830435 +0200
@@ -61,9 +61,7 @@ libtapioca_core_glib_@TAPIOCA_MAJORMINOR
 	tpa-room-list.h \
 	tpa-session.h \
 	tpa-text.h \
-	tpa-manager.h \
 	tpa-account.h \
-	tpa-session.h \
     tpa-ifaces.h
 
 noinst_HEADERS = \
--- tapioca-glib-0.14.0.1.orig//tapioca/core/Makefile.in	2007-06-05 15:46:57.000000000 +0200
+++ tapioca-glib-0.14.0.1//tapioca/core/Makefile.in	2010-06-29 13:49:41.747324856 +0200
@@ -306,9 +306,7 @@ libtapioca_core_glib_@TAPIOCA_MAJORMINOR
 	tpa-room-list.h \
 	tpa-session.h \
 	tpa-text.h \
-	tpa-manager.h \
 	tpa-account.h \
-	tpa-session.h \
     tpa-ifaces.h
 
 noinst_HEADERS = \
