diff -Nur nautilus-elementary-build.orig/configure.in nautilus-elementary-build/configure.in
--- nautilus-elementary-build.orig/configure.in	2010-10-29 16:42:53.000000000 +0200
+++ nautilus-elementary-build/configure.in	2010-12-16 17:04:05.415682392 +0100
@@ -23,7 +23,7 @@
 m4_define(nautilus_extension_current,  3)
 m4_define(nautilus_extension_revision, 0)
 
-AC_INIT([nautilus],[2.32.0],
+AC_INIT([nautilus],[2.32.2],
 	[http://bugzilla.gnome.org/enter_bug.cgi?product=nautilus])
 
 dnl ===========================================================================
diff -Nur nautilus-elementary-build.orig/src/nautilus-pathbar.c nautilus-elementary-build/src/nautilus-pathbar.c
--- nautilus-elementary-build.orig/src/nautilus-pathbar.c	2010-10-29 16:25:40.000000000 +0200
+++ nautilus-elementary-build/src/nautilus-pathbar.c	2010-12-16 17:02:54.062349334 +0100
@@ -1609,14 +1609,19 @@
 	} else if (nautilus_file_is_gone (file)) {
 		gint idx, position;
 
-		/* remove this and the following buttons */
-		position = g_list_position (path_bar->button_list,
+		/* if the current location is gone, don't do anything, as the view
+		 * will get the event too and call us back.
+		 */
+		if (nautilus_file_compare_location (file, current_button_data->file) != 0) {
+			/* remove this and the following buttons */
+			position = g_list_position (path_bar->button_list,
 					    g_list_find (path_bar->button_list, button_data));
 
-		if (position != -1) {
-			for (idx = 0; idx <= position; idx++) {
-				gtk_container_remove (GTK_CONTAINER (path_bar),
-						      BUTTON_DATA (path_bar->button_list->data)->button);
+			if (position != -1) {
+				for (idx = 0; idx <= position; idx++) {
+					gtk_container_remove (GTK_CONTAINER (path_bar),
+							      BUTTON_DATA (path_bar->button_list->data)->button);
+				}
 			}
 		}
 
