--- f_login.c.orig	2009-08-30 13:15:56.798155006 +0800
+++ f_login.c	2009-08-30 13:21:01.774794346 +0800
@@ -351,8 +351,11 @@
 							       _
 							       ("Invalid Password or Mobileno"));
 			cfg_size = get_token(buf, "Content-Length: ", "\r\n");
-			if (cfg_size == NULL)
-				return;
+			if (cfg_size == NULL) {
+				cfg_size = get_token(buf, "Content-length: ", "\r\n");
+				if (cfg_size == NULL)
+					return;
+			}
 			cur += 4;
 			sip->SysCfg.size = atoi(cfg_size);
 			sip->SysCfg.buf = g_malloc(sip->SysCfg.size);
