--- pidgin-2.5.5/libpurple/protocols/Makefile.am	2007-11-14 09:22:06.000000000 +0800
+++ pidgin-2.5.5/libpurple/protocols/Makefile.am	2009-04-01 15:10:16.943016058 +0800
@@ -1,5 +1,5 @@
 EXTRA_DIST = Makefile.mingw
 
-DIST_SUBDIRS = bonjour gg irc jabber msn msnp9 myspace novell null oscar qq sametime silc silc10 simple yahoo zephyr
+DIST_SUBDIRS = bonjour gg irc jabber msn msnp9 myspace novell null oscar qq sametime silc silc10 simple yahoo zephyr fetion
 
 SUBDIRS = $(DYNAMIC_PRPLS) $(STATIC_PRPLS)
