--- main.cpp	2008-04-12 08:55:56.000000000 +0200
+++ main.cpp	2008-10-24 11:44:47.000000000 +0200
@@ -29,20 +29,20 @@
 	QfNavBar *navbar = new QfNavBar;
 
 	QfNavBarGroup *group = navbar->addGroup("APPLICATIONS");
-	navbar->addItem(group, QPixmap("icons/z_allApps.png"), "All Applications");
-	navbar->addItem(group, QPixmap("icons/z_updates.png"), "Updates");
-	navbar->addItem(group, QPixmap("icons/z_upToDate.png"), "Up To Date");
-	navbar->addItem(group, QPixmap("icons/z_unknown.png"), "Unknown");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_allApps.png"), "All Applications");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_updates.png"), "Updates");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_upToDate.png"), "Up To Date");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_unknown.png"), "Unknown");
 
 	group = navbar->addGroup("BY KIND");
-	navbar->addItem(group, QPixmap("icons/z_apps.png"), "Applications");
-	navbar->addItem(group, QPixmap("icons/z_plugins.png"), "Plugins");
-	navbar->addItem(group, QPixmap("icons/z_widgets.png"), "Widgets");
-	navbar->addItem(group, QPixmap("icons/z_prefApp.png"), "Preference Panes");
-	navbar->addItem(group, QPixmap("icons/z_systemUpdates.png"), "System Updates");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_apps.png"), "Applications");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_plugins.png"), "Plugins");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_widgets.png"), "Widgets");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_prefApp.png"), "Preference Panes");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_systemUpdates.png"), "System Updates");
 
 	group = navbar->addGroup("VENDORS");
-	navbar->addItem(group, QPixmap("icons/z_apple.png"), "Apple");
+	navbar->addItem(group, QPixmap("/usr/share/qf-macnavbar/icons/z_apple.png"), "Apple");
 
 	navbar->show();
 
