diff --git a/Makefile b/Makefile
index 652123c..714949a 100644
--- a/Makefile
+++ b/Makefile
@@ -1208,7 +1208,6 @@ _modinst_:
 PHONY += _modinst_post
 _modinst_post: _modinst_
 	$(Q)$(MAKE) -f $(srctree)/scripts/Makefile.fwinst obj=firmware __fw_modinst
-	$(call cmd,depmod)
 
 else # CONFIG_MODULES
 
@@ -1451,9 +1450,6 @@ _emodinst_:
 
 PHONY += _emodinst_post
 _emodinst_post: _emodinst_
-ifeq ($(INSTALL_MOD_PATH),)
-	$(call cmd,depmod)
-endif
 
 clean-dirs := $(addprefix _clean_,$(KBUILD_EXTMOD))
 
@@ -1598,15 +1594,6 @@ quiet_cmd_rmdirs = $(if $(wildcard $(rm-dirs)),CLEAN   $(wildcard $(rm-dirs)))
 quiet_cmd_rmfiles = $(if $(wildcard $(rm-files)),CLEAN   $(wildcard $(rm-files)))
       cmd_rmfiles = rm -f $(rm-files)
 
-# Run depmod only if we have System.map and depmod is executable
-quiet_cmd_depmod = DEPMOD  $(KERNELRELEASE)
-      cmd_depmod = \
-	if [ -r System.map -a -x $(DEPMOD) ]; then                              \
-		$(DEPMOD) -ae -F System.map                                     \
-		$(if $(strip $(INSTALL_MOD_PATH)), -b $(INSTALL_MOD_PATH) )     \
-		$(KERNELRELEASE);                                               \
-	fi
-
 # Create temporary dir for module support files
 # clean it up only when building all modules
 cmd_crmodverdir = $(Q)mkdir -p $(MODVERDIR) \
