--- kactus2-2.2.0/Kactus2_Solution.pro	2013-04-23 12:26:32.000000000 +0400
+++ kactus2-2.2.0.new/Kactus2_Solution.pro	2013-08-29 23:36:56.672621097 +0400
@@ -25,10 +25,10 @@
 unix:help.files = ./Help/Kactus2Help.qch ./Help/Kactus2Help.qhc
 
 unix:config.path = /etc/xdg/TUT
+unix:config.extra = cp ./releaseFiles/DefaultSettingsLinux.ini $(INSTALL_ROOT)/etc/xdg/TUT/Kactus2.ini
 
-UNAME = $$system(uname -m)
-equals(UNAME, x86_64): unix:config.extra = cp ./releaseFiles/DefaultSettingsLinux.ini /etc/xdg/TUT/Kactus2.ini; (test -d /usr/lib64 && ln -f -s /usr/bin/Kactus2 /usr/lib64/libKactus2.so) || (test -d /lib/x86_64-linux-gnu && ln -f -s /usr/bin/Kactus2 /lib/x86_64-linux-gnu/libKactus2.so)
-!equals(UNAME, x86_64): unix:config.extra = cp ./releaseFiles/DefaultSettingsLinux.ini /etc/xdg/TUT/Kactus2.ini; ln -f -s /usr/bin/Kactus2 /usr/lib/libKactus2.so
+unix:lib.path = /usr/lib
+unix:lib.extra = ln -f -s /usr/bin/Kactus2 $(INSTALL_ROOT)/usr/lib/libKactus2.so
 
 unix:library.path = /usr/share/kactus2/library
 unix:library.files = ./releaseFiles/Library/*
@@ -39,5 +39,4 @@
 unix:desktop.path = /usr/share/applications
 unix:desktop.files = ./releaseFiles/kactus2.desktop
 
-INSTALLS += help doc upgrade config library icons desktop
-
+INSTALLS += help doc upgrade config lib library icons desktop
