diff -rauN luafilesystem-1.5.0/config luafilesystem-arch-patch-1.5.0/config
--- luafilesystem-1.5.0/config	2009-10-20 22:54:35.000000000 +0200
+++ luafilesystem-arch-patch-1.5.0/config	2010-12-17 22:07:24.000000000 +0100
@@ -1,7 +1,7 @@
 # Installation directories
 
 # Default installation prefix
-PREFIX=/usr/local
+PREFIX=$(DESTDIR)/usr
 
 # System's libraries directory (where binary libraries are installed)
 LUA_LIBDIR= $(PREFIX)/lib/lua/5.1
