--- multicoreware-x265-82bbd2bf3b49/source/CMakeLists.txt	2014-04-03 08:00:44.000000000 +0200
+++ patched/source/CMakeLists.txt	2014-04-11 10:42:36.393369572 +0200
@@ -286,6 +286,7 @@
         # shared library is not installed if a tag is not found
         set_target_properties(x265-shared PROPERTIES VERSION ${X265_LATEST_TAG} SOVERSION ${X265_BUILD})
         install(TARGETS x265-shared
+                RUNTIME DESTINATION ${BIN_INSTALL_DIR}
                 LIBRARY DESTINATION ${LIB_INSTALL_DIR}
                 ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
     endif()
diff -ru multicoreware-x265-82bbd2bf3b49/source/common/CMakeLists.txt patched/source/common/CMakeLists.txt
--- multicoreware-x265-82bbd2bf3b49/source/common/CMakeLists.txt	2014-04-03 08:00:44.000000000 +0200
+++ patched/source/common/CMakeLists.txt	2014-04-11 09:43:05.204709743 +0200
@@ -129,8 +129,12 @@
     source_group(Assembly FILES ${ASM_PRIMITIVES})
 endif(ENABLE_ASSEMBLY)
 
+if(MINGW)
+set_source_files_properties(param.cpp PROPERTIES COMPILE_FLAGS -DHAVE_STRTOK_R="1")
+else(MINGW)
 check_symbol_exists(strtok_r "string.h" HAVE_STRTOK_R)
 set_source_files_properties(param.cpp PROPERTIES COMPILE_FLAGS -DHAVE_STRTOK_R=${HAVE_STRTOK_R})
+endif(MINGW)
 
 if(GCC AND GCC_HAS_NO_NARROWING)
     set_source_files_properties(cpu.cpp PROPERTIES COMPILE_FLAGS -Wno-narrowing)
diff -ru multicoreware-x265-82bbd2bf3b49/source/common/param.cpp patched/source/common/param.cpp
--- multicoreware-x265-82bbd2bf3b49/source/common/param.cpp	2014-04-03 08:00:44.000000000 +0200
+++ patched/source/common/param.cpp	2014-04-11 09:45:15.396795288 +0200
@@ -37,7 +37,7 @@
 #define strcasecmp _stricmp 
 #endif
 
-#if !HAVE_STROTOK_R
+#if !HAVE_STRTOK_R
 /* 
  * adapted from public domain strtok_r() by Charlie Gordon
  *
