--- a/common/config.py	2011-12-07 18:06:38.720838665 -0500
+++ b/common/config.py	2011-12-07 18:06:46.937600015 -0500
@@ -35,7 +35,7 @@
 
 class Config( configfile.ConfigFileWithProfiles ):
 	APP_NAME = 'Back In Time'
-	VERSION = '1.0.7'
+	VERSION = '1.0.8'
 	COPYRIGHT = 'Copyright (c) 2008-2009 Oprea Dan, Bart de Koning, Richard Bailey'
 	CONFIG_VERSION = 5
 
--- a/kde4/docbook/en/index.docbook-old	2011-12-07 18:14:26.196724807 -0500
+++ b/kde4/docbook/en/index.docbook	2011-12-07 18:13:59.319695812 -0500
@@ -3,9 +3,9 @@
 <!ENTITY legal SYSTEM "legal.xml">
 <!ENTITY appname "Back In Time">
 <!ENTITY app "<application>&appname;</application>">
-<!ENTITY appversion "1.0.7">
-<!ENTITY manrevision "1.0.7">
-<!ENTITY date "Juliy 2010">
+<!ENTITY appversion "1.0.8">
+<!ENTITY manrevision "1.0.8">
+<!ENTITY date "June 2011">
 ]>
 <!--
       (Do not remove this comment block.)
@@ -24,7 +24,7 @@
     </abstract>         
 
     <copyright> 
-      <year>2008-2009</year> 
+      <year>2008-2011</year> 
       <holder>Oprea Dan, Bart de Koning, Richard Bailey</holder> 
     </copyright> 
 
--- a/common/debian_specific/control-old	2011-12-07 18:15:17.394088448 -0500
+++ b/common/debian_specific/control	2011-12-07 18:15:27.114221742 -0500
@@ -1,5 +1,5 @@
 Package: backintime-common
-Version: 1.0.7
+Version: 1.0.8
 Source: backintime-common
 Section: utils
 Priority: optional
--- a/gnome/docbook/C/backintime-C.omf-old	2011-12-07 18:15:50.617878655 -0500
+++ b/gnome/docbook/C/backintime-C.omf	2011-12-07 18:16:16.948243213 -0500
@@ -4,8 +4,8 @@
     <creator>(BIT Team)</creator>
     <maintainer>(BIT Team)</maintainer>
     <title>Back In Time Manual</title>
-    <date>July 2010</date>
-    <version identifier="Back In Time Manual 1.0.7" date="July 2010"/>
+    <date>June 2011</date>
+    <version identifier="Back In Time Manual 1.0.8" date="June 2011"/>
     <subject category="GNOME|Applications|System Tools"/>
     <description>
 		Back In Time is a simple backup system for GNOME desktop.
--- a/gnome/docbook/C/backintime.xml-old	2011-12-07 18:16:43.211942300 -0500
+++ b/gnome/docbook/C/backintime.xml	2011-12-07 18:17:24.309185413 -0500
@@ -3,9 +3,9 @@
 <!ENTITY legal SYSTEM "legal.xml">
 <!ENTITY appname "Back In Time">
 <!ENTITY app "<application>&appname;</application>">
-<!ENTITY appversion "1.0.7">
-<!ENTITY manrevision "1.0.7">
-<!ENTITY date "July 2010">
+<!ENTITY appversion "1.0.8">
+<!ENTITY manrevision "1.0.8">
+<!ENTITY date "June 2011">
 ]>
 <!--
       (Do not remove this comment block.)

