Index: Makefile.am
===================================================================
--- Makefile.am	(revision 12383)
+++ Makefile.am	(working copy)
@@ -51,7 +51,7 @@
 # default.supp, as it is built from the base .supp files at compile-time.
 dist_noinst_DATA = $(SUPP_FILES)
 
-pkglib_DATA = default.supp
+pkgdata_DATA = default.supp
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = valgrind.pc
Index: coregrind/Makefile.am
===================================================================
--- coregrind/Makefile.am	(revision 12383)
+++ coregrind/Makefile.am	(working copy)
@@ -513,7 +513,7 @@
 	m_gdbserver/powerpc-altivec64l.xml
 
 # so as to make sure these get copied into the install tree
-pkglib_DATA  = $(GDBSERVER_XML_FILES)
+pkgdata_DATA  = $(GDBSERVER_XML_FILES)
 
 # so as to make sure these get copied into the tarball
 EXTRA_DIST  += $(GDBSERVER_XML_FILES)
