commit 8ce78f01543587a6a991382f03536ed8bf8affa0
Author: Benjamin Chrétien <chretien@lirmm.fr>
Date:   Sun Mar 2 14:36:53 2014 +0100

    Fix MongoDB errors.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 68390de..af0742b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,7 @@ catkin_package(
   INCLUDE_DIRS ${CATKIN_DEVEL_PREFIX}/include include
   LIBRARIES warehouse_ros
   CATKIN_DEPENDS roscpp geometry_msgs rostime std_msgs
-  DEPENDS Boost
+  DEPENDS Boost MongoDB
   )
 
 include(cmake/FindMongoDB.cmake)
