--- a/CMakeLists.txt	2012-08-29 05:48:41.000000000 -0400
+++ b/CMakeLists.txt	2013-09-27 23:17:29.453674078 -0400
@@ -72,17 +72,23 @@
         "Use tutorial provided math implementation" OFF) 
 if (COMPIL_STATIC_LIBRARY)  
   add_library(mmg3dlib4.0 STATIC ${source_files} ${lib_file})
-  target_link_libraries(mmg3dlib4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR})
+  target_link_libraries(mmg3dlib4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR} ${PTHREAD_LIB} ${Z_LIB})
 endif (COMPIL_STATIC_LIBRARY)   
 option (COMPIL_SHARED_LIBRARY 
         "Use tutorial provided math implementation" OFF) 
 if (COMPIL_SHARED_LIBRARY)  
   add_library(mmg3dlib4.0 SHARED ${source_files} ${lib_file})
-  target_link_libraries(mmg3dlib4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR})
+  target_link_libraries(mmg3dlib4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR} ${PTHREAD_LIB} ${Z_LIB})
 endif (COMPIL_SHARED_LIBRARY)   
 
 find_library(M_LIB m)
-target_link_libraries(mmg3d4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR})    
+target_link_libraries(mmg3d4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR} ${PTHREAD_LIB} ${Z_LIB})    
+
+find_library(PTHREAD_LIB pthread)
+target_link_libraries(mmg3d4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR} ${PTHREAD_LIB} ${Z_LIB})    
+
+find_library(Z_LIB z)
+target_link_libraries(mmg3d4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR} ${PTHREAD_LIB} ${Z_LIB})    
 
 #add testlib   
 if (COMPIL_STATIC_LIBRARY) 
@@ -94,7 +100,7 @@
   )
   add_executable(testlib ${source_testlib}) 
   include_directories(sources/)                                    
-  target_link_libraries(testlib mmg3dlib4.0)    
+  target_link_libraries(testlib mmg3dlib4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR} ${PTHREAD_LIB} ${Z_LIB})    
 endif (COMPIL_STATIC_LIBRARY)   
 
 if (COMPIL_SHARED_LIBRARY) 
@@ -106,5 +112,5 @@
   )
   add_executable(testlib ${source_testlib}) 
   include_directories(sources/)                                    
-  target_link_libraries(testlib mmg3dlib4.0)    
+  target_link_libraries(testlib mmg3dlib4.0 ${M_LIB} ${LIBS_SCOTCH} ${LIBS_SCOTCHERR} ${PTHREAD_LIB} ${Z_LIB})    
 endif (COMPIL_SHARED_LIBRARY)
