--- src/sys_sdl_img_png.c_old	2012-02-11 21:20:28.000000000 +0000
+++ src/sys_sdl_img_png.c	2012-02-11 21:26:45.000000000 +0000
@@ -39,6 +39,10 @@
 #define int_p_NULL (int*)NULL
 #endif
 
+#if defined( TCOD_LINUX )
+#define png_infopp_NULL (png_infopp)NULL 
+#endif
+
 bool TCOD_sys_check_png(const char *filename) {
 	static uint8 magic_number[]={137, 80, 78, 71, 13, 10, 26, 10};
 	return TCOD_sys_check_magic_number(filename,sizeof(magic_number),magic_number);
@@ -121,7 +125,7 @@
 		&png_interlace_type,NULL,NULL);
 
 	/* create the SDL surface */
-	bitmap=TCOD_sys_get_surface(png_width,png_height,info_ptr->channels == 4);
+	bitmap=TCOD_sys_get_surface(png_width,png_height, png_get_channels(png_ptr,info_ptr) == 4);
 
 	/* get row data */
 	row_pointers=(png_bytep *)malloc(sizeof(png_bytep)*png_height);
