--- config.h	2004-09-07 16:28:35.000000000 +0200
+++ config.PATH	2004-12-09 15:16:00.000000000 +0100
@@ -79,19 +79,19 @@
   version number to this path.  You can override the default by
   uncommenting the defintion of TKGATE_HOMEDIR below.
  */
-#define TKGATE_HOMEDIRBASE	"/usr/X11R6/lib"
+#define TKGATE_HOMEDIRBASE	"/usr/share"
 
 /*
   The location of the tkgate and gmac executables.  If you change this, you will
   need to run the configure script again.
  */
-#define TKGATE_BINDIR		/usr/X11R6/bin
+#define TKGATE_BINDIR		/usr/bin
 
 /*
   The location of the tkgate and gmac man pages.  If you change this, you will
   need to run the configure script again.
  */
-#define TKGATE_MANDIR		/usr/X11R6/man/man1
+#define TKGATE_MANDIR		/usr/man/man1
 
 /*
   Digital circuit editor and simulator.
@@ -101,7 +101,7 @@
 /*
   URL for local documentation
  */
-#define TKGATE_LOCALDOC		"file:/usr/X11R6/lib/tkgate-" TKGATE_VERSION "/doc/index.html"
+#define TKGATE_LOCALDOC		"file:/usr/share/tkgate-" TKGATE_VERSION "/doc/index.html"
 
 /*
   The TkGate Copyright message.
@@ -122,13 +122,13 @@
   compile flags to use for gcc.  If you change this, you will
   need to run the configure script again.
  */
-#define TKGATE_GCCFLAGS		-O
+#define TKGATE_GCCFLAGS		-O2 -march=i686 -pipe
 
 /*
   compile flags to use for cc.  If you change this, you will
   need to run the configure script again.
  */
-#define TKGATE_CCFLAGS		
+#define TKGATE_CCFLAGS		-O2 -march=i686	-pipe
 
 /*
   Set this to '1' if you get an error about Tk_ConfigureWidget
