--- xyscanWindow.cpp	2010-01-15 02:08:27.000000000 +0100
+++ xyscanWindow-pathed.cpp	2010-02-27 16:05:55.000000000 +0100
@@ -370,27 +370,9 @@
 {
     mHelpBrowser = 0;
 
-    QString path = qApp->applicationDirPath() + "/docs";
+    QString path = "/usr/share/doc/xyscan";
     QDir dir(path);
     if (!dir.exists()) {
-        path = qApp->applicationDirPath() + "/../docs";
-        dir.setPath(path);
-    }
-#if defined(Q_OS_MAC)   
-    if (!dir.exists()) {
-        path = qApp->applicationDirPath() + "/../Resources/docs";
-        dir.setPath(path);
-    }
-#endif
-    //
-    //  For developing on Mac/Xcode only: avoids having to copy
-    //  the docs into the build substructure.
-    //
-    if (!dir.exists()) { 
-        path = "/Users/ullrich/Documents/Projects/xyscan/docs";
-        dir.setPath(path);
-    }
-    if (!dir.exists()) {
         QMessageBox::warning( 0, "xyscan",
                               "Cannot find the directory holding the documentation (docs). "
                               "No online help will be available. "
