--- blamframe-old.cpp	2014-03-01 22:04:54.104258685 +0800
+++ blamframe.cpp	2014-03-01 22:04:50.197591892 +0800
@@ -37,7 +37,7 @@
 	
 	/**** Icon *****/
 
-	wxIcon icon("wxwin.ico", wxBITMAP_TYPE_ICO);
+	wxIcon icon("/usr/share/pixmaps/k-map/wxwin.ico", wxBITMAP_TYPE_ICO);
 	SetIcon(icon);
 
 	/**** Menu *****/

--- blamframe-old.h	2014-03-01 20:50:51.684110222 +0800
+++ blamframe.h	2014-03-01 20:49:07.470773404 +0800
@@ -18,7 +18,7 @@
 class blamFrame : public wxFrame
 {
 	public:
-		blamFrame( const wxString& title, const wxPoint& pos, const wxSize& pos, wxLocale& locale );
+		blamFrame( const wxString& title, const wxPoint& pos, const wxSize& size, wxLocale& locale );
 		void OnQuit( wxCommandEvent& event );
 		void OnAbout( wxCommandEvent& event );
 		void OnCellAdresses( wxCommandEvent& event );
