--- scidavis.pro.orig	2010-03-12 19:40:49.000000000 +0100
+++ scidavis.pro	2010-03-12 19:42:34.000000000 +0100
@@ -87,32 +87,32 @@
 ### are compiled against Qt4), dynamically against everything else.
 #############################################################################
 
-unix:INCLUDEPATH  += ../3rdparty/qwtplot3d/include
-unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
+#unix:INCLUDEPATH  += ../3rdparty/qwtplot3d/include
+#unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
 
-unix:INCLUDEPATH  += ../3rdparty/qwt/src
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:INCLUDEPATH  += ../3rdparty/qwt/src
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
 
-unix:LIBS         += -L/usr/lib$${libsuff}
-unix:LIBS         += -lgsl -lgslcblas -lz
-unix:LIBS         += -lmuparser
-unix:INCLUDEPATH  += /usr/include/muParser
+#unix:LIBS         += -L/usr/lib$${libsuff}
+#unix:LIBS         += -lgsl -lgslcblas -lz
+#unix:LIBS         += -lmuparser
+#unix:INCLUDEPATH  += /usr/include/muParser
 
 #############################################################################
 ### Link everything dynamically
 #############################################################################
 
-#unix:INCLUDEPATH  += /usr/include/qwt5
-#unix:LIBS         += -L/usr/lib$${libsuff}
+unix:INCLUDEPATH  += /usr/include/qwt
+unix:LIBS         += -L/usr/lib$${libsuff}
 ## dynamically link against Qwt(3D) installed system-wide
 ## WARNING: make sure they are compiled against >= Qt4.2
 ## Mixing Qt 4.2 and Qt >= 4.3 compiled stuff may also 
 ## cause problems.
-#unix:INCLUDEPATH  += /usr/include/qwtplot3d
-#unix:LIBS         += -lqwtplot3d
-#unix:LIBS         += -lqwt
+unix:INCLUDEPATH  += /usr/include/qwtplot3d
+unix:LIBS         += -lqwtplot3d
+unix:LIBS         += -lqwt
 ##dynamically link against GSL and zlib installed system-wide
-#unix:LIBS         += -lgsl -lgslcblas -lz -lmuparser
+unix:LIBS         += -lgsl -lgslcblas -lz -lmuparser -lGLU
 
 #############################################################################
 ### Default settings for Windows
