--- toycars-0.3.10.orig/toycars/src/MyFisix/Convex.h	2008-09-09 16:14:27.000000000 +0600
+++ toycars-0.3.10/toycars/src/MyFisix/Convex.h	2012-02-24 04:13:00.376838629 +0600
@@ -22,6 +22,8 @@
 #endif
 
 #include <list>
+#include <cstddef>
+
 using namespace std;
 
 #include "Contact.h"
--- toycars-0.3.10.orig/toycars_track_editor/src/TrackView.cxx	2009-06-28 09:13:24.000000000 +0600
+++ toycars-0.3.10/toycars_track_editor/src/TrackView.cxx	2012-02-24 04:26:35.660144216 +0600
@@ -6,12 +6,12 @@
 #include "ScAnimation.h"
 #include "tinyxml.h"
 #include "ScException.h"
-#include <Fl/Fl_JPEG_Image.H>
-#include <Fl/Fl_PNG_Image.H>
-#include <Fl/Fl_BMP_Image.H>
-#include <Fl/Fl_PNM_Image.H>
-#include <Fl/Fl_GIF_Image.H>
-#include <Fl/Fl_XPM_Image.H>
+#include <FL/Fl_JPEG_Image.H>
+#include <FL/Fl_PNG_Image.H>
+#include <FL/Fl_BMP_Image.H>
+#include <FL/Fl_PNM_Image.H>
+#include <FL/Fl_GIF_Image.H>
+#include <FL/Fl_XPM_Image.H>
 #include <FL/Fl_File_Chooser.H>   // fl_alert
 #include "GameConstants.h"
 #include "TrackEditorUI.h"
