--- libwebcam-0.2.0/libwebcam/CMakeLists.txt.orig       2010-03-12 21:37:02.000000000 -0600
+++ libwebcam-0.2.0/libwebcam/CMakeLists.txt    2010-03-12 21:58:32.844272522 -0600
@@ -82,6 +82,8 @@
 # PACKAGING
 #
 
+
+if (EXISTS "/usr/bin/dpkg")
 # Prerequisites
 execute_process (
        COMMAND                         /usr/bin/dpkg --print-architecture
@@ -146,3 +148,5 @@
 
        include (CPack)
 endif (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
+
+endif (EXISTS "/usr/bin/dpkg")
--- libwebcam-0.2.0/uvcdynctrl/CMakeLists.txt.orig      2010-03-12 21:37:02.000000000 -0600
+++ libwebcam-0.2.0/uvcdynctrl/CMakeLists.txt   2010-03-12 22:01:27.567275338 -0600
@@ -103,6 +103,8 @@
 # Packaging
 #
 
+
+if (EXISTS "/usr/bin/dpkg")
 # Prerequisites
 execute_process (
        COMMAND                         /usr/bin/dpkg --print-architecture
@@ -168,3 +170,5 @@
 
        include (CPack)
 endif (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
+
+endif (EXISTS "/usr/bin/dpkg")
