--- a/libwebcam/CMakeLists.txt	2011-06-06 22:33:56.000000000 +0200
+++ b/libwebcam/CMakeLists.txt	2012-01-16 20:44:27.000000000 +0100
@@ -134,6 +134,8 @@
 # PACKAGING
 #
 
+
+if (EXISTS "/usr/bin/dpkg")
 # Prerequisites
 execute_process (
 	COMMAND				/usr/bin/dpkg --print-architecture
@@ -198,3 +200,5 @@
 
 	include (CPack)
 endif (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
+
+endif (EXISTS "/usr/bin/dpkg")
--- a/uvcdynctrl/CMakeLists.txt	2011-06-06 22:33:56.000000000 +0200
+++ b/uvcdynctrl/CMakeLists.txt	2012-01-16 20:49:19.000000000 +0100
@@ -103,6 +103,8 @@
 # Packaging
 #
 
+
+if (EXISTS "/usr/bin/dpkg")
 # Prerequisites
 execute_process (
 	COMMAND				/usr/bin/dpkg --print-architecture
@@ -168,3 +170,5 @@
 
 	include (CPack)
 endif (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
+
+endif (EXISTS "/usr/bin/dpkg")
