diff -Naur libwebcam/libwebcam/compat.h libwebcam-build//libwebcam/compat.h
--- libwebcam/libwebcam/compat.h	2011-04-03 17:59:21.634084436 -0300
+++ libwebcam-build//libwebcam/compat.h	2011-04-03 18:03:43.929330128 -0300
@@ -27,7 +27,6 @@
 #ifndef C_COMPAT_H
 #define C_COMPAT_H
 
-#include <linux/videodev.h>
 #include <linux/videodev2.h>
 
 #ifdef USE_UVCVIDEO
diff -Naur libwebcam/libwebcam/dynctrl.c libwebcam-build//libwebcam/dynctrl.c
--- libwebcam/libwebcam/dynctrl.c	2011-04-03 17:59:21.627417805 -0300
+++ libwebcam-build//libwebcam/dynctrl.c	2011-04-03 18:04:12.439175677 -0300
@@ -35,7 +35,7 @@
 #include <unistd.h>
 #include <stdarg.h>
 #include <sys/ioctl.h>
-#include <linux/videodev.h>
+#include <linux/videodev2.h>
 #include <errno.h>
 #include <iconv.h>
 
