--- src/utils/wxCas/src/wxcasframe.cpp	2010-02-03 18:14:39.000000000 +0100
+++ src/utils/wxCas/src/wxcasframe.cpp.new	2010-05-01 15:01:42.823357214 +0200
@@ -303,11 +303,11 @@
 #ifdef __WXMSW__
 
 	memdc.
-	SetFont ( wxFont::wxFont ( 6, wxSWISS, wxNORMAL, wxBOLD ) );
+	SetFont ( wxFont ( 6, wxSWISS, wxNORMAL, wxBOLD ) );
 #else
 
 	memdc.
-	SetFont ( wxFont::wxFont ( 8, wxSWISS, wxNORMAL, wxBOLD ) );
+	SetFont ( wxFont ( 8, wxSWISS, wxNORMAL, wxBOLD ) );
 #endif
 
 	memdc.
