--- classes/fpbase.cc	2003-05-29 20:25:25.000000000 +0200
+++ classes/fpbase.cc.new	2006-06-24 16:18:00.000000000 +0200
@@ -26,11 +26,11 @@ fpbase::fpbase()
  pstream::init(buf);
 }
 
-fpbase::fpbase(int f)
+/*fpbase::fpbase(int f)
 {
  CLY_NewFBFromFD(buf,f);
  pstream::init(buf);
-}
+}*/
 
 fpbase::fpbase( const char *name, CLY_OpenModeT omode, int prot )
 {
@@ -39,12 +39,12 @@ fpbase::fpbase( const char *name, CLY_Op
  pstream::init(buf);
 }
 
-fpbase::fpbase( int f, char *b, int len )
+/*fpbase::fpbase( int f, char *b, int len )
 {
  CLY_NewFBFromFD(buf,f);
  buf->CLY_PubSetBuf(b,len);
  pstream::init(buf);
-}
+}*/
 
 void fpbase::setbuf(char* b, int len)
 {
