Index: holyspirit/trunk/Source/Moteurs/moteurGraphique.cpp
===================================================================
--- holyspirit.orig/trunk/Source/Moteurs/moteurGraphique.cpp
+++ holyspirit/trunk/Source/Moteurs/moteurGraphique.cpp
@@ -1051,7 +1051,7 @@ bool MoteurGraphique::getEvent(sf::Event
 
 sf::Vector2f MoteurGraphique::getPositionSouris()
 {
-    return m_ecran.convertCoords(sf::Mouse::getPosition(m_ecran), m_camera);
+    return m_ecran.mapPixelToCoords(sf::Mouse::getPosition(m_ecran), m_camera);
 }
 
 sf::Vector2f MoteurGraphique::getPositionSourisEcran()
