--- Makefile.original	2013-05-15 16:13:45.106410846 -0400
+++ Makefile	2013-05-15 16:12:30.299749303 -0400
@@ -7,13 +7,9 @@
 SRCS = allpix_dm.cc listHandler.cpp MediPixWriteToEntuple.cc
 HDRS = allpix_dm.h listHandler.h MediPixWriteToEntuple.h allpix_dm_consts.h
 
-ifndef ROOTSYS
-  DO_FLAG=FALSE  
-else
-  DO_FLAG=TRUE
-  INCLUDE += $(shell $(ROOTSYS)/bin/root-config --prefix=$(ROOTSYS) --cflags)
-  LIBS += $(shell $(ROOTSYS)/bin/root-config --prefix=$(ROOTSYS) --libs)
-endif
+DO_FLAG=TRUE
+INCLUDE += $(shell root-config --cflags)
+LIBS += $(shell root-config --libs)
 
 GLOBAL_CFLAGS=-g3 -W -Wall
 #GLOBAL_CFLAGS=-O2 -W -Wall
