*** metadata.json.old	2012-04-27 15:23:44.126056131 +0300
--- metadata.json	2012-04-27 15:25:26.586055100 +0300
***************
*** 3,9 ****
    "description": "Switch between desktops scrolling at right edge of the screen.", 
    "name": "Desktop Scroller", 
    "shell-version": [
!     "3.2"
    ], 
    "url": "https://extensions.gnome.org", 
    "uuid": "desktop-scroller@gnome.org", 
--- 3,10 ----
    "description": "Switch between desktops scrolling at right edge of the screen.", 
    "name": "Desktop Scroller", 
    "shell-version": [
!     "3.2",
!     "3.4"
    ], 
    "url": "https://extensions.gnome.org", 
    "uuid": "desktop-scroller@gnome.org", 
