Index: climage.sh
===================================================================
--- climage.sh	(révision 30)
+++ climage.sh	(copie de travail)
@@ -19,6 +19,8 @@
 
 #main script for climage
 
+[ -e '/usr/bin/python2' ] && _python='/usr/bin/python2' || _python='/usr/bin/python'
+
 test_first_launch(){
 	#testing if "img" is a konwn command.
 	dont_remind=0
@@ -79,7 +81,7 @@
 #default
 pic_per_line=3
 #or conf file
-source ~/.climage 2> /dev/null
+source ~/.climage > /dev/null
 
 result=0
 
@@ -88,7 +90,7 @@
 
 #case: "ls" on the current directory
 if [ $# -eq 0 ]; then
-	python $script_path/src/ls-climage.py $pic_per_line $columns
+	$_python $script_path/src/ls-climage.py $pic_per_line $columns
 	result=$?
 	folder=$(pwd)
 	echo "$folder" > /tmp/climage_mem_path
@@ -105,21 +107,21 @@
 		if [ $# -ge 1 ]; then
 			cd "$1" 2> /dev/null
 		fi
-		python $script_path/src/ls-climage.py $pic_per_line $columns
+		$_python $script_path/src/ls-climage.py $pic_per_line $columns
 		result=$? 
 		#memorize folder
 		folder=$(pwd)
 		echo "$folder" > /tmp/climage_mem_path
 	elif [ -f "$1" ]; then
 		#display one single picture from its path
-		python $script_path/src/cat-climage.py "$1" $columns
+		$_python $script_path/src/cat-climage.py "$1" $columns
 		result=$?
 	else
 		#display one single picture from its id
 		if [ -e "/tmp/climage_mem_path" ]; then
 			folder=$(cat /tmp/climage_mem_path)
 			cd "$folder"
-			python $script_path/src/id-climage.py $1 $columns
+			$_python $script_path/src/id-climage.py $1 $columns
 			result=$?
 		fi
 		if [ $? -eq 2 ]; then
