--- packages/PyTrilinos/src/EpetraExt.i
+++ packages/PyTrilinos/src/EpetraExt.i
@@ -450,18 +450,22 @@
 /////////////////////////////////
 %include "EpetraExt_Transform.h"
 %template () std::vector<Epetra_IntVector>;
-
-%template () EpetraExt::Transform<Epetra_CrsGraph, Epetra_MapColoring>;
+%include "Epetra_IntVector.h"
+%include "Epetra_CrsGraph.h"
+%include "Epetra_MapColoring.h"
+%template (aa) EpetraExt::Transform<Epetra_CrsGraph, Epetra_MapColoring>;
+%template (bb) EpetraExt::Transform<Epetra_CrsGraph,
+				    std::vector<Epetra_IntVector > >;
 %template () EpetraExt::Transform<Epetra_CrsGraph,
 				  std::vector<Epetra_IntVector,
 					      std::allocator<Epetra_IntVector> > >;
-%template () EpetraExt::Transform<Epetra_CrsMatrix, Epetra_CrsMatrix >;
+%template (cc) EpetraExt::Transform<Epetra_CrsMatrix, Epetra_CrsMatrix >;
 
-%template () EpetraExt::StructuralTransform<Epetra_CrsGraph, Epetra_MapColoring>;
-%template () EpetraExt::StructuralTransform<Epetra_CrsGraph,
+%template (dd) EpetraExt::StructuralTransform<Epetra_CrsGraph, Epetra_MapColoring>;
+%template (ee) EpetraExt::StructuralTransform<Epetra_CrsGraph,
 					    std::vector<Epetra_IntVector> >;
 
-%template () EpetraExt::SameTypeTransform<Epetra_CrsMatrix >;
+%template (ff) EpetraExt::SameTypeTransform<Epetra_CrsMatrix >;
 
 ///////////////////////////////////
 // EpetraExt_MapColoring support //
