--- opt/rightscale/bin/post_install.sh   2012-06-20 11:21:06.000000000 +1000
+++ opt/rightscale/bin/post_install.sh   2012-11-03 15:40:37.009184929 +1100
@@ -13,22 +13,22 @@
 
 install_right_link_scripts
 
-echo "Setting up System V init and motd"
-
-ln -f /opt/rightscale/etc/init.d/rightboot /etc/init.d/rightboot
-ln -f /opt/rightscale/etc/init.d/rightscale /etc/init.d/rightscale
-ln -f /opt/rightscale/etc/init.d/rightlink /etc/init.d/rightlink
+if [ "$RS_BASE_OS" != 'archlinux' ]; then
+  echo "Setting up System V init and motd"
+  ln -f /opt/rightscale/etc/init.d/rightboot /etc/init.d/rightboot
+  ln -f /opt/rightscale/etc/init.d/rightscale /etc/init.d/rightscale
+  ln -f /opt/rightscale/etc/init.d/rightlink /etc/init.d/rightlink
+  chmod +x /etc/init.d/rightboot
+  chmod +x /etc/init.d/rightscale
+  chmod +x /etc/init.d/rightlink
+fi
 
 mkdir -p /var/lib/rightscale/right_link/certs
 ln -s /var/lib/rightscale/right_link/certs /opt/rightscale/right_link/certs
 
 chmod -R +x /opt/rightscale/bin/*
 chmod +x /opt/rightscale/etc/init.d/*
-chmod +x /etc/init.d/rightboot
-chmod +x /etc/init.d/rightscale
-chmod +x /etc/init.d/rightlink
 chmod +x /opt/rightscale/right_link/bin/*
-
 mkdir -p /etc/rightscale.d
 
 if [ "$RS_BASE_OS" == "debian" ]; then
