diff -Nur ./include/ARToolKitPlus/TrackerImpl.h ../srcFixed/include/ARToolKitPlus/TrackerImpl.h
--- ./include/ARToolKitPlus/TrackerImpl.h	2009-08-12 10:14:10.000000000 -0300
+++ ../srcFixed/include/ARToolKitPlus/TrackerImpl.h	2009-08-12 10:14:29.000000000 -0300
@@ -198,7 +198,7 @@
 	 * in an ARToolKit camera calibration file.
 	 * Returns true if loading of the camera file succeeded.
 	 */
-	static bool calcCameraMatrix(const char* nCamParamFile, int nWidth, int nHeight,
+	/*static*/ bool calcCameraMatrix(const char* nCamParamFile, int nWidth, int nHeight,
 								 ARFloat nNear, ARFloat nFar, ARFloat *nMatrix);
 
 	
diff -Nur ./src/librpp/rpp.cpp ../srcFixed/src/librpp/rpp.cpp
--- ./src/librpp/rpp.cpp	2009-08-12 10:14:09.000000000 -0300
+++ ../srcFixed/src/librpp/rpp.cpp	2009-08-12 10:15:55.000000000 -0300
@@ -50,6 +50,7 @@
 #include "rpp.h"
 #include "rpp_const.h"
 #include "rpp_vecmat.h"
+#include <cstring>
 
 namespace rpp {
 
diff -Nur ./src/librpp/rpp_vecmat.cpp ../srcFixed/src/librpp/rpp_vecmat.cpp
--- ./src/librpp/rpp_vecmat.cpp	2009-08-12 10:14:09.000000000 -0300
+++ ../srcFixed/src/librpp/rpp_vecmat.cpp	2009-08-12 10:15:55.000000000 -0300
@@ -44,6 +44,8 @@
 #include "rpp_vecmat.h"
 #include "math.h"
 #include "assert.h"
+#include <cstdlib>
+#include <cstdio>
 
 
 namespace rpp {
diff -Nur ./tools/IdPatGen/src/PN/Image.cpp ../srcFixed/tools/IdPatGen/src/PN/Image.cpp
--- ./tools/IdPatGen/src/PN/Image.cpp	2009-08-12 10:14:09.000000000 -0300
+++ ../srcFixed/tools/IdPatGen/src/PN/Image.cpp	2009-08-12 10:15:55.000000000 -0300
@@ -35,7 +35,7 @@
 
 #include <cstdio>
 #include <cstdlib>
-#include <string>
+#include <cstring>
 
 #if defined(TARGET_HOST_WIN32) || defined(TARGET_HOST_WINCE)
 #include <windows.h>
diff -Nur ./tools/IdPatGen/src/PN/ImageTool.cpp ../srcFixed/tools/IdPatGen/src/PN/ImageTool.cpp
--- ./tools/IdPatGen/src/PN/ImageTool.cpp	2009-08-12 10:14:09.000000000 -0300
+++ ../srcFixed/tools/IdPatGen/src/PN/ImageTool.cpp	2009-08-12 10:15:55.000000000 -0300
@@ -33,7 +33,7 @@
 #include "ImageTool.h"
 
 #include <cstdio>
-#include <string>
+#include <cstring>
 
 #if defined(TARGET_HOST_WIN32) || defined(TARGET_HOST_WINCE)
 #include <windows.h>
