diff -Naur pkg/usr/include/ARToolKitPlus/TrackerImpl.h pkgFixed/usr/include/ARToolKitPlus/TrackerImpl.h
--- pkg/usr/include/ARToolKitPlus/TrackerImpl.h	2006-07-25 16:05:46.000000000 -0300
+++ pkgFixed/usr/include/ARToolKitPlus/TrackerImpl.h	2009-08-11 02:20:06.000000000 -0300
@@ -659,40 +659,40 @@
 
 // this is templated code, so we need to include all this here...
 //
-#include "../../src/extra/FixedPoint.h"
-#include "../../src/core/arBitFieldPattern.cxx"
-#include "../../src/core/arDetectMarker.cxx"
-#include "../../src/core/arDetectMarker2.cxx"
-#include "../../src/core/arGetCode.cxx"
-#include "../../src/core/arGetMarkerInfo.cxx"
-#include "../../src/core/arGetTransMat.cxx"
-#include "../../src/core/arGetTransMat2.cxx"
-#include "../../src/core/arGetTransMat3.cxx"
-#include "../../src/core/rppGetTransMat.cxx" // RPP integration -- [t.pintaric]
-#include "../../src/core/arGetTransMatCont.cxx"
-#include "../../src/core/arLabeling.cxx"
-#include "../../src/core/arMultiActivate.cxx"
-#include "../../src/core/arMultiGetTransMat.cxx"
-#include "../../src/core/rppMultiGetTransMat.cxx" 	// RPP integration -- [t.pintaric]
-#include "../../src/core/arMultiReadConfigFile.cxx"
-#include "../../src/core/arUtil.cxx"
-#include "../../src/core/matrix.cxx"
-#include "../../src/core/mPCA.cxx"
-//#include "../../src/core/paramChangeSize.cxx"
-#include "../../src/core/paramDecomp.cxx"
-#include "../../src/core/paramDistortion.cxx"
-#include "../../src/core/byteSwap.cxx"
-#include "../../src/core/paramFile.cxx"
-#include "../../src/core/vector.cxx"
+#include "src/extra/FixedPoint.h"
+#include "src/core/arBitFieldPattern.cxx"
+#include "src/core/arDetectMarker.cxx"
+#include "src/core/arDetectMarker2.cxx"
+#include "src/core/arGetCode.cxx"
+#include "src/core/arGetMarkerInfo.cxx"
+#include "src/core/arGetTransMat.cxx"
+#include "src/core/arGetTransMat2.cxx"
+#include "src/core/arGetTransMat3.cxx"
+#include "src/core/rppGetTransMat.cxx" // RPP integration -- [t.pintaric]
+#include "src/core/arGetTransMatCont.cxx"
+#include "src/core/arLabeling.cxx"
+#include "src/core/arMultiActivate.cxx"
+#include "src/core/arMultiGetTransMat.cxx"
+#include "src/core/rppMultiGetTransMat.cxx" 	// RPP integration -- [t.pintaric]
+#include "src/core/arMultiReadConfigFile.cxx"
+#include "src/core/arUtil.cxx"
+#include "src/core/matrix.cxx"
+#include "src/core/mPCA.cxx"
+//#include "src/core/paramChangeSize.cxx"
+#include "src/core/paramDecomp.cxx"
+#include "src/core/paramDistortion.cxx"
+#include "src/core/byteSwap.cxx"
+#include "src/core/paramFile.cxx"
+#include "src/core/vector.cxx"
 
-#include "../../src/CameraImpl.cxx"
-#include "../../src/CameraAdvImpl.cxx"
-#include "../../src/CameraFactory.cxx"
-#include "../../src/extra/BCH.cxx"
+#include "src/CameraImpl.cxx"
+#include "src/CameraAdvImpl.cxx"
+#include "src/CameraFactory.cxx"
+#include "src/extra/BCH.cxx"
 
-#include "../../src/TrackerImpl.cxx"
-//#include "../../src/extra/harrisCornerDetector.cxx"
-//#include "../../src/extra/cornerRefinement.cxx"
+#include "src/TrackerImpl.cxx"
+//#include "src/extra/harrisCornerDetector.cxx"
+//#include "src/extra/cornerRefinement.cxx"
 
 
 #endif //__ARTOOLKIT_TRACKERIMPL_HEADERFILE__
diff -Naur pkg/usr/include/ARToolKitPlus/TrackerMultiMarkerImpl.h pkgFixed/usr/include/ARToolKitPlus/TrackerMultiMarkerImpl.h
--- pkg/usr/include/ARToolKitPlus/TrackerMultiMarkerImpl.h	2006-07-25 16:05:46.000000000 -0300
+++ pkgFixed/usr/include/ARToolKitPlus/TrackerMultiMarkerImpl.h	2009-08-11 02:21:27.000000000 -0300
@@ -183,7 +183,7 @@
 };	// namespace ARToolKitPlus
 
 
-#include "../src/TrackerMultiMarkerImpl.cxx"
+#include "src/TrackerMultiMarkerImpl.cxx"
 
 
 #endif //__ARTOOLKITPLUS_TRACKERMULTIMARKERIMPL_HEADERFILE__
diff -Naur pkg/usr/include/ARToolKitPlus/TrackerSingleMarkerImpl.h pkgFixed/usr/include/ARToolKitPlus/TrackerSingleMarkerImpl.h
--- pkg/usr/include/ARToolKitPlus/TrackerSingleMarkerImpl.h	2006-07-25 16:05:46.000000000 -0300
+++ pkgFixed/usr/include/ARToolKitPlus/TrackerSingleMarkerImpl.h	2009-08-11 02:21:09.000000000 -0300
@@ -178,7 +178,7 @@
 
 }	// namespace ARToolKitPlus
 
-#include "../src/TrackerSingleMarkerImpl.cxx"
+#include "src/TrackerSingleMarkerImpl.cxx"
 
 
 #endif //__ARTOOLKITPLUS_TRACKERSINGLEMARKERIMPL_HEADERFILE__
