Only in nfqueue-bindings-0.3/: build
diff -aur -x '*~' nfq-orig/nfqueue-bindings-0.3/CMakeLists.txt nfqueue-bindings-0.3/CMakeLists.txt
--- nfq-orig/nfqueue-bindings-0.3/CMakeLists.txt	2009-10-18 08:37:28.000000000 -0700
+++ nfqueue-bindings-0.3/CMakeLists.txt	2012-02-20 14:43:53.719672995 -0800
@@ -26,7 +26,7 @@
 SET( LIB_INSTALL_DIR  "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" )
 
 ADD_SUBDIRECTORY(python)
-ADD_SUBDIRECTORY(perl)
+# ADD_SUBDIRECTORY(perl)
 
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/nfq_version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/nfq_version.h)
 
diff -aur -x '*~' nfq-orig/nfqueue-bindings-0.3/Makefile nfqueue-bindings-0.3/Makefile
--- nfq-orig/nfqueue-bindings-0.3/Makefile	2009-10-18 08:37:28.000000000 -0700
+++ nfqueue-bindings-0.3/Makefile	2012-02-20 14:41:33.537099097 -0800
@@ -1,5 +1,6 @@
 PREFIX = /usr/local
-CMAKE_OPTIONS = -DDEBUG=1 -DCMAKE_VERBOSE_MAKEFILE=0 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=$(PREFIX)
+PYTHON_EXE = `which python2`
+CMAKE_OPTIONS = -DDEBUG=1 -DCMAKE_VERBOSE_MAKEFILE=0 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=$(PREFIX) -DPYTHON_EXECUTABLE=$(PYTHON_EXE)
 
 all:
 	[ -d build ] || mkdir build; \
diff -aur -x '*~' nfq-orig/nfqueue-bindings-0.3/nfq_common.c nfqueue-bindings-0.3/nfq_common.c
--- nfq-orig/nfqueue-bindings-0.3/nfq_common.c	2009-10-18 08:37:28.000000000 -0700
+++ nfqueue-bindings-0.3/nfq_common.c	2012-02-20 15:34:13.513313368 -0800
@@ -35,7 +35,7 @@
 
 int queue_bind(struct queue *self, int af_family)
 {
-        if (nfq_bind_pf(self->_h, af_family)) {
+        if (nfq_bind_pf(self->_h, af_family) < 0) {
                 throw_exception("error during nfq_bind_pf()");
                 return -1;
         }
@@ -44,7 +44,7 @@
 
 int queue_unbind(struct queue *self, int af_family)
 {
-        if (nfq_unbind_pf(self->_h, af_family)) {
+        if (nfq_unbind_pf(self->_h, af_family) < 0) {
                 throw_exception("error during nfq_unbind_pf()");
                 return -1;
         }
