diff -rupN mosquitto-0.12/client/Makefile mosquitto-0.12.edit//client/Makefile
--- mosquitto-0.12/client/Makefile	2011-07-25 21:45:36.000000000 +0100
+++ mosquitto-0.12.edit//client/Makefile	2011-08-15 22:17:44.926606708 +0100
@@ -2,7 +2,7 @@ CC=cc
 CFLAGS=-I../lib -Wall -ggdb
 LDFLAGS=-L../lib ../lib/libmosquitto.so.0
 INSTALL=install
-prefix=/usr/local
+prefix=/usr
 
 .PHONY: all install uninstall clean
 
diff -rupN mosquitto-0.12/config.mk mosquitto-0.12.edit//config.mk
--- mosquitto-0.12/config.mk	2011-07-25 21:45:37.000000000 +0100
+++ mosquitto-0.12.edit//config.mk	2011-08-15 22:17:31.383273337 +0100
@@ -23,6 +23,6 @@ MSGMERGE=msgmerge
 MSGFMT=msgfmt
 DOCBOOK2MAN=docbook2man.pl
 
-prefix=/usr/local
+prefix=/usr
 mandir=${prefix}/share/man
 localedir=${prefix}/share/locale
