--- a/src/Makefile	2014-10-08 23:50:07.000000000 +0200
+++ b/src/Makefile	2014-11-16 22:43:34.280059788 +0100
@@ -99,15 +99,15 @@
 	${CC} $(CFLAGS) ${CPPFLAGS} -c $< -o $@
 
 install : all
-	$(INSTALL) -d ${DESTDIR}$(prefix)/sbin
-	$(INSTALL) -s --strip-program=$(STRIP) mosquitto ${DESTDIR}${prefix}/sbin/mosquitto
+	$(INSTALL) -d ${DESTDIR}$(prefix)/bin
+	$(INSTALL) -s --strip-program=$(STRIP) mosquitto ${DESTDIR}${prefix}/bin/mosquitto
 	$(INSTALL) mosquitto_plugin.h ${DESTDIR}${prefix}/include/mosquitto_plugin.h
 ifeq ($(WITH_TLS),yes)
 	$(INSTALL) -s --strip-program=$(STRIP) mosquitto_passwd ${DESTDIR}${prefix}/bin/mosquitto_passwd
 endif
 
 uninstall :
-	-rm -f ${DESTDIR}${prefix}/sbin/mosquitto
+	-rm -f ${DESTDIR}${prefix}/bin/mosquitto
 	-rm -f ${DESTDIR}${prefix}/include/mosquitto_plugin.h
 	-rm -f ${DESTDIR}${prefix}/bin/mosquitto_passwd
 
