--- 12.6/common/lib/modules/fglrx/build_mod/firegl_public.c	2012-06-09 11:58:17.000000000 +0200
+++ 12.6/common/lib/modules/fglrx/build_mod/firegl_public.c	2012-06-09 11:53:54.000000000 +0200
@@ -34,6 +34,10 @@
 #include <linux/autoconf.h>
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,4,0)
+#include <asm/fpu-internal.h>
+#endif
+
 #if !defined(CONFIG_X86) 
 #if !defined(CONFIG_X86_PC) 
 #if !defined(CONFIG_X86_XEN) 
