--- Reveal.pro.orig	2005-11-05 08:05:25.000000000 -0800
+++ Reveal.pro	2005-11-05 08:18:09.000000000 -0800
@@ -37,7 +37,7 @@
 #===========================================================================================
 
 #Comment this in if you want to manually set the include and linking paths
-#CONFIG += manualPaths
+CONFIG += manualPaths
 
 #Comment this out if you don't want to UPX compress the binary
 #!mac:unix:CONFIG += useUpxCompression
@@ -51,11 +51,10 @@
 #------------ Linux/FreeBSD Include Paths + Linking Options ---------------
 !mac:unix {
   contains( CONFIG, manualPaths ) {
-    INCLUDEPATH += ../libs/exiv2-svn/src
-    INCLUDEPATH += ../libs/libjpeg
-    LIBS += ../libs/libexiv2.a
-    LIBS += ../libs/libjpeg.a
-    LIBS += ../libs/libqt-mt.a
+    INCLUDEPATH += /usr/include/exiv2
+    INCLUDEPATH += /usr/include
+    LIBS += -lexiv2
+    LIBS += -ljpeg
   } else {
     LIBS += -lexiv2
     LIBS += -ljpeg
