--- copy	2012-01-24 11:29:27.790561756 -0800
+++ rdmd.d	2012-01-24 11:30:35.914955004 -0800
@@ -473,7 +473,7 @@
 {
     // Escape trailing backslashes, so they don't escape the ending quote.
     // Backslashes elsewhere should NOT be escaped.
-    for (int i=arg.length-1; i>=0 && arg[i]=='\\'; i--)
+    for (int i=(cast(int)arg.length)-1; i>=0 && arg[i]=='\\'; i--)
         arg ~= '\\';
     return '"' ~ std.array.replace(arg, `"`, `\"`) ~ '"';
 }
