diff -aurN squid-3.1.14-orig/src/Makefile.in squid-3.1.14/src/Makefile.in
--- squid-3.1.14-orig/src/Makefile.in	2011-07-04 07:39:51.000000000 +0300
+++ squid-3.1.14/src/Makefile.in	2011-07-31 22:33:21.802735751 +0300
@@ -2061,7 +2061,7 @@
 DEFAULT_STORE_LOG = $(DEFAULT_LOG_PREFIX)/store.log
 DEFAULT_PID_FILE = $(DEFAULT_PIDFILE)
 DEFAULT_NETDB_FILE = $(DEFAULT_LOG_PREFIX)/netdb.state
-DEFAULT_SWAP_DIR = $(localstatedir)/cache
+DEFAULT_SWAP_DIR = $(localstatedir)/cache/squid
 DEFAULT_SSL_DB_DIR = $(localstatedir)/lib/ssl_db
 DEFAULT_PINGER = $(libexecdir)/`echo pinger | sed '$(transform);s/$$/$(EXEEXT)/'`
 DEFAULT_UNLINKD = $(libexecdir)/`echo unlinkd | sed '$(transform);s/$$/$(EXEEXT)/'`
diff -aurN squid-3.1.14-orig/src/cf.data.pre squid-3.1.14/src/cf.data.pre
--- squid-3.1.14-orig/src/cf.data.pre	2011-07-04 07:38:34.000000000 +0300
+++ squid-3.1.14/src/cf.data.pre	2011-07-31 22:33:58.388278368 +0300
@@ -4230,7 +4230,7 @@
 
 NAME: cache_effective_group
 TYPE: string
-DEFAULT: none
+DEFAULT: proxy
 LOC: Config.effectiveGroup
 DOC_START
 	Squid sets the GID to the effective user's default group ID
