diff --git a/configure b/configure
index 31ff979..f142729 100755
--- a/configure
+++ b/configure
@@ -485,7 +485,7 @@ pkgxx-install: pkgxx has-install # has-install is needed for \$(LIBEXECDIR)
 	`DIR \$\(ETCDIR\)/pkg++/defaults`
 	`DIR \$\(ETCDIR\)/pkg++/uses`
 	`INSTALL 0755 pkg++ \$\(BINDIR\)`
-	`ZCOMPILE '$(BINDIR)/pkg++'`
+	`ZCOMPILE '$(DESTDIR)$(BINDIR)/pkg++'`
 	`for include in $(ls includes/); do
 		INSTALL 0644 includes/$include \$\(SHAREDIR\)/pkg++/includes
 		echo -ne "\t"
