diff -Nur transmission-remote-cli/transmission-remote-cli.py transmission-remote-cli-python2//transmission-remote-cli.py
--- transmission-remote-cli/transmission-remote-cli.py	2010-10-19 10:33:49.544594708 -0300
+++ transmission-remote-cli-python2//transmission-remote-cli.py	2010-10-19 10:42:51.278983745 -0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ########################################################################
 # This is transmission-remote-cli, whereas 'cli' stands for 'Curses    #
 # Luminous Interface', a client for the daemon of the BitTorrent       #
