--- CMakeLists.orig.txt	2015-04-24 02:59:54.023123455 +0200
+++ CMakeLists.txt	2015-04-24 02:58:01.641595167 +0200
@@ -25,7 +25,7 @@
 option(googleapis_build_samples "Build googleapis sample programs." ON)
 option(googleapis_build_service_apis "Build specialized service APIs." ON)
 
-set(GOOGLEAPIS_EXTERNAL_DEPENDENCIES_DIR ${CMAKE_BINARY_DIR}/external_dependencies/install CACHE STRING "Root directory where external dependencies are located")
+set(GOOGLEAPIS_EXTERNAL_DEPENDENCIES_DIR "${CMAKE_BINARY_DIR}/external_dependencies/install" CACHE STRING "Root directory where external dependencies are located")
 
 if (MSVC)
   option(gtest_force_shared_crt "Force Gmock to use standard compiler flags" ON)
