diff -Nru kdenetwork-4.7.97/krdc/application.cpp kdenetwork-4.7.97-new//krdc/application.cpp
--- kdenetwork-4.7.97/krdc/application.cpp	1970-01-01 01:00:00.000000000 +0100
+++ kdenetwork-4.7.97-new//krdc/application.cpp	2012-01-25 14:06:47.193661049 +0100
@@ -0,0 +1,75 @@
+/****************************************************************************
+**
+** Copyright (C) 2001-2003 Tim Jansen <tim@tjansen.de>
+** Copyright (C) 2007 - 2012 Urs Wolfer <uwolfer @ kde.org>
+**
+** This file is part of KDE.
+**
+** This program is free software; you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation; either version 2 of the License, or
+** (at your option) any later version.
+**
+** This program is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program; see the file COPYING. If not, write to
+** the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+** Boston, MA 02110-1301, USA.
+**
+****************************************************************************/
+
+#include "application.h"
+#include "mainwindow.h"
+
+#include <KCmdLineArgs>
+#include <KDebug>
+
+Application::Application()
+        : KUniqueApplication()
+{ 
+    m_mainWindow = new MainWindow;
+    m_mainWindow->show();
+}
+
+Application::~Application()
+{
+    delete m_mainWindow;
+}
+
+int Application::newInstance()
+{
+    KCmdLineArgs *args = KCmdLineArgs::parsedArgs();
+
+    if (args->count() > 0) {
+        for (int i = 0; i < args->count(); ++i) {
+            KUrl u(args->url(i));
+
+            if (u.scheme().isEmpty() || u.host().isEmpty()) { // unusable url; try to recover it...
+                QString arg(args->url(i).url());
+
+                kDebug(5010) << "unusable url; try to recover it:" << arg;
+
+                if (arg.lastIndexOf('/') != 0)
+                    arg = arg.right(arg.length() - arg.lastIndexOf('/') - 1);
+
+                if (!arg.contains("://"))
+                    arg.prepend("vnc://"); // vnc was default in kde3 times...
+
+                kDebug(5010) << "recovered url:" << arg;
+
+                u = arg;
+            }
+
+            if (!u.isValid())
+                continue;
+
+            m_mainWindow->newConnection(u, ((args->isSet("fullscreen")) && (args->count() == 1))); 
+        }
+    }
+    
+    return true;
+}
\ Pas de fin de ligne à la fin du fichier
diff -Nru kdenetwork-4.7.97/krdc/application.h kdenetwork-4.7.97-new//krdc/application.h
--- kdenetwork-4.7.97/krdc/application.h	1970-01-01 01:00:00.000000000 +0100
+++ kdenetwork-4.7.97-new//krdc/application.h	2012-01-25 14:06:47.213661048 +0100
@@ -0,0 +1,45 @@
+/****************************************************************************
+**
+** Copyright (C) 2007 - 2008 Urs Wolfer <uwolfer @ kde.org>
+** Copyright (C) 2009 - 2010 Tony Murray <murraytony @ gmail.com>
+**
+** This file is part of KDE.
+**
+** This program is free software; you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation; either version 2 of the License, or
+** (at your option) any later version.
+**
+** This program is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program; see the file COPYING. If not, write to
+** the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+** Boston, MA 02110-1301, USA.
+**
+****************************************************************************/
+
+#ifndef APPLICATION_H
+#define APPLICATION_H
+
+#include <KUniqueApplication>
+
+class MainWindow;
+
+class Application : public KUniqueApplication
+{
+    Q_OBJECT
+public:
+    Application();
+    ~Application();
+
+    int newInstance();
+    
+private:
+    MainWindow *m_mainWindow;
+};
+
+#endif
diff -Nru kdenetwork-4.7.97/krdc/CMakeLists.txt kdenetwork-4.7.97-new//krdc/CMakeLists.txt
--- kdenetwork-4.7.97/krdc/CMakeLists.txt	2011-07-27 20:25:38.000000000 +0200
+++ kdenetwork-4.7.97-new//krdc/CMakeLists.txt	2012-01-25 14:06:47.213661048 +0100
@@ -78,6 +78,7 @@
     systemtrayicon.cpp
     tabbedviewwidget.cpp
     mainwindow.cpp
+    application.cpp
     main.cpp
 )
 
diff -Nru kdenetwork-4.7.97/krdc/main.cpp kdenetwork-4.7.97-new//krdc/main.cpp
--- kdenetwork-4.7.97/krdc/main.cpp	2011-07-27 20:25:38.000000000 +0200
+++ kdenetwork-4.7.97-new//krdc/main.cpp	2012-01-25 14:06:47.213661048 +0100
@@ -22,12 +22,11 @@
 **
 ****************************************************************************/
 
-#include "mainwindow.h"
+#include "application.h"
 
-#include <KApplication>
-#include <KLocale>
 #include <KCmdLineArgs>
 #include <KAboutData>
+#include <KLocale>
 #include <KDebug>
 
 #include <QTime>
@@ -36,6 +35,7 @@
 {
     QTime startupTimer;
     startupTimer.start();
+
     KAboutData aboutData("krdc", 0, ki18n("KRDC"), KDE_VERSION_STRING,
                          ki18n("KDE Remote Desktop Client"), KAboutData::License_GPL,
                          ki18n("(c) 2007-2010, Urs Wolfer\n"
@@ -64,40 +64,8 @@
     options.add("!+[URL]", ki18n("URLs to connect after startup"));
 
     KCmdLineArgs::addCmdLineOptions(options);
-
-    KApplication app;
-
-    MainWindow *mainwindow = new MainWindow;
-    mainwindow->show();
-
-    KCmdLineArgs *args = KCmdLineArgs::parsedArgs();
-
-    if (args->count() > 0) {
-        for (int i = 0; i < args->count(); ++i) {
-            KUrl u(args->url(i));
-
-            if (u.scheme().isEmpty() || u.host().isEmpty()) { // unusable url; try to recover it...
-                QString arg(args->url(i).url());
-
-                kDebug(5010) << "unusable url; try to recover it:" << arg;
-
-                if (arg.lastIndexOf('/') != 0)
-                    arg = arg.right(arg.length() - arg.lastIndexOf('/') - 1);
-
-                if (!arg.contains("://"))
-                    arg.prepend("vnc://"); // vnc was default in kde3 times...
-
-                kDebug(5010) << "recovered url:" << arg;
-
-                u = arg;
-            }
-
-            if (!u.isValid())
-                continue;
-
-            mainwindow->newConnection(u, ((args->isSet("fullscreen")) && (args->count() == 1)));
-        }
-    }
+    
+    Application app;
 
     kDebug(5010) << "########## KRDC ready:" << startupTimer.elapsed() << "ms ##########";
 
