--- a/panel.h	2004-05-24 10:01:48.000000000 +0300
+++ b/panel.h	2011-10-05 11:04:04.195774940 +0300
@@ -17,7 +17,7 @@ enum { POS_NONE, POS_START, POS_END };
 
 #define PANEL_HEIGHT_DEFAULT  26
 #define PANEL_HEIGHT_MAX      200
-#define PANEL_HEIGHT_MIN      16
+#define PANEL_HEIGHT_MIN      14
 
 
 typedef struct {
--- a/panel.c	2011-10-05 11:01:30.127765268 +0300
+++ b/panel.c	2011-10-05 11:02:50.668770327 +0300
@@ -428,6 +428,7 @@ panel_start_gui(panel *p)
 
     /************************/
     /* Window Mapping Point */
+    gdk_window_set_override_redirect(p->topgwin->window, TRUE);
     gtk_widget_show_all(p->topgwin);
     Xclimsg(p->topxwin, a_NET_WM_DESKTOP, 0xFFFFFFFF, 0, 0, 0, 0);
 
@@ -541,6 +542,7 @@ panel_start(panel *p)
     if (!panel_parse_plugin(p))
         RET(0);
 
+    gdk_window_set_override_redirect(p->topgwin->window, TRUE);
     gtk_widget_show_all(p->topgwin);
     print_wmdata(p);
     RET(1);
