diff -uNr src/Pale-Moon/config/system-headers src.infinality/Pale-Moon/config/system-headers
--- src/Pale-Moon/config/system-headers	2015-05-10 00:38:52.000000000 +0900
+++ src.infinality/Pale-Moon/config/system-headers	2015-05-11 11:43:47.000000000 +0900
@@ -411,7 +411,7 @@
 freetype/ftsizes.h
 freetype/ftadvanc.h
 freetype/ftbitmap.h
-freetype/ftxf86.h
+freetype/ftfntfmt.h
 freetype.h
 ftcache.h
 ftglyph.h
@@ -424,7 +424,7 @@
 ftsizes.h
 ftadvanc.h
 ftbitmap.h
-ftxf86.h
+ftfntfmt.h
 fribidi/fribidi.h
 FSp_fopen.h
 fstream
diff -uNr src/Pale-Moon/js/src/config/system-headers src.infinality/Pale-Moon/js/src/config/system-headers
--- src/Pale-Moon/js/src/config/system-headers	2015-05-10 00:39:08.000000000 +0900
+++ src.infinality/Pale-Moon/js/src/config/system-headers	2015-05-11 11:42:15.000000000 +0900
@@ -411,7 +411,7 @@
 freetype/ftsizes.h
 freetype/ftadvanc.h
 freetype/ftbitmap.h
-freetype/ftxf86.h
+freetype/ftfntfmt.h
 freetype.h
 ftcache.h
 ftglyph.h
@@ -424,7 +424,7 @@
 ftsizes.h
 ftadvanc.h
 ftbitmap.h
-ftxf86.h
+ftfntfmt.h
 fribidi/fribidi.h
 FSp_fopen.h
 fstream
