--- Linux-PAM-1.1.6-orig/modules/pam_unix/pam_unix_acct.c	2012-11-10 12:49:26.743723557 +0100
+++ Linux-PAM-1.1.6/modules/pam_unix/pam_unix_acct.c	2012-11-10 12:50:11.470154845 +0100
@@ -48,6 +48,7 @@
 #include <time.h>		/* for time() */
 #include <errno.h>
 #include <sys/wait.h>
+#include <sys/resource.h>
 
 #include <security/_pam_macros.h>
 
--- Linux-PAM-1.1.6-orig/modules/pam_unix/pam_unix_passwd.c	2012-11-10 12:49:26.743723557 +0100
+++ Linux-PAM-1.1.6/modules/pam_unix/pam_unix_passwd.c	2012-11-10 12:55:15.751878688 +0100
@@ -58,6 +58,7 @@
 #include <signal.h>
 #include <errno.h>
 #include <sys/wait.h>
+#include <sys/resource.h>
 
 #include <security/_pam_macros.h>
 
