
diff -ur klineakconfig-0.9-mod/klineakconfig/klineakdef.cpp klineakconfig-0.9/klineakconfig/klineakdef.cpp
--- klineakconfig-0.9-mod/klineakconfig/klineakdef.cpp	2008-06-22 14:05:13.000000000 +0200
+++ klineakconfig-0.9/klineakconfig/klineakdef.cpp	2006-03-14 17:56:32.000000000 +0100
@@ -28,7 +28,7 @@
 #include <lineak/ldef.h>
 #include <lineak/lkey.h>
 #include <lineak/defloader.h>
-
+#include <stdlib.h>
 
 extern "C" {
 #include <X11/Xlib.h>
