diff -ruN gw6c-6_0-RELEASE.orig/tspc-advanced/conf/gw6c.conf.in gw6c-6_0-RELEASE/tspc-advanced/conf/gw6c.conf.in
--- gw6c-6_0-RELEASE.orig/tspc-advanced/conf/gw6c.conf.in	2009-05-26 12:52:48.000000000 +0200
+++ gw6c-6_0-RELEASE/tspc-advanced/conf/gw6c.conf.in	2009-05-26 12:53:47.000000000 +0200
@@ -120,7 +120,7 @@
 #   Directory where the Gateway6 Client will be installed. This value has been 
 #   set during installation.
 #
-gw6_dir=@conf_gw6_dir@
+gw6_dir=/etc/gw6c
 
 #
 # Auto-Retry Connect, Retry Delay and Max Retry Delay:
@@ -224,7 +224,7 @@
 #
 #   broker_list=<file_name>
 #  
-broker_list=tsp-broker-list.txt
+broker_list=/var/run/gw6c/tsp-broker-list.txt
 
 #
 # Last Server Used File Name:
@@ -234,7 +234,7 @@
 #
 #   last_server=<file_name>
 #
-last_server=tsp-last-server.txt
+last_server=/var/run/gw6c/tsp-last-server.txt
 
 #
 # Always Use Last Known Working Server:
@@ -282,10 +282,10 @@
 #     log_file=0
 #     log_syslog=0
 #
-#log_console=
-#log_stderr=
-#log_file=
-#log_syslog=
+log_console=0
+log_stderr=0
+log_file=1
+log_syslog=1
 
 #
 # Log File Name:
@@ -294,7 +294,7 @@
 #
 #   log_filename=<file_name>
 #
-log_filename=gw6c.log
+log_filename=/var/log/gw6c.log
 
 #
 # Log File Rotation:
diff -ruN gw6c-6_0-RELEASE.orig/tspc-advanced/platform/linux/tsp_local.c gw6c-6_0-RELEASE/tspc-advanced/platform/linux/tsp_local.c
--- gw6c-6_0-RELEASE.orig/tspc-advanced/platform/linux/tsp_local.c	2009-05-26 12:52:48.000000000 +0200
+++ gw6c-6_0-RELEASE/tspc-advanced/platform/linux/tsp_local.c	2009-05-26 12:53:47.000000000 +0200
@@ -36,11 +36,11 @@
 
 /* these globals are defined by US used by alot of things in  */
 
-char *FileName  = "gw6c.conf";
+char *FileName  = "/etc/gw6c/gw6c.conf";
 char *ScriptInterpretor = "/bin/sh";
 char *ScriptExtension = "sh";
 char *ScriptDir = NULL;
-char *TspHomeDir = "/usr/local/etc/gw6";
+char *TspHomeDir = "/etc/gw6";
 char DirSeparator = '/';
 
 int indSigHUP = 0;    // Set to 1 when HUP signal is trapped.
diff -ruN gw6c-6_0-RELEASE.orig/tspc-advanced/src/tsp/tsp_auth_passdss.c gw6c-6_0-RELEASE/tspc-advanced/src/tsp/tsp_auth_passdss.c
--- gw6c-6_0-RELEASE.orig/tspc-advanced/src/tsp/tsp_auth_passdss.c	2009-05-26 12:52:48.000000000 +0200
+++ gw6c-6_0-RELEASE/tspc-advanced/src/tsp/tsp_auth_passdss.c	2009-05-26 12:53:13.000000000 +0200
@@ -35,7 +35,7 @@
 
 #define TSP_AUTH_PASSDSS_STRING "PASSDSS-3DES-1"
 #define TSP_AUTH_PASSDSS_BUFFERSIZE 4096
-#define TSPC_DSA_KEYFILE "gw6ckeys.pub"
+#define TSPC_DSA_KEYFILE "/var/run/gw6c/gw6ckeys.pub"
 
 #if defined(WIN32) && !defined(WINCE)
 extern BOOL IsService;    // Declared in winpc/tsp_local.c
diff -ruN gw6c-6_0-RELEASE.orig/tspc-advanced/template/linux.sh gw6c-6_0-RELEASE/tspc-advanced/template/linux.sh
--- gw6c-6_0-RELEASE.orig/tspc-advanced/template/linux.sh	2009-05-26 12:52:48.000000000 +0200
+++ gw6c-6_0-RELEASE/tspc-advanced/template/linux.sh	2009-05-26 12:53:47.000000000 +0200
@@ -78,7 +78,7 @@
 
 ifconfig=/sbin/ifconfig
 route=/sbin/route
-ipconfig=/sbin/ip
+ipconfig=/usr/sbin/ip
 rtadvd=/usr/sbin/radvd
 rtadvd_pid=/var/run/radvd/radvd.pid
 sysctl=/sbin/sysctl
