diff --git a/ant/build.gant b/ant/build.gant
index de47cb8..2682f2b 100644
--- a/ant/build.gant
+++ b/ant/build.gant
@@ -657,13 +657,29 @@ def initPlugins(){
 
   defaults.each{ name -> pluginInclude(name)}
 
-  new File(getVariable('eclipse') + '/features').eachDir{ dir ->
-    features << dir.getName().replaceFirst(/_\d.*/, '')
+  try {
+    eclipseFeaturesPath = getVariable('eclipse') + '/features'
+    new File(eclipseFeaturesPath).eachDir{ dir ->
+      features << dir.getName().replaceFirst(/_\d.*/, '')
+    }
+  } catch (IOException e) {
+    ant.echo("Error reading from ${eclipseFeaturesPath}");
   }
-  new File(getVariable('eclipse') + '/dropins').eachDir{ dropin ->
-    new File(dropin.getAbsolutePath() + '/eclipse/features').eachFile{ file ->
-      features << file.getName().replaceFirst(/_\d.*/, '')
+
+  try {
+    eclipseDropinsPath = getVariable('eclipse') + '/dropins'
+    new File(eclipseDropinsPath).eachDir{ dropin ->
+      try {
+        eclipseDropinFeaturePath = dropin.getAbsolutePath() + '/eclipse/features'
+        new File(eclipseDropinFeaturePath).eachFile{ file ->
+          features << file.getName().replaceFirst(/_\d.*/, '')
+        }
+      } catch (IOException e) {
+        ant.echo("Error reading from ${eclipseDropinFeaturePath}")
+      }
     }
+  } catch (IOException e) {
+    ant.echo("Error reading from ${eclipseDropinsPath}");
   }
 
   new File('.').eachDirMatch(~/org\.eclim\.\w*/){ dir ->
