Index: main.cpp
===================================================================
--- main.cpp	(revision 657)
+++ main.cpp	(working copy)
@@ -47,8 +47,6 @@
 main( int argc,
       char ** argv )
 {
-    std::locale::global( std::locale::classic() );
-
     std::cout << PACKAGE"-"VERSION << "\n\n"
               << "Copyright (C) 2009 - 2012 RoboCup Soccer Simulator Maintenance Group.\n"
               << std::endl;
@@ -55,6 +53,8 @@
 
     QApplication app( argc, argv );
 
+    std::locale::global( std::locale::classic() );
+
     if ( ! Options::instance().parseCmdLine( argc, argv ) )
     {
         return 1;
