commit 2a34ed66fed772631f655d86e2d182dc10766132
Author: Carl Hetherington <cth@carlh.net>
Date:   Fri Nov 15 13:36:34 2013 +0000

    Update for libquickmail 0.1.16.

diff --git a/src/lib/kdm.cc b/src/lib/kdm.cc
index 05c4cd0..4d8f297 100644
--- a/src/lib/kdm.cc
+++ b/src/lib/kdm.cc
@@ -229,7 +229,7 @@ email_kdms (
 		boost::algorithm::replace_all (body, "$DCP_NAME", film->dcp_name ());
 		
 		quickmail_set_body (mail, body.c_str());
-		quickmail_add_attachment_file (mail, zip_file.string().c_str());
+		quickmail_add_attachment_file (mail, zip_file.string().c_str(), "application/zip");
 		char const* error = quickmail_send (mail, Config::instance()->mail_server().c_str(), 25, "", "");
 		if (error) {
 			quickmail_destroy (mail);
