--- src/dcpomatic-1.70.0/src/wx/wscript	2014-06-27 16:55:21.833189938 +0200
+++ src/dcpomatic-1.70.0/src/wx/wscript_patched	2014-06-27 16:55:53.863189059 +0200
@@ -67,8 +67,8 @@
     conf.in_msg = 1
     wx_version = conf.check_cfg(package='', path=conf.options.wx_config, args='--version').strip()
     conf.im_msg = 0
-    if wx_version != '3.0.0':
-        conf.fatal('wxwidgets version 3.0.0 is required; %s found' % wx_version)
+    if wx_version < '3.0.0':
+        conf.fatal('at least wxwidgets version 3.0.0 is required; %s found' % wx_version)
 
 def build(bld):
     if bld.env.BUILD_STATIC:
