diff -Naur gnome-calendar-3.15.3.1.orig/configure.ac gnome-calendar-3.15.3.1/configure.ac
--- gnome-calendar-3.15.3.1.orig/configure.ac	2014-12-24 21:15:35.000000000 +0100
+++ gnome-calendar-3.15.3.1/configure.ac	2015-01-08 14:43:39.245222763 +0100
@@ -81,7 +81,7 @@
 MAINTAINER_COMPILER_FLAGS=${MAINTAINER_COMPILER_FLAGS#*  }
 AC_SUBST(MAINTAINER_COMPILER_FLAGS)
 
-GLIB_MIN_VERSION=2.43.2
+GLIB_MIN_VERSION=2.42.0
 GTK_MIN_VERSION=3.14.0
 ECAL_REQUIRED=3.5.3
 EDATASERVER_REQUIRED=3.5.3
diff -Naur gnome-calendar-3.15.3.1.orig/src/gcal-manager.c gnome-calendar-3.15.3.1/src/gcal-manager.c
--- gnome-calendar-3.15.3.1.orig/src/gcal-manager.c	2014-12-24 19:32:06.000000000 +0100
+++ gnome-calendar-3.15.3.1/src/gcal-manager.c	2015-01-08 14:43:19.751417068 +0100
@@ -267,6 +267,21 @@
     }
 }
 
+static gboolean
+strv_contains (const gchar * const *strv,
+               const gchar  *s)
+{
+  gint i;
+
+  for (i = 0; strv[i]; i++)
+    {
+      if (g_strcmp0 (strv[i], s) == 0)
+        return TRUE;
+    }
+
+  return FALSE;
+}
+
 static void
 on_client_connected (GObject      *source_object,
                      GAsyncResult *result,
@@ -300,7 +315,7 @@
       /* notify the readonly property */
       g_signal_connect (client, "notify::readonly", G_CALLBACK (on_client_readonly_changed), user_data);
 
-      if (g_strv_contains ((const gchar * const *) priv->disabled_sources, e_source_get_uid (source)))
+      if (strv_contains ((const gchar * const *) priv->disabled_sources, e_source_get_uid (source)))
         {
           unit->enabled = FALSE;
         }
