Index: exactimage-0.8.5/edisplay/edisplay.cc
===================================================================
--- exactimage-0.8.5.orig/edisplay/edisplay.cc	2012-05-30 04:26:28.824014192 +0000
+++ exactimage-0.8.5/edisplay/edisplay.cc	2012-05-30 04:26:45.740005367 +0000
@@ -31,6 +31,7 @@
 #include <algorithm>
 #include <iostream>
 #include <sstream>
+#include <unistd.h>
 
 #include "ArgumentList.hh"
 using namespace Utility;
