diff -up vavoom-1.30/source/p_switch.cpp~ vavoom-1.30/source/p_switch.cpp
--- vavoom-1.30/source/p_switch.cpp~	2008-05-18 18:46:18.000000000 +0200
+++ vavoom-1.30/source/p_switch.cpp	2009-03-20 09:31:58.000000000 +0100
@@ -54,6 +54,7 @@ enum EBWhere
 class VButton : public VThinker
 {
 	DECLARE_CLASS(VButton, VThinker, 0)
+	VButton() {}
 
 	vint32		Side;
 	vuint8		Where;
diff -up vavoom-1.30/source/ui_widget.h~ vavoom-1.30/source/ui_widget.h
--- vavoom-1.30/source/ui_widget.h~	2008-01-03 09:36:55.000000000 +0100
+++ vavoom-1.30/source/ui_widget.h	2009-03-20 09:31:59.000000000 +0100
@@ -45,6 +45,7 @@ struct VClipRect
 class VWidget : public VObject
 {
 	DECLARE_CLASS(VWidget, VObject, 0)
+	VWidget() {}
 
 private:
 	//	Parent container widget.
