Index: cblrepo/distributed-process-task-0.1.0/distributed-process-task.cabal
===================================================================
--- cblrepo.orig/distributed-process-task-0.1.0/distributed-process-task.cabal
+++ cblrepo/distributed-process-task-0.1.0/distributed-process-task.cabal
@@ -33,9 +33,9 @@ library
                    base >= 4.4 && < 5,
                    data-accessor >= 0.2.2.3,
                    distributed-process >= 0.5.2 && < 0.6,
-                   distributed-process-extras >= 0.1.1 && < 0.2,
+                   distributed-process-extras >= 0.1.1,
                    distributed-process-async >= 0.2 && < 0.3,
-                   distributed-process-client-server >= 0.1.1 && <0.2,
+                   distributed-process-client-server >= 0.1.1,
                    binary >= 0.6.3.0 && < 0.8,
                    deepseq >= 1.3.0.1 && < 1.4,
                    mtl,
