diff -ur a/fldigi-3.20.34/src/main.cxx b/fldigi-3.20.34/src/main.cxx
--- a/fldigi-3.20.34/src/main.cxx	2010-12-28 02:58:32.000000000 +0200
+++ b/fldigi-3.20.34/src/main.cxx	2011-01-19 23:30:02.000000000 +0200
@@ -275,7 +275,7 @@
 	Fl::scheme(progdefaults.ui_scheme.c_str());
 	progdefaults.initFonts();
 
-	dxcc_open(string(HomeDir).append("cty.dat").c_str());
+	dxcc_open("/usr/share/cty/cty.dat");
 	qsl_open(string(HomeDir).append("lotw1.txt").c_str(), QSL_LOTW);
 	if (!qsl_open(string(HomeDir).append("eqsl.txt").c_str(), QSL_EQSL))
 		qsl_open(string(HomeDir).append("AGMemberList.txt").c_str(), QSL_EQSL);
