diff -ur a/fldigi-3.21.11/src/main.cxx b/fldigi-3.21.11/src/main.cxx
--- a/fldigi-3.21.11/src/main.cxx	2011-06-11 23:11:57.000000000 +0300
+++ b/fldigi-3.21.11/src/main.cxx	2011-06-22 23:45:58.000000000 +0300
@@ -276,7 +276,7 @@
 	Fl::scheme(progdefaults.ui_scheme.c_str());
 	progdefaults.initFonts();
 
-	dxcc_open(string(HomeDir).append("cty.dat").c_str());
+	dxcc_open("/usr/share/cty/cty.dat");
 	qsl_open(string(HomeDir).append("lotw1.txt").c_str(), QSL_LOTW);
 	if (!qsl_open(string(HomeDir).append("eqsl.txt").c_str(), QSL_EQSL))
 		qsl_open(string(HomeDir).append("AGMemberList.txt").c_str(), QSL_EQSL);
