diff -ur ./src/QGlib/CMakeLists.txt ../RELEASE-0.10.1-FIXED/src/QGlib/CMakeLists.txt
--- ./src/QGlib/CMakeLists.txt	2011-01-23 13:35:19.000000000 +0100
+++ ../RELEASE-0.10.1-FIXED/src/QGlib/CMakeLists.txt	2011-12-21 03:17:38.446929193 +0100
@@ -29,6 +29,10 @@
     error.h       Error
 )
 
+set(QtGLib_MOC_HEADERS
+    connect.h
+)
+
 file(GLOB QtGLib_CODEGEN_HEADERS RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "*.h")
 list(REMOVE_ITEM QtGLib_CODEGEN_HEADERS "connectimpl.h" "emitimpl.h")
 
@@ -45,7 +49,8 @@
 run_codegen("QGlib" "${QtGLib_CODEGEN_INCLUDES}" "${QtGLib_CODEGEN_HEADERS}")
 
 # Build and link QtGLib
-automoc4_add_library(QtGLib ${SHARED_OR_STATIC} ${QtGLib_SRCS})
+qt4_wrap_cpp(QtGLib_MOC_SRCS ${QtGLib_MOC_HEADERS} OPTIONS -DBOOST_TT_HAS_OPERATOR_HPP_INCLUDED)
+add_library(QtGLib ${SHARED_OR_STATIC} ${QtGLib_SRCS} ${QtGLib_MOC_SRCS})
 set_target_properties(QtGLib PROPERTIES OUTPUT_NAME QtGLib-${QTGLIB_API_VERSION}
                                           SOVERSION ${QTGLIB_SOVERSION}
                                             VERSION ${QTGSTREAMER_VERSION})
diff -ur ./src/QGlib/connect.cpp ../RELEASE-0.10.1-FIXED/src/QGlib/connect.cpp
--- ./src/QGlib/connect.cpp	2011-01-23 13:35:19.000000000 +0100
+++ ../RELEASE-0.10.1-FIXED/src/QGlib/connect.cpp	2011-12-21 03:17:51.583595779 +0100
@@ -505,5 +505,3 @@
 
 } //namespace Private
 } //namespace QGlib
-
-#include "connect.moc"
diff -ur ./src/QGst/CMakeLists.txt ../RELEASE-0.10.1-FIXED/src/QGst/CMakeLists.txt
--- ./src/QGst/CMakeLists.txt	2011-01-23 13:35:19.000000000 +0100
+++ ../RELEASE-0.10.1-FIXED/src/QGst/CMakeLists.txt	2011-12-21 03:18:11.010262334 +0100
@@ -37,6 +37,10 @@
     Ui/videowidget.cpp
 )
 
+set(QtGStreamerUi_MOC_HEADERS
+    Ui/videowidget.h
+)
+
 set(QtGStreamerUtils_SRCS
     Utils/applicationsink.cpp
     Utils/applicationsource.cpp
@@ -126,7 +130,8 @@
 target_link_libraries(QtGStreamer LINK_INTERFACE_LIBRARIES ${QTGLIB_LIBRARY})
 
 # Build and link QtGStreamerUi
-automoc4_add_library(QtGStreamerUi ${SHARED_OR_STATIC} ${QtGStreamerUi_SRCS})
+qt4_wrap_cpp(QtGStreamerUi_MOC_SRCS ${QtGStreamerUi_MOC_HEADERS} OPTIONS -DBOOST_TT_HAS_OPERATOR_HPP_INCLUDED)
+add_library(QtGStreamerUi ${SHARED_OR_STATIC} ${QtGStreamerUi_SRCS} ${QtGStreamerUi_HEADERS})
 set_target_properties(QtGStreamerUi PROPERTIES OUTPUT_NAME QtGStreamerUi-${QTGSTREAMER_API_VERSION}
                                                  SOVERSION ${QTGSTREAMER_UI_SOVERSION}
                                                    VERSION ${QTGSTREAMER_VERSION})
