Changes to Makefile required for building the Debian package

--- simutrans-pak64-102.2.orig/Makefile
+++ simutrans-pak64-102.2/Makefile
@@ -2,7 +2,7 @@
 #   make clean all archives
 # to get fresh and ready to deploy .tbz2 and .zip archives
 
-MAKEOBJ ?= ./makeobj
+MAKEOBJ ?= /usr/bin/makeobj
 
 DESTDIR  ?= simutrans
 PAKDIR   ?= $(DESTDIR)/pak
@@ -42,7 +42,7 @@
 
 .PHONY: $(DIRS) copy tar zip
 
-all: copy $(DIRS)
+all: $(DIRS)
 
 archives: tar zip
 
@@ -59,12 +59,13 @@
 
 copy:
 	@echo "===> COPY"
-	@mkdir -p $(PAKDIR)/sound $(PAKDIR)/text $(PAKDIR)/config $(PAKDIR)/scenario
-	@cp -p compat/compat.tab $(PAKDIR)
-	@cp -p sound/* $(PAKDIR)/sound
-	@cp -p config/* $(PAKDIR)/config
-	@cp -p scenario/* $(PAKDIR)/scenario
-	@cp -p pak.text/* $(PAKDIR)/text
+	install -m 755 -d $(PAKDIR)/sound $(PAKDIR)/text $(PAKDIR)/config $(PAKDIR)/scenario
+	install -m 644 compat/compat.tab $(PAKDIR)
+	install -m 644 sound/* $(PAKDIR)/sound
+	install -m 644 config/* $(PAKDIR)/config
+	install -m 644 scenario/* $(PAKDIR)/scenario
+	install -m 644 pak.text/* $(PAKDIR)/text
+	install -m 644 simutrans/pak/* $(PAKDIR)/
 
 $(DIRS64):
 	@echo "===> PAK64 $@"
