--- pt1-b14397800eae.old/recpt1/pt1_dev.h	2012-06-27 06:56:33.000000000 +0900
+++ pt1-b14397800eae/recpt1/pt1_dev.h	2012-09-22 01:01:27.195716727 +0900
@@ -7,20 +7,20 @@
     "/dev/pt1video0",
     "/dev/pt1video5",
     "/dev/pt1video4",
-    "/dev/pt1video9",
-    "/dev/pt1video8",
-    "/dev/pt1video13",
-    "/dev/pt1video12"
+    "/dev/pt3video1",
+    "/dev/pt3video0",
+    "/dev/pt3video4",
+    "/dev/pt3video5"
 };
 char *isdb_t_dev[NUM_ISDB_T_DEV] = {
     "/dev/pt1video2",
     "/dev/pt1video3",
     "/dev/pt1video6",
     "/dev/pt1video7",
-    "/dev/pt1video10",
-    "/dev/pt1video11",
-    "/dev/pt1video14",
-    "/dev/pt1video15"
+    "/dev/pt3video2",
+    "/dev/pt3video3",
+    "/dev/pt3video6",
+    "/dev/pt3video7"
 };
 
 // 変換テーブル(ISDB-T用)
