--- usr/share/autowifi/action	2009-09-27 15:45:10.000000000 +0200
+++ usr/share/autowifi/action.new	2010-04-09 14:44:37.137491193 +0200
@@ -10,7 +10,7 @@
 
 IFCONFIG="/sbin/ifconfig"
 ROUTE="/sbin/route"
-DHCLIENT="/sbin/dhclient"
+DHCLIENT="/sbin/dhcpcd"
 
 # obtain the ssid
 INTERFACE="${1}"
@@ -56,7 +56,7 @@
   lecho "Using profile '${PROFILE}'"
   case "${NETWORK}" in
     "dhcp")
-      ${DHCLIENT} -q -pf /var/run/dhclient-${INTERFACE}.pid ${INTERFACE}
+      ${DHCLIENT} ${INTERFACE}
       ;;
     "static")
       ${IFCONFIG} ${INTERFACE} ${ADDRESS} $([ -n "${NETMASK}" ] && echo -n "netmask ${NETMASK}") up
@@ -74,16 +74,17 @@
       lecho "Unknown profile type \"${NETWORK}\", assuming \"custom\""
       ;;
   esac
-  # execute "connect" script
-  autowifi_connect "${INTERFACE}" "${SSID}" "${PROFILE}"
+
   # execute network-independant "connect" script
   autowifi_generic_event_connected "${INTERFACE}" "${SSID}" "${PROFILE}"
+  # execute "connect" script
+  autowifi_connect "${INTERFACE}" "${SSID}" "${PROFILE}"
+
 elif [ "${ACTION}" = "DISCONNECT" ]; then
   # NOTE: Never bring the interface down
   case "${NETWORK}" in
     "dhcp")
-      [ -f /var/run/dhclient-${INTERFACE}.pid ] && kill $(cat /var/run/dhclient-${INTERFACE}.pid)
-      rm -f /var/run/dhclient-${INTERFACE}.pid
+      ${DHCLIENT} -k ${INTERFACE}
       ;;
     "static")
       [ -n "${DNS}" -a -f /etc/resolv.conf.wpabak ] && mv /etc/resolv.conf.wpabak /etc/resolv.conf
