diff -U 3 -H -d -r -N -- a/configure b/configure
--- a/configure	2013-01-20 00:08:40.000000000 -0600
+++ b/configure	2014-06-21 08:38:35.000000000 -0600
@@ -15996,10 +15996,10 @@
 find the right location for the libpq library and its headers.
 " "$LINENO" 5
 fi
-with_postgres_include=`$PG_CONFIG --includedir`
+with_postgres_include=/usr/${ARCH}/include
 { $as_echo "$as_me:${as_lineno-$LINENO}: using PostgreSQL headers at $with_postgres_include" >&5
 $as_echo "$as_me: using PostgreSQL headers at $with_postgres_include" >&6;}
-with_postgres_lib=`$PG_CONFIG --libdir`
+with_postgres_lib=/usr/${ARCH}/lib
 { $as_echo "$as_me:${as_lineno-$LINENO}: using PostgreSQL libraries at $with_postgres_lib" >&5
 $as_echo "$as_me: using PostgreSQL libraries at $with_postgres_lib" >&6;}
 
diff -U 3 -H -d -r -N -- a/configure.ac b/configure.ac
--- a/configure.ac	2013-01-20 00:08:31.000000000 -0600
+++ b/configure.ac	2014-06-21 08:38:40.000000000 -0600
@@ -354,9 +354,9 @@
 find the right location for the libpq library and its headers.
 ])
 fi
-with_postgres_include=`$PG_CONFIG --includedir`
+with_postgres_include=/usr/${ARCH}/include
 AC_MSG_NOTICE([using PostgreSQL headers at $with_postgres_include])
-with_postgres_lib=`$PG_CONFIG --libdir`
+with_postgres_lib=/usr/${ARCH}/lib
 AC_MSG_NOTICE([using PostgreSQL libraries at $with_postgres_lib])
 
 AC_SUBST(with_postgres_lib)
diff -U 3 -H -d -r -N -- a/configure.ac.in b/configure.ac.in
--- a/configure.ac.in	2013-01-17 01:25:15.000000000 -0600
+++ b/configure.ac.in	2014-06-21 08:38:46.000000000 -0600
@@ -354,9 +354,9 @@
 find the right location for the libpq library and its headers.
 ])
 fi
-with_postgres_include=`$PG_CONFIG --includedir`
+with_postgres_include=/usr/${ARCH}/include
 AC_MSG_NOTICE([using PostgreSQL headers at $with_postgres_include])
-with_postgres_lib=`$PG_CONFIG --libdir`
+with_postgres_lib=/usr/${ARCH}/lib
 AC_MSG_NOTICE([using PostgreSQL libraries at $with_postgres_lib])
 
 AC_SUBST(with_postgres_lib)
