diff -u a/MonoGame.Framework/Desktop/Input/Mouse.cs b/MonoGame.Framework/Desktop/Input/Mouse.cs
--- a/MonoGame.Framework/Desktop/Input/Mouse.cs	2012-03-27 10:37:10.000000000 -0700
+++ b/MonoGame.Framework/Desktop/Input/Mouse.cs	2012-04-29 18:15:45.286650189 -0700
@@ -37,7 +37,7 @@
 purpose and non-infringement.
 */
 #endregion License
-﻿
+
 using MouseInfo = OpenTK.Input.Mouse;
 using System;
 
diff -u a/MonoGame.Framework/Desktop/Input/MouseState.cs b/MonoGame.Framework/Desktop/Input/MouseState.cs
--- a/MonoGame.Framework/Desktop/Input/MouseState.cs	2012-03-27 10:37:10.000000000 -0700
+++ b/MonoGame.Framework/Desktop/Input/MouseState.cs	2012-04-29 18:16:07.146651062 -0700
@@ -37,7 +37,7 @@
 purpose and non-infringement.
 */
 #endregion License
-﻿using System;
+using System;
 
 namespace Microsoft.Xna.Framework.Input
 {    
