diff -u a/MonoGame.Framework/Media/MediaPlayer.cs b/MonoGame.Framework/Media/MediaPlayer.cs
--- a/MonoGame.Framework/Media/MediaPlayer.cs	2012-03-27 10:37:18.000000000 -0700
+++ b/MonoGame.Framework/Media/MediaPlayer.cs	2012-04-29 18:17:59.873322560 -0700
@@ -41,7 +41,7 @@
 using System;
 using Microsoft.Xna.Framework.Audio;
 
-﻿namespace Microsoft.Xna.Framework.Media
+namespace Microsoft.Xna.Framework.Media
 {
     public static class MediaPlayer
     {
diff -u a/MonoGame.Framework/Media/MediaState.cs b/MonoGame.Framework/Media/MediaState.cs
--- a/MonoGame.Framework/Media/MediaState.cs	2012-03-27 10:37:18.000000000 -0700
+++ b/MonoGame.Framework/Media/MediaState.cs	2012-04-29 18:18:24.676656938 -0700
@@ -39,7 +39,7 @@
 #endregion License
 
 using System;
-﻿
+
 namespace Microsoft.Xna.Framework.Media
 {
     public enum MediaState
diff -u a/MonoGame.Framework/Media/Playlist.cs b/MonoGame.Framework/Media/Playlist.cs
--- a/MonoGame.Framework/Media/Playlist.cs	2012-03-27 10:37:18.000000000 -0700
+++ b/MonoGame.Framework/Media/Playlist.cs	2012-04-29 18:18:40.849991399 -0700
@@ -38,7 +38,7 @@
 */
 #endregion License
 
-using System;﻿
+using System;
 
 namespace Microsoft.Xna.Framework.Media
 {
diff -u a/MonoGame.Framework/Media/PlaylistCollection.cs b/MonoGame.Framework/Media/PlaylistCollection.cs
--- a/MonoGame.Framework/Media/PlaylistCollection.cs	2012-03-27 10:37:18.000000000 -0700
+++ b/MonoGame.Framework/Media/PlaylistCollection.cs	2012-04-29 18:19:00.733324976 -0700
@@ -41,7 +41,7 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
-﻿
+
 namespace Microsoft.Xna.Framework.Media
 {
 
diff -u a/MonoGame.Framework/Media/Song.cs b/MonoGame.Framework/Media/Song.cs
--- a/MonoGame.Framework/Media/Song.cs	2012-03-27 10:37:18.000000000 -0700
+++ b/MonoGame.Framework/Media/Song.cs	2012-04-29 18:19:21.933326019 -0700
@@ -42,7 +42,7 @@
 using System.IO;
 
 using Microsoft.Xna.Framework.Audio;
-﻿
+
 namespace Microsoft.Xna.Framework.Media
 {
     public class Song : IEquatable<Song>, IDisposable
