--- cmake_install.cmake.orig	2012-07-23 09:42:06.000000000 +0200
+++ cmake_install.cmake	2012-07-23 10:04:26.254650113 +0200
@@ -103,16 +103,16 @@
 IF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
   
      include(BundleUtilities)
-     message( "${CMAKE_INSTALL_PREFIX}/Unvanquished/daemon.i386" )
-     fixup_bundle("${CMAKE_INSTALL_PREFIX}/Unvanquished/daemon.i386"   "${CMAKE_INSTALL_PREFIX}/Unvanquished/librendererGLi386.so;${CMAKE_INSTALL_PREFIX}/Unvanquished/librendererGL3i386.so"   "")
+     message( "$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/Unvanquished/daemon.i386" )
+     fixup_bundle("$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/Unvanquished/daemon.i386"   "$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/Unvanquished/librendererGLi386.so;$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/Unvanquished/librendererGL3i386.so"   "")
      
 ENDIF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
 
 IF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
   
      include(BundleUtilities)
-     message( "${CMAKE_INSTALL_PREFIX}/Unvanquished/daemonded.i386" )
-     fixup_bundle("${CMAKE_INSTALL_PREFIX}/Unvanquished/daemonded.i386"   "${CMAKE_INSTALL_PREFIX}/Unvanquished/librendererGLi386.so;${CMAKE_INSTALL_PREFIX}/Unvanquished/librendererGL3i386.so"   "")
+     message( "$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/Unvanquished/daemonded.i386" )
+     fixup_bundle("$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/Unvanquished/daemonded.i386"   "$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/Unvanquished/librendererGLi386.so;$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/Unvanquished/librendererGL3i386.so"   "")
      
 ENDIF(NOT CMAKE_INSTALL_COMPONENT OR "${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified")
 
