diff -Naur grafana-2.0.2.ori/build.go grafana-2.0.2/build.go
--- grafana-2.0.2.ori/build.go	2015-04-22 17:51:50.000000000 +0200
+++ grafana-2.0.2/build.go	2015-04-29 21:31:26.145099831 +0200
@@ -74,7 +74,6 @@
 		case "package":
 			//verifyGitRepoIsClean()
 			grunt("release")
-			createLinuxPackages()
 
 		case "latest":
 			makeLatestDistCopies()
diff -Naur grafana-2.0.2.ori/conf/defaults.ini grafana-2.0.2/conf/defaults.ini
--- grafana-2.0.2.ori/conf/defaults.ini	2015-04-22 17:51:50.000000000 +0200
+++ grafana-2.0.2/conf/defaults.ini	2015-04-29 21:31:58.114442024 +0200
@@ -9,11 +9,11 @@
 [paths]
 # Path to where grafana can store temp files, sessions, and the sqlite3 db (if that is useD)
 #
-data = data
+data = /var/lib/grafana
 #
 # Directory where grafana can store logs
 #
-logs = data/log
+logs = /var/log/grafana
 
 #################################### Server ####################################
 [server]
@@ -36,7 +36,7 @@
 router_logging = false
 
 # the path relative working path
-static_root_path = public
+static_root_path = /usr/share/grafana/public
 
 # enable gzip
 enable_gzip = false
@@ -58,7 +58,7 @@
 ssl_mode = disable
 
 # For "sqlite3" only, path relative to data_path setting
-path = grafana.db
+path = /var/lib/grafana/grafana.db
 
 #################################### Session ####################################
 [session]
diff -Naur grafana-2.0.2.ori/pkg/components/renderer/renderer.go grafana-2.0.2/pkg/components/renderer/renderer.go
--- grafana-2.0.2.ori/pkg/components/renderer/renderer.go	2015-04-22 17:51:50.000000000 +0200
+++ grafana-2.0.2/pkg/components/renderer/renderer.go	2015-04-29 21:32:26.993847808 +0200
@@ -21,7 +21,7 @@
 
 func RenderToPng(params *RenderOpts) (string, error) {
 	log.Info("PhantomRenderer::renderToPng url %v", params.Url)
-	binPath, _ := filepath.Abs(filepath.Join(setting.PhantomDir, "phantomjs"))
+	binPath := "/usr/bin/phantomjs"
 	scriptPath, _ := filepath.Abs(filepath.Join(setting.PhantomDir, "render.js"))
 	pngPath, _ := filepath.Abs(filepath.Join(setting.ImagesDir, util.GetRandomString(20)))
 	pngPath = pngPath + ".png"
