diff -Naur grafana-2.0.0-beta3.ori/build.go grafana-2.0.0-beta3/build.go
--- grafana-2.0.0-beta3.ori/build.go	2015-04-12 16:23:43.000000000 +0200
+++ grafana-2.0.0-beta3/build.go	2015-04-13 21:48:17.784798265 +0200
@@ -71,7 +71,6 @@
 		case "package":
 			//verifyGitRepoIsClean()
 			grunt("release", "--pkgVer="+version)
-			createLinuxPackages()
 
 		case "latest":
 			makeLatestDistCopies()
diff -Naur grafana-2.0.0-beta3.ori/conf/defaults.ini grafana-2.0.0-beta3/conf/defaults.ini
--- grafana-2.0.0-beta3.ori/conf/defaults.ini	2015-04-12 16:23:43.000000000 +0200
+++ grafana-2.0.0-beta3/conf/defaults.ini	2015-04-13 21:52:57.832497080 +0200
@@ -9,11 +9,11 @@
 [paths]
 # Path to where grafana can store temp files, sessions, and the sqlite3 db (if that is useD)
 #
-data = data
+data = /var/lib/grafana
 #
 # Directory where grafana can store logs
 #
-logs = data/log
+logs = /var/log/grafana
 
 #################################### Server ####################################
 [server]
@@ -36,7 +36,7 @@
 router_logging = false
 
 # the path relative working path
-static_root_path = public
+static_root_path = /usr/share/grafana/public
 
 # enable gzip
 enable_gzip = false
@@ -58,7 +58,7 @@
 ssl_mode = disable
 
 # For "sqlite3" only, path relative to data_path setting
-path = grafana.db
+path = /var/lib/grafana/grafana.db
 
 #################################### Session ####################################
 [session]
diff -Naur grafana-2.0.0-beta3.ori/pkg/components/renderer/renderer.go grafana-2.0.0-beta3/pkg/components/renderer/renderer.go
--- grafana-2.0.0-beta3.ori/pkg/components/renderer/renderer.go	2015-04-12 16:23:43.000000000 +0200
+++ grafana-2.0.0-beta3/pkg/components/renderer/renderer.go	2015-04-13 22:07:12.055339411 +0200
@@ -21,7 +21,7 @@
 
 func RenderToPng(params *RenderOpts) (string, error) {
 	log.Info("PhantomRenderer::renderToPng url %v", params.Url)
-	binPath, _ := filepath.Abs(filepath.Join(setting.PhantomDir, "phantomjs"))
+	binPath := "/usr/bin/phantomjs"
 	scriptPath, _ := filepath.Abs(filepath.Join(setting.PhantomDir, "render.js"))
 	pngPath, _ := filepath.Abs(filepath.Join(setting.ImagesDir, util.GetRandomString(20)))
 	pngPath = pngPath + ".png"
