--- setup_pjsip.py	2014-08-05 13:31:13.488011554 +0200
+++ setup_pjsip.py.new	2014-08-05 13:31:03.638349274 +0200
@@ -150,6 +150,8 @@
             cmd = ["bash", "configure", "--disable-video"]
         else:
             cmd = ["./configure", "--disable-video"]
+        self.distutils_exec_process(["chmod", "u+x", "aconfigure"], silent=not self.pjsip_verbose_build, cwd=self.build_dir, env=env)
+        self.distutils_exec_process(["chmod", "u+x", "configure"], silent=not self.pjsip_verbose_build, cwd=self.build_dir, env=env)
         self.distutils_exec_process(cmd, silent=not self.pjsip_verbose_build, cwd=self.build_dir, env=env)
         if "#define PJ_HAS_SSL_SOCK 1\n" not in open(os.path.join(self.build_dir, "pjlib", "include", "pj", "compat", "os_auto.h")).readlines():
             os.remove(os.path.join(self.build_dir, "build.mak"))
