--- png_textures.c.ORIG	2011-06-17 21:56:49.000000000 +1000
+++ png_textures.c	2011-06-17 21:57:03.000000000 +1000
@@ -30,13 +30,13 @@
   info_ptr = png_create_info_struct(png_ptr);
 
   if (setjmp(png_jmpbuf(png_ptr))) {
-    png_destroy_read_struct(&png_ptr, &info_ptr, png_infopp_NULL);
+    png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
     fclose(fp);
     return 0;
   }
 
   png_init_io(png_ptr, fp);
-  png_read_png(png_ptr, info_ptr, PNG_TRANSFORM_EXPAND, png_voidp_NULL);
+  png_read_png(png_ptr, info_ptr, PNG_TRANSFORM_EXPAND, NULL);
 
   row_pointers = png_get_rows(png_ptr, info_ptr);
 
@@ -69,7 +69,7 @@
 	   png_get_rowbytes(png_ptr, info_ptr));
   }
 
-  png_destroy_read_struct(&png_ptr, &info_ptr, png_infopp_NULL);
+  png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
   fclose(fp);
 
   return r;
