--- nxnode.old	2010-12-14 02:24:52.000000000 +0300
+++ nxnode	2010-12-15 01:40:27.000000000 +0300
@@ -1531,6 +1531,10 @@
 
 node_start_monitor()
 {
+#oldbay//
+# wait timeout for start suspend
+	sleep $NODE_START_MONITOR_WAIT_TIMEOUT
+#oldbay\\
 	nxlog "$FUNCNAME ($$): staring"
 	if [ "$ENABLE_1_5_0_BACKEND" = "1" ]
 	then
@@ -1875,7 +1879,7 @@
 if which "$NODE_AUTOSTART" >/dev/null 2>&1
 then
 	# go into background immediately
-	NXSESSIONID="$sess_id" DISPLAY=:$display "$NODE_AUTOSTART" "$1" >/dev/null 2>&1 &
+	NXSESSIONID="$sess_id" DISPLAY=:$display cmd_node_autostart &
 	# dont't wait for this child!
 	disown $!
 fi
@@ -1913,6 +1917,15 @@
 nxlog "$FUNCNAME ($$): end"
 }
 
+cmd_node_autostart()
+{
+#oldbay//
+# wait to run node_autostart
+	sleep $NODE_START_MONITOR_WAIT_TIMEOUT
+	$NODE_AUTOSTART >/dev/null 2>&1 &
+#oldbay\\
+}
+
 #
 # -----------------------------------------------------------------------------
 # cmd_node functions - changes lots of small variables
