--- src/core/FlaimProvider/FlaimWrapper/Makefile.am	2010-01-05 13:22:40.000000000 +0100
+++ src/core/FlaimProvider/FlaimWrapper/Makefile.am	2011-10-27 20:04:13.323423935 +0200
@@ -93,7 +93,7 @@
 # NOTE: The following is a _hack_. 
 LIBFLAIM_LIBS_HACK = "-lpthread -lrt -lstdc++ -ldl -lncurses /usr/$(lib)/libflaim.a"
 
-libFlaimWrapper_la_LDFLAGS = $(LIBFLAIM_LIBS_HACK)
+libFlaimWrapper_la_LDFLAGS = $(LIBFLAIM_LIBS)
 
 install-exec-local: libFlaimWrapper.la
 	$(mkinstalldirs) $(DESTDIR)$(webbindir)
