--- src/core/Simias.Web/SharedCollection.cs	2010-04-15 10:16:23.000000000 -0400
+++ src/core/Simias.Web/SharedCollection.cs	2010-07-19 12:38:09.905200825 -0400
@@ -1106,7 +1106,7 @@ namespace Simias.Web
 						while(mntLine != null)
 						{
 							// verify it's a device on this box
-							if(mntLine.StartsWith("/dev") && (mntLine.IndexOf("iso9660") == -1))
+							if((mntLine.StartsWith("/dev") || mntLine.StartsWith("/home")) && (mntLine.IndexOf("iso9660") == -1))
 							{
 								Stat stat;
 								string[] entries;
