Index: src/e_mod_main.c
===================================================================
--- src/e_mod_main.c	(revision 480)
+++ src/e_mod_main.c	(working copy)
@@ -20,7 +20,8 @@
 #include "Efreet.h"
 #include "Efreet_Trash.h"
 
-#define FILE_MANAGER "thunar"
+#define FILE_MANAGER "dolphin"
+// #define FILE_MANAGER "thunar"
 // #define FILE_MANAGER "konqueror"
 // #define FILE_MANAGER "nautilus --no-desktop"
 
@@ -251,7 +252,10 @@
    ev = event_info;
    if ((ev->button == 1))
      {
-        _trash_cb_menu_show(NULL, NULL, NULL);
+       if (ev->flags & (EVAS_BUTTON_DOUBLE_CLICK | EVAS_BUTTON_TRIPLE_CLICK))
+         {
+           _trash_cb_menu_show(NULL, NULL, NULL);	 
+         }
      }
    else if ((ev->button == 3) && (!inst->menu))
      {
