diff -ur mrbayes-3.1.2.orig/mcmc.c mrbayes-3.1.2/mcmc.c
--- mrbayes-3.1.2.orig/mcmc.c	2005-12-24 05:26:28.000000000 +1300
+++ mrbayes-3.1.2/mcmc.c	2007-08-17 11:58:14.858900600 +1200
@@ -7871,7 +7871,7 @@
 					free(tempStr);
 					return (ERROR);
 					}
-				if (chainParams.saveBrlens == YES)
+				if (chainParams.mcmcDiagn == YES)
 					MrBayesPrint ("%s   Calculating MCMC diagnostics\n", spacer);
 				else
 					MrBayesPrint ("%s   Not calculating MCMC diagnostics\n", spacer);
