--- Makefile.in.orig	2013-07-27 19:37:53.311437783 +0200
+++ Makefile.in	2013-07-27 19:39:55.756892228 +0200
@@ -21,12 +21,12 @@
 	$(MAKE) -C $(testgtk_dir)
 
 install: 
-	$(MAKE) -C $(src_dir) install
-	$(MAKE) -C $(projects_dir) install
-	$(MAKE) -C $(testgtk_dir) install
-	$(MAKE) -C $(doc_dir) install
+	$(MAKE) -C $(src_dir) DESTDIR=$(DESTDIR) install
+	$(MAKE) -C $(projects_dir) DESTDIR=$(DESTDIR) install
+	$(MAKE) -C $(testgtk_dir) DESTDIR=$(DESTDIR) install
+	$(MAKE) -C $(doc_dir) DESTDIR=$(DESTDIR) install
 	#$(MAKE) -C $(example_dir) install
-	$(MAKE) -C $(xml_dir) install
+	$(MAKE) -C $(xml_dir) DESTDIR=$(DESTDIR) install
 	@echo '------------------------------------------------------------------'
 	@echo '--  GtkAda has now been installed.                              --'
 	@echo '--  To be able to use the library, you may need to update your  --'
