--- original/src/ol_lrc_fetch.c	2013-09-28 09:52:17.574805803 -0300
+++ new/src/ol_lrc_fetch.c	2013-09-28 09:52:41.878804944 -0300
@@ -21,6 +21,7 @@
 #include <gtk/gtk.h>
 #include "ol_lrc_fetch.h"
 #include "string.h"
+#include "ol_lrc_fetch_viewlyric.h"
 #include "ol_lrc_fetch_xiami.h"
 #include "ol_lrc_fetch_ttplayer.h"
 #include "ol_utils.h"
@@ -89,6 +90,7 @@
 {
   ol_lrc_fetch_add_engine (ol_lrc_fetch_xiami_engine ());
   ol_lrc_fetch_add_engine (ol_lrc_fetch_ttplayer_engine ());
+  ol_lrc_fetch_add_engine (ol_lrc_fetch_viewlyric_engine ());
 }
 
 static int
--- original/src/Makefile.am	2013-09-28 09:52:17.568805803 -0300
+++ new/src/Makefile.am	2013-09-28 09:52:41.872804945 -0300
@@ -60,6 +60,7 @@
 	ol_lrc_fetch_module.h \
 	ol_lrc_fetch_xiami.h \
 	ol_lrc_fetch_ttplayer.h \
+	ol_lrc_fetch_viewlyric.h \
 	ol_lrc_fetch_ui.h \
 	ol_lrc_fetch_utils.h \
 	ol_lrc.h \
@@ -164,6 +165,7 @@
 	ol_lrc_fetch_ui.c \
 	ol_lrc_fetch_xiami.c \
 	ol_lrc_fetch_ttplayer.c \
+	ol_lrc_fetch_viewlyric.c \
 	ol_lrc_fetch_utils.c \
 	ol_lrc_candidate_list.c \
 	ol_lrc_engine_list.c \
--- original/src/Makefile.in	2012-06-04 10:30:46.000000000 -0300
+++ new/src/Makefile.in	2013-09-28 10:19:56.164747237 -0300
@@ -93,7 +93,7 @@
 	ol_lrc.$(OBJEXT) ol_search_dialog.$(OBJEXT) \
 	ol_trayicon.$(OBJEXT) ol_menu.$(OBJEXT) ol_lrc_fetch.$(OBJEXT) \
 	ol_lrc_fetch_module.$(OBJEXT) ol_lrc_fetch_ui.$(OBJEXT) \
-	ol_lrc_fetch_xiami.$(OBJEXT) ol_lrc_fetch_ttplayer.$(OBJEXT) \
+	ol_lrc_fetch_xiami.$(OBJEXT) ol_lrc_fetch_ttplayer.$(OBJEXT) ol_lrc_fetch_viewlyric.$(OBJEXT) \
 	ol_lrc_fetch_utils.$(OBJEXT) ol_lrc_candidate_list.$(OBJEXT) \
 	ol_lrc_engine_list.$(OBJEXT) ol_music_info.$(OBJEXT) \
 	ol_color.$(OBJEXT) ol_eggaccelerators.$(OBJEXT) \
@@ -382,6 +382,7 @@
 	ol_lrc_fetch_module.h \
 	ol_lrc_fetch_xiami.h \
 	ol_lrc_fetch_ttplayer.h \
+	ol_lrc_fetch_viewlyric.h \
 	ol_lrc_fetch_ui.h \
 	ol_lrc_fetch_utils.h \
 	ol_lrc.h \
@@ -486,6 +487,7 @@
 	ol_lrc_fetch_ui.c \
 	ol_lrc_fetch_xiami.c \
 	ol_lrc_fetch_ttplayer.c \
+	ol_lrc_fetch_viewlyric.c \
 	ol_lrc_fetch_utils.c \
 	ol_lrc_candidate_list.c \
 	ol_lrc_engine_list.c \
@@ -618,6 +620,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ol_lrc_fetch.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ol_lrc_fetch_module.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ol_lrc_fetch_ttplayer.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ol_lrc_fetch_viewlyric.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ol_lrc_fetch_ui.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ol_lrc_fetch_utils.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ol_lrc_fetch_xiami.Po@am__quote@
