diff -Naur a/editor.c b/editor.c
--- a/editor.c	2012-07-04 11:57:05.000000000 +0000
+++ b/editor.c	2012-07-04 11:57:42.000000000 +0000
@@ -27,7 +27,6 @@
 #include <gdk/gdkkeysyms.h>
 #include <pcre.h>
 #include <glib.h>
-#include <glib/gunicode.h>
 #include <ctype.h>
 
 #ifdef HAVE_GTK3
diff -Naur a/unformat.c b/unformat.c
--- a/unformat.c	2012-06-01 04:38:42.000000000 +0000
+++ b/unformat.c	2012-07-04 12:03:29.000000000 +0000
@@ -19,7 +19,6 @@
 
 #include "config.h"
 #include <glib.h>
-#include <glib/gunicode.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
