Index: otool/Makefile.in
===================================================================
--- cctools-836.1/otool.orig/Makefile.in
+++ cctools-836.1.new/otool/Makefile.in
@@ -52,7 +52,7 @@ CFILES = main.c ofile_print.c m68k_disas
 ifeq ($(HAVE_OBJC_RUNTIME_H),YES)
 OBJCFILES = print_objc.c print_objc2_32bit.c print_objc2_64bit.c
 OBJCOBJS = $(OBJCFILES:.c=.o) 
-LDFLAGS += -lobjc -lpthread -ldispatch
+LDFLAGS += -lobjc
 endif
 OBJS = $(CFILES:.c=.o) 
 
