Index: driver/mysql_art_resultset.cpp
===================================================================
--- driver/mysql_art_resultset.cpp.orig
+++ driver/mysql_art_resultset.cpp
@@ -16,6 +16,7 @@
 
 #include <cppconn/exception.h>
 #include <sstream>
+#include <stdio.h>
 #include <math.h>
 
 
Index: driver/mysql_resultbind.cpp
===================================================================
--- driver/mysql_resultbind.cpp.orig
+++ driver/mysql_resultbind.cpp
@@ -14,6 +14,7 @@
 #include "mysql_resultbind.h"
 
 #include <string.h>
+#include <stdio.h>
 
 namespace sql
 {
Index: test/unit/classes/resultsetmetadata.cpp
===================================================================
--- test/unit/classes/resultsetmetadata.cpp.orig
+++ test/unit/classes/resultsetmetadata.cpp
@@ -15,6 +15,7 @@
 #include "resultsetmetadata.h"
 #include <sstream>
 #include <stdlib.h>
+#include <stdio.h>
 #include <cppconn/resultset.h>
 #include <cppconn/datatype.h>
 #include <cppconn/prepared_statement.h>
