From 4137a71fccd5bdf8422d71d5baf1a5f12d73afcc Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcel.huber@hsr.ch>
Date: Mon, 14 Jul 2014 13:36:06 +0200
Subject: (naively) corrected plugin code to run with current set of libs

---
 plugins/CapitalizeEmphasis.hs | 2 +-
 plugins/PigLatin.hs           | 2 +-

diff --git a/plugins/CapitalizeEmphasis.hs b/plugins/CapitalizeEmphasis.hs
index 59055aa..da2b960 100644
--- a/plugins/CapitalizeEmphasis.hs
+++ b/plugins/CapitalizeEmphasis.hs
@@ -11,7 +11,7 @@ plugin :: Plugin
 plugin = mkPageTransform capsTransform
 
 capsTransform :: [Inline] -> [Inline]
-capsTransform (Emph x : xs) = processWith capStr x ++ capsTransform xs
+capsTransform (Emph x : xs) = bottomUp capStr x ++ capsTransform xs
 capsTransform (x:xs)        = x : capsTransform xs
 capsTransform []            = []
 
diff --git a/plugins/PigLatin.hs b/plugins/PigLatin.hs
index a306462..f60c636 100644
--- a/plugins/PigLatin.hs
+++ b/plugins/PigLatin.hs
@@ -12,7 +12,7 @@ plugin = PageTransform $ \doc -> do
   meta <- askMeta
   case lookup "language" meta of
        Just s | map toLower s == "pig latin" ->
-         return $ processWith pigLatinStr doc
+         return $ bottomUp pigLatinStr doc
        _ -> return doc
 
 pigLatinStr :: Inline -> Inline
-- 
2.0.1

