--- src/irc/proxy/listen.c	2009-07-21 20:48:09.000000000 +0200
+++ src/irc/proxy/listen.c	2010-07-21 15:56:31.000000000 +0200
@@ -192,6 +192,11 @@
 		return;
 	}
 
+	if (strcmp(cmd, "PUSH") == 0) {
+		signal_emit("proxy push command", 2, client, args);
+		return;
+	}
+
 	if (client->server == NULL || !client->server->connected) {
 		proxy_outdata(client, ":%s NOTICE %s :Not connected to server\n",
 			      client->proxy_address, client->nick);
