diff -ruN mupen64_src-0.5/Makefile mupen64_src-0.5-new/Makefile
--- mupen64_src-0.5/Makefile	2005-08-26 19:52:13.000000000 +0000
+++ mupen64_src-0.5-new/Makefile	2007-06-10 12:16:46.000000000 +0000
@@ -405,7 +405,7 @@
 	
 clean:
 	find . -name '*.o' -print0 | xargs -0r rm -f
-	rm mupen64 mupen64_nogui mupen64_dbg plugins/mupen64_input.so blight_input/arial.ttf.c blight_input/ttftoh plugins/blight_input.so plugins/mupen64_hle_rsp_azimer.so plugins/dummyaudio.so plugins/mupen64_audio.so plugins/jttl_audio.so plugins/mupen64_soft_gfx.so plugins/glN64.so
+	rm -f mupen64 mupen64_nogui mupen64_dbg plugins/mupen64_input.so blight_input/arial.ttf.c blight_input/ttftoh plugins/blight_input.so plugins/mupen64_hle_rsp_azimer.so plugins/dummyaudio.so plugins/mupen64_audio.so plugins/jttl_audio.so plugins/mupen64_soft_gfx.so plugins/glN64.so
 
 clean_o:
 	find . -name '*.o' -print0 | xargs -0r rm -f
diff -ruN mupen64_src-0.5/blight_input/SDL_ttf.c mupen64_src-0.5-new/blight_input/SDL_ttf.c
--- mupen64_src-0.5/blight_input/SDL_ttf.c	2005-08-26 19:52:16.000000000 +0000
+++ mupen64_src-0.5-new/blight_input/SDL_ttf.c	2007-06-10 12:16:46.000000000 +0000
@@ -47,12 +47,13 @@
 #include <freetype/freetype.h>
 #include <freetype/ftoutln.h>
 #include <freetype/ttnameid.h>
-*/
+
 #include <freetype/internal/ftobjs.h>
 
 #ifndef FT_OPEN_STREAM
 #define FT_OPEN_STREAM ft_open_stream
 #endif
+*/
 
 #include "SDL.h"
 #include "SDL_endian.h"
@@ -275,7 +276,7 @@
 	}
 	memset(stream, 0, sizeof(*stream));
 
-	stream->memory = library->memory;
+	stream->memory = NULL;  /* set by FT_Open_Face */
 	stream->read = RWread;
 	stream->descriptor.pointer = src;
 	stream->pos = (unsigned long)position;
File mupen64_src-0.5/compare_pipe is a fifo while file mupen64_src-0.5-new/compare_pipe is a fifo
diff -ruN mupen64_src-0.5/glN64/glATI.h mupen64_src-0.5-new/glN64/glATI.h
--- mupen64_src-0.5/glN64/glATI.h	2005-08-26 19:52:17.000000000 +0000
+++ mupen64_src-0.5-new/glN64/glATI.h	2007-06-10 12:17:37.000000000 +0000
@@ -864,8 +864,8 @@
 typedef GLuint (APIENTRY *PFNGLGENFRAGMENTSHADERSATIPROC)(GLuint range);
 typedef GLvoid (APIENTRY *PFNGLBINDFRAGMENTSHADERATIPROC)(GLuint id);
 typedef GLvoid (APIENTRY *PFNGLDELETEFRAGMENTSHADERATIPROC)(GLuint id);
-typedef GLvoid (APIENTRY *PFNGLBEGINFRAGMENTSHADERATIPROC)(GLvoid);
-typedef GLvoid (APIENTRY *PFNGLENDFRAGMENTSHADERATIPROC)(GLvoid);
+typedef GLvoid (APIENTRY *PFNGLBEGINFRAGMENTSHADERATIPROC)();
+typedef GLvoid (APIENTRY *PFNGLENDFRAGMENTSHADERATIPROC)();
 typedef GLvoid (APIENTRY *PFNGLPASSTEXCOORDATIPROC)(GLuint dst, GLuint coord, GLenum swizzle);
 typedef GLvoid (APIENTRY *PFNGLSAMPLEMAPATIPROC)(GLuint dst, GLuint interp, GLenum swizzle);
 typedef GLvoid (APIENTRY *PFNGLCOLORFRAGMENTOP1ATIPROC)(GLenum op, GLuint dst, GLuint dstMask,
@@ -2194,7 +2194,7 @@
 typedef GLvoid    (APIENTRY * PFNGLDELETEOCCLUSIONQUERIESNVPROC) (GLsizei n, const GLuint *ids);
 typedef GLboolean (APIENTRY * PFNGLISOCCLUSIONQUERYNVPROC) (GLuint id);
 typedef GLvoid    (APIENTRY * PFNGLBEGINOCCLUSIONQUERYNVPROC) (GLuint id);
-typedef GLvoid    (APIENTRY * PFNGLENDOCCLUSIONQUERYNVPROC) (GLvoid);
+typedef GLvoid    (APIENTRY * PFNGLENDOCCLUSIONQUERYNVPROC) ();
 typedef GLvoid    (APIENTRY * PFNGLGETOCCLUSIONQUERYIVNVPROC) (GLuint id, GLenum pname, GLint *params);
 typedef GLvoid    (APIENTRY * PFNGLGETOCCLUSIONQUERYUIVNVPROC) (GLuint id, GLenum pname, GLuint *params);
 
diff -ruN mupen64_src-0.5/main/main.c mupen64_src-0.5-new/main/main.c
--- mupen64_src-0.5/main/main.c	2005-08-26 19:52:11.000000000 +0000
+++ mupen64_src-0.5-new/main/main.c	2007-06-10 12:16:46.000000000 +0000
@@ -37,6 +37,8 @@
 
 #include <stdlib.h>
 #include <unistd.h>
+#include <dirent.h>
+#include <sys/stat.h>
 
 #include "main.h"
 #include "guifuncs.h"
@@ -666,7 +668,7 @@
 	i4 = old_i + atoi(s) - 1;
      }
    
-   printf("\n\nSelected Plugins: %s, %s, %s, %s\n", plugins[i1], plugins[i2], plugins[3], plugins[i4]);
+   printf("\n\nSelected Plugins: %s, %s, %s, %s\n", plugins[i1], plugins[i2], plugins[i3], plugins[i4]);
    
    plugin_load_plugins(plugins[i1], plugins[i2], plugins[i3], plugins[i4]);
    romOpen_gfx();
diff -ruN mupen64_src-0.5/mupen64_audio/main.c mupen64_src-0.5-new/mupen64_audio/main.c
--- mupen64_src-0.5/mupen64_audio/main.c	2005-08-26 19:52:17.000000000 +0000
+++ mupen64_src-0.5-new/mupen64_audio/main.c	2007-06-10 12:16:46.000000000 +0000
@@ -16,6 +16,7 @@
 
 #include "../main/winlnxdefs.h"
 #include "Audio_#1.1.h"
+#include "../config.h"
 
 static AUDIO_INFO AudioInfo;
 static int dsp;
@@ -172,6 +173,11 @@
 
 static void GetPluginDir(char *Directory)
 {
+#ifdef WITH_HOME
+   strcpy(Directory, getenv("HOME"));
+   strcat(Directory, "/.mupen64/");
+   strcat(Directory, "plugins/");
+#else
    int n = readlink("/proc/self/exe", Directory, PATH_MAX);
    if (n == -1)
      strcpy(Directory, "./");
@@ -182,6 +188,7 @@
 	  Directory[strlen(Directory)-1] = '\0';
      }
    strcat(Directory, "plugins/");
+#endif
 }
 
 #ifdef USE_GTK
