diff -aur sonata/sonata/main.py sonata.new/sonata/main.py
--- sonata/sonata/main.py	2009-09-22 01:02:16.000000000 +0400
+++ sonata.new/sonata/main.py	2013-11-01 15:59:08.919054767 +0400
@@ -993,13 +993,7 @@
                             self.status['random'] == '1')
                     if not self.last_consume or self.last_consume != self.status['consume']:
                         self.consumemenu.set_active(self.status['consume'] == '1')
-                    if self.status['xfade'] == '0':
-                        self.config.xfade_enabled = False
-                    else:
-                        self.config.xfade_enabled = True
-                        self.config.xfade = int(self.status['xfade'])
-                        if self.config.xfade > 30:
-                            self.config.xfade = 30
+                    self.config.xfade_enabled = False
                     self.last_repeat = self.status['repeat']
                     self.last_random = self.status['random']
                     self.last_consume = self.status['consume']
