--- pyqwt3d-0.1.7~cvs20090625~/sip/Qwt3D_Qt3_Module.sip 2010-02-07 16:36:56.000000000 +0100
+++ pyqwt3d-0.1.7~cvs20090625/sip/Qwt3D_Qt3_Module.sip  2010-02-07 16:37:33.000000000 +0100
@@ -35,7 +35,7 @@

 typedef unsigned int    GLenum;
 typedef unsigned char   GLboolean;
-typedef unsigned int    GLbitfield;
+//typedef unsigned int    GLbitfield;
 typedef void            GLvoid;
 typedef /*signed*/ char GLbyte;         /* 1-byte signed */
 typedef short           GLshort;        /* 2-byte signed */
diff -urNad pyqwt3d-0.1.7~cvs20090625~/sip/Qwt3D_Qt4_Module.sip pyqwt3d-0.1.7~cvs20090625/sip/Qwt3D_Qt4_Module.sip
--- pyqwt3d-0.1.7~cvs20090625~/sip/Qwt3D_Qt4_Module.sip 2010-02-07 16:36:56.000000000 +0100
+++ pyqwt3d-0.1.7~cvs20090625/sip/Qwt3D_Qt4_Module.sip  2010-02-07 16:36:59.000000000 +0100
@@ -36,7 +36,7 @@

 // typedef unsigned int    GLenum;
 typedef unsigned char   GLboolean;
-typedef unsigned int    GLbitfield;
+//typedef unsigned int    GLbitfield;
 typedef void            GLvoid;
 typedef /*signed*/ char GLbyte;         /* 1-byte signed */
 typedef short           GLshort;        /* 2-byte signed */
@@ -45,7 +45,7 @@
 typedef unsigned short  GLushort;       /* 2-byte unsigned */
 // typedef unsigned int    GLuint;         /* 4-byte unsigned */
 typedef int             GLsizei;        /* 4-byte signed */
-typedef float           GLfloat;        /* single precision float */
+//typedef float           GLfloat;        /* single precision float */
 typedef float           GLclampf;       /* single precision float in [0,1] */
 typedef double          GLdouble;       /* double precision float */
 typedef double          GLclampd;       /* double precision float in [0,1] */
--- pyqwt3d-0.1.7~cvs20090625~/sip/qwt3d_plot.sip 2010-02-07 16:36:56.000000000 +0100
+++ pyqwt3d-0.1.7~cvs20090625/sip/qwt3d_plot.sip  2010-02-07 16:36:59.000000000 +0100
@@ -45,8 +45,8 @@
         sipWrapperType **type;
         int yes, no;
     } graph[] = {
-        {sipName_Plot3D, &sipClass_Plot3D, 1, -1},
-        {sipName_SurfacePlot, &sipClass_SurfacePlot, -1, -1},
+        {(char*)sipName_Plot3D, &sipClass_Plot3D, 1, -1},
+        {(char*)sipName_SurfacePlot, &sipClass_SurfacePlot, -1, -1},
     };
     int i = 0;
     sipClass = NULL;
