diff -Nur djv-0.9.0.orig/plugins/djv_openexr/djv_openexr.cpp djv-0.9.0/plugins/djv_openexr/djv_openexr.cpp
--- djv-0.9.0.orig/plugins/djv_openexr/djv_openexr.cpp	2012-07-31 10:40:03.000000000 +1000
+++ djv-0.9.0/plugins/djv_openexr/djv_openexr.cpp	2014-07-04 14:57:46.924801156 +1000
@@ -35,10 +35,9 @@
 
 #include <djv_assert.h>
 
-#include <ImfChannelList.h>
-#include <ImfFramesPerSecond.h>
-#include <ImfStandardAttributes.h>
-#include <ImfThreading.h>
+#include <OpenEXR/ImfFramesPerSecond.h>
+#include <OpenEXR/ImfStandardAttributes.h>
+#include <OpenEXR/ImfThreading.h>
 
 namespace djv_openexr
 {
diff -Nur djv-0.9.0.orig/plugins/djv_openexr/djv_openexr.h djv-0.9.0/plugins/djv_openexr/djv_openexr.h
--- djv-0.9.0.orig/plugins/djv_openexr/djv_openexr.h	2012-08-06 07:25:00.000000000 +1000
+++ djv-0.9.0/plugins/djv_openexr/djv_openexr.h	2014-07-04 14:57:57.561467426 +1000
@@ -38,16 +38,9 @@
 #include <djv_image_io.h>
 
 #include <ImathBox.h>
-#include <ImfPixelType.h>
-
-namespace Imf
-{
-
-struct Channel;
-class ChannelList;
-class Header;
-
-} // Imf
+#include <OpenEXR/ImfChannelList.h>
+#include <OpenEXR/ImfHeader.h>
+#include <OpenEXR/ImfPixelType.h>
 
 //! \namespace djv_openexr
 //
diff -Nur djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_base.cpp djv-0.9.0/plugins/djv_openexr/djv_openexr_base.cpp
--- djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_base.cpp	2012-07-31 10:29:31.000000000 +1000
+++ djv-0.9.0/plugins/djv_openexr/djv_openexr_base.cpp	2014-07-04 14:56:49.761469956 +1000
@@ -35,7 +35,7 @@
 
 #include <djv_assert.h>
 
-#include <ImfThreading.h>
+#include <OpenEXR/ImfThreading.h>
 
 namespace djv_openexr
 {
diff -Nur djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_load.cpp djv-0.9.0/plugins/djv_openexr/djv_openexr_load.cpp
--- djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_load.cpp	2012-08-06 08:31:14.000000000 +1000
+++ djv-0.9.0/plugins/djv_openexr/djv_openexr_load.cpp	2014-07-04 14:59:03.071464980 +1000
@@ -35,10 +35,9 @@
 
 #include <djv_gl_image.h>
 
-#include <ImfChannelList.h>
-#include <ImfHeader.h>
-#include <ImfInputFile.h>
-#include <ImfRgbaYca.h>
+#include <OpenEXR/ImfChannelList.h>
+#include <OpenEXR/ImfHeader.h>
+#include <OpenEXR/ImfRgbaYca.h>
 
 #include <algorithm>
 
diff -Nur djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_load.h djv-0.9.0/plugins/djv_openexr/djv_openexr_load.h
--- djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_load.h	2012-07-30 06:03:05.000000000 +1000
+++ djv-0.9.0/plugins/djv_openexr/djv_openexr_load.h	2014-07-04 14:58:49.764798810 +1000
@@ -35,13 +35,7 @@
 #define DJV_OPENEXR_LOAD_H
 
 #include <djv_openexr.h>
-
-namespace Imf
-{
-
-class InputFile;
-
-} // Imf
+#include <OpenEXR/ImfInputFile.h>
 
 namespace djv_openexr
 {
diff -Nur djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_save.cpp djv-0.9.0/plugins/djv_openexr/djv_openexr_save.cpp
--- djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_save.cpp	2012-07-29 05:14:15.000000000 +1000
+++ djv-0.9.0/plugins/djv_openexr/djv_openexr_save.cpp	2014-07-04 14:59:24.934797498 +1000
@@ -35,10 +35,9 @@
 
 #include <djv_gl_image.h>
 
-#include <ImfChannelList.h>
-#include <ImfCompressionAttribute.h>
-#include <ImfHeader.h>
-#include <ImfOutputFile.h>
+#include <OpenEXR/ImfChannelList.h>
+#include <OpenEXR/ImfCompressionAttribute.h>
+#include <OpenEXR/ImfHeader.h>
 
 namespace djv_openexr
 {
diff -Nur djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_save.h djv-0.9.0/plugins/djv_openexr/djv_openexr_save.h
--- djv-0.9.0.orig/plugins/djv_openexr/djv_openexr_save.h	2012-07-29 05:14:16.000000000 +1000
+++ djv-0.9.0/plugins/djv_openexr/djv_openexr_save.h	2014-07-04 14:59:20.561464327 +1000
@@ -35,13 +35,7 @@
 #define DJV_OPENEXR_SAVE_H
 
 #include <djv_openexr.h>
-
-namespace Imf
-{
-
-class OutputFile;
-
-} // Imf
+#include <OpenEXR/ImfOutputFile.h>
 
 namespace djv_openexr
 {
