diff -Nur djv-0.9.0.orig/lib/djv_core/djv_file.h djv-0.9.0/lib/djv_core/djv_file.h
--- djv-0.9.0.orig/lib/djv_core/djv_file.h	2012-07-29 04:35:44.000000000 +1000
+++ djv-0.9.0/lib/djv_core/djv_file.h	2014-07-04 15:01:25.678126324 +1000
@@ -37,6 +37,7 @@
 #include <djv_debug.h>
 #include <djv_seq.h>
 #include <djv_set.h>
+#include <unistd.h>
 
 namespace djv
 {
diff -Nur djv-0.9.0.orig/lib/djv_core/djv_file_inline.h djv-0.9.0/lib/djv_core/djv_file_inline.h
--- djv-0.9.0.orig/lib/djv_core/djv_file_inline.h	2012-07-27 09:37:05.000000000 +1000
+++ djv-0.9.0/lib/djv_core/djv_file_inline.h	2014-07-04 15:01:00.021460615 +1000
@@ -32,6 +32,7 @@
 //! \file djv_file_inline.h
 
 #include <djv_math.h>
+#include <unistd.h>
 
 namespace djv
 {
diff -Nur djv-0.9.0.orig/lib/djv_core/djv_user.h djv-0.9.0/lib/djv_core/djv_user.h
--- djv-0.9.0.orig/lib/djv_core/djv_user.h	2012-07-27 09:37:05.000000000 +1000
+++ djv-0.9.0/lib/djv_core/djv_user.h	2014-07-04 15:01:12.701460142 +1000
@@ -35,6 +35,7 @@
 #define DJV_USER_H
 
 #include <djv_string.h>
+#include <unistd.h>
 
 namespace djv
 {
