diff -rc cnijfilter-source-3.40-1.orig/cnijfilter/src/bjfimage.c cnijfilter-source-3.40-1/cnijfilter/src/bjfimage.c
*** cnijfilter-source-3.40-1.orig/cnijfilter/src/bjfimage.c	2010-02-19 06:29:46.000000000 +0100
--- cnijfilter-source-3.40-1/cnijfilter/src/bjfimage.c	2012-02-06 15:52:41.823962703 +0100
***************
*** 39,45 ****
  
  #include <stdio.h>
  #include <stdlib.h>
! #include <png.h>
  
  #include "cpktypes.h"
  #include "cncl.h"
--- 39,45 ----
  
  #include <stdio.h>
  #include <stdlib.h>
! #include <libpng14/png.h>
  
  #include "cpktypes.h"
  #include "cncl.h"
diff -rc cnijfilter-source-3.40-1.orig/cnijfilter/src/Makefile.am cnijfilter-source-3.40-1/cnijfilter/src/Makefile.am
*** cnijfilter-source-3.40-1.orig/cnijfilter/src/Makefile.am	2010-03-09 11:51:24.000000000 +0100
--- cnijfilter-source-3.40-1/cnijfilter/src/Makefile.am	2012-02-06 15:48:32.713595785 +0100
***************
*** 24,29 ****
  	uitypes.h 
  
  
! cif_LDADD = -lcnbpcmcm@CNCL_LIB_ID@ -lcnbpess@CNCL_LIB_ID@ -lm -ldl -ltiff -lpng\
  			-lcnbpcnclapi@CNCL_LIB_ID@ -lcnbpcnclbjcmd@CNCL_LIB_ID@ -lcnbpcnclui@CNCL_LIB_ID@ -lpopt
  cif_LDFLAGS = -L../../@CNCL_LIB_ID@/libs_bin@ARC@
--- 24,29 ----
  	uitypes.h 
  
  
! cif_LDADD = -lcnbpcmcm@CNCL_LIB_ID@ -lcnbpess@CNCL_LIB_ID@ -lm -ldl -ltiff -lpng14\
  			-lcnbpcnclapi@CNCL_LIB_ID@ -lcnbpcnclbjcmd@CNCL_LIB_ID@ -lcnbpcnclui@CNCL_LIB_ID@ -lpopt
  cif_LDFLAGS = -L../../@CNCL_LIB_ID@/libs_bin@ARC@