--- setupPrintcap2	2010-05-23 23:10:01.000000000 -0400
+++ setupPrintcap2.new	2012-06-02 15:59:42.561759149 -0400
@@ -17,18 +17,14 @@
 # this program; if not, write to the Free Software Foundation, Inc., 59 Temple
 # Place, Suite 330, Boston, MA  02111-1307  USA
 #
 
 BRPRINTER_NAME=$1
-PRINTCAP_NAME=/etc/printcap.local
+PRINTCAP_NAME=/etc/printcap
 SPOOLER_NAME=/var/spool/lpd/$BRPRINTER_NAME
 DEVICE_IF=/dev/lp0
 
-if [ ! -f $PRINTCAP_NAME ]; then
-	PRINTCAP_NAME=/etc/printcap
-fi
-
 case "$2" in
 	-i)
 	if [ ! -d $SPOOLER_NAME ]; then
 	        mkdir $SPOOLER_NAME
 	fi
@@ -44,11 +40,11 @@
 ${BRPRINTER_NAME}:\\
         :mx=0:\\
         :sd=/var/spool/lpd/$BRPRINTER_NAME:\\
         :sh:\\
         :lp=$DEVICE_IF:\\
-        :if=/usr/local/Brother/Printer/$BRPRINTER_NAME/lpd/filter$BRPRINTER_NAME:
+        :if=/usr/share/Brother/Printer/$BRPRINTER_NAME/lpd/filter$BRPRINTER_NAME:
 EOF
 	;;
 	-e)
 	set +o noclobber
 	cp $PRINTCAP_NAME "$PRINTCAP_NAME".tmp
