Index: clutter-sharp/clutter-gtk/clutter-gtk-sharp.pc.in
===================================================================
--- clutter-sharp.orig/clutter-gtk/clutter-gtk-sharp.pc.in	2010-12-29 22:38:57.642717380 +0800
+++ clutter-sharp/clutter-gtk/clutter-gtk-sharp.pc.in	2010-12-29 22:39:04.312791090 +0800
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-bundlefiles=${libdir}/clutter-sharp/clutter-gtk-sharp.dll ${libdir}/clutter-sharp/clutter-gtk-sharp.dll.mdb ${libdir}/clutter-sharp/clutter-gtk-sharp.dll.config
+bundlefiles=${libdir}/clutter-sharp/clutter-gtk-sharp.dll ${libdir}/clutter-sharp/clutter-gtk-sharp.dll ${libdir}/clutter-sharp/clutter-gtk-sharp.dll.config
 
 Name: clutter-gtk-sharp
 Description: clutter-sharp - Clutter-Gtk C#/Mono Bindings
Index: clutter-sharp/clutter/clutter-sharp.pc.in
===================================================================
--- clutter-sharp.orig/clutter/clutter-sharp.pc.in	2010-12-29 22:38:57.616050418 +0800
+++ clutter-sharp/clutter/clutter-sharp.pc.in	2010-12-29 22:39:04.312791090 +0800
@@ -2,7 +2,7 @@
 exec_prefix=@exec_prefix@
 libdir=${exec_prefix}/lib
 gapidir=${prefix}/share/gapi-2.0
-bundlefiles=${libdir}/clutter-sharp/clutter-sharp.dll ${libdir}/clutter-sharp/clutter-sharp.dll.mdb ${libdir}/clutter-sharp/clutter-sharp.dll.config ${libdir}/clutter-sharp/glib-sharp.dll ${libdir}/clutter-sharp/glib-sharp.dll.config
+bundlefiles=${libdir}/clutter-sharp/clutter-sharp.dll ${libdir}/clutter-sharp/clutter-sharp.dll ${libdir}/clutter-sharp/clutter-sharp.dll.config ${libdir}/clutter-sharp/glib-sharp.dll ${libdir}/clutter-sharp/glib-sharp.dll.config
 
 Name: clutter-sharp
 Description: clutter-sharp - clutter C#/Mono Bindings
