From: Chow Loong Jin <hyperair@ubuntu.com>
Date: Sat, 28 May 2011 22:20:26 +0800
Subject: Fix bashism in Makefile

---
 glib/Makefile.am |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/glib/Makefile.am b/glib/Makefile.am
index 9bda459..2773d37 100644
--- a/glib/Makefile.am
+++ b/glib/Makefile.am
@@ -17,10 +17,12 @@ patch.dll: patch.cs glib-sharp.dll
 	$(MCS) -out:$@ -target:library -nowarn:0169 -r:glib-sharp.dll $<
 
 %.dll:
-	@for lib in $(GTKSHARP_LIBS); do \
+	for lib in $(GTKSHARP_LIBS); do \
 		case "$$lib" in *$@*) \
-			echo "Copying $${lib:3}"; \
-			cp `readlink -e $${lib:3}`{,.config} . ;; \
+			slib="$$(expr substr $$lib 4 \( length $$lib \))"; \
+			echo "Copying $$slib"; \
+			rlib="$$(readlink -e $$slib)"; \
+			cp $$rlib $$rlib.config . ;; \
 		esac; \
 	done
 
-- 
