--- configure	2011-11-05 15:29:03.000000000 +0100
+++ configure2	2013-05-11 23:12:22.723713873 +0200
@@ -7353,17 +7353,8 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for xft-config" >&5
 $as_echo_n "checking for xft-config... " >&6; }
     XFTCONFIG=`which xft-config`
-    XFT_CFLAGS=""
-    XFT_LIBS=""
-    if test -x ${XFTCONFIG}; then
-    	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $XFTCONFIG" >&5
-$as_echo "$XFTCONFIG" >&6; }
-    	XFT_CFLAGS=`${XFTCONFIG} --cflags`
-    	XFT_LIBS=`${XFTCONFIG} --libs`
-    else
-    	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-    fi
+    XFT_CFLAGS=$(pkg-config --cflags xft)
+    XFT_LIBS=$(pkg-config --libs xft)
 
     # Had to move this, Xft 2 again
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for XRenderQueryExtension in -lXrender" >&5
