Description: lockdown locks the menu editor entry too
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=499148
Bug-Ubuntu: https://bugs.launchpad.net/bugs/4712

Index: gnome-panel-2.30.0/gnome-panel/panel-menu-bar.c
===================================================================
--- gnome-panel-2.30.0.orig/gnome-panel/panel-menu-bar.c	2010-03-05 03:20:12.000000000 +1100
+++ gnome-panel-2.30.0/gnome-panel/panel-menu-bar.c	2010-06-03 13:58:03.123946697 +1000
@@ -345,8 +345,9 @@
 				   _("_Help"),
 				   NULL);
 
-	if (panel_is_program_in_path ("alacarte") ||
-	    panel_is_program_in_path ("gmenu-simple-editor")) {
+	if (!panel_lockdown_get_locked_down () && 
+	    (panel_is_program_in_path ("alacarte") ||
+	    panel_is_program_in_path ("gmenu-simple-editor"))) {
 		panel_applet_add_callback (menubar->priv->info,
 					   "edit",
 					   NULL,
Index: gnome-panel-2.30.0/gnome-panel/panel-menu-button.c
===================================================================
--- gnome-panel-2.30.0.orig/gnome-panel/panel-menu-button.c	2010-03-10 09:52:58.000000000 +1100
+++ gnome-panel-2.30.0/gnome-panel/panel-menu-button.c	2010-06-03 13:58:03.123946697 +1000
@@ -662,8 +662,9 @@
 
 	panel_applet_add_callback (info, "help", GTK_STOCK_HELP, _("_Help"), NULL);
 
-        if (panel_is_program_in_path ("alacarte") ||
-	    panel_is_program_in_path ("gmenu-simple-editor"))
+        if (!panel_lockdown_get_locked_down () && 
+	    (panel_is_program_in_path ("alacarte") ||
+	    panel_is_program_in_path ("gmenu-simple-editor")))
 		panel_applet_add_callback (info, "edit", NULL,
 					   _("_Edit Menus"), NULL);
 
