=== modified file 'src/idooffscreenproxy.c'
--- src/idooffscreenproxy.c	2011-09-06 17:52:37 +0000
+++ src/idooffscreenproxy.c	2012-01-11 23:49:05 +0000
@@ -211,6 +211,10 @@
     | GDK_ENTER_NOTIFY_MASK
     | GDK_LEAVE_NOTIFY_MASK;
   attributes.visual = gdk_screen_get_rgba_visual (gdk_screen_get_default ());//gtk_widget_get_visual (widget);
+  if (!attributes.visual)
+    {
+      attributes.visual = gdk_screen_get_system_visual (gdk_screen_get_default ());
+    }
   attributes.wclass = GDK_INPUT_OUTPUT;
   
   attributes_mask = GDK_WA_X | GDK_WA_Y | GDK_WA_VISUAL;

