--- a/kernel/time/tick-sched.c	2012-07-21 13:58:29.000000000 -0700
+++ b/kernel/time/tick-sched.c	2012-08-12 11:48:41.213442655 -0700
@@ -406,7 +406,9 @@
 		 */
 		if (!ts->tick_stopped) {
 			select_nohz_load_balancer(1);
+#ifndef CONFIG_NO_HZ
 			calc_load_enter_idle();
+#endif
 
 			ts->idle_tick = hrtimer_get_expires(&ts->sched_timer);
 			ts->tick_stopped = 1;
@@ -582,7 +584,9 @@
 	/* Update jiffies first */
 	select_nohz_load_balancer(0);
 	tick_do_update_jiffies64(now);
+#ifndef CONFIG_NO_HZ
 	update_cpu_load_nohz();
+#endif
 
 #ifndef CONFIG_VIRT_CPU_ACCOUNTING
 	/*
@@ -598,7 +602,9 @@
 		account_idle_ticks(ticks);
 #endif
 
+#ifndef CONFIG_NO_HZ
 	calc_load_exit_idle();
+#endif
 	touch_softlockup_watchdog();
 	/*
 	 * Cancel the scheduled timer and restore the tick

