diff -aur projectzomboid/media/lua/Camping/BuildingObjects/campingCampfire.lua projectzomboid-fixed/media/lua/Camping/BuildingObjects/campingCampfire.lua
--- projectzomboid/media/lua/Camping/BuildingObjects/campingCampfire.lua	2013-05-22 11:12:37.000000000 -0700
+++ projectzomboid-fixed/media/lua/Camping/BuildingObjects/campingCampfire.lua	2013-06-02 16:51:26.130544373 -0700
@@ -3,6 +3,8 @@
 --** Help you to place your campfire by let you dragging a ghost render of the tent around **
 --***********************************************************
 
+require "BuildingObjects/ISBuildingObject"
+
 campingCampfire = ISBuildingObject:derive("campingCampfire");
 
 --************************************************************************--
diff -aur projectzomboid/media/lua/Camping/BuildingObjects/campingTent.lua projectzomboid-fixed/media/lua/Camping/BuildingObjects/campingTent.lua
--- projectzomboid/media/lua/Camping/BuildingObjects/campingTent.lua	2013-05-22 11:12:37.000000000 -0700
+++ projectzomboid-fixed/media/lua/Camping/BuildingObjects/campingTent.lua	2013-06-02 16:51:33.353864127 -0700
@@ -3,6 +3,8 @@
 --** Help you to place your tent by let you dragging a ghost render of the tent around **
 --***********************************************************
 
+require "BuildingObjects/ISBuildingObject"
+
 campingTent = ISBuildingObject:derive("campingTent");
 
 --************************************************************************--
diff -aur projectzomboid/media/lua/Farming/BuildingObjects/farmingPlot.lua projectzomboid-fixed/media/lua/Farming/BuildingObjects/farmingPlot.lua
--- projectzomboid/media/lua/Farming/BuildingObjects/farmingPlot.lua	2013-05-22 11:12:37.000000000 -0700
+++ projectzomboid-fixed/media/lua/Farming/BuildingObjects/farmingPlot.lua	2013-06-02 16:51:52.380496504 -0700
@@ -2,6 +2,8 @@
 --**                    ROBERT JOHNSON                     **
 --***********************************************************
 
+require "BuildingObjects/ISBuildingObject"
+
 farmingPlot = ISBuildingObject:derive("farmingPlot");
 
 --************************************************************************--
diff -aur projectzomboid/media/lua/OptionScreens/MainOptions.lua projectzomboid-fixed/media/lua/OptionScreens/MainOptions.lua
--- projectzomboid/media/lua/OptionScreens/MainOptions.lua	2013-05-22 11:12:37.000000000 -0700
+++ projectzomboid-fixed/media/lua/OptionScreens/MainOptions.lua	2013-06-02 16:50:50.093950478 -0700
@@ -3,6 +3,7 @@
 --**            Screen with all our options                **
 --***********************************************************
 
+require "ISUI/ISScrollingListBox"
 require "ISUI/ISPanel"
 require "ISUI/ISButton"
 require "ISUI/ISInventoryPane"
diff -aur projectzomboid/media/lua/Reloading/ISReloadUtil.lua projectzomboid-fixed/media/lua/Reloading/ISReloadUtil.lua
--- projectzomboid/media/lua/Reloading/ISReloadUtil.lua	2013-05-22 11:12:37.000000000 -0700
+++ projectzomboid-fixed/media/lua/Reloading/ISReloadUtil.lua	2013-06-02 16:52:37.507089681 -0700
@@ -1,4 +1,7 @@
 require "ISBaseObject"
+require "Reloading/ISReloadableMagazine"
+require "Reloading/ISSemiAutoWeapon"
+require "Reloading/ISShotgunWeapon"
 
 ISReloadUtil = ISBaseObject:derive("ISReloadUtil");
 
