diff -crB pulseaudio-equalizer/usr/bin/pulseaudio-equalizer-gtk pulseaudio-equalizer.new//usr/bin/pulseaudio-equalizer-gtk
*** pulseaudio-equalizer/usr/bin/pulseaudio-equalizer-gtk	2010-10-23 22:16:45.092802000 +0200
--- pulseaudio-equalizer.new//usr/bin/pulseaudio-equalizer-gtk	2010-10-23 22:30:40.493342779 +0200
***************
*** 4,7 ****
  # Author: Conn O'Griofa <connogriofa AT gmail DOT com>
  # Version: (see '/usr/pulseaudio-equalizer' script)
  
! python /usr/share/pulseaudio-equalizer/pulseaudio-equalizer.py
--- 4,7 ----
  # Author: Conn O'Griofa <connogriofa AT gmail DOT com>
  # Version: (see '/usr/pulseaudio-equalizer' script)
  
! python2 /usr/share/pulseaudio-equalizer/pulseaudio-equalizer.py
