From 469b124d5b636d2df711915b5063d9e40885fbe7 Mon Sep 17 00:00:00 2001
From: Alfredo Palhares <masterkorp@masterkorp.net>
Date: Wed, 8 May 2013 00:04:37 +0100
Subject: [PATCH] Ack the libserialIO path.

---
 FreeEMSLoader.pro | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/FreeEMSLoader.pro b/FreeEMSLoader.pro
index 8d3fbaa..1a696fd 100644
--- a/FreeEMSLoader.pro
+++ b/FreeEMSLoader.pro
@@ -87,9 +87,9 @@ unix:INSTALLS += target
 
 # Default make specs
 INCLUDEPATH += src/inc
-unix:INCLUDEPATH += $$quote(/usr/local/include/)
-unix:LIBS += $$quote(/usr/local/lib/libSerialIO.so.$$LIB_VERSION)
-unix:PRE_TARGETDEPS += $$quote(/usr/local/lib/libSerialIO.so.$$LIB_VERSION)
+unix:INCLUDEPATH += $$quote(/usr/include/)
+unix:LIBS += $$quote(/usr/lib/libSerialIO.so.$$LIB_VERSION)
+unix:PRE_TARGETDEPS += $$quote(/usr/lib/libSerialIO.so.$$LIB_VERSION)
 unix:DEFINES += GIT_HASH=$$system(git describe --dirty=-DEV --always)
 unix:DEFINES += GIT_HASH_FULL=$$system(git rev-parse HEAD)
 
-- 
1.8.2.2

