diff -aur xdg-utils-1.1.0-rc1/scripts/xdg-email xdg-utils-mimeo/scripts/xdg-email
--- xdg-utils-1.1.0-rc1/scripts/xdg-email	2011-01-01 04:03:14.000000000 -0600
+++ xdg-utils-mimeo/scripts/xdg-email	2012-01-23 20:21:05.258527175 -0600
@@ -361,6 +361,7 @@
     elif `dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:org.gnome.SessionManager > /dev/null 2>&1` ; then DE=gnome;
     elif xprop -root _DT_SAVE_MODE 2> /dev/null | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
     elif [ x"$DESKTOP_SESSION" == x"LXDE" ]; then DE=lxde;
+    elif which mimeo 2> /dev/null; then DE=mimeo;
     else DE=""
     fi
 }
@@ -509,6 +510,18 @@
     fi
 }
 
+open_mimeo()
+{
+    DEBUG 1 "Running mimeo \"$1\""
+    mimeo "$1"
+
+    if [ $? -eq 0 ]; then
+        exit_success
+    else
+        exit_failure_operation_failed
+    fi
+}
+
 open_generic()
 {
     IFS=":"
@@ -716,6 +729,10 @@
     open_xfce "${mailto}"
     ;;
 
+    mimeo)
+    open_mimeo "${mailto}"
+    ;;
+
     generic|lxde)
     open_generic "${mailto}"
     ;;
diff -aur xdg-utils-1.1.0-rc1/scripts/xdg-mime xdg-utils-mimeo/scripts/xdg-mime
--- xdg-utils-1.1.0-rc1/scripts/xdg-mime	2011-01-01 04:03:14.000000000 -0600
+++ xdg-utils-mimeo/scripts/xdg-mime	2012-01-23 20:21:05.258527175 -0600
@@ -431,6 +431,7 @@
     elif `dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:org.gnome.SessionManager > /dev/null 2>&1` ; then DE=gnome;
     elif xprop -root _DT_SAVE_MODE 2> /dev/null | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
     elif [ x"$DESKTOP_SESSION" == x"LXDE" ]; then DE=lxde;
+    elif which mimeo 2> /dev/null; then DE=mimeo;
     else DE=""
     fi
 }
@@ -512,6 +513,16 @@
     fi
 }
 
+info_mimeo()
+{
+    mimeo -m "$1"
+    if [ $? -eq 0 ]; then
+        exit_success
+    else
+        exit_failure_operation_failed
+    fi
+}
+
 info_generic()
 {
     DEBUG 1 "Running file -i \"$1\""
@@ -848,6 +859,10 @@
         info_gnome "$filename"
         ;;
 
+        mimeo)
+        info_mimeo "$filename"
+        ;;
+
         *)
         info_generic "$filename"
         ;;
diff -aur xdg-utils-1.1.0-rc1/scripts/xdg-open xdg-utils-mimeo/scripts/xdg-open
--- xdg-utils-1.1.0-rc1/scripts/xdg-open	2012-01-23 20:21:11.654904878 -0600
+++ xdg-utils-mimeo/scripts/xdg-open	2012-01-23 20:21:05.258527175 -0600
@@ -311,6 +311,7 @@
     elif `dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:org.gnome.SessionManager > /dev/null 2>&1` ; then DE=gnome;
     elif xprop -root _DT_SAVE_MODE 2> /dev/null | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
     elif xprop -root 2> /dev/null | grep -i '^xfce_desktop_window' >/dev/null 2>&1; then DE=xfce
+    elif which mimeo 2> /dev/null; then DE=mimeo;
     fi
 
     if [ x"$DE" = x"" ]; then
@@ -432,6 +433,17 @@
     fi
 }
 
+open_mimeo()
+{
+    DEBUG 1 "Running mimeo \"$1\""
+    mimeo "$1"
+    if [ $? -eq 0 ]; then
+        exit_success
+    else
+        exit_failure_operation_failed
+    fi
+}
+
 open_generic()
 {
     # Paths or file:// URLs
@@ -570,6 +582,10 @@
     open_lxde "$url"
     ;;
 
+    mimeo)
+    open_mimeo "$url"
+    ;;
+
     generic)
     open_generic "$url"
     ;;
diff -aur xdg-utils-1.1.0-rc1/scripts/xdg-settings xdg-utils-mimeo/scripts/xdg-settings
--- xdg-utils-1.1.0-rc1/scripts/xdg-settings	2011-01-01 04:03:14.000000000 -0600
+++ xdg-utils-mimeo/scripts/xdg-settings	2012-01-23 20:21:05.258527175 -0600
@@ -316,6 +316,7 @@
     elif `dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:org.gnome.SessionManager > /dev/null 2>&1` ; then DE=gnome;
     elif xprop -root _DT_SAVE_MODE 2> /dev/null | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
     elif [ x"$DESKTOP_SESSION" == x"LXDE" ]; then DE=lxde;
+    elif which mimeo; then DE=mimeo;
     else DE=""
     fi
 }
@@ -750,6 +751,42 @@
 }
 
 # }}} xfce
+# {{{ mimeo
+
+get_browser_mimeo()
+{
+    DEBUG 1 "Running mimeo -c 'http:'"
+    mimeo -d http: | sed -rn '2s/  ([^ ]+) +.*/\1/p'
+    if [ $? -eq 0 ]; then
+        exit_success
+    else
+        exit_failure_operation_failed
+    fi
+}
+
+check_browser_mimeo()
+{
+    browser=`get_browser_mimeo $1`
+    if [ x"$browser" != x"$1" ]; then
+        echo no
+        exit_success
+    fi
+    echo yes
+    exit_success
+}
+
+set_browser_mimeo()
+{
+    DEBUG 1 "Running mimeo --prefer 'regex:x-scheme-handler/https?' \"$1\""
+    mimeo --prefer 'regex:x-scheme-handler/https?' \"$1\"
+    if [ $? -eq 0 ]; then
+        exit_success
+    else
+        exit_failure_operation_failed
+    fi
+}
+
+# }}} mimeo
 # }}} default browser
 
 dispatch_specific()
@@ -860,7 +897,7 @@
 detectDE
 
 case "$DE" in
-    kde|gnome|xfce)
+    kde|gnome|xfce|mimeo)
     dispatch_specific "$@"
     ;;
 
