Index: configure.ac
===================================================================
--- configure.ac	(revision 7238)
+++ configure.ac	(working copy)
@@ -78,7 +78,7 @@
   OCCAM_CONFIG_TARGET_SUBDIRS(modules/occade/libsrc)
   OCCAM_CONFIG_TARGET_SUBDIRS(modules/occGL/libsrc)
   OCCAM_CONFIG_TARGET_SUBDIRS(modules/button/libsrc)
-  OCCAM_CONFIG_TARGET_SUBDIRS(modules/video/libsrc)
+#  OCCAM_CONFIG_TARGET_SUBDIRS(modules/video/libsrc)
   OCCAM_CONFIG_TARGET_SUBDIRS(modules/graphics3d/libsrc)
   OCCAM_CONFIG_TARGET_SUBDIRS(modules/player/libsrc)
 
@@ -105,7 +105,7 @@
   OCCAM_CONFIG_TARGET_SUBDIRS(modules/ttyutil/examples)
   OCCAM_CONFIG_TARGET_SUBDIRS(modules/udc/examples)
   OCCAM_CONFIG_TARGET_SUBDIRS(modules/useful/examples)
-  OCCAM_CONFIG_TARGET_SUBDIRS(modules/video/examples)
+#  OCCAM_CONFIG_TARGET_SUBDIRS(modules/video/examples)
 
   OCCAM_CONFIG_TARGET_SUBDIRS(demos/bar)
   OCCAM_CONFIG_TARGET_SUBDIRS(demos/game)
