--- ddclient.old	2011-11-28 06:20:43.993483966 +0900
+++ ddclient	2011-11-28 06:47:18.383505412 +0900
@@ -563,6 +563,16 @@
 			  $variables{'service-common-defaults'},
 		        ),
     },
+    'valuedomain' => {
+    'updateable' => undef,
+    'update'     => \&nic_valuedomain_update,
+    'examples'   => \&nic_valuedomain_examples,
+    'variables'  => merge(
+              { 'server' => setv(T_FQDNP, 1, 0, 1, 'dyn.value-domain.com', undef) },
+              { 'login'  => setv(T_FQDN , 1, 0, 1, '', undef) },
+              $variables{'service-common-defaults'}
+                ),
+    },
 );
 $variables{'merged'} = merge($variables{'global-defaults'},
 			     $variables{'service-common-defaults'},
@@ -2183,16 +2193,16 @@
 
     } elsif (defined($sub) && &$sub($host)) {
 	$update = 1;
-    } elsif ((defined($cache{$host}{'static'}) && defined($config{$host}{'static'}) &&
-              ($cache{$host}{'static'} ne $config{$host}{'static'})) ||
-             (defined($cache{$host}{'wildcard'}) && defined($config{$host}{'wildcard'}) &&
-              ($cache{$host}{'wildcard'} ne $config{$host}{'wildcard'})) ||
-             (defined($cache{$host}{'mx'}) && defined($config{$host}{'mx'}) &&
-              ($cache{$host}{'mx'} ne $config{$host}{'mx'})) ||
-             (defined($cache{$host}{'backupmx'}) && defined($config{$host}{'backupmx'}) &&
-              ($cache{$host}{'backupmx'} ne $config{$host}{'backupmx'})) ) {
-	info("updating %s because host settings have been changed.", $host);
-	$update = 1;
+#    } elsif ((defined($cache{$host}{'static'}) && defined($config{$host}{'static'}) &&
+#              ($cache{$host}{'static'} ne $config{$host}{'static'})) ||
+#             (defined($cache{$host}{'wildcard'}) && defined($config{$host}{'wildcard'}) &&
+#              ($cache{$host}{'wildcard'} ne $config{$host}{'wildcard'})) ||
+#             (defined($cache{$host}{'mx'}) && defined($config{$host}{'mx'}) &&
+#              ($cache{$host}{'mx'} ne $config{$host}{'mx'})) ||
+#             (defined($cache{$host}{'backupmx'}) && defined($config{$host}{'backupmx'}) &&
+#              ($cache{$host}{'backupmx'} ne $config{$host}{'backupmx'})) ) {
+#	info("updating %s because host settings have been changed.", $host);
+#	$update = 1;
 
     } else {
 	success("%s: skipped: IP address was already set to %s.", $host, $ip)
@@ -3671,6 +3681,117 @@
     }
 }
 
+######################################################################
+
+######################################################################
+## nic_valuedomain_examples
+######################################################################
+sub nic_valuedomain_examples {
+    return <<EoEXAMPLE;
+
+o 'valuedomain'
+
+The 'valuedomain' protocol is used by DNS services offered by www.value-domain.com.
+
+Configuration variables applicable to the 'valuedomain' protocol are:
+  protocol=valuedomain         ##
+  server=fqdn.of.service       ## defaults to dyn.value-domain.com
+  login=my-domain.name         ## your domain name and password  registered with the service
+  password=service-password    ##
+  host.my-domain.name          ## the host registered with the service.
+
+Example ${program}.conf file entries:
+  ## single host update
+  protocol=valuedomain,                                          \\
+  login=my-domain.name,                                          \\
+  password=my-domainname.com-password                            \\
+  myhost.my-domainname
+
+  ## single host update for wild (*.my-domain.name) support
+  protocol=valuedomain,                                          \\
+  login=my-domain.name,                                          \\
+  password=my-domainname.com-password                            \\
+  *.my-domain.name
+
+  ## multiple host update
+  protocol=valuedomain,                                          \\
+  login=my-domain.name,                                          \\
+  password=my-domainname.com-password                            \\
+  myhost.my-domain.name,my2ndhost.my-domain.name
+
+EoEXAMPLE
+}
+######################################################################
+## nic_valuedomain_update
+##
+## written by Hideo Sato
+##  
+## update by irgaly <irgaly@gmail.com>
+##   2011/1/16: Support normal style host name.
+##              You can set host name like 'www.example.com' otherwise 'www'.
+##              This change lets you to manage more than one domain on Value Domain.
+##
+## based on http://www.value-domain.com/ddns.php?action=howto
+## needs this url to update:
+## http://dyn.value-domain.com/cgi-bin/dyn.fcg?d=domain-name&p=password&h=hostname&i=IPaddress
+##
+######################################################################
+sub nic_valuedomain_update {
+
+
+    debug("\nnic_valuedomain_update -------------------");
+
+    ## update each configured host
+    foreach my $h (@_) {
+        my $ip = delete $config{$h}{'wantip'};
+        info("setting IP address to %s for %s", $ip, $h);
+        verbose("UPDATE:","updating %s", $h);
+
+        # convert host name to valuedomain style.
+        # ex.) example.com -> @, www.example.com -> www
+        my $exp = $config{$h}{'login'};
+        $exp =~ s/\./\\./g;
+        my $vd_host = $h;
+        $vd_host =~ s/\.?$exp$//g;
+        $vd_host = '@' unless length($vd_host);
+        debug("VALUE DOMAIN host name is %s", $vd_host);
+
+        my $url;
+        $url   = "http://$config{$h}{'server'}/cgi-bin/dyn.fcg";
+        $url  .= "?d=$config{$h}{'login'}";
+        $url  .= "&p=$config{$h}{'password'}";
+        $url  .= "&h=$vd_host";
+        $url  .= "&i=";
+        $url  .= $ip if $ip;
+
+        my $reply = geturl(opt('proxy'), $url);
+        if (!defined($reply) || !$reply) {
+            failed("updating %s: Could not connect to %s.", $h, $config{$h}{'server'});
+            last;
+        }
+        last if !header_ok($h, $reply);
+
+        my @reply = split /\n/, $reply;
+        my ($return_code, $return_description) = ('','');
+        foreach my $line (@reply) {
+            $return_code = $1 if $line =~ m%^status=(.*)\s*$%i;
+            $return_description = $line;
+        }
+
+        if ($return_code !~ /0/) {
+            $config{$h}{'status'} = 'failed';
+            warning("SENT:    %s", $url) unless opt('verbose');
+            warning("REPLIED: %s", $reply);
+            failed("updating %s", $h);
+        } else {
+            $config{$h}{'ip'}     = $ip;
+            $config{$h}{'mtime'}  = $now;
+            $config{$h}{'status'} = 'good';
+            success("updating %s: %s: %s: IP address set to %s", $h, $return_code, $return_description, $ip);
+        }
+    }
+}
+
 ######################################################################
 # vim: ai ts=4 sw=4 tw=78 :
 
--- sample-etc_ddclient.conf.old	2011-11-28 06:47:34.346836170 +0900
+++ sample-etc_ddclient.conf	2011-11-28 06:50:41.560172977 +0900
@@ -186,3 +186,11 @@
 # client=ddclient,
 # password=my-dtdns.com-password
 # myhost.dtdns.net, otherhost.dtdns.net
+
+##
+## VALUE-DOMAIN (www.value-domain.com)
+##
+# protocol=valuedomain,
+# login=my-domain.name,
+# password=my-domainname.com-password
+# myhost.my-domain.name,my2ndhost.my-domain.name
