diff -ruN blt2.4z.orig/demos/bitmap2.tcl blt2.4z/demos/bitmap2.tcl
--- blt2.4z.orig/demos/bitmap2.tcl	2002-08-12 00:00:30.000000000 +0200
+++ blt2.4z/demos/bitmap2.tcl	2007-02-28 09:05:07.000000000 +0100
@@ -212,4 +212,4 @@
     incr row
 }
 
-blt::table configure . c* -padx 2
\ Pas de fin de ligne à la fin du fichier.
+blt::table configure . c* -padx 2
diff -ruN blt2.4z.orig/demos/hierbox2.tcl blt2.4z/demos/hierbox2.tcl
--- blt2.4z.orig/demos/hierbox2.tcl	2001-02-07 06:42:17.000000000 +0100
+++ blt2.4z/demos/hierbox2.tcl	2007-02-28 09:05:18.000000000 +0100
@@ -97,4 +97,4 @@
 eval .h entry configure $nodes -labelcolor red 
 
 wm protocol . WM_DELETE_WINDOW { destroy . }
-#blt::bltdebug 100
\ Pas de fin de ligne à la fin du fichier.
+#blt::bltdebug 100
diff -ruN blt2.4z.orig/demos/Makefile.in blt2.4z/demos/Makefile.in
--- blt2.4z.orig/demos/Makefile.in	2002-02-19 01:11:46.000000000 +0100
+++ blt2.4z/demos/Makefile.in	2007-02-28 08:58:34.000000000 +0100
@@ -79,7 +79,7 @@
             : ; \
           else \
             echo " mkdir $(INSTALL_ROOT)$$i" ; \
-            mkdir $(INSTALL_ROOT)"$$i" ; \
+            mkdir -p $(INSTALL_ROOT)"$$i" ; \
           fi ; \
         done
 
diff -ruN blt2.4z.orig/demos/scripts/page.tcl blt2.4z/demos/scripts/page.tcl
--- blt2.4z.orig/demos/scripts/page.tcl	2001-02-07 06:42:23.000000000 +0100
+++ blt2.4z/demos/scripts/page.tcl	2007-02-28 08:58:04.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/local/bin/tclsh
+#!/usr/bin/tclsh
 
 array set page "
     rows	2
diff -ruN blt2.4z.orig/library/dnd.tcl blt2.4z/library/dnd.tcl
--- blt2.4z.orig/library/dnd.tcl	2002-02-03 21:00:43.000000000 +0100
+++ blt2.4z/library/dnd.tcl	2007-02-28 09:05:46.000000000 +0100
@@ -99,4 +99,4 @@
     if { $info(state) & 0x10 } {
 	puts "NumLock-Drop"
     }
-}
\ Pas de fin de ligne à la fin du fichier.
+}
diff -ruN blt2.4z.orig/library/Makefile.in blt2.4z/library/Makefile.in
--- blt2.4z.orig/library/Makefile.in	2002-07-15 01:33:24.000000000 +0200
+++ blt2.4z/library/Makefile.in	2007-02-28 08:44:41.000000000 +0100
@@ -58,7 +58,7 @@
 	for i in $(miscFiles) ; do \
 	    $(INSTALL_DATA) $(srcdir)/$$i $(INSTALL_ROOT)$(scriptdir) ; \
 	done 
-	$(INSTALL_DATA) pkgIndex.tcl $(scriptdir)
+	$(INSTALL_DATA) pkgIndex.tcl $(INSTALL_ROOT)$(scriptdir)
 
 mkdirs:
 	@for i in $(instdirs) ; do \
diff -ruN blt2.4z.orig/Makefile.in blt2.4z/Makefile.in
--- blt2.4z.orig/Makefile.in	2002-02-19 01:11:43.000000000 +0100
+++ blt2.4z/Makefile.in	2007-02-28 08:53:12.000000000 +0100
@@ -47,9 +47,9 @@
 	(cd library; $(MAKE) install)
 	(cd man; $(MAKE) install)
 	(cd demos; $(MAKE) install)
-	$(INSTALL_DATA) $(srcdir)/README $(INSTALL_DIR)$(scriptdir)
-	$(INSTALL_DATA) $(srcdir)/PROBLEMS $(INSTALL_DIR)$(scriptdir)
-	$(INSTALL_DATA) $(srcdir)/NEWS $(INSTALL_DIR)$(scriptdir)
+	$(INSTALL_DATA) $(srcdir)/README $(INSTALL_ROOT)$(scriptdir)
+	$(INSTALL_DATA) $(srcdir)/PROBLEMS $(INSTALL_ROOT)$(scriptdir)
+	$(INSTALL_DATA) $(srcdir)/NEWS $(INSTALL_ROOT)$(scriptdir)
 
 mkdirs:
 	@for i in $(instdirs) ; do \
@@ -57,7 +57,7 @@
 	    : ; \
 	  else \
             echo " mkdir $(INSTALL_ROOT)$$i" ; \
-            mkdir $(INSTALL_ROOT)$$i ; \
+            mkdir -p $(INSTALL_ROOT)$$i ; \
           fi ; \
         done
 
diff -ruN blt2.4z.orig/man/Blt_TreeExists.man3 blt2.4z/man/Blt_TreeExists.man3
--- blt2.4z.orig/man/Blt_TreeExists.man3	2001-02-07 06:42:27.000000000 +0100
+++ blt2.4z/man/Blt_TreeExists.man3	2007-02-28 09:05:37.000000000 +0100
@@ -63,4 +63,4 @@
 .CE
 .SH KEYWORDS
 tree, token
-Tcl_TreeCreate, Tcl_TreeGetToken, Tcl_TreeReleaseToken
\ Pas de fin de ligne à la fin du fichier.
+Tcl_TreeCreate, Tcl_TreeGetToken, Tcl_TreeReleaseToken
