--- blt2.4z/library/Makefile.in.orig	2008-01-06 20:06:26.000000000 -0800
+++ blt2.4z/library/Makefile.in	2008-01-07 14:52:17.000000000 -0800
@@ -9,6 +9,9 @@
 srcdir       = @srcdir@
 libdir       = @libdir@
 scriptdir    = @BLT_LIBRARY@
+# This should really be set by a configure switch.  For now it's a hack
+# for Fedora.
+pkglibdir    = @libdir@/tcl@TCL_VERSION@/blt$(version)
 
 INSTALL      = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
@@ -39,7 +42,8 @@
 		tclIndex
 
 instdirs     =  $(prefix) $(exec_prefix) \
-		$(libdir) $(prefix)/lib $(scriptdir) $(scriptdir)/dd_protocols
+		$(libdir) $(prefix)/lib $(scriptdir) \
+		$(pkglibdir) $(scriptdir)/dd_protocols
 
 all: pkgIndex
 
@@ -58,15 +62,15 @@
 	for i in $(miscFiles) ; do \
 	    $(INSTALL_DATA) $(srcdir)/$$i $(INSTALL_ROOT)$(scriptdir) ; \
 	done 
-	$(INSTALL_DATA) pkgIndex.tcl $(scriptdir)
+	$(INSTALL_DATA) pkgIndex.tcl $(INSTALL_ROOT)$(pkglibdir)
 
 mkdirs:
 	@for i in $(instdirs) ; do \
           if test -d $(INSTALL_ROOT)$$i ; then \
             : ; \
           else \
-            echo " mkdir $(INSTALL_ROOT)$$i" ; \
-            mkdir $(INSTALL_ROOT)$$i ; \
+            echo " mkdir -p $(INSTALL_ROOT)$$i" ; \
+            mkdir -p $(INSTALL_ROOT)$$i ; \
           fi ; \
         done
 
--- blt2.4z/configure.in.orig	2008-01-06 22:48:37.000000000 -0800
+++ blt2.4z/configure.in	2008-01-06 22:49:57.000000000 -0800
@@ -976,7 +976,9 @@
 BLT_VERSION=${BLT_MAJOR_VERSION}.${BLT_MINOR_VERSION}
 
 # Add BLT to the run path
-libdir=${exec_prefix}/lib
+if test "x${libdir}" == "x" ; then
+    libdir=${exec_prefix}/lib
+fi
 
 if test "x${libdir}" != "x"  -a \
 	"${libdir}" != "/usr/lib" -a  \
--- blt2.4z/demos/Makefile.in.orig	2008-01-06 20:11:25.000000000 -0800
+++ blt2.4z/demos/Makefile.in	2008-01-06 22:02:50.000000000 -0800
@@ -6,7 +6,7 @@
 exec_prefix =	@exec_prefix@
 version =	@BLT_VERSION@
 libdir =	@libdir@
-scriptdir =	$(prefix)/lib/blt$(version)
+scriptdir =	@BLT_LIBRARY@
 destdir =	$(scriptdir)/demos
 srcdir =	@srcdir@
 
--- blt2.4z/configure.orig	2008-01-06 22:48:41.000000000 -0800
+++ blt2.4z/configure	2008-01-06 22:49:57.000000000 -0800
@@ -3756,7 +3756,9 @@
 BLT_VERSION=${BLT_MAJOR_VERSION}.${BLT_MINOR_VERSION}
 
 # Add BLT to the run path
-libdir=${exec_prefix}/lib
+if test "x${libdir}" == "x" ; then
+    libdir=${exec_prefix}/lib
+fi
 
 if test "x${libdir}" != "x"  -a \
 	"${libdir}" != "/usr/lib" -a  \
--- blt2.4z/Makefile.in.orig	2008-01-06 13:27:01.000000000 -0800
+++ blt2.4z/Makefile.in	2008-01-06 22:03:26.000000000 -0800
@@ -13,7 +13,7 @@
 includedir =	@includedir@
 libdir =	@libdir@
 version =	@BLT_VERSION@
-scriptdir =	$(prefix)/lib/blt$(version)
+scriptdir =	@BLT_LIBRARY@
 
 instdirs =	$(prefix) \
 		$(exec_prefix) \
@@ -28,7 +28,7 @@
 
 INSTALL =	@INSTALL@
 INSTALL_DATA =	@INSTALL_DATA@
-INSTALL_ROOT = 
+INSTALL_ROOT =  @INSTALL_ROOT@
 SHELL =		/bin/sh
 RM =		rm -f
 
@@ -47,17 +47,17 @@
 	(cd library; $(MAKE) install)
 	(cd man; $(MAKE) install)
 	(cd demos; $(MAKE) install)
-	$(INSTALL_DATA) $(srcdir)/README $(INSTALL_DIR)$(scriptdir)
-	$(INSTALL_DATA) $(srcdir)/PROBLEMS $(INSTALL_DIR)$(scriptdir)
-	$(INSTALL_DATA) $(srcdir)/NEWS $(INSTALL_DIR)$(scriptdir)
+	$(INSTALL_DATA) $(srcdir)/README $(INSTALL_ROOT)$(scriptdir)
+	$(INSTALL_DATA) $(srcdir)/PROBLEMS $(INSTALL_ROOT)$(scriptdir)
+	$(INSTALL_DATA) $(srcdir)/NEWS $(INSTALL_ROOT)$(scriptdir)
 
 mkdirs:
 	@for i in $(instdirs) ; do \
           if test -d $(INSTALL_ROOT)$$i ; then \
 	    : ; \
 	  else \
-            echo " mkdir $(INSTALL_ROOT)$$i" ; \
-            mkdir $(INSTALL_ROOT)$$i ; \
+            echo " mkdir -p $(INSTALL_ROOT)$$i" ; \
+            mkdir -p $(INSTALL_ROOT)$$i ; \
           fi ; \
         done
 
--- blt2.4z/src/Makefile.in.orig	2008-01-06 22:27:58.000000000 -0800
+++ blt2.4z/src/Makefile.in	2008-01-06 22:28:06.000000000 -0800
@@ -22,7 +22,7 @@
 
 bindir =	$(exec_prefix)/bin
 exec_prefix =	@exec_prefix@
-incdir =	$(prefix)/include
+incdir =	@includedir@
 libdir = 	@libdir@
 scriptdir =	$(exec_prefix)/lib
 
--- blt2.4z/man/Makefile.in.orig	2008-01-06 20:09:02.000000000 -0800
+++ blt2.4z/man/Makefile.in	2008-01-06 22:57:34.000000000 -0800
@@ -10,9 +10,9 @@
 
 instdirs	= $(mandir) $(mandir)/mann $(mandir)/man3
 
-MAN_N		= BLT.n barchart.n beep.n bgexec.n bitmap.n \
+MAN_N		= BLT.n barchart.n beep.n bgexec.n bltbitmap.n \
 		  bltdebug.n busy.n container.n cutbuffer.n \
-		  dragdrop.n eps.n graph.n hierbox.n  \
+		  dragdrop.n eps.n bltgraph.n hierbox.n  \
 		  hiertable.n htext.n spline.n stripchart.n \
 	 	  table.n tabset.n tile.n tree.n treeview.n vector.n \
 		  watch.n winop.n
@@ -51,8 +51,8 @@
           if test -d $(INSTALL_ROOT)$$i ; then \
             : ; \
           else \
-            echo " mkdir $(INSTALL_ROOT)$$i" ; \
-            mkdir $(INSTALL_ROOT)$$i ; \
+            echo " mkdir -p $(INSTALL_ROOT)$$i" ; \
+            mkdir -p $(INSTALL_ROOT)$$i ; \
           fi ; \
 	done
 
