diff -Naur umoria/source/config.h umoria-aur/source/config.h
--- umoria/source/config.h	1994-07-21 18:47:13.000000000 -0700
+++ umoria-aur/source/config.h	2009-10-26 13:55:47.000000000 -0700
@@ -200,7 +200,7 @@
 
 /* This must be unix; change MORIA_LIB as appropriate.  */
 #define MORIA_SAV	"moria.save"
-#define MORIA_LIB(xxx)  "/home/math/grabiner/moria/files/xxx"
+#define MORIA_LIB(xxx)  "/usr/share/umoria/files/"#xxx
 #define MORIA_HOU	MORIA_LIB(hours)
 #define MORIA_MOR	MORIA_LIB(news)
 #define MORIA_TOP	MORIA_LIB(scores)
@@ -266,6 +266,7 @@
 
 /* Define USG for many systems, this is basically to select SYS V style
    system calls (as opposed to BSD style).  */
+#define USG
 #if defined(SYS_III) || defined(SYS_V) || defined(MSDOS) || defined(MAC)
 #ifndef USG
 #define USG
diff -Naur umoria/source/io.c umoria-aur/source/io.c
--- umoria/source/io.c	1994-07-21 18:47:26.000000000 -0700
+++ umoria-aur/source/io.c	2009-10-26 13:47:42.000000000 -0700
@@ -7,6 +7,7 @@
    included in all such copies. */
 
 #include <stdio.h>
+#include <termios.h>
 #ifndef STDIO_LOADED
 #define STDIO_LOADED
 #endif
@@ -424,7 +425,7 @@
   /* disable all of the normal special control characters */
   tbuf.c_cc[VINTR] = (char)3; /* control-C */
   tbuf.c_cc[VQUIT] = (char)-1;
-  tbuf.c_cc[VERASE] = (char)-1;
+  /*tbuf.c_cc[VERASE] = (char)-1;*/
   tbuf.c_cc[VKILL] = (char)-1;
   tbuf.c_cc[VEOF] = (char)-1;
 
diff -Naur umoria/unix/Makefile umoria-aur/unix/Makefile
--- umoria/unix/Makefile	1994-07-21 15:37:08.000000000 -0700
+++ umoria-aur/unix/Makefile	2009-10-26 13:50:21.000000000 -0700
@@ -3,10 +3,10 @@
 # LIBDIR must be the same directory defined in config.h
 # OWNER is who you want the game to be chown to.
 # GROUP is who you wnat the game to be chgrp to.
-BINDIR = /home/zariski/grabiner/moria
-LIBDIR = /home/zariski/grabiner/moria/files
-OWNER = grabiner
-GROUP = grad
+BINDIR = /usr/bin
+LIBDIR = /usr/share/umoria/files
+OWNER = root
+GROUP = root
 
 # For testing and debugging the program, it is best to use this line.
 # CFLAGS = -g
@@ -14,9 +14,9 @@
 CFLAGS = -O
 
 # For BSD Systems
-CURSES = -lcurses -ltermcap
+# CURSES = -lcurses -ltermcap
 # For SYS V Systems
-# CURSES = -lcurses
+CURSES = -lcurses
 # For XENIX, some XENIX systems may need -ltinfo
 # CURSES = -ltcap -ltermcap -lx
 
diff -Naur umoria/unix/unix.c umoria-aur/unix/unix.c
--- umoria/unix/unix.c	1994-07-21 15:37:08.000000000 -0700
+++ umoria-aur/unix/unix.c	2009-10-26 13:48:46.000000000 -0700
@@ -261,7 +261,7 @@
 #endif
 
 #ifdef USG
-unsigned short getuid();
+/*unsigned short getuid();*/
 #else
 #ifndef SECURE
 #ifdef BSD4_3
