diff -Naur core-4.1-orig/python/src/Makefile.am core-4.1/python/src/Makefile.am
--- core-4.1-orig/python/src/Makefile.am	2010-12-21 15:27:06.000000000 -0500
+++ core-4.1/python/src/Makefile.am	2011-01-05 14:34:42.757727909 -0500
@@ -32,7 +32,7 @@
 noinst_LIBRARIES = libnetns.a
 libnetns_a_SOURCES = netnsmodule.c vcmdmodule.c
 libnetns.a:
-	SBINDIR=@SBINDIR@ CFLAGS=@libev_CFLAGS@ $(PYTHON) setup.py build
+	SBINDIR=@SBINDIR@ CFLAGS="@libev_CFLAGS@" $(PYTHON) setup.py build
 
 install:	install-exec-hook
 
@@ -57,7 +57,7 @@
 	-rm -rf build
 
 rpmbuild.sh:
-	echo SBINDIR=@SBINDIR@ CFLAGS=@libev_CFLAGS@ $(PYTHON) setup.py build > rpmbuild.sh
+	echo SBINDIR=@SBINDIR@ CFLAGS="@libev_CFLAGS@" $(PYTHON) setup.py build > rpmbuild.sh
 	chmod a+x rpmbuild.sh
 
 rpm:	rpmbuild.sh
