diff -Naur core-4.2-orig/python/core/misc/LatLongUTMconversion.py core-4.2-patch/python/core/misc/LatLongUTMconversion.py
--- core-4.2-orig/python/core/misc/LatLongUTMconversion.py	2010-05-21 11:21:43.000000000 -0400
+++ core-4.2-patch/python/core/misc/LatLongUTMconversion.py	2011-08-19 18:26:27.436234190 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # this file is from http://pygps.org/
 
 # Lat Long - UTM, UTM - Lat Long conversions
diff -Naur core-4.2-orig/python/examples/findcore.py core-4.2-patch/python/examples/findcore.py
--- core-4.2-orig/python/examples/findcore.py	2010-12-16 16:51:53.000000000 -0500
+++ core-4.2-patch/python/examples/findcore.py	2011-08-19 18:26:39.196170482 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Search for installed CORE library files and Python bindings.
 #
diff -Naur core-4.2-orig/python/examples/stopsession.py core-4.2-patch/python/examples/stopsession.py
--- core-4.2-orig/python/examples/stopsession.py	2011-08-15 13:17:39.000000000 -0400
+++ core-4.2-patch/python/examples/stopsession.py	2011-08-19 18:26:45.132804985 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # (c)2010-2011 the Boeing Company
 # author: Jeff Ahrenholz <jeffrey.m.ahrenholz@boeing.com>
 #
diff -Naur core-4.2-orig/python/sbin/cored.py core-4.2-patch/python/sbin/cored.py
--- core-4.2-orig/python/sbin/cored.py	2011-08-15 13:17:39.000000000 -0400
+++ core-4.2-patch/python/sbin/cored.py	2011-08-19 18:26:09.656330511 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # CORE
 # Copyright (c)2010-2011 the Boeing Company.
diff -Naur core-4.2-orig/python/sbin/coresendmsg.py core-4.2-patch/python/sbin/coresendmsg.py
--- core-4.2-orig/python/sbin/coresendmsg.py	2011-08-03 16:59:40.000000000 -0400
+++ core-4.2-patch/python/sbin/coresendmsg.py	2011-08-19 18:26:13.679642049 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # CORE
 # Copyright (c)2011 the Boeing Company.
