diff -Naur core-orig/daemon/core/misc/LatLongUTMconversion.py core/daemon/core/misc/LatLongUTMconversion.py
--- core-orig/daemon/core/misc/LatLongUTMconversion.py	2010-05-21 11:21:43.000000000 -0400
+++ core/daemon/core/misc/LatLongUTMconversion.py	2012-02-08 14:12:44.607595128 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # this file is from http://pygps.org/
 
 # Lat Long - UTM, UTM - Lat Long conversions
diff -Naur core-orig/daemon/examples/findcore.py core/daemon/examples/findcore.py
--- core-orig/daemon/examples/findcore.py	2010-12-16 16:51:53.000000000 -0500
+++ core/daemon/examples/findcore.py	2012-02-08 14:12:31.780689193 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Search for installed CORE library files and Python bindings.
 #
diff -Naur core-orig/daemon/examples/stopsession.py core/daemon/examples/stopsession.py
--- core-orig/daemon/examples/stopsession.py	2012-02-03 13:33:51.000000000 -0500
+++ core/daemon/examples/stopsession.py	2012-02-08 14:12:37.574130592 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # (c)2010-2012 the Boeing Company
 # author: Jeff Ahrenholz <jeffrey.m.ahrenholz@boeing.com>
 #
diff -Naur core-orig/daemon/sbin/cored core/daemon/sbin/cored
--- core-orig/daemon/sbin/cored	2012-02-03 13:33:51.000000000 -0500
+++ core/daemon/sbin/cored	2012-02-08 14:12:19.773798556 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # CORE
 # Copyright (c)2010-2012 the Boeing Company.
diff -Naur core-orig/daemon/sbin/coresendmsg core/daemon/sbin/coresendmsg
--- core-orig/daemon/sbin/coresendmsg	2012-02-03 13:33:51.000000000 -0500
+++ core/daemon/sbin/coresendmsg	2012-02-08 14:12:24.413885109 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # CORE
 # Copyright (c)2011-2012 the Boeing Company.
diff -Naur core-orig/scripts/perf/perflogserver.py core/scripts/perf/perflogserver.py
--- core-orig/scripts/perf/perflogserver.py	2012-02-03 13:33:51.000000000 -0500
+++ core/scripts/perf/perflogserver.py	2012-02-08 14:12:54.087771969 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # (c)2011-2012 the Boeing Company
 #
