--- usr/share/pyjamas/examples/buildall.py.old	2012-02-18 15:46:16.296746553 -0500
+++ usr/share/pyjamas/examples/buildall.py	2012-02-18 15:47:01.603167773 -0500
@@ -1,15 +1,10 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import os, sys, glob, shutil
 
 options = " ".join(sys.argv[1:])
 if not options:
     options = "-O"
-for pyjsbuild in ('../bin/pyjsbuild.py', '../bin/pyjsbuild', None):
-    if os.path.exists(pyjsbuild):
-        break
-if not pyjsbuild:
-    sys.stderr.write("Cannot find pyjsbuild")
-    sys.exit(1)
+pyjsbuild = "/usr/bin/pyjsbuild"
 
 def guessMainScriptName(d):
     if os.path.isfile('build.sh'):
@@ -36,10 +31,10 @@
             print("********** Building %s **********" % name.upper())
             if os.path.isdir('output'): 
                 shutil.rmtree(os.path.join(".", "output")) # Why should we skip compilation here ? Just clean first.
-            res = os.system("python ../%s %s %s" % (pyjsbuild, options, f))
+            res = os.system("python2 %s %s %s" % (pyjsbuild, options, f))
         except Exception, e:
             print "Error encountered for %s - %r" % (d, e)
         finally:
             #raw_input('Press any key')
             os.chdir("..")
- 
\ No newline at end of file
+ 
