diff --git a/src/frontend/gtk2/client.c b/src/frontend/gtk2/client.c
index 8329071..52c5c9a 100644
--- a/src/frontend/gtk2/client.c
+++ b/src/frontend/gtk2/client.c
@@ -31,7 +31,6 @@
 #include "client.h"
 #include "marshall.h"
 
-#define LOG_LEVEL DEBUG
 #define IC_NAME_MAX 64
 
 #define PREEDIT_TYPE_STRING_INT \
@@ -129,7 +128,7 @@ FcitxIMClient* FcitxIMClientOpen(FcitxIMClientConnectCallback connectcb, FcitxIM
 
 static void _changed_cb(DBusGProxy* proxy, char* service, char* old_owner, char* new_owner, gpointer user_data)
 {
-    FcitxLog(LOG_LEVEL, "_changed_cb");
+    FcitxLog(DEBUG, "_changed_cb");
     FcitxIMClient* client = (FcitxIMClient*) user_data;
     if (g_str_equal(service, client->servicename)) {
         gboolean new_owner_good = new_owner && (new_owner[0] != '\0');
@@ -151,7 +150,7 @@ static void _changed_cb(DBusGProxy* proxy, char* service, char* old_owner, char*
 
 static void _destroy_cb(DBusGProxy *proxy, gpointer user_data)
 {
-    FcitxLog(LOG_LEVEL, "_destroy_cb");
+    FcitxLog(DEBUG, "_destroy_cb");
     FcitxIMClient* client = (FcitxIMClient*) user_data;
     if (client->proxy == proxy) {
         g_object_unref(client->proxy);
diff --git a/src/frontend/ipc/ipc.c b/src/frontend/ipc/ipc.c
index a29b1de..e126c4f 100644
--- a/src/frontend/ipc/ipc.c
+++ b/src/frontend/ipc/ipc.c
@@ -33,6 +33,10 @@
 
 #define GetIPCIC(ic) ((FcitxIPCIC*) (ic)->privateic)
 
+#ifndef DBUS_ERROR_UNKNOWN_PROPERTY
+#define DBUS_ERROR_UNKNOWN_PROPERTY           "org.freedesktop.DBus.Error.UnknownProperty"
+#endif
+
 typedef struct _FcitxIPCCreateICPriv {
     DBusMessage* message;
     DBusConnection* conn;
diff --git a/src/lib/fcitx-qt/fcitxqtconnection.cpp b/src/lib/fcitx-qt/fcitxqtconnection.cpp
index 24fc6b6..08a17ea 100644
--- a/src/lib/fcitx-qt/fcitxqtconnection.cpp
+++ b/src/lib/fcitx-qt/fcitxqtconnection.cpp
@@ -28,8 +28,9 @@
 #include "fcitx-utils/utils.h"
 #include <QX11Info>
 #include <QFile>
-#include <qtimer.h>
+#include <QTimer>
 #include <QDir>
+#include <QCoreApplication>
 #include <X11/Xlib.h>
 
 FcitxQtConnection::FcitxQtConnection(QObject* parent): QObject(parent)
