--- bsfl	2011-04-10 00:01:33.000000000 +0200
+++ bsfl.modified	2012-12-05 14:10:13.123808961 +0100
@@ -38,11 +38,11 @@
     # 
     # Syslog style log messages
     #
-    if ! defined LOGDATEFORMAT
+    if ! defined $LOGDATEFORMAT
     then
         LOGDATEFORMAT="%b %e %H:%M:%S"
     fi
-    if ! defined LOG_FILE
+    if ! defined $LOG_FILE
     then
         LOG_FILE=$0.log
     fi
@@ -51,15 +51,15 @@
     # Enable / disable logging to a file
     # Value: yes or no (y / n)
     #
-    if ! defined LOG_ENABLED
+    if ! defined $LOG_ENABLED
     then
         LOG_ENABLED=no
     fi
-    if ! defined SYSLOG_ENABLED
+    if ! defined $SYSLOG_ENABLED
     then
         SYSLOG_ENABLED=no
     fi
-    if ! defined SYSLOG_TAG
+    if ! defined $SYSLOG_TAG
     then
         SYSLOG_TAG=$0
     fi
@@ -96,7 +96,11 @@
 }
 
 function defined {
-    [[ ${!1-X} == ${!1-Y} ]]
+    if [ ! -z $1 ]; then 
+      return 0 
+    else  
+      return 1
+    fi 
 }
 
 #
@@ -105,11 +109,10 @@
 #
 function has_value {
     if defined $1; then
-        if [[ -n ${!1} ]]; then
-            return 0
-        fi
-    fi
-    return 1
+      return 0
+    else 
+      return 1
+    fi 
 }
 
 #
@@ -208,7 +211,7 @@
         LOG_MESSAGE="$1" 
         DATE=`date +"$LOGDATEFORMAT"`
 
-        if has_value LOG_MESSAGE
+        if has_value $LOG_MESSAGE
         then
             LOG_STRING="$DATE $LOG_MESSAGE"
         else
@@ -244,12 +247,12 @@
     MESSAGE="$1"
     COLOR="$2"
 
-    if ! has_value COLOR
+    if ! has_value $COLOR
     then
         COLOR="$DEFAULT"
     fi
 
-    if has_value "MESSAGE"
+    if has_value $MESSAGE
     then
         $COLOR
         echo "$MESSAGE" 
@@ -634,7 +637,7 @@
     local DEST="$2"
     local FILE="$3"
 
-    has_value FILE 
+    has_value $FILE 
     die_if_false $? "Empty argument 'file'"
     file_exists "$FILE"
     die_if_false $? "File does not exist"
