Index: gears/localserver/common/localserver_db.h
===================================================================
--- gears/localserver/common/localserver_db.h	(révision 3407)
+++ gears/localserver/common/localserver_db.h	(copie de travail)
@@ -27,6 +27,8 @@
 #define GEARS_LOCALSERVER_COMMON_LOCALSERVER_DB_H__
 
 #include <vector>
+#include <cstdio>
+
 #include "gears/base/common/common.h"
 #include "gears/base/common/name_value_table.h"
 #include "gears/base/common/sqlite_wrapper.h"
Index: gears/tools/config.mk
===================================================================
--- gears/tools/config.mk	(révision 3407)
+++ gears/tools/config.mk	(copie de travail)
@@ -96,7 +96,7 @@
 # Set default build mode
 #   dbg = debug build
 #   opt = release build
-MODE = dbg
+MODE = opt
 
 # Set default OS architecture
 #   OSX and Symbian builds will override this value (see rules.mk).
@@ -235,8 +235,9 @@
 endif
 endif
 endif
-GECKO_BIN = $(GECKO_SDK)/gecko_sdk/bin
-GECKO_LIB = $(GECKO_SDK)/gecko_sdk/lib
+
+GECKO_BIN = /usr/lib/xulrunner-devel-1.9.2/bin/
+GECKO_LIB = /usr/lib/xulrunner-devel-1.9.2/lib/
 # GECKO_SDK gets defined below (different for each OS).
 
 $(BROWSER)_CPPFLAGS += -DBROWSER_$(BROWSER)=1
Index: gears/geolocation/backoff_manager.cc
===================================================================
--- gears/geolocation/backoff_manager.cc	(révision 3407)
+++ gears/geolocation/backoff_manager.cc	(copie de travail)
@@ -26,6 +26,7 @@
 #include "gears/geolocation/backoff_manager.h"
 
 #include <assert.h>
+#include <cstdlib>
 
 // The baseline minimum period between network requests.
 static const int kBaselineMinimumRequestInterval = 1000 * 5;  // 5 seconds
Index: gears/base/firefox/install.rdf.m4
===================================================================
--- gears/base/firefox/install.rdf.m4	(révision 3407)
+++ gears/base/firefox/install.rdf.m4	(copie de travail)
@@ -79,13 +79,9 @@
 |^,PRODUCT_OS,^|wince|^,^|m4_dnl
     <!-- WE DON'T CURRENTLY BUILD FOR WINCE FIREFOX. -->
 |^,PRODUCT_OS,^|linux|^,^|m4_dnl
-m4_ifelse(PRODUCT_ARCH,^|x86_64|^,^|m4_dnl
-    <em:targetPlatform>Linux_x86_64-gcc3</em:targetPlatform>
-|^,^|m4_dnl // PRODUCT_ARCH else...
-    <em:targetPlatform>Linux_x86-gcc3</em:targetPlatform>
-    <!-- Ubuntu Edgy Eft requires "linux-gnu" for the OS_TARGET prefix. -->
-    <em:targetPlatform>linux-gnu_x86-gcc3</em:targetPlatform>
-|^)m4_dnl // PRODUCT_ARCH end
+    m4_format(`<em:targetPlatform>Linux_%s-gcc3</em:targetPlatform>', PRODUCT_ARCH)
+	<!-- Ubuntu Edgy Eft requires "linux-gnu" for the OS_TARGET prefix. -->
+	m4_format(`<em:targetPlatform>linux-gnu_%s-gcc3</em:targetPlatform>', PRODUCT_ARCH)
 |^,PRODUCT_OS,^|osx|^,^|m4_dnl
     <em:targetPlatform>Darwin</em:targetPlatform>
 |^)
Index: gears/base/common/string_utils.h
===================================================================
--- gears/base/common/string_utils.h	(révision 3407)
+++ gears/base/common/string_utils.h	(copie de travail)
@@ -29,6 +29,7 @@
 #include <algorithm>
 #include <assert.h>
 #include <cctype>
+#include <climits>
 #include <string>
 #include <vector>
 #include "gears/base/common/string16.h" // for char16
Index: gears/base/common/string16.h
===================================================================
--- gears/base/common/string16.h	(révision 3407)
+++ gears/base/common/string16.h	(copie de travail)
@@ -46,6 +46,7 @@
 #define GEARS_BASE_COMMON_STRING16_H__
 
 #include <string>
+#include <cstring>
 #include "gears/base/common/basictypes.h"
 
 // Need to cast literals (Linux, OSX) and SQLite void* retvals (all platforms)
Index: gears/base/common/http_utils.cc
===================================================================
--- gears/base/common/http_utils.cc	(révision 3407)
+++ gears/base/common/http_utils.cc	(copie de travail)
@@ -25,6 +25,7 @@
 //
 // This file is branched from /google3/webutil/http/httputils.cc
 
+#include <cstdlib>
 #include <iterator>
 #include <vector>
 #include "gears/base/common/http_utils.h"
Index: gears/base/common/basictypes.h
===================================================================
--- gears/base/common/basictypes.h	(révision 3407)
+++ gears/base/common/basictypes.h	(copie de travail)
@@ -27,7 +27,7 @@
 #define GEARS_BASE_COMMON_BASICTYPES_H__
 
 #include <stddef.h>  // for NULL, size_t
-
+#include <stdint.h>  // for intptr_t
 #ifdef _MSC_VER
 #include <float.h>  // for _isnan() on VC++
 #define isnan(x) _isnan(x)  // VC++ uses _isnan() instead of isnan()
Index: third_party/jsoncpp/json_reader.cc
===================================================================
--- third_party/jsoncpp/json_reader.cc	(révision 3407)
+++ third_party/jsoncpp/json_reader.cc	(copie de travail)
@@ -1,5 +1,6 @@
 #include "third_party/jsoncpp/reader.h"
 #include "third_party/jsoncpp/value.h"
+#include <cstring>
 #include <utility>
 #include <stdio.h>
 #include <assert.h>
Index: third_party/jsoncpp/json_value.cc
===================================================================
--- third_party/jsoncpp/json_value.cc	(révision 3407)
+++ third_party/jsoncpp/json_value.cc	(copie de travail)
@@ -1,5 +1,6 @@
 #include "third_party/jsoncpp/value.h"
 #include "third_party/jsoncpp/writer.h"
+#include <cstring>
 #include <utility>
 #include "assert.h"
 #ifdef JSON_USE_CPPTL
Index: third_party/jsoncpp/json_writer.cc
===================================================================
--- third_party/jsoncpp/json_writer.cc	(révision 3407)
+++ third_party/jsoncpp/json_writer.cc	(copie de travail)
@@ -1,4 +1,5 @@
 #include "third_party/jsoncpp/writer.h"
+#include <cstring>
 #include <utility>
 #include <assert.h>
 #include <stdio.h>
Index: third_party/googleurl/src/url_parse.cc
===================================================================
--- third_party/googleurl/src/url_parse.cc	(révision 3407)
+++ third_party/googleurl/src/url_parse.cc	(copie de travail)
@@ -33,7 +33,7 @@
  * the terms of any one of the MPL, the GPL or the LGPL.
  *
  * ***** END LICENSE BLOCK ***** */
-
+#include <cstdlib>
 #include "googleurl/src/url_parse.h"
 
 #include "base/logging.h"
Index: gears/desktop/desktop.cc
===================================================================
--- gears/desktop/desktop.cc	(révision 3408)
+++ gears/desktop/desktop.cc	(copie de travail)
@@ -216,14 +216,14 @@
   }
 
   // Resolve the icon urls
-  if (!shortcut_info->icon16x16.url.empty() &&
-      !ResolveUrl(&shortcut_info->icon16x16.url, &error_) ||
-      !shortcut_info->icon32x32.url.empty() &&
-      !ResolveUrl(&shortcut_info->icon32x32.url, &error_) ||
-      !shortcut_info->icon48x48.url.empty() &&
-      !ResolveUrl(&shortcut_info->icon48x48.url, &error_) ||
-      !shortcut_info->icon128x128.url.empty() &&
-      !ResolveUrl(&shortcut_info->icon128x128.url, &error_)) {
+  if ((!shortcut_info->icon16x16.url.empty() &&
+      !ResolveUrl(&shortcut_info->icon16x16.url, &error_)) ||
+      (!shortcut_info->icon32x32.url.empty() &&
+      !ResolveUrl(&shortcut_info->icon32x32.url, &error_)) ||
+      (!shortcut_info->icon48x48.url.empty() &&
+      !ResolveUrl(&shortcut_info->icon48x48.url, &error_)) ||
+      (!shortcut_info->icon128x128.url.empty() &&
+      !ResolveUrl(&shortcut_info->icon128x128.url, &error_))) {
     return false;
   }

Index: gears/cctests/test.cc
===================================================================
--- gears/cctests/test.cc	(révision 3408)
+++ gears/cctests/test.cc	(copie de travail)
@@ -700,15 +700,15 @@
 
   bool ok = false;
   JsParamType t = context->GetArgumentType(1);
-  if (type == STRING16(L"bool") && t == JSPARAM_BOOL ||
-      type == STRING16(L"int") && t == JSPARAM_INT ||
-      type == STRING16(L"double") && t == JSPARAM_DOUBLE ||
-      type == STRING16(L"string") && t == JSPARAM_STRING16 ||
-      type == STRING16(L"null") && t == JSPARAM_NULL ||
-      type == STRING16(L"undefined") && t == JSPARAM_UNDEFINED ||
-      type == STRING16(L"array") && t == JSPARAM_ARRAY ||
-      type == STRING16(L"function") && t == JSPARAM_FUNCTION ||
-      type == STRING16(L"object") && t == JSPARAM_OBJECT) {
+  if ((type == STRING16(L"bool") && t == JSPARAM_BOOL) ||
+      (type == STRING16(L"int") && t == JSPARAM_INT) ||
+      (type == STRING16(L"double") && t == JSPARAM_DOUBLE) ||
+      (type == STRING16(L"string") && t == JSPARAM_STRING16) ||
+      (type == STRING16(L"null") && t == JSPARAM_NULL) ||
+      (type == STRING16(L"undefined") && t == JSPARAM_UNDEFINED) ||
+      (type == STRING16(L"array") && t == JSPARAM_ARRAY) ||
+      (type == STRING16(L"function") && t == JSPARAM_FUNCTION) ||
+      (type == STRING16(L"object") && t == JSPARAM_OBJECT)) {
     ok = true;
   }
   context->SetReturnValue(JSPARAM_BOOL, &ok);
Index: gears/tools/rules.mk
===================================================================
--- gears/tools/rules.mk	(révision 3423)
+++ gears/tools/rules.mk	(copie de travail)
@@ -1301,8 +1301,8 @@
 	cp -R $(COMMON_OUTDIR)/genfiles/i18n/* $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/chrome/chromeFiles/locale
 	cp $(FF36_MODULE_TYPELIB) $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/components
 	cp $(FF36_MODULE_DLL) $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff36/$(DLL_PREFIX)$(MODULE)$(DLL_SUFFIX)
-	cp $(FF31_MODULE_DLL) $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff35/$(DLL_PREFIX)$(MODULE)$(DLL_SUFFIX)
-	cp $(FF3_MODULE_DLL) $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff30/$(DLL_PREFIX)$(MODULE)$(DLL_SUFFIX)
+	#cp $(FF31_MODULE_DLL) $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff35/$(DLL_PREFIX)$(MODULE)$(DLL_SUFFIX)
+	#cp $(FF3_MODULE_DLL) $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff30/$(DLL_PREFIX)$(MODULE)$(DLL_SUFFIX)
 ifneq ($(ARCH),x86_64)
 	cp $(FF2_MODULE_DLL) $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff2/$(DLL_PREFIX)$(MODULE)$(DLL_SUFFIX)
 endif
@@ -1314,9 +1314,9 @@
 ifeq ($(MODE),dbg)
 ifdef IS_WIN32_OR_WINCE
 	cp $(FF36_OUTDIR)/$(MODULE).pdb $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff36/$(MODULE).pdb
-	cp $(FF31_OUTDIR)/$(MODULE).pdb $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff35/$(MODULE).pdb
-	cp $(FF3_OUTDIR)/$(MODULE).pdb $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff30/$(MODULE).pdb
-	cp $(FF2_OUTDIR)/$(MODULE).pdb $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff2/$(MODULE).pdb
+	#cp $(FF31_OUTDIR)/$(MODULE).pdb $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff35/$(MODULE).pdb
+	#cp $(FF3_OUTDIR)/$(MODULE).pdb $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff30/$(MODULE).pdb
+	#cp $(FF2_OUTDIR)/$(MODULE).pdb $(INSTALLERS_OUTDIR)/$(INSTALLER_BASE_NAME)/lib/ff2/$(MODULE).pdb
 endif
 endif
 endif # not LINUX

