diff -aur cnijfilter-source-3.20-1/backend/autogen.sh cnijfilter-source-3.20-1_new/backend/autogen.sh
--- cnijfilter-source-3.20-1/backend/autogen.sh	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/backend/autogen.sh	2013-01-08 09:37:15.882100727 +0100
@@ -112,7 +112,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/backend/configure.in cnijfilter-source-3.20-1_new/backend/configure.in
--- cnijfilter-source-3.20-1/backend/configure.in	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/backend/configure.in	2013-01-08 09:37:15.878767393 +0100
@@ -1,7 +1,7 @@
 AC_INIT(src/cnij_backend_common.c)
 AM_INIT_AUTOMAKE(backend, 3.20-1)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 AC_ARG_ENABLE(progpath)
 if test "${enable_progpath}" = "" ; then
diff -aur cnijfilter-source-3.20-1/backendnet/autogen.sh cnijfilter-source-3.20-1_new/backendnet/autogen.sh
--- cnijfilter-source-3.20-1/backendnet/autogen.sh	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/backendnet/autogen.sh	2013-01-08 09:37:15.882100727 +0100
@@ -112,7 +112,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/backendnet/configure.in cnijfilter-source-3.20-1_new/backendnet/configure.in
--- cnijfilter-source-3.20-1/backendnet/configure.in	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/backendnet/configure.in	2013-01-08 09:37:15.878767393 +0100
@@ -1,7 +1,7 @@
 AC_INIT(backend/cnijnetlm.c)
 AM_INIT_AUTOMAKE(backendnet, 3.10-1)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 AC_ARG_ENABLE(progpath)
 if test "${enable_progpath}" = "" ; then
diff -aur cnijfilter-source-3.20-1/cngpij/autogen.sh cnijfilter-source-3.20-1_new/cngpij/autogen.sh
--- cnijfilter-source-3.20-1/cngpij/autogen.sh	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/cngpij/autogen.sh	2013-01-08 09:37:15.885434060 +0100
@@ -112,7 +112,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/cngpij/configure.in cnijfilter-source-3.20-1_new/cngpij/configure.in
--- cnijfilter-source-3.20-1/cngpij/configure.in	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/cngpij/configure.in	2013-01-08 09:37:15.878767393 +0100
@@ -1,7 +1,7 @@
 AC_INIT(cngpij/bjcups.c)
 AM_INIT_AUTOMAKE(cngpij, 3.20-1)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 AC_ARG_ENABLE(progpath)
 if test "${enable_progpath}" = "" ; then
diff -aur cnijfilter-source-3.20-1/cngpijmon/autogen.sh cnijfilter-source-3.20-1_new/cngpijmon/autogen.sh
--- cnijfilter-source-3.20-1/cngpijmon/autogen.sh	2009-04-24 08:12:49.000000000 +0200
+++ cnijfilter-source-3.20-1_new/cngpijmon/autogen.sh	2013-01-08 09:37:15.885434060 +0100
@@ -122,7 +122,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/cngpijmon/cnijnpr/autogen.sh cnijfilter-source-3.20-1_new/cngpijmon/cnijnpr/autogen.sh
--- cnijfilter-source-3.20-1/cngpijmon/cnijnpr/autogen.sh	2009-04-24 08:12:49.000000000 +0200
+++ cnijfilter-source-3.20-1_new/cngpijmon/cnijnpr/autogen.sh	2013-01-08 09:37:15.885434060 +0100
@@ -112,7 +112,7 @@
 #      fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/cngpijmon/cnijnpr/configure.in cnijfilter-source-3.20-1_new/cngpijmon/cnijnpr/configure.in
--- cnijfilter-source-3.20-1/cngpijmon/cnijnpr/configure.in	2009-04-24 08:12:49.000000000 +0200
+++ cnijfilter-source-3.20-1_new/cngpijmon/cnijnpr/configure.in	2013-01-08 09:37:15.878767393 +0100
@@ -1,8 +1,7 @@
 AC_INIT(configure.in)
 AM_INIT_AUTOMAKE(cnijnpr, 3.20-1)
 
-AM_CONFIG_HEADER(config.h)
-
+AC_CONFIG_HEADERS([config.h])
 
 AC_ARG_ENABLE(libpath)
 if test "${enable_libpath}" = "" ; then
diff -aur cnijfilter-source-3.20-1/cngpijmon/configure.in cnijfilter-source-3.20-1_new/cngpijmon/configure.in
--- cnijfilter-source-3.20-1/cngpijmon/configure.in	2009-04-24 08:12:49.000000000 +0200
+++ cnijfilter-source-3.20-1_new/cngpijmon/configure.in	2013-01-08 09:37:15.885434060 +0100
@@ -18,11 +18,10 @@
 
 
 AM_INIT_AUTOMAKE(cngpijmon${program_suffix}, 3.20-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 
 AM_PATH_GTK_2_0(2.0.0, ,
diff -aur cnijfilter-source-3.20-1/cnijfilter/autogen.sh cnijfilter-source-3.20-1_new/cnijfilter/autogen.sh
--- cnijfilter-source-3.20-1/cnijfilter/autogen.sh	2009-03-26 06:11:05.000000000 +0100
+++ cnijfilter-source-3.20-1_new/cnijfilter/autogen.sh	2013-01-08 09:37:15.885434060 +0100
@@ -125,7 +125,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/cnijfilter/configure.in cnijfilter-source-3.20-1_new/cnijfilter/configure.in
--- cnijfilter-source-3.20-1/cnijfilter/configure.in	2009-03-26 06:11:05.000000000 +0100
+++ cnijfilter-source-3.20-1_new/cnijfilter/configure.in	2013-01-08 09:37:15.885434060 +0100
@@ -25,7 +25,7 @@
 
 
 AM_INIT_AUTOMAKE(cif${program_suffix}, 3.20-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 dnl Checks for programs.
 AC_ISC_POSIX
@@ -42,7 +42,6 @@
 AC_SUBST(CNCL_LIB_ID)
 
 AC_PROG_CC
-AM_PROG_CC_STDC
 
 dnl Checks for libraries.
 
diff -aur cnijfilter-source-3.20-1/lgmon/autogen.sh cnijfilter-source-3.20-1_new/lgmon/autogen.sh
--- cnijfilter-source-3.20-1/lgmon/autogen.sh	2009-04-24 08:19:35.000000000 +0200
+++ cnijfilter-source-3.20-1_new/lgmon/autogen.sh	2013-01-08 09:37:15.885434060 +0100
@@ -106,7 +106,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/lgmon/configure.in cnijfilter-source-3.20-1_new/lgmon/configure.in
--- cnijfilter-source-3.20-1/lgmon/configure.in	2009-04-24 08:19:35.000000000 +0200
+++ cnijfilter-source-3.20-1_new/lgmon/configure.in	2013-01-08 09:37:15.885434060 +0100
@@ -9,7 +9,7 @@
 fi
 
 AM_INIT_AUTOMAKE(lgmon{program_suffix}, 3.20-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 dnl AM_MAINTAINER_MODE
 
 AC_DEFINE_UNQUOTED(PACKAGE_PRINTER_MODEL, "${program_suffix}")
@@ -37,7 +37,6 @@
 AC_SUBST(CNCL_LIB_ID)
 
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AC_PROG_RANLIB
 
diff -aur cnijfilter-source-3.20-1/libs/autogen.sh cnijfilter-source-3.20-1_new/libs/autogen.sh
--- cnijfilter-source-3.20-1/libs/autogen.sh	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/libs/autogen.sh	2013-01-08 09:37:15.885434060 +0100
@@ -127,7 +127,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/libs/configure.in cnijfilter-source-3.20-1_new/libs/configure.in
--- cnijfilter-source-3.20-1/libs/configure.in	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/libs/configure.in	2013-01-08 09:37:15.882100727 +0100
@@ -1,7 +1,7 @@
 AC_INIT(paramlist/paramlist.h)
 AM_INIT_AUTOMAKE(buftool, 1.0.0)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 # Checks for programs.
 AC_PROG_CXX
diff -aur cnijfilter-source-3.20-1/ppd/autogen.sh cnijfilter-source-3.20-1_new/ppd/autogen.sh
--- cnijfilter-source-3.20-1/ppd/autogen.sh	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/ppd/autogen.sh	2013-01-08 09:37:15.885434060 +0100
@@ -126,7 +126,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/printui/autogen.sh cnijfilter-source-3.20-1_new/printui/autogen.sh
--- cnijfilter-source-3.20-1/printui/autogen.sh	2009-04-06 13:06:02.000000000 +0200
+++ cnijfilter-source-3.20-1_new/printui/autogen.sh	2013-01-08 09:37:15.885434060 +0100
@@ -138,7 +138,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/printui/configure.in cnijfilter-source-3.20-1_new/printui/configure.in
--- cnijfilter-source-3.20-1/printui/configure.in	2009-04-06 13:06:02.000000000 +0200
+++ cnijfilter-source-3.20-1_new/printui/configure.in	2013-01-08 09:37:15.888767393 +0100
@@ -9,7 +9,7 @@
 fi
 
 AM_INIT_AUTOMAKE(printui${program_suffix}, 3.20-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 AM_MAINTAINER_MODE
 
 AC_DEFINE_UNQUOTED(PACKAGE_PRINTER_MODEL, "${program_suffix}")
@@ -35,7 +35,6 @@
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 
 AM_PATH_GTK_2_0(2.6.0, ,
diff -aur cnijfilter-source-3.20-1/pstocanonij/autogen.sh cnijfilter-source-3.20-1_new/pstocanonij/autogen.sh
--- cnijfilter-source-3.20-1/pstocanonij/autogen.sh	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/pstocanonij/autogen.sh	2013-01-08 09:37:15.885434060 +0100
@@ -114,7 +114,7 @@
       fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
-      if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
+      if grep "^AC_CONFIG_HEADERS" configure.in >/dev/null; then
 	echo "Running autoheader..."
 	autoheader
       fi
diff -aur cnijfilter-source-3.20-1/pstocanonij/configure.in cnijfilter-source-3.20-1_new/pstocanonij/configure.in
--- cnijfilter-source-3.20-1/pstocanonij/configure.in	2009-04-06 11:49:22.000000000 +0200
+++ cnijfilter-source-3.20-1_new/pstocanonij/configure.in	2013-01-08 09:37:15.882100727 +0100
@@ -1,7 +1,7 @@
 AC_INIT(filter/pstocanonij.c)
 AM_INIT_AUTOMAKE(pstocanonij, 1.0.0)
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 AC_ARG_ENABLE(progpath)
 if test "${enable_progpath}" = "" ; then
