--- a/src/ppb_flash_file_modulelocal.c  2014-03-31 22:23:09.625187699 +0400
+++ b/src/ppb_flash_file_modulelocal.c  2014-03-31 22:27:12.298535687 +0400
@@ -29,7 +29,7 @@
 #include <ppapi/c/ppb_file_io.h>
 #include <stdlib.h>
 #include <stdio.h>
-#include <Uri.h>
+#include <uriparser/Uri.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <errno.h>
--- a/src/ppb_url_util_dev.c    2014-03-31 22:40:11.501916076 +0400
+++ b/src/ppb_url_util_dev.c    2014-03-31 22:43:16.001927217 +0400
@@ -26,7 +26,7 @@
 #include "ppb_var.h"
 #include <stdlib.h>
 #include <string.h>
-#include <Uri.h>
+#include <uriparser/Uri.h>
 #include "trace.h"
 #include "tables.h"
