--- desk.h.orig	2008-04-08 10:15:25.000000000 -0500
+++ desk.h	2008-04-07 10:46:18.000000000 -0500
@@ -110,7 +110,7 @@
   void LoadIcons (void);
   void setCheckMountProc (Icon * I);
   static void procIsMounted (Icon * I);
-  Boolean Desk::isMounted (char *mnt);
+  Boolean isMounted (char *mnt);
   static Boolean procExecJobs (XtPointer D);
   Boolean ExecJobs (void);
   inline Pixmap getUPixmap(void){return UnknownPixmap;};
@@ -128,7 +128,7 @@
   inline int getCaptionBorder(void) { return CaptionBorder; };
   inline int getsnapX (void) { return snapX; };
   inline int getsnapY (void) { return snapY; };
-  void Desk::LowerIcons(Icon *I);
+  void LowerIcons(Icon *I);
 };
 
 #endif
--- menu.h.orig	2008-04-08 10:15:41.000000000 -0500
+++ menu.h	2008-04-07 10:46:48.000000000 -0500
@@ -36,7 +36,7 @@
   };
   void AddCommand (char *item, char *command);
   void AddProps(char *item, Icon *I);
-  static void PopupMenu::procExec (Widget w, XtPointer client_data,
+  static void procExec (Widget w, XtPointer client_data,
 				   XtPointer call_data);
 };
 
--- icon.h.orig	2008-04-08 10:15:33.000000000 -0500
+++ icon.h	2008-04-07 10:47:05.000000000 -0500
@@ -106,8 +106,8 @@
   void updateMount (Boolean state);
   inline char* getConfigFile(void) { return ConfigFile; };
   void checkAccess( char *path, int mode ); 
-  void Icon::raiseme (void);
-  void Icon::lowerme (void);
+  void raiseme (void);
+  void lowerme (void);
 };
 
 #endif
--- desk.cc.orig	2015-01-29 19:37:47.388001955 -0600
+++ desk.cc	2015-01-29 19:37:53.771162476 -0600
@@ -174,7 +174,7 @@
 }
 static int Filter_Links (const struct dirent * dir)
 {
- char *p;
+ const char *p;
  static char *ext =".lnk";
  static int  extlen = strlen(ext);
  if( !(p = strstr(dir->d_name,  ext)) ) return 0;
--- Makefile.linux.orig	2015-01-29 19:41:30.541959462 -0600
+++ Makefile.linux	2015-01-29 19:49:22.139183281 -0600
@@ -6,7 +6,7 @@
 ## for Linux
 CFLAGS   = -w -I/usr/X11R6/include -DLINUX 
 #-DPRNDEBUG
-LIBS     = -L/usr/X11R6/lib -lImlib  -lXaw  -ldl
+LIBS     = -L/usr/X11R6/lib -lImlib  -lXaw  -ldl -lXext -lXt -lX11
 
 ## for FreeBSD
 #CFLAGS   = -w -I/usr/X11R6/include -DFREEBSD 
