diff --git a/sources/4th.h b/sources/4th.h
index dfdb36f..7778729 100644
--- a/sources/4th.h
+++ b/sources/4th.h
@@ -60,11 +60,11 @@
 #define M4DUPNAM 26
 #define M4CABORT 27
 
-#define CELL_MIN INT_MIN
-#define CELL_MAX INT_MAX
+#define CELL_MIN LONG_MIN
+#define CELL_MAX LONG_MAX
 #define strtocell(a,b,c) strtol((a),(b),(c))
 
-typedef int cell;
+typedef long cell;
 typedef unsigned char unit;
 
 typedef struct {
