diff -ruN spfs,a/fs/Makefile spfs,b/fs/Makefile
--- spfs,a/fs/Makefile	2006-09-07 22:05:07.000000000 +0200
+++ spfs,b/fs/Makefile	2010-10-13 21:29:09.000000000 +0200
@@ -11,6 +11,10 @@
 ufs: ufs.o
 	$(CC) -o ufs $(CFLAGS) ufs.o $(LFLAGS) $(NPFS_LFLAGS)
 
+install:
+	mkdir -p $(INSTALLPREFIX)/bin
+	cp ufs $(INSTALLPREFIX)/bin
+
 clean:
 	rm -f *.o *~ $(FS) core.*
 
diff -ruN spfs,a/libspclient/Makefile spfs,b/libspclient/Makefile
--- spfs,a/libspclient/Makefile	2007-07-03 21:53:40.000000000 +0200
+++ spfs,b/libspclient/Makefile	2010-10-13 21:29:23.000000000 +0200
@@ -26,8 +26,10 @@
 	ranlib libspclient.a
 
 install:
-	cp ../include/spclient.h /usr/local/include
-	cp libspclient.a /usr/local/lib
+	mkdir -p $(INSTALLPREFIX)/include
+	mkdir -p $(INSTALLPREFIX)/lib
+	cp ../include/spclient.h $(INSTALLPREFIX)/include
+	cp libspclient.a $(INSTALLPREFIX)/lib
 
 clean:
 	rm -f *.o *.a *~
diff -ruN spfs,a/libspfs/Makefile spfs,b/libspfs/Makefile
--- spfs,a/libspfs/Makefile	2007-07-06 22:35:18.000000000 +0200
+++ spfs,b/libspfs/Makefile	2010-10-13 21:26:59.000000000 +0200
@@ -22,8 +22,10 @@
 	ranlib libspfs.a
 
 install:
-	cp ../include/spfs.h /usr/local/include
-	cp libspfs.a /usr/local/lib
+	mkdir -p $(INSTALLPREFIX)/include
+	mkdir -p $(INSTALLPREFIX)/lib
+	cp ../include/spfs.h $(INSTALLPREFIX)/include
+	cp libspfs.a $(INSTALLPREFIX)/lib
 
 clean:
 	rm -f *.o *.a *~ core.*
diff -ruN spfs,a/Makefile spfs,b/Makefile
--- spfs,a/Makefile	2007-08-26 06:21:25.000000000 +0200
+++ spfs,b/Makefile	2010-10-13 21:24:53.000000000 +0200
@@ -1,5 +1,7 @@
 SYSNAME:=${shell uname}
 SYSNAME!=uname
+INSTALLPREFIX:=/usr/local
+export INSTALLPREFIX
 
 all: 
 	make -C libspfs
@@ -13,5 +15,6 @@
 	make -C fs clean
 
 install:
-	cp include/spfs.h /usr/local/include
-	cp libspfs/libspfs.a /usr/local/lib
+	make -C libspfs install
+	make -C libspclient install
+	make -C fs install
