From dc2e15d615b4e5341676828ec0e01558bcd30e4f Mon Sep 17 00:00:00 2001
From: Jon Nordby <jononor@gmail.com>
Date: Mon, 20 Dec 2010 20:36:15 +0100
Subject: [PATCH] Fix undefined reference to 'chmod'

---
 .../update-contextkit-providers.cpp                |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/libcontextsubscriber/update-contextkit-providers/update-contextkit-providers.cpp b/libcontextsubscriber/update-contextkit-providers/update-contextkit-providers.cpp
index 754f968..514a6b6 100644
--- a/libcontextsubscriber/update-contextkit-providers/update-contextkit-providers.cpp
+++ b/libcontextsubscriber/update-contextkit-providers/update-contextkit-providers.cpp
@@ -22,6 +22,7 @@
 #include <QCoreApplication>
 #include <QDir>
 #include <stdlib.h>
+#include <sys/stat.h>
 #include "contextregistryinfo.h"
 #include "contextpropertyinfo.h"
 #include "contextproviderinfo.h"
-- 
1.7.3.4

