diff -aur xdg-utils/scripts/xdg-open.in xdg-utils.new/scripts/xdg-open.in
--- xdg-utils/scripts/xdg-open.in	2009-11-07 23:15:29.022492880 -0600
+++ xdg-utils.new/scripts/xdg-open.in	2009-11-08 10:13:52.807798529 -0600
@@ -72,6 +72,17 @@
     fi
 }
 
+open_rox()
+{
+    rox "$1"
+
+    if [ $? -eq 0 ]; then
+        exit_success
+    else
+        exit_failure_operation_failed
+    fi
+}
+
 open_generic()
 {
     if mimeopen -v 2>/dev/null 1>&2; then
@@ -163,6 +174,10 @@
     open_xfce "$url"
     ;;
 
+    rox)
+    open_rox "$url"
+    ;;
+
     generic)
     open_generic "$url"
     ;;
diff -aur xdg-utils/scripts/xdg-utils-common.in xdg-utils.new/scripts/xdg-utils-common.in
--- xdg-utils/scripts/xdg-utils-common.in	2009-11-07 23:15:29.022492880 -0600
+++ xdg-utils.new/scripts/xdg-utils-common.in	2009-11-08 10:20:16.207739088 -0600
@@ -196,6 +196,7 @@
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde${KDE_SESSION_VERSION};
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
     elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif [ -e $HOME/.config/rox.sourceforge.net ]; then DE=rox;
     fi
 }
 
Only in xdg-utils.new/scripts: xdg-utils-common.in~
