--- src/qtsingleapplication/qtlocalpeer.cpp	2013-11-12 23:55:32.421886269 +0100
+++ src/qtsingleapplication/qtlocalpeer.cpp	2013-11-29 21:09:14.971598661 +0100
@@ -100,7 +100,7 @@
         socketName += QLatin1Char('-') + QString::number(sessionId, 16);
     }
 #else
-    socketName += QLatin1Char('-') + QString::number(::getuid(), 16);
+    socketName += QLatin1Char('-') + QString::number(QtLP_Private::getuid(), 16);
 #endif
 
     server = new QLocalServer(this);
