--- data/gshot.ui	2010-11-29 08:31:53.000000000 -0500
+++ /tmp/trunk/data/gshot.ui	2011-05-29 09:00:03.316329020 -0400
@@ -1,8 +1,14 @@
-<?xml version="1.0"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <interface>
   <requires lib="gtk+" version="2.16"/>
   <!-- interface-naming-policy project-wide -->
+  <object class="GtkAdjustment" id="delay_adjustment">
+    <property name="upper">60</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+  </object>
   <object class="GtkDialog" id="dialog">
+    <property name="can_focus">False</property>
     <property name="border_width">5</property>
     <property name="title" translatable="yes">Screenshot</property>
     <property name="resizable">False</property>
@@ -10,12 +16,44 @@
     <property name="type_hint">normal</property>
     <property name="skip_taskbar_hint">True</property>
     <property name="skip_pager_hint">True</property>
-    <signal name="delete_event" handler="exit_app"/>
+    <signal name="delete-event" handler="exit_app" swapped="no"/>
     <child internal-child="vbox">
       <object class="GtkVBox" id="dialog-vbox">
         <property name="visible">True</property>
-        <property name="orientation">vertical</property>
+        <property name="can_focus">False</property>
         <property name="spacing">2</property>
+        <child internal-child="action_area">
+          <object class="GtkHButtonBox" id="dialog-action_area">
+            <property name="visible">True</property>
+            <property name="can_focus">False</property>
+            <property name="layout_style">end</property>
+            <child>
+              <placeholder/>
+            </child>
+            <child>
+              <object class="GtkButton" id="close_button">
+                <property name="label">gtk-close</property>
+                <property name="visible">True</property>
+                <property name="can_focus">True</property>
+                <property name="receives_default">True</property>
+                <property name="use_action_appearance">False</property>
+                <property name="use_stock">True</property>
+                <signal name="clicked" handler="exit_app" swapped="no"/>
+              </object>
+              <packing>
+                <property name="expand">False</property>
+                <property name="fill">False</property>
+                <property name="position">1</property>
+              </packing>
+            </child>
+          </object>
+          <packing>
+            <property name="expand">False</property>
+            <property name="fill">True</property>
+            <property name="pack_type">end</property>
+            <property name="position">0</property>
+          </packing>
+        </child>
         <child>
           <object class="GtkNotebook" id="notebook">
             <property name="visible">True</property>
@@ -25,19 +63,22 @@
             <child>
               <object class="GtkVBox" id="main_box">
                 <property name="visible">True</property>
+                <property name="can_focus">False</property>
                 <property name="border_width">5</property>
-                <property name="orientation">vertical</property>
                 <child>
                   <object class="GtkHBox" id="hbox">
                     <property name="visible">True</property>
+                    <property name="can_focus">False</property>
                     <child>
                       <object class="GtkFrame" id="preview">
                         <property name="visible">True</property>
+                        <property name="can_focus">False</property>
                         <property name="border_width">5</property>
                         <property name="label_xalign">0</property>
                         <child>
                           <object class="GtkAlignment" id="preview_alignment">
                             <property name="visible">True</property>
+                            <property name="can_focus">False</property>
                             <property name="border_width">5</property>
                             <property name="top_padding">5</property>
                             <property name="bottom_padding">5</property>
@@ -48,6 +89,7 @@
                                 <property name="width_request">350</property>
                                 <property name="height_request">250</property>
                                 <property name="visible">True</property>
+                                <property name="can_focus">False</property>
                                 <property name="stock">gtk-missing-image</property>
                               </object>
                             </child>
@@ -56,12 +98,15 @@
                         <child type="label">
                           <object class="GtkLabel" id="preview_label">
                             <property name="visible">True</property>
+                            <property name="can_focus">False</property>
                             <property name="label" translatable="yes">&lt;b&gt;Preview&lt;/b&gt;</property>
                             <property name="use_markup">True</property>
                           </object>
                         </child>
                       </object>
                       <packing>
+                        <property name="expand">True</property>
+                        <property name="fill">True</property>
                         <property name="padding">2</property>
                         <property name="position">0</property>
                       </packing>
@@ -69,8 +114,8 @@
                     <child>
                       <object class="GtkVButtonBox" id="buttonbox">
                         <property name="visible">True</property>
+                        <property name="can_focus">False</property>
                         <property name="border_width">5</property>
-                        <property name="orientation">vertical</property>
                         <property name="spacing">5</property>
                         <property name="layout_style">start</property>
                         <child>
@@ -80,8 +125,10 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">True</property>
                             <property name="tooltip_text" translatable="yes">Make a screenshot</property>
+                            <property name="use_action_appearance">False</property>
                             <property name="use_stock">True</property>
-                            <signal name="clicked" handler="capture_mode"/>
+                            <property name="xalign">0</property>
+                            <signal name="clicked" handler="capture_mode" swapped="no"/>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -93,6 +140,7 @@
                         <child>
                           <object class="GtkHSeparator" id="separator1">
                             <property name="visible">True</property>
+                            <property name="can_focus">False</property>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -107,8 +155,10 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">True</property>
                             <property name="tooltip_text" translatable="yes">Save screenshot to file</property>
+                            <property name="use_action_appearance">False</property>
                             <property name="use_stock">True</property>
-                            <signal name="clicked" handler="save_clicked"/>
+                            <property name="xalign">0</property>
+                            <signal name="clicked" handler="save_clicked" swapped="no"/>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -124,8 +174,10 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">True</property>
                             <property name="tooltip_text" translatable="yes">Print screenshot</property>
+                            <property name="use_action_appearance">False</property>
                             <property name="use_stock">True</property>
-                            <signal name="clicked" handler="print_clicked"/>
+                            <property name="xalign">0</property>
+                            <signal name="clicked" handler="print_clicked" swapped="no"/>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -141,8 +193,10 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">True</property>
                             <property name="tooltip_text" translatable="yes">Copy screenshot to clipboard</property>
+                            <property name="use_action_appearance">False</property>
                             <property name="use_stock">True</property>
-                            <signal name="clicked" handler="copy_clicked"/>
+                            <property name="xalign">0</property>
+                            <signal name="clicked" handler="copy_clicked" swapped="no"/>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -154,6 +208,7 @@
                         <child>
                           <object class="GtkHSeparator" id="separator2">
                             <property name="visible">True</property>
+                            <property name="can_focus">False</property>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -168,8 +223,10 @@
                             <property name="can_focus">True</property>
                             <property name="receives_default">True</property>
                             <property name="tooltip_text" translatable="yes">Show about dialog</property>
+                            <property name="use_action_appearance">False</property>
                             <property name="use_stock">True</property>
-                            <signal name="clicked" handler="show_about"/>
+                            <property name="xalign">0</property>
+                            <signal name="clicked" handler="show_about" swapped="no"/>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -178,13 +235,20 @@
                             <property name="position">6</property>
                           </packing>
                         </child>
+                        <child>
+                          <placeholder/>
+                        </child>
                       </object>
                       <packing>
+                        <property name="expand">False</property>
+                        <property name="fill">True</property>
                         <property name="position">1</property>
                       </packing>
                     </child>
                   </object>
                   <packing>
+                    <property name="expand">True</property>
+                    <property name="fill">True</property>
                     <property name="padding">2</property>
                     <property name="position">0</property>
                   </packing>
@@ -192,14 +256,17 @@
                 <child>
                   <object class="GtkHBox" id="mode_box">
                     <property name="visible">True</property>
+                    <property name="can_focus">False</property>
                     <property name="spacing">1</property>
                     <child>
                       <object class="GtkLabel" id="mode_label">
                         <property name="visible">True</property>
+                        <property name="can_focus">False</property>
                         <property name="label" translatable="yes">Mode:</property>
                       </object>
                       <packing>
                         <property name="expand">False</property>
+                        <property name="fill">True</property>
                         <property name="padding">2</property>
                         <property name="position">0</property>
                       </packing>
@@ -207,9 +274,10 @@
                     <child>
                       <object class="GtkComboBox" id="mode_combo">
                         <property name="visible">True</property>
+                        <property name="can_focus">False</property>
                         <property name="tooltip_text" translatable="yes">Choose screenshot mode</property>
                         <property name="model">mode_store</property>
-                        <signal name="changed" handler="mode_changed"/>
+                        <signal name="changed" handler="mode_changed" swapped="no"/>
                         <child>
                           <object class="GtkCellRendererText" id="mode_cell"/>
                           <attributes>
@@ -218,6 +286,8 @@
                         </child>
                       </object>
                       <packing>
+                        <property name="expand">True</property>
+                        <property name="fill">True</property>
                         <property name="padding">2</property>
                         <property name="position">1</property>
                       </packing>
@@ -225,6 +295,7 @@
                   </object>
                   <packing>
                     <property name="expand">False</property>
+                    <property name="fill">True</property>
                     <property name="padding">2</property>
                     <property name="position">1</property>
                   </packing>
@@ -232,14 +303,17 @@
                 <child>
                   <object class="GtkHBox" id="delay_box">
                     <property name="visible">True</property>
+                    <property name="can_focus">False</property>
                     <property name="spacing">1</property>
                     <child>
                       <object class="GtkLabel" id="delay_label">
                         <property name="visible">True</property>
+                        <property name="can_focus">False</property>
                         <property name="label" translatable="yes">Delay:</property>
                       </object>
                       <packing>
                         <property name="expand">False</property>
+                        <property name="fill">True</property>
                         <property name="padding">2</property>
                         <property name="position">0</property>
                       </packing>
@@ -252,9 +326,11 @@
                         <property name="adjustment">delay_adjustment</property>
                         <property name="digits">0</property>
                         <property name="value_pos">right</property>
-                        <signal name="value_changed" handler="set_delay"/>
+                        <signal name="value-changed" handler="set_delay" swapped="no"/>
                       </object>
                       <packing>
+                        <property name="expand">True</property>
+                        <property name="fill">True</property>
                         <property name="padding">2</property>
                         <property name="position">1</property>
                       </packing>
@@ -262,6 +338,7 @@
                   </object>
                   <packing>
                     <property name="expand">False</property>
+                    <property name="fill">True</property>
                     <property name="padding">2</property>
                     <property name="position">2</property>
                   </packing>
@@ -271,6 +348,7 @@
             <child type="tab">
               <object class="GtkLabel" id="main_label">
                 <property name="visible">True</property>
+                <property name="can_focus">False</property>
                 <property name="label" translatable="yes">Main</property>
               </object>
               <packing>
@@ -280,8 +358,8 @@
             <child>
               <object class="GtkVBox" id="settings_box">
                 <property name="visible">True</property>
+                <property name="can_focus">False</property>
                 <property name="border_width">5</property>
-                <property name="orientation">vertical</property>
                 <child>
                   <object class="GtkCheckButton" id="frame_check">
                     <property name="label" translatable="yes">Include window frame</property>
@@ -289,12 +367,14 @@
                     <property name="can_focus">True</property>
                     <property name="receives_default">False</property>
                     <property name="tooltip_text" translatable="yes">Capture window with decorations</property>
+                    <property name="use_action_appearance">False</property>
                     <property name="active">True</property>
                     <property name="draw_indicator">True</property>
-                    <signal name="toggled" handler="frame_toggled"/>
+                    <signal name="toggled" handler="frame_toggled" swapped="no"/>
                   </object>
                   <packing>
                     <property name="expand">False</property>
+                    <property name="fill">True</property>
                     <property name="padding">3</property>
                     <property name="position">0</property>
                   </packing>
@@ -306,12 +386,14 @@
                     <property name="can_focus">True</property>
                     <property name="receives_default">False</property>
                     <property name="tooltip_text" translatable="yes">Hide this window before capture</property>
+                    <property name="use_action_appearance">False</property>
                     <property name="active">True</property>
                     <property name="draw_indicator">True</property>
-                    <signal name="toggled" handler="hide_toggled"/>
+                    <signal name="toggled" handler="hide_toggled" swapped="no"/>
                   </object>
                   <packing>
                     <property name="expand">False</property>
+                    <property name="fill">True</property>
                     <property name="padding">3</property>
                     <property name="position">1</property>
                   </packing>
@@ -319,9 +401,11 @@
                 <child>
                   <object class="GtkHBox" id="folder_box">
                     <property name="visible">True</property>
+                    <property name="can_focus">False</property>
                     <child>
                       <object class="GtkLabel" id="folder_label">
                         <property name="visible">True</property>
+                        <property name="can_focus">False</property>
                         <property name="label" translatable="yes">Default folder:</property>
                       </object>
                       <packing>
@@ -334,12 +418,15 @@
                     <child>
                       <object class="GtkFileChooserButton" id="folder_button">
                         <property name="visible">True</property>
+                        <property name="can_focus">False</property>
                         <property name="tooltip_text" translatable="yes">Set default folder for screenshots</property>
                         <property name="action">select-folder</property>
                         <property name="title" translatable="yes">Select folder</property>
-                        <signal name="file_set" handler="set_folder"/>
+                        <signal name="file-set" handler="set_folder" swapped="no"/>
                       </object>
                       <packing>
+                        <property name="expand">True</property>
+                        <property name="fill">True</property>
                         <property name="padding">3</property>
                         <property name="position">1</property>
                       </packing>
@@ -347,6 +434,7 @@
                   </object>
                   <packing>
                     <property name="expand">False</property>
+                    <property name="fill">True</property>
                     <property name="padding">3</property>
                     <property name="position">2</property>
                   </packing>
@@ -358,11 +446,13 @@
                     <property name="can_focus">True</property>
                     <property name="receives_default">False</property>
                     <property name="tooltip_text" translatable="yes">Save settings on exit</property>
+                    <property name="use_action_appearance">False</property>
                     <property name="draw_indicator">True</property>
-                    <signal name="toggled" handler="save_toggled"/>
+                    <signal name="toggled" handler="save_toggled" swapped="no"/>
                   </object>
                   <packing>
                     <property name="expand">False</property>
+                    <property name="fill">True</property>
                     <property name="padding">3</property>
                     <property name="position">3</property>
                   </packing>
@@ -375,12 +465,14 @@
                     <property name="receives_default">False</property>
                     <property name="events">GDK_POINTER_MOTION_MASK | GDK_POINTER_MOTION_HINT_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK</property>
                     <property name="tooltip_text" translatable="yes">Show icon in system notification area</property>
+                    <property name="use_action_appearance">False</property>
                     <property name="active">True</property>
                     <property name="draw_indicator">True</property>
-                    <signal name="toggled" handler="tray_toggled"/>
+                    <signal name="toggled" handler="tray_toggled" swapped="no"/>
                   </object>
                   <packing>
                     <property name="expand">False</property>
+                    <property name="fill">True</property>
                     <property name="padding">3</property>
                     <property name="position">4</property>
                   </packing>
@@ -388,6 +480,7 @@
                 <child>
                   <object class="GtkAlignment" id="tray_alignment">
                     <property name="visible">True</property>
+                    <property name="can_focus">False</property>
                     <property name="left_padding">15</property>
                     <child>
                       <object class="GtkCheckButton" id="tray_hide_check">
@@ -397,8 +490,9 @@
                         <property name="receives_default">False</property>
                         <property name="events">GDK_POINTER_MOTION_MASK | GDK_POINTER_MOTION_HINT_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK</property>
                         <property name="tooltip_text" translatable="yes">Hide window on startup</property>
+                        <property name="use_action_appearance">False</property>
                         <property name="draw_indicator">True</property>
-                        <signal name="toggled" handler="tray_hide_toggled"/>
+                        <signal name="toggled" handler="tray_hide_toggled" swapped="no"/>
                       </object>
                     </child>
                   </object>
@@ -417,6 +511,7 @@
             <child type="tab">
               <object class="GtkLabel" id="settings_label">
                 <property name="visible">True</property>
+                <property name="can_focus">False</property>
                 <property name="label" translatable="yes">Settings</property>
               </object>
               <packing>
@@ -426,39 +521,12 @@
             </child>
           </object>
           <packing>
+            <property name="expand">True</property>
+            <property name="fill">True</property>
             <property name="padding">1</property>
             <property name="position">1</property>
           </packing>
         </child>
-        <child internal-child="action_area">
-          <object class="GtkHButtonBox" id="dialog-action_area">
-            <property name="visible">True</property>
-            <property name="layout_style">end</property>
-            <child>
-              <placeholder/>
-            </child>
-            <child>
-              <object class="GtkButton" id="close_button">
-                <property name="label">gtk-close</property>
-                <property name="visible">True</property>
-                <property name="can_focus">True</property>
-                <property name="receives_default">True</property>
-                <property name="use_stock">True</property>
-                <signal name="clicked" handler="exit_app"/>
-              </object>
-              <packing>
-                <property name="expand">False</property>
-                <property name="fill">False</property>
-                <property name="position">1</property>
-              </packing>
-            </child>
-          </object>
-          <packing>
-            <property name="expand">False</property>
-            <property name="pack_type">end</property>
-            <property name="position">0</property>
-          </packing>
-        </child>
       </object>
     </child>
     <action-widgets>
@@ -484,39 +552,49 @@
   </object>
   <object class="GtkMenu" id="tray_menu">
     <property name="visible">True</property>
+    <property name="can_focus">False</property>
     <child>
       <object class="GtkMenuItem" id="tray_screen">
         <property name="visible">True</property>
+        <property name="can_focus">False</property>
+        <property name="use_action_appearance">False</property>
         <property name="label" translatable="yes">Entire screen</property>
         <property name="use_underline">True</property>
-        <signal name="activate" handler="capture_screen" after="yes"/>
+        <signal name="activate" handler="capture_screen" after="yes" swapped="no"/>
       </object>
     </child>
     <child>
       <object class="GtkMenuItem" id="tray_window">
         <property name="visible">True</property>
+        <property name="can_focus">False</property>
+        <property name="use_action_appearance">False</property>
         <property name="label" translatable="yes">Single window</property>
         <property name="use_underline">True</property>
-        <signal name="activate" handler="capture_window" after="yes"/>
+        <signal name="activate" handler="capture_window" after="yes" swapped="no"/>
       </object>
     </child>
     <child>
       <object class="GtkMenuItem" id="tray_region">
         <property name="visible">True</property>
+        <property name="can_focus">False</property>
+        <property name="use_action_appearance">False</property>
         <property name="label" translatable="yes">Selected region</property>
         <property name="use_underline">True</property>
-        <signal name="activate" handler="capture_region" after="yes"/>
+        <signal name="activate" handler="capture_region" after="yes" swapped="no"/>
       </object>
     </child>
     <child>
       <object class="GtkSeparatorMenuItem" id="tray_separator1">
         <property name="visible">True</property>
+        <property name="can_focus">False</property>
       </object>
     </child>
     <child>
       <object class="GtkImageMenuItem" id="tray_save">
         <property name="label">gtk-save-as</property>
         <property name="visible">True</property>
+        <property name="can_focus">False</property>
+        <property name="use_action_appearance">False</property>
         <property name="use_underline">True</property>
         <property name="use_stock">True</property>
       </object>
@@ -525,6 +603,8 @@
       <object class="GtkImageMenuItem" id="tray_print">
         <property name="label">gtk-print</property>
         <property name="visible">True</property>
+        <property name="can_focus">False</property>
+        <property name="use_action_appearance">False</property>
         <property name="use_underline">True</property>
         <property name="use_stock">True</property>
       </object>
@@ -532,21 +612,19 @@
     <child>
       <object class="GtkSeparatorMenuItem" id="tray_separator2">
         <property name="visible">True</property>
+        <property name="can_focus">False</property>
       </object>
     </child>
     <child>
       <object class="GtkImageMenuItem" id="tray_quit">
         <property name="label">gtk-quit</property>
         <property name="visible">True</property>
+        <property name="can_focus">False</property>
+        <property name="use_action_appearance">False</property>
         <property name="use_underline">True</property>
         <property name="use_stock">True</property>
-        <signal name="activate" handler="gtk_main_quit"/>
+        <signal name="activate" handler="gtk_main_quit" swapped="no"/>
       </object>
     </child>
   </object>
-  <object class="GtkAdjustment" id="delay_adjustment">
-    <property name="upper">60</property>
-    <property name="step_increment">1</property>
-    <property name="page_increment">10</property>
-  </object>
 </interface>
--- src/callback.c	2010-11-29 11:28:05.000000000 -0500
+++ ../trunk/src/callback.c	2011-05-29 09:00:03.116330104 -0400
@@ -32,6 +32,7 @@
 capture_image (gint mode)
 {
   gboolean state;
+  gint cmode = mode;
   GtkWidget *win = GTK_WIDGET (gtk_builder_get_object (data.xml, "dialog"));
 
   /* Free previously captured image */
@@ -48,10 +49,16 @@
 	gtk_main_iteration ();
     }
 
+  /* sleep if needed */
+  if (data.delay)
+    {
+      sleep (data.delay);
+      while (gtk_events_pending ())
+        gtk_main_iteration ();
+    }
+  
   /* Capture in the specified mode and store image in global data */
-  sleep (data.delay);
-
-  if (data.image = capture (win, mode, data.frame))
+  if (data.image = capture (win, cmode, data.frame))
     {
       GdkPixbuf *p;
       gdouble factor;
@@ -68,8 +75,7 @@
 				       GDK_INTERP_BILINEAR))
         {
           gtk_image_set_from_pixbuf (GTK_IMAGE (gtk_builder_get_object
-						(data.xml, "preview_image")),
-				     p);
+						(data.xml, "preview_image")), p);
           g_object_unref (p);
         }
     }
--- src/grabber.c	2010-11-29 08:22:08.000000000 -0500
+++ ../trunk/src/grabber.c	2011-05-29 09:00:03.112996789 -0400
@@ -133,7 +133,7 @@
   else
     {
       gint xpos, ypos;
-      window = gdk_window_foreign_new (xid);
+      window = GDK_WINDOW (gdk_window_foreign_new (xid));
 
       gdk_drawable_get_size (window, &width, &height);
       gdk_window_get_origin (window, &xpos, &ypos);
--- src/main.c	2010-11-29 12:47:50.000000000 -0500
+++ ../trunk/src/main.c	2011-05-29 09:00:03.109663474 -0400
@@ -203,10 +203,16 @@
     {
       show_tray ();
       if (data.tray_hide)
-	gtk_widget_hide (GTK_WIDGET (gtk_builder_get_object (data.xml, "dialog")));
+        gtk_widget_hide (GTK_WIDGET (gtk_builder_get_object (data.xml, "dialog")));
+      else
+        gtk_widget_show (GTK_WIDGET (gtk_builder_get_object (data.xml, "dialog")));
     }
   else
-    gtk_widget_set_sensitive (w, FALSE);
+    {
+      gtk_widget_show (GTK_WIDGET (gtk_builder_get_object (data.xml, "dialog")));
+      gtk_widget_set_sensitive (w, FALSE);
+    }
+  
 
 #ifdef HAVE_UNIQUE
   /* register unique application */
