--- mkspecs/features/qt_module.prf.orig	2013-07-31 20:37:56.789883201 +0200
+++ mkspecs/features/qt_module.prf	2013-07-31 20:38:06.013034478 +0200
@@ -25,7 +25,7 @@
 
 # Compile as shared/DLL or static according to the option given to configure
 # unless overridden. Host builds are always static
-host_build|staticlib: CONFIG += static
+#host_build|staticlib: CONFIG += static
 
 host_build {
     QT -= gui   # no host module will ever use gui
