--- configure.orig	2014-01-02 11:31:45.972854453 +0100
+++ configure	2014-01-02 11:30:37.076201181 +0100
@@ -4923,6 +4923,7 @@ if [ "$XPLATFORM_MINGW" = "yes" ]; then
             exit 1
         fi
     elif [ "$CFG_OPENGL" = "desktop" ]; then
+        CFG_OPENGL=angle
         # Desktop OpenGL support
         compileTest x11/opengl "OpenGL"
         if [ $? != "0" ]; then
@@ -4959,7 +4960,7 @@ fi
 # auto-detect OpenGL support (es2 = OpenGL ES 2.x)
 if [ "$CFG_OPENGL" = "auto" ] || [ "$CFG_OPENGL" = "yes" ]; then
     if compileTest unix/opengldesktop "OpenGL"; then
-        CFG_OPENGL=desktop
+        CFG_OPENGL=angle
     elif compileTest unix/opengles2 "OpenGL ES 2.x"; then
         CFG_OPENGL=es2
     else
@@ -4993,6 +4994,7 @@ elif [ "$CFG_OPENGL" = "es2" ]; then
         exit 1
     fi
 elif [ "$CFG_OPENGL" = "desktop" ]; then
+    CFG_OPENGL=angle
     # Desktop OpenGL support
     compileTest unix/opengldesktop "OpenGL"
     if [ $? != "0" ]; then
