--- Farmerjoe.conf	2009-06-03 12:14:58.029554703 +0200
+++ Farmerjoe.new	2009-06-03 12:20:26.329523561 +0200
@@ -6,16 +6,8 @@
 logs = logs
 
 linux_root = /render
-linux_blender = /render/bin/linux/blender_2.42a/blender
+linux_blender = /usr/bin/blender
 linux_composite = /usr/bin/composite
 
-windows_root = r:
-windows_blender = r:\bin\windows\blender\Blender.exe
-windows_composite = composite
-
-osx_root = /Volumes/farmerjoe
-osx_blender = /Volumes/farmerjoe/bin/osx/blender/blender.app/Contents/MacOS/blender
-osx_composite = /usr/local/bin/composite
-
 # Application server Configuration
 appserver_port = 2007
