--- src/CMakeLists.txt	(revision 521)
+++ src/CMakeLists.txt	(working copy)
@@ -90,6 +90,6 @@
 
 #${LIBTCOD_CPP_NAME} is empty on non-linux platforms
 INSTALL(TARGETS ${LIBTCOD_NAME} ${LIBTCOD_GUI_NAME} ${LIBTCOD_CPP_NAME}
-  RUNTIME DESTINATION ${INSTALL_DIR}
-  LIBRARY DESTINATION ${INSTALL_DIR}
+  RUNTIME DESTINATION ${DESTDIR}${CMAKE_INSTALL_PREFIX}/bin
+  LIBRARY DESTINATION ${DESTDIR}${CMAKE_INSTALL_PREFIX}/lib
 )
--- samples/CMakeLists.txt	(revision 521)
+++ samples/CMakeLists.txt	(working copy)
@@ -64,6 +64,6 @@
 ENDIF()
 
 INSTALL(TARGETS samples_c samples_cpp hmtool
-  RUNTIME DESTINATION ${INSTALL_DIR}
-  LIBRARY DESTINATION ${INSTALL_DIR}
+  RUNTIME DESTINATION ${DESTDIR}${CMAKE_INSTALL_PREFIX}/share/libtcod/bin
+  LIBRARY DESTINATION ${DESTDIR}${CMAKE_INSTALL_PREFIX}/share/libtcod/lib
 )
--- CMakeLists.txt	(revision 521)
+++ CMakeLists.txt	(working copy)
@@ -53,6 +53,9 @@
 	IF(CMAKE_BUILD_TYPE MATCHES "Debug")
 		find_library(EFENCE_LIBRARY efence PATHS /usr/lib /usr/local/lib)
 	ENDIF()
+	INSTALL(DIRECTORY include/ DESTINATION ${DESTDIR}${CMAKE_INSTALL_PREFIX}/include/libtcod
+	    PATTERN ".svn" EXCLUDE
+	    PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ)
 ENDIF()
 
 add_subdirectory(src) #Needs to come first since samples depend on it
