Index: src/sys_sdl_img_png.c
===================================================================
--- src/sys_sdl_img_png.c	(revision 521)
+++ src/sys_sdl_img_png.c	(working copy)
@@ -34,10 +34,8 @@
 #include "libtcod.h"
 #include "libtcod_int.h"
 
-#if defined( TCOD_MACOSX ) || defined( TCOD_HAIKU ) 
 #define png_infopp_NULL (png_infop)NULL
 #define int_p_NULL (int*)NULL
-#endif
 
 bool TCOD_sys_check_png(const char *filename) {
 	static uint8 magic_number[]={137, 80, 78, 71, 13, 10, 26, 10};
