--- mindi-2.1.1/install.sh	2012-02-24 01:36:32.000000000 +0000
+++ mindi-2.1.1/install-new.sh	2012-02-28 17:42:32.000000000 +0000
@@ -87,6 +87,22 @@
 sed -e "s~^MINDI_PREFIX=XXX~MINDI_PREFIX=$sublocal~" -e "s~^MINDI_CONF=YYY~MINDI_CONF=$subconf~" -e "s~^MINDI_LIB=LLL~MINDI_LIB=$sublocallib~" mindi > $local/sbin/mindi
 chmod 755 $local/sbin/mindi $local/sbin/mindi-bkphw
 
+# patching udev paths
+UDEVADM_PATH=$(type -p udevadm)
+UDEVD_PATH=$(type -p udevd)
+GREP_PATH=$(type -p grep)
+
+UDEVADM_PATH=$(echo $UDEVADM_PATH | sed 's/\//\\\//g')
+UDEVD_PATH=$(echo $UDEVD_PATH | sed 's/\//\\\//g')
+GREP_PATH=$(echo $GREP_PATH | sed 's/\//\\\//g')
+
+sed -i.tmp -e "s/\/bin\/grep/$GREP_PATH/g" -e "s/\/sbin\/udevd/$UDEVD_PATH/g" $local/sbin/mindi 
+"rm" -f $local/sbin/mindi.tmp
+
+sed -i.tmp -e "s/\/sbin\/udevd/$UDEVD_PATH/g" -e "s/\/sbin\/udevadm/$UDEVADM_PATH/g" $locallib/mindi/rootfs/etc/init.d/rcS 
+"rm" -f $locallib/mindi/rootfs/etc/init.d/rcS.tmp
+####
+
 install -m 644 mindi.8 $MANDIR
 #install -m 644 ChangeLog COPYING README README.busybox README.ia64 README.pxe TODO INSTALL svn.log $DOCDIR
 
