--- mindi-2.0.7.6/mindi	2011-04-03 18:34:54.886666669 +0100
+++ mindi-2.0.7.6/mindi-new	2011-04-03 18:36:30.303333336 +0100
@@ -2998,9 +2998,9 @@
 fi
 
 # for Mandrake 9.2, which comes with two aes.o.gz modules :-/
-insmod /lib/modules/$KERVERRUN/*/*/misc/aes.*o.gz >> $LOGFILE 2>> $LOGFILE
+#insmod /lib/modules/$KERVERRUN/*/*/misc/aes.*o.gz >> $LOGFILE 2>> $LOGFILE
 for i in loop cdrom ide-cd isofs linear raid0 raid1 raid5 ; do
-	insmod $i >> $LOGFILE 2>> $LOGFILE
+	modprobe $i >> $LOGFILE 2>> $LOGFILE
 done
 
 KERN_DISK_MADE=""
