--- mindi-2.1.1/deplist.d/minimal.conf	2012-02-16 02:18:39.000000000 +0000
+++ mindi-2.1.1/deplist.d/minimal.conf.new	2012-02-27 23:22:52.000000000 +0000
@@ -19,11 +19,11 @@
 /usr/bin/loadkeys
 /bin/loadkeys
 /sbin/pidof
-# On ArchLinux
-/bin/pidof
 /etc/init.d/functions
 # On ArchLinux
+/bin/pidof
 /etc/rc.d/functions
+/usr/bin/grep
 
 # Compression
 /bin/gzip
@@ -79,8 +79,14 @@
 /sbin/hwup
 /sbin/path_id
 /sbin/scsi_tur
-# On ArchLinux
-/sbin/udevadm
+
+# On ArchLinux # new udev paths
+/usr/bin/udevadm
+/usr/bin/udevd
+/usr/bin/kmod
+/usr/bin/insmod
+/usr/bin/lsmod
+
 /sbin/udevstart.static
 /sbin/udev_volume_id
 /sbin/start_udev
