--- mindi-2.1.1/deplist.d/net.conf	2012-02-16 13:06:08.000000000 +0000
+++ mindi-2.1.1/deplist.d/net.conf.new	2012-02-27 23:23:50.845138257 +0000
@@ -13,12 +13,10 @@
 /var/lib/dhcp/dhclient.leases
 /var/lib/dhclient/dhclient.leases
 /var/lib/dhclient
-# On ArchLinnux
-/sbin/dhcpcd
-/var/lib/dhcpd
-# On ArchLinnux
-/sbin/dhcpcd
-/var/lib/dhcpd
+
+# On ArchLinux
+/usr/sbin/dhcpcd
+/var/lib/dhcpcd
 
 # Useful for debug
 /usr/sbin/ethtool
