--- mindi-2.1.1/udev.files	2009-02-14 01:57:37.000000000 +0000
+++ mindi-2.1.1/udev.files.new	2012-02-27 23:40:55.981748037 +0000
@@ -6,9 +6,12 @@
 # 
 # $Id$
 #
+/usr/bin/kmod
 /sbin/udevd
+/usr/bin/udevd
 /sbin/modprobe
 /sbin/insmod
+/usr/bin/insmod
 /sbin/udevstart
 /sbin/udevcontrol
 /sbin/udevtrigger
@@ -18,5 +21,6 @@
 /sbin/create_static_dev_nodes
 /sbin/start_udev
 /sbin/udevadm
+/usr/bin/udevadm
 /sbin/usb_id
 /sbin/vol_id
