--- src/KDAB-Charm-6b7117a/CMakeLists.txt	2011-12-07 14:35:34.000000000 +0100
+++ ./CMakeLists.txt	2011-12-13 11:54:35.445317832 +0100
@@ -19,6 +19,11 @@
 SET( Charm_VERSION_PATCH "${CMAKE_MATCH_3}" )
 SET( Charm_VERSION_COUNT 3 )
 
+IF( NOT ( Charm_VERSION_MAJOR AND Charm_VERSION_MINOR AND Charm_VERSION_PATCH ) )
+    MESSAGE( FATAL_ERROR "No Git executable or valid Charm version argument found.\n"
+             "Please pass a version to CMake e.g. cmake -DCharm_VERSION=1.0.0" )
+ENDIF()
+
 IF( NOT CMAKE_BUILD_TYPE )
         SET( CMAKE_BUILD_TYPE "Release" CACHE STRING "" FORCE )
 ENDIF()
@@ -30,6 +35,7 @@
 MESSAGE( STATUS "Building Charm ${Charm_VERSION} in ${CMAKE_BUILD_TYPE} mode" )
 
 CMAKE_MINIMUM_REQUIRED( VERSION 2.6.0 )
+SET( CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/CMake" )
 FIND_PACKAGE( Qt4 4.6.0 COMPONENTS QtMain QtCore QtGui QtSql QtXml QtNetwork REQUIRED )
 INCLUDE( UseQt4 )
 ENABLE_TESTING()
