--- smstools3.orig/src/locking.c	2010-08-18 15:34:57.000000000 +0400
+++ smstools3/src/locking.c	2012-03-19 21:03:11.000000000 +0400
@@ -42,7 +42,8 @@ int lockfile( char*  filename)
     lockfile=open(lockfilename,O_CREAT|O_EXCL|O_WRONLY,0644);
     if (lockfile>=0)
     {
-      snprintf(pid, sizeof(pid), "%i %s\n", (int)getpid(), DEVICE.name);
+      snprintf(pid, sizeof(pid), "%i %s\n", (int)getpid(), 
+               (process_id == -1) ? "MAINPROCESS" : DEVICE.name);
       write(lockfile, pid, strlen(pid));
       close(lockfile);
       sync();
