diff -ur mdbtools-0.6pre1/include/mdbtools.h mdbtools-0.6pre1/include/mdbtools.h
--- mdbtools-0.6pre1/include/mdbtools.h	2012-03-15 13:40:01.475887724 -0400
+++ mdbtools-0.6pre1/include/mdbtools.h	2012-03-15 13:40:34.786290876 -0400
@@ -147,7 +147,7 @@
 #define IS_JET3(mdb) (mdb->f->jet_version==MDB_VER_JET3)
 
 /* hash to store registered backends */
-extern GHashTable	*mdb_backends;
+static GHashTable	*mdb_backends;
 
 /* forward declarations */
 typedef struct mdbindex MdbIndex;
