diff -aur xfirepower-0.84.orig/client/libsprite/color.c xfirepower-0.84/client/libsprite/color.c
--- xfirepower-0.84.orig/client/libsprite/color.c	1995-05-24 04:02:23.000000000 +0600
+++ xfirepower-0.84/client/libsprite/color.c	2012-05-15 16:55:21.866317844 +0600
@@ -83,6 +83,10 @@
 {
     XColor xcolor;
     
+    /* XStoreColor crashes TrueColor and Mono displays! */
+    if(DisplayPlanes(W_Display, W_Screen) != 8)
+	return;
+
     xcolor.pixel = colortable[color].pixelValue;
     xcolor.red = (unsigned short)r;
     xcolor.green = (unsigned short)g;
diff -aur xfirepower-0.84.orig/client/libsprite/data.c xfirepower-0.84/client/libsprite/data.c
--- xfirepower-0.84.orig/client/libsprite/data.c	1995-05-27 08:12:53.000000000 +0600
+++ xfirepower-0.84/client/libsprite/data.c	2012-05-15 16:55:57.139651375 +0600
@@ -80,7 +80,9 @@
 W_Color W_Green2 = GREEN2, W_Blue = BLUE, W_DarkGreen = DARKGREEN, W_Brown = BROWN;
 W_Color W_Yellow = YELLOW, W_Cyan = CYAN, W_Grey = LIGHT_GREY, W_DarkGrey = DARK_GREY;
 int     W_Textwidth, W_Textheight, W_BigTextwidth, W_BigTextheight;
+#ifndef strdup
 char   *strdup();
+#endif
 
 int     W_in_message = 0;	/* jfy -- for Jerry's warp message hack */
 
diff -aur xfirepower-0.84.orig/client/libsprite/init.c xfirepower-0.84/client/libsprite/init.c
--- xfirepower-0.84.orig/client/libsprite/init.c	1995-12-17 13:24:24.000000000 +0500
+++ xfirepower-0.84/client/libsprite/init.c	2012-05-15 16:56:18.992984833 +0600
@@ -338,7 +338,7 @@
     XSetBackground(W_Display, borderGC, colortable[DARK_GREY].pixelValue);
     XSetForeground(W_Display, borderGC, colortable[LIGHT_GREY].pixelValue);
 
-    XSetBackground(W_Display, maskGC, 0xff);
+    XSetBackground(W_Display, maskGC, 0xffffffff);
     XSetForeground(W_Display, maskGC, 0);
 
     backColor = W_Black;
diff -aur xfirepower-0.84.orig/client/libsprite/libsdata.h xfirepower-0.84/client/libsprite/libsdata.h
--- xfirepower-0.84.orig/client/libsprite/libsdata.h	1995-05-27 08:12:40.000000000 +0600
+++ xfirepower-0.84/client/libsprite/libsdata.h	2012-05-15 16:57:04.726318425 +0600
@@ -20,6 +20,7 @@
 
 #include "Wlib.h"
 #include "libsdefs.h"
+#include "libsstruct.h"
 
 extern int	nplanes;
 extern int	xpmORplanes ;
diff -aur xfirepower-0.84.orig/client/libsprite/libsdefs.h xfirepower-0.84/client/libsprite/libsdefs.h
--- xfirepower-0.84.orig/client/libsprite/libsdefs.h	1995-05-27 08:05:59.000000000 +0600
+++ xfirepower-0.84/client/libsprite/libsdefs.h	2012-05-15 16:57:19.802985178 +0600
@@ -59,7 +59,7 @@
 #define BOLD_FONT2	"-*-clean-bold-r-normal--10-100-75-75-c-60-*"
 #define ITALIC_FONT	"6x10"
 #define ITALIC_FONT2	"-*-clean-bold-r-normal--10-100-75-75-c-60-*"
-#define BIG_FONT	"-adobe-helvetica-bold-r-normal--34-*-*-*-*-*-*-*"
+#define BIG_FONT	"-*-*-bold-r-normal--34-*-*-*-*-*-*-*"
 #define IND_FONT        "-*-clean-bold-r-normal--10-100-75-75-c-60-*"
 
 #define FONTS 4
