--- xfirepower-0.84.orig/server/log.c	1995-11-22 13:36:19.000000000 +0500
+++ xfirepower-0.84/server/log.c	2012-05-15 20:12:46.887303744 +0600
@@ -2,6 +2,8 @@
 /* Log handling for XFirepower server */
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <time.h>
 #include "data.h"
 #include "struct.h"
@@ -20,10 +22,13 @@
     time_t t = time(0);
     struct tm *tmv = localtime(&t);
     char tbuf[40];
+    char *logfile;
 
     /* open the file if it isn't already */
     if (!ml) {
-        ml = fopen("logs/message.log", "a");
+        logfile = getenv("HOME");
+        strcat(logfile, "/.firepower.log");
+        ml = fopen(logfile, "a");
         if (!ml) {
             fprintf(stderr, "Can't open log file!\n");
             return;
--- xfirepower-0.84.orig/server/motd.c	2012-01-20 14:20:37.000000000 +0600
+++ xfirepower-0.84/server/motd.c	2012-01-20 14:21:07.586649794 +0600
@@ -13,7 +13,7 @@
 
 /* motd_read_motd()
 
-   Read the motd from "./.motd"
+   Read the motd from "/usr/share/xfirepower/motd"
    Save it memory, send it to each player when they log on
 
    Return: none
@@ -26,7 +26,7 @@
 
     GLO_motdLines = 0;
 
-    if ((mf = fopen("./.motd", "r")) == 0) {
+    if ((mf = fopen("/usr/share/xfirepower/motd", "r")) == 0) {
         printf("No motd file, I'm bummed :-(\n");
         return;
     }
@@ -42,7 +42,7 @@
     GLO_motd = malloc(GLO_motdLines * sizeof(char *));
 
     /* read into motd array */
-    mf = fopen("./.motd", "r");
+    mf = fopen("/usr/share/xfirepower/motd", "r");
     for (i = 0; i < GLO_motdLines; i++) {
         GLO_motd[i] = malloc(100);
         fgets(GLO_motd[i], 100, mf);
