--- labrea-2.5-stable-1.orig//autogen.sh	1970-01-01 01:00:00.000000000 +0100
+++ labrea-2.5-stable-1//autogen.sh	2010-03-24 17:16:22.309843093 +0100
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+autoreconf -vfi
--- labrea-2.5-stable-1.orig//src/Makefile.am	2003-01-09 19:13:19.000000000 +0100
+++ labrea-2.5-stable-1//src/Makefile.am	2010-03-24 17:23:38.715172653 +0100
@@ -2,7 +2,7 @@
 ##
 ## $Id: Makefile.am,v 1.1.1.1 2003/01/09 18:13:19 lorgor Exp $
 
-AM_CPPFLAGS = -I../inc -I- @DNETINC@ @PCAPINC@
+AM_CPPFLAGS = -iquote../inc @DNETINC@ @PCAPINC@
 
 sbin_PROGRAMS = labrea
 
