--- Makefile.old	2006-12-11 14:58:52.000000000 +0100
+++ Makefile	2013-03-28 17:43:27.113400386 +0100
@@ -6,11 +6,11 @@
 
 CXX = g++
 INCLUDES = -I./
-COPTS  = -Wall -O3
+COPTS  = -Wall -O2
 CDEBUG = -Wall -g 
 CXXFLAGS = $(COPTS) $(INCLUDES) -D$(ENDIANESS)
 
-PREFIX = /usr/local
+PREFIX = /usr
 INSTALL = /usr/bin/install -c
 
 #Name definitions
@@ -71,14 +71,15 @@
 
 #Installs the library on a system global location
 install: lib
-	@$(INSTALL) -m 755 $(LIBFILENAME) $(PREFIX)/lib/$(LIBFILENAME)
-	@ln -s -f $(PREFIX)/lib/$(LIBFILENAME) $(PREFIX)/lib/$(LIBSONAME) 
-	@mkdir  -p $(PREFIX)/include/oscpack/ip $(PREFIX)/include/oscpack/osc
-	@$(INSTALL) -m 644 ip/*.h $(PREFIX)/include/oscpack/ip
-	@$(INSTALL) -m 644 osc/*.h $(PREFIX)/include/oscpack/osc
-	@echo "SUCCESS! oscpack has been installed in $(PREFIX)/lib and $(PREFIX)/include/ospack/"
-	@echo "now doing ldconfig..."
-	@ldconfig
+	@$(INSTALL) -d -m 755 $(DESTDIR)/$(PREFIX)/lib
+	@$(INSTALL) -d -m 755 $(DESTDIR)/$(PREFIX)/include/oscpack/osc
+	@$(INSTALL) -d -m 755 $(DESTDIR)/$(PREFIX)/include/oscpack/ip
+	
+	@$(INSTALL) -m 755 $(LIBFILENAME) $(DESTDIR)/$(PREFIX)/lib/$(LIBFILENAME)
+	@ln -s -f $(LIBFILENAME) $(DESTDIR)/$(PREFIX)/lib/$(LIBSONAME) 
+#	@mkdir  -p $(PREFIX)/include/oscpack/ip $(DESTDIR)/$(PREFIX)/include/oscpack/osc
+	@$(INSTALL) -m 644 ip/*.h $(DESTDIR)/$(PREFIX)/include/oscpack/ip
+	@$(INSTALL) -m 644 osc/*.h $(DESTDIR)/$(PREFIX)/include/oscpack/osc
 
 #Installs the include/lib structure locally
 install-local: lib
