diff -Naur oscpack-org/examples/OscDump.cpp oscpack/examples/OscDump.cpp
--- oscpack-org/examples/OscDump.cpp	2006-11-20 14:49:40.000000000 +0100
+++ oscpack/examples/OscDump.cpp	2008-04-23 19:45:38.000000000 +0200
@@ -36,6 +36,8 @@
 
 
 #include <iostream>
+#include <string.h>
+#include <stdlib.h>
 
 #include "osc/OscReceivedElements.h"
 #include "osc/OscPrintReceivedElements.h"
diff -Naur oscpack-org/osc/OscPrintReceivedElements.cpp oscpack/osc/OscPrintReceivedElements.cpp
--- oscpack-org/osc/OscPrintReceivedElements.cpp	2006-11-20 14:49:38.000000000 +0100
+++ oscpack/osc/OscPrintReceivedElements.cpp	2008-04-23 19:41:06.000000000 +0200
@@ -29,6 +29,8 @@
 */
 #include "OscPrintReceivedElements.h"
 
+#include <string.h>
+
 #include <iostream>
 #include <iomanip>
 #include <ctime>
diff -Naur oscpack-org/tests/OscReceiveTest.cpp oscpack/tests/OscReceiveTest.cpp
--- oscpack-org/tests/OscReceiveTest.cpp	2006-11-20 14:49:38.000000000 +0100
+++ oscpack/tests/OscReceiveTest.cpp	2008-04-23 19:41:40.000000000 +0200
@@ -30,6 +30,7 @@
 #include "OscReceiveTest.h"
 
 #include <string.h>
+#include <stdlib.h>
 #include <iostream>
 
 #include "osc/OscReceivedElements.h"
diff -Naur oscpack-org/tests/OscSendTests.cpp oscpack/tests/OscSendTests.cpp
--- oscpack-org/tests/OscSendTests.cpp	2006-11-20 14:49:38.000000000 +0100
+++ oscpack/tests/OscSendTests.cpp	2008-04-23 19:39:27.000000000 +0200
@@ -30,6 +30,7 @@
 #include "OscSendTests.h"
 
 #include <iostream>
+#include <stdlib.h>
 #include <string.h>
 
 #include "osc/OscOutboundPacketStream.h"
diff -Naur oscpack-org/tests/OscUnitTests.cpp oscpack/tests/OscUnitTests.cpp
--- oscpack-org/tests/OscUnitTests.cpp	2006-11-20 14:49:38.000000000 +0100
+++ oscpack/tests/OscUnitTests.cpp	2008-04-23 19:37:33.000000000 +0200
@@ -29,6 +29,8 @@
 */
 #include "OscUnitTests.h"
 
+#include <string.h>
+
 #include <iostream>
 #include <iomanip>
 
