From 4fa9ba1d0c84f54c9cde2ac036d388f94c8696ec Mon Sep 17 00:00:00 2001
From: Jon Nordby <jononor@gmail.com>
Date: Tue, 21 Dec 2010 00:39:21 +0100
Subject: [PATCH] Disable building of tests (hacky)

---
 m-keyboard.pro |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/m-keyboard.pro b/m-keyboard.pro
index f1fed99..fd73fb8 100644
--- a/m-keyboard.pro
+++ b/m-keyboard.pro
@@ -2,7 +2,6 @@ CONFIG += ordered
 TARGET = meego-im-plugins
 TEMPLATE = subdirs
 SUBDIRS = m-keyboard \
-      tests \
       fixture_virtualkeyboard \
       translations \
 
-- 
1.7.3.4

