diff --git a/69-yubikey.rules b/69-yubikey.rules
index d12ce23..d6da287 100644
--- a/69-yubikey.rules
+++ b/69-yubikey.rules
@@ -4,7 +4,7 @@ ACTION!="add|change", GOTO="yubico_end"
 # device node, needed for challenge/response to work correctly.
 
 # Yubico Yubikey II
-ATTRS{idVendor}=="1050", ATTRS{idProduct}=="0010|0110|0111|114|116", \
+ATTRS{idVendor}=="1050", ATTRS{idProduct}=="0010|0110|0111|0114|0116|0120", \
     ENV{ID_SECURITY_TOKEN}="1"
 
 LABEL="yubico_end"
