From 2789530e055f98b1f59569b90483e75aa390559d Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Fri, 13 Dec 2013 10:50:27 +0100
Subject: [PATCH 1/1] work around timing issues

---
 ykcore/ykcore_libusb-1.0.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ykcore/ykcore_libusb-1.0.c b/ykcore/ykcore_libusb-1.0.c
index a3541c1..7c8a847 100644
--- a/ykcore/ykcore_libusb-1.0.c
+++ b/ykcore/ykcore_libusb-1.0.c
@@ -83,6 +83,11 @@ int _ykusb_write(void *dev, int report_type, int report_number,
 		if (ykl_errno > 0 && rc2 < 0)
 			ykl_errno = rc2;
 	}
+
+	/* For any reason Yubikey is not ready yet in some rare cases,
+	 * resulting in a timeouts. Waiting for a moment works around this. */
+	usleep(1);
+
 	if (ykl_errno > 0)
 		return 1;
 	yk_errno = YK_EUSBERR;
-- 
1.8.5.1

