--- CMakeLists.txt.orig	2008-08-27 17:02:17.588535328 +0200
+++ CMakeLists.txt	2008-08-27 17:04:10.541899478 +0200
@@ -1,5 +1,15 @@
 project(plasma-news)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${PLASMA_INCLUDE_DIR})
+
+find_package(KDE4 REQUIRED)
+include(KDE4Defaults)
+find_package(Plasma REQUIRED)
+
+add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}
+		    ${PLASMA_INCLUDE_DIR}
+		    ${KDE4_INCLUDES}
+)
 
 set(news_SRCS
     news.cpp
@@ -16,4 +26,4 @@
 install(FILES
 up.svg
 down.svg
-DESTINATION ${DATA_INSTALL_DIR}/desktoptheme/default/news/)
\ Pas de fin de ligne à la fin du fichier.
+DESTINATION ${DATA_INSTALL_DIR}/desktoptheme/default/news/)
