--- news.cpp.orig	2008-08-27 17:06:34.659225689 +0200
+++ news.cpp	2008-08-27 17:13:46.378457262 +0200
@@ -81,7 +81,7 @@
     m_headerColor = Plasma::Theme::defaultTheme()->color(Plasma::Theme::HighlightColor);
     m_textColor = Plasma::Theme::defaultTheme()->color(Plasma::Theme::TextColor);
     m_backgroundColor = Plasma::Theme::defaultTheme()->color(Plasma::Theme::BackgroundColor);
-    m_altBackgroundColor = Plasma::Theme::defaultTheme()->color(Plasma::Theme::ButtonBackgroundColor);
+    m_altBackgroundColor = Plasma::Theme::defaultTheme()->color(Plasma::Theme::BackgroundColor);
 
 
     m_feeds = cg.readEntry("feeds",
