--- elflibviewer-0.9/CMakeLists.txt	2009-10-11 04:25:45.000000000 +0200
+++ elflibviewer-0.9/CMakeLists.txt.new	2009-10-13 11:31:32.000000000 +0200
@@ -22,5 +22,5 @@
 install(TARGETS elflibviewer DESTINATION bin)
 
 # Add icon and desktop files
-install(CODE "EXECUTE_PROCESS(COMMAND xdg-icon-resource install --novendor --size 48 ${CMAKE_SOURCE_DIR}/elflibviewer.png)")
-install(CODE "EXECUTE_PROCESS(COMMAND xdg-desktop-menu install --novendor ${CMAKE_SOURCE_DIR}/elflibviewer.desktop)")
+install(FILES ${CMAKE_SOURCE_DIR}/elflibviewer.png DESTINATION share/icons/hicolor/48x48/apps/)
+install(FILES ${CMAKE_SOURCE_DIR}/elflibviewer.desktop DESTINATION share/applications/)
