diff -Naur hxplay-1.0.8.orig/audio/fixptutil/pub/math64.h hxplay-1.0.8/audio/fixptutil/pub/math64.h
--- hxplay-1.0.8.orig/audio/fixptutil/pub/math64.h	2004-09-17 14:22:38.000000000 -0400
+++ hxplay-1.0.8/audio/fixptutil/pub/math64.h	2007-06-21 00:43:48.000000000 -0400
@@ -124,7 +124,7 @@
 // GCC / i386
 ///////////////////////////////////////////////////////////////////////////////////////
 
-#if defined(__GNUC__) && defined(__i386__) && !defined(_NO_GNU_AS)
+#if defined(__GNUC__) && defined(__i386__) || defined(__amd64__) && !defined(_NO_GNU_AS)
 
 #define HAVE_PLATFORM_MACROS
 
diff -Naur hxplay-1.0.8.orig/build/umakecf/linux-2.6-glibc23-amd64.cf hxplay-1.0.8/build/umakecf/linux-2.6-glibc23-amd64.cf
--- hxplay-1.0.8.orig/build/umakecf/linux-2.6-glibc23-amd64.cf	2007-06-21 00:44:10.000000000 -0400
+++ hxplay-1.0.8/build/umakecf/linux-2.6-glibc23-amd64.cf	2007-06-21 00:43:48.000000000 -0400
@@ -61,8 +61,6 @@
 # ***** END LICENSE BLOCK *****
 # 
 """Linux 2.6 using GCC/G++ 3.3 and glibc-2.3 on AMD64 (x86_64)."""
-
-
 exec_config_file('linux-2.6-glibc23.cf')
 
 ## list of PCF files to run
@@ -75,8 +73,7 @@
 platform.cc.args["default"] = platform.cc.args["default"] + \
         " -m64 -D__amd64__"
 
-project.RemoveSystemPaths("-L/usr/lib/X11")
-project.AddSystemPaths("-L/usr/lib/X1164")
+project.AddSystemPaths("-L/usr/lib/X11")
 
 # platform defines
 project.AddDefines("_LITTLE_ENDIAN")
diff -Naur hxplay-1.0.8.orig/common/import/gecko-sdk/nspr/include/prcpucfg.h hxplay-1.0.8/common/import/gecko-sdk/nspr/include/prcpucfg.h
--- hxplay-1.0.8.orig/common/import/gecko-sdk/nspr/include/prcpucfg.h	2004-12-22 15:00:40.000000000 -0500
+++ hxplay-1.0.8/common/import/gecko-sdk/nspr/include/prcpucfg.h	2007-06-21 00:43:48.000000000 -0400
@@ -182,6 +182,52 @@
 #define PR_BYTES_PER_WORD_LOG2  3
 #define PR_BYTES_PER_DWORD_LOG2 3
 
+#elif defined(__amd64__)
+
+#define IS_LITTLE_ENDIAN 1
+#undef  IS_BIG_ENDIAN
+#define IS_64
+
+#define PR_BYTES_PER_BYTE   1
+#define PR_BYTES_PER_SHORT  2
+#define PR_BYTES_PER_INT    4
+#define PR_BYTES_PER_INT64  8
+#define PR_BYTES_PER_LONG   8
+#define PR_BYTES_PER_FLOAT  4
+#define PR_BYTES_PER_DOUBLE 8
+#define PR_BYTES_PER_WORD   8
+#define PR_BYTES_PER_DWORD  8
+
+#define PR_BITS_PER_BYTE    8
+#define PR_BITS_PER_SHORT   16
+#define PR_BITS_PER_INT     32
+#define PR_BITS_PER_INT64   64
+#define PR_BITS_PER_LONG    64
+#define PR_BITS_PER_FLOAT   32
+#define PR_BITS_PER_DOUBLE  64
+#define PR_BITS_PER_WORD    64
+
+#define PR_BITS_PER_BYTE_LOG2   3
+#define PR_BITS_PER_SHORT_LOG2  4
+#define PR_BITS_PER_INT_LOG2    5
+#define PR_BITS_PER_INT64_LOG2  6
+#define PR_BITS_PER_LONG_LOG2   6
+#define PR_BITS_PER_FLOAT_LOG2  5
+#define PR_BITS_PER_DOUBLE_LOG2 6
+#define PR_BITS_PER_WORD_LOG2   6
+
+#define PR_ALIGN_OF_SHORT   2
+#define PR_ALIGN_OF_INT     4
+#define PR_ALIGN_OF_LONG    8
+#define PR_ALIGN_OF_INT64   8
+#define PR_ALIGN_OF_FLOAT   4
+#define PR_ALIGN_OF_DOUBLE  8
+#define PR_ALIGN_OF_POINTER 8
+#define PR_ALIGN_OF_WORD    8
+
+#define PR_BYTES_PER_WORD_LOG2  3
+#define PR_BYTES_PER_DWORD_LOG2 3
+
 #elif defined(__mc68000__)
 
 #undef  IS_LITTLE_ENDIAN
diff -Naur hxplay-1.0.8.orig/datatype/image/jpg/import/jpeg-6b/jmorecfg.h hxplay-1.0.8/datatype/image/jpg/import/jpeg-6b/jmorecfg.h
--- hxplay-1.0.8.orig/datatype/image/jpg/import/jpeg-6b/jmorecfg.h	2003-06-18 12:17:45.000000000 -0400
+++ hxplay-1.0.8/datatype/image/jpg/import/jpeg-6b/jmorecfg.h	2007-06-21 00:44:43.000000000 -0400
@@ -161,7 +161,7 @@
 #if (defined _UNIX && (defined _ALPHA || defined _OSF1))
 typedef int INT32;
 #else
-typedef long INT32;
+typedef int INT32;
 #endif
 #endif
 
