diff --git a/libnotifymm-0.6.1/libnotify/libnotifymm/notification.cc b/libnotifymm-0.6.1/libnotify/libnotifymm/notification.cc
index acce672..dad7473 100644
--- a/libnotifymm-0.6.1/libnotify/libnotifymm/notification.cc
+++ b/libnotifymm-0.6.1/libnotify/libnotifymm/notification.cc
@@ -286,17 +286,19 @@ bool Notification::update(const Glib::ustring& summary, const Glib::ustring& bod
 
 void Notification::attach_to_widget(Gtk::Widget& attach)
 {
-notify_notification_attach_to_widget(gobj(), (attach).gobj()); 
+  (void)attach;
 }
 
 void Notification::attach_to_status_icon(const Glib::RefPtr<Gtk::StatusIcon>& status_icon)
 {
-notify_notification_attach_to_status_icon(gobj(), Glib::unwrap(status_icon)); 
+  (void)status_icon;
 }
 
 void Notification::set_geometry_hints(const Glib::RefPtr<Gdk::Screen>& screen, int x, int y)
 {
-notify_notification_set_geometry_hints(gobj(), Glib::unwrap(screen), x, y); 
+  (void)screen;
+  (void)x;
+  (void)y;
 }
 
 #ifdef GLIBMM_EXCEPTIONS_ENABLED
