--- setup.py.orig	2015-01-08 22:54:12.617906696 +0100
+++ setup.py	2015-01-08 23:08:39.611303117 +0100
@@ -19,6 +19,7 @@
 
 import os
 import sys
+import io
 
 try:
     import DistUtilsExtra.auto
@@ -32,14 +33,14 @@
     filename = os.path.join(libdir, 'uberwriter_lib/uberwriterconfig.py')
     oldvalues = {}
     try:
-        fin = open(filename, 'r', encoding="utf-8")
-        fout = open(filename + '.new', 'w', encoding="utf-8")
+        fin = io.open(filename, 'r', encoding="utf-8")
+        fout = io.open(filename + '.new', 'w', encoding="utf-8")
 
         for line in fin:
             fields = line.split(' = ') # Separate variable from value
             if fields[0] in values:
                 oldvalues[fields[0]] = fields[1].strip()
-                line = "%s = %s\n" % (fields[0], values[fields[0]])
+                line = u"%s = %s\n" % (fields[0], values[fields[0]])
             fout.write(line)
 
         fout.flush()
@@ -69,9 +70,6 @@
     if not os.path.exists(old_desktop_file):
         print ("ERROR: Can't find", old_desktop_file)
         sys.exit(1)
-    if os.path.exists(desktop_file):
-        print("REMOVING DESKTOP FILE")
-        os.remove(desktop_file)
     if target_data != prefix + '/':
         # This is an /opt install, so rename desktop file to use extras-
         desktop_file = desktop_path + '/extras-uberwriter.desktop'
@@ -89,17 +87,17 @@
 def update_desktop_file(filename, target_pkgdata, target_scripts):
     print("updateing deskop file: %s" % filename)
     try:
-        fin = open(filename, 'r', encoding="utf-8")
-        fout = open(filename + '.new', 'w', encoding="utf-8")
+        fin = io.open(filename, 'r', encoding="utf-8")
+        fout = io.open(filename + '.new', 'w', encoding="utf-8")
 
         for line in fin:
             if 'Icon=' in line:
-                line = "Icon=%s\n" % (target_pkgdata + 'media/uberwriter.svg')
+                line = u"Icon=%s\n" % (target_pkgdata + 'media/uberwriter.svg')
             elif 'Exec=' in line:
                 cmd = line.split("=")[1].split(None, 1)
-                line = "Exec=%s" % (target_scripts + 'uberwriter')
+                line = u"Exec=%s" % (target_scripts + 'uberwriter')
                 if len(cmd) > 1:
-                    line += " %s" % cmd[1].strip()  # Add script arguments back
+                    line += u" %s" % cmd[1].strip()  # Add script arguments back
                 line += "\n"
             fout.write(line)
         fout.flush()
