diff -Naur fluxbox-1.1.1~/util/fbrun/main.cc fluxbox-1.1.1/util/fbrun/main.cc
--- fluxbox-1.1.1~/util/fbrun/main.cc	2008-01-15 03:50:53.000000000 -0600
+++ fluxbox-1.1.1/util/fbrun/main.cc	2009-01-26 17:56:58.000000000 -0600
@@ -63,7 +63,7 @@
         "   -fg [color name]            Foreground text color"<<endl<<
         "   -bg [color name]            Background color"<<endl<<
         "   -na                         Disable antialias"<<endl<<
-        "   -hf [history file]          History file to load (default ~/.fluxbox/fbrun_history)"<<endl<<
+        "   -hf [history file]          History file to load (default ~/.fbrun_history)"<<endl<<
         "   -help                       Show this help"<<endl<<endl<<
         "Example: fbrun -fg black -bg white -text xterm -title \"run xterm\""<<endl;
 }
@@ -81,7 +81,7 @@
     string foreground("black");   // text color
     string background("white");   // text background color
     string display_name; // name of the display connection
-    string history_file("~/.fluxbox/fbrun_history"); // command history file
+    string history_file("~/.fbrun_history"); // command history file
     // parse arguments
     for (int i=1; i<argc; i++) {
         if (strcmp(argv[i], "-font") == 0 && i+1 < argc) {
