--- linux-2.6.42.noarch/drivers/staging/nvec/Kconfig.orig	2012-02-02 08:16:12.512727480 -0600
+++ linux-2.6.42.noarch/drivers/staging/nvec/Kconfig	2012-02-01 18:44:56.674990109 -0600
@@ -1,6 +1,6 @@
 config MFD_NVEC
 	bool "NV Tegra Embedded Controller SMBus Interface"
-	depends on I2C && GPIOLIB && ARCH_TEGRA
+	depends on I2C && GPIOLIB && ARCH_TEGRA && MFD_CORE=y
 	help
 	    Say Y here to enable support for a nVidia compliant embedded
 	    controller.
