diff -Naur xtide-2.13/Global.cc xtide-2.13-aur/Global.cc
--- xtide-2.13/Global.cc	2012-05-26 14:47:38.000000000 -0700
+++ xtide-2.13-aur/Global.cc	2013-05-28 23:13:30.757618495 -0700
@@ -213,7 +213,7 @@
 #ifdef UseLocalFiles
   static const char confFile[] = "xtide.conf";
 #else
-  static const char confFile[] = "/etc/xtide.conf";
+  static const char confFile[] = "/etc/xtide/xtide.conf";
 #endif
   static bool firstCall (true);
   static SafeVector<Dstr> xtideConfLines (2U);
@@ -400,10 +400,10 @@
     break;
   case Error::NO_HFILE_PATH:
     text_out += "NO_HFILE_PATH\n\
-If /etc/xtide.conf is not provided, you must set the environment variable\n\
+If /etc/xtide/xtide.conf is not provided, you must set the environment variable\n\
 HFILE_PATH to point to your harmonics files.  Example:\n\
   export HFILE_PATH=/usr/local/share/xtide/harmonics.tcd\n\
-Please refer to the documentation for usage of /etc/xtide.conf.";
+Please refer to the documentation for usage of /etc/xtide/xtide.conf.";
     break;
   case Error::NO_HFILE_IN_PATH:
     text_out += "NO_HFILE_IN_PATH\n\
