--- Makefile.linux	2012-05-23 10:46:14.000000000 +0200
+++ Makefile.linux.new	2012-07-09 13:09:03.387944487 +0200
@@ -21,14 +21,7 @@
 # $ IPTABLESPATH=/path/to/iptables-1.4.1 make -f Makefile.linux
 #
 #CFLAGS = -O -g -DDEBUG
-CFLAGS ?= -Os
-CFLAGS += -fno-strict-aliasing
-CFLAGS += -fno-common
-CFLAGS += -D_GNU_SOURCE
-CFLAGS += -Wall
-CFLAGS += -Wextra -Wstrict-prototypes -Wdeclaration-after-statement
-#CFLAGS += -Wno-missing-field-initializers
-CFLAGS += -ansi
+CFLAGS ?= -Wall -Os -D_GNU_SOURCE -fno-strict-aliasing -Wstrict-prototypes -Wdeclaration-after-statement -ansi
 CC ?= gcc
 RM = rm -f
 INSTALL = install
