--- Makefile.media_orig	2007-04-24 18:57:50.000000000 -0400
+++ Makefile.media	2007-04-24 18:58:17.000000000 -0400
@@ -561,7 +561,7 @@
 	@install -d $(DESTDIR)$(KDIR26)/video/zc0301
 	@for i in zc0301.ko;do if [ -e "$$i" ]; then echo -n "$$i "; install -m 644 -c $$i $(DESTDIR)$(KDIR26)/video/zc0301; fi; done; echo;
 
-	/sbin/depmod -a $(KERNELRELEASE) $(if $(DESTDIR),-b $(DESTDIR))
+	#/sbin/depmod -a $(KERNELRELEASE) $(if $(DESTDIR),-b $(DESTDIR))
 
 media-rminstall::
 	@echo -e "\nRemoving old $(DEST) files\n"
