diff -r d402a183765a kiklib/src/kik_pty_streams.c
--- a/kiklib/src/kik_pty_streams.c	Sun Apr 22 03:32:15 2012 +0900
+++ b/kiklib/src/kik_pty_streams.c	Sat Apr 28 16:18:52 2012 +0900
@@ -2,10 +2,12 @@
  *	$Id$
  */
 
-/* _GNU_SOURCE must be defined before including <features.h> to take effect.
- * since standard headers/kik_types.h #includes features.h indirectly,
- * ecplicitly evaluate only the autoconf's result here */
-#include "kik_def.h"
+/*
+ * _GNU_SOURCE must be defined before including <features.h> to take effect.
+ * since standard headers, kik_types.h and kik_def.h include features.h indirectly,
+ * ecplicitly evaluate only the autoconf's result here.
+ */
+#include  "kik_config.h"
 #ifdef HAVE_GNU_SOURCE
 #define _GNU_SOURCE
 #endif
