diff -rupN cnijfilter-common-2.70-old/cngpijmon/backend/cnij_backend_common.c cnijfilter-common-2.70/cngpijmon/backend/cnij_backend_common.c
--- cnijfilter-common-2.70-old/cngpijmon/backend/cnij_backend_common.c	2011-09-25 09:17:00.882146933 -0400
+++ cnijfilter-common-2.70/cngpijmon/backend/cnij_backend_common.c	2011-09-25 09:30:33.424326112 -0400
@@ -33,6 +33,7 @@
 
 // CUPS Header
 #include <cups/cups.h>
+#include <cups/ppd.h>
 #include <cups/ipp.h>
 
 // Header file for CANON
diff -rupN cnijfilter-common-2.70-old/cngpijmon/src/bjcupsmon_cups.c cnijfilter-common-2.70/cngpijmon/src/bjcupsmon_cups.c
--- cnijfilter-common-2.70-old/cngpijmon/src/bjcupsmon_cups.c	2011-09-25 09:17:00.883146973 -0400
+++ cnijfilter-common-2.70/cngpijmon/src/bjcupsmon_cups.c	2011-09-25 09:18:30.218656719 -0400
@@ -20,6 +20,7 @@
 
 /*** Includes ***/
 #include <cups/cups.h>
+#include <cups/ppd.h>
 #include <cups/language.h>
 #include <sys/types.h>
 #include <unistd.h>
