diff --git a/src/eventcomm.c b/src/eventcomm.c
index 7a56cb7..76509a8 100644
--- a/src/eventcomm.c
+++ b/src/eventcomm.c
@@ -527,6 +527,7 @@ SynapticsReadEvent(InputInfoPtr pInfo, struct input_event *ev)
     return rc;
 }
 
+#ifdef HAVE_MTDEV
 static Bool
 EventTouchSlotPreviouslyOpen(SynapticsPrivate *priv, int slot)
 {
@@ -537,6 +538,7 @@ EventTouchSlotPreviouslyOpen(SynapticsPrivate *priv, int slot)
 
     return FALSE;
 }
+#endif
 
 static void
 EventProcessTouchEvent(InputInfoPtr pInfo, struct SynapticsHwState *hw,
