diff -Naupr powermanga-0.90.orig/data/curves/Makefile powermanga-0.90/data/curves/Makefile
--- powermanga-0.90.orig/data/curves/Makefile	2007-09-04 10:41:12.000000000 +0200
+++ powermanga-0.90/data/curves/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/data/curves
+pkgdatadir = $(datadir)/powermanga/data/curves
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/data/curves/Makefile.am powermanga-0.90/data/curves/Makefile.am
--- powermanga-0.90.orig/data/curves/Makefile.am	2007-07-13 07:31:49.000000000 +0200
+++ powermanga-0.90/data/curves/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/data/curves
+pkgdatadir = $(datadir)/powermanga/data/curves
 
 dist_pkgdata_DATA = \
   bezier_curve_000.bin \
diff -Naupr powermanga-0.90.orig/data/curves/Makefile.in powermanga-0.90/data/curves/Makefile.in
--- powermanga-0.90.orig/data/curves/Makefile.in	2007-09-04 10:41:02.000000000 +0200
+++ powermanga-0.90/data/curves/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/data/curves
+pkgdatadir = $(datadir)/powermanga/data/curves
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/data/levels/curves_phase/Makefile powermanga-0.90/data/levels/curves_phase/Makefile
--- powermanga-0.90.orig/data/levels/curves_phase/Makefile	2007-09-04 10:41:12.000000000 +0200
+++ powermanga-0.90/data/levels/curves_phase/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/data/levels/curves_phase
+pkgdatadir = $(datadir)/powermanga/data/levels/curves_phase
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/data/levels/curves_phase/Makefile.am powermanga-0.90/data/levels/curves_phase/Makefile.am
--- powermanga-0.90.orig/data/levels/curves_phase/Makefile.am	2007-07-13 07:31:50.000000000 +0200
+++ powermanga-0.90/data/levels/curves_phase/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/data/levels/curves_phase
+pkgdatadir = $(datadir)/powermanga/data/levels/curves_phase
 
 dist_pkgdata_DATA = \
   curves_00.bin \
diff -Naupr powermanga-0.90.orig/data/levels/curves_phase/Makefile.in powermanga-0.90/data/levels/curves_phase/Makefile.in
--- powermanga-0.90.orig/data/levels/curves_phase/Makefile.in	2007-09-04 10:41:02.000000000 +0200
+++ powermanga-0.90/data/levels/curves_phase/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/data/levels/curves_phase
+pkgdatadir = $(datadir)/powermanga/data/levels/curves_phase
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/data/levels/grids_phase/Makefile powermanga-0.90/data/levels/grids_phase/Makefile
--- powermanga-0.90.orig/data/levels/grids_phase/Makefile	2007-09-04 10:41:12.000000000 +0200
+++ powermanga-0.90/data/levels/grids_phase/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/data/levels/grids_phase
+pkgdatadir = $(datadir)/powermanga/data/levels/grids_phase
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/data/levels/grids_phase/Makefile.am powermanga-0.90/data/levels/grids_phase/Makefile.am
--- powermanga-0.90.orig/data/levels/grids_phase/Makefile.am	2007-07-13 08:22:53.000000000 +0200
+++ powermanga-0.90/data/levels/grids_phase/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/data/levels/grids_phase
+pkgdatadir = $(datadir)/powermanga/data/levels/grids_phase
 
 dist_pkgdata_DATA = \
  grid_00.bin \
diff -Naupr powermanga-0.90.orig/data/levels/grids_phase/Makefile.in powermanga-0.90/data/levels/grids_phase/Makefile.in
--- powermanga-0.90.orig/data/levels/grids_phase/Makefile.in	2007-09-04 10:41:03.000000000 +0200
+++ powermanga-0.90/data/levels/grids_phase/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/data/levels/grids_phase
+pkgdatadir = $(datadir)/powermanga/data/levels/grids_phase
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/data/Makefile powermanga-0.90/data/Makefile
--- powermanga-0.90.orig/data/Makefile	2007-09-04 10:41:12.000000000 +0200
+++ powermanga-0.90/data/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/data
+pkgdatadir = $(datadir)/powermanga/data
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/data/Makefile.am powermanga-0.90/data/Makefile.am
--- powermanga-0.90.orig/data/Makefile.am	2007-07-13 09:07:28.000000000 +0200
+++ powermanga-0.90/data/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/data
+pkgdatadir = $(datadir)/powermanga/data
 
 dist_pkgdata_DATA = \
   menu_order_keystrokes.bin 
diff -Naupr powermanga-0.90.orig/data/Makefile.in powermanga-0.90/data/Makefile.in
--- powermanga-0.90.orig/data/Makefile.in	2007-09-04 10:41:02.000000000 +0200
+++ powermanga-0.90/data/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/data
+pkgdatadir = $(datadir)/powermanga/data
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/graphics/bitmap/fonts/Makefile powermanga-0.90/graphics/bitmap/fonts/Makefile
--- powermanga-0.90.orig/graphics/bitmap/fonts/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/graphics/bitmap/fonts/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/bitmap/fonts
+pkgdatadir = $(datadir)/powermanga/graphics/bitmap/fonts
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/graphics/bitmap/fonts/Makefile.am powermanga-0.90/graphics/bitmap/fonts/Makefile.am
--- powermanga-0.90.orig/graphics/bitmap/fonts/Makefile.am	2007-09-04 10:26:27.000000000 +0200
+++ powermanga-0.90/graphics/bitmap/fonts/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/graphics/bitmap/fonts
+pkgdatadir = $(datadir)/powermanga/graphics/bitmap/fonts
 
 dist_pkgdata_DATA = \
   font_big.spr \
diff -Naupr powermanga-0.90.orig/graphics/bitmap/fonts/Makefile.in powermanga-0.90/graphics/bitmap/fonts/Makefile.in
--- powermanga-0.90.orig/graphics/bitmap/fonts/Makefile.in	2007-09-04 10:41:03.000000000 +0200
+++ powermanga-0.90/graphics/bitmap/fonts/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/bitmap/fonts
+pkgdatadir = $(datadir)/powermanga/graphics/bitmap/fonts
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/graphics/bitmap/Makefile powermanga-0.90/graphics/bitmap/Makefile
--- powermanga-0.90.orig/graphics/bitmap/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/graphics/bitmap/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/bitmap
+pkgdatadir = $(datadir)/powermanga/graphics/bitmap
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/graphics/bitmap/Makefile.am powermanga-0.90/graphics/bitmap/Makefile.am
--- powermanga-0.90.orig/graphics/bitmap/Makefile.am	2007-07-14 09:19:18.000000000 +0200
+++ powermanga-0.90/graphics/bitmap/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/graphics/bitmap
+pkgdatadir = $(datadir)/powermanga/graphics/bitmap
 
 dist_pkgdata_DATA = \
   main_menu.spr \
diff -Naupr powermanga-0.90.orig/graphics/bitmap/Makefile.in powermanga-0.90/graphics/bitmap/Makefile.in
--- powermanga-0.90.orig/graphics/bitmap/Makefile.in	2007-09-04 10:41:03.000000000 +0200
+++ powermanga-0.90/graphics/bitmap/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/bitmap
+pkgdatadir = $(datadir)/powermanga/graphics/bitmap
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/graphics/Makefile powermanga-0.90/graphics/Makefile
--- powermanga-0.90.orig/graphics/Makefile	2007-09-04 10:41:12.000000000 +0200
+++ powermanga-0.90/graphics/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics
+pkgdatadir = $(datadir)/powermanga/graphics
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/graphics/Makefile.am powermanga-0.90/graphics/Makefile.am
--- powermanga-0.90.orig/graphics/Makefile.am	2007-08-02 18:19:51.000000000 +0200
+++ powermanga-0.90/graphics/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/graphics
+pkgdatadir = $(datadir)/powermanga/graphics
 
 dist_pkgdata_DATA = \
   256_colors_palette.bin \
diff -Naupr powermanga-0.90.orig/graphics/Makefile.in powermanga-0.90/graphics/Makefile.in
--- powermanga-0.90.orig/graphics/Makefile.in	2007-09-04 10:41:03.000000000 +0200
+++ powermanga-0.90/graphics/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics
+pkgdatadir = $(datadir)/powermanga/graphics
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/graphics/sprites/guardians/Makefile powermanga-0.90/graphics/sprites/guardians/Makefile
--- powermanga-0.90.orig/graphics/sprites/guardians/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/graphics/sprites/guardians/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/guardians
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/guardians
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/graphics/sprites/guardians/Makefile.am powermanga-0.90/graphics/sprites/guardians/Makefile.am
--- powermanga-0.90.orig/graphics/sprites/guardians/Makefile.am	2007-07-13 21:25:51.000000000 +0200
+++ powermanga-0.90/graphics/sprites/guardians/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/guardians
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/guardians
 
 dist_pkgdata_DATA = \
   guardian_00.spr \
diff -Naupr powermanga-0.90.orig/graphics/sprites/guardians/Makefile.in powermanga-0.90/graphics/sprites/guardians/Makefile.in
--- powermanga-0.90.orig/graphics/sprites/guardians/Makefile.in	2007-09-04 10:41:03.000000000 +0200
+++ powermanga-0.90/graphics/sprites/guardians/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/guardians
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/guardians
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/graphics/sprites/Makefile powermanga-0.90/graphics/sprites/Makefile
--- powermanga-0.90.orig/graphics/sprites/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/graphics/sprites/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites
+pkgdatadir = $(datadir)/powermanga/graphics/sprites
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/graphics/sprites/Makefile.am powermanga-0.90/graphics/sprites/Makefile.am
--- powermanga-0.90.orig/graphics/sprites/Makefile.am	2007-07-13 21:46:20.000000000 +0200
+++ powermanga-0.90/graphics/sprites/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites
+pkgdatadir = $(datadir)/powermanga/graphics/sprites
 
 dist_pkgdata_DATA = \
   all_enemies.spr \
diff -Naupr powermanga-0.90.orig/graphics/sprites/Makefile.in powermanga-0.90/graphics/sprites/Makefile.in
--- powermanga-0.90.orig/graphics/sprites/Makefile.in	2007-09-04 10:41:03.000000000 +0200
+++ powermanga-0.90/graphics/sprites/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites
+pkgdatadir = $(datadir)/powermanga/graphics/sprites
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/graphics/sprites/meteors/Makefile powermanga-0.90/graphics/sprites/meteors/Makefile
--- powermanga-0.90.orig/graphics/sprites/meteors/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/graphics/sprites/meteors/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/meteors
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/meteors
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/graphics/sprites/meteors/Makefile.am powermanga-0.90/graphics/sprites/meteors/Makefile.am
--- powermanga-0.90.orig/graphics/sprites/meteors/Makefile.am	2007-07-12 21:30:55.000000000 +0200
+++ powermanga-0.90/graphics/sprites/meteors/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/meteors
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/meteors
 
 dist_pkgdata_DATA = \
   meteor_00.spr \
diff -Naupr powermanga-0.90.orig/graphics/sprites/meteors/Makefile.in powermanga-0.90/graphics/sprites/meteors/Makefile.in
--- powermanga-0.90.orig/graphics/sprites/meteors/Makefile.in	2007-09-04 10:41:03.000000000 +0200
+++ powermanga-0.90/graphics/sprites/meteors/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/meteors
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/meteors
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/graphics/sprites/spaceships/Makefile powermanga-0.90/graphics/sprites/spaceships/Makefile
--- powermanga-0.90.orig/graphics/sprites/spaceships/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/graphics/sprites/spaceships/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/spaceships
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/spaceships
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/graphics/sprites/spaceships/Makefile.am powermanga-0.90/graphics/sprites/spaceships/Makefile.am
--- powermanga-0.90.orig/graphics/sprites/spaceships/Makefile.am	2007-07-13 13:06:05.000000000 +0200
+++ powermanga-0.90/graphics/sprites/spaceships/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/spaceships
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/spaceships
 
 dist_pkgdata_DATA = \
   spaceship_1_frame_0.spr \
diff -Naupr powermanga-0.90.orig/graphics/sprites/spaceships/Makefile.in powermanga-0.90/graphics/sprites/spaceships/Makefile.in
--- powermanga-0.90.orig/graphics/sprites/spaceships/Makefile.in	2007-09-04 10:41:03.000000000 +0200
+++ powermanga-0.90/graphics/sprites/spaceships/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/spaceships
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/spaceships
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/graphics/sprites/stars/Makefile powermanga-0.90/graphics/sprites/stars/Makefile
--- powermanga-0.90.orig/graphics/sprites/stars/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/graphics/sprites/stars/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/stars
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/stars
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/graphics/sprites/stars/Makefile.am powermanga-0.90/graphics/sprites/stars/Makefile.am
--- powermanga-0.90.orig/graphics/sprites/stars/Makefile.am	2007-07-13 20:22:31.000000000 +0200
+++ powermanga-0.90/graphics/sprites/stars/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/stars
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/stars
 
 dist_pkgdata_DATA = \
   star_big_0.spr \
diff -Naupr powermanga-0.90.orig/graphics/sprites/stars/Makefile.in powermanga-0.90/graphics/sprites/stars/Makefile.in
--- powermanga-0.90.orig/graphics/sprites/stars/Makefile.in	2007-09-04 10:41:04.000000000 +0200
+++ powermanga-0.90/graphics/sprites/stars/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/graphics/sprites/stars
+pkgdatadir = $(datadir)/powermanga/graphics/sprites/stars
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/Makefile powermanga-0.90/Makefile
--- powermanga-0.90.orig/Makefile	2007-09-04 10:41:12.000000000 +0200
+++ powermanga-0.90/Makefile	2007-10-22 13:24:16.000000000 +0200
@@ -82,7 +82,7 @@ DIST_ARCHIVES = $(distdir).tar.gz
 GZIP_ENV = --best
 distuninstallcheck_listfiles = find . -type f -print
 distcleancheck_listfiles = find . -type f -print
-pkgdatadir = $(mandir)
+pkgdatadir = $(mandir)/man6
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/Makefile.am powermanga-0.90/Makefile.am
--- powermanga-0.90.orig/Makefile.am	2007-08-13 08:43:21.000000000 +0200
+++ powermanga-0.90/Makefile.am	2007-10-22 13:24:23.000000000 +0200
@@ -2,6 +2,6 @@
 SUBDIRS = texts src graphics graphics/bitmap graphics/bitmap/fonts graphics/sprites graphics/sprites/guardians graphics/sprites/stars graphics/sprites/meteors graphics/sprites/spaceships data data/curves data/levels/curves_phase data/levels/grids_phase sounds sounds/handheld_console
 EXTRA_DIST = bootstrap
 
-pkgdatadir = $(mandir)
+pkgdatadir = $(mandir)/man6
 dist_pkgdata_DATA = powermanga.6
 
diff -Naupr powermanga-0.90.orig/Makefile.in powermanga-0.90/Makefile.in
--- powermanga-0.90.orig/Makefile.in	2007-09-04 10:41:05.000000000 +0200
+++ powermanga-0.90/Makefile.in	2007-10-22 13:24:29.000000000 +0200
@@ -82,7 +82,7 @@ DIST_ARCHIVES = $(distdir).tar.gz
 GZIP_ENV = --best
 distuninstallcheck_listfiles = find . -type f -print
 distcleancheck_listfiles = find . -type f -print
-pkgdatadir = $(mandir)
+pkgdatadir = $(mandir)/man6
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/sounds/handheld_console/Makefile powermanga-0.90/sounds/handheld_console/Makefile
--- powermanga-0.90.orig/sounds/handheld_console/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/sounds/handheld_console/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/sound/handheld_console
+pkgdatadir = $(datadir)/powermanga/sound/handheld_console
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/sounds/handheld_console/Makefile.am powermanga-0.90/sounds/handheld_console/Makefile.am
--- powermanga-0.90.orig/sounds/handheld_console/Makefile.am	2007-08-13 08:43:23.000000000 +0200
+++ powermanga-0.90/sounds/handheld_console/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/sound/handheld_console
+pkgdatadir = $(datadir)/powermanga/sound/handheld_console
 dist_pkgdata_DATA = \
   music_congratulations.ogg \
   music_game.ogg \
diff -Naupr powermanga-0.90.orig/sounds/handheld_console/Makefile.in powermanga-0.90/sounds/handheld_console/Makefile.in
--- powermanga-0.90.orig/sounds/handheld_console/Makefile.in	2007-09-04 10:41:04.000000000 +0200
+++ powermanga-0.90/sounds/handheld_console/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/sound/handheld_console
+pkgdatadir = $(datadir)/powermanga/sound/handheld_console
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/sounds/Makefile powermanga-0.90/sounds/Makefile
--- powermanga-0.90.orig/sounds/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/sounds/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/sounds
+pkgdatadir = $(datadir)/powermanga/sounds
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/sounds/Makefile.am powermanga-0.90/sounds/Makefile.am
--- powermanga-0.90.orig/sounds/Makefile.am	2007-08-13 08:43:22.000000000 +0200
+++ powermanga-0.90/sounds/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/sounds
+pkgdatadir = $(datadir)/powermanga/sounds
 
 dist_pkgdata_DATA = \
   music_congratulations.zik \
diff -Naupr powermanga-0.90.orig/sounds/Makefile.in powermanga-0.90/sounds/Makefile.in
--- powermanga-0.90.orig/sounds/Makefile.in	2007-09-04 10:41:04.000000000 +0200
+++ powermanga-0.90/sounds/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/sounds
+pkgdatadir = $(datadir)/powermanga/sounds
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
diff -Naupr powermanga-0.90.orig/src/Makefile powermanga-0.90/src/Makefile
--- powermanga-0.90.orig/src/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/src/Makefile	2007-10-22 13:08:59.000000000 +0200
@@ -220,7 +220,7 @@ target_alias = 
 target_cpu = i686
 target_os = linux-gnu
 target_vendor = pc
-gamesdir = $(prefix)/games
+gamesdir = $(bindir)
 scoredir = /var/games
 score = powermanga.hi
 powermanga_SOURCES = $(SOURCES_MAIN) $(SOURCES_EXTRA)
diff -Naupr powermanga-0.90.orig/src/Makefile.am powermanga-0.90/src/Makefile.am
--- powermanga-0.90.orig/src/Makefile.am	2007-08-31 22:46:43.000000000 +0200
+++ powermanga-0.90/src/Makefile.am	2007-10-22 13:09:13.000000000 +0200
@@ -1,5 +1,5 @@
 
-gamesdir = $(prefix)/games
+gamesdir = $(bindir)
 scoredir = /var/games
 score = powermanga.hi
 
diff -Naupr powermanga-0.90.orig/src/Makefile.in powermanga-0.90/src/Makefile.in
--- powermanga-0.90.orig/src/Makefile.in	2007-09-04 10:41:04.000000000 +0200
+++ powermanga-0.90/src/Makefile.in	2007-10-22 13:09:37.000000000 +0200
@@ -220,7 +220,7 @@ target_alias = @target_alias@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-gamesdir = $(prefix)/games
+gamesdir = $(bindir)
 scoredir = /var/games
 score = powermanga.hi
 powermanga_SOURCES = $(SOURCES_MAIN) $(SOURCES_EXTRA)
diff -Naupr powermanga-0.90.orig/src/tools.c powermanga-0.90/src/tools.c
--- powermanga-0.90.orig/src/tools.c	2007-08-31 22:46:43.000000000 +0200
+++ powermanga-0.90/src/tools.c	2007-10-22 13:09:59.000000000 +0200
@@ -474,7 +474,7 @@ static const char
 {
   /* normally unused, except when running from the source directory... */
   ".",
-  /* special value meaning "$(PREFIX)/share/games/powermanga/" */
+  /* special value meaning "$(PREFIX)/share/powermanga/" */
   /* also marks end of list */
   0
 };
@@ -498,7 +498,7 @@ locate_data_file (const char *const name
 #else
   struct stat s;
 #endif
-  const char *subdir = "/share/games/powermanga/";
+  const char *subdir = "/share/powermanga/";
 
   if (name == NULL)
     {
diff -Naupr powermanga-0.90.orig/texts/Makefile powermanga-0.90/texts/Makefile
--- powermanga-0.90.orig/texts/Makefile	2007-09-04 10:41:13.000000000 +0200
+++ powermanga-0.90/texts/Makefile	2007-10-22 13:06:03.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/texts
+pkgdatadir = $(datadir)/powermanga/texts
 ACLOCAL = ${SHELL} /home/bubux/dev/tempo/powermanga-0.90/autotools/missing --run aclocal-1.9
 AMDEP_FALSE = #
 AMDEP_TRUE = 
diff -Naupr powermanga-0.90.orig/texts/Makefile.am powermanga-0.90/texts/Makefile.am
--- powermanga-0.90.orig/texts/Makefile.am	2007-07-13 13:24:49.000000000 +0200
+++ powermanga-0.90/texts/Makefile.am	2007-10-22 13:06:53.000000000 +0200
@@ -1,5 +1,5 @@
 
-pkgdatadir = $(datadir)/games/powermanga/texts
+pkgdatadir = $(datadir)/powermanga/texts
 
 dist_pkgdata_DATA = \
   about_en.txt \
diff -Naupr powermanga-0.90.orig/texts/Makefile.in powermanga-0.90/texts/Makefile.in
--- powermanga-0.90.orig/texts/Makefile.in	2007-09-04 10:41:05.000000000 +0200
+++ powermanga-0.90/texts/Makefile.in	2007-10-22 13:06:49.000000000 +0200
@@ -58,7 +58,7 @@ am__installdirs = "$(DESTDIR)$(pkgdatadi
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/games/powermanga/texts
+pkgdatadir = $(datadir)/powermanga/texts
 ACLOCAL = @ACLOCAL@
 AMDEP_FALSE = @AMDEP_FALSE@
 AMDEP_TRUE = @AMDEP_TRUE@
