diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 318d4f8..382c53a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,7 +1,11 @@
+option(BUILD_SHARED_LIBS "build shared library" OFF) 
+
 include_directories(${LIBUECC_SOURCE_DIR}/include)
 
-add_library(uecc STATIC ec25519.c ec25519_gf.c)
+add_library(uecc ec25519.c ec25519_gf.c)
 
 install(TARGETS uecc
-  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib${LIB_SUFFIX}
+  ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
