diff --git a/session/budgie-session.c b/session/budgie-session.c
index 27be04d..a95ae0b 100644
--- a/session/budgie-session.c
+++ b/session/budgie-session.c
@@ -29,7 +29,7 @@
 #define DESKTOP_WM "budgie-wm"
 #define DESKTOP_PANEL "budgie-panel"
 /* Must re-address this at some point. Have a systemd-user session for it */
-#define DESKTOP_SETTINGS "/usr/lib/gnome-settings-daemon-3.0/gnome-settings-daemon"
+#define DESKTOP_SETTINGS "/usr/lib/gnome-settings-daemon/gnome-settings-daemon"
 #define FILE_MANAGER "nautilus -n"
 
 int main(int argc, char **argv)
