*** metadata.json	2012-05-06 18:43:13.000000000 +0400
--- /home/masroor/vibou.gTile/metadata.json	2012-05-06 18:44:50.986220881 +0400
***************
*** 1 ****
! {"shell-version": ["3.2"], "uuid": "gTile@vibou", "name": "gTile", "description": "Tile your windows as you like. It even supports multiscreen ! -EDIT: - See README file for configuration -", "url" : "https://github.com/vibou/vibou.gTile"}
--- 1 ----
! {"shell-version": ["3.2", "3.4", "3.4.1"], "uuid": "gTile@vibou", "name": "gTile", "description": "Tile your windows as you like. It even supports multiscreen ! -EDIT: - See README file for configuration -", "url" : "https://github.com/vibou/vibou.gTile"}
*** extension.js	2012-05-06 18:43:13.000000000 +0400
--- /home/masroor/vibou.gTile/extension.js	2012-05-06 18:45:41.722889825 +0400
***************
*** 70,76 ****
          gridSettings[SETTINGS_ANIMATION] = true;
       // Key Binding to Toggle the Window
       // uncomment this, and set the following command to set the keybinding:
!      Main.wm.setKeybindingHandler('run_command_9', toggleTiling);
       
       /**
       Use the following command to set the keybinding:
--- 70,76 ----
          gridSettings[SETTINGS_ANIMATION] = true;
       // Key Binding to Toggle the Window
       // uncomment this, and set the following command to set the keybinding:
!      Meta.keybindings_set_custom_handler('run_command_9', toggleTiling);
       
       /**
       Use the following command to set the keybinding:
***************
*** 108,114 ****
      initSettings();
      initGrids(); 
  
! 	Main.panel._rightBox.insert_actor(launcher.actor, 0);
  	
  }
  
--- 108,114 ----
      initSettings();
      initGrids(); 
  
! 	Main.panel._rightBox.add_actor(launcher.actor, 0);
  	
  }
  
