--- orig/setup.py	2013-03-29 17:17:18.920823455 +0100
+++ pyminuit2-1.1.0/setup.py	2013-03-29 17:16:30.981656192 +0100
@@ -8,7 +8,7 @@
 from subprocess import Popen, PIPE
 def root_config(*opts):
     args = ['root-config'] + ['--'+o for o in opts]
-    return Popen(args, stdout=PIPE).communicate()[0].strip().split(' ')
+    return Popen(args, stdout=PIPE).communicate()[0].decode('utf8').strip().split(' ')
 
 def find_library(lib):
     from distutils import ccompiler
