--- src/osd/sdl/sdl.mak	2011-10-19 16:29:42.000000000 +0200
+++ src/osd/sdl/sdl.mak	2011-10-19 16:31:17.250808385 +0200
@@ -485,9 +485,9 @@
 
 # the new debugger relies on GTK+ in addition to the base SDLMAME needs
 # Non-X11 builds can not use the debugger
-INCPATH += `pkg-config --cflags-only-I gtk+-2.0` `pkg-config --cflags-only-I gconf-2.0`
-CCOMFLAGS += `pkg-config --cflags-only-other gtk+-2.0` `pkg-config --cflags-only-other gconf-2.0`
-LIBS += `pkg-config --libs gtk+-2.0` `pkg-config --libs gconf-2.0`
+INCPATH +=
+CCOMFLAGS +=
+LIBS +=
 #CCOMFLAGS += -DGTK_DISABLE_DEPRECATED
 
 # some systems still put important things in a different prefix

--- src/osd/sdl/dview.c.orig	2013-01-12 21:52:02.982355413 +0100
+++ src/osd/sdl/dview.c	2013-01-12 21:57:12.141158943 +0100
@@ -1,7 +1,6 @@
 
 #include "dview.h"
 #ifndef SDLMAME_WIN32
-#include <gconf/gconf-client.h>
 #endif
 
 G_DEFINE_TYPE(DView, dview, GTK_TYPE_CONTAINER);
@@ -347,13 +346,9 @@
 static void dview_class_init(DViewClass *dvc)
 {
 #ifndef SDLMAME_WIN32
-	GConfClient *conf = gconf_client_get_default();
 	char *name = 0;
 	dvc->fixedfont = 0;
 
-	if(conf)
-		name = gconf_client_get_string(conf, "/desktop/gnome/interface/monospace_font_name", 0);
-
 	if(name) {
 		dvc->fixedfont = pango_font_description_from_string(name);
 		g_free(name);
