--- src/guymager-0.5.2/compileinfo.sh	2010-03-24 16:16:44.000000000 +0100
+++ compileinfo.sh	2010-05-18 13:53:35.072625452 +0200
@@ -2,10 +2,6 @@
 
 echo '// Automatically generated file. See project file and compileinfo.sh for further information.'
 date '+const char *pCompileInfoTimestamp = "%Y-%m-%d-%H.%M.%S";'
-head -n 1 debian/changelog | awk '{
-                                    Version = $2
-                                    gsub ("\\(", "", Version)
-                                    gsub ("\\)", "", Version)
-                                    print "const char *pCompileInfoVersion   = \"" Version "\";"}'
+printf "const char *pCompileInfoVersion   = "\"0.5.2\"\;"";
 
 
