--- src/guymager-0.5.2/main.cpp	2010-05-12 11:49:07.381453890 +0200
+++ main.cpp	2010-05-12 12:15:29.387545746 +0200
@@ -29,9 +29,9 @@
 #undef free     // in order to be able to access std::malloc and std::free
 
 #include "toolconstants.h"
-#include "signal/toolsignal.h"
-#include "sysinfo/toolsysinfo.h"
-#include "cfg/toolcfg.h"
+#include "toolsignal.h"
+#include "toolsysinfo.h"
+#include "toolcfg.h"
 
 #include "libewf.h"
 
@@ -70,8 +70,8 @@
 #define MAIN_LANGUAGE_FILENAME_DEFAULT "guymager_en"
 #define MAIN_LANGUAGE_FILENAME_QT      "qt_"
 #define MAIN_DEFAULT_MISCFILES_DIR     "/usr/share/guymager/"
-#define MAIN_DEFAULT_LANGUAGE_DIR_QT   "/usr/share/qt4/translations/"
-
+#define MAIN_DEFAULT_LANGUAGE_DIR_QT   "/usr/share/qt/translations/"
+#define SPLASH_DIR                     MAIN_DEFAULT_MISCFILES_DIR
 // ------------------------------------
 //   Application-wide global variables
 // ------------------------------------
