Index: mango/icu/UBreakIterator.d
===================================================================
--- mango/icu/UBreakIterator.d	(revision 1122)
+++ mango/icu/UBreakIterator.d	(working copy)
@@ -300,7 +300,7 @@
 
 *******************************************************************************/
 
-private class UBreakIterator : ICU
+class UBreakIterator : ICU
 {       
         package Handle  handle;  
 
Index: mango/icu/UString.d
===================================================================
--- mango/icu/UString.d	(revision 1122)
+++ mango/icu/UString.d	(working copy)
@@ -765,7 +765,8 @@
 
         final override uint toHash ()
         {
-                return typeid(wchar[]).getHash (&content[0..len]);
+		wchar [] tmp = content[0..len];
+                return (typeid(wchar[]).getHash (&tmp));
         }
 
         /***********************************************************************
Index: mango/xml/rpc/Client.d
===================================================================
--- mango/xml/rpc/Client.d	(revision 1122)
+++ mango/xml/rpc/Client.d	(working copy)
@@ -39,7 +39,8 @@
 
 private import tango.text.convert.UnicodeBom,
 	Double = tango.text.convert.Float,
-	Text = tango.text.Util;
+	Text = tango.text.Util,
+	Integer = tango.text.convert.Integer;
 
 private import mango.xml.sax.model.ISAXParser,
 	mango.xml.sax.model.ISAXHandler,
@@ -753,4 +754,4 @@
 		}
 	} else
 		static assert(0);
-}
\ No newline at end of file
+}
