From fb2165860a0dac001739c11898d7bcdc70d7badb Mon Sep 17 00:00:00 2001
From: Martin Thornton
Date: 2013-11-26
Subject: [PATCH] Create lock file to prevent more than one logout dialog

---
 lxsession-logout/lxsession-logout.c | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/lxsession-logout/lxsession-logout.c b/lxsession-logout/lxsession-logout.c
index 9ace78d..a430dcd 100644
--- a/lxsession-logout/lxsession-logout.c
+++ b/lxsession-logout/lxsession-logout.c
@@ -23,6 +23,7 @@
 #include <gdk/gdk.h>
 #include <gdk/gdkx.h>
 #include <glib/gi18n.h>
+#include <sys/file.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
@@ -440,6 +441,14 @@ gboolean expose_event(GtkWidget * widget, GdkEventExpose * event, GdkPixbuf * pi
     return FALSE;
 }
 
+static char lockfile[PATH_MAX];
+
+/* Unlink lockfile on exit. */
+static void main_at_exit(void)
+{
+    unlink(lockfile);
+}
+
 /* Main program. */
 int main(int argc, char * argv[])
 {
@@ -470,6 +479,18 @@ int main(int argc, char * argv[])
     const char * p = g_getenv("_LXSESSION_PID");
     if (p != NULL) handler_context.lxsession_pid = atoi(p);
 
+    /* Create lock file to prevent more than one logout dialog per lxsession process. */
+    sprintf(lockfile, "/tmp/.lxsession-logout-%d.lock", handler_context.lxsession_pid);
+    int fd = open(lockfile, O_RDONLY|O_CREAT, 00600);
+    if (fd >= 0)
+    {
+        if (flock(fd, LOCK_EX | LOCK_NB))
+        {
+            exit(EXIT_FAILURE);
+        }
+    }
+    atexit(main_at_exit);
+
     /* Initialize capabilities of the systemd mechanism. */
     if (dbus_systemd_CanPowerOff())
     {
@@ -593,6 +614,7 @@ int main(int argc, char * argv[])
     gtk_window_set_default_size(GTK_WINDOW(window), gdk_screen_get_width(screen), gdk_screen_get_height(screen));
     gtk_widget_set_app_paintable(window, TRUE);
     g_signal_connect(G_OBJECT(window), "expose_event", G_CALLBACK(expose_event), pixbuf);
+    g_signal_connect(G_OBJECT(window), "destroy", G_CALLBACK(gtk_main_quit), NULL);
 
     /* Toplevel container */
     GtkWidget* alignment = gtk_alignment_new(0.5, 0.5, 0.0, 0.0);
-- 
2.0.0

