--- configure.ac.new	2010-08-16 07:39:51.000000000 +0300
+++ configure.ac	2008-03-24 17:27:24.000000000 +0200
@@ -18,10 +18,7 @@
 AC_SUBST(X_LIBS)
 AC_SUBST(X_EXTRA_LIBS)
 
-AC_CHECK_LIB([X11], [XOpenDisplay], HAVE_X11="yes", HAVE_X11="no")
-if test "x$HAVE_X11" = "xno" ; then
-  AC_MSG_ERROR([libX11 is required])
-fi
+AC_CHECK_LIB([X11], [XOpenDisplay])
 
 # Error out on compile warnings
 dnl Add some useful warnings if we have gcc.
