diff --git a/sketches/dump_to_html.lua b/sketches/dump_to_html.lua
index 5fb8f7d..e70b102 100644
--- a/sketches/dump_to_html.lua
+++ b/sketches/dump_to_html.lua
@@ -160,6 +160,7 @@ function dump_to_html(src_file, grammar, dir)
     intfa_file:close()
     os.execute(string.format("dot -Tpng -o %s %s", png_file, dot_file))
     local w = 800
+    local h = 600
     if have_imagemagick then
       local img_info = io.popen(string.format("identify %s", png_file)):read("*a")
       w, h = string.match(img_info, "[^ ]+ [^ ]+ (%d+)x(%d+)")
