diff --git a/Makefile b/Makefile
index ac313fe..6aa0405 100644
--- a/Makefile
+++ b/Makefile
@@ -3,6 +3,7 @@
 BINDIR := $(DESTDIR)$(PREFIX)/bin
 LIBDIR := $(DESTDIR)$(PREFIX)/lib
 INCDIR := $(DESTDIR)$(PREFIX)/include
+DOCDIR := $(DESTDIR)$(PREFIX)/share/doc
 
 IMGDIR := /usr/share/asciidoc/images
 
@@ -39,6 +40,7 @@
 LIB := $(LUALIB) runtime/libgazelle.a
 INC := $(wildcard runtime/include/gazelle/*.h)
 IMG := $(foreach img,$(wildcard $(IMGDIR)/*.png),docs/images/$(notdir $(img)))
+DOC := $(wildcard docs/*.html docs/*.gif docs/*.png)
 
 .PHONY: all clean doc install test
 
@@ -98,6 +100,10 @@
 	install -m 0644 runtime/libgazelle.a $(LIBDIR)
 	install -d $(INCDIR)/gazelle
 	install -m 0644 $(INC) $(INCDIR)/gazelle
+ 
+installdoc: docs/manual.html $(DOC)
+	install -d $(DOCDIR)/gazelle
+	install -m 0644 $(DOC) $(DOCDIR)/gazelle
 
 clean:
 	$(RM) $(OBJ)
