Index: cblrepo/purescript-0.6.4/purescript.cabal
===================================================================
--- cblrepo.orig/purescript-0.6.4/purescript.cabal
+++ cblrepo/purescript-0.6.4/purescript.cabal
@@ -32,7 +32,7 @@ library
                    mtl >= 2.1.0 && < 2.3.0,
                    parsec -any,
                    transformers >= 0.3 && < 0.5,
-                   utf8-string >= 1 && < 2,
+                   utf8-string,
                    pattern-arrows >= 0.0.2 && < 0.1,
                    monad-unify >= 0.2.2 && < 0.3,
                    file-embed >= 0.0.7 && < 0.0.8,
