diff -u quadra/skelton/include/dict.h quadra/skelton/include.new/dict.h
--- quadra/skelton/include/dict.h	2011-09-23 16:06:20.936832915 +0200
+++ quadra/skelton/include.new/dict.h	2011-09-23 16:08:38.791691192 +0200
@@ -21,6 +21,7 @@
 #ifndef _HEADER_DICT
 #define _HEADER_DICT
 
+#include <cstddef>
 #include <vector>
 
 #include "types.h"
diff -u quadra/skelton/include/net.h quadra/skelton/include.new/net.h
--- quadra/skelton/include/net.h	2011-09-23 16:06:20.953499766 +0200
+++ quadra/skelton/include.new/net.h	2011-09-23 16:08:23.564855751 +0200
@@ -21,6 +21,7 @@
 #ifndef _HEADER_NET
 #define _HEADER_NET
 
+#include <cstddef>
 #include <list>
 #include <vector>
 
Subdirectorios comunes: quadra/skelton/include/.svn y quadra/skelton/include.new/.svn
