diff -urN fvwm-2.5.10.orig/bin/fvwm-menu-xlock.in fvwm-2.5.10/bin/fvwm-menu-xlock.in
--- fvwm-2.5.10.orig/bin/fvwm-menu-xlock.in	2003-03-22 02:01:24.000000000 +0000
+++ fvwm-2.5.10/bin/fvwm-menu-xlock.in	2004-05-24 12:26:14.641330208 +0100
@@ -42,7 +42,7 @@
 my $start = 0;
 my $special = 0;
 $itemF =~ s/\\t/\t/g;
-open(XL, "xlock -h 2>&1 |") || die "Exec echo 'Could not run xlock'\n";
+open(XL, "xlock -help 2>&1 |") || die "Exec echo 'Could not run xlock'\n";
 
 print "DestroyMenu $name\n";
 print "AddToMenu $name \"$iconTStr$title\" Title\n";
