--- old/setup.py	2010-10-27 00:44:19.999594258 +0200
+++ new/setup.py	2010-10-27 00:43:26.549594258 +0200
@@ -12,7 +12,7 @@
        author      = "Thomas Breuel",
        description = """Python bindings for OCRopus""",
        packages = ["ocropy"],
-       scripts = glob.glob("ocropus-*"),
+       scripts = [i for i in glob.glob("ocropus-*") if not i.endswith('.glade')],
        data_files=[('/usr/local/share/ocropus/models',
                     ["2m2-reject.cmodel","multi3.cmodel"]),
                    ('/usr/local/share/ocropus/gui',
