diff -urN dosbox.orig/src/dosbox.cpp dosbox/src/dosbox.cpp
--- dosbox.orig/src/dosbox.cpp	2014-10-26 17:07:51.406125243 -0300
+++ dosbox/src/dosbox.cpp	2014-10-26 17:07:56.582771721 -0300
@@ -465,6 +465,9 @@
 	Pbool = secprop->Add_bool("nosound",Property::Changeable::OnlyAtStart,false);
 	Pbool->Set_help("Enable silent mode, sound is still emulated though.");
 
+	Pbool = secprop->Add_bool("swapstereo",Property::Changeable::OnlyAtStart,false);
+	Pbool->Set_help("Swaps the left and right stereo channels.");
+
 	Pint = secprop->Add_int("rate",Property::Changeable::OnlyAtStart,44100);
 	Pint->Set_values(rates);
 	Pint->Set_help("Mixer sample rate, setting any device's rate higher than this will probably lower their sound quality.");
diff -urN dosbox.orig/src/hardware/mixer.cpp dosbox/src/hardware/mixer.cpp
--- dosbox.orig/src/hardware/mixer.cpp	2014-10-26 17:07:51.399458602 -0300
+++ dosbox/src/hardware/mixer.cpp	2014-10-26 17:07:56.582771721 -0300
@@ -72,6 +72,7 @@
 	bool nosound;
 	Bit32u freq;
 	Bit32u blocksize;
+	bool swapstereo;
 } mixer;
 
 Bit8u MixTemp[MIXER_BUFSIZE];
@@ -167,6 +168,9 @@
 	Bitu mixpos=mixer.pos+done;
 	freq_index&=MIXER_REMAIN;
 	Bitu pos=0;Bitu new_pos;
+	int offset[2];
+	offset[0] = mixer.swapstereo ? 1:0;
+	offset[1] = mixer.swapstereo ? 0:1;
 
 	goto thestart;
 	for (;;) {
@@ -180,15 +184,15 @@
 			if ( sizeof( Type) == 1) {
 				if (!signeddata) {
 					if (stereo) {
-						diff[0]=(((Bit8s)(data[pos*2+0] ^ 0x80)) << 8)-last[0];
-						diff[1]=(((Bit8s)(data[pos*2+1] ^ 0x80)) << 8)-last[1];
+						diff[0]=(((Bit8s)(data[pos*2+offset[0]] ^ 0x80)) << 8)-last[0];
+						diff[1]=(((Bit8s)(data[pos*2+offset[1]] ^ 0x80)) << 8)-last[1];
 					} else {
 						diff[0]=(((Bit8s)(data[pos] ^ 0x80)) << 8)-last[0];
 					}
 				} else {
 					if (stereo) {
-						diff[0]=(data[pos*2+0] << 8)-last[0];
-						diff[1]=(data[pos*2+1] << 8)-last[1];
+						diff[0]=(data[pos*2+offset[0]] << 8)-last[0];
+						diff[1]=(data[pos*2+offset[1]] << 8)-last[1];
 					} else {
 						diff[0]=(data[pos] << 8)-last[0];
 					}
@@ -198,8 +202,8 @@
 				if (signeddata) {
 					if (stereo) {
 						if (nativeorder) {
-							diff[0]=data[pos*2+0]-last[0];
-							diff[1]=data[pos*2+1]-last[1];
+							diff[0]=data[pos*2+offset[0]]-last[0];
+							diff[1]=data[pos*2+offset[1]]-last[1];
 						} else {
 							if ( sizeof( Type) == 2) {
 								diff[0]=(Bit16s)host_readw((HostPt)&data[pos*2+0])-last[0];
@@ -223,8 +227,8 @@
 				} else {
 					if (stereo) {
 						if (nativeorder) {
-							diff[0]=(Bits)data[pos*2+0]-32768-last[0];
-							diff[1]=(Bits)data[pos*2+1]-32768-last[1];
+							diff[0]=(Bits)data[pos*2+offset[0]]-32768-last[0];
+							diff[1]=(Bits)data[pos*2+offset[1]]-32768-last[1];
 						} else {
 							if ( sizeof( Type) == 2) {
 								diff[0]=(Bits)host_readw((HostPt)&data[pos*2+0])-32768-last[0];
@@ -613,6 +617,7 @@
 	mixer.freq=section->Get_int("rate");
 	mixer.nosound=section->Get_bool("nosound");
 	mixer.blocksize=section->Get_int("blocksize");
+	mixer.swapstereo=section->Get_bool("swapstereo");
 
 	/* Initialize the internal stuff */
 	mixer.channels=0;
