diff -urN dosbox-svn.orig/src/hardware/cmos.cpp dosbox-svn/src/hardware/cmos.cpp
--- dosbox-svn.orig/src/hardware/cmos.cpp	2013-10-29 05:23:42.694829284 -0300
+++ dosbox-svn/src/hardware/cmos.cpp	2013-10-29 05:24:06.604736035 -0300
@@ -194,9 +194,11 @@
 		return ((drive_a << 4) | (drive_b));
 	/* First harddrive info */
 	case 0x12:
+		/* NTS: DOSBox 0.74 mainline has these backwards: the upper nibble is the first hard disk,
+		   the lower nibble is the second hard disk. */
 		hdparm = 0;
-		if(imageDiskList[2] != NULL) hdparm |= 0xf;
-		if(imageDiskList[3] != NULL) hdparm |= 0xf0;
+		if(imageDiskList[3] != NULL) hdparm |= 0xf;
+		if(imageDiskList[2] != NULL) hdparm |= 0xf0;
 		return hdparm;
 	case 0x19:
 		if(imageDiskList[2] != NULL) return 47; /* User defined type */
