diff -ur dosbox.orig/configure.ac dosbox/configure.ac
--- dosbox.orig/configure.ac	2013-06-27 15:18:35.639527241 -0300
+++ dosbox/configure.ac	2013-06-27 15:26:38.608395440 -0300
@@ -320,12 +320,8 @@
    AC_MSG_RESULT(no)
 else
   if test x$enable_fpu = xyes; then
-    if test x$c_targetcpu = xx86 ; then
-        AC_DEFINE(C_FPU_X86,1)
-        AC_MSG_RESULT(yes)
-    else
-        AC_MSG_RESULT(no)
-    fi
+    AC_DEFINE(C_FPU_X86,1)
+    AC_MSG_RESULT(yes)
   else
       AC_MSG_RESULT(no)
   fi
