diff -ur dosbox.orig/src/dos/dos_programs.cpp dosbox/src/dos/dos_programs.cpp
--- dosbox.orig/src/dos/dos_programs.cpp	2012-09-29 00:57:44.473335051 -0300
+++ dosbox/src/dos/dos_programs.cpp	2012-09-29 01:02:52.276671037 -0300
@@ -1440,10 +1440,10 @@
 	MSG_Add("PROGRAM_MOUNT_STATUS_FORMAT","%-5s  %-58s %-12s\n");
 	MSG_Add("PROGRAM_MOUNT_STATUS_2","Drive %c is mounted as %s\n");
 	MSG_Add("PROGRAM_MOUNT_STATUS_1","The currently mounted drives are:\n");
-	MSG_Add("PROGRAM_MOUNT_ERROR_1","Directory %s doesn't exist.\n");
-	MSG_Add("PROGRAM_MOUNT_ERROR_2","%s isn't a directory\n");
-	MSG_Add("PROGRAM_MOUNT_ILL_TYPE","Illegal type %s\n");
-	MSG_Add("PROGRAM_MOUNT_ALREADY_MOUNTED","Drive %c already mounted with %s\n");
+	MSG_Add("PROGRAM_MOUNT_ERROR_1","\aDirectory %s doesn't exist.\n");
+	MSG_Add("PROGRAM_MOUNT_ERROR_2","\a%s isn't a directory\n");
+	MSG_Add("PROGRAM_MOUNT_ILL_TYPE","\aIllegal type %s\n");
+	MSG_Add("PROGRAM_MOUNT_ALREADY_MOUNTED","\aDrive %c already mounted with %s\n");
 	MSG_Add("PROGRAM_MOUNT_USAGE",
 		"Usage \033[34;1mMOUNT Drive-Letter Local-Directory\033[0m\n"
 		"For example: MOUNT c %s\n"
@@ -1451,7 +1451,7 @@
 		"The directory has to exist.\n");
 	MSG_Add("PROGRAM_MOUNT_UMOUNT_NOT_MOUNTED","Drive %c isn't mounted.\n");
 	MSG_Add("PROGRAM_MOUNT_UMOUNT_SUCCESS","Drive %c has successfully been removed.\n");
-	MSG_Add("PROGRAM_MOUNT_UMOUNT_NO_VIRTUAL","Virtual Drives can not be unMOUNTed.\n");
+	MSG_Add("PROGRAM_MOUNT_UMOUNT_NO_VIRTUAL","\aVirtual Drives can not be unMOUNTed.\n");
 	MSG_Add("PROGRAM_MOUNT_WARNING_WIN","\033[31;1mMounting c:\\ is NOT recommended. Please mount a (sub)directory next time.\033[0m\n");
 	MSG_Add("PROGRAM_MOUNT_WARNING_OTHER","\033[31;1mMounting / is NOT recommended. Please mount a (sub)directory next time.\033[0m\n");
 
@@ -1463,16 +1463,16 @@
 	MSG_Add("PROGRAM_LOADFIX_ALLOC","%d kb allocated.\n");
 	MSG_Add("PROGRAM_LOADFIX_DEALLOC","%d kb freed.\n");
 	MSG_Add("PROGRAM_LOADFIX_DEALLOCALL","Used memory freed.\n");
-	MSG_Add("PROGRAM_LOADFIX_ERROR","Memory allocation error.\n");
+	MSG_Add("PROGRAM_LOADFIX_ERROR","\aMemory allocation error.\n");
 
 	MSG_Add("MSCDEX_SUCCESS","MSCDEX installed.\n");
-	MSG_Add("MSCDEX_ERROR_MULTIPLE_CDROMS","MSCDEX: Failure: Drive-letters of multiple CDRom-drives have to be continuous.\n");
+	MSG_Add("MSCDEX_ERROR_MULTIPLE_CDROMS","\aMSCDEX: Failure: Drive-letters of multiple CDRom-drives have to be continuous.\n");
 	MSG_Add("MSCDEX_ERROR_NOT_SUPPORTED","MSCDEX: Failure: Not yet supported.\n");
-	MSG_Add("MSCDEX_ERROR_OPEN","MSCDEX: Failure: Invalid file or unable to open.\n");
-	MSG_Add("MSCDEX_TOO_MANY_DRIVES","MSCDEX: Failure: Too many CDRom-drives (max: 5). MSCDEX Installation failed.\n");
+	MSG_Add("MSCDEX_ERROR_OPEN","\aMSCDEX: Failure: Invalid file or unable to open.\n");
+	MSG_Add("MSCDEX_TOO_MANY_DRIVES","\aMSCDEX: Failure: Too many CDRom-drives (max: 5). MSCDEX Installation failed.\n");
 	MSG_Add("MSCDEX_LIMITED_SUPPORT","MSCDEX: Mounted subdirectory: limited support.\n");
-	MSG_Add("MSCDEX_INVALID_FILEFORMAT","MSCDEX: Failure: File is either no iso/cue image or contains errors.\n");
-	MSG_Add("MSCDEX_UNKNOWN_ERROR","MSCDEX: Failure: Unknown error.\n");
+	MSG_Add("MSCDEX_INVALID_FILEFORMAT","\aMSCDEX: Failure: File is either no iso/cue image or contains errors.\n");
+	MSG_Add("MSCDEX_UNKNOWN_ERROR","\aMSCDEX: Failure: Unknown error.\n");
 
 	MSG_Add("PROGRAM_RESCAN_SUCCESS","Drive cache cleared.\n");
 
@@ -1571,8 +1571,8 @@
 		"\033[33;1mCTRL-F12\033[0m    : Speed up emulation (Increase DOSBox Cycles).\n"
 		"\033[33;1mALT-F12\033[0m     : Unlock speed (turbo button/fast forward).\n"
 		);
-	MSG_Add("PROGRAM_BOOT_NOT_EXIST","Bootdisk file does not exist.  Failing.\n");
-	MSG_Add("PROGRAM_BOOT_NOT_OPEN","Cannot open bootdisk file.  Failing.\n");
+	MSG_Add("PROGRAM_BOOT_NOT_EXIST","\aBootdisk file does not exist.  Failing.\n");
+	MSG_Add("PROGRAM_BOOT_NOT_OPEN","\aCannot open bootdisk file.  Failing.\n");
 	MSG_Add("PROGRAM_BOOT_WRITE_PROTECTED","Image file is read-only! Might create problems.\n");
 	MSG_Add("PROGRAM_BOOT_PRINT_ERROR","This command boots DOSBox from either a floppy or hard disk image.\n\n"
 		"For this command, one can specify a succession of floppy disks swappable\n"
@@ -1584,36 +1584,36 @@
 		"The syntax of this command is:\n\n"
 		"\033[34;1mBOOT [diskimg1.img diskimg2.img] [-l driveletter]\033[0m\n"
 		);
-	MSG_Add("PROGRAM_BOOT_UNABLE","Unable to boot off of drive %c");
+	MSG_Add("PROGRAM_BOOT_UNABLE","\aUnable to boot off of drive %c");
 	MSG_Add("PROGRAM_BOOT_IMAGE_OPEN","Opening image file: %s\n");
-	MSG_Add("PROGRAM_BOOT_IMAGE_NOT_OPEN","Cannot open %s");
+	MSG_Add("PROGRAM_BOOT_IMAGE_NOT_OPEN","\aCannot open %s");
 	MSG_Add("PROGRAM_BOOT_BOOT","Booting from drive %c...\n");
 	MSG_Add("PROGRAM_BOOT_CART_WO_PCJR","PCjr cartridge found, but machine is not PCjr");
 	MSG_Add("PROGRAM_BOOT_CART_LIST_CMDS","Available PCjr cartridge commandos:%s");
 	MSG_Add("PROGRAM_BOOT_CART_NO_CMDS","No PCjr cartridge commandos found");
 
-	MSG_Add("PROGRAM_IMGMOUNT_SPECIFY_DRIVE","Must specify drive letter to mount image at.\n");
-	MSG_Add("PROGRAM_IMGMOUNT_SPECIFY2","Must specify drive number (0 or 3) to mount image at (0,1=fda,fdb;2,3=hda,hdb).\n");
+	MSG_Add("PROGRAM_IMGMOUNT_SPECIFY_DRIVE","\aMust specify drive letter to mount image at.\n");
+	MSG_Add("PROGRAM_IMGMOUNT_SPECIFY2","\aMust specify drive number (0 or 3) to mount image at (0,1=fda,fdb;2,3=hda,hdb).\n");
 	MSG_Add("PROGRAM_IMGMOUNT_SPECIFY_GEOMETRY",
 		"For \033[33mCD-ROM\033[0m images:   \033[34;1mIMGMOUNT drive-letter location-of-image -t iso\033[0m\n"
 		"\n"
 		"For \033[33mhardrive\033[0m images: Must specify drive geometry for hard drives:\n"
 		"bytes_per_sector, sectors_per_cylinder, heads_per_cylinder, cylinder_count.\n"
 		"\033[34;1mIMGMOUNT drive-letter location-of-image -size bps,spc,hpc,cyl\033[0m\n");
-	MSG_Add("PROGRAM_IMGMOUNT_INVALID_IMAGE","Could not load image file.\n"
+	MSG_Add("PROGRAM_IMGMOUNT_INVALID_IMAGE","\aCould not load image file.\n"
 		"Check that the path is correct and the image is accessible.\n");
-	MSG_Add("PROGRAM_IMGMOUNT_INVALID_GEOMETRY","Could not extract drive geometry from image.\n"
+	MSG_Add("PROGRAM_IMGMOUNT_INVALID_GEOMETRY","\aCould not extract drive geometry from image.\n"
 		"Use parameter -size bps,spc,hpc,cyl to specify the geometry.\n");
-	MSG_Add("PROGRAM_IMGMOUNT_TYPE_UNSUPPORTED","Type \"%s\" is unsupported. Specify \"hdd\" or \"floppy\" or\"iso\".\n");
-	MSG_Add("PROGRAM_IMGMOUNT_FORMAT_UNSUPPORTED","Format \"%s\" is unsupported. Specify \"fat\" or \"iso\" or \"none\".\n");
-	MSG_Add("PROGRAM_IMGMOUNT_SPECIFY_FILE","Must specify file-image to mount.\n");
-	MSG_Add("PROGRAM_IMGMOUNT_FILE_NOT_FOUND","Image file not found.\n");
-	MSG_Add("PROGRAM_IMGMOUNT_MOUNT","To mount directories, use the \033[34;1mMOUNT\033[0m command, not the \033[34;1mIMGMOUNT\033[0m command.\n");
-	MSG_Add("PROGRAM_IMGMOUNT_ALREADY_MOUNTED","Drive already mounted at that letter.\n");
-	MSG_Add("PROGRAM_IMGMOUNT_CANT_CREATE","Can't create drive from file.\n");
+	MSG_Add("PROGRAM_IMGMOUNT_TYPE_UNSUPPORTED","\aType \"%s\" is unsupported. Specify \"hdd\" or \"floppy\" or\"iso\".\n");
+	MSG_Add("PROGRAM_IMGMOUNT_FORMAT_UNSUPPORTED","\aFormat \"%s\" is unsupported. Specify \"fat\" or \"iso\" or \"none\".\n");
+	MSG_Add("PROGRAM_IMGMOUNT_SPECIFY_FILE","\aMust specify file-image to mount.\n");
+	MSG_Add("PROGRAM_IMGMOUNT_FILE_NOT_FOUND","\aImage file not found.\n");
+	MSG_Add("PROGRAM_IMGMOUNT_MOUNT","\aTo mount directories, use the \033[34;1mMOUNT\033[0m command, not the \033[34;1mIMGMOUNT\033[0m command.\n");
+	MSG_Add("PROGRAM_IMGMOUNT_ALREADY_MOUNTED","\aDrive already mounted at that letter.\n");
+	MSG_Add("PROGRAM_IMGMOUNT_CANT_CREATE","\aCan't create drive from file.\n");
 	MSG_Add("PROGRAM_IMGMOUNT_MOUNT_NUMBER","Drive number %d mounted as %s\n");
-	MSG_Add("PROGRAM_IMGMOUNT_NON_LOCAL_DRIVE", "The image must be on a host or local drive.\n");
-	MSG_Add("PROGRAM_IMGMOUNT_MULTIPLE_NON_CUEISO_FILES", "Using multiple files is only supported for cue/iso images.\n");
+	MSG_Add("PROGRAM_IMGMOUNT_NON_LOCAL_DRIVE", "\aThe image must be on a host or local drive.\n");
+	MSG_Add("PROGRAM_IMGMOUNT_MULTIPLE_NON_CUEISO_FILES", "\aUsing multiple files is only supported for cue/iso images.\n");
 
 	MSG_Add("PROGRAM_KEYB_INFO","Codepage %i has been loaded\n");
 	MSG_Add("PROGRAM_KEYB_INFO_LAYOUT","Codepage %i has been loaded for layout %s\n");
@@ -1625,10 +1625,10 @@
 		"  \033[32;1mKEYB\033[0m sp 850: Load the spanish (SP) layout, use codepage 850.\n"
 		"  \033[32;1mKEYB\033[0m sp 850 mycp.cpi: Same as above, but use file mycp.cpi.\n");
 	MSG_Add("PROGRAM_KEYB_NOERROR","Keyboard layout %s loaded for codepage %i\n");
-	MSG_Add("PROGRAM_KEYB_FILENOTFOUND","Keyboard file %s not found\n\n");
-	MSG_Add("PROGRAM_KEYB_INVALIDFILE","Keyboard file %s invalid\n");
-	MSG_Add("PROGRAM_KEYB_LAYOUTNOTFOUND","No layout in %s for codepage %i\n");
-	MSG_Add("PROGRAM_KEYB_INVCPFILE","None or invalid codepage file for layout %s\n\n");
+	MSG_Add("PROGRAM_KEYB_FILENOTFOUND","\aKeyboard file %s not found\n\n");
+	MSG_Add("PROGRAM_KEYB_INVALIDFILE","\aKeyboard file %s invalid\n");
+	MSG_Add("PROGRAM_KEYB_LAYOUTNOTFOUND","\aNo layout in %s for codepage %i\n");
+	MSG_Add("PROGRAM_KEYB_INVCPFILE","\aNone or invalid codepage file for layout %s\n\n");
 
 	/*regular setup*/
 	PROGRAMS_MakeFile("MOUNT.COM",MOUNT_ProgramStart);
diff -ur dosbox.orig/src/ints/int10_char.cpp dosbox/src/ints/int10_char.cpp
--- dosbox.orig/src/ints/int10_char.cpp	2012-09-29 00:57:44.493334205 -0300
+++ dosbox/src/ints/int10_char.cpp	2012-09-29 00:58:04.912471318 -0300
@@ -24,6 +24,7 @@
 #include "mem.h"
 #include "inout.h"
 #include "int10.h"
+#include "callback.h"
 
 static void CGA2_CopyRow(Bit8u cleft,Bit8u cright,Bit8u rold,Bit8u rnew,PhysPt base) {
 	Bit8u cheight = real_readb(BIOSMEM_SEG,BIOSMEM_CHAR_HEIGHT);
@@ -584,9 +585,19 @@
 	Bit8u cur_row=CURSOR_POS_ROW(page);
 	Bit8u cur_col=CURSOR_POS_COL(page);
 	switch (chr) {
-	case 7:
-	//TODO BEEP
-	break;
+	case 7: {
+		// set timer (this should not be needed as the timer already is programmed 
+		// with those values, but the speaker stays silent without it)
+		IO_Write(0x43,0xb6);
+		IO_Write(0x42,1320&0xff);
+		IO_Write(0x42,1320>>8);
+		// enable speaker
+		IO_Write(0x61,IO_Read(0x61)|0x3);
+		for(Bitu i=0; i < 333; i++) CALLBACK_Idle();
+		IO_Write(0x61,IO_Read(0x61)&~0x3);
+		break;
+	}
+
 	case 8:
 		if(cur_col>0) cur_col--;
 		break;
