diff -urN dosbox.orig/src/ints/int10_char.cpp dosbox/src/ints/int10_char.cpp
--- dosbox.orig/src/ints/int10_char.cpp	2012-01-18 11:50:32.647557003 -0300
+++ dosbox/src/ints/int10_char.cpp	2012-01-18 11:50:58.823195976 -0300
@@ -24,6 +24,7 @@
 #include "mem.h"
 #include "inout.h"
 #include "int10.h"
+#include "callback.h"
 
 static void CGA2_CopyRow(Bit8u cleft,Bit8u cright,Bit8u rold,Bit8u rnew,PhysPt base) {
 	Bit8u cheight = real_readb(BIOSMEM_SEG,BIOSMEM_CHAR_HEIGHT);
@@ -572,9 +573,18 @@
 	Bit8u cur_row=CURSOR_POS_ROW(page);
 	Bit8u cur_col=CURSOR_POS_COL(page);
 	switch (chr) {
-	case 7:
-	//TODO BEEP
-	break;
+	case 7: {
+		// set timer (this should not be needed as the timer already is programmed
+		// with those values, but the speaker stays silent without it)
+		IO_Write(0x43,0xb6);
+		IO_Write(0x42,1320&0xff);
+		IO_Write(0x42,1320>>8);
+		// enable speaker
+		IO_Write(0x61,IO_Read(0x61)|0x3);
+		for(Bitu i=0; i < 333; i++) CALLBACK_Idle();
+		IO_Write(0x61,IO_Read(0x61)&~0x3);
+		break;
+	}
 	case 8:
 		if(cur_col>0) cur_col--;
 		break;
