--- Makefile.ori	2007-04-01 21:21:51.000000000 +0200
+++ Makefile	2008-01-21 20:48:03.000000000 +0100
@@ -16,5 +16,7 @@
 	-rm gexec
 	-rm *.o
 install:
-	-cp gexec /usr/local/bin
-	-cp gexec.1 /usr/local/man/man1/
+	-test -d ${DESTDIR}/usr/bin || mkdir -p ${DESTDIR}/usr/bin
+	-test -d ${DESTDIR}/usr/man/man1 || mkdir -p ${DESTDIR}/usr/man/man1
+	-cp gexec ${DESTDIR}/usr/bin/
+	-cp gexec.1 ${DESTDIR}/usr/man/man1/
