--- build.orig	2012-12-17 21:39:20.791118451 +0100
+++ build.new	2012-12-17 21:38:15.000000000 +0100
@@ -19,15 +19,17 @@
 my $firmware_url = "http://www.linuxtv.org/downloads/firmware/";
 my $firmware_tarball = "dvb-firmwares.tar.bz2";
 
-GetOptions('v|verbose' => \$level,
-	   'help|?' => \$help,
-	    man => \$man,
-	   'check_only|check-only' => \$check_only,
-	   'main_git|main-git' => \$main_git,
-	   'git=s{2}' => \@git,
-	  ) or pod2usage(2);
-pod2usage(1) if $help;
-pod2usage(-exitstatus => 0, -verbose => 2) if $man;
+my $pkgdir = $ARGV[1];
+
+#GetOptions('v|verbose' => \$level,
+#	   'help|?' => \$help,
+#	    man => \$man,
+#	   'check_only|check-only' => \$check_only,
+#	   'main_git|main-git' => \$main_git,
+#	   'git=s{2}' => \@git,
+#	  ) or pod2usage(2);
+#pod2usage(1) if $help;
+#pod2usage(-exitstatus => 0, -verbose => 2) if $man;
 
 #############
 # Static vars
@@ -411,7 +413,7 @@
 				or die "Can't update local branch $rname";
 	}
 
-	system ("make -C linux dir DIR=../media/") == 0
+	system ("make DESTDIR='$pkgdir' -C linux dir DIR=../media/") == 0
 		or die "Can't link the building system to the media directory.";
 } else {
 	print "\n";
@@ -430,8 +432,8 @@
 	print "****************************\n";
 	system("git pull git://linuxtv.org/media_build.git master");
 
-	system ("make -C linux/ download") == 0 or die "Download failed";
-	system ("make -C linux/ untar") == 0 or die "Untar failed";
+	system ("make DESTDIR='$pkgdir' -C linux/ download") == 0 or die "Download failed";
+	system ("make DESTDIR='$pkgdir' -C linux/ untar") == 0 or die "Untar failed";
 }
 
 print "**********************************************************\n";
@@ -448,8 +450,8 @@
 print "* Start building *\n";
 print "******************\n";
 
-system ("make allyesconfig") == 0 or die "can't select all drivers";
-system ("make") == 0 or die "build failed";
+system ("make DESTDIR='$pkgdir' allyesconfig") == 0 or die "can't select all drivers";
+system ("make DESTDIR='$pkgdir'") == 0 or die "build failed";
 
 print "**********************************************************\n";
 print "* Compilation finished. Use 'make install' to install them\n";
