--- Makefile.in	2009-03-10 12:22:42.000000000 +0100
+++ Makefile.in.ok	2009-09-15 16:19:55.000000000 +0200
@@ -103,7 +103,7 @@
 	@INSTALL@ -m 755 -d ${DESTDIR}@prefix@/bin
 	@INSTALL@ -m 755 -d ${DEVDESTDIR}@prefix@/include/modglue
 	${IPHASE} libmodglue.la ${DESTDIR}@prefix@/lib
-	mv ${DESTDIR}@prefix@/lib/libmodglue*.so ${DEVDESTDIR}@prefix@/lib
+#	mv ${DESTDIR}@prefix@/lib/libmodglue*.so ${DEVDESTDIR}@prefix@/lib
 #	${IPHASE} libmid.la ${DESTDIR}@prefix@/lib
 	${FPHASE}
 	${IPHASE} ptywrap ${DESTDIR}@prefix@/bin
