diff -aur asis-gpl-2013-src.pristine/Makefile.stub asis-gpl-2013-src.new/Makefile.stub
--- asis-gpl-2013-src.pristine/Makefile.stub	2012-09-06 17:07:12.000000000 +0800
+++ asis-gpl-2013-src.new/Makefile.stub	2013-07-28 13:40:01.524005288 +0800
@@ -4,7 +4,7 @@
 # Where Asis library should be installed
 # Default is into current GNAT directory
 INSTALL_DIR = $(dir $(shell which gnatls))..
-prefix=$(INSTALL_DIR)
+prefix=$(DESTDIR)$(INSTALL_DIR)
 
 # Asis specific directories
 GNAT_DIR=../gnat
@@ -32,7 +32,7 @@
 AR_FLAGS = rc
 RANLIB = ranlib
 RANLIB_TEST = [ -f /usr/bin/ranlib -o -f /bin/ranlib ]
-GNATMAKE = gnatmake
+GNATMAKE = gnatmake -R
 GNATMAKE_FLAGS =
 
 # External Variable to pass to project file
