--- ngplant/SConscript  2012-12-03 22:08:15.000000000 +0100
+++ ngplant/SConscript.new      2012-12-07 19:43:52.022890220 +0100
@@ -52,6 +52,7 @@
 NGPlantEnv.Append(CPPPATH=NGPlantEnv['GLEXT_INC'])
 NGPlantEnv.Append(LIBPATH=NGPlantEnv['GLEXT_LIBPATH'])
 NGPlantEnv.Append(LIBS=NGPlantEnv['GLEXT_LIBS'])
+NGPlantEnv.Append(LIBS=['GL'])
 
 if TimingsEnabled:
     NGPlantEnv.Append(CPPDEFINES=[('P3D_TIMINGS_ENABLED','1')])

--- ngpshot/SConscript  2012-12-07 19:55:17.997118800 +0100
+++ ngpshot/SConscript.new      2012-12-07 19:55:27.813152771 +0100
@@ -32,6 +32,7 @@
 NGPShotEnv.Append(CPPPATH=NGPShotEnv['GLEXT_INC'])
 NGPShotEnv.Append(LIBPATH=NGPShotEnv['GLEXT_LIBPATH'])
 NGPShotEnv.Append(LIBS=NGPShotEnv['GLEXT_LIBS'])
+NGPShotEnv.Append(LIBS=['X11'])
 
 AppendLibPngConf(NGPShotEnv)
 AppendLibJpegConf(NGPShotEnv)

--- pywrapper/SConscript  2012-12-03 22:08:15.000000000 +0100
+++ pywrapper/SConscript.new      2012-12-07 20:20:13.367989729 +0100
@@ -82,7 +82,7 @@
                     source = SRCS,
                     PYEXT_NAME='_ngp',
                     PYEXT_VERSION='1.0',
-                    CPPPATH=['.','pywrapper'])
+                    CPPPATH=['.','pywrapper','/usr/include/python2.7'])
 
 env.Command('_ngp.so','setup.py',"python pywrapper/setup.py build --build-lib pywrapper --build-temp pywrapper/build")
 

