--- asterisk-1.6.2.0-beta1.old/codecs/gsm/Makefile	2009-05-06 06:29:32.646721595 +0800
+++ asterisk-1.6.2.0-beta1.new/codecs/gsm/Makefile	2009-05-06 07:23:25.423060878 +0800
@@ -30,7 +30,7 @@
 ######### define this, and read about the GSM_OPT_WAV49 option in the
 ######### manual page on gsm_option(3).
 
-#K6OPT	= -DK6OPT
+K6OPT	= -DK6OPT
 #K6OPT	=
 ######### Define to enable MMXTM optimizations for x86 architecture CPU's
 ######### which support MMX instructions.  This should be newer pentiums,
