--- asterisk-1.6.2.0-beta1.old/Makefile	2009-05-06 06:29:32.846721620 +0800
+++ asterisk-1.6.2.0-beta1.new/Makefile	2009-05-06 07:26:19.249712139 +0800
@@ -683,7 +683,7 @@
 	@if [ "$(OVERWRITE)" = "y" ] || [ ! -f $(DESTDIR)$(ASTCONFPATH) ]; then \
 		echo "Creating asterisk.conf"; \
 		( \
-		echo "[directories](!) ; remove the (!) to enable this" ; \
+		echo "[directories] ; remove the (!) to enable this" ; \
 		echo "astetcdir => $(ASTETCDIR)" ; \
 		echo "astmoddir => $(MODULES_DIR)" ; \
 		echo "astvarlibdir => $(ASTVARLIBDIR)" ; \
