diff --git a/sigrok-qt/sigrok-qt.pro b/sigrok-qt/sigrok-qt.pro
index c5ad0f3..ffb77a3 100644
--- a/sigrok-qt/sigrok-qt.pro
+++ b/sigrok-qt/sigrok-qt.pro
@@ -73,12 +73,12 @@ win32 {
 }
 
 # Installation
-PREFIX        = /usr/local
+PREFIX        = /usr
 target.path   = $$PREFIX/bin
 locale.path   = $$PREFIX/share/sigrok-qt/translations
 locale.extra  = lrelease sigrok-qt.pro
 locale.files  = locale/*.qm
-man.path      = $$PREFIX/man/man1
+man.path      = $$PREFIX/share/man/man1
 man.files     = doc/sigrok-qt.1
 INSTALLS     += target locale man
 
