--- findwine.old	2007-10-11 07:56:10.000000000 -0400
+++ findwine	2007-10-11 07:56:32.000000000 -0400
@@ -102,13 +102,13 @@
     mkdir /tmp/wine-$$
     touch /tmp/wine-$$/.no_debug_window_flag
     touch /tmp/wine-$$/.no_prelaunch_window_flag
-    WINEVER="`WINEPREFIX=/tmp/wine-$$ wine --version 2>&1 | grep "Wine [0-9]" | sed -e \"s/Wine //\"`"
+    WINEVER="`WINEPREFIX=/tmp/wine-$$ wine --version 2>&1 | grep "wine-[0-9]" | sed -e \"s/wine-//\"`"
     rm -Rf /tmp/wine-$$
   fi
 else
   # find out the Wine version if WINEVER is not set in environment
   echo -n "`eval_gettext "detecting Wine version... "`"
-  [ "$WINEVER" = "" ] && WINEVER="`wine --version 2>&1 | grep "Wine [0-9]" | sed -e \"s/Wine //\"`"
+  [ "$WINEVER" = "" ] && WINEVER="`wine --version 2>&1 | grep "wine-[0-9]" | sed -e \"s/wine-//\"`"
 
   echo "done."
 
