diff -Naur src/getfem-4.1.1/interface/src/python/setup.py src2/getfem-4.1.1/interface/src/python/setup.py
--- src/getfem-4.1.1/interface/src/python/setup.py	2011-08-23 05:45:22.855256113 -0400
+++ src2/getfem-4.1.1/interface/src/python/setup.py	2011-08-23 05:47:20.496960733 -0400
@@ -45,10 +45,18 @@
 #                                      '@GETFEM_STATICLIBS@'],
                       sources = ['getfem_python_c.c'])
 
-setup (name = 'getfem-interface',
-       license='LGPL',
-       version = '4.0',
-       maintainer="Yves Renard",
-       maintainer_email="Yves.Renard@insa-lyon.fr",
-       description = "This is the getfem-python-interface module",
-       ext_modules = [getfemmod])
+dist = setup (name = 'getfem-interface',
+              license='LGPL',
+              version = '4.0',
+              maintainer="Yves Renard",
+              maintainer_email="Yves.Renard@insa-lyon.fr",
+              description = "This is the getfem-python-interface module",
+              ext_modules = [getfemmod])
+
+os.remove('_getfem.so')
+try:
+    iter(dist.command_obj['build_ext'].compiler.linker_so)
+    dist.command_obj['build_ext'].compiler.linker_so = [i for i in dist.command_obj['build_ext'].compiler.linker_so if not '--as-needed' in i]
+except:
+    pass
+dist.command_obj['build_ext'].build_extensions()
