--- TeaCooker/CMakeLists.txt.org	2009-11-30 20:38:07.964877910 +0100
+++ TeaCooker/CMakeLists.txt	2009-11-30 20:38:59.807778748 +0100
@@ -6,7 +6,6 @@
 # Find the required Libaries
 find_package(KDE4 REQUIRED)
 include(KDE4Defaults)
-find_package(Plasma REQUIRED)
  
 add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 include_directories(
@@ -23,7 +22,7 @@
 kde4_add_ui_files(teacooker_SRCS)
 kde4_add_plugin(teacooker ${teacooker_SRCS})
 target_link_libraries(teacooker
-                      ${PLASMA_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_PHONON_LIBS})
+                      ${KDE4_PLASMA_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_PHONON_LIBS})
  
 install(TARGETS teacooker
         DESTINATION ${PLUGIN_INSTALL_DIR})
--- TeaCooker/plasma-teacooker.cpp.org	2009-11-30 20:50:08.534790285 +0100
+++ TeaCooker/plasma-teacooker.cpp	2009-11-30 20:50:26.267674399 +0100
@@ -44,7 +44,6 @@
 
 #include <plasma/svg.h>
 #include <plasma/theme.h>
-#include <plasma/widgets/icon.h>
 #include <plasma/containment.h>
 // #include <plasma/tooltipmanager.h>
 
