Index: client/gui_qt/WidgetDownloadQueue.cpp
===================================================================
--- client/gui_qt/WidgetDownloadQueue.cpp	(revision 1800)
+++ client/gui_qt/WidgetDownloadQueue.cpp	(working copy)
@@ -51,6 +51,14 @@
     ui->treeView->setContextMenuPolicy(Qt::CustomContextMenu);
     connect(ui->treeView, SIGNAL(customContextMenuRequested(const QPoint &)), SLOT(showContextMenu(const QPoint &)));
 
+#if QT_VERSION < 0x040600
+    /* Moving files requires Qt 4.6.0 or higher */
+    ui->btnQueueMoveToTop->hide();
+    ui->btnQueueMoveUp->hide();
+    ui->btnQueueMoveDown->hide();
+    ui->btnQueueMoveToBottom->hide();
+#endif
+
     /* Catch the DEL key */
     ui->treeView->installEventFilter(this);
 
@@ -159,7 +167,11 @@
         Q_ASSERT(collection_item != NULL);
 
         if (selectionModel->isSelected(idx_collection)) {
-            if (i == model->rowCount() - 1 && direction == MOVE_DIRECTION_DOWN) {
+            if (i == 0 && (direction == MOVE_DIRECTION_UP || direction == MOVE_DIRECTION_TOP)) {
+                /* We're already at the top, no need to move again */
+                continue;
+            }
+            if (i == model->rowCount() - 1 && (direction == MOVE_DIRECTION_DOWN || direction == MOVE_DIRECTION_BOTTOM)) {
                 /* We're already at the bottom, no need to move again */
                 continue;
             }
@@ -183,7 +195,12 @@
             QModelIndex idx_subject = model->index(subject_item->row(), 0, idx_collection);
 
             if (selectionModel->isSelected(idx_subject)) {
-                if (j == collection_item->childCount() - 1 && direction == MOVE_DIRECTION_DOWN) {
+                if (j == 0 && (direction == MOVE_DIRECTION_UP || direction == MOVE_DIRECTION_TOP)) {
+                    /* We're already at the top, no need to move again */
+                    continue;
+                }
+
+                if (j == collection_item->childCount() - 1 && (direction == MOVE_DIRECTION_DOWN || direction == MOVE_DIRECTION_BOTTOM)) {
                     /* We're already at the bottom, no need to move again */
                     continue;
                 }
Index: client/gui_qt/DownloadQueueModel.cpp
===================================================================
--- client/gui_qt/DownloadQueueModel.cpp	(revision 1800)
+++ client/gui_qt/DownloadQueueModel.cpp	(working copy)
@@ -522,6 +522,14 @@
     Q_UNUSED(subject);
     Q_UNUSED(new_collection_name);
 
+#if QT_VERSION < 0x040600
+    static bool beenhere = false;
+    if (!beenhere) {
+        beenhere = true;
+        qWarning("Moving files requires a more recent version of Qt. Please upgrade");
+        return;
+    }
+#else
     DownloadQueueItem *collection_node = lookupCollectionItem(orig_collection_name);
     DownloadQueueItem *subject_node = lookupSubjectItem(collection_node, subject);
 
@@ -548,6 +556,7 @@
     collection_node->insertChild(new_position, subject_node);
 
     endMoveRows();
+#endif
 }
 
 void DownloadQueueModel::onCollectionAdded(QString collection_name, QString poster)
@@ -609,6 +618,14 @@
 
 void DownloadQueueModel::onCollectionMoved(QString collection_name, int old_position, int new_position)
 {
+#if QT_VERSION < 0x040600
+    static bool beenhere = false;
+    if (!beenhere) {
+        beenhere = true;
+        qWarning("Moving collections requires a more recent version of Qt. Please upgrade");
+        return;
+    }
+#else
 #if 0
     bool was_expanded;
 #endif
@@ -641,6 +658,7 @@
         collection_node->setExpanded(true);
     }
 #endif
+#endif
 }
 
 void DownloadQueueModel::onTrafficMonitorUpdate(int bytes_received1, int bytes_received2, int bytes_received3, int bytes_received4, int bytes_received5, int bytes_received6, int bytes_received7, int bytes_received8, int bytes_received9, int bytes_received10, QDateTime stamp, double average)
