diff -aur src/fsl/extras/src/tk/library/listbox.tcl srcnew/fsl/extras/src/tk/library/listbox.tcl
--- src/fsl/extras/src/tk/library/listbox.tcl	2007-07-10 11:05:17.000000000 -0400
+++ srcnew/fsl/extras/src/tk/library/listbox.tcl	2012-07-02 17:28:20.229872177 -0400
@@ -178,24 +178,24 @@
 # The MouseWheel will typically only fire on Windows and Mac OS X.
 # However, someone could use the "event generate" command to produce
 # one on other platforms.
-if {[tk windowingsystem] eq "classic" || [tk windowingsystem] eq "aqua"} {
-    bind Listbox <MouseWheel> {
-        %W yview scroll [expr {- (%D)}] units
-    }
-    bind Listbox <Option-MouseWheel> {
-        %W yview scroll [expr {-10 * (%D)}] units
-    }
-    bind Listbox <Shift-MouseWheel> {
-        %W xview scroll [expr {- (%D)}] units
-    }
-    bind Listbox <Shift-Option-MouseWheel> {
-        %W xview scroll [expr {-10 * (%D)}] units
-    }
-} else {
-    bind Listbox <MouseWheel> {
-        %W yview scroll [expr {- (%D / 120) * 4}] units
-    }
-}
+#if {[tk windowingsystem] eq "classic" || [tk windowingsystem] eq "aqua"} {
+#    bind Listbox <MouseWheel> {
+#        %W yview scroll [expr {- (%D)}] units
+#    }
+#    bind Listbox <Option-MouseWheel> {
+#        %W yview scroll [expr {-10 * (%D)}] units
+#    }
+#    bind Listbox <Shift-MouseWheel> {
+#        %W xview scroll [expr {- (%D)}] units
+#    }
+#    bind Listbox <Shift-Option-MouseWheel> {
+#        %W xview scroll [expr {-10 * (%D)}] units
+#    }
+#} else {
+#    bind Listbox <MouseWheel> {
+#        %W yview scroll [expr {- (%D / 120) * 4}] units
+#    }
+#}
 
 if {"x11" eq [tk windowingsystem]} {
     # Support for mousewheels on Linux/Unix commonly comes through mapping
diff -aur src/fsl/extras/src/tk/library/text.tcl srcnew/fsl/extras/src/tk/library/text.tcl
--- src/fsl/extras/src/tk/library/text.tcl	2007-07-10 11:05:17.000000000 -0400
+++ srcnew/fsl/extras/src/tk/library/text.tcl	2012-07-02 17:29:11.044734134 -0400
@@ -454,24 +454,24 @@
 # However, someone could use the "event generate" command to produce
 # one on other platforms.
 
-if {[tk windowingsystem] eq "classic" || [tk windowingsystem] eq "aqua"} {
-    bind Text <MouseWheel> {
-        %W yview scroll [expr {- (%D)}] units
-    }
-    bind Text <Option-MouseWheel> {
-        %W yview scroll [expr {-10 * (%D)}] units
-    }
-    bind Text <Shift-MouseWheel> {
-        %W xview scroll [expr {- (%D)}] units
-    }
-    bind Text <Shift-Option-MouseWheel> {
-        %W xview scroll [expr {-10 * (%D)}] units
-    }
-} else {
-    bind Text <MouseWheel> {
-        %W yview scroll [expr {- (%D / 120) * 4}] units
-    }
-}
+#if {[tk windowingsystem] eq "classic" || [tk windowingsystem] eq "aqua"} {
+#    bind Text <MouseWheel> {
+#        %W yview scroll [expr {- (%D)}] units
+#    }
+#    bind Text <Option-MouseWheel> {
+#        %W yview scroll [expr {-10 * (%D)}] units
+#    }
+#    bind Text <Shift-MouseWheel> {
+#        %W xview scroll [expr {- (%D)}] units
+#    }
+#    bind Text <Shift-Option-MouseWheel> {
+#        %W xview scroll [expr {-10 * (%D)}] units
+#    }
+#} else {
+#    bind Text <MouseWheel> {
+#        %W yview scroll [expr {- (%D / 120) * 4}] units
+#    }
+#}
 
 if {"x11" eq [tk windowingsystem]} {
     # Support for mousewheels on Linux/Unix commonly comes through mapping
