diff -u -r vmblock-only/linux/filesystem.c vmblock-only/linux/filesystem.c
--- vmblock-only/linux/filesystem.c	2012-01-18 22:01:07.000000000 +0100
+++ vmblock-only/linux/filesystem.c	2012-04-01 11:48:24.000000000 +0200
@@ -525,7 +525,11 @@
       return -EINVAL;
    }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 4, 0)
    rootDentry = d_alloc_root(rootInode);
+#else
+   rootDentry = d_make_root(rootInode);
+#endif
    if (!rootDentry) {
       iput(rootInode);
       return -ENOMEM;
