diff -ru vmci-only.orig/include/compat_semaphore.h vmci-only/include/compat_semaphore.h
--- vmci-only.orig/include/compat_semaphore.h	2011-01-11 21:55:57.016999997 +0100
+++ vmci-only/include/compat_semaphore.h	2011-01-11 21:57:39.719999997 +0100
@@ -45,5 +45,13 @@
    #endif
 #endif
 
+#if (defined CONFIG_PREEMPT_RT && LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31)) || LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)
+   #ifndef DECLARE_MUTEX
+      #define DECLARE_MUTEX(_m)  DEFINE_SEMAPHORE(_m)
+   #endif
+   #ifndef init_MUTEX
+      #define init_MUTEX(_m) sema_init(_m,1)
+   #endif
+#endif
 
 #endif /* __COMPAT_SEMAPHORE_H__ */
