diff --git vmmon-only/include/compat_pci.h vmmon-only/include/compat_pci.h
index fc88ae8..86e31c7 100644
--- a/include/compat_pci.h
+++ b/include/compat_pci.h
@@ -600,4 +600,9 @@ pci_get_drvdata(struct pci_dev *pdev)
        device_set_wakeup_enable(dev, val)
 #endif
 
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 34)
+#define iommu_map_range iommu_map
+#define iommu_unmap_range iommu_unmap
+#endif
+
 #endif /* __COMPAT_PCI_H__ */
