diff -u -r vsock-only/linux/af_vsock.c vsock-only/linux/af_vsock.c
--- vsock-only/linux/af_vsock.c	2011-03-26 04:03:08.000000000 +0100
+++ vsock-only/linux/af_vsock.c	2011-04-02 13:33:39.000000000 +0200
@@ -102,7 +102,6 @@
 #include <linux/miscdevice.h>
 #include <linux/poll.h>
 #include <linux/smp.h>
-#include <linux/smp_lock.h>
 #include <asm/io.h>
 #if defined(__x86_64__) && LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 12)
 #   if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 0)
diff -u -r vsock-only/linux/util.c vsock-only/linux/util.c
--- vsock-only/linux/util.c	2011-03-26 04:03:08.000000000 +0100
+++ vsock-only/linux/util.c	2011-04-02 13:35:45.000000000 +0200
@@ -34,7 +34,7 @@
 struct list_head vsockBindTable[VSOCK_HASH_SIZE + 1];
 struct list_head vsockConnectedTable[VSOCK_HASH_SIZE];
 
-spinlock_t vsockTableLock = SPIN_LOCK_UNLOCKED;
+DEFINE_SPINLOCK(vsockTableLock);
 
 /*
  * snprintf() wasn't exported until 2.4.10: fall back on sprintf in those
