diff -Naur a/gui/core.in b/gui/core.in
--- a/gui/core.in	2012-09-21 12:25:46.000000000 -0400
+++ b/gui/core.in	2013-01-09 10:46:46.261145265 -0500
@@ -70,6 +70,7 @@
 # locate wish8.5 or wish8.4 binaries
 WISHLIST="/usr/local/bin/wish8.5 /usr/bin/wish8.5"
 WISHLIST="$WISHLIST /usr/local/bin/wish8.4 /usr/bin/wish8.4"
+WISHLIST="$WISHLIST /usr/local/bin/wish /usr/bin/wish"
 for wishbin in $WISHLIST
 do
     if [ -x $wishbin ]
