--- CMakeLists.txt.old	2009-12-21 15:02:41.000000000 +0100
+++ CMakeLists.txt.new	2009-12-21 15:02:47.000000000 +0100
@@ -198,7 +198,7 @@ add_definitions(
 
 if(UNIX)
 	add_definitions(-DPOSIX)
-	set(CONF_INST_DIR bin)
+	set(CONF_INST_DIR /etc)
 endif(UNIX)
 
 if(USE_BOEHM_GC)
@@ -268,7 +268,7 @@ get_target_property(LDC_LOC ${LDC_EXE} L
 # TODO: testrun
 
 install(TARGETS ${LDC_EXE} DESTINATION bin)
-install(FILES ${PROJECT_SOURCE_DIR}/bin/${LDMD_EXE} DESTINATION bin)
+install(PROGRAMS ${PROJECT_SOURCE_DIR}/bin/${LDMD_EXE} DESTINATION bin)
 install(FILES ${PROJECT_BINARY_DIR}/bin/${LDC_EXE}.conf DESTINATION ${CONF_INST_DIR})
 install(FILES ${PROJECT_BINARY_DIR}/bin/${LDC_EXE}.rebuild.conf DESTINATION ${CONF_INST_DIR})
 install(DIRECTORY ${PROJECT_BINARY_DIR}/lib DESTINATION . USE_SOURCE_PERMISSIONS)
