diff -wbBur boss/include/boss.h boss.my/include/boss.h
--- boss/include/boss.h	2007-04-01 09:53:25.000000000 +0400
+++ boss.my/include/boss.h	2011-12-12 15:13:00.000000000 +0400
@@ -431,8 +431,8 @@
 
 // TO DO: Move these to boss.cpp when done testing
 #include <SDL.h>
-#include <SDL_TTF.h>
-#include <SDL_Mixer.h>
+#include <SDL_ttf.h>
+#include <SDL_mixer.h>
 
 /// Boss Namespace
 namespace Boss {
diff -wbBur boss/src/boss.cpp boss.my/src/boss.cpp
--- boss/src/boss.cpp	2007-04-01 09:53:25.000000000 +0400
+++ boss.my/src/boss.cpp	2011-12-12 15:13:12.000000000 +0400
@@ -1,7 +1,7 @@
 #include "boss.h"
 #include <SDL.h>
-#include <SDL_TTF.h>
-#include <SDL_Mixer.h>
+#include <SDL_ttf.h>
+#include <SDL_mixer.h>
 
 #include <assert.h>
 #include <math.h>
Only in boss.my/src: boss.o
