--- configure.ac~	2011-01-20 16:25:45.612430798 +0000
+++ configure.ac	2011-01-20 16:29:14.777316928 +0000
@@ -86,7 +86,7 @@
 fi
 AM_CONDITIONAL(HAVE_QUEUE_CONFIG, test "x$enable_queue_config" = "xyes")
 
-PKG_CHECK_MODULES(BOGNOR, gtk+-2.0 glib-2.0 dbus-glib-1 gio-2.0 libnotify tracker-client-0.9 gupnp-1.0 >= 0.13.0 gupnp-av-1.0 >= 0.5)
+PKG_CHECK_MODULES(BOGNOR, gtk+-2.0 glib-2.0 dbus-glib-1 gio-2.0 libnotify tracker-client-0.10 gupnp-1.0 >= 0.13.0 gupnp-av-1.0 >= 0.5)
 PKG_CHECK_MODULES(BOGNOR_REGIS, glib-2.0 dbus-glib-1 $gconf_pkg)
 PKG_CHECK_MODULES(CLIENT, gio-2.0)
 
