diff -ur recipe-0.1.7/terra/VteObject.py recipe-0.1.7-new/terra/VteObject.py
--- recipe-0.1.7/terra/VteObject.py	2013-03-04 12:11:28.000000000 -0300
+++ recipe-0.1.7-new/terra/VteObject.py	2013-09-24 20:45:32.885314050 -0300
@@ -149,11 +149,9 @@
 
         self.vte.set_scroll_on_keystroke(ConfigManager.get_conf('scroll-on-keystroke'))
 
-        self.vte.set_background_saturation(ConfigManager.get_conf('transparency') / 100.0)
 
         self.vte.set_opacity(int((100 - ConfigManager.get_conf(('transparency'))) / 100.0 * 65535))
 
-        self.vte.set_background_transparent(ConfigManager.use_fake_transparency)
 
         self.vte.set_word_chars(ConfigManager.get_conf('select-by-word'))
 
@@ -162,8 +160,6 @@
             Gdk.color_parse(ConfigManager.get_conf('color-background')),
             [])
 
-        self.vte.set_background_image_file(
-            ConfigManager.get_conf('background-image'))
 
         if not ConfigManager.get_conf('use-default-font'):
             self.vte.set_font_from_string(ConfigManager.get_conf('font-name'))
@@ -460,4 +456,4 @@
         self.vte.copy_clipboard()
 
     def paste_clipboard(self):
-        self.vte.paste_clipboard()
\ No newline at end of file
+        self.vte.paste_clipboard()
