--- ../apertium-3.2.0/configure.ac	2010-02-28 18:24:47.000000000 +0100
+++ apertium-3.2.0/configure.ac	2010-09-21 15:16:03.234072111 +0200
@@ -184,10 +184,13 @@
 if test x$(uname) != xDarwin;
 then 
 AC_CHECK_HEADER(pcreposix.h,
-  AC_CHECK_LIB(pcre, pcre_info,[
-  LIBS="$LIBS -lpcreposix -lpcre"
-  no_comp_check=yes],
-  AC_MSG_ERROR([*** unable to locate pcre library ***])),
+  AC_CHECK_LIB(pcre, pcre_fullinfo,[
+    LIBS="$LIBS -lpcreposix -lpcre"
+    no_comp_check=yes],
+    AC_CHECK_LIB(pcre, pcre_info,[ dnl info is deprecated in favor of fullinfo, but leave it in case old versions don't have fullinfo
+      LIBS="$LIBS -lpcreposix -lpcre"
+      no_comp_check=yes],
+      AC_MSG_ERROR([*** unable to locate pcre library ***]))),
   AC_MSG_ERROR([*** unable to locate pcreposix.h include file ***]))
 
 AC_CHECK_HEADER(pcrecpp.h,
