--- pspp-0.6.2/lib/gtksheet/gtkitementry.c 2007-06-23 04:17:47.000000000 +0200
+++ pspp-0.6.2/lib/gtksheet/gtkitementry.c.new 2010-06-20 00:02:54.000000000 +0200
@@ -131,7 +131,6 @@
 static PangoLayout *gtk_entry_ensure_layout            (GtkEntry       *entry,
                                                         gboolean        include_preedit);
 static void         gtk_entry_queue_draw               (GtkEntry       *entry);
-static void         gtk_entry_reset_im_context         (GtkEntry       *entry);
 static void         gtk_entry_recompute                (GtkEntry       *entry);
 static void         gtk_entry_get_cursor_locations     (GtkEntry       *entry,
  CursorType      type,
@@ -1695,16 +1694,6 @@
 }
 
 static void
-gtk_entry_reset_im_context (GtkEntry *entry)
-{
-  if (entry->need_im_reset)
-    {
-      entry->need_im_reset = 0;
-      gtk_im_context_reset (entry->im_context);
-    }
-}
-
-static void
 gtk_entry_get_cursor_locations (GtkEntry   *entry,
  CursorType  type,
  gint       *strong_x, 