diff -crB ubuntu-sso-client-1.3.2//bin/ubuntu-sso-login ubuntu-sso-client-1.3.2-python2//bin/ubuntu-sso-login
*** ubuntu-sso-client-1.3.2//bin/ubuntu-sso-login	2011-07-22 23:39:10.000000000 +0400
--- ubuntu-sso-client-1.3.2-python2//bin/ubuntu-sso-login	2011-07-26 22:07:52.149560749 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  
  # ubuntu-sso-login - Client side log-in utility for Ubuntu One
  #
--- 1,4 ----
! #!/usr/bin/env python2
  
  # ubuntu-sso-login - Client side log-in utility for Ubuntu One
  #
diff -crB ubuntu-sso-client-1.3.2//bin/windows-ubuntu-sso-login ubuntu-sso-client-1.3.2-python2//bin/windows-ubuntu-sso-login
*** ubuntu-sso-client-1.3.2//bin/windows-ubuntu-sso-login	2011-07-22 23:39:10.000000000 +0400
--- ubuntu-sso-client-1.3.2-python2//bin/windows-ubuntu-sso-login	2011-07-26 22:07:42.079578790 +0400
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  # Authors: Manuel de la Pena <manuel@canonical.com>
  #          Alejandro J. Cura <alecu@canonical.com>
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  # Authors: Manuel de la Pena <manuel@canonical.com>
  #          Alejandro J. Cura <alecu@canonical.com>
diff -crB ubuntu-sso-client-1.3.2//setup.py ubuntu-sso-client-1.3.2-python2//setup.py
*** ubuntu-sso-client-1.3.2//setup.py	2011-07-22 23:39:24.000000000 +0400
--- ubuntu-sso-client-1.3.2-python2//setup.py	2011-07-26 22:07:14.876294137 +0400
***************
*** 1,4 ****
! #!/usr/bin/env python
  # setup.py - Build system for Ubuntu SSO Client package
  #
  # Authors: Natalia B. Bidart <natalia.bidart@canonical.com>
--- 1,4 ----
! #!/usr/bin/env python2
  # setup.py - Build system for Ubuntu SSO Client package
  #
  # Authors: Natalia B. Bidart <natalia.bidart@canonical.com>
diff -crB ubuntu-sso-client-1.3.2//ubuntu_sso/tests/bin/show_gui ubuntu-sso-client-1.3.2-python2//ubuntu_sso/tests/bin/show_gui
*** ubuntu-sso-client-1.3.2//ubuntu_sso/tests/bin/show_gui	2011-07-22 23:39:10.000000000 +0400
--- ubuntu-sso-client-1.3.2-python2//ubuntu_sso/tests/bin/show_gui	2011-07-26 22:08:13.686188795 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  
  # ubuntu-sso-login-gui - GUI for registration and login for Ubuntu SSO
  #
--- 1,4 ----
! #!/usr/bin/env python2
  
  # ubuntu-sso-login-gui - GUI for registration and login for Ubuntu SSO
  #
diff -crB ubuntu-sso-client-1.3.2//ubuntu_sso/tests/bin/show_nm_state ubuntu-sso-client-1.3.2-python2//ubuntu_sso/tests/bin/show_nm_state
*** ubuntu-sso-client-1.3.2//ubuntu_sso/tests/bin/show_nm_state	2011-07-22 23:39:10.000000000 +0400
--- ubuntu-sso-client-1.3.2-python2//ubuntu_sso/tests/bin/show_nm_state	2011-07-26 22:08:22.209506829 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  
  # show_nm_state - Show the state of the NetworkManager daemon
  #
--- 1,4 ----
! #!/usr/bin/env python2
  
  # show_nm_state - Show the state of the NetworkManager daemon
  #
