diff -crB ubuntu-sso-client-2.99.1/bin/ubuntu-sso-login ubuntu-sso-client-2.99.1-python2/bin/ubuntu-sso-login
*** ubuntu-sso-client-2.99.1/bin/ubuntu-sso-login	2012-01-03 17:10:01.000000000 +0400
--- ubuntu-sso-client-2.99.1-python2/bin/ubuntu-sso-login	2012-01-07 22:21:48.140071085 +0400
***************
*** 1,4 ****
! #!/usr/bin/env python
  # -*- coding: utf-8 -*-
  #
  # Copyright 2009-2010 Canonical Ltd.
--- 1,4 ----
! #!/usr/bin/env python2
  # -*- coding: utf-8 -*-
  #
  # Copyright 2009-2010 Canonical Ltd.
diff -crB ubuntu-sso-client-2.99.1/setup.py ubuntu-sso-client-2.99.1-python2/setup.py
*** ubuntu-sso-client-2.99.1/setup.py	2012-01-03 22:04:03.000000000 +0400
--- ubuntu-sso-client-2.99.1-python2/setup.py	2012-01-07 22:23:47.292955238 +0400
***************
*** 1,4 ****
! #!/usr/bin/env python
  # setup.py - Build system for Ubuntu SSO Client package
  #
  # Authors: Natalia B. Bidart <natalia.bidart@canonical.com>
--- 1,4 ----
! #!/usr/bin/env python2
  # setup.py - Build system for Ubuntu SSO Client package
  #
  # Authors: Natalia B. Bidart <natalia.bidart@canonical.com>
diff -crB ubuntu-sso-client-2.99.1/ubuntu_sso/tests/bin/show_gui ubuntu-sso-client-2.99.1-python2/ubuntu_sso/tests/bin/show_gui
*** ubuntu-sso-client-2.99.1/ubuntu_sso/tests/bin/show_gui	2011-12-19 23:00:28.000000000 +0400
--- ubuntu-sso-client-2.99.1-python2/ubuntu_sso/tests/bin/show_gui	2012-01-07 22:22:09.699990871 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  
  # ubuntu-sso-login-gui - GUI for registration and login for Ubuntu SSO
  #
--- 1,4 ----
! #!/usr/bin/env python2
  
  # ubuntu-sso-login-gui - GUI for registration and login for Ubuntu SSO
  #
diff -crB ubuntu-sso-client-2.99.1/ubuntu_sso/tests/bin/show_nm_state ubuntu-sso-client-2.99.1-python2/ubuntu_sso/tests/bin/show_nm_state
*** ubuntu-sso-client-2.99.1/ubuntu_sso/tests/bin/show_nm_state	2011-12-19 23:00:28.000000000 +0400
--- ubuntu-sso-client-2.99.1-python2/ubuntu_sso/tests/bin/show_nm_state	2012-01-07 22:22:18.266625532 +0400
***************
*** 1,4 ****
! #!/usr/bin/python
  
  # show_nm_state - Show the state of the NetworkManager daemon
  #
--- 1,4 ----
! #!/usr/bin/env python2
  
  # show_nm_state - Show the state of the NetworkManager daemon
  #
