diff --git a/src/voglperfrun.cpp b/src/voglperfrun.cpp
index fcedfc2..e225350 100644
--- a/src/voglperfrun.cpp
+++ b/src/voglperfrun.cpp
@@ -469,8 +469,8 @@ static void init_logfile(launch_data_t &ld)
 static void init_LD_PRELOAD(launch_data_t &ld)
 { 
     // set up LD_PRELOAD string
-    std::string voglperf32 = get_full_path("./libvoglperf32.so");
-    std::string voglperf64 = get_full_path("./libvoglperf64.so");
+    std::string voglperf32 = get_full_path("../lib32/libvoglperf32.so");
+    std::string voglperf64 = get_full_path("../lib/libvoglperf64.so");
 
     ld.LD_PRELOAD = "LD_PRELOAD=";
 
