--- soplex-1.7.0/doc/soplex.dxy.orig	2012-08-01 10:54:57.000000000 +0200
+++ soplex-1.7.0/doc/soplex.dxy	2012-08-01 10:50:53.000000000 +0200
@@ -893,7 +893,7 @@ HTML_HEADER            = soplexheader.ht
 # each generated HTML page. If it is left blank doxygen will generate a
 # standard footer.
 
-HTML_FOOTER            = soplexfooter.html
+HTML_FOOTER            = 
 
 # The HTML_STYLESHEET tag can be used to specify a user-defined cascading
 # style sheet that is used by each HTML page. It can be used to
