--- libjacksms/netclientqhttp.cpp	2011-07-04 01:17:05.000000000 +0200
+++ libjacksms/netclientqhttp.cpp	2012-05-06 00:36:14.510707159 +0200
@@ -190,6 +190,7 @@
         QString netClientQHttp::readPage(bool _ret){
 
             QNetworkRequest r(url);
+            r.setRawHeader("Content-Type", "application/x-www-form-urlencoded");
             QSslConfiguration config = r.sslConfiguration();
             config.setPeerVerifyMode(QSslSocket::VerifyNone);
             r.setSslConfiguration(config);
@@ -257,6 +258,7 @@
         {
             url = QUrl(QString::fromUtf8(_url.toStdString().c_str(),_url.length()));
             QNetworkRequest r(url);
+            r.setRawHeader("Content-Type", "application/x-www-form-urlencoded");
             QSslConfiguration config = r.sslConfiguration();
             config.setPeerVerifyMode(QSslSocket::VerifyNone);
             r.setSslConfiguration(config);
@@ -339,6 +341,7 @@
             url.setQueryItems(fields);
 
             QNetworkRequest r(url);
+            r.setRawHeader("Content-Type", "application/x-www-form-urlencoded");
             QSslConfiguration config = r.sslConfiguration();
             config.setPeerVerifyMode(QSslSocket::VerifyNone);
             r.setSslConfiguration(config);
