--- thumbnail-checker.py	2008-08-21 15:32:51.263109070 +0200
+++ thumbnail-checker.after.py	2008-08-21 15:33:56.263345762 +0200
@@ -18,7 +18,7 @@
 		self.id = None
 		self.first_time = True
 
-		xml = gtk.glade.XML('thumbnail-checker.glade', None, None)
+		xml = gtk.glade.XML('/usr/share/thumbnail-checker/thumbnail-checker.glade', None, None)
 		self.model = gtk.TreeStore(str, str, str, int)
 
 		self.window = xml.get_widget('window')
