*** pinject/pinject/object_providers.py	2013-08-02 00:03:50.579354099 +0200
--- ../pinject/pinject/object_providers.py	2013-08-02 00:06:14.976010850 +0200
***************
*** 49,55 ****
          return provided
  
      def provide_class(self, cls, injection_context):
!         if type(cls.__init__) is types.MethodType:
              init_kwargs = self.get_injection_kwargs(
                  cls.__init__, injection_context)
          else:
--- 49,55 ----
          return provided
  
      def provide_class(self, cls, injection_context):
!         if type(cls.__init__) is types.MethodType or type(cls.__init__) is types.FunctionType:
              init_kwargs = self.get_injection_kwargs(
                  cls.__init__, injection_context)
          else:
