diff -Naur Healpix_2.12a/src/cxx/Healpix_cxx/planck.make Healpix_2.12a_patched/src/cxx/Healpix_cxx/planck.make
--- Healpix_2.12a/src/cxx/Healpix_cxx/planck.make	2006-12-04 20:31:45.000000000 +0100
+++ Healpix_2.12a_patched/src/cxx/Healpix_cxx/planck.make	2009-08-17 22:55:59.508994368 +0200
@@ -44,42 +44,42 @@
 libhealpix_cxx.a: $(HEALPIX_OBJ) $(HEADERS)
 	$(ARCREATE) libhealpix_cxx.a $(HEALPIX_OBJ)
 
-syn_alm_cxx.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+syn_alm_cxx.o: libhealpix_cxx.a libcxxsupport.a 
 syn_alm_cxx: syn_alm_cxx.o
 	$(CXXL) $(CXXLFLAGS) -o $@ syn_alm_cxx.o -lhealpix_cxx \
 	  -lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-alm2map_cxx.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a libfftpack.a
+alm2map_cxx.o: libhealpix_cxx.a libcxxsupport.a  libfftpack.a
 alm2map_cxx: alm2map_cxx.o
 	$(CXXL) $(CXXLFLAGS) -o $@ alm2map_cxx.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-anafast_cxx.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a libfftpack.a
+anafast_cxx.o: libhealpix_cxx.a libcxxsupport.a  libfftpack.a
 anafast_cxx: anafast_cxx.o
 	$(CXXL) $(CXXLFLAGS) -o $@ anafast_cxx.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-map2tga.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+map2tga.o: libhealpix_cxx.a libcxxsupport.a 
 map2tga: map2tga.o
 	$(CXXL) $(CXXLFLAGS) -o $@ map2tga.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-udgrade_cxx.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+udgrade_cxx.o: libhealpix_cxx.a libcxxsupport.a 
 udgrade_cxx: udgrade_cxx.o
 	$(CXXL) $(CXXLFLAGS) -o $@ udgrade_cxx.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-hotspots_cxx.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+hotspots_cxx.o: libhealpix_cxx.a libcxxsupport.a 
 hotspots_cxx: hotspots_cxx.o
 	$(CXXL) $(CXXLFLAGS) -o $@ hotspots_cxx.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-calc_powspec.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+calc_powspec.o: libhealpix_cxx.a libcxxsupport.a 
 calc_powspec: calc_powspec.o
 	$(CXXL) $(CXXLFLAGS) -o $@ calc_powspec.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-median_filter.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+median_filter.o: libhealpix_cxx.a libcxxsupport.a 
 median_filter: median_filter.o
 	$(CXXL) $(CXXLFLAGS) -o $@ median_filter.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
@@ -89,7 +89,7 @@
 	$(CXXL) $(CXXLFLAGS) -o $@ hpxtest.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-smoothing_cxx.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a libfftpack.a
+smoothing_cxx.o: libhealpix_cxx.a libcxxsupport.a  libfftpack.a
 smoothing_cxx: smoothing_cxx.o
 	$(CXXL) $(CXXLFLAGS) -o $@ smoothing_cxx.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
@@ -99,7 +99,7 @@
 	$(CXXL) $(CXXLFLAGS) -o $@ rotalm_cxx.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-mult_alm.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+mult_alm.o: libhealpix_cxx.a libcxxsupport.a 
 mult_alm: mult_alm.o
 	$(CXXL) $(CXXLFLAGS) -o $@ mult_alm.o -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
diff -Naur Healpix_2.12a/src/cxx/Makefile Healpix_2.12a_patched/src/cxx/Makefile
--- Healpix_2.12a/src/cxx/Makefile	2006-01-31 11:31:18.000000000 +0100
+++ Healpix_2.12a_patched/src/cxx/Makefile	2009-08-17 22:56:41.483724397 +0200
@@ -4,7 +4,7 @@
 
 ACTIVE_SUBDIRS := Healpix_cxx alice
 
-SUBDIRS	:= $(ACTIVE_SUBDIRS) libcfitsio cxxsupport libfftpack
+SUBDIRS	:= $(ACTIVE_SUBDIRS) cxxsupport libfftpack
 
 LEVELS_SRC:=$(shell pwd)
 export LEVELS_SRC
@@ -14,7 +14,7 @@
 
 include $(PARAMFILE)
 
-cxxsupport.all: libcfitsio.all
+cxxsupport.all: 
 Healpix_cxx.all: cxxsupport.all libfftpack.all
 alice.all: Healpix_cxx.all
 
diff -Naur Healpix_2.12a/src/cxx/alice/planck.make Healpix_2.12a_patched/src/cxx/alice/planck.make
--- Healpix_2.12a/src/cxx/alice/planck.make	2009-01-08 17:19:58.000000000 +0100
+++ Healpix_2.12a_patched/src/cxx/alice/planck.make	2009-08-17 22:53:46.828171615 +0200
@@ -20,28 +20,28 @@
 alice2.o: alice_usage.h OrthogonalSkyMap.h
 testOrthogonalSkyMap.o: OrthogonalSkyMap.h
 
-generateTexture.o: libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+generateTexture.o: libhealpix_cxx.a libcxxsupport.a 
 generateTexture: generateTexture.o
 	$(CXXL) $(CXXLFLAGS) -o $@ generateTexture.o  -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-$(ALICE2_OBJ): libhealpix_cxx.a libcxxsupport.a libcfitsio.a
-alice2: $(ALICE2_OBJ) libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+$(ALICE2_OBJ): libhealpix_cxx.a libcxxsupport.a 
+alice2: $(ALICE2_OBJ) libhealpix_cxx.a libcxxsupport.a 
 	$(CXXL) $(CXXLFLAGS) -o $@ $(ALICE2_OBJ) -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)
 
-alice_test: $(TEST_OBJ) libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+alice_test: $(TEST_OBJ) libhealpix_cxx.a libcxxsupport.a 
 	$(CXXL) $(CXXLFLAGS) -o $@ $(TEST_OBJ) -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)	
 
-testSoSSkyMap: $(TESTSOS_OBJ) libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+testSoSSkyMap: $(TESTSOS_OBJ) libhealpix_cxx.a libcxxsupport.a 
 	$(CXXL) $(CXXLFLAGS) -o $@ $(TESTSOS_OBJ) -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)		
 
-testMollweideSkyMap: $(TESTMOL_OBJ) libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+testMollweideSkyMap: $(TESTMOL_OBJ) libhealpix_cxx.a libcxxsupport.a 
 	$(CXXL) $(CXXLFLAGS) -o $@ $(TESTMOL_OBJ) -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)	
 	
-testOrthogonalSkyMap: $(TESTORTH_OBJ) OrthogonalSkyMap.h  libhealpix_cxx.a libcxxsupport.a libcfitsio.a
+testOrthogonalSkyMap: $(TESTORTH_OBJ) OrthogonalSkyMap.h  libhealpix_cxx.a libcxxsupport.a 
 	$(CXXL) $(CXXLFLAGS) -o $@ $(TESTORTH_OBJ) -lhealpix_cxx \
 	-lcxxsupport -lcfitsio -lfftpack $(CXX_EXTRALIBS)		
diff -Naur Healpix_2.12a/src/cxx/alice/testMollweideSkyMap.cc Healpix_2.12a_patched/src/cxx/alice/testMollweideSkyMap.cc
--- Healpix_2.12a/src/cxx/alice/testMollweideSkyMap.cc	2007-05-01 19:29:27.000000000 +0200
+++ Healpix_2.12a_patched/src/cxx/alice/testMollweideSkyMap.cc	2009-08-17 22:54:14.428168716 +0200
@@ -1,3 +1,4 @@
+#include <cstdio>
 #include <iostream>
 #include <assert.h>
 #include <math.h>
diff -Naur Healpix_2.12a/src/cxx/alice/testOrthogonalSkyMap.cc Healpix_2.12a_patched/src/cxx/alice/testOrthogonalSkyMap.cc
--- Healpix_2.12a/src/cxx/alice/testOrthogonalSkyMap.cc	2007-05-01 19:29:27.000000000 +0200
+++ Healpix_2.12a_patched/src/cxx/alice/testOrthogonalSkyMap.cc	2009-08-17 22:54:31.492334434 +0200
@@ -1,3 +1,4 @@
+#include <cstdio>
 #include <iostream>
 #include <assert.h>
 #include <math.h>
diff -Naur Healpix_2.12a/src/cxx/config/config.generic_gcc_pic Healpix_2.12a_patched/src/cxx/config/config.generic_gcc_pic
--- Healpix_2.12a/src/cxx/config/config.generic_gcc_pic	1970-01-01 01:00:00.000000000 +0100
+++ Healpix_2.12a_patched/src/cxx/config/config.generic_gcc_pic	2009-08-17 23:16:11.751368511 +0200
@@ -0,0 +1,17 @@
+CC= gcc
+CCFLAGS_NO_C= -W -Wall -I$(INCDIR) -O2 -g0 -fno-strict-aliasing -fomit-frame-pointer -ffast-math
+CCFLAGS= $(CCFLAGS_NO_C) -fPIC -c
+
+CXX= g++
+CXXL= g++
+CXXWARNFLAGS= -W -Wall -Wshadow -Wwrite-strings -Wredundant-decls -Woverloaded-virtual -Wcast-qual -Wcast-align -Wpointer-arith -Wconversion -Wold-style-cast -Wno-unknown-pragmas
+CXXCFLAGS_NO_C= $(CXXWARNFLAGS) -ansi -I$(INCDIR) -O2 -g0 -ffast-math -fomit-frame-pointer
+CXXCFLAGS= $(CXXCFLAGS_NO_C) -fPIC -c
+CXXLFLAGS= -L. -L$(LIBDIR) -ffast-math
+ifeq ($(shell uname),SunOS)
+CXX_EXTRALIBS= -lnsl -lsocket
+endif
+
+ARCREATE= ar crv
+
+include $(LEVELS_SRC)/config/rules.common
diff -Naur Healpix_2.12a/src/cxx/cxxsupport/planck.make Healpix_2.12a_patched/src/cxx/cxxsupport/planck.make
--- Healpix_2.12a/src/cxx/cxxsupport/planck.make	2008-02-19 11:58:45.000000000 +0100
+++ Healpix_2.12a_patched/src/cxx/cxxsupport/planck.make	2009-08-17 22:55:16.301497835 +0200
@@ -20,7 +20,7 @@
 SIMPARAMS_H := simparams.h $(CXXUTILS_H)
 PARAMFILE_H := paramfile.h $(SIMPARAMS_H) $(CXXUTILS_H)
 ARR_H := arr.h $(CXXUTILS_H)
-FITSHANDLE_H := fitshandle.h fitsio.h $(ARR_H) $(DATATYPES_H)
+FITSHANDLE_H := fitshandle.h $(ARR_H) $(DATATYPES_H)
 VEC3_H := vec3.h
 ROTMATRIX_H := rotmatrix.h $(CXXUTILS_H) $(VEC3_H)
 POINTING_H := pointing.h $(VEC3_H) $(CXXUTILS_H)
