--- a/tests/test_dateparse.py
+++ b/tests/test_dateparse.py
@@ -280,10 +280,11 @@
                            dict(year=5000, month=6),
                            dict(year=5001, month=1))
     
-    # oct 2010 to feb -> oct 1 2010 to feb 28 2011
+    now = datetime.now()
+    # oct 2010 to feb -> oct 1 2010 to feb 28 this year (29th if leap year)
     assert_unamb_span(p.date_from("oct 2010 to feb"),
                            dict(year=2010, month=10),
-                           dict(year=2011, month=2))
+                           dict(year=now.year, month=2))
     
     assert_unamb_span(p.date_from("5pm to 3am", basedate),
                            dict(year=2010, month=9, day=20, hour=17),
