--- common.mk	2009-05-08 17:24:25.887869719 +0900
+++ common.mk.gcc34	2009-05-08 17:12:24.221725553 +0900
@@ -61,10 +61,10 @@
 COMMONDIR  := $(ROOTDIR)/../common
 
 # Compilers
-NVCC       := $(CUDA_INSTALL_PATH)/bin/nvcc 
-CXX        := g++
-CC         := gcc
-LINK       := g++ -fPIC
+NVCC       := $(CUDA_INSTALL_PATH)/bin/nvcc --compiler-bindir GCC34DIR
+CXX        := g++-3.4
+CC         := gcc-3.4
+LINK       := g++-3.4 -fPIC
 
 # Includes
 INCLUDES  += -I. -I$(CUDA_INSTALL_PATH)/include -I$(COMMONDIR)/inc
