diff -rupN drm/configure.ac drm.new/configure.ac
--- drm/configure.ac	2012-01-01 02:37:24.915941147 +0100
+++ drm.new/configure.ac	2012-01-01 02:48:52.431435747 +0100
@@ -47,10 +47,6 @@ LT_PREREQ([2.2])
 LT_INIT([disable-static])
 
 
-PKG_CHECK_MODULES(PTHREADSTUBS, pthread-stubs)
-AC_SUBST(PTHREADSTUBS_CFLAGS)
-AC_SUBST(PTHREADSTUBS_LIBS)
-
 PKG_CHECK_MODULES(PCIACCESS, [pciaccess >= 0.10])
 AC_SUBST(PCIACCESS_CFLAGS)
 AC_SUBST(PCIACCESS_LIBS)
diff -rupN drm/intel/Makefile.am drm.new/intel/Makefile.am
--- drm/intel/Makefile.am	2012-01-01 02:37:24.917941111 +0100
+++ drm.new/intel/Makefile.am	2012-01-01 02:49:03.353245193 +0100
@@ -26,7 +26,6 @@ AM_CFLAGS = \
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/intel \
-	$(PTHREADSTUBS_CFLAGS) \
 	$(PCIACCESS_CFLAGS) \
 	-I$(top_srcdir)/include/drm
 
@@ -34,7 +33,6 @@ libdrm_intel_la_LTLIBRARIES = libdrm_int
 libdrm_intel_ladir = $(libdir)
 libdrm_intel_la_LDFLAGS = -version-number 1:0:0 -no-undefined
 libdrm_intel_la_LIBADD = ../libdrm.la \
-	@PTHREADSTUBS_LIBS@ \
 	@PCIACCESS_LIBS@ \
 	@CLOCK_LIB@
 
diff -rupN drm/nouveau/Makefile.am drm.new/nouveau/Makefile.am
--- drm/nouveau/Makefile.am	2012-01-01 02:37:24.921941042 +0100
+++ drm.new/nouveau/Makefile.am	2012-01-01 02:49:25.184863291 +0100
@@ -2,13 +2,12 @@ AM_CFLAGS = \
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/nouveau \
-	$(PTHREADSTUBS_CFLAGS) \
 	-I$(top_srcdir)/include/drm
 
 libdrm_nouveau_la_LTLIBRARIES = libdrm_nouveau.la
 libdrm_nouveau_ladir = $(libdir)
 libdrm_nouveau_la_LDFLAGS = -version-number 1:0:0 -no-undefined
-libdrm_nouveau_la_LIBADD = ../libdrm.la @PTHREADSTUBS_LIBS@
+libdrm_nouveau_la_LIBADD = ../libdrm.la
 
 libdrm_nouveau_la_SOURCES = \
 			    nouveau_device.c \
diff -rupN drm/radeon/Makefile.am drm.new/radeon/Makefile.am
--- drm/radeon/Makefile.am	2012-01-01 02:37:24.922941025 +0100
+++ drm.new/radeon/Makefile.am	2012-01-01 02:49:15.237037163 +0100
@@ -26,13 +26,12 @@ AM_CFLAGS = \
 	$(WARN_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/radeon \
-	$(PTHREADSTUBS_CFLAGS) \
 	-I$(top_srcdir)/include/drm
 
 libdrm_radeon_la_LTLIBRARIES = libdrm_radeon.la
 libdrm_radeon_ladir = $(libdir)
 libdrm_radeon_la_LDFLAGS = -version-number 1:0:0 -no-undefined
-libdrm_radeon_la_LIBADD = ../libdrm.la @PTHREADSTUBS_LIBS@
+libdrm_radeon_la_LIBADD = ../libdrm.la
 
 libdrm_radeon_la_SOURCES = \
 	radeon_bo_gem.c \
