--- Makefile.orig	2010-09-12 20:56:54.000000000 +0200
+++ Makefile	2010-09-16 21:18:33.720000008 +0200
@@ -9,7 +9,8 @@
 
 
 # Installation program
-INSTALL = cp -i
+INSTALL = install -m644
+INSTALL_BIN = install -m755
 
 # C++ compiler
 CXX=g++
@@ -88,11 +89,8 @@
   -DFLEXIBLE  \
   -DNUMBERS \
   -DA_TERGO \
-  -DGENERALIZE \
-  -DSORT_ON_FREQ \
   -DSHOW_FILLERS  \
   -DSTOPBIT \
-  -DNEXTBIT \
   -DMORE_COMPR \
   -DCASECONV \
   -DRUNON_WORDS \
@@ -101,7 +99,6 @@
   -DCHCLASS \
   -DGUESS_LEXEMES -DGUESS_PREFIX \
   -DGUESS_MMORPH \
-  -DDUMP_ALL \
   -DLOOSING_RPM #-DDMALLOC
 
 
@@ -145,6 +142,11 @@
 # to which man section man pages for fsa belong
 MANSECT1 = 1
 MANSECT5 = 5
+# Hardcoded paths for in-file replacements
+BINDIRFILE = ${BINDIR}
+DICTDIRFILE = ${DICTDIR}
+TCLMACQDIRFILE = ${TCLMACQDIR}
+TCLMACQBINDIRFILE = ${TCLMACQBINDIR}
 
 ########################################################################
 
@@ -332,7 +334,7 @@
 	else echo "Directory ${MANDIR}/man${MANSECT5} does not exist!" ; fi
 
 installbin: ${ALL_PROGS}
-	${INSTALL} ${ALL_PROGS} ${BINDIR}
+	${INSTALL_BIN} ${ALL_PROGS} ${BINDIR}
 
 installbuild: fsa_build
 	${INSTALL} fsa_build ${BINDIR}
@@ -340,16 +342,16 @@
 installscripts:
 	if [ -d ${BINDIR} ] ; then \
 	for s in ${SKL_SCRIPTS} ; \
-	do sed -e "s|XXX_DICT_DIR|${DICTDIR}|g" \
-		-e "s|XXX_EXEC_DIR|${BINDIR}|g" < $${s}-skeleton > $${s} ; \
+	do sed -e "s|XXX_DICT_DIR|${DICTDIRFILE}|g" \
+		-e "s|XXX_EXEC_DIR|${BINDIRFILE}|g" < $${s}-skeleton > $${s} ; \
 	done ; \
 	for s in ${TCL_SCRIPTS} ; \
-	do sed -e "s|@TCLMACQDIR@|${TCLMACQDIR}|g" \
-		-e "s|@TCLMACQBINDIR@|${TCLMACQBINDIR}|g" \
+	do sed -e "s|@TCLMACQDIR@|${TCLMACQDIRFILE}|g" \
+		-e "s|@TCLMACQBINDIR@|${TCLMACQBINDIRFILE}|g" \
 		-e "s|@Q@|${PREP_FCONF}|g" < $${s}.in > $${s} ; \
 	done; \
         chmod 0755 ${ALL_SCRIPTS} ; \
-        ${INSTALL} ${ALL_SCRIPTS} ${BINDIR} ;\
+        ${INSTALL_BIN} ${ALL_SCRIPTS} ${BINDIR} ;\
 	${INSTALL} ${TCL_SUPP_FILES} ${TCLMACQDIR} ; \
 	else echo "Directory ${BINDIR} does not exist!" ; fi
 
