--- wqy-zenhei/44-wqy-zenhei.conf	2011-01-19 09:29:49.960192626 +0800
+++ 44-wqy-zenhei.conf	2012-10-17 09:26:49.599085833 +0800
@@ -6,7 +6,11 @@
   <match target="font">
         <test qual="any" name="family">
                 <string>WenQuanYi Zen Hei</string>
+        </test> 
+        <test qual="any" name="family">
                 <string>文泉驿正黑</string>
+        </test> 
+        <test qual="any" name="family">
                 <string>文泉驛正黑</string>
         </test> 
         <edit name="globaladvance"><bool>false</bool></edit>
