diff -Naur indicator-appmenu-0.3.2.orig/configure.ac indicator-appmenu-0.3.2/configure.ac
--- indicator-appmenu-0.3.2.orig/configure.ac	2011-10-13 20:24:15.000000000 +0200
+++ indicator-appmenu-0.3.2/configure.ac	2011-11-23 03:16:35.032863857 +0100
@@ -85,9 +85,23 @@
 
 DBUSMENU_JSONLOADER_REQUIRED_VERSION=0.3.3
 
-PKG_CHECK_MODULES(INDICATORTEST,
-                  dbusmenu-jsonloader-0.4 >= $DBUSMENU_JSONLOADER_REQUIRED_VERSION
-                 )
+AC_ARG_ENABLE([tests],
+	AC_HELP_STRING([--disable-tests], [Disable test scripts and tools]),,
+	[enable_tests=auto])
+
+if test x"$enable_tests" != x"no" ; then
+   PKG_CHECK_MODULES(INDICATORTEST,
+                     dbusmenu-jsonloader-0.4 >= $DBUSMENU_JSONLOADER_REQUIRED_VERSION,
+                     [have_dbusmenu_jsonloader=yes],
+                     [have_dbusmenu_jsonloader=no]
+                    )
+   if test x${enable_tests} = xyes && test x${have_dbusmenu_jsonloader} = xno; then
+     AC_MSG_ERROR([Test scripts and tools configured but dbusmenu-jsonloader not found])
+   fi
+else
+   have_dbusmenu-jsonloader=no
+fi
+AM_CONDITIONAL(BUILD_TESTS, test x${have_dbusmenu_jsonloader} = xyes)
 
 AC_SUBST(INDICATORTEST_CFLAGS)
 AC_SUBST(INDICATORTEST_LIBS)
@@ -154,4 +168,5 @@
 
 	Prefix:        $prefix
 	Indicator Dir: $INDICATORDIR
+        Test tools:    $have_dbusmenu_jsonloader
 ])
diff -Naur indicator-appmenu-0.3.2.orig/Makefile.am indicator-appmenu-0.3.2/Makefile.am
--- indicator-appmenu-0.3.2.orig/Makefile.am	2011-07-22 23:05:41.000000000 +0200
+++ indicator-appmenu-0.3.2/Makefile.am	2011-11-23 03:16:55.071636476 +0100
@@ -1,8 +1,11 @@
 
+SUBDIRS = src
+
+if BUILD_TESTS
+SUBDIRS += \
-SUBDIRS = \
-	src \
 	scripts \
 	tools
+endif
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-localinstall --enable-deprecations
 
