diff -ur racoon2-20100526a/lib/configure racoon2-20100526a-b/lib/configure
--- racoon2-20100526a/lib/configure	2008-07-05 19:41:35.000000000 -0700
+++ racoon2-20100526a-b/lib/configure	2012-03-22 14:34:11.680306711 -0700
@@ -1348,7 +1348,7 @@
 fi
 
   if test x"$enable_pedant" = xyes; then
-	OPTFLAG="-Werror $OPTFLAG"
+	OPTFLAG="-Wno-unused-but-set-variable  $OPTFLAG"
 	disable_pedant=no
   else
 	disable_pedant=yes
diff -ur racoon2-20100526a/lib/configure.in racoon2-20100526a-b/lib/configure.in
--- racoon2-20100526a/lib/configure.in	2008-03-05 04:17:29.000000000 -0700
+++ racoon2-20100526a-b/lib/configure.in	2012-03-22 14:34:24.646973384 -0700
@@ -89,7 +89,7 @@
   AC_ARG_ENABLE(pedant, [  --disable-pedant        no pedantic compiler options],
 	[], [enable_pedant=yes])
   if test x"$enable_pedant" = xyes; then
-	OPTFLAG="-Werror $OPTFLAG"
+	OPTFLAG="-Wno-unused-but-set-variable  $OPTFLAG"
 	disable_pedant=no
   else
 	disable_pedant=yes
diff -ur racoon2-20100526a/lib/configure.lineno racoon2-20100526a-b/lib/configure.lineno
--- racoon2-20100526a/lib/configure.lineno	2004-06-07 00:51:54.000000000 -0700
+++ racoon2-20100526a-b/lib/configure.lineno	2012-03-22 14:34:39.713640058 -0700
@@ -2866,7 +2866,7 @@
 # Check whether --enable-pedant or --disable-pedant was given.
 if test "${enable_pedant+set}" = set; then
   enableval="$enable_pedant"
-  OPTFLAG="-Werror $OPTFLAG"; enable_pedant=yes
+  OPTFLAG="-Wno-unused-but-set-variable  $OPTFLAG"; enable_pedant=yes
 else
   enable_pedant=no
 fi;
