diff -urN SoXt-1.2.2/src/Inventor/Xt/widgets/SoXtGLArea.cpp SoXt-1.2.2-new/src/Inventor/Xt/widgets/SoXtGLArea.cpp
--- SoXt-1.2.2/src/Inventor/Xt/widgets/SoXtGLArea.cpp	2004-08-24 13:01:19.000000000 +0200
+++ SoXt-1.2.2-new/src/Inventor/Xt/widgets/SoXtGLArea.cpp	2006-11-16 19:28:25.000000000 +0100
@@ -61,7 +61,10 @@
 #include <X11/IntrinsicP.h>
 #include <X11/StringDefs.h>
 #include <GL/gl.h>
+
+extern "C" {
 #include <Xm/PrimitiveP.h>
+}
 
 #include <Inventor/errors/SoDebugError.h>
 
diff -urN SoXt-1.2.2/src/Inventor/Xt/widgets/SoXtThumbWheelP.h SoXt-1.2.2-new/src/Inventor/Xt/widgets/SoXtThumbWheelP.h
--- SoXt-1.2.2/src/Inventor/Xt/widgets/SoXtThumbWheelP.h	2004-09-21 16:56:01.000000000 +0200
+++ SoXt-1.2.2-new/src/Inventor/Xt/widgets/SoXtThumbWheelP.h	2006-11-16 19:34:02.000000000 +0100
@@ -26,8 +26,12 @@
 
 /* $Id: SoXtThumbWheelP.h,v 1.13 2004/09/21 14:56:01 larsa Exp $ */
 
+/* XXX: or is there only need to include one of them in xtern C?!
+        -fritschy, 20061116 */
+extern "C" {
 #include <Xm/Xm.h>
 #include <Xm/PrimitiveP.h>
+}
 
 #include <Inventor/Xt/widgets/SoXtThumbWheel.h>
 
