*** rtmp_init.c	2009-05-12 11:17:35.000000000 +0200
--- rtmp_init.c.orig	2011-03-19 11:12:53.350002232 +0100
***************
*** 794,802 ****
  		pAd->MLMEThr_pid= -1;
  		pAd->RTUSBCmdThr_pid= -1;
  
! 		init_MUTEX_LOCKED(&(pAd->usbdev_semaphore));
! 		init_MUTEX_LOCKED(&(pAd->mlme_semaphore));
! 		init_MUTEX_LOCKED(&(pAd->RTUSBCmd_semaphore));
  
  		init_completion (&pAd->mlmenotify);	// event initially non-signalled
  		init_completion (&pAd->cmdnotify); 	// event initially non-signalled
--- 794,802 ----
  		pAd->MLMEThr_pid= -1;
  		pAd->RTUSBCmdThr_pid= -1;
  
! 		sema_init(&(pAd->usbdev_semaphore), 0);
! 		sema_init(&(pAd->mlme_semaphore), 0);
! 		sema_init(&(pAd->RTUSBCmd_semaphore), 0);
  
  		init_completion (&pAd->mlmenotify);	// event initially non-signalled
  		init_completion (&pAd->cmdnotify); 	// event initially non-signalled
