--- a/Makefile.am
+++ b/Makefile.am
@@ -6,7 +6,7 @@ include $(top_srcdir)/Makefile.decl
 
 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
 
-SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po
 DIST_SUBDIRS = $(SUBDIRS) build
 if BUILD_MODULAR_TESTS
 SUBDIRS += tests 
@@ -68,11 +68,8 @@ EXTRA_DIST += 			\
 
 # These may be in the builddir too
 BUILT_EXTRA_DIST = 		\
-	README			\
 	INSTALL			\
-	ChangeLog		\
-	config.h.win32		\
-	gtk-doc.make
+	ChangeLog
 
 CONFIGURE_DEPENDENCIES = acglib.m4
 
@@ -121,7 +118,6 @@ files:
 
 release:
 	rm -rf .deps */.deps
-	cd docs && make glib.html
 	$(MAKE) distcheck
 
 sanity:

--- a/configure.ac
+++ b/configure.ac
@@ -3720,15 +3720,6 @@ gio/tests/gdbus-object-manager-example/Makefile
 gio/tests/services/Makefile
 gio/tests/services/org.gtk.GDBus.Examples.ObjectManager.service
 po/Makefile.in
-docs/Makefile
-docs/reference/Makefile
-docs/reference/glib/Makefile
-docs/reference/glib/version.xml
-docs/reference/gobject/Makefile
-docs/reference/gobject/version.xml
-docs/reference/gio/Makefile
-docs/reference/gio/gdbus-object-manager-example/Makefile
-docs/reference/gio/version.xml
 tests/Makefile
 tests/gobject/Makefile
 tests/refcount/Makefile
