--- src/rlog/rlog.h~	2009-10-14 22:29:17.265283827 +0600
+++ src/rlog/rlog.h	2009-10-14 22:29:26.515285193 +0600
@@ -4,6 +4,7 @@
 #include <syslog.h>
 #include <stdarg.h>
 #include <string>
+#include <cstdio>
 
 namespace rlog {
 
--- src/boost/iostreams/filter/lzma.hpp~	2009-10-14 22:41:53.878541893 +0600
+++ src/boost/iostreams/filter/lzma.hpp	2009-10-14 22:44:12.078529051 +0600
@@ -47,10 +47,10 @@
 
                     // Compression levels
 
-BOOST_IOSTREAMS_DECL extern const lzma_easy_level no_compression;
-BOOST_IOSTREAMS_DECL extern const lzma_easy_level best_speed;
-BOOST_IOSTREAMS_DECL extern const lzma_easy_level best_compression;
-BOOST_IOSTREAMS_DECL extern const lzma_easy_level default_compression;
+BOOST_IOSTREAMS_DECL extern const uint32_t no_compression;
+BOOST_IOSTREAMS_DECL extern const uint32_t best_speed;
+BOOST_IOSTREAMS_DECL extern const uint32_t best_compression;
+BOOST_IOSTREAMS_DECL extern const uint32_t default_compression;
 
                     // Status codes
 
@@ -88,10 +88,10 @@
 struct lzma_params {
 
     // Non-explicit constructor.
-    lzma_params( lzma_easy_level level = lzma::default_compression )
+    lzma_params( uint32_t level = lzma::default_compression )
         : level(level)
         { }
-    lzma_easy_level level;
+    uint32_t level;
 };
 
 //
--- src/boost/iostreams/filter/lzma.cpp~	2009-10-14 22:44:21.335193915 +0600
+++ src/boost/iostreams/filter/lzma.cpp	2009-10-14 22:45:44.868520963 +0600
@@ -28,10 +28,10 @@
 
                     // Compression levels
 
-const lzma_easy_level no_compression       = LZMA_EASY_COPY;
-const lzma_easy_level best_speed           = LZMA_EASY_LZMA2_1;
-const lzma_easy_level best_compression     = LZMA_EASY_LZMA_9;
-const lzma_easy_level default_compression  = LZMA_EASY_LZMA_7;
+const uint32_t no_compression       = 0;
+const uint32_t best_speed           = 1;
+const uint32_t best_compression     = 9;
+const uint32_t default_compression  = 2;
 
                     // Status codes
 
@@ -39,7 +39,7 @@
 const int stream_end           = LZMA_STREAM_END;
 const int unsupported_check    = LZMA_UNSUPPORTED_CHECK;
 const int mem_error            = LZMA_MEM_ERROR;
-const int header_error         = LZMA_HEADER_ERROR;
+const int header_error         = LZMA_OPTIONS_ERROR;
 const int data_error           = LZMA_DATA_ERROR;
 const int buf_error            = LZMA_BUF_ERROR;
 const int prog_error           = LZMA_PROG_ERROR;
@@ -135,7 +135,7 @@
 
     lzma_error::check(
         compress ?
-            lzma_easy_encoder(s, p.level) :
+            lzma_easy_encoder(s, p.level, LZMA_CHECK_CRC32) :
             lzma_stream_decoder(s, 100 * 1024 * 1024, 0 )
     );
 }
