--- pyobd-0.9.2/pyobd.orig	2010-06-20 10:47:06.000000000 -0400
+++ pyobd-0.9.2/pyobd	2013-06-17 09:28:15.000000000 -0400
@@ -38,7 +38,7 @@
 import ConfigParser #safe application configuration
 import webbrowser #open browser from python
 
-from obd2_codes import pcodes
+from obd2_codes import pcodes, ucodes
 from obd2_codes import ptest
 
 from wx.lib.mixins.listctrl import ListCtrlAutoWidthMixin
@@ -550,7 +550,7 @@
 
         root = tree.AddRoot("Code Reference")
         proot = tree.AppendItem(root,"Powertrain (P) Codes")
-        codes = obd_io.pcodes.keys()
+        codes = pcodes.keys()
         codes.sort()
         group = ""
         for c in codes:
@@ -558,10 +558,10 @@
                 group_root = tree.AppendItem(proot, c[:3]+"XX")
                 group = c[:3]
             leaf = tree.AppendItem(group_root, c)
-            tree.AppendItem(leaf, obd_io.pcodes[c]) 
+            tree.AppendItem(leaf, pcodes[c]) 
 
         uroot = tree.AppendItem(root,"Network (U) Codes")
-        codes = obd_io.ucodes.keys()
+        codes = ucodes.keys()
         codes.sort()
         group = ""
         for c in codes:
@@ -569,8 +569,9 @@
                 group_root = tree.AppendItem(uroot, c[:3]+"XX")
                 group = c[:3]
             leaf = tree.AppendItem(group_root, c)
-            tree.AppendItem(leaf, obd_io.ucodes[c]) 
+            tree.AppendItem(leaf, ucodes[c]) 
 
+        tree.Expand(root)
         diag.SetSize((400,500))
         diag.Show(True)
 
--- pyobd-0.9.2/obd2_codes.py.orig	2009-09-14 03:17:36.000000000 -0400
+++ pyobd-0.9.2/obd2_codes.py	2013-06-17 09:09:13.000000000 -0400
@@ -1788,8 +1788,10 @@
 	"P3494": "Cylinder 12 Exhaust Valve Control Performance",
 	"P3495": "Cylinder 12 Exhaust Valve Control Circuit Low",
 	"P3496": "Cylinder 12 Exhaust Valve Control Circuit High",
-	"P3497": "Cylinder Deactivation System",
-        "U0001" : "High Speed CAN Communication Bus" , 
+	"P3497": "Cylinder Deactivation System"
+}
+ucodes = {
+	"U0001" : "High Speed CAN Communication Bus" , 
 	"U0002" : "High Speed CAN Communication Bus (Performance)" , 
 	"U0003" : "High Speed CAN Communication Bus (Open)" , 
 	"U0004" : "High Speed CAN Communication Bus (Low)" , 
