diff -ur FreeImage.orig/Source/OpenEXR/IlmImf/ImfAutoArray.h FreeImage/Source/OpenEXR/IlmImf/ImfAutoArray.h
--- FreeImage.orig/Source/OpenEXR/IlmImf/ImfAutoArray.h	2012-07-18 07:27:26.000000000 +0200
+++ FreeImage/Source/OpenEXR/IlmImf/ImfAutoArray.h	2013-05-25 16:46:45.461198283 +0200
@@ -45,6 +45,7 @@
 //-----------------------------------------------------------------------------
 
 #include "OpenEXRConfig.h"
+#include <cstring>
 
 namespace Imf {
 
