diff -c ./lastfm-priv.h ../../../liblastfm/src/lastfm-priv.h
*** ./lastfm-priv.h	2011-09-23 06:56:09.695770901 -0300
--- ../../../liblastfm/src/lastfm-priv.h	2011-09-23 06:54:09.896072726 -0300
***************
*** 1,6 ****
  #include <stdio.h>
  #include <curl/curl.h>
- #include <curl/types.h>
  #include <curl/easy.h>
  
  #ifndef LASTFM_PRIV_H
--- 1,5 ----
diff -c ./lfm_helper.c ../../../liblastfm/src/lfm_helper.c
*** ./lfm_helper.c	2011-09-23 06:56:09.695770901 -0300
--- ../../../liblastfm/src/lfm_helper.c	2011-09-23 06:54:40.865994640 -0300
***************
*** 10,16 ****
  #include <ctype.h>
  
  #include <curl/curl.h>
- #include <curl/types.h>
  #include <curl/easy.h>
  
  #include "lfm_helper.h"
--- 10,15 ----
diff -c ./lfm_helper.h ../../../liblastfm/src/lfm_helper.h
*** ./lfm_helper.h	2011-09-23 06:56:09.695770901 -0300
--- ../../../liblastfm/src/lfm_helper.h	2011-09-23 06:54:25.582699836 -0300
***************
*** 1,7 ****
  #include <stdlib.h>
  #include "lastfm-priv.h"
  #include <curl/curl.h>
- #include <curl/types.h>
  #include <curl/easy.h>
  
  #ifndef HELPERS_H
--- 1,6 ----
