diff -ur lineakd-0.9/lineak/cdromctrl.cpp lineakd-0.9-mod/lineak/cdromctrl.cpp
--- lineakd-0.9/lineak/cdromctrl.cpp	2006-05-22 03:45:18.000000000 +0200
+++ lineakd-0.9-mod/lineak/cdromctrl.cpp	2008-06-23 16:15:10.000000000 +0200
@@ -17,6 +17,7 @@
 
 #include <lineak/cdromctrl.h>
 #include <lineak/lineak_core_functions.h>
+#include <string.h>
 
 extern "C" {
 #include <fcntl.h>
diff -ur lineakd-0.9/lineak/defloader.cpp lineakd-0.9-mod/lineak/defloader.cpp
--- lineakd-0.9/lineak/defloader.cpp	2006-03-01 18:27:55.000000000 +0100
+++ lineakd-0.9-mod/lineak/defloader.cpp	2008-06-23 16:16:35.000000000 +0200
@@ -21,6 +21,7 @@
 #include <lineak/lkey.h>
 #include <lineak/lobject.h>
 #include <lineak/lbutton.h>
+#include <stdlib.h>
 
 using namespace lineak_util_functions;
 
diff -ur lineakd-0.9/lineak/pluginmanager.cpp lineakd-0.9-mod/lineak/pluginmanager.cpp
--- lineakd-0.9/lineak/pluginmanager.cpp	2006-04-11 23:47:49.000000000 +0200
+++ lineakd-0.9-mod/lineak/pluginmanager.cpp	2008-06-23 16:06:49.000000000 +0200
@@ -27,6 +27,7 @@
 #include <lineak/displayctrl.h>
 #include <lineak/lkbd.h>
 #include <lineak/lconfig.h>
+#include <string.h>
 
 extern "C" {
 #include <sys/types.h>
diff -ur lineakd-0.9/lineak/xmgr.cpp lineakd-0.9-mod/lineak/xmgr.cpp
--- lineakd-0.9/lineak/xmgr.cpp	2006-03-01 18:27:55.000000000 +0100
+++ lineakd-0.9-mod/lineak/xmgr.cpp	2008-06-23 16:14:10.000000000 +0200
@@ -19,6 +19,8 @@
 #include <lineak/lkbd.h>
 #include <lineak/lkey.h>
 #include <lineak/lbutton.h>
+#include <stdlib.h>
+#include <string.h>
 
 extern "C" {
    #include <signal.h>
