From 44e7c2b928b8c71dfe189208e35ca49ab210f058 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 9 Aug 2011 12:30:49 -0400
Subject: [PATCH] Add %.S and %.E rules to make debugging easier.

---
 Make.rules |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/Make.rules b/Make.rules
index 1f24ebd..cb029c3 100644
--- a/Make.rules
+++ b/Make.rules
@@ -31,3 +31,8 @@
 %.o: %.c
 	$(CC) $(INCDIR) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
+%.S: %.c
+	$(CC) $(INCDIR) $(CFLAGS) $(CPPFLAGS) -S $< -o $@
+
+%.E: %.c
+	$(CC) $(INCDIR) $(CFLAGS) $(CPPFLAGS) -E $< -o $@
-- 
1.7.6

