From 9f31f438d128035a0b38122f50de713dece12a95 Mon Sep 17 00:00:00 2001
From: Adaptee <adaptee@gmail.com>
Date: Mon, 3 Jan 2011 16:23:01 +0800
Subject: [PATCH] change the path of parted to /usr/sbin/parted, to suit with archlinux

---
 bin/usb-creator-helper |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/bin/usb-creator-helper b/bin/usb-creator-helper
index 08c0811..2427b1f 100755
--- a/bin/usb-creator-helper
+++ b/bin/usb-creator-helper
@@ -120,7 +120,7 @@ class USBCreator(dbus.service.Object):
         num = deviceobj.Get(device, 'partition-number',
                             dbus_interface=PROPS_IFACE)
         try:
-            popen(['/sbin/parted', parent, 'set', str(num), 'boot', 'on'])
+            popen(['/usr/sbin/parted', parent, 'set', str(num), 'boot', 'on'])
         except USBCreatorProcessException:
             # Don't worry about not being able to re-read the partition table.
             # TODO: As this will still be a problem for KVM users, this should
-- 
1.7.3.4

