From d3fc3577b2c4b7b373f49b1cfcae59b847648329 Mon Sep 17 00:00:00 2001
From: Adaptee <adaptee@gmail.com>
Date: Mon, 3 Jan 2011 16:31:39 +0800
Subject: [PATCH] remove the blackcode using lsb-release and python-debian, because archlinux can not provide such equivalent

---
 usbcreator/install.py |   60 +++++++++++-------------------------------------
 1 files changed, 14 insertions(+), 46 deletions(-)

diff --git a/usbcreator/install.py b/usbcreator/install.py
index bfbd817..943753d 100644
--- a/usbcreator/install.py
+++ b/usbcreator/install.py
@@ -79,7 +79,7 @@ class install(Thread):
 
     def success(self):
         pass
-    
+
     def _success(self):
         if self.progress_thread and self.progress_thread.is_alive():
             logging.debug('Shutting down the progress thread.')
@@ -113,7 +113,7 @@ class install(Thread):
         '''Will be called when we need to know if the user wants to try a
         failed operation again.  Must return a boolean value.'''
         pass
-    
+
     def join(self, timeout=None):
         self._stopevent.set()
         Thread.join(self, timeout)
@@ -124,7 +124,7 @@ class install(Thread):
             if self.progress_thread and self.progress_thread.is_alive():
                 self.progress_thread.join()
             sys.exit(0)
-    
+
     # Exception catching wrapper.
 
     def run(self):
@@ -150,7 +150,7 @@ class install(Thread):
             self._failure(_('An uncaught exception was raised:\n%s') % str(e))
 
     # Helpers for core routines.
-    
+
     def initialize_progress_thread(self):
         logging.debug('initialize_progress_thread')
         if os.path.isfile(self.source):
@@ -173,7 +173,7 @@ class install(Thread):
         self.progress_thread.progress = self.progress
         self.progress_thread.start()
         self.check()
-    
+
     def remove_extras(self):
         logging.debug('remove_extras')
         '''Remove files created by usb-creator.'''
@@ -235,27 +235,14 @@ class install(Thread):
                 'Maybe "%s" is not an Ubuntu image?') %
                 (self.target, e, self.source))
         self.check()
-        
+
         # Mangle the configuration files based on the options we've selected.
         import glob
-        import lsb_release
-        try:
-            from debian import debian_support
-        except ImportError:
-            from debian_bundle import debian_support
+
         for filename in glob.iglob(os.path.join(self.target, 'syslinux', '*.cfg')):
             if os.path.basename(filename) == 'gfxboot.cfg':
                 continue
             f = None
-            target_os_ver = None
-            our_os_ver = debian_support.Version(
-                lsb_release.get_distro_information()['RELEASE'])
-
-            if os.path.exists(os.path.join(self.target, '.disk', 'info')):
-                with open(os.path.join(self.target, '.disk', 'info'),'r') as f:
-                    contents = f.readline().split()
-                if len(contents) > 2:
-                    target_os_ver = debian_support.Version(contents[1])
             try:
                 f = open(filename, 'r')
                 label = ''
@@ -273,25 +260,6 @@ class install(Thread):
                                 line.insert(1, 'cdrom-detect/try-usb=true')
                             if label not in ('memtest', 'hd'):
                                 line.insert(1, 'noprompt')
-                        #OS version specific mangles
-                        #The syntax in syslinux changed with the version
-                        #shipping in Ubuntu 10.10
-                        elif (target_os_ver and our_os_ver and
-                              target_os_ver != our_os_ver):
-                            lucid = debian_support.Version('10.04')
-                            maverick = debian_support.Version('10.10')
-                            #10.10 or newer image, burning on 10.04 or lower
-                            if (command.lower() == 'ui' and
-                                our_os_ver <= lucid and
-                                target_os_ver >= maverick):
-                                line.remove('ui')
-                            #10.04 or earlier image, burning on 10.10 or higher
-                            #Currently still broke.
-                            #elif (command.lower() == 'gfxboot' and
-                            #      our_os_ver >= maverick and
-                            #      target_os_ver <= lucid):
-                            #    line.insert(0, 'ui')
-
                     to_write.append(' '.join(line) + '\n')
                 f.close()
                 f = open(filename, 'w')
@@ -322,7 +290,7 @@ class install(Thread):
             else:
                 # FIXME evand 2009-07-23: Need a copy of mke2fs.exe.
                 mkfs_cmd = []
-            
+
             self.progress_message(_('Creating a persistence file...'))
             popen(dd_cmd)
             self.check()
@@ -364,7 +332,7 @@ class install(Thread):
         self.progress_message(_('Writing disk image...'))
         failure_msg = _('Could not write the disk image (%s) to the device'
                         ' (%s).') % (self.source, self.device)
-        
+
         cmd = ['dd', 'if=%s' % str(self.source), 'of=%s' % str(self.device),
                'bs=1M']
         if sys.platform == 'win32':
@@ -400,7 +368,7 @@ class install(Thread):
                 continue
             listing.append(line.split())
         self.check()
-        
+
         # Clear.
 
         self.progress_message(_('Removing files...'))
@@ -418,11 +386,11 @@ class install(Thread):
                     shutil.rmtree(t)
         self.check()
         self.remove_extras()
-        
+
         self.initialize_progress_thread()
 
         # Copy.
-        
+
         cmd = ['7z', 'x', self.source, 'md5sum.txt', '-so']
         md5sums = {}
         try:
@@ -511,7 +479,7 @@ class install(Thread):
                     shutil.rmtree(f)
         self.remove_extras()
         self.check()
-        
+
         self.initialize_progress_thread()
 
         self.progress_message(_('Copying files...'))
@@ -557,7 +525,7 @@ class install(Thread):
 
         self.create_persistence()
         self.sync()
-    
+
     def copy_file(self, sourcepath, targetpath):
         self.check()
         sourcefh = None
-- 
1.7.3.4

