diff -urN quickly-12.05/bin/quickly quickly-12.05.py2/bin/quickly
--- quickly-12.05/bin/quickly	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/bin/quickly	2012-06-18 10:43:27.518464940 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/add.py quickly-12.05.py2/data/templates/ubuntu-application/add.py
--- quickly-12.05/data/templates/ubuntu-application/add.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/add.py	2012-06-18 10:43:27.521798254 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/configure.py quickly-12.05.py2/data/templates/ubuntu-application/configure.py
--- quickly-12.05/data/templates/ubuntu-application/configure.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/configure.py	2012-06-18 10:43:27.521798254 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/create.py quickly-12.05.py2/data/templates/ubuntu-application/create.py
--- quickly-12.05/data/templates/ubuntu-application/create.py	2012-03-05 11:08:32.000000000 -0600
+++ quickly-12.05.py2/data/templates/ubuntu-application/create.py	2012-06-18 10:43:27.525131568 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/debug.py quickly-12.05.py2/data/templates/ubuntu-application/debug.py
--- quickly-12.05/data/templates/ubuntu-application/debug.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/debug.py	2012-06-18 10:43:27.528464881 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2010 Travis B. Hartwell
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/design.py quickly-12.05.py2/data/templates/ubuntu-application/design.py
--- quickly-12.05/data/templates/ubuntu-application/design.py	2011-12-22 08:12:55.000000000 -0600
+++ quickly-12.05.py2/data/templates/ubuntu-application/design.py	2012-06-18 10:43:27.531798195 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/edit.py quickly-12.05.py2/data/templates/ubuntu-application/edit.py
--- quickly-12.05/data/templates/ubuntu-application/edit.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/edit.py	2012-06-18 10:43:27.535131510 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 # Copyright 2010 Tony Byrne
diff -urN quickly-12.05/data/templates/ubuntu-application/internal/apportutils.py quickly-12.05.py2/data/templates/ubuntu-application/internal/apportutils.py
--- quickly-12.05/data/templates/ubuntu-application/internal/apportutils.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/internal/apportutils.py	2012-06-18 10:43:27.535131510 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/license.py quickly-12.05.py2/data/templates/ubuntu-application/license.py
--- quickly-12.05/data/templates/ubuntu-application/license.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/license.py	2012-06-18 10:43:27.538464825 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/package.py quickly-12.05.py2/data/templates/ubuntu-application/package.py
--- quickly-12.05/data/templates/ubuntu-application/package.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/package.py	2012-06-18 10:43:27.541798140 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/project_root/bin/project_name quickly-12.05.py2/data/templates/ubuntu-application/project_root/bin/project_name
--- quickly-12.05/data/templates/ubuntu-application/project_root/bin/project_name	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/project_root/bin/project_name	2012-06-18 10:43:27.545131454 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
diff -urN quickly-12.05/data/templates/ubuntu-application/project_root/setup.py quickly-12.05.py2/data/templates/ubuntu-application/project_root/setup.py
--- quickly-12.05/data/templates/ubuntu-application/project_root/setup.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/project_root/setup.py	2012-06-18 10:43:27.548464767 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
diff -urN quickly-12.05/data/templates/ubuntu-application/project_root/tests/test_example.py quickly-12.05.py2/data/templates/ubuntu-application/project_root/tests/test_example.py
--- quickly-12.05/data/templates/ubuntu-application/project_root/tests/test_example.py	2012-04-12 09:57:42.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/project_root/tests/test_example.py	2012-06-18 10:43:27.548464767 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
diff -urN quickly-12.05/data/templates/ubuntu-application/project_root/tests/test_lint.py quickly-12.05.py2/data/templates/ubuntu-application/project_root/tests/test_lint.py
--- quickly-12.05/data/templates/ubuntu-application/project_root/tests/test_lint.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/project_root/tests/test_lint.py	2012-06-18 10:43:27.551798081 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
diff -urN quickly-12.05/data/templates/ubuntu-application/release.py quickly-12.05.py2/data/templates/ubuntu-application/release.py
--- quickly-12.05/data/templates/ubuntu-application/release.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/release.py	2012-06-18 10:43:27.555131396 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/run.py quickly-12.05.py2/data/templates/ubuntu-application/run.py
--- quickly-12.05/data/templates/ubuntu-application/run.py	2012-03-05 11:08:32.000000000 -0600
+++ quickly-12.05.py2/data/templates/ubuntu-application/run.py	2012-06-18 10:43:27.558464711 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/save.py quickly-12.05.py2/data/templates/ubuntu-application/save.py
--- quickly-12.05/data/templates/ubuntu-application/save.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/save.py	2012-06-18 10:43:27.561798026 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/share.py quickly-12.05.py2/data/templates/ubuntu-application/share.py
--- quickly-12.05/data/templates/ubuntu-application/share.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/share.py	2012-06-18 10:43:27.565131340 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/store/python/indicator.py quickly-12.05.py2/data/templates/ubuntu-application/store/python/indicator.py
--- quickly-12.05/data/templates/ubuntu-application/store/python/indicator.py	2011-12-15 13:48:28.000000000 -0600
+++ quickly-12.05.py2/data/templates/ubuntu-application/store/python/indicator.py	2012-06-18 10:43:27.565131340 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
diff -urN quickly-12.05/data/templates/ubuntu-application/submitubuntu.py quickly-12.05.py2/data/templates/ubuntu-application/submitubuntu.py
--- quickly-12.05/data/templates/ubuntu-application/submitubuntu.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/submitubuntu.py	2012-06-18 10:43:27.568464653 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/test/apportutils.py quickly-12.05.py2/data/templates/ubuntu-application/test/apportutils.py
--- quickly-12.05/data/templates/ubuntu-application/test/apportutils.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/test/apportutils.py	2012-06-18 10:43:27.575131282 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
@@ -18,7 +18,7 @@
 
 class TestApportUtils(unittest.TestCase):
     def test_lpi_existing(self):
-        lines = """#!/usr/bin/python
+        lines = """#!/usr/bin/python2
 import sys
 import os
 import gtk
@@ -59,7 +59,7 @@
         self.failIf(apportutils.detect_or_insert_lpi(lines, "project_name1", "helpMenu1"))
 
     def test_partial_lpi_import_only(self):
-        lines = """#!/usr/bin/python
+        lines = """#!/usr/bin/python2
 import sys
 import os
 import gtk
@@ -94,7 +94,7 @@
         self.failIf(apportutils.detect_or_insert_lpi(lines, "project_name1", "helpMenu1"))
 
     def test_no_lpi(self):
-        lines = """#!/usr/bin/python
+        lines = """#!/usr/bin/python2
 import sys
 import os
 import gtk
@@ -118,7 +118,7 @@
         response = about.run()
         about.destroy()
 """.splitlines(True)
-        expected = """#!/usr/bin/python
+        expected = """#!/usr/bin/python2
 import sys
 import os
 import gtk
diff -urN quickly-12.05/data/templates/ubuntu-application/test/builder/builder.py quickly-12.05.py2/data/templates/ubuntu-application/test/builder/builder.py
--- quickly-12.05/data/templates/ubuntu-application/test/builder/builder.py	2011-12-15 13:48:28.000000000 -0600
+++ quickly-12.05.py2/data/templates/ubuntu-application/test/builder/builder.py	2012-06-18 10:43:27.578464597 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 
 import unittest
diff -urN quickly-12.05/data/templates/ubuntu-application/test.py quickly-12.05.py2/data/templates/ubuntu-application/test.py
--- quickly-12.05/data/templates/ubuntu-application/test.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/test.py	2012-06-18 10:43:27.571797967 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2011 Tony Byrne
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/tutorial.py quickly-12.05.py2/data/templates/ubuntu-application/tutorial.py
--- quickly-12.05/data/templates/ubuntu-application/tutorial.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/tutorial.py	2012-06-18 10:43:27.581797912 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-application/upgrade.py quickly-12.05.py2/data/templates/ubuntu-application/upgrade.py
--- quickly-12.05/data/templates/ubuntu-application/upgrade.py	2012-05-31 12:22:01.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-application/upgrade.py	2012-06-18 10:43:27.585131226 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-cli/project_root/bin/project_name quickly-12.05.py2/data/templates/ubuntu-cli/project_root/bin/project_name
--- quickly-12.05/data/templates/ubuntu-cli/project_root/bin/project_name	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-cli/project_root/bin/project_name	2012-06-18 10:43:27.585131226 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
diff -urN quickly-12.05/data/templates/ubuntu-cli/project_root/setup.py quickly-12.05.py2/data/templates/ubuntu-cli/project_root/setup.py
--- quickly-12.05/data/templates/ubuntu-cli/project_root/setup.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-cli/project_root/setup.py	2012-06-18 10:43:27.588464539 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
diff -urN quickly-12.05/data/templates/ubuntu-flash-game/create.py quickly-12.05.py2/data/templates/ubuntu-flash-game/create.py
--- quickly-12.05/data/templates/ubuntu-flash-game/create.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-flash-game/create.py	2012-06-18 10:43:27.591797853 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-flash-game/package.py quickly-12.05.py2/data/templates/ubuntu-flash-game/package.py
--- quickly-12.05/data/templates/ubuntu-flash-game/package.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-flash-game/package.py	2012-06-18 10:43:27.595131168 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/data/templates/ubuntu-flash-game/project_root/bin/project_name quickly-12.05.py2/data/templates/ubuntu-flash-game/project_root/bin/project_name
--- quickly-12.05/data/templates/ubuntu-flash-game/project_root/bin/project_name	2012-05-02 10:33:22.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-flash-game/project_root/bin/project_name	2012-06-18 10:43:27.595131168 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
diff -urN quickly-12.05/data/templates/ubuntu-flash-game/project_root/setup.py quickly-12.05.py2/data/templates/ubuntu-flash-game/project_root/setup.py
--- quickly-12.05/data/templates/ubuntu-flash-game/project_root/setup.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/data/templates/ubuntu-flash-game/project_root/setup.py	2012-06-18 10:43:27.598464483 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 ### BEGIN LICENSE
 # This file is in the public domain
diff -urN quickly-12.05/quickly/version.py quickly-12.05.py2/quickly/version.py
--- quickly-12.05/quickly/version.py	2011-11-01 11:48:06.000000000 -0500
+++ quickly-12.05.py2/quickly/version.py	2012-06-18 10:43:27.601797798 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # Copyright 2009 Didier Roche
 #
diff -urN quickly-12.05/setup.py quickly-12.05.py2/setup.py
--- quickly-12.05/setup.py	2012-05-31 12:43:36.000000000 -0500
+++ quickly-12.05.py2/setup.py	2012-06-18 10:43:27.605131112 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 # quickly: quickly project handler
 #
diff -urN quickly-12.05/ubuntudevbar.py quickly-12.05.py2/ubuntudevbar.py
--- quickly-12.05/ubuntudevbar.py	2011-12-15 13:48:28.000000000 -0600
+++ quickly-12.05.py2/ubuntudevbar.py	2012-06-18 10:43:27.605131112 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 
 from gi.repository import GLib, GdkPixbuf, Gtk, Nautilus, Vte # pylint: disable=E0611
