diff -ur aewm-1.3.12.orig/Makefile aewm-1.3.12/Makefile
--- aewm-1.3.12.orig/Makefile	2008-01-03 02:09:41.000000000 +0900
+++ aewm-1.3.12/Makefile	2008-01-03 02:16:27.000000000 +0900
@@ -3,7 +3,7 @@
 
 # Set this to the location where you want to install
 DESTDIR =
-XROOT = /usr/X11R6
+XROOT = /usr
 
 # Uncomment to enable Shape extension support
 #OPT_WMFLAGS += -DSHAPE
@@ -78,6 +78,7 @@
 
 AEMAN = aewm.1x aeclients.1x
 AERC = aewmrc clientsrc
+AEDOC = ./doc
 
 install: all
 	mkdir -p $(BINDIR) $(MANDIR) $(CFGDIR)
