diff --git a/jhbuild/defaults.jhbuildrc b/jhbuild/defaults.jhbuildrc
index e467a49..de715c4 100644
--- a/jhbuild/defaults.jhbuildrc
+++ b/jhbuild/defaults.jhbuildrc
@@ -78,7 +78,10 @@ repos = {}
 cvsroots = {}
 svnroots = {}
 branches = {}
-module_autogenargs = {}
+# Arch-specific setting: we need to pass PYTHON=/usr/bin/python2 when building gobject-introspection
+module_autogenargs = {
+    'gobject-introspection': autogenargs + ' PYTHON=/usr/bin/python2',
+}
 module_cmakeargs = {}
 module_makeargs = {}
 module_extra_env = {}
