--- a/jhbuild/defaults.jhbuildrc
+++ b/jhbuild/defaults.jhbuildrc
@@ -60,7 +60,8 @@
         jobs = 2
 
 # override environment variables, command line arguments, etc
-autogenargs = '--disable-static --disable-gtk-doc'
+# the PYTHON argument is needed to make some modules compile on Arch
+autogenargs = '--disable-static --disable-gtk-doc PYTHON=/usr/bin/python2'
 cmakeargs = ''
 makeargs = ''
 cflags = ''
