diff -rc paragui-1.1.8/include/paragui.h paragui/include/paragui.h
*** paragui-1.1.8/include/paragui.h	2004-04-18 00:39:07.000000000 +0400
--- paragui/include/paragui.h	2010-01-28 12:32:06.000000000 +0300
***************
*** 71,77 ****
  #if defined(HAVE_HASH_MAP)
  #define HASH_MAP_INC <hash_map>
  #define MAP_INC <hash_map>
! #define STL_MAP std::hash_map
  #elif defined(HAVE_EXT_HASH_MAP)
  #define HASH_MAP_INC <ext/hash_map>
  #define MAP_INC <ext/hash_map>
--- 71,77 ----
  #if defined(HAVE_HASH_MAP)
  #define HASH_MAP_INC <hash_map>
  #define MAP_INC <hash_map>
! #define STL_MAP __gnu_cxx::hash_map
  #elif defined(HAVE_EXT_HASH_MAP)
  #define HASH_MAP_INC <ext/hash_map>
  #define MAP_INC <ext/hash_map>
diff -rc paragui-1.1.8/src/core/pgfilearchive.cpp paragui/src/core/pgfilearchive.cpp
*** paragui-1.1.8/src/core/pgfilearchive.cpp	2004-03-13 16:45:44.000000000 +0300
--- paragui/src/core/pgfilearchive.cpp	2010-01-28 12:32:41.000000000 +0300
***************
*** 100,106 ****
  	const char* sep = GetDirSeparator();
  	std::string::size_type pos = 0, incr;
  	newpath = new std::string(path);
! 	incr = std::strlen(sep);
  	if(incr == 1 && sep[0] == '/')
  		return newpath;
  
--- 100,106 ----
  	const char* sep = GetDirSeparator();
  	std::string::size_type pos = 0, incr;
  	newpath = new std::string(path);
! 	incr = strlen(sep);
  	if(incr == 1 && sep[0] == '/')
  		return newpath;
  
