--- BeastieWorker-0.4.orig/src/Makefile	2012-02-21 08:59:28.890581178 +0600
+++ BeastieWorker-0.4/src/Makefile	2012-02-21 08:59:39.520581239 +0600
@@ -3,8 +3,8 @@
 DEL=rm
 
 LFLAGS=      -O2 -pthread
-INCS=        -I/usr/local/include/SDL11 -I/usr/X11R6/include/
-LIBS=        -L/usr/local/lib -lSDL-1.1 -L/usr/X11R6/lib -lGL -lGLU
+INCS=        -I/usr/include/SDL
+LIBS=        -lSDL -lGL -lGLU
 OBJS=        cursor.o font.o main.o matrixmath.o menu.o model3d.o player.o scene.o sound.o strings.o texture.o sky.o
 MAIN=        BeastieWorker
 
--- BeastieWorker-0.4.orig/src/main.cpp	2012-02-21 08:59:28.890581178 +0600
+++ BeastieWorker-0.4/src/main.cpp	2012-02-21 08:59:39.520581239 +0600
@@ -20,17 +20,17 @@
 #define _MENU_GAME_  2
 #define _MENU_WIN_   3
 
-char *const_file_level_index= "levels/index";
-char *const_path_model=       "model/";
-char *const_file_font=        "textures/font.bmp";
-char *const_path_cursor=      "textures/";
-char *const_file_menu_skin=   "textures/skin.bmp";
-char *const_file_menu_image=  "textures/beasty.bmp";
-char *const_file_sound_level= "sounds/level.wav";
-char *const_file_sound_push=  "sounds/puch.wav";
-char *const_file_sound_step=  "sounds/step.wav";
-char *const_file_cloud=       "textures/cloud.bmp";
-char *const_file_mist=        "textures/mist.bmp";
+char *const_file_level_index= "/usr/share/beastieworker/levels/index";
+char *const_path_model=       "/usr/share/beastieworker/model/";
+char *const_file_font=        "/usr/share/beastieworker/textures/font.bmp";
+char *const_path_cursor=      "/usr/share/beastieworker/textures/";
+char *const_file_menu_skin=   "/usr/share/beastieworker/textures/skin.bmp";
+char *const_file_menu_image=  "/usr/share/beastieworker/textures/beasty.bmp";
+char *const_file_sound_level= "/usr/share/beastieworker/sounds/level.wav";
+char *const_file_sound_push=  "/usr/share/beastieworker/sounds/puch.wav";
+char *const_file_sound_step=  "/usr/share/beastieworker/sounds/step.wav";
+char *const_file_cloud=       "/usr/share/beastieworker/textures/cloud.bmp";
+char *const_file_mist=        "/usr/share/beastieworker/textures/mist.bmp";
 
 bool bFullScreen= false;
 
--- BeastieWorker-0.4.orig/model/left.3d	2012-02-21 08:59:28.893914509 +0600
+++ BeastieWorker-0.4/model/left.3d	2012-02-21 09:01:19.443915135 +0600
@@ -1,4 +1,4 @@
-T model/beastie.bmp
+T /usr/share/beastieworker/model/beastie.bmp
 M
 1 1.454459 -4.069280 16.820227 0.921623 -0.163685 -0.351877 0.338465 0.550959
 1 1.732314 -3.945513 17.490400 0.966073 -0.145701 0.213248 0.328628 0.577800
--- BeastieWorker-0.4.orig/model/look.3d	2012-02-21 08:59:28.893914509 +0600
+++ BeastieWorker-0.4/model/look.3d	2012-02-21 09:01:19.500581802 +0600
@@ -1,4 +1,4 @@
-T model/beastie.bmp
+T /usr/share/beastieworker/model/beastie.bmp
 
 M
 1 1.454459 -4.069280 16.820227 0.921623 -0.163685 -0.351877 0.338465 0.550959
--- BeastieWorker-0.4.orig/model/push.3d	2012-02-21 08:59:28.897247842 +0600
+++ BeastieWorker-0.4/model/push.3d	2012-02-21 09:01:19.557248469 +0600
@@ -1,4 +1,4 @@
-T model/beastie.bmp
+T /usr/share/beastieworker/model/beastie.bmp
 
 M
 1 1.454459 11.930719 16.820227 0.921623 -0.163686 -0.351877 0.338465 0.550959
--- BeastieWorker-0.4.orig/model/right.3d	2012-02-21 08:59:28.897247842 +0600
+++ BeastieWorker-0.4/model/right.3d	2012-02-21 09:01:19.603915136 +0600
@@ -1,4 +1,4 @@
-T model/beastie.bmp
+T /usr/share/beastieworker/model/beastie.bmp
 
 M
 1 1.454459 -4.069280 16.820227 0.921623 -0.163685 -0.351877 0.338465 0.550959
--- BeastieWorker-0.4.orig/model/step.3d	2012-02-21 08:59:28.893914509 +0600
+++ BeastieWorker-0.4/model/step.3d	2012-02-21 09:01:19.653915136 +0600
@@ -1,4 +1,4 @@
-T model/beastie.bmp
+T /usr/share/beastieworker/model/beastie.bmp
 
 M
 1 1.454459 11.930719 16.820227 0.921623 -0.163686 -0.351877 0.338465 0.550959
