--- dvdstyler.cpp.old	2014-01-12 13:46:17.358458174 +0100
+++ dvdstyler.cpp	2014-01-12 13:45:58.275124878 +0100
@@ -177,14 +177,14 @@
 	// parse command line
 	static const wxCmdLineEntryDesc cmdLineDesc[] = {
 #if wxCHECK_VERSION(2,9,0)
-		{ wxCMD_LINE_SWITCH, "s", "start", _("automatically starts the generation and burning") },
-		{ wxCMD_LINE_OPTION, "t", "tempDir", _("temporary directory") },
-		{ wxCMD_LINE_OPTION, "o", "outputDir", _("output directory") },
-		{ wxCMD_LINE_OPTION, "i", "isoFile", _("output ISO file name") },
-		{ wxCMD_LINE_OPTION, "d", "device", _("DVD device to burn") },
-		{ wxCMD_LINE_SWITCH, "e", "stderr", _("write log messages to standard error") },
-		{ wxCMD_LINE_SWITCH, "v", "version", _("show version of program") },
-		{ wxCMD_LINE_SWITCH, "h", "help", _("show summary of options") },
+		{ wxCMD_LINE_SWITCH, "s", "start", _("automatically starts the generation and burning").mb_str() },
+		{ wxCMD_LINE_OPTION, "t", "tempDir", _("temporary directory").mb_str() },
+		{ wxCMD_LINE_OPTION, "o", "outputDir", _("output directory").mb_str() },
+		{ wxCMD_LINE_OPTION, "i", "isoFile", _("output ISO file name").mb_str() },
+		{ wxCMD_LINE_OPTION, "d", "device", _("DVD device to burn").mb_str() },
+		{ wxCMD_LINE_SWITCH, "e", "stderr", _("write log messages to standard error").mb_str() },
+		{ wxCMD_LINE_SWITCH, "v", "version", _("show version of program").mb_str() },
+		{ wxCMD_LINE_SWITCH, "h", "help", _("show summary of options").mb_str() },
 #else
 		{ wxCMD_LINE_SWITCH, wxT("s"), wxT("start"), _("automatically starts the generation and burning") },
 		{ wxCMD_LINE_OPTION, wxT("t"), wxT("tempDir"), _("temporary directory") },
@@ -195,7 +195,7 @@
 		{ wxCMD_LINE_SWITCH, wxT("v"), wxT("version"), _("show version of program") },
 		{ wxCMD_LINE_SWITCH, wxT("h"), wxT("help"), _("show summary of options") },
 #endif
-		{ wxCMD_LINE_PARAM, NULL, NULL, _("input file"), wxCMD_LINE_VAL_STRING,
+		{ wxCMD_LINE_PARAM, NULL, NULL, _("input file").mb_str(), wxCMD_LINE_VAL_STRING,
 				wxCMD_LINE_PARAM_OPTIONAL | wxCMD_LINE_PARAM_MULTIPLE },
 		{ wxCMD_LINE_NONE }
 	};
