diff -ur old/omnibus/factories.py new/omnibus/factories.py
--- old/omnibus/factories.py	2014-07-29 09:26:37.000000000 -0700
+++ new/omnibus/factories.py	2014-12-21 01:29:02.912402630 -0800
@@ -33,6 +33,9 @@
         authenticator_class = auth_class
         pubsub = pubsub_instance
 
+        def check_origin(self, origin):
+            return True
+
         def open(self):
             self.on_open(None)
 
