--- src/grouplist.cpp~	2007-04-02 10:09:18.000000000 +0000
+++ src/grouplist.cpp	2007-04-02 10:09:24.000000000 +0000
@@ -22,6 +22,7 @@
 #include "grouplist.h"
 #include "binheader.h"
 #include <kmessagebox.h>
+#include <cstdlib>
 #include <assert.h>
 #include <kiconloader.h>
 #include "progress.h"
--- src/headerlist.cpp~	2007-04-02 10:12:28.000000000 +0000
+++ src/headerlist.cpp	2007-04-02 10:12:34.000000000 +0000
@@ -24,6 +24,7 @@
 #include <kmessagebox.h>
 #include <klocale.h>
 #include "progress.h"
+#include <cstdlib>
 #include <errno.h>
 #include <kprogress.h>
 #include <qregexp.h>
--- src/hlistviewitem.cpp~	2007-04-02 10:14:54.000000000 +0000
+++ src/hlistviewitem.cpp	2007-04-02 10:14:59.000000000 +0000
@@ -21,6 +21,7 @@
 #include "newsgroup.h"
 #include <qpainter.h>
 #include <klocale.h>
+#include <cstdlib>
 /*
 static const char *patterns[] = { "\\d+\\.(?:avi|mpg|mpeg|wmv)",  //.nn.avi
 	"(?:avi|mpg|mpeg|wmv)\\.\\d+)",  //.avi.nn
--- src/nntpthreadsocket.cpp~	2007-04-02 10:15:35.000000000 +0000
+++ src/nntpthreadsocket.cpp	2007-04-02 10:15:41.000000000 +0000
@@ -22,6 +22,7 @@
 #include "nntpthreadsocket.h"
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <cstdlib>
 #include <assert.h>
 #include <netdb.h>
 #include <sys/poll.h>
--- src/queueparts.cpp~	2007-04-02 10:16:09.000000000 +0000
+++ src/queueparts.cpp	2007-04-02 10:16:15.000000000 +0000
@@ -20,6 +20,7 @@
 
 #include "queueparts.h"
 #include <qfileinfo.h>
+#include <cstdlib>
 
 
 
--- src/qmgr.cpp~	2007-04-02 10:17:13.000000000 +0000
+++ src/qmgr.cpp	2007-04-02 10:17:20.000000000 +0000
@@ -29,6 +29,7 @@
 #include <qdatetime.h>
 #include "nzbform.h"
 #include "mylistview.h"
+#include <cstdlib>
 
 
 extern bool g_qResume;
--- src/updatedb.cpp~	2007-04-02 10:18:26.000000000 +0000
+++ src/updatedb.cpp	2007-04-02 10:18:31.000000000 +0000
@@ -20,6 +20,7 @@
 
 #include "updatedb.h"
 #include "binheader.h"
+#include <cstdlib>
 
 //CACHE PARAMETERS
 #define CACHESIZE 503

--- src/newsgroup.h.orig	2008-06-08 16:03:20.000000000 +0200
+++ src/newsgroup.h	2008-06-08 16:03:37.000000000 +0200
@@ -20,6 +20,7 @@
 #ifndef NEWSGROUP_H
 #define NEWSGROUP_H
 
+#include <cstdlib>
 #include "globals.h"
 // #include <kmdimainfrm.h>
 #include <klistview.h>
