Index: makefiles/src/subdir.mk
===================================================================
--- makefiles/src/subdir.mk	(revision 360)
+++ makefiles/src/subdir.mk	(working copy)
@@ -16,6 +16,7 @@
 ../src/image.cpp \
 ../src/interface.cpp \
 ../src/morph.cpp \
+../src/netrender.cpp \
 ../src/primitives.cpp \
 ../src/settings.cpp \
 ../src/shaders.cpp \
@@ -36,6 +37,7 @@
 ./src/image.o \
 ./src/interface.o \
 ./src/morph.o \
+./src/netrender.o \
 ./src/primitives.o \
 ./src/settings.o \
 ./src/shaders.o \
@@ -56,6 +58,7 @@
 ./src/image.d \
 ./src/interface.d \
 ./src/morph.d \
+./src/netrender.d \
 ./src/primitives.d \
 ./src/settings.d \
 ./src/shaders.d \
Index: src/texture.cpp
===================================================================
--- src/texture.cpp	(revision 360)
+++ src/texture.cpp	(working copy)
@@ -5,6 +5,7 @@
  *      Author: krzysztof
  */
 #include <stdio.h>
+#include <cstring>
 #include "texture.hpp"
 #include "files.h"
 
Index: src/timeline.cpp
===================================================================
--- src/timeline.cpp	(revision 360)
+++ src/timeline.cpp	(working copy)
@@ -6,7 +6,7 @@
  */
 
 #include <cstdlib>
-
+#include <cstring>
 #include "timeline.hpp"
 #include "files.h"
 #include "interface.h"
