--- src/cl_support.cpp	2011-10-10 22:56:41.485716013 -0400
+++ src/cl_support.new	2011-10-10 22:59:30.398134269 -0400
@@ -65,7 +65,7 @@
 	devices[0].getInfo(CL_DEVICE_GLOBAL_MEM_SIZE, &memorySize);
 	printf("OpenCL Memory size  %ld MB\n", memorySize/1024/1024);
 
-	std::ifstream file("cl_engine.cl");
+	std::ifstream file("/usr/share/mandelbulber/cl_engine.cl");
 	checkErr(file.is_open() ? CL_SUCCESS : -1, "Can't open file: cl_engine.cl");
 
 	std::string prog(std::istreambuf_iterator<char>(file), (std::istreambuf_iterator<char>()));

