--- window.py.orig	2010-03-21 23:46:13.510085876 +0900
+++ window.py	2010-03-21 23:48:18.720090426 +0900
@@ -120,7 +120,7 @@
         atexit.register(self.signal_handler)
         signal.signal(15, self.signal_handler)
         # Read the configuration from oxc.conf file
-        self.config = ConfigObj("oxc.conf") 
+        self.config = ConfigObj( os.environ['HOME'] + "/.openxencenter/oxc.conf" )
         # Read from configuration saved servers
         if self.config['servers']['hosts']:
             self.config_hosts = self.config['servers']['hosts']
