--- Sendmail-PMilter-0.97/lib/Sendmail/PMilter/Context.pm.orig	2009-02-24 01:38:20.000000000 +0200
+++ Sendmail-PMilter-0.97/lib/Sendmail/PMilter/Context.pm	2009-04-13 17:48:58.408517322 +0300
@@ -465,7 +465,7 @@
 	my $this = shift;
 	my $key = shift;
 
-	foreach my $code (SMFIC_RCPT, SMFIC_MAIL, SMFIC_HELO, SMFIC_CONNECT) {
+	foreach my $code (SMFIC_BODYEOB, SMFIC_BODY, SMFIC_EOH, SMFIC_HEADER, SMFIC_RCPT, SMFIC_MAIL, SMFIC_HELO, SMFIC_CONNECT) {
 		my $val = $this->{symbols}{$code}{$key};
 
 		return $val if defined($val);
