From 6d7a6f0f9fe4a475157d6b39a92739b859b69749 Mon Sep 17 00:00:00 2001
From: Dennis Schubert <mail@dennis-schubert.de>
Date: Thu, 25 Dec 2014 18:32:46 +0100
Subject: [PATCH] fix example log paths

---
 doc/example.conf | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/doc/example.conf b/doc/example.conf
index ba01072..f668118 100755
--- a/doc/example.conf
+++ b/doc/example.conf
@@ -94,15 +94,15 @@ admin {
 };
 
 log {
-	fname_userlog = "logs/userlog";
-	#fname_fuserlog = "logs/fuserlog";
-	fname_operlog = "logs/operlog";
-	#fname_foperlog = "logs/foperlog";
-	fname_serverlog = "logs/serverlog";
-	#fname_klinelog = "logs/klinelog";
-	fname_killlog = "logs/killlog";
-	fname_operspylog = "logs/operspylog";
-	#fname_ioerrorlog = "logs/ioerror";
+	fname_userlog = "/var/log/charybdis/userlog";
+	#fname_fuserlog = "/var/log/charybdis/fuserlog";
+	fname_operlog = "/var/log/charybdis/operlog";
+	#fname_foperlog = "/var/log/charybdis/foperlog";
+	fname_serverlog = "/var/log/charybdis/serverlog";
+	#fname_klinelog = "/var/log/charybdis/klinelog";
+	fname_killlog = "/var/log/charybdis/killlog";
+	fname_operspylog = "/var/log/charybdis/operspylog";
+	#fname_ioerrorlog = "/var/log/charybdis/ioerror";
 };
 
 /* class {} blocks MUST be specified before anything that uses them.  That
-- 
2.1.2

