diff -Nur paxd-libre-1.3.orig/paxd-libre.c paxd-libre-1.3/paxd-libre.c
--- paxd-libre-1.3.orig/paxd-libre.c	2014-11-08 19:40:27.000000000 -0200
+++ paxd-libre-1.3/paxd-libre.c	2014-11-28 15:30:42.304537269 -0200
@@ -28,7 +28,7 @@
         perror("inotify");
         return EXIT_FAILURE;
     }
-    int watch_conf = inotify_add_watch_x(inotify, "/etc/paxd.conf", IN_MODIFY);
+    int watch_conf = inotify_add_watch_x(inotify, "/etc/paxd-libre.conf", IN_MODIFY);
     int watch_conf_dir = inotify_add_watch_x(inotify, "/etc/", IN_CREATE | IN_MOVED_TO);
     int watch_pacman = inotify_add_watch_x(inotify, "/var/lib/pacman/", IN_DELETE);
 
@@ -45,10 +45,10 @@
 
             if (event->wd == watch_conf) {
                 fprintf(stderr, "configuration modified, updating attributes\n");
-            } else if (event->wd == watch_conf_dir && !strcmp(event->name, "paxd.conf")) {
+            } else if (event->wd == watch_conf_dir && !strcmp(event->name, "paxd-libre.conf")) {
                 fprintf(stderr, "configuration created or moved to, updating attributes\n");
                 close(watch_conf);
-                watch_conf = inotify_add_watch_x(inotify, "/etc/paxd.conf", IN_MODIFY);
+                watch_conf = inotify_add_watch_x(inotify, "/etc/paxd-libre.conf", IN_MODIFY);
             } else if (event->wd == watch_pacman && !strcmp(event->name, "db.lck")) {
                 fprintf(stderr, "pacman finished a transaction, updating attributes\n");
             } else {
