From 70949a77477c08c870767d6b3a5d3871c7b65375 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20Mal=C3=BD?= <madcatxster@gmail.com>
Date: Fri, 12 Jul 2013 19:08:33 +0200
Subject: [PATCH 2/3] Add support for ICH9 LPC

---
 compal.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/compal.c b/compal.c
index 766aec5..b35d388 100644
--- a/compal.c
+++ b/compal.c
@@ -92,6 +92,7 @@ const struct pci_device_id lpc_bridge_table[] = {
 	{PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_ICH7_30, PCI_ANY_ID, PCI_ANY_ID, 0, 0,},
 	{PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_ICH7_31, PCI_ANY_ID, PCI_ANY_ID, 0, 0,},
         {PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_ICH8_4, PCI_ANY_ID, PCI_ANY_ID, 0, 0,},
+        {PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_ICH9_5, PCI_ANY_ID, PCI_ANY_ID, 0, 0,},
 	{PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_SB400, PCI_ANY_ID, PCI_ANY_ID, 0, 0,},
 	{0,},			/* End of list */
 };
@@ -191,6 +192,7 @@ static int enable_cdimode(void)
 		case PCI_DEVICE_ID_INTEL_ICH7_30:
 		case PCI_DEVICE_ID_INTEL_ICH7_31:
 		case PCI_DEVICE_ID_INTEL_ICH8_4:	/* ICH8 */
+		case PCI_DEVICE_ID_INTEL_ICH9_5:	/* ICH9 */
 			pci_read_config_dword(lpc_bridge, INTEL_LPC_GEN4_DEC, &(value.dword));
 			pci_reg_state.dword = value.dword;
 			value.dword = 0x3CFF21;
-- 
1.8.3.4

