From c32c74f5e75d1c44dbd4ebfeb690ab3e883d49df Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20Mal=C3=BD?= <madcatxster@prifuk.cz>
Date: Tue, 20 Aug 2013 13:49:09 +0200
Subject: [PATCH 3/3] Unlock mutex in fail path.

---
 cooling.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/cooling.c b/cooling.c
index d7bae1b..80d1215 100755
--- a/cooling.c
+++ b/cooling.c
@@ -72,9 +72,8 @@ static int omnibook_cooling_write(char *buffer, struct omnibook_operation *io_op
 	if (!retval)
 		io_op->backend->cooling_state = *buffer - '0' ;
 
-	mutex_unlock(&io_op->backend->mutex);
-
 	out:
+	mutex_unlock(&io_op->backend->mutex);
 	return retval;
 }
 
-- 
1.8.3.4

