--- ../src/usr/local/Brother/Printer/mfc9465cdn/cupswrapper/cupswrappermfc9465cdn	2010-07-21 08:32:52.000000000 +0100
+++ usr/local/Brother/Printer/mfc9465cdn/cupswrapper/cupswrappermfc9465cdn	2012-07-24 18:03:06.658975145 +0100
@@ -26,11 +26,11 @@
 tmp_filter=/var/tmp/brlpdwrapper${printer_model}
 if [ "$1" = '-e' ]; then
   lpadmin -x ${printer_name}
-  rm -f /usr/share/cups/model/br${printer_model}.ppd
-  rm -f /usr/share/ppd/br${printer_model}.ppd
-  rm -f /usr/lib/cups/filter/brlpdwrapper${printer_model}
-  rm -f /usr/lib64/cups/filter/brlpdwrapper${printer_model}
-  rm -f /usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1
+  rm -f usr/share/cups/model/br${printer_model}.ppd
+  rm -f usr/share/ppd/br${printer_model}.ppd
+  rm -f usr/lib/cups/filter/brlpdwrapper${printer_model}
+  rm -f usr/lib64/cups/filter/brlpdwrapper${printer_model}
+  rm -f usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1
 if [  -e /etc/init.d/cups ]; then
    /etc/init.d/cups restart
 elif [  -e /etc/init.d/cupsys ]; then
@@ -56,33 +56,33 @@
   echo   '       -r : remove printer'
   exit 0
 fi
-#mkdir -p /usr/local/Brother/${device_model}/${printer_model}/filter
-#mkdir -p /usr/lib/cups/filter
+#mkdir -p usr/local/Brother/${device_model}/${printer_model}/filter
+#mkdir -p usr/lib/cups/filter
 
-if [ -e "/usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model}" ]; then
+if [ -e "usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model}" ]; then
   :
 else
   echo "ERROR : Brother LPD filter is not installed."
 fi
-rm -f /usr/share/cups/model/br${printer_model}.ppd
-if [ -d "/usr/share/cups/model" ]; then
-  ppd_file_name=/usr/share/cups/model/br${printer_model}.ppd
+rm -f usr/share/cups/model/br${printer_model}.ppd
+if [ -d "usr/share/cups/model" ]; then
+  ppd_file_name=usr/share/cups/model/br${printer_model}.ppd
 else
-  ppd_file_name=/usr/share/ppd/br${printer_model}.ppd
+  ppd_file_name=usr/share/ppd/br${printer_model}.ppd
 fi
 
-if [ -e "/usr/local/Brother/${device_model}/${printer_model}/cupswrapper/${printer_model}.ppd" ];	then
-cp "/usr/local/Brother/${device_model}/${printer_model}/cupswrapper/${printer_model}.ppd" $ppd_file_name
+if [ -e "usr/local/Brother/${device_model}/${printer_model}/cupswrapper/${printer_model}.ppd" ];	then
+cp "usr/local/Brother/${device_model}/${printer_model}/cupswrapper/${printer_model}.ppd" $ppd_file_name
 fi
 
 chmod 644 $ppd_file_name
 
-if [ -d /usr/share/ppd ]
+if [ -d usr/share/ppd ]
 then
-if [ -d /usr/share/cups/model ]
+if [ -d usr/share/cups/model ]
 then
-	cp $ppd_file_name /usr/share/ppd/br${printer_model}.ppd
-	chmod 644 /usr/share/ppd/br${printer_model}.ppd
+	cp $ppd_file_name usr/share/ppd/br${printer_model}.ppd
+	chmod 644 usr/share/ppd/br${printer_model}.ppd
 fi
 fi
 
@@ -128,7 +128,7 @@
 PPDC=\`echo \$PPDC | sed -e 's/PPD=//'\`
 
 if [ "\$PPDC" = "" ]; then
-    PPDC="/usr/share/cups/model/br${printer_model}.ppd"
+    PPDC="usr/share/cups/model/br${printer_model}.ppd"
 fi
 
 
@@ -198,25 +198,25 @@
       cat    > \$INPUT_TEMP_PS
    fi
 fi
-if [ -e "/usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model}" ]; then
+if [ -e "usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model}" ]; then
        :
 else
-    echo "ERROR: /usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model} does not exist"   >>\$LOGFILE
+    echo "ERROR: usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model} does not exist"   >>\$LOGFILE
     errorcode=30
     exit
 fi
 
 CUPSOPTION=\`echo "\$5 Copies=\$4" | sed -e 's/PageSize/media/' -e 's/BrSheets/Sheets/' -e 's/multiple-document-handling/Collate/' -e 's/separate-documents-collated-copies/ON/' -e 's/separate-documents-uncollated-copies/OFF/'\`
-if [ -e "/usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1" ]; then
+if [ -e "usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1" ]; then
   if [ \$DEBUG = 0 ]; then
-     /usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1  ${printer_name}  \$PPDC 0 "\$CUPSOPTION" "${printer_model}">> /dev/null
+     usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1  ${printer_name}  \$PPDC 0 "\$CUPSOPTION" "${printer_model}">> /dev/null
   else
-     /usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1  ${printer_name}  \$PPDC \$LOGCLEVEL "\$CUPSOPTION" "${printer_model}">>\$LOGFILE
+     usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1  ${printer_name}  \$PPDC \$LOGCLEVEL "\$CUPSOPTION" "${printer_model}">>\$LOGFILE
   fi
 fi
 
 if [ \$DEBUG -lt 10 ]; then
-    cat    \$INPUT_TEMP_PS | /usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model} "\$\$" "CUPS" "USB"
+    cat    \$INPUT_TEMP_PS | usr/local/Brother/${device_model}/${printer_model}/lpd/filter${printer_model} "\$\$" "CUPS" "USB"
 
     if [ \$LOGLEVEL -gt 2 ];  then
 	   if [ \$LOGFILE != "/dev/null" ]; then
@@ -235,19 +235,19 @@
 chmod 755 $tmp_filter
 
 #
-#	check /usr/lib/cups/filter
+#	check usr/lib/cups/filter
 #
-if [ -d /usr/lib/cups/filter ]; then
-	brotherlpdwrapper=/usr/lib/cups/filter/brlpdwrapper${printer_model}
+if [ -d usr/lib/cups/filter ]; then
+	brotherlpdwrapper=usr/lib/cups/filter/brlpdwrapper${printer_model}
 	rm -f  $brotherlpdwrapper
 	cp $tmp_filter	$brotherlpdwrapper
 fi
 
 #
-#	check /usr/lib64/cups/filter
+#	check usr/lib64/cups/filter
 #
-if [ -e /usr/lib64/cups/filter ]; then
-	brotherlpdwrapper64=/usr/lib64/cups/filter/brlpdwrapper${printer_model}
+if [ -e usr/lib64/cups/filter ]; then
+	brotherlpdwrapper64=usr/lib64/cups/filter/brlpdwrapper${printer_model}
 	rm -f  $brotherlpdwrapper64
 	cp $tmp_filter	$brotherlpdwrapper64
 fi
@@ -257,8 +257,8 @@
 #
 rm -f  $tmp_filter
 
-chmod a+w /usr/local/Brother/${device_model}/${printer_model}/inf/br${printer_model}rc
-chmod a+w /usr/local/Brother/${device_model}/${printer_model}/inf
+chmod a+w usr/local/Brother/${device_model}/${printer_model}/inf/br${printer_model}rc
+chmod a+w usr/local/Brother/${device_model}/${printer_model}/inf
 if [ -e /etc/init.d/lpd ]; then
    /etc/init.d/lpd stop
 fi
