diff -aur mwcollect/src/dynserv-nfqueue/dynserv-nfqueue.cpp mwcollect-build/src/dynserv-nfqueue/dynserv-nfqueue.cpp
--- mwcollect/src/dynserv-nfqueue/dynserv-nfqueue.cpp	2011-11-28 10:21:13.000000000 -0500
+++ mwcollect-build/src/dynserv-nfqueue/dynserv-nfqueue.cpp	2011-11-28 10:19:50.000000000 -0500
@@ -153,7 +153,7 @@
 	struct nfgenmsg * message, struct nfq_data * data)
 {
 	struct nfqnl_msg_packet_hdr * header;
-	char * payload;
+	unsigned char * payload;
 	int length;
 
 	if((header = nfq_get_msg_packet_hdr(data)) == 0)
