diff -urN comm-release.orig/config/autoconf.mk.in comm-release/config/autoconf.mk.in
--- comm-release.orig/config/autoconf.mk.in	2012-10-13 21:24:23.187962799 -0300
+++ comm-release/config/autoconf.mk.in	2012-10-13 21:25:50.534447407 -0300
@@ -30,14 +30,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= $(datadir)/idl/$(MOZ_APP_NAME)
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel
 
 MOZDEPTH	= $(DEPTH)/mozilla
 DIST		= $(MOZDEPTH)/dist
diff -urN comm-release.orig/mozilla/config/baseconfig.mk comm-release/mozilla/config/baseconfig.mk
--- comm-release.orig/mozilla/config/baseconfig.mk	2012-10-13 21:24:25.017889151 -0300
+++ comm-release/mozilla/config/baseconfig.mk	2012-10-13 21:26:25.089723346 -0300
@@ -1,9 +1,9 @@
 INCLUDED_AUTOCONF_MK = 1
 
-includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+includedir := $(includedir)/$(MOZ_APP_NAME)
+idldir = $(datadir)/idl/$(MOZ_APP_NAME)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 DIST = $(DEPTH)/dist
 
 # We do magic with OBJ_SUFFIX in config.mk, the following ensures we don't
diff -urN comm-release.orig/mozilla/js/src/config/baseconfig.mk comm-release/mozilla/js/src/config/baseconfig.mk
--- comm-release.orig/mozilla/js/src/config/baseconfig.mk	2012-10-13 21:24:27.134470632 -0300
+++ comm-release/mozilla/js/src/config/baseconfig.mk	2012-10-13 21:27:29.293806029 -0300
@@ -1,7 +1,7 @@
 INCLUDED_AUTOCONF_MK = 1
 
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 
 ifneq (,$(filter /%,$(TOP_DIST)))
 DIST = $(TOP_DIST)
