diff -ruN ipp2p-0.8.2.orig/libipt_ipp2p.c ipp2p-0.8.2/libipt_ipp2p.c
--- ipp2p-0.8.2.orig/libipt_ipp2p.c	2009-02-06 15:24:09.000000000 +0100
+++ ipp2p-0.8.2/libipt_ipp2p.c	2009-02-06 15:39:25.000000000 +0100
@@ -58,7 +58,7 @@
 	{ "waste", 0, 0, 'h' },
 	{ "xdcc", 0, 0, 'i' },
 	{ "debug", 0, 0, 'j' },
-        {0}
+        { .name = NULL }
 };
 
 	
@@ -373,13 +373,14 @@
 
 
 static 
-struct iptables_match ipp2p= 
+struct xtables_match ipp2p= 
 { 
     .next           = NULL,
     .name           = "ipp2p",
+    .family         = PF_INET,
     .version        = IPTABLES_VERSION,
-    .size           = IPT_ALIGN(sizeof(struct ipt_p2p_info)),
-    .userspacesize  = IPT_ALIGN(sizeof(struct ipt_p2p_info)),
+    .size           = XT_ALIGN(sizeof(struct ipt_p2p_info)),
+    .userspacesize  = XT_ALIGN(sizeof(struct ipt_p2p_info)),
     .help           = &help,
     .init           = &init,
     .parse          = &parse,
@@ -393,6 +394,6 @@
 
 void _init(void)
 {
-    register_match(&ipp2p);
+    xtables_register_match(&ipp2p);
 }
 
