diff -Nru ipp2p-0.8.2.orig/libipt_ipp2p.c ipp2p-0.8.2/libipt_ipp2p.c
--- ipp2p-0.8.2.orig/libipt_ipp2p.c	2006-01-04 18:24:19.000000000 +0200
+++ ipp2p-0.8.2/libipt_ipp2p.c	2007-12-30 20:20:26.000000000 +0200
@@ -64,12 +64,10 @@
 	
 
 static void
-init(struct ipt_entry_match *m, unsigned int *nfcache)
+init(struct ipt_entry_match *m)
 {
     struct ipt_p2p_info *info = (struct ipt_p2p_info *)m->data;
 
-    *nfcache |= NFC_UNKNOWN;
-
     /*init the module with default values*/
     info->cmd = 0;
     info->debug = 0;
@@ -79,8 +77,7 @@
 
 static int
 parse(int c, char **argv, int invert, unsigned int *flags,
-	const struct ipt_entry *entry,
-	unsigned int *nfcache,
+	const void *entry,
 	struct ipt_entry_match **match)
 {
     struct ipt_p2p_info *info = (struct ipt_p2p_info *)(*match)->data;
@@ -315,7 +312,7 @@
 
 
 static void
-print(const struct ipt_ip *ip,
+print(const void *ip,
         const struct ipt_entry_match *match,
 	int numeric)
 {
@@ -347,7 +344,7 @@
 
 
 static void
-save(const struct ipt_ip *ip, const struct ipt_entry_match *match)
+save(const void *ip, const struct ipt_entry_match *match)
 {
     struct ipt_p2p_info *info = (struct ipt_p2p_info *)match->data;
     
