--- Makefile.old	2008-08-29 02:16:29.000000000 +0200
+++ Makefile	2008-08-29 01:20:18.000000000 +0200
@@ -5,6 +5,7 @@
 
 KERNEL_SRC ?= $(firstword $(wildcard /lib/modules/$(shell uname -r)/build /usr/src/linux))
+KERNEL_SRC = /usr/src/linux-$(shell uname -r)
 ifeq ($(KERNEL_SRC),)
 $(error You need to define KERNEL_SRC)
 endif
 
@@ -36,7 +37,9 @@
 $(warning You need to install iptables sources and maybe set IPTABLES_SRC)
 endif
 
+IPTABLES_SRC = ../iptables
 IPTABLES_INCLUDE = -I$(IPTABLES_SRC)/include
+IPTABLES_INCLUDE += -I$(KERNEL_SRC)/include
 
 ifneq ($(IPTVER),)
 	IPTABLES_VERSION = $(IPTVER)
@@ -64,7 +67,7 @@
 
 libipt_ipp2p.so: libipt_ipp2p.c ipt_ipp2p.h
 	$(CC) $(CFLAGS) $(IPTABLES_OPTION) $(IPTABLES_INCLUDE) -fPIC -c libipt_ipp2p.c
-	ld -shared -o libipt_ipp2p.so libipt_ipp2p.o
+	$(CC) -shared -o libipt_ipp2p.so libipt_ipp2p.o
 
 clean:
 	-rm -f *.o *.so *.ko .*.cmd *.mod.c
