Description: Adapt to LibreOffice 3.4 paths to fix FTFBS - LP: #833889

--- lo-menubar-0.1.0.orig/wscript
+++ lo-menubar-0.1.0/wscript
@@ -24,21 +24,21 @@ def options(opt):
 def configure(conf):
 	conf.load('compiler_cxx')
 	
-	uno_sdk_libpath = '%s/basis3.3/sdk/lib' % Options.options.LO_PREFIX
+	uno_sdk_libpath = '%s/basis3.4/sdk/lib' % Options.options.LO_PREFIX
 	
 	conf.check_cxx(lib='uno_sal', uselib_store='SALLIB', libpath=uno_sdk_libpath, mandatory=True)
 	conf.check_cxx(lib='uno_cppu', uselib_store='CPPULIB', libpath=uno_sdk_libpath, mandatory=True)
 	#conf.check_cc(lib='uno_salhelpergcc3', uselib_store='SALHELPERLIB',  mandatory=True)
 	conf.check_cxx(lib='uno_cppuhelpergcc3', uselib_store='CPPUHELPERLIB', libpath=uno_sdk_libpath,  mandatory=True)
 	
-	conf.find_program('cppumaker', var='CPPUMAKER', path_list=[Options.options.LO_PREFIX + '/basis3.3/sdk/bin'], mandatory=True)
+	conf.find_program('cppumaker', var='CPPUMAKER', path_list=[Options.options.LO_PREFIX + '/basis3.4/sdk/bin'], mandatory=True)
 
 	conf.env['TYPES_RDB'] = conf.find_file('types.rdb',
 	                                       path_list=[Options.options.URE_PREFIX + '/share/misc/'],
 	                                       mandatory=True)
 	
 	conf.env['OFFAPI_RDB'] = conf.find_file('offapi.rdb',
-	                                        path_list=[Options.options.LO_PREFIX + '/basis3.3/program'],
+	                                        path_list=[Options.options.LO_PREFIX + '/basis3.4/program'],
 	                                        mandatory=True)
 	
 	conf.check_cfg(package='gio-2.0', uselib_store='GIO',
@@ -202,7 +202,7 @@ def build(bld):
 #	def post_install (ctx):
 #		if ctx.cmd != 'install':
 #			return
-#		ctx.exec_command ('/usr/lib/ure/bin/regcomp -register -wop="vnd.sun.star.expand:\$OOO_BASE_DIR/program/" -r /usr/lib/libreoffice/basis3.3/program/appmenu.rdb -c /usr/lib/libreoffice/basis3.3/program/%s.uno.so' % 'menubar')
+#		ctx.exec_command ('/usr/lib/ure/bin/regcomp -register -wop="vnd.sun.star.expand:\$OOO_BASE_DIR/program/" -r /usr/lib/libreoffice/basis3.4/program/appmenu.rdb -c /usr/lib/libreoffice/basis3.4/program/%s.uno.so' % 'menubar')
 #	
 #	bld.add_post_fun (post_install)
 
