diff -Naur wirbelscan-0.0.6.orig//scanfilter.c wirbelscan-0.0.6/scanfilter.c
--- wirbelscan-0.0.6.orig//scanfilter.c	2011-02-05 14:00:27.000000000 +0000
+++ wirbelscan-0.0.6/scanfilter.c	2011-02-12 22:55:18.764005300 +0000
@@ -90,12 +90,14 @@
         }
       }
 #endif
+#if defined PLUGINPARAMPATCHVERSNUM || VDRVERSNUM > 10713
     if (IsPvrinput(newChannel)) {
       if ((channel->Source() == newChannel->Source()) && is_nearly_same_frequency(channel, newChannel) &&
           ! strcmp(channel->Parameters(), newChannel->Parameters())) {
         return (true);
         }
       }
+#endif
     if (newChannel->IsSat()) {
       if (!is_different_transponder_deep_scan(newChannel, channel, auto_allowed)) {
         return (true);
