--- src/sakura.c	2010-04-17 03:36:17.000000000 +0200
+++ src/sakura.c	2010-05-12 13:11:09.000000000 +0200
@@ -599,7 +599,7 @@
 	npages = gtk_notebook_get_n_pages(GTK_NOTEBOOK(sakura.notebook));
 
 	/* Check for each tab if there are running processes. Use tcgetpgrp to compare to the shell PGID */
-	for (i=0; i < npages; i++) {
+/*	for (i=0; i < npages; i++) {
 
 		term = sakura_get_page_term(sakura, i);
 		pgid = tcgetpgrp(vte_terminal_get_pty(VTE_TERMINAL(term->vte)));
@@ -619,7 +619,7 @@
 			}
 		}
 	}
-
+*/
 	return FALSE;
 }
 
