--- src/sakura.c	2011-07-30 21:35:11.000000000 +0200
+++ src/sakura.c	2011-08-01 09:31:58.011616313 +0200
@@ -690,7 +690,7 @@
 	npages = gtk_notebook_get_n_pages(GTK_NOTEBOOK(sakura.notebook));
 
 	/* Check for each tab if there are running processes. Use tcgetpgrp to compare to the shell PGID */
-	for (i=0; i < npages; i++) {
+/*	for (i=0; i < npages; i++) {
 
 		term = sakura_get_page_term(sakura, i);
 		pgid = tcgetpgrp(vte_terminal_get_pty(VTE_TERMINAL(term->vte)));
@@ -711,7 +711,7 @@
 			}
 		} 
 	}
-
+*/
 	sakura_config_done();
 	return FALSE;
 }
