--- libgettext-camomile-ocaml/gettextCamomile.ml	2008-04-30 02:32:41.000000000 +0400
+++ libgettext-camomile-ocaml/gettextCamomile.ml.patched	2011-05-26 00:04:56.829494395 +0400
@@ -21,8 +21,8 @@
 (**************************************************************************)
 
 
-open CamomileLibrary;;
-open Default.Camomile;;
+open CamomileLibraryDefault;;
+open Camomile;;
 open GettextTypes;;
 
 (** Error reported when something goes wrong during Camomile initialization.
