Description: Add DESTDIR support to Makefile.
Author: Benjamin Drung <bdrung@debian.org>

--- a/Makefile.in
+++ b/Makefile.in
@@ -20,7 +20,7 @@
 #.. Absolute path to the ProjectDefs file in this directory
 OCSPROJECT=@ocs_abssrcdir@/ProjectDefs
 #.. Path to the place where the distribution is to be installed
-OCSHOME=@prefix@
+OCSHOME=$(prefix)
 #.. Description of your system 
 OSARCH=@host_os@-@host_cpu@
 #.. Path to the GNU sed program
@@ -68,13 +68,13 @@
 MINSRCPACKAGES = pkg.examples pkg.tmpl
 PACKAGES = $(MINPACKAGES) $(STDPACKAGES)
 
-OCSADMIN = OCSPROJECT=$(OCSPROJECT) $(OCSHOME)/bin/ocsadmin -ocshome $(OCSHOME) -ocssrc $(OCSSRC) -ocsspecs $(OCSSRC)/om/specs
-OCSADMININIT = OCSPROJECT=$(OCSPROJECT) $(OCSSRC)/om/scripts/ocsadmin -ocssrc $(OCSSRC) -ocshome $(OCSHOME) -ocsspecs $(OCSSRC)/om/specs
+OCSADMIN = OCSPROJECT=$(OCSPROJECT) $(DESTDIR)$(OCSHOME)/bin/ocsadmin -ocshome $(DESTDIR)$(OCSHOME) -ocssrc $(OCSSRC) -ocsspecs $(OCSSRC)/om/specs
+OCSADMININIT = OCSPROJECT=$(OCSPROJECT) $(OCSSRC)/om/scripts/ocsadmin -ocssrc $(OCSSRC) -ocshome $(DESTDIR)$(OCSHOME) -ocsspecs $(OCSSRC)/om/specs
 
 install: init $(BOOTPACKAGES) update-inter pkg.getopalname $(STDPACKAGES) $(EXTRAS)
 	$(OCSADMIN) install $(MINSRCPACKAGES)
-	chmod -R a+r @prefix@/*
-	chmod a+x @prefix@/bin/*
+	chmod -R a+r $(DESTDIR)$(prefix)/*
+	chmod a+x $(DESTDIR)$(prefix)/bin/*
 
 update-inter:
 	$(OCSADMIN) update_inter lib.opal_base
@@ -88,8 +88,8 @@
 
 reinstall: reinit $(PACKAGES) $(EXTRAS)
 	$(OCSADMIN) install $(MINSRCPACKAGES)
-	chmod -R a+r @prefix@/*
-	chmod a+x @prefix@/bin/*
+	chmod -R a+r $(DESTDIR)$(prefix)/*
+	chmod a+x $(DESTDIR)$(prefix)/bin/*
 
 init:
 	$(OCSADMININIT) init -sed $(SED) "$(LONGVERSION)"
--- a/src/om/scripts/oasys.in
+++ b/src/om/scripts/oasys.in
@@ -1,7 +1,7 @@
 #!/bin/sh
 # $Id: /opal/ocs/tags/2.4/src/om/scripts/oasys.in 10088 2011-09-28T22:00:14.597904Z florenz@TU-BERLIN.DE  $
 
-OCS=@prefix@
+OCS=${DESTDIR}@prefix@
 export OCS
 
 SHELL=/bin/sh
--- a/src/om/scripts/pureDosfop.in
+++ b/src/om/scripts/pureDosfop.in
@@ -1,7 +1,7 @@
 #!/bin/sh
 # $Id: /opal/ocs/tags/2.4/src/om/scripts/pureDosfop.in 10088 2011-09-28T22:00:14.597904Z florenz@TU-BERLIN.DE  $
 
-OCS=@prefix@
+OCS=${DESTDIR}@prefix@
 export OCS
 
 enable_dosfop=@ocs_cv_feature_dosfop@
--- a/src/om/scripts/ocs.in
+++ b/src/om/scripts/ocs.in
@@ -1,7 +1,7 @@
 #!/bin/sh
 # $Id: /opal/ocs/tags/2.4/src/om/scripts/ocs.in 10088 2011-09-28T22:00:14.597904Z florenz@TU-BERLIN.DE  $
 
-OCS=@prefix@
+OCS=${DESTDIR}@prefix@
 export OCS
 
 SHELL=/bin/sh
--- a/src/om/scripts/dosfop.in
+++ b/src/om/scripts/dosfop.in
@@ -8,7 +8,7 @@
     exit 1
 fi
 
-OCS=@prefix@
+OCS=${DESTDIR}@prefix@
 export OCS
 
 SHELL=/bin/sh
