Description: Fix typos.
Author: Benjamin Drung <bdrung@debian.org>

--- a/src/browser/BrCommands.impl
+++ b/src/browser/BrCommands.impl
@@ -36,7 +36,7 @@
 DEF desc12 == "12             C-c C-b i     Anzeige der Importe aus der Struktur am Cursor"
 DEF com12(IO,Args) == 
   LET	OptName	== name(Args)(name?)
-  IN	IF OptName nil? THEN %("Name: error, name not avaliable")
+  IN	IF OptName nil? THEN %("Name: error, name not available")
 	ELSE LET SeqPair== com12a(IO,Args)(OptName cont)
 		SortSeq	== (\\A.(A 1st)++(A 2nd))*
 				(msort(\\A,B.(A 1st)<(B 1st))(SeqPair))
@@ -72,7 +72,7 @@
 			Str2	== IF Large? THEN "\n"
 				   ELSE ""! FI 
 		   IN(&(Str1,"\tONLY"++Str2++SeqName))::Seq),<>)(SeqPair)
-  IN	IF OptLevel nil? THEN %(&("Name: error, level not avaliable",""))
+  IN	IF OptLevel nil? THEN %(&("Name: error, level not available",""))
 	ELSE SeqStr
 	FI
 
@@ -94,7 +94,7 @@
 	SortSeq	== msort(\\A,B.(A 1st)<(B 1st))(SeqPInfo)
 	SeqStr	== (\\Pair. `("","\n", "")(info2Str)
 			((Pair 1st)::(msort(<)(Pair 2nd))))*SortSeq
-  IN	IF OptLevel nil? THEN %("Name: error, level not avaliable")
+  IN	IF OptLevel nil? THEN %("Name: error, level not available")
 	ELSE SeqStr
 	FI
 
@@ -112,7 +112,7 @@
 								(SortRt)
 				SortRt	== msort(<)(Seq rt)
 			  FI)*SeqSort
-  IN	IF OptLevel nil? THEN %("Name: error, level not avaliable")
+  IN	IF OptLevel nil? THEN %("Name: error, level not available")
 	ELSE SeqStr
 	FI
 
--- a/src/browser/IndependentCom.impl
+++ b/src/browser/IndependentCom.impl
@@ -29,7 +29,7 @@
 			"?              C-cC-bC-vC-f  Anzeigelevel auf fct setzen",
 			"?              C-cC-bC-vC-o  Anzeigelevel auf optimize setzen",
 			"")
-	AktLev	== IF OptLevel nil? THEN %("Name: error, level not avaliable")
+	AktLev	== IF OptLevel nil? THEN %("Name: error, level not available")
 		   IF OptLevel avail? THEN %(OptLevel cont)
 		   FI
 	OptLevel== name(Args)(level?)
--- a/src/oasys/debugger/dbcontrol/DebuggerInterface.impl
+++ b/src/oasys/debugger/dbcontrol/DebuggerInterface.impl
@@ -186,7 +186,7 @@
     )
     ++ 
     cmd(lit("oasys-intern-db-kill") @ terminateClient & succeed("ok."),
-        "Kill the evaluator proces."
+        "Kill the evaluator process."
     )
     ++ 
     cmd(lit("oasys-intern-db-sync") 
--- a/src/om/genmake/genmake.1
+++ b/src/om/genmake/genmake.1
@@ -78,7 +78,7 @@
 is given, genmake generates a maintenance file for the
 top-level system of an OPAL program. If
 .I top-level-command
-is ommitted, the name of the the top-level structure is used instead.
+is omitted, the name of the the top-level structure is used instead.
 If
 .BI \-S " system"
 is given, genmake generates a maintenance file for a
@@ -184,7 +184,7 @@
 ./ .BI \-keep
 ./ Generate a description file which keeps all derived compilation
 ./ objects for later reuse. If shape is used, this objects will also be
-./ cached. This is is usefull if compilation options such as debugging
+./ cached. This is is useful if compilation options such as debugging
 ./ and optimization are frequently changed, since it avoids reproduction
 ./ of objects not affected by the changes.
 .TP
