--- certmaster-0.27/certmaster/SSLConnection.py
+++ certmaster-0.27/certmaster/SSLConnection.py
@@ -62,7 +62,7 @@
         c, a = self.__dict__["conn"].accept()
         return (SSLConnection(c), a)
 
-    def makefile(self, mode, bufsize):
+    def makefile(self, mode='r', bufsize=-1):
         """
         We need to use socket._fileobject Because SSL.Connection
         doesn't have a 'dup'. Not exactly sure WHY this is, but
