--- Makefile	2013-03-31 14:15:14.000000000 +0200
+++ ../../Makefile	2013-05-11 21:34:32.431640251 +0200
@@ -1,6 +1,18 @@
+# LUAV: 5.1 5.2
+# lua version to build against
+LUAV?=5.1
+
+# prefix: /usr/local /usr /opt/local /sw
+# the top of the default install tree
+prefix?=/usr/local
+
+# DESTDIR: (no default)
+# used by package managers to install into a temporary destination
+DESTDIR=
+
 # Inform the location to intall the modules
-LUAPATH=/usr/local/share/lua/5.1
-LUACPATH=/usr/local/lib/lua/5.1
+LUAPATH=$(prefix)/share/lua/$(LUAV)
+LUACPATH=$(prefix)/lib/lua/$(LUAV)
 
 # Compile with build-in LuaSocket's help files.
 # Comment this lines if you will link with non-internal LuaSocket's help files
@@ -30,7 +42,7 @@
 	@echo "  * macosx"
 
 install:
-	@cd src && $(MAKE) LUACPATH="$(LUACPATH)" LUAPATH="$(LUAPATH)" install
+	@cd src && $(MAKE) LUACPATH="$(DESTDIR)$(LUACPATH)" LUAPATH="$(DESTDIR)$(LUAPATH)" install
 
 linux:
 	@echo "---------------------"
