diff -urN visual-5.71_releaseorig//src/gtk2/makefile visual-5.71_release/src/gtk2/makefile
--- visual-5.71_releaseorig//src/gtk2/makefile	2011-07-03 12:48:38.000000000 +0200
+++ visual-5.71_release/src/gtk2/makefile	2011-07-03 12:48:53.000000000 +0200
@@ -59,7 +59,7 @@
 #CVISUAL_LIBS += -lstdc++
 
 CVISUAL_LIBS += $(GTK_LIBS) $(GTHREAD_LIBS) \
-       -lpython$(PYTHON_VERSION) -lboost_python-mt -lboost_signals-mt -lboost_thread-mt -lstdc++
+       -lpython$(PYTHON_VERSION) -lboost_python -lboost_signals -lboost_thread -lstdc++
 
 CVISUAL_OBJS = arrayprim.o arrow.o axial.o box.o cone.o cylinder.o display_kernel.o ellipsoid.o \
 	frame.o label.o material.o mouse_manager.o mouseobject.o primitive.o pyramid.o \
