diff -urN visual-5.71_releaseorig/bin/vpython.in visual-5.71_release/bin/vpython.in
--- visual-5.71_releaseorig/bin/vpython.in	2011-05-08 19:21:49.000000000 +0200
+++ visual-5.71_release/bin/vpython.in	2011-05-08 19:22:55.000000000 +0200
@@ -2,7 +2,7 @@
 # Variables determined by configure.
 PYTHON=@PYTHON@
 visualexampledir=@visualexampledir@
-IDLE=idle
+IDLE=idle2
 PYTHONPATH=@PYTHONPATH@
 export PYTHONPATH
 
@@ -17,8 +17,8 @@
     ;;
 '')
     cd $visualexampledir
-    $PYTHON $IDLE
+    $IDLE
     ;;
-*)  $PYTHON $IDLE "$*"
+*)  $IDLE "$*"
     ;;
 esac
