diff -crB ./Makefile ../../src2/notecase-1.9.8/Makefile
*** ./Makefile	2008-12-09 05:07:06.000000000 -0500
--- ../../src2/notecase-1.9.8/Makefile	2010-10-28 15:52:02.266666675 -0400
***************
*** 694,705 ****
  		echo 'pkg-config error detected $(shell pkg-config --cflags $(LIBS))';\
  		exit 1;\
  	fi;  
! 	$(Q)if [ "$(HAVE_GTKSOURCEVIEW)" = "1" ]; then \
! 		if [[ "$(TEST_SRCVIEW_VER)" < "2.4.1" ]]; then \
! 		echo 'ERROR: gtksourceview package is too old ($(shell pkg-config --modversion gtksourceview-2.0))!';\
! 		exit 1;\
! 		fi;\
! 	fi;
  	$(Q)echo 'Configuration: GNOME_VFS=$(HAVE_GNOME_VFS);GTKSOURCEVIEW=$(HAVE_GTKSOURCEVIEW);DEBUG=$(DEBUG);PROFILE=$(PROFILE);BYTE_ORDER=$(BYTE_ORDER);PLATFORM=$(PLATFORM)' 
  	$(Q)$(SED) 's/^Version:.*/Version:        $(VERSION)/' ./docs/notecase.spec
  	$(Q)$(SED) 's/^Source:.*/Source:         $(archive)_src.tar.gz/' ./docs/notecase.spec
--- 694,705 ----
  		echo 'pkg-config error detected $(shell pkg-config --cflags $(LIBS))';\
  		exit 1;\
  	fi;  
! #	$(Q)if [ "$(HAVE_GTKSOURCEVIEW)" = "1" ]; then \
! #		if [[  "$(TEST_SRCVIEW_VER)" < "2.4.1" ]]; then \
! #		echo 'ERROR: gtksourceview package is too old ($(shell pkg-config --modversion gtksourceview-2.0))!';\
! #		exit 1;\
! #		fi;\
! #	fi;
  	$(Q)echo 'Configuration: GNOME_VFS=$(HAVE_GNOME_VFS);GTKSOURCEVIEW=$(HAVE_GTKSOURCEVIEW);DEBUG=$(DEBUG);PROFILE=$(PROFILE);BYTE_ORDER=$(BYTE_ORDER);PLATFORM=$(PLATFORM)' 
  	$(Q)$(SED) 's/^Version:.*/Version:        $(VERSION)/' ./docs/notecase.spec
  	$(Q)$(SED) 's/^Source:.*/Source:         $(archive)_src.tar.gz/' ./docs/notecase.spec
diff -crB ./src/lib/IniFile.cpp ../../src2/notecase-1.9.8/src/lib/IniFile.cpp
*** ./src/lib/IniFile.cpp	2008-10-24 14:54:50.000000000 -0400
--- ../../src2/notecase-1.9.8/src/lib/IniFile.cpp	2010-10-28 14:33:35.196666673 -0400
***************
*** 99,104 ****
--- 99,105 ----
  
  bool IniFile::Save()
  {
+     #define snprintf
  	//return true;
      m_bDirty = false;   //reset flag
  
