diff -rupN ../../src.orig/notecase-1.9.8//Makefile ./Makefile
--- ../../src.orig/notecase-1.9.8//Makefile	2011-02-20 18:57:21.602838389 -0500
+++ ./Makefile	2011-02-20 20:48:11.143865652 -0500
@@ -694,12 +694,12 @@ updatesrc:
 		echo 'pkg-config error detected $(shell pkg-config --cflags $(LIBS))';\
 		exit 1;\
 	fi;  
-	$(Q)if [ "$(HAVE_GTKSOURCEVIEW)" = "1" ]; then \
-		if [[ "$(TEST_SRCVIEW_VER)" < "2.4.1" ]]; then \
-		echo 'ERROR: gtksourceview package is too old ($(shell pkg-config --modversion gtksourceview-2.0))!';\
-		exit 1;\
-		fi;\
-	fi;
+#	$(Q)if [ "$(HAVE_GTKSOURCEVIEW)" = "1" ]; then \
+#		if [[ "$(TEST_SRCVIEW_VER)" < "2.4.1" ]]; then \
+#		echo 'ERROR: gtksourceview package is too old ($(shell pkg-config --modversion gtksourceview-2.0))!';\
+#		exit 1;\
+#		fi;\
+#	fi;
 	$(Q)echo 'Configuration: GNOME_VFS=$(HAVE_GNOME_VFS);GTKSOURCEVIEW=$(HAVE_GTKSOURCEVIEW);DEBUG=$(DEBUG);PROFILE=$(PROFILE);BYTE_ORDER=$(BYTE_ORDER);PLATFORM=$(PLATFORM)' 
 	$(Q)$(SED) 's/^Version:.*/Version:        $(VERSION)/' ./docs/notecase.spec
 	$(Q)$(SED) 's/^Source:.*/Source:         $(archive)_src.tar.gz/' ./docs/notecase.spec
@@ -820,7 +820,7 @@ weexpc: updatesrc touch
 #
 # Note: first "make" command forces generating correct help file path in "config.h"
 #
-DEBSIZE:=$(shell du -ks ./notecase/ | cut -f 1)
+#DEBSIZE:=$(shell du -ks ./notecase/ | cut -f 1)
 
 deb:
 	$(Q)mkdir ./notecase
diff -rupN ../../src.orig/notecase-1.9.8//src/lib/IniFile.cpp ./src/lib/IniFile.cpp
--- ../../src.orig/notecase-1.9.8//src/lib/IniFile.cpp	2011-02-20 18:57:21.616165057 -0500
+++ ./src/lib/IniFile.cpp	2011-02-20 20:36:45.217580257 -0500
@@ -9,6 +9,7 @@
 #include "IniFile.h"
 #include <algorithm>    //find() function
 #include <string>
+#include <cstdio>
 #include "File64.h"
 #include "FilePath.h"
 #include "debug.h"
@@ -121,14 +122,14 @@ bool IniFile::Save()
     for(int i=0; i<size; i++)
     {
         //write line with section name
-        snprintf(szBuffer, sizeof(szBuffer), "[%s]\n", m_contents[i].m_name.c_str());
+		std::snprintf(szBuffer, sizeof(szBuffer), "[%s]\n", m_contents[i].m_name.c_str());
 		file.Write(szBuffer, strlen(szBuffer));
 
         int count = m_contents[i].m_Keys.size();
         for(int j=0; j<count; j++)
         {
             //write "key = value"
-            snprintf(szBuffer, sizeof(szBuffer), "%s=%s\n",
+		std::snprintf(szBuffer, sizeof(szBuffer), "%s=%s\n",
                 m_contents[i].m_Keys[j].m_name.c_str(),
                 m_contents[i].m_Keys[j].m_value.c_str());
 			file.Write(szBuffer, strlen(szBuffer));
@@ -210,7 +211,7 @@ bool IniFile::GetValue(const char *szSec
 bool IniFile::SetValue(const char *szSection, const char *szKey, const int nValue)
 {
     char szNumber[30] = "";
-    snprintf(szNumber, sizeof(szNumber), "%d", nValue);
+    std::snprintf(szNumber, sizeof(szNumber), "%d", nValue);
 
     return SetValue(szSection, szKey, szNumber);
 }
