diff -urb kterm-6.2.0.orig/button.c kterm-6.2.0/button.c
--- kterm-6.2.0.orig/button.c	1996-07-02 08:01:46.000000000 +0300
+++ kterm-6.2.0/button.c	2008-11-27 12:09:50.000000000 +0200
@@ -41,8 +41,6 @@
 #include "error.h"
 #include "menu.h"
 
-extern char *malloc();
-
 extern void DoSecureKeyboard();
 
 #define KeyState(x) (((x) & (ShiftMask|ControlMask)) + (((x) & Mod1Mask) ? 2 : 0))
diff -urb kterm-6.2.0.orig/charproc.c kterm-6.2.0/charproc.c
--- kterm-6.2.0.orig/charproc.c	1996-07-12 08:01:36.000000000 +0300
+++ kterm-6.2.0/charproc.c	2008-11-27 12:10:33.000000000 +0200
@@ -74,6 +74,7 @@
 #include <errno.h>
 #include <setjmp.h>
 #include <ctype.h>
+#include <stdlib.h>
 #include "VTparse.h"
 #include "data.h"
 #include "error.h"
@@ -101,9 +102,6 @@
 
 extern XtAppContext app_con;
 extern Widget toplevel;
-extern void exit();
-extern char *malloc();
-extern char *realloc();
 extern fd_set Select_mask;
 extern fd_set X_mask;
 extern fd_set pty_mask;
diff -urb kterm-6.2.0.orig/screen.c kterm-6.2.0/screen.c
--- kterm-6.2.0.orig/screen.c	1996-06-23 11:00:26.000000000 +0300
+++ kterm-6.2.0/screen.c	2008-11-27 12:10:48.000000000 +0200
@@ -45,8 +45,8 @@
 #include <sys/stream.h>			/* get typedef used in ptem.h */
 #include <sys/ptem.h>
 #endif
+#include <stdlib.h>
 
-extern Char *calloc(), *malloc(), *realloc();
 extern void free();
 
 ScrnBuf Allocate (nrow, ncol, addr)
diff -urb kterm-6.2.0.orig/scrollbar.c kterm-6.2.0/scrollbar.c
--- kterm-6.2.0.orig/scrollbar.c	1996-07-12 08:01:39.000000000 +0300
+++ kterm-6.2.0/scrollbar.c	2008-11-27 12:11:14.000000000 +0200
@@ -30,6 +30,7 @@
 
 #include <stdio.h>
 #include <ctype.h>
+#include <stdlib.h>
 #include <X11/Xatom.h>
 
 #include <X11/StringDefs.h>
@@ -324,7 +325,6 @@
 	register TScreen *screen = &xw->screen;
 	register int border = 2 * screen->border;
 	register int i;
-	Char *realloc(), *calloc();
 
 	if(screen->scrollbar)
 		return;
