diff -Nru freeipa-3.0.0.orig/checks/check-ra.py freeipa-3.0.0/checks/check-ra.py
--- freeipa-3.0.0.orig/checks/check-ra.py	2012-11-15 15:13:09.617353890 -0500
+++ freeipa-3.0.0/checks/check-ra.py	2012-11-15 17:26:14.829260760 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors:
 #   Jason Gerard DeRose <jderose@redhat.com>
 #   John Dennis <jdennis@redhat.com>
diff -Nru freeipa-3.0.0.orig/contrib/RHEL4/ipa-client-setup freeipa-3.0.0/contrib/RHEL4/ipa-client-setup
--- freeipa-3.0.0.orig/contrib/RHEL4/ipa-client-setup	2012-11-15 15:13:09.370687151 -0500
+++ freeipa-3.0.0/contrib/RHEL4/ipa-client-setup	2012-11-15 17:30:47.565980980 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Simo Sorce <ssorce@redhat.com>
 #          Karl MacMillan <kmacmillan@mentalrootkit.com>
 #
diff -Nru freeipa-3.0.0.orig/contrib/RHEL4/setup.py freeipa-3.0.0/contrib/RHEL4/setup.py
--- freeipa-3.0.0.orig/contrib/RHEL4/setup.py	2012-11-15 15:13:09.370687151 -0500
+++ freeipa-3.0.0/contrib/RHEL4/setup.py	2012-11-15 17:30:39.885979496 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Copyright (C) 2007  Red Hat
 # see file 'COPYING' for use and warranty information
 #
diff -Nru freeipa-3.0.0.orig/doc/examples/python-api.py freeipa-3.0.0/doc/examples/python-api.py
--- freeipa-3.0.0.orig/doc/examples/python-api.py	2012-11-15 15:13:09.370687151 -0500
+++ freeipa-3.0.0/doc/examples/python-api.py	2012-11-15 17:30:55.142649480 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors:
 #   Jason Gerard DeRose <jderose@redhat.com>
 #
diff -Nru freeipa-3.0.0.orig/init/systemd/freeipa-systemd-upgrade freeipa-3.0.0/init/systemd/freeipa-systemd-upgrade
--- freeipa-3.0.0.orig/init/systemd/freeipa-systemd-upgrade	2012-11-15 15:13:09.344020476 -0500
+++ freeipa-3.0.0/init/systemd/freeipa-systemd-upgrade	2012-11-15 17:31:21.352654393 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 from ipaserver.install.krbinstance import update_key_val_in_file
 from ipapython import ipautil, config
 from ipapython import services as ipaservices
diff -Nru freeipa-3.0.0.orig/install/certmonger/dogtag-ipa-retrieve-agent-submit freeipa-3.0.0/install/certmonger/dogtag-ipa-retrieve-agent-submit
--- freeipa-3.0.0.orig/install/certmonger/dogtag-ipa-retrieve-agent-submit	2012-11-15 15:13:09.444020503 -0500
+++ freeipa-3.0.0/install/certmonger/dogtag-ipa-retrieve-agent-submit	2012-11-15 17:28:05.059283071 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python -E
+#!/usr/bin/python2 -E
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/install/po/pygettext.py freeipa-3.0.0/install/po/pygettext.py
--- freeipa-3.0.0.orig/install/po/pygettext.py	2012-11-15 15:13:09.437353834 -0500
+++ freeipa-3.0.0/install/po/pygettext.py	2012-11-15 17:29:44.942635841 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python2
 # -*- coding: iso-8859-1 -*-
 # Originally written by Barry Warsaw <barry@zope.com>
 #
diff -Nru freeipa-3.0.0.orig/install/restart_scripts/renew_ca_cert freeipa-3.0.0/install/restart_scripts/renew_ca_cert
--- freeipa-3.0.0.orig/install/restart_scripts/renew_ca_cert	2012-11-15 15:13:09.474020510 -0500
+++ freeipa-3.0.0/install/restart_scripts/renew_ca_cert	2012-11-15 17:27:33.199276389 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python -E
+#!/usr/bin/python2 -E
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/install/restart_scripts/renew_ra_cert freeipa-3.0.0/install/restart_scripts/renew_ra_cert
--- freeipa-3.0.0.orig/install/restart_scripts/renew_ra_cert	2012-11-15 15:13:09.474020510 -0500
+++ freeipa-3.0.0/install/restart_scripts/renew_ra_cert	2012-11-15 17:27:50.549280342 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python -E
+#!/usr/bin/python2 -E
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/install/restart_scripts/restart_dirsrv freeipa-3.0.0/install/restart_scripts/restart_dirsrv
--- freeipa-3.0.0.orig/install/restart_scripts/restart_dirsrv	2012-11-15 15:13:09.474020510 -0500
+++ freeipa-3.0.0/install/restart_scripts/restart_dirsrv	2012-11-15 17:27:58.169281755 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python -E
+#!/usr/bin/python2 -E
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/install/restart_scripts/restart_httpd freeipa-3.0.0/install/restart_scripts/restart_httpd
--- freeipa-3.0.0.orig/install/restart_scripts/restart_httpd	2012-11-15 15:13:09.474020510 -0500
+++ freeipa-3.0.0/install/restart_scripts/restart_httpd	2012-11-15 17:27:18.532606981 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python -E
+#!/usr/bin/python2 -E
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/install/restart_scripts/restart_pkicad freeipa-3.0.0/install/restart_scripts/restart_pkicad
--- freeipa-3.0.0.orig/install/restart_scripts/restart_pkicad	2012-11-15 15:13:09.474020510 -0500
+++ freeipa-3.0.0/install/restart_scripts/restart_pkicad	2012-11-15 17:27:25.369274945 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python -E
+#!/usr/bin/python2 -E
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-adtrust-install freeipa-3.0.0/install/tools/ipa-adtrust-install
--- freeipa-3.0.0.orig/install/tools/ipa-adtrust-install	2012-11-15 15:13:09.444020503 -0500
+++ freeipa-3.0.0/install/tools/ipa-adtrust-install	2012-11-15 17:29:27.362631982 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python2
 #
 # Authors: Sumit Bose <sbose@redhat.com>
 # Based on ipa-server-install by Karl MacMillan <kmacmillan@mentalrootkit.com>
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-ca-install freeipa-3.0.0/install/tools/ipa-ca-install
--- freeipa-3.0.0.orig/install/tools/ipa-ca-install	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-ca-install	2012-11-15 17:28:30.422621051 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Rob Crittenden <rcritten@redhat.com>
 #
 # Copyright (C) 2011  Red Hat
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-compat-manage freeipa-3.0.0/install/tools/ipa-compat-manage
--- freeipa-3.0.0.orig/install/tools/ipa-compat-manage	2012-11-15 15:13:09.444020503 -0500
+++ freeipa-3.0.0/install/tools/ipa-compat-manage	2012-11-15 17:29:19.182630373 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors: Rob Crittenden <rcritten@redhat.com>
 # Authors: Simo Sorce <ssorce@redhat.com>
 #
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-compliance freeipa-3.0.0/install/tools/ipa-compliance
--- freeipa-3.0.0.orig/install/tools/ipa-compliance	2012-11-15 15:13:09.444020503 -0500
+++ freeipa-3.0.0/install/tools/ipa-compliance	2012-11-15 17:29:15.012629553 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-csreplica-manage freeipa-3.0.0/install/tools/ipa-csreplica-manage
--- freeipa-3.0.0.orig/install/tools/ipa-csreplica-manage	2012-11-15 15:13:09.444020503 -0500
+++ freeipa-3.0.0/install/tools/ipa-csreplica-manage	2012-11-15 17:29:23.729297946 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Rob Crittenden <rcritten@redhat.com>
 #
 # Based on ipa-replica-manage by Karl MacMillan <kmacmillan@mentalrootkit.com>
diff -Nru freeipa-3.0.0.orig/install/tools/ipactl freeipa-3.0.0/install/tools/ipactl
--- freeipa-3.0.0.orig/install/tools/ipactl	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipactl	2012-11-15 17:28:45.715957216 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors: Simo Sorce <ssorce@redhat.com>
 #
 # Copyright (C) 2008-2010  Red Hat
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-dns-install freeipa-3.0.0/install/tools/ipa-dns-install
--- freeipa-3.0.0.orig/install/tools/ipa-dns-install	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-dns-install	2012-11-15 17:28:35.125955298 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Martin Nagy <mnagy@redhat.com>
 # Based on ipa-server-install by Karl MacMillan <kmacmillan@mentalrootkit.com>
 #
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-ldap-updater freeipa-3.0.0/install/tools/ipa-ldap-updater
--- freeipa-3.0.0.orig/install/tools/ipa-ldap-updater	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-ldap-updater	2012-11-15 17:28:41.925956524 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors: Rob Crittenden <rcritten@redhat.com>
 #
 # Copyright (C) 2008  Red Hat
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-managed-entries freeipa-3.0.0/install/tools/ipa-managed-entries
--- freeipa-3.0.0.orig/install/tools/ipa-managed-entries	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-managed-entries	2012-11-15 17:29:05.642627732 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors: Jr Aquino <jr.aquino@citrix.com>
 #
 # Copyright (C) 2011  Red Hat
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-nis-manage freeipa-3.0.0/install/tools/ipa-nis-manage
--- freeipa-3.0.0.orig/install/tools/ipa-nis-manage	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-nis-manage	2012-11-15 17:28:49.732624692 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors: Rob Crittenden <rcritten@redhat.com>
 # Authors: Simo Sorce <ssorce@redhat.com>
 #
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-replica-conncheck freeipa-3.0.0/install/tools/ipa-replica-conncheck
--- freeipa-3.0.0.orig/install/tools/ipa-replica-conncheck	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-replica-conncheck	2012-11-15 17:29:39.272634691 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Martin Kosek <mkosek@redhat.com>
 #
 # Copyright (C) 2011  Red Hat
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-replica-install freeipa-3.0.0/install/tools/ipa-replica-install
--- freeipa-3.0.0.orig/install/tools/ipa-replica-install	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-replica-install	2012-11-15 17:28:54.572625627 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Karl MacMillan <kmacmillan@mentalrootkit.com>
 #
 # Copyright (C) 2007  Red Hat
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-replica-manage freeipa-3.0.0/install/tools/ipa-replica-manage
--- freeipa-3.0.0.orig/install/tools/ipa-replica-manage	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-replica-manage	2012-11-15 17:28:23.655953147 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Karl MacMillan <kmacmillan@mentalrootkit.com>
 #
 # Copyright (C) 2007  Red Hat
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-replica-prepare freeipa-3.0.0/install/tools/ipa-replica-prepare
--- freeipa-3.0.0.orig/install/tools/ipa-replica-prepare	2012-11-15 15:13:09.437353834 -0500
+++ freeipa-3.0.0/install/tools/ipa-replica-prepare	2012-11-15 17:28:17.825952026 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Karl MacMillan <kmacmillan@mentalrootkit.com>
 #
 # Copyright (C) 2007  Red Hat
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-server-certinstall freeipa-3.0.0/install/tools/ipa-server-certinstall
--- freeipa-3.0.0.orig/install/tools/ipa-server-certinstall	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-server-certinstall	2012-11-15 17:29:00.115959983 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Karl MacMillan <kmacmillan@mentalrootkit.com>
 #
 # Copyright (C) 2007  Red Hat
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-server-install freeipa-3.0.0/install/tools/ipa-server-install
--- freeipa-3.0.0.orig/install/tools/ipa-server-install	2012-11-15 15:13:09.440687168 -0500
+++ freeipa-3.0.0/install/tools/ipa-server-install	2012-11-15 17:29:10.995962102 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Karl MacMillan <kmacmillan@mentalrootkit.com>
 #          Simo Sorce <ssorce@redhat.com>
 #          Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/install/tools/ipa-upgradeconfig freeipa-3.0.0/install/tools/ipa-upgradeconfig
--- freeipa-3.0.0.orig/install/tools/ipa-upgradeconfig	2012-11-15 16:24:59.041702000 -0500
+++ freeipa-3.0.0/install/tools/ipa-upgradeconfig	2012-11-15 17:29:30.852632679 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/ipa freeipa-3.0.0/ipa
--- freeipa-3.0.0.orig/ipa	2012-11-15 15:13:09.617353890 -0500
+++ freeipa-3.0.0/ipa	2012-11-15 17:26:19.052595147 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Authors:
 #   Jason Gerard DeRose <jderose@redhat.com>
diff -Nru freeipa-3.0.0.orig/ipa-client/ipa-install/ipa-client-automount freeipa-3.0.0/ipa-client/ipa-install/ipa-client-automount
--- freeipa-3.0.0.orig/ipa-client/ipa-install/ipa-client-automount	2012-11-15 15:13:09.364020482 -0500
+++ freeipa-3.0.0/ipa-client/ipa-install/ipa-client-automount	2012-11-15 17:31:14.025986303 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/ipa-client/ipa-install/ipa-client-install freeipa-3.0.0/ipa-client/ipa-install/ipa-client-install
--- freeipa-3.0.0.orig/ipa-client/ipa-install/ipa-client-install	2012-11-15 16:24:59.045035334 -0500
+++ freeipa-3.0.0/ipa-client/ipa-install/ipa-client-install	2012-11-15 17:31:05.055984694 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 # Authors: Simo Sorce <ssorce@redhat.com>
 #          Karl MacMillan <kmacmillan@mentalrootkit.com>
 #
diff -Nru freeipa-3.0.0.orig/ipa-client/py-compile freeipa-3.0.0/ipa-client/py-compile
--- freeipa-3.0.0.orig/ipa-client/py-compile	2012-11-15 15:13:09.360687148 -0500
+++ freeipa-3.0.0/ipa-client/py-compile	2012-11-15 17:33:07.176011529 -0500
@@ -29,7 +29,7 @@
 # <automake-patches@gnu.org>.
 
 if [ -z "$PYTHON" ]; then
-  PYTHON=python
+  PYTHON=python2
 fi
 
 me=py-compile
diff -Nru freeipa-3.0.0.orig/ipalib/x509.py freeipa-3.0.0/ipalib/x509.py
--- freeipa-3.0.0.orig/ipalib/x509.py	2012-11-15 15:13:09.344020476 -0500
+++ freeipa-3.0.0/ipalib/x509.py	2012-11-15 17:32:09.929331825 -0500
@@ -238,7 +238,7 @@
 
 if __name__ == '__main__':
     # this can be run with:
-    # python ipalib/x509.py < /etc/ipa/ca.crt
+    # python2 ipalib/x509.py < /etc/ipa/ca.crt
 
     from ipalib import api
     api.bootstrap()
diff -Nru freeipa-3.0.0.orig/ipapython/Makefile freeipa-3.0.0/ipapython/Makefile
--- freeipa-3.0.0.orig/ipapython/Makefile	2012-11-15 15:13:09.617353890 -0500
+++ freeipa-3.0.0/ipapython/Makefile	2012-11-15 17:37:29.926067739 -0500
@@ -1,4 +1,4 @@
-PYTHONLIBDIR ?= $(shell  python -c "from distutils.sysconfig import *; print get_python_lib()")
+PYTHONLIBDIR ?= $(shell  python2 -c "from distutils.sysconfig import *; print get_python_lib()")
 PACKAGEDIR ?= $(DESTDIR)/$(PYTHONLIBDIR)/ipa
 CONFIGDIR ?= $(DESTDIR)/etc/ipa
 TESTS = $(wildcard test/*.py)
@@ -12,9 +12,9 @@
 
 install:
 	if [ "$(DESTDIR)" = "" ]; then \
-		python setup.py install; \
+		python2 setup.py install; \
 	else \
-		python setup.py install --root $(DESTDIR); \
+		python2 setup.py install --root $(DESTDIR); \
 	fi
 	@for subdir in $(SUBDIRS); do \
 		(cd $$subdir && $(MAKE) $@) || exit 1; \
@@ -42,4 +42,4 @@
 test: $(subst .py,.tst,$(TESTS))
 
 %.tst: %.py
-	python $<
+	python2 $<
diff -Nru freeipa-3.0.0.orig/ipapython/py_default_encoding/Makefile freeipa-3.0.0/ipapython/py_default_encoding/Makefile
--- freeipa-3.0.0.orig/ipapython/py_default_encoding/Makefile	2012-11-15 15:13:09.617353890 -0500
+++ freeipa-3.0.0/ipapython/py_default_encoding/Makefile	2012-11-15 17:37:03.089395314 -0500
@@ -1,15 +1,15 @@
-PYTHONLIBDIR ?= $(shell  python -c "from distutils.sysconfig import *; print get_python_lib()")
+PYTHONLIBDIR ?= $(shell  python2 -c "from distutils.sysconfig import *; print get_python_lib()")
 PACKAGEDIR ?= $(DESTDIR)/$(PYTHONLIBDIR)/ipa
 CONFIGDIR ?= $(DESTDIR)/etc/ipa
 
 all:
-	python setup.py build
+	python2 setup.py build
 
 install:
 	if [ "$(DESTDIR)" = "" ]; then \
-		python setup.py install; \
+		python2 setup.py install; \
 	else \
-		python setup.py install --root $(DESTDIR); \
+		python2 setup.py install --root $(DESTDIR); \
 	fi
 
 clean:
diff -Nru freeipa-3.0.0.orig/ipapython/setup.py freeipa-3.0.0/ipapython/setup.py
--- freeipa-3.0.0.orig/ipapython/setup.py	2012-11-15 15:13:09.614020557 -0500
+++ freeipa-3.0.0/ipapython/setup.py	2012-11-15 17:26:28.759263676 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Copyright (C) 2007  Red Hat
 # see file 'COPYING' for use and warranty information
 #
diff -Nru freeipa-3.0.0.orig/ipapython/setup.py.in freeipa-3.0.0/ipapython/setup.py.in
--- freeipa-3.0.0.orig/ipapython/setup.py.in	2012-11-15 15:13:09.614020557 -0500
+++ freeipa-3.0.0/ipapython/setup.py.in	2012-11-15 17:26:37.205931898 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Copyright (C) 2007  Red Hat
 # see file 'COPYING' for use and warranty information
 #
diff -Nru freeipa-3.0.0.orig/ipapython/test/test_ipautil.py freeipa-3.0.0/ipapython/test/test_ipautil.py
--- freeipa-3.0.0.orig/ipapython/test/test_ipautil.py	2012-11-15 15:13:09.614020557 -0500
+++ freeipa-3.0.0/ipapython/test/test_ipautil.py	2012-11-15 17:26:44.399266593 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 #
 # Copyright (C) 2007    Red Hat
 # see file 'COPYING' for use and warranty information
diff -Nru freeipa-3.0.0.orig/ipapython/test/test_ipavalidate.py freeipa-3.0.0/ipapython/test/test_ipavalidate.py
--- freeipa-3.0.0.orig/ipapython/test/test_ipavalidate.py	2012-11-15 15:13:09.614020557 -0500
+++ freeipa-3.0.0/ipapython/test/test_ipavalidate.py	2012-11-15 17:26:50.729267769 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/python -E
+#! /usr/bin/python2 -E
 #
 # Copyright (C) 2007    Red Hat
 # see file 'COPYING' for use and warranty information
diff -Nru freeipa-3.0.0.orig/ipaserver/install/ipa_ldap_updater.py freeipa-3.0.0/ipaserver/install/ipa_ldap_updater.py
--- freeipa-3.0.0.orig/ipaserver/install/ipa_ldap_updater.py	2012-11-15 15:13:09.607353888 -0500
+++ freeipa-3.0.0/ipaserver/install/ipa_ldap_updater.py	2012-11-15 17:27:03.125937426 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors: Rob Crittenden <rcritten@redhat.com>
 #          Petr Viktorin <pviktori@redhat.com>
 #
diff -Nru freeipa-3.0.0.orig/lite-server.py freeipa-3.0.0/lite-server.py
--- freeipa-3.0.0.orig/lite-server.py	2012-11-15 15:13:09.617353890 -0500
+++ freeipa-3.0.0/lite-server.py	2012-11-15 17:26:22.972595910 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Authors:
 #   Jason Gerard DeRose <jderose@redhat.com>
diff -Nru freeipa-3.0.0.orig/makeapi freeipa-3.0.0/makeapi
--- freeipa-3.0.0.orig/makeapi	2012-11-15 15:13:09.434020500 -0500
+++ freeipa-3.0.0/makeapi	2012-11-15 17:30:11.452640745 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
 #   John Dennis <jdennis@redhat.com>
diff -Nru freeipa-3.0.0.orig/Makefile freeipa-3.0.0/Makefile
--- freeipa-3.0.0.orig/Makefile	2012-11-15 15:13:09.617353890 -0500
+++ freeipa-3.0.0/Makefile	2012-11-15 17:44:56.979508580 -0500
@@ -44,7 +44,7 @@
 LINT_OPTIONS=--no-fail
 endif
 
-PYTHON ?= $(shell rpm -E %__python)
+PYTHON ?= /usr/bin/python2
 
 all: bootstrap-autogen server
 	@for subdir in $(SUBDIRS); do \
diff -Nru freeipa-3.0.0.orig/make-lint freeipa-3.0.0/make-lint
--- freeipa-3.0.0.orig/make-lint	2012-11-15 15:13:09.434020500 -0500
+++ freeipa-3.0.0/make-lint	2012-11-15 17:30:06.652639612 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Authors:
 #   Jakub Hrozek <jhrozek@redhat.com>
@@ -181,7 +181,7 @@
         line = file.readline(128)
         file.close()
 
-        if line[:2] == '#!' and line.find('python') >= 0:
+        if line[:2] == '#!' and line.find('python2') >= 0:
             result.append(filepath)
 
     return result
diff -Nru freeipa-3.0.0.orig/make-test freeipa-3.0.0/make-test
--- freeipa-3.0.0.orig/make-test	2012-11-15 15:13:09.340687141 -0500
+++ freeipa-3.0.0/make-test	2012-11-15 17:37:52.159407513 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 """
 Run IPA unit tests under multiple versions of Python (if present).
@@ -11,7 +11,7 @@
 from subprocess import call
 
 versions = ('2.4', '2.5', '2.6', '2.7')
-python = '/usr/bin/python'
+python = '/usr/bin/python2'
 nose = '/usr/bin/nosetests'
 ran = []
 fail = []
diff -Nru freeipa-3.0.0.orig/make-testcert freeipa-3.0.0/make-testcert
--- freeipa-3.0.0.orig/make-testcert	2012-11-15 15:13:09.340687141 -0500
+++ freeipa-3.0.0/make-testcert	2012-11-15 17:26:03.199258201 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Authors:
 #   Rob Crittenden <rcritten@redhat.com>
diff -Nru freeipa-3.0.0.orig/setup-client.py freeipa-3.0.0/setup-client.py
--- freeipa-3.0.0.orig/setup-client.py	2012-11-15 15:13:09.404020492 -0500
+++ freeipa-3.0.0/setup-client.py	2012-11-15 17:30:15.739308261 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Authors:
 #   Jason Gerard DeRose <jderose@redhat.com>
diff -Nru freeipa-3.0.0.orig/setup.py freeipa-3.0.0/setup.py
--- freeipa-3.0.0.orig/setup.py	2012-11-15 15:13:09.404020492 -0500
+++ freeipa-3.0.0/setup.py	2012-11-15 17:30:19.365975639 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Authors:
 #   Jason Gerard DeRose <jderose@redhat.com>
diff -Nru freeipa-3.0.0.orig/tests/i18n.py freeipa-3.0.0/tests/i18n.py
--- freeipa-3.0.0.orig/tests/i18n.py	2012-11-15 15:13:09.404020492 -0500
+++ freeipa-3.0.0/tests/i18n.py	2012-11-15 17:30:23.495976487 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Authors:
 #   John Dennis <jdennis@redhat.com>
 #
diff -Nru freeipa-3.0.0.orig/tests/test_ipapython/test_dn.py freeipa-3.0.0/tests/test_ipapython/test_dn.py
--- freeipa-3.0.0.orig/tests/test_ipapython/test_dn.py	2012-11-15 15:13:09.374020486 -0500
+++ freeipa-3.0.0/tests/test_ipapython/test_dn.py	2012-11-15 17:30:34.322645128 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 import unittest
 from ipapython.dn import *
