diff -Nru freeipa-3.0.1.orig/ipa-client/ipaclient/ntpconf.py freeipa-3.0.1/ipa-client/ipaclient/ntpconf.py
--- freeipa-3.0.1.orig/ipa-client/ipaclient/ntpconf.py	2012-11-16 14:16:12.578349000 -0500
+++ freeipa-3.0.1/ipa-client/ipaclient/ntpconf.py	2012-11-16 14:16:43.722680000 -0500
@@ -123,9 +123,9 @@
     __write_config(path_ntp_conf, nc)
     ipaservices.restore_context(path_ntp_conf)
 
-    __backup_config(path_ntp_sysconfig, fstore)
-    __write_config(path_ntp_sysconfig, ntp_sysconfig)
-    ipaservices.restore_context(path_ntp_sysconfig)
+    #__backup_config(path_ntp_sysconfig, fstore)
+    #__write_config(path_ntp_sysconfig, ntp_sysconfig)
+    #ipaservices.restore_context(path_ntp_sysconfig)
 
     # Set the ntpd to start on boot
     ipaservices.knownservices.ntpd.enable()
@@ -146,7 +146,7 @@
     if os.path.exists(ntpdate):
         # retry several times -- logic follows /etc/init.d/ntpdate
         # implementation
-        cmd = [ntpdate, "-U", "ntp", "-s", "-b", "-v", server_fqdn]
+        cmd = [ntpdate, "-s", "-b", "-v", server_fqdn]
         for retry in range(0, 3):
             try:
                 ipautil.run(cmd)
