--- a/ipa-client/ipaclient/ntpconf.py
+++ b/ipa-client/ipaclient/ntpconf.py
@@ -110,9 +110,9 @@
     if os.path.exists(path_step_tickers):
         config_step_tickers = True
         ns = ipautil.template_str(ntp_step_tickers, sub_dict)
-        __backup_config(path_step_tickers, fstore)
-        __write_config(path_step_tickers, ns)
-        ipaservices.restore_context(path_step_tickers)
+        #__backup_config(path_step_tickers, fstore)
+        #__write_config(path_step_tickers, ns)
+        #ipaservices.restore_context(path_step_tickers)
 
     if sysstore:
         module = 'ntp'
@@ -147,7 +147,7 @@
     if os.path.exists(ntpdate):
         # retry several times -- logic follows /etc/init.d/ntpdate
         # implementation
-        cmd = [ntpdate, "-U", "ntp", "-s", "-b", "-v", server_fqdn]
+        cmd = [ntpdate, "-s", "-b", "-v", server_fqdn]
         for retry in range(0, 3):
             try:
                 ipautil.run(cmd)
