diff -ur freeipa-2.1.4/ipaserver/ipaldap.py freeipa-2.1.4-new/ipaserver/ipaldap.py
--- freeipa-2.1.4/ipaserver/ipaldap.py	2011-12-06 16:27:06.000000000 +0200
+++ freeipa-2.1.4-new/ipaserver/ipaldap.py	2012-02-03 16:16:30.520563730 +0200
@@ -30,7 +30,7 @@
 import time
 import struct
 import ldap.sasl
-from ldap.controls import LDAPControl,DecodeControlTuples,EncodeControlTuples
+from ldap.controls import LDAPControl,DecodeControlTuples,RequestControlTuples
 from ldap.ldapobject import SimpleLDAPObject
 from ipaserver import ipautil
 from ipalib import errors
