diff -Nur tvmaxe.orig/mpylayer/gen_properties.py tvmaxe/mpylayer/gen_properties.py
--- tvmaxe.orig/mpylayer/gen_properties.py	2008-12-21 01:04:54.000000000 +0200
+++ tvmaxe/mpylayer/gen_properties.py	2010-12-17 22:48:13.210960000 +0200
@@ -7,7 +7,7 @@
 If it changes, you can just pass the name of a text file containing the 
 property table as an argument to this module:
 
-$ python -m mpylayer.gen_properties prop.txt
+$ python2 -m mpylayer.gen_properties prop.txt
 
 and prop_table.py will be generated accordingly.
 
diff -Nur tvmaxe.orig/mpylayer/mpylayer_control.py tvmaxe/mpylayer/mpylayer_control.py
--- tvmaxe.orig/mpylayer/mpylayer_control.py	2010-10-30 04:48:46.000000000 +0300
+++ tvmaxe/mpylayer/mpylayer_control.py	2010-12-17 22:47:19.694293334 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 #### Copyright (c) Clovis Fabricio Costa
 
diff -Nur tvmaxe.orig/tvmaxe.py tvmaxe/tvmaxe.py
--- tvmaxe.orig/tvmaxe.py	2010-12-17 15:17:08.000000000 +0200
+++ tvmaxe/tvmaxe.py	2010-12-17 22:47:19.687626667 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #-*- coding: utf-8 -*-
 
 version = 0.03
diff -Nur tvmaxe.orig/vlc.py tvmaxe/vlc.py
--- tvmaxe.orig/vlc.py	2010-11-02 23:13:22.000000000 +0200
+++ tvmaxe/vlc.py	2010-12-17 22:48:43.694293334 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#!/usr/bin/python2
 
 #
 # Python ctypes bindings for VLC
diff -Nur tvmaxe.orig/which.py tvmaxe/which.py
--- tvmaxe.orig/which.py	2010-11-04 01:04:20.000000000 +0200
+++ tvmaxe/which.py	2010-12-17 22:47:19.687626667 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 def which(program):
     import os
