--- src/vtun-3.0.3/linux/tun_dev.c	2008-01-08 01:36:22.000000000 +0300
+++ new/vtun-3.0.3/linux/tun_dev.c	2013-04-15 20:57:04.100000871 +0400
@@ -87,7 +87,7 @@ static int tun_open_common(char *dev, in
     struct ifreq ifr;
     int fd;
 
-    if ((fd = open("/dev/net/tun", O_RDWR)) < 0)
+    if ((fd = open("/dev/tun", O_RDWR)) < 0)
        return tun_open_common0(dev, istun);
 
     memset(&ifr, 0, sizeof(ifr));
