diff -ur peapod-0.7.2-orig/btclient.py peapod-0.7.2/btclient.py
--- peapod-0.7.2-orig/btclient.py	2008-06-11 03:37:57.000000000 -0500
+++ peapod-0.7.2/btclient.py	2011-02-10 13:57:17.116666677 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # The contents of this file are subject to the BitTorrent Open Source License
 # Version 1.0 (the License).  You may not copy or use this file, in either
diff -ur peapod-0.7.2-orig/Peapod/btclient.py peapod-0.7.2/Peapod/btclient.py
--- peapod-0.7.2-orig/Peapod/btclient.py	2008-06-11 03:37:57.000000000 -0500
+++ peapod-0.7.2/Peapod/btclient.py	2011-02-10 13:57:17.126666678 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # The contents of this file are subject to the BitTorrent Open Source License
 # Version 1.0 (the License).  You may not copy or use this file, in either
diff -ur peapod-0.7.2-orig/Peapod/download.py peapod-0.7.2/Peapod/download.py
--- peapod-0.7.2-orig/Peapod/download.py	2008-06-11 03:37:57.000000000 -0500
+++ peapod-0.7.2/Peapod/download.py	2011-02-10 13:57:17.123333344 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 """
 this module contains the code that actually does the downloading
 
diff -ur peapod-0.7.2-orig/Peapod/peapod.py peapod-0.7.2/Peapod/peapod.py
--- peapod-0.7.2-orig/Peapod/peapod.py	2008-06-11 03:39:12.000000000 -0500
+++ peapod-0.7.2/Peapod/peapod.py	2011-02-10 13:57:17.156666678 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 """
 Python podcast client
 
diff -ur peapod-0.7.2-orig/Peapod/tagging.py peapod-0.7.2/Peapod/tagging.py
--- peapod-0.7.2-orig/Peapod/tagging.py	2008-06-11 03:37:57.000000000 -0500
+++ peapod-0.7.2/Peapod/tagging.py	2011-02-10 13:57:17.146666677 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 import sys
 import string
diff -ur peapod-0.7.2-orig/peapod.py peapod-0.7.2/peapod.py
--- peapod-0.7.2-orig/peapod.py	2008-06-11 03:37:57.000000000 -0500
+++ peapod-0.7.2/peapod.py	2011-02-10 13:57:17.180000010 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 """
 peapod is a command-line podcast downloader
 (C) 2005 Billy Allan.  billy@monkeytwizzle.com
