diff -ur pads-1.2-orig//src/Makefile.in pads-1.2//src/Makefile.in
--- pads-1.2-orig//src/Makefile.in	2005-06-16 05:40:05.000000000 +0200
+++ pads-1.2//src/Makefile.in	2010-12-21 00:22:48.633334281 +0100
@@ -176,7 +176,7 @@
                util.c util.h \
                global.h
 
-pads_LDADD = $(top_srcdir)/lib/bstring/libbstring.a output/liboutput.a
+pads_LDADD = $(top_srcdir)/lib/bstring/libbstring.a output/liboutput.a -lpcre
 bin_SCRIPTS = pads-report
 EXTRA_DIST = pads-report.pl
 SUBDIRS = output
