diff --git a/common/Make.rules b/common/Make.rules
index 3e46eb4..ae651d2 100644
--- a/common/Make.rules
+++ b/common/Make.rules
@@ -213,8 +213,8 @@ list_af_names:
 # manpages
 # =====================
 
-POD2MAN                         = /usr/bin/pod2man
-POD2HTML                        = /usr/bin/pod2html
+POD2MAN                         = pod2man
+POD2HTML                        = pod2html
 MANDIR                          = /usr/share/man
 DOCDIR                          = /usr/share/doc/${NAME}-${VERSION}
 
diff --git a/parser/Makefile b/parser/Makefile
index fb2e14e..a3da763 100644
--- a/parser/Makefile
+++ b/parser/Makefile
@@ -27,7 +27,7 @@ common/Make.rules: $(COMMONDIR)/Make.rules
 endif
 
 DESTDIR=/
-APPARMOR_BIN_PREFIX=${DESTDIR}/lib/apparmor
+APPARMOR_BIN_PREFIX=${DESTDIR}/usr/lib/apparmor
 CONFDIR=/etc/apparmor
 INSTALL_CONFDIR=${DESTDIR}${CONFDIR}
 LOCALEDIR=/usr/share/locale
@@ -343,8 +343,8 @@ install: install-indep install-arch
 
 .PHONY: install-arch
 install-arch: $(INSTALLDEPS)
-	install -m 755 -d $(DESTDIR)/sbin
-	install -m 755 ${TOOLS} $(DESTDIR)/sbin
+	install -m 755 -d $(DESTDIR)/usr/bin
+	install -m 755 ${TOOLS} $(DESTDIR)/usr/bin
 
 .PHONY: install-indep
 install-indep:
diff --git a/parser/tst/Makefile b/parser/tst/Makefile
index d254809..7cb14ec 100644
--- a/parser/tst/Makefile
+++ b/parser/tst/Makefile
@@ -1,5 +1,5 @@
 #
-PROVE=/usr/bin/prove
+PROVE=prove
 TESTS=simple.pl
 PARSER_DIR=..
 PARSER_BIN=apparmor_parser
diff --git a/utils/Makefile b/utils/Makefile
index 3d85aaf..8055cf4 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -42,7 +42,7 @@ all: ${MANPAGES} ${HTMLMANPAGES}
 
 # need some better way of determining this
 DESTDIR=/
-BINDIR=${DESTDIR}/usr/sbin
+BINDIR=${DESTDIR}/usr/bin
 CONFDIR=${DESTDIR}/etc/apparmor
 PYPREFIX=/usr
 
