diff --git a/apparmor/changehat/pam_apparmor/Makefile b/apparmor/changehat/pam_apparmor/Makefile
index f234f0e..3faa70a 100644
--- a/apparmor/changehat/pam_apparmor/Makefile
+++ b/apparmor/changehat/pam_apparmor/Makefile
@@ -80,7 +80,7 @@ $(NAME).so: ${OBJECTS}
 
 # need some better way of determining this
 DESTDIR=/
-SECDIR ?= ${DESTDIR}/lib/security
+SECDIR ?= ${DESTDIR}/usr/lib/security
 
 .PHONY: install
 install: $(NAME).so
