--- Coob.csproj.old	2013-07-23 14:04:50.000000000 -0700
+++ Coob.csproj	2013-07-23 14:06:37.079105471 -0700
@@ -39,7 +39,7 @@
       <HintPath>.\Ionic.Zlib.dll</HintPath>
     </Reference>
     <Reference Include="Jint">
-      <HintPath>..\..\..\..\Libs\Jint\Jint.dll</HintPath>
+      <HintPath>.\Jint.dll</HintPath>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
@@ -102,11 +102,11 @@
   </ItemGroup>
   <ItemGroup />
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
+       Other similar extension points exist, see Microsoft.Common.targets.
+  <Target Name="BeforeBuild">
+  </Target>
+  <Target Name="AfterBuild">
+  </Target>
   -->
 </Project>
\ No newline at end of file
