--- umurmur-0.2.6-old/src/Makefile	2010-03-24 21:53:23.000000000 +0100
+++ umurmur-0.2.6/src/Makefile	        2010-07-17 17:50:54.919740242 +0200
@@ -15,11 +15,11 @@
 
 # Build support and link to PolarSSL
 # If PolarSSL is in a nonstandard place.
-# POLARSSL_LIB:=../../../polarssl/polarssl-0.12.1/library/libpolarssl.a
-# SSL_LIB:=$(POLARSSL_LIB) # If statically linking
-# POLARSSL_INCLUDE:=../../../polarssl/polarssl-0.12.1/include
-# EXTRA_CFLAGS:=-DUSE_POLARSSL -I$(POLARSSL_INCLUDE)
-# EXTRA_LDLAGS:=-lpolarssl
+ POLARSSL_LIB:=/usr/lib/libpolarssl.a
+ SSL_LIB:=$(POLARSSL_LIB) # If statically linking
+ POLARSSL_INCLUDE:=/usr/include/polarssl
+ EXTRA_CFLAGS:=-DUSE_POLARSSL -I$(POLARSSL_INCLUDE)
+ EXTRA_LDLAGS:=-lpolarssl
 
 
 # OpenSSL - usually installed at a standard place
