From 3e40cae5eec96bdc07cbe12692436edf97fbb24e Mon Sep 17 00:00:00 2001
From: Paul Gideon Dann <pdgiddie@gmail.com>
Date: Wed, 20 May 2009 13:17:29 +0100
Subject: [PATCH 2/4] Setting default Qt4 backend to Cairo

---
 qt4/src/poppler-private.cc |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qt4/src/poppler-private.cc b/qt4/src/poppler-private.cc
index ffb5b92..917b7b5 100644
--- a/qt4/src/poppler-private.cc
+++ b/qt4/src/poppler-private.cc
@@ -240,7 +240,7 @@ namespace Debug {
     void DocumentData::init()
     {
         m_fontInfoIterator = 0;
-        m_backend = Document::SplashBackend;
+        m_backend = Document::CairoBackend;
         m_outputDev = 0;
         paperColor = Qt::white;
         m_hints = 0;
-- 
1.7.10.2

