diff -Nurw src/python-fam-1.1.1/_fam.c src2/python-fam-1.1.1/_fam.c
--- src/python-fam-1.1.1/_fam.c	2005-04-10 07:48:45.000000000 +0000
+++ src2/python-fam-1.1.1/_fam.c	2008-02-26 09:37:10.000000000 +0000
@@ -112,7 +112,7 @@
   fprintf(stderr, "_fam: deleting connection object.\n");
 #endif
   _fam_close(self, NULL);
-  PyMem_DEL(self);
+  PyObject_Del(self);
 }
 
 static PyTypeObject _fam_connection_type = {
@@ -215,7 +215,7 @@
   fprintf(stderr, "_fam: deleting request object.\n");
 #endif
   _fam_cancel_monitor(self, NULL);
-  PyMem_DEL(self);
+  PyObject_Del(self);
 }
 
 static PyTypeObject _fam_request_type = {
@@ -357,7 +357,7 @@
       fe_obj->fc_obj = NULL;
     }
     Py_XDECREF(fe_obj->attr);
-    PyMem_DEL(fe_obj);
+    PyObject_Del(fe_obj);
   }
 }
 
@@ -435,7 +435,7 @@
 
   int rc;
   
-  _fam_connection_object *fc_obj = PyObject_NEW(_fam_connection_object, &_fam_connection_type);
+  _fam_connection_object *fc_obj = PyObject_New(_fam_connection_object, &_fam_connection_type);
   if (!fc_obj)
     return NULL;
 
@@ -498,7 +498,7 @@
     if (!PyArg_ParseTuple(args, "sO", &filename, &userData))
       return NULL;
 
-    fr_obj = PyObject_NEW(_fam_request_object, &_fam_request_type);
+    fr_obj = PyObject_New(_fam_request_object, &_fam_request_type);
     if (!fr_obj)
       return NULL;
 
@@ -545,7 +545,7 @@
     if (!PyArg_ParseTuple(args, "sO", &filename, &userData))
       return NULL;
 
-    fr_obj = PyObject_NEW(_fam_request_object, &_fam_request_type);
+    fr_obj = PyObject_New(_fam_request_object, &_fam_request_type);
     if (!fr_obj)
       return NULL;
 
@@ -647,7 +647,7 @@
 
   if (fc_obj->fc) {
 
-    fe_obj = PyObject_NEW(_fam_event_object, &_fam_event_type);
+    fe_obj = PyObject_New(_fam_event_object, &_fam_event_type);
     if (!fe_obj)
       return NULL;
 
Binary files src/python-fam-1.1.1/build/lib.linux-i686-2.5/_fam.so and src2/python-fam-1.1.1/build/lib.linux-i686-2.5/_fam.so differ
Binary files src/python-fam-1.1.1/build/temp.linux-i686-2.5/_fam.o and src2/python-fam-1.1.1/build/temp.linux-i686-2.5/_fam.o differ
diff -Nurw src/python-fam-1.1.1/test.py src2/python-fam-1.1.1/test.py
--- src/python-fam-1.1.1/test.py	2005-04-10 07:48:45.000000000 +0000
+++ src2/python-fam-1.1.1/test.py	2008-02-26 09:41:24.000000000 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.2
+#!/usr/bin/python
 # This file is part of Python FAM.
 # Copyright (C) 2002 Robert Sander <robert.sander@epigenomics.com>
 # 
diff -Nurw src/python-fam-1.1.1/test2.py src2/python-fam-1.1.1/test2.py
--- src/python-fam-1.1.1/test2.py	2005-04-10 07:48:45.000000000 +0000
+++ src2/python-fam-1.1.1/test2.py	2008-02-26 09:41:31.000000000 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.2
+#!/usr/bin/python
 # This file is part of Python FAM.
 # Copyright (C) 2002 Robert Sander <robert.sander@epigenomics.com>
 # 
