diff -Naur ./gui/PluginManager.cpp ../../retroshare-gui.patched/src//gui/PluginManager.cpp
--- ./gui/PluginManager.cpp	2010-03-12 21:15:27.000000000 +0100
+++ ../../retroshare-gui.patched/src//gui/PluginManager.cpp	2010-04-16 21:02:38.113407976 +0200
@@ -42,7 +42,7 @@
 PluginManager::PluginManager()
 {
     baseFolder =  //qApp->applicationDirPath()+"///plugins" ;
-Rshare::dataDirectory() + "/plugins" ;
+"/usr/lib/retroshare/plugins" ;
     lastError = "No error.";
 
     viewWidget = 0;
diff -Naur ./RetroShare.pro ../../retroshare-gui.patched/src//RetroShare.pro
--- ./RetroShare.pro	2010-03-26 16:31:06.000000000 +0100
+++ ../../retroshare-gui.patched/src//RetroShare.pro	2010-04-16 21:02:02.623156063 +0200
@@ -1,10 +1,10 @@
-CONFIG += qt gui uic qrc resources uitools newsettings
+CONFIG += qt gui uic qrc resources uitools newsettings pluginmgr blogs
 QT     += network xml script opengl
 
 TEMPLATE = app
 TARGET = RetroShare
 
-DEFINES += RS_RELEASE_VERSION
+# DEFINES += RS_RELEASE_VERSION
 RCC_DIR = temp/qrc
 UI_DIR  = temp/ui
 MOC_DIR = temp/moc
