diff -Naur v0.5.0/libretroshare/src/upnp/UPnPBase.h v0.5.0.patched//libretroshare/src/upnp/UPnPBase.h
--- v0.5.0.unpatched/libretroshare/src/upnp/UPnPBase.h	2011-01-03 18:50:48.067689656 +0100
+++ v0.5.0/libretroshare/src/upnp/UPnPBase.h	2011-01-03 18:52:03.996681493 +0100
@@ -25,7 +25,7 @@
 //This file uses libupnp
 
 #include <map>
-#include <string>
+#include <string.h>
 #include <sstream>
 #include <memory>
 #include <vector>
diff -Naur v0.5.0/retroshare-gui/src/gui/NetworkDialog.cpp v0.5.0.patched//retroshare-gui/src/gui/NetworkDialog.cpp
--- v0.5.0.unpatched/retroshare-gui/src/gui/NetworkDialog.cpp	2011-01-03 18:50:48.091024727 +0100
+++ v0.5.0/retroshare-gui/src/gui/NetworkDialog.cpp	2011-01-03 18:52:18.611103946 +0100
@@ -782,29 +782,29 @@
              
       if(config.netUpnpOk)
       {
-         ui.iconlabel_upnp->setPixmap(QPixmap::QPixmap(":/images/ledon1.png"));
+         ui.iconlabel_upnp->setPixmap(QPixmap(":/images/ledon1.png"));
       }
       else
       {    
-         ui.iconlabel_upnp->setPixmap(QPixmap::QPixmap(":/images/ledoff1.png"));
+         ui.iconlabel_upnp->setPixmap(QPixmap(":/images/ledoff1.png"));
       }
                               
       if (config.netLocalOk)
       {
-	  ui.iconlabel_netLimited->setPixmap(QPixmap::QPixmap(":/images/ledon1.png"));
+	  ui.iconlabel_netLimited->setPixmap(QPixmap(":/images/ledon1.png"));
       }
       else
       {          
-          ui.iconlabel_netLimited->setPixmap(QPixmap::QPixmap(":/images/ledoff1.png"));
+          ui.iconlabel_netLimited->setPixmap(QPixmap(":/images/ledoff1.png"));
       }
 
       if (config.netExtraAddressOk)
       {
-	  ui.iconlabel_ext->setPixmap(QPixmap::QPixmap(":/images/ledon1.png"));
+	  ui.iconlabel_ext->setPixmap(QPixmap(":/images/ledon1.png"));
       }
       else
       {
-	  ui.iconlabel_ext->setPixmap(QPixmap::QPixmap(":/images/ledoff1.png"));
+	  ui.iconlabel_ext->setPixmap(QPixmap(":/images/ledoff1.png"));
       }
 
     rsiface->unlockData(); /* UnLock Interface */
diff -Naur v0.5.0/retroshare-gui/src/gui/PeersDialog.cpp v0.5.0.patched//retroshare-gui/src/gui/PeersDialog.cpp
--- v0.5.0.unpatched/retroshare-gui/src/gui/PeersDialog.cpp	2011-01-03 18:50:48.094358309 +0100
+++ v0.5.0/retroshare-gui/src/gui/PeersDialog.cpp	2011-01-03 18:52:18.611103946 +0100
@@ -284,7 +284,7 @@
       hbox->setSpacing(6);
     
       iconLabel = new QLabel( this );
-      iconLabel->setPixmap(QPixmap::QPixmap(":/images/user/friends24.png"));
+      iconLabel->setPixmap(QPixmap(":/images/user/friends24.png"));
       iconLabel->setMaximumSize( iconLabel->frameSize().height() + 24, 24 );
       hbox->addWidget(iconLabel);
        
diff -Naur v0.5.0/retroshare-gui/src/gui/statusbar/hashingstatus.cpp v0.5.0.patched//retroshare-gui/src/gui/statusbar/hashingstatus.cpp
--- v0.5.0.unpatched/retroshare-gui/src/gui/statusbar/hashingstatus.cpp	2011-01-03 18:50:48.097691891 +0100
+++ v0.5.0/retroshare-gui/src/gui/statusbar/hashingstatus.cpp	2011-01-03 18:52:18.611103946 +0100
@@ -44,7 +44,7 @@
     hbox->setSpacing(6);
         
     //iconLabel = new QLabel( this );
-    //iconLabel->setPixmap(QPixmap::QPixmap(":/images/user/identitygray16.png"));
+    //iconLabel->setPixmap(QPixmap(":/images/user/identitygray16.png"));
     // iconLabel doesn't change over time, so we didn't need a minimum size
     //hbox->addWidget(iconLabel);
     
