--- include/pyamanith.h	2006-03-28 06:43:22.000000000 +1000
+++ include/pyamanith.h	2008-04-05 01:55:33.000000000 +1000
@@ -59,7 +59,7 @@
 #include <amanith/numerics/geigen.h>
 #include <amanith/numerics/gfilter.h>
 #include <amanith/numerics/gintegration.h>
-#include <amanith/rendering/gdrawboard.h>
+#include <amanith/rendering/gdrawboard.swig.h>
 #include <amanith/rendering/gdrawstyle.h>
 #include <amanith/rendering/gopenglboard.h>
 #include <amanith/support/gavltree.h>
--- setup.py	2007-11-19 07:10:03.000000000 +1000
+++ setup.py	2008-04-05 01:55:33.000000000 +1000
@@ -16,7 +16,7 @@
 inc = [join("changes", "include"), join(amanith_base, "include"), "include"]
 lib = [join(amanith_base, "lib")]
 cc_args = []
-swig_args = ["-Ibuild/amanith"]
+swig_args = ["-Ichanges/include", "-Iinclude", "-I/usr/include"]
 
 if os.name == "nt":
   if debug:
@@ -38,20 +38,9 @@
         relPath = join(root, file).replace(source + os.sep, "")
         shutil.copy(join(source, relPath), join(target, relPath))
   
-  def run(self):
-    # Ultimately this should use patch(1) or something, but for now
-    # we just copy the stuff over.
-    if not os.path.isdir(join("build", "amanith")):
-      os.makedirs(join("build", "amanith"))
-      shutil.copytree(join(amanith_base, "include"), join("build", "amanith", "include"))
-    self.copy("changes", join("build", "amanith"))
-    _build_ext.run(self)
 
 ext_amanith = Extension('_amanith', ['amanith.i'],
-                        include_dirs = inc,
-                        library_dirs = lib,
-                        libraries = libs,
-                        extra_compile_args = cc_args)
+                        swig_opts = swig_args, extra_compile_args = swig_args, libraries = libs)
                         
 setup(name = 'PyAmanith',
       version = '0.3.35',
