From a13e114c5be6d9d9558686a90fd5a778f55318f3 Mon Sep 17 00:00:00 2001
From: xantares <xantares09@hotmail.com>
Date: Tue, 26 Aug 2014 10:24:51 +0000
Subject: [PATCH 1/5] cmake: force HAVE_TIMERSUB

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bd7bcc6..86c9c85 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,6 +77,8 @@ set (CMAKE_REQUIRED_INCLUDES "sys/time.h")
 #we could use CMakePushCheckState to push / pop CMAKE_REQUIRED_INCLUDES
 #but I think it has no influence on the rest of the build...
 check_function_exists (timersub HAVE_TIMERSUB)
+message(WARNING "Forcing HAVE_TIMERSUB")
+set(HAVE_TIMERSUB 1)
 
 ###### Platform-specific function checks
 
@@ -174,4 +176,4 @@ set (CPACK_SOURCE_IGNORE_FILES "xcf"
 set (CPACK_INCLUDE_TOPLEVEL_DIRECTORY 1)
 
 # I think this has to stay near the end, after CPack options
-include (CPack)
\ No newline at end of file
+include (CPack)
-- 
2.1.0

