From 10b0dc9ed7aefd568bf8e55559f25af36586d656 Mon Sep 17 00:00:00 2001
From: xantares <xantares09@hotmail.com>
Date: Tue, 26 Aug 2014 10:32:55 +0000
Subject: [PATCH 4/5] cmake: fix missing scantool_diag.c

---
 scantool/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scantool/CMakeLists.txt b/scantool/CMakeLists.txt
index b80af58..24c88bf 100644
--- a/scantool/CMakeLists.txt
+++ b/scantool/CMakeLists.txt
@@ -54,7 +54,7 @@ set (LIBDIAG_SRCS diag_l0_me.c diag_l0_br.c diag_l0_elm.c diag_l0_sim.c
 set (LIBDYNO_SRCS dyno.c)
 set (DIAGTEST_SRCS diag_test.c)
 set (SCANTOOL_SRCS scantool.c scantool_cli.c ${ST_DIAG} scantool_set.c
-	scantool_test.c scantool_debug.c scantool_vag.c scantool_dyno.c
+	scantool_test.c scantool_diag.c scantool_debug.c scantool_vag.c scantool_dyno.c
 	scantool_aif.c)
 #and GLOB all the headers. This is *only* so that the headers end up in
 #the file list of IDE projects (at least Code::blocks)
-- 
2.1.0

