diff -rwup volumeicon-0.3.0/Makefile.in ./Makefile.in
--- volumeicon-0.3.0/Makefile.in	2011-05-09 17:49:35.000000000 -0400
+++ ./Makefile.in	2011-05-09 17:04:08.000000000 -0400
@@ -126,6 +126,8 @@ INSTALL_DATA = @INSTALL_DATA@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
+KEYBINDER_CFLAGS = @KEYBINDER_CFLAGS@
+KEYBINDER_LIBS = @KEYBINDER_LIBS@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
Only in .: autom4te.cache
Only in .: autoscan.log
Only in volumeicon-0.3.0: config.log
Only in volumeicon-0.3.0: configure
diff -rwup volumeicon-0.3.0/configure.ac ./configure.ac
--- volumeicon-0.3.0/configure.ac	2011-05-09 17:49:35.000000000 -0400
+++ ./configure.ac	2011-05-09 17:03:48.000000000 -0400
@@ -26,6 +26,10 @@ PKG_CHECK_MODULES([GTK], [gtk+-2.0 >= 2.
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
+PKG_CHECK_MODULES([KEYBINDER], [keybinder])
+AC_SUBST(KEYBINDER_CFLAGS)
+AC_SUBST(KEYBINDER_LIBS)
+
 
 if test "x${oss}" = xno; then
 # Check for alsa
Only in .: configure.scan
diff -rwup volumeicon-0.3.0/data/Makefile.in ./data/Makefile.in
--- volumeicon-0.3.0/data/Makefile.in	2011-05-09 17:49:35.000000000 -0400
+++ ./data/Makefile.in	2011-05-09 17:04:08.000000000 -0400
@@ -72,6 +72,8 @@ INSTALL_DATA = @INSTALL_DATA@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
+KEYBINDER_CFLAGS = @KEYBINDER_CFLAGS@
+KEYBINDER_LIBS = @KEYBINDER_LIBS@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
Only in .: hotkeys.patch
diff -rwup volumeicon-0.3.0/src/Makefile.am ./src/Makefile.am
--- volumeicon-0.3.0/src/Makefile.am	2011-05-09 17:49:35.000000000 -0400
+++ ./src/Makefile.am	2011-05-09 17:04:55.000000000 -0400
@@ -1,7 +1,7 @@
 AM_CFLAGS = -Wall -DDATADIR=\"@datadir@/volumeicon\"
-AM_CFLAGS += @GTK_CFLAGS@ @ALSA_CFLAGS@ @OSS_CFLAGS@
+AM_CFLAGS += @GTK_CFLAGS@ @ALSA_CFLAGS@ @OSS_CFLAGS@ @KEYBINDER_CFLAGS@
 
-LIBS = @GTK_LIBS@ @ALSA_LIBS@
+LIBS = @GTK_LIBS@ @ALSA_LIBS@ @KEYBINDER_LIBS@
 
 bin_PROGRAMS = volumeicon
 
diff -rwup volumeicon-0.3.0/src/Makefile.in ./src/Makefile.in
--- volumeicon-0.3.0/src/Makefile.in	2011-05-09 17:49:35.000000000 -0400
+++ ./src/Makefile.in	2011-05-09 17:04:58.000000000 -0400
@@ -94,9 +94,11 @@ INSTALL_DATA = @INSTALL_DATA@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
+KEYBINDER_CFLAGS = @KEYBINDER_CFLAGS@
+KEYBINDER_LIBS = @KEYBINDER_LIBS@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @GTK_LIBS@ @ALSA_LIBS@
+LIBS = @GTK_LIBS@ @ALSA_LIBS@ @KEYBINDER_LIBS@
 LTLIBOBJS = @LTLIBOBJS@
 MAKEINFO = @MAKEINFO@
 MKDIR_P = @MKDIR_P@
@@ -160,7 +162,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AM_CFLAGS = -Wall -DDATADIR=\"@datadir@/volumeicon\" @GTK_CFLAGS@ \
-	@ALSA_CFLAGS@ @OSS_CFLAGS@ $(am__empty)
+	@ALSA_CFLAGS@ @OSS_CFLAGS@ @KEYBINDER_CFLAGS@ $(am__empty)
 @ENABLE_OSS_FALSE@BACKEND = alsa_backend.c alsa_backend.h
 @ENABLE_OSS_TRUE@BACKEND = oss_backend.c oss_backend.h
 volumeicon_SOURCES = \
diff -rwup volumeicon-0.3.0/src/config.c ./src/config.c
--- volumeicon-0.3.0/src/config.c	2011-05-09 17:49:35.000000000 -0400
+++ ./src/config.c	2011-05-09 17:32:34.000000000 -0400
@@ -40,6 +40,9 @@ static char * m_config_file = NULL;
 static char * m_helper_program = NULL;
 static gchar * m_theme = NULL;
 static gchar * m_channel = NULL;
+static gchar * m_up = NULL;
+static gchar * m_down = NULL;
+static gchar * m_mute = NULL;
 static int m_stepsize = 0;
 static gboolean m_lmb_slider = FALSE;
 static gboolean m_mmb_mute = FALSE;
@@ -76,6 +79,9 @@ static void config_read()
 	m_theme = g_key_file_get_value(kf, "StatusIcon", "theme", NULL);
 	m_lmb_slider = g_key_file_get_boolean(kf, "StatusIcon", "lmb_slider", NULL);
 	m_mmb_mute = g_key_file_get_boolean(kf, "StatusIcon", "mmb_mute", NULL);
+    m_up = g_key_file_get_value(kf, "Hotkeys", "up", NULL);
+    m_down = g_key_file_get_value(kf, "Hotkeys", "down", NULL);
+    m_mute = g_key_file_get_value(kf, "Hotkeys", "mute", NULL);
 	g_key_file_free(kf);
 
 	// Load default values for unset keys
@@ -153,11 +159,29 @@ gboolean config_get_middle_mouse_mute()
 	return m_mmb_mute;
 }
 
+const gchar * config_get_hotkey_up()
+{
+    return m_up;
+}
+
+const gchar * config_get_hotkey_down()
+{
+    return m_down;
+}
+
+const gchar * config_get_hotkey_mute()
+{
+    return m_mute;
+}
+
 void config_write()
 {
 	assert(m_config_file != NULL);
 
 	GKeyFile * kf = g_key_file_new();
+    g_key_file_set_value(kf, "Hotkeys", "up", "XF86_AudioRaiseVolume");
+    g_key_file_set_value(kf, "Hotkeys", "down", "XF86_AudioLowerVolume");
+    g_key_file_set_value(kf, "Hotkeys", "mute", "XF86_AudioMute");
 	g_key_file_set_integer(kf, "StatusIcon", "stepsize", m_stepsize);
 	g_key_file_set_boolean(kf, "StatusIcon", "lmb_slider", m_lmb_slider);
 	g_key_file_set_boolean(kf, "StatusIcon", "mmb_mute", m_mmb_mute);
diff -rwup volumeicon-0.3.0/src/config.h ./src/config.h
--- volumeicon-0.3.0/src/config.h	2011-05-09 17:49:35.000000000 -0400
+++ ./src/config.h	2011-05-09 17:33:04.000000000 -0400
@@ -41,5 +41,8 @@ gboolean config_get_use_gtk_theme();
 gboolean config_get_left_mouse_slider();
 gboolean config_get_middle_mouse_mute();
 int config_get_stepsize();
+const gchar * config_get_hotkey_up();
+const gchar * config_get_hotkey_down();
+const gchar * config_get_hotkey_mute();
 
 #endif
diff -rwup volumeicon-0.3.0/src/volumeicon.c ./src/volumeicon.c
--- volumeicon-0.3.0/src/volumeicon.c	2011-05-09 17:49:35.000000000 -0400
+++ ./src/volumeicon.c	2011-05-09 18:31:08.000000000 -0400
@@ -21,10 +21,12 @@
 // with this program.  If not, see <http://www.gnu.org/licenses/>.
 //##############################################################################
 
+#include <stdio.h>
 #include <stdlib.h>
 #include <assert.h>
 #include <gtk/gtk.h>
 #include <glib/gstdio.h>
+#include <keybinder.h>
 #include <signal.h>
 #include <unistd.h>
 
@@ -35,6 +37,13 @@
 #endif
 #include "config.h"
 
+enum HOTKEY
+{
+    UP,
+    DOWN,
+    MUTE
+};
+
 //##############################################################################
 // Definitions
 //##############################################################################
@@ -642,6 +651,24 @@ static void scale_setup()
 		G_CALLBACK(scale_value_changed), NULL);
 }
 
+void handle_hotkey(const char *keystring, void *user_data)
+{
+    if ((enum HOTKEY)user_data != MUTE)
+    {
+        int step, volume;
+        step = config_get_stepsize();
+        volume = backend_get_volume() + (user_data == UP ? step : -step);
+        backend_set_volume((volume > 100 ? 100 : (volume < 0 ? 0 : volume)));
+        status_icon_update(FALSE, FALSE);
+    }
+    else {
+        int mute = !backend_get_mute();
+        backend_set_mute(mute);
+        status_icon_update(mute, FALSE);
+    }
+    scale_update();
+}
+
 //##############################################################################
 // Exported functions
 //##############################################################################
@@ -680,6 +707,11 @@ int main(int argc, char * argv[])
 	status_icon_setup(m_mute);
 	scale_setup();
 
+    keybinder_init();
+    keybinder_bind(config_get_hotkey_up(), handle_hotkey, (void *)UP);
+    keybinder_bind(config_get_hotkey_down(), handle_hotkey, (void *)DOWN);
+    keybinder_bind(config_get_hotkey_mute(), handle_hotkey, (void *)MUTE);
+
 	// Main Loop
 	gtk_main();
 
Only in .: volumeicon-0.3.0
