diff -ur pianobarfly-build/.git/config pianobarfly-build/.git/config
diff -ur pianobarfly-build/src/fly_id3.c pianobarfly-build/src/fly_id3.c
--- pianobarfly-build/src/fly_id3.c	2012-03-13 18:35:23.796005228 -0400
+++ pianobarfly-build/src/fly_id3.c	2012-03-13 18:36:14.666772707 -0400
@@ -39,6 +39,11 @@
 #include "fly_id3.h"
 #include "ui.h"
 
+/**
+ * Name of the temporary file used in tagging.
+ */
+#define BAR_FLY_TMP_MP3_FILE_NAME "pianobarfly-tmp.mp3"
+
 int BarFlyID3AddCover(struct id3_tag* tag, uint8_t const* cover_art,
 		size_t cover_size, BarSettings_t const* settings)
 {
@@ -228,15 +233,13 @@
 		BarSettings_t const* settings)
 {
 	int exit_status = 0;
-	int status_int;
+	int status;
 	id3_length_t size1;
 	id3_length_t size2;
 	id3_byte_t* tag_buffer = NULL;
 	FILE* audio_file = NULL;
 	FILE* tmp_file = NULL;
 	uint8_t audio_buffer[BAR_FLY_COPY_BLOCK_SIZE];
-	char tmp_file_path[L_tmpnam];
-	char* junk;
 	size_t read_count;
 	size_t write_count;
 
@@ -286,16 +289,11 @@
 
 	/*
 	 * Open the tmp file.
-	 *
-	 * Assigning the return value of tmpnam() to a junk pointer to get the
-	 * compiler to be quiet.
-	 */
-	junk = tmpnam(tmp_file_path);
-	junk = junk;
-	tmp_file = fopen(tmp_file_path, "w+b");
+	 */
+	tmp_file = fopen(BAR_FLY_TMP_MP3_FILE_NAME, "w+b");
 	if (tmp_file == NULL) {
 		BarUiMsg(settings, MSG_ERR, "Could not open the temporary file (%s) "
-				"(%d:%s).\n", tmp_file_path, errno, strerror(errno));
+				"(%d:%s).\n", BAR_FLY_TMP_MP3_FILE_NAME, errno, strerror(errno));
 		goto error;
 	}
 
@@ -305,7 +303,7 @@
 	write_count = fwrite(tag_buffer, 1, size2, tmp_file);
 	if (write_count != size2) {
 		BarUiMsg(settings, MSG_ERR, "Could not write the tag to the file (%s) "
-				"(%d:%s).\n", tmp_file_path, errno, strerror(errno));
+				"(%d:%s).\n", BAR_FLY_TMP_MP3_FILE_NAME, errno, strerror(errno));
 		goto error;
 	}
 
@@ -326,7 +324,7 @@
 		write_count = fwrite(audio_buffer, 1, read_count, tmp_file);
 		if (write_count != read_count) {
 			BarUiMsg(settings, MSG_ERR, "Failed to write to the tmp file "
-					"(%s).\n", tmp_file_path);
+					"(%s).\n", BAR_FLY_TMP_MP3_FILE_NAME);
 			goto error;
 		}
 	}
@@ -344,8 +342,8 @@
 	/*
 	 * Overwrite the audio file with the tmp file.
 	 */
-	status_int = rename(tmp_file_path, file_path);
-	if (status_int != 0) {
+	status = rename(BAR_FLY_TMP_MP3_FILE_NAME, file_path);
+	if (status != 0) {
 		BarUiMsg(settings, MSG_ERR, "Could not overwrite the audio file "
 				"(%d:%s).\n", errno, strerror(errno));
 		goto error;
@@ -357,7 +355,7 @@
 	/*
 	 * Delete the tmp file if it exists.
 	 */
-	unlink(tmp_file_path);
+	unlink(BAR_FLY_TMP_MP3_FILE_NAME);
 
 	exit_status = -1;
 
diff -ur pianobarfly-build/src/fly_mp4.c pianobarfly-build/src/fly_mp4.c
--- pianobarfly-build/src/fly_mp4.c	2012-03-13 18:35:23.796005228 -0400
+++ pianobarfly-build/src/fly_mp4.c	2012-03-13 18:36:14.666772707 -0400
@@ -138,6 +138,11 @@
 #define BAR_FLY_MP4_ATOM_YEAR_CLASS {0x00, 0x00, 0x00, 0x01}
 
 /**
+ * Name of the tmp file used in tagging.
+ */
+#define BAR_FLY_TMP_MP4_FILE_NAME "pianobarfly-tmp.m4a"
+
+/**
  * The MP4 atom structure.  This structure represents a single atom in an MP4 
  * file.
  */
@@ -1902,8 +1907,6 @@
 	size_t audio_buf_size;
 	size_t read_count;
 	size_t write_count;
-	char tmp_file_path[L_tmpnam];
-	char* junk;
 	size_t atom_size;
 	BarFlyMp4Atom_t* atom;
 
@@ -1912,20 +1915,19 @@
 
 	/*
 	 * Open the tmp file.
-	 *
-	 * Assigning the return value of tmpnam() to a junk pointer to get the
-	 * compiler to be quiet.
 	 */
-	junk = tmpnam(tmp_file_path);
-	junk = junk;
-	tmp_file = fopen(tmp_file_path, "wb");
+	tmp_file = fopen(BAR_FLY_TMP_MP4_FILE_NAME, "wb");
 	if (tmp_file == NULL) {
 		BarUiMsg(settings, MSG_ERR,
 				"Error opening the temporary file (%s) (%d:%s).\n",
-				tmp_file_path, errno, strerror(errno));
+				BAR_FLY_TMP_MP4_FILE_NAME, errno, strerror(errno));
 		goto error;
 	}
 
+	/*
+	 * Copy everything up to the start of the moov atom from the MP4 file to the
+	 * tmp file.
+	 */
 	status = fseek(tag->mp4_file, 0, SEEK_SET);
 	if (status != 0) {
 		BarUiMsg(settings, MSG_ERR, "Error seeking in the file (%d:%s).\n",
@@ -1933,10 +1935,6 @@
 		goto error;
 	}
 
-	/*
-	 * Copy everything up to the start of the moov atom from the MP4 file to the
-	 * tmp file.
-	 */
 	atom = _BarFlyMp4TagFindAtom(tag, "moov");
 	if (atom == NULL) {
 		goto error;
@@ -2025,7 +2023,7 @@
 	fclose(tag->mp4_file);
 	tag->mp4_file = NULL;
 
-	status = rename(tmp_file_path, tag->file_path);
+	status = rename(BAR_FLY_TMP_MP4_FILE_NAME, tag->file_path);
 	if (status != 0) {
 		BarUiMsg(settings, MSG_ERR, "Error overwriting the MP4 file (%d:%s).\n",
 				errno, strerror(errno));
@@ -2035,6 +2033,11 @@
 	goto end;
 
 error:
+	/*
+	 * Delete the tmp file if it exists.
+	 */
+	unlink(BAR_FLY_TMP_MP4_FILE_NAME);
+
 	exit_status = -1;
 
 end:
