--- build/SConscript.configure_org	2012-01-13 17:22:56.752747258 +0100
+++ build/SConscript.configure	2012-01-13 17:24:42.440670512 +0100
@@ -23,13 +23,6 @@
 	print '\nA configuration file must be selected! Have a look at http://www.mitsuba-renderer.org/docs.html'
 	Exit(1)
 
-if not os.path.exists(GetBuildPath('#dependencies')):
-	print '\nThe build dependency files are missing -- please check them out'
-	print 'from Mercurial by running the following command *inside* the '
-	print 'Mitsuba directory:\n'
-	print '$ hg clone https://www.mitsuba-renderer.org/hg/dependencies\n'
-	Exit(1)
-
 # Parse configuration options
 vars = Variables(configFile)
 vars.Add('BUILDDIR',      'Target directory for intermediate files')
@@ -243,17 +236,6 @@
 		if MTS_VERSION != depVersion:
 			versionMismatch = True
 
-if versionMismatch:
-		print '\nThe dependency directory and your Mitsuba codebase have different version'
-		print 'numbers! Your copy of Mitsuba has version %s, whereas the dependencies ' % MTS_VERSION
-		print 'have version %s. Please bring them into sync, either by running\n' % depVersion
-		print '$ hg update -r v%s\n' % depVersion 
-		print 'in the Mitsuba directory, or by running\n'
-		print '$ cd dependencies'
-		print '$ hg pull'
-		print '$ hg update -r v%s\n' % MTS_VERSION
-		Exit(1)
-
 env = conf.Finish()
 
 dist = GetOption('dist') != None
