Index: QtStalkerScript/QtStalkerScript.cpp
===================================================================
RCS file: /cvsroot/qtstalker/qtstalker/QtStalkerScript/QtStalkerScript.cpp,v
retrieving revision 1.9
diff -u -d -p -r1.9 QtStalkerScript.cpp
--- QtStalkerScript/QtStalkerScript.cpp	10 Sep 2011 20:08:52 -0000	1.9
+++ QtStalkerScript/QtStalkerScript.cpp	17 Sep 2011 12:09:07 -0000
@@ -188,7 +189,7 @@ void QtStalkerScript::scriptFinished (in
   l << QObject::tr("Script");
   l << _script->name();
   l << QObject::tr("completed");
-qDebug() << l.join(" ");
+  qDebug() << l.join(" ");
 */
 
   QCoreApplication::exit(0);
