--- src/hdjcpl-1.09/Makefile.in	2009-01-20 16:23:10.000000000 +0100
+++ Makefile.in	2009-04-03 14:03:14.000000000 +0200
@@ -14,9 +14,9 @@
 
 @SET_MAKE@
 VPATH = @srcdir@
-pkgdatadir = $(datadir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
-pkgincludedir = $(includedir)/@PACKAGE@
+pkgdatadir = $(DESTDIR)/$(datadir)/@PACKAGE@
+pkglibdir = $(DESTDIR)/$(libdir)/@PACKAGE@
+pkgincludedir = $(DESTDIR)/$(includedir)/@PACKAGE@
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 install_sh_DATA = $(install_sh) -c -m 644
 install_sh_PROGRAM = $(install_sh) -c
@@ -168,7 +168,7 @@
 SUBDIRS = src
 DIST_OTHER = COPYING
 DATA_NAME_HDJCPL = hercules-hdjcpl
-HDJCPL_DATADIR = $(datadir)/$(DATA_NAME_HDJCPL)
+HDJCPL_DATADIR = $(DESTDIR)/$(datadir)/$(DATA_NAME_HDJCPL)
 SKIN_FILES = data/*.png
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
@@ -612,8 +612,8 @@
 install-data-local: $(SKIN_FILES)
 	$(MKDIR_P) $(HDJCPL_DATADIR)
 	cp $(SKIN_FILES) $(HDJCPL_DATADIR)
-	cp install-data/hdjcpl.desktop $(datadir)/applications
-	cp install-data/hdjcpl.xpm $(datadir)/icons
+	cp install-data/hdjcpl.desktop $(DESTDIR)/$(datadir)/applications
+	cp install-data/hdjcpl.xpm $(DESTDIR)/$(datadir)/pixmaps
 
 uninstall-local:
 	rm -f $(HDJCPL_DATADIR)/*.png
