Index: a/qemu/vl.c
===================================================================
--- a/qemu/vl.c
+++ b/qemu/vl.c
@@ -5658,7 +5658,8 @@ int main(int argc, char **argv, char **e
 #ifdef USE_KVM
     if (kvm_enabled()) {
 	if (kvm_qemu_init() < 0) {
-	    fprintf(stderr, "Could not initialize KVM, will disable KVM support\n");
+	    fprintf(stderr, "Could not initialize KVM. Do you have kvm-amd or kvm-intel modprobe'd?\nIf you want to use CPU emulation, start with -no-kvm.\n");
+	    exit(1);
 #ifdef NO_CPU_EMULATION
 	    fprintf(stderr, "Compiled with --disable-cpu-emulation, exiting.\n");
 	    exit(1);
