--- files/usr/lib/cinnamon-settings/cinnamon-settings.py	2012-05-03 17:25:53.665079828 +0200
+++ files/usr/lib/cinnamon-settings/cinnamon-settings.py	2012-05-03 17:26:08.921746257 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 try:
     import os
@@ -616,7 +616,7 @@
         self.label = Gtk.Label(label)
 
         self.content_widget = Gtk.FontButton()
-        self.content_widget.set_font_name(self.value)
+        #self.content_widget.set_font_name(self.value)
         
         if (label != ""):
             self.pack_start(self.label, False, False, 2)
@@ -636,7 +636,7 @@
         self.label = Gtk.Label(label)
 
         self.content_widget = Gtk.FontButton()
-        self.content_widget.set_font_name(self.value)
+        #self.content_widget.set_font_name(self.value)
         
         if (label != ""):
             self.pack_start(self.label, False, False, 2)
@@ -1121,7 +1121,7 @@
         
         sidePage = SidePage(_("Calendar"), "clock.svg", self.content_box)
         self.sidePages.append((sidePage, "calendar"))
-        self.changeTimeWidget = ChangeTimeWidget()     
+        #self.changeTimeWidget = ChangeTimeWidget()     
         sidePage.add_widget(GSettingsCheckButton(_("Show week dates in calendar"), "org.cinnamon.calendar", "show-weekdate"))         
         sidePage.add_widget(GSettingsEntry(_("Date format for the panel"), "org.cinnamon.calendar", "date-format"))                                 
         sidePage.add_widget(GSettingsEntry(_("Date format inside the date applet"), "org.cinnamon.calendar", "date-format-full"))                                 
@@ -1132,7 +1132,7 @@
             sidePage.add_widget(self.ntpCheckButton)
         except:
             pass
-        sidePage.add_widget(self.changeTimeWidget)
+        #sidePage.add_widget(self.changeTimeWidget)
         try:
             sidePage.add_widget(TimeZoneSelectorWidget())
         except:
@@ -1140,7 +1140,7 @@
         
         if self.ntpCheckButton != None:
             self.ntpCheckButton.connect('toggled', self._ntp_toggled)
-            self.changeTimeWidget.change_using_ntp( self.ntpCheckButton.get_active() )
+            #self.changeTimeWidget.change_using_ntp( self.ntpCheckButton.get_active() )
         
         sidePage = SidePage(_("Hot corner"), "overview.svg", self.content_box)
         self.sidePages.append((sidePage, "hotcorner"))
@@ -1358,7 +1358,8 @@
         self.side_view_sw.show_all()
         
     def _ntp_toggled(self, widget):
-        self.changeTimeWidget.change_using_ntp( self.ntpCheckButton.get_active() )
+        #self.changeTimeWidget.change_using_ntp( self.ntpCheckButton.get_active() )
+        pass
                 
                 
 if __name__ == "__main__":