diff -Naur BibTool.orig/configure.in BibTool.new/configure.in
--- BibTool.orig/configure.in	2012-03-17 23:40:33.000000000 +0100
+++ BibTool.new/configure.in	2012-03-18 00:12:53.000000000 +0100
@@ -70,7 +70,6 @@
 	AC_MSG_CHECKING(for kpathsea library)
 	kpathsea_dir=
 	kpathsea_lib=
-	kpathsea_lib_static=
 	kpathsea_def=
 	AC_TRY_COMPILE(
 	[
diff -Naur BibTool.orig/regex-0.12/regex.c BibTool.new/regex-0.12/regex.c
--- BibTool.orig/regex-0.12/regex.c	2012-03-17 23:40:33.000000000 +0100
+++ BibTool.new/regex-0.12/regex.c	2012-03-17 23:58:04.000000000 +0100
@@ -63,14 +63,6 @@
 #include <strings.h>
 #endif
 
-#ifdef STDC_HEADERS
-#include <stdlib.h>
-#else
-char *malloc ();
-char *realloc ();
-#endif
-
-
 /* Define the syntax stuff for \<, \>, etc.  */
 
 /* This must be nonzero for the wordchar and notwordchar pattern
