diff -Naur BibTool.orig/regex-0.12/regex.c BibTool.new/regex-0.12/regex.c
--- BibTool.orig/regex-0.12/regex.c	2012-03-17 23:40:33.000000000 +0100
+++ BibTool.new/regex-0.12/regex.c	2012-03-17 23:58:04.000000000 +0100
@@ -63,14 +63,6 @@
 #include <strings.h>
 #endif
 
-#ifdef STDC_HEADERS
-#include <stdlib.h>
-#else
-char *malloc ();
-char *realloc ();
-#endif
-
-
 /* Define the syntax stuff for \<, \>, etc.  */
 
 /* This must be nonzero for the wordchar and notwordchar pattern
