--- src/systemUtils.py.orig	2008-07-05 06:31:42.000000000 -0400
+++ src/systemUtils.py	2008-07-17 17:17:46.000000000 -0400
@@ -178,6 +178,8 @@
 
 def dumpSkillTree (skillTree):
     skillDumpFilename = getCacheDir () + "skillTreeDump.obj"
+    if not os.path.exists(getCacheDir ()):
+        os.makedirs(getCacheDir ())    
     skillDumpFile = open (skillDumpFilename, 'w')
     pickle.dump (skillTree, skillDumpFile)
     skillDumpFile.close ()
