From 1900184485673e3a49cc611b286bbfd882eee96e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A1s=20Czig=C3=A1ny?= <andras.czigany.13@gmail.com>
Date: Wed, 10 Dec 2014 17:09:46 +0100
Subject: [PATCH] rules.mk: fix mkdir

---
 project/rules.mk | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/project/rules.mk b/project/rules.mk
index f7a728a..3956c6c 100644
--- a/project/rules.mk
+++ b/project/rules.mk
@@ -3,31 +3,31 @@
 ###############################################################################
 
 $(TARGET_PATH):
-	mkdir $@
+	mkdir -p $@
 
 $(TARGET_BASE): | $(TARGET_PATH)
-	mkdir $@
+	mkdir -p $@
 
 $(TARGET_BIN): | $(TARGET_BASE)
-	mkdir $@
+	mkdir -p $@
 
 $(TARGET_DOC): | $(TARGET_BASE)
-	mkdir $@
+	mkdir -p $@
 
 $(TARGET_LIB): | $(TARGET_BASE)
-	mkdir $@
+	mkdir -p $@
 
 $(TARGET_OBJ): | $(TARGET_BASE)
-	mkdir $@
+	mkdir -p $@
 
 $(TARGET_SRC): | $(TARGET_BASE)
-	mkdir $@
+	mkdir -p $@
 
 $(TARGET_INCL): | $(TARGET_BASE)
-	mkdir $@
+	mkdir -p $@
 
 $(TARGET_TEST): | $(TARGET_BASE)
-	mkdir $@
+	mkdir -p $@
 
 .PHONY: info
 info:
-- 
2.1.3

