From e286e8e3d7b9e3d2d3b443bbf1ef156d8c84c985 Mon Sep 17 00:00:00 2001
From: Yclept Nemo <orbisvicis@gmail.com>
Date: Sat, 25 Jan 2014 15:43:36 -0500
Subject: [PATCH] Require CMake >= 2.8 to avoid CMP0012 warnings

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4637deb..789e537 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.8)
 
 project(flandmark)
 
-- 
1.8.2.2

