Index: rabbitvcs/vcs/svn/__init__.py
===================================================================
--- rabbitvcs/vcs/svn/__init__.py	(revision 3046)
+++ rabbitvcs/vcs/svn/__init__.py	(revision 3047)
@@ -276,6 +276,16 @@
             path = realpath(path)
         return self.client.info(path)
 
+    def find_repository_path(self, path):
+        path_to_check = path
+        while path_to_check != "/" and path_to_check != "":
+            if os.path.isdir(os.path.join(path_to_check, ".svn")):
+                return path_to_check
+            
+            path_to_check = os.path.split(path_to_check)[0]
+        
+        return None
+
     def status(self, path, summarize=True, invalidate=False):
         if path in self.cache:
             if invalidate:
@@ -323,14 +333,17 @@
             return False
 
     def is_in_a_or_a_working_copy(self, path):
-        return self.is_working_copy(path) or self.is_working_copy(os.path.split(path)[0])
+        if self.is_working_copy(path):
+            return True
 
+        return (self.find_repository_path(os.path.split(path)[0]) != "")
+
     def is_versioned(self, path):
         if self.is_working_copy(path):
             return True
         else:
             # info will return nothing for an unversioned file inside a working copy
-            if (self.is_working_copy(os.path.split(path)[0]) and
+            if (self.is_in_a_or_a_working_copy(path) and
                     self.client_info(path)):
                 return True
 
