--- liblcfg/code/configure.ac	2010-11-05 23:56:18.369969667 +0100
+++ liblcfg/code/configure.ac	2011-04-29 00:54:17.142240905 +0200
@@ -24,11 +24,11 @@
 case "$host" in
 *-*-freebsd*)
   CPPFLAGS="$CPPFLAGS -I/usr/local/include -I/usr/src/contrib/file/"
-  LDFLAGS="$LDFLAGS -L/usr/local/lib -R/usr/local/lib"
+  LDFLAGS="$LDFLAGS -L/usr/local/lib"
   ;;
 *-*-linux*)
   CPPFLAGS="$CPPFLAGS -D _GNU_SOURCE -I/usr/local/include"
-  LDFLAGS="$LDFLAGS -L/usr/local/lib -R/usr/local/lib"
+  LDFLAGS="$LDFLAGS -L/usr/local/lib"
   ;;
 *-*-darwin*)
   CPPFLAGS="$CPPFLAGS -I/opt/local/include"
