diff -ru gracegtk-0.4.5.orig/src/Makefile gracegtk-0.4.5/src/Makefile
--- gracegtk-0.4.5.orig/src/Makefile	2012-02-08 16:07:48.000000000 -0800
+++ gracegtk-0.4.5/src/Makefile	2012-02-08 16:08:23.000000000 -0800
@@ -60,6 +60,7 @@
 		$(JPEG_LIB) \
 		$(PNG_LIB) \
 		$(Z_LIB) \
+		$(DL_LIB) \
 		$(CUPS_LIBS) \
 		$(LIBS_GTK)
 
diff -ru gracegtk-0.4.5.orig/src/rstdrv.c gracegtk-0.4.5/src/rstdrv.c
--- gracegtk-0.4.5.orig/src/rstdrv.c	2012-02-08 16:07:48.000000000 -0800
+++ gracegtk-0.4.5/src/rstdrv.c	2012-02-08 16:10:14.000000000 -0800
@@ -55,6 +55,7 @@
 
 #ifdef HAVE_LIBPNG
 #  include <png.h>
+#  include <zlib.h>
 #endif
 
 static void rstImagePnm(gdImagePtr ihandle, FILE *prstream);
@@ -888,7 +889,7 @@
         return;
     }
 
-    if (setjmp(png_ptr->jmpbuf)) {
+    if (setjmp(png_jmpbuf(png_ptr))) {
         png_destroy_write_struct(&png_ptr, &info_ptr);
         return;
     }
