diff -ur mfiler2-4.1.0.orig/configure mfiler2-4.1.0/configure
--- mfiler2-4.1.0.orig/configure	2009-11-04 10:13:38.000000000 +0900
+++ mfiler2-4.1.0/configure	2009-12-11 22:00:57.048220299 +0900
@@ -3198,7 +3198,7 @@
 test -z "$INSTALL_DATA" && INSTALL_DATA='${INSTALL} -m 644'
 
 
-CFLAGS="-I . -I/usr/local/include -L/usr/local/lib -g"
+CFLAGS="-I . -I/usr/local/include -L/usr/local/lib -L/opt/ruby1.8/lib -Wl,-rpath,/opt/ruby1.8/lib -g"
 
 
 ###############################################################################
@@ -3652,10 +3652,10 @@
 ###############################################################################
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking Ruby" >&5
 $as_echo_n "checking Ruby... " >&6; }
-if test -e "/usr/bin/ruby" || test -e "/usr/local/bin/ruby"; then
+if test -e "/usr/bin/ruby-1.8" || test -e "/usr/local/bin/ruby"; then
     echo "ruby"
 
-    RUBY="ruby"
+    RUBY="ruby-1.8"
 elif test -e "/usr/bin/ruby18" || test -e "/usr/local/bin/ruby18"; then
     echo "ruby18"
 
