--- makefile.orig	2012-06-15 10:04:17.307159232 +0200
+++ makefile	2012-06-15 10:11:08.070259240 +0200
@@ -76,17 +76,18 @@
 #
 
 # Warning do not set both USE_NETCDF3 and USE_NETCDF4 to one
-USE_NETCDF3=1
-USE_NETCDF4=0
+USE_NETCDF3=0
+USE_NETCDF4=1
 USE_REGEX=1
 USE_TIGGE=1
-USE_MYSQL=0
-USE_IPOLATES=0
+USE_MYSQL=1
+USE_IPOLATES=1
 USE_UDF=0
 USE_OPENMP=0
 
 # often enviroment variable FC=fortran compiler, if not, define it here
-# FC=gfortran
+FC=gfortran
+FFLAGS=$(CFLAGS)
 
 ifeq ($(USE_NETCDF3),1)
   ifeq ($(USE_NETCDF4),1)
@@ -111,7 +112,7 @@
 # wLDFLAGS has the directory/name of the library
 
 ifeq ($(CC),gcc)
-   wCPPFLAGS+=-Wall -Wmissing-prototypes -Wold-style-definition -ffast-math  -O3 -g
+   wCPPFLAGS+=$(CPPFLAGS)
 endif
 ifeq ($(CC),opencc)
    wCPPFLAGS+=-O3 -Wall -ffast-math -opencc
@@ -152,10 +153,9 @@
 
 # grib2c library
 
-g:=${cwd}/g2clib-1.2.1
-glib:=$g/libgrib2c.a
-wLDFLAGS+=-L$g -lgrib2c
-wCPPFLAGS+=-I$g
+glib:=/usr/lib/libgrib2c.a
+wLDFLAGS+=-L/usr/lib -lgrib2c
+wCPPFLAGS+=-I/usr/include
 
 # gctpc library
 gctpc:=${cwd}/gctpc
@@ -165,11 +165,9 @@
 
 # Jasper
 
-j=${cwd}/jasper-1.900.1
-jlib=$j/src/libjasper/.libs/libjasper.a
-wLDFLAGS+=-L$j/src/libjasper/.libs -ljasper
-wCPPFLAGS+=-I$j/src/libjasper/include
-
+jlib=/usr/lib/libjasper.a
+wLDFLAGS+=-L/usr/lib -ljasper
+wCPPFLAGS+=-I/usr/include
 
 ifeq ($(USE_NETCDF3),1)
    n:=${cwd}/netcdf-3.6.2
@@ -182,13 +180,11 @@
 endif
 
 ifeq ($(USE_NETCDF4),1)
-   n4:=${cwd}/netcdf-4.1.3
-   n4lib:=${n4}/libsrc4/.libs/libnetcdf.a
-   h5:=${cwd}/hdf5-1.8.6
-   h5lib:=${h5}/src/.libs/libhdf5.a
-#   wLDFLAGS+=-L${n4}/libsrc/.libs -lnetcdf
-   wLDFLAGS+=-L${n4}/libsrc4/.libs -lnetcdf -L${h5}/hl/src/.libs -lhdf5_hl -L${h5}/src/.libs -lhdf5
-   wCPPFLAGS+=-I${n4}/libsrc4
+   n4lib:=/usr/lib/libnetcdf.a
+   h5:=
+   h5lib:=
+	 wLDFLAGS+=`nc-config --libs`
+	 wCPPFLAGS+=`nc-config --cflags`
    a:=$(shell echo "\#define USE_NETCDF4" >> ${CONFIG_H})
 else
    a:=$(shell echo "//\#define USE_NETCDF4" >> ${CONFIG_H})
@@ -223,7 +219,7 @@
    ifeq ($(FC),gfortran)
       wLDFLAGS+=-lgfortran
       wCPPFLAGS+=-DGFORTRAN
-      wFFLAGS+=-O2
+      wFFLAGS+=$(CFLAGS)
    endif
 
 # for open64 fortran - personal system
@@ -307,17 +303,15 @@
 
 # png 
 
-p=${cwd}/libpng-1.2.44
-plib=$p/.libs/libpng.a
-wLDFLAGS+=-L$p/.libs -lpng
-wCPPFLAGS+=-I$p
+plib=/usr/lib/libpng.a
+wLDFLAGS+=-L/usr/lib -lpng
+wCPPFLAGS+=-I/usr/include
 
 # z
 
-z=${cwd}/zlib-1.2.5
-zlib=$z/libz.a
-wLDFLAGS+=-L$z -lz
-wCPPFLAGS+=-I$z
+zlib=/usr/lib/libz.a
+wLDFLAGS+=-L/usr/lib -lz
+wCPPFLAGS+=-I/usr/include
 
 wLDFLAGS+=-lm
 wCPPFLAGS+=-I/usr/include ${CPPFLAGS}
@@ -337,9 +331,6 @@
    MAKE:=gmake
 endif
 
-
-
-
 w=wgrib2
 prog=$w/wgrib2
 
@@ -352,37 +343,6 @@
 fast:        $w/*.c $w/*.h ${jlib} ${nlib} ${zlib} ${plib} ${h5lib} ${glib} ${n4lib} ${iplib} ${gctpclib}
 	cd $w && export LDFLAGS="${wLDFLAGS}" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
 
-
-${jlib}:
-	cp $j.tar.gz tmpj.tar.gz
-	gunzip -f tmpj.tar.gz
-	tar -xvf tmpj.tar
-	rm tmpj.tar
-	cd $j && export CFLAGS="${wCPPFLAGS}" && ./configure --without-x --disable-libjpeg --disable-opengl && ${MAKE}
-
-${plib}:	${zlib}
-	cp $p.tar.gz tmpp.tar.gz
-	gunzip -f tmpp.tar.gz
-	tar -xvf tmpp.tar
-	rm tmpp.tar
-#       for OSX
-#	export LDFLAGS="-L$z" && cd $p && export CPPFLAGS="${wCPPFLAGS}" && make -f scripts/makefile.darwin
-#	for everybody else
-	export LDFLAGS="-L$z" && cd $p && export CPPFLAGS="${wCPPFLAGS}" && ./configure --disable-shared && ${MAKE}
-
-${zlib}:
-	cp $z.tar.gz tmpz.tar.gz
-	gunzip -f tmpz.tar.gz
-	tar -xvf tmpz.tar
-	rm tmpz.tar
-	cd $z && export CFLAGS="${wCPPFLAGS}" && ./configure && ${MAKE}
-
-
-${glib}:	${jlib} ${plib} ${zlib}
-	touch ${glib}
-	rm ${glib}
-	cd $g && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
-
 ${gctpclib}:
 	cp gctpc20.tar.Z tmpgctpc.tar.Z
 	gunzip -f tmpgctpc.tar.Z
@@ -391,39 +351,8 @@
 	cp makefile.gctpc proj.h ${gctpc}/source/
 	cd ${gctpc}/source && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE} -f makefile.gctpc
 
-${nlib}:
-	cp netcdf.tar.gz tmpn.tar.gz
-	gunzip -f tmpn.tar.gz
-	tar -xvf tmpn.tar
-	rm tmpn.tar
-	cd $n && export CPPFLAGS="${wCPPFLAGS}" && ./configure --enable-c-only && ${MAKE} check
-
-${n4lib}:	${zlib} ${h5lib}
-	mkdir -p ${cwd}/zlib/include
-	mkdir -p ${cwd}/zlib/lib
-	cp ${z}/*.h ${cwd}/zlib/include/
-	cp ${z}/*.a ${cwd}/zlib/lib/
-
-	mkdir -p ${cwd}/hdf5/include
-	cp  ${h5}/src/*.h ${h5}/hl/src/*.h ${cwd}/hdf5/include
-	mkdir -p ${cwd}/hdf5/lib
-	cp ${h5}/src/.libs/*a ${h5}/hl/src/.libs/*a ${cwd}/hdf5/lib/
-
-	cp ${n4}.tar.gz tmpn.tar.gz
-	gunzip -f tmpn.tar.gz
-	tar -xvf tmpn.tar
-	rm tmpn.tar
-	cd ${n4} && export CPPFLAGS="${wCPPFLAGS}" && ./configure --disable-fortran --disable-cxx --disable-dap --enable-netcdf-4 --with-zlib=${cwd}/zlib --with-hdf5=${cwd}/hdf5 && ${MAKE}
-
-${h5lib}:
-	cp ${h5}.tar.gz tmph5.tar.gz
-	gunzip -f tmph5.tar.gz
-	tar -xvf tmph5.tar
-	rm tmph5.tar
-	cd ${h5} && export CPPFLAGS="${CPPFLAGS}" && ./configure --disable-shared --disable-fortran --with-zlib=$z && ${MAKE} && ${MAKE}
-
 ${iplib}:
-	cd ${ip} && export F90=${F90} && export FFLAGS="${wFFLAGS}" && ${MAKE}
+	cd ${ip} && ${MAKE} FC=${FC} FFLAGS="${wFFLAGS}" 
 
 aux_progs/gmerge:	aux_progs/gmerge.c		
 	cd aux_progs && ${MAKE} -f gmerge.make
