--- makefile~	2009-10-09 23:09:53.000000000 +0600
+++ makefile	2009-11-12 15:25:25.669357605 +0500
@@ -27,8 +27,8 @@
 #
 # the flags are stored in wgrib2/config.h
 #
-USE_NETCDF4=0
-USE_NETCDF3=1
+USE_NETCDF4=1
+USE_NETCDF3=0
 USE_REGEX=1
 USE_TIGGE=1
 USE_MYSQL=0
@@ -74,13 +74,10 @@
 endif
 
 ifeq ($(USE_NETCDF4),1)
-   n4:=${cwd}/netcdf-4.0.1
-   n4lib:=${n4}/libsrc4/.libs/libnetcdf.a
-   h5:=${cwd}/hdf5-1.8.2
-   h5lib:=${h5}/src/.libs/libhdf5.a
-#   wLDFLAGS+=-L${n4}/libsrc/.libs -lnetcdf
-   wLDFLAGS+=-L${n4}/libsrc4/.libs -lnetcdf -L${h5}/hl/src/.libs -lhdf5_hl -L${h5}/src/.libs -lhdf5
-   wCPPFLAGS+=-I${n4}/libsrc4
+   n4lib:=/usr/lib/libnetcdf.a
+   h5lib:=/usr/lib/libhdf5.a
+   wLDFLAGS+=-L/usr/lib -lnetcdf -lhdf5_hl -lhdf5
+   wCPPFLAGS+=-I/usr/include
    a:=$(shell echo "\#define USE_NETCDF4" >> ${CONFIG_H})
 else
    a:=$(shell echo "//\#define USE_NETCDF4" >> ${CONFIG_H})
@@ -102,24 +99,21 @@
 
 # Jasper
 
-j=${cwd}/jasper-1.900.1
-jlib=$j/src/libjasper/.libs/libjasper.a
-wLDFLAGS+=-L$j/src/libjasper/.libs -ljasper
-wCPPFLAGS+=-I$j/src/libjasper/include
+jlib=/usr/lib/libjasper.a
+wLDFLAGS+=-L/usr/lib -ljasper
+wCPPFLAGS+=-I/usr/include
 
 # png 
 
-p=${cwd}/libpng-1.2.34
-plib=$p/.libs/libpng.a
-wLDFLAGS+=-L$p/.libs -lpng
-wCPPFLAGS+=-I$p
+plib=/usr/lib/libpng.a
+wLDFLAGS+=-L/usr/lib -lpng
+wCPPFLAGS+=-I/usr/include
 
 # z
 
-z=${cwd}/zlib-1.2.3
-zlib=$z/libz.a
-wLDFLAGS+=-L$z -lz
-wCPPFLAGS+=-I$z
+zlib=/usr/lib/libz.a
+wLDFLAGS+=-L/usr/lib -lz
+wCPPFLAGS+=-I/usr/include
 
 wLDFLAGS+=-lm
 wCPPFLAGS+=-I/usr/include ${CPPFLAGS}
@@ -147,78 +141,12 @@
 	cd $w && export LDFLAGS="${wLDFLAGS}" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
 
 
-${jlib}:
-	cp $j.tar.gz tmpj.tar.gz
-	gunzip -n tmpj.tar.gz
-	tar -xvf tmpj.tar
-	rm tmpj.tar
-	cd $j && ./configure --without-x --disable-libjpeg --disable-opengl && ${MAKE}
-
-${plib}:	${zlib}
-	cp $p.tar.gz tmpp.tar.gz
-	gunzip -n tmpp.tar.gz
-	tar -xvf tmpp.tar
-	rm tmpp.tar
-#       for OSX
-#	export LDFLAGS="-L $z" && cd $p && export CPPFLAGS="${wCPPFLAGS}" && make -f scripts/makefile.darwin
-#	for everybody else
-	export LDFLAGS="-L $z" && cd $p && export CPPFLAGS="${wCPPFLAGS}" && ./configure --disable-shared && ${MAKE}
-
-${zlib}:
-	cp $z.tar.gz tmpz.tar.gz
-	gunzip -n tmpz.tar.gz
-	tar -xvf tmpz.tar
-	rm tmpz.tar
-	cd $z && ./configure && ${MAKE}
-
-
 ${glib}:	${jlib} ${plib} ${zlib}
 	touch ${glib}
 	rm ${glib}
 	cd $g && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
 
 
-${nlib}:
-	cp netcdf.tar.gz tmpn.tar.gz
-	gunzip -n tmpn.tar.gz
-	tar -xvf tmpn.tar
-	rm tmpn.tar
-	cd $n && ./configure --enable-c-only && ${MAKE} check
-
-${n4lib}:	${zlib} ${h5lib}
-	mkdir -p ${cwd}/zlib/include
-	mkdir -p ${cwd}/zlib/lib
-	cp ${z}/*.h ${cwd}/zlib/include/
-	cp ${z}/*.a ${cwd}/zlib/lib/
-
-	mkdir -p ${cwd}/hdf5/include
-	cp  ${h5}/src/*.h ${h5}/hl/src/*.h ${cwd}/hdf5/include
-	mkdir -p ${cwd}/hdf5/lib
-	cp ${h5}/src/.libs/*a ${h5}/hl/src/.libs/*a ${cwd}/hdf5/lib/
-
-	cp ${n4}.tar.gz tmpn.tar.gz
-	gunzip -n tmpn.tar.gz
-	tar -xvf tmpn.tar
-	rm tmpn.tar
-	cd ${n4} && ./configure --enable-netcdf-4 --enable-c-only --with-zlib=${cwd}/zlib --with-hdf5=${cwd}/hdf5 && ${MAKE}
-
-#	cd ${n4} && ./configure --enable-c-only && ${MAKE} check
-
-#	cd ${n4} && ./configure --enable-netcdf-4 --enable-c-only --with-zlib=${cwd}/zlib --with-hdf5=${cwd}/hdf5 && ${MAKE} check
-
-#	cd ${n4} && ./configure --enable-c-only && ${MAKE} check
-
-#	cd ${n4} && ./configure --enable-netcdf-4 --enable-c-only && ${MAKE} check
-
-
-${h5lib}:
-	cp ${h5}.tar.gz tmph5.tar.gz
-	gunzip -n tmph5.tar.gz
-	tar -xvf tmph5.tar
-	rm tmph5.tar
-	cd ${h5} && ./configure --disable-shared --disable-fortran --with-zlib=$z && ${MAKE} && ${MAKE}
-#	cd ${h5} && ./configure --disable-shared --disable-fortran --with-zlib=$z && ${MAKE} && ${MAKE} check
-
 clean:
 	cd $w && ${MAKE} clean
 	cd $g && touch junk.a junk.o && rm *.o *.a
