--- makefile.orig	2011-08-04 20:03:32.000000000 +0200
+++ makefile	2011-11-08 14:09:07.605580448 +0100
@@ -33,12 +33,12 @@
 #
 
 # Warning do not set both USE_NETCDF3 and USE_NETCDF4 to one
-USE_NETCDF3=1
-USE_NETCDF4=0
+USE_NETCDF3=0
+USE_NETCDF4=1
 USE_REGEX=1
 USE_TIGGE=1
-USE_MYSQL=0
-USE_IPOLATES=0
+USE_MYSQL=1
+USE_IPOLATES=1
 USE_UDF=0
 
 ifeq ($(USE_NETCDF3),1)
@@ -72,18 +72,15 @@
 
 # grib2c library
 
-g:=${cwd}/g2clib-1.2.1
-glib:=$g/libgrib2c.a
-wLDFLAGS+=-L$g -lgrib2c
-wCPPFLAGS+=-I$g
+glib:=/usr/lib/libgrib2c.a
+wLDFLAGS+=-L/usr/lib -lgrib2c
+wCPPFLAGS+=-I/usr/include
 
 # Jasper
 
-j=${cwd}/jasper-1.900.1
-jlib=$j/src/libjasper/.libs/libjasper.a
-wLDFLAGS+=-L$j/src/libjasper/.libs -ljasper
-wCPPFLAGS+=-I$j/src/libjasper/include
-
+jlib=/usr/lib/libjasper.a
+wLDFLAGS+=-L/usr/lib -ljasper
+wCPPFLAGS+=-I/usr/include
 
 ifeq ($(USE_NETCDF3),1)
    n:=${cwd}/netcdf-3.6.2
@@ -96,13 +93,12 @@
 endif
 
 ifeq ($(USE_NETCDF4),1)
-   n4:=${cwd}/netcdf-4.1.1
-   n4lib:=${n4}/libsrc4/.libs/libnetcdf.a
-   h5:=${cwd}/hdf5-1.8.6
-   h5lib:=${h5}/src/.libs/libhdf5.a
+   n4lib:=/usr/lib/libnetcdf.a
+   h5:=
+   h5lib:=
 #   wLDFLAGS+=-L${n4}/libsrc/.libs -lnetcdf
-   wLDFLAGS+=-L${n4}/libsrc4/.libs -lnetcdf -L${h5}/hl/src/.libs -lhdf5_hl -L${h5}/src/.libs -lhdf5
-   wCPPFLAGS+=-I${n4}/libsrc4
+   wLDFLAGS+=`nc-config --libs`
+   wCPPFLAGS+=`nc-config --cflags`
    a:=$(shell echo "\#define USE_NETCDF4" >> ${CONFIG_H})
 else
    a:=$(shell echo "//\#define USE_NETCDF4" >> ${CONFIG_H})
@@ -162,17 +158,15 @@
 
 # png 
 
-p=${cwd}/libpng-1.2.44
-plib=$p/.libs/libpng.a
-wLDFLAGS+=-L$p/.libs -lpng
-wCPPFLAGS+=-I$p
+plib=/usr/lib/libpng.a
+wLDFLAGS+=-L/usr/lib -lpng
+wCPPFLAGS+=-I/usr/include
 
 # z
 
-z=${cwd}/zlib-1.2.5
-zlib=$z/libz.a
-wLDFLAGS+=-L$z -lz
-wCPPFLAGS+=-I$z
+zlib=/usr/lib/libz.a
+wLDFLAGS+=-L/usr/lib -lz
+wCPPFLAGS+=-I/usr/include
 
 wLDFLAGS+=-lm
 wCPPFLAGS+=-I/usr/include ${CPPFLAGS}
@@ -204,78 +198,6 @@
 	cd $w && export LDFLAGS="${wLDFLAGS}" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
 
 
-${jlib}:
-	cp $j.tar.gz tmpj.tar.gz
-	gunzip -n tmpj.tar.gz
-	tar -xvf tmpj.tar
-	rm tmpj.tar
-	cd $j && ./configure --without-x --disable-libjpeg --disable-opengl && ${MAKE}
-
-${plib}:	${zlib}
-	cp $p.tar.gz tmpp.tar.gz
-	gunzip -n tmpp.tar.gz
-	tar -xvf tmpp.tar
-	rm tmpp.tar
-#       for OSX
-#	export LDFLAGS="-L$z" && cd $p && export CPPFLAGS="${wCPPFLAGS}" && make -f scripts/makefile.darwin
-#	for everybody else
-	export LDFLAGS="-L$z" && cd $p && export CPPFLAGS="${wCPPFLAGS}" && ./configure --disable-shared && ${MAKE}
-
-${zlib}:
-	cp $z.tar.gz tmpz.tar.gz
-	gunzip -n tmpz.tar.gz
-	tar -xvf tmpz.tar
-	rm tmpz.tar
-	cd $z && ./configure && ${MAKE}
-
-
-${glib}:	${jlib} ${plib} ${zlib}
-	touch ${glib}
-	rm ${glib}
-	cd $g && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
-
-
-${nlib}:
-	cp netcdf.tar.gz tmpn.tar.gz
-	gunzip -n tmpn.tar.gz
-	tar -xvf tmpn.tar
-	rm tmpn.tar
-	cd $n && ./configure --enable-c-only && ${MAKE} check
-
-${n4lib}:	${zlib} ${h5lib}
-	mkdir -p ${cwd}/zlib/include
-	mkdir -p ${cwd}/zlib/lib
-	cp ${z}/*.h ${cwd}/zlib/include/
-	cp ${z}/*.a ${cwd}/zlib/lib/
-
-	mkdir -p ${cwd}/hdf5/include
-	cp  ${h5}/src/*.h ${h5}/hl/src/*.h ${cwd}/hdf5/include
-	mkdir -p ${cwd}/hdf5/lib
-	cp ${h5}/src/.libs/*a ${h5}/hl/src/.libs/*a ${cwd}/hdf5/lib/
-
-	cp ${n4}.tar.gz tmpn.tar.gz
-	gunzip -n tmpn.tar.gz
-	tar -xvf tmpn.tar
-	rm tmpn.tar
-	cd ${n4} && ./configure --disable-fortran --disable-cxx --disable-dap --enable-netcdf-4 --with-zlib=${cwd}/zlib --with-hdf5=${cwd}/hdf5 && ${MAKE}
-
-#	cd ${n4} && ./configure --enable-c-only && ${MAKE} check
-
-#	cd ${n4} && ./configure --enable-netcdf-4 --enable-c-only --with-zlib=${cwd}/zlib --with-hdf5=${cwd}/hdf5 && ${MAKE} check
-
-#	cd ${n4} && ./configure --enable-c-only && ${MAKE} check
-
-#	cd ${n4} && ./configure --enable-netcdf-4 --enable-c-only && ${MAKE} check
-
-
-${h5lib}:
-	cp ${h5}.tar.gz tmph5.tar.gz
-	gunzip -n tmph5.tar.gz
-	tar -xvf tmph5.tar
-	rm tmph5.tar
-	cd ${h5} && ./configure --disable-shared --disable-fortran --with-zlib=$z && ${MAKE} && ${MAKE}
-#	cd ${h5} && ./configure --disable-shared --disable-fortran --with-zlib=$z && ${MAKE} && ${MAKE} check
-
 ${iplib}:
 	cd ${ip} && export F90=${F90} && make
 
