--- src/sakura.c	2010-04-17 05:36:17.000000000 +0400
+++ src/sakura.c	2010-11-05 16:58:18.090740617 +0300
@@ -152,6 +152,7 @@
 	guint height;
 	glong columns;
 	glong rows;
+	gint scroll_lines;
 	gint char_width;
 	gint char_height;
 	gint label_count;
@@ -206,9 +207,9 @@
 
 
 #define ICON_FILE "terminal-tango.svg"
-#define SCROLL_LINES 4096
 #define HTTP_REGEXP "(ftp|http)s?://[-a-zA-Z0-9.?$%&/=_~#.,:;+]*"
 #define CONFIGFILE "sakura.conf"
+#define DEFAULT_SCROLL_LINES 4096
 #define DEFAULT_COLUMNS 80
 #define DEFAULT_ROWS 24
 #define DEFAULT_FONT "monospace 11"
@@ -1578,6 +1579,11 @@
 	}
 	g_free(cfgtmp);
 
+	if (!g_key_file_has_key(sakura.cfg, cfg_group, "scroll_lines", NULL)) {
+		g_key_file_set_integer(sakura.cfg, cfg_group, "scroll_lines", DEFAULT_SCROLL_LINES);
+	}
+	sakura.scroll_lines = g_key_file_get_integer(sakura.cfg, cfg_group, "scroll_lines", NULL);
+
 	if (!g_key_file_has_key(sakura.cfg, cfg_group, "add_tab_accelerator", NULL)) {
 		g_key_file_set_integer(sakura.cfg, cfg_group, "add_tab_accelerator", DEFAULT_ADD_TAB_ACCELERATOR);
 	}
@@ -2109,7 +2115,7 @@
 	gtk_widget_show_all(tab_hbox);
 
 	/* Init vte */
-	vte_terminal_set_scrollback_lines(VTE_TERMINAL(term->vte), SCROLL_LINES);
+	vte_terminal_set_scrollback_lines(VTE_TERMINAL(term->vte), sakura.scroll_lines);
 	vte_terminal_match_add_gregex(VTE_TERMINAL(term->vte), sakura.http_regexp, 0);
 	vte_terminal_set_mouse_autohide(VTE_TERMINAL(term->vte), TRUE);
 
