diff -aur tapestrea-0.1.0.6/scripting/chuck-1.2.1.2/src/makefile.jack tapestrea-0.1.0.6.patched/scripting/chuck-1.2.1.2/src/makefile.jack
--- tapestrea-0.1.0.6/scripting/chuck-1.2.1.2/src/makefile.jack	2009-05-29 21:30:43.000000000 +0200
+++ tapestrea-0.1.0.6.patched/scripting/chuck-1.2.1.2/src/makefile.jack	2012-04-12 20:26:50.241790534 +0200
@@ -6,7 +6,7 @@
 INCLUDES=
 CFLAGS?= -O3
 FLAGS= -D__LINUX_JACK__ -c $(CFLAGS)
-LIBS=-lasound -ljack -lstdc++ -ldl -lm
+LIBS=-lasound -ljack -lstdc++ -ldl -lm -lpthread
 SF_OBJ=util_sndfile.o
 
 ifneq ($(CHUCK_DEBUG),)
diff -aur tapestrea-0.1.0.6/scripting/chuck-1.2.1.2/src/util_string.h tapestrea-0.1.0.6.patched/scripting/chuck-1.2.1.2/src/util_string.h
--- tapestrea-0.1.0.6/scripting/chuck-1.2.1.2/src/util_string.h	2009-05-29 21:30:43.000000000 +0200
+++ tapestrea-0.1.0.6.patched/scripting/chuck-1.2.1.2/src/util_string.h	2012-04-12 20:15:28.641249810 +0200
@@ -36,7 +36,7 @@
 #include "chuck_def.h"
 #include <string>
 #include <vector>
-
+#include <stdio.h>
 
 // itoa
 std::string itoa( t_CKINT val );
diff -aur tapestrea-0.1.0.6/src/makefile.jack tapestrea-0.1.0.6.patched/src/makefile.jack
--- tapestrea-0.1.0.6/src/makefile.jack	2009-05-29 21:52:00.000000000 +0200
+++ tapestrea-0.1.0.6.patched/src/makefile.jack	2012-04-12 20:27:09.168749904 +0200
@@ -3,7 +3,7 @@
 CHUCK_DIR=../scripting/chuck-1.2.1.2/src
 INCLUDES=-I$(CHUCK_DIR)
 FLAGS=-D__LINUX_JACK__ `pkg-config --cflags gtk+-2.0` -O3 -c
-LIBS=-lasound -lstdc++ -lm -lsndfile -lGL -lGLU -lglut -lpthread -ljack `pkg-config --libs gtk+-2.0`
+LIBS=-lasound -lstdc++ -lm -lsndfile -lGL -lGLU -lglut -lpthread -ljack -ldl `pkg-config --libs gtk+-2.0`
 
 TAPS_OBJS= taps_analysis.o taps_birdbrain.o taps_driver.o \
 	taps_featurelibrary.o taps_pvc.o taps_regioncomparer.o \
diff -aur tapestrea-0.1.0.6/src/taps_birdbrain.h tapestrea-0.1.0.6.patched/src/taps_birdbrain.h
--- tapestrea-0.1.0.6/src/taps_birdbrain.h	2009-05-07 04:58:10.000000000 +0200
+++ tapestrea-0.1.0.6.patched/src/taps_birdbrain.h	2012-04-12 20:00:37.740805676 +0200
@@ -43,7 +43,7 @@
 #include <memory.h>
 #include <vector>
 #include <string>
-
+#include <stdio.h>
 
 class BirdBrain
 {
@@ -67,7 +67,7 @@
     static void goto_dir( const std::string & path );
 
     static void ola( SAMPLE *buffer, SAMPLE *to_add, int shift, int size );
-	static void ola( SAMPLE *buffer, SAMPLE *to_add, int shift, int buffer_size, int to_add_size);
+	static void ola( SAMPLE *buffer, SAMPLE *to_add, int shift, int buffer_size, int to_add_size);
     static void filter( const SAMPLE *x, const SAMPLE *winxp, SAMPLE *y, int len, float *B, int lenb, float *A, int lena );
     static void scale( SAMPLE *buffer, t_TAPUINT len, float factor );
     static void scale_fft( SAMPLE *buffer, t_TAPUINT len, t_TAPUINT fft_size = fft_size(), t_TAPUINT wnd_size = wnd_size() );
