diff -up alienblaster/src/options.cc~ alienblaster/src/options.cc
--- alienblaster/src/options.cc~	2007-07-22 21:13:29.000000000 +0200
+++ alienblaster/src/options.cc	2007-07-22 21:13:29.000000000 +0200
@@ -48,7 +48,7 @@ Options::Options(const string lFilename)
       string newLine;
       getline( inputFile, newLine );
       // search delimiter
-      unsigned int pos = newLine.find("=");
+      string::size_type pos = newLine.find("=");
       // comment line or no delimiter found
       if (( newLine[0] != '#' ) && ( pos != string::npos )) {
 	string key = newLine.substr(0,pos);
diff -up alienblaster/src/font.cc~ alienblaster/src/font.cc
--- alienblaster/src/font.cc~	2007-07-22 21:14:44.000000000 +0200
+++ alienblaster/src/font.cc	2007-07-22 21:14:44.000000000 +0200
@@ -84,7 +84,7 @@ void Font::drawInt(SDL_Surface *screen, 
       destR.w = charWidth;
       destR.h = charHeight;
       
-      unsigned int charsetpos = charset.find( (char)((val % 10) + '0') );
+      string::size_type charsetpos = charset.find( (char)((val % 10) + '0') );
       if (charsetpos != string::npos ) {
 	srcR.x = charsetpos * charWidth;
       } else {
@@ -127,7 +127,7 @@ void Font::drawStr(SDL_Surface *screen, 
   SDL_Rect srcR;
   
   int x = 0;
-  unsigned int charsetpos;
+  string::size_type charsetpos;
   for(unsigned int i=0; i < text.size(); ++i) {
     x = 0;
     charsetpos = charset.find(text[i]);
