diff -up alienblaster/src/video.h~ alienblaster/src/video.h
--- alienblaster/src/video.h~	2007-07-22 21:34:45.000000000 +0200
+++ alienblaster/src/video.h	2007-07-22 21:34:45.000000000 +0200
@@ -36,8 +36,6 @@ public:
   ~Video();
   SDL_Surface *init();
 
-  bool fullscreen;
-  
   void clearScreen();
   void toggleFullscreen();
 };
diff -up alienblaster/src/video.cc~ alienblaster/src/video.cc
--- alienblaster/src/video.cc~	2007-07-22 21:33:10.000000000 +0200
+++ alienblaster/src/video.cc	2007-07-22 21:34:35.000000000 +0200
@@ -38,12 +38,11 @@ SDL_Surface *Video::init(){
   // --------------------------------------------------
   // SDL initialisation
   // -----------------------------------------------------
-  fullscreen = false;
   if (SDL_InitSubSystem(SDL_INIT_VIDEO) < 0) {
     printf("Couldn't initialize SDL video subsystem: %s\n", SDL_GetError());
     exit(1);
   }
-  screen = SDL_SetVideoMode( SCREEN_WIDTH, SCREEN_HEIGHT, BIT_DEPTH, SDL_DOUBLEBUF /* | SDL_FULLSCREEN */ );
+  screen = SDL_SetVideoMode( SCREEN_WIDTH, SCREEN_HEIGHT, BIT_DEPTH, SDL_DOUBLEBUF | SDL_FULLSCREEN );
   if (!screen) {
     printf("Couldn't set %dx%d, %dbit video mode: %s\n", SCREEN_WIDTH, SCREEN_HEIGHT, BIT_DEPTH, SDL_GetError());
     exit(2);
@@ -68,10 +67,5 @@ void Video::clearScreen() {
 }
 
 void Video::toggleFullscreen() {
-  if ( fullscreen ) {
-    screen = SDL_SetVideoMode( SCREEN_WIDTH, SCREEN_HEIGHT, BIT_DEPTH, SDL_DOUBLEBUF );
-  } else {
-    screen = SDL_SetVideoMode( SCREEN_WIDTH, SCREEN_HEIGHT, BIT_DEPTH, SDL_DOUBLEBUF | SDL_FULLSCREEN );
-  }
-  fullscreen = !fullscreen;
+  SDL_WM_ToggleFullScreen(screen);
 }
