--- etc/init.d.slurm	2010-09-20 19:05:02.000000000 +0200
+++ .//init.d.slurm	2011-02-04 12:30:49.253250631 +0100
@@ -11,7 +11,7 @@
 # processname: /usr/sbin/slurmctld
 # pidfile: /var/run/slurmctld.pid
 #
-# config: /etc/sysconfig/slurm
+# config: /etc/default/slurm
 #
 ### BEGIN INIT INFO
 # Provides:          slurm
@@ -69,8 +69,8 @@
 # Source slurm specific configuration
 # SLURMCTLD_OPTIONS defines slurmctld command line options. See "man slurmctld"
 # SLURMD_OPTIONS defines slurmd command line options. See "man slurmd"
-if [ -f /etc/sysconfig/slurm ] ; then
-    . /etc/sysconfig/slurm
+if [ -f /etc/default/slurm ] ; then
+    . /etc/default/slurm
 else
     SLURMCTLD_OPTIONS=""
     SLURMD_OPTIONS=""
