--- ./etc/slurmd.service	2015-03-08 17:43:36.484824729 +0100
+++ ./etc/slurmd.service	2015-03-08 17:50:21.786157899 +0100
@@ -1,12 +1,12 @@
 [Unit]
 Description=Slurm node daemon
-After=network.target
+After=network.target systemd-tmpfiles-clean.service
 ConditionPathExists=/etc/slurm-llnl/slurm.conf

 [Service]
-Type=forking
-EnvironmentFile=/etc/default/slurmd
-ExecStart=/usr/bin/slurmd $SLURMD_OPTIONS
+Type=simple
+EnvironmentFile=/etc/default/slurm-llnl
+ExecStart=/usr/bin/slurmd -D $SLURMD_OPTIONS
 PIDFile=/var/run/slurm/slurmd.pid

 [Install]
