--- pigale-1.3.14.orig/ClientServer/ClientServer.pro	2012-02-16 18:23:05.000000000 +0600
+++ pigale-1.3.14/ClientServer/ClientServer.pro	2012-05-12 10:35:57.483232969 +0600
@@ -33,9 +33,9 @@
 DESTDIR=$$PWD
    
 #Installation
-target.path = $$DISTPATH/bin	
+target.path = /usr/bin	
 data.files =ClientData.txt
-data.path = $$DISTPATH/bin
+data.path = /usr/share/pigale
 INSTALLS += target data
 
 build_pass:CONFIG(debug, debug|release) {
