diff -rupN pigale-1.3.14.orig/ClientServer/ClientServer.pro pigale-1.3.14/ClientServer/ClientServer.pro
--- pigale-1.3.14.orig/ClientServer/ClientServer.pro	2012-02-16 07:23:05.000000000 -0500
+++ pigale-1.3.14/ClientServer/ClientServer.pro	2013-09-26 18:51:47.235341295 -0400
@@ -33,9 +33,9 @@ QMAKE_EXTRA_TARGETS += distdir
 DESTDIR=$$PWD
    
 #Installation
-target.path = $$DISTPATH/bin	
+target.path = /usr/bin
 data.files =ClientData.txt
-data.path = $$DISTPATH/bin
+data.path = /usr/share/pigale
 INSTALLS += target data
 
 build_pass:CONFIG(debug, debug|release) {
diff -rupN pigale-1.3.14.orig/qt/qt.pro pigale-1.3.14/qt/qt.pro
--- pigale-1.3.14.orig/qt/qt.pro	2012-02-20 10:17:28.000000000 -0500
+++ pigale-1.3.14/qt/qt.pro	2013-09-26 18:50:43.522004269 -0400
@@ -60,7 +60,7 @@ CONFIG(debug, debug|release) {
        win32-msvc2010:LIBGLUT=freeglut_debug.lib 
        }  else  {
        LIBTGRAPH=libtgraph.a
-       LIBGLUT=libfreeglut.a
+       #LIBGLUT=libfreeglut.a
        win32-msvc2010:LIBTGRAPH=tgraph.lib 
        win32-msvc2010:LIBGLUT=freeglut.lib 
        }
@@ -80,7 +80,7 @@ macx {
      } else {
      DEFINES += FREEGLUT FREEGLUT_STATIC
      #unix:LIBS += $$GLUTLIB
-     unix:LIBS += ../$$LIBGLUT
+     unix:LIBS += -lglut
      win32:LIBS +=../lib/$$LIBGLUT -lopengl32 -lglu32 -lgdi32 -luser32 -lwinmm
      }
      
