diff -aur cciss_vol_status-1.06.pristine/cciss_vol_status.c cciss_vol_status-1.06.new/cciss_vol_status.c
--- cciss_vol_status-1.06.pristine/cciss_vol_status.c	2009-11-18 17:13:06.000000000 +0000
+++ cciss_vol_status-1.06.new/cciss_vol_status.c	2010-06-14 16:01:39.000000000 +0100
@@ -1004,7 +1004,7 @@
 	if (status < 0) 
 		return 0;
 	for (i = 0; prod[i] != NULL; i++) {
-		if (strncmp(std_inq+8, prod[i], strnlen(prod[i])) == 0)
+		if (strncmp(std_inq+8, prod[i], strlen(prod[i])) == 0)
 			return 1;
 	}
 	return 0;
Only in cciss_vol_status-1.06.new: cciss_vol_status.c~
