--- src/platform-win32.cc.old	2012-06-15 13:38:53.264129788 -0500
+++ src/platform-win32.cc	2012-06-15 13:40:33.240414435 -0500
@@ -885,7 +885,7 @@
                    bool is_executable) {
   // VirtualAlloc rounds allocated size to page size automatically.
   size_t msize = RoundUp(requested, static_cast<int>(GetPageSize()));
-  void* address = 0;
+  //void* address = 0;
 
   // Windows XP SP2 allows Data Excution Prevention (DEP).
   int prot = is_executable ? PAGE_EXECUTE_READWRITE : PAGE_READWRITE;
