From 85bc2777a45471b152f1fc945a360d40c1ce5be6 Mon Sep 17 00:00:00 2001
From: Gergely Imreh <imrehg@gmail.com>
Date: Thu, 29 Oct 2009 12:05:03 +0800
Subject: [PATCH] giggle-view-file: depreciated gtk_source_view_set_mark_category_pixbuf

Signed-off-by: Gergely Imreh <imrehg@gmail.com>
---
 src/giggle-view-file.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/giggle-view-file.c b/src/giggle-view-file.c
index 1d692da..0db7133 100644
--- a/src/giggle-view-file.c
+++ b/src/giggle-view-file.c
@@ -560,7 +560,7 @@ create_category (GiggleViewFilePriv *priv,
 		pixbuf = create_category_icon (priv, name);
 	
 	if (pixbuf) {
-		gtk_source_view_set_mark_category_pixbuf
+		gtk_source_view_set_mark_category_icon_from_pixbuf
 			(GTK_SOURCE_VIEW (priv->source_view),
 			 name, pixbuf);
 
-- 
1.6.5.2

