--- a/nv-linux.h	2013-11-27 23:47:09.000000000 +0100
+++ b/nv-linux.h	2014-03-27 10:53:46.000000000 +0100
@@ -206,8 +206,12 @@
 #endif
 
 #if defined(CONFIG_ACPI) 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
+#include <linux/acpi.h>
+#else
 #include <acpi/acpi.h>
 #include <acpi/acpi_drivers.h>
+#endif
 #if defined(NV_ACPI_DEVICE_OPS_HAS_MATCH) || defined(ACPI_VIDEO_HID)
 #define NV_LINUX_ACPI_EVENTS_SUPPORTED 1
 #endif
