diff -Naur FreeCAD-0.11.4422.orig/src/Mod/Image/Gui/GLImageBox.cpp FreeCAD-0.11.4422/src/Mod/Image/Gui/GLImageBox.cpp
--- FreeCAD-0.11.4422.orig/src/Mod/Image/Gui/GLImageBox.cpp	2011-12-25 18:58:11.067683491 +0100
+++ FreeCAD-0.11.4422/src/Mod/Image/Gui/GLImageBox.cpp	2011-12-25 19:00:20.670314707 +0100
@@ -25,6 +25,16 @@
 #if defined(__MINGW32__)
 # include <GL/gl.h>
 # include <GL/glext.h>
+#elif defined (FC_OS_MACOSX)
+# include <OpenGL/gl.h>
+# include <OpenGL/glu.h>
+#elif defined (FC_OS_WIN32)
+# include <Windows.h>
+# include <GL/gl.h>
+# include <GL/glu.h>
+#else
+# include <GL/gl.h>
+# include <GL/glu.h>
 #endif
 
 #include "GLImageBox.h"
diff -Naur FreeCAD-0.11.4422.orig/src/Mod/Part/App/AppPartPy.cpp FreeCAD-0.11.4422/src/Mod/Part/App/AppPartPy.cpp
--- FreeCAD-0.11.4422.orig/src/Mod/Part/App/AppPartPy.cpp	2011-12-25 18:58:11.131016481 +0100
+++ FreeCAD-0.11.4422/src/Mod/Part/App/AppPartPy.cpp	2011-12-25 19:02:02.659762181 +0100
@@ -426,7 +426,7 @@
             d.SetCoord(vec.x, vec.y, vec.z);
         }
         Handle_Geom_Plane aPlane = new Geom_Plane(p, d);
-        BRepBuilderAPI_MakeFace Face(aPlane, 0.0, length, 0.0, width);
+        BRepBuilderAPI_MakeFace Face(aPlane, 0.0, length, 0.0, width,Precision::Confusion());
         return new TopoShapeFacePy(new TopoShape((Face.Face()))); 
     }
     catch (Standard_DomainError) {
@@ -971,7 +971,7 @@
             mkSweep.Perform(tolerance, Standard_False, GeomAbs_C1, BSplCLib::MaxDegree(), 1000);
 
             const Handle_Geom_Surface& surf = mkSweep.Surface();
-            BRepBuilderAPI_MakeFace mkBuilder(surf, umin, umax, vmin, vmax);
+            BRepBuilderAPI_MakeFace mkBuilder(surf, umin, umax, vmin, vmax,Precision::Confusion());
             const TopoDS_Face& face = mkBuilder.Face();
             return new TopoShapeFacePy(new TopoShape(face));
         }
@@ -1022,7 +1022,7 @@
             double u1,u2,v1,v2;
             surf->Bounds(u1,u2,v1,v2);
 
-            BRepBuilderAPI_MakeFace mkBuilder(surf, umin, umax, v1, v2);
+            BRepBuilderAPI_MakeFace mkBuilder(surf, umin, umax, v1, v2,Precision::Confusion());
             const TopoDS_Face& face = mkBuilder.Face();
             return new TopoShapeFacePy(new TopoShape(face));
         }
diff -Naur FreeCAD-0.11.4422.orig/src/Mod/Part/App/Geometry.cpp FreeCAD-0.11.4422/src/Mod/Part/App/Geometry.cpp
--- FreeCAD-0.11.4422.orig/src/Mod/Part/App/Geometry.cpp	2011-12-25 18:58:11.127683166 +0100
+++ FreeCAD-0.11.4422/src/Mod/Part/App/Geometry.cpp	2011-12-25 19:02:57.802796780 +0100
@@ -735,7 +735,7 @@
     Handle_Geom_Surface s = Handle_Geom_Surface::DownCast(handle());
     Standard_Real u1,u2,v1,v2;
     s->Bounds(u1,u2,v1,v2);
-    BRepBuilderAPI_MakeFace mkBuilder(s, u1, u2, v1, v2);
+    BRepBuilderAPI_MakeFace mkBuilder(s, u1, u2, v1, v2,Precision::Confusion());
     return mkBuilder.Shape();
 }
 
diff -Naur FreeCAD-0.11.4422.orig/src/Mod/Part/App/GeometrySurfacePyImp.cpp FreeCAD-0.11.4422/src/Mod/Part/App/GeometrySurfacePyImp.cpp
--- FreeCAD-0.11.4422.orig/src/Mod/Part/App/GeometrySurfacePyImp.cpp	2011-12-25 18:58:11.131016481 +0100
+++ FreeCAD-0.11.4422/src/Mod/Part/App/GeometrySurfacePyImp.cpp	2011-12-25 19:03:49.039185876 +0100
@@ -77,7 +77,7 @@
             s->Bounds(u1,u2,v1,v2);
             if (!PyArg_ParseTuple(args, "|dddd", &u1,&u2,&v1,&v2))
                 return 0;
-            BRepBuilderAPI_MakeFace mkBuilder(s, u1, u2, v1, v2);
+            BRepBuilderAPI_MakeFace mkBuilder(s, u1, u2, v1, v2, Precision::Confusion());
             TopoDS_Shape sh = mkBuilder.Shape();
             return new TopoShapeFacePy(new TopoShape(sh));
         }
diff -Naur FreeCAD-0.11.4422.orig/src/Mod/Part/App/PrimitiveFeature.cpp FreeCAD-0.11.4422/src/Mod/Part/App/PrimitiveFeature.cpp
--- FreeCAD-0.11.4422.orig/src/Mod/Part/App/PrimitiveFeature.cpp	2011-12-25 18:58:11.127683166 +0100
+++ FreeCAD-0.11.4422/src/Mod/Part/App/PrimitiveFeature.cpp	2011-12-25 19:05:23.842005617 +0100
@@ -123,7 +123,7 @@
     gp_Pnt pnt(0.0,0.0,0.0);
     gp_Dir dir(0.0,0.0,1.0);
     Handle_Geom_Plane aPlane = new Geom_Plane(pnt, dir);
-    BRepBuilderAPI_MakeFace mkFace(aPlane, 0.0, L, 0.0, W);
+    BRepBuilderAPI_MakeFace mkFace(aPlane, 0.0, L, 0.0, W, Precision::Confusion());
 
     const char *error=0;
     switch (mkFace.Error())
@@ -141,9 +141,11 @@
     case BRepBuilderAPI_ParametersOutOfRange:
         error = "parameters out of range";
         break;
+#if OCC_HEX_VERSION < 0x060500
     case BRepBuilderAPI_SurfaceNotC2:
         error = "surface not C2";
         break;
+#endif
     default:
         error = "unknown error";
         break;
diff -Naur FreeCAD-0.11.4422.orig/src/Mod/Part/App/TopoShape.cpp FreeCAD-0.11.4422/src/Mod/Part/App/TopoShape.cpp
--- FreeCAD-0.11.4422.orig/src/Mod/Part/App/TopoShape.cpp	2011-12-25 18:58:11.134349796 +0100
+++ FreeCAD-0.11.4422/src/Mod/Part/App/TopoShape.cpp	2011-12-25 19:06:04.638451272 +0100
@@ -140,8 +140,10 @@
         return "Curve projection failed";
     case BRepBuilderAPI_ParametersOutOfRange:
         return "Parameters out of range";
+#if OCC_HEX_VERSION < 0x060500
     case BRepBuilderAPI_SurfaceNotC2:
         return "Surface not C2-continous";
+#endif
     default:
         return "Unknown creation error";
     }
diff -Naur FreeCAD-0.11.4422.orig/src/Mod/Part/App/TopoShapeFacePyImp.cpp FreeCAD-0.11.4422/src/Mod/Part/App/TopoShapeFacePyImp.cpp
--- FreeCAD-0.11.4422.orig/src/Mod/Part/App/TopoShapeFacePyImp.cpp	2011-12-25 18:58:11.131016481 +0100
+++ FreeCAD-0.11.4422/src/Mod/Part/App/TopoShapeFacePyImp.cpp	2011-12-25 19:12:35.892998331 +0100
@@ -22,32 +22,32 @@
 
 
 #include "PreCompiled.h"
-#ifndef _PreComp_
-# include <BRep_Tool.hxx>
-# include <BRepTools.hxx>
-# include <BRepBuilderAPI_MakeFace.hxx>
-# include <ShapeAnalysis.hxx>
-# include <BRepAdaptor_Surface.hxx>
-# include <BRepLProp_SLProps.hxx>
-# include <BRepOffsetAPI_MakeOffset.hxx>
+#ifndef _PreComp_
+# include <BRep_Tool.hxx>
+# include <BRepTools.hxx>
+# include <BRepBuilderAPI_MakeFace.hxx>
+# include <ShapeAnalysis.hxx>
+# include <BRepAdaptor_Surface.hxx>
+# include <BRepLProp_SLProps.hxx>
+# include <BRepOffsetAPI_MakeOffset.hxx>
 # include <Geom_BezierSurface.hxx>
 # include <Geom_BSplineSurface.hxx>
-# include <Geom_Plane.hxx>
-# include <Geom_CylindricalSurface.hxx>
-# include <Geom_ConicalSurface.hxx>
+# include <Geom_Plane.hxx>
+# include <Geom_CylindricalSurface.hxx>
+# include <Geom_ConicalSurface.hxx>
 # include <Geom_SphericalSurface.hxx>
 # include <Geom_ToroidalSurface.hxx>
 # include <Handle_Geom_Surface.hxx>
 # include <TopoDS.hxx>
-# include <TopoDS_Face.hxx>
-# include <TopoDS_Wire.hxx>
-# include <gp_Pnt2d.hxx>
-# include <gp_Pln.hxx>
-# include <gp_Cylinder.hxx>
-# include <gp_Cone.hxx>
+# include <TopoDS_Face.hxx>
+# include <TopoDS_Wire.hxx>
+# include <gp_Pnt2d.hxx>
+# include <gp_Pln.hxx>
+# include <gp_Cylinder.hxx>
+# include <gp_Cone.hxx>
 # include <gp_Sphere.hxx>
 # include <gp_Torus.hxx>
-#endif
+#endif
 
 #include <BRepTopAdaptor_FClass2d.hxx>
 #include <BRepGProp.hxx>
@@ -97,7 +97,7 @@
 int TopoShapeFacePy::PyInit(PyObject* args, PyObject* /*kwd*/)
 {
     PyObject *pW;
-    if (PyArg_ParseTuple(args, "O!", &(Part::TopoShapePy::Type), &pW)) {
+    if (PyArg_ParseTuple(args, "O!", &(Part::TopoShapePy::Type), &pW)) {
         try {
             const TopoDS_Shape& sh = static_cast<Part::TopoShapePy*>(pW)->getTopoShapePtr()->_Shape;
             if (sh.IsNull()) {
@@ -109,17 +109,17 @@
                 BRepBuilderAPI_MakeFace mkFace(TopoDS::Wire(sh));
                 getTopoShapePtr()->_Shape = mkFace.Face();
                 return 0;
-            }
+            }
             else if (sh.ShapeType() == TopAbs_FACE) {
                 getTopoShapePtr()->_Shape = sh;
                 return 0;
             }
         }
-        catch (Standard_Failure) {
+        catch (Standard_Failure) {
             Handle_Standard_Failure e = Standard_Failure::Caught();
-            PyErr_SetString(PyExc_Exception, e->GetMessageString());
+            PyErr_SetString(PyExc_Exception, e->GetMessageString());
             return -1;
-        }
+        }
     }
 
     PyErr_Clear();
@@ -133,7 +133,7 @@
                 return -1;
             }
 
-            BRepBuilderAPI_MakeFace mkFace(S);
+            BRepBuilderAPI_MakeFace mkFace(S, Precision::Confusion());
             if (bound) {
                 Py::List list(bound);
                 for (Py::List::iterator it = list.begin(); it != list.end(); ++it) {
