diff -Naur FreeCAD-0.11.4422.orig//src/Mod/Part/App/PrimitiveFeature.cpp FreeCAD-0.11.4422/src/Mod/Part/App/PrimitiveFeature.cpp
--- FreeCAD-0.11.4422.orig//src/Mod/Part/App/PrimitiveFeature.cpp	2011-04-17 22:45:25.446781473 +0200
+++ FreeCAD-0.11.4422/src/Mod/Part/App/PrimitiveFeature.cpp	2011-04-17 22:46:58.816275648 +0200
@@ -141,9 +141,11 @@
     case BRepBuilderAPI_ParametersOutOfRange:
         error = "parameters out of range";
         break;
+#if OCC_HEX_VERSION < 0x060500
     case BRepBuilderAPI_SurfaceNotC2:
         error = "surface not C2";
         break;
+#endif
     default:
         error = "unknown error";
         break;
diff -Naur FreeCAD-0.11.4422.orig//src/Mod/Part/App/TopoShape.cpp FreeCAD-0.11.4422/src/Mod/Part/App/TopoShape.cpp
--- FreeCAD-0.11.4422.orig//src/Mod/Part/App/TopoShape.cpp	2011-04-17 22:45:25.946778765 +0200
+++ FreeCAD-0.11.4422/src/Mod/Part/App/TopoShape.cpp	2011-04-17 22:47:58.335953203 +0200
@@ -140,8 +140,10 @@
         return "Curve projection failed";
     case BRepBuilderAPI_ParametersOutOfRange:
         return "Parameters out of range";
+#if OCC_HEX_VERSION < 0x060500
     case BRepBuilderAPI_SurfaceNotC2:
         return "Surface not C2-continous";
+#endif
     default:
         return "Unknown creation error";
     }
