*** src/connman-dbus.conf	2011-12-04 22:13:34.658620137 +0100
--- src/connman-dbus.conf	2011-12-04 22:49:10.293722693 +0100
***************
*** 8,13 ****
--- 8,18 ----
          <allow send_interface="net.connman.Counter"/>
          <allow send_interface="net.connman.Notification"/>
      </policy>
+     <policy group="network">
+ 	<allow send_destination="net.connman"/>
+ 	<allow send_interface="net.connman.Agent"/>
+ 	<allow send_interface="net.connman.Counter"/>
+     </policy>
      <policy at_console="true">
          <allow send_destination="net.connman"/>
      </policy>
