From 235bb7e292a6d4d5e23a954f81d71a5699a0b676 Mon Sep 17 00:00:00 2001
From: march1993 <march511@gmail.com>
Date: Mon, 19 Aug 2013 21:49:21 +0000
Subject: [PATCH] bug fixed

---
 eaccelerator.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/eaccelerator.c b/eaccelerator.c
index f6be48f..84cd279 100644
--- a/eaccelerator.c
+++ b/eaccelerator.c
@@ -63,7 +63,9 @@
 
 #include "php.h"
 #include "php_ini.h"
+#if PHP_VERSION_ID < 50500
 #include "php_logos.h"
+#endif
 #include "main/fopen_wrappers.h"
 #include "ext/standard/info.h"
 #include "ext/standard/php_incomplete_class.h"
@@ -2269,8 +2271,10 @@ ZEND_DLEXPORT int eaccelerator_zend_startup(zend_extension *extension)
         }
     }
 
+#if PHP_VERSION_ID < 50500
     php_register_info_logo(EACCELERATOR_VERSION_GUID, "text/plain", (unsigned char*)EACCELERATOR_VERSION_STRING, sizeof(EACCELERATOR_VERSION_STRING));
     php_register_info_logo(EACCELERATOR_LOGO_GUID,    "image/gif",  (unsigned char*)eaccelerator_logo, sizeof(eaccelerator_logo));
+#endif
 
     return SUCCESS;
 }
-- 
1.8.5.1

