=== modified file 'unity-misc/na-tray-child.c'
--- unity-misc/na-tray-child.c	2011-06-16 13:15:05 +0000
+++ unity-misc/na-tray-child.c	2011-09-29 14:27:49 +0000
@@ -218,6 +218,18 @@
 }
 
 static void
+na_tray_child_get_preferred_width (GtkWidget* widget, gint *mwidth, gint *nwidth)
+{
+  *mwidth = *nwidth = 22;
+}
+
+static void
+na_tray_child_get_preferred_height (GtkWidget *widget, gint *mheight, gint *nheight)
+{
+  *mheight = *nheight = 22;
+}
+
+static void
 na_tray_child_init (NaTrayChild *child)
 {
 }
@@ -236,6 +248,8 @@
   widget_class->realize = na_tray_child_realize;
   widget_class->size_allocate = na_tray_child_size_allocate;
   widget_class->draw = na_tray_child_draw;
+  widget_class->get_preferred_width = na_tray_child_get_preferred_width;
+  widget_class->get_preferred_height = na_tray_child_get_preferred_height;
 }
 
 GtkWidget *

