--- bin/activemq.orig	2013-12-16 13:31:57.860172222 +0100
+++ bin/activemq	2013-12-16 13:34:22.616840102 +0100
@@ -308,6 +308,19 @@
     CLASSPATH="`cygpath --path --unix "$CLASSPATH"`"
 fi
 
+# Detect JAVA_HOME of the installed java-runtime
+if [ -z "$JAVA_HOME" ]; then
+  if [ -e /etc/profile.d/jre6.sh ]; then
+    source /etc/profile.d/jre6.sh
+  elif [ -e /etc/profile.d/jre.sh ]; then
+    source /etc/profile.d/jre.sh
+  elif [ -e /etc/profile.d/jdk6.sh ]; then
+    source /etc/profile.d/jdk6.sh
+  elif [ -e /etc/profile.d/jdk.sh ]; then
+    source /etc/profile.d/jdk.sh
+  fi
+fi
+
 # Detect the location of the java binary
 if [ -z "$JAVACMD" ] || [ "$JAVACMD" = "auto" ] ; then
   if [ -n "$JAVA_HOME"  ] ; then
@@ -691,4 +704,4 @@
     ;;
   *)
     invoke_task
-esac
\ No newline at end of file
+esac
