--- bin/activemq.orig	2013-06-03 11:22:23.389091695 +0200
+++ bin/activemq	2013-06-03 13:01:31.288286403 +0200
@@ -308,6 +308,15 @@
     CLASSPATH="`cygpath --path --unix "$CLASSPATH"`"
 fi
 
+# Detect JAVA_HOME of the installed java-runtime
+if [ -z "$JAVA_HOME" ]; then
+  if [ -e /etc/profile.d/jre6.sh ]; then
+    source /etc/profile.d/jre6.sh
+  elif [ -e /etc/profile.d/jre.sh ]; then
+    source /etc/profile.d/jre.sh
+  fi
+fi
+
 # Detect the location of the java binary
 if [ -z "$JAVACMD" ] || [ "$JAVACMD" = "auto" ] ; then
   if [ -n "$JAVA_HOME"  ] ; then
@@ -691,4 +700,4 @@
     ;;
   *)
     invoke_task
-esac
\ No newline at end of file
+esac
