diff -Naur A/easy-ssh-connection/contents/code/main.py B/easy-ssh-connection/contents/code/main.py
--- A/easy-ssh-connection/contents/code/main.py	2013-09-10 23:44:21.290602129 -0400
+++ B/easy-ssh-connection/contents/code/main.py	2013-09-10 23:53:10.702886795 -0400
@@ -25,9 +25,10 @@
 
 		QObject.connect(self.svg, SIGNAL("clicked()"), self.showPopup)
 
+		self.layout.setSpacing(0)
 		self.layout.addItem(self.svg)
 		self.applet.setLayout(self.layout)
-		self.resize(32, 32)
+		self.resize(128, 128)
 
 		self.servers = {}
 		self.selectedServer = ""
@@ -114,7 +115,7 @@
 			subprocess.Popen(str("konsole " + newTab + " -e " + command), shell=True)
 
 	def getConfig(self):
-		return KConfig(self.package().filePath("config", "main.rc"), KConfig.SimpleConfig)
+		return KConfig(unicode(KGlobal.dirs().localkdedir()) + "share/apps/easy-ssh-connection/main.rc", KConfig.SimpleConfig)
 
 	def showConfigurationInterface(self):
 		self.dlg = KDialog()
