=== modified file 'ubuntuone/couch/auth.py'
--- ubuntuone/couch/auth.py	2011-05-25 14:04:22 +0000
+++ ubuntuone/couch/auth.py	2011-08-09 13:09:48 +0000
@@ -134,10 +134,13 @@
     for key, value in urlparse.parse_qs(query).items():
         parameters[key] = value[0]
 
+    request_len = len(request_body) if request_body else 0
+    timeout = 10 * (request_len / 1024 / 1024 + 1)  # 10 seconds per megabyte
+
     oauth_header = get_oauth_request_header(
         consumer, token, url, http_method, signature_method, parameters)
     headers = headers or {}
     headers.update(oauth_header)
-    http = httplib2.Http(timeout=10)
+    http = httplib2.Http(timeout=timeout)
     return http.request(
         url, method=http_method, headers=headers, body=request_body)

