=== modified file 'ubuntuone/couch/auth.py'
--- ubuntuone/couch/auth.py	2011-10-19 16:07:17 +0000
+++ ubuntuone/couch/auth.py	2012-07-11 17:40:23 +0000
@@ -23,7 +23,7 @@
 from oauth import oauth
 from dbus.mainloop.glib import DBusGMainLoop
 
-from ubuntuone.platform.credentials.linux import (
+from ubuntuone.platform.credentials.dbus_service import (
     DBUS_BUS_NAME,
     DBUS_CREDENTIALS_PATH,
 )
@@ -141,7 +141,7 @@
         parameters[key] = value[0]
 
     request_len = len(request_body) if request_body else 0
-    timeout = 10 * (request_len / 1024 / 1024 + 1)  # 10 seconds per megabyte
+    timeout = 30 * (request_len / 1024 / 1024 + 1)  # 30 seconds per megabyte
 
     oauth_header = get_oauth_request_header(
         consumer, token, url, http_method, signature_method, parameters)

=== modified file 'ubuntuone/tests/test_auth.py'
--- ubuntuone/tests/test_auth.py	2011-10-19 16:07:17 +0000
+++ ubuntuone/tests/test_auth.py	2012-07-11 17:40:23 +0000
@@ -22,7 +22,7 @@
 from ubuntuone.couch.auth import (
     HMAC_SHA1, get_oauth_request_header, get_oauth_credentials,
     get_oauth_token, request, CredentialsNotFound, NO_CREDENTIALS)
-from ubuntuone.platform.credentials.linux import (
+from ubuntuone.platform.credentials.dbus_service import (
     DBUS_BUS_NAME,
     DBUS_CREDENTIALS_PATH,
 )
