diff -aur old/Makefile new/Makefile
--- old/Makefile	2011-10-23 03:56:55.000000000 +1100
+++ new/Makefile	2012-05-24 18:32:28.227669893 +1000
@@ -15,21 +15,9 @@
 
 ifneq ($(KERNELRELEASE),)
 
-  LINUXVER_GOODFOR_CFG80211:=$(strip $(shell \
-    if [ "$(VERSION)" -ge "2" -a "$(PATCHLEVEL)" -ge "6" -a "$(SUBLEVEL)" -ge "32" ]; then \
-      echo TRUE; \
-    else \
-      echo FALSE; \
-    fi \
-  ))
+  LINUXVER_GOODFOR_CFG80211:=TRUE
 
-    LINUXVER_WEXT_ONLY:=$(strip $(shell \
-    if [ "$(VERSION)" -ge "2" -a "$(PATCHLEVEL)" -ge "6" -a "$(SUBLEVEL)" -ge "17" ]; then \
-      echo FALSE; \
-    else \
-      echo TRUE; \
-    fi \
-  ))
+  LINUXVER_WEXT_ONLY:=FALSE
 
   ifneq ($(API),)
     ifeq ($(API), CFG80211)
diff -aur old/src/wl/sys/wl_iw.h new/src/wl/sys/wl_iw.h
--- old/src/wl/sys/wl_iw.h	2011-10-23 03:56:55.000000000 +1100
+++ new/src/wl/sys/wl_iw.h	2012-05-24 18:32:31.354336075 +1000
@@ -15,6 +15,7 @@
 #ifndef _wl_iw_h_
 #define _wl_iw_h_
 
+#include <linux/semaphore.h>
 #include <linux/wireless.h>
 
 #include <typedefs.h>
diff -aur old/src/wl/sys/wl_linux.c new/src/wl/sys/wl_linux.c
--- old/src/wl/sys/wl_linux.c	2011-10-23 03:56:55.000000000 +1100
+++ new/src/wl/sys/wl_linux.c	2012-05-24 18:32:35.214336356 +1000
@@ -40,7 +40,6 @@
 #include <linux/pci_ids.h>
 #define WLC_MAXBSSCFG		1	
 
-#include <asm/system.h>
 #include <asm/io.h>
 #include <asm/irq.h>
 #include <asm/pgtable.h>
@@ -385,7 +384,7 @@
 #endif
 	.ndo_get_stats = wl_get_stats,
 	.ndo_set_mac_address = wl_set_mac_address,
-	.ndo_set_multicast_list = wl_set_multicast_list,
+	.ndo_set_rx_mode = wl_set_multicast_list,
 	.ndo_do_ioctl = wl_ioctl
 };
 
