--- src/xevdevkeyinfo.cpp	2008-02-28 19:37:06.000000000 +0100
+++ src/xevdevkeyinfo.cpp	2009-03-20 12:35:00.000000000 +0100
@@ -28,7 +28,9 @@
 #include <sys/time.h>
 #include <sys/wait.h>
 #include <sys/ioctl.h>
+#include <sys/types.h>
 #include <unistd.h>
+#include <stdlib.h>
 #include <iostream>
 #include <vector>
 
--- src/xevdevserver.cpp	2008-02-28 19:21:30.000000000 +0100
+++ src/xevdevserver.cpp	2009-03-20 12:35:23.000000000 +0100
@@ -30,6 +30,7 @@
 #include <sys/time.h>
 #include <sys/wait.h>
 #include <unistd.h>
+#include <stdlib.h>
 #include <iostream>
 #include <vector>
 
--- src/xkeyboardeventdevice.cpp	2008-02-28 19:52:04.000000000 +0100
+++ src/xkeyboardeventdevice.cpp	2009-03-20 12:36:17.000000000 +0100
@@ -28,6 +28,8 @@
 #include "keyboardaction.h"
 #include <linux/input.h>
 #include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
 
 #include <iostream>
 
--- src/xserverconnection.cpp 2008-02-27 00:17:58.000000000 +0300
+++ src/xserverconnection.cpp 2009-06-23 20:03:07.000000000 +0400
@@ -27,5 +27,6 @@
 #include <X11/extensions/XTest.h>
 #include <X11/extensions/XKB.h>
 #include <iostream>
+#include <stdint.h>
 
 XServerConnection::XServerConnection()
 {
