diff -Naur old/CPAN/XML/Parser/Expat.pm new/CPAN/XML/Parser/Expat.pm
--- old/CPAN/XML/Parser/Expat.pm	2011-09-12 03:35:08.000000000 +1000
+++ new/CPAN/XML/Parser/Expat.pm	2012-06-27 07:16:38.491684969 +1000
@@ -490,7 +490,7 @@
   local(*FILE);
   open(FILE, $_[0]) or  croak "Couldn't open $_[0]:\n$!";
   binmode(FILE);
-  my $ret = $self->parse(*FILE);
+  my $ret = $self->parse(\*FILE);
   close(FILE);
   $ret;
 }
@@ -676,7 +676,7 @@
                       'End'   => \&eh,
                       'Char'  => \&ch);
  open(FOO, '<', 'info.xml') or die "Couldn't open";
- $parser->parse(*FOO);
+ $parser->parse(\*FOO);
  close(FOO);
  # $parser->parse('<foo id="me"> here <em>we</em> go </foo>');
 
diff -Naur old/CPAN/XML/Simple.pm new/CPAN/XML/Simple.pm
--- old/CPAN/XML/Simple.pm	2011-09-12 03:35:08.000000000 +1000
+++ new/CPAN/XML/Simple.pm	2012-06-27 07:16:38.495018303 +1000
@@ -333,7 +333,7 @@
     # $tree = $xp->parsefile($filename);  # Changed due to prob w/mod_perl
     local(*XML_FILE);
     open(XML_FILE, '<', $filename) || croak qq($filename - $!);
-    $tree = $xp->parse(*XML_FILE);
+    $tree = $xp->parse(\*XML_FILE);
     close(XML_FILE);
   }
   else {
diff -Naur old/Slim/Control/Request.pm new/Slim/Control/Request.pm
--- old/Slim/Control/Request.pm	2012-03-14 23:28:25.000000000 +1100
+++ new/Slim/Control/Request.pm	2012-06-27 07:17:57.325049118 +1000
@@ -1531,7 +1531,8 @@
 
 # get a parameter by name
 sub getParam {
-	return $_[0]->{'_params'}->{ $_[1] };
+	my $r = $_[0]->{'_params'}->{ $_[1] };
+	return $r if ($r) or undef;
 }
 
 # delete a parameter by name
