diff -aur old/CPAN/XML/Parser/Expat.pm new/CPAN/XML/Parser/Expat.pm
--- old/CPAN/XML/Parser/Expat.pm	2011-09-12 03:35:08.000000000 +1000
+++ new/CPAN/XML/Parser/Expat.pm	2011-11-03 17:02:06.238832780 +1100
@@ -490,7 +490,7 @@
   local(*FILE);
   open(FILE, $_[0]) or  croak "Couldn't open $_[0]:\n$!";
   binmode(FILE);
-  my $ret = $self->parse(*FILE);
+  my $ret = $self->parse(\*FILE);
   close(FILE);
   $ret;
 }
@@ -676,7 +676,7 @@
                       'End'   => \&eh,
                       'Char'  => \&ch);
  open(FOO, '<', 'info.xml') or die "Couldn't open";
- $parser->parse(*FOO);
+ $parser->parse(\*FOO);
  close(FOO);
  # $parser->parse('<foo id="me"> here <em>we</em> go </foo>');
 
diff -aur old/CPAN/XML/Simple.pm new/CPAN/XML/Simple.pm
--- old/CPAN/XML/Simple.pm	2011-09-12 03:35:08.000000000 +1000
+++ new/CPAN/XML/Simple.pm	2011-11-03 16:59:38.348520576 +1100
@@ -333,7 +333,7 @@
     # $tree = $xp->parsefile($filename);  # Changed due to prob w/mod_perl
     local(*XML_FILE);
     open(XML_FILE, '<', $filename) || croak qq($filename - $!);
-    $tree = $xp->parse(*XML_FILE);
+    $tree = $xp->parse(\*XML_FILE);
     close(XML_FILE);
   }
   else {
