diff -Narup kaconnect-1.1.1/connector.h kaconnect-1.1.1.new/connector.h
--- kaconnect-1.1.1/connector.h	2002-02-25 17:53:07.000000000 +0100
+++ kaconnect-1.1.1.new/connector.h	2007-11-17 12:55:24.000000000 +0100
@@ -24,7 +24,7 @@ class ConnectorWidget : public QWidget {
         void arrow(int x1, int y1, int x2, int y2); 
    
     public:
-        ConnectorWidget::ConnectorWidget(QWidget *parent, snd_seq_t *pseq); 
+        ConnectorWidget(QWidget *parent, snd_seq_t *pseq); 
         void set_r_listBox(PortListBox *p_r_portlist);
         void set_w_listBox(PortListBox *p_w_portlist);
                    
diff -Narup kaconnect-1.1.1/make_kaconnect kaconnect-1.1.1.new/make_kaconnect
--- kaconnect-1.1.1/make_kaconnect	2002-07-16 13:00:36.000000000 +0200
+++ kaconnect-1.1.1.new/make_kaconnect	2007-11-17 12:54:57.000000000 +0100
@@ -1,4 +1,4 @@
-QT_LIB_DIR=-L/usr/lib/qt3/lib
+QT_LIB_DIR=-L/opt/qt/lib
 
 kaconnect: kaconnect.o alsa_access.o alsa_access.moc.o connector.o connector.moc.o portlistbox.o portlistbox.moc.o 
 	gcc kaconnect.o alsa_access.o alsa_access.moc.o \
@@ -10,37 +10,37 @@ kaconnect: kaconnect.o alsa_access.o als
 
 kaconnect.o: kaconnect.cpp
 	gcc -c kaconnect.cpp \
-	-I/usr/lib/qt3/include 
+	-I/opt/qt/include 
 
 alsa_access.o: alsa_access.h alsa_access.cpp  
 	gcc -c alsa_access.cpp \
-	-I/usr/lib/qt3/include
+	-I/opt/qt/include
 
 alsa_access.moc.o: alsa_access.moc.cpp
 	gcc -c alsa_access.moc.cpp \
-	-I/usr/lib/qt3/include
+	-I/opt/qt/include
 
 alsa_access.moc.cpp: alsa_access.h alsa_access.cpp 
-	/usr/lib/qt3/bin/moc alsa_access.h -o alsa_access.moc.cpp
+	/opt/qt/bin/moc alsa_access.h -o alsa_access.moc.cpp
 
 connector.o: connector.h connector.cpp
 	gcc -c connector.cpp \
-	-I/usr/lib/qt3/include
+	-I/opt/qt/include
 
 connector.moc.o: connector.moc.cpp
 	gcc -c connector.moc.cpp \
-	-I/usr/lib/qt3/include
+	-I/opt/qt/include
 
 connector.moc.cpp: connector.h connector.cpp portlistbox.h
-	/usr/lib/qt3/bin/moc connector.h -o connector.moc.cpp
+	/opt/qt/bin/moc connector.h -o connector.moc.cpp
 
 portlistbox.o: portlistbox.h portlistbox.cpp
 	gcc -c portlistbox.cpp \
-	-I/usr/lib/qt3/include
+	-I/opt/qt/include
 
 portlistbox.moc.o: portlistbox.moc.cpp
 	gcc -c portlistbox.moc.cpp \
-	-I/usr/lib/qt3/include
+	-I/opt/qt/include
 
 portlistbox.moc.cpp: portlistbox.h portlistbox.cpp 
-	/usr/lib/qt3/bin/moc portlistbox.h -o portlistbox.moc.cpp
+	/opt/qt/bin/moc portlistbox.h -o portlistbox.moc.cpp
