--- applet/SmoothTasks/Applet.cpp.orig	2011-08-11 23:08:37.000000000 -0400
+++ applet/SmoothTasks/Applet.cpp	2011-08-11 23:09:25.000000000 -0400
@@ -260,7 +260,7 @@
 	KConfigGroup launcherCg(&cg, "Launchers");
 	
 	QVariantList launcherProperties;
-	launcherProperties.append(launcherItem->url().url());
+	launcherProperties.append(launcherItem->launcherUrl().url());
 	launcherProperties.append(launcherItem->icon().name());
 	launcherProperties.append(launcherItem->name());
 	launcherProperties.append(launcherItem->genericName());
