From 0a9833f1a733e68086588f730acc3bb48f006dc3 Mon Sep 17 00:00:00 2001
From: Yosef Or Boczko <yosefor3@walla.com>
Date: Sun, 28 Apr 2013 17:36:03 +0300
Subject: [PATCH] Deleted libgd

---
 gedit/Makefile.am | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/gedit/Makefile.am b/gedit/Makefile.am
index 0b06442..c21b594 100644
--- a/gedit/Makefile.am
+++ b/gedit/Makefile.am
@@ -8,7 +8,6 @@ pkglib_LTLIBRARIES = libgedit-private.la
 INCLUDES =								\
 	-I$(top_srcdir)							\
 	-I$(srcdir)							\
-	-I$(top_srcdir)/libgd						\
 	$(GEDIT_CFLAGS)							\
 	$(GTK_MAC_CFLAGS)						\
 	$(WARN_CFLAGS)							\
@@ -35,8 +34,7 @@ endif
 
 libgedit_la_LDFLAGS = -export-dynamic -no-undefined -export-symbols-regex "^[[^_]].*"
 
-libgedit_la_LIBADD = \
-	$(top_builddir)/libgd/libgd.la
+libgedit_la_LIBADD = 
 
 # GEDIT_LIBS must be the last to ensure correct order on some platforms
 libgedit_la_LIBADD += $(GEDIT_LIBS) $(GTK_MAC_LIBS)
-- 
1.8.2.2

