diff -rauN cdp-0.33-13/Makefile cdp-0.33-13-compress-patch/Makefile
--- cdp-0.33-13/Makefile	1999-08-18 23:23:26.000000000 +0200
+++ cdp-0.33-13-compress-patch/Makefile	2014-02-06 15:21:11.628902891 +0100
@@ -29,13 +29,13 @@
 .c.o:
 	gcc $(COMP_OPT) -c $(DBG) $*.c 
 
-all: cdp cdp.1.Z
+all: cdp cdp.1.gz
 	
 cdp: $(OBJS)
 	gcc -o cdp $(OBJS) $(LINK_OPT) $(DBG) $(LIBS)
 
-cdp.1.Z: cdp.1
-	groff -Tascii -man cdp.1 | compress >cdp.1.Z
+cdp.1.gz: cdp.1
+	groff -Tascii -man cdp.1 | gzip >cdp.1.gz
 		
 display.o: display.c display.h struct.h
 
@@ -53,13 +53,13 @@
 
 cdp.o: cdp.c cdp.h misc.h display.h struct.h
 
-install: cdp cdp.1.Z
+install: cdp cdp.1.gz
 	chmod 755 cdp
-	chmod 644 cdp.1.Z
+	chmod 644 cdp.1.gz
 	install -m 755 -s cdp $(DESTDIR)/usr/bin
 	ln -sf cdp $(DESTDIR)/usr/bin/cdplay
 	install -m 644 cdp.1 $(DESTDIR)/usr/man/man1
 
 clean:
-	rm -f cdp cdp.1.Z *.o core *~
+	rm -f cdp cdp.1.gz *.o core *~
 
