diff -rauN cdp-0.33-13/display.c cdp-0.33-13-stdio-getline-patch/display.c
--- cdp-0.33-13/display.c	1995-11-10 07:10:42.000000000 +0100
+++ cdp-0.33-13-stdio-getline-patch/display.c	2010-07-27 18:49:06.000000000 +0200
@@ -540,7 +540,7 @@
     displayStatusPrintf( pInfo, pCD->trk[ track - 1 ].songname );
   
     line = TOC_TOP_LINE + 1 + track - 1 - pInfo->scrStartTrack;
-    code = getline( line, STARTX_TRK_NAME, WIDTH_TRK_NAME + 2,REVERSE_COLOR,
+    code = cdp_getline( line, STARTX_TRK_NAME, WIDTH_TRK_NAME + 2,REVERSE_COLOR,
 		    NORMAL_COLOR, MARKED_COLOR | A_BOLD, 
  		    lpszNewName, SONG_SIZE, 
 		    pCD->trk[ track - 1 ].songname );
@@ -573,7 +573,7 @@
     resetCurrTrackMark( pInfo );
     pCD = pInfo->pCD;
 
-    code = getline( 2, 19, WIDTH_CDNAME, REVERSE_COLOR,
+    code = cdp_getline( 2, 19, WIDTH_CDNAME, REVERSE_COLOR,
 		    NORMAL_COLOR, MARKED_COLOR | A_BOLD, 
 		    lpszNewName, sizeof( pCD->cdname ), 
 		    pCD->cdname );
@@ -604,7 +604,7 @@
     resetCurrTrackMark( pInfo );
     pCD = pInfo->pCD;
 
-    code = getline( 1, 19, WIDTH_ARTIST,REVERSE_COLOR,
+    code = cdp_getline( 1, 19, WIDTH_ARTIST,REVERSE_COLOR,
 		    NORMAL_COLOR, MARKED_COLOR | A_BOLD, 
 		    lpszNewName, sizeof( pCD->artist ), 
 		    pCD->artist );
diff -rauN cdp-0.33-13/getline.c cdp-0.33-13-stdio-getline-patch/getline.c
--- cdp-0.33-13/getline.c	1995-11-10 07:10:42.000000000 +0100
+++ cdp-0.33-13-stdio-getline-patch/getline.c	2010-07-27 18:48:51.000000000 +0200
@@ -139,7 +139,7 @@
 }
 			     
 
-int     getline( int      y,
+int     cdp_getline( int      y,
 		 int      x, 
 		 int      width,
  		 chtype   attrText,
diff -rauN cdp-0.33-13/getline.h cdp-0.33-13-stdio-getline-patch/getline.h
--- cdp-0.33-13/getline.h	1995-11-10 07:10:42.000000000 +0100
+++ cdp-0.33-13-stdio-getline-patch/getline.h	2010-07-27 18:48:43.000000000 +0200
@@ -7,7 +7,7 @@
 #define  __getline__H
 
 
-int     getline( int      y, 
+int     cdp_getline( int      y, 
 		 int      x,
 		 int      width,
  		 chtype   attrText,
