=== modified file 'sc_linux.c'
--- x/sc_linux.c	2009-08-12 13:20:43 +0000
+++ x/sc_linux.c	2011-11-08 23:36:06 +0000
@@ -39,7 +39,7 @@
 
 #ifdef HAVE_LIBESD
 static int esdblock = 0;	/* 1 to block ESD; 0 to non-block */
-static int esdrecord = 0;	/* 1 to use ESD record mode; 0 to use ESD monitor mode */
+static int esdrecord = 1;	/* 1 to use ESD record mode; 0 to use ESD monitor mode */
 #endif
 
 static int sc_chans = 0;
@@ -87,7 +87,7 @@
 
 /* show system error and close sound device if given ioctl status is bad */
 static void
-check_status_ioctl(int d, int request, void *argp, int line)
+check_status_ioctl(int d, unsigned int request, void *argp, int line)
 {
   if (ioctl(d, request, argp) < 0) {
     snd_errormsg1 = "sound ioctl";

