diff -Naur trunk.orig/qtikzconfig.pri trunk.new/qtikzconfig.pri
--- orig/qtikzconfig.pri	2013-04-08 22:23:08.780509972 +0100
+++ new/qtikzconfig.pri	2013-04-08 22:24:54.250467478 +0100
@@ -40,11 +40,11 @@
 CONFIG += release
 
 # qmake command:
-#QMAKECOMMAND = qmake-qt4
+QMAKECOMMAND = qmake-qt4
 # lrelease command:
-#LRELEASECOMMAND = lrelease-qt4
+LRELEASECOMMAND = lrelease-qt4
 # qcollectiongenerator command:
-#QCOLLECTIONGENERATORCOMMAND = qcollectiongenerator
+QCOLLECTIONGENERATORCOMMAND = qcollectiongenerator-qt4
 
 # TikZ documentation default file path:
 TIKZ_DOCUMENTATION_DEFAULT =
"/usr/share/doc/texmf/pgf/pgfmanual.pdf.gz"
