diff -u xterm-304/button.c xterm/button.c
--- xterm-304/button.c	2014-04-22 02:50:10.000000000 +0200
+++ xterm/button.c	2014-05-14 14:02:31.533255078 +0200
@@ -2353,6 +2353,55 @@
     }
 }
 
+/* ARGSUSED */
+void
+HandleSelectNeedle(Widget w,
+		   XEvent *event GCC_UNUSED,
+		   String *params,
+		   Cardinal *num_params)
+{
+    if (!num_params || *num_params < 1)
+	return;
+
+    XtermWidget xw;
+    String needle = params[0];
+    size_t len = strlen(needle);
+
+    if ((xw = getXtermWidget(w)) != 0) {
+	TScreen *screen = TScreenOf(xw);
+	TRACE(("HandleSelectNeedle\n"));
+
+	CELL start = screen->cursorp;
+	if(!isSameCELL(&screen->startH, &screen->endH))
+	    start = screen->startH;
+
+	CELL cell = start;
+	for (; cell.row >= 0; cell.row--) {
+	    /* we could go to -screen->savedlines, but it doesn't
+	     * seem to work as intended */
+	    LineData *ld = getLineData(screen, cell.row);
+	    Dimension limit = ld->lineSize;
+	    if (cell.row == start.row)
+		limit = cell.col;
+	    for (cell.col = limit - len; cell.col >= 0; cell.col--) {
+		size_t i;
+		for (i = 0; i < len; i++) {
+		    /* are String and CharData* the same thing? */
+		    if (ld->charData[cell.col + i] != needle[i])
+			break;
+		}
+		if (i == len) { /* needle found */
+		    screen->selectUnit = Select_WORD;
+		    ComputeSelect(xw, &cell, &cell, False);
+		    return;
+		}
+	    }
+	}
+	/* Deselect so that we can loop again. */
+	DisownSelection(xw);
+    }
+}
+
 static void
 TrackDown(XtermWidget xw, XButtonEvent *event)
 {
diff -u xterm-304/charproc.c xterm/charproc.c
--- xterm-304/charproc.c	2014-05-02 23:48:33.000000000 +0200
+++ xterm/charproc.c	2014-05-14 14:19:33.093237072 +0200
@@ -263,6 +263,7 @@
     { "select-cursor-start",	HandleKeyboardSelectStart },
     { "select-end",		HandleSelectEnd },
     { "select-extend",		HandleSelectExtend },
+    { "select-needle",		HandleSelectNeedle },
     { "select-set",		HandleSelectSet },
     { "select-start",		HandleSelectStart },
     { "send-signal",		HandleSendSignal },
diff -u xterm-304/xterm.h xterm/xterm.h
--- xterm-304/xterm.h	2014-05-03 21:30:17.000000000 +0200
+++ xterm/xterm.h	2014-05-14 14:19:43.483235439 +0200
@@ -804,6 +804,7 @@
 extern void HandleKeyboardStartExtend  PROTO_XT_ACTIONS_ARGS;
 extern void HandleSelectEnd            PROTO_XT_ACTIONS_ARGS;
 extern void HandleSelectExtend         PROTO_XT_ACTIONS_ARGS;
+extern void HandleSelectNeedle         PROTO_XT_ACTIONS_ARGS;
 extern void HandleSelectSet            PROTO_XT_ACTIONS_ARGS;
 extern void HandleSelectStart          PROTO_XT_ACTIONS_ARGS;
 extern void HandleStartExtend          PROTO_XT_ACTIONS_ARGS;
diff -u xterm-304/xterm.man xterm/xterm.man
--- xterm-304/xterm.man	2014-04-14 20:42:54.000000000 +0200
+++ xterm/xterm.man	2014-05-14 15:00:15.749856506 +0200
@@ -5920,6 +5920,10 @@
 This action tracks the pointer and extends the selection.
 It should only be bound to Motion events.
 .TP 8
+.B "select-needle(\fIneedle\fP)"
+This action makes a word selection on the first occurrence of the needle string
+preceding the current selection, if any, or the cursor.
+.TP 8
 .B "select-set()"
 This action stores text that corresponds to the current selection,
 without affecting the selection mode.
