--- openfire_src/src/java/org/jivesoftware/openfire/server/ServerDialback.java.of443	2011-04-04 19:36:09.000000000 -0700
+++ openfire_src/src/java/org/jivesoftware/openfire/server/ServerDialback.java	2011-04-04 19:36:46.000000000 -0700
@@ -236,8 +236,7 @@
             stream.append(" xmlns=\"jabber:server\"");
             stream.append(" to=\"").append(remoteDomain).append("\"");
             stream.append(" from=\"").append(localDomain).append("\"");
-            stream.append(" xmlns:db=\"jabber:server:dialback\"");
-            stream.append(" version=\"1.0\">");
+            stream.append(" xmlns:db=\"jabber:server:dialback\">");
             connection.deliverRawText(stream.toString());
 
             // Set a read timeout (of 5 seconds) so we don't keep waiting forever
