--- GoogleCalendar-old/GoogleCalendarScreenlet.py	2010-11-27 06:35:06.000000000 -0500
+++ GoogleCalendar/GoogleCalendarScreenlet.py	2010-12-20 21:39:51.098375189 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf -*-
 
 # This application is released under the GNU General Public License 
@@ -896,9 +896,9 @@
 		# events pane tooltip area
 		if (self.showEvents and not self.autoHide) or (self.showEvents and self.autoHide and self.eventsFound):
 			if self.__events_area:
-				self.__events_area.set_size_request(self.eventsWidth * self.scale, self.eventsHeight * self.scale)
+				self.__events_area.set_size_request(int (self.eventsWidth * self.scale), int (self.eventsHeight * self.scale))
 			if self.__all_area:
-				self.__all_area.move(self.__events_area, self.xpos() * self.scale, self.padding * self.scale)
+				self.__all_area.move(self.__events_area, int (self.xpos() * self.scale), int (self.padding * self.scale))
 			self.__events_area.set_tooltip_text(self.scroll_tooltip)
 			self.window.show_all()        
 		else:

