From d8fa8db1aadf3012cb6bfc6cf038c5343c30ceb7 Mon Sep 17 00:00:00 2001
From: Bill Kolokithas <kolokithas.b@gmail.com>
Date: Wed, 17 Jul 2013 00:55:04 +0300
Subject: [PATCH] check return value of strchr

---
 src/desktop.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/desktop.cc b/src/desktop.cc
index 810f163..782f7ff 100644
--- a/src/desktop.cc
+++ b/src/desktop.cc
@@ -59,6 +59,9 @@ bool read_desktop_file(FILE *file, desktop_file_t &values, string_mapper sm)
             char *key=line, *value;
             // Split that string in place
             value = strchr(line, '=');
+            if (value == NULL)
+                continue;
+
             value[0] = 0;
             value++;
 
-- 
1.8.3.3

