diff -r -U 3 a/get-iab b/get-iab
--- a/get-iab	2007-01-24 12:42:54.000000000 +0530
+++ b/get-iab	2007-06-24 16:00:16.000000000 +0530
@@ -31,7 +31,6 @@
 use warnings;
 use strict;
 use Getopt::Std;
-use LWP::Simple;
 #
 my $default_url = 'http://standards.ieee.org/regauth/oui/iab.txt';
 my $default_filename='ieee-iab.txt';
@@ -86,7 +85,7 @@
 # Fetch the content from the URL
 #
 print "Fetching IAB data from $url\n" if $verbose;
-my $content = get $url;
+my $content = `wget -q -O - $url`;
 die "Could not get IAB data from $url\n" unless defined $content;
 my $content_length = length($content);
 die "Zero-sized response from from $url\n" unless ($content_length > 0);
diff -r -U 3 a/get-oui b/get-oui
--- a/get-oui	2007-01-24 12:42:54.000000000 +0530
+++ b/get-oui	2007-06-24 16:00:23.000000000 +0530
@@ -27,7 +27,6 @@
 use warnings;
 use strict;
 use Getopt::Std;
-use LWP::Simple;
 #
 my $default_url = 'http://standards.ieee.org/regauth/oui/oui.txt';
 my $default_filename='ieee-oui.txt';
@@ -80,7 +79,7 @@
 # Fetch the content from the URL
 #
 print "Fetching OUI data from $url\n" if $verbose;
-my $content = get $url;
+my $content = `wget -q -O - $url`;
 die "Could not get OUI data from $url\n" unless defined $content;
 my $content_length = length($content);
 die "Zero-sized response from from $url\n" unless ($content_length > 0);
Only in a: ieee-oui.txt
Only in b: ieee-oui.txt.bak
