--- creavision/libwebcam/libwebcam.c.orig	2013-05-30 18:59:07.000000000 +0200
+++ creavision/libwebcam/libwebcam.c	2013-09-12 10:16:36.047604026 +0200
@@ -2137,8 +2137,6 @@
 		case V4L2_CID_GAIN:					return CC_GAIN;
 		case V4L2_CID_HFLIP:				break;	// not supported by libwebcam
 		case V4L2_CID_VFLIP:				break;	// not supported by libwebcam
-		case V4L2_CID_HCENTER:				break;	// deprecated
-		case V4L2_CID_VCENTER:				break;	// deprecated
 #ifdef V4L2_CID_POWER_LINE_FREQUENCY
 		case V4L2_CID_POWER_LINE_FREQUENCY:		return CC_POWER_LINE_FREQUENCY;
 #endif