--- zsnes_1_51/src/autogen.sh	2013-05-19 22:31:27.000000000 +1000
+++ zsnes_1_51_lirc/src/autogen.sh	2013-05-19 22:30:42.000000000 +1000
@@ -3,7 +3,7 @@
 echo "Generating build information using aclocal and autoconf..."
 
 # Regenerate configuration files
-aclocal  --acdir=`sdl-config --prefix`/share/aclocal	# thanks asfand
+aclocal  --system-acdir=`sdl-config --prefix`/share/aclocal	# thanks asfand
 autoconf
 
 # Run configure for this platform, or simply update Makefile
