diff -rupN nop/scripts/drbd drbd-8.3.7/scripts/drbd
--- nop/scripts/drbd	2010-01-07 11:09:34.000000000 +0200
+++ drbd-8.3.7/scripts/drbd	2010-02-06 22:46:46.037783913 +0200
@@ -217,7 +217,7 @@ case "$1" in
 	log_end_msg 0
 	;;
     *)
-	echo "Usage: /etc/init.d/drbd {start|stop|status|reload|restart|force-reload}"
+	echo "Usage: /etc/rc.d/drbd {start|stop|status|reload|restart|force-reload}"
 	exit 1
 	;;
 esac
