--- src/compositor/meta-shadow-factory.c	2013-04-30 23:21:27.000000000 +0200
+++ src/compositor/meta-shadow-factory-changed.c	2013-06-16 13:05:33.778689650 +0200
@@ -123,17 +123,17 @@
 /* The first element in this array also defines the default parameters
  * for newly created classes */
 MetaShadowClassInfo default_shadow_classes[] = {
-  { "normal",       { 6, -1, 0, 3, 255 }, { 3, -1, 0, 3, 128 } },
-  { "dialog",       { 6, -1, 0, 3, 255 }, { 3, -1, 0, 3, 128 } },
-  { "modal_dialog", { 6, -1, 0, 1, 255 }, { 3, -1, 0, 3, 128 } },
-  { "utility",      { 3, -1, 0, 1, 255 }, { 3, -1, 0, 1, 128 } },
-  { "border",       { 6, -1, 0, 3, 255 }, { 3, -1, 0, 3, 128 } },
-  { "menu",         { 6, -1, 0, 3, 255 }, { 3, -1, 0, 0, 128 } },
+  { "normal",       { 25, -1, 0, 20, 200 }, { 8, -1, 0, 4, 128 } },
+  { "dialog",       { 20, -1, 0, 3, 200 }, { 8, -1, 0, 4, 128 } },
+  { "modal_dialog", { 20, -1, 0, 1, 200 }, { 8, -1, 0, 4, 128 } },
+  { "utility",      { 10, -1, 0, 1, 200 }, { 8, -1, 0, 3, 128 } },
+  { "border",       { 20, -1, 0, 3, 200 }, { 8, -1, 0, 4, 128 } },
+  { "menu",         { 20, -1, 0, 3, 200 }, { 8, -1, 0, 2, 128 } },
 
-  { "popup-menu",    { 1, -1, 0, 1, 128 }, { 1, -1, 0, 1, 128 } },
+  { "popup-menu",    { 20, -1, 0, 1, 128 }, { 8, -1, 0, 1, 128 } },
 
-  { "dropdown-menu", { 1, 10, 0, 1, 128 }, { 1, 10, 0, 1, 128 } },
-  { "attached",      { 1, -1, 0, 1, 128 }, { 1, -1, 0, 1, 128 } }
+  { "dropdown-menu", { 20, 10, 0, 3, 128 }, { 8, 10, 0, 3, 128 } },
+  { "attached",      { 20, -1, 0, 3, 128 }, { 8, -1, 0, 3, 128 } }
 };
 
 G_DEFINE_TYPE (MetaShadowFactory, meta_shadow_factory, G_TYPE_OBJECT);
