diff -rupN linuxconsoletools-1.4.1-original/utils/inputattach.c linuxconsoletools-1.4.1/utils/inputattach.c
--- linuxconsoletools-1.4.1-original/inputattach.c	2011-06-25 15:16:58.000000000 +0200
+++ linuxconsoletools-1.4.1/inputattach.c	2011-07-13 13:15:37.231000078 +0200
@@ -459,6 +459,17 @@ static int dump_init(int fd, unsigned lo
 		}
 }
 
+static int wacom_v_init(int fd, unsigned long *id, unsigned long *extra)
+{
+	if (write(fd, "BA19\r", 5) != 5)
+                return -1;
+
+	usleep(100 * 1000);
+
+	setline(fd, CS8, B19200);
+	return 0;
+}
+
 struct input_types {
 	const char *name;
 	const char *name2;
@@ -588,6 +599,12 @@ static struct input_types input_types[]
 { "--w8001",		"-w8001",	"Wacom W8001",
 	B38400, CS8,
 	SERIO_W8001,		0x00,	0x00,	0,	NULL },
+{ "--wacom_iv",		"-wacom_iv",	"Wacom protocol 4 tablet",
+	B9600, CS8,
+	SERIO_WACOM_IV,		0x00,	0x00,	0,	NULL },
+{ "--wacom_v",		"-wacom_v",	"Wacom protocol 5 tablet",
+	B9600, CS8,
+	SERIO_WACOM_V,		0x00,	0x00,	0,	wacom_v_init },
 { NULL, NULL, NULL, 0, 0, 0, 0, 0, 0, NULL }
 };
 
diff -rupN linuxconsoletools-1.4.1-original/utils/serio-ids.h linuxconsoletools-1.4.1/utils/serio-ids.h
--- linuxconsoletools-1.4.1-original/serio-ids.h	2010-11-04 07:15:02.000000000 +0100
+++ linuxconsoletools-1.4.1/serio-ids.h	2011-07-13 13:11:55.223000078 +0200
@@ -125,5 +125,11 @@
 #ifndef SERIO_PSMULT
 # define SERIO_PS2MULT		0x3c
 #endif
+#ifndef SERIO_WACOM_IV
+# define SERIO_WACOM_IV		0x3d
+#endif
+#ifndef SERIO_WACOM_V
+# define SERIO_WACOM_V		0x3e
+#endif
 
 #endif
