--- ihtc.cpp.orig	2009-12-17 20:20:27.000000000 +0100
+++ ihtc.cpp	2011-02-02 10:31:37.383333338 +0100
@@ -116,7 +116,7 @@
     foreach (QGraphicsItem *kid, desktop->childItems())
     {
         o = dynamic_cast<QObject*>(kid);
-        if (o && o->inherits("Plasma::ToolBox") || o->inherits("Plasma::DesktopToolBox"))
+        if (o && o->inherits("Plasma::ToolBox") || o->inherits("Plasma::DesktopToolBox") || o->inherits("DesktopToolBox"))
             return kid;
     }
     return 0;
