--- fly8.orig/UNIX/makelx.top	1999-06-27 03:00:09.000000000 +0600
+++ fly8/UNIX/makelx.top	2012-05-18 11:34:56.027491023 +0600
@@ -64,9 +64,9 @@
 OOUT=		-o ./
 EOUT=		-o ./
 
-INSTPATH=	/usr/local/games
-INSTDIR=	fly8
-INSTFLY8=	$(INSTPATH)/$(INSTDIR)
+BINDIR=		$(DESTDIR)/usr/bin
+DATADIR=	$(DESTDIR)/usr/share/fly8
+DOCDIR=		$(DESTDIR)/usr/share/doc/fly8
 RELDIR=		/tmp
 VERNAME=	f8112
 
@@ -150,19 +150,19 @@
 	-(cd utils; $(MAKE) clean)
 
 install:
-	-mkdir $(INSTFLY8)
-	cp	fly8 *.map misc/*.max misc/*.adv misc/*.b50 misc/*.f22 \
-		utils/mac2max utils/max2mac \
+	mkdir -p $(BINDIR) $(DATADIR) $(DOCDIR)
+	cp	fly8		$(BINDIR)/fly8bin
+	cp	utils/mac2max utils/max2mac $(SYSTEM)/fly8udp $(SYSTEM)/udpmgr \
+		$(BINDIR)
+	cp	*.map misc/*.max misc/*.adv misc/*.b50 misc/*.f22 \
 		shapes/*.vxx \
 		parms/*.prm parms/*.nav parms/*.lnd \
 		$(SYSTEM)/*.ini $(SYSTEM)/*.vmd \
 		$(SYSTEM)/fly $(SYSTEM)/flytty $(SYSTEM)/flyfifo \
-		$(SYSTEM)/flyudp $(SYSTEM)/fly8udp $(SYSTEM)/udpmgr \
-		$(SYSTEM)/server $(SYSTEM)/mgr \
-		docs/*.doc docs/bugs.txt \
-		$(INSTFLY8)
-	cp $(SYSTEM)/linux.txt	$(INSTFLY8)/README
-	cp shapes/wf16.vxx	$(INSTFLY8)/plane.vxx
+		$(SYSTEM)/flyudp $(SYSTEM)/server $(SYSTEM)/mgr \
+		$(DATADIR)
+	cp docs/fly8.doc	$(DOCDIR)/fly8.doc
+	cp shapes/wf16.vxx	$(DATADIR)/plane.vxx
 
 reinstall:
 	cp	fly8 *.map \
