diff -aur fly8.orig/parms/expr.awk fly8/parms/expr.awk
--- fly8.orig/parms/expr.awk	1997-05-21 07:19:36.000000000 +0600
+++ fly8/parms/expr.awk	2012-05-18 00:58:23.377037595 +0600
@@ -191,16 +191,16 @@
 }
 
 function exclusive_or(  e, e1) {
-	e = and()
+	e = and_new()
 	while (tok == "^") {
 		advance()
-		e1 = and()
+		e1 = and_new()
 		e = bitwise(e, e1, "^");
 	}
 	return e
 }
 
-function and(  e, e1) {
+function and_new(  e, e1) {
 	e = equality()
 	while (tok == "&") {
 		advance()
diff -aur fly8.orig/shapes/f8shape.awk fly8/shapes/f8shape.awk
--- fly8.orig/shapes/f8shape.awk	1997-05-21 07:19:38.000000000 +0600
+++ fly8/shapes/f8shape.awk	2012-05-18 00:58:35.030370994 +0600
@@ -177,16 +177,16 @@
 }
 
 function exclusive_or(  e, e1) {
-	e = and()
+	e = and_new()
 	while (tok == "^") {
 		advance()
-		e1 = and()
+		e1 = and_new()
 		e = bitwise(e, e1, "^");
 	}
 	return e
 }
 
-function and(  e, e1) {
+function and_new(  e, e1) {
 	e = equality()
 	while (tok == "&") {
 		advance()
