--- fly8.orig/UNIX/fly.ini	1997-08-31 10:14:24.000000000 +0600
+++ fly8/UNIX/fly.ini	2012-05-18 01:13:09.073709268 +0600
@@ -7,7 +7,7 @@
 
 # Name to identify myself, team and home:
 #
-	NEyal		My prefered name
+	NPlayer		My prefered name
 	Tone		My team name
 	Ha18		My home runway
 
@@ -30,14 +30,14 @@
 #
 # To use the X11/PixMap based driver:
 #
-	dvgrx:direct	for fast, NO double buffer operation
-	Vgrx
-	mm
-#
-#	dvgrx:single	double buffered - may be slow
+#	dvgrx:direct	for fast, NO double buffer operation
 #	Vgrx
 #	mm
 #
+	dvgrx:single	double buffered - may be slow
+	Vgrx
+	m1024x768
+#
 # To use the X11/Image based driver:
 #
 #	dvgri:single	standard X11 Image based, may be (very) slow
@@ -84,10 +84,11 @@
 #	dpAstick:zero:d=046:rd=2
 #
 # A mouse:
-	dpmouse:smx=3:smy=3:d=0
+#	dpmouse:smx=3:smy=3:d=0
 #
 # The keypad:
 #	dpkeypad:linear
+	dpkeypad:+x+y:linear
 
 # Sound driver:
 #
@@ -117,7 +118,7 @@
 #
 #	Pplane:classic
 #	Pplane:basic
-#	Pplane:f16
+	Pplane:f16
 
 # Set a color, HUD for example:
 #
