--- Makefile	2011-12-09 23:24:18.648368747 +0100
+++ Makefile_new	2011-12-09 23:25:41.183523559 +0100
@@ -28,7 +28,7 @@
 
 AEOLUS_O =	main.o audio.o model.o slave.o imidi.o addsynth.o scales.o \
 		reverb.o asection.o division.o rankwave.o rngen.o exp2ap.o lfqueue.o
-aeolus:	LDLIBS += -lclalsadrv -lclthreads -lpthread -ljack -lasound -lpthread -lrt
+aeolus:	LDLIBS += -lclalsadrv -lclthreads -lpthread -ljack -lasound -lpthread -lrt -ldl
 aeolus: LDFLAGS += -L$(PREFIX)/$(LIBDIR)
 aeolus:	$(AEOLUS_O)
 	g++ $(LDFLAGS) -o $@ $(AEOLUS_O) $(LDLIBS)
