diff -ruN rox-font/AppRun rox-font_a/AppRun
--- rox-font/AppRun	2012-05-09 00:22:46.000000000 -0400
+++ rox-font_a/AppRun	2012-05-09 00:24:50.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import findrox; findrox.version(1, 9, 13)
 
 import os
diff -ruN rox-font/makedist.sh rox-font_a/makedist.sh
--- rox-font/makedist.sh	2012-05-09 00:22:46.000000000 -0400
+++ rox-font_a/makedist.sh	2012-05-09 00:25:15.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 SCRIPT=`echo "$0" | sed 's|^\./||'`
 echo "$SCRIPT" | grep -q '^/' || SCRIPT=`pwd`/"$SCRIPT"
diff -ruN rox-font/Messages/dist rox-font_a/Messages/dist
--- rox-font/Messages/dist	2012-05-09 00:22:46.000000000 -0400
+++ rox-font_a/Messages/dist	2012-05-09 00:24:12.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 echo Making all .mo files...
 echo
diff -ruN rox-font/Messages/update-po rox-font_a/Messages/update-po
--- rox-font/Messages/update-po	2012-05-09 00:22:46.000000000 -0400
+++ rox-font_a/Messages/update-po	2012-05-09 00:24:31.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 echo Extracting messages from source files...
 echo
