diff -upr linux.orig/arch/arm/Kconfig linux/arch/arm/Kconfig
--- linux.orig/arch/arm/Kconfig	2014-01-04 12:28:30.470000001 +0100
+++ linux/arch/arm/Kconfig	2014-01-03 14:56:58.680000000 +0100
@@ -626,7 +626,7 @@ config ARCH_TEGRA
 	select ARCH_HAS_CPUFREQ
 	select ARCH_PROVIDES_UDELAY
 	select FIQ
-	select PCI
+	select MIGHT_HAVE_PCI
 	help
 	  This enables support for NVIDIA Tegra based systems (Tegra APX,
 	  Tegra 6xx and Tegra 2 series).
