--- a/src/libdecodeqr/Makefile.in	2008-04-04 13:05:25.000000000 +0900
+++ a/src/libdecodeqr/Makefile.in	2008-04-04 13:06:33.000000000 +0900
@@ -31,9 +31,9 @@
 ARFLAGS=rcs
 
 LD=@LD@
-LDFLAGS=@LDFLAGS@ -shared -soname $(SONAME)
+LDFLAGS=@LDFLAGS@ -shared -Wl,-soname,$(SONAME)
 LIBCV=@LIBCV@
 LIBS=$(LIBCV)
 
 INSTALL=@INSTALL@
 LDCONFIG=/sbin/ldconfig
@@ -53,7 +53,7 @@
 	$(AR) $(ARFLAGS) $@ $(objs)
 
 $(LIBNAME_SO): $(objs)
-	$(LD) -o $@ $(objs) $(LDFLAGS) $(LIBS)
+	$(CXX) -o $@ $(objs) $(LDFLAGS) $(LIBS)
 
 install: $(dsts)
 	$(INSTALL) -m 0755 $(LIBNAME_A) $(LIBNAME_SO) $(libdir)
