diff -u a/configure b/configure
--- a/configure	2010-12-31 06:11:33.816672557 +0000
+++ b/configure	2010-12-31 06:14:10.773339223 +0000
@@ -2816,6 +2816,8 @@
   LIBCV=-l$LIBCVNAME
 fi
 
+LIBCV=`pkg-config --libs opencv`
+
 if test -z "$LIBCV"; then
    echo
    echo "If your system already has OpenCV, append place of the libcv.la"
@@ -2920,6 +2922,8 @@
   LIBHIGHGUI=-l$LIBHIGHGUINAME
 fi
 
+LIBHIGHGUI=`pkg-config --libs opencv`
+
    if test -z "$LIBHIGHGUI"; then
       echo
       echo "If your system already has OpenCV, append place of the libhighgui.la"
