*** src/com/asi/gnu/callbacks/JNUCallbackFrame.c.orig	2013-03-06 20:34:01.187989866 +0100
--- src/com/asi/gnu/callbacks/JNUCallbackFrame.c	2013-03-06 19:36:22.919588292 +0100
***************
*** 135,141 ****
  
  
    if (fid == NULL && !error_flag) {
!     string FIELDdtor("L");
      FIELDdtor += qcn;
      FIELDdtor += ";";
      fid = env->GetStaticFieldID(jc, "FIELD", FIELDdtor.c_str());
--- 135,141 ----
  
  
    if (fid == NULL && !error_flag) {
!     std::string FIELDdtor("L");
      FIELDdtor += qcn;
      FIELDdtor += ";";
      fid = env->GetStaticFieldID(jc, "FIELD", FIELDdtor.c_str());
