--- obmenugen-0.4.3-r56/src/config.d.old	2009-12-26 19:07:52.000000000 +0100
+++ obmenugen-0.4.3-r56/src/config.d	2011-12-15 00:07:59.000000000 +0100
@@ -67,7 +67,7 @@ class Config
             ConfigPairs["obmenugen schema file"] = ConfigPairs["obmenugen config dir"] ~ "obmenugen.schema";
             ConfigPairs["obmenugen exclusions file"] = ConfigPairs["obmenugen config dir"] ~ "exclusions";
             
-            ConfigPairs["locales full"] = getenv("LANG");
+            ConfigPairs["locales full"] = getenv("LC_MESSAGES");
             ConfigPairs["locales long"] = std.string.split(ConfigPairs["locales full"], ".")[0];
             ConfigPairs["locales short"] = std.string.split(ConfigPairs["locales long"], "_")[0];
             
