Index: cblrepo/vty-5.2.6/vty.cabal
===================================================================
--- cblrepo.orig/vty-5.2.6/vty.cabal
+++ cblrepo/vty-5.2.6/vty.cabal
@@ -50,7 +50,7 @@ library
                        deepseq >= 1.1 && < 1.4,
                        directory,
                        filepath >= 1.0 && < 2.0,
-                       lens >= 3.9.0.2 && < 4.7,
+                       lens >= 3.9.0.2,
                        -- required for nice installation with yi
                        hashable >= 1.2,
                        mtl >= 1.1.1.0 && < 2.3,
