Index: configure.in
===================================================================
--- configure.in	(revision 1541)
+++ configure.in	(working copy)
@@ -4,12 +4,14 @@
 AC_PREREQ(2.59)
 AC_COPYRIGHT([(c) TicTacTec 2005, 2006])
 AC_INIT([ta-lib], [SVN], [http://sourceforge.net/tracker/?group_id=8903&atid=108903])
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR([src/ta_func/ta_AD.c])
 AC_CONFIG_HEADER([include/ta_config.h])
 AM_INIT_AUTOMAKE([ta-lib], [SVN])
 
 # Checks for programs.
 AC_PROG_CC
+AM_PROG_CC_C_O
 AC_PROG_LIBTOOL
 
 # Checks for header files.
