Only in gd/c: sysdep.h
diff -rc scsh-0.6.7/c/unix/dynamo.c gd/c/unix/dynamo.c
*** scsh-0.6.7/c/unix/dynamo.c	2001-01-22 20:48:00.000000000 +0800
--- gd/c/unix/dynamo.c	2013-12-17 13:56:54.935175605 +0800
***************
*** 4,9 ****
--- 4,10 ----
   */
  
  #include <stdlib.h>
+ #include <string.h>
  #include <unistd.h>
  #include "sysdep.h"
  #include "scheme48.h"
diff -rc scsh-0.6.7/c/unix/socket.c gd/c/unix/socket.c
*** scsh-0.6.7/c/unix/socket.c	2003-06-25 16:19:34.000000000 +0800
--- gd/c/unix/socket.c	2013-12-17 13:55:23.551843919 +0800
***************
*** 5,10 ****
--- 5,11 ----
   * An interface to Unix sockets.
   */
  
+ #include <stdlib.h>
  #include <sys/types.h>
  #include <sys/socket.h>
  #include <sys/param.h>
Only in gd/scsh: endian.scm
Only in gd/scsh: machine
diff -rc scsh-0.6.7/scsh/network1.c gd/scsh/network1.c
*** scsh-0.6.7/scsh/network1.c	2006-03-03 05:46:15.000000000 +0800
--- gd/scsh/network1.c	2013-12-17 13:59:38.385172657 +0800
***************
*** 409,415 ****
  			     s48_value optname)
  {
    int optval;
!   size_t optlen=sizeof(optval);
    
    if (getsockopt(s48_extract_fixnum (s),
  		 s48_extract_fixnum (level),
--- 409,415 ----
  			     s48_value optname)
  {
    int optval;
!   socklen_t optlen=sizeof(optval);
    
    if (getsockopt(s48_extract_fixnum (s),
  		 s48_extract_fixnum (level),
***************
*** 426,432 ****
  				    s48_value optname)
  {
    struct linger optval;
!   size_t optlen=sizeof(optval);
    
    if (getsockopt(s48_extract_fixnum (s),
  		 s48_extract_fixnum (level),
--- 426,432 ----
  				    s48_value optname)
  {
    struct linger optval;
!   socklen_t optlen=sizeof(optval);
    
    if (getsockopt(s48_extract_fixnum (s),
  		 s48_extract_fixnum (level),
***************
*** 444,450 ****
  				     s48_value optname)
  {
    struct timeval optval;
!   size_t optlen=sizeof(optval);
    
    if (getsockopt(s48_extract_fixnum (s),
  		 s48_extract_fixnum (level),
--- 444,450 ----
  				     s48_value optname)
  {
    struct timeval optval;
!   socklen_t optlen=sizeof(optval);
    
    if (getsockopt(s48_extract_fixnum (s),
  		 s48_extract_fixnum (level),
diff -rc scsh-0.6.7/scsh/tty1.c gd/scsh/tty1.c
*** scsh-0.6.7/scsh/tty1.c	2006-05-16 17:27:47.000000000 +0800
--- gd/scsh/tty1.c	2013-12-17 14:35:31.661800492 +0800
***************
*** 21,26 ****
--- 21,27 ----
  #include <fcntl.h>
  #include <signal.h>
  #include "scheme48.h"
+ #define BUFSIZE 1024
  
  #ifdef HAVE_STROPTS_H
  #include <stropts.h>
***************
*** 491,508 ****
  static const char*
  allocate_slave_name(int master_fd, const char* clone){
  
!   char * slave_name;
  
  #if defined (HAVE_PTSNAME)
!   slave_name = ptsname (master_fd);
    if (slave_name)
      return slave_name;
  #endif
! 
    /* kludge for AIX */
    if (clone
        && !strcmp (clone, "/dev/ptc")
!       && (slave_name = ttyname (master_fd)) != NULL)
      return slave_name;
  
    return NULL;
--- 492,511 ----
  static const char*
  allocate_slave_name(int master_fd, const char* clone){
  
!   char *slave_name;
  
  #if defined (HAVE_PTSNAME)
!   /*slave_name = ptsname (master_fd);*/
!   ptsname_r(master_fd, slave_name, BUFSIZE);
    if (slave_name)
      return slave_name;
  #endif
! 		ttyname_r( master_fd, slave_name, BUFSIZE);
    /* kludge for AIX */
    if (clone
        && !strcmp (clone, "/dev/ptc")
! 			&& slave_name != NULL)
!       /*&& (slave_name = ttyname (master_fd)) != NULL)*/
      return slave_name;
  
    return NULL;
Only in gd: scsh-config
