diff -urN oolua-release-1.4.0/file_generator/premake4.lua oolua-release-1.4.0.fix_lua51/file_generator/premake4.lua
--- oolua-release-1.4.0/file_generator/premake4.lua	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/file_generator/premake4.lua	2013-03-04 10:50:40.192385680 -0600
@@ -10,7 +10,7 @@
 }
 includedirs 
 {
-	"include/lua",
+	"include/lua51",
 	"/usr/local/include",
 	"/usr/include",
 } 
diff -urN oolua-release-1.4.0/include/lua_includes.h oolua-release-1.4.0.fix_lua51/include/lua_includes.h
--- oolua-release-1.4.0/include/lua_includes.h	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/include/lua_includes.h	2013-03-04 10:53:58.368479760 -0600
@@ -9,18 +9,18 @@
 extern "C"
 {
 #if defined _MSC_VER
-#	include "lua/lua.h"
-#	include "lua/lauxlib.h"
-#	include "lua/lualib.h"
+#	include "lua.h"
+#	include "lauxlib.h"
+#	include "lualib.h"
 #elif defined __MINGW32__ //you may need to change this
-#	include "lua/lua.h"
-#	include "lua/lauxlib.h"
-#	include "lua/lualib.h"
+#	include "lua.h"
+#	include "lauxlib.h"
+#	include "lualib.h"
 #elif defined __GNUC__
 	
-#	include "lua/lua.h"
-#	include "lua/lauxlib.h"
-#	include "lua/lualib.h"
+#	include "lua.h"
+#	include "lauxlib.h"
+#	include "lualib.h"
 
 /*	
 #	include "lua.h"
diff -urN oolua-release-1.4.0/oolua4.lua oolua-release-1.4.0.fix_lua51/oolua4.lua
--- oolua-release-1.4.0/oolua4.lua	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/oolua4.lua	2013-03-04 10:50:40.192385680 -0600
@@ -8,7 +8,7 @@
 files { "include/*.h", "src/*.cpp" }
 includedirs 
 					{
-						"include/lua/",
+						"include/lua51/",
 						"include/",
 						"/usr/local/include",
 						"/usr/include"
diff -urN oolua-release-1.4.0/profile/premake4.lua oolua-release-1.4.0.fix_lua51/profile/premake4.lua
--- oolua-release-1.4.0/profile/premake4.lua	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/profile/premake4.lua	2013-03-04 10:50:40.192385680 -0600
@@ -17,7 +17,7 @@
 	root .. "profile/"
 	,root .. "include/"
 	,"./"
-	,"include/lua"
+	,"include/lua51"
 --	,"include/boost"
 } 
 
diff -urN oolua-release-1.4.0/profile/swig_profile_wrap.cpp oolua-release-1.4.0.fix_lua51/profile/swig_profile_wrap.cpp
--- oolua-release-1.4.0/profile/swig_profile_wrap.cpp	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/profile/swig_profile_wrap.cpp	2013-03-04 10:53:58.372479762 -0600
@@ -741,8 +741,8 @@
 extern "C" {
 #endif
 
-#include "lua/lua.h"
-#include "lua/lauxlib.h"
+#include "lua.h"
+#include "lauxlib.h"
 #include <stdlib.h>  /* for malloc */
 #include <assert.h>  /* for a few sanity tests */
 
diff -urN oolua-release-1.4.0/unit_tests/string_is_integral/premake4.lua oolua-release-1.4.0.fix_lua51/unit_tests/string_is_integral/premake4.lua
--- oolua-release-1.4.0/unit_tests/string_is_integral/premake4.lua	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/unit_tests/string_is_integral/premake4.lua	2013-03-04 10:50:40.192385680 -0600
@@ -19,7 +19,7 @@
 {
 	"include/cppunit",
 	"include/gmock",
-	"include/lua",
+	"include/lua51",
 	"include/",
 	root .. "include/",
 	"/usr/local/include",
diff -urN oolua-release-1.4.0/unit_tests/test_classes/error_check.cpp oolua-release-1.4.0.fix_lua51/unit_tests/test_classes/error_check.cpp
--- oolua-release-1.4.0/unit_tests/test_classes/error_check.cpp	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/unit_tests/test_classes/error_check.cpp	2013-03-04 10:53:58.372479762 -0600
@@ -12,7 +12,7 @@
 	 even when the path is pointing to luajit header directory.
 	 If we include the luajit header directly it will fail to compile for Rio Lua
 	 */
-//#	include "lua/lua.hpp"
+//#	include "lua.hpp"
 #endif
 
 namespace  
diff -urN oolua-release-1.4.0/unit_tests/test.coverage.lua oolua-release-1.4.0.fix_lua51/unit_tests/test.coverage.lua
--- oolua-release-1.4.0/unit_tests/test.coverage.lua	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/unit_tests/test.coverage.lua	2013-03-04 10:50:40.192385680 -0600
@@ -22,7 +22,7 @@
 	{
 		"include/cppunit",
 		"include/gmock",
-		"include/lua",
+		"include/lua51",
 		"include/",
 		"/usr/local/include",
 		"/usr/include",
diff -urN oolua-release-1.4.0/unit_tests/tests_may_fail/premake4.lua oolua-release-1.4.0.fix_lua51/unit_tests/tests_may_fail/premake4.lua
--- oolua-release-1.4.0/unit_tests/tests_may_fail/premake4.lua	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/unit_tests/tests_may_fail/premake4.lua	2013-03-04 10:50:40.192385680 -0600
@@ -16,7 +16,7 @@
 includedirs 
 {
 	"include/cppunit",
-	"include/lua",
+	"include/lua51",
 	"include/",
 	root .. "include/",
 	"/usr/local/include",
diff -urN oolua-release-1.4.0/unit_tests/test.unit.exceptions.lua oolua-release-1.4.0.fix_lua51/unit_tests/test.unit.exceptions.lua
--- oolua-release-1.4.0/unit_tests/test.unit.exceptions.lua	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/unit_tests/test.unit.exceptions.lua	2013-03-04 10:51:30.572410392 -0600
@@ -23,7 +23,7 @@
 {
 	"include/cppunit",
 	"include/gmock",
-	"include/lua",
+	"include/lua51",
 	"include/",
 	root .. "include/",
 	"/usr/local/include",
diff -urN oolua-release-1.4.0/unit_tests/test.unit.lua oolua-release-1.4.0.fix_lua51/unit_tests/test.unit.lua
--- oolua-release-1.4.0/unit_tests/test.unit.lua	2013-02-18 19:15:20.000000000 -0600
+++ oolua-release-1.4.0.fix_lua51/unit_tests/test.unit.lua	2013-03-04 10:51:30.572410392 -0600
@@ -20,7 +20,7 @@
 {
 	"include/cppunit",
 	"include/gmock",
-	"include/lua",
+	"include/lua51",
 	"include/",
 	root .. "include/",
 	"/usr/local/include",
