--- a/JAUS++-5.130429-src/src/cxutils/Packet.cpp	2013-04-23 12:14:18.848492500 -0400
+++ b/JAUS++-5.130429-src/src/cxutils/Packet.cpp	2013-05-31 13:45:20.776509813 -0400
@@ -46,7 +46,7 @@
 #include "cxutils/Packet.h"
 #include <stdlib.h>
 #include <string>
-#include <string.h>
+#include <cstring>
 #include <assert.h>
 #include <iostream>
 #include <iomanip>
--- a/JAUS++-5.130429-src/src/cxutils/Packet.h	2013-04-23 12:14:19.830365900 -0400
+++ b/JAUS++-5.130429-src/src/cxutils/Packet.h	2013-05-31 13:48:17.323087531 -0400
@@ -48,6 +48,7 @@
 
 #include <limits.h>
 #include <string>
+#include <cstring>
 #include <vector>
 #include <list>
 #include "cxutils/CxBase.h"
