Index: lib/Frontend/InitHeaderSearch.cpp
===================================================================
--- lib/Frontend/InitHeaderSearch.cpp	(revision 126310)
+++ lib/Frontend/InitHeaderSearch.cpp	(working copy)
@@ -732,10 +732,10 @@
     AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5",
                                 "x86_64-suse-linux", "", "", triple);
 
-    // Arch Linux 2008-06-24
-    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.3.1",
+    // Arch Linux 2010-12-19
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5.2",
                                 "i686-pc-linux-gnu", "", "", triple);
-    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.3.1",
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5.2",
                                 "x86_64-unknown-linux-gnu", "", "", triple);
     // Gentoo x86 2010.0 stable
     AddGnuCPlusPlusIncludePaths(
