Index: lib/Frontend/InitHeaderSearch.cpp
===================================================================
--- lib/Frontend/InitHeaderSearch.cpp	(revision 122175)
+++ lib/Frontend/InitHeaderSearch.cpp	(working copy)
@@ -714,10 +714,10 @@
     AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5",
                                 "x86_64-suse-linux", "", "", triple);
 
-    // Arch Linux 2008-06-24
-    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.3.1",
+    // Arch Linux 2010-12-19
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5.2",
                                 "i686-pc-linux-gnu", "", "", triple);
-    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.3.1",
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5.2",
                                 "x86_64-unknown-linux-gnu", "", "", triple);
     // Gentoo x86 2010.0 stable
     AddGnuCPlusPlusIncludePaths(
Index: lib/Driver/ToolChains.cpp
===================================================================
--- lib/Driver/ToolChains.cpp	(revision 122175)
+++ lib/Driver/ToolChains.cpp	(working copy)
@@ -1344,8 +1344,8 @@
       GccTriple = "i586-suse-linux";
   }
 
-  const char* GccVersions[] = {"4.5.1", "4.5", "4.4.5", "4.4.4", "4.4.3", "4.4",
-                               "4.3.4", "4.3.3", "4.3.2"};
+  const char* GccVersions[] = {"4.5.2", "4.5.1", "4.5", "4.4.5", "4.4.4",
+                               "4.4.3", "4.4", "4.3.4", "4.3.3", "4.3.2"};
   std::string Base = "";
   for (unsigned i = 0; i < sizeof(GccVersions)/sizeof(char*); ++i) {
     std::string Suffix = GccTriple + "/" + GccVersions[i];
