--- cmake/Modules/FindJSONC.cmake 2014-05-28 20:01:28.176224670 -0400
+++ cmake/Modules/FindJSONC.cmake	2014-05-28 20:01:34.363067146 -0400
@@ -6,7 +6,7 @@
 #  JSONC_INCLUDE_DIR - The JSON-C include directory.
 #  JSONC_LIBRARY     - The JSON-C library to link against.

-FIND_PATH(JSONC_INCLUDE_DIR json/json.h)
+FIND_PATH(JSONC_INCLUDE_DIR json-c/json.h)
 FIND_LIBRARY(JSONC_LIBRARY NAMES json-c json)

 IF (JSONC_INCLUDE_DIR AND JSONC_LIBRARY)
diff --git a/libgrive/src/protocol/Json.cc b/libgrive/src/protocol/Json.cc
index 4c115d6..a7f0373 100644
--- libgrive/src/protocol/Json.cc
+++ libgrive/src/protocol/Json.cc
@@ -29,8 +29,8 @@
        #pragma warning(push)
        #pragma warning(disable: 4005)
 #endif
-#include <json/json_tokener.h>
-#include <json/linkhash.h>
+#include <json-c/json_tokener.h>
+#include <json-c/linkhash.h>
 #ifdef _MSC_VER
        #pragma warning(pop)
 #endif
