diff -ur lineak-kdeplugins-0.9/kmix_plugin/kmixclient.h lineak-kdeplugins-0.9-mod/kmix_plugin/kmixclient.h
--- lineak-kdeplugins-0.9/kmix_plugin/kmixclient.h	2003-12-09 22:46:45.000000000 +0100
+++ lineak-kdeplugins-0.9-mod/kmix_plugin/kmixclient.h	2008-06-22 15:02:36.000000000 +0200
@@ -17,6 +17,7 @@
 
 #ifndef KMIXCLIENT_H
 #define KMIXCLIENT_H
+#include <stdlib.h>
 #include <string>
 #include <dcopclient.h>
 
