--- pyami/fft/fftwsetup.py	2014-06-09 12:19:23.374240418 -0700
+++ pyami/fft/fftwsetup.py	2014-06-09 19:56:45.197775389 -0700
@@ -65,8 +65,10 @@ def run_timing():
 		plan = make_plan(a, rigor)
 		init(a, plan)
 		run(plan)
-
-	pyami.fft.calc_fftw3.store_wisdom()
+	if len(sys.argv) > 4:
+		pyami.fft.calc_fftw3.store_system_wisdom()
+	else:
+		pyami.fft.calc_fftw3.store_wisdom()
 
 	for key in ['create','plan','init','run']:
 		print key, timing[key]

--- pyami/fft/calc_fftw3.py	2014-06-09 19:53:35.945881480 -0700
+++ pyami/fft/calc_fftw3.py	2014-06-09 19:53:48.725334167 -0700
@@ -35,7 +35,7 @@ global_plan_kwargs = {
 
 ## where to look for user customized wisdom
 home = os.path.expanduser('~')
-wisdom_file = 'fftw3-wisdom-' + platform.node()
+wisdom_file = '.fftw3-wisdom-' + platform.node()
 wisdom_file = os.path.join(home, wisdom_file)
 
 def load_wisdom():
@@ -57,6 +57,9 @@ def store_wisdom():
 			os.remove(wisdom_file)
 		fftw3.export_wisdom_to_file(wisdom_file)
 
+def store_system_wisdom():
+		fftw3.export_wisdom_to_file("/etc/fftw/wisdom")
+
 class FFTW3Calculator(calc_base.Calculator):
 	def __init__(self):
 		calc_base.Calculator.__init__(self)

--- pyami/fileutil.py	2013-12-14 17:53:13.000000000 +0100
+++ pyami/fileutil.py	2014-01-06 12:08:50.244555780 +0100
@@ -51,6 +51,12 @@
 	module.  If you want to force a certain module, pass it to this
 	function in the optional argument.
 	'''
+	# add portable dir
+	if sys.platform == 'win32':
+		portable_dir = os.path.join(os.path.dirname(os.path.split(sys.executable)[0]), "etc", "myami")
+	else:
+		portable_dir = os.path.expanduser("~/.leginon")
+
 	# system config location is /etc/myami on unix like systems or
 	# under PROGRAMFILES on windows
 	if sys.platform == 'win32':
@@ -68,7 +74,7 @@
 	# user home dir
 	user_dir = os.path.expanduser('~')
 
-	return [system_dir, installed_dir, user_dir]
+	return [system_dir, installed_dir, user_dir, portable_dir]
 
 def open_if_not_exists(filename):
 	'''Creates a new file for read/write access.  Raises exception if file exists'''
