diff -aur astropy-0.4.4/astropy/wcs/src/wcslib_wrap.c astropy-0.4.4.patched/astropy/wcs/src/wcslib_wrap.c
--- astropy-0.4.4/astropy/wcs/src/wcslib_wrap.c	2015-01-21 20:20:33.000000000 +0000
+++ astropy-0.4.4.patched/astropy/wcs/src/wcslib_wrap.c	2015-01-27 21:00:44.890710318 +0000
@@ -751,7 +751,7 @@
 
   wcsprm_python2c(&self->x);
   wcsprm_python2c(&other->x);
-  status = wcscompare(cmp, &self->x, &other->x, &equal);
+  status = wcscompare(cmp, 0.0, &self->x, &other->x, &equal);
   wcsprm_c2python(&self->x);
   wcsprm_c2python(&other->x);
 
@@ -1697,7 +1697,7 @@
     wcsprm_python2c(ax);
     wcsprm_python2c(bx);
     status = wcscompare(
-        WCSCOMPARE_ANCILLARY,
+        WCSCOMPARE_ANCILLARY, 0.0,
         ax, bx, &equal);
     wcsprm_c2python(ax);
     wcsprm_c2python(bx);
