--- a/usr/bin/eee-control-daemon	2010-05-23 23:55:33.000000000 +0200
+++ b/usr/bin/eee-control-daemon	2010-05-27 16:27:31.998495261 +0200
@@ -84,6 +84,8 @@ if __name__ == "__main__":
         brn_path = os.path.join(acpi_base, "brn")
     elif os.path.exists("/sys/class/backlight/eeepc/brightness"):
         brn_path = "/sys/class/backlight/eeepc/brightness"
+    elif os.path.exists("/sys/class/backlight/acpi_video0/brightness"):
+        brn_path = "/sys/class/backlight/acpi_video0/brightness"
     else:
         brn_path = "/dev/null"
 
