diff -aur a/utils/gdm3setup-daemon.in b/utils/gdm3setup-daemon.in
--- a/utils/gdm3setup-daemon.in	2013-09-28 22:42:18.000000000 +0200
+++ b/utils/gdm3setup-daemon.in	2013-09-28 23:24:06.721234684 +0200
@@ -118,16 +118,10 @@
 					sender_keyword='sender', connection_keyword='connexion')
 	def SetUI(self,name,value,sender=None, connexion=None):
 		if self.policykit_test(sender,connexion,'apps.nano77.gdm3setup.set') :
-			if name!='SHELL_THEME':
-				if name == "WALLPAPER" :
-					value = "file://"+value
-				bus_adrress , bus_pid = Get_Bus()
-				subprocess.call('su - %GDM_USER% -s /bin/bash -c "LANG='+LANG+' DBUS_SESSION_BUS_ADDRESS='+bus_adrress+' DBUS_SESSION_BUS_PID='+bus_pid+' set_gdm -n '+name+' -v '+"'"+value+"'"+'"',shell=True)
-			else :
-				if value == "" or value == "Adwaita" :
-					subprocess.call("gdmshelltheme -d",shell=True)
-				else :
-					subprocess.call("gdmshelltheme -n "+value,shell=True)
+			if name == "WALLPAPER" :
+				value = "file://"+value
+			bus_adrress , bus_pid = Get_Bus()
+			subprocess.call('su - %GDM_USER% -s /bin/bash -c "LANG='+LANG+' DBUS_SESSION_BUS_ADDRESS='+bus_adrress+' DBUS_SESSION_BUS_PID='+bus_pid+' set_gdm -n '+name+' -v '+"'"+value+"'"+'"',shell=True)
 			return "OK"
 		else :
 			return "ERROR : YOU ARE NOT ALLOWED !"
@@ -141,16 +135,13 @@
 		os.remove("/tmp/GET_GDM")
 		p = subprocess.Popen("gdmshelltheme -g ",stdout=subprocess.PIPE,shell=True)
 		shelltheme = p.stdout.read().strip()
-		settings.append("SHELL='"+shelltheme+"'\n")
 		return settings
 
 	@dbus.service.method('apps.nano77.gdm3setup','s','b',sender_keyword='sender', connection_keyword='connexion')
 	def SetShellTheme(self,SHELL_THEME,sender=None, connexion=None):
 		if self.policykit_test(sender,connexion,'apps.nano77.gdm3setup.set') :
-			if SHELL_THEME == "" or SHELL_THEME == "Adwaita" :
-				subprocess.call("gdmshelltheme -d",shell=True)
-			else :
-				subprocess.call("gdmshelltheme -n "+SHELL_THEME,shell=True)
+			bus_adrress , bus_pid = Get_Bus()
+			subprocess.call('su - %GDM_USER% -s /bin/bash -c "LANG='+LANG+' DBUS_SESSION_BUS_ADDRESS='+bus_adrress+' DBUS_SESSION_BUS_PID='+bus_pid+' set_gdm -n SHELL_THEME -v '+"'"+SHELL_THEME+"'"+'"',shell=True)
 			return True
 		else :
 			return False
diff -aur a/utils/get_gdm.in b/utils/get_gdm.in
--- a/utils/get_gdm.in	2013-02-07 10:34:18.000000000 +0100
+++ b/utils/get_gdm.in	2013-02-09 22:13:23.908160239 +0100
@@ -9,6 +9,8 @@
 
 echo -n "GTK=" > /tmp/GET_GDM
 gsettings get org.gnome.desktop.interface gtk-theme >> /tmp/GET_GDM
+echo -n "SHELL=" >> /tmp/GET_GDM
+gsettings get org.gnome.shell theme-name >> /tmp/GET_GDM
 echo -n "ICON=" >> /tmp/GET_GDM
 gsettings get org.gnome.desktop.interface icon-theme >> /tmp/GET_GDM
 echo -n "FONT=" >> /tmp/GET_GDM
diff -aur a/utils/set_gdm b/utils/set_gdm
--- a/utils/set_gdm	2013-02-07 10:34:18.000000000 +0100
+++ b/utils/set_gdm	2013-02-09 22:14:24.881816065 +0100
@@ -63,6 +63,10 @@
 				echo "$parameter_name = $parameter_value"
 				gsettings set org.gnome.desktop.interface gtk-theme "$parameter_value"
 				;;
+			SHELL_THEME)
+				echo "$parameter_name = $parameter_value"
+				gsettings set org.gnome.shell theme-name "$parameter_value"
+				;;
 			FONT)
 				echo "$parameter_name = $parameter_value"
 				gsettings set org.gnome.desktop.interface font-name "$parameter_value"
