commit da22691733bfaaf6a1a953d3e9b53e1ab2dd854a
Author: Eugene Petrov <dhamp@ya.ru>
Date:   Wed Apr 11 18:25:09 2012 +0400

    QT UI: fix build with gcc 4.7.x

diff --git a/eiskaltdcpp-qt/src/PublicHubs.cpp b/eiskaltdcpp-qt/src/PublicHubs.cpp
index 9b33a5c..523d7d4 100644
--- a/eiskaltdcpp-qt/src/PublicHubs.cpp
+++ b/eiskaltdcpp-qt/src/PublicHubs.cpp
@@ -146,7 +146,7 @@ void PublicHubs::slotContextMenu(){
         return;
 
     if (proxy)
-        std::transform(indexes.begin(), indexes.end(), indexes.begin(), [&proxy](QModelIndex i) { return proxy->mapToSource(i); });
+        std::transform(indexes.begin(), indexes.end(), indexes.begin(), [&](QModelIndex i) { return proxy->mapToSource(i); });
 
     WulforUtil *WU = WulforUtil::getInstance();
 
