diff -ru xmlm-1.0.2_ori/build xmlm-1.0.2/build
--- xmlm-1.0.2_ori/build	2009-11-11 14:08:24.000000000 +0100
+++ xmlm-1.0.2/build	2010-08-31 22:43:32.963333559 +0200
@@ -8,7 +8,8 @@
 SHORTDESCRIPTION="Streaming XML input\/output for OCaml"
 EMAIL="daniel.buenzl i\\\@erratique.ch"
 
-INSTALLDIR=${INSTALLDIR:="`ocamlc -where`/$PROJECTNAME"}
+#INSTALLDIR=${INSTALLDIR:="`ocamlc -where`/$PROJECTNAME"}
+INSTALLDIR=$INSTALLDIR
 DOCDIR=${DOCDIR:="doc"}
 F=${F:="_build/src/$PROJECTNAME"}
 
@@ -20,7 +21,7 @@
 # Misc tools
 CP=${CP:="cp"}
 RM=${RM:="rm"}
-MKDIR=${MKDIR:="mkdir"}
+#MKDIR=${MKDIR:="mkdir"}
 
 ocb () { $OCAMLBUILD $BUILDFLAGS $* ; }
 
@@ -28,26 +29,28 @@
 {
   case $1 in
     module-byte)
-      ocb $PROJECTNAME.cmo ;;
+      ocb $PROJECTNAME.cma
+      ocb $PROJECTNAME.cmi ;;
     module-native) 
-      ocb $PROJECTNAME.cmx ;;
+      ocb $PROJECTNAME.cmx
+      ocb $PROJECTNAME.cmxa ;;
     module-plugin) 
       ocb $PROJECTNAME.cmxs ;;
     module)
       action module-byte
       action module-native ;;
     doc)
-      ocb $PROJECTNAME.cmi
+#      ocb $PROJECTNAME.cmi
       $OCAMLDOC -html -I src -colorize-code -d $DOCDIR src/$PROJECTNAME.mli ;;
     distrib) distrib ;;
     install-byte)
       action module-byte
-      $MKDIR -p $INSTALLDIR
-      $CP $F.ml $F.mli $F.cmi $F.cmo src/META $INSTALLDIR ;;
+#      $MKDIR -p $INSTALLDIR
+      $CP $F.mli $F.cmi $F.cma src/META $INSTALLDIR ;;
     install)
       action install-byte
       action module-native
-      $CP $F.cmx $F.o $INSTALLDIR ;;
+      $CP $F.cmx $F.cmxa $F.a $INSTALLDIR ;;
     install-plugin)
       action install
       action module-plugin
diff -ru xmlm-1.0.2_ori/src/META xmlm-1.0.2/src/META
--- xmlm-1.0.2_ori/src/META	2009-11-11 14:08:24.000000000 +0100
+++ xmlm-1.0.2/src/META	2010-08-31 22:42:40.243333558 +0200
@@ -1,6 +1,5 @@
 version = "1.0.2"
 description = "Streaming XML input/output for OCaml"
-archive(byte) = "xmlm.cmo"
-archive(native) = "xmlm.cmx"
+archive(byte) = "xmlm.cma"
+archive(native) = "xmlm.cmxa"
 archive(plugin,native) = "xmlm.cmxs"
-directory = "+xmlm"
