--- a/libMRML/cc/CAttributeList.cc	2002-07-26 13:23:23.000000000 +0200
+++ b/libMRML/cc/CAttributeList.cc	2004-12-22 19:42:18.000000000 +0100
@@ -94,7 +94,7 @@
       
     if(strcmp(i->second.first,
 		   i->second.second)){
-      cout << "DELETING FAILED" << flush << int(i->second.second) << flush << i->second.first << i->second.second << endl;
+      cout << "DELETING FAILED" << flush << long(i->second.second) << flush << i->second.first << i->second.second << endl;
       assert(0);
     }
     delete[] i->second.first;

