--- a/scripts/perl/Makefile.am	2013-03-20 16:18:08.416315573 +0100
+++ b/scripts/perl/Makefile.am	2013-03-20 16:18:14.999613241 +0100
@@ -146,7 +146,7 @@
 	$(PERLCONFIG)
 
 
-bin_SCRIPTS= @GIFT_INTERNAL_PERLS@ gift-url-to-fts.pl gift-diagnose-print-all-ADI.pl gift-mrml-client.pl CFeedbackClient.pm CFCMovingTarget.pm gift-perl-example-server.pl gift-add-collection.pl gift-remove-collection.pl gift-dtd-to-tex.pl gift-dtd-to-keywords.pl CQueryParadigmMatcher.pm CTripleTransfer.pl CGLDAudreyMTam.pl CGLStructuredAnnotation.pm CGIFTLink.pm CGLDistanceMatrix.pm CGIFTLinkDemo.pl CXMLTreeBuilder.pm CXMLTreeWriter.pm CXMLTreeVisitor.pm CXTVTripletList.pm CXTVWriter.pm CGraphCollection.pm gift-old-to-new-url2fts.pl gift-start.pl gift-diagnose-print-all-ADI.pl
+bin_SCRIPTS= @GIFT_INTERNAL_PERLS@ gift-url-to-fts.pl gift-diagnose-print-all-ADI.pl gift-mrml-client.pl CFeedbackClient.pm CFCMovingTarget.pm gift-perl-example-server.pl gift-add-collection.pl gift-remove-collection.pl gift-dtd-to-tex.pl gift-dtd-to-keywords.pl CQueryParadigmMatcher.pm CTripleTransfer.pl CGLDAudreyMTam.pl CGLStructuredAnnotation.pm CGIFTLink.pm CGLDistanceMatrix.pm CGIFTLinkDemo.pl CXMLTreeBuilder.pm CXMLTreeWriter.pm CXMLTreeVisitor.pm CXTVTripletList.pm CXTVWriter.pm CGraphCollection.pm gift-old-to-new-url2fts.pl gift-start.pl
 
 EXTRA_SCRIPTS=gift-browser-benchmark.pl gift-extract-text-features.pl gift-annotation-to-html.pl gift-ranking-comparison.pl CFCRankingComparison.pm CGraphCollection.pm
 
--- a/libMRML/include/Makefile.am	2013-03-20 16:22:22.834937269 +0100
+++ b/libMRML/include/Makefile.am	2013-03-20 16:22:59.238073389 +0100
@@ -21,7 +21,7 @@
 
 libMRMLdir=$(includedir)/libMRML/include
 
-libMRML_HEADERS= directory.h \
+libMRML_HEADERS=  \
 	CAFPlugin.h		     CMRMLStringConstants.h   CTimeStampGenerator.h \
 	CAccessor.h		     CMagic.h		      CXEVCommunication.h \
 	CAccessorAdmin.h	     CMutex.h		      CXEVTripletList.h \
