--- ../../orig/anytun-0.3.1/src/configure	2009-10-29 01:29:31.000000000 +0100
+++ configure	2009-10-30 23:54:00.427183812 +0100
@@ -34,7 +34,7 @@
 TARGET=`uname -s`
 
 CXXFLAGS='-g -Wall -O2 -DLOG_SYSLOG -DLOG_FILE -DLOG_STDOUT'
-LDFLAGS='-g -Wall -O2 -lboost_thread -lboost_serialization -lboost_system -lboost_date_time'
+LDFLAGS='-g -Wall -O2 -lboost_thread-mt -lboost_serialization -lboost_system -lboost_date_time'
 
 CRYPTO_LIB='gcrypt'
 PASSPHRASE=1
--- ../../orig/anytun-0.3.1/src/anyrtpproxy/Makefile	2009-10-29 01:29:31.000000000 +0100
+++ anyrtpproxy/Makefile	2009-10-30 23:54:07.031174716 +0100
@@ -34,7 +34,7 @@
 CXX = g++
 CXXFLAGS = -g -Wall
 LD = g++
-LDFLAGS = -g -Wall -O2 -lboost_thread -lboost_serialization -lboost_system
+LDFLAGS = -g -Wall -O2 -lboost_thread-mt -lboost_serialization -lboost_system
 
 OBJS = anyrtpproxy.o \
        ../signalController.o \
