--- Scripting.py	2013-03-07 01:09:29.668452488 +0900
+++ Scripting.py.new	2013-03-07 01:13:20.820757681 +0900
@@ -6,6 +6,7 @@
 
 import os, shutil, traceback, datetime, inspect, errno, sys, stat
 from waflib import Utils, Configure, Logs, Options, ConfigSet, Context, Errors, Build, Node
+from waflib.Context import WAFVERSION
 
 build_dir_override = None
 
@@ -26,8 +27,8 @@
 	Logs.init_log()
 
 	if Context.WAFVERSION != version:
-		Logs.error('Waf script %r and library %r do not match (directory %r)' % (version, WAFVERSION, wafdir))
-		sys.exit(1)
+		Logs.warn('Waf script %r and library %r do not match (directory %r)' % (version, WAFVERSION, wafdir))
+		#sys.exit(1)
 
 	if '--version' in sys.argv:
 		opt_obj = Options.OptionsContext()
