--- src/ja/centre/util/io/nio/MappedByteBufferWrapper.java.old	2006-12-10 23:32:25.000000000 +0200
+++ src/ja/centre/util/io/nio/MappedByteBufferWrapper.java	2008-01-16 13:42:15.000000000 +0200
@@ -59,9 +59,8 @@
         this.fileName = fileName;
 
         try {
-            raf = new RandomAccessFile( fileName, "rw" );
-
             try {
+                raf = new RandomAccessFile( fileName, "rw" );
                 initBuffer();
             } catch ( IOException e ) {
                 LOG.info( "Could not lock file \"" + fileName + "\". Switching to non-locking mode..." );
