--- src/ja/lingo/application/util/plaf/JaLingoLookAndFeel.java.old	2006-10-30 16:05:26.000000000 +0200
+++ src/ja/lingo/application/util/plaf/JaLingoLookAndFeel.java	2008-01-16 14:32:57.000000000 +0200
@@ -36,13 +36,14 @@
         // children dynamic re-layout on resize
         Toolkit.getDefaultToolkit().setDynamicLayout( true );
 
-        // kunstoff
+        String slaf = System.getProperty("ja.lingo.laf");
+        JaLingoLookAndFeel laf = null;
 
-        JaLingoLookAndFeel laf = new JaLingoLookAndFeel();
-        setCurrentTheme( new RainyTheme( fontSize ) );
-        setCurrentGradientTheme( new RainyGradientTheme() );
-
-        //LookAndFeel laf = new WindowsLookAndFeel();
+        if( slaf == null) { // kunstoff
+            laf = new JaLingoLookAndFeel();
+            setCurrentTheme( new RainyTheme( fontSize ) );
+            setCurrentGradientTheme( new RainyGradientTheme() );
+        }
 
         // children LF
         //Plastic3DLookAndFeel laf = new Plastic3DLookAndFeel();
@@ -50,7 +51,10 @@
 
         try {
             //UIManager.setLookAndFeel( UIManager.getSystemLookAndFeelClassName() );
-            UIManager.setLookAndFeel( laf );
+            if( laf != null )
+                UIManager.setLookAndFeel( laf );
+            else
+                UIManager.setLookAndFeel( slaf );
             //Wrapper.wrap();
         } catch ( Exception e ) {
             LOG.warn( "Could not initialize LF", e );
