=== modified file 'src/applet-device-wifi.c'
---
 src/applet-device-wifi.c |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

Index: network-manager-applet-0.8~a~git.20090830t224938.0334788/src/applet-device-wifi.c
===================================================================
--- network-manager-applet-0.8~a~git.20090830t224938.0334788.orig/src/applet-device-wifi.c	2009-08-31 19:30:00.000000000 -0400
+++ network-manager-applet-0.8~a~git.20090830t224938.0334788/src/applet-device-wifi.c	2009-08-31 19:32:23.000000000 -0400
@@ -1164,7 +1164,6 @@
 {
 	NMSettingsConnectionInterface *connection;
 	NMAccessPoint *new = NULL;
-	char *msg;
 	char *esc_ssid = NULL;
 
 	new = update_active_ap (device, new_state, applet);
@@ -1187,12 +1186,11 @@
 			nm_settings_connection_interface_update (connection, bssid_update_cb, NULL);
 	}
 
-	msg = g_strdup_printf (_("You are now connected to the wireless network '%s'."),
-	                       esc_ssid ? esc_ssid : _("(none)"));
-	applet_do_notify_with_pref (applet, _("Connection Established"),
-	                            msg, "nm-device-wireless",
+	applet_do_notify_with_pref (applet,
+	                            esc_ssid ? esc_ssid : _("(none)"),
+	                            _("Connection Established"),
+                                   "nm-device-wireless",
 	                            PREF_DISABLE_CONNECTED_NOTIFICATIONS);
-	g_free (msg);
 	g_free (esc_ssid);
 }
 
