=== modified file 'src/applet-device-gsm.c'
--- a/src/applet-device-gsm.c	2009-02-17 16:32:22 +0000
+++ b/src/applet-device-gsm.c	2009-02-17 16:36:47 +0000
@@ -287,24 +287,19 @@
 	if (new_state == NM_DEVICE_STATE_ACTIVATED) {
 		NMConnection *connection;
 		NMSettingConnection *s_con = NULL;
-		char *str = NULL;
+		const char *str = NULL;
 
 		connection = applet_find_active_connection_for_device (device, applet, NULL);
 		if (connection) {
-			const char *id;
-
 			s_con = NM_SETTING_CONNECTION (nm_connection_get_setting (connection, NM_TYPE_SETTING_CONNECTION));
-			id = s_con ? nm_setting_connection_get_id (s_con) : NULL;
-			if (id)
-				str = g_strdup_printf (_("You are now connected to '%s'."), id);
+			str = s_con ? nm_setting_connection_get_id (s_con) : NULL;
 		}
 
 		applet_do_notify_with_pref (applet,
-		                            _("Connection Established"),
-		                            str ? str : _("You are now connected to the GSM network."),
+		                            str ? str : _("GSM network."),
+		                            _("Connection Established"),
 		                            "nm-device-wwan",
 		                            PREF_DISABLE_CONNECTED_NOTIFICATIONS);
-		g_free (str);
 	}
 }
 

