Index: network-manager-applet-0.7.1~rc4.1/src/applet.c
===================================================================
--- network-manager-applet-0.7.1~rc4.1.orig/src/applet.c
+++ network-manager-applet-0.7.1~rc4.1/src/applet.c
@@ -1974,7 +1974,7 @@
 		applet->icon_layers[layer] = g_object_ref (pixbuf);
 
 	if (!applet->icon_layers[0]) {
-		nma_icon_check_and_load ("nm-no-connection", &applet->no_connection_icon, applet);
+		nma_icon_check_and_load ("notification-network-disconnected", &applet->no_connection_icon, applet);
 		pixbuf = g_object_ref (applet->no_connection_icon);
 	} else {
 		pixbuf = gdk_pixbuf_copy (applet->icon_layers[0]);
@@ -2060,7 +2060,7 @@
 				info->icon = "notification-gsm-disconnected";
 				break;
 			default:
-				info->icon = "nm-no-connection";
+				info->icon = "notification-network-disconnected";
 				break;
 		}
 		g_debug("going for offline with icon: %s", info->icon);
@@ -2560,11 +2560,11 @@
 	switch (state) {
 	case NM_STATE_UNKNOWN:
 	case NM_STATE_ASLEEP:
-		pixbuf = nma_icon_check_and_load ("nm-no-connection", &applet->no_connection_icon, applet);
+		pixbuf = nma_icon_check_and_load ("notification-network-disconnected", &applet->no_connection_icon, applet);
 		dev_tip = g_strdup (_("Networking disabled"));
 		break;
 	case NM_STATE_DISCONNECTED:
-		pixbuf = nma_icon_check_and_load ("nm-no-connection", &applet->no_connection_icon, applet);
+		pixbuf = nma_icon_check_and_load ("notification-network-disconnected", &applet->no_connection_icon, applet);
 		dev_tip = g_strdup (_("No network connection"));
 		break;
 	default:
