Index: network-manager-applet-0.8/src/applet.c
===================================================================
--- network-manager-applet-0.8.orig/src/applet.c	2010-03-09 14:58:49.000000000 +0100
+++ network-manager-applet-0.8/src/applet.c	2010-03-09 15:02:59.000000000 +0100
@@ -615,17 +615,6 @@
 	return item;
 }
 
-static void
-applet_clear_notify (NMApplet *applet)
-{
-	if (applet->notification == NULL)
-		return;
-
-	notify_notification_close (applet->notification, NULL);
-	g_object_unref (applet->notification);
-	applet->notification = NULL;
-}
-
 static gboolean
 applet_notify_server_has_actions ()
 {
@@ -675,15 +664,23 @@
 		return;
 #endif
 
-	applet_clear_notify (applet);
-
 	escaped = utils_escape_notify_message (message);
-	notify = notify_notification_new (summary,
-	                                  escaped,
-	                                  icon ? icon : GTK_STOCK_NETWORK,
-	                                  NULL);
-	g_free (escaped);
-	applet->notification = notify;
+	if (applet->notification == NULL) {
+		notify = notify_notification_new (summary,
+		                                  escaped,
+		                                  icon ? icon : GTK_STOCK_NETWORK,
+		                                  NULL);
+
+		applet->notification = notify;
+	} else {
+		notify = applet->notification;
+		notify_notification_update (notify,
+		                            summary,
+		                            escaped,
+		                            icon ? icon : GTK_STOCK_NETWORK);
+	}
+
+	g_free (escaped);
 
 	notify_notification_attach_to_status_icon (notify, applet->status_icon);
 	notify_notification_set_urgency (notify, urgency);
@@ -2973,7 +2973,6 @@
 	/* Have clicking on the applet act also as acknowledgement
 	 * of the notification. 
 	 */
-	applet_clear_notify (applet);
 
 	nma_menu_clear (applet);
 	gtk_menu_popup (GTK_MENU (applet->menu), NULL, NULL,
@@ -2990,7 +2989,6 @@
 	/* Have clicking on the applet act also as acknowledgement
 	 * of the notification. 
 	 */
-	applet_clear_notify (applet);
 
 	nma_context_menu_update (applet);
 	gtk_menu_popup (GTK_MENU (applet->context_menu), NULL, NULL,
