diff -Nru tpad.orig/bin/tpad tpad/bin/tpad
--- tpad.orig/bin/tpad	2008-11-19 14:43:56.000000000 +0100
+++ tpad/bin/tpad	2008-11-19 14:45:59.000000000 +0100
@@ -2,7 +2,7 @@
 # restart using wish
 # please alter the name and/or path of
 # the wish shell below, if it doesn't work \
-exec wish8.4 "$0" "$@"
+exec wish "$0" "$@"
 
 package require -exact tpad 1.3
 namespace import ::tpad::*
diff -Nru tpad.orig/etc/tpad.conf tpad/etc/tpad.conf
--- tpad.orig/etc/tpad.conf	2004-08-31 18:18:26.000000000 +0200
+++ tpad/etc/tpad.conf	2005-12-21 16:58:33.000000000 +0100
@@ -74,10 +74,10 @@
 variable maxundo 1  ;# maximum number of compound undo actions
 
 # command to run an external browser to read the HTML help
-variable browsercmd mozilla
+variable browsercmd firefox
 
 # path to where the HTML documentation lives
-variable helpath {/usr/local/share/doc/tpad}
+variable helpath {http://tclpad.sourceforge.net/docs}
 
 # initial placing of non-modal dialogs ({},pointer,widget,none)
 variable placing widget
@@ -88,10 +88,11 @@
 # file where the "open recent" list is saved ({} for non-permanent list)
 variable openrecfile {~/.tpad_history}
 
-variable fmtkind BSD  ;# kind of fmt(1) (GNU, BSD, none)
+variable fmtkind GNU  ;# kind of fmt(1) (GNU, BSD, none)
 variable fmt {fmt}    ;# fmt(1) path (ignored if fmtkind is none)
 
-variable xmlwf {xmlwf}  ;# xmlwf(1) path (empty if you don't have it)
+# please install lib/expat before enabling the xmlwf plugin
+variable xmlwf {}  ;# xmlwf(1) path (empty if you don't have it)
 
 # note that you will also need to install Tcllib for this plugin to work
 variable tidy {}  ;# tidy(1) path (empty if you don't have it)
@@ -101,12 +102,13 @@
 variable fortune {fortune}
 
 # display fortune tip at startup (no, yes); ignored if fortune is empty
-variable tip yes
+# please install extra/games/fortune-mod before enabling tips
+variable tip no
 
 # you will need to change this only if you move or rename the fortune
 # database file and its index (which has the added extension .dat)
 # ignored if fortune is empty
-variable fortunefile {/usr/local/share/tpad/tpad-tips}
+variable fortunefile {/usr/share/tpad/tpad-tips}
 
 set tk_strictMotif 0  ;# strict Motif compliance (0, 1)
 ## end of static vars or constants initialization
diff -Nru tpad.orig/lib/tpad1.3/tpad.tcl tpad/lib/tpad1.3/tpad.tcl
--- tpad.orig/lib/tpad1.3/tpad.tcl	2004-08-31 18:18:26.000000000 +0200
+++ tpad/lib/tpad1.3/tpad.tcl	2005-12-21 16:58:47.000000000 +0100
@@ -50,7 +50,7 @@
 
 ## private commands
 proc ::tpad::LoadConf {} {
-  set global /usr/local/etc/tpad.conf
+  set global /etc/tpad.conf
 
   set peruser ~/.tpadrc
 
diff -Nru tpad.orig/man/man1/tpad.1 tpad/man/man1/tpad.1
--- tpad.orig/man/man1/tpad.1	2004-08-31 18:18:26.000000000 +0200
+++ tpad/man/man1/tpad.1	2005-12-21 16:59:41.000000000 +0100
@@ -26,10 +26,10 @@
 to get the default host and display number.
 .SH FILES
 .TP
-.B /usr/local/bin/tpad
+.B /usr/bin/tpad
 The executable, a wish(1) shell script.
 .TP
-.B /usr/local/etc/tpad.conf
+.B /etc/tpad.conf
 Default global configuration file.
 .TP
 .B ~/.tpadrc
@@ -40,7 +40,7 @@
 .SH SEE ALSO
 wish(1), text(n), vi(1),
 .br
-Detailed HTML documentation \fB/usr/local/share/doc/tpad/index.html\fR,
+Detailed HTML documentation <\fIhttp://tclpad.sourceforge.net/docs\fR>,
 .br
 tpad web site <\fIhttp://monitor.deis.unical.it/ant/tpad\fR>
 .SH AUTHOR
