diff -rupN lsnes/options.build lsnes_new/options.build
--- lsnes/options.build	2013-02-01 22:53:50.655065122 +0000
+++ lsnes_new/options.build	2013-02-01 23:54:03.130770383 +0000
@@ -25,7 +25,7 @@ DOT_EXECUTABLE_SUFFIX=
 
 # C++ compiler (GCC 4.6) for the target system.
 # CROSS_PREFIX is prepended to this command.
-CC=g++-4.6
+CC=g++
 
 # Linker for the target system.
 # CROSS_PREFIX is prepended to this command.
@@ -44,7 +44,7 @@ HOSTCC=$(CC)
 USER_HOSTCCFLAGS=
 
 # C++ complier extra flags.
-USER_CFLAGS=-g
+USER_CFLAGS=-g -O3
 
 # Linker extra flags.
 USER_LDFLAGS=
@@ -93,11 +93,11 @@ SOUND=PORTAUDIO
 # - WIN32MM: Windows Multimedia (Win32 only).
 # - SDL: Simple Directmedia Layer (GRAPHICS=SDL only).
 # - WXWIDGETS: wxWidgets (GRAPHICS=WXWIDGETS only)
-JOYSTICK=DUMMY
+JOYSTICK=EVDEV
 
 # Set to non-empty value (e.g. 'yes') to enable use of Secret Rabbit Code (a.k.a. libsamperate).
 # This is used for high-quality samplerate conversion for dumping.
-SECRET_RABBIT_CODE=
+SECRET_RABBIT_CODE=yes
 
 # Set to non-empty value (e.g. 'yes') to build the gambatte core.
 BUILD_GAMBATTE=
@@ -107,7 +107,7 @@ BUILD_GAMBATTE=
 BSNES_VERSION=085
 
 # Set to non-empty value (e.g. 'yes') if bsnes build uses compatiblity core, not accuracy core.
-BSNES_IS_COMPAT=
+BSNES_IS_COMPAT=yes
 
 # Set to non-empty value (e.g. 'yes') if boost libraries need the '-mt' prefix.
 # Useful on Win32 and Mac OS X.
@@ -124,4 +124,4 @@ NO_DLFCN=
 BOOST_FILESYSTEM3=
 
 # Set to non-empty value (e.g. 'yes') to enable commentary track support (requires Opus codec).
-OPUS_CODEC=
+OPUS_CODEC=yes
