--- vkolev-gsharkdown-gsharkdown-0.4.1/gsharkdown.py	2011-10-12 17:35:05.000000000 +0200
+++ vkolev-gsharkdown-gsharkdown-0.4.1.patch/gsharkdown.py	2011-10-18 13:00:21.000000000 +0200
@@ -121,7 +121,8 @@
     except:
         pass
     CONFIG = ConfigObj("%s/.gsharkdown/gsharkdown.ini" % os.environ.get("HOME"))
-    CONFIG['down_path'] = os.path.join(glib.get_user_special_dir(glib.USER_DIRECTORY_MUSIC), "Grooveshark")
+    MUSIC_DIR = glib.get_user_special_dir(glib.USER_DIRECTORY_MUSIC) or os.path.join(os.environ.get("HOME"), "Music")
+    CONFIG['down_path'] = os.path.join(MUSIC_DIR, "Grooveshark")
     if os.path.isdir(CONFIG['down_path']) == False:
         try:
             os.makedirs(CONFIG['down_path'])
