diff -aur vkolev-gsharkdown-1ad5361b56d9/lib/groove.py vkolev-gsharkdown-1ad5361b56d9.patched/lib/groove.py
--- vkolev-gsharkdown-1ad5361b56d9/lib/groove.py	2012-06-19 23:38:33.000000000 +0200
+++ vkolev-gsharkdown-1ad5361b56d9.patched/lib/groove.py	2013-07-17 15:47:43.018194607 +0200
@@ -19,7 +19,7 @@
 def dummy(error = None):
     pass
 
-_referer = "http://grooveshark.com/JSQueue.swf?20120312.08"
+_referer = "http://grooveshark.com/JSQueue.swf?20130520"
 _token = None
 _isInitialized = False
 _initializingLock = threading.Lock()
@@ -90,7 +90,7 @@
     p["method"] = "getCommunicationToken"
     p["header"] = h
     p["header"]["client"] = "htmlshark"
-    p["header"]["clientRevision"] = "20120312"
+    p["header"]["clientRevision"] = "20130520"
 
     conn = createCurl("https://grooveshark.com/more.php?" + p["method"])
     conn.setopt(pycurl.POST, True)
@@ -120,8 +120,8 @@
     p["parameters"]["query"] = query
     p["header"] = h
     p["header"]["client"] = "htmlshark"
-    p["header"]["clientRevision"] = "20120312"
-    p["header"]["token"] = prepToken("getResultsFromSearch", ":reallyHotSauce:")
+    p["header"]["clientRevision"] = "20130520"
+    p["header"]["token"] = prepToken("getResultsFromSearch", ":nuggetsOfBaller:")
     p["method"] = "getResultsFromSearch"
 
     conn = createCurl("https://grooveshark.com/more.php?" + p["method"])
@@ -158,8 +158,8 @@
     p["parameters"]["country"] = h["country"]
     p["header"] = h
     p["header"]["client"] = "jsqueue"
-    p["header"]["clientRevision"] = "20120312.08"
-    p["header"]["token"] = prepToken("getStreamKeysFromSongIDs", ":circlesAndSquares:")
+    p["header"]["clientRevision"] = "20130520"
+    p["header"]["token"] = prepToken("getStreamKeysFromSongIDs", ":chickenFingers:")
     p["method"] = "getStreamKeysFromSongIDs"
 
     conn = createCurl("https://grooveshark.com/more.php?" + p["method"])
