diff --git a/liboctave/oct-alloc.h b/liboctave/oct-alloc.h
index 13fb9f0..2b57693 100644
--- a/liboctave/oct-alloc.h
+++ b/liboctave/oct-alloc.h
@@ -1,7 +1,6 @@
 /*
 
-Copyright (C) 1996, 1997, 1998, 2002, 2003, 2005, 2006, 2007
-              John W. Eaton
+Copyright (C) 1996-2011 John W. Eaton
 
 This file is part of Octave.
 
@@ -24,6 +23,8 @@ along with Octave; see the file COPYING.  If not, see
 #if !defined (octave_oct_alloc_h)
 #define octave_oct_alloc_h 1
 
+#include <cstddef>
+
 class
 OCTAVE_API
 octave_allocator
diff --git a/src/DLD-FUNCTIONS/urlwrite.cc b/src/DLD-FUNCTIONS/urlwrite.cc
index e190642..c721bc1 100644
--- a/src/DLD-FUNCTIONS/urlwrite.cc
+++ b/src/DLD-FUNCTIONS/urlwrite.cc
@@ -43,7 +43,7 @@ along with Octave; see the file COPYING.  If not, see
 #if defined (HAVE_CURL)
 
 #include <curl/curl.h>
-#include <curl/types.h>
+#include <curl/curlver.h>
 #include <curl/easy.h>
 
 // Write callback function for curl.
