From c318744bcc401979d12ad45b2dbefbfcada18181 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Vadinsk=C3=BD?= <Nicky726@google.com>
Date: Sun, 6 Feb 2011 13:56:45 +0100
Subject: [PATCH 1/2] Added arch as distro option to the build.conf.

---
 build.conf |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/build.conf b/build.conf
index d13e236..f746daa 100644
--- a/build.conf
+++ b/build.conf
@@ -25,9 +25,9 @@ NAME = refpolicy
 # for programs or configurations specific to the
 # distribution.  Setting this will enable options
 # for the distribution.
-# redhat, gentoo, debian, suse, and rhel4 are current options.
+# redhat, gentoo, debian, suse, arch, and rhel4 are current options.
 # Fedora users should enable redhat.
-#DISTRO = redhat
+DISTRO = arch
 
 # Unknown Permissions Handling
 # The behavior for handling permissions defined in the
-- 
1.7.4

From 6cb1c2cdc2d6175d84a94a7ac3b538709f247a44 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Vadinsk=C3=BD?= <Nicky726@google.com>
Date: Sun, 6 Feb 2011 14:03:43 +0100
Subject: [PATCH 2/2] Fixed path of init scripts for Arch Linux (/etc/rc.d/).

---
 policy/modules/system/init.fc |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/policy/modules/system/init.fc b/policy/modules/system/init.fc
index 9775375..f0cebdc 100644
--- a/policy/modules/system/init.fc
+++ b/policy/modules/system/init.fc
@@ -16,6 +16,10 @@ ifdef(`distro_gentoo',`
 /etc/x11/startDM\.sh	--	gen_context(system_u:object_r:initrc_exec_t,s0)
 ')
 
+ifdef(`distro_arch',`
+/etc/rc\.d/.*		--	gen_context(system_u:object_r:initrc_exec_t,s0)
+')
+
 #
 # /dev
 #
-- 
1.7.4

