--- fs/Makefile.in	2013-04-17 18:13:05.000000000 +0200
+++ fs/Makefile.in.new	2014-05-02 23:01:51.103023189 +0200
@@ -368,9 +368,9 @@
 bfsyncca_SOURCES = bfsyncca.cc
 bfsyncca_LDADD = $(GLIB_LIBS)
 bfdefrag_SOURCES = bfdefrag.cc
-bfdefrag_LDADD = $(GLIB_LIBS) $(FUSE_LIBS) libbfsync.la
+bfdefrag_LDADD = $(GLIB_LIBS) $(FUSE_LIBS) $(BDB_LIBS) libbfsync.la
 bfrandomize_SOURCES = bfrandomize.cc
-bfrandomize_LDADD = $(GLIB_LIBS) $(FUSE_LIBS) libbfsync.la
+bfrandomize_LDADD = $(GLIB_LIBS) $(FUSE_LIBS) $(BDB_LIBS) libbfsync.la
 bfperf_SOURCES = bfperf.cc
 bfperf_LDADD = $(GLIB_LIBS) $(FUSE_LIBS) libbfsync.la
 bfdbdump_SOURCES = bfdbdump.cc
