diff -uBr ksimus-0.3.6/ksimus/componentmap.cpp ksimus-0.3.6cppfiles/ksimus/componentmap.cpp
--- ksimus-0.3.6/ksimus/componentmap.cpp	2002-03-28 17:16:09.000000000 +0100
+++ ksimus-0.3.6cppfiles/ksimus/componentmap.cpp	2010-01-16 02:34:19.000000000 +0100
@@ -18,6 +18,7 @@
 #include <limits.h>
 #include <qpainter.h>
 #include <queue>
+#include <stdlib.h>
 
 
 #include "resource.h"
Nur in ksimus-0.3.6cppfiles/ksimus: componentmap.cpp~.
diff -uBr ksimus-0.3.6/ksimus/ksimdata.cpp ksimus-0.3.6cppfiles/ksimus/ksimdata.cpp
--- ksimus-0.3.6/ksimus/ksimdata.cpp	2003-02-16 17:34:32.000000000 +0100
+++ ksimus-0.3.6cppfiles/ksimus/ksimdata.cpp	2010-01-16 02:26:41.000000000 +0100
@@ -70,7 +70,7 @@
 
 //#########################################################################################
 
-EnumDict<KSimData::eVersionType>::tData EnumDict<KSimData::eVersionType>::data[]
+template<> EnumDict<KSimData::eVersionType>::tData EnumDict<KSimData::eVersionType>::data[]
       = { {"error",        KSimData::versionError},
           {"as is",        KSimData::versionAsIs},
           {"unknown",      KSimData::versionUnknown},
Nur in ksimus-0.3.6cppfiles/ksimus: ksimdata.cpp~.
diff -uBr ksimus-0.3.6/ksimus/ksimus.cpp ksimus-0.3.6cppfiles/ksimus/ksimus.cpp
--- ksimus-0.3.6/ksimus/ksimus.cpp	2003-05-01 18:11:55.000000000 +0200
+++ ksimus-0.3.6cppfiles/ksimus/ksimus.cpp	2010-01-16 02:26:58.000000000 +0100
@@ -75,7 +75,7 @@
 //################################################################################
 
 
-EnumDict<eAppViewType>::tData EnumDict<eAppViewType>::data[]
+template<> EnumDict<eAppViewType>::tData EnumDict<eAppViewType>::data[]
       = { {"Sheet View",   APP_SHEET_VIEW},
           {"User View",    APP_USER_VIEW},
           {"Module View",  APP_MODULE_VIEW},
Nur in ksimus-0.3.6cppfiles/ksimus: ksimus.cpp~.
diff -uBr ksimus-0.3.6/ksimus/lib/componentinfo.cpp ksimus-0.3.6cppfiles/ksimus/lib/componentinfo.cpp
--- ksimus-0.3.6/ksimus/lib/componentinfo.cpp	2002-08-05 23:43:32.000000000 +0200
+++ ksimus-0.3.6cppfiles/ksimus/lib/componentinfo.cpp	2010-01-16 02:04:43.000000000 +0100
@@ -32,7 +32,7 @@
 //##########################################################
 
 
-EnumDict<ViewAttribute>::tData EnumDict<ViewAttribute>::data[]
+template<> EnumDict<ViewAttribute>::tData EnumDict<ViewAttribute>::data[]
 			= { {"VA_SHEETVIEW", 			VA_SHEETVIEW},
           {"VA_USERVIEW",  			VA_USERVIEW},
 					{"VA_SHEET_AND_USER", VA_SHEET_AND_USER},
diff -uBr ksimus-0.3.6/ksimus/module/moduledata.cpp ksimus-0.3.6cppfiles/ksimus/module/moduledata.cpp
--- ksimus-0.3.6/ksimus/module/moduledata.cpp	2002-12-29 15:58:37.000000000 +0100
+++ ksimus-0.3.6cppfiles/ksimus/module/moduledata.cpp	2010-01-16 02:05:08.000000000 +0100
@@ -68,7 +68,7 @@
 static const char * sPixmapStore	= "Pixmap Store";
 static const char * sPixmapData = "Pixmap Data";
 
-EnumDict<ModuleViewType>::tData EnumDict<ModuleViewType>::data[]
+template<> EnumDict<ModuleViewType>::tData EnumDict<ModuleViewType>::data[]
       = { {"Generic",    MV_GENERIC},
           {"User View",  MV_USERVIEW},
           {"Pixmap",     MV_PIXMAP},
@@ -82,7 +82,7 @@
 }
 
 
-EnumDict<ModulePixmapStoreType>::tData EnumDict<ModulePixmapStoreType>::data[]
+template<> EnumDict<ModulePixmapStoreType>::tData EnumDict<ModulePixmapStoreType>::data[]
       = { {"Absolute",         MPS_ABSOLTUE},
           {"Relative Module",  MPS_RELATIVE_MODULE},
           {"Internal",         MPS_INTERNAL},
diff -uBr ksimus-0.3.6/ksimus/tools/ksimbooltristate.cpp ksimus-0.3.6cppfiles/ksimus/tools/ksimbooltristate.cpp
--- ksimus-0.3.6/ksimus/tools/ksimbooltristate.cpp	2003-02-28 00:06:19.000000000 +0100
+++ ksimus-0.3.6cppfiles/ksimus/tools/ksimbooltristate.cpp	2010-01-16 02:06:01.000000000 +0100
@@ -32,7 +32,7 @@
 
 //#########################################################################################
 
-EnumDict<eKSimBoolTristate>::tData EnumDict<eKSimBoolTristate>::data[]
+template<> EnumDict<eKSimBoolTristate>::tData EnumDict<eKSimBoolTristate>::data[]
       = { {"False",        KSIMBOOLTRISTATE_FALSE},
           {"True",         KSIMBOOLTRISTATE_TRUE},
           {"Inactive",     KSIMBOOLTRISTATE_INACTIVE},
diff -uBr ksimus-0.3.6/ksimus/tools/ksimfilename.cpp ksimus-0.3.6cppfiles/ksimus/tools/ksimfilename.cpp
--- ksimus-0.3.6/ksimus/tools/ksimfilename.cpp	2002-09-27 21:33:15.000000000 +0200
+++ ksimus-0.3.6cppfiles/ksimus/tools/ksimfilename.cpp	2010-01-16 02:06:03.000000000 +0100
@@ -46,7 +46,7 @@
 
 static EnumDict<KSimFilename::ePathType> pathTypeDict;
 
-EnumDict<KSimFilename::ePathType>::tData EnumDict<KSimFilename::ePathType>::data[]
+template<> EnumDict<KSimFilename::ePathType>::tData EnumDict<KSimFilename::ePathType>::data[]
 			= { {"Absolute",          KSimFilename::PATH_ABSOLUTE},
 					{"Relative Document", KSimFilename::PATH_RELATIVE_DOCUMENT},
 					{"Relative Module",   KSimFilename::PATH_RELATIVE_MODULE},
diff -uBr ksimus-0.3.6/ksimus/tools/ksimtimebase.cpp ksimus-0.3.6cppfiles/ksimus/tools/ksimtimebase.cpp
--- ksimus-0.3.6/ksimus/tools/ksimtimebase.cpp	2002-08-05 23:44:00.000000000 +0200
+++ ksimus-0.3.6cppfiles/ksimus/tools/ksimtimebase.cpp	2010-01-16 02:06:03.000000000 +0100
@@ -31,7 +31,7 @@
 #include "ksimtimeserver.h"
 #include "enumdict.h"
 
-EnumDict<eTimeUnit>::tData EnumDict<eTimeUnit>::data[]
+template<> EnumDict<eTimeUnit>::tData EnumDict<eTimeUnit>::data[]
       = { {"Ticks",    unit_ticks},
           {"1/10ns",   unit_tenth_nsec},
           {"1/10nsec", unit_tenth_nsec},
