diff -rupN ov51x-jpeg-1.5.9-old/ov51x-jpeg-core.c ov51x-jpeg-1.5.9/ov51x-jpeg-core.c
--- ov51x-jpeg-1.5.9-old/ov51x-jpeg-core.c	2011-02-19 18:36:23.386643339 +0100
+++ ov51x-jpeg-1.5.9/ov51x-jpeg-core.c	2011-02-19 18:37:45.912406161 +0100
@@ -8298,11 +8298,11 @@ ov51x_probe(struct usb_interface *intf,

 	init_waitqueue_head(&ov->wq);

-	init_MUTEX(&ov->lock);	/* to 1 == available */
-	init_MUTEX(&ov->buf_lock);
-	init_MUTEX(&ov->param_lock);
-	init_MUTEX(&ov->i2c_lock);
-	init_MUTEX(&ov->cbuf_lock);
+	sema_init(&ov->lock,1);	/* to 1 == available */
+	sema_init(&ov->buf_lock,1);
+	sema_init(&ov->param_lock,1);
+	sema_init(&ov->i2c_lock,1);
+	sema_init(&ov->cbuf_lock,1);

 	ov->buf_state = BUF_NOT_ALLOCATED;
