diff -Nur ov51x-jpeg-1.5.9-old/ov511-decomp.c ov51x-jpeg-1.5.9/ov511-decomp.c
--- ov51x-jpeg-1.5.9-old/ov511-decomp.c	2009-05-14 19:45:52.101366869 +0300
+++ ov51x-jpeg-1.5.9/ov511-decomp.c	2009-05-14 19:46:45.229739888 +0300
@@ -508,6 +508,5 @@
 struct ov51x_decomp_ops ov511_decomp_ops = {
 	.decomp_400 =	Decompress400,
 	.decomp_420 =	Decompress420,
-	.owner =	THIS_MODULE,
 };
 
diff -Nur ov51x-jpeg-1.5.9-old/ov518-decomp.c ov51x-jpeg-1.5.9/ov518-decomp.c
--- ov51x-jpeg-1.5.9-old/ov518-decomp.c	2009-05-14 19:45:52.102366089 +0300
+++ ov51x-jpeg-1.5.9/ov518-decomp.c	2009-05-14 19:46:45.242740219 +0300
@@ -1488,6 +1488,5 @@
 struct ov51x_decomp_ops ov518_decomp_ops = {
 	.decomp_400 =	Decompress400,	
 	.decomp_420 =	Decompress420,	
-	.owner =	THIS_MODULE,
 };
 
diff -Nur ov51x-jpeg-1.5.9-old/ov519-decomp.c ov51x-jpeg-1.5.9/ov519-decomp.c
--- ov51x-jpeg-1.5.9-old/ov519-decomp.c	2009-05-14 19:45:52.102366089 +0300
+++ ov51x-jpeg-1.5.9/ov519-decomp.c	2009-05-14 19:46:45.256740398 +0300
@@ -1534,7 +1534,6 @@
 struct ov51x_decomp_ops ov519_decomp_ops = {
 	.decomp_400 =	Decompress400,	
 	.decomp_420 =	Decompress420,	
-	.owner =	THIS_MODULE,
 };
 
 
diff -Nur ov51x-jpeg-1.5.9-old/ov51x-jpeg-core.c ov51x-jpeg-1.5.9/ov51x-jpeg-core.c
--- ov51x-jpeg-1.5.9-old/ov51x-jpeg-core.c	2009-05-14 19:45:52.102366089 +0300
+++ ov51x-jpeg-1.5.9/ov51x-jpeg-core.c	2009-05-14 19:46:45.327739952 +0300
@@ -6607,7 +6606,6 @@
 
 #ifdef OV511_OLD_V4L
 static struct video_device vdev_template = {
-	.owner =	THIS_MODULE,
 	.name =		"OV511 USB Camera",
 	.type =		VID_TYPE_CAPTURE,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 24)
@@ -6623,7 +6621,6 @@
 #else	/* New V4L API */
 
 static struct v4l2_file_operations ov511_fops = {
-	.owner =	THIS_MODULE,
 	.open =		ov51x_v4l1_open,
 	.release =	ov51x_v4l1_close,
 	.read =		ov51x_v4l1_read,
@@ -6636,7 +6633,6 @@
 
 static struct video_device vdev_template = {
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 27)
-	.owner =	THIS_MODULE,
 #endif
 	.name =		"OV51x USB Camera",
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 27)
diff -Nur ov51x-jpeg-1.5.9-old/ov51x-jpeg.h ov51x-jpeg-1.5.9/ov51x-jpeg.h
--- ov51x-jpeg-1.5.9-old/ov51x-jpeg.h	2009-05-14 19:45:52.101366869 +0300
+++ ov51x-jpeg-1.5.9/ov51x-jpeg.h	2009-05-14 19:46:45.336739493 +0300
@@ -588,7 +588,6 @@
 			  int, int, int);
 	int (*decomp_422)(unsigned char *, unsigned char *, unsigned char *,
 			  int, int, int);
-	struct module *owner;
 };
 
 extern struct ov51x_decomp_ops ov511_decomp_ops;
