diff -Nur ov51x-jpeg-1.5.9-old/ov51x-jpeg-core.c ov51x-jpeg-1.5.9/ov51x-jpeg-core.c
--- ov51x-jpeg-1.5.9-old/ov51x-jpeg-core.c	2009-05-01 03:07:54.000000000 +0300
+++ ov51x-jpeg-1.5.9/ov51x-jpeg-core.c	2009-05-01 03:40:38.000000000 +0300
@@ -12,6 +12,9 @@
 * To support OV7670 sensors as found in Creative Live! Vista IM webcam
 * With reference to One Laptop Per Child sourced kernel patches in 2.6.20 kernel
 * contributed to kernel by Jonathan Corbet.
+*
+* Modified to work with 2.6.29 kernels by Onur Küçük <onur@pardus.org.tr>
+*
 */
 
 
@@ -678,7 +681,6 @@
 	ov->proc_devdir = create_proc_entry(dirname, S_IFDIR, ov51x_jpeg_proc_entry);
 	if (!ov->proc_devdir)
 		return;
-	ov->proc_devdir->owner = THIS_MODULE;
 
 	/* Create "info" entry (human readable device information) */
 	PDEBUG(4, "creating /proc/ov51x-jpeg/ov51x/%s/info", dirname);
@@ -686,7 +688,6 @@
 		ov->proc_devdir, ov511_read_proc_info, ov);
 	if (!ov->proc_info)
 		return;
-	ov->proc_info->owner = THIS_MODULE;
 
 	/* Don't create it if old snapshot mode on (would cause race cond.) */
 	if (!snapshot) {
@@ -697,7 +698,6 @@
 			ov511_read_proc_button, ov);
 		if (!ov->proc_button)
 			return;
-		ov->proc_button->owner = THIS_MODULE;
 	}
 
 	/* Create "control" entry (ioctl() interface) */
@@ -709,7 +709,6 @@
 		unlock_kernel();
 		return;
 	}
-	ov->proc_control->owner = THIS_MODULE;
 	ov->proc_control->data = ov;
 	ov->proc_control->proc_fops = &ov511_control_fops;
 	unlock_kernel();
@@ -762,9 +761,7 @@
 	 */
 
 	ov51x_jpeg_proc_entry = create_proc_entry("ov51x-jpeg", S_IFDIR, &proc_root);
-	if (ov51x_jpeg_proc_entry)
-		ov51x_jpeg_proc_entry->owner = THIS_MODULE;
-	else
+	if (!ov51x_jpeg_proc_entry)
 		err("Unable to create /proc/ov51x-jpeg");
 
 }
@@ -6383,7 +6380,7 @@
 	if (down_interruptible(&ov->lock))
 		return -EINTR;
 
-	rc = video_usercopy(inode, file, cmd, arg, ov51x_v4l1_ioctl_internal);
+	rc = video_usercopy(file, cmd, arg, ov51x_v4l1_ioctl_internal);
 
 	up(&ov->lock);
 	return rc;
@@ -8374,7 +8371,7 @@
 
 	memcpy(ov->vdev, &vdev_template, sizeof(*ov->vdev));
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 27)
-	ov->vdev->dev = dev->dev;
+	ov->vdev->parent = &dev->dev;
 #else
 	ov->vdev->dev = &dev->dev;
 #endif
ov51x-jpeg-1.5.9-old/ov51x-jpeg-core.o ve ov51x-jpeg-1.5.9/ov51x-jpeg-core.o dosyaları birbirinden farklı
