diff --git a/CMakeLists.txt b/CMakeLists.txt
index f362298..4fa4598 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,7 +36,7 @@ set(TPM_DEVICE_NAME "/dev/tpm")
 else()
 set(TPM_LOG_FILE "/var/log/tpmd.log")
 set(TPM_SOCKET_NAME "/var/run/tpm/tpmd_socket:0")
-set(TPM_STORAGE_NAME "/var/lib/tpm/tpm_emulator-1_2_${${PROJECT_NAME}_VERSION_MAJOR}_${${PROJECT_NAME}_VERSION_MINOR}")
+set(TPM_STORAGE_NAME "/var/lib/tpmd/tpm_emulator-1_2_${${PROJECT_NAME}_VERSION_MAJOR}_${${PROJECT_NAME}_VERSION_MINOR}")
 set(TPM_DEVICE_NAME "/dev/tpm")
 endif()
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h)
diff --git a/tpmd/unix/tpmd.c b/tpmd/unix/tpmd.c
index 349f704..d0ec2ee 100644
--- a/tpmd/unix/tpmd.c
+++ b/tpmd/unix/tpmd.c
@@ -268,16 +268,18 @@ static int init_socket(const char *name)
         error("socket(AF_UNIX) failed: %s", strerror(errno));
         return -1;
     }
+    unlink(name);
     mkdirs(name);
     addr.sun_family = AF_UNIX;
     strncpy(addr.sun_path, name, sizeof(addr.sun_path));
-    umask(0177);
+    umask(0117);
     if (bind(sock, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
         error("bind(%s) failed: %s", addr.sun_path, strerror(errno));
         close(sock);
         return -1;
     }
     listen(sock, 1);
+    chown(name, opt_uid, opt_gid);
     return sock;
 }
 
@@ -296,6 +298,8 @@ static void main_loop(void)
     /* open UNIX socket */
     sock = init_socket(opt_socket_name);
     if (sock < 0) exit(EXIT_FAILURE);
+    /* switch uid/gid if required */
+    switch_uid_gid();
     /* init tpm emulator */
     debug("initializing TPM emulator");
     if (tpm_emulator_init(tpm_startup, tpm_config) != 0) {
@@ -389,8 +393,6 @@ int main(int argc, char **argv)
     info("starting TPM Emulator daemon (1.2.%d.%d-%d)",
          VERSION_MAJOR, VERSION_MINOR, VERSION_BUILD);
     parse_options(argc, argv);
-    /* switch uid/gid if required */
-    switch_uid_gid();
     /* init signal handlers */
     init_signal_handler();
     /* unless requested otherwiese, fork and daemonize process */
diff --git a/tpmd_dev/linux/Makefile b/tpmd_dev/linux/Makefile
index a69f8cc..2d0c97c 100644
--- a/tpmd_dev/linux/Makefile
+++ b/tpmd_dev/linux/Makefile
@@ -23,7 +23,7 @@ clean:
 	@$(MAKE) -C $(KERNEL_BUILD) M=$(CURDIR) clean
 	@rm -f Modules.symvers tpmd_dev.rules
 
-TPM_GROUP ?= tss
+TPM_GROUP ?= tpm
 INSTALL ?= install
 
 tpmd_dev.rules: tpmd_dev.rules.in
