--- xotcl-1.6.7.orig/Makefile.in
+++ xotcl-1.6.7/Makefile.in
@@ -30,7 +30,9 @@
 src_generic_dir		= ${srcdir}/generic
 src_man_dir		= ${srcdir}/man
 TCL_LIB_SPEC		= @TCL_LIB_SPEC@
+TCL_STUB_LIB_SPEC	= @TCL_STUB_LIB_SPEC@
 TK_LIB_SPEC		= @TK_LIB_SPEC@
+TK_STUB_LIB_SPEC	= @TK_STUB_LIB_SPEC@
 subdirs			= @subdirs@
 aol_prefix		= @aol_prefix@
 
@@ -429,13 +431,13 @@
 xotclsh: tclAppInit.o $(PKG_OBJECTS) $(CONDITIONAL_STUB_OBJECTS)
 	$(CC) -rdynamic -o $@ tclAppInit.o \
 		$(PKG_OBJECTS) \
-		$(CFLAGS)  $(TCL_LIB_SPEC) \
+		$(CFLAGS)  $(TCL_LIB_SPEC) $(TCL_STUB_LIB_SPEC)\
 		$(DMALLOC_LIB) $(CONDITIONAL_STUB_OBJECTS)
 
 xowish: tkAppInit.o $(PKG_OBJECTS) $(CONDITIONAL_STUB_OBJECTS)
 	$(CC) -rdynamic -o $@ tkAppInit.o \
 		$(PKG_OBJECTS) \
-		$(CFLAGS)  $(TCL_LIB_SPEC) $(TK_LIB_SPEC) \
+		$(CFLAGS)  $(TCL_LIB_SPEC) $(TCL_STUB_LIB_SPEC) $(TK_LIB_SPEC) $(TK_STUB_LIB_SPEC) \
 		$(DMALLOC_LIB) $(CONDITIONAL_STUB_OBJECTS)
 
 install-shells:
