diff -ur mutt-1.5.21.orig/curs_main.c mutt-1.5.21/curs_main.c
--- mutt-1.5.21.orig/curs_main.c	2012-06-24 06:13:26.585601881 +0300
+++ mutt-1.5.21/curs_main.c	2012-06-24 06:14:05.932270391 +0300
@@ -1823,6 +1823,7 @@
 	MAYBE_REDRAW (menu->redraw);
 	break;
 
+      case OP_PURGE_MESSAGE:
       case OP_DELETE:
 
 	CHECK_MSGCOUNT;
@@ -1833,6 +1834,7 @@
 	if (tag)
 	{
 	  mutt_tag_set_flag (M_DELETE, 1);
+	  mutt_tag_set_flag (M_PURGED, (op != OP_PURGE_MESSAGE) ? 0 : 1);
 	  if (option (OPTDELETEUNTAG))
 	    mutt_tag_set_flag (M_TAG, 0);
 	  menu->redraw = REDRAW_INDEX;
@@ -1840,6 +1842,8 @@
 	else
 	{
 	  mutt_set_flag (Context, CURHDR, M_DELETE, 1);
+	  mutt_set_flag (Context, CURHDR, M_PURGED,
+			 (op != OP_PURGE_MESSAGE) ? 0 : 1);
 	  if (option (OPTDELETEUNTAG))
 	    mutt_set_flag (Context, CURHDR, M_TAG, 0);
 	  if (option (OPTRESOLVE))
@@ -2141,11 +2145,13 @@
 	if (tag)
 	{
 	  mutt_tag_set_flag (M_DELETE, 0);
+	  mutt_tag_set_flag (M_PURGED, 0);
 	  menu->redraw = REDRAW_INDEX;
 	}
 	else
 	{
 	  mutt_set_flag (Context, CURHDR, M_DELETE, 0);
+	  mutt_set_flag (Context, CURHDR, M_PURGED, 0);
 	  if (option (OPTRESOLVE) && menu->current < Context->vcount - 1)
 	  {
 	    menu->current++;
@@ -2166,9 +2172,11 @@
 	CHECK_ACL(M_ACL_DELETE, _("undelete message(s)"));
 
 	rc = mutt_thread_set_flag (CURHDR, M_DELETE, 0,
-				   op == OP_UNDELETE_THREAD ? 0 : 1);
+				   op == OP_UNDELETE_THREAD ? 0 : 1)
+	  + mutt_thread_set_flag (CURHDR, M_PURGED, 0,
+				  op == OP_UNDELETE_THREAD ? 0 : 1);
 
-	if (rc != -1)
+	if (rc > -1)
 	{
 	  if (option (OPTRESOLVE))
 	  {
diff -ur mutt-1.5.21.orig/flags.c mutt-1.5.21/flags.c
--- mutt-1.5.21.orig/flags.c	2012-06-24 06:13:26.572268547 +0300
+++ mutt-1.5.21/flags.c	2012-06-24 06:14:05.932270391 +0300
@@ -104,6 +104,16 @@
       }
       break;
 
+    case M_PURGED:
+      if (bf)
+      {
+	if (!h->purged)
+	  h->purged = 1;
+      }
+      else if (h->purged)
+	h->purged = 0;
+      break;
+
     case M_NEW:
 
       if (!mutt_bit_isset(ctx->rights,M_ACL_SEEN))
diff -ur mutt-1.5.21.orig/functions.h mutt-1.5.21/functions.h
--- mutt-1.5.21.orig/functions.h	2012-06-24 06:13:26.585601881 +0300
+++ mutt-1.5.21/functions.h	2012-06-24 06:14:05.932270391 +0300
@@ -121,6 +121,7 @@
   { "toggle-write",		OP_TOGGLE_WRITE,		"%" },
   { "next-thread",		OP_MAIN_NEXT_THREAD,		"\016" },
   { "next-subthread",		OP_MAIN_NEXT_SUBTHREAD,		"\033n" },
+  { "purge-message",		OP_PURGE_MESSAGE,		NULL },
   { "query",			OP_QUERY,			"Q" },
   { "quit",			OP_QUIT,			"q" },
   { "reply",			OP_REPLY,			"r" },
@@ -213,6 +214,7 @@
   { "print-message",	OP_PRINT,			"p" },
   { "previous-thread",	OP_MAIN_PREV_THREAD,		"\020" },
   { "previous-subthread",OP_MAIN_PREV_SUBTHREAD,	"\033p" },
+  { "purge-message",	OP_PURGE_MESSAGE,		NULL },
   { "quit",		OP_QUIT,			"Q" },
   { "exit",		OP_EXIT,			"q" },
   { "reply",		OP_REPLY,			"r" },
diff -ur mutt-1.5.21.orig/mutt.h mutt-1.5.21/mutt.h
--- mutt-1.5.21.orig/mutt.h	2012-06-24 06:13:26.572268547 +0300
+++ mutt-1.5.21/mutt.h	2012-06-24 06:14:05.932270391 +0300
@@ -188,6 +188,7 @@
   M_UNDELETE,
   M_DELETED,
   M_APPENDED,
+  M_PURGED,
   M_FLAG,
   M_TAG,
   M_UNTAG,
@@ -708,6 +709,7 @@
   unsigned int flagged : 1; 		/* marked important? */
   unsigned int tagged : 1;
   unsigned int appended : 1; /* has been saved */
+  unsigned int purged : 1;   /* bypassing the trash folder */
   unsigned int deleted : 1;
   unsigned int changed : 1;
   unsigned int attach_del : 1; 		/* has an attachment marked for deletion */
diff -ur mutt-1.5.21.orig/mx.c mutt-1.5.21/mx.c
--- mutt-1.5.21.orig/mx.c	2012-06-24 06:13:26.572268547 +0300
+++ mutt-1.5.21/mx.c	2012-06-24 06:14:21.622271126 +0300
@@ -806,6 +806,7 @@
     {
       for (i = 0 ; i < ctx->msgcount ; i++)
        if (ctx->hdrs[i]->deleted && !ctx->hdrs[i]->appended
+           && !ctx->hdrs[i]->purged
            && mutt_append_message (ctx_trash, ctx, ctx->hdrs[i], 0, 0) == -1)
          {
            mx_close_mailbox (ctx_trash, NULL);
diff -ur mutt-1.5.21.orig/OPS mutt-1.5.21/OPS
--- mutt-1.5.21.orig/OPS	2012-06-24 06:13:26.585601881 +0300
+++ mutt-1.5.21/OPS	2012-06-24 06:14:05.932270391 +0300
@@ -142,6 +142,7 @@
 OP_PREV_LINE "scroll up one line"
 OP_PREV_PAGE "move to the previous page"
 OP_PRINT "print the current entry"
+OP_PURGE_MESSAGE "really delete the current entry, bypassing the trash folder"
 OP_QUERY "query external program for addresses"
 OP_QUERY_APPEND "append new query results to current results"
 OP_QUIT "save changes to mailbox and quit"
diff -ur mutt-1.5.21.orig/pager.c mutt-1.5.21/pager.c
--- mutt-1.5.21.orig/pager.c	2012-06-24 06:13:26.585601881 +0300
+++ mutt-1.5.21/pager.c	2012-06-24 06:14:05.932270391 +0300
@@ -2336,12 +2336,15 @@
 	MAYBE_REDRAW (redraw);
 	break;
 
+      case OP_PURGE_MESSAGE:
       case OP_DELETE:
 	CHECK_MODE(IsHeader (extra));
 	CHECK_READONLY;
 	CHECK_ACL(M_ACL_DELETE, _("delete message"));
 
 	mutt_set_flag (Context, extra->hdr, M_DELETE, 1);
+	mutt_set_flag (Context, extra->hdr, M_PURGED,
+		       ch != OP_PURGE_MESSAGE ? 0 : 1);
         if (option (OPTDELETEUNTAG))
 	  mutt_set_flag (Context, extra->hdr, M_TAG, 0);
 	redraw = REDRAW_STATUS | REDRAW_INDEX;
@@ -2668,6 +2671,7 @@
 	CHECK_ACL(M_ACL_DELETE, _("undelete message"));
 
 	mutt_set_flag (Context, extra->hdr, M_DELETE, 0);
+	mutt_set_flag (Context, extra->hdr, M_PURGED, 0);
 	redraw = REDRAW_STATUS | REDRAW_INDEX;
 	if (option (OPTRESOLVE))
 	{
@@ -2683,9 +2687,11 @@
 	CHECK_ACL(M_ACL_DELETE, _("undelete message(s)"));
 
 	r = mutt_thread_set_flag (extra->hdr, M_DELETE, 0,
+				  ch == OP_UNDELETE_THREAD ? 0 : 1)
+	  + mutt_thread_set_flag (extra->hdr, M_PURGED, 0,
 				  ch == OP_UNDELETE_THREAD ? 0 : 1);
 
-	if (r != -1)
+	if (r > -1)
 	{
 	  if (option (OPTRESOLVE))
 	  {
diff -ur mutt-1.5.21.orig/PATCHES mutt-1.5.21/PATCHES
--- mutt-1.5.21.orig/PATCHES	2012-06-24 06:13:26.572268547 +0300
+++ mutt-1.5.21/PATCHES	2012-06-24 06:14:05.935603725 +0300
@@ -1 +1,2 @@
 patch-1.5.16hg.cd.trash_folder.vl.1
+patch-1.5.5.1.cd.purge_message.3.4
diff -ur mutt-1.5.21.orig/pattern.c mutt-1.5.21/pattern.c
--- mutt-1.5.21.orig/pattern.c	2012-06-24 06:13:26.585601881 +0300
+++ mutt-1.5.21/pattern.c	2012-06-24 06:14:05.935603725 +0300
@@ -1357,8 +1357,10 @@
       {
 	switch (op)
 	{
-	  case M_DELETE:
 	  case M_UNDELETE:
+	    mutt_set_flag (Context, Context->hdrs[Context->v2r[i]], M_PURGED,
+			   0);
+	  case M_DELETE:
 	    mutt_set_flag (Context, Context->hdrs[Context->v2r[i]], M_DELETE, 
 			  (op == M_DELETE));
 	    break;
