--- libmysqld/Makefile.in.orig	2010-09-15 11:23:18.489669108 +0200
+++ libmysqld/Makefile.in	2010-09-15 11:23:59.450379311 +0200
@@ -500,7 +500,7 @@
 
 noinst_LIBRARIES = libmysqld_int.a
 pkglib_LIBRARIES = libmysqld.a
-SUBDIRS = . examples
+SUBDIRS = .
 libmysqld_sources = libmysqld.c lib_sql.cc emb_qcache.cc
 libmysqlsources = errmsg.c get_password.c libmysql.c client.c pack.c \
                         my_time.c
