diff --git libviso2/CMakeLists.txt libviso2/CMakeLists.txt
index dafc14c..32ee5b2 100644
--- libviso2/CMakeLists.txt
+++ libviso2/CMakeLists.txt
@@ -31,7 +31,7 @@ install(TARGETS viso2
   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
 )
 
-install(DIRECTORY ${PROJECT_NAME}/src
+FILE(GLOB viso2_headers "${PROJECT_NAME}/src/*.h")
+install(FILES ${viso2_headers}  
   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-  PATTERN ".cpp" EXCLUDE
 )
