diff --git a/setup.py b/setup.py
index f822aad..c5b5b5d 100755
--- a/setup.py
+++ b/setup.py
@@ -3,7 +3,7 @@
 
 # Contest Management System - http://cms-dev.github.io/
 # Copyright © 2010-2013 Giovanni Mascellani <mascellani@poisson.phc.unipi.it>
-# Copyright © 2010-2013 Stefano Maggiolo <s.maggiolo@gmail.com>
+# Copyright © 2010-2014 Stefano Maggiolo <s.maggiolo@gmail.com>
 # Copyright © 2010-2012 Matteo Boscariol <boscarim@hotmail.com>
 # Copyright © 2013 Luca Wehrstedt <luca.wehrstedt@gmail.com>
 #
@@ -40,6 +40,12 @@ from glob import glob
 from setuptools import setup
 
 
+# Root directories for the /usr and /var trees.
+ROOT = "MAKEPKG_INSTALL_ROOT"
+USR_ROOT = os.path.join(ROOT, "usr")
+VAR_ROOT = os.path.join(ROOT, "var")
+
+
 def do_setup():
     """Execute the setup thanks to setuptools.
 
@@ -270,15 +276,15 @@ def install():
     cmsuser_grp = grp.getgrnam("cmsuser")
 
     print("copying isolate to /usr/local/bin/.")
-    makedir(os.path.join("/", "usr", "local", "bin"), root, 0755)
+    makedir(os.path.join(USR_ROOT, "bin"), root, 0755)
     copyfile(os.path.join(".", "isolate", "isolate"),
-             os.path.join("/", "usr", "local", "bin", "isolate"),
+             os.path.join(USR_ROOT, "bin", "isolate"),
              root, 04750, group=cmsuser_grp)
 
     print("copying configuration to /usr/local/etc/.")
-    makedir(os.path.join("/", "usr", "local", "etc"), root, 0755)
+    makedir(os.path.join(USR_ROOT, "etc"), root, 0755)
     for conf_file_name in ["cms.conf", "cms.ranking.conf"]:
-        conf_file = os.path.join("/", "usr", "local", "etc", conf_file_name)
+        conf_file = os.path.join(USR_ROOT, "etc", conf_file_name)
         # Skip if destination is a symlink
         if os.path.islink(conf_file):
             continue
@@ -295,7 +301,7 @@ def install():
         country_code = re.search("/([^/]*)\.po", locale).groups()[0]
         print("  %s" % country_code)
         path = os.path.join("cms", "server", "mo", country_code, "LC_MESSAGES")
-        dest_path = os.path.join("/", "usr", "local", "share", "locale",
+        dest_path = os.path.join(USR_ROOT, "share", "locale",
                                  country_code, "LC_MESSAGES")
         makedir(dest_path, root, 0755)
         copyfile(os.path.join(path, "cms.mo"),
@@ -303,11 +309,12 @@ def install():
                  root, 0644)
 
     print("creating directories.")
-    dirs = [os.path.join("/", "var", "local", "log"),
-            os.path.join("/", "var", "local", "cache"),
-            os.path.join("/", "var", "local", "lib"),
-            os.path.join("/", "var", "local", "run"),
-            os.path.join("/", "usr", "local", "share")]
+    dirs = [os.path.join(VAR_ROOT, "log"),
+            os.path.join(VAR_ROOT, "cache"),
+            os.path.join(VAR_ROOT, "lib"),
+            os.path.join(ROOT, "run"),  # On Archlinux /var/run is a symlink to /run
+            os.path.join(USR_ROOT, "include"),
+            os.path.join(USR_ROOT, "share")]
     for _dir in dirs:
         # Skip if destination is a symlink
         if os.path.islink(os.path.join(_dir, "cms")):
