Index: cblrepo/diagrams-core-1.2.0.4/diagrams-core.cabal
===================================================================
--- cblrepo.orig/diagrams-core-1.2.0.4/diagrams-core.cabal
+++ cblrepo/diagrams-core-1.2.0.4/diagrams-core.cabal
@@ -44,7 +44,7 @@ Library
                        newtype >= 0.2 && < 0.3,
                        monoid-extras >= 0.3 && < 0.4,
                        dual-tree >= 0.2 && < 0.3,
-                       lens >= 4.0 && < 4.7
+                       lens >= 4.0
   hs-source-dirs:      src
 
   Other-extensions:    DeriveDataTypeable
