Index: cblrepo/diagrams-core-1.2.0.4/diagrams-core.cabal
===================================================================
--- cblrepo.orig/diagrams-core-1.2.0.4/diagrams-core.cabal
+++ cblrepo/diagrams-core-1.2.0.4/diagrams-core.cabal
@@ -38,13 +38,13 @@ Library
   Build-depends:       base >= 4.2 && < 4.8,
                        containers >= 0.4.2 && < 0.6,
                        semigroups >= 0.8.4 && < 0.17,
-                       vector-space >= 0.8.4 && < 0.9,
+                       vector-space >= 0.8.4,
                        vector-space-points >= 0.1 && < 0.3,
                        MemoTrie >= 0.4.7 && < 0.7,
                        newtype >= 0.2 && < 0.3,
                        monoid-extras >= 0.3 && < 0.4,
                        dual-tree >= 0.2 && < 0.3,
-                       lens >= 4.0 && < 4.7
+                       lens >= 4.0
   hs-source-dirs:      src
 
   Other-extensions:    DeriveDataTypeable
