diff -aur SOEM/test/linux/ebox/ebox.c SOEM.new/test/linux/ebox/ebox.c
--- SOEM/test/linux/ebox/ebox.c	2013-02-26 22:07:18.000000000 +0100
+++ SOEM.new/test/linux/ebox/ebox.c	2013-08-26 15:35:19.896813499 +0200
@@ -20,14 +20,7 @@
 #include <time.h>
 #include <pthread.h>
 #include <math.h>
-
-#include "ethercattype.h"
-#include "nicdrv.h"
-#include "ethercatbase.h"
-#include "ethercatmain.h"
-#include "ethercatcoe.h"
-#include "ethercatconfig.h"
-#include "ethercatdc.h"
+#include <soem.h>
 
 #define NSEC_PER_SEC 1000000000
 
diff -aur SOEM/test/linux/eepromtool/eepromtool.c SOEM.new/test/linux/eepromtool/eepromtool.c
--- SOEM/test/linux/eepromtool/eepromtool.c	2013-02-26 22:07:18.000000000 +0100
+++ SOEM.new/test/linux/eepromtool/eepromtool.c	2013-08-26 15:35:32.666813493 +0200
@@ -20,15 +20,7 @@
 #include <sys/time.h>
 #include <unistd.h>
 #include <time.h>
-
-#include "ethercattype.h"
-#include "nicdrv.h"
-#include "ethercatbase.h"
-#include "ethercatmain.h"
-#include "ethercatcoe.h"
-//#include "ethercatfoe.h"
-//#include "ethercatconfig.h"
-//#include "ethercatprint.h"
+#include <soem.h>
 
 #define MAXBUF 32768
 #define STDBUF 2048
diff -aur SOEM/test/linux/firm_update/firm_update.c SOEM.new/test/linux/firm_update/firm_update.c
--- SOEM/test/linux/firm_update/firm_update.c	2013-02-26 22:07:18.000000000 +0100
+++ SOEM.new/test/linux/firm_update/firm_update.c	2013-08-26 15:35:45.583480240 +0200
@@ -17,15 +17,7 @@
 #include <string.h>
 #include <sys/time.h>
 #include <unistd.h>
-
-#include "ethercattype.h"
-#include "nicdrv.h"
-#include "ethercatbase.h"
-#include "ethercatmain.h"
-#include "ethercatcoe.h"
-#include "ethercatfoe.h"
-#include "ethercatconfig.h"
-#include "ethercatprint.h"
+#include <soem.h>
 
 #define FWBUFSIZE (8 * 1024 * 1024)
 
diff -aur SOEM/test/linux/red_test/red_test.c SOEM.new/test/linux/red_test/red_test.c
--- SOEM/test/linux/red_test/red_test.c	2013-02-26 22:07:18.000000000 +0100
+++ SOEM.new/test/linux/red_test/red_test.c	2013-08-26 15:35:57.316813584 +0200
@@ -20,15 +20,7 @@
 #include <time.h>
 #include <pthread.h>
 #include <math.h>
-
-#include "ethercattype.h"
-#include "nicdrv.h"
-#include "ethercatbase.h"
-#include "ethercatmain.h"
-#include "ethercatcoe.h"
-#include "ethercatconfig.h"
-#include "ethercatdc.h"
-#include "ethercatprint.h"
+#include <soem.h>
 
 #define NSEC_PER_SEC 1000000000
 #define EC_TIMEOUTMON 500
diff -aur SOEM/test/linux/simple_test/simple_test.c SOEM.new/test/linux/simple_test/simple_test.c
--- SOEM/test/linux/simple_test/simple_test.c	2013-02-26 22:07:18.000000000 +0100
+++ SOEM.new/test/linux/simple_test/simple_test.c	2013-08-26 15:36:12.010147060 +0200
@@ -14,16 +14,7 @@
 #include <sys/time.h>
 #include <unistd.h>
 #include <pthread.h>
-
-#include "ethercattype.h"
-#include "nicdrv.h"
-#include "ethercatbase.h"
-#include "ethercatmain.h"
-#include "ethercatdc.h"
-#include "ethercatcoe.h"
-#include "ethercatfoe.h"
-#include "ethercatconfig.h"
-#include "ethercatprint.h"
+#include <soem.h>
 
 #define EC_TIMEOUTMON 500
 
diff -aur SOEM/test/linux/slaveinfo/slaveinfo.c SOEM.new/test/linux/slaveinfo/slaveinfo.c
--- SOEM/test/linux/slaveinfo/slaveinfo.c	2013-02-26 22:07:18.000000000 +0100
+++ SOEM.new/test/linux/slaveinfo/slaveinfo.c	2013-08-26 15:36:23.370147007 +0200
@@ -13,15 +13,7 @@
 
 #include <stdio.h>
 #include <string.h>
-
-#include "ethercattype.h"
-#include "nicdrv.h"
-#include "ethercatbase.h"
-#include "ethercatmain.h"
-#include "ethercatconfig.h"
-#include "ethercatcoe.h"
-#include "ethercatdc.h"
-#include "ethercatprint.h"
+#include <soem.h>
 
 char IOmap[4096];
 ec_ODlistt ODlist;
