diff --git a/sgp/SGP.cc b/sgp/SGP.cc
index caaca1b..4d575f5 100644
--- a/sgp/SGP.cc
+++ b/sgp/SGP.cc
@@ -392,7 +392,7 @@ try
   std::string configPath = FileMan::joinPaths(configFolderPath, "ja2.ini");
   std::string gameResRootPath = findRootGameResFolder(configPath);
 
-  std::string externalizedDataPath = FileMan::joinPaths(exeFolder, "externalized");
+  std::string externalizedDataPath = "/usr/share/ja2/externalized";
 
   DefaultContentManager *cm;
 
