diff -Naur runvdr-extreme-0.4.2-orig//runvdr.conf.example runvdr-extreme-0.4.2/runvdr.conf.example
--- runvdr-extreme-0.4.2-orig//runvdr.conf.example	2009-06-21 00:27:59.000000000 +0100
+++ runvdr-extreme-0.4.2/runvdr.conf.example	2010-04-26 21:52:13.287259117 +0100
@@ -27,7 +27,7 @@
 ##### -------------------------
 
 # The VDR binary to run:
-VDRPRG="/usr/local/bin/vdr"
+VDRPRG="/usr/bin/vdr"
 
 # Command to load the DVB drivers
 # DVBLOAD will be launched before VDR starts and whenever the drivers
@@ -43,16 +43,16 @@
 #}
 
 # Use VIDEODIR as video directory. The default is /video.
-VIDEODIR="/video"
+VIDEODIR="/var/spool/video"
 
 # Read config files from directory CONFIGDIR (default is to read them
 # from the video directory).
-CONFIGDIR="/video"
+CONFIGDIR="/etc/vdr"
 
 # Location of locale files for VDR 1.5.14 and newer. Default is to use
 # the builtin "./locale" or whatever was compiled as LOCDIR into VDR.
 # 
-# LOCALEDIR="/usr/src/vdr/locale"
+LOCALEDIR="/usr/share/locale"
 
 ##### -----------
 #####   Plugins
@@ -82,7 +82,7 @@
 AddPlugin
 
 # Search for plugins in directory LIBDIR. Default is ./PLUGINS/lib.
-LIBDIR=""
+LIBDIR="/usr/lib/vdr"
 
 # If you use PluginSetup: Where is the the plugin_setup_runvdr.conf file?
 # PLUGINSETUPCONF="$CONFIGDIR/plugins/plugin_setup_runvdr.conf"
@@ -98,12 +98,12 @@
 # 'root' is necessary if the system time shall be set from the transponder 
 # data, but for security reasons vdr can switch to a lesser privileged user
 # id during normal operation.
-USER=""
+USER="vdr"
 
 # Write the EPG data into EPGFILE. Default is /video/epg.data. 
 # Use EPGFILE=- to disable this. If EPGFILE is a directory, the file 
 # epg.data will be created in that directory.
-EPGFILE=""
+EPGFILE="/var/spool/epg"
 
 # Call RECORDCMD before and after a recording.
 RECORDCMD=""
@@ -147,7 +147,7 @@
 # grabbing images to disk is disabled.
 GRAB=
 
-# Use a LIRC remote control device. If set to 1, vdr uses /dev/lircd.
+# Use a LIRC remote control device. If set to 1, vdr uses /var/run/lirc/lircd.
 # If not set, or set to 0, don't use LIRC.
 LIRC=
 
@@ -184,11 +184,11 @@
 RUNVDRPID=/var/run/runvdr.pid
 
 # Set the controlling terminal. For example, /dev/tty8
-TERMINAL="/dev/tty8"
+TERMINAL=""
 
 # Switch console to some terminal? Needs terminal number. 
 # For example, SWITCHTERMINAL=8
-SWITCHTERMINAL=8
+SWITCHTERMINAL=
 
 # Set language for locale to run VDR on.
 # This affects sort options in recordings.
