--- src/dart-sdk/bin/dart2js	2013-10-22 05:51:13.000000000 -0600
+++ dart2js	2013-10-28 11:38:35.969349331 -0600
@@ -4,11 +4,12 @@
 # BSD-style license that can be found in the LICENSE file.
 
 function follow_links() {
-  while [ -h "$1" ]; do
+  _f=$1
+  while [ -h "$_f" ]; do
     # On Mac OS, readlink -f doesn't work.
-    1="$(readlink "$1")"
+    _f="$(readlink "$_f")"
   done
-  echo "$1"
+  echo "$_f"
 }
 
 # Unlike $0, $BASH_SOURCE points to the absolute path of this file.
