diff -ruN FLENS-lite.orig/config.ubuntu FLENS-lite.patched/config.ubuntu
--- FLENS-lite.orig/config.ubuntu	2013-11-14 14:01:53.180588379 +0100
+++ FLENS-lite.patched/config.ubuntu	2013-11-14 14:03:54.873447108 +0100
@@ -1,6 +1,8 @@
 CXX         = g++
-CXXFLAGS   += -Wall -Wextra -Wstrict-overflow=0 -Werror -g -O3 -fPIC
+CXXFLAGS   += -Wall -Wextra -Wstrict-overflow=0 -g -O3 -fPIC
 INCDIRS    += -I.
 DYLIB_EXT   = so
 CXXDYLIB    = -shared
-LDFLAGS    += -llapack -latlas -lblas
+LDFLAGS    += -llapack -lblas
+MAJORVERSION = 20090712cvs
+MINORVERSION = 0
diff -ruN FLENS-lite.orig/flens/Makefile FLENS-lite.patched/flens/Makefile
--- FLENS-lite.orig/flens/Makefile	2013-11-14 14:01:53.177255058 +0100
+++ FLENS-lite.patched/flens/Makefile	2013-11-14 14:03:54.870113788 +0100
@@ -5,6 +5,9 @@
 include ../Makefile.common
 
 DYLIB   = libflens.$(DYLIB_EXT)
+DYLIBMAJOR = $(DYLIB).$(MAJORVERSION)
+DYLIBFULL = $(DYLIB).$(MAJORVERSION).$(MINORVERSION)
+LDEXTRAFLAGS = -Wl,-soname,$(DYLIBMAJOR)
 OBJDIR  = .obj/
 DEPDIR  = .dep/
 SOURCES := $(wildcard *.cc)
@@ -14,8 +17,10 @@
 
 .PHONY : all
 all: $(OBJDIR) $(DEPDIR) $(OBJECTS)
-	$(CXX) $(CXXDYLIB) $(CXXFLAGS) -o $(DYLIB) $(OBJDIR)*.o $(LDFLAGS)
-	cp $(DYLIB) $(FLENS_HOME)
+	$(CXX) $(CXXDYLIB) $(CXXFLAGS) -o $(DYLIBFULL) $(OBJDIR)*.o $(LDFLAGS) $(LDEXTRAFLAGS)
+	ln -s $(DYLIBFULL) $(DYLIBMAJOR)
+	ln -s $(DYLIBMAJOR) $(DYLIB)
+	cp -d $(DYLIB) $(DYLIBMAJOR) $(DYLIBFULL) $(FLENS_HOME)
 	
 ifneq ($(MAKECMDGOALS),clean)
 -include $(DEPS)
diff -ruN FLENS-lite.orig/flens/scalaroperations.tcc FLENS-lite.patched/flens/scalaroperations.tcc
--- FLENS-lite.orig/flens/scalaroperations.tcc	2013-11-14 14:01:53.177255058 +0100
+++ FLENS-lite.patched/flens/scalaroperations.tcc	2013-11-14 14:03:54.866780467 +0100
@@ -252,9 +252,9 @@
     eval(const X &x, const Y &y)
     {
         typedef typename Promotion<X, Y>::Type T;
-        const T &X = x;
-        const T &Y = y;
-        return std::max(X,Y);
+        const T &xx = x;
+        const T &yy = y;
+        return std::max(xx,yy);
     }
 };
 
@@ -306,9 +306,9 @@
     eval(const X &x, const Y &y)
     {
         typedef typename Promotion<X, Y>::Type T;
-        const T &X = x;
-        const T &Y = y;
-        return std::min(X,Y);
+        const T &xx = x;
+        const T &yy = y;
+        return std::min(xx,yy);
     }
 };
 
diff -ruN FLENS-lite.orig/Makefile FLENS-lite.patched/Makefile
--- FLENS-lite.orig/Makefile	2013-11-14 14:01:53.197254980 +0100
+++ FLENS-lite.patched/Makefile	2013-11-14 14:03:54.886780389 +0100
@@ -19,7 +19,7 @@
 install:
 	$(MKDIR) $(PREFIX)/lib
 	$(MKDIR) $(PREFIX)/include/flens/
-	cp libflens.$(DYLIB_EXT) $(PREFIX)/lib
+	cp -d libflens.$(DYLIB_EXT)* $(PREFIX)/lib
 	cp flens/*.h flens/*.tcc $(PREFIX)/include/flens/
 
 uninstall:
diff -ruN FLENS-lite.orig/Makefile.common FLENS-lite.patched/Makefile.common
--- FLENS-lite.orig/Makefile.common	2013-11-14 14:01:53.197254980 +0100
+++ FLENS-lite.patched/Makefile.common	2013-11-14 14:03:54.886780389 +0100
@@ -1,12 +1,12 @@
 ifndef FLENS_HOME
-export FLENS_HOME = $(PWD)
+export FLENS_HOME = $(shell pwd)
 endif
 
-PREFIX = /usr/local/
+PREFIX = $(DESTDIR)/usr
 
 LIBDIRS  = flens
 EXECDIRS = tutorials examples tests
-SUBDIRS = $(LIBDIRS) $(EXECDIRS)
+SUBDIRS = $(LIBDIRS)
 INCDIRS = -I $(FLENS_HOME) -I/sw/include
 
 # some shell commands
