diff --git a/conf/example.gitolite.rc b/conf/example.gitolite.rc
index 4adcb5d..678308e 100644
--- a/conf/example.gitolite.rc
+++ b/conf/example.gitolite.rc
@@ -9,8 +9,8 @@
 
 # --------------------------------------
 # Do not change the next two lines unless you know what you're doing
-# $GL_PACKAGE_CONF = "";
-# $GL_PACKAGE_HOOKS = "";
+$GL_PACKAGE_CONF = "/etc/gitolite";
+$GL_PACKAGE_HOOKS = "/usr/share/gitolite/hooks";
 
 # --------------------------------------
 # MIRRORING SUPPORT
diff --git a/src/gl-setup b/src/gl-setup
index f9f50a2..54b1b50 100755
--- a/src/gl-setup
+++ b/src/gl-setup
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-GL_PACKAGE_CONF=/tmp/share/gitolite/conf
+GL_PACKAGE_CONF=/etc/gitolite
 # must be the same as the value for the same variable in
 # $GL_PACKAGE_CONF/example.gitolite.rc.  Sorry about the catch-22 :)
 
