diff --git a/src/playlist/playlistcontainer.ui b/src/playlist/playlistcontainer.ui
index 8429618..dff4095 100644
--- a/src/playlist/playlistcontainer.ui
+++ b/src/playlist/playlistcontainer.ui
@@ -148,7 +148,7 @@
       <bool>false</bool>
      </property>
      <property name="sortingEnabled">
-      <bool>true</bool>
+      <bool>false</bool>
      </property>
      <property name="allColumnsShowFocus">
       <bool>true</bool>
