diff --git a/wikipediaDumpReader-0.2.9/src/bz2module.c b/wikipediaDumpReader-0.2.9/src/bz2module.c
index 90a0bbb..41d25ea 100644
--- a/wikipediaDumpReader-0.2.9/src/bz2module.c
+++ b/wikipediaDumpReader-0.2.9/src/bz2module.c
@@ -12,15 +12,6 @@ Copyright (c) 2002  Python Software Foundation; All Rights Reserved
 #include "bzlib.h"
 #include "structmember.h"
 
-/* Py_ssize_t is a signed integral type such that sizeof(Py_ssize_t) ==
- * sizeof(size_t).  C99 doesn't define such a thing directly (size_t is an
- * unsigned integral type).  See PEP 353 for details.
- */
-#ifndef Py_ssize_t
-// on Python 2.4, not 2.5
-typedef ssize_t		Py_ssize_t;
-#endif
-
 #ifdef WITH_THREAD
 #include "pythread.h"
 #endif
