Index: oneiric/panel/applets/CMakeLists.txt
===================================================================
--- oneiric.orig/panel/applets/CMakeLists.txt	2011-07-10 21:27:57.764440782 +0200
+++ oneiric/panel/applets/CMakeLists.txt	2011-07-10 23:30:22.820634209 +0200
@@ -15,14 +15,14 @@
 pkg_check_modules(QTBAMF REQUIRED libqtbamf)
 pkg_check_modules(DBUSMENUQT REQUIRED dbusmenu-qt)
 pkg_check_modules(GTK REQUIRED gtk+-2.0)
-pkg_check_modules(INDICATOR REQUIRED indicator)
+pkg_check_modules(INDICATOR REQUIRED indicator-0.4)
 pkg_check_modules(WNCK REQUIRED libwnck-1.0)
 
 find_package(X11 REQUIRED)
 
 # Get indicator dirs from pkgconfig
-read_pkg_variable(INDICATOR_DIR       indicator indicatordir)
-read_pkg_variable(INDICATOR_ICONS_DIR indicator iconsdir)
+read_pkg_variable(INDICATOR_DIR       indicator-0.4 indicatordir)
+read_pkg_variable(INDICATOR_ICONS_DIR indicator-0.4 iconsdir)
 configure_file(indicator-config.h.in indicator-config.h)
 
 # Sources
Index: oneiric/libunity-2d-private/Unity2d/CMakeLists.txt
===================================================================
--- oneiric.orig/libunity-2d-private/Unity2d/CMakeLists.txt	2011-07-10 23:30:39.180715334 +0200
+++ oneiric/libunity-2d-private/Unity2d/CMakeLists.txt	2011-07-10 23:30:57.044803923 +0200
@@ -8,7 +8,7 @@
 pkg_check_modules(GDK REQUIRED gdk-2.0)
 pkg_check_modules(GIO REQUIRED gio-2.0)
 pkg_check_modules(STARTUPNOTIFICATION REQUIRED libstartup-notification-1.0)
-pkg_check_modules(INDICATOR REQUIRED indicator)
+pkg_check_modules(INDICATOR REQUIRED indicator-0.4)
 
 find_package(X11 REQUIRED)
 
