--- launcher/UnityApplications/CMakeLists.txt.old	2011-01-18 16:14:55.909999999 +0100
+++ launcher/UnityApplications/CMakeLists.txt	2011-01-18 16:10:48.569999998 +0100
@@ -3,6 +3,7 @@
 pkg_check_modules(QTGCONF REQUIRED libqtgconf)
 pkg_check_modules(QTDEE REQUIRED libqtdee)
 pkg_check_modules(GDK REQUIRED gdk-2.0)
+pkg_check_modules(GIO REQUIRED gio-unix-2.0)
 pkg_check_modules(GIO REQUIRED gio-2.0)
 pkg_check_modules(WNCK REQUIRED libwnck-1.0)
 
