*** unity-2d/launcher/UnityApplications/CMakeLists.txt	2011-06-01 12:45:29.000000000 +1100
--- build/launcher/UnityApplications/CMakeLists.txt	2011-06-01 12:45:36.000000000 +1100
***************
*** 6,11 ****
--- 6,12 ----
  pkg_check_modules(GLIB REQUIRED glib-2.0)
  pkg_check_modules(GDK REQUIRED gdk-2.0)
  pkg_check_modules(GIO REQUIRED gio-2.0)
+ pkg_check_modules(GIO_UNIX REQUIRED gio-unix-2.0)
  pkg_check_modules(WNCK REQUIRED libwnck-1.0)
  pkg_check_modules(STARTUPNOTIFICATION REQUIRED libstartup-notification-1.0)
  pkg_check_modules(INDICATOR REQUIRED indicator)
***************
*** 71,76 ****
--- 72,78 ----
      ${GLIB_INCLUDE_DIRS}
      ${GDK_INCLUDE_DIRS}
      ${GIO_INCLUDE_DIRS}
+     ${GIO_UNIX_INCLUDE_DIRS}
      ${WNCK_INCLUDE_DIRS}
      ${STARTUPNOTIFICATION_INCLUDE_DIRS}
      ${INDICATOR_INCLUDE_DIRS}
***************
*** 88,93 ****
--- 90,96 ----
      ${GLIB_LDFLAGS}
      ${GDK_LDFLAGS}
      ${GIO_LDFLAGS}
+     ${GIO_UNIX_LDFLAGS}
      ${WNCK_LDFLAGS}
      ${STARTUPNOTIFICATION_LDFLAGS}
      ${INDICATOR_LDFLAGS}
Only in build/launcher/UnityApplications: CMakeLists.txt.orig
diff -cr unity-2d/libunity-2d-private/Unity2d/dragitemwithurl.cpp build/libunity-2d-private/Unity2d/dragitemwithurl.cpp
*** unity-2d/libunity-2d-private/Unity2d/dragitemwithurl.cpp	2011-06-01 12:45:29.000000000 +1100
--- build/libunity-2d-private/Unity2d/dragitemwithurl.cpp	2011-06-01 12:47:23.000000000 +1100
***************
*** 25,31 ****
  // GIO
  #undef signals
  extern "C" {
!     #include <gio/gdesktopappinfo.h>
  }
  
  // Qt
--- 25,31 ----
  // GIO
  #undef signals
  extern "C" {
!     #include <gio-unix-2.0/gio/gdesktopappinfo.h>
  }
  
  // Qt
Only in build/libunity-2d-private/Unity2d: dragitemwithurl.cpp~
