diff -pruN unity-2d.orig/libunity-2d-private/Unity2d/dragitemwithurl.cpp unity-2d/libunity-2d-private/Unity2d/dragitemwithurl.cpp
--- unity-2d.orig/libunity-2d-private/Unity2d/dragitemwithurl.cpp	2011-06-10 12:25:28.000000000 +1100
+++ unity-2d/libunity-2d-private/Unity2d/dragitemwithurl.cpp	2011-06-10 12:27:41.000000000 +1100
@@ -25,7 +25,7 @@
 // GIO
 #undef signals
 extern "C" {
-    #include <gio/gdesktopappinfo.h>
+    #include <gio-unix-2.0/gio/gdesktopappinfo.h>
 }
 
 // Qt
diff -pruN unity-2d.orig/libunity-2d-private/Unity2d/launcherapplication.h unity-2d/libunity-2d-private/Unity2d/launcherapplication.h
--- unity-2d.orig/libunity-2d-private/Unity2d/launcherapplication.h	2011-06-10 12:25:28.000000000 +1100
+++ unity-2d/libunity-2d-private/Unity2d/launcherapplication.h	2011-06-10 12:26:27.000000000 +1100
@@ -17,7 +17,7 @@
 #ifndef LAUNCHERAPPLICATION_H
 #define LAUNCHERAPPLICATION_H
 
-#include <gio/gdesktopappinfo.h>
+#include <gio-unix-2.0/gio/gdesktopappinfo.h>
 #include <libwnck/libwnck.h>
 
 // libindicator
