diff -u exitx.orig//main.c exitx/main.c
--- exitx.orig//main.c	2009-05-17 14:41:55.000000000 +0800
+++ exitx/main.c	2010-05-09 21:29:11.000000000 +0800
@@ -11,12 +11,19 @@
 #include <memory.h>
 #include <signal.h>
 #include <stdio.h>
+#include <glib/gi18n.h>
+#include <locale.h>
 
 #include "halt.h"
 #include "logout.h"
 #include "reboot.h"
 #include "hibernate.h"
 
+#define PACKAGE "exitx" //软件包名
+#define LOCALEDIR "/usr/share/locale" //locale所在目录
+//#define _(string)   gettext(string)
+//#define N_(string)  string
+
 #define COLOR			"#b6c4d7"
 #define BORDER			6
 
@@ -352,6 +359,10 @@
 
 	gtk_init(&argc, &argv);
 
+    bindtextdomain(PACKAGE, LOCALEDIR);
+    //以上函数用来设定国际化翻译包所在位置
+    textdomain(PACKAGE);
+    //      //以上函数用来设定国际化翻译包名称，省略了.mo
 	/* get screen with pointer */
 	screen = xfce_gdk_display_locate_monitor_with_pointer(NULL, &monitor);
 	if (screen == NULL) {
@@ -404,7 +415,8 @@
 
 	/* - -------------------------------- - */
 	/* label */
-	label = gtk_label_new("<b><big>退出系统</big></b>");
+	//label = gtk_label_new("<b><big>退出系统</big></b>");
+	label = gtk_label_new(_("<b><big>Exit System</big></b>"));
 	gtk_label_set_use_markup(GTK_LABEL(label), TRUE);
 	gtk_widget_show(label);
 	gtk_box_pack_start(GTK_BOX(hbox), label, TRUE, TRUE, 0);
@@ -433,7 +445,8 @@
 	gtk_box_pack_start(GTK_BOX(vbox2), image, FALSE, FALSE, 0);
 	g_object_unref(icon);
 
-	label = gtk_label_new("退出");
+	//label = gtk_label_new("退出");
+	label = gtk_label_new(_("Exit"));
 	gtk_widget_show(label);
 	gtk_box_pack_start(GTK_BOX(vbox2), label, FALSE, FALSE, 0);
 
@@ -455,7 +468,8 @@
 	gtk_box_pack_start(GTK_BOX(vbox2), image, FALSE, FALSE, 0);
 	g_object_unref(icon);
 
-	label = gtk_label_new("待机");
+	//label = gtk_label_new("待机");
+	label = gtk_label_new(_("Standby"));
 	gtk_widget_show(label);
 	gtk_box_pack_start(GTK_BOX(vbox2), label, FALSE, FALSE, 0);
 
@@ -477,7 +491,8 @@
 	gtk_box_pack_start(GTK_BOX(vbox2), image, FALSE, FALSE, 0);
 	g_object_unref(icon);
 
-	label = gtk_label_new("重启");
+	//label = gtk_label_new("重启");
+	label = gtk_label_new(_("Reboot"));
 	gtk_widget_show(label);
 	gtk_box_pack_start(GTK_BOX(vbox2), label, FALSE, FALSE, 0);
 
@@ -499,7 +514,8 @@
 	gtk_box_pack_start(GTK_BOX(vbox2), image, FALSE, FALSE, 0);
 	g_object_unref(icon);
 
-	label = gtk_label_new("关机");
+	//label = gtk_label_new("关机");
+	label = gtk_label_new(_("Halt"));
 	gtk_widget_show(label);
 	gtk_box_pack_start(GTK_BOX(vbox2), label, FALSE, FALSE, 0);
 
@@ -520,7 +536,8 @@
 	gtk_widget_show(vbox2);
 	gtk_box_pack_start(GTK_BOX(hbox), vbox2, TRUE, TRUE, 0);
 
-	cancel_button = gtk_button_new_with_label("取消");
+	//cancel_button = gtk_button_new_with_label("取消");
+	cancel_button = gtk_button_new_with_label(_("Cancel"));
 	g_signal_connect(cancel_button, "clicked", G_CALLBACK(cancel_button_clicked), &shutdownType);
 	gtk_widget_show(cancel_button);
 	gtk_box_pack_start(GTK_BOX(hbox), cancel_button, TRUE, TRUE, 0);
diff -u exitx.orig//Makefile exitx/Makefile
--- exitx.orig//Makefile	2008-01-05 19:05:28.000000000 +0800
+++ exitx/Makefile	2010-05-09 21:33:26.000000000 +0800
@@ -1,5 +1,12 @@
+all:exitx exitx.po
 exitx: main.c halt.h hibernate.h logout.h reboot.h
-	gcc -O2 -Wall -o exitx main.c `pkg-config gtk+-2.0 --cflags` `pkg-config gtk+-2.0 --libs`
+	gcc -O2 -Wall -o exitx main.c `pkg-config gtk+-2.0 --cflags` `pkg-config gtk+-2.0 --libs` `pkg-config --cflags --libs glib2.0` -lX11
 
 indent: main.c
 	indent -npro -kr -i8 -ts8 -sob -l120 -ss -ncs -cp1 main.c
+exitx.po:*.c 
+	xgettext -k_ --language=C -o $@  $^ --from-code=utf-8
+	#xgettext -j -k_ -o $@  $^
+.PHONY:clean
+clean:
+	-rm -f exitx.po exitx
