--- ./drbd/Makefile	2011-07-07 13:55:39.000000000 +0300
+++ ./drbd/Makefile.new	2011-11-30 14:19:42.734829115 +0200
@@ -77,11 +77,6 @@
     endif
   endif
 
-  KDIR_Makefile_PATCHLEVEL = $(shell test -e $(KDIR)/Makefile && grep "^PATCHLEVEL = " $(KDIR)/Makefile | cut -d " " -f 3)
-  ifneq ($(findstring $(KDIR_Makefile_PATCHLEVEL),12345),)
-    $(error "won't compile with this kernel version")
-  endif
-
   .PHONY: drbd.o default all greeting clean kbuild install dep tags
 
   drbd.o: greeting kbuild
