--- interfacemplayer.cpp	2011-05-30 11:05:42.000000000 +0200
+++ interfacemplayer-fix.cpp	2011-12-06 18:47:32.555766310 +0100
@@ -25,10 +25,10 @@
  *
  *****************************************************************************/
 
-#define	RX_CATCH_MPLAYER_OUTPUT_STATUS		"(mplayer\\s+[^\\s]+.*MPlayer Team)"
+#define	RX_CATCH_MPLAYER_OUTPUT_STATUS		"(mplayer\\d*\\s+[^\\s]+.*MPlayer Team)"
 #define	RX_CATCH_MPLAYER_OUTPUT_HELP		"Available video output drivers:(.*)Available audio output drivers:(.*)"
 
-#define	RX_PARSE_MPLAYER_SIGNATURE			"(mplayer)\\s+([^\\s]+).*(MPlayer Team)"
+#define	RX_PARSE_MPLAYER_SIGNATURE			"(mplayer\\d*)\\s+([^\\s]+).*(MPlayer Team)"
 
 #define	RX_PARSE_MPLAYER_ID_DVD_DISCID		"ID_DVD_VOLUME_ID=(.*)"
 #define RX_PARSE_MPLAYER_ID_DVD_TITLES		"ID_DVD_TITLES=(\\d+)"
