--- a/k3dsdk/gil/boost/gil/extension/io/png_io_private.hpp	2010-04-18 07:51:06.000000000 +0200
+++ b/k3dsdk/gil/boost/gil/extension/io/png_io_private.hpp	2010-09-12 08:51:51.000000000 +0200
@@ -14,7 +14,8 @@
 /// \author Hailin Jin and Lubomir Bourdev \n
 ///         Adobe Systems Incorporated
 /// \date   2005-2007 \n Last updated September 24, 2006
-
+#define png_infopp_NULL (png_infopp)NULL
+#define int_p_NULL (int*)NULL
 #include <algorithm>
 #include <vector>
 #include <boost/static_assert.hpp>
