--- libXcm-0.4.2/configure
+++ libXcm-0.4.2/configure
@@ -76,7 +76,7 @@
 X_ADD_2="X11 Xfixes Xmu"
 LIBS_TEST="m"
 
-STRIPOPT="sed s/-O.//"
+STRIPOPT="cat"
 
 if [ $# -gt 0 ]; then
   prefix=""
+++ libXcm-0.4.2/configure_tests.sh
--- libXcm-0.4.2/configure_tests.sh
@@ -3,7 +3,7 @@
 set > testset.txt
 ERROR=0
 WARNING=0
-STRIPOPT='s/-O.// ; s/-isysroot [[:graph:]]*// ; s/-arch ppc// ; s/-arch i386//'
+STRIPOPT='s/-isysroot [[:graph:]]*// ; s/-arch ppc// ; s/-arch i386//'
 
 mkdir -p tests
 
--- libXcm-0.4.2/makefile.in
+++ libXcm-0.4.2/makefile.in
@@ -51,7 +51,6 @@
 
 DOKU =  AUTHORS \
         ChangeLog \
-        COPYING \
         README
 SUB_DOKU = \
 	docs/net-color-spec
