--- miniupnpc-1.8/setup.py	2012-05-23 16:50:10.000000000 +0800
+++ miniupnpc-1.8/setup.py	2012-05-23 16:50:10.000000000 +0800
@@ -10,9 +10,9 @@
 from distutils import sysconfig
 sysconfig.get_config_vars()["OPT"] = ''
 sysconfig.get_config_vars()["CFLAGS"] = ''
-setup(name="miniupnpc", version="1.7",
+setup(name="miniupnpc", version="1.8",
       ext_modules=[
 	         Extension(name="miniupnpc", sources=["miniupnpcmodule.c"],
-			           extra_objects=["libminiupnpc.a"])
+				extra_link_args=["-lminiupnpc"])
 			 ])
 
