--- miniupnpc-1.8/Makefile	2013-01-28 18:10:25.000000000 +0800
+++ miniupnpc-1.8/Makefile	2013-01-28 18:10:25.000000000 +0800
@@ -137,11 +137,11 @@
 everything:	all $(EXECUTABLES_ADDTESTS)
 
 pythonmodule:	$(LIBRARY) miniupnpcmodule.c setup.py
-	python setup.py build
+	python2 setup.py build
 	touch $@
 
 installpythonmodule:	pythonmodule
-	python setup.py install
+	python2 setup.py install
 
 pythonmodule3:	$(LIBRARY) miniupnpcmodule.c setup.py
 	python3 setup.py build
