diff -urN openvas-libraries-4.0.5.orig/misc/network.c openvas-libraries-4.0.5/misc/network.c
--- openvas-libraries-4.0.5.orig/misc/network.c	2011-06-01 15:38:37.000000000 +0200
+++ openvas-libraries-4.0.5/misc/network.c	2011-10-23 13:24:56.416944706 +0200
@@ -824,8 +824,6 @@
     }
 
   unblock_socket (fp->fd);
-  /* for non-blocking sockets, gnutls requires a 0 lowat value */
-  gnutls_transport_set_lowat (fp->tls_session, 0);
 
   gnutls_transport_set_ptr (fp->tls_session,
                             (gnutls_transport_ptr_t) GSIZE_TO_POINTER (fp->fd));
