Index: lib/src/Uncertainty/Distribution/ComposedDistribution.cxx
===================================================================
--- lib/src/Uncertainty/Distribution/ComposedDistribution.cxx	(révision 4031)
+++ lib/src/Uncertainty/Distribution/ComposedDistribution.cxx	(copie de travail)
@@ -788,11 +788,11 @@
   // Special case for the elliptical distribution case: linear transformation
   if (isElliptical())
   {
-    const TriangularMatrix inverseCholesky(getInverseCholesky());
+    const TriangularMatrix cholesky(copula_.getShapeMatrix().computeCholesky());
     const NumericalPoint mean(getMean());
     InverseIsoProbabilisticTransformation inverseTransform;
-    inverseTransform.setEvaluationImplementation(new InverseNatafEllipticalDistributionEvaluation(mean, inverseCholesky));
-    inverseTransform.setGradientImplementation(new InverseNatafEllipticalDistributionGradient(inverseCholesky));
+    inverseTransform.setEvaluationImplementation(new InverseNatafEllipticalDistributionEvaluation(mean, cholesky));
+    inverseTransform.setGradientImplementation(new InverseNatafEllipticalDistributionGradient(cholesky));
     inverseTransform.setHessianImplementation(new InverseNatafEllipticalDistributionHessian(dimension));
     inverseTransform.setParameters(parameters);
     return inverseTransform;
