diff --git a/autogen.sh b/autogen.sh
index 7424428..2bcc607 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -54,8 +54,8 @@ for extra in contrib/*/Makefile.core.def; do
   fi
 done
 
-python gentpl.py $UTIL_DEFS > Makefile.util.am
-python gentpl.py $CORE_DEFS > grub-core/Makefile.core.am
+python2 gentpl.py $UTIL_DEFS > Makefile.util.am
+python2 gentpl.py $CORE_DEFS > grub-core/Makefile.core.am
 
 for extra in contrib/*/Makefile.common; do
   if test -e "$extra"; then
