--- ggz-gtk-client-0.0.14.1/ggz-gtk/xtext.c	2008-01-05 20:03:36.000000000 +0100
+++ ggz-gtk-client-0.0.14.1.PATCHED/ggz-gtk/xtext.c	2010-08-04 01:52:36.446764446 +0200
@@ -720,7 +720,7 @@
 	xtext->overdraw = FALSE;
 	xtext->tint_red = xtext->tint_green = xtext->tint_blue = TINT_VALUE;
 
-	xtext->adj = (GtkAdjustment *) gtk_adjustment_new (0, 0, 1, 1, 1, 1);
+	xtext->adj = (GtkAdjustment *) gtk_adjustment_new (0, 0, 1, 1, 1, 0);
 	g_object_ref (G_OBJECT (xtext->adj));
 	gtk_object_sink ((GtkObject *) xtext->adj);
 
