--- src/client.c.orig	2014-05-10 16:09:00.932692351 +0200
+++ src/client.c	2014-05-10 16:11:18.348582846 +0200
@@ -981,8 +981,12 @@
         wc.width = c->width;
         wc.height = c->height;

+        if (FLAG_TEST (c->flags, CLIENT_FLAG_FULLSCREEN))
+        {
+            clientUpdateFullscreenSize (c);
+        }
         /* If client is maximized, we need to update its coordonates and size as well */
-        if (FLAG_TEST (c->flags, CLIENT_FLAG_MAXIMIZED))
+        else if (FLAG_TEST (c->flags, CLIENT_FLAG_MAXIMIZED))
         {
             GdkRectangle rect;
             myScreenFindMonitorAtPoint (screen_info,
@@ -3534,8 +3538,12 @@
             continue;
         }

+        if( FLAG_TEST (c->flags, CLIENT_FLAG_FULLSCREEN))
+        {
+            clientUpdateFullscreenSize (c);
+        }
         /* Recompute size and position of maximized windows */
-        if (FLAG_TEST (c->flags, CLIENT_FLAG_MAXIMIZED))
+        else if (FLAG_TEST (c->flags, CLIENT_FLAG_MAXIMIZED))
         {
             maximization_flags = c->flags & CLIENT_FLAG_MAXIMIZED;

@@ -3549,10 +3557,6 @@
             wc.height = c->height;
             clientConfigure (c, &wc, CWX | CWY | CWWidth | CWHeight, CFG_NOTIFY);
         }
-        else if (FLAG_TEST (c->flags, CLIENT_FLAG_FULLSCREEN))
-        {
-            clientUpdateFullscreenSize (c);
-        }
         else
         {
             configure_flags = CFG_CONSTRAINED | CFG_REQUEST;
