diff -Naur tp_smapi-0.40/Makefile tp_smapi-0.40.new/Makefile
--- tp_smapi-0.40/Makefile	2008-12-15 21:03:06.000000000 -0800
+++ tp_smapi-0.40.new/Makefile	2008-12-16 13:13:30.000000000 -0800
@@ -3,7 +3,7 @@
 X:=$(shell false)
 KVER        := $(shell uname -r)
 KBASE       := /lib/modules/$(KVER)
-KSRC        := $(KBASE)/source
+KSRC        := $(KBASE)/build
 KBUILD      := $(KBASE)/build
 MOD_DIR     := $(KBASE)/kernel
 PWD         := $(shell pwd)
@@ -87,7 +87,7 @@
 	rm -f $(MOD_DIR)/drivers/hwmon/hdaps.ko
 	rm -f $(MOD_DIR)/extra/hdaps.ko
 endif
-	$(MAKE) -C $(KBUILD) M=$(PWD) O=$(KBUILD) modules_install
+	$(MAKE) -C $(KBUILD) M=$(PWD) modules_install
 	depmod -a
 
 
