diff -ur src/pdns-gui.0.3.3/apps/frontend/modules/domain/actions/actions.class.php patched/apps/frontend/modules/domain/actions/actions.class.php
--- src/pdns-gui.0.3.3/apps/frontend/modules/domain/actions/actions.class.php	2010-04-25 13:03:55.000000000 +0200
+++ patched/apps/frontend/modules/domain/actions/actions.class.php	2010-12-30 10:52:00.000000000 +0100
@@ -272,7 +272,10 @@
           return false;
         }
         
-        if (!preg_match('/^[a-z]{1}[a-z0-9\.\-]+[a-z0-9]{1}$/',$data['name']) || strlen($data['name']) > 63)
+        //if (!preg_match('/^[a-z]{1}[a-z0-9\.\-]+[a-z0-9]{1}$/',$data['name']) || strlen($data['name']) > 63)
+        // validates domains based on (RFCs 952 and 1123 )
+        // (Author Remi Sabourin) all domains: ^([a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?\.)*[a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?$
+        if (!preg_match('/^([a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,6}$/',$data['name']) || strlen($data['name']) > 253)
         {
           $this->getRequest()->setError('record',"Row $i: invalid name (only letters, digits and hyphen allowed).");
           return false;
diff -ur src/pdns-gui.0.3.3/apps/frontend/modules/template/actions/actions.class.php patched/apps/frontend/modules/template/actions/actions.class.php
--- src/pdns-gui.0.3.3/apps/frontend/modules/template/actions/actions.class.php	2010-04-25 13:03:55.000000000 +0200
+++ patched/apps/frontend/modules/template/actions/actions.class.php	2010-12-30 10:41:00.000000000 +0100
@@ -267,7 +267,10 @@
         return false;
       }
       
-      if (!preg_match('/^[a-z]{1}[a-z0-9\.\-]+[a-z0-9]{1}$/',$data['name']) || strlen($data['name']) > 63)
+      //if (!preg_match('/^[a-z]{1}[a-z0-9\.\-]+[a-z0-9]{1}$/',$data['name']) || strlen($data['name']) > 63)
+      // original test is useless. my test maybe could be less strict
+      if (!preg_match('/^(([a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?\.)+(([a-zA-Z]{2,6})|(%DOMAIN%))|(%DOMAIN%))$/',$data['name']))
+      
       {
         $this->getRequest()->setError('record',"Row $i: invalid name (only letters, digits and hyphen allowed).");
         return false;
diff -ur src/pdns-gui.0.3.3/data/symfony/config/php.yml patched/data/symfony/config/php.yml
--- src/pdns-gui.0.3.3/data/symfony/config/php.yml	2009-09-25 12:50:40.000000000 +0200
+++ patched/data/symfony/config/php.yml	2010-12-30 11:25:31.000000000 +0100
@@ -4,8 +4,9 @@
   arg_separator.output:        |
     &amp;
 
-check:
-  zend.ze1_compatibility_mode: off
+//removed for php5.3 compatibility
+//check:
+//  zend.ze1_compatibility_mode: off
 
 warn:
   magic_quotes_gpc:            off
