From 0c800fce355928b9510492b55a92bf1dad46238c Mon Sep 17 00:00:00 2001
From: Thomas Haider <t.haider@deprecate.de>
Date: Fri, 3 May 2013 15:24:23 +0200
Subject: [PATCH] keep log_file setting in debug mode
 
---
bley | 3 ---
 1 file changed, 3 deletions(-)
 
diff --git a/bley b/bley
index 52ae1d9..c61e890 100755
--- a/bley
+++ b/bley
@@ -263,9 +263,6 @@ def bley_start():
 
     settings.exim_workaround  = config.getboolean('bley', 'exim_workaround')
 
-    if settings.debug:
-        settings.log_file = None
-
     if settings.log_file == 'syslog':
         import syslog
         syslog.openlog('bley', syslog.LOG_PID, syslog.LOG_MAIL)
-- 
1.8.2.2
