--- simpleviewergl-git/src/formats/formatgif.cpp	2015-01-30 08:47:03.000000000 +0500
+++ simpleviewergl-git/src/formats/formatgif.cpp.new	2015-01-30 08:49:40.000000000 +0500
@@ -29,7 +29,7 @@
     m_size = file.getSize();
     file.close();
 
-    GifFileType* gif_file = DGifOpenFileName(filename);
+    GifFileType* gif_file = DGifOpenFileName(filename, NULL);
     if(gif_file == 0)
     {
         std::cout << "Error Opening GIF image" << std::endl;
@@ -40,7 +40,7 @@
     if(res != GIF_OK || gif_file->ImageCount < 1)
     {
         std::cout << "Error Opening GIF image" << std::endl;
-        DGifCloseFile(gif_file);
+        DGifCloseFile(gif_file, NULL);
         return false;
     }
 
@@ -147,7 +147,7 @@
     //
     //	std::cout << "Record Type" << (int)recordType << std::endl;
 
-    DGifCloseFile(gif_file);
+    DGifCloseFile(gif_file, NULL);
 
     return true;
 }
