diff --git a/src/dwarftherapist.cpp b/src/dwarftherapist.cpp
index 7b88278..1401d20 100644
--- a/src/dwarftherapist.cpp
+++ b/src/dwarftherapist.cpp
@@ -117,7 +117,7 @@ void DwarfTherapist::setup_search_paths() {
     // Dwarf-Therapist/release/../share/game_data.ini
     paths << QString("%1/../share").arg(appdir);
     // /usr/bin/../share/dwarftherapist/game_data.ini
-    paths << QString("%1/../share/dwarftherapist").arg(appdir);
+    paths << QString("%1/../share/splintermind-attributes").arg(appdir);
     // cwd/game_data.ini
     paths << QString("%1").arg(working_dir);
     // cwd/share/game_data.ini
