--- configure.ac	2010-09-02 11:58:21.000000000 +0200
+++ configure.ac    2010-10-27 18:42:58.453333606 +0200
@@ -62,8 +62,8 @@
 
 dnl CUPS_LIBS='-lcups -lm'
 dnl CUPS_IMAGE_LIBS='-lcupsimage -lcups -ljpeg -lm'
-CUPS_LIBS="/opt/lsb/lib"$LSB_ARCH"/libcups.so /opt/lsb/lib"$LSB_ARCH"/libm.so"
-CUPS_IMAGE_LIBS="/opt/lsb/lib"$LSB_ARCH"/libcupsimage.so /opt/lsb/lib"$LSB_ARCH"/libcups.so /opt/lsb/lib"$LSB_ARCH"/libjpeg.so"
+CUPS_LIBS="/usr/lib/libcups.so /usr/lib/libm.so"
+CUPS_IMAGE_LIBS="/usr/lib/libcupsimage.so /usr/lib/libcups.so /usr/lib/libjpeg.so"
 
 AC_SUBST(CUPS_LIBS)
 AC_SUBST(CUPS_IMAGE_LIBS)
