diff --git a/Makefile b/Makefile
index 7c86f39..cb1ac50 100644
--- a/Makefile
+++ b/Makefile
@@ -25,9 +25,9 @@
 #** -LICENSE-END-
 
 CXX=g++
-SDK_PATH=../../include
+SDK_PATH=/usr/src/decklink-sdk
 CFLAGS=-Wno-multichar -I $(SDK_PATH) -fno-rtti -D__STDC_CONSTANT_MACROS -g
-LDFLAGS=-lm -ldl -lpthread `pkg-config --libs libavformat`
+LDFLAGS=-lm -ldl -lpthread -lavformat -lavutil -lavcodec
 
 all: bmdcapture bmdplay bmdgenlock
 
diff --git a/bmdcapture.cpp b/bmdcapture.cpp
index 2f86174..dad25d9 100644
--- a/bmdcapture.cpp
+++ b/bmdcapture.cpp
@@ -26,6 +26,8 @@
 ** -LICENSE-END-
 */
 
+#define AVIO_FLAG_WRITE 2
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <pthread.h>
