diff -ru texstudio/qcodeedit/lib/document/qdocumentcommand.cpp texstudio-patched/qcodeedit/lib/document/qdocumentcommand.cpp
--- texstudio/qcodeedit/lib/document/qdocumentcommand.cpp	2012-05-07 23:36:09.949491283 +0200
+++ texstudio-patched/qcodeedit/lib/document/qdocumentcommand.cpp	2012-05-07 23:38:48.861238301 +0200
@@ -22,6 +22,8 @@
 
 #include "qdocument_p.h"
 
+#include "stdint.h"
+
 /*!
 	\ingroup document
 	@{
diff -ru texstudio/qtsingleapplication/qtlocalpeer.cpp texstudio-patched/qtsingleapplication/qtlocalpeer.cpp
--- texstudio/qtsingleapplication/qtlocalpeer.cpp	2012-05-07 23:36:09.889493139 +0200
+++ texstudio-patched/qtsingleapplication/qtlocalpeer.cpp	2012-05-07 23:40:47.240906793 +0200
@@ -60,6 +60,7 @@
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
diff -ru texstudio/tests/qsearchreplacepanel_t.cpp texstudio-patched/tests/qsearchreplacepanel_t.cpp
--- texstudio/tests/qsearchreplacepanel_t.cpp	2012-05-07 23:36:09.889493139 +0200
+++ texstudio-patched/tests/qsearchreplacepanel_t.cpp	2012-05-07 23:40:05.192208554 +0200
@@ -213,9 +213,9 @@
 				widget->cFind->lineEdit()->cursorForward(false,search.length());	
 				QTest::keyClick(widget->cFind->lineEdit(),search[search.length()-1].toLatin1());
 			}
-			QDocumentCursor s=ed->cursor().selectionStart();
-			QEQUAL2(s.lineNumber(), cy, search+" "+ed->cursor().selectedText()+"  "+QString::number(loop));
-			QEQUAL2(s.columnNumber(), cx, search+" "+ed->cursor().selectedText());
+			QDocumentCursor c=ed->cursor().selectionStart();
+			QEQUAL2(c.lineNumber(), cy, search+" "+ed->cursor().selectedText()+"  "+QString::number(loop));
+			QEQUAL2(c.columnNumber(), cx, search+" "+ed->cursor().selectedText());
 			QEQUAL2(ed->cursor().selectedText(), res, search+" "+ed->cursor().selectedText());
 			//searching shouldn't change highlighted selection
 			QCEMULTIEQUAL(getHighlightedSelection(ed), panel->getSearchScope(), ed->document()->cursor(scopey1,scopex1,scopey2,scopex2));			
