--- app/actions/tool-options-commands.c.orig	2010-02-08 13:04:02.000000000 +0100
+++ app/actions/tool-options-commands.c	2010-02-08 13:09:08.000000000 +0100
@@ -121,14 +121,14 @@
   if (options)
     {
       GdkModifierType state;
-      guint flags = GIMP_CONFIG_PARAM_DONT_RESTORE;
+      guint flags = 0;
 
       if (gtk_get_current_event_state (&state))
         {
           if (state & GDK_SHIFT_MASK && ! (state & GDK_CONTROL_MASK))
-            flags = 0;
+            flags = GIMP_CONFIG_PARAM_DONT_RESTORE;
           else if (state & GDK_CONTROL_MASK && ! (state & GDK_SHIFT_MASK))
-            flags |= GIMP_CONFIG_PARAM_DONT_RESTORE_SECONDARY;
+            flags = GIMP_CONFIG_PARAM_DONT_RESTORE | GIMP_CONFIG_PARAM_DONT_RESTORE_SECONDARY;
         }
 
       tool_options_sync (G_OBJECT (options),
