--- app/dialogs/about-dialog.c.orig	2010-02-08 13:15:37.000000000 +0100
+++ app/dialogs/about-dialog.c	2010-02-08 13:17:52.000000000 +0100
@@ -609,7 +609,7 @@
   gtk_widget_show (label);
 #endif
 
-  label = gtk_label_new (_("gimp-painter- (release 20090715)"));
+  label = gtk_label_new (_("Gimp Studio PKGVER"));
   gtk_box_pack_start (GTK_BOX (vbox), label, FALSE, FALSE, 0);
   gtk_box_reorder_child (GTK_BOX (vbox), label, 2);
   gtk_widget_show (label);
