--- app/gui/splash.c.orig	2010-02-09 20:01:25.000000000 +0100
+++ app/gui/splash.c	2010-02-09 20:41:11.000000000 +0100
@@ -37,6 +37,8 @@
 
 /*  #define STARTUP_TIMER 1  */
 
+static const gchar const *version_text = "Version PKGVER (Gimp GIMPVER)";
+static const gint version_correction   = VERT_CORRECTION;
 
 typedef struct
 {
@@ -46,6 +48,9 @@
   gint            height;
   GtkWidget      *progress;
   GdkGC          *gc;
+  PangoLayout    *version;
+  gint            version_x;
+  gint            version_y;
   PangoLayout    *upper;
   gint            upper_x;
   gint            upper_y;
@@ -157,6 +162,7 @@
   gtk_widget_set_size_request (splash->area, splash->width, splash->height);
 
   /*  create the pango layouts  */
+  splash->version = gtk_widget_create_pango_layout (splash->area, "");
   splash->upper = gtk_widget_create_pango_layout (splash->area, "");
   splash->lower = gtk_widget_create_pango_layout (splash->area, "");
   gimp_pango_layout_set_scale (splash->lower, PANGO_SCALE_SMALL);
@@ -212,6 +218,7 @@
   gtk_widget_destroy (splash->window);
 
   g_object_unref (splash->gc);
+  g_object_unref (splash->version);
   g_object_unref (splash->upper);
   g_object_unref (splash->lower);
 
@@ -266,6 +273,9 @@
   gdk_gc_set_clip_region (splash->gc, event->region);
 
   gdk_draw_layout (widget->window, splash->gc,
+                   splash->version_x, splash->version_y, splash->version);
+
+  gdk_draw_layout (widget->window, splash->gc,
                    splash->upper_x, splash->upper_y, splash->upper);
 
   gdk_draw_layout (widget->window, splash->gc,
@@ -284,6 +294,20 @@
   PangoRectangle  ink;
   PangoRectangle  logical;
 
+  pango_layout_get_pixel_extents (splash->version, &ink, NULL);
+
+  if (area)
+    splash_rectangle_union (area, &ink, splash->version_x, splash->version_y);
+
+  pango_layout_set_text (splash->version, version_text, -1);
+  pango_layout_get_pixel_extents (splash->version, &ink, &logical);
+
+  splash->version_x = (splash->width - logical.width) / 2;
+  splash->version_y = splash->height/2 + version_correction;
+
+  if (area)
+    splash_rectangle_union (area, &ink, splash->version_x, splash->version_y);
+
   if (text1)
     {
       pango_layout_get_pixel_extents (splash->upper, &ink, NULL);
