--- themes/Default/images/Makefile.in.orig	2010-02-05 14:06:40.000000000 +0100
+++ themes/Default/images/Makefile.in	2010-02-05 14:08:22.000000000 +0100
@@ -707,6 +707,8 @@
 	tools/stock-tool-heal-22.png			\
 	tools/stock-tool-ink-16.png			\
 	tools/stock-tool-ink-22.png			\
+	tools/stock-tool-gink-16.png			\
+	tools/stock-tool-gink-22.png			\
 	tools/stock-tool-iscissors-16.png		\
 	tools/stock-tool-iscissors-22.png		\
 	tools/stock-tool-levels-16.png			\
@@ -717,6 +719,8 @@
 	tools/stock-tool-move-22.png			\
 	tools/stock-tool-paintbrush-16.png		\
 	tools/stock-tool-paintbrush-22.png		\
+	tools/stock-tool-mixbrush-16.png		\
+	tools/stock-tool-mixbrush-22.png		\
 	tools/stock-tool-path-16.png			\
 	tools/stock-tool-path-22.png			\
 	tools/stock-tool-pencil-16.png			\
--- libgimpwidgets/gimpstock.h.orig	2010-02-05 14:19:44.000000000 +0100
+++ libgimpwidgets/gimpstock.h	2010-02-05 14:20:43.000000000 +0100
@@ -118,12 +118,14 @@
 #define GIMP_STOCK_TOOL_FREE_SELECT         "gimp-tool-free-select"
 #define GIMP_STOCK_TOOL_FOREGROUND_SELECT   "gimp-tool-foreground-select"
 #define GIMP_STOCK_TOOL_FUZZY_SELECT        "gimp-tool-fuzzy-select"
+#define GIMP_STOCK_TOOL_GINK                "gimp-tool-gink"
 #define GIMP_STOCK_TOOL_HEAL                "gimp-tool-heal"
 #define GIMP_STOCK_TOOL_HUE_SATURATION      "gimp-tool-hue-saturation"
 #define GIMP_STOCK_TOOL_INK                 "gimp-tool-ink"
 #define GIMP_STOCK_TOOL_ISCISSORS           "gimp-tool-iscissors"
 #define GIMP_STOCK_TOOL_LEVELS              "gimp-tool-levels"
 #define GIMP_STOCK_TOOL_MEASURE             "gimp-tool-measure"
+#define GIMP_STOCK_TOOL_MIXBRUSH            "gimp-tool-mixbrush"
 #define GIMP_STOCK_TOOL_MOVE                "gimp-tool-move"
 #define GIMP_STOCK_TOOL_PAINTBRUSH          "gimp-tool-paintbrush"
 #define GIMP_STOCK_TOOL_PATH                "gimp-tool-path"
--- libgimpwidgets/gimpstock.c.orig	2010-02-05 14:16:01.000000000 +0100
+++ libgimpwidgets/gimpstock.c	2010-02-05 14:19:15.000000000 +0100
@@ -302,12 +302,14 @@
   { GIMP_STOCK_TOOL_FREE_SELECT,         NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_FOREGROUND_SELECT,   NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_FUZZY_SELECT,        NULL,        0, 0, LIBGIMP_DOMAIN },
-  { GIMP_STOCK_TOOL_HUE_SATURATION,      NULL,        0, 0, LIBGIMP_DOMAIN },
+  { GIMP_STOCK_TOOL_GINK,                NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_HEAL,                NULL,        0, 0, LIBGIMP_DOMAIN },
+  { GIMP_STOCK_TOOL_HUE_SATURATION,      NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_INK,                 NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_ISCISSORS,           NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_LEVELS,              NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_MEASURE,             NULL,        0, 0, LIBGIMP_DOMAIN },
+  { GIMP_STOCK_TOOL_MIXBRUSH,            NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_MOVE,                NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_PAINTBRUSH,          NULL,        0, 0, LIBGIMP_DOMAIN },
   { GIMP_STOCK_TOOL_PATH,                NULL,        0, 0, LIBGIMP_DOMAIN },
@@ -452,12 +454,14 @@
   { GIMP_STOCK_TOOL_FREE_SELECT,         stock_tool_free_select_22         },
   { GIMP_STOCK_TOOL_FOREGROUND_SELECT,   stock_tool_foreground_select_22   },
   { GIMP_STOCK_TOOL_FUZZY_SELECT,        stock_tool_fuzzy_select_22        },
+  { GIMP_STOCK_TOOL_GINK,                stock_tool_gink_22                },
   { GIMP_STOCK_TOOL_HEAL,                stock_tool_heal_22                },
   { GIMP_STOCK_TOOL_HUE_SATURATION,      stock_tool_hue_saturation_22      },
   { GIMP_STOCK_TOOL_INK,                 stock_tool_ink_22                 },
   { GIMP_STOCK_TOOL_ISCISSORS,           stock_tool_iscissors_22           },
   { GIMP_STOCK_TOOL_LEVELS,              stock_tool_levels_22              },
   { GIMP_STOCK_TOOL_MEASURE,             stock_tool_measure_22             },
+  { GIMP_STOCK_TOOL_MIXBRUSH,            stock_tool_mixbrush_22            },
   { GIMP_STOCK_TOOL_MOVE,                stock_tool_move_22                },
   { GIMP_STOCK_TOOL_PAINTBRUSH,          stock_tool_paintbrush_22          },
   { GIMP_STOCK_TOOL_PATH,                stock_tool_path_22                },
@@ -614,12 +618,14 @@
   { GIMP_STOCK_TOOL_FREE_SELECT,         stock_tool_free_select_16         },
   { GIMP_STOCK_TOOL_FOREGROUND_SELECT,   stock_tool_foreground_select_16   },
   { GIMP_STOCK_TOOL_FUZZY_SELECT,        stock_tool_fuzzy_select_16        },
+  { GIMP_STOCK_TOOL_GINK,                stock_tool_gink_16                },
   { GIMP_STOCK_TOOL_HEAL,                stock_tool_heal_16                },
   { GIMP_STOCK_TOOL_HUE_SATURATION,      stock_tool_hue_saturation_16      },
   { GIMP_STOCK_TOOL_INK,                 stock_tool_ink_16                 },
   { GIMP_STOCK_TOOL_ISCISSORS,           stock_tool_iscissors_16           },
   { GIMP_STOCK_TOOL_LEVELS,              stock_tool_levels_16              },
   { GIMP_STOCK_TOOL_MEASURE,             stock_tool_measure_16             },
+  { GIMP_STOCK_TOOL_MIXBRUSH,            stock_tool_mixbrush_16            },
   { GIMP_STOCK_TOOL_MOVE,                stock_tool_move_16                },
   { GIMP_STOCK_TOOL_PAINTBRUSH,          stock_tool_paintbrush_16          },
   { GIMP_STOCK_TOOL_PATH,                stock_tool_path_16                },
--- app/tools/gimpinktool2.c.orig	2010-02-05 14:24:07.000000000 +0100
+++ app/tools/gimpinktool2.c	2010-02-05 14:23:22.000000000 +0100
@@ -56,7 +56,7 @@
                 _("Draw in G-pen -2-"),
                 N_("G-pen"), "G",
                 NULL, GIMP_HELP_TOOL_INK,
-                GIMP_STOCK_TOOL_INK,
+                GIMP_STOCK_TOOL_GINK,
                 data);
 }
 
--- app/tools/gimpmixbrushtool.c.orig	2010-02-05 14:24:47.000000000 +0100
+++ app/tools/gimpmixbrushtool.c	2010-02-05 14:23:04.000000000 +0100
@@ -56,7 +56,7 @@
                 _("Mixbrush Tool: Brush with mixing color feature"),
                 N_("_Mixbrush"), "W",
                 NULL, GIMP_HELP_TOOL_PAINTBRUSH,
-                GIMP_STOCK_TOOL_PAINTBRUSH,
+                GIMP_STOCK_TOOL_MIXBRUSH,
                 data);
 }
 
