--- a/caffeine/core.py	2014-01-29 20:52:36.218279030 +0100
+++ b/caffeine/core.py	2014-01-29 20:53:23.666046037 +0100
@@ -127,7 +127,7 @@
         try:
             ## look for files opened by flashplayer that begin with 'Flash'
 
-            output = commands.getoutput("python flash_detect.py")
+            output = commands.getoutput("python2 flash_detect.py")
 
             if output == "1":
                 raise escape
 
