diff -r 6c97c487cf50 indra/cmake/FindOculusSDK.cmake
--- a/indra/cmake/FindOculusSDK.cmake	Wed Dec 10 20:30:50 2014 -0800
+++ b/indra/cmake/FindOculusSDK.cmake	Sun Dec 28 08:43:47 2014 -0600
@@ -9,16 +9,27 @@
 FIND_PATH(OCULUS_INCLUDE_DIR
   NAMES
     OVR.h
-    OVRVersion.h
+    OVR_Version.h
   PATHS
     /usr/include
     /usr/local/include
     /opt/local/include
+    /usr/include/ovr-0.4.4/LibOVR/Include
   )
 
+FIND_PATH(OCULUS_SRC_DIR
+  NAMES
+    OVR_CAPI_GL.h
+  PATHS
+    /usr/include
+    /usr/local/include
+    /opt/local/include
+    /usr/include/ovr-0.4.4/LibOVR/Src
+)
+
 FIND_PATH(OCULUS_LIBRARY_DIR
   NAMES
-    libovr.a
+    libOculusVR.a
   PATHS
     /usr/lib
     /usr/local/lib
@@ -27,7 +38,7 @@
 
 FIND_LIBRARY(OCULUS_LIBRARY
   NAMES
-    ovr
+    OculusVR
   PATHS
     /usr/lib
     /usr/local/lib
@@ -40,6 +51,7 @@
 
 SET(OCULUS_INCLUDE_DIRS
   ${OCULUS_INCLUDE_DIR}
+  ${OCULUS_SRC_DIR}
   )
 
 if (OCULUS_FOUND)
@@ -52,12 +64,12 @@
 if (OCULUS_INCLUDE_DIRS AND OCULUS_LIBRARY)
   SET(OCULUS_FOUND "YES")
 else (OCULUS_INCLUDE_DIRS AND OCULUS_LIBRARY)
-  SET(OCULUS_FOUND "NO"
+  SET(OCULUS_FOUND "NO")
 endif (OCULUS_INCLUDE_DIRS AND OCULUS_LIBRARY)
 
 if (OCULUS_FOUND)
   if (NOT OCULUS_FIND_QUIETLY)
-    MESSAGE(STATUS "Found OculusSDK: ${OCULUS_LIBRARY}")
+    MESSAGE(STATUS "Found OculusSDK: ${OCULUS_INCLUDE_DIRS}")
   endif (NOT OCULUS_FIND_QUIETLY)
 else (OCULUS_FOUND)
   if (OCULUS_FIND_REQUIRED)
diff -r 6c97c487cf50 indra/cmake/OculusSDK.cmake
--- a/indra/cmake/OculusSDK.cmake	Wed Dec 10 20:30:50 2014 -0800
+++ b/indra/cmake/OculusSDK.cmake	Sun Dec 28 08:43:47 2014 -0600
@@ -2,25 +2,28 @@
 
 include(Prebuilt)
 
-set(OCULUSSDK_FIND_QUIETLY ON)
+#set(OCULUSSDK_FIND_QUIETLY ON)
 set(OCULUSSDK_FIND_REQUIRED ON)
 
-if (STANDALONE)
-  include(FindOculusSDK)
-else (STANDALONE)
-  use_prebuilt_binary(OVR)
-  if (WINDOWS)
-    set(OCULUS_LIBRARY
-      debug libovrd.lib
-      optimized libovr.lib)
-  elseif (DARWIN)
-    set(OCULUS_LIBRARY ovr)
-  elseif (LINUX)
-    set(OCULUS_LIBRARY ovr)
-  endif (WINDOWS)
-  set(OCULUS_INCLUDE_DIRS 
-    "${LIBS_PREBUILT_DIR}/include"
-	"${LIBS_PREBUILT_DIR}/include/OVR"
-	"${LIBS_PREBUILT_DIR}/include/OVR/Kernel"
-	"${LIBS_PREBUILT_DIR}/include/OVR/Util")
-endif (STANDALONE)
+include(FindOculusSDK)
+
+#if (STANDALONE)
+#  include(FindOculusSDK)
+#else (STANDALONE)
+#  use_prebuilt_binary(OVR)
+#  if (WINDOWS)
+#    set(OCULUS_LIBRARY
+#      debug libovrd.lib
+#      optimized libovr.lib)
+#  elseif (DARWIN)
+#    set(OCULUS_LIBRARY ovr)
+#  elseif (LINUX)
+#    #set(OCULUS_LIBRARY ovr)
+#    set(OCULUS_LIBRARY OculusVR)
+#  endif (WINDOWS)
+#  set(OCULUS_INCLUDE_DIRS 
+#    "${LIBS_PREBUILT_DIR}/include"
+#	"${LIBS_PREBUILT_DIR}/include/OVR"
+#	"${LIBS_PREBUILT_DIR}/include/OVR/Kernel"
+#	"${LIBS_PREBUILT_DIR}/include/OVR/Util")
+#endif (STANDALONE)
diff -r 6c97c487cf50 indra/cmake/Python.cmake
--- a/indra/cmake/Python.cmake	Wed Dec 10 20:30:50 2014 -0800
+++ b/indra/cmake/Python.cmake	Sun Dec 28 08:43:47 2014 -0600
@@ -35,7 +35,7 @@
 
   string(REPLACE ":" ";" PATH_LIST "$ENV{PATH}")
   find_program(PYTHON_EXECUTABLE
-    NAMES python python25 python24 python23
+    NAMES python2 python25 python24 python23
     NO_DEFAULT_PATH # Avoid searching non-standard locations first
     PATHS
     /bin
diff -r 6c97c487cf50 indra/llcommon/llmemory.h
--- a/indra/llcommon/llmemory.h	Wed Dec 10 20:30:50 2014 -0800
+++ b/indra/llcommon/llmemory.h	Sun Dec 28 08:43:47 2014 -0600
@@ -26,6 +26,7 @@
 #ifndef LLMEMORY_H
 #define LLMEMORY_H
 
+#include <stdint.h>
 #include "linden_common.h"
 
 class LLMutex ;
diff -r 6c97c487cf50 indra/llwindow/llwindow.h
--- a/indra/llwindow/llwindow.h	Wed Dec 10 20:30:50 2014 -0800
+++ b/indra/llwindow/llwindow.h	Sun Dec 28 08:43:47 2014 -0600
@@ -223,7 +223,7 @@
 // </FS:ND>
 
 	static BOOL getDisplayResolution(S32 &width, S32 &height, S32 &bits, S32 &refresh);  // <CV:David>
-	virtual void* getHwnd() = 0; // <CV:David>
+	void* getHwnd(); // <CV:David>
 };
 
 
diff -r 6c97c487cf50 indra/llwindow/llwindowsdl.cpp
--- a/indra/llwindow/llwindowsdl.cpp	Wed Dec 10 20:30:50 2014 -0800
+++ b/indra/llwindow/llwindowsdl.cpp	Sun Dec 28 08:43:47 2014 -0600
@@ -195,7 +195,7 @@
 			 BOOL fullscreen, BOOL clearBg,
 			 BOOL disable_vsync, BOOL use_gl,
 			 BOOL ignore_pixel_depth, U32 fsaa_samples, U32 output_type)
-	: LLWindow(callbacks, fullscreen, flags),
+	: LLWindow(callbacks, fullscreen, flags, output_type),
 	  Lock_Display(NULL),
 	  Unlock_Display(NULL), mGamma(1.0f)
 {
@@ -1127,6 +1127,11 @@
 	return convertCoords(screen_pos, position);
 }
 
+ECursorType LLWindowSDL::getCursor() const
+{
+    return mCurrentCursor;
+}
+
 
 F32 LLWindowSDL::getNativeAspectRatio()
 {
diff -r 6c97c487cf50 indra/newview/llviewerdisplay.cpp
--- a/indra/newview/llviewerdisplay.cpp	Wed Dec 10 20:30:50 2014 -0800
+++ b/indra/newview/llviewerdisplay.cpp	Sun Dec 28 08:43:47 2014 -0600
@@ -83,6 +83,10 @@
 // [/RLVa:KB]
 #include "llviewermenu.h"  // <CV:David> DJRTODO: Temporarily needed for going fullscreen with DK2 in extended mode
 
+#ifdef LL_LINUX
+#include "llwindowsdl.h"
+#endif
+
 extern LLPointer<LLViewerTexture> gStartTexture;
 extern bool gShiftFrame;
 
@@ -2035,6 +2039,8 @@
 
 		// Optional according to pop-up text; OculusWorldDemo doesn't use it ...
 		//gRiftConfig.OGL.DC = GetDC(window);
+#elif LL_LINUX
+		gRiftConfig.OGL.Disp = glXGetCurrentDisplay();
 #endif
 
 		calculateRiftDistortionCaps();
diff -r 6c97c487cf50 indra/newview/viewer_manifest.py
--- a/indra/newview/viewer_manifest.py	Wed Dec 10 20:30:50 2014 -0800
+++ b/indra/newview/viewer_manifest.py	Sun Dec 28 08:43:47 2014 -0600
@@ -1320,7 +1320,7 @@
     def copy_finish(self):
         # Force executable permissions to be set for scripts
         # see CHOP-223 and http://mercurial.selenic.com/bts/issue1802
-        for script in 'firestorm', 'bin/update_install':
+        for script in 'casviewer', 'bin/update_install':
             self.run_command("chmod +x %r" % os.path.join(self.get_dst_prefix(), script))
 
     def package_finish(self):
