--- byobu-5.20/usr/bin/byobu-config.orig        2012-07-08 15:43:32.460101372 +0100
+++ byobu-5.20/usr/bin/byobu-config     2012-07-08 15:43:48.926768669 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 #    byobu-config
 #    Copyright (C) 2008 Canonical Ltd.
 --- byobu-5.20/usr/bin/byobu-select-session.orig        2012-07-08 15:44:23.400103315 +0100
+++ byobu-5.20/usr/bin/byobu-select-session     2012-07-08 15:44:27.833436814 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 #    byobu-select-session
 #    Copyright (C) 2010 Canonical Ltd.
 --- byobu-5.20/etc/byobu/socketdir.orig 2012-07-08 15:49:39.093448671 +0100
+++ byobu-5.20/etc/byobu/socketdir      2012-07-08 15:50:23.553450363 +0100
@@ -8,4 +8,4 @@
 #  * the variable name is SOCKETDIR
 #  * there is no space around the "="
 #  * and that the path value is quoted
-SOCKETDIR="/var/run/screen"
+SOCKETDIR="/tmp/screens"
