diff -Naur byobu-3.5.old/etc/byobu/socketdir byobu-3.5/etc/byobu/socketdir
--- byobu-3.5.old/etc/byobu/socketdir	2009-12-17 15:32:04.000000000 +0100
+++ byobu-3.5/etc/byobu/socketdir	2009-12-27 02:57:00.000000000 +0100
@@ -7,4 +7,4 @@
 #  * the variable name is SOCKETDIR
 #  * there is no space around the "="
 #  * and that the path value is quoted
-SOCKETDIR="/var/run/screen"
+SOCKETDIR="/tmp/screens"
diff -Naur byobu-3.5.old/usr/lib/byobu/wifi_quality byobu-3.5/usr/lib/byobu/wifi_quality
--- byobu-3.5.old/usr/lib/byobu/wifi_quality	2009-12-17 15:32:04.000000000 +0100
+++ byobu-3.5/usr/lib/byobu/wifi_quality	2009-12-27 02:57:00.000000000 +0100
@@ -21,11 +21,11 @@
 color 2>/dev/null || color() { true; }
 
 if [ "$1" = "--detail" ]; then
-	/sbin/iwconfig 2>/dev/null
+	/usr/sbin/iwconfig 2>/dev/null
 	exit 0
 fi
 
-iwconfig=`/sbin/iwconfig 2>/dev/null`
+iwconfig=`/usr/sbin/iwconfig 2>/dev/null`
 bitrate=`echo "$iwconfig" | grep "Bit Rate." | sed -e "s/^.*Bit Rate.//" -e "s/ .*$//g"`
 [ -z "$bitrate" ] && exit 0
 quality=`echo "$iwconfig" | grep "Link Quality." | sed -e "s/^.*Link Quality.//" -e "s/ .*$//g" | awk -F/ '{printf "%.0f", 100*$1/$2}'`
diff -Naur byobu-3.5.old/usr/bin/byobu-config byobu-3.5/usr/bin/byobu-config
--- byobu-3.5.old/usr/bin/byobu-config	2010-07-28 20:25:31.000000000 +0100
+++ byobu-3.5/usr/bin/byobu-config	2010-10-20 20:48:48.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
 #
 #    byobu-config
 #    Copyright (C) 2008 Canonical Ltd.
@@ -59,7 +59,7 @@
 
 # Command presets for windows creation
 cmd=(   ("System activity", "top", "top"),
-        ("System log", "log", "watch -n 10 tail -n 5 /var/log/syslog /var/log/auth.log /var/log/dmesg"),
+        ("System log", "log", "watch -n 10 tail -n 5 /var/log/syslog.log /var/log/auth.log /var/log/dmesg.log"),
         ("Disk and ram usage", "mem", 'watch -n 30 "df -h; echo ""; free -mt"'))
 
 
diff -Naur byobu-3.5.old/usr/bin/byobu-select-session byobu-3.5/usr/bin/byobu-select-session
--- byobu-3.5.old/usr/bin/byobu-select-session	2010-07-28 20:25:31.000000000 +0100
+++ byobu-3.5/usr/bin/byobu-select-session	2010-10-20 20:46:47.000000000 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python2
 #
 #    byobu-select-session
 #    Copyright (C) 2010 Canonical Ltd.
