diff -Naur byobu-4.17.orig/usr/bin/byobu-config byobu-4.17/usr/bin/byobu-config
--- byobu-4.17.orig/usr/bin/byobu-config	2011-06-29 16:31:53.530926728 +0200
+++ byobu-4.17/usr/bin/byobu-config	2011-06-29 16:32:11.030831922 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 #    byobu-config
 #    Copyright (C) 2008 Canonical Ltd.
diff -Naur byobu-4.17.orig/usr/bin/byobu-select-session byobu-4.17/usr/bin/byobu-select-session
--- byobu-4.17.orig/usr/bin/byobu-select-session	2011-06-29 16:31:53.534260043 +0200
+++ byobu-4.17/usr/bin/byobu-select-session	2011-06-29 16:32:25.527420054 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 #
 #    byobu-select-session
 #    Copyright (C) 2010 Canonical Ltd.
