diff -aur gwakeonlan-0.5.1.original/setup.py gwakeonlan-0.5.1/setup.py
--- gwakeonlan-0.5.1.original/setup.py	2012-08-12 17:31:15.090187502 +0000
+++ gwakeonlan-0.5.1/setup.py	2012-08-12 17:31:31.670188079 +0000
@@ -73,7 +73,7 @@
       data_files=[
                   ('share/applications', ['data/gwakeonlan.desktop']),
                   ('share/man/man1', ['man/gwakeonlan.1']),
-                  ('share/doc/gwakeonlan', ['doc/README', 'doc/changelog', 'doc/translators']),
+                  ('share/doc/gwakeonlan', ['doc/README', 'doc/changelog', 'doc/translators', 'doc/copyright']),
                   ('share/gwakeonlan', ['data/gwakeonlan.glade', 'data/gwakeonlan.svg']),
                  ],
       cmdclass={'install_data': InstallData}
