
diff --git a/Makefile.in b/Makefile.in
--- a/mozilla/Makefile.in
+++ b/mozilla/Makefile.in
@@ -69,18 +69,16 @@ config.status: $(topsrcdir)/configure
 	@echo "Please rerun configure."
 	@echo "To ignore this message, touch 'config.status' in the build directory,"
 	@echo "but your build might not succeed."
 	@exit 1
 
 # Build pseudo-external modules first when export is explicitly called
 export::
 	$(RM) -r $(DIST)/sdk
-	$(MAKE) -C config export
-	$(MAKE) tier_nspr
 
 ifdef ENABLE_TESTS
 # Additional makefile targets to call automated test suites
 include $(topsrcdir)/testing/testsuite-targets.mk
 endif
 
 # Hacky way for precompile tier to bypass default tier traversal mechanism.
 TIER_precompile_CUSTOM := 1
