--- Makefile.orig	2007-09-12 11:31:56.000000000 +0200
+++ Makefile	2009-06-29 23:57:03.000000000 +0200
@@ -18,15 +18,11 @@
 install : install-package install-bin install-etc install-log install-man
 
 install-package : all
-	[ -d "$(PACKAGEDIR)" ] || mkdir -p "$(PACKAGEDIR)"
-	[ -d "$(PACKAGEDIR)/sbin" ] || mkdir -p "$(PACKAGEDIR)/sbin"
-	[ -d "$(PACKAGEDIR)/etc" ] || mkdir -p "$(PACKAGEDIR)/etc"
-	[ -d "$(PACKAGEDIR)/man/man5" ] || mkdir -p "$(PACKAGEDIR)/man/man5"
-	[ -d "$(PACKAGEDIR)/man/man8" ] || mkdir -p "$(PACKAGEDIR)/man/man8"
-	cp $(TARGETS) "$(PACKAGEDIR)/sbin"
-	cp conf/* "$(PACKAGEDIR)/etc"
-	cp muhttpd.conf.5 "$(PACKAGEDIR)/man/man5"
-	cp muhttpd.8 "$(PACKAGEDIR)/man/man8"
+	install -Dps $(TARGETS) "$(PREFIX)/sbin/$(TARGETS)"
+	install -Dp -m644 conf/muhttpd.conf "$(PREFIX)/etc/$(TARGETS)/muhttpd.conf"
+	install -Dp -m644 conf/types.conf "$(PREFIX)/etc/$(TARGETS)/types.conf"
+	install -Dp -m644 muhttpd.conf.5 "$(PREFIX)/usr/share/man/man5/muhttpd.conf.5"
+	install -Dp -m644 muhttpd.8 "$(PREFIX)/usr/share/man/man8/muhttpd.8"
 
 install-bin :
 	[ -d "$(SBINDIR)" ] || mkdir -p "$(SBINDIR)"
