diff -urN src.orig/archup.c src/archup.c
--- src.orig/archup.c	2010-04-01 19:34:29.000000000 +0200
+++ src/archup.c	2012-02-06 22:45:50.300669859 +0100
@@ -147,7 +147,7 @@
         GError *error = NULL;
 
         /* Those are needed for the output. */
-	char *output_string=malloc(23); 
+	char *output_string=malloc(24); 
 	sprintf(output_string,"There are updates for:\n");
 	bool got_updates = FALSE;
 
@@ -198,7 +198,7 @@
 			output_string[strlen(output_string)-1] = '\0';
 		}
 		/* Constructs the notification. */
-		my_notify = notify_notification_new("New updates for Archlinux available!",output_string,NULL,NULL);
+		my_notify = notify_notification_new("New updates for Archlinux available!",output_string,NULL);
 		/* Sets the timeout until the notification disappears. */
 		notify_notification_set_timeout(my_notify,timeout);
 		/* We set the category.*/
