--- Makefile.old	2013-07-31 02:49:35.000000000 +0300
+++ Makefile	2013-07-31 12:35:33.185996869 +0300
@@ -5,10 +5,10 @@
 #   - Use "amd64" for a build on x86_64 (64-bits)
 #   - Use "ppc32" for a build on powerpc (32-bits)
 #   - Use "nojit" for unsupported architectures.
-ifeq ($(shell arch),x86_64)
+ifeq ($(shell uname -m),x86_64)
 export DYNAMIPS_ARCH?=amd64
 else
-ifeq ($(shell arch),i686)
+ifeq ($(shell uname -m),i686)
 export DYNAMIPS_ARCH?=x86
 endif
 endif
