Index: configure.ac
===================================================================
--- configure.ac	(revision 3036)
+++ configure.ac	(working copy)
@@ -243,6 +243,7 @@
 	AC_MSG_ERROR([*** mysql client library not found. Please install it.])
 else 
 	LDFLAGS="$LDFLAGS $MYSQL_LDFLAGS"
+	LDFLAGS="${LDFLAGS//-Wl,--as-needed}"
 	CFLAGS="$CFLAGS $MYSQL_CFLAGS"
 	CXXFLAGS="$CXXFLAGS $MYSQL_CFLAGS"
 fi
