--- configure.ac	2010-05-01 16:26:33.016870258 +0200
+++ configure.ac.new	2010-05-01 16:41:26.576771361 +0200
@@ -99,8 +99,8 @@
 # Checks for libraries.
 
 # Reset 
-CFLAGS="$CFLAGS $INCLUDES -Wall -fno-strict-aliasing"
-CXXFLAGS="$CXXFLAGS $INCLUDES -Wall -fno-strict-aliasing"
+CFLAGS="$CFLAGS $INCLUDES -w -Wfatal-errors -fno-strict-aliasing"
+CXXFLAGS="$CXXFLAGS $INCLUDES -w -Wfatal-errors -fno-strict-aliasing"
 
 # Check for c++0x functionnality (unordered_map)
 AC_CXX_COMPILE_STDCXX_0X
