diff -Naur gtkmm-utils-0.4.1~/glibmm-utils/date.cc gtkmm-utils-0.4.1/glibmm-utils/date.cc
--- gtkmm-utils-0.4.1~/glibmm-utils/date.cc	2012-11-21 11:43:42.726346909 -0600
+++ gtkmm-utils-0.4.1/glibmm-utils/date.cc	2012-11-21 11:44:27.172760954 -0600
@@ -25,7 +25,7 @@
 
 #include <sys/time.h>
 
-#include <glib/gmessages.h>
+#include <glib.h>
 
 #include <glibmm/convert.h>
 #include <glibmm/date.h>
diff -Naur gtkmm-utils-0.4.1~/glibmm-utils/ustring.cc gtkmm-utils-0.4.1/glibmm-utils/ustring.cc
--- gtkmm-utils-0.4.1~/glibmm-utils/ustring.cc	2012-11-21 11:43:42.726346909 -0600
+++ gtkmm-utils-0.4.1/glibmm-utils/ustring.cc	2012-11-21 11:44:55.645930936 -0600
@@ -20,8 +20,7 @@
  */
 
 #include <cstring> // for g++ 4.3
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include "ustring.h"
 
 namespace Glib {
