diff -crB pkg//etc/rc.d/iowolfet-ded pkg.new//etc/rc.d/iowolfet-ded
*** pkg//etc/rc.d/iowolfet-ded	2010-08-26 20:55:12.340000165 +0200
--- pkg.new//etc/rc.d/iowolfet-ded	2010-08-26 22:31:14.696666828 +0200
***************
*** 3,9 ****
  . /etc/rc.conf
  . /etc/rc.d/functions
  
! ET_BIN=/opt/iowolfet/ioetded.i386
  PID=`pidof -o %PPID $ET_BIN`
  
  case "$1" in
--- 3,9 ----
  . /etc/rc.conf
  . /etc/rc.d/functions
  
! ET_BIN=/opt/iowolfet/ioetded.x86_64
  PID=`pidof -o %PPID $ET_BIN`
  
  case "$1" in
diff -crB pkg//opt/iowolfet/iowolfet pkg.new//opt/iowolfet/iowolfet
*** pkg//opt/iowolfet/iowolfet	2010-08-26 20:55:12.326666827 +0200
--- pkg.new//opt/iowolfet/iowolfet	2010-08-26 22:30:48.106666831 +0200
***************
*** 3,6 ****
  # the binaries must run with correct working directory
  cd "/opt/iowolfet"
  export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:.
! exec ./ioet.i386 $*
--- 3,6 ----
  # the binaries must run with correct working directory
  cd "/opt/iowolfet"
  export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:.
! exec ./ioet.x86_64 $*
diff -crB pkg//opt/iowolfet/iowolfet-ded pkg.new//opt/iowolfet/iowolfet-ded
*** pkg//opt/iowolfet/iowolfet-ded	2010-08-26 20:55:12.333333498 +0200
--- pkg.new//opt/iowolfet/iowolfet-ded	2010-08-26 22:30:58.873333496 +0200
***************
*** 3,6 ****
  # the binaries must run with correct working directory
  cd "/opt/iowolfet"
  export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:.
! exec ./ioetded.i386 +exec server.cfg $*
--- 3,6 ----
  # the binaries must run with correct working directory
  cd "/opt/iowolfet"
  export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:.
! exec ./ioetded.x86_64 +exec server.cfg $*
