--- cudamat.py	2013-11-06 10:19:42.062701774 +0100
+++ cudamat.py	2013-11-06 10:05:42.384031000 +0100
@@ -4,10 +4,11 @@
 
 MAX_ONES = 1024*256
 
+_curdir = os.path.dirname(os.path.realpath(__file__))
 if platform.system() == 'Windows':
-    _cudamat = ct.cdll.LoadLibrary('libcudamat.dll')
+    _cudamat = ct.cdll.LoadLibrary(os.path.join(_curdir, 'libcudamat.dll'))
 else:
-    _cudamat = ct.cdll.LoadLibrary('libcudamat.so')
+    _cudamat = ct.cdll.LoadLibrary(os.path.join(_curdir, 'libcudamat.so'))
 
 _cudamat.get_last_cuda_error.restype = ct.c_char_p
 _cudamat.cublas_init.restype = ct.c_int
