diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba148a6..859061a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -530,8 +530,8 @@ if(NOT DISABLE_WX)
 			add_definitions(-D__WXGTK__)
 
 			# Check for required libs
-			check_lib(GTHREAD2 gthread-2.0 glib/gthread.h REQUIRED)
-			check_lib(PANGOCAIRO pangocairo pango/pangocairo.h REQUIRED)
+			check_lib(GTHREAD2 gthread-2.0 glib-2.0/glib/gthread.h REQUIRED)
+			check_lib(PANGOCAIRO pangocairo-1.0 pango-1.0/pango/pangocairo.h REQUIRED)
 		elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
 			add_definitions(-D__WXMSW__)
 		else()
diff --git a/Externals/wxWidgets3/CMakeLists.txt b/Externals/wxWidgets3/CMakeLists.txt
index 5b9891b..3b047f4 100644
--- a/Externals/wxWidgets3/CMakeLists.txt
+++ b/Externals/wxWidgets3/CMakeLists.txt
@@ -877,8 +877,8 @@ elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 		${SRCS_UNIXGTK})
 	set(LIBS
 		png
-		${GTHREAD2_LIBRARIES}
-		${PANGOCAIRO_LIBRARIES}
+		gthread-2.0
+		pangocairo-1.0
 		${GTK2_LIBRARIES})
 else()
 	add_definitions(-D__WXMSW__)