From 730ab0c9a88cf329e4f51d4ec75dcd0f3e0ef28f Mon Sep 17 00:00:00 2001
From: yar <yardenack@gmail.com>
Date: Sun, 4 May 2014 09:43:45 -0700
Subject: [PATCH 1/1] pass processlist flags to readOpenVZData (fixes build
 error)

---
 ProcessList.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ProcessList.c b/ProcessList.c
index 46dc0a6..7559623 100644
--- a/ProcessList.c
+++ b/ProcessList.c
@@ -584,7 +584,7 @@ static bool ProcessList_readStatmFile(Process* process, const char* dirname, con
 
 #ifdef HAVE_OPENVZ
 
-static void ProcessList_readOpenVZData(Process* process, const char* dirname, const char* name) {
+static void ProcessList_readOpenVZData(ProcessList* this, Process* process, const char* dirname, const char* name) {
    if ( (!(this->flags & PROCESS_FLAG_OPENVZ)) || (access("/proc/vz", R_OK) != 0)) {
       process->vpid = process->pid;
       process->ctid = 0;
@@ -812,7 +812,7 @@ static bool ProcessList_processEntries(ProcessList* this, const char* dirname, P
          process->user = UsersTable_getRef(this->usersTable, process->st_uid);
 
          #ifdef HAVE_OPENVZ
-         ProcessList_readOpenVZData(process, dirname, name);
+         ProcessList_readOpenVZData(this, process, dirname, name);
          #endif
          
          #ifdef HAVE_VSERVER
-- 
1.9.1

