diff -up src/bbkeys-0.9.1/configure.ac.orig src/bbkeys-0.9.1/configure.ac
--- src/bbkeys-0.9.1/configure.ac.orig	2013-07-29 13:42:47.000000000 -0600
+++ src/bbkeys-0.9.1/configure.ac	2013-07-29 13:43:33.000000000 -0600
@@ -131,8 +131,9 @@ AC_CHECK_LIB(Xext, XMissingExtension, LI
 
 dnl pkgconfig checks for blackbox library
 PKG_CHECK_MODULES(LIBBT, libbt)
-CXXFLAGS="$CXXFLAGS $LIBBT_CFLAGS"
-LIBS="$LIBS $LIBBT_LIBS"
+PKG_CHECK_MODULES(XFT, xft)
+CXXFLAGS="$CXXFLAGS $LIBBT_CFLAGS $XFT_CFLAGS"
+LIBS="$LIBS $LIBBT_LIBS $XFT_LIBS"
 
 dnl generate the config header
 AM_CONFIG_HEADER(config.h)
