diff -crB rmahjong-0.4-orig/client/client.py rmahjong-0.4/client/client.py
*** rmahjong-0.4-orig/client/client.py	2011-02-22 19:38:05.000000000 +0000
--- rmahjong-0.4/client/client.py	2013-01-13 20:55:08.759793224 +0000
***************
*** 298,304 ****
  
  			
  def main_init(config):
! 	logging.basicConfig(filename = "client.log", format = "%(asctime)s - %(levelname)s - %(message)s", level = logging.DEBUG)
  	pygame.display.init()
  	pygame.font.init()
  	config.preinit()
--- 298,304 ----
  
  			
  def main_init(config):
! 	logging.basicConfig(filename = "/tmp/rmahjong-client.log", format = "%(asctime)s - %(levelname)s - %(message)s", level = logging.DEBUG)
  	pygame.display.init()
  	pygame.font.init()
  	config.preinit()
diff -crB rmahjong-0.4-orig/server/server.py rmahjong-0.4/server/server.py
*** rmahjong-0.4-orig/server/server.py	2011-02-22 19:38:05.000000000 +0000
--- rmahjong-0.4/server/server.py	2013-01-13 20:55:32.198172433 +0000
***************
*** 91,97 ****
  if len(sys.argv) == 1:
  	print "Usage:", sys.argv[0], "<number_of_players>"
  else:
! 	logging.basicConfig(filename = "server.log", format = "%(asctime)s - %(levelname)s - %(message)s", level = logging.DEBUG)
  	server = Server(4500, int(sys.argv[1]))
  	sys.stdout.write("Init done\n")
  	sys.stdout.flush()
--- 91,97 ----
  if len(sys.argv) == 1:
  	print "Usage:", sys.argv[0], "<number_of_players>"
  else:
! 	logging.basicConfig(filename = "/tmp/rmahjong-server.log", format = "%(asctime)s - %(levelname)s - %(message)s", level = logging.DEBUG)
  	server = Server(4500, int(sys.argv[1]))
  	sys.stdout.write("Init done\n")
  	sys.stdout.flush()
