diff -crB rmahjong-0.4-orig/client/graphics.py rmahjong-0.4/client/graphics.py
*** rmahjong-0.4-orig/client/graphics.py	2011-02-22 19:38:05.000000000 +0000
--- rmahjong-0.4/client/graphics.py	2013-01-13 20:38:01.983362880 +0000
***************
*** 19,24 ****
--- 19,25 ----
  from pygame import display, event
  import pygame
  import OpenGL.GL as gl
+ import OpenGL.GL.EXT.texture_filter_anisotropic as tfa
  import OpenGL.GLU as glu
  
  font = None
***************
*** 154,160 ****
  		gl.glTexEnvf( gl.GL_TEXTURE_ENV, gl.GL_TEXTURE_ENV_MODE, gl.GL_MODULATE );
  		gl.glTexParameteri(gl.GL_TEXTURE_2D, gl.GL_TEXTURE_MAG_FILTER, gl.GL_LINEAR)
  		gl.glTexParameteri(gl.GL_TEXTURE_2D, gl.GL_TEXTURE_MIN_FILTER, gl.GL_LINEAR_MIPMAP_LINEAR)
! 		gl.glTexParameterf(gl.GL_TEXTURE_2D, gl.GL_TEXTURE_MAX_ANISOTROPY_EXT, 1000.0)
  
  		gl.glTexImage2D(gl.GL_TEXTURE_2D, 0, gl.GL_RGBA, width, height, 0, gl.GL_RGBA, gl.GL_UNSIGNED_BYTE, data)
  		glu.gluBuild2DMipmaps(gl.GL_TEXTURE_2D, 4, width, height, gl.GL_RGBA, gl.GL_UNSIGNED_BYTE, data );
--- 155,161 ----
  		gl.glTexEnvf( gl.GL_TEXTURE_ENV, gl.GL_TEXTURE_ENV_MODE, gl.GL_MODULATE );
  		gl.glTexParameteri(gl.GL_TEXTURE_2D, gl.GL_TEXTURE_MAG_FILTER, gl.GL_LINEAR)
  		gl.glTexParameteri(gl.GL_TEXTURE_2D, gl.GL_TEXTURE_MIN_FILTER, gl.GL_LINEAR_MIPMAP_LINEAR)
! 		gl.glTexParameterf(gl.GL_TEXTURE_2D, tfa.GL_TEXTURE_MAX_ANISOTROPY_EXT, 1000.0)
  
  		gl.glTexImage2D(gl.GL_TEXTURE_2D, 0, gl.GL_RGBA, width, height, 0, gl.GL_RGBA, gl.GL_UNSIGNED_BYTE, data)
  		glu.gluBuild2DMipmaps(gl.GL_TEXTURE_2D, 4, width, height, gl.GL_RGBA, gl.GL_UNSIGNED_BYTE, data );
