Description: Fix compatibility with pcsc-lite >= 1.7.3
Author: Wolfgang Rosenauer <wr@rosenauer.org> 
--- a/src/ccid_ifdhandler.h
+++ b/src/ccid_ifdhandler.h
@@ -31,6 +31,12 @@
 	SCARD_CTL_CODE(FEATURE_VERIFY_PIN_DIRECT + CLASS2_IOCTL_MAGIC)
 #define IOCTL_FEATURE_MODIFY_PIN_DIRECT \
 	SCARD_CTL_CODE(FEATURE_MODIFY_PIN_DIRECT + CLASS2_IOCTL_MAGIC)
+
+// Fix problem using pcsc-lite 1.7.3 or later header files
+#ifndef FEATURE_MCT_READERDIRECT
+#define FEATURE_MCT_READERDIRECT	FEATURE_MCT_READER_DIRECT
+#endif
+
 #define IOCTL_FEATURE_MCT_READERDIRECT \
 	SCARD_CTL_CODE(FEATURE_MCT_READERDIRECT + CLASS2_IOCTL_MAGIC)
 
