diff -ruN rox-volume/AppletRun rox-volume_a/AppletRun
--- rox-volume/AppletRun	2008-12-13 23:56:36.000000000 -0300
+++ rox-volume_a/AppletRun	2012-05-11 03:05:19.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import findrox; findrox.version(2,0,0)
 import rox, os, sys
 
diff -ruN rox-volume/AppRun rox-volume_a/AppRun
--- rox-volume/AppRun	2008-12-13 23:56:36.000000000 -0300
+++ rox-volume_a/AppRun	2012-05-11 03:05:26.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import findrox; findrox.version(2,0,0)
 import rox, os, sys
 
diff -ruN rox-volume/Messages/make_tips rox-volume_a/Messages/make_tips
--- rox-volume/Messages/make_tips	2008-12-13 23:56:36.000000000 -0300
+++ rox-volume_a/Messages/make_tips	2012-05-11 03:05:02.000000000 -0400
@@ -1,4 +1,4 @@
-#!/bin/env python
+#!/bin/env python2
 
 from xmllib import *
 import string, os
