--- libcap-ng/utils/pscap.c	2012-02-17 10:29:28.413944941 +0100
+++ libcap-ng/utils/pscap.c	2012-02-27 17:17:51.943962066 +0100
@@ -57,11 +57,6 @@ int main(int argc, char *argv[])
 	}
 
 	/* Drop capabilities so we aren't listed */
-	caps = capng_have_capabilities(CAPNG_SELECT_CAPS);
-	if (caps == CAPNG_FULL) {
-		capng_clear(CAPNG_SELECT_BOTH);
-		capng_apply(CAPNG_SELECT_BOTH);
-	}
 
 	d = opendir("/proc");
 	if (d == NULL) {

