diff -Naur Weather-Underground-3.0.2~/Underground.pm Weather-Underground-3.02/Underground.pm
--- Weather-Underground-3.0.2~/Underground.pm	2011-03-05 11:13:28.261594137 -0600
+++ Weather-Underground-3.02/Underground.pm	2011-03-05 11:14:41.509890127 -0600
@@ -443,7 +443,7 @@
 		$ua->timeout($self->{timeout});
 	}
 	$oldagent = $ua->agent();
-	$ua->agent("Weather::Underground version $VERSION");
+	$ua->agent("curl version $VERSION");
 	$document = get($self->{_url});
 	$ua->agent($oldagent);
 
@@ -763,8 +763,8 @@
 	}
 
 	$stateref->{"content_TEMPERATURE"} =~ s/\s//g;
-	($temperature_celsius)    = ($stateref->{"content_TEMPERATURE"} =~ /(-?(?:\d|\.)+)[^a-z0-9]*?c/i);
-	($temperature_fahrenheit) = ($stateref->{"content_TEMPERATURE"} =~ /(-?(?:\d|\.)+)[^a-z0-9]*?f/i);
+  ($temperature_celsius)    = ($stateref->{"content_TEMPERATURE"} =~  /(-?([?[0-9]*\.?[0-9]+))&deg;c/i);
+  ($temperature_fahrenheit) = ($stateref->{"content_TEMPERATURE"} =~ /(-?([?[0-9]*\.?[0-9]+))&deg;f/i);
 	if (!length($temperature_celsius) && length($temperature_fahrenheit)) {
 		$temperature_celsius = ($temperature_fahrenheit - 32) / 1.8;
 	}
@@ -773,8 +773,8 @@
 	}
 
 	$stateref->{"content_DEW_POINT"} =~ s/\s//g;
-	($dewpoint_celsius)    = ($stateref->{"content_DEW_POINT"} =~ /(-?(?:\d|\.)+)[^a-z0-9]*?c/i);
-	($dewpoint_fahrenheit) = ($stateref->{"content_DEW_POINT"} =~ /(-?(?:\d|\.)+)[^a-z0-9]*?f/i);
+  ($dewpoint_celsius)    = ($stateref->{"content_DEW_POINT"} =~  /(-?([?[0-9]*\.?[0-9]+))&deg;c/i);
+  ($dewpoint_fahrenheit) = ($stateref->{"content_DEW_POINT"} =~ /(-?([?[0-9]*\.?[0-9]+))&deg;f/i);
 	if (!length($dewpoint_celsius) && length($dewpoint_fahrenheit)) {
 		$dewpoint_celsius = ($dewpoint_fahrenheit - 32) / 1.8;
 	}
