*** opennurbs_array_defs.h	2012-09-07 07:35:48.000000000 +0200
--- ../../opennurbs_array_defs.h	2012-09-07 07:07:51.000000000 +0200
***************
*** 14,19 ****
--- 14,29 ----
  ////////////////////////////////////////////////////////////////
  */
  
+ #include <stddef.h>
+ 
+ void ON_qsort( 
+         void* base,
+         size_t count,
+         size_t sizeof_element,
+         int (*compare)(const void*,const void*)
+         );
+ 
+ 
  #if !defined(ON_ARRAY_DEFS_INC_)
  #define ON_ARRAY_DEFS_INC_
  
