--- sta/sta_cfg.c.orig	2011-04-27 04:22:21.000000000 +0200
+++ sta/sta_cfg.c	2011-08-02 14:56:55.000000000 +0200
@@ -4923,7 +4923,7 @@
             wrq->u.data.length = strlen(extra) + 1; /* 1: size of '\0' */
             break;
         case SHOW_DRVIER_VERION:
-            snprintf(extra, size, "Driver version-%s, %s %s\n", STA_DRIVER_VERSION, __DATE__, __TIME__ );
+            snprintf(extra, size, "Driver version-%s, %s %s\n", STA_DRIVER_VERSION, NULL, NULL );
             wrq->u.data.length = strlen(extra) + 1; /* 1: size of '\0' */
             break;
 #ifdef DOT11_N_SUPPORT
