--- include/os/rt_linux.h.orig	2010-12-15 13:04:34.000000000 +0100
+++ include/os/rt_linux.h	2010-12-31 20:33:24.000000000 +0100
@@ -138,10 +138,10 @@
 #ifdef CONFIG_STA_SUPPORT
 
 #ifdef RTMP_MAC_USB
-#define STA_PROFILE_PATH			"/etc/Wireless/RT2870STA/RT2870STA.dat"
+#define STA_PROFILE_PATH			"/etc/Wireless/RT3572STA/RT3572STA.dat"
 #define STA_DRIVER_VERSION			"2.5.0.0"
 #ifdef MULTIPLE_CARD_SUPPORT
-#define CARD_INFO_PATH			"/etc/Wireless/RT2870STA/RT2870STACard.dat"
+#define CARD_INFO_PATH			"/etc/Wireless/RT3572STA/RT3572STACard.dat"
 #endif /* MULTIPLE_CARD_SUPPORT */
 #endif /* RTMP_MAC_USB */
 
@@ -1440,7 +1440,7 @@
 
 #ifdef RTMP_MAC_USB
 #ifdef CONFIG_STA_SUPPORT
-#define EEPROM_BIN_FILE_NAME  "/etc/Wireless/RT2870STA/e2p.bin"
+#define EEPROM_BIN_FILE_NAME  "/etc/Wireless/RT3572STA/e2p.bin"
 #endif /* CONFIG_STA_SUPPORT */
 #endif /* RTMP_MAC_USB */
 
--- include/os/rt_linux_cmm.h.orig	2010-12-31 20:37:40.000000000 +0100
+++ include/os/rt_linux_cmm.h	2010-12-31 20:38:52.000000000 +0100
@@ -88,7 +88,7 @@
 #else
 
 #ifdef RTMP_MAC_USB
-#define RTMP_DRV_NAME	"rt2870"
+#define RTMP_DRV_NAME	"rt3572"
 #else
 #define RTMP_DRV_NAME	"rt2860"
 #endif /* RTMP_MAC_USB */
--- os/linux/usb_main_dev.c.orig	2010-12-15 13:15:33.000000000 +0100
+++ os/linux/usb_main_dev.c	2010-12-31 20:35:02.000000000 +0100
@@ -35,7 +35,7 @@
 /* *** If you have a solution for the bug in current version of driver, please mail to me. */
 /* Otherwise post to forum in ralinktech's web site(www.ralinktech.com) and let all users help you. *** */
 MODULE_AUTHOR("Paul Lin <paul_lin@ralinktech.com>");
-MODULE_DESCRIPTION("RT2870 Wireless Lan Linux Driver");
+MODULE_DESCRIPTION("RT3572 Wireless Lan Linux Driver");
 #ifdef CONFIG_STA_SUPPORT
 #ifdef MODULE_VERSION
 MODULE_VERSION(STA_DRIVER_VERSION);
