diff -aur jack_snapshot.orig/jack_snapshot.cc jack_snapshot.patched/jack_snapshot.cc
--- jack_snapshot.orig/jack_snapshot.cc	2009-09-15 01:50:27.000000000 +0200
+++ jack_snapshot.patched/jack_snapshot.cc	2009-09-15 01:53:36.000000000 +0200
@@ -8,6 +8,8 @@
 #include <iostream>
 #include <fstream>
 #include <cerrno>
+#include <string.h>
+#include <cstdlib>
 
 int main(int argc, char *argv[]) {
   std::cout << "Jack connection snapshot [(C) 2004 - inf. Florian Schmidt]" << std::endl;
