--- nightingale.orig	2012-02-05 01:04:49.000000000 -0800
+++ nightingale	2012-02-05 01:10:19.000000000 -0800
@@ -39,18 +39,7 @@
 #set -e
 #set -x
 
-# this depends on your system's gstreamer location
-# while sometimes, the make process actually uses the location properly
-# it doesn't always, and this is better than symlinking
-for dir in /usr/lib64 /usr/lib ; do
-  if [ -f ${dir}/gstreamer-0.10/libgstcoreelements.so ] ; then
-    export GST_PLUGIN_PATH=${dir}/gstreamer\-0.10
-    break
-  elif [ -f ${dir}/gstreamer0.10/libgstcoreelements.so ] ; then
-	export GST_PLUIN_PATH=${dir}/gstreamer0.10
-	break
-  fi
-done
+export GST_PLUGIN_PATH="/usr/lib/gstreamer-0.10"
 
 cmdname=`basename "$0"`
 MOZ_DIST_BIN=`dirname "$0"`
