diff -u org/common-CMakeLists.txt new/common-CMakeLists.txt
--- org/common-CMakeLists.txt	2011-10-08 09:30:36.377277095 -0400
+++ new/common-CMakeLists.txt	2011-10-08 09:31:15.068117049 -0400
@@ -164,7 +164,7 @@
 # TODO but gcc on the Mac is at version 4.0 and this
 # TODO option was added in some later version.
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wdeclaration-after-statement -Werror -Wstrict-prototypes")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wdeclaration-after-statement -Wstrict-prototypes -Wno-unused")
 
 # the apple_unicode code violates the strict-aliasing rules
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-strict-aliasing")
