diff -u org/common-CMakeLists.txt new/common-CMakeLists.txt
--- org/common-CMakeLists.txt	2011-08-15 13:24:40.183902314 -0400
+++ new/common-CMakeLists.txt	2011-08-15 14:33:43.226919628 -0400
@@ -138,7 +138,7 @@
 # TODO but gcc on the Mac is at version 4.0 and this
 # TODO option was added in some later version.
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wdeclaration-after-statement -Werror -Wstrict-prototypes")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wdeclaration-after-statement -Werror -Wstrict-prototypes -Wno-unused")
 
 # the apple_unicode code violates the strict-aliasing rules
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-strict-aliasing")
