--- target-ethos/general/presentation.cpp	2012-08-13 20:30:53.008742985 +0200
+++ target-ethos/general/presentation.cpp	2012-08-13 20:30:41.952076712 +0200
@@ -205,6 +205,7 @@
 void Presentation::loadShaders() {
   string pathname = application->path("Video Shaders/");
   lstring files = directory::files(pathname);
+  if(files.size() == 0) files = directory::files("/usr/share/bsnes/shaders/");
   for(auto &filename : files) {
     auto shader = new RadioItem;
     string name = filename;
