--- snes/chip/icd2/interface/interface.cpp.orig	2011-06-18 08:02:08.000000000 +0200
+++ snes/chip/icd2/interface/interface.cpp	2011-06-18 08:02:19.000000000 +0200
@@ -101,7 +101,7 @@
     case 3: data = ~r6007; break;
   }
 
-  switch(id) {
+  switch((GameBoy::Input)id) {
     case GameBoy::Input::Start:  return data & 0x80;
     case GameBoy::Input::Select: return data & 0x40;
     case GameBoy::Input::B:      return data & 0x20;
