--- ui/input/input.cpp.orig	2011-06-18 08:01:59.000000000 +0200
+++ ui/input/input.cpp	2011-06-18 08:03:35.000000000 +0200
@@ -82,7 +82,7 @@
 }
 
 int16_t InputMapper::Gamepad::poll(unsigned id) {
-  switch(id) {
+  switch((SNES::Input::JoypadID)id) {
     case SNES::Input::JoypadID::Up:     return up.poll();
     case SNES::Input::JoypadID::Down:   return down.poll() & !up.poll();
     case SNES::Input::JoypadID::Left:   return left.poll();
@@ -118,7 +118,7 @@
 }
 
 int16_t InputMapper::Mouse::poll(unsigned id) {
-  switch(id) {
+  switch((SNES::Input::MouseID)id) {
     case SNES::Input::MouseID::X: return x.poll();
     case SNES::Input::MouseID::Y: return y.poll();
     case SNES::Input::MouseID::Left: return left.poll();
@@ -150,7 +150,7 @@
 }
 
 int16_t InputMapper::SuperScope::poll(unsigned id) {
-  switch(id) {
+  switch((SNES::Input::SuperScopeID)id) {
     case SNES::Input::SuperScopeID::X: return x.poll();
     case SNES::Input::SuperScopeID::Y: return y.poll();
     case SNES::Input::SuperScopeID::Trigger: return trigger.poll();
@@ -182,7 +182,7 @@
 }
 
 int16_t InputMapper::Justifier::poll(unsigned id) {
-  switch(id) {
+  switch((SNES::Input::JustifierID)id) {
     case SNES::Input::JustifierID::X: return x.poll();
     case SNES::Input::JustifierID::Y: return y.poll();
     case SNES::Input::JustifierID::Trigger: return trigger.poll();
--- snes/chip/icd2/interface/interface.cpp.orig	2011-06-18 08:02:08.000000000 +0200
+++ snes/chip/icd2/interface/interface.cpp	2011-06-18 08:02:19.000000000 +0200
@@ -101,7 +101,7 @@
     case 3: data = ~r6007; break;
   }
 
-  switch(id) {
+  switch((GameBoy::Input)id) {
     case GameBoy::Input::Start:  return data & 0x80;
     case GameBoy::Input::Select: return data & 0x40;
     case GameBoy::Input::B:      return data & 0x20;
