diff -Nur perl-poppler/lib/Poppler.pm perl-poppler-test/lib/Poppler.pm
--- perl-poppler/lib/Poppler.pm	2011-01-24 16:22:54.946531212 +0100
+++ perl-poppler-test/lib/Poppler.pm	2011-01-24 16:24:19.946197790 +0100
@@ -34,6 +34,11 @@
     my $path = 'file:///path/to/some.pdf';
 
     my $o = Poppler::Document->new_from_file($path);
+    # or if you want to open the pdf in perl, or use some othe source:
+    #  open (PDF, "<FILE.PDF");
+    #  read (PDF, my $data, -s "FILE.PDF");
+    #  close (PDF);
+    #  my $o = Poppler::Document->new_from_data($data, length($data));
 
     my $page = $o->get_page( 0 );
 
@@ -61,6 +66,8 @@
 
 =head2 L<Poppler::Document> Blessed Object = Poppler::Document->new_from_file( STRING uri )
 
+=head2 L<Poppler::Document> Blessed Object = Poppler::Document->new_from_data( STRING pdf_data, length ( STRING pdf_data ) )
+
 =head2 BOOLEAN = $poppler_document->save( STRING uri )
 
 =head2 BOOLEAN = $poppler_document->save_a_copy( STRING uri )
diff -Nur perl-poppler/Poppler.xs perl-poppler-test/Poppler.xs
--- perl-poppler/Poppler.xs	2011-01-24 16:22:54.946531212 +0100
+++ perl-poppler-test/Poppler.xs	2011-01-24 16:23:34.548114551 +0100
@@ -38,6 +38,24 @@
 OUTPUT:
     RETVAL
 
+hPopplerDocument*
+new_from_data( class , data , length )
+		char * class;
+		char * data;
+		int * length;
+PREINIT:
+		PopplerDocument *document;
+CODE:
+		g_type_init();
+		Newz(0, RETVAL, 1, hPopplerDocument );
+		document = poppler_document_new_from_data( data, length, NULL, NULL );
+		if( document == NULL ) {
+        croak("poppler_document_new_from_data failed");
+		}
+		RETVAL->handle = document;
+OUTPUT:
+		RETVAL
+
 int
 hPopplerDocument::save( uri )
     char * uri;
diff -Nur perl-poppler/typemap perl-poppler-test/typemap
--- perl-poppler/typemap	2011-01-24 16:22:54.947531126 +0100
+++ perl-poppler-test/typemap	2011-01-24 16:23:34.548114551 +0100
@@ -14,6 +14,8 @@
 cairo_t * T_OBJECT
 cairo_surface_t * T_OBJECT
 
+int * T_INT
+
 INPUT
 
 # newSVGooString
