--- GNUmakefile.in.orig	2012-05-01 12:34:55.674940696 +0000
+++ GNUmakefile.in	2012-05-01 12:37:24.206818332 +0000
@@ -47,7 +47,7 @@
 INSTALL_DIR=		@INSTALL_DIR@
 
 # install(1) commands
-INSTALL_DATA=		${INSTALL} -m 444
+INSTALL_DATA=		${INSTALL} -m 444 -D
 INSTALL_PROGRAM=	${INSTALL} -m 555
 INSTALL_SCRIPT=		${INSTALL_PROGRAM}
 
@@ -70,11 +70,14 @@
 	rm -rf config.status
 	rm -rf configure configure.lineno
 
-install: ${DESTDIR}${bindir}/mkcue
+install: ${DESTDIR}${bindir}/mkcue ${DESTDIR}${mandir}/mkcue.1.gz
 
 ${DESTDIR}${bindir}/%: %
 	${INSTALL_PROGRAM} $< $@
 
+${DESTDIR}${mandir}/%: %
+	${INSTALL_DATA} $< $@
+
 # BLECH
 dist:
 	svn export . ${distname}
