From: Aurelien Gateau <aurelien.gateau@canonical.com>
Date: Thu, 16 Dec 2010 11:17:14 +0100
Subject: [PATCH] Added an option to make it possible to hide maximized titlebars



bzr-revid: aurelien.gateau@canonical.com-20101216101714-tyht142o66k25uaj

diff --git a/src/ui/frames.c b/src/ui/frames.c
index f5ea044..622201f 100644
--- a/src/ui/frames.c
+++ b/src/ui/frames.c
@@ -93,7 +93,7 @@
 
 static void meta_frames_font_changed          (MetaFrames *frames);
 static void meta_frames_button_layout_changed (MetaFrames *frames);
-
+static void meta_frames_show_maximized_titlebars_changed (MetaFrames *frames);
 
 static GdkRectangle*    control_rect (MetaFrameControl   control,
                                       MetaFrameGeometry *fgeom);
@@ -195,6 +195,9 @@
     case META_PREF_BUTTON_LAYOUT:
       meta_frames_button_layout_changed (META_FRAMES (data));
       break;
+    case META_PREF_SHOW_MAXIMIZED_TITLEBARS:
+      meta_frames_show_maximized_titlebars_changed (META_FRAMES (data));
+      break;
     default:
       break;
     }
@@ -421,6 +424,16 @@
 {
   g_hash_table_foreach (frames->frames,
                         queue_draw_func, frames);
+}
+
+void meta_frames_show_maximized_titlebars_changed (MetaFrames *frames)
+{
+  MetaTheme *theme = meta_theme_get_current ();
+  if (theme)
+    {
+      meta_theme_set_show_maximized_titlebars (theme,
+                                               meta_prefs_get_show_maximized_titlebars ());
+    }
 }
 
 static void
diff --git a/src/core/prefs.c b/src/core/prefs.c
index 405f401..06da206 100644
--- a/src/core/prefs.c
+++ b/src/core/prefs.c
@@ -99,6 +99,7 @@
 static gboolean resize_with_right_button = FALSE;
 static gboolean force_fullscreen = TRUE;
 static gboolean hide_decorator_tooltip = FALSE;
+static gboolean show_maximized_titlebars = TRUE;
 
 static MetaVisualBellType visual_bell_type = META_VISUAL_BELL_FULLSCREEN_FLASH;
 static MetaButtonLayout button_layout;
@@ -419,6 +420,11 @@
       &hide_decorator_tooltip,
       FALSE,
     },
+    { "/apps/metacity/general/show_maximized_titlebars",
+      META_PREF_SHOW_MAXIMIZED_TITLEBARS,
+      &show_maximized_titlebars,
+      FALSE,
+    },
     { NULL, 0, NULL, FALSE },
   };
 
@@ -1273,6 +1279,12 @@
 meta_prefs_get_hide_decorator_tooltip (void)
 {
   return hide_decorator_tooltip;
+}
+
+gboolean
+meta_prefs_get_show_maximized_titlebars (void)
+{
+  return show_maximized_titlebars;
 }
 
 const char*
@@ -1770,6 +1782,9 @@
 
     case META_PREF_HIDE_DECORATOR_TOOLTIP:
       return "HIDE_DECORATOR_TOOLTIP";
+
+    case META_PREF_SHOW_MAXIMIZED_TITLEBARS:
+      return "META_PREF_SHOW_MAXIMIZED_TITLEBARS";
     }
 
   return "(unknown)";
diff --git a/src/ui/ui.c b/src/ui/ui.c
index 9fdf346..fadb90f 100644
--- a/src/ui/ui.c
+++ b/src/ui/ui.c
@@ -881,6 +881,12 @@
                            gboolean    force_reload)
 {
   meta_theme_set_current (name, force_reload);
+  MetaTheme *theme = meta_theme_get_current ();
+  if (theme)
+    {
+      meta_theme_set_show_maximized_titlebars (theme,
+                                               meta_prefs_get_show_maximized_titlebars ());
+    }
   meta_invalidate_default_icons ();
 }
 
diff --git a/src/include/prefs.h b/src/include/prefs.h
index 30d63b8..73296b9 100644
--- a/src/include/prefs.h
+++ b/src/include/prefs.h
@@ -61,7 +61,8 @@
   META_PREF_COMPOSITING_MANAGER,
   META_PREF_RESIZE_WITH_RIGHT_BUTTON,
   META_PREF_FORCE_FULLSCREEN,
-  META_PREF_HIDE_DECORATOR_TOOLTIP
+  META_PREF_HIDE_DECORATOR_TOOLTIP,
+  META_PREF_SHOW_MAXIMIZED_TITLEBARS
 } MetaPreference;
 
 typedef void (* MetaPrefsChangedFunc) (MetaPreference pref,
@@ -82,6 +83,7 @@
 MetaFocusNewWindows         meta_prefs_get_focus_new_windows  (void);
 gboolean                    meta_prefs_get_raise_on_click     (void);
 gboolean                    meta_prefs_get_hide_decorator_tooltip (void);
+gboolean                    meta_prefs_get_show_maximized_titlebars (void);
 const char*                 meta_prefs_get_theme              (void);
 /* returns NULL if GTK default should be used */
 const PangoFontDescription* meta_prefs_get_titlebar_font      (void);
diff --git a/src/ui/theme.c b/src/ui/theme.c
index b6a5f62..32d162d 100644
--- a/src/ui/theme.c
+++ b/src/ui/theme.c
@@ -406,7 +406,8 @@
                                int                   *top_height,
                                int                   *bottom_height,
                                int                   *left_width,
-                               int                   *right_width)
+                               int                   *right_width,
+                               MetaTheme             *theme)
 {
   int buttons_height, title_height;
   
@@ -441,6 +442,9 @@
       else
         *bottom_height = layout->bottom_height;
     }
+
+  if ((flags & META_FRAME_MAXIMIZED) && !theme->show_maximized_titlebars)
+    *top_height = 0;
 
   if (flags & META_FRAME_FULLSCREEN)
     {
@@ -614,7 +618,8 @@
                                  &fgeom->top_height,
                                  &fgeom->bottom_height,
                                  &fgeom->left_width,
-                                 &fgeom->right_width);
+                                 &fgeom->right_width,
+                                 theme);
 
   width = client_width + fgeom->left_width + fgeom->right_width;
 
@@ -4847,6 +4852,9 @@
   theme->quark_icon_height = g_quark_from_static_string ("icon_height");
   theme->quark_title_width = g_quark_from_static_string ("title_width");
   theme->quark_title_height = g_quark_from_static_string ("title_height");
+
+  theme->show_maximized_titlebars = TRUE;
+
   return theme;
 }
 
@@ -5271,7 +5279,8 @@
                                  text_height,
                                  flags,
                                  top_height, bottom_height,
-                                 left_width, right_width);
+                                 left_width, right_width,
+                                 theme);
 }
 
 void
@@ -6635,3 +6644,9 @@
       return 1; 
     }
 }
+
+void
+meta_theme_set_show_maximized_titlebars (MetaTheme *theme, gboolean show)
+{
+  theme->show_maximized_titlebars = show;
+}
diff --git a/src/metacity.schemas.in.in b/src/metacity.schemas.in.in
index 97cd280..e7eb082 100644
--- a/src/metacity.schemas.in.in
+++ b/src/metacity.schemas.in.in
@@ -36,6 +36,20 @@
            with the middle button while holding down the key given in
            "mouse_button_modifier"; set it to false to make it work the
            opposite way around.
+         </long>
+      </locale>
+    </schema>
+
+    <schema>
+      <key>/schemas/apps/metacity/general/show_maximized_titlebars</key>
+      <applyto>/apps/metacity/general/show_maximized_titlebars</applyto>
+      <owner>metacity</owner>
+      <type>bool</type>
+      <default>true</default>
+      <locale name="C">
+         <short>Whether to show title bars for maximized windows</short>
+         <long>
+           Set this to false to hide title bars for maximized windows.
          </long>
       </locale>
     </schema>
diff --git a/src/ui/theme.h b/src/ui/theme.h
index 39d9e21..8d90e64 100644
--- a/src/ui/theme.h
+++ b/src/ui/theme.h
@@ -821,6 +821,8 @@
   GQuark quark_icon_height;
   GQuark quark_title_width;
   GQuark quark_title_height;
+
+  gboolean show_maximized_titlebars;
 };
 
 struct _MetaPositionExprEnv
@@ -854,7 +856,8 @@
                                                   int                   *top_height,
                                                   int                   *bottom_height,
                                                   int                   *left_width,
-                                                  int                   *right_width);
+                                                  int                   *right_width,
+                                                  MetaTheme             *theme);
 void             meta_frame_layout_calc_geometry (const MetaFrameLayout  *layout,
                                                   int                     text_height,
                                                   MetaFrameFlags          flags,
@@ -1175,6 +1178,8 @@
 
 guint meta_theme_earliest_version_with_button (MetaButtonType type);
 
+void meta_theme_set_show_maximized_titlebars (MetaTheme *theme, gboolean);
+
 
 #define META_THEME_ALLOWS(theme, feature) (theme->format_version >= feature)
 
