From: Aurelien Gateau <aurelien.gateau@canonical.com>
Date: Thu, 16 Dec 2010 11:17:14 +0100
Subject: [PATCH] Added an option to make it possible to hide maximized titlebars



bzr-revid: aurelien.gateau@canonical.com-20101216101714-tyht142o66k25uaj

Index: metacity-2.34.1/src/ui/frames.c
===================================================================
--- metacity-2.34.1.orig/src/ui/frames.c	2011-08-18 00:09:47.559519247 -0400
+++ metacity-2.34.1/src/ui/frames.c	2011-08-18 00:10:05.569953587 -0400
@@ -95,7 +95,7 @@
 
 static void meta_frames_font_changed          (MetaFrames *frames);
 static void meta_frames_button_layout_changed (MetaFrames *frames);
-
+static void meta_frames_show_maximized_titlebars_changed (MetaFrames *frames);
 
 static GdkRectangle*    control_rect (MetaFrameControl   control,
                                       MetaFrameGeometry *fgeom);
@@ -197,6 +197,9 @@
     case META_PREF_BUTTON_LAYOUT:
       meta_frames_button_layout_changed (META_FRAMES (data));
       break;
+    case META_PREF_SHOW_MAXIMIZED_TITLEBARS:
+      meta_frames_show_maximized_titlebars_changed (META_FRAMES (data));
+      break;
     default:
       break;
     }
@@ -425,6 +428,16 @@
                         queue_draw_func, frames);
 }
 
+void meta_frames_show_maximized_titlebars_changed (MetaFrames *frames)
+{
+  MetaTheme *theme = meta_theme_get_current ();
+  if (theme)
+    {
+      meta_theme_set_show_maximized_titlebars (theme,
+                                               meta_prefs_get_show_maximized_titlebars ());
+    }
+}
+
 static void
 reattach_style_func (gpointer key, gpointer value, gpointer data)
 {
Index: metacity-2.34.1/src/core/prefs.c
===================================================================
--- metacity-2.34.1.orig/src/core/prefs.c	2011-08-18 00:09:47.559519247 -0400
+++ metacity-2.34.1/src/core/prefs.c	2011-08-18 00:10:05.569953587 -0400
@@ -99,6 +99,7 @@
 static gboolean resize_with_right_button = FALSE;
 static gboolean force_fullscreen = TRUE;
 static gboolean hide_decorator_tooltip = FALSE;
+static gboolean show_maximized_titlebars = TRUE;
 
 static MetaVisualBellType visual_bell_type = META_VISUAL_BELL_FULLSCREEN_FLASH;
 static MetaButtonLayout button_layout;
@@ -419,6 +420,11 @@
       &hide_decorator_tooltip,
       FALSE,
     },
+    { "/apps/metacity/general/show_maximized_titlebars",
+      META_PREF_SHOW_MAXIMIZED_TITLEBARS,
+      &show_maximized_titlebars,
+      FALSE,
+    },
     { NULL, 0, NULL, FALSE },
   };
 
@@ -1275,6 +1281,12 @@
   return hide_decorator_tooltip;
 }
 
+gboolean
+meta_prefs_get_show_maximized_titlebars (void)
+{
+  return show_maximized_titlebars;
+}
+
 const char*
 meta_prefs_get_theme (void)
 {
@@ -1770,6 +1782,9 @@
 
     case META_PREF_HIDE_DECORATOR_TOOLTIP:
       return "HIDE_DECORATOR_TOOLTIP";
+
+    case META_PREF_SHOW_MAXIMIZED_TITLEBARS:
+      return "META_PREF_SHOW_MAXIMIZED_TITLEBARS";
     }
 
   return "(unknown)";
Index: metacity-2.34.1/src/ui/ui.c
===================================================================
--- metacity-2.34.1.orig/src/ui/ui.c	2011-08-18 00:09:59.519807670 -0400
+++ metacity-2.34.1/src/ui/ui.c	2011-08-18 00:10:05.569953587 -0400
@@ -883,6 +883,12 @@
                            gboolean    force_reload)
 {
   meta_theme_set_current (name, force_reload);
+  MetaTheme *theme = meta_theme_get_current ();
+  if (theme)
+    {
+      meta_theme_set_show_maximized_titlebars (theme,
+                                               meta_prefs_get_show_maximized_titlebars ());
+    }
   meta_invalidate_default_icons ();
 }
 
Index: metacity-2.34.1/src/include/prefs.h
===================================================================
--- metacity-2.34.1.orig/src/include/prefs.h	2011-08-18 00:09:47.559519247 -0400
+++ metacity-2.34.1/src/include/prefs.h	2011-08-18 00:10:05.569953587 -0400
@@ -61,7 +61,8 @@
   META_PREF_COMPOSITING_MANAGER,
   META_PREF_RESIZE_WITH_RIGHT_BUTTON,
   META_PREF_FORCE_FULLSCREEN,
-  META_PREF_HIDE_DECORATOR_TOOLTIP
+  META_PREF_HIDE_DECORATOR_TOOLTIP,
+  META_PREF_SHOW_MAXIMIZED_TITLEBARS
 } MetaPreference;
 
 typedef void (* MetaPrefsChangedFunc) (MetaPreference pref,
@@ -82,6 +83,7 @@
 MetaFocusNewWindows         meta_prefs_get_focus_new_windows  (void);
 gboolean                    meta_prefs_get_raise_on_click     (void);
 gboolean                    meta_prefs_get_hide_decorator_tooltip (void);
+gboolean                    meta_prefs_get_show_maximized_titlebars (void);
 const char*                 meta_prefs_get_theme              (void);
 /* returns NULL if GTK default should be used */
 const PangoFontDescription* meta_prefs_get_titlebar_font      (void);
Index: metacity-2.34.1/src/ui/theme.c
===================================================================
--- metacity-2.34.1.orig/src/ui/theme.c	2011-08-18 00:09:52.879647547 -0400
+++ metacity-2.34.1/src/ui/theme.c	2011-08-18 00:10:05.579953821 -0400
@@ -406,7 +406,8 @@
                                int                   *top_height,
                                int                   *bottom_height,
                                int                   *left_width,
-                               int                   *right_width)
+                               int                   *right_width,
+                               MetaTheme             *theme)
 {
   int buttons_height, title_height;
   
@@ -442,6 +443,9 @@
         *bottom_height = layout->bottom_height;
     }
 
+  if ((flags & META_FRAME_MAXIMIZED) && !theme->show_maximized_titlebars)
+    *top_height = 0;
+
   if (flags & META_FRAME_FULLSCREEN)
     {
       if (top_height)
@@ -646,7 +650,8 @@
                                  &fgeom->top_height,
                                  &fgeom->bottom_height,
                                  &fgeom->left_width,
-                                 &fgeom->right_width);
+                                 &fgeom->right_width,
+                                 theme);
 
   width = client_width + fgeom->left_width + fgeom->right_width;
 
@@ -5025,6 +5030,9 @@
   theme->quark_icon_height = g_quark_from_static_string ("icon_height");
   theme->quark_title_width = g_quark_from_static_string ("title_width");
   theme->quark_title_height = g_quark_from_static_string ("title_height");
+
+  theme->show_maximized_titlebars = TRUE;
+
   return theme;
 }
 
@@ -5449,7 +5457,8 @@
                                  text_height,
                                  flags,
                                  top_height, bottom_height,
-                                 left_width, right_width);
+                                 left_width, right_width,
+                                 theme);
 }
 
 void
@@ -6813,3 +6822,9 @@
       return 1; 
     }
 }
+
+void
+meta_theme_set_show_maximized_titlebars (MetaTheme *theme, gboolean show)
+{
+  theme->show_maximized_titlebars = show;
+}
Index: metacity-2.34.1/src/metacity.schemas.in.in
===================================================================
--- metacity-2.34.1.orig/src/metacity.schemas.in.in	2011-03-08 04:31:53.000000000 -0500
+++ metacity-2.34.1/src/metacity.schemas.in.in	2011-08-18 00:10:05.579953821 -0400
@@ -41,6 +41,20 @@
     </schema>
 
     <schema>
+      <key>/schemas/apps/metacity/general/show_maximized_titlebars</key>
+      <applyto>/apps/metacity/general/show_maximized_titlebars</applyto>
+      <owner>metacity</owner>
+      <type>bool</type>
+      <default>true</default>
+      <locale name="C">
+         <short>Whether to show title bars for maximized windows</short>
+         <long>
+           Set this to false to hide title bars for maximized windows.
+         </long>
+      </locale>
+    </schema>
+
+    <schema>
       <key>/schemas/apps/metacity/general/button_layout</key>
       <applyto>/apps/metacity/general/button_layout</applyto>
       <owner>metacity</owner>
Index: metacity-2.34.1/src/ui/theme.h
===================================================================
--- metacity-2.34.1.orig/src/ui/theme.h	2011-08-18 00:09:52.879647547 -0400
+++ metacity-2.34.1/src/ui/theme.h	2011-08-18 00:10:05.579953821 -0400
@@ -883,6 +883,8 @@
   GQuark quark_icon_height;
   GQuark quark_title_width;
   GQuark quark_title_height;
+
+  gboolean show_maximized_titlebars;
 };
 
 struct _MetaPositionExprEnv
@@ -916,7 +918,8 @@
                                                   int                   *top_height,
                                                   int                   *bottom_height,
                                                   int                   *left_width,
-                                                  int                   *right_width);
+                                                  int                   *right_width,
+                                                  MetaTheme             *theme);
 void             meta_frame_layout_calc_geometry (const MetaFrameLayout  *layout,
                                                   int                     text_height,
                                                   MetaFrameFlags          flags,
@@ -1244,6 +1247,8 @@
 
 guint meta_theme_earliest_version_with_button (MetaButtonType type);
 
+void meta_theme_set_show_maximized_titlebars (MetaTheme *theme, gboolean);
+
 
 #define META_THEME_ALLOWS(theme, feature) (theme->format_version >= feature)
 
