
Index: metacity-2.34.1/src/core/prefs.c
===================================================================
--- metacity-2.34.1.orig/src/core/prefs.c	2011-08-18 00:10:11.320092262 -0400
+++ metacity-2.34.1/src/core/prefs.c	2011-08-18 00:10:19.830297517 -0400
@@ -102,6 +102,7 @@
 static gboolean force_fullscreen = TRUE;
 static gboolean hide_decorator_tooltip = FALSE;
 static gboolean show_maximized_titlebars = TRUE;
+static gboolean workspace_switcher_keyboard_cycle = FALSE;
 
 static MetaVisualBellType visual_bell_type = META_VISUAL_BELL_FULLSCREEN_FLASH;
 static MetaButtonLayout button_layout;
@@ -437,6 +438,11 @@
       &capture_before_unmap,
       FALSE,
     },
+    { "/apps/metacity/general/workspace_switcher_keyboard_cycle",
+      META_PREF_WORKSPACE_SWITCHER_KEYBOARD_CYCLE,
+      &workspace_switcher_keyboard_cycle,
+      FALSE,
+    },
     { NULL, 0, NULL, FALSE },
   };
 
@@ -1803,6 +1809,9 @@
 
     case META_PREF_SHOW_MAXIMIZED_TITLEBARS:
       return "META_PREF_SHOW_MAXIMIZED_TITLEBARS";
+      
+    case META_PREF_WORKSPACE_SWITCHER_KEYBOARD_CYCLE:
+      return "WORKSPACE_SWITCHER_KEYBOARD_CYCLE";
     }
 
   return "(unknown)";
@@ -2758,6 +2767,12 @@
   return compositing_manager;
 }
 
+gboolean
+meta_prefs_get_workspace_switcher_keyboard_cycle (void)
+{
+  return workspace_switcher_keyboard_cycle;
+}
+
 guint
 meta_prefs_get_mouse_button_resize (void)
 {
Index: metacity-2.34.1/src/core/workspace.c
===================================================================
--- metacity-2.34.1.orig/src/core/workspace.c	2011-08-18 00:09:43.479420855 -0400
+++ metacity-2.34.1/src/core/workspace.c	2011-08-18 00:10:19.830297517 -0400
@@ -816,8 +816,9 @@
 {
   MetaWorkspaceLayout layout;  
   int i, current_space, num_workspaces;
-  gboolean ltr;
+  gboolean ltr, cycle;
 
+  cycle = meta_prefs_get_workspace_switcher_keyboard_cycle ();
   current_space = meta_workspace_index (workspace);
   num_workspaces = meta_screen_get_n_workspaces (workspace->screen);
   meta_screen_calc_workspace_layout (workspace->screen, num_workspaces,
@@ -844,14 +845,52 @@
       break;
     }
 
-  if (layout.current_col < 0)
-    layout.current_col = 0;
-  if (layout.current_col >= layout.cols)
-    layout.current_col = layout.cols - 1;
-  if (layout.current_row < 0)
-    layout.current_row = 0;
-  if (layout.current_row >= layout.rows)
-    layout.current_row = layout.rows - 1;
+  if (cycle)
+    {
+      /* right to left */
+      if (layout.current_col < 0)
+        {
+          layout.current_col = layout.cols - 1;
+          if (layout.rows > 1)
+            {
+              /* first row */
+              if (layout.current_row == 0)
+                layout.current_row = layout.rows - 1;
+              else
+                layout.current_row -= 1;
+            }
+        }
+
+      /* left to right */
+      if (layout.current_col >= layout.cols)
+        {
+          layout.current_col = 0;
+
+          if (layout.rows > 1)
+            {
+              /* last row */
+              if (layout.current_row == (layout.rows -1))
+                layout.current_row = 0;
+              else
+                layout.current_row += 1;
+            }
+        }
+      if (layout.current_row < 0)
+        layout.current_row = 0;
+      if (layout.current_row >= layout.rows)
+        layout.current_row = layout.rows - 1;
+    }
+  else
+    {
+      if (layout.current_col < 0)
+        layout.current_col = 0;
+      if (layout.current_col >= layout.cols)
+        layout.current_col = layout.cols - 1;
+      if (layout.current_row < 0)
+        layout.current_row = 0;
+      if (layout.current_row >= layout.rows)
+        layout.current_row = layout.rows - 1;
+    }
 
   i = layout.grid[layout.current_row * layout.cols + layout.current_col];
 
Index: metacity-2.34.1/src/include/prefs.h
===================================================================
--- metacity-2.34.1.orig/src/include/prefs.h	2011-08-18 00:10:11.330092500 -0400
+++ metacity-2.34.1/src/include/prefs.h	2011-08-18 00:10:19.830297517 -0400
@@ -64,7 +64,8 @@
   META_PREF_RESIZE_WITH_RIGHT_BUTTON,
   META_PREF_FORCE_FULLSCREEN,
   META_PREF_HIDE_DECORATOR_TOOLTIP,
-  META_PREF_SHOW_MAXIMIZED_TITLEBARS
+  META_PREF_SHOW_MAXIMIZED_TITLEBARS,
+  META_PREF_WORKSPACE_SWITCHER_KEYBOARD_CYCLE
 } MetaPreference;
 
 typedef void (* MetaPrefsChangedFunc) (MetaPreference pref,
@@ -124,6 +125,8 @@
 gboolean    meta_prefs_get_force_fullscreen  (void);
 gboolean    meta_prefs_get_compositor_effects (void);
 gboolean    meta_prefs_get_capture_before_unmap (void);
+gboolean    meta_prefs_get_workspace_switcher_keyboard_cycle (void);
+
 
 /**
  * Sets whether the compositor is turned on.
Index: metacity-2.34.1/src/metacity.schemas.in.in
===================================================================
--- metacity-2.34.1.orig/src/metacity.schemas.in.in	2011-08-18 00:10:11.330092500 -0400
+++ metacity-2.34.1/src/metacity.schemas.in.in	2011-08-18 00:10:19.830297517 -0400
@@ -337,6 +337,26 @@
     </schema>
 
     <schema>
+      <key>/schemas/apps/metacity/general/workspace_switcher_keyboard_cycle</key>
+      <applyto>/apps/metacity/general/workspace_switcher_keyboard_cycle</applyto>
+      <owner>metacity</owner>
+      <type>bool</type>
+      <default>false</default>
+      <locale name="C">
+        <short>Workspace Switcher Keyboard Cycle</short>
+        <long>
+          Determines if the workspace switcher enables cycling
+          of workspaces using keyboard shortcuts. (typically right and
+          left arrow keys). If true and single row layout, keys will
+          cycle from last to first and first to last workspace when
+          pressed on first or last workspace. If true and multi-row
+          layout, keys will cycle left to right, top to bottom
+          and vice versa.
+        </long>
+      </locale>
+    </schema>
+
+    <schema>
       <key>/schemas/apps/metacity/general/audible_bell</key>
       <applyto>/apps/metacity/general/audible_bell</applyto>      
       <owner>metacity</owner>
