Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Description: Do not initially position windows over the launcher.
Bug: https://bugs.launchpad.net/unity-2d/+bug/688816

Index: metacity-2.30.3/src/core/place.c
===================================================================
--- metacity-2.30.3.orig/src/core/place.c	2010-09-04 18:09:53.000000000 +0200
+++ metacity-2.30.3/src/core/place.c	2011-05-20 18:00:42.809514001 +0200
@@ -138,6 +138,32 @@
   current = meta_screen_get_current_xinerama (window->screen);
   meta_window_get_work_area_for_xinerama (window, current->number, &work_area);
 
+  /* Restrict the work area further to take into account possible dock windows
+     that haven’t reserved struts. This is for example the case of a launcher
+     that sits on the edge of the screen with an intellihide behaviour. */
+  tmp = windows;
+  while (tmp != NULL)
+    {
+      MetaWindow *w = tmp->data;
+      if (w->type == META_WINDOW_DOCK && w->struts == NULL)
+        {
+          MetaRectangle rect;
+          meta_window_get_outer_rect (w, &rect);
+          /* Note: this caters only for the case of a dock on the left edge of
+             the screen. Other cases (right, top, bottom) are not handled at
+             the moment. */
+          if (rect.x == work_area.x && rect.width < work_area.width &&
+              rect.height > rect.width)
+            {
+              if (rect.x + rect.width > work_area.x)
+                {
+                  work_area.x = rect.x + rect.width;
+                }
+            }
+        }
+      tmp = tmp->next;
+    }
+
   cascade_x = MAX (0, work_area.x);
   cascade_y = MAX (0, work_area.y);
   
@@ -389,13 +415,13 @@
 
       switch (other->type)
         {
-        case META_WINDOW_DOCK:
         case META_WINDOW_SPLASHSCREEN:
         case META_WINDOW_DESKTOP:
         case META_WINDOW_DIALOG:
         case META_WINDOW_MODAL_DIALOG:
           break;
 
+        case META_WINDOW_DOCK:
         case META_WINDOW_NORMAL:
         case META_WINDOW_UTILITY:
         case META_WINDOW_TOOLBAR:
