--- src/pl/plpgsql/src/Makefile.orig	2014-07-21 14:10:42.000000000 -0500
+++ src/pl/plpgsql/src/Makefile	2014-08-16 21:21:43.303005362 -0500
@@ -14,7 +14,7 @@
 NAME= plpgsql
 
 override CPPFLAGS := -I. -I$(srcdir) $(CPPFLAGS)
-SHLIB_LINK = $(filter -lintl, $(LIBS))
+SHLIB_LINK = $(filter -lintl, $(LIBS)) ../../../../src/backend/access/common/heaptuple.o ../../../../src/backend/access/heap/tuptoaster.o ../../../../src/backend/executor/execTuples.o
 rpath =
 
 OBJS = pl_gram.o pl_handler.o pl_comp.o pl_exec.o pl_funcs.o pl_scanner.o
