--- postgresql-9.4.0/src/pl/tcl/Makefile.orig	2014-12-15 19:07:34.000000000 -0600
+++ postgresql-9.4.0/src/pl/tcl/Makefile	2015-02-03 09:09:31.504321466 -0600
@@ -32,6 +32,10 @@
 # On Windows, we don't link directly with the Tcl library; see below
 ifneq ($(PORTNAME), win32)
 SHLIB_LINK = $(TCL_LIB_SPEC) $(TCL_LIBS) -lc
+else
+ifeq ($(host_os), mingw32)
+SHLIB_LINK = $(TCL_LIB_SPEC) $(TCL_LIBS)
+endif
 endif
 
 
@@ -51,6 +55,7 @@
 # which are not compatible with mingw gcc. Therefore we need to build a
 # new import library to link with.
 ifeq ($(PORTNAME), win32)
+ifneq ($(host_os), mingw32)
 
 tclwithver = $(subst -l,,$(filter -l%, $(TCL_LIB_SPEC)))
 TCLDLL = $(dir $(TCLSH))/$(tclwithver).dll
@@ -63,6 +68,7 @@
 $(tclwithver).def: $(TCLDLL)
 	pexports $^ > $@
 
+endif # mingw32
 endif # win32
 
 
@@ -117,6 +123,8 @@
 	rm -f $(OBJS)
 	rm -rf $(pg_regress_clean_files)
 ifeq ($(PORTNAME), win32)
+ifneq ($(host_os), mingw32)
 	rm -f $(tclwithver).def
 endif
+endif
 	$(MAKE) -C modules $@
