--- splashy-0.3.13/src/splashy_video.c	2008-12-24 17:14:55.000000000 +0100
+++ splashy-0.3.13-new/src/splashy_video.c	2010-09-02 20:11:13.000000000 +0200
@@ -884,7 +884,8 @@
 
         video.fontdesc.flags = DFDESC_HEIGHT;
 
-        fontface = g_build_filename (splashy_get_config_string (SPL_THEMES_DIR), splashy_get_config_string ("/splashy/textbox/text/font/file"), NULL);
+        /*fontface = g_build_filename (splashy_get_config_string (SPL_THEMES_DIR), splashy_get_config_string ("/splashy/textbox/text/font/file"), NULL);*/
+        fontface = g_build_filename (splashy_get_theme_dir(), splashy_get_config_string ("/splashy/textbox/text/font/file"), NULL);
         temp = splashy_get_config_int ("/splashy/textbox/text/font/height",
                                        10);
 
