--- pr/src/Makefile.in.orig	2008-11-10 16:34:55.000000000 +0000
+++ pr/src/Makefile.in	2008-11-10 17:12:24.000000000 +0000
@@ -193,9 +193,12 @@
 OS_LIBS		= -lsocket -lnsl -lgen -lresolv
 endif
 
-ifeq ($(OS_ARCH),WINNT)
+ifeq ($(OS_TARGET),WINNT)
 ifdef NS_USE_GCC
+IMPORT_LIBRARY  = libnspr4.dll.a
+TARGETS        += libnspr4.dll.a
 OS_LIBS		= -ladvapi32 -lwsock32 -lwinmm
+libnspr4.dll.a: libnspr4.dll
 else
 OS_LIBS		= advapi32.lib wsock32.lib winmm.lib
 endif
--- lib/ds/Makefile.in.orig	2008-11-10 16:45:35.000000000 +0000
+++ lib/ds/Makefile.in	2008-11-10 17:12:51.000000000 +0000
@@ -62,10 +62,10 @@
 
 HEADERS := $(addprefix $(srcdir)/, $(HEADERS))
 
-ifeq ($(OS_ARCH), WINNT)
-RES=$(OBJDIR)/plds.res
-RESNAME=plds.rc
-endif # WINNT
+#ifeq ($(OS_TARGET), WINNT)
+#RES=$(OBJDIR)/plds.res
+#RESNAME=plds.rc
+#endif # WINNT
 
 ifeq ($(OS_ARCH), AIX)
 ifeq ($(CLASSIC_NSPR),1)
@@ -121,7 +121,10 @@
 endif
 
 LIBRARY_NAME	= plds
+IMPORT_LIBRARY  = libplds.dll.a
+OS_LIBS         = -ladvapi32 -lwsock32 -lwinmm
 LIBRARY_VERSION	= $(MOD_MAJOR_VERSION)
+libplds.dll.a: libplds.dll
 
 RELEASE_HEADERS = $(HEADERS)
 RELEASE_HEADERS_DEST = $(RELEASE_INCLUDE_DIR)
--- lib/libc/src/Makefile.in.orig	2008-11-10 16:55:40.000000000 +0000
+++ lib/libc/src/Makefile.in	2008-11-10 16:59:59.000000000 +0000
@@ -71,10 +71,17 @@
 
 RELEASE_LIBS = $(TARGETS)
 
-ifeq ($(OS_ARCH),WINNT)
-RES=$(OBJDIR)/plc.res
-RESNAME=plc.rc
-endif # WINNT
+ifeq ($(OS_TARGET),WINNT)
+IMPORT_LIBRARY  = libplc4.dll.a
+TARGETS        += libplc4.dll.a
+OS_LIBS		= -ladvapi32 -lwsock32 -lwinmm
+libplc4.dll.a: libplc4.dll
+endif
+
+#ifeq ($(OS_TARGET),WINNT)
+#RES=$(OBJDIR)/plc.res
+#RESNAME=plc.rc
+#endif # WINNT
 
 ifeq ($(OS_ARCH), AIX)
 ifeq ($(CLASSIC_NSPR),1)
