diff -ur nspr.orig/config/rules.mk nspr/config/rules.mk
--- nspr.orig/config/rules.mk	2014-08-22 15:01:17.699244639 +0000
+++ nspr/config/rules.mk	2014-08-22 16:34:36.039098503 +0000
@@ -83,12 +83,12 @@
 ifeq (,$(filter-out WIN95 WINCE WINMO OS2,$(OS_TARGET)))
 LIBRARY		= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION)_s.$(LIB_SUFFIX)
 SHARED_LIBRARY	= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY	= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
+IMPORT_LIBRARY ?= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
 SHARED_LIB_PDB	= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).pdb
 else
 LIBRARY		= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION)_s.$(LIB_SUFFIX)
 SHARED_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
+IMPORT_LIBRARY ?= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
 SHARED_LIB_PDB	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).pdb
 endif
 
diff -ur nspr.orig/pr/src/Makefile.in nspr/pr/src/Makefile.in
--- nspr.orig/pr/src/Makefile.in	2014-08-22 15:01:17.709243627 +0000
+++ nspr/pr/src/Makefile.in	2014-08-22 16:31:15.982437156 +0000
@@ -163,9 +163,13 @@
 OS_LIBS		= -lsocket -lc
 endif
 
-ifeq ($(OS_ARCH),WINNT)
+ifeq ($(OS_TARGET),WINNT)
+RELEASE_BINS += libnspr4.dll
+IMPORT_LIBRARY  = libnspr4.dll.a
+TARGETS         += libnspr4.dll.a 
 ifdef NS_USE_GCC
-OS_LIBS		= -ladvapi32 -lws2_32 -lwinmm
+OS_LIBS		= -ladvapi32 -lws2_32 -lwinmm -lmswsock
+libnspr4.dll.a: libnspr4.dll
 else
 OS_LIBS		= advapi32.lib ws2_32.lib winmm.lib
 endif
@@ -291,7 +295,7 @@
 
 LIBRARY_NAME = nspr
 LIBRARY_VERSION = $(MOD_MAJOR_VERSION)
-
+IMPORT_LIBRARY = libnspr4.dll.a
 RELEASE_LIBS = $(TARGETS)
 
 include $(topsrcdir)/config/rules.mk
