diff -ur nspr.orig/config/rules.mk nspr/config/rules.mk
--- nspr.orig/config/rules.mk	2014-08-23 10:44:27.349585007 +0200
+++ nspr/config/rules.mk	2014-08-23 10:44:49.022903720 +0200
@@ -83,12 +83,12 @@
 ifeq (,$(filter-out WIN95 WINCE WINMO OS2,$(OS_TARGET)))
 LIBRARY		= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION)_s.$(LIB_SUFFIX)
 SHARED_LIBRARY	= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY	= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
+IMPORT_LIBRARY ?= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
 SHARED_LIB_PDB	= $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).pdb
 else
 LIBRARY		= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION)_s.$(LIB_SUFFIX)
 SHARED_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
+IMPORT_LIBRARY ?= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(LIB_SUFFIX)
 SHARED_LIB_PDB	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).pdb
 endif
 
diff -ur nspr.orig/lib/ds/Makefile.in nspr/lib/ds/Makefile.in
--- nspr.orig/lib/ds/Makefile.in	2014-08-23 10:44:27.349585007 +0200
+++ nspr/lib/ds/Makefile.in	2014-08-23 10:46:30.379501838 +0200
@@ -86,7 +86,10 @@
 endif
 
 LIBRARY_NAME	= plds
+IMPORT_LIBRARY  = libplds.dll.a
+OS_LIBS         = -ladvapi32 -lwsock32 -lwinmm
 LIBRARY_VERSION	= $(MOD_MAJOR_VERSION)
+libplds.dll.a: libplds.dll
 
 RELEASE_HEADERS = $(HEADERS)
 RELEASE_HEADERS_DEST = $(RELEASE_INCLUDE_DIR)
diff -ur nspr.orig/lib/libc/src/Makefile.in nspr/lib/libc/src/Makefile.in
--- nspr.orig/lib/libc/src/Makefile.in	2014-08-23 10:44:27.349585007 +0200
+++ nspr/lib/libc/src/Makefile.in	2014-08-23 10:47:23.619465719 +0200
@@ -43,6 +43,12 @@
 RES=$(OBJDIR)/plc.res
 RESNAME=plc.rc
 endif # WINNT
+ifeq ($(OS_TARGET),WINNT)
+IMPORT_LIBRARY  = libplc4.dll.a
+TARGETS        += libplc4.dll.a
+OS_LIBS		= -ladvapi32 -lwsock32 -lwinmm
+libplc4.dll.a: libplc4.dll
+endif
 
 ifeq ($(OS_ARCH), AIX)
 ifeq ($(CLASSIC_NSPR),1)
diff -ur nspr.orig/pr/src/Makefile.in nspr/pr/src/Makefile.in
--- nspr.orig/pr/src/Makefile.in	2014-08-23 10:44:27.359585000 +0200
+++ nspr/pr/src/Makefile.in	2014-08-23 10:44:49.022903720 +0200
@@ -163,9 +163,13 @@
 OS_LIBS		= -lsocket -lc
 endif
 
-ifeq ($(OS_ARCH),WINNT)
+ifeq ($(OS_TARGET),WINNT)
+RELEASE_BINS += libnspr4.dll
+IMPORT_LIBRARY  = libnspr4.dll.a
+TARGETS         += libnspr4.dll.a 
 ifdef NS_USE_GCC
-OS_LIBS		= -ladvapi32 -lws2_32 -lwinmm
+OS_LIBS		= -ladvapi32 -lws2_32 -lwinmm -lmswsock
+libnspr4.dll.a: libnspr4.dll
 else
 OS_LIBS		= advapi32.lib ws2_32.lib winmm.lib
 endif
@@ -291,7 +295,7 @@
 
 LIBRARY_NAME = nspr
 LIBRARY_VERSION = $(MOD_MAJOR_VERSION)
-
+IMPORT_LIBRARY = libnspr4.dll.a
 RELEASE_LIBS = $(TARGETS)
 
 include $(topsrcdir)/config/rules.mk
