From d44e96f26193a5141e95892654f7e613e1a2150c Mon Sep 17 00:00:00 2001
From: Danny N <dnoest@gmail.com>
Date: Fri, 7 Dec 2012 01:54:57 -0500
Subject: [PATCH] Fixed crtc bug.

---
 panel.cpp |   20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/panel.cpp b/panel.cpp
index efaccd1..7eaf795 100644
--- a/panel.cpp
+++ b/panel.cpp
@@ -552,6 +552,8 @@ Rectangle Panel::GetPrimaryViewport() {
     XRRScreenResources *resources;
     XRRCrtcInfo *crtc_info;
 
+    int crtc;
+
     fallback.x = 0;
     fallback.y = 0;
     fallback.width = DisplayWidth(Dpy, Scr);
@@ -572,8 +574,24 @@ Rectangle Panel::GetPrimaryViewport() {
         XRRFreeScreenResources(resources);
         return fallback;
     }
+    if (primary_info->crtc < 1) {
+        if (primary_info->ncrtc > 0) {
+            // just use the first one. 
+            // i do not know what the consequences
+            // of this will be. it seems to be that the index
+            // is which monitor to use.  
+            crtc = primary_info->crtcs[0];
+        } else {
+            cerr << "Cannot get crtc from xrandr.\n";
+            exit(EXIT_FAILURE);
+        }
+    } else {
+        crtc = primary_info->crtc;
+    }
+    // this is the function call its dying on 
+    crtc_info = XRRGetCrtcInfo(Dpy, resources, crtc);
+
 
-    crtc_info = XRRGetCrtcInfo(Dpy, resources, primary_info->crtc);
     if (!crtc_info) {
         XRRFreeOutputInfo(primary_info);
         XRRFreeScreenResources(resources);
-- 
1.7.10

