*** source/game/Mod.cpp	2012-09-03 17:55:15.183644949 -0500
--- Mod.cpp.change	2012-09-03 19:28:30.733756915 -0500
***************
*** 2,8 ****
  #include "core/File.h"
  #include "game/Mod.h"
  #include <sstream>
! #define BOOST_FILESYSTEM_VERSION 2
  #include <boost/filesystem.hpp>
  
  #if NERO_PLATFORM_WINDOWS
--- 2,8 ----
  #include "core/File.h"
  #include "game/Mod.h"
  #include <sstream>
! #define BOOST_FILESYSTEM_VERSION 3
  #include <boost/filesystem.hpp>
  
  #if NERO_PLATFORM_WINDOWS
