*** source/core/File.cpp	2012-09-03 19:24:15.917334798 -0500
--- File.cpp.change	2012-09-03 19:25:25.933351962 -0500
***************
*** 6,12 ****
  #include "core/Common.h"
  #include "File.h"
  #include <fstream>
! #define BOOST_FILESYSTEM_VERSION 2
  #include <boost/filesystem.hpp>   // includes all needed Boost.Filesystem declarations
  namespace fs = boost::filesystem;
  
--- 6,13 ----
  #include "core/Common.h"
  #include "File.h"
  #include <fstream>
! #define BOOST_FILESYSTEM_VERSION 3
! #define BOOST_FILESYSTEM_DEPRECATED 1
  #include <boost/filesystem.hpp>   // includes all needed Boost.Filesystem declarations
  namespace fs = boost::filesystem;
  
