--- otherlibs/octomap/include/octomap/OcTreeBaseImpl.h	2013-05-10 12:57:48.035571958 -0300
+++ otherlibs/octomap/include/octomap/OcTreeBaseImpl.h	2013-05-10 12:59:01.154607123 -0300
@@ -151,9 +151,6 @@ namespace octomap {
     /// Deletes the complete tree structure (only the root node will remain)
     void clear();
 
-    OcTreeBaseImpl deepCopy() const;
-
-
     /// Lossless compression of OcTree: merge children to parent when there are
     /// eight children with identical values
     virtual void prune();
