diff -ur picasa-orig/opt/google/picasa/3.0/bin/picasa picasa/opt/google/picasa/3.0/bin/picasa
--- picasa-orig/opt/google/picasa/3.0/bin/picasa	2008-09-16 13:38:03.000000000 -0600
+++ picasa/opt/google/picasa/3.0/bin/picasa	2008-10-20 19:01:45.000000000 -0600
@@ -2,6 +2,12 @@
 #   picasa
 #   Simple script to launch Picasa
 
+# Run using linux32 if architecture is x86_64
+if [ "`uname -m`" = "x86_64" ] ; then
+	/usr/bin/linux32 "$0" "$@"
+	exit $?
+fi
+
 close_splash() {
   if [ -n "$SPLASH_PID" ]; then
     kill "$SPLASH_PID" > /dev/null 2>&1
@@ -128,35 +134,6 @@
     WINEPREFIX="$PICASA_WINEPREFIX"
 fi
 
-# Check for the 32-bit OpenSSL libs that Picasa needs.
-if [ "`uname -m`" = "x86_64" ] ; then
-    SSLSETUP="$PIC_BINDIR/repackage32.sh"
-    SSLPROMPT="Many online features of Picasa will not work
-until you install a 32-bit version of OpenSSL.
-If you skip this step, you can perform it later by running:
-$SSLSETUP openssl
-For additional assistance, please visit:
-http://groups.google.com/group/Google-Labs-Picasa-for-Linux
-
-Would you like to install 32-bit OpenSSL now?"
-    "$SSLSETUP" -p "$SSLPROMPT" openssl
-
-    MDNSFIX="$PIC_BINDIR"/mdns.sh
-    MDNSPROMPT="
-With certain 64-bit Linux configuration, you may not be
-able to access Picasa's internet features unless you install
-the lib32nss-mdns package.
-
-If you skip this step, you can perform it later by running:
-$MDNSFIX
-For additional assistance, please visit:
-http://groups.google.com/group/Google-Labs-Picasa-for-Linux
-
-Would you like to install the lib32nss-mdns package now?
-"
-  "$MDNSFIX" -p "$MDNSPROMPT" -c "$WINEPREFIX/.mdnsfix"
-fi
-
 # Try to avoid Wine crashes due to font problem, as described at:
 # http://www.codeweavers.com/support/wiki/faq/fontissues
 BADVER="Version 0.1.20060928"
Only in picasa/opt/google/picasa/3.0/bin: picasa~
diff -ur picasa-orig/opt/google/picasa/3.0/bin/wrapper picasa/opt/google/picasa/3.0/bin/wrapper
--- picasa-orig/opt/google/picasa/3.0/bin/wrapper	2008-09-16 13:38:03.000000000 -0600
+++ picasa/opt/google/picasa/3.0/bin/wrapper	2008-10-20 18:57:47.000000000 -0600
@@ -335,6 +335,7 @@
 
 # Set the environment up
 export PATH=$parentdir/wine/bin:$mydir:"$mydir/xdg-utils-1.0.2/scripts":"$mydir/xdg-user-dirs-0.8/bin":$PATH
+[ "`uname -m`" = "x86_64" ] && export LD_LIBRARY_PATH=/opt/lib32/usr/lib/:/opt/lib32/lib/:$LD_LIBRARY_PATH
 export LD_LIBRARY_PATH=$parentdir/wine/lib:$parentdir/lib:$LD_LIBRARY_PATH
 export WINEDLLPATH=$parentdir/wine/lib/wine
 export WINELOADER=$parentdir/wine/bin/wine
