--- modules/libpr0n/encoders/png/nsPNGEncoder.cpp	2009-05-02 11:39:49.000000000 -0500
+++ modules/libpr0n/encoders/png/nsPNGEncoder1.cpp	2010-02-23 09:49:38.902117049 -0600
@@ -111,9 +111,10 @@ NS_IMETHODIMP nsPNGEncoder::InitFromData
 
   // initialize
   png_struct* png_ptr = png_create_write_struct(PNG_LIBPNG_VER_STRING,
-                                                png_voidp_NULL,
-                                                png_error_ptr_NULL,
-                                                png_error_ptr_NULL);
+                                                NULL,
+                                                NULL,
+                                                NULL);
+
   if (! png_ptr)
     return NS_ERROR_OUT_OF_MEMORY;
   png_info* info_ptr = png_create_info_struct(png_ptr);
