diff -ru bombic-0.0.1-src.orig/src/D3DXApp.cpp bombic-0.0.1-src/src/D3DXApp.cpp
--- bombic-0.0.1-src.orig/src/D3DXApp.cpp	2005-03-19 09:28:28.000000000 +0100
+++ bombic-0.0.1-src/src/D3DXApp.cpp	2010-07-08 19:42:14.000000000 +0200
@@ -229,8 +229,6 @@
 
 bool load_samples()
 {	
-	
-/*
 	#define SND_MENU_CHNG         0 
 	#define SND_MENU_SEL          1
 	#define SND_GAME_EXPLODE      2
@@ -239,10 +237,10 @@
 	#define SND_GAME_BONUS        5
 	#define SND_GAME_TIME_BONUS   6
 	#define SND_GAME_DEAD_BOMBER  7
+/*
 	napalmbomba   8
 	9 - psani na stroji
 */
-/*
 	g_sb[0].LoadFromRes("_menu1.wav");
 	g_sb[1].LoadFromRes("_menu2.wav");
 	g_sb[SND_GAME_EXPLODE].LoadFromRes("bomba.wav");
@@ -270,6 +268,6 @@
 //	g_sb[22].LoadFromRes("SMRT_MESIC");
 //	g_sb[23].LoadFromRes("SMRT_MESIC");
 	g_sb[24].LoadFromRes("_napalm.wav"); // cervenokulka
-*/
+
 	return true;
 }
diff -ru bombic-0.0.1-src.orig/src/HDSound.cpp bombic-0.0.1-src/src/HDSound.cpp
--- bombic-0.0.1-src.orig/src/HDSound.cpp	2004-12-19 15:19:27.000000000 +0100
+++ bombic-0.0.1-src/src/HDSound.cpp	2010-07-08 19:43:41.000000000 +0200
@@ -61,12 +61,12 @@
 		}
 // open 44.1KHz, signed 16bit, system byte order,
 //      stereo audio, using 1024 byte chunks
-/*	if(Mix_OpenAudio(MIX_DEFAULT_FREQUENCY, MIX_DEFAULT_FORMAT, 2, 2048)==-1) {
+        if(Mix_OpenAudio(MIX_DEFAULT_FREQUENCY, MIX_DEFAULT_FORMAT, 2, 2048)==-1) {
 		cerr << "Mix_OpenAudio: " <<  Mix_GetError() << endl;
 		return false;
 	}
-*/	
-//	Mix_AllocateChannels(16);
+
+        Mix_AllocateChannels(16);
 	
 	return true;
 	
@@ -103,13 +103,11 @@
 
 bool HDSoundBuffer::LoadFromRes(const char *resname)
 {
-/*
 	const char *c = locate_file(resname);
 	if (c == NULL) return false;
 	m_sample = Mix_LoadWAV(c);
 	m_sample_channel = -1;
-	return true;
-*/	
+        return true;
 }
 
 void HDSoundBuffer::SetPan(int pan)
--- bombic-0.0.1-src.orig/data/samples/Makefile.in	2005-03-19 11:53:05.000000000 +0100
+++ bombic-0.0.1-src/data/samples/Makefile.in	2010-07-08 19:59:12.000000000 +0200
@@ -160,7 +160,19 @@
 B5.wav \
 megabomba.wav \
 smrt.wav \
-test.wav
+test.wav \
+bonus_casovy.wav \
+bonus.wav \
+_menu1.wav \
+_menu2.wav \
+_napalm.wav \
+_polozeni_bomby.wav \
+_smrt_dudlik.wav \
+_smrt_koule.wav \
+_smrt_mesicka.wav \
+_smrt_slizky.wav \
+_smrt1.wav \
+_smrt2.wav
 
 all: all-am
 
