--- makefile.lnx~	2009-10-08 00:19:03.050892802 +0600
+++ makefile.lnx	2009-10-08 00:19:27.637558977 +0600
@@ -29,7 +29,7 @@
 # omit frame pointer option needed for 25% speed improvment
 OPTFLGS=-fno-strength-reduce -fomit-frame-pointer
 # option good in general, change if you know your X86 architecture
-ARCHFLGS= -march=pentiumpro
+ARCHFLGS= -march=i686
 
 # select one of the sets of C flags - comment out others
 
--- makefile.amd64~	2009-10-08 00:34:52.874467958 +0600
+++ makefile.amd64	2009-10-08 00:35:15.477463911 +0600
@@ -29,19 +29,19 @@
 # omit frame pointer option needed for 25% speed improvment
 OPTFLGS=-fno-strength-reduce -fomit-frame-pointer
 # option good in general, change if you know your X86 architecture
-ARCHFLGS= -march=pentiumpro
+ARCHFLGS= -march=x86-64
 
 # select one of the sets of C flags - comment out others
 
 # cflags when making optimized cver
-CFLAGS= $(ARCHFLGS) -pipe $(WARNS) $(INCS) $(OPTFLGS) -O2 -m32
+CFLAGS= $(ARCHFLGS) -pipe $(WARNS) $(INCS) $(OPTFLGS) -O2
 
 # cflags when making -g debugging version 
-# CFLAGS= $(ARCHFLGS) -pipe $(WARNS) $(INCS) -g -m32
+# CFLAGS= $(ARCHFLGS) -pipe $(WARNS) $(INCS) -g
 
 # cflags when making malloc and -g debugging version 
 # assumes dbmalloc-14L2.tgz .a lib and malloc.h installed one level up
-# CFLAGS= $(ARCHFLGS) -pipe $(WARNS) $(INCS) -D__DBMALLOC__ -g -m32
+# CFLAGS= $(ARCHFLGS) -pipe $(WARNS) $(INCS) -D__DBMALLOC__ -g
 
 # change gcc version if needed 
 CC=gcc
