--- /home/abaddon/Documents/AUR/vokoscreen-git/src/vokoscreen/applications/vokoscreen.desktop	2014-12-08 14:18:31.009971545 -0600
+++ /home/abaddon/Documents/AUR/vokoscreen-git/src/vokoscreen/applications/vokoscreen.desktop.fix	2014-12-08 14:19:59.770014215 -0600
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Comment=screencast
+Comment=Screencast creator
 Exec=vokoscreen
 Icon=vokoscreen
 Name=vokoscreen
@@ -7,4 +7,4 @@
 Terminal=false
 Type=Application
 Categories=AudioVideo;Recorder;
-Keywords=Audio;Video;Recorder;Screencast;
+Keywords=Audio;Video;Recorder;Screencast;
\ No newline at end of file
