--- a/applications/vokoscreen.desktop	2015-03-17 12:40:28.387652229 -0600
+++ b/applications/vokoscreen.desktop	2015-03-17 12:41:54.677613007 -0600
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Comment=screencast
+Comment=An easy to use screencast creator
 Exec=vokoscreen
 Icon=vokoscreen
 Name=vokoscreen
@@ -7,4 +7,4 @@ StartupNotify=false
 Terminal=false
 Type=Application
 Categories=AudioVideo;Recorder;
-Keywords=Audio;Video;Recorder;Screencast;
+Keywords=Audio;Video;Recorder;Screencast;
\ No newline at end of file
