--- vanilla/bison-mode.el	2013-04-02 23:10:46.240738359 +0200
+++ patched/bison-mode.el	2013-04-02 23:12:56.127410548 +0200
@@ -85,9 +85,9 @@
 ;; *************** dependencies ***************
 
 (require 'derived)			;; define-derived-mode
-;(require 'flex-mode)			;; for flex-mode derivation
+(require 'flex-mode)			;; for flex-mode derivation
 (require 'make-regexp)			;; make-regexp
-
+(require 'cc-fonts)	            ;; c-font-lock-keywords
 
 ;; *************** internal vars ***************
 
@@ -986,4 +986,4 @@
 
 (provide 'bison-mode)
 
-;; *************** end of code ***************
\ No newline at end of file
+;; *************** end of code ***************
