--- systemd/apt-cacher-ng.service	2013-09-18 21:51:06.000000000 +0200
+++ systemd/apt-cacher-ng.service.arch	2013-11-26 22:13:36.007264116 +0100
@@ -4,10 +4,10 @@
 After=network.target
 
 [Service]
-ExecStart=/usr/sbin/apt-cacher-ng SocketPath=/var/run/apt-cacher-ng/socket -c /etc/apt-cacher-ng ForeGround=1
+ExecStart=/usr/bin/apt-cacher-ng SocketPath=/run/apt-cacher-ng/socket -c /etc/apt-cacher-ng ForeGround=1
 # this ExecStopPost command can be removed together with the SocketPath option
 # above if the bridge functionality is not needed
-ExecStopPost=/bin/rm -f /var/run/apt-cacher-ng/*
+ExecStopPost=/bin/rm -f /run/apt-cacher-ng/*
 User=apt-cacher-ng
 Group=apt-cacher-ng
 
