--- install.rdf.orig	2010-08-02 00:01:12.000000000 +0200
+++ install.rdf	2011-07-14 16:01:45.000000000 +0200
@@ -17,7 +17,7 @@
       <Description>
        <em:id>{3550f703-e582-4d05-9a08-453d09bdfdc6}</em:id>
        <em:minVersion>2.0</em:minVersion>
-       <em:maxVersion>3.1.*</em:maxVersion>
+       <em:maxVersion>5.*</em:maxVersion>
       </Description>
     </em:targetApplication>
 
