--- unmscheme.c.orig	2013-12-29 15:47:25.774866875 +0800
+++ unmscheme.c	2013-12-29 15:50:46.274863256 +0800
@@ -452,13 +452,10 @@
   int i;
   sexpr *input, *result;
 
-  /* crashes under lion */
-  /*
-  unmscheme_directory = strdup(dirname(argv[0]));
-  */
-
-  unmscheme_directory = strdup(argv[0]);
-  unmscheme_directory[strlen(unmscheme_directory) - strlen("/unmscheme")] = '\0';
+	if ( getenv("UNMSCHEME_HOME") == NULL )
+  	unmscheme_directory = "/usr/share/unmscheme";
+	else
+  	unmscheme_directory = getenv("UNMSCHEME_HOME");
 
   gargc = argc;
   gargv = argv;
