--- src/skulltag/skulltagmain.cpp	2011-12-04 16:56:33.881561048 -0800
+++ src/skulltag/skulltagmain.cpp.new	2011-12-04 16:58:08.361554958 -0800
@@ -139,7 +139,7 @@
 		void						masterHost(QString &host, unsigned short &port) const
 		{
 			QString str = pConfig->setting("Masterserver");
-			Strings::translateServerAddress(str, host, port, "skulltag.servegame.com", 15300);
+			Strings::translateServerAddress(str, host, port, "master.skulltag.net", 15300);
 		}
 
 		void						registerIRCServer(QVector<IRCNetworkEntity> &networks) const
@@ -187,6 +187,6 @@
 	config.createSetting("ServerBinaryPath", pf.findFile("skulltag-server"));
 #endif
 
-	config.createSetting("Masterserver", "skulltag.servegame.com:15300");
+	config.createSetting("Masterserver", "master.skulltag.net:15300");
 	config.createSetting("EnableTesting", true);
 }
