diff -Nur asterisk-1.6.1.5.orig/Makefile asterisk-1.6.1.5/Makefile
--- asterisk-1.6.1.5.orig/Makefile	2009-08-05 00:54:45.000000000 +1000
+++ asterisk-1.6.1.5/Makefile	2009-09-11 10:43:29.193684531 +1000
@@ -747,11 +747,11 @@
 	echo "PROJECT_NUMBER=$(ASTERISKVERSION)") | doxygen - 
 
 install-logrotate:
-	if [ ! -d $(ASTETCDIR)/../logrotate.d ]; then \
-		mkdir $(ASTETCDIR)/../logrotate.d ; \
+	if [ ! -d $(DESTDIR)$(ASTETCDIR)/../logrotate.d ]; then \
+		mkdir $(DESTDIR)$(ASTETCDIR)/../logrotate.d ; \
 	fi
 	sed 's#__LOGDIR__#$(ASTLOGDIR)#g' < contrib/scripts/asterisk.logrotate | sed 's#__SBINDIR__#$(ASTSBINDIR)#g' > contrib/scripts/asterisk.logrotate.tmp
-	install -m 0644 contrib/scripts/asterisk.logrotate.tmp $(ASTETCDIR)/../logrotate.d/asterisk
+	install -m 0644 contrib/scripts/asterisk.logrotate.tmp $(DESTDIR)$(ASTETCDIR)/../logrotate.d/asterisk
 	rm -f contrib/scripts/asterisk.logrotate.tmp
 
 config:
diff -Nur asterisk-1.6.1.5.orig/contrib/scripts/asterisk.logrotate asterisk-1.6.1.5/contrib/scripts/asterisk.logrotate
--- asterisk-1.6.1.5.orig/contrib/scripts/asterisk.logrotate	2008-07-24 15:21:00.000000000 +1000
+++ asterisk-1.6.1.5/contrib/scripts/asterisk.logrotate	2009-09-11 10:45:13.377002211 +1000
@@ -5,7 +5,7 @@
 	compress
 	delaycompress
 	notifempty
-	create 640 root root 
+	create 640 asterisk asterisk
 	sharedscripts
 	postrotate
 		__SBINDIR__/asterisk -rx 'logger reload' > /dev/null 2> /dev/null	
