diff -Nur asterisk-1.6.1.5.orig/Makefile asterisk-1.6.1.5/Makefile
--- asterisk-1.6.1.5.orig/Makefile	2009-08-05 00:54:45.000000000 +1000
+++ asterisk-1.6.1.5/Makefile	2009-09-11 10:47:48.643658740 +1000
@@ -638,7 +638,7 @@
 	@if [ "$(OVERWRITE)" = "y" ] || [ ! -f $(DESTDIR)$(ASTCONFPATH) ]; then \
 		echo "Creating asterisk.conf"; \
 		( \
-		echo "[directories](!) ; remove the (!) to enable this" ; \
+		echo "[directories]" ; \
 		echo "astetcdir => $(ASTETCDIR)" ; \
 		echo "astmoddir => $(MODULES_DIR)" ; \
 		echo "astvarlibdir => $(ASTVARLIBDIR)" ; \
