--- mozilla.orig/config/system-headers	2010-04-16 17:31:32.000000000 +0100
+++ mozilla/config/system-headers	2010-04-23 00:44:38.560196628 +0100
@@ -267,6 +267,7 @@
 freetype/freetype.h
 freetype/ftcache.h
 freetype/ftglyph.h
+freetype/ftlcdfil.h
 freetype/ftsynth.h
 freetype/ftoutln.h
 freetype/ttnameid.h
--- a/configure.in	Tue Jan 04 10:40:46 2011 -0800
+++ b/configure.in	Wed Jan 05 00:18:26 2011 +0100
@@ -3555,7 +3555,7 @@
                            $HAVE_FT_BITMAP_SIZE_Y_PPEM,
                            [FT_Bitmap_Size structure includes y_ppem field])
 
-        AC_CHECK_FUNCS(FT_GlyphSlot_Embolden FT_Load_Sfnt_Table FT_Select_Size)
+        AC_CHECK_FUNCS(FT_GlyphSlot_Embolden FT_Load_Sfnt_Table FT_Select_Size FT_Library_SetLcdFilter)
 
     	LIBS="$_SAVE_LIBS"
     	CFLAGS="$_SAVE_CFLAGS"
