diff --git a/sbt b/sbt
index 540b6e7..7c39d28 100755
--- a/sbt
+++ b/sbt
@@ -122,8 +122,8 @@ declare -r script_name="$(basename $script_path)"
 
 # some non-read-onlies set with defaults
 declare java_cmd=java
-declare sbt_launch_dir="$script_dir/.lib"
-declare sbt_universal_launcher="$script_dir/lib/sbt-launch.jar"
+declare sbt_launch_dir="$HOME/sbt-dir/.lib"
+declare sbt_universal_launcher="$HOME/sbt-dir/lib/sbt-launch.jar"
 declare sbt_mem=$default_sbt_mem
 declare sbt_jar=$sbt_universal_launcher
 declare trace_level=$default_trace_level
