--- encrypt_install	2011-06-30 19:57:13.000000000 +0200
+++ encrypt_install	2011-07-30 09:19:15.251305984 +0200
@@ -18,2 +18,9 @@
     add_file "/lib/initcpio/udev/11-dm-initramfs.rules" "/lib/udev/rules.d/11-dm-initramfs.rules"
+    PLYMOUTH_PLUGIN_PATH="$(plymouth --get-splash-plugin-path)"
+    add_binary ${PLYMOUTH_PLUGIN_PATH}/label.so
+    add_binary ${PLYMOUTH_PLUGIN_PATH}/text.so
+    add_file "/usr/share/fonts/TTF/DejaVuSans.ttf"
+    add_file "/etc/fonts/fonts.conf"
+    add_file "/etc/fonts/conf.d/60-latin.conf"
+    add_file "/usr/lib/pango/1.6.0/modules/pango-basic-x.so"
 }
