--- systemd-units/systemd-ask-password-plymouth.path.in	2012-07-15 13:13:50.424285560 +0300
+++ systemd-units/systemd-ask-password-plymouth.path.ina	2012-07-15 13:16:56.015043174 +0300
@@ -6,7 +6,7 @@
 After=plymouth-start.service
 Before=basic.target shutdown.target
 ConditionKernelCommandLine=!plymouth.enable=0
-ConditionPathExists=/run/plymouth/pid
+ConditionPathExists=/run/plymouth.pid
 
 [Path]
 DirectoryNotEmpty=/run/systemd/ask-password
