diff --git a/src/Padre-Plugin-Autodia-0.01/lib/Padre/Plugin/Autodia.pm b/src/Padre-Plugin-Autodia-0.01/lib/Padre/Plugin/Autodia.pm
index 5256968..088e5de 100644
--- a/src/Padre-Plugin-Autodia-0.01/lib/Padre/Plugin/Autodia.pm
+++ b/src/Padre-Plugin-Autodia-0.01/lib/Padre/Plugin/Autodia.pm
@@ -20,7 +20,7 @@ use Cwd;
 use Autodia;
 
 use Padre::Wx         ();
-use Padre::Wx::Dialog ();
+use Padre::Wx::Role::Dialog ();
 use Padre::Util       ();
 use Padre::Current    ();
 
diff --git a/src/Padre-Plugin-Autodia-0.01/lib/Padre/Plugin/Autodia.pm b/src/Padre-Plugin-Autodia-0.01/lib/Padre/Plugin/Autodia.pm
index 5256968..af4a5d5 100644
--- a/src/Padre-Plugin-Autodia-0.01/lib/Padre/Plugin/Autodia.pm
+++ b/src/Padre-Plugin-Autodia-0.01/lib/Padre/Plugin/Autodia.pm
@@ -114,8 +114,8 @@ sub draw_this_file {
 
     my $outfile = "${filename}.draw_this_file.jpg";
 
-    (my $language = lc($document->get_mimetype) ) =~ s|application/[x\-]*||;
-    warn "language : $language, mimetype : ",$document->get_mimetype, "\n";
+    (my $language = lc($document->mimetype) ) =~ s|application/[x\-]*||;
+    warn "language : $language, mimetype : ",$document->mimetype, "\n";
 
     my $autodia_handler = $self->_get_handler({filenames => [ $filename ], outfile => $outfile, graphviz => 1, language => $language});
 
