--- spoon.sh	2012-11-21 22:07:36.000000000 +0100
+++ spoon-patched.sh	2013-08-04 16:25:43.144894608 +0200
@@ -6,13 +6,6 @@
 # ** Gtk2 and Fte enabled.
 # **************************************************
 
-# set MOZILLA_FIVE_HOME=/usr/local/mozilla
-# set LD_LIBRARY_PATH=/usr/local/mozilla
-
-# Try to guess xulrunner location - change this if you need to
-MOZILLA_FIVE_HOME=$(find /usr/lib -maxdepth 1 -name xulrunner-[0-9]* | head -1)
-LD_LIBRARY_PATH=${MOZILLA_FIVE_HOME}:${LD_LIBRARY_PATH}
-export MOZILLA_FIVE_HOME LD_LIBRARY_PATH
 
 # Fix for GTK Windows issues with SWT
 export GDK_NATIVE_WINDOWS=1
@@ -27,7 +20,7 @@
 BASEDIR=`dirname $0`
 cd $BASEDIR
 DIR=`pwd`
-cd -
+cd - > /dev/null
 
 . "$DIR/set-pentaho-env.sh"
 
@@ -155,7 +148,7 @@
     PENTAHO_DI_JAVA_OPTIONS="-Xmx512m -XX:MaxPermSize=256m"
 fi
 
-OPT="$OPT $PENTAHO_DI_JAVA_OPTIONS -Djava.library.path=$LIBPATH -DKETTLE_HOME=$KETTLE_HOME -DKETTLE_REPOSITORY=$KETTLE_REPOSITORY -DKETTLE_USER=$KETTLE_USER -DKETTLE_PASSWORD=$KETTLE_PASSWORD -DKETTLE_PLUGIN_PACKAGES=$KETTLE_PLUGIN_PACKAGES -DKETTLE_LOG_SIZE_LIMIT=$KETTLE_LOG_SIZE_LIMIT"
+OPT="$OPT -Dorg.eclipse.swt.browser.DefaultType=webkit $PENTAHO_DI_JAVA_OPTIONS -Djava.library.path=$LIBPATH:/usr/lib -DKETTLE_HOME=$KETTLE_HOME -DKETTLE_REPOSITORY=$KETTLE_REPOSITORY -DKETTLE_USER=$KETTLE_USER -DKETTLE_PASSWORD=$KETTLE_PASSWORD -DKETTLE_PLUGIN_PACKAGES=$KETTLE_PLUGIN_PACKAGES -DKETTLE_LOG_SIZE_LIMIT=$KETTLE_LOG_SIZE_LIMIT"
 
 # ***************
 # ** Run...    **
