--- configure.ac	2014-08-08 21:56:02.221368250 -0700
+++ configure_new.ac	2014-08-08 22:04:29.588015883 -0700
@@ -20,8 +20,6 @@
 dnl Check for programs.
 AC_PROG_CC
 AC_PROG_CXX
-AC_CHECK_PROG(DOXYGEN, doxygen, yes,)
-AM_CONDITIONAL(build_reference, test "x$DOXYGEN" = "xyes")
 
 dnl Check for internationalization.
 AM_GNU_GETTEXT
@@ -66,35 +64,12 @@
     AC_DEFINE([HAVE_POPPLER_0_17_0], [1], [Define to 1 if you have Poppler version 0.15.0 or higher.])
 fi
 
+AM_CONDITIONAL([OS_WIN32], false)
 
-AC_MSG_CHECKING([for native Win32])
-case "$host" in
-  *-*-mingw*)
-    os_win32=yes
-    ;;
-  *)
-    os_win32=no
-    ;;
-esac
-AC_MSG_RESULT([$os_win32])
-
-AM_CONDITIONAL([OS_WIN32],[test "$os_win32" = "yes"])
-
-if test "$os_win32" != "yes"; then
-  EPDFVIEW_PATH_CUPS([CUPS], [have_cups=yes])
-  AM_CONDITIONAL(cups_printing, test "x$have_cups" = "xyes")
-  if test "x$have_cups" = "xyes"; then
-      AC_SUBST([CUPS_CFLAGS])
-      AC_SUBST([CUPS_LIBS])
-      AC_DEFINE([HAVE_CUPS], [1], [Define to 1 if you have the CUPS API library.])
-  fi
-else
-  AM_CONDITIONAL(cups_printing, false)
-fi
+AM_CONDITIONAL(cups_printing, false)
 
 dnl CPPUnit library
-AM_PATH_CPPUNIT([1.10.2], CPPUNIT="yes")
-AM_CONDITIONAL(build_tests, test "x$CPPUNIT" = "xyes" -a "x$enable_debug" = "xyes")
+AM_CONDITIONAL(build_tests, false)
 
 dnl Check for header files.
 dnl Check for types.
@@ -109,8 +84,5 @@
 intl/Makefile   \
 po/Makefile.in  \
 src/Makefile    \
-src/gtk/Makefile    \
-tests/Makefile  \
-doc/Makefile    \
-doc/Doxyfile])
+src/gtk/Makefile])
 AC_OUTPUT
