--- smw-old/_src/map.cpp	2009-12-20 16:17:27.654541498 -0600
+++ smw/_src/map.cpp	2009-12-20 16:19:24.557028628 -0600
@@ -628,7 +628,7 @@
 			
 			for(short iBackground = 0; iBackground < 26; iBackground++)
 			{
-				char * szFindUnderscore = strstr(g_szBackgroundConversion[iBackground], "_");
+				const char * szFindUnderscore = strstr(g_szBackgroundConversion[iBackground], "_");
 
 				if(szFindUnderscore)
 					szFindUnderscore++;
