diff -ru libindi_3rdparty_0.9.9-orig/indi-qhy/CMakeLists.txt libindi_3rdparty_0.9.9/indi-qhy/CMakeLists.txt
--- libindi_3rdparty_0.9.9-orig/indi-qhy/CMakeLists.txt	2014-08-30 13:02:24.000000000 +0200
+++ libindi_3rdparty_0.9.9/indi-qhy/CMakeLists.txt	2015-02-04 23:00:44.152088317 +0100
@@ -2,11 +2,11 @@
 PROJECT(qhyccd CXX)
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules/")
-if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-  set(FIRMWARE_INSTALL_DIR "/lib/firmware")
-else ()
+#if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+#  set(FIRMWARE_INSTALL_DIR "/lib/firmware")
+#else ()
   set(FIRMWARE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib/firmware")
-endif ()
+#endif ()
 set(UDEVRULES_INSTALL_DIR "/etc/udev/rules.d")
 
 set (VERSION_MAJOR 1)
