--- dmenu-4.2.1/dmenu.c	2010-11-20 03:24:33.000000000 -0600
+++ dmenu-4.2.1-patched/dmenu.c	2010-12-07 02:53:25.376666673 -0600
@@ -11,6 +11,7 @@
 #include <X11/extensions/Xinerama.h>
 #endif
 #include "draw.h"
+#include <wordexp.h>
 
 #define INRECT(x,y,rx,ry,rw,rh) ((x) >= (rx) && (x) < (rx)+(rw) && (y) >= (ry) && (y) < (ry)+(rh))
 #define MIN(a,b)                ((a) < (b) ? (a) : (b))
@@ -37,6 +38,7 @@
 static void run(void);
 static void setup(void);
 static void usage(void);
+static void matchfile(char *filestart);
 
 static char text[BUFSIZ];
 static int bh, mw, mh;
@@ -357,6 +359,10 @@
 		}
 		break;
 	case XK_Tab:
+		if( strchr(text, ' ')!=NULL ) {
+			matchfile( strchr(text, ' ')+1 );
+			break;
+		}
 		if(!sel)
 			return;
 		strncpy(text, sel->text, sizeof text);
@@ -431,6 +437,35 @@
 }
 
 void
+matchfile(char *filestart) {
+	wordexp_t exp;
+	int i, j, k, p=strlen(filestart);
+	filestart[ p+1 ] = 0;
+	filestart[ p ] = '*';
+
+	wordexp(filestart, &exp, 0);
+	if( exp.we_wordc > 0 ) {
+		for(j=0,i=0; exp.we_wordv[0][i]!=0; i++,j++) {
+			if( exp.we_wordv[0][i]==' ' ) filestart[j++]='\\';
+			filestart[j]=exp.we_wordv[0][i];
+		}
+		filestart[j]=0;
+
+		for(k=1; k<exp.we_wordc; k++)
+			for(j=0, i=0; exp.we_wordv[k][i]; i++,j++) {
+				if( filestart[j]=='\\' ) j++;
+				if( filestart[j]!=exp.we_wordv[k][i] ) {
+					filestart[j]=0;
+					break;
+				}
+			}
+	} else {
+		filestart[ p ] = 0;
+	}
+	wordfree(&exp);
+}
+
+void
 readstdin(void) {
 	char buf[sizeof text], *p;
 	Item *item, **end;
