diff -ur reprepro-4.3.0.orig//database.c reprepro-4.3.0/database.c
--- reprepro-4.3.0.orig//database.c	2010-07-11 13:15:59.000000000 +0300
+++ reprepro-4.3.0/database.c	2010-12-23 16:51:36.570000651 +0300
@@ -233,7 +233,7 @@
 		}
 	}
 
-#if DB_VERSION_MAJOR == 4
+#if DB_VERSION_MAJOR >= 4
 #define DB_OPEN(database,filename,name,type,flags) database->open(database,NULL,filename,name,type,flags,0664)
 #else
 #if DB_VERSION_MAJOR == 3
