diff -aur src/wxLua/modules/wxbind/src/wxcore_image.cpp src2//wxLua/modules/wxbind/src/wxcore_image.cpp
--- src/wxLua/modules/wxbind/src/wxcore_image.cpp	2009-03-31 06:23:42.000000000 +0200
+++ src2//wxLua/modules/wxbind/src/wxcore_image.cpp	2011-05-06 17:32:46.000000000 +0200
@@ -2292,8 +2292,11 @@
 {
     // get this
     wxImageHistogram::iterator * self = (wxImageHistogram::iterator *)wxluaT_getuserdatatype(L, 1, wxluatype_wxImageHistogram_iterator);
+    // gcc 4.6 fix
+    wxImageHistogram::iterator* temp = self;
     // call op_inc
-    wxImageHistogram::iterator* returns = &((*self)++);
+    //wxImageHistogram::iterator* returns = &((*self)++);
+    wxImageHistogram::iterator* returns = temp++;
     // push the result datatype
     wxluaT_pushuserdatatype(L, returns, wxluatype_wxImageHistogram_iterator);
 
