diff -Naur omniNotify.orig//examples/demo_add_filter.cc omniNotify/examples/demo_add_filter.cc
--- omniNotify.orig//examples/demo_add_filter.cc	2010-06-25 00:13:49.943940542 +0200
+++ omniNotify/examples/demo_add_filter.cc	2010-06-25 00:26:33.466791409 +0200
@@ -1,5 +1,5 @@
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 
 #include "CosNotifyShorthands.h"
diff -Naur omniNotify.orig//examples/demo_offer_change.cc omniNotify/examples/demo_offer_change.cc
--- omniNotify.orig//examples/demo_offer_change.cc	2010-06-25 00:13:50.090622714 +0200
+++ omniNotify/examples/demo_offer_change.cc	2010-06-25 00:26:33.496784774 +0200
@@ -1,5 +1,5 @@
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 
 #include "CosNotifyShorthands.h"
diff -Naur omniNotify.orig//examples/demo_subscription_change.cc omniNotify/examples/demo_subscription_change.cc
--- omniNotify.orig//examples/demo_subscription_change.cc	2010-06-25 00:13:49.943940542 +0200
+++ omniNotify/examples/demo_subscription_change.cc	2010-06-25 00:26:33.483453396 +0200
@@ -1,5 +1,5 @@
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 
 #include "CosNotifyShorthands.h"
diff -Naur omniNotify.orig//examples/get_channel.h omniNotify/examples/get_channel.h
--- omniNotify.orig//examples/get_channel.h	2010-06-25 00:13:50.107289240 +0200
+++ omniNotify/examples/get_channel.h	2010-06-25 00:32:47.993411421 +0200
@@ -6,6 +6,8 @@
 #include <stdio.h>
 #include "CosNotifyShorthands.h"
 
+using namespace std;
+
 // SAMPLE code to get a channel reference either from the
 // naming service or from a stringified IOR stored in a file
 
diff -Naur omniNotify.orig//examples/legacy_clients.cc omniNotify/examples/legacy_clients.cc
--- omniNotify.orig//examples/legacy_clients.cc	2010-06-25 00:13:50.107289240 +0200
+++ omniNotify/examples/legacy_clients.cc	2010-06-25 00:35:46.187142224 +0200
@@ -4,13 +4,15 @@
 // See legacy_clients.h
 // -------------------------------------------------------------- //
 
-#include <iostream.h>
-#include <iomanip.h>
+#include <iostream>
+#include <iomanip>
 #include "thread_wrappers.h"
 
 #include "CosNotifyShorthands.h"
 #include "legacy_clients.h"
 
+using namespace std;
+
 // ----------------- CosEvC::PushConsumer ---------------- //
 
 LegacyPushConsumer_i::
diff -Naur omniNotify.orig//examples/ndadmin.cc omniNotify/examples/ndadmin.cc
--- omniNotify.orig//examples/ndadmin.cc	2010-06-25 00:13:49.943940542 +0200
+++ omniNotify/examples/ndadmin.cc	2010-06-25 00:26:33.473451929 +0200
@@ -6,7 +6,7 @@
 #error COS_USES_BOA should not be set for this example
 #endif
 
-#include <iostream.h>
+#include <iostream>
 
 #include "orb_init_name.h"
 #include "get_channel.h"
diff -Naur omniNotify.orig//examples/parse_cmd_line.h omniNotify/examples/parse_cmd_line.h
--- omniNotify.orig//examples/parse_cmd_line.h	2010-06-25 00:13:50.097286027 +0200
+++ omniNotify/examples/parse_cmd_line.h	2010-06-25 00:34:19.880484288 +0200
@@ -4,7 +4,7 @@
 #define _PARSE_CMD_LINE_H
 
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 
 #include "CosNotifyShorthands.h"
 
@@ -16,6 +16,8 @@
 #  include "getopt_impl.h"
 #endif
 
+using namespace std;
+
 static void usage(const char* pname, CORBA::Boolean supports_filters)
 {
   cout << "Usage: " << pname << " [-d #] [-b #] [-m #] [-M #] [-n name] [-f file] [-N name] [-F file] [-p file] [-a file] [-v]";
diff -Naur omniNotify.orig//examples/sample_clients.cc omniNotify/examples/sample_clients.cc
--- omniNotify.orig//examples/sample_clients.cc	2010-06-25 00:13:49.943940542 +0200
+++ omniNotify/examples/sample_clients.cc	2010-06-25 00:34:59.603397802 +0200
@@ -7,9 +7,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <fstream.h>
-#include <iostream.h>
-#include <iomanip.h>
+#include <fstream>
+#include <iostream>
+#include <iomanip>
 
 #ifdef HAVE_UNISTD_H
 #  include <unistd.h>
@@ -18,6 +18,7 @@
 #include "thread_wrappers.h"
 #include "sample_clients.h"
 
+using namespace std;
 
 ///////////////////////////////////////////////////////////////////
 //                 Optimization Notes                            //
diff -Naur omniNotify.orig//examples/sample_functions.cc omniNotify/examples/sample_functions.cc
--- omniNotify.orig//examples/sample_functions.cc	2010-06-25 00:13:49.943940542 +0200
+++ omniNotify/examples/sample_functions.cc	2010-06-25 00:33:32.170074805 +0200
@@ -2,10 +2,12 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
-#include <iomanip.h>
+#include <iostream>
+#include <iomanip>
 #include "sample_functions.h"
 
+using namespace std;
+
 ///////////////////////////////////////////////////////////////////
 //             USER-DEFINED FUNCTION EXAMPLES                    //
 ///////////////////////////////////////////////////////////////////
diff -Naur omniNotify.orig//examples/some_notify_clients.cc omniNotify/examples/some_notify_clients.cc
--- omniNotify.orig//examples/some_notify_clients.cc	2010-06-25 00:13:49.943940542 +0200
+++ omniNotify/examples/some_notify_clients.cc	2010-06-25 00:26:33.480114266 +0200
@@ -4,7 +4,7 @@
 #endif
 
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 
 #include "orb_init_name.h"
 #include "get_channel.h"
diff -Naur omniNotify.orig//include/ModFlexLexer.h omniNotify/include/ModFlexLexer.h
--- omniNotify.orig//include/ModFlexLexer.h	2010-06-25 00:13:50.167286027 +0200
+++ omniNotify/include/ModFlexLexer.h	2010-06-25 00:29:10.850520256 +0200
@@ -88,7 +88,9 @@
 // Never included before - need to define base class.
 #define __FLEX_LEXER_H
 
-#include <iostream.h>
+#include <iostream>
+
+using namespace std;
 
 extern "C++" {
 
diff -Naur omniNotify.orig//lib/fixlexout.pl omniNotify/lib/fixlexout.pl
--- omniNotify.orig//lib/fixlexout.pl	2010-06-25 00:13:50.195898972 +0200
+++ omniNotify/lib/fixlexout.pl	2010-06-25 00:26:33.513452418 +0200
@@ -13,7 +13,7 @@
 #   (2) change forward declaration of isteam
 #                class istream;
 #       to
-#                #include <iostream.h>
+#                #include <iostream>
 #
 #   (3) change
 #                #include <FlexLexer.h>
@@ -74,7 +74,7 @@
     # skip
     $declstart = 0;
   } elsif (/^\s*class\s+istream\s*;\s*$/) {
-    print OUTFILE "#include <iostream.h>\n"; $outlines++;
+    print OUTFILE "#include <iostream>\n"; $outlines++;
     $declstart = 0;
   } elsif (/^\#include\s+\<FlexLexer.h\>/) {
     print OUTFILE "#include \"ModFlexLexer.h\"\n"; $outlines++;
diff -Naur omniNotify.orig//lib/RDIParser_l.cc omniNotify/lib/RDIParser_l.cc
--- omniNotify.orig//lib/RDIParser_l.cc	2010-06-25 00:13:50.203952624 +0200
+++ omniNotify/lib/RDIParser_l.cc	2010-06-25 00:30:35.557174909 +0200
@@ -53,7 +53,9 @@
 #ifdef __cplusplus
 
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
+
+using namespace std;
 
 /* Use prototypes in function declarations. */
 #define YY_USE_PROTOS
diff -Naur omniNotify.orig//ReleaseNote_omniNotify_11 omniNotify/ReleaseNote_omniNotify_11
--- omniNotify.orig//ReleaseNote_omniNotify_11	2010-06-25 00:13:50.117286027 +0200
+++ omniNotify/ReleaseNote_omniNotify_11	2010-06-25 00:26:33.526785193 +0200
@@ -98,7 +98,7 @@
   still more can be done.
 
 * One side-effect of new logging facility is none of our .h files
-  include <iostream.h> anymore.  The flex-generated lexer still
+  include <iostream> anymore.  The flex-generated lexer still
   includes it, however :-<  It would be great to get rid of this
   last case.
 
