--- xcf2png.c.orig	2011-12-31 18:02:37.514216921 +0000
+++ xcf2png.c	2011-12-31 17:59:36.464222561 +0000
@@ -71,9 +71,9 @@
   
   outfile = openout(flatspec.output_filename);
   libpng = png_create_write_struct(PNG_LIBPNG_VER_STRING,
-                                   png_voidp_NULL,
+                                   (png_voidp)NULL,
                                    my_error_callback,
-                                   png_error_ptr_NULL);
+                                   (png_voidp)NULL);
   if( !libpng )
     FatalUnexpected(_("Couldn't initialize libpng library"));
   
