--- Makefile.orig	2012-02-24 15:45:29.000000000 -0500
+++ Makefile	2012-02-29 20:51:19.052093856 -0500
@@ -31,7 +31,7 @@
 
 VERSION := $(shell head -n 1 VERSION)
 install:
-	ocamlfind install -patch-version ${VERSION} $(LIBRARY) lib/META $(INTF) $(IMPL) $(OTHERS) $(DOC)
+	ocamlfind install -ldconf ignore -patch-version ${VERSION} $(LIBRARY) lib/META $(INTF) $(IMPL) $(OTHERS) $(DOC)
 	install -d -m 755 $(BINDIR)
 	install $(BIN) $(BINDIR)
 
