diff -Nur Puma-EM.orig/code/MoM/makefile Puma-EM/code/MoM/makefile
--- Puma-EM.orig/code/MoM/makefile	2014-05-21 17:09:36.000000000 +0200
+++ Puma-EM/code/MoM/makefile	2014-06-02 14:45:17.318043256 +0200
@@ -68,7 +68,7 @@
 mesh_cubes: mesh_cubes.o
 	$(MPICC) $(INCLUDE_PATH) mesh_cubes.o readWriteBlitzArrayFromFile.o $(LIB_SEARCH_PATH) -lblitz -lm -o mesh_cubes
 
-libs: libMoM libMLFMA libAMOS libITERATIVE libZGELS libLOCAL_LAPACK
+libs: libMoM libMLFMA libAMOS libITERATIVE libZGELS
 
 libMoM: $(OBJECTS_LIBMOM)
 	ar -r libMoM.a $(OBJECTS_LIBMOM)
diff -Nur Puma-EM.orig/code/PyGmsh.py Puma-EM/code/PyGmsh.py
--- Puma-EM.orig/code/PyGmsh.py	2014-05-21 17:09:36.000000000 +0200
+++ Puma-EM/code/PyGmsh.py	2014-06-02 14:48:01.644724789 +0200
@@ -117,7 +117,7 @@
 def executeGmsh(path, targetName, ViewMesh):
     isGeoFileThere(path, targetName)
     #CommandString = 'gmsh -2 ' + os.path.join(path, targetName) + '.geo'
-    CommandString = 'gmsh -2 -algo del2d ' + os.path.join(path, targetName) + '.geo' + ' -string "General.ExpertMode=1;"'
+    CommandString = 'gmsh -2 -algo del2d ' + os.path.join(path, targetName) + '.geo' + ' -string "General.ExpertMode=1;Mesh.Binary=0;"'
     print("  Meshing. Command: ", CommandString)
     commands.getoutput(CommandString)
     if ViewMesh == 1:
diff -Nur Puma-EM.orig/code/setup_GMSH.py Puma-EM/code/setup_GMSH.py
--- Puma-EM.orig/code/setup_GMSH.py	2014-05-21 17:09:36.000000000 +0200
+++ Puma-EM/code/setup_GMSH.py	2014-06-02 14:47:55.718057588 +0200
@@ -51,7 +51,7 @@
         isGeoFileThere(geoDirName, params_simu.targetName)
         f = open(filename, 'w')
         GMSH_command = 'rm -f ' + os.path.join(geoDirName, params_simu.targetName) + '.msh* \n'
-        GMSH_command += 'gmsh -2 -algo del2d -rand 1e-06 ' + os.path.join(geoDirName, params_simu.targetName) + '.geo' + ' -string "General.ExpertMode=1;"\n'
+        GMSH_command += 'gmsh -2 -algo del2d -rand 1e-06 ' + os.path.join(geoDirName, params_simu.targetName) + '.geo' + ' -string "General.ExpertMode=1;Mesh.Binary=0;"\n'
         f.write(GMSH_command)
         f.write("exit 0" + "\n")
     else:
diff -Nur Puma-EM.orig/makefile.inc Puma-EM/makefile.inc
--- Puma-EM.orig/makefile.inc	2014-05-21 17:09:36.000000000 +0200
+++ Puma-EM/makefile.inc	2014-06-02 14:43:37.578034236 +0200
@@ -36,7 +36,7 @@
 F_FLAGS:= -c -O3 -fPIC -pthread -ffast-math -mfpmath=both
 
 # we can use vendor-supplied lapack. If empty, puma-em lapack is used
-#LIBLAPACK:= -L/usr/lib/ -llapack
+LIBLAPACK:= -L/usr/lib/ -llapack
 
 #INCLUDE_PATH= -I/path/to/include
 #LIB_SEARCH_PATH= -L/path/to/lib
