diff -aur ike.pristine/source/iked/main.cpp ike.new/source/iked/main.cpp
--- ike.pristine/source/iked/main.cpp	2013-04-19 01:15:05.000000000 -0300
+++ ike.new/source/iked/main.cpp	2013-05-24 20:01:13.038241618 -0400
@@ -544,8 +544,9 @@
 	// create our pid file
 	//
 
-	if( !daemon_pidfile_create( path_pid ) )
-		return -1;
+	if( path_pid[ 0 ] )
+		if( !daemon_pidfile_create( path_pid ) )
+			return -1;
 
 	//
 	// run daemon main loop
@@ -557,7 +558,8 @@
 	// remove our pidfile
 	//
 
-	daemon_pidfile_remove( path_pid );
+	if( path_pid[ 0 ] )
+		daemon_pidfile_remove( path_pid );
 
 #endif
 
