--- trunk/prosody.cfg.lua.dist.orig 2013-04-02 20:03:54.862132070 +0200
+++ trunk/prosody.cfg.lua.dist 2013-04-02 20:04:50.668472745 +0200
@@ -21,6 +21,8 @@
 -- (see http://prosody.im/doc/creating_accounts for info)
 -- Example: admins = { "user1@example.com", "user2@example.net" }
 admins = { }
+daemonize = true
+pidfile = "/var/run/prosody/prosody.pid"

 -- Enable use of libevent for better performance under high load
 -- For more information see: http://prosody.im/doc/libevent
@@ -141,8 +141,8 @@
 -- Logging configuration
 -- For advanced logging see http://prosody.im/doc/logging
 log = {
-	info = "prosody.log"; -- Change 'info' to 'debug' for verbose logging
-	error = "prosody.err";
+	info = "/var/log/prosody/prosody.log"; -- Change 'info' to 'debug' for verbose logging
+	error = "/var/log/prosody/prosody.err";
 	-- "*syslog"; -- Uncomment this for logging to syslog
 	-- "*console"; -- Log to the console, useful for debugging with daemonize=false
 }
