--- linux-3.13.9-uksm_orig/mm/uksm.c	2014-04-04 14:56:29.738462696
-0300
+++ linux-3.13.9-devel/mm/uksm.c	2014-04-04 14:55:36.165528224 -0300
@@ -1604,7 +1604,11 @@
 static struct page *page_trans_compound_anon(struct page *page)
 {
 	if (PageTransCompound(page)) {
-		struct page *head = compound_trans_head(page);
+		/*benatto: old compound_trans_head
+ 		 *	   was merged into standard
+ 		 *	   compound_head function on linux/mm.h
+ 		 */
+		struct page *head = compound_head(page);
 		/*
 		 * head may actually be splitted and freed from under
 		 * us but it's ok here.
