diff --git a/Makefile b/Makefile
index ec70365..679a060 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-all:	$(MAKE)-simavr $(MAKE)-tests $(MAKE)-examples
+all:	$(MAKE)-simavr
 
 $(MAKE)-simavr:
 	$(MAKE) -C simavr
diff --git a/Makefile.common b/Makefile.common
index a73a059..7cc91b4 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -41,6 +41,7 @@ ARCH 		= ${shell $(CC) -dumpmachine | sed -e 's/^x/i/' -e 's/\(.\).*/\1/'}
 
 CFLAGS		+= -O2 -Wall
 CFLAGS		+= -g
+CFLAGS		+= -fPIC
 
 ifeq (${shell uname}, Darwin)
 # gcc 4.2 from MacOS is really not up to scratch anymore 
@@ -54,7 +55,7 @@ IPATH		+= /opt/local/include
 LFLAGS		= -L/opt/local/lib/
 endif
 else
-AVR_ROOT 	:= /usr/lib/avr
+AVR_ROOT 	:= /usr/avr
 AVR_INC 	:= ${AVR_ROOT}
 AVR 		:= avr-
 CFLAGS 		+= -fPIC
diff --git a/simavr/Makefile b/simavr/Makefile
index 7a67d32..183371f 100644
--- a/simavr/Makefile
+++ b/simavr/Makefile
@@ -22,6 +22,7 @@ SIMAVR_REVISION	= 1
 target	= run_avr
 
 CFLAGS	+= -Werror
+CFLAGS	+= -fPIC
 # tracing id useful especialy if you develop simavr core.
 # it otherwise eat quite a bit of few cycles, even disabled
 #CFLAGS	+= -DCONFIG_SIMAVR_TRACE=1
