diff --git a/Makefile b/Makefile
index 65085af..9a6e7d5 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 
 
-all:	build-simavr build-tests build-examples
+all:	build-simavr
 
 build-simavr:
 	$(MAKE) -C simavr
diff --git a/Makefile.common b/Makefile.common
index a73a059..b18080f 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -41,6 +41,7 @@ ARCH 		= ${shell $(CC) -dumpmachine | sed -e 's/^x/i/' -e 's/\(.\).*/\1/'}
 
 CFLAGS		+= -O2 -Wall
 CFLAGS		+= -g
+CFLAGS      += -fPIC
 
 ifeq (${shell uname}, Darwin)
 # gcc 4.2 from MacOS is really not up to scratch anymore 
@@ -54,7 +55,7 @@ IPATH		+= /opt/local/include
 LFLAGS		= -L/opt/local/lib/
 endif
 else
-AVR_ROOT 	:= /usr/lib/avr
+AVR_ROOT 	:= /usr/avr
 AVR_INC 	:= ${AVR_ROOT}
 AVR 		:= avr-
 CFLAGS 		+= -fPIC
