Index: src/Util.h
===================================================================
--- src/Util.h	(Revision 1217)
+++ src/Util.h	(Arbeitskopie)
@@ -23,6 +23,7 @@
 bool utilIsZipFile(const char *);
 void utilStripDoubleExtension(const char *, char *);
 IMAGE_TYPE utilFindType(const char *);
+IMAGE_TYPE utilFindType(const char *, char (&)[2048]);
 uint8_t *utilLoad(const char *, bool (*)(const char*), uint8_t *, int &);
 
 void utilPutDword(uint8_t *, uint32_t);
