Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(Revision 1199)
+++ CMakeLists.txt	(Arbeitskopie)
@@ -66,6 +66,11 @@
 
 if( ENABLE_LINK )
     FIND_PACKAGE ( SFML REQUIRED )
+    INCLUDE_DIRECTORIES( ${SFML_INCLUDE_DIR} )
+
+    # Hotfix: SOIL is needed by SFML 1.6, but the linker can't find
+    #         it on my system. Let's try to fix that.
+    find_library( LIBSOIL_LIBRARY NAMES SOIL libSOIL )
 endif( ENABLE_LINK )
 # set the standard libraries all ports use
 SET(VBAMCORE_LIBS
@@ -75,7 +80,8 @@
     ${SFML_LIBRARY}
     ${OPENGL_LIBRARIES}
     ${ZLIB_LIBRARY}
-    ${PNG_LIBRARY})
+    ${PNG_LIBRARY}
+    ${LIBSOIL_LIBRARY})
 
 
 
Index: CMakeScripts/FindSFML.cmake
===================================================================
--- CMakeScripts/FindSFML.cmake	(Revision 1199)
+++ CMakeScripts/FindSFML.cmake	(Arbeitskopie)
@@ -19,6 +19,7 @@
 
 SET(SFML_INCLUDE_SEARCH_DIR 
     ~/Library/Frameworks 
+    /usr/include/sfml-1.6/SFML
     /Library/Frameworks 
     /usr/local/include/SFML 
     /usr/include/SFML 
@@ -49,7 +50,7 @@
     STRING(TOUPPER ${COMPONENT} UPPERCOMPONENT) 
     STRING(TOLOWER ${COMPONENT} LOWERCOMPONENT) 
     FIND_LIBRARY(SFML_${UPPERCOMPONENT}_LIBRARY 
-        NAMES sfml-${LOWERCOMPONENT} 
+        NAMES sfml-${LOWERCOMPONENT}-1.6 sfml-${LOWERCOMPONENT}
         HINTS 
         $ENV{SFMLDIR} 
         PATH_SUFFIXES lib64 lib 
@@ -63,7 +64,7 @@
     ) 
     IF(SFML_${UPPERCOMPONENT}_INCLUDE_DIR AND SFML_${UPPERCOMPONENT}_LIBRARY) 
         LIST(APPEND SFML_LIBRARY ${SFML_${UPPERCOMPONENT}_LIBRARY}) 
-        LIST(APPEND SFML_INCLUDE_DIR ${SFML_${UPPERCOMPONENT}_INCLUDE_DIR}) 
+        LIST(APPEND SFML_INCLUDE_DIR ${SFML_${UPPERCOMPONENT}_INCLUDE_DIR} ${SFML_${UPPERCOMPONENT}_INCLUDE_DIR}/..) 
         LIST(REMOVE_DUPLICATES SFML_LIBRARY) 
         LIST(REMOVE_DUPLICATES SFML_INCLUDE_DIR) 
     ENDIF(SFML_${UPPERCOMPONENT}_INCLUDE_DIR AND SFML_${UPPERCOMPONENT}_LIBRARY) 
Index: src/Util.h
===================================================================
--- src/Util.h	(Revision 1217)
+++ src/Util.h	(Arbeitskopie)
@@ -23,6 +23,7 @@
 bool utilIsZipFile(const char *);
 void utilStripDoubleExtension(const char *, char *);
 IMAGE_TYPE utilFindType(const char *);
+IMAGE_TYPE utilFindType(const char *, char (&)[2048]);
 uint8_t *utilLoad(const char *, bool (*)(const char*), uint8_t *, int &);
 
 void utilPutDword(uint8_t *, uint32_t);
