diff -Naur gems-1.1.orig/server/source/defaults.h gems-1.1/server/source/defaults.h
--- gems-1.1.orig/server/source/defaults.h	2007-03-27 12:26:29.000000000 +0000
+++ gems-1.1/server/source/defaults.h	2007-03-27 12:26:44.000000000 +0000
@@ -51,7 +51,7 @@
 #define SCRIPT_OPTIONS  "-fq"		/* See man script */
 
 /* Lock filename prefix (followed by TCP port) */
-#define LOCK_PREFIX	"/var/lock/gems-server."
+#define LOCK_PREFIX	"/tmp/gems-server."
 
 /* Suggested maximum terminal size */
 #define SUGGESTED_MAX_COLS	80
