--- lockers.h.orig	2012-06-26 19:54:26.741329744 -0400
+++ lockers.h	2012-06-26 19:54:36.484663182 -0400
@@ -78,7 +78,7 @@
 		// (adding this now, will make it easier to use hash tables later)
 
 	int count_used_lockers_in_room(int roomvnum);
-	int lockers_object::get_next_free_locker_number(int roomvnum); // return the next available locker number in the room
+	int get_next_free_locker_number(int roomvnum); // return the next available locker number in the room
 	void info(char_data *ch, int locker_number);
 
 	OBJ_DATA *locker_contains;
