Fix gcc warnings.
--- wmxmms2-0.6.orig/src/wmgeneral.c
+++ wmxmms2-0.6/src/wmgeneral.c
@@ -445,7 +445,7 @@ void scroller(char *s, int x, int y){
 	static	char		S_HOLD [100];
 	static	unsigned int	s_offset;
 	static	int		start_offset;
-	if(s==NULL || s=="" || s=="\0" || strlen(s)==0 ){
+	if(s==NULL || *s=='\0' ){
 		s=" ";
 		s_offset=0;
 		strcpy(S_HOLD,s);
--- wmxmms2-0.6.orig/src/xmms2cif.h
+++ wmxmms2-0.6/src/xmms2cif.h
@@ -19,7 +19,7 @@ volatile unsigned int STATE;
 #define S_ERROR 99
 
 struct numberinfo {
-	uint length;
+	int32_t length;
 	uint played;
 	char title_info[100];
 };
