--- kwine_startmenu-0.1.orig/src/kwine_startmenu.h	2006-08-04 06:30:22.000000000 +0800
+++ kwine_startmenu-0.1/src/kwine_startmenu.h	2006-09-03 21:35:18.000000000 +0800
@@ -50,7 +50,7 @@
 	void mouseMoveEvent(QMouseEvent* e);
 
     private:
-        bool KWineMenu::loadicon(const QString &path, QPixmap &pix);
+        bool loadicon(const QString &path, QPixmap &pix);
         void append(const QString &icon, const KURL &url, const QString &label);
         void append_sub(const QString &icon, const KURL &url, const QString &label);
         KDirLister m_dirLister;
