diff -Nurd pyra-0.5.0/udev//60-roccat-pyra-wired.rules pyra-0.5.0.2/udev//60-roccat-pyra-wired.rules
--- pyra-0.5.0/udev//60-roccat-pyra-wired.rules	2011-04-08 19:46:06.000000000 +0300
+++ pyra-0.5.0.2/udev//60-roccat-pyra-wired.rules	2011-05-02 20:00:56.097199293 +0300
@@ -14,7 +14,6 @@
 # Transfer roccat event device to user group roccat.
 # On kernels prior to 2.6.38 the event device is of class roccat, from
 # 2.6.38 onwards class is named after device name
-SUBSYSTEM=="roccat", ACTION=="add", ATTRS{idVendor}=="1e7d", ATTRS{idProduct}=="2c24", GROUP="roccat", MODE="0660"
 SUBSYSTEM=="pyra", ACTION=="add", ATTRS{idVendor}=="1e7d", ATTRS{idProduct}=="2c24", GROUP="roccat", MODE="0660"
 
 # Transfer drivers sysfs attributes to user group roccat.
diff -Nurd pyra-0.5.0/udev//60-roccat-pyra-wireless.rules pyra-0.5.0.2/udev//60-roccat-pyra-wireless.rules
--- pyra-0.5.0/udev//60-roccat-pyra-wireless.rules	2011-04-08 19:46:06.000000000 +0300
+++ pyra-0.5.0.2/udev//60-roccat-pyra-wireless.rules	2011-05-02 20:01:03.643825075 +0300
@@ -14,7 +14,6 @@
 # Transfer roccat event device to user group roccat.
 # On kernels prior to 2.6.38 the event device is of class roccat, from
 # 2.6.38 onwards class is named after device name
-SUBSYSTEM=="roccat", ACTION=="add", ATTRS{idVendor}=="1e7d", ATTRS{idProduct}=="2cf6", GROUP="roccat", MODE="0660"
 SUBSYSTEM=="pyra", ACTION=="add", ATTRS{idVendor}=="1e7d", ATTRS{idProduct}=="2cf6", GROUP="roccat", MODE="0660"
 
 # Transfer drivers sysfs attributes to user group roccat.
