--- Jamtop	2012-04-19 09:50:31.000000000 +0200
+++ Jamtop.new	2012-05-12 20:53:10.000000000 +0200
@@ -67,6 +67,8 @@
 				LibWinD = /usr/lib/x86_64-linux-gnu ;
 			} else if [ GLOB /usr/lib64 : libX11.so ] {
 				LibWinD = /usr/lib64 ;
+			} else if [ GLOB /usr/lib : libX11.so ] {
+				LibWinD = /usr/lib ;
 			} else {
 				ECHO Unable to locate the 64 bit X11 library files ;
 			}
@@ -113,13 +115,13 @@
 # See if we have a system JPEG library.
 if ! $(BUILTIN_JPEG) && $(UNIX) {
 	if [ GLOB /usr/include : jpeglib.h ]  {
-		if  ! $(HOST64) && ( [ GLOB /usr/lib : libjpg.so ] || [ GLOB /usr/lib : libjpg.a ] )
-		 ||   $(HOST64) && ( [ GLOB /usr/lib64 : libjpg.so ] || [ GLOB /usr/lib64 : libjpg.a ] )
-		 ||   $(HOST64) && ( [ GLOB /usr/lib/x86_64-linux-gnu : libjpg.so ] || [ GLOB /usr/lib/x86_64-linux-gnu : libjpg.a ] ) {
+		if  ! $(HOST64) && ( [ GLOB /usr/lib : libjpeg.so ] || [ GLOB /usr/lib : libjpeg.a ] )
+		 ||   $(HOST64) && ( [ GLOB /usr/lib64 : libjpeg.so ] || [ GLOB /usr/lib64 : libjpeg.a ] )
+		 ||   $(HOST64) && ( [ GLOB /usr/lib/x86_64-linux-gnu : libjpeg.so ] || [ GLOB /usr/lib/x86_64-linux-gnu : libjpeg.a ] ) {
 			echo "Using system JPEG library" ;
 			JPEGLIB = ;
 			JPEGINC = ;
-			LINKFLAGS += $(LINKFLAG)tiff ;
+			LINKFLAGS += $(LINKFLAG)jpeg ;
 			HAVE_JPEG = true ;
 		}
 	}
