diff -aur xf86-video-nouveau-1.0.10/src/drmmode_display.c xf86-video-nouveau-1.0.10.new/src/drmmode_display.c
--- xf86-video-nouveau-1.0.10/src/drmmode_display.c	2013-11-07 04:59:45.000000000 +0000
+++ xf86-video-nouveau-1.0.10.new/src/drmmode_display.c	2014-06-25 10:26:59.151407253 +0100
@@ -1286,8 +1286,9 @@
 	if (!drmmode->mode_res)
 		return FALSE;
 
-	xf86CrtcSetSizeRange(pScrn, 320, 200, drmmode->mode_res->max_width,
+	xf86CrtcSetSizeRange(pScrn, 160, 100, drmmode->mode_res->max_width,
 			     drmmode->mode_res->max_height);
+	xf86DrvMsg(pScrn->scrnIndex, X_INFO, "arcade-drm: range min set (160x100)\n");
 
 	if (!drmmode->mode_res->count_connectors ||
 	    !drmmode->mode_res->count_crtcs) {
