--- xen-4.0.1.orig/Config.mk	2010-08-25 12:22:44.000000000 +0200
+++ xen-4.0.1/Config.mk	2010-11-02 23:38:11.575000000 +0100
@@ -187,4 +187,4 @@
 CONFIG_MINITERM    ?= n
 CONFIG_LOMOUNT     ?= n
 
--include $(XEN_ROOT)/.config
+#-include $(XEN_ROOT)/.config

--- xen-4.0.1/Config.mk.orig	2010-08-25 11:22:44.000000000 +0100
+++ xen-4.0.1/Config.mk	2011-01-29 17:40:43.000000000 +0000
@@ -135,6 +135,8 @@
 
 LDFLAGS += $(foreach i, $(EXTRA_LIB), -L$(i)) 
 CFLAGS += $(foreach i, $(EXTRA_INCLUDES), -I$(i))
+# temporary compile fix for rawhide
+CFLAGS += -Wunused-but-set-variable -Wno-error=unused-but-set-variable -Wuninitialized -Wno-error=uninitialized
 
 EMBEDDED_EXTRA_CFLAGS := -nopie -fno-stack-protector -fno-stack-protector-all
 EMBEDDED_EXTRA_CFLAGS += -fno-exceptions
