From 9539e80e33755599d2fd3c92c824651c4189be3f Mon Sep 17 00:00:00 2001
From: Yosef Or Boczko <yoseforb@gmail.com>
Date: Thu, 21 Nov 2013 21:25:04 +0200
Subject: chatView: Don't align the status message to right

---
 src/chatView.js | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/chatView.js b/src/chatView.js
index 5d530ab..491b5f8 100644
--- a/src/chatView.js
+++ b/src/chatView.js
@@ -156,6 +156,7 @@ const ChatView = new Lang.Class({
     _createTags: function() {
         let buffer = this._view.get_buffer();
         let tagTable = buffer.get_tag_table();
+        let rtl = Gtk.Widget.get_default_direction() == Gtk.TextDirection.RTL;
         let tags = [
           { name: 'nick',
             left_margin: 0 },
@@ -168,12 +169,12 @@ const ChatView = new Lang.Class({
           { name: 'status',
             left_margin: 0,
             indent: 0,
-            justification: Gtk.Justification.RIGHT },
+            justification: rtl ? Gtk.Justification.LEFT : Gtk.Justification.RIGHT },
           { name: 'timestamp',
             left_margin: 0,
             indent: 0,
             weight: Pango.Weight.BOLD,
-            justification: Gtk.Justification.RIGHT },
+            justification: rtl ? Gtk.Justification.LEFT : Gtk.Justification.RIGHT },
           { name: 'action',
             left_margin: 0 },
           { name: 'url',
-- 
1.8.4.2

