diff -ur PyQwt-5.2.0.orig/configure/configure.py PyQwt-5.2.0/configure/configure.py
--- PyQwt-5.2.0.orig/configure/configure.py	2014-01-31 15:56:00.000000000 +0100
+++ PyQwt-5.2.0/configure/configure.py	2014-01-31 15:59:07.000000000 +0100
@@ -183,7 +183,7 @@
 ''' % ' '.join(extra_include_dirs))
     f.close()
 
-    qmake = os.path.join(configuration.qt_dir, 'bin', 'qmake')
+    qmake = os.path.join(configuration.qt_dir, 'bin', 'qmake-qt4')
     os.system('%s -spec %s -o %s %s' % (
         qmake, configuration.platform, make_file, pro_file))
 
