diff -wNaur sawfish-1.9.1/src/windows.c mySawfish-1.9.1/src/windows.c 
--- sawfish-1.9.1/src/windows.c	2012-09-01 10:47:01.000000000 +0200
+++ mySawfish-1.9.1/src/windows.c	2012-11-24 15:37:29.967950229 +0100
@@ -1297,6 +1297,11 @@
     hints = &VWIN(win)->hints;
     flags = hints->flags;
 
+    if (hints->max_width >= 2000)
+        hints->max_width = 2000;
+    if (hints->max_height >= 2000)
+        hints->max_height = 2000;
+
     /* Some sanity checking */
     if ((flags & PMinSize) 
 	&& (hints->min_width < 0 || hints->min_height < 0))
