--- setup.py	2014-09-16 14:52:51.000000000 +0200
+++ setup.py	2015-02-14 08:42:38.442082122 +0100
@@ -69,14 +69,14 @@
         ],
     },
     install_requires=[
-        "Sphinx==1.2.2",
-        "beautifulsoup4==4.3.2",
-        "characteristic==14.1.0",
-        "click==3.1",
-        "colorama==0.3.1",
-        "lxml==3.3.5",
-        "six==1.7.3",
-        "zope.interface==4.1.1",
+        "Sphinx>=1.2.2",
+        "beautifulsoup4>=4.3.2",
+        "characteristic>=14.1.0",
+        "click>=3.1",
+        "colorama>=0.3.1",
+        "lxml>=3.3.5",
+        "six>=1.7.3",
+        "zope.interface>=4.1.1",
     ],
     tests_require=[
         "pytest",
