--- pycawm-0.2/Makefile
+++ pycawm-0.2/Makefile
@@ -2,7 +2,7 @@
 lt = $(shell expr $(1) \< $(2))
 
 # python version checks
-PYTHONVERSION=$(shell python -c "print __import__('sys').version[:3]")
+PYTHONVERSION=$(shell python2 -c "print __import__('sys').version[:3]")
 PYTHONMAJOR=$(shell echo $(PYTHONVERSION) | cut -d'.' -f1)
 PYTHONMINOR=$(shell echo $(PYTHONVERSION) | cut -d'.' -f2)
 ifeq ($(PYTHONVERSION),)
@@ -31,7 +31,7 @@
 	install -m644 conf/*.png $(DATADIR)/PycaWM
 	install -m644 pycawm/*.py $(MODULEDIR)/pycawm
 	install -m644 pycawm/plugins/*.py  $(MODULEDIR)/pycawm/plugins
-	python -O -c "from compileall import compile_dir; compile_dir('$(MODULEDIR)/pycawm')"
+	python2 -O -c "from compileall import compile_dir; compile_dir('$(MODULEDIR)/pycawm')"
 
 
 uninstall:
