diff --git a/src/Makefile b/src/Makefile
index 569a033..cfc9eb3 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,7 +1,7 @@
 VERSION :=0.1.1
 
-PREFIX :=/usr/local
-BINDIR :=$(PREFIX)/bin
+PREFIX :=/usr
+BINDIR :=$(DESTDIR)$(PREFIX)/bin
 
 PROG := mdz
 
@@ -39,7 +39,8 @@ clean:
 	rm -f $(PROG) $(OBJS)
 
 install: $(PROG)
-	install $(PROG) $(BINDIR)
+	mkdir -p $(BINDIR)
+	install -D $(PROG) $(BINDIR)/$(PROG)
 
 dist:
 	mkdir $(DISTDIR)
