--- siag-3.6.1-old/common/kdeinst	2006-11-17 22:00:02.000000000 -0500
+++ siag-3.6.1/common/kdeinst	2006-11-17 22:09:55.000000000 -0500
@@ -50,23 +50,6 @@
 	shift
 done
 
-if [ "$KDEDIR" = "" -o ! -w "$KDEDIR" ]; then
-	if [ -d /opt/kde -a -w /opt/kde ]; then
-		KDEDIR=/opt/kde			# SUSE, Slackware
-	elif [ -d /usr/local/kde -a -w /usr/local/kde ]; then
-		KDEDIR=/usr/local/kde		# YGIAGAM
-	elif [ -d /usr/kde -a -w /usr/kde ]; then
-		KDEDIR=/usr/kde			# Another guess
-	elif [ -d "$HOME/.kde" -a -w "$HOME/.kde" ]; then
-		KDEDIR="$HOME/.kde"		# Home directory
-	else
-		if [ $VERBOSE = yes ]; then
-			echo Don\'t know where to put anything
-		fi
-		exit
-	fi
-fi
-
 if [ "$APPLNK" ]; then
 	if [ "$MODE" = uninstall ]; then
 		rm -f "$KDEDIR/share/applnk/Siag Office/$APPLNK"
