--- xfce4-settings-4.10.0/dialogs/mouse-settings/main.c	2012-07-01 19:47:43.383911090 +0300
+++ xfce4-settings-4.10.0/dialogs/mouse-settings/main.c	2012-07-01 19:47:03.113910956 +0300
@@ -757,8 +757,8 @@
 
                 /* format: RT, RB, LT, LB, F1, F2, F3 */
                 data[4] = tap_to_click ? 1 : 0;
-                data[5] = tap_to_click ? 3 : 0;
-                data[6] = tap_to_click ? 2 : 0;
+                data[5] = tap_to_click ? 2 : 0;
+                data[6] = tap_to_click ? 3 : 0;
 
                 array = g_ptr_array_sized_new (n_items);
                 for (n = 0; n < n_items; n++)
