--- uberwriter/uberwriter/UberwriterWindow.py	2012-10-25 15:11:54.313025000 +0200
+++ uberwriter.fix/uberwriter/UberwriterWindow.py	2012-10-25 15:21:33.302002860 +0200
@@ -19,7 +19,10 @@
 import os
 import codecs
 import webbrowser
-import apt
+try:
+    import apt
+except ImportError:
+    apt = None
 import urllib
 
 from locale import gettext as _
@@ -529,12 +532,18 @@
 
     def export_as_pdf(self, widget, data=None):
         if self.texlive_installed == False:
-            try:
-                cache = apt.Cache()
-                inst = cache["texlive"].is_installed
-            except:
-                inst = True
-
+            if apt is not None:
+                try:
+                    cache = apt.Cache()
+                    inst = cache["texlive"].is_installed
+                except:
+                    inst = True
+            else:
+                try:
+                    subprocess.call(['pdflatex', '--version'])
+                    inst = True
+                except OSError:
+                    inst = False
             if inst == False:
                 dialog = Gtk.MessageDialog(self,
                     Gtk.DialogFlags.MODAL | Gtk.DialogFlags.DESTROY_WITH_PARENT,
