*** epsxe.plx	2013-07-21 01:53:44.198789767 -0400
--- epsxe.plx.new	2013-07-21 02:10:24.513035038 -0400
***************
*** 1,7 ****
  #!/usr/bin/perl
  
  $pkgname = 'epsxe'; 
! $pkgver = '1.6.0';
  $pkgpath = "/opt/$pkgname/$pkgver";
  
  system("cd $pkgpath ; ./$pkgname");
--- 1,7 ----
  #!/usr/bin/perl
  
  $pkgname = 'epsxe'; 
! $pkgver = '1.5.2';
  $pkgpath = "/opt/$pkgname/$pkgver";
  
  system("cd $pkgpath ; ./$pkgname");
