--- ./source/standard.c.~1~	2009-08-16 21:35:21.000000000 +0200
+++ ./source/standard.c	2010-07-22 05:14:45.000000000 +0200
@@ -3142,7 +3142,9 @@
 A68_ENV_REAL (genie_cgs_astronomical_unit, GSL_CONST_CGS_ASTRONOMICAL_UNIT)
 A68_ENV_REAL (genie_cgs_bar, GSL_CONST_CGS_BAR)
 A68_ENV_REAL (genie_cgs_barn, GSL_CONST_CGS_BARN)
+#ifdef GSL_CONST_CGS_BOHR_MAGNETON
 A68_ENV_REAL (genie_cgs_bohr_magneton, GSL_CONST_CGS_BOHR_MAGNETON)
+#endif
 A68_ENV_REAL (genie_cgs_bohr_radius, GSL_CONST_CGS_BOHR_RADIUS)
 A68_ENV_REAL (genie_cgs_boltzmann, GSL_CONST_CGS_BOLTZMANN)
 A68_ENV_REAL (genie_cgs_btu, GSL_CONST_CGS_BTU)
@@ -3153,13 +3155,23 @@
 A68_ENV_REAL (genie_cgs_curie, GSL_CONST_CGS_CURIE)
 A68_ENV_REAL (genie_cgs_day, GSL_CONST_CGS_DAY)
 A68_ENV_REAL (genie_cgs_dyne, GSL_CONST_CGS_DYNE)
+#ifdef GSL_CONST_CGS_ELECTRON_CHARGE
 A68_ENV_REAL (genie_cgs_electron_charge, GSL_CONST_CGS_ELECTRON_CHARGE)
+#endif
+#ifdef GSL_CONST_CGS_ELECTRON_MAGNETIC_MOMENT
 A68_ENV_REAL (genie_cgs_electron_magnetic_moment, GSL_CONST_CGS_ELECTRON_MAGNETIC_MOMENT)
+#endif
 A68_ENV_REAL (genie_cgs_electron_volt, GSL_CONST_CGS_ELECTRON_VOLT) A68_ENV_REAL (genie_cgs_erg, GSL_CONST_CGS_ERG)
-A68_ENV_REAL (genie_cgs_faraday, GSL_CONST_CGS_FARADAY) A68_ENV_REAL (genie_cgs_fathom, GSL_CONST_CGS_FATHOM)
+#ifdef GSL_CONST_CGS_FARADAY
+A68_ENV_REAL (genie_cgs_faraday, GSL_CONST_CGS_FARADAY)
+#endif
+A68_ENV_REAL (genie_cgs_fathom, GSL_CONST_CGS_FATHOM)
 A68_ENV_REAL (genie_cgs_fluid_ounce, GSL_CONST_CGS_FLUID_OUNCE) A68_ENV_REAL (genie_cgs_foot, GSL_CONST_CGS_FOOT)
 A68_ENV_REAL (genie_cgs_footcandle, GSL_CONST_CGS_FOOTCANDLE) A68_ENV_REAL (genie_cgs_footlambert, GSL_CONST_CGS_FOOTLAMBERT)
-A68_ENV_REAL (genie_cgs_gauss, GSL_CONST_CGS_GAUSS) A68_ENV_REAL (genie_cgs_gram_force, GSL_CONST_CGS_GRAM_FORCE)
+#ifdef GSL_CONST_CGS_GAUSS
+A68_ENV_REAL (genie_cgs_gauss, GSL_CONST_CGS_GAUSS)
+#endif
+A68_ENV_REAL (genie_cgs_gram_force, GSL_CONST_CGS_GRAM_FORCE)
 A68_ENV_REAL (genie_cgs_grav_accel, GSL_CONST_CGS_GRAV_ACCEL)
 A68_ENV_REAL (genie_cgs_gravitational_constant, GSL_CONST_CGS_GRAVITATIONAL_CONSTANT)
 A68_ENV_REAL (genie_cgs_hectare, GSL_CONST_CGS_HECTARE) A68_ENV_REAL (genie_cgs_horsepower, GSL_CONST_CGS_HORSEPOWER)
@@ -3178,14 +3190,19 @@
 A68_ENV_REAL (genie_cgs_mil, GSL_CONST_CGS_MIL) A68_ENV_REAL (genie_cgs_mile, GSL_CONST_CGS_MILE)
 A68_ENV_REAL (genie_cgs_miles_per_hour, GSL_CONST_CGS_MILES_PER_HOUR) A68_ENV_REAL (genie_cgs_minute, GSL_CONST_CGS_MINUTE)
 A68_ENV_REAL (genie_cgs_molar_gas, GSL_CONST_CGS_MOLAR_GAS) A68_ENV_REAL (genie_cgs_nautical_mile, GSL_CONST_CGS_NAUTICAL_MILE)
-A68_ENV_REAL (genie_cgs_newton, GSL_CONST_CGS_NEWTON) A68_ENV_REAL (genie_cgs_nuclear_magneton, GSL_CONST_CGS_NUCLEAR_MAGNETON)
+A68_ENV_REAL (genie_cgs_newton, GSL_CONST_CGS_NEWTON) 
+#ifdef GSL_CONST_CGS_NUCLEAR_MAGNETON
+A68_ENV_REAL (genie_cgs_nuclear_magneton, GSL_CONST_CGS_NUCLEAR_MAGNETON)
+#endif
 A68_ENV_REAL (genie_cgs_ounce_mass, GSL_CONST_CGS_OUNCE_MASS) A68_ENV_REAL (genie_cgs_parsec, GSL_CONST_CGS_PARSEC)
 A68_ENV_REAL (genie_cgs_phot, GSL_CONST_CGS_PHOT) A68_ENV_REAL (genie_cgs_pint, GSL_CONST_CGS_PINT)
 A68_ENV_REAL (genie_cgs_planck_constant_h, 6.6260693e-27) A68_ENV_REAL (genie_cgs_planck_constant_hbar, 6.6260693e-27 / (2 * A68G_PI))
 A68_ENV_REAL (genie_cgs_point, GSL_CONST_CGS_POINT) A68_ENV_REAL (genie_cgs_poise, GSL_CONST_CGS_POISE)
 A68_ENV_REAL (genie_cgs_pound_force, GSL_CONST_CGS_POUND_FORCE) A68_ENV_REAL (genie_cgs_pound_mass, GSL_CONST_CGS_POUND_MASS)
 A68_ENV_REAL (genie_cgs_poundal, GSL_CONST_CGS_POUNDAL)
+#ifdef GSL_CONST_CGS_PROTON_MAGNETIC_MOMENT
 A68_ENV_REAL (genie_cgs_proton_magnetic_moment, GSL_CONST_CGS_PROTON_MAGNETIC_MOMENT)
+#endif
 A68_ENV_REAL (genie_cgs_psi, GSL_CONST_CGS_PSI) A68_ENV_REAL (genie_cgs_quart, GSL_CONST_CGS_QUART)
 A68_ENV_REAL (genie_cgs_rad, GSL_CONST_CGS_RAD) A68_ENV_REAL (genie_cgs_roentgen, GSL_CONST_CGS_ROENTGEN)
 A68_ENV_REAL (genie_cgs_rydberg, GSL_CONST_CGS_RYDBERG) A68_ENV_REAL (genie_cgs_solar_mass, GSL_CONST_CGS_SOLAR_MASS)
--- ./source/prelude.c.~1~	2009-08-16 21:35:21.000000000 +0200
+++ ./source/prelude.c	2010-07-22 05:25:23.000000000 +0200
@@ -1829,10 +1829,10 @@
   a68_idf (A68_EXT, "cgsmassneutron", MODE (REAL), genie_cgs_mass_neutron);
   a68_idf (A68_EXT, "cgsrydberg", MODE (REAL), genie_cgs_rydberg);
   a68_idf (A68_EXT, "cgsboltzmann", MODE (REAL), genie_cgs_boltzmann);
-  a68_idf (A68_EXT, "cgsbohrmagneton", MODE (REAL), genie_cgs_bohr_magneton);
-  a68_idf (A68_EXT, "cgsnuclearmagneton", MODE (REAL), genie_cgs_nuclear_magneton);
-  a68_idf (A68_EXT, "cgselectronmagneticmoment", MODE (REAL), genie_cgs_electron_magnetic_moment);
-  a68_idf (A68_EXT, "cgsprotonmagneticmoment", MODE (REAL), genie_cgs_proton_magnetic_moment);
+//  a68_idf (A68_EXT, "cgsbohrmagneton", MODE (REAL), genie_cgs_bohr_magneton);
+//  a68_idf (A68_EXT, "cgsnuclearmagneton", MODE (REAL), genie_cgs_nuclear_magneton);
+//  a68_idf (A68_EXT, "cgselectronmagneticmoment", MODE (REAL), genie_cgs_electron_magnetic_moment);
+//  a68_idf (A68_EXT, "cgsprotonmagneticmoment", MODE (REAL), genie_cgs_proton_magnetic_moment);
   a68_idf (A68_EXT, "cgsmolargas", MODE (REAL), genie_cgs_molar_gas);
   a68_idf (A68_EXT, "cgsstandardgasvolume", MODE (REAL), genie_cgs_standard_gas_volume);
   a68_idf (A68_EXT, "cgsminute", MODE (REAL), genie_cgs_minute);
@@ -1891,9 +1891,9 @@
   a68_idf (A68_EXT, "cgspsi", MODE (REAL), genie_cgs_psi);
   a68_idf (A68_EXT, "cgspoise", MODE (REAL), genie_cgs_poise);
   a68_idf (A68_EXT, "cgsstokes", MODE (REAL), genie_cgs_stokes);
-  a68_idf (A68_EXT, "cgsfaraday", MODE (REAL), genie_cgs_faraday);
-  a68_idf (A68_EXT, "cgselectroncharge", MODE (REAL), genie_cgs_electron_charge);
-  a68_idf (A68_EXT, "cgsgauss", MODE (REAL), genie_cgs_gauss);
+//  a68_idf (A68_EXT, "cgsfaraday", MODE (REAL), genie_cgs_faraday);
+//  a68_idf (A68_EXT, "cgselectroncharge", MODE (REAL), genie_cgs_electron_charge);
+//  a68_idf (A68_EXT, "cgsgauss", MODE (REAL), genie_cgs_gauss);
   a68_idf (A68_EXT, "cgsstilb", MODE (REAL), genie_cgs_stilb);
   a68_idf (A68_EXT, "cgslumen", MODE (REAL), genie_cgs_lumen);
   a68_idf (A68_EXT, "cgslux", MODE (REAL), genie_cgs_lux);
