diff -Nur Mesa-7.11.orig/src/gallium/drivers/llvmpipe/Makefile Mesa-7.11/src/gallium/drivers/llvmpipe/Makefile
--- Mesa-7.11.orig/src/gallium/drivers/llvmpipe/Makefile	2011-07-09 11:37:09.000000000 +1000
+++ Mesa-7.11/src/gallium/drivers/llvmpipe/Makefile	2011-08-04 11:22:30.307658447 +1000
@@ -70,7 +70,7 @@
 	$(PYTHON2) $(PYTHON_FLAGS) lp_tile_soa.py ../../auxiliary/util/u_format.csv > $@
 
 LDFLAGS += $(LLVM_LDFLAGS)
-LIBS += -L../../auxiliary/ -lgallium libllvmpipe.a $(LLVM_LIBS) $(GL_LIB_DEPS)
+LIBS += -L../../auxiliary/ -lgallium -lpthread libllvmpipe.a $(LLVM_LIBS) $(GL_LIB_DEPS) $(DLOPEN_LIBS)
 LD=$(CXX)
 
 $(PROGS): lp_test_main.o libllvmpipe.a
