--- fluxStyle   2008-12-11 23:16:54.000000000 +1300
+++ fluxStyle   2008-12-11 23:21:35.000000000 +1300
@@ -26,10 +26,10 @@
 from fluxstyle import findStyles,parseCo
 from os.path import isfile,expanduser,isdir
 from fluxstyle import errorMessage

-#GLADEFILE="/usr/share/fluxstyle/glade/main.glade"
-#N_IMAGE="/usr/share/fluxstyle/images/none.jpg"
-GLADEFILE="./glade/main.glade"
-N_IMAGE="./images/none.jpg"
+GLADEFILE="/usr/share/fluxstyle/glade/main.glade"
+N_IMAGE="/usr/share/fluxstyle/images/none.jpg"
+#GLADEFILE="./glade/main.glade"
+#N_IMAGE="./images/none.jpg"

 class StyleChange:
     """Class wrapper for changing styles in fluxbox"""
