diff -Naur pytelemeter-1.5/pytelemeter/Telemeter4ToolsParser.py pytelemeter-1.5_patched/pytelemeter/Telemeter4ToolsParser.py
--- pytelemeter-1.5/pytelemeter/Telemeter4ToolsParser.py	2009-01-27 23:15:51.000000000 +0100
+++ pytelemeter-1.5_patched/pytelemeter/Telemeter4ToolsParser.py	2009-08-29 04:17:32.000000000 +0200
@@ -117,7 +117,7 @@
     def __init__(self, node):
         self.node = node
     def __getitem__(self, key):
-        children = self.node.getElementsByTagName('ns1:'+key)
+        children = self.node.getElementsByTagName(key)
         if len(children) == 1:
             return _Ns1Node(children[0])
         elif len(children) > 1:
