diff -aur zyGrib-5.0.2/src/zyGrib.pro zyGrib-5.0.2-patch/src/zyGrib.pro
--- zyGrib-5.0.2/src/zyGrib.pro	2011-07-15 12:41:40.000000000 +0200
+++ zyGrib-5.0.2-patch/src/zyGrib.pro	2011-07-19 18:10:53.138961167 +0200
@@ -28,8 +28,8 @@
 	ICON = ../img/zyGrib.icns
 }
 else {
-	INCLUDEPATH += /usr/include/qwt-qt4/
-	LIBS += -lqwt-qt4
+	INCLUDEPATH += /usr/include/qwt/
+	LIBS += -lqwt
 	DESTDIR = .
 }
 }
Seulement dans zyGrib-5.0.2-patch/src: zyGrib.pro~
