diff -aur zyGrib-3.9.9.1.vanilla/src/GUI/ui_DataPlotDialog.h zyGrib-3.9.9.1.new/src/GUI/ui_DataPlotDialog.h
--- zyGrib-3.9.9.1.vanilla/src/GUI/ui_DataPlotDialog.h	2011-03-11 17:35:43.716671672 +0100
+++ zyGrib-3.9.9.1.new/src/GUI/ui_DataPlotDialog.h	2011-03-11 18:11:18.303338335 +0100
@@ -20,7 +20,7 @@
 #include <QtGui/QHBoxLayout>
 #include <QtGui/QHeaderView>
 #include <QtGui/QVBoxLayout>
-#include "qwt_plot.h"
+#include <qwt_plot.h>
 
 QT_BEGIN_NAMESPACE
 
diff -aur zyGrib-3.9.9.1.vanilla/src/zyGrib.pro zyGrib-3.9.9.1.new/src/zyGrib.pro
--- zyGrib-3.9.9.1.vanilla/src/zyGrib.pro	2011-03-11 17:35:43.843338328 +0100
+++ zyGrib-3.9.9.1.new/src/zyGrib.pro	2011-03-11 18:24:18.540004911 +0100
@@ -5,9 +5,9 @@
 TARGET   = zyGrib
 
 DEPENDPATH  += . util map GUI
-INCLUDEPATH += . util map GUI curvedrawer bzip2 zlib-1.2.3 proj-4.6.0/src
+INCLUDEPATH += . util map GUI curvedrawer
 
-LIBS += -Lbzip2 -lbz2 -Lzlib-1.2.3 -lz -Lproj-4.6.0/src -lproj
+LIBS += -lbz2 -lz -lproj
 
 # ----------------------------------------------------
 # platform specific
@@ -27,8 +27,8 @@
 	ICON = ../img/zyGrib.icns
 }
 else {
-	INCLUDEPATH += /usr/include/qwt-qt4/
-	LIBS += -lqwt-qt4
+	INCLUDEPATH += /usr/include/qwt/
+	LIBS += -lqwt
 	DESTDIR = .
 }
 }
