Index: gnome-osd-0.12.2/configure.ac
===================================================================
--- gnome-osd-0.12.2.orig/configure.ac
+++ gnome-osd-0.12.2/configure.ac
@@ -17,7 +17,6 @@ AM_GLIB_GNU_GETTEXT
 AC_PROG_INTLTOOL
 
 
-AM_PATH_PYTHON_VERSION(2.5, 2.5.0, 2.4, 2.4.0, 2.3, 2.3.4)
 AC_PREFIX_PROGRAM(bonobo-activation-run-query)
 
 PKG_CHECK_MODULES(GNOME_OSD, pygtk-2.0 >= 2.4.0 gnome-python-2.0 >= 2.6.0)
