=== modified file 'gtk/gtk.symbols'
Index: gtk+-2.21.7/gtk/gtk.symbols
===================================================================
--- gtk+-2.21.7.orig/gtk/gtk.symbols	2010-09-01 12:21:02.742617000 +1000
+++ gtk+-2.21.7/gtk/gtk.symbols	2010-09-01 12:21:03.902617000 +1000
@@ -5281,6 +5281,7 @@
 gtk_widget_get_realized
 gtk_widget_set_mapped
 gtk_widget_get_mapped
+ubuntu_gtk_widget_set_has_grab
 #endif
 #endif
 
Index: gtk+-2.21.7/gtk/gtkwidget.c
===================================================================
--- gtk+-2.21.7.orig/gtk/gtkwidget.c	2010-08-31 01:09:37.000000000 +1000
+++ gtk+-2.21.7/gtk/gtkwidget.c	2010-09-01 12:21:03.912617000 +1000
@@ -11401,5 +11401,15 @@
   return res;
 }
 
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  if (has_grab)
+    GTK_OBJECT_FLAGS (widget) |= GTK_HAS_GRAB;
+  else
+    GTK_OBJECT_FLAGS (widget) &= ~(GTK_HAS_GRAB);
+}
+
 #define __GTK_WIDGET_C__
 #include "gtkaliasdef.c"
Index: gtk+-2.21.7/gtk/gtkwidget.h
===================================================================
--- gtk+-2.21.7.orig/gtk/gtkwidget.h	2010-08-31 01:09:37.000000000 +1000
+++ gtk+-2.21.7/gtk/gtkwidget.h	2010-09-01 12:21:03.912617000 +1000
@@ -1342,6 +1342,9 @@
 						       GtkWidget *toplevel,
 						       gpointer   user_data);
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
