=== modified file 'gtk/gtk.symbols'
Index: gtk+-2.24.3/gtk/gtk.symbols
===================================================================
--- gtk+-2.24.3.orig/gtk/gtk.symbols	2011-03-15 18:00:27.832882477 +1100
+++ gtk+-2.24.3/gtk/gtk.symbols	2011-03-15 18:00:29.232966723 +1100
@@ -5329,6 +5329,7 @@
 gtk_widget_get_realized
 gtk_widget_set_mapped
 gtk_widget_get_mapped
+ubuntu_gtk_widget_set_has_grab
 #endif
 #endif
 
Index: gtk+-2.24.3/gtk/gtkwidget.c
===================================================================
--- gtk+-2.24.3.orig/gtk/gtkwidget.c	2011-02-20 02:10:21.000000000 +1100
+++ gtk+-2.24.3/gtk/gtkwidget.c	2011-03-15 18:00:29.242967328 +1100
@@ -11419,5 +11419,15 @@
   return res;
 }
 
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  if (has_grab)
+    GTK_OBJECT_FLAGS (widget) |= GTK_HAS_GRAB;
+  else
+    GTK_OBJECT_FLAGS (widget) &= ~(GTK_HAS_GRAB);
+}
+
 #define __GTK_WIDGET_C__
 #include "gtkaliasdef.c"
Index: gtk+-2.24.3/gtk/gtkwidget.h
===================================================================
--- gtk+-2.24.3.orig/gtk/gtkwidget.h	2011-02-20 01:31:37.000000000 +1100
+++ gtk+-2.24.3/gtk/gtkwidget.h	2011-03-15 18:00:29.242967328 +1100
@@ -1346,6 +1346,9 @@
 						       GtkWidget *toplevel,
 						       gpointer   user_data);
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
