Index: gtk+2.0-2.21.5/gtk/gtkrc.c
===================================================================
--- gtk+2.0-2.21.5.orig/gtk/gtkrc.c	2010-07-22 17:55:14.000000000 +0200
+++ gtk+2.0-2.21.5/gtk/gtkrc.c	2010-07-22 17:57:19.657052790 +0200
@@ -450,7 +450,20 @@
       if (im_module_file)
 	result = g_strdup (im_module_file);
       else
-	result = g_build_filename (GTK_LIBDIR, "gtk-2.0", GTK_BINARY_VERSION, "gtk.immodules", NULL);
+        {
+#if defined(__linux__) && ( defined(__i386__) || defined (__x86_64__) || defined(__ia64__) )
+# if defined (__i386__)
+         result = g_build_filename (GTK_LIBDIR, "gtk-2.0", GTK_BINARY_VERSION, "gtk.immodules.32", NULL);
+# else
+         result = g_build_filename (GTK_LIBDIR, "gtk-2.0", GTK_BINARY_VERSION, "gtk.immodules.64", NULL);
+# endif
+          /* Prefer compat gtk.immodules file if it's usable. */
+          if (g_file_test(result, G_FILE_TEST_EXISTS))
+            return result;
+          g_free (result);
+#endif
+          result = g_build_filename (GTK_LIBDIR, "gtk-2.0", GTK_BINARY_VERSION, "gtk.immodules", NULL);
+        }
     }
 
   return result;
@@ -522,7 +535,21 @@
   else
     {
       const gchar *home;
-      str = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtkrc", NULL);
+#if defined(__linux__) && ( defined(__i386__) || defined (__x86_64__) || defined(__ia64__) )
+# if defined (__i386__)
+      str = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtkrc.32", NULL);
+# else
+      str = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtkrc.64", NULL);
+# endif
+      /* Prefer compat gtkrc if it's usable. */
+      if (!g_file_test(str, G_FILE_TEST_EXISTS))
+        {
+          g_free (str), /* continue in next line */
+#endif
+          str = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtkrc", NULL);
+#if defined(__linux__) && ( defined(__i386__) || defined (__x86_64__) || defined(__ia64__) )
+        }
+#endif
 
       gtk_rc_add_default_file (str);
       g_free (str);
@@ -530,7 +557,21 @@
       home = g_get_home_dir ();
       if (home)
 	{
-	  str = g_build_filename (home, ".gtkrc-2.0", NULL);
+#if defined(__linux__) && ( defined(__i386__) || defined (__x86_64__) || defined(__ia64__) )
+# if defined (__i386__)
+          str = g_build_filename (home, ".gtkrc-2.0.32", NULL);
+# else
+          str = g_build_filename (home, ".gtkrc-2.0.64", NULL);
+# endif
+          /* Prefer compat .gtkrc-2.0 if it's usable. */
+          if (!g_file_test(str, G_FILE_TEST_EXISTS))
+            {
+              g_free (str), /* continue in next line */
+#endif
+              str = g_build_filename (home, ".gtkrc-2.0", NULL);
+#if defined(__linux__) && ( defined(__i386__) || defined (__x86_64__) || defined(__ia64__) )
+            }
+#endif
 	  gtk_rc_add_default_file (str);
 	  g_free (str);
 	}
