Index: gtk+2.0-2.21.6/gtk/gtktreeview.c
===================================================================
--- gtk+2.0-2.21.6.orig/gtk/gtktreeview.c	2010-08-16 22:57:56.000000000 +0200
+++ gtk+2.0-2.21.6/gtk/gtktreeview.c	2010-08-17 11:52:07.864006434 +0200
@@ -9616,7 +9616,8 @@
 
       if (cursor_path)
 	{
-	  if (tree_view->priv->selection->type == GTK_SELECTION_MULTIPLE)
+	  if (tree_view->priv->selection->type == GTK_SELECTION_MULTIPLE ||
+	      tree_view->priv->selection->type == GTK_SELECTION_SINGLE)
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, FALSE, FALSE);
 	  else
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, TRUE, FALSE);
