diff -ur scheme48-1.8/Makefile.in scheme48-1.8.sic/Makefile.in
--- scheme48-1.8/Makefile.in	2008-01-26 09:06:27.000000000 -0600
+++ scheme48-1.8.sic/Makefile.in	2009-04-11 17:22:17.000000000 -0500
@@ -8,7 +8,7 @@
 VPATH = @srcdir@
 CC = @CC@
 LD = @LD@
-DEFS = @DEFS@ -D__COMPILING_SCHEME48_ITSELF__ -DS48_HOST_ARCHITECTURE=\"$(HOST_ARCHITECTURE)\"
+DEFS = @DEFS@ -D__COMPILING_SCHEME48_ITSELF__ -DS48_HOST_ARCHITECTURE=\"$(HOST_ARCHITECTURE)\" -D_GNU_SOURCE
 LIBS = @LIBS@
 CFLAGS = @CFLAGS@
 CPPFLAGS =
@@ -416,8 +416,8 @@
 	    -e 's,@LD\@,$(LD),g'					\
 	    -e 's,@DYNAMIC_EXTERNALS_CFLAGS\@,$(DYNAMIC_EXTERNALS_CFLAGS),g' \
 	    -e 's,@DYNAMIC_EXTERNALS_LDFLAGS\@,$(DYNAMIC_EXTERNALS_LDFLAGS),g' \
-	< $(srcdir)/build/$(CONFIG_SCRIPT).in > $$config_script &&	\
-	chmod +x $$config_script
+	< $(srcdir)/build/$(CONFIG_SCRIPT).in > $(DESTDIR)/$$config_script &&	\
+	chmod +x $(DESTDIR)/$$config_script
 
 inst-externals:
 	$(INSTALL_PROGRAM) $(POSIX_EXTERNAL) $(SRFI_27_EXTERNAL) $(DESTDIR)$(LIB)/
