--- a/bin/pytrainer
+++ b/bin/pytrainer
@@ -95,25 +95,6 @@
 sys.path.insert(0, site_path)
 from pytrainer.main import pyTrainer
 
-#Determine xulrunner location
-xulrunners = glob.glob("/usr/li*/xulrunner*/xulrunner")
-if len(xulrunners) == 0:
-    #Didnt find any - fall back to old approach
-    xul_env = commands.getstatusoutput("find /usr/li* -name xulrunner -exec dirname {} \; 2>/dev/null")[1]
-elif len(xulrunners) == 1:
-    #Found just one location - use that
-    xul_env = os.path.dirname(xulrunners[0])
-else:
-    #Found more than one - need to choose one...
-    max_version = None
-    for item in xulrunners:
-        xul_dir = os.path.dirname(item)
-        max_version = _max(max_version, xul_dir)
-    xul_env = max_version
-
-print "Using xulrunner dir: %s" % xul_env
-os.environ['MOZILLA_FIVE_HOME'] = xul_env
-
 def main():
     pytrainer = pyTrainer(None, data_path)
 
--- a/pytrainer/extensions/googlemaps.py
+++ b/pytrainer/extensions/googlemaps.py
@@ -35,9 +35,7 @@
     def drawMap(self,activity):
         '''Draw google map
             create html file using Google API version3
-            render using embedded Mozilla
-
-            info at http://www.pygtk.org/pygtkmozembed/class-gtkmozembed.html
+            render using embedded Webkit
         '''
         logging.debug(">>")
         points = []
--- a/pytrainer/extensions/mapviewer.py
+++ b/pytrainer/extensions/mapviewer.py
@@ -16,7 +16,7 @@
 #along with this program; if not, write to the Free Software
 #Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-import gtkmozembed
+import webkit
 import logging
 import os
 
@@ -31,14 +31,13 @@
 			logging.debug("Display box (%s) is None" % ( str(box)))
 			return
 		self.box = box
-		gtkmozembed.set_profile_path("/tmp", "foobar") # http://faq.pygtk.org/index.py?req=show&file=faq19.018.htp  #TODO FIX???
-		self.moz = gtkmozembed.MozEmbed()
+		self.wkview = webkit.WebView()
 		self.pack_box()
 		logging.debug("<<")
 
 	def pack_box(self):
 		logging.debug(">>")
-		self.box.pack_start(self.moz, True, True)
+		self.box.pack_start(self.wkview, True, True)
 		self.box.show_all()
 		logging.debug("<<")
 
@@ -46,7 +45,7 @@
 		logging.debug(">>")
 		if htmlfile is None:
 			htmlfile = self.createErrorHtml()
-		self.moz.load_url("file://%s" % (htmlfile))
+		self.wkview.load_uri("file://%s" % (htmlfile))
 		#self.box.show_all()
 		logging.debug("<<")
 
--- a/pytrainer/extensions/osm.py
+++ b/pytrainer/extensions/osm.py
@@ -2,7 +2,6 @@
 # TODO: store OpenLayers.js locally (1MB file)
 # TODO: Add google satelite images layers ?
 
-import gtkmozembed
 import os
 import re
 import logging
@@ -24,9 +23,7 @@
 	def drawMap(self,activity):
 		'''Draw osm map
 			create html file using Open Layers and Open Street Map
-			render using embedded Mozilla
-
-			info at http://www.pygtk.org/pygtkmozembed/class-gtkmozembed.html
+			render using embedded Webkit
 		'''
 		logging.debug(">>")
 		points = []
--- a/pytrainer/extensions/waypointeditor.py
+++ b/pytrainer/extensions/waypointeditor.py
@@ -16,7 +16,7 @@
 #along with this program; if not, write to the Free Software
 #Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
-import gtkmozembed
+import webkit
 import os
 import re
 
@@ -33,9 +33,9 @@
 		logging.debug(">>")
 		self.data_path = data_path
 		self.extension = Extension()
-		self.moz = gtkmozembed.MozEmbed()
-		self.moz.connect('title', self.handle_title_changed) 
-		vbox.pack_start(self.moz, True, True)
+		self.wkview = webkit.WebView()
+		self.wkview.connect('notify::title', self.handle_title_changed) 
+		vbox.pack_start(self.wkview, True, True)
 		vbox.show_all()
 		self.htmlfile = ""
 		self.waypoint=waypoint
@@ -43,7 +43,9 @@
 		logging.debug("<<")
 		
 	def handle_title_changed(self, *args): 
-		title = self.moz.get_title() 
+		title = self.wkview.get_title()
+		if title == None:
+			return
 		logging.debug("Received title: "+ title)
 		m = re.match("call:([a-zA-Z]*)[(](.*)[)]", title) 
 		if m: 
@@ -86,7 +88,7 @@
 		tmpdir = self.pytrainer_main.profile.tmpdir
 		htmlfile = tmpdir+"/waypointeditor.html"
 		logging.debug("HTML file: "+str(htmlfile))
-		self.moz.load_url("file://"+htmlfile)
+		self.wkview.load_uri("file://"+htmlfile)
 		logging.debug("<<")
 	
 	def createHtml(self,default_waypoint=None):
