Index: cblrepo/elm-compiler-0.14.1/elm-compiler.cabal
===================================================================
--- cblrepo.orig/elm-compiler-0.14.1/elm-compiler.cabal
+++ cblrepo/elm-compiler-0.14.1/elm-compiler.cabal
@@ -117,7 +117,7 @@ Library
         directory >= 1.0 && < 2.0,
         filepath >= 1 && < 2.0,
         indents >= 0.3 && < 0.4,
-        language-ecmascript >=0.15 && < 0.17,
+        language-ecmascript,
         language-glsl >= 0.0.2 && < 0.2,
         mtl >= 2 && < 3,
         parsec >= 3.1.1 && < 3.5,
