diff -wbBur qc-usb-0.6.6/qc-driver.c qc-usb-0.6.6.my/qc-driver.c
--- qc-usb-0.6.6/qc-driver.c	2006-10-24 02:06:19.000000000 +0000
+++ qc-usb-0.6.6.my/qc-driver.c	2011-02-03 15:44:45.000000000 +0000
@@ -43,15 +43,15 @@
 */
 /* }}} */
 /* {{{ [fold] Includes  */
+
+#define UTS_RELEASE ""
+
 #ifdef NOKERNEL
 #include "quickcam.h"
 #else
 #include <linux/quickcam.h>
 #endif
 #include <linux/version.h>
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)) || (HAVE_UTSRELEASE_H == 1)
-#include <linux/utsrelease.h>
-#endif
 #include <linux/module.h>
 
 #include "qc-memory.h"
@@ -190,7 +190,7 @@
 };
 
 static LIST_HEAD(quickcam_list);		/* Linked list containing all QuickCams */
-static DECLARE_MUTEX(quickcam_list_lock);	/* Always lock first quickcam_list_lock, then qc->lock */
+static struct semaphore quickcam_list_lock;	/* Always lock first quickcam_list_lock, then qc->lock */
 
 /* Default values for user-specified lookup-table; may be overwritten by user */
 static unsigned char userlut_contents[QC_LUT_SIZE] = {
@@ -994,7 +996,7 @@
 		PRINTK(KERN_WARNING,"Could not register procfs file entry");
 		return -ENXIO;
 	}
-	entry->owner = THIS_MODULE;
+//	entry->owner = THIS_MODULE;
 	entry->data = qc;
 	entry->read_proc = qc_proc_read;
 	entry->write_proc = qc_proc_write;
@@ -1032,7 +1034,7 @@
 			return -ENXIO;
 		}
 	}
-	qc_proc_entry->owner = THIS_MODULE;
+//	qc_proc_entry->owner = THIS_MODULE;
 	return 0;
 }
 /* }}} */
@@ -2997,7 +3023,7 @@
 	release:	qc_v4l_close,
 	read:		qc_v4l_read,
 //	write:		qc_v4l_write,
-	ioctl:		qc_v4l_ioctl,
+	compat_ioctl:		qc_v4l_ioctl,
 	mmap:		qc_v4l_mmap,
 	poll:		qc_v4l_poll,
 };
@@ -3103,7 +3130,7 @@
 PDEBUG("poisoning qc in qc_usb_init");
 		POISON(*qc);
 		if (qcdebug&QC_DEBUGMUTEX) PDEBUG("init down(%p) in qc_usb_init()", qc);
-		init_MUTEX_LOCKED(&qc->lock);
+		sema_init(&qc->lock, 1);
 		qc->users = 0;
 		if ((r=qc_i2c_init(qc))<0) goto fail2;
 	}
diff -wbBur qc-usb-0.6.6/qc-memory.c qc-usb-0.6.6.my/qc-memory.c
--- qc-usb-0.6.6/qc-memory.c	2006-11-04 13:34:32.000000000 +0000
+++ qc-usb-0.6.6.my/qc-memory.c	2011-02-03 15:45:30.000000000 +0000
@@ -27,7 +27,7 @@
  */
 /* }}} */
 /* {{{ [fold] Includes */
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #include <linux/version.h>
 
 #ifdef CONFIG_SMP
diff -wbBur qc-usb-0.6.6/quickcam.h qc-usb-0.6.6.my/quickcam.h
--- qc-usb-0.6.6/quickcam.h	2006-11-04 13:38:27.000000000 +0000
+++ qc-usb-0.6.6.my/quickcam.h	2011-02-03 15:38:57.000000000 +0000
@@ -76,7 +76,7 @@
 
 #ifdef __KERNEL__
 
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #include <linux/version.h>
 
 #ifdef CONFIG_SMP
