--- acm-5.0.debian/dis/sdbm/dbe.c	2012-05-24 04:18:21.033310326 +0600
+++ acm-5.0/dis/sdbm/dbe.c	2012-05-24 04:19:18.943310653 +0600
@@ -1,7 +1,8 @@
 #include <stdio.h>
+#include <gdbm.h>
 #ifndef VMS
 #include <sys/file.h>
-#include <gdbm-ndbm.h>
+#include <ndbm.h>
 #else
 #include "file.h"
 #include "ndbm.h"
--- acm-5.0.debian/src/audio.c	2012-05-24 04:18:21.103310327 +0600
+++ acm-5.0/src/audio.c	2012-05-24 04:16:39.916643088 +0600
@@ -578,7 +578,7 @@
 #else							/* defined(HPAUDIO) */
 
 int
-initializeAudio(craft * c, viewer * v, char *p)
+initializeAudio(craft * c, viewer * v, char *p, char *p1)
 {
 	int       i;
 
--- acm-5.0.debian/src/main.c	2012-05-24 04:43:48.000000000 +0600
+++ acm-5.0/src/main.c	2012-05-24 04:52:35.756655274 +0600
@@ -457,7 +457,7 @@
 #endif
 #endif
 
-	init("/usr/share/games/acm/");
+	init("/usr/share/acm/");
 
 	dpy = XOpenDisplay(display);
 	splash_file = acm_fopen ("acmlogo-unix.xwd", "r");
--- acm-5.0.debian/configure	2012-05-24 04:41:29.000000000 +0600
+++ acm-5.0/configure	2012-05-24 04:42:30.693318520 +0600
@@ -530,11 +530,7 @@
 
 
 
-if test -d /usr/share/games ; then
-    OBVDIR="/usr/share/games/acm"
-else
-    OBVDIR=\$\(prefix\)/games/acm
-fi
+OBVDIR="/usr/share/acm"
 
 
 rand=0
