--- mesa-build/src/gallium/targets/xorg-radeon/radeon_xorg.c	2010-09-27 17:21:47.131657039 +0200
+++ mesa-build/src/gallium/targets/xorg-radeon/radeon_xorg.c.new	2010-09-27 19:44:48.827779987 +0200
@@ -53,7 +53,7 @@
 };
 
 static XF86ModuleVersionInfo radeon_xorg_version = {
-    "radeong",
+    "radeon",
     MODULEVENDORSTRING,
     MODINFOSTRING1,
     MODINFOSTRING2,
@@ -69,9 +69,9 @@
  * Xorg driver exported structures
  */
 
-_X_EXPORT DriverRec radeong = {
+_X_EXPORT DriverRec radeon = {
     1,
-    "radeong",
+    "radeon",
     radeon_xorg_identify,
     NULL,
     xorg_tracker_available_options,
@@ -84,7 +84,7 @@
 
 static MODULESETUPPROTO(radeon_xorg_setup);
 
-_X_EXPORT XF86ModuleData radeongModuleData = {
+_X_EXPORT XF86ModuleData radeonModuleData = {
     &radeon_xorg_version,
     radeon_xorg_setup,
     NULL
@@ -103,7 +103,7 @@
      */
     if (!setupDone) {
 	setupDone = 1;
-	xf86AddDriver(&radeong, module, HaveDriverFuncs);
+	xf86AddDriver(&radeon, module, HaveDriverFuncs);
 
 	/*
 	 * The return value must be non-NULL on success even though there
@@ -120,7 +120,7 @@
 static void
 radeon_xorg_identify(int flags)
 {
-    xf86PrintChipsets("radeong", "Driver for Radeon Gallium with KMS",
+    xf86PrintChipsets("radeon", "Driver for Radeon Gallium with KMS",
 		      radeon_xorg_chipsets);
 }
 
@@ -135,8 +135,8 @@
 			       NULL, NULL, NULL, NULL, NULL);
     if (scrn != NULL) {
 	scrn->driverVersion = 1;
-	scrn->driverName = "radeong";
-	scrn->name = "radeong";
+	scrn->driverName = "radeon";
+	scrn->name = "radeon";
 	scrn->Probe = NULL;
 
 	entity = xf86GetEntityInfo(entity_num);
