diff -Nur -x '*.orig' -x '*~' pidgin-2.6.4/pidgin/gtkprefs.c pidgin-2.6.4.new/pidgin/gtkprefs.c
--- pidgin-2.6.4/pidgin/gtkprefs.c	2009-11-30 04:13:55.000000000 +0100
+++ pidgin-2.6.4.new/pidgin/gtkprefs.c	2009-12-10 17:00:47.000000000 +0100
@@ -1344,6 +1344,7 @@
 	GtkWidget *label;
 	GtkSizeGroup *sg;
 	GList *names = NULL;
+	GtkWidget *scrolledWindow1024x600;
 
 	ret = gtk_vbox_new(FALSE, PIDGIN_HIG_CAT_SPACE);
 	gtk_container_set_border_width(GTK_CONTAINER(ret), PIDGIN_HIG_BORDER);
@@ -1421,6 +1422,15 @@
 	keyboard_shortcuts(ret);
 
 	gtk_widget_show_all(ret);
+
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 700) {
+		scrolledWindow1024x600 = gtk_scrolled_window_new(NULL,NULL);
+		gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),GTK_POLICY_NEVER,GTK_POLICY_AUTOMATIC);
+		gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),ret);
+		ret = scrolledWindow1024x600;
+		gtk_widget_show(ret);
+	}
+
 	g_object_unref(sg);
 	return ret;
 }
@@ -1445,6 +1455,7 @@
 	GtkWidget *iconpref2;
 	GtkWidget *imhtml;
 	GtkWidget *frame;
+	GtkWidget *scrolledWindow1024x600;
 
 	ret = gtk_vbox_new(FALSE, PIDGIN_HIG_CAT_SPACE);
 	gtk_container_set_border_width(GTK_CONTAINER(ret), PIDGIN_HIG_BORDER);
@@ -1574,6 +1574,14 @@
 					 G_CALLBACK(formatting_clear_cb), NULL);
 	sample_imhtml = imhtml;
 
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 700) {
+		scrolledWindow1024x600 = gtk_scrolled_window_new(NULL,NULL);
+		gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),GTK_POLICY_NEVER,GTK_POLICY_AUTOMATIC);
+		gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),ret);
+		gtk_widget_show(ret);
+		ret = scrolledWindow1024x600;
+	}
+
 	gtk_widget_show(ret);
 
 	return ret;
@@ -1893,7 +1911,8 @@
 	GtkWidget *ret, *vbox, *hbox, *label, *entry, *browser_button;
 	GtkSizeGroup *sg;
 	GList *browsers = NULL;
-
+	GtkWidget *scrolledWindow1024x600;
+	
 	ret = gtk_vbox_new(FALSE, PIDGIN_HIG_CAT_SPACE);
 	gtk_container_set_border_width (GTK_CONTAINER (ret), PIDGIN_HIG_BORDER);
 
@@ -1969,6 +1988,14 @@
 		g_object_unref(sg);
 	}
 
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 700) {
+		scrolledWindow1024x600 = gtk_scrolled_window_new(NULL,NULL);
+		gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),GTK_POLICY_NEVER,GTK_POLICY_AUTOMATIC);
+		gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),ret);
+		ret = scrolledWindow1024x600;
+		gtk_widget_show(ret);
+	}
+
 	return ret;
 }
 #endif /*_WIN32*/
@@ -2405,6 +2432,7 @@
 	int j;
 	const char *file;
 	char *pref;
+	GtkWidget *scrolledWindow1024x600;
 #ifndef _WIN32
 	GtkWidget *dd;
 	GtkWidget *entry;
@@ -2580,6 +2608,15 @@
 	gtk_box_pack_start(GTK_BOX(hbox), button, FALSE, FALSE, 1);
 
 	gtk_widget_show_all(ret);
+
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 700) {
+		scrolledWindow1024x600 = gtk_scrolled_window_new(NULL,NULL);
+		gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),GTK_POLICY_NEVER,GTK_POLICY_AUTOMATIC);
+		gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),ret);
+		ret = scrolledWindow1024x600;
+		gtk_widget_show(ret);
+	}
+
 	g_object_unref(sg);
 
 	return ret;
