diff -Nur -x '*.orig' -x '*~' pidgin-2.5.4/pidgin/gtkpounce.c pidgin-2.5.4.new/pidgin/gtkpounce.c
--- pidgin-2.5.4/pidgin/gtkpounce.c	2009-01-13 03:05:39.000000000 +0800
+++ pidgin-2.5.4.new/pidgin/gtkpounce.c	2009-02-26 00:38:15.445274448 +0800
@@ -498,6 +498,7 @@
 	GPtrArray *sound_widgets;
 	GPtrArray *exec_widgets;
 	GtkWidget *send_msg_imhtml;
+	GtkWidget *scrolledWindow1024x600 = NULL;
 
 	g_return_if_fail((cur_pounce != NULL) ||
 	                 (account != NULL) ||
@@ -547,8 +548,14 @@
 
 	/* Create the vbox that will contain all the prefs stuff. */
 	vbox2 = gtk_vbox_new(FALSE, PIDGIN_HIG_BOX_SPACE);
-	gtk_box_pack_start(GTK_BOX(vbox1), vbox2, TRUE, TRUE, 0);
-
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 600) {
+	  scrolledWindow1024x600 = gtk_scrolled_window_new(NULL,NULL);
+	  gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),GTK_POLICY_NEVER,GTK_POLICY_AUTOMATIC);
+	  gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),vbox2);
+	  gtk_box_pack_start(GTK_BOX(vbox1), scrolledWindow1024x600, TRUE, TRUE, 0);
+	} else {
+	  gtk_box_pack_start(GTK_BOX(vbox1), vbox2, TRUE, TRUE, 0);
+	}
 	/* Create the "Pounce on Whom" frame. */
 	frame = pidgin_make_frame(vbox2, _("Pounce on Whom"));
 
@@ -1015,6 +1022,12 @@
 	}
 
 	gtk_widget_show(vbox2);
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 600) {
+	  gint oldw,oldh;
+	  gtk_widget_show(scrolledWindow1024x600);
+	  gtk_window_get_size(GTK_WINDOW(window),&oldw,&oldh);
+	  gtk_window_resize(GTK_WINDOW (window),oldw,448);
+	}
 	gtk_widget_show(window);
 }
 
