diff --git a/Makefile b/Makefile
index b85ca97..28569ab 100644
--- a/Makefile
+++ b/Makefile
@@ -38,6 +38,9 @@
 #   -DTRACE        N  This enables the "trace" command so that the search tree
 #                     can be dumped while running.
 
+PROFILE_USE_FLAGS=-fprofile-use -fprofile-correction -Wno-coverage-mismatch -freorder-blocks-and-partition
+PROFILE_GEN_FLAGS=-fprofile-generate -fprofile-correction
+
 default:
 	$(MAKE) linux-64
 help:
@@ -129,20 +132,21 @@ linux-amd64:
 linux:
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=g++ \
-		CFLAGS='$(CFLAGS) -g -Wall -pipe -O3 \
-			-fno-gcse -mpreferred-stack-boundary=8' \
-		CXFLAGS=$(CFLAGS) \
-		LDFLAGS='$(LDFLAGS) -g -lpthread -lstdc++' \
-		opt='$(opt) -DDEBUG -DTRACE -DINLINE64 -DCPUS=8' \
+		CFLAGS='$(CFLAGS) $(PROFILE_USE_FLAGS)' \
+		CXFLAGS='$(CXXFLAGS)' \
+		LDFLAGS='$(LDFLAGS) $(PROFILE_USE_FLAGS) -Wl,--as-needed' \
+		LIBS='-lpthread -lstdc++' \
+		opt='$(opt) -DTRACE -DINLINE32 -DCPUS=8 -DBOOKDIR="\"/usr/share/crafty\""' \
 		crafty-make
 
 linux-profile:
 	$(MAKE) target=LINUX \
 		CC=gcc CXX=g++ \
-		CFLAGS='-Wall -pipe -O3 -fprofile-arcs' \
-		CXFLAGS='' \
-		LDFLAGS='$(LDFLAGS) -fprofile-arcs -lstdc++ ' \
-		opt='$(opt) -DINLINE32 -DCPUS=2' \
+		CFLAGS='$(CFLAGS) $(PROFILE_GEN_FLAGS)' \
+		CXFLAGS='$(CXXFLAGS)' \
+		LDFLAGS='$(LDFLAGS) $(PROFILE_GEN_FLAGS) -Wl,--as-needed' \
+		LIBS='-lpthread -lstdc++ -lgcov' \
+		opt='$(opt) -DINLINE32 -DCPUS=8 -DBOOKDIR="\"..\""' \
 		crafty-make
 
 debug:
@@ -218,11 +222,9 @@ solaris-gcc:
 		crafty-make
 
 profile:
-	@rm -rf profdir
-	@rm -rf position.bin
-	@mkdir profdir
+	@rm -rf *.gcda
 	@touch *.c *.cpp *.h
-	$(MAKE) linux-64-profile
+	$(MAKE) linux-profile
 	@echo "#!/bin/csh" > runprof
 	@echo "./crafty <<EOF" >>runprof
 	@echo "st=10" >>runprof
@@ -289,7 +291,7 @@ profile:
 	@echo "quit" >>runprof
 	@echo "EOF" >>runprof
 	@chmod +x runprof
-	@./runprof
+	@bash ./runprof
 	@echo "#!/bin/csh" > runprof
 	@echo "./crafty <<EOF" >>runprof
 	@echo "st=10" >>runprof
@@ -301,10 +303,10 @@ profile:
 	@echo "quit" >>runprof
 	@echo "EOF" >>runprof
 	@chmod +x runprof
-	@./runprof
+	@bash ./runprof
 	@rm runprof
 	@touch *.c *.cpp *.h
-	$(MAKE) linux-64
+	$(MAKE) linux
 
 #
 #  one of the two following definitions for "objects" should be used.  The
diff --git a/option.c b/option.c
index 9d4eec5..1b06449 100644
--- a/option.c
+++ b/option.c
@@ -1097,7 +1097,7 @@ int Option(TREE * RESTRICT tree) {
     char *readstat = (char *) -1;
     int lines = 0;
 
-    helpfile = fopen("crafty.hlp", "r");
+    helpfile = fopen(BOOKDIR"/crafty.hlp", "r");
     if (!helpfile) {
       printf("ERROR.  Unable to open \"crafty.hlp\" -- help unavailable\n");
       return (1);
