diff --git a/Makefile b/Makefile
index eb06678..3c86cea 100644
--- a/Makefile
+++ b/Makefile
@@ -159,8 +159,8 @@ linux-64:
 		CC=icc CXX=icc \
 		CFLAGS='-w -xP -O2 -fno-alias -prof-use -prof_dir ./profdir' \
 		CXFLAGS='-w -xP -O2 -prof-use -prof-dir ./profdir' \
-		LDFLAGS='$(LDFLAGS) -lpthread -lstdc++' \
-		opt='$(opt) -DINLINE64 -DCPUS=2' \
+		LDFLAGS='$(LDFLAGS) -lpthread -lstdc++ -Wl,--as-needed' \
+		opt='$(opt) -DINLINE64 -DCPUS=8 -DBOOKDIR="\"/usr/share/crafty\""' \
 		crafty-make
 
 linux-64-profile:
@@ -168,8 +168,8 @@ linux-64-profile:
 		CC=icc CXX=icc \
 		CFLAGS='-w -xP -O2 -fno-alias -prof-gen=srcpos -prof-dir ./profdir' \
 		CXFLAGS='-w -xP -O2 -ip -prof-gen=srcpos -prof-dir ./profdir' \
-		LDFLAGS='$(LDFLAGS) -lpthread -lstdc++ ' \
-		opt='$(opt) -DINLINE64 -DCPUS=2' \
+		LDFLAGS='$(LDFLAGS) -lpthread -lstdc++ -Wl,--as-needed' \
+		opt='$(opt) -DINLINE64 -DCPUS=8 -DBOOKDIR="\"/usr/share/crafty\""' \
 		crafty-make
 
 linux-icc:
@@ -289,7 +289,7 @@ profile:
 	@echo "quit" >>runprof
 	@echo "EOF" >>runprof
 	@chmod +x runprof
-	@./runprof
+	@bash ./runprof
 	@echo "#!/bin/csh" > runprof
 	@echo "./crafty <<EOF" >>runprof
 	@echo "st=10" >>runprof
@@ -301,7 +301,7 @@ profile:
 	@echo "quit" >>runprof
 	@echo "EOF" >>runprof
 	@chmod +x runprof
-	@./runprof
+	@bash ./runprof
 	@rm runprof
 	@touch *.c *.cpp *.h
 	$(MAKE) linux-64
diff --git a/option.c b/option.c
index 4aacdee..5ba49a2 100644
--- a/option.c
+++ b/option.c
@@ -1093,7 +1093,7 @@ int Option(TREE * RESTRICT tree) {
     char *readstat = (char *) -1;
     int lines = 0;
 
-    helpfile = fopen("crafty.hlp", "r");
+    helpfile = fopen(BOOKDIR"/crafty.hlp", "r");
     if (!helpfile) {
       printf("ERROR.  Unable to open \"crafty.hlp\" -- help unavailable\n");
       return (1);
