diff -r fd2883a53fc1 src/client.c
--- a/src/client.c	Mon Dec 12 16:07:47 2011 +0100
+++ b/src/client.c	Mon Dec 12 16:31:23 2011 +0100
@@ -3778,6 +3778,12 @@
 
     screen_info = c->screen_info;
 
+    if ((c->button_status[button] == BUTTON_STATE_PRELIGHT) &&
+         clientGetButtonPixmap (c, button, PRELIGHT))
+    {
+        return (PRELIGHT);
+    }
+
     if (state == INACTIVE)
     {
         return (state);
@@ -3789,12 +3795,6 @@
         return (PRESSED);
     }
 
-    if ((c->button_status[button] == BUTTON_STATE_PRELIGHT) &&
-        clientGetButtonPixmap (c, button, PRELIGHT))
-    {
-        return (PRELIGHT);
-    }
-
     return (ACTIVE);
 }
 
diff -r fd2883a53fc1 src/events.c
--- a/src/events.c	Mon Dec 12 16:07:47 2011 +0100
+++ b/src/events.c	Mon Dec 12 16:31:23 2011 +0100
@@ -1382,23 +1382,20 @@
                 clientClearDelayedFocus ();
             }
         }
-        if (c == clientGetFocus ())
+        for (b = 0; b < BUTTON_COUNT; b++)
         {
-            for (b = 0; b < BUTTON_COUNT; b++)
+            if (MYWINDOW_XWINDOW(c->buttons[b]) == ev->window)
             {
-                if (MYWINDOW_XWINDOW(c->buttons[b]) == ev->window)
+                if (!xfwmPixmapNone(clientGetButtonPixmap(c, b, PRELIGHT)))
                 {
-                    if (!xfwmPixmapNone(clientGetButtonPixmap(c, b, PRELIGHT)))
-                    {
-                        c->button_status[b] = BUTTON_STATE_PRELIGHT;
-                        need_redraw = TRUE;
-                    }
+                    c->button_status[b] = BUTTON_STATE_PRELIGHT;
+                    need_redraw = TRUE;
                 }
             }
-            if (need_redraw)
-            {
-                frameQueueDraw (c, FALSE);
-            }
+        }
+        if (need_redraw)
+        {
+            frameQueueDraw (c, FALSE);
         }
 
         /* No need to process the event any further */
