diff --git a/dlls/ntdll/virtual.c b/dlls/ntdll/virtual.c
index d67f2d5..c69b840 100644
--- a/dlls/ntdll/virtual.c
+++ b/dlls/ntdll/virtual.c
@@ -1394,7 +1394,6 @@ static int alloc_virtual_heap( void *base, size_t size, void *arg )
 {
     void **heap_base = arg;
 
-    if (is_beyond_limit( base, size, address_space_limit )) address_space_limit = (char *)base + size;
     if (size < VIRTUAL_HEAP_SIZE) return 0;
     if (is_win64 && base < (void *)0x80000000) return 0;
     *heap_base = wine_anon_mmap( (char *)base + size - VIRTUAL_HEAP_SIZE,
