diff -crB src/usr/share/cura/Cura/gui/util/openglGui.py src.old/usr/share/cura/Cura/gui/util/openglGui.py
*** src/usr/share/cura/Cura/gui/util/openglGui.py	2013-06-25 11:01:31.135097531 +0200
--- src.old/usr/share/cura/Cura/gui/util/openglGui.py	2013-06-25 10:59:28.347502866 +0200
***************
*** 124,130 ****
  
  class glGuiPanel(glcanvas.GLCanvas):
  	def __init__(self, parent):
! 		attribList = (glcanvas.WX_GL_RGBA, glcanvas.WX_GL_DOUBLEBUFFER, glcanvas.WX_GL_DEPTH_SIZE, 32, glcanvas.WX_GL_STENCIL_SIZE, 8)
  		glcanvas.GLCanvas.__init__(self, parent, style=wx.WANTS_CHARS, attribList = attribList)
  		self._base = self
  		self._focus = None
--- 124,130 ----
  
  class glGuiPanel(glcanvas.GLCanvas):
  	def __init__(self, parent):
! 		attribList = (glcanvas.WX_GL_RGBA, glcanvas.WX_GL_DOUBLEBUFFER, glcanvas.WX_GL_DEPTH_SIZE, 24, glcanvas.WX_GL_STENCIL_SIZE, 8)
  		glcanvas.GLCanvas.__init__(self, parent, style=wx.WANTS_CHARS, attribList = attribList)
  		self._base = self
  		self._focus = None
*** src/usr/share/cura/Cura/gui/tools/superformula.py	2013-06-25 11:20:51.746171537 +0200
--- src.old/usr/share/cura/Cura/gui/tools/superformula.py	2013-06-25 11:20:18.907379836 +0200
***************
*** 101,107 ****
  		sizer = wx.BoxSizer()
  		self.SetSizer(sizer)
  
! 		attribList = (glcanvas.WX_GL_RGBA, glcanvas.WX_GL_DOUBLEBUFFER, glcanvas.WX_GL_DEPTH_SIZE, 32, glcanvas.WX_GL_STENCIL_SIZE, 8)
  		self._glCanvas = glcanvas.GLCanvas(self, style=wx.WANTS_CHARS, attribList = attribList)
  		self._glCanvas.SetMinSize((800,600))
  		sizer.Add(self._glCanvas, 1, flag=wx.EXPAND)
--- 101,107 ----
  		sizer = wx.BoxSizer()
  		self.SetSizer(sizer)
  
! 		attribList = (glcanvas.WX_GL_RGBA, glcanvas.WX_GL_DOUBLEBUFFER, glcanvas.WX_GL_DEPTH_SIZE, 24, glcanvas.WX_GL_STENCIL_SIZE, 8)
  		self._glCanvas = glcanvas.GLCanvas(self, style=wx.WANTS_CHARS, attribList = attribList)
  		self._glCanvas.SetMinSize((800,600))
  		sizer.Add(self._glCanvas, 1, flag=wx.EXPAND)
***************
*** 468,474 ****
  		self.addButton = wx.Button(self.panel, -1, 'Add to print')
  		sizer.Add(self.addButton, pos=(13,1))
  
! 		attribList = (glcanvas.WX_GL_RGBA, glcanvas.WX_GL_DOUBLEBUFFER, glcanvas.WX_GL_DEPTH_SIZE, 32, glcanvas.WX_GL_STENCIL_SIZE, 8)
  		self._glCanvas = glcanvas.GLCanvas(self.panel, style=wx.WANTS_CHARS, attribList = attribList)
  		self._glCanvas.SetMinSize((800,600))
  		sizer.Add(self._glCanvas, pos=(0,2), span=(14,1), flag=wx.EXPAND)
--- 468,474 ----
  		self.addButton = wx.Button(self.panel, -1, 'Add to print')
  		sizer.Add(self.addButton, pos=(13,1))
  
! 		attribList = (glcanvas.WX_GL_RGBA, glcanvas.WX_GL_DOUBLEBUFFER, glcanvas.WX_GL_DEPTH_SIZE, 24, glcanvas.WX_GL_STENCIL_SIZE, 8)
  		self._glCanvas = glcanvas.GLCanvas(self.panel, style=wx.WANTS_CHARS, attribList = attribList)
  		self._glCanvas.SetMinSize((800,600))
  		sizer.Add(self._glCanvas, pos=(0,2), span=(14,1), flag=wx.EXPAND)
