--- jahplayer-0.2.2.orig/jahwidgets/src/qt3/olibs.pri	2007-04-04 16:00:17.000000000 +0800
+++ jahplayer-0.2.2/jahwidgets/src/qt3/olibs.pri	2007-04-29 14:29:03.000000000 +0800
@@ -1,7 +1,7 @@
 # Common qmake include for OpenLibraries
 
 unix {
-    OLIB_PREFIX=$$system(pkg-config --variable=prefix openlibaries 2>/dev/null)
+    OLIB_PREFIX=$$system(pkg-config --variable=prefix openlibraries 2>/dev/null)
     isEmpty( OLIB_PREFIX ) {
         OLIB_PREFIX=/usr/local
     }
