--- Nagstamon.2/Config.py	2014-07-28 10:46:23.000000000 +0200
+++ Nagstamon/Config.py	2014-07-31 10:19:27.440120648 +0200
@@ -24,6 +24,8 @@
 import base64
 import zlib
 
+from xdg import BaseDirectory
+
 # global flag for keyring availabilty
 keyring_available = False
 # Determine if keyring module and an implementation is available for secure password storage
@@ -199,9 +201,12 @@
             # ~/.nagstamon/nagstamon.conf is the user conf file
             # os.path.expanduser('~') finds out the user HOME dir where
             # nagstamon expects its conf file to be
-            self.configdir = os.path.expanduser('~') + os.sep + ".nagstamon"
+            self.configdir = os.path.join(BaseDirectory.xdg_config_home, "nagstamon")
+
+        if not os.path.isdir(self.configdir):
+            os.mkdir(self.configdir)
 
-        self.configfile = self.configdir + os.sep + "nagstamon.conf"
+        self.configfile = os.path.join(self.configdir, "nagstamon.conf")
 
         # make path fit for actual os, normcase for letters and normpath for path
         self.configfile = os.path.normpath(os.path.normcase(self.configfile))
