Index: lib/Driver/ToolChains.cpp
===================================================================
--- lib/Driver/ToolChains.cpp	(revision 123373)
+++ lib/Driver/ToolChains.cpp	(working copy)
@@ -1398,7 +1398,7 @@
   PPaths.push_back(Twine(GCCInstallation.getParentLibPath() + "/../" +
                          GCCInstallation.getTriple() + "/bin").str());
 
-  Linker = GetProgramPath("ld");
+  Linker = GetProgramPath("ld.gold");
 
   LinuxDistro Distro = DetectLinuxDistro(Arch);
