diff -Naur usr-orig/bin/gufw usr/bin/gufw
--- usr-orig/bin/gufw	2012-01-12 18:43:02.000000000 +0000
+++ usr/bin/gufw	2012-03-11 01:11:13.356646235 +0000
@@ -9,7 +9,7 @@
 for ((i = 0; i < ${#LOCATIONS[@]}; i++))
 do
     if [[ -e "${LOCATIONS[${i}]}" ]]; then
-        python ${LOCATIONS[${i}]}
+        gksudo --preserve-env python2 ${LOCATIONS[${i}]}
         successful="True"
         break
     fi
diff -Naur usr-orig/share/applications/gufw.desktop usr/share/applications/gufw.desktop
--- usr-orig/share/applications/gufw.desktop	2012-03-11 01:09:56.000000000 +0000
+++ usr/share/applications/gufw.desktop	2012-03-11 01:11:58.755552450 +0000
@@ -2,7 +2,7 @@
 Version=1.0
 Type=Application
 Terminal=false
-Exec=gufw
+Exec=gksudo --preserve-env gufw
 Name=Firewall Configuration
 Name[zh_TW]=防火牆設定
 Name[vi]=Cấu hình tường lửa
