Index: xkeymap.c
===================================================================
--- xkeymap.c	(revision 1658)
+++ xkeymap.c	(working copy)
@@ -545,9 +545,17 @@
 			    && (get_key_state(state, XK_Control_L)
 				|| get_key_state(state, XK_Control_R)))
 			{
-				/* Ctrl-Alt-Enter: toggle full screen */
-				if (pressed)
-					xwin_toggle_fullscreen();
+                /* Ctrl-Win-Alt-Enter: exit */
+                if (get_key_state(state, XK_Super_L) || get_key_state(state, XK_Super_R)) {
+				    if(pressed)
+                        exit(0);
+                }
+                /* Ctrl-Alt-Enter: toggle full screen */
+                else {
+                    if(pressed)
+                        xwin_toggle_fullscreen();
+                }
+
 				return True;
 			}
 			break;
