diff -rU 1 ramlog-2.0.0-orig//ramlog ramlog-2.0.0/ramlog
--- ramlog-2.0.0-orig//ramlog	2010-04-12 09:23:23.000000000 +0200
+++ ramlog-2.0.0/ramlog	2010-05-15 12:33:09.000000000 +0200
@@ -34,3 +34,3 @@
 VERBOSE=1
-CONFIGFILE=/etc/init.d/ramlog
+CONFIGFILE=/etc/rc.d/ramlog
 
@@ -56,3 +56,3 @@
 prog="ramlog"
-DAEMONDIR=/etc/init.d
+DAEMONDIR=/etc/rc.d
 UNINSTALL_FLAG_FILE=/var/tmp/ramlog_uninstall_flag_file	#If exists, ramlog will be uninstalled during shutdown or startup
@@ -376,7 +376,7 @@
 daemA1="-e ^plymouthd"			#daemon start 1
-DEFSTOP rsyslogd /etc/init.d/rsyslog
-DEFSTOP acpid /etc/init.d/acpid
-DEFSTOP syslogd /etc/init.d/sysklogd
-DEFSTOP cupsd /etc/init.d/cups
-DEFSTOP gdm-simpl /etc/init.d/gdm
+DEFSTOP rsyslogd /etc/rc.d/rsyslog
+DEFSTOP acpid /etc/rc.d/acpid
+DEFSTOP syslogd /etc/rc.d/sysklogd
+DEFSTOP cupsd /etc/rc.d/cups
+DEFSTOP gdm-simpl /etc/rc.d/gdm
 DEFSTOP Xorg SLEEP
diff -rU 1 ramlog-2.0.0-orig//ramlog.conf.upstart ramlog-2.0.0/ramlog.conf.upstart
--- ramlog-2.0.0-orig//ramlog.conf.upstart	2010-04-07 00:35:07.000000000 +0200
+++ ramlog-2.0.0/ramlog.conf.upstart	2010-05-15 19:23:05.000000000 +0200
@@ -9,4 +9,4 @@
 
-pre-start exec /etc/init.d/ramlog upstart
-#post-stop exec /etc/init.d/ramlog upstop
+pre-start exec /etc/rc.d/ramlog upstart
+#post-stop exec /etc/rc.d/ramlog upstop
 
diff -rU 1 ramlog-2.0.0-orig//ramlog.cron ramlog-2.0.0/ramlog.cron
--- ramlog-2.0.0-orig//ramlog.cron	2008-06-08 04:28:38.000000000 +0200
+++ ramlog-2.0.0/ramlog.cron	2010-05-15 19:22:49.000000000 +0200
@@ -2,2 +2,2 @@
 #save logs from ramdisk to harddrive:
-/etc/init.d/ramlog restart
+/etc/rc.d/ramlog restart
