diff -urNw libavg-1.7.1/configure.in libavg-1.7.1.patched/configure.in
--- libavg-1.7.1/configure.in	2012-01-26 17:42:39.000000000 +0100
+++ libavg-1.7.1.patched/configure.in	2012-07-24 09:59:53.380220160 +0200
@@ -75,7 +75,7 @@
 fi
     
 AX_BOOST_THREAD
-if test -z "$BOOST_THREAD_LIB"; then
+if test -z "$BOOST_THREAD_LIBS"; then
    AC_MSG_ERROR([boost.thread not found. Aborting.])
 fi
 
diff -urNw libavg-1.7.1/m4/ax_boost_thread.m4 libavg-1.7.1.patched/m4/ax_boost_thread.m4
--- libavg-1.7.1/m4/ax_boost_thread.m4	2011-12-07 13:58:13.000000000 +0100
+++ libavg-1.7.1.patched/m4/ax_boost_thread.m4	2012-07-24 10:07:58.000000000 +0200
@@ -6,9 +6,9 @@
 dnl name or suffix and then just the raw library.
 dnl
 dnl If the library is found, HAVE_BOOST_THREAD is defined and
-dnl BOOST_THREAD_LIB is set to the name of the library.
+dnl BOOST_THREAD_LIBS is set to the name of the library.
 dnl
-dnl This macro calls AC_SUBST(BOOST_THREAD_LIB).
+dnl This macro calls AC_SUBST(BOOST_THREAD_LIBS).
 dnl
 dnl @category InstalledPackages
 dnl @author Michael Tindal <mtindal@paradoxpoint.com>
@@ -17,20 +17,22 @@
 
 AC_DEFUN([AX_BOOST_THREAD],
 [AC_REQUIRE([AC_CXX_NAMESPACES])dnl
-AC_CACHE_CHECK(whether the Boost::Thread library is available,
-ax_cv_boost_thread,
-[AC_LANG_SAVE
+
+AC_LANG_SAVE
  AC_LANG_CPLUSPLUS
  CXXFLAGS_SAVE=$CXXFLAGS
+LIBS_SAVE=$LIBS
 dnl FIXME: need to include a generic way to check for the flag
 dnl to turn on threading support.
  CXXFLAGS="-pthread $CXXFLAGS"
- AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <boost/thread/thread.hpp>]],
+
+AC_CACHE_CHECK(whether the Boost::Thread library is available,
+ax_cv_boost_thread,
+[AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <boost/thread/thread.hpp>]],
 			           [[boost::thread_group thrds; return 0;]])],
   	           ax_cv_boost_thread=yes, ax_cv_boost_thread=no)
- CXXFLAGS=$CXXFLAGS_SAVE
- AC_LANG_RESTORE
 ])
+
 if test "$ax_cv_boost_thread" = yes; then
   AC_DEFINE(HAVE_BOOST_THREAD,,[define if the Boost::Thread library is available])
   dnl Now determine the appropriate file names
@@ -41,9 +43,32 @@
     ax_boost_thread_lib=boost_thread-$with_boost_thread
   fi])
   for ax_lib in $ax_thread_lib $ax_boost_thread_lib boost_thread boost_thread-mt; do
-    AC_CHECK_LIB($ax_lib, main, [BOOST_THREAD_LIB=$ax_lib
+    AC_CHECK_LIB($ax_lib, main, [BOOST_THREAD_LIBS=-l$ax_lib
 break])
   done
-  AC_SUBST(BOOST_THREAD_LIB)
+
+  # OXullo 2012-07-18: since boost 1.50, boost::thread depends on boost::system
+  AC_CACHE_CHECK(whether Boost::Thread needs Boost::System library,
+  ax_cv_boost_thread_system,
+  [LIBS="$LIBS $BOOST_THREAD_LIBS"
+  AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <boost/thread/thread.hpp>]],
+            [[boost::thread_group thrds; return 0;]])],
+            [ax_cv_boost_thread_system=no],
+            [LIBS="$LIBS $BOOST_THREAD_LIBS -lboost_system"
+                AC_LINK_IFELSE([
+                    AC_LANG_PROGRAM([[#include <boost/thread/thread.hpp>]],
+                        [[boost::thread_group thrds; return 0;]])
+                    ],
+                    [BOOST_THREAD_LIBS="$BOOST_THREAD_LIBS -lboost_system"
+                        ax_cv_boost_thread_system=yes],
+                    [AC_ERROR([Cannot use Boost::Thread])]
+                    )])
+  ])
+  
+  CXXFLAGS=$CXXFLAGS_SAVE
+  LIBS=$LIBS_SAVE
+  AC_LANG_RESTORE
+
+  AC_SUBST(BOOST_THREAD_LIBS)
 fi
 ])dnl
diff -urNw libavg-1.7.1/src/audio/Makefile.am libavg-1.7.1.patched/src/audio/Makefile.am
--- libavg-1.7.1/src/audio/Makefile.am	2011-12-07 13:59:08.000000000 +0100
+++ libavg-1.7.1.patched/src/audio/Makefile.am	2012-07-24 10:18:00.395229279 +0200
@@ -15,4 +15,4 @@
 
 testlimiter_SOURCES = testlimiter.cpp $(ALL_H)
 testlimiter_LDADD = ./libaudio.la ../base/libbase.la \
-        -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@
+        @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@
diff -urNw libavg-1.7.1/src/base/Makefile.am libavg-1.7.1.patched/src/base/Makefile.am
--- libavg-1.7.1/src/base/Makefile.am	2012-01-26 17:01:32.000000000 +0100
+++ libavg-1.7.1.patched/src/base/Makefile.am	2012-07-24 10:18:11.050229369 +0200
@@ -26,6 +26,6 @@
 
 noinst_PROGRAMS=testbase
 testbase_SOURCES=testbase.cpp $(ALL_H)
-testbase_LDADD = ./libbase.la -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@
+testbase_LDADD = ./libbase.la @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@
 # -rdynamic needed only for testBacktrace to work under linux.
 testbase_LDFLAGS = -rdynamic
diff -urNw libavg-1.7.1/src/graphics/Makefile.am libavg-1.7.1.patched/src/graphics/Makefile.am
--- libavg-1.7.1/src/graphics/Makefile.am	2012-01-26 17:01:31.000000000 +0100
+++ libavg-1.7.1.patched/src/graphics/Makefile.am	2012-07-24 10:18:23.172229472 +0200
@@ -50,15 +50,15 @@
 noinst_PROGRAMS=testgraphics testgpu benchmarkgraphics
 testgraphics_SOURCES=testgraphics.cpp $(ALL_H)
 testgraphics_LDADD = ./libgraphics.la ../base/libbase.la \
-        @XML2_LIBS@ -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@ $(X_LIBS) @GDK_PIXBUF_LIBS@
+        @XML2_LIBS@ @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@ $(X_LIBS) @GDK_PIXBUF_LIBS@
 
 benchmarkgraphics_SOURCES=benchmarkgraphics.cpp $(ALL_H)
 benchmarkgraphics_LDADD = ./libgraphics.la ../base/libbase.la \
-        @XML2_LIBS@ -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@ @GDK_PIXBUF_LIBS@
+        @XML2_LIBS@ @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@ @GDK_PIXBUF_LIBS@
 
 testgpu_SOURCES=testgpu.cpp $(ALL_H)
 testgpu_LDADD = ./libgraphics.la ../base/libbase.la -ldl \
-        @XML2_LIBS@ -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@ $(X_LIBS) \
+        @XML2_LIBS@ @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@ $(X_LIBS) \
         @GL_LIBS@ @GLU_LIBS@ @SDL_LIBS@ \
         @GDK_PIXBUF_LIBS@
 
diff -urNw libavg-1.7.1/src/imaging/Makefile.am libavg-1.7.1.patched/src/imaging/Makefile.am
--- libavg-1.7.1/src/imaging/Makefile.am	2012-01-26 17:01:32.000000000 +0100
+++ libavg-1.7.1.patched/src/imaging/Makefile.am	2012-07-24 10:18:34.683229569 +0200
@@ -38,8 +38,8 @@
 
 testimaging_SOURCES=testimaging.cpp $(ALL_H)
 testimaging_LDADD = ./libimaging.la ../graphics/libgraphics.la ../base/libbase.la \
-        @XML2_LIBS@ -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@ @GDK_PIXBUF_LIBS@
+        @XML2_LIBS@ @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@ @GDK_PIXBUF_LIBS@
 
 checktracking_SOURCES=checktracking.cpp $(ALL_H)
 checktracking_LDADD = ./libimaging.la ../graphics/libgraphics.la ../base/libbase.la \
-        @XML2_LIBS@ -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@ @GDK_PIXBUF_LIBS@
+        @XML2_LIBS@ @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@ @GDK_PIXBUF_LIBS@
diff -urNw libavg-1.7.1/src/player/Makefile.am libavg-1.7.1.patched/src/player/Makefile.am
--- libavg-1.7.1/src/player/Makefile.am	2012-01-26 17:01:31.000000000 +0100
+++ libavg-1.7.1.patched/src/player/Makefile.am	2012-07-24 10:18:46.999229671 +0200
@@ -67,7 +67,7 @@
 testplayer_LDADD = libplayer.la ../video/libvideo.la ../audio/libaudio.la \
         ../imaging/libimaging.la ../graphics/libgraphics.la ../base/libbase.la \
         ../lmfit/liblmfit.la  ../oscpack/liboscpack.la \
-        @XML2_LIBS@ -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@ @PANGOFT2_LIBS@ @LIBRSVG_LIBS@\
+        @XML2_LIBS@ @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@ @PANGOFT2_LIBS@ @LIBRSVG_LIBS@\
         @DC1394_2_LIBS@ @GLU_LIBS@ $(ALL_GL_LIBS) $(XI2_1_LIBS) $(XI2_2_LIBS) \
         @LIBFFMPEG@ $(BOOST_PYTHON_LIBS) $(PYTHON_LIBS) @GDK_PIXBUF_LIBS@ \
         @FONTCONFIG_LIBS@
@@ -78,7 +78,7 @@
 testcalibrator_LDADD = libplayer.la ../video/libvideo.la ../audio/libaudio.la \
         ../imaging/libimaging.la ../graphics/libgraphics.la ../base/libbase.la \
         ../lmfit/liblmfit.la ../oscpack/liboscpack.la \
-        @XML2_LIBS@ -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@ @LIBRSVG_LIBS@ @GDK_PIXBUF_LIBS@
+        @XML2_LIBS@ @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@ @LIBRSVG_LIBS@ @GDK_PIXBUF_LIBS@
 
 libplayer_la_LIBADD = $(BOOST_PYTHON_LIBS) $(PYTHON_LIBS) $(MTDEV_LIBS)
 libplayer_la_SOURCES = $(GL_SOURCES) \
diff -urNw libavg-1.7.1/src/video/Makefile.am libavg-1.7.1.patched/src/video/Makefile.am
--- libavg-1.7.1/src/video/Makefile.am	2011-12-07 13:59:59.000000000 +0100
+++ libavg-1.7.1.patched/src/video/Makefile.am	2012-07-24 10:18:54.586229738 +0200
@@ -41,7 +41,7 @@
 testvideo_SOURCES = testvideo.cpp $(ALL_H)
 testvideo_LDADD = ./libvideo.la ../audio/libaudio.la ../graphics/libgraphics.la \
         ../base/libbase.la \
-        @XML2_LIBS@  -l@BOOST_THREAD_LIB@ @PTHREAD_LIBS@ \
+        @XML2_LIBS@  @BOOST_THREAD_LIBS@ @PTHREAD_LIBS@ \
         @LIBFFMPEG@ $(X_LIBS) \
         @GDK_PIXBUF_LIBS@
 
diff -urNw libavg-1.7.1/src/wrapper/Makefile.am libavg-1.7.1.patched/src/wrapper/Makefile.am
--- libavg-1.7.1/src/wrapper/Makefile.am	2012-01-26 17:01:30.000000000 +0100
+++ libavg-1.7.1.patched/src/wrapper/Makefile.am	2012-07-24 10:19:02.181229799 +0200
@@ -32,7 +32,7 @@
                 ../lmfit/liblmfit.la \
                 ../oscpack/liboscpack.la \
                 @PANGOFT2_LIBS@ @LIBRSVG_LIBS@ $(BOOST_PYTHON_LIBS)\
-                -l@BOOST_THREAD_LIB@ @XML2_LIBS@ \
+                @BOOST_THREAD_LIBS@ @XML2_LIBS@ \
                 @DC1394_2_LIBS@ @GLU_LIBS@ $(XI2_1_LIBS) $(XI2_2_LIBS) \
                 $(ALL_GL_LIBS) @LIBFFMPEG@ @PTHREAD_LIBS@ \
                 @GDK_PIXBUF_LIBS@ @FONTCONFIG_LIBS@
