--- usr/lib/python2.7/site-packages/nautilus_image_manipulator/nautilus_image_manipulatorconfig.py.old	2012-01-26 14:06:34.000000000 -0200
+++ usr/lib/python2.7/site-packages/nautilus_image_manipulator/nautilus_image_manipulatorconfig.py	2012-01-26 14:09:49.205675437 -0200
@@ -28,7 +28,7 @@
 
 # Where your project will look for your data (for instance, images and ui
 # files). By default, this is ../data, relative your trunk layout
-__nautilus_image_manipulator_data_directory__ = '../data/'
+__nautilus_image_manipulator_data_directory__ = '/usr/share/nautilus-image-manipulator/'
 __license__ = 'GPL-3'
 
 import os
