diff -aur src/tintii-2.5.2/config.py src2//tintii-2.5.2/config.py
--- src/tintii-2.5.2/config.py	2011-02-23 12:33:01.000000000 +0100
+++ src2//tintii-2.5.2/config.py	2011-04-22 15:31:55.000000000 +0200
@@ -13,5 +13,5 @@
 # Linux/g++ compiler options
 CXX='g++'
 CXXFLAGS='-O3 -Wall -funroll-loops -fomit-frame-pointer `wx-config --cxxflags` -fopenmp -ffast-math -g'
-LINKFLAGS='`wx-config --libs` `wx-config --libs aui` -fopenmp -g'
+LINKFLAGS='`wx-config --libs | sed "s|-Wl.*ded ||"` `wx-config --libs aui | sed "s|-Wl.*ded ||"` -fopenmp -g'
 #NDEBUG='yes'
diff -aur src/tintii-2.5.2/src/indii/tint/gui/PreviewImage.cpp src2//tintii-2.5.2/src/indii/tint/gui/PreviewImage.cpp
--- src/tintii-2.5.2/src/indii/tint/gui/PreviewImage.cpp	2011-04-21 08:53:44.000000000 +0200
+++ src2//tintii-2.5.2/src/indii/tint/gui/PreviewImage.cpp	2011-04-22 15:23:59.000000000 +0200
@@ -251,11 +251,9 @@
   GetViewStart(&x, &y);
   GetScrollPixelsPerUnit(&xUnit, &yUnit);
   
-  if (evt.GetWheelAxis() == 0) {
-	  y -= evt.GetWheelRotation();
-  } else {
-    x -= evt.GetWheelRotation();
-  }
+  y -= evt.GetWheelRotation();
+  x -= evt.GetWheelRotation();
+
   Scroll(x, y);
 }
 
diff -aur src/tintii-2.5.2/src/indii/tint/gui/TintAbout.cpp src2//tintii-2.5.2/src/indii/tint/gui/TintAbout.cpp
--- src/tintii-2.5.2/src/indii/tint/gui/TintAbout.cpp	2011-04-21 08:53:44.000000000 +0200
+++ src2//tintii-2.5.2/src/indii/tint/gui/TintAbout.cpp	2011-04-22 15:23:59.000000000 +0200
@@ -22,5 +22,5 @@
   info.SetWebSite(_("http://www.indii.org/tintii"));
   info.SetCopyright(_("Copyright 2007-2011 Lawrence Murray"));
 
-  wxAboutBox(info, parent);
+  wxAboutBox(info);
 }
diff -aur src/tintii-2.5.2/src/indii/tint/gui/TintHelp.cpp src2//tintii-2.5.2/src/indii/tint/gui/TintHelp.cpp
--- src/tintii-2.5.2/src/indii/tint/gui/TintHelp.cpp	2011-04-21 08:53:44.000000000 +0200
+++ src2//tintii-2.5.2/src/indii/tint/gui/TintHelp.cpp	2011-04-22 15:23:59.000000000 +0200
@@ -84,8 +84,8 @@
     }
 		#endif
   } else {
-    wxMessageBox(wxString("The tintii online help system could not be loaded.\n\nMake sure "
-        "that the tintii.htb file is in the ") <<
+    wxMessageBox(wxString(_T("The tintii online help system could not be loaded.\n\nMake sure ")
+        _T("that the tintii.htb file is in the ")) <<
         wxStandardPaths::Get().GetResourcesDir() <<
         _(" directory, and restart tintii."),
         _("Error"), wxOK|wxICON_HAND, NULL);
diff -aur src/tintii-2.5.2/src/indii/tint/gui/TintRegistration.cpp src2//tintii-2.5.2/src/indii/tint/gui/TintRegistration.cpp
--- src/tintii-2.5.2/src/indii/tint/gui/TintRegistration.cpp	2011-04-21 08:53:44.000000000 +0200
+++ src2//tintii-2.5.2/src/indii/tint/gui/TintRegistration.cpp	2011-04-22 15:23:59.000000000 +0200
@@ -75,7 +75,7 @@
   fntUrl.SetUnderlined(true);
   butUrl->SetForegroundColour(*wxBLUE);
 	#endif
-  lblTitle->SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT).Bold());
+  lblTitle->SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
   lblDesc->SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
         
   txtKey = new wxTextCtrl(this, wxID_ANY, wxEmptyString, wxDefaultPosition,
diff -aur src/tintii-2.5.2/src/indii/tint/gui/TintSplash.cpp src2//tintii-2.5.2/src/indii/tint/gui/TintSplash.cpp
--- src/tintii-2.5.2/src/indii/tint/gui/TintSplash.cpp	2011-04-21 08:53:44.000000000 +0200
+++ src2//tintii-2.5.2/src/indii/tint/gui/TintSplash.cpp	2011-04-22 15:23:59.000000000 +0200
@@ -51,7 +51,7 @@
     } else {
 	    dc.SetBrush(*wxBLACK);
     }
-    dc.SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT).Bold());
+    dc.SetFont(wxSystemSettings::GetFont(wxSYS_DEFAULT_GUI_FONT));
     dc.SetPen(wxNullPen);
     dc.SetTextBackground(*wxWHITE);
     dc.SetTextForeground(*wxWHITE);
diff -aur src/tintii-2.5.2/src/indii/tint/gui/TintWindow.cpp src2//tintii-2.5.2/src/indii/tint/gui/TintWindow.cpp
--- src/tintii-2.5.2/src/indii/tint/gui/TintWindow.cpp	2011-04-21 08:53:44.000000000 +0200
+++ src2//tintii-2.5.2/src/indii/tint/gui/TintWindow.cpp	2011-04-22 15:23:59.000000000 +0200
@@ -96,7 +96,7 @@
   /* save AUI perspective */
   wxConfig config(_("tintii"), _("indii.org"));
   wxString persp = manager->SavePerspective();
-  config.Write(wxString("Perspective"), persp);
+  config.Write(wxString(_T("Perspective")), persp);
 
   manager->UnInit();
 }
@@ -637,8 +637,8 @@
   	tbTop->AddControl(new wxContextHelpButton(tbTop), _T("Context help"));
   }
   #else
-  tbTop->AddStretchableSpace();
-  tbTop->AddControl(new wxContextHelpButton(tbTop), _T("Context help"));
+  tbTop->AddSeparator();
+  tbTop->AddControl(new wxContextHelpButton(tbTop));
   #endif
   tbTop->Realize();  
   
@@ -672,7 +672,7 @@
   width = MIN_THUMB_PANE_WIDTH;
   height = 512;
   
-  paneInfo.Name("Thumbs");
+  paneInfo.Name(_T("Thumbs"));
   paneInfo.MinSize(wxSize(width, 110));
   paneInfo.BestSize(wxSize(width, height));
   paneInfo.FloatingSize(width + EXTRA_PANE_WIDTH, height + EXTRA_PANE_HEIGHT);
@@ -686,7 +686,7 @@
   width = prePane->GetMinSize().GetWidth();
   height = prePane->GetMinSize().GetHeight();
   
-  paneInfo.Name("Preprocessing");
+  paneInfo.Name(_T("Preprocessing"));
   paneInfo.MinSize(wxSize(width, height));
   paneInfo.BestSize(wxSize(width, height));
   paneInfo.FloatingSize(width + EXTRA_PANE_WIDTH, height + EXTRA_PANE_HEIGHT);
@@ -703,7 +703,7 @@
   width = postPane->GetMinSize().GetWidth();
   height = postPane->GetMinSize().GetHeight();
   
-  paneInfo.Name("Postprocessing");
+  paneInfo.Name(_T("Postprocessing"));
   paneInfo.MinSize(wxSize(width, height));
   paneInfo.BestSize(wxSize(width, height));
   paneInfo.FloatingSize(width + EXTRA_PANE_WIDTH, height + EXTRA_PANE_HEIGHT);
@@ -718,7 +718,7 @@
   width = mixerPane->GetMinSize().GetWidth();
   height = mixerPane->GetMinSize().GetHeight();
   
-  paneInfo.Name("Mixer");
+  paneInfo.Name(_T("Mixer"));
   paneInfo.MinSize(wxSize(width, height));
   paneInfo.BestSize(wxSize(width, height));
   paneInfo.FloatingSize(width + EXTRA_PANE_WIDTH, height + EXTRA_PANE_HEIGHT);
@@ -770,7 +770,7 @@
 
     wxAuiPaneInfo paneInfo;
     paneInfo.DefaultPane();
-    paneInfo.Name("Buttons");
+    paneInfo.Name(_T("Buttons"));
     paneInfo.MinSize(wxSize(width, height));
     paneInfo.BestSize(wxSize(width, height));
     paneInfo.CaptionVisible(false);
diff -aur src/tintii-2.5.2/src/indii/tint/ImageCache.cpp src2//tintii-2.5.2/src/indii/tint/ImageCache.cpp
--- src/tintii-2.5.2/src/indii/tint/ImageCache.cpp	2011-04-21 08:53:44.000000000 +0200
+++ src2//tintii-2.5.2/src/indii/tint/ImageCache.cpp	2011-04-22 15:23:59.000000000 +0200
@@ -40,7 +40,7 @@
     if (find != cache.end()) {
       image = find->second;
     } else {
-      wxImageResizeQuality quality;
+      int quality;
       if (high) {
         quality = wxIMAGE_QUALITY_HIGH;
       } else {
