diff -crB CGI-SpeedyCGI-2.22-unpatched/src/speedy_backend_main.h CGI-SpeedyCGI-2.22-patched/src/speedy_backend_main.h
*** CGI-SpeedyCGI-2.22-unpatched/src/speedy_backend_main.h	2003-10-07 07:03:48.000000000 +0300
--- CGI-SpeedyCGI-2.22/src/speedy_backend_main.h		2011-03-09 16:43:07.216350379 +0200
***************
*** 38,44 ****
  
  #else
  
! #define speedy_new(s,n,t)	New(123,s,n,t)
  #define speedy_renew		Renew
  #define speedy_free		Safefree
  
--- 38,44 ----
  
  #else
  
! #define speedy_new(s,n,t)	Newx(s,n,t)
  #define speedy_renew		Renew
  #define speedy_free		Safefree
  
