diff -crB CGI-SpeedyCGI-2.22-unpatched//Makefile.PL CGI-SpeedyCGI-2.22-patched//Makefile.PL
*** CGI-SpeedyCGI-2.22-unpatched//Makefile.PL	2011-03-09 18:28:18.665423958 +0200
--- CGI-SpeedyCGI-2.22/Makefile.PL		2011-03-09 18:30:25.644020175 +0200
***************
*** 6,33 ****
      import SpeedyMake qw(%write_makefile_common);
  }
  
- print <<END;
- 
- Optional mod_speedycgi support.
- 
- Mod_speedycgi increases performance under Apache by avoiding the fork/exec
- overhead associated with each request under normal SpeedyCGI.  However, it
- requires a working copy of "apxs" in your path, Apache with mod_so
- support, and additional Apache configuration.
- 
- END
- print "Compile mod_speedycgi (default no)? ";
- 
  my @dirs = qw(src speedy_backend speedy);
  my $macro = $write_makefile_common{macro};
  my $apache_module = 0;
- if (<STDIN> =~ /y/i) {
-     die "ERROR: Command 'apxs -q CC' failed.\n"
- 	unless $macro->{APACHE_APXS_WORKS};
-     print "Compiling for Apache version $macro->{APACHE_VERSION}\n";
-     push(@dirs, $macro->{MOD_SPEEDYCGI_DIR});
-     $apache_module = 1;
- }
  
  WriteMakefile(
      NAME		=> 'CGI::SpeedyCGI',
--- 6,14 ----
diff -crB CGI-SpeedyCGI-2.22-unpatched//src/speedy_backend_main.h CGI-SpeedyCGI-2.22-patched//src/speedy_backend_main.h
*** CGI-SpeedyCGI-2.22-unpatched//src/speedy_backend_main.h	2011-03-09 18:28:18.722089998 +0200
--- CGI-SpeedyCGI-2.22/src/speedy_backend_main.h		2011-03-09 18:28:43.578481719 +0200
***************
*** 38,44 ****
  
  #else
  
! #define speedy_new(s,n,t)	New(123,s,n,t)
  #define speedy_renew		Renew
  #define speedy_free		Safefree
  
--- 38,44 ----
  
  #else
  
! #define speedy_new(s,n,t)	Newx(s,n,t)
  #define speedy_renew		Renew
  #define speedy_free		Safefree
  
