diff -Nru clementine-1.0.1/src/main.cpp clementine-1.0.1.new/src/main.cpp
--- clementine-1.0.1/src/main.cpp	2012-01-22 13:43:26.000000000 +0100
+++ clementine-1.0.1.new/src/main.cpp	2012-08-18 08:53:27.378700849 +0200
@@ -453,10 +453,7 @@
   QFile self_maps("/proc/self/maps");
   if (self_maps.open(QIODevice::ReadOnly)) {
     QByteArray data = self_maps.readAll();
-    if (data.contains("libnvidia-tls.so.285.03") ||
-        data.contains("libnvidia-tls.so.280.13") ||
-        data.contains("libnvidia-tls.so.275.28") ||
-        data.contains("libnvidia-tls.so.275.19")) {
+    if (data.contains("libnvidia-tls.so")) {
       qLog(Warning) << "Exiting immediately to work around NVIDIA driver bug";
       _exit(ret);
     }
