--- groundfire-0.25.orig/src/buttons.cc	2004-04-03 03:11:23.000000000 +0600
+++ groundfire-0.25/src/buttons.cc	2012-04-29 00:21:26.770389401 +0600
@@ -29,6 +29,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 #include "buttons.hh"
 #include "font.hh"
+#include <cstring>
 
 ////////////////////////////////////////////////////////////////////////////////
 // Public Member Functions
--- groundfire-0.25.orig/src/inifile.cc	2004-04-04 01:44:44.000000000 +0600
+++ groundfire-0.25/src/inifile.cc	2012-04-29 00:21:27.463722740 +0600
@@ -33,6 +33,7 @@
 #include <windows.h>
 #endif
 #include "inifile.hh"
+#include <cstdlib>
 
 ////////////////////////////////////////////////////////////////////////////////
 // Public Member Functions
--- groundfire-0.25.orig/src/landscape.cc	2004-05-13 16:45:06.000000000 +0600
+++ groundfire-0.25/src/landscape.cc	2012-04-29 00:21:26.760389401 +0600
@@ -35,6 +35,7 @@
 
 #include <math.h>
 #include <cstdio>
+#include <cstdlib>
 
 ////////////////////////////////////////////////////////////////////////////////
 // Public Member Functions
--- groundfire-0.25.orig/src/selector.cc	2004-04-03 03:53:52.000000000 +0600
+++ groundfire-0.25/src/selector.cc	2012-04-29 00:21:26.797056067 +0600
@@ -29,6 +29,8 @@
 ////////////////////////////////////////////////////////////////////////////////
 #include "selector.hh"
 #include "font.hh"
+#include <cstdlib>
+#include <string.h>
 
 ////////////////////////////////////////////////////////////////////////////////
 // Public Member Functions
--- groundfire-0.25.orig/Makefile	2004-05-15 17:45:04.000000000 +0600
+++ groundfire-0.25/Makefile	2012-04-29 00:26:07.687057656 +0600
@@ -21,13 +21,13 @@
 
 UNIXFLAGS    = -O3 -fomit-frame-pointer -ffast-math
 UNIXINCLUDE  = -I/usr/X11R6/include -I/usr/local/include
-UNIXLIBS     = -L/usr/X11R6/lib -L/usr/local/lib -lglfw -lGL -lGLU -lopenal -lX11 -lXxf86vm -lXext -pthread -lm
+UNIXLIBS     = -L/usr/X11R6/lib -L/usr/local/lib -lglfw -lGL -lGLU -lopenal -lX11 -lXxf86vm -lXext -pthread -lm -lalut
 
 win32::
 	cd src ; $(MAKE) TARGET=$(EXENAME) "FLAGS=$(WIN32FLAGS)" "INCLUDE=$(WIN32INCLUDE)" "LIBS=$(WIN32LIBS)"
 
 unix:
-	cd src ; $(MAKE) TARGET=$(EXENAME) "FLAGS=$(UNIXFLAGS)" "INCLUDE=$(UNIXINCLUDE)" "LIBS=$(UNIXLIBS)"
+	cd src ; $(MAKE) TARGET=$(EXENAME) "FLAGS=$(CFLAGS)" "INCLUDE=$(UNIXINCLUDE)" "LIBS=$(UNIXLIBS)"
 
 clean:
 	rm -f $(EXENAME)*
