=== modified file 'src/MainWindow.cs'
--- src/MainWindow.cs	2011-01-19 23:48:39 +0000
+++ src/MainWindow.cs	2011-01-21 23:05:47 +0000
@@ -192,11 +192,11 @@
         {
             if ( ( bool ) Config.Client.Get ( Config.Settings.StartInTray ) )
             {
-                this.Hide ();
+                window.Hide ();
             }
             else
             {
-                this.Show ();
+                window.Show ();
             }
         }
         else

