--- textadept.c	2014-03-27 15:11:07.000000000 +0100
+++ textadept.c.new	2014-03-27 15:13:06.000000000 +0100
@@ -2391,9 +2391,10 @@
   char *last_slash = NULL;
 #if __linux__
   textadept_home = malloc(FILENAME_MAX);
-  int len = readlink("/proc/self/exe", textadept_home, FILENAME_MAX);
-  textadept_home[len] = '\0';
-  if ((last_slash = strrchr(textadept_home, '/'))) *last_slash = '\0';
+//  int len = readlink("/proc/self/exe", textadept_home, FILENAME_MAX);
+//  textadept_home[len] = '\0';
+//  if ((last_slash = strrchr(textadept_home, '/'))) *last_slash = '\0';
+  strcpy(textadept_home, "/usr/share/textadept");
 #elif _WIN32
   textadept_home = malloc(FILENAME_MAX);
   GetModuleFileName(0, textadept_home, FILENAME_MAX);
