Index: pidgin-libnotify/src/pidgin-libnotify.c
===================================================================
--- pidgin-libnotify.orig/src/pidgin-libnotify.c	2011-08-10 22:46:33.003107473 -0400
+++ pidgin-libnotify/src/pidgin-libnotify.c	2011-08-10 22:46:33.963108883 -0400
@@ -438,7 +438,7 @@
 		return;
 	}
 
-	notification = notify_notification_new (title, tr_body, "notification-message-im", NULL);
+	notification = notify_notification_new (title, tr_body, "notification-message-im");
 	purple_debug_info (PLUGIN_ID, "notify(), new: "
 					 "title: '%s', body: '%s', buddy: '%s'\n",
 					 title, tr_body, buddy != NULL ? best_name (buddy) : "(null)");
