diff --git a/configure.ac b/configure.ac
index e3044b8..72a22ca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -121,7 +121,7 @@ PKG_CHECK_MODULES(DISPLAY_PANEL, $COMMON_MODULES cinnamon-desktop x11)
 PKG_CHECK_MODULES(MEDIA_PANEL, $COMMON_MODULES)
 
 PKG_CHECK_MODULES(NETWORK_PANEL, $COMMON_MODULES)
-PKG_CHECK_MODULES(POWER_PANEL, $COMMON_MODULES upower-glib >= 0.9.1
+PKG_CHECK_MODULES(POWER_PANEL, $COMMON_MODULES upower-glib >= 0.9.2
                   cinnamon-settings-daemon >= $CSD_REQUIRED_VERSION)
 PKG_CHECK_MODULES(COLOR_PANEL, $COMMON_MODULES colord >= 0.1.8)
 PKG_CHECK_MODULES(REGION_PANEL, $COMMON_MODULES libgnomekbd >= 2.91.91
diff --git a/panels/power/cc-power-panel.c b/panels/power/cc-power-panel.c
index fb8e1b4..5ab0b21 100644
--- a/panels/power/cc-power-panel.c
+++ b/panels/power/cc-power-panel.c
@@ -934,10 +934,10 @@ disable_unavailable_combo_items (CcPowerPanel *self,
                           -1);
       switch (value_tmp) {
       case CSD_POWER_ACTION_SUSPEND:
-        enabled = cc_login1("CanSuspend") || up_client_get_can_suspend (self->priv->up_client);
+        enabled = cc_login1("CanSuspend");
         break;
       case CSD_POWER_ACTION_HIBERNATE:
-        enabled = cc_login1("CanHibernate") || up_client_get_can_hibernate (self->priv->up_client);
+        enabled = cc_login1("CanHibernate");
         break;
       default:
         enabled = TRUE;
