diff -Naur ntlmaps-0.9.9.0.1/__init__.py ntlmaps-0.9.9.0.1_new/__init__.py
--- ntlmaps-0.9.9.0.1/__init__.py	2005-06-10 05:49:35.000000000 +0200
+++ ntlmaps-0.9.9.0.1_new/__init__.py	2007-05-28 16:02:14.000000000 +0200
@@ -20,10 +20,10 @@
 # 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
 #
 
-import os, sys
+#import os, sys
 
-ntlmaps_dir = os.path.dirname(os.path.abspath(__file__))
-ntlmaps_libdir = ntlmaps_dir + '/lib'
-sys.path.append(ntlmaps_libdir)
+#ntlmaps_dir = os.path.dirname(os.path.abspath(__file__))
+#ntlmaps_libdir = ntlmaps_dir
+#sys.path.append(ntlmaps_libdir)
 
-del os, sys
+#del os, sys
diff -Naur ntlmaps-0.9.9.0.1/lib/config.py ntlmaps-0.9.9.0.1_new/lib/config.py
--- ntlmaps-0.9.9.0.1/lib/config.py	2005-06-10 05:46:43.000000000 +0200
+++ ntlmaps-0.9.9.0.1_new/lib/config.py	2007-05-28 15:59:49.000000000 +0200
@@ -52,7 +52,7 @@
     the config file. Otherwise the function falls back to
     'server.cfg' in the current directory. """
 
-    configFileName = configFileDir+'server.cfg'
+    configFileName = configFileDir+'ntlmaps'
 
     optionsList, notUsedArguments = getopt.getopt(argv[1:], 'c:')
 
diff -Naur ntlmaps-0.9.9.0.1/main.py ntlmaps-0.9.9.0.1_new/main.py
--- ntlmaps-0.9.9.0.1/main.py	2006-01-27 02:35:00.000000000 +0100
+++ ntlmaps-0.9.9.0.1_new/main.py	2007-05-28 16:00:22.000000000 +0200
@@ -18,17 +18,13 @@
 # Free Software Foundation, Inc.,
 # 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
 #
-import __init__
-
 import sys
-
-import server, config, config_affairs
-
+from ntlmaps import server, config, config_affairs
 
 #--------------------------------------------------------------
 # config affairs
 # look for default config name in lib/config.py
-conf = config.read_config(config.findConfigFileNameInArgv(sys.argv, __init__.ntlmaps_dir+'/'))
+conf = config.read_config(config.findConfigFileNameInArgv(sys.argv, '/etc/conf.d/'))
 
 conf['GENERAL']['VERSION'] = '0.9.9.0.1'
 
