diff -aur package.pristine/xdxf_makedict/src/log.cpp package.new/xdxf_makedict/src/log.cpp
--- package.pristine/xdxf_makedict/src/log.cpp	2014-09-21 18:01:48.511463611 +0200
+++ package.new/xdxf_makedict/src/log.cpp	2014-09-21 18:02:31.072079909 +0200
@@ -68,11 +68,11 @@
 	g_log_set_default_handler(log, this);
 }
 
-void g_info(const gchar *fmt, ...)
+/*void g_info(const gchar *fmt, ...)
 {
 	va_list va;
 
 	va_start(va, fmt);
 	g_logv(G_LOG_DOMAIN, G_LOG_LEVEL_INFO, fmt, va);
 	va_end(va);
-}
+}*/
diff -aur package.pristine/xdxf_makedict/src/log.hpp package.new/xdxf_makedict/src/log.hpp
--- package.pristine/xdxf_makedict/src/log.hpp	2014-09-21 18:01:48.511463611 +0200
+++ package.new/xdxf_makedict/src/log.hpp	2014-09-21 18:02:17.555218920 +0200
@@ -19,6 +19,6 @@
 			gpointer user_data);
 };
 
-extern void g_info(const gchar *fmt, ...);
+//extern void g_info(const gchar *fmt, ...);
 
 #endif//!_LOG_HPP_
