--- src/rt3573sta/common/rtusb_dev_id.c	2013-10-21 16:29:34.342946124 -0200
+++ rtusb_dev_id.c	2013-09-26 17:51:42.000000000 -0300
@@ -56,6 +56,7 @@
 	{USB_DEVICE(0x0b05,0x1784)}, /* Asus USB-N13 */
     {USB_DEVICE(0x0B05,0x17AD)}, /* ASUS USB-N66 */
 	{USB_DEVICE(0x0df6,0x006e)}, /* Sitecom WLA-6100 */
+	{USB_DEVICE(0x13b1,0x003b)}, /* Cisco LinkSys AE3000 */
 #endif /* RT3573 */
 	{ }/* Terminating entry */
 };
