diff -Nur template2pdf-0.6.4.orig/template2pdf/t2p/trml2pdf.py template2pdf-0.6.4/template2pdf/t2p/trml2pdf.py
--- template2pdf-0.6.4.orig/template2pdf/t2p/trml2pdf.py	2011-06-24 15:05:47.000000000 +1000
+++ template2pdf-0.6.4/template2pdf/t2p/trml2pdf.py	2011-06-24 19:34:05.242224924 +1000
@@ -177,7 +177,7 @@
         from reportlab.pdfbase.ttfonts import TTFont
         faceName = params.get('faceName', '')
         fileName = params.get('fileName', '')
-        subfontIndex = int(params.get('subfontIndes', '0'))        
+        subfontIndex = int(params.get('subfontIndex', '0'))        
         font = FONT_CACHE.setdefault(
             (faceName, fileName), TTFont(faceName, fileName, False, subfontIndex))
     return font
