--- Makefile	2004-10-16 17:57:26.000000000 -0500
+++ Makefile	2011-02-24 18:41:32.170000298 -0600
@@ -85,7 +85,7 @@
 #         AUDIO THEN UNCOMMENT THE NEXT LINES, ELSE COMMENT THEM OUT
 #
 # ------------------------------------------------------------
-AUDIOLIB = -lopenal
+AUDIOLIB = -lopenal -lalut
 AUDIODEF = -I/usr/include/AL -DSOUND 
 
 
@@ -103,7 +103,7 @@
 
 #-------------------------------------------------------------------
 
-TARGET 	= battalion
+TARGET 	= battalion2004
 
 OBJ 	= main.o audio.o gprim.o graphics.o objects.o text.o update.o  font.o tgaload.o
 LIBS    = -L/usr/X11R6/lib -lm  
