diff -rupN a/src/gengine.c b/src/gengine.c
--- a/src/gengine.c	2011-03-22 16:57:41.000000000 +0000
+++ b/src/gengine.c	2014-01-29 10:57:28.933098365 +0000
@@ -81,7 +81,7 @@ FilmInfo* get_movie_info(const char *fna
 	av_register_all();
 	movie_info->filename = g_strdup(fname);
 	movie_info->size = get_file_size(fname);
-	if (av_open_input_file(&pFormatCtx, fname, NULL, 0, NULL) != 0)
+	if (avformat_open_input(&pFormatCtx, fname, NULL, NULL) != 0)
 		return NULL;
 	if (av_find_stream_info(pFormatCtx) < 0)
 		return NULL;
@@ -93,7 +93,7 @@ FilmInfo* get_movie_info(const char *fna
 	AVCodec *pCodec = avcodec_find_decoder(pCodecCtx->codec_id);
 	if (pCodec == NULL)
 		return NULL;
-	if (avcodec_open(pCodecCtx, pCodec) < 0) return NULL;
+	if (avcodec_open2(pCodecCtx, pCodec, NULL) < 0) return NULL;
 	pCodecCtx = (pFormatCtx)->streams[videoStream]->codec;
 	streams = pFormatCtx->nb_streams;
 #if LIBAVCODEC_BUILD > 4753
