--- ppp.orig/scripts/ipv6-up.sample	Mon Aug 23 10:07:07 1999
+++ ppp/scripts/ipv6-up.sample	Thu Jul 13 19:16:26 2000
@@ -8,26 +8,27 @@
 # Start router advertisements on this link.
 # Based on radvd 0.5.0 behaviour
 
-DEVICE=$1
+DEVICE="$1"
 
-CFGFILE=/usr/inet6/etc/radvd.conf-$DEVICE
-PIDFILE=/var/run/radvd-$DEVICE.pid
+CFGFILE="/usr/inet6/etc/radvd.conf-$DEVICE"
+PIDFILE="/var/run/radvd-$DEVICE.pid"
+EXEFILE="/usr/inet6/sbin/radvd"
 
-if [ -x /usr/inet6/sbin/radvd && -f $CFGFILE ]; then
-    touch $PIDFILE
-    if [ ! -f $PIDFILE ]; then
+if [ -x "$EXEFILE" -a -f "$CFGFILE" ]; then
+    touch "$PIDFILE"
+    if [ ! -f "$PIDFILE" ]; then
 	echo "error: $PIDFILE is not a regular file. Aborting"
 	exit 0
     fi
 
-    PID=`cat $PIDFILE`
-    if [ "$PID" != "" ]; then
-	ps h $PID >/dev/null 2>&1 && exit 0
+    PID="$(cat "$PIDFILE")"
+    if [ -n "$PID" ]; then
+	ps h "$PID" &> /dev/null && exit 0
     fi
 
     # radvd 0.5.0 doesn't write a pid-file so we do it here
     # enabling debugging keeps radvd in foreground, putting it
     # on background gives us the PID.
-    /usr/inet6/sbin/radvd -d 1 -C $CFGFILE &
-    echo $! >$PIDFILE
+    "$EXEFILE" -d 1 -C "$CFGFILE" &
+    echo $! >"$PIDFILE"
 fi
--- ppp.orig/scripts/pon	2002-11-25 02:30:44 +0300
+++ ppp/scripts/pon	2003-10-08 19:27:03 +0400
@@ -1,10 +1,12 @@
 #!/bin/sh
 
+PPP_ON_BOOT=/etc/ppp/ppp_on_boot
+
 case "$1" in
   -*) echo "
 Usage: pon [provider] [arguments]
 
-If pon is invoked without arguments, /etc/ppp/ppp_on_boot file will be
+If pon is invoked without arguments, $PPP_ON_BOOT file will be
 run, presuming it exists and is executable. Otherwise, a PPP connection
 will be started using settings from /etc/ppp/peers/provider.
 If you specify one argument, a PPP connection will be started using
@@ -16,8 +18,8 @@
       ;;
 esac
 
-if [ -z "$1" -a -x /etc/ppp/ppp_on_boot ]; then
-  exec /etc/ppp/ppp_on_boot
+if [ -z "$1" -a -x "$PPP_ON_BOOT" ]; then
+  exec "$PPP_ON_BOOT"
 fi
 
 if [ -z "$1" -a ! -f /etc/ppp/peers/provider ]; then
