--- ppp-2.4.2/pppd/ipcp.c.vns	2004-01-13 06:59:06 +0300
+++ ppp-2.4.2/pppd/ipcp.c	2004-05-28 12:08:36 +0400
@@ -90,6 +90,7 @@
 static int default_route_set[NUM_PPP];	/* Have set up a default route */
 static int proxy_arp_set[NUM_PPP];	/* Have created proxy arp entry */
 static bool usepeerdns;			/* Ask peer for DNS addrs */
+static bool noresolvconf;		/* Don't modify /etc/resolv.conf */
 static int ipcp_is_up;			/* have called np_up() */
 static int ipcp_is_open;		/* haven't called np_finished() */
 static bool ask_for_local;		/* request our address from peer */
@@ -206,6 +207,9 @@
       "disable proxyarp option", OPT_ALIAS | OPT_A2CLR,
       &ipcp_wantoptions[0].proxy_arp },
 
+    { "noresolvconf", o_bool, &noresolvconf,
+      "Don't modify /etc/resolv.conf when ask peer for DNS address(es)", 1 },
+
     { "usepeerdns", o_bool, &usepeerdns,
       "Ask peer for DNS address(es)", 1 },
 
@@ -1693,7 +1697,8 @@
 	script_setenv("DNS2", ip_ntoa(go->dnsaddr[1]), 0);
     if (usepeerdns && (go->dnsaddr[0] || go->dnsaddr[1])) {
 	script_setenv("USEPEERDNS", "1", 0);
-	create_resolv(go->dnsaddr[0], go->dnsaddr[1]);
+	if (!noresolvconf)
+	    create_resolv(go->dnsaddr[0], go->dnsaddr[1]);
     }
 
     /*
