diff -urN ppp-2.4.1-old/scripts/callback ppp-2.4.1/scripts/callback
--- ppp-2.4.1-old/scripts/callback	Wed Aug  9 06:49:51 1995
+++ ppp-2.4.1/scripts/callback	Mon Jan 21 13:09:01 2002
@@ -33,9 +33,9 @@
 	ABORT		'\nNO ANSWER\r'			\
 	ABORT		'\nRINGING\r\n\r\nRINGING\r'	\
 	''		AT				\
-	'OK-+++\c-OK'	'AT&C0&D2S0=0H0			\
+	'OK-+++\c-OK'	'AT&C0&D2S0=0H0'		\
 	TIMEOUT		30				\
-	OK		ATDT$TELEPHONE			\
+	OK		ATDP$TELEPHONE			\
 	CONNECT		''				\
 	assword:	$MODEMPASS			\
 	"\nNO CARRIER\r"
