diff -rupN armstrong.old/src/libzzub/archive.h armstrong/src/libzzub/archive.h
--- armstrong.old/src/libzzub/archive.h	2009-01-09 16:17:44.000000000 -0500
+++ armstrong/src/libzzub/archive.h	2009-02-26 17:25:03.000000000 -0500
@@ -18,7 +18,7 @@ Foundation, Inc., 51 Franklin Street, Fi
 */
 
 #pragma once
-
+#include <cstdio>
 #include <vector>
 #include <list>
 
diff -rupN armstrong.old/src/libzzub/bmxreader.cpp armstrong/src/libzzub/bmxreader.cpp
--- armstrong.old/src/libzzub/bmxreader.cpp	2009-02-26 17:23:25.000000000 -0500
+++ armstrong/src/libzzub/bmxreader.cpp	2009-02-26 17:24:24.000000000 -0500
@@ -15,7 +15,7 @@ You should have received a copy of the G
 License along with this library; if not, write to the Free Software
 Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
-
+#include <cstdio>
 #include "common.h"
 #include <sstream>
 #include "archive.h"
diff -rupN armstrong.old/src/libzzub/host.cpp armstrong/src/libzzub/host.cpp
--- armstrong.old/src/libzzub/host.cpp	2009-01-09 16:17:44.000000000 -0500
+++ armstrong/src/libzzub/host.cpp	2009-02-26 17:27:47.000000000 -0500
@@ -15,7 +15,7 @@ You should have received a copy of the G
 License along with this library; if not, write to the Free Software
 Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
-
+#include <cstdio>
 #include "common.h"
 
 using namespace std;
diff -rupN armstrong.old/src/libzzub/pluginloader.cpp armstrong/src/libzzub/pluginloader.cpp
--- armstrong.old/src/libzzub/pluginloader.cpp	2009-01-09 16:17:44.000000000 -0500
+++ armstrong/src/libzzub/pluginloader.cpp	2009-02-26 17:30:48.000000000 -0500
@@ -16,7 +16,7 @@ You should have received a copy of the G
 License along with this library; if not, write to the Free Software
 Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
-
+#include <cstdio>
 #include <string>
 #include <algorithm>
 #include "common.h"
diff -rupN armstrong.old/src/libzzub/tools.cpp armstrong/src/libzzub/tools.cpp
--- armstrong.old/src/libzzub/tools.cpp	2009-01-09 16:17:44.000000000 -0500
+++ armstrong/src/libzzub/tools.cpp	2009-02-26 17:33:27.000000000 -0500
@@ -15,7 +15,7 @@ You should have received a copy of the G
 License along with this library; if not, write to the Free Software
 Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
-
+#include <cstdio>
 #include <string>
 
 #if defined(POSIX)
diff -rupN armstrong.old/src/plugins/bigyo_frequencyshifter/Bigyo_FrequencyShifter.cpp armstrong/src/plugins/bigyo_frequencyshifter/Bigyo_FrequencyShifter.cpp
--- armstrong.old/src/plugins/bigyo_frequencyshifter/Bigyo_FrequencyShifter.cpp	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/bigyo_frequencyshifter/Bigyo_FrequencyShifter.cpp	2009-02-26 17:45:09.000000000 -0500
@@ -16,6 +16,7 @@ License along with this library; if not,
 Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
 #define _USE_MATH_DEFINES
+#include <cstdio>
 #include <math.h>
 #include <float.h>
 #include <zzub/signature.h>
diff -rupN armstrong.old/src/plugins/btdsys_ringmod/ringmod.cpp armstrong/src/plugins/btdsys_ringmod/ringmod.cpp
--- armstrong.old/src/plugins/btdsys_ringmod/ringmod.cpp	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/btdsys_ringmod/ringmod.cpp	2009-02-26 17:47:29.000000000 -0500
@@ -27,7 +27,7 @@ ANY THEORY OF LIABILITY, WHETHER IN CONT
 (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
-
+#include <cstdio>
 #include <zzub/signature.h>
 #include <zzub/plugin.h>
 
diff -rupN armstrong.old/src/plugins/CyanPhase_DTMF-1/CyanPhase_DTMF-1.cpp armstrong/src/plugins/CyanPhase_DTMF-1/CyanPhase_DTMF-1.cpp
--- armstrong.old/src/plugins/CyanPhase_DTMF-1/CyanPhase_DTMF-1.cpp	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/CyanPhase_DTMF-1/CyanPhase_DTMF-1.cpp	2009-02-26 17:39:51.000000000 -0500
@@ -4,7 +4,7 @@
 // CyanPhase DTMF-1
 // Copyright 2000 CyanPhase aka Edward L. Blake
 // Enjoy
-
+#include <cstdio>
 #include <zzub/signature.h>
 #include <zzub/plugin.h>
 
diff -rupN armstrong.old/src/plugins/Intoxicat_ACloud/Cloud.cpp armstrong/src/plugins/Intoxicat_ACloud/Cloud.cpp
--- armstrong.old/src/plugins/Intoxicat_ACloud/Cloud.cpp	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/Intoxicat_ACloud/Cloud.cpp	2009-02-26 17:42:22.000000000 -0500
@@ -1,6 +1,6 @@
 #include <zzub/signature.h>
 #include <zzub/plugin.h>
-
+#include <cstdio>
 #include <stdlib.h>
 #include <math.h>
 #include <string.h>
diff -rupN armstrong.old/src/plugins/jmmcd_Crossfade/crossfade.cpp armstrong/src/plugins/jmmcd_Crossfade/crossfade.cpp
--- armstrong.old/src/plugins/jmmcd_Crossfade/crossfade.cpp	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/jmmcd_Crossfade/crossfade.cpp	2009-02-26 17:51:20.000000000 -0500
@@ -4,7 +4,7 @@
 
 // Lots of code from BTDSys RingMod -- thanks!
 
-
+#include <cstdio>
 #include <zzub/signature.h>
 #include <zzub/plugin.h>
 //#include "../buzz2zzub/mdk.h"
diff -rupN armstrong.old/src/plugins/jmmcd_Note_Pool/jmmcd_Note_Pool.cpp armstrong/src/plugins/jmmcd_Note_Pool/jmmcd_Note_Pool.cpp
--- armstrong.old/src/plugins/jmmcd_Note_Pool/jmmcd_Note_Pool.cpp	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/jmmcd_Note_Pool/jmmcd_Note_Pool.cpp	2009-02-26 17:52:03.000000000 -0500
@@ -6,7 +6,7 @@
 
 // thanks to btdsys, 7900, and usr for answering questions, but especially btdsys
 // for making peerlib, without which this wouldn't be possible.
-
+#include <cstdio>
 #define MACHINE_NAME "Peer Note-Pool"
 #define AUTHOR "jmmcd"
 #define FULL_NAME AUTHOR " " MACHINE_NAME
diff -rupN armstrong.old/src/plugins/ladspadapter/ladspadapter.cpp armstrong/src/plugins/ladspadapter/ladspadapter.cpp
--- armstrong.old/src/plugins/ladspadapter/ladspadapter.cpp	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/ladspadapter/ladspadapter.cpp	2009-02-26 17:55:34.000000000 -0500
@@ -17,7 +17,7 @@
 
 
 // ladspadapter allows running LADSPA plugins as zzub plugins
-
+#include <cstdio>
 #include <ladspa.h>
 #include <zzub/signature.h>
 #include 	"zzub/plugin.h"
diff -rupN armstrong.old/src/plugins/ladspadapter/paramtools.h armstrong/src/plugins/ladspadapter/paramtools.h
--- armstrong.old/src/plugins/ladspadapter/paramtools.h	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/ladspadapter/paramtools.h	2009-02-26 17:54:52.000000000 -0500
@@ -17,7 +17,7 @@
 
 #if !defined(__LADSPA_PARAMTOOLS_H__)
 #define __LADSPA_PARAMTOOLS_H__
-
+#include <cstdio>
 #include <math.h>
 #include <algorithm>
 
diff -rupN armstrong.old/src/plugins/lunar/pugxml.h armstrong/src/plugins/lunar/pugxml.h
--- armstrong.old/src/plugins/lunar/pugxml.h	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/lunar/pugxml.h	2009-02-26 18:00:33.000000000 -0500
@@ -24,7 +24,7 @@
 #	endif
 #endif
 
-
+#include <cstdio>
 #define TCHAR char
 #define _T(x) x
 
diff -rupN armstrong.old/src/plugins/zzub_miditracker/miditracker.cpp armstrong/src/plugins/zzub_miditracker/miditracker.cpp
--- armstrong.old/src/plugins/zzub_miditracker/miditracker.cpp	2009-01-09 16:17:45.000000000 -0500
+++ armstrong/src/plugins/zzub_miditracker/miditracker.cpp	2009-02-26 18:03:39.000000000 -0500
@@ -7,7 +7,7 @@
 #include <zzub/signature.h>
 #include <zzub/plugin.h>
 #include "miditracker.h"
-
+#include <cstdio>
 using namespace std;
 
 namespace miditracker {
