diff --git a/armstrong/src/libzzub/libzzub.cpp b/armstrong/src/libzzub/libzzub.cpp
index 643b02c..fa7a1b3 100644
--- a/src/libzzub/libzzub.cpp
+++ b/src/libzzub/libzzub.cpp
@@ -1447,7 +1447,7 @@ int zzub_plugin_describe_value(zzub_plugin_t *plugin, int group, int column, int
 	int index = -1;
 	zzub_plugin_pattern_to_linear_no_connections(plugin, group, 0, column, &index);
 
-	const parameter* para = plugin->_player->back.plugin_get_parameter_info(plugin->id, group, 0, column);
+	const zzub::parameter* para = plugin->_player->back.plugin_get_parameter_info(plugin->id, group, 0, column);
 	if (index != -1) {
 		if (value != getNoValue(para)) {	// infector crashen when trying to describe novalues (and out-of-range-values)
 			
