--- src/kernels/Makefile	2011-01-28 19:44:57.000000000 +0100
+++ src/kernels/Makefile	2011-03-08 10:48:44.783442947 +0100
@@ -4,13 +4,13 @@ all: install
 
 
 install:
-	mkdir -p /usr/share/hashkill/kernels
-	chmod ugo+r /usr/share/hashkill/kernels
-	cp -f *.cl /usr/share/hashkill/kernels
+	mkdir -p ../../../../pkg/usr/share/hashkill/kernels
+	chmod ugo+r ../../../../pkg/usr/share/hashkill/kernels
+	cp -f *.cl ../../../../pkg/usr/share/hashkill/kernels
 
 clean:
 
 uninstall:
-	rm -rf /usr/share/hashkill/markov 2>/dev/null
+	rm -rf ../../../../pkg/usr/share/hashkill/markov 2>/dev/null
 
 distclean:
--- src/markov/Makefile	2010-06-19 14:50:20.000000000 +0200
+++ src/markov/Makefile	2011-03-08 10:45:55.644557427 +0100
@@ -9,13 +9,13 @@ all:
 
 
 install:
-	mkdir -p /usr/share/hashkill/markov
-	chmod ugo+r /usr/share/hashkill/markov
-	cp -f *.stat /usr/share/hashkill/markov
+	mkdir -p ../../../../pkg/usr/share/hashkill/markov
+	chmod ugo+r ../../../../pkg/usr/share/hashkill/markov
+	cp -f *.stat ../../../../pkg/usr/share/hashkill/markov
 
 clean:
 
 uninstall:
-	rm -rf /usr/share/hashkill/markov 2>/dev/null
+	rm -rf ../../../../pkg/usr/share/hashkill/markov 2>/dev/null
 
 distclean:
--- src/plugins/Makefile	2011-02-27 00:20:21.000000000 +0100
+++ src/plugins/Makefile	2011-03-08 10:45:51.950603664 +0100
@@ -128,15 +128,15 @@ plugin_zip:
 
 
 install:
-	mkdir -p /usr/share/hashkill/plugins
-	mkdir -p /usr/share/hashkill/sessions
-	chmod ugo+w /usr/share/hashkill/sessions
-	cp -f *.so /usr/share/hashkill/plugins 
+	mkdir -p ../../../../pkg/usr/share/hashkill/plugins
+	mkdir -p ../../../../pkg/usr/share/hashkill/sessions
+	chmod ugo+w ../../../../pkg/usr/share/hashkill/sessions
+	cp -f *.so ../../../../pkg/usr/share/hashkill/plugins 
 
 clean:
 	rm *.so
 
 uninstall:
-	rm -rf /usr/share/hashkill
+	rm -rf ../../../../pkg/usr/share/hashkill
 
 distclean:
