--- src/lib/utils.py	2011-08-22 11:54:41.152111468 +0200
+++ src/lib/utils.py.old	2011-08-22 11:53:37.663720706 +0200
@@ -77,7 +77,7 @@
 else:
 	# we run as normal user, install into $HOME
 	USER = 1
+	DIR_USER = os.path.join(xdg_data_home,'screenlets')
-	DIR_USER = os.path.join(os.environ['HOME'],'.screenlets')
 	DIR_AUTOSTART = get_autostart_dir()
 
 # ------------------------------------------------------------------------------
