diff src/main.cpp src/main.cpp
index 4dba74b..2f657d8 100644
--- src/main.cpp
+++ src/main.cpp
@@ -238,8 +238,8 @@ struct structures::ret analysefile(const std::string &dateiname,
 
 
     // generate current date in the form YYYY-MM-DD
-    std::time_t today_time = std::time(NULL);       // today as calendar time
-    struct std::tm *today = std::gmtime(&today_time); // today in GMT
+    time_t today_time = time(NULL);       // today as calendar time
+    struct tm *today = gmtime(&today_time); // today in GMT
     // year
     // CONTINUE
     // (1900+today.tm_year)
