diff -rupN kbib-0.6.6//kbib/kbibgroupviewgroup.cpp kbib-0.6.6-prl//kbib/kbibgroupviewgroup.cpp
--- kbib-0.6.6//kbib/kbibgroupviewgroup.cpp	2009-11-03 07:30:54.000000000 +0000
+++ kbib-0.6.6-prl//kbib/kbibgroupviewgroup.cpp	2010-09-04 11:27:22.882331620 +0100
@@ -67,7 +67,7 @@ void KBibGroupViewGroup::addEntry(BibEnt
     if (groupBy == QString(i18n("Reference type")) )
     {
         st = entry->getEntryType();
-        valueList = QStringList::QStringList(st);
+        valueList = QStringList(st);
     }
     else
     {
@@ -78,7 +78,7 @@ void KBibGroupViewGroup::addEntry(BibEnt
 			valueList = QStringList::split(field->connectingString, st);	
 		}
         else
-            valueList =  QStringList::QStringList(st);
+            valueList =  QStringList(st);
 
 
     }
