--- blatSrc/inc/common.mk	2010-01-05 15:31:32.916902685 -0700
+++ blatSrc/inc/common.mk2	2010-01-05 15:30:06.476911176 -0700
@@ -9,12 +9,12 @@
 
 # Stronger warning checks, and warnings-->errors, for libraries and CGIs:
 ifeq (darwin,$(findstring darwin,${OSTYPE}))
+    HG_WARN_ERR = -DJK_WARN -Wall -Wno-unused-variable
-    HG_WARN_ERR = -DJK_WARN -Wall -Werror -Wno-unused-variable
 else
   ifeq (solaris,$(findstring solaris,${OSTYPE}))
     HG_WARN_ERR = -DJK_WARN -Wall
   else
+    HG_WARN_ERR = -DJK_WARN -Wall
-    HG_WARN_ERR = -DJK_WARN -Wall -Werror
   endif
 endif
 # Apply the stronger checks to all code on our development machine:
@@ -33,7 +33,7 @@
     CGI_BIN=/usr/local/apache/cgi-bin
 endif
 ifeq (${BINDIR},)
+    BINDIR = ${INSTALL_ROOT}/usr/bin
-    BINDIR = ${HOME}/bin/${MACHTYPE}
 endif
 MKDIR=mkdir -p
 ifeq (${STRIP},)
