diff -Naur therion/extern/lxMath.h therion_aur/extern/lxMath.h
--- therion/extern/lxMath.h	2006-12-12 04:35:40.000000000 -0800
+++ therion_aur/extern/lxMath.h	2011-08-26 10:44:45.158507955 -0700
@@ -34,6 +34,7 @@
 #include <list>
 #include <vector>
 #include <map>
+#include <cstring>
 #endif  
 //LXDEPCHECK - standart libraries
 
diff -Naur therion/loch/lxLRUD.h therion_aur/loch/lxLRUD.h
--- therion/loch/lxLRUD.h	2006-12-12 04:35:40.000000000 -0800
+++ therion_aur/loch/lxLRUD.h	2011-08-26 11:02:14.393779292 -0700
@@ -3,6 +3,7 @@
 
 // Standard libraries
 #ifndef LXDEPCHECK
+#include <cstring>
 #endif  
 //LXDEPCHECK - standart libraries
 
diff -Naur therion/loch/lxMath.h therion_aur/loch/lxMath.h
--- therion/loch/lxMath.h	2006-12-12 04:35:40.000000000 -0800
+++ therion_aur/loch/lxMath.h	2011-08-26 11:03:07.036862925 -0700
@@ -34,6 +34,7 @@
 #include <list>
 #include <vector>
 #include <map>
+#include <cstring>
 #endif  
 //LXDEPCHECK - standart libraries
 
diff -Naur therion/loch/Makefile therion_aur/loch/Makefile
--- therion/loch/Makefile	2010-02-17 02:18:25.000000000 -0800
+++ therion_aur/loch/Makefile	2011-08-26 11:07:21.202895903 -0700
@@ -6,7 +6,8 @@
   lxGUI.o lxGLC.o lxOptDlg.o lxAboutDlg.o \
   img.o
 
-VTKVERSION = 5.4
+#VTKVERSION = 5.4
+#VTKVERSION = 5.6
 VTKLIBS = -lvtkHybrid -lvtkImaging -lvtkIO -lvtkGraphics -lvtkFiltering -lvtkCommon -lvtkjpeg -lvtkpng -lvtkzlib -lvtksys -lfreetype
 POSTMAKE =
 STRIPFLAG = -s 
diff -Naur therion/thbezier.h therion_aur/thbezier.h
--- therion/thbezier.h	2007-05-26 18:27:54.000000000 -0700
+++ therion_aur/thbezier.h	2011-08-26 10:42:15.352031702 -0700
@@ -31,6 +31,7 @@
 
 #include <list>
 #include <vector>
+#include <cstring>
 
 
 struct thbezier_point {
diff -Naur therion/thconfig.cxx therion_aur/thconfig.cxx
--- therion/thconfig.cxx	2009-06-04 06:25:36.000000000 -0700
+++ therion_aur/thconfig.cxx	2011-08-26 10:41:21.002395223 -0700
@@ -148,7 +148,7 @@
   const char * winini = "C:/WINDOWS;C:/WINNT;C:/Program files/Therion";
   const char * wincfg = "C:/Program files/Therion";
 #else
-  const char * unixini = "/etc:/usr/etc:/usr/local/etc";
+  const char * unixini = "/etc/therion:/etc:/usr/etc:/usr/local/etc";
   const char * unixcfg = "/usr/share/therion:/usr/local/share/therion";
 #endif
   
diff -Naur therion/thdb2dcp.cxx therion_aur/thdb2dcp.cxx
--- therion/thdb2dcp.cxx	2007-05-25 14:51:30.000000000 -0700
+++ therion_aur/thdb2dcp.cxx	2011-08-26 10:42:58.652419666 -0700
@@ -26,6 +26,7 @@
  */
  
 #include "thdb2dcp.h"
+#include <cstring>
 
 thdb2dcp::thdb2dcp() {
 
diff -Naur therion/thscrapen.cxx therion_aur/thscrapen.cxx
--- therion/thscrapen.cxx	2007-05-25 14:51:30.000000000 -0700
+++ therion_aur/thscrapen.cxx	2011-08-26 10:43:36.626503207 -0700
@@ -26,6 +26,7 @@
  */
  
 #include "thscrapen.h"
+#include <cstring>
 
 thscrapen::thscrapen() {
 
diff -Naur therion/thscraplo.cxx therion_aur/thscraplo.cxx
--- therion/thscraplo.cxx	2007-05-25 14:51:30.000000000 -0700
+++ therion_aur/thscraplo.cxx	2011-08-26 10:43:21.184220316 -0700
@@ -26,6 +26,7 @@
  */
  
 #include "thscraplo.h"
+#include <cstring>
 
 thscraplo::thscraplo() {
 
diff -Naur therion/thwarp.h therion_aur/thwarp.h
--- therion/thwarp.h	2007-05-25 14:51:30.000000000 -0700
+++ therion_aur/thwarp.h	2011-08-26 10:45:20.526101989 -0700
@@ -31,6 +31,7 @@
 
 #include "thpic.h"
 #include "thsketch.h"
+#include <cstring>
 
 
 class thwarp {
