diff -Naur therion/loch/Makefile therion_aur/loch/Makefile
--- therion/loch/Makefile	2012-08-12 13:20:24.000000000 -0700
+++ therion_aur/loch/Makefile	2012-08-13 22:37:14.484887763 -0700
@@ -36,7 +36,7 @@
 CXXPFLAGS = -DLXLINUX $(shell wx-config --cxxflags) -Wno-deprecated $(shell freetype-config --cflags) -I$(VTKPATH)
 CCPFLAGS = -DLXLINUX  $(shell wx-config --cflags)
 LXLIBDIR = linux
-PLIBS = $(shell wx-config --libs --gl-libs) -L$(VTKLIBPATH) $(VTKLIBS) -lGLU -lpthread -lz 
+PLIBS = $(shell wx-config --libs --gl-libs) -L$(VTKLIBPATH) $(VTKLIBS) -lGL -lGLU -lpthread -lz -lpng -ljpeg -lX11
 LXPLATFORM = LINUX
 
 # PLATFORM WIN32
diff -Naur therion/thconfig.cxx therion_aur/thconfig.cxx
--- therion/thconfig.cxx	2011-04-08 06:54:38.000000000 -0700
+++ therion_aur/thconfig.cxx	2012-08-13 22:37:31.457079226 -0700
@@ -150,7 +150,7 @@
   const char * winini = "C:/WINDOWS;C:/WINNT;C:/Program files/Therion";
   const char * wincfg = "C:/Program files/Therion";
 #else
-  const char * unixini = "/etc:/usr/etc:/usr/local/etc";
+  const char * unixini = "/etc/therion:/usr/etc:/usr/local/etc";
   const char * unixcfg = "/usr/share/therion:/usr/local/share/therion";
 #endif
   
diff -Naur therion/thtmpdir.cxx therion_aur/thtmpdir.cxx
--- therion/thtmpdir.cxx	2007-05-25 15:51:30.000000000 -0700
+++ therion_aur/thtmpdir.cxx	2012-10-12 13:50:16.434844811 -0700
@@ -203,12 +203,14 @@
       DIR *tmpdir = opendir(this->name);
       struct dirent *tmpf;
       thbuffer tmpfname;
+//thwarning(("tmpdir is: %s", this->name.get_buffer()))
       if (tmpdir != NULL) {
         tmpf = readdir(tmpdir);
         while (tmpf != NULL) {
           tmpfname = this->name;
           tmpfname += THPATHSEPARATOR;
           tmpfname += tmpf->d_name;
+//thwarning(("unlinking: %s", tmpfname.get_buffer()))
           unlink(tmpfname);
           tmpf = readdir(tmpdir);
         }
@@ -216,9 +218,11 @@
       }
   
       // remove directory
-      if (rmdir(this->name) != 0)
+//thwarning(("rmdir: %s", this->name.get_buffer()))
+      if (rmdir(this->name) != 0) {
+		//thwarning(("rmdir: %s", this->name.get_buffer()))
         thwarning(("error deleting temporary directory -- %s",this->name.get_buffer()))
-      else {
+      } else {
         this->name = ".";
         this->tried = false;
         this->exist = false;
