diff -Naur therion/loch/Makefile therion_aur/loch/Makefile
--- therion/loch/Makefile	2010-06-21 06:42:07.000000000 -0700
+++ therion_aur/loch/Makefile	2010-06-21 07:19:52.000000000 -0700
@@ -6,7 +6,7 @@
   lxGUI.o lxGLC.o lxOptDlg.o lxAboutDlg.o \
   img.o
 
-VTKVERSION = 5.4
+#VTKVERSION = 5.4
 VTKLIBS = -lvtkHybrid -lvtkImaging -lvtkIO -lvtkGraphics -lvtkFiltering -lvtkCommon -lvtkjpeg -lvtkpng -lvtkzlib -lvtksys -lfreetype
 POSTMAKE =
 STRIPFLAG = -s 
diff -Naur therion/thconfig.cxx therion_aur/thconfig.cxx
--- therion/thconfig.cxx	2009-06-04 06:25:36.000000000 -0700
+++ therion_aur/thconfig.cxx	2010-05-18 12:35:28.000000000 -0700
@@ -148,7 +148,7 @@
   const char * winini = "C:/WINDOWS;C:/WINNT;C:/Program files/Therion";
   const char * wincfg = "C:/Program files/Therion";
 #else
-  const char * unixini = "/etc:/usr/etc:/usr/local/etc";
+  const char * unixini = "/etc/therion:/etc:/usr/etc:/usr/local/etc";
   const char * unixcfg = "/usr/share/therion:/usr/local/share/therion";
 #endif
   
