Index: nxcl/nxcl.cpp
===================================================================
--- nxcl/nxcl.cpp	(revision 507)
+++ nxcl/nxcl.cpp	(working copy)
@@ -27,6 +27,7 @@
  * found, for example, in Ubuntu 6.06. */
 #define DBUS_API_SUBJECT_TO_CHANGE 1
 extern "C" {
+#include <stdlib.h>
 #include <dbus/dbus.h>
 #include <X11/Xlib.h>
 }
Index: lib/notQt.cpp
===================================================================
--- lib/notQt.cpp	(revision 507)
+++ lib/notQt.cpp	(working copy)
@@ -21,6 +21,7 @@
 #include <sstream>
 extern "C" {
 #include <unistd.h>
+#include <stdlib.h>
 #include <errno.h>
 #include <sys/types.h>
 #include <sys/wait.h>
Index: lib/nxclientlib.cpp
===================================================================
--- lib/nxclientlib.cpp	(revision 507)
+++ lib/nxclientlib.cpp	(working copy)
@@ -29,6 +29,8 @@
 #include <fstream>
 
 extern "C" {
+    #include <stdlib.h>
+    #include <string.h>
     #include <errno.h>
     #include <sys/types.h>
     #include <sys/stat.h>
