From 3e5077983e3cdfa09e741978f8c4fb7150c65011 Mon Sep 17 00:00:00 2001
From: Vladimir Chizhov <master@mirlord.com>
Date: Sun, 20 May 2012 20:38:40 +0400
Subject: [PATCH] Use '-a' (short and more compatible form of '--multiple')
 option for basename.

---
 bin/mad |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/mad b/bin/mad
index 2d8147b..b381fcb 100755
--- a/bin/mad
+++ b/bin/mad
@@ -21,7 +21,7 @@ list_pages() {
     test ! -z $path \
       && test -d $path \
       && find $path -type f -print0 \
-      | xargs -0 basename \
+      | xargs -0 basename -a \
       | grep -iv 'readme*' \
       | grep '.md$' \
       | perl -pe 's|^(.*)\.md$|    \1|;'
-- 
1.7.10.2

