diff -ru winwrangler-0.1-old/src/ww-tray.c winwrangler-0.1-dev/src/ww-tray.c
--- winwrangler-0.1-old/src/ww-tray.c	2008-10-29 23:41:22.000000000 +0100
+++ winwrangler-0.1-dev/src/ww-tray.c	2008-11-01 15:11:14.000000000 +0100
@@ -166,7 +166,7 @@
 	}
 	
 	layouts = ww_get_layouts ();
-	tray_icon = gtk_status_icon_new_from_icon_name ("winwrangler");
+	tray_icon = gtk_status_icon_new_from_file ("/usr/share/icons/hicolor/scalable/apps/winwrangler.svg");
 	
 	gtk_status_icon_set_visible (tray_icon, TRUE);
 	gtk_status_icon_set_tooltip (tray_icon, "Perform advanced window layout");

