diff -cBr simpleburn-1.6.3/scripts/simpleburn-udev-media-detection.sh simpleburn-1.6.3_patched/scripts/simpleburn-udev-media-detection.sh
*** simpleburn-1.6.3/scripts/simpleburn-udev-media-detection.sh	2011-06-19 00:58:31.000000000 +0200
--- simpleburn-1.6.3_patched/scripts/simpleburn-udev-media-detection.sh	2012-05-08 12:49:17.586869312 +0200
***************
*** 11,21 ****
  source simpleburn-burning-suite quiet
  
  
! if /lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA"
  then 
! 	if /lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_DVD"
  	then mediatype="dvd"
! 	else if /lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_CD"
  		then mediatype="cd"
  		else exit 1
  		fi
--- 11,21 ----
  source simpleburn-burning-suite quiet
  
  
! if /usr/lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA"
  then 
! 	if /usr/lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_DVD"
  	then mediatype="dvd"
! 	else if /usr/lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_CD"
  		then mediatype="cd"
  		else exit 1
  		fi
***************
*** 35,70 ****
  	
  	if [ "$mediatype" == "cd" ]
  	then
! 		if /lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_CD_RW"
  		then rewritablemedia=1
  		else rewritablemedia=0
  		fi
  		
! 		if /lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_TRACK_COUNT_AUDIO"
  		then mediacontent="audio" #overrides "data" if it is an hybrid CD
  			mediasize=`$CDRECORD -toc dev=$device 2>&1 | grep "track:lout" | sed 's/track:lout lba: \+\([0-9]\+\) .*/\1/'`
  			let mediasize=mediasize*2
  		fi
  		
! 		if /lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_CD_R"
  		then mediacapacity=`$CDRECORD -atip dev=$device 2>&1 | grep "ATIP start of lead out:" | sed 's/.*: \([0-9]\+\) .*/\1/'` 
  			let mediacapacity=mediacapacity*2
  		else mediacapacity=0
  		fi
  	else #DVD
! 		if /lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_DVD.*RW"
  		then rewritablemedia=1
  		else rewritablemedia=0
  		fi
  		
! 		if /lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_DVD.*R"
  		then mediacapacity=`$CDRECORD -atip dev=$device 2>&1 | grep "rzone size:" | tail -1 | sed 's/rzone size: \+//'` #doesn't work with wodim or cdrskin => mediacapacity = 0 (unknown)
  			let mediacapacity=mediacapacity*2
  		else mediacapacity=0
  		fi
  	fi
  	
! 	trackscount=`/lib/udev/cdrom_id $device | grep "ID_CDROM_MEDIA_TRACK_COUNT=" | cut -f2 -d=`
  	
  	if [ "$2" == "oneline" ]
  	then echo "$mediatype:$mediacontent:$mediacapacity:$mediasize:$rewritablemedia:$trackscount"
--- 35,70 ----
  	
  	if [ "$mediatype" == "cd" ]
  	then
! 		if /usr/lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_CD_RW"
  		then rewritablemedia=1
  		else rewritablemedia=0
  		fi
  		
! 		if /usr/lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_TRACK_COUNT_AUDIO"
  		then mediacontent="audio" #overrides "data" if it is an hybrid CD
  			mediasize=`$CDRECORD -toc dev=$device 2>&1 | grep "track:lout" | sed 's/track:lout lba: \+\([0-9]\+\) .*/\1/'`
  			let mediasize=mediasize*2
  		fi
  		
! 		if /usr/lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_CD_R"
  		then mediacapacity=`$CDRECORD -atip dev=$device 2>&1 | grep "ATIP start of lead out:" | sed 's/.*: \([0-9]\+\) .*/\1/'` 
  			let mediacapacity=mediacapacity*2
  		else mediacapacity=0
  		fi
  	else #DVD
! 		if /usr/lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_DVD.*RW"
  		then rewritablemedia=1
  		else rewritablemedia=0
  		fi
  		
! 		if /usr/lib/udev/cdrom_id $device | grep -q "ID_CDROM_MEDIA_DVD.*R"
  		then mediacapacity=`$CDRECORD -atip dev=$device 2>&1 | grep "rzone size:" | tail -1 | sed 's/rzone size: \+//'` #doesn't work with wodim or cdrskin => mediacapacity = 0 (unknown)
  			let mediacapacity=mediacapacity*2
  		else mediacapacity=0
  		fi
  	fi
  	
! 	trackscount=`/usr/lib/udev/cdrom_id $device | grep "ID_CDROM_MEDIA_TRACK_COUNT=" | cut -f2 -d=`
  	
  	if [ "$2" == "oneline" ]
  	then echo "$mediatype:$mediacontent:$mediacapacity:$mediasize:$rewritablemedia:$trackscount"
diff -cBr simpleburn-1.6.3/src/common-detection.c simpleburn-1.6.3_patched/src/common-detection.c
*** simpleburn-1.6.3/src/common-detection.c	2011-04-14 08:43:53.000000000 +0200
--- simpleburn-1.6.3_patched/src/common-detection.c	2012-05-08 12:50:16.723534051 +0200
***************
*** 153,159 ****
  				g_print ("DEBUG: udev/listcddevices: n°%d: '%s'\n", i, cddevices.gl_pathv[i]);
  			#endif
  			devicename = g_path_get_basename( cddevices.gl_pathv[i] );
! 			command = g_strdup_printf( "/lib/udev/cdrom_id /dev/%s", devicename );
  			g_spawn_command_line_sync( command, &output, NULL, &exitvalue, NULL );
  			if ( exitvalue == 0 ) { //if it is a CD / DVD device
  				#ifdef DEBUG
--- 153,159 ----
  				g_print ("DEBUG: udev/listcddevices: n°%d: '%s'\n", i, cddevices.gl_pathv[i]);
  			#endif
  			devicename = g_path_get_basename( cddevices.gl_pathv[i] );
! 			command = g_strdup_printf( "/usr/lib/udev/cdrom_id /dev/%s", devicename );
  			g_spawn_command_line_sync( command, &output, NULL, &exitvalue, NULL );
  			if ( exitvalue == 0 ) { //if it is a CD / DVD device
  				#ifdef DEBUG
