--- loconvert.orig	2011-04-16 16:48:02.853333458 +0300
+++ loconvert	2011-04-16 16:47:52.560000127 +0300
@@ -116,7 +116,8 @@
 # run the tests
 echo "Doing the conversion..."
 export PYTHONPATH="$ooo_home/basis-link/program"
-python $document_converter_py "$OOCONV_INPUT" "$OOCONV_OUTPUT"
+export URE_MORE_TYPES=$ooo_home/basis-link/program/offapi.rdb
+python2 $document_converter_py "$OOCONV_INPUT" "$OOCONV_OUTPUT"
 
 # kill the OOo
 killall -9 soffice.bin
