diff --git PDBReader.java PDBReader.java
index 71cdac7..178cebe 100644
--- PDBReader.java
+++ PDBReader.java
@@ -673,7 +673,12 @@ public class PDBReader extends DefaultChemObjectReader {
 		if (lineLength >= 79) {
             String frag = cLine.substring(78, 80).trim();
             if (frag.length() > 0) {
-                oAtom.setCharge(Double.parseDouble(frag));
+                // see Format_v33_A4.pdf, p. 178
+                try {
+                    oAtom.setCharge(Double.parseDouble(frag));
+                } catch (NumberFormatException e) {
+                    oAtom.setCharge(Double.parseDouble(new StringBuilder(frag).reverse().toString()));
+                }
             }
 		}
 		
