diff -rupN -x entries recastnavigation-read-only//CMakeLists.txt recastnavigation-read-only-new//CMakeLists.txt
--- recastnavigation-read-only//CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ recastnavigation-read-only-new//CMakeLists.txt	2010-03-10 10:08:44.366086396 +0100
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 2.6 FATAL_ERROR)
+
+project (RecastNavigation)
+
+SET (LIBRARY_OUTPUT_PATH
+  ${PROJECT_BINARY_DIR}/bin
+  CACHE PATH
+  "Single Directory for all Libraries"
+)
+
+SET (EXECUTABLE_OUTPUT_PATH
+  ${PROJECT_BINARY_DIR}/bin
+  CACHE PATH
+  "Single Directory for all Executables."
+)
+
+add_subdirectory (Detour)
+add_subdirectory (Recast)
+add_subdirectory (DebugUtils)
+add_subdirectory (RecastDemo)
diff -rupN -x entries recastnavigation-read-only//DebugUtils/CMakeLists.txt recastnavigation-read-only-new//DebugUtils/CMakeLists.txt
--- recastnavigation-read-only//DebugUtils/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ recastnavigation-read-only-new//DebugUtils/CMakeLists.txt	2010-03-08 13:54:14.385138498 +0100
@@ -0,0 +1,21 @@
+include_directories(
+  ${RecastNavigation_SOURCE_DIR}/DebugUtils/Include
+  ${RecastNavigation_SOURCE_DIR}/Detour/Include
+  ${RecastNavigation_SOURCE_DIR}/Recast/Include
+)
+
+link_directories (
+  ${RecastNavigation_BINARY_DIR}/Detour
+  ${RecastNavigation_BINARY_DIR}/Recast
+)
+
+add_library (
+  DebugUtils SHARED
+
+  Source/DebugDraw.cpp
+  Source/DetourDebugDraw.cpp
+  Source/RecastDebugDraw.cpp
+  Source/RecastDump.cpp
+)
+
+TARGET_LINK_LIBRARIES( DebugUtils Detour Recast )
diff -rupN -x entries recastnavigation-read-only//Detour/CMakeLists.txt recastnavigation-read-only-new//Detour/CMakeLists.txt
--- recastnavigation-read-only//Detour/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ recastnavigation-read-only-new//Detour/CMakeLists.txt	2010-03-08 13:48:04.882066694 +0100
@@ -0,0 +1,13 @@
+include_directories( ${RecastNavigation_SOURCE_DIR}/Detour/Include )
+link_directories ( ${RecastNavigation_BINARY_DIR}/Recast ) 
+
+add_library (
+  Detour SHARED
+
+  Source/DetourCommon.cpp
+  Source/DetourNavMesh.cpp
+  Source/DetourNavMeshBuilder.cpp
+  Source/DetourNode.cpp
+)
+
+TARGET_LINK_LIBRARIES( Detour Recast )
diff -rupN -x entries recastnavigation-read-only//Recast/CMakeLists.txt recastnavigation-read-only-new//Recast/CMakeLists.txt
--- recastnavigation-read-only//Recast/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ recastnavigation-read-only-new//Recast/CMakeLists.txt	2010-03-08 13:49:49.398704162 +0100
@@ -0,0 +1,19 @@
+include_directories( ${RecastNavigation_SOURCE_DIR}/Recast/Include )
+#link_directories ( ${RecastNavigation_BINARY_DIR}/Recast ) 
+
+add_library (
+  Recast SHARED
+
+  Source/Recast.cpp
+  Source/RecastArea.cpp
+  Source/RecastContour.cpp
+  Source/RecastFilter.cpp
+  Source/RecastLog.cpp
+  Source/RecastMesh.cpp
+  Source/RecastMeshDetail.cpp
+  Source/RecastRasterization.cpp
+  Source/RecastRegion.cpp
+  Source/RecastTimer.cpp
+)
+
+#TARGET_LINK_LIBRARIES( Detour Recast )
diff -rupN -x entries recastnavigation-read-only//RecastDemo/CMakeLists.txt recastnavigation-read-only-new//RecastDemo/CMakeLists.txt
--- recastnavigation-read-only//RecastDemo/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ recastnavigation-read-only-new//RecastDemo/CMakeLists.txt	2010-03-10 10:08:31.589433396 +0100
@@ -0,0 +1,58 @@
+
+FIND_PACKAGE(OpenGL REQUIRED)
+FIND_PACKAGE(SDL REQUIRED)
+
+IF (OPENGL_FOUND)
+  MESSAGE("OPENGL FOUND [${OPENGL_LIBRARIES}]")
+ENDIF (OPENGL_FOUND)
+
+IF (SDL_FOUND)
+  MESSAGE("SDL FOUND [${SDL_LIBRARY}]")
+ENDIF (SDL_FOUND)
+
+include_directories(
+  ${SDL_INCLUDE_DIR}
+  ${RecastNavigation_SOURCE_DIR}/RecastDemo/Include
+  ${RecastNavigation_SOURCE_DIR}/RecastDemo/Contrib
+  ${RecastNavigation_SOURCE_DIR}/DebugUtils/Include
+  ${RecastNavigation_SOURCE_DIR}/Detour/Include
+  ${RecastNavigation_SOURCE_DIR}/Recast/Include
+)
+
+link_directories (
+  ${RecastNavigation_BINARY_DIR}/Detour
+  ${RecastNavigation_BINARY_DIR}/Recast
+  ${RecastNavigation_BINARY_DIR}/DebugUtils
+)
+
+add_executable (
+  RecastDemo
+
+  Source/ChunkyTriMesh.cpp
+  Source/ConvexVolumeTool.cpp
+  Source/imgui.cpp
+  Source/imguiRenderGL.cpp
+  Source/InputGeom.cpp
+  Source/main.cpp
+  Source/MeshLoaderObj.cpp
+  Source/NavMeshTesterTool.cpp
+  Source/OffMeshConnectionTool.cpp
+  Source/Sample.cpp
+  Source/Sample_Debug.cpp
+  Source/Sample_SoloMeshSimple.cpp
+  Source/Sample_SoloMeshTiled.cpp
+  Source/Sample_TileMesh.cpp
+  Source/TestCase.cpp
+  #Source/SDLMain.m
+)
+
+TARGET_LINK_LIBRARIES( RecastDemo DebugUtils Detour Recast ${SDL_LIBRARY} ${OPENGL_LIBRARIES} )
+
+# create_symlink
+execute_process(COMMAND ${CMAKE_COMMAND} -E copy_directory ${RecastNavigation_SOURCE_DIR}/RecastDemo/Bin/Meshes ${EXECUTABLE_OUTPUT_PATH}/Meshes)
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E copy_directory ${RecastNavigation_SOURCE_DIR}/RecastDemo/Bin/Tests  ${EXECUTABLE_OUTPUT_PATH}/Tests)
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E copy ${RecastNavigation_SOURCE_DIR}/RecastDemo/Bin/DroidSans.ttf ${EXECUTABLE_OUTPUT_PATH})
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E copy ${RecastNavigation_SOURCE_DIR}/RecastDemo/Bin/test.chf ${EXECUTABLE_OUTPUT_PATH})
