Index: libmusclecard-1.3.6/src/musclecard.c
===================================================================
--- libmusclecard-1.3.6.orig/src/musclecard.c
+++ libmusclecard-1.3.6/src/musclecard.c
@@ -89,7 +89,7 @@ MSC_RV MSCListTokens(MSCULong32 listScop
 	MSCPULong32 arrayLength)
 {
 	MSCLong32 rv;
-	SCARD_READERSTATE_A rgReaderStates;
+	SCARD_READERSTATE rgReaderStates;
 	MSCTokenInfo tokenInfo;
 	MSCLPTokenInfo currentToken;
 	MSCULong32 tokensFound;
@@ -611,14 +611,14 @@ MSC_RV MSCWaitForTokenEvent(MSCLPTokenIn
 	MSCULong32 arraySize, MSCULong32 timeoutValue)
 {
 	MSCLong32 rv, rt;
-	LPSCARD_READERSTATE_A rgReaderStates;
+	LPSCARD_READERSTATE rgReaderStates;
 	MSCTokenInfo tokenInfo;
 	int i;
 
 	rgReaderStates = NULL;
 
 	/*
-	 * Allocate array of SCARD_READERSTATE_A structures, set UNAWARE on
+	 * Allocate array of SCARD_READERSTATE structures, set UNAWARE on
 	 * all of the structures to get the current status and then send them
 	 * to GetStatusChange for blocking event
 	 */
@@ -646,8 +646,8 @@ MSC_RV MSCWaitForTokenEvent(MSCLPTokenIn
 	}
 	mscUnLockThread();
 
-	rgReaderStates = (LPSCARD_READERSTATE_A)
-		malloc(sizeof(SCARD_READERSTATE_A) * arraySize);
+	rgReaderStates = (LPSCARD_READERSTATE)
+		malloc(sizeof(SCARD_READERSTATE) * arraySize);
 
 	if (rgReaderStates == NULL)
 		return MSC_INTERNAL_ERROR;
@@ -1853,8 +1853,6 @@ MSC_RV pcscToMSC(MSCLong32 pcscCode)
 			return MSC_TOKEN_REMOVED;
 		case SCARD_W_RESET_CARD:
 			return MSC_TOKEN_RESET;
-		case SCARD_W_INSERTED_CARD:
-			return MSC_TOKEN_INSERTED;
 		case SCARD_E_NO_SERVICE:
 			return MSC_SERVICE_UNRESPONSIVE;
 		case SCARD_E_UNKNOWN_CARD:
Index: libmusclecard-1.3.6/utils/bundleTool/bundleTool.c
===================================================================
--- libmusclecard-1.3.6.orig/utils/bundleTool/bundleTool.c
+++ libmusclecard-1.3.6/utils/bundleTool/bundleTool.c
@@ -55,7 +55,7 @@ int main(int argc, char **argv)
 {
 	LONG rv;
 	SCARDCONTEXT hContext;
-	SCARD_READERSTATE_A rgReaderStates;
+	SCARD_READERSTATE rgReaderStates;
 	DWORD readerListSize;
 	struct stat statBuffer;
 	char spAtrValue[100];
