--- batterymon.py	2009-05-01 07:14:37.000000000 -0400
+++ batterymon.py	2011-10-19 17:51:53.000000000 -0400
@@ -128,15 +128,19 @@
         charge_level = self.__battery.GetProperty('battery.charge_level.percentage')
         logger.debug("battery.percentage: %s" % str(charge_level))
 
-        try:
-            remaining_time = self.__battery.GetProperty('battery.remaining_time')
-            logger.debug("battery.remaining_time: %s" % str(remaining_time))
 
-        except dbus.DBusException, e:
-            logger.error(e)
-            remaining_time = -1
+	if self.__battery.GetProperty('battery.rechargeable.is_discharging'):
+		try:
+			remaining_time = self.__battery.GetProperty('battery.remaining_time')
+			logger.debug("battery.remaining_time: %s" % str(remaining_time))
 
-        remaining_time = self.__str_time(remaining_time)
+		except dbus.DBusException, e:
+			logger.error(e)
+			remaining_time = -1
+
+		remaining_time = self.__str_time(remaining_time)
+	else:
+		remaining_time=0
 
         if self.property_modified_handler:
             self.property_modified_handler(BatteryInfo(charge_level, remaining_time, is_charging, is_discharging, present))
