diff -rNaud slim-1.3.0-orig/cfg.cpp slim-1.3.0/cfg.cpp
--- slim-1.3.0-orig/cfg.cpp	2007-09-01 23:41:51.000000000 +0200
+++ slim-1.3.0/cfg.cpp	2007-09-01 23:42:03.000000000 +0200
@@ -41,6 +41,17 @@
     options.insert(option("sessionstop_cmd",""));
     options.insert(option("console_cmd","/usr/X11R6/bin/xterm -C -fg white -bg black +sb -g %dx%d+%d+%d -fn %dx%d -T ""Console login"" -e /bin/sh -c ""/bin/cat /etc/issue; exec /bin/login"""));
     options.insert(option("screenshot_cmd","import -window root /slim.png"));
+    options.insert(option("f2_cmd",""));    
+    options.insert(option("f3_cmd",""));     
+    options.insert(option("f4_cmd",""));   
+    options.insert(option("f5_cmd",""));   
+    options.insert(option("f6_cmd","")); 
+    options.insert(option("ctrl_f1_cmd",""));      
+    options.insert(option("ctrl_f2_cmd",""));    
+    options.insert(option("ctrl_f3_cmd",""));     
+    options.insert(option("ctrl_f4_cmd",""));   
+    options.insert(option("ctrl_f5_cmd",""));   
+    options.insert(option("ctrl_f6_cmd",""));                
     options.insert(option("welcome_msg","Welcome to %host"));
     options.insert(option("default_user",""));
     options.insert(option("current_theme","default"));
diff -rNaud slim-1.3.0-orig/panel.cpp slim-1.3.0/panel.cpp
--- slim-1.3.0-orig/panel.cpp	2007-09-01 23:41:51.000000000 +0200
+++ slim-1.3.0/panel.cpp	2007-09-01 23:42:11.000000000 +0200
@@ -345,13 +345,59 @@
     string formerString = "";
     
     XLookupString(&event.xkey, &ascii, 1, &keysym, &compstatus);
+    
+    
+    // User defined custom keys (Ctrl)
+    if (reinterpret_cast<XKeyEvent&>(event).state & ControlMask) {
+    	switch(keysym){
+			case XK_F1:
+				system(cfg->getOption("ctrl_f1_cmd").c_str());
+				return true;     
+			case XK_F2:
+				system(cfg->getOption("ctrl_f2_cmd").c_str());
+				return true;           
+			case XK_F3:
+				system(cfg->getOption("ctrl_f3_cmd").c_str());
+				return true;  
+			case XK_F4:
+				system(cfg->getOption("ctrl_f4_cmd").c_str());
+				return true;    
+			case XK_F5:
+				system(cfg->getOption("ctrl_f5_cmd").c_str());
+				return true;    
+			case XK_F6:
+				system(cfg->getOption("ctrl_f6_cmd").c_str());
+				return true; 
+    	}     		
+    }
+    
+    
     switch(keysym){
+    	
+    	// Session chooser 
         case XK_F1:
             SwitchSession();
             return true;
-
+            
+        // User defined custom keys    
+        case XK_F2:
+            system(cfg->getOption("f2_cmd").c_str());
+            return true;
+        case XK_F3:
+            system(cfg->getOption("f3_cmd").c_str());
+            return true;
+        case XK_F4:
+            system(cfg->getOption("f4_cmd").c_str());
+            return true;
+        case XK_F5:
+            system(cfg->getOption("f5_cmd").c_str());
+            return true;    
+        case XK_F6:
+            system(cfg->getOption("f6_cmd").c_str());
+            return true;                         
+                    
+        // Take a screenshot    
         case XK_F11:
-            // Take a screenshot
             system(cfg->getOption("screenshot_cmd").c_str());
             return true;

--- slim-1.3.1/slim.conf.old	2009-02-23 20:54:06.000000000 +0800
+++ slim-1.3.1/slim.conf	2009-02-23 20:57:30.240104118 +0800
@@ -60,8 +60,22 @@
 # Executed when pressing F11 (requires imagemagick)
 screenshot_cmd      import -window root /slim.png
 
+# Custom commands for F2 to F6 and Ctrl-F1 to Ctrl-F6 keys 
+#f2_cmd      some command
+#f3_cmd      some command
+#f4_cmd      some command
+#f5_cmd      some command
+#f6_cmd      some command
+# ctrl_f1_cmd    some command 
+# ctrl_f2_cmd    some command 
+# ctrl_f3_cmd    some command  
+# ctrl_f4_cmd    some command  
+# ctrl_f5_cmd    some command  
+# ctrl_f6_cmd    some command 
+
 # welcome message. Available variables: %host, %domain
 welcome_msg         Welcome to %host
+#welcome_msg         F1-session F2-locale F3-keyboard F4-suspend F5-reboot F6-halt
 
 # Session message. Prepended to the session name when pressing F1
 # session_msg         Session: 
