--- silo-4.8/configure	2010-09-15 07:48:51.000000000 -0700
+++ configure	2012-10-02 17:29:20.149214928 -0700
@@ -26995,40 +26995,13 @@
     fi
     { echo "$as_me:$LINENO: checking for supported Qt version" >&5
 echo $ECHO_N "checking for supported Qt version... $ECHO_C" >&6; }
-    QtVersion=""
-    QtVersion2=""
-    consumeCount=-1
-    toks=`$QT_BIN/qmake --version`
-    for tok in $toks; do
-        if test "$tok" == "Qt"; then
-            consumeCount=2
-        fi
-        if test $consumeCount == 0; then
-            QtVersion2=$tok
-            break
-        else
-            consumeCount=$((consumeCount-1))
-        fi
-    done
 
-    ## Check for supported versions of Qt.
-    for ver in  4.4.0 4.4.1 4.4.2 4.4.3 4.4.4 4.4.5 \
-                4.5.0 4.5.1 4.5.2 4.5.3 4.5.4 4.5.5 \
-                4.6.0 4.6.1 4.6.2 4.6.3 4.6.4 4.6.5 ; do
-       if test "$QtVersion2" == "$ver"; then
-           QtVersion=$ver
-           break
-       fi
-    done
-
-    if test -z "$QtVersion"; then
-        { echo "$as_me:$LINENO: WARNING:   Sorry, but you need Qt version 4.4.0 or higher to compile Silo's silex tool. silex is being disabled" >&5
-echo "$as_me: WARNING:   Sorry, but you need Qt version 4.4.0 or higher to compile Silo's silex tool. silex is being disabled" >&2;}
-        SILEX=""
-    else
-        { echo "$as_me:$LINENO: result: Qt $QtVersion in $QT_INCLUDE" >&5
-echo "${ECHO_T}Qt $QtVersion in $QT_INCLUDE" >&6; }
-    fi
+    QtMinVer=440
+    as_save_IFS=$IFS; IFS=.
+    QtVersion=($($QT_BIN/qmake --version | egrep -o [0-9]+[.][0-9]+[.][0-9]+))
+    IFS=$as_save_IFS
+    QtVersion=$((QtVersion[0]*100 + QtVersion[1]*10 + QtVersion[2]))
+    if [ $QtVersion -lt $QtMinVer ]; then QtVersion=""; fi
 
     # Set the Qt bin,include,lib directories
 
