diff --git i/include/freetype/config/ftoption.h w/include/freetype/config/ftoption.h
index d162780..e37dd64 100644
--- i/include/freetype/config/ftoption.h
+++ w/include/freetype/config/ftoption.h
@@ -225,7 +225,7 @@ FT_BEGIN_HEADER
   /*                                                                       */
   /*   Define this macro if you want to enable this `feature'.             */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_USE_PNG */
+#define FT_CONFIG_OPTION_USE_PNG
 
 
   /*************************************************************************/
