commit 02a8843fe6aeea3be5b0d971e7cb5c1ede304907
Author: Benjamin Chrétien <chretien@lirmm.fr>
Date:   Sun Mar 2 15:51:08 2014 +0100

    Find OpenCV.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 295ebe8..4a89596 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,7 @@ project(pano_core)
 
 find_package(catkin REQUIRED COMPONENTS cv_bridge roscpp)
 find_package(Boost REQUIRED COMPONENTS program_options)
+find_package(OpenCV REQUIRED)
 
 ###################################
 ## catkin specific configuration ##
