--- /home/bas/webcamVideoV4L.cxx        2011-05-22 16:13:34.785441458 +0200
+++ src/panda3d-1.7.2/panda/src/vision/webcamVideoV4L.cxx       2011-05-24 16:10:09.883188149 +0200
@@ -21,7 +21,6 @@

 #include <fcntl.h>
 #include <sys/ioctl.h>
-#include <linux/videodev.h>
 #include <linux/videodev2.h>

 TypeHandle WebcamVideoV4L::_type_handle;
@@ -33,7 +32,6 @@
 //               the global list _all_webcams.
 ////////////////////////////////////////////////////////////////////
 void find_all_webcams_v4l() {
-  struct video_capability cap;
   struct v4l2_capability cap2;

   vector_string devs;
@@ -113,14 +111,6 @@
           continue;
         }
       }
-
-      // Check for Video4Linux capabilities
-      if (ioctl(fd, VIDIOCGCAP, &cap) != -1) {
-        if (cap.type & VID_TYPE_CAPTURE) {
-          //TODO: Video4Linux support
-          continue;
-        }
-      }
     }
     close(fd);
   }

--- /home/bas/webcamVideoCursorV4L.cxx  2011-05-22 16:13:39.522115838 +0200
+++ webcamVideoCursorV4L.cxx    2011-05-22 16:14:13.848838406 +0200
@@ -19,7 +19,6 @@
 #include <fcntl.h>
 #include <sys/mman.h>
 #include <sys/ioctl.h>
-#include <linux/videodev.h>
 #include <linux/videodev2.h>

 #ifdef HAVE_JPEG
