--- /home/bas/webcamVideoV4L.cxx    2011-05-22 16:13:34.785441458 +0200
+++ webcamVideoV4L.cxx  2011-05-22 16:15:53.545667433 +0200
@@ -21,9 +21,22 @@

 #include <fcntl.h>
 #include <sys/ioctl.h>
-#include <linux/videodev.h>
 #include <linux/videodev2.h>

+#define VIDIOCGCAP _IOR('v',1,struct video_capability) /* Get capabilities */
+
+struct video_capability
+{
+  char name[32];
+  int type;
+  int channels; /* Num channels */
+  int audios; /* Num audio devices */
+  int maxwidth; /* Supported width */
+  int maxheight; /* And height */
+  int minwidth; /* Supported width */
+  int minheight; /* And height */
+};
+
 TypeHandle WebcamVideoV4L::_type_handle;

 ////////////////////////////////////////////////////////////////////
@@ -113,14 +126,6 @@
           continue;
         }
       }
-
-      // Check for Video4Linux capabilities
-      if (ioctl(fd, VIDIOCGCAP, &cap) != -1) {
-        if (cap.type & VID_TYPE_CAPTURE) {
-          //TODO: Video4Linux support
-          continue;
-        }
-      }
     }
     close(fd);
   }

--- /home/bas/webcamVideoCursorV4L.cxx  2011-05-22 16:13:39.522115838 +0200
+++ webcamVideoCursorV4L.cxx    2011-05-22 16:14:13.848838406 +0200
@@ -19,7 +19,6 @@
 #include <fcntl.h>
 #include <sys/mman.h>
 #include <sys/ioctl.h>
-#include <linux/videodev.h>
 #include <linux/videodev2.h>

 #ifdef HAVE_JPEG
