diff -Nur dahdi-linux-2.2.1-rc2.orig/Makefile dahdi-linux-2.2.1-rc2/Makefile
--- dahdi-linux-2.2.1-rc2.orig/Makefile	2009-11-10 06:32:45.000000000 +1100
+++ dahdi-linux-2.2.1-rc2/Makefile	2010-01-05 21:55:05.518240715 +1100
@@ -128,7 +128,6 @@
 	for hdr in $(INST_HEADERS); do \
 		install -D -m 644 include/dahdi/$$hdr $(DESTDIR)/usr/include/dahdi/$$hdr; \
 	done
-	@rm -rf $(DESTDIR)/usr/include/zaptel
 
 uninstall-include:
 	for hdr in $(INST_HEADERS); do \
@@ -174,16 +173,9 @@
 
 install-modules: modules
 ifndef DESTDIR
-	@if modinfo zaptel > /dev/null 2>&1; then \
-		echo -n "Removing Zaptel modules for kernel $(KVERS), please wait..."; \
-		build_tools/uninstall-modules zaptel $(KVERS); \
-		rm -rf /lib/modules/$(KVERS)/zaptel; \
-		echo "done."; \
-	fi
 	build_tools/uninstall-modules dahdi $(KVERS)
 endif
 	$(KMAKE) INSTALL_MOD_PATH=$(DESTDIR) INSTALL_MOD_DIR=dahdi modules_install
-	[ `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
 
 uninstall-modules:
 ifdef DESTDIR
@@ -196,7 +188,6 @@
 		rm -rf /lib/modules/$(KVERS)/dahdi; \
 		echo "done."; \
 	fi
-	[ `id -u` = 0 ] && /sbin/depmod -a $(KVERS) || :
 endif
 
 update:
