--- definings.hh	2009-06-23 13:40:23.000000000 +0300
+++ definings.hh.patched	2011-07-11 21:21:06.000000000 +0300
@@ -71,7 +71,7 @@
 
 // Constants
 #define BUFFER_SIZE    512             // (Byte) Read-Buffer (used at class_window_main::get_pagenumber())
-#define FILES_MAX       26             // Maximum Files at Merge-List (class_window_main_merge::at merge_count_sourcefiles())
+#define FILES_MAX     9999             // Maximum Files at Merge-List (class_window_main_merge::at merge_count_sourcefiles())
 
 /*** Namespace Pdfchain ***/
 namespace Pdfchain{
@@ -332,7 +332,7 @@
 	const Glib::ustring msg_QUIT                     = "Do you really want to quit PDF Chain?";
 	const Glib::ustring msg_PDFTK                    = "PDFTK returns success\n(but that's not guaranteed)";
 	const Glib::ustring msg_FILE_OVERWRITE           = "File alredy exists!\nDo you want to overwrite?";
-	const Glib::ustring msg_MAX_26_FILES             = "26 Files Maximum!";
+	const Glib::ustring msg_MAX_26_FILES             = "9999 Files Maximum!";
 	
 	
 	
@@ -344,7 +344,7 @@
 	const Glib::ustring err_FILE_DOES_NOT_EXIST                   = "File does not exist";
 	
 	const Glib::ustring err_FILE_HAS_NO_PAGES                     = "File has no pages";
-	const Glib::ustring err_TO_MUCH_FILES                         = "To much files (max. 26)";
+	const Glib::ustring err_TO_MUCH_FILES                         = "Too many files (max. 9999)";
 	const Glib::ustring err_NO_SOURCE_FILE                        = "No Source File (PDF)";
 	const Glib::ustring err_NO_BACKGROUND_STAMP_FILE              = "No Background / Stamp File (PDF)";
 	const Glib::ustring err_NO_ATTACHMENT_FILE                    = "No Attachment(s)";
