diff -Naur a/pv/view/dsosignal.cpp b/pv/view/dsosignal.cpp
--- a/pv/view/dsosignal.cpp	2014-09-24 13:24:46.000000000 +0200
+++ b/pv/view/dsosignal.cpp	2014-09-24 18:20:25.476798193 +0200
@@ -113,9 +113,9 @@
     _trig_vpos(probe->index * 0.5 + 0.25),
     _zeroPos(probe->index * 0.5 + 0.25)
 {
-    QVector<uint64_t> vValue;
+    QVector<quint64> vValue;
     QVector<QString> vUnit;
-    QVector<uint64_t> hValue;
+    QVector<quint64> hValue;
     QVector<QString> hUnit;
     for(quint64 i = 0; i < vDialValueCount; i++)
         vValue.append(vDialValue[i]);
