diff --git a/framework/core.py b/framework/core.py
index b20fd89..ce3ef34 100644
--- a/framework/core.py
+++ b/framework/core.py
@@ -204,7 +204,7 @@ def checkDir(dirname, failifexists):
 if 'PIGLIT_BUILD_DIR' in os.environ:
     testBinDir = os.environ['PIGLIT_BUILD_DIR'] + '/bin/'
 else:
-    testBinDir = os.path.dirname(__file__) + '/../bin/'
+    testBinDir = '/usr/lib/piglit/'
 
 if 'PIGLIT_SOURCE_DIR' not in os.environ:
     p = os.path
