diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c207f9..ef42f63 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -293,24 +293,24 @@ install (
 		piglit-run.py
 		piglit-summary-html.py
 		piglit-summary-junit.py
-	DESTINATION .
+	DESTINATION lib/piglit
 )
 
 install (
 	DIRECTORY framework
-	DESTINATION .
+	DESTINATION lib/piglit
 	FILES_MATCHING PATTERN "*.py"
 )
 
 install (
 	DIRECTORY tests
-	DESTINATION .
+	DESTINATION share/piglit
 	FILES_MATCHING REGEX ".*\\.(tests|program_test|shader_test|frag|vert|ktx|cl)$"
 )
 
 install (
 	DIRECTORY generated_tests
-	DESTINATION .
+	DESTINATION share/piglit
 	FILES_MATCHING REGEX ".*\\.(shader_test|program_test|frag|vert)$"
 )
 
