diff -Naur et-sdl-sound.orig/Makefile et-sdl-sound.new/Makefile
--- et-sdl-sound.orig/Makefile	2008-08-22 20:45:07.306507251 +0400
+++ et-sdl-sound.new/Makefile	2008-08-22 20:45:22.253894000 +0400
@@ -14,8 +14,8 @@
 SCRIPTS = et-sdl-sound wolf-sdl-sound wolfsp-sdl-sound quake3-sdl-sound
 SCRIPT_IN = launcher-script.in
 
-CFLAGS = -m32 -O2 -fomit-frame-pointer -Wall -fPIC -D__SDL -D__DLOPEN_SDL -D__DEFAULT_BACKEND=SDL
-LDFLAGS = -m32 -shared
+CFLAGS = -O2 -fomit-frame-pointer -Wall -fPIC -D__SDL -D__DLOPEN_SDL -D__DEFAULT_BACKEND=SDL
+LDFLAGS = -shared
 
 all: $(SCRIPTS)
 
