--- src/rt3593-2.5.0.0/include/rtmp_def.h	2013-04-26 12:12:08.992580571 +0300
+++ src/rt3593-2.5.0.0/include/rtmp_def_new.h	2013-04-26 12:12:21.022580888 +0300
@@ -1482,13 +1482,8 @@
 #define IS_OPMODE_AP(_x)		((_x)->OpMode == OPMODE_AP)
 #define IS_OPMODE_STA(_x)		((_x)->OpMode == OPMODE_STA)
 
-#ifdef ANDROID_SUPPORT
 #define INF_MAIN_DEV_NAME		"wlan"
 #define INF_MBSSID_DEV_NAME		"wlan"
-#else
-#define INF_MAIN_DEV_NAME		"ra"
-#define INF_MBSSID_DEV_NAME		"ra"
-#endif /* ANDROID_SUPPORT */
 #define INF_WDS_DEV_NAME		"wds"
 #define INF_APCLI_DEV_NAME		"apcli"
 #define INF_MESH_DEV_NAME		"mesh"
