--- src/rt3593-2.5.0.0/common/rtusb_dev_id.c	2012-09-11 10:18:22.000000000 +0300
+++ rtusb_dev_id.c	2013-04-24 12:42:39.000000000 +0300
@@ -52,6 +52,7 @@
 	{USB_DEVICE(0x148F,0x3573)}, /* Ralink 3573 */
 	{USB_DEVICE(0x7392,0x7733)}, /* Edimax */
 	{USB_DEVICE(0x0846,0x9012)}, /* Netgear WNDA4100 N900*/
+	{USB_DEVICE(0x13B1,0x003B)}, /* Cisco LinkSys AE3000 */
 #endif /* RT3573 */
 	{ }/* Terminating entry */
 };
