diff -aur orientdb-community-1.6.2.pristine/bin/console.sh orientdb-community-1.6.2.new/bin/console.sh
--- orientdb-community-1.6.2.pristine/bin/console.sh	2013-12-09 16:04:23.000000000 +0000
+++ orientdb-community-1.6.2.new/bin/console.sh	2013-12-17 18:28:45.650107878 +0000
@@ -4,7 +4,7 @@
 #
 
 #set current working directory
-cd `dirname $0`
+#cd `dirname $0`
 
 # resolve links - $0 may be a softlink
 PRG="$0"
