--- a/nodelet/CMakeLists.txt	2013-05-22 16:54:54.787695832 -0400
+++ b/nodelet/CMakeLists.txt	2013-05-22 16:53:08.754362969 -0400
@@ -21,7 +21,7 @@
 include_directories(SYSTEM ${catkin_INCLUDE_DIRS}
                            ${BOOST_INCLUDE_DIRS}
 )
-include_directories(include)
+include_directories(include ${CATKIN_DEVEL_PREFIX}/include)
 
 # Deal with the services
 add_service_files(DIRECTORY srv FILES NodeletList.srv  NodeletLoad.srv  NodeletUnload.srv)
