From 93340174b7e13ec85ce82f299ffcd577a33902fa Mon Sep 17 00:00:00 2001
From: Rui Matos <tiagomatos@gmail.com>
Date: Fri, 29 Nov 2013 18:15:34 +0100
Subject: [PATCH] Disable GOA page in new user mode

---
 gnome-initial-setup/gnome-initial-setup.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gnome-initial-setup/gnome-initial-setup.c b/gnome-initial-setup/gnome-initial-setup.c
index 0cb370a..145d3be 100644
--- a/gnome-initial-setup/gnome-initial-setup.c
+++ b/gnome-initial-setup/gnome-initial-setup.c
@@ -105,6 +105,8 @@ should_skip_page (GisDriver    *driver,
     }
     break;
   case GIS_DRIVER_MODE_NEW_USER:
+    if (g_strcmp0 ("goa", page_id) == 0)
+      return TRUE;
   default:
     break;
   }
-- 
1.8.3.1

