diff -rud odf-converter-r5224.old/trunk/source/Shell/OdfConverter/Makefile odf-converter-r5224/trunk/source/Shell/OdfConverter/Makefile
--- odf-converter-r5224.old/trunk/source/Shell/OdfConverter/Makefile	2009-06-15 12:26:53.136293718 -0400
+++ odf-converter-r5224/trunk/source/Shell/OdfConverter/Makefile	2009-06-15 12:45:19.542225268 -0400
@@ -2,7 +2,7 @@
 include $(top_srcdir)/build/common.mk
 
 ifeq ($(OS),Cygwin)
-MACHINE_CONFIG_FILE=--machine-config $(shell cygpath -m $(shell pkg-config --variable=prefix mono))/etc/mono/2.0/machine.config
+MACHINE_CONFIG_FILE=--machine-config /etc/mono/2.0/machine.config
 CC=gcc -mno-cygwin
 EXEEXT=.exe
 PATHSEP=;
@@ -16,7 +16,7 @@
 ifeq ($(OS),Darwin)
 MACHINE_CONFIG_FILE=
 else
-MACHINE_CONFIG_FILE=--machine-config "`pkg-config --variable=prefix mono`/etc/mono/2.0/machine.config"
+MACHINE_CONFIG_FILE=--machine-config /etc/mono/2.0/machine.config
 endif
 EXEEXT=
 PATHSEP=:
@@ -72,7 +72,7 @@
 	    $(EXPORTDYNAMIC) \
 	    -Wl,--undefined=$(ZIPOPEN) -Wl,--undefined=$(UNZOPEN) \
 	    `pkg-config --libs-only-L mono` -L../../AdditionalTools/zlib123/contrib/minizip \
-	    -Wl,-Bstatic -lmono -lzlibwapi -Wl,-Bdynamic  \
+	    -Wl,-Bstatic -lmono -L../../../lib/zlibwapi.dll -Wl,-Bdynamic  \
 	    `pkg-config --libs-only-l mono | sed -e "s/\-lmono //"` \
 	    $(ZLIB) $(OTHERLIBS)
 endif
diff -rud odf-converter-r5224.old/trunk/source/Shell/OdfConverter/OdfValidator.cs odf-converter-r5224/trunk/source/Shell/OdfConverter/OdfValidator.cs
--- odf-converter-r5224.old/trunk/source/Shell/OdfConverter/OdfValidator.cs	2009-06-15 12:26:53.136293718 -0400
+++ odf-converter-r5224/trunk/source/Shell/OdfConverter/OdfValidator.cs	2009-06-15 12:39:05.099138148 -0400
@@ -49,9 +49,9 @@
 	
 #if MONO
 	// Dummy validation class
-	public class OdfValidator
+	public class OdfValidator : IValidator
 	{
-		public OdfValidator(Report report)
+		public OdfValidator(Object report)
 		{ 
 		}
 		public void validate(String fileName)
diff -rud odf-converter-r5224.old/trunk/source/Spreadsheet/Converter/Makefile odf-converter-r5224/trunk/source/Spreadsheet/Converter/Makefile
--- odf-converter-r5224.old/trunk/source/Spreadsheet/Converter/Makefile	2009-06-15 12:27:21.062335802 -0400
+++ odf-converter-r5224/trunk/source/Spreadsheet/Converter/Makefile	2009-06-15 12:26:19.275673369 -0400
@@ -30,12 +30,16 @@
 	$(CSC) $(CSC_DEBUG) $(RESOURCE_FLAGS) -target:library \
 		-reference:../../Common/OdfZipUtils/OdfZipUtils.dll \
 		-reference:../../Common/OdfConverterLib/OdfConverterLib.dll \
+		-reference:System.Drawing \
+		-reference:System.Windows.Forms \
 		-out:$@ $(SOURCE)
 
 SpreadsheetStatic.dll: $(SOURCE) Makefile $(OOX2ODF_RES) $(ODF2OOX_RES)
 	$(CSC) $(CSC_DEBUG) $(RESOURCE_FLAGS) -target:library \
 		-reference:../../Common/OdfZipUtils/OdfZipUtilsStatic.dll \
 		-reference:../../Common/OdfConverterLib/OdfConverterLibStatic.dll \
+		-reference:System.Drawing \
+		-reference:System.Windows.Forms \
 		-out:$@ $(SOURCE)
 
 clean-local:
