--- src/root/core/unix/src/TUnixSystem.cxx	2012-07-06 17:33:05.714731352 -0700
+++ nsrc/root/core/unix/src/TUnixSystem.cxx	2012-07-06 17:39:58.256455766 -0700
@@ -4919,6 +4919,10 @@
 
 //---- System, CPU and Memory info ---------------------------------------------
 
+#if defined(R__LINUX)
+#include <sys/resource.h>
+#endif
+
 #if defined(R__MACOSX)
 #include <sys/resource.h>
 #include <mach/mach.h>
--- src/root/build/rmkdepend/main.c	2012-07-06 17:33:07.404724762 -0700
+++ nsrc/root/build/rmkdepend/main.c	2012-07-06 17:35:28.780840256 -0700
@@ -60,7 +60,7 @@
 #  endif
 # elif defined(__CYGWIN__)
 extern int fchmod(int, mode_t);
-#else
+# elif !defined(__linux)
 extern int fchmod(int, int);
 # endif
 #endif
