diff -rupN a/bindings/pyroot/ROOT.py b/bindings/pyroot/ROOT.py
--- a/bindings/pyroot/ROOT.py	2014-12-21 15:44:54.496436050 -0500
+++ b/bindings/pyroot/ROOT.py	2014-12-21 15:49:42.889771871 -0500
@@ -341,7 +341,11 @@ def _displayhook( v ):
 
 
 ### set import hook to be able to trigger auto-loading as appropriate
-import __builtin__
+try:
+    import __builtin__
+except ImportError:
+    import builtins as __builtin__
+
 _orig_ihook = __builtin__.__import__
 def _importhook( name, glbls = {}, lcls = {}, fromlist = [], level = -1 ):
    if name[0:5] == 'ROOT.':
