diff -up ../sfftobmp3_orig/src/Makefile.am src/Makefile.am
--- ../sfftobmp3_orig/src/Makefile.am	2012-06-29 16:39:42.996175613 +0200
+++ src/Makefile.am	2012-06-29 16:47:13.233194186 +0200
@@ -3,4 +3,4 @@
 INCLUDES            =  -I$(top_builddir) -I$(top_srcdir)
 bin_PROGRAMS = sfftobmp
 sfftobmp_SOURCES = my_getopt.c cmdline.cpp codes.cpp common.cpp decoder.cpp input.cpp main.cpp output.cpp
-sfftobmp_LDADD = @TIFF_LIB@ @JPEG_LIB@ -lboost_filesystem-mt -lboost_system
+sfftobmp_LDADD = @TIFF_LIB@ @JPEG_LIB@ -lboost_filesystem -lboost_system
diff -up ../sfftobmp3_orig/src/Makefile.in src/Makefile.in
--- ../sfftobmp3_orig/src/Makefile.in	2012-06-29 16:39:42.996175613 +0200
+++ src/Makefile.in	2012-06-29 16:47:44.999650497 +0200
@@ -163,7 +163,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 INCLUDES = -I$(top_builddir) -I$(top_srcdir)
 sfftobmp_SOURCES = my_getopt.c cmdline.cpp codes.cpp common.cpp decoder.cpp input.cpp main.cpp output.cpp
-sfftobmp_LDADD = @TIFF_LIB@ @JPEG_LIB@ -lboost_filesystem-mt -lboost_system
+sfftobmp_LDADD = @TIFF_LIB@ @JPEG_LIB@ -lboost_filesystem -lboost_system
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-am
 
