diff -aur quickfast_1_4_0/setup.sh quickfast_1_4_0.patched/setup.sh
--- quickfast_1_4_0/setup.sh	2009-03-19 21:24:50.000000000 +0100
+++ quickfast_1_4_0.patched/setup.sh	2012-04-02 14:41:48.955559338 +0200
@@ -3,19 +3,19 @@
 # QuickFAST depends on BOOST V 1.36.0 or later. (http://www.boost.org/)
 # QuickFAST depends on Xerces V3.0 or later. (http://xerces.apache.org/xerces-c/)
 # Customize this file by setting variables to suit your environment
-export QUICKFAST_ROOT=`pwd`
+export QUICKFAST_ROOT=$(pwd)
 
-export MPC_ROOT=~/MPC
+export MPC_ROOT=$(readlink -f ../MPC)
 
-export BOOST_ROOT=~/boost/boost_1_38_0
-export BOOST_ROOT_LIB=$BOOST_ROOT/lib
-export BOOST_CFG=-gcc41-mt
-export BOOST_VERSION=boost-1_38
-export XERCES_ROOT=~/xerces/xerces-c-3.0.1
-export XERCES_LIBPATH=$XERCES_ROOT/src/.libs
-export XERCES_LIBNAME=xerces-c-3.0
-export XERCES_INCLUDE=$XERCES_ROOT/src
+export BOOST_ROOT=/usr/include/boost
+export BOOST_ROOT_LIB=/usr/lib
+export BOOST_CFG=
+export BOOST_VERSION=boost-1_49
+export XERCES_ROOT=/usr/include/xercesc
+export XERCES_LIBPATH=/usr/lib
+export XERCES_LIBNAME=xerces-c-3.1
+export XERCES_INCLUDE=/usr/include/xercesc
 
 
 export PATH=$QUICKFAST_ROOT/bin:$MPC_ROOT:$PATH
-export LD_LIBRARY_PATH=$XERCES_LIBPATH:$QUICKFAST_ROOT/lib:$BOOST_ROOT_LIB:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=/usr/lib:$LD_LIBRARY_PATH
