diff --git a/src/device.c b/src/device.c
index 91c6cd1..c53da95 100644
--- a/src/device.c
+++ b/src/device.c
@@ -4485,7 +4485,8 @@ update_info (Device *device)
         }
       else
         {
-          if (g_udev_device_get_property_as_boolean (device->priv->d, "ID_CDROM_MEDIA"))
+          if (g_udev_device_get_property_as_boolean (device->priv->d, "ID_CDROM_MEDIA") ||
+              g_udev_device_get_property_as_boolean (device->priv->d, "ID_DRIVE_FLOPPY"))
             {
               media_available = TRUE;
             }
