diff -up tvtime-1.0.2/docs/html/default.tvtime.xml.alsamixer2 tvtime-1.0.2/docs/html/default.tvtime.xml
--- tvtime-1.0.2/docs/html/default.tvtime.xml.alsamixer2	2005-09-08 06:06:57.000000000 +0200
+++ tvtime-1.0.2/docs/html/default.tvtime.xml	2009-06-28 11:55:05.000000000 +0200
@@ -116,13 +116,15 @@
   <option name="VBIDevice" value="/dev/vbi0"/>
 
   <!--
-    This sets the mixer device and channel to use.  The format is device
-    name:channel name.  Valid channels are:
+    This sets the mixer device and channel to use.  The format for OSS 
+    is device name:channel name.  Valid OSS channels are:
       vol, bass, treble, synth, pcm, speaker, line, mic, cd, mix, pcm2,
       rec, igain, ogain, line1, line2, line3, dig1, dig2, dig3, phin,
       phout, video, radio, monitor
+    The format for ALSA mixer is device/channel (e.g., "default/Line"
+    or "hw:0/CD")
    -->
-  <option name="MixerDevice" value="/dev/mixer:line"/>
+   <option name="MixerDevice" value="default/Line"/>
 
   <!--
     This option enables 16:9 aspect ratio mode by default on startup.
diff -up tvtime-1.0.2/docs/man/en/tvtime.xml.5.alsamixer2 tvtime-1.0.2/docs/man/en/tvtime.xml.5
--- tvtime-1.0.2/docs/man/en/tvtime.xml.5.alsamixer2	2005-09-08 06:05:35.000000000 +0200
+++ tvtime-1.0.2/docs/man/en/tvtime.xml.5	2009-06-28 11:55:56.000000000 +0200
@@ -234,7 +234,10 @@ This sets which device to use for VBI de
 .TP
 <option name="MixerDevice" value="/dev/mixer:line"/>
 This sets the mixer device and channel to use.  The format is device
-name:channel name.  Valid channels are:
+name:channel name for OSS mixer (e.g., "/dev/mixer:Line") or device/channel
+for ALSA (e.g., "hw:0/CD").
+
+Valid OSS channels are:
 
 .nh
 .IR vol ", " bass ", " treble ", " synth ", " pcm ", " speaker ", "
diff -up tvtime-1.0.2/src/tvtimeconf.c.alsamixer2 tvtime-1.0.2/src/tvtimeconf.c
--- tvtime-1.0.2/src/tvtimeconf.c.alsamixer2	2005-09-08 06:07:56.000000000 +0200
+++ tvtime-1.0.2/src/tvtimeconf.c	2009-06-28 11:55:56.000000000 +0200
@@ -629,9 +629,11 @@ static void print_usage( char **argv )
     lfputs( _("  -l, --xmltvlanguage=LANG   Use XMLTV data in given language, if available.\n"), stderr );
     lfputs( _("  -v, --verbose              Print debugging messages to stderr.\n"), stderr );
     lfputs( _("  -X, --display=DISPLAY      Use the given X display to connect to.\n"), stderr );
-    lfputs( _("  -x, --mixer=DEVICE[:CH]    The mixer device and channel to control.\n"
-              "                             (defaults to /dev/mixer:line)\n\n"
-              "                             Valid channels are:\n"
+    lfputs( _("  -x, --mixer=<DEVICE[:CH]>|<DEVICE/CH>\n"
+              "                             The mixer device and channel to control. The first\n"
+              "                             variant sets the OSS mixer the second one ALSA.\n"
+              "                             (defaults to default/Line)\n\n"
+              "                             Valid channels for OSS are:\n"
               "                                 vol, bass, treble, synth, pcm, speaker, line,\n"
               "                                 mic, cd, mix, pcm2, rec, igain, ogain, line1,\n"
               "                                 line2, line3, dig1, dig2, dig3, phin, phout,\n"
@@ -677,9 +679,11 @@ static void print_config_usage( char **a
     lfputs( _("  -R, --priority=PRI         Sets the process priority to run tvtime at.\n"), stderr );
     lfputs( _("  -t, --xmltv=FILE           Read XMLTV listings from the given file.\n"), stderr );
     lfputs( _("  -l, --xmltvlanguage=LANG   Use XMLTV data in given language, if available.\n"), stderr );
-    lfputs( _("  -x, --mixer=DEVICE[:CH]    The mixer device and channel to control.\n"
-              "                             (defaults to /dev/mixer:line)\n\n"
-              "                             Valid channels are:\n"
+    lfputs( _("  -x, --mixer=<DEVICE[:CH]>|<DEVICE/CH>\n"
+              "                             The mixer device and channel to control. The first\n"
+              "                             variant sets the OSS mixer the second one ALSA.\n"
+              "                             (defaults to default/Line)\n\n"
+              "                             Valid channels for OSS are:\n"
               "                                 vol, bass, treble, synth, pcm, speaker, line,\n"
               "                                 mic, cd, mix, pcm2, rec, igain, ogain, line1,\n"
               "                                 line2, line3, dig1, dig2, dig3, phin, phout,\n"
@@ -764,7 +768,7 @@ config_t *config_new( void )
 
     ct->uid = getuid();
 
-    ct->mixerdev = strdup( "/dev/mixer:line" );
+    ct->mixerdev = strdup( "default/Line" );
 
     ct->deinterlace_method = strdup( "GreedyH" );
     ct->check_freq_present = 1;
