diff -urN o3read-0.0.4.orig/Makefile o3read-0.0.4/Makefile
--- o3read-0.0.4.orig/Makefile	2005-01-07 09:28:47.000000000 +0100
+++ o3read-0.0.4/Makefile	2008-01-07 10:20:07.000000000 +0100
@@ -4,7 +4,7 @@
 
 CC = gcc
 CFLAGS = -Wall -ansi -pedantic -O -g
-PREFIX = /usr/local
+PREFIX = $(DESTDIR)/usr
 BINDIR = $(PREFIX)/bin
 MANDIR = $(PREFIX)/man/man1
 DOCS = COPYING INSTALL README TODO ChangeLog filformat.sxw
