--- src/nm-openvpn-service.c.orig	2013-02-21 02:32:10.000000000 +0400
+++ src/nm-openvpn-service.c	2013-02-24 19:44:09.175987616 +0400
@@ -120,6 +120,7 @@
 	{ NM_OPENVPN_KEY_TA,                   G_TYPE_STRING, 0, 0, FALSE },
 	{ NM_OPENVPN_KEY_TA_DIR,               G_TYPE_INT, 0, 1, FALSE },
 	{ NM_OPENVPN_KEY_TAP_DEV,              G_TYPE_BOOLEAN, 0, 0, FALSE },
+	{ NM_OPENVPN_KEY_TUN_IPV6,             G_TYPE_BOOLEAN, 0, 0, TRUE },
 	{ NM_OPENVPN_KEY_TLS_REMOTE,           G_TYPE_STRING, 0, 0, FALSE },
 	{ NM_OPENVPN_KEY_REMOTE_CERT_TLS,      G_TYPE_STRING, 0, 0, FALSE },
 	{ NM_OPENVPN_KEY_TUNNEL_MTU,           G_TYPE_INT, 0, G_MAXINT, FALSE },
@@ -986,9 +987,15 @@
 	/* Query on the management socket for user/pass */
 	add_openvpn_arg (args, "--management-query-passwords");
 
+	/* tun-ipv6 */
+	tmp = nm_setting_vpn_get_data_item (s_vpn, NM_OPENVPN_KEY_TUN_IPV6);
+	if (tmp && !strcmp (tmp, "yes")) {
+		add_openvpn_arg (args, "--tun-ipv6");
+	} else {
 	/* do not let openvpn setup routes or addresses, NM will handle it */
 	add_openvpn_arg (args, "--route-noexec");
 	add_openvpn_arg (args, "--ifconfig-noexec");
+	}
 
 	/* Now append configuration options which are dependent on the configuration type */
 	if (!strcmp (connection_type, NM_OPENVPN_CONTYPE_TLS)) {
--- src/nm-openvpn-service.h.orig	2011-12-10 00:11:09.823142496 +0400
+++ src/nm-openvpn-service.h	2011-12-10 00:11:55.643213924 +0400
@@ -64,6 +64,7 @@
 #define NM_OPENVPN_KEY_TUNNEL_MTU "tunnel-mtu"
 #define NM_OPENVPN_KEY_USERNAME "username"
 #define NM_OPENVPN_KEY_TAP_DEV "tap-dev"
+#define NM_OPENVPN_KEY_TUN_IPV6 "tun-ipv6"
 #define NM_OPENVPN_KEY_TLS_REMOTE "tls-remote"
 
 #define NM_OPENVPN_KEY_PASSWORD "password"
--- properties/nm-openvpn-dialog.ui.orig	2011-08-23 03:16:35.000000000 +0400
+++ properties/nm-openvpn-dialog.ui	2011-12-09 23:51:44.688826184 +0400
@@ -315,6 +315,22 @@
                   </packing>
                 </child>
                 <child>
+                   <object class="GtkCheckButton" id="tunipv6_checkbutton">
+                    <property name="label" translatable="yes">Build an IP_v6 capable link</property>
+                    <property name="visible">True</property>
+                    <property name="can_focus">True</property>
+                    <property name="receives_default">False</property>
+                    <property name="use_action_appearance">False</property>
+                    <property name="use_underline">True</property>
+                    <property name="draw_indicator">True</property>
+                  </object>
+                  <packing>
+                    <property name="expand">False</property>
+                    <property name="fill">True</property>
+                    <property name="position">5</property>
+                  </packing>
+                </child>
+                <child>
                   <object class="GtkHBox" id="hbox4">
                     <property name="visible">True</property>
                     <property name="can_focus">False</property>
@@ -358,7 +374,7 @@
                   <packing>
                     <property name="expand">False</property>
                     <property name="fill">True</property>
-                    <property name="position">5</property>
+                    <property name="position">6</property>
                   </packing>
                 </child>
                 <child>
@@ -405,7 +421,7 @@
                   <packing>
                     <property name="expand">False</property>
                     <property name="fill">True</property>
-                    <property name="position">6</property>
+                    <property name="position">7</property>
                   </packing>
                 </child>
                 <child>
@@ -423,7 +439,7 @@
                   <packing>
                     <property name="expand">False</property>
                     <property name="fill">True</property>
-                    <property name="position">7</property>
+                    <property name="position">8</property>
                   </packing>
                 </child>
               </object>
@@ -441,7 +457,7 @@
                   <packing>
                     <property name="expand">False</property>
                     <property name="fill">True</property>
-                    <property name="position">8</property>
+                    <property name="position">9</property>
                   </packing>
                 </child>
               </object>
--- properties/auth-helpers.c.orig	2011-12-10 00:03:19.606575008 +0400
+++ properties/auth-helpers.c	2011-12-10 00:18:54.851969772 +0400
@@ -934,6 +934,7 @@
 	NM_OPENVPN_KEY_TUNNEL_MTU,
 	NM_OPENVPN_KEY_FRAGMENT_SIZE,
 	NM_OPENVPN_KEY_TAP_DEV,
+	NM_OPENVPN_KEY_TUN_IPV6,
 	NM_OPENVPN_KEY_PROTO_TCP,
 	NM_OPENVPN_KEY_PROXY_TYPE,
 	NM_OPENVPN_KEY_PROXY_SERVER,
@@ -1511,12 +1512,6 @@
 		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (widget), TRUE);
 	}
 
-	value = g_hash_table_lookup (hash, NM_OPENVPN_KEY_MSSFIX);
-	if (value && !strcmp (value, "yes")) {
-		widget = GTK_WIDGET (gtk_builder_get_object (builder, "mssfix_checkbutton"));
-		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (widget), TRUE);
-	}
-
 	value = g_hash_table_lookup (hash, NM_OPENVPN_KEY_PROTO_TCP);
 	if (value && !strcmp (value, "yes")) {
 		widget = GTK_WIDGET (gtk_builder_get_object (builder, "tcp_checkbutton"));
@@ -1529,6 +1524,12 @@
 		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (widget), TRUE);
 	}
 
+	value = g_hash_table_lookup (hash, NM_OPENVPN_KEY_TUN_IPV6);
+	if (value && !strcmp (value, "yes")) {
+		widget = GTK_WIDGET (gtk_builder_get_object (builder, "tunipv6_checkbutton"));
+		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (widget), TRUE);
+	}
+
 	widget = GTK_WIDGET (gtk_builder_get_object (builder, "cipher_combo"));
 	value = g_hash_table_lookup (hash, NM_OPENVPN_KEY_CIPHER);
 	populate_cipher_combo (GTK_COMBO_BOX (widget), value);
@@ -1726,6 +1727,10 @@
 	if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (widget)))
 		g_hash_table_insert (hash, g_strdup (NM_OPENVPN_KEY_TAP_DEV), g_strdup ("yes"));
 
+	widget = GTK_WIDGET (gtk_builder_get_object (builder, "tunipv6_checkbutton"));
+	if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (widget)))
+		g_hash_table_insert (hash, g_strdup (NM_OPENVPN_KEY_TUN_IPV6), g_strdup ("yes"));
+
 	widget = GTK_WIDGET (gtk_builder_get_object (builder, "cipher_combo"));
 	model = gtk_combo_box_get_model (GTK_COMBO_BOX (widget));
 	if (gtk_combo_box_get_active_iter (GTK_COMBO_BOX (widget), &iter)) {
--- po/ru.po.orig	2011-12-11 01:33:04.558729218 +0400
+++ po/ru.po	2011-12-11 01:36:06.430413125 +0400
@@ -322,6 +322,9 @@
 msgid "Use a TA_P device"
 msgstr "Использовать устройство TA_P"
 
+msgid "Build an IP_v6 capable link"
+msgstr "Включить поддержку IP_v6"
+
 #: ../properties/nm-openvpn-dialog.ui.h:39
 msgid "Use a _TCP connection"
 msgstr "Использовать соединение _TCP"
