--- libgcc/libgcov.c	2013-02-04 14:06:20.000000000 -0500
+++ libgcov.c	2013-04-12 18:54:28.276519039 -0400
@@ -137,11 +137,7 @@
 
         /* Try to make directory if it doesn't already exist.  */
         if (access (filename, F_OK) == -1
-#ifdef TARGET_POSIX_IO
-            && mkdir (filename, 0755) == -1
-#else
             && mkdir (filename) == -1
-#endif
             /* The directory might have been made by another process.  */
 	    && errno != EEXIST)
 	  {
