--- outgoing.c	2010-05-22 20:41:30.000000000 +0200
+++ outgoing.c.bu	2010-05-22 20:41:12.000000000 +0200
@@ -470,7 +470,7 @@
                             sizeof(struct iovec) * (conn->vec_len - i));
                     conn->vec_len -= i;
                 }
-                ((char *) conn->vec[0].iov_base) += conn->vec[0].iov_len - (len - written);
+                conn->vec[0].iov_base = (char *)conn->vec[0].iov_base + (conn->vec[0].iov_len - (len - written));
                 conn->vec[0].iov_len = len - written;
                 break;
             }
