diff -rupN arping-2.09/src/arping.c arping-2.09.new/src/arping.c
--- arping-2.09/src/arping.c	2010-04-01 12:48:36.000000000 -0400
+++ arping-2.09.new/src/arping.c	2011-12-01 03:06:59.318850553 -0500
@@ -1271,7 +1271,7 @@ int main(int argc, char **argv)
 		       pcap_get_selectable_fd(pcap));
 	}
 
-#if HAVE_NET_BPF_H
+#ifdef BIOCIMMEDIATE
 	{
 		uint32_t on = 1;
 		if (0 < (ioctl(pcap_fileno(pcap), BIOCIMMEDIATE,
