diff -uNr a/configure.ac b/configure.ac
--- a/configure.ac	2010-10-02 00:47:20.950000000 +0800
+++ b/configure.ac	2010-10-02 00:49:30.303000001 +0800
@@ -56,8 +56,8 @@
 if test "x$enable_notify" = "xyes"; then
   PKG_CHECK_MODULES(LIBNOTIFY, libnotify)
   AC_DEFINE(HAVE_NOTIFY, 1, [Define to 1 if libnotify support is required.])
-  AC_SUBST(NOTIFY_CFLAGS)
-  AC_SUBST(NOTIFY_LIBS)
+  AC_SUBST(LIBNOTIFY_CFLAGS)
+  AC_SUBST(LIBNOTIFY_LIBS)
 fi
 
 ## ----------------
@@ -71,6 +71,8 @@
 if test "x$enable_gstreamer" = "xyes"; then
   PKG_CHECK_MODULES(GSTREAMER, gstreamer-0.10)
   AC_DEFINE(HAVE_GSTREAMER, 1, [Define to 1 if gstreamer support is required.])
+  AC_SUBST(GSTREAMER_CFLAGS)
+  AC_SUBST(GSTREAMER_LIBS)
 fi
 
 ## ----------------
diff -uNr a/uget-cmd/Makefile.am b/uget-cmd/Makefile.am
--- a/uget-cmd/Makefile.am	2010-10-02 00:47:20.965000000 +0800
+++ b/uget-cmd/Makefile.am	2010-10-02 00:49:51.446000001 +0800
@@ -4,7 +4,7 @@
 uget_cmd_CPPFLAGS = -DDATADIR='"$(datadir)"' -I$(top_srcdir)/include -I$(top_srcdir)/uget-cmd
 uget_cmd_CFLAGS   = @LFS_CFLAGS@  @GIO_CFLAGS@ @GTHREAD_CFLAGS@ @CURL_CFLAGS@ @GSTREAMER_CFLAGS@ @LIBPWMD_CFLAGS@
 uget_cmd_LDFLAGS  = @LFS_LDFLAGS@ @GIO_LIBS@   @GTHREAD_LIBS@   @CURL_LIBS@   @GSTREAMER_LIBS@   @LIBPWMD_LIBS@
-uget_cmd_LDADD    = $(top_builddir)/src/libuglib.a
+uget_cmd_LDADD    = $(top_builddir)/src/libuglib.a $(GSTREAMER_LIBS)
 
 uget_cmd_SOURCES  = \
 	ug_category-cmd.c \
diff -uNr a/uget-gtk/Makefile.am b/uget-gtk/Makefile.am
--- a/uget-gtk/Makefile.am	2010-10-02 00:47:20.962000000 +0800
+++ b/uget-gtk/Makefile.am	2010-10-02 00:48:29.289000001 +0800
@@ -4,7 +4,7 @@
 uget_gtk_CPPFLAGS = -DDATADIR='"$(datadir)"' -I$(top_srcdir)/include -I$(top_srcdir)/uget-gtk
 uget_gtk_CFLAGS   = @LFS_CFLAGS@  @GTK_CFLAGS@ @GIO_CFLAGS@ @GTHREAD_CFLAGS@ @CURL_CFLAGS@ @LIBNOTIFY_CFLAGS@ @GSTREAMER_CFLAGS@ @LIBPWMD_CFLAGS@
 uget_gtk_LDFLAGS  = @LFS_LDFLAGS@ @GTK_LIBS@   @GIO_LIBS@   @GTHREAD_LIBS@   @CURL_LIBS@   @LIBNOTIFY_LIBS@   @GSTREAMER_LIBS@   @LIBPWMD_LIBS@
-uget_gtk_LDADD    = $(top_builddir)/src/libuglib.a $(NOTIFY_LIBS)
+uget_gtk_LDADD    = $(top_builddir)/src/libuglib.a $(LIBNOTIFY_LIBS) $(GSTREAMER_LIBS)
 
 uget_gtk_SOURCES  = \
 	ug_selector.c  ug_summary.c  ug_setting_form.c \
