diff -ur SFML-1.6.orig/include/SFML/System/Resource.hpp SFML-1.6/include/SFML/System/Resource.hpp
--- SFML-1.6.orig/include/SFML/System/Resource.hpp	2010-01-27 14:53:54.000000000 +0100
+++ SFML-1.6/include/SFML/System/Resource.hpp	2012-09-12 00:08:34.313630269 +0200
@@ -28,6 +28,7 @@
 ////////////////////////////////////////////////////////////
 // Headers
 ////////////////////////////////////////////////////////////
+#include <cstddef>
 #include <set>
 
 
diff -ur SFML-1.6.orig/src/SFML/Audio/Makefile SFML-1.6/src/SFML/Audio/Makefile
--- SFML-1.6.orig/src/SFML/Audio/Makefile	2010-01-27 15:00:19.000000000 +0100
+++ SFML-1.6/src/SFML/Audio/Makefile	2012-09-12 00:10:38.136964035 +0200
@@ -4,21 +4,21 @@
 OBJVORBIS = $(SRCVORBIS:.c=.o)
 
 ifeq ($(STATIC), yes)
-    LIB     = libsfml-audio-s.a
+    LIB     = libsfml1.6-audio-s.a
     LIBNAME = $(LIBPATH)/$(LIB)
     INSTALL = 
 else
-    LIB     = libsfml-audio.so
+    LIB     = libsfml1.6-audio.so
     LIBNAME = $(LIBPATH)/$(LIB).$(VERSION)
     INSTALL = && $(LN) $(LNFLAGS) $(LIB).$(VERSION) $(DESTLIBDIR)/$(LIB)
 endif
 
 all: $(LIB)
 
-libsfml-audio-s.a: $(OBJ) $(OBJVORBIS)
+libsfml1.6-audio-s.a: $(OBJ) $(OBJVORBIS)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) $(OBJVORBIS)
 
-libsfml-audio.so: $(OBJ) $(OBJVORBIS)
+libsfml1.6-audio.so: $(OBJ) $(OBJVORBIS)
 	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJVORBIS) -lsndfile -lopenal
 
 $(OBJ): %.o: %.cpp
diff -ur SFML-1.6.orig/src/SFML/Graphics/Makefile SFML-1.6/src/SFML/Graphics/Makefile
--- SFML-1.6.orig/src/SFML/Graphics/Makefile	2010-01-27 15:00:05.000000000 +0100
+++ SFML-1.6/src/SFML/Graphics/Makefile	2012-09-12 00:10:38.130297367 +0200
@@ -12,21 +12,21 @@
 OBJZLIB = $(SRCZLIB:.c=.o)
 
 ifeq ($(STATIC), yes)
-    LIB     = libsfml-graphics-s.a
+    LIB     = libsfml1.6-graphics-s.a
     LIBNAME = $(LIBPATH)/$(LIB)
     INSTALL = 
 else
-    LIB     = libsfml-graphics.so
+    LIB     = libsfml1.6-graphics.so
     LIBNAME = $(LIBPATH)/$(LIB).$(VERSION)
     INSTALL = && $(LN) $(LNFLAGS) $(LIB).$(VERSION) $(DESTLIBDIR)/$(LIB)
 endif
 
 all: $(LIB)
 
-libsfml-graphics-s.a: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
+libsfml1.6-graphics-s.a: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
 
-libsfml-graphics.so: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
+libsfml1.6-graphics.so: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
 	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB) -lfreetype -lX11 -lGL
 
 $(OBJ): %.o: %.cpp
diff -ur SFML-1.6.orig/src/SFML/Network/Makefile SFML-1.6/src/SFML/Network/Makefile
--- SFML-1.6.orig/src/SFML/Network/Makefile	2010-01-27 15:00:37.000000000 +0100
+++ SFML-1.6/src/SFML/Network/Makefile	2012-09-12 00:10:38.150297362 +0200
@@ -2,21 +2,21 @@
 OBJ = $(SRC:.cpp=.o)
 
 ifeq ($(STATIC), yes)
-    LIB     = libsfml-network-s.a
+    LIB     = libsfml1.6-network-s.a
     LIBNAME = $(LIBPATH)/$(LIB)
     INSTALL = 
 else
-    LIB     = libsfml-network.so
+    LIB     = libsfml1.6-network.so
     LIBNAME = $(LIBPATH)/$(LIB).$(VERSION)
     INSTALL = && $(LN) $(LNFLAGS) $(LIB).$(VERSION) $(DESTLIBDIR)/$(LIB)
 endif
 
 all: $(LIB)
 
-libsfml-network-s.a: $(OBJ)
+libsfml1.6-network-s.a: $(OBJ)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ)
 
-libsfml-network.so: $(OBJ)
+libsfml1.6-network.so: $(OBJ)
 	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ)
 
 $(OBJ): %.o: %.cpp
diff -ur SFML-1.6.orig/src/SFML/System/Makefile SFML-1.6/src/SFML/System/Makefile
--- SFML-1.6.orig/src/SFML/System/Makefile	2010-01-27 15:00:23.000000000 +0100
+++ SFML-1.6/src/SFML/System/Makefile	2012-09-12 00:10:38.140297366 +0200
@@ -2,21 +2,21 @@
 OBJ = $(SRC:.cpp=.o)
 
 ifeq ($(STATIC), yes)
-    LIB     = libsfml-system-s.a
+    LIB     = libsfml1.6-system-s.a
     LIBNAME = $(LIBPATH)/$(LIB)
     INSTALL = 
 else
-    LIB     = libsfml-system.so
+    LIB     = libsfml1.6-system.so
     LIBNAME = $(LIBPATH)/$(LIB).$(VERSION)
     INSTALL = && $(LN) $(LNFLAGS) $(LIB).$(VERSION) $(DESTLIBDIR)/$(LIB)
 endif
 
 all: $(LIB)
 
-libsfml-system-s.a: $(OBJ)
+libsfml1.6-system-s.a: $(OBJ)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ)
 
-libsfml-system.so: $(OBJ)
+libsfml1.6-system.so: $(OBJ)
 	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) -lpthread
 
 $(OBJ): %.o: %.cpp
diff -ur SFML-1.6.orig/src/SFML/Window/Linux/Joystick.hpp SFML-1.6/src/SFML/Window/Linux/Joystick.hpp
--- SFML-1.6.orig/src/SFML/Window/Linux/Joystick.hpp	2010-03-18 14:15:05.000000000 +0100
+++ SFML-1.6/src/SFML/Window/Linux/Joystick.hpp	2012-09-12 00:08:34.313630269 +0200
@@ -31,6 +31,7 @@
 #if defined(SFML_SYSTEM_LINUX)
     #include <linux/joystick.h>
     #include <fcntl.h>
+    #include <unistd.h>
 #elif defined(SFML_SYSTEM_FREEBSD)
     // #include <sys/joystick.h> ?
     #define ABS_MAX 1
diff -ur SFML-1.6.orig/src/SFML/Window/Makefile SFML-1.6/src/SFML/Window/Makefile
--- SFML-1.6.orig/src/SFML/Window/Makefile	2010-01-27 15:00:37.000000000 +0100
+++ SFML-1.6/src/SFML/Window/Makefile	2012-09-12 00:10:38.146964029 +0200
@@ -3,21 +3,21 @@
 OBJGLEW = $(SRCGLEW:.c=.o)
 
 ifeq ($(STATIC), yes)
-    LIB     = libsfml-window-s.a
+    LIB     = libsfml1.6-window-s.a
     LIBNAME = $(LIBPATH)/$(LIB)
     INSTALL = 
 else
-    LIB     = libsfml-window.so
+    LIB     = libsfml1.6-window.so
     LIBNAME = $(LIBPATH)/$(LIB).$(VERSION)
     INSTALL = && $(LN) $(LNFLAGS) $(LIB).$(VERSION) $(DESTLIBDIR)/$(LIB)
 endif
 
 all: $(LIB)
 
-libsfml-window-s.a: $(OBJ)
+libsfml1.6-window-s.a: $(OBJ)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ)
 
-libsfml-window.so: $(OBJ)
+libsfml1.6-window.so: $(OBJ)
 	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) -lX11 -lXrandr -lGL
 
 $(OBJ): %.o: %.cpp
