--- Makefile	2010-01-26 00:55:56.000000000 +0100
+++ Makefile2	2010-11-27 19:40:40.000000000 +0100
@@ -3,40 +3,32 @@
 ##########################################################################
 
 PREFIX=/usr
-ALT_PREFIX=/usr/local
 
 # Install with individual configuration for every user
 install: uninstall
-	sudo mkdir -p $(PREFIX)/bin
-	sudo mkdir -p $(PREFIX)/lib/FreeArc
-	sudo mkdir -p $(PREFIX)/lib/FreeArc/Documentation
-	sudo cp -r bin/* $(PREFIX)/bin
-	sudo cp -r lib/* $(PREFIX)/lib/FreeArc
-	sudo cp -r Documentation/* $(PREFIX)/lib/FreeArc/Documentation
-	mkdir -p ~/.FreeArc
-	cp -r cfg/* ~/.FreeArc
+	 mkdir -p ${DESTDIR}/bin
+	 mkdir -p ${DESTDIR}/lib/FreeArc
+	 mkdir -p ${DESTDIR}/lib/FreeArc/Documentation
+	 cp -r bin/* ${DESTDIR}/bin
+	 cp -r lib/* ${DESTDIR}/lib/FreeArc
+	 cp -r Documentation/* ${DESTDIR}/lib/FreeArc/Documentation
+	 mkdir -p ~/.FreeArc
+	 cp -r cfg/* ~/.FreeArc
 
 uninstall:
-	sudo rm -rf ~/.FreeArc
-	sudo rm -rf /etc/arc.languages
-	sudo rm -rf /etc/FreeArc
-	sudo rm -rf $(PREFIX)/lib/FreeArc
-	sudo rm -rf $(ALT_PREFIX)/lib/FreeArc
-	sudo rm -f /etc/arc.groups
-	sudo rm -f /etc/arc.ini
-	sudo rm -f /etc/winarc.history
-	sudo rm -f /etc/winarc.ini
-	sudo rm -f /etc/freearc.history
-	sudo rm -f /etc/freearc.ini
-	sudo rm -f $(PREFIX)/bin/arc
-	sudo rm -f $(PREFIX)/bin/unarc
-	sudo rm -f $(PREFIX)/bin/winarc
-	sudo rm -f $(PREFIX)/bin/freearc
-	sudo rm -f $(PREFIX)/lib/*arc*.sfx
-	sudo rm -f $(ALT_PREFIX)/bin/arc
-	sudo rm -f $(ALT_PREFIX)/bin/unarc
-	sudo rm -f $(ALT_PREFIX)/bin/winarc
-	sudo rm -f $(ALT_PREFIX)/bin/freearc
-	sudo rm -f $(ALT_PREFIX)/lib/*arc*.sfx
+	 rm -rf /etc/arc.languages
+	 rm -rf /etc/FreeArc
+	 rm -rf ${DESTDIR}/lib/FreeArc
+	 rm -f /etc/arc.groups
+	 rm -f /etc/arc.ini
+	 rm -f /etc/winarc.history
+	 rm -f /etc/winarc.ini
+	 rm -f /etc/freearc.history
+	 rm -f /etc/freearc.ini
+	 rm -f ${DESTDIR}/bin/arc
+	 rm -f ${DESTDIR}/bin/unarc
+	 rm -f ${DESTDIR}/bin/winarc
+	 rm -f ${DESTDIR}/bin/freearc
+	 rm -f ${DESTDIR}/lib/*arc*.sfx
 
 
