diff -Nur NetworkManager-0.9.0.orig/src/nm-policy.c NetworkManager-0.9.0/src/nm-policy.c
--- a/src/nm-policy.c	2011-08-23 06:41:02.099686450 +0000
+++ b/src/nm-policy.c	     2011-08-23 06:43:38.227791737 +0000
@@ -320,6 +320,7 @@
		return;
	}

+#if 0
	/* Try automatically determined hostname from the best device's IP config */
	if (!best4)
		best4 = get_best_ip4_device (policy->manager);
@@ -374,6 +374,7 @@
		}
	}

+#endif
	/* If no automatically-configured hostname, try using the hostname from
	 * when NM started up.
	 */
