--- hsspiffy.e	2011-05-28 17:01:07.000000000 +0300
+++ hsspiffy.e.revert	2011-05-29 15:04:51.081617497 +0300
@@ -24,11 +24,11 @@
 --globals--
 global integer PATH_SLASH
 
-ifdef WINDOWS then
-  PATH_SLASH='\\'  
-elsedef
+if platform()=LINUX then  --yes, believe it or not, HSPEAK can be compiled to run on Linux
   PATH_SLASH='/'
-end ifdef
+else
+  PATH_SLASH='\\'  
+end if
 
 --redundant to the standard upper() and lower() functions I know, but I
 --use them constantly, and almost never use the other wildcard functs.
@@ -209,11 +209,11 @@
 
 --normalise a pathname to use the platform specific path delimiter--
 global function normalize_filename(sequence s)
-  ifdef WINDOWS then
-    return filenamos(s)
-  elsedef
+  if platform()=LINUX then
     return filenamix(s)
-  end ifdef
+  else
+    return filenamos(s)
+  end if
 end function
 
 --extract only portion of a string after the last of a delimiter--
@@ -604,3 +604,4 @@
     return(s)
   end if
 end function
+
