diff -aur python-poppler-qt4-0.16.3.pristine/setup.py python-poppler-qt4-0.16.3.new/setup.py
--- python-poppler-qt4-0.16.3.pristine/setup.py	2013-12-29 15:56:36.315973198 +0100
+++ python-poppler-qt4-0.16.3.new/setup.py	2013-12-29 15:59:25.528520481 +0100
@@ -67,6 +67,11 @@
     except OSError:
         return attrs
     flag_map = {'-I': 'include_dirs', '-L': 'library_dirs', '-l': 'libraries'}
+
+    # for python3 turn bytes back to string
+    if sys.version_info.major > 2:
+        output = output.decode('utf-8')
+
     for flag in shlex.split(output):
         option, path = flag[:2], flag[2:]
         if option in flag_map:
@@ -80,6 +85,11 @@
     cmd = ['pkg-config', '--modversion', package]
     try:
         output = subprocess.Popen(cmd, stdout=subprocess.PIPE).communicate()[0]
+
+        # for python3 turn bytes back to string
+        if sys.version_info.major > 2:
+            output = output.decode('utf-8')
+
         return tuple(map(int, re.findall(r'\d+', output)))
     except OSError:
         sys.stderr.write("Can't determine version of %s\n" % package)
