diff -Naurb ./qt-creator-2.6.2-src/src/plugins/cpptools/cppchecksymbols.cpp ./qt-creator-2.6.2-src-patched/src/plugins/cpptools/cppchecksymbols.cpp
--- ./qt-creator-2.6.2-src/src/plugins/cpptools/cppchecksymbols.cpp	2013-02-10 22:05:02.011650554 +0400
+++ ./qt-creator-2.6.2-src-patched/src/plugins/cpptools/cppchecksymbols.cpp	2013-02-10 22:05:02.011650554 +0400
@@ -53,7 +53,7 @@
 // Whenever this feature is considered "accepted" the switch below should be permanently
 // removed, unless we decide to actually make this a user setting - that is why it's
 // currently a bool instead of a define.
-static const bool highlightCtorDtorAsType = true;
+static const bool highlightCtorDtorAsType = false;
 
 using namespace CPlusPlus;
 using namespace CppTools;
