diff -Naurb ./qt-creator-2.6.0-src/src/libs/3rdparty/cplusplus/Token.h ./qt-creator-2.6.0-src-patched/src/libs/3rdparty/cplusplus/Token.h
--- ./qt-creator-2.6.0-src/src/libs/3rdparty/cplusplus/Token.h	2012-11-06 17:01:49.000000000 +0400
+++ ./qt-creator-2.6.0-src-patched/src/libs/3rdparty/cplusplus/Token.h	2012-11-16 18:01:42.000000000 +0400
@@ -117,11 +117,9 @@
     T_FIRST_KEYWORD,
     T_ASM = T_FIRST_KEYWORD,
     T_AUTO,
-    T_BOOL,
     T_BREAK,
     T_CASE,
     T_CATCH,
-    T_CHAR,
     T_CHAR16_T,
     T_CHAR32_T,
     T_CLASS,
@@ -133,7 +131,6 @@
     T_DEFAULT,
     T_DELETE,
     T_DO,
-    T_DOUBLE,
     T_DYNAMIC_CAST,
     T_ELSE,
     T_ENUM,
@@ -141,14 +138,11 @@
     T_EXPORT,
     T_EXTERN,
     T_FALSE,
-    T_FLOAT,
     T_FOR,
     T_FRIEND,
     T_GOTO,
     T_IF,
     T_INLINE,
-    T_INT,
-    T_LONG,
     T_MUTABLE,
     T_NAMESPACE,
     T_NEW,
@@ -161,8 +155,6 @@
     T_REGISTER,
     T_REINTERPRET_CAST,
     T_RETURN,
-    T_SHORT,
-    T_SIGNED,
     T_SIZEOF,
     T_STATIC,
     T_STATIC_ASSERT,
@@ -178,12 +170,9 @@
     T_TYPEID,
     T_TYPENAME,
     T_UNION,
-    T_UNSIGNED,
     T_USING,
     T_VIRTUAL,
-    T_VOID,
     T_VOLATILE,
-    T_WCHAR_T,
     T_WHILE,

     T___ATTRIBUTE__,
@@ -245,6 +234,21 @@
     T_Q_GADGET,
     T_LAST_KEYWORD = T_Q_GADGET,

+    // POD types
+    T_FIRST_POD,
+    T_BOOL = T_FIRST_POD,
+    T_CHAR,
+    T_DOUBLE,
+    T_FLOAT,
+    T_INT,
+    T_LONG,
+    T_SHORT,
+    T_SIGNED,
+    T_UNSIGNED,
+    T_VOID,
+    T_WCHAR_T,
+    T_LAST_POD = T_WCHAR_T,
+
     // aliases
     T_OR = T_PIPE_PIPE,
     T_AND = T_AMPER_AMPER,
@@ -317,6 +321,9 @@
     inline bool isKeyword() const
     { return f.kind >= T_FIRST_KEYWORD && f.kind < T_FIRST_QT_KEYWORD; }

+    inline bool isPodType() const
+    { return f.kind >= T_FIRST_POD && f.kind <= T_LAST_POD; }
+
     inline bool isComment() const
     { return f.kind == T_COMMENT || f.kind == T_DOXY_COMMENT ||
       f.kind == T_CPP_COMMENT || f.kind == T_CPP_DOXY_COMMENT; }
diff -Naurb ./qt-creator-2.6.0-src/src/plugins/cppeditor/cppeditor.cpp ./qt-creator-2.6.0-src-patched/src/plugins/cppeditor/cppeditor.cpp
--- ./qt-creator-2.6.0-src/src/plugins/cppeditor/cppeditor.cpp	2012-11-06 17:01:49.000000000 +0400
+++ ./qt-creator-2.6.0-src-patched/src/plugins/cppeditor/cppeditor.cpp	2012-11-16 18:09:12.000000000 +0400
@@ -2198,6 +2198,7 @@
         categories << TextEditor::C_NUMBER
                    << TextEditor::C_STRING
                    << TextEditor::C_TYPE
+                   << TextEditor::C_POD_TYPE
                    << TextEditor::C_KEYWORD
                    << TextEditor::C_OPERATOR
                    << TextEditor::C_PREPROCESSOR
diff -Naurb ./qt-creator-2.6.0-src/src/plugins/cppeditor/cppeditorenums.h ./qt-creator-2.6.0-src-patched/src/plugins/cppeditor/cppeditorenums.h
--- ./qt-creator-2.6.0-src/src/plugins/cppeditor/cppeditorenums.h	2012-11-06 17:01:49.000000000 +0400
+++ ./qt-creator-2.6.0-src-patched/src/plugins/cppeditor/cppeditorenums.h	2012-11-16 18:09:12.000000000 +0400
@@ -42,6 +42,7 @@
     CppNumberFormat,
     CppStringFormat,
     CppTypeFormat,
+    CppPodTypeFormat,
     CppKeywordFormat,
     CppOperatorFormat,
     CppPreprocessorFormat,
diff -Naurb ./qt-creator-2.6.0-src/src/plugins/cppeditor/cpphighlighter.cpp ./qt-creator-2.6.0-src-patched/src/plugins/cppeditor/cpphighlighter.cpp
--- ./qt-creator-2.6.0-src/src/plugins/cppeditor/cpphighlighter.cpp	2012-11-06 17:01:49.000000000 +0400
+++ ./qt-creator-2.6.0-src-patched/src/plugins/cppeditor/cpphighlighter.cpp	2012-11-16 18:09:12.000000000 +0400
@@ -195,6 +195,9 @@
         } else if (tk.isKeyword() || CppTools::isQtKeyword(text.midRef(tk.begin(), tk.length())) || tk.isObjCAtKeyword())
             setFormat(tk.begin(), tk.length(), m_formats[CppKeywordFormat]);

+        else if (tk.isPodType())
+            setFormat(tk.begin(), tk.length(), m_formats[CppPodTypeFormat]);
+
         else if (tk.isOperator())
             setFormat(tk.begin(), tk.length(), m_formats[CppOperatorFormat]);

diff -Naurb ./qt-creator-2.6.0-src/src/plugins/texteditor/texteditorconstants.h ./qt-creator-2.6.0-src-patched/src/plugins/texteditor/texteditorconstants.h
--- ./qt-creator-2.6.0-src/src/plugins/texteditor/texteditorconstants.h	2012-11-06 17:01:49.000000000 +0400
+++ ./qt-creator-2.6.0-src-patched/src/plugins/texteditor/texteditorconstants.h	2012-11-16 18:20:16.000000000 +0400
@@ -59,6 +59,7 @@
     C_VIRTUAL_METHOD,
     C_FUNCTION,
     C_KEYWORD,
+    C_POD_TYPE,
     C_OPERATOR,
     C_PREPROCESSOR,
     C_LABEL,
diff -Naurb ./qt-creator-2.6.0-src/src/plugins/texteditor/texteditorsettings.cpp ./qt-creator-2.6.0-src-patched/src/plugins/texteditor/texteditorsettings.cpp
--- ./qt-creator-2.6.0-src/src/plugins/texteditor/texteditorsettings.cpp	2012-11-06 17:01:49.000000000 +0400
+++ ./qt-creator-2.6.0-src-patched/src/plugins/texteditor/texteditorsettings.cpp	2012-11-16 18:18:44.000000000 +0400
@@ -156,6 +156,7 @@
                                          Qt::darkBlue));
     formatDescr.append(FormatDescription(C_STRING, tr("String"),
                                          tr("Character and string literals."), Qt::darkGreen));
+    formatDescr.append(FormatDescription(C_POD_TYPE, tr("POD type"), tr("Plain-old data types"), Qt::darkYellow));
     formatDescr.append(FormatDescription(C_TYPE, tr("Type"), tr("Name of a type."),
                                          Qt::darkMagenta));
     formatDescr.append(FormatDescription(C_LOCAL, tr("Local"), tr("Local variables.")));
@@ -220,9 +221,12 @@
                                          tr("Variables defined outside the script."),
                                          jsGlobalFormat));

+    Format keywordFormat;
+    keywordFormat.setForeground(Qt::black);
+    keywordFormat.setBold(true);
     formatDescr.append(FormatDescription(C_KEYWORD, tr("Keyword"),
-                                         tr("Reserved keywords of the programming language."),
-                                         Qt::darkYellow));
+                       tr("Reserved keywords of the programming language."), keywordFormat));
+
     formatDescr.append(FormatDescription(C_OPERATOR, tr("Operator"),
                                          tr("Operators. (for example operator++ operator-=)")));
     formatDescr.append(FormatDescription(C_PREPROCESSOR, tr("Preprocessor"),
