diff -aur meh-0.3.pristine/src/gif.c meh-0.3.new/src/gif.c
--- meh-0.3.pristine/src/gif.c	2013-05-25 03:47:16.098847514 -0700
+++ meh-0.3.new/src/gif.c	2013-05-25 03:47:57.745288563 -0700
@@ -57,7 +57,6 @@
 	SavedImage *s;
 
 	if(DGifSlurp(g->gif) == GIF_ERROR){
-		PrintGifError();
 		return 1;
 	}
 
@@ -68,7 +67,6 @@
 	else if(g->gif->SColorMap)
 		colormap = g->gif->SColorMap->Colors;
 	else{
-		PrintGifError();
 		return 1;
 	}
 
