x.orig	2007-05-10 22:56:44.000000000 +0800
+++ makefile.unix	2007-05-10 23:00:21.000000000 +0800
@@ -127,7 +127,7 @@
 ###########################################################################
 
 # GCC on x86
-CFLAGS = -O2
+#CFLAGS = -O2
 
 # GCC on x86 with some optimizations
 # CFLAGS = -O2 -mtune=i686 -fomit-frame-pointer -fstrength-reduce
-ffast-math
@@ -683,7 +683,7 @@
 # JOY_PS2 = 1
 
 # SDL joystick support.
-# JOY_SDL = 1
+JOY_SDL = 1
 
 # This will enable the use of the Happ Controls UGCI(tm) USB
devices'
 # Coin/Start inputs using the libugci library available from
@@ -765,7 +765,7 @@
 # X11_DGA = 1
 
 # Build support for OpenGL rendering into the X-Window version
-# X11_OPENGL = 1
+X11_OPENGL = 1
 
 # Build support for rendering through Glide into the X-Window
version
 # X11_GLIDE = 1
@@ -778,12 +778,12 @@
 # Choose the location of your X headers and libraries.
 
 # Standard location for X11 
-# X11INC		= -I/usr/include/X11
-# X11LIB		= -L/usr/lib/X11
+X11INC		= -I/usr/include/X11
+X11LIB		= -L/usr/lib/X11
 
 # Standard location for XFree86
-X11INC		= -I/usr/X11R6/include
-X11LIB		= -L/usr/X11R6/lib
+# X11INC		= -I/usr/X11R6/include
+# X11LIB		= -L/usr/X11R6/lib
 
 # Standard location for Sun systems
 # X11INC		= -I/usr/openwin/include
