--- orig/bucd/usr/local/bin/buc	2009-05-20 04:03:00.000000000 -0500
+++ new/bucd/usr/local/bin/buc	2010-12-18 13:13:53.793333336 -0600
@@ -1,5 +1,5 @@
  #!/bin/bash
- dirname=/usr/local/buc/
+ dirname=/usr/share/buc
   
  percorso=`echo $1 | sed 's/\/[^\/]*$//'`
  argomento=`echo $1 | sed 's/^.*\///g'`

--- orig/bucd/usr/local/buc/welcome.mc	2009-05-20 04:03:00.000000000 -0500
+++ new/bucd/usr/local/buc/welcome.mc	2010-12-18 13:28:41.353333336 -0600
@@ -12,7 +12,7 @@
 		</label>
 
 		<button title="Vai alla cartella">
-			nautilus /usr/local/buc/ &
+			nautilus /usr/share/buc/ &
 		</button>
 	</tab>
 </config>
