Remove README.ibmvstgt entry from doc/Makefile.

Signed-off-by: Zdenek Kaspar <zkaspar82@gmail.com>

--- tgt-1.0.14.orig/doc/Makefile	2011-03-01 21:36:05.000000000 +0100
+++ tgt-1.0.14/doc/Makefile	2011-03-01 21:36:23.000000000 +0100
@@ -4,7 +4,7 @@
 MANPAGES = manpages/tgtadm.8 manpages/tgt-admin.8 manpages/tgtimg.8 \
 		manpages/tgt-setup-lun.8 manpages/tgtd.8
 
-DOCS = README.fcoe README.ibmvstgt README.iscsi README.iser \
+DOCS = README.fcoe README.iscsi README.iser \
 	    README.lu_configuration README.mmc tmf.txt
 
 XSLTPROC = /usr/bin/xsltproc
