--- doc/Makefile.am	2007-02-26 21:22:55.000000000 +0100
+++ doc/Makefile.am.new	2010-07-31 01:32:59.615113725 +0200
@@ -4,7 +4,7 @@
 ## If you don't want it to overwrite it,
 ## 	Please disable it in the Anjuta project configuration
 
-linux_mtp_streaming_server_docdir = $(prefix)/@NO_PREFIX_PACKAGE_DOC_DIR@
+linux_mtp_streaming_server_docdir = $(prefix)/share/@NO_PREFIX_PACKAGE_DOC_DIR@
 
 linux_mtp_streaming_server_doc_DATA = \
 	ChangeLog
--- Makefile.am	2007-03-06 16:41:42.000000000 +0100
+++ Makefile.am.new	2010-07-31 01:34:50.215103635 +0200
@@ -5,7 +5,7 @@
 
 SUBDIRS = include src data doc
 
-linux_mtp_streaming_serverdocdir = ${prefix}/doc/linux_mtp_streaming_server
+linux_mtp_streaming_serverdocdir = ${prefix}/share/doc/linux_mtp_streaming_server
 linux_mtp_streaming_serverdoc_DATA = \
 	README\
 	COPYING\
