--- a/src/plan9port-build/LICENSE	2010-05-17 02:39:12.000000000 +0000
+++ b/src/plan9port-build/LICENSE	2010-05-17 02:43:17.000000000 +0000
@@ -14,7 +14,10 @@
 
 The bitmap fonts in the font/luc, font/lucm, font/lucsans, and font/pelm
 directory are copyright B&H Inc. and distributed under more restricted
-terms under agreement with B&H.  See the NOTICE file in those directories.
+terms under agreement with B&H.  See the NOTICE files in those directories.
+
+Many of the COPYRIGHT, LICENSE, NOTICE and, or README informative files
+are installed, but in /usr/local/plan9/unix or $PLAN9/unix directory.
 
 ===================================================================
 
