diff -Nurp guitarix2-0.21.0.orig/src/gx_head/gx_internal_plugins.cpp guitarix2-0.21.0/src/gx_head/gx_internal_plugins.cpp
--- guitarix2-0.21.0.orig/src/gx_head/gx_internal_plugins.cpp	2012-01-16 11:39:03.000000000 +0100
+++ guitarix2-0.21.0/src/gx_head/gx_internal_plugins.cpp	2012-01-17 10:22:46.128763526 +0100
@@ -24,7 +24,6 @@
  *
  * --------------------------------------------------------------------------
  */
-
 #include "engine.h"
 #include "gx_faust_support.h"
 
diff -Nurp guitarix2-0.21.0.orig/src/gx_head/gxw_mm_controllers.cpp guitarix2-0.21.0/src/gx_head/gxw_mm_controllers.cpp
--- guitarix2-0.21.0.orig/src/gx_head/gxw_mm_controllers.cpp	2012-01-16 11:33:22.000000000 +0100
+++ guitarix2-0.21.0/src/gx_head/gxw_mm_controllers.cpp	2012-01-17 11:31:32.946999158 +0100
@@ -160,7 +160,7 @@ UiSelector<T>::~UiSelector() {
 
 template <class T>
 void UiSelector<T>::on_value_changed() {
-    modifyZone(get_value());
+    this->modifyZone(get_value());
 }
 
 template <class T>
diff -Nurp guitarix2-0.21.0.orig/src/headers/engine.h guitarix2-0.21.0/src/headers/engine.h
--- guitarix2-0.21.0.orig/src/headers/engine.h	2011-11-22 17:48:43.000000000 +0100
+++ guitarix2-0.21.0/src/headers/engine.h	2012-01-17 10:22:32.953163490 +0100
@@ -35,6 +35,7 @@
 #include <glibmm/i18n.h>     // NOLINT
 #include <glibmm/optioncontext.h>   // NOLINT
 #include <glibmm/dispatcher.h>
+#include <glibmm/miscutils.h>
 #include <giomm/file.h>
 
 /* waf generated defines */
diff -Nurp guitarix2-0.21.0.orig/src/ladspa/ladspa_guitarix.cpp guitarix2-0.21.0/src/ladspa/ladspa_guitarix.cpp
--- guitarix2-0.21.0.orig/src/ladspa/ladspa_guitarix.cpp	2011-11-29 22:55:14.000000000 +0100
+++ guitarix2-0.21.0/src/ladspa/ladspa_guitarix.cpp	2012-01-17 11:58:00.002840732 +0100
@@ -3,6 +3,9 @@
 #include <ladspa.h>
 #include <iostream>
 
+#include <glibmm/init.h>
+#include <glibmm/thread.h>
+
 #include <jack/jack.h>
 #include <jack/thread.h>
 
