diff -aur libopensync-0.22/opensync/opensync_time.c libopensync-0.22.patched/opensync/opensync_time.c
--- libopensync-0.22/opensync/opensync_time.c	2007-03-27 13:49:18.000000000 +0200
+++ libopensync-0.22.patched/opensync/opensync_time.c	2011-07-16 10:27:49.528914597 +0200
@@ -889,16 +889,13 @@
 	osync_trace(TRACE_ENTRY, "%s(%s, %p)", __func__, vtime, tzid);
 
 	int year;
-	char *newyear = NULL;
-	time_t newyear_t, timestamp;
+	time_t timestamp;
 	struct tm *std_changetime, *dst_changetime;
 	time_t dstStamp, stdStamp;
 	xmlNode *current = NULL;
 
 	sscanf(vtime, "%4d%*2d%*2dT%*2d%*d%*2d%*c", &year);
 
-	newyear = g_strdup_printf("%4d0101T000000", year);
-	newyear_t = osync_time_vtime2unix(newyear, 0);
 	timestamp = osync_time_vtime2unix(vtime, 0);
 
 	/* Handle XML Timezone field */
diff -aur libopensync-0.22/tools/osynctest.c libopensync-0.22.patched/tools/osynctest.c
--- libopensync-0.22/tools/osynctest.c	2007-03-27 13:49:19.000000000 +0200
+++ libopensync-0.22.patched/tools/osynctest.c	2011-07-16 10:28:19.242782087 +0200
@@ -81,8 +81,6 @@
 double _second()     /* note that some compilers like AIX xlf do not require the trailing  '_' */
 {
     struct timeval tp;
-    int rtn;
-    rtn=gettimeofday(&tp, NULL);
 
     return ((double)tp.tv_sec+(1.e-6)*tp.tv_usec);
 }
