diff --git a/Makefile b/Makefile
index 955506c..c905d47 100644
--- a/Makefile
+++ b/Makefile
@@ -4,16 +4,16 @@ DEV_CFLAGS = -g -Wall -DELF_PLAYER -DELF_LINUX
 STA_CFLAGS = -Wall -O2 -DELF_PLAYER -DELF_LINUX
 SHR_CFLAGS = -fPIC -Wall -O2 -DELF_LINUX
 
-INCS = -Igfx -Ielf -I/usr/include/lua5.1 -I/usr/include/freetype2
+INCS = -Igfx -Ielf -I/usr/include/lua5.1 -I/usr/include/freetype2 -I/usr/include/bullet
 
 BLENDELF_LIBS = -lGL -lGLEW -lglfw -lXxf86vm -lXrandr -lXrender -pthread \
-	-lfreeimage -lvorbisfile -lvorbis -logg -lopenal -llua5.1 -lfreetype \
+	-lfreeimage -lvorbisfile -lvorbis -logg -lopenal -llua -lfreetype \
 	-lBulletDynamics -lLinearMath -lBulletCollision -lassimp
 
 BLENDELF_STATIC_LIBS = -lGL -lGLU /usr/lib/libGLEW.a /usr/lib/libglfw.a \
 	/usr/lib/libXxf86vm.a /usr/lib/libXrandr.a /usr/lib/libXrender.a \
 	/usr/lib/libvorbisfile.a /usr/lib/libvorbis.a /usr/lib/libogg.a \
-	/usr/lib/liblua5.1.a \
+	/usr/lib/liblua.a \
 	/usr/local/lib/libBulletDynamics.a \
 	/usr/local/lib/libLinearMath.a \
 	/usr/local/lib/libBulletCollision.a \
@@ -27,7 +27,7 @@ all:
 	gcc -c elf/scripting.c $(DEV_CFLAGS) $(INCS)
 	g++ -c elf/physics.cpp $(DEV_CFLAGS) $(INCS)
 	gcc -c elf/binds.c $(DEV_CFLAGS) $(INCS)
-	gcc -Wl,-rpath,linux_libraries -o blendelf *.o $(DEV_CFLAGS) $(BLENDELF_LIBS)
+	gcc -Wl,-rpath,linux_libraries -o blendelfd *.o $(DEV_CFLAGS) $(BLENDELF_LIBS)
 	rm *.o
 
 static:
@@ -38,7 +38,7 @@ static:
 	gcc -c elf/scripting.c $(STA_CFLAGS) $(INCS)
 	g++ -c elf/physics.cpp $(STA_CFLAGS) $(INCS)
 	gcc -c elf/binds.c $(STA_CFLAGS) $(INCS)
-	gcc -Wl,-rpath,linux_libraries -o blendelf *.o $(STA_CFLAGS) $(BLENDELF_STATIC_LIBS)
+	gcc -Wl,-rpath,linux_libraries -o blendelf *.o $(STA_CFLAGS) $(BLENDELF_LIBS)
 	rm *.o
 
 shared:
@@ -49,6 +49,6 @@ shared:
 	gcc -c -fPIC elf/scripting.c $(SHR_CFLAGS) $(INCS)
 	g++ -c -fPIC elf/physics.cpp $(SHR_CFLAGS) $(INCS)
 	gcc -c -fPIC elf/binds.c $(SHR_CFLAGS) $(INCS)
-	gcc -Wl,-rpath,linux_libraries -shared -o libblendelf.so *.o $(SHR_CFLAGS) $(BLENDELF_STATIC_LIBS)
+	gcc -Wl,-rpath,linux_libraries -shared -o libblendelf.so *.o $(SHR_CFLAGS) $(BLENDELF_LIBS)
 	rm *.o
 
diff --git a/genwraps.py b/genwraps.py
index 0424c66..aeb2bec 100644
--- a/genwraps.py
+++ b/genwraps.py
@@ -677,7 +677,7 @@ class Function:
 		for param in self.params:
 			dstr += param.type+'['+param.orig_type+'] '+param.name+', '
 		dstr += ')'
-		print dstr
+		print(dstr)
 
 def get_parts(line):
 	tstr = ''
