--- src/filesystem.c.orig	2009-01-22 17:29:16.000000000 -0500
+++ src/filesystem.c	2010-12-22 16:00:44.000000000 -0500
@@ -633,6 +633,10 @@
   return metalist;
 }
 
+void my_gtk_widget_destroy(GtkWidget *widget) {
+ if (GTK_IS_WIDGET(widget)) gtk_widget_destroy(widget);
+}
+
 static gpointer dirfill_thread(gpointer thread_args)
 {
   dirfill_thread_arg_t *args = (dirfill_thread_arg_t *) thread_args;
@@ -653,7 +657,7 @@
   metarg->metalist = metalist;
   g_idle_add(update_metadata,metarg);
 
-  g_idle_add((GSourceFunc) gtk_widget_destroy, args->dialog);
+  g_idle_add((GSourceFunc) my_gtk_widget_destroy, args->dialog);
   g_free(args->path);
   g_free(args);
   return NULL;
