--- ../gtkaml-0.5.91/src/Makefile.in	2011-12-31 12:49:33.000000000 +0200
+++ src/Makefile.in	2012-01-04 21:01:08.000000000 +0200
@@ -51,9 +51,9 @@
 nodist_gtkamlc_OBJECTS = $(am__objects_1)
 gtkamlc_OBJECTS = $(nodist_gtkamlc_OBJECTS)
 am__DEPENDENCIES_1 =
-gtkamlc_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
+gtkamlc_DEPENDENCIES = $(top_srcdir)/libgtkaml/libgtkaml.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
-	$(top_srcdir)/libgtkaml/libgtkaml.la
+	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 AM_V_lt = $(am__v_lt_$(V))
 am__v_lt_ = $(am__v_lt_$(AM_DEFAULT_VERBOSITY))
 am__v_lt_0 = --silent
@@ -235,11 +235,11 @@
 	$(gtkamlc_VALASOURCES:.vala=.c) 
 
 gtkamlc_LDADD = \
+	$(top_srcdir)/libgtkaml/libgtkaml.la \
 	$(GLIB_LIBS) \
 	$(GIO_LIBS) \
 	$(VALA_LIBS) \
-	$(LIBXML2_LIBS) \
-	$(top_srcdir)/libgtkaml/libgtkaml.la
+	$(LIBXML2_LIBS)
 
 EXTRA_DIST = $(gtkamlc_VALASOURCES)
 all: $(BUILT_SOURCES)
