diff -aur expyriment-0.6.3.orig/CHANGES.txt expyriment-0.6.3/CHANGES.txt
--- expyriment-0.6.3.orig/CHANGES.txt	2013-06-26 15:13:12.821359146 +0100
+++ expyriment-0.6.3/CHANGES.txt	2013-06-26 15:17:47.479869390 +0100
@@ -147,7 +147,7 @@
   - extra plugins can be now also included in the folder 
     $HOME/.expyriment/<module_name>/  (or ~expyriment)
 - command line interface: 
-  - see "python -m expyriment.cli -h" for help
+  - see "python2 -m expyriment.cli -h" for help
 - better on- and offline documentation:
   - new function: expyriment.show_documentation()
   - new Api Reference Tool (API browse and search GUI)
diff -aur expyriment-0.6.3.orig/documentation/html/Examples.html expyriment-0.6.3/documentation/html/Examples.html
--- expyriment-0.6.3.orig/documentation/html/Examples.html	2013-06-26 15:13:13.191366163 +0100
+++ expyriment-0.6.3/documentation/html/Examples.html	2013-06-26 15:17:48.793228447 +0100
@@ -65,7 +65,7 @@
 <h2>Simon task<a class="headerlink" href="#simon-task" title="Permalink to this headline">¶</a></h2>
 <p>An experiment to asses a spatial stimulus-response compatibility effect (see
 <a class="reference external" href="http://en.wikipedia.org/wiki/Simon_effect">wikipedia</a>).</p>
-<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python</span>
+<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python2</span>
 
 <span class="sd">&quot;&quot;&quot;</span>
 <span class="sd">A simple behavioural task to asses a Simon effect.</span>
@@ -131,7 +131,7 @@
 <h2>Word fragment completion task<a class="headerlink" href="#word-fragment-completion-task" title="Permalink to this headline">¶</a></h2>
 <p>Task as used for instance in <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/1829476">Weldon, 1991</a>.
 The script read in a stimulus list file (<a class="reference download internal" href="_downloads/word_fragment_completion_stimuluslist.csv"><tt class="xref download docutils literal"><span class="pre">demo</span> <span class="pre">stimulus</span> <span class="pre">list</span></tt></a>).</p>
-<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python</span>
+<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python2</span>
 
 <span class="sd">&quot;&quot;&quot;</span>
 <span class="sd">Word fragment completion task as used in the study of Weldon (1991).</span>
@@ -209,7 +209,7 @@
 <p>A full experiment to access SNARC and SNARC-like effects in a number and a
 letter classification task (e.g., <a class="reference external" href="http://dx.doi.org/10.1016/S0010-0277(02)00234-2">Gevers, Reynvoer, &amp; Fias (2003)</a>) with two response
 mappings, error feedback and between-subject factors.</p>
-<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python</span>
+<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python2</span>
 
 <span class="sd">&quot;&quot;&quot;</span>
 <span class="sd">A parity judgment task to assess the SNARC effect.</span>
@@ -344,7 +344,7 @@
 <h2>Really short example<a class="headerlink" href="#really-short-example" title="Permalink to this headline">¶</a></h2>
 <p>Expyriment is efficient!. See here a very short example of an functioning
 experiment in less than 20 lines of pure code.</p>
-<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python</span>
+<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python2</span>
 <span class="sd">&quot;&quot;&quot;</span>
 <span class="sd">A very short example experiment in 16 lines of pure code.</span>
 
@@ -380,7 +380,7 @@
 <div class="section" id="data-preprocessing">
 <h2>Data preprocessing<a class="headerlink" href="#data-preprocessing" title="Permalink to this headline">¶</a></h2>
 <p>Preprocessing the data of the SNARC experiment for further statistical analysis.</p>
-<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python</span>
+<div class="highlight-python"><div class="highlight"><pre><span class="c">#!/usr/bin/env python2</span>
 
 <span class="sd">&quot;&quot;&quot;</span>
 <span class="sd">Example analysis script for snarc_experiment.py</span>
diff -aur expyriment-0.6.3.orig/documentation/html/_sources/Tools.txt expyriment-0.6.3/documentation/html/_sources/Tools.txt
--- expyriment-0.6.3.orig/documentation/html/_sources/Tools.txt	2013-06-26 15:13:13.191366163 +0100
+++ expyriment-0.6.3/documentation/html/_sources/Tools.txt	2013-06-26 15:17:48.619891719 +0100
@@ -76,7 +76,7 @@
 
 Usage::
 
-    python -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]
+    python2 -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]
 
 .. FIXME better docu command line interface
 
diff -aur expyriment-0.6.3.orig/documentation/html/_sources/Tutorial.txt expyriment-0.6.3/documentation/html/_sources/Tutorial.txt
--- expyriment-0.6.3.orig/documentation/html/_sources/Tutorial.txt	2013-06-26 15:13:13.191366163 +0100
+++ expyriment-0.6.3/documentation/html/_sources/Tutorial.txt	2013-06-26 15:17:48.659892503 +0100
@@ -24,7 +24,7 @@
 Now run the file by either double clicking on it (if you are on Windows) or by 
 typing the following into a command line:
     
-    ``python first_experiment.py``
+    ``python2 first_experiment.py``
 
 The following should happen:
 
diff -aur expyriment-0.6.3.orig/documentation/html/Tools.html expyriment-0.6.3/documentation/html/Tools.html
--- expyriment-0.6.3.orig/documentation/html/Tools.html	2013-06-26 15:13:13.138031818 +0100
+++ expyriment-0.6.3/documentation/html/Tools.html	2013-06-26 15:17:48.296552053 +0100
@@ -148,7 +148,7 @@
 <h2>Command line interface<a class="headerlink" href="#command-line-interface" title="Permalink to this headline">¶</a></h2>
 <p>The command line interface can be used to start the Expyriment Reference Tool, or to run the Expyriment test suite or any Python script with predefined Expyriment default settings.</p>
 <p>Usage:</p>
-<div class="highlight-python"><pre>python -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]</pre>
+<div class="highlight-python"><pre>python2 -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]</pre>
 </div>
 </div>
 <div class="section" id="import-expyriment-data-into-r">
diff -aur expyriment-0.6.3.orig/documentation/sphinx/create_rst_api_reference.py expyriment-0.6.3/documentation/sphinx/create_rst_api_reference.py
--- expyriment-0.6.3.orig/documentation/sphinx/create_rst_api_reference.py	2013-06-26 15:13:13.194699559 +0100
+++ expyriment-0.6.3/documentation/sphinx/create_rst_api_reference.py	2013-06-26 15:17:49.083234128 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 make rst files for the expyriment API reference
diff -aur expyriment-0.6.3.orig/documentation/sphinx/examples/really_short_exp.py expyriment-0.6.3/documentation/sphinx/examples/really_short_exp.py
--- expyriment-0.6.3.orig/documentation/sphinx/examples/really_short_exp.py	2013-06-26 15:13:13.194699559 +0100
+++ expyriment-0.6.3/documentation/sphinx/examples/really_short_exp.py	2013-06-26 15:17:49.063233737 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 A very short example experiment in 16 lines of pure code.
 
diff -aur expyriment-0.6.3.orig/documentation/sphinx/examples/simon_task.py expyriment-0.6.3/documentation/sphinx/examples/simon_task.py
--- expyriment-0.6.3.orig/documentation/sphinx/examples/simon_task.py	2013-06-26 15:13:13.194699559 +0100
+++ expyriment-0.6.3/documentation/sphinx/examples/simon_task.py	2013-06-26 15:17:49.066567136 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A simple behavioural task to asses a Simon effect.
diff -aur expyriment-0.6.3.orig/documentation/sphinx/examples/snarc_data_preprocessing.py expyriment-0.6.3/documentation/sphinx/examples/snarc_data_preprocessing.py
--- expyriment-0.6.3.orig/documentation/sphinx/examples/snarc_data_preprocessing.py	2013-06-26 15:13:13.194699559 +0100
+++ expyriment-0.6.3/documentation/sphinx/examples/snarc_data_preprocessing.py	2013-06-26 15:17:49.079900728 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Example analysis script for snarc_experiment.py
diff -aur expyriment-0.6.3.orig/documentation/sphinx/examples/snarc_experiment.py expyriment-0.6.3/documentation/sphinx/examples/snarc_experiment.py
--- expyriment-0.6.3.orig/documentation/sphinx/examples/snarc_experiment.py	2013-06-26 15:13:13.194699559 +0100
+++ expyriment-0.6.3/documentation/sphinx/examples/snarc_experiment.py	2013-06-26 15:17:49.073233929 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A parity judgment task to assess the SNARC effect.
diff -aur expyriment-0.6.3.orig/documentation/sphinx/examples/word_fragment_completion_task.py expyriment-0.6.3/documentation/sphinx/examples/word_fragment_completion_task.py
--- expyriment-0.6.3.orig/documentation/sphinx/examples/word_fragment_completion_task.py	2013-06-26 15:13:13.194699559 +0100
+++ expyriment-0.6.3/documentation/sphinx/examples/word_fragment_completion_task.py	2013-06-26 15:17:49.076567328 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Word fragment completion task as used in the study of Weldon (1991).
diff -aur expyriment-0.6.3.orig/documentation/sphinx/Tools.rst expyriment-0.6.3/documentation/sphinx/Tools.rst
--- expyriment-0.6.3.orig/documentation/sphinx/Tools.rst	2013-06-26 15:13:13.194699559 +0100
+++ expyriment-0.6.3/documentation/sphinx/Tools.rst	2013-06-26 15:17:48.973231973 +0100
@@ -76,7 +76,7 @@
 
 Usage::
 
-    python -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]
+    python2 -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]
 
 .. FIXME better docu command line interface
 
diff -aur expyriment-0.6.3.orig/documentation/sphinx/Tutorial.rst expyriment-0.6.3/documentation/sphinx/Tutorial.rst
--- expyriment-0.6.3.orig/documentation/sphinx/Tutorial.rst	2013-06-26 15:13:13.194699559 +0100
+++ expyriment-0.6.3/documentation/sphinx/Tutorial.rst	2013-06-26 15:17:49.039899946 +0100
@@ -24,7 +24,7 @@
 Now run the file by either double clicking on it (if you are on Windows) or by 
 typing the following into a command line:
     
-    ``python first_experiment.py``
+    ``python2 first_experiment.py``
 
 The following should happen:
 
diff -aur expyriment-0.6.3.orig/examples/really_short_exp.py expyriment-0.6.3/examples/really_short_exp.py
--- expyriment-0.6.3.orig/examples/really_short_exp.py	2013-06-26 15:13:12.961361801 +0100
+++ expyriment-0.6.3/examples/really_short_exp.py	2013-06-26 15:17:47.853210036 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 A very short example experiment in 16 lines of pure code.
 
diff -aur expyriment-0.6.3.orig/examples/simon_task.py expyriment-0.6.3/examples/simon_task.py
--- expyriment-0.6.3.orig/examples/simon_task.py	2013-06-26 15:13:12.961361801 +0100
+++ expyriment-0.6.3/examples/simon_task.py	2013-06-26 15:17:47.856543435 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A simple behavioural task to asses a Simon effect.
diff -aur expyriment-0.6.3.orig/examples/snarc_data_preprocessing.py expyriment-0.6.3/examples/snarc_data_preprocessing.py
--- expyriment-0.6.3.orig/examples/snarc_data_preprocessing.py	2013-06-26 15:13:12.961361801 +0100
+++ expyriment-0.6.3/examples/snarc_data_preprocessing.py	2013-06-26 15:17:47.869877029 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Example analysis script for snarc_experiment.py
diff -aur expyriment-0.6.3.orig/examples/snarc_experiment.py expyriment-0.6.3/examples/snarc_experiment.py
--- expyriment-0.6.3.orig/examples/snarc_experiment.py	2013-06-26 15:13:12.961361801 +0100
+++ expyriment-0.6.3/examples/snarc_experiment.py	2013-06-26 15:17:47.863210232 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A parity judgment task to assess the SNARC effect.
diff -aur expyriment-0.6.3.orig/examples/word_fragment_completion_task.py expyriment-0.6.3/examples/word_fragment_completion_task.py
--- expyriment-0.6.3.orig/examples/word_fragment_completion_task.py	2013-06-26 15:13:12.961361801 +0100
+++ expyriment-0.6.3/examples/word_fragment_completion_task.py	2013-06-26 15:17:47.866543630 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Word fragment completion task as used in the study of Weldon (1991).
diff -aur expyriment-0.6.3.orig/expyriment/_api_reference_tool.py expyriment-0.6.3/expyriment/_api_reference_tool.py
--- expyriment-0.6.3.orig/expyriment/_api_reference_tool.py	2013-06-26 15:13:12.834692732 +0100
+++ expyriment-0.6.3/expyriment/_api_reference_tool.py	2013-06-26 15:17:47.516536775 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 The Expyriment documentation
diff -aur expyriment-0.6.3.orig/expyriment/cli.py expyriment-0.6.3/expyriment/cli.py
--- expyriment-0.6.3.orig/expyriment/cli.py	2013-06-26 15:13:12.888027077 +0100
+++ expyriment-0.6.3/expyriment/cli.py	2013-06-26 15:17:47.563204355 +0100
@@ -1,11 +1,11 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """A command line interface for Expyriment.
 
 Use this to start the Expyriment Reference Tool, or to run the Expyriment Test
 Suite or any Python script with predefined Expyriment default settings.
 
-Usage: python -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]
+Usage: python2 -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]
 
 OPTIONS:
       -g              No OpenGL
@@ -74,7 +74,7 @@
                         sys.exit()
                     elif arg == 'h':
                             print """
-Usage: python -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]
+Usage: python2 -m expyriment.cli [EXPYRIMENT SCRIPT] [OPTIONS]
 
     OPTIONS:
           -g              No OpenGL
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_blankscreen.py expyriment-0.6.3/expyriment/stimuli/_blankscreen.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_blankscreen.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_blankscreen.py	2013-06-26 15:17:47.746541279 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A blank screen stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_canvas.py expyriment-0.6.3/expyriment/stimuli/_canvas.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_canvas.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_canvas.py	2013-06-26 15:17:47.743207882 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A Canvas stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_circle.py expyriment-0.6.3/expyriment/stimuli/_circle.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_circle.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_circle.py	2013-06-26 15:17:47.769875070 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A circle stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_dot.py expyriment-0.6.3/expyriment/stimuli/_dot.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_dot.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_dot.py	2013-06-26 15:17:47.773208468 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A dot stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_ellipse.py expyriment-0.6.3/expyriment/stimuli/_ellipse.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_ellipse.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_ellipse.py	2013-06-26 15:17:47.776541867 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 An ellipse stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_dotcloud.py expyriment-0.6.3/expyriment/stimuli/extras/_dotcloud.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_dotcloud.py	2013-06-26 15:13:12.938028025 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_dotcloud.py	2013-06-26 15:17:47.689873504 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A dotcloud stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_lcdsymbol.py expyriment-0.6.3/expyriment/stimuli/extras/_lcdsymbol.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_lcdsymbol.py	2013-06-26 15:13:12.938028025 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_lcdsymbol.py	2013-06-26 15:17:47.686540105 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A LCD symbol.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_noisetone.py expyriment-0.6.3/expyriment/stimuli/extras/_noisetone.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_noisetone.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_noisetone.py	2013-06-26 15:17:47.719874092 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 The noise tone stimulus module.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_polygondot.py expyriment-0.6.3/expyriment/stimuli/extras/_polygondot.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_polygondot.py	2013-06-26 15:13:12.938028025 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_polygondot.py	2013-06-26 15:17:47.673206511 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A dot stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_polygonellipse.py expyriment-0.6.3/expyriment/stimuli/extras/_polygonellipse.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_polygonellipse.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_polygonellipse.py	2013-06-26 15:17:47.713207293 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 An ellipse stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_polygonline.py expyriment-0.6.3/expyriment/stimuli/extras/_polygonline.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_polygonline.py	2013-06-26 15:13:12.938028025 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_polygonline.py	2013-06-26 15:17:47.669873113 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A Line stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_polygonrectangle.py expyriment-0.6.3/expyriment/stimuli/extras/_polygonrectangle.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_polygonrectangle.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_polygonrectangle.py	2013-06-26 15:17:47.723207491 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A Rectangle stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_stimuluscircle.py expyriment-0.6.3/expyriment/stimuli/extras/_stimuluscircle.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_stimuluscircle.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_stimuluscircle.py	2013-06-26 15:17:47.706540497 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A stimulus circle stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_stimuluscloud.py expyriment-0.6.3/expyriment/stimuli/extras/_stimuluscloud.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_stimuluscloud.py	2013-06-26 15:13:12.938028025 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_stimuluscloud.py	2013-06-26 15:17:47.669873113 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A stimulus cloud stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/extras/_visualmask.py expyriment-0.6.3/expyriment/stimuli/extras/_visualmask.py
--- expyriment-0.6.3.orig/expyriment/stimuli/extras/_visualmask.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/extras/_visualmask.py	2013-06-26 15:17:47.699873699 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A Visual Mask.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_fixcross.py expyriment-0.6.3/expyriment/stimuli/_fixcross.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_fixcross.py	2013-06-26 15:13:12.931361233 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_fixcross.py	2013-06-26 15:17:47.659872914 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A fixation cross stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_frame.py expyriment-0.6.3/expyriment/stimuli/_frame.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_frame.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_frame.py	2013-06-26 15:17:47.756541476 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A frame stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_line.py expyriment-0.6.3/expyriment/stimuli/_line.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_line.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_line.py	2013-06-26 15:17:47.806542454 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A Line stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_picture.py expyriment-0.6.3/expyriment/stimuli/_picture.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_picture.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_picture.py	2013-06-26 15:17:47.799875659 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A picture stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_rectangle.py expyriment-0.6.3/expyriment/stimuli/_rectangle.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_rectangle.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_rectangle.py	2013-06-26 15:17:47.803209056 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A Rectangle stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_shape.py expyriment-0.6.3/expyriment/stimuli/_shape.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_shape.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_shape.py	2013-06-26 15:17:47.796542260 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A Shape stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_stimulus.py expyriment-0.6.3/expyriment/stimuli/_stimulus.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_stimulus.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_stimulus.py	2013-06-26 15:17:47.736541085 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 This module contains the base classes for stimuli.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_textbox.py expyriment-0.6.3/expyriment/stimuli/_textbox.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_textbox.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_textbox.py	2013-06-26 15:17:47.739874484 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A text box stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_textline.py expyriment-0.6.3/expyriment/stimuli/_textline.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_textline.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_textline.py	2013-06-26 15:17:47.763208275 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A text line stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_textscreen.py expyriment-0.6.3/expyriment/stimuli/_textscreen.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_textscreen.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_textscreen.py	2013-06-26 15:17:47.766541673 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 A text screen stimulus.
diff -aur expyriment-0.6.3.orig/expyriment/stimuli/_tone.py expyriment-0.6.3/expyriment/stimuli/_tone.py
--- expyriment-0.6.3.orig/expyriment/stimuli/_tone.py	2013-06-26 15:13:12.941361422 +0100
+++ expyriment-0.6.3/expyriment/stimuli/_tone.py	2013-06-26 15:17:47.753208076 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 The tone stimulus module.
diff -aur expyriment-0.6.3.orig/setup.py expyriment-0.6.3/setup.py
--- expyriment-0.6.3.orig/setup.py	2013-06-26 15:13:13.194699559 +0100
+++ expyriment-0.6.3/setup.py	2013-06-26 15:17:49.296571638 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 
 import stat
