Index: trunk/Tools/qmake/mkspecs/features/functions.prf
===================================================================
--- trunk/Tools/qmake/mkspecs/features/functions.prf	(revision 150222)
+++ trunk/Tools/qmake/mkspecs/features/functions.prf	(revision 150223)
@@ -205,5 +205,5 @@
 
 defineTest(programExistsInPath) {
-    win_cmd_shell: program = $${1}.exe
+    win32: program = $${1}.exe
     else: program = $$1
 
