diff -aur a/configure b/configure
--- a/configure	2012-01-08 18:12:37.606881465 -0800
+++ b/configure	2012-01-08 18:14:12.833542637 -0800
@@ -287,7 +287,7 @@
 
 # --- QT4 START ---
 QT4_IM='N'
-QT4_IM_DIR=/usr/$LIB/qt4/plugins/inputmethods
+QT4_IM_DIR=/usr/$LIB/qt/plugins/inputmethods
 echo -n ".... Testing Qt 4.x      :  "
 pkg-config --libs QtCore QtGui > /dev/null 2> /dev/null
 if [ $? = 0 ]; then
diff -aur a/src/qt4-im/Makefile b/src/qt4-im/Makefile
--- a/src/qt4-im/Makefile	2012-01-08 18:12:39.846881337 -0800
+++ b/src/qt4-im/Makefile	2012-01-08 18:13:49.563543979 -0800
@@ -1,5 +1,5 @@
 include ../../config.mak
-QT=qt4
+QT=qt
 QTIM=$(QT)/plugins/inputmethods
 IMMODULES=$(libdir)/$(QTIM)
 
