Index: sundials-2.5.0/sundialsTB/install_STB.m
===================================================================
--- sundials-2.5.0.orig/sundialsTB/install_STB.m	2012-03-21 23:47:00.000000000 +0100
+++ sundials-2.5.0/sundialsTB/install_STB.m	2012-06-07 10:47:47.000000000 +0200
@@ -9,7 +9,7 @@
 % MEX compiler command
 % --------------------
 
-mexcompiler = 'mex -v';
+mexcompiler = 'mkoctfile --mex -v';
 
 % Location of sundialsTB and top of sundials source tree
 % ------------------------------------------------------
@@ -46,48 +46,18 @@
 % Figure out what modules exist and which ones will be built
 % ----------------------------------------------------------
 
-fprintf('\n\nSelect modules to be built\n');
+fprintf('\n\nBuild Modules for CVODES, IDAS, KINSOL\n');
 
-q = fullfile(sun,'src','cvodes');
-if exist(q, 'dir')
-  answ = input('    Compile CVODES interface? (y/n) ','s');
-  if answ == 'y'
-    cvm_ok = true;
-  else
-    cvm_ok = false;
-  end
-end
-
-if exist(q, 'dir')
-  answ = input('    Compile IDAS interface? (y/n) ','s');
-  if answ == 'y'
-    idm_ok = true;
-  else
-    idm_ok = false;
-  end
-end
-
-q = fullfile(sun,'src','kinsol');
-if exist(q, 'dir')
-  answ = input('    Compile KINSOL interface? (y/n) ','s');
-  if answ == 'y'
-    kim_ok = true;
-  else
-    kim_ok = false;
-  end
-end
-
-if ~cvm_ok && ~idm_ok && ~kim_ok
-  fprintf('\nOK. All done.\n');
-  return
-end
+cvm_ok = true;
+idm_ok = true;
+kim_ok = true;
 
 % Create sundials_config.h
 % ------------------------
 
 mkdir('sundials');
 fi = fopen(fullfile('sundials','sundials_config.h'),'w');
-fprintf(fi,'#define SUNDIALS_PACKAGE_VERSION "2.4.0"\n');
+fprintf(fi,'#define SUNDIALS_PACKAGE_VERSION "2.5.0"\n');
 fprintf(fi,'#define SUNDIALS_DOUBLE_PRECISION 1\n');
 fprintf(fi,'#define SUNDIALS_USE_GENERIC_MATH 1\n');
 fprintf(fi,'#define SUNDIALS_EXPORT\n');
@@ -117,17 +87,15 @@
 % ------------------
 
 fprintf('\n\nMEX files were successfully created.\n');
-answ = input('    Install toolbox? (y/n) ','s');
-if answ ~= 'y'
-  fprintf('\n\nOK. All done.\n');
-  return
-end
+fprintf('\n\nInstall toolbox.\n');
+answ = 'y'
 
 while true
   fprintf('\n\nSpecify the location where you wish to install the toolbox.\n');
   fprintf('The toolbox will be installed in a subdirectory "sundialsTB".\n');
   fprintf('Enter return to cancel the installation.\n');
-  where = input('    Installation directory: ','s');
+  %where = input('    Installation directory: ','s');
+  where = './octave';
   if isempty(where)
     go = 0;
     break;
@@ -149,7 +117,8 @@
 go = 1;
 if exist(stbi,'dir')
   fprintf('\n\nDirectory %s exists!\n',stbi);
-  answ = input('    Replace? (y/n) ','s');
+  %answ = input('    Replace? (y/n) ','s');
+  answ = 'y';
   if answ == 'y'
     rmdir(stbi,'s');
     go = 1;
@@ -210,11 +179,14 @@
 fprintf('add a call to %s\n',fullfile(stbi,'startup_STB.m'));
 fprintf('\nEnjoy!\n\n');
 
+exit
+endfunction
+
 %---------------------------------------------------------------------------------
 % Check if mex works and if the user accepts the current mexopts
 %---------------------------------------------------------------------------------
 
-function mex_ok = check_mex(mexcompiler)
+function mex_ok = check_mex (mexcompiler)
 
 % Create a dummy file
 fid = fopen('foo.c', 'w');
@@ -232,7 +204,8 @@
 delete(sprintf('foo.%s', mexext))
 
 fprintf('\n\nMEX files will be compiled and built using the above options\n');
-answ = input('    Proceed? (y/n) ','s');
+%answ = input('    Proceed? (y/n) ','s');
+answ = 'y';
 if answ == 'y'
   mex_ok = true;
 else
@@ -240,7 +213,7 @@
   mex_ok = false;
 end
 
-return
+endfunction
 
 %---------------------------------------------------------------------------------
 % compilation of cvm MEX file
@@ -352,6 +325,8 @@
 
 cd(stb)
 
+endfunction
+
 %---------------------------------------------------------------------------------
 % compilation of idm MEX file
 %---------------------------------------------------------------------------------
@@ -461,6 +436,7 @@
 
 cd(stb)
 
+endfunction
 
 %---------------------------------------------------------------------------------
 % compilation of KINSOL MEX file
@@ -569,6 +545,8 @@
 
 cd(stb)
 
+endfunction
+
 %---------------------------------------------------------------------------------
 % Installation of common sundialsTB files
 %---------------------------------------------------------------------------------
@@ -633,6 +611,7 @@
     break;
   end
 end
+endfunction
 
 %---------------------------------------------------------------------------------
 % Installation of CVODES files
@@ -795,6 +774,8 @@
   disp(msg);
 end
 
+endfunction
+
 %---------------------------------------------------------------------------------
 % Installation of IDAS files
 %---------------------------------------------------------------------------------
@@ -949,7 +930,7 @@
   disp(msg);
 end
 
-
+endfunction
 
 %---------------------------------------------------------------------------------
 % Installation of KINSOL files
@@ -1023,3 +1004,6 @@
 end
 
 
+endfunction
+
+install_STB;
