diff -Nurd jutoh-2.23/jutoh2.desktop jutoh-2.23.patched/jutoh2.desktop
--- jutoh-2.23/jutoh2.desktop	2014-10-15 20:11:28.000000000 +0300
+++ jutoh-2.23.patched/jutoh2.desktop	2014-10-16 16:21:35.377684640 +0300
@@ -5,8 +5,8 @@
 Terminal=false
 Exec=jutoh2 %f
 TryExec=jutoh2
-Icon[en_GB]=jutoh2.png
-Icon=jutoh2.png
+Icon[en_GB]=jutoh
+Icon=jutoh
 Name[en_GB]=Jutoh 2
 Name=Jutoh 2
 GenericName=Ebook Editor
diff -Nurd jutoh-2.23/jutoh2.keys jutoh-2.23.patched/jutoh2.keys
--- jutoh-2.23/jutoh2.keys	2014-10-15 20:11:28.000000000 +0300
+++ jutoh-2.23.patched/jutoh2.keys	2014-10-16 16:22:13.227494357 +0300
@@ -1,7 +1,7 @@
 application/x-jutohproject
         default_action_type=application
         description=Jutoh Project
-        icon_filename=jutoh2.png
+        icon_filename=jutoh-document.png
         short_list_application_ids_for_novice_user=Jutoh
         short_list_application_ids_for_intermediat_user=Jutoh
         short_list_application_ids_for_advanced_user=Jutoh
@@ -11,7 +11,7 @@
 application/x-jutohscript
         default_action_type=application
         description=Jutoh Script
-        icon_filename=jutoh2.png
+        icon_filename=jutoh-document-script.png
         short_list_application_ids_for_novice_user=Jutoh
         short_list_application_ids_for_intermediat_user=Jutoh
         short_list_application_ids_for_advanced_user=Jutoh
diff -Nurd jutoh-2.23/jutoh2.xml jutoh-2.23.patched/jutoh2.xml
--- jutoh-2.23/jutoh2.xml	2014-10-15 20:11:28.000000000 +0300
+++ jutoh-2.23.patched/jutoh2.xml	2014-10-16 16:23:03.920572768 +0300
@@ -3,6 +3,7 @@
    <mime-type type="application/x-jutohproject">
      <comment>Jutoh 2 Project</comment>
      <glob pattern="*.jutoh"/>
+     <icon name="jutoh-document"/>
      <sub-class-of type="application/x-sqlite3"/>
      <magic priority="80">
        <match type="string" offset="0" value="SQLite format 3"/>
@@ -11,5 +12,6 @@
    <mime-type type="application/x-jutohscript">
      <comment>Jutoh Script</comment>
      <glob pattern="*.jutohscript"/>
+     <icon name="jutoh-document-script"/>
    </mime-type>
 </mime-info>
