diff -Nurd JutohData/jutoh.desktop jutoh-1.71.patched/jutoh.desktop
--- JutohData/jutoh.desktop	2013-07-20 18:22:08.000000000 +0300
+++ jutoh-1.71.patched/jutoh.desktop	2013-10-26 13:34:59.414361241 +0300
@@ -5,8 +5,8 @@
 Terminal=false
 Exec=jutoh %f
 TryExec=jutoh
-Icon[en_GB]=jutoh.png
-Icon=jutoh.png
+Icon[en_GB]=jutoh
+Icon=jutoh
 Name[en_GB]=Jutoh
 Name=Jutoh
 GenericName=Ebook Editor
diff -Nurd JutohData/jutoh.keys jutoh-1.71.patched/jutoh.keys
--- JutohData/jutoh.keys	2013-07-20 18:22:08.000000000 +0300
+++ jutoh-1.71.patched/jutoh.keys	2013-10-26 13:34:59.414361241 +0300
@@ -1,7 +1,7 @@
 application/x-jutohproject
         default_action_type=application
         description=Jutoh Project
-        icon_filename=jutoh.png
+        icon_filename=jutoh-document.png
         short_list_application_ids_for_novice_user=Jutoh
         short_list_application_ids_for_intermediat_user=Jutoh
         short_list_application_ids_for_advanced_user=Jutoh
@@ -11,7 +11,7 @@
 application/x-jutohscript
         default_action_type=application
         description=Jutoh Script
-        icon_filename=jutoh.png
+        icon_filename=jutoh-document-script.png
         short_list_application_ids_for_novice_user=Jutoh
         short_list_application_ids_for_intermediat_user=Jutoh
         short_list_application_ids_for_advanced_user=Jutoh
diff -Nurd JutohData/jutoh.xml jutoh-1.71.patched/jutoh.xml
--- JutohData/jutoh.xml	2013-07-20 18:22:08.000000000 +0300
+++ jutoh-1.71.patched/jutoh.xml	2013-10-26 13:37:22.176869725 +0300
@@ -3,6 +3,7 @@
    <mime-type type="application/x-jutohproject">
      <comment>Jutoh Project</comment>
      <glob pattern="*.jutoh"/>
+     <icon name="jutoh-document"/>
      <sub-class-of type="application/x-sqlite3"/>
      <magic priority="80">
        <match type="string" offset="0" value="SQLite format 3"/>
@@ -11,5 +12,6 @@
    <mime-type type="application/x-jutohscript">
      <comment>Jutoh Script</comment>
      <glob pattern="*.jutohscript"/>
+     <icon name="jutoh-document-script"/>
    </mime-type>
 </mime-info>
