--- lib-src/enigma-core/ecl_dict.hh.old	2011-04-29 08:41:47.649582258 +0200
+++ lib-src/enigma-core/ecl_dict.hh	2011-04-29 08:42:05.404691426 +0200
@@ -21,6 +21,7 @@
 
 #include "ecl_error.hh"
 #include <utility>
+#include <cstddef>
 
 namespace ecl
 {
--- lib-src/zipios++/src/directory.h.old	2011-04-29 08:40:31.063798354 +0200
+++ lib-src/zipios++/src/directory.h	2011-04-29 08:40:52.656147037 +0200
@@ -31,6 +31,7 @@
 #include <string>
 #include <ctime>
 #include <stdexcept>
+#include <cstddef>
 
 // #include <boost.h>  Contents of boost.h
 
