--- KCemu-0.5.1.orig/src/libdisk/load_22dsk.c	2010-03-08 00:50:22.000000000 +0500
+++ KCemu-0.5.1/src/libdisk/load_22dsk.c	2012-05-20 16:14:44.308167308 +0600
@@ -81,7 +81,7 @@
 
 typedef struct {
   FILE *f;
-  gzFile *gzf;
+  gzFile gzf;
   int read_only;
   int len;
   struct {
--- KCemu-0.5.1.orig/src/ui/gtk/plotter.cc	2010-03-08 00:50:22.000000000 +0500
+++ KCemu-0.5.1/src/ui/gtk/plotter.cc	2012-05-20 16:22:00.711503108 +0600
@@ -18,7 +18,7 @@
  */
 
 #include <cairo/cairo.h>
-#include <glib-2.0/glib/gmain.h>
+#include <glib.h>
 #include <gtk-2.0/gtk/gtkspinbutton.h>
 
 #include "kc/kc.h"
