diff -x .git -urN linux-3.8/arch/x86/Kconfig.cpu zen/arch/x86/Kconfig.cpu
--- linux-3.8/arch/x86/Kconfig.cpu	2013-02-19 01:58:34.000000000 +0200
+++ zen/arch/x86/Kconfig.cpu	2013-03-29 10:56:39.214608089 +0200
@@ -290,6 +290,17 @@
 	  This is really intended for distributors who need more
 	  generic optimizations.
 
+config X86_MARCH_NATIVE
+	bool "Use -march=native cflag (EXPERIMENTAL)"
+	depends on EXPERIMENTAL
+	help
+	  Setting Y here, will result in passing the -march=native and
+	  -mtune=native cflags to GCC while compiling the kernel, which
+	  makes GCC check the CPU capabilities and use the best cflags
+	  for your computer.
+
+	  Set Y here only if you use >=gcc-4.2.0.
+
 #
 # Define implied options from the CPU selection here
 config X86_INTERNODE_CACHE_SHIFT
diff -x .git -urN linux-3.8/arch/x86/Makefile_32.cpu zen/arch/x86/Makefile_32.cpu
--- linux-3.8/arch/x86/Makefile_32.cpu	2013-02-19 01:58:34.000000000 +0200
+++ zen/arch/x86/Makefile_32.cpu	2013-03-29 10:56:39.214608089 +0200
@@ -10,6 +10,9 @@
 endif
 
 align := $(cc-option-align)
+ifeq ($(CONFIG_X86_MARCH_NATIVE),y)
+cflags-y			+= -march=native
+else
 cflags-$(CONFIG_M486)		+= -march=i486
 cflags-$(CONFIG_M586)		+= -march=i586
 cflags-$(CONFIG_M586TSC)	+= -march=i586
@@ -41,6 +44,8 @@
 # Geode GX1 support
 cflags-$(CONFIG_MGEODEGX1)	+= -march=pentium-mmx
 cflags-$(CONFIG_MGEODE_LX)	+= $(call cc-option,-march=geode,-march=pentium-mmx)
+endif
+
 # add at the end to overwrite eventual tuning options from earlier
 # cpu entries
 cflags-$(CONFIG_X86_GENERIC) 	+= $(call tune,generic,$(call tune,i686))
