--- linux-3.18/kernel/time/posix-cpu-timers.c	2014-04-15 09:06:17.904901803 -0400
+++ linux-3.18/kernel/time/posix-cpu-timers.c	2014-04-16 21:55:55.305243142 -0400
@@ -1338,7 +1338,7 @@
 		p->flags |= PF_NOFREEZE;
 		kthread_bind(p, cpu);
 		/* Must be high prio to avoid getting starved */
-		param.sched_priority = MAX_RT_PRIO-1;
+		param.sched_priority = MAX_RT_PRIO-2;
 		sched_setscheduler(p, SCHED_FIFO, &param);
 		per_cpu(posix_timer_task,cpu) = p;
 		break;
