--- pyhalebarde-0.3.orig/halebarde/boardwidget.py	2009-07-01 20:58:03.000000000 +0600
+++ pyhalebarde-0.3/halebarde/boardwidget.py	2012-05-04 07:36:29.000000000 +0600
@@ -86,8 +86,9 @@
         self.stage.connect('button-press-event', self.on_button_press_event)
 
     def on_button_press_event(self, actor, event):
-        print self.screen_to_game(event.x, event.y)
-        self.clic((event.x, event.y), event.button)
+	x, y = int(event.x), int(event.y)
+        print self.screen_to_game(x, y)
+        self.clic((x, y), event.button)
 
     def game_to_screen(self, *args):
         "resize and re-position in game positions to screen position"
@@ -100,7 +101,7 @@
     def draw_background (self):
         #TODO clean this shit. move to a class?
         import cairo
-        from cluttercairo import CairoTexture
+        from clutter import CairoTexture
         background_surface = CairoTexture(*self.screen_size)
         self.stage.add(background_surface)
         self.bs = background_surface
@@ -246,9 +247,9 @@
 
             # show that the unit is selected
             actor = self.unit_actor[self.game.get_unit(gx, gy)]
-            timeline = clutter.Timeline(fps=60, duration=3000)
+            timeline = clutter.Timeline(3000)
             timeline.set_loop(True)
-            alpha = clutter.Alpha(timeline, clutter.sine_func)
+            alpha = clutter.Alpha(timeline, clutter.EASE_IN_OUT_SINE)
 
             g_behaviour = clutter.BehaviourScale(alpha=alpha,
                     x_scale_start= 1.1, x_scale_end = 0.9, y_scale_start = 1.1,
@@ -266,7 +267,7 @@
                 rect = clutter.Rectangle()
                 rect.set_position(sx+2, sy+3)
                 rect.set_size(self.theme.case -5, self.theme.case -5)
-                rect.set_color(clutter.Color(*color, alpha = 50))
+                rect.set_color(clutter.Color(alpha = 50, *color))
 
                 self.stage.add(rect)
                 self.BAD.append(rect)
