--- linux-2.6.37.orig/kernel/sched_bfs.c	2011-02-18 23:09:39.000000000 +0200
+++ linux-2.6.37/kernel/sched_bfs.c	2011-02-18 23:13:17.368000023 +0200
@@ -3352,7 +3352,7 @@
  * This waits for either a completion of a specific task to be signaled or for a
  * specified timeout to expire. It is interruptible. The timeout is in jiffies.
  */
-unsigned long __sched
+long __sched
 wait_for_completion_interruptible_timeout(struct completion *x,
 					  unsigned long timeout)
 {
@@ -3385,7 +3385,7 @@
  * signaled or for a specified timeout to expire. It can be
  * interrupted by a kill signal. The timeout is in jiffies.
  */
-unsigned long __sched
+long __sched
 wait_for_completion_killable_timeout(struct completion *x,
 				     unsigned long timeout)
 {
