diff -up -r a/Makefile b/Makefile
--- a/Makefile	2011-05-19 07:06:34.000000000 +0300
+++ b/Makefile	2011-05-26 19:06:30.746999996 +0300
@@ -559,6 +559,10 @@ ifndef CONFIG_CC_STACKPROTECTOR
 KBUILD_CFLAGS += $(call cc-option, -fno-stack-protector)
 endif
 
+# This warning generated too much noise in a regular build.
+# Use make W=1 to enable this warning (see scripts/Makefile.build)
+KBUILD_CFLAGS += $(call cc-option, -Wno-unused-but-set-variable)
+
 ifdef CONFIG_FRAME_POINTER
 KBUILD_CFLAGS	+= -fno-omit-frame-pointer -fno-optimize-sibling-calls
 else
diff -up -r a/scripts/Makefile.build b/scripts/Makefile.build
--- a/scripts/Makefile.build	2011-05-19 07:06:34.000000000 +0300
+++ b/scripts/Makefile.build	2011-05-26 19:06:30.747999996 +0300
@@ -80,6 +80,7 @@ KBUILD_EXTRA_WARNINGS += -Wredundant-dec
 KBUILD_EXTRA_WARNINGS += -Wshadow
 KBUILD_EXTRA_WARNINGS += -Wswitch-default
 KBUILD_EXTRA_WARNINGS += $(call cc-option, -Wvla,)
+KBUILD_EXTRA_WARNINGS += $(call cc-option, -Wunused-but-set-variable)
 KBUILD_CFLAGS += $(KBUILD_EXTRA_WARNINGS)
 endif
 
