--- configure	2011-02-02 20:12:51.645130002 +0100
+++ configure.old	2011-02-02 20:12:00.317473334 +0100
@@ -16276,7 +16276,7 @@
 if test "${with_python+set}" = set; then :
   withval=$with_python; PYTHON="$withval/python"
 else
-  PYTHON="python"
+  PYTHON="python2"
 fi
 
 
--- ./scripts/dir.py.old	2011-01-07 13:50:14.000000000 +0100
+++ ./scripts/dir.py	2011-02-02 20:58:49.152353335 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import fileinput
 
--- ./scripts/csiparse2.py.old	2011-01-07 13:50:14.000000000 +0100
+++ ./scripts/csiparse2.py	2011-02-02 20:58:23.731736668 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Generate instances for the pairwise dependency prediction task.
--- ./scripts/csidp.py.old	2011-01-07 13:50:14.000000000 +0100
+++ ./scripts/csidp.py	2011-02-02 20:58:04.274803333 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Constraint satisfaction inference for dependency parsing.
--- ./scripts/rels.py.old	2011-01-07 13:50:14.000000000 +0100
+++ ./scripts/rels.py	2011-02-02 20:59:26.553643333 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import fileinput
 
--- ./scripts/pairs.py.old	2011-01-07 13:50:14.000000000 +0100
+++ ./scripts/pairs.py	2011-02-02 20:59:07.406555000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Generate instances for the pairwise dependency prediction task.
