diff --git a/Makefile b/Makefile
index 6c7fd4e..6f3d28c 100644
--- a/Makefile
+++ b/Makefile
@@ -4,9 +4,9 @@ PREFIX    = /usr/local
 MANPREFIX = $(PREFIX)/share/man
 
 CC      = gcc
-CFLAGS  = -std=c99 -Wall -pedantic -O2 -I$(PREFIX)/include -DHAVE_GIFLIB
+CFLAGS  = -std=c99 -Wall -pedantic -O2 -I/usr/include/freetype2 -DHAVE_GIFLIB
 LDFLAGS = -L$(PREFIX)/lib
-LIBS    = -lX11 -lImlib2 -lgif
+LIBS    = -lX11 -lImlib2 -lgif -lutil -lXext -lXft -lfontconfig
 
 SRC = commands.c exif.c image.c main.c options.c thumbs.c util.c window.c
 OBJ = $(SRC:.c=.o)
diff --git a/commands.c b/commands.c
index 49298bf..70e2935 100644
--- a/commands.c
+++ b/commands.c
@@ -323,6 +323,35 @@ bool i_scroll_to_edge(arg_t a)
 		return false;
 }
 
+bool i_scroll_to_edge_seq(arg_t a)
+{
+	if (mode == MODE_IMAGE) {
+		switch (img.edgedir) {
+			case 1:
+				img.edgedir++;
+				img_pan_edge(&img, DIR_DOWN);
+				break;
+			case 2:
+				img.edgedir++;
+				img_pan_edge(&img, DIR_UP);
+				img_pan_edge(&img, DIR_LEFT);
+				break;
+			case 3:
+				img.edgedir++;
+				img_pan_edge(&img, DIR_DOWN);
+				break;
+			case 4:
+				img.edgedir = 1;
+				img_pan_edge(&img, DIR_UP);
+				img_pan_edge(&img, DIR_RIGHT);
+				break;
+		}
+		return true;
+	} else {
+		return false;
+	}
+}
+
 /* Xlib helper function for i_drag() */
 Bool is_motionnotify(Display *d, XEvent *e, XPointer a)
 {
diff --git a/commands.h b/commands.h
index ba40eed..7de7b59 100644
--- a/commands.h
+++ b/commands.h
@@ -60,6 +60,7 @@ bool it_navigate_marked(arg_t);
 bool it_scroll_move(arg_t);
 bool it_scroll_screen(arg_t);
 bool i_scroll_to_edge(arg_t);
+bool i_scroll_to_edge_seq(arg_t);
 bool i_drag(arg_t);
 bool i_zoom(arg_t);
 bool i_set_zoom(arg_t);
diff --git a/config.def.h b/config.def.h
index b1e9c02..5532cc6 100644
--- a/config.def.h
+++ b/config.def.h
@@ -9,7 +9,7 @@ enum {
 /* bar font:
  * (see X(7) section "FONT NAMES" for valid values)
  */
-static const char * const BAR_FONT = "-*-fixed-medium-r-*-*-13-*-*-*-*-60-*-*";
+static const char * const BAR_FONT = "Monospace:size=9";
 
 /* colors:
  * (see X(7) section "COLOR NAMES" for valid values)
@@ -100,8 +100,8 @@ static const keymap_t keys[] = {
 	{ true,   XK_p,             i_navigate_frame,     (arg_t) -1 },
 	{ true,   XK_space,         i_toggle_animation,   (arg_t) None },
 
-	{ false,  XK_m,             it_toggle_image_mark, (arg_t) None },
-	{ false,  XK_M,             it_reverse_marks,     (arg_t) None },
+	{ false,  XK_M,             it_toggle_image_mark, (arg_t) None },
+	{ true,   XK_M,             it_reverse_marks,     (arg_t) None },
 	{ false,  XK_N,             it_navigate_marked,   (arg_t) +1 },
 	{ false,  XK_P,             it_navigate_marked,   (arg_t) -1 },
 
@@ -127,6 +127,7 @@ static const keymap_t keys[] = {
 	{ false,  XK_J,             i_scroll_to_edge,     (arg_t) DIR_DOWN },
 	{ false,  XK_K,             i_scroll_to_edge,     (arg_t) DIR_UP },
 	{ false,  XK_L,             i_scroll_to_edge,     (arg_t) DIR_RIGHT },
+	{ false,  XK_m,             i_scroll_to_edge_seq, (arg_t) None },
 
 	{ false,  XK_plus,          i_zoom,               (arg_t) +1 },
 	{ false,  XK_KP_Add,        i_zoom,               (arg_t) +1 },
diff --git a/image.c b/image.c
index ee49076..acaf2b2 100644
--- a/image.c
+++ b/image.c
@@ -446,10 +446,11 @@ void img_render(img_t *img)
 	if (!img->re) {
 		/* rendered for the first time */
 		img->re = true;
+		img->edgedir = 1;
 		if (img->zoom * img->w <= win->w)
 			img->x = (win->w - img->w * img->zoom) / 2;
 		else
-			img->x = 0;
+			img->x = options->manga_mode ? (win->w - img->w * img->zoom) : 0;
 		if (img->zoom * img->h <= win->h)
 			img->y = (win->h - img->h * img->zoom) / 2;
 		else
diff --git a/image.h b/image.h
index 09322c2..f6ecf5f 100644
--- a/image.h
+++ b/image.h
@@ -54,6 +54,7 @@ typedef struct {
 	bool dirty;
 	bool aa;
 	bool alpha;
+	int edgedir;
 
 	Imlib_Color_Modifier cmod;
 	int gamma;
diff --git a/main.c b/main.c
index b1140cb..489e308 100644
--- a/main.c
+++ b/main.c
@@ -360,8 +360,8 @@ void update_info(void)
 	if (ow_info) {
 		fn = strlen(files[sel].name);
 		if (fn < llen &&
-		    win_textwidth(files[sel].name, fn, true) +
-		    win_textwidth(rt, n, true) < win.w)
+		    win_textwidth(&win, files[sel].name, fn, true) +
+		    win_textwidth(&win, rt, n, true) < win.w)
 		{
 			strncpy(lt, files[sel].name, llen);
 		} else {
diff --git a/options.c b/options.c
index e4d597c..0a1235c 100644
--- a/options.c
+++ b/options.c
@@ -33,7 +33,7 @@ const options_t *options = (const options_t*) &_options;
 
 void print_usage(void)
 {
-	printf("usage: sxiv [-bcdFfhioqrstvZ] [-G GAMMA] [-g GEOMETRY] [-n NUM] "
+	printf("usage: sxiv [-bcdFfhimoqrstvZ] [-G GAMMA] [-g GEOMETRY] [-n NUM] "
 	       "[-N name] [-z ZOOM] FILES...\n");
 }
 
@@ -64,9 +64,10 @@ void parse_options(int argc, char **argv)
 
 	_options.quiet = false;
 	_options.thumb_mode = false;
+	_options.manga_mode = false;
 	_options.clean_cache = false;
 
-	while ((opt = getopt(argc, argv, "bcdFfG:g:hin:N:oqrstvZz:")) != -1) {
+	while ((opt = getopt(argc, argv, "bcdFfG:g:hin:N:moqrstvZz:")) != -1) {
 		switch (opt) {
 			case '?':
 				print_usage();
@@ -104,6 +105,11 @@ void parse_options(int argc, char **argv)
 			case 'i':
 				_options.from_stdin = true;
 				break;
+			case 'm':
+				_options.manga_mode = true;
+				_options.scalemode = SCALE_ZOOM;
+				_options.zoom = 1.0;
+				break;
 			case 'n':
 				if (sscanf(optarg, "%d", &t) <= 0 || t < 1) {
 					fprintf(stderr, "sxiv: invalid argument for option -n: %s\n",
diff --git a/options.h b/options.h
index 982602f..58af49f 100644
--- a/options.h
+++ b/options.h
@@ -46,6 +46,7 @@ typedef struct {
 	/* misc flags: */
 	bool quiet;
 	bool thumb_mode;
+	bool manga_mode;
 	bool clean_cache;
 } options_t;
 
diff --git a/sxiv.1 b/sxiv.1
index 1fca2dd..98d1fd0 100644
--- a/sxiv.1
+++ b/sxiv.1
@@ -57,6 +57,10 @@ more information on
 .BI "\-N " NAME
 Set the resource name of sxiv's X window to NAME.
 .TP
+.BI "\-m "
+Start in manga mode, scale all images to 100% and read images from right to
+left.
+.TP
 .BI "\-n " NUM
 Start at picture number NUM.
 .TP
@@ -262,6 +266,9 @@ Pan to top image edge.
 .B L
 Pan to right image edge.
 .TP
+.B m
+Pan in manga way (topright, bottomright, topleft, bottomleft).
+.TP
 .BR Ctrl-h ", " Ctrl-Left
 Pan image one window width left.
 .TP
diff --git a/thumbs.c b/thumbs.c
index a1ee779..7486182 100644
--- a/thumbs.c
+++ b/thumbs.c
@@ -382,7 +382,7 @@ void tns_mark(tns_t *tns, int n, bool mark)
 		return;
 
 	if (n >= 0 && n < tns->cnt) {
-		unsigned long col;
+		XftColor col;
 		thumb_t *t = &tns->thumbs[n];
 		win_t *win = tns->win;
 		int x = t->x, y = t->y, w = t->w, h = t->h;
@@ -411,7 +411,7 @@ void tns_highlight(tns_t *tns, int n, bool hl)
 		return;
 
 	if (n >= 0 && n < tns->cnt) {
-		unsigned long col;
+		XftColor col;
 		thumb_t *t = &tns->thumbs[n];
 		win_t *win = tns->win;
 
diff --git a/window.c b/window.c
index e0b7273..6fe6be0 100644
--- a/window.c
+++ b/window.c
@@ -22,6 +22,8 @@
 #include <string.h>
 #include <locale.h>
 #include <X11/cursorfont.h>
+#include <fontconfig/fontconfig.h>
+#include <X11/Xft/Xft.h>
 
 #include "options.h"
 #include "util.h"
@@ -44,59 +46,38 @@ Atom wm_delete_win;
 static struct {
 	int ascent;
 	int descent;
-	XFontStruct *xfont;
-	XFontSet set;
+	XftFont *xfont;
 } font;
 
 static int fontheight;
 static int barheight;
 
-void win_init_font(Display *dpy, const char *fontstr)
+void win_init_font(Display *dpy, int screen, const char *fontstr)
 {
-	int n;
-	char *def, **missing;
-
-	font.set = XCreateFontSet(dpy, fontstr, &missing, &n, &def);
-	if (missing)
-		XFreeStringList(missing);
-	if (font.set) {
-		XFontStruct **xfonts;
-		char **font_names;
-
-		font.ascent = font.descent = 0;
-		XExtentsOfFontSet(font.set);
-		n = XFontsOfFontSet(font.set, &xfonts, &font_names);
-		while (n--) {
-			font.ascent  = MAX(font.ascent, (*xfonts)->ascent);
-			font.descent = MAX(font.descent,(*xfonts)->descent);
-			xfonts++;
-		}
-	} else {
-		if ((font.xfont = XLoadQueryFont(dpy, fontstr)) == NULL &&
-		    (font.xfont = XLoadQueryFont(dpy, "fixed")) == NULL)
-		{
-			die("could not load font: %s", fontstr);
-		}
-		font.ascent  = font.xfont->ascent;
-		font.descent = font.xfont->descent;
-	}
+	if(!(font.xfont = XftFontOpenName(dpy, screen, fontstr))
+	&& !(font.xfont = XftFontOpenName(dpy, screen, "fixed")))
+		die("error, cannot load font: '%s'\n", fontstr);
+
+	font.ascent = font.xfont->ascent;
+	font.descent = font.xfont->descent;
+
 	fontheight = font.ascent + font.descent;
 	barheight = fontheight + 2 * V_TEXT_PAD;
 }
 
-unsigned long win_alloc_color(win_t *win, const char *name)
+XftColor win_alloc_color(win_t *win, const char *name)
 {
-	XColor col;
+	XftColor col;
 
-	if (win == NULL)
-		return 0UL;
-	if (XAllocNamedColor(win->env.dpy,
-	                     DefaultColormap(win->env.dpy, win->env.scr),
-	                     name, &col, &col) == 0)
-	{
-		die("could not allocate color: %s", name);
+	if (win) {
+		if (!XftColorAllocName(win->env.dpy, DefaultVisual(win->env.dpy, win->env.scr),
+	                         DefaultColormap(win->env.dpy, win->env.scr),
+	                         name, &col))
+		{
+			die("could not allocate color: %s", name);
+		}
 	}
-	return col.pixel;
+	return col;
 }
 
 void win_init(win_t *win)
@@ -122,9 +103,9 @@ void win_init(win_t *win)
 	if (setlocale(LC_CTYPE, "") == NULL || XSupportsLocale() == 0)
 		warn("no locale support");
 
-	win_init_font(e->dpy, BAR_FONT);
+	win_init_font(e->dpy, e->scr, BAR_FONT);
 
-	win->white     = WhitePixel(e->dpy, e->scr);
+	win->white     = win_alloc_color(win, "#FFFFFF");
 	win->bgcol     = win_alloc_color(win, WIN_BG_COLOR);
 	win->fscol     = win_alloc_color(win, WIN_FS_COLOR);
 	win->selcol    = win_alloc_color(win, SEL_COLOR);
@@ -213,7 +194,7 @@ void win_open(win_t *win)
 		win->y = (e->scrh - win->h) / 2;
 	}
 
-	attr.background_pixel = win->bgcol;
+	attr.background_pixel = win->bgcol.pixel;
 	attr_mask = CWBackPixel;
 
 	win->xwin = XCreateWindow(e->dpy, RootWindow(e->dpy, e->scr),
@@ -392,7 +373,7 @@ void win_clear(win_t *win)
 	if (win->pm == None)
 		win->pm = XCreatePixmap(e->dpy, win->xwin, win->w, h, e->depth);
 
-	XSetForeground(e->dpy, gc, win->fullscreen ? win->fscol : win->bgcol);
+	XSetForeground(e->dpy, gc, win->fullscreen ? win->fscol.pixel : win->bgcol.pixel);
 	XFillRectangle(e->dpy, win->pm, gc, 0, 0, win->w, h);
 }
 
@@ -402,6 +383,7 @@ void win_draw_bar(win_t *win)
 	char rest[3];
 	const char *dots = "...";
 	win_env_t *e;
+	XftDraw *d;
 
 	if (win == NULL || win->xwin == None || win->pm == None)
 		return;
@@ -410,25 +392,24 @@ void win_draw_bar(win_t *win)
 	y = win->h + font.ascent + V_TEXT_PAD;
 	w = win->w;
 
-	XSetForeground(e->dpy, gc, win->bar.bgcol);
+	XSetForeground(e->dpy, gc, win->bar.bgcol.pixel);
 	XFillRectangle(e->dpy, win->pm, gc, 0, win->h, win->w, win->bar.h);
 
-	XSetForeground(e->dpy, gc, win->bar.fgcol);
-	XSetBackground(e->dpy, gc, win->bar.bgcol);
+	XSetForeground(e->dpy, gc, win->bar.fgcol.pixel);
+	XSetBackground(e->dpy, gc, win->bar.bgcol.pixel);
 
 	if ((len = strlen(win->bar.r)) > 0) {
-		if ((tw = win_textwidth(win->bar.r, len, true)) > w)
+		if ((tw = win_textwidth(win, win->bar.r, len, true)) > w)
 			return;
 		x = win->w - tw + H_TEXT_PAD;
 		w -= tw;
-		if (font.set)
-			XmbDrawString(e->dpy, win->pm, font.set, gc, x, y, win->bar.r, len);
-		else
-			XDrawString(e->dpy, win->pm, gc, x, y, win->bar.r, len);
+		d = XftDrawCreate(e->dpy, win->pm, DefaultVisual(e->dpy, e->scr), DefaultColormap(e->dpy, e->scr));
+		XftDrawStringUtf8(d, &win->bar.fgcol, font.xfont, x, y, (XftChar8 *)win->bar.r, len);
+		XftDrawDestroy(d);
 	}
 	if ((len = strlen(win->bar.l)) > 0) {
 		olen = len;
-		while (len > 0 && (tw = win_textwidth(win->bar.l, len, true)) > w)
+		while (len > 0 && (tw = win_textwidth(win, win->bar.l, len, true)) > w)
 			len--;
 		if (len > 0) {
 			if (len != olen) {
@@ -439,10 +420,9 @@ void win_draw_bar(win_t *win)
 				memcpy(win->bar.l + len - w, dots, w);
 			}
 			x = H_TEXT_PAD;
-			if (font.set)
-				XmbDrawString(e->dpy, win->pm, font.set, gc, x, y, win->bar.l, len);
-			else
-				XDrawString(e->dpy, win->pm, gc, x, y, win->bar.l, len);
+			d = XftDrawCreate(e->dpy, win->pm, DefaultVisual(e->dpy, e->scr), DefaultColormap(e->dpy, e->scr));
+			XftDrawStringUtf8(d, &win->bar.fgcol, font.xfont, x, y, (XftChar8 *)win->bar.l, len);
+			XftDrawDestroy(d);
 			if (len != olen)
 			  memcpy(win->bar.l + len - w, rest, w);
 		}
@@ -462,7 +442,7 @@ void win_draw(win_t *win)
 }
 
 void win_draw_rect(win_t *win, Pixmap pm, int x, int y, int w, int h,
-                   bool fill, int lw, unsigned long col)
+                   bool fill, int lw, XftColor col)
 {
 	XGCValues gcval;
 
@@ -470,7 +450,7 @@ void win_draw_rect(win_t *win, Pixmap pm, int x, int y, int w, int h,
 		return;
 
 	gcval.line_width = lw;
-	gcval.foreground = col;
+	gcval.foreground = col.pixel;
 	XChangeGC(win->env.dpy, gc, GCForeground | GCLineWidth, &gcval);
 
 	if (fill)
@@ -491,17 +471,13 @@ void win_update_bar(win_t *win)
 	}
 }
 
-int win_textwidth(const char *text, unsigned int len, bool with_padding)
+int win_textwidth(win_t *win, const char *text, unsigned int len, bool with_padding)
 {
-	XRectangle r;
+	XGlyphInfo ext;
 	int padding = with_padding ? 2 * H_TEXT_PAD : 0;
 
-	if (font.set) {
-		XmbTextExtents(font.set, text, len, NULL, &r);
-		return r.width + padding;
-	} else {
-		return XTextWidth(font.xfont, text, len) + padding;
-	}
+	XftTextExtentsUtf8(win->env.dpy, font.xfont, (XftChar8 *)text, len, &ext);
+	return ext.xOff + padding;
 }
 
 void win_set_title(win_t *win, const char *title)
diff --git a/window.h b/window.h
index 19a7171..b1f49ad 100644
--- a/window.h
+++ b/window.h
@@ -21,6 +21,7 @@
 
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
+#include <X11/Xft/Xft.h>
 
 #include "types.h"
 
@@ -42,10 +43,10 @@ typedef struct {
 	Window xwin;
 	win_env_t env;
 
-	unsigned long white;
-	unsigned long bgcol;
-	unsigned long fscol;
-	unsigned long selcol;
+	XftColor white;
+	XftColor bgcol;
+	XftColor fscol;
+	XftColor selcol;
 	Pixmap pm;
 
 	int x;
@@ -62,8 +63,8 @@ typedef struct {
 		unsigned int h;
 		char l[BAR_L_LEN];
 		char r[BAR_R_LEN];
-		unsigned long bgcol;
-		unsigned long fgcol;
+		XftColor bgcol;
+		XftColor fgcol;
 	} bar;
 } win_t;
 
@@ -83,11 +84,11 @@ void win_toggle_bar(win_t*);
 void win_clear(win_t*);
 void win_draw(win_t*);
 void win_draw_rect(win_t*, Pixmap, int, int, int, int, bool, int,
-                   unsigned long);
+                   XftColor);
 
 void win_update_bar(win_t*);
 
-int win_textwidth(const char*, unsigned int, bool);
+int win_textwidth(win_t*, const char*, unsigned int, bool);
 
 void win_set_title(win_t*, const char*);
 void win_set_cursor(win_t*, cursor_t);
