diff --git a/Makefile b/Makefile
index ee7077b..f91f355 100644
--- a/Makefile
+++ b/Makefile
@@ -4,9 +4,9 @@ PREFIX    = /usr/local
 MANPREFIX = $(PREFIX)/share/man
 
 CC      = gcc
-CFLAGS  = -std=c99 -Wall -pedantic -O2 -I$(PREFIX)/include -DHAVE_GIFLIB
+CFLAGS  = -std=c99 -Wall -pedantic -O2 -I$(PREFIX)/include -I$(PREFIX)/include/freetype2 -DHAVE_GIFLIB
 LDFLAGS = -L$(PREFIX)/lib
-LIBS    = -lX11 -lImlib2 -lgif
+LIBS    = -lX11 -lImlib2 -lgif -lutil -lXext -lXft -lfontconfig
 
 SRC = commands.c exif.c image.c main.c options.c thumbs.c util.c window.c
 OBJ = $(SRC:.c=.o)
diff --git a/main.c b/main.c
index 90916e3..24914df 100644
--- a/main.c
+++ b/main.c
@@ -374,8 +374,8 @@ void update_info(void)
 	if (ow_info) {
 		fn = strlen(files[sel].name);
 		if (fn < llen &&
-		    win_textwidth(files[sel].name, fn, true) +
-		    win_textwidth(rt, n, true) < win.w)
+		    win_textwidth(&win, files[sel].name, fn, true) +
+		    win_textwidth(&win, rt, n, true) < win.w)
 		{
 			strncpy(lt, files[sel].name, llen);
 		} else {
diff --git a/thumbs.c b/thumbs.c
index fe0ad84..30fbefe 100644
--- a/thumbs.c
+++ b/thumbs.c
@@ -385,7 +385,7 @@ void tns_mark(tns_t *tns, int n, bool mark)
 		return;
 
 	if (n >= 0 && n < tns->cnt) {
-		unsigned long col;
+		XftColor col;
 		thumb_t *t = &tns->thumbs[n];
 		win_t *win = tns->win;
 		int x = t->x, y = t->y, w = t->w, h = t->h;
@@ -414,7 +414,7 @@ void tns_highlight(tns_t *tns, int n, bool hl)
 		return;
 
 	if (n >= 0 && n < tns->cnt) {
-		unsigned long col;
+		XftColor col;
 		thumb_t *t = &tns->thumbs[n];
 		win_t *win = tns->win;
 
diff --git a/window.c b/window.c
index 8086a4c..36c250a 100644
--- a/window.c
+++ b/window.c
@@ -24,6 +24,8 @@
 #include <locale.h>
 #include <X11/cursorfont.h>
 #include <X11/Xatom.h>
+#include <fontconfig/fontconfig.h>
+#include <X11/Xft/Xft.h>
 
 #include "options.h"
 #include "util.h"
@@ -45,8 +47,7 @@ static GC gc;
 static struct {
 	int ascent;
 	int descent;
-	XFontStruct *xfont;
-	XFontSet set;
+	XftFont *xfont;
 } font;
 
 static int fontheight;
@@ -57,52 +58,31 @@ Atom atoms[ATOM_COUNT];
 static Bool fs_support;
 static Bool fs_warned;
 
-void win_init_font(Display *dpy, const char *fontstr)
+void win_init_font(Display *dpy, int screen, const char *fontstr)
 {
-	int n;
-	char *def, **missing;
-
-	font.set = XCreateFontSet(dpy, fontstr, &missing, &n, &def);
-	if (missing)
-		XFreeStringList(missing);
-	if (font.set) {
-		XFontStruct **xfonts;
-		char **font_names;
-
-		font.ascent = font.descent = 0;
-		XExtentsOfFontSet(font.set);
-		n = XFontsOfFontSet(font.set, &xfonts, &font_names);
-		while (n--) {
-			font.ascent  = MAX(font.ascent, (*xfonts)->ascent);
-			font.descent = MAX(font.descent,(*xfonts)->descent);
-			xfonts++;
-		}
-	} else {
-		if ((font.xfont = XLoadQueryFont(dpy, fontstr)) == NULL &&
-		    (font.xfont = XLoadQueryFont(dpy, "fixed")) == NULL)
-		{
-			die("could not load font: %s", fontstr);
-		}
-		font.ascent  = font.xfont->ascent;
-		font.descent = font.xfont->descent;
-	}
+	if (!(font.xfont = XftFontOpenName(dpy, screen, fontstr))
+	&&  !(font.xfont = XftFontOpenName(dpy, screen, "fixed")))
+		die("error, cannot load font: '%s'\n", fontstr);
+
+	font.ascent = font.xfont->ascent;
+	font.descent = font.xfont->descent;
+
 	fontheight = font.ascent + font.descent;
 	barheight = fontheight + 2 * V_TEXT_PAD;
 }
 
-unsigned long win_alloc_color(win_t *win, const char *name)
+XftColor win_alloc_color(win_t *win, const char *name)
 {
-	XColor col;
+	XftColor col;
 
-	if (win == NULL)
-		return 0UL;
-	if (XAllocNamedColor(win->env.dpy,
-	                     DefaultColormap(win->env.dpy, win->env.scr),
-	                     name, &col, &col) == 0)
-	{
-		die("could not allocate color: %s", name);
+	if (win) {
+		if (!XftColorAllocName(win->env.dpy, DefaultVisual(win->env.dpy, win->env.scr),
+																				DefaultColormap(win->env.dpy, win->env.scr),
+																				name, &col)) {
+			die("could not allocate color: %s\n", name);
+		}
 	}
-	return col.pixel;
+	return col;
 }
 
 void win_check_wm_support(Display *dpy, Window root)
@@ -161,9 +141,9 @@ void win_init(win_t *win)
 	if (setlocale(LC_CTYPE, "") == NULL || XSupportsLocale() == 0)
 		warn("no locale support");
 
-	win_init_font(e->dpy, BAR_FONT);
+	win_init_font(e->dpy, e->scr, BAR_FONT);
 
-	win->white     = WhitePixel(e->dpy, e->scr);
+	win->white     = win_alloc_color(win, "#FFFFFF");
 	win->bgcol     = win_alloc_color(win, WIN_BG_COLOR);
 	win->fscol     = win_alloc_color(win, WIN_FS_COLOR);
 	win->selcol    = win_alloc_color(win, SEL_COLOR);
@@ -262,7 +242,7 @@ void win_open(win_t *win)
 		win->y = (e->scrh - win->h) / 2;
 	}
 
-	attr.background_pixel = win->bgcol;
+	attr.background_pixel = win->bgcol.pixel;
 	attr_mask = CWBackPixel;
 
 	win->xwin = XCreateWindow(e->dpy, RootWindow(e->dpy, e->scr),
@@ -467,7 +447,7 @@ void win_clear(win_t *win)
 	if (win->pm == None)
 		win->pm = XCreatePixmap(e->dpy, win->xwin, win->w, h, e->depth);
 
-	XSetForeground(e->dpy, gc, win->fullscreen ? win->fscol : win->bgcol);
+	XSetForeground(e->dpy, gc, win->fullscreen ? win->fscol.pixel : win->bgcol.pixel);
 	XFillRectangle(e->dpy, win->pm, gc, 0, 0, win->w, h);
 }
 
@@ -477,6 +457,7 @@ void win_draw_bar(win_t *win)
 	char rest[3];
 	const char *dots = "...";
 	win_env_t *e;
+	XftDraw *d;
 
 	if (win == NULL || win->xwin == None || win->pm == None)
 		return;
@@ -485,25 +466,24 @@ void win_draw_bar(win_t *win)
 	y = win->h + font.ascent + V_TEXT_PAD;
 	w = win->w;
 
-	XSetForeground(e->dpy, gc, win->bar.bgcol);
+	XSetForeground(e->dpy, gc, win->bar.bgcol.pixel);
 	XFillRectangle(e->dpy, win->pm, gc, 0, win->h, win->w, win->bar.h);
 
-	XSetForeground(e->dpy, gc, win->bar.fgcol);
-	XSetBackground(e->dpy, gc, win->bar.bgcol);
+	XSetForeground(e->dpy, gc, win->bar.fgcol.pixel);
+	XSetBackground(e->dpy, gc, win->bar.bgcol.pixel);
 
 	if ((len = strlen(win->bar.r)) > 0) {
-		if ((tw = win_textwidth(win->bar.r, len, true)) > w)
+		if ((tw = win_textwidth(win, win->bar.r, len, true)) > w)
 			return;
 		x = win->w - tw + H_TEXT_PAD;
 		w -= tw;
-		if (font.set)
-			XmbDrawString(e->dpy, win->pm, font.set, gc, x, y, win->bar.r, len);
-		else
-			XDrawString(e->dpy, win->pm, gc, x, y, win->bar.r, len);
+		d = XftDrawCreate(e->dpy, win->pm, DefaultVisual(e->dpy, e->scr), DefaultColormap(e->dpy, e->scr));
+		XftDrawStringUtf8(d, &win->bar.fgcol, font.xfont, x, y, (XftChar8 *)win->bar.r, len);
+		XftDrawDestroy(d);
 	}
 	if ((len = strlen(win->bar.l)) > 0) {
 		olen = len;
-		while (len > 0 && (tw = win_textwidth(win->bar.l, len, true)) > w)
+		while (len > 0 && (tw = win_textwidth(win, win->bar.l, len, true)) > w)
 			len--;
 		if (len > 0) {
 			if (len != olen) {
@@ -514,10 +494,9 @@ void win_draw_bar(win_t *win)
 				memcpy(win->bar.l + len - w, dots, w);
 			}
 			x = H_TEXT_PAD;
-			if (font.set)
-				XmbDrawString(e->dpy, win->pm, font.set, gc, x, y, win->bar.l, len);
-			else
-				XDrawString(e->dpy, win->pm, gc, x, y, win->bar.l, len);
+			d = XftDrawCreate(e->dpy, win->pm, DefaultVisual(e->dpy, e->scr), DefaultColormap(e->dpy, e->scr));
+			XftDrawStringUtf8(d, &win->bar.fgcol, font.xfont, x, y, (XftChar8 *)win->bar.l, len);
+			XftDrawDestroy(d);
 			if (len != olen)
 			  memcpy(win->bar.l + len - w, rest, w);
 		}
@@ -537,7 +516,7 @@ void win_draw(win_t *win)
 }
 
 void win_draw_rect(win_t *win, Pixmap pm, int x, int y, int w, int h,
-                   bool fill, int lw, unsigned long col)
+                   bool fill, int lw, XftColor col)
 {
 	XGCValues gcval;
 
@@ -545,7 +524,7 @@ void win_draw_rect(win_t *win, Pixmap pm, int x, int y, int w, int h,
 		return;
 
 	gcval.line_width = lw;
-	gcval.foreground = col;
+	gcval.foreground = col.pixel;
 	XChangeGC(win->env.dpy, gc, GCForeground | GCLineWidth, &gcval);
 
 	if (fill)
@@ -566,17 +545,13 @@ void win_update_bar(win_t *win)
 	}
 }
 
-int win_textwidth(const char *text, unsigned int len, bool with_padding)
+int win_textwidth(win_t *win, const char *text, unsigned int len, bool with_padding)
 {
-	XRectangle r;
+	XGlyphInfo ext;
 	int padding = with_padding ? 2 * H_TEXT_PAD : 0;
 
-	if (font.set) {
-		XmbTextExtents(font.set, text, len, NULL, &r);
-		return r.width + padding;
-	} else {
-		return XTextWidth(font.xfont, text, len) + padding;
-	}
+	XftTextExtentsUtf8(win->env.dpy, font.xfont, (XftChar8 *)text, len, &ext);
+	return ext.xOff + padding;
 }
 
 void win_set_title(win_t *win, const char *title)
diff --git a/window.h b/window.h
index ec71319..22faf32 100644
--- a/window.h
+++ b/window.h
@@ -21,6 +21,7 @@
 
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
+#include <X11/Xft/Xft.h>
 
 #include "types.h"
 
@@ -53,10 +54,10 @@ typedef struct {
 	Window xwin;
 	win_env_t env;
 
-	unsigned long white;
-	unsigned long bgcol;
-	unsigned long fscol;
-	unsigned long selcol;
+	XftColor white;
+	XftColor bgcol;
+	XftColor fscol;
+	XftColor selcol;
 	Pixmap pm;
 
 	int x;
@@ -73,8 +74,8 @@ typedef struct {
 		unsigned int h;
 		char l[BAR_L_LEN];
 		char r[BAR_R_LEN];
-		unsigned long bgcol;
-		unsigned long fgcol;
+		XftColor bgcol;
+		XftColor fgcol;
 	} bar;
 } win_t;
 
@@ -94,11 +95,11 @@ void win_toggle_bar(win_t*);
 void win_clear(win_t*);
 void win_draw(win_t*);
 void win_draw_rect(win_t*, Pixmap, int, int, int, int, bool, int,
-                   unsigned long);
+                   XftColor);
 
 void win_update_bar(win_t*);
 
-int win_textwidth(const char*, unsigned int, bool);
+int win_textwidth(win_t*, const char*, unsigned int, bool);
 
 void win_set_title(win_t*, const char*);
 void win_set_cursor(win_t*, cursor_t);
