diff --git a/Makefile b/Makefile
index a00d366..b4caef6 100644
--- a/Makefile
+++ b/Makefile
@@ -5,9 +5,9 @@ MANPREFIX = $(PREFIX)/share/man
 
 CC       = gcc
 CFLAGS   = -std=c99 -Wall -pedantic -O2
-CPPFLAGS = -I$(PREFIX)/include -D_XOPEN_SOURCE=500 -DHAVE_LIBEXIF -DHAVE_GIFLIB
+CPPFLAGS = -I$(PREFIX)/include -D_XOPEN_SOURCE=500 -DHAVE_LIBEXIF -DHAVE_GIFLIB -I$(PREFIX)/include/freetype2
 LDFLAGS  = -L$(PREFIX)/lib
-LIBS     = -lX11 -lImlib2 -lexif -lgif
+LIBS     = -lX11 -lImlib2 -lexif -lgif -lutil -lXext -lXft -lfontconfig
 
 SRC = commands.c image.c main.c options.c thumbs.c util.c window.c
 OBJ = $(SRC:.c=.o)
diff --git a/image.c b/image.c
index 8f14461..3be824a 100644
--- a/image.c
+++ b/image.c
@@ -450,7 +450,7 @@ void img_render(img_t *img)
 	int sx, sy, sw, sh;
 	int dx, dy, dw, dh;
 	Imlib_Image bg;
-	unsigned long c;
+	XftColor c;
 
 	if (img == NULL || img->im == NULL || img->win == NULL)
 		return;
@@ -522,7 +522,7 @@ void img_render(img_t *img)
 			imlib_image_put_back_data(data);
 		} else {
 			c = win->fullscreen ? win->fscol : win->bgcol;
-			imlib_context_set_color(c >> 16 & 0xFF, c >> 8 & 0xFF, c & 0xFF, 0xFF);
+			imlib_context_set_color(c.color.red & 0xFF, c.color.blue & 0xFF, c.color.green & 0xFF, 0xFF);
 			imlib_image_fill_rectangle(0, 0, dw, dh);
 		}
 		imlib_blend_image_onto_image(img->im, 0, sx, sy, sw, sh, 0, 0, dw, dh);
diff --git a/main.c b/main.c
index 070d419..cb07cd6 100644
--- a/main.c
+++ b/main.c
@@ -392,8 +392,8 @@ void update_info(void)
 	if (ow_info) {
 		fn = strlen(files[fileidx].name);
 		if (fn < llen &&
-		    win_textwidth(files[fileidx].name, fn, true) +
-		    win_textwidth(rt, n, true) < win.w)
+		    win_textwidth(&win, files[fileidx].name, fn, true) +
+		    win_textwidth(&win, rt, n, true) < win.w)
 		{
 			strncpy(lt, files[fileidx].name, llen);
 		} else {
diff --git a/thumbs.c b/thumbs.c
index ea91292..701dfda 100644
--- a/thumbs.c
+++ b/thumbs.c
@@ -421,7 +421,7 @@ void tns_mark(tns_t *tns, int n, bool mark)
 	if (n >= 0 && n < tns->cnt) {
 		win_t *win = tns->win;
 		thumb_t *t = &tns->thumbs[n];
-		unsigned long col = win->fullscreen ? win->fscol : win->bgcol;
+		XftColor col = win->fullscreen ? win->fscol : win->bgcol;
 		int x = t->x + t->w, y = t->y + t->h;
 
 		win_draw_rect(win, x - 2, y + 1, 1, 2, true, 1, col);
@@ -446,7 +446,7 @@ void tns_highlight(tns_t *tns, int n, bool hl)
 	if (n >= 0 && n < tns->cnt) {
 		win_t *win = tns->win;
 		thumb_t *t = &tns->thumbs[n];
-		unsigned long col;
+		XftColor col;
 
 		if (hl)
 			col = win->selcol;
diff --git a/window.c b/window.c
index 581ca76..af237fa 100644
--- a/window.c
+++ b/window.c
@@ -24,6 +24,8 @@
 #include <locale.h>
 #include <X11/cursorfont.h>
 #include <X11/Xatom.h>
+#include <fontconfig/fontconfig.h>
+#include <X11/Xft/Xft.h>
 
 #include "options.h"
 #include "util.h"
@@ -45,8 +47,7 @@ static GC gc;
 static struct {
 	int ascent;
 	int descent;
-	XFontStruct *xfont;
-	XFontSet set;
+	XftFont *xfont;
 } font;
 
 static int fontheight;
@@ -57,52 +58,31 @@ Atom atoms[ATOM_COUNT];
 static Bool fs_support;
 static Bool fs_warned;
 
-void win_init_font(Display *dpy, const char *fontstr)
+void win_init_font(Display *dpy, int screen, const char *fontstr)
 {
-	int n;
-	char *def, **missing;
-
-	font.set = XCreateFontSet(dpy, fontstr, &missing, &n, &def);
-	if (missing)
-		XFreeStringList(missing);
-	if (font.set) {
-		XFontStruct **xfonts;
-		char **font_names;
-
-		font.ascent = font.descent = 0;
-		XExtentsOfFontSet(font.set);
-		n = XFontsOfFontSet(font.set, &xfonts, &font_names);
-		while (n--) {
-			font.ascent  = MAX(font.ascent, (*xfonts)->ascent);
-			font.descent = MAX(font.descent,(*xfonts)->descent);
-			xfonts++;
-		}
-	} else {
-		if ((font.xfont = XLoadQueryFont(dpy, fontstr)) == NULL &&
-		    (font.xfont = XLoadQueryFont(dpy, "fixed")) == NULL)
-		{
-			die("could not load font: %s", fontstr);
-		}
-		font.ascent  = font.xfont->ascent;
-		font.descent = font.xfont->descent;
-	}
+	if (!(font.xfont = XftFontOpenName(dpy, screen, fontstr))
+	&&  !(font.xfont = XftFontOpenName(dpy, screen, "fixed")))
+		die("error, cannot load font: '%s'\n", fontstr);
+
+	font.ascent = font.xfont->ascent;
+	font.descent = font.xfont->descent;
+
 	fontheight = font.ascent + font.descent;
 	barheight = fontheight + 2 * V_TEXT_PAD;
 }
 
-unsigned long win_alloc_color(win_t *win, const char *name)
+XftColor win_alloc_color(win_t *win, const char *name)
 {
-	XColor col;
+	XftColor col;
 
-	if (win == NULL)
-		return 0UL;
-	if (XAllocNamedColor(win->env.dpy,
-	                     DefaultColormap(win->env.dpy, win->env.scr),
-	                     name, &col, &col) == 0)
-	{
-		die("could not allocate color: %s", name);
+	if (win) {
+		if (!XftColorAllocName(win->env.dpy, DefaultVisual(win->env.dpy, win->env.scr),
+																				DefaultColormap(win->env.dpy, win->env.scr),
+																				name, &col)) {
+			die("could not allocate color: %s\n", name);
+		}
 	}
-	return col.pixel;
+	return col;
 }
 
 void win_check_wm_support(Display *dpy, Window root)
@@ -161,7 +141,7 @@ void win_init(win_t *win)
 	if (setlocale(LC_CTYPE, "") == NULL || XSupportsLocale() == 0)
 		warn("no locale support");
 
-	win_init_font(e->dpy, BAR_FONT);
+	win_init_font(e->dpy, e->scr, BAR_FONT);
 
 	win->bgcol     = win_alloc_color(win, WIN_BG_COLOR);
 	win->fscol     = win_alloc_color(win, WIN_FS_COLOR);
@@ -299,7 +279,7 @@ void win_open(win_t *win)
 	win->buf.h = e->scrh;
 	win->buf.pm = XCreatePixmap(e->dpy, win->xwin,
 	                            win->buf.w, win->buf.h, e->depth);
-	XSetForeground(e->dpy, gc, fullscreen ? win->fscol : win->bgcol);
+	XSetForeground(e->dpy, gc, fullscreen ? win->fscol.pixel : win->bgcol.pixel);
 	XFillRectangle(e->dpy, win->buf.pm, gc, 0, 0, win->buf.w, win->buf.h);
 	XSetWindowBackgroundPixmap(e->dpy, win->xwin, win->buf.pm);
 
@@ -406,7 +386,7 @@ void win_clear(win_t *win)
 		win->buf.pm = XCreatePixmap(e->dpy, win->xwin,
 		                            win->buf.w, win->buf.h, e->depth);
 	}
-	XSetForeground(e->dpy, gc, win->fullscreen ? win->fscol : win->bgcol);
+	XSetForeground(e->dpy, gc, win->fullscreen ? win->fscol.pixel : win->bgcol.pixel);
 	XFillRectangle(e->dpy, win->buf.pm, gc, 0, 0, win->buf.w, win->buf.h);
 }
 
@@ -416,6 +396,7 @@ void win_draw_bar(win_t *win)
 	char rest[3];
 	const char *dots = "...";
 	win_env_t *e;
+	XftDraw *d;
 
 	if (win == NULL || win->xwin == None)
 		return;
@@ -424,25 +405,24 @@ void win_draw_bar(win_t *win)
 	y = win->h + font.ascent + V_TEXT_PAD;
 	w = win->w;
 
-	XSetForeground(e->dpy, gc, win->bar.bgcol);
+	XSetForeground(e->dpy, gc, win->bar.bgcol.pixel);
 	XFillRectangle(e->dpy, win->buf.pm, gc, 0, win->h, win->w, win->bar.h);
 
-	XSetForeground(e->dpy, gc, win->bar.fgcol);
-	XSetBackground(e->dpy, gc, win->bar.bgcol);
+	XSetForeground(e->dpy, gc, win->bar.fgcol.pixel);
+	XSetBackground(e->dpy, gc, win->bar.bgcol.pixel);
 
 	if ((len = strlen(win->bar.r)) > 0) {
-		if ((tw = win_textwidth(win->bar.r, len, true)) > w)
+		if ((tw = win_textwidth(win, win->bar.r, len, true)) > w)
 			return;
 		x = win->w - tw + H_TEXT_PAD;
 		w -= tw;
-		if (font.set)
-			XmbDrawString(e->dpy, win->buf.pm, font.set, gc, x, y, win->bar.r, len);
-		else
-			XDrawString(e->dpy, win->buf.pm, gc, x, y, win->bar.r, len);
+		d = XftDrawCreate(e->dpy, win->buf.pm, DefaultVisual(e->dpy, e->scr), DefaultColormap(e->dpy, e->scr));
+		XftDrawStringUtf8(d, &win->bar.fgcol, font.xfont, x, y, (XftChar8 *)win->bar.r, len);
+		XftDrawDestroy(d);
 	}
 	if ((len = strlen(win->bar.l)) > 0) {
 		olen = len;
-		while (len > 0 && (tw = win_textwidth(win->bar.l, len, true)) > w)
+		while (len > 0 && (tw = win_textwidth(win, win->bar.l, len, true)) > w)
 			len--;
 		if (len > 0) {
 			if (len != olen) {
@@ -453,10 +433,9 @@ void win_draw_bar(win_t *win)
 				memcpy(win->bar.l + len - w, dots, w);
 			}
 			x = H_TEXT_PAD;
-			if (font.set)
-				XmbDrawString(e->dpy, win->buf.pm, font.set, gc, x, y, win->bar.l, len);
-			else
-				XDrawString(e->dpy, win->buf.pm, gc, x, y, win->bar.l, len);
+			d = XftDrawCreate(e->dpy, win->buf.pm, DefaultVisual(e->dpy, e->scr), DefaultColormap(e->dpy, e->scr));
+			XftDrawStringUtf8(d, &win->bar.fgcol, font.xfont, x, y, (XftChar8 *)win->bar.l, len);
+			XftDrawDestroy(d);
 			if (len != olen)
 			  memcpy(win->bar.l + len - w, rest, w);
 		}
@@ -477,7 +456,7 @@ void win_draw(win_t *win)
 }
 
 void win_draw_rect(win_t *win, int x, int y, int w, int h, bool fill, int lw,
-                   unsigned long col)
+                   XftColor col)
 {
 	XGCValues gcval;
 
@@ -485,7 +464,7 @@ void win_draw_rect(win_t *win, int x, int y, int w, int h, bool fill, int lw,
 		return;
 
 	gcval.line_width = lw;
-	gcval.foreground = col;
+	gcval.foreground = col.pixel;
 	XChangeGC(win->env.dpy, gc, GCForeground | GCLineWidth, &gcval);
 
 	if (fill)
@@ -494,17 +473,13 @@ void win_draw_rect(win_t *win, int x, int y, int w, int h, bool fill, int lw,
 		XDrawRectangle(win->env.dpy, win->buf.pm, gc, x, y, w, h);
 }
 
-int win_textwidth(const char *text, unsigned int len, bool with_padding)
+int win_textwidth(win_t *win, const char *text, unsigned int len, bool with_padding)
 {
-	XRectangle r;
+	XGlyphInfo ext;
 	int padding = with_padding ? 2 * H_TEXT_PAD : 0;
 
-	if (font.set) {
-		XmbTextExtents(font.set, text, len, NULL, &r);
-		return r.width + padding;
-	} else {
-		return XTextWidth(font.xfont, text, len) + padding;
-	}
+	XftTextExtentsUtf8(win->env.dpy, font.xfont, (XftChar8 *)text, len, &ext);
+	return ext.xOff + padding;
 }
 
 void win_set_title(win_t *win, const char *title)
diff --git a/window.h b/window.h
index 1178007..9bdd5a8 100644
--- a/window.h
+++ b/window.h
@@ -21,6 +21,7 @@
 
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
+#include <X11/Xft/Xft.h>
 
 #include "types.h"
 
@@ -53,9 +54,9 @@ typedef struct {
 	Window xwin;
 	win_env_t env;
 
-	unsigned long bgcol;
-	unsigned long fscol;
-	unsigned long selcol;
+	XftColor bgcol;
+	XftColor fscol;
+	XftColor selcol;
 
 	int x;
 	int y;
@@ -75,8 +76,8 @@ typedef struct {
 		unsigned int h;
 		char l[BAR_L_LEN];
 		char r[BAR_R_LEN];
-		unsigned long bgcol;
-		unsigned long fgcol;
+		XftColor bgcol;
+		XftColor fgcol;
 	} bar;
 } win_t;
 
@@ -93,9 +94,9 @@ void win_toggle_bar(win_t*);
 
 void win_clear(win_t*);
 void win_draw(win_t*);
-void win_draw_rect(win_t*, int, int, int, int, bool, int, unsigned long);
+void win_draw_rect(win_t*, int, int, int, int, bool, int, XftColor);
 
-int win_textwidth(const char*, unsigned int, bool);
+int win_textwidth(win_t*, const char*, unsigned int, bool);
 
 void win_set_title(win_t*, const char*);
 void win_set_cursor(win_t*, cursor_t);
