--- ./src/GNS3-0.8.2-BETA-src/setup.py	2011-12-12 05:29:59.000000000 +0100
+++ setup.py	2011-12-16 22:19:24.000000000 +0100
@@ -219,7 +219,6 @@
                 'GNS3.Ui.ConfigurationPages',
                 'GNS3.Langs'],
           package_data = { 'GNS3': ['Langs/*.qm', 'Dynagen/configspec'] },
-          data_files = [ ('/usr/local/libexec/gns3/', ['qemuwrapper/qemuwrapper.py', 'vboxwrapper/vboxcontroller_4_1.py', 'vboxwrapper/vboxwrapper.py']),
-                        ('/usr/local/share/examples/gns3/', ['baseconfig.txt'])]
+          data_files = [ ]
     )
 
