--- mozilla-1.9.2/toolkit/system/gnome/nsAlertsIconListener.cpp.old	2011-03-28 22:05:22.282469799 +0200
+++ mozilla-1.9.2/toolkit/system/gnome/nsAlertsIconListener.cpp	2011-03-28 21:34:05.534299783 +0200
@@ -204,7 +204,7 @@
 {
   NotifyNotification* notify = notify_notification_new(mAlertTitle.get(),
                                                        mAlertText.get(),
-                                                       NULL, NULL);
+                                                       NULL);
   if (!notify)
     return NS_ERROR_OUT_OF_MEMORY;
 
