--- sdl1/pdcsdl.h	2008-07-14 07:24:52.000000000 +0300
+++ sdl1/pdcsdl.h	2010-07-11 00:59:49.553272703 +0300
@@ -2,12 +2,10 @@
 
 /* $Id: pdcsdl.h,v 1.17 2008/07/14 04:24:52 wmcbrine Exp $ */
 
-#include <curspriv.h>
-
 #include <SDL/SDL.h>
 
-PDCEX SDL_Surface *pdc_screen, *pdc_font, *pdc_icon, *pdc_back;
-PDCEX int pdc_sheight, pdc_swidth, pdc_yoffset, pdc_xoffset;
+extern SDL_Surface *pdc_screen, *pdc_font, *pdc_icon, *pdc_back;
+extern int pdc_sheight, pdc_swidth, pdc_yoffset, pdc_xoffset;
 
 extern SDL_Surface *pdc_tileback;    /* used to regenerate the background
                                         of "transparent" cells */
