From 4a277b5515c7e2cd0286add232b064486796ff4c Mon Sep 17 00:00:00 2001
From: Yosef Or Boczko <yoseforb@gmail.com>
Date: Mon, 15 Jul 2013 16:54:31 +0300
Subject: Add class style "titlebar" to the headerbar for the csd

---
 src/baobab-main-window.ui | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/baobab-main-window.ui b/src/baobab-main-window.ui
index 76ba263..9f3104c 100644
--- a/src/baobab-main-window.ui
+++ b/src/baobab-main-window.ui
@@ -50,6 +50,9 @@
       <object class="GtkHeaderBar" id="header_bar">
         <property name="visible">True</property>
         <property name="can_focus">False</property>
+        <style>
+            <class name="titlebar"/>
+        </style>
         <child>
           <object class="GtkButton" id="back_button">
             <property name="valign">center</property>
-- 
1.8.3.2

