Index: src/Config.py
===================================================================
RCS file: /cvsroot/krank/krank/src/Config.py,v
retrieving revision 1.13
diff -u -r1.13 Config.py
--- src/Config.py	28 Feb 2007 13:23:43 -0000	1.13
+++ src/Config.py	14 Jun 2008 23:31:03 -0000
@@ -91,6 +91,8 @@
     def getConfigFilePath(self):
         if os.sys.platform == 'darwin':
             return os.path.expanduser('~/Library/Preferences/krank.cfg')
+        elif os.sys.platform == 'linux2':
+            return os.path.expanduser('~/.krankcfg')
         else:
             return os.path.join(os.environ['APPDATA'], 'krank.cfg')
         
@@ -185,4 +187,4 @@
             stage = self.stage
         return len([s for s in self['scores'][stage-1] if s['score'] < sys.maxint])
     
-    
\ No newline at end of file
+    
