#Description: Include the texts subdir in the build.
Index: tecnoballz-0.92/configure.ac
===================================================================
--- tecnoballz-0.92.orig/configure.ac	2009-04-16 15:35:02.000000000 -0400
+++ tecnoballz-0.92/configure.ac	2009-04-16 15:35:33.000000000 -0400
@@ -81,5 +81,5 @@
 
 
 dnl  Output makefiles
-AC_OUTPUT(Makefile src/Makefile src/TecnoballZ/Makefile src/TecnoballZ/hires/Makefile src/TecnoballZ/lores/Makefile src/TecnoballZ/textures/Makefile src/TecnoballZ/musics/Makefile src/TecnoballZ/sounds/Makefile)
+AC_OUTPUT(Makefile src/Makefile src/TecnoballZ/Makefile src/TecnoballZ/hires/Makefile src/TecnoballZ/lores/Makefile src/TecnoballZ/textures/Makefile src/TecnoballZ/musics/Makefile src/TecnoballZ/sounds/Makefile src/TecnoballZ/texts/Makefile)
 
Index: tecnoballz-0.92/src/TecnoballZ/Makefile.am
===================================================================
--- tecnoballz-0.92.orig/src/TecnoballZ/Makefile.am	2009-04-16 15:35:46.000000000 -0400
+++ tecnoballz-0.92/src/TecnoballZ/Makefile.am	2009-04-16 15:36:10.000000000 -0400
@@ -1,6 +1,6 @@
 NULL =
 
-SUBDIRS = lores hires textures sounds musics
+SUBDIRS = lores hires textures sounds musics texts
 
 pkgdatadir = $(datadir)/tecnoballz
 dist_pkgdata_DATA = \
Index: tecnoballz-0.92/src/TecnoballZ/texts/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ tecnoballz-0.92/src/TecnoballZ/texts/Makefile.am	2009-04-16 15:37:27.000000000 -0400
@@ -0,0 +1,15 @@
+NULL =
+pkgdatadir = $(datadir)/tecnoballz/texts
+dist_pkgdata_DATA = \
+	main_menu_en.txt \
+	main_menu_fr.txt \
+	popup_menu_en.txt \
+	popup_menu_fr.txt \
+	scrolltext_en.txt \
+	scrolltext_fr.txt \
+	shop_en.txt \
+	shop_fr.txt \
+	short_info_messages_en.txt \
+	short_info_messages_fr.txt \
+	$(NULL)
+
