#Description: Include levels-data.xml file in build.
Index: tecnoballz-0.92/src/TecnoballZ/Makefile.am
===================================================================
--- tecnoballz-0.92.orig/src/TecnoballZ/Makefile.am	2009-04-16 14:41:28.000000000 -0400
+++ tecnoballz-0.92/src/TecnoballZ/Makefile.am	2009-04-16 14:41:48.000000000 -0400
@@ -6,6 +6,7 @@
 dist_pkgdata_DATA = \
 	cosinus128.list \
 	gard_lissa.list \
+	levels-data.xml \
 	min60map.png \
 	tableau.data \
   tilemap-congratulation_20.data \
