diff -dur tecnoballz-0.92.orig/configure.ac tecnoballz-0.92/configure.ac
--- tecnoballz-0.92.orig/configure.ac	2007-11-19 13:44:15.000000000 +0100
+++ tecnoballz-0.92/configure.ac	2011-11-26 18:04:22.221420518 +0100
@@ -50,23 +50,6 @@
 
 LDFLAGS_="${LDFLAGS_save}"
 
-AC_CHECK_LIB(vorbisfile, ov_raw_seek, [
-	MM_LIBS="${MM_LIBS} -lvorbisfile"
-],[
-	AC_MSG_ERROR(Could not find the vorbisfile lib : -lvorbisfile)
-])
-AC_CHECK_LIB(ogg, ogg_stream_packetin, [
-	MM_LIBS="${MM_LIBS} -logg"
-],[
-	AC_MSG_ERROR(Could not find the ogg lib : -logg)
-])
-AC_CHECK_LIB(smpeg, SMPEG_play, [
-	MM_LIBS="${MM_LIBS} -lsmpeg"
-],[
-	AC_MSG_ERROR(Could not find the smpeg lib : -lsmpeg)
-])
-AC_SUBST(MM_LIBS)
-
 AC_CHECK_LIB(mikmod, Player_Stop, [
 	MIKMOD_LIBS="-lmikmod"
 	MIKMOD_FLAGS="-pthread -D_REENTRANT"
diff -dur tecnoballz-0.92.orig/src/Makefile.am tecnoballz-0.92/src/Makefile.am
--- tecnoballz-0.92.orig/src/Makefile.am	2007-10-31 08:35:29.000000000 +0100
+++ tecnoballz-0.92/src/Makefile.am	2011-11-26 18:04:44.634636369 +0100
@@ -9,7 +9,7 @@
 games_PROGRAMS = tecnoballz
 tecnoballz_SOURCES = $(SOURCES_MAIN) 
 tecnoballz_CXXFLAGS = -DDATADIR=\"$(datadir)/tecnoballz\" -DSCOREFILE=\"$(scoredir)/$(score)\" @XLIB_CFLAGS@ @SDL_CFLAGS@ @MIKMOD_FLAGS@ 
-tecnoballz_LDADD = @XLIB_LIBS@ @SDL_LIBS@ @MM_LIBS@ @MIKMOD_LIBS@ -L.
+tecnoballz_LDADD = @XLIB_LIBS@ @SDL_LIBS@ @MIKMOD_LIBS@ -L.
 
 SOURCES_MAIN = \
   bitmap_data.cc \
