diff -Nur glc_lib.orig/GLC_lib.pro glc_lib/GLC_lib.pro
--- glc_lib.orig/GLC_lib.pro	2011-06-19 17:16:28.000000000 +1000
+++ glc_lib/GLC_lib.pro	2012-01-02 11:44:16.385034999 +1100
@@ -21,6 +21,7 @@
 DEPENDPATH += .
 INCLUDEPATH += .
 INCLUDEPATH += ./3rdparty/zlib
+!win32:LIBS += -lGLU
 
 # Input					
 HEADERS_QUAZIP +=	3rdparty/quazip/crypt.h \
diff -Nur glc_lib.orig/glc_object.h glc_lib/glc_object.h
--- glc_lib.orig/glc_object.h	2011-06-19 17:16:30.000000000 +1000
+++ glc_lib/glc_object.h	2012-01-02 11:42:51.918469751 +1100
@@ -26,6 +26,11 @@
 #define GLC_OBJECT_H_
 
 #include <QtOpenGL>
+#ifdef Q_WS_MAC
+# include <OpenGL/glu.h>
+#else
+# include <GL/glu.h>
+#endif
 #include <QString>
 #include <QUuid>
 #include <QMutex>
