--- xcdroast-0.98alpha15/src/init.c.nowarn	2003-10-30 12:58:26.000000000 +0100
+++ xcdroast-0.98alpha15/src/init.c	2004-09-09 16:50:31.651482768 +0200
@@ -736,7 +736,7 @@
 #if (defined(__MACH__) && defined(__APPLE__)) || (USE_NONROOTMODE == 0)
 	dodebug(1,"-- X-CD-Roast compiled with disabled non-root-mode.\n");
 
-#if !(defined(__MACH__) && defined(__APPLE__))
+#if !(defined(__MACH__) && defined(__APPLE__)) && 0
 	if (!isroot()) {
 		g_print("\nWARNING: Using X-CD-Roast as non-root and was compiled without non-root-support.\n");
 	}
@@ -1125,7 +1125,10 @@
 		if (isroot()) {
 			show_fancy_dialog(_("Warning:"), _("This copy of X-CD-Roast has been compiled without\nnon-root-support. If you want to use X-CD-Roast without entering\nthe root-password, you have to install a fully enabled version.\n\nVisit http://www.xcdroast.org for more information."), T_OK);
 		} else {
+#if 0
 			show_fancy_dialog(_("Warning:"), _("This copy of X-CD-Roast has been compiled without\nnon-root-support. Therefore you will not be able to use it as\nnon-root user unless you have take special precautions. It is\nrecommended that you install a fully enabled version.\n\nVisit http://www.xcdroast.org for more information."), T_OK);
+#endif
+;
 		}
 		gtk_widget_destroy(toplevel);
 	}
