Index: src/io.c
===================================================================
--- src/io.c.orig	2006-12-19 13:02:15.000000000 +0100
+++ src/io.c	2006-12-19 13:18:53.000000000 +0100
@@ -985,7 +985,11 @@
 
 	/* add the found devices to the writerreader structure */
 	count = 0;
-	g_snprintf(tmp,MAXLINE,"%s:", transport);
+	if (strstr(transport,"ATAPI")) {
+		g_snprintf(tmp,MAXLINE,"OLD%s:", transport);
+	} else {
+		g_snprintf(tmp,MAXLINE,"%s:", transport);
+	}
 	while(scsidevices[count] != NULL) {
 		add_writerreader(scsidevices[count], tmp, txt);
 		g_free(scsidevices[count]);
