diff -up xcdroast-0.98alpha16/src/menus.c.p23 xcdroast-0.98alpha16/src/menus.c
--- xcdroast-0.98alpha16/src/menus.c.p23	2008-08-20 17:41:28.000000000 +0200
+++ xcdroast-0.98alpha16/src/menus.c	2009-02-09 16:55:07.000000000 +0100
@@ -2701,7 +2701,7 @@ static const gchar *blanktrans[] = BLANK
 	gtk_container_set_border_width (GTK_CONTAINER (box1), 5);
         gtk_container_add(GTK_CONTAINER(dialog),box1);
 
-        if (!curset.isProDVD) {
+        if (!curset.isDVD) {
 		f1 = gtk_frame_new(_("Blank CD-RW"));
         } else {
 		f1 = gtk_frame_new(_("Blank CD/DVD+-RW"));
@@ -2811,7 +2811,7 @@ static const gchar *blanktrans[] = BLANK
 	gtk_box_pack_start(GTK_BOX(box1),box3,FALSE,TRUE,5);
 	gtk_widget_show(box3);
 
-        if (!curset.isProDVD) {
+        if (!curset.isDVD) {
 		button1 = gtk_button_new_with_label(_("Blank CD-RW"));
         } else {
 		button1 = gtk_button_new_with_label(_("Blank CD/DVD+-RW"));
diff -up xcdroast-0.98alpha16/src/xcdrdata.h.p23 xcdroast-0.98alpha16/src/xcdrdata.h
--- xcdroast-0.98alpha16/src/xcdrdata.h.p23	2008-08-20 17:57:53.000000000 +0200
+++ xcdroast-0.98alpha16/src/xcdrdata.h	2009-02-09 16:55:07.000000000 +0100
@@ -60,7 +60,9 @@ typedef struct setup_data {
 	gboolean root_option_change_imagedirs;
 	gboolean root_option_change_logoptions;
 
+#ifndef CDRECORDDVD_IS_FREE
 	gchar *ProDVDkey;
+#endif
 	geometry_t mainwindow;
 	geometry_t burnwindow;
 	
@@ -120,7 +122,7 @@ typedef struct writer_driver {
 } writer_driver_t;
 
 typedef struct current_set {
-	gint isProDVD;
+	gint isDVD;
 	gint support_ontheflyaudio;
 	gint writer_devnr;
 	gint reader_devnr;
diff -up xcdroast-0.98alpha16/src/create.c.p23 xcdroast-0.98alpha16/src/create.c
--- xcdroast-0.98alpha16/src/create.c.p23	2008-08-20 17:57:53.000000000 +0200
+++ xcdroast-0.98alpha16/src/create.c	2009-02-09 16:55:07.000000000 +0100
@@ -6792,7 +6792,7 @@ gint sel, btn;
 			fill_session_available();
 			/* force recalc of session size if dvd */
 			/* so he will get the multisession warning */
-			if (curset.cdrtype >= 1000 && ( curset.isProDVD & 1 ) && curset.multisession == 1) {
+			if (curset.cdrtype >= 1000 && ( curset.isDVD & 1 ) && curset.multisession == 1) {
 				set_sessionsize_unknown();	
 			}
 			break;
diff -up xcdroast-0.98alpha16/src/xcdroast.h.p23 xcdroast-0.98alpha16/src/xcdroast.h
--- xcdroast-0.98alpha16/src/xcdroast.h.p23	2008-08-21 12:36:52.000000000 +0200
+++ xcdroast-0.98alpha16/src/xcdroast.h	2009-02-09 17:01:08.000000000 +0100
@@ -112,22 +112,30 @@
    then in $PREFIX (e.g. /usr/bin/cdrecord instead of 
    /usr/local/lib/xcdroast-0.98/bin/cdrecord) */
 
-#define CDRECORD "bin/cdrecord" 
-#define CDDA2WAV "bin/cdda2wav"
-#define READCD "bin/readcd"
-#define MKISOFS "bin/mkisofs"
+#define CDRECORD "bin/wodim" 
+#define CDDA2WAV "bin/icedax"
+#define READCD "bin/readom"
+#define MKISOFS "bin/genisoimage"
+
+#define CDRECORD_NAME "wodim"
+#define CDRECORDDVD_NAME "wodim"
+#define CDRECORDDVD_IS_FREE
+
+#define CDDA2WAV_NAME "icedax"
+#define READCD_NAME "readom"
+#define MKISOFS_NAME "genisoimage"
 
 /* MINIMUM required versions. Greater versions are now silently
    accepted. 
    Distribution-Vendors: DO NOT TOUCH THESE!
    IT WONT WORK WITH LOWER VERSIONS, EVEN WHEN YOU THINK IT DOES! */
  
-#define CDRECORD_VERSION "2.01"
-#define CDRECORD_PRODVD_VERSION "2.01a11"
+#define CDRECORD_VERSION "1.1.9"
+#define CDRECORDDVD_VERSION "2.01a11"
 #define CDRECORD_VERSION_WITH_ONTHFLYAUDIO "2.01a13"
-#define MKISOFS_VERSION "2.01"
-#define CDDA2WAV_VERSION "2.01"
-#define READCD_VERSION "2.01"
+#define MKISOFS_VERSION "1.1.9"
+#define CDDA2WAV_VERSION "1.1.9"
+#define READCD_VERSION "1.1.9"
 
 /* group to use for non-root-mode */
 /* #define NONROOTMODEGROUP "xcdwrite"  - HISTORIC */
diff -up xcdroast-0.98alpha16/src/duplicate.c.p23 xcdroast-0.98alpha16/src/duplicate.c
--- xcdroast-0.98alpha16/src/duplicate.c.p23	2003-10-26 14:14:19.000000000 +0100
+++ xcdroast-0.98alpha16/src/duplicate.c	2009-02-09 16:55:07.000000000 +0100
@@ -1156,7 +1156,7 @@ GtkWidget *scrolled_win;
 	gtk_table_set_col_spacings(GTK_TABLE(tbl),10);
 	gtk_box_pack_start(GTK_BOX(actionspace),tbl,TRUE,TRUE,10);
 	gtk_widget_show(tbl);
-	if (!curset.isProDVD) {
+	if (!curset.isDVD) {
 		f1 = gtk_frame_new(_("CD-Information"));
 	} else {
 		f1 = gtk_frame_new(_("CD/DVD-Information"));
@@ -1739,7 +1739,7 @@ GtkWidget *tbl, *check;
 	gtk_widget_show(tbl);
 	vbox = gtk_vbox_new(FALSE,10);
 	gtk_table_attach_defaults(GTK_TABLE(tbl),vbox,0,1,0,1);
-	if (!curset.isProDVD) {
+	if (!curset.isDVD) {
 		f1 = gtk_frame_new(_("CD-Information"));
 	} else {
 		f1 = gtk_frame_new(_("CD/DVD-Information"));
@@ -2241,7 +2241,7 @@ GtkWidget *tbl, *omenu, *check;
 	gtk_widget_show(tbl);
 	vbox = gtk_vbox_new(FALSE,10);
 	gtk_table_attach_defaults(GTK_TABLE(tbl),vbox,0,1,0,1);
-	if (!curset.isProDVD) {
+	if (!curset.isDVD) {
 		f1 = gtk_frame_new(_("CD-Information"));
 	} else {
 		f1 = gtk_frame_new(_("CD/DVD-Information"));
@@ -3090,7 +3090,7 @@ gint menuidx, menuhistory;
 	gtk_table_set_col_spacings(GTK_TABLE(tbl),10);
 	gtk_box_pack_start(GTK_BOX(actionspace),tbl,TRUE,TRUE,10);
 	gtk_widget_show(tbl);
-	if (!curset.isProDVD) {
+	if (!curset.isDVD) {
 		f1 = gtk_frame_new(_("CD to write"));
 	} else {
 		f1 = gtk_frame_new(_("CD/DVD to write"));
@@ -3107,7 +3107,7 @@ gint menuidx, menuhistory;
 	hbox = gtk_hbox_new(TRUE,10);
 	gtk_box_pack_start(GTK_BOX(actionspace),hbox,FALSE,TRUE,5);
 	gtk_widget_show(hbox);
-	if (!curset.isProDVD) {
+	if (!curset.isDVD) {
 		b1 = gtk_button_new_with_label(_("Write CD"));
 	} else {
 		b1 = gtk_button_new_with_label(_("Write CD/DVD"));
@@ -3118,7 +3118,7 @@ gint menuidx, menuhistory;
 		GTK_SIGNAL_FUNC(writetracks_clicked), NULL);
 	define_tooltip(b1,_("Writes all tracks specified by the TOC file to a CD."));
 
-	if (!curset.isProDVD) {
+	if (!curset.isDVD) {
 		b1 = gtk_button_new_with_label(_("Blank CD-RW"));
 	} else {
 		b1 = gtk_button_new_with_label(_("Blank CD/DVD+-RW"));
@@ -3239,7 +3239,7 @@ gint menuidx, menuhistory;
 		gtk_widget_show (menu_item);
 		i++;
 	}
-	if (curset.isProDVD) {
+	if (curset.isDVD) {
 		i = 0;
 		tmpval = curset.cdrtype;
 		while (dvdtypes[i] != 0) {
@@ -3984,7 +3984,7 @@ GtkWidget *b1;
 	side_widgets[0] = b1;
 	define_tooltip(b1,_("Displays information about the currently inserted CD and tracks stored on the harddrive."));
 
-	if (!curset.isProDVD) {
+	if (!curset.isDVD) {
 		b1 = gtk_toggle_button_new_with_label(_("Read CD"));
 	} else {
 		b1 = gtk_toggle_button_new_with_label(_("Read CD/DVD"));
@@ -3996,7 +3996,7 @@ GtkWidget *b1;
 	side_widgets[1] = b1;
 	define_tooltip(b1,_("Reads all tracks of a CD and copy them to the harddrive."));
 
-	if (!curset.isProDVD) {
+	if (!curset.isDVD) {
 		b1 = gtk_toggle_button_new_with_label(_("Verify CD"));
 	} else {
 		b1 = gtk_toggle_button_new_with_label(_("Verify CD/DVD"));
@@ -4016,7 +4016,7 @@ GtkWidget *b1;
 	side_widgets[3] = b1;
 	define_tooltip(b1,_("Plays audio tracks (.wav files) from the harddrive via soundcard."));
 
-	if (!curset.isProDVD) {
+	if (!curset.isDVD) {
 		b1 = gtk_toggle_button_new_with_label(_("Write CD"));
 	} else {
 		b1 = gtk_toggle_button_new_with_label(_("Write CD/DVD"));
diff -up xcdroast-0.98alpha16/src/main.c.p23 xcdroast-0.98alpha16/src/main.c
--- xcdroast-0.98alpha16/src/main.c.p23	2008-08-21 11:45:34.000000000 +0200
+++ xcdroast-0.98alpha16/src/main.c	2009-02-09 16:55:07.000000000 +0100
@@ -583,7 +583,7 @@ gint stat;
 */
 
         /* warn about non-existing LFS support in dvd mode */
-        if ((curset.isProDVD & 1) && sizeof(off_t) != 8) {
+        if ((curset.isDVD & 1) && sizeof(off_t) != 8) {
                 show_dialog(ICO_WARN, _("Warning: Your system is not able to create files larger\nthan 2 GB. Writing DVDs will therefore not possible."), T_OK, NULL, NULL, 0);
         }
 
