diff -aur ttf2png-0.3.old/Makefile ttf2png-0.3.new/Makefile
--- ttf2png-0.3.old/Makefile	2012-02-14 21:39:39.510252812 -0300
+++ ttf2png-0.3.new/Makefile	2012-02-14 21:40:06.150108491 -0300
@@ -1,5 +1,5 @@
 ttf2png: ttf2png.c
-	gcc -Wall $^ -o $@ $(shell freetype-config --cflags --libs) $(shell pkg-config --cflags --libs libpng12)
+	gcc -Wall $^ -o $@ $(shell freetype-config --cflags --libs) $(shell pkg-config --cflags --libs libpng)
 
 VER=0.3
 
