--- a/cyrussasl.c	2013-04-17 01:49:32.461705182 +0100
+++ b/cyrussasl.c	2013-04-17 02:28:26.442663377 +0100
@@ -813,13 +813,13 @@
 }
 
 /* metatable, hook for calling gc_context on context structs */
-static const luaL_reg meta[] = {
+static const luaL_Reg meta[] = {
   { "__gc", gc_context },
   { NULL,   NULL        }
 };
 
 /* function table for this module */
-static const struct luaL_reg methods[] = {
+static const struct luaL_Reg methods[] = {
   { "setssf",       cyrussasl_setssf            },
   { "setprop",      cyrussasl_sasl_setprop      },
   { "listmech",     cyrussasl_sasl_listmech     },
@@ -841,12 +841,12 @@
 /* Module initializer, called from Lua when the module is loaded. */
 int luaopen_cyrussasl(lua_State *l)
 {
-  /* Construct a new namespace table for Lua, and register it as the global 
+  /* Construct a new namespace table for Lua, and register it as the global
    * named "cyrussasl".
    */
   luaL_openlib(l, MODULENAME, methods, 0);
 
-  /* Create metatable, which is used to tie C data structures to our garbage 
+  /* Create metatable, which is used to tie C data structures to our garbage
    * collection function. */
   luaL_newmetatable(l, MODULENAME);
   luaL_openlib(l, 0, meta, 0);
