diff -Naur kdebase-4.3.80.orig/apps/konsole/data/profiles/Shell.profile kdebase-4.3.80/apps/konsole/data/profiles/Shell.profile
--- kdebase-4.3.80.orig/apps/konsole/data/profiles/Shell.profile	2009-11-21 10:43:48.000000000 +0100
+++ kdebase-4.3.80/apps/konsole/data/profiles/Shell.profile	2009-12-06 20:45:03.000000000 +0100
@@ -161,12 +161,16 @@
 LocalTabTitleFormat=%d : %n
 RemoteTabTitleFormat=%h : %u
 
+[General]
+ShowNewAndCloseTabButtons=true
+
 [Appearance]
+AntiAliasFonts=true
 ColorScheme=Linux
+Font=Bitstream Vera Sans Mono,10,-1,0,75,0,0,0,0,0
 
 [Scrolling]
-ScrollMode=FixedLineCount
-ScrollLineCount=1000
-
-
+HistoryMode=2
 
+[Terminal Features]
+BlinkingCursorEnabled=true
