diff -ur softbeep.orig/Makefile softbeep/Makefile
--- softbeep.orig/Makefile	2002-04-06 01:52:59.000000000 +0400
+++ softbeep/Makefile	2010-01-10 22:55:18.000000000 +0300
@@ -1,4 +1,4 @@
-PREFIX=/usr/local
+PREFIX=/usr
 SO=libsoftbeep.so
 SOURCES=libsoftbeep.c
 CFLAGS=-Wall -g -O2
@@ -23,9 +23,10 @@
 	chmod ug+rx softbeep
 
 install: all
-	install -groot -m755 -oroot softbeep $(PREFIX)/bin
-	install -groot -m755 -oroot sb-beep $(PREFIX)/bin
-	install -groot -m644 -oroot -s $(SO) $(PREFIX)/lib
+	install -d $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/lib
+	install -groot -m755 -oroot softbeep $(DESTDIR)$(PREFIX)/bin
+	install -groot -m755 -oroot sb-beep $(DESTDIR)$(PREFIX)/bin
+	install -groot -m644 -oroot -s $(SO) $(DESTDIR)$(PREFIX)/lib
 
 README: README.in
 	sed s/VERSION/$(VERSION)/ < README.in > README
diff -ur softbeep.orig/sb-beep softbeep/sb-beep
--- softbeep.orig/sb-beep	2002-03-31 17:19:19.000000000 +0400
+++ softbeep/sb-beep	2010-01-10 23:08:22.000000000 +0300
@@ -2,10 +2,14 @@
 
 FILE=/usr/share/sounds/warning.wav
 
-echo "$@" >> /tmp/sb-beep
-
 if [ "$1" == "irssi" ] || [ "$1" == "xchat" ] ; then
 	FILE=/usr/share/sounds/email.wav
 fi
 
-exec esdplay $FILE
+# use the player specified in $SB_PLAYER if defined
+
+if [ -z "$SB_PLAYER" ]; then
+        exec aplay $FILE &> /dev/null &
+else
+        exec $SB_PLAYER $FILE &> /dev/null &
+fi
