diff --git a/configure.ac b/configure.ac
index 2eb82fb..03734df 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,7 @@ AC_COPYRIGHT([Copyright (c) 2011 3scale])
 AC_INIT([libvmod-threescale], [0.1])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR(src/vmod_threescale.vcc)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_CANONICAL_SYSTEM
 AC_LANG(C)
@@ -57,9 +57,9 @@ AC_CHECK_FILE([$VARNISHSRC/include/varnishapi.h],
 )
 
 # Check that varnishtest is built in the varnish source directory
-AC_CHECK_FILE([$VARNISHSRC/bin/varnishtest/varnishtest],
+AC_CHECK_FILE([/usr/bin/varnishtest],
 [],
-[AC_MSG_FAILURE([Can't find "$VARNISHSRC/bin/varnishtest/varnishtest". Please build your varnish source directory])]
+[AC_MSG_FAILURE([Can't find "/usr/bin/varnishtest". Please build your varnish source directory])]
 )
 
 # vmod installation dir
