diff -up original/yweather.bac new/yweather.bac
--- original/yweather.bac	2013-01-07 21:44:58.000000000 +0800
+++ new/yweather.bac	2014-05-13 11:33:52.000000000 +0800
@@ -229,7 +229,6 @@ END FUNCTION
 FUNCTION EXTRACTL$(STRING Main$,STRING Delim$)
 ' ------------------
 	' Similar to LEFT$ but uses a delimiter rather the position
-    LOCAL i TYPE int
     LOCAL sRet$ TYPE STRING
     sRet$ = ""
     IF (LEN(Main$) EQ 0) | (LEN(Delim$) EQ 0) THEN
@@ -246,7 +245,7 @@ END FUNCTION
 FUNCTION EXTRACTR$(STRING Main$,STRING Delim$)
 ' ------------------
 	' Similar to RIGHT$ but uses a delimiter rather the position
-    LOCAL i, j, k, mlen, dlen TYPE int
+    LOCAL j, k, mlen, dlen TYPE int
     LOCAL sRet$ TYPE STRING
     i = 0
     j = 0
@@ -279,7 +278,7 @@ END FUNCTION
 ' --------------
 FUNCTION GETWORD$(STRING mystring$, NUMBER word_number, STRING delimiter$)
 ' --------------
-	LOCAL  i, mypos, n TYPE NUMBER
+	LOCAL mypos, n TYPE NUMBER
 	LOCAL string_temp$ TYPE STRING
 	i = 0
 	mypos  = 0
