--- src/libgtop-2.28.4/sysdeps/linux/fsusage.c	2011-06-01 17:40:47.000000000 +0200
+++ fsusage.c	2012-11-01 22:12:55.344474306 +0100
@@ -41,7 +41,11 @@
 
 	while ((mnt = getmntent(fp)))
 	{
-		if (!strcmp(mountpoint, mnt->mnt_dir)) {
+        /* There can be multiple root mount entries, skip the unuseful one */
+        if (!strcmp(mnt->mnt_fsname, "rootfs"))
+            continue;
+        
+        if (!strcmp(mountpoint, mnt->mnt_dir)) {
 			if (!strncmp(mnt->mnt_fsname, "/dev/", 5)) {
 				g_strlcpy(device, mnt->mnt_fsname + 5, device_size);
 				found = TRUE;
