diff -Naur wp2latex.3.58.orig/sources.cc/images/ras_png.cc wp2latex.3.58.new/sources.cc/images/ras_png.cc
--- wp2latex.3.58.orig/sources.cc/images/ras_png.cc	2012-05-12 15:08:15.000000000 +0200
+++ wp2latex.3.58.new/sources.cc/images/ras_png.cc	2012-05-12 15:16:07.000000000 +0200
@@ -6,7 +6,8 @@
 #include "raster.h"
 
 #include <png.h>
-
+#define png_infopp_NULL (png_infopp)NULL
+#define int_p_NULL (int*)NULL
 
 #define SupportPNG  2
 
