diff --git a/util/grub-mkconfig.in b/util/grub-mkconfig.in
index e317279..a457b26 100644
--- a/util/grub-mkconfig.in
+++ b/util/grub-mkconfig.in
@@ -49,6 +49,7 @@ usage () {
 Usage: $self [OPTION]
 Generate a grub config file
 
+  -p, --prefix=PATH       grub prefix dir [default=/boot/grub]
   -o, --output=FILE       output generated config to FILE [default=stdout]
   -h, --help              print this message and exit
   -v, --version           print the version information and exit
@@ -81,6 +82,11 @@ do
     -v | --version)
 	echo "$self (${PACKAGE_NAME}) ${PACKAGE_VERSION}"
 	exit 0 ;;
+    -p | --prefix)
+	GRUB_PREFIX=`argument $option "$@"`; shift;;
+    --prefix=*)
+	GRUB_PREFIX=`echo "$option" | sed "s,//*,/,g"`
+	;;
     -o | --output)
 	grub_cfg=`argument $option "$@"`; shift;;
     --output=*)
