--- pkg-sed/opt/brother/Printers/dcp195c/cupswrapper/cupswrapperdcp195c	2012-08-31 20:49:40.093666698 +0200
+++ pkg/opt/brother/Printers/dcp195c/cupswrapper/cupswrapperdcp195c	2012-08-31 20:48:14.953673037 +0200
@@ -47,7 +47,9 @@
   rm -f /usr/lib64/cups/filter/brlpdwrapper${printer_model}
   rm -f /opt/brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1
 # rm -f /usr/local/Brother/${device_model}/${printer_model}/cupswrapper/brcupsconfpt1
-if [  -e /etc/rc.d/cups ]; then
+if which systemctl &> /dev/null; then
+    systemctl restart cupsd.service
+elif [  -e /etc/rc.d/cups ]; then
    /etc/rc.d/cups restart
 elif [  -e /etc/rc.d/cupsys ]; then
    /etc/rc.d/cupsys restart
@@ -57,7 +59,9 @@
 fi
 if [ "$1" = "-r" ]; then
   lpadmin -x ${printer_name}
-if [  -e /etc/rc.d/cups ]; then
+if which systemctl &> /dev/null; then
+    systemctl restart cupsd.service
+elif [  -e /etc/rc.d/cups ]; then
    /etc/rc.d/cups restart
 elif [  -e /etc/rc.d/cupsys ]; then
    /etc/rc.d/cupsys restart
@@ -287,7 +291,9 @@
 fi


-if [  -e /etc/rc.d/cups ]; then
+if which systemctl &> /dev/null; then
+    systemctl restart cupsd.service
+elif [  -e /etc/rc.d/cups ]; then
    /etc/rc.d/cups restart
 elif [  -e /etc/rc.d/cupsys ]; then
    /etc/rc.d/cupsys restart
