Index: build_mozc.py
===================================================================
--- build_mozc.py	(revision 130)
+++ build_mozc.py	(working copy)
@@ -214,15 +214,6 @@
     if not PkgExists('ibus-1.0 >= 1.4.1'):
       logging.info('removing ibus.gyp.')
       gyp_file_names.remove('%s/unix/ibus/ibus.gyp' % SRC_DIR)
-    # Check if Qt libraries are installed.
-    qt_found = PkgExists('QtCore', 'QtGui')
-    if options.ensure_value('qtdir', None):
-      qt_found = True
-    if not qt_found:
-      # Fall back to --noqt mode.  Ensure options.noqt is available and
-      # force it to be True.
-      options.ensure_value('noqt', True)
-      options.noqt = True
   if options.target_platform == 'NaCl':
     # Add chrome NaCl Mozc gyp scripts.
     gyp_file_names.append('%s/chrome/nacl/nacl_extension.gyp' % SRC_DIR)
@@ -620,10 +611,18 @@
 
   if options.branding:
     command_line.extend(['-D', 'branding=%s' % options.branding])
+
   if options.noqt:
     command_line.extend(['-D', 'use_qt=NO'])
   else:
-    command_line.extend(['-D', 'use_qt=YES'])
+    if options.target_platform == 'Linux' and not options.qtdir:
+      # Check if Qt libraries are installed.
+      system_qt_found = PkgExists('QtCore >= 4.0', 'QtCore < 5.0',
+                                  'QtGui >= 4.0', 'QtGui < 5.0')
+      if not system_qt_found:
+        PrintErrorAndExit('Qt4 is required to build GUI Tool. '
+                          'Specify --noqt to skip building GUI Tool.')
+      command_line.extend(['-D', 'use_qt=YES'])
   if options.qtdir:
     command_line.extend(['-D', 'qt_dir=%s' % os.path.abspath(options.qtdir)])
   else:
