diff -bruN src.orig/mupdf/CMakeLists-fitz.txt src/mupdf/CMakeLists-fitz.txt
--- src.orig/mupdf/CMakeLists-fitz.txt	2013-01-08 21:29:35.788524000 +0900
+++ src/mupdf/CMakeLists-fitz.txt	2013-03-23 12:01:26.933197499 +0900
@@ -134,6 +134,9 @@
   fitz/stm_comp_buf.c
   fitz/stm_open.c
   fitz/stm_read.c
+  fitz/stm_output.c
+  pdf/pdf_device.c
+  pdf/pdf_field.c
   pdf/pdf_annot.c
   pdf/pdf_cmap.c
   pdf/pdf_cmap_load.c
diff -bruN src.orig/mupdf/get-mupdf.sh src/mupdf/get-mupdf.sh
--- src.orig/mupdf/get-mupdf.sh	2012-12-12 20:55:24.974995000 +0900
+++ src/mupdf/get-mupdf.sh	2013-03-23 12:59:08.003048698 +0900
@@ -2,7 +2,7 @@
 if [ -d mupdf/.git ]; then
   (cd mupdf ; git pull)
 else
-  git clone git://git.ghostscript.com/mupdf.git
+  git clone --depth=1 git://git.ghostscript.com/mupdf.git
 fi
 if [ ! -L mupdf/CMakeLists.txt ]; then
   ln -s ../CMakeLists-fitz.txt mupdf/CMakeLists.txt
diff -bruN src.orig/mupdf/qmupdf.cpp src/mupdf/qmupdf.cpp
--- src.orig/mupdf/qmupdf.cpp	2012-12-12 23:20:58.245276000 +0900
+++ src/mupdf/qmupdf.cpp	2013-03-23 12:14:20.076497592 +0900
@@ -334,7 +334,8 @@
 
 QSizeF Page::size() const
 {
-    return convert_fz_rect(pdf_bound_page(d->doc->xref, d->page)).size();
+	fz_rect rect;
+	return convert_fz_rect(*pdf_bound_page(d->doc->xref, d->page, &rect)).size();
 }
 
 qreal Page::duration() const
@@ -348,15 +349,15 @@
 {
     const QSizeF s = size();
 
-    fz_matrix ctm;
+	fz_matrix ctm, tmp;
     ctm = fz_identity;
-    ctm = fz_concat(ctm, fz_scale(width / s.width(), height / s.height()));
+	fz_concat(&ctm, &ctm, fz_scale(&tmp, width / s.width(), height / s.height()));
 
     fz_cookie cookie = { 0, 0, 0, 0 };
     fz_pixmap *image = fz_new_pixmap(d->doc->ctxt, fz_device_rgb, width, height);
     fz_clear_pixmap_with_value(d->doc->ctxt, image, 0xff);
     fz_device *device = fz_new_draw_device(d->doc->ctxt, image);
-    pdf_run_page(d->doc->xref, d->page, device, ctm, &cookie);
+	pdf_run_page(d->doc->xref, d->page, device, &ctm, &cookie);
     fz_free_device(device);
 
     QImage img;
@@ -374,10 +375,11 @@
     fz_matrix ctm;
     ctm = fz_identity;
     fz_cookie cookie = { 0, 0, 0, 0 };
-    fz_text_page *page = fz_new_text_page(d->doc->ctxt, pdf_bound_page(d->doc->xref, d->page));
+	fz_rect rect;
+	fz_text_page *page = fz_new_text_page(d->doc->ctxt, pdf_bound_page(d->doc->xref, d->page, &rect));
     fz_text_sheet *sheet = fz_new_text_sheet(d->doc->ctxt);
     fz_device *device = fz_new_text_device(d->doc->ctxt, sheet, page);
-    pdf_run_page(d->doc->xref, d->page, device, ctm, &cookie);
+	pdf_run_page(d->doc->xref, d->page, device, &ctm, &cookie);
     fz_free_device(device);
     if (cookie.errors) {
         fz_free_text_page(d->doc->ctxt, page);
