=== modified file 'libutouch-geis/backend/xcb/geis_xcb_backend.c'
--- a/libutouch-geis/backend/xcb/geis_xcb_backend.c	2011-03-23 17:48:10 +0000
+++ b/libutouch-geis/backend/xcb/geis_xcb_backend.c	2011-03-24 02:18:16 +0000
@@ -110,6 +110,7 @@
   GeisBoolean is_direct = GEIS_FALSE;
   GeisBoolean is_independent = GEIS_FALSE;
 
+#ifdef XInput_2_1
   if (xi2_mode == XIDirectTouch)
     is_direct = GEIS_TRUE;
 
@@ -139,6 +140,7 @@
   {
     geis_device_add_attr(geis_device, device_attr);
   }
+#endif
 }
 
 
@@ -727,6 +729,7 @@
   for (class_index = 0; class_index < xcb_device->num_classes; ++class_index)
   {
     XIAnyClassInfo *any = xcb_device->classes[class_index];
+#ifdef XInput_2_1
     if (any->type == XITouchClass)
     {
       XITouchClassInfo *v = (XITouchClassInfo *)any;
@@ -796,6 +799,7 @@
 	geis_device_add_attr(geis_device, device_attr);
       }
     }
+#endif
   }
   geis_post_event(be->geis, device_event);
   goto final_exit;
@@ -829,11 +833,13 @@
          ++class_index)
     {
       XIAnyClassInfo *any = devices[device_index].classes[class_index];
+#ifdef XInput_2_1
       if (any->type == XITouchClass)
       {
 	_report_an_xcb_device(be, &devices[device_index]);
 	break;
       }
+#endif
     }
   }
 

