=== modified file 'services/panel-service.c'
--- services/panel-service.c	2012-12-13 07:47:24 +0000
+++ services/panel-service.c	2013-02-04 21:12:57 +0000
@@ -108,7 +108,7 @@
   {"libapplication.so", "gsd-keyboard-xkb"},  /* keyboard layout selector */
   {"libmessaging.so", NULL},                  /* indicator-messages */
   {"libpower.so", NULL},                      /* indicator-power */
-  {"libapplication.so", "bluetooth-manager"}, /* bluetooth manager */
+  {"libbluetooth.so", NULL},                  /* indicator-bluetooth */
   {"libnetwork.so", NULL},                    /* indicator-network */
   {"libnetworkmenu.so", NULL},                /* indicator-network */
   {"libapplication.so", "nm-applet"},         /* network manager */

