=== modified file 'panel/PanelMenuView.cpp'
--- panel/PanelMenuView.cpp	2012-11-08 09:10:21 +0000
+++ panel/PanelMenuView.cpp	2012-12-03 09:35:59 +0000
@@ -1591,7 +1591,7 @@
   const gchar *title = g_variant_get_string(data, 0);
   _panel_title = (title ? title : "");
 
-  Refresh();
+  Refresh(true);
   QueueDraw();
 }
 

=== modified file 'tests/autopilot/unity/tests/launcher/test_keynav.py'
--- tests/autopilot/unity/tests/launcher/test_keynav.py	2012-09-13 10:56:42 +0000
+++ tests/autopilot/unity/tests/launcher/test_keynav.py	2012-12-03 09:35:59 +0000
@@ -234,3 +234,8 @@
         self.assertThat(self.dash.visible, Eventually(Equals(True)))
         self.assertThat(self.launcher.key_nav_is_active, Eventually(Equals(False)))
 
+    def test_launcher_keynav_changes_panel(self):
+        """The panel title must change when in key nav mode."""
+
+        self.start_keynav_with_cleanup_cancel()
+        self.assertThat(self.panels.get_active_panel().title, Eventually(Equals("Dash Home")))

