diff -cr original/panel/PanelMenuView.cpp modified/panel/PanelMenuView.cpp
*** original/panel/PanelMenuView.cpp	2013-10-10 18:08:30.000000000 +0100
--- modified/panel/PanelMenuView.cpp	2014-04-01 11:04:49.976988324 +0100
***************
*** 72,78 ****
    , new_app_menu_shown_(false)
    , monitor_(0)
    , active_xid_(0)
!   , desktop_name_(_("Ubuntu Desktop"))
    , menus_fadein_(DEFAULT_MENUS_FADEIN)
    , menus_fadeout_(DEFAULT_MENUS_FADEOUT)
    , menus_discovery_(DEFAULT_MENUS_DISCOVERY)
--- 72,78 ----
    , new_app_menu_shown_(false)
    , monitor_(0)
    , active_xid_(0)
!   , desktop_name_(_("Unity Desktop"))
    , menus_fadein_(DEFAULT_MENUS_FADEIN)
    , menus_fadeout_(DEFAULT_MENUS_FADEOUT)
    , menus_discovery_(DEFAULT_MENUS_DISCOVERY)
Only in modified/panel: PanelMenuView.cpp~
diff -cr original/tests/autopilot/unity/tests/launcher/test_keynav.py modified/tests/autopilot/unity/tests/launcher/test_keynav.py
*** original/tests/autopilot/unity/tests/launcher/test_keynav.py	2013-10-10 18:08:30.000000000 +0100
--- modified/tests/autopilot/unity/tests/launcher/test_keynav.py	2014-04-01 11:06:10.459827138 +0100
***************
*** 152,158 ****
          self.launcher_instance.key_nav_activate()
          self.addCleanup(self.keybinding, "expo/cancel")
  
!         self.assertThat(self.unity.panels.get_active_panel().title, Eventually(Equals("Ubuntu Desktop")))
  
      def test_launcher_keynav_expo_exit_on_esc(self):
          """Esc should quit expo when entering it from KeyNav."""
--- 152,158 ----
          self.launcher_instance.key_nav_activate()
          self.addCleanup(self.keybinding, "expo/cancel")
  
!         self.assertThat(self.unity.panels.get_active_panel().title, Eventually(Equals("Unity Desktop")))
  
      def test_launcher_keynav_expo_exit_on_esc(self):
          """Esc should quit expo when entering it from KeyNav."""
Only in modified/tests/autopilot/unity/tests/launcher: test_keynav.py~
