diff -Naur unity-5.0.0.orig/CMakeLists.txt unity-5.0.0/CMakeLists.txt
--- unity-5.0.0.orig/CMakeLists.txt	2012-01-12 17:47:07.000000000 +0100
+++ unity-5.0.0/CMakeLists.txt	2012-01-15 09:45:17.010953206 +0100
@@ -119,7 +119,7 @@
 #
 # Compiz Plugins
 #
-set (UNITY_PLUGIN_DEPS "compiz;nux-2.0 >= 2.0.0;libbamf3;dee-1.0;gio-2.0;gio-unix-2.0;dbusmenu-glib-0.4;x11;libstartup-notification-1.0;gthread-2.0;indicator3-0.4;atk;unity-misc >= 0.4.0;gconf-2.0;libutouch-geis;gtk+-3.0 >= 3.1;sigc++-2.0;json-glib-1.0;libnotify;gnome-desktop-3.0;gdu")
+set (UNITY_PLUGIN_DEPS "compiz;nux-2.0 >= 2.0.0;libbamf3;dee-1.0;gio-2.0;gio-unix-2.0;dbusmenu-glib-0.4;x11;libstartup-notification-1.0;gthread-2.0;indicator3-0.4;atk;unity-misc >= 0.4.0;gconf-2.0;gtk+-3.0 >= 3.1;sigc++-2.0;json-glib-1.0;libnotify;gnome-desktop-3.0;gdu")
 
 add_subdirectory(plugins/unityshell)
 add_subdirectory(plugins/gtkloader)
diff -Naur unity-5.0.0.orig/plugins/unityshell/src/Launcher.cpp unity-5.0.0/plugins/unityshell/src/Launcher.cpp
--- unity-5.0.0.orig/plugins/unityshell/src/Launcher.cpp	2012-01-12 17:47:07.000000000 +0100
+++ unity-5.0.0/plugins/unityshell/src/Launcher.cpp	2012-01-15 09:55:01.696575516 +0100
@@ -192,11 +192,6 @@
   plugin_adapter.compiz_screen_viewport_switch_started.connect(sigc::mem_fun(this, &Launcher::OnViewPortSwitchStarted));
   plugin_adapter.compiz_screen_viewport_switch_ended.connect(sigc::mem_fun(this, &Launcher::OnViewPortSwitchEnded));
 
-  GeisAdapter& adapter = *(GeisAdapter::Default());
-  adapter.drag_start.connect(sigc::mem_fun(this, &Launcher::OnDragStart));
-  adapter.drag_update.connect(sigc::mem_fun(this, &Launcher::OnDragUpdate));
-  adapter.drag_finish.connect(sigc::mem_fun(this, &Launcher::OnDragFinish));
-
   display.changed.connect(sigc::mem_fun(this, &Launcher::OnDisplayChanged));
 
   _current_icon       = NULL;
@@ -402,50 +397,6 @@
 }
 
 void
-Launcher::OnDragStart(GeisAdapter::GeisDragData* data)
-{
-  if (_drag_out_id && _drag_out_id == data->id)
-    return;
-
-  if (data->touches == 4)
-  {
-    _drag_out_id = data->id;
-    if (_hidden)
-    {
-      _drag_out_delta_x = 0.0f;
-    }
-    else
-    {
-      _drag_out_delta_x = DRAG_OUT_PIXELS;
-      _hide_machine->SetQuirk(LauncherHideMachine::MT_DRAG_OUT, false);
-    }
-  }
-}
-
-void
-Launcher::OnDragUpdate(GeisAdapter::GeisDragData* data)
-{
-  if (data->id == _drag_out_id)
-  {
-    _drag_out_delta_x = CLAMP(_drag_out_delta_x + data->delta_x, 0.0f, DRAG_OUT_PIXELS);
-    EnsureAnimation();
-  }
-}
-
-void
-Launcher::OnDragFinish(GeisAdapter::GeisDragData* data)
-{
-  if (data->id == _drag_out_id)
-  {
-    if (_drag_out_delta_x >= DRAG_OUT_PIXELS - 90.0f)
-      _hide_machine->SetQuirk(LauncherHideMachine::MT_DRAG_OUT, true);
-    SetTimeStruct(&_times[TIME_DRAG_OUT], &_times[TIME_DRAG_OUT], ANIM_DURATION_SHORT);
-    _drag_out_id = 0;
-    EnsureAnimation();
-  }
-}
-
-void
 Launcher::startKeyNavMode()
 {
   SetStateKeyNav(true);
diff -Naur unity-5.0.0.orig/plugins/unityshell/src/Launcher.h unity-5.0.0/plugins/unityshell/src/Launcher.h
--- unity-5.0.0.orig/plugins/unityshell/src/Launcher.h	2012-01-12 17:47:07.000000000 +0100
+++ unity-5.0.0/plugins/unityshell/src/Launcher.h	2012-01-15 09:51:33.363116092 +0100
@@ -32,7 +32,6 @@
 #include "BackgroundEffectHelper.h"
 #include "DNDCollectionWindow.h"
 #include "DndData.h"
-#include "GeisAdapter.h"
 #include "Introspectable.h"
 #include "LauncherDragWindow.h"
 #include "LauncherHideMachine.h"
@@ -246,10 +245,6 @@
   void OnWindowMapped(guint32 xid);
   void OnWindowUnmapped(guint32 xid);
 
-  void OnDragStart(GeisAdapter::GeisDragData* data);
-  void OnDragUpdate(GeisAdapter::GeisDragData* data);
-  void OnDragFinish(GeisAdapter::GeisDragData* data);
-
   void OnPluginStateChanged();
 
   void OnViewPortSwitchStarted();
diff -Naur unity-5.0.0.orig/plugins/unityshell/src/unityshell.cpp unity-5.0.0/plugins/unityshell/src/unityshell.cpp
--- unity-5.0.0.orig/plugins/unityshell/src/unityshell.cpp	2012-01-12 17:47:07.000000000 +0100
+++ unity-5.0.0/plugins/unityshell/src/unityshell.cpp	2012-01-15 09:49:39.521844903 +0100
@@ -29,7 +29,6 @@
 #include "Launcher.h"
 #include "LauncherIcon.h"
 #include "LauncherController.h"
-#include "GeisAdapter.h"
 #include "DevicesSettings.h"
 #include "PluginAdapter.h"
 #include "QuicklistManager.h"
@@ -97,7 +96,6 @@
   , screen(screen)
   , cScreen(CompositeScreen::get(screen))
   , gScreen(GLScreen::get(screen))
-  , gestureEngine(nullptr)
   , wt(nullptr)
   , panelWindow(nullptr)
   , debugger(nullptr)
@@ -311,9 +309,6 @@
      g_idle_add_full (G_PRIORITY_DEFAULT, &UnityScreen::initPluginActions, this, NULL);
      super_keypressed_ = false;
 
-     GeisAdapter::Default()->Run();
-     gestureEngine = new GestureEngine(screen);
-
      CompString name(PKGDATADIR"/panel-shadow.png");
      CompString pname("unityshell");
      CompSize size(1, 20);
diff -Naur unity-5.0.0.orig/plugins/unityshell/src/unityshell.h unity-5.0.0/plugins/unityshell/src/unityshell.h
--- unity-5.0.0.orig/plugins/unityshell/src/unityshell.h	2012-01-12 17:47:07.000000000 +0100
+++ unity-5.0.0/plugins/unityshell/src/unityshell.h	2012-01-15 09:49:39.521844903 +0100
@@ -43,7 +43,6 @@
 #include "PanelController.h"
 #include "PanelStyle.h"
 #include "UScreen.h"
-#include "GestureEngine.h"
 #include "DebugDBusInterface.h"
 #include "SwitcherController.h"
 #include "UBusWrapper.h"
@@ -253,7 +252,6 @@
   panel::Controller::Ptr    panel_controller_;
   switcher::Controller::Ptr switcher_controller_;
 
-  GestureEngine*                        gestureEngine;
   nux::WindowThread*                    wt;
   nux::BaseWindow*                      panelWindow;
   nux::Geometry                         lastTooltipArea;
diff -Naur unity-5.0.0.orig/standalone-clients/CMakeLists.txt unity-5.0.0/standalone-clients/CMakeLists.txt
--- unity-5.0.0.orig/standalone-clients/CMakeLists.txt	2012-01-12 17:47:07.000000000 +0100
+++ unity-5.0.0/standalone-clients/CMakeLists.txt	2012-01-15 09:46:58.576139550 +0100
@@ -235,8 +235,6 @@
                 ${UNITY_SRC}/LauncherHoverMachine.h
                 ${UNITY_SRC}/LauncherDragWindow.cpp
                 ${UNITY_SRC}/LauncherDragWindow.h
-                ${UNITY_SRC}/GeisAdapter.cpp
-                ${UNITY_SRC}/GeisAdapter.h
                 ${UNITY_SRC}/WindowManager.h
                 ${UNITY_SRC}/WindowManager.cpp
                 ${UNITY_SRC}/IconRenderer.cpp
