diff -Naur unity-6.12.0.orig/launcher/LauncherIcon.cpp unity-6.12.0/launcher/LauncherIcon.cpp
--- unity-6.12.0.orig/launcher/LauncherIcon.cpp	2012-11-06 15:34:13.000000000 +0100
+++ unity-6.12.0/launcher/LauncherIcon.cpp	2013-03-10 10:38:28.999256173 +0100
@@ -361,7 +361,7 @@
 
   // FIXME: we need to create some kind of -unity postfix to see if we are looking to the unity-icon-theme
   // for dedicated unity icons, then remove the postfix and degrade to other icon themes if not found
-  if (icon_name == "workspace-switcher" && IsMonoDefaultTheme())
+  if (icon_name == "workspace-switcher")
     result = TextureFromSpecificGtkTheme(GetUnityTheme(), icon_name, size, update_glow_colors);
 
   if (!result)
