diff -Naur unity-6.12.0.orig/plugins/unityshell/src/DebugDBusInterface.cpp unity-6.12.0/plugins/unityshell/src/DebugDBusInterface.cpp
--- unity-6.12.0.orig/plugins/unityshell/src/DebugDBusInterface.cpp	2012-11-23 00:11:02.856475000 +0100
+++ unity-6.12.0/plugins/unityshell/src/DebugDBusInterface.cpp	2012-11-23 00:11:23.230260777 +0100
@@ -18,6 +18,7 @@
  */
 
 #include <queue>
+#include <iostream>
 #include <fstream>
 #include <sstream>
 #include <boost/algorithm/string.hpp>
