diff -cr original/services/CMakeLists.txt modified/services/CMakeLists.txt
*** original/services/CMakeLists.txt	2013-10-10 18:08:30.000000000 +0100
--- modified/services/CMakeLists.txt	2014-04-01 09:06:01.300778556 +0100
***************
*** 12,20 ****
      indicator3-0.4>=12.10.2
      x11
      libido3-0.1>=13.0.0
-     libupstart
-     libnih
-     libnih-dbus
      dbus-1
  )
  
--- 12,17 ----
***************
*** 55,60 ****
  add_executable(unity-panel-service ${PANEL_SOURCES})
  target_link_libraries(unity-panel-service ${LIBS})
  install(TARGETS unity-panel-service DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/unity/)
- 
- configure_file(unity-panel-service.conf.in ${CMAKE_CURRENT_BINARY_DIR}/unity-panel-service.conf)
- install(FILES ${CMAKE_CURRENT_BINARY_DIR}/unity-panel-service.conf DESTINATION ${CMAKE_INSTALL_PREFIX}/share/upstart/sessions)
--- 52,54 ----
Only in modified/services: CMakeLists.txt~
diff -cr original/services/panel-service.c modified/services/panel-service.c
*** original/services/panel-service.c	2013-10-10 18:08:30.000000000 +0100
--- modified/services/panel-service.c	2014-04-01 09:08:55.413042343 +0100
***************
*** 33,41 ****
  #include <X11/XKBlib.h>
  #include <X11/extensions/XInput2.h>
  
- #include <upstart.h>
- #include <nih/alloc.h>
- 
  G_DEFINE_TYPE (PanelService, panel_service, G_TYPE_OBJECT);
  
  #define GET_PRIVATE(o) \
--- 33,38 ----
***************
*** 80,87 ****
  
    IndicatorObjectEntry *pressed_entry;
    gboolean use_event;
- 
-   NihDBusProxy * upstart;
  };
  
  /* Globals */
--- 77,82 ----
***************
*** 148,165 ****
    g_idle_remove_by_data (self);
    gdk_window_remove_filter (NULL, (GdkFilterFunc)event_filter, self);
  
-   if (priv->upstart != NULL)
-     {
-       int event_sent = 0;
-       event_sent = upstart_emit_event_sync (NULL, priv->upstart,
-                                             "indicator-services-end", NULL, 0);
-       if (event_sent != 0)
-          g_warning("Unable to signal for indicator services to start");
- 
-       nih_unref (priv->upstart, NULL);
-       priv->upstart = NULL;
-     }
- 
    if (GTK_IS_WIDGET (priv->menubar) &&
        gtk_widget_get_realized (GTK_WIDGET (priv->menubar)))
      {
--- 143,148 ----
***************
*** 541,560 ****
    return FALSE;
  }
  
- static gboolean
- ready_signal (PanelService *self)
- {
-   if (PANEL_IS_SERVICE (self) && self->priv->upstart != NULL)
-     {
-       int event_sent = 0;
-       event_sent = upstart_emit_event_sync (NULL, self->priv->upstart, "indicator-services-start", NULL, 0);
-       if (event_sent != 0)
-          g_warning ("Unable to signal for indicator services to start");
-     }
- 
-   return FALSE;
- }
- 
  static void
  update_keybinding (GSettings *settings, const gchar *key, gpointer data)
  {
--- 524,529 ----
***************
*** 670,692 ****
    update_keybinding (priv->gsettings, MENU_TOGGLE_KEYBINDING_KEY, &priv->menu_toggle);
    update_keybinding (priv->gsettings, SHOW_DASH_KEY, &priv->show_dash);
    update_keybinding (priv->gsettings, SHOW_HUD_KEY, &priv->show_hud);
- 
-   const gchar *upstartsession = g_getenv ("UPSTART_SESSION");
-   if (upstartsession != NULL)
-     {
-       DBusConnection *conn = dbus_connection_open (upstartsession, NULL);
-       if (conn != NULL)
-         {
-           priv->upstart = nih_dbus_proxy_new (NULL, conn,
-                                               NULL,
-                                               DBUS_PATH_UPSTART,
-                                               NULL, NULL);
-           dbus_connection_unref (conn);
-         }
-     }
- 
-   if (priv->upstart != NULL)
-     priv->upstart->auto_start = FALSE;
  }
  
  static gboolean
--- 639,644 ----
***************
*** 829,835 ****
    suppress_signals = FALSE;
  
    g_idle_add ((GSourceFunc)initial_resync, self);
-   g_idle_add ((GSourceFunc)ready_signal, self);
  }
  
  PanelService *
--- 781,786 ----
Only in modified/services: panel-service.c~
diff -cr original/services/unity-panel-service.conf.in modified/services/unity-panel-service.conf.in
*** original/services/unity-panel-service.conf.in	2013-10-10 18:08:30.000000000 +0100
--- modified/services/unity-panel-service.conf.in	2014-04-01 09:09:11.459610440 +0100
***************
*** 1,18 ****
- description "Backing Service for the Unity Panel"
- author "Ted Gould <ted@ubuntu.com>"
- 
- # These should be used when we have Unity full under upstart, but while
- # we have the pre-start work around for gnome-session starting it we're
- # going to comment them out.
- #
- # start on started unity7
- # stop on stopped unity7
- 
- start on desktop-start DESKTOP_SESSION=ubuntu
- stop on desktop-end
- 
- emits indicator-services-start
- emits indicator-services-end
- 
- respawn
- exec ${CMAKE_INSTALL_PREFIX}/lib/unity/unity-panel-service
--- 0 ----
Only in modified/services: unity-panel-service.conf.in~
diff -cr original/tests/CMakeLists.txt modified/tests/CMakeLists.txt
*** original/tests/CMakeLists.txt	2013-10-10 18:08:30.000000000 +0100
--- modified/tests/CMakeLists.txt	2014-04-01 09:10:02.029299798 +0100
***************
*** 36,42 ****
  #
  # Unit tests
  #
! set (TEST_DEPS "${UNITY_PLUGIN_DEPS};unity>=4.99.0;libupstart;libnih;libnih-dbus;dbus-1")
  pkg_check_modules (TEST_UNIT_DEPS REQUIRED ${TEST_DEPS})
  
  string (REPLACE ";" " " TEST_UNIT_DEPS_CFLAGS_OTHER "${TEST_UNIT_CFLAGS_OTHER}")
--- 36,42 ----
  #
  # Unit tests
  #
! set (TEST_DEPS "${UNITY_PLUGIN_DEPS};unity>=4.99.0;dbus-1")
  pkg_check_modules (TEST_UNIT_DEPS REQUIRED ${TEST_DEPS})
  
  string (REPLACE ";" " " TEST_UNIT_DEPS_CFLAGS_OTHER "${TEST_UNIT_CFLAGS_OTHER}")
Only in modified/tests: CMakeLists.txt~
