diff -cr original/CMakeLists.txt modified/CMakeLists.txt
*** original/CMakeLists.txt	2013-10-10 18:08:30.000000000 +0100
--- modified/CMakeLists.txt	2014-04-01 09:19:11.402591749 +0100
***************
*** 195,201 ****
      SET (BOOT_LOGGER_FLAG "-DENABLE_LOGGER")
  endif (BOOT_LOGGER)
  
! SET (MAINTAINER_CXXFLAGS "-Werror -Wall -Wcast-align -Wempty-body -Wformat-security -Winit-self -Warray-bounds")
  
  option (DISABLE_ERROR_ON_LOCAL_TYPEDEFS_WARNINGS "Disable errors when local typedefs are unused" ON)
  if (DISABLE_ERROR_ON_LOCAL_TYPEDEFS_WARNINGS)
--- 195,201 ----
      SET (BOOT_LOGGER_FLAG "-DENABLE_LOGGER")
  endif (BOOT_LOGGER)
  
! SET (MAINTAINER_CXXFLAGS "-Wall -Wcast-align -Wempty-body -Wformat-security -Winit-self -Warray-bounds")
  
  option (DISABLE_ERROR_ON_LOCAL_TYPEDEFS_WARNINGS "Disable errors when local typedefs are unused" ON)
  if (DISABLE_ERROR_ON_LOCAL_TYPEDEFS_WARNINGS)
Only in modified: CMakeLists.txt~
diff -cr original/services/CMakeLists.txt modified/services/CMakeLists.txt
*** original/services/CMakeLists.txt	2013-10-10 18:08:30.000000000 +0100
--- modified/services/CMakeLists.txt	2014-04-01 09:20:05.012262434 +0100
***************
*** 38,44 ****
  set(CFLAGS
      ${SERVICE_DEPS_CFLAGS}
      ${SERVICE_DEPS_CFLAGS_OTHER}
!     "-Werror -Wall"
      )
  
  string (REPLACE ";" " " CFLAGS "${CFLAGS}")
--- 38,44 ----
  set(CFLAGS
      ${SERVICE_DEPS_CFLAGS}
      ${SERVICE_DEPS_CFLAGS_OTHER}
!     "-Wall"
      )
  
  string (REPLACE ";" " " CFLAGS "${CFLAGS}")
Only in modified/services: CMakeLists.txt~
