diff -Nru lightdm-1.4.0.orig/debian/guest-account lightdm-1.4.0/debian/guest-account
--- lightdm-1.4.0.orig/debian/guest-account	2013-01-09 03:38:59.881423049 -0500
+++ lightdm-1.4.0/debian/guest-account	2013-01-09 04:15:16.693442050 -0500
@@ -117,6 +117,14 @@
       . /etc/guest-session/prefs.sh
   fi
 
+  # Load default dconf settings
+  if [ -d /etc/guest-session/gsettings/ ]; then
+    for i in /etc/guest-session/gsettings/*; do
+      cat "${i}" | sed '/^\[.*\]$/ s/\./\//g' | \
+        su - $USER -c 'dbus-launch dconf load /'
+    done
+  fi
+
   echo $USER  
 }
 
