--- state.c.old	2012-01-04 21:21:45.230816194 -0800
+++ state.c	2012-01-04 21:28:21.497138767 -0800
@@ -92,10 +92,14 @@
 			++content_count;
 		}
 
-		if (content_count < state->level+1) {
-			fprintf(stderr, "You must have at least %d 'content' files in different disks.\n", state->level+1);
+		if (content_count < 1) {
+			fprintf(stderr, "*** ERROR : Must have at least one 'content' file configured.\n");
 			exit(EXIT_FAILURE);
 		}
+		if (content_count < state->level+1) {
+			fprintf(stderr, "*** WARNING : Only %d 'content' files are configured\n", content_count);
+ 			fprintf(stderr, "SUGGESTION : You should have at least %d 'content' files in different disks.\n", state->level+1);
+		}
 	}
 
 	/* check if all the data and parity disks are different */
