--- ftplugin/php/phpErrorMarker.vim.old	2010-01-28 01:44:59.546374768 +0100
+++ ftplugin/php/phpErrorMarker.vim	2010-01-28 01:45:49.455540583 +0100
@@ -27,7 +27,6 @@
 
 		command -buffer MarkPhpErrors call phpErrorMarker#markErrors()
 		command -buffer UnmarkPhpErrors call phpErrorMarker#unmarkErrors()
-		command -nargs=0 PhpErrorMarkerVimball call phpErrorMarker#makeVimball()
 
 		let &cpo = s:cpo
 		unlet s:cpo
