--- caffeine-build/caffeine/main.py     2011-11-11 05:12:50 +0000
+++ caffeine-build/caffeine/main.py     2011-11-20 01:40:33 +0000
@@ -324,5 +324,7 @@
     def on_R_click(self, status_icon, mbutton, time, data=None):
         ## popdown menu
-        self.menu.popup(None, None,
-                Gtk.status_icon_position_menu, 3, time, self.status_icon)
+        self.menu.show_all()
+        def func(menu, user_data):
+            return status_icon.position_menu(self.menu, status_icon)
+        self.menu.popup(None, None, func, self.status_icon, 3, time)

