diff -crB plasma-widget-quickaccess-0.8.1/popupdialog.cpp plasma-widget-quickaccess-0.8.1_patched/popupdialog.cpp
*** plasma-widget-quickaccess-0.8.1/popupdialog.cpp	2009-05-27 21:02:59.000000000 +0200
--- plasma-widget-quickaccess-0.8.1_patched/popupdialog.cpp	2011-06-25 16:04:54.694094678 +0200
***************
*** 247,256 ****
    m_view->verticalScrollBar()->setValue(0);
    m_label->setFileItem(m_current);
    m_backButton->hide();
!   if(m_settings->enableDolphinSorting())
!   {
!     checkDolphinSorting(&(m_current.url()));
!   }
    QWidget::hideEvent( event );
    emit signal_hide();
  }
--- 247,256 ----
    m_view->verticalScrollBar()->setValue(0);
    m_label->setFileItem(m_current);
    m_backButton->hide();
!   //if(m_settings->enableDolphinSorting())
!   //{
!   //  checkDolphinSorting(&(m_current.url()));
!   //}
    QWidget::hideEvent( event );
    emit signal_hide();
  }
***************
*** 322,329 ****
      m_current = m_start;
    }
  
!   if(m_settings->enableDolphinSorting())
!     checkDolphinSorting(&(m_current.url()));
  
    if(m_current.isFile() || !m_settings->allowNavigation()) {
      m_current.run();
--- 322,329 ----
      m_current = m_start;
    }
  
!   //if(m_settings->enableDolphinSorting())
!   //  checkDolphinSorting(&(m_current.url()));
  
    if(m_current.isFile() || !m_settings->allowNavigation()) {
      m_current.run();
