--- shedskin/__init__.py.old	2010-10-19 19:37:41.704036667 +0200
+++ shedskin/__init__.py	2010-10-19 19:37:56.067370004 +0200
@@ -36,7 +36,7 @@
 
     # --- some checks
     major, minor = sys.version_info[:2]
-    if (major, minor) not in [(2, 4), (2, 5), (2, 6)]:
+    if (major, minor) not in [(2, 4), (2, 5), (2, 6), (2, 7)]:
         print '*ERROR* Shed Skin is not compatible with this version of Python'
         sys.exit(1)
     if sys.platform == 'win32' and os.path.isdir('c:/mingw'):
