=== modified file 'example/Makefile.am'
--- example/Makefile.am	2010-12-04 04:03:03 +0000
+++ example/Makefile.am	2010-12-04 04:03:03 +0000
@@ -11,9 +11,13 @@
 
 simple_client_CFLAGS = \
 	$(LIBRARY_CFLAGS) \
+	-DLOCAL_ICON="\"$(abs_srcdir)/simple-client-test-icon.png\"" \
 	-Wall -Werror \
 	-I$(top_srcdir)/src
 
 simple_client_LDADD = \
 	$(LIBRARY_LIBS) \
 	$(top_builddir)/src/libappindicator.la
+
+EXTRA_DIST = \
+	simple-client-test-icon.png

=== added file 'example/simple-client-test-icon.png'
Binary files example/simple-client-test-icon.png	1970-01-01 00:00:00 +0000 and example/simple-client-test-icon.png	2010-12-04 04:03:03 +0000 differ
=== modified file 'example/simple-client.c'
--- example/simple-client.c	2010-08-05 19:45:50 +0000
+++ example/simple-client.c	2010-12-04 04:03:03 +0000
@@ -60,6 +60,20 @@
 }
 
 static void
+local_icon_toggle_cb (GtkWidget *widget, gpointer data)
+{
+	AppIndicator * ci = APP_INDICATOR(data);
+
+	if (gtk_check_menu_item_get_active(GTK_CHECK_MENU_ITEM(widget))) {
+		app_indicator_set_icon(ci, LOCAL_ICON);
+	} else {
+		app_indicator_set_icon(ci, "indicator-messages");
+	}
+
+	return;
+}
+
+static void
 item_clicked_cb (GtkWidget *widget, gpointer data)
 {
   const gchar *text = (const gchar *)data;
@@ -192,6 +206,12 @@
         gtk_menu_shell_append (GTK_MENU_SHELL (menu), item);
 		gtk_widget_show(item);
 
+        item = gtk_check_menu_item_new_with_label ("Set Local Icon");
+        g_signal_connect (item, "activate",
+                          G_CALLBACK (local_icon_toggle_cb), ci);
+        gtk_menu_shell_append (GTK_MENU_SHELL (menu), item);
+		gtk_widget_show(item);
+
         app_indicator_set_menu (ci, GTK_MENU (menu));
 
 	mainloop = g_main_loop_new(NULL, FALSE);

