Index: Makefile
===================================================================
--- Makefile	(revision 377)
+++ Makefile	(working copy)
@@ -135,8 +135,10 @@
 
 # build targets and rules
 
-all:       setup libs $(TRANSLATE_DRIVER_EXE) $(AML_DRIVER_EXE) $(EXPANSION_DRIVER_EXE) $(EVAL_DRIVER_EXE) $(AML2MAD_ETC_EXE)
+all:       setup libs antlr $(TRANSLATE_DRIVER_EXE) $(AML_DRIVER_EXE) $(EXPANSION_DRIVER_EXE) $(EVAL_DRIVER_EXE) $(AML2MAD_ETC_EXE)
 
+noantlr:   setup libs $(TRANSLATE_DRIVER_EXE) $(AML_DRIVER_EXE) $(EXPANSION_DRIVER_EXE) $(EVAL_DRIVER_EXE) $(AML2MAD_ETC_EXE)
+
 aml:       setup libs $(AML_DRIVER_EXE) 
 
 expansion: setup libs $(EXPANSION_DRIVER_EXE)
@@ -153,8 +155,10 @@
 
 unittests: setup libs $(UNIT_TESTS_EXE)
 
-libs:      setup lib/libantlr.a lib/libuap.a modules/uap_fortran.mod
+libs:      setup lib/libuap.a modules/uap_fortran.mod
 
+antlr:     setup lib/libantlr.a
+
 setup:
 	@mkdir -p bin lib modules
 
@@ -224,7 +228,7 @@
 	@echo "Building ANTLR 2.7.0 library"
 	@ar cr $@ $(ANTLR_OBJ)
 
-lib/libuap.a: $(UAP_OBJ) lib/libantlr.a
+lib/libuap.a: $(UAP_OBJ) 
 	@echo "Building UAP library"
 	@ar cr $@ $(UAP_OBJ)
 
