From 1e0e472c43781fca3a14c0a799cdc54dd336625c Mon Sep 17 00:00:00 2001
From: Timofey Titovets <nefelim4ag@gmail.com>
Date: Thu, 5 Feb 2015 01:04:59 +0300
Subject: [PATCH] fix_python2

---
 atom.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/atom.sh b/atom.sh
index ecd7da0..89d6cd8 100755
--- a/atom.sh
+++ b/atom.sh
@@ -77,6 +77,8 @@ elif [ $OS == 'Linux' ]; then
   ATOM_PATH="$USR_DIRECTORY/share/atom/atom"
   ATOM_HOME="${ATOM_HOME:-$HOME/.atom}"
 
+  export PYTHON=/usr/bin/python2
+
   mkdir -p "$ATOM_HOME"
 
   : ${TMPDIR:=/tmp}
-- 
2.2.2

