--- ./gcc/gcc.c.orig	2012-04-02 12:45:18.000000000 +0200
+++ ./gcc/gcc.c	2012-04-02 13:04:52.000000000 +0200
@@ -3106,13 +3106,10 @@
   fputs (_("  -print-prog-name=<prog>  Display the full path to compiler component <prog>\n"), stdout);
   fputs (_("  -print-multi-directory   Display the root directory for versions of libgcc\n"), stdout);
 #ifdef _PIC30_H_
-  fputs (_("\
-  -fast-math               Use alternative floating point support routines\n"), stdout);
-  fputs (_("\
-  -legacy-libc             Use legacy (pre v3.25) lib C routines\n"), stdout);
+  fputs (_("  -fast-math               Use alternative floating point support routines\n"), stdout);
+  fputs (_("  -legacy-libc             Use legacy (pre v3.25) lib C routines\n"), stdout);
 #endif
-  fputs (_("\
-  -print-multi-lib         Display the mapping between command line options and\n\
+  fputs (_("  -print-multi-lib         Display the mapping between command line options and\n\
                            multiple library search directories\n"), stdout);
   fputs (_("  -print-multi-os-directory Display the relative path to OS libraries\n"), stdout);
   fputs (_("  -Wa,<options>            Pass comma-separated <options> on to the assembler\n"), stdout);
@@ -3412,7 +3409,7 @@
 #endif
         {
           const char *startp, *endp;
-          char *nstore = (char *) alloca (strlen (temp) + 3);
+          char *nstore = (char *) alloca (strlen (temp) + 3 + strlen(base_path) + strlen(dir_separator_str));
 
           startp = endp = temp;
           while (1)
@@ -3423,6 +3420,7 @@
                     strcpy (nstore, concat (".", dir_separator_str, NULL));
                   else if (IS_DIR_SEPARATOR(*startp)) {
                     strncpy (nstore, startp, endp - startp);
+		    nstore[endp-startp] = 0;
                   } else {
                     nstore[0] = 0;
                     strcat(nstore, base_path);
