--- gcc-4.0.2/gcc-4.0.2/libiberty/make-relative-prefix.c.orig	2008-12-13 11:25:52.000000000 +0800
+++ ./libiberty/make-relative-prefix.c	2008-12-13 11:28:55.000000000 +0800
@@ -321,7 +321,9 @@
 	    break;
 	}
 
-      if (prog_num <= 0 || i == bin_num)
+      // kunilkuda-131208: Force function to specify relative directories
+      // even if we're installed in standard location
+      if (prog_num <= 0)
 	{
 	  free_split_directories (prog_dirs);
 	  free_split_directories (bin_dirs);
