--- ./gcc/config.gcc.orig	2009-01-31 18:53:24.000000000 +0100
+++ ./gcc/config.gcc	2009-01-31 19:19:35.000000000 +0100
@@ -1564,6 +1564,7 @@
         host_can_use_collect2=no
         host_xm_defines="PROGRAM_PREFIX=\\\"${program_prefix}\\\" ${host_xm_defines}"
         extra_modes=pic30/pic30-modes.h
+	tm_file="pic30/defines.h ${tm_file}"
         LANGUAGES="c"
         LIBGCCMK=
         ;;
@@ -1573,6 +1574,7 @@
         tm_defines="OBJECT_FORMAT_ELF $tm_defines"
         host_xm_defines="PROGRAM_PREFIX=\\\"${program_prefix}\\\" ${host_xm_defines}"
         extra_modes=pic30/pic30-modes.h
+	tm_file="pic30/defines.h ${tm_file}"
         LANGUAGES="c"
         LIBGCCMK=
         ;;
--- /dev/null	2009-01-31 13:17:10.543334184 +0100
+++ ./gcc/config/pic30/defines.h	2009-01-31 19:22:32.000000000 +0100
@@ -0,0 +1,23 @@
+#define	MPLABC30_COMMON_INCLUDE_PATH	"/usr/pic30/include"
+#define	MPLABC30_PIC24F_INCLUDE_PATH	"/usr/pic30/include/PIC24F"
+#define	MPLABC30_PIC24H_INCLUDE_PATH	"/usr/pic30/include/PIC24H"
+#define	MPLABC30_PIC30F_INCLUDE_PATH	"/usr/pic30/include/PIC30F"
+#define	MPLABC30_PIC33F_INCLUDE_PATH	"/usr/pic30/include/PIC33F"
+#define	MPLABC30_COMMON_LIB_PATH	"/usr/pic30/lib"
+#define	MPLABC30_PIC24F_LIB_PATH	"/usr/pic30/lib/PIC24F"
+#define	MPLABC30_PIC24H_LIB_PATH	"/usr/pic30/lib/PIC24H"
+#define	MPLABC30_PIC30F_LIB_PATH	"/usr/pic30/lib/PIC30F"
+#define	MPLABC30_PIC33F_LIB_PATH	"/usr/pic30/lib/PIC33F"
+/*#define	STANDARD_INCLUDE_DIR		"/usr/pic30/include"*/
+/* 
+ * Major simplification hack.
+ * I just hope this does the trick....
+ * /Karro
+ */
+#undef INCLUDE_DEFAULTS
+#define INCLUDE_DEFAULTS			\
+  {						\
+    { "/usr/pic30/include", "GCC", 0, 0 },		\
+    { 0, 0, 0, 0 }				\
+  }
+
