--- ./c30_resource/src/generator/resource.c.~1~	2008-02-20 18:19:36.000000000 +0100
+++ ./c30_resource/src/generator/resource.c	2009-01-31 07:30:30.000000000 +0100
@@ -65,29 +65,7 @@
   unsigned int i;
 
   buffer_size = 80;
-  if (stat(name,&fileinfo) == -1) {
-    /* name cannot be statted */
-    /* this could be because of the reoganization of the install directory 
-     * some of the binaries are in the same directory as the resource file,
-     * some are one directory below it... try moving one up and looking again
-     */
-    char *c;
-    char *ultimate = (char *)name,
-         *penultimate = (char *)name;
-    char *temp;
-
-    for (c = (char *)name; *c; c++) {
-      if ((*c == '/') || (*c == '\\')) {
-        penultimate = ultimate;
-        ultimate = c;
-        while (c[1] == *c) c++;  /* ignore duplicate slashes */
-      }
-    }
-    temp = (char *)xcalloc(strlen(ultimate)+1,1);
-    strcpy(temp, ultimate);
-    strcpy(penultimate,ultimate);
-    stat(name,&fileinfo);
-  }
+  stat(name,&fileinfo);
   /* under WINDOZE, the "b" is required because text files are subject to
      translation and ftell and fseek might not agree */
   input_file = fopen(name,"rb");
