--- ../../../orig/gcc-4.0.2/gcc-4.0.2/gcc/config/pic30/t-pic30	2011-01-23 11:46:42.000000000 +0100
+++ ./gcc/config/pic30/t-pic30	2011-01-23 11:52:51.000000000 +0100
@@ -1,8 +1,9 @@
-LIBGCC1 = libgcc1.null
-CROSS_LIBGCC1 = libgcc1.null
-
-# forget the libgcc1...
-LIBGCC1_TEST =
-
-LIBGCC2 = libgcc1.null
-
+# Replacement t-pic30, by John Steele Scott
+# 22/2/2004
+# The original t-pic30 file from Microchip C30 1.30 has no effect, this one
+# lets the build succeed by disabling the libgcc target.
+ 
+LIBGCC = 
+INSTALL_LIBGCC = 
+ 
+CROSS_SYSTEM_HEADER_DIR = /usr/share/pic30/include
