--- macutils/hexbin/buffer.c.orig	2005-12-27 02:41:46.000000000 +0100
+++ macutils/hexbin/buffer.c	2005-12-27 02:41:17.000000000 +0100
@@ -1,12 +1,11 @@
+
+#include <stdlib.h>
+
 #include "globals.h"
 #include "../util/util.h"
 #include "buffer.h"
 #include "../fileio/wrfile.h"
 
-extern char *malloc();
-extern char *realloc();
-extern void exit();
-
 char *data_fork, *rsrc_fork;
 int data_size, rsrc_size;
 static int max_data_size, max_rsrc_size;

--- macutils/macunpack/lzh.h.orig	2005-12-27 02:41:45.000000000 +0100
+++ macutils/macunpack/lzh.h	2005-12-27 02:24:19.000000000 +0100
@@ -58,10 +58,4 @@
 #define	lz5	7
 #define	lzs	8
 
-extern char *lzh_pointer;
-extern char *lzh_data;
-extern char *lzh_finfo;
-extern int lzh_fsize;
-extern int lzh_kind;
-extern char *lzh_file;
 
