diff -Naur gnome-bluetooth-3.6.1.orig/applet/bluetooth-applet.desktop.in.in gnome-bluetooth-3.6.1/applet/bluetooth-applet.desktop.in.in
--- gnome-bluetooth-3.6.1.orig/applet/bluetooth-applet.desktop.in.in	2012-11-15 16:58:31.000000000 +0100
+++ gnome-bluetooth-3.6.1/applet/bluetooth-applet.desktop.in.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,14 +0,0 @@
-[Desktop Entry]
-_Name=Bluetooth Manager
-_Comment=Bluetooth Manager applet
-Icon=bluetooth
-Exec=bluetooth-applet
-Terminal=false
-Type=Application
-Categories=
-OnlyShowIn=GNOME;XFCE;LXDE;
-X-GNOME-Bugzilla-Bugzilla=GNOME
-X-GNOME-Bugzilla-Product=gnome-bluetooth
-X-GNOME-Bugzilla-Component=applet
-X-GNOME-Bugzilla-Version=@VERSION@
-AutostartCondition=GNOME3 if-session gnome-fallback
diff -Naur gnome-bluetooth-3.6.1.orig/applet/indicator-bluetooth.desktop.in.in gnome-bluetooth-3.6.1/applet/bluetooth-indicator.desktop.in.in
--- gnome-bluetooth-3.6.1.orig/applet/indicator-bluetooth.desktop.in.in	1970-01-01 01:00:00.000000000 +0100
+++ gnome-bluetooth-3.6.1/applet/indicator-bluetooth.desktop.in.in	2012-11-24 22:59:47.321399610 +0100
@@ -0,0 +1,14 @@
+[Desktop Entry]
+_Name=Bluetooth Manager
+_Comment=Bluetooth Manager applet
+Icon=bluetooth
+Exec=indicator-bluetooth
+Terminal=false
+Type=Application
+Categories=
+OnlyShowIn=GNOME;
+X-GNOME-Bugzilla-Bugzilla=GNOME
+X-GNOME-Bugzilla-Product=gnome-bluetooth
+X-GNOME-Bugzilla-Component=applet
+X-GNOME-Bugzilla-Version=@VERSION@
+AutostartCondition=GNOME3 if-session unity
diff -Naur gnome-bluetooth-3.6.1.orig/applet/Makefile.am gnome-bluetooth-3.6.1/applet/Makefile.am
--- gnome-bluetooth-3.6.1.orig/applet/Makefile.am	2012-11-24 22:56:02.146418000 +0100
+++ gnome-bluetooth-3.6.1/applet/Makefile.am	2012-11-24 23:10:37.115777689 +0100
@@ -1,10 +1,10 @@
 include Makefile-lib.am
 
-bin_PROGRAMS = bluetooth-applet
+bin_PROGRAMS = indicator-bluetooth
 
-bluetooth_applet_SOURCES = main.c notify.h notify.c agent.h agent.c
+indicator_bluetooth_SOURCES = main.c notify.h notify.c agent.h agent.c
 
-bluetooth_applet_LDADD = $(top_builddir)/lib/libgnome-bluetooth.la $(builddir)/libgnome-bluetooth-applet.la $(APPLET_LIBS)
+indicator_bluetooth_LDADD = $(top_builddir)/lib/libgnome-bluetooth.la $(builddir)/libgnome-bluetooth-applet.la $(APPLET_LIBS)
 
 noinst_PROGRAMS = test-agentdialog test-icon
 
@@ -19,7 +19,7 @@
 INCLUDES = -I$(top_srcdir)/lib
 
 if HAVE_APP_INDICATOR
-bluetooth_applet_LDADD += $(APP_INDICATOR_LIBS)
+indicator_bluetooth_LDADD += $(APP_INDICATOR_LIBS)
 test_agentdialog_LDADD += $(APP_INDICATOR_LIBS)
 test_icon_LDADD += $(APP_INDICATOR_LIBS)
 INCLUDES += -DHAVE_APP_INDICATOR $(APP_INDICATOR_CFLAGS)
@@ -32,8 +32,8 @@
 
 autostartdir = $(sysconfdir)/xdg/autostart
 
-autostart_in_in_files = bluetooth-applet.desktop.in.in
-autostart_in_files = bluetooth-applet.desktop.in
+autostart_in_in_files = indicator-bluetooth.desktop.in.in
+autostart_in_files = indicator-bluetooth.desktop.in
 
 autostart_DATA = $(autostart_in_files:.desktop.in=.desktop)
 
diff -Naur gnome-bluetooth-3.6.1.orig/configure.ac gnome-bluetooth-3.6.1/configure.ac
--- gnome-bluetooth-3.6.1.orig/configure.ac	2012-11-24 22:56:02.149751000 +0100
+++ gnome-bluetooth-3.6.1/configure.ac	2012-11-24 23:09:22.937469002 +0100
@@ -184,7 +184,7 @@
 	  lib/Makefile
 	  lib/plugins/Makefile
 	  applet/Makefile
-	  applet/bluetooth-applet.desktop.in
+	  applet/indicator-bluetooth.desktop.in
 	  wizard/Makefile
 	  wizard/bluetooth-wizard.desktop.in
 	  sendto/Makefile
diff -Naur gnome-bluetooth-3.6.1.orig/po/POTFILES.in gnome-bluetooth-3.6.1/po/POTFILES.in
--- gnome-bluetooth-3.6.1.orig/po/POTFILES.in	2012-11-24 22:56:02.149751000 +0100
+++ gnome-bluetooth-3.6.1/po/POTFILES.in	2012-11-24 23:16:29.676915026 +0100
@@ -9,7 +9,7 @@
 applet/main.c
 applet/notify.c
 applet/agent.c
-applet/bluetooth-applet.desktop.in.in
+applet/indicator-bluetooth.desktop.in.in
 applet/com.canonical.indicator.bluetooth.gschema.xml.in
 [type: gettext/glade] applet/popup-menu.ui
 [type: gettext/glade] applet/authorisation-dialogue.ui
