diff -aur kde-plasma-cpufrequtility-1.4/CMakeLists.txt kde-plasma-cpufrequtility-1.4/CMakeLists.txt
--- kde-plasma-cpufrequtility-1.4/CMakeLists.txt	2011-08-29 19:54:46.000000000 +0700
+++ kde-plasma-cpufrequtility-1.4/CMakeLists.txt	2011-08-30 23:17:03.158319568 +0700
@@ -19,13 +19,13 @@
 	      contents/icons/performance.png
 	      contents/icons/powersave.png)
 set(IKDE4_DATA_DIR share/kde4)
-set(IKDE4_APP_DIR ${IKDE4_DATA_DIR}/apps)
+set(IKDE4_APP_DIR share/apps)
 set(IKDE4_SERVICE_DIR ${IKDE4_DATA_DIR}/services)
-set(IKDE4_PLASMOID_DIR ${IKDE4_APP_DIR}/plasma/plasmoids/${PROJECT_NAME})
+set(IKDE4_PLASMOID_DIR ${IKDE4_APP_DIR}/${PROJECT_NAME})
 if (${CMAKE_INSTALL_PREFIX} STREQUAL /usr)
- set(IKDE4_LIBEXEC_DIR libexec/kde4)
+ set(IKDE4_LIBEXEC_DIR lib/kde4)
 else ()
- set(IKDE4_LIBEXEC_DIR /usr/libexec/kde4)
+ set(IKDE4_LIBEXEC_DIR /usr/lib/kde4)
 endif ()
 
 kde4_add_executable(cpu_freq_helper ${cpu_freq_helper_SRCS})
