diff -rup src/common/inbound.c src/common/inbound.c
--- src/common/inbound.c	2007-06-20 05:48:42.000000000 -0300
+++ src/common/inbound.c	2008-01-14 17:06:48.000000000 -0300
@@ -220,7 +220,7 @@ inbound_privmsg (server *serv, char *fro
 static int
 SearchNick (char *text, char *nicks)
 {
-	char S[300];	/* size of irc_extra_hilight in xchatprefs */
+	char S[1000];	/* size of irc_extra_hilight in xchatprefs */
 	char *n;
 	char *p;
 	char *t;
diff -rup src/common/xchat.h src/common/xchat.h
--- src/common/xchat.h	2007-06-18 22:54:46.000000000 -0300
+++ src/common/xchat.h	2008-01-14 17:06:24.000000000 -0300
@@ -124,7 +124,7 @@ struct xchatprefs
 	char background[PATHLEN + 1];
 	char dccdir[PATHLEN + 1];
 	char dcc_completed_dir[PATHLEN + 1];
-	char irc_extra_hilight[300];
+	char irc_extra_hilight[1000];
 	char irc_no_hilight[300];
 	char irc_nick_hilight[300];
 	char dnsprogram[72];
