diff -upNr athcool-0.3.12.orign/Makefile athcool-0.3.12/Makefile
--- athcool-0.3.12.orign/Makefile	2007-11-05 13:47:46.000000000 +0100
+++ athcool-0.3.12/Makefile	2008-02-11 00:22:53.000000000 +0100
@@ -20,7 +20,7 @@ CFLAGS	= -O2 -Wall
 DEFS    = -I. -I$(includedir) -DPACKAGE=\"$(PACKAGE)\" -DVERSION=\"$(VERSION)\"
 #DEFS   += -DENABLE_FORCEID=1
 #DEFS   += -DDISABLE_WRITE_REG=1
-LIBS	= -lpci
+LIBS	= -lpci -lz
 
 ### rules ###
 
