--- ./src/aqualung/src/decoder/dec_lavc.c.orig	2013-01-21 23:11:26.074432616 +0100
+++ ./src/aqualung/src/decoder/dec_lavc.c	2013-01-21 23:12:38.314782104 +0100
@@ -112,7 +112,7 @@
 	file_decoder_t * fdec = dec->fdec;
 	int i;
 
-	if (av_open_input_file(&pd->avFormatCtx, filename, NULL, 0, NULL) != 0)
+    if (avformat_open_input(&pd->avFormatCtx, filename, NULL, NULL) != 0)
 		return DECODER_OPEN_BADLIB;
 
 	if (av_find_stream_info(pd->avFormatCtx) < 0)
