diff -rupN phreeqpy0.1/phreeqpy/iphreeqc/phreeqc_dll.py new/phreeqpy/iphreeqc/phreeqc_dll.py
--- phreeqpy0.1/phreeqpy/iphreeqc/phreeqc_dll.py	2011-05-25 15:00:38.000000000 +0200
+++ new/phreeqpy/iphreeqc/phreeqc_dll.py	2011-12-19 20:44:11.000000000 +0100
@@ -18,7 +18,7 @@ class IPhreeqc(object):
         if sys.platform == 'win32':
             dll_name = 'IPhreeqc.dll'
         elif sys.platform == 'linux2':
-            dll_name = './libiphreeqc.so.0.0.0'
+            dll_name = '/usr/lib/libiphreeqc.so'
         else:
             msg = 'Platform is %s not supported' % sys.platform
             raise NotImplementedError(msg)
