--- a/setup.py	2011-06-26 15:43:53.606745495 -0400
+++ b/setup.py	2011-06-26 15:43:43.283312861 -0400
@@ -24,14 +24,15 @@
 
 
 def get_yesno( prompt, default ):
-    while 1:
-        answer = raw_input(prompt)
-        if answer == '':
-            return default
-        elif answer[0] == 'Y' or answer[0] == 'y':
-            return 1
-        elif answer[0] == 'N' or answer[0] == 'n':
-            return 0
+	return 1
+    #while 1:
+        #answer = raw_input(prompt)
+        #if answer == '':
+        #    return default
+        #elif answer[0] == 'Y' or answer[0] == 'y':
+        #    return 1
+        #elif answer[0] == 'N' or answer[0] == 'n':
+        #    return 0
 
 try:
     import Image
@@ -68,9 +69,9 @@
     print
     print 'Enter the full pathname of the Ghostscript interpreter,'
     print 'or "+foo" to search the PATH for "foo" at runtime:'
-    gs = raw_input( '[%s] ' % (default,) )
-    if gs == '':
-        gs = default
+    #gs = raw_input( '[%s] ' % (default,) )
+    #if gs == '':
+    gs = default
 
     f = open( 'epstoimg', 'w' )
     f.write( '#!/usr/bin/python\n' )
