diff -Nur xdg-utils-1.1.0-rc1.orig/scripts/xdg-email xdg-utils-1.1.0-rc1/scripts/xdg-email
--- xdg-utils-1.1.0-rc1.orig/scripts/xdg-email	2011-01-01 20:03:14.000000000 +1000
+++ xdg-utils-1.1.0-rc1/scripts/xdg-email	2011-03-06 20:18:04.886429689 +1000
@@ -509,21 +509,25 @@
     fi
 }
 
-open_generic()
+open_generic_browser()
 {
     IFS=":"
     for browser in $BROWSER; do
         if [ x"$browser" != x"" ]; then
-
+	    
             browser_with_arg=`printf "$browser" "$1" 2>/dev/null`
-            if [ $? -ne 0 ]; then browser_with_arg=$browser;
+            if [ $? -ne 0 ]; then 
+		browser_with_arg=$browser;
             fi
 
-            if [ x"$browser_with_arg" = x"$browser" ]; then "$browser" "$1";
-            else $browser_with_arg;
+            if [ x"$browser_with_arg" = x"$browser" ]; then 
+		"$browser" "$1";
+            else 
+		$browser_with_arg;
             fi
-
-            if [ $? -eq 0 ]; then exit_success;
+	    
+            if [ $? -eq 0 ]; then 
+		exit_success;
             fi
         fi
     done
@@ -531,6 +535,21 @@
     exit_failure_operation_impossible "no method available for opening '$1'"
 }
 
+open_generic()
+{
+    if [ ! -z "$XDG_GENERIC_EMAIL" ]; then
+        $XDG_GENERIC_EMAIL "$1"
+
+        if [ $? -eq 0 ]; then
+            exit_success
+        else
+            exit_failure_operation_failed
+        fi
+    else
+        open_generic_browser $@
+    fi
+}
+
 url_encode()
 {
 result=$(echo "$1" | $utf8 | awk '
@@ -703,6 +722,30 @@
     fi
 fi
 
+# check system config
+if [ ! -z "$XDG_CONFIG_DIRS" ]; then
+    IFS=":"
+    for dir in $XDG_CONFIG_DIRS
+    do
+        cfg="$dir/xdg-utils/xdg-utils.rc"
+        if [ -e "$cfg" ]; then
+            source "$cfg"
+            break;
+        fi
+    done
+fi
+# check users config
+if [ ! -z "$XDG_CONFIG_HOME" ]; then
+    cfg="$XDG_CONFIG_HOME/xdg-utils/xdg-utils.rc"
+    if [ -e "$cfg" ]; then
+        source "$cfg"
+    fi
+fi
+# Update BROWSER variable, based on config file
+if [ ! -z "$XDG_GENERIC_BROWSER" ]; then
+    BROWSER="$XDG_GENERIC_BROWSER:$BROWSER"
+fi
+
 case "$DE" in
     kde)
     open_kde "${mailto}"
diff -Nur xdg-utils-1.1.0-rc1.orig/scripts/xdg-mime xdg-utils-1.1.0-rc1/scripts/xdg-mime
--- xdg-utils-1.1.0-rc1.orig/scripts/xdg-mime	2011-01-01 20:03:14.000000000 +1000
+++ xdg-utils-1.1.0-rc1/scripts/xdg-mime	2011-03-06 20:53:59.947273327 +1000
@@ -217,7 +217,11 @@
 
 Synopsis
 
-xdg-mime query { filetype | default } ...
+xdg-mime query { filetype | default[ --command] } ...
+	filetype -- detect MIME type of file
+	default  -- print default handler for MIME
+	            with option --command - print binary command instead
+	            desktop entry
 
 xdg-mime default application mimetype(s)
 
@@ -669,9 +673,35 @@
     mv ${default_file}.new $default_file
 }
 
+# This handles backslashes but not quote marks.
+first_word()
+{
+    read first rest
+    echo "$first"
+}
+
+desktop_file_to_binary()
+{
+    search="${XDG_DATA_HOME:-$HOME/.local/share}:${XDG_DATA_DIRS:-/usr/local/share:/usr/share}"
+    desktop="`basename "$1"`"
+    IFS=:
+    for dir in $search; do
+        unset IFS
+        [ "$dir" -a -d "$dir/applications" ] || continue
+        file="$dir/applications/$desktop"
+        [ -r "$file" ] || continue
+        # Remove any arguments (%F, %f, %U, %u, etc.).
+        command="`grep -E "^Exec(\[[^]=]*])?=" "$file" | cut -d= -f 2- | first_word`"
+        command="`which "$command"`"
+        readlink -f "$command"
+        return
+    done
+}
+
 defapp_generic()
 {
     MIME="$1"
+    ARG="$2"
     xdg_user_dir="$XDG_DATA_HOME"
     [ -n "$xdg_user_dir" ] || xdg_user_dir="$HOME/.local/share"
     xdg_user_dir="$xdg_user_dir/$xdg_dir_name"
@@ -682,8 +712,12 @@
        DEBUG 2 "Checking $x/applications/defaults.list"
        trader_result=`grep "$MIME=" $x/applications/defaults.list 2> /dev/null | cut -d '=' -f 2 | cut -d ';' -f 1`
        if [ -n "$trader_result" ] ; then
-          echo $trader_result
-          exit_success
+	   if [ "$ARG" == "--command" ]; then
+	       desktop_file_to_binary "$trader_result"
+	   else
+               echo $trader_result
+	   fi
+           exit_success
        fi
     done
     exit_success
@@ -756,6 +790,7 @@
       default)
         action=defapp
         mimetype="$2"
+	mimetype_arg="$3"
         if [ -z "$mimetype" ] ; then
             exit_failure_syntax "mimetype argument missing"
         fi
@@ -864,7 +899,7 @@
         ;;
 
         *)
-        defapp_generic "$mimetype"
+        defapp_generic "$mimetype" "$mimetype_arg"
         ;;
     esac
     exit_failure_operation_impossible "no method available for quering default application for '$mimetype'"
diff -Nur xdg-utils-1.1.0-rc1.orig/scripts/xdg-open xdg-utils-1.1.0-rc1/scripts/xdg-open
--- xdg-utils-1.1.0-rc1.orig/scripts/xdg-open	2011-01-01 20:03:14.000000000 +1000
+++ xdg-utils-1.1.0-rc1/scripts/xdg-open	2011-03-06 20:27:54.136450426 +1000
@@ -33,6 +33,8 @@
 #
 #---------------------------------------------
 
+oldIFS=$IFS
+
 manualpage()
 {
 cat << _MANUALPAGE
@@ -409,8 +411,39 @@
     fi
 }
 
+
+open_generic_browser()
+{    
+    IFS=":"
+    for browser in $BROWSER; do
+        if [ x"$browser" != x"" ]; then
+
+            browser_with_arg=`printf "$browser" "$1" 2>/dev/null`
+            if [ $? -ne 0 ]; then
+                browser_with_arg=$browser;
+            fi
+
+            if [ x"$browser_with_arg" = x"$browser" ]; then
+                "$browser" "$1";
+            else eval '$browser_with_arg'$xdg_redirect_output;
+            fi
+
+            if [ $? -eq 0 ]; then
+                exit_success;
+            fi
+        fi
+    done
+}
+
 open_generic()
 {
+    if [ ! -z "$XDG_GENERIC_OPEN" ]; then
+        $XDG_GENERIC_OPEN "$1"
+        if [ $? -eq 0 ]; then
+            exit_success
+        fi
+    fi
+
     # Paths or file:// URLs
     if (echo "$1" | grep -q '^file://' ||
         ! echo "$1" | egrep -q '^[a-zA-Z+\.\-]+:'); then
@@ -442,25 +475,7 @@
         fi
     fi
 
-    IFS=":"
-    for browser in $BROWSER; do
-        if [ x"$browser" != x"" ]; then
-
-            browser_with_arg=`printf "$browser" "$1" 2>/dev/null`
-            if [ $? -ne 0 ]; then
-                browser_with_arg=$browser;
-            fi
-
-            if [ x"$browser_with_arg" = x"$browser" ]; then
-                "$browser" "$1";
-            else eval '$browser_with_arg'$xdg_redirect_output;
-            fi
-
-            if [ $? -eq 0 ]; then
-                exit_success;
-            fi
-        fi
-    done
+    open_generic_browser "$1"
 
     exit_failure_operation_impossible "no method available for opening '$1'"
 }
@@ -516,6 +531,15 @@
     exit_failure_syntax "file or URL argument missing"
 fi
 
+if which xdg-open-hook.sh > /dev/null 2> /dev/null; then
+    xdg-email-hook.sh "${mailto}"
+    if [ $? -eq 0 ]; then
+        exit_success
+    else
+        exit_failure_operation_failed
+    fi
+fi
+
 detectDE
 
 if [ x"$DE" = x"" ]; then
@@ -530,6 +554,30 @@
     fi
 fi
 
+# check system config
+if [ ! -z "$XDG_CONFIG_DIRS" ]; then
+    IFS=":"
+    for dir in $XDG_CONFIG_DIRS
+    do
+        cfg="$dir/xdg-utils/xdg-utils.rc"
+        if [ -e "$cfg" ]; then
+            source "$cfg"
+            break;
+        fi
+    done
+fi
+# check users config
+if [ ! -z "$XDG_CONFIG_HOME" ]; then
+    cfg="$XDG_CONFIG_HOME/xdg-utils/xdg-utils.rc"
+    if [ -e "$cfg" ]; then
+        source "$cfg"
+    fi
+fi
+# Update BROWSER variable, based on config file
+if [ ! -z "$XDG_GENERIC_BROWSER" ]; then
+    BROWSER="$XDG_GENERIC_BROWSER:$BROWSER"
+fi
+
 case "$DE" in
     kde)
     open_kde "$url"
diff -Nur xdg-utils-1.1.0-rc1.orig/scripts/xdg-settings xdg-utils-1.1.0-rc1/scripts/xdg-settings
--- xdg-utils-1.1.0-rc1.orig/scripts/xdg-settings	2011-01-01 20:03:14.000000000 +1000
+++ xdg-utils-1.1.0-rc1/scripts/xdg-settings	2011-03-06 20:35:22.903937128 +1000
@@ -859,6 +859,35 @@
 
 detectDE
 
+if [ x"$DE" = x"" ]; then
+    DE=generic
+fi
+
+# check system config
+if [ ! -z "$XDG_CONFIG_DIRS" ]; then
+    IFS=":"
+    for dir in $XDG_CONFIG_DIRS
+    do
+        cfg="$dir/xdg-utils/xdg-utils.rc"
+        if [ -e "$cfg" ]; then
+            source "$cfg"
+            break;
+        fi
+    done
+fi
+# check users config
+if [ ! -z "$XDG_CONFIG_HOME" ]; then
+    cfg="$XDG_CONFIG_HOME/xdg-utils/xdg-utils.rc"
+    if [ -e "$cfg" ]; then
+        source "$cfg"
+    fi
+fi
+# Update BROWSER variable, based on config file
+# TODO: do more clean
+if [ ! -z "$XDG_GENERIC_BROWSER" ]; then
+    BROWSER="$XDG_GENERIC_BROWSER"
+fi
+
 case "$DE" in
     kde|gnome|xfce)
     dispatch_specific "$@"
