diff -Nur xdg-utils-1.0.2.orig/scripts/xdg-email xdg-utils-1.0.2/scripts/xdg-email
--- xdg-utils-1.0.2.orig/scripts/xdg-email	2007-06-25 06:58:04.000000000 +1100
+++ xdg-utils-1.0.2/scripts/xdg-email	2009-11-02 11:45:51.196846916 +1000
@@ -429,7 +429,7 @@
     fi
 }
 
-open_generic()
+open_generic_browser()
 {
     IFS=":"
     for browser in $BROWSER; do
@@ -449,6 +449,21 @@
     exit_failure_operation_impossible "no method available for opening '$1'"
 }
 
+open_generic()
+{
+    if [ ! -z "$XDG_GENERIC_EMAIL" ]; then
+        $XDG_GENERIC_EMAIL "$1"
+
+        if [ $? -eq 0 ]; then
+            exit_success
+        else
+            exit_failure_operation_failed
+        fi
+    else
+        open_generic_browser $@
+    fi
+}
+
 url_encode()
 {
 result=$(echo "$1" | $utf8 | awk '
@@ -617,6 +632,26 @@
     DE=generic
 fi
 
+# check system config
+if [ ! -z "$XDG_CONFIG_DIRS" ]; then
+    IFS=":"
+    for dir in $XDG_CONFIG_DIRS
+    do
+        cfg="$dir/xdg-utils/xdg-utils.rc"
+        if [ -e "$cfg" ]; then
+            source "$cfg"
+            break;
+        fi
+    done
+fi
+# check users config
+if [ ! -z "$XDG_CONFIG_HOME" ]; then
+    cfg="$XDG_CONFIG_HOME/xdg-utils/xdg-utils.rc"
+    if [ -e "$cfg" ]; then
+        source "$cfg"
+    fi
+fi
+
 case "$DE" in
     kde)
     open_kde "${mailto}"
diff -Nur xdg-utils-1.0.2.orig/scripts/xdg-open xdg-utils-1.0.2/scripts/xdg-open
--- xdg-utils-1.0.2.orig/scripts/xdg-open	2007-06-25 06:58:01.000000000 +1100
+++ xdg-utils-1.0.2/scripts/xdg-open	2009-11-02 11:43:42.040195917 +1000
@@ -31,6 +31,8 @@
 #
 #---------------------------------------------
 
+oldIFS=$IFS
+
 manualpage()
 {
 cat << _MANUALPAGE
@@ -358,7 +360,7 @@
     fi
 }
 
-open_generic()
+open_generic_browser()
 {
     IFS=":"
     for browser in $BROWSER; do
@@ -378,6 +380,21 @@
     exit_failure_operation_impossible "no method available for opening '$1'"
 }
 
+open_generic()
+{
+    if [ ! -z "$XDG_GENERIC_OPEN" ]; then
+        $XDG_GENERIC_OPEN "$1"
+
+        if [ $? -eq 0 ]; then
+            exit_success
+        else
+            exit_failure_operation_failed
+        fi
+    else
+        open_generic_browser $@
+    fi
+}
+
 [ x"$1" != x"" ] || exit_failure_syntax
 
 url=
@@ -403,6 +420,15 @@
     exit_failure_syntax "file or URL argument missing"
 fi
 
+if which xdg-open-hook.sh > /dev/null 2> /dev/null; then
+    xdg-email-hook.sh "${mailto}"
+    if [ $? -eq 0 ]; then
+        exit_success
+    else
+        exit_failure_operation_failed
+    fi
+fi
+
 detectDE
 
 if [ x"$DE" = x"" ]; then
@@ -413,6 +439,26 @@
     DE=generic
 fi
 
+# check system config
+if [ ! -z "$XDG_CONFIG_DIRS" ]; then
+    IFS=":"
+    for dir in $XDG_CONFIG_DIRS
+    do
+        cfg="$dir/xdg-utils/xdg-utils.rc"
+        if [ -e "$cfg" ]; then
+            source "$cfg"
+            break;
+        fi
+    done
+fi
+# check users config
+if [ ! -z "$XDG_CONFIG_HOME" ]; then
+    cfg="$XDG_CONFIG_HOME/xdg-utils/xdg-utils.rc"
+    if [ -e "$cfg" ]; then
+        source "$cfg"
+    fi
+fi
+
 case "$DE" in
     kde)
     open_kde "$url"
