diff -Naur ParaView-v4.1.0.orig/Utilities/VisItBridge/Library/module.cmake ParaView-v4.1.0/Utilities/VisItBridge/Library/module.cmake
--- ParaView-v4.1.0.orig/Utilities/VisItBridge/Library/module.cmake	2014-11-29 10:05:58.574771757 +0100
+++ ParaView-v4.1.0/Utilities/VisItBridge/Library/module.cmake	2014-11-29 10:24:30.057870396 +0100
@@ -3,14 +3,16 @@
     vtkCommonDataModel
     vtkCommonExecutionModel
     vtkCommonMisc
-    vtkFiltersExtraction
     vtkFiltersAMR
+    vtkFiltersExtraction
     vtkFiltersFlowPaths
     vtkImagingHybrid
     vtkIOGeometry
     vtkIOImage
+    vtkIOLegacy
     vtkParallelCore
+    vtkpng
     vtkRenderingVolumeOpenGL
-    vtkIOLegacy
+    vtkzlib
   EXCLUDE_FROM_WRAPPING
   )
diff -Naur ParaView-v4.1.0.orig/Utilities/VisItBridge/Library/VisItLib/CMake/VisItCommon.cmake ParaView-v4.1.0/Utilities/VisItBridge/Library/VisItLib/CMake/VisItCommon.cmake
--- ParaView-v4.1.0.orig/Utilities/VisItBridge/Library/VisItLib/CMake/VisItCommon.cmake	2014-11-29 10:05:58.611438225 +0100
+++ ParaView-v4.1.0/Utilities/VisItBridge/Library/VisItLib/CMake/VisItCommon.cmake	2014-11-29 10:19:43.812754453 +0100
@@ -86,19 +86,6 @@
     ENDIF(VISIT_PARALLEL_CXXFLAGS)
 ENDFUNCTION(ADD_PARALLEL_LIBRARY)
 
-MACRO(VISIT_VTK_THIRD_PARTY_INCLUDE upper lower)
-  if(VTK_USE_SYSTEM_${upper})
-    if(${upper}_INCLUDE_DIR)
-      include_directories(${${upper}_INCLUDE_DIR})
-    endif(${upper}_INCLUDE_DIR)
-  else(VTK_USE_SYSTEM_${upper})
-    include_directories(
-      ${VTK_BINARY_DIR}/Utilities/${lower}
-      ${VTK_SOURCE_DIR}/Utilities/${lower}
-    )
-  endif(VTK_USE_SYSTEM_${upper})
-ENDMACRO(VISIT_VTK_THIRD_PARTY_INCLUDE)
-
 #called from readers that are being built into paraview
 FUNCTION(ADD_VISIT_READER NAME VERSION)
   set(PLUGIN_NAME "vtk${NAME}")
@@ -612,19 +599,6 @@
   set(HAVE_LIBMILI ${MILI_FOUND})
 endif(MILI_FOUND)
 
-#setup zlib
-include_directories(${VTK_ZLIB_INCLUDE_DIRS})
-
-#setup netcdf
-visit_vtk_third_party_include(NETCDF vtknetcdf/include)
-
-#setup png
-visit_vtk_third_party_include(PNG vtkpng)
-
-#setup tiff
-visit_vtk_third_party_include(TIFF vtktiff)
-
-
 #-----------------------------------------------------------------------------
 # Detect packages here. We could probably write macros that we can include from
 # elsewhere for this.
diff -Naur ParaView-v4.1.0.orig/Utilities/VisItBridge/Library/VisItLib/visit_vtk/full/CMakeLists.txt ParaView-v4.1.0/Utilities/VisItBridge/Library/VisItLib/visit_vtk/full/CMakeLists.txt
--- ParaView-v4.1.0.orig/Utilities/VisItBridge/Library/VisItLib/visit_vtk/full/CMakeLists.txt	2014-11-29 10:05:58.604771595 +0100
+++ ParaView-v4.1.0/Utilities/VisItBridge/Library/VisItLib/visit_vtk/full/CMakeLists.txt	2014-11-29 10:22:52.341733104 +0100
@@ -88,7 +88,7 @@
 vtkVisItStreamLine.C
 vtkVisItStructuredGrid.C
 vtkVisItStructuredGridNormals.C
-vtkVisItTIFFWriter.C
+#vtkVisItTIFFWriter.C
 vtkVisItTensorGlyph.C
 vtkVolumeFromVolume.C
 )
@@ -130,7 +130,6 @@
     vtkInteractionStyle
     vtkIOLegacy
     vtkRenderingOpenGL
-    vtktiff
   )
 
 VISIT_INSTALL_TARGETS(visit_vtk)
