diff -Naur ParaView-v4.1.0.orig/ThirdParty/mpi4py/vtkmpi4py/CMakeLists.txt ParaView-v4.1.0/ThirdParty/mpi4py/vtkmpi4py/CMakeLists.txt
--- ParaView-v4.1.0.orig/ThirdParty/mpi4py/vtkmpi4py/CMakeLists.txt	2014-11-29 10:05:58.374772841 +0100
+++ ParaView-v4.1.0/ThirdParty/mpi4py/vtkmpi4py/CMakeLists.txt	2014-11-29 10:28:42.423169883 +0100
@@ -165,8 +165,9 @@
   RUNTIME_OUTPUT_DIRECTORY "${mpi4py_BINARY_DIR}"
   LINKER_LANGUAGE C
   )
-TARGET_LINK_LIBRARIES(mpi4py.MPI ${PYTHON_LIBRARY})
-TARGET_LINK_LIBRARIES(mpi4py.MPI ${MPI_C_LIBRARIES})
+TARGET_LINK_LIBRARIES(mpi4py.MPI ${PYTHON_LIBRARY}
+				 ${MPI_C_LIBRARIES}
+				 ${CMAKE_DL_LIBS})
 
 
 # --- mpi4py.MPE ---
