diff -Naur ParaView-v4.1.0.orig/VTK/Rendering/OpenGL/vtkXOpenGLRenderWindow.cxx ParaView-v4.1.0/VTK/Rendering/OpenGL/vtkXOpenGLRenderWindow.cxx
--- ParaView-v4.1.0.orig/VTK/Rendering/OpenGL/vtkXOpenGLRenderWindow.cxx	2014-11-29 10:05:58.168107294 +0100
+++ ParaView-v4.1.0/VTK/Rendering/OpenGL/vtkXOpenGLRenderWindow.cxx	2014-11-29 10:33:05.881742605 +0100
@@ -27,7 +27,7 @@
 
 // define GLX_GLXEXT_LEGACY to prevent glx.h to include glxext.h provided by
 // the system
-//#define GLX_GLXEXT_LEGACY
+#define GLX_GLXEXT_LEGACY
 #include "GL/glx.h"
 
 #include "vtkgl.h"
