diff -Naur orig/libao2/ao_arts.c patched/libao2/ao_arts.c
--- orig/libao2/ao_arts.c	2007-10-07 21:49:27.000000000 +0200
+++ patched/libao2/ao_arts.c	2008-01-24 10:11:28.000000000 +0100
@@ -46,7 +46,7 @@
 	int frag_spec;
 
 	if( (err=arts_init()) ) {
-		mp_msg(MSGT_AO, MSGL_ERR, MSGTR_AO_ARTS_CantInit, arts_error_text(err));
+		mp_msg(MSGT_AO, MSGL_INFO, MSGTR_AO_ARTS_CantInit, arts_error_text(err));
 		return 0;
 	}
 	mp_msg(MSGT_AO, MSGL_INFO, MSGTR_AO_ARTS_ServerConnect);
diff -Naur orig/libao2/ao_esd.c patched/libao2/ao_esd.c
--- orig/libao2/ao_esd.c	2007-10-07 21:49:27.000000000 +0200
+++ patched/libao2/ao_esd.c	2008-01-24 10:12:04.000000000 +0100
@@ -151,7 +151,7 @@
     if (esd_fd < 0) {
 	esd_fd = esd_open_sound(server);
 	if (esd_fd < 0) {
-	    mp_msg(MSGT_AO, MSGL_ERR, MSGTR_AO_ESD_CantOpenSound,
+	    mp_msg(MSGT_AO, MSGL_INFO, MSGTR_AO_ESD_CantOpenSound,
 		   strerror(errno));
 	    return 0;
 	}
diff -Naur orig/libvo/vosub_vidix.c patched/libvo/vosub_vidix.c
--- orig/libvo/vosub_vidix.c	2007-10-07 21:49:28.000000000 +0200
+++ patched/libvo/vosub_vidix.c	2008-01-24 10:13:08.000000000 +0100
@@ -723,7 +723,7 @@
               
 	if(vidix_handler == NULL)
 	{
-		mp_msg(MSGT_VO,MSGL_ERR, MSGTR_LIBVO_SUB_VIDIX_CouldntFindWorkingVidixDriver);
+		mp_msg(MSGT_VO,MSGL_INFO, MSGTR_LIBVO_SUB_VIDIX_CouldntFindWorkingVidixDriver);
 		return -1;
 	}
 	if((err=vdlGetCapability(vidix_handler,&vidix_cap)) != 0)
diff -Naur orig/mplayer.c patched/mplayer.c
--- orig/mplayer.c	2007-10-07 21:49:33.000000000 +0200
+++ patched/mplayer.c	2008-01-24 10:13:47.000000000 +0100
@@ -2146,7 +2146,6 @@
       gCpuCaps.has3DNow, gCpuCaps.has3DNowExt,
       gCpuCaps.hasSSE, gCpuCaps.hasSSE2);
 #ifdef RUNTIME_CPUDETECT
-  mp_msg(MSGT_CPLAYER,MSGL_INFO, MSGTR_CompiledWithRuntimeDetection);
 #else
   mp_msg(MSGT_CPLAYER,MSGL_INFO, MSGTR_CompiledWithCPUExtensions);
 #ifdef HAVE_MMX
