diff -Naurd MPlayer-1.0rc2.orig/etc/mplayer.desktop MPlayer-1.0rc2/etc/mplayer.desktop
--- MPlayer-1.0rc2.orig/etc/mplayer.desktop	2007-10-07 21:49:33.000000000 +0200
+++ MPlayer-1.0rc2/etc/mplayer.desktop	2008-01-24 14:12:57.000000000 +0100
@@ -7,6 +7,7 @@
 Comment[de]=Multimedia-Player
 Comment[es]=Reproductor multimedia
 Comment[fr]=Lecteur multimédia
+Comment[hu]=Multimédia lejátszó
 Comment[it]=Lettore multimediale
 Icon=mplayer.xpm
 TryExec=gmplayer
