diff -Naur original/data/Makefile.am modified/data/Makefile.am
--- original/data/Makefile.am	2014-08-12 14:00:21.000000000 +0100
+++ modified/data/Makefile.am	2014-08-26 17:24:05.113480986 +0100
@@ -32,7 +32,7 @@
 
 sessiondir = $(datadir)/gnome-session/sessions
 session_in_files = gnome-flashback.session.desktop.in gnome-flashback-compiz.session.desktop.in
-session_DATA = gnome-flashback.session
+session_DATA = $(session_in_files:.session.desktop.in=.session)
 
 xsessiondir = $(datadir)/xsessions
 xsession_in_in_files = gnome-fallback.desktop.in.in gnome-flashback-compiz.desktop.in.in
diff -Naur original/data/Makefile.in modified/data/Makefile.in
--- original/data/Makefile.in	2014-08-22 16:40:54.000000000 +0100
+++ modified/data/Makefile.in	2014-08-26 17:24:57.929894972 +0100
@@ -452,7 +452,7 @@
 gsettings_SCHEMAS = $(gsettings_schemas_in:.xml.in=.xml)
 sessiondir = $(datadir)/gnome-session/sessions
 session_in_files = gnome-flashback.session.desktop.in gnome-flashback-compiz.session.desktop.in
-session_DATA = gnome-flashback.session
+session_DATA = $(session_in_files:.session.desktop.in=.session)
 xsessiondir = $(datadir)/xsessions
 xsession_in_in_files = gnome-fallback.desktop.in.in gnome-flashback-compiz.desktop.in.in
 xsession_in_files = gnome-fallback.desktop.in gnome-flashback-compiz.desktop.in
