diff -Naur original/data/Makefile.am modified/data/Makefile.am
--- original/data/Makefile.am	2014-08-12 14:00:21.000000000 +0100
+++ modified/data/Makefile.am	2014-09-19 10:11:41.951536792 +0100
@@ -32,12 +32,12 @@
 
 sessiondir = $(datadir)/gnome-session/sessions
 session_in_files = gnome-flashback.session.desktop.in gnome-flashback-compiz.session.desktop.in
-session_DATA = gnome-flashback.session
+session_DATA =
 
 xsessiondir = $(datadir)/xsessions
 xsession_in_in_files = gnome-fallback.desktop.in.in gnome-flashback-compiz.desktop.in.in
 xsession_in_files = gnome-fallback.desktop.in gnome-flashback-compiz.desktop.in
-xsession_DATA = $(xsession_in_files:.desktop.in=.desktop)
+xsession_DATA =
 
 %.session: %.session.desktop.in
 	$(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; LC_ALL=C $(INTLTOOL_MERGE) -d -u -c $(top_builddir)/po/.intltool-merge-cache $(top_srcdir)/po $< $@
diff -Naur original/data/Makefile.in modified/data/Makefile.in
--- original/data/Makefile.in	2014-08-22 16:40:54.000000000 +0100
+++ modified/data/Makefile.in	2014-09-19 10:12:03.271416736 +0100
@@ -452,11 +452,11 @@
 gsettings_SCHEMAS = $(gsettings_schemas_in:.xml.in=.xml)
 sessiondir = $(datadir)/gnome-session/sessions
 session_in_files = gnome-flashback.session.desktop.in gnome-flashback-compiz.session.desktop.in
-session_DATA = gnome-flashback.session
+session_DATA =
 xsessiondir = $(datadir)/xsessions
 xsession_in_in_files = gnome-fallback.desktop.in.in gnome-flashback-compiz.desktop.in.in
 xsession_in_files = gnome-fallback.desktop.in gnome-flashback-compiz.desktop.in
-xsession_DATA = $(xsession_in_files:.desktop.in=.desktop)
+xsession_DATA =
 default_layoutdir = $(datadir)/gnome-panel
 default_layout_DATA = panel-default-layout.layout
 EXTRA_DIST = \
