Description: fix moving applets in panel
Origin: upstream, https://git.gnome.org/browse/gnome-panel/commit/?id=d7d0efa55a26cc
Last-Update: 2014-02-24

--- a/gnome-panel/applet.c
+++ b/gnome-panel/applet.c
@@ -557,7 +557,7 @@
 {
 	GtkAllocation   allocation;
 	GtkRequisition  requisition;
-	GdkModifierType modifier_mask;
+	GdkDevice      *device;
 	GdkScreen      *screen;
 	GtkWidget      *parent;
 	int             menu_x = 0;
@@ -574,7 +574,8 @@
 	gtk_widget_get_preferred_size (GTK_WIDGET (menu), &requisition, NULL);
 
 	gdk_window_get_origin (gtk_widget_get_window (applet), &menu_x, &menu_y);
-	gdk_window_get_device_position(gtk_widget_get_window (applet), gtk_get_current_event_device (), &pointer_x, &pointer_y, &modifier_mask);
+	device = gdk_device_manager_get_client_pointer (gdk_display_get_device_manager (gtk_widget_get_display (applet)));
+	gdk_window_get_device_position(gtk_widget_get_window (applet), device, &pointer_x, &pointer_y, NULL);
 
 	gtk_widget_get_allocation (applet, &allocation);
 
--- a/gnome-panel/panel-toplevel.c
+++ b/gnome-panel/panel-toplevel.c
@@ -925,13 +925,14 @@
 {
 	GdkScreen 	*screen;
 	GdkWindow 	*root_window;
-	GdkModifierType	 modifier_mask;
+	GdkDevice 	*device;
 	int        	 new_x, new_y;
 
 	screen = gtk_window_get_screen (GTK_WINDOW (toplevel));
 	root_window = gdk_screen_get_root_window (screen);
 
-	gdk_window_get_device_position(root_window, gtk_get_current_event_device (), &new_x, &new_y, &modifier_mask);
+	device = gdk_device_manager_get_client_pointer(gdk_display_get_device_manager(gtk_widget_get_display(GTK_WIDGET(toplevel))));
+	gdk_window_get_device_position(root_window, device, &new_x, &new_y, NULL);
 
 	switch (keyval) {
 	case GDK_KEY_Up:
--- a/gnome-panel/panel-widget.c
+++ b/gnome-panel/panel-widget.c
@@ -1509,7 +1509,7 @@
 {
 	GtkWidget       *widget;
 	GtkAllocation   allocation;
-	GdkModifierType modifier_mask;
+	GdkDevice       *device;
 	int             x,y;
 	int             w,h;
 
@@ -1522,7 +1522,8 @@
 	   !gtk_widget_get_visible(widget))
 		return FALSE;
 
-	gdk_window_get_device_position(gtk_widget_get_window (widget), gtk_get_current_event_device (), &x, &y, &modifier_mask);
+	device = gdk_device_manager_get_client_pointer (gdk_display_get_device_manager (gtk_widget_get_display (widget)));
+	gdk_window_get_device_position(gtk_widget_get_window (widget), device, &x, &y, NULL);
 
 	gtk_widget_get_allocation (widget, &allocation);
 	w = allocation.width;
@@ -1895,11 +1896,12 @@
 {
 	int             x, y;
 	gboolean        rtl;
-	GdkModifierType modifier_mask;
+	GdkDevice      *device;
 
 	g_return_val_if_fail (PANEL_IS_WIDGET (panel), -1);
 
-	gdk_window_get_device_position(gtk_widget_get_window (GTK_WIDGET (panel)), gtk_get_current_event_device (), &x, &y, &modifier_mask);
+	device = gdk_device_manager_get_client_pointer (gdk_display_get_device_manager (gtk_widget_get_display (GTK_WIDGET (panel))));
+	gdk_window_get_device_position(gtk_widget_get_window (GTK_WIDGET (panel)), device, &x, &y, NULL);
 	rtl = gtk_widget_get_direction (GTK_WIDGET (panel)) == GTK_TEXT_DIR_RTL;
 	
 	if (panel->orient == GTK_ORIENTATION_HORIZONTAL)
@@ -2074,13 +2076,14 @@
 	if(panel->currently_dragged_applet && repeat_if_outside) {
 		GtkWidget       *widget;
 		GtkAllocation   allocation;
-		GdkModifierType modifier_mask;
+		GdkDevice       *device;
 		int             x,y;
 		int             w,h;
 
 		widget = panel->currently_dragged_applet->applet;
 
-		gdk_window_get_device_position(gtk_widget_get_window (widget), gtk_get_current_event_device (), &x, &y, &modifier_mask);
+		device = gdk_device_manager_get_client_pointer (gdk_display_get_device_manager (gtk_widget_get_display (widget)));
+		gdk_window_get_device_position(gtk_widget_get_window (widget), device, &x, &y, NULL);
 
 		gtk_widget_get_allocation (widget, &allocation);
 		w = allocation.width;
--- a/libpanel-applet/panel-applet.c
+++ b/libpanel-applet/panel-applet.c
@@ -1130,7 +1130,7 @@
 	PanelApplet    *applet;
 	GtkAllocation   allocation;
 	GtkRequisition  requisition;
-	GdkModifierType modifier_mask;
+	GdkDevice      *device;
 	GdkScreen      *screen;
 	int             menu_x = 0;
 	int             menu_y = 0;
@@ -1148,7 +1148,8 @@
 	gtk_widget_get_preferred_size (GTK_WIDGET (menu), &requisition, NULL);
 	gdk_window_get_origin (gtk_widget_get_window (widget),
 			       &menu_x, &menu_y);
-	gdk_window_get_device_position(gtk_widget_get_window (widget), gtk_get_current_event_device (), &pointer_x, &pointer_y, &modifier_mask);
+	device = gdk_device_manager_get_client_pointer (gdk_display_get_device_manager (gtk_widget_get_display (widget)));
+	gdk_window_get_device_position(gtk_widget_get_window (widget), device, &pointer_x, &pointer_y, NULL);
 
 	gtk_widget_get_allocation (widget, &allocation);
 
