--- include/common.h	2003-02-22 16:26:04.000000000 -0500
+++ ../../src.new/htun-0.9.6/include/common.h	2011-03-20 00:50:46.210001372 -0400
@@ -30,6 +30,7 @@
 #include "server.h"
 #include "iprange.h"
 
+#define PATH_MAX (FILENAME_MAX) 
 #define HTUN_MAXPACKET 65536
 #define HTUN_DEFAULT_CFGFILE "/etc/htund.conf"
 #define HTUN_MAXCLIENTS 10
--- src/gram.y	2003-01-23 20:20:50.000000000 -0500
+++ ../../src.new/htun-0.9.6/src/gram.y	2011-03-20 01:13:46.976668039 -0400
@@ -329,7 +329,7 @@
     die_error(yylineno, s1, s2);
 }
 
-static void set_ip(struct in_addr *a, char *dst, char *src)
+void set_ip(struct in_addr *a, char *dst, char *src)
 {
     struct hostent host, *result = &host;
     char buf[4096];
@@ -343,7 +343,7 @@
     strncpy(dst, inet_ntoa(*a), 15);
 }
 
-static int get_answer(char *name, char *true_val, char *false_val)
+int get_answer(char *name, char *true_val, char *false_val)
 {
     if(strstr(name, true_val) != NULL)
         return 1;
@@ -353,7 +353,7 @@
         yy_error("invalid answer", "answer must be 'yes' or 'no'" );
 }
 
-static inline set_base64_user_pass( void ) 
+inline set_base64_user_pass( void ) 
 {
     char user_pass[90];
 
