--- tomoe-0.6.0.orig//module/dict/Makefile.am	2007-06-18 03:35:40.000000000 +0200
+++ tomoe-0.6.0//module/dict/Makefile.am	2010-08-30 09:53:11.760000963 +0200
@@ -34,7 +34,7 @@ INCLUDES =				\
 
 LIBADD = $(TOMOE_LIBS)			\
 	$(top_builddir)/lib/libtomoe.la
-LDFLAGS =							\
+AM_LDFLAGS =							\
   -rpath $(dict_moduledir) -avoid-version -module		\
   -export-dynamic $(no_undefined) $(LIBTOOL_EXPORT_OPTIONS)
 
--- tomoe-0.6.0.orig//module/recognizer/Makefile.am	2007-04-10 09:24:20.000000000 +0200
+++ tomoe-0.6.0//module/recognizer/Makefile.am	2010-08-30 09:52:52.690001016 +0200
@@ -23,7 +23,7 @@ CLEANFILES = *~ *.bak
 AM_CPPFLAGS =
 INCLUDES = $(TOMOE_CFLAGS)
 
-LDFLAGS =							\
+AM_LDFLAGS =							\
   -rpath $(recognizer_moduledir) -avoid-version -module		\
   -export-dynamic $(no_undefined) $(LIBTOOL_EXPORT_OPTIONS)
 
