--- ./lib/tomoe-dict-ptr-array.c.bz502662	2007-06-21 11:48:57.000000000 +1000
+++ ./lib/tomoe-dict-ptr-array.c	2009-06-02 13:23:40.000000000 +1000
@@ -50,7 +50,7 @@ struct _TomoeDictPtrArrayPrivate
 
 static TomoeDictClass *parent_class;
 
-G_DEFINE_ABSTRACT_TYPE (TomoeDictPtrArray, _tomoe_dict_ptr_array, TOMOE_TYPE_DICT)
+G_DEFINE_ABSTRACT_TYPE (TomoeDictPtrArray, tomoe_dict_ptr_array, TOMOE_TYPE_DICT)
 
 static void         dispose                    (GObject       *object);
 static void         set_property               (GObject       *object,
@@ -75,7 +75,7 @@ static gboolean     is_editable         
 static gchar       *get_available_private_utf8 (TomoeDict     *dict);
 
 static void
-_tomoe_dict_ptr_array_class_init (TomoeDictPtrArrayClass *klass)
+tomoe_dict_ptr_array_class_init (TomoeDictPtrArrayClass *klass)
 {
     GObjectClass *gobject_class;
     TomoeDictClass *dict_class;
@@ -123,7 +123,7 @@ _tomoe_dict_ptr_array_class_init (TomoeD
 }
 
 static void
-_tomoe_dict_ptr_array_init (TomoeDictPtrArray *dict)
+tomoe_dict_ptr_array_init (TomoeDictPtrArray *dict)
 {
     TomoeDictPtrArrayPrivate *priv = TOMOE_DICT_PTR_ARRAY_GET_PRIVATE (dict);
     priv->chars    = g_ptr_array_new();
@@ -196,7 +196,7 @@ char_compare_func (gconstpointer a, gcon
 }
 
 void
-_tomoe_dict_ptr_array_sort (TomoeDictPtrArray *dict)
+tomoe_dict_ptr_array_sort (TomoeDictPtrArray *dict)
 {
     TomoeDictPtrArrayPrivate *priv;
 
@@ -219,7 +219,7 @@ register_char (TomoeDict *dict, TomoeCha
 
     unregister_char (dict, tomoe_char_get_utf8 (chr));
     g_ptr_array_add (priv->chars, g_object_ref (G_OBJECT (chr)));
-    _tomoe_dict_ptr_array_sort (TOMOE_DICT_PTR_ARRAY (dict));
+    tomoe_dict_ptr_array_sort (TOMOE_DICT_PTR_ARRAY (dict));
 
     priv->modified = TRUE;
 
@@ -523,7 +523,7 @@ get_available_private_utf8 (TomoeDict *d
 }
 
 GPtrArray *
-_tomoe_dict_ptr_array_get_array (TomoeDictPtrArray *dict)
+tomoe_dict_ptr_array_get_array (TomoeDictPtrArray *dict)
 {
     g_return_val_if_fail (TOMOE_IS_DICT_PTR_ARRAY (dict), NULL);
 
--- ./lib/tomoe-dict-ptr-array.h.bz502662	2007-06-21 11:48:57.000000000 +1000
+++ ./lib/tomoe-dict-ptr-array.h	2009-06-02 13:27:12.000000000 +1000
@@ -29,7 +29,7 @@ G_BEGIN_DECLS
 
 #include "tomoe-dict.h"
 
-#define TOMOE_TYPE_DICT_PTR_ARRAY            (_tomoe_dict_ptr_array_get_type ())
+#define TOMOE_TYPE_DICT_PTR_ARRAY            (tomoe_dict_ptr_array_get_type ())
 #define TOMOE_DICT_PTR_ARRAY(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), TOMOE_TYPE_DICT_PTR_ARRAY, TomoeDictPtrArray))
 #define TOMOE_DICT_PTR_ARRAY_CLASS(klass)    (G_TYPE_CHECK_CLASS_CAST ((klass), TOMOE_TYPE_DICT_PTR_ARRAY, TomoeDictPtrArrayClass))
 #define TOMOE_IS_DICT_PTR_ARRAY(obj)         (G_TYPE_CHECK_INSTANCE_TYPE ((obj), TOMOE_TYPE_DICT_PTR_ARRAY))
@@ -49,9 +49,11 @@ struct _TomoeDictPtrArrayClass
     TomoeDictClass parent_class;
 };
 
-GType     _tomoe_dict_ptr_array_get_type        (void) G_GNUC_CONST;
-void      _tomoe_dict_ptr_array_sort            (TomoeDictPtrArray *dict);
-GPtrArray *_tomoe_dict_ptr_array_get_array      (TomoeDictPtrArray *dict);
+GType     tomoe_dict_ptr_array_get_type        (void) G_GNUC_CONST;
+void      tomoe_dict_ptr_array_sort            (TomoeDictPtrArray *dict);
+#define  _tomoe_dict_ptr_array_sort tomoe_dict_ptr_array_sort
+GPtrArray *tomoe_dict_ptr_array_get_array      (TomoeDictPtrArray *dict);
+#define _tomoe_dict_ptr_array_get_array tomoe_dict_ptr_array_get_array
 
 G_END_DECLS
 
--- ./lib/tomoe-xml-parser.c.bz502662	2007-04-10 17:24:21.000000000 +1000
+++ ./lib/tomoe-xml-parser.c	2009-06-02 13:23:40.000000000 +1000
@@ -428,7 +428,7 @@ init_parse_data (ParseData *data, TomoeX
 }
 
 gboolean
-_tomoe_xml_parser_parse_dictionary_file (const gchar *filename,
+tomoe_xml_parser_parse_dictionary_file (const gchar *filename,
                                          TomoeXMLParsedData *result)
 {
     GMarkupParseContext *context;
@@ -469,7 +469,7 @@ _tomoe_xml_parser_parse_dictionary_file 
 }
 
 TomoeChar *
-_tomoe_xml_parser_parse_char_data (const gchar *xml, gssize len)
+tomoe_xml_parser_parse_char_data (const gchar *xml, gssize len)
 {
     GMarkupParseContext *context;
     TomoeXMLParsedData result;
--- ./lib/tomoe-xml-parser.h.bz502662	2007-04-10 17:24:21.000000000 +1000
+++ ./lib/tomoe-xml-parser.h	2009-06-02 13:23:40.000000000 +1000
@@ -37,11 +37,12 @@ struct _TomoeXMLParsedData
     GPtrArray *chars;
 };
 
-gboolean   _tomoe_xml_parser_parse_dictionary_file (const gchar *filename,
+gboolean   tomoe_xml_parser_parse_dictionary_file (const gchar *filename,
                                                     TomoeXMLParsedData *result);
-TomoeChar *_tomoe_xml_parser_parse_char_data       (const gchar *xml,
+#define   _tomoe_xml_parser_parse_dictionary_file tomoe_xml_parser_parse_dictionary_file
+TomoeChar *tomoe_xml_parser_parse_char_data       (const gchar *xml,
                                                     gssize      len);
-
+#define   _tomoe_xml_parser_parse_char_data tomoe_xml_parser_parse_char_data
 
 G_END_DECLS
 
