diff -rc foxtrotgps-1.0.1_original/src/friends.c foxtrotgps-1.0.1/src/friends.c
*** foxtrotgps-1.0.1_original/src/friends.c	2011-07-22 13:49:08.065173977 +0200
--- foxtrotgps-1.0.1/src/friends.c	2011-07-22 13:49:32.885557770 +0200
***************
*** 17,23 ****
  #include <gtk/gtk.h>
  
  #include <curl/curl.h>
! #include <curl/types.h>
  #include <curl/easy.h>
  
  
--- 17,23 ----
  #include <gtk/gtk.h>
  
  #include <curl/curl.h>
! //#include <curl/types.h>
  #include <curl/easy.h>
  
  
diff -rc foxtrotgps-1.0.1_original/src/tile_management.c foxtrotgps-1.0.1/src/tile_management.c
*** foxtrotgps-1.0.1_original/src/tile_management.c	2011-07-22 13:49:08.068507375 +0200
--- foxtrotgps-1.0.1/src/tile_management.c	2011-07-22 13:49:47.485783617 +0200
***************
*** 5,11 ****
  #include <glib/gprintf.h>
  #include <string.h>
  #include <curl/curl.h>
! #include <curl/types.h> 
  #include <curl/easy.h> 
  #include <math.h>
  #include "globals.h"
--- 5,11 ----
  #include <glib/gprintf.h>
  #include <string.h>
  #include <curl/curl.h>
! //#include <curl/types.h> 
  #include <curl/easy.h> 
  #include <math.h>
  #include "globals.h"
diff -rc foxtrotgps-1.0.1_original/src/util.h foxtrotgps-1.0.1/src/util.h
*** foxtrotgps-1.0.1_original/src/util.h	2011-07-22 13:49:08.068507375 +0200
--- foxtrotgps-1.0.1/src/util.h	2011-07-22 13:50:01.726003809 +0200
***************
*** 2,8 ****
  #include <gtk/gtk.h>
  
  #include <curl/curl.h>
! #include <curl/types.h>
  #include <curl/easy.h>
  int
  sql_execute(char *db_name, char *sql, int (*cb_func)(void*,int,char**,char**));
--- 2,8 ----
  #include <gtk/gtk.h>
  
  #include <curl/curl.h>
! //#include <curl/types.h>
  #include <curl/easy.h>
  int
  sql_execute(char *db_name, char *sql, int (*cb_func)(void*,int,char**,char**));