--- nettop.c~	2001-11-09 04:50:03.000000000 +0100
+++ nettop.c	2008-05-11 01:52:40.000000000 +0200
@@ -216,22 +216,26 @@
 		}
 	}
 
-	ether.count = ether.size = (int) ether.l = (int) ether.r = 0;
+	ether.count = ether.size = 0;
+	ether.l = ether.r = NULL;
 	ether.type = -1;
 	for (i = 0; i < HISTORY_SIZE; i++)
 		ether.size_h[i] = 0;
 
-	ip.count = ip.size = (int) ip.l = (int) ip.r = 0;
+	ip.count = ip.size = 0;
+	ip.l = ip.r = NULL;
 	ip.type = -1;
 	for (i = 0; i < HISTORY_SIZE; i++)
 		ip.size_h[i] = 0;
 
-	tcp.count = tcp.size = (int) tcp.l = (int) tcp.r = 0;
+	tcp.count = tcp.size = 0;
+	tcp.l = tcp.r = NULL;
 	tcp.type = -1;
 	for (i = 0; i < HISTORY_SIZE; i++)
 		tcp.size_h[i] = 0;
 
-	udp.count = udp.size = (int) udp.l = (int) udp.r = 0;
+	udp.count = udp.size = 0;
+	udp.l = udp.r = NULL;
 	udp.type = -1;
 	for (i = 0; i < HISTORY_SIZE; i++)
 		udp.size_h[i] = 0;
@@ -361,6 +365,7 @@
 				}
 				break;
 			default:
+		break;
 			}
 		}
 	}
