--- doc/doc.pri	2009-08-30 07:57:42.577787183 +0200
+++ doc/doc.pri	2009-08-30 07:58:41.695854702 +0200
@@ -16,8 +16,8 @@
 }
 
 unix {
-    QDOC = SRCDIR=$$PWD OUTDIR=$$OUT_PWD/doc/html $$QDOC_BIN 
-    HELPGENERATOR = $$(QTDIR)/bin/qhelpgenerator
+    QDOC = SRCDIR=$$PWD OUTDIR=$$OUT_PWD/doc/html /usr/bin/qdoc3
+    HELPGENERATOR = /usr/bin/qhelpgenerator
 } else {
     QDOC = set SRCDIR=$$PWD&& set OUTDIR=$$OUT_PWD/doc/html&& $$QDOC_BIN 
     HELPGENERATOR = $$(QTDIR)\bin\qhelpgenerator.exe
