--- src/Pkg/Package.hs.old	2012-11-27 20:43:18.481535534 -0800
+++ src/Pkg/Package.hs	2012-11-27 20:45:00.938159875 -0800
@@ -4,6 +4,7 @@
 import System.Directory
 import System.Exit
 import System.IO
+import System.Environment (lookupEnv)
 
 import Util.System
 
@@ -106,7 +107,8 @@
 
 installIBC :: String -> Name -> IO ()
 installIBC p m = do let f = toIBCFile m
-                    d <- getDataDir
+                    target <- lookupEnv "TARGET"
+                    d <- maybe getDataDir return target
                     let destdir = d ++ "/" ++ p ++ "/" ++ getDest m
                     putStrLn $ "Installing " ++ f ++ " to " ++ destdir
                     system $ "mkdir -p " ++ destdir 
