--- irserver-vanilla/lowlevel.c      2014-07-25 19:50:25.471623406 +0200
+++ irserver-patched/lowlevel.c  2014-07-25 20:05:44.771600822 +0200
@@ -3430,10 +3430,17 @@
 #endif
 #ifdef LINUX
 		//LINUX Autofind USB devices
-		if ((!strncmp (st + q,"usb",3) || !strncmp (st + q,"USB",3)) && !autoflag) {
+                if( !strncmp( st, "/dev/irtrans", 12 ) && !autoflag ) {
+                  autoflag = 1;
+                  res = 0;
+                  get_detail_deviceinfo ("",st,IF_USB);
+                }
+		else if ((!strncmp (st + q,"usb",3) || !strncmp (st + q,"USB",3)) && !autoflag) {
 			autoflag = 1;
 			res = 0;
 			for (i=0;i < 16;i++) {
+                           sprintf (dst,"/dev/irtrans%d",i);
+                           if (get_detail_deviceinfo ("",dst,IF_USB)) {
 				sprintf (dst,"/dev/ttyUSB%d",i);
 				if (get_detail_deviceinfo ("",dst,IF_USB)) {
 					sprintf (dst,"/dev/usb/ttyUSB%d",i);
@@ -3445,6 +3452,7 @@
 						}
 					}
 				}
+			   }
 			}
 		}
 		else if ((!strncmp (st + q,"/dev/usb/tty",12) || !strncmp (st + q,"/dev/ttyUSB",11) || !strncmp (st + q,"/dev/usb/tts",12) ||
