From 4e84c272439c5335d78492c2a753436067d77c9b Mon Sep 17 00:00:00 2001
From: TonySeek <tonyseek@gmail.com>
Date: Sun, 3 Feb 2013 21:32:40 +0800
Subject: [PATCH] Add "create_pthread" to libs searching list.

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index cd0d235..a5ce162 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,6 +9,7 @@ AC_CONFIG_HEADERS([config.h])
 AC_SEARCH_LIBS(socket, socket)
 AC_SEARCH_LIBS(gethostbyname, nsl)
 AC_SEARCH_LIBS(mallinfo, malloc)
+AC_SEARCH_LIBS(pthread_create, pthread)
 
 AC_CHECK_LIB([rt], [clock_gettime])
 AC_CHECK_FUNC(daemon,AC_DEFINE([HAVE_DAEMON],,[Define this if you have daemon()]),[AC_LIBOBJ(daemon)])
-- 
1.8.1.2

