Index: src/lib/azy_net.c
===================================================================
--- src/lib/azy_net.c
+++ src/lib/azy_net.c
@@ -521,8 +521,8 @@
 
    net->http.content_length = length;
    snprintf(buf, sizeof(buf), "%i", length);
-   azy_net_header_set(net, "content-length", NULL);
-   azy_net_header_set(net, "content-length", buf);
+   azy_net_header_set(net, "Content-Length", NULL);
+   azy_net_header_set(net, "Content-Length", buf);
 }
 
 /**
Index: src/lib/azy_server_events.c
===================================================================
--- src/lib/azy_server_events.c
+++ src/lib/azy_server_events.c
@@ -861,7 +861,7 @@
 
       case AZY_NET_TYPE_POST:
         if (!client->current->transport)
-          client->current->transport = azy_transport_get(azy_net_header_get(client->current, "content-type"));
+          client->current->transport = azy_transport_get(azy_net_header_get(client->current, "Content-Type"));
         switch (client->current->transport)
           {
            case AZY_NET_TRANSPORT_JSON:
