diff --git a/kdevmercurial.desktop b/kdevmercurial.desktop
index b49e1f5..7698581 100644
--- a/kdevmercurial.desktop
+++ b/kdevmercurial.desktop
@@ -46,7 +46,7 @@ X-KDE-PluginInfo-Email=batyiev@gmail.com
 X-KDE-PluginInfo-Version=0.9
 X-KDE-PluginInfo-License=GPL
 X-KDE-PluginInfo-Category=Version Control
-X-KDevelop-Version=11
+X-KDevelop-Version=12
 X-KDevelop-Category=Global
 X-KDevelop-Interfaces=org.kdevelop.IBasicVersionControl, org.kdevelop.IDistributedVersionControl
 X-KDevelop-Mode=GUI
