--- SConstruct.old	2011-05-09 02:49:06.986670742 +0200
+++ SConstruct	2011-05-09 02:50:01.947173125 +0200
@@ -32,7 +32,7 @@
 # Documentation
 #  This only needs to be run during the release/packaging process
 man_fixup = "|sed 's/.SH DESCRIPTION/\\n.SH DESCRIPTION/'"
-man_builder = Builder(action='docbook2x-man --to-stdout $SOURCE'
+man_builder = Builder(action='docbook2man --to-stdout $SOURCE'
                       + man_fixup + '| gzip -9 > $TARGET',
                       suffix = '.gz',
                       src_suffix = '.docbook')
@@ -43,7 +43,10 @@
 man_reglookup_timeline = env.ManPage('doc/reglookup-timeline.1.docbook')
 
 # Installation
-prefix='/usr/local/'
+if 'PREFIX' in os.environ:
+	prefix = os.environ['PREFIX']
+else:
+	prefix = '/usr/local/'
 install_items = [prefix+'bin',
                  prefix+'lib', 
                  prefix+'include/regfi',
@@ -58,13 +61,13 @@
 if sys.version_info[0] == 2:
    install_items.append('pyregfi2-install.log')
    env.Command('pyregfi2-install.log', ['python/pyregfi/__init__.py', 'python/pyregfi/structures.py'], 
-               "python pyregfi-distutils.py install | tee pyregfi2-install.log")
+               "PREFIX='" + prefix + "' python pyregfi-distutils.py install | tee pyregfi2-install.log")
 
 python_path = os.popen('which python3').read()
 if python_path != '':
    install_items.append('pyregfi3-install.log')
    env.Command('pyregfi3-install.log', ['python/pyregfi/__init__.py', 'python/pyregfi/structures.py'], 
-               "python3 pyregfi-distutils.py install | tee pyregfi3-install.log")
+               "PREFIX='" + prefix + "' python3 pyregfi-distutils.py install | tee pyregfi3-install.log")
 
 # API documentation
 regfi_doc = env.Command('doc/devel/regfi/index.html', 
