--- a/usr/lib/dkms/dkms_autoinstaller	2011-07-21 23:06:55.000000000 +0000
+++ b/usr/lib/dkms/dkms_autoinstaller	2011-08-02 04:45:06.000000000 +0000
@@ -27,6 +27,13 @@
     log_end_msg() { if [ "$1" = "0" ]; then echo " Done. "; else echo " Failed. "; fi }
     alias log_action_begin_msg=log_daemon_msg
     alias log_action_end_msg=log_end_msg
+elif [ -f /etc/rc.d/functions ]; then
+    . /etc/rc.conf
+    . /etc/rc.d/functions
+    alias log_daemon_msg=stat_busy
+    log_end_msg() { if [ "$1" = "0" ]; then stat_done; else stat_fail; fi }
+    alias log_action_begin_msg=printhl
+    alias log_action_end_msg=true
 fi
 
 if [ -n "$2" ]; then
