--- ole.c	2014-07-27 05:23:26.291469781 -0400
+++ ole.new.c	2014-07-27 05:23:37.651608715 -0400
@@ -565,9 +565,8 @@
 \------------------------------------------------------------------*/
 int OLE_get_header( struct OLE_object *ole )
 {
-	int result = 0;
 	ole->header.sector_size = OLE_HEADER_BLOCK_SIZE;
-	result = OLE_get_block( ole, -1, ole->header_block );
+	OLE_get_block( ole, -1, ole->header_block );
 
 	if (OLE_is_file_OLE( ole ) == 0)
 	{
