diff -Naur bad/dated_main.h all_cc_progs/dated_main.h
--- bad/dated_main.h	2012-04-01 22:11:01.000000000 +0000
+++ all_cc_progs/dated_main.h	2012-12-06 23:46:54.817291386 +0000
@@ -22,7 +22,7 @@
     return *this;
 };
 
-template<class T> dated<T>::~dated();
+// template<class T> dated<T>::~dated();
 
 template<class T> bool dated<T>::empty() const { return (dates_.size()<1); };  
 
diff -Naur bad/makefile all_cc_progs/makefile
--- bad/makefile	2007-04-09 09:35:21.000000000 +0000
+++ all_cc_progs/makefile	2012-12-06 23:46:41.073956163 +0000
@@ -28,9 +28,9 @@
              security_price_history_search.o security_price_history_sizes.o security_price_history_utils.o
 
 # stock price history
-HXX_FILES += stock_price_history.h
-CXX_FILES += stock_price_history.cc stock_price_history_adjustments.cc stock_price_history_dividends.cc
-LIBOBJS   += stock_price_history.o stock_price_history_adjustments.o stock_price_history_dividends.o
+#HXX_FILES += stock_price_history.h
+#CXX_FILES += stock_price_history.cc stock_price_history_adjustments.cc stock_price_history_dividends.cc
+#LIBOBJS   += stock_price_history.o stock_price_history_adjustments.o stock_price_history_dividends.o
 
 LIB = libempirdata.a 
 $(LIB): $(LIBOBJS)
@@ -52,4 +52,4 @@
 	unzip -u -o $(ZIPFILE)
 install-h:
 install-lib:
-install-www:
\ No newline at end of file
+install-www:
