--- pymongo-2.3/doc/conf.py	2012-07-07 03:00:11.000000000 +0800
+++ pymongo-2.3-new/doc/conf.py	2012-10-16 23:40:27.558775336 +0800
@@ -81,6 +81,7 @@
 # The theme to use for HTML and HTML Help pages.  Major themes that come with
 # Sphinx are currently 'default' and 'sphinxdoc'.
 html_theme = 'default'
+html_theme_options = {'collapsiblesidebar': True}
 
 # The name for this set of Sphinx documents.  If None, it defaults to
 # "<project> v<release> documentation".
