--- a/src/gq-tab-search.c	2011-08-05 17:58:52.308905062 +0200
+++ b/src/gq-tab-search.c	2011-08-05 18:01:15.455360637 +0200
@@ -2307,6 +2307,8 @@ static void gq_tab_search_init(GqTabSear
     /* results paned */
 #warning "FIXME: save the paned position"
     paned = gtk_hpaned_new();
+/* set pane position to something useful... */
+gtk_paned_set_position(paned, 400);
     gtk_widget_show(paned);
     gtk_box_pack_start(GTK_BOX(searchmode_vbox), paned, TRUE, TRUE, 0);
 
@@ -2316,7 +2320,7 @@ static void gq_tab_search_init(GqTabSear
      * display to be set */
     P(self)->results_display = gq_input_form_new();
     gtk_widget_show(P(self)->results_display);
-    gtk_paned_pack2(GTK_PANED(paned), P(self)->results_display, FALSE, FALSE);
+    gtk_paned_pack2(GTK_PANED(paned), P(self)->results_display, TRUE, FALSE);
     gq_input_form_set_editable(GQ_INPUT_FORM(P(self)->results_display), TRUE);
 
     /* results treeview */
