From 2de8322082d82acadd15bc9b24f6812b5a875df7 Mon Sep 17 00:00:00 2001
From: Lucas De Marchi <lucas.demarchi@profusion.mobi>
Date: Thu, 18 Nov 2010 23:21:57 -0200
Subject: [PATCH 2/6] mandir is at /usr/share

---
 MCONFIG |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/MCONFIG b/MCONFIG
index c76b591..72944e9 100644
--- a/MCONFIG
+++ b/MCONFIG
@@ -23,7 +23,7 @@ SBINDIR  = /sbin
 LIBDIR   = /usr/lib
 DATADIR  = /usr/share
 AUXDIR   = $(DATADIR)/syslinux
-MANDIR	 = /usr/man
+MANDIR	 = /usr/share/man
 INCDIR   = /usr/include
 TFTPBOOT = /tftpboot
 COM32DIR = $(AUXDIR)/com32
-- 
1.7.3.2

