From abefafdd33eec443d74119b72b2d1e15a1e19856 Mon Sep 17 00:00:00 2001
From: Lucas De Marchi <lucas.demarchi@profusion.mobi>
Date: Thu, 18 Nov 2010 23:22:32 -0200
Subject: [PATCH 3/6] Do not build dos and windows executables

---
 Makefile |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index da90cdc..841bd90 100644
--- a/Makefile
+++ b/Makefile
@@ -43,9 +43,7 @@ BOBJECTS = $(BTARGET) \
 	mbr/mbr_c.bin mbr/altmbr_c.bin mbr/gptmbr_c.bin \
 	mbr/mbr_f.bin mbr/altmbr_f.bin mbr/gptmbr_f.bin \
 	core/pxelinux.0 core/isolinux.bin core/isolinux-debug.bin \
-	gpxe/gpxelinux.0 dos/syslinux.com \
-	win32/syslinux.exe win64/syslinux64.exe \
-	dosutil/*.com dosutil/*.sys \
+	gpxe/gpxelinux.0 \
 	$(MODULES)
 
 # BSUBDIRs build the on-target binary components.
@@ -55,7 +53,7 @@ BOBJECTS = $(BTARGET) \
 # files that depend only on the B phase, but may have to be regenerated
 # for "make installer".
 BSUBDIRS = codepage com32 lzo core memdisk modules mbr memdump gpxe sample \
-	   libinstaller dos win32 win64 dosutil
+	   libinstaller
 ITARGET  =
 IOBJECTS = $(ITARGET) \
 	utils/gethostip utils/isohybrid utils/mkdiskimage \
@@ -69,7 +67,6 @@ INSTALL_SBIN  = extlinux/extlinux
 # Things to install in /usr/lib/syslinux
 INSTALL_AUX   =	core/pxelinux.0 gpxe/gpxelinux.0 core/isolinux.bin \
 		core/isolinux-debug.bin \
-		dos/syslinux.com \
 		mbr/*.bin $(MODULES)
 INSTALL_AUX_OPT = win32/syslinux.exe win64/syslinux64.exe
 
-- 
1.7.3.2

