--- tools/plyheader.old.cpp	2011-04-15 23:15:24.574312747 -0700
+++ tools/plyheader.cpp	2011-04-15 23:17:02.857440649 -0700
@@ -1,3 +1,5 @@
+#include <cstdlib>
+#include <cstring>
 #include <fstream>
 #include <iostream>
 
@@ -38,7 +40,7 @@
       }
     }
 
-    if ((short_opt == 'h') || (std::strcmp(long_opt, "help") == 0)) {
+    if ((short_opt == 'h') || (strcmp(long_opt, "help") == 0)) {
       std::cout << "Usage: plyheader [OPTION] [[INFILE] OUTFILE]\n";
       std::cout << "Extract the header from a PLY file.\n";
       std::cout << "\n";
@@ -51,7 +53,7 @@
       return EXIT_SUCCESS;
     }
 
-    else if ((short_opt == 'v') || (std::strcmp(long_opt, "version") == 0)) {
+    else if ((short_opt == 'v') || (strcmp(long_opt, "version") == 0)) {
       std::cout << "plyheader (" << PACKAGE_NAME << ") " << PACKAGE_VERSION << "\n";
       std::cout << "Copyright (C) 2007 " << PACKAGE_AUTHOR << "\n";
       std::cout << "\n";
@@ -90,7 +92,7 @@
   const char* ifilename = "";
   if (parc > 0) {
     ifilename = parv[0];
-    if (std::strcmp(ifilename, "-") != 0) {
+    if (strcmp(ifilename, "-") != 0) {
       ifstream.open(ifilename);
       if (!ifstream.is_open()) {
         std::cerr << "plyheader: " << ifilename << ": " << "no such file or directory" << "\n";
@@ -103,7 +105,7 @@
   const char* ofilename = "";
   if (parc > 1) {
     ofilename = parv[1];
-    if (std::strcmp(ofilename, "-") != 0) {
+    if (strcmp(ofilename, "-") != 0) {
       ofstream.open(ofilename);
       if (!ofstream.is_open()) {
         std::cerr << "plyheader: " << ofilename << ": " << "could not open file" << "\n";
