diff -c src/mkinit/src/mkinit src/build/src/mkinit
*** src/mkinit/src/mkinit	2011-03-22 06:42:38.000000000 +0000
--- src/build/src/mkinit	2011-03-22 07:01:58.000000000 +0000
***************
*** 72,78 ****
  		export MKINIT_STATE=/tmp/mkinit_state
  		mkdir -p $MKINIT_STATE
  	fi
! 	/usr/lib/plan9/bin/mk \
  		-f /etc/init.mk \
  		-i -k "$@"
  }
--- 72,78 ----
  		export MKINIT_STATE=/tmp/mkinit_state
  		mkdir -p $MKINIT_STATE
  	fi
! 	/opt/plan9/bin/mk \
  		-f /etc/init.mk \
  		-i -k "$@"
  }
diff -c src/mkinit/src/mkinit.rc src/build/src/mkinit.rc
*** src/mkinit/src/mkinit.rc	2011-03-22 06:42:38.000000000 +0000
--- src/build/src/mkinit.rc	2011-03-22 07:02:32.000000000 +0000
***************
*** 1,4 ****
! #!/usr/lib/plan9/bin/rc
  
  # Copyright (C) 2009 Andy Spencer
  # See ../COPYING for terms
--- 1,4 ----
! #!/opt/plan9/bin/rc
  
  # Copyright (C) 2009 Andy Spencer
  # See ../COPYING for terms
***************
*** 78,84 ****
  		MKINIT_STATE=/tmp/mkinit_state
  		mkdir -p $MKINIT_STATE
  	}
! 	/usr/lib/plan9/bin/mk \
  		-f /etc/init.mk \
  		-i -k $*
  }
--- 78,84 ----
  		MKINIT_STATE=/tmp/mkinit_state
  		mkdir -p $MKINIT_STATE
  	}
! 	/opt/plan9/bin/mk \
  		-f /etc/init.mk \
  		-i -k $*
  }
Lika underkataloger: src/mkinit/src/.svn och src/build/src/.svn
