--- src/rxvtdaemon.C	2012-01-19 14:34:39.000000000 +0100
+++ src/rxvtdaemon.C	2012-10-16 19:18:17.344715230 +0200
@@ -37,6 +37,7 @@
 {
   char name[PATH_MAX];
   char *path = getenv ("RXVT_SOCKET");
+  char *display = getenv("DISPLAY");
 
   if (!path)
     {
@@ -50,9 +51,10 @@
       snprintf (name, PATH_MAX, "%s/.urxvt", path);
       mkdir (name, 0777);
 
-      snprintf (name, PATH_MAX, "%s/.urxvt/urxvtd-%s",
-                path,
-                u.nodename);
+      if (!display)
+         snprintf (name, PATH_MAX, "%s/.urxvt/urxvtd-%s", path, u.nodename);
+      else
+         snprintf (name, PATH_MAX, "%s/.urxvt/urxvtd-%s-%s", path, u.nodename, display);
 
       path = name;
     }
