--- src/command.C	Wed Dec 21 22:59:04 2011 +0100
+++ src/command.C	Wed Dec 21 23:01:28 2011 +0100
@@ -2197,10 +2197,46 @@
                 }
               else
 # endif
+#ifndef NO_SECONDARY_SCREEN
                 {
-                  scr_page (v, i);
-                  scrollBar.show (1);
+                  /* on SECONDARY screen, we send "fake" UP/DOWN keys instead
+                   * (this allows to scroll within man, less, etc) */
+                  if (option (Opt_secondaryWheel) && current_screen != PRIMARY)
+                    {
+                       XKeyEvent event;
+                       event.display     = ev.display;
+                       event.window      = ev.window;
+                       event.root        = ev.root;
+                       event.subwindow   = ev.subwindow;
+                       event.time        = ev.time;
+                       event.x           = ev.x;
+                       event.y           = ev.y;
+                       event.x_root      = ev.x_root;
+                       event.y_root      = ev.y_root;
+                       event.same_screen = ev.same_screen;
+                       event.state       = 0;
+                       event.keycode     = XKeysymToKeycode(ev.display,
+                                            (v == UP) ? XK_Up : XK_Down);
+                       for (i = 0; i < 3; ++i)
+                         {
+                            event.type = KeyPress;
+                            XSendEvent (event.display, event.window, TRUE,
+                                        KeyPressMask, (XEvent *) &event);
+                            event.type = KeyRelease;
+                            XSendEvent (event.display, event.window, TRUE,
+                                        KeyPressMask, (XEvent *) &event);
+                          }
+                    }
+                  /* on PRIMARY screen, we scroll in the buffer */
+                  else
+#endif
+                    {
+                       scr_page (v, i);
+                       scrollBar.show (1);
+                    }
+#ifndef NO_SECONDARY_SCREEN
                 }
+#endif
             }
             break;
 #endif
diff -r d5f9ea7306c4 -r cca1997c1a85 src/optinc.h
--- src/optinc.h	Wed Dec 21 22:59:04 2011 +0100
+++ src/optinc.h	Wed Dec 21 23:01:28 2011 +0100
@@ -26,6 +26,7 @@
  def(cursorBlink)
  def(secondaryScreen)
  def(secondaryScroll)
+ def(secondaryWheel)
  def(pastableTabs)
  def(cursorUnderline)
 #if ENABLE_FRILLS
diff -r d5f9ea7306c4 -r cca1997c1a85 src/rsinc.h
--- src/rsinc.h	Wed Dec 21 22:59:04 2011 +0100
+++ src/rsinc.h	Wed Dec 21 23:01:28 2011 +0100
@@ -102,6 +102,7 @@
 #ifndef NO_SECONDARY_SCREEN
   def (secondaryScreen)
   def (secondaryScroll)
+  def (secondaryWheel)
 #endif
 #ifdef OFF_FOCUS_FADING
   def (fade)
diff -r d5f9ea7306c4 -r cca1997c1a85 src/xdefaults.C
--- src/xdefaults.C	Wed Dec 21 22:59:04 2011 +0100
+++ src/xdefaults.C	Wed Dec 21 23:01:28 2011 +0100
@@ -261,6 +261,7 @@
 #ifndef NO_SECONDARY_SCREEN
               BOOL (Rs_secondaryScreen, "secondaryScreen", "ssc", Opt_secondaryScreen, 0, "enable secondary screen"),
               BOOL (Rs_secondaryScroll, "secondaryScroll", "ssr", Opt_secondaryScroll, 0, "enable secondary screen scroll"),
+              BOOL (Rs_secondaryWheel, "secondaryWheel", "ssw", Opt_secondaryWheel, 0, "enable secondary screen wheel"),
 #endif
 #if ENABLE_PERL
               RSTRG (Rs_perl_lib, "perl-lib", "string"), //, "colon-separated directories with extension scripts"),TODO
