diff -aur package.pristine/include/defines_linux.h package.new/include/defines_linux.h
--- package.pristine/include/defines_linux.h	2015-02-24 13:27:37.419344000 +0100
+++ package.new/include/defines_linux.h	2015-02-24 13:29:37.808653542 +0100
@@ -1,6 +1,8 @@
 #ifndef DEFINES_LINUX_H
 #define DEFINES_LINUX_H
 
+#define RESOURCES_DIR "/usr/share/equalx/resources"
+
 // Editor
 #define DEFAULT_FONT "Monospace,10,-1,5,50,0,0,0,0,0"
 
