--- MsnSniff.cpp~	2009-10-13 08:14:34.760367106 +0600
+++ MsnSniff.cpp	2009-10-13 08:14:53.977034437 +0600
@@ -12,6 +12,8 @@
 #include <arpa/inet.h>
 #include <csignal>
 #include <regex.h>
+#include <cstring>
+#include <cstdlib>
 
 #include "MsnSniffSettings.h"
 #include "MsnSniffUI.h"
--- MsnSniffUI.cpp~	2009-10-13 08:16:21.293689654 +0600
+++ MsnSniffUI.cpp	2009-10-13 08:16:57.073686584 +0600
@@ -5,6 +5,7 @@
 #include <string>
 #include <cassert>
 #include <ctime>
+#include <cstdlib>
 
 namespace MsnSniffUI
 {
--- MsnSniffSettings.cpp~	2009-10-13 08:17:21.597016736 +0600
+++ MsnSniffSettings.cpp	2009-10-13 08:18:22.797011742 +0600
@@ -1,4 +1,5 @@
 #include "MsnSniffSettings.h"
+#include <cstring>
 
 namespace MsnSniffSettings
 {
--- MsnMsg.cpp~	2009-10-13 08:19:03.493673800 +0600
+++ MsnMsg.cpp	2009-10-13 08:20:48.680810558 +0600
@@ -1,5 +1,7 @@
 #include "MsnMsg.h"
 #include "UrlDecode.h"
+#include <cstring>
+#include <cstdio>
 
 using namespace std;
 
--- UrlDecode.cpp~	2009-10-13 08:21:25.024493466 +0600
+++ UrlDecode.cpp	2009-10-13 08:22:23.106986181 +0600
@@ -1,4 +1,6 @@
 #include "UrlDecode.h"
+#include <cstring>
+#include <cstdio>
 
 using namespace std;
 
