*** ./configure	2010-06-10 21:58:00.478849399 +0200
--- ./configure.new	2010-06-10 22:07:36.649625270 +0200
***************
*** 9481,9485 ****
--- 9481,9493 ----
  EOF
  fi
  
+ # copy disabled features to a file
+ cat > disabled_features << EOF
+ Input: $noinputmodules
+ Codecs: $nocodecmodules
+ Audio output: $noaomodules
+ Video output: $novomodules
+ EOF
+ 
  # Last move:
  rm -f "$TMPEXE" "$TMPC" "$TMPS" "$TMPCPP" "$TMPH"
