--- xorg-server-1.8.2/configure.ac	2010-04-02 09:12:10.000000000 +0200
+++ xorg-server-1.8.2/configure.ac	2010-04-07 03:14:49.928744549 +0200
@@ -26,7 +26,7 @@
 dnl Process this file with autoconf to create configure.

 AC_PREREQ(2.57)
-AC_INIT([xorg-server], 1.8.2, [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg], xorg-server)
+AC_INIT([xorg-server], 1.7.5.1, [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg], xorg-server)
 RELEASE_DATE="2010-07-01"
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
