--- gtk+-2.91.7/gdk/x11/gdkxid.c	2010-12-21 19:07:07.000000000 +0200
+++ gtk+-2.91.7/gdk/x11/gdkxid.c.new	2010-12-22 09:17:34.180000003 +0200
@@ -21,7 +21,7 @@
  * Modified by the GTK+ Team and others 1997-2000.  See the AUTHORS
  * file for a list of people on the GTK+ Team.  See the ChangeLog
  * files for a list of changes.  These files are distributed with
- * GTK+ at ftp://ftp.gtk.org/pub/gtk/. 
+ * GTK+ at ftp://ftp.gtk.org/pub/gtk/.
  */
 
 #include "config.h"
@@ -58,10 +58,10 @@
   if (!display_x11->xid_ht)
     display_x11->xid_ht = g_hash_table_new ((GHashFunc) gdk_xid_hash,
                                             (GEqualFunc) gdk_xid_equal);
-
+/*
   if (g_hash_table_lookup (display_x11->xid_ht, xid))
     g_warning ("XID collision, trouble ahead");
-
+*/
   g_hash_table_insert (display_x11->xid_ht, xid, data);
 }
 
