--- test.c.old	2009-08-06 23:07:00.000000000 +0100
+++ test.c.new	2013-05-28 00:16:02.640490919 +0100
@@ -4,6 +4,7 @@
 
 #include <stdio.h>
 #include <sys/stat.h>
+#include <stdlib.h>
 #include "test.h"
 
 #define LZWVL_PACKER_VERSION 1
@@ -115,7 +116,7 @@
 		return 0;
 	}
 
-	fstat(inFile->_file, &f);
+	fstat(fileno(inFile), &f);
 	inFileSize = f.st_size;
 
 	if(!quiet) printf("Input file size : %d\n", inFileSize);
@@ -142,10 +143,10 @@
  * How to use this.
  */
 void printUsage() {
-		printf("Usage : test.exe -[vql] infile outfile\n");
-		printf("                  -v   verbose mode\n");
-		printf("                  -l   long mode\n");
-		printf("                  -q   quiet mode\n");
+		printf("Usage : lzwvl -[vql] infile outfile\n");
+		printf("              -v   verbose mode\n");
+		printf("              -l   long mode\n");
+		printf("              -q   quiet mode\n");
 }
 
 
@@ -251,7 +252,7 @@
 
 
 	struct stat f;
-	fstat(outFile->_file, &f);
+	fstat(fileno(outFile), &f);
 	int outFileSize = f.st_size;
 
     if(!quiet) {
