--- Makefile		2008-08-20 22:58:24.000000000 +0200
+++ Makefile		2008-08-20 23:15:14.000000000 +0200
@@ -1,6 +1,6 @@
-glib_inc = /usr/local/include/glib-2.0
-glib_dir = /usr/local/lib/glib-2.0
-irssi_dir = /usr/local/include/irssi
+glib_inc = /usr/include/glib-2.0
+glib_dir = /usr/lib/glib-2.0
+irssi_dir = /usr/include/irssi
 
 # *** to find glib directories, try:
 # >>> pkg-config --cflags glib-2.0
@@ -73,7 +73,7 @@
 	@echo    "   cp libfish.so ~/.irssi/modules; cp libfish.so /usr/local/lib/irssi/modules/"
 	@echo    "--------------------------------------------------------------------------------"
 	@echo
-	@echo "Press ENTER to continue or CTRL+C to abort..."; read junk
+	@echo "Press ENTER to continue or CTRL+C to abort..."
 
 	@if [ ! -f "miracl.a" ] && [ ! -f "MIRACL/miracl.a" ]; then echo "ERROR: miracl.a not found! You didn't compile MIRACL, did you?"; echo; exit 1; fi
 	@if [ -f "MIRACL/miracl.a" ]; then echo "Okay: Using $(PWD)/MIRACL/miracl.a ..."; cp MIRACL/miracl.a miracl.a; echo; fi
