--- mutt-1.5.21/sendlib.c	2010-09-13 19:19:55.000000000 +0200
+++ mutt-android/sendlib.c	2012-10-02 16:41:24.466737391 +0200
@@ -363,8 +363,6 @@
 
     if (a->disposition < sizeof(dispstr)/sizeof(char*))
     {
-      fprintf (f, "Content-Disposition: %s", dispstr[a->disposition]);
-
       if (a->use_disp)
       {
 	if (!(fn = a->d_filename))
@@ -372,6 +370,9 @@
 
 	if (fn)
 	{
+      /* We only print this line if there is a file. */
+      fprintf (f, "Content-Disposition: %s", dispstr[a->disposition]);
+
 	  char *tmp;
 
 	  /* Strip off the leading path... */
@@ -386,10 +387,10 @@
 	  rfc822_cat (buffer, sizeof (buffer), tmp, MimeSpecials);
 	  FREE (&tmp);
 	  fprintf (f, "; filename%s=%s", encode ? "*" : "", buffer);
+      fputc ('\n', f);
 	}
       }
 
-      fputc ('\n', f);
     }
     else
     {
