--- configure.orig	2008-07-24 17:20:51.000000000 +0300
+++ configure	2008-09-05 06:52:38.000000000 +0300
@@ -4379,7 +4379,7 @@ echo "${ECHO_T}$am_cv_python_pyexecdir" 
 
 # Find Python compiler and linker flags
 PYTHON_LIBS=$($PYTHON -c "from distutils import sysconfig
-print '/'.join(sysconfig.get_config_vars('LIBPL','LDLIBRARY'))+' '+sysconfig.get_config_var('LIBS')+' '+sysconfig.get_config_var('LINKFORSHARED')")
+print '/'.join(sysconfig.get_config_vars('LIBDIR','LDLIBRARY'))+' '+sysconfig.get_config_var('LIBS')+' '+sysconfig.get_config_var('LINKFORSHARED')")
 PYTHON_CFLAGS=$($PYTHON -c "from distutils import sysconfig
 print sysconfig.get_python_inc()")
 PYTHON_CFLAGS="-I$PYTHON_CFLAGS"
