diff -Naur src/http.c src2/http.c
--- src/http.c	2011-04-13 13:24:26.000000000 +0100
+++ src2/http.c	2011-07-09 21:51:56.000000000 +0100
@@ -22,7 +22,7 @@
 #include <string.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
+//#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <glib-object.h>
diff -Naur src/main.c src2/main.c
--- src/main.c	2011-04-23 22:09:56.000000000 +0100
+++ src2/main.c	2011-07-09 21:51:35.000000000 +0100
@@ -24,7 +24,7 @@
 #include <stdlib.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
+//#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <glib/gi18n.h>