diff -aur id34mtp/id34mtp.c id34mtp.diff/id34mtp.c
--- id34mtp/id34mtp.c	2006-03-21 17:00:25.000000000 +0100
+++ id34mtp.diff/id34mtp.c	2006-05-06 20:46:53.000000000 +0200
@@ -40,7 +40,8 @@
 uint64_t filesize;
 struct stat sb;
 
-static int progress (uint64_t const sent, uint64_t const total, uint8_t const * const buf, uint32_t const len, void const * const data)
+static const LIBMTP_progressfunc_t mtp_progress;
+static int mtp_progress (uint64_t const sent, uint64_t const total, void const * const data)
 {
   int percent = (sent*100)/total;
   printf("Progress: %llu of %llu (%d%%)\r", sent, total, percent);
@@ -254,7 +255,7 @@
             	LIBMTP_track_t *trackmeta;
             	LIBMTP_Init();
             	trackmeta = LIBMTP_new_track_t();
-            	trackmeta->codec = LIBMTP_CODEC_MP3;
+            	trackmeta->filetype = LIBMTP_FILETYPE_MP3;
             	trackmeta->title = strdup(mp3.id3.title);
             	trackmeta->album = strdup(mp3.id3.album);
             	trackmeta->artist = strdup(mp3.id3.artist);
@@ -275,7 +276,7 @@
 					LIBMTP_destroy_track_t(trackmeta);
 					exit(1);
 					}
-				ret = LIBMTP_Send_Track_From_File(device, path, trackmeta, progress, NULL);
+				ret = LIBMTP_Send_Track_From_File(device, path, trackmeta, mtp_progress, NULL, 0);
 				printf("\n");
 		LIBMTP_Release_Device(device);
 				printf("New track ID: %d\n", trackmeta->item_id);
