*** init/rdiff-web	2008-03-05 17:49:11.000000000 +0100
--- init/rdiff-web.new	2010-04-15 21:40:59.144537697 +0200
***************
*** 1,4 ****
! #!/bin/sh
  
  # Basic support for the Linux Standard Base Specification 3.1.0
  ### BEGIN INIT INFO
--- 1,4 ----
! #!/bin/bash
  
  # Basic support for the Linux Standard Base Specification 3.1.0
  ### BEGIN INIT INFO
***************
*** 10,53 ****
  # Description: Manages the rdiffWeb daemon
  ### END INIT INFO
  
! PIDFILE="/var/run/rdiff-web"
  
! SUCCESS_MSG="[71G done"
! ERROR_MSG="[71Gfailed"
  
- if [ -f /etc/init.d/functions ]; then
-  . /etc/init.d/functions
- fi
- 
- # This script won't be able to run without root privileges.
- if [ `id -u` -ne 0 ]; then
-    echo "Error: this script must be run as root."
-    exit 1
- fi
- 
- 
- # See how we were called.
  case "$1" in
     start)
        if [ ! -f /etc/rdiffweb/rdw.conf ]; then
           echo -n "rdiffWeb is not configured!  Please run 'rdiff-web-config'."
!          echo $ERROR_MSG
!          exit 1
        fi
!       echo -n "Starting rdiffWeb..."
!       /usr/bin/rdiff-web --pid-file="$PIDFILE" \
!       --background > /dev/null 2>&1
!       echo $SUCCESS_MSG
!       exit 0
     ;;
  
     stop)
        if [ -e "$PIDFILE" ]; then
!          echo -n "Stopping rdiffWeb..."
           kill `cat "$PIDFILE" 2>/dev/null` > /dev/null 2>&1
           rm $PIDFILE > /dev/null 2>&1
!          echo $SUCCESS_MSG
!          exit 0
        else
           echo -n "rdiffWeb is not running"
           echo $ERROR_MSG
--- 10,39 ----
  # Description: Manages the rdiffWeb daemon
  ### END INIT INFO
  
! PIDFILE="/var/run/rdiff-web.pid"
  
! . /etc/rc.conf
! . /etc/rc.d/functions
  
  case "$1" in
     start)
+    	  stat_busy "Starting rdiff WebInterface"
        if [ ! -f /etc/rdiffweb/rdw.conf ]; then
           echo -n "rdiffWeb is not configured!  Please run 'rdiff-web-config'."
! 		 stat_fail
        fi
!       /usr/bin/rdiff-web --pid-file="$PIDFILE" --background > /dev/null 2>&1 -d
! 	  add_daemon rdiffweb
! 	  stat_done
     ;;
  
     stop)
        if [ -e "$PIDFILE" ]; then
!          stat_busy "Stopping rdiff WebInterface"
           kill `cat "$PIDFILE" 2>/dev/null` > /dev/null 2>&1
           rm $PIDFILE > /dev/null 2>&1
! 		 rm_daemon rdiffweb
! 		 stat_done
        else
           echo -n "rdiffWeb is not running"
           echo $ERROR_MSG
***************
*** 56,68 ****
     ;;
  
     status)
-       echo -n "rdiffWeb "
        if [ -e "$PIDFILE" ]; then
!          echo "is running."
!          exit 0
        else
!          echo "is not running."
!          exit 1
        fi
     ;;
  
--- 42,53 ----
     ;;
  
     status)
        if [ -e "$PIDFILE" ]; then
!          stat_busy "rdiffWeb is running."
! 		 stat_done
        else
!          stat_busy "rdiffWeb is not running."
! 		 stat_fail
        fi
     ;;
  
***************
*** 74,78 ****
        echo "Usage: `basename "$0"` {start|stop|status|restart}"
        exit 1
  esac
- 
  exit 0
--- 59,62 ----
