--- setup.py	2014-12-07 11:57:50.932533773 +0100
+++ setup.py.new	2014-12-07 11:58:05.879119467 +0100
@@ -18,7 +18,6 @@ setup(name='rdiffWeb',
    packages=['rdiffWeb'],
    package_data={'rdiffWeb': ['templates/*.html', 'templates/*.xml', 'templates/*.txt', 'static/*.png', 'static/*.js', 'static/*.css', 'static/images/*']},
    data_files=[('/etc/rdiffweb', ['rdw.conf.sample']),
-               ('/etc/init.d', ['init/rdiff-web']),
                ('/usr/bin', ['rdiff-web']), # Manually place rdiff-web in /usr/bin, so the init script can find it
                ],
    scripts=['rdiff-web-config']
