*** setup.py	2008-03-05 17:49:13.000000000 +0100
--- setup.py.new	2010-04-15 21:31:41.807926469 +0200
***************
*** 18,24 ****
     packages=['rdiffWeb'],
     package_data={'rdiffWeb': ['templates/*.html', 'templates/*.xml', 'templates/*.txt', 'static/*.png', 'static/*.js', 'static/*.css', 'static/images/*']},
     data_files=[('/etc/rdiffweb', ['rdw.conf.sample']),
!                ('/etc/init.d', ['init/rdiff-web']),
                 ('/usr/bin', ['rdiff-web']), # Manually place rdiff-web in /usr/bin, so the init script can find it
                 ],
     scripts=['rdiff-web-config']
--- 18,24 ----
     packages=['rdiffWeb'],
     package_data={'rdiffWeb': ['templates/*.html', 'templates/*.xml', 'templates/*.txt', 'static/*.png', 'static/*.js', 'static/*.css', 'static/images/*']},
     data_files=[('/etc/rdiffweb', ['rdw.conf.sample']),
!                ('/etc/rc.d', ['init/rdiff-web']),
                 ('/usr/bin', ['rdiff-web']), # Manually place rdiff-web in /usr/bin, so the init script can find it
                 ],
     scripts=['rdiff-web-config']
