--- mondo-2.2.9.5/src/mondorestore/mondo-rstr-tools.c	2010-09-09 23:13:40.000000000 +0100
+++ mondo-2.2.9.5/src/mondorestore/mondo-rstr-tools-new.c	2011-04-06 01:23:24.000000000 +0100
@@ -1239,7 +1239,11 @@
 	if (ask_me_yes_or_no
 		("You will now be able to re-generate your initrd.\nThis is especially useful if you changed of hardware configuration, cloned, made P2V, used multipath...\nDo you need to do it ?")) {
 		log_msg(1,"Launching shell for manual initrd recreation");
-		popup_and_OK("You'll now be chrooted under your future / partition.\nGo under /boot and rebuild your initrd with\nmkinitrd -f -v initrd-2.x.y.img 2.x.y e.g.\nThen type exit to finish.");
+		if (does_file_exist(MNT_RESTORING"/etc/arch-release")) {
+			popup_and_OK("You'll now be chrooted under your future / partition.\nEdit /etc/mkinitcpio.conf if needed and rebuild your initrd with the kernel preset name e.g.\nmkinitcpio -p kernel26\nThen type exit to finish.\n");
+		} else {
+			popup_and_OK("You'll now be chrooted under your future / partition.\nGo under /boot and rebuild your initrd with\nmkinitrd -f -v initrd-2.x.y.img 2.x.y e.g.\nThen type exit to finish.");
+		}
 		mvaddstr_and_log_it(g_currentY, 0, "Modifying initrd...");
 		if (!g_text_mode) {
 			newtSuspend();
