diff --git a/jni/pmdmini/Makefile b/jni/pmdmini/Makefile
index 2bd9364..a315c54 100644
--- a/jni/pmdmini/Makefile
+++ b/jni/pmdmini/Makefile
@@ -9,6 +9,7 @@ SRC_PMDDIR = $(SRC_DIR)/pmdwin
 
 CFLAGS += -I$(SRC_DIR)
 CFLAGS += -I$(SRC_PMDDIR)
+CFLAGS += -fpermissive
 
 SDL_LIBS = `sdl-config --libs`
 SDL_CFLAGS = `sdl-config --cflags`
diff --git a/jni/pmdmini/mak/lib.mak b/jni/pmdmini/mak/lib.mak
index eb0862d..e384b04 100644
--- a/jni/pmdmini/mak/lib.mak
+++ b/jni/pmdmini/mak/lib.mak
@@ -4,7 +4,7 @@ PMDMINI_DIR = src
 FMGEN_DIR = src/fmgen
 PMDWIN_DIR = src/pmdwin
 
-FMGEN_CFLAGS = -I$(FMGEN_DIR)
+FMGEN_CFLAGS = -I$(FMGEN_DIR) -fpermissive
 PMDWIN_CFLAGS = -I$(PMDWIN_DIR)
 
 
