--- woof-2009-12-27.py	2011-03-31 20:29:40.000000000 +0200
+++ woof-2009-12-27.py	2011-03-31 21:28:32.355233473 +0200
@@ -34,7 +34,6 @@
 
 maxdownloads = 1
 TM = object
-cpid = -1
 compressed = 'gz'
 upload = False
 
@@ -209,7 +208,7 @@
       
 
    def do_GET (self):
-      global maxdownloads, cpid, compressed, upload
+      global maxdownloads, compressed, upload
 
       # Form for uploading a file
       if upload:
@@ -266,11 +265,7 @@
       # let a separate process handle the actual download, so that
       # multiple downloads can happen simultaneously.
 
-      cpid = os.fork ()
-
-      if cpid == 0:
-         # Child process
-         child = None
+      if True:
          type = None
          
          if os.path.isfile (self.filename):
@@ -340,7 +335,7 @@
    if ip_addr:
       print("Now serving on http://%s:%s/" % (ip_addr, httpd.server_port))
 
-   while cpid != 0 and maxdownloads > 0:
+   while maxdownloads > 0:
       httpd.handle_request ()
 
 
@@ -389,7 +384,7 @@
 
 
 def main ():
-   global cpid, upload, compressed
+   global upload, compressed
 
    maxdown = 1
    port = 8080
@@ -491,16 +486,6 @@
 
    serve_files (filename, maxdown, ip_addr, port)
 
-   # wait for child processes to terminate
-   if cpid != 0:
-      try:
-         while 1:
-            os.wait ()
-      except OSError:
-         pass
-
-
-
 if __name__=='__main__':
    try:
       main ()
