From ceca84f81e462255de03aeea0fa2b8aedb064645 Mon Sep 17 00:00:00 2001
From: Reihar <rar501@gmail.com>
Date: Sun, 31 Aug 2014 00:48:30 +0200
Subject: [PATCH] Change setup.py's shebang

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 6bcc7f7..b0d3f29 100755
--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 import os
-- 
2.1.0

