diff -Naur spider.orig/defs.h spider/defs.h
--- spider.orig/defs.h	1991-09-28 19:46:21.000000000 +0200
+++ spider/defs.h	2011-08-20 03:21:42.411243136 +0200
@@ -17,6 +17,7 @@
 #include	<X11/Xutil.h>
 #include	<X11/Xos.h>
 #include	<stdio.h>
+#include	<stdlib.h>
 #include	"assert.h"
 #include	"copyright.h"
 
@@ -31,7 +32,7 @@
 
 typedef enum	{Faceup, Facedown, Joker}	Type;
 
-#else DEBUG
+#else /* DEBUG */
 
 typedef	char	Suit;
 typedef	char	Rank;
@@ -60,7 +61,7 @@
 #define	Facedown	1
 #define	Joker		2
 
-#endif DEBUG
+#endif /* DEBUG */
 
 #define	NUM_DECKS	2
 #define	NUM_PILES	8
diff -Naur spider.orig/gfx.c spider/gfx.c
--- spider.orig/gfx.c	1991-09-28 19:46:10.000000000 +0200
+++ spider/gfx.c	2011-08-20 04:07:56.200657689 +0200
@@ -166,7 +166,7 @@
 	gcflags |= GCFont;
 
 	textgc = XCreateGC(dpy, RootWindow(dpy, screen), gcflags, &gcv);
-#endif KITLESS
+#endif /* KITLESS */
 
 	tmpmap = XCreateBitmapFromData(dpy, RootWindow(dpy, screen),
 		logo_bits, logo_width, logo_height);
@@ -1321,7 +1321,7 @@
 	XDrawImageString(dpy, message_win, textgc, MESSAGE_X, message_y,
 		last_message, strlen(last_message));
 }
-#endif	KITLESS
+#endif	/* KITLESS */
 
 void
 card_message(str, card)
diff -Naur spider.orig/globals.h spider/globals.h
--- spider.orig/globals.h	1991-09-28 19:46:20.000000000 +0200
+++ spider/globals.h	2011-08-20 03:18:35.033442225 +0200
@@ -90,12 +90,12 @@
 void	button_press();
 void	button_release();
 void	do_expand();
-#endif KITLESS
+#endif /* KITLESS */
 
 #ifdef XAW
 Bool	can_get_help_files();
 #endif
 
-extern char	*malloc();
-extern char	*calloc();
-extern char	*realloc();
+extern void	*malloc();
+extern void	*calloc();
+extern void	*realloc();
diff -Naur spider.orig/movelog.c spider/movelog.c
--- spider.orig/movelog.c	1991-09-28 19:46:17.000000000 +0200
+++ spider/movelog.c	2011-08-20 03:22:50.550140339 +0200
@@ -33,6 +33,21 @@
 
 extern int	cheat_count;
 
+Rank	flip_ranks[NUM_RANKS] =	{
+	King, Queen, Jack, Ten, Nine, Eight, Seven, 
+	Six, Five, Four, Three, Deuce, Ace
+};
+
+static int
+card_to_int(card)
+CardPtr	card;
+{
+int	val;
+
+	val = card->suit * 13 + flip_ranks[card->rank];
+	return (val);
+}
+
 make_deck_cache()
 {
 CardPtr	tmp;
@@ -280,11 +295,6 @@
 	}
 }
 
-Rank	flip_ranks[NUM_RANKS] =	{
-	King, Queen, Jack, Ten, Nine, Eight, Seven, 
-	Six, Five, Four, Three, Deuce, Ace
-};
-
 static void
 int_to_card(val, suit, rank)
 int	val;
@@ -298,16 +308,6 @@
 	assert (*rank >= Ace && *suit <= King);
 }
 
-static int
-card_to_int(card)
-CardPtr	card;
-{
-int	val;
-
-	val = card->suit * 13 + flip_ranks[card->rank];
-	return (val);
-}
-
 static CardPtr
 find_card(cache, suit, rank)
 CardPtr	*cache;
@@ -709,6 +709,23 @@
 	}
 }
 
+static int
+restore_game(str, str2)
+char	*str, *str2;
+{
+
+	if (read_position(str) != 0)	{
+		return (-1);
+	}
+
+	init_cache();		/* clear out the move cache */
+
+	if (str2)
+		read_moves(str2);
+
+	return (0);
+}
+
 read_selection(buf)
 char	*buf;
 {
@@ -786,23 +803,6 @@
 	show_message(buf);
 }
 
-static int
-restore_game(str, str2)
-char	*str, *str2;
-{
-
-	if (read_position(str) != 0)	{
-		return (-1);
-	}
-
-	init_cache();		/* clear out the move cache */
-
-	if (str2)
-		read_moves(str2);
-
-	return (0);
-}
-
 /*
  * play the same deck again
  */
diff -Naur spider.orig/spider.c spider/spider.c
--- spider.orig/spider.c	1991-09-28 19:46:12.000000000 +0200
+++ spider/spider.c	2011-08-20 04:19:23.259260596 +0200
@@ -619,7 +619,7 @@
 
 	return (type_names[type]);
 }
-#endif DEBUG
+#endif /* DEBUG */
 
 
 /*
diff -Naur spider.orig/windows.c spider/windows.c
--- spider.orig/windows.c	1991-09-28 19:46:15.000000000 +0200
+++ spider/windows.c	2011-08-20 03:23:34.517277892 +0200
@@ -116,7 +116,7 @@
 	XMapWindow(dpy, message_win);
 	XMapWindow(dpy, table);
 }
-#endif KITLESS
+#endif /* KITLESS */
 
 #ifndef	KITLESS
 table_init(win)
@@ -140,4 +140,4 @@
         }
         XChangeWindowAttributes(dpy, table, winmask, &winattr);
 }
-#endif	KITLESS
+#endif	/* KITLESS */
diff -Naur spider.orig/xaw_ui.c spider/xaw_ui.c
--- spider.orig/xaw_ui.c	1991-09-28 19:46:51.000000000 +0200
+++ spider/xaw_ui.c	2011-08-20 04:03:09.000000000 +0200
@@ -429,30 +429,25 @@
 	confirm_box = XtCreatePopupShell("prompt", transientShellWidgetClass,
 		toplevel, args, n);
 
-	n = 0;
 	confirmpane = XtCreateManagedWidget("confirmbox", panedWidgetClass,
 		confirm_box, args, n);
 
-	n = 0;
 	XtSetArg(args[n], XtNshowGrip, False);	n++;
 	confirm_label = XtCreateManagedWidget("confirmlabel", labelWidgetClass,
 		confirmpane, args, n);
 
-	n = 0;
 	XtSetArg(args[n], XtNshowGrip, False);	n++;
 	XtSetArg(args[n], XtNorientation, XtorientHorizontal);	n++;
 	confirmbox = XtCreateManagedWidget("confirmbox", boxWidgetClass,
 		confirmpane, args, n);
 
-	n = 0;
 	button = XtCreateManagedWidget("confirmok", commandWidgetClass,
 		confirmbox, args, n);
-	XtAddCallback(button, XtNcallback, confirm_callback, 1);
+	XtAddCallback(button, XtNcallback, confirm_callback, (int *) 1);
 
-	n = 0;
 	button = XtCreateManagedWidget("confirmcancel", commandWidgetClass,
 		confirmbox, args, n);
-	XtAddCallback(button, XtNcallback, confirm_callback, 2);
+	XtAddCallback(button, XtNcallback, confirm_callback, (int *) 2);
 }
 
 
@@ -489,66 +484,54 @@
 {
 Widget	help, helpbox, helpform, helpview;
 Widget	button;
-int	n;
+int	n = 0;
 Arg	args[10];
 
-	n = 0;
 	help = XtCreatePopupShell("Spider Help", topLevelShellWidgetClass,
 		toplevel, args, n);
 
-	n = 0;
 	helpbox = XtCreateManagedWidget("helpbox", panedWidgetClass,
 		help, args, n);
 
-	n = 0;
 	XtSetArg(args[n], XtNshowGrip, False);	n++;
 	XtSetArg(args[n], XtNorientation, XtorientHorizontal);	n++;
 	helpform = XtCreateManagedWidget("helpform", boxWidgetClass,
 		helpbox, args, n);
 
-	n = 0;
 	button = XtCreateManagedWidget("helpintro", commandWidgetClass,
 		helpform, args, n);
-	XtAddCallback(button, XtNcallback, change_help, 1);
+	XtAddCallback(button, XtNcallback, change_help, (int *) 1);
 
-	n = 0;
 	button = XtCreateManagedWidget("helprules", commandWidgetClass,
 		helpform, args, n);
-	XtAddCallback(button, XtNcallback, change_help, 2);
+	XtAddCallback(button, XtNcallback, change_help, (int *) 2);
 
-	n = 0;
 	button = XtCreateManagedWidget("helpcontrols", commandWidgetClass,
 		helpform, args, n);
-	XtAddCallback(button, XtNcallback, change_help, 3);
+	XtAddCallback(button, XtNcallback, change_help, (int *) 3);
 
-	n = 0;
 	button = XtCreateManagedWidget("helpexamples", commandWidgetClass,
 		helpform, args, n);
-	XtAddCallback(button, XtNcallback, change_help, 4);
+	XtAddCallback(button, XtNcallback, change_help, (int *) 4);
 
-	n = 0;
 	button = XtCreateManagedWidget("helpmisc", commandWidgetClass,
 		helpform, args, n);
-	XtAddCallback(button, XtNcallback, change_help, 5);
+	XtAddCallback(button, XtNcallback, change_help, (int *) 5);
 
-	n = 0;
 	button = XtCreateManagedWidget("helpsummary", commandWidgetClass,
 		helpform, args, n);
-	XtAddCallback(button, XtNcallback, change_help, 6);
+	XtAddCallback(button, XtNcallback, change_help, (int *) 6);
 
-	n = 0;
 	button = XtCreateManagedWidget("helpexit", commandWidgetClass,
 		helpform, args, n);
-	XtAddCallback(button, XtNcallback, change_help, 7);
+	XtAddCallback(button, XtNcallback, change_help, (int *) 7);
 
-	n = 0;
 	XtSetArg(args[n], XtNshowGrip, False);	n++;
 	XtSetArg(args[n], XtNallowHoriz, True);	n++;
 	XtSetArg(args[n], XtNallowVert, True);	n++;
 	helpview = XtCreateManagedWidget("helpview", viewportWidgetClass,
 		helpbox, args, n);
 
-	n = 0;
 	XtSetArg(args[n], XtNheight, 300);	n++;
 	XtSetArg(args[n], XtNwidth, 550);	n++;
 	XtSetArg(args[n], XtNtype, XawAsciiFile);	n++;
