--- a/daemon/memcached.c	2012-05-24 15:28:11.098970777 -0500
+++ b/daemon/memcached.c	2012-05-24 15:36:27.736651204 -0500
@@ -6833,7 +6833,6 @@
     int c;
     bool lock_memory = false;
     bool do_daemonize = false;
-    bool preallocate = false;
     int maxcore = 0;
     char *username = NULL;
     char *pid_file = NULL;
@@ -7029,7 +7028,6 @@
             break;
         case 'L' :
             if (enable_large_pages() == 0) {
-                preallocate = true;
                 old_opts += sprintf(old_opts, "preallocate=true;");
             }
             break;
@@ -7434,7 +7432,7 @@
         udp_port = settings.udpport ? settings.udpport : settings.port;
 
         /* create the UDP listening socket and bind it */
-        if (settings.udpport && server_sockets(settings.udpport, udp_transport,
+        if (settings.udpport && server_sockets(udp_port, udp_transport,
                                                portnumber_file)) {
             settings.extensions.logger->log(EXTENSION_LOG_WARNING, NULL,
                                             "failed to listen on UDP port %d: %s",
