--- old/configure	2005-12-16 08:17:03.000000000 -0600
+++ patch/configure	2010-11-28 01:21:27.800000072 -0600
@@ -2642,18 +2642,18 @@
 fi
 echo "$as_me:$LINENO: result: $ac_cv_lib_ssl_RSA_new" >&5
 echo "${ECHO_T}$ac_cv_lib_ssl_RSA_new" >&6
-if test $ac_cv_lib_ssl_RSA_new = yes; then
+#if test $ac_cv_lib_ssl_RSA_new = yes; then
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBSSL 1
 _ACEOF
 
   LIBS="-lssl $LIBS"
 
-else
-
-		    echo "Error! You need libssl."
-		    exit -1
-fi
+#else
+#
+#		    echo "Error! You need libssl."
+#		    exit -1
+#fi
 
 
 
