diff -aur old/aexcl/aexcl_play.cxx patch/aexcl/aexcl_play.cxx
--- old/aexcl/aexcl_play.cxx	2010-11-28 01:26:40.913333406 -0600
+++ patch/aexcl/aexcl_play.cxx	2010-11-28 01:29:56.546666737 -0600
@@ -206,7 +206,7 @@
 	if(!aexd.agui->aex_chooser->size()) return -1;
 	if(aexd.agui->aex_chooser->value()<0) return -1;
 	rargv[1]=aexd.agui->aex_chooser->text(aexd.agui->aex_chooser->value());
-	if(!(astr=strstr(rargv[1],IP_NAME_DELIM))) return -1;
+	if(!(astr=(char*)strstr(rargv[1],IP_NAME_DELIM))) return -1;
 	memset(ip,0,sizeof(ip));
 	memcpy(ip,rargv[1],((__u32)(astr-rargv[1])<sizeof(ip))?astr-rargv[1]:sizeof(ip));
 	rargv[1]=ip;
diff -aur old/aexcl/Makefile.in patch/aexcl/Makefile.in
--- old/aexcl/Makefile.in	2010-11-28 01:26:40.913333406 -0600
+++ patch/aexcl/Makefile.in	2010-11-28 01:35:33.883333405 -0600
@@ -12,7 +12,7 @@
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 
 TARGET=aexcl_play
-GLIB_SUBST=1 # remove this line, if you want to use libglib-2.0
+#GLIB_SUBST=1 # remove this line, if you want to use libglib-2.0
 DESTDIR =
 
 CXXFLAGS += -Wall -D_GNU_SOURCE -I../raop_play -I../rendezvous
