Only in packer.orig: .git
diff -aur packer.orig/packer packer.new/packer
--- packer.orig/packer	2011-08-07 13:36:04.000000000 +1000
+++ packer.new/packer	2011-08-07 13:37:58.000000000 +1000
@@ -595,8 +595,14 @@
       results="$(pacman -Ssq -- "${packageargs[@]}")"
     else
       results="$(pacman -Ss -- "${packageargs[@]}")"
+      # Repo
       results="$(sed -r "s|^[^ ][^/]*/|$S${COLOR3}&$S${COLOR1}|" <<< "$results")"
+      # Package
       results="$(sed -r "s|^([^ ]+) ([^ ]+)(.*)$|\1 $S${COLOR2}\2$S${ENDCOLOR}\3|" <<< "$results")"
+      # Group
+      results="$(sed -r "s|^([^ ]+ [^ ]+ \[.*\]) (\(.*\))(.*)$|\1 $S${COLOR5}\2$S${ENDCOLOR}\3|" <<< "$results")"
+      # Status
+      results="$(sed -r "s|^([^ ]+ [^ ]+ \[.*\].*) (\[.*\])$|\1 $S${COLOR4}\2$S${ENDCOLOR}|" <<< "$results")"
     fi
     if [[ $option = search ]]; then
       echo -e "$results" | fmt -"$_WIDTH" -s
