--- src/master/src/gcconfig.pri.in	2013-03-10 18:42:17.755389782 +0000
+++ src/master/src/gcconfig.pri	2013-03-10 19:19:17.030033605 +0000
@@ -6,7 +6,7 @@
 
 # We recommend a debug build for development and a release build for deployment
 #CONFIG += debug
-#CONFIG += release
+CONFIG += release
 
 # If you want a console window to appear on Windows machines
 # then uncomment the following two lines.
@@ -16,16 +16,16 @@
 # To compile translation you need QT tool - lrelease
 # If it is not found using he defaults in src/src.pro
 # then set the full path and filename here.
-#QMAKE_LRELEASE = /usr/bin/lrelease
+QMAKE_LRELEASE = /usr/bin/lrelease-qt4
 
 # We use g++ on all platforms so switch on auto vectorization amongst other
 # things to speed up looping over ride file points
-#QMAKE_CXXFLAGS += -O3
+QMAKE_CXXFLAGS += -O3
 
 # Let us know where flex and bison are installed.
 # You may need to specify the full path if things don't work.
-#QMAKE_LEX  = flex
-#QMAKE_YACC = bison
+QMAKE_LEX  = flex
+QMAKE_YACC = bison
 #win32 {
 #  QMAKE_YACC = bison --file-prefix=y -t
 #  QMAKE_MOVE = cmd /c move
@@ -50,7 +50,7 @@
 #    http://www.ftdichip.com/Drivers/D2XX.htm
 # Set path to where the D2XX include file (d2xx.h) is located
 # If the files are in /usr/include/d2xx then set
-# D2XX = /usr/include/d2xx
+D2XX = /usr/include/
 # If for some reason you need a library to compile d2xx support
 # specify it on the D2XX_LIBS = line.
 # You *must* define D2XX_INCLUDE to use this feature.
@@ -67,9 +67,9 @@
 # LIBOAUTH_LIBS    = $${LIBOAUTH_INSTALL}/lib/liboauth.a -lcurl -lcrypto -lz
 # You may override the INCLUDE and LIB files if you like.
 # You *must* define LIBOAUTH_INSTALL to use this feature.
-#LIBOAUTH_INSTALL = 
-#LIBOAUTH_INCLUDE = 
-#LIBOAUTH_LIBS    = 
+LIBOAUTH_INSTALL = /usr 
+LIBOAUTH_INCLUDE =  /usr/include
+LIBOAUTH_LIBS = /usr/lib/liboauth.so -lcurl -lcrypto -lz
 
 # If you want 3D plotting, you need to install qwtplot3d
 #     http://qwtplot3d.sourceforge.net/
@@ -122,9 +122,9 @@
 # ICAL_LIBS    = $${ICAL_INSTALL}/lib/libical.a
 # You may override the INCLUDE and LIB files if you like.
 # You *must* define ICAL_INSTALL to use this feature.
-#ICAL_INSTALL = 
-#ICAL_INCLUDE = 
-#ICAL_LIBS    =
+ICAL_INSTALL = whatever 
+ICAL_INCLUDE =  /usr/include/libical
+ICAL_LIBS    = /usr/lib/libical.so
 
 # If you want support for using USB1 sticks in Train View on Windows
 # then install the SiLabs USBXpress Software Development Kit (SDK)
@@ -167,15 +167,15 @@
 # VLC_LIBS    = -lvlc -lvlccore
 # You may override the INCLUDE and LIB files if you like.
 # You *must* define VLC_INSTALL to use this feature.
-#VLC_INSTALL = 
+VLC_INSTALL = /usr/bin 
 #VLC_INCLUDE = 
 #VLC_LIBS    = 
 
 #If you want search functionality then uncomment the following
 #two lines once you habve installed clucene developer libraries
 #and runtimes. See the INSTALL guide for your platform. 
-#CLUCENE_INCLUDE = /usr/include/CLucene
-#CLUCENE_LIBS    = -lclucene
+CLUCENE_INCLUDE = /usr/include/CLucene
+CLUCENE_LIBS    = -lclucene-core
 
 
 # *** Mac users NOTE ***
