diff -urN sabre-0.2.4b/src/aaaunit.C tmp/sabre-0.2.4b/src/aaaunit.C
--- sabre-0.2.4b/src/aaaunit.C	1998-12-31 20:13:37.000000000 -0500
+++ tmp/sabre-0.2.4b/src/aaaunit.C	2008-01-31 12:16:20.000000000 -0500
@@ -24,8 +24,8 @@
  * Author : Dan Hammer                           *
  *************************************************/
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <string.h>
 #include <math.h>
 #include <limits.h>
@@ -139,7 +139,7 @@
 		lp = target->position;
 }
 
-void AAA_Unit::read(istream &is)
+void AAA_Unit::read(std::istream &is)
 {
 char c;
 
@@ -157,7 +157,7 @@
 	init();
 }
 
-void AAA_Unit::write(ostream &os)
+void AAA_Unit::write(std::ostream &os)
 {
 	os << "{\n";
 	os << view << '\n';
diff -urN sabre-0.2.4b/src/aigunner.h tmp/sabre-0.2.4b/src/aigunner.h
--- sabre-0.2.4b/src/aigunner.h	1998-12-31 03:53:35.000000000 -0500
+++ tmp/sabre-0.2.4b/src/aigunner.h	2008-01-30 19:26:59.000000000 -0500
@@ -31,7 +31,7 @@
 #define aiGUN_YAW_RATE        (_degree * 20.0)
 
 #ifdef __cplusplus
-#include "fstream.h"
+#include "fstream"
 #include "sobject.h"
 #include "sarray.h"
 #include "sfltmdl.h"
diff -urN sabre-0.2.4b/src/clip.C tmp/sabre-0.2.4b/src/clip.C
--- sabre-0.2.4b/src/clip.C	1999-02-12 17:33:40.000000000 -0500
+++ tmp/sabre-0.2.4b/src/clip.C	2008-01-30 19:12:44.000000000 -0500
@@ -26,7 +26,7 @@
  * clipping algorythm.                           *
  *************************************************/
 #include <stdio.h>
-#include <iostream.h>
+#include <iostream>
 #include <math.h>
 #include "defs.h"
 #include "grafix.h"
diff -urN sabre-0.2.4b/src/clip.h tmp/sabre-0.2.4b/src/clip.h
--- sabre-0.2.4b/src/clip.h	1999-02-12 17:33:40.000000000 -0500
+++ tmp/sabre-0.2.4b/src/clip.h	2008-01-31 10:44:33.000000000 -0500
@@ -105,14 +105,14 @@
   void set(const bounding_cube &bc);
   R_3DPoint centerPoint();
 
-  void write(ostream &os)
+  void write(std::ostream &os)
     {
       os << "[ " << min_x << " " << max_x << " ]\n"; 
       os << "[ " << min_y << " " << max_y << " ]\n"; 
       os << "[ " << min_z << " " << max_z << " ]\n"; 
     }
 
-  friend ostream &operator <<(ostream &os, bounding_cube &bc)
+  friend std::ostream &operator <<(std::ostream &os, bounding_cube &bc)
     {
       bc.write(os);
       return (os);
diff -urN sabre-0.2.4b/src/cockpit.C tmp/sabre-0.2.4b/src/cockpit.C
--- sabre-0.2.4b/src/cockpit.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/cockpit.C	2008-01-30 19:19:46.000000000 -0500
@@ -31,8 +31,8 @@
 #include <string.h>
 #include <math.h>
 #include <ctype.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include "defs.h"
 #include "grafix.h"
 #include "vga_13.h"
diff -urN sabre-0.2.4b/src/colormap.C tmp/sabre-0.2.4b/src/colormap.C
--- sabre-0.2.4b/src/colormap.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/colormap.C	2008-01-31 11:45:11.000000000 -0500
@@ -27,13 +27,13 @@
 #include <stdio.h>
 #include <values.h>
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 #include <string.h>
 #include "simerr.h"
 #include "simfile.h"
 #include "colormap.h"
 
-void ColorMap::read(istream &is)
+void ColorMap::read(std::istream &is)
 {
   char c;
   READ_TOKI('(',is,c)
@@ -42,7 +42,7 @@
   READ_TOK(')',is,c)
 }
 
-void ColorMap::write(ostream &os)
+void ColorMap::write(std::ostream &os)
 {
   os << "( " << id_color << ' ' << base_color << ' ' << n << " )";
 }
diff -urN sabre-0.2.4b/src/colormap.h tmp/sabre-0.2.4b/src/colormap.h
--- sabre-0.2.4b/src/colormap.h	1998-12-13 10:32:45.000000000 -0500
+++ tmp/sabre-0.2.4b/src/colormap.h	2008-01-31 10:38:05.000000000 -0500
@@ -69,16 +69,16 @@
       n = cm.n;
     }
 
-  void read(istream &is);
-  void write(ostream &os);
+  void read(std::istream &is);
+  void write(std::ostream &os);
   
-  friend istream &operator >>(istream &is, ColorMap &cm)
+  friend std::istream &operator >>(std::istream &is, ColorMap &cm)
     {
       cm.read(is);
       return(is);
     }
 
-  friend ostream &operator <<(ostream &os, ColorMap &cm)
+  friend std::ostream &operator <<(std::ostream &os, ColorMap &cm)
     {
       cm.write(os);
       return(os);
diff -urN sabre-0.2.4b/src/colorspc.C tmp/sabre-0.2.4b/src/colorspc.C
--- sabre-0.2.4b/src/colorspc.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/colorspc.C	2008-01-31 12:15:36.000000000 -0500
@@ -18,11 +18,11 @@
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include "colorspc.h"
 
-istream &operator >>(istream &is, color_spec &cs)
+std::istream &operator >>(std::istream &is, color_spec &cs)
 {
   char c;
   is >> c;
diff -urN sabre-0.2.4b/src/colorspc.h tmp/sabre-0.2.4b/src/colorspc.h
--- sabre-0.2.4b/src/colorspc.h	1998-12-13 10:32:45.000000000 -0500
+++ tmp/sabre-0.2.4b/src/colorspc.h	2008-01-31 10:36:21.000000000 -0500
@@ -47,8 +47,8 @@
       red = r; green = g; blue = b;
     }
 
-  friend istream &operator >>(istream &is, color_spec &cs);
-  friend ostream &operator <<(ostream &os, color_spec &cs)
+  friend std::istream &operator >>(std::istream &is, color_spec &cs);
+  friend std::ostream &operator <<(std::ostream &os, color_spec &cs)
     {
       os << '(' << ' ' << cs.color << ' ' << cs.red << ' ';
       os << cs.green << ' ' << cs.blue << " )";
diff -urN sabre-0.2.4b/src/copoly.C tmp/sabre-0.2.4b/src/copoly.C
--- sabre-0.2.4b/src/copoly.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/copoly.C	2008-01-31 11:37:09.000000000 -0500
@@ -27,7 +27,7 @@
 #include <stdio.h>
 #include <values.h>
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 #include "simerr.h"
 #include "vga_13.h"
 #include "pen.h"
@@ -185,7 +185,7 @@
     }
 }
 
-void C_Oriented_Poly::write(ostream &os)
+void C_Oriented_Poly::write(std::ostream &os)
 {
   int tflag = 0;
   char cc = '0';
@@ -218,13 +218,13 @@
   os << ']';
 }
 
-ostream & operator <<(ostream &os, C_Oriented_Poly &cp)
+std::ostream & operator <<(std::ostream &os, C_Oriented_Poly &cp)
 {
   cp.write(os);
   return os;
 }
 
-ostream &operator <<(ostream &os, C_Oriented_Shape &cs)
+std::ostream &operator <<(std::ostream &os, C_Oriented_Shape &cs)
 {
   os << '{' << '\n';
   os << cs.npolys << '\n';
@@ -277,7 +277,7 @@
     delete [] o_polys;
 }
 
-istream &operator >>(istream &is, C_Oriented_Shape &cs)
+std::istream &operator >>(std::istream &is, C_Oriented_Shape &cs)
 {
   for (int i=0;i<cs.npolys;i++)
     is >> cs.o_polys[i];
diff -urN sabre-0.2.4b/src/copoly.h tmp/sabre-0.2.4b/src/copoly.h
--- sabre-0.2.4b/src/copoly.h	1998-12-13 10:32:46.000000000 -0500
+++ tmp/sabre-0.2.4b/src/copoly.h	2008-01-31 10:58:08.000000000 -0500
@@ -71,8 +71,8 @@
 		     int shadow_color,
 		     Port_3D &ref_port);
   void draw(Port_3D &theport, Port_3D &ref_port, int = 1);
-  friend ostream & operator <<(ostream &, C_Oriented_Poly &);
-  void write(ostream &os);
+  friend std::ostream & operator <<(std::ostream &, C_Oriented_Poly &);
+  void write(std::ostream &os);
   int getScreenPoints(Port_3D &, Port_3D &, R_2DPoint *spoints);
   int getScreenPoints(Port_3D &, Port_3D &, TR_2DPoint *spoints);
 
@@ -101,9 +101,9 @@
   void set_poly_icolor(Vector &light);
   void set_poly_color(int n, int color);
   int drawn_flag;
-  friend istream &operator >>(istream &, C_Oriented_Shape &);
+  friend std::istream &operator >>(std::istream &, C_Oriented_Shape &);
   void set_params(shape_params *);
-  friend ostream &operator <<(ostream &, C_Oriented_Shape &);
+  friend std::ostream &operator <<(std::ostream &, C_Oriented_Shape &);
   void create_points(Port_3D &ref_port);
   void delete_points();
 };
diff -urN sabre-0.2.4b/src/cpoly.C tmp/sabre-0.2.4b/src/cpoly.C
--- sabre-0.2.4b/src/cpoly.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/cpoly.C	2008-01-31 11:34:06.000000000 -0500
@@ -29,7 +29,7 @@
 #include <stdio.h>
 #include <values.h>
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 #include <string.h>
 #include "vmath.h"
 #include "rendpoly.h"
@@ -56,7 +56,7 @@
   file_context = fcontext_buff;
 }
 
-void C_PolyInfo::read(istream &is)
+void C_PolyInfo::read(std::istream &is)
 {
   char c;
   int txtflag;
@@ -104,7 +104,7 @@
 }
 
 
-void C_PolyInfo::write(ostream &os)
+void C_PolyInfo::write(std::ostream &os)
 {
   int txtflg = (tpoints != NULL);
   os << '[' << npoints << " " << txtflg << " ";
@@ -181,7 +181,7 @@
 
 }
 
-void C_ShapeInfo::read(istream &is)
+void C_ShapeInfo::read(std::istream &is)
 {
   char c;
   set_context();
@@ -202,7 +202,7 @@
   READ_TOKX('}','{',is,c)
 }
 
-void C_ShapeInfo::write(ostream &os)
+void C_ShapeInfo::write(std::ostream &os)
 {
   os << "{\n";
   os << npolys <<  "\n";
@@ -268,7 +268,7 @@
   return (*this);
 }
 
-void poly_params::read(istream &is)
+void poly_params::read(std::istream &is)
 {
   char c;
   READ_TOKI('(',is,c)
@@ -277,12 +277,12 @@
   READ_TOK(')',is,c)
 }
 
-void poly_params::write(ostream &os)
+void poly_params::write(std::ostream &os)
 {
   os << '(' << color << " " << tmap << " " << flags << " " << color_range << ')';
 }
 
-void shape_params::read(istream &is)
+void shape_params::read(std::istream &is)
 {
   char c;
   if (p_params)
@@ -308,7 +308,7 @@
     READ_TOK(']',is,c)
 }
 
-void shape_params::write(ostream &os)
+void shape_params::write(std::ostream &os)
 {
   if (p_params)
     {
@@ -365,7 +365,7 @@
 
 void C_3DObjectInfo::readFile(char *path)
 {
-  ifstream is;
+  std::ifstream is;
   if (this->path)
     delete this->path;
   this->path = strdup(path);
@@ -375,7 +375,7 @@
 
 void C_3DObjectInfo::writeFile(char *path)
 {
-  ofstream os;
+  std::ofstream os;
   MYCHECK(path != NULL);
   if (this->path == NULL)
     this->path = strdup(path);
@@ -388,7 +388,7 @@
     write(os);
 }
 
-void C_3DObjectInfo::read(istream &is)
+void C_3DObjectInfo::read(std::istream &is)
 {
   char c;
   if (shapes)
@@ -398,7 +398,7 @@
   READ_TOKI('{',is,c)
     is >> id;
   is >> nshapes;
-  // cout << id << ": " << nshapes << " shapes\n";
+  // std::cout << id << ": " << nshapes << " shapes\n";
   shapes = new C_ShapeInfo[nshapes];
   MYCHECK(shapes != NULL);
   dflt_params = new shape_params[nshapes];
@@ -416,7 +416,7 @@
     
     }
 
-void C_3DObjectInfo::write(ostream &os)
+void C_3DObjectInfo::write(std::ostream &os)
 {
   if (shapes)
     {
@@ -472,7 +472,7 @@
   base_color  = cp.base_color;
 }
 
-istream & operator >>(istream &is, C_Poly &cp)
+std::istream & operator >>(std::istream &is, C_Poly &cp)
 {
   char c;
   is >> c;
@@ -750,7 +750,7 @@
     delete [] polys;
 }
 
-istream & operator >>(istream &is, C_Shape &cs)
+std::istream & operator >>(std::istream &is, C_Shape &cs)
 {
   is >> cs.flags;
   if (cs.flags & CPY_PARAMS)
diff -urN sabre-0.2.4b/src/cpoly.h tmp/sabre-0.2.4b/src/cpoly.h
--- sabre-0.2.4b/src/cpoly.h	1998-12-13 10:32:46.000000000 -0500
+++ tmp/sabre-0.2.4b/src/cpoly.h	2008-01-31 10:51:46.000000000 -0500
@@ -139,10 +139,10 @@
 
   friend inline int operator ==(const poly_params & p1, const poly_params & p2);
 
-  void read(istream &);
-  void write(ostream &);
-  friend inline istream &operator >>(istream & is, poly_params &ps);
-  friend inline ostream &operator <<(ostream & os, poly_params &ps);
+  void read(std::istream &);
+  void write(std::ostream &);
+  friend inline std::istream &operator >>(std::istream & is, poly_params &ps);
+  friend inline std::ostream &operator <<(std::ostream & os, poly_params &ps);
 } ;
 
 
@@ -153,13 +153,13 @@
 	  && (p1.tmap == p2.tmap));
 }
 
-inline istream &operator >>(istream &is, poly_params &ps)
+inline std::istream &operator >>(std::istream &is, poly_params &ps)
 {
   ps.read(is);
   return(is);
 }
 
-inline ostream &operator <<(ostream &os, poly_params &ps)
+inline std::ostream &operator <<(std::ostream &os, poly_params &ps)
 {
   ps.write(os);
   return(os);
@@ -206,19 +206,19 @@
 
   void add(const poly_params &);
 
-  void read(istream &);
-  void write(ostream &);
-  friend inline istream & operator >>(istream &, shape_params &);
-  friend inline ostream & operator <<(ostream &, shape_params &);
+  void read(std::istream &);
+  void write(std::ostream &);
+  friend inline std::istream & operator >>(std::istream &, shape_params &);
+  friend inline std::ostream & operator <<(std::ostream &, shape_params &);
 } ;
 
-inline istream & operator >>(istream &is, shape_params &sp)
+inline std::istream & operator >>(std::istream &is, shape_params &sp)
 {
   sp.read(is);
   return(is);
 }
 
-inline ostream & operator <<(ostream &os, shape_params &sp)
+inline std::ostream & operator <<(std::ostream &os, shape_params &sp)
 {
   sp.write(os);
   return(os);
@@ -268,21 +268,21 @@
 
   void copy(const C_PolyInfo &);
 
-  void read(istream &);
-  void write(ostream &);
+  void read(std::istream &);
+  void write(std::ostream &);
 
-  friend inline istream &operator >>(istream &is, C_PolyInfo &cp);
-  friend inline ostream &operator <<(ostream &os, C_PolyInfo &cp);
+  friend inline std::istream &operator >>(std::istream &is, C_PolyInfo &cp);
+  friend inline std::ostream &operator <<(std::ostream &os, C_PolyInfo &cp);
 
 };
 
-inline istream &operator >>(istream &is, C_PolyInfo &cp)
+inline std::istream &operator >>(std::istream &is, C_PolyInfo &cp)
 {
   cp.read(is);
   return(is);
 }
 
-inline ostream &operator <<(ostream &os, C_PolyInfo &cp)
+inline std::ostream &operator <<(std::ostream &os, C_PolyInfo &cp)
 {
   cp.write(os);
   return(os);
@@ -329,20 +329,20 @@
   void copy(const C_ShapeInfo &);
   void add(const C_PolyInfo &ip);
 
-  void read(istream &);
-  void write(ostream &);
+  void read(std::istream &);
+  void write(std::ostream &);
 
-  friend inline istream &operator >>(istream &, C_ShapeInfo &);
-  friend inline ostream &operator <<(ostream &, C_ShapeInfo &);
+  friend inline std::istream &operator >>(std::istream &, C_ShapeInfo &);
+  friend inline std::ostream &operator <<(std::ostream &, C_ShapeInfo &);
 };
 
-inline istream &operator >>(istream &is, C_ShapeInfo &cp)
+inline std::istream &operator >>(std::istream &is, C_ShapeInfo &cp)
 {
   cp.read(is);
   return(is);
 }
 
-inline ostream &operator <<(ostream &os, C_ShapeInfo &cp)
+inline std::ostream &operator <<(std::ostream &os, C_ShapeInfo &cp)
 {
   cp.write(os);
   return(os);
@@ -379,19 +379,19 @@
 
   void readFile(char *path);
   void writeFile(char *path);
-  void read(istream &);
-  void write(ostream &);
-  friend inline istream &operator >>(istream &is, C_3DObjectInfo &co);
-  friend inline ostream &operator <<(ostream &os, C_3DObjectInfo &co);
+  void read(std::istream &);
+  void write(std::ostream &);
+  friend inline std::istream &operator >>(std::istream &is, C_3DObjectInfo &co);
+  friend inline std::ostream &operator <<(std::ostream &os, C_3DObjectInfo &co);
 };
 
-inline istream &operator >>(istream &is, C_3DObjectInfo &co)
+inline std::istream &operator >>(std::istream &is, C_3DObjectInfo &co)
 {
   co.read(is);
   return(is);
 }
 
-inline ostream &operator <<(ostream &os, C_3DObjectInfo &co)
+inline std::ostream &operator <<(std::ostream &os, C_3DObjectInfo &co)
 {
   co.write(os);
   return(os);
@@ -428,7 +428,7 @@
   void set_shadow(REAL_TYPE z_value, Vector &light_source);
   void draw_shadow(Port_3D &theport, int z_clip, int shadow_color);
   virtual void set_poly_icolor(Vector &);
-  friend istream &operator >>(istream &, C_Poly &);
+  friend std::istream &operator >>(std::istream &, C_Poly &);
   void set_params(poly_params *prms);
   void set_params(const C_Poly &);
   virtual void set_scale(REAL_TYPE x)
@@ -461,7 +461,7 @@
   virtual void set_shadow(REAL_TYPE , Vector &);
   virtual void draw(Port_3D &, R_3DPoint &);
   void set_poly_icolor(Vector &);
-  friend istream & operator >>(istream &, C_Shape &);
+  friend std::istream & operator >>(std::istream &, C_Shape &);
   virtual void set_params(shape_params *);
   virtual void set_scale(REAL_TYPE);
   virtual void getBounds(bounding_cube &);
diff -urN sabre-0.2.4b/src/dvector.C tmp/sabre-0.2.4b/src/dvector.C
--- sabre-0.2.4b/src/dvector.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/dvector.C	2008-01-31 11:46:10.000000000 -0500
@@ -22,14 +22,14 @@
  * floating-point equivalents of vectors                        *
  ****************************************************************/
 #include <stdio.h>
-#include <iostream.h>
+#include <iostream>
 #include <math.h>
 #ifdef USING_FIXED
 #include "fixpoint.h"
 #endif
 #include "vmath.h"
 #include "dvmath.h"
-istream &operator >> (istream &is, DVector &v)
+std::istream &operator >> (std::istream &is, DVector &v)
 {
 	char c;
 	is >> c;
diff -urN sabre-0.2.4b/src/dvmath.h tmp/sabre-0.2.4b/src/dvmath.h
--- sabre-0.2.4b/src/dvmath.h	1998-12-13 10:32:46.000000000 -0500
+++ tmp/sabre-0.2.4b/src/dvmath.h	2008-01-31 11:00:16.000000000 -0500
@@ -60,9 +60,9 @@
   friend int operator ==(const DVector &v1, const DVector &v2)
   { return (v1.X==v2.X && v1.Y==v2.Y && v1.Z==v2.Z); }
 
-  friend ostream &operator << (ostream &os,DVector &v)
+  friend std::ostream &operator << (std::ostream &os,DVector &v)
 	 { return os  << '<' << v.X << ' ' << v.Y << ' ' << v.Z << '>' ; }
-  friend istream &operator >> (istream &is,DVector &v);
+  friend std::istream &operator >> (std::istream &is,DVector &v);
 
   void SetZero( void )
   {
diff -urN sabre-0.2.4b/src/earth.C tmp/sabre-0.2.4b/src/earth.C
--- sabre-0.2.4b/src/earth.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/earth.C	2008-01-31 11:43:56.000000000 -0500
@@ -26,8 +26,8 @@
  * background rendering of earth & sky           *
  *************************************************/
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
@@ -101,7 +101,7 @@
   return (result);
 }
 
-void Earth_Watch::read(istream &is)
+void Earth_Watch::read(std::istream &is)
 {
   char c;
   int i,j,idx;
@@ -183,7 +183,7 @@
   READ_TOKI('}',is,c)
 }
 
-istream &operator >>(istream &is, Earth_Watch &ew)
+std::istream &operator >>(std::istream &is, Earth_Watch &ew)
 {
   ew.read(is);
   return is;
@@ -488,13 +488,13 @@
   t_render_ground(port);
 }
 
-void Earth_Watch2::read(istream &is)
+void Earth_Watch2::read(std::istream &is)
 {
   char c;
   int i,j;
   int rsrv;
   char terrain_path[100];
-  ifstream ts;
+  std::ifstream ts;
   float dab_min;
 
   R_3DPoint loc;
diff -urN sabre-0.2.4b/src/earth.h tmp/sabre-0.2.4b/src/earth.h
--- sabre-0.2.4b/src/earth.h	1998-12-13 10:32:46.000000000 -0500
+++ tmp/sabre-0.2.4b/src/earth.h	2008-01-31 11:12:44.000000000 -0500
@@ -75,8 +75,8 @@
   void draw_horizon_grade(Port_3D &);
   void render_ground(Port_3D &);
   void t_render_ground(Port_3D &);
-  friend istream &operator >>(istream &is, Earth_Watch &);
-  virtual void read(istream &is);
+  friend std::istream &operator >>(std::istream &is, Earth_Watch &);
+  virtual void read(std::istream &is);
 };
 
 class Earth_Watch2 : public Earth_Watch
@@ -84,7 +84,7 @@
 public:
   Terrain_Shape *tshapes;
   int            ntshapes;
-  virtual void read(istream &is);
+  virtual void read(std::istream &is);
   REAL_TYPE getGroundLevel(R_3DPoint &p);
 };
 
diff -urN sabre-0.2.4b/src/flight.C tmp/sabre-0.2.4b/src/flight.C
--- sabre-0.2.4b/src/flight.C	1999-11-21 08:01:36.000000000 -0500
+++ tmp/sabre-0.2.4b/src/flight.C	2008-01-31 11:47:02.000000000 -0500
@@ -55,8 +55,8 @@
  * ain't no function of velocity
  */
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
@@ -82,7 +82,7 @@
 REAL_TYPE Flight::rot_rate_max = _PI34;
 REAL_TYPE Flight::rot_max = _PI2 * 0.4;
 
-void Flight::read(istream &is)
+void Flight::read(std::istream &is)
 {
   char c;
   READ_TOKI('{',is,c);
@@ -94,7 +94,7 @@
   state.weight = specs->weight;
 }
 
-void Flight::write(ostream &os)
+void Flight::write(std::ostream &os)
 {
   os << "{\n " << controls << " " << state << "\n}\n";
 }
diff -urN sabre-0.2.4b/src/flight.h tmp/sabre-0.2.4b/src/flight.h
--- sabre-0.2.4b/src/flight.h	1999-11-21 08:02:21.000000000 -0500
+++ tmp/sabre-0.2.4b/src/flight.h	2008-01-31 11:02:53.000000000 -0500
@@ -52,14 +52,14 @@
   void calcRotations(float);
   void applyForces(float);
   void applyRotations(float);
-  void read(istream &);
-  void write(ostream &);
-  friend istream & operator >>(istream & is, Flight &f)
+  void read(std::istream &);
+  void write(std::ostream &);
+  friend std::istream & operator >>(std::istream & is, Flight &f)
     {
       f.read(is);
       return(is);
     }
-  friend ostream & operator <<(ostream & os, Flight &f)
+  friend std::ostream & operator <<(std::ostream & os, Flight &f)
     {
       f.write(os);
       return(os);
diff -urN sabre-0.2.4b/src/fltlite.C tmp/sabre-0.2.4b/src/fltlite.C
--- sabre-0.2.4b/src/fltlite.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/fltlite.C	2008-01-31 11:51:23.000000000 -0500
@@ -21,8 +21,8 @@
  * flight.cpp                                          *
  *******************************************************/
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
@@ -57,7 +57,7 @@
   return (DVector(r_w));
 }
 
-void FlightLight_Specs::read(istream &is)
+void FlightLight_Specs::read(std::istream &is)
 {
   char c = ' ';
   READ_TOK('{',is,c);
@@ -68,7 +68,7 @@
   mass = weight / g;
 }
 
-void FlightLight_Specs::write(ostream &os)
+void FlightLight_Specs::write(std::ostream &os)
 {
   os << "{\n";
   os << weight << thrust;
@@ -77,13 +77,13 @@
   os << "}\n";
 }
 
-istream &operator >>(istream &is, FlightLight_Specs &fs)
+std::istream &operator >>(std::istream &is, FlightLight_Specs &fs)
 {
   fs.read(is);
   return (is);
 }
 
-ostream &operator <<(ostream &os, FlightLight_Specs &fs)
+std::ostream &operator <<(std::ostream &os, FlightLight_Specs &fs)
 {
   fs.write(os);
   return (os);
diff -urN sabre-0.2.4b/src/fltlite.h tmp/sabre-0.2.4b/src/fltlite.h
--- sabre-0.2.4b/src/fltlite.h	1998-12-13 10:32:46.000000000 -0500
+++ tmp/sabre-0.2.4b/src/fltlite.h	2008-01-31 11:04:03.000000000 -0500
@@ -96,10 +96,10 @@
     return *this;
   }
   
-  void read(istream &);
-  void write(ostream &);
-  friend istream &operator >>(istream &, FlightLight_Specs &fs);
-  friend ostream &operator <<(ostream &, FlightLight_Specs &fs);
+  void read(std::istream &);
+  void write(std::ostream &);
+  friend std::istream &operator >>(std::istream &, FlightLight_Specs &fs);
+  friend std::ostream &operator <<(std::ostream &, FlightLight_Specs &fs);
 };
 
 
diff -urN sabre-0.2.4b/src/fltmngr.C tmp/sabre-0.2.4b/src/fltmngr.C
--- sabre-0.2.4b/src/fltmngr.C	1999-04-19 16:17:31.000000000 -0400
+++ tmp/sabre-0.2.4b/src/fltmngr.C	2008-01-31 11:57:26.000000000 -0500
@@ -24,8 +24,8 @@
  * Author : Dan Hammer                           *
  *************************************************/
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
@@ -58,7 +58,7 @@
 #define F2FIX(a) (a)
 #define FIX2F(a) (a)
 
-void Flight_Node::read(istream &is)
+void Flight_Node::read(std::istream &is)
 {
   char buff[64];
   int n;
@@ -157,7 +157,7 @@
   viewer->affiliation = pilot->getAffiliation();
 }
 
-istream &operator >>(istream &is, Flight_Node &fn)
+std::istream &operator >>(std::istream &is, Flight_Node &fn)
 {
   fn.read(is);
   return is;
@@ -413,7 +413,7 @@
 int Flight_Manager::read_file(char *path)
 {
   int result = 0;
-  ifstream infile;
+  std::ifstream infile;
 
   if (open_is(infile,path))
     {
@@ -424,11 +424,11 @@
   return result;
 }
 
-void Flight_Manager::read(istream &is)
+void Flight_Manager::read(std::istream &is)
 {
   char buff[BUFSIZ];
   int got_line;
-  ifstream is2;
+  std::ifstream is2;
   int i;
 
   is.seekg(0L);
@@ -530,13 +530,13 @@
     }
 }
 
-istream & operator >>(istream &is, Flight_Manager &fm)
+std::istream & operator >>(std::istream &is, Flight_Manager &fm)
 {
   fm.read(is);
   return is;
 }
 
-void Flight_Manager::readNodes(istream &is)
+void Flight_Manager::readNodes(std::istream &is)
 {
   int i;
 
@@ -561,14 +561,14 @@
 
 void Flight_Manager::readNodeFile(char *path)
 {
-  ifstream is;
+  std::ifstream is;
 
   if (open_is(is,path))
     readNodes(is);
 }
 
 
-void Flight_Manager::readFlites(istream &is)
+void Flight_Manager::readFlites(std::istream &is)
 {
 #define MAX_SWAYPOINT_INFOS 16
   swaypoint_info waypoints[MAX_SWAYPOINT_INFOS];
@@ -722,7 +722,7 @@
 
 void Flight_Manager::readFliteFile(char *path)
 {
-  ifstream is;
+  std::ifstream is;
 
   if (open_is(is,path))
     readFlites(is);
diff -urN sabre-0.2.4b/src/fltmngr.h tmp/sabre-0.2.4b/src/fltmngr.h
--- sabre-0.2.4b/src/fltmngr.h	1999-02-12 17:33:40.000000000 -0500
+++ tmp/sabre-0.2.4b/src/fltmngr.h	2008-01-31 11:08:46.000000000 -0500
@@ -77,8 +77,8 @@
 			delete [] wi;
 	}
 
-	friend istream &operator >>(istream &, Flight_Node &);
-	void read(istream &);
+	friend std::istream &operator >>(std::istream &, Flight_Node &);
+	void read(std::istream &);
 
 	void start()
 	{
@@ -134,8 +134,8 @@
 
 	~Flight_Manager();
 	int read_file(char *);
-	friend istream &operator >>(istream &, Flight_Manager &);
-	void read(istream &);
+	friend std::istream &operator >>(std::istream &, Flight_Manager &);
+	void read(std::istream &);
 	void update(Unguided_Manager *um = NULL);
 	void start();
 	void pause();
@@ -180,9 +180,9 @@
 			return NULL;
 	}
 	void readNodeFile(char *);
-	void readNodes(istream &is);
+	void readNodes(std::istream &is);
 	void readFliteFile(char *);
-	void readFlites(istream &is);
+	void readFlites(std::istream &is);
 };
 
 #endif
diff -urN sabre-0.2.4b/src/fltobj.C tmp/sabre-0.2.4b/src/fltobj.C
--- sabre-0.2.4b/src/fltobj.C	1999-11-21 08:01:13.000000000 -0500
+++ tmp/sabre-0.2.4b/src/fltobj.C	2008-01-31 12:20:16.000000000 -0500
@@ -25,8 +25,8 @@
  *************************************************/
 #include <stdio.h>
 #include <string.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
@@ -130,7 +130,7 @@
 int Flight_Specs::read_file(char *path)
 {
 int result = 0;
-ifstream infile;
+std::ifstream infile;
 
 	if (open_is(infile,path))
 	{
@@ -145,7 +145,7 @@
 {
 int result = 0;
 
-	ofstream outfile;
+	std::ofstream outfile;
 	if (open_libos(outfile,path))
 	{
 		result = 1;
@@ -155,7 +155,7 @@
 	return (result);
 }
 
-void Flight_Specs::read(istream &is)
+void Flight_Specs::read(std::istream &is)
 {
 char c;
 
@@ -204,7 +204,7 @@
 	READ_TOK('}',is,c);
 }
 
-void Flight_Specs::write(ostream &os)
+void Flight_Specs::write(std::ostream &os)
 {
 	os << "{\n";
 	os << weight << max_thrust << fps2kts(max_speed) << fps2kts(corner_speed);
@@ -228,7 +228,7 @@
 	os << "}\n";
 }
 
-void Flight_Controls::read(istream &is)
+void Flight_Controls::read(std::istream &is)
 {
 	is >> throttle >> landing_gear >> wheel_brakes;
 	is >> cockpit >> hud_on >> vextern;
@@ -241,14 +241,14 @@
 	radar = 1;
 }
 
-void Flight_Controls::write(ostream &os)
+void Flight_Controls::write(std::ostream &os)
 {
 	os << elevators << ' ' << ailerons << ' '  << throttle << '\n';
 	os << rudder << ' ' << speed_brakes << '\n' ;
 	os << flaps << ' ' << wheel_brakes << ' ' << landing_gear << '\n';
 }
 
-istream &operator >>(istream &is, Vector_Q &vq)
+std::istream &operator >>(std::istream &is, Vector_Q &vq)
 {
 	char c;
 	is >> c;
@@ -298,7 +298,7 @@
 	return Vector_Q(r_mag,v_result);
 }
 
-void Flight_State::read(istream &is)
+void Flight_State::read(std::istream &is)
 {
 float kph;
 
diff -urN sabre-0.2.4b/src/fltobj.h tmp/sabre-0.2.4b/src/fltobj.h
--- sabre-0.2.4b/src/fltobj.h	1999-11-21 08:03:08.000000000 -0500
+++ tmp/sabre-0.2.4b/src/fltobj.h	2008-01-31 11:01:47.000000000 -0500
@@ -167,14 +167,14 @@
 
   Flight_Specs();
 
-  void read(istream &);
-  void write(ostream &);
-  friend istream &operator >>(istream &is, Flight_Specs &fs)
+  void read(std::istream &);
+  void write(std::ostream &);
+  friend std::istream &operator >>(std::istream &is, Flight_Specs &fs)
     {
       fs.read(is);
       return(is);
     }
-  friend ostream &operator <<(ostream &os, Flight_Specs &fs)
+  friend std::ostream &operator <<(std::ostream &os, Flight_Specs &fs)
     {
       fs.write(os);
       return(os);
@@ -286,14 +286,14 @@
 	 auto_coord = 0;
   }
 
-  void read(istream &is);
-  void write(ostream &os);
-  friend istream & operator >>(istream &is, Flight_Controls &fc)
+  void read(std::istream &is);
+  void write(std::ostream &os);
+  friend std::istream & operator >>(std::istream &is, Flight_Controls &fc)
     {
       fc.read(is);
       return(is);
     }
-  friend ostream & operator <<(ostream &os, Flight_Controls &fc)
+  friend std::ostream & operator <<(std::ostream &os, Flight_Controls &fc)
     {
       fc.write(os);
       return(os);
@@ -331,13 +331,13 @@
   friend Vector_Q operator +(Vector_Q &, Vector_Q &);
   friend DVector operator +(DVector &, Vector_Q &);
 
-  friend ostream &operator <<(ostream &os, Vector_Q &vq)
+  friend std::ostream &operator <<(std::ostream &os, Vector_Q &vq)
   {
     os << '(' << vq.magnitude << ' ' << vq.direction << ')';
     return os;
   }
 
-  friend istream &operator >>(istream &is, Vector_Q &vq);
+  friend std::istream &operator >>(std::istream &is, Vector_Q &vq);
 };
 
 /***********************************************************
@@ -445,17 +445,17 @@
     fuel = 0.0;
     elapsed_time = 0.0;
   }
-  void read(istream &is);
-  void write(ostream &os)
+  void read(std::istream &is);
+  void write(std::ostream &os)
     {
       os << velocity.magnitude << flight_port;
     }
-  friend istream &operator >>(istream &is, Flight_State &fp)
+  friend std::istream &operator >>(std::istream &is, Flight_State &fp)
     {
       fp.read(is);
       return(is);
     }
-  friend ostream &operator <<(ostream &os, Flight_State &fp)
+  friend std::ostream &operator <<(std::ostream &os, Flight_State &fp)
     {
       fp.write(os);
       return(os);
diff -urN sabre-0.2.4b/src/fltzview.C tmp/sabre-0.2.4b/src/fltzview.C
--- sabre-0.2.4b/src/fltzview.C	1998-12-31 03:21:23.000000000 -0500
+++ tmp/sabre-0.2.4b/src/fltzview.C	2008-01-30 19:10:24.000000000 -0500
@@ -26,8 +26,8 @@
  *************************************************/
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
diff -urN sabre-0.2.4b/src/game.C tmp/sabre-0.2.4b/src/game.C
--- sabre-0.2.4b/src/game.C	1999-11-24 18:17:36.000000000 -0500
+++ tmp/sabre-0.2.4b/src/game.C	2008-01-31 11:24:16.000000000 -0500
@@ -32,8 +32,8 @@
 #include <math.h>
 #include <signal.h>
 #include <ctype.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <setjmp.h>
 #include <stdarg.h>
 #include "sim.h"
@@ -114,12 +114,12 @@
 		sound_off(SNDGLOBAL_INTRO);
 	}
 	sound_off_all();
-	printResults(cout);
+	printResults(std::cout);
 }
 
 void GameSession::readWorldFile()
 {
-ifstream is;
+std::ifstream is;
 char c;
 Port_3D port;
 float pixl_ratio;
@@ -669,9 +669,9 @@
 }
 
 #ifndef SABREWIN
-inline void PRINTLN(ostream &os, char *s, ...)
+inline void PRINTLN(std::ostream &os, char *s, ...)
 #else
-void __cdecl PRINTLN(ostream &os, char *s, ...)
+void __cdecl PRINTLN(std::ostream &os, char *s, ...)
 #endif
 {
   va_list ap;
@@ -688,7 +688,7 @@
 }
 
 
-void GameSession::printResults(ostream &os)
+void GameSession::printResults(std::ostream &os)
 {
   int i;
   PRINTLN(os, "\nMission Debrief\n");
@@ -702,7 +702,7 @@
 }
 
 
-void GameSession::printResult(Flight_Node &fn, ostream &os)
+void GameSession::printResult(Flight_Node &fn, std::ostream &os)
 {
 
   Flight *flight = fn.flight;
diff -urN sabre-0.2.4b/src/game.h tmp/sabre-0.2.4b/src/game.h
--- sabre-0.2.4b/src/game.h	1998-12-31 03:21:24.000000000 -0500
+++ tmp/sabre-0.2.4b/src/game.h	2008-01-31 11:13:00.000000000 -0500
@@ -73,8 +73,8 @@
 	void update_flight();
 	void draw_flight(Port_3D *pport = NULL);
 	void doPaletteEffect();
-	void printResults(ostream &);
-	void printResult(Flight_Node &, ostream &);
+	void printResults(std::ostream &);
+	void printResult(Flight_Node &, std::ostream &);
 	void __cdecl show_message(int bot, char *str, ... );
 	Flight_Controls *getFlightControls()
 	{
diff -urN sabre-0.2.4b/src/grndunit.C tmp/sabre-0.2.4b/src/grndunit.C
--- sabre-0.2.4b/src/grndunit.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/grndunit.C	2008-01-31 11:58:53.000000000 -0500
@@ -25,8 +25,8 @@
  * Author : Dan Hammer                           *
  *************************************************/
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <string.h>
 #include <math.h>
 #include <limits.h>
@@ -65,7 +65,7 @@
     */
 }
 
-void Ground_Unit_Specs::read(istream &is)
+void Ground_Unit_Specs::read(std::istream &is)
 {
   char c;
   char specs_id[32];
@@ -230,7 +230,7 @@
   return (&position);
 }
 
-void Ground_Unit::read(istream &is)
+void Ground_Unit::read(std::istream &is)
 {
   char c;
   file_context = "Ground_Unit";
@@ -247,7 +247,7 @@
   init();
 }
 
-void Ground_Unit::write(ostream &os)
+void Ground_Unit::write(std::ostream &os)
 {
   os << "{\n";
   os << affiliation << view << '\n';
@@ -255,13 +255,13 @@
   os << "}\n";
 }
 
-istream &operator >>(istream &is, Ground_Unit &gu)
+std::istream &operator >>(std::istream &is, Ground_Unit &gu)
 {
   gu.read(is);
   return(is);
 }
 
-ostream &operator <<(ostream &os, Ground_Unit &gu)
+std::ostream &operator <<(std::ostream &os, Ground_Unit &gu)
 {
   gu.write(os);
   return(os);
@@ -306,7 +306,7 @@
       units[i]->add_draw_list(dl,port);
 }
 
-istream &operator >>(istream &is, Ground_Unit_Manager &gum)
+std::istream &operator >>(std::istream &is, Ground_Unit_Manager &gum)
 {
   gum.read(is);
   return(is);
@@ -314,13 +314,13 @@
 
 void Ground_Unit_Manager::read_file(char *path)
 {
-  ifstream is;
+  std::ifstream is;
   file_context = "Ground_Unit_Manager";
   if (open_is(is,path))
     read(is);
 }
 
-void Ground_Unit_Manager::read(istream &is)
+void Ground_Unit_Manager::read(std::istream &is)
 {
   char c;
   int gu_idx;
@@ -370,7 +370,7 @@
     }  
 }
 
-ostream &operator >>(ostream &os, Ground_Unit_Manager &gum)
+std::ostream &operator >>(std::ostream &os, Ground_Unit_Manager &gum)
 {
   gum.write(os);
   return(os);
@@ -378,12 +378,12 @@
 
 void Ground_Unit_Manager::write_file(char *path)
 {
-  ofstream os;
+  std::ofstream os;
   if (open_libos(os,path))
     write(os);
 }
 
-void Ground_Unit_Manager::write(ostream &)
+void Ground_Unit_Manager::write(std::ostream &)
 {
 
 }
diff -urN sabre-0.2.4b/src/grndunit.h tmp/sabre-0.2.4b/src/grndunit.h
--- sabre-0.2.4b/src/grndunit.h	1998-12-31 04:30:45.000000000 -0500
+++ tmp/sabre-0.2.4b/src/grndunit.h	2008-01-31 11:10:06.000000000 -0500
@@ -57,15 +57,15 @@
 
   ~Ground_Unit_Specs();
 
-  void read(istream &);
-  void write(ostream &);
+  void read(std::istream &);
+  void write(std::ostream &);
 
-  friend istream &operator >>(istream &is, Ground_Unit_Specs &gsp)
+  friend std::istream &operator >>(std::istream &is, Ground_Unit_Specs &gsp)
     {
       gsp.read(is);
       return (is);
     }
-  friend ostream &operator <<(ostream &os, Ground_Unit_Specs &gsp)
+  friend std::ostream &operator <<(std::ostream &os, Ground_Unit_Specs &gsp)
     {
       gsp.write(os);
       return (os);
@@ -122,10 +122,10 @@
 
   void updatePosition();
   void setPosition();
-  virtual void read(istream &);
-  virtual void write(ostream &);
-  friend istream &operator >>(istream &, Ground_Unit &);
-  friend ostream &operator <<(ostream &, Ground_Unit &);
+  virtual void read(std::istream &);
+  virtual void write(std::ostream &);
+  friend std::istream &operator >>(std::istream &, Ground_Unit &);
+  friend std::ostream &operator <<(std::ostream &, Ground_Unit &);
 };
 
 class AAA_Unit : public Ground_Unit
@@ -160,8 +160,8 @@
   void calcLead(R_3DPoint &, R_3DPoint &);
   void youHit(Target *tg);
   void update(Unguided_Manager *, Target_List &);
-  void read(istream &);
-  void write(ostream &);
+  void read(std::istream &);
+  void write(std::ostream &);
 };
 
 class Ground_Unit_Manager
@@ -179,12 +179,12 @@
   void update(Unguided_Manager *, Target_List &);
   void add_draw_list(DrawList &, Port_3D &);
 
-  void read(istream &);
-  friend istream &operator >>(istream &, Ground_Unit_Manager &);
+  void read(std::istream &);
+  friend std::istream &operator >>(std::istream &, Ground_Unit_Manager &);
   void read_file(char *);
 
-  void write(ostream &);
-  friend ostream &operator <<(ostream &, Ground_Unit_Manager &);
+  void write(std::ostream &);
+  friend std::ostream &operator <<(std::ostream &, Ground_Unit_Manager &);
   void write_file(char *);
   void addTargetList(Target_List &);
   Ground_Unit_Specs *getSpecs(char *);
diff -urN sabre-0.2.4b/src/group_3d.C tmp/sabre-0.2.4b/src/group_3d.C
--- sabre-0.2.4b/src/group_3d.C	1998-12-13 10:32:43.000000000 -0500
+++ tmp/sabre-0.2.4b/src/group_3d.C	2008-01-30 19:08:16.000000000 -0500
@@ -27,7 +27,7 @@
 #include <stdio.h>
 #include <values.h>
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 #include "simerr.h"
 #include "grafix.h"
 #include "vmath.h"
diff -urN sabre-0.2.4b/src/hud.C tmp/sabre-0.2.4b/src/hud.C
--- sabre-0.2.4b/src/hud.C	1998-12-31 20:35:11.000000000 -0500
+++ tmp/sabre-0.2.4b/src/hud.C	2008-01-31 12:09:35.000000000 -0500
@@ -28,8 +28,8 @@
 #include <values.h>
 #include <string.h>
 #include <math.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include "defs.h"
 #include "pc_keys.h"
 #include "grafix.h"
@@ -71,12 +71,12 @@
 
 void Hud::read_file(char *path)
 {
-	ifstream is;
+	std::ifstream is;
 	if (open_is(is,path))
 		is >> *this;
 }
 
-istream &operator >>(istream &is, Hud &hd)
+std::istream &operator >>(std::istream &is, Hud &hd)
 {
 	char c = ' ';
 
diff -urN sabre-0.2.4b/src/hud.h tmp/sabre-0.2.4b/src/hud.h
--- sabre-0.2.4b/src/hud.h	1998-12-31 20:13:37.000000000 -0500
+++ tmp/sabre-0.2.4b/src/hud.h	2008-01-31 11:12:21.000000000 -0500
@@ -69,7 +69,7 @@
   void show_gunsight(Pilot &, Port_3D &);
   void do_hud(Flight_Node &);
   void do_hud_stuff(int, int, Flight &);
-  friend istream & operator >>(istream &, Hud &);
+  friend std::istream & operator >>(std::istream &, Hud &);
   void read_file(char *path);
 
   void do_hud_formation_geometry(const sTargetGeometry &, Port_3D &);
diff -urN sabre-0.2.4b/src/input.C tmp/sabre-0.2.4b/src/input.C
--- sabre-0.2.4b/src/input.C	1998-12-31 04:22:18.000000000 -0500
+++ tmp/sabre-0.2.4b/src/input.C	2008-01-30 19:22:41.000000000 -0500
@@ -31,8 +31,8 @@
 #include <math.h>
 #include <string.h>
 #include <time.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include "defs.h"
 #include "vga_13.h"
 #include "input.h"
diff -urN sabre-0.2.4b/src/instrmnt.C tmp/sabre-0.2.4b/src/instrmnt.C
--- sabre-0.2.4b/src/instrmnt.C	1998-12-13 10:32:44.000000000 -0500
+++ tmp/sabre-0.2.4b/src/instrmnt.C	2008-01-31 12:10:40.000000000 -0500
@@ -31,8 +31,8 @@
 #include <string.h>
 #include <math.h>
 #include <ctype.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include "defs.h"
 #include "grafix.h"
 #include "vga_13.h"
@@ -315,7 +315,7 @@
     indicators[indicators_idx++] = ind;
 }
 
-istream &operator >>(istream &is, Instrument_Panel &ip)
+std::istream &operator >>(std::istream &is, Instrument_Panel &ip)
 {
   char c;
   int i;
@@ -393,7 +393,7 @@
   ip.add_indicator(&compass);
   ip.add_indicator(&throttle_i);
 
-  ifstream is;
+  std::ifstream is;
   if (open_is(is,path))
     is >> ip;
   is.close();
diff -urN sabre-0.2.4b/src/instrmnt.h tmp/sabre-0.2.4b/src/instrmnt.h
--- sabre-0.2.4b/src/instrmnt.h	1998-12-13 10:32:46.000000000 -0500
+++ tmp/sabre-0.2.4b/src/instrmnt.h	2008-01-31 11:12:05.000000000 -0500
@@ -36,7 +36,7 @@
   int x,y;
   virtual void underpaint(Flight &) = 0;
   virtual void display(Flight &) = 0;
-  virtual void read_from(istream &) = 0;
+  virtual void read_from(std::istream &) = 0;
   virtual void init() = 0;
 };
 
@@ -46,7 +46,7 @@
   int width,color;
   float scale;
   float range;
-  void read_from(istream &is)
+  void read_from(std::istream &is)
   {
     is >> x >> y >> width >> range;
     is >> scale >> color;
@@ -168,7 +168,7 @@
   ~Instrument_Panel()
   { if (indicators) delete indicators; }
   void add_indicator(Indicator *);
-  friend istream &operator >>(istream &is, Instrument_Panel &);
+  friend std::istream &operator >>(std::istream &is, Instrument_Panel &);
   void underpaint(Flight &);
   void display(Flight &);
   void init();
diff -urN sabre-0.2.4b/src/main.C tmp/sabre-0.2.4b/src/main.C
--- sabre-0.2.4b/src/main.C	2008-01-31 12:27:37.000000000 -0500
+++ tmp/sabre-0.2.4b/src/main.C	2008-01-31 11:22:27.000000000 -0500
@@ -39,8 +39,8 @@
 #include <math.h>
 #include <signal.h>
 #include <ctype.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <setjmp.h>
 #include <stdarg.h>
 #ifdef DEBIAN
@@ -363,7 +363,7 @@
 		      demo,
 		      messageLoop);
   theGame.doGame();
-  theGame.printResults(cout);
+  theGame.printResults(std::cout);
 }
 
 void doSceneEdit()
diff -urN sabre-0.2.4b/src/moveable.C tmp/sabre-0.2.4b/src/moveable.C
--- sabre-0.2.4b/src/moveable.C	1998-12-13 10:32:44.000000000 -0500
+++ tmp/sabre-0.2.4b/src/moveable.C	2008-01-30 19:22:09.000000000 -0500
@@ -22,8 +22,8 @@
  ***********************************************************/
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
diff -urN sabre-0.2.4b/src/obj_3d.C tmp/sabre-0.2.4b/src/obj_3d.C
--- sabre-0.2.4b/src/obj_3d.C	2008-01-31 12:27:37.000000000 -0500
+++ tmp/sabre-0.2.4b/src/obj_3d.C	2008-01-31 11:35:40.000000000 -0500
@@ -29,7 +29,7 @@
 #include <stdio.h>
 #include <values.h>
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 #include <string.h>
 #include "vmath.h"
 #include "rendpoly.h"
@@ -316,7 +316,7 @@
 
 void C_3DInfoManager::readFile(char *path)
 {
-  ifstream is;
+  std::ifstream is;
   if (this->path)
     delete this->path;
   this->path = strdup(path);
@@ -326,7 +326,7 @@
 
 void C_3DInfoManager::writeFile(char *path)
 {
-  ofstream os;
+  std::ofstream os;
   MYCHECK(path != NULL);
   if (this->path == NULL)
     this->path = strdup(path);
@@ -339,7 +339,7 @@
     write(os);
 }
 
-void C_3DInfoManager::read(istream &is)
+void C_3DInfoManager::read(std::istream &is)
 {
   char c;
   char buff[256];
@@ -361,7 +361,7 @@
   READ_TOKI('}',is,c)
 }
 
-void C_3DInfoManager::write(ostream &os)
+void C_3DInfoManager::write(std::ostream &os)
 {
   os << "{\n" << ninfos << "\n";
   for (int i=0;i<ninfos;i++)
@@ -377,7 +377,7 @@
  * C_3DObject2 Methods                                         *
  ***************************************************************/
 
-void C_3DObject2::read(istream &is)
+void C_3DObject2::read(std::istream &is)
 {
   char c;
   int i;
@@ -389,7 +389,7 @@
   ci = im->getInfo(obj_id);
   if (ci == NULL)
     {
-      cout << "C_3DObject2: object id " << obj_id << "not found!\n";
+      std::cout << "C_3DObject2: object id " << obj_id << "not found!\n";
       READ_TOKI('}',is,c);
       return;
     }
@@ -635,7 +635,7 @@
     delete [] groups;
 }
 
-void C_DrawList2::read(istream &is)
+void C_DrawList2::read(std::istream &is)
 {
   int i,j,idx,n;
   char c;
@@ -667,7 +667,7 @@
     }
 }
 
-istream &operator >>(istream &is, C_DrawList2 &cd)
+std::istream &operator >>(std::istream &is, C_DrawList2 &cd)
 {
   cd.read(is);
   return is;
diff -urN sabre-0.2.4b/src/obj_3d.h tmp/sabre-0.2.4b/src/obj_3d.h
--- sabre-0.2.4b/src/obj_3d.h	1998-12-13 10:32:47.000000000 -0500
+++ tmp/sabre-0.2.4b/src/obj_3d.h	2008-01-31 10:55:42.000000000 -0500
@@ -144,19 +144,19 @@
 
   void readFile(char *path);
   void writeFile(char *path);
-  void read(istream &);
-  void write(ostream &);
-  friend inline istream &operator >>(istream &, C_3DInfoManager &);
-  friend inline ostream &operator <<(ostream &, C_3DInfoManager &);
+  void read(std::istream &);
+  void write(std::ostream &);
+  friend inline std::istream &operator >>(std::istream &, C_3DInfoManager &);
+  friend inline std::ostream &operator <<(std::ostream &, C_3DInfoManager &);
 } ;
 
-istream &operator >>(istream &is, C_3DInfoManager &cm)
+std::istream &operator >>(std::istream &is, C_3DInfoManager &cm)
 {
   cm.read(is);
   return(is);
 }
 
-ostream &operator <<(ostream &os, C_3DInfoManager &cm)
+std::ostream &operator <<(std::ostream &os, C_3DInfoManager &cm)
 {
   cm.write(os);
   return(os);
@@ -236,19 +236,19 @@
       return (C_3DOBJECT_T); 
     }
 
-  void read(istream &);
-  void write(ostream &);
-  friend inline istream &operator >>(istream & is, C_3DObject2 &ob);
-  friend inline ostream &operator <<(ostream & os, C_3DObject2 &ob);
+  void read(std::istream &);
+  void write(std::ostream &);
+  friend inline std::istream &operator >>(std::istream & is, C_3DObject2 &ob);
+  friend inline std::ostream &operator <<(std::ostream & os, C_3DObject2 &ob);
 };
 
-istream & operator >>(istream &is, C_3DObject2 &ob)
+std::istream & operator >>(std::istream &is, C_3DObject2 &ob)
 {
   ob.read(is);
   return(is);
 }
 
-ostream & operator <<(ostream &os, C_3DObject2 &ob)
+std::ostream & operator <<(std::ostream &os, C_3DObject2 &ob)
 {
   ob.write(os);
   return(os);
@@ -273,8 +273,8 @@
   }
 
   ~C_DrawList2();
-  friend istream & operator >>(istream &is, C_DrawList2 &cs);
-  void read(istream &is);
+  friend std::istream & operator >>(std::istream &is, C_DrawList2 &cs);
+  void read(std::istream &is);
   void translate(R_3DPoint &);
   void add_draw_list(DrawList &, Port_3D &port);
   void get_targets(Target_List &);
diff -urN sabre-0.2.4b/src/pilobj.C tmp/sabre-0.2.4b/src/pilobj.C
--- sabre-0.2.4b/src/pilobj.C	1999-01-04 19:42:25.000000000 -0500
+++ tmp/sabre-0.2.4b/src/pilobj.C	2008-01-31 12:23:05.000000000 -0500
@@ -19,8 +19,8 @@
 */
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <string.h>
 #include <math.h>
 #include <limits.h>
@@ -37,7 +37,7 @@
 #include "flight.h"
 #include "pilobj.h"
 
-istream &operator >>(istream &is, Pilot_Params &ps)
+std::istream &operator >>(std::istream &is, Pilot_Params &ps)
 {
 	char c= ' ';
 	READ_TOK('(',is,c);
diff -urN sabre-0.2.4b/src/pilobj.h tmp/sabre-0.2.4b/src/pilobj.h
--- sabre-0.2.4b/src/pilobj.h	1999-02-12 17:33:41.000000000 -0500
+++ tmp/sabre-0.2.4b/src/pilobj.h	2008-01-31 11:07:39.000000000 -0500
@@ -53,7 +53,7 @@
 	 negG = -3.0;
 	 log = 0;
   }
-  friend istream &operator>>(istream &, Pilot_Params &);
+  friend std::istream &operator>>(std::istream &, Pilot_Params &);
 };
 
 class Pilot;
@@ -78,4 +78,4 @@
   int set_message(char *mss, int priority, Pilot *fr);
 };
 
-#endif
\ No newline at end of file
+#endif
diff -urN sabre-0.2.4b/src/pilot.C tmp/sabre-0.2.4b/src/pilot.C
--- sabre-0.2.4b/src/pilot.C	1999-11-21 19:38:53.000000000 -0500
+++ tmp/sabre-0.2.4b/src/pilot.C	2008-01-30 19:12:01.000000000 -0500
@@ -29,8 +29,8 @@
  *************************************************/
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <string.h>
 #include <math.h>
 #include <limits.h>
diff -urN sabre-0.2.4b/src/port_3d.C tmp/sabre-0.2.4b/src/port_3d.C
--- sabre-0.2.4b/src/port_3d.C	1999-02-25 09:26:24.000000000 -0500
+++ tmp/sabre-0.2.4b/src/port_3d.C	2008-01-31 11:29:58.000000000 -0500
@@ -26,8 +26,8 @@
  *************************************************/
 #include <stdio.h>
 #include <math.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include "vmath.h"
 #include "rtkey.h"
 #include "vga_13.h"
@@ -52,7 +52,7 @@
 REAL_TYPE Port_3D::fovy = 280;
 extern REAL_TYPE world_scale;
 
-void Port_3D::read(istream &is)
+void Port_3D::read(std::istream &is)
 {
   /* used to be fovx */
   float ignore;
@@ -68,20 +68,20 @@
   cos_roll = cos(roll);
 }
 
-istream &operator >>(istream &is, Port_3D &port)
+std::istream &operator >>(std::istream &is, Port_3D &port)
 {
   port.read(is);
   return is;
 }
 
-void Port_3D::write(ostream &os)
+void Port_3D::write(std::ostream &os)
 {
   os << slook_from  << "\n";
   os << look_from << "\n";
   os << look_at << "\n";
 }
 
-ostream &operator <<(ostream &os, Port_3D &port)
+std::ostream &operator <<(std::ostream &os, Port_3D &port)
 {
   port.write(os);
   return os;
diff -urN sabre-0.2.4b/src/port_3d.h tmp/sabre-0.2.4b/src/port_3d.h
--- sabre-0.2.4b/src/port_3d.h	1999-02-12 17:33:40.000000000 -0500
+++ tmp/sabre-0.2.4b/src/port_3d.h	2008-01-31 10:45:50.000000000 -0500
@@ -134,10 +134,10 @@
 	inline void set_look_from(const R_3DPoint &);
 	inline const Vector &calc_view_normal();
 	friend class C_Shape;
-	friend istream &operator >>(istream &is, Port_3D &port);
-	void read(istream &is);
-	friend ostream &operator <<(ostream &os, Port_3D &port);
-	void write(ostream &os);
+	friend std::istream &operator >>(std::istream &is, Port_3D &port);
+	void read(std::istream &is);
+	friend std::ostream &operator <<(std::ostream &os, Port_3D &port);
+	void write(std::ostream &os);
 
 
 	int IsVisible(int x, int y)
diff -urN sabre-0.2.4b/src/portkey.C tmp/sabre-0.2.4b/src/portkey.C
--- sabre-0.2.4b/src/portkey.C	1999-02-12 17:33:41.000000000 -0500
+++ tmp/sabre-0.2.4b/src/portkey.C	2008-01-30 19:17:32.000000000 -0500
@@ -23,8 +23,8 @@
 #include <math.h>
 #include <signal.h>
 #include <ctype.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <setjmp.h>
 #include "defs.h"
 #include "pc_keys.h"
diff -urN sabre-0.2.4b/src/ppm.C tmp/sabre-0.2.4b/src/ppm.C
--- sabre-0.2.4b/src/ppm.C	1998-12-13 10:32:44.000000000 -0500
+++ tmp/sabre-0.2.4b/src/ppm.C	2008-01-31 12:14:26.000000000 -0500
@@ -28,8 +28,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include "vga_13.h"
 
 static int fcount = 0;
@@ -46,11 +46,11 @@
   if (!buffer_ptr)
 	return;
   sprintf(buff,"sbr%04d.ppm",fcount++);
-  ofstream os(buff);
+  std::ofstream os(buff);
   if (os)
     {
       os << "P3\n";
-      os << "# CREATOR: Sabre Fighter Plane Simulator Version " << VERSION << " " << __DATE__ << endl;
+      os << "# CREATOR: Sabre Fighter Plane Simulator Version " << VERSION << " " << __DATE__ << std::endl;
       os << SCREEN_WIDTH << " " << SCREEN_HEIGHT << "\n";
       os << 63 << '\n';
       pptr = buffer_ptr;
diff -urN sabre-0.2.4b/src/rendpoly.C tmp/sabre-0.2.4b/src/rendpoly.C
--- sabre-0.2.4b/src/rendpoly.C	2008-01-31 12:27:37.000000000 -0500
+++ tmp/sabre-0.2.4b/src/rendpoly.C	2008-01-30 19:18:09.000000000 -0500
@@ -18,8 +18,8 @@
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
diff -urN sabre-0.2.4b/src/rndrpoly.C tmp/sabre-0.2.4b/src/rndrpoly.C
--- sabre-0.2.4b/src/rndrpoly.C	1999-11-21 09:27:51.000000000 -0500
+++ tmp/sabre-0.2.4b/src/rndrpoly.C	2008-01-30 19:15:39.000000000 -0500
@@ -20,8 +20,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
diff -urN sabre-0.2.4b/src/rndzpoly.C tmp/sabre-0.2.4b/src/rndzpoly.C
--- sabre-0.2.4b/src/rndzpoly.C	1998-12-13 10:32:44.000000000 -0500
+++ tmp/sabre-0.2.4b/src/rndzpoly.C	2008-01-30 19:11:41.000000000 -0500
@@ -28,8 +28,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
diff -urN sabre-0.2.4b/src/rotate.C tmp/sabre-0.2.4b/src/rotate.C
--- sabre-0.2.4b/src/rotate.C	1998-12-13 10:32:44.000000000 -0500
+++ tmp/sabre-0.2.4b/src/rotate.C	2008-01-30 19:21:28.000000000 -0500
@@ -27,7 +27,7 @@
 #include <time.h>
 #include <string.h>
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 #include "defs.h"
 #include "grafix.h"
 #include "vga_13.h"
diff -urN sabre-0.2.4b/src/scnedit.C tmp/sabre-0.2.4b/src/scnedit.C
--- sabre-0.2.4b/src/scnedit.C	1999-02-12 17:33:41.000000000 -0500
+++ tmp/sabre-0.2.4b/src/scnedit.C	2008-01-31 11:25:08.000000000 -0500
@@ -33,8 +33,8 @@
 #include <math.h>
 #include <signal.h>
 #include <ctype.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <setjmp.h>
 #include <stdarg.h>
 #include "sim.h"
@@ -68,7 +68,7 @@
 
 void SceneEdit::readWorldFile()
 {
-ifstream is;
+std::ifstream is;
 char c;
 float pixl_ratio;
 char buff[100];
diff -urN sabre-0.2.4b/src/simfile.C tmp/sabre-0.2.4b/src/simfile.C
--- sabre-0.2.4b/src/simfile.C	1999-04-19 16:59:18.000000000 -0400
+++ tmp/sabre-0.2.4b/src/simfile.C	2008-01-31 12:02:58.000000000 -0500
@@ -27,8 +27,8 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include "vmath.h"
 #include "defs.h"
@@ -63,7 +63,7 @@
 
 void read_texture_file(char *path)
 {
-  ifstream is;
+  std::ifstream is;
 #ifdef SABREWIN
   if (map_man)
     delete map_man;
@@ -84,7 +84,7 @@
 void read_palette_file(char *path)
 {
   color_spec cs;
-  ifstream is;
+  std::ifstream is;
   current_path = path;
   if (open_is(is,path))
     {
@@ -96,7 +96,7 @@
 
 void read_sound_file(char *path)
 {
-  ifstream is;
+  std::ifstream is;
   int		nsounds;
   int		gotline;
   char		buff[BUFSIZ];
@@ -128,7 +128,7 @@
     }
 }
 
-int get_line(istream &is, char *buff, int size)
+int get_line(std::istream &is, char *buff, int size)
 {
   int l;
   while (is)
@@ -154,7 +154,7 @@
   return 0;
 }
 
-int open_is(ifstream &is, char *path)
+int open_is(std::ifstream &is, char *path)
 {
   char *lpath = build_libpath(path);
   current_path = lpath;
@@ -167,7 +167,7 @@
 }
 
 
-int open_os(ofstream &os, char *path)
+int open_os(std::ofstream &os, char *path)
 {
   os.open(path);
   if (!os.good())
@@ -175,7 +175,7 @@
   return (os.good());
 }
 
-int open_libos(ofstream &os, char *path)
+int open_libos(std::ofstream &os, char *path)
 {
   char *lpath = build_libpath(path);
   os.open(lpath);
@@ -185,7 +185,7 @@
   return (os.good());
 }
 
-int read_int(istream &is)
+int read_int(std::istream &is)
 {
   char c;
   int result = 0;
diff -urN sabre-0.2.4b/src/simfile.h tmp/sabre-0.2.4b/src/simfile.h
--- sabre-0.2.4b/src/simfile.h	1998-12-31 03:58:34.000000000 -0500
+++ tmp/sabre-0.2.4b/src/simfile.h	2008-01-31 10:17:17.000000000 -0500
@@ -21,18 +21,18 @@
 #define __simfile_h
 #include "sim.h"
 #include "simerr.h"
-#include "iostream.h"
-#include "fstream.h"
+#include <iostream>
+#include <fstream>
 
 void read_palette_file(char *);
 void read_texture_file(char *);
 void read_sound_file(char *);
-int get_line(istream &, char *, int);
-int open_is(ifstream &, char *);
-int open_os(ofstream &, char *);
-int open_libos(ofstream &, char *);
+int get_line(std::istream &, char *, int);
+int open_is(std::ifstream &, char *);
+int open_os(std::ofstream &, char *);
+int open_libos(std::ofstream &, char *);
 char *build_libpath(char *);
-int read_int(istream &);
+int read_int(std::istream &);
 
 extern char *current_path;
 extern char *file_context;
diff -urN sabre-0.2.4b/src/simfilex.C tmp/sabre-0.2.4b/src/simfilex.C
--- sabre-0.2.4b/src/simfilex.C	1999-04-19 17:01:02.000000000 -0400
+++ tmp/sabre-0.2.4b/src/simfilex.C	2008-01-31 12:21:05.000000000 -0500
@@ -24,7 +24,7 @@
 #include "simfile.h"
 #include "simfilex.h"
 
-int simfileX::readint(istream &is, char tok0, char tok1)
+int simfileX::readint(std::istream &is, char tok0, char tok1)
 {
   int result;
   char c;
@@ -41,7 +41,7 @@
   return (result);
 }
 
-int simfileX::readstr(istream &is, char *buff, int maxlen, char tok0, char tok1)
+int simfileX::readstr(std::istream &is, char *buff, int maxlen, char tok0, char tok1)
 {
   char c0,c1=0;
   int l;
@@ -89,7 +89,7 @@
   return (0);
 }
 
-int simfileX::readinput(istream &is, char *buff, int maxlen, int &i, 
+int simfileX::readinput(std::istream &is, char *buff, int maxlen, int &i, 
 			char tok0,
 			char tok1)
 {
@@ -103,7 +103,7 @@
   return (result);
 }
 
-int simfileX::readdictinput(istream &is, char *buff, int maxlen, int &val,
+int simfileX::readdictinput(std::istream &is, char *buff, int maxlen, int &val,
 			    dict *d, int d_cnt,
 			    char tok0,
 			    char tok1)
diff -urN sabre-0.2.4b/src/simfilex.h tmp/sabre-0.2.4b/src/simfilex.h
--- sabre-0.2.4b/src/simfilex.h	1998-12-13 10:32:47.000000000 -0500
+++ tmp/sabre-0.2.4b/src/simfilex.h	2008-01-31 11:47:46.000000000 -0500
@@ -31,13 +31,13 @@
   char *str;
   int  val;
 } dict;
-static int readint(istream &is, char tok0 = 0, char tok1 = 0);
-static int readstr(istream &is, char *bff, int maxlen, char tok0 = 0, 
+static int readint(std::istream &is, char tok0 = 0, char tok1 = 0);
+static int readstr(std::istream &is, char *bff, int maxlen, char tok0 = 0, 
 		   char tok1 = 0);
-static int readinput(istream &is, char *buff, int maxlen, int &i, 
+static int readinput(std::istream &is, char *buff, int maxlen, int &i, 
 	      char tok0 = 0,
 	      char tok1 = 0);
-static int readdictinput(istream &is, char *buff, int maxlen, int &i,
+static int readdictinput(std::istream &is, char *buff, int maxlen, int &i,
 			dict *d, int d_cnt,
 			char tok0 = 0,
 			char tok1 = 0);
diff -urN sabre-0.2.4b/src/siminput.C tmp/sabre-0.2.4b/src/siminput.C
--- sabre-0.2.4b/src/siminput.C	1999-11-24 18:27:36.000000000 -0500
+++ tmp/sabre-0.2.4b/src/siminput.C	2008-01-30 19:16:17.000000000 -0500
@@ -27,8 +27,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
diff -urN sabre-0.2.4b/src/simsnd.C tmp/sabre-0.2.4b/src/simsnd.C
--- sabre-0.2.4b/src/simsnd.C	1999-04-19 15:23:39.000000000 -0400
+++ tmp/sabre-0.2.4b/src/simsnd.C	2008-01-30 19:20:52.000000000 -0500
@@ -28,7 +28,7 @@
 #include <windowsx.h> 
 #include <mmsystem.h>
 #include <objbase.h>
-#include <iostream.h>
+#include <iostream>
 #include <conio.h>
 #include <stdlib.h>
 #include <malloc.h>
diff -urN sabre-0.2.4b/src/target.C tmp/sabre-0.2.4b/src/target.C
--- sabre-0.2.4b/src/target.C	1998-12-13 10:32:45.000000000 -0500
+++ tmp/sabre-0.2.4b/src/target.C	2008-01-30 19:15:21.000000000 -0500
@@ -27,7 +27,7 @@
 #include <stdio.h>
 #include <values.h>
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 #include <string.h>
 #include "vmath.h"
 #include "target.h"
diff -urN sabre-0.2.4b/src/terrain.C tmp/sabre-0.2.4b/src/terrain.C
--- sabre-0.2.4b/src/terrain.C	1999-02-12 17:33:41.000000000 -0500
+++ tmp/sabre-0.2.4b/src/terrain.C	2008-01-30 19:16:57.000000000 -0500
@@ -27,7 +27,7 @@
 #include <stdio.h>
 #include <values.h>
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 #include <string.h>
 #include "vmath.h"
 #include "rendpoly.h"
diff -urN sabre-0.2.4b/src/txtrmap.C tmp/sabre-0.2.4b/src/txtrmap.C
--- sabre-0.2.4b/src/txtrmap.C	2008-01-31 12:27:37.000000000 -0500
+++ tmp/sabre-0.2.4b/src/txtrmap.C	2008-01-31 11:40:40.000000000 -0500
@@ -33,8 +33,8 @@
 #include <windows.h>
 #endif
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
@@ -85,7 +85,7 @@
 	}
 }
 
-void TextrMap::read(istream &is)
+void TextrMap::read(std::istream &is)
 {
 char *cptr;
 int ii;
@@ -141,20 +141,20 @@
 	READ_TOK('}',is,c);
 }
 
-void TextrMap::write(ostream &os)
+void TextrMap::write(std::ostream &os)
 {
   os << "{\n" << id << "*\n" << trans_colr << "\n";
   compress();
   os << "}\n";
 }
 
-istream & operator >>(istream &is, TextrMap &tm)
+std::istream & operator >>(std::istream &is, TextrMap &tm)
 {
   tm.read(is);
   return (is);
 }
 
-ostream & operator <<(ostream &os, TextrMap &tm)
+std::ostream & operator <<(std::ostream &os, TextrMap &tm)
 {
   tm.write(os);
   return (os);
@@ -167,7 +167,7 @@
 unsigned long srcsize;
 char     *tgt;
 char     *fname;
-ofstream os;
+std::ofstream os;
 
 	if (bytes != NULL)
 	{
@@ -812,7 +812,7 @@
     return 0;
 }
 
-void TextrMap_Manager::read(istream &is)
+void TextrMap_Manager::read(std::istream &is)
 {
 int n;
 
@@ -844,7 +844,7 @@
 	}
 }
 
-istream &operator >>(istream &is, TextrMap_Manager &tm)
+std::istream &operator >>(std::istream &is, TextrMap_Manager &tm)
 {
   tm.read(is);
   return (is);
@@ -852,12 +852,12 @@
 
 void TextrMap_Manager::read_file(char *path)
 {
-  ifstream is;
+  std::ifstream is;
   if (open_is(is,path))
     is >> *this;
 }
 
-void TextrMap_Manager::write(ostream &os)
+void TextrMap_Manager::write(std::ostream &os)
 {
   os << nxt << '\n';
   for (int i=0;i<nxt;i++)
@@ -865,7 +865,7 @@
 }
 
 
-ostream &operator <<(ostream &os, TextrMap_Manager &tm)
+std::ostream &operator <<(std::ostream &os, TextrMap_Manager &tm)
 {
   tm.write(os);
   return (os);
@@ -873,7 +873,7 @@
 
 void TextrMap_Manager::write_file(char *path)
 {
-  ofstream os;
+  std::ofstream os;
   if (open_libos(os,path))
     write(os);
 }
diff -urN sabre-0.2.4b/src/txtrmap.h tmp/sabre-0.2.4b/src/txtrmap.h
--- sabre-0.2.4b/src/txtrmap.h	2008-01-31 12:27:37.000000000 -0500
+++ tmp/sabre-0.2.4b/src/txtrmap.h	2008-01-31 10:48:13.000000000 -0500
@@ -152,10 +152,10 @@
 			delete [] mc;
 	}
 
-	friend istream &operator >>(istream &is, TextrMap &tm);
-	friend ostream &operator >>(ostream &os, TextrMap &tm);
-	void read(istream &is);
-	void write(ostream &os);
+	friend std::istream &operator >>(std::istream &is, TextrMap &tm);
+	friend std::ostream &operator >>(std::ostream &os, TextrMap &tm);
+	void read(std::istream &is);
+	void write(std::ostream &os);
 	void compress();
 	void read_compressed(int = 0);
 	unsigned char *getBytes(int = 0);
@@ -220,13 +220,13 @@
 
 	int add_map(TextrMap &);
 
-	void read(istream &is);
+	void read(std::istream &is);
 	void read_file(char *);
-	friend istream &operator >>(istream &is, TextrMap_Manager &);
+	friend std::istream &operator >>(std::istream &is, TextrMap_Manager &);
 
-	void write(ostream &os);
+	void write(std::ostream &os);
 	void write_file(char *);
-	friend ostream &operator <<(ostream &os, TextrMap_Manager &);
+	friend std::ostream &operator <<(std::ostream &os, TextrMap_Manager &);
 
 
 	TextrMap &get_map(int n)
diff -urN sabre-0.2.4b/src/unguided.C tmp/sabre-0.2.4b/src/unguided.C
--- sabre-0.2.4b/src/unguided.C	1998-12-31 03:21:23.000000000 -0500
+++ tmp/sabre-0.2.4b/src/unguided.C	2008-01-30 19:13:19.000000000 -0500
@@ -27,8 +27,8 @@
  *************************************************/
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <limits.h>
 #include <values.h>
 #include "defs.h"
diff -urN sabre-0.2.4b/src/viewobj.C tmp/sabre-0.2.4b/src/viewobj.C
--- sabre-0.2.4b/src/viewobj.C	1998-12-13 10:32:45.000000000 -0500
+++ tmp/sabre-0.2.4b/src/viewobj.C	2008-01-30 19:14:09.000000000 -0500
@@ -26,8 +26,8 @@
  *************************************************/
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include "defs.h"
 #include "vga_13.h"
diff -urN sabre-0.2.4b/src/vmath.C tmp/sabre-0.2.4b/src/vmath.C
--- sabre-0.2.4b/src/vmath.C	1998-12-13 10:32:45.000000000 -0500
+++ tmp/sabre-0.2.4b/src/vmath.C	2008-01-31 11:26:16.000000000 -0500
@@ -21,7 +21,7 @@
  * vector.cpp                                                   *
  ****************************************************************/
 #include <stdio.h>
-#include <iostream.h>
+#include <iostream>
 #include <math.h>
 #include "simfile.h"
 #include "vmath.h"
@@ -33,7 +33,7 @@
 const REAL_TYPE zfact = REAL_TYPE(1.0);
 #endif
 
-istream &operator >> (istream &is, Vector &v)
+std::istream &operator >> (std::istream &is, Vector &v)
 {
   char c;
   READ_TOKI('<',is,c);
@@ -42,7 +42,7 @@
   return is;
 }
 
-istream &operator >> (istream &is,R_3DPoint &r3D)
+std::istream &operator >> (std::istream &is,R_3DPoint &r3D)
 {
   char c;
   READ_TOKI('<',is,c)
@@ -56,7 +56,7 @@
 
 
 
-istream &operator >> (istream &is, S_3DPoint &s3D)
+std::istream &operator >> (std::istream &is, S_3DPoint &s3D)
 {
   char c;
   READ_TOKI('<',is,c)
@@ -65,7 +65,7 @@
     return is;
 }
 
-istream &operator >> (istream &is, TxtPoint &tp)
+std::istream &operator >> (std::istream &is, TxtPoint &tp)
 {
   char c;
   READ_TOKI('<',is,c)
diff -urN sabre-0.2.4b/src/vmath.h tmp/sabre-0.2.4b/src/vmath.h
--- sabre-0.2.4b/src/vmath.h	1999-02-12 17:33:41.000000000 -0500
+++ tmp/sabre-0.2.4b/src/vmath.h	2008-01-31 10:42:47.000000000 -0500
@@ -21,7 +21,7 @@
 #define __vmath_h
 
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 
 
 #ifndef REAL_TYPE
@@ -118,10 +118,10 @@
     X -= v.X; Y -= v.Y; Z -= v.Z;
   }
 
-  friend ostream &operator << (ostream &os,Vector &v)
+  friend std::ostream &operator << (std::ostream &os,Vector &v)
   { return os  << '<' << v.X << ' ' << v.Y << ' ' << v.Z << '>' ; }
 
-  friend istream &operator >> (istream &is,Vector &v);
+  friend std::istream &operator >> (std::istream &is,Vector &v);
 
   void SetZero( void )
   {
@@ -338,10 +338,10 @@
   REAL_TYPE _z()
     { return z; }
 
-  friend ostream &operator << (ostream &os,R_3DPoint &r3D)
+  friend std::ostream &operator << (std::ostream &os,R_3DPoint &r3D)
     { return os  << '<' << r3D.x << ' ' << r3D.y << ' ' << r3D.z << '>' ; }
 
-  friend istream &operator >> (istream &is,R_3DPoint &r3D);
+  friend std::istream &operator >> (std::istream &is,R_3DPoint &r3D);
 
   friend R_3DPoint operator +(const R_3DPoint &r1, const R_3DPoint &r2)
     { return R_3DPoint(r1.x+r2.x,r1.y+r2.y,r1.z+r2.z); }
@@ -509,10 +509,10 @@
   TXTP_TYPE _i()
     { return i; }
 
-  friend ostream &operator << (ostream &os,TxtPoint &r3D)
+  friend std::ostream &operator << (std::ostream &os,TxtPoint &r3D)
     { return os  << '<' << r3D.u << ' ' << r3D.v << ' ' << r3D.i << '>' ; }
 
-  friend istream &operator >> (istream &is,TxtPoint &r3D);
+  friend std::istream &operator >> (std::istream &is,TxtPoint &r3D);
 
   friend TxtPoint operator +(const TxtPoint &r1, const TxtPoint &r2)
     { return TxtPoint(r1.u+r2.u,r1.v+r2.v,r1.i+r2.i); }
@@ -753,9 +753,9 @@
   REAL_TYPE _rho()
     { return rho; }
 
-  friend istream &operator >>( istream &is, S_3DPoint &s3D);
+  friend std::istream &operator >>( std::istream &is, S_3DPoint &s3D);
 
-  friend ostream &operator << ( ostream &os, S_3DPoint &s3D)
+  friend std::ostream &operator << ( std::ostream &os, S_3DPoint &s3D)
     { return os  << "<" << s3D.theta << " " << s3D.phi << " " << s3D.rho << ">"; }
 
   friend S_3DPoint operator +(const S_3DPoint &s1, const S_3DPoint &s2)
diff -urN sabre-0.2.4b/src/waypoint.C tmp/sabre-0.2.4b/src/waypoint.C
--- sabre-0.2.4b/src/waypoint.C	1998-12-13 10:32:45.000000000 -0500
+++ tmp/sabre-0.2.4b/src/waypoint.C	2008-01-31 12:19:18.000000000 -0500
@@ -26,8 +26,8 @@
  *************************************************/
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <string.h>
 #include <math.h>
 #include <limits.h>
@@ -43,7 +43,7 @@
 WayPoint ** WayPoint::gwaypoints = NULL;
 int WayPoint::n_waypoints = 0;
 
-void WayPoint::read(istream &is)
+void WayPoint::read(std::istream &is)
 {
   char c = ' ';
   READ_TOK('(',is,c)
@@ -54,14 +54,14 @@
     average_speed = kts2fps(average_speed);
 }
 
-void WayPoint::write(ostream &os)
+void WayPoint::write(std::ostream &os)
 {
   os << "(\n" << mode << " " <<  mode_x << " " << mode_y << "\n";
   os << location << " " << fps2kts(average_speed) << "\n";
   os << ")\n";
 }
 
-void WayPoint::readWayPoints(istream &is)
+void WayPoint::readWayPoints(std::istream &is)
 {
   int i;
   int n;
@@ -78,7 +78,7 @@
     }
 }
 
-WayPoint *WayPoint::readWayPointChain(istream &is, int &n)
+WayPoint *WayPoint::readWayPointChain(std::istream &is, int &n)
 {
   char c;
   WayPoint *result = NULL;
@@ -102,19 +102,19 @@
 
 void WayPoint::readWayPointFile(char *path)
 {
-  ifstream is;
+  std::ifstream is;
   if (open_is(is,path))
     readWayPoints(is);
 }
 
-void WayPoint::writeWayPoints(ostream &)
+void WayPoint::writeWayPoints(std::ostream &)
 {
 
 }
 
 void WayPoint::writeWayPointFile(char *path)
 {
-  ofstream os;
+  std::ofstream os;
   if (open_libos(os,path))
     writeWayPoints(os);
 }
diff -urN sabre-0.2.4b/src/waypoint.h tmp/sabre-0.2.4b/src/waypoint.h
--- sabre-0.2.4b/src/waypoint.h	1998-12-31 03:21:24.000000000 -0500
+++ tmp/sabre-0.2.4b/src/waypoint.h	2008-01-31 12:17:38.000000000 -0500
@@ -75,24 +75,24 @@
 		return *this;
 	}
 
-  void read(istream &);
-  friend istream &operator >> (istream & is, WayPoint &wp)
+  void read(std::istream &);
+  friend std::istream &operator >> (std::istream & is, WayPoint &wp)
     {
       wp.read(is);
       return(is);
     }
-  void write(ostream &);
-  friend ostream &operator << (ostream &os, WayPoint &wp)
+  void write(std::ostream &);
+  friend std::ostream &operator << (std::ostream &os, WayPoint &wp)
     {
       wp.write(os);
       return(os);
     }
 
-  static void readWayPoints(istream &is);
-  static void writeWayPoints(ostream &os);
+  static void readWayPoints(std::istream &is);
+  static void writeWayPoints(std::ostream &os);
   static void readWayPointFile(char *);
   static void writeWayPointFile(char *);
-  static WayPoint *readWayPointChain(istream &is, int &);
+  static WayPoint *readWayPointChain(std::istream &is, int &);
   static WayPoint **gwaypoints;
   static int        n_waypoints;
   static WayPoint *getWayPoint(char *);
diff -urN sabre-0.2.4b/src/weapons.C tmp/sabre-0.2.4b/src/weapons.C
--- sabre-0.2.4b/src/weapons.C	1998-12-31 03:21:23.000000000 -0500
+++ tmp/sabre-0.2.4b/src/weapons.C	2008-01-31 11:49:37.000000000 -0500
@@ -27,8 +27,8 @@
  *************************************************/
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <limits.h>
 #include <values.h>
@@ -51,7 +51,7 @@
 /***************************************************
  * Weapon_Specs methods                            *
  ***************************************************/
-void Weapon_Specs::read(istream &is)
+void Weapon_Specs::read(std::istream &is)
 {
   is >> wep_name;
   is >> flt_specs;
@@ -60,7 +60,7 @@
   is >> damage_factor;
 }
 
-void Weapon_Specs::write(ostream &os)
+void Weapon_Specs::write(std::ostream &os)
 {
   os << wep_name << '\n';
   os << flt_specs << '\n';
@@ -70,7 +70,7 @@
 /***************************************************
  * Gun_Specs methods                               *
  ***************************************************/
-void Gun_Specs::read(istream &is)
+void Gun_Specs::read(std::istream &is)
 {
   int clr,dummy;
   char c =  ' ';
@@ -87,7 +87,7 @@
   rounds_per_second = 1.0 / (rounds_per_second / 60);
 }
 
-void Gun_Specs::write(ostream &os)
+void Gun_Specs::write(std::ostream &os)
 {
   color_spec cs(tracer_color);
   os << '(' << '\n';
@@ -100,7 +100,7 @@
 /***************************************************
  * Cannon_Specs methods                            *
  ***************************************************/
-void Cannon_Specs::read(istream &is)
+void Cannon_Specs::read(std::istream &is)
 {
   char c;
   int clr;
@@ -128,7 +128,7 @@
   rounds_per_second = 1.0 / (rounds_per_second / 60);
 }
 
-void Cannon_Specs::write(ostream &os)
+void Cannon_Specs::write(std::ostream &os)
 {
   color_spec cs(tracer_color);
   os << '(' << '\n';
@@ -162,7 +162,7 @@
   zm->read_file(path);
 }
 
-void Bomb_Specs::read(istream &is)
+void Bomb_Specs::read(std::istream &is)
 {
   char c;
   READ_TOKI('(',is,c);
@@ -179,7 +179,7 @@
   blast_map_size *= world_scale;
 }
 
-void Bomb_Specs::write(ostream &os)
+void Bomb_Specs::write(std::ostream &os)
 {
   os << '(' << '\n';
   Weapon_Specs::write(os);
@@ -195,7 +195,7 @@
 /***************************************************
  * Rocket_Specs methods                            *
  ***************************************************/
-void Rocket_Specs::read(istream &is)
+void Rocket_Specs::read(std::istream &is)
 {
   char c;
   READ_TOKI('(',is,c);
@@ -220,7 +220,7 @@
   flume_delta_size *= world_scale;
 }
 
-void Rocket_Specs::write(ostream &os)
+void Rocket_Specs::write(std::ostream &os)
 {
   os << '(' << '\n';
   Weapon_Specs::write(os);
@@ -273,7 +273,7 @@
   wep_i = NULL;
 }
 
-void Weapon::read(istream &is)
+void Weapon::read(std::istream &is)
 {
   char c;
   READ_TOKI('{',is,c);
@@ -297,7 +297,7 @@
   READ_TOK('}',is,c);
 }
 
-void Weapon::write(ostream &)
+void Weapon::write(std::ostream &)
 {
 
 }
@@ -401,7 +401,7 @@
 /*****************************************************
  * Guns methods                                      *
  *****************************************************/
-istream & operator >>(istream &is, Guns &gp)
+std::istream & operator >>(std::istream &is, Guns &gp)
 {
   ((Weapon &)gp).read(is);
   return is;
@@ -445,7 +445,7 @@
 /********************************************************
  * Bombs methods                                        *
  ********************************************************/
-istream & operator >>(istream &is, Bombs &b)
+std::istream & operator >>(std::istream &is, Bombs &b)
 {
   ((Weapon &)b).read(is);
   return is;
@@ -480,7 +480,7 @@
 /********************************************************
  * Rockets methods                                        *
  ********************************************************/
-istream & operator >>(istream &is, Rockets &r)
+std::istream & operator >>(std::istream &is, Rockets &r)
 {
   ((Weapon &)r).read(is);
   return is;
@@ -516,7 +516,7 @@
 /********************************************************
  * Missiles methods                                     *
  ********************************************************/
-istream & operator >>(istream &is, Missiles &r)
+std::istream & operator >>(std::istream &is, Missiles &r)
 {
   ((Weapon &)r).read(is);
   return is;
@@ -551,7 +551,7 @@
 /********************************************************
  * FuelTanks methods                                     *
  ********************************************************/
-istream & operator >>(istream &is, FuelTanks &r)
+std::istream & operator >>(std::istream &is, FuelTanks &r)
 {
   ((Weapon &)r).read(is);
   return is;
@@ -809,7 +809,7 @@
 
 void Weapons_Manager::read_file(char *path)
 {
-  ifstream infile;
+  std::ifstream infile;
   if (open_is(infile,path))
     infile >> *this;
   infile.close();
@@ -826,7 +826,7 @@
   { "cannon_type",cannon_t }
 };
 
-void Weapons_Manager::read(istream &is)
+void Weapons_Manager::read(std::istream &is)
 {
   int i,j,ndx,ndx2;
   int wep_type,sz;
@@ -993,7 +993,7 @@
   READ_TOKI('}',is,c);
 }
 
-istream & operator >>(istream &is, Weapons_Manager &wp)
+std::istream & operator >>(std::istream &is, Weapons_Manager &wp)
 {
   wp.read(is);
   return is;
diff -urN sabre-0.2.4b/src/weapons.h tmp/sabre-0.2.4b/src/weapons.h
--- sabre-0.2.4b/src/weapons.h	1998-12-31 03:21:24.000000000 -0500
+++ tmp/sabre-0.2.4b/src/weapons.h	2008-01-31 11:06:37.000000000 -0500
@@ -45,8 +45,8 @@
   float max_range;
   int damage_factor;
   int wep_type;
-  void read(istream &is);
-  void write(ostream &os);
+  void read(std::istream &is);
+  void write(std::ostream &os);
  
   Weapon_Specs()
     {
@@ -55,12 +55,12 @@
       wep_type = -1;
     }
 
-  friend istream &operator >>(istream &is, Weapon_Specs &ws)
+  friend std::istream &operator >>(std::istream &is, Weapon_Specs &ws)
     {
       ws.read(is);
       return(is);
     }
-  friend ostream &operator <<(ostream &os, Weapon_Specs &ws)
+  friend std::ostream &operator <<(std::ostream &os, Weapon_Specs &ws)
     {
       ws.write(os);
       return(os);
@@ -116,14 +116,14 @@
       wep_type = gun_t;
       rounds_per_second = 0.0;
     }
-  void read(istream &is);
-  void write(ostream &os);
-  friend istream &operator >>(istream &is, Gun_Specs &ws)
+  void read(std::istream &is);
+  void write(std::ostream &os);
+  friend std::istream &operator >>(std::istream &is, Gun_Specs &ws)
     {
       ws.read(is);
       return(is);
     }
-  friend ostream &operator <<(ostream &os, Gun_Specs &ws)
+  friend std::ostream &operator <<(std::ostream &os, Gun_Specs &ws)
     {
       ws.write(os);
       return(os);
@@ -175,14 +175,14 @@
       return (smoke_map);
     }
 
-  void read(istream &is);
-  void write(ostream &os);
-  friend istream &operator >>(istream &is, Cannon_Specs &cs)
+  void read(std::istream &is);
+  void write(std::ostream &os);
+  friend std::istream &operator >>(std::istream &is, Cannon_Specs &cs)
     {
       cs.read(is);
       return(is);
     }
-  friend ostream &operator <<(ostream &os, Cannon_Specs &cs)
+  friend std::ostream &operator <<(std::ostream &os, Cannon_Specs &cs)
     {
       cs.write(os);
       return(os);
@@ -212,14 +212,14 @@
   
   virtual ~Bomb_Specs();
   void createZManager(char *path);
-  void read(istream &is);
-  void write(ostream &os);
-  friend istream &operator >>(istream &is, Bomb_Specs &ws)
+  void read(std::istream &is);
+  void write(std::ostream &os);
+  friend std::istream &operator >>(std::istream &is, Bomb_Specs &ws)
     {
       ws.read(is);
       return(is);
     }
-  friend ostream &operator <<(ostream &os, Bomb_Specs &ws)
+  friend std::ostream &operator <<(std::ostream &os, Bomb_Specs &ws)
     {
       ws.write(os);
       return(os);
@@ -272,14 +272,14 @@
       strcpy(flume_flame_map,"fire");
       strcpy(flume_smoke_map,"smoke");
     }
-  void read(istream &is);
-  void write(ostream &os);
-  friend istream &operator >>(istream &is, Rocket_Specs &rs)
+  void read(std::istream &is);
+  void write(std::ostream &os);
+  friend std::istream &operator >>(std::istream &is, Rocket_Specs &rs)
     {
       rs.read(is);
       return(is);
     }
-  friend ostream &operator <<(ostream &os, Rocket_Specs &rs)
+  friend std::ostream &operator <<(std::ostream &os, Rocket_Specs &rs)
     {
       rs.write(os);
       return(os);
@@ -368,14 +368,14 @@
     {
       return (n);
     }
-  void read(istream &is);
-  void write(ostream &os);
-  friend istream &operator >>(istream &is, Weapon &wp)
+  void read(std::istream &is);
+  void write(std::ostream &os);
+  friend std::istream &operator >>(std::istream &is, Weapon &wp)
     {
       wp.read(is);
       return(is);
     }
-  friend ostream &operator <<(ostream &os, Weapon &wp)
+  friend std::ostream &operator <<(std::ostream &os, Weapon &wp)
     {
       wp.write(os);
       return(os);
@@ -393,7 +393,7 @@
       gun_specs = spcs;
     }
 
-  friend istream &operator >>(istream &is, Guns &gp);
+  friend std::istream &operator >>(std::istream &is, Guns &gp);
   int update(Flight &host_flight, Unguided_Manager *um, 
 	     Weapon_Instance *, Target *);
   int getMaxRounds();
@@ -408,7 +408,7 @@
     {
       bomb_specs = spcs;
     }
-  friend istream &operator >>(istream &is, Bombs &b);
+  friend std::istream &operator >>(std::istream &is, Bombs &b);
   int update(Flight &host_flight, Unguided_Manager *um, 
 	     Weapon_Instance *, Target *);
 };
@@ -422,7 +422,7 @@
     {
       rocket_specs = spcs;
     }
-  friend istream &operator >>(istream &is, Rockets &r);
+  friend std::istream &operator >>(std::istream &is, Rockets &r);
   int update(Flight &host_flight, Unguided_Manager *um, 
 	     Weapon_Instance *, Target *);
 };
@@ -436,7 +436,7 @@
     {
       missile_specs = spcs;
     }
-  friend istream &operator >>(istream &is, Missiles &m);
+  friend std::istream &operator >>(std::istream &is, Missiles &m);
   int update(Flight &host_flight, Unguided_Manager *um, 
 	     Weapon_Instance *, Target *);
 };
@@ -450,7 +450,7 @@
     {
       fueltank_specs = spcs;
     }
-  friend istream &operator >>(istream &is, FuelTanks &m);
+  friend std::istream &operator >>(std::istream &is, FuelTanks &m);
   int update(Flight &host_flight, Unguided_Manager *um, 
 	     Weapon_Instance *, Target *);
 };
@@ -635,8 +635,8 @@
   Weapons_Manager();
   ~Weapons_Manager();
   void read_file(char *);
-  friend istream & operator >>(istream &is, Weapons_Manager &wp);
-  void read(istream &);
+  friend std::istream & operator >>(std::istream &is, Weapons_Manager &wp);
+  void read(std::istream &);
   Weapon_List *get_list(int n);
   Weapon_List *get_list(char *);
   Weapon_Instance *build_instance_list(int n, int *cnt, char *id = NULL);
diff -urN sabre-0.2.4b/src/zview.C tmp/sabre-0.2.4b/src/zview.C
--- sabre-0.2.4b/src/zview.C	1998-12-13 10:32:45.000000000 -0500
+++ tmp/sabre-0.2.4b/src/zview.C	2008-01-31 11:42:25.000000000 -0500
@@ -39,8 +39,8 @@
  * z-buffer                                        *
  ***************************************************/
 #include <stdio.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 #include <math.h>
 #include <string.h>
 #include <limits.h>
@@ -127,7 +127,7 @@
 int Z_Node_Manager::read_file(char *path)
 {
   int result = 0;
-  ifstream infile;
+  std::ifstream infile;
   if (open_is(infile,path))
     {
       zzpath = path;
@@ -139,7 +139,7 @@
 }
 extern int shape_cnt;
 
-void Z_Node_Manager::read(istream &is)
+void Z_Node_Manager::read(std::istream &is)
 {
   char c;
   READ_TOKI('{',is,c)
@@ -190,7 +190,7 @@
     reference_shape[i].create(&(reference_info[i]));
 }
 
-istream & operator >>(istream &is, Z_Node_Manager &zm)
+std::istream & operator >>(std::istream &is, Z_Node_Manager &zm)
 {
   zm.read(is);
   return(is);
@@ -293,7 +293,7 @@
 int Z_Node_Manager::write_file(char *path)
 {
   int result = 0;
-  ofstream ofile;
+  std::ofstream ofile;
   if (open_os(ofile,path))
     {
       result = 1;
@@ -303,7 +303,7 @@
   return result;
 }
 
-void Z_Node_Manager::write(ostream &os)
+void Z_Node_Manager::write(std::ostream &os)
 {
   int i;
   os << "{\n";
@@ -337,7 +337,7 @@
   os << "}\n";
 }
 
-void Z_Node_Manager::write_params(ostream &os, int i)
+void Z_Node_Manager::write_params(std::ostream &os, int i)
 {
   if (i >= 0 && i < n_shapes)
     {
@@ -347,7 +347,7 @@
     }
 }
 
-ostream & operator <<(ostream &os, Z_Node_Manager &zm)
+std::ostream & operator <<(std::ostream &os, Z_Node_Manager &zm)
 {
   zm.write(os);
   return os;
@@ -582,7 +582,7 @@
     }
 }
 
-void Z_Viewer::write(ostream &os)
+void Z_Viewer::write(std::ostream &os)
 {
   os << "{\n";
   os << "XXXXXX\n";
@@ -602,7 +602,7 @@
   os << "}\n";
 }
 
-ostream &operator <<(ostream &os, Z_Viewer &zv)
+std::ostream &operator <<(std::ostream &os, Z_Viewer &zv)
 {
   zv.write(os);
   return os;
diff -urN sabre-0.2.4b/src/zview.h tmp/sabre-0.2.4b/src/zview.h
--- sabre-0.2.4b/src/zview.h	1998-12-13 10:32:48.000000000 -0500
+++ tmp/sabre-0.2.4b/src/zview.h	2008-01-31 10:59:26.000000000 -0500
@@ -71,15 +71,15 @@
   Z_Node_Manager();
   ~Z_Node_Manager();
 
-  friend istream & operator >>(istream &, Z_Node_Manager &);
-  void read(istream &is);
+  friend std::istream & operator >>(std::istream &, Z_Node_Manager &);
+  void read(std::istream &is);
   int read_file(char *);
   void create();
-  friend ostream & operator <<(ostream &, Z_Node_Manager &);
-  void write(ostream &os);
+  friend std::ostream & operator <<(std::ostream &, Z_Node_Manager &);
+  void write(std::ostream &os);
   int write_file(char *);
   void add(C_ShapeInfo &new_shape, shape_params &new_params);
-  void write_params(ostream &os, int);
+  void write_params(std::ostream &os, int);
   void deleteShape(int which);
   void buildShapes();
   void setId(char *);
@@ -132,8 +132,8 @@
   REAL_TYPE draw_prep(Port_3D &);
   void build_bounding_cube();
   void calc_screen_bounds(Port_3D &, R_3DPoint *);
-  friend ostream &operator <<(ostream &os, Z_Viewer &zv);
-  void write(ostream &os);
+  friend std::ostream &operator <<(std::ostream &os, Z_Viewer &zv);
+  void write(std::ostream &os);
   void setReferencePort(Port_3D *prt)
     {
       reference_port = prt;
