--- src/apq-2.1/configure.orig	2013-07-27 14:26:41.103737504 +0200
+++ src/apq-2.1/configure	2013-07-27 14:27:59.886290883 +0200
@@ -7,39 +7,13 @@
 	Database="$2"
 
 	set +eu
-	set -- $(type $Command 2>/dev/null)
+	set -- $(LC_ALL=C type $Command 2>/dev/null)
 	if [ "$2" != "is" ] ; then
 		HAVE_DB=0
 	else
 		HAVE_DB=1
 	fi
 
-	if [ $HAVE_DB -eq 0 ] ; then
-		cat >/dev/tty <<EOF
--------------------------------------------------------------------------------
-	WARNING: Unable to locate the $Command command.
-
-	This suggests that you do not have $Database installed.
-
-	Alternatively, you may need to alter your PATH variable to
-	include $Database commands. Please check your PATH
-	variable if you believe that $Database is installed on
-	your system.
--------------------------------------------------------------------------------
-
-EOF
-		echo "Is it OK to build APQ without $Database support (Yes/No)?"
-		read reply </dev/tty
-		case "$reply" in
-		Y* | y* )
-			echo "Very well then. No $Database support it is."
-			echo
-			;;
-		* )	echo "Configuration aborted. Please check your PATH" >/dev/tty
-			echo "variable, and try again." >/dev/tty
-			exit 13;;
-		esac
-	fi
 	return $HAVE_DB
 }
 
