--- configure.orig	2012-10-29 21:11:42.957716619 +0100
+++ configure	2013-07-27 14:36:14.698058864 +0200
@@ -90,35 +90,6 @@
 # system type and processor architecture type by uname.
 
 echo "Configuring SparForte..."
-sleep 1
-echo
-LOCATE="glocate"
-echo -n "checking for locate ($LOCATE)... "
-"$LOCATE" --help >/dev/null 2>/dev/null
-if [ $? -ne 0 ] ; then
-  echo "no"
-  LOCATE="locate"
-  echo -n "checking for locate ($LOCATE)... "
-  "$LOCATE" --help >/dev/null 2>/dev/null
-  if [ $? -ne 0 ] ; then
-     echo "no"
-     # freebsd 8 has different parameters
-     "$LOCATE" thisisafakepatternbecausefreebsdlocatehasnohelp >/dev/null 2>/dev/null
-  else
-     echo "yes"
-  fi
-else
-  echo "yes"
-fi
-if [ $? -ne 0 ] ; then
-   echo "Could not run locate"
-   echo
-   echo "Locate is used by this script to ensure you have all the required"
-   echo "software to build SparForte."
-   echo
-   echo "Please install GNU locate and run updatedb (as root)."
-   exit 1
-fi
 cd src
 echo -n "checking for OS (uname -s)... "
 OS=`uname -s 2>/dev/null`
@@ -325,8 +296,8 @@
 
 # Check for the Ada Run-Time Library
 
-echo -n "checking for libgnat (ldconfig -p, Linux)... "
-TMP=`/sbin/ldconfig -p 2>/dev/null | grep libgnat`
+echo -n "checking for libgnat... "
+TMP=`find /usr/lib/gcc/*-unknown-linux-gnu/*/adalib/ -name "libgnat*.so"`
 if [ -z "$TMP" ] ; then
    echo "no"
    echo -n "checking for libgnat (ldconfig -r, FreeBSD)... "
@@ -695,9 +666,7 @@
 # ADAVOX library
 
 echo
-sleep 3
 echo "Configuring third-party ADAVOX (AU/WAV Sound Library)..."
-sleep 1
 
 # /dev/dsp
 
@@ -859,12 +828,8 @@
 if [ -z "$HAS_POSTGRES" ] ; then
    PGLIBSSUB=""
 else
-   PGLIBSSUB="`pg_config --libdir` `pg_config --libs`"
-   # There is a known bug with pg_config in that it reports the compiling
-   # libraries for the PostgreSQL server, not libpg.  So it reports more
-   # libraries than we need.
-   PGLIBSSUB=`echo "$PGLIBSSUB" | sed 's/\ \-lxslt//g;s/\ \-lxml2//g;s/\ \-lkrb5//g;s/\ \-lgssapi_krb5//g;s/\ \-lcom_err//g;s/\ \-lcrypto//g;s/\ \-lcrypt//g'`
-   PGLIBSSUB=`echo "-L""$PGLIBSSUB" | sed 's/\ /\\\ /g;s/\//\\\\\//g'`
+   # pg_config is broken!
+   PGLIBSSUB="-L\/usr\/lib\ -lpq"
 fi
 if [ -z "$HAS_MYSQL" ] ; then
    MYSQLLIBSSUB=""
