--- configure.orig	2012-02-07 20:35:02.105889493 +0100
+++ configure	2012-02-07 21:11:36.943999034 +0100
@@ -88,23 +88,6 @@
 echo "Configuring SparForte..."
 sleep 1
 echo
-echo -n "checking for locate command... "
-locate --help >/dev/null 2>/dev/null
-if [ $? -ne 0 ] ; then
-   # freebsd has different parameters
-   locate thisisafakepatternbecausefreebsdlocatehasnohelp >/dev/null 2>/dev/null
-fi
-if [ $? -ne 0 ] ; then
-   echo "Could not run locate"
-   echo
-   echo "Locate is used by this script to ensure you have all the required"
-   echo "software to build SparForte."
-   echo
-   echo "Please install GNU locate and run updatedb (as root)."
-   exit 1
-else
-   echo "yes"
-fi
 cd src
 echo -n "checking for OS (uname -s)... "
 OS=`uname -s 2>/dev/null`
@@ -646,40 +629,10 @@
 
 echo "SparForte is configured"
 
-# ADAVOX library
-
-echo
-sleep 3
-echo "Configuring third-party ADAVOX (AU/WAV Sound Library)..."
-sleep 1
-
-# /dev/dsp
-
-echo
-echo -n "adavox library: checking for /dev/dsp... "
-if test ! -z "$BASH" ; then
-   test -a /dev/dsp
-else
-   test -r /dev/dsp
-fi
-if [ $? -eq 0 ] ; then
-   echo "yes"
-else
-      echo "no"
-      echo
-      echo "ADAVOX requires /dev/dsp.  Create a /dev/dsp symlink to your real audio device"
-      echo "(If you don't care about audio support, try ln -s /dev/null /dev/dsp)"
-      exit 1
-fi
-
-echo "ADAVOX library is configured"
-
 # APQ library
 
 echo
-sleep 3
 echo "Configuring third-party APQ (Database Library)..."
-sleep 1
 
 echo -n "apq library: checking for one or more database config scripts... "
 if [ ! -z "$NO_MYSQL" -a ! -z "$NO_POSTGRES" ] ; then
