--- src/main.h.orig	2008-02-02 04:02:47.000000000 +0200
+++ src/main.h	2009-05-21 12:34:54.861111456 +0300
@@ -535,7 +535,7 @@
 
 //DB// lang.c variables for DB Hub 0.4
 
-
+char bindaddr[256];
 
 /* Functions */
 int    new_human_user(int sock);
--- src/main.c.orig	2008-02-02 20:47:46.000000000 +0200
+++ src/main.c	2009-05-21 12:44:12.277722034 +0300
@@ -3868,6 +3868,7 @@
  		{ "version", no_argument, 0, 'v'},
  		{ "workdir", required_argument, 0, 'w'},
  		{ "logfile", required_argument, 0, 'l'},
+ 		{ "bind", required_argument, 0, 'b'},
  		{ "debug", optional_argument, 0, 'd'},
 #ifdef SWITCH_USER
 		{ "uid", required_argument, 0, 'u'},
@@ -4028,17 +4029,17 @@
 	snprintf(pmspammer, 50, "noone");
 
 	total_share = 0;
-
  	/* Switch to option processing routines to handle arguments */
  
 #ifdef SWITCH_USER 
- 	while ( (optres=getopt_long(argc, argv, "shvu:g:w:l:d::", long_options, &optindex))!=-1){
+ 	while ( (optres=getopt_long(argc, argv, "shvb:u:g:w:l:d::", long_options, &optindex))!=-1){
 #else
-	while ( (optres=getopt_long(argc, argv, "shvw:l:d::", long_options, &optindex))!=-1){
+	while ( (optres=getopt_long(argc, argv, "shvb:w:l:d::", long_options, &optindex))!=-1){
 #endif
  	switch (optres){
  	   case 'h':
  		printf("DB Hub, version %s\n", VERSION);
+ 		printf("  -b, --bind ip           : Bind to ip.\n");
  		printf("  -d, --debug [level]     : Debug mode. Prevents DB Hub from making itself a\n                            background daemon. Optional level ist verbosity.\n");
  		printf("  -h, --help              : Print this help and exit.\n");+ 		printf("  -v, --version           : Print version.\n");
  		printf("  -l, --logfile <logfile> : Set logfile.\n");
@@ -4079,6 +4080,9 @@
 	   case 's':
 	   	logusesyslog=1;
 		break;
+ 	   case 'b':
+		strncpy(bindaddr, optarg, (sizeof(bindaddr)-1));
+		break;
  	   case 'd':
  	   	debug=1;
  		if (optarg) verbosity=atoi(optarg);
--- src/network.c.orig	2008-02-01 01:41:36.000000000 +0200
+++ src/network.c	2009-05-21 12:46:01.171044985 +0300
@@ -377,7 +377,10 @@
    if (set_to_localhost) {
      inet_pton(AF_INET, "127.0.0.1", &hub_addr.sin_addr);
    } else {
-     hub_addr.sin_addr.s_addr = INADDR_ANY;
+    if(bindaddr)
+        inet_pton(AF_INET, bindaddr, &hub_addr.sin_addr);
+    else
+        hub_addr.sin_addr.s_addr = INADDR_ANY;
    }
    hub_addr.sin_port = htons(port);
    
@@ -498,7 +501,10 @@
      }                    
    
    hub_addr.sin_family = AF_INET;
-   hub_addr.sin_addr.s_addr = INADDR_ANY;
+   if(bindaddr)
+	inet_pton(AF_INET, bindaddr, &hub_addr.sin_addr);
+   else
+	hub_addr.sin_addr.s_addr = INADDR_ANY;
    hub_addr.sin_port = htons(port);
    memset(hub_addr.sin_zero, 0, 8);
    
