--- 13.8/common/lib/modules/fglrx/build_mod/firegl_public.c	2013-08-19 16:58:55.000000000 +0200
+++ 13.8/common/lib/modules/fglrx/build_mod/firegl_public.c	2013-09-30 18:59:02.267341585 +0200
@@ -259,7 +259,11 @@
 #endif

 #ifdef MODULE_LICENSE
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,12,0)
 MODULE_LICENSE("Proprietary. (C) 2002 - ATI Technologies, Starnberg, GERMANY");
+#else
+MODULE_LICENSE("GPL\0Proprietary. (C) 2002 - ATI Technologies, Starnberg, GERMANY");
+#endif
 #endif
 #ifdef MODULE_DEVICE_TABLE
 MODULE_DEVICE_TABLE(pci, fglrx_pci_table);
