--- a/opt/sky_linux/sky.sh	2015-05-11 04:46:03.000000000 +0300
+++ b/opt/sky_linux/sky.sh	2015-05-13 15:44:57.964647515 +0300
@@ -1,7 +1,13 @@
 #!/bin/bash
-export QT_IM_MODULE="ibus"
-export QT4_IM_MODULE="ibus"
-export XMODIFIERS="@im=ibus"
-export GTK_IM_MODULE="ibus"
-export DEPENDS="ibus, ibus-gtk, ibus-qt4, ibus-clutter"
+
+# test ibus-daemon is running in current session
+if ps -ef | grep -e "^`whoami`\b.*[i]bus-daemon\b.*x" >/dev/null
+then
+    export QT_IM_MODULE="ibus"
+    export QT4_IM_MODULE="ibus"
+    export XMODIFIERS="@im=ibus"
+    export GTK_IM_MODULE="ibus"
+    export DEPENDS="ibus, ibus-gtk, ibus-qt4, ibus-clutter"
+fi
+
 LD_LIBRARY_PATH=/opt/sky_linux/lib /opt/sky_linux/sky "$@"
