--- /home/sega/test/r54/aircrackgui-m4-read-only/DEFINES.h
+++ /home/sega/test/r54-build/aircrackgui-m4-read-only/DEFINES.h
@@ -39,23 +39,15 @@
 static const int MAX_BSSID_CON = 700;
 static const int MAX_ESSID = 500;
 static const QString BUFFER_AIRODUMP_FILENAME = "airodump.M4"; //same that in common.h
-#ifdef __i386__
-static const QString AIRODUMP_COM =     "aircrack-ng-1.1-M4/x32/airodump-ng";
-static const QString AIREPLAY_COM =     "aircrack-ng-1.1-M4/x32/aireplay-ng";
-static const QString AIRCRACK_COM =     "aircrack-ng-1.1-M4/x32/aircrack-ng";
-static const QString PACKETFORGE_COM =  "aircrack-ng-1.1-M4/x32/packetforge-ng";
-static const QString AIRMON_COM =       "aircrack-ng-1.1-M4/x32/airmon-ng";
-#else
-static const QString AIRODUMP_COM =     "aircrack-ng-1.1-M4/x64/airodump-ng";
-static const QString AIREPLAY_COM =     "aircrack-ng-1.1-M4/x64/aireplay-ng";
-static const QString AIRCRACK_COM =     "aircrack-ng-1.1-M4/x64/aircrack-ng";
-static const QString PACKETFORGE_COM =  "aircrack-ng-1.1-M4/x64/packetforge-ng";
-static const QString AIRMON_COM =       "aircrack-ng-1.1-M4/x64/airmon-ng";
-#endif
+static const QString AIRODUMP_COM =     "/usr/sbin/airodump-ng";
+static const QString AIREPLAY_COM =     "/usr/sbin/aireplay-ng";
+static const QString AIRCRACK_COM =     "/usr/bin/aircrack-ng";
+static const QString PACKETFORGE_COM =  "/usr/bin/packetforge-ng";
+static const QString AIRMON_COM =       "/usr/sbin/airmon-ng";
 
 //HAVE TO BE INSTALLED ON THE SYSTEM. Because depending of the version of linux, macchanger have diferent
 //'folder or paths to acces'
-static const QString MACCHANGER_COM = "macchanger";
+static const QString MACCHANGER_COM = "/usr/bin/macchanger";
 static const QString CAPTURE_FOLDER = "captures/";
 static const QString ARP_FOLDER = "arp_replays/";
 static const QString FRAG_FOLDER = "frag_caps/";
