diff --git a/python-src/Parser/asdl_c.py b/python-src/Parser/asdl_c.py
index 1cd77aa..99554d3 100755
--- a/python-src/Parser/asdl_c.py
+++ b/python-src/Parser/asdl_c.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 """Generate C code from an ASDL description."""
 
 # TO DO
@@ -1019,7 +1019,7 @@ class ObjVisitor(PickleVisitor):
             self.emit("case %s:" % t.name, 2)
             self.emit("Py_INCREF(%s_singleton);" % t.name, 3)
             self.emit("return %s_singleton;" % t.name, 3)
-        self.emit("default:" % name, 2)
+        self.emit("default:", 2)
         self.emit('/* should never happen, but just in case ... */', 3)
         code = "PyErr_Format(PyExc_SystemError, \"unknown %s found\");" % name
         self.emit(code, 3, reflow=False)
