diff -aur ShutdownApplet@shelley-pristine/applet.js ShutdownApplet@shelley-new/applet.js
--- ShutdownApplet@shelley-pristine/applet.js	2012-04-16 09:12:50.000000000 -0300
+++ ShutdownApplet@shelley-new/applet.js	2013-05-27 19:24:18.025412699 -0400
@@ -37,15 +37,15 @@
             });                                                                         
 
             this.menu.addAction(_("Suspend"), function(event) {
-                Util.spawnCommandLine("dbus-send --print-reply --system --dest=org.freedesktop.UPower /org/freedesktop/UPower org.freedesktop.UPower.Suspend");
+                Util.spawnCommandLine("systemctl suspend");
             });
 	    
               this.menu.addAction(_("Hibernate"), function(event) {
-                Util.spawnCommandLine("dbus-send --print-reply --system --dest=org.freedesktop.UPower /org/freedesktop/UPower org.freedesktop.UPower.Suspend");
+                Util.spawnCommandLine("systemctl hibernate");
             });  
             
              this.menu.addAction(_("Restart"), function(event) {
-                Util.spawnCommandLine("dbus-send --system --print-reply --system --dest=org.freedesktop.ConsoleKit /org/freedesktop/ConsoleKit/Manager org.freedesktop.ConsoleKit.Manager.Restart");
+                Util.spawnCommandLine("systemctl reboot");
             });  
             
         this.menu.addAction(_("Log Out"), function(event) {
@@ -53,7 +53,7 @@
             });
             
 	     this.menu.addAction(_("Shutdown"), function(event) {
-                Util.spawnCommandLine("dbus-send --system --print-reply --system --dest=org.freedesktop.ConsoleKit /org/freedesktop/ConsoleKit/Manager org.freedesktop.ConsoleKit.Manager.Stop");
+                Util.spawnCommandLine("systemctl poweroff");
             });                        
         }
         catch (e) {
