--- setup.py.orig	2010-06-21 18:20:13.000000000 -0500
+++ setup.py	2010-06-21 18:21:34.704278540 -0500
@@ -60,8 +60,8 @@
 _libpci = Extension(
     'pylibpci._libpci', ['pylibpci/_libpci.pyx', ],
     libraries=['pci', ],
-    extra_compile_args=['-Wall', '-Werror', ],
-    extra_link_args=['-Wall', '-Werror', ],
+    extra_compile_args=['-Wall', ],
+    extra_link_args=['-Wall', '-Werror', ],
 )
 
 setup(
