diff -aur a/gdm3setup-daemon.py b/gdm3setup-daemon.py
--- a/gdm3setup-daemon.py	2012-05-27 01:51:32.000000000 +0200
+++ b/gdm3setup-daemon.py	2012-10-19 17:36:08.809101641 +0200
@@ -77,54 +77,6 @@
 
 	return dbus_address,dbus_pid
 
-def Get_Shell_theme():
-	shell_theme = 'Adwaita'
-	ifile = file('/usr/share/gnome-shell/theme/gnome-shell.css',"r")
-	lines = ifile.readlines()
-	for i in range(len(lines)) :
-		line = lines[i].strip()
-		if line[0:2] == "/*" and line[len(line)-2:len(line)] == "*/" :
-			comment = line[2:len(line)-2]
-			if comment[0:6] == "theme=" :
-				shell_theme = comment[6:len(comment)]
-
-	return shell_theme
-
-def Set_Shell_theme(new_theme) :
-	theme = Get_Shell_theme()
-	if theme == "Adwaita" and new_theme != "Adwaita":
-		if os.path.isdir('/usr/share/themes/'+new_theme+'/gnome-shell') :
-			if not os.path.islink('/usr/share/gnome-shell/theme') :
-				if os.path.isdir('/usr/share/gnome-shell/theme.original/') :
-					shutil.rmtree('/usr/share/gnome-shell/theme.original/')
-				shutil.move('/usr/share/gnome-shell/theme/','/usr/share/gnome-shell/theme.original/')
-				shutil.copytree('/usr/share/themes/'+new_theme+'/gnome-shell','/usr/share/gnome-shell/theme/')
-			elif os.path.isdir('/usr/share/gnome-shell/theme.original/') :
-				os.remove('/usr/share/gnome-shell/theme')
-				shutil.copytree('/usr/share/themes/'+new_theme+'/gnome-shell','/usr/share/gnome-shell/theme/')
-		ifile = file('/usr/share/gnome-shell/theme/gnome-shell.css','r')
-		lines = ifile.readlines()
-		lines.insert(0,"/*theme="+new_theme+"*/\n")
-		ifile.close()
-		ofile = file('/usr/share/gnome-shell/theme/gnome-shell.css','w')
-		ofile.writelines(lines)
-		ofile.close()
-
-	if theme != "Adwaita" and new_theme == "Adwaita" and os.path.isdir('/usr/share/gnome-shell/theme.original/') :
-		shutil.rmtree('/usr/share/gnome-shell/theme/')
-		shutil.move('/usr/share/gnome-shell/theme.original/','/usr/share/gnome-shell/theme/')
-
-	if theme != "Adwaita" and new_theme != "Adwaita" and os.path.isdir('/usr/share/themes/'+new_theme+'/gnome-shell') :
-		shutil.rmtree('/usr/share/gnome-shell/theme/')
-		shutil.copytree('/usr/share/themes/'+new_theme+'/gnome-shell','/usr/share/gnome-shell/theme/')
-		ifile = file('/usr/share/gnome-shell/theme/gnome-shell.css','r')
-		lines = ifile.readlines()
-		lines.insert(0,"/*theme="+new_theme+"*/\n")
-		ifile.close()
-		ofile = file('/usr/share/gnome-shell/theme/gnome-shell.css','w')
-		ofile.writelines(lines)
-		ofile.close()
-
 class GDM3SetupDBusService(dbus.service.Object):
 	def __init__(self):
 		bus=dbus.SystemBus()
@@ -148,11 +100,8 @@
 					sender_keyword='sender', connection_keyword='connexion')
 	def SetUI(self,name,value,sender=None, connexion=None):
 		if self.policykit_test(sender,connexion,'apps.nano77.gdm3setup.set') :
-			if name!='SHELL_THEME':
-				bus_adrress , bus_pid = Get_Bus()
-				subprocess.call('su - '+GDM_USER_NAME+' -s /bin/bash -c "LANG='+LANG+' DBUS_SESSION_BUS_ADDRESS='+bus_adrress+' DBUS_SESSION_BUS_PID='+bus_pid+' set_gdm.sh -n '+name+' -v '+"'"+value+"'"+'"',shell=True)
-			else :
-				Set_Shell_theme(value)
+			bus_adrress , bus_pid = Get_Bus()
+			subprocess.call('su - '+GDM_USER_NAME+' -s /bin/bash -c "LANG='+LANG+' DBUS_SESSION_BUS_ADDRESS='+bus_adrress+' DBUS_SESSION_BUS_PID='+bus_pid+' set_gdm.sh -n '+name+' -v '+"'"+value+"'"+'"',shell=True)
 
 			return "OK"
 		else :
@@ -165,7 +114,6 @@
 		settings = ifile.readlines()
 		ifile.close()
 		os.remove("/tmp/GET_GDM")
-		settings.append("SHELL='"+Get_Shell_theme()+"'\n")
 		return settings
 
 	@dbus.service.method('apps.nano77.gdm3setup','bsbi','s',sender_keyword='sender', connection_keyword='connexion')
diff -aur a/get_gdm.sh b/get_gdm.sh
--- a/get_gdm.sh	2012-09-02 18:07:51.118359000 +0200
+++ b/get_gdm.sh	2012-10-19 17:32:35.655588425 +0200
@@ -9,6 +9,8 @@
 
 echo -n "GTK=" > /tmp/GET_GDM
 gsettings get org.gnome.desktop.interface gtk-theme >> /tmp/GET_GDM
+echo -n "SHELL=" >> /tmp/GET_GDM
+gsettings get org.gnome.shell theme-name >> /tmp/GET_GDM
 echo -n "ICON=" >> /tmp/GET_GDM
 gsettings get org.gnome.desktop.interface icon-theme >> /tmp/GET_GDM
 echo -n "FONT=" >> /tmp/GET_GDM
diff -aur a/set_gdm.sh b/set_gdm.sh
--- a/set_gdm.sh	2012-10-10 18:48:49.964897000 +0200
+++ b/set_gdm.sh	2012-10-19 17:33:15.132328029 +0200
@@ -63,6 +63,10 @@
 				echo "$parameter_name = $parameter_value"
 				gsettings set org.gnome.desktop.interface gtk-theme "$parameter_value"
 				;;
+			SHELL_THEME)
+				echo "$parameter_name = $parameter_value"
+				gsettings set org.gnome.shell theme-name "$parameter_value"
+				;;
 			FONT)
 				echo "$parameter_name = $parameter_value"
 				gsettings set org.gnome.desktop.interface font-name "$parameter_value"
