--- TextSTAT.pyw	2009-12-29 11:48:40.000000000 -0500
+++ TextSTAT.pyw_new	2010-01-16 17:38:56.000000000 -0500
@@ -20,7 +20,7 @@ except:
 # Dies scheint besser zu funktionieren:
 SkriptDir = os.path.dirname(sys.argv[0])
 ImgDir = os.path.join(SkriptDir, 'Images')
-cfgFile = os.path.join(SkriptDir, 'TextSTAT.ini')
+cfgFile = os.path.join(os.path.expanduser("~"), '.textstatrc')
 
 try:
     f = open(cfgFile)
