--- Bombermaaan_1.4.0.627_20081018_src.orig/Bombermaaan/winreplace.cpp	2008-10-18 17:26:05.000000000 +0600
+++ Bombermaaan_1.4.0.627_20081018_src/Bombermaaan/winreplace.cpp	2012-05-13 15:42:40.851506643 +0600
@@ -71,7 +71,7 @@
 #endif
 
 long _findfirst(const char *pattern, _finddata_t *findData) {
-	char *backslash = NULL;				// backslash position
+	const char *backslash = NULL;				// backslash position
 	const char *searchPattern = NULL;	// actual search pattern
 	char *directory = NULL;				// directory
 	char *pos;
