diff --git a/gtk/gtkdialog.c b/gtk/gtkdialog.c
index d95be71..f769a48 100644
--- a/gtk/gtkdialog.c
+++ b/gtk/gtkdialog.c
@@ -273,45 +273,15 @@ add_cb (GtkContainer *container,
 }
 
 static void
-update_title (GObject    *dialog,
-              GParamSpec *pspec,
-              GtkWidget  *label)
-{
-  const gchar *title;
-
-  title = gtk_window_get_title (GTK_WINDOW (dialog));
-  gtk_label_set_label (GTK_LABEL (label), title);
-  gtk_widget_set_visible (label, title && title[0]);
-}
-
-static void
 apply_use_header_bar (GtkDialog *dialog)
 {
   GtkDialogPrivate *priv = dialog->priv;
 
   gtk_widget_set_visible (priv->action_area, !priv->use_header_bar);
   gtk_widget_set_visible (priv->headerbar, priv->use_header_bar);
-  if (!priv->use_header_bar)
-    {
-      GtkWidget *box = NULL;
-
-      if (gtk_window_get_type_hint (GTK_WINDOW (dialog)) == GDK_WINDOW_TYPE_HINT_DIALOG)
-        {
-          GtkWidget *label;
-
-          box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
-          gtk_widget_show (box);
-          gtk_widget_set_size_request (box, -1, 16);
-          label = gtk_label_new ("");
-          gtk_widget_set_margin_top (label, 6);
-          gtk_widget_set_margin_bottom (label, 6);
-          gtk_style_context_add_class (gtk_widget_get_style_context (label), "title");
-          gtk_box_set_center_widget (GTK_BOX (box), label);
-          g_signal_connect_object (dialog, "notify::title", G_CALLBACK (update_title), label, 0);
-        }
 
-      gtk_window_set_titlebar (GTK_WINDOW (dialog), box);
-    }
+  if (!priv->use_header_bar)
+    gtk_window_set_titlebar (GTK_WINDOW (dialog), NULL);
   if (priv->use_header_bar)
     g_signal_connect (priv->action_area, "add", G_CALLBACK (add_cb), dialog);
 }
