diff -crB -x '*.o' -x '*.Po' -x Makefile openeuclide/src/MyArchive.cc openeuclide-patched/src/MyArchive.cc
*** openeuclide/src/MyArchive.cc	2006-11-02 00:13:02.000000000 +0100
--- openeuclide-patched/src/MyArchive.cc	2009-03-23 12:50:34.969361759 +0100
***************
*** 20,25 ****
--- 20,26 ----
  
  #include "MyArchive.h"
  #include <iostream>
+ #include <string.h>
  
  CMyArchive::CMyArchive() {
  
diff -crB -x '*.o' -x '*.Po' -x Makefile openeuclide/src/euclideConf.cc openeuclide-patched/src/euclideConf.cc
*** openeuclide/src/euclideConf.cc	2006-11-02 00:13:02.000000000 +0100
--- openeuclide-patched/src/euclideConf.cc	2009-03-23 11:16:45.083248012 +0100
***************
*** 2,7 ****
--- 2,8 ----
  #include <envz.h>
  #include <iostream>
  #include <glib.h>
+ #include <stdlib.h>
  
  #define MAX_CONF_LEN 8000
  
diff -crB -x '*.o' -x '*.Po' -x Makefile openeuclide/src/euclideMain.cc openeuclide-patched/src/euclideMain.cc
*** openeuclide/src/euclideMain.cc	2006-11-02 00:13:02.000000000 +0100
--- openeuclide-patched/src/euclideMain.cc	2009-03-23 12:49:42.166023497 +0100
***************
*** 21,26 ****
--- 21,27 ----
  #include <gdk/gdkkeysyms.h>
  #include <string>
  #include <iostream>
+ #include <string.h>
  #include "diaBase.h"
  #include "euclideMain.h"
  #include "euclideView.h"
