From db0f0197a2e1740ec2e13cf254d0e6a968bf6db4 Mon Sep 17 00:00:00 2001
From: "Arnel A. Borja" <kyoushuu@yahoo.com>
Date: Thu, 13 Sep 2012 22:20:49 +0800
Subject: [PATCH 3/4] gio/tests: Compile gdbus-proxy only when target is Unix

---
 gio/tests/Makefile.am |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/gio/tests/Makefile.am b/gio/tests/Makefile.am
index e352383..68ebcf0 100644
--- a/gio/tests/Makefile.am
+++ b/gio/tests/Makefile.am
@@ -71,7 +71,6 @@ TEST_PROGS +=			\
 	gdbus-connection-loss	\
 	gdbus-connection-slow	\
 	gdbus-names		\
-	gdbus-proxy		\
 	gdbus-proxy-threads	\
 	gdbus-proxy-well-known-name \
 	gdbus-introspection	\
@@ -81,6 +80,12 @@ TEST_PROGS +=			\
 	gdbus-bz627724		\
 	gmenumodel		\
 	$(NULL)
+
+if OS_UNIX
+TEST_PROGS +=			\
+	gdbus-proxy		\
+	$(NULL)
+endif
 endif
 
 
-- 
1.7.10.4
