--- launchy/src/src.pro	2012-06-25 07:13:13.173505684 -0400
+++ launchy/src/src.pro.new	2012-06-25 07:20:35.760185788 -0400
@@ -65,6 +65,7 @@
         platform_base_hotkey.h \
         platform_base_hottrigger.h
     PREFIX = /usr
+    LIBS += -lX11
     DEFINES += SKINS_PATH=\\\"$$PREFIX/share/launchy/skins/\\\" \
         PLUGINS_PATH=\\\"$$PREFIX/lib/launchy/plugins/\\\" \
         PLATFORMS_PATH=\\\"$$PREFIX/lib/launchy/\\\"
