--- Makefile.orig	2011-02-23 18:10:10.000000000 -0500
+++ Makefile	2011-02-23 18:12:23.000000000 -0500
@@ -1,6 +1,7 @@
 PYVERSION = python2.7
 SETUP = ./setup.py
 FIND = find
+DESTDIR ?= /
 
 .PHONY: dist install test install-local test-local uninstall-local clean dangerously-clean
 
@@ -18,8 +19,8 @@ uninstall-local:
 test-local: uninstall-local install-local
 	$(PYVERSION) $(HOME)/.local/lib/$(PYVERSION)/site-packages/lib3to2/tests/test_all_fixers.py
 
-install:
-	$(SETUP) install
+install: 
+	$(SETUP) install --root=$(DESTDIR) 
 
 test:
 	lib3to2/tests/test_all_fixers.py
