diff -urN razercfg-0.20.orig/razerd/CMakeLists.txt razercfg-0.20/razerd/CMakeLists.txt
--- razercfg-0.20.orig/razerd/CMakeLists.txt	2013-04-26 09:56:45.000000000 +0200
+++ razercfg-0.20/razerd/CMakeLists.txt	2013-06-04 10:26:13.376802172 +0200
@@ -8,4 +8,4 @@
 include_directories("${razer_SOURCE_DIR}/librazer")
 
 target_link_libraries(razerd razer)
-install(TARGETS razerd DESTINATION sbin)
+install(TARGETS razerd DESTINATION bin)
diff -urN razercfg-0.20.orig/razerd.initscript razercfg-0.20/razerd.initscript
--- razercfg-0.20.orig/razerd.initscript	2013-04-26 09:56:45.000000000 +0200
+++ razercfg-0.20/razerd.initscript	2013-06-04 10:26:13.620137641 +0200
@@ -9,7 +9,7 @@
 # Description:       Razer device state daemon
 ### END INIT INFO
 
-export PATH="/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin"
+export PATH="/usr/local/bin:/usr/bin"
 NAME=razerd
 DAEMON="$(which $NAME)"
 PIDFILE="/run/razerd/razerd.pid"
