diff -c razercfg-0.16/ui/pyrazer.py razercfg-0.16/ui-python2/pyrazer.py
*** razercfg-0.16/ui/pyrazer.py	2011-07-11 21:13:43.000000000 +1000
--- razercfg-0.16/ui-python2/pyrazer.py	2011-07-21 20:12:02.503138783 +1000
***************
*** 1,4 ****
! #!/usr/bin/env python
  """
  #   Razer device configuration
  #   High level user interface library
--- 1,4 ----
! #!/usr/bin/env python2
  """
  #   Razer device configuration
  #   High level user interface library
diff -c razercfg-0.16/ui/qrazercfg razercfg-0.16/ui-python2/qrazercfg
*** razercfg-0.16/ui/qrazercfg	2011-07-11 21:13:43.000000000 +1000
--- razercfg-0.16/ui-python2/qrazercfg	2011-07-21 20:12:10.063097826 +1000
***************
*** 1,4 ****
! #!/usr/bin/env python
  #
  #   Razer device QT configuration tool
  #
--- 1,4 ----
! #!/usr/bin/env python2
  #
  #   Razer device QT configuration tool
  #
diff -c razercfg-0.16/ui/razercfg razercfg-0.16/ui-python2/razercfg
*** razercfg-0.16/ui/razercfg	2011-07-11 21:13:43.000000000 +1000
--- razercfg-0.16/ui-python2/razercfg	2011-07-21 20:12:16.799727999 +1000
***************
*** 1,4 ****
! #!/usr/bin/env python
  #
  #   Razer device commandline configuration tool
  #
--- 1,4 ----
! #!/usr/bin/env python2
  #
  #   Razer device commandline configuration tool
  #
