diff -NaruB grid-updates-1.1.3-orig/share/tahoe.css.patched19 grid-updates-1.1.3/share/tahoe.css.patched19
--- grid-updates-1.1.3-orig/share/tahoe.css.patched19	2012-06-03 13:25:20.000000000 -0400
+++ grid-updates-1.1.3/share/tahoe.css.patched19	2012-07-05 02:15:38.093238331 -0400
@@ -22,7 +22,7 @@
 }
 th { 
   color: white;
-  background-color: #58a1c3;
+  background-color: #3289b4;
 }
 td {
     padding: .3em .3em;
@@ -87,54 +87,48 @@
   font-size: 60%;
   border: .2em solid #3289b4;
   float: right;
-  width: 40%;
   margin: 0 0 .5em .5em;
-  padding: 3px;
+  width: 42%;
 }
 #this-client .nodeid { font-size: inherit; }
 #this-client h2 {
   text-align: center;
   background: #3289b4;
-  color: white;
   margin: -2px -2px 0 -2px; /* matches padding */
   padding: .3em;
 }
 #this-client table { 
   font-size: inherit;
   margin: 0 -3px -3px -3px; /* matches padding */
+  border: none;
 }
 #this-client td > ul {
-  list-style-type: outside;
-  margin: 0 0 0 2.3em;
-  padding-left: 0;
+  padding-left: 10;
 }
 
 #grid-updates {
-  width: 55%;
+  width: 56%;
   font-size: 60%;
-  color: white;
   text-align: center;
-  background: white;
-  border: .3em solid #3289b4;
+  border: .2em solid #3289b4;
 }
 #grid-updates h2 {
   background: #3289b4;
-  border: 1px solid white;
-  margin: 0;
+  margin: -2px -2px 0 -2px; /* matches padding */
   padding: .3em;
 }
 div.grid-updates{
-  background: #58a1c3;
-  border: 1px solid white;
-  font-family: sans-serif;
+  background: #3289b4;
+  margin: 0 -2px -2px -2px;  /*matches padding */
+  border: 1px solid #3289b4;
 }
 a.grid-updates {
-  color: white;
+  color: black;
+  padding: .2em;
 }
 iframe#grid-updates {
-  background-color: white;
   width: 98%;
-  height: 25%;
+  height: 52%;
   border: none;
 }
 /* End grid-updates news section */
diff -NaruB grid-updates-1.1.3-orig/share/welcome.xhtml.patched19 grid-updates-1.1.3/share/welcome.xhtml.patched19
--- grid-updates-1.1.3-orig/share/welcome.xhtml.patched19	2012-07-05 01:57:08.364138413 -0400
+++ grid-updates-1.1.3/share/welcome.xhtml.patched19	2012-07-05 02:16:40.245972027 -0400
@@ -6,42 +6,42 @@
   <link href="/static/NEWS.atom" type="application/atom+xml" rel="alternate" title="grid-updates NEWS Atom Feed" />
 </head><body>
 
-<h1>Welcome to Tahoe-LAFS!</h1>
+<h1>Welcome to Tahoe-LAFS</h1>
+
 
 <div class="section" id="this-client">
   <h2>This Client</h2>
-
-  <p>
-    <a href="status/">Recent Uploads and Downloads</a>,
-    <a href="statistics">Operational Statistics</a>
-  </p>
-
   <table class="node-info table-headings-left">
-    <tr><th>My nickname:</th> <td class="nickname mine" n:render="string" n:data="my_nickname" /></tr>
-    <tr><th>My nodeid:</th> <td class="nodeid mine data-chars" n:render="string" n:data="my_nodeid" /></tr>
-    <tr><th>My versions:</th> <td n:render="string" n:data="version" /></tr>
-    <tr><th>Tahoe-LAFS code imported from:</th> <td n:render="string" n:data="import_path" /></tr>
-    <tr><th>Services running:</th> <td n:render="services" /></tr>
+    <tr><th>Nickname:</th> <td class="nickname mine" n:render="string" n:data="my_nickname" /></tr>
+    <tr><th>Nodeid:</th> <td class="nodeid mine data-chars" n:render="string" n:data="my_nodeid" /></tr>
+    <tr><th>Versions:</th> <td n:render="string" n:data="version" /></tr>
+    <tr><th>Tahoe-LAFS module location:</th> <td n:render="string" n:data="import_path" /></tr>
+    <tr><th>Services:</th> <td n:render="services" /></tr>
   </table>
+  <p>
+    <center>
+      <a href="status/">Recent Uploads and Downloads</a>,
+      <a href="statistics">Operational Statistics</a>
+	</center>
+  </p>
 </div>
 
+
 <div id="controls">
 <div class="section" id="grid-updates">
-       <h2>Grid News</h2>
-       <iframe id="grid-updates" scrolling="auto" name="grid-updates-news" src="static/NEWS.html" />
-       <div class="grid-updates">
-               Please note: the above are <a class="grid-updates" href="static/NEWS.html">news</a>
-               retrieved by <a class="grid-updates"
-               href="/uri/URI:DIR2-RO:hgh5ylzzj6ey4a654ir2yxxblu:hzk3e5rbsefobeqhliytxpycop7ep6qlscmw4wzj5plicg3ilotq">grid-updates</a>.
-               You need to run it regularly to update these news.
-       </div>
+  <h2>Grid News</h2>
+    <iframe id="grid-updates" scrolling="auto" name="grid-updates-news" src="static/NEWS.html" />
+      <div class="grid-updates">
+        Note: the above <a class="grid-updates" href="static/NEWS.html">news</a> is retrieved by 
+        <a class="grid-updates" href="/uri/URI:DIR2-RO:hgh5ylzzj6ey4a654ir2yxxblu:hzk3e5rbsefobeqhliytxpycop7ep6qlscmw4wzj5plicg3ilotq">grid-updates</a>.
+        You need to run it regularly to recieve updates.
+      </div>
 </div>
-<!-- End grid-updates patch -->
 
-<h2>Controls</h2>
 
+<br />
+<h2>Controls</h2>
   <p>There are also controls for each directory on that directory's page.</p>
-
   <div n:render="mkdir_form" />
   <div n:render="view_form" />
   <div n:render="upload_form" />
@@ -48,8 +48,9 @@
   <div n:render="download_form" />
 </div>
 
-<h2>Connected Introducer(s)</h2>
 
+<br />
+<h2>Connected Introducers</h2>
 <div>
 <table n:render="sequence" n:data="introducers">
   <tr n:pattern="header">
@@ -68,16 +69,10 @@
 </div>
 
 
-
+<br />
 <div class="section" id="grid">
   <h2>Status of the Storage Grid</h2>
 
-  <div>
-    <n:attr name="class">connected-<n:invisible n:render="string" n:data="connected_to_helper" /></n:attr>
-    <div>Helper: <span n:render="string" n:data="helper_furl" /></div>
-    <div>Connected to helper?: <span n:render="string" n:data="connected_to_helper" /></div>
-  </div>
-
   <p>Connected to <span n:render="string" n:data="connected_storage_servers" />
      of <span n:render="string" n:data="known_storage_servers" /> known storage servers:</p>
 
@@ -109,15 +104,31 @@
       <tr n:pattern="empty"><td>no peers!</td></tr>
     </table>
   </div>
+
+  <div>
+    <n:attr name="class">connected-<n:invisible n:render="string" n:data="connected_to_helper" /></n:attr>
+    <div>Helper: <span n:render="string" n:data="helper_furl" /></div>
+    <div>Connected to helper?: <span n:render="string" n:data="connected_to_helper" /></div>
+  </div>
+
 </div>
 
+
+<br />
 <div class="section" id="other-resources">
   <h2>Other Resources</h2>
 
-  <div>Please visit the <a target="_blank" href="https://tahoe-lafs.org/">Tahoe-LAFS home page</a> for
-  code updates and bug reporting.</div>
+  <div><li>
+	The <a href="provisioning">provisioning tool</a> and <a
+    href="reliability">reliability calculator</a> may also be useful.
+  </li></div>
+
+  <div><li>
+	Please visit the <a target="_blank" href="https://tahoe-lafs.org/">Tahoe-LAFS home page</a> for
+    code updates and bug reporting.
+  </li></div>
 
-  <div n:render="incident_button" />
+  <p><div n:render="incident_button" /></p>
 </div>
 
 </body></html>
