diff -ru rev-570/sc/cam.c sc-build/cam.c
--- rev-570/sc/cam.c	2013-07-07 14:30:54.464044999 +1200
+++ sc-build/cam.c	2013-07-07 14:30:57.173044939 +1200
@@ -824,6 +824,7 @@
     prg.sid=-1;
     idleTime.Set();
     prg.pids.Clear();
+    prg.caDescr.Clear();
     trigger=true;
     }
   dataMutex.Unlock();
@@ -957,7 +958,7 @@
 
     case 0:
       StopEcm();
-      if(IsIdle()) { mode=-1; break; }
+      if(filterSid<0 || IsIdle()) { mode=-1; break; }
 
       dolog=LOG_COUNT;
       NewEcm();
diff -ru rev-570/sc/contrib/sasc-ng/configure sc-build/contrib/sasc-ng/configure
--- rev-570/sc/contrib/sasc-ng/configure	2013-07-07 14:30:54.466044999 +1200
+++ sc-build/contrib/sasc-ng/configure	2013-07-07 14:30:57.175044939 +1200
@@ -79,6 +79,7 @@
   echo "                           long: Try all known optimizations"
   echo "                           no: Don't do any optimizations"
   echo "  --ffdecsa_mode=<val>     use <val> optimization"
+  echo "                           like  --ffdecsa_mode=PARALLEL_128_SSE"
   echo "  -cxx=<c++ compiler>      command for C++ compilation (default: g++)"
   exit 0
 }
@@ -207,9 +208,17 @@
    echo "Choosing PARALLEL_MODE = ${MAX_MODE}"
    echo "FFDECSA_OPTS = \"$FLAGS\" PARALLEL_MODE=${MAX_MODE} COMPILER=$CXX" >> config.mak
 else
-   if test "x$ffdecsa_flags" != "x"; then
-     echo "FFDECSA_OPTS = \"$ffdecsa_flags\" PARALLEL_MODE=${MAX_MODE} COMPILER=$CXX" >> config.mak
-   elif test "x$MAX_MODE" != "xPARALLEL_32_INT"; then
+   if ! test -f /proc/cpuinfo; then
+     echo "\nYou should not set '--ffdecsa_mode='\n\nAborting...\n"
+     rm -rf "${TMPDIR}"
+     exit 1
+   fi
+   get_cpu_optimization
+   if test "x$MAX_MODE" != "xPARALLEL_32_INT"; then
+     echo "Using PARALLEL_MODE = ${MAX_MODE}"
+     echo "FFDECSA_OPTS = \"$FLAGS\" PARALLEL_MODE=${MAX_MODE} COMPILER=$CXX" >> config.mak
+   else
+     echo "Using PARALLEL_MODE = ${MAX_MODE}"
      echo "FFDECSA_OPTS = PARALLEL_MODE=${MAX_MODE} COMPILER=$CXX" >> config.mak
    fi
 fi
@@ -221,6 +230,9 @@
   elif test -e "${dvb_path}/linux/include/linux/dvb/frontend.h"; then
     echo "DVB_DIR=${dvb_path}/linux" >> config.mak
     echo "Using DVB_DIR: ${dvb_path}/linux"
+  elif test -e "${dvb_path}/include/uapi/linux/dvb/frontend.h"; then
+    echo "DVB_DIR=${dvb_path}" >> config.mak
+    echo "Using DVB_DIR: ${dvb_path}"
   else
     echo "Could not find DVB headers within $dvb_path"
   fi
@@ -229,3 +241,4 @@
 echo "CXX=$CXX" >> config.mak
 date >> config.log
 echo "	$0 $*" >> config.log
+
diff -ru rev-570/sc/contrib/sasc-ng/dvblb_plugins/plugin_cam.c sc-build/contrib/sasc-ng/dvblb_plugins/plugin_cam.c
--- rev-570/sc/contrib/sasc-ng/dvblb_plugins/plugin_cam.c	2013-07-07 14:30:54.467044999 +1200
+++ sc-build/contrib/sasc-ng/dvblb_plugins/plugin_cam.c	2013-07-07 14:30:57.176044939 +1200
@@ -260,6 +260,7 @@
 
   if (sidmsg->calen == 0) {
     free_sidmsg(sidmsg);
+    msg->type = MSG_PROCESSED;
     return;
   }
   for(ch=Channels.First(); ch; ch=Channels.Next(ch)) {
@@ -280,6 +281,7 @@
         }
       }
       free_sidmsg(sidmsg);
+      msg->type = MSG_PROCESSED;
       return;
     }
   } else {
diff -ru rev-570/sc/contrib/sasc-ng/dvblb_plugins/plugin_ffdecsa.c sc-build/contrib/sasc-ng/dvblb_plugins/plugin_ffdecsa.c
--- rev-570/sc/contrib/sasc-ng/dvblb_plugins/plugin_ffdecsa.c	2013-07-07 14:30:54.467044999 +1200
+++ sc-build/contrib/sasc-ng/dvblb_plugins/plugin_ffdecsa.c	2013-07-07 14:30:57.176044939 +1200
@@ -123,11 +123,13 @@
           pthread_mutex_unlock(&csa->state_lock);
           return;
         }
+        pthread_mutex_lock(&csa->keylock);
         if(! csa->keyindex[index].valid) {
           csa->keyindex[index].valid = 1;
           csa->keyindex[index].status = 0;
           csa->keyindex[index].queued = 0;
         }
+        pthread_mutex_unlock(&csa->keylock);
         pop_entry_from_queue_l(pid_ll, &pidmap_empty_queue, struct pid, &list_lock);
         pid_ll->pid = pid;
         pid_ll->index = index;
@@ -146,7 +148,9 @@
           ll_find_elem(pid_ll, csa->pid_map, index, index, struct pid);
           if(pid_ll == NULL) {
             //no valid pids on this index
+            pthread_mutex_lock(&csa->keylock);
             csa->keyindex[index].status = 0;
+            pthread_mutex_unlock(&csa->keylock);
             if(list_empty(&csa->pid_map)) {
               //state = ENCRYPTED_NOT_READY;
               csa->state = NOT_ENCRYPTED;
diff -ru rev-570/sc/contrib/sasc-ng/dvbloopback/module/config-dvb/Makefile sc-build/contrib/sasc-ng/dvbloopback/module/config-dvb/Makefile
--- rev-570/sc/contrib/sasc-ng/dvbloopback/module/config-dvb/Makefile	2013-07-07 14:30:54.468044999 +1200
+++ sc-build/contrib/sasc-ng/dvbloopback/module/config-dvb/Makefile	2013-07-07 14:30:57.177044939 +1200
@@ -4,7 +4,7 @@
 
 EXTRA_CFLAGS ?= -Idrivers/media/dvb/dvb-core/ -I$(PWD)
 
-BUILD_DIR := /lib/modules/$(shell uname -r)/build
+BUILD_DIR ?= /lib/modules/$(shell uname -r)/build
 
 all: clean
 	make -C $(BUILD_DIR) M=$(PWD) V=1 modules
Only in sc-build/contrib/sasc-ng/dvbloopback/module: config_dvb_dvbdir.pl
diff -ru rev-570/sc/contrib/sasc-ng/dvbloopback/module/config_dvb.pl sc-build/contrib/sasc-ng/dvbloopback/module/config_dvb.pl
--- rev-570/sc/contrib/sasc-ng/dvbloopback/module/config_dvb.pl	2013-07-07 14:30:54.469044999 +1200
+++ sc-build/contrib/sasc-ng/dvbloopback/module/config_dvb.pl	2013-07-07 14:30:57.177044939 +1200
@@ -17,79 +17,17 @@
   $cmd = "cd config-dvb && make $vars" . ($debug ? "" : "2>/dev/null 1>/dev/null");
   print "$cmd\n" if($debug);
 
-  #test linux-version >= 2.6.22
   system("ln -sf chkdvb-2.6.v4l.c config-dvb/chkdvb.c");
-  if(system("$cmd") == 0) {
-    print "Found dvbdev.h from 2.6.22 or later\n";
+  system("ln -sf ../dvbdev-3.7.x.h config-dvb/dvbdev.h");
+   print "Assuming kernel 3.7.1 or later\n";
+   print "Using canned header\n";
+    if(system("$cmd") == 0) {
     `echo "DVB_DEFINE_MOD_OPT_ADAPTER_NR(adapter_nr);" >> dvbdevwrap.h`; 
     `echo "#define wrap_dvb_reg_adapter(a, b, c) dvb_register_adapter(a, b, c, &dvblb_basedev->dev, adapter_nr)" >> dvbdevwrap.h`; 
+     system("ln -sf dvbdev-3.7.x.h dvbdev.h"); 
     return 0;
   }
-  
-  #test linux-version >= 2.6.18
-  system("ln -sf chkdvb-2.6.18.c config-dvb/chkdvb.c");
-  if(system("$cmd") == 0) {
-    print "Found dvbdev.h from 2.6.18 or later\n";
-    `echo "#define wrap_dvb_reg_adapter(a, b, c) dvb_register_adapter(a, b, c, &dvblb_basedev->dev)" >> dvbdevwrap.h`;
-    return 0;
-  }
-  
-  #test linux-version >= 2.6.14
-  system("ln -sf chkdvb-2.6.14.c config-dvb/chkdvb.c");
-  if(system("$cmd") == 0) {
-    print "Found dvbdev.h from 2.6.14 or later\n";
-    `echo "#define wrap_dvb_reg_adapter dvb_register_adapter" >> dvbdevwrap.h`;
-    return 0;
-  }
-
-  #test linux-version >= 2.6.5
-  system("ln -sf chkdvb-2.6.5.c config-dvb/chkdvb.c");
-  if(system("$cmd") == 0) {
-    print "Found dvbdev.h from 2.6.5 or later\n";
-    print "But this is an unsupported kernel!\n";
-    return 1;
-  }
-
-  #maybe kernel headers aren't available.  let's use canned dvbdev.h
-  #this is dangerous!
-  $uname = `uname -r`;
-  if($uname =~ /2\.6\.(\d\d)/ && $1 >= 22) {
-    system("ln -sf ../dvbdev-2.6.v4l.h config-dvb/dvbdev.h"); 
-    system("ln -sf chkdvb-2.6.v4l.c config-dvb/chkdvb.c"); 
-    if(system("$cmd") == 0) {
-      print "Found 2.6.22 or later kernel, but no dvbdev.h\n";
-      print "Using canned header\n";
-      `echo "DVB_DEFINE_MOD_OPT_ADAPTER_NR(adapter_nr);" >> dvbdevwrap.h`; 
-      `echo "#define wrap_dvb_reg_adapter(a, b, c) dvb_register_adapter(a, b, c, &dvblb_basedev->dev, adapter_nr)" >> dvbdevwrap.h`; 
-      system("ln -sf dvbdev-2.6.v4l.h dvbdev.h"); 
-      return 0;
-    }
-  }
-  elsif($uname =~ /2\.6\.2[01]/ ||
-        $uname =~ /2\.6\.1[89]/) {
-    system("ln -sf ../dvbdev-2.6.18.h config-dvb/dvbdev.h");
-    system("ln -sf chkdvb-2.6.18.c config-dvb/chkdvb.c");
-    if(system("$cmd") == 0) {
-      print "Found 2.6.18 or later kernel, but no dvbdev.h\n";
-      print "Using canned header\n";
-      `echo "#define wrap_dvb_reg_adapter(a, b, c) dvb_register_adapter(a, b, c, &dvblb_basedev->dev)" >> dvbdevwrap.h`;
-      system("ln -sf dvbdev-2.6.18.h dvbdev.h");
-      return 0;
-    }
-  }
-  elsif($uname =~ /2\.6\.1[4-7]/) {
-    system("ln -sf ../dvbdev-2.6.14.h config-dvb/dvbdev.h");
-    system("ln -sf chkdvb-2.6.14.c config-dvb/chkdvb.c");
-    if(system("$cmd") == 0) {
-      print "Found 2.6.14 or later kernel, but no dvbdev.h\n";
-      print "Using canned header\n";
-      `echo "#define wrap_dvb_reg_adapter dvb_register_adapter" >> dvbdevwrap.h`;
-      system("ln -sf dvbdev-2.6.14.h dvbdev.h");
-      return 0;
-    }
-  }
-  print "Could not identify kernel\n";
-  return 1;
 }
 
 exit(test_dvb_adapter(@ARGV));
+
Only in sc-build/contrib/sasc-ng/dvbloopback/module: dvbdev-3.7.x.h
diff -ru rev-570/sc/contrib/sasc-ng/dvbloopback/module/dvblb_forward.c sc-build/contrib/sasc-ng/dvbloopback/module/dvblb_forward.c
--- rev-570/sc/contrib/sasc-ng/dvbloopback/module/dvblb_forward.c	2013-07-07 14:30:54.470044999 +1200
+++ sc-build/contrib/sasc-ng/dvbloopback/module/dvblb_forward.c	2013-07-07 14:30:57.179044939 +1200
@@ -166,9 +166,9 @@
 	struct file *ftmp = find_forwardmap(lbdev, f->private_data);
 	if (!ftmp || IS_ERR(ftmp))
 		return -EFAULT;
-	if (lbdev->forward_dev->fops &&lbdev->forward_dev->fops->ioctl)
-		return lbdev->forward_dev->fops->ioctl(
-		           ftmp->f_dentry->d_inode, ftmp, cmd, arg);
+	if (lbdev->forward_dev->fops &&lbdev->forward_dev->fops->unlocked_ioctl)
+		return lbdev->forward_dev->fops->unlocked_ioctl(
+		           ftmp, cmd, arg);
 	return -EFAULT;
 }
 
diff -ru rev-570/sc/contrib/sasc-ng/dvbloopback/module/dvb_loopback.c sc-build/contrib/sasc-ng/dvbloopback/module/dvb_loopback.c
--- rev-570/sc/contrib/sasc-ng/dvbloopback/module/dvb_loopback.c	2013-07-07 14:30:54.470044999 +1200
+++ sc-build/contrib/sasc-ng/dvbloopback/module/dvb_loopback.c	2013-07-07 14:30:57.178044939 +1200
@@ -118,9 +118,9 @@
 /* This is a copy of dvb_usercopy.  We need to do this because it isn't exported
    by dvbdev
 */
-static int dvblb_usercopy(struct inode *inode, struct file *file,
+static int dvblb_usercopy(struct file *file,
 		     unsigned int cmd, unsigned long arg,
-		     int (*func)(struct inode *inode, struct file *file,
+		     int (*func)(struct file *file,
 		     unsigned int cmd, void *arg))
 {
 	char    sbuf[128];
@@ -180,7 +180,7 @@
 	}
 
 	/* call driver */
-	if ((err = func(inode, file, cmd, parg)) == -ENOIOCTLCMD)
+	if ((err = func(file, cmd, parg)) == -ENOIOCTLCMD)
 		err = -EINVAL;
 
 	if (err < 0)
@@ -663,7 +663,7 @@
    dvb_generic_ioctl) which is called by dvblb_ioctl for device-0.  It is
    used to forward ioctl commands back to the userspace application
 */
-static int dvblb_looped_ioctl(struct inode *inode, struct file *f,
+static int dvblb_looped_ioctl(struct file *f,
 	unsigned int cmd, void *parg)
 {
 	int ret;
@@ -692,7 +692,7 @@
 	return ret;
 }
 
-static int dvblb_ioctl(struct inode *inode, struct file *f,
+static long dvblb_ioctl(struct file *f,
 	unsigned int cmd, unsigned long arg)
 {
 	void * parg = (void *)arg;
@@ -722,8 +722,7 @@
 		/* This is the looped device */
 		if (lbdev->forward_dev)
 			return dvblb_forward_ioctl(lbdev, f, cmd, arg);
-
-		return dvblb_usercopy (inode, f, cmd, arg,
+			return dvblb_usercopy (f, cmd, arg,
 		                       dvbdev->kernel_ioctl);
 	}
 	/* This is the userspace control device */
@@ -978,7 +977,7 @@
 	.write		= dvblb_write,
 	.poll		= dvblb_poll,
 	.mmap		= dvblb_mmap,
-	.ioctl		= dvblb_ioctl,
+	.unlocked_ioctl	= dvblb_ioctl,
 };
 
 static struct dvb_device dvbdev_looped = {
@@ -998,7 +997,7 @@
 	.write		= dvblb_write,
 	.poll		= dvblb_poll,
 	.mmap		= dvblb_mmap,
-	.ioctl		= dvblb_ioctl,
+        .unlocked_ioctl = dvblb_ioctl,
 };
 
 static struct dvb_device dvbdev_userspace = {
diff -ru rev-570/sc/contrib/sasc-ng/dvbloopback/module/Makefile sc-build/contrib/sasc-ng/dvbloopback/module/Makefile
--- rev-570/sc/contrib/sasc-ng/dvbloopback/module/Makefile	2013-07-07 14:30:54.468044999 +1200
+++ sc-build/contrib/sasc-ng/dvbloopback/module/Makefile	2013-07-07 14:30:57.177044939 +1200
@@ -4,26 +4,32 @@
 dvbloopback-objs := dvb_loopback.o dvblb_proc.o dvblb_forward.o
 
 ifdef DVB_DIR
-  EXTRA_CFLAGS = -I$(DVB_DIR) -I$(DVB_DIR)/drivers/media/dvb/dvb-core
+  EXTRA_CFLAGS = -I$(SOURCE_DIR) -I$(SOURCE_DIR)/include \
+     -I$(SOURCE_DIR)/include/uapi -I$(SOURCE_DIR)/arch/x86/include \
+     -I$(DVB_DIR)/drivers/media/dvb-core
   DVB_SRC = $(DVB_DIR)
-  SYMVER = $(DVB_DIR)/../v4l/Module.symvers
+  SYMVER = /lib/modules/$(shell uname -r)/build/Module.symvers
   have_modver := $(wildcard $(SYMVER))
-endif
-EXTRA_CFLAGS += -Idrivers/media/dvb/dvb-core/ -I$(PWD)
-
-BUILD_DIR ?= /lib/modules/$(shell uname -r)/build
-
-all: add_modver
+  BUILD_DIR ?= /lib/modules/$(shell uname -r)/build
+  all: add_modver
+	./config_dvb_dvbdir.pl "BUILD_DIR=$(BUILD_DIR)" "EXTRA_CFLAGS=$(EXTRA_CFLAGS)"
+	make -C $(BUILD_DIR) M=$(PWD) modules
+else
+  EXTRA_CFLAGS += -Idrivers/media/dvb-core/ -I$(PWD)
+  SYMVER = /lib/modules/$(shell uname -r)/build/Module.symvers
+  have_modver := $(wildcard $(SYMVER))
+  BUILD_DIR ?= /lib/modules/$(shell uname -r)/build
+  all: add_modver
 	./config_dvb.pl "BUILD_DIR=$(BUILD_DIR)" "EXTRA_CFLAGS=$(EXTRA_CFLAGS)"
 	make -C $(BUILD_DIR) M=$(PWD) modules
+endif
+
 ifeq ($(strip $(have_modver)),) 
 add_modver:
 	echo "Skipping Modver $(SYMVER)"
-
 else
 add_modver: $(SYMVER)
 	cp -f $(SYMVER) .
-	
 endif
 
 clean:
@@ -33,3 +39,4 @@
 	rm -f dvbdevwrap.h
 	rm -f dvbdev.h
 	rm -f Module.symvers
+
diff -ru rev-570/sc/contrib/sasc-ng/dvbloopback/src/msg_passing.c sc-build/contrib/sasc-ng/dvbloopback/src/msg_passing.c
--- rev-570/sc/contrib/sasc-ng/dvbloopback/src/msg_passing.c	2013-07-07 14:30:54.471044999 +1200
+++ sc-build/contrib/sasc-ng/dvbloopback/src/msg_passing.c	2013-07-07 14:30:57.180044939 +1200
@@ -46,11 +46,16 @@
 {
   struct msgctrl *msgctrl;
   int priority;
+  int x;
   for (priority=0; priority <= MSG_HIGH_PRIORITY; priority++) {
     msgctrl = &message_control[priority];
     bzero(msgctrl, sizeof(struct msgctrl));
     INIT_LIST_HEAD(&msgctrl->msglist);
     INIT_LIST_HEAD(&msgctrl->empty_queue);
+    for (x = 0; x < 10; x++) {
+      struct msg *msg = (struct msg *)malloc(sizeof(struct msg));
+      list_add(&msg->list, &msgctrl->empty_queue);
+    }
     pthread_mutex_init(&msgctrl->mutex, NULL);
     pthread_cond_init(&msgctrl->cond, NULL);
   }
@@ -95,6 +100,8 @@
         list_del(&msg->list);
         break;
       }
+      if (!msg)
+        break;
       //If we've seen all elements on the queue, or the queue is empty,
       //we are done
       if(ptr == &msgctrl->msglist)
diff -ru rev-570/sc/contrib/sasc-ng/Makefile sc-build/contrib/sasc-ng/Makefile
--- rev-570/sc/contrib/sasc-ng/Makefile	2013-07-07 14:30:54.465044999 +1200
+++ sc-build/contrib/sasc-ng/Makefile	2013-07-07 14:30:57.175044939 +1200
@@ -8,16 +8,16 @@
 
 CC       ?= gcc
 CXX      ?= g++
-CXXFLAGS ?= -Wall -D__user= -Werror 
+CXXFLAGS ?= -Wall -D__user= 
 CFLAGS   ?= -Wall -D__user= 
 
 ifdef DVB_DIR
-  INCLUDES = -I$(DVB_DIR)/include
+  INCLUDES = -I$(SOURCE_DIR)/include/uapi -I$(SOURCE_DIR)/arch/x86/include -I$(SOURCE_DIR)/include
   DVB_MOD_DIR = DVB_DIR=$(DVB_DIR)
 endif
 
 DEFINES += -DRELEASE_VERSION=\"$(VERSION)\" -D__KERNEL_STRICT_NAMES
-INCLUDES += -Idvbloopback/module -I/lib/modules/$(shell uname -r)/build/include
+INCLUDES += -Idvbloopback/module
 LBDIR = dvbloopback/src
 SCDIR = sc/PLUGINS/src/$(SCVER)
 SC_FLAGS = -O2 -fPIC -Wall -Woverloaded-virtual
@@ -57,7 +57,7 @@
 INC_DEPS := $(shell ls $(LBDIR)/*.h) dvbloopback/module/dvbloopback.h
 INC_DEPS_LB := $(shell ls dvblb_plugins/*.h)
 
-LIBS = -lpthread -lcrypto -lcrypt
+LIBS = -lpthread -lcrypto -lcrypt -lv4l1
 
 all: $(TOOL) libscanwrap.so
 
diff -ru rev-570/sc/contrib/sasc-ng/sc/dvbdevice.cpp sc-build/contrib/sasc-ng/sc/dvbdevice.cpp
--- rev-570/sc/contrib/sasc-ng/sc/dvbdevice.cpp	2013-07-07 14:30:54.473044999 +1200
+++ sc-build/contrib/sasc-ng/sc/dvbdevice.cpp	2013-07-07 14:30:57.184044939 +1200
@@ -10,7 +10,7 @@
 #include "include/vdr/dvbdevice.h"
 #include <errno.h>
 #include <limits.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include <linux/dvb/audio.h>
 #include <linux/dvb/dmx.h>
 #include <linux/dvb/frontend.h>
diff -ru rev-570/sc/contrib/sasc-ng/sc/sasccam.cpp sc-build/contrib/sasc-ng/sc/sasccam.cpp
--- rev-570/sc/contrib/sasc-ng/sc/sasccam.cpp	2013-07-07 14:30:54.481044999 +1200
+++ sc-build/contrib/sasc-ng/sc/sasccam.cpp	2013-07-07 14:30:57.196044938 +1200
@@ -43,7 +43,7 @@
 extern void _SetCaDescr(int adapter, ca_descr_t *ca_descr);
 bool cSascDvbDevice::SetCaDescr(ca_descr_t *ca_descr, bool initial)
 {
-  printf("Called cSascDvbDevice::SetCaDescr\n");
+  //printf("Called cSascDvbDevice::SetCaDescr\n");
   _SetCaDescr(cardidx, ca_descr);
   return true;
 }
@@ -51,7 +51,7 @@
 extern void _SetCaPid(int adapter, ca_pid_t *ca_pid);
 bool cSascDvbDevice::SetCaPid(ca_pid_t *ca_pid)
 {
-  printf("Called cSascDvbDevice::SetCaPid\n");
+  //printf("Called cSascDvbDevice::SetCaPid\n");
   _SetCaPid(cardidx, ca_pid);
   return true;
 }

