--- installwatch-0.6.3/Makefile.orig	2002-12-11 13:33:09.000000000 +0100
+++ installwatch-0.6.3/Makefile	2002-12-11 13:33:34.000000000 +0100
@@ -27,7 +27,7 @@
 install: all
 	if [ -r $(LIBDIR)/installwatch.so ]; then rm  $(LIBDIR)/installwatch.so; export LD_PRELOAD=""; cp installwatch.so $(LIBDIR); LD_PRELOAD=$(LIBDIR)/installwatch.so; else cp installwatch.so $(LIBDIR); fi
 	
-	sed -e "s|#PREFIX#|$(PREFIX)|" < installwatch > $(BINDIR)/installwatch
+	sed -e "s|#PREFIX#|/usr|" < installwatch > $(BINDIR)/installwatch
 	chmod 755 $(BINDIR)/installwatch
 
 uninstall:
