Index: autogen.sh
===================================================================
--- autogen.sh	(revision 6940)
+++ autogen.sh	(working copy)
@@ -9,10 +9,10 @@
 # tools and you shouldn't use this script.  Just call ./configure
 # directly.
 
-ACLOCAL=${ACLOCAL-aclocal-1.10}
+ACLOCAL=${ACLOCAL-aclocal-1.12}
 LIBTOOLIZE=${LIBTOOLIZE-libtoolize}
 AUTOHEADER=${AUTOHEADER-autoheader}
-AUTOMAKE=${AUTOMAKE-automake-1.10}
+AUTOMAKE=${AUTOMAKE-automake-1.12}
 AUTOCONF=${AUTOCONF-autoconf}
 GETTEXTIZE=${GETTEXTIZE-glib-gettextize}
 INTLTOOLIZE=${INTLTOOLIZE-intltoolize}