--- fsaa1655g.c	2006-03-28 13:14:10.000000000 +0200
+++ fsaa1655gmod.c	2008-09-23 23:19:39.000000000 +0200
@@ -35,7 +35,6 @@
  ***************************************************************************/
 
 #include <linux/compiler.h>
-#include <linux/config.h>
 #include <linux/errno.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
@@ -145,7 +144,7 @@
 {
 	if (dir_base) {
 		remove_proc_entry("radio", dir_base);
-		remove_proc_entry(DRV_NAME, &proc_root);
+		remove_proc_entry(DRV_NAME, NULL);
 		dir_base = NULL;
 	}
 }
@@ -156,7 +155,7 @@
 	struct proc_dir_entry *ent;
 	int err = 0;
 
-	dir_base = create_proc_entry(DRV_NAME, S_IFDIR, &proc_root);
+	dir_base = create_proc_entry(DRV_NAME, S_IFDIR, NULL);
 	if (dir_base == NULL) {
 		printk(KERN_ERR DRV_NAME ": Unable to initialise /proc/" 
 		       DRV_NAME "\n");
