diff -Nwaur src-jf-org/switchconf/Makefile src-jf/switchconf/Makefile
--- src-jf-org/switchconf/Makefile	2008-01-03 14:43:53.000000000 +0100
+++ src-jf/switchconf/Makefile	2008-02-06 12:28:31.000000000 +0100
@@ -28,7 +28,7 @@
 	cp conf $(DESTDIR)/$(ETCDIR)
 	cp plugins.before/* $(DESTDIR)/$(ETCDIR)/before.d/
 	cp plugins.after/* $(DESTDIR)/$(ETCDIR)/after.d/
-	install -D init.d $(DESTDIR)/$(INITDIR)/$(PACKNAME)
+	install -D rc.d $(DESTDIR)/$(INITDIR)/$(PACKNAME)
 	install -D $(PACKNAME) $(DESTDIR)/$(BINDIR)/$(PACKNAME)
 
 test:
diff -Naur src/switchconf/Makefile src-jf/switchconf/Makefile
--- src/switchconf/Makefile	2007-09-09 01:33:13.000000000 +0200
+++ src-jf/switchconf/Makefile	2008-01-03 14:43:53.000000000 +0100
@@ -6,7 +6,7 @@
 ETCDIR=/etc/$(PACKNAME)
 MANDIR=/usr/share/man/man8
 LOGROTATEDIR=/etc/logrotate.d
-INITDIR=/etc/init.d
+INITDIR=/etc/rc.d
 
 
 all:
@@ -24,7 +24,7 @@
 	[ -d $(DESTDIR)/$(ETCDIR)/before.d ] || mkdir -p $(DESTDIR)/$(ETCDIR)/before.d
 	[ -d $(DESTDIR)/$(ETCDIR)/after.d ] || mkdir -p $(DESTDIR)/$(ETCDIR)/after.d
 	[ -d $(DESTDIR)/$(LOGROTATEDIR)/ ] || mkdir -p $(DESTDIR)/$(LOGROTATEDIR)
-	cp debian/logrotate $(DESTDIR)/$(LOGROTATEDIR)/$(PACKNAME)
+	cp logrotate $(DESTDIR)/$(LOGROTATEDIR)/$(PACKNAME)
 	cp conf $(DESTDIR)/$(ETCDIR)
 	cp plugins.before/* $(DESTDIR)/$(ETCDIR)/before.d/
 	cp plugins.after/* $(DESTDIR)/$(ETCDIR)/after.d/
--- src/switchconf/rc.d	1970-01-01 01:00:00.000000000 +0100
+++ src-jf/switchconf/rc.d	2008-02-06 13:53:36.000000000 +0100
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+. /etc/rc.conf
+. /etc/rc.d/functions
+
+CMDLINE="/proc/cmdline"
+
+case "$1" in
+  start)
+    stat_busy "Starting $daemon_name"
+      conf="$(sed -rne 's/(.*[[:space:]]|^)switchconf=([^[:space:]]+).*/\2/p' "$CMDLINE" | tr -d \\n)"
+      if [ -z ${conf} ]; then
+        echo "No switchconf kernel option"
+        stat_fail
+        exit 1
+      else
+        /usr/sbin/switchconf ${conf}
+      fi
+      if [ $? -gt 0 ]; then
+        stat_fail
+        exit 1
+      else
+        stat_done
+      fi
+    ;;
+
+  stop)
+    ;;
+
+  restart)
+    $0 stop
+    sleep 3
+    $0 start
+    ;;
+  *)
+    echo "usage: $0 {start|stop|restart}"
+esac
+exit 0
diff -Naur src-org/switchconf/Makefile src-jf/switchconf/Makefile
--- src-org/switchconf/Makefile	2008-02-06 14:11:04.000000000 +0100
+++ src-jf/switchconf/Makefile	2008-02-06 14:15:13.000000000 +0100
@@ -26,8 +26,6 @@
 	[ -d $(DESTDIR)/$(LOGROTATEDIR)/ ] || mkdir -p $(DESTDIR)/$(LOGROTATEDIR)
 	cp logrotate $(DESTDIR)/$(LOGROTATEDIR)/$(PACKNAME)
 	cp conf $(DESTDIR)/$(ETCDIR)
-	cp plugins.before/* $(DESTDIR)/$(ETCDIR)/before.d/
-	cp plugins.after/* $(DESTDIR)/$(ETCDIR)/after.d/
 	install -D rc.d $(DESTDIR)/$(INITDIR)/$(PACKNAME)
 	install -D $(PACKNAME) $(DESTDIR)/$(BINDIR)/$(PACKNAME)
 
