--- font-manager/src/core/__init__.py.in	2010-12-24 11:05:14.000000000 +0800
+++ font-manager/src/core/__init__.py.in	2013-06-12 21:15:28.294418494 +0800
@@ -882,7 +882,7 @@
 # Make sure we have everything we need to run properly
 _setup_logging()
 for folder in CACHE_DIR, CONFIG_DIR, APP_CONFIG_DIR, AUTOSTART_DIR, DATA_DIR, \
-                        USER_FONT_CONFIG_DIR, USER_FONT_DIR, USER_LIBRARY_DIR:
+                        USER_FONT_CONFIG_DIR, USER_FONT_DIR, USER_LIBRARY_DIR, join(CONFIG_DIR, 'fontconfig'):
     if not exists(folder):
         os.makedirs(folder, 0755)
         if folder == USER_FONT_DIR:
