--- font-manager/src/utils/common.py	2010-12-20 12:47:37.000000000 +0800
+++ font-manager/src/utils/common.py	2013-06-12 22:09:06.296985078 +0800
@@ -146,7 +146,7 @@
 
     If delete is True, folder will be deleted afterwards
     """
-    archiver = """file-roller -a '{0}.{1}' '{2}'""".format(arch_name,
+    archiver = """file-roller --add-to='{0}.{1}' '{2}'""".format(arch_name,
                                                             arch_type, folder)
     os.chdir(destination)
     roller = subprocess.Popen(shlex.split(archiver))
--- font-manager/src/core/fonts.py	2010-12-20 13:14:43.000000000 +0800
+++ font-manager/src/core/fonts.py	2013-06-12 22:26:29.084669145 +0800
@@ -524,7 +524,7 @@
     tmp_dir = tempfile.mkdtemp(suffix='-font-manager', prefix='tmp-')
     arch_dir = join(tmp_dir, dir_name)
     os.mkdir(arch_dir)
-    subprocess.call(['file-roller', '-e', arch_dir, filepath])
+    subprocess.call(['file-roller', '--extract-to=' + arch_dir, filepath])
     # Todo: Need to check whether archive actually contained any fonts
     # if user_is_stupid:
     #     self.notify()
