--- font-manager.orig/src/core/fonts.py	2010-12-20 13:14:43.000000000 +0800
+++ font-manager/src/core/fonts.py	2012-09-24 22:43:35.086389030 +0800
@@ -33,6 +33,7 @@
 import glob
 import cPickle
 import shelve
+import bsddb3
 import pango
 import tempfile
 import shutil
@@ -340,7 +341,7 @@
         Delete cache if it's in an outdated format.
         """
         try:
-            self.cache = shelve.open(CACHE_FILE,
+            self.cache = shelve.Shelf(bsddb3.hashopen(CACHE_FILE),
                                         protocol=cPickle.HIGHEST_PROTOCOL)
             test = self.cache[self.cache.keys()[0]]
             assert (test.version == 1)
@@ -412,7 +413,7 @@
     def _load_new_cache(self):
         self.cache.close()
         delete_cache()
-        self.cache = shelve.open(CACHE_FILE, protocol=cPickle.HIGHEST_PROTOCOL)
+        self.cache = shelve.Shelf(bsddb3.hashopen(CACHE_FILE), protocol=cPickle.HIGHEST_PROTOCOL)
         return
 
     def _load_user_collections(self):
--- font-manager.orig/src/ui/export.py.in	2010-12-16 10:50:37.000000000 +0800
+++ font-manager/src/ui/export.py.in	2012-09-24 22:44:09.622868597 +0800
@@ -33,6 +33,7 @@
 import shutil
 import cPickle
 import shelve
+import bsddb3
 import subprocess
 import time
 import tempfile
@@ -225,7 +226,7 @@
         if outp is None:
             outp = self.workdir
         not exists(SAMPLER_CACHE) or os.unlink(SAMPLER_CACHE)
-        cache = shelve.open(SAMPLER_CACHE, protocol=cPickle.HIGHEST_PROTOCOL)
+        cache = shelve.Shelf(bsddb3.hashopen(SAMPLER_CACHE), protocol=cPickle.HIGHEST_PROTOCOL)
         cache['collection'] = self.current_collection
         cache['fontlist'] = inp
         cache['outfile'] = join(outp, '{0}.pdf'.format(self.current_collection))
--- font-manager.orig/src/ui/fontconfig.py	2010-12-16 10:50:37.000000000 +0800
+++ font-manager/src/ui/fontconfig.py	2012-09-24 22:45:20.262485910 +0800
@@ -35,6 +35,7 @@
 import pango
 import cPickle
 import shelve
+import bsddb3
 import UserDict
 
 from os.path import exists, join
@@ -320,7 +321,7 @@
         return box
 
     def _load_cache(self):
-        cache = shelve.open(join(CACHE_DIR, CACHED_SETTINGS),
+        cache = shelve.Shelf(bsddb3.hashopen(join(CACHE_DIR, CACHED_SETTINGS)),
                                             protocol=cPickle.HIGHEST_PROTOCOL)
         families = self.objects['FontManager'].list_families()
         for family in cache:
@@ -385,7 +386,7 @@
         """
         Save dialog state to file.
         """
-        cache = shelve.open(join(CACHE_DIR, CACHED_SETTINGS),
+        cache = shelve.Shelf(bsddb3.hashopen(join(CACHE_DIR, CACHED_SETTINGS)),
                                             protocol=cPickle.HIGHEST_PROTOCOL)
         families = self.objects['FontManager'].list_families()
         for family in self.cache.iterkeys():
--- font-manager.orig/src/font-sampler.in	2010-12-20 10:36:19.000000000 +0800
+++ font-manager/src/font-sampler.in	2012-09-24 22:46:22.942146346 +0800
@@ -34,6 +34,7 @@
 import gobject
 import cPickle
 import shelve
+import bsddb3
 import sys
 import time
 import UserDict
@@ -529,7 +530,7 @@
 
 if __name__ == '__main__':
     if exists(CACHE_FILE):
-        cache = shelve.open(CACHE_FILE, protocol=cPickle.HIGHEST_PROTOCOL)
+        cache = shelve.Shelf(bsddb3.hashopen(CACHE_FILE), protocol=cPickle.HIGHEST_PROTOCOL)
         objects = FontSampler()
         objects.collection = cache['collection']
         objects.fontlist = cache['fontlist']
