--- ad_oss.c		2008-05-11 06:29:38.000000000 +0200
+++ ad_oss.c.new	2010-03-07 21:54:46.661728104 +0100
@@ -211,19 +211,6 @@
     }
 
     if (mixerFD >= 0) {
-        if (ioctl(mixerFD, SOUND_MIXER_WRITE_RECSRC, &sourceMic) < 0) {
-            if (errno == ENXIO)
-                fprintf(stderr,
-                        "%s %d: can't set mic source for this device.\n",
-                        __FILE__, __LINE__);
-            else {
-                fprintf(stderr,
-                        "%s %d: mixer set to mic: %s\n",
-                        __FILE__, __LINE__, strerror(errno));
-                exit(1);
-            }
-        }
-
         /* Set the same gain for left and right channels. */
         inputGain = inputGain << 8 | inputGain;
 
