diff -Nur GLC_Player_src_2.2.0.orig/opengl_view/OpenglView.cpp GLC_Player_src_2.2.0/opengl_view/OpenglView.cpp
--- GLC_Player_src_2.2.0.orig/opengl_view/OpenglView.cpp	2010-07-01 11:49:56.000000000 +1000
+++ GLC_Player_src_2.2.0/opengl_view/OpenglView.cpp	2011-06-15 10:23:23.172582337 +1000
@@ -979,7 +979,7 @@
 	const double scaleFactor= 0.08;
 	glTranslated(1.0 - (scaleFactor * displayRatio * 1.4), - panelRatio + (scaleFactor * 1.3), 0.0);
 	glScaled(scaleFactor * displayRatio, scaleFactor, scaleFactor);
-	glMultMatrixd(uiMatrix.data());
+	glMultMatrixd(uiMatrix.getData());
 
 	qglColor(Qt::red);
 	renderText(1.0, 0.0, 0.0, "X");
diff -Nur GLC_Player_src_2.2.0.orig/ui_class/AboutPlayer.cpp GLC_Player_src_2.2.0/ui_class/AboutPlayer.cpp
--- GLC_Player_src_2.2.0.orig/ui_class/AboutPlayer.cpp	2010-07-01 11:49:56.000000000 +1000
+++ GLC_Player_src_2.2.0/ui_class/AboutPlayer.cpp	2011-06-15 10:25:48.388424571 +1000
@@ -33,5 +33,5 @@
 {
     setupUi(this);
 
-    label_GLC_Lib->setText(label_GLC_Lib->text() + QString(" ") + QString(GLC_VERSION));
+    label_GLC_Lib->setText(label_GLC_Lib->text() + QString(" ") + QString(glc::version));
 }
diff -Nur GLC_Player_src_2.2.0.orig/ui_class/EditPositionDialog.cpp GLC_Player_src_2.2.0/ui_class/EditPositionDialog.cpp
--- GLC_Player_src_2.2.0.orig/ui_class/EditPositionDialog.cpp	2010-07-01 11:49:56.000000000 +1000
+++ GLC_Player_src_2.2.0/ui_class/EditPositionDialog.cpp	2011-06-15 10:22:57.148799282 +1000
@@ -34,9 +34,9 @@
 	setupUi(this);
 
 	// Get Matrix translation
-	tx->setValue(m_Matrix.data()[12]);
-	ty->setValue(m_Matrix.data()[13]);
-	tz->setValue(m_Matrix.data()[14]);
+	tx->setValue(m_Matrix.getData()[12]);
+	ty->setValue(m_Matrix.getData()[13]);
+	tz->setValue(m_Matrix.getData()[14]);
 
 	// Get Matrix Euler angle
 	QVector<double> angles= m_Matrix.toEuler();
