--- src/Makefile.old	2013-07-21 11:51:35.356061462 -0500
+++ src/Makefile	2013-07-21 11:52:32.192731686 -0500
@@ -246,7 +246,7 @@
 	$(CROSS)$(CC) -c $(CFLAGS) $(LUAFLAGS) -Ilua/src $^
 $(luajit_objs): lua/src/lib/*.c lua/src/lib/lpeg/*.c
 	$(CROSS)$(CC) -c $(CFLAGS) $(LUAFLAGS) -Iluajit/src $^
-	for lib in $(luajit_objs); do mv $$(echo $$lib | sed 's/jit//g') $$lib; done
+	for lib in $(luajit_objs); do if test -e $lib;then cp $$(echo $$lib | sed 's/jit//g') $$lib; fi; done
 libluajit.a:
 	cd luajit && $(MAKE) CC="$(CC) $(CFLAGS)"
 	cp luajit/src/$@ .
