--- textadept_5.2/src/Makefile.old	2012-04-26 17:56:09.477622976 -0500
+++ textadept_5.2/src/Makefile	2012-04-26 19:21:46.560716449 -0500
@@ -4,7 +4,7 @@
 
 # Linux (default)
 CC = gcc
-CPP = g++
+CPP = g++ -lgmodule-2.0
 PKG_CONFIG = pkg-config
 PKG_CONFIG_PATH = $PKG_CONFIG_PATH
 ifndef BSD
@@ -287,3 +287,20 @@
 	rm -r $(MODULESDIR)
 	# Done
 	rm -r $(RELEASEDIR32)
+
+install:
+	mkdir -p -m755 $(DESTDIR)/usr/share/textadept/
+	cp -R ../core/ $(DESTDIR)/usr/share/textadept/
+	cp -R ../doc $(DESTDIR)/usr/share/textadept/
+	cp -R ../lexers $(DESTDIR)/usr/share/textadept/
+	cp -R ../modules $(DESTDIR)/usr/share/textadept/
+	cp -R ../scripts $(DESTDIR)/usr/share/textadept/
+	cp -R ../themes $(DESTDIR)/usr/share/textadept/
+	chmod -R 755 $(DESTDIR)/usr/share/textadept
+	install -D -m644 ../init.lua $(DESTDIR)/usr/share/textadept/
+	install -D -m755 ../textadept $(DESTDIR)/usr/share/textadept/textadept
+	mkdir -p -m755 $(DESTDIR)/usr/bin/
+	install -D -m644 ../core/images/textadept.png $(DESTDIR)/usr/share/pixmaps/textadept.png
+	install -D -m644 ../core/images/textadept.png $(DESTDIR)/usr/share/icons/hicolor/48x48/apps/textadept.png
+	install -D -m644 ../core/images/textadept.svg $(DESTDIR)/usr/share/icons/hicolor/scalable/apps/textadept.svg
+	install -D -m644 ../LICENSE $(DESTDIR)/usr/share/licenses/textadept/LICENSE
