--- ./Amaya/amaya/query.c.orig	2012-04-08 14:08:24.000000000 -0700
+++ ./Amaya/amaya/query.c	2012-04-08 14:11:34.000000000 -0700
@@ -2585,6 +2585,7 @@ void QueryInit ()
 #endif
 
 #ifdef SSL
+#if 0 /* HTSetSocketBufSize not in libwww any more */
   ptr = TtaGetEnvString ("BUF_SIZE_SOCKET");
   if (ptr && *ptr) {
     int bufSize;
@@ -2596,6 +2597,7 @@ void QueryInit ()
   else {
     HTUnSetSocketBufSize ();
   }
+#endif
 #endif /* SSL */
 }
 
