diff -Naur 1/Amaya11.4.4/Amaya/configure 2/Amaya11.4.4/Amaya/configure
--- 1/Amaya11.4.4/Amaya/configure	2011-12-21 11:42:33.000000000 +0100
+++ 2/Amaya11.4.4/Amaya/configure	2012-09-29 20:44:22.000000000 +0200
@@ -13579,7 +13579,7 @@
     fi
     # this is the flags used to build amaya with OpenGL
     GL_INCLUDES="-I${GL_BUILDDIR}/include"
-    GL_LIBRARIES="-Wl,-rpath,${GL_BUILDDIR}/lib -L${GL_BUILDDIR}/lib -lGL -lGLU"
+    GL_LIBRARIES="-L${GL_BUILDDIR}/lib -lGL -lGLU"
   fi
   GL_OPTIONS="-D_GL"
 
