--- libressl-2.0.0.org/crypto/compat/getentropy_linux.c	2014-07-12 10:49:35.360915885 +0000
+++ libressl-2.0.0/crypto/compat/getentropy_linux.c	2014-07-12 10:51:06.879916029 +0000
@@ -65,7 +65,6 @@
 
 int	getentropy(void *buf, size_t len);
 
-extern int main(int, char *argv[]);
 static int gotdata(char *buf, size_t len);
 static int getentropy_urandom(void *buf, size_t len);
 #ifdef CTL_MAXNAME
@@ -338,7 +337,6 @@
 			HX(sigprocmask(SIG_BLOCK, NULL, &sigset) == -1,
 			    sigset);
 
-			HD(main);		/* an addr in program */
 			HD(getentropy);	/* an addr in this library */
 			HD(printf);		/* an addr in libc */
 			p = (char *)&p;
