--- gst-rtsp-0.10.8/configure.orig	2011-01-24 10:59:40.000000000 +0000
+++ gst-rtsp-0.10.8/configure	2011-01-25 20:12:26.333133389 +0000
@@ -15100,7 +15100,7 @@
 rm -f conftest.err conftest.i conftest.$ac_ext
 CPPFLAGS="$save_CPPFLAGS"
 
-
+PYGOBJECT_REQ=2.11.2
 
 pkg_failed=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for PYGOBJECT" >&5
@@ -15173,7 +15173,7 @@
 	HAVE_PYGOBJECT="yes"
 fi
 
-PYGOBJECT_REQ=2.11.2
+
 
 
 
