diff -ur dvbhdhomerun-old/kernel/Makefile dvbhdhomerun/kernel/Makefile
--- dvbhdhomerun-old/kernel/Makefile	2010-11-18 00:00:06.000000000 +0200
+++ dvbhdhomerun/kernel/Makefile	2012-08-07 18:32:29.000000000 +0300
@@ -21,7 +21,6 @@
 
 install: dvb_hdhomerun
 	$(MAKE) -C $(KERNEL_DIR) SUBDIRS=$(PWD) modules_install
-	depmod -aeF /boot/System.map-$(KERNEL_VERSION)
 
 clean:
 	rm -f *~
diff -ur dvbhdhomerun-old/userhdhomerun/CMakeLists.txt dvbhdhomerun/userhdhomerun/CMakeLists.txt
--- dvbhdhomerun-old/userhdhomerun/CMakeLists.txt	2012-04-28 19:01:08.000000000 +0300
+++ dvbhdhomerun/userhdhomerun/CMakeLists.txt	2012-08-07 18:30:01.000000000 +0300
@@ -6,6 +6,7 @@
 ENDIF(COMMAND cmake_policy)
 
 SET(LIBHDHOMERUN_PATH /usr/lib/libhdhomerun)
+SET(LIBHDHOMERUN_INCPATH /usr/include/hdhomerun)
 
 SET(CMAKE_BUILD_TYPE Debug)
 
@@ -33,7 +34,7 @@
 
 INCLUDE_DIRECTORIES(
 	${CMAKE_CURRENT_BINARY_DIR} 
-	${LIBHDHOMERUN_PATH}
+	${LIBHDHOMERUN_INCPATH}
 )
 
 LINK_DIRECTORIES(
