--- ghosd/configure.ac	2009-02-15 17:23:29.557561023 +0100
+++ ghosd-build/configure.ac	2009-02-15 17:34:42.734158966 +0100
@@ -7,11 +7,8 @@
 
 # libtool versioning information.
 # we're gonna break this API regularly until the first release, though.
-LT_CURRENT=0
-LT_REVISION=0
-LT_AGE=0
 
-VERSION_INFO="$LT_CURRENT:$LT_REVISION:$LT_AGE"
+VERSION_INFO="0"
 AC_SUBST(VERSION_INFO)
 
 AM_MAINTAINER_MODE
