Index: eman2/src/EMAN2/src/eman2/libEM/sparx/util_sparx.cpp
===================================================================
--- eman2.orig/src/EMAN2/src/eman2/libEM/sparx/util_sparx.cpp
+++ eman2/src/EMAN2/src/eman2/libEM/sparx/util_sparx.cpp
@@ -59,6 +59,8 @@ using namespace EMAN;
 using namespace std;
 using std::complex;
 
+namespace EMAN{
+
 /* Subroutine */ 
 int circum_(double *, double *, double *, double *, int *);
 long int left_(double *, double *, double *, double *, double *, double *, double *, double *, double *);
@@ -21703,3 +21705,5 @@ int Util::branch_factor_0(int* costlist,
 	
 
 }
+
+}
