--- cupswrapperMFC7460DN-2.0.4	2010-06-11 09:55:15.000000000 +0200
+++ cupswrapperMFC7460DN-2.0.4_NEW	2011-12-04 22:56:55.608305583 +0100
@@ -18,52 +18,7 @@
 # Place, Suite 330, Boston, MA  02111-1307  USA
 #
 
-if [ "$1" = '-e' ]; then
-  lpadmin -x MFC7460DN
-  rm -f /usr/share/cups/model/MFC7460DN.ppd
-  rm -f /usr/lib/cups/filter/brlpdwrapperMFC7460DN
-  rm -f /usr/lib64/cups/filter/brlpdwrapperMFC7460DN
-  rm -f /usr/local/Brother/Printer/MFC7460DN/cupswrapper/brcupsconfig4
-  if [ -f /usr/share/ppd/MFC7460DN.ppd ];then
-  rm -f /usr/share/ppd/MFC7460DN.ppd
-  fi
-  if [ -e /etc/init.d/cups ]; then
-    /etc/init.d/cups restart
-  fi
-  if [ -e /etc/init.d/cupsys ]; then
-    /etc/init.d/cupsys restart
-  fi
-  exit 0 
-fi
-if [ "$1" = "-r" ]; then
-  lpadmin -x MFC7460DN
-  if [ -e /etc/init.d/cups ]; then
-    /etc/init.d/cups restart
-  fi
-  if [ -e /etc/init.d/cupsys ]; then
-    /etc/init.d/cupsys restart
-  fi
-  exit 0 
-fi
-if [ "$1" = "help" ] || [ "$1" = "-h" ]; then
-  echo   'option -h : help'
-  echo   '       -i : install'
-  echo   '       -e : uninstall'
-  echo   '       -r : remove printer'
-  exit 0
-fi
-mkdir -p /usr/lib/cups/filter
-
-if [ -e "/usr/local/Brother/Printer/MFC7460DN/lpd/filterMFC7460DN" ]; then
-  :
-else
-  echo "ERROR : Brother LPD filter is not installed."
-fi
-rm -f /usr/share/cups/model/MFC7460DN.ppd
-ppd_file_name=/usr/share/cups/model/MFC7460DN.ppd
-if [ ! -e /usr/share/cups/model ]; then
-   ppd_file_name=/usr/share/ppd/MFC7460DN.ppd
-fi
+ppd_file_name=MFC7460DN
 
 cat <<ENDOFPPDFILE >$ppd_file_name
 *PPD-Adobe: "4.3"
@@ -347,23 +302,7 @@
 
 ENDOFPPDFILE
 
-chmod 755  $ppd_file_name
-if [ -e '/usr/share/ppd' ];then
-  if [ -e '/usr/share/cups/model' ];then
-    cp  $ppd_file_name  /usr/share/ppd
-  fi
-fi
-
-brotherlpdwrapper=/usr/lib/cups/filter/brlpdwrapperMFC7460DN
-brotherlpdwrapper64=/usr/lib64/cups/filter/brlpdwrapperMFC7460DN
-rm -f  $brotherlpdwrapper
-if [ ! -e /usr/lib/cups/filter ];then
-  if [ -e /usr/lib64/cups/filter ];then
-    brotherlpdwrapper=/usr/lib64/cups/filter/brlpdwrapperMFC7460DN
-  fi
-fi
-
-
+brotherlpdwrapper=brlpdwrapperMFC7460DN
 cat <<!ENDOFWFILTER! >$brotherlpdwrapper
 #! /bin/sh
 #
@@ -493,25 +432,25 @@
 fi
 
 
-if [ -e "/usr/local/Brother/Printer/\$PRINTER/lpd/filter\$PRINTER" ]; then
+if [ -e "/usr/share/brother/Printer/\$PRINTER/lpd/filter\$PRINTER" ]; then
        :
 else
-    echo "ERROR: /usr/local/Brother/Printer/\$PRINTER/lpd/filter\$PRINTER does not exist"   >>\$LOGFILE
+    echo "ERROR: /usr/share/brother/Printer/\$PRINTER/lpd/filter\$PRINTER does not exist"   >>\$LOGFILE
     errorcode=30
     exit $errorcode
 fi
 
-if [ -e "/usr/local/Brother/Printer/MFC7460DN/cupswrapper/brcupsconfig4" ]; then
+if [ -e "/usr/share/brother/Printer/MFC7460DN/cupswrapper/brcupsconfig4" ]; then
   if [ \$DEBUG = 0 ]; then
-     /usr/local/Brother/Printer/MFC7460DN/cupswrapper/brcupsconfig4  \$PRINTER  \$PPD 0 "\$options" >> /dev/null
+     /usr/share/brother/Printer/MFC7460DN/cupswrapper/brcupsconfig4  \$PRINTER  \$PPD 0 "\$options" >> /dev/null
   else
-     /usr/local/Brother/Printer/MFC7460DN/cupswrapper/brcupsconfig4  \$PRINTER  \$PPD \$LOGCLEVEL "\$options" >>\$LOGFILE
+     /usr/share/brother/Printer/MFC7460DN/cupswrapper/brcupsconfig4  \$PRINTER  \$PPD \$LOGCLEVEL "\$options" >>\$LOGFILE
   fi
 fi
 
 
 if [ \$DEBUG -le 2 ]; then
-    cat    \$INPUT_TEMP_PS | /usr/local/Brother/Printer/\$PRINTER/lpd/filter\$PRINTER
+    cat    \$INPUT_TEMP_PS | /usr/share/brother/Printer/\$PRINTER/lpd/filter\$PRINTER
 fi
 
 if [ \$DEBUG -ge 2 ];  then
@@ -527,39 +466,3 @@
 
 !ENDOFWFILTER!
 
-
-chmod 755 $brotherlpdwrapper
-if [ -e /usr/lib/cups/filter ]; then
- if [ -e /usr/lib64/cups/filter ]; then
-   cp $brotherlpdwrapper  $brotherlpdwrapper64
- fi
-fi
-
-chmod a+w /usr/local/Brother/Printer/MFC7460DN/inf/brMFC7460DNrc
-chmod a+w /usr/local/Brother/Printer/MFC7460DN/inf
-if [ -e /etc/init.d/lpd ]; then
-   /etc/init.d/lpd stop
-fi
-if [  -e /etc/init.d/lprng ]; then
-   /etc/init.d/lprng stop
-fi
-  if [ -e /etc/init.d/cups ]; then
-    /etc/init.d/cups restart
-  fi
-  if [ -e /etc/init.d/cupsys ]; then
-    /etc/init.d/cupsys restart
-  fi
-sleep 2s
-port2=`lpinfo -v | grep -i 'usb://Brother/MFC7460DN' | head -1`
-if [ "$port2" = '' ];then
-  port2=`lpinfo -v | grep -i 'usb://Brother/' | head -1`
-  if [ "$port2" = '' ];then
-	   port2=`lpinfo -v | grep 'usb://' | head -1`
-  fi
-fi
-port=`echo $port2| sed s/direct//g`
-if [ "$port" = '' ];then
-	port=usb:/dev/usb/lp0
-fi
-lpadmin -p MFC7460DN -E -v $port -P $ppd_file_name
-  exit 0 
