diff -urp gruler/gruler.rb gruler.new/gruler.rb
--- gruler/gruler.rb	2008-06-06 10:36:32.000000000 +1000
+++ gruler/gruler.rb	2008-07-08 06:03:31.843709043 +1000
@@ -57,6 +57,15 @@ puts 'Connecting to GConf...'
 	$ppi = gconf[GCONF_PPI_PATH] || DEFAULT_PPI
 	gconf.set_root(GCONF_ROOT)
 
+class GladeXML
+    alias_method :guard_source_from_gc_without_nil_check, 
+:guard_source_from_gc
+    def guard_source_from_gc(source)
+      return if source.nil?
+      guard_source_from_gc_without_nil_check(source)
+    end
+  end
+
 puts 'Creating windows...'
 	$preferences_window = PreferencesWindow.new
 	$ruler_window = RulerWindow.new
