--- a/ui-libsnes/libsnes.cpp	2011-06-26 16:06:46.256986573 +0200
+++ b/ui-libsnes/libsnes.cpp	2011-06-26 16:11:51.496993143 +0200
@@ -17,7 +17,8 @@
     unsigned height = overscan ? 239 : 224;
     if(interlace) height <<= 1;
     data += 9 * 1024;  //skip front porch
-    if(pvideo_refresh) return pvideo_refresh(data, width, height);
+    if(pvideo_refresh) pvideo_refresh(data, width, height);
+    input_poll();
   }
 
   void audio_sample(uint16_t left, uint16_t right) {
