diff -pu construo-0.2.2/command_line.cxx construo-0.2.2-new/command_line.cxx
--- construo-0.2.2/command_line.cxx	2003-07-26 20:47:47.000000000 +0200
+++ construo-0.2.2-new/command_line.cxx	2008-04-26 23:58:17.000000000 +0200
@@ -17,6 +17,8 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <stdlib.h>
+#include <string.h>
 #include <config.h>
 #include <iostream>
 #include <stdio.h>
diff -pu construo-0.2.2/config.hxx construo-0.2.2-new/config.hxx
--- construo-0.2.2/config.hxx	2003-01-11 16:37:26.000000000 +0100
+++ construo-0.2.2-new/config.hxx	2008-04-26 23:58:30.000000000 +0200
@@ -20,6 +20,7 @@
 #ifndef CONFIG_HH
 #define CONFIG_HH
 
+#include <stdlib.h>
 #include <string>
 
 class Config
diff -pu construo-0.2.2/lisp_reader.cxx construo-0.2.2-new/lisp_reader.cxx
--- construo-0.2.2/lisp_reader.cxx	2003-01-11 16:37:26.000000000 +0100
+++ construo-0.2.2-new/lisp_reader.cxx	2008-04-26 23:58:44.000000000 +0200
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <string.h>
 #include "construo_error.hxx"
 #include "lisp_reader.hxx"
 
diff -pu construo-0.2.2/unix_system.cxx construo-0.2.2-new/unix_system.cxx
--- construo-0.2.2/unix_system.cxx	2003-07-26 20:47:49.000000000 +0200
+++ construo-0.2.2-new/unix_system.cxx	2008-04-26 23:59:02.000000000 +0200
@@ -17,6 +17,8 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <stdlib.h>
+#include <string.h>
 #include <stdio.h>
 #include <assert.h>
 #include <unistd.h>
diff -pu construo-0.2.2/vector.cxx construo-0.2.2-new/vector.cxx
--- construo-0.2.2/vector.cxx	2003-01-11 16:37:26.000000000 +0100
+++ construo-0.2.2-new/vector.cxx	2008-04-26 23:59:22.000000000 +0200
@@ -66,7 +66,7 @@ float CL_Vector::angle(const CL_Vector& 
 	#ifdef WIN32
 	return (float)acos(dot(v)/(norm()*v.norm()));  
 	#else
-	return std::acos(dot(v)/(norm()*v.norm()));  
+	return acos(dot(v)/(norm()*v.norm()));  
 	#endif
 }
 
@@ -87,8 +87,8 @@ CL_Vector CL_Vector::rotate(float angle,
 	float s = (float)sin(angle);
 	float c = (float)cos(angle);
 	#else
-	float s = std::sin(angle);
-	float c = std::cos(angle);
+	float s = sin(angle);
+	float c = cos(angle);
 	#endif
 
 	tmp.x = x*(a.x*a.x*(1-c)+c)     + y*(a.x*a.y*(1-c)-a.z*s) + z*(a.x*a.z*(1-c)+a.y*s);
diff -pu construo-0.2.2/world.cxx construo-0.2.2-new/world.cxx
--- construo-0.2.2/world.cxx	2003-07-28 22:57:41.000000000 +0200
+++ construo-0.2.2-new/world.cxx	2008-04-26 23:59:32.000000000 +0200
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <string.h>
 #include <assert.h>
 #include <algorithm>
 #include "config.h"
diff -pu construo-0.2.2/x11_display.cxx construo-0.2.2-new/x11_display.cxx
--- construo-0.2.2/x11_display.cxx	2003-07-29 00:46:48.000000000 +0200
+++ construo-0.2.2-new/x11_display.cxx	2008-04-26 23:59:42.000000000 +0200
@@ -17,6 +17,7 @@
 //  along with this program; if not, write to the Free Software
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
+#include <string.h>
 #include <config.h>
 #include <iostream>
 #include <X11/Xutil.h>
