--- blackmagic_core.c.orig	2011-11-18 17:32:51.279116818 +0100
+++ blackmagic_core.c	2011-11-18 17:34:01.593525761 +0100
@@ -36,6 +36,7 @@
 #include <linux/fcntl.h>
 #include <linux/poll.h>
 #include <linux/version.h>
+#include <linux/dma-mapping.h>
 
 #include "blackmagic_core.h"
 #include "blackmagic_bh.h"
@@ -327,9 +328,9 @@ blackmagic_probe(struct pci_dev *pdev, c
 	pci_enable_device(pdev);
 	pci_set_master(pdev);
 
-	if (pci_set_dma_mask(pdev, DMA_64BIT_MASK) < 0)
+	if (pci_set_dma_mask(pdev, DMA_BIT_MASK(64)) < 0)
 	{
-		if (pci_set_dma_mask(pdev, DMA_32BIT_MASK) < 0)
+		if (pci_set_dma_mask(pdev, DMA_BIT_MASK(32)) < 0)
 			goto fail;
 	}
	
--- blackmagic_lib.c.orig	2012-03-31 01:35:56.413196500 +0200
+++ blackmagic_lib.c	2012-03-31 01:37:05.427720246 +0200
@@ -69,6 +69,8 @@ static DEFINE_SPINLOCK(__dl_wait_queue_l
 static spinlock_t __dl_wait_queue_list_lock = SPIN_LOCK_UNLOCKED;
 #endif
 
+#define TS_USEDFPU 0x0001
+
 struct dl_wait_queue_head_t
 {
 	struct list_head entry;
	
