--- a/extensions/globalmenu/package/install.rdf	2013-01-23 15:10:52.517443592 +0100
+++ b/extensions/globalmenu/package/install.rdf	2013-01-23 15:12:32.636901199 +0100
@@ -34,4 +34,13 @@
 		</Description>
 		</em:targetApplication>
+		
+		<em:targetApplication>
+		<!-- Seamonkey -->
+		<Description>
+			<em:id>{92650c4d-4b8e-4d2a-b7eb-24ecf4f6b63a}</em:id>
+			<em:minVersion>2.15.1</em:minVersion>
+			<em:maxVersion>2.*</em:maxVersion>
+		</Description>
+		</em:targetApplication>
 
 		<!-- front-end metadata -->
 		<em:name>Global Menu Bar integration</em:name>
		<em:description>This extension integrates the menu bar in to the Unity panel</em:description>
