--- telldus-core-2.1.1/client/CMakeLists.txt.orig	2014-01-09 22:23:47.302984232 +0100
+++ telldus-core-2.1.1/client/CMakeLists.txt	2014-01-09 22:24:09.303162766 +0100
@@ -105,6 +105,7 @@
 ENDIF (UNIX)
 
 TARGET_LINK_LIBRARIES( ${telldus-core_TARGET}	${telldus-core_LIBRARIES} )
+TARGET_LINK_LIBRARIES( ${telldus-core_TARGET}	${telldus-core_LIBRARIES} )
 
 SET_TARGET_PROPERTIES(${telldus-core_TARGET} PROPERTIES
 	FRAMEWORK TRUE
