--- todo-plugin/todooutputpane.cpp~
+++ todo-plugin/todooutputpane.cpp
@@ -131,27 +131,27 @@ void TodoOutputPane::setFocus()
     todoList->setFocus();
 }
 
-bool TodoOutputPane::hasFocus()
+bool TodoOutputPane::hasFocus() const
 {
     return todoList->hasFocus();
 }
 
-bool TodoOutputPane::canFocus()
+bool TodoOutputPane::canFocus() const
 {
     return true;
 }
 
-bool TodoOutputPane::canNavigate()
+bool TodoOutputPane::canNavigate() const
 {
     return todoList->count() > 1;
 }
 
-bool TodoOutputPane::canNext()
+bool TodoOutputPane::canNext() const
 {
     return todoList->currentRow() < todoList->count() && todoList->count() > 1;
 }
 
-bool TodoOutputPane::canPrevious()
+bool TodoOutputPane::canPrevious() const
 {
     return todoList->currentRow() > 0 && todoList->count() > 1;
 }
--- todo-plugin/todooutputpane.h~
+++ todo-plugin/todooutputpane.h
@@ -53,15 +53,15 @@ public:
     void clearContents(QString filename);
     void visibilityChanged(bool visible);
 
-    void setFocus();
-    bool hasFocus();
-    bool canFocus();
+    virtual void setFocus();
+    virtual bool hasFocus() const;
+    virtual bool canFocus() const;
 
-    bool canNavigate();
-    bool canNext();
-    bool canPrevious();
-    void goToNext();
-    void goToPrev();
+    virtual bool canNavigate() const;
+    virtual bool canNext() const;
+    virtual bool canPrevious() const;
+    virtual void goToNext();
+    virtual void goToPrev();
 
     void sort();
 
-- 
1.7.6
