--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,9 @@
 
 for helplang in listdir('help'):
     data_files.append(('share/gnome/help/%s/%s' % (PACKAGE, helplang),
-                      glob(join('help', helplang, '*'))))
+                      glob(join('help', helplang, '*.page'))))
+    data_files.append(('share/gnome/help/%s/%s/figures' % (PACKAGE, helplang),
+                      glob(join('help', helplang, 'figures', '*'))))
 
 build_info_template = """# -*- coding: UTF-8 -*-
 
