diff -Naur caledonia-tux-orig/caledonia-tux.plymouth caledonia-tux/caledonia-tux.plymouth
--- caledonia-tux-orig/caledonia-tux.plymouth	2011-07-15 12:02:19.000000000 +0200
+++ caledonia-tux/caledonia-tux.plymouth	2011-09-13 09:41:12.141741528 +0200
@@ -4,5 +4,5 @@
 ModuleName=script
 
 [script]
-ImageDir=/lib/plymouth/themes/caledonia-tux
-ScriptFile=/lib/plymouth/themes/caledonia-tux/caledonia-tux.script
\ Manca newline alla fine del file
+ImageDir=/usr/share/plymouth/themes/caledonia-tux
+ScriptFile=/usr/share/plymouth/themes/caledonia-tux/caledonia-tux.script
\ Manca newline alla fine del file
