diff -ruN sizzleopt-0.0.3.orig/gdbm/gdbm.c sizzleopt-0.0.3.new/gdbm/gdbm.c
--- sizzleopt-0.0.3.orig/gdbm/gdbm.c	2008-11-06 20:43:53.000000000 +0100
+++ sizzleopt-0.0.3.new/gdbm/gdbm.c	2008-11-06 20:52:05.000000000 +0100
@@ -126,7 +126,6 @@
   if (gdbm_open_p (tagged))
     {
       gdbm_close (info->dbf);
-      gdbm_open_p (tagged) = 0;
     }
   zzz_free (info, sizeof (struct gdbm_info));
 }
@@ -237,7 +236,6 @@
   CHECK_TYPE (1, gdbm_p (gdbm) && gdbm_open_p (gdbm), "open gdbm file");
   gdbm_close (gdbm_info (gdbm)->dbf);
   gdbm_info (gdbm)->dbf = NULL;
-  gdbm_open_p (gdbm) = 0;
   *result = zzz_unspecified;
   return RESULT_SUCCESS;
 }
diff -ruN sizzleopt-0.0.3.orig/x/color.c sizzleopt-0.0.3.new/x/color.c
--- sizzleopt-0.0.3.orig/x/color.c	2008-11-06 20:43:53.000000000 +0100
+++ sizzleopt-0.0.3.new/x/color.c	2008-11-06 20:51:09.000000000 +0100
@@ -82,7 +82,6 @@
 	  Display * display = display_disp (color_disp (tagged));
 	  /* FIXME: Add color freeing code here.  */
 	}
-      color_open_p (tagged) = 0;
     }
   zzz_free (color_col (tagged), sizeof (XColor));
 }
@@ -148,7 +147,6 @@
   zzz_scm_t display;
 
   CHECK_TYPE (1, color_p (color) && color_open_p (color), "open color");
-  color_open_p (color) = 0;
   *result = zzz_unspecified;
   return RESULT_SUCCESS;
 }
diff -ruN sizzleopt-0.0.3.orig/x/display.c sizzleopt-0.0.3.new/x/display.c
--- sizzleopt-0.0.3.orig/x/display.c	2008-11-06 20:43:53.000000000 +0100
+++ sizzleopt-0.0.3.new/x/display.c	2008-11-06 20:46:47.000000000 +0100
@@ -79,7 +79,6 @@
   if (display_open_p (tagged))
     {
       XCloseDisplay (display_disp (tagged));
-      display_open_p (tagged) = 0;
       zzz_x_display = NULL;
     }
 }
@@ -133,7 +132,6 @@
 
   CHECK_TYPE (1, display_p (display) && display_open_p (display),
 	      "open display");
-  display_open_p (display) = 0;
   XCloseDisplay (display_disp (display));
   if (zzz_eq (zzz_default_display, display))
     {
diff -ruN sizzleopt-0.0.3.orig/x/gctxt.c sizzleopt-0.0.3.new/x/gctxt.c
--- sizzleopt-0.0.3.orig/x/gctxt.c	2008-11-06 20:43:53.000000000 +0100
+++ sizzleopt-0.0.3.new/x/gctxt.c	2008-11-06 20:50:14.000000000 +0100
@@ -78,8 +78,7 @@
          window descriptor might be already freed.  */
       if (!zzz_terminating)
 	XFreeGC (display_disp (gc_disp (tagged)), gc_gc (tagged));
-      gc_open_p (tagged) = 0;
-    }
+      }
 }
 
 static void
@@ -142,7 +141,6 @@
 
   CHECK_TYPE (1, gc_p (gc) && gc_open_p (gc), "open GC");
   XFreeGC (display_disp (gc_disp (gc)), gc_gc (gc));
-  gc_open_p (gc) = 0;
   *result = zzz_unspecified;
   return RESULT_SUCCESS;
 }
diff -ruN sizzleopt-0.0.3.orig/x/window.c sizzleopt-0.0.3.new/x/window.c
--- sizzleopt-0.0.3.orig/x/window.c	2008-11-06 20:43:53.000000000 +0100
+++ sizzleopt-0.0.3.new/x/window.c	2008-11-06 20:48:55.000000000 +0100
@@ -82,7 +82,6 @@
       if (!zzz_terminating)
 	XDestroyWindow (display_disp (window_disp (tagged)),
 			window_win (tagged));
-      window_open_p (tagged) = 0;
     }
 }
 
@@ -176,7 +175,6 @@
   XUnmapWindow (display_disp (display), window_win (window));
   XDestroyWindow (display_disp (display), window_win (window));
   XFlush (display_disp (display));
-  window_open_p (window) = 0;
   *result = zzz_unspecified;
   return RESULT_SUCCESS;
 }
