--- scangearmp-source-1.80-1/scangearmp/config.h.in	1970-01-01 01:00:00.000000000 +0100
+++ scangearmp-source-1.80-1.new/scangearmp/config.h.in	2013-04-20 17:00:50.834342887 +0200
@@ -0,0 +1 @@
+/* config.h.in. Generated automatically from configure.in by autoheader. */
--- scangearmp-source-1.80-1/scangearmp/configure.in	2011-02-14 07:19:36.000000000 +0100
+++ scangearmp-source-1.80-1.new/scangearmp/configure.in	2013-04-20 17:03:46.091465396 +0200
@@ -21,14 +21,13 @@
 AC_SUBST(enable_libpath)
 
 AM_INIT_AUTOMAKE(scangearmp, 1.80-1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_PROG_LIBTOOL
 AC_PROG_LIBTOOL
 AM_MAINTAINER_MODE
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AC_PROG_RANLIB
 
