--- scangearmp-source-1.80-1/scangearmp/src/scanfile.c	2011-02-14 07:19:36.000000000 +0100
+++ scangearmp-source-1.80-1.new/scangearmp/src/scanfile.c	2012-02-11 22:27:48.492011813 +0100
@@ -45,6 +45,7 @@
 #include "raw2pdf.h"
 #include "scanmsg.h"
 
+#define Z_BEST_SPEED 1
 #define GRAY2MONO_BK		(0)
 #define GRAY2MONO_GRAY		(1)
 
@@ -386,7 +387,7 @@
 
 void write_data_for_png( png_structp png_ptr, png_bytep data, png_size_t length )
 {
-	FileControlWriteFile( *( (int *)png_ptr->io_ptr ), (CNMSLPSTR)data, length );
+	FileControlWriteFile( *( (int *)png_get_io_ptr(png_ptr) ), (CNMSLPSTR)data, length );
 
 	return;
 }
