--- node/transport.py	2014-11-09 01:16:41.000000000 +0100
+++ node/transporta.py	2015-01-06 15:51:01.139098140 +0100
@@ -253,7 +253,7 @@
             try:
                 self.log.info('Generating PGP keypair. This may take several minutes...')
                 print 'Generating PGP keypair. This may take several minutes...'
-                gpg = gnupg.GPG()
+                gpg = gnupg.GPG(gpgbinary='gpg1')
                 input_data = gpg.gen_key_input(key_type="RSA",
                                                key_length=2048,
                                                name_email='gfy@gfy.com',
