diff --git a/SConstruct b/SConstruct
index 5e9bdb8..ccf41a0 100644
--- a/SConstruct
+++ b/SConstruct
@@ -799,7 +799,7 @@ if not tclhfound:
         tclhfound = tclhfound or configure.CheckHeader(tmp, language = "C")
 zlibhfound = configure.CheckHeader("zlib.h", language = "C")
 midiPluginSdkFound = configure.CheckHeader("funknown.h", language = "C++")
-luaFound = configure.CheckHeader("lua.h", language = "C")
+luaFound = configure.CheckHeader("luajit-2.0/lua.h", language = "C")
 print 'Checking for LUA: %s' % (['no', 'yes'][int(luaFound)])
 swigFound = 'swig' in commonEnvironment['TOOLS']
 print 'Checking for SWIG... %s' % (['no', 'yes'][int(swigFound)])
@@ -1470,7 +1470,7 @@ else:
         print 'CONFIGURATION DECISION: Building Lua wrapper to Csound C++ interface library.'
         luaWrapperEnvironment = csoundWrapperEnvironment.Clone()
         if getPlatform() != 'win32':
-            csoundWrapperEnvironment.Append(CPPPATH=['/usr/include/lua5.1'])
+            luaWrapperEnvironment.Append(CPPPATH=['/usr/include/luajit-2.0'])
         if getPlatform() == 'win32':
             csoundLuaInterface = luaWrapperEnvironment.SharedObject(
                     'interfaces/lua_interface.i',
@@ -2229,7 +2229,7 @@ else:
        if(luaFound == 1):
          luaEnvironment.Append(LIBS = ['luajit-5.1'])
          luaEnvironment.Append(LIBS = ['util', 'dl', 'm'])
-         luaEnvironment.Append(CPPPATH = '/usr/local/include/luajit-2.0')
+         luaEnvironment.Append(CPPPATH = '/usr/include/luajit-2.0')
     elif getPlatform() == 'win32':
        if(luaFound == 1):
         luaEnvironment.Append(LIBS = ['lua51'])
@@ -2563,6 +2563,7 @@ else:
        if getPlatform() == 'win32':
                  luaCsoundACWrapperEnvironment.Prepend(LIBS = Split('luaCsnd lua51 CsoundAC csnd fltk_images'))
        else:
+                 luaCsoundACWrapperEnvironment.Append(CPPPATH = ['/usr/include/luajit-2.0'])
                  luaCsoundACWrapperEnvironment.Prepend(LIBS = [luaWrapper])
                  luaCsoundACWrapperEnvironment.Prepend(LIBS = Split('luajit-5.1 CsoundAC csnd fltk_images'))
        luaCsoundACWrapper = luaCsoundACWrapperEnvironment.SharedObject(
