*** mdsplus/Makefile.in.org	2012-10-23 20:48:02.000000000 +0200
--- mdsplus/Makefile.in	2013-09-11 13:21:30.291486492 +0200
***************
*** 101,107 ****
  RPM :
  	devscripts/MakeRpm
  
! MISC_PARTS = tdi idl trees setup.sh setup.csh ChangeLog java LabView include mdsobjects/python desktop pixmaps matlab php epics
  
  install: 
  	devscripts/mkdirhier @exec_prefix@/local/tdi
--- 101,107 ----
  RPM :
  	devscripts/MakeRpm
  
! MISC_PARTS = tdi idl trees setup.sh setup.csh ChangeLog java LabView include mdsobjects/python desktop pixmaps matlab php epics rtevents
  
  install: 
  	devscripts/mkdirhier @exec_prefix@/local/tdi
