diff --git a/cmd/gear/support.go b/cmd/gear/support.go
index 46eaa44..6418885 100644
--- a/cmd/gear/support.go
+++ b/cmd/gear/support.go
@@ -187,7 +187,7 @@ func InitPostStart(dockerSocket string, id containers.Identifier) error {
 
 func getHostIPFromNamespace(name string) (*net.IPAddr, error) {
 	// Resolve the containers local IP
-	cmd := exec.Command("ip", "netns", "exec", name, "hostname", "-I")
+	cmd := exec.Command("ip", "netns", "exec", name, "hostname", "-i")
 	cmd.Stderr = os.Stderr
 	source, erro := cmd.Output()
 	if erro != nil {
