diff -Naur1 vboot_reference/Makefile vboot_reference-ld/Makefile
--- vboot_reference/Makefile	2013-10-16 05:33:18.160004369 +0000
+++ vboot_reference-ld/Makefile	2013-10-16 05:36:38.825004465 +0000
@@ -872,7 +872,2 @@
 
-# Because we play some clever linker script games to add new commands without
-# changing any header files, futility must be linked with ld.bfd, not gold.
-${FUTIL_BIN}: LDFLAGS += -fuse-ld=bfd
-${FUTIL_STATIC_BIN}: LDFLAGS += -fuse-ld=bfd
-
 # Some utilities need external crypto functions
