diff --git a/osquery/core/init_osquery.cpp b/osquery/core/init_osquery.cpp
index b6cd956..eda108a 100644
--- a/osquery/core/init_osquery.cpp
+++ b/osquery/core/init_osquery.cpp
@@ -56,7 +56,7 @@ void initOsquery(int argc, char* argv[]) {
   FLAGS_max_log_size = 1024; // max size for individual log file is 1GB
 
   // Let gflags parse the non-help options/flags.
-  google::ParseCommandLineNonHelpFlags(&argc, &argv, false);
+  gflags::ParseCommandLineNonHelpFlags(&argc, &argv, false);
 
   if (isWritable(FLAGS_osquery_log_dir.c_str()).ok()) {
     FLAGS_log_dir = FLAGS_osquery_log_dir;
diff --git a/tools/flag_test.cpp b/tools/flag_test.cpp
index 5d9dd3f..f24fa42 100644
--- a/tools/flag_test.cpp
+++ b/tools/flag_test.cpp
@@ -12,7 +12,7 @@ DEFINE_string(log_receiver, "", "log_receiver");
 DEFINE_string(log_path, "", "log_path");
 
 int main(int argc, char* argv[]) {
-  google::ParseCommandLineFlags(&argc, &argv, true);
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
   google::InitGoogleLogging(argv[0]);
   FLAGS_logtostderr = 1;
 
