--- ws	2007-12-29 01:46:16.000000000 +0000
+++ ws-arch	2010-02-28 19:07:24.000000000 +0000
@@ -99,7 +99,7 @@
 CP="/bin/cp"
 LN="/bin/ln"
 LS="/bin/ls"
-SORT="/bin/sort"
+SORT="/usr/bin/sort"
 SMBTREE="/usr/bin/smbtree"
 SMBCLIENT="/usr/bin/smbclient"
 MOUNTCIFS="/sbin/mount.cifs"
@@ -109,10 +109,10 @@
 SUDO="/usr/bin/sudo"
 TEST="/usr/bin/test"
 TOUCH="/bin/touch"
-INSTALL="/usr/bin/install"
-MYEDITOR="/bin/vi"
+INSTALL="/bin/install"
+MYEDITOR="/usr/bin/vi"
 NMBLOOKUP="/usr/bin/nmblookup"
-FUSER="/sbin/fuser"
+FUSER="/usr/bin/fuser"
 if [ $UID -eq 0 ]; then
 	# Disable sudo when
 	# using "ws -U on"
