diff -aur qedit-2.4.1/base/Key.h qedit-2.4.1.patched/base/Key.h
--- qedit-2.4.1/base/Key.h	2010-07-15 20:46:57.000000000 +0200
+++ qedit-2.4.1.patched/base/Key.h	2012-04-11 01:09:52.954775108 +0200
@@ -281,7 +281,7 @@
       for( typename Key::key_set::const_iterator iter = keys.begin(); iter !=  keys.end(); iter++)
       {
         T* t( dynamic_cast<T*>( *iter ) );
-        if( t ) insert( t );
+        if( t ) this->insert( t );
       }
 
     }
@@ -297,7 +297,7 @@
       for( typename Key::key_set::const_iterator iter = keys.begin(); iter !=  keys.end(); iter++)
       {
         T* t( dynamic_cast<T*>( *iter ) );
-        if( t ) insert( t );
+        if( t ) this->insert( t );
       }
 
     }
diff -aur qedit-2.4.1/src/WindowServer.cpp qedit-2.4.1.patched/src/WindowServer.cpp
--- qedit-2.4.1/src/WindowServer.cpp	2011-11-24 11:48:03.000000000 +0100
+++ qedit-2.4.1.patched/src/WindowServer.cpp	2012-04-11 05:53:53.391634043 +0200
@@ -354,11 +354,11 @@
         File& file( *iter );
 
         // find matching window
-        BASE::KeySet<MainWindow>::iterator iter = find_if( windows.begin(), windows.end(), MainWindow::SameFileFTor( file ) );
-        assert( iter != windows.end() );
+        BASE::KeySet<MainWindow>::iterator windowIter = find_if( windows.begin(), windows.end(), MainWindow::SameFileFTor( file ) );
+        assert( windowIter != windows.end() );
 
         // loop over views
-        BASE::KeySet<TextView> views( *iter );
+        BASE::KeySet<TextView> views( *windowIter );
         for( BASE::KeySet<TextView>::iterator viewIter = views.begin(); viewIter != views.end(); ++viewIter )
         {
             if( !(*viewIter)->selectDisplay( file ) ) continue;
