From df1b9b946a822703a6731f91db31b5ebdd0e93e8 Mon Sep 17 00:00:00 2001
From: daddz <aka.vince@gmail.com>
Date: Tue, 5 Jun 2012 19:40:27 +0200
Subject: [PATCH] fixed deprecated mkdir_p usage in Makefile

---
 main/po/Makefile.am |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main/po/Makefile.am b/main/po/Makefile.am
index afdcea3..166acd2 100644
--- a/main/po/Makefile.am
+++ b/main/po/Makefile.am
@@ -13,7 +13,7 @@ update-po:
 	$(MDTOOL_RUN) gettext-update -f:$(top_srcdir)/Main.sln
 
 $(GMO_FILES): $(LC_BUILD)/%/LC_MESSAGES/$(PACKAGE).mo: %.po
-	$(mkdir_p) $(dir $@)
+	$(MKDIR_P) $(dir $@)
 	msgfmt '$<' -o '$@'
 
 statistics:
@@ -23,7 +23,7 @@ statistics:
 	done
 	
 $(MO_FILES): $(INSTALL_DIR)/%/LC_MESSAGES/$(PACKAGE).mo : $(LC_BUILD)/%/LC_MESSAGES/$(PACKAGE).mo
-	$(mkdir_p) $(dir $@)
+	$(MKDIR_P) $(dir $@)
 	$(install_sh) '$<' '$@'
 
 install-data-local: $(MO_FILES)
-- 
1.7.10.3

