diff -Naur maint-viewer/indra/newview/viewer_manifest.py maint-viewer-c/indra/newview/viewer_manifest.py
--- maint-viewer/indra/newview/viewer_manifest.py	2009-02-01 03:17:09.000000000 +0100
+++ maint-viewer-c/indra/newview/viewer_manifest.py	2009-02-09 14:39:57.000000000 +0100
@@ -652,29 +652,41 @@
 
         if self.prefix("../../libraries/i686-linux/lib_release_client", dst="lib"):
 #            self.path("libkdu_v42R.so", "libkdu.so")
-            self.path("libfmod-3.75.so")
-            self.path("libapr-1.so.0")
-            self.path("libaprutil-1.so.0")
-            self.path("libdb-4.2.so")
-            self.path("libcrypto.so.0.9.7")
-            self.path("libexpat.so.1")
-            self.path("libssl.so.0.9.7")
-            self.path("libuuid.so", "libuuid.so.1")
-            self.path("libSDL-1.2.so.0")
-            self.path("libELFIO.so")
-            self.path("libopenjpeg.so.1.3.0", "libopenjpeg.so.1.3")
-            self.path("libalut.so")
-            self.path("libopenal.so", "libopenal.so.1")
-            self.end_prefix("lib")
+	    try:
+                self.path("libfmod-3.75.so")
+                self.path("libapr-1.so.0")
+                self.path("libaprutil-1.so.0")
+                self.path("libdb-4.2.so")
+                self.path("libcrypto.so.0.9.7")
+                self.path("libexpat.so.1")
+                self.path("libssl.so.0.9.7")
+                self.path("libuuid.so", "libuuid.so.1")
+                self.path("libSDL-1.2.so.0")
+                self.path("libELFIO.so")
+                self.path("libopenjpeg.so.1.3.0", "libopenjpeg.so.1.3")
+                self.path("libalut.so")
+                self.path("libopenal.so", "libopenal.so.1")
+                self.end_prefix("lib")
+            except:
+                print "Skipping prebuilt libs - not found"
+                self.end_prefix("lib")
 
             # Vivox runtimes
             if self.prefix(src="vivox-runtime/i686-linux", dst="bin"):
-                    self.path("SLVoice")
-                    self.end_prefix()
+                    try:
+                        self.path("SLVoice")
+                        self.end_prefix()
+                    except:
+                        print "Skipping Vivox/bin - not found"
+                        self.end_prefix()
             if self.prefix(src="vivox-runtime/i686-linux", dst="lib"):
-                    self.path("libortp.so")
-                    self.path("libvivoxsdk.so")
-                    self.end_prefix("lib")
+                    try:
+                        self.path("libortp.so")
+                        self.path("libvivoxsdk.so")
+                        self.end_prefix("lib")
+                    except:
+                        print "Skipping Vivox/lib - not found"
+                        self.end_prefix("lib")
 
 class Linux_x86_64Manifest(LinuxManifest):
     def construct(self):
