diff -Naur maint-viewer/indra/newview/llappviewerlinux.cpp maint-viewer-test/indra/newview/llappviewerlinux.cpp
--- maint-viewer/indra/newview/llappviewerlinux.cpp	2009-02-01 03:17:10.193548975 +0100
+++ maint-viewer-test/indra/newview/llappviewerlinux.cpp	2009-02-02 00:59:32.512405213 +0100
@@ -187,7 +187,7 @@
 		for (i = 0; i < size; i++)
 		{
 			// the format of the StraceFile is very specific, to allow (kludgy) machine-parsing
-			fprintf(StraceFile, "%-3d ", i);
+			fprintf(StraceFile, "%-3d ", (int) i);
 			fprintf(StraceFile, "%-32s\t", "unknown");
 			fprintf(StraceFile, "%p ", stackarray[i]);
 			fprintf(StraceFile, "%s\n", strings[i]);
