diff -Naur o/indra/llxml/llxmlnode.cpp p/indra/llxml/llxmlnode.cpp
--- o/indra/llxml/llxmlnode.cpp	2009-08-17 23:34:39.000000000 +0200
+++ p/indra/llxml/llxmlnode.cpp	2009-08-19 19:39:03.000000000 +0200
@@ -667,6 +667,7 @@
 {
 	// Read file
 	LL_DEBUGS("XMLNode") << "parsing XML file: " << filename << LL_ENDL;
+        std::string saved_locale = setlocale(LC_ALL,"C");
 	LLFILE* fp = LLFile::fopen(filename, "rb");		/* Flawfinder: ignore */
 	if (fp == NULL)
 	{
@@ -684,6 +685,7 @@
 
 	bool rv = parseBuffer(buffer, nread, node, defaults_tree);
 	delete [] buffer;
+        setlocale(LC_ALL, saved_locale.c_str() );
 	return rv;
 }
 
diff -Naur o/indra/llxml/llxmlparser.cpp p/indra/llxml/llxmlparser.cpp
--- o/indra/llxml/llxmlparser.cpp	2009-08-17 23:34:39.000000000 +0200
+++ p/indra/llxml/llxmlparser.cpp	2009-08-19 19:40:01.000000000 +0200
@@ -78,6 +78,7 @@
 	
 	BOOL success = TRUE;
 
+        std::string saved_locale = setlocale(LC_ALL,"C");
 	LLFILE* file = LLFile::fopen(path, "rb");		/* Flawfinder: ignore */
 	if( !file )
 	{
@@ -129,7 +130,7 @@
 	{
 		llwarns << mAuxErrorString << llendl;
 	}
-
+        setlocale(LC_ALL, saved_locale.c_str() );
 	return success;
 }
 
