diff -aur jsoncpp-src-0.5.0/SConstruct jsoncpp-src-0.5.0.new/SConstruct
--- jsoncpp-src-0.5.0/SConstruct	2010-03-12 08:31:04.000000000 +0100
+++ jsoncpp-src-0.5.0.new/SConstruct	2012-01-15 15:01:02.000000000 +0100
@@ -33,6 +33,15 @@
         LD_LIBRARY_PATH = "%s:libs/%s" %(LD_LIBRARY_PATH, platform)
         os.environ['LD_LIBRARY_PATH'] = LD_LIBRARY_PATH
         print "LD_LIBRARY_PATH =", LD_LIBRARY_PATH
+# XXX[patch]
+    elif platform == 'mingw':
+        CXX = 'i486-mingw32-g++'
+        import commands
+        version = commands.getoutput('%s -dumpversion' %CXX)
+        print "Using platform '%s'" %platform
+        LD_LIBRARY_PATH = os.environ.get('LD_LIBRARY_PATH', '')
+        os.environ['LD_LIBRARY_PATH'] = LD_LIBRARY_PATH
+        print "LD_LIBRARY_PATH =", LD_LIBRARY_PATH
 except KeyError:
     print 'You must specify a "platform"'
     sys.exit(2)
@@ -104,6 +113,11 @@
 elif platform == 'mingw':
     env.Tool( 'mingw' )
     env.Append( CPPDEFINES=[ "WIN32", "NDEBUG", "_MT" ] )
+# XXX[patch]
+    env['SHARED_LIB_ENABLED'] = True
+    env["CXX"] = CXX
+    env["AR"] = 'i486-mingw32-ar'
+    env["RANLIB"] = 'i486-mingw32-ranlib'
 elif platform.startswith('linux-gcc'):
     env.Tool( 'default' )
     env.Append( LIBS = ['pthread'], CCFLAGS = "-Wall" )
@@ -228,8 +242,7 @@
     """.split() )
 env.Alias( 'src-dist', srcdist_cmd )
 
-buildProjectInDirectory( 'src/jsontestrunner' )
+#buildProjectInDirectory( 'src/jsontestrunner' )
 buildProjectInDirectory( 'src/lib_json' )
-buildProjectInDirectory( 'src/test_lib_json' )
+#buildProjectInDirectory( 'src/test_lib_json' )
 #print env.Dump()
-
