--- undistract-me/long-running.bash	2013-02-20 19:36:55.223620458 +0100
+++ undistract-me-build/long-running.bash	2013-02-20 23:06:00.526691302 +0100
@@ -13,6 +13,14 @@
     LONG_RUNNING_COMMAND_TIMEOUT=10
 fi
 
+if [ -z "$LONG_RUNNING_COMMAND_SOUND" ]; then
+    LONG_RUNNING_COMMAND_SOUND="/usr/share/undistract-me/101246__tomlija__metal-glass-makes-long-ping-sound.aif"
+fi
+
+if [ -z "$LONG_RUNNING_COMMAND_SOUND_PLAYER" ]; then
+    LONG_RUNNING_COMMAND_SOUND_PLAYER=default_player
+fi
+
 # The pre-exec hook functionality is in a separate branch.
 if [ -z "$LONG_RUNNING_PREEXEC_LOCATION" ]; then
     LONG_RUNNING_PREEXEC_LOCATION=/usr/share/undistract-me/preexec.bash
@@ -52,6 +60,10 @@
 
         echo $H$M$S
     }
+    
+    function default_player() {
+        paplay --stream-name="Command finished" "$1"
+    }
 
     function precmd () {
 
@@ -66,11 +78,8 @@
                 local time_taken_human=$(sec_to_human $time_taken)
                 if [[ $time_taken -gt $LONG_RUNNING_COMMAND_TIMEOUT ]] &&
                     [[ -n $DISPLAY ]] ; then
-                    notify-send \
-                        -i utilities-terminal \
-                        -u low \
-                        "Long command completed" \
-                        "\"$__udm_last_command\" took $time_taken_human"
+                    ( ( $LONG_RUNNING_COMMAND_SOUND_PLAYER "$LONG_RUNNING_COMMAND_SOUND" & ) & )
+                    ( ( /usr/share/undistract-me/notify.py $__udm_last_window "\"$__udm_last_command\" took $time_taken_human"  & ) & )
                 fi
                 if [[ -n $LONG_RUNNING_COMMAND_CUSTOM_TIMEOUT ]] &&
                     [[ -n $LONG_RUNNING_COMMAND_CUSTOM ]] &&
