diff -ur a/FDLog1-148v-R0001.zip/fdlog.py b/FDLog1-148v-R0001.zip/fdlog.py
--- a/FDLog1-148v-R0001.zip/fdlog.py	2010-08-21 14:34:20.000000000 +0300
+++ b/FDLog1-148v-R0001.zip/fdlog.py	2010-10-08 22:08:38.000000000 +0300
@@ -2865,9 +2865,10 @@
 showbc   = 0                    # show broadcasts
 debug    = 0
 
-logdbf   = "fdlog.fdd"          # persistent file copy of log database
-logfile  = "fdlog.log"          # printable log file (contest entry)
-globf    = "fdlog.dat"          # persistent global file
+homedir  = os.path.expanduser('~')
+logdbf   = homedir + "/fdlog/fdlog.fdd" # persistent file copy of log database
+logfile  = homedir + "/fdlog/fdlog.log" # printable log file (contest entry)
+globf    = homedir + "/fdlog/fdlog.dat" # persistent global file
             
 kbuf     = ""                   # keyboard line buffer
 
@@ -2914,7 +2915,7 @@
 
 port_base += port_offset
 print "Using Network Port:",port_base
-logdbf = "fdlog%s.fdd"%(authk[0:3])
+logdbf = homedir + "/fdlog/fdlog%s.fdd"%(authk[0:3])
 print "Using Log Data file:",logdbf
 
 print "Starting Network"
@@ -2936,7 +2937,7 @@
 menu.add_cascade(label="File",menu=filemenu)
 filemenu.add_command(label="Save Entry File",command=lambda:contestlog(1))
 filemenu.add_command(label="PreView Saved Entry File",\
-                     command=lambda:viewtextf('fdlog.log'))
+                     command=lambda:viewtextf(logfile))
 filemenu.add_command(label="View Log Data File",\
                      command=lambda:viewtextf(logdbf))
 filemenu.add_command(label="Exit",command=root.quit)
