--- einstein-2.0/convert.h	2005-08-14 04:40:58.000000000 +0200
+++ convert.h	2009-05-24 17:13:35.941186331 +0200
@@ -5,6 +5,7 @@
 #include <iostream>
 #include <sstream>
 #include <string>
+#include <typeinfo>
 
 #include "exceptions.h"
 #include "unicode.h"
--- einstein-2.0/unicode.h	2005-08-14 04:40:58.000000000 +0200
+++ unicode.h	2009-05-24 17:13:22.227852925 +0200
@@ -7,6 +7,7 @@
 
 
 #include <string>
+#include <cstring>
 #include <iostream>
 
 
--- einstein-2.0/formatter.cpp	2005-08-14 04:40:58.000000000 +0200
+++ formatter.cpp	2009-06-26 07:40:23.854480611 +0200
@@ -58,7 +58,7 @@
             if ((c.type == INT_ARG) || (c.type == STRING_ARG) ||
                     (c.type == FLOAT_ARG) || (c.type == DOUBLE_ARG))
             {
-                int no = (int)c.data;
+                long int no = (long int)c.data;
                 args[no - 1] = c.type;
             }
         }
@@ -135,7 +135,7 @@
                 
             case STRING_ARG:
             case INT_ARG:
-                no = (int)cmd->data - 1;
+                no = (long int)cmd->data - 1;
                 if (no < (int)argValues.size())
                     s += argValues[no]->format(cmd);
                 break;
