--- Makefile.original	2011-12-18 12:59:07.000000000 +0000
+++ Makefile	2011-12-18 13:00:40.000000000 +0000
@@ -1,6 +1,6 @@
 
 LIBTCODDIR=src/libtcod-1.5.1
-CFLAGS=-Isrc/brogue -Isrc/platform -Wall -Wno-parentheses ${DEFINES}
+CFLAGS=-Isrc/brogue -Isrc/platform -I/usr/include/libtcod -Wno-parentheses ${DEFINES}
 RELEASENAME=brogue-1.5.2
 
 %.o : %.c
@@ -42,7 +42,7 @@
 
 both : DEPENDENCIES += ${TCOD_DEP}
 both : DEFINES += ${TCOD_DEF} ${CURSES_DEF}
-both : LIBRARIES += ${TCOD_LIB} ${CURSES_LIB}
+both : LIBRARIES += ${TCOD_LIB} ${CURSES_LIB} -lSDL
 
 both : bin/brogue
 
