--- src/platform/tcod-platform.c	2012-06-06 21:02:22.000000000 +0100
+++ ./tcod-platform.c	2012-07-29 22:40:02.060285007 +0100
@@ -290,7 +290,9 @@
 	TCOD_sys_sleep_milli((unsigned int) milliseconds);
 
 	if (bufferedKey.vk == TCODK_NONE) {
-		bufferedKey = TCOD_console_check_for_keypress(TCOD_KEY_PRESSED);
+		TCOD_sys_check_for_event(TCOD_EVENT_KEY_PRESS | TCOD_EVENT_MOUSE, &bufferedKey, &mouse);
+	} else {
+		TCOD_sys_check_for_event(TCOD_EVENT_MOUSE, &bufferedKey, &mouse);
 	}
 	
 	if (missedMouse.lmb == 0 && missedMouse.rmb == 0) {
@@ -387,7 +389,7 @@
 		}
 
 		
-		key = TCOD_console_check_for_keypress(TCOD_KEY_PRESSED);
+		TCOD_sys_check_for_event(TCOD_EVENT_KEY_PRESS | TCOD_EVENT_MOUSE, &key, &mouse);
 		rewriteKey(&key, textInput);
 		if (processKeystroke(key, returnEvent, textInput)) {
 			return;
