--- usr/autodesk/maya2012-x64/bin/maya2012	2011-07-28 03:27:15.000000000 +0300
+++ usr/autodesk/maya2012-x64/bin/maya2012.new	2011-10-18 22:56:39.669211445 +0300
@@ -204,6 +204,9 @@
 # performance degredation, remove or comment out the following line:
 unsetenv QT_PLUGIN_PATH
 
+# Set the libjpeg library to use
+setenv LD_PRELOAD "/usr/lib/libjpeg.so.62"
+
 # Set up the location of the libquicktime plugins
 setenv LIBQUICKTIME_PLUGIN_DIR "$MAYA_LOCATION/lib"
 
