diff -ruN a/plugins/xsettings/gsd-xsettings-manager.c b/plugins/xsettings/gsd-xsettings-manager.c
--- a/plugins/xsettings/gsd-xsettings-manager.c	2013-01-10 14:37:58.000000000 +0100
+++ b/plugins/xsettings/gsd-xsettings-manager.c	2013-01-24 14:20:51.455212927 +0100
@@ -215,7 +215,7 @@
  * - ajax
  * 
  */
-#define DPI_FALLBACK 96
+#define DPI_FALLBACK 120
 
 typedef struct _TranslationEntry TranslationEntry;
 typedef void (* TranslationFunc) (GnomeXSettingsManager *manager,
