--- aur-litecoin/src/litecoin-project-litecoin-7fd4c83/litecoin-qt.pro	2013-03-13 02:23:02.000000000 -0700
+++ aur-litecoin2/src/litecoin-project-litecoin-7fd4c83/litecoin-qt.pro	2013-03-25 20:07:01.324758834 -0700
@@ -4,7 +4,7 @@
 INCLUDEPATH += src src/json src/qt
 DEFINES += QT_GUI BOOST_THREAD_USE_LIB BOOST_SPIRIT_THREADSAFE USE_IPV6
 CONFIG += no_include_pwd
-
+UPNP_BUILD = 
 # for boost 1.37, add -mt to the boost libraries
 # use: qmake BOOST_LIB_SUFFIX=-mt
 # for boost thread win32 with _win32 sufix
@@ -46,15 +46,20 @@
 # miniupnpc (http://miniupnp.free.fr/files/) must be installed for support
 contains(USE_UPNP, -) {
     message(Building without UPNP support)
+    UPNP_BUILD += 0
 } else {
-    message(Building with UPNP support)
-    count(USE_UPNP, 0) {
-        USE_UPNP=1
+    contains(UPNP_BUILD, 0) {
+        message(UPNP support already defined, skipping this!)
+    } else {
+        message(Building with UPNP support)
+        count(USE_UPNP, 0) {
+            USE_UPNP=1
+        }
+        DEFINES += USE_UPNP=$$USE_UPNP STATICLIB
+        INCLUDEPATH += $$MINIUPNPC_INCLUDE_PATH
+        LIBS += $$join(MINIUPNPC_LIB_PATH,,-L,) -lminiupnpc
+        win32:LIBS += -liphlpapi
     }
-    DEFINES += USE_UPNP=$$USE_UPNP STATICLIB
-    INCLUDEPATH += $$MINIUPNPC_INCLUDE_PATH
-    LIBS += $$join(MINIUPNPC_LIB_PATH,,-L,) -lminiupnpc
-    win32:LIBS += -liphlpapi
 }
 
 # use: qmake "USE_DBUS=1"
