=== modified file 'src/applet.c'
---
 src/applet.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: ubuntu.0.7/src/applet.c
===================================================================
--- ubuntu.0.7.orig/src/applet.c
+++ ubuntu.0.7/src/applet.c
@@ -1719,16 +1719,18 @@
 	                  G_CALLBACK (foo_device_added_cb),
 	                  applet);
 	g_signal_connect (applet->nm_client, "notify::manager-running",
 	                  G_CALLBACK (foo_manager_running_cb),
 	                  applet);
 
 	if (nm_client_get_manager_running (applet->nm_client))
 		g_idle_add (foo_set_initial_state, applet);
+
+	applet_schedule_update_icon (applet);
 }
 
 static GdkPixbuf *
 applet_common_get_device_icon (NMDeviceState state, NMApplet *applet)
 {
 	GdkPixbuf *pixbuf = NULL;
 	int stage = -1;
 
@@ -1825,16 +1827,18 @@
 		pixbuf = dclass->get_icon (device, state, connection, tip, applet);
 		if (!*tip)
 			*tip = get_tip_for_device_state (device, state, connection);
 	}
 
 out:
 	if (!pixbuf)
 		pixbuf = applet_common_get_device_icon (state, applet);
+	if (!pixbuf)
+		pixbuf = applet->no_connection_icon;
 	return pixbuf;
 }
 
 static char *
 get_tip_for_vpn (NMActiveConnection *active, NMVPNConnectionState state, NMApplet *applet)
 {
 	NMConnectionScope scope;
 	char *tip = NULL;
@@ -1893,17 +1897,16 @@
 	GString *tip;
 	NMVPNConnectionState vpn_state = NM_VPN_SERVICE_STATE_UNKNOWN;
 	gboolean nm_running;
 	NMActiveConnection *active_vpn = NULL;
 
 	applet->update_icon_id = 0;
 
 	nm_running = nm_client_get_manager_running (applet->nm_client);
-	gtk_status_icon_set_visible (applet->status_icon, nm_running);
 
 	/* Handle device state first */
 
 	state = nm_client_get_state (applet->nm_client);
 	if (!nm_running)
 		state = NM_STATE_UNKNOWN;
 
 	switch (state) {
