=== modified file 'src/applet-device-gsm.c'
---
 src/applet-device-gsm.c   |    2 +-
 src/applet-device-wifi.c  |    4 ++--
 src/applet-device-wired.c |    2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

Index: network-manager-applet-0.7.1~rc3/src/applet-device-gsm.c
===================================================================
--- network-manager-applet-0.7.1~rc3.orig/src/applet-device-gsm.c
+++ network-manager-applet-0.7.1~rc3/src/applet-device-gsm.c
@@ -495,17 +495,17 @@
 		if (connection) {
 			s_con = NM_SETTING_CONNECTION (nm_connection_get_setting (connection, NM_TYPE_SETTING_CONNECTION));
 			str = s_con ? nm_setting_connection_get_id (s_con) : NULL;
 		}
 
 		applet_do_notify_with_pref (applet,
 		                            str ? str : _("GSM network."),
 		                            _("Connection Established"),
-		                            "nm-device-wwan",
+		                            "notification-gsm-high",
 		                            PREF_DISABLE_CONNECTED_NOTIFICATIONS);
 	}
 }
 
 #ifdef WITH_MBCA
 typedef struct {
 	NMDevice* device;
 	NMApplet* applet;
Index: network-manager-applet-0.7.1~rc3/src/applet-device-wifi.c
===================================================================
--- network-manager-applet-0.7.1~rc3.orig/src/applet-device-wifi.c
+++ network-manager-applet-0.7.1~rc3/src/applet-device-wifi.c
@@ -971,17 +971,17 @@
 	/* Avoid notifying too often */
 	g_get_current_time (&timeval);
 	data->last_notification_time = timeval.tv_sec;
 
 	applet_do_notify (applet,
 	                  NOTIFY_URGENCY_LOW,
 	                  _("Wireless Networks Available"),
 	                  _("Click on this icon to connect to a wireless network"),
-	                  "nm-device-wireless",
+	                  "notification-network-wireless-full",
 	                  "dont-show",
 	                  _("Don't show this message again"),
 	                  wifi_available_dont_show_cb,
 	                  applet);
 	return FALSE;
 }
 
 static void
@@ -1176,17 +1176,17 @@
 		gconf_connection = applet_get_exported_connection_for_device (device, applet);
 		if (gconf_connection && add_seen_bssid (gconf_connection, new))
 			nma_gconf_connection_save (gconf_connection);
 	}
 
 	applet_do_notify_with_pref (applet,
 	                            esc_ssid ? esc_ssid : _("(none)"),
 	                            _("Connection Established"),
-                                   "nm-device-wireless",
+	                            "notification-network-wireless-full",
 	                            PREF_DISABLE_CONNECTED_NOTIFICATIONS);
 	g_free (esc_ssid);
 }
 
 static GdkPixbuf *
 wireless_get_icon (NMDevice *device,
                    NMDeviceState state,
                    NMConnection *connection,
Index: network-manager-applet-0.7.1~rc3/src/applet-device-wired.c
===================================================================
--- network-manager-applet-0.7.1~rc3.orig/src/applet-device-wired.c
+++ network-manager-applet-0.7.1~rc3/src/applet-device-wired.c
@@ -256,17 +256,17 @@
 		if (connection) {
 			s_con = NM_SETTING_CONNECTION (nm_connection_get_setting (connection, NM_TYPE_SETTING_CONNECTION));
 			str = s_con ? nm_setting_connection_get_id (s_con) : NULL;
 		}
 
 		applet_do_notify_with_pref (applet,
 		                            str ? str : _("Wired network"),
 		                            _("Connection Established"),
-		                            "nm-device-wired",
+		                            "notification-network-ethernet-connected",
 		                            PREF_DISABLE_CONNECTED_NOTIFICATIONS);
 	}
 }
 
 static GdkPixbuf *
 wired_get_icon (NMDevice *device,
                 NMDeviceState state,
                 NMConnection *connection,
