Index: OvitoConfig.cmake
===================================================================
--- OvitoConfig.cmake	(revision 416)
+++ OvitoConfig.cmake	(working copy)
@@ -280,6 +280,9 @@
 	# Make sure Python is installed.
 	FIND_PACKAGE(PythonLibs REQUIRED)
 	INCLUDE_DIRECTORIES(${PYTHON_INCLUDE_PATH})
+
+	find_package( ZLIB REQUIRED )
+	include_directories( ${ZLIB_INCLUDE_DIRS} )
 	
 	# Modules that use Python have to link libutil too.
 	IF(NOT WIN32)
Index: cmake/Plugins.cmake
===================================================================
--- cmake/Plugins.cmake	(revision 416)
+++ cmake/Plugins.cmake	(working copy)
@@ -10,7 +10,7 @@
 	TARGET_LINK_LIBRARIES(${target_name} ${plugin_name})
 
 	IF(${plugin_name} STREQUAL "Scripting")
-		TARGET_LINK_LIBRARIES(${target_name} ${Boost_PYTHON_LIBRARY} ${PYTHON_UTIL_LIBRARY} ${PYTHON_LIBRARIES})
+		TARGET_LINK_LIBRARIES(${target_name} ${Boost_PYTHON_LIBRARY} ${PYTHON_UTIL_LIBRARY} ${PYTHON_LIBRARIES} ${ZLIB_LIBRARIES})
 	ENDIF()
 
 ENDMACRO(OVITO_ADD_PLUGIN_DEPENDENCY)
Index: src/core/CMakeLists.txt
===================================================================
--- src/core/CMakeLists.txt	(revision 416)
+++ src/core/CMakeLists.txt	(working copy)
@@ -404,7 +404,7 @@
 
 # Add Boost.Python libraries for scripting
 IF(OVITO_BUILD_PLUGIN_SCRIPTING)
-	TARGET_LINK_LIBRARIES(Core ${Boost_PYTHON_LIBRARY} ${PYTHON_UTIL_LIBRARY} ${PYTHON_LIBRARIES})
+	TARGET_LINK_LIBRARIES(Core ${Boost_PYTHON_LIBRARY} ${PYTHON_UTIL_LIBRARY} ${PYTHON_LIBRARIES} ${ZLIB_LIBRARIES})
 ENDIF(OVITO_BUILD_PLUGIN_SCRIPTING)
 
 IF(NOT OVITO_MONOLITHIC_BUILD)
Index: src/scripting/CMakeLists.txt
===================================================================
--- src/scripting/CMakeLists.txt	(revision 416)
+++ src/scripting/CMakeLists.txt	(working copy)
@@ -55,7 +55,7 @@
 )
 
 # Add Python libraries for scripting
-TARGET_LINK_LIBRARIES(Scripting ${Boost_PYTHON_LIBRARY} ${PYTHON_UTIL_LIBRARY} ${PYTHON_LIBRARIES})
+TARGET_LINK_LIBRARIES(Scripting ${Boost_PYTHON_LIBRARY} ${PYTHON_UTIL_LIBRARY} ${PYTHON_LIBRARIES} ${ZLIB_LIBRARIES})
 
 # Create sub directories for the script files
 FILE(MAKE_DIRECTORY "${OVITO_SHARE_DIRECTORY}/scripts")
