--- src/main.c.orig	2010-07-30 09:53:48.000000000 -0700
+++ src/main.c	2011-04-19 18:58:51.000000000 -0700
@@ -37,6 +37,9 @@
 
 #ifdef ENABLE_NOTIFICATION
 #include <libnotify/notify.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
 static NotifyNotification *notification = NULL;
 #endif /* ENABLE_NOTIFICATION */
 
@@ -591,7 +594,11 @@
 			g_list_free (caps);
 		}
 
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+		notification = notify_notification_new (title, msg, "goobox");
+#else
 		notification = notify_notification_new_with_status_icon (title, msg, "goobox", status_icon);
+#endif
 		notify_notification_set_urgency (notification, NOTIFY_URGENCY_LOW);
 
 		if (supports_actions) {
