--- kencfs-1.2/qtlocalpeer.cpp	2009-12-16 11:43:42.000000000 +0100
+++ kencfs-1.2-new/qtlocalpeer.cpp	2012-12-28 19:15:39.989521971 +0100
@@ -100,7 +100,7 @@
         socketName += QLatin1Char('-') + QString::number(sessionId, 16);
     }
 #else
-    socketName += QLatin1Char('-') + QString::number(::getuid(), 16);
+    socketName += QLatin1Char('-') + QString::number(QtLP_Private::getuid(), 16);
 #endif
 
     server = new QLocalServer(this);
