diff -Naur alarm-clock-applet-0.3.1.orig/src/ui.c alarm-clock-applet-0.3.1/src/ui.c
--- alarm-clock-applet-0.3.1.orig/src/ui.c	2010-03-06 11:20:23.000000000 -0400
+++ alarm-clock-applet-0.3.1/src/ui.c	2011-03-17 09:55:29.497565359 -0400
@@ -176,8 +176,15 @@
     NotifyNotification *n;
     GError *error = NULL;
 
-    n = notify_notification_new_with_status_icon (summary, body, icon,
-                                                  applet->status_icon);
+	#ifdef NOTIFY_CHECK_VERSION
+	#if NOTIFY_CHECK_VERSION (0, 7, 0)
+		n = notify_notification_new(summary, body, icon);
+	#else
+		n = notify_notification_new(summary, body, icon, applet->status_icon);
+	#endif
+	#else
+		n = notify_notification_new(summary, body, icon, applet->status_icon);
+	#endif
     
     if (!notify_notification_show (n, &error)) {
         g_warning ("Failed to send notification: %s", error->message);
