diff -rupN comm-1.9.2.bu/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc comm-1.9.2/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc
--- comm-1.9.2.bu/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc	2010-08-20 14:24:22.436288001 +0200
+++ comm-1.9.2/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc	2010-08-20 14:24:38.886288004 +0200
@@ -43,6 +43,7 @@
 #include <string.h>
 #include <sys/mman.h>
 #include <unistd.h>
+#include <sys/stat.h> 
 
 #include <algorithm>
 #include <cassert>
