diff -Naur fgetty-0.7/checkpassword-pam.c fgetty-0.7-pam/checkpassword-pam.c
--- fgetty-0.7/checkpassword-pam.c	2002-09-18 01:33:36.000000000 +0200
+++ fgetty-0.7-pam/checkpassword-pam.c	2011-07-23 18:48:38.393108788 +0200
@@ -70,5 +70,9 @@
       return 111;
     }
   }
+  if (isatty(0) && isatty(1)) {
+    puts("checkpassword: wrong password.");
+    sleep(5);
+  }
   return 1;
 }
diff -Naur fgetty-0.7/Makefile fgetty-0.7-pam/Makefile
--- fgetty-0.7/Makefile	2010-02-07 11:53:18.000000000 +0100
+++ fgetty-0.7-pam/Makefile	2011-07-23 18:48:25.233180055 +0200
@@ -2,7 +2,7 @@
 # install to somewhere else than /
 #DESTDIR=/tmp/fefix
 
-all: fgetty login login2 checkpassword
+all: fgetty login login2 checkpassword-pam
 
 DIET=diet -Os
 #CROSS=arm-linux-
@@ -23,8 +23,12 @@
 login: login.o
 login2: login2.o
 checkpassword: checkpassword.o
+checkpassword-pam.o: checkpassword-pam.c
+	$(CROSS)$(CC) -pipe -Os -fomit-frame-pointer -c checkpassword-pam.c
+checkpassword-pam2.o: checkpassword-pam2.c
+	$(CROSS)$(CC) -pipe -Os -fomit-frame-pointer -c checkpassword-pam2.c 
 checkpassword-pam: checkpassword-pam.o checkpassword-pam2.o
-	$(CROSS)$(CC) -o $@ $^ -lmisc $(LDFLAGS)
+	$(CROSS)$(CC) -o $@ $^ -lpam_misc -lpam $(LDFLAGS)
 
 debug: fgetty.c fmt_ulong.o
 	gcc -g -o debug fgetty.c fmt_ulong.o -DDEBUG
@@ -34,15 +38,15 @@
 	install login $(DESTDIR)/bin/login1
 	install login2 $(DESTDIR)/bin
 	install fgetty $(DESTDIR)/sbin
-	install checkpassword $(DESTDIR)/bin/checkpassword.login
+	install checkpassword-pam $(DESTDIR)/bin/checkpassword.login
 	install -m 644 fgetty.8 $(DESTDIR)/usr/man/man8/fgetty.8
 	@echo "now change your /etc/inittab to do something like"
 	@echo "  1:123:respawn:/sbin/fgetty /dev/vc/1 --noclear"
 
 clean:
-	rm -f *.o fgetty debug dietgetty login login2 checkpassword core
+	rm -f *.o fgetty debug dietgetty login login2 checkpassword-pam core
 
-sigs: fgetty.sig login.sig login2.sig checkpassword.sig
+sigs: fgetty.sig login.sig login2.sig checkpassword-pam.sig
 
 .SUFFIXES: .sig
 %.sig: %
