diff -urN smpq-1.4/CMakeLists.txt smpq-1.4-stormlib-find/CMakeLists.txt
--- smpq-1.4/CMakeLists.txt	2011-10-19 18:09:09.000000000 +0200
+++ smpq-1.4-stormlib-find/CMakeLists.txt	2012-07-22 15:41:05.260511327 +0200
@@ -19,10 +19,10 @@
 
 project(SMPQ)
 set(VERSION 1.4)
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.8)
 
 option(WITH_CMD "Enable smpq console application" ON)
-option(WITH_KDE "Enable KDE4 kio_smpq plugin" ON)
+option(WITH_KDE "Enable KDE4 kio_smpq plugin" OFF)
 
 if(WIN32)
 	option(WITH_NSIS "Create Windows NSIS Installer" OFF)
@@ -31,7 +31,11 @@
 if(NOT STORMLIB_INCLUDE_DIR OR NOT STORMLIB_LIBRARY)
 
 	find_path(STORMLIB_INCLUDE_DIR StormLib.h)
-	find_library(STORMLIB_LIBRARY NAMES StormLib)
+	if(WIN32)
+		find_library(STORMLIB_LIBRARY NAMES StormLib)
+	else()
+		find_library(STORMLIB_LIBRARY NAMES storm)
+	endif()
 
 	if(NOT STORMLIB_INCLUDE_DIR OR NOT STORMLIB_LIBRARY)
 		message(FATAL_ERROR "StormLib header or library not found")
@@ -130,7 +134,7 @@
 			set(MAKENSIS_OUTPUT SMPQ-${VERSION}.exe)
 
 		else(CMAKE_SIZEOF_VOID_P STREQUAL 4)
-	
+
 			set(MAKENSIS_FLAGS -DX86_64 ${MAKENSIS_FLAGS})
 			set(MAKENSIS_OUTPUT SMPQ-${VERSION}-x86_64.exe)
 
