diff -urN smpq-1.4/info.cpp smpq-1.4-stormlib-streamflags/info.cpp
--- smpq-1.4/info.cpp	2011-10-19 17:38:52.000000000 +0200
+++ smpq-1.4-stormlib-streamflags/info.cpp	2012-07-22 16:14:29.321274160 +0200
@@ -68,12 +68,12 @@
 
 	streamFlags = GetInfo(SArchive, SFILE_INFO_STREAM_FLAGS);
 
-	if ( streamFlags & STREAM_FLAG_PART_FILE )
+	if ( streamFlags & STREAM_PROVIDER_PARTIAL )
 		printMessage("Archive partial: Yes");
 	else
 		printMessage("Archive partial: No");
 
-	if ( streamFlags & STREAM_FLAG_ENCRYPTED_FILE )
+	if ( streamFlags & STREAM_PROVIDER_ENCRYPTED )
 		printMessage("Archive encryped: Yes");
 	else
 		printMessage("Archive encryped: No");
