diff --git a/Makefile.am b/Makefile.am
index 87e9fa9..cbd3868 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -47,7 +47,7 @@ LOG_PL = \
         print $$log."\n"; \
     }
 
-$(srcdir)/ChangeLog: log
+$(srcdir)/ChangeLog: 
 log:
 	@test -z "$(srcdir)" || cd $(srcdir) && \
 	 ( ! test -d .git || \
@@ -75,4 +75,4 @@ rpm:
 	cp $(PACKAGE)-$(VERSION).tar.gz /usr/src/rpm/SOURCES
 	CFLAGS="-O2 -mcpu=i686" rpm -ba --clean isync.spec
 
-doc_DATA = README TODO NEWS ChangeLog AUTHORS
+doc_DATA = README TODO NEWS AUTHORS
diff --git a/autogen.sh b/autogen.sh
index 9480c36..522cd5c 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -1,6 +1,5 @@
 #! /bin/sh
 set -e -v
-make -f Makefile.am log
 aclocal
 autoheader
 automake --add-missing
