diff -Naur exaile_0.2.14-orig/Makefile exaile_0.2.14-new/Makefile
--- exaile_0.2.14-orig/Makefile	2008-10-07 03:30:59.000000000 +1100
+++ exaile_0.2.14-new/Makefile	2010-11-03 11:28:18.973369965 +1100
@@ -68,6 +68,8 @@
 	cd $(DESTDIR)$(PREFIX)/bin && \
 	  /bin/echo -e \
 	    "#!/bin/sh\n" \
+	    "export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/xulrunner-1.9.2\n" \
+	    "export MOZILLA_FIVE_HOME=/usr/lib/xulrunner-1.9.2\n" \
 	    "cd $(PREFIX)/share/exaile\n" \
 	    "exec python $(PREFIX)$(LIBDIR)/exaile/exaile.py \"\$$@\"" \
 	    > exaile && \
