diff -Naur mp3valgui~/makefile mp3valgui/makefile
--- mp3valgui~/makefile	1969-12-31 18:00:00.000000000 -0600
+++ mp3valgui/makefile	2011-06-17 20:27:41.454225131 -0500
@@ -0,0 +1,7 @@
+install:
+	install -Dm755 mp3valgui.py $(DESTDIR)/usr/bin/mp3valgui
+	install -Dm644 mp3valgui.glade $(DESTDIR)/usr/share/mp3valgui/mp3valgui.glade
+	install -Dm644 mp3valgui.xpm $(DESTDIR)/usr/share/pixmaps/mp3valgui.xpm
+	install -Dm644 mp3valgui.desktop $(DESTDIR)/usr/share/applications/mp3valgui.desktop
+
+.PHONY: install
diff -Naur mp3valgui~/mp3valgui.desktop mp3valgui/mp3valgui.desktop
--- mp3valgui~/mp3valgui.desktop	1969-12-31 18:00:00.000000000 -0600
+++ mp3valgui/mp3valgui.desktop	2011-06-17 20:27:41.454225131 -0500
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=mp3valgui
+Comment=A PyGTK interface to mp3val
+TryExec=/usr/bin/mp3valgui
+Exec=mp3valgui
+Icon=/usr/share/pixmaps/mp3valgui.xpm
+Categories=Audio;AudioVideoEditing;
diff -Naur mp3valgui~/mp3valgui.py mp3valgui/mp3valgui.py
--- mp3valgui~/mp3valgui.py	2011-06-17 20:27:28.323902436 -0500
+++ mp3valgui/mp3valgui.py	2011-06-17 20:27:45.907667914 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # Copyright (C) 2007, Erich Schubert <erich@debian.org>
 #
 # This program is free software; you can redistribute it and/or modify
@@ -265,9 +265,9 @@
 		UI for the mp3val MP3 validator
 		"""
 		# Load Glade
-		gladefile = locate("mp3valgui.glade")
+		gladefile = locate("/usr/share/mp3valgui/mp3valgui.glade")
 		if not gladefile:
-			gobject.idle_add(doErrormessage, "UI description file mp3valgui.glade not found.")
+			gobject.idle_add(doErrormessage, "UI description file /usr/share/mp3valgui/mp3valgui.glade not found.")
 			
 		self.windowname = "windowMP3valGui"
 		self.wTree = gtk.glade.XML(gladefile, self.windowname)
diff -Naur mp3valgui~/mp3valgui.xpm mp3valgui/mp3valgui.xpm
--- mp3valgui~/mp3valgui.xpm	1969-12-31 18:00:00.000000000 -0600
+++ mp3valgui/mp3valgui.xpm	2011-06-17 20:27:41.454225131 -0500
@@ -0,0 +1,183 @@
+/* XPM */
+static const char *mp3valgui_xpm[] = {
+"32 32 148 2",
+"  	c None",
+". 	c #8B8B8B",
+"+ 	c #8A8A8A",
+"@ 	c #858585",
+"# 	c #848484",
+"$ 	c #8E8E8E",
+"% 	c #878787",
+"& 	c #B0B0B0",
+"* 	c #CDCDCD",
+"= 	c #D8D8D8",
+"- 	c #E1E1E1",
+"; 	c #E5E5E5",
+"> 	c #DFDFDF",
+", 	c #D9D9D9",
+"' 	c #C3C3C3",
+") 	c #999999",
+"! 	c #888888",
+"~ 	c #C2C2C2",
+"{ 	c #D4D4D4",
+"] 	c #D3D3D3",
+"^ 	c #E4E4E4",
+"/ 	c #EBEBEB",
+"( 	c #F1F1F1",
+"_ 	c #E6E6E6",
+": 	c #A7A7A7",
+"< 	c #868686",
+"[ 	c #A5A5A5",
+"} 	c #D7D7D7",
+"| 	c #D2D2D2",
+"1 	c #C6C6C6",
+"2 	c #C9C9C9",
+"3 	c #CCCCCC",
+"4 	c #DCDCDC",
+"5 	c #EAEAEA",
+"6 	c #F0F0F0",
+"7 	c #F5F5F5",
+"8 	c #DBDBDB",
+"9 	c #8C8C8C",
+"0 	c #BABABA",
+"a 	c #C5C5C5",
+"b 	c #BDBDBD",
+"c 	c #C0C0C0",
+"d 	c #C7C7C7",
+"e 	c #DDDDDD",
+"f 	c #E2E2E2",
+"g 	c #EEEEEE",
+"h 	c #F2F2F2",
+"i 	c #F7F7F7",
+"j 	c #949494",
+"k 	c #BFBFBF",
+"l 	c #D5D5D5",
+"m 	c #BCBCBC",
+"n 	c #B7B7B7",
+"o 	c #BEBEBE",
+"p 	c #CFCFCF",
+"q 	c #DADADA",
+"r 	c #E8E8E8",
+"s 	c #ECECEC",
+"t 	c #EFEFEF",
+"u 	c #F9F9F9",
+"v 	c #F3F3F3",
+"w 	c #A9A9A9",
+"x 	c #DEDEDE",
+"y 	c #BBBBBB",
+"z 	c #B6B6B6",
+"A 	c #B8B8B8",
+"B 	c #D1D1D1",
+"C 	c #E0E0E0",
+"D 	c #E9E9E9",
+"E 	c #EDEDED",
+"F 	c #F4F4F4",
+"G 	c #F6F6F6",
+"H 	c #F8F8F8",
+"I 	c #C1C1C1",
+"J 	c #C8C8C8",
+"K 	c #CECECE",
+"L 	c #E3E3E3",
+"M 	c #E7E7E7",
+"N 	c #A6A6A6",
+"O 	c #C4C4C4",
+"P 	c #DADAD9",
+"Q 	c #E1E1E0",
+"R 	c #E4E4E3",
+"S 	c #E7E7E6",
+"T 	c #EAEAE9",
+"U 	c #D6D6D6",
+"V 	c #CBCBCB",
+"W 	c #CACACA",
+"X 	c #CFCFCE",
+"Y 	c #DDDDDC",
+"Z 	c #E0E0DF",
+"` 	c #E3E3E2",
+" .	c #E6E6E5",
+"..	c #E9E9E8",
+"+.	c #EBEBEA",
+"@.	c #EDEDEC",
+"#.	c #EFEFEE",
+"$.	c #F3F3F2",
+"%.	c #B3B3B3",
+"&.	c #DFDFDE",
+"*.	c #E2E2E1",
+"=.	c #E8E8E7",
+"-.	c #F0F0EF",
+";.	c #B2B2B2",
+">.	c #838383",
+",.	c #8D8D8D",
+"'.	c #DEDEDD",
+").	c #D4D4D3",
+"!.	c #DCDCDB",
+"~.	c #D6D6D5",
+"{.	c #D0D0D0",
+"].	c #989898",
+"^.	c #828282",
+"/.	c #ECECEB",
+"(.	c #AAAAAA",
+"_.	c #E5E5E4",
+":.	c #DBDBDA",
+"<.	c #D8D8D7",
+"[.	c #A2A2A2",
+"}.	c #EEEEED",
+"|.	c #979797",
+"1.	c #F1F1F0",
+"2.	c #F4F4F3",
+"3.	c #D9D9D8",
+"4.	c #8F8F8F",
+"5.	c #FBFBFB",
+"6.	c #A0A0A0",
+"7.	c #939393",
+"8.	c #FDFDFD",
+"9.	c #D3D3D2",
+"0.	c #D5D5D4",
+"a.	c #969696",
+"b.	c #FEFEFE",
+"c.	c #FCFCFC",
+"d.	c #FAFAFA",
+"e.	c #D7D7D6",
+"f.	c #D2D2D1",
+"g.	c #B9B9B9",
+"h.	c #898989",
+"i.	c #757575",
+"j.	c #696969",
+"k.	c #9C9C9C",
+"l.	c #5A5A5A",
+"m.	c #9A9A9A",
+"n.	c #9B9B9B",
+"o.	c #6D6D6D",
+"p.	c #7B7B7B",
+"q.	c #818181",
+"                                                                ",
+"                          . + @ # + $ %                         ",
+"                    + + & * = - ; > , ' ) %                     ",
+"                ! + ~ > , { ] { , > ^ / ( _ : +                 ",
+"              < [ } | ' 1 2 3 | } 4 - _ 5 6 7 8 9               ",
+"            @ 0 } a b c ' d * ] = e f _ 5 g h i g j             ",
+"          < k l m n 0 o ' 2 p { q > ^ r s t h 7 u v .           ",
+"        + w x ' y z A o a 3 B } 4 C ; D E ( F G H ( * !         ",
+"        ! > B ' k 0 0 I J K ] q > L M / t h 7 G r ; s N         ",
+"      + ' C 3 d ' ' ~ O 3 P Q R S T T t ( F v - > f r x <       ",
+"      + D U p * V W W X Y Z `  ...+.@.#.$.E , q e C L 5 w       ",
+"      %.r = l ] ] | | &.Z *.S =.5 #.#.-.T U { } q x - M W 9     ",
+"    $ l ; e 4 q , , Z ` ` ` ;.$ 9 : @...Y Y | l = 8 x L ^ >.    ",
+"    . _ r L - C > x  . .r N         ,.> '.Y ).] U , 4 > D +     ",
+"    @ ( s D M ; ^ ; ....Z !           m f !.~.{.] U q e _ ].    ",
+"    ^.H t E s 5 D 5 /.@.8             (._.:.<.K B { } q L [.    ",
+"    # G F ( 6 g g g }.}.5 <           m f P ] V K | l = - |.    ",
+"    9 ( H 7 F h h h -.1.2.:         ,.*.8 3.K 2 3 p | l C .     ",
+"    4.M 5.H G G 7 G v $.+.&.6.+ + 7.3.*.~.~.a 1 W * {.{ , >.    ",
+"      W 8.5.u H u u H T Q Q M L *...'.9.0.d I O d W * { I 9     ",
+"      a.b.c.c.5.5.d.r &.*.Z '.Y :.e.0.f.J y o ~ a J V l [       ",
+"      9 _ b.8.8.5.s ^ C e &.&.Y :.3.9.I z g.m k ~ a 3 * @       ",
+"        j b.b.c.h E r L > q U ] B J ~ b g.z 0 b c O p [.<       ",
+"        + , b.d.7 6 s M f e , { p V 1 I m A n 0 b J 0 @         ",
+"          + x c.u F t / _ - 4 = ] K 2 a c y n A c c %           ",
+"            h.> c.H v g D ; C 8 } | * J O k y k y 9 i.          ",
+"              < 2 c.i h E r ^ > q l B 3 d 1 J ;.< j.            ",
+"                < ) L i v E M f x q U { | d k.>.                ",
+"                  l.< m.' C r / M C ] g.n.# o.                  ",
+"                        p.! # >.@ ^.< q.                        ",
+"                                                                ",
+"                                                                "};
