--- cm15a.c	2012-06-12 13:35:40.747513969 +0200
+++ cm15a.new.c	2012-06-12 13:36:58.717520096 +0200
@@ -179,7 +179,7 @@
 #define AUTOCONF
 
 #if defined(AUTOCONF)
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #else
 // #include <linux/config.h> // http://forums.labjack.com/index.php?showtopic=4753 Labjack support
 //                           // suggests that the config.h file is deprecated
@@ -190,7 +190,7 @@
 #include <linux/init.h>
 #include <linux/slab.h>
 #include <linux/module.h>
-#include <linux/smp_lock.h>
+//#include <linux/smp_lock.h>
 #include <linux/completion.h>
 #include <asm/uaccess.h>
 #include <linux/usb.h>
