diff -urp "/Original/configure" "/Parchao/configure"
--- "/Original/configure"	2013-10-17 11:21:32.000000000 -0300
+++ "/Parchao/configure"	2013-11-21 04:42:14.366457476 -0300
@@ -4052,7 +4052,7 @@ case $kernel_arch in
 
 		BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_SOURCES/include/asm/mach-default"
 		if test -d "$KERNEL_SOURCES/arch/x86/include" ; then
-			BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_SOURCES/arch/x86/include -I$KERNEL_SOURCES/arch/x86/include/asm/mach-default"
+			BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_SOURCES/arch/x86/include -I$KERNEL_SOURCES/arch/x86/include/asm/mach-default -I$KERNEL_SOURCES/arch/x86/include/generated"
 		fi
 		if test -d "$KERNEL_BUILD/include2/asm/mach-default" ; then
 		    BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_BUILD/include2/asm/mach-default"
@@ -4075,7 +4075,7 @@ case $kernel_arch in
 			BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_SOURCES/include/asm/mach-default"
 		fi
 		if test -d "$KERNEL_SOURCES/arch/x86/include" ; then
-			BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_SOURCES/arch/x86/include -I$KERNEL_SOURCES/arch/x86/include/asm/mach-default"
+			BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_SOURCES/arch/x86/include -I$KERNEL_SOURCES/arch/x86/include/asm/mach-default -I$KERNEL_SOURCES/arch/x86/include/generated"
 		fi
 	;;
 	mips*)
@@ -4097,7 +4097,7 @@ case $kernel_arch in
 
 		BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_SOURCES/include/asm/mach-default"
 		if test -d "$KERNEL_SOURCES/arch/x86/include" ; then
-			BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_SOURCES/arch/x86/include -I$KERNEL_SOURCES/arch/x86/include/asm/mach-default"
+			BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_SOURCES/arch/x86/include -I$KERNEL_SOURCES/arch/x86/include/asm/mach-default -I$KERNEL_SOURCES/arch/x86/include/generated"
 		fi
 		if test -d "$KERNEL_BUILD/include2/asm/mach-default" ; then
 		    BUILD_INC_DIRS="$BUILD_INC_DIRS -I$KERNEL_BUILD/include2/asm/mach-default"
