diff -rupN playedit-0.1.5/src/app_window.cpp playedit-0.1.5-patched/src/app_window.cpp
--- playedit-0.1.5/src/app_window.cpp	2011-12-09 09:09:32.000000000 +0530
+++ playedit-0.1.5-patched/src/app_window.cpp	2012-11-13 00:48:01.031792718 +0530
@@ -21,6 +21,9 @@
 #include "basic_dialogs.h"
 
 #include <glibmm/i18n.h>
+#include <glibmm/miscutils.h>
+#include <glibmm/thread.h>
+#include <glibmm/timer.h>
 #include <gtkmm/action.h>
 #include <gtkmm/main.h>
 #include <gtkmm/icontheme.h>
diff -rupN playedit-0.1.5/src/basic_dialogs.cpp playedit-0.1.5-patched/src/basic_dialogs.cpp
--- playedit-0.1.5/src/basic_dialogs.cpp	2011-12-08 10:55:41.000000000 +0530
+++ playedit-0.1.5-patched/src/basic_dialogs.cpp	2012-11-13 00:48:42.228241210 +0530
@@ -18,6 +18,7 @@
 #include "basic_dialogs.h"
 
 #include <glibmm/i18n.h>
+#include <glibmm/miscutils.h>
 #include <giomm/appinfo.h>
 #include <gtkmm/messagedialog.h>
 #include <gtkmm/aboutdialog.h>
diff -rupN playedit-0.1.5/src/metadata_editor.cpp playedit-0.1.5-patched/src/metadata_editor.cpp
--- playedit-0.1.5/src/metadata_editor.cpp	2011-12-08 11:02:29.000000000 +0530
+++ playedit-0.1.5-patched/src/metadata_editor.cpp	2012-11-13 00:49:06.361446811 +0530
@@ -18,6 +18,7 @@
 #include "metadata_editor.h"
 
 #include <glibmm/i18n.h>
+#include <glibmm/stringutils.h>
 #include <gtkmm/stock.h>
 #include <iomanip>
 
diff -rupN playedit-0.1.5/src/playlist_drag_dest.cpp playedit-0.1.5-patched/src/playlist_drag_dest.cpp
--- playedit-0.1.5/src/playlist_drag_dest.cpp	2011-12-09 07:46:35.000000000 +0530
+++ playedit-0.1.5-patched/src/playlist_drag_dest.cpp	2012-11-13 00:49:32.571307860 +0530
@@ -19,6 +19,7 @@
 #include "playlist.h"
 #include "music.h"
 
+#include <glibmm/main.h>
 #include <iostream>
 
 const unsigned int EXPAND_TIMEOUT = 1000;
