diff --git a/Makefile b/Makefile
index 2499ec3..cc0abaa 100644
--- a/Makefile
+++ b/Makefile
@@ -40,7 +40,7 @@ MACHINE=$(shell $(CC) -dumpmachine | sed "s/\(-\).*$$//")
 
 ifeq ($(MACHINE),x86_64)
 	ARCH=$(MACHINE)
-	LIBDIR=/usr/lib64
+	LIBDIR=/usr/lib
 	FORMAT=efi-app-x86-64
 else
 	ARCH=ia32
@@ -48,10 +48,11 @@ else
 	FORMAT=efi-app-$(ARCH)
 endif
 
-CRT0=$(LIBDIR)/gnuefi/crt0-efi-$(ARCH).o
-LDSCRIPT=$(LIBDIR)/gnuefi/elf_$(ARCH)_efi.lds
+CRT0=$(LIBDIR)/crt0-efi-$(ARCH).o
+LDSCRIPT=$(LIBDIR)/elf_$(ARCH)_efi.lds
 
 CFLAGS=-I. -I/usr/include/efi -I/usr/include/efi/$(ARCH) \
+		-I/usr/include/efi/protocol \
 		-DEFI_FUNCTION_WRAPPER -fPIC -fshort-wchar -ffreestanding \
 		-Wall -Ifs/ -Iloaders/ -D$(ARCH) -Werror
 LDFLAGS=-T $(LDSCRIPT) -Bsymbolic -shared -nostdlib -L$(LIBDIR) $(CRT0)
