--- compat-wireless-2.6-old-build/Makefile	2008-09-15 23:45:37.000000000 +0200
+++ compat-wireless-2.6-old-build/Makefile	2008-09-16 21:24:41.000000000 +0200
@@ -1,10 +1,10 @@
 KMODDIR?=       updates
 KMODDIR_ARG:=   "INSTALL_MOD_DIR=$(KMODDIR)"
-ifneq ($(origin $(KLIB)), undefined)
 KMODPATH_ARG:=  "INSTALL_MOD_PATH=$(KLIB)"
-else
+ifeq ($(origin $(KLIB)), undefined)
 KLIB:=          /lib/modules/$(shell uname -r)
 endif
+KMODPATH_ARG:=  "INSTALL_MOD_PATH=$(DESTDIR)"
 export KLIB_BUILD ?=	$(KLIB)/build
 # Sometimes not available in the path
 MODPROBE := /sbin/modprobe
@@ -57,75 +57,18 @@
 	@touch $@
 	@md5sum $(COMPAT_CONFIG) > $(CONFIG_CHECK)
 
-install: uninstall modules
+install: modules
 	$(MAKE) -C $(KLIB_BUILD) M=$(PWD) $(KMODDIR_ARG) $(KMODPATH_ARG) \
-		modules_install
+	modules_install
 	@# All the scripts we can use
-	@mkdir -p /usr/lib/compat-wireless/
-	@install scripts/modlib.sh	/usr/lib/compat-wireless/
-	@install scripts/madwifi-unload	/usr/sbin/
+	@mkdir -p $(DESTDIR)/usr/lib/compat-wireless/
+	@mkdir -p $(DESTDIR)/usr/sbin/
+	@install scripts/modlib.sh      $(DESTDIR)/usr/lib/compat-wireless/
+	@install scripts/madwifi-unload $(DESTDIR)/usr/sbin/
 	@# This is to allow switching between drivers without blacklisting
-	@install scripts/athenable	/usr/sbin/
-	@install scripts/b43enable	/usr/sbin/
-	@install scripts/athload	/usr/sbin/
-	@install scripts/b43load	/usr/sbin/
-	@if [ ! -z $(MADWIFI) ]; then \
-		echo ;\
-		echo -n "Note: madwifi detected, we're going to disable it. "  ;\
-		echo "If you would like to enable it later you can run:"  ;\
-		echo "    sudo athenable madwifi"  ;\
-		echo ;\
-		echo Running athenable ath5k...;\
-		/usr/sbin/athenable ath5k ;\
-	fi
-	@/sbin/depmod -ae
-	@echo
-	@echo "Currently detected wireless subsystem modules:"
-	@echo 
-	@$(MODPROBE) -l mac80211
-	@# rc80211_simple is a module only on 2.6.22 and 2.6.23
-	@$(MODPROBE) -l cfg80211
-	@$(MODPROBE) -l adm8211
-	@$(MODPROBE) -l at76_usb
-	@$(MODPROBE) -l ath5k
-	@$(MODPROBE) -l ath9k
-	@$(MODPROBE) -l b43
-	@$(MODPROBE) -l b43legacy
-	@$(MODPROBE) -l ssb
-	@$(MODPROBE) -l iwlcore
-	@$(MODPROBE) -l iwl3945
-	@$(MODPROBE) -l iwlagn
-	@$(MODPROBE) -l ipw2100
-	@$(MODPROBE) -l ipw2200
-	@$(MODPROBE) -l ieee80211
-	@$(MODPROBE) -l ieee80211_crypt
-	@$(MODPROBE) -l libertas_cs
-	@$(MODPROBE) -l ub8xxx
-	@$(MODPROBE) -l p54_pci
-	@$(MODPROBE) -l p54_usb
-	@$(MODPROBE) -l rt2400pci
-	@$(MODPROBE) -l rt2500pci
-	@$(MODPROBE) -l rt2500usb
-	@$(MODPROBE) -l rt61pci
-	@$(MODPROBE) -l rt73usb
-	@$(MODPROBE) -l usbnet
-	@$(MODPROBE) -l cdc_ether
-	@$(MODPROBE) -l rndis_host
-	@$(MODPROBE) -l rndis_wlan
-	@$(MODPROBE) -l rtl8180
-	@$(MODPROBE) -l rtl8187
-	@# rc80211_simple is no longer a module
-	@#$(MODPROBE) -l rc80211_simple
-	@$(MODPROBE) -l zd1211rw
-	@echo 
-	@echo Now run:
-	@echo 
-	@echo make unload
-	@echo
-	@echo And then load the wireless module you need. If unsure run:
-	@echo
-	@echo make load
-	@echo
+	@install scripts/athenable      $(DESTDIR)/usr/sbin/
+	@install scripts/b43enable      $(DESTDIR)/usr/sbin/
+	@install scripts/athload        $(DESTDIR)/usr/sbin/
 
 uninstall:
 	@# New location, matches upstream
