diff -ur Mnemosyne-2.3.1-orig/mnemosyne/pyqt_ui/statistics_wdgts_plotting.py Mnemosyne-2.3.1/mnemosyne/pyqt_ui/statistics_wdgts_plotting.py
--- Mnemosyne-2.3.1-orig/mnemosyne/pyqt_ui/statistics_wdgts_plotting.py	2013-12-03 11:42:02.000000000 +0100
+++ Mnemosyne-2.3.1/mnemosyne/pyqt_ui/statistics_wdgts_plotting.py	2014-09-03 21:17:38.817060365 +0200
@@ -4,6 +4,9 @@
 
 from PyQt4 import QtGui
 
+from matplotlib import use
+use("Qt4Agg")
+
 from matplotlib import rcParams
 from matplotlib.figure import Figure
 from matplotlib.backends.backend_qt4agg import FigureCanvasQTAgg as FigureCanvas
