Build configuration for Arch Linux

diff -Naur src.orig/config.default src/config.default
--- simutrans.orig/config.default	2011-01-25 21:40:41.000000000 +0100
+++ simutrans/config.default	2011-09-27 16:08:36.176581292 +0200
@@ -6,25 +6,25 @@
 #BACKEND = allegro
 #BACKEND = gdi
 #BACKEND = sdl
-#BACKEND = mixer_sdl
+BACKEND = mixer_sdl
 #BACKEND = x11
 #BACKEND = posix
 
 #COLOUR_DEPTH = 0
 #COLOUR_DEPTH = 8
-#COLOUR_DEPTH = 16
+COLOUR_DEPTH = 16
 
 #OSTYPE = amiga
 #OSTYPE = beos
 #OSTYPE = cygwin
 #OSTYPE = freebsd
 #OSTYPE = haiku
-#OSTYPE = linux
+OSTYPE = linux
 #OSTYPE = mingw
 #OSTYPE = mac
 
 #DEBUG = 3    # Level 1-3, higher number means more debug-friendly, see Makefile
-#OPTIMISE = 1 # Add umpteen optimisation flags
+OPTIMISE = 1 # Add umpteen optimisation flags
 #PROFILE = 1
 
 #WITH_REVISION = 1 # adds the revision from svn; required for networkgames
@@ -33,9 +33,9 @@
 # Define these as empty strings, if you don't have the respective config program
 #ALLEGRO_CONFIG = allegro-config
 #PNG_CONFIG     = pkg-config libpng
-#SDL_CONFIG     = sdl-config
+SDL_CONFIG     = sdl-config
 
-#VERBOSE = 1
+VERBOSE = 1
 
 # The following useful conditional compilation flags exist
 #
@@ -63,3 +63,6 @@
 #
 # In order to use the flags, add a line like this: (-Dxxx)
 # FLAGS = -DUSE_C
+ifeq ($(shell uname -m),x86_64)
+  FLAGS += -DUSE_C
+endif
