diff -Naur dataobj/loadsave.cc src/dataobj/loadsave.cc
--- dataobj/loadsave.cc	2012-02-28 21:49:16.000000000 +0100
+++ dataobj/loadsave.cc	2012-03-02 16:55:33.440159003 +0100
@@ -246,7 +246,7 @@
 	const char *success = NULL;
 
 	if(  is_xml()  &&  saving  &&  (!is_bzip2()  ||  fd->bse==BZ_OK)
-	     &&  (is_zipped()  ?  fd->gzfp  :  fd->fp) ) {
+	     &&  (is_zipped()  ?  fd->gzfp  :  (gzFile) fd->fp) ) {
 		// only write when close and no error occurred
 		const char *end = "\n</Simutrans>\n";
 		write( end, strlen(end) );
