diff -uar simutrans-src-110-0/config.default simutrans-src-110-0-new/config.default
--- simutrans-src-110-0/config.default	2011-01-25 14:40:41.000000000 -0600
+++ simutrans-src-110-0-new/config.default	2011-05-20 21:51:32.232261655 -0500
@@ -5,21 +5,21 @@
 
 #BACKEND = allegro
 #BACKEND = gdi
-#BACKEND = sdl
+BACKEND = sdl
 #BACKEND = mixer_sdl
 #BACKEND = x11
 #BACKEND = posix
 
 #COLOUR_DEPTH = 0
 #COLOUR_DEPTH = 8
-#COLOUR_DEPTH = 16
+COLOUR_DEPTH = 16
 
 #OSTYPE = amiga
 #OSTYPE = beos
 #OSTYPE = cygwin
 #OSTYPE = freebsd
 #OSTYPE = haiku
-#OSTYPE = linux
+OSTYPE = linux
 #OSTYPE = mingw
 #OSTYPE = mac
 
@@ -33,7 +33,7 @@
 # Define these as empty strings, if you don't have the respective config program
 #ALLEGRO_CONFIG = allegro-config
 #PNG_CONFIG     = pkg-config libpng
-#SDL_CONFIG     = sdl-config
+SDL_CONFIG     = sdl-config
 
 #VERBOSE = 1
 
@@ -62,4 +62,6 @@
 # DESTINATION_CITYCARS: Citycars can have a destination (not recommended)
 #
 # In order to use the flags, add a line like this: (-Dxxx)
-# FLAGS = -DUSE_C
+ifeq ($(shell uname -m),x86_64)
+FLAGS += -DUSE_C
+endif
Only in simutrans-src-110-0-new: config.default~
diff -uar simutrans-src-110-0/simmain.cc simutrans-src-110-0-new/simmain.cc
--- simutrans-src-110-0/simmain.cc	2011-02-03 15:10:06.000000000 -0600
+++ simutrans-src-110-0-new/simmain.cc	2011-05-20 21:52:58.458923954 -0500
@@ -437,11 +437,13 @@
 #endif
 	{
 		// save the current directories
-		getcwd(umgebung_t::program_dir, lengthof(umgebung_t::program_dir));
-		strcat( umgebung_t::program_dir, path_sep );
+		//getcwd(umgebung_t::program_dir, lengthof(umgebung_t::program_dir));
++		//strcat( umgebung_t::program_dir, path_sep );
++		strcpy(umgebung_t::program_dir, "/usr/share/simutrans/" );
++		chdir(umgebung_t::program_dir);
 	}
 	else {
-		strcpy( umgebung_t::program_dir, argv[0] );
+		strcpy( umgebung_t::program_dir, "/usr/share/simutrans/" );
 		*(strrchr( umgebung_t::program_dir, path_sep[0] )+1) = 0;
 
 		chdir( umgebung_t::program_dir );
Only in simutrans-src-110-0-new: simmain.cc~
diff -uar simutrans-src-110-0/simsys_d.cc simutrans-src-110-0-new/simsys_d.cc
--- simutrans-src-110-0/simsys_d.cc	2011-01-25 14:40:41.000000000 -0600
+++ simutrans-src-110-0-new/simsys_d.cc	2011-05-20 21:53:55.705587723 -0500
@@ -324,7 +324,7 @@
 	return NULL;
 #else
 #ifndef __APPLE__
-	sprintf( buffer, "%s/simutrans", getenv("HOME") );
+	sprintf( buffer, "%s/.simutrans", getenv("HOME") );
 #else
 	sprintf( buffer, "%s/Library/Simutrans", getenv("HOME") );
 #endif
Only in simutrans-src-110-0-new: simsys_d.cc~
diff -uar simutrans-src-110-0/simsys_s.cc simutrans-src-110-0-new/simsys_s.cc
--- simutrans-src-110-0/simsys_s.cc	2011-02-14 14:50:10.000000000 -0600
+++ simutrans-src-110-0-new/simsys_s.cc	2011-05-20 21:54:49.782251651 -0500
@@ -317,7 +317,7 @@
 	return NULL;
 #else
 #ifndef __APPLE__
-	sprintf( buffer, "%s/simutrans", getenv("HOME") );
+	sprintf( buffer, "%s/.simutrans", getenv("HOME") );
 #else
 	sprintf( buffer, "%s/Library/Simutrans", getenv("HOME") );
 #endif
Only in simutrans-src-110-0-new: simsys_s.cc~
diff -uar simutrans-src-110-0/tpl/slist_tpl.h simutrans-src-110-0-new/tpl/slist_tpl.h
--- simutrans-src-110-0/tpl/slist_tpl.h	2010-08-16 15:14:54.000000000 -0500
+++ simutrans-src-110-0-new/tpl/slist_tpl.h	2011-05-20 21:55:23.745583265 -0500
@@ -10,6 +10,7 @@
 
 #include <iterator>
 #include <typeinfo>
+#include <cstddef>
 #include "../dataobj/freelist.h"
 #include "../simdebug.h"
 
Only in simutrans-src-110-0-new/tpl: slist_tpl.h~
