diff --git a/src/buffers.c b/src/buffers.c
index fe0d50a..4a37d58 100644
--- a/src/buffers.c
+++ b/src/buffers.c
@@ -2590,7 +2590,7 @@ substituteText(const char *line)
 	    replaceLine[replaceLineLen] = '\n';
 	    if(!linecount) {
 /* normal substitute */
-		char newnum[LNWIDTH];
+		char newnum[LNWIDTH+1];
 		textLines[textLinesCount] = allocMem(replaceLineLen + 1);
 		memcpy(textLines[textLinesCount], replaceLine,
 		   replaceLineLen + 1);
