--- applet_old.js	2012-09-22 00:29:10.000000000 +0200
+++ applet.js	2013-01-19 18:17:24.981647000 +0100
@@ -97,8 +97,8 @@
     let schema = schemaSource.lookup('org.cinnamon.applets.system-monitor', true);
     Schema = new Gio.Settings({ settings_schema: schema });
     
-    Background = new Clutter.Color();
-    Background.from_string(Schema.get_string('background'));
+    let [res, clutterColor] = new Clutter.Color.from_string(Schema.get_string('background'));
+    Background = clutterColor;
     //IconSize = Math.round(Panel.PANEL_ICON_SIZE * 4 / 5);
     //Constant doesn't exist. Took me ages to figure out WHAT caused Net() to break...
     IconSize = 16;
@@ -226,9 +226,8 @@
 
         this.colors = [];
         for(let color in this.color_name) {
-            let clutterColor = new Clutter.Color();
-            let name = this.elt + '-' + this.color_name[color] + '-color';
-            clutterColor.from_string(Schema.get_string(name));
+            let name = this.elt + '-' + this.color_name[color] + '-color';            
+            let [res, clutterColor] = new Clutter.Color.from_string(Schema.get_string(name));
             Schema.connect('changed::' + name, Lang.bind(
                 clutterColor, function (schema, key) {
                     this.from_string(Schema.get_string(key));
@@ -864,8 +863,7 @@
         // FIXME Handle colors correctly
         this.colors = ["#444", "#666", "#888", "#aaa", "#ccc", "#eee"];
         for(let color in this.colors) {
-            let clutterColor = new Clutter.Color();
-            clutterColor.from_string(this.colors[color]);
+            let [res, clutterColor] = new Clutter.Color.from_string(this.colors[color]);
             this.colors[color] = clutterColor;
         }
     },
