--- ./src/cdemu.cpp.old	2012-01-13 13:19:49.000000000 -0500
+++ ./src/cdemu.cpp	2012-12-29 14:42:23.586371880 -0500
@@ -30,11 +30,11 @@
     connect(&m_watcher, SIGNAL(serviceUnregistered(QString)), this, SLOT(slotServiceUnregistered(QString)));
 
     m_watcher.setConnection(QDBusConnection::sessionBus());
-    m_watcher.addWatchedService("net.sf.cdemu.CDEMUD_Daemon");
+    m_watcher.addWatchedService("net.sf.cdemu.CDEmuDaemon");
 
-    QDBusConnection::sessionBus().interface()->startService("net.sf.cdemu.CDEMUD_Daemon");
-    QDBusConnection::sessionBus().connect("net.sf.cdemu.CDEMUD_Daemon", "/CDEMUD_Daemon",
-                                          "net.sf.cdemu.CDEMUD_Daemon", "DeviceStatusChanged",
+    QDBusConnection::sessionBus().interface()->startService("net.sf.cdemu.CDEmuDaemon");
+    QDBusConnection::sessionBus().connect("net.sf.cdemu.CDEmuDaemon", "/Daemon",
+                                          "net.sf.cdemu.CDEmuDaemon", "DeviceStatusChanged",
                                           this, SLOT(slotDeviceChanged(int)));
 }
 
@@ -46,14 +46,14 @@
 
 bool CDEmu::isDaemonRunning() const
 {
-    return QDBusConnection::sessionBus().interface()->isServiceRegistered("net.sf.cdemu.CDEMUD_Daemon");
+    return QDBusConnection::sessionBus().interface()->isServiceRegistered("net.sf.cdemu.CDEmuDaemon");
 }
 
 
 int CDEmu::deviceCount() const
 {
-    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEMUD_Daemon", "/CDEMUD_Daemon",
-                                                    "net.sf.cdemu.CDEMUD_Daemon", "GetNumberOfDevices");
+    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEmuDaemon", "/Daemon",
+                                                    "net.sf.cdemu.CDEmuDaemon", "GetNumberOfDevices");
 
     QDBusReply<int> reply = QDBusConnection::sessionBus().call(m);
     if (reply.isValid())
@@ -78,8 +78,8 @@
 
 bool CDEmu::isLoaded(unsigned int device) const
 {
-    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEMUD_Daemon", "/CDEMUD_Daemon",
-                                                    "net.sf.cdemu.CDEMUD_Daemon", "DeviceGetStatus");
+    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEmuDaemon", "/Daemon",
+                                                    "net.sf.cdemu.CDEmuDaemon", "DeviceGetStatus");
     m << (int)device;
 
     QDBusMessage reply = QDBusConnection::sessionBus().call(m);
@@ -95,8 +95,8 @@
 
 QString CDEmu::filename(unsigned int device) const
 {
-    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEMUD_Daemon", "/CDEMUD_Daemon",
-                                                    "net.sf.cdemu.CDEMUD_Daemon", "DeviceGetStatus");
+    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEmuDaemon", "/Daemon",
+                                                    "net.sf.cdemu.CDEmuDaemon", "DeviceGetStatus");
     m << (int)device;
 
     QDBusMessage reply = QDBusConnection::sessionBus().call(m);
@@ -132,8 +132,8 @@
 
     QVariantMap parameters; // Unused for now
 
-    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEMUD_Daemon", "/CDEMUD_Daemon",
-                                                    "net.sf.cdemu.CDEMUD_Daemon", "DeviceLoad");
+    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEmuDaemon", "/Daemon",
+                                                    "net.sf.cdemu.CDEmuDaemon", "DeviceLoad");
     m << (int)device << filenames << parameters;
 
     QDBusMessage reply = QDBusConnection::sessionBus().call(m);
@@ -152,8 +152,8 @@
     if ((int)device >= deviceCount())
         return DeviceNotAvailable;
 
-    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEMUD_Daemon", "/CDEMUD_Daemon",
-                                                    "net.sf.cdemu.CDEMUD_Daemon", "DeviceUnload");
+    QDBusMessage m = QDBusMessage::createMethodCall("net.sf.cdemu.CDEmuDaemon", "/Daemon",
+                                                    "net.sf.cdemu.CDEmuDaemon", "DeviceUnload");
     m << (int)device;
 
     QDBusMessage reply = QDBusConnection::sessionBus().call(m);
@@ -166,14 +166,14 @@
 
 void CDEmu::slotServiceRegistered(const QString &service)
 {
-    if (service == "net.sf.cdemu.CDEMUD_Daemon")
+    if (service == "net.sf.cdemu.CDEmuDaemon")
         emit daemonChanged(true);
 }
 
 
 void CDEmu::slotServiceUnregistered(const QString &service)
 {
-    if (service == "net.sf.cdemu.CDEMUD_Daemon")
+    if (service == "net.sf.cdemu.CDEmuDaemon")
         emit daemonChanged(false);
 }
 
