--- old/server/sock.c	2009-07-03 18:28:33.000000000 +0200
+++ new/server/sock.c	2009-07-14 20:38:13.000000000 +0200
@@ -398,6 +398,10 @@
 
         if (event & POLLOUT)
         {
+               /* According to WS2 spec, FD_WRITE is only delivered
+               after connect and WSAEWOULDBLOCK while sending,
+               so we clear FD_WRITE here. */
+            sock->state &= ~FD_WRITE;
             sock->pmask |= FD_WRITE;
             sock->hmask |= FD_WRITE;
             sock->errors[FD_WRITE_BIT] = 0;
