--- /usr/bin/fontpage	2014-10-20 15:53:09.000000000 +0200
+++ /tmp/test	2014-10-20 16:01:31.510344641 +0200
@@ -21,7 +21,8 @@
 ###############################################
 ##  importing with system requirement checks #################
 
-import sys, os, time, commands, string, shutil, re, ConfigParser, Image, ImageChops
+import sys, os, time, commands, string, shutil, re, ConfigParser
+from PIL import Image, ImageChops
 import pygtk
 
 #import gettext 

