diff -Naur netgen-4.9.13_orig/libsrc/occ/Partition_Inter3d.cxx netgen-4.9.13/libsrc/occ/Partition_Inter3d.cxx
--- netgen-4.9.13_orig/libsrc/occ/Partition_Inter3d.cxx	2009-08-24 04:12:24.000000000 +0200
+++ netgen-4.9.13/libsrc/occ/Partition_Inter3d.cxx	2011-06-23 12:12:41.974521692 +0200
@@ -243,7 +243,7 @@
       Standard_Integer i, nbExt = anExtPS.NbExt();
       Extrema_POnSurf aPOnSurf;
       for (i = 1; i <= nbExt; ++i )
-        if (anExtPS.Value( i ) <= TolE) {
+        if (anExtPS.SquareDistance( i ) <= TolE) {
           aPOnSurf = anExtPS.Point( i );
           break;
         }
diff -Naur netgen-4.9.13_orig/libsrc/occ/Partition_Loop2d.cxx netgen-4.9.13/libsrc/occ/Partition_Loop2d.cxx
--- netgen-4.9.13_orig/libsrc/occ/Partition_Loop2d.cxx	2009-08-24 04:12:24.000000000 +0200
+++ netgen-4.9.13/libsrc/occ/Partition_Loop2d.cxx	2011-06-21 14:58:56.000000000 +0200
@@ -22,7 +22,7 @@
 #include <BRepAdaptor_Surface.hxx>
 #include <BRepAlgo_AsDes.hxx>
 #include <BRepAlgo_FaceRestrictor.hxx>
-#include <BRepOffset_DataMapOfShapeReal.hxx>
+#include <TopTools_DataMapOfShapeReal.hxx>
 #include <BRepTopAdaptor_FClass2d.hxx>
 #include <BRep_Builder.hxx>
 #include <BRep_Tool.hxx>
@@ -519,7 +519,7 @@
     DC.Initialize( DegEdge, F );
 
   // avoid intersecting twice the same edge
-  BRepOffset_DataMapOfShapeReal EUMap ( EdgesList.Extent() );
+  TopTools_DataMapOfShapeReal EUMap ( EdgesList.Extent() );
 
   Standard_Real U, f, l;
   BRep_Tool::Range (DegEdge, f, l);
diff -Naur netgen-4.9.13_orig/libsrc/occ/Partition_Spliter.cxx netgen-4.9.13/libsrc/occ/Partition_Spliter.cxx
--- netgen-4.9.13_orig/libsrc/occ/Partition_Spliter.cxx	2009-08-24 04:12:24.000000000 +0200
+++ netgen-4.9.13/libsrc/occ/Partition_Spliter.cxx	2011-06-21 15:05:19.000000000 +0200
@@ -1169,7 +1169,7 @@
           for (; j<=nbj && ok; ++j) {
             if (Extrema.IsMin(j)) {
 	      hasMin = Standard_True;
-              ok = Extrema.Value(j) <= tol;
+              ok = Extrema.SquareDistance(j) <= tol;
 	    }
           }
         }
