diff -up plasma-widget-quickaccess-0.8.1/popupdialog.cpp.fix-address-of-temporary-object-error plasma-widget-quickaccess-0.8.1/popupdialog.cpp
--- plasma-widget-quickaccess-0.8.1/popupdialog.cpp.fix-address-of-temporary-object-error   2009-05-27 21:02:59.000000000 +0200
+++ plasma-widget-quickaccess-0.8.1/popupdialog.cpp 2011-02-17 15:56:42.205288780 +0100
@@ -249,7 +249,7 @@ void PopupDialog::hideEvent ( QHideEvent
   m_backButton->hide();
   if(m_settings->enableDolphinSorting())
   {
-    checkDolphinSorting(&(m_current.url()));
+    checkDolphinSorting(m_current.url());
   }
   QWidget::hideEvent( event );
   emit signal_hide();
@@ -258,7 +258,7 @@ void PopupDialog::hideEvent ( QHideEvent
 void PopupDialog::setStartUrl(const KUrl &url)
 {
   if(m_settings->enableDolphinSorting())
-    checkDolphinSorting(&url);
+    checkDolphinSorting(url);
   if( !m_model->dirLister()->openUrl( url ) )
     kDebug() << "can not open url: " << url;
   connect(m_model->dirLister(), SIGNAL(completed()), this, SLOT(dirListerCompleted()));
@@ -323,7 +323,7 @@ void PopupDialog::slot_open(const QModel
   }
  
   if(m_settings->enableDolphinSorting())
-    checkDolphinSorting(&(m_current.url()));
+    checkDolphinSorting(m_current.url());
  
   if(m_current.isFile() || !m_settings->allowNavigation()) {
     m_current.run();
@@ -332,8 +332,8 @@ void PopupDialog::slot_open(const QModel
   m_label->setFileItem(m_current);
 }
  
-void PopupDialog::checkDolphinSorting(const KUrl *dir) {
-  KConfig conf(dir->path(KUrl::AddTrailingSlash)+".directory");
+void PopupDialog::checkDolphinSorting(const KUrl& dir) {
+  KConfig conf(dir.path(KUrl::AddTrailingSlash)+".directory");
   KConfigGroup cg(&conf, "Dolphin");
   int sortingType = cg.readEntry("Sorting", 0);
   int sortOrder = cg.readEntry("SortOrder", 0);
diff -up plasma-widget-quickaccess-0.8.1/popupdialog.h.fix-address-of-temporary-object-error plasma-widget-quickaccess-0.8.1/popupdialog.h
--- plasma-widget-quickaccess-0.8.1/popupdialog.h.fix-address-of-temporary-object-error 2009-05-27 21:01:52.000000000 +0200
+++ plasma-widget-quickaccess-0.8.1/popupdialog.h   2011-02-17 15:53:46.015491457 +0100
@@ -58,7 +58,7 @@ class PopupDialog : public ResizeDialog
     KFileItem startItem()
       { return m_start; }
        
-    void checkDolphinSorting(const KUrl *dir);
+    void checkDolphinSorting(const KUrl &dir);
  
   signals:
     void signal_hide();

