--- media_build_experimental/linux/drivers/media/Makefile	2013-12-08 09:14:07.655609925 +0100
+++ media_build_experimental/linux/drivers/media/Makefile	2013-12-08 09:14:14.443668504 +0100
@@ -22,7 +22,7 @@
 obj-$(CONFIG_DVB_CORE)  += dvb-core/
 
 # There are both core and drivers at RC subtree - merge before drivers
-obj-y += rc/
+# obj-y += rc/
 
 #
 # Finally, merge the drivers that require the core

