diff --git a/Source/WebKit/qt/WebCoreSupport/DeviceMotionProviderQt.h b/Source/WebKit/qt/WebCoreSupport/DeviceMotionProviderQt.h
index 466490f..667bd3c 100644
--- a/Source/WebKit/qt/WebCoreSupport/DeviceMotionProviderQt.h
+++ b/Source/WebKit/qt/WebCoreSupport/DeviceMotionProviderQt.h
@@ -40,7 +40,7 @@ class DeviceOrientationProviderQt;
 class DeviceMotionProviderQt : public QAccelerometerFilter {
 public:
     DeviceMotionProviderQt();
-    ~DeviceMotionProviderQt();
+    virtual ~DeviceMotionProviderQt();
 
     void setController(DeviceMotionController*);
 
diff --git a/Source/WebKit/qt/WebCoreSupport/DeviceOrientationProviderQt.cpp b/Source/WebKit/qt/WebCoreSupport/DeviceOrientationProviderQt.cpp
index de46d24..84b596f 100644
--- a/Source/WebKit/qt/WebCoreSupport/DeviceOrientationProviderQt.cpp
+++ b/Source/WebKit/qt/WebCoreSupport/DeviceOrientationProviderQt.cpp
@@ -29,6 +29,11 @@ DeviceOrientationProviderQt::DeviceOrientationProviderQt()
     m_lastOrientation = DeviceOrientation::create();
 }
 
+DeviceOrientationProviderQt::~DeviceOrientationProviderQt()
+{
+
+}
+
 void DeviceOrientationProviderQt::setController(DeviceOrientationController* controller)
 {
     if (!controller)
diff --git a/Source/WebKit/qt/WebCoreSupport/DeviceOrientationProviderQt.h b/Source/WebKit/qt/WebCoreSupport/DeviceOrientationProviderQt.h
index d07f0a9..3f58b7d 100644
--- a/Source/WebKit/qt/WebCoreSupport/DeviceOrientationProviderQt.h
+++ b/Source/WebKit/qt/WebCoreSupport/DeviceOrientationProviderQt.h
@@ -37,6 +37,7 @@ namespace WebCore {
 class DeviceOrientationProviderQt : public QRotationFilter {
 public:
     DeviceOrientationProviderQt();
+    virtual ~DeviceOrientationProviderQt();
 
     void setController(DeviceOrientationController*);
 
