diff --git a/setup.py b/setup.py
index 4d4e8c5..d5fe6c4 100755
--- a/setup.py
+++ b/setup.py
@@ -42,8 +42,9 @@ from setuptools import setup
 
 
 # Root directories for the /usr and /var trees.
-USR_ROOT = os.path.join("/", "usr", "local")
-VAR_ROOT = os.path.join("/", "var", "local")
+ROOT = "MAKEPKG_INSTALL_ROOT"
+USR_ROOT = os.path.join(ROOT, "usr")
+VAR_ROOT = os.path.join(ROOT, "var")
 
 
 def do_setup():
@@ -316,7 +317,7 @@ def install():
     dirs = [os.path.join(VAR_ROOT, "log"),
             os.path.join(VAR_ROOT, "cache"),
             os.path.join(VAR_ROOT, "lib"),
-            os.path.join(VAR_ROOT, "run"),
+            os.path.join(ROOT, "run"),  # On Archlinux /var/run is a symlink to /run
             os.path.join(USR_ROOT, "include"),
             os.path.join(USR_ROOT, "share")]
     for _dir in dirs:
