Index: REminiscence-0.1.9/systemstub_sdl.cpp
===================================================================
--- REminiscence-0.1.9.orig/systemstub_sdl.cpp
+++ REminiscence-0.1.9/systemstub_sdl.cpp
@@ -366,12 +366,12 @@ void SystemStub_SDL::processEvents() {
 			if (ev.key.keysym.mod & KMOD_ALT) {
 				if (ev.key.keysym.sym == SDLK_RETURN) {
 					switchGfxMode(!_fullscreen, _scaler);
-				} else if (ev.key.keysym.sym == SDLK_KP_PLUS) {
+				} else if (ev.key.keysym.sym == SDLK_PLUS) {
 					uint8 s = _scaler + 1;
 					if (s < NUM_SCALERS) {
 						switchGfxMode(_fullscreen, s);
 					}
-				} else if (ev.key.keysym.sym == SDLK_KP_MINUS) {
+				} else if (ev.key.keysym.sym == SDLK_MINUS) {
 					int8 s = _scaler - 1;
 					if (_scaler > 0) {
 						switchGfxMode(_fullscreen, s);
@@ -392,9 +392,9 @@ void SystemStub_SDL::processEvents() {
 					_pi.save = true;
 				} else if (ev.key.keysym.sym == SDLK_l) {
 					_pi.load = true;
-				} else if (ev.key.keysym.sym == SDLK_KP_PLUS) {
+				} else if (ev.key.keysym.sym == SDLK_PLUS) {
 					_pi.stateSlot = 1;
-				} else if (ev.key.keysym.sym == SDLK_KP_MINUS) {
+				} else if (ev.key.keysym.sym == SDLK_MINUS) {
 					_pi.stateSlot = -1;
 				} else if (ev.key.keysym.sym == SDLK_r) {
 					_pi.inpRecord = true;
