diff -aur xbic-1.0_qtlib2.orig/src/Makefile xbic-1.0_qtlib2/src/Makefile
--- xbic-1.0_qtlib2.orig/src/Makefile	2000-01-14 08:26:46.000000000 +0500
+++ xbic-1.0_qtlib2/src/Makefile	2012-05-20 07:00:09.538252867 +0600
@@ -3,16 +3,16 @@
 
 CC = g++
 
-MOC = moc
+MOC = /opt/qt2/bin/moc
 
 CFLAGS = -Wall -O6 -fomit-frame-pointer
 #CFLAGS = -Wall -g
 
-INCDIR = -I/usr/lib/qt/include
+INCDIR = -I/opt/qt2/include
 
-LIBDIR = -L/usr/X11R6/lib
+LIBDIR = -L/usr/lib -L/opt/qt2/lib
 
-LIBS = -lqt -lqimgio -lpng -lz
+LIBS = -lqt -lpng -lz
 
 OBJ1 =  battle_main.o factory.o bewertung.o mytogglebutton.o \
 	computerspieler.o depot_window.o depot.o land.o landschaft.o \
