diff -crB pidgin-facebookchat/libfacebook.h pidgin-facebookchat2/libfacebook.h
*** pidgin-facebookchat/libfacebook.h	2010-01-20 05:42:26.315434731 -0500
--- pidgin-facebookchat2/libfacebook.h	2010-01-20 06:24:07.974772505 -0500
***************
*** 22,28 ****
  #define LIBFACEBOOK_H
  
  #define FACEBOOK_PLUGIN_VERSION "1.64"
! #define FACEBOOK_PLUGIN_ID "prpl-bigbrownchunx-facebookim"
  #define FACEBOOK_CAPTCHA_SITE "6LezHAAAAAAAADqVjseQ3ctG3ocfQs2Elo1FTa_a"
  
  #include <glib.h>
--- 22,28 ----
  #define LIBFACEBOOK_H
  
  #define FACEBOOK_PLUGIN_VERSION "1.64"
! #define FACEBOOK_PLUGIN_ID "prpl-facebookim"
  #define FACEBOOK_CAPTCHA_SITE "6LezHAAAAAAAADqVjseQ3ctG3ocfQs2Elo1FTa_a"
  
  #include <glib.h>
