diff -u s3c2410_boot_usb/boot_usb.c s3c2410_boot_usb_orgin/boot_usb.c
--- s3c2410_boot_usb/boot_usb.c	2006-08-07 15:00:27.000000000 +0800
+++ s3c2410_boot_usb_orgin/boot_usb.c	2009-10-13 21:52:43.000000000 +0800
@@ -94,7 +94,7 @@
 	return csum;
 }
 
-#define CHUNK_SIZE 100
+#define CHUNK_SIZE 1024
 static int qt2410_send_file(u_int32_t addr, void *data, u_int32_t len)
 {
 	int ret = 0;
@@ -139,8 +139,10 @@
 #endif
 
 		ret = usb_bulk_write(hdl, QT2410_OUT_EP, cur, remain, 0);
-		if (ret < 0)
+		if (ret < 0) {
+            printf("usb_bulk_write() fail... abort!\n");
 			break;
+        }
 	}
 
 	free(buf);
@@ -172,7 +174,7 @@
 #endif
 
 //#define KERNEL_RAM_BASE	0x30008000
-#define KERNEL_RAM_BASE	0x33F80000
+#define KERNEL_RAM_BASE	0x32000000
 
 int main(int argc, char **argv)
 {
@@ -225,9 +227,11 @@
 	if (!prog)
 		exit(1);
 
-	if (qt2410_send_file(KERNEL_RAM_BASE, prog, st.st_size)) {
+	if (qt2410_send_file(KERNEL_RAM_BASE, prog, st.st_size) < 0) {
 		printf("Error downloading program\n");
 		exit(1);
-	}
+	} else {
+        printf("donwloading program success!\n");
+    }
 	exit(0);
 }
