diff -Nur src/jstest-gtk/src/joystick_list_widget.cpp src_new/jstest-gtk/src/joystick_list_widget.cpp
--- src/jstest-gtk/src/joystick_list_widget.cpp	2013-01-08 19:18:12.605692837 -0200
+++ src_new/jstest-gtk/src/joystick_list_widget.cpp	2013-01-08 19:04:13.426781606 -0200
@@ -123,7 +123,7 @@
   for(std::vector<JoystickDescription>::const_iterator i = joysticks.begin(); i != joysticks.end(); ++i)
     {
       Gtk::ListStore::iterator it = device_list->append();
-      (*it)[DeviceListColumns::instance().icon] = Gdk::Pixbuf::create_from_file("data/generic.png");
+      (*it)[DeviceListColumns::instance().icon] = Gdk::Pixbuf::create_from_file("/usr/share/jstest-gtk/data/generic.png");
       (*it)[DeviceListColumns::instance().path] = i->filename;
 
       std::ostringstream out;
