# Thanks to Michael Stone for the patch

diff --git a/bin/sugar-session b/bin/sugar-session
index cc8358c..b4a68cc 100755
--- a/bin/sugar-session
+++ b/bin/sugar-session
@@ -240,7 +240,8 @@
  
      # this must be added early, so that it executes and unfreezes the screen
      # even when we initially get blocked on the intro screen
-    gobject.idle_add(unfreeze_dcon_cb)
+    if os.path.exists("/sys/devices/platform/dcon"):
+        gobject.idle_add(unfreeze_dcon_cb)
  
      intro.check_profile()
