diff -crB sugar-0.92.3-orig/src/jarabe/desktop/favoritesview.py sugar-0.92.3/src/jarabe/desktop/favoritesview.py
*** sugar-0.92.3-orig/src/jarabe/desktop/favoritesview.py	2011-05-25 15:15:33.000000000 +0200
--- sugar-0.92.3/src/jarabe/desktop/favoritesview.py	2011-07-13 17:24:00.000000000 +0200
***************
*** 665,675 ****
          self.changed = dispatch.Signal()
  
      def get_layout(self):
          return self._layout
  
      def set_layout(self, layout):
          logging.debug('set_layout %r %r', layout, self._layout)
!         if layout != self._layout:
              self._layout = layout
  
              client = gconf.client_get_default()
--- 665,677 ----
          self.changed = dispatch.Signal()
  
      def get_layout(self):
+         if self._layout == None:
+             self._layout = favoriteslayout.RingLayout.key
          return self._layout
  
      def set_layout(self, layout):
          logging.debug('set_layout %r %r', layout, self._layout)
!         if layout != self._layout and layout != None:
              self._layout = layout
  
              client = gconf.client_get_default()
