diff --git a/Makefile b/Makefile
index bae9e6b..95f2695 100644
--- a/Makefile
+++ b/Makefile
@@ -61,16 +61,16 @@ bindings_install: python2_install python3_install
 bindings_clean: python_clean
 
 python2: libfiu
-	cd bindings/python && python setup.py build
+	cd bindings/python && python2 setup.py build
 
 python2_install: python2
-	cd bindings/python && python setup.py install
+	cd bindings/python && python2 setup.py install --prefix=${PREFIX} 
 
 python3: libfiu
 	cd bindings/python && python3 setup.py build
 
 python3_install: python3
-	cd bindings/python && python3 setup.py install
+	cd bindings/python && python3 setup.py install --prefix=${PREFIX} 
 
 python_clean:
 	cd bindings/python && rm -rf build/
diff --git a/preload/posix/generate b/preload/posix/generate
index e7aac43..41f7d6f 100755
--- a/preload/posix/generate
+++ b/preload/posix/generate
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Reads function information and generates code for the preloader library.
