diff -Naur blue~/plugins/graphics/png/png.c blue/plugins/graphics/png/png.c
--- blue~/plugins/graphics/png/png.c	2013-01-09 20:54:17.324599005 -0600
+++ blue/plugins/graphics/png/png.c	2013-01-09 20:55:29.816434967 -0600
@@ -68,13 +68,13 @@
 
     Image self = obj->value.vptr;
 
-    self->width = info->width;
+    self->width = png_get_image_width(self, info);
 
-    self->height = info->height;
+    self->height = png_get_image_height(self, info);
 
-	width = info->width;
+	width = png_get_image_width(self, info);
 
-	height = info->height;
+	height = png_get_image_height(self, info);
 
     //info->bit_depth;
     //info->channels;
@@ -87,7 +87,7 @@
 
     /* allocate memory for rows */
 	row_pointers = malloc(sizeof(png_bytep) * height);
-	for (y=0; y<height; y++) row_pointers[y] = malloc(info->rowbytes);
+	for (y=0; y<height; y++) row_pointers[y] = malloc(png_get_image_rowbytes(self, info));
 
     /* read image into allocated memory */
 	png_read_image(png, row_pointers);
@@ -96,7 +96,7 @@
 
     png_byte * pdata;
 
-    self->format = info->channels;
+    self->format = png_get_image_channels(self, info);
     self->data = pdata = malloc(self->width * self->height * self->format);
     for (y=0; y<height; y++) {
         row = row_pointers[height-y-1];
