--- etc/pacman.conf.orig	2010-06-18 15:47:20.000000000 -0430
+++ etc/pacman.conf		2010-06-17 05:45:18.000000000 -0430
@@ -53,15 +53,30 @@
 # uncommented to enable the repo.
 #
 
-# An example of a disabled remote package repository with multiple servers
-# available. To enable, uncomment the following lines. You can add preferred
-# servers immediately after the header and they will be used before the
-# default mirrors.
-#[core]
-#Server = ftp://ftp.example.com/foobar/$repo/os/$arch/
-# The file referenced here should contain a list of 'Server = ' lines.
+# The testing repositories are disabled by default. To enable, uncomment the
+# repo name header and Include lines. You can add preferred servers immediately
+# after the header, and they will be used before the default mirrors.
+
+#[testing]
+## Add your preferred servers here, they will be used first
+#Include = /etc/pacman.d/mirrorlist
+
+[core]
+# Add your preferred servers here, they will be used first
+Include = /etc/pacman.d/mirrorlist
+
+[extra]
+# Add your preferred servers here, they will be used first
+Include = /etc/pacman.d/mirrorlist
+
+#[community-testing]
+## Add your preferred servers here, they will be used first
 #Include = /etc/pacman.d/mirrorlist
 
+[community]
+# Add your preferred servers here, they will be used first
+Include = /etc/pacman.d/mirrorlist
+
 # An example of a custom package repository.  See the pacman manpage for
 # tips on creating your own repositories.
 #[custom]
