--- yle-dl-1.99.9/Makefile	2012-06-10 17:21:12.000000000 +0300
+++ yle-dl-1.99.9-arch/Makefile	2012-06-13 00:15:05.759600908 +0300
@@ -1,4 +1,4 @@
-prefix?=/usr/local
+prefix=/usr
 BINDIR=$(DESTDIR)/$(prefix)/bin
 
 all:
diff -Naur yle-dl-1.99.9/yle-dl yle-dl-1.99.9-arch/yle-dl
--- yle-dl-1.99.9/yle-dl	2012-06-10 17:21:12.000000000 +0300
+++ yle-dl-1.99.9-arch/yle-dl	2012-06-13 00:14:44.726728450 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """
