diff --git a/README b/README
index 6a43496..b54ad44 100644
--- a/README
+++ b/README
@@ -6,9 +6,8 @@ Source code: https://github.com/aajanki/yle-dl
 
 This is a plugin for librtmp for downloading media files from the
 video streaming services of the Finnish national broadcasting company
-Yle: Yle Areena (http://areena-beta.yle.fi/ng/areena,
-http://ylex.yle.fi/ylex-areena) and Elävä Arkisto
-(http://www.yle.fi/elavaarkisto/).
+Yle: Yle Areena (http://areena.yle.fi, http://ylex.yle.fi/ylex-areena)
+and Elävä Arkisto (http://www.yle.fi/elavaarkisto/).
 
 Installation
 ------------
@@ -73,6 +72,6 @@ Firewall must allow outgoing traffic on ports 80 and 1935.
 Examples
 --------
 
-yle-dl http://areena-beta.yle.fi/ng/areena/tv/1373722/#/play -o video.flv
+yle-dl http://areena.yle.fi/tv/1544491 -o video.flv
 
 yle-dl "http://www.yle.fi/elavaarkisto/?s=s&g=4&ag=28&t=&a=9390"
diff --git a/README.fi b/README.fi
index a768ba2..5defd38 100644
--- a/README.fi
+++ b/README.fi
@@ -5,9 +5,8 @@ Kotisivu: http://www.iki.fi/aoa/rtmpdump-yle
 Lähdekoodi: https://github.com/aajanki/yle-dl
 
 Tämä on librtmp-kirjaston plugin tiedostojen lataamiseen (uudesta) Yle
-Areenasta (http://areena-beta.yle.fi/ng/areena,
-http://ylex.yle.fi/ylex-areena) ja Elävästä arkistosta
-(http://www.yle.fi/elavaarkisto/).
+Areenasta (http://areena.yle.fi, http://ylex.yle.fi/ylex-areena) ja
+Elävästä arkistosta (http://www.yle.fi/elavaarkisto/).
 
 Asennus
 -------
@@ -112,14 +111,14 @@ Selvitä Areena-videon osoite komennolla "yle-dl --showurl" ja syötä
 komennon tulostama osoite mplayerille tai muulle librtmp-pohjaiselle
 ohjelmalle. Esimerkiksi:
 
-mplayer "`yle-dl --showurl http://areena-beta.yle.fi/ng/areena/tv/1373722/`"
+mplayer "`yle-dl --showurl http://areena.yle.fi/tv/1544491`"
 
 Esimerkkejä
 -----------
 
 * Areenan ohjelman lataaminen automaattisesti nimettävään tiedostoon:
 
-  yle-dl http://areena-beta.yle.fi/ng/areena/tv/1373722/
+  yle-dl http://areena.yle.fi/tv/1544491
 
 * Elävän arkiston ohjelman kaikkien jaksojen lataaminen:
 
@@ -127,12 +126,12 @@ Esimerkkejä
 
 * Ohjelman lataaminen tiedostoon uutiset.flv:
 
-  yle-dl http://areena-beta.yle.fi/ng/areena/tv/1377212/ -o uutiset.flv
+  yle-dl http://areena.yle.fi/tv/1544491 -o uutiset.flv
 
 * Keskeytyneen latauksen jatkaminen tiedostoon uutiset.flv:
 
-  yle-dl http://areena-beta.yle.fi/ng/areena/tv/1377212/ -o uutiset.flv --resume
+  yle-dl http://areena.yle.fi/tv/1544491 -o uutiset.flv --resume
 
 * Lataa uusimmat uutiset
 
-  yle-dl --latestepisode http://areena-beta.yle.fi/ng/areena?q=uutiset
+  yle-dl --latestepisode http://areena.yle.fi/?q=uutiset
diff --git a/yle-dl b/yle-dl
index 65eb43e..f8803a4 100755
--- a/yle-dl
+++ b/yle-dl
@@ -7,10 +7,9 @@ yle-dl - rtmpdump frontend for Yle Areena, Elävä Arkisto and YleX Areena
 Copyright (C) 2010-2012 Antti Ajanki <antti.ajanki@iki.fi>
 
 This script extracts RTMP stream information from Yle Areena
-(http://areena-beta.yle.fi/ng/areena/tv), YleX Areena
-(http://ylex.yle.fi/ylex-areena), Elävä Arkisto
-(http://yle.fi/elavaarkisto/index.html) web pages and calls rtmpdump
-with correct parameters.
+(http://areena.yle.fi), YleX Areena (http://ylex.yle.fi/ylex-areena),
+Elävä Arkisto (http://yle.fi/elavaarkisto/index.html) web pages and
+calls rtmpdump with correct parameters.
 """
 
 import sys
@@ -33,9 +32,9 @@ version = "1.99.6 (2.0 beta 6)"
 AREENA_NG_RTMP = 'rtmp://flashk.yle.fi/mediakanta//x'
 # x above is a placeholder, yle plugin will construct the real playpath.
 # Two slashes after mediakanta is similar to what YLE player does.
-AREENA_NG_SWF = 'http://areena-beta.yle.fi/static/player/1.1/flowplayer/flowplayer.cluster-3.2.3.swf'
+AREENA_NG_SWF = 'http://static.yle.fi/global/player/version/1.2.7/flowplayer/flowplayer.commercial-3.2.7-encrypted.swf'
 AREENA_NG_TCURL = 'rtmp://flashk.yle.fi/mediakanta/'
-AREENA_NG_CLIP_TEMPLATE = 'http://areena-beta.yle.fi/ng/areena/%s/%s/#/play'
+AREENA_NG_CLIP_TEMPLATE = 'http://areena.yle.fi/%s/%s'
 AREENA_NG_HTTP_HEADERS = {'User-Agent': 'yle-dl/' + version.split(' ')[0]}
 
 AREENA_LIVE_SWF = 'http://areena.yle.fi/player/Application.swf?build=2'
