--- Settings.cs	2011-09-13 17:45:50.706897029 +0200
+++ Settings.cs.new	2011-09-13 17:39:27.555639405 +0200
@@ -49,7 +49,7 @@
 		{
 			if (!supertuxdatadir_cached)
 			{
-				supertuxdatadir = GuessDataDir();
+				supertuxdatadir = "/usr/share/games/supertux2";
 				supertuxdatadir_cached = true;
 			}
 			return supertuxdatadir;
@@ -200,7 +200,7 @@
 			String supertux_datadir_best = null;
 			while (!String.IsNullOrEmpty(supertux_datadir = supertux_process.StandardOutput.ReadLine())) {
 				if(System.IO.Path.IsPathRooted(supertux_datadir)) {
-					supertux_datadir_best = supertux_datadir;
+					supertux_datadir_best = "/usr/share/games/supertux2";
 				}
 			}
 
