*** a/qmmp-foou/librarys/qtsingleapplication/src/qtlocalpeer.cpp	2012-06-26 15:34:44.931539749 +0800
--- b/qmmp-foou/librarys/qtsingleapplication/src/qtlocalpeer.cpp	2012-06-26 15:32:36.618210836 +0800
***************
*** 44,50 ****
  ** 
  ****************************************************************************/
  
! 
  #include "qtlocalpeer.h"
  #include <QtCore/QCoreApplication>
  #include <QtCore/QTime>
--- 44,50 ----
  ** 
  ****************************************************************************/
  
! #include <unistd.h>
  #include "qtlocalpeer.h"
  #include <QtCore/QCoreApplication>
  #include <QtCore/QTime>
