diff -ru instantbird-0.3a2-src.orig/configure.in instantbird-0.3a2-src/configure.in
--- instantbird-0.3a2-src.orig/configure.in	2011-05-24 22:16:18.000000000 -0400
+++ instantbird-0.3a2-src/configure.in	2011-05-24 22:21:49.000000000 -0400
@@ -4277,7 +4277,7 @@
 if test -n "$MOZ_NATIVE_NSS"; then
    NSS_LIBS="$NSS_LIBS -lcrmf"
 else
-   NSS_CFLAGS='-I$(LIBXUL_DIST)/include/nss'
+   NSS_CFLAGS=`$NSS_CONFIG $nss_config_args --cflags`
    NSS_DEP_LIBS="\
         \$(LIBXUL_DIST)/lib/\$(LIB_PREFIX)crmf.\$(LIB_SUFFIX) \
         \$(LIBXUL_DIST)/lib/\$(DLL_PREFIX)smime$NSS_VERSION\$(DLL_SUFFIX) \
diff -ru instantbird-0.3a2-src.orig/mozilla/configure.in instantbird-0.3a2-src/mozilla/configure.in
--- instantbird-0.3a2-src.orig/mozilla/configure.in	2011-05-24 22:17:28.000000000 -0400
+++ instantbird-0.3a2-src/mozilla/configure.in	2011-05-24 22:22:26.000000000 -0400
@@ -4784,7 +4784,7 @@
 if test -n "$MOZ_NATIVE_NSS"; then
    NSS_LIBS="$NSS_LIBS -lcrmf"
 else
-   NSS_CFLAGS='-I$(LIBXUL_DIST)/include/nss'
+   NSS_CFLAGS=`$NSS_CONFIG $nss_config_args --cflags`
    NSS_DEP_LIBS="\
         \$(LIBXUL_DIST)/lib/\$(LIB_PREFIX)crmf.\$(LIB_SUFFIX) \
         \$(LIBXUL_DIST)/lib/\$(DLL_PREFIX)smime$NSS_VERSION\$(DLL_SUFFIX) \
diff -ru instantbird-0.3a2-src.orig/purple/libpurple/Makefile.in instantbird-0.3a2-src/purple/libpurple/Makefile.in
--- instantbird-0.3a2-src.orig/purple/libpurple/Makefile.in	2011-05-24 22:19:16.000000000 -0400
+++ instantbird-0.3a2-src/purple/libpurple/Makefile.in	2011-05-24 22:22:47.000000000 -0400
@@ -51,7 +51,7 @@
 VISIBILITY_FLAGS=
 
 ifneq (,$(filter-out WINNT WINCE Darwin,$(OS_ARCH)))
-LOCAL_INCLUDES	= $(GLIB_CFLAGS) -I/usr/include/libxml2/ -I/usr/local/include/libxml2/
+LOCAL_INCLUDES	= $(NSS_CFLAGS) $(GLIB_CFLAGS) -I/usr/include/libxml2/ -I/usr/local/include/libxml2/
 else
 LOCAL_INCLUDES  = -I$(DIST)/include/glib
 
