--- instantbird-0.2-src.orig/mozilla/configure.in	2010-10-12 10:44:37.000000000 -0700
+++ instantbird-0.2-src/mozilla/configure.in	2010-10-19 09:51:49.660019364 -0700
@@ -811,7 +811,7 @@
     AC_MSG_RESULT([yes])    
 fi
 
-MOZ_PATH_PROGS(PYTHON, $PYTHON python2.5 python2.4 python)
+MOZ_PATH_PROGS(PYTHON, $PYTHON python2.7 python2)
 if test -z "$PYTHON"; then
     AC_MSG_ERROR([python was not found in \$PATH])
 fi
--- instantbird-0.2-src.orig/configure.in	2010-10-12 10:44:37.000000000 -0700
+++ instantbird-0.2-src/configure.in	2010-10-19 09:51:49.660019364 -07--
@@ -811,7 +811,7 @$
     AC_MSG_RESULT([yes])
 fi

-AC_PATH_PROGS(PYTHON, $PYTHON python)
+AC_PATH_PROGS(PYTHON, $PYTHON python2.7 python2)
 if test -z "$PYTHON"; then
     AC_MSG_ERROR([python was not found in \$PATH])
 fi
--- instantbird-0.2-src.orig/mozilla/client.mk	2011-02-19 19:57:43.000000000 -0500
+++ instantbird-0.2-src/mozilla/client.mk	2011-02-19 19:57:51.000000000 -0500
@@ -103,7 +103,7 @@
 MAKE := gmake
 endif
 PERL ?= perl
-PYTHON ?= python
+PYTHON ?= python2
 
 RUN_AUTOCONF_LOCALLY = 1
 CONFIG_GUESS_SCRIPT := $(wildcard $(TOPSRCDIR)/build/autoconf/config.guess)
--- instantbird-0.2-src.orig/client.mk	2011-02-19 19:57:43.000000000 -0500
+++ instantbird-0.2-src/client.mk	2011-02-19 19:57:51.000000000 -0500
@@ -105,7 +105,7 @@
 MAKE := gmake
 endif
 PERL ?= perl
-PYTHON ?= python
+PYTHON ?= python2
 
 CONFIG_GUESS_SCRIPT := $(wildcard $(TOPSRCDIR)/mozilla/build/autoconf/config.guess)
 ifdef CONFIG_GUESS_SCRIPT
