diff --git a/Makefile b/Makefile
index bff8de7..caa0e16 100644
--- a/Makefile
+++ b/Makefile
@@ -11,9 +11,12 @@ install: all
 	mkdir -p ${DESTDIR}${PREFIX}/bin
 	install tdm tdmctl ${DESTDIR}${PREFIX}/bin/
 	cp -Rv share ${DESTDIR}${PREFIX}/
+	install -Dm644 tdm_comp.sh ${DESTDIR}/usr/share/bash-completion/completions/tdmctl
 
 uninstall:
-	rm ${PREFIX}/bin/{tdm,tdmctl}
+	rm -f ${PREFIX}/bin/{tdm,tdmctl}
+	rm -rf ${PREFIX}/share/tdm
+	rm -f /usr/share/bash-completion/completions/tdmctl
 	
 clean:
 	rm hprefix tdmctl
diff --git a/tdm_comp.sh b/tdm_comp.sh
new file mode 100644
index 0000000..d58796d
--- /dev/null
+++ b/tdm_comp.sh
@@ -0,0 +1,34 @@
+_tdmctl() 
+{
+	local cur prev opts base
+	COMPREPLY=()
+	cur="${COMP_WORDS[COMP_CWORD]}"
+	prev="${COMP_WORDS[COMP_CWORD-1]}"
+
+	#
+	#  The basic options we'll complete.
+	#
+	opts="init list add cache check default enable disable"
+
+	#
+	#  Complete the arguments to some of the basic commands.
+	#
+	case "${prev}" in
+		check|default|disable)
+			local sessions=$(tdmctl list)
+			COMPREPLY=( $(compgen -W "${sessions}" -- ${cur}) )
+			return 0
+			;;
+		enable)
+			local sessions=$(tdmctl cache)
+			COMPREPLY=( $(compgen -W "${sessions}" -- ${cur}) )
+			return 0
+			;;
+		*)
+			;;
+	esac
+
+	COMPREPLY=($(compgen -W "${opts}" -- ${cur}))  
+	return 0
+}
+complete -F _tdmctl tdmctl
diff --git a/tdmctl.sh b/tdmctl.sh
index 40bd028..ed56573 100755
--- a/tdmctl.sh
+++ b/tdmctl.sh
@@ -4,7 +4,9 @@ usage(){
 	echo "tdmctl cache: list cached files."
 	echo "tdmctl check <session>: see what <session> is."
 	echo "tdmctl default [session]: show/set default X session."
+	echo "tdmctl add <name> <path> [X(default)/extra]: add a session."
 	echo "tdmctl enable/disable <session>: enable/disable session."
+	exit
 }
 
 check(){
@@ -56,6 +58,16 @@ case "$1" in
 			exit 1
 		fi
 		;;
+	add)
+		[ -n "$3" ]||usage
+		if [[ "$4" == "X" || "$4" == "" ]]; then
+			ln -s "$3" "${CONFDIR}/sessions/$2"
+		elif [ "$4" == "extra" ]; then
+			ln -s "$3" "${CONFDIR}/extra/$2"
+		else
+			usage
+		fi
+		;;
 	enable)
 		if [ -f "${CACHEDIR}/X$2" ]; then
 			cp -v "${CACHEDIR}/X$2" "${CONFDIR}/sessions/$2"
