--- m-tsudo-pt3-5d78838/Makefile	2012-08-30 20:58:04.000000000 +0900
+++ m-tsudo-pt3-5d78838/Makefile.new	2012-09-18 06:25:35.927378852 +0900
@@ -3,14 +3,17 @@
 REL_VERSION = "0.0.1"
 REL_DATE = "2012-07-07"
 EXTRA_CFLAGS += -Wformat=2
+PREFIX=/usr
+LIBDIR=$(PREFIX)/lib
+UDEVPREFIX=/etc
 
 all: ${TARGET}
 
 pt3_drv.ko: pt3_pci.c pt3_bus.c pt3_i2c.c pt3_tc.c pt3_qm.c pt3_mx.c pt3_dma.c version.h
-	make -C /lib/modules/`uname -r`/build M=`pwd` V=$(VERBOSITY) modules
+	make -C $(LIBDIR)/modules/`uname -r`/build M=`pwd` V=$(VERBOSITY) modules
 
 clean:
-	make -C /lib/modules/`uname -r`/build M=`pwd` V=$(VERBOSITY) clean
+	make -C $(LIBDIR)/modules/`uname -r`/build M=`pwd` V=$(VERBOSITY) clean
 
 obj-m := pt3_drv.o
 
@@ -27,8 +30,7 @@
 	fi
 
 install: $(TARGET)
-	install -m 644 $(TARGET) /lib/modules/`uname -r`/kernel/drivers/video
-	if [ -d /etc/udev/rules.d -a ! -f /etc/udev/rules.d/99-pt3.rules ] ; then \
-		install -m 644 etc/99-pt3.rules /etc/udev/rules.d ; \
+	install -m 644 $(TARGET) $(LIBDIR)/modules/`uname -r`/kernel/drivers/video
+	if [ -d $(UDEVPREFIX)/udev/rules.d -a ! -f $(UDEVPREFIX)/udev/rules.d/99-pt3.rules ] ; then \
+		install -m 644 etc/99-pt3.rules $(UDEVPREFIX)/udev/rules.d ; \
 	fi
-	depmod -a
