--- os_dep/linux/usb_intf.c.orig	2011-01-26 10:13:11.000000000 -0200
+++ os_dep/linux/usb_intf.c	2011-05-18 20:28:36.997804589 -0300
@@ -914,8 +914,6 @@
 	if( padapter->registrypriv.power_mgnt != PS_MODE_ACTIVE )
 	{
 		if(padapter->registrypriv.usbss_enable ){ 	/* autosuspend (2s delay) */
-			pdvobjpriv->pusbdev->autosuspend_delay = 0 * HZ;//15 * HZ; idle-delay time		 	
-
 			#if (LINUX_VERSION_CODE>=KERNEL_VERSION(2,6,35))
 			usb_enable_autosuspend(padapter->dvobjpriv.pusbdev);
 			#elif  (LINUX_VERSION_CODE>=KERNEL_VERSION(2,6,22) && LINUX_VERSION_CODE<=KERNEL_VERSION(2,6,34))
