diff -Naur old/opt/ankama/dofus/transition.conf new/opt/ankama/dofus/transition.conf
--- old/opt/ankama/dofus/transition.conf	2013-08-14 11:30:17.124450778 -0400
+++ new/opt/ankama/dofus/transition.conf	2013-08-14 11:28:59.719276754 -0400
@@ -6,3 +6,9 @@
 
 include "share/dofus.ici"
 
+# Use adl instead of Air binaries on Arch Linux
+dofus.reg.path = "${root}/share/reg/bin/air-generic-launcher.sh"
+launcher.command = """
+"${root}/bin/air-generic-launcher.sh" --lang=${i18n.lang} --update-server-port=${service_port} --updater_version=v2
+"""
+
