Index: src/ffgtk-0.8.4/configure.ac
===================================================================
--- src/ffgtk-0.8.4/configure.ac	(Revision 1809)
+++ src/ffgtk-0.8.4/configure.ac	(Arbeitskopie)
@@ -2,7 +2,7 @@
 AC_INIT(ffgtk, 0.8.4, http://www.tabos.org/ffgtk)
 AC_PREFIX_DEFAULT(/usr)
 AM_INIT_AUTOMAKE
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 # Check for compilers.
 AC_PROG_CC

