diff -ur a/hamlib-1.2.12/configure b/hamlib-1.2.12/configure
--- a/hamlib-1.2.12/configure	2010-09-04 00:25:46.000000000 +0300
+++ b/hamlib-1.2.12/configure	2011-03-11 23:31:53.000000000 +0200
@@ -19707,7 +19707,7 @@
 	if test -z "$PYTHON_EXTRA_LIBS"; then
 	   PYTHON_EXTRA_LIBS=`$PYTHON -c "import distutils.sysconfig; \
                 conf = distutils.sysconfig.get_config_var; \
-                conf('LIBS'))"`
+                print (conf('LIBS'))"`
 	fi
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $PYTHON_EXTRA_LIBS" >&5
 $as_echo "$PYTHON_EXTRA_LIBS" >&6; }
