--- murrine-engine/autogen.sh	2012-05-14 00:41:50.597959910 +0200
+++ murrine-engine-new/autogen.sh	2012-05-14 00:43:51.780444920 +0200
@@ -16,7 +16,10 @@
 	DIE=1
 }
 
-if automake-1.11 --version < /dev/null > /dev/null 2>&1 ; then
+if automake-1.12 --version < /dev/null > /dev/null 2>&1 ; then
+	AUTOMAKE=automake-1.12
+	ACLOCAL=aclocal-1.12
+elif automake-1.11 --version < /dev/null > /dev/null 2>&1 ; then
 	AUTOMAKE=automake-1.11
 	ACLOCAL=aclocal-1.11
 elif automake-1.10 --version < /dev/null > /dev/null 2>&1 ; then
