diff -aur sleepyhead.orig/sleepyhead/sleepyhead.pro sleepyhead/sleepyhead/sleepyhead.pro
--- sleepyhead.orig/sleepyhead/sleepyhead.pro	2014-09-21 10:18:46.279864977 -0500
+++ sleepyhead/sleepyhead/sleepyhead.pro	2014-09-21 10:23:15.949872696 -0500
@@ -4,7 +4,7 @@
 #
 #-------------------------------------------------
 
-QT += core gui network xml
+QT += core gui network xml printsupport
 
 greaterThan(QT_MAJOR_VERSION,4) {
     QT += widgets webkitwidgets
@@ -65,7 +65,7 @@
 
 
 unix:!macx {
-    LIBS        += -lX11 -lz -lGLU
+    LIBS        += -lX11 -lz
     DEFINES         += _TTY_POSIX_
 }
 
