diff -aur a/modules/access/rdp.c b/modules/access/rdp.c
--- a/modules/access/rdp.c	2015-01-26 17:03:07.912143945 -0500
+++ b/modules/access/rdp.c	2015-01-26 17:05:14.985473923 -0500
@@ -428,8 +428,6 @@
     if ( p_sys->f_fps <= 0 ) p_sys->f_fps = 1.0;
     p_sys->i_frame_interval = 1000000 / p_sys->f_fps;
 
-    freerdp_channels_global_init();
-
     p_sys->p_instance = freerdp_new();
     if ( !p_sys->p_instance )
     {
@@ -504,7 +502,6 @@
 
     freerdp_disconnect( p_sys->p_instance );
     freerdp_free( p_sys->p_instance );
-    freerdp_channels_global_uninit();
 
     if ( p_sys->p_block )
         block_Release( p_sys->p_block );
