--- kdocker-4.6.orig/solutions/qtsingleapplication/qtlocalpeer.cpp	2011-05-30 05:05:45.000000000 +0200
+++ kdocker-4.6/solutions/qtsingleapplication/qtlocalpeer.cpp	2012-04-30 10:10:07.731231805 +0200
@@ -100,7 +100,7 @@
         socketName += QLatin1Char('-') + QString::number(sessionId, 16);
     }
 #else
-    socketName += QLatin1Char('-') + QString::number(::getuid(), 16);
+    socketName += QLatin1Char('-') + QString::number(QtLP_Private::getuid(), 16);
 #endif
 
     server = new QLocalServer(this);
