diff -u -r cirnosay/src/cirnosay.cpp cirnosay-arch/src/cirnosay.cpp
--- cirnosay/src/cirnosay.cpp	2012-04-30 04:04:06.136061488 +0400
+++ cirnosay-arch/src/cirnosay.cpp	2012-04-30 04:06:04.502730765 +0400
@@ -21,8 +21,8 @@
 	int x = 30, y = 9;
 	int width = -1;
 	std::wstring text;
-	std::string image_file = BUILD_DIR "../pictures/l_Touhoudex_Chibi_Cirno.png";
-	std::string palette_file = BUILD_DIR "../pal.png";
+	std::string image_file = "/usr/share/cirnosay/pictures/l_Touhoudex_Chibi_Cirno.png";
+	std::string palette_file = "/usr/share/cirnosay/pal.png";
 };
 
 void show_palette()
