--- a/drivers/usb/core/message.c	2014-10-04 11:04:44.783785005 -0400
+++ b/drivers/usb/core/message.c	2014-10-04 11:05:28.167029025 -0400
@@ -770,9 +770,7 @@
 		dev->string_langid = 0x0409;
 		dev->have_langid = 1;
 		dev_err(&dev->dev,
-			"string descriptor 0 malformed (err = %d), "
-			"defaulting to 0x%04x\n",
-				err, dev->string_langid);
+			"language id specifier not provided by device, defaulting to English\n");
 		return 0;
 	}
 
