diff -rup freepv-0.3.0/CMakeLists.txt freepv-0.3.0.new//CMakeLists.txt
--- freepv-0.3.0/CMakeLists.txt	2009-05-02 01:12:31.000000000 +0200
+++ freepv-0.3.0.new//CMakeLists.txt	2011-08-14 12:02:57.462015099 +0200
@@ -89,7 +89,7 @@ find_package(JPEG REQUIRED)
 find_package(FREEGLUT REQUIRED)
 find_package(OpenGL REQUIRED)
 find_package(X11 REQUIRED)
-find_package(GECKO REQUIRED)
+#find_package(GECKO REQUIRED)
 find_package(LibXml2 REQUIRED)
 find_package(PNG REQUIRED)
 find_package(XF86VM REQUIRED)
diff -rup freepv-0.3.0/src/CMakeLists.txt freepv-0.3.0.new//src/CMakeLists.txt
--- freepv-0.3.0/src/CMakeLists.txt	2009-05-02 01:12:31.000000000 +0200
+++ freepv-0.3.0.new//src/CMakeLists.txt	2011-08-14 12:03:10.395160335 +0200
@@ -5,7 +5,7 @@ add_subdirectory (libfreepv)
 add_subdirectory (freepv-glut)
 add_subdirectory (utils)
 if(UNIX)
-	add_subdirectory (freepv-mozilla)
+#	add_subdirectory (freepv-mozilla)
 	add_subdirectory (freepv-glx)
 endif(UNIX)
 
