--- a/process/operations.hpp		2009-04-20 15:10:02.000000000 +0400
+++ b/process/operations.hpp		2011-06-13 20:47:38.275078401 +0400
@@ -42,7 +42,7 @@
 #include <boost/process/detail/file_handle.hpp> 
 #include <boost/process/detail/pipe.hpp> 
 #include <boost/process/detail/stream_info.hpp> 
-#include <boost/filesystem/path.hpp> 
+#include <boost/filesystem.hpp> 
 #include <boost/algorithm/string/predicate.hpp> 
 #include <boost/system/system_error.hpp> 
 #include <boost/throw_exception.hpp> 
