diff -Nuar a/configure.ac b/configure.ac
--- a/configure.ac	2013-02-01 22:36:42.767817072 +0000
+++ b/configure.ac	2013-02-01 22:38:17.297323063 +0000
@@ -5,7 +5,7 @@
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 AC_CONFIG_SRCDIR(alacarte.in)
 AC_CONFIG_HEADERS([config.h])
-AM_INIT_AUTOMAKE([1.7 no-dist-gzip dist-xz])
+AM_INIT_AUTOMAKE([1.7])
 AM_MAINTAINER_MODE
 
 dnl put the ACLOCAL flags in the makefile
