diff -Nuar a/Alacarte/util.py b/Alacarte/util.py
--- a/Alacarte/util.py	2013-06-17 05:34:24.000000000 +0000
+++ b/Alacarte/util.py	2013-06-17 05:35:28.000000000 +0000
@@ -22,7 +22,7 @@
 from gi.repository import Gtk, GdkPixbuf, GMenu, GLib
 
 # XXX: look into pygobject error marshalling
-from gi._glib import GError
+from gi.repository.GLib import GError
 
 DESKTOP_GROUP = GLib.KEY_FILE_DESKTOP_GROUP
 KEY_FILE_FLAGS = GLib.KeyFileFlags.KEEP_COMMENTS | GLib.KeyFileFlags.KEEP_TRANSLATIONS
