--- netgen-code_org/ng/Makefile	2014-01-29 12:29:32.508509238 -0500
+++ netgen-code/ng/Makefile	2014-01-29 12:29:46.161901138 -0500
@@ -273,7 +273,7 @@
 JPEGLIB_INCLUDES = -DJPEGLIB
 JPEGLIB_LIBS = -ljpeg
 LD = /usr/bin//ld -m elf_x86_64
-LDFLAGS = -Wl,-O1,--sort-common,--as-needed,-z,relro
+LDFLAGS = -Wl,-O1,--sort-common,--as-needed,-z,relro -lX11 -ldl
 LIBGLU = -lGLU
 LIBOBJS = 
 LIBS = -lpthread 
