diff -Naur netgen-code_org/ng/ngpkg.cpp netgen-code/ng/ngpkg.cpp
--- netgen-code_org/ng/ngpkg.cpp	2013-08-15 23:11:35.555992326 +0200
+++ netgen-code/ng/ngpkg.cpp	2013-08-15 23:19:38.981057953 +0200
@@ -2260,7 +2260,7 @@
     static int gopsize = DEFAULT_GOP_SIZE;
     static int bframes = DEFAULT_B_FRAMES;
     static int MPGbufsize = DEFAULT_MPG_BUFSIZE;
-    static CodecID codec_id = CODEC_ID_MPEG1VIDEO;
+    static AVCodecID codec_id = AV_CODEC_ID_MPEG1VIDEO;
     static FILE *MPGfile;
     static buffer_t buff;
     static struct SwsContext *img_convert_ctx;
@@ -2442,7 +2442,7 @@
 
         // Add sequence end code:
         //-----------------------
-        if( codec_id == CODEC_ID_MPEG1VIDEO ) {
+        if( codec_id == AV_CODEC_ID_MPEG1VIDEO ) {
           buff.MPG[0] = 0x00;
           buff.MPG[1] = 0x00;
           buff.MPG[2] = 0x01;
