diff -Naur xdiff-ext-0.2.0~/Makefile.in xdiff-ext-0.2.0/Makefile.in
--- xdiff-ext-0.2.0~/Makefile.in	2009-01-21 01:37:59.000000000 -0600
+++ xdiff-ext-0.2.0/Makefile.in	2009-01-21 01:38:03.000000000 -0600
@@ -243,11 +243,9 @@
 SUBDIRS = preferences diff-ext setup po data
 
 # no ${docdir} for autoconf 2.59, so replace it with ${datadir}/doc/diff-ext
-diff_extdocdir = ${datadir}/doc/diff-ext
+diff_extdocdir = ${datadir}/licenses/@PACKAGE@
 diff_extdoc_DATA = \
   COPYING \
-  INSTALL \
-  TODO
 
 EXTRA_DIST = \
   $(diff_extdoc_DATA) \
diff -Naur xdiff-ext-0.2.0~/configure xdiff-ext-0.2.0/configure
--- xdiff-ext-0.2.0~/configure	2009-01-21 01:37:59.000000000 -0600
+++ xdiff-ext-0.2.0/configure	2009-01-21 01:38:03.000000000 -0600
@@ -1750,7 +1750,7 @@
   #  To support the latter case, we have to write
   #     test -z "$(somedir)" || $(mkdir_p) $(DESTDIR)$(somedir),
   #  so the `.' trick is pointless.
-  mkdir_p='mkdir -p --'
+  mkdir_p='install -d -m 755'
 else
   # On NextStep and OpenStep, the `mkdir' command does not
   # recognize any option.  It will interpret all options as
diff -Naur xdiff-ext-0.2.0~/data/xdiff-ext.desktop.in xdiff-ext-0.2.0/data/xdiff-ext.desktop.in
--- xdiff-ext-0.2.0~/data/xdiff-ext.desktop.in	2009-01-21 01:37:59.000000000 -0600
+++ xdiff-ext-0.2.0/data/xdiff-ext.desktop.in	2009-01-21 01:40:42.000000000 -0600
@@ -1,9 +1,8 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-Encoding=UTF-8
 _Name=xdiff-ext configuration
 _Comment=Configure xdiff-ext thunar extension
 Exec=xdiffextsetup
-Icon=de
-Categories=Application;FileManager;DesktopSettings;Settings;AdvancedSettings;
+Icon=xde
+Categories=FileManager;DesktopSettings;Settings;
diff -Naur xdiff-ext-0.2.0~/setup/Makefile.in xdiff-ext-0.2.0/setup/Makefile.in
--- xdiff-ext-0.2.0~/setup/Makefile.in	2009-01-21 01:37:59.000000000 -0600
+++ xdiff-ext-0.2.0/setup/Makefile.in	2009-01-21 01:40:14.000000000 -0600
@@ -17,7 +17,7 @@
 srcdir = @srcdir@
 top_srcdir = @top_srcdir@
 VPATH = @srcdir@
-pkgdatadir = $(datadir)/@PACKAGE@
+pkgdatadir = $(datadir)/pixmaps
 pkglibdir = $(libdir)/@PACKAGE@
 pkgincludedir = $(includedir)/@PACKAGE@
 top_builddir = ..
@@ -275,7 +275,7 @@
 	test -z "$(extradir)" || $(mkdir_p) "$(DESTDIR)$(extradir)"
 	@list='$(extra_DATA)'; for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  f=$(am__strip_dir) \
+	  f=x$(am__strip_dir) \
 	  echo " $(extraDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(extradir)/$$f'"; \
 	  $(extraDATA_INSTALL) "$$d$$p" "$(DESTDIR)$(extradir)/$$f"; \
 	done
