--- /tmp/wmsetup	2008-03-26 19:27:12.000000000 +0100
+++ /opt/inferno-os/lib/wmsetup	2008-03-26 19:28:24.000000000 +0100
@@ -27,6 +27,14 @@
 	} >[2] /dev/null
 }
 
+##Added by archlinux PKGBUILD, furst bind is for the copy-paste to and from X, the fn acme is a fix for acme's font problem
+bind -b '#^' /chan 
+fn acme {
+       builtin acme -f /fonts/vera/Vera/unicode.14.font -F /fonts/vera/VeraMono/VeraMono.14.font $*
+}
+
+
+
 plumber >/chan/wmstdout >[2=1]
 menu Shell			{wmrun wm/sh}
 menu Acme			{wmrun acme}
