From 77c76b0f4c58c83027cd88bcc5591577950e0679 Mon Sep 17 00:00:00 2001
From: Yosef Or Boczko <yosefor3@walla.com>
Date: Mon, 10 Jun 2013 23:42:14 +0300
Subject: Make headerbar a titlebar

---
 src/dh-window.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/dh-window.c b/src/dh-window.c
index 1373919..abec66d 100644
--- a/src/dh-window.c
+++ b/src/dh-window.c
@@ -509,8 +509,6 @@ dh_window_init (DhWindow *window)
         priv = GET_PRIVATE (window);
         window->priv = priv;
 
-        gtk_window_set_hide_titlebar_when_maximized (GTK_WINDOW (window), TRUE);
-
         priv->selected_search_link = NULL;
 
         /* handle settings */
@@ -693,9 +691,8 @@ window_populate (DhWindow *window)
         menu = gtk_builder_get_object (priv->builder, "window-menu");
         gtk_menu_button_set_menu_model (GTK_MENU_BUTTON (menu_button), G_MENU_MODEL (menu));
 
-        /* Add toolbar to main box */
-        gtk_box_pack_start (GTK_BOX (priv->main_box), priv->header_bar,
-                            FALSE, FALSE, 0);
+        /* Make headerbar a titlebar */
+        gtk_window_set_titlebar(GTK_WINDOW(window), priv->header_bar);
 
         priv->hpaned = gtk_paned_new (GTK_ORIENTATION_HORIZONTAL);
         gtk_box_pack_start (GTK_BOX (priv->main_box), priv->hpaned, TRUE, TRUE, 0);
-- 
1.8.3

