From bab4f8c319eda9e7ec948d9923bb7274c5a01983 Mon Sep 17 00:00:00 2001
From: Yosef Or Boczko <yosefor3@walla.com>
Date: Tue, 11 Jun 2013 01:30:09 +0300
Subject: Bump required GTK+

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index a27b89c..b37d624 100644
--- a/configure.ac
+++ b/configure.ac
@@ -55,7 +55,7 @@ PKG_CHECK_MODULES(DEVHELP, [
 ])
 
 PKG_CHECK_MODULES(LIBDEVHELP, [
-  gtk+-3.0 >= 3.5.6
+  gtk+-3.0 >= 3.9.2
   gio-2.0 >= 2.32
   webkit2gtk-3.0 >= 2.0.0
 ])
-- 
1.8.3

