diff -aur fritz.org/src/defs.h fritz.new/src/defs.h
--- fritz.org/src/defs.h	2007-07-05 15:00:00.000000000 +0200
+++ fritz.new/src/defs.h	2008-02-06 05:07:38.000000000 +0100
@@ -60,7 +60,7 @@
 
 #define	UNUSED_ARG(x)	(x)=(x)
 
-typedef unsigned long uintptr_t;
+//typedef unsigned long uintptr_t;
 
 /*---------------------------------------------------------------------------*\
 \*---------------------------------------------------------------------------*/
