diff --git a/configure.ac b/configure.ac
index afa17b3..6c905d6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -101,6 +101,7 @@ for driver in `echo ${drivers} | sed -e 's/,/ /g' -e 's/,$//g'`; do
 		vfs301)
 			AC_DEFINE([ENABLE_VFS301], [], [Build Validity VFS301/VFS300 driver])
 			enable_vfs301="yes"
+		;;
 		etes603)
 			AC_DEFINE([ENABLE_ETES603], [], [Build EgisTec ES603 driver])
 			enable_etes603="yes"
@@ -313,6 +314,7 @@ if test x$enable_vfs301 != xno ; then
 	AC_MSG_NOTICE([** vfs301 driver enabled])
 else
 	AC_MSG_NOTICE([   vfs301 driver disabled])
+fi
 if test x$enable_etes603 != xno ; then
 	AC_MSG_NOTICE([** etes603 driver enabled])
 else
