diff -uNr avro-cpp-1.7.6/CMakeLists.txt avro-cpp-1.7.6-patched/CMakeLists.txt
--- avro-cpp-1.7.6/CMakeLists.txt	2014-01-10 21:04:02.000000000 +0000
+++ avro-cpp-1.7.6-patched/CMakeLists.txt	2014-03-10 17:50:03.454140872 +0000
@@ -101,7 +101,7 @@
     add_custom_command (OUTPUT ${ns}.hh
         COMMAND precompile ${CMAKE_CURRENT_SOURCE_DIR}/jsonschemas/${file}
             ${file}
-        COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/scripts/gen-cppcode.py
+        COMMAND python2 ${CMAKE_CURRENT_SOURCE_DIR}/scripts/gen-cppcode.py
             -n ${ns} -i ${file} -o ${ns}.hh
         DEPENDS precompile ${CMAKE_CURRENT_SOURCE_DIR}/jsonschemas/${file})
     add_custom_target(${ns} DEPENDS ${ns}.hh)
diff -uNr avro-cpp-1.7.6/scripts/gen-cppcode.py avro-cpp-1.7.6-patched/scripts/gen-cppcode.py
--- avro-cpp-1.7.6/scripts/gen-cppcode.py	2014-01-10 21:04:02.000000000 +0000
+++ avro-cpp-1.7.6-patched/scripts/gen-cppcode.py	2014-03-10 17:49:35.077407526 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 license = '''/**
  * Licensed to the Apache Software Foundation (ASF) under one
