diff -aur yoshimi-0.060.10/src/UI/BankUI.fl yoshimi-0.060.10.patched/src/UI/BankUI.fl
--- yoshimi-0.060.10/src/UI/BankUI.fl	2011-04-12 01:20:30.000000000 +0200
+++ yoshimi-0.060.10.patched/src/UI/BankUI.fl	2011-07-12 00:36:03.000000000 +0200
@@ -51,7 +51,7 @@
 class BankProcess_ {open
 } {
   Function {process(void)} {open return_type {virtual void}
-  } {}
+  } {code{;}}
   decl {Bank *bank;} {public
   }
 }
diff -aur yoshimi-0.060.10/src/UI/OscilGenUI.fl yoshimi-0.060.10.patched/src/UI/OscilGenUI.fl
--- yoshimi-0.060.10/src/UI/OscilGenUI.fl	2011-04-12 01:20:30.000000000 +0200
+++ yoshimi-0.060.10.patched/src/UI/OscilGenUI.fl	2011-07-12 00:36:57.000000000 +0200
@@ -166,7 +166,7 @@
 
 class PSlider {open : {public Fl_Slider}
 } {
-  Function {PSlider(int x,int y, int w, int h, const char *label=0):Fl_Slider(x,y,w,h,label)} {} {}
+  Function {PSlider(int x,int y, int w, int h, const char *label=0):Fl_Slider(x,y,w,h,label)} {} {code{;}}
   Function {handle(int event)} {return_type int
   } {
     code {int X=x(),Y=y(),W=w(),H=h();
diff -aur yoshimi-0.060.10/src/UI/PartUI.fl yoshimi-0.060.10.patched/src/UI/PartUI.fl
--- yoshimi-0.060.10/src/UI/PartUI.fl	2011-04-12 01:20:30.000000000 +0200
+++ yoshimi-0.060.10.patched/src/UI/PartUI.fl	2011-07-12 00:37:30.000000000 +0200
@@ -89,7 +89,7 @@
 class PartUI_ {open
 } {
   Function {showparameters(int kititem,int engine)} {open return_type virtual
-  } {}
+  } {code{;}}
 } 
 
 class PartKitItem {: {public Fl_Group}
diff -aur yoshimi-0.060.10/src/UI/PresetsUI.fl yoshimi-0.060.10.patched/src/UI/PresetsUI.fl
--- yoshimi-0.060.10/src/UI/PresetsUI.fl	2011-04-12 01:20:30.000000000 +0200
+++ yoshimi-0.060.10.patched/src/UI/PresetsUI.fl	2011-07-12 18:54:26.000000000 +0200
@@ -44,8 +44,8 @@
 
 class PresetsUI_ {} {
   Function {refresh()} {return_type {virtual void}
-  } {}
-  Function {~PresetsUI_()} {} {}
+  } {code{;}}
+  Function {~PresetsUI_()} {} {code{;}}
 } 
 
 class PresetsUI {} {