--- ../cuda-z-0.5/src/czdialog.cpp	2009-04-12 16:14:04.000000000 +0200
+++ src/czdialog.cpp	2012-06-18 15:36:57.687840248 +0200
@@ -537,7 +537,7 @@
 	if(fileName.isEmpty())
 		return;
 
-	CZLog(CZLogLevelMid, "Export to text as %s", fileName);
+//	CZLog(CZLogLevelMid, "Export to text as %s", fileName);
 
 	QFile file(fileName);
 	if(!file.open(QFile::WriteOnly | QFile::Text)) {
@@ -675,7 +675,7 @@
 	if(fileName.isEmpty())
 		return;
 
-	CZLog(CZLogLevelMid, "Export to HTML as %s", fileName);
+//	CZLog(CZLogLevelMid, "Export to HTML as %s", fileName);
 
 	QFile file(fileName);
 	if(!file.open(QFile::WriteOnly | QFile::Text)) {
@@ -861,7 +861,7 @@
 	bool error			/*!< HTTP operation error state. */
 ) {
 	if(error) {
-		CZLog(CZLogLevelWarning, "Get version request done with error %d: %s", http->error(), http->errorString());
+//		CZLog(CZLogLevelWarning, "Get version request done with error %d: %s", http->error(), http->errorString());
 
 		labelAppUpdate->setText(tr("Can't load version information.\n") + http->errorString());
 	} else {
