diff -r 6ffa317d13a8 -r a40508fcb472 setup.py
--- a/setup.py	Wed Dec 12 22:31:16 2012 -0800
+++ b/setup.py	Thu Dec 13 00:34:34 2012 -0800
@@ -9,7 +9,7 @@
 
 setup(name='Seascope',
 	version='0.7',
-	description='A multi-platform multi-language source code browsing tool'
+	description='A multi-platform multi-language source code browsing tool',
 	long_description='A pyQt GUI front-end for idutils, cscope and gtags. Written in python using pyQt, QScintilla libraries.',
 	url='http://seascope.googlecode.com',
 	packages=['Seascope',
