Nur in src/grubng-0.9.6/: bin.
diff -ur src/grubng-0.9.6//Grubng/Console/ConsoleMain.cs grubng-0.9.6//Grubng/Console/ConsoleMain.cs
--- src/grubng-0.9.6//Grubng/Console/ConsoleMain.cs	2011-04-14 09:50:14.572801355 +0200
+++ grubng-0.9.6//Grubng/Console/ConsoleMain.cs	2011-01-31 11:36:25.000000000 +0100
@@ -224,7 +224,7 @@
 				//Create sitemap
 				if (answer == '7')
 				{
-					this.CreateSitemap();
+					CreateSitemap();
 				}
 			}
 			//Start program in automatic mode
@@ -253,7 +253,7 @@
 			bool stoped;
 			while (answer != 'q')
 			{
-				answer = this.Menu();
+				answer = Menu();
 				Console.Clear();
 				//Resume/pause crawling
 				if (answer == 'r')
@@ -409,7 +409,7 @@
 				//Upload sitemap on server
 				if (answer == 'u')
 				{
-					this.CreateSitemap();
+					CreateSitemap();
 				}
 			}
 			//Quit from program
diff -ur src/grubng-0.9.6//grubng-sharp/7Zip/LzBinTree.cs grubng-0.9.6//grubng-sharp/7Zip/LzBinTree.cs
--- src/grubng-0.9.6//grubng-sharp/7Zip/LzBinTree.cs	2011-04-14 09:47:55.296889211 +0200
+++ grubng-0.9.6//grubng-sharp/7Zip/LzBinTree.cs	2009-06-09 23:34:13.000000000 +0200
@@ -84,7 +84,7 @@
 				_hash[i] = 0;
 			_cyclicBufferPos = 0;
 			ReduceOffsets(-1);
-			this.CRC();
+			CRC();
 		}
 
 		public new void MovePos()
