diff -Naur a/mscore-1.1/mscore/mscore/preferences.cpp b/mscore-1.1/mscore/mscore/preferences.cpp
--- a/mscore-1.1/mscore/mscore/preferences.cpp	2011-07-27 15:32:44.000000000 +0200
+++ b/mscore-1.1/mscore/mscore/preferences.cpp	2011-12-23 22:23:20.000000000 +0100
@@ -148,7 +148,7 @@
 #else
       useAlsaAudio       = true;
       useJackAudio       = false;
-      usePortaudioAudio  = false;
+      usePortaudioAudio  = true;
       useJackMidi        = false;
 #endif
       alsaDevice         = "default";
@@ -381,7 +381,7 @@
       useAlsaAudio       = s.value("useAlsaAudio", true).toBool();
       useJackAudio       = s.value("useJackAudio", false).toBool();
       useJackMidi        = s.value("useJackMidi",  false).toBool();
-      usePortaudioAudio  = s.value("usePortaudioAudio", false).toBool();
+      usePortaudioAudio  = s.value("usePortaudioAudio", true).toBool();
 #endif
 
       alsaDevice         = s.value("alsaDevice", "default").toString();
