--- jack2-1.9.7~dfsg.orig/wscript
+++ jack2-1.9.7~dfsg/wscript
@@ -201,8 +201,7 @@ def configure(conf):
     conf.define('JACKMP', 1)
     if conf.env['BUILD_JACKDBUS'] == True:
         conf.define('JACK_DBUS', 1)
-        if conf.env['BUILD_JACKD'] == False:
-            conf.define('USE_LIBDBUS_AUTOLAUNCH', 1)
+        conf.define('USE_LIBDBUS_AUTOLAUNCH', 1)
     if conf.env['BUILD_WITH_PROFILE'] == True:
         conf.define('JACK_MONITOR', 1)
     if conf.env['BUILD_WITH_32_64'] == True:
--- jack2-1.9.7~dfsg.orig/common/wscript
+++ jack2-1.9.7~dfsg/common/wscript
@@ -178,7 +178,7 @@ def build(bld):
     clientlib.defines = 'HAVE_CONFIG_H'
     clientlib.uselib = uselib
     clientlib.install_path = '${LIBDIR}'
-    if bld.env['BUILD_JACKDBUS'] == True and bld.env['BUILD_JACKD'] == False:
+    if bld.env['BUILD_JACKDBUS'] == True:
         clientlib.uselib.append('DBUS-1')
     clientlib.includes = includes
     clientlib.name         = 'clientlib'
