--- gmbox-gtk/libgmbox/core.py	2011-07-17 00:00:05.000000000 +0800
+++ gmbox-gtk/libgmbox/core.py.flashplayer_key	2011-10-13 15:47:02.938282039 +0800
@@ -99,7 +99,7 @@
 
         if not hasattr(self, "songUrl"):
             template = "http://www.google.cn/music/songstreaming?id=%s&cd&sig=%s&output=xml"
-            flashplayer_key = "a3230bc2ef1939edabc39ddd03009439"
+            flashplayer_key = "343aef99312bc33a887761529031cbad"
             sig = hashlib.md5(flashplayer_key + self.id).hexdigest()
             url = template % (self.id, sig)
 
