diff -aur pdfmasher/ebooks/oeb/stylizer.py pdfmasher-fix/ebooks/oeb/stylizer.py
--- pdfmasher/ebooks/oeb/stylizer.py	2012-11-06 18:56:57.000000000 -0800
+++ pdfmasher-fix/ebooks/oeb/stylizer.py	2012-11-06 18:58:54.000000000 -0800
@@ -15,7 +15,9 @@
 from cssutils import (profile as cssprofiles, parseString, parseStyle, log as
         cssutils_log, CSSParser, profiles)
 from lxml import etree
-from lxml.cssselect import css_to_xpath, ExpressionError, SelectorSyntaxError
+from cssselect import HTMLTranslator
+from cssutils import (profile as cssprofiles, parseString, parseStyle, log as
+        cssutils_log, CSSParser, profiles, replaceUrls)
 from .base import XHTML, XHTML_NS, CSS_MIME, OEB_STYLES
 from .base import XPNSMAP, xpath, urlnormalize
 from .htmlcss import HTML_CSS
@@ -23,6 +25,7 @@
 cssutils_log.setLevel(logging.WARN)
 
 _html_css_stylesheet = None
+css_to_xpath = HTMLTranslator().css_to_xpath
 
 def html_css_stylesheet():
     global _html_css_stylesheet
