--- a/Ochi.pro	2014-05-15 10:47:04.553331197 +0200
+++ b/Ochi.pro	2014-05-15 13:04:03.515773872 +0200
@@ -5,24 +5,24 @@
 #-------------------------------------------------
 
 INCLUDEPATH += /usr/include/opencv
-LIBS += /usr/lib64/libopencv_core.so \
-    /usr/lib64/libopencv_gpu.so \
-    /usr/lib64/libopencv_calib3d.so \
-    /usr/lib64/libopencv_contrib.so \
-    /usr/lib64/libopencv_features2d.so \
-    /usr/lib64/libopencv_flann.so \
-    /usr/lib64/libopencv_highgui.so \
-    /usr/lib64/libopencv_imgproc.so \
-    /usr/lib64/libopencv_legacy.so \
-    /usr/lib64/libopencv_ml.so \
-    /usr/lib64/libopencv_nonfree.so \
-    /usr/lib64/libopencv_objdetect.so \
-    /usr/lib64/libopencv_photo.so \
-    /usr/lib64/libopencv_stitching.so \
-    /usr/lib64/libopencv_superres.so \
-    /usr/lib64/libopencv_ts.so \
-    /usr/lib64/libopencv_video.so \
-    /usr/lib64/libopencv_videostab.so
+LIBS += /usr/lib/libopencv_core.so \
+    /usr/lib/libopencv_gpu.so \
+    /usr/lib/libopencv_calib3d.so \
+    /usr/lib/libopencv_contrib.so \
+    /usr/lib/libopencv_features2d.so \
+    /usr/lib/libopencv_flann.so \
+    /usr/lib/libopencv_highgui.so \
+    /usr/lib/libopencv_imgproc.so \
+    /usr/lib/libopencv_legacy.so \
+    /usr/lib/libopencv_ml.so \
+    /usr/lib/libopencv_nonfree.so \
+    /usr/lib/libopencv_objdetect.so \
+    /usr/lib/libopencv_photo.so \
+    /usr/lib/libopencv_stitching.so \
+    /usr/lib/libopencv_superres.so \
+    /usr/lib/libopencv_ts.a \
+    /usr/lib/libopencv_video.so \
+    /usr/lib/libopencv_videostab.so
 
 QT       += core gui 