diff -rup sb0x-project/api/std.py sb0x-project-2.0.1rc2/api/std.py
--- sb0x-project-2.0.1rc2/api/std.py	2014-05-25 06:51:47.900850440 +0300
+++ sb0x-project-2.0.1rc2/api/std.py	2014-05-25 06:52:16.790848681 +0300
@@ -70,4 +70,4 @@ def cles():
 
 
 #the current path
-home = home()
+home = "/opt/sb0x-project/"
diff -rup sb0x-project/core/tests.py sb0x-project-2.0.1rc2/core/tests.py
--- sb0x-project-2.0.1rc2/core/tests.py	2014-05-25 06:51:47.900850440 +0300
+++ sb0x-project-2.0.1rc2/core/tests.py	2014-05-25 06:52:42.167513798 +0300
@@ -18,7 +18,7 @@ from time import sleep
 class tests(object):
 	"""testing user OS"""
 	def __init__(self):
-		self.home = os.getcwd()
+		self.home = "/opt/sb0x-project"
 		self.outputdir = "%s/output" % (self.home)
 	#================================#
 	#[1] test python version function for python 2.7
