diff -aur pymazon.orig/pymazon/core/downloader.py pymazon.patched/pymazon/core/downloader.py
--- pymazon.orig/pymazon/core/downloader.py	2013-09-30 22:00:38.024566519 -0700
+++ pymazon.patched/pymazon/core/downloader.py	2013-09-30 22:02:43.413264947 -0700
@@ -89,7 +89,10 @@
         return handle   
 
     def _download(self, obj, handle):
-        fs = int(obj.filesize)
+        if obj.filesize:
+            fs = int(obj.filesize)
+        else:
+            fs = int(handle.info()['Content-Length'])
         chunk_size = fs / 100 #1% of file per chunk              
         buf = StringIO()
         perc_complete = 0  
