diff --git a/libfprint/60-fprint-autosuspend.rules b/libfprint/60-fprint-autosuspend.rules
index 914c06e..9a5c0b5 100644
--- a/libfprint/60-fprint-autosuspend.rules
+++ b/libfprint/60-fprint-autosuspend.rules
@@ -78,6 +78,7 @@ SUBSYSTEM=="usb", ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0008", ATTRS{dev}=
 
 # Validity VFS5011
 SUBSYSTEM=="usb", ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0011", ATTRS{dev}=="*", ATTR{power/control}="auto"
+SUBSYSTEM=="usb", ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0017", ATTRS{dev}=="*", ATTR{power/control}="auto"
 SUBSYSTEM=="usb", ATTRS{idVendor}=="138a", ATTRS{idProduct}=="0018", ATTRS{dev}=="*", ATTR{power/control}="auto"
 
 # UPEK TouchChip/Eikon Touch 300
diff --git a/libfprint/drivers/vfs5011.c b/libfprint/drivers/vfs5011.c
index 02a8bd4..73015fa 100644
--- a/libfprint/drivers/vfs5011.c
+++ b/libfprint/drivers/vfs5011.c
@@ -1130,6 +1130,7 @@ static void dev_deactivate(struct fp_img_dev *dev)
 static const struct usb_id id_table[] =
 {
 	{ .vendor = 0x138a, .product = 0x0011 /* vfs5011 */ },
+	{ .vendor = 0x138a, .product = 0x0017 /* Validity device from Lenovo T440 laptops */ },
 	{ .vendor = 0x138a, .product = 0x0018 /* one more Validity device */ },
 	{ 0, 0, 0, },
 };
