--- ifled.c.orig	2007-11-15 16:01:02.000000000 +0100
+++ ifled.c	2007-11-15 16:02:49.000000000 +0100
@@ -17,7 +17,9 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
  *                                         
  * Last modified: 991211
- * 
+ *
+ * 20071115 - Minor fixes (GCN)
+ *
  */
 
 #include <stdio.h>
@@ -27,6 +29,7 @@
 #include <sys/utsname.h>
 #include <signal.h>
 #include <unistd.h>
+#include <stdlib.h>
 
 const char *banner = 
 		"ifled v0.6 - (c)1999 Mattias Wadman <napolium@sudac.org>\n"
