From 17065ba1cb2658f5273b064ee874992affdb7b92 Mon Sep 17 00:00:00 2001
From: Roman Anasal <roman.anasal@academyconsult.de>
Date: Wed, 10 Dec 2014 12:42:44 +0100
Subject: [PATCH] Bugfix: always respect $(DESTDIR) in Makefile

---
 Makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 997bd0a..80058dc 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -64,7 +64,7 @@ strip: $(PRPL_LIBNAME)
 # TODO: Find a better place for server.pub
 install: $(PRPL_LIBNAME)
 	install -D $(PRPL_LIBNAME) $(DESTDIR)$(PLUGIN_DIR_PURPLE)/$(PRPL_NAME)
-	install -D tg-server.pub /etc/telegram-purple/server.pub
+	install -D tg-server.pub $(DESTDIR)/etc/telegram-purple/server.pub
 	install -D imgs/telegram16.png $(DESTDIR)$(DATA_ROOT_DIR_PURPLE)/pixmaps/pidgin/protocols/16/telegram.png
 	install -D imgs/telegram22.png $(DESTDIR)$(DATA_ROOT_DIR_PURPLE)/pixmaps/pidgin/protocols/22/telegram.png
 	install -D imgs/telegram48.png $(DESTDIR)$(DATA_ROOT_DIR_PURPLE)/pixmaps/pidgin/protocols/48/telegram.png
@@ -76,7 +76,7 @@ local_install:
 .PHONY: uninstall
 uninstall:
 	rm -f $(DESTDIR)$(PLUGIN_DIR_PURPLE)/$(PRPL_NAME)
-	rm -f /etc/telegram-purple/server.pub
+	rm -f $(DESTDIR)/etc/telegram-purple/server.pub
 	rm -f $(DESTDIR)$(DATA_ROOT_DIR_PURPLE)/pixmaps/pidgin/protocols/16/telegram.png
 	rm -f $(DESTDIR)$(DATA_ROOT_DIR_PURPLE)/pixmaps/pidgin/protocols/22/telegram.png
 	rm -f $(DESTDIR)$(DATA_ROOT_DIR_PURPLE)/pixmaps/pidgin/protocols/48/telegram.png
-- 
2.2.1

