From 1f9554d88956ade1e5fb52c5038604a5cd3ec013 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Wed, 3 Jul 2013 20:51:45 +0100
Subject: [PATCH] Fix LADSPA path reset

---
 source/carla.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/carla.py b/source/carla.py
index 194f222..f23b0ed 100755
--- a/source/carla.py
+++ b/source/carla.py
@@ -502,7 +502,7 @@ def slot_resetSettings(self):
                 paths.sort()
                 self.ui.lw_ladspa.clear()
 
-                for ladspa in Carla.paths:
+                for ladspa in paths:
                     self.ui.lw_ladspa.addItem(ladspa)
 
             elif self.ui.tw_paths.currentIndex() == 1:
-- 
1.8.1.6

