diff -ur a/src/CustomWidgets/quickedit.cpp b/src/CustomWidgets/quickedit.cpp
--- a/src/CustomWidgets/quickedit.cpp	2013-05-20 21:57:17.000000000 +0200
+++ b/src/CustomWidgets/quickedit.cpp	2013-07-12 14:57:32.775070514 +0200
@@ -695,7 +695,7 @@
 
     QList<BrowserNode*> children = classNode->children(validTypes);
     std::reverse(children.begin(), children.end());
-    QList<TreeItemInterface> items;
+    //QList<TreeItemInterface> items;
     for(BrowserNode* child : children)
     {
         if(!child)
