diff -ur a/Makefile b/Makefile
--- a/Makefile	2013-05-20 21:57:17.000000000 +0200
+++ b/Makefile	2013-07-12 19:18:37.164710907 +0200
@@ -117,11 +117,11 @@
 	cp -p *.lang "$(DESTDIR)$(DOUML_LOCALE_DIR)"
 	for i in $(PROGS); do cp -p bin/$$i "$(DESTDIR)$(DOUML_LIB)" ; done
 	echo "#!/bin/sh" >$(DESTDIR)$(DOUML_DIR)/douml
-	echo "PATH=$(DEB_DESTDIR)$(DOUML_LIB):$$"PATH >>$(DESTDIR)$(DOUML_DIR)/douml
-	echo "DOUML_LIB_DIR=$(DEB_DESTDIR)$(DOUML_LIB)" >>$(DESTDIR)$(DOUML_DIR)/douml
+	echo "PATH=$(DOUML_LIB):$$"PATH >>$(DESTDIR)$(DOUML_DIR)/douml
+	echo "DOUML_LIB_DIR=$(DOUML_LIB)" >>$(DESTDIR)$(DOUML_DIR)/douml
 	echo "export PATH" >>$(DESTDIR)$(DOUML_DIR)/douml
 	echo "export DOUML_LIB_DIR" >>$(DESTDIR)$(DOUML_DIR)/douml
-	echo "exec $(DEB_DESTDIR)$(DOUML_LIB)/douml \"$$"@"\"" >>$(DESTDIR)$(DOUML_DIR)/douml
+	echo "exec $(DOUML_LIB)/douml \"$$"@"\"" >>$(DESTDIR)$(DOUML_DIR)/douml
 	chmod +x "$(DESTDIR)$(DOUML_DIR)/douml"
 	if test -n "$(DOUML_DESKTOP_DIR)" ; \
 	then \
@@ -131,7 +131,7 @@
 		echo "Name=DoUML" >> "$(DESTDIR)$(DOUML_DESKTOP_DIR)/douml.desktop" ; \
 		echo "Type=Application" >> "$(DESTDIR)$(DOUML_DESKTOP_DIR)/douml.desktop" ; \
 		echo "Comment=Free UML 2 modeler" >> "$(DESTDIR)$(DOUML_DESKTOP_DIR)/douml.desktop" ; \
-		echo "Exec=$(DEB_DESTDIR)$(DOUML_DIR)/douml" >> "$(DESTDIR)$(DOUML_DESKTOP_DIR)/douml.desktop" ; \
+		echo "Exec=$(DOUML_DIR)/douml" >> "$(DESTDIR)$(DOUML_DESKTOP_DIR)/douml.desktop" ; \
 		echo "TryExec=douml" >> "$(DESTDIR)$(DOUML_DESKTOP_DIR)/douml.desktop" ; \
 		echo "Icon=douml" >> "$(DESTDIR)$(DOUML_DESKTOP_DIR)/douml.desktop" ; \
 		echo "Categories=Development;" >> "$(DESTDIR)$(DOUML_DESKTOP_DIR)/douml.desktop" ; \
diff -ur a/src/CustomWidgets/quickedit.cpp b/src/CustomWidgets/quickedit.cpp
--- a/src/CustomWidgets/quickedit.cpp	2013-05-20 21:57:17.000000000 +0200
+++ b/src/CustomWidgets/quickedit.cpp	2013-07-12 14:57:32.775070514 +0200
@@ -695,7 +695,7 @@
 
     QList<BrowserNode*> children = classNode->children(validTypes);
     std::reverse(children.begin(), children.end());
-    QList<TreeItemInterface> items;
+    //QList<TreeItemInterface> items;
     for(BrowserNode* child : children)
     {
         if(!child)
