diff -au ceton_infinitv_linux_driver/ctn91xx_pci.c.orig ceton_infinitv_linux_driver/ctn91xx_pci.c
--- ceton_infinitv_linux_driver/ctn91xx_pci.c.orig	2013-03-25 10:38:03.044080591 -0400
+++ ceton_infinitv_linux_driver/ctn91xx_pci.c	2013-03-25 10:39:41.460681103 -0400
@@ -225,7 +225,7 @@
     return -EIO;
 }
 
-static void __devexit ctn91xx_unregister(struct pci_dev *pdev)
+static void ctn91xx_unregister(struct pci_dev *pdev)
 {
     ctn91xx_dev_t* dev = (ctn91xx_dev_t*)pci_get_drvdata(pdev);
 
@@ -264,7 +264,7 @@
     kfree( dev );
 }
 
-static struct pci_device_id ctn91xx_table[] __devinitdata = {
+static struct pci_device_id ctn91xx_table[] = {
     { CETON_VENDOR_ID, CTN91XX_DEVICE_ID, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 },
     { ALTERA_VENDOR_ID, CTN91XX_DEVICE_ID, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 },
     { 0, }
@@ -274,7 +274,7 @@
     .name = DEVICE_NAME,
     .id_table = ctn91xx_table,
     .probe = ctn91xx_register,
-    .remove = __devexit_p(ctn91xx_unregister),
+    .remove = ctn91xx_unregister,
 };
 
 int ctn91xx_register_pci_driver()
