--- rlab-2.2.9-old/579.f	2006-02-06 12:40:04.000000000 -0500
+++ rlab-2.2.9/579.f	2006-11-16 00:49:17.000000000 -0500
@@ -253,7 +253,7 @@
 C
 C  CALCULATION OF THE EXTRAPOLATION CONSTANTS.
 C
-  220 FC(M) = 1.5+SIGN(.5,FACT-1.)
+  220 FC(M) = 1.5+SIGN(dble(.5),dble(FACT-1.))
       IF(M.EQ.2) FC(M)=FC(M)+RA
       IF(FACT.GT.1.) FC(M)=-FC(M)
       GOTO 30
@@ -296,4 +296,4 @@
 
       DRCMULT = DCMPLX(RE*DBLE(CO),RE*DIMAG(CO))
       RETURN
-      END
\ No newline at end of file
+      END
