--- rlab-2.2.9-old/flibs/contin/src/coqual.f	2006-01-31 17:28:47.000000000 -0500
+++ rlab-2.2.9/flibs/contin/src/coqual.f	2006-11-16 00:59:39.000000000 -0500
@@ -121,12 +121,12 @@
       BASE=STEPX/CORDIS
       ESAB=CORDIS/STEPX
       IF((IWORK(28).LT.NAVE.AND.TEST.GT.BASE)
-          1  .OR.(IWORK(28).GT.NAVE.AND.TEST.LT.BASE))THEN
+     1  .OR.(IWORK(28).GT.NAVE.AND.TEST.LT.BASE))THEN
         QUAL=EIGHT
         GO TO 88
         ENDIF
       IF((IWORK(28).LT.NAVE.AND.TEST.GT.ESAB)
-          1  .OR.(IWORK(28).GT.NAVE.AND.TEST.LT.ESAB))THEN
+     1  .OR.(IWORK(28).GT.NAVE.AND.TEST.LT.ESAB))THEN
         QUAL=THGIE
         GO TO 88
         ENDIF
