diff -urN blogofile-0.7.1-orig/setup.py blogofile-0.7.1/setup.py
--- blogofile-0.7.1-orig/setup.py	2011-11-07 17:51:16.875924816 +0100
+++ blogofile-0.7.1/setup.py	2011-11-07 17:51:39.299136673 +0100
@@ -21,7 +21,6 @@
                          'pyyaml',
                          'textile',
                          'markdown',
-                         'argparse',
                          'pygments',
                          'docutils'],
       entry_points="""
