--- src/fontcache.cpp	2014-12-20 09:06:35.358645697 +0100
+++ src/fontcache.cpp	2014-12-20 09:06:50.775311981 +0100
@@ -536,8 +536,8 @@
 	aa = (slot->bitmap.pixel_mode == FT_PIXEL_MODE_GRAY);
 
 	/* Add 1 pixel for the shadow on the medium font. Our sprite must be at least 1x1 pixel */
-	int width  = max(1, slot->bitmap.width + (this->fs == FS_NORMAL));
-	int height = max(1, slot->bitmap.rows  + (this->fs == FS_NORMAL));
+	int width  = max(1, (int)slot->bitmap.width + (this->fs == FS_NORMAL));
+	int height = max(1, (int)slot->bitmap.rows  + (this->fs == FS_NORMAL));
 
 	/* Limit glyph size to prevent overflows later on. */
 	if (width > 256 || height > 256) usererror("Font glyph is too large");
