From 60207bc8f2a361158980e167ae3774f366275184 Mon Sep 17 00:00:00 2001
From: Marc-Andre Lamothe <marc7141@gmail.com>
Date: Sat, 17 Jan 2015 22:45:48 -0500
Subject: [PATCH] Applied patch for 64 bit Linux

---
 Source/Core/Core/CoreParameter.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/Core/Core/CoreParameter.cpp b/Source/Core/Core/CoreParameter.cpp
index d690c3b..06236f7 100644
--- a/Source/Core/Core/CoreParameter.cpp
+++ b/Source/Core/Core/CoreParameter.cpp
@@ -98,7 +98,7 @@ void SCoreStartupParameter::LoadDefaults()
 	bJITOff = false; // debugger only settings
 	bJITLoadStoreOff = false;
 	bJITLoadStoreFloatingOff = false;
-	bJITLoadStorePairedOff = false; // XXX not 64-bit clean
+	bJITLoadStorePairedOff = true; // XXX not 64-bit clean
 	bJITFloatingPointOff = false;
 	bJITIntegerOff = false;
 	bJITPairedOff = false;
-- 
2.2.1

