diff -Nurp mms-1.1.1.orig/movie-library/movie-library.cpp mms-1.1.1/movie-library/movie-library.cpp
--- mms-1.1.1.orig/movie-library/movie-library.cpp	2010-05-04 16:00:37.000000000 +0200
+++ mms-1.1.1/movie-library/movie-library.cpp	2010-05-04 16:02:59.000000000 +0200
@@ -109,7 +109,7 @@ bool create_thumbnail(const string& file
 }
 
 #else
-bool create_thumbnail(const string& file, const string& thumbnail, int width, int height, thumbnailer::thumbnailer& thumb){
+bool create_thumbnail(const string& file, const string& thumbnail, int width, int height, thumbnailer& thumb){
   std::string _dir = thumbnail.substr(0, thumbnail.rfind('/')+1);
   abs_recurse_mkdir(_dir, NULL);
   thumbnailer::movie_specs ms;
@@ -322,7 +322,7 @@ void check_path(SQLDatabase& db, std::li
 #ifndef use_ffmpeg_thumbnailer
   Mplayer thumbnailer;
 #else
-  thumbnailer::thumbnailer  thumbnailer;
+  thumbnailer thumbnailer;
   thumbnailer.init((conf->p_debug_level() > 2));
   thumbnailer.deepval = 
 	  (movie_conf->p_thumbnail_accur() >= 0 && movie_conf->p_thumbnail_accur() < 10) ? movie_conf->p_thumbnail_accur() : 3;
diff -Nurp mms-1.1.1.orig/plugins/feature/movie/movie_db.cpp mms-1.1.1/plugins/feature/movie/movie_db.cpp
--- mms-1.1.1.orig/plugins/feature/movie/movie_db.cpp	2010-05-04 16:00:31.000000000 +0200
+++ mms-1.1.1/plugins/feature/movie/movie_db.cpp	2010-05-04 16:01:50.000000000 +0200
@@ -1186,7 +1186,7 @@ string MovieDB::get_name_from_file(const
 bool MovieDB::create_thumbnail(const string& file, const string& thumbnail, int width, int height)
 {
 #ifdef use_ffmpeg_thumbnailer
-  thumbnailer::thumbnailer thumb;
+  thumbnailer thumb;
   thumb.init((thumb_verbosity > 2));
   thumb.deepval = thumb_deepval;
   thumb.skipblanks = thumb_skipblanks;