--- /mnt/docs/intelfx/devel/__auxiliary/archlinux-aur/acpica/acpica/src/acpica-unix-20120518/source/tools/acpisrc/Makefile	2012-05-19 00:40:16.000000000 +0400
+++ /tmp/mnt_docs_intelfx_devel___auxiliary_archlinux-aur_acpica_acpica_src_acpica-unix-20120518_source_tools_acpisrc_Makefile.tmp	2012-06-03 19:49:00.000000000 +0400
@@ -80,7 +80,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
