diff -uNr lxr-0.9.8.org/diff lxr-0.9.8/diff
--- lxr-0.9.8.org/diff	2009-05-10 13:54:29.000000000 +0200
+++ lxr-0.9.8/diff	2011-01-30 19:34:50.730000019 +0100
@@ -26,6 +26,7 @@
 $CVSID = '$Id: diff,v 1.18 2009/05/10 11:54:29 adrianissott Exp $ ';
 
 use strict;
+use lib '/usr/share/perl5/vendor_perl/LXR';
 use lib '.'; # for Local.pm
 use lib do { $0 =~ m{(.*)/} ? "$1/lib" : "lib" }; # if LXR modules are in ./lib
 
diff -uNr lxr-0.9.8.org/find lxr-0.9.8/find
--- lxr-0.9.8.org/find	2009-05-10 13:54:29.000000000 +0200
+++ lxr-0.9.8/find	2011-01-30 19:34:08.346666684 +0100
@@ -26,6 +26,7 @@
 $CVSID = '$Id: find,v 1.26 2009/05/10 11:54:29 adrianissott Exp $ ';
 
 use strict;
+use lib '/usr/share/perl5/vendor_perl/LXR';
 use lib '.'; # for Local.pm
 use lib do { $0 =~ m{(.*)/} ? "$1/lib" : "lib" }; # if LXR modules are in ./lib
 
diff -uNr lxr-0.9.8.org/genxref lxr-0.9.8/genxref
--- lxr-0.9.8.org/genxref	2009-05-14 23:13:07.000000000 +0200
+++ lxr-0.9.8/genxref	2011-01-30 19:29:18.866666683 +0100
@@ -191,7 +191,7 @@
 		open($swish,
 			    "| "
 			  . $config->swishbin
-			  . " -S prog -i stdin -v 1 -c swish-e.conf -f "
+			  . " -S prog -i stdin -v 1 -c /etc/lxr/swish-e.conf -f "
 			  . $config->swishdir . "/"
 			  . $releaseid
 			  . ".index")
diff -uNr lxr-0.9.8.org/ident lxr-0.9.8/ident
--- lxr-0.9.8.org/ident	2010-01-05 19:00:58.000000000 +0100
+++ lxr-0.9.8/ident	2011-01-30 19:35:14.556666686 +0100
@@ -26,6 +26,7 @@
 $CVSID = '$Id: ident,v 1.25 2010/01/05 18:00:58 mbox Exp $ ';
 
 use strict;
+use lib '/usr/share/perl5/vendor_perl/LXR';
 use lib '.'; # for Local.pm
 use lib do { $0 =~ m{(.*)/} ? "$1/lib" : "lib" }; # if LXR modules are in ./lib
 
@@ -49,7 +50,7 @@
 	my $templ = shift;
 	my $ret   = '';
 
-	my @refs = $index->getindex($identifier, $release);
+	my @refs = $index->getindex($identifier, $releaseid);
 
 	my $file_hits = 0;
 	my $last_file;
@@ -86,7 +87,7 @@
 	my $templ = shift;
 	my $ret   = '';
 
-	my @uses = $index->getreference($identifier, $release);
+	my @uses = $index->getreference($identifier, $releaseid);
 	my $file_hits = 0;
 	my $last_file;
 	foreach my $ref (sort { $$a[0] cmp $$b[0] } @uses) {
diff -uNr lxr-0.9.8.org/lib/LXR/Common.pm lxr-0.9.8/lib/LXR/Common.pm
--- lxr-0.9.8.org/lib/LXR/Common.pm	2010-01-16 00:23:20.000000000 +0100
+++ lxr-0.9.8/lib/LXR/Common.pm	2011-01-30 19:29:18.866666683 +0100
@@ -23,6 +23,7 @@
 $CVSID = '$Id: Common.pm,v 1.64 2010/01/15 23:23:20 mbox Exp $ ';
 
 use strict;
+use lib '/usr/share/perl5/vendor_perl/LXR';
 
 require Exporter;
 
diff -uNr lxr-0.9.8.org/lib/LXR/Config.pm lxr-0.9.8/lib/LXR/Config.pm
--- lxr-0.9.8.org/lib/LXR/Config.pm	2009-04-08 23:17:06.000000000 +0200
+++ lxr-0.9.8/lib/LXR/Config.pm	2011-01-30 19:29:18.866666683 +0100
@@ -67,7 +67,7 @@
 	$url .= '/' unless $url =~ m#/$#;    # append / if necessary
 
 	unless ($confpath) {
-		($confpath) = ($0 =~ /(.*?)[^\/]*$/);
+		$confpath = '/etc/lxr/';
 		$confpath .= $confname;
 	}
 
diff -uNr lxr-0.9.8.org/search lxr-0.9.8/search
--- lxr-0.9.8.org/search	2010-01-05 19:00:58.000000000 +0100
+++ lxr-0.9.8/search	2011-01-30 19:35:09.686666685 +0100
@@ -26,6 +26,7 @@
 $CVSID = '$Id: search,v 1.34 2010/01/05 18:00:58 mbox Exp $ ';
 
 use strict;
+use lib '/usr/share/perl5/vendor_perl/LXR';
 use lib '.'; # for Local.pm
 use lib do { $0 =~ m{(.*)/} ? "$1/lib" : "lib" }; # if LXR modules are in ./lib
 
diff -uNr lxr-0.9.8.org/source lxr-0.9.8/source
--- lxr-0.9.8.org/source	2009-05-10 13:54:29.000000000 +0200
+++ lxr-0.9.8/source	2011-01-30 19:35:04.486666685 +0100
@@ -27,6 +27,7 @@
 $CVSID = '$Id: source,v 1.47 2009/05/10 11:54:29 adrianissott Exp $ ';
 
 use strict;
+use lib '/usr/share/perl5/vendor_perl/LXR';
 use lib '.';    # for Local.pm
 use lib do { $0 =~ m{(.*)/} ? "$1/lib" : "lib" };  # if LXR modules are in ./lib
 
