diff -ur a/jetty-hightide-8.1.8.v20121106/bin/jetty.sh b/jetty-hightide-8.1.8.v20121106/bin/jetty.sh
--- a/jetty-hightide-8.1.8.v20121106/bin/jetty.sh	2012-11-06 19:55:49.000000000 +0200
+++ b/jetty-hightide-8.1.8.v20121106/bin/jetty.sh	2012-12-19 13:22:27.000000000 +0200
@@ -443,7 +443,7 @@
         touch "$JETTY_PID"
         chown "$JETTY_USER" "$JETTY_PID"
         # FIXME: Broken solution: wordsplitting, pathname expansion, arbitrary command execution, etc.
-        su - "$JETTY_USER" -c "
+        su -s /bin/sh - "$JETTY_USER" -c "
           exec ${RUN_CMD[*]} --daemon &
           disown \$!
           echo \$! > '$JETTY_PID'"
