diff -rup gcc-3.4.6.orig//libiberty/cplus-dem.c gcc-3.4.6/libiberty/cplus-dem.c
--- gcc-3.4.6.orig//libiberty/cplus-dem.c	2004-01-02 21:33:55.000000000 +0100
+++ gcc-3.4.6/libiberty/cplus-dem.c	2010-07-23 14:23:24.816480990 +0200
@@ -49,12 +49,12 @@ Boston, MA 02111-1307, USA.  */
 #include <string.h>
 #include <stdio.h>
 
-#ifdef HAVE_STDLIB_H
+/*#ifdef HAVE_STDLIB_H*/
 #include <stdlib.h>
-#else
+/*#else
 char * malloc ();
 char * realloc ();
-#endif
+#endif*/
 
 #include <demangle.h>
 #undef CURRENT_DEMANGLING_STYLE
diff -rup gcc-3.4.6.orig//libiberty/getpwd.c gcc-3.4.6/libiberty/getpwd.c
--- gcc-3.4.6.orig//libiberty/getpwd.c	2001-10-07 23:53:31.000000000 +0200
+++ gcc-3.4.6/libiberty/getpwd.c	2010-07-23 14:10:09.189729471 +0200
@@ -32,13 +32,13 @@ extern int errno;
 #ifdef HAVE_SYS_PARAM_H
 #include <sys/param.h>
 #endif
-#if HAVE_SYS_STAT_H
+/*#if HAVE_SYS_STAT_H*/
 #include <sys/stat.h>
-#endif
+/*#endif*/
 
 /* Prototype these in case the system headers don't provide them. */
-extern char *getpwd ();
-extern char *getwd ();
+/*extern char *getpwd ();
+extern char *getwd ();*/
 
 #include "libiberty.h"
 
diff -rup gcc-3.4.6.orig//libiberty/strsignal.c gcc-3.4.6/libiberty/strsignal.c
--- gcc-3.4.6.orig//libiberty/strsignal.c	2003-04-15 22:36:33.000000000 +0200
+++ gcc-3.4.6/libiberty/strsignal.c	2010-07-23 14:10:09.189729471 +0200
@@ -554,7 +554,7 @@ followed by a newline.
 
 */
 
-#ifndef HAVE_PSIGNAL
+/*#ifndef HAVE_PSIGNAL
 
 void
 psignal (signo, message)
