Index: twill-0.9/twill/shell.py
===================================================================
--- twill-0.9/twill/shell.py	2007-03-06 23:00:05.000000000 +0000
+++ twill-0.9/twill/shell.py	2007-12-21 02:56:21.000000000 +0000
@@ -102,10 +102,17 @@
         # initialize a new local namespace.
         namespaces.new_local_dict()
 
+        # The history file name is located in the home directory of
+        # the user
+        import os
+        self._history_file = os.environ.has_key("HOME") and \
+            os.path.join(os.environ["HOME"], ".twill_history") or \
+            None
+
         # import readline history, if available.
-        if readline:
+        if self._history_file and readline:
             try:
-                readline.read_history_file('.twill-history')
+                readline.read_history_file(self._history_file)
             except IOError:
                 pass
 
@@ -236,8 +243,8 @@
 
     def do_EOF(self, *args):
         "Exit on CTRL-D"
-        if readline:
-            readline.write_history_file('.twill-history')
+        if self._history_file and readline:
+            readline.write_history_file(self._history_file)
             
         raise SystemExit()
 
