--- usr/bin/cowsay	2015-03-09 21:47:02.035896594 +0100
+++ usr/bin/cowsay-new	2015-03-09 21:49:19.018487829 +0100
@@ -2,6 +2,8 @@
 
 ##
 ## Cowsay 3.03
+## patched UTF-8 support by
+## https://bugs.debian.org/cgi-bin/bugreport.cgi?msg=27;filename=cowsay-utf8.patch;att=1;bug=254557
 ##
 ## This file is part of cowsay.  (c) 1999-2000 Tony Monroe.
 ##
@@ -12,6 +14,13 @@
 use Getopt::Std;
 use Cwd;
 
+if (${^UTF8LOCALE}) {
+    binmode STDIN, ':utf8';
+    binmode STDOUT, ':utf8';
+    require Encode;
+    eval { $_ = Encode::decode_utf8($_,1) } for @ARGV;
+}
+
 $version = "3.03";
 $progname = basename($0);
 $eyes = "oo";
