diff -Nur blahtexml-0.9/Source/main.cpp blahtexml-0.9.new/Source/main.cpp
--- blahtexml-0.9/Source/main.cpp	2010-11-27 22:40:52.000000000 +0800
+++ blahtexml-0.9.new/Source/main.cpp	2012-11-08 20:57:36.886227241 +0800
@@ -24,6 +24,7 @@
 #include <stdlib.h>
 #include <sstream>
 #include <stdexcept>
+#include <unistd.h>
 
 using namespace std;
 using namespace blahtex;
diff -Nur blahtexml-0.9/Source/mainPng.cpp blahtexml-0.9.new/Source/mainPng.cpp
--- blahtexml-0.9/Source/mainPng.cpp	2010-04-08 00:45:30.000000000 +0800
+++ blahtexml-0.9.new/Source/mainPng.cpp	2012-11-08 20:58:12.876038563 +0800
@@ -27,6 +27,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <sstream>
+#include <unistd.h>
 
 
 using namespace std;
