--- game/g_save.c.old	2008-03-26 10:30:07.566890048 +0500
+++ game/g_save.c	2008-03-26 13:21:28.040059549 +0500
@@ -750,7 +750,6 @@ void WriteGame (char *filename, qboolean
 		gi.error ("Couldn't open %s", filename);
 
 	memset (str, 0, sizeof(str));
-	strcpy (str, __DATE__);
 	fwrite (str, sizeof(str), 1, f);
 
 	game.autosaved = autosave;
@@ -779,11 +778,6 @@ void ReadGame (char *filename)
 		gi.error ("Couldn't open %s", filename);
 
 	fread (str, sizeof(str), 1, f);
-	if (strcmp (str, __DATE__))
-	{
-		fclose (f);
-		gi.error ("Savegame from an older version.\n");
-	}
 
 	g_edicts =  gi.TagMalloc (game.maxentities * sizeof(g_edicts[0]), TAG_GAME);
 	globals.edicts = g_edicts;
