*** Makefile	2011-05-04 04:29:42.000000000 -0400
--- Makefile_new	2012-05-03 12:49:36.384820277 -0400
***************
*** 28,34 ****
  CFLAGS	+=	-DWITHOUT_XATTR
  endif
  
! LDFLAGS	=	$(shell pkg-config fuse --libs)
  
  FORTAR	=	src COPYING LICENSE README Makefile \
  		README.ru.UTF-8 ChangeLog mhddfs.1 \
--- 28,34 ----
  CFLAGS	+=	-DWITHOUT_XATTR
  endif
  
! LDFLAGS	=	-lpthread $(shell pkg-config fuse --libs)
  
  FORTAR	=	src COPYING LICENSE README Makefile \
  		README.ru.UTF-8 ChangeLog mhddfs.1 \
