--- argot-1.0/Makefile	2011-11-03 02:39:54.000000000 +0300
+++ argot-1.0/Makefile	2011-11-03 02:40:25.000000000 +0300
@@ -70,10 +70,10 @@
 install: all
 	if [ -x "$(PATH_OCAMLFIND)" ]; then \
 	  $(PATH_OCAMLFIND) query $(PROJECT_NAME) && $(PATH_OCAMLFIND) remove $(PROJECT_NAME) || true; \
-	  $(PATH_OCAMLFIND) install $(PROJECT_NAME) META $(PATH_BUILD)/*.cmo $(PATH_BUILD)/*.cm?s; \
+	  $(PATH_OCAMLFIND) install $(PROJECT_NAME) META $(PATH_BUILD)/*.cmo $(PATH_BUILD)/*.cmi $(PATH_BUILD)/*.cm?s; \
 	else \
 	  mkdir -p $(PATH_INSTALL); \
-	  for ext in cmo cmxs cmjs; do \
+	  for ext in cmo cmi cmxs cmjs; do \
 	    test -f $(PATH_BUILD)/$(PROJECT_NAME).$$ext && cp $(PATH_BUILD)/$(PROJECT_NAME).$$ext $(PATH_INSTALL) || true; \
 	  done \
 	fi
