--- asus_acpi.c.orig	2007-11-26 05:46:58.000000000 +0100
+++ asus_acpi.c	2007-12-21 03:15:45.000000000 +0100
@@ -158,10 +158,15 @@ static struct eeepc_hotk *ehotk;
 static int eeepc_hotk_add(struct acpi_device *device);
 static int eeepc_hotk_remove(struct acpi_device *device, int type);
 
+static const struct acpi_device_id eee_device_ids[] = {
+	{EEEPC_HOTK_HID, 0},
+	{"", 0}
+};
+
 static struct acpi_driver eeepc_hotk_driver = {
 	.name = "eeepc_acpi",
 	.class = EEEPC_HOTK_CLASS,
-	.ids = EEEPC_HOTK_HID,
+	.ids = eee_device_ids,
 	.ops = {
 		.add = eeepc_hotk_add,
 		.remove = eeepc_hotk_remove,
@@ -547,7 +552,7 @@ static int asus_hotk_remove(struct acpi_
 static struct acpi_driver asus_hotk_driver = {
 	.name = "asus_acpi",
 	.class = ACPI_HOTK_CLASS,
-	.ids = ACPI_HOTK_HID,
+	.ids = eee_device_ids,
 	.ops = {
 		.add = asus_hotk_add,
 		.remove = asus_hotk_remove,
@@ -1178,7 +1183,7 @@ static void asus_hotk_notify(acpi_handle
 		hotk->brightness = (event & ~((u32) BR_DOWN));
 	}
 
-	acpi_bus_generate_event(hotk->device, event,
+	acpi_bus_generate_proc_event(hotk->device, event,
 				hotk->event_count[event % 128]++);
 
 	return;
@@ -1785,7 +1790,7 @@ eeepc_hotk_notify(acpi_handle handle, u3
 				event = 0x11;
 		}
 	}
-	acpi_bus_generate_event(ehotk->device, event, ehotk->event_count[event % 128]++);
+	acpi_bus_generate_proc_event(ehotk->device, event, ehotk->event_count[event % 128]++);
 
 	return;
 }
