diff -ur ../bibutils_4.1/Makefile_start ./Makefile_start
--- ../bibutils_4.1/Makefile_start	2008-12-21 19:31:06.000000000 +0100
+++ ./Makefile_start	2008-12-26 11:26:01.000000000 +0100
@@ -1,7 +1,13 @@
 POSTFIX=REPLACE_POSTFIX
 CC = REPLACE_CC
 RANLIB=REPLACE_RANLIB
-INSTALLDIR=REPLACE_INSTALLDIR
+PREFIX=REPLACE_PREFIX
+LIBDIR=${PREFIX}/lib
+SHCC=REPLACE_SHCC
+SHLD=REPLACE_SHLD
+INSTALL = /bin/install -c
+SHLIB = lib/libbibutils.so
+#STLIB = lib/libbibutils.a
 
 VERSION=4.1
 DATE=12/21/08
@@ -10,7 +16,7 @@
 	xml2ads xml2bib xml2end xml2isi xml2ris xml2wordbib modsclean
 
 all : FORCE
-	cd lib; make -k $(CC) -k $(RANLIB); cd ..
+	cd lib; make -k $(SHCC) -k $(SHLD) -k $(RANLIB); cd ..
 	cd bin; make -k $(CC) -k VERSION="$(VERSION)" -k DATE="$(DATE)"; cd ..
 
 clean: FORCE
@@ -30,9 +36,13 @@
 
 install: all FORCE
 	sed 's/VERSION/${VERSION}/g' packageconfig_start > lib/bibutils.pc
-#	@for p in ${PROGRAMS}; \
-#		do ( cp bin/$$p ${INSTALLDIR}/$$p ); \
-#	done
+	@for p in ${PROGRAMS}; \
+		do ( cp bin/$$p ${INSTALLDIR}/$$p ); \
+	done
+
+installlib:
+	${INSTALL} -d ${DESTDIR}${LIBDIR} ; ${INSTALL} ${SHLIB} ${DESTDIR}${LIBDIR}
+#	${INSTALL} ${STLIB} ${DESTDIR}${LIBDIR}
 
 package: all FORCE
 	csh -f maketgz.csh ${VERSION} ${POSTFIX}
diff -ur ../bibutils_4.1/configure ./configure
--- ../bibutils_4.1/configure	2008-12-20 00:48:40.000000000 +0100
+++ ./configure	2008-12-26 11:18:16.000000000 +0100
@@ -3,15 +3,15 @@
 set INPUT_FILE = Makefile_start
 set OUTPUT_FILE = Makefile
 
-set INSTALLDIR = /usr/local/bin
+set PREFIX = /usr/local
 
 if ( ${#argv} > 0 ) then
-	if ( $1 == "--install-dir" ) then
+	if ( $1 == "--prefix" ) then
 		if ( ${#argv} < 2 ) then
-			echo "--install-dir requires a directory"
+			echo "--prefix requires a directory"
 			exit
 		else
-			set INSTALLDIR = $2
+			set PREFIX = $2
 		endif
 	else
 		echo "Unidentified argument $1"
@@ -76,81 +76,105 @@
 #POSTFIX=REPLACE_POSTFIX
 #CC = REPLACE_CC
 #RANLIB=REPLACE_RANLIB
-#INSTALLDIR=REPLACE_INSTALLDIR
+#PREFIX=REPLACE_PREFIX
 
 if ( $type == "Linux_x86" ) then
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/CC="cc -Wall"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -Wall -fPIC"/' | \
+	sed 's/REPLACE_SHLD/SHLD="gcc -shared -I."/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="ranlib"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_i386/' > $OUTPUT_FILE
 else if ( $type == "Linux_x86_64" ) then
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/CC="cc -Wall"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -Wall -fPIC"/' | \
+	sed 's/REPLACE_SHLD/SHLD="gcc -shared -I."/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="ranlib"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_i386/' > $OUTPUT_FILE
 else if ( $type == "Linux_Unknown" ) then
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/CC="cc -Wall"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -Wall -fPIC"/' | \
+	sed 's/REPLACE_SHLD/SHLD="gcc -shared -I."/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="ranlib"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX//' > $OUTPUT_FILE
 else if ( $type == "MacOSX_ppc" && $universal_binary == "TRUE" ) then
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/CC="cc -arch i386 -arch ppc -Wall"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -arch i386 -arch ppc -Wall"/' | \
+	sed 's/REPLACE_SHLD/SHLD="cc -bundle -flat_namespace -undefined suppress"/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="ranlib -s"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_osx_universal/' > $OUTPUT_FILE
 else if ( $type == "MacOSX_intel" && $universal_binary == "TRUE" ) then
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/CC="cc -arch i386 -arch ppc -Wall"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -arch i386 -arch ppc -Wall"/' | \
+	sed 's/REPLACE_SHLD/SHLD="cc -bundle -flat_namespace -undefined suppress"/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="ranlib -s"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_osx_universal/' > $OUTPUT_FILE
 else if ( $type == "MacOSX_ppc" || $type == "MacOSX_intel" || \
 	$type == "MacOSX_unknown" ) then
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/CC="cc -Wall"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -arch i386 -arch ppc -Wall"/' | \
+	sed 's/REPLACE_SHLD/SHLD="cc -bundle -flat_namespace -undefined suppress"/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="ranlib -s"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_osx/' > $OUTPUT_FILE
 else if ( $type == "SunOS5" ) then
 	cat $INPUT_FILE | \
 	sed 's|REPLACE_CC|CC="gcc"|' | \
+	sed 's/REPLACE_SHCC/CC="gcc -Wall -fPIC"/' | \
+	sed 's/REPLACE_SHLD/SHLD="gcc -shared -I."/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="echo Skipping ranlib"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_sunos5/' > $OUTPUT_FILE
 else if ( $type == "IRIX" ) then
 	cat $INPUT_FILE | \
 	sed 's|REPLACE_CC|CC="/tsri/gnu/sgi4DIRIX6/bin/gcc"|' | \
+	sed 's/REPLACE_SHCC/CC="/tsri/gnu/sgi4DIRIX6/bin/gcc -Wall -fPIC"/' | \
+	sed 's/REPLACE_SHLD/SHLD="/tsri/gnu/sgi4DIRIX6/bin/gcc -shared -I."/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="echo Skipping ranlib"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_irix/' > $OUTPUT_FILE
 else if ( $type == "NetBSD" ) then
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/CC="cc -Wall"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -Wall -fPIC"/' | \
+	sed 's/REPLACE_SHLD/SHLD="gcc -shared -I."/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="ranlib"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_netbsd/' > $OUTPUT_FILE
 else if ( $type == "FreeBSD" ) then
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/CC="cc -Wall"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -Wall -fPIC"/' | \
+	sed 's/REPLACE_SHLD/SHLD="gcc -shared -I."/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="ranlib"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_freebsd/' > $OUTPUT_FILE
 else if ( $type == "Cygwin" ) then
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/"CC=cc"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -fPIC"/' | \
+	sed 's/REPLACE_SHLD/SHLD="gcc -shared -I."/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="echo Skipping ranlib"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX/_cygwin/' > $OUTPUT_FILE
 else
 	# Unknown operating system
 	cat $INPUT_FILE | \
 	sed 's/REPLACE_CC/"CC=cc"/' | \
+	sed 's/REPLACE_SHCC/CC="cc -fPIC"/' | \
+	sed 's/REPLACE_SHLD/SHLD="gcc -shared -I."/' | \
 	sed 's/REPLACE_RANLIB/RANLIB="echo Skipping ranlib"/' | \
-	sed "s|REPLACE_INSTALLDIR|${INSTALLDIR}|" | \
+	sed "s|REPLACE_PREFIX|${PREFIX}|" | \
 	sed 's/REPLACE_POSTFIX//' > $OUTPUT_FILE
 endif
 
@@ -164,13 +188,14 @@
 echo "    e-mail cdputnam@ucsd.edu with the system type and output of"
 echo "    the command: uname -a"
 echo
-echo "Set installation directory to $INSTALLDIR."
-echo "    To modify install directory type: configure --install-dir DIR"
+echo "Set installation directory to $PREFIX."
+echo "    To modify install directory type: configure --prefix DIR"
 echo "    where DIR is the desired directory."
 echo
 if ( $OUTPUT_FILE == "Makefile" ) then
   echo "To compile,                  type: make"
   echo "To install,                  type: make install"
+  echo "To install the library,      type: make installlib"
   echo "To make tgz package,         type: make package"
   echo "To make deb package,         type: make deb"
   echo
diff -ur ../bibutils_4.1/lib/Makefile ./lib/Makefile
--- ../bibutils_4.1/lib/Makefile	2008-12-21 00:17:31.000000000 +0100
+++ ./lib/Makefile	2008-12-26 11:19:48.000000000 +0100
@@ -24,6 +24,7 @@
 all:  libbibutils.a
 
 libbibutils.a: $(BIBUTILS_OBJS)
+	${SHLD} -o libbibutils.so $(BIBUTILS_OBJS)
 	ar -cr libbibutils.a $(BIBUTILS_OBJS)
 	$(RANLIB) libbibutils.a
 
