diff -Nuarp usr/src/alx-3.9/drivers/net/ethernet/atheros/alx_orig/alx_ethtool.c usr/src/alx-3.9/drivers/net/ethernet/atheros/alx/alx_ethtool.c
--- usr/src/alx-3.9/drivers/net/ethernet/atheros/alx_orig/alx_ethtool.c	2013-06-22 13:57:36.262985520 -0500
+++ usr/src/alx-3.9/drivers/net/ethernet/atheros/alx/alx_ethtool.c	2013-06-22 13:57:49.310986214 -0500
@@ -731,6 +731,7 @@ static int alx_diag_register(struct alx_
 
 	switch (ALX_DID(hw)) {
 	case ALX_DEV_ID_AR8161:
+	case ALX_DEV_ID_E2200:
 	case ALX_DEV_ID_AR8162:
 	case ALX_DEV_ID_AR8171:
 	case ALX_DEV_ID_AR8172:
diff -Nuarp usr/src/alx-3.9/drivers/net/ethernet/atheros/alx_orig/alx_main.c usr/src/alx-3.9/drivers/net/ethernet/atheros/alx/alx_main.c
--- usr/src/alx-3.9/drivers/net/ethernet/atheros/alx_orig/alx_main.c	2013-06-22 13:57:36.262985520 -0500
+++ usr/src/alx-3.9/drivers/net/ethernet/atheros/alx/alx_main.c	2013-06-22 13:57:49.310986214 -0500
@@ -57,6 +57,7 @@ static const char alx_drv_desc[] =
 	PCI_DEVICE(ALX_VENDOR_ID, device_id)}
 static DEFINE_PCI_DEVICE_TABLE(alx_pci_tbl) = {
 	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8161),
+	ALX_ETHER_DEVICE(ALX_DEV_ID_E2200),
 	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8162),
 	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8171),
 	ALX_ETHER_DEVICE(ALX_DEV_ID_AR8172),
@@ -1016,6 +1017,7 @@ static int alx_identify_hw(struct alx_ad
 
 	switch (ALX_DID(hw)) {
 	case ALX_DEV_ID_AR8161:
+	case ALX_DEV_ID_E2200:
 	case ALX_DEV_ID_AR8162:
 	case ALX_DEV_ID_AR8171:
 	case ALX_DEV_ID_AR8172:
diff -Nuarp usr/src/alx-3.9/drivers/net/ethernet/atheros/alx_orig/alx_reg.h usr/src/alx-3.9/drivers/net/ethernet/atheros/alx/alx_reg.h
--- usr/src/alx-3.9/drivers/net/ethernet/atheros/alx_orig/alx_reg.h	2013-06-22 13:57:36.262985520 -0500
+++ usr/src/alx-3.9/drivers/net/ethernet/atheros/alx/alx_reg.h	2013-06-22 13:57:50.798986293 -0500
@@ -25,6 +25,7 @@
 
 /* pci dev-ids */
 #define ALX_DEV_ID_AR8161               0x1091
+#define ALX_DEV_ID_E2200                0xe091
 #define ALX_DEV_ID_AR8162               0x1090
 #define ALX_DEV_ID_AR8171               0x10A1
 #define ALX_DEV_ID_AR8172               0x10A0
