--- usr/src/alx-3.9/Makefile.old	2013-06-09 10:40:32.513890452 -0500
+++ usr/src/alx-3.9/Makefile	2013-06-09 10:41:14.696889538 -0500
@@ -19,7 +19,32 @@
 ifneq ($(origin KLIB), undefined)
 KMODPATH_ARG:=  "INSTALL_MOD_PATH=$(KLIB)"
 else
-export KLIB:=          /lib/modules/$(shell uname -r)
+for i in \
+	`echo "linux linux-lts linux-ck linuxkernel26-ck linux-ck-corex linux-ck-p4 \
+	linux-ck-pentm linux-ck-atom linux-ck-core2 linux-ck-nehalem  \
+	linux-ck-ivybridge linux-ck-haswell linux-ck-kx linux-ck-k10 linux-ck-barcelona \
+	linux-ck-bulldozer linux-ck-piledriver linux-ck-sandybridge"`;
+do
+	pacman -Qi ${i} &> /dev/null
+	if [ "$?" = "0" ]; then
+		_KVER=`LC_ALL=C pacman -Qi ${i} | grep Version | sed "s/.\+: //g"`
+
+		_KTYPE=`echo ${i} | grep -o -e "ck\|lts"`
+		
+		case "${_KTYPE}" in
+			"ck")
+				export _KLIBVER=${_KVER}-CK
+				;;
+			"lts")
+				export _KLIBVER=${_KVER}-LTS
+				;;
+			*)
+				export _KLIBVER=${_KVER}-ARCH
+				;;
+		esac
+	fi
+done
+export KLIB:=          /lib/modules/${_KLIBVER}
 endif
 export KLIB_BUILD ?=	$(KLIB)/build
 export MAKE
