diff -rup nip2-7.30.0/src/parser.h nip2-7.30.0-new/src/parser.h
--- nip2-7.30.0/src/parser.h	2012-04-18 12:38:51.000000000 +0200
+++ nip2-7.30.0-new/src/parser.h	2012-07-26 17:51:13.313778170 +0200
@@ -55,7 +55,7 @@ extern InputState input_state;
 void nip2yyerror( const char *sub, ... )
 	__attribute__((format(printf, 1, 2)));
 void yyerror( const char *msg ); 
-extern int yyleng;			/* lex stuff */
+extern size_t yyleng;			/* lex stuff */
 
 /* Lex gathers tokens here for workspace.c
  */
