diff -aur PsychoPy-orig/docs/source/changelog.rst PsychoPy-1.77.02/docs/source/changelog.rst
--- PsychoPy-orig/docs/source/changelog.rst	2013-07-15 15:41:57.062334913 +0100
+++ PsychoPy-1.77.02/docs/source/changelog.rst	2013-07-15 15:44:56.504484049 +0100
@@ -876,7 +876,7 @@
 (released Jul 09)
 
 * DotStim can have fieldShape of 'sqr', 'square' or 'circle' (the first two are equiv)
-* CHANGED intepreters in all .py scripts to be the same (#!/usr/bin/env python). Use PATH env variable to choose non-default python version for your Python scripts
+* CHANGED intepreters in all .py scripts to be the same (#!/usr/bin/env python2). Use PATH env variable to choose non-default python version for your Python scripts
 * CHANGED pyglet textures to use numpy->ctypes rather than numpy->string
 * FIXED systemInfo assigned on Linux systems
 
diff -aur PsychoPy-orig/docs/source/demos/face.py PsychoPy-1.77.02/docs/source/demos/face.py
--- PsychoPy-orig/docs/source/demos/face.py	2013-07-15 15:41:57.149002570 +0100
+++ PsychoPy-1.77.02/docs/source/demos/face.py	2013-07-15 15:47:28.289771681 +0100
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python2.5
+#!/usr/bin/env python2
 from psychopy import core, visual, event
 import psychopy.sound
 #create a window to draw in
diff -aur PsychoPy-orig/docs/source/demos/plaid.py PsychoPy-1.77.02/docs/source/demos/plaid.py
--- PsychoPy-orig/docs/source/demos/plaid.py	2013-07-15 15:41:57.149002570 +0100
+++ PsychoPy-1.77.02/docs/source/demos/plaid.py	2013-07-15 15:47:48.633375407 +0100
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#!/usr/bin/env python2
 from psychopy import *
 
 #create a window to draw in
diff -aur PsychoPy-orig/docs/source/demos/text.py PsychoPy-1.77.02/docs/source/demos/text.py
--- PsychoPy-orig/docs/source/demos/text.py	2013-07-15 15:41:57.149002570 +0100
+++ PsychoPy-1.77.02/docs/source/demos/text.py	2013-07-15 15:44:57.001156918 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import *
 """
 Text rendering has changed a lot (for the better) under pyglet. This
diff -aur PsychoPy-orig/docs/source/developers/demoStyleGuide.rst PsychoPy-1.77.02/docs/source/developers/demoStyleGuide.rst
--- PsychoPy-orig/docs/source/developers/demoStyleGuide.rst	2013-07-15 15:41:57.099001999 +0100
+++ PsychoPy-1.77.02/docs/source/developers/demoStyleGuide.rst	2013-07-15 15:44:56.791154296 +0100
@@ -15,7 +15,7 @@
 
 - Standardize the top stuff to have 1) a shbang, 2) utf-8 encoding, and 3) a comment::
 
-    #!/usr/bin/env python
+    #!/usr/bin/env python2
     # -*- coding: utf-8 -*-
     """Demo name, purpose, description (1-2 sentences, although some demos need more explanation).
     """
diff -aur PsychoPy-orig/psychopy/app/builder/experiment.py PsychoPy-1.77.02/psychopy/app/builder/experiment.py
--- PsychoPy-orig/psychopy/app/builder/experiment.py	2013-07-15 15:41:57.752342804 +0100
+++ PsychoPy-1.77.02/psychopy/app/builder/experiment.py	2013-07-15 15:44:58.547842910 +0100
@@ -130,7 +130,7 @@
         else:
             localDateTime = data.getDateStr(format="%B %d, %Y, at %H:%M")
 
-        script.write('#!/usr/bin/env python\n' +
+        script.write('#!/usr/bin/env python2\n' +
                     '# -*- coding: utf-8 -*-\n' +
                     '"""\nThis experiment was created using PsychoPy2 Experiment Builder (v%s), %s\n' % (
                         self.psychopyVersion, localDateTime ) +
diff -aur PsychoPy-orig/psychopy/app/psychopyApp.py PsychoPy-1.77.02/psychopy/app/psychopyApp.py
--- PsychoPy-orig/psychopy/app/psychopyApp.py	2013-07-15 15:41:57.759009548 +0100
+++ PsychoPy-1.77.02/psychopy/app/psychopyApp.py	2013-07-15 15:44:58.681177910 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Part of the PsychoPy library
 # Copyright (C) 2013 Jonathan Peirce
diff -aur PsychoPy-orig/psychopy/CHANGELOG.txt PsychoPy-1.77.02/psychopy/CHANGELOG.txt
--- PsychoPy-orig/psychopy/CHANGELOG.txt	2013-07-15 15:41:57.415672288 +0100
+++ PsychoPy-1.77.02/psychopy/CHANGELOG.txt	2013-07-15 15:44:57.741166163 +0100
@@ -843,7 +843,7 @@
 (released Jul 09)
 
 * DotStim can have fieldShape of 'sqr', 'square' or 'circle' (the first two are equiv)
-* CHANGED intepreters in all .py scripts to be the same (#!/usr/bin/env python). Use PATH env variable to choose non-default python version for your Python scripts
+* CHANGED intepreters in all .py scripts to be the same (#!/usr/bin/env python2). Use PATH env variable to choose non-default python version for your Python scripts
 * CHANGED pyglet textures to use numpy->ctypes rather than numpy->string
 * FIXED systemInfo assigned on Linux systems
 
diff -aur PsychoPy-orig/psychopy/contrib/http/benchmark_report.py PsychoPy-1.77.02/psychopy/contrib/http/benchmark_report.py
--- PsychoPy-orig/psychopy/contrib/http/benchmark_report.py	2013-07-15 15:41:57.819010234 +0100
+++ PsychoPy-1.77.02/psychopy/contrib/http/benchmark_report.py	2013-07-15 15:44:59.117850035 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Go from multiple flat raw data files (.pkl) to html page, save .cache.pkl
diff -aur PsychoPy-orig/psychopy/contrib/mseq.py PsychoPy-1.77.02/psychopy/contrib/mseq.py
--- PsychoPy-orig/psychopy/contrib/mseq.py	2013-07-15 15:41:57.819010234 +0100
+++ PsychoPy-1.77.02/psychopy/contrib/mseq.py	2013-07-15 15:44:59.087849659 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 '''
 %        Maximum length sequence assuming 2,3,5 distinct values
 %
diff -aur PsychoPy-orig/psychopy/contrib/mseqSearch.py PsychoPy-1.77.02/psychopy/contrib/mseqSearch.py
--- PsychoPy-orig/psychopy/contrib/mseqSearch.py	2013-07-15 15:41:57.819010234 +0100
+++ PsychoPy-1.77.02/psychopy/contrib/mseqSearch.py	2013-07-15 15:44:59.107849910 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 %        Maximum length sequence assuming distinct values = baseVal
 %
diff -aur PsychoPy-orig/psychopy/contrib/opensslwrap.py PsychoPy-1.77.02/psychopy/contrib/opensslwrap.py
--- PsychoPy-orig/psychopy/contrib/opensslwrap.py	2013-07-15 15:41:57.819010234 +0100
+++ PsychoPy-1.77.02/psychopy/contrib/opensslwrap.py	2013-07-15 15:44:59.104516535 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """Basic file encryption for psych & human neuro research, using openssl."""
diff -aur PsychoPy-orig/psychopy/contrib/quest.py PsychoPy-1.77.02/psychopy/contrib/quest.py
--- PsychoPy-orig/psychopy/contrib/quest.py	2013-07-15 15:41:57.819010234 +0100
+++ PsychoPy-1.77.02/psychopy/contrib/quest.py	2013-07-15 15:44:59.094516410 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 1996-2002 Denis G. Pelli
 # Copyright (c) 1996-9 David Brainard
diff -aur "PsychoPy-orig/psychopy/demos/coder/experiment control/autoDraw_autoLog.py" "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/autoDraw_autoLog.py"
--- "PsychoPy-orig/psychopy/demos/coder/experiment control/autoDraw_autoLog.py"	2013-07-15 15:41:57.659008403 +0100
+++ "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/autoDraw_autoLog.py"	2013-07-15 15:44:58.204505286 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 
 """One easy way to handle stimuli that are drawn repeatedly is to 
diff -aur "PsychoPy-orig/psychopy/demos/coder/experiment control/fMRI_launchScan.py" "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/fMRI_launchScan.py"
--- "PsychoPy-orig/psychopy/demos/coder/experiment control/fMRI_launchScan.py"	2013-07-15 15:41:57.662341775 +0100
+++ "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/fMRI_launchScan.py"	2013-07-15 15:44:58.217838786 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """This demo illustrates using hardware.emulator.launchScan() to either start a
 real scan, or emulate sync pulses. Emulation is to allow debugging script timing
diff -aur "PsychoPy-orig/psychopy/demos/coder/experiment control/gammaMotionAnalysis.py" "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/gammaMotionAnalysis.py"
--- "PsychoPy-orig/psychopy/demos/coder/experiment control/gammaMotionAnalysis.py"	2013-07-15 15:41:57.659008403 +0100
+++ "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/gammaMotionAnalysis.py"	2013-07-15 15:44:58.187838410 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """Use this script to analyse data from the gammaMotionNull.py
 script. 
 
diff -aur "PsychoPy-orig/psychopy/demos/coder/experiment control/gammaMotionNull.py" "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/gammaMotionNull.py"
--- "PsychoPy-orig/psychopy/demos/coder/experiment control/gammaMotionNull.py"	2013-07-15 15:41:57.659008403 +0100
+++ "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/gammaMotionNull.py"	2013-07-15 15:44:58.177838286 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 """Determine screen gamma using motion-nulling method 
 of Ledgeway and Smith, 1994, Vision Research, 34, 2727–2740
diff -aur "PsychoPy-orig/psychopy/demos/coder/experiment control/JND_staircase_analysis.py" "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/JND_staircase_analysis.py"
--- "PsychoPy-orig/psychopy/demos/coder/experiment control/JND_staircase_analysis.py"	2013-07-15 15:41:57.659008403 +0100
+++ "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/JND_staircase_analysis.py"	2013-07-15 15:44:58.181171661 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #This analysis script takes one or more staircase datafiles as input from a GUI
 #It then plots the staircases on top of each other on the left 
diff -aur "PsychoPy-orig/psychopy/demos/coder/experiment control/JND_staircase_exp.py" "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/JND_staircase_exp.py"
--- "PsychoPy-orig/psychopy/demos/coder/experiment control/JND_staircase_exp.py"	2013-07-15 15:41:57.659008403 +0100
+++ "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/JND_staircase_exp.py"	2013-07-15 15:44:58.204505286 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """measure your JND in orientation using a staircase method"""
 
 from psychopy import core, visual, gui, data, misc, event
diff -aur "PsychoPy-orig/psychopy/demos/coder/experiment control/logFiles.py" "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/logFiles.py"
--- "PsychoPy-orig/psychopy/demos/coder/experiment control/logFiles.py"	2013-07-15 15:41:57.659008403 +0100
+++ "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/logFiles.py"	2013-07-15 15:44:58.167838161 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from psychopy import logging, core, visual
 
diff -aur "PsychoPy-orig/psychopy/demos/coder/experiment control/TrialHandler.py" "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/TrialHandler.py"
--- "PsychoPy-orig/psychopy/demos/coder/experiment control/TrialHandler.py"	2013-07-15 15:41:57.662341775 +0100
+++ "PsychoPy-1.77.02/psychopy/demos/coder/experiment control/TrialHandler.py"	2013-07-15 15:44:58.214505411 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from random import random
 from psychopy import data
 
diff -aur PsychoPy-orig/psychopy/demos/coder/hardware/bits_plus_plus.py PsychoPy-1.77.02/psychopy/demos/coder/hardware/bits_plus_plus.py
--- PsychoPy-orig/psychopy/demos/coder/hardware/bits_plus_plus.py	2013-07-15 15:41:57.532340290 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/hardware/bits_plus_plus.py	2013-07-15 15:44:57.981169161 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 import numpy
 
diff -aur PsychoPy-orig/psychopy/demos/coder/hardware/cedrusRB730.py PsychoPy-1.77.02/psychopy/demos/coder/hardware/cedrusRB730.py
--- PsychoPy-orig/psychopy/demos/coder/hardware/cedrusRB730.py	2013-07-15 15:41:57.532340290 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/hardware/cedrusRB730.py	2013-07-15 15:44:57.997836036 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy.hardware import cedrus
 from psychopy import core
 import sys
diff -aur PsychoPy-orig/psychopy/demos/coder/hardware/egi_netstation.py PsychoPy-1.77.02/psychopy/demos/coder/hardware/egi_netstation.py
--- PsychoPy-orig/psychopy/demos/coder/hardware/egi_netstation.py	2013-07-15 15:41:57.532340290 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/hardware/egi_netstation.py	2013-07-15 15:44:58.001169411 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """This demo comes from the simple_distilled example provided with pynetstation.
 Note that egi pynetstation can also be used in a multi-threaded form.
diff -aur PsychoPy-orig/psychopy/demos/coder/hardware/ioLab_bbox.py PsychoPy-1.77.02/psychopy/demos/coder/hardware/ioLab_bbox.py
--- PsychoPy-orig/psychopy/demos/coder/hardware/ioLab_bbox.py	2013-07-15 15:41:57.532340290 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/hardware/ioLab_bbox.py	2013-07-15 15:44:58.004502786 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """Demo program to illustrate using ioLabs button box
 
diff -aur PsychoPy-orig/psychopy/demos/coder/hardware/labjack_u3.py PsychoPy-1.77.02/psychopy/demos/coder/hardware/labjack_u3.py
--- PsychoPy-orig/psychopy/demos/coder/hardware/labjack_u3.py	2013-07-15 15:41:57.532340290 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/hardware/labjack_u3.py	2013-07-15 15:44:57.991169286 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """ demo for using labjack DAC devices
 
diff -aur PsychoPy-orig/psychopy/demos/coder/hardware/parallelPortOutput.py PsychoPy-1.77.02/psychopy/demos/coder/hardware/parallelPortOutput.py
--- PsychoPy-orig/psychopy/demos/coder/hardware/parallelPortOutput.py	2013-07-15 15:41:57.532340290 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/hardware/parallelPortOutput.py	2013-07-15 15:44:57.994502661 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #this is win32 only - I have no idea how to use parallel ports on a Mac! jwp
 from psychopy import visual, core, logging
 from psychopy import parallel 
diff -aur PsychoPy-orig/psychopy/demos/coder/hardware/testSoundLatency.py PsychoPy-1.77.02/psychopy/demos/coder/hardware/testSoundLatency.py
--- PsychoPy-orig/psychopy/demos/coder/hardware/testSoundLatency.py	2013-07-15 15:41:57.532340290 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/hardware/testSoundLatency.py	2013-07-15 15:44:57.987835911 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """This demo allows for automated testing of the sound latency on your system.
 To use it you need a labjack (or adapt for a similar device) and a cable to 
diff -aur PsychoPy-orig/psychopy/demos/coder/input/customMouse.py PsychoPy-1.77.02/psychopy/demos/coder/input/customMouse.py
--- PsychoPy-orig/psychopy/demos/coder/input/customMouse.py	2013-07-15 15:41:57.519006801 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/input/customMouse.py	2013-07-15 15:44:57.957835536 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # demo of CustomMouse()
 # author Jeremy Gray
diff -aur PsychoPy-orig/psychopy/demos/coder/input/GUI.py PsychoPy-1.77.02/psychopy/demos/coder/input/GUI.py
--- PsychoPy-orig/psychopy/demos/coder/input/GUI.py	2013-07-15 15:41:57.519006801 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/input/GUI.py	2013-07-15 15:44:57.951168787 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import gui
 
 # create a DlgFromDict
diff -aur PsychoPy-orig/psychopy/demos/coder/input/joystick_universal.py PsychoPy-1.77.02/psychopy/demos/coder/input/joystick_universal.py
--- PsychoPy-orig/psychopy/demos/coder/input/joystick_universal.py	2013-07-15 15:41:57.519006801 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/input/joystick_universal.py	2013-07-15 15:44:57.964502286 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 from psychopy.hardware import joystick
 
diff -aur PsychoPy-orig/psychopy/demos/coder/input/latencyFromTone.py PsychoPy-1.77.02/psychopy/demos/coder/input/latencyFromTone.py
--- PsychoPy-orig/psychopy/demos/coder/input/latencyFromTone.py	2013-07-15 15:41:57.519006801 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/input/latencyFromTone.py	2013-07-15 15:44:57.961168911 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """demo to illustrate and test microphone.AdvAudioCapture, and also permit
 external latency testing (e.g., BlackBox Toolkit) by providing visual-tone synchrony
 
diff -aur PsychoPy-orig/psychopy/demos/coder/input/mouse.py PsychoPy-1.77.02/psychopy/demos/coder/input/mouse.py
--- PsychoPy-orig/psychopy/demos/coder/input/mouse.py	2013-07-15 15:41:57.519006801 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/input/mouse.py	2013-07-15 15:44:57.977835786 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event, misc
 """
 As of version 1.51 the mouse coordinates for
diff -aur PsychoPy-orig/psychopy/demos/coder/input/speech_recognition.py PsychoPy-1.77.02/psychopy/demos/coder/input/speech_recognition.py
--- PsychoPy-orig/psychopy/demos/coder/input/speech_recognition.py	2013-07-15 15:41:57.519006801 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/input/speech_recognition.py	2013-07-15 15:44:57.967835662 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """Demo to illustrate speech recognition using microphone.Speech2Text
 
diff -aur PsychoPy-orig/psychopy/demos/coder/misc/captureFrames.py PsychoPy-1.77.02/psychopy/demos/coder/misc/captureFrames.py
--- PsychoPy-orig/psychopy/demos/coder/misc/captureFrames.py	2013-07-15 15:41:57.565674003 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/misc/captureFrames.py	2013-07-15 15:44:58.014502911 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual
 
 #copy pixels from the frame buffer
diff -aur PsychoPy-orig/psychopy/demos/coder/misc/http_upload.py PsychoPy-1.77.02/psychopy/demos/coder/misc/http_upload.py
--- PsychoPy-orig/psychopy/demos/coder/misc/http_upload.py	2013-07-15 15:41:57.565674003 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/misc/http_upload.py	2013-07-15 15:44:58.011169536 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """Illustrates using psychopy.web.upload() to send a file over the internet to a configured server.
 
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/bufferImageStim.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/bufferImageStim.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/bufferImageStim.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/bufferImageStim.py	2013-07-15 15:44:58.084503786 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """Demo of class psychopy.visual.BufferImageStim():
     - take a snapshot of a multi-item screen image
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/clockface.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/clockface.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/clockface.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/clockface.py	2013-07-15 15:44:58.117837536 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 import numpy, time    
 win = visual.Window([800,800], monitor='testMonitor')
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/colors.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/colors.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/colors.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/colors.py	2013-07-15 15:44:58.074503661 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, event, core, logging
 
 #demo color spaces
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/counterphase.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/counterphase.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/counterphase.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/counterphase.py	2013-07-15 15:44:58.141171161 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 There are many ways to generate counter-phase e.g. 
 vary the contrast of a grating sinusoidally between 1 and -1, 
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/customTextures.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/customTextures.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/customTextures.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/customTextures.py	2013-07-15 15:44:58.104504036 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, filters, event, core
 import numpy as np
 """Textures (e.g. for a GratingStim) can be created from custom numpy arrays. 
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/dot_gabors.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/dot_gabors.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/dot_gabors.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/dot_gabors.py	2013-07-15 15:44:58.037836536 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 
 #create a window to draw in
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/dots.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/dots.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/dots.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/dots.py	2013-07-15 15:44:58.064503535 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, event, core
 
 #create a window to draw in
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/elementArrays.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/elementArrays.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/elementArrays.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/elementArrays.py	2013-07-15 15:44:58.071170286 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """This script demonstrates the use of the ElementArrayStim, a highly optimised stimulus for generating 
 arrays of similar (but not identical) elements, such as in global form arrays or random dot stimuli.
 
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/embeddedOpenGL.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/embeddedOpenGL.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/embeddedOpenGL.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/embeddedOpenGL.py	2013-07-15 15:44:58.147837910 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 This demo shows how you can make standard opengl calls within a
 psychopy script, allowing you to draw anything that OpenGL can draw
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/face_jpg.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/face_jpg.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/face_jpg.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/face_jpg.py	2013-07-15 15:44:58.091170536 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import core, visual, event
 import psychopy.sound
 #create a window to draw in
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/gabor.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/gabor.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/gabor.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/gabor.py	2013-07-15 15:44:58.097837286 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import core, visual, event
 
 #create a window to draw in
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/imagesAndPatches.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/imagesAndPatches.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/imagesAndPatches.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/imagesAndPatches.py	2013-07-15 15:44:58.094503911 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import core, visual, event
 
 #create a window to draw in
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/maskReveal.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/maskReveal.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/maskReveal.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/maskReveal.py	2013-07-15 15:44:58.057836786 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """When you want to reveal an image gradually from behind a mask, 
 the tempting thing to do is to alter a stimulus mask using .setMask()
 
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/plaid.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/plaid.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/plaid.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/plaid.py	2013-07-15 15:44:58.101170661 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, logging, event, core
 
 #create a window to draw in
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/ratingScale.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/ratingScale.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/ratingScale.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/ratingScale.py	2013-07-15 15:44:58.044503286 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """ demo for the class psychopy.visual.RatingScale()
 author: Jeremy Gray, modified by Henrik Singmann
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/rotatingFlashingWedge.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/rotatingFlashingWedge.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/rotatingFlashingWedge.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/rotatingFlashingWedge.py	2013-07-15 15:44:58.077837036 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #rotate flashing wedge
 from psychopy import visual, event, core
 
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/screensAndWindows.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/screensAndWindows.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/screensAndWindows.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/screensAndWindows.py	2013-07-15 15:44:58.111170785 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """Using multiple screens and windows with PsychoPy
 """
 
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/soundStimuli.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/soundStimuli.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/soundStimuli.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/soundStimuli.py	2013-07-15 15:44:58.114504161 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Sound stimuli are currently an area of development in PsychoPy
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/starField.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/starField.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/starField.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/starField.py	2013-07-15 15:44:58.061170161 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """This demo requires a graphics card that supports OpenGL2 extensions. 
 
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/textStimuli.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/textStimuli.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/textStimuli.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/textStimuli.py	2013-07-15 15:44:58.041169911 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 """
 Text rendering has changed a lot (for the better) under pyglet. This
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/textStimuli_pygame.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/textStimuli_pygame.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/textStimuli_pygame.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/textStimuli_pygame.py	2013-07-15 15:44:58.121170911 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 import sys
 
diff -aur PsychoPy-orig/psychopy/demos/coder/stimuli/visual_noise.py PsychoPy-1.77.02/psychopy/demos/coder/stimuli/visual_noise.py
--- PsychoPy-orig/psychopy/demos/coder/stimuli/visual_noise.py	2013-07-15 15:41:57.632341433 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/stimuli/visual_noise.py	2013-07-15 15:44:58.054503411 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #demo arbitrary numpy array
 
 from psychopy import visual, core, event, logging
diff -aur PsychoPy-orig/psychopy/demos/coder/timing/timeByFrames.py PsychoPy-1.77.02/psychopy/demos/coder/timing/timeByFrames.py
--- PsychoPy-orig/psychopy/demos/coder/timing/timeByFrames.py	2013-07-15 15:41:57.592340975 +0100
+++ PsychoPy-1.77.02/psychopy/demos/coder/timing/timeByFrames.py	2013-07-15 15:44:58.024503036 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #The most accurate way to time your stimulus presentation is to
 #present for a certain number of frames. For that to work you need
diff -aur PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/app/base.py PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/app/base.py
--- PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/app/base.py	2013-07-15 15:41:57.239003600 +0100
+++ PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/app/base.py	2013-07-15 15:44:57.184492542 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/carbon_hid.py PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/carbon_hid.py
--- PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/carbon_hid.py	2013-07-15 15:41:57.209003257 +0100
+++ PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/carbon_hid.py	2013-07-15 15:44:57.124491792 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/carbon_tablet.py PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/carbon_tablet.py
--- PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/carbon_tablet.py	2013-07-15 15:41:57.209003257 +0100
+++ PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/carbon_tablet.py	2013-07-15 15:44:57.117825043 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/directinput.py PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/directinput.py
--- PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/directinput.py	2013-07-15 15:41:57.209003257 +0100
+++ PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/directinput.py	2013-07-15 15:48:04.303584966 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
 # $Id:$
 
 import ctypes
diff -aur PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/evdev_constants.py PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/evdev_constants.py
--- PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/evdev_constants.py	2013-07-15 15:41:57.209003257 +0100
+++ PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/evdev_constants.py	2013-07-15 15:44:57.171159042 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''Event constants from /usr/include/linux/input.h
 '''
diff -aur PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/evdev.py PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/evdev.py
--- PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/evdev.py	2013-07-15 15:41:57.209003257 +0100
+++ PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/evdev.py	2013-07-15 15:44:57.177825791 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/wintab.py PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/wintab.py
--- PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/wintab.py	2013-07-15 15:41:57.209003257 +0100
+++ PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/wintab.py	2013-07-15 15:48:27.143892370 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
 # $Id:$
 
 import ctypes
diff -aur PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/x11_xinput.py PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/x11_xinput.py
--- PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/x11_xinput.py	2013-07-15 15:41:57.209003257 +0100
+++ PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/x11_xinput.py	2013-07-15 15:44:57.164492292 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/x11_xinput_tablet.py PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/x11_xinput_tablet.py
--- PsychoPy-orig/psychopy/hardware/joystick/pyglet_input/x11_xinput_tablet.py	2013-07-15 15:41:57.209003257 +0100
+++ PsychoPy-1.77.02/psychopy/hardware/joystick/pyglet_input/x11_xinput_tablet.py	2013-07-15 15:44:57.107824917 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur PsychoPy-orig/psychopy/monitors/MonitorCenter.py PsychoPy-1.77.02/psychopy/monitors/MonitorCenter.py
--- PsychoPy-orig/psychopy/monitors/MonitorCenter.py	2013-07-15 15:41:57.429005772 +0100
+++ PsychoPy-1.77.02/psychopy/monitors/MonitorCenter.py	2013-07-15 15:44:57.807833662 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Part of the PsychoPy library
 # Copyright (C) 2013 Jonathan Peirce
diff -aur PsychoPy-orig/psychopy/preferences/generateSpec.py PsychoPy-1.77.02/psychopy/preferences/generateSpec.py
--- PsychoPy-orig/psychopy/preferences/generateSpec.py	2013-07-15 15:41:57.819010234 +0100
+++ PsychoPy-1.77.02/psychopy/preferences/generateSpec.py	2013-07-15 15:44:59.161183910 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # generate .spec files for all OS's based on differences from baseNoArch.spec
 # copies & tweaks baseNoArch.spec -> write out as platform specific .spec file
diff -aur PsychoPy-orig/psychopy/tests/run.py PsychoPy-1.77.02/psychopy/tests/run.py
--- PsychoPy-orig/psychopy/tests/run.py	2013-07-15 15:41:57.809010120 +0100
+++ PsychoPy-1.77.02/psychopy/tests/run.py	2013-07-15 15:44:59.061182660 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import sys, os
 import subprocess
 
diff -aur PsychoPy-orig/psychopy/tests/runPytest.py PsychoPy-1.77.02/psychopy/tests/runPytest.py
--- PsychoPy-orig/psychopy/tests/runPytest.py	2013-07-15 15:41:57.809010120 +0100
+++ PsychoPy-1.77.02/psychopy/tests/runPytest.py	2013-07-15 15:44:59.047849160 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 
 sources = """
 eNrsvV2TI0eSIDZ3J93qIO3trnRafdzuKYlSX2ayUeiqJjkfOIIjTrN7t3c5ZBu7e5dnxTIwC8hC
diff -aur PsychoPy-orig/psychopy/web.py PsychoPy-1.77.02/psychopy/web.py
--- PsychoPy-orig/psychopy/web.py	2013-07-15 15:41:57.172336169 +0100
+++ PsychoPy-1.77.02/psychopy/web.py	2013-07-15 15:44:57.057824293 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """Library for working with internet connections"""
diff -aur PsychoPy-orig/psychopy/wizard.py PsychoPy-1.77.02/psychopy/wizard.py
--- PsychoPy-orig/psychopy/wizard.py	2013-07-15 15:41:57.425672401 +0100
+++ PsychoPy-1.77.02/psychopy/wizard.py	2013-07-15 15:44:57.787833413 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """Libraries for wizards, currently firstrun configuration and benchmark."""
diff -aur PsychoPy-orig/setup.py PsychoPy-1.77.02/setup.py
--- PsychoPy-orig/setup.py	2013-07-15 15:41:57.819010234 +0100
+++ PsychoPy-1.77.02/setup.py	2013-07-15 15:44:59.174517411 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """Requires setuptools and uses the manifest.in file for data files"""
 
 from setuptools import setup, Extension, find_packages
