diff -aur OLD.Psychopy/docs/source/changelog.rst PsychoPy-1.73.04/docs/source/changelog.rst
--- OLD.Psychopy/docs/source/changelog.rst	2012-02-08 12:07:20.000000000 +0000
+++ PsychoPy-1.73.04/docs/source/changelog.rst	2012-03-02 11:24:58.466424242 +0000
@@ -663,7 +663,7 @@
 (released Jul 09)
 
 * DotStim can have fieldShape of 'sqr', 'square' or 'circle' (the first two are equiv)
-* CHANGED intepreters in all .py scripts to be the same (#!/usr/bin/env python). Use PATH env variable to choose non-default python version for your Python scripts
+* CHANGED intepreters in all .py scripts to be the same (#!/usr/bin/env python2). Use PATH env variable to choose non-default python version for your Python scripts
 * CHANGED pyglet textures to use numpy->ctypes rather than numpy->string
 * FIXED systemInfo assigned on Linux systems
 
diff -aur OLD.Psychopy/docs/source/demos/text.py PsychoPy-1.73.04/docs/source/demos/text.py
--- OLD.Psychopy/docs/source/demos/text.py	2012-01-30 17:44:00.000000000 +0000
+++ PsychoPy-1.73.04/docs/source/demos/text.py	2012-03-02 11:24:58.853097378 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import *
 """
 Text rendering has changed a lot (for the better) under pyglet. This
diff -aur OLD.Psychopy/psychopy/app/builder/experiment.py PsychoPy-1.73.04/psychopy/app/builder/experiment.py
--- OLD.Psychopy/psychopy/app/builder/experiment.py	2012-02-08 11:59:46.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/app/builder/experiment.py	2012-03-02 11:24:59.639777205 +0000
@@ -106,7 +106,7 @@
         """
         self.expPath = expPath
         script = IndentingBuffer(u'') #a string buffer object
-        script.write('#!/usr/bin/env python\n' +
+        script.write('#!/usr/bin/env python2\n' +
                     '# -*- coding: utf-8 -*-\n' +
                     '"""\nThis experiment was created using PsychoPy2 Experiment Builder (v%s), %s\n' % (
                         self.psychopyVersion, data.getDateStr(format="%B %d, %Y, at %H:%M") ) +
diff -aur OLD.Psychopy/psychopy/app/psychopyApp.py PsychoPy-1.73.04/psychopy/app/psychopyApp.py
--- OLD.Psychopy/psychopy/app/psychopyApp.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/app/psychopyApp.py	2012-03-02 11:24:59.756445823 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Part of the PsychoPy library
 # Copyright (C) 2011 Jonathan Peirce
diff -aur OLD.Psychopy/psychopy/CHANGELOG.txt PsychoPy-1.73.04/psychopy/CHANGELOG.txt
--- OLD.Psychopy/psychopy/CHANGELOG.txt	2012-02-08 12:07:06.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/CHANGELOG.txt	2012-03-02 11:24:59.113101728 +0000
@@ -645,7 +645,7 @@
 (released Jul 09)
 
 * DotStim can have fieldShape of 'sqr', 'square' or 'circle' (the first two are equiv)
-* CHANGED intepreters in all .py scripts to be the same (#!/usr/bin/env python). Use PATH env variable to choose non-default python version for your Python scripts
+* CHANGED intepreters in all .py scripts to be the same (#!/usr/bin/env python2). Use PATH env variable to choose non-default python version for your Python scripts
 * CHANGED pyglet textures to use numpy->ctypes rather than numpy->string
 * FIXED systemInfo assigned on Linux systems
 
diff -aur OLD.Psychopy/psychopy/contrib/mseq.py PsychoPy-1.73.04/psychopy/contrib/mseq.py
--- OLD.Psychopy/psychopy/contrib/mseq.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/contrib/mseq.py	2012-03-02 11:24:59.996449838 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 '''
 %        Maximum length sequence assuming 2,3,5 distinct values
 %
diff -aur OLD.Psychopy/psychopy/contrib/mseqSearch.py PsychoPy-1.73.04/psychopy/contrib/mseqSearch.py
--- OLD.Psychopy/psychopy/contrib/mseqSearch.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/contrib/mseqSearch.py	2012-03-02 11:25:00.006450006 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 %        Maximum length sequence assuming distinct values = baseVal
 %
diff -aur OLD.Psychopy/psychopy/contrib/quest.py PsychoPy-1.73.04/psychopy/contrib/quest.py
--- OLD.Psychopy/psychopy/contrib/quest.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/contrib/quest.py	2012-03-02 11:24:59.999783227 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 1996-2002 Denis G. Pelli
 # Copyright (c) 1996-9 David Brainard
diff -aur OLD.Psychopy/psychopy/demos/coder/experiment control/autoDraw_autoLog.py PsychoPy-1.73.04/psychopy/demos/coder/experiment control/autoDraw_autoLog.py
--- OLD.Psychopy/psychopy/demos/coder/experiment control/autoDraw_autoLog.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/experiment control/autoDraw_autoLog.py	2012-03-02 11:24:59.359772520 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 
 """One easy way to handle stimuli that are drawn repeatedly is to 
diff -aur OLD.Psychopy/psychopy/demos/coder/experiment control/fMRI_launchScan.py PsychoPy-1.73.04/psychopy/demos/coder/experiment control/fMRI_launchScan.py
--- OLD.Psychopy/psychopy/demos/coder/experiment control/fMRI_launchScan.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/experiment control/fMRI_launchScan.py	2012-03-02 11:24:59.369772688 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """This demo illustrates using hardware.emulator.launchScan() to either start a real scan, 
 or emulate sync pulses and user responses. Emulation is to allow debugging script timing
diff -aur OLD.Psychopy/psychopy/demos/coder/experiment control/gammaMotionAnalysis.py PsychoPy-1.73.04/psychopy/demos/coder/experiment control/gammaMotionAnalysis.py
--- OLD.Psychopy/psychopy/demos/coder/experiment control/gammaMotionAnalysis.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/experiment control/gammaMotionAnalysis.py	2012-03-02 11:24:59.346438964 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """Use this script to analyse data from the gammaMotionNull.py
 script. 
 
diff -aur OLD.Psychopy/psychopy/demos/coder/experiment control/gammaMotionNull.py PsychoPy-1.73.04/psychopy/demos/coder/experiment control/gammaMotionNull.py
--- OLD.Psychopy/psychopy/demos/coder/experiment control/gammaMotionNull.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/experiment control/gammaMotionNull.py	2012-03-02 11:24:59.339772186 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 """Determine screen gamma using motion-nulling method 
 of Ledgeway and Smith, 1994, Vision Research, 34, 2727–2740
diff -aur OLD.Psychopy/psychopy/demos/coder/experiment control/JND_staircase_analysis.py PsychoPy-1.73.04/psychopy/demos/coder/experiment control/JND_staircase_analysis.py
--- OLD.Psychopy/psychopy/demos/coder/experiment control/JND_staircase_analysis.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/experiment control/JND_staircase_analysis.py	2012-03-02 11:24:59.343105575 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #This analysis script takes one or more staircase datafiles as input from a GUI
 #It then plots the staircases on top of each other on the left 
diff -aur OLD.Psychopy/psychopy/demos/coder/experiment control/JND_staircase_exp.py PsychoPy-1.73.04/psychopy/demos/coder/experiment control/JND_staircase_exp.py
--- OLD.Psychopy/psychopy/demos/coder/experiment control/JND_staircase_exp.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/experiment control/JND_staircase_exp.py	2012-03-02 11:24:59.363105909 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """measure your JND in orientation using a staircase method"""
 
 from psychopy import core, visual, gui, data, misc, event
diff -aur OLD.Psychopy/psychopy/demos/coder/experiment control/logFiles.py PsychoPy-1.73.04/psychopy/demos/coder/experiment control/logFiles.py
--- OLD.Psychopy/psychopy/demos/coder/experiment control/logFiles.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/experiment control/logFiles.py	2012-03-02 11:24:59.336438797 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from psychopy import logging, core, visual
 
diff -aur OLD.Psychopy/psychopy/demos/coder/experiment control/TrialHandler.py PsychoPy-1.73.04/psychopy/demos/coder/experiment control/TrialHandler.py
--- OLD.Psychopy/psychopy/demos/coder/experiment control/TrialHandler.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/experiment control/TrialHandler.py	2012-03-02 11:24:59.366439299 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from random import random
 from psychopy import data
 
diff -aur OLD.Psychopy/psychopy/demos/coder/hardware/bits_plus_plus.py PsychoPy-1.73.04/psychopy/demos/coder/hardware/bits_plus_plus.py
--- OLD.Psychopy/psychopy/demos/coder/hardware/bits_plus_plus.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/hardware/bits_plus_plus.py	2012-03-02 11:24:59.189769676 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 import numpy
 
diff -aur OLD.Psychopy/psychopy/demos/coder/hardware/cedrusRB730.py PsychoPy-1.73.04/psychopy/demos/coder/hardware/cedrusRB730.py
--- OLD.Psychopy/psychopy/demos/coder/hardware/cedrusRB730.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/hardware/cedrusRB730.py	2012-03-02 11:24:59.199769844 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy.hardware import cedrus
 from psychopy import core
 import sys
diff -aur OLD.Psychopy/psychopy/demos/coder/hardware/egi_netstation.py PsychoPy-1.73.04/psychopy/demos/coder/hardware/egi_netstation.py
--- OLD.Psychopy/psychopy/demos/coder/hardware/egi_netstation.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/hardware/egi_netstation.py	2012-03-02 11:24:59.203103233 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """This demo comes from the simple_distilled example provided with pynetstation.
 Note that egi pynetstation can also be used in a multi-threaded form.
diff -aur OLD.Psychopy/psychopy/demos/coder/hardware/ioLab_bbox.py PsychoPy-1.73.04/psychopy/demos/coder/hardware/ioLab_bbox.py
--- OLD.Psychopy/psychopy/demos/coder/hardware/ioLab_bbox.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/hardware/ioLab_bbox.py	2012-03-02 11:24:59.206436622 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """Demo program to illustrate using ioLabs button box
 
diff -aur OLD.Psychopy/psychopy/demos/coder/hardware/labjack_u3.py PsychoPy-1.73.04/psychopy/demos/coder/hardware/labjack_u3.py
--- OLD.Psychopy/psychopy/demos/coder/hardware/labjack_u3.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/hardware/labjack_u3.py	2012-03-02 11:24:59.193103066 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """ demo for using labjack DAC devices
 
diff -aur OLD.Psychopy/psychopy/demos/coder/hardware/parallelPortOutput.py PsychoPy-1.73.04/psychopy/demos/coder/hardware/parallelPortOutput.py
--- OLD.Psychopy/psychopy/demos/coder/hardware/parallelPortOutput.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/hardware/parallelPortOutput.py	2012-03-02 11:24:59.196436455 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #this is win32 only - I have no idea how to use parallel ports on a Mac! jwp
 from psychopy import visual, core, logging
 from psychopy import parallel 
diff -aur OLD.Psychopy/psychopy/demos/coder/input/customMouse.py PsychoPy-1.73.04/psychopy/demos/coder/input/customMouse.py
--- OLD.Psychopy/psychopy/demos/coder/input/customMouse.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/input/customMouse.py	2012-03-02 11:24:59.176436120 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # demo of CustomMouse()
 # author Jeremy Gray
diff -aur OLD.Psychopy/psychopy/demos/coder/input/GUI.py PsychoPy-1.73.04/psychopy/demos/coder/input/GUI.py
--- OLD.Psychopy/psychopy/demos/coder/input/GUI.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/input/GUI.py	2012-03-02 11:24:59.176436120 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import gui
 
 # create a DlgFromDict
diff -aur OLD.Psychopy/psychopy/demos/coder/input/joystick_universal.py PsychoPy-1.73.04/psychopy/demos/coder/input/joystick_universal.py
--- OLD.Psychopy/psychopy/demos/coder/input/joystick_universal.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/input/joystick_universal.py	2012-03-02 11:24:59.179769510 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 from psychopy.hardware import joystick
 
diff -aur OLD.Psychopy/psychopy/demos/coder/input/mouse.py PsychoPy-1.73.04/psychopy/demos/coder/input/mouse.py
--- OLD.Psychopy/psychopy/demos/coder/input/mouse.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/input/mouse.py	2012-03-02 11:24:59.186436287 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event, misc
 """
 As of version 1.51 the mouse coordinates for
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/bufferImageStim.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/bufferImageStim.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/bufferImageStim.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/bufferImageStim.py	2012-03-02 11:24:59.269771015 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 bufferImageStim.py
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/clockface.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/clockface.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/clockface.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/clockface.py	2012-03-02 11:24:59.293104738 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 import numpy, time    
 win = visual.Window([800,800], monitor='testMonitor')
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/colors.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/colors.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/colors.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/colors.py	2012-03-02 11:24:59.263104237 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, event, core, logging
 
 #demo color spaces
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/counterphase.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/counterphase.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/counterphase.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/counterphase.py	2012-03-02 11:24:59.313105073 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 There are many ways to generate counter-phase e.g. 
 vary the contrast of a grating sinusoidally between 1 and -1, 
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/customTextures.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/customTextures.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/customTextures.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/customTextures.py	2012-03-02 11:24:59.283104572 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, filters, event, core
 import numpy as np
 """Textures (e.g. for a PatchStim) can be created from custom numpy arrays. 
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/dot_gabors.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/dot_gabors.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/dot_gabors.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/dot_gabors.py	2012-03-02 11:24:59.233103735 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 
 #create a window to draw in
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/dots.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/dots.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/dots.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/dots.py	2012-03-02 11:24:59.253104070 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, event, core
 
 #create a window to draw in
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/elementArrays.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/elementArrays.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/elementArrays.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/elementArrays.py	2012-03-02 11:24:59.259770847 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """This script demonstrates the use of the ElementArrayStim, a highly optimised stimulus for generating 
 arrays of similar (but not identical) elements, such as in global form arrays or random dot stimuli.
 
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/embeddedOpenGL.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/embeddedOpenGL.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/embeddedOpenGL.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/embeddedOpenGL.py	2012-03-02 11:24:59.316438463 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 This demo shows how you can make standard opengl calls within a
 psychopy script, allowing you to draw anything that OpenGL can draw
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/face_jpg.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/face_jpg.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/face_jpg.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/face_jpg.py	2012-03-02 11:24:59.273104404 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import core, visual, event
 import psychopy.sound
 #create a window to draw in
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/gabor.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/gabor.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/gabor.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/gabor.py	2012-03-02 11:24:59.279771182 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import core, visual, event
 
 #create a window to draw in
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/imagesAndPatches.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/imagesAndPatches.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/imagesAndPatches.py	2012-01-30 17:44:02.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/imagesAndPatches.py	2012-03-02 11:24:59.276437793 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import core, visual, event
 
 """There are two options for drawing image-based stimuli in PsychoPy:
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/maskReveal.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/maskReveal.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/maskReveal.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/maskReveal.py	2012-03-02 11:24:59.249770681 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """When you want to reveal an image gradually from behind a mask, 
 the tempting thing to do is to alter a stimulus mask using .setMask()
 
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/plaidcm.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/plaidcm.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/plaidcm.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/plaidcm.py	2012-03-02 11:24:59.309771684 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 
 #create a window to draw in
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/plaid.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/plaid.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/plaid.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/plaid.py	2012-03-02 11:24:59.279771182 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, logging, event, core
 
 #create a window to draw in
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/ratingScale.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/ratingScale.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/ratingScale.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/ratingScale.py	2012-03-02 11:24:59.236437124 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """ demo for the class psychopy.visual.RatingScale()
 """
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/rotatingFlashingWedge.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/rotatingFlashingWedge.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/rotatingFlashingWedge.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/rotatingFlashingWedge.py	2012-03-02 11:24:59.266437626 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #rotate flashing wedge
 from psychopy import visual, event, core
 
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/screensAndWindows.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/screensAndWindows.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/screensAndWindows.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/screensAndWindows.py	2012-03-02 11:24:59.286437961 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """Using multiple screens and windows with PsychoPy
 """
 
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/soundStimuli.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/soundStimuli.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/soundStimuli.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/soundStimuli.py	2012-03-02 11:24:59.289771349 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 For sound use, I really recommend installing pygame (v1.8 or later). 
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/starField.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/starField.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/starField.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/starField.py	2012-03-02 11:24:59.249770681 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """This demo requires a graphics card that supports OpenGL2 extensions. 
 
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/textStimuli.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/textStimuli.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/textStimuli.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/textStimuli.py	2012-03-02 11:24:59.233103735 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 """
 Text rendering has changed a lot (for the better) under pyglet. This
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/textStimuli_pygame.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/textStimuli_pygame.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/textStimuli_pygame.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/textStimuli_pygame.py	2012-03-02 11:24:59.296438128 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, event
 import sys
 
diff -aur OLD.Psychopy/psychopy/demos/coder/stimuli/visual_noise.py PsychoPy-1.73.04/psychopy/demos/coder/stimuli/visual_noise.py
--- OLD.Psychopy/psychopy/demos/coder/stimuli/visual_noise.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/stimuli/visual_noise.py	2012-03-02 11:24:59.246437291 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #demo arbitrary numpy array
 
 from psychopy import visual, core, event, logging
diff -aur OLD.Psychopy/psychopy/demos/coder/timing/pylabPlotFrameTimes.py PsychoPy-1.73.04/psychopy/demos/coder/timing/pylabPlotFrameTimes.py
--- OLD.Psychopy/psychopy/demos/coder/timing/pylabPlotFrameTimes.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/timing/pylabPlotFrameTimes.py	2012-03-02 11:24:59.219770178 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 from psychopy import visual, core, logging
 import pylab
 
diff -aur OLD.Psychopy/psychopy/demos/coder/timing/timeByFrames.py PsychoPy-1.73.04/psychopy/demos/coder/timing/timeByFrames.py
--- OLD.Psychopy/psychopy/demos/coder/timing/timeByFrames.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/demos/coder/timing/timeByFrames.py	2012-03-02 11:24:59.219770178 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #The most accurate way to time your stimulus presentation is to
 #present for a certain number of frames. For that to work you need
diff -aur OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/app/base.py PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/app/base.py
--- OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/app/base.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/app/base.py	2012-03-02 11:24:59.006433276 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/carbon_hid.py PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/carbon_hid.py
--- OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/carbon_hid.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/carbon_hid.py	2012-03-02 11:24:58.969765996 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/carbon_tablet.py PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/carbon_tablet.py
--- OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/carbon_tablet.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/carbon_tablet.py	2012-03-02 11:24:58.963099218 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/evdev_constants.py PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/evdev_constants.py
--- OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/evdev_constants.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/evdev_constants.py	2012-03-02 11:24:58.989766331 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''Event constants from /usr/include/linux/input.h
 '''
diff -aur OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/evdev.py PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/evdev.py
--- OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/evdev.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/evdev.py	2012-03-02 11:24:58.996433109 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/x11_xinput.py PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/x11_xinput.py
--- OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/x11_xinput.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/x11_xinput.py	2012-03-02 11:24:58.983099552 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/x11_xinput_tablet.py PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/x11_xinput_tablet.py
--- OLD.Psychopy/psychopy/hardware/joystick/pyglet_input/x11_xinput_tablet.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/hardware/joystick/pyglet_input/x11_xinput_tablet.py	2012-03-02 11:24:58.956432440 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 '''
diff -aur OLD.Psychopy/psychopy/monitors/MonitorCenter.py PsychoPy-1.73.04/psychopy/monitors/MonitorCenter.py
--- OLD.Psychopy/psychopy/monitors/MonitorCenter.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/monitors/MonitorCenter.py	2012-03-02 11:24:59.146435619 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Part of the PsychoPy library
 # Copyright (C) 2011 Jonathan Peirce
diff -aur OLD.Psychopy/psychopy/preferences/generateSpec.py PsychoPy-1.73.04/psychopy/preferences/generateSpec.py
--- OLD.Psychopy/psychopy/preferences/generateSpec.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/preferences/generateSpec.py	2012-03-02 11:25:00.046450674 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # generate .spec files for all OS's based on differences from baseNoArch.spec
 # copies & tweaks baseNoArch.spec -> write out as platform specific .spec file
diff -aur OLD.Psychopy/psychopy/tests/run.py PsychoPy-1.73.04/psychopy/tests/run.py
--- OLD.Psychopy/psychopy/tests/run.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/tests/run.py	2012-03-02 11:24:59.959782558 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import nose, sys, os
 
 thisDir,filename = os.path.split(os.path.abspath(__file__))
diff -aur OLD.Psychopy/psychopy/tests/testAllVisual/testAllStimuli.py PsychoPy-1.73.04/psychopy/tests/testAllVisual/testAllStimuli.py
--- OLD.Psychopy/psychopy/tests/testAllVisual/testAllStimuli.py	2012-01-30 17:44:04.000000000 +0000
+++ PsychoPy-1.73.04/psychopy/tests/testAllVisual/testAllStimuli.py	2012-03-02 11:24:59.986449671 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import nose, sys, os, copy
 from psychopy import visual, misc, core, monitors, filters
 from psychopy.tests import utils
diff -aur OLD.Psychopy/setup.py PsychoPy-1.73.04/setup.py
--- OLD.Psychopy/setup.py	2012-01-30 17:44:06.000000000 +0000
+++ PsychoPy-1.73.04/setup.py	2012-03-02 11:25:00.056450842 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """Requires setuptools and uses the manifest.in file for data files"""
 
 from setuptools import setup, Extension, find_packages
