--- Makefile.orig	2010-03-07 16:31:11.832208042 +0100
+++ Makefile	2010-03-07 16:22:27.662260898 +0100
@@ -23,7 +23,7 @@
 CPPFLAGS = $(OPTIM_FLAGS) -Wall
 LDFLAGS +=
 LDLIBS +=
-
+DESTDIR = ""
 
 # Targets
 #
@@ -39,6 +39,8 @@
 acwjack.o:	acwjack.cc 
 	g++ -c $(CPPFLAGS) $(SHOBJ_FLAGS) acwjack.cc
 
+acweight.o:	acweight.cc 
+	g++ -c $(CPPFLAGS) $(SHOBJ_FLAGS) acweight.cc
 
 
 ACWPLUG_O = acwplug.o acwplug_if.o acweight.o
@@ -57,9 +59,8 @@
 	g++ -o acwtest acwtest.o acweight.o -lfftw3f -lm
 
 install:	acwjack.so acwplug.so	
-	/bin/cp  acwjack.so /usr/lib/jack
-	/bin/cp  acwplug.so /usr/lib/ladspa
-
+	/bin/install -Dm 755 acwjack.so $(DESTDIR)/usr/lib/jack/acwjack.so
+	/bin/install -Dm 755 acwplug.so $(DESTDIR)/usr/lib/ladspa/acwplug.so
 clean:
 	/bin/rm -f *.o *.u *.so *~
 
