diff --git a/Makefile b/Makefile
index 3ddf8b9..dc2c5b0 100644
--- a/Makefile
+++ b/Makefile
@@ -36,11 +36,12 @@
 
 OBJCOPY=objcopy
 
-MACHINE=$(shell $(CC) -dumpmachine | sed "s/\(-\).*$$//")
+# MACHINE=$(shell $(CC) -dumpmachine | sed "s/\(-\).*$$//")
+MACHINE=x86_64
 
 ifeq ($(MACHINE),x86_64)
 	ARCH=$(MACHINE)
-	LIBDIR=/usr/lib64
+	LIBDIR=/usr/lib
 	FORMAT=efi-app-x86-64
 else
 	ARCH=ia32
@@ -48,10 +49,11 @@ else
 	FORMAT=efi-app-$(ARCH)
 endif
 
-CRT0=$(LIBDIR)/gnuefi/crt0-efi-$(ARCH).o
-LDSCRIPT=$(LIBDIR)/gnuefi/elf_$(ARCH)_efi.lds
+CRT0=$(LIBDIR)/crt0-efi-$(ARCH).o
+LDSCRIPT=$(LIBDIR)/elf_$(ARCH)_efi.lds
 
 CFLAGS=-I. -I/usr/include/efi -I/usr/include/efi/$(ARCH) \
+		-I/usr/include/efi/protocol \
 		-DEFI_FUNCTION_WRAPPER -fPIC -fshort-wchar -ffreestanding \
 		-Wall -Ifs/ -Iloaders/ -D$(ARCH) -Werror
 LDFLAGS=-T $(LDSCRIPT) -Bsymbolic -shared -nostdlib -znocombreloc \
