--- eqn2img.c.orig	2013-01-27 05:17:45.000000000 +0100
+++ eqn2img.c	2013-08-07 21:54:27.711072356 +0200
@@ -623,19 +623,19 @@
   };
 
   if(img_name) {
-    fp = EGifOpenFileName(img_name, 0);
+    fp = EGifOpenFileName(img_name, 0, NULL);
     if(!fp)
       return -1;
   }
   else
-    fp = EGifOpenFileHandle(STDOUT_FILENO);
+    fp = EGifOpenFileHandle(STDOUT_FILENO, NULL);
 
   for(i=0; i<256; i++) {
     pal[i].Red = (i*background.red + (255-i)*foreground.red)/255;
     pal[i].Green = (i*background.green + (255-i)*foreground.green)/255;
     pal[i].Blue = (i*background.blue + (255-i)*foreground.blue)/255;
   }
-  color_map = MakeMapObject(256, pal);
+  color_map = GifMakeMapObject(256, pal);
 
   /* EGifSetGifVersion("89a"); this causes segfault (but is really required for transparency, I think) */
   EGifPutScreenDesc(fp, width, height, 256, 255, color_map);
