diff -crB libgpod/gnome-autogen.sh libgpod-fixed/gnome-autogen.sh
*** libgpod/gnome-autogen.sh	2012-05-20 11:12:24.143746500 +0200
--- libgpod-fixed/gnome-autogen.sh	2012-05-20 10:28:57.533924473 +0200
***************
*** 316,328 ****
  
  case $REQUIRED_AUTOMAKE_VERSION in
      1.4*) automake_progs="automake-1.4" ;;
!     1.5*) automake_progs="automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7 automake-1.6 automake-1.5" ;;
!     1.6*) automake_progs="automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7 automake-1.6" ;;
!     1.7*) automake_progs="automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7" ;;
!     1.8*) automake_progs="automake-1.11 automake-1.10 automake-1.9 automake-1.8" ;;
!     1.9*) automake_progs="automake-1.11 automake-1.10 automake-1.9" ;;
!     1.10*) automake_progs="automake-1.11 automake-1.10" ;;
!     1.11*) automake_progs="automake-1.11" ;;
  esac
  version_check automake AUTOMAKE "$automake_progs" $REQUIRED_AUTOMAKE_VERSION \
      "http://ftp.gnu.org/pub/gnu/automake/automake-$REQUIRED_AUTOMAKE_VERSION.tar.gz"
--- 316,329 ----
  
  case $REQUIRED_AUTOMAKE_VERSION in
      1.4*) automake_progs="automake-1.4" ;;
!     1.5*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7 automake-1.6 automake-1.5" ;;
!     1.6*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7 automake-1.6" ;;
!     1.7*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9 automake-1.8 automake-1.7" ;;
!     1.8*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9 automake-1.8" ;;
!     1.9*) automake_progs="automake-1.12 automake-1.11 automake-1.10 automake-1.9" ;;
!     1.10*) automake_progs="automake-1.12 automake-1.11 automake-1.10" ;;
!     1.11*) automake_progs="automake-1.12 automake-1.11" ;;
!     1.12*) automake_progs="automake-1.12" ;;
  esac
  version_check automake AUTOMAKE "$automake_progs" $REQUIRED_AUTOMAKE_VERSION \
      "http://ftp.gnu.org/pub/gnu/automake/automake-$REQUIRED_AUTOMAKE_VERSION.tar.gz"
diff -crB libgpod/src/db-parse-context.c libgpod-fixed/src/db-parse-context.c
*** libgpod/src/db-parse-context.c	2012-05-20 11:12:24.147079859 +0200
--- libgpod-fixed/src/db-parse-context.c	2012-05-20 11:11:14.009879152 +0200
***************
*** 64,70 ****
  	g_return_if_fail (ctx != NULL);
  
  	if (ctx->mapped_file) {
! 		g_mapped_file_free(ctx->mapped_file);
  	}
  
  	g_free (ctx);
--- 64,70 ----
  	g_return_if_fail (ctx != NULL);
  
  	if (ctx->mapped_file) {
! 		g_mapped_file_unref(ctx->mapped_file);
  	}
  
  	g_free (ctx);
***************
*** 215,221 ****
  					device->byte_order);
  
  	if (ctx == NULL) {
! 		g_mapped_file_free(mapped_file);
  		return NULL;
  	}
  	ctx->db = db;
--- 215,221 ----
  					device->byte_order);
  
  	if (ctx == NULL) {
! 		g_mapped_file_unref(mapped_file);
  		return NULL;
  	}
  	ctx->db = db;

diff -crB libgpod-orig/src/itdb_artwork.c libgpod-fixed/src/itdb_artwork.c
*** libgpod-orig/src/itdb_artwork.c	2012-05-20 11:14:29.588034945 +0200
--- libgpod-fixed/src/itdb_artwork.c	2012-05-20 11:50:28.774479577 +0200
***************
*** 882,889 ****
  					       pad_x, pad_y,
  					       width, height);
  	pixbuf = gdk_pixbuf_copy (pixbuf_sub);
! 	gdk_pixbuf_unref (pixbuf_full);
! 	gdk_pixbuf_unref (pixbuf_sub);
  
          return pixbuf;
  }
--- 882,889 ----
  					       pad_x, pad_y,
  					       width, height);
  	pixbuf = gdk_pixbuf_copy (pixbuf_sub);
! 	g_object_unref (pixbuf_full);
! 	g_object_unref (pixbuf_sub);
  
          return pixbuf;
  }
diff -crB libgpod-orig/tests/get-timezone.c libgpod-fixed/tests/get-timezone.c
*** libgpod-orig/tests/get-timezone.c	2012-05-20 11:14:29.571368152 +0200
--- libgpod-fixed/tests/get-timezone.c	2012-05-20 11:50:30.961162891 +0200
***************
*** 28,34 ****
      if (argc >= 2) {
          mountpoint = argv[1];
      } else {
!         g_print ("Usage: %s <mountpoint>\n\n", g_basename(argv[0]));
          return -1;
      }
  
--- 28,34 ----
      if (argc >= 2) {
          mountpoint = argv[1];
      } else {
!         g_print ("Usage: %s <mountpoint>\n\n", g_path_get_basename(argv[0]));
          return -1;
      }
  
diff -crB libgpod-orig/tests/itdb_main.c libgpod-fixed/tests/itdb_main.c
*** libgpod-orig/tests/itdb_main.c	2012-05-20 11:14:29.574701511 +0200
--- libgpod-fixed/tests/itdb_main.c	2012-05-20 11:50:28.787813013 +0200
***************
*** 55,61 ****
  
    if (infile == 0)
    {
!       printf ("Usage: %s <infile> [<outfile>]\n",  g_basename(argv[0]));
        exit (0);
    }
  
--- 55,61 ----
  
    if (infile == 0)
    {
!       printf ("Usage: %s <infile> [<outfile>]\n",  g_path_get_basename(argv[0]));
        exit (0);
    }
  
diff -crB libgpod-orig/tests/test-covers.c libgpod-fixed/tests/test-covers.c
*** libgpod-orig/tests/test-covers.c	2012-05-20 11:14:29.571368152 +0200
--- libgpod-fixed/tests/test-covers.c	2012-05-20 11:50:28.794479730 +0200
***************
*** 53,59 ****
          }
          g_print ("  %s\n", filename);
          gdk_pixbuf_save (pixbuf, filename, "png", NULL, NULL);
!         gdk_pixbuf_unref (pixbuf);
          /*		g_print ("Saved %s\n", filename); */
          g_free (filename);
  }
--- 53,59 ----
          }
          g_print ("  %s\n", filename);
          gdk_pixbuf_save (pixbuf, filename, "png", NULL, NULL);
!         g_object_unref (pixbuf);
          /*		g_print ("Saved %s\n", filename); */
          g_free (filename);
  }
diff -crB libgpod-orig/tests/test-fw-id.c libgpod-fixed/tests/test-fw-id.c
*** libgpod-orig/tests/test-fw-id.c	2012-05-20 11:14:29.571368152 +0200
--- libgpod-fixed/tests/test-fw-id.c	2012-05-20 11:50:28.794479730 +0200
***************
*** 42,48 ****
      char *fwid;
  
      if (argc < 2) {
!         g_print ("Usage: %s <mountpoint>\n", g_basename (argv[0]));
          return 1;
  
      }
--- 42,48 ----
      char *fwid;
  
      if (argc < 2) {
!         g_print ("Usage: %s <mountpoint>\n", g_path_get_basename (argv[0]));
          return 1;
  
      }
diff -crB libgpod-orig/tests/test-ls.c libgpod-fixed/tests/test-ls.c
*** libgpod-orig/tests/test-ls.c	2012-05-20 11:14:29.571368152 +0200
--- libgpod-fixed/tests/test-ls.c	2012-05-20 11:50:28.791146372 +0200
***************
*** 121,127 ****
        g_print ("Usage: %s <mountpoint>|-l [<playlistname>]\n\n"
                 "-l - List from the local repository (~" LOCALDB ")\n"
  	       "<playlistname> - name of the playlist to list (optional)\n",  
!                 g_basename(argv[0]));
        exit (0);
    }
  
--- 121,127 ----
        g_print ("Usage: %s <mountpoint>|-l [<playlistname>]\n\n"
                 "-l - List from the local repository (~" LOCALDB ")\n"
  	       "<playlistname> - name of the playlist to list (optional)\n",  
!                 g_path_get_basename(argv[0]));
        exit (0);
    }
  
diff -crB libgpod-orig/tests/test-photos.c libgpod-fixed/tests/test-photos.c
*** libgpod-orig/tests/test-photos.c	2012-05-20 11:14:29.571368152 +0200
--- libgpod-fixed/tests/test-photos.c	2012-05-20 11:50:28.797813088 +0200
***************
*** 81,87 ****
  		path = g_build_filename (dir, filename, NULL);
  		g_free (filename);
  		gdk_pixbuf_save (pixbuf, path, "png", NULL, NULL);
! 		gdk_pixbuf_unref (pixbuf);
  		g_free (path);
  	}
          g_list_free (thumbnails);
--- 81,87 ----
  		path = g_build_filename (dir, filename, NULL);
  		g_free (filename);
  		gdk_pixbuf_save (pixbuf, path, "png", NULL, NULL);
! 		g_object_unref (pixbuf);
  		g_free (path);
  	}
          g_list_free (thumbnails);
diff -crB libgpod-orig/tools/ipod-time-sync.c libgpod-fixed/tools/ipod-time-sync.c
*** libgpod-orig/tools/ipod-time-sync.c	2012-05-20 11:14:29.578034870 +0200
--- libgpod-fixed/tools/ipod-time-sync.c	2012-05-20 11:50:28.784479654 +0200
***************
*** 34,40 ****
  main (int argc, char **argv)
  {
      if (argc < 2) {
!         g_print (_("usage: %s <device> [timezone]\n"), g_basename (argv[0]));
  	return 1;
      }
  
--- 34,40 ----
  main (int argc, char **argv)
  {
      if (argc < 2) {
!         g_print (_("usage: %s <device> [timezone]\n"), g_path_get_basename (argv[0]));
  	return 1;
      }
  
diff -crB libgpod-orig/tools/read-sysinfoextended.c libgpod-fixed/tools/read-sysinfoextended.c
*** libgpod-orig/tools/read-sysinfoextended.c	2012-05-20 11:14:29.578034870 +0200
--- libgpod-fixed/tools/read-sysinfoextended.c	2012-05-20 11:50:28.781146295 +0200
***************
*** 78,84 ****
      char *xml;
      
      if (argc < 3) {
!       g_print (_("usage: %s <device|uuid|bus device> <mountpoint>\n"), g_basename (argv[0]));
  	return 1;
      }
  
--- 78,84 ----
      char *xml;
      
      if (argc < 3) {
!       g_print (_("usage: %s <device|uuid|bus device> <mountpoint>\n"), g_path_get_basename (argv[0]));
  	return 1;
      }
  
