--- ./bfd/cpu-pic30.c.~1~	2007-02-02 22:49:02.000000000 +0100
+++ ./bfd/cpu-pic30.c	2009-01-28 07:25:02.000000000 +0100
@@ -308,8 +308,16 @@
   }
   rib = read_rib(pic30_resource_file);
   if (rib == 0) {
+#if 0
     fprintf(stderr,"Could not open resource file: %s\n", pic30_resource_file);
     err_return = 1;
+#else
+    /* fake missing resource file */
+    pic30_resource_version = xmalloc(strlen(version_part1) + 
+                                     strlen(version_part2) + 40);
+    sprintf(pic30_resource_version,"%s %s", version_part1, version_part2);
+    close_rib();
+#endif
     return;
   }
   if (strcmp(rib->tool_chain,"C30")) {
