diff -ur ../pic30-binutils-v3_25.orig/acme/bfd/config.bfd ./bfd/config.bfd
--- ../pic30-binutils-v3_25.orig/acme/bfd/config.bfd	2011-01-22 20:22:06.000000000 +0100
+++ ./bfd/config.bfd	2011-01-22 19:38:42.000000000 +0100
@@ -859,13 +859,13 @@
     targ_underscore=yes
     ;;
 
-  pic30-*-coff)
-    targ_defvec=pic30_coff_vec
+  pic30-*-elf)
+    targ_defvec=pic30_elf32_vec
     targ_underscore=yes
     ;;
 
-  pic30-*-elf)
-    targ_defvec=pic30_elf32_vec
+  pic30-*-coff | pic30-*-*)
+    targ_defvec=pic30_coff_vec
     targ_underscore=yes
     ;;
 
diff -ur ../pic30-binutils-v3_25.orig/acme/binutils/readelf.c ./binutils/readelf.c
--- ../pic30-binutils-v3_25.orig/acme/binutils/readelf.c	2011-01-22 20:22:07.000000000 +0100
+++ ./binutils/readelf.c	2011-01-22 20:21:45.000000000 +0100
@@ -22,6 +22,7 @@
    02111-1307, USA.  */
 
 
+#include "../mchp_version.h"
 #include <assert.h>
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ur ../pic30-binutils-v3_25.orig/acme/gas/configure ./gas/configure
--- ../pic30-binutils-v3_25.orig/acme/gas/configure	2011-01-22 20:22:07.000000000 +0100
+++ ./gas/configure	2011-01-22 20:14:31.000000000 +0100
@@ -2597,7 +2597,7 @@
       or32-*-rtems*)			fmt=coff ;;
       or32-*-coff)			fmt=coff ;;
       or32-*-elf)			fmt=elf ;;
-      pic30-*-coff)                     fmt=coff need_libm=yes;;
+      pic30-*-coff | pic30-unknown-none)                     fmt=coff need_libm=yes;;
       pic30-*-elf)                      fmt=elf need_libm=yes;;
       pj*)				fmt=elf ;;
 
diff -ur ../pic30-binutils-v3_25.orig/acme/gas/configure.in ./gas/configure.in
--- ../pic30-binutils-v3_25.orig/acme/gas/configure.in	2011-01-22 20:22:07.000000000 +0100
+++ ./gas/configure.in	2011-01-22 20:07:56.000000000 +0100
@@ -386,7 +386,7 @@
       or32-*-rtems*)			fmt=coff ;;
       or32-*-coff)			fmt=coff ;;
       or32-*-elf)			fmt=elf ;;
-      pic30-*-coff)                     fmt=coff need_libm=yes;;
+      pic30-*-coff | pic30-unknown-none)                     fmt=coff need_libm=yes;;
       pic30-*-elf)                      fmt=elf need_libm=yes;;
       pj*)				fmt=elf ;;
 
diff -ur ../pic30-binutils-v3_25.orig/acme/libiberty/Makefile.in ./libiberty/Makefile.in
--- ../pic30-binutils-v3_25.orig/acme//ld/configure.tgt	2011-01-22 20:22:08.000000000 +0100
+++ ./ld/configure.tgt	2011-01-22 19:38:45.000000000 +0100
@@ -542,8 +542,8 @@
 			  tdir_elf64_s390=`echo ${targ_alias} | sed -e 's/s390/s390x/'`
 			fi ;;
 *-*-ieee*)		targ_emul=vanilla ;;
-pic30-*-coff)           targ_emul=pic30_coff ;;
 pic30-*-elf)            targ_emul=pic30_elf32 ;;
+pic30-*-*)           targ_emul=pic30_coff ;;
 ip2k-*-elf)		targ_emul=elf32ip2k ;;
 
 *)
