--- ./bfd/cpu-pic30.c.~1~	2007-02-02 22:49:02.000000000 +0100
+++ ./bfd/cpu-pic30.c	2009-01-28 08:24:37.000000000 +0100
@@ -21,6 +21,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
+#include "../mchp_version.h"
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
--- ./binutils/readelf.c.~1~	2006-07-11 19:20:50.000000000 +0200
+++ ./binutils/readelf.c	2009-01-28 08:25:49.000000000 +0100
@@ -22,6 +22,7 @@
    02111-1307, USA.  */
 
 
+#include "../mchp_version.h"
 #include <assert.h>
 #include <sys/types.h>
 #include <sys/stat.h>
