diff -crB starspan-1.2.03/src/csv/Csv.h patched/src/csv/Csv.h
*** starspan-1.2.03/src/csv/Csv.h	2007-12-14 08:43:01.000000000 +0100
--- patched/src/csv/Csv.h	2010-09-13 18:29:55.191874268 +0200
***************
*** 10,15 ****
--- 10,16 ----
  #include <iostream>
  #include <string>
  #include <vector>
+ #include <cstdio>
  
  using namespace std;
  
diff -crB starspan-1.2.03/src/stats/Stats.cc patched/src/stats/Stats.cc
*** starspan-1.2.03/src/stats/Stats.cc	2006-03-24 18:10:49.000000000 +0100
--- patched/src/stats/Stats.cc	2010-09-13 18:30:25.909767734 +0200
***************
*** 6,11 ****
--- 6,12 ----
  
  #include "Stats.h"
  
+ #include <algorithm>
  #include <string>
  #include <map>
  #include <cstdio>
