diff -aur ratox-0.1.pristine/ratox.c ratox-0.1/ratox.c
--- ratox-0.1.pristine/ratox.c	2014-10-14 17:17:19.542161227 +0200
+++ ratox-0.1/ratox.c	2014-10-14 17:18:36.818526515 +0200
@@ -694,7 +694,7 @@
 	if (read(fd, data, sz) != sz)
 		eprintf("read %s:", DATAFILE);
 
-	if (tox_is_data_encrypted(data) == 1) {
+	if (tox_is_save_encrypted(data) == 1) {
 		if (encryptdatafile == 0)
 			printout("Data : %s > Encrypted, but saving unencrypted\n", DATAFILE);
 		while (readpass("Data : Passphrase > ") < 0 ||
