--- cmake_modules/libpocoConfig.cmake	2011-04-25 09:22:20.401611285 +0200
+++ cmake_modules/libpocoConfig.cmake	2011-04-25 09:22:23.358129212 +0200
@@ -10,11 +10,11 @@
 
 find_library(POCO_DATA NAMES PocoData)
 
-find_library(POCO_MYSQL NAMES PocoMySQL)
+find_library(POCO_MYSQL NAMES PocoDataMySQL)
 
-find_library(POCO_SQLITE NAMES PocoSQLite)
+find_library(POCO_SQLITE NAMES PocoDataSQLite)
 
-find_library(POCO_ODBC NAMES PocoODBC)
+find_library(POCO_ODBC NAMES PocoDataODBC)
 
 if(LIBPOCO_INCLUDE_DIR AND POCO_DATA AND POCO_FOUNDATION)
 	set(LIBPOCO_LIBRARIES ${POCO_DATA} ${POCO_FOUNDATION})
