diff -ru ifplugd-0.28.orig/src/nlapi.c ifplugd-0.28/src/nlapi.c
--- ifplugd-0.28.orig/src/nlapi.c	2005-01-05 01:25:31.000000000 +0100
+++ ifplugd-0.28/src/nlapi.c	2014-05-03 02:31:15.782219000 +0200
@@ -71,7 +71,7 @@
     
     for (;;) {
         int bytes;
-        char replybuf[1024];
+        char replybuf[4096];
         struct nlmsghdr *p = (struct nlmsghdr *) replybuf;
         
         if ((bytes = recv(nlapi_fd, &replybuf, sizeof(replybuf), block ? 0 : MSG_DONTWAIT)) < 0) {
