commit 7d6e5c7c891bffd2b4e0608c6feb6ca0eee2fad4
Author: Benjamin Chrétien <chretien@lirmm.fr>
Date:   Mon Mar 3 12:45:54 2014 +0100

    Fix Python 2 call in hook.

diff --git a/env-hooks/10.ros.sh.em b/env-hooks/10.ros.sh.em
index 316ebac..7d8b2d9 100644
--- a/env-hooks/10.ros.sh.em
+++ b/env-hooks/10.ros.sh.em
@@ -1,7 +1,7 @@
 # generated from ros/env-hooks/10.ros.sh.em
 
 # scrub old ROS bin dirs, to avoid accidentally finding the wrong executables
-export PATH="`python -c \"import os; print(os.pathsep.join([x for x in \\\"$PATH\\\".split(os.pathsep) if not any([d for d in ['cturtle', 'diamondback', 'electric', 'fuerte'] if d in x])]))\"`"
+export PATH="`/usr/bin/env python2 -c \"import os; print(os.pathsep.join([x for x in \\\"$PATH\\\".split(os.pathsep) if not any([d for d in ['cturtle', 'diamondback', 'electric', 'fuerte'] if d in x])]))\"`"
 
 if [ -n "$ROS_DISTRO" -a "$ROS_DISTRO" != "hydro" ]; then
   echo "ROS_DISTRO was set to '$ROS_DISTRO' before. Please make sure that the environment does not mix paths from different distributions."
