--- a/configure
+++ b/configure
@@ -713,7 +713,6 @@ CFLAGS
 CC
 READLINK
 RPMBUILD
-DOXYGEN
 SSH_PROG
 UNAME
 XSLT_CONFIG
@@ -3174,86 +3173,6 @@ fi
 	fi
 fi
 
-# Check for doxygen
-# Extract the first word of "doxygen", so it can be a program name with args.
-set dummy doxygen; ac_word=$2
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
-$as_echo_n "checking for $ac_word... " >&6; }
-if ${ac_cv_prog_DOXYGEN+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  if test -n "$DOXYGEN"; then
-  ac_cv_prog_DOXYGEN="$DOXYGEN" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-    for ac_exec_ext in '' $ac_executable_extensions; do
-  if as_fn_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_DOXYGEN="doxygen"
-    $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-  done
-IFS=$as_save_IFS
-
-  test -z "$ac_cv_prog_DOXYGEN" && ac_cv_prog_DOXYGEN="no"
-fi
-fi
-DOXYGEN=$ac_cv_prog_DOXYGEN
-if test -n "$DOXYGEN"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $DOXYGEN" >&5
-$as_echo "$DOXYGEN" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-
-
-
-# Check for rpmbuild
-# Extract the first word of "rpmbuild", so it can be a program name with args.
-set dummy rpmbuild; ac_word=$2
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
-$as_echo_n "checking for $ac_word... " >&6; }
-if ${ac_cv_prog_RPMBUILD+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  if test -n "$RPMBUILD"; then
-  ac_cv_prog_RPMBUILD="$RPMBUILD" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-    for ac_exec_ext in '' $ac_executable_extensions; do
-  if as_fn_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_RPMBUILD="rpmbuild"
-    $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-  done
-IFS=$as_save_IFS
-
-  test -z "$ac_cv_prog_RPMBUILD" && ac_cv_prog_RPMBUILD="no"
-fi
-fi
-RPMBUILD=$ac_cv_prog_RPMBUILD
-if test -n "$RPMBUILD"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $RPMBUILD" >&5
-$as_echo "$RPMBUILD" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-
-
-
 # Check whether readlink supports the -f option
 READLINK_FKNOWN=`$readlink -f xxx 1>/dev/null 2>&1 && echo yes`
 if test "x${READLINK_FKNOWN}" != "xyes";
@@ -13601,18 +13520,7 @@ fi
 
 
 
-if test -z "$DOXYGEN"; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Due to missing doxygen you will not able to generate Doxygen documentation." >&5
-$as_echo "$as_me: WARNING: Due to missing doxygen you will not able to generate Doxygen documentation." >&2;}
-fi
-
-
-if test -z "$RPMBUILD"; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Due to missing rpmbuild you will not able to generate RPM package." >&5
-$as_echo "$as_me: WARNING: Due to missing rpmbuild you will not able to generate RPM package." >&2;}
-fi
-
-ac_config_files="$ac_config_files Makefile src/config.h libnetconf.spec libnetconf.pc Doxyfile"
+ac_config_files="$ac_config_files Makefile src/config.h libnetconf.spec libnetconf.pc"
 
 ac_config_files="$ac_config_files headers/libnetconf.h headers/libnetconf_xml.h headers/libnetconf_ssh.h"
 
@@ -14614,7 +14522,6 @@ do
     "src/config.h") CONFIG_FILES="$CONFIG_FILES src/config.h" ;;
     "libnetconf.spec") CONFIG_FILES="$CONFIG_FILES libnetconf.spec" ;;
     "libnetconf.pc") CONFIG_FILES="$CONFIG_FILES libnetconf.pc" ;;
-    "Doxyfile") CONFIG_FILES="$CONFIG_FILES Doxyfile" ;;
     "headers/libnetconf.h") CONFIG_FILES="$CONFIG_FILES headers/libnetconf.h" ;;
     "headers/libnetconf_xml.h") CONFIG_FILES="$CONFIG_FILES headers/libnetconf_xml.h" ;;
     "headers/libnetconf_ssh.h") CONFIG_FILES="$CONFIG_FILES headers/libnetconf_ssh.h" ;;
--- a/Makefile.in
+++ b/Makefile.in
@@ -245,7 +245,7 @@ rpm: tarball
 	fi
 
 .PHONY: install
-install: all install-devel install-doc
+install: all install-devel
 	$(INSTALL) -m 775 -d $(DESTDIR)/$(libdir)
 	$(LIBTOOL) --mode=install cp $(NAME).la $(DESTDIR)/$(libdir)
 	$(LIBTOOL) --mode=finish $(DESTDIR)/$(libdir)
