=== modified file 'Makefile.am'
--- Makefile.am	2010-01-19 22:02:50 +0000
+++ Makefile.am	2010-10-12 17:00:58 +0000
@@ -16,4 +16,4 @@
 DISTCLEANFILES = libido.pc
 EXTRA_DIST = libido.pc.in
 
-DISTCHECK_CONFIGURE_FLAGS = --disable-gtk-doc
+DISTCHECK_CONFIGURE_FLAGS = --disable-gtk-doc --enable-deprecations

=== modified file 'configure.ac'
--- configure.ac	2010-09-14 00:31:26 +0000
+++ configure.ac	2010-10-12 17:00:58 +0000
@@ -76,6 +76,16 @@
 AC_SUBST(GEIS_CFLAGS)
 AC_SUBST(GEIS_LIBS)
 
+dnl ==========================================================================
+
+AC_ARG_ENABLE([deprecations],
+  [AS_HELP_STRING([--enable-deprecations],
+    [allow deprecated API usage @<:@default=yes@:>@])],
+     [],
+     [enable_deprecations=yes])
+AS_IF([test "x$enable_deprecations" = xno],
+  [CFLAGS="$CFLAGS -DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED -DGSEAL_ENABLE -DGTK_DISABLE_SINGLE_INCLUDES"])
+
 dnl ===========================================================================
 
 if test "x$GCC" = "xyes"; then

=== modified file 'example/gesture.c'
--- example/gesture.c	2010-09-14 21:40:31 +0000
+++ example/gesture.c	2010-10-12 17:00:58 +0000
@@ -18,7 +18,7 @@
   gdouble radians;
   gint width = (in_touch > 0) ? 10 : 1;
 
-  cr = gdk_cairo_create (widget->window);
+  cr = gdk_cairo_create (gtk_widget_get_window (widget));
 
   cairo_set_source_rgb (cr, 0, 0, 0);
   cairo_set_line_width (cr, width);

=== modified file 'src/idocalendarmenuitem.c'
--- src/idocalendarmenuitem.c	2010-07-29 21:07:05 +0000
+++ src/idocalendarmenuitem.c	2010-10-12 17:00:58 +0000
@@ -124,7 +124,7 @@
     gtk_widget_grab_focus (widget);
 
   event->focus_change.type = GDK_FOCUS_CHANGE;
-  event->focus_change.window = g_object_ref (widget->window);
+  event->focus_change.window = g_object_ref (gtk_widget_get_window (widget));
   event->focus_change.in = in;
 
   gtk_widget_event (widget, event);
@@ -145,13 +145,14 @@
   if (menuitem->priv->selected)
     {
       GtkWidget *calendar = menuitem->priv->calendar;
+      GdkWindow *window = gtk_widget_get_window (calendar);
 
       gtk_widget_event (calendar,
                         ((GdkEvent *)(void*)(event)));
 
-      if (calendar->window != NULL)
+      if (window != NULL)
         {
-          gdk_window_raise (calendar->window);
+          gdk_window_raise (window);
         }
 
       if (!gtk_widget_has_focus (calendar))
@@ -170,12 +171,13 @@
                                      GdkEventButton *event)
 {
   GtkWidget *calendar = IDO_CALENDAR_MENU_ITEM (widget)->priv->calendar;
+  GdkWindow *window = gtk_widget_get_window (calendar);
 
   if (event->button == 1)
     {
-      if (calendar->window != NULL)
+      if (window != NULL)
         {
-          gdk_window_raise (calendar->window);
+          gdk_window_raise (window);
         }
 
       if (!gtk_widget_has_focus (calendar))
@@ -225,12 +227,14 @@
 calendar_realized_cb (GtkWidget        *widget,
                       IdoCalendarMenuItem *item)
 {
-  if (widget->window != NULL)
+  GdkWindow *window = gtk_widget_get_window (widget);
+
+  if (window != NULL)
     {
-      gdk_window_raise (widget->window);
+      gdk_window_raise (window);
     }
 
-  g_signal_connect (GTK_WIDGET (item)->parent,
+  g_signal_connect (gtk_widget_get_parent (GTK_WIDGET (item)),
                     "key-press-event",
                     G_CALLBACK (ido_calendar_menu_item_key_press),
                     item);

=== modified file 'src/idoentrymenuitem.c'
--- src/idoentrymenuitem.c	2010-09-10 21:26:16 +0000
+++ src/idoentrymenuitem.c	2010-10-12 17:00:58 +0000
@@ -164,7 +164,7 @@
   g_object_ref (widget);
 
   event->focus_change.type = GDK_FOCUS_CHANGE;
-  event->focus_change.window = g_object_ref (widget->window);
+  event->focus_change.window = g_object_ref (gtk_widget_get_window (widget));
   event->focus_change.in = in;
 
   gtk_widget_event (widget, event);
@@ -180,12 +180,13 @@
                                   GdkEventButton *event)
 {
   GtkWidget *entry = IDO_ENTRY_MENU_ITEM (widget)->priv->entry;
+  GdkWindow *window = gtk_widget_get_window (entry);
 
   if (event->button == 1)
     {
-      if (entry->window != NULL)
+      if (window != NULL)
         {
-          gdk_window_raise (entry->window);
+          gdk_window_raise (window);
         }
 
       if (!gtk_widget_has_focus (entry))
@@ -235,12 +236,14 @@
 entry_realized_cb (GtkWidget        *widget,
                    IdoEntryMenuItem *item)
 {
-  if (widget->window != NULL)
+  GdkWindow *window = gtk_widget_get_window (widget);
+
+  if (window != NULL)
     {
-      gdk_window_raise (widget->window);
+      gdk_window_raise (window);
     }
 
-  g_signal_connect (GTK_WIDGET (item)->parent,
+  g_signal_connect (gtk_widget_get_parent (GTK_WIDGET (item)),
                     "key-press-event",
                     G_CALLBACK (ido_entry_menu_item_key_press),
                     item);

=== modified file 'src/idogesturemanager.c'
--- src/idogesturemanager.c	2010-09-24 14:46:19 +0000
+++ src/idogesturemanager.c	2010-10-12 17:00:58 +0000
@@ -668,7 +668,7 @@
       GeisXcbWinInfo xcb_win_info = {
         .display_name = NULL,
         .screenp      = NULL,
-        .window_id    = GDK_DRAWABLE_XID (GTK_WIDGET (window)->window)
+        .window_id    = GDK_DRAWABLE_XID (gtk_widget_get_window (GTK_WIDGET (window)))
       };
       GeisWinInfo win_info = {
         GEIS_XCB_FULL_WINDOW,

=== modified file 'src/idomessagedialog.c'
--- src/idomessagedialog.c	2010-07-08 18:49:37 +0000
+++ src/idomessagedialog.c	2010-10-12 17:00:58 +0000
@@ -82,7 +82,7 @@
    *
    * See: https://bugs.launchpad.net/ubuntu/+source/compiz/+bug/240794
    */
-  gdk_window_set_functions (widget->window,
+  gdk_window_set_functions (gtk_widget_get_window (widget),
                             GDK_FUNC_RESIZE | GDK_FUNC_MOVE);
 
   ido_message_dialog_get_secondary_label (IDO_MESSAGE_DIALOG (widget));
@@ -166,7 +166,7 @@
       IdoTimeline *timeline;
       IdoMessageDialogMorphContext *context;
 
-      start = GTK_WIDGET (dialog)->requisition;
+      gtk_widget_get_requisition (GTK_WIDGET (dialog), &start);
 
       priv->expanded = TRUE;
 
@@ -209,7 +209,7 @@
   event_box = gtk_event_box_new ();
   gtk_widget_show (event_box);
 
-  vbox = GTK_DIALOG (object)->vbox;
+  vbox = gtk_dialog_get_content_area (GTK_DIALOG (object));
   priv->action_area = gtk_dialog_get_action_area (GTK_DIALOG (object));
 
   g_object_ref (G_OBJECT (vbox));
@@ -296,7 +296,7 @@
       msg = g_strdup_vprintf (message_format, args);
       va_end (args);
 
-      gtk_label_set_text (GTK_LABEL (GTK_MESSAGE_DIALOG (widget)->label),
+      gtk_label_set_text (ido_message_dialog_get_primary_label (IDO_MESSAGE_DIALOG (widget)),
                           msg);
 
       g_free (msg);
@@ -397,7 +397,7 @@
 
                       label = GTK_LABEL (vlist->data);
 
-                      if (strcmp ((primary ? text : secondary_text), label->label) == 0)
+                      if (strcmp ((primary ? text : secondary_text), gtk_label_get_label (label)) == 0)
                         {
                           return GTK_WIDGET (label);
                         }

=== modified file 'src/idomessagedialog.h'
--- src/idomessagedialog.h	2010-07-08 16:11:17 +0000
+++ src/idomessagedialog.h	2010-10-12 17:00:58 +0000
@@ -29,8 +29,6 @@
 #ifndef __IDO_MESSAGE_DIALOG_H__
 #define __IDO_MESSAGE_DIALOG_H__
 
-#include <gtk/gtkmessagedialog.h>
-
 #define IDO_TYPE_MESSAGE_DIALOG         (ido_message_dialog_get_type ())
 #define IDO_MESSAGE_DIALOG(o)           (G_TYPE_CHECK_INSTANCE_CAST ((o), IDO_TYPE_MESSAGE_DIALOG, IdoMessageDialog))
 #define IDO_MESSAGE_DIALOG_CLASS(k)     (G_TYPE_CHECK_CLASS_CAST ((k), IDO_TYPE_MESSAGE_DIALOG, IdoMessageDialogClass))

=== modified file 'src/idoscalemenuitem.c'
--- src/idoscalemenuitem.c	2010-10-07 17:19:20 +0000
+++ src/idoscalemenuitem.c	2010-10-12 17:00:58 +0000
@@ -139,6 +139,8 @@
   GtkRequisition secondary_req;
   gint horizontal_padding;
   gint primary_padding, secondary_padding;
+  guint border_width;
+  GtkStyle * style;
 
   GTK_WIDGET_CLASS (ido_scale_menu_item_parent_class)->size_allocate (widget, allocation);
 
@@ -185,8 +187,11 @@
       priv->right_padding = primary_padding;
     }
 
-  priv->child_allocation.x = GTK_CONTAINER (widget)->border_width + widget->style->xthickness;
-  priv->child_allocation.y = GTK_CONTAINER (widget)->border_width + widget->style->ythickness;
+  border_width = gtk_container_get_border_width (GTK_CONTAINER (widget));
+  style = gtk_widget_get_style (widget);
+
+  priv->child_allocation.x = border_width + style->xthickness;
+  priv->child_allocation.y = border_width + style->ythickness;
 
   priv->child_allocation.x += horizontal_padding;
   priv->child_allocation.x += priv->toggle_size;
@@ -554,7 +559,7 @@
       return TRUE;
     }
 
-  event->window = GTK_RANGE (scale)->event_window;
+  event->window = gtk_widget_get_parent_window (scale);
 
   translate_event_coordinates (menuitem, event->x, &x);
   event->x = x;

=== modified file 'src/idoscalemenuitem.h'
--- src/idoscalemenuitem.h	2010-07-15 13:44:29 +0000
+++ src/idoscalemenuitem.h	2010-10-12 17:00:58 +0000
@@ -26,7 +26,6 @@
 #ifndef __IDO_SCALE_MENU_ITEM_H__
 #define __IDO_SCALE_MENU_ITEM_H__
 
-#include <gtk/gtkmenuitem.h>
 #include "idorange.h"
 
 G_BEGIN_DECLS

=== modified file 'src/idotimeline.c'
--- src/idotimeline.c	2010-07-27 18:02:13 +0000
+++ src/idotimeline.c	2010-10-12 17:00:58 +0000
@@ -22,7 +22,6 @@
 #include "idotimeline.h"
 #include "idotypebuiltins.h"
 
-#include <gtk/gtksettings.h>
 #include <math.h>
 
 #define IDO_TIMELINE_GET_PRIV(obj) (G_TYPE_INSTANCE_GET_PRIVATE ((obj), IDO_TYPE_TIMELINE, IdoTimelinePriv))

=== modified file 'src/idotimeline.h'
--- src/idotimeline.h	2010-07-22 12:21:06 +0000
+++ src/idotimeline.h	2010-10-12 17:00:58 +0000
@@ -21,7 +21,6 @@
 #define __IDO_TIMELINE_H__
 
 #include <glib-object.h>
-#include <gtk/gtkenums.h>
 #include <gdk/gdk.h>
 
 G_BEGIN_DECLS

