--- generate-mozc-ut.sh.orig	2012-12-08 14:35:43.000000000 +0900
+++ generate-mozc-ut.sh	2012-12-08 17:23:53.446647243 +0900
@@ -20,12 +20,8 @@
 rm -f ut-*.txt*
 rm -f */*.txt.r
 
-rm -f chimei/*.CSV
-rm -f chimei/*.zip
 rm -f chimei/chimei.txt
-rm -f chimei/gen_zip_code_seed.py
 
-rm -rf ../mozc-$MOZCVER/
 rm -rf ../mozc-ut-$MOZCVER.$DICVER/
 
 
@@ -34,18 +30,11 @@
 # generate the mozc-ut dictionary
 ##################################################
 
-# unpack official mozc
-echo "unpack official mozc..."
-
-cd ..
-tar jxf mozc-$MOZCVER.tar.bz2
-cd -
-
 # get official mozcdic
 cat ../mozc-$MOZCVER/data/dictionary_oss/dictionary*.txt > mozcdic_all.txt
 
 # get mozcdic costlist
-ruby 32-* mozcdic_all.txt
+ruby 32-* mozcdic_all.txt || exit 1
 mv mozcdic_all.txt.cost costlist
 
 # get hinsi ID
@@ -53,22 +42,10 @@
 
 
 
-# generate zip code dic
-echo "generate zip code dic..."
-cd chimei/
-wget http://www.post.japanpost.jp/zipcode/dl/kogaki/zip/ken_all.zip
-wget http://www.post.japanpost.jp/zipcode/dl/jigyosyo/zip/jigyosyo.zip
-unzip ken_all.zip
-unzip jigyosyo.zip
-cp ../../mozc-$MOZCVER/dictionary/gen_zip_code_seed.py .
-python gen_zip_code_seed.py --zip_code=KEN_ALL.CSV --jigyosyo=JIGYOSYO.CSV >> \
-../../mozc-$MOZCVER/data/dictionary_oss/dictionary09.txt
-cd -
-
 # generate chimei.txt
 echo "generate chimei.txt..."
 cd chimei/
-ruby get-entries.rb KEN_ALL.CSV
+ruby get-entries.rb ../../KEN_ALL.CSV || exit 1
 cd -
 
 
@@ -77,44 +54,49 @@
 echo "check major ut dictionaries..."
 
 # ruby 12-* dicfile min_hits
-ruby 12-* altcanna/altcanna.txt 300
-ruby 12-* jinmei/jinmei.txt 20
-ruby 12-* ekimei/ekimei.txt 300
-ruby 12-* chimei/chimei.txt 300
+ruby 12-* altcanna/altcanna.txt 300 || exit 1
+ruby 12-* jinmei/jinmei.txt 20 || exit 1
+ruby 12-* ekimei/ekimei.txt 300 || exit 1
+ruby 12-* chimei/chimei.txt 300 || exit 1
 
 cat altcanna/altcanna.txt.r jinmei/jinmei.txt.r ekimei/ekimei.txt.r \
-chimei/chimei.txt.r > ut-dic1.txt
+chimei/chimei.txt.r > ut-dic1.txt || exit 1
 
-ruby 44-* mozcdic_all.txt ut-dic1.txt
-ruby 36-* ut-dic1.txt.yomihyouki
+ruby 44-* mozcdic_all.txt ut-dic1.txt || exit 1
+ruby 36-* ut-dic1.txt.yomihyouki || exit 1
 
-cat ut-dic1.txt.yomihyouki.cost mozcdic_all.txt > mozcdic_all.txt.utmajor
+cat ut-dic1.txt.yomihyouki.cost mozcdic_all.txt > mozcdic_all.txt.utmajor || exit 1
 
 
 
 # check minor ut dictionaries
 echo "check minor ut dictionaries..."
 
-ruby 12-* skk/skk.txt 300
-ruby 12-* edict/edict.txt 300
-ruby 12-* hatena/hatena.txt 300
+ruby 12-* skk/skk.txt 300 || exit 1
+[ "$EDICT" = "yes" ] && ruby 12-* edict/edict.txt 300 || exit 1
+ruby 12-* hatena/hatena.txt 300 || exit 1
 
-if [ $NICODIC = "true" ]
+if [ "$NICODIC" = "true" ]
 then
-	ruby 12-* niconico/niconico.txt 300
+	ruby 12-* niconico/niconico.txt 300 || exit 1
 fi
 
-cat skk/skk.txt.r edict/edict.txt.r hatena/hatena.txt.r > ut-dic2.txt
+cat skk/skk.txt.r hatena/hatena.txt.r > ut-dic2.txt || exit 1
+
+if [ "$EDICT" = "yes" ]; then
+    cat edict/edict.txt.r ut-dic2.txt > ut-dic2.txt.new || exit 1
+    mv ut-dic2.txt.new ut-dic2.txt
+fi
 
-if [ $NICODIC = "true" ]
+if [ "$NICODIC" = "true" ]
 then
-	cat niconico/niconico.txt.r ut-dic2.txt > ut-dic2.txt.new
+	cat niconico/niconico.txt.r ut-dic2.txt > ut-dic2.txt.new || exit 1
 	mv ut-dic2.txt.new ut-dic2.txt
 fi
 
-ruby 42-* mozcdic_all.txt.utmajor ut-dic2.txt
-ruby 40-* mozcdic_all.txt.utmajor ut-dic2.txt.yomi
-ruby 36-* ut-dic2.txt.yomi.hyouki
+ruby 42-* mozcdic_all.txt.utmajor ut-dic2.txt || exit 1
+ruby 40-* mozcdic_all.txt.utmajor ut-dic2.txt.yomi || exit 1
+ruby 36-* ut-dic2.txt.yomi.hyouki || exit 1
 
 
 
@@ -122,9 +104,9 @@
 # 「べーとーべん」から「ベートーヴェン」に変換できるようにする
 echo "generate babibubebo from vavivuvevo..."
 
-cat *.cost mozcdic_all.txt > ut-check-va.txt
-ruby 60-* ut-check-va.txt
-ruby 62-* ut-check-va.txt.va
+cat *.cost mozcdic_all.txt > ut-check-va.txt || exit 1
+ruby 60-* ut-check-va.txt || exit 1
+ruby 62-* ut-check-va.txt.va || exit 1
 
 
 
@@ -133,8 +115,8 @@
 ##################################################
 
 # install mozcdic-ut
-cat *.cost *.va.to_ba > dictionary-ut.txt
-cat dictionary-ut.txt ../mozc-$MOZCVER/data/dictionary_oss/dictionary00.txt > dictionary00.txt
+cat *.cost *.va.to_ba > dictionary-ut.txt || exit 1
+cat dictionary-ut.txt ../mozc-$MOZCVER/data/dictionary_oss/dictionary00.txt > dictionary00.txt || exit 1
 mv dictionary00.txt ../mozc-$MOZCVER/data/dictionary_oss/
 
 # change mozc branding
@@ -147,8 +129,10 @@
 cp altcanna/doc/* ../mozc-$MOZCVER/doc-ut/dictionary/altcanna
 mkdir -p ../mozc-$MOZCVER/doc-ut/dictionary/chimei
 cp chimei/doc/* ../mozc-$MOZCVER/doc-ut/dictionary/chimei
-mkdir -p ../mozc-$MOZCVER/doc-ut/dictionary/edict
-cp edict/doc/* ../mozc-$MOZCVER/doc-ut/dictionary/edict
+if [ "$EDICT" = "yes" ]; then
+    mkdir -p ../mozc-$MOZCVER/doc-ut/dictionary/edict
+    cp edict/doc/* ../mozc-$MOZCVER/doc-ut/dictionary/edict
+fi
 mkdir -p ../mozc-$MOZCVER/doc-ut/dictionary/ekimei
 cp ekimei/doc/* ../mozc-$MOZCVER/doc-ut/dictionary/ekimei
 mkdir -p ../mozc-$MOZCVER/doc-ut/dictionary/hatena
@@ -158,7 +142,7 @@
 mkdir -p ../mozc-$MOZCVER/doc-ut/dictionary/skk
 cp skk/doc/* ../mozc-$MOZCVER/doc-ut/dictionary/skk
 
-if [ $NICODIC = "true" ]
+if [ "$NICODIC" = "true" ]
 then
 	mkdir -p ../mozc-$MOZCVER/doc-ut/dictionary/niconico
 	cp niconico/doc/* ../mozc-$MOZCVER/doc-ut/dictionary/niconico
@@ -169,9 +153,6 @@
 cp README ../mozc-$MOZCVER/doc-ut/
 cp ChangeLog ../mozc-$MOZCVER/doc-ut/
 
-# debian/rules
-cp -r debian/ ../mozc-$MOZCVER/
-
 # rename mozc dir to mozc-ut
 cd ..
 mv mozc-$MOZCVER mozc-ut-$MOZCVER.$DICVER
