Index: unix/emacs/mozc.el
===================================================================
--- unix/emacs/mozc.el	(revision 151)
+++ unix/emacs/mozc.el	(working copy)
@@ -356,6 +356,33 @@
                  ('hiragana-katakana 'kana)
                  ('next 'pagedown)
                  ('prior 'pageup)
+                 ('kp-decimal 'decimal)
+                 ('kp-0 'numpad0)
+                 ('kp-1 'numpad1)
+                 ('kp-2 'numpad2)
+                 ('kp-3 'numpad3)
+                 ('kp-4 'numpad4)
+                 ('kp-5 'numpad5)
+                 ('kp-6 'numpad6)
+                 ('kp-7 'numpad7)
+                 ('kp-8 'numpad8)
+                 ('kp-9 'numpad9)
+                 ('kp-delete 'delete)  ; .
+                 ('kp-insert 'insert)  ; 0
+                 ('kp-end 'right)      ; 1
+                 ('kp-down 'down)      ; 2
+                 ('kp-next 'pagedown)  ; 3
+                 ('kp-left 'left)      ; 4
+                 ('kp-begin 'clear)    ; 5
+                 ('kp-right 'right)    ; 6
+                 ('kp-home 'home)      ; 7
+                 ('kp-up 'up)          ; 8
+                 ('kp-prior 'pageup)   ; 9
+                 ('kp-add 'add)
+                 ('kp-subtract 'subtract)
+                 ('kp-multiply 'multiply)
+                 ('kp-divide 'divide)
+                 ('kp-enter 'enter)
                  (t basic-type))))
       (cond
        ;; kana + shift + rest => katakana + rest
Index: data/usage_stats/stats.def
===================================================================
--- data/usage_stats/stats.def	(revision 151)
+++ data/usage_stats/stats.def	(working copy)
@@ -388,6 +388,8 @@
 HANJA
 # Numpad [,]
 COMMA
+# Numpad [5]
+CLEAR
 
 # Stats for config
 # Keymap style
Index: session/key_parser.cc
===================================================================
--- session/key_parser.cc	(revision 151)
+++ session/key_parser.cc	(working copy)
@@ -170,6 +170,7 @@
     keycode_map_["divide"] = KeyEvent::DIVIDE;
     keycode_map_["equals"] = KeyEvent::EQUALS;
     keycode_map_["comma"] = KeyEvent::COMMA;
+    keycode_map_["clear"] = KeyEvent::CLEAR;
   }
 
   SpecialKeysMap keycode_map_;
Index: session/commands.proto
===================================================================
--- session/commands.proto	(revision 151)
+++ session/commands.proto	(working copy)
@@ -131,7 +131,8 @@
     CAPS_LOCK = 68;
     HANJA = 69;
     COMMA = 70;  // Numpad [,]
-    NUM_SPECIALKEYS = 71;
+    CLEAR = 71;  // Numpad [5] without NUMLOCK
+    NUM_SPECIALKEYS = 72;
   };
 
   // CTRL, ALT and SHIFT cover both LEFT and RIGHT keys.  If the
