diff -uNr mozcdic-ut-20110727.orig//20-limit-jinmei.rb mozcdic-ut-20110727/20-limit-jinmei.rb
--- mozcdic-ut-20110727.orig//20-limit-jinmei.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/20-limit-jinmei.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # limit_jinmei
diff -uNr mozcdic-ut-20110727.orig//22-limit-meisi.rb mozcdic-ut-20110727/22-limit-meisi.rb
--- mozcdic-ut-20110727.orig//22-limit-meisi.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/22-limit-meisi.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # limit_meisi
diff -uNr mozcdic-ut-20110727.orig//24-change-vu.rb mozcdic-ut-20110727/24-change-vu.rb
--- mozcdic-ut-20110727.orig//24-change-vu.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/24-change-vu.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # change_vu
diff -uNr mozcdic-ut-20110727.orig//32-get-mozcdic-cost-range.rb mozcdic-ut-20110727/32-get-mozcdic-cost-range.rb
--- mozcdic-ut-20110727.orig//32-get-mozcdic-cost-range.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/32-get-mozcdic-cost-range.rb	2011-07-27 23:15:50.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # get_cost
@@ -21,7 +22,7 @@
 		s = lines[i].chomp.split("	")
 
 		# get Japanese words
-		if s[-1].length == s[-1].split(//u).length * 3
+		if s[-1].bytesize == s[-1].split(//u).length * 3
 			t[p] = s[3].to_i
 			p = p + 1
 		end
diff -uNr mozcdic-ut-20110727.orig//34-change-to-mozcdic-order.rb mozcdic-ut-20110727/34-change-to-mozcdic-order.rb
--- mozcdic-ut-20110727.orig//34-change-to-mozcdic-order.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/34-change-to-mozcdic-order.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # hindo_order
diff -uNr mozcdic-ut-20110727.orig//36-apply-cost-and-hinsi.rb mozcdic-ut-20110727/36-apply-cost-and-hinsi.rb
--- mozcdic-ut-20110727.orig//36-apply-cost-and-hinsi.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/36-apply-cost-and-hinsi.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # set_entries
diff -uNr mozcdic-ut-20110727.orig//38-change-mozc-branding.rb mozcdic-ut-20110727/38-change-mozc-branding.rb
--- mozcdic-ut-20110727.orig//38-change-mozc-branding.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/38-change-mozc-branding.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # change_branding
diff -uNr mozcdic-ut-20110727.orig//40-mozc_to_ut-remove-hyouki-duplicates.rb mozcdic-ut-20110727/40-mozc_to_ut-remove-hyouki-duplicates.rb
--- mozcdic-ut-20110727.orig//40-mozc_to_ut-remove-hyouki-duplicates.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/40-mozc_to_ut-remove-hyouki-duplicates.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # remove duplicates
diff -uNr mozcdic-ut-20110727.orig//42-mozc_to_ut-remove-yomigana-duplicates.rb mozcdic-ut-20110727/42-mozc_to_ut-remove-yomigana-duplicates.rb
--- mozcdic-ut-20110727.orig//42-mozc_to_ut-remove-yomigana-duplicates.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/42-mozc_to_ut-remove-yomigana-duplicates.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # remove duplicates
diff -uNr mozcdic-ut-20110727.orig//44-mozc_to_ut-get-same-hyouki.rb mozcdic-ut-20110727/44-mozc_to_ut-get-same-hyouki.rb
--- mozcdic-ut-20110727.orig//44-mozc_to_ut-get-same-hyouki.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/44-mozc_to_ut-get-same-hyouki.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # remove duplicates
diff -uNr mozcdic-ut-20110727.orig//46-mozc_to_ut-get-different-yomigana.rb mozcdic-ut-20110727/46-mozc_to_ut-get-different-yomigana.rb
--- mozcdic-ut-20110727.orig//46-mozc_to_ut-get-different-yomigana.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/46-mozc_to_ut-get-different-yomigana.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # remove duplicates
diff -uNr mozcdic-ut-20110727.orig//48-mozc_to_ut-generate-betuyomi-entries.rb mozcdic-ut-20110727/48-mozc_to_ut-generate-betuyomi-entries.rb
--- mozcdic-ut-20110727.orig//48-mozc_to_ut-generate-betuyomi-entries.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/48-mozc_to_ut-generate-betuyomi-entries.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # remove duplicates
diff -uNr mozcdic-ut-20110727.orig//68-add-ba-to-va.rb mozcdic-ut-20110727/68-add-ba-to-va.rb
--- mozcdic-ut-20110727.orig//68-add-ba-to-va.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/68-add-ba-to-va.rb	2011-07-27 23:14:24.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # convert_vu
@@ -29,7 +30,7 @@
 
 		if s[0].split(//u).length >= 4
 		if s[0] != yomi
-		if s[-1].length == s[-1].split(//u).length * 3
+		if s[-1].bytesize == s[-1].split(//u).length * 3
 			dicfile.puts s.join("	")
 		end
 		end
diff -uNr mozcdic-ut-20110727.orig//generate-mozc-ut.sh mozcdic-ut-20110727/generate-mozc-ut.sh
--- mozcdic-ut-20110727.orig//generate-mozc-ut.sh	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/generate-mozc-ut.sh	2011-07-27 23:19:31.000000000 +0900
@@ -18,14 +18,8 @@
 rm -f utdic.txt*
 rm -f utall.txt
 
-rm -rf ../mozc-$MOZCVER/
 rm -rf ../mozc-ut-$MOZCVER/
 
-# unpack official mozc
-cd ..
-tar jxf mozc-$MOZCVER.tar.bz2
-cd -
-
 # get mozcdic
 cat ../mozc-$MOZCVER/data/dictionary/dictionary*.txt > mozcdic_all.txt
 
diff -uNr mozcdic-ut-20110727.orig//hindo-mozc-orders.rb mozcdic-ut-20110727/hindo-mozc-orders.rb
--- mozcdic-ut-20110727.orig//hindo-mozc-orders.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/hindo-mozc-orders.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # hindo_order
diff -uNr mozcdic-ut-20110727.orig//hindo-ut-orders.rb mozcdic-ut-20110727/hindo-ut-orders.rb
--- mozcdic-ut-20110727.orig//hindo-ut-orders.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/hindo-ut-orders.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # hindo_order
diff -uNr mozcdic-ut-20110727.orig//remove-duplicates.rb mozcdic-ut-20110727/remove-duplicates.rb
--- mozcdic-ut-20110727.orig//remove-duplicates.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/remove-duplicates.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # remove_duplicates
diff -uNr mozcdic-ut-20110727.orig//reverse-sortdic.rb mozcdic-ut-20110727/reverse-sortdic.rb
--- mozcdic-ut-20110727.orig//reverse-sortdic.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/reverse-sortdic.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # sort lines
diff -uNr mozcdic-ut-20110727.orig//sortdic.rb mozcdic-ut-20110727/sortdic.rb
--- mozcdic-ut-20110727.orig//sortdic.rb	2011-07-27 01:02:46.000000000 +0900
+++ mozcdic-ut-20110727/sortdic.rb	2011-07-27 23:12:29.000000000 +0900
@@ -1,4 +1,5 @@
 #!/usr/bin/ruby
+# -*- encoding: utf-8 -*-
 
 #############################
 # sort lines
