Index: src/system.h
===================================================================
--- src.orig/src/system.h
+++ src/src/system.h
@@ -137,7 +137,22 @@ enum
 /* Redirection and wildcarding when done by the utility itself.
    Generally a noop, but used in particular for native VMS. */
 #ifndef initialize_main
-# define initialize_main(ac, av)
+# define initialize_main(ac, av)                                                        \
+  do {                                                                                  \
+  fprintf (stderr,                                                                      \
+           "Academic tradition requires you to cite works you base your article on.\n"  \
+           "When using programs that use GNU %s to process data for publication\n"      \
+           "please cite:\n"                                                             \
+           "\n",                                                                        \
+           PROGRAM_NAME);                                                               \
+  version_etc (stderr, (const char *) NULL, PROGRAM_NAME, VERSION, AUTHORS,             \
+               (const char *) NULL);                                                    \
+  fprintf (stderr,                                                                      \
+           "\n"                                                                         \
+           "This helps funding further development; and it won't cost you a cent.\n"    \
+           "If you pay 10000 EUR you should feel free to use GNU %s without citing.\n", \
+           PROGRAM_NAME);                                                               \
+  } while (0)
 #endif
 
 #include "stat-macros.h"
