=== modified file 'mnemosyne/mnemosyne/pyqt_ui/makefile'
--- mnemosyne/mnemosyne/pyqt_ui/makefile	2011-12-23 18:52:03 +0000
+++ mnemosyne/mnemosyne/pyqt_ui/makefile	2012-02-12 16:01:24 +0000
@@ -15,13 +15,13 @@
 	pyrcc4 mnemosyne.qrc -o mnemosyne_rc.py
 
 ui_review_wdgt.py: review_wdgt.ui
-	./pyuic4 $< > tmp
+	python2 ./pyuic4 $< > tmp
 	sed -f fix_review_widget_ui.sed <  tmp > $@
 
 $(WIDGETS): ui_%.py: %.ui
-	./pyuic4 $< > $@
+	python2 ./pyuic4 $< > $@
 
 clean:
 	rm -f *~ *.pyc
 	rm -f $(WIDGETS) ui_review_wdgt.py
-	rm -f mnemosyne_rc.py
\ No newline at end of file
+	rm -f mnemosyne_rc.py

=== modified file 'mnemosyne/po/makefile'
--- mnemosyne/po/makefile	2011-10-13 10:24:00 +0000
+++ mnemosyne/po/makefile	2012-02-12 15:39:57 +0000
@@ -4,7 +4,7 @@
 
 $(LANGS): ../mo/%/LC_MESSAGES/mnemosyne.mo: %.po
 	mkdir -p `dirname $@`
-	./msgfmt.py -o $@ $<
+	python2 ./msgfmt.py -o $@ $<
 
 clean:
 	rm -rf *~

