diff -Naur ../plotmmNew/plotmm/curve.h plotmm-0.1.2/plotmm/curve.h
--- ../plotmmNew/plotmm/curve.h	2007-11-22 17:20:25.000000000 +0100
+++ plotmm-0.1.2/plotmm/curve.h	2005-11-09 17:47:21.000000000 +0100
@@ -140,11 +140,7 @@
 			  int from = 0, int to = -1);
 
 	//! Signals that this curve has changed in some way or the other
-	#if (GTKMM_MAJOR_VERSION == 2 && GTKMM_MINOR_VERSION >= 4)
-	sigc::signal0<void> signal_curve_changed;
-	#else
 	SigC::Signal0<void> signal_curve_changed;
-	#endif
 
     protected:
 	virtual void init(const Glib::ustring &title);
diff -Naur ../plotmmNew/plotmm/plot.cc plotmm-0.1.2/plotmm/plot.cc
--- ../plotmmNew/plotmm/plot.cc	2007-11-22 17:20:25.000000000 +0100
+++ plotmm-0.1.2/plotmm/plot.cc	2005-11-09 17:59:48.000000000 +0100
@@ -406,31 +406,19 @@
 }
 
 //! Return the signal owned by plot canvas
-#if (GTKMM_MAJOR_VERSION == 2 && GTKMM_MINOR_VERSION >= 4)
-sigc::signal3<void,int,int,GdkEventButton*> Plot::signal_plot_mouse_press()
-#else
 SigC::Signal3<void,int,int,GdkEventButton*> Plot::signal_plot_mouse_press()
-#endif
 {
     return canvas_.signal_plot_mouse_press;
 }
 
 //! Return the signal owned by plot canvas
-#if (GTKMM_MAJOR_VERSION == 2 && GTKMM_MINOR_VERSION >= 4)
-sigc::signal3<void,int,int,GdkEventButton*> Plot::signal_plot_mouse_release()
-#else
 SigC::Signal3<void,int,int,GdkEventButton*> Plot::signal_plot_mouse_release()
-#endif
 {
     return canvas_.signal_plot_mouse_release;
 }
 
 //! Return the signal owned by plot canvas
-#if (GTKMM_MAJOR_VERSION == 2 && GTKMM_MINOR_VERSION >= 4)
-sigc::signal3<void,int,int,GdkEventMotion*> Plot::signal_plot_mouse_move()
-#else
 SigC::Signal3<void,int,int,GdkEventMotion*> Plot::signal_plot_mouse_move()
-#endif
 {
     return canvas_.signal_plot_mouse_move;
 }
diff -Naur ../plotmmNew/plotmm/plot.h plotmm-0.1.2/plotmm/plot.h
--- ../plotmmNew/plotmm/plot.h	2007-11-22 17:20:25.000000000 +0100
+++ plotmm-0.1.2/plotmm/plot.h	2004-03-09 10:51:55.000000000 +0100
@@ -78,16 +78,10 @@
 	virtual bool begin_replot();
 	virtual void clear();
 	virtual void end_replot();
-	
-	#if (GTKMM_MAJOR_VERSION == 2 && GTKMM_MINOR_VERSION >= 4)
-	sigc::signal3<void,int,int,GdkEventButton*> signal_plot_mouse_press;
-	sigc::signal3<void,int,int,GdkEventButton*> signal_plot_mouse_release;
-	sigc::signal3<void,int,int,GdkEventMotion*> signal_plot_mouse_move;
-	#else
+
 	SigC::Signal3<void,int,int,GdkEventButton*> signal_plot_mouse_press;
 	SigC::Signal3<void,int,int,GdkEventButton*> signal_plot_mouse_release;
 	SigC::Signal3<void,int,int,GdkEventMotion*> signal_plot_mouse_move;
-	#endif
 
     protected:
 	virtual bool on_expose_event(GdkEventExpose* event);
@@ -146,17 +140,11 @@
 	PlotLabel *title() { return &title_; }
 	Scale *scale(PlotAxisID id) { return tickMark_[id]; }
 	PlotLabel *label(PlotAxisID id) { return axisLabel_[id]; }
-	
-	#if (GTKMM_MAJOR_VERSION == 2 && GTKMM_MINOR_VERSION >= 4)
-	sigc::signal3<void,int,int,GdkEventButton*> signal_plot_mouse_press();
-	sigc::signal3<void,int,int,GdkEventButton*> signal_plot_mouse_release();
-	sigc::signal3<void,int,int,GdkEventMotion*> signal_plot_mouse_move();
-	#else
+
 	SigC::Signal3<void,int,int,GdkEventButton*> signal_plot_mouse_press();
 	SigC::Signal3<void,int,int,GdkEventButton*> signal_plot_mouse_release();
 	SigC::Signal3<void,int,int,GdkEventMotion*> signal_plot_mouse_move();
-	#endif
-	
+
 	void set_selection(const Rectangle &r);
 	Rectangle get_selection() { return select_; }
 	void enable_selection();
diff -Naur ../plotmmNew/plotmm/scale.h plotmm-0.1.2/plotmm/scale.h
--- ../plotmmNew/plotmm/scale.h	2007-11-22 17:20:25.000000000 +0100
+++ plotmm-0.1.2/plotmm/scale.h	2004-03-05 16:25:43.000000000 +0100
@@ -133,11 +133,7 @@
 	/*! This signal is thrown whenever the scale is enabled or disabled
 	 *  \sa set_enabled, enabled
 	 */
-	#if (GTKMM_MAJOR_VERSION == 2 && GTKMM_MINOR_VERSION >= 4)
-	sigc::signal1<void,bool> signal_enabled;
-	#else
 	SigC::Signal1<void,bool> signal_enabled;
-	#endif
 
     protected:
 	virtual void on_realize();
