--- conf-cc	2004-08-17 03:13:15.000000000 +0200
+++ ../conf-cc	2008-07-29 22:19:12.240922551 +0200
@@ -1,4 +1,4 @@
-cc -O2
+cc -march=native -O3
 
 This will be used to compile .c files. Add any needed flags here; for
 example, if you want support for files larger than 2GB on Linux systems,
--- conf-root	2000-02-29 05:10:12.000000000 +0100
+++ ../conf-root	2008-07-29 22:19:12.267588529 +0200
@@ -1,4 +1,4 @@
-/usr/local
+TMPDIR_PKGBUILD
 
 The first line of this file gives the top-level directory for installation.
 Files will be installed in the appropriate subdirectories; bin, lib,
--- conf-version	2006-04-09 17:11:00.000000000 +0200
+++ ../conf-version	2008-07-29 22:19:12.267588529 +0200
@@ -1,4 +1,4 @@
-1.13
+1.13.arch
 
 The revision number of the software being distributed. This should not
 need to be changed by the user. If you modify this code, and then use
