--- chromium-9.0.576.0/third_party/ffmpeg/ffmpeg.gyp.orig	2010-11-10 18:34:47.665578900 +0600
+++ chromium-9.0.576.0/third_party/ffmpeg/ffmpeg.gyp	2010-11-10 18:48:56.621192286 +0600
@@ -907,6 +907,29 @@
                   '<!@(pkg-config --cflags libavcodec libavformat libavutil)',
                 ],
               },
+              'conditions': [
+                ['target_arch!="arm"', {
+                  'link_settings': {
+                    'libraries': [
+                      '-lvpx',
+                    ],
+                  },
+                }],
+                ['use_system_vpx!=0', {
+                  'include_dirs': [
+                    '/usr/include/vpx',
+                  ],
+                }, { # use_system_vpx==0
+                  'include_dirs': [
+                    '<(libvpx_incl_dir)',
+                  ],
+                  'link_settings': {
+                    'ldflags': [
+                      '-L<(libvpx_hack_dir)/lib/<(OS)/<(target_arch)',
+                    ],
+                  },
+                }],
+              ],
             }, {  # else use_system_ffmpeg==0, add local copy to include path
               'include_dirs': [
                 'config',
