--- gpu/command_buffer/common/types.h~	2011-03-31 13:32:58.000000000 +0400
+++ gpu/command_buffer/common/types.h	2011-03-31 13:33:12.000000000 +0400
@@ -11,6 +11,7 @@
 #include <stdint.h>
 #endif
 #include <string>
+#include <cstddef>
 
 typedef signed char         schar;
 typedef signed char         int8;
--- ui/gfx/codec/jpeg_codec.h~	2011-03-31 14:11:18.000000000 +0400
+++ ui/gfx/codec/jpeg_codec.h	2011-03-31 14:11:27.000000000 +0400
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <vector>
+#include <cstddef>
 
 class SkBitmap;
 
--- chrome/browser/search_engines/template_url_prepopulate_data.h	2011-03-31 14:48:03.000000000 +0400
+++ chrome/browser/search_engines/template_url_prepopulate_data.h~	2011-03-31 14:49:52.000000000 +0400
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <vector>
+#include <cstddef>
 
 class GURL;
 class PrefService;
--- third_party/tcmalloc/chromium/src/common.cc~	2011-03-29 20:46:14.000000000 +0400
+++ third_party/tcmalloc/chromium/src/common.cc	2011-03-29 20:47:02.000000000 +0400
@@ -30,6 +30,7 @@
 // ---
 // Author: Sanjay Ghemawat <opensource@google.com>
 
+#include <cstddef>
 #include "config.h"
 #include "system-alloc.h"
 #include "config.h"
--- third_party/libjingle/source/talk/base/sigslot.h~	2011-03-31 13:43:11.000000000 +0400
+++ third_party/libjingle/source/talk/base/sigslot.h	2011-03-31 13:52:32.000000000 +0400
@@ -80,6 +80,7 @@
 
 #include <set>
 #include <list>
+#include <cstddef>
 
 // On our copy of sigslot.h, we force single threading
 #define SIGSLOT_PURE_ISO
--- third_party/ots/src/os2.cc~	2011-03-31 14:03:13.000000000 +0400
+++ third_party/ots/src/os2.cc	2011-03-31 14:03:25.000000000 +0400
@@ -5,6 +5,7 @@
 #include "os2.h"
 
 #include "head.h"
+#include <cstddef>
 
 // OS/2 - OS/2 and Windows Metrics
 // http://www.microsoft.com/opentype/otspec/os2.htm
--- third_party/tcmalloc/chromium/src/base/vdso_support.h~	2011-03-29 23:20:20.000000000 +0400
+++ third_party/tcmalloc/chromium/src/base/vdso_support.h	2011-03-29 23:20:37.000000000 +0400
@@ -37,6 +37,7 @@
 
 #define HAVE_VDSO_SUPPORT 1
 
+#include <cstddef>
 #include <stdlib.h>     // for NULL
 #include <link.h>  // for ElfW
 #include "base/basictypes.h"
--- third_party/tcmalloc/chromium/src/base/stl_allocator.h~	2011-03-31 12:53:13.000000000 +0400
+++ third_party/tcmalloc/chromium/src/base/stl_allocator.h	2011-03-31 12:53:32.000000000 +0400
@@ -38,6 +38,7 @@
 #include <config.h>
 
 #include <limits>
+#include <cstddef>
 
 #include "base/basictypes.h"
 #include "base/logging.h"
--- third_party/tcmalloc/chromium/src/symbolize.h~	2011-03-31 12:58:42.000000000 +0400
+++ third_party/tcmalloc/chromium/src/symbolize.h	2011-03-31 12:59:00.000000000 +0400
@@ -38,6 +38,7 @@
 #include <stdint.h>  // for uintptr_t
 #endif
 #include <map>
+#include <cstddef>
 
 using std::map;
 
--- third_party/tcmalloc/chromium/src/system-alloc.h~	2011-03-31 13:01:01.000000000 +0400
+++ third_party/tcmalloc/chromium/src/system-alloc.h	2011-03-31 13:01:15.000000000 +0400
@@ -37,6 +37,7 @@
 #define TCMALLOC_SYSTEM_ALLOC_H_
 
 #include <config.h>
+#include <cstddef>
 #include "internal_logging.h"
 
 // REQUIRES: "alignment" is a power of two or "0" to indicate default alignment
--- ppapi/cpp/paint_aggregator.h.orig	2011-04-07 15:09:18.135353854 +0700
+++ ppapi/cpp/paint_aggregator.h	2011-04-07 15:10:24.806763997 +0700
@@ -6,6 +6,7 @@
 #define PPAPI_CPP_PAINT_AGGREGATOR_H_
 
 #include <vector>
+#include <cstddef>
 
 #include "ppapi/cpp/point.h"
 #include "ppapi/cpp/rect.h"
