Index: base/system_monitor/system_monitor_unittest.cc
diff --git a/base/system_monitor/system_monitor_unittest.cc b/base/system_monitor/system_monitor_unittest.cc
index 9517247c149efe5fc8e1853c2ef28252e09ab05f..8756c58a04370f26b57659d10af3f49f279c0556 100644
--- a/base/system_monitor/system_monitor_unittest.cc
+++ b/base/system_monitor/system_monitor_unittest.cc
@@ -47,8 +47,6 @@ TEST(SystemMonitor, PowerNotifications) {
 
   // Initialize a message loop for this to run on.
   MessageLoop loop;
-  // Initialize time() since it registers as a SystemMonitor observer.
-  base::Time now = base::Time::Now();
 
   SystemMonitor system_monitor;
   PowerTest test[kObservers];
