Index: base/id_map_unittest.cc
diff --git a/base/id_map_unittest.cc b/base/id_map_unittest.cc
index 54475b678919c968a2dca5db28170711d7ef7ac4..bf82eb686e85e7286eecb984c38dcceeae9a2593 100644
--- a/base/id_map_unittest.cc
+++ b/base/id_map_unittest.cc
@@ -165,11 +165,9 @@ TEST_F(IDMapTest, OwningPointersDeletesThemOnDestruct) {
 
   int external_del_count = 0;
   DestructorCounter* external_obj[kCount];
-  int map_external_ids[kCount];
 
   int owned_del_count = 0;
   DestructorCounter* owned_obj[kCount];
-  int map_owned_ids[kCount];
 
   {
     IDMap<DestructorCounter> map_external;
@@ -177,10 +175,10 @@ TEST_F(IDMapTest, OwningPointersDeletesThemOnDestruct) {
 
     for (int i = 0; i < kCount; ++i) {
       external_obj[i] = new DestructorCounter(&external_del_count);
-      map_external_ids[i] = map_external.Add(external_obj[i]);
+      map_external.Add(external_obj[i]);
 
       owned_obj[i] = new DestructorCounter(&owned_del_count);
-      map_owned_ids[i] = map_owned.Add(owned_obj[i]);
+      map_owned.Add(owned_obj[i]);
     }
   }
 
