--- chrome/nacl.gypi	2011-09-08 21:48:53.235859194 +0200
+++ chrome/nacl.gypi	2011-09-08 21:49:11.602677395 +0200
@@ -111,7 +111,7 @@
               'action_name': 'nacl_irt',
               'message': 'Building NaCl IRT',
               'inputs': [
-                '<!@(<(irt_inputs_cmd) --platform=x86-32)',
+                #'<!@(<(irt_inputs_cmd) --platform=x86-32)',
               ],
               'outputs': ['<(PRODUCT_DIR)/nacl_irt_x86_32.nexe'],
               'action': [
@@ -127,7 +127,7 @@
               'action_name': 'nacl_irt',
               'message': 'Building NaCl IRT',
               'inputs': [
-                '<!@(<(irt_inputs_cmd) --platform=x86-64)',
+                #'<!@(<(irt_inputs_cmd) --platform=x86-64)',
               ],
               'outputs': ['<(PRODUCT_DIR)/nacl_irt_x86_64.nexe'],
               'action': [
