--- src/chrome/nacl.gypi 2011-09-02 00:33:33.277071461 -0500
+++ src/chrome/nacl.gypi 2011-09-02 19:13:22.035161396 -0500
@@ -127,7 +127,7 @@
               'action_name': 'nacl_irt',
               'message': 'Building NaCl IRT',
               'inputs': [
-                '<!@(<(irt_inputs_cmd) --platform=x86-64)',
+                #'<!@(<(irt_inputs_cmd) --platform=x86-64)',
               ],
               'outputs': ['<(PRODUCT_DIR)/nacl_irt_x86_64.nexe'],
               'action': [
