*** third_party/ffmpeg/avcodec-53.sigs	2012-01-18 03:04:31.000000000 +0100
--- third_party/ffmpeg/avcodec-53.sigs	2012-01-22 02:13:56.359631164 +0100
***************
*** 9,15 ****
  AVFrame *avcodec_alloc_frame(void);
  int av_new_packet(AVPacket *pkt, int size);
  int avcodec_decode_audio3(AVCodecContext *avctx, int16_t *samples, int *frame_size_ptr, AVPacket *avpkt);
! int avcodec_decode_video2(AVCodecContext *avctx, AVFrame *picture, int *got_picture_ptr, AVPacket *avpkt);
  int avcodec_open(AVCodecContext *avctx, AVCodec *codec);
  int avcodec_close(AVCodecContext *avctx);
  int avcodec_thread_init(AVCodecContext *s, int thread_count);
--- 9,15 ----
  AVFrame *avcodec_alloc_frame(void);
  int av_new_packet(AVPacket *pkt, int size);
  int avcodec_decode_audio3(AVCodecContext *avctx, int16_t *samples, int *frame_size_ptr, AVPacket *avpkt);
! int avcodec_decode_video2(AVCodecContext *avctx, AVFrame *picture, int *got_picture_ptr, const AVPacket *avpkt);
  int avcodec_open(AVCodecContext *avctx, AVCodec *codec);
  int avcodec_close(AVCodecContext *avctx);
  int avcodec_thread_init(AVCodecContext *s, int thread_count);
*** content/plugin/plugin_thread.cc	2012-01-18 03:01:20.000000000 +0100
--- content/plugin/plugin_thread.cc	2012-01-22 02:41:10.837442947 +0100
***************
*** 81,87 ****
    {
      // XEmbed plugins assume they are hosted in a Gtk application, so we need
      // to initialize Gtk in the plugin process.
!     g_thread_init(NULL);
  
      // Flash has problems receiving clicks with newer GTKs due to the
      // client-side windows change.  To be safe, we just always set the
--- 81,94 ----
    {
      // XEmbed plugins assume they are hosted in a Gtk application, so we need
      // to initialize Gtk in the plugin process.
!     std::string glib_version = glib_check_version (2,24,0);
!     std::string major_version("GLib version too old (major mismatch)");
!     std::string minor_version("GLib version too old (micro mismatch)");
!     if (glib_version == major_version) {
! 	g_thread_init (NULL);
!     } else if (glib_version == minor_version) {
! 	g_thread_init (NULL);
!     }
  
      // Flash has problems receiving clicks with newer GTKs due to the
      // client-side windows change.  To be safe, we just always set the
*** ui/base/l10n/l10n_util.cc	2012-01-18 03:03:14.000000000 +0100
--- ui/base/l10n/l10n_util.cc	2012-01-22 02:36:04.792434322 +0100
***************
*** 5,11 ****
  #include "ui/base/l10n/l10n_util.h"
  
  #if defined(USE_X11)
! #include <glib/gutils.h>
  #endif
  
  #include <algorithm>
--- 5,11 ----
  #include "ui/base/l10n/l10n_util.h"
  
  #if defined(USE_X11)
! #include <glib.h>
  #endif
  
  #include <algorithm>