*** third_party/tcmalloc/chromium/src/base/linuxthreads.cc	2012-07-08 13:07:06.125973769 +0200
--- third_party/tcmalloc/chromium/src/base/linuxthreads.cc	2012-07-08 13:07:06.139307032 +0200
***************
*** 193,199 ****
  /* Signal handler to help us recover from dying while we are attached to
   * other threads.
   */
! static void SignalHandler(int signum, siginfo_t *si, void *data) {
    if (sig_pids != NULL) {
      if (signum == SIGABRT) {
        while (sig_num_threads-- > 0) {

--- 193,199 ----
  /* Signal handler to help us recover from dying while we are attached to
   * other threads.
   */
! static void SignalHandler(int signum, siginfo *si, void *data) {
    if (sig_pids != NULL) {
      if (signum == SIGABRT) {
        while (sig_num_threads-- > 0) {
