--- src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp.orig	2010-05-28 13:44:54.165839939 +0700
+++ src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp	2010-05-28 13:47:08.362081806 +0700
@@ -931,6 +931,12 @@
             # run-time determination.
             'WEBCORE_NAVIGATOR_PLATFORM="Linux i686"',
           ],
+          'configurations': {
+            'Debug': {
+              'cflags!': [ '-O0' ],
+              'cflags': [ '-O1' ],
+            }
+          },
         }],
         ['OS=="mac"', {
           'dependencies': [
