--- client/mysqlimport.c.old	2013-08-20 15:32:09.367131119 +0200
+++ client/mysqlimport.c	2013-08-20 15:49:44.847324537 +0200
@@ -50,7 +50,7 @@
 static char *add_load_option(char *ptr,const char *object,
 			     const char *statement);
 
-static my_bool	verbose=0,lock_tables=0,ignore_errors=0,opt_delete=0,
+static my_bool	verbose=0,lock_tables=0,ignore_errors=0,opt_delete=0, ignore_foreign_keys=0,
 		replace=0,silent=0,ignore=0,opt_compress=0,
                 opt_low_priority= 0, tty_password= 0;
 static my_bool debug_info_flag= 0, debug_check_flag= 0;
@@ -126,6 +126,10 @@
    &current_host, 0, GET_STR, REQUIRED_ARG, 0, 0, 0, 0, 0, 0},
   {"ignore", 'i', "If duplicate unique key was found, keep old row.",
    &ignore, &ignore, 0, GET_BOOL, NO_ARG, 0, 0, 0, 0, 0, 0},
+  {"ignore-foreign-keys", 'k',
+   "Ignore foreign key checks while importing data.",
+   (uchar**) &ignore_foreign_keys, (uchar**) &ignore_foreign_keys, 0, GET_BOOL, NO_ARG, 
+   0, 0, 0, 0, 0, 0},
   {"ignore-lines", OPT_IGN_LINES, "Ignore first n lines of data infile.",
    &opt_ignore_lines, &opt_ignore_lines, 0, GET_LL,
    REQUIRED_ARG, 0, 0, 0, 0, 0, 0},
@@ -467,6 +471,9 @@
     ignore_errors=0;
     db_error(mysql);
   }
+  if (ignore_foreign_keys) {
+    mysql_query(mysql, "SET FOREIGN_KEY_CHECKS = 0;");
+  }
   return mysql;
 }
 
--- client/mysql.cc.old	2013-08-20 15:36:41.513850176 +0200
+++ client/mysql.cc	2013-08-20 15:48:07.317307811 +0200
@@ -131,7 +131,7 @@
 typedef enum enum_info_type INFO_TYPE;
 
 static MYSQL mysql;			/* The connection */
-static my_bool ignore_errors=0,wait_flag=0,quick=0,
+static my_bool ignore_errors=0,ignore_foreign_keys=0,wait_flag=0,quick=0,
                connected=0,opt_raw_data=0,unbuffered=0,output_tables=0,
 	       opt_rehash=1,skip_updates=0,safe_updates=0,one_database=0,
 	       opt_compress=0, using_opt_local_infile=0,
@@ -1235,6 +1235,12 @@
 
   put_info(ORACLE_WELCOME_COPYRIGHT_NOTICE("2000"), INFO_INFO);
 
+  if (ignore_foreign_keys){
+          mysql_query(&mysql, "SET FOREIGN_KEY_CHECKS = 0;");
+          put_info("Foreign key checks are disabled! \\g.",
+                   INFO_INFO);
+  }
+
 #ifdef HAVE_READLINE
   initialize_readline((char*) my_progname);
   if (!status.batch && !quick && !opt_html && !opt_xml)
@@ -1482,6 +1488,10 @@
   {"force", 'f', "Continue even if we get an SQL error. Sets abort-source-on-error to 0",
    &ignore_errors, &ignore_errors, 0, GET_BOOL, NO_ARG, 0, 0,
    0, 0, 0, 0},
+  {"ignore-foreign-keys", 'k',
+     "Ignore foreign key checks while importing data.",
+     (uchar**) &ignore_foreign_keys, (uchar**) &ignore_foreign_keys, 0, GET_BOOL, NO_ARG,
+     0, 0, 0, 0, 0, 0},
   {"named-commands", 'G',
    "Enable named commands. Named commands mean this program's internal "
    "commands; see mysql> help . When enabled, the named commands can be "
