--- urlbar.cpp.orig	2010-06-16 12:33:45.000000000 +0200
+++ urlbar.cpp	2010-06-16 12:39:36.902919839 +0200
@@ -185,7 +185,7 @@
     }
     else
     {
-        QColor loadingColor = Application::palette().color(QPalette::ToolTipBase);
+        QColor loadingColor = Application::palette().color(QPalette::Highlight);
 
         QLinearGradient gradient( QPoint(0, 0), QPoint(width(), height()) );
         gradient.setColorAt(0, loadingColor);
