--- findbar.cpp	2010-06-16 12:05:31.000000000 +0200
+++ src/mainline/src/findbar.cpp	2010-06-16 12:19:15.279730230 +0200
@@ -162,16 +162,19 @@
     if (m_lineEdit->text().isEmpty())
     {
         p.setColor(QPalette::Base, QColor(KApplication::palette().color(QPalette::Active, QPalette::Base)));
+        p.setColor(QPalette::Text, QColor(KApplication::palette().color(QPalette::Text)));
     }
     else
     {
         if (match)
         {
             p.setColor(QPalette::Base, QColor(186, 249, 206));
+            p.setColor(QPalette::Text, Qt::black);
         }
         else
         {
             p.setColor(QPalette::Base, QColor(247, 130, 130)); // previous were 247, 230, 230
+            p.setColor(QPalette::Text, Qt::black);
         }
     }
     m_lineEdit->setPalette(p);
