Index: cblrepo/haskell-hsmagick/src/hsmagick-0.5/Graphics/Transform/Magick/FFIHelpers.hsc
===================================================================
--- cblrepo.orig/haskell-hsmagick/src/hsmagick-0.5/Graphics/Transform/Magick/FFIHelpers.hsc
+++ cblrepo/haskell-hsmagick/src/hsmagick-0.5/Graphics/Transform/Magick/FFIHelpers.hsc
@@ -44,6 +44,7 @@ import qualified Foreign.Concurrent as F
 import Control.Exception
 import Prelude hiding (maximum, minimum)
 import Control.Monad
+import System.IO.Unsafe (unsafePerformIO)
 
 -- functions to help with doing FFI 
 
Index: cblrepo/haskell-hsmagick/src/hsmagick-0.5/Graphics/Transform/Magick/Images.hsc
===================================================================
--- cblrepo.orig/haskell-hsmagick/src/hsmagick-0.5/Graphics/Transform/Magick/Images.hsc
+++ cblrepo/haskell-hsmagick/src/hsmagick-0.5/Graphics/Transform/Magick/Images.hsc
@@ -64,6 +64,7 @@ import Graphics.Transform.Magick.Util
 import Data.Char
 import Data.List
 import System.Directory
+import System.IO.Unsafe (unsafePerformIO)
 
 -- The externally-visible Haskell API for GraphicsMagick.
 
