diff -Naur gnome-power-manager-2.32.0.orig/configure.ac gnome-power-manager-2.32.0/configure.ac
--- gnome-power-manager-2.32.0.orig/configure.ac	2010-09-22 13:02:17.000000000 +0200
+++ gnome-power-manager-2.32.0/configure.ac	2011-03-15 23:43:49.800699458 +0100
@@ -104,7 +104,7 @@
 GNOME_KEYRING_REQUIRED=0.6.0
 DBUS_REQUIRED=1.0
 DBUS_GLIB_REQUIRED=0.70
-LIBNOTIFY_REQUIRED=0.4.3
+LIBNOTIFY_REQUIRED=0.7.0
 LIBWNCK_REQUIRED=2.10.0
 CAIRO_REQUIRED=1.0.0
 UNIQUE_REQUIRED=0.9.4
diff -Naur gnome-power-manager-2.32.0.orig/src/gpm-manager.c gnome-power-manager-2.32.0/src/gpm-manager.c
--- gnome-power-manager-2.32.0.orig/src/gpm-manager.c	2010-09-22 13:02:17.000000000 +0200
+++ gnome-power-manager-2.32.0/src/gpm-manager.c	2011-03-15 23:43:00.815623248 +0100
@@ -480,12 +480,8 @@
 	/* close any existing notification of this class */
 	gpm_manager_notify_close (manager, *notification_class);
 
-	/* if the status icon is hidden, don't point at it */
-	if (manager->priv->status_icon != NULL &&
-	    gtk_status_icon_is_embedded (manager->priv->status_icon))
-		notification = notify_notification_new_with_status_icon (title, message, icon, manager->priv->status_icon);
-	else
-		notification = notify_notification_new (title, message, icon, NULL);
+	/* create a new notification */
+	notification = notify_notification_new (title, message, icon);
 	notify_notification_set_timeout (notification, timeout);
 	notify_notification_set_urgency (notification, urgency);
 	g_signal_connect (notification, "closed", G_CALLBACK (gpm_manager_notification_closed_cb), notification_class);
