--- installer.rb.old	2010-06-11 14:39:17.586452869 +0200
+++ installer.rb	2010-06-11 14:39:32.473118124 +0200
@@ -425,13 +425,13 @@
 		
 		# Fix the shebang lines of scripts in 'bin' folder.
 		fix_shebang_lines("#{@destdir}#{@prefix}/bin", "#{@prefix}/bin/ruby")
-		Dir.chdir("#{@destdir}#{@prefix}/lib/ruby/gems/1.8/gems") do
-			if !Dir["sqlite3-ruby*"].empty?
-				# The sqlite3-ruby gem installs files with wrong
-				# permissions. We fix this.
-				sh "chmod -R g+r,o+r,o-w sqlite3-ruby*"
-			end
-		end
+		#Dir.chdir("#{@destdir}#{@prefix}/lib/ruby/gems/1.8/gems") do
+		#	if !Dir["sqlite3-ruby*"].empty?
+		#		# The sqlite3-ruby gem installs files with wrong
+		#		# permissions. We fix this.
+		#		sh "chmod -R g+r,o+r,o-w sqlite3-ruby*"
+		#	end
+		#end
 	end
 	
 	def show_finalization_screen
