--- CMakeLists.txt.orig	2008-08-27 16:04:23.365956729 +0200
+++ CMakeLists.txt	2008-08-27 16:05:14.792210522 +0200
@@ -1,5 +1,15 @@
 project(plasma-system-monitor)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${PLASMA_INCLUDE_DIR})
+
+find_package(KDE4 REQUIRED)
+include(KDE4Defaults)
+find_package(Plasma REQUIRED)
+
+add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}
+			${PLASMA_INCLUDE_DIR}
+			${KDE4_INCLUDES}
+)
 
 # lib
 
