--- a/catkin/cmake/python.cmake	2013-05-25 16:20:44.158812486 -0400
+++ b/catkin/cmake/python.cmake	2013-05-25 16:33:33.828811885 -0400
@@ -6,7 +6,7 @@
 set(PYTHON_VERSION_XDOTY ${PYTHON_VERSION_XDOTY} CACHE STRING "Python version")
 
 #This should be resolved automatically one day...
-option(SETUPTOOLS_DEB_LAYOUT "ON for debian style python packages layout" ON)
+option(SETUPTOOLS_DEB_LAYOUT "ON for debian style python packages layout" OFF)
 
 if(APPLE OR WIN32)
   set(SETUPTOOLS_DEB_LAYOUT OFF)
