--- adl.c.orig	2014-01-16 00:56:37.588878903 -0600
+++ adl.c	2014-01-16 00:56:31.317879127 -0600
@@ -468,6 +468,8 @@
 		else
 			ga->has_fanspeed = true;
 
+		ga->DefFanSpeedValue.iSpeedType=ADL_DL_FANCTRL_SPEED_TYPE_PERCENT;
+
 		/* Save the fanspeed values as defaults in case we reset later */
 		ADL_Overdrive5_FanSpeed_Get(ga->iAdapterIndex, 0, &ga->DefFanSpeedValue);
 		if (gpus[gpu].gpu_fan)
