diff --git a/Version b/Version
index b490fe5..151b81d 100644
--- a/Version
+++ b/Version
@@ -1 +1 @@
-omnetpp-3.3p1
+omnetpp-3.3p2
diff --git a/setenv b/setenv
old mode 100644
new mode 100755
diff --git a/src/plove/engine/filetokenizer.cc b/src/plove/engine/filetokenizer.cc
index 96d8659..cddf79c 100644
--- a/src/plove/engine/filetokenizer.cc
+++ b/src/plove/engine/filetokenizer.cc
@@ -13,6 +13,7 @@
 *--------------------------------------------------------------*/
 
 
+#include <stdio.h>
 #include <string.h>
 #include <assert.h>
 #include <sstream>
diff --git a/src/plove/engine/filewriter.cc b/src/plove/engine/filewriter.cc
index 9f34527..cb1c325 100644
--- a/src/plove/engine/filewriter.cc
+++ b/src/plove/engine/filewriter.cc
@@ -16,6 +16,7 @@
 #pragma warning(disable:4786)
 #endif
 
+#include <stdio.h>
 #include "channel.h"
 #include "filewriter.h"
 
diff --git a/src/plove/engine/vectorfilewriter.cc b/src/plove/engine/vectorfilewriter.cc
index 529af8c..3ecd78d 100644
--- a/src/plove/engine/vectorfilewriter.cc
+++ b/src/plove/engine/vectorfilewriter.cc
@@ -16,6 +16,7 @@
 #pragma warning(disable:4786)
 #endif
 
+#include <stdio.h>
 #include <stdlib.h>
 #include "channel.h"
 #include "vectorfilewriter.h"
diff --git a/src/scalars/engine/filetokenizer.cc b/src/scalars/engine/filetokenizer.cc
index 2fc2a3e..1b730dd 100644
--- a/src/scalars/engine/filetokenizer.cc
+++ b/src/scalars/engine/filetokenizer.cc
@@ -13,6 +13,7 @@
 *--------------------------------------------------------------*/
 
 
+#include <stdio.h>
 #include <assert.h>
 #include <string.h>
 #include <sstream>
