diff -uNr bin.org/dbtoepub bin/dbtoepub
--- bin.org/dbtoepub	2011-10-30 12:11:53.386513024 +0100
+++ bin/dbtoepub	2011-10-30 13:07:20.867922640 +0100
@@ -24,7 +24,7 @@
 require 'optparse'
 require 'tmpdir'
 
-require 'docbook'
+require 'dbtoepub/docbook'
 
 verbose = false
 debug = false
diff -uNr bin.org/lib/docbook.rb bin/lib/docbook.rb
--- bin.org/lib/docbook.rb	2011-10-30 12:11:53.386513024 +0100
+++ bin/lib/docbook.rb	2011-10-30 13:07:20.867922640 +0100
@@ -5,9 +5,10 @@
 
   class Epub
     CHECKER = "epubcheck"
-    STYLESHEET = File.expand_path(File.join(File.dirname(__FILE__), '..', '..', "docbook.xsl"))
+    STYLESHEET_PATH = "/usr/share/xml/docbook/xsl-stylesheets-1.77.1/epub"
+    STYLESHEET = File.expand_path(File.join(STYLESHEET_PATH, "docbook.xsl"))
     CALLOUT_PATH = File.join('images', 'callouts')
-    CALLOUT_FULL_PATH = File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..', CALLOUT_PATH))
+    CALLOUT_FULL_PATH = File.expand_path(File.join(STYLESHEET_PATH, CALLOUT_PATH))
     CALLOUT_LIMIT = 15
     CALLOUT_EXT = ".png"
     XSLT_PROCESSOR = "xsltproc"
