--- a/handleMenu.py	2013-11-26 11:53:01.493833103 -0500
+++ b/handleMenu.py	2013-11-26 11:52:52.517201324 -0500
@@ -11,6 +11,7 @@
 #       characters [, ], {, }, (, ), <, or >
 
 def read_menu(menufile):
+    menu = []
     if isfile(menufile):
 
         # Read menu contents
@@ -20,7 +21,6 @@
         cFile.close()
 
         # Parse the file
-        menu = []
         for menuItem in menuContents:
             # TODO: Check for comments, save them as 4th field in the menu... 
             # then I can implement a "display in menu" -field :)
