--- a/configure	2012-04-16 11:09:10.889423950 +0100
+++ b/configure	2012-04-16 11:09:45.641890001 +0100
@@ -1555,7 +1555,7 @@
 
   --enable-debug          turn on debugging output and compile options
   --enable-gexec          turn on gexec support (platform-specific)
-  --enable-setuid=USER    turn on setuid support (default setuid=nobody)
+  --enable-setuid=USER    turn on setuid support (default setuid=ganglia)
   --enable-setgid=GROUP   turn on setgid support (default setgid=no)
   --enable-pedantic       turn on pedantic mode during compile
   --enable-memcheck       turn on memory checking during compile
@@ -11088,7 +11088,7 @@
 fi
 
 
-setuid_user=nobody
+setuid_user=ganglia
 # Check whether --enable-setuid was given.
 if test "${enable_setuid+set}" = set; then :
   enableval=$enable_setuid; if test x"$enableval" != xno; then no_setuid=0; setuid_user=$enableval ; fi
--- a/gmetad/gmetad.conf.in	2012-03-07 13:25:17.649872310 +0000
+++ b/gmetad/gmetad.conf.in	2012-03-07 13:25:33.419936478 +0000
@@ -99,8 +99,8 @@
 #-------------------------------------------------------------------------------
 # User gmetad will setuid to (defaults to "nobody")
 # default: "nobody"
-# setuid_username "nobody"
-#
+setuid_username "ganglia"
+
 #-------------------------------------------------------------------------------
 # Umask to apply to created rrd files and grid directory structure
 # default: 0 (files are public)
