*** src/trunk/common/im_png.cpp	2010-05-20 12:04:49.878773535 +0200
--- im_png.cpp	2010-05-20 12:05:09.295436080 +0200
***************
*** 47,53 ****
    FreeData ();
  
    file.Read (sig, 8);
!   if (!png_check_sig(sig, 8))
      return false;	// bad signature
  
    png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
--- 47,53 ----
    FreeData ();
  
    file.Read (sig, 8);
!   if (!(png_sig_cmp(sig, 0, 8) == 0))
      return false;	// bad signature
  
    png_ptr = png_create_read_struct(PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
