diff -aur opensesame-2.8.0.orig/opensesame opensesame-2.8.0/opensesame
--- opensesame-2.8.0.orig/opensesame	2014-02-17 16:38:27.674038759 +0000
+++ opensesame-2.8.0/opensesame	2014-02-17 16:46:49.612491557 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.0.orig/opensesamerun opensesame-2.8.0/opensesamerun
--- opensesame-2.8.0.orig/opensesamerun	2014-02-17 16:38:27.674038759 +0000
+++ opensesame-2.8.0/opensesamerun	2014-02-17 16:47:02.938709802 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.0.orig/setup.py opensesame-2.8.0/setup.py
--- opensesame-2.8.0.orig/setup.py	2014-02-17 16:38:27.674038759 +0000
+++ opensesame-2.8.0/setup.py	2014-02-17 16:47:12.271731264 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
