diff -aur opensesame-ORIG/libopensesame/misc.py opensesame-0.26/libopensesame/misc.py
--- opensesame-ORIG/libopensesame/misc.py	2012-04-01 13:02:29.000000000 +0100
+++ opensesame-0.26/libopensesame/misc.py	2012-04-03 16:34:25.623398722 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-ORIG/libqtopensesame/qtopensesame.py opensesame-0.26/libqtopensesame/qtopensesame.py
--- opensesame-ORIG/libqtopensesame/qtopensesame.py	2012-03-29 12:04:12.000000000 +0100
+++ opensesame-0.26/libqtopensesame/qtopensesame.py	2012-04-03 16:36:28.212139694 +0100
@@ -890,7 +890,7 @@
 			cmd.append(py_exe)
 
 		if sys.argv[0] == "opensesame" and os.name != "nt":
-			cmd.append("python")
+			cmd.append("python2")
 
 		cmd.append(sys.argv[0])
 		cmd.append(self.current_path)
diff -aur opensesame-ORIG/opensesame opensesame-0.26/opensesame
--- opensesame-ORIG/opensesame	2012-03-12 10:13:17.000000000 +0000
+++ opensesame-0.26/opensesame	2012-04-03 16:38:05.773790519 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-ORIG/opensesamerun opensesame-0.26/opensesamerun
--- opensesame-ORIG/opensesamerun	2012-03-12 10:13:17.000000000 +0000
+++ opensesame-0.26/opensesamerun	2012-04-03 16:38:12.740575072 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
