diff -aur opensesame-2.8.1.orig/dev-scripts/architecture.py opensesame-2.8.1/dev-scripts/architecture.py
--- opensesame-2.8.1.orig/dev-scripts/architecture.py	2014-04-10 11:56:03.141795073 +0100
+++ opensesame-2.8.1/dev-scripts/architecture.py	2014-04-10 11:57:22.147852829 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
@@ -18,7 +18,7 @@
 along with OpenSesame.  If not, see <http://www.gnu.org/licenses/>.
 """
 
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/dev-scripts/check_ts.py opensesame-2.8.1/dev-scripts/check_ts.py
--- opensesame-2.8.1.orig/dev-scripts/check_ts.py	2014-04-10 11:56:03.141795073 +0100
+++ opensesame-2.8.1/dev-scripts/check_ts.py	2014-04-10 11:57:22.147852829 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/dev-scripts/code_check.py opensesame-2.8.1/dev-scripts/code_check.py
--- opensesame-2.8.1.orig/dev-scripts/code_check.py	2014-04-10 11:56:03.141795073 +0100
+++ opensesame-2.8.1/dev-scripts/code_check.py	2014-04-10 11:57:22.151186138 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/dev-scripts/docstruct.py opensesame-2.8.1/dev-scripts/docstruct.py
--- opensesame-2.8.1.orig/dev-scripts/docstruct.py	2014-04-10 11:56:04.271786364 +0100
+++ opensesame-2.8.1/dev-scripts/docstruct.py	2014-04-10 11:57:22.154519447 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/dev-scripts/item-help.py opensesame-2.8.1/dev-scripts/item-help.py
--- opensesame-2.8.1.orig/dev-scripts/item-help.py	2014-04-10 11:56:04.271786364 +0100
+++ opensesame-2.8.1/dev-scripts/item-help.py	2014-04-10 11:57:22.157852755 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/dev-scripts/update-hidden.py opensesame-2.8.1/dev-scripts/update-hidden.py
--- opensesame-2.8.1.orig/dev-scripts/update-hidden.py	2014-04-10 11:56:04.271786364 +0100
+++ opensesame-2.8.1/dev-scripts/update-hidden.py	2014-04-10 11:57:22.161186063 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/opensesame opensesame-2.8.1/opensesame
--- opensesame-2.8.1.orig/opensesame	2014-04-10 11:56:08.941750372 +0100
+++ opensesame-2.8.1/opensesame	2014-04-10 11:59:50.580042177 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/opensesameandroid.py opensesame-2.8.1/opensesameandroid.py
--- opensesame-2.8.1.orig/opensesameandroid.py	2014-04-10 11:56:08.941750372 +0100
+++ opensesame-2.8.1/opensesameandroid.py	2014-04-10 11:57:22.524516593 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/opensesamerun opensesame-2.8.1/opensesamerun
--- opensesame-2.8.1.orig/opensesamerun	2014-04-10 11:56:08.941750372 +0100
+++ opensesame-2.8.1/opensesamerun	2014-04-10 11:57:37.271069606 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/plugins/pp_io/parallelppdev.py opensesame-2.8.1/plugins/pp_io/parallelppdev.py
--- opensesame-2.8.1.orig/plugins/pp_io/parallelppdev.py	2014-04-10 11:56:08.951750295 +0100
+++ opensesame-2.8.1/plugins/pp_io/parallelppdev.py	2014-04-10 11:57:22.567849594 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # parallel port access using the ppdev driver
 
 import sys
diff -aur opensesame-2.8.1.orig/setup-android.py opensesame-2.8.1/setup-android.py
--- opensesame-2.8.1.orig/setup-android.py	2014-04-10 11:56:09.458413057 +0100
+++ opensesame-2.8.1/setup-android.py	2014-04-10 11:57:22.601182670 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/setup-mac.py opensesame-2.8.1/setup-mac.py
--- opensesame-2.8.1.orig/setup-mac.py	2014-04-10 11:56:09.458413057 +0100
+++ opensesame-2.8.1/setup-mac.py	2014-04-10 11:57:22.604515977 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/setup.py opensesame-2.8.1/setup.py
--- opensesame-2.8.1.orig/setup.py	2014-04-10 11:56:09.458413057 +0100
+++ opensesame-2.8.1/setup.py	2014-04-10 11:57:22.611182593 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-2.8.1.orig/setup-win32.py opensesame-2.8.1/setup-win32.py
--- opensesame-2.8.1.orig/setup-win32.py	2014-04-10 11:56:09.458413057 +0100
+++ opensesame-2.8.1/setup-win32.py	2014-04-10 11:57:22.607849285 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
