diff -aur opensesame-0.27.orig/libqtopensesame/qtopensesame.py opensesame-0.27/libqtopensesame/qtopensesame.py
--- opensesame-0.27.orig/libqtopensesame/qtopensesame.py	2012-12-13 12:13:49.651719990 +0000
+++ opensesame-0.27/libqtopensesame/qtopensesame.py	2012-12-13 12:16:19.337695255 +0000
@@ -737,7 +737,7 @@
 			cmd.append(py_exe)
 
 		if sys.argv[0] == "opensesame" and os.name != "nt":
-			cmd.append("python")
+			cmd.append("python2")
 
 		cmd.append(sys.argv[0])
 		cmd.append(self.current_path)
diff -aur opensesame-0.27.orig/opensesame opensesame-0.27/opensesame
--- opensesame-0.27.orig/opensesame	2012-12-13 12:13:49.668386953 +0000
+++ opensesame-0.27/opensesame	2012-12-13 12:17:19.318754316 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
diff -aur opensesame-0.27.orig/opensesamerun opensesame-0.27/opensesamerun
--- opensesame-0.27.orig/opensesamerun	2012-12-13 12:13:49.668386953 +0000
+++ opensesame-0.27/opensesamerun	2012-12-13 12:17:26.655550878 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #-*- coding:utf-8 -*-
 
 """
