diff --git a/colortheme b/colortheme
index b954d8a..c27da6f 100644
--- a/colortheme
+++ b/colortheme
@@ -4,7 +4,15 @@
 # Modified by Aaron Griffin
 # and further by Kazuo Teramoto
 
-
+case "${1}" in
+#help
+  *h|*help)
+    echo "uses unicode version as default"
+    echo "\"${0} -a\" ascii version"
+    echo "\"${0} -u\" unicode version"
+  ;;
+#ascii version
+  *a|*ascii)
 FGNAMES=(' black ' '  red  ' ' green ' ' yellow' '  blue ' 'magenta' '  cyan ' ' white ')
 BGNAMES=('DFT' 'BLK' 'RED' 'GRN' 'YEL' 'BLU' 'MAG' 'CYN' 'WHT')
 echo "     .--------------------------------------------------------------------------."
@@ -31,13 +39,9 @@ for b in $(seq 0 8); do
   fi
 done
 echo "     '--------------------------------------------------------------------------'"
-#!/bin/bash
-# Original: http://frexx.de/xterm-256-notes/
-#           http://frexx.de/xterm-256-notes/data/colortable16.sh
-# Modified by Aaron Griffin
-# and further by Kazuo Teramoto
-
-
+  ;;
+#unicode version
+  *u|*unicode|*)
 FGNAMES=(' black ' '  red  ' ' green ' ' yellow' '  blue ' 'magenta' '  cyan ' ' white ')
 BGNAMES=('DFT' 'BLK' 'RED' 'GRN' 'YEL' 'BLU' 'MAG' 'CYN' 'WHT')
 echo "     ┌──────────────────────────────────────────────────────────────────────────┐"
@@ -64,3 +68,5 @@ for b in $(seq 0 8); do
   fi
 done
 echo "     └──────────────────────────────────────────────────────────────────────────┘"
+  ;;
+esac
