diff -rupN chickennix-0.1.orig/Makefile chickennix-0.1/Makefile
--- chickennix-0.1.orig/Makefile	2009-02-17 08:37:22.000000000 +0100
+++ chickennix-0.1/Makefile	2009-02-26 17:24:12.000000000 +0100
@@ -1,12 +1,11 @@
 # Makefile for chickennix
-CC			=	gcc
-OBJECTS		=	 main.o enemy.o menu.o alphabet.o box.o highscore.o keyboard_input.o sound.o
+CC		=	gcc
+OBJECTS		=	main.o enemy.o menu.o alphabet.o box.o highscore.o keyboard_input.o sound.o
 LIBS		=	-lSDL_image -lSDL_mixer
 CFLAGS		=	-O2 -Wall -pedantic -std=c99
-SRC			=	src
+SRC		=	src
 bindir		=	$(PREFIX)/bin
 datarootdir	=	$(PREFIX)/share
-datagames	=	$(PREFIX)/share/games
 pixmapsdir	=	$(datarootdir)/pixmaps
 desktopdir	=	$(datarootdir)/applications
 mandir		=	$(datarootdir)/man
@@ -14,7 +13,8 @@ INSTALL		=	install
 INSTALL-DATA=	cd .. && install
 TARGET		=	chickennix
 
-PREFIX	:=	/usr
+PREFIX		=	/usr
+DESTDIR		=	~/
 
 
 chickennix:	$(OBJECTS)
@@ -50,38 +50,38 @@ sound.o: ${SRC}/sound.c
 	$(CC) `sdl-config --cflags` $(CFLAGS) -c ${SRC}/$*.c
 
 install: chickennix container data
-	$(INSTALL) -m 755 $(TARGET) ${PREFIX}/games
-	$(INSTALL) -m 755 $(TARGET) $(bindir)
-	$(INSTALL) -m 644 data/images/background/*.png $(datagames)/$(TARGET)/data/images/background
-	$(INSTALL) -m 644 data/images/background/*.JPG $(datagames)/$(TARGET)/data/images/background
-	$(INSTALL) -m 644 data/images/enemy/*.png	$(datagames)/$(TARGET)/data/images/enemy
-	$(INSTALL) -m 644 data/images/font/*.png 	$(datagames)/$(TARGET)/data/images/font
-	$(INSTALL) -m 644 data/images/ingame/*.png	$(datagames)/$(TARGET)/data/images/ingame
-	$(INSTALL) -m 644 data/images/menu/*.png	$(datagames)/$(TARGET)/data/images/menu
-	$(INSTALL) -m 644 data/images/other/*.png	$(datagames)/$(TARGET)/data/images/other
-	$(INSTALL) -m 644 data/images/*.bmp			$(datagames)/$(TARGET)/data/images
-	$(INSTALL) -m 644 data/sound/music/*		$(datagames)/$(TARGET)/data/sound/music
-	$(INSTALL) -m 644 data/sound/soundeffect/*	$(datagames)/$(TARGET)/data/sound/soundeffect
-	$(INSTALL) -m 644 data/user/License.txt		$(datagames)/$(TARGET)
-	$(INSTALL) -m 644 Makefile 					$(datagames)/$(TARGET)
-	$(INSTALL) -m 644 chickennix.png 			$(pixmapsdir)
-	$(INSTALL) -m 644 chickennix.desktop 		$(desktopdir)
+	$(INSTALL) -m 755 $(TARGET) $(DESTDIR)$(bindir)
+	$(INSTALL) -m 644 data/images/background/*.png	$(DESTDIR)$(datarootdir)/$(TARGET)/data/images/background
+	$(INSTALL) -m 644 data/images/background/*.JPG	$(DESTDIR)$(datarootdir)/$(TARGET)/data/images/background
+	$(INSTALL) -m 644 data/images/enemy/*.png	$(DESTDIR)$(datarootdir)/$(TARGET)/data/images/enemy
+	$(INSTALL) -m 644 data/images/font/*.png 	$(DESTDIR)$(datarootdir)/$(TARGET)/data/images/font
+	$(INSTALL) -m 644 data/images/ingame/*.png	$(DESTDIR)$(datarootdir)/$(TARGET)/data/images/ingame
+	$(INSTALL) -m 644 data/images/menu/*.png	$(DESTDIR)$(datarootdir)/$(TARGET)/data/images/menu
+	$(INSTALL) -m 644 data/images/other/*.png	$(DESTDIR)$(datarootdir)/$(TARGET)/data/images/other
+	$(INSTALL) -m 644 data/images/*.bmp		$(DESTDIR)$(datarootdir)/$(TARGET)/data/images
+	$(INSTALL) -m 644 data/sound/music/*		$(DESTDIR)$(datarootdir)/$(TARGET)/data/sound/music
+	$(INSTALL) -m 644 data/sound/soundeffect/*	$(DESTDIR)$(datarootdir)/$(TARGET)/data/sound/soundeffect
+	$(INSTALL) -m 644 $(TARGET).png 		$(DESTDIR)$(pixmapsdir)/
+	$(INSTALL) -m 644 $(TARGET).desktop 		$(DESTDIR)$(desktopdir)/
 	
 container:
-	mkdir -p $(datagames)/$(TARGET)
-	mkdir -p $(datagames)/$(TARGET)/data
-	mkdir -p $(datagames)/$(TARGET)/data/user
-	mkdir -p $(datagames)/$(TARGET)/data/level
-	mkdir -p $(datagames)/$(TARGET)/data/sound
-	mkdir -p $(datagames)/$(TARGET)/data/sound/music
-	mkdir -p $(datagames)/$(TARGET)/data/sound/soundeffect
-	mkdir -p $(datagames)/$(TARGET)/data/images
-	mkdir -p $(datagames)/$(TARGET)/data/images/background
-	mkdir -p $(datagames)/$(TARGET)/data/images/enemy
-	mkdir -p $(datagames)/$(TARGET)/data/images/font
-	mkdir -p $(datagames)/$(TARGET)/data/images/ingame
-	mkdir -p $(datagames)/$(TARGET)/data/images/menu
-	mkdir -p $(datagames)/$(TARGET)/data/images/other
+	mkdir -p $(DESTDIR)$(bindir)
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/user
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/level
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/sound
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/sound/music
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/sound/soundeffect
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/images
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/images/background
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/images/enemy
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/images/font
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/images/ingame
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/images/menu
+	mkdir -p $(DESTDIR)$(datarootdir)/$(TARGET)/data/images/other
+	mkdir -p $(DESTDIR)$(pixmapsdir)
+	mkdir -p $(DESTDIR)$(desktopdir)
 
 clean:
 	@echo "Cleaning:"
@@ -89,8 +89,8 @@ clean:
 
 uninstall:
 	@echo "Remove chickennix:"
-	rm -rf $(datagames)/$(TARGET)
+	rm -rf $(DESTDIR)$(datarootdir)/$(TARGET)
 	rm $(bindir)/$(TARGET)
-	rm $(pixmapsdir)/$(TARGET).png
-	rm $(desktopdir)/$(TARGET).desktop
-#	rm $(mandir)/man1/chickennix.1
+	rm $(DESTDIR)$(pixmapsdir)/$(TARGET).png
+	rm $(DESTDIR)$(desktopdir)/$(TARGET).desktop
+#	rm $(DESTDIR)$(mandir)/man1/chickennix.1
