*** extension.js	2011-05-15 21:31:00.000000000 +0200
--- extension.js	2011-05-27 09:51:28.989953860 +0200
***************
*** 21,33 ****
     // the icon must exist in the appropriate directory
     let logo = new St.Icon({ icon_type: St.IconType.FULLCOLOR, 
                              icon_size: hotCornerButton.height, 
!                             icon_name: 'fedora-logo-icon' });
  
     // comment out this line if you do not want an icon displayed
     box.add_actor(logo);
  
     // comment out this line if you do not want the label displayed
     box.add_actor(label);
  
     Main.panel.button.set_child(box);
  }
--- 21,33 ----
     // the icon must exist in the appropriate directory
     let logo = new St.Icon({ icon_type: St.IconType.FULLCOLOR, 
                              icon_size: hotCornerButton.height, 
!                             icon_name: 'distributor-logo' });
  
     // comment out this line if you do not want an icon displayed
     box.add_actor(logo);
  
     // comment out this line if you do not want the label displayed
     box.add_actor(label);
  
     Main.panel.button.set_child(box);
  }
