diff -u ../fpcbuild-2.6.2.orig/fpcsrc/Makefile fpcsrc/Makefile
--- ../fpcbuild-2.6.2.orig/fpcsrc/Makefile	2013-08-14 23:31:18.223956923 +0400
+++ fpcsrc/Makefile	2013-08-14 23:40:50.467274325 +0400
@@ -203,6 +203,14 @@
 ifeq ($(OS_TARGET),linux)
 linuxHier=1
 endif
+ifndef CROSSCOMPILE
+BUILDFULLNATIVE=1
+export BUILDFULLNATIVE
+endif
+ifdef BUILDFULLNATIVE
+BUILDNATIVE=1
+export BUILDNATIVE
+endif
 export OS_TARGET OS_SOURCE ARCH CPU_TARGET CPU_SOURCE FULL_TARGET FULL_SOURCE TARGETSUFFIX SOURCESUFFIX CROSSCOMPILE
 ifdef FPCDIR
 override FPCDIR:=$(subst \,/,$(FPCDIR))
@@ -396,9 +404,11 @@
 ifneq ($(wildcard utils),)
 NOUTILSTARGETS=embedded gba
 ifeq ($(findstring $(OS_TARGET),$(NOUTILSTARGETS)),)
+ifdef BUILDFULLNATIVE
 UTILS=1
 endif
 endif
+endif
 ifeq ($(FULL_TARGET),i386-linux)
 override TARGET_DIRS+=compiler rtl utils packages ide installer
 endif
diff -u ../fpcbuild-2.6.2.orig/fpcsrc/Makefile.fpc fpcsrc/Makefile.fpc
--- ../fpcbuild-2.6.2.orig/fpcsrc/Makefile.fpc	2013-08-14 23:31:18.183956922 +0400
+++ fpcsrc/Makefile.fpc	2013-08-14 23:39:19.330610192 +0400
@@ -160,9 +160,11 @@
 ifneq ($(wildcard utils),)
 NOUTILSTARGETS=embedded gba
 ifeq ($(findstring $(OS_TARGET),$(NOUTILSTARGETS)),)
+ifdef BUILDFULLNATIVE
 UTILS=1
 endif
 endif
+endif
