diff -rupN ariba-0.7.1/source/ariba/communication/networkinfo/NetworkInformation.h ariba-0.7.1.1/source/ariba/communication/networkinfo/NetworkInformation.h
--- ariba-0.7.1/source/ariba/communication/networkinfo/NetworkInformation.h	2009-10-10 17:12:59.000000000 +0200
+++ ariba-0.7.1.1/source/ariba/communication/networkinfo/NetworkInformation.h	2012-04-09 14:41:56.425583553 +0200
@@ -39,6 +39,7 @@
 #ifndef __NETWORK_INFORMATION_H
 #define __NETWORK_INFORMATION_H
 
+#include <unistd.h>
 #include <vector>
 #include <string>
 #include <algorithm>
diff -rupN ariba-0.7.1/source/ariba/utility/serialization/Data.hpp ariba-0.7.1.1/source/ariba/utility/serialization/Data.hpp
--- ariba-0.7.1/source/ariba/utility/serialization/Data.hpp	2011-04-07 13:44:18.000000000 +0200
+++ ariba-0.7.1.1/source/ariba/utility/serialization/Data.hpp	2012-04-09 14:42:17.785467835 +0200
@@ -41,6 +41,7 @@
 
 //== library includes ==
 #include <stdlib.h>
+#include <string.h>
 #include <iostream>
 #include <boost/cstdint.hpp>
 #include <boost/type_traits/make_unsigned.hpp>
