diff -Naur old/smplayer/src/basegui.cpp new/smplayer/src/basegui.cpp
--- old/smplayer/src/basegui.cpp	2011-01-25 18:16:30.000000000 +0200
+++ new/smplayer/src/basegui.cpp	2011-01-25 18:29:21.442435183 +0200
@@ -2894,7 +2894,7 @@
 		for (n=0; n < core->mdat.chapters; n++) {
 			QAction *a = new QAction(chapterGroup);
 			a->setCheckable(true);
-			a->setText( QString::number(n+1) );
+			a->setText(core->mdat.chapters_name[n]);
 			a->setData( n + Core::firstChapter() );
 		}
 	} else {
@@ -2918,7 +2918,7 @@
 			for (n=0; n < core->mdat.mkv_chapters; n++) {
 				QAction *a = new QAction(chapterGroup);
 				a->setCheckable(true);
-				a->setText( QString::number(n+1) );
+				a->setText(core->mdat.mkv_chapters_name[n]);
 				a->setData( n + Core::firstChapter() );
 			}
 		} else {
diff -Naur old/smplayer/src/mediadata.cpp new/smplayer/src/mediadata.cpp
--- old/smplayer/src/mediadata.cpp	2011-01-25 18:16:30.000000000 +0200
+++ new/smplayer/src/mediadata.cpp	2011-01-25 18:31:42.311989276 +0200
@@ -51,11 +51,13 @@
 
 #if GENERIC_CHAPTER_SUPPORT
 	chapters = 0;
+	chapters_name.clear();
 #else
 	//chapters=0;
 	//angles=0;
 
 	mkv_chapters=0;
+	mkv_chapters_name.clear();
 #endif
 
 	initialized=false;
diff -Naur old/smplayer/src/mediadata.h new/smplayer/src/mediadata.h
--- old/smplayer/src/mediadata.h	2011-01-25 18:16:30.000000000 +0200
+++ new/smplayer/src/mediadata.h	2011-01-25 18:27:16.182462317 +0200
@@ -28,7 +28,7 @@
 
 #include <QString>
 #include <QSettings>
-
+#include <QMap>
 
 // Types of media
 
@@ -76,11 +76,13 @@
 
 #if GENERIC_CHAPTER_SUPPORT
 	int chapters;
+	QMap<int,QString> chapters_name;
 #else
 	//int chapters, angles; // for DVDs
 
 	// Matroshka chapters
 	int mkv_chapters;
+	QMap<int,QString> mkv_chapters_name;
 #endif
 
 	// Clip info
diff -Naur old/smplayer/src/mplayerprocess.cpp new/smplayer/src/mplayerprocess.cpp
--- old/smplayer/src/mplayerprocess.cpp	2011-01-25 18:16:30.000000000 +0200
+++ new/smplayer/src/mplayerprocess.cpp	2011-01-25 18:25:13.595751066 +0200
@@ -112,6 +112,7 @@
 static QRegExp rx_screenshot("^\\*\\*\\* screenshot '(.*)'");
 static QRegExp rx_endoffile("^Exiting... \\(End of file\\)|^ID_EXIT=EOF");
 static QRegExp rx_mkvchapters("\\[mkv\\] Chapter (\\d+) from");
+static QRegExp rx_mkvchapters_name("^ID_CHAPTER_(\\d+)_NAME=(.*)");
 static QRegExp rx_aspect2("^Movie-Aspect is ([0-9,.]+):1");
 static QRegExp rx_fontcache("^\\[ass\\] Updating font cache|^\\[ass\\] Init");
 static QRegExp rx_scanning_font("Scanning file");
@@ -504,6 +505,20 @@
 		}
 		else
 
+		if (rx_mkvchapters_name.indexIn(line)!=-1) {
+            int id = rx_mkvchapters_name.cap(1).toInt();
+            QString s = rx_mkvchapters_name.cap(2);
+            qDebug("MplayerProcess::parseLine: mkv chapters: %d", id);
+            qDebug("MplayerProcess::parseLine: mkv chapters name: %s", s.toUtf8().data());
+#if GENERIC_CHAPTER_SUPPORT
+            md.chapters_name.insert(id,s);
+#else
+            md.mkv_chapters_name.insert(id,s);
+#endif
+        }
+
+		else
+
 		// VCD titles
 		if (rx_vcd.indexIn(line) > -1 ) {
 			int ID = rx_vcd.cap(1).toInt();
