--- vpnclient_init.orig	2011-09-20 15:22:52.901415716 -0400
+++ vpnclient_init	2011-09-20 15:28:15.013003963 -0400
@@ -16,14 +16,7 @@
 VPNCLIENT="/opt/cisco-vpnclient/bin/vpnclient"
 VPNDEV="cipsec0"
 VPNMOD=cisco_ipsec
-case `uname -r` in
-2.[56].*)
-    VPNMOD_FILE="${VPNMOD}.ko"
-    ;;
-*)
-    VPNMOD_FILE="$VPNMOD"
-    ;;
-esac    
+VPNMOD_FILE="${VPNMOD}.ko"
 
 WHOAMI=`id | sed -e 's/(.*//'`
 
@@ -35,7 +28,7 @@
 		echo "Failed (super user access required)"
 		exit 1
 	fi
-	/sbin/lsmod | grep -q "${VPNMOD}"
+	/bin/lsmod | grep -q "${VPNMOD}"
         if [ "$?" = "0" ] ; then 
 		echo "module ${VPNMOD} is already running. Use restart instead."
 		exit 1
@@ -82,9 +75,6 @@
 		fi
 		;;
 	*)
-		echo "Failed (unsupported Linux version)"
-		/sbin/rmmod ${VPNMOD}
-		exit 1
 		;;
 	esac
 	
@@ -98,7 +88,7 @@
 	fi
     killall cvpnd > /dev/null 2>&1
 
-	/sbin/lsmod | grep -q "${VPNMOD}"
+	/bin/lsmod | grep -q "${VPNMOD}"
     if [ "$?" != "0" ] ; then 
 		echo "module ${VPNMOD} is not running."
 		exit 1
@@ -116,8 +106,8 @@
 	echo "Done"
 	;;
   status)
-	/sbin/lsmod | egrep 'Module'
-	/sbin/lsmod | egrep "${VPNMOD}"
+	/bin/lsmod | egrep 'Module'
+	/bin/lsmod | egrep "${VPNMOD}"
 	if [ "$?" != "0" ] ; then
 			echo
 			echo "Status Failed (lsmod ${VPNMOD}) - The VPN module is not loaded."
