--- a/lib/swiotlb.c.orig	2011-01-05 02:50:19.000000000 +0200
+++ b/lib/swiotlb.c	2011-02-02 09:52:57.246672523 +0200
@@ -686,8 +686,10 @@
 	/*
 	 * Ensure that the address returned is DMA'ble
 	 */
-	if (!dma_capable(dev, dev_addr, size))
-		panic("map_single: bounce buffer is not DMA'ble");
+        if (!dma_capable(dev, dev_addr, size)) {
+		swiotlb_tbl_unmap_single(dev, map, size, dir);
+		dev_addr = swiotlb_virt_to_bus(dev, io_tlb_overflow_buffer);
+}
 
 	return dev_addr;
 }
