diff -Naur Kognite~/build-linux Kognite/build-linux
--- Kognite~/build-linux	2013-01-12 00:11:25.903239915 -0600
+++ Kognite/build-linux	2013-01-12 00:11:40.276286137 -0600
@@ -1,6 +1,7 @@
 #!/bin/bash
 
-qmake -project -norecursive "INCPATH+=-I/usr/include /usr/include/poppler/qt4
+qmake -project -norecursive "INCLUDEPATH+=-I/usr/include /usr/include/poppler/qt4
+CONFIG+=no_keywords
 LIBS+=-L/usr/lib -lxapian -lpoppler-qt4 -lz"
-qmake 
-make
\ No newline at end of file
+qmake
+make
diff -Naur Kognite~/header.h Kognite/header.h
--- Kognite~/header.h	2013-01-12 00:11:25.906573182 -0600
+++ Kognite/header.h	2013-01-12 00:11:31.359797681 -0600
@@ -29,11 +29,11 @@
 	{
 		Q_OBJECT
 		
-		private slots:
+		private Q_SLOTS:
 		
 		void dropEvent(QDropEvent *);
 		
-	signals:
+	Q_SIGNALS:
 		
 		void fileMoveRequested(QString, QPoint);
 		void directoryMoveRequested(QString, QPoint);
@@ -44,7 +44,7 @@
 	{
 		Q_OBJECT
 		
-		public slots:
+		public Q_SLOTS:
 		
 		void processTextChange();
 		void setCurrentFileName(const QString&);
diff -Naur Kognite~/window.cpp Kognite/window.cpp
--- Kognite~/window.cpp	2013-01-12 00:11:25.906573182 -0600
+++ Kognite/window.cpp	2013-01-12 00:11:31.359797681 -0600
@@ -1067,7 +1067,7 @@
 		
 		if (dirStructure.isDir(dirStructure.index(event->mimeData()->urls()[0].path())))
 		{
-			emit directoryMoveRequested(event->mimeData()->urls()[0].path(), event->pos());
+			Q_EMIT directoryMoveRequested(event->mimeData()->urls()[0].path(), event->pos());
 			setState(QAbstractItemView::NoState);
 			event->accept();
 			return;
@@ -1075,7 +1075,7 @@
 		
 		else
 		{
-			emit fileMoveRequested(event->mimeData()->urls()[0].path(), event->pos());
+			Q_EMIT fileMoveRequested(event->mimeData()->urls()[0].path(), event->pos());
 			setState(QAbstractItemView::NoState);
 			event->accept();
 			return;
