diff -Naur rsbac-admin-1.4.0~/Makefile rsbac-admin-1.4.0/Makefile
--- rsbac-admin-1.4.0~/Makefile	2009-01-15 10:52:53.000000000 -0600
+++ rsbac-admin-1.4.0/Makefile	2009-01-15 10:53:51.000000000 -0600
@@ -49,7 +49,7 @@
 build: libs pam rklogd tools nss
 	
 install: headers-install libs-install pam-install rklogd-install \
-tools-install nss-install
+tools-install nss-install scripts-install
 
 install-strip: headers-install libs-install pam-install rklogd-install-strip \
 tools-install-strip nss-install
@@ -124,6 +124,11 @@
 	$(call E, "Installing RSBAC tools \(stripped\)...")
 	@$(MAKE) install-strip -C main/tools
 
+scripts-install:
+	$(call E, "Installing RSBAC scripts...")
+	@install -d -m755 $(DESTDIR)/etc
+	@install -c -m644 {config,functions}.rsbac $(DESTDIR)/etc
+
 #
 # Uninstallations
 #
diff -Naur rsbac-admin-1.4.0~/config.rsbac rsbac-admin-1.4.0/config.rsbac
--- rsbac-admin-1.4.0~/config.rsbac	1969-12-31 18:00:00.000000000 -0600
+++ rsbac-admin-1.4.0/config.rsbac	2009-01-15 10:53:01.000000000 -0600
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+. /etc/rc.conf
+. /etc/rc.d/functions
+. /secoff/.rsbac/functions
+
+case "$1" in
+  auth)
+    stat_busy "Configuring AUTH module"
+    config_auth &> /dev/null
+    [ $? -gt 0 ] && stat fail || stat_done
+    ;;
+  cap)
+    stat_busy "Configuring CAP module"
+    config_cap &> /dev/null
+    [ $? -gt 0 ] && stat fail || stat_done
+    ;;
+  daz)
+    stat_busy "Configuring DAZ module"
+    config_daz &> /dev/null
+    [ $? -gt 0 ] && stat fail || stat_done
+    ;;
+  ff)
+    stat_busy "Configuring FF module"
+    config_ff &> /dev/null
+    [ $? -gt 0 ] && stat fail || stat_done
+    ;;
+  jail)
+    stat_busy "Configuring JAIL module"
+    config_jail &> /dev/null
+    [ $? -gt 0 ] && stat fail || stat_done
+    ;;
+  rsbac)
+    echo
+    printhl "This script will only configure the most basic settings."
+    printhl "It is NOT sufficient for a fully usable, hardened system."
+    printsep
+    echo
+    $0 auth
+    $0 cap
+    $0 daz
+    $0 ff
+    $0 jail
+    ;;
+  *)
+    echo "usage: $0 {auth:cap:daz:ff:jail:rsbac}"
+esac
+exit 0
diff -Naur rsbac-admin-1.4.0~/functions.rsbac rsbac-admin-1.4.0/functions.rsbac
--- rsbac-admin-1.4.0~/functions.rsbac	1969-12-31 18:00:00.000000000 -0600
+++ rsbac-admin-1.4.0/functions.rsbac	2009-01-15 10:53:01.000000000 -0600
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+SECOFF=secoff
+ADMIN=root
+E_STAT=0
+
+config_auth() {
+  attr_set_user $SECOFF auth_role 1 || E_STAT=1
+  attr_set_user $ADMIN auth_role 2 || E_STAT=1
+  attr_set_fd AUTH FD auth_may_setuid 1 /bin/login || E_STAT=1
+  attr_set_fd AUTH FD auth_may_setuid 1 /bin/su || E_STAT=1
+  if [ -n "`pacman -Q sudo 2> /dev/null`" ]; then
+    attr_set_fd AUTH FD auth_may_setuid 1 /usr/bin/sudo || E_STAT=1
+  fi
+  return $E_STAT
+}
+
+config_cap() {
+  attr_set_user $SECOFF cap_role 1 || E_STAT=1
+  attr_set_user $ADMIN cap_role 2 || E_STAT=1
+  return $E_STAT
+}
+
+config_daz() {
+  attr_set_user $SECOFF daz_role 1 || E_STAT=1
+  attr_set_user $ADMIN daz_role 2 || E_STAT=1
+  return $E_STAT
+}
+
+config_ff() {
+  attr_set_user $SECOFF ff_role 1 || E_STAT=1
+  attr_set_user $ADMIN ff_role 2 || E_STAT=1
+  return $E_STAT
+}
+
+config_jail() {
+  attr_set_user $SECOFF jail_role 1 || E_STAT=1
+  attr_set_user $ADMIN jail_role 2 || E_STAT=1
+  return $E_STAT
+}
diff -Naur rsbac-admin-1.4.0~/main/headers/Makefile rsbac-admin-1.4.0/main/headers/Makefile
--- rsbac-admin-1.4.0~/main/headers/Makefile	2009-01-15 10:52:54.000000000 -0600
+++ rsbac-admin-1.4.0/main/headers/Makefile	2009-01-15 10:52:57.000000000 -0600
@@ -13,7 +13,7 @@
 endif
 
 DESTDIR		:= 
-PREFIX		:= /usr/local
+PREFIX		:= /usr
 DIR_INC		:= $(PREFIX)/include/rsbac
 
 
diff -Naur rsbac-admin-1.4.0~/main/libs/Makefile rsbac-admin-1.4.0/main/libs/Makefile
--- rsbac-admin-1.4.0~/main/libs/Makefile	2009-01-15 10:52:53.000000000 -0600
+++ rsbac-admin-1.4.0/main/libs/Makefile	2009-01-15 10:52:57.000000000 -0600
@@ -19,10 +19,10 @@
 HOST		:= $(shell uname -m)
 
 DESTDIR		:= 
-PREFIX		:= /usr/local
+PREFIX		:= /usr
 ifeq ($(DIR_LIBS),)
 ifeq ($(HOST), x86_64)
-LIBDIR		:= $(PREFIX)/lib64
+LIBDIR		:= $(PREFIX)/lib
 else
 LIBDIR		:= $(PREFIX)/lib
 endif
diff -Naur rsbac-admin-1.4.0~/main/nss/Makefile rsbac-admin-1.4.0/main/nss/Makefile
--- rsbac-admin-1.4.0~/main/nss/Makefile	2009-01-15 10:52:54.000000000 -0600
+++ rsbac-admin-1.4.0/main/nss/Makefile	2009-01-15 10:52:57.000000000 -0600
@@ -19,7 +19,7 @@
 endif
 
 DESTDIR		:= 
-PREFIX		:= /usr/local
+PREFIX		:= /usr
 ifeq ($(DIR_NSS),)
 LIBDIR		:= $(PREFIX)/lib
 else
diff -Naur rsbac-admin-1.4.0~/main/pam/Makefile rsbac-admin-1.4.0/main/pam/Makefile
--- rsbac-admin-1.4.0~/main/pam/Makefile	2009-01-15 10:52:54.000000000 -0600
+++ rsbac-admin-1.4.0/main/pam/Makefile	2009-01-15 10:52:57.000000000 -0600
@@ -19,9 +19,9 @@
 HOST		:= $(shell uname -m)
 
 DESTDIR		:= 
-PREFIX		:= /usr/local
+PREFIX		:= /usr
 ifeq ($(HOST), x86_64)
-LIBDIR		:= /lib64
+LIBDIR		:= /lib
 else
 LIBDIR		:= /lib
 endif
diff -Naur rsbac-admin-1.4.0~/main/rklogd/Makefile rsbac-admin-1.4.0/main/rklogd/Makefile
--- rsbac-admin-1.4.0~/main/rklogd/Makefile	2009-01-15 10:52:54.000000000 -0600
+++ rsbac-admin-1.4.0/main/rklogd/Makefile	2009-01-15 10:52:57.000000000 -0600
@@ -18,7 +18,7 @@
 endif
 
 DESTDIR		:= 
-PREFIX		:= /usr/local
+PREFIX		:= /usr
 LOCALEDIR	:= $(PREFIX)/share/locale
 DIR_BIN		:= $(PREFIX)/bin
 DIR_SBIN	:= $(PREFIX)/sbin
diff -Naur rsbac-admin-1.4.0~/main/tools/Makefile rsbac-admin-1.4.0/main/tools/Makefile
--- rsbac-admin-1.4.0~/main/tools/Makefile	2009-01-15 10:52:54.000000000 -0600
+++ rsbac-admin-1.4.0/main/tools/Makefile	2009-01-15 10:52:57.000000000 -0600
@@ -21,10 +21,10 @@
 endif
 
 DESTDIR		:= 
-PREFIX		:= /usr/local
+PREFIX		:= /usr
 LOCALEDIR	:= $(PREFIX)/share/locale
 DIR_PO		:= de ru fr
-DIR_DOC		:= $(PREFIX)/share/doc/$(PACKAGE)-$(VERSION)
+DIR_DOC		:= $(PREFIX)/share/doc/rsbac-admin
 DIR_MAN		:= $(PREFIX)/share/man/man1
 DIR_BIN		:= $(PREFIX)/bin
 DIR_RBIN	:= /bin
@@ -115,8 +115,6 @@
 	$(GZIP) -9c $(f) > $(f:.1=.1.gz);)
 	$(foreach f, $(FILES_MAN:.1=.1.gz), $(ECHO) -e "  INSTALL\t$(f)"; \
 	$(INSTALL) -m 644 $(f) $(DESTDIR)/$(DIR_MAN);)
-	$(E) "INSTALL\t\tAUTHORS INSTALL README COPYING Changes"
-	$(INSTALL) -m644 AUTHORS INSTALL README COPYING Changes $(DESTDIR)/$(DIR_DOC)
 	$(E) "INSTALL\t\t $(DIR_EXAMPLE)"
 	$(CP) -r $(DIR_EXAMPLE) $(DESTDIR)/$(DIR_DOC)
 
diff -Naur rsbac-admin-1.4.0~/main/tools/examples/reg/Makefile rsbac-admin-1.4.0/main/tools/examples/reg/Makefile
--- rsbac-admin-1.4.0~/main/tools/examples/reg/Makefile	2009-01-15 10:52:54.000000000 -0600
+++ rsbac-admin-1.4.0/main/tools/examples/reg/Makefile	2009-01-15 10:52:57.000000000 -0600
@@ -11,7 +11,7 @@
 #MODVERSIONS=1
 
 DESTDIR		:=
-PREFIX		:= $(DESTDIR)/usr/local
+PREFIX		:= $(DESTDIR)/usr
 
 # comment out, if RSBAC kernel has no PROC support
 CONFIG_RSBAC_PROC := 1
