--- a/qucs-git/qucs-core/adms/autogen_lin.sh	2013-04-10 21:26:28.657036096 +0200
+++ b/qucs-git/qucs-core/adms/autogen_lin.sh	2013-04-10 21:26:37.200369969 +0200
@@ -51,7 +51,7 @@
 #
 # run configure
 #
-CMD="./configure --enable-maintainer-mode"
+CMD="./configure --enable-maintainer-mode --prefix=/usr"
 echo "Running $CMD $@ ..."
 $CMD "$@"
 #
@@ -60,7 +60,3 @@
 echo "Running make clean"
 make clean
 
-echo
-echo "Continue with:"
-echo "make"
-echo "make install"
