--- bootstrap.sh	2014-06-07 02:29:46.505020592 +0200
+++ bootstrap.sh	2014-06-07 02:31:09.817640264 +0200
@@ -35,7 +35,6 @@ if [ -d "./deps/adms" ]; then
     ./deps/adms/autogen.sh "$@";
   else
     echo "Could not locate adms autogen script in ./adms, you may use configure with --disable-adms to use installed version"
-    exit
   fi
 else
   echo "No local adms source folder found (you may need to use the configure --diable-adms option to build with installed version)"
