diff -u a/ropevim.vim b/ropevim.vim
--- a/ropevim.vim	2012-08-09 14:49:56.632247167 -0700
+++ b/ropevim.vim	2012-08-09 14:50:13.295610618 -0700
@@ -1,5 +1,5 @@
 function! LoadRope()
-python << EOF
+python3 << EOF
 import ropevim
 EOF
 endfunction
diff -u a/setup.py b/setup.py
--- a/setup.py	2012-08-09 13:28:18.268912317 -0700
+++ b/setup.py	2012-08-09 13:29:01.525622838 -0700
@@ -36,6 +36,6 @@
       license='GNU GPL',
       classifiers=classifiers,
       requires=['ropemode'],
-      data_files=[('share/vim/plugin', ['ropevim.vim'])],
+      data_files=[('share/vim/vimfiles/plugin', ['ropevim.vim'])],
       **extra_kwargs
 )
diff -u a/ropevim.py b/ropevim.py
--- a/ropevim.py	2012-08-09 14:57:41.873040691 -0700
+++ b/ropevim.py	2012-08-09 14:58:31.286447402 -0700
@@ -270,7 +270,7 @@
         globals()[name] = callback
         arg = 'None' if prefix else ''
         vim.command('function! %s()\n' % _vim_name(name) +
-                    'python ropevim.%s(%s)\n' % (name, arg) +
+                    'python3 ropevim.%s(%s)\n' % (name, arg) +
                     'endfunction\n')
 
     def _completion_data(self, proposal):
@@ -377,10 +377,10 @@
 
     def __init__(self):
         self.values = []
-        vim.command('python import vim')
+        vim.command('python3 import vim')
         vim.command('function! RopeValueCompleter(A, L, P)\n'
-                    'python args = [vim.eval("a:" + p) for p in "ALP"]\n'
-                    'python ropevim._completer(*args)\n'
+                    'python3 args = [vim.eval("a:" + p) for p in "ALP"]\n'
+                    'python3 ropevim._completer(*args)\n'
                     'return s:completions\n'
                     'endfunction\n')
 
