--- activestate.py	2013-08-15 00:21:03.000000000 +0200
+++ activestate.py.new	2013-11-13 12:11:35.067199371 +0100
@@ -182,7 +182,7 @@
	     os.chmod(path, perm) # restore permissions
 
 
-def relocate_python(install_prefix, verbose=False):
+def relocate_python(install_prefix, dest_prefix=None, verbose=False):
     """Relocate this Python installation.
     
     "Relocation" involves updating hardcoded shebang lines in Python scripts
@@ -210,16 +210,19 @@
                            % (install_prefix, len(install_prefix),
                                shortest_original_prefix_length))
 
-    log("relocate this Python to '%s'" % install_prefix)
+    if dest_prefix is None:
+        dest_prefix = install_prefix
+
+    log("relocate this Python to '%s'" % dest_prefix)
     for prefix in prefixes:
-        if prefix == install_prefix:
+        if prefix == dest_prefix:
             continue
         for dirpath, dirnames, filenames in os.walk(install_prefix):
             for filename in filenames:
                 if splitext(filename)[1] in (".pyo", ".pyc"):
                     continue
                 _relocate_path(join(dirpath, filename),
-                               prefix, install_prefix, log)
+                               prefix, dest_prefix, log)
 
 
 #---- mainline
