reverted:
--- b/kwin/effects/presentwindows/presentwindows.h
+++ a/kwin/effects/presentwindows/presentwindows.h
@@ -137,7 +137,8 @@
         WindowExitAction = 2, // Deactivates the effect without activating new window
         WindowToCurrentDesktopAction = 3, // Brings window to current desktop
         WindowToAllDesktopsAction = 4, // Brings window to all desktops
+        WindowMinimizeAction = 5, // Minimize the window
+        WindowCloseAction = 6 // Closes the window
-        WindowMinimizeAction = 5 // Minimize the window
     };
     enum DesktopMouseAction {
         DesktopNoAction = 0, // nothing
reverted:
--- b/kwin/effects/presentwindows/presentwindows.cpp
+++ a/kwin/effects/presentwindows/presentwindows.cpp
@@ -670,6 +670,11 @@
                 m_highlightedWindow->minimize();
         }
         break;
+    case WindowCloseAction:
+        if (m_highlightedWindow) {
+            m_highlightedWindow->closeWindow();
+        }
+        break;
     default:
         break;
     }
reverted:
--- b/kwin/effects/presentwindows/presentwindows_config.ui
+++ a/kwin/effects/presentwindows/presentwindows_config.ui
@@ -161,6 +161,11 @@
           <string>(Un-)Minimize window</string>
          </property>
         </item>
+        <item>
+         <property name="text">
+          <string>Close window</string>
+         </property>
+        </item>
        </widget>
       </item>
       <item row="1" column="0">
@@ -205,6 +210,11 @@
           <string>(Un-)Minimize window</string>
          </property>
         </item>
+        <item>
+         <property name="text">
+          <string>Close window</string>
+         </property>
+        </item>
        </widget>
       </item>
       <item row="2" column="0">
@@ -249,6 +259,11 @@
           <string>(Un-)Minimize window</string>
          </property>
         </item>
+        <item>
+         <property name="text">
+          <string>Close window</string>
+         </property>
+        </item>
        </widget>
       </item>
      </layout>