diff -Naur a/src/fwbackups/cron.py b/src/fwbackups/cron.py
--- a/src/fwbackups/cron.py	2010-12-20 09:06:44.000000000 +0100
+++ b/src/fwbackups/cron.py	2012-11-20 21:07:31.122778814 +0100
@@ -168,8 +168,8 @@
   of 0 or 1 to indicate a failure writing to the crontab or a success
   respectively."""
   if LINUX:
-    environ = {'EDITOR': 'python %s/cronwriter.py' % INSTALL_DIR,
-               'VISUAL': 'python %s/cronwriter.py' % INSTALL_DIR}
+    environ = {'EDITOR': 'python2 %s/cronwriter.py' % INSTALL_DIR,
+               'VISUAL': 'python2 %s/cronwriter.py' % INSTALL_DIR}
   elif DARWIN:
     environ = {'EDITOR': '/fwbackups-cronwriter.py',
                'VISUAL': '/fwbackups-cronwriter.py'}
