*** config.h	2011-07-03 01:53:34.000000000 -0700
--- config.h.new	2011-07-28 12:53:21.000000000 -0700
***************
*** 35,42 ****
  
  /* Default paths to runtime files: */
  
! #define ASSETS_DIR              "assets"
! #define DEF_WORDLIST            "skipfish.wl"
  
  /* Various default settings for HTTP client (cmdline override): */
  
--- 35,42 ----
  
  /* Default paths to runtime files: */
  
! #define ASSETS_DIR              "/usr/share/skipfish/assets"
! #define DEF_WORDLIST            "/usr/share/skipfish/skipfish.wl"
  
  /* Various default settings for HTTP client (cmdline override): */
  
