diff -ru lzfs-build.old/etc/Makefile.in lzfs-build/etc/Makefile.in
--- lzfs-build.old/etc/Makefile.in	2011-03-04 22:55:51.133653776 +0300
+++ lzfs-build/etc/Makefile.in	2011-03-04 22:56:15.313587004 +0300
@@ -2,7 +2,7 @@
 etcdir = /etc
 initddir = $(etcdir)/init.d
 zfsdir = $(etcdir)/zfs
-INSTALL = /usr/bin/install -c
+INSTALL = /bin/install -c
 
 all: 
 install: install_initd install_zfs
diff -ru lzfs-build.old/module/Makefile.in lzfs-build/module/Makefile.in
--- lzfs-build.old/module/Makefile.in	2011-03-04 22:55:51.180320333 +0300
+++ lzfs-build/module/Makefile.in	2011-03-04 22:56:30.033536622 +0300
@@ -17,7 +17,7 @@
 lzfs-objs += lzfs_xattr_acl.o
 
 
-INSTALL=/usr/bin/install
+INSTALL=/bin/install
 
 
 modules:
diff -ru lzfs-build.old/usr/Makefile.in lzfs-build/usr/Makefile.in
--- lzfs-build.old/usr/Makefile.in	2011-03-04 22:55:51.190320310 +0300
+++ lzfs-build/usr/Makefile.in	2011-03-04 22:56:42.633488039 +0300
@@ -1,6 +1,6 @@
 usrdir = /usr
 sbindir = $(usrdir)/sbin
-INSTALL = /usr/bin/install -c
+INSTALL = /bin/install -c
 
 all:
 
