diff -Naur gloobus-preview-bzr.orig/configure.ac gloobus-preview-bzr/configure.ac
--- gloobus-preview-bzr.orig/configure.ac	2012-05-13 16:52:57.000000000 +0200
+++ gloobus-preview-bzr/configure.ac	2012-05-13 16:55:38.133223954 +0200
@@ -16,6 +16,7 @@
 
 # *** checks for programs **************************************************** #
 AC_PROG_CXX             # Check g++
+m4_ifdef([AM_PROG_AR], [AM_PROG_AR]) # required by automake 1.12
 AC_PROG_LIBTOOL         # To create libraries
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET        # Compile more than one Makefiles
