diff -Naur gloobus-preview.orig/configure.ac gloobus-preview/configure.ac
--- gloobus-preview.orig/configure.ac	2011-05-10 23:21:26.070604000 +0200
+++ gloobus-preview/configure.ac	2011-05-10 23:24:13.113429460 +0200
@@ -48,12 +48,6 @@
 PKG_CHECK_MODULES(GLIB, glib-2.0)
 PKG_CHECK_MODULES(DBUS, dbus-glib-1)
 
-PKG_CHECK_MODULES(GMODULE, gmodule-2.0)											# To compile XCF loader
-PKG_CHECK_MODULES(GDKPIXBUF, gdk-pixbuf-2.0)									# To compile XCF loader
-	
-AC_CHECK_HEADER(bzlib.h,,AC_MSG_ERROR(Can not find bzlib header))				# To compile XCF loader
-AC_CHECK_LIB(bz2,BZ2_bzDecompressInit,,AC_MSG_ERROR(Can not find libbz2))		# To compile XCF loader ==> sudo apt-get install libtool libbz2-dev
-
 # Checks for typedefs, structures, and compiler characteristics.
 AC_HEADER_STDBOOL
 
@@ -84,8 +78,6 @@
 	src/plugin-ps/Makefile
 	src/plugin-text/Makefile
 	src/plugin-ttf/Makefile
-	src/loaders/PSD/Makefile
-	src/loaders/XCF/Makefile
    	data/Makefile
    	data/images/Makefile
    	data/gtkrc/Makefile
diff -Naur gloobus-preview.orig/src/Makefile.am gloobus-preview/src/Makefile.am
--- gloobus-preview.orig/src/Makefile.am	2011-05-10 23:21:26.070604000 +0200
+++ gloobus-preview/src/Makefile.am	2011-05-10 23:26:02.260096766 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = loaders/PSD loaders/XCF plugin-comic plugin-folder plugin-compressed plugin-djvu plugin-icns plugin-gstreamer-audio plugin-gstreamer-video plugin-openoffice plugin-pdf plugin-pixbuf plugin-ps plugin-text plugin-ttf .
+SUBDIRS = plugin-comic plugin-folder plugin-compressed plugin-djvu plugin-icns plugin-gstreamer-audio plugin-gstreamer-video plugin-openoffice plugin-pdf plugin-pixbuf plugin-ps plugin-text plugin-ttf .
 
 #CXXFLAGS = -Werror -g				#If -O2  iDocument threads don't work
 AM_CXXFLAGS = -Wall -Werror #-g -O0
