diff -ur meshlab.orig/src/fgt/filter_trioptimize/curvedgeflip.h meshlab/src/fgt/filter_trioptimize/curvedgeflip.h
--- meshlab.orig/src/fgt/filter_trioptimize/curvedgeflip.h	2011-01-25 18:49:26.000000000 +1100
+++ meshlab/src/fgt/filter_trioptimize/curvedgeflip.h	2012-04-21 16:34:08.130214602 +1000
@@ -326,7 +326,7 @@
 				for (unsigned int i = 0; i < 3; i++)
 					if ((*fi).V1(i) - (*fi).V0(i) > 0) {
 						PosType newpos(&*fi, i);
-            Insert(heap, newpos, tri::IMark(m));
+						CurvEdgeFlip::Insert(heap, newpos, tri::IMark(m));
 					}
 	}
 }; // end CurvEdgeFlip class
diff -ur meshlab.orig/src/meshlabplugins/decorate_base/colorhistogram.h meshlab/src/meshlabplugins/decorate_base/colorhistogram.h
--- meshlab.orig/src/meshlabplugins/decorate_base/colorhistogram.h	2011-01-25 18:50:01.000000000 +1100
+++ meshlab/src/meshlabplugins/decorate_base/colorhistogram.h	2012-04-21 16:28:01.547218426 +1000
@@ -27,7 +27,7 @@
    */
   void Add(ScalarType v,Color4b c,float increment);
 
-  Color4b BinColorAvg(ScalarType v) { return BinColorAvgInd(BinIndex(v)); }
+  Color4b BinColorAvg(ScalarType v) { return BinColorAvgInd(this->BinIndex(v)); }
 
   Color4b BinColorAvgInd(int index) {
     return Color4b( (unsigned char)((CV[index][0] / float(this->H[index]))),
@@ -40,7 +40,7 @@
 
   //! Reset histogram data.
   void Clear() {
-    this->::Clear();
+    this->Clear();
     CV.clear();
   }
   /*
@@ -82,7 +82,7 @@
 template <class ScalarType>
 void ColorHistogram<ScalarType>::Add(ScalarType v,Color4b c,float increment=1.0f)
 {
-  int pos=BinIndex(v);
+  int pos=this->BinIndex(v);
   if(v<this->minElem) this->minElem=v;
   if(v>this->maxElem) this->maxElem=v;
   if(pos>=0 && pos<=this->n)
diff -ur meshlab.orig/src/meshlabplugins/filter_isoparametrization/param_collapse.h meshlab/src/meshlabplugins/filter_isoparametrization/param_collapse.h
--- meshlab.orig/src/meshlabplugins/filter_isoparametrization/param_collapse.h	2011-01-25 18:50:05.000000000 +1100
+++ meshlab/src/meshlabplugins/filter_isoparametrization/param_collapse.h	2012-04-21 16:08:01.428881823 +1000
@@ -60,7 +60,7 @@
   }
 
 	inline bool IsFeasible(){
-    return LinkConditions(Super::pos);
+    return this->LinkConditions(Super::pos);
 	}
 
   inline void SetHlevMeshUV(const std::vector<BaseFace*> &LowFace,
@@ -225,7 +225,7 @@
 	for (unsigned int i=0;i<created.face.size();i++)
 		domain.face[i].areadelta=created.face[i].areadelta;
 
-	DoCollapse(created,posEdge, newPos); // v0 is deleted and v1 take the new position
+	this->DoCollapse(created,posEdge, newPos); // v0 is deleted and v1 take the new position
 	
 	UpdateTopologies<BaseMesh>(&created);
 
@@ -617,7 +617,7 @@
 		ScalarType area0=Area<BaseFace>(orderedFaces0);
 
 		///do the collapse
-		DoCollapse(m, this->pos, newPos); // v0 is deleted and v1 take the new position
+		this->DoCollapse(m, this->pos, newPos); // v0 is deleted and v1 take the new position
 		//vcg::tri::UpdateTopology<BaseMesh>::TestVertexFace(m); ///TEST
 		//---------------------------///
 		///create a parametrized submesh post-collapse #1
diff -ur meshlab.orig/src/meshlabplugins/filter_meshing/quadric_tex_simp.h meshlab/src/meshlabplugins/filter_meshing/quadric_tex_simp.h
--- meshlab.orig/src/meshlabplugins/filter_meshing/quadric_tex_simp.h	2011-01-25 18:50:06.000000000 +1100
+++ meshlab/src/meshlabplugins/filter_meshing/quadric_tex_simp.h	2012-04-21 16:16:42.580712135 +1000
@@ -424,7 +424,7 @@
 			}
 
 
-			this->_priority = max(priority1, priority2)*(1 + (Params().ExtraTCoordWeight)*(QH::Vd(this->pos.V(0)).size()+QH::Vd(this->pos.V(1)).size() - 2));
+			this->_priority = std::max(priority1, priority2)*(1 + (Params().ExtraTCoordWeight)*(QH::Vd(this->pos.V(0)).size()+QH::Vd(this->pos.V(1)).size() - 2));
 		
 			return this->_priority;
 		}
@@ -668,7 +668,7 @@
 												constraint has been imposed during the re-computation of the other minimal */
 
 		
-	 DoCollapse(m, this->pos, newPos ); // v0 is deleted and v1 take the new position	 
+	 this->DoCollapse(m, this->pos, newPos ); // v0 is deleted and v1 take the new position	 
 
 	vcg::TexCoord2f newtcoord;
 	Quadric5<double> newq;
diff -ur meshlab.orig/src/meshlabplugins/filter_plymc/tri_edge_collapse_mc.h meshlab/src/meshlabplugins/filter_plymc/tri_edge_collapse_mc.h
--- meshlab.orig/src/meshlabplugins/filter_plymc/tri_edge_collapse_mc.h	2011-01-25 18:50:04.000000000 +1100
+++ meshlab/src/meshlabplugins/filter_plymc/tri_edge_collapse_mc.h	2012-04-21 16:18:45.524050507 +1000
@@ -98,7 +98,7 @@
 //        assert(	(p0[0]==p1[0]) ||
 //                        (p0[1]==p1[1]) ||
 //                        (p0[2]==p1[2]) );
-      DoCollapse(m, this->pos, MidPoint);
+      this->DoCollapse(m, this->pos, MidPoint);
 	}
 						
 						
