diff -Nur meshlab.orig/src/meshlab_full.pro meshlab/src/meshlab_full.pro
--- meshlab.orig/src/meshlab_full.pro	2011-02-17 03:17:41.000000000 +1100
+++ meshlab/src/meshlab_full.pro	2011-02-19 11:36:57.737776916 +1100
@@ -63,7 +63,7 @@
 # projects
                 fgt/filter_aging \
                 fgt/filter_colorproc \
-#			fgt/filter_csg \
+			fgt/filter_csg \
                 fgt/filter_func \
                 fgt/filter_ssynth \
                 fgt/filter_qhull \
@@ -74,4 +74,4 @@
                 fgt/editpaint\
                 fgt/edit_quality \
                 fgt/edit_texture \
-                ##				fgt/edit_topo \
+                				fgt/edit_topo \
