diff -Nur meshlab.orig/src/meshlab/meshlab.qrc meshlab/src/meshlab/meshlab.qrc
--- meshlab.orig/src/meshlab/meshlab.qrc	2010-04-30 23:10:45.000000000 +1000
+++ meshlab/src/meshlab/meshlab.qrc	2010-05-13 18:24:12.404074717 +1000
@@ -34,6 +34,5 @@
         <file>images/100mesh.html</file>
         <file>images/selected_face.png</file>
         <file>images/selected_vert.png</file>
-        <file>images/facebook.gif</file>
     </qresource>
 </RCC>
