--- kpl.orig/kpl.cpp	2011-06-11 16:30:07.000000000 +0600
+++ kpl/kpl.cpp	2012-05-09 17:45:56.154018830 +0600
@@ -323,9 +323,9 @@
   actionCollection()->action(
     KStandardAction::name(KStandardAction::ReportBug))->
       setStatusTip(i18n("Displays a dialog for submitting a bug report"));
-  actionCollection()->action(
-    KStandardAction::name(KStandardAction::SwitchApplicationLanguage))->
-      setStatusTip(i18n("Switches the language of the application"));
+//  actionCollection()->action(
+//    KStandardAction::name(KStandardAction::SwitchApplicationLanguage))->
+//      setStatusTip(i18n("Switches the language of the application"));
   actionCollection()->action(
     KStandardAction::name(KStandardAction::AboutApp))->
       setStatusTip(i18n("Displays essential information about Kpl"));
