diff -aur vovoid-vsxu-8bd519f/plugins/src/sound.rtaudio/CMakeLists.txt vovoid-vsxu-8bd519f.patched/plugins/src/sound.rtaudio/CMakeLists.txt
--- vovoid-vsxu-8bd519f/plugins/src/sound.rtaudio/CMakeLists.txt	2013-06-18 21:47:52.000000000 +0200
+++ vovoid-vsxu-8bd519f.patched/plugins/src/sound.rtaudio/CMakeLists.txt	2013-08-03 02:47:09.004827543 +0200
@@ -19,7 +19,7 @@
 
   find_package(JACK)
   if(JACK_FOUND)
-    add_definitions(-D__LINUX_JACK__)
+    add_definitions(-D__UNIX_JACK__)
     set( WITH_MIDI TRUE )
 
     include_directories(${JACK_INCLUDE_DIRS})
diff -aur vovoid-vsxu-8bd519f/plugins/src/sound.rtaudio/lib/RtAudio/RtAudio.cpp vovoid-vsxu-8bd519f.patched/plugins/src/sound.rtaudio/lib/RtAudio/RtAudio.cpp
--- vovoid-vsxu-8bd519f/plugins/src/sound.rtaudio/lib/RtAudio/RtAudio.cpp	2013-06-18 21:47:52.000000000 +0200
+++ vovoid-vsxu-8bd519f.patched/plugins/src/sound.rtaudio/lib/RtAudio/RtAudio.cpp	2013-08-03 02:47:09.004827543 +0200
@@ -41,6 +41,7 @@
 // RtAudio: Version 4.0.11
 
 #include "RtAudio.h"
+#include <iostream>
 #include <cstdlib>
 #include <cstring>
 #include <climits>
diff -aur vovoid-vsxu-8bd519f/plugins/src/sound.rtaudio/main.cpp vovoid-vsxu-8bd519f.patched/plugins/src/sound.rtaudio/main.cpp
--- vovoid-vsxu-8bd519f/plugins/src/sound.rtaudio/main.cpp	2013-06-18 21:47:52.000000000 +0200
+++ vovoid-vsxu-8bd519f.patched/plugins/src/sound.rtaudio/main.cpp	2013-08-03 02:47:09.004827543 +0200
@@ -131,14 +131,14 @@
 {
   printf("Parameters for vsx_listener (visualization sound input):\n");
   #if (PLATFORM == PLATFORM_LINUX)
-  printf("  PulseAudio is the default sound driver, but you can override with any\n"
+  printf("  JACK is the default sound driver, but you can override with any\n"
          "  of the following:\n\n");
   #if defined(__LINUX_ALSA__)
   printf("  -sound_type_alsa            - ALSA sound system\n");
   #endif
 
-  #if defined(__LINUX_JACK__)
-  printf("  -sound_type_jack            - JACK sound system\n");
+  #if defined(__LINUX_PULSE__)
+  printf("  -sound_type_pulse           - PulseAudio sound system\n");
   #endif
 
   #if defined(__LINUX_OSS__)
@@ -172,10 +172,10 @@
         // ALSA
         rtaudio_type = RtAudio::LINUX_ALSA;
       } else
-      if (internal_args->has_param("sound_type_jack"))
+      if (internal_args->has_param("sound_type_pulse"))
       {
-        // JACK
-        rtaudio_type = RtAudio::UNIX_JACK;
+        // PulseAudio
+        rtaudio_type = RtAudio::LINUX_PULSE;
       } else
       if (internal_args->has_param("sound_type_oss"))
       {
@@ -183,8 +183,8 @@
         rtaudio_type = RtAudio::LINUX_OSS;
       } else
       {
-        // default - PulseAudio
-        rtaudio_type = RtAudio::LINUX_PULSE;
+        // default - JACK
+        rtaudio_type = RtAudio::UNIX_JACK;
       }
       #endif
       #if (PLATFORM == PLATFORM_WINDOWS)
diff -aur vovoid-vsxu-8bd519f/plugins/src/sound.rtaudio/rtaudio_handler.h vovoid-vsxu-8bd519f.patched/plugins/src/sound.rtaudio/rtaudio_handler.h
--- vovoid-vsxu-8bd519f/plugins/src/sound.rtaudio/rtaudio_handler.h	2013-06-18 21:47:52.000000000 +0200
+++ vovoid-vsxu-8bd519f.patched/plugins/src/sound.rtaudio/rtaudio_handler.h	2013-08-03 02:47:23.454827832 +0200
@@ -113,7 +113,7 @@
                               log(
                                 10.0f
                                 +
-                                44100.0f * ( (float)ii * one_div_512 )
+                                48000.0f * ( (float)ii * one_div_512 )
                                 ),
                               1.0f
                               );
@@ -194,7 +194,7 @@
   parameters.deviceId = padc->getDefaultInputDevice();
   parameters.nChannels = 2;
   parameters.firstChannel = 0;
-  unsigned int sampleRate = 44100;
+  unsigned int sampleRate = 48000;
   unsigned int bufferFrames = 512;
 
   RtAudio::StreamOptions options;
