commit 5d599233537510c1b14f7da8e4a1f82714c39388
Author: Benjamin Chrétien <chretien@lirmm.fr>
Date:   Sat Mar 8 00:47:04 2014 +0100

    Arch-specifix fix.

diff --git a/Makefile.openrtm_aist_python b/Makefile.openrtm_aist_python
index a65223a..44ac403 100644
--- a/Makefile.openrtm_aist_python
+++ b/Makefile.openrtm_aist_python
@@ -11,8 +11,9 @@ MD5SUM_FILE = $(FILENAME).md5sum
 MK_DIR      = $(shell rospack find mk)
 
 include $(MK_DIR)/download_unpack_build.mk
-PYTHONDIR   = `python -c 'import sys; print "python%s"%sys.version[:3]'`
+PYTHONBIN   = python2
+PYTHONDIR   = `$(PYTHONBIN) -c 'import sys; print "python%s"%sys.version[:3]'`
 
 installed: $(SOURCE_DIR)/unpacked
-	cd $(SOURCE_DIR) && python ./setup.py build && python ./setup.py install --prefix=$(INSTALL_DIR) --install-scripts $(INSTALL_SCRIPTS_DIR) --install-layout=deb 
+	cd $(SOURCE_DIR) && $(PYTHONBIN) ./setup.py build && $(PYTHONBIN) ./setup.py install --prefix=$(INSTALL_DIR) --install-scripts $(INSTALL_SCRIPTS_DIR)
 	touch installed
