diff -Naur indicator-session-0.3.7.1.orig/src/indicator-session.c indicator-session-0.3.7.1/src/indicator-session.c
--- indicator-session-0.3.7.1.orig/src/indicator-session.c	2011-10-25 18:35:45.000000000 +0200
+++ indicator-session-0.3.7.1/src/indicator-session.c	2011-11-23 22:05:38.057026375 +0100
@@ -277,10 +277,10 @@
 {  
 	IndicatorSession * self = INDICATOR_SESSION (io);
   if (entry == &self->users){
-    return 1;
+    return 0;
   }
   else if (entry == &self->devices){
-    return 0;
+    return 1;
   }
   g_warning ("IOEntry handed to us to position but we don't own it!");
   return 0;
