diff -aur pptp-1.7.2/pptp.c pptp-1.7.2-copy/pptp.c
--- pptp-1.7.2/pptp.c	2008-05-14 08:33:55.000000000 +0200
+++ pptp-1.7.2-copy/pptp.c	2012-05-09 19:55:07.773599800 +0200
@@ -158,7 +158,6 @@
     LOG("rx accepted  = %d", rx_accepted);
     LOG("rx lost      = %d", rx_lost);
     LOG("rx under win = %d", rx_underwin);
-    LOG("rx over  win = %d", rx_overwin);
     LOG("rx buffered  = %d", rx_buffered);
     LOG("rx OS errors = %d", rx_errors);
     LOG("rx truncated = %d", rx_truncated);
diff -aur pptp-1.7.2/pptp_gre.c pptp-1.7.2-copy/pptp_gre.c
--- pptp-1.7.2/pptp_gre.c	2008-05-14 08:33:55.000000000 +0200
+++ pptp-1.7.2-copy/pptp_gre.c	2012-05-09 19:54:28.349160768 +0200
@@ -405,8 +405,7 @@
                 seq, seq_recv + 1);
         stats.rx_underwin++;
     /* sequence number too high, is it reasonably close? */
-    } else if ( seq < seq_recv + MISSING_WINDOW ||
-                WRAPPED(seq, seq_recv + MISSING_WINDOW) ) {
+    } else {
 	stats.rx_buffered++;
         if ( log_level >= 1 )
             log("%s packet %d (expecting %d, lost or reordered)",
@@ -418,14 +417,8 @@
             return callback(cl, buffer + ip_len + headersize, payload_len);
         } else {
             pqueue_add(seq, buffer + ip_len + headersize, payload_len);
-	}
-    /* no, packet must be discarded */
-    } else {
-	if ( log_level >= 1 )
-            warn("discarding bogus packet %d (expecting %d)", 
-		 seq, seq_recv + 1);
-        stats.rx_overwin++;
-    }
+		}
+    } 
     return 0;
 }
 
diff -aur pptp-1.7.2/pptp_gre.h pptp-1.7.2-copy/pptp_gre.h
--- pptp-1.7.2/pptp_gre.h	2008-05-14 08:33:55.000000000 +0200
+++ pptp-1.7.2-copy/pptp_gre.h	2012-05-09 19:55:38.168549044 +0200
@@ -26,8 +26,6 @@
   uint32_t rx_lost;      // data packet did not arrive before timeout
   uint32_t rx_underwin;  // data packet was under window (arrived too late
                          // or duplicate packet)
-  uint32_t rx_overwin;   // data packet was over window
-                         // (too many packets lost?)
   uint32_t rx_buffered;  // data packet arrived earlier than expected,
                          // packet(s) before it were lost or reordered
   uint32_t rx_errors;    // OS error on receive
diff -aur pptp-1.7.2/pqueue.h pptp-1.7.2-copy/pqueue.h
--- pptp-1.7.2/pqueue.h	2008-05-14 08:33:55.000000000 +0200
+++ pptp-1.7.2-copy/pqueue.h	2012-05-09 19:50:09.617113166 +0200
@@ -8,9 +8,6 @@
 #define DEFAULT_PACKET_TIMEOUT 0.3
 extern int packet_timeout_usecs;
 
-/* assume packet is bad/spoofed if it's more than this many seqs ahead */
-#define MISSING_WINDOW 300
-
 /* Packet queue structure: linked list of packets received out-of-order */
 typedef struct pqueue {
   struct pqueue *next;
