diff --git a/dlls/kernel32/sync.c b/dlls/kernel32/sync.c
index 08385f1..ec8c16a 100644
--- a/dlls/kernel32/sync.c
+++ b/dlls/kernel32/sync.c
@@ -1826,12 +1826,12 @@ HANDLE WINAPI CreateIoCompletionPort(HANDLE hFileHandle, HANDLE hExistingComplet
     TRACE("(%p, %p, %08lx, %08x)\n",
           hFileHandle, hExistingCompletionPort, CompletionKey, dwNumberOfConcurrentThreads);
 
-    if (hExistingCompletionPort && hFileHandle == INVALID_HANDLE_VALUE)
+/*    if (hExistingCompletionPort && hFileHandle == INVALID_HANDLE_VALUE)*/
     {
         SetLastError( ERROR_INVALID_PARAMETER);
         return NULL;
     }
-
+#if 0
     if (hExistingCompletionPort)
         ret = hExistingCompletionPort;
     else
@@ -1858,6 +1858,7 @@ fail:
         CloseHandle( ret );
     SetLastError( RtlNtStatusToDosError(status) );
     return 0;
+#endif
 }
 
 /******************************************************************************
