diff -Naur asterisk-1.8.32.3.orig/configure asterisk-1.8.32.3/configure
--- asterisk-1.8.32.3.orig/configure	2014-08-18 00:31:23.000000000 +0200
+++ asterisk-1.8.32.3/configure	2015-04-24 15:12:53.984258923 +0200
@@ -4704,7 +4704,7 @@
 astkeydir='${astvarlibdir}'
 astspooldir='${localstatedir}/spool/asterisk'
 astlogdir='${localstatedir}/log/asterisk'
-astvarrundir='${localstatedir}/run/asterisk'
+astvarrundir='/run/asterisk'
 
 case "${host_os}" in
      *bsd*)
