Index: darcs-bridge/src/darcs-fastconvert-gsoc/Bridge.hs
===================================================================
--- darcs-bridge.orig/src/darcs-fastconvert-gsoc/Bridge.hs
+++ darcs-bridge/src/darcs-fastconvert-gsoc/Bridge.hs
@@ -476,7 +476,7 @@ createPreHook bridgeDir = unlines hook w
          , "if [ \"$1\" = \"darcs\" ]; "
            ++ "then other=\"git\"; else other=\"darcs\"; fi"
          , "echo \"Pulling in any changes from the $other repo...\"\n"
-         , "darcs-fastconvert sync " ++ bridgeDir
+         , "darcs-bridge sync " ++ bridgeDir
            ++ " $1\n" ]
 
 -- |listBranches returns a list of branches that are managed by the bridge.
Index: darcs-bridge/src/darcs-fastconvert-gsoc/MergeTags.hs
===================================================================
--- darcs-bridge.orig/src/darcs-fastconvert-gsoc/MergeTags.hs
+++ darcs-bridge/src/darcs-fastconvert-gsoc/MergeTags.hs
@@ -19,7 +19,7 @@ import Darcs.Patch.Info ( isTag )
 import Darcs.Patch.PatchInfoAnd ( PatchInfoAnd, info )
 
 mergeTagPrefix :: String
-mergeTagPrefix = "darcs-fastconvert merge "
+mergeTagPrefix = "darcs-bridge merge "
 
 preTargetPrefix :: String
 preTargetPrefix = "pre-target: "
