--- wordcloud-original.py	2014-10-14 19:48:07.394940410 -0700
+++ wordcloud-patched.py	2014-10-14 19:49:04.092104991 -0700
@@ -17,7 +17,10 @@
 
 item1 = itemgetter(1)
 
-FONT_PATH = "/usr/share/fonts/truetype/droid/DroidSansMono.ttf"
+# Patched for Arch Linux
+FONT_PATH = "/usr/share/fonts/TTF/DroidSansMono.ttf"
+#FONT_PATH = "/usr/share/fonts/truetype/droid/DroidSansMono.ttf"
+
 STOPWORDS = set([x.strip() for x in open(os.path.join(os.path.dirname(__file__),
                                                       'stopwords')).read().split('\n')])
 
