diff -Nurp a/plugins/wb.admin/backend/wb_server_control.py
--- a/plugins/wb.admin/backend/wb_server_control.py   2013-08-0$
+++ b/plugins/wb.admin/backend/wb_server_control.py   2013-08-22
@@ -34,8 +34,8 @@ import socket
 
 #-------------------------------------------------------------------------------
 def get_local_ip_list():
-    cmd    = "/bin/sh -c ifconfig"
-    regexp = "inet addr:([0-9a-f:\.]+)"
+    cmd    = "/bin/sh -c 'ip addr'"
+    regexp = "inet[6]? ([0-9a-f:\.]+)"
 
     if hasattr(sys, 'getwindowsversion'):
         cmd    = 'ipconfig'
