--- a/plugins/wb.admin/backend/wb_admin_ssh.py	2014-03-26 16:47:23.000000000 +0100
+++ b/plugins/wb.admin/backend/wb_admin_ssh.py	2014-04-09 15:03:16.118452901 +0200
@@ -67,7 +67,7 @@ def normalize_windows_path_for_ftp(path)
 
 if paramiko and server_version_str2tuple(paramiko.__version__) >= (1, 7, 4):
     from paramiko.message import Message
-    from paramiko.common import sys, MSG_CHANNEL_OPEN
+    from paramiko.common import MSG_CHANNEL_OPEN
     from paramiko.channel import Channel
     from paramiko.ssh_exception import SSHException 
     import threading
