From 25cc754a9d6dcb8e42159be712e889766bc4a8f8 Mon Sep 17 00:00:00 2001
From: Eric Schulte <schulte.eric@gmail.com>
Date: Mon, 10 Mar 2014 22:54:01 -0600
Subject: [PATCH 5/6] not building guest-tools32

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4050bad..684290e 100644
--- a/Makefile
+++ b/Makefile
@@ -35,7 +35,7 @@ MAKE = make -j$(JOBS)
 
 all: all-release guest-tools
 
-guest-tools: stamps/guest-tools32-make stamps/guest-tools64-make
+guest-tools: stamps/guest-tools64-make
 
 all-release all-asan-release: stamps/tools-release-make
 all-release: stamps/qemu-release-make
-- 
1.9.0

