diff -Naur multispeech-2.4.5.orig/m4/boost.m4 multispeech-2.4.5/m4/boost.m4
--- multispeech-2.4.5.orig/m4/boost.m4	2010-02-08 07:21:00.000000000 +0600
+++ multispeech-2.4.5/m4/boost.m4	2010-11-24 10:12:09.439106643 +0600
@@ -78,7 +78,7 @@
 
 		dnl check for file system support
 		AC_CHECK_HEADERS([boost/filesystem.hpp boost/filesystem/fstream.hpp], [
-			LIBS="$saved_LIBS $BOOST_LIBS -lboost_filesystem-mt"
+			LIBS="$saved_LIBS $BOOST_LIBS -lboost_filesystem-mt -lboost_system-mt"
 			AC_LINK_IFELSE([
 				AC_LANG_PROGRAM([[#include <boost/filesystem.hpp>]],
 				[[boost::filesystem::path file("/dev/null");]])], [], [
