--- src/UI/Linux/TrayIcon/libtrayicon/eggtrayicon.c	2010-09-14 19:46:15.000000000 +0200
+++ src/UI/Linux/TrayIcon/libtrayicon/eggtrayicon.c	2011-05-23 16:16:09.273011462 +0200
@@ -27,6 +27,7 @@
 
 #include "eggtrayicon.h"
 
+#include <gtk/gtk.h>
 #include <gdk/gdkx.h>
 #include <X11/Xatom.h>
 
@@ -234,8 +235,13 @@
     {
       GdkWindow *gdkwin;
 
+#if GTK_CHECK_VERSION(2, 24, 0)
+    gdkwin = gdk_x11_window_lookup_for_display (gtk_widget_get_display (widget),
+                                               icon->manager_window);
+#else
       gdkwin = gdk_window_lookup_for_display (gtk_widget_get_display (widget),
                                               icon->manager_window);
+#endif
 
       gdk_window_remove_filter (gdkwin, egg_tray_icon_manager_filter, icon);
     }
@@ -299,8 +305,13 @@
     {
       GdkWindow *gdkwin;
 
+#if GTK_CHECK_VERSION(2, 24, 0)
+      gdkwin = gdk_x11_window_lookup_for_display (gtk_widget_get_display (GTK_WIDGET (icon)),
+                              icon->manager_window);
+#else
       gdkwin = gdk_window_lookup_for_display (gtk_widget_get_display (GTK_WIDGET (icon)),
 					      icon->manager_window);
+#endif
       
       gdk_window_remove_filter (gdkwin, egg_tray_icon_manager_filter, icon);
     }
@@ -321,8 +332,13 @@
     {
       GdkWindow *gdkwin;
 
+#if GTK_CHECK_VERSION(2, 24, 0)
+    gdkwin = gdk_x11_window_lookup_for_display (gtk_widget_get_display (GTK_WIDGET (icon)),         
+                            icon->manager_window);
+#else
       gdkwin = gdk_window_lookup_for_display (gtk_widget_get_display (GTK_WIDGET (icon)),
 					      icon->manager_window);
+#endif
       
       gdk_window_add_filter (gdkwin, egg_tray_icon_manager_filter, icon);
 
