--- ./pagespeed.conf	2011-01-03 13:21:03.877045676 +0100
+++ /etc/httpd/conf/extra/pagespeed.conf	2011-01-03 13:31:33.133440346 +0100
@@ -1,8 +1,8 @@
-LoadModule pagespeed_module @@APACHE_MODULEDIR@@/mod_pagespeed.so
+LoadModule pagespeed_module /etc/httpd/modules/mod_pagespeed.so
 
 # Only attempt to load mod_deflate if it hasn't been loaded already.
 <IfModule !mod_deflate.c>
-@@COMMENT_OUT_DEFLATE@@ LoadModule deflate_module @@APACHE_MODULEDIR@@/mod_deflate.so
+    LoadModule deflate_module /etc/httpd/modules/mod_deflate.so
 </IfModule>
 <IfModule pagespeed_module>
     # Turn on mod_pagespeed. To completely disable mod_pagespeed, you
@@ -17,8 +17,8 @@
     # ModPagespeedGeneratedFilePrefix directories must exist and be
     # writable by the apache user (as specified by the User
     # directive).
-    ModPagespeedFileCachePath            "@@MODPAGESPEED_CACHE_ROOT@@/cache/"
-    ModPagespeedGeneratedFilePrefix      "@@MODPAGESPEED_CACHE_ROOT@@/files/"
+    ModPagespeedFileCachePath            "/var/www/wwwtmp/cache/"
+    ModPagespeedGeneratedFilePrefix      "/var/www/wwwtmp/files/"
 
     # Override the mod_pagespeed 'rewrite level'. The default level
     # "CoreFilters" uses a set of rewrite filters that are generally
