--- ldc.rebuild.conf.in	2010-02-25 15:52:52.522648371 +0100
+++ ../../ldc.rebuild.conf.in	2010-03-18 00:04:26.729655444 +0100
@@ -1,2 +1,2 @@
 [Environment] 
-DFLAGS=-I@RUNTIME_DIR@ -I@RUNTIME_DIR@/lib/common -L-L%@P%/../lib -d-version=Tango -defaultlib=@RUNTIME_AIO@ -debuglib=@RUNTIME_AIO@ 
+DFLAGS=-I@RUNTIME_DIR@ -I@RUNTIME_DIR@/tango/core/vendor -L-L%@P%/../lib -d-version=Tango -defaultlib=@RUNTIME_AIO@ -debuglib=@RUNTIME_AIO@ 
