diff -ru DevIL-1.6.8/include/IL/il.h DevIL-1.6.8-gcc4.1/include/IL/il.h
--- DevIL-1.6.8/include/IL/il.h	2006-08-10 15:20:10.000000000 +0200
+++ DevIL-1.6.8-gcc4.1/include/IL/il.h	2008-07-02 09:39:40.000000000 +0200
@@ -445,15 +445,15 @@
 ILAPI ILvoid    ILAPIENTRY ilBindImage(ILuint Image);
 ILAPI ILboolean ILAPIENTRY ilBlit(ILuint Source, ILint DestX, ILint DestY, ILint DestZ, ILuint SrcX, ILuint SrcY, ILuint SrcZ, ILuint Width, ILuint Height, ILuint Depth);
 ILAPI ILvoid    ILAPIENTRY ilClearColour(ILclampf Red, ILclampf Green, ILclampf Blue, ILclampf Alpha);
-ILAPI ILboolean ILAPIENTRY ilClearImage(ILvoid);
-ILAPI ILuint    ILAPIENTRY ilCloneCurImage(ILvoid);
+ILAPI ILboolean ILAPIENTRY ilClearImage(void);
+ILAPI ILuint    ILAPIENTRY ilCloneCurImage(void);
 ILAPI ILboolean ILAPIENTRY ilCompressFunc(ILenum Mode);
 ILAPI ILboolean ILAPIENTRY ilConvertImage(ILenum DestFormat, ILenum DestType);
 ILAPI ILboolean ILAPIENTRY ilConvertPal(ILenum DestFormat);
 ILAPI ILboolean ILAPIENTRY ilCopyImage(ILuint Src);
-ILAPI ILuint    ILAPIENTRY ilCopyPixels(ILuint XOff, ILuint YOff, ILuint ZOff, ILuint Width, ILuint Height, ILuint Depth, ILenum Format, ILenum Type, ILvoid *Data);
+ILAPI ILuint    ILAPIENTRY ilCopyPixels(ILuint XOff, ILuint YOff, ILuint ZOff, ILuint Width, ILuint Height, ILuint Depth, ILenum Format, ILenum Type, void *Data);
 ILAPI ILuint    ILAPIENTRY ilCreateSubImage(ILenum Type, ILuint Num);
-ILAPI ILboolean ILAPIENTRY ilDefaultImage(ILvoid);
+ILAPI ILboolean ILAPIENTRY ilDefaultImage(void);
 ILAPI ILvoid    ILAPIENTRY ilDeleteImage(const ILuint Num);
 ILAPI ILvoid    ILAPIENTRY ilDeleteImages(ILsizei Num, const ILuint *Images);
 ILAPI ILboolean ILAPIENTRY ilDisable(ILenum Mode);
@@ -464,71 +464,71 @@
 ILAPI ILubyte*  ILAPIENTRY ilGetAlpha(ILenum Type);
 ILAPI ILboolean ILAPIENTRY ilGetBoolean(ILenum Mode);
 ILAPI ILvoid    ILAPIENTRY ilGetBooleanv(ILenum Mode, ILboolean *Param);
-ILAPI ILubyte*  ILAPIENTRY ilGetData(ILvoid);
-ILAPI ILuint    ILAPIENTRY ilGetDXTCData(ILvoid *Buffer, ILuint BufferSize, ILenum DXTCFormat);
-ILAPI ILenum    ILAPIENTRY ilGetError(ILvoid);
+ILAPI ILubyte*  ILAPIENTRY ilGetData(void);
+ILAPI ILuint    ILAPIENTRY ilGetDXTCData(void *Buffer, ILuint BufferSize, ILenum DXTCFormat);
+ILAPI ILenum    ILAPIENTRY ilGetError(void);
 ILAPI ILint     ILAPIENTRY ilGetInteger(ILenum Mode);
 ILAPI ILvoid    ILAPIENTRY ilGetIntegerv(ILenum Mode, ILint *Param);
-ILAPI ILuint    ILAPIENTRY ilGetLumpPos(ILvoid);
-ILAPI ILubyte*  ILAPIENTRY ilGetPalette(ILvoid);
+ILAPI ILuint    ILAPIENTRY ilGetLumpPos(void);
+ILAPI ILubyte*  ILAPIENTRY ilGetPalette(void);
 ILAPI ILstring  ILAPIENTRY ilGetString(ILenum StringName);
 ILAPI ILvoid    ILAPIENTRY ilHint(ILenum Target, ILenum Mode);
-ILAPI ILvoid    ILAPIENTRY ilInit(ILvoid);
+ILAPI ILvoid    ILAPIENTRY ilInit(void);
 ILAPI ILboolean ILAPIENTRY ilIsDisabled(ILenum Mode);
 ILAPI ILboolean ILAPIENTRY ilIsEnabled(ILenum Mode);
 ILAPI ILboolean ILAPIENTRY ilIsImage(ILuint Image);
 ILAPI ILboolean ILAPIENTRY ilIsValid(ILenum Type, ILstring FileName);
 ILAPI ILboolean ILAPIENTRY ilIsValidF(ILenum Type, ILHANDLE File);
-ILAPI ILboolean ILAPIENTRY ilIsValidL(ILenum Type, ILvoid *Lump, ILuint Size);
+ILAPI ILboolean ILAPIENTRY ilIsValidL(ILenum Type, void *Lump, ILuint Size);
 ILAPI ILvoid    ILAPIENTRY ilKeyColour(ILclampf Red, ILclampf Green, ILclampf Blue, ILclampf Alpha);
 ILAPI ILboolean ILAPIENTRY ilLoad(ILenum Type, const ILstring FileName);
 ILAPI ILboolean ILAPIENTRY ilLoadF(ILenum Type, ILHANDLE File);
 ILAPI ILboolean ILAPIENTRY ilLoadImage(const ILstring FileName);
-ILAPI ILboolean ILAPIENTRY ilLoadL(ILenum Type, const ILvoid *Lump, ILuint Size);
+ILAPI ILboolean ILAPIENTRY ilLoadL(ILenum Type, const void *Lump, ILuint Size);
 ILAPI ILboolean ILAPIENTRY ilLoadPal(const ILstring FileName);
 ILAPI ILvoid    ILAPIENTRY ilModAlpha( ILdouble AlphaValue );
 ILAPI ILboolean ILAPIENTRY ilOriginFunc(ILenum Mode);
 ILAPI ILboolean ILAPIENTRY ilOverlayImage(ILuint Source, ILint XCoord, ILint YCoord, ILint ZCoord);
-ILAPI ILvoid    ILAPIENTRY ilPopAttrib(ILvoid);
+ILAPI ILvoid    ILAPIENTRY ilPopAttrib(void);
 ILAPI ILvoid    ILAPIENTRY ilPushAttrib(ILuint Bits);
 ILAPI ILvoid    ILAPIENTRY ilRegisterFormat(ILenum Format);
 ILAPI ILboolean ILAPIENTRY ilRegisterLoad(const ILstring Ext, IL_LOADPROC Load);
 ILAPI ILboolean ILAPIENTRY ilRegisterMipNum(ILuint Num);
 ILAPI ILboolean ILAPIENTRY ilRegisterNumImages(ILuint Num);
 ILAPI ILvoid    ILAPIENTRY ilRegisterOrigin(ILenum Origin);
-ILAPI ILvoid    ILAPIENTRY ilRegisterPal(ILvoid *Pal, ILuint Size, ILenum Type);
+ILAPI ILvoid    ILAPIENTRY ilRegisterPal(void *Pal, ILuint Size, ILenum Type);
 ILAPI ILboolean ILAPIENTRY ilRegisterSave(const ILstring Ext, IL_SAVEPROC Save);
 ILAPI ILvoid    ILAPIENTRY ilRegisterType(ILenum Type);
 ILAPI ILboolean ILAPIENTRY ilRemoveLoad(const ILstring Ext);
 ILAPI ILboolean ILAPIENTRY ilRemoveSave(const ILstring Ext);
-ILAPI ILvoid    ILAPIENTRY ilResetMemory(ILvoid);
-ILAPI ILvoid    ILAPIENTRY ilResetRead(ILvoid);
-ILAPI ILvoid    ILAPIENTRY ilResetWrite(ILvoid);
+ILAPI ILvoid    ILAPIENTRY ilResetMemory(void);
+ILAPI ILvoid    ILAPIENTRY ilResetRead(void);
+ILAPI ILvoid    ILAPIENTRY ilResetWrite(void);
 ILAPI ILboolean ILAPIENTRY ilSave(ILenum Type, ILstring FileName);
 ILAPI ILuint    ILAPIENTRY ilSaveF(ILenum Type, ILHANDLE File);
 ILAPI ILboolean ILAPIENTRY ilSaveImage(const ILstring FileName);
-ILAPI ILuint    ILAPIENTRY ilSaveL(ILenum Type, ILvoid *Lump, ILuint Size);
+ILAPI ILuint    ILAPIENTRY ilSaveL(ILenum Type, void *Lump, ILuint Size);
 ILAPI ILboolean ILAPIENTRY ilSavePal(const ILstring FileName);
 ILAPI ILvoid    ILAPIENTRY ilSetAlpha( ILdouble AlphaValue );
-ILAPI ILboolean ILAPIENTRY ilSetData(ILvoid *Data);
+ILAPI ILboolean ILAPIENTRY ilSetData(void *Data);
 ILAPI ILboolean ILAPIENTRY ilSetDuration(ILuint Duration);
 ILAPI ILvoid    ILAPIENTRY ilSetInteger(ILenum Mode, ILint Param);
 ILAPI ILvoid    ILAPIENTRY ilSetMemory(mAlloc, mFree);
-ILAPI ILvoid    ILAPIENTRY ilSetPixels(ILint XOff, ILint YOff, ILint ZOff, ILuint Width, ILuint Height, ILuint Depth, ILenum Format, ILenum Type, ILvoid *Data);
+ILAPI ILvoid    ILAPIENTRY ilSetPixels(ILint XOff, ILint YOff, ILint ZOff, ILuint Width, ILuint Height, ILuint Depth, ILenum Format, ILenum Type, void *Data);
 ILAPI ILvoid    ILAPIENTRY ilSetRead(fOpenRProc, fCloseRProc, fEofProc, fGetcProc, fReadProc, fSeekRProc, fTellRProc);
 ILAPI ILvoid    ILAPIENTRY ilSetString(ILenum Mode, const char *String);
 ILAPI ILvoid    ILAPIENTRY ilSetWrite(fOpenWProc, fCloseWProc, fPutcProc, fSeekWProc, fTellWProc, fWriteProc);
-ILAPI ILvoid    ILAPIENTRY ilShutDown(ILvoid);
-ILAPI ILboolean ILAPIENTRY ilTexImage(ILuint Width, ILuint Height, ILuint Depth, ILubyte numChannels, ILenum Format, ILenum Type, ILvoid *Data);
+ILAPI ILvoid    ILAPIENTRY ilShutDown(void);
+ILAPI ILboolean ILAPIENTRY ilTexImage(ILuint Width, ILuint Height, ILuint Depth, ILubyte numChannels, ILenum Format, ILenum Type, void *Data);
 ILAPI ILenum    ILAPIENTRY ilTypeFromExt(const ILstring FileName);
 ILAPI ILboolean ILAPIENTRY ilTypeFunc(ILenum Mode);
 ILAPI ILboolean ILAPIENTRY ilLoadData(const ILstring FileName, ILuint Width, ILuint Height, ILuint Depth, ILubyte Bpp);
 ILAPI ILboolean ILAPIENTRY ilLoadDataF(ILHANDLE File, ILuint Width, ILuint Height, ILuint Depth, ILubyte Bpp);
-ILAPI ILboolean ILAPIENTRY ilLoadDataL(ILvoid *Lump, ILuint Size, ILuint Width, ILuint Height, ILuint Depth, ILubyte Bpp);
+ILAPI ILboolean ILAPIENTRY ilLoadDataL(void *Lump, ILuint Size, ILuint Width, ILuint Height, ILuint Depth, ILubyte Bpp);
 ILAPI ILboolean ILAPIENTRY ilSaveData(const ILstring FileName);
 
-ILAPI ILboolean ILAPIENTRY ilLoadFromJpegStruct(ILvoid* JpegDecompressorPtr);
-ILAPI ILboolean ILAPIENTRY ilSaveFromJpegStruct(ILvoid* JpegCompressorPtr);
+ILAPI ILboolean ILAPIENTRY ilLoadFromJpegStruct(void* JpegDecompressorPtr);
+ILAPI ILboolean ILAPIENTRY ilSaveFromJpegStruct(void* JpegCompressorPtr);
 
 // For all those weirdos that spell "colour" without the 'u'.
 #define ilClearColor	ilClearColour
diff -ru DevIL-1.6.8/include/IL/ilu.h DevIL-1.6.8-gcc4.1/include/IL/ilu.h
--- DevIL-1.6.8/include/IL/ilu.h	2006-08-16 11:18:21.000000000 +0200
+++ DevIL-1.6.8-gcc4.1/include/IL/ilu.h	2008-07-02 09:35:47.000000000 +0200
@@ -122,37 +122,37 @@
 
 
 // ImageLib Utility Functions
-ILAPI ILboolean      ILAPIENTRY iluAlienify(ILvoid);
+ILAPI ILboolean      ILAPIENTRY iluAlienify(void);
 ILAPI ILboolean      ILAPIENTRY iluBlurAvg(ILuint Iter);
 ILAPI ILboolean      ILAPIENTRY iluBlurGaussian(ILuint Iter);
-ILAPI ILboolean      ILAPIENTRY iluBuildMipmaps(ILvoid);
-ILAPI ILuint         ILAPIENTRY iluColoursUsed(ILvoid);
+ILAPI ILboolean      ILAPIENTRY iluBuildMipmaps(void);
+ILAPI ILuint         ILAPIENTRY iluColoursUsed(void);
 ILAPI ILboolean      ILAPIENTRY iluCompareImage(ILuint Comp);
 ILAPI ILboolean      ILAPIENTRY iluContrast(ILfloat Contrast);
 ILAPI ILboolean      ILAPIENTRY iluCrop(ILuint XOff, ILuint YOff, ILuint ZOff, ILuint Width, ILuint Height, ILuint Depth);
 ILAPI ILvoid         ILAPIENTRY iluDeleteImage(ILuint Id);
-ILAPI ILboolean      ILAPIENTRY iluEdgeDetectE(ILvoid);
-ILAPI ILboolean      ILAPIENTRY iluEdgeDetectP(ILvoid);
-ILAPI ILboolean      ILAPIENTRY iluEdgeDetectS(ILvoid);
-ILAPI ILboolean      ILAPIENTRY iluEmboss(ILvoid);
+ILAPI ILboolean      ILAPIENTRY iluEdgeDetectE(void);
+ILAPI ILboolean      ILAPIENTRY iluEdgeDetectP(void);
+ILAPI ILboolean      ILAPIENTRY iluEdgeDetectS(void);
+ILAPI ILboolean      ILAPIENTRY iluEmboss(void);
 ILAPI ILboolean      ILAPIENTRY iluEnlargeCanvas(ILuint Width, ILuint Height, ILuint Depth);
 ILAPI ILboolean      ILAPIENTRY iluEnlargeImage(ILfloat XDim, ILfloat YDim, ILfloat ZDim);
-ILAPI ILboolean      ILAPIENTRY iluEqualize(ILvoid);
+ILAPI ILboolean      ILAPIENTRY iluEqualize(void);
 ILAPI ILstring 		 ILAPIENTRY iluErrorString(ILenum Error);
 ILAPI ILboolean      ILAPIENTRY iluConvolution(ILint *matrix, ILint scale, ILint bias);
-ILAPI ILboolean      ILAPIENTRY iluFlipImage(ILvoid);
+ILAPI ILboolean      ILAPIENTRY iluFlipImage(void);
 ILAPI ILboolean      ILAPIENTRY iluGammaCorrect(ILfloat Gamma);
-ILAPI ILuint         ILAPIENTRY iluGenImage(ILvoid);
+ILAPI ILuint         ILAPIENTRY iluGenImage(void);
 ILAPI ILvoid         ILAPIENTRY iluGetImageInfo(ILinfo *Info);
 ILAPI ILint          ILAPIENTRY iluGetInteger(ILenum Mode);
 ILAPI ILvoid         ILAPIENTRY iluGetIntegerv(ILenum Mode, ILint *Param);
 ILAPI ILstring 		 ILAPIENTRY iluGetString(ILenum StringName);
 ILAPI ILvoid         ILAPIENTRY iluImageParameter(ILenum PName, ILenum Param);
-ILAPI ILvoid         ILAPIENTRY iluInit(ILvoid);
-ILAPI ILboolean      ILAPIENTRY iluInvertAlpha(ILvoid);
+ILAPI ILvoid         ILAPIENTRY iluInit(void);
+ILAPI ILboolean      ILAPIENTRY iluInvertAlpha(void);
 ILAPI ILuint         ILAPIENTRY iluLoadImage(const ILstring FileName);
-ILAPI ILboolean      ILAPIENTRY iluMirror(ILvoid);
-ILAPI ILboolean      ILAPIENTRY iluNegative(ILvoid);
+ILAPI ILboolean      ILAPIENTRY iluMirror(void);
+ILAPI ILboolean      ILAPIENTRY iluNegative(void);
 ILAPI ILboolean      ILAPIENTRY iluNoisify(ILclampf Tolerance);
 ILAPI ILboolean      ILAPIENTRY iluPixelize(ILuint PixSize);
 ILAPI ILvoid         ILAPIENTRY iluRegionfv(ILpointf *Points, ILuint n);
@@ -166,7 +166,7 @@
 ILAPI ILboolean      ILAPIENTRY iluScaleAlpha(ILfloat scale);
 ILAPI ILboolean      ILAPIENTRY iluScaleColours(ILfloat r, ILfloat g, ILfloat b);
 ILAPI ILboolean      ILAPIENTRY iluSharpen(ILfloat Factor, ILuint Iter);
-ILAPI ILboolean      ILAPIENTRY iluSwapColours(ILvoid);
+ILAPI ILboolean      ILAPIENTRY iluSwapColours(void);
 ILAPI ILboolean      ILAPIENTRY iluWave(ILfloat Angle);
 
 #define iluColorsUsed   iluColoursUsed
diff -ru DevIL-1.6.8/include/IL/ilut.h DevIL-1.6.8-gcc4.1/include/IL/ilut.h
--- DevIL-1.6.8/include/IL/ilut.h	2006-07-04 16:43:23.000000000 +0200
+++ DevIL-1.6.8-gcc4.1/include/IL/ilut.h	2008-07-02 09:35:47.000000000 +0200
@@ -198,10 +198,10 @@
 ILAPI ILint			ILAPIENTRY ilutGetInteger(ILenum Mode);
 ILAPI ILvoid		ILAPIENTRY ilutGetIntegerv(ILenum Mode, ILint *Param);
 ILAPI ILstring		ILAPIENTRY ilutGetString(ILenum StringName);
-ILAPI ILvoid		ILAPIENTRY ilutInit(ILvoid);
+ILAPI ILvoid		ILAPIENTRY ilutInit(void);
 ILAPI ILboolean		ILAPIENTRY ilutIsDisabled(ILenum Mode);
 ILAPI ILboolean		ILAPIENTRY ilutIsEnabled(ILenum Mode);
-ILAPI ILvoid		ILAPIENTRY ilutPopAttrib(ILvoid);
+ILAPI ILvoid		ILAPIENTRY ilutPopAttrib(void);
 ILAPI ILvoid		ILAPIENTRY ilutPushAttrib(ILuint Bits);
 ILAPI ILvoid		ILAPIENTRY ilutSetInteger(ILenum Mode, ILint Param);
 
@@ -211,11 +211,11 @@
 // ImageLib Utility Toolkit's OpenGL Functions
 #ifdef ILUT_USE_OPENGL
 	ILAPI GLuint	ILAPIENTRY ilutGLBindTexImage();
-	ILAPI GLuint	ILAPIENTRY ilutGLBindMipmaps(ILvoid);
-	ILAPI ILboolean	ILAPIENTRY ilutGLBuildMipmaps(ILvoid);
+	ILAPI GLuint	ILAPIENTRY ilutGLBindMipmaps(void);
+	ILAPI ILboolean	ILAPIENTRY ilutGLBuildMipmaps(void);
 	ILAPI GLuint	ILAPIENTRY ilutGLLoadImage(ILstring FileName);
-	ILAPI ILboolean	ILAPIENTRY ilutGLScreen(ILvoid);
-	ILAPI ILboolean	ILAPIENTRY ilutGLScreenie(ILvoid);
+	ILAPI ILboolean	ILAPIENTRY ilutGLScreen(void);
+	ILAPI ILboolean	ILAPIENTRY ilutGLScreenie(void);
 	ILAPI ILboolean	ILAPIENTRY ilutGLSaveImage(ILstring FileName, GLuint TexID);
 	ILAPI ILboolean	ILAPIENTRY ilutGLSetTex(GLuint TexID);
 	ILAPI ILboolean	ILAPIENTRY ilutGLTexImage(GLuint Level);
@@ -259,13 +259,13 @@
 	ILAPI HBITMAP	ILAPIENTRY ilutConvertSliceToHBitmap(HDC hDC, ILuint slice);
 	ILAPI ILvoid	ILAPIENTRY ilutFreePaddedData(ILubyte *Data);
 	ILAPI ILvoid	ILAPIENTRY ilutGetBmpInfo(BITMAPINFO *Info);
-	ILAPI HPALETTE	ILAPIENTRY ilutGetHPal(ILvoid);
-	ILAPI ILubyte*	ILAPIENTRY ilutGetPaddedData(ILvoid);
-	ILAPI ILboolean	ILAPIENTRY ilutGetWinClipboard(ILvoid);
+	ILAPI HPALETTE	ILAPIENTRY ilutGetHPal(void);
+	ILAPI ILubyte*	ILAPIENTRY ilutGetPaddedData(void);
+	ILAPI ILboolean	ILAPIENTRY ilutGetWinClipboard(void);
 	ILAPI ILboolean	ILAPIENTRY ilutLoadResource(HINSTANCE hInst, ILint ID, ILstring ResourceType, ILenum Type);
 	ILAPI ILboolean	ILAPIENTRY ilutSetHBitmap(HBITMAP Bitmap);
 	ILAPI ILboolean	ILAPIENTRY ilutSetHPal(HPALETTE Pal);
-	ILAPI ILboolean	ILAPIENTRY ilutSetWinClipboard(ILvoid);
+	ILAPI ILboolean	ILAPIENTRY ilutSetWinClipboard(void);
 	ILAPI HBITMAP	ILAPIENTRY ilutWinLoadImage(ILstring FileName, HDC hDC);
 	ILAPI ILboolean	ILAPIENTRY ilutWinLoadUrl(ILstring Url);
 	ILAPI ILboolean ILAPIENTRY ilutWinPrint(ILuint XPos, ILuint YPos, ILuint Width, ILuint Height, HDC hDC);
