--- rivendell-2.1.5/cae/Makefile.am	2010-07-29 15:32:30.000000000 -0400
+++ rivendell-2.1.5-dirty/cae/Makefile.am	2012-08-16 12:11:54.076253345 -0400
@@ -23,7 +23,7 @@
 
 AM_CPPFLAGS = -Wall @QT_CXXFLAGS@
 INCLUDES = -I$(top_srcdir)/lib -I$(top_srcdir)/rdhpi
-LIBS = @QT_LIBS@ -L$(top_srcdir)/lib -L$(top_srcdir)/rdhpi
+LIBS = @QT_LIBS@ -L$(top_srcdir)/lib -L$(top_srcdir)/rdhpi -lpthread
 MOC = @QT_MOC@
 
 # The dependency for qt's Meta Object Compiler (moc)
