--- rivendell-2.1.5/conf/rd-bin.conf.in	2011-06-23 15:47:28.000000000 -0400
+++ rivendell-2.1.5-dirty/conf/rd-bin.conf.in	2012-08-17 02:21:33.637805096 -0400
@@ -20,7 +20,7 @@
 #   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 #
 
-<Directory "@libexecdir@">
+<Directory "@libexecdir@/rivendell/web">
   Options ExecCGI FollowSymLinks
   AllowOverride None
   Order allow,deny
@@ -47,5 +47,5 @@
     SetHandler cgi-script
   </Files>
 </Directory>
-ScriptAlias /rd-bin/ "@libexecdir@/"
+ScriptAlias /rd-bin/ "@libexecdir@/rivendell/web"
 TimeOut 1200
