--- ../../../gmap-gsnap/gmap-2014-02-18/config/config.sub	2014-02-18 14:05:33.000000000 -0800
+++ config.sub	2014-07-25 16:23:55.979536389 -0700
@@ -299,6 +299,9 @@
 	ms1)
 		basic_machine=mt-unknown
 		;;
+  x86_64-Arch-Linux)
+    basic_machine=x86_64-pc
+    ;;
 
 	# We use `pc' rather than `unknown'
 	# because (1) that's what they normally are, and
@@ -1187,6 +1190,9 @@
 	pmac | pmac-mpw)
 		basic_machine=powerpc-apple
 		;;
+  x86_64-Arch-Linux)
+    basic_machine=x86_64-pc
+    ;;
 	*-unknown)
 		# Make sure to match an already-canonicalized machine name.
 		;;
@@ -1388,6 +1394,9 @@
 	-zvmoe)
 		os=-zvmoe
 		;;
+  x86_64-Arch-Linux)
+    basic_machine=x86_64-pc
+    ;;
 	-none)
 		;;
 	*)
