diff -u ../../Allim_1.1/AllimOptions_2Pexpr ./AllimOptions_2Pexpr
--- ../../Allim_1.1/AllimOptions_2Pexpr	2013-03-18 02:20:46.000000000 -0700
+++ ./AllimOptions_2Pexpr	2014-08-01 12:33:46.232207152 -0700
@@ -214,24 +214,19 @@
 
 ### Get samtools to manipulate SAM and BAM files from http://samtools.sourceforge.net/
 # Provide the full path of the executables relative to the <Allim_1.0>
-SAMTOOLS = /Allim_1.0/executables/samtools
+SAMTOOLS = /usr/bin/samtools
 
 
-### Get intersectBed executables from a collection of useful utilities called bedtools-
-### from http://code.google.com/p/bedtools/
+### Get intersectBed executables from a collection of useful utilities called bedtools from
+### http://code.google.com/p/bedtools/
 # Provide the full paths of the executables relative to the <Allim_1.0>
-INTERSECTBED = /Allim_1.0/executables/intersectBed
-SORTBED = /Allim_1.0/executables/sortBed
-BAMTOBED = /Allim_1.0/executables/bamToBed
+INTERSECTBED = /usr/bin/intersectBed
+SORTBED = /usr/bin/sortBed
+BAMTOBED = /usr/bin/bamToBed
 
 
 #### Three PICARD JAR files from http://picard.sourceforge.net/index.shtml
 # Provide the full paths of the executables relative to the <Allim_1.0>
-SORTSAM = /Allim_1.0/executables/SortSam.jar
-MERGESAMFILES = /Allim_1.0/executables/MergeSamFiles.jar
-BUILDBAMINDEX = /Allim_1.0/executables/BuildBamIndex.jar
-
-
-
-
-
+SORTSAM = /usr/share/java/picard-tools/SortSam.jar
+MERGESAMFILES = /usr/share/java/picard-tools/MergeSamFiles.jar
+BUILDBAMINDEX = /usr/share/java/picard-tools/BuildBamIndex.jar
diff -u ../../Allim_1.1/AllimOptions_2Pgenomes ./AllimOptions_2Pgenomes
--- ../../Allim_1.1/AllimOptions_2Pgenomes	2012-10-19 05:02:24.000000000 -0700
+++ ./AllimOptions_2Pgenomes	2014-08-01 12:33:24.626029799 -0700
@@ -124,22 +124,22 @@
 
 ### Get samtools to manipulate SAM and BAM files from http://samtools.sourceforge.net/
 # Provide the full path of the executables relative to the <Allim_1.0>
-SAMTOOLS = /Allim_1.0/executables/samtools
+SAMTOOLS = /usr/bin/samtools
 
 
 ### Get intersectBed executables from a collection of useful utilities called bedtools from
 ### http://code.google.com/p/bedtools/
 # Provide the full paths of the executables relative to the <Allim_1.0>
-INTERSECTBED = /Allim_1.0/executables/intersectBed
-SORTBED = /Allim_1.0/executables/sortBed
-BAMTOBED = /Allim_1.0/executables/bamToBed
+INTERSECTBED = /usr/bin/intersectBed
+SORTBED = /usr/bin/sortBed
+BAMTOBED = /usr/bin/bamToBed
 
 
 #### Three PICARD JAR files from http://picard.sourceforge.net/index.shtml
 # Provide the full paths of the executables relative to the <Allim_1.0>
-SORTSAM = /Allim_1.0/executables/SortSam.jar
-MERGESAMFILES = /Allim_1.0/executables/MergeSamFiles.jar
-BUILDBAMINDEX = /Allim_1.0/executables/BuildBamIndex.jar
+SORTSAM = /usr/share/java/picard-tools/SortSam.jar
+MERGESAMFILES = /usr/share/java/picard-tools/MergeSamFiles.jar
+BUILDBAMINDEX = /usr/share/java/picard-tools/BuildBamIndex.jar
 
 
 
Only in ./: allim.patch
Common subdirectories: ../../Allim_1.1/classes and ./classes
Common subdirectories: ../../Allim_1.1/executables and ./executables
