From 368f6fb598dbd22f03319df0d6ba380865ab146f Mon Sep 17 00:00:00 2001
From: Robson Roberto Souza Peixoto <robsonpeixoto@gmail.com>
Date: Wed, 20 Oct 2010 17:07:19 +0000
Subject: Fix build on Mono 2.8

Signed-off-by: Gabriel Burt <gabriel.burt@gmail.com>
---
diff --git src/PdfMod/Gui/Client.cs src/PdfMod/Gui/Client.cs
index 19e0af6..d7a408b 100644
--- src/PdfMod/Gui/Client.cs
+++ src/PdfMod/Gui/Client.cs
@@ -415,7 +415,7 @@ namespace PdfMod.Gui
         static void OnLogNotify (LogNotifyArgs args)
         {
             ThreadAssist.ProxyToMain (delegate {
-                Gtk.MessageType mtype;
+                Gtk.MessageType mtype = Gtk.MessageType.Error;
                 var entry = args.Entry;
 
                 switch (entry.Type) {
--
cgit v0.8.3.1
