--- interfaces/Makefile.am~	2011-02-18 08:46:19.446685569 -0800
+++ interfaces/Makefile.am	2011-02-18 08:46:33.170018783 -0800
@@ -19,7 +19,7 @@
 	     sw-collections.xml
 
 %-ginterface.h %-ginterface.c: %.xml Makefile.am
-	$(AM_V_GEN)python $(top_srcdir)/tools/glib-ginterface-gen.py --include='"sw-marshals.h"' --filename=$(basename $@) $< Sw_
+	$(AM_V_GEN)python2 $(top_srcdir)/tools/glib-ginterface-gen.py --include='"sw-marshals.h"' --filename=$(basename $@) $< Sw_
 
 GINTERFACE_SOURCES = \
 	$(DBUS_SPECS:.xml=-ginterface.h) \
