diff -ur numdisplay/lib/displaydev.py numdisplay.a/lib/displaydev.py
--- numdisplay/lib/displaydev.py	2009-11-27 17:11:29.000000000 +0100
+++ numdisplay.a/lib/displaydev.py	2011-11-24 15:27:56.676930843 +0100
@@ -536,7 +536,7 @@
         
         a = n.array([tid,thingct,subunit,0,x,y,z,t],dtype=n.uint16)
         # Compute the checksum
-        sum = n.add.reduce(a)
+        sum = n.add.reduce(a, dtype=n.uint16)
         sum = 0xffff - (sum & 0xffff)
         a[3] = sum
         self._write(a.tostring())
