diff -x config.log -x config.status -ru swbis-1.11/swsupplib/taru/to_oct.c swbis-1.11/swsupplib/taru/to_oct.c
--- swbis-1.11/swsupplib/taru/to_oct.c	2010-06-26 17:51:57.000000000 +0200
+++ swbis-1.11/swsupplib/taru/to_oct.c	2013-04-04 08:32:15.313357321 +0200
@@ -59,6 +59,10 @@
 	>> CHAR_BIT * (sizeof (int) - 1)))
 #endif
 
+#ifndef __USE_XOPEN
+ #define __USE_XOPEN 1
+#endif
+
 # include <stdio.h>
 # include <unistd.h>
 # include <sys/types.h>
