--- pyffmpeg.pyx   2011-01-24 10:25:06.000000000 +0100
+++ pyffmpeg-fixed.pyx  2011-01-24 10:33:04.000000000 +0100
@@ -804,7 +804,7 @@
  #   self.prepacket=<AVPacket *>None
 #   self.packet=&self.packetbufa
 
-    def __new__(self):
+    def __cinit__(self):
         pass
 
     def dump(self):
@@ -1835,7 +1835,7 @@
     cdef int with_readahead
     cdef unsigned long long int seek_before_security_interval
 
-    def __new__(self,with_readahead=True,seek_before=4000):
+    def __cinit__(self,with_readahead=True,seek_before=4000):
         self.filename = None
         self.tracks=[]
         self.ctracks=NULL
