diff -rNup async.orig/boost/async/wait_for_any.hpp async/boost/async/wait_for_any.hpp
--- async.orig/boost/async/wait_for_any.hpp	2010-08-16 14:35:44.892372168 -0300
+++ async/boost/async/wait_for_any.hpp	2010-08-16 14:36:12.602373474 -0300
@@ -189,7 +189,7 @@ wait_for_any_and_interrupt( AE& ae, F1 f
         >::type >::move_dest_type
     >::type fut1= get_future<AE>()(fusion::at_c<1>(handles));
     unsigned r = boost::wait_for_any(fut0, fut1);
-    std::cout << "boost::wait_for_any=" << r << std::endl;
+    //std::cout << "boost::wait_for_any=" << r << std::endl;
     typename result_of::wait_for_any<AE, fusion::tuple<F1,F2> >::type res =
         detail::partial<2,typename boost::result_of<F1()>::type >::make(r,handles);
 
diff -rNup async.orig/boost/futures/future.hpp async/boost/futures/future.hpp
--- async.orig/boost/futures/future.hpp	2010-08-16 14:35:44.892372168 -0300
+++ async/boost/futures/future.hpp	2010-08-16 14:36:49.642373611 -0300
@@ -24,6 +24,7 @@
 #include <boost/ref.hpp>
 #include <boost/scoped_array.hpp>
 #include <boost/utility/enable_if.hpp>
+#include <vector>
 #include <list>
 #include <boost/next_prior.hpp>
 
