diff -aur dictionary-1.8.7.pristine/dictionary.el dictionary-1.8.7.new/dictionary.el
--- dictionary-1.8.7.pristine/dictionary.el	2004-10-02 02:31:20.000000000 -0400
+++ dictionary-1.8.7.new/dictionary.el	2012-06-02 09:00:34.072489684 -0400
@@ -329,8 +329,9 @@
   
   (make-local-variable 'dictionary-default-dictionary)
   (make-local-variable 'dictionary-default-strategy)
-  
-  (make-local-hook 'kill-buffer-hook)
+
+  (when (featurep 'xemacs)
+    (make-local-hook 'kill-buffer-hook))
   (add-hook 'kill-buffer-hook 'dictionary-close t t)
   (run-hooks 'dictionary-mode-hook))
 
