diff --git a/src/gnome-plist-editor.c b/src/gnome-plist-editor.c
index 7dc4ae7..eb19a38 100644
--- a/src/gnome-plist-editor.c
+++ b/src/gnome-plist-editor.c
@@ -557,7 +557,7 @@ int main(int argc, char **argv) {
 
 	/* Create the main window */
 	gtk_builder = gtk_builder_new();
-	gtk_builder_add_from_file(gtk_builder, "gnome-plist-editor.ui", NULL);
+	gtk_builder_add_from_file(gtk_builder, "/usr/share/gnome-plist-editor.ui", NULL);
 	gtk_builder_connect_signals(gtk_builder, NULL);
 
 	app.main_window = GTK_WINDOW(gtk_builder_get_object(gtk_builder, "main_window"));
