Index: i3status-2.6/src/print_volume.c
===================================================================
--- i3status-2.6.orig/src/print_volume.c
+++ i3status-2.6/src/print_volume.c
@@ -49,6 +49,8 @@ static void free_hdl(struct mixer_hdl *h
 }
 #endif
 
+static unsigned char graph[3] = {0xe2, 0x96, 0x81};
+
 void print_volume(yajl_gen json_gen, char *buffer, const char *fmt, const char *device, const char *mixer, int mixer_idx) {
         char *outwalk = buffer;
 
@@ -172,6 +174,12 @@ void print_volume(yajl_gen json_gen, cha
 			outwalk += sprintf(outwalk, "%d%%", avg);
 			walk += strlen("volume");
 		}
+                if (BEGINS_WITH(walk+1, "graph")) {
+                        graph[2] = 0x81 + (avg / 14);
+                        memcpy(outwalk, graph, 3);
+                        outwalk += 3;
+                        walk += strlen("graph");
+                }
 	}
 #endif
 #if defined(__FreeBSD__) || defined(__OpenBSD__)
