=== modified file 'src/sakura.c'
--- src/sakura.c	2010-10-06 13:45:12 +0000
+++ src/sakura.c	2010-10-09 00:58:40 +0000
@@ -242,6 +242,7 @@
             term_data_id, term, (GDestroyNotify)g_free);
 
 /* Callbacks */
+static gboolean sakura_dbl_click (GtkWidget *, GdkEventButton *, gpointer);
 static gboolean sakura_key_press (GtkWidget *, GdkEventKey *, gpointer);
 static void     sakura_increase_font (GtkWidget *, void *);
 static void     sakura_decrease_font (GtkWidget *, void *);
@@ -312,6 +313,14 @@
     { NULL }
 };
 
+static
+gboolean sakura_dbl_click (GtkWidget *widget, GdkEventButton *event, gpointer user_data)
+{
+	if(event->type==GDK_2BUTTON_PRESS || event->type==GDK_3BUTTON_PRESS) {
+		sakura_new_tab (widget, user_data);
+	}
+	return FALSE;
+}
 
 static
 gboolean sakura_key_press (GtkWidget *widget, GdkEventKey *event, gpointer user_data)
@@ -1739,6 +1748,7 @@
 	g_signal_connect(G_OBJECT(sakura.main_window), "destroy", G_CALLBACK(sakura_destroy_window), NULL);
 	g_signal_connect(G_OBJECT(sakura.main_window), "key-press-event", G_CALLBACK(sakura_key_press), NULL);
 	g_signal_connect(G_OBJECT(sakura.main_window), "configure-event", G_CALLBACK(sakura_resized_window), NULL);
+	g_signal_connect(G_OBJECT(sakura.main_window), "button-press-event", G_CALLBACK(sakura_dbl_click), NULL);
 }
 
 

