Index: hsbuild/haskell-cspm-interpreter/src/CSPM-Interpreter-0.7.0.0/CSPM-Interpreter.cabal
===================================================================
--- hsbuild.orig/haskell-cspm-interpreter/src/CSPM-Interpreter-0.7.0.0/CSPM-Interpreter.cabal
+++ hsbuild/haskell-cspm-interpreter/src/CSPM-Interpreter-0.7.0.0/CSPM-Interpreter.cabal
@@ -18,7 +18,7 @@ Maintainer:          Marc Fontaine <font
 Stability:           experimental
 Tested-With:         GHC == 7.8.2
 
-cabal-Version:       >= 1.20
+cabal-Version:       >= 1.18
 build-type: Simple
 Extra-source-files:
 Library
@@ -57,4 +57,4 @@ Library
     CSPM.Interpreter.SSet
     Data.Digest.Pure.MD5
     Data.Digest.Pure.HashMD5
-    Paths_CSPM_Interpreter
\ No newline at end of file
+    Paths_CSPM_Interpreter
