--- OpenMesh-3.0/Doc/CMakeLists.txt.orig	2014-04-05 20:48:18.768489253 -0700
+++ OpenMesh-3.0/Doc/CMakeLists.txt	2014-04-05 20:48:44.195056756 -0700
@@ -7,13 +7,12 @@
   endif ()
 
   # Add a documentation install target
-  add_custom_target(doc-install COMMENT "Installing Documentation" VERBATIM )
+  install(DIRECTORY ${CMAKE_BINARY_DIR}/Build/${ACG_PROJECT_DATADIR}/Doc/html/
+    DESTINATION ${ACG_PROJECT_DATADIR}/doc/html
+    COMPONENT documentation)
+  add_custom_target(doc-install
+    COMMAND ${CMAKE_COMMAND} -DCMAKE_INSTALL_COMPONENT=documentation -P ${CMAKE_BINARY_DIR}/cmake_install.cmake
+    DEPENDS doc
+    COMMENT "Installing Documentation" VERBATIM )
 
-  # Build Documentation before installing it
-  add_dependencies(doc-install doc)
-
-  if (TARGET doc-install)
-    acg_copy_after_build (doc-install "${CMAKE_BINARY_DIR}/Build/${ACG_PROJECT_DATADIR}/Doc/html"   "${CMAKE_INSTALL_PREFIX}/${ACG_PROJECT_DATADIR}/doc/html")
-  endif()
-
-ENDIF(DOXYGEN_FOUND)
\ No newline at end of file
+ENDIF(DOXYGEN_FOUND)
