--- configure.in.old	2005-07-20 17:28:53.000000000 +0200
+++ configure.in	2013-05-27 16:14:25.948913287 +0200
@@ -144,8 +144,8 @@
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
-AC_TYPE_UID_T
-AC_TYPE_PID_T
+#AC_TYPE_UID_T
+#AC_TYPE_PID_T
 AC_TYPE_SIZE_T
 AC_HEADER_TIME
 AC_STRUCT_TM
@@ -158,7 +158,7 @@
 AC_FUNC_STRCOLL
 AC_FUNC_VFORK
 AC_FUNC_VPRINTF
-AC_FUNC_GETLOADAVG
+#AC_FUNC_GETLOADAVG
 
 dnl Do this again so the AC_CHECK_FUNCS works.
 
