Index: vtkKWResourceUtilities.cxx
===================================================================
RCS file: /cvsroot/KWWidgets/KWWidgets/vtkKWResourceUtilities.cxx,v
retrieving revision 1.24
diff -u -r1.24 vtkKWResourceUtilities.cxx
--- vtkKWResourceUtilities.cxx	16 Oct 2008 22:32:29 -0000	1.24
+++ vtkKWResourceUtilities.cxx	24 Apr 2011 11:32:40 -0000
@@ -165,7 +165,7 @@
     {
     if (bit_depth < 8) 
       {
-      png_set_gray_1_2_4_to_8(png_ptr);
+      png_set_expand_gray_1_2_4_to_8(png_ptr);
       }
     png_set_gray_to_rgb(png_ptr);
     }
