--- Makefile	2010-05-27 10:57:45.180556613 +0200
+++ Makefile.new	2010-05-27 10:57:14.656429459 +0200
@@ -12,14 +12,15 @@
 prefix=/usr
 exec_prefix=${prefix}
 
+DESTDIR=?
 INSTALLDIR = ${exec_prefix}/sbin
 INSTALLLIB = ${exec_prefix}/lib
 INSTALLMAN = ${prefix}/man
 TOT_CONFIG_FILE=/etc/totd.conf
 
-INSTALL = /usr/bin/install
+INSTALL = /bin/install
 
-CFLAGS  = -march=x86-64 -mtune=generic -O2 -pipe -DHAVE_CONFIG_H  -Werror -Wall -DTOTCONF=\"$(TOT_CONFIG_FILE)\" -DSCOPED_REWRITE -DUSE_INET4 -DUSE_INET6  $(INCLUDEPATH)
+CFLAGS  = -march=x86-64 -mtune=generic -O2 -pipe -DHAVE_CONFIG_H -Wall -DTOTCONF=\"$(TOT_CONFIG_FILE)\" -DSCOPED_REWRITE -DUSE_INET4 -DUSE_INET6  $(INCLUDEPATH)
 
 # When debugging is enabled by --enable-malloc-debug flag to the configure
 # script, the # substitution will contain the empty string, thus
@@ -66,8 +67,9 @@
 	-ranlib libswill.a
 
 install: $(PROG)
-	$(INSTALL) -c -s -m 0555 -o bin -g bin $(PROG) $(INSTALLDIR)
-	$(INSTALL) -c -m 0444 -o root -g wheel $(MAN) $(INSTALLMAN)/man8
+	$(INSTALL) $(PROG) $(DESTDIR)$(INSTALLDIR)
+	$(INSTALL) $(MAN) $(DESTDIR)$(INSTALLMAN)/man8
+
 
 lint:
 	lint ${SRCS}
