--- ./libpurple/protocols/oscar/oscar.c	2010-01-10 14:33:32.579045106 -0500
+++ ./libpurple/protocols/oscar/oscar.c.new	2010-01-10 15:14:44.271896000 -0500
@@ -457,10 +457,10 @@
 		charsetstr1 = "ASCII";
 		charsetstr2 = purple_account_get_string(account, "encoding", OSCAR_DEFAULT_CUSTOM_ENCODING);
 	} else if (charset == 0x000d) {
-		/* iChat sending unicode over a Direct IM connection = Unicode */
-		/* Mobile AIM client on a Nokia 3100 and an LG VX6000 = ISO-8859-1 */
-		charsetstr1 = "UTF-16BE";
-		charsetstr2 = "UTF-8";
+		/* iChat sending unicode over a Direct IM connection = UTF-8 */
+		/* Mobile AIM client on multiple devices (including Blackberry Tour, Nokia 3100, and LG VX6000) = ISO-8859-1 */
+		charsetstr1 = "UTF-8";
+		charsetstr2 = "ISO-8859-1";
 		charsetstr3 = purple_account_get_string(account, "encoding", OSCAR_DEFAULT_CUSTOM_ENCODING);
 	} else {
 		/* Unknown, hope for valid UTF-8... */
@@ -3576,9 +3576,9 @@
 
 			purple_debug_misc("oscar",
 					"created room: %s %hu %hu %hu %u %hu %hu %hhu %hu %s %s\n",
-					fqcn, exchange, instance, flags, createtime,
+					fqcn ? fqcn : "(null)", exchange, instance, flags, createtime,
 					maxmsglen, maxoccupancy, createperms, unknown,
-					name, ck);
+					name ? name : "(null)", ck);
 			aim_chat_join(od, exchange, ck, instance);
 			}
 			break;
