From 1b7ea533ea79bacbb14979e92416fa03b22ee774 Mon Sep 17 00:00:00 2001
From: Robert Kuska <rkuska@redhat.com>
Date: Sun, 24 Aug 2014 15:42:41 +0200
Subject: [PATCH] Change exception check of TypeError

---
 click/core.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/click/core.py b/click/core.py
index b4c7368..fd179a2 100644
--- a/click/core.py
+++ b/click/core.py
@@ -432,7 +432,7 @@ class Context(object):
             except TypeError as e:
                 if not injected_arguments:
                     raise
-                if 'got multiple values for keyword argument' in str(e):
+                if 'got multiple values for' in str(e):
                     raise RuntimeError(
                         'You called .invoke() on the context with a command '
                         'but provided parameters as positional arguments.  '
-- 
1.9.3
