--- fms-1.0/fmplayer.cpp.orig	2008-04-09 15:33:41.000000000 -0300
+++ fms-1.0/fmplayer.cpp	2008-06-26 19:31:24.000000000 -0300
@@ -26,6 +26,8 @@
 #include "include/fmplayer_modulator.cpp"
 #include "include/fmplayer_midi.cpp"
 
+#include <limits.h>
+
 const double exact_limit = 0.0021; // exacticity problems with fmod or anywhere
 
 void writeSoundByte(char * byte, bool wav, int bytes, int dspfd) {
--- fms-1.0/include/fmplayer_mixer.cpp.orig	2008-04-09 15:33:41.000000000 -0300
+++ fms-1.0/include/fmplayer_mixer.cpp	2008-06-26 19:32:46.000000000 -0300
@@ -17,6 +17,8 @@
 
 /*----------------------------------------------------------------*/
 
+#include <limits.h>
+
 int FMMixer::lfd_nr = 1;
 
 FMMixer::FMMixer(FMChannel *parentChannel){


