From 77a8dfc6b7f5cd7525bd733ed840dd065b2a152d Mon Sep 17 00:00:00 2001
From: Yosef Or Boczko <yosefor3@walla.com>
Date: Mon, 25 Mar 2013 15:36:20 +0200
Subject: [PATCH] Add exec_prefix to geoip.pc.in

---
 geoip.pc.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/geoip.pc.in b/geoip.pc.in
index d943880..31b8fff 100644
--- a/geoip.pc.in
+++ b/geoip.pc.in
@@ -1,4 +1,5 @@
 prefix=@prefix@
+exec_prefix=@prefix@
 libdir=@libdir@
 includedir=@includedir@
 datadir=@datarootdir@
-- 
1.8.2

