--- Makefile	2010-04-03 01:55:51.000000000 +0200
+++ Makefile	2010-04-03 01:55:21.000000000 +0200
@@ -8,12 +8,12 @@
 # What is this package?
 NLSPACKAGE	= pykickstart
 POTFILE		= $(NLSPACKAGE).pot
-INSTALL		= /usr/bin/install -c
+INSTALL		= /bin/install -c
 INSTALL_DATA	= $(INSTALL) -m 644
-INSTALL_DIR	= /usr/bin/install -d
+INSTALL_DIR	= /bin/install -d
 
 # destination directory
-INSTALL_NLS_DIR = $(RPM_BUILD_ROOT)/usr/share/locale
+INSTALL_NLS_DIR = $(DESTDIR)/usr/share/locale
 
 # PO catalog handling
 MSGMERGE	= msgmerge -v
