diff -urN amarokFS-0.5/amarokFS-xml.pro amarokFS-0.5-arch/amarokFS-xml.pro
--- amarokFS-0.5/amarokFS-xml.pro	2007-03-19 13:17:58.000000000 +0100
+++ amarokFS-0.5-arch/amarokFS-xml.pro	2007-04-30 11:48:25.000000000 +0200
@@ -3,12 +3,8 @@
 
 CONFIG		+= qt warn_on release
 
-INCLUDEPATH += /usr/include/kde
-INCLUDEPATH += /usr/local/include/kde
 INCLUDEPATH += /usr/include
-INCLUDEPATH += /opt/kde3/include 
-INCLUDEPATH += /opt/kde/include
-INCLUDEPATH += /usr/kde/3.5/include
+INCLUDEPATH += /opt/kde/include 
 		
 				
 HEADERS	= amfs-xml.h \
@@ -25,30 +21,26 @@
 	label-pixmap.cpp
 
 LIBS += -lkdecore
-LIBS += -L/opt/kde3/lib
-LIBS += -L/usr/local/kde3/lib
-LIBS += -L/usr/lib/qt3
 LIBS += -L/opt/kde/lib
-LIBS += -L/usr/kde/3.5/lib
 LIBS += -lkio
 		
-images.path = /usr/share/amarokFS/images
+images.path = $(DESTDIR)/opt/kde/share/apps/amarokFS/images
 images.files = images/*
 
-defaultTheme.path = /usr/share/amarokFS
+defaultTheme.path = $(DESTDIR)/opt/kde/share/apps/amarokFS
 defaultTheme.files = theme.xml
 		
-target.path = /usr/local/bin
+target.path = $(DESTDIR)/opt/kde/bin
 		
 icon.files = amarokFS.png
-icon.path = /usr/share/icons/
+icon.path = $(DESTDIR)/opt/kde/share/icons/
 		
 desktop.files = amarokFS.desktop
-desktop.path = /usr/share/applications/
+desktop.path = $(DESTDIR)/opt/kde/share/applications/
 		
 docs.files = README
 docs.files += theme-howto.txt
-docs.path = /usr/share/doc/amarokFS
+docs.path = $(DESTDIR)/opt/kde/share/doc/amarokFS
 
 
 INSTALLS += target
diff -urN amarokFS-0.5/amarokFS.desktop amarokFS-0.5-arch/amarokFS.desktop
--- amarokFS-0.5/amarokFS.desktop	2007-02-16 20:41:04.000000000 +0100
+++ amarokFS-0.5-arch/amarokFS.desktop	2007-04-30 11:34:23.000000000 +0200
@@ -5,7 +5,7 @@
 Exec=amarokFS
 GenericName=Full screen mode for the Amarok player
 GenericName[en_US]=Full screen mode for the Amarok player
-Icon=/usr/share/icons/amarokFS.png
+Icon=/opt/kde/share/icons/amarokFS.png
 MimeType=
 Name=Amarok Full Screen
 Name[en_US]=Amarok Full Screen
diff -urN amarokFS-0.5/amfs-xml.cpp amarokFS-0.5-arch/amfs-xml.cpp
--- amarokFS-0.5/amfs-xml.cpp	2007-03-19 13:35:38.000000000 +0100
+++ amarokFS-0.5-arch/amfs-xml.cpp	2007-04-30 11:38:27.000000000 +0200
@@ -45,9 +45,9 @@
 	//cout << themeCustomPrepath << "\n";
 	
 	//set the default path
-	QString themePath = "/usr/share/amarokFS/theme.xml";
+	QString themePath = "/opt/kde/share/apps/amarokFS/theme.xml";
 	//default prepath for the images directory	
-	QString themeImagesPrePath = "/usr/share/amarokFS/";
+	QString themeImagesPrePath = "/opt/kde/share/apps/amarokFS/";
 		
 	//if the custom theme file exists, use it, otherwise use the default theme
 	if(QFile(themeCustomPath).exists()) {
@@ -120,7 +120,7 @@
 
 	contextMenu->insertSeparator();
 	aboutAct = new QAction(tr("&About"), 0, this);
-	aboutAct->setIconSet(QIconSet(QPixmap("/usr/share/icons/amarokFS.png")));
+	aboutAct->setIconSet(QIconSet(QPixmap("/opt/kde/share/icons/amarokFS.png")));
 	aboutAct->setStatusTip(tr("Show About Dialog."));
 	connect(aboutAct, SIGNAL(activated()), this, SLOT(launchAbout()));
 	aboutAct->addTo(contextMenu);
@@ -597,7 +597,7 @@
 	reflectionPixmap = cdReflected;
 	
 	QImage gradientImage;
-	gradientImage.load( "/usr/share/amarokFS/images/gradient.png");
+	gradientImage.load( "/opt/kde/share/apps/amarokFS/images/gradient.png");
 	gradientImage = gradientImage.scale(reflectionPixmap.width(), 120);
 	gradientPixmap = gradientImage;
 	gradient->setPixmap(gradientPixmap);
@@ -684,8 +684,8 @@
 		result.load(coverPathDCOPtagcover);
 	}
 	if (result.isNull()) {
-		if(transDef == TRUE) result.load("/usr/share/amarokFS/images/logoB.png");
-		else result.load("/usr/share/amarokFS/images/logo.png");
+		if(transDef == TRUE) result.load("/opt/kde/share/apps/amarokFS/images/logoB.png");
+		else result.load("/opt/kde/share/apps/amarokFS/images/logo.png");
 	}
 	return result;
 	
