--- bootstrap.py	2011-10-18 12:20:42.084632296 -0500
+++ bootstrap.py	2011-10-18 12:21:52.198363940 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import os, stat, sys, platform, subprocess
 home = 'https://subversion.xor.aps.anl.gov/EXPGUI/'
 
@@ -49,31 +49,3 @@
     print ""
     p = subprocess.call(cmd)
 
-if sys.platform.startswith('linux'):
-    os.chmod(gsaspath + "/expgui/expgui", 
-             stat.S_IXUSR | stat.S_IRUSR | stat.S_IRGRP | stat.S_IXGRP | stat.S_IXOTH) 
-    if os.path.exists(os.path.expanduser('~/Desktop/')):
-        open(os.path.expanduser('~/Desktop/EXPGUI.desktop'),'w').write('''
-[Desktop Entry]
-Encoding=UTF-8
-Version=1.0
-Type=Application
-Terminal=false
-Exec=%s/expgui/expgui
-Name=EXPGUI
-Icon=%s/EXPGUI.png
-''' % (gsaspath,gsaspath))
-        os.chmod(os.path.expanduser('~/Desktop/EXPGUI.desktop'),
-                 stat.S_IWUSR | stat.S_IXUSR | stat.S_IRUSR | stat.S_IRGRP | stat.S_IXGRP | stat.S_IXOTH) 
-        open(os.path.expanduser('~/Desktop/GSAS.desktop'),'w').write('''
-[Desktop Entry]
-Encoding=UTF-8
-Version=1.0
-Type=Application
-Terminal=true
-Exec=%s/GSAS
-Name=GSAS
-Icon=%s/GSAS.png
-''' % (gsaspath,gsaspath))
-        os.chmod(os.path.expanduser('~/Desktop/GSAS.desktop'),
-                 stat.S_IWUSR | stat.S_IXUSR | stat.S_IRUSR | stat.S_IRGRP | stat.S_IXGRP | stat.S_IXOTH) 
