diff -ur unsort-0.5/Makefile unsort-0.5-ott/Makefile
--- unsort-0.5/Makefile	2005-02-23 10:17:59.000000000 +0100
+++ unsort-0.5-ott/Makefile	2005-06-20 19:54:23.000000000 +0200
@@ -3,20 +3,20 @@
 
 OBJS=unsort.o
 
+prefix=/usr
+
 all: unsort
 
 unsort: $(OBJS)
 	$(CC) -Wall -W $(OBJS) $(LDFLAGS) $(LDFLAGSf) -o unsort
-	strip unsort
 
 install: unsort
-	cp unsort /usr/local/bin
-	install -D unsort.1 /usr/local/man/man1/unsort.1
-	gzip -9 /usr/local/man/man1/unsort.1
+	cp unsort $(prefix)/bin
+	install -D unsort.1 $(prefix)/man/man1/unsort.1
 
 uninstall: clean
-	rm -f /usr/local/bin/unsort
-	rm -f /usr/local/man/man1/unsort.1.gz
+	rm -f $(prefix)/bin/unsort
+	rm -f $(prefix)/man/man1/unsort.1.gz
 
 clean:
 	rm -f $(OBJS) unsort core unsort-$(VERSION).tgz
