diff -rupN surf/config.def.h surf-new/config.def.h
--- surf/config.def.h	2012-09-20 13:01:21.430100363 +0200
+++ surf/config.def.h	2012-09-20 13:02:06.966767099 +0200
@@ -3,9 +3,9 @@ static char *useragent      = "Surf/"VER
 static char *progress       = "#0066FF";
 static char *progress_untrust = "#FF6600";
 static char *progress_trust = "#00FF00";
-static char *stylefile      = ".surf/style.css";
-static char *scriptfile     = ".surf/script.js";
-static char *cookiefile     = ".surf/cookies.txt";
+static char *stylefile      = "surf/style.css";
+static char *scriptfile     = "surf/script.js";
+static char *cookiefile     = "surf/cookies.txt";
 static time_t sessiontime   = 3600;
 static char *cafile         = "/etc/ssl/certs/ca-certificates.crt";
 static char *strictssl      = FALSE; /* Refuse untrusted SSL connections */
diff -rupN surf/surf.c surf-new/surf.c
--- surf/surf.c	2012-09-20 13:01:21.430100363 +0200
+++ surf/surf.c	2012-09-20 13:02:22.376767119 +0200
@@ -125,7 +125,7 @@ buildpath(const char *path) {
 	if(path[0] == '/')
 		apath = g_strdup(path);
 	else
-		apath = g_strconcat(g_get_home_dir(), "/", path, NULL);
+		apath = g_strconcat(g_get_user_config_dir(), "/", path, NULL);
 	if((p = strrchr(apath, '/'))) {
 		*p = '\0';
 		g_mkdir_with_parents(apath, 0700);
