--- parser/mpTokenReader.cpp	2011-02-14 02:05:26.000000000 +0100
+++ parser/mpTokenReader.cpp	2011-09-05 16:25:58.000000000 +0200
@@ -224,7 +224,11 @@
   */
   void TokenReader::SetExpr(const string_type &a_sFormula)
   {
-    m_sExpr = a_sFormula;
+    #if (__GNUC__ == 4) && (__GNUC_MINOR__ > 5)
+       m_sExpr = a_sFormula + string_type(_T(" "));
+    #else
+       m_sExpr = a_sFormula;
+    #endif
     m_vTokens.clear();
     ReInit();
   }
