diff --git a/Hellanzb/NZBLeecher/Protocol.py b/Hellanzb/NZBLeecher/Protocol.py
index 0c67779..1a5c91f 100644
--- a/Hellanzb/NZBLeecher/Protocol.py
+++ b/Hellanzb/NZBLeecher/Protocol.py
@@ -655,7 +655,7 @@ class NZBLeecher(NNTPClient, TimeoutMixin):
         reactor.callInThread(decode, segment)
 
     def gotGroup(self, group):
-        group = group[3]
+        group = self.gettingGroup
         self.activeGroups.append(group)
         self.gettingGroup = None
         debug(str(self) + ' got GROUP: ' + group)
