--- wscript	2010-03-30 19:20:36.000000000 +0200
+++ wscript	2010-05-01 21:37:00.279390037 +0200
@@ -167,11 +167,7 @@
     register = [os.path.join(bld.env['PYGTK_DEFSDIR'], 'pango-types.defs'),
                 os.path.join(bld.env['PYGTK_DEFSDIR'], 'gdk-types.defs'),
                 os.path.join(bld.env['PYGTK_DEFSDIR'], 'gtk-types.defs'),
-                os.path.join(bld.env['GNOME_PYTHON_DEFSDIR'], 'bonobo-types.defs'),
-                os.path.join(bld.env['GNOME_PYTHON_DEFSDIR'], 'bonoboui-types.defs'),
                 ] + list(register)
-
-    cmd.command.extend(['--load-types', os.path.join(bld.env['GNOME_PYTHON_ARG_TYPES_DIR'], 'bonobo-arg-types.py')])
     
     for load in local_load_types:
         cmd.source.append(load)
