--- data/fonts/Makefile.in	2011-02-26 19:07:32.000000000 +0100
+++ data/fonts/Makefile1.in	2012-07-31 18:53:27.014016568 +0200
@@ -511,23 +511,23 @@
 
 
 install: install-data-am
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	@echo "Preparing fontdir, please wait..." ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	if [ ! `fc-cache --list > /dev/null 2>&1` ] ; then \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  if test "x${HOME}" != "x" ; then \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    mkdir ${HOME}/.fonts > /dev/null 2>&1 ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    cp $(fontsdir)/CBM.ttf ${HOME}/.fonts ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    fc-cache ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  else \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	    echo '$$HOME not set... giving up.' ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  fi \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	else \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	  echo 'fc-cache not available... giving up.' ; \
-@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	fi
-@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	@echo "Preparing fontdir..." ; \
-@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	cd $(DESTDIR)$(fontsdir) ; \
-@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	bdftopcf -o vice-cbm.pcf vice-cbm.bdf ; \
-@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	rm vice-cbm.bdf ; \
-@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	mkfontdir .
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#@echo "Preparing fontdir, please wait..." ; \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#if [ ! `fc-cache --list > /dev/null 2>&1` ] ; then \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#  if test "x${HOME}" != "x" ; then \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#    mkdir ${HOME}/.fonts > /dev/null 2>&1 ; \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#    cp $(fontsdir)/CBM.ttf ${HOME}/.fonts ; \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#    fc-cache ; \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#  else \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#    echo '$$HOME not set... giving up.' ; \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#  fi \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#else \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#  echo 'fc-cache not available... giving up.' ; \
+@HAVE_FC_CACHE_TRUE@@SDL_COMPILE_FALSE@	#fi
+@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	#@echo "Preparing fontdir..." ; \
+@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	#cd $(DESTDIR)$(fontsdir) ; \
+@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	#bdftopcf -o vice-cbm.pcf vice-cbm.bdf ; \
+@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	#rm vice-cbm.bdf ; \
+@HAVE_FC_CACHE_FALSE@@SDL_COMPILE_FALSE@	#mkfontdir .
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
