--- conftest.orig	2012-05-06 14:53:27.624398648 +0100
+++ conftest.sh	2012-05-06 14:54:04.994193275 +0100
@@ -512,7 +512,7 @@
             # and if it as an 'event' member.
             #
             echo "$CONFTEST_PREAMBLE
-            #include <asm/system.h>
+            #include <asm/switch_to.h>
             #include <linux/pm.h>
             void conftest_pm_message_t(pm_message_t state) {
                 pm_message_t *p = &state;
