--- conftest.sh.old	2012-02-14 09:47:06.748005709 +0100
+++ conftest.sh	2012-02-14 09:47:21.434671886 +0100
@@ -126,9 +126,9 @@
     CFLAGS="$BASE_CFLAGS $MACH_CFLAGS $OUTPUT_CFLAGS -I$HEADERS $AUTOCONF_CFLAGS"
 
     if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" ]; then
-        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include"
+        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include -I$SOURCES/arch/x86/include/generated"
     elif [ "$ARCH" = "ARMv7" ]; then
-        CFLAGS="$CFLAGS -I$SOURCES/arch/arm/include"
+        CFLAGS="$CFLAGS -I$SOURCES/arch/arm/include -I$SOURCES/arch/arm/include/generated"
     fi
     if [ -n "$BUILD_PARAMS" ]; then
         CFLAGS="$CFLAGS -D$BUILD_PARAMS"
