diff -ruN StormLib/Makefile.am StormLib.fix/Makefile.am
--- StormLib/Makefile.am	1970-01-01 08:00:00.000000000 +0800
+++ StormLib.fix/Makefile.am	2008-11-19 18:42:05.000000000 +0800
@@ -0,0 +1 @@
+SUBDIRS = stormlib
diff -ruN StormLib/Makefile.config StormLib.fix/Makefile.config
--- StormLib/Makefile.config	2003-04-09 10:50:48.000000000 +0800
+++ StormLib.fix/Makefile.config	1970-01-01 08:00:00.000000000 +0800
@@ -1,21 +0,0 @@
-########################################################################
-#
-# Makefile.config for compiling StormLib under linux
-#  
-#   set constants here (CFLAGS, LDFLAGS, the compiler to use, etc.)
-#
-# Author: Marko Friedemann <marko.friedemann@bmx-chemnitz.de>
-# Created at: Mon Jan 29 18:26:01 CEST 2001
-# Computer: whiplash.flachland-chemnitz.de 
-# System: Linux 2.4.0 on i686
-#    
-# Copyright (c) 2001 BMX-Chemnitz.DE All rights reserved.
-#
-#####################################################################G++ = g++
-#G++ = g++-3.0.1
-
-CFLAGS_ADD = -Wall
-LDFLAGS_ADD = 
-
-# installation flags: "-s" => strip; dont use together with CFLAG -g if you want to actually debug ;)
-I_FLAGS_ADD = -s
diff -ruN StormLib/autogen.sh StormLib.fix/autogen.sh
--- StormLib/autogen.sh	1970-01-01 08:00:00.000000000 +0800
+++ StormLib.fix/autogen.sh	2008-11-19 18:42:05.000000000 +0800
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+
+touch AUTHORS COPYING ChangeLog INSTALL NEWS README
+libtool_dir=/usr/share/libtool
+cp $libtool_dir/config/ltmain.sh .
+cp $libtool_dir/config/compile .
+cp $libtool_dir/config/depcomp .
+
+set -x
+aclocal
+autoheader
+autoconf
+automake -a
+
diff -ruN StormLib/configure.in StormLib.fix/configure.in
--- StormLib/configure.in	1970-01-01 08:00:00.000000000 +0800
+++ StormLib.fix/configure.in	2008-11-19 19:33:47.000000000 +0800
@@ -0,0 +1,54 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.63])
+AC_INIT([stormlib], [6.23], [[email]zarraxx@gmail.com[/email]])
+AC_CONFIG_SRCDIR([stormlib/SFileOpenFileEx.cpp])
+AC_CONFIG_HEADERS([config.h])
+lt_major=6
+lt_age=23
+lt_revision=0
+dist_version=6.0.23
+AM_INIT_AUTOMAKE(stormlib, $dist_version)
+AC_SUBST(lt_major)
+AC_SUBST(lt_revision)
+AC_SUBST(lt_age)
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_INSTALL
+AC_PROG_LN_S
+AC_PROG_RANLIB
+AC_PROG_LIBTOOL
+# Checks for libraries.
+# FIXME: Replace `main' with a function in `-lbz2':
+AC_CHECK_LIB([bz2], [BZ2_bzopen])
+# FIXME: Replace `main' with a function in `-lz':
+AC_CHECK_LIB([z], [gzopen])
+
+# Checks for header files.
+AC_CHECK_HEADERS([fcntl.h stdint.h stdlib.h string.h unistd.h utime.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_TYPE_INT16_T
+AC_TYPE_INT32_T
+AC_TYPE_INT64_T
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_CHOWN
+AC_FUNC_ERROR_AT_LINE
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([ftruncate memmove memset pow rint sqrt strerror strrchr strstr utime])
+
+AC_CONFIG_FILES([Makefile
+                 stormlib/Makefile
+                 stormlib/pklib/Makefile
+		])
+#AC_CONFIG_SUBDIRS([stormlib])
+AC_OUTPUT(stormlib/Stormlib.pc)
diff -ruN StormLib/makefile StormLib.fix/makefile
--- StormLib/makefile	2004-06-12 13:14:34.000000000 +0800
+++ StormLib.fix/makefile	1970-01-01 08:00:00.000000000 +0800
@@ -1,58 +0,0 @@
-########################################################################
-#
-# Makefile for compiling StormLib under linux
-#
-# Author: Marko Friedemann <marko.friedemann@bmx-chemnitz.de>
-# Created at: Mon Jan 29 18:26:01 CEST 2001
-# Computer: whiplash.flachland-chemnitz.de 
-# System: Linux 2.4.0 on i686
-#    
-# Copyright (c) 2001 BMX-Chemnitz.DE All rights reserved.
-#
-########################################################################
-
-FILES.cpp = SCommon.cpp SFileCreateArchiveEx.cpp SFileFindFile.cpp \
-            SFileOpenArchive.cpp SFileOpenFileEx.cpp SFileReadFile.cpp \
-            SFileCompress.cpp SFileDecompress.cpp \
-            SListFile.cpp \
-            wavunp/wavunp1.cpp wavunp/wavunp2.cpp \
-            dcl/explode.c \
-FILES.o = $(FILES.cpp:.cpp=.o)
-
-include Makefile.config
-
-LIB = libStorm.so
-
-CFLAGS = $(CFLAGS_ADD) -D__SYS_ZLIB
-I_FLAGS = $(I_FLAGS_ADD)
-LDFLAGS = $(LDFLAGS_ADD) -lz
-
-all: .dep $(LIB)
-
-$(LIB): $(FILES.o)
-	ld -shared $(LDFLAGS) -o $(LIB) $(FILES.o)
-
-include .dep
-
-%.o: %.cpp
-	$(G++) $(CFLAGS) -c $< -o $@
-
-clean:
-	-rm $(FILES.o)
-
-new: clean all
-
-mrproper: clean
-	-rm $(LIB) .dep
-
-mrnew: mrproper new
-
-install: $(LIB)
-	install $(I_FLAGS) $(LIB) /usr/local/lib
-	mkdir -p /usr/local/include/StormLib
-	cp Storm.h /usr/local/include/StormLib
-	cp StormPort.h /usr/local/include/StormLib
-	ldconfig
-
-.dep:
-	$(G++) -MM $(CFLAGS) $(FILES.cpp) > .dep
diff -ruN StormLib/stormlib/Makefile StormLib.fix/stormlib/Makefile
--- StormLib/stormlib/Makefile	2008-10-06 12:38:44.000000000 +0800
+++ StormLib.fix/stormlib/Makefile	1970-01-01 08:00:00.000000000 +0800
@@ -1,63 +0,0 @@
-#####################################################################
-###
-#
-# Makefile for compiling StormLib under linux
-#
-# Author: Marko Friedemann <marko.friedemann@bmx-chemnitz.de>
-# Created at: Mon Jan 29 18:26:01 CEST 2001
-# Computer: whiplash.flachland-chemnitz.de 
-# System: Linux 2.4.0 on i686
-#    
-# Copyright (c) 2001 BMX-Chemnitz.DE All rights reserved.
-#
-#####################################################################
-###
-
-##############################################
-# updated on October 6, 2008 by Trevor Hogan #
-##############################################
-
-C++ = g++
-CC = gcc
-DFLAGS = -D__SYS_ZLIB
-OFLAGS =
-LFLAGS = -lbz2 -lz
-CFLAGS = -fPIC
-CFLAGS += $(OFLAGS) $(DFLAGS)
-
-OBJS = SCommon.o SCompression.o SFileCompactArchive.o \
-       SFileCreateArchiveEx.o SFileExtractFile.o SFileFindFile.o \
-       SListFile.o SFileOpenArchive.o SFileOpenFileEx.o \
-       SAttrFile.o SFileReadFile.o StormPortLinux.o \
-       wave/wave.o \
-       huffman/huff.o \
-       misc/crc32.o misc/md5.o
-
-COBJS = pklib/crc32.o pklib/explode.o pklib/implode.o
-
-LIB = libStorm.so
-
-all: $(OBJS) $(COBJS) $(LIB)
-
-$(LIB): $(OBJS) $(COBJS)
-	$(C++) -shared -o $(LIB) $(OBJS) $(COBJS) $(LFLAGS)
-
-clean:
-	rm -f $(OBJS) $(COBJS) $(LIB)
-
-$(OBJS): %.o: %.cpp
-	$(C++) -o $@ $(CFLAGS) -c $<
-
-$(COBJS): %.o: %.c
-	$(CC) -o $@ $(CFLAGS) -c $<
-
-$(LIB): $(OBJS) $(COBJS)
-
-all: $(LIB)
-
-install: $(LIB)
-	install $(LIB) /usr/local/lib
-	mkdir -p /usr/local/include/StormLib
-	cp StormLib.h /usr/local/include/StormLib
-	cp StormPort.h /usr/local/include/StormLib
-	ldconfig
diff -ruN StormLib/stormlib/Makefile.am StormLib.fix/stormlib/Makefile.am
--- StormLib/stormlib/Makefile.am	1970-01-01 08:00:00.000000000 +0800
+++ StormLib.fix/stormlib/Makefile.am	2008-11-19 19:36:54.000000000 +0800
@@ -0,0 +1,33 @@
+#MAINTAINERCLEANFILES = Makefile.in
+SUBDIRS = pklib
+INCLUDES = -I../ -I./
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = Stormlib.pc
+installincludedir=$(includedir)/Stormlib
+installinclude_HEADERS= StormLib.h StormPort.h
+#AM_CPPFLAGS=-DINSTALL_PREFIX="\"$(prefix)\""
+lib_LTLIBRARIES = libStorm.la
+libStorm_la_LDFLAGS = -version-info @lt_major@:@lt_age@:@lt_revision@
+libStorm_la_SOURCES = \
+					  StormLib.h \
+                      StormPort.h \
+                      SCommon.cpp \
+                      SCompression.cpp \
+                      SFileCompactArchive.cpp \
+                      SFileCreateArchiveEx.cpp \
+                      SFileExtractFile.cpp \
+                      SFileFindFile.cpp \
+                      SListFile.cpp \
+		              SFileOpenArchive.cpp \
+                      SFileOpenFileEx.cpp \
+                      SAttrFile.cpp \
+                      SFileReadFile.cpp \
+                      StormPortLinux.cpp \
+		              wave/wave.cpp \
+                      huffman/huff.cpp \
+                      misc/crc32.cpp \
+                      misc/md5.cpp 
+libStorm_la_LIBADD = pklib/libpk.la
+                      
+
+
diff -ruN StormLib/stormlib/SFileFindFile.cpp StormLib.fix/stormlib/SFileFindFile.cpp
--- StormLib/stormlib/SFileFindFile.cpp	2008-04-22 15:15:46.000000000 +0800
+++ StormLib.fix/stormlib/SFileFindFile.cpp	2008-11-19 19:25:30.000000000 +0800
@@ -147,7 +147,7 @@
         pNode = ha->pListFile[hs->dwNextIndex++];
 
         // If this entry is free, do nothing
-        if(pHash->dwBlockIndex < HASH_ENTRY_DELETED && (DWORD_PTR)pNode < LISTFILE_ENTRY_DELETED)
+        if(pHash->dwBlockIndex < HASH_ENTRY_DELETED && (unsigned long)pNode < (unsigned long)LISTFILE_ENTRY_DELETED)
         {
             // Check the file name.
             if(CheckWildCard(pNode->szFileName, hs->szSearchMask))
diff -ruN StormLib/stormlib/StormPortLinux.cpp StormLib.fix/stormlib/StormPortLinux.cpp
--- StormLib/stormlib/StormPortLinux.cpp	2008-10-07 15:33:46.000000000 +0800
+++ StormLib.fix/stormlib/StormPortLinux.cpp	2008-11-19 21:10:10.000000000 +0800
@@ -72,22 +72,24 @@
 }
 
 HANDLE CreateFile(const char *sFileName, DWORD ulMode, DWORD ulSharing, void *pSecAttrib, DWORD ulCreation, DWORD ulFlags, HANDLE hFile)
-{
+{
+#define FILE_MODE (S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH)
     switch (ulCreation)
     {
     case OPEN_EXISTING:
         return (HANDLE)open(sFileName, O_RDONLY | O_LARGEFILE);
 
     case OPEN_ALWAYS:
-        return (HANDLE)open(sFileName, O_RDWR | O_CREAT);
+        return (HANDLE)open(sFileName, O_RDWR | O_CREAT ,FILE_MODE);
 
     case CREATE_ALWAYS:
     case CREATE_NEW:
-        return (HANDLE)open(sFileName, O_RDWR | O_CREAT | O_TRUNC);
+        return (HANDLE)open(sFileName, O_RDWR | O_CREAT | O_TRUNC ,FILE_MODE);
 
     default:
         return INVALID_HANDLE_VALUE;
-    }
+    }
+#undef FILE_MODE
 }
 
 BOOL CloseHandle(HANDLE hFile)
diff -ruN StormLib/stormlib/Stormlib.pc.in StormLib.fix/stormlib/Stormlib.pc.in
--- StormLib/stormlib/Stormlib.pc.in	1970-01-01 08:00:00.000000000 +0800
+++ StormLib.fix/stormlib/Stormlib.pc.in	2008-11-19 19:37:08.000000000 +0800
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: StormLib
+Description: StormLib library.
+Requires: 
+Version: @VERSION@
+Libs: -L${libdir} -lStorm -lbz2 -lz
+Cflags: -I${includedir}/Stormlib
diff -ruN StormLib/stormlib/pklib/Makefile.am StormLib.fix/stormlib/pklib/Makefile.am
--- StormLib/stormlib/pklib/Makefile.am	1970-01-01 08:00:00.000000000 +0800
+++ StormLib.fix/stormlib/pklib/Makefile.am	2008-11-19 19:10:59.000000000 +0800
@@ -0,0 +1,2 @@
+noinst_LTLIBRARIES = libpk.la
+libpk_la_SOURCES = crc32.c explode.c implode.c pklib.h
