--- animea	2011-07-03 22:19:23.000000000 +0200
+++ /home/jorge/Descargas/linux-manga-downloader_0.2.0.1-5Beta_all/usr/local/bin/Correcciones/animea	2013-02-28 17:04:08.062797678 +0100
@@ -3,7 +3,8 @@
 
 # /* Import options *\ #
 
-source ~/.LMD/options;
+source ~/.LMD/options
+source /usr/share/lmd/langs/$LMLANG;
 
 function MANGAEA_DN(){
 if ! echo $ODOWN | egrep "http://manga.animea.net/.*page-1.html" ;
@@ -33,28 +34,28 @@
 then
 	echo "$ANE1"
 else
-	echo "Se creara el directorio para el manga $BDIR"
+	echo "$ANE2"
 	mkdir $CDIR"$BDIR"
 fi
 
 
 if ls $CDIR"$BDIR" | grep "$DIRNAME" > /dev/null
 then
-	echo el directorio $DIRNAME ya existe introduce otro nombre.
+	echo $ANE3
 		read -p "" DIRNAME
 		if ls $CDIR"$BDIR" | grep "$DIRNAME" > /dev/null
 		then
-			echo Error con el nombre de directorio,\n compruebe el nombre y ejecute de nuevo el script.
+			echo $ANE4
 			exit
 		else
 			cd $CDIR"$BDIR"
-			echo se va a crear el directorio "$DIRNAME"
+			echo $ANE5
 			mkdir "$DIRNAME"
 			cd "$DIRNAME"
 		fi
 else
 	cd $CDIR"$BDIR"
-	echo se va a crear el directorio "$DIRNAME"
+	echo $ANE5
 	mkdir "$DIRNAME"
 	cd "$DIRNAME"
 fi
@@ -68,7 +69,7 @@
 echo
 #echo '\n' $DWNLINK \| $LAN \| $a
 #exit
-		echo Descargando pagina $i.
+		echo $ANE6
 		wget $DWNLINK -o /tmp/.logsesion
 done
 }
@@ -82,7 +83,7 @@
 		;;
 	*)
 		#Error
-		echo "error ese parametro no existe"
+		echo $ANE7
 		exit
 		;;
 esac
