--- ca_ES	2011-07-03 22:19:23.000000000 +0200
+++ correcciones/ca_CA	2013-03-06 16:10:00.270816425 +0100
@@ -1,5 +1,5 @@
 #
-#Catalan					0.1
+#Català					0.1
 #
 
 # /* GUI *\ #
@@ -7,25 +7,25 @@
 GULA1="Submanga en aquest moments compta amb dues modalitats:
 1- La normal
 2- La bulk
-Aquesta segona permet descarregar un manga complet, tots els capitols publicats de manera automatica:
+Aquesta segona permet descarregar un manga complet, tots els capítols publicats de manera automàtica:
 http://submanga.com/claymore/completa
 També permet 
-la opció de descarrega de un fansub especific:
+la opció de descàrrega de un fansub específic:
 http://submanga.com/The_World_God_Only_Knows/scanlation/Nikasub"
 ###
 ###
-GULA2="<b>1- 4chan</b> - Per a 4chan es necessari introduir una url del thread
+GULA2="<b>1- 4chan</b> - Per a 4chan és necessari introduir una url del thread
 ej: http://boards.4chan.org/h/res/3025102
-(no intenteu accedir amb aquesta url, el thread no existira XD)
-Després de introduir la url s'ha de introduir un nom per el directori en el segon bloc, de no posarlo es aplicara un nom random.
+(no intenteu accedir amb aquesta url, el thread no existirà XD)
+Després de introduir la url s'ha de introduir un nom per el directori en el segon bloc, de no posar-ho s'aplicarà un nom aleatori.
 
 <b>2- Zerochan</b> - Posar la url de la galeria a descarregar
-ej: http://www.zerochan.net/Durarara%21%21?q=durarara, la pagina 1 per a menys problemes ;)
+ej: http://www.zerochan.net/Durarara%21%21?q=durarara, la pàgina 1 per a menys problemes ;)
 
-<b>3- Boru</b> - Aquest es per a les pagines fetes amb Danboru un CMS per a debian especific de galeries de imatges, aquest plug es funcional en varies pagines fetes amb Danboru, si en alguna falla aviseu i es intentara fer compatible.
+<b>3- Boru</b> - Aquest és per a les pagines fetes amb Danboru un CMS per a debian éspecific de galeries de imatges, aquest plug és funcional en varies pagines fetes amb Danboru, si en alguna falla aviseu i es intentara fer compatible.
 url ej: http://konachan.com/post/index?tags=angel_beats%21
 
-<b>**Es avisa els Danboru son famosos per la seva quantitat nefasta de hentai, tot i aixis solen tenir molt bons wallpapers**</b>
+<b>**Es avisa els Danboru son famosos per la seva quantitat nefasta de hentai, tot i aixi solen tindre molt bons wallpapers**</b>
 
 <b>4- Pown-it</b> - Flashes super repetitius i mega adictius.
 url ej:http://www.pown.it/5224"
@@ -33,39 +33,39 @@
 ## GUI buttons and actions. ##
 
 GULA_HELPB="Ajuda"
-GULA_DOWNLOAD="Descarrega"
+GULA_DOWNLOAD="Descàrrega"
 GULA_APLY="Aplicar"
-GULA_CLOSE="Tencar"
+GULA_CLOSE="Tancar"
 GULA_OK="OK"
 GULA_OPC="Opcions"
-GULA_SERV="Servidores"
-GULA_CFGB="Configuració del LMD"
+GULA_SERV="Servidors"
+GULA_CFGB="Configuració de LMD"
 
 ## GUI CONF DIALOG
 
-GULA_CONF="Configuració del LMD"
+GULA_CONF="Configuració de LMD"
 GULA_CONF1="Configuració"
 GULA_CONF2="Directori Personalitzat"
 GULA_CONF3="Selecciona un Directori"
 GULA_CONF4="Selecciona Idioma"
-GULA_CONF5="Guarda Configuració?"
+GULA_CONF5="Guardar Configuració?"
 
 ## GUI SUBMENU SMA
 
-GULA_SMA="Descarrega Normal - Descarrega un capítol de submanga"
+GULA_SMA="Descàrrega Normal - Descarrega un capítol de submanga"
 GULA_SMA1="Bulk Downloader - Descarrega un manga complet **help**"
-GULA_SMA2="Submanga List – Llistà els últims mangas publicats"
+GULA_SMA2="Submanga List – Llista els últims mangas publicats"
 
 ## GUI SUBMENU MCA
 
-GULA_MCA="Descarga Normal - Descarga un capitulo de mcanime"
-GULA_MCA1="Mcanime List - Lista los ultimos mangas publidados"
+GULA_MCA="Descàrrega Normal - Descarrega un capítol de mcanime"
+GULA_MCA1="Mcanime List – Llista els últims mangas publicats"
 
 ## GUI SUBMENU RAWS
 
 GULA_RAWS="MangaHead - Raws i E-Scans"
 GULA_RAWS1="MangaFox - E-Scans"
-GULA_RAWS2="MangaShare  - El manga compartit es millor"
+GULA_RAWS2="MangaShare  - El manga compartit és millor"
 GULA_RAWS3="MangaReader"
 GULA_RAWS4="Animea"
 
@@ -82,38 +82,51 @@
 GULA_OTS3="Danboru - Galeries"
 GULA_OTS4="Pown.it - Flashes super repetitius i mega adictius"
 
+##GENERAL
+
+conf="s"
+CONF="S"
+neg="n"
+NEG="N"
+
 
 # Submanga
 
-SMA1="El directori $BDIR ja existeix no es necessari crear ne un altre."
-SMA2="Es creara el directori per al manga"
+SMA1="El directori $BDIR ja existeix no és necessari crear-ne un altre."
+SMA2="Es crearà el directori per al manga"
 SMA3="El directori $DIRNAME ja existeix introdueix un altre nom."
 SMA4="Error amb el nombre de directori,\n comprova el nom i executa de nou el script."
 SMA5="Es creara el directori"
-SMA6="Descarregant la pagina"
-SMA7="Error al descarregar la pagina"
-SMA8="Descarrega Completa"
-SMA9="Es descarregara el manga"
+SMA6="Descarregant la pàgina"
+SMA7="Error al descarregar la pàgina"
+SMA8="Descàrrega Completa"
+SMA9="Es descarregarà el manga"
 SMA10="A continuació es descarregaran $BUL capítols."
-SMA11="¿Desitja continuar la descarrega des de un capítol determinat?"
-SMA12="Introdueix el numero del capítol \"ej: 37\" "
+SMA11="¿Desitja continuar la descàrrega des de un capítol determinat?"
+SMA12="Introdueix el número del capítol \"Ej: 37\" "
 SMA13="Descarregat Correctament. "
-SMA14="Error aquest paràmetre no existeix"
+SMA14="Error, aquest paràmetre no existeix"
 SMA15="Submanga - Nous Mangas (Beta)"
 SMA16="Selecciona"
-SMA17="Numº"
+SMA17="Nº"
 SMA18="Manga"
 SMA19="Direcció"
 
+#4chan
+FCH1="Aquest manga ha estat descarregat anteriorment"
+FCH2="Es procedeix a descarregar les noves imatges del thread"
+FCH3="Es descarregaran $FILESN imatges. No tanquis el programa."
+FCH4="Descarregant........."
+FCH5="Descàrrega completa, s'han descarregat $FILESN imatges."
 
 # Mcanime
 
-MCA1="El directori $BDIR ja existeix no es necessari crear ne un altre."
+MCA1="El directori $BDIR ja existeix no és necessari crear-ne un altre."
 MCA2="El directori $DIRNAME ja existeix introdueix un altre nom."
 MCA3="Error amb el nombre de directori,\n comprova el nom i executa de nou el script."
 MCA4="Es creara el directori $BDIR"
 MCA5="Es creara el directori $DIRNAME"
-MCA6="Error Descarregant la pagina $i"
+MCA6="Error Descarregant la pàgina $i"
 MCA7="Descarrega Completa."
 MCA8="Descarregant"
 
@@ -122,9 +135,9 @@
 
 MAS1="El directori $DIRNAME ja existeix introdueix un altre nom."
 MAS2="Error amb el nombre de directori,\n comprova el nom i executa de nou el script"
-MAS3="Es creara el directori \"$DIRNAME\""
-MAS4="Error Descarregant la pagina $i"
-MAS5="Descarrega completa"
+MAS3="Es crearà el directori \"$DIRNAME\""
+MAS4="Error Descarregant la pàgina $i"
+MAS5="Descàrrega completa"
 
 
 # MangaReader
@@ -132,27 +145,33 @@
 MAR1="El directori $DIRNAME ja existeix introdueix un altre nom."
 MAR2="Error amb el nombre de directori,\n comprova el nom i executa de nou el script."
 MAR3="Es creara el directori \"$DIRNAME\""
-MAR4="Descarregant pagina 1"
-MAR5="Descarregant la pagina $i"
-MAR6="Descarrega completa"
+MAR4="Descarregant pàgina 1"
+MAR5="Descarregant la pàgina $i"
+MAR6="Descàrrega completa"
 
 # G.E-Hentai
 
 GEH1="El directori $DIRNAME ja existeix introdueix un altre nom."
 GEH2="Error amb el nombre de directori,\n comprova el nom i executa de nou el script."
-GEH3="Es creara el directori "$DIRNAME""
-GEH4="Pagina $xi descarregada"
+GEH3="Es crearà el directori "$DIRNAME""
+GEH4="Pàgina $xi descarregada"
 
 # FutaHentai
 
 FTH1="El directori $DIRNAME ja existeix introdueix un altre nom."
 FTH2="Error amb el nombre de directori,\n comprova el nom i executa de nou el script."
-FTH3="Es creara el directori "$DIRNAME""
-FTH4="Pagina $i descarregada"
+FTH3="Es crearà el directori "$DIRNAME""
+FTH4="Pàgina $i descarregada"
 
 # Animea
 
-ANE1="El directori $BDIR ja existeix no es necessari crear ne un de nou"
+ANE1="El directori $BDIR ja existeix, no és necessari crear-ne un de nou"
+ANE2="Es crearà el directori per al manga $BDIR"
+ANE3="El directori $DIRNAME ja existeix, introdueix un altre nom."
+ANE4="Error amb el nom de directori,\n comprova el nom i ejecuta de nou el script."
+ANE5"Es va a crear el directori "$DIRNAME""
+ANE6="Descarregant pàgina $i."
+ANE7="Error eixe parametre no existeix"
 
 ##
 ##OtherS
@@ -160,19 +179,19 @@
 
 # Danboru
 
-DAN1="Descarregades $pga pagines."
+DAN1="Descarregades $pga pàgines."
 DAN2="Descarregant...."
 DAN3="No tanquis la terminal"
-DAN4="Descarrega completada, se han descarregat $FILESN imatges."
+DAN4="Descàrrega completada, se han descarregat $FILESN imatges."
 
 #Pown.it
 
-PNI1="0 no es un pown.it descarrega-ble"
+PNI1="0 no és un pown.it descarrega-ble"
 
 # Zerochan
 
 ZEO1="Imatge $pga de $FILESN descarregada"
-ZEO2="Pagina $pgs descarregada"
+ZEO2="Pàgina $pgs descarregada"
 
 
 
