--- es_ES	2013-03-01 11:47:07.840898081 +0100
+++ correcciones/es_ES	2013-03-06 16:09:46.787274584 +0100
@@ -9,20 +9,20 @@
 2- La bulk
 Esta segunda permite descargar un manga completo poniendo solo la url:
 http://submanga.com/claymore/completa
-o si lo queremos de un fansub especifico:
+o si lo queremos de un fansub específico:
 http://submanga.com/The_World_God_Only_Knows/scanlation/Nikasub"
 ###
 ###
 GULA2="<b>1- 4chan</b> - Para 4chan hay que introducir la url del thread ej: http://boards.4chan.org/h/res/3025102
-(no intenteis acceder por que el thread no existira XD)
-Despues de introducir la url en el primer bloque hay que introducir un nombre para el directorio en el segundo bloque, de no hacerlo se le aplicara uno random
+(no intenten acceder por que el thread no existirá XD)
+Después de introducir la url en el primer bloque hay que introducir un nombre para el directorio en el segundo bloque, de no hacerlo se le aplicará uno aleatorio
 
-<b>2- Zerochan</b> - Poner la url de la galeria que se desea descargar ej: http://www.zerochan.net/Durarara%21%21?q=durarara, la pagina 1 para menos problemas ;)
+<b>2- Zerochan</b> - Poner la url de la galería que se desea descargar ej: http://www.zerochan.net/Durarara%21%21?q=durarara, la página 1 para menos problemas ;)
 
-<b>3- Boru</b> - Este es para las paginas que estan echas con Danboru un CMS para debian especifico de galeria de imagenes, este plug es funcional en varias paginas echas con Danboru, si en alguna no funciona avisad y se provara su compatibilidad
+<b>3- Boru</b> - Este es para las páginas que están echas con Danboru un CMS para debian específico de galería de imágenes, este plug es funcional en varias páginas echas con Danboru, si en alguna no funciona avisen y se probará su compatibilidad
 url ej: http://konachan.com/post/index?tags=angel_beats%21
 
-<b>**Se avisa los Danboru son famosos por la cantidad nefasta de hentai, aun asi suelen tener muy buenos wallpapers**</b>
+<b>**Se avisa los Danboru son famosos por la cantidad nefasta de hentai, aun así suelen tener muy buenos wallpapers**</b>
 
 <b>4- Pown-it</b> - Flashes super repetitivos mega adictivos.
 url ej:http://www.pown.it/5224"
@@ -36,27 +36,27 @@
 GULA_OK="OK"
 GULA_OPC="Opciones"
 GULA_SERV="Servidores"
-GULA_CFGB="Configuracion del LMD"
+GULA_CFGB="Configuración de LMD"
 
 ## GUI CONF DIALOG
 
-GULA_CONF="Configuracion del LMD"
-GULA_CONF1="Configuracion"
+GULA_CONF="Configuración de LMD"
+GULA_CONF1="Configuración"
 GULA_CONF2="Directorio Personalizado"
-GULA_CONF3="Seleciona un Directorio"
-GULA_CONF4="Selecionar Idioma"
-GULA_CONF5="Guardar Configuracion?"
+GULA_CONF3="Selecciona un Directorio"
+GULA_CONF4="Seleccionar Idioma"
+GULA_CONF5="¿Guardar Configuración?"
 
 ## GUI SUBMENU SMA
 
-GULA_SMA="Descarga Normal - Descarga un capitulo de submanga"
+GULA_SMA="Descarga Normal - Descarga un capítulo de submanga"
 GULA_SMA1="Bulk Downloader - Descarga un manga completo **help**"
-GULA_SMA2="Submanga List - Lista los ultimos mangas publidados"
+GULA_SMA2="Submanga List - Lista los últimos mangas publicados"
 
 ## GUI SUBMENU MCA
 
 GULA_MCA="Descarga Normal - Descarga un capitulo de mcanime"
-GULA_MCA1="Mcanime List - Lista los ultimos mangas publidados"
+GULA_MCA1="Mcanime List - Lista los últimos mangas publicados"
 
 ## GUI SUBMENU RAWS
 
@@ -69,87 +69,106 @@
 ## GUI SUBMENU HEN
 
 GULA_HEN="FutaHentai - El mejor site de manga futa gratis +18"
-GULA_HEN1="E-Hentai - Grandes galerias de puro hentai & chicas monas +18"
+GULA_HEN1="E-Hentai - Grandes galerías de puro hentai & chicas monas +18"
 
 ## GUI SUBMENU OTS
 
 GULA_OTS="Otros Scripts"
-GULA_OTS1="4chan Thread Downloader - Descarga threads y galerias de 4chan"
-GULA_OTS2="Zerochan - Galerias y galerias de imagenes en HQ"
-GULA_OTS3="Danboru - Galerias"
+GULA_OTS1="4chan Thread Downloader - Descarga threads y galerías de 4chan"
+GULA_OTS2="Zerochan - Galerías y galerías de imágenes en HQ"
+GULA_OTS3="Danboru - Galerías"
 GULA_OTS4="Pown.it - Flashes super repetitivos mega adictivos"
 
+##GENERAL
+
+conf="s"
+CONF="S"
+neg="n"
+NEG="N"
 
 # Submanga
 
 SMA1="El directorio $BDIR ya existe no hace falta crear uno nuevo"
 SMA2="Se creara el directorio para el manga"
-SMA3="el directorio $DIRNAME ya existe introduce otro nombre."
+SMA3="El directorio $DIRNAME ya existe introduce otro nombre."
 SMA4="Error con el nombre de directorio,\n compruebe el nombre y ejecute de nuevo el script."
-SMA5="se va a crear el directorio"
+SMA5="Se va a crear el directorio"
 SMA6="Descargando pagina"
 SMA7="Error al descargar la pagina"
 SMA8="Descarga Completa"
-SMA9="Se descargara el manga"
-SMA10="A continuacion se descargaran $BUL capitulos."
-SMA11="¿Desea continuar la descarga des de un capitulo determinado?"
-SMA12="Introduzca el numero del capitulo \"ej: 37\" "
+SMA9="Se descargará el manga"
+SMA10="A continuación se descargaran $BUL capítulos."
+SMA11="¿Desea continuar la descarga desde un capítulo determinado?"
+SMA12="Introduzca el numero del capítulo \"ej: 37\" "
 SMA13="Descargado Correctamente. "
-SMA14="error ese parametro no existe"
+SMA14="Error, ese parámetro no existe"
 SMA15="Submanga - Nuevos Mangas (POC)"
 SMA16="Seleccionar"
-SMA17="Numº"
+SMA17="Nº"
 SMA18="Manga"
-SMA19="Direccion"
+SMA19="Dirección"
 
+#4chan
+FCH1="Este manga ha sido descargado anteriormente"
+FCH2="Se procederá a descargar las imágenes nuevas del thread"
+FCH3="Se descargarán $FILESN imágenes. No cierres el programa."
+FCH4="Descargando........."
+FCH5="Imagen $i de $FILESN descargada"
+FCH6="Descarga Completa, han sido descargadas $FILESN imágenes."
 
 # Mcanime
 
 MCA1="El directorio $BDIR ya existe no hace falta crear otro."
-MCA2="el directorio $DIRNAME ya existe introduce otro nombre."
+MCA2="El directorio $DIRNAME ya existe introduce otro nombre."
 MCA3="Error con el nombre de directorio,\n compruebe el nombre y ejecute de nuevo el script."
-MCA4="se va a crear el directorio $BDIR"
-MCA5="se va a crear el directorio $DIRNAME"
-MCA6="Error al descargar la pagina $i"
+MCA4="Se va a crear el directorio $BDIR"
+MCA5="Se va a crear el directorio $DIRNAME"
+MCA6="Error al descargar la página $i"
 MCA7="Descarga Completa."
 MCA8="Descargando"
 
 
 # MangaShare
 
-MAS1="el directorio $DIRNAME ya existe introduce otro nombre."
+MAS1="El directorio $DIRNAME ya existe introduce otro nombre."
 MAS2="Error con el nombre de directorio,'\n' compruebe el nombre y ejecute de nuevo el script."
-MAS3="se va a crear el directorio \"$DIRNAME\""
-MAS4="Error al descargar la pagina $i"
+MAS3="Se va a crear el directorio \"$DIRNAME\""
+MAS4="Error al descargar la página $i"
 MAS5="Descarga completada"
 
 
 # MangaReader
 
-MAR1="el directorio $DIRNAME ya existe introduce otro nombre."
+MAR1="El directorio $DIRNAME ya existe introduce otro nombre."
 MAR2="Error con el nombre de directorio,\n compruebe el nombre y ejecute de nuevo el script."
-MAR3="se va a crear el directorio \"$DIRNAME\""
-MAR4="Descargando pagina 1"
-MAR5="Descargando pagina $i"
+MAR3="Se va a crear el directorio \"$DIRNAME\""
+MAR4="Descargando página 1"
+MAR5="Descargando página $i"
 MAR6="Descarga completa"
 
 # G.E-Hentai
 
-GEH1="el directorio $DIRNAME ya existe introduce otro nombre."
+GEH1="El directorio $DIRNAME ya existe introduce otro nombre."
 GEH2="Error con el nombre de directorio,'\n' compruebe el nombre y ejecute de nuevo el script."
-GEH3="se va a crear el directorio "$DIRNAME""
-GEH4="pagina $xi descargada"
+GEH3="Se va a crear el directorio "$DIRNAME""
+GEH4="Página $xi descargada"
 
 # FutaHentai
 
-FTH1="el directorio $DIRNAME ya existe introduce otro nombre."
+FTH1="El directorio $DIRNAME ya existe introduce otro nombre."
 FTH2="Error con el nombre de directorio,\n compruebe el nombre y ejecute de nuevo el script."
-FTH3="se va a crear el directorio "$DIRNAME""
-FTH4="Pagina $i descargada"
+FTH3="Se va a crear el directorio "$DIRNAME""
+FTH4="Página $i descargada"
 
 # Animea
 
 ANE1="El directorio $BDIR ya existe no hace falta crear uno nuevo"
+ANE2="Se creara el directorio para el manga $BDIR"
+ANE3="El directorio $DIRNAME ya existe introduce otro nombre."
+ANE4="Error con el nombre de directorio,\n compruebe el nombre y ejecute de nuevo el script."
+ANE5="Se va a crear el directorio "$DIRNAME""
+ANE6="Descargando pagina $i."
+ANE7="Error ese parametro no existe"
 
 ##
 ##OtherS
@@ -157,10 +176,10 @@
 
 # Danboru
 
-DAN1="Descargadas $pga paginas."
+DAN1="Descargadas $pga páginas."
 DAN2="Descargando...."
 DAN3="No cierres la terminal"
-DAN4="Descarga completa, se han descargado $FILESN imagenes."
+DAN4="Descarga completa, se han descargado $FILESN imágenes."
 
 #Pown.it
 
@@ -169,7 +188,7 @@
 # Zerochan
 
 ZEO1="Imagen $pga de $FILESN descargada"
-ZEO2="Pagina $pgs descargada"
+ZEO2="Página $pgs descargada"
 
 
 
