--- lmd-fu	2011-07-03 22:19:24.000000000 +0200
+++ /home/jorge/Descargas/linux-manga-downloader_0.2.0.1-5Beta_all/usr/local/bin/Correcciones/lmd-fu	2013-02-28 16:11:42.407484288 +0100
@@ -112,14 +112,13 @@
 
 CDIR=\"$HOME/manga\"
 LMLANG=\"es_ES\"" > $HOME/.LMD/options
-
 echo "completed"
-notify-send "LMD" "La configuracion ha sido reseteada a valores por defecto"
+notify-send "LMD" "La configuración ha sido reseteada a valores por defecto"
 }
 
 if [ $USER == "root" ];
 then
-	echo "root user is not supported" 
+	echo "Root user is not supported" 
 else
 
 case $1 in
