--- /home/jorge/Descargas/linux-manga-downloader_0.2.0.1-5Beta_all/usr/local/bin/lmd-fu	2011-07-03 22:19:24.000000000 +0200
+++ /home/jorge/Descargas/linux-manga-downloader_0.2.0.1-5Beta_all/usr/local/bin/lmd-fu2	2013-02-28 03:17:31.317121334 +0100
@@ -110,7 +110,7 @@
 #Options File 						0.0.2.3
 #
 
-CDIR=\"$HOME/manga\"
+CDIR=\"$HOME/manga/\"
 LMLANG=\"es_ES\"" > $HOME/.LMD/options
 
 echo "completed"
