--- /home/jorge/Descargas/linux-manga-downloader_0.2.0.1-5Beta_all/usr/local/bin/lmd	2011-07-03 22:19:24.000000000 +0200
+++ /home/jorge/Descargas/linux-manga-downloader_0.2.0.1-5Beta_all/usr/local/bin/lmd2	2013-02-28 03:21:33.970873917 +0100
@@ -1,7 +1,8 @@
 #!/bin/bash
 #
 
-
+[ ! -d ~/.LMD ] && lmd-fu -reconf
+[ $? -ne 0 ] exit 1
 source ~/.LMD/options;
 source /usr/share/lmd/langs/$LMLANG;
 #function ESCAS(){
@@ -84,7 +85,7 @@
 <window title=\"Linux Manga Downloader v2 (Beta)\" default_height=\"250\" default_width=\"450\">
 <vbox>
 
-	<notebook labels=\"Inició|Submanga|Mcanime|Raws & E-Scans|Hentai|OtherS|Creditos\">
+	<notebook labels=\"Inicio|Submanga|Mcanime|Raws & E-Scans|Hentai|Others|Créditos\">
 	<vbox>
 		<frame>
 			<pixmap>
