--- /home/jorge/Descargas/linux-manga-downloader_0.2.0.1-5Beta_all/usr/share/applications/LMD.desktop	2011-07-03 22:19:24.000000000 +0200
+++ /home/jorge/Descargas/linux-manga-downloader_0.2.0.1-5Beta_all/usr/share/applications/LMD2.desktop	2013-02-28 18:30:33.729144585 +0100
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Name=LMD
 Version=1.0
-Exec=/usr/local/bin/lmd
 Comment=Download all your favorite manga
+Exec=lmd
 Icon=/usr/share/icons/pen.png
-Type=Application
 Terminal=true
+Type=Application
 StartupNotify=true
 Encoding=UTF-8
 Categories=Graphics;
\ No hay ningún carácter de nueva línea al final del fichero
