--- submanga	2011-07-03 22:19:24.000000000 +0200
+++ Correcciones/submanga	2013-03-06 15:51:55.080700922 +0100
@@ -4,7 +4,7 @@
 # /* Import options *\ #
 
 source ~/.LMD/options;
-source ~/.LMD/lang/$LMLANG;
+source /usr/share/lmd/langs/$LMLANG;
 
 ##### /* Functions *\ #####
 
@@ -20,7 +20,7 @@
 
 NAUL=`echo $ODOWN | cut -d "/" -f 4`
 BDIR=`echo $NAUL | tr "_" " "`
-#echo $CDIR"$BDIR"/"$DIRNAME"
+#echo $CDIR/"$BDIR"/"$DIRNAME"
 #exit
 
 if ls $CDIR | grep "$BDIR" > /dev/null
@@ -28,26 +28,26 @@
 	echo "$SMA1"
 else
 	echo "$SMA2 $BDIR"
-	mkdir "$CDIR$BDIR"
+	mkdir $CDIR/$BDIR
 fi
 
 
-if ls $CDIR"$BDIR" | grep "$DIRNAME" > /dev/null
+if ls $CDIR/"$BDIR" | grep "$DIRNAME" > /dev/null
 then
 	echo $SMA3
 		read -p "" DIRNAME
-		if ls $CDIR"$BDIR" | grep "$DIRNAME" > /dev/null
+		if ls $CDIR/"$BDIR" | grep "$DIRNAME" > /dev/null
 		then
 			echo $SMA4
 			exit
 		else
-			cd $CDIR"$BDIR"
+			cd $CDIR/"$BDIR"
 			echo $SMA5 "$DIRNAME"
 			mkdir "$DIRNAME"
 			cd "$DIRNAME"
 		fi
 else
-	cd $CDIR"$BDIR"
+	cd $CDIR/$BDIR
 	echo $SMA5 "$DIRNAME"
 	mkdir "$DIRNAME"
 	cd "$DIRNAME"
@@ -61,7 +61,7 @@
 	notify-send --icon="pen" -u normal "$DIRNAME" "Se procedera a su descarga en HD"
 	cd "/tmp"
 	wget --referer="http://submanga.com/c/$LINK" http://submanga.com/hd -o /tmp/.hdlog -O /tmp/smhd
-	cd "$CDIR""$BDIR"/"$DIRNAME"
+	cd "$CDIR"/"$BDIR"/"$DIRNAME"
 	SCDNB=`cat /tmp/smhd | grep "http://img.*.submanga.com/hd.*" | cut -d "%" -f 7 | cut -d "\"" -f 11 | cut -d "/" -f 6 `
 	FSTNB=`cat /tmp/smhd | grep "http://img.*.submanga.com/hd.*" | cut -d "%" -f 7 | cut -d "\"" -f 11 | cut -d "/" -f 5 `
 	for i in `seq 1 100`;
@@ -116,7 +116,7 @@
 	echo "$SMA1"
 else
 	echo "$SMA2 $BDIR"
-	mkdir "$CDIR$BDIR"
+	mkdir "$CDIR/$BDIR"
 fi
 
 WCLFILE=`wc -l /tmp/.smbulklynks | cut -d " " -f 1`
@@ -133,7 +133,7 @@
 echo "$SMA9 $BDIR"
 echo "$SMA10"
 echo "$SMA11"
-read -p "[y/N] " YESNO
+read -p "$CONFIRM " YESNO
 if [ $YESNO = y ];
 then
 	read -p "$SMA12" FCH
@@ -148,17 +148,17 @@
 		FNF=`echo $FNCX | cut -d "&" -f1`
 		DIRNAME="$(echo $FNF)"
 		
-		if ls $CDIR"$BDIR" | grep "$DIRNAME" > /dev/null
+		if ls $CDIR/"$BDIR" | grep "$DIRNAME" > /dev/null
 		then
 			echo $SMA3
 			read -p "" DIRNAME
-			if ls $CDIR"$BDIR" | grep "$DIRNAME" > /dev/null
+			if ls $CDIR/"$BDIR" | grep "$DIRNAME" > /dev/null
 			then
 				echo $SMA4
 			exit
 			fi
 		else
-			cd $CDIR"$BDIR"
+			cd $CDIR/"$BDIR"
 			echo $SMA5 "$DIRNAME"
 			mkdir "$DIRNAME"
 			cd "$DIRNAME"
@@ -171,7 +171,7 @@
 			notify-send --icon="pen" -u normal "$DIRNAME" "Se procedera a su descarga en HD"
 			cd "/tmp"
 			wget --referer="http://submanga.com/c/$LINK" http://submanga.com/hd -o /tmp/.hdlog -O /tmp/smhd
-			cd "$CDIR""$BDIR"/"$DIRNAME"
+			cd "$CDIR"/"$BDIR"/"$DIRNAME"
 			SCDNB=`cat /tmp/smhd | grep "http://img.*.submanga.com/hd.*" | cut -d "%" -f 7 | cut -d "\"" -f 11 | cut -d "/" -f 6 `
 			FSTNB=`cat /tmp/smhd | grep "http://img.*.submanga.com/hd.*" | cut -d "%" -f 7 | cut -d "\"" -f 11 | cut -d "/" -f 5 `
 			for i in `seq 1 100`;
@@ -224,17 +224,17 @@
 		FNF=`echo $FNCX | cut -d "&" -f1`
 		DIRNAME="$(echo $FNF)"
 
-		if ls $CDIR"$BDIR" | grep "$DIRNAME" > /dev/null
+		if ls $CDIR/"$BDIR" | grep "$DIRNAME" > /dev/null
 		then
 			echo $SMA3
 			read -p "" DIRNAME
-			if ls $CDIR"$BDIR" | grep "$DIRNAME" > /dev/null
+			if ls $CDIR/"$BDIR" | grep "$DIRNAME" > /dev/null
 			then
 				echo $SMA4
 			exit
 			fi
 		else
-			cd $CDIR"$BDIR"
+			cd $CDIR/"$BDIR"
 			echo $SMA5 "$DIRNAME"
 			mkdir "$DIRNAME"
 			cd "$DIRNAME"
@@ -247,7 +247,7 @@
 			notify-send --icon="pen" -u normal "$DIRNAME" "Se procedera a su descarga en HD"
 			cd "/tmp"
 			wget --referer="http://submanga.com/c/$LINK" http://submanga.com/hd -o /tmp/.hdlog -O /tmp/smhd
-			cd "$CDIR""$BDIR"/"$DIRNAME"
+			cd "$CDIR"/"$BDIR"/"$DIRNAME"
 			SCDNB=`cat /tmp/smhd | grep "http://img.*.submanga.com/hd.*" | cut -d "%" -f 7 | cut -d "\"" -f 11 | cut -d "/" -f 6 `
 			FSTNB=`cat /tmp/smhd | grep "http://img.*.submanga.com/hd.*" | cut -d "%" -f 7 | cut -d "\"" -f 11 | cut -d "/" -f 5 `
 			for i in `seq 1 100`;
