--- submanga	2011-07-03 22:19:24.000000000 +0200
+++ /home/jorge/Descargas/linux-manga-downloader_0.2.0.1-5Beta_all/usr/local/bin/Correcciones/submanga	2013-02-28 17:03:58.315980249 +0100
@@ -4,7 +4,7 @@
 # /* Import options *\ #
 
 source ~/.LMD/options;
-source ~/.LMD/lang/$LMLANG;
+source /usr/share/lmd/langs/$LMLANG;
 
 ##### /* Functions *\ #####
 
@@ -28,11 +28,11 @@
 	echo "$SMA1"
 else
 	echo "$SMA2 $BDIR"
-	mkdir "$CDIR$BDIR"
+	mkdir $CDIR/$BDIR
 fi
 
 
-if ls $CDIR"$BDIR" | grep "$DIRNAME" > /dev/null
+if ls $CDIR/"$BDIR" | grep "$DIRNAME" > /dev/null
 then
 	echo $SMA3
 		read -p "" DIRNAME
@@ -116,7 +116,7 @@
 	echo "$SMA1"
 else
 	echo "$SMA2 $BDIR"
-	mkdir "$CDIR$BDIR"
+	mkdir "$CDIR/$BDIR"
 fi
 
 WCLFILE=`wc -l /tmp/.smbulklynks | cut -d " " -f 1`
@@ -133,7 +133,7 @@
 echo "$SMA9 $BDIR"
 echo "$SMA10"
 echo "$SMA11"
-read -p "[y/N] " YESNO
+read -p "$CONFIRM " YESNO
 if [ $YESNO = y ];
 then
 	read -p "$SMA12" FCH
