--- libgksu-2.0.12/libgksuui/gksuui-dialog.c.orig	2012-07-22 18:48:28.001292562 -0300
+++ libgksu-2.0.12/libgksuui/gksuui-dialog.c	2012-07-22 19:44:44.911174828 -0300
@@ -216,39 +216,12 @@
   gtk_box_pack_start (GTK_BOX(GKSUUI_DIALOG(dialog)->entry_vbox), vbox, TRUE, TRUE, 0);
   gtk_widget_show (vbox);
 
-  check_button = gtk_check_button_new_with_label (_("Remember password"));
-  g_signal_connect (G_OBJECT(check_button), "toggled", G_CALLBACK(cb_toggled_cb), "save-to-keyring");
-
-  remember_password = gconf_client_get_bool (gconf_client, BASE_PATH"save-to-keyring", NULL);
-  gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(check_button), remember_password);
-  gtk_box_pack_start (GTK_BOX(vbox), check_button, TRUE, TRUE, 0);
-  gtk_widget_show (check_button);
-
   alignment = gtk_alignment_new (0.5, 0.5, 0.6, 1);
   gtk_box_pack_start (GTK_BOX(vbox), alignment, TRUE, TRUE, 2);
   gtk_widget_show (alignment);
 
-  radio_vbox = gtk_vbox_new (2, TRUE);
-  gtk_container_add (GTK_CONTAINER(alignment), radio_vbox);
-  gtk_widget_set_sensitive (radio_vbox, remember_password);
-  gtk_widget_show (radio_vbox);
-
-  radio_session = gtk_radio_button_new_with_label (NULL, _("Save for this session"));
-  g_signal_connect (G_OBJECT(radio_session), "toggled", G_CALLBACK(cb_toggled_cb), "save-keyring");
-  gtk_box_pack_start (GTK_BOX(radio_vbox), radio_session, TRUE, TRUE, 0);
-  gtk_widget_show (radio_session);
-
-  radio_default = gtk_radio_button_new_with_label_from_widget (GTK_RADIO_BUTTON(radio_session), _("Save in the keyring"));
-  gtk_box_pack_start (GTK_BOX(radio_vbox), radio_default, TRUE, TRUE, 0);
-  gtk_widget_show (radio_default);
-
   g_signal_connect (G_OBJECT(check_button), "toggled", G_CALLBACK(set_sensitivity_cb), radio_vbox);
 
-  tmp = gconf_client_get_string (gconf_client, BASE_PATH"save-keyring", NULL);
-  if (tmp && (!strcmp (tmp, "default")))
-    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(radio_default), TRUE);
-  g_free (tmp);
-
   g_object_unref (gconf_client);
 }
 
