--- pulseaudio-equalizer-2.7/usr/bin/pulseaudio-equalizer.old	2010-09-20 19:28:09.000000000 +0000
+++ pulseaudio-equalizer-2.7/usr/bin/pulseaudio-equalizer	2010-09-20 19:28:58.000000000 +0000
@@ -128,7 +128,9 @@ fi 
 if [ -f $HOME/.pulse/default.pa ]; then
     if [ "$(cat $HOME/.pulse/default.pa | grep -m1 "### BEGIN: Equalized audio configuration")" = "### BEGIN: Equalized audio configuration" ]; then
     PA_EQUALIZER_PERSISTENCE=1
-  fi
+    else
+    PA_EQUALIZER_PERSISTENCE=0
+    fi
 else
   PA_EQUALIZER_PERSISTENCE=0
 fi
