diff --git a/src/tilda_terminal.c b/src/tilda_terminal.c
index d1b66c1..a0281a5 100644
--- a/src/tilda_terminal.c
+++ b/src/tilda_terminal.c
@@ -845,7 +845,7 @@ static int button_press_cb (G_GNUC_UNUSED GtkWidget *widget, GdkEventButton *eve
             tilda_window_prev_tab (tt->tw);
             break;
         case 3: /* Right Click */
-            popup_menu (tt->tw, tt);
+            /* popup_menu (tt->tw, tt); */
             break;
         case 2: /* Middle Click */
             break;
