--- via_chrome9_irq.c.orig	2014-10-09 12:43:38.000000000 +0900
+++ via_chrome9_irq.c	2014-12-31 01:27:14.394735475 +0900
@@ -426,7 +426,7 @@
 	/* via_chrome9_hpd_irq_set(dev); */
 }
 
-irqreturn_t via_chrome9_driver_irq_handler(DRM_IRQ_ARGS)
+irqreturn_t via_chrome9_driver_irq_handler(int irq, void *arg)
 {
 	struct drm_device *dev = (struct drm_device *) arg;
 	struct drm_via_chrome9_private *dev_priv =
@@ -524,7 +524,7 @@
 			INTERRUPT_ENABLE);
 }
 
-irqreturn_t via_chrome9_driver_irq_handler_nonKMS(DRM_IRQ_ARGS)
+irqreturn_t via_chrome9_driver_irq_handler_nonKMS(int irq, void *arg)
 {
 	struct drm_via_chrome9_private *dev_priv =
 		(struct drm_via_chrome9_private *)arg;
