--- via_chrome9_reloc.c.orig	2014-10-09 12:43:38.000000000 +0900
+++ via_chrome9_reloc.c	2014-12-31 01:21:05.731110225 +0900
@@ -148,7 +148,7 @@
 		return -ENOMEM;
 	}
 
-	if (DRM_COPY_FROM_USER(parse->exec_objects,
+	if (copy_from_user(parse->exec_objects,
 		(void __user *)(unsigned long)data->buffer_ptr,
 		sizeof(struct drm_via_chrome9_gem_exec_object))) {
 		return -EFAULT;
@@ -449,12 +449,12 @@
 		return -ENOMEM;
 	}
 
-	if (DRM_COPY_FROM_USER(parse->relocs,
+	if (copy_from_user(parse->relocs,
 			(void __user *)
 			(unsigned long)parse->exec_objects->relocs_ptr,
 			parse->exec_objects->relocation_count *
 			sizeof(struct drm_via_chrome9_gem_relocation_entry))) {
-		DRM_ERROR("DRM_COPY_FROM_USER relocs error\n");
+		DRM_ERROR("copy_from_user relocs error\n");
 		return -EFAULT;
 	}
 
@@ -473,14 +473,14 @@
 
 	/* update the buffer object offset to user space */
 	if (parse->need_correct) {
-		if (DRM_COPY_TO_USER(
+		if (copy_to_user(
 			(void __user *)(unsigned long)
 			parse->exec_objects->relocs_ptr,
 			parse->relocs,
 			parse->exec_objects->relocation_count *
 			sizeof(struct drm_via_chrome9_gem_relocation_entry))) {
 
-			DRM_ERROR("DRM_COPY_TO_USER relocs error\n");
+			DRM_ERROR("copy_to_user relocs error\n");
 			return -EFAULT;
 		}
 	}
