--- PythonChecker.vala	2012-02-25 19:38:43.586816604 +0100
+++ PythonChecker.vala	2012-02-25 19:39:19.163484022 +0100
@@ -53,7 +53,7 @@
   AsyncCommand cmd;
   construct {
     string import = "import %s".printf(module);
-    string[] argv = {"python", "-c", import};
+    string[] argv = {"python2", "-c", import};
     cmd = new AsyncCommand(argv);
     cmd.done.connect((s) => {
       available = s;