--- ./src/signsrch.c	2013-10-05 03:43:38.000000000 +0200
+++ ./src/signsrch.c	2013-10-20 16:33:10.919862203 +0200
@@ -70,7 +70,7 @@
 #define mystrdup            strdup
 #define PATHSZ              2000
 #define MAX_AND_DISTANCE    ((pattern_len < (3000/32)) ? (pattern_len * 32) : 3000) // was 3000
-#define SIGNFILE            "signsrch.sig"
+#define SIGNFILE            "/usr/share/signsrch/signsrch.sig"
 #define SIGNFILEWEB         "http://aluigi.org/mytoolz/signsrch.sig.zip"
 #define INVALID_OFFSET      (-1)
 #define FREEZ(X)            if(X) { free(X); X = NULL; }
