--- net-bubble-0.9.1.orig/chargement.c	2004-08-22 22:55:35.000000000 +0600
+++ net-bubble-0.9.1/chargement.c	2012-04-20 16:22:00.865781851 +0600
@@ -167,45 +167,45 @@
 
 	/* PNG 8 */
 
-	tab_fonts[0]=LoadSurface("gfx/fonts/16/bleu.png");
-	tab_fonts[1]=LoadSurface("gfx/fonts/16/bleufonce.png");
-	tab_fonts[2]=LoadSurface("gfx/fonts/16/cyan.png");
-	tab_fonts[3]=LoadSurface("gfx/fonts/16/jaune.png");
-	tab_fonts[4]=LoadSurface("gfx/fonts/16/mauve.png");
-	tab_fonts[5]=LoadSurface("gfx/fonts/16/orange.png");
-	tab_fonts[6]=LoadSurface("gfx/fonts/16/rose.png");
-	tab_fonts[7]=LoadSurface("gfx/fonts/16/rouge.png");
-	tab_fonts[8]=LoadSurface("gfx/fonts/16/vert.png");
-
-	tab_fonts[9]=LoadSurface("gfx/fonts/32/bleu.png");
-	tab_fonts[10]=LoadSurface("gfx/fonts/32/bleufonce.png");
-	tab_fonts[11]=LoadSurface("gfx/fonts/32/cyan.png");
-	tab_fonts[12]=LoadSurface("gfx/fonts/32/jaune.png");
-	tab_fonts[13]=LoadSurface("gfx/fonts/32/mauve.png");
-	tab_fonts[14]=LoadSurface("gfx/fonts/32/orange.png");
-	tab_fonts[15]=LoadSurface("gfx/fonts/32/rose.png");
-	tab_fonts[16]=LoadSurface("gfx/fonts/32/rouge.png");
-	tab_fonts[17]=LoadSurface("gfx/fonts/32/vert.png");
-
-	tab_fonts[18]=LoadSurface("gfx/fonts/64/bleu.png");
-	tab_fonts[19]=LoadSurface("gfx/fonts/64/bleufonce.png");
-	tab_fonts[20]=LoadSurface("gfx/fonts/64/cyan.png");
-	tab_fonts[21]=LoadSurface("gfx/fonts/64/jaune.png");
-	tab_fonts[22]=LoadSurface("gfx/fonts/64/mauve.png");
-	tab_fonts[23]=LoadSurface("gfx/fonts/64/orange.png");
-	tab_fonts[24]=LoadSurface("gfx/fonts/64/rose.png");
-	tab_fonts[25]=LoadSurface("gfx/fonts/64/rouge.png");
-	tab_fonts[26]=LoadSurface("gfx/fonts/64/vert.png");
+	tab_fonts[0]=LoadSurface("/usr/share/net-bubble/gfx/fonts/16/bleu.png");
+	tab_fonts[1]=LoadSurface("/usr/share/net-bubble/gfx/fonts/16/bleufonce.png");
+	tab_fonts[2]=LoadSurface("/usr/share/net-bubble/gfx/fonts/16/cyan.png");
+	tab_fonts[3]=LoadSurface("/usr/share/net-bubble/gfx/fonts/16/jaune.png");
+	tab_fonts[4]=LoadSurface("/usr/share/net-bubble/gfx/fonts/16/mauve.png");
+	tab_fonts[5]=LoadSurface("/usr/share/net-bubble/gfx/fonts/16/orange.png");
+	tab_fonts[6]=LoadSurface("/usr/share/net-bubble/gfx/fonts/16/rose.png");
+	tab_fonts[7]=LoadSurface("/usr/share/net-bubble/gfx/fonts/16/rouge.png");
+	tab_fonts[8]=LoadSurface("/usr/share/net-bubble/gfx/fonts/16/vert.png");
+
+	tab_fonts[9]=LoadSurface("/usr/share/net-bubble/gfx/fonts/32/bleu.png");
+	tab_fonts[10]=LoadSurface("/usr/share/net-bubble/gfx/fonts/32/bleufonce.png");
+	tab_fonts[11]=LoadSurface("/usr/share/net-bubble/gfx/fonts/32/cyan.png");
+	tab_fonts[12]=LoadSurface("/usr/share/net-bubble/gfx/fonts/32/jaune.png");
+	tab_fonts[13]=LoadSurface("/usr/share/net-bubble/gfx/fonts/32/mauve.png");
+	tab_fonts[14]=LoadSurface("/usr/share/net-bubble/gfx/fonts/32/orange.png");
+	tab_fonts[15]=LoadSurface("/usr/share/net-bubble/gfx/fonts/32/rose.png");
+	tab_fonts[16]=LoadSurface("/usr/share/net-bubble/gfx/fonts/32/rouge.png");
+	tab_fonts[17]=LoadSurface("/usr/share/net-bubble/gfx/fonts/32/vert.png");
+
+	tab_fonts[18]=LoadSurface("/usr/share/net-bubble/gfx/fonts/64/bleu.png");
+	tab_fonts[19]=LoadSurface("/usr/share/net-bubble/gfx/fonts/64/bleufonce.png");
+	tab_fonts[20]=LoadSurface("/usr/share/net-bubble/gfx/fonts/64/cyan.png");
+	tab_fonts[21]=LoadSurface("/usr/share/net-bubble/gfx/fonts/64/jaune.png");
+	tab_fonts[22]=LoadSurface("/usr/share/net-bubble/gfx/fonts/64/mauve.png");
+	tab_fonts[23]=LoadSurface("/usr/share/net-bubble/gfx/fonts/64/orange.png");
+	tab_fonts[24]=LoadSurface("/usr/share/net-bubble/gfx/fonts/64/rose.png");
+	tab_fonts[25]=LoadSurface("/usr/share/net-bubble/gfx/fonts/64/rouge.png");
+	tab_fonts[26]=LoadSurface("/usr/share/net-bubble/gfx/fonts/64/vert.png");
 		
-	tab_bubbles_img[0]=LoadSurface("gfx/bubbles_stuff/ballesvert.png");
-	tab_bubbles_img[1]=LoadSurface("gfx/bubbles_stuff/ballesorange.png");
-	tab_bubbles_img[2]=LoadSurface("gfx/bubbles_stuff/ballesbleu.png");
-	tab_bubbles_img[3]=LoadSurface("gfx/bubbles_stuff/ballesrouge.png");
-	tab_bubbles_img[4]=LoadSurface("gfx/bubbles_stuff/ballesmagenta.png");
-	tab_bubbles_img[5]=LoadSurface("gfx/bubbles_stuff/ballesjaune.png");
-	tab_bubbles_img[6]=LoadSurface("gfx/bubbles_stuff/ballescyan.png");
-	tab_bubbles_img[7]=LoadSurface("gfx/bubbles_stuff/ballesnoir.png");
-	tab_bubbles_img[8]=LoadSurface("gfx/bubbles_stuff/ballesinv.png");
+	tab_bubbles_img[0]=LoadSurface("/usr/share/net-bubble/gfx/bubbles_stuff/ballesvert.png");
+	tab_bubbles_img[1]=LoadSurface("/usr/share/net-bubble/gfx/bubbles_stuff/ballesorange.png");
+	tab_bubbles_img[2]=LoadSurface("/usr/share/net-bubble/gfx/bubbles_stuff/ballesbleu.png");
+	tab_bubbles_img[3]=LoadSurface("/usr/share/net-bubble/gfx/bubbles_stuff/ballesrouge.png");
+	tab_bubbles_img[4]=LoadSurface("/usr/share/net-bubble/gfx/bubbles_stuff/ballesmagenta.png");
+	tab_bubbles_img[5]=LoadSurface("/usr/share/net-bubble/gfx/bubbles_stuff/ballesjaune.png");
+	tab_bubbles_img[6]=LoadSurface("/usr/share/net-bubble/gfx/bubbles_stuff/ballescyan.png");
+	tab_bubbles_img[7]=LoadSurface("/usr/share/net-bubble/gfx/bubbles_stuff/ballesnoir.png");
+	tab_bubbles_img[8]=LoadSurface("/usr/share/net-bubble/gfx/bubbles_stuff/ballesinv.png");
 
 	/* Loading */
 
@@ -279,13 +279,13 @@
 
 	/* JPG ok */
 
-	plage_1j=LoadSurface("gfx/background/plage_1j.jpg");
-	plage_2j=LoadSurface("gfx/background/plage_2j.jpg");
-	bg_full_menu=LoadSurface("gfx/background/bg_full_menu.jpg"); 
-	bg_playable_1p_court=LoadSurface("gfx/background/bg_playable_1p_court.jpg"); /* NB: Tout est playable sauf la bordure. */
-	bg_playable_1p_long=LoadSurface("gfx/background/bg_playable_1p_long.jpg"); /* NB: Tout est playable sauf la bordure. */
-	bg_playable_2p_gauche=LoadSurface("gfx/background/bg_playable_2p_gauche.jpg"); /* NB: Tout est playable sauf la bordure. */
-	bg_playable_2p_droite=LoadSurface("gfx/background/bg_playable_2p_droite.jpg"); /* NB: Tout est playable sauf la bordure. */
+	plage_1j=LoadSurface("/usr/share/net-bubble/gfx/background/plage_1j.jpg");
+	plage_2j=LoadSurface("/usr/share/net-bubble/gfx/background/plage_2j.jpg");
+	bg_full_menu=LoadSurface("/usr/share/net-bubble/gfx/background/bg_full_menu.jpg"); 
+	bg_playable_1p_court=LoadSurface("/usr/share/net-bubble/gfx/background/bg_playable_1p_court.jpg"); /* NB: Tout est playable sauf la bordure. */
+	bg_playable_1p_long=LoadSurface("/usr/share/net-bubble/gfx/background/bg_playable_1p_long.jpg"); /* NB: Tout est playable sauf la bordure. */
+	bg_playable_2p_gauche=LoadSurface("/usr/share/net-bubble/gfx/background/bg_playable_2p_gauche.jpg"); /* NB: Tout est playable sauf la bordure. */
+	bg_playable_2p_droite=LoadSurface("/usr/share/net-bubble/gfx/background/bg_playable_2p_droite.jpg"); /* NB: Tout est playable sauf la bordure. */
 
 	/* Loading */
 
@@ -303,21 +303,21 @@
 
 	/* BMP */
 
-	rond_angle=LoadSurfaceBMP("gfx/bubbles_stuff/rond_angle.bmp",1,255,255,255);
-	offset_court=LoadSurfaceBMP("gfx/bubbles_stuff/offset_court.bmp",1,255,255,255);
-	offset_long=LoadSurfaceBMP("gfx/bubbles_stuff/offset_long.bmp",1,255,255,255);
-	malus1=LoadSurfaceBMP("gfx/bubbles_stuff/malus1.bmp",1,255,255,255);
-	malus2=LoadSurfaceBMP("gfx/bubbles_stuff/malus2.bmp",1,255,255,255);
+	rond_angle=LoadSurfaceBMP("/usr/share/net-bubble/gfx/bubbles_stuff/rond_angle.bmp",1,255,255,255);
+	offset_court=LoadSurfaceBMP("/usr/share/net-bubble/gfx/bubbles_stuff/offset_court.bmp",1,255,255,255);
+	offset_long=LoadSurfaceBMP("/usr/share/net-bubble/gfx/bubbles_stuff/offset_long.bmp",1,255,255,255);
+	malus1=LoadSurfaceBMP("/usr/share/net-bubble/gfx/bubbles_stuff/malus1.bmp",1,255,255,255);
+	malus2=LoadSurfaceBMP("/usr/share/net-bubble/gfx/bubbles_stuff/malus2.bmp",1,255,255,255);
 
 	/* PNG ... à faire */
 
-	tab_pix[0]=LoadSurface("gfx/pix/etat0.png");
-	tab_pix[1]=LoadSurface("gfx/pix/etat1.png");
-	tab_pix[2]=LoadSurface("gfx/pix/etat2.png");
-	tab_pix[3]=LoadSurface("gfx/pix/etat3.png");
-	tab_pix[4]=LoadSurface("gfx/pix/etat4.png");
-	tab_pix[5]=LoadSurface("gfx/pix/etat5.png");
-	tab_pix[6]=LoadSurface("gfx/pix/etat6.png");
+	tab_pix[0]=LoadSurface("/usr/share/net-bubble/gfx/pix/etat0.png");
+	tab_pix[1]=LoadSurface("/usr/share/net-bubble/gfx/pix/etat1.png");
+	tab_pix[2]=LoadSurface("/usr/share/net-bubble/gfx/pix/etat2.png");
+	tab_pix[3]=LoadSurface("/usr/share/net-bubble/gfx/pix/etat3.png");
+	tab_pix[4]=LoadSurface("/usr/share/net-bubble/gfx/pix/etat4.png");
+	tab_pix[5]=LoadSurface("/usr/share/net-bubble/gfx/pix/etat5.png");
+	tab_pix[6]=LoadSurface("/usr/share/net-bubble/gfx/pix/etat6.png");
 	
 	/* Loading */
 	
@@ -349,8 +349,8 @@
 
 	/* PNG ... à faire */
 
-	ip_fond=LoadSurface("gfx/background/ip_fond.png");
-	cache_ip=LoadSurface("gfx/background/cache_ip.png");
+	ip_fond=LoadSurface("/usr/share/net-bubble/gfx/background/ip_fond.png");
+	cache_ip=LoadSurface("/usr/share/net-bubble/gfx/background/cache_ip.png");
 
 	/* Loading */
 
@@ -368,16 +368,16 @@
 
 	/* main (principal) */
 
-	menu_single_off=LoadSurface("gfx/buttons/1joueur_off.jpg");
-	menu_single_on=LoadSurface("gfx/buttons/1joueur_on.jpg");
+	menu_single_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/1joueur_off.jpg");
+	menu_single_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/1joueur_on.jpg");
 
-	menu_multiplayer_off=LoadSurface("gfx/buttons/multi_off.jpg");
-	menu_multiplayer_on=LoadSurface("gfx/buttons/multi_on.jpg");
+	menu_multiplayer_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/multi_off.jpg");
+	menu_multiplayer_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/multi_on.jpg");
 
 	/* menu 1 joueur */
 
-	menu_classic_off=LoadSurface("gfx/buttons/seul_off.jpg");
-	menu_classic_on=LoadSurface("gfx/buttons/seul_on.jpg");
+	menu_classic_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/seul_off.jpg");
+	menu_classic_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/seul_on.jpg");
 
 	/* Loading */
 
@@ -391,20 +391,20 @@
 	
 	/* - Fin part loading - */
 	
-	menu_vs_cpu_off=LoadSurface("gfx/buttons/vs_cpu_off.jpg");
-	menu_vs_cpu_on=LoadSurface("gfx/buttons/vs_cpu_on.jpg");
+	menu_vs_cpu_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/vs_cpu_off.jpg");
+	menu_vs_cpu_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/vs_cpu_on.jpg");
 	
-	menu_cpue_off=LoadSurface("gfx/buttons/vs_cpu_easy_off.jpg");
-	menu_cpue_on=LoadSurface("gfx/buttons/vs_cpu_easy_on.jpg");
+	menu_cpue_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/vs_cpu_easy_off.jpg");
+	menu_cpue_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/vs_cpu_easy_on.jpg");
 
-	menu_cpum_off=LoadSurface("gfx/buttons/vs_cpu_medium_off.jpg");
-	menu_cpum_on=LoadSurface("gfx/buttons/vs_cpu_medium_on.jpg");
+	menu_cpum_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/vs_cpu_medium_off.jpg");
+	menu_cpum_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/vs_cpu_medium_on.jpg");
 
-	menu_cpuh_off=LoadSurface("gfx/buttons/vs_cpu_crazy_off.jpg");
-	menu_cpuh_on=LoadSurface("gfx/buttons/vs_cpu_crazy_on.jpg");
+	menu_cpuh_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/vs_cpu_crazy_off.jpg");
+	menu_cpuh_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/vs_cpu_crazy_on.jpg");
 
-	menu_cpu_vs_cpu_off=LoadSurface("gfx/buttons/cpu_vs_cpu_off.jpg");
-	menu_cpu_vs_cpu_on=LoadSurface("gfx/buttons/cpu_vs_cpu_on.jpg");
+	menu_cpu_vs_cpu_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/cpu_vs_cpu_off.jpg");
+	menu_cpu_vs_cpu_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/cpu_vs_cpu_on.jpg");
 
 	/* Loading */
 
@@ -420,20 +420,20 @@
 		
 	/* menu 2 joueurs */
 
-	menu_local_off=LoadSurface("gfx/buttons/local_off.jpg");
-	menu_local_on=LoadSurface("gfx/buttons/local_on.jpg");
+	menu_local_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/local_off.jpg");
+	menu_local_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/local_on.jpg");
 
-	menu_net_off=LoadSurface("gfx/buttons/lan_net_off.jpg");
-	menu_net_on=LoadSurface("gfx/buttons/lan_net_on.jpg");
+	menu_net_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/lan_net_off.jpg");
+	menu_net_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/lan_net_on.jpg");
 
-	menu_join_off=LoadSurface("gfx/buttons/join_off.jpg");
-	menu_join_on=LoadSurface("gfx/buttons/join_on.jpg");
+	menu_join_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/join_off.jpg");
+	menu_join_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/join_on.jpg");
 
-	menu_host_off=LoadSurface("gfx/buttons/host_off.jpg");
-	menu_host_on=LoadSurface("gfx/buttons/host_on.jpg");
+	menu_host_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/host_off.jpg");
+	menu_host_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/host_on.jpg");
 
-	menu_connect_off=LoadSurface("gfx/buttons/connect_off.jpg");
-	menu_connect_on=LoadSurface("gfx/buttons/connect_on.jpg");
+	menu_connect_off=LoadSurface("/usr/share/net-bubble/gfx/buttons/connect_off.jpg");
+	menu_connect_on=LoadSurface("/usr/share/net-bubble/gfx/buttons/connect_on.jpg");
 
 	/* Loading */
 
--- net-bubble-0.9.1.orig/main.c	2004-09-16 15:27:34.000000000 +0600
+++ net-bubble-0.9.1/main.c	2012-04-20 16:22:29.595782015 +0600
@@ -3530,7 +3530,7 @@
 
 	/* Fin des déclarations. */
 
-	sprintf(level_tab_char,"levels/level%d.txt",level); /* sprintf copie la chaine "levels/levelX" dans level_tab_char en ajoutant le caractère de fin de chaine '\0'. En effet, %d de level_tab_char[17](17=lg renvoyée par sprintf=nbre de char de ":levels/level1.txt")=>level_tab_char[17] est celui d'après et j'ai mis juste avant le sprintf: level_tab_char[17]='a' et pourtant après le sprintf: %d de level_tab_char[17]=0 (valeur décimale du code ascii '\0' =0) => sprintf fait ce qu'il faut. */
+	sprintf(level_tab_char,"/usr/share/net-bubble/levels/level%d.txt",level); /* sprintf copie la chaine "/usr/share/net-bubble/levels/levelX" dans level_tab_char en ajoutant le caractère de fin de chaine '\0'. En effet, %d de level_tab_char[17](17=lg renvoyée par sprintf=nbre de char de ":/usr/share/net-bubble/levels/level1.txt")=>level_tab_char[17] est celui d'après et j'ai mis juste avant le sprintf: level_tab_char[17]='a' et pourtant après le sprintf: %d de level_tab_char[17]=0 (valeur décimale du code ascii '\0' =0) => sprintf fait ce qu'il faut. */
 	
 	/* Ouverture du fichier du fichier "levelX". */
 
@@ -3539,7 +3539,7 @@
 		/* fprintf(stderr, "Dans le fichier main.c (fonction fopen()): Impossible d'ouvrir le fichier \"%s\". Raison: %s\n",level_tab_char, strerror(errno)); 
 		exit(EXIT_FAILURE); */
 		
-		fp=fopen("levels/level1.txt", "r"); /* On recommence a niveau 1 lq fini. */
+		fp=fopen("/usr/share/net-bubble/levels/level1.txt", "r"); /* On recommence a niveau 1 lq fini. */
 		level=1;
 	}
 
