diff -aur cairo-1.12.16-orig/configure.ac cairo-1.12.16-patched/configure.ac
--- cairo-1.12.16-orig/configure.ac	2013-08-26 19:11:51.000000000 +0300
+++ cairo-1.12.16-patched/configure.ac	2014-05-06 02:55:08.699485493 +0300
@@ -845,6 +845,9 @@
 PKG_CHECK_MODULES(gtk, "gtk+-2.0",have_gtk=yes, have_gtk=no)
 AM_CONDITIONAL(HAVE_GTK, test "x$have_gtk" = "xyes")
 
+# gcc 4.9 fix
+CFLAGS+=" -ffat-lto-objects"
+
 AC_CONFIG_FILES([
 Makefile
 boilerplate/Makefile
