--- /home/peto/x/thinkfan-0.7.1/thinkfan.c	2010-06-14 01:31:18.000000000 +0200
+++ ./thinkfan.c	2010-10-11 17:12:20.000000000 +0200
@@ -43,13 +43,17 @@
 	prefix = "\n"; // It is set to "" by the output macros
 
 	// Set initial fan level...
-	lvl_idx = config->num_limits - 1;
+	lvl_idx = 0;
 	b_temp = temp = config->get_temp();
-	if (errcnt) return errcnt;
-	while ((temp <= config->limits[lvl_idx].low) \
-	 && (lvl_idx > 0)) lvl_idx--;
+	if (errcnt)
+		return errcnt;
+	while ((temp > config->limits[lvl_idx].high) \
+	 && (lvl_idx > config->num_limits- 1))
+	{
+		lvl_idx++;
+	}
 	set_fan;
-
+	
 	/**********************************************
 	 * Main loop. This is the actual fan control.
 	 **********************************************/
@@ -59,6 +63,13 @@
 
 		// depending on the command line, this might also call depulse()
 		temp = config->get_temp();
+		//if (temp != 0 && last_temp == 0)
+		//	last_temp = temp;
+		
+		// check whether the value is correct
+		if (temp == 0 || last_temp == 0) {
+			continue;
+		}
 
 		// Write current fan level to IBM_FAN one cycle before the watchdog
 		// timeout ends, to let it know we're alive.
