Only in libevdevc: .git
diff -aur libevdevc/src/module.mk libevdevc.new/src/module.mk
--- libevdevc/src/module.mk	2015-03-13 14:06:46.775440352 +0000
+++ libevdevc.new/src/module.mk	2015-03-13 14:09:43.186917250 +0000
@@ -21,8 +21,6 @@
 	install -D -m 0755 src/libevdevc_hollow.so.0 \
 		$(DESTDIR)$(LIBDIR)/libevdevc_hollow.so.0
 	ln -f -s libevdevc_hollow.so.0 $(DESTDIR)$(LIBDIR)/libevdevc_hollow.so
-	install -D -m 0644 $(SRC)/libevdevc-cros.pc \
-			$(DESTDIR)$(LIBDIR)/pkgconfig/libevdevc-cros.pc
 
 setup-lib-in-place:
 	mkdir -p $(SRC)/in-place || true
