*** zhu3d.pri-old	2012-01-10 20:07:33.937297943 +0100
--- zhu3d.pri	2012-01-10 20:08:23.453696355 +0100
***************
*** 13,32 ****
  #
  # To change the locations, just fill in your desired path like e.g:
  #
! # SYSDIR=/usr/share/zhu3d-4.2.4/system
! # DOCDIR=/usr/share/doc/zhu3d-4.2.4/doc
! # WORKDIR=$$(HOME)/zhu3d-4.2.4/work
! # TEXDIR=$$(HOME)/zhu3d-4.2.4/work/textures
  #
  # In the example above the variable $$(HOME) denotes the users home-directory.
  # If an entry stays blank, the directory path for this will be relative to the
  # application path. As default the texture-directory resides under the work-
  # directory.
  
! SYSDIR=
! DOCDIR=
! WORKDIR=
! TEXDIR=
  
  
  # DETERMINING A DEFAULT GUI-STYLE
--- 13,32 ----
  #
  # To change the locations, just fill in your desired path like e.g:
  #
! # SYSDIR=/usr/share/zhu3d/system/usr/share/zhu3d-4.2.4/system
! # DOCDIR=/usr/share/zhu3d/doc/usr/share/doc/zhu3d-4.2.4/doc
! # WORKDIR=/usr/share/zhu3d/work$$(HOME)/zhu3d-4.2.4/work
! # TEXDIR=/usr/share/zhu3d/work/textures$$(HOME)/zhu3d-4.2.4/work/textures
  #
  # In the example above the variable $$(HOME) denotes the users home-directory.
  # If an entry stays blank, the directory path for this will be relative to the
  # application path. As default the texture-directory resides under the work-
  # directory.
  
! SYSDIR=/usr/share/zhu3d/system
! DOCDIR=/usr/share/zhu3d/doc
! WORKDIR=/usr/share/zhu3d/work
! TEXDIR=/usr/share/zhu3d/work/textures
  
  
  # DETERMINING A DEFAULT GUI-STYLE
