diff -crB envytools/include/yy.h envytools2/include/yy.h
*** envytools/include/yy.h	2012-09-08 12:11:20.937977038 +0200
--- envytools2/include/yy.h	2012-09-08 12:12:01.012201793 +0200
***************
*** 61,64 ****
--- 61,69 ----
  
  void yy_str_deescape(const char *str, struct astr *astr);
  
+ #ifndef YY_TYPEDEF_YY_SCANNER_T
+ #define YY_TYPEDEF_YY_SCANNER_T
+ typedef void* yyscan_t;
+ #endif
+ 
  #endif
