--- Gallery.pm	2011-06-08 20:47:46.000000000 +0200
+++ Gallery.pm.new	2011-09-11 19:43:25.456065848 +0200
@@ -14,9 +14,6 @@
 	if (exists($ENV{MOD_PERL_API_VERSION})
 		and ($ENV{MOD_PERL_API_VERSION}==2)) {
 		require mod_perl2;
-		if ($mod_perl::VERSION >= 1.99 && $mod_perl::VERSION < 2.0) {
-			die "mod_perl 2.0.0 or later is now required";
-		}
 		require Apache2::ServerRec;
 		require Apache2::RequestRec;
 		require Apache2::Log;
@@ -28,15 +25,6 @@
 		Apache2::Const->import(-compile => 'OK','DECLINED','FORBIDDEN','NOT_FOUND','HTTP_NOT_MODIFIED');
 
 		$::MP2 = 1;
-	} else {
-		require mod_perl;
-
-		require Apache;
-		require Apache::Constants;
-		require Apache::Request;
-	
-		Apache::Constants->import('OK','DECLINED','FORBIDDEN','NOT_FOUND');
-		$::MP2 = 0;
 	}
 }
 
