diff -Nru -x .git mirall/src/CMakeLists.txt mirall_new/src/CMakeLists.txt
--- mirall/src/CMakeLists.txt	2013-03-20 10:19:26.900058655 +0100
+++ mirall_new/src/CMakeLists.txt	2013-03-20 10:02:27.146777089 +0100
@@ -151,6 +151,7 @@
 
 set(mirall_SRCS
     mirall/application.cpp
+    mirall/systray.cpp
     mirall/folderwizard.cpp
     mirall/statusdialog.cpp
     mirall/owncloudwizard.cpp
@@ -165,6 +166,7 @@
 
 set(mirall_HEADERS
     mirall/application.h
+    mirall/systray.h
     mirall/folderwizard.h
     mirall/owncloudsetupwizard.h
     mirall/owncloudwizard.h
@@ -184,6 +186,7 @@
 
 # csync is required.
 include_directories(${CSYNC_INCLUDE_DIR}/csync ${CSYNC_INCLUDE_DIR} ${CMAKE_BINARY_DIR})
+include_directories(${QT_QTDBUS_INCLUDE_DIR})
 include_directories(${3rdparty_INC})
 
 qt4_wrap_cpp(mirallMoc ${mirall_HEADERS})
@@ -232,7 +235,7 @@
     # mirall is not built for the bundle
     add_executable( mirall WIN32 main.cpp ${final_src})
 
-    target_link_libraries(mirall ${QT_LIBRARIES} )
+    target_link_libraries(mirall ${QT_LIBRARIES} ${QT_QTDBUS_LIBRARIES} )
     target_link_libraries(mirall owncloudsync)
     target_link_libraries(mirall ${CSYNC_LIBRARY})
 
@@ -274,7 +277,7 @@
 set_target_properties( ${APPLICATION_EXECUTABLE} PROPERTIES
         RUNTIME_OUTPUT_DIRECTORY  ${BIN_OUTPUT_DIRECTORY}
 )
-target_link_libraries( ${APPLICATION_EXECUTABLE} ${QT_LIBRARIES} )
+target_link_libraries( ${APPLICATION_EXECUTABLE} ${QT_LIBRARIES} ${QT_QTDBUS_LIBRARIES} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} owncloudsync )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${CSYNC_LIBRARY} )
 
diff -Nru -x .git mirall/src/mirall/application.cpp mirall_new/src/mirall/application.cpp
--- mirall/src/mirall/application.cpp	2013-03-20 10:19:26.903391988 +0100
+++ mirall_new/src/mirall/application.cpp	2013-03-20 10:02:27.146777089 +0100
@@ -15,6 +15,7 @@
 #include <iostream>
 
 #include "mirall/application.h"
+#include "mirall/systray.h"
 #include "mirall/folder.h"
 #include "mirall/folderwatcher.h"
 #include "mirall/folderwizard.h"
@@ -444,7 +445,7 @@
 {
     // Setting a parent heres will crash on X11 since by the time qapp runs
     // its childrens dtors, the X11->screen variable queried for is gone -> crash
-    _tray = new QSystemTrayIcon;
+    _tray = new Systray();
     _tray->setIcon( _theme->syncStateIcon( SyncResult::NotYetStarted, true ) );
 
     connect(_tray,SIGNAL(activated(QSystemTrayIcon::ActivationReason)),
diff -Nru -x .git mirall/src/mirall/application.h mirall_new/src/mirall/application.h
--- mirall/src/mirall/application.h	2013-03-20 10:19:26.903391988 +0100
+++ mirall_new/src/mirall/application.h	2013-03-20 10:02:27.146777089 +0100
@@ -16,7 +16,6 @@
 #define APPLICATION_H
 
 #include <QApplication>
-#include <QSystemTrayIcon>
 #include <QNetworkReply>
 #include <QSslError>
 
@@ -27,6 +26,7 @@
 #include "mirall/logbrowser.h"
 #include "mirall/folderman.h"
 #include "mirall/fileitemdialog.h"
+#include "mirall/systray.h"
 
 class QAction;
 class QMenu;
@@ -104,7 +104,7 @@
     void raiseDialog( QWidget* );
 
     // configuration file -> folder
-    QSystemTrayIcon *_tray;
+    Systray *_tray;
     QAction *_actionQuit;
     QAction *_actionAddFolder;
     QAction *_actionOpenStatus;
diff -Nru -x .git mirall/src/mirall/systray.cpp mirall_new/src/mirall/systray.cpp
--- mirall/src/mirall/systray.cpp	1970-01-01 01:00:00.000000000 +0100
+++ mirall_new/src/mirall/systray.cpp	2013-03-20 10:03:24.126774196 +0100
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) by Cédric Bellegarde <gnumdk@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "systray.h"
+
+#ifdef Q_OS_LINUX
+#include <QDBusConnection>
+#include <QDBusMessage>
+#include <QDBusPendingCall>
+#endif
+
+void Systray::showMessage(const QString & title, const QString & message, MessageIcon icon, int millisecondsTimeoutHint)
+{
+#ifdef Q_OS_LINUX
+    QList<QVariant> args = QList<QVariant>() << "owncloud" << quint32(0) << "dialog-information"
+                                             << title << message << QStringList () << QVariantMap() << qint32(-1);
+    QDBusMessage method = QDBusMessage::createMethodCall("org.freedesktop.Notifications","/org/freedesktop/Notifications", "", "Notify");
+    method.setArguments(args);
+    QDBusConnection::sessionBus().asyncCall(method);
+#else
+    QSystemTrayIcon::showMessage(title, message, icon, millisecondsTimeoutHint);
+#endif
+}
\ Pas de fin de ligne à la fin du fichier
diff -Nru -x .git mirall/src/mirall/systray.h mirall_new/src/mirall/systray.h
--- mirall/src/mirall/systray.h	1970-01-01 01:00:00.000000000 +0100
+++ mirall_new/src/mirall/systray.h	2013-03-20 10:03:24.126774196 +0100
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) by Cédric Bellegarde <gnumdk@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef SYSTRAY_H
+#define SYSTRAY_H
+
+#include <QSystemTrayIcon>
+
+class QIcon;
+
+class Systray : public QSystemTrayIcon
+{
+    Q_OBJECT
+public:
+    void showMessage(const QString & title, const QString & message, MessageIcon icon = Information, int millisecondsTimeoutHint = 10000);
+};
+
+#endif //SYSTRAY_H
\ Pas de fin de ligne à la fin du fichier
