diff -ur openct-0.6.18/etc/openct.conf.in openct-0.6.18-static/etc/openct.conf.in
--- openct-0.6.18/etc/openct.conf.in	2009-07-29 10:04:36.000000000 +0300
+++ openct-0.6.18-static/etc/openct.conf.in	2009-12-19 18:02:16.000000000 +0200
@@ -6,26 +6,26 @@
 
 #
 # Path to ifdhandler
-ifdhandler {
-	program		= @sbindir@/ifdhandler;
+#ifdhandler {
+#	program		= @sbindir@/ifdhandler;
 	#
 	# Safe to disable force_poll:
 	#  >=linux-2.6.27.14
 	#  >=linux-2.6.28.3
 	#
-	force_poll	= 1;
-@ENABLE_NON_PRIVILEGED@	user		= @daemon_user@;
-@ENABLE_NON_PRIVILEGED@	groups = {
-@ENABLE_NON_PRIVILEGED@		@daemon_groups@,
-@ENABLE_NON_PRIVILEGED@	};
-};
+#	force_poll	= 1;
+#@ENABLE_NON_PRIVILEGED@	user		= @daemon_user@;
+#@ENABLE_NON_PRIVILEGED@	groups = {
+#@ENABLE_NON_PRIVILEGED@		@daemon_groups@,
+#@ENABLE_NON_PRIVILEGED@	};
+#};
 
 #
 # Configuration for ifdproxy (if you use it)
-ifdproxy {
+#ifdproxy {
 #	server-port     = @OPENCT_SOCKET_PATH@/.ifdproxy,
 #	device-port     = :6666;
-};
+#};
 
 # Configure static, non-hotplug aware readers here
 #
diff -ur openct-0.6.18/etc/openct_usb.in openct-0.6.18-static/etc/openct_usb.in
--- openct-0.6.18/etc/openct_usb.in	2008-12-27 17:51:56.000000000 +0200
+++ openct-0.6.18-static/etc/openct_usb.in	2009-12-19 18:05:17.000000000 +0200
@@ -4,6 +4,9 @@
 [ "$ACTION" = "add" ] || exit 0
 [ -e "@OPENCT_SOCKET_PATH@/status" ] || exit 0
 
+# try to get the device path
+cd /sys$DEVPATH && cd .. && DEVPATH=`pwd`
+
 # try to get the device node from the parent device
 if [ -z "$DEVNAME" ]; then
 	#
@@ -15,7 +18,7 @@
 	else
 		UDEVINFO="udevadm info"
 	fi
-	DEVNAME=/dev/$($UDEVINFO --query=name --path=$(dirname $DEVPATH))
+	DEVNAME=/dev/$($UDEVINFO --query=name --path=$DEVPATH)
 fi
 
 # if udev supplied a device node directly from the usb-device, we use it,
@@ -31,9 +34,9 @@
 fi
 
 if [ -z "$PRODUCT" ]; then
-	V=$(cat /sys$(dirname $DEVPATH)/idVendor | sed -e 's/^0*//')
-	P=$(cat /sys$(dirname $DEVPATH)/idProduct | sed -e 's/^0*//')
-	D=$(cat /sys$(dirname $DEVPATH)/bcdDevice | sed -e 's/^0*//')
+	V=$(cat $DEVPATH/idVendor | sed -e 's/^0*//')
+	P=$(cat $DEVPATH/idProduct | sed -e 's/^0*//')
+	D=$(cat $DEVPATH/bcdDevice | sed -e 's/^0*//')
 	PRODUCT="$V/$P/$D"
 fi
 
diff -ur openct-0.6.18/src/ifd/Makefile.am openct-0.6.18-static/src/ifd/Makefile.am
--- openct-0.6.18/src/ifd/Makefile.am	2008-12-27 17:51:56.000000000 +0200
+++ openct-0.6.18-static/src/ifd/Makefile.am	2009-12-19 18:02:16.000000000 +0200
@@ -27,12 +27,14 @@
 	-I$(top_srcdir)/src/include \
 	-I$(top_builddir)/src/include \
 	$(OPTIONAL_PCSC_CFLAGS) $(OPTIONAL_LIBUSB_CFLAGS) $(LTLIB_CFLAGS)
+libifd_la_LDFLAGS = -all-static
 
 ifdhandler_SOURCES = ifdhandler.c
-ifdhandler_LDADD = libifd.la
+ifdhandler_LDADD = libifd.la -ldl
 ifdhandler_CFLAGS = $(AM_CFLAGS) \
 	-I$(top_srcdir)/src/include \
 	-I$(top_builddir)/src/include
+ifdhandler_LDFLAGS = -all-static
 
 ifdproxy_SOURCES = ifdproxy.c ria-device.c ria-server.c
 ifdproxy_LDADD = libifd.la
diff -ur openct-0.6.18/src/tools/Makefile.am openct-0.6.18-static/src/tools/Makefile.am
--- openct-0.6.18/src/tools/Makefile.am	2008-07-27 12:54:36.000000000 +0300
+++ openct-0.6.18-static/src/tools/Makefile.am	2009-12-19 18:02:16.000000000 +0200
@@ -11,7 +11,8 @@
 	-I$(top_builddir)/src/include
 
 openct_control_SOURCES = openct-control.c
-openct_control_LDADD = $(top_builddir)/src/ifd/libifd.la $(top_builddir)/src/ct/libopenct.la
+openct_control_LDADD = $(top_builddir)/src/ifd/libifd.la $(top_builddir)/src/ct/libopenct.la -ldl
 openct_control_CFLAGS = $(AM_CFLAGS) \
 	-I$(top_srcdir)/src/include \
 	-I$(top_builddir)/src/include
+openct_control_LDFLAGS = -all-static
