diff -ur opensc-0.11.11/etc/opensc.conf.in opensc-0.11.11-static/etc/opensc.conf.in
--- opensc-0.11.11/etc/opensc.conf.in	2009-02-01 10:28:49.000000000 +0200
+++ opensc-0.11.11-static/etc/opensc.conf.in	2009-12-19 19:46:35.000000000 +0200
@@ -46,7 +46,7 @@
 	# NOTE: if "internal" keyword is used, must be the
 	# last entry in reader_drivers list
 	#
-	# reader_drivers = openct, pcsc, ctapi;
+	reader_drivers = openct; #, pcsc, ctapi;
 
 	reader_driver ctapi {
 		# module /usr/local/towitoko/lib/libtowitoko.so {
@@ -64,7 +64,7 @@
 	# but the same set of variables are applicable to ctapi and
 	# openct readers, simply by using "reader_driver ctapi" and
 	# "reader_driver openct", respectively.
-	reader_driver pcsc {
+	# reader_driver pcsc {
 		# This sets the maximum send and receive sizes.
 		# Some reader drivers have limitations, so you need
 		# to set these values. For usb devices check the
@@ -92,7 +92,7 @@
 		# Use specific pcsc provider.
 		# Default: @DEFAULT_PCSC_PROVIDER@
 		# provider_library = @DEFAULT_PCSC_PROVIDER@
-	}
+	# }
 
 	# Options for OpenCT support
 	reader_driver openct {
diff -ur opensc-0.11.11/src/tools/Makefile.am opensc-0.11.11-static/src/tools/Makefile.am
--- opensc-0.11.11/src/tools/Makefile.am	2009-09-17 10:08:44.000000000 +0300
+++ opensc-0.11.11-static/src/tools/Makefile.am	2009-12-19 21:26:54.000000000 +0200
@@ -28,6 +28,7 @@
 opensc_explorer_LDADD = $(OPTIONAL_READLINE_LIBS)
 pkcs15_tool_SOURCES = pkcs15-tool.c util.c
 pkcs15_tool_LDADD = $(OPTIONAL_OPENSSL_LIBS)
+pkcs15_tool_LDFLAGS = -all-static
 pkcs11_tool_SOURCES = pkcs11-tool.c util.c
 pkcs11_tool_LDADD = $(OPTIONAL_OPENSSL_LIBS) \
 	$(top_builddir)/src/pkcs11/libpkcs11.la
