diff -u a/kotlinc-js b/kotlinc-js
--- a/kotlinc-js	2015-03-18 19:02:16.000000000 +0100
+++ b/kotlinc-js	2015-03-22 00:36:20.055904426 +0100
@@ -9,34 +9,7 @@
 # PARTICULAR PURPOSE.
 ##############################################################################
 
-cygwin=false;
-case "`uname`" in
-    CYGWIN*) cygwin=true ;;
-esac
-
-# Finding the root folder for this Kotlin distribution
-SOURCE=$0;
-SCRIPT=`basename "$SOURCE"`;
-while [ -h "$SOURCE" ]; do
-    SCRIPT=`basename "$SOURCE"`;
-    LOOKUP=`ls -ld "$SOURCE"`;
-    TARGET=`expr "$LOOKUP" : '.*-> \(.*\)$'`;
-    if expr "${TARGET:-.}/" : '/.*/$' > /dev/null; then
-        SOURCE=${TARGET:-.};
-    else
-        SOURCE=`dirname "$SOURCE"`/${TARGET:-.};
-    fi;
-done;
-
-# see #2092
-KOTLIN_HOME=`dirname "$SOURCE"`
-KOTLIN_HOME=`cd "$KOTLIN_HOME"; pwd -P`
-KOTLIN_HOME=`cd "$KOTLIN_HOME"/..; pwd`
-
-if $cygwin; then
-    # Remove spaces from KOTLIN_HOME on windows
-    KOTLIN_HOME=`cygpath --windows --short-name "$KOTLIN_HOME"`
-fi
+: KOTLIN_HOME:=/usr/share/kotlin
 
 [ -n "$JAVA_OPTS" ] || JAVA_OPTS="-Xmx256M -Xms32M -noverify"
 
diff -u a/kotlinc-jvm b/kotlinc-jvm
--- a/kotlinc-jvm	2015-03-18 19:02:16.000000000 +0100
+++ b/kotlinc-jvm	2015-03-22 00:36:27.655904670 +0100
@@ -9,34 +9,7 @@
 # PARTICULAR PURPOSE.
 ##############################################################################
 
-cygwin=false;
-case "`uname`" in
-    CYGWIN*) cygwin=true ;;
-esac
-
-# Finding the root folder for this Kotlin distribution
-SOURCE=$0;
-SCRIPT=`basename "$SOURCE"`;
-while [ -h "$SOURCE" ]; do
-    SCRIPT=`basename "$SOURCE"`;
-    LOOKUP=`ls -ld "$SOURCE"`;
-    TARGET=`expr "$LOOKUP" : '.*-> \(.*\)$'`;
-    if expr "${TARGET:-.}/" : '/.*/$' > /dev/null; then
-        SOURCE=${TARGET:-.};
-    else
-        SOURCE=`dirname "$SOURCE"`/${TARGET:-.};
-    fi;
-done;
-
-# see #2092
-KOTLIN_HOME=`dirname "$SOURCE"`
-KOTLIN_HOME=`cd "$KOTLIN_HOME"; pwd -P`
-KOTLIN_HOME=`cd "$KOTLIN_HOME"/..; pwd`
-
-if $cygwin; then
-    # Remove spaces from KOTLIN_HOME on windows
-    KOTLIN_HOME=`cygpath --windows --short-name "$KOTLIN_HOME"`
-fi
+: KOTLIN_HOME:=/usr/share/kotlin
 
 [ -n "$JAVA_OPTS" ] || JAVA_OPTS="-Xmx256M -Xms32M -noverify"
 
