--- squashfs4.0/squashfs-tools/mksquashfs.c	2009-05-28 14:54:46 +0400
+++ squashfs4.0/squashfs-tools/mksquashfs.c.orig	2009-04-06 01:22:48 +0400
@@ -4491,7 +4491,7 @@
 	if(stat(argv[source + 1], &buf) == -1) {
 		if(errno == ENOENT) { /* Does not exist */
 			fd = open(argv[source + 1], O_CREAT | O_TRUNC | O_RDWR,
-				S_IRWXU);
+				S_IRUSR |S_IWUSR|S_IRGRP|S_IROTH);
 			if(fd == -1) {
 				perror("Could not create destination file");
 				exit(1);
