--- src/unikey-gtk/Makefile.in_original	2008-09-07 17:08:03.000000000 +0700
+++ src/unikey-gtk/Makefile.in	2008-09-07 17:08:44.000000000 +0700
@@ -566,7 +566,7 @@
 
 install-data-am: install-moduleLTLIBRARIES
 	@$(NORMAL_INSTALL)
-	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
+#	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
 
 install-exec-am:
 
@@ -596,7 +596,7 @@
 
 uninstall-am: uninstall-info-am uninstall-moduleLTLIBRARIES
 	@$(NORMAL_INSTALL)
-	$(MAKE) $(AM_MAKEFLAGS) uninstall-hook
+#	$(MAKE) $(AM_MAKEFLAGS) uninstall-hook
 
 .PHONY: CTAGS GTAGS all all-am check check-am clean clean-generic \
 	clean-libtool clean-moduleLTLIBRARIES ctags distclean \
