diff --git a/exo/exo/Makefile.am b/exo/exo/Makefile.am
index 3fc51df..3f36cb5 100644
--- a/exo/exo/Makefile.am
+++ b/exo/exo/Makefile.am
@@ -102,7 +102,7 @@ libexo_0_3_la_CFLAGS =							\
 libexo_0_3_la_LDFLAGS =							\
 	-export-dynamic							\
 	-version-info $(LIBEXO_VERINFO)					\
-	-export-symbols-regex "^[^_].*"					\
+	-Wl,--version-script=libexo.ver					\
 	-no-undefined
 
 libexo_0_3_la_LIBADD =							\
