exo-mount: use 'noatime' if HAL allows it.

diff --git a/exo/exo-mount/exo-mount-hal.c b/exo/exo-mount/exo-mount-hal.c
--- a/exo/exo-mount/exo-mount-hal.c
+++ b/exo/exo-mount/exo-mount-hal.c
@@ -670,6 +703,10 @@ exo_mount_hal_device_mount (ExoMountHalDevice *device,
               /* non-pollable drive... */
               options[n++] = g_strdup ("sync");
             }
+          else if (strcmp (device->fsoptions[m], "noatime") == 0)
+            {
+              options[n++] = g_strdup ("noatime");
+            }
           else if (strcmp (device->fsoptions[m], "longnames") == 0
                 && strcmp (device->fstype, "vfat") == 0)
             {
