--- ./src/linpsk/src/linpsk.cpp.orig	2013-04-28 18:13:37.044407760 +0200
+++ ./src/linpsk/src/linpsk.cpp	2013-04-28 18:16:12.552173566 +0200
@@ -690,7 +690,8 @@
     RxDisplay->RxFreq->setAfcDisplayMode ( settings.ActChannel->AfcProperties() );
     RxDisplay->RxFreq->setAfcMode ( settings.ActChannel->getAfcMode() );
     Control->Display->setPhasePointer ( settings.ActChannel->getPhasePointer() );
-    settings.ActChannel->setParameter ( Extra, &Menu.getParameter() );
+    ExtraParameter ActParam = Menu.getParameter();
+    settings.ActChannel->setParameter ( Extra, &ActParam );
   }
   if ( settings.ActChannel != 0 )
     switch ( settings.ActChannel->getModulationType() )
