--- makepanda/makepanda.py	2010-01-23 12:54:21.000000000 +0100
+++ makepanda/makepanda.py	2010-02-05 18:10:17.334869077 +0100
@@ -478,7 +478,7 @@ if (COMPILER=="LINUX"):
     if (RTDIST):
         SmartPkgEnable("WX",    tool = "wx-config")
     if (RUNTIME):
-        SmartPkgEnable("NPAPI", "",          (), ("xulrunner-*/stable", "xulrunner-*/stable/npapi.h", "nspr*/prtypes.h", "nspr*"))
+        SmartPkgEnable("NPAPI", "",          (), ("xulrunner-*/", "xulrunner-*/npapi.h", "nspr*/prtypes.h", "nspr*"))
     if (sys.platform != "darwin"):
         # CgGL is covered by the Cg framework, and we don't need X11 components on OSX
         if (PkgSkip("NVIDIACG")==0 and not RUNTIME):
