--- notify_orig.c	2011-03-17 21:45:19.471540111 -0300
+++ notify.c	2011-03-17 21:45:34.310354600 -0300
@@ -94,7 +94,7 @@
     if(notification == (NotifyNotification*)0)
     {
         /*notify_notification_close(notification, NULL);*/
-        notification = notify_notification_new(summary, body, icon, NULL);
+        notification = notify_notification_new(summary, body, icon);
         notify_notification_set_urgency(notification, level);
       /*printf("NEW\n");*/
     }
