Index: src/kex/CMakeLists.txt
===================================================================
--- src/kex/CMakeLists.txt	(revision 1231)
+++ src/kex/CMakeLists.txt	(working copy)
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 project(Doom64Ex)
 
-set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/kex/")
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}")
 find_package(ZLIB REQUIRED)
 find_package(PNG REQUIRED)
 find_package(SDL REQUIRED)
@@ -9,7 +9,7 @@
 find_package(SDL_net REQUIRED)
 find_package(OpenGL REQUIRED)
 
-include_directories(${CMAKE_SOURCE_DIR}/kex ${CMAKE_SOURCE_DIR}/kex/Ext ${FluidSynth_INCLUDE_DIR} ${SDL_INCLUDE_DIR} ${PNG_INCLUDE_DIR} ${OPENGL_INCLUDE_DIR})
+include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/Ext ${FluidSynth_INCLUDE_DIR} ${SDL_INCLUDE_DIR} ${PNG_INCLUDE_DIR})
 
 # Get around a issue
 add_definitions(-Ddprintf=_dprintf -D__cdecl=)
@@ -26,7 +26,7 @@
 execute_process(
 	COMMAND svn info
 	COMMAND grep Revision
-	COMMAND grep -o -e "[0-9]\\+"
+	COMMAND grep -m 1 -o -e "[0-9]\\+"
 	COMMAND tr -d "\\n"
 	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
 	OUTPUT_VARIABLE SVNREVISION
Index: src/wadgen/Png.c
===================================================================
--- src/wadgen/Png.c	(revision 1231)
+++ src/wadgen/Png.c	(working copy)
@@ -114,7 +114,7 @@
 		PNG_FILTER_TYPE_DEFAULT);
 
 	// setup palette
-	palette = (png_colorp)Mem_Alloc((16 * numpal) * png_sizeof(png_color));
+	palette = (png_colorp)Mem_Alloc((16 * numpal) * sizeof(png_color));
 
 	// copy dPalette_t data over to png_colorp
 	for(x = 0, j = 0; x < numpal; x++)
