diff -Nur qt3d-1.0-src.orig/util/qt3d/assetviewer/main.cpp qt3d-1.0-src/util/qt3d/assetviewer/main.cpp
--- qt3d-1.0-src.orig/util/qt3d/assetviewer/main.cpp	2012-04-11 12:44:30.000000000 +1000
+++ qt3d-1.0-src/util/qt3d/assetviewer/main.cpp	2012-10-02 20:31:32.588428854 +1000
@@ -57,7 +57,7 @@
     QApplication app(argc, argv);
     QDeclarativeView3D view;
 
-    QString qml = q_get_qmldir(QLatin1String("qml/AssetViewer.qml"));
+    QString qml = QLatin1String("/usr/share/qt/quick3d/resources/examples/assetviewer/qml/AssetViewer.qml");
     view.setSource(QUrl::fromLocalFile(qml));
     view.setWindowTitle(QString("3D Asset Viewer"));
 
