--- drivers/amrmo_init.c~	2010-01-06 20:12:51.000000000 -0300
+++ drivers/amrmo_init.c	2014-05-16 12:28:08.228231227 -0300
@@ -247,7 +247,11 @@
 };
 
 
-static struct pci_device_id amrmo_pci_tbl [] __devinitdata = {
+static struct pci_device_id amrmo_pci_tbl []
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 8, 0)
+__devinitdata
+#endif
+ = {
 	{PCI_VENDOR_ID_SILICON, PCI_DEVICE_ID_SL1900,	   /* SiliconInstr SL1543:3052 */
 	 PCI_ANY_ID, PCI_ANY_ID, 0, 0, SL1543_CARD},
 	{PCI_VENDOR_ID_MOTOROLA, PCI_DEVICE_ID_SL1900,     /* Motorola 1057:3052 */

--- drivers/Makefile
+++ drivers/Makefile
@@ -82,7 +82,7 @@
            $(RM) ${DESTDIR}/dev/slusb$(minor) ; ) echo -n

 else
-ifeq ($(shell grep 'UTS.*"2\.6' $(KERNEL_DIR)/include/generated/utsrelease.h $(KERNEL_DIR)/include/linux/utsrelease.h $(KERNEL_DIR)/include/linux/version.h 2>/dev/null),)
+ifeq ($(shell grep 'UTS.*"\(2\.6\|3\)' $(KERNEL_DIR)/include/generated/utsrelease.h $(KERNEL_DIR)/include/linux/utsrelease.h $(KERNEL_DIR)/include/linux/version.h 2>/dev/null),)

 slusb-objs:= old_st7554.o
 obj:=.

