=== modified file 'opensand-core/src/dvb_rcs/core/BlockDvbTal.cpp'
--- opensand-core/src/dvb_rcs/core/BlockDvbTal.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/dvb_rcs/core/BlockDvbTal.cpp	2014-04-30 21:02:41 +0000
@@ -53,6 +53,7 @@
 
 #include <sstream>
 #include <assert.h>
+#include <unistd.h>
 
 int BlockDvbTal::Downward::Downward::qos_server_sock = -1;
 

=== modified file 'opensand-core/src/dvb_rcs/fmt/FmtSimulation.cpp'
--- opensand-core/src/dvb_rcs/fmt/FmtSimulation.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/dvb_rcs/fmt/FmtSimulation.cpp	2014-04-30 20:54:36 +0000
@@ -42,6 +42,7 @@
 #include <cstdlib>
 #include <errno.h>
 #include <string.h>
+#include <unistd.h>
 
 
 // TODO we say that if modcod id1 < modcod id2, then

=== modified file 'opensand-core/src/lan_adaptation/BlockLanAdaptation.cpp'
--- opensand-core/src/lan_adaptation/BlockLanAdaptation.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/lan_adaptation/BlockLanAdaptation.cpp	2014-04-30 21:50:01 +0000
@@ -48,6 +48,7 @@
 #include <linux/if_tun.h>
 #include <net/if.h>
 #include <errno.h>
+#include <unistd.h>
 
 #define TUNTAP_FLAGS_LEN 4 // Flags [2 bytes] + Proto [2 bytes]
 

=== modified file 'opensand-core/src/physical_layer/plugins/minimal_condition/acm_loop/AcmLoop.cpp'
--- opensand-core/src/physical_layer/plugins/minimal_condition/acm_loop/AcmLoop.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/physical_layer/plugins/minimal_condition/acm_loop/AcmLoop.cpp	2014-04-30 22:07:36 +0000
@@ -41,6 +41,7 @@
 
 #include <errno.h>
 #include <string.h>
+#include <unistd.h>
 
 AcmLoop::AcmLoop():
 	MinimalConditionPlugin(), modcod_table()

=== modified file 'opensand-core/src/sat_carrier/sat_carrier_channel.cpp'
--- opensand-core/src/sat_carrier/sat_carrier_channel.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/sat_carrier/sat_carrier_channel.cpp	2014-04-30 21:57:33 +0000
@@ -37,6 +37,7 @@
 
 #include <netinet/in.h>
 #include <cstring>
+#include <unistd.h>
 
 
 /**

=== modified file 'opensand-core/src/sat_carrier/sat_carrier_udp_channel.cpp'
--- opensand-core/src/sat_carrier/sat_carrier_udp_channel.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/sat_carrier/sat_carrier_udp_channel.cpp	2014-04-30 21:59:33 +0000
@@ -42,6 +42,7 @@
 #include <strings.h>
 #include <arpa/inet.h>
 #include <errno.h>
+#include <unistd.h>
 
 
 /**

=== modified file 'opensand-core/src/sat_carrier/tests/TestSatCarriers.cpp'
--- opensand-core/src/sat_carrier/tests/TestSatCarriers.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/sat_carrier/tests/TestSatCarriers.cpp	2014-04-30 22:01:51 +0000
@@ -53,6 +53,7 @@
 
 #include <sys/ioctl.h>
 #include <fcntl.h>
+#include <unistd.h>
 #include <linux/if_tun.h>
 
 

=== modified file 'opensand-core/src/sat_carrier/tests/test.cpp'
--- opensand-core/src/sat_carrier/tests/test.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/sat_carrier/tests/test.cpp	2014-04-30 22:03:43 +0000
@@ -44,6 +44,7 @@
 #include <string.h>
 #include <signal.h>
 #include <sched.h>
+#include <unistd.h>
 
 
 

=== modified file 'opensand-core/src/system/gw.cpp'
--- opensand-core/src/system/gw.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/system/gw.cpp	2014-04-30 22:13:33 +0000
@@ -70,6 +70,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <signal.h>
+#include <unistd.h>
 
 
 /**

=== modified file 'opensand-core/src/system/sat.cpp'
--- opensand-core/src/system/sat.cpp	2014-04-22 09:41:18 +0000
+++ opensand-core/src/system/sat.cpp	2014-04-30 22:13:13 +0000
@@ -66,6 +66,7 @@
 #include <string.h>
 #include <signal.h>
 #include <sched.h>
+#include <unistd.h>
 
 
 

