--- comm-central/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp~	2011-03-08 18:10:56.964001137 +0200
+++ comm-central/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp	2011-03-08 18:10:56.954000736 +0200
@@ -218,7 +218,7 @@ nsAlertsIconListener::ShowAlert(GdkPixbu
 {
   mNotification = notify_notification_new(mAlertTitle.get(),
                                           mAlertText.get(),
-                                          NULL, NULL);
+                                          NULL);
   if (!mNotification)
     return NS_ERROR_OUT_OF_MEMORY;
