--- HippoDraw-1.21.3.orig/axes/Range.cxx	2007-05-09 00:47:27.000000000 +0600
+++ HippoDraw-1.21.3/axes/Range.cxx	2012-05-08 12:38:48.806176887 +0600
@@ -20,6 +20,7 @@
 
 #include <cassert>
 #include <cmath>
+#include <ostream>
 
 #ifndef _MSC_VER
 using std::floor;
--- HippoDraw-1.21.3.orig/fits/FitsFile.cxx	2007-05-17 23:57:10.000000000 +0600
+++ HippoDraw-1.21.3/fits/FitsFile.cxx	2012-05-08 12:38:48.809510221 +0600
@@ -13,6 +13,7 @@
 
 #include "pattern/string_convert.h"
 
+#include <cstring>
 #include <algorithm>
 #include <stdexcept>
 
--- HippoDraw-1.21.3.orig/minimizers/Fitter.cxx	2006-01-29 03:54:45.000000000 +0500
+++ HippoDraw-1.21.3/minimizers/Fitter.cxx	2012-05-08 12:38:48.812843555 +0600
@@ -13,6 +13,8 @@
 
 #include "StatedFCN.h"
 
+#include <climits>
+#include <cstdlib>
 #include <stdexcept>
 
 using std::string;
--- HippoDraw-1.21.3.orig/minimizers/LMFitter.cxx	2007-04-24 03:57:21.000000000 +0600
+++ HippoDraw-1.21.3/minimizers/LMFitter.cxx	2012-05-08 12:38:48.812843555 +0600
@@ -20,6 +20,7 @@
 
 #include <algorithm>
 
+#include <climits>
 #include <cmath>
 #include <cassert>
 
--- HippoDraw-1.21.3.orig/minimizers/NumLinAlg.cxx	2005-05-17 00:39:20.000000000 +0600
+++ HippoDraw-1.21.3/minimizers/NumLinAlg.cxx	2012-05-08 12:38:48.812843555 +0600
@@ -19,6 +19,7 @@
 #include <cmath>
 #include <cfloat>
 #include <cassert>
+#include <cstdlib>
 
 using std::ofstream;
 using std::ifstream;
--- HippoDraw-1.21.3.orig/projectors/DyHist1DProjector.cxx	2007-04-24 03:57:21.000000000 +0600
+++ HippoDraw-1.21.3/projectors/DyHist1DProjector.cxx	2012-05-08 12:38:48.812843555 +0600
@@ -22,6 +22,7 @@
 #include "datasrcs/NTuple.h"
 
 #include <cassert>
+#include <climits>
 
 using namespace hippodraw;
 
--- HippoDraw-1.21.3.orig/projectors/DyHist2DProjector.cxx	2005-09-12 22:55:10.000000000 +0600
+++ HippoDraw-1.21.3/projectors/DyHist2DProjector.cxx	2012-05-08 12:38:48.812843555 +0600
@@ -26,6 +26,7 @@
 #include "datasrcs/NTuple.h"
 
 #include <cassert>
+#include <climits>
 
 using namespace hippodraw;
 
--- HippoDraw-1.21.3.orig/projectors/EqualEntriesHist1DProjector.cxx	2007-04-24 03:57:21.000000000 +0600
+++ HippoDraw-1.21.3/projectors/EqualEntriesHist1DProjector.cxx	2012-05-08 12:38:48.812843555 +0600
@@ -26,6 +26,7 @@
 #include <numeric>
 
 #include <cfloat>
+#include <climits>
 
 #include <cassert>
 
--- HippoDraw-1.21.3.orig/projectors/Map1Projector.cxx	2007-04-28 00:14:52.000000000 +0600
+++ HippoDraw-1.21.3/projectors/Map1Projector.cxx	2012-05-08 12:38:48.816176889 +0600
@@ -25,6 +25,7 @@
 #include <numeric>
 
 #include <cfloat>
+#include <climits>
 
 #include <cassert>
 
--- HippoDraw-1.21.3.orig/projectors/Map2Projector.cxx	2007-04-24 03:57:21.000000000 +0600
+++ HippoDraw-1.21.3/projectors/Map2Projector.cxx	2012-05-08 12:38:48.816176889 +0600
@@ -24,6 +24,7 @@
 #include <algorithm>
 
 #include <cfloat>
+#include <climits>
 
 #include <cassert>
 
--- HippoDraw-1.21.3.orig/projectors/MapMatrixProjector.cxx	2007-04-24 03:57:21.000000000 +0600
+++ HippoDraw-1.21.3/projectors/MapMatrixProjector.cxx	2012-05-08 12:38:48.816176889 +0600
@@ -25,6 +25,7 @@
 #include <numeric>
 
 #include <cfloat>
+#include <climits>
 #include <cmath>
 
 #include <cassert>
--- HippoDraw-1.21.3.orig/projectors/MeshProjector.cxx	2007-04-24 03:57:21.000000000 +0600
+++ HippoDraw-1.21.3/projectors/MeshProjector.cxx	2012-05-08 12:38:48.816176889 +0600
@@ -24,6 +24,7 @@
 
 #include <algorithm>
 #include <cfloat>
+#include <climits>
 
 #include <cassert>
 
--- HippoDraw-1.21.3.orig/projectors/NTupleProjector.cxx	2007-04-24 03:57:21.000000000 +0600
+++ HippoDraw-1.21.3/projectors/NTupleProjector.cxx	2012-05-08 12:38:48.816176889 +0600
@@ -23,6 +23,7 @@
 #include <algorithm>
 #include <functional>
 #include <stdexcept>
+#include <climits>
 
 #include <cassert>
 
--- HippoDraw-1.21.3.orig/projectors/ProfileProjector.cxx	2007-04-24 03:57:21.000000000 +0600
+++ HippoDraw-1.21.3/projectors/ProfileProjector.cxx	2012-05-08 12:38:48.816176889 +0600
@@ -27,6 +27,7 @@
 #include "datasrcs/NTuple.h"
 
 #include <algorithm>
+#include <climits>
 
 #include <cassert>
 
--- HippoDraw-1.21.3.orig/projectors/StripChartProjector.cxx	2005-10-30 05:10:14.000000000 +0500
+++ HippoDraw-1.21.3/projectors/StripChartProjector.cxx	2012-05-08 12:38:48.819510223 +0600
@@ -20,6 +20,7 @@
 #include "datasrcs/DataSource.h"
 
 #include <algorithm>
+#include <climits>
 
 #include <cassert>
 
--- HippoDraw-1.21.3.orig/qt/QtApp.cxx	2007-03-15 04:28:28.000000000 +0500
+++ HippoDraw-1.21.3/qt/QtApp.cxx	2012-05-08 12:38:48.819510223 +0600
@@ -26,6 +26,7 @@
 
 #include "qdir.h"
 #include <cassert>
+#include <cstdlib>
 
 using std::string;
 
--- HippoDraw-1.21.3.orig/qt/QtViewImp.cxx	2007-06-29 02:03:02.000000000 +0600
+++ HippoDraw-1.21.3/qt/QtViewImp.cxx	2012-05-08 12:38:48.819510223 +0600
@@ -42,6 +42,7 @@
 #endif
 
 #include <cassert>
+#include <cstdlib>
 
 using namespace hippodraw;
 
--- HippoDraw-1.21.3.orig/reps/AxisRepBase.cxx	2007-07-03 00:17:16.000000000 +0600
+++ HippoDraw-1.21.3/reps/AxisRepBase.cxx	2012-05-08 12:41:16.842844394 +0600
@@ -30,11 +30,13 @@
 
 #include <cmath>
 #include <cassert>
+#include <cstdio>
 
 using std::max;
 using std::min;
 using std::string;
 using std::vector;
+using std::sprintf;
 #ifdef __USE_ISOC99
 using std::isfinite;
 #endif
--- HippoDraw-1.21.3.orig/python/PyDataSource.cxx	2007-08-25 04:19:16.000000000 +0600
+++ HippoDraw-1.21.3/python/PyDataSource.cxx	2012-05-08 12:42:00.002844638 +0600
@@ -37,6 +37,10 @@
 #include "numarray/num_util.h"
 #endif
 
+#ifdef HAVE_NUMPY
+#include "numpy/oldnumeric.h"
+#endif
+
 #ifdef HAVE_CFITSIO
 #include "fits/FitsNTuple.h"
 #endif
