--- a/src/coolkey/log.cpp
+++ b/src/coolkey/log.cpp
@@ -18,6 +18,8 @@
  * ***** END COPYRIGHT BLOCK *****/
 
 #include <string>
+#include <string.h>
+#include <stdlib.h>
 #include "mypkcs11.h"
 #include <assert.h>
 #include <stdio.h>
--- a/src/coolkey/machdep.cpp
+++ b/src/coolkey/machdep.cpp
@@ -33,6 +33,8 @@
 #include <sys/stat.h>
 #include <sys/mman.h>
 #include <pthread.h>
+#include <string.h>
+#include <stdlib.h>
 #endif
 
 #ifdef _WIN32
--- a/src/coolkey/object.cpp
+++ b/src/coolkey/object.cpp
@@ -21,6 +21,7 @@
 #include "PKCS11Exception.h"
 #include "object.h"
 #include <algorithm>
+#include <string.h>
 
 using std::find_if;
 
--- a/src/coolkey/slot.cpp
+++ b/src/coolkey/slot.cpp
@@ -25,7 +25,6 @@
 #include "PKCS11Exception.h"
 #include <winscard.h>
 #include "slot.h"
-#include <memory.h>
 #include "zlib.h"
 #include "params.h"
 
@@ -33,7 +32,6 @@
 
 #define MIN(x, y) ((x) < (y) ? (x) : (y))
 
-using std::auto_ptr;
 
 
 #ifdef DEBUG
