--- Makefile.in.patched	2015-04-08 00:45:39.993729211 +0900
+++ Makefile.in	2015-04-08 00:56:17.405672563 +0900
@@ -1165,10 +1165,10 @@
 
 
 HTTPCommonHeaders.h: HTTPCommonHeaders.template.h HTTPCommonHeaders.txt
-	FBCODE_DIR=$(top_srcdir)/.. INSTALL_DIR=$(srcdir) HEADERS_LIST=$(srcdir)/HTTPCommonHeaders.txt gen_HTTPCommonHeaders.h.sh
+	FBCODE_DIR=$(top_srcdir)/.. INSTALL_DIR=$(srcdir) HEADERS_LIST=$(srcdir)/HTTPCommonHeaders.txt $(SHELL) gen_HTTPCommonHeaders.h.sh
 
 HTTPCommonHeaders.cpp: HTTPCommonHeaders.template.gperf HTTPCommonHeaders.txt
-	FBCODE_DIR=$(top_srcdir)/.. INSTALL_DIR=$(srcdir) HEADERS_LIST=$(srcdir)/HTTPCommonHeaders.txt gen_HTTPCommonHeaders.cpp.sh
+	FBCODE_DIR=$(top_srcdir)/.. INSTALL_DIR=$(srcdir) HEADERS_LIST=$(srcdir)/HTTPCommonHeaders.txt $(SHELL) gen_HTTPCommonHeaders.cpp.sh
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
