diff -uNr cat11.3/common/lib/modules/fglrx/build_mod/drmP.h build_mod2/drmP.h
--- cat11.3/common/lib/modules/fglrx/build_mod/drmP.h	2011-03-24 17:00:28.000000000 +0100
+++ cat11.3/common/lib/modules/fglrx/build_mod2/drmP.h	2011-03-30 04:23:15.000000000 +0200
@@ -57,7 +57,6 @@
 #include <linux/pci.h>
 #include <linux/version.h>
 #include <linux/sched.h>
-#include <linux/smp_lock.h>	/* For (un)lock_kernel */
 #include <linux/mm.h>
 #include <linux/pagemap.h>
 #if defined(__alpha__) || defined(__powerpc__)
diff -uNr cat11.3/common/lib/modules/fglrx/build_mod/firegl_public.c build_mod2/firegl_public.c
--- cat11.3/common/lib/modules/fglrx/build_mod/firegl_public.c	2011-03-24 17:00:28.000000000 +0100
+++ cat11.3/common/lib/modules/fglrx/build_mod2/firegl_public.c	2011-03-30 04:32:59.773179890 +0200
@@ -114,7 +114,6 @@
 #include <linux/pci.h>
 #include <linux/wait.h>
 #include <linux/miscdevice.h>
-#include <linux/smp_lock.h>
 // newer SuSE kernels need this
 #include <linux/highmem.h>
 
@@ -1042,9 +1041,6 @@
 
     dev->pubdev.signature = FGL_DEVICE_SIGNATURE;
 
-    for (i = 0; i < __KE_MAX_SPINLOCKS; i++)
-        dev->spinlock[i] = SPIN_LOCK_UNLOCKED;
-
     for (i=0; i < __KE_MAX_SEMAPHORES; i++)
         sema_init(&dev->struct_sem[i], 1);
 
