--- dnsflood.old	2003-12-30 04:53:38.000000000 +0100
+++ dnsflood	2008-03-29 11:43:27.000000000 +0100
@@ -3 +3 @@
-PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
+PATH=/sbin:/bin:/usr/sbin:/usr/bin
@@ -5 +5 @@
-test -f /usr/local/sbin/dns_flood_detector || exit 0
+test -f /usr/sbin/dns_flood_detector || exit 0
@@ -10 +10 @@
-  		start-stop-daemon --start --quiet --exec /usr/local/sbin/dns_flood_detector -- -d
+  		start-stop-daemon --start --quiet --exec /usr/sbin/dns_flood_detector -- -d
@@ -15 +15 @@
-  		start-stop-daemon --stop --quiet --exec /usr/local/sbin/dns_flood_detector
+  		start-stop-daemon --stop --quiet --exec /usr/sbin/dns_flood_detector
@@ -21 +21 @@
-  		start-stop-daemon --stop --quiet --exec /usr/local/sbin/dns_flood_detector
+  		start-stop-daemon --stop --quiet --exec /usr/sbin/dns_flood_detector
@@ -23 +23 @@
-  		start-stop-daemon --stop --quiet --exec /usr/local/sbin/dns_flood_detector
+  		start-stop-daemon --stop --quiet --exec /usr/sbin/dns_flood_detector
@@ -27 +27 @@
-  		start-stop-daemon --start --quiet --exec /usr/local/sbin/dns_flood_detector -- -d
+  		start-stop-daemon --start --quiet --exec /usr/sbin/dns_flood_detector -- -d
@@ -31 +31 @@
-  		echo "Usage: /etc/init.d/dnsflood {start|stop|restart|force-reload}"
+  		echo "Usage: /etc/rc.d/dnsflood {start|stop|restart|force-reload}"
