diff -ru acgmake/configs/config.Linux acgmake.new/configs/config.Linux
--- acgmake/configs/config.Linux	2008-04-15 11:35:56.000000000 -0400
+++ acgmake.new/configs/config.Linux	2009-01-30 15:43:43.000000000 -0500
@@ -88,10 +88,10 @@
 
 QT4_CFLAGS := -DQT_CLEAN_NAMESPACE -DQT_NO_COMPAT -DQT_THREAD_SUPPORT
 QT4_LIBS   := QtCore QtGui QtOpenGL QtSvg QtNetwork QtXml QtTest QtSql QtDesigner
-QT4_INCDIR := /usr/include/qt4 /usr/include/qt4/Qt/ /usr/include/qt4/QtCore/ /usr/include/qt4/QtGui/ /usr/include/qt4/QtAssistant /usr/include/qt4/QtDesigner /usr/include/qt4/QtNetwork /usr/include/qt4/QtOpenGL /usr/include/qt4/QtSql /usr/include/qt4/QtSvg /usr/include/qt4/QtTest /usr/include/qt4/QtUiTools /usr/include/qt4/QtXml
+QT4_INCDIR := /usr/include/Qt /usr/include/QtCore/ /usr/include/QtGui/ /usr/include/QtAssistant /usr/include/QtDesigner /usr/include/QtNetwork /usr/include/QtOpenGL /usr/include/QtSql /usr/include/QtSvg /usr/include/QtTest /usr/include/QtUiTools /usr/include/QtXml
 QT4_LIBDIR := /usr/lib
-MOC4       := /usr/bin/moc-qt4
-UIC4       := /usr/bin/uic-qt4
+MOC4       := /usr/bin/moc
+UIC4       := /usr/bin/uic
 
 
 CUDA_CFLAGS := 
Only in acgmake.new/configs: config.Linux~
