From 8b2ecac801732053b96920aa6ad90853b5223963 Mon Sep 17 00:00:00 2001
From: Raimar <no@email.de>
Date: Sat, 21 Jun 2014 20:06:05 +0200
Subject: [PATCH] changed minimal height of panel from 2 to 1 for autohide

---
 src/configurator.c | 1 +
 src/panel.c        | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/configurator.c b/src/configurator.c
index 62762e2..c3c6081 100644
--- a/src/configurator.c
+++ b/src/configurator.c
@@ -1064,6 +1064,7 @@ void panel_configure( Panel* p, int sel_page )
 
     w = (GtkWidget*)gtk_builder_get_object( builder, "height_when_minimized" );
     gtk_spin_button_set_value(GTK_SPIN_BUTTON(w), p->height_when_hidden);
+    gtk_spin_button_set_range(GTK_SPIN_BUTTON(w), 1, 10);
     g_signal_connect( w, "value-changed",
                       G_CALLBACK(set_height_when_minimized), p);
 
diff --git a/src/panel.c b/src/panel.c
index c9ad576..4077dec 100644
--- a/src/panel.c
+++ b/src/panel.c
@@ -101,7 +101,7 @@ static Panel* panel_allocate(void)
     p->round_corners = 0;
     p->autohide = 0;
     p->visible = TRUE;
-    p->height_when_hidden = 2;
+    p->height_when_hidden = 1;
     p->transparent = 0;
     p->alpha = 255;
     gdk_color_parse("white", &p->gtintcolor);
-- 
2.0.0

