diff -aur xf86-video-sis-0.9.1.pristine/src/sis_vga.c xf86-video-sis-0.9.1.new/src/sis_vga.c
--- xf86-video-sis-0.9.1.pristine/src/sis_vga.c 2012-03-17 03:06:21.148507957 +1100
+++ xf86-video-sis-0.9.1.new/src/sis_vga.c      2012-03-17 03:37:39.487555978 +1100
@@ -1946,11 +1946,17 @@
 SISVGABlankScreen(ScrnInfoPtr pScrn, Bool on)
 {
     SISPtr pSiS = SISPTR(pScrn);
-    UChar  tmp;
+    UChar  tmp, orig;
 
     inSISIDXREG(SISSR, 0x01, tmp);
+    orig = tmp;
     if(on) tmp &= ~0x20;
     else   tmp |= 0x20;
+    /* Only update the hardware if the state changes because the reset will
+     * disrupt the output requiring the screen to resync.
+     */
+    if(orig == tmp)
+        return;
     SiS_SeqReset(pSiS, TRUE);
     outSISIDXREG(SISSR, 0x01, tmp);
     SiS_SeqReset(pSiS, FALSE);
