diff -aur pino3.pristine/CMakeLists.txt pino3.new/CMakeLists.txt
--- pino3.pristine/CMakeLists.txt	2014-10-24 00:57:38.535042858 -0400
+++ pino3.new/CMakeLists.txt	2014-10-24 01:00:11.601961580 -0400
@@ -62,6 +62,8 @@
 pkg_check_modules(REST REQUIRED "rest-0.7 >= 0.7.10")
 #pkg_check_modules(RESTEXTRAS REQUIRED "rest-extras-0.7 >= 0.7.10")
 #pkg_check_modules(LIBINDICATE "indicate >= 0.3")
+pkg_check_modules(GTHREAD REQUIRED "gthread-2.0 >= 2.14.0")
+
 
 if(${LIBINDICATE_FOUND})
   message(STATUS "libindicate found, support enabled...")
@@ -141,6 +143,7 @@
   ${LIBINDICATE_LIBRARIES}
   ${REST_LIBRARIES}
   #${RESTEXTRAS_LIBRARIES}
+  ${GTHREAD_LIBRARIES}
 )
 
 include_directories(src/vapi/)
