diff -Nur tigervnc.orig/CMakeLists.txt tigervnc/CMakeLists.txt
--- tigervnc.orig/CMakeLists.txt	2011-12-07 18:42:34.472840117 +0100
+++ tigervnc/CMakeLists.txt	2011-12-07 20:24:25.933520509 +0100
@@ -292,6 +292,11 @@
 if(USE_INCLUDED_FLTK)
   set(FLTK_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/common/fltk)
   set(FLTK_LIBRARIES)
+  set(HAVE_FLTK_DEAD_KEYS TRUE)
+  set(HAVE_FLTK_CLIPBOARD TRUE)
+  set(HAVE_FLTK_MEDIAKEYS TRUE)
+  set(HAVE_FLTK_FULLSCREEN TRUE)
+  set(HAVE_FLTK_CURSOR TRUE)
   if(APPLE)
     set(FLTK_LIBRARIES "-framework Carbon -framework Cocoa -framework ApplicationServices")
   elseif(NOT WIN32)
@@ -390,8 +395,6 @@
 endif()
 add_subdirectory(vncviewer)
 
-include(cmake/BuildPackages.cmake)
-
 # uninstall
 configure_file("${CMAKE_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
   "cmake_uninstall.cmake" IMMEDIATE @ONLY)
diff -Nur tigervnc.orig/unix/CMakeLists.txt tigervnc/unix/CMakeLists.txt
--- tigervnc.orig/unix/CMakeLists.txt	2011-12-07 18:41:07.151844683 +0100
+++ tigervnc/unix/CMakeLists.txt	2011-12-07 20:24:25.953520309 +0100
@@ -5,4 +5,4 @@
 add_subdirectory(x0vncserver)
 
 install(PROGRAMS vncserver DESTINATION bin)
-install(FILES vncserver.man DESTINATION man/man1 RENAME vncserver.1)
+install(FILES vncserver.man DESTINATION share/man/man1 RENAME vncserver.1)
diff -Nur tigervnc.orig/unix/vncconfig/CMakeLists.txt tigervnc/unix/vncconfig/CMakeLists.txt
--- tigervnc.orig/unix/vncconfig/CMakeLists.txt	2011-12-07 18:41:06.586844687 +0100
+++ tigervnc/unix/vncconfig/CMakeLists.txt	2011-12-07 20:24:25.962520309 +0100
@@ -12,4 +12,4 @@
 target_link_libraries(vncconfig tx rfb network rdr ${X11_LIBRARIES})
 
 install(TARGETS vncconfig DESTINATION bin)
-install(FILES vncconfig.man DESTINATION man/man1 RENAME vncconfig.1)
+install(FILES vncconfig.man DESTINATION share/man/man1 RENAME vncconfig.1)
diff -Nur tigervnc.orig/unix/vncpasswd/CMakeLists.txt tigervnc/unix/vncpasswd/CMakeLists.txt
--- tigervnc.orig/unix/vncpasswd/CMakeLists.txt	2011-12-07 18:41:05.011844797 +0100
+++ tigervnc/unix/vncpasswd/CMakeLists.txt	2011-12-07 20:24:25.962520309 +0100
@@ -6,4 +6,4 @@
 target_link_libraries(vncpasswd tx rfb rdr os)
 
 install(TARGETS vncpasswd DESTINATION bin)
-install(FILES vncpasswd.man DESTINATION man/man1 RENAME vncpasswd.1)
+install(FILES vncpasswd.man DESTINATION share/man/man1 RENAME vncpasswd.1)
diff -Nur tigervnc.orig/unix/x0vncserver/CMakeLists.txt tigervnc/unix/x0vncserver/CMakeLists.txt
--- tigervnc.orig/unix/x0vncserver/CMakeLists.txt	2011-12-07 18:41:00.443844979 +0100
+++ tigervnc/unix/x0vncserver/CMakeLists.txt	2011-12-07 20:24:25.962520309 +0100
@@ -24,4 +24,4 @@
 endif()
 
 install(TARGETS x0vncserver DESTINATION bin)
-install(FILES x0vncserver.man DESTINATION man/man1 RENAME x0vncserver.1)
+install(FILES x0vncserver.man DESTINATION share/man/man1 RENAME x0vncserver.1)
diff -Nur tigervnc.orig/vncviewer/CMakeLists.txt tigervnc/vncviewer/CMakeLists.txt
--- tigervnc.orig/vncviewer/CMakeLists.txt	2011-12-07 18:42:32.226840228 +0100
+++ tigervnc/vncviewer/CMakeLists.txt	2011-12-07 20:25:12.665517869 +0100
@@ -40,7 +40,7 @@
 if(USE_INCLUDED_FLTK)
   target_link_libraries(vncviewer fltk_static)
 endif()
-target_link_libraries(vncviewer rfb network rdr os Xregion ${FLTK_LIBRARIES} ${GETTEXT_LIBRARIES})
+target_link_libraries(vncviewer rfb network rdr os Xregion fontconfig ${FLTK_LIBRARIES} ${GETTEXT_LIBRARIES})
 
 # When building with GnuTLS, librdr depends on ws2_32, so in order to make
 # MinGW happy, we need to put ws2_32 in librdr's target_link_libraries string,
@@ -55,5 +55,5 @@
   install(TARGETS vncviewer DESTINATION bin)
 endif()
 if(UNIX)
-  install(FILES vncviewer.man DESTINATION man/man1 RENAME vncviewer.1)
+  install(FILES vncviewer.man DESTINATION share/man/man1 RENAME vncviewer.1)
 endif()
