diff -ur tigervnc.orig/vncviewer/CMakeLists.txt tigervnc/vncviewer/CMakeLists.txt
--- tigervnc.orig/vncviewer/CMakeLists.txt	2012-08-10 23:47:44.000000000 +0200
+++ tigervnc/vncviewer/CMakeLists.txt	2012-08-11 20:14:08.965227281 +0200
@@ -45,7 +45,7 @@
   add_executable(vncviewer ${VNCVIEWER_SOURCES})
 endif()
 
-target_link_libraries(vncviewer rfb network rdr os Xregion ${FLTK_LIBRARIES} ${GETTEXT_LIBRARIES})
+target_link_libraries(vncviewer rfb network rdr os Xregion ${FLTK_LIBRARIES} png ${GETTEXT_LIBRARIES})
 
 # When building with GnuTLS, librdr depends on ws2_32, so in order to make
 # MinGW happy, we need to put ws2_32 in librdr's target_link_libraries string,
