diff -ur tigervnc.orig/CMakeLists.txt tigervnc/CMakeLists.txt
--- tigervnc.orig/CMakeLists.txt	2012-03-29 14:21:47.116228737 +0200
+++ tigervnc/CMakeLists.txt	2012-04-08 22:47:10.276864008 +0200
@@ -322,13 +322,13 @@
   set(HAVE_FLTK_MEDIAKEYS 1)
   set(HAVE_FLTK_FULLSCREEN 1)
   set(HAVE_FLTK_CURSOR 1)
-  set(HAVE_FLTK_ICONS 1)
+  set(HAVE_FLTK_ICONS 0)
   set(FLTK_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/common/fltk)
   set(FLTK_LIBRARIES)
   if(APPLE)
     set(FLTK_LIBRARIES "-framework Carbon -framework Cocoa -framework ApplicationServices")
   elseif(NOT WIN32)
-    set(FLTK_LIBRARIES "-ldl")
+    set(FLTK_LIBRARIES "-ldl" ${FONTCONFIG_LIB})
   endif()
   message(STATUS "Using included FLTK library")
 endif()
