diff -Nur tigervnc.orig/CMakeLists.txt tigervnc/CMakeLists.txt
--- tigervnc.orig/CMakeLists.txt	2011-11-03 20:01:18.215753000 +0100
+++ tigervnc/CMakeLists.txt	2011-11-05 12:56:27.541329882 +0100
@@ -322,6 +322,11 @@
 if(USE_INCLUDED_FLTK)
   set(FLTK_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/common/fltk)
   set(FLTK_LIBRARIES)
+  set(HAVE_FLTK_DEAD_KEYS TRUE)
+  set(HAVE_FLTK_CLIPBOARD TRUE)
+  set(HAVE_FLTK_MEDIAKEYS TRUE)
+  set(HAVE_FLTK_FULLSCREEN TRUE)
+  set(HAVE_FLTK_CURSOR TRUE)
   if(APPLE)
     set(FLTK_LIBRARIES "-framework Carbon -framework Cocoa -framework ApplicationServices")
   elseif(NOT WIN32)
@@ -420,8 +425,6 @@
 endif()
 add_subdirectory(vncviewer)
 
-include(cmake/BuildPackages.cmake)
-
 # uninstall
 configure_file("${CMAKE_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
   "cmake_uninstall.cmake" IMMEDIATE @ONLY)
diff -Nur tigervnc.orig/unix/CMakeLists.txt tigervnc/unix/CMakeLists.txt
--- tigervnc.orig/unix/CMakeLists.txt	2011-11-03 20:01:18.215753000 +0100
+++ tigervnc/unix/CMakeLists.txt	2011-11-05 12:56:47.126085042 +0100
@@ -5,4 +5,4 @@
 add_subdirectory(x0vncserver)
 
 install(PROGRAMS vncserver DESTINATION bin)
-install(FILES vncserver.man DESTINATION man/man1 RENAME vncserver.1)
+install(FILES vncserver.man DESTINATION share/man/man1 RENAME vncserver.1)
diff -Nur tigervnc.orig/unix/vncconfig/CMakeLists.txt tigervnc/unix/vncconfig/CMakeLists.txt
--- tigervnc.orig/unix/vncconfig/CMakeLists.txt	2011-06-25 09:32:34.047296000 +0200
+++ tigervnc/unix/vncconfig/CMakeLists.txt	2011-11-05 12:57:08.438818547 +0100
@@ -12,4 +12,4 @@
 target_link_libraries(vncconfig tx rfb network rdr ${X11_LIBRARIES})
 
 install(TARGETS vncconfig DESTINATION bin)
-install(FILES vncconfig.man DESTINATION man/man1 RENAME vncconfig.1)
+install(FILES vncconfig.man DESTINATION share/man/man1 RENAME vncconfig.1)
diff -Nur tigervnc.orig/unix/vncpasswd/CMakeLists.txt tigervnc/unix/vncpasswd/CMakeLists.txt
--- tigervnc.orig/unix/vncpasswd/CMakeLists.txt	2011-06-25 09:32:34.047296000 +0200
+++ tigervnc/unix/vncpasswd/CMakeLists.txt	2011-11-05 12:57:29.526554959 +0100
@@ -6,4 +6,4 @@
 target_link_libraries(vncpasswd tx rfb rdr os)
 
 install(TARGETS vncpasswd DESTINATION bin)
-install(FILES vncpasswd.man DESTINATION man/man1 RENAME vncpasswd.1)
+install(FILES vncpasswd.man DESTINATION share/man/man1 RENAME vncpasswd.1)
diff -Nur tigervnc.orig/unix/x0vncserver/CMakeLists.txt tigervnc/unix/x0vncserver/CMakeLists.txt
--- tigervnc.orig/unix/x0vncserver/CMakeLists.txt	2011-08-19 18:41:06.828544000 +0200
+++ tigervnc/unix/x0vncserver/CMakeLists.txt	2011-11-05 12:57:54.131247362 +0100
@@ -24,4 +24,4 @@
 endif()
 
 install(TARGETS x0vncserver DESTINATION bin)
-install(FILES x0vncserver.man DESTINATION man/man1 RENAME x0vncserver.1)
+install(FILES x0vncserver.man DESTINATION share/man/man1 RENAME x0vncserver.1)
diff -Nur tigervnc.orig/vncviewer/CMakeLists.txt tigervnc/vncviewer/CMakeLists.txt
--- tigervnc.orig/vncviewer/CMakeLists.txt	2011-10-05 11:32:28.143081000 +0200
+++ tigervnc/vncviewer/CMakeLists.txt	2011-11-05 12:58:20.376919245 +0100
@@ -55,5 +55,5 @@
   install(TARGETS vncviewer DESTINATION bin)
 endif()
 if(UNIX)
-  install(FILES vncviewer.man DESTINATION man/man1 RENAME vncviewer.1)
+  install(FILES vncviewer.man DESTINATION share/man/man1 RENAME vncviewer.1)
 endif()
