Binary files dmenu-4.3.1.orig/dmenu and dmenu-4.3.1/dmenu differ
diff -p dmenu-4.3.1.orig/dmenu.1 dmenu-4.3.1/dmenu.1
*** dmenu-4.3.1.orig/dmenu.1	2011-05-20 13:19:54.750580965 +0800
--- dmenu-4.3.1/dmenu.1	2011-05-20 13:32:24.873183868 +0800
*************** dmenu \- dynamic menu
*** 20,25 ****
--- 20,27 ----
  .IR color ]
  .RB [ \-sf
  .IR color ]
+ .RB [ \-m
+ .IR screen ]
  .RB [ \-v ]
  .P
  .BR dmenu_run " ..."
*************** defines the selected background color.
*** 77,82 ****
--- 79,87 ----
  .BI \-sf " color"
  defines the selected foreground color.
  .TP
+ .BI \-sf " screen"
+ selects screen to be displayed on.
+ .TP
  .B \-v
  prints version information to stdout, then exits.
  .SH USAGE
diff -p dmenu-4.3.1.orig/dmenu.c dmenu-4.3.1/dmenu.c
*** dmenu-4.3.1.orig/dmenu.c	2011-05-20 13:19:54.750580965 +0800
--- dmenu-4.3.1/dmenu.c	2011-05-20 13:33:41.539435198 +0800
*************** static char text[BUFSIZ] = "";
*** 44,49 ****
--- 44,50 ----
  static int bh, mw, mh;
  static int inputw, promptw;
  static int lines = 0;
+ static int selscreen = -1;
  static size_t cursor = 0;
  static const char *font = NULL;
  static const char *prompt = NULL;
*************** main(int argc, char *argv[]) {
*** 97,102 ****
--- 98,105 ----
  			selbgcolor = argv[++i];
  		else if(!strcmp(argv[i], "-sf"))
  			selfgcolor = argv[++i];
+ 		else if(!strcmp(argv[i], "-m"))
+ 			selscreen = atoi(argv[++i]);
  		else
  			usage();
  
*************** setup(void) {
*** 518,527 ****
  		unsigned int du;
  		Window dw;
  
! 		XQueryPointer(dc->dpy, root, &dw, &dw, &x, &y, &di, &di, &du);
! 		for(i = 0; i < n-1; i++)
! 			if(INRECT(x, y, info[i].x_org, info[i].y_org, info[i].width, info[i].height))
! 				break;
  		x = info[i].x_org;
  		y = info[i].y_org + (topbar ? 0 : info[i].height - mh);
  		mw = info[i].width;
--- 521,534 ----
  		unsigned int du;
  		Window dw;
  
! 		if (selscreen >= 0 && selscreen < n)
! 			i = selscreen;
! 		else {
! 			XQueryPointer(dc->dpy, root, &dw, &dw, &x, &y, &di, &di, &du);
! 			for(i = 0; i < n-1; i++)
! 				if(INRECT(x, y, info[i].x_org, info[i].y_org, info[i].width, info[i].height))
! 					break;
! 		}
  		x = info[i].x_org;
  		y = info[i].y_org + (topbar ? 0 : info[i].height - mh);
  		mw = info[i].width;
*************** setup(void) {
*** 555,560 ****
  void
  usage(void) {
  	fputs("usage: dmenu [-b] [-f] [-i] [-l lines] [-p prompt] [-fn font]\n"
! 	      "             [-nb color] [-nf color] [-sb color] [-sf color] [-v]\n", stderr);
  	exit(EXIT_FAILURE);
  }
--- 562,567 ----
  void
  usage(void) {
  	fputs("usage: dmenu [-b] [-f] [-i] [-l lines] [-p prompt] [-fn font]\n"
! 	      "             [-nb color] [-nf color] [-sb color] [-sf color] [-m screen] [-v]\n", stderr);
  	exit(EXIT_FAILURE);
  }
Binary files dmenu-4.3.1.orig/dmenu.o and dmenu-4.3.1/dmenu.o differ
