diff -c src/mkinit/init.mk src/build/init.mk
*** src/mkinit/init.mk	2011-03-22 06:42:38.000000000 +0000
--- src/build/init.mk	2011-03-22 07:01:20.000000000 +0000
***************
*** 2,8 ****
  # See COPYING for terms
  
  # Config
! MKSHELL=/usr/lib/plan9/bin/rc
  NPROC=10
  
  hostname-opts = c
--- 2,8 ----
  # See COPYING for terms
  
  # Config
! MKSHELL=/opt/plan9/bin/rc
  NPROC=10
  
  hostname-opts = c
diff -c src/mkinit/services.mk src/build/services.mk
*** src/mkinit/services.mk	2011-03-22 06:42:38.000000000 +0000
--- src/build/services.mk	2011-03-22 07:00:48.000000000 +0000
***************
*** 1,4 ****
! MKSHELL=/usr/lib/plan9/bin/rc
  
  # Example
  #test-start:VQPservice -u: foo-start
--- 1,4 ----
! MKSHELL=/opt/plan9/bin/rc
  
  # Example
  #test-start:VQPservice -u: foo-start
Lika underkataloger: src/mkinit/src och src/build/src
Lika underkataloger: src/mkinit/.svn och src/build/.svn
