diff -aur mvsis-1.3/helpers/config.sub src/mvsis-1.3/helpers/config.sub
--- mvsis-1.3/helpers/config.sub	2003-02-24 23:24:10.000000000 +0100
+++ src/mvsis-1.3/helpers/config.sub	2011-10-01 16:23:32.000000000 +0200
@@ -164,6 +164,8 @@
 	i[3456]86)
 	  basic_machine=$basic_machine-pc
 	  ;;
+        x86_64-pc)
+          ;;
 	# Object if more than one company name word.
 	*-*-*)
 		echo Invalid configuration \`$1\': machine \`$basic_machine\' not recognized 1>&2
diff -aur mvsis-1.3/src/base/ntki/ntki.h src/mvsis-1.3/src/base/ntki/ntki.h
--- mvsis-1.3/src/base/ntki/ntki.h	2005-05-18 06:14:37.000000000 +0200
+++ src/mvsis-1.3/src/base/ntki/ntki.h	2011-10-01 16:29:13.000000000 +0200
@@ -80,7 +80,7 @@
 extern void              Net_NetworkEquiv( Ntk_Network_t * pNet, int fFuncRed, int fFeedBack, int fDoSparse, int fVerbose );
 /*=== ntkiFrames.c =====================================================*/
 extern Ntk_Network_t *   Ntk_NetworkDeriveTimeFrames( Ntk_Network_t * pNet, int nFrames );
-extern void              Ntk_NetworkReorderCiCo( Ntk_Network_t * pNet );
+//extern void              Ntk_NetworkReorderCiCo( Ntk_Network_t * pNet );
 extern void              Ntk_NetworkMakeCombinational( Ntk_Network_t * pNet );
 /*=== ntkiFxu.c =========================================================*/
 extern bool              Ntk_NetworkFastExtract( Ntk_Network_t * pNet, Fxu_Data_t * p );
diff -aur mvsis-1.3/src/graph/wn/wnStrashBin.c src/mvsis-1.3/src/graph/wn/wnStrashBin.c
--- mvsis-1.3/src/graph/wn/wnStrashBin.c	2004-05-12 06:30:11.000000000 +0200
+++ src/mvsis-1.3/src/graph/wn/wnStrashBin.c	2011-10-01 16:39:07.000000000 +0200
@@ -77,7 +77,8 @@
 
     // clean the data of the nodes in the window
     Ntk_NetworkForEachNodeSpecial( pWnd->pNet, pNode )
-        pNode->pCopy = (Ntk_Node_t *)pNode->pData = NULL;
+    pNode->pData = NULL;
+        pNode->pCopy = (Ntk_Node_t *)pNode->pData;
 
     // set the leaves
     pgInputs = Sh_ManagerReadVars( pMan );
