diff -Naur a/src/config.py b/src/config.py
--- a/src/config.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/config.py	2010-12-06 02:38:19.273333338 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
diff -Naur a/src/dat.py b/src/dat.py
--- a/src/dat.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/dat.py	2010-12-06 02:37:22.950000003 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
diff -Naur a/src/db.py b/src/db.py
--- a/src/db.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/db.py	2010-12-06 02:37:38.266666670 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
diff -Naur a/src/downloaders.py b/src/downloaders.py
--- a/src/downloaders.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/downloaders.py	2010-12-06 02:37:51.596666670 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
diff -Naur a/src/drm.py b/src/drm.py
--- a/src/drm.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/drm.py	2010-12-06 02:37:58.510000004 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
diff -Naur a/src/files.py b/src/files.py
--- a/src/files.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/files.py	2010-12-06 02:38:12.506666672 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
diff -Naur a/src/game.py b/src/game.py
--- a/src/game.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/game.py	2010-12-06 02:37:30.943333337 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
diff -Naur a/src/globals.py b/src/globals.py
--- a/src/globals.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/globals.py	2010-12-06 02:37:15.783333339 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
diff -Naur a/src/gui.py b/src/gui.py
--- a/src/gui.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/gui.py	2010-12-06 02:37:44.010000004 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
diff -Naur a/src/updaters.py b/src/updaters.py
--- a/src/updaters.py	2010-12-01 17:19:36.000000000 +0100
+++ b/src/updaters.py	2010-12-06 02:38:05.623333337 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # DRM - DsRomsManager
 #
