--- build_python.sh.original	2009-04-05 07:10:59.000000000 +0200
+++ build_python.sh	2010-04-08 15:07:30.000000000 +0200
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-SOURCE=`dirname "$0"`
+SOURCE=`pwd`
 PWD=`pwd`
 BUILD=$PWD/build
 INSTALL=$PWD/install
@@ -72,7 +72,7 @@
     try cp -Rp "$SOURCE/bzip2-1.0.3" "$BUILD/bzip2-1.0.3"
     try cd "$BUILD/bzip2-1.0.3"
 
-    try make CC="$CC" LD="$LD" CXX="$CXX" CXXLD="$CXXLD"
+    try make CC="$CC -fPIC" LD="$LD" CXX="$CXX -fPIC" CXXLD="$CXXLD"
     try make install PREFIX="$INSTALL"
     try cd "$BUILD"
     try touch built.bz2

