--- distribute.rpy	2010-04-08 15:02:24.541624144 +0200
+++ distribute.rpy.original	2010-01-15 03:59:27.000000000 +0100
@@ -245,14 +245,12 @@
         # Do we have the files?
         has_windows = dist_exists("renpy.exe")
         has_linux = dist_exists("lib/linux-x86")
-        has_linux64 = dist_exists("lib/linux-x86_64")
         has_mac = dist_exists("renpy.app")
-        has_all = has_windows and has_mac and has_linux and has_linux64
+        has_all = has_windows and has_mac and has_linux
 
         # Should we build these distributions?
         build_windows = has_windows and project.info.get("build_windows", has_windows)
         build_linux = has_linux and project.info.get("build_linux", has_linux)
-        build_linux64 = has_linux64 and project.info.get("build_linux64", has_linux64)
         build_mac = has_mac and project.info.get("build_mac", has_mac)
         build_all = has_all and project.info.get("build_all", False)
 
@@ -301,10 +299,6 @@
                 toggle_button(_(u"Linux x86"), build_linux, ui.returns("build_linux"),
                               _(u"Tar.Bz2 distribution for the Linux x86 platform."))
 
-            if has_linux64:
-                toggle_button(_(u"Linux x86_64"), build_linux64, ui.returns("build_linux64"),
-                              _(u"Tar.Bz2 distribution for the Linux x86_64 platform."))
-
             if has_mac:
                 toggle_button(_(u"Macintosh Universal"), build_mac, ui.returns("build_mac"),
                               _(u"Single application distribution for the Macintosh x86 and ppc platforms."))
@@ -327,8 +321,6 @@
                 build_windows = not build_windows
             elif act == "build_linux":
                 build_linux = not build_linux
-            elif act == "build_linux64":
-                build_linux64 = not build_linux64
             elif act == "build_mac":
                 build_mac = not build_mac
             elif act == "build_all":
@@ -373,7 +365,6 @@
 
         project.info["build_windows"] = build_windows
         project.info["build_linux"] = build_linux
-        project.info["build_linux64"] = build_linux64
         project.info["build_mac"] = build_mac
         project.info["build_all"] = build_all
 
@@ -395,7 +386,6 @@
         multi_files = [ ]
         win_files = [ ]
         linux_files = [ ]
-        linux_files64 = [ ]
         mac_files = [ ]
 
         # A map from source file name to replacement data to be placed in
@@ -471,19 +461,6 @@
 
             
         ######################################################################
-        # Linux x86_64 files.
-        
-        if build_linux64 or build_all:
-
-            linux_files64.append((rb + "renpy.sh", "/" + executable_name + ".sh"))
-            linux_files64.append((rb + "lib", "/lib"))
-            linux_files64.append((rb + "lib/python", "/lib/python"))
-            linux_files64.extend(tree(rb + "lib/linux-x86_64", "/lib/linux-x86_64"))
-
-            # Warning: The tar.bz2 builder doesn't support file_data.
-
-            
-        ######################################################################
         # Mac (non-app) files.
         
         if build_mac or build_all:
@@ -526,13 +503,6 @@
                 multi_files + linux_files)
                 
 
-        if build_linux64:
-            make_tar(
-                _(u"Building Linux x86_64..."),
-                base_name + "-linux-x86_64",
-                multi_files + linux_files64)
-                
-
         if build_mac:
 
             # Reorganize the files so all the non application files live inside
@@ -564,7 +534,7 @@
             make_zip(
                 _(u"Building Combined..."),
                 base_name + "-all",
-                multi_files + win_files + linux_files + linux_files64 + mac_files,
+                multi_files + win_files + linux_files + mac_files,
                 file_data)
 
         # Report success to the user.
