--- build.original.sh	2009-12-12 23:08:18.000000000 +0100
+++ build.sh	2010-04-07 02:38:08.412228052 +0200
@@ -43,14 +43,14 @@
 # export LDFLAGS="-gstabs -L$INSTALL/lib $LDFLAGS"
 
 # Production
-export CFLAGS="$CFLAGS -O3 -I$INSTALL/include -I$INSTALL/include/freetype2"
-export CXXFLAGS="$CXXFLAGS -O3 -I$INSTALL/include -I$INSTALL/include/freetype2"
+export CFLAGS="$CFLAGS -O3 -I$INSTALL/include -fPIC -I$INSTALL/include/freetype2"
+export CXXFLAGS="$CXXFLAGS -O3 -I$INSTALL/include -fPIC -I$INSTALL/include/freetype2"
 export LDFLAGS="-O3 -L$INSTALL/lib $LDFLAGS"
 
 
 if [ "x$MSYSTEM" != "x" ]; then
-  export CFLAGS="$CFLAGS -fno-strict-aliasing"
-  export CXXFLAGS="$CXXFLAGS -fno-strict-aliasing"
+  export CFLAGS="$CFLAGS -fno-strict-aliasing -fPIC"
+  export CXXFLAGS="$CXXFLAGS -fno-strict-aliasing -fPIC"
 fi
 
 OLD_CC="$CC"
