diff --git a/server/cloud9/middleware.js b/server/cloud9/middleware.js
index dac6234..5971d88 100644
--- a/server/cloud9/middleware.js
+++ b/server/cloud9/middleware.js
@@ -24,16 +24,9 @@ exports.staticProvider = function (root, mount) {
 
         if (pathname.indexOf(mount) === 0) {
             request.url = url.replace(mount, "") || "/";
-            staticGzip(request, response, function (err) {
-                if (err) {
-                    request.url = url;
-                    return next(err);
-                }
-
-                staticProvider(request, response, function (err) {
-                    request.url = url;
-                    next(err);
-                });
+            staticProvider(request, response, function (err) {
+                request.url = url;
+                next(err);
             });
         } else
             next();
