--- Makefile.am.orig	2011-01-23 16:01:26.000000000 +0100
+++ Makefile.am	2011-01-23 16:03:08.000000000 +0100
@@ -76,8 +76,8 @@ clean_lib:
 	cd $(top_builddir)/third-party/opcodes/ && make distclean
 	cd $(top_builddir)/third-party/readline/ && make distclean
 install-data-hook:
-	rm -f -r $(prefix)/testsuite && mkdir $(prefix)/testsuite
-	cp -a $(top_srcdir)/testsuite/arm_hello $(prefix)/testsuite/arm_hello/
+	rm -f -r ${DESTDIR}$(prefix)/testsuite && mkdir ${DESTDIR}$(prefix)/testsuite
+	cp -a $(top_srcdir)/testsuite/arm_hello ${DESTDIR}$(prefix)/testsuite/arm_hello/
 #	rm -f -r $(prefix)/conf && mkdir $(prefix)/conf
 #	cp -a $(top_srcdir)/conf/* $(prefix)/conf
 
