--- gnome15-panel-applet-0.0.1_orig/g15-applet	2011-03-12 22:18:30.836666681 +0100
+++ gnome15-panel-applet-0.0.1/g15-applet	2011-03-12 22:21:52.263333347 +0100
@@ -42,7 +42,7 @@
 loop = gobject.MainLoop()
 
 # Allow running from local path
-path = os.path.join(os.path.abspath(os.path.dirname(sys.argv[0])), "..", "gnome15", "src", "main", "python")
+path = os.path.join(os.path.abspath(os.path.dirname(sys.argv[0])), "..", "gnome15", "src", "main", "python2")
 if os.path.exists(path):
     sys.path.insert(0, path)
 
