diff -ur PyQt-gpl-5.0.orig/configure.py PyQt-gpl-5.0/configure.py
--- PyQt-gpl-5.0.orig/configure.py	2013-06-29 08:34:48.257986767 -0400
+++ PyQt-gpl-5.0/configure.py	2013-06-29 08:35:03.671376052 -0400
@@ -59,7 +59,7 @@
 
 # The module meta-data.
 MODULE_METADATA = {
-    'dbus':             ModuleMetadata(qmake_QT=['-gui'], qmake_TARGET='qt',
+    'dbus':             ModuleMetadata(qmake_QT=['-gui'], qmake_TARGET='qt5',
                                 in_consolidated=False),
     'QAxContainer':     ModuleMetadata(qmake_QT=['axcontainer']),
     'Qt':               ModuleMetadata(qmake_QT=['-core', '-gui'],
diff -ur PyQt-gpl-5.0.orig/dbus/dbus.cpp PyQt-gpl-5.0/dbus/dbus.cpp
--- PyQt-gpl-5.0.orig/dbus/dbus.cpp	2013-06-29 08:34:48.304653603 -0400
+++ PyQt-gpl-5.0/dbus/dbus.cpp	2013-06-29 08:36:10.278284654 -0400
@@ -401,29 +401,29 @@
 
 // The module entry point.
 #if PY_MAJOR_VERSION >= 3
-PyMODINIT_FUNC PyInit_qt()
+PyMODINIT_FUNC PyInit_qt5()
 {
     static PyModuleDef module_def = {
         PyModuleDef_HEAD_INIT,
-        "qt",
+        "qt5",
         NULL,
         -1,
         module_functions,
     };
 
     // Import the generic part of the Python DBus bindings.
-    if (import_dbus_bindings("dbus.mainloop.qt") < 0)
+    if (import_dbus_bindings("dbus.mainloop.qt5") < 0)
         return 0;
 
     return PyModule_Create(&module_def);
 }
 #else
-PyMODINIT_FUNC initqt()
+PyMODINIT_FUNC initqt5()
 {
     // Import the generic part of the Python DBus bindings.
-    if (import_dbus_bindings("dbus.mainloop.qt") < 0)
+    if (import_dbus_bindings("dbus.mainloop.qt5") < 0)
         return;
 
-    Py_InitModule("qt", module_functions);
+    Py_InitModule("qt5", module_functions);
 }
 #endif
