--- configure~	2011-05-07 19:42:51.000000000 +0900
+++ configure	2011-05-19 21:02:38.073679278 +0900
@@ -2895,9 +2895,9 @@
 ##########################################################################
 if test "$OS" = CYGWIN
 then
-    CFLAGS="-I. -I$myprefix/include -L$myprefix/lib -L."
+    CFLAGS="-I. -L. -I$myprefix/include -L$myprefix/lib"
 else
-    CFLAGS="-I. -I$myprefix/include -L$myprefix/lib -fPIC -L."
+    CFLAGS="-I. -L. -I$myprefix/include -L$myprefix/lib -fPIC"
 fi
 
 if test -d /usr/local/include
