diff -rauN wt-3.1.9/examples/filetreetable/FileTreeTableNode.C wt-3.1.9-filetreetable-example-patch/examples/filetreetable/FileTreeTableNode.C
--- wt-3.1.9/examples/filetreetable/FileTreeTableNode.C	2011-04-07 16:09:11.000000000 +0200
+++ wt-3.1.9-filetreetable-example-patch/examples/filetreetable/FileTreeTableNode.C	2011-04-09 09:49:24.000000000 +0200
@@ -22,12 +22,12 @@
 FileTreeTableNode::FileTreeTableNode(const boost::filesystem::path& path)
 #if BOOST_FILESYSTEM_VERSION < 3
 #ifndef WT_NO_STD_WSTRING
-  : WTreeTableNode(Wt::widen(path.leaf()), createIcon(path)),
+  : WTreeTableNode(Wt::widen(path.leaf().string()), createIcon(path)),
 #else
-  : WTreeTableNode(path.leaf(), createIcon(path)),
+  : WTreeTableNode(Wt::WString(path.leaf().string()), createIcon(path)),
 #endif
 #else
-  : WTreeTableNode(path.leaf(), createIcon(path)),
+  : WTreeTableNode(Wt::WString(path.leaf().string()), createIcon(path)),
 #endif
     path_(path)
 {
