diff -rauN wt-3.1.9/src/Wt/FontSupportSimple.C wt-3.1.9-fontsupport-boost-fs-3-patch/src/Wt/FontSupportSimple.C
--- wt-3.1.9/src/Wt/FontSupportSimple.C	2011-02-23 10:21:36.000000000 +0100
+++ wt-3.1.9-fontsupport-boost-fs-3-patch/src/Wt/FontSupportSimple.C	2011-05-16 18:15:00.000000000 +0200
@@ -208,7 +208,11 @@
 			    const boost::filesystem::path& path,
 			    FontMatch& match) const
 {
+#if BOOST_FILESYSTEM_VERSION < 3
   std::string f = Utils::lowerCase(path.leaf());
+#else
+  std::string f = Utils::lowerCase(path.leaf().string());
+#endif
 
   if (boost::ends_with(f, ".ttf")
       || boost::ends_with(f, ".ttc")) {
