Index: src/wx/CMakeLists.txt
===================================================================
--- src/wx/CMakeLists.txt	(Revision 1231)
+++ src/wx/CMakeLists.txt	(Arbeitskopie)
@@ -58,6 +58,12 @@
     FIND_PACKGE ( DirectX REQUIRED )
 ENDIF(WIN32 AND ENABLE_DIRECTX)
 
+# Favour wxrc-2.8 over wxrc.
+FIND_PROGRAM( WXRC_COMMAND NAMES wxrc-2.8 wxrc )
+IF( NOT WXRC_COMMAND )
+    MESSAGE( FATAL_ERROR "wxrc-2.8 or wxrc not found in your PATH! Is wxgtk installed?" )
+ENDIF( NOT WXRC_COMMAND )
+
 # contrib widgets
 include_directories(widgets)
 
@@ -100,7 +106,7 @@
                    # doing this in its own dir prevents name prefixes
 		   COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/wxvbam.xrc wxvbam.xrc
 		   COMMAND ${CMAKE_COMMAND} -E copy ${ICO_PARENT_DIR}/${WX_XRC_ICON} ${WX_XRC_ICON}
-                   COMMAND wxrc wxvbam.xrc -o wxvbam.xrs
+                   COMMAND ${WXRC_COMMAND} wxvbam.xrc -o wxvbam.xrs
                    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 	           DEPENDS ${ICO_PARENT_DIR}/${XRC_ITEM}
 		   DEPENDS wxvbam.xrc)
