diff -aur rdiff-backup-1.2.8/rdiff_backup/rpath.py rdiff-backup-1.2.8-kyr/rdiff_backup/rpath.py
--- rdiff-backup-1.2.8/rdiff_backup/rpath.py	2009-03-16 16:36:21.000000000 +0200
+++ rdiff-backup-1.2.8-kyr/rdiff_backup/rpath.py	2010-06-01 00:04:40.883735847 +0300
@@ -353,10 +353,14 @@
 
 	"""
 	assert rpath.conn is Globals.local_connection
-	s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
-	try: s.bind(rpath.path)
-	except socket.error, exc:
-		raise SkipFileException("Socket error: " + str(exc))
+	try: rpath.conn.os.mknod(rpath.path, stat.S_IFSOCK)
+	except (OSError, AttributeError):
+ 		# AttributeError will be raised by Python 2.2, which
+ 		# doesn't have os.mknod
+ 		s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+ 		try: s.bind(rpath.path)
+ 		except socket.error, exc:
+			raise SkipFileException("Socket error: " + str(exc))
 
 def gzip_open_local_read(rpath):
 	"""Return open GzipFile.  See security note directly above"""
