# Sets paths so launcher will launch correct game

--- platform.cpp.orig	2013-07-19 15:56:04.469189831 +0200
+++ platform.cpp	2013-07-19 15:56:29.614663988 +0200
@@ -82,8 +82,8 @@
     rlp.rlim_cur = min(OPEN_MAX, rlp.rlim_max);
     setrlimit(RLIMIT_NOFILE, &rlp);
 #elif defined Q_WS_X11
-    *game_config_dir()          = "/usr/share/spicyhorse/" + executable_file_info.baseName() + "/game_config/";
-    *game_config_file()         = "/usr/share/spicyhorse/" + executable_file_info.baseName() + "/game_config/game.cfg";
+    *game_config_dir()          = "/usr/share/akaneiro-launcher/";
+    *game_config_file()         = "/usr/share/akaneiro-launcher/game.cfg";
     *platform_id()              = "lin";
 
     // TODO: check if it is good idea to do that with user privileges.

--- tests/game_config/game.cfg	2013-07-19 15:59:24.056391181 +0200
+++ game.cfg	2013-04-22 07:54:00.000000000 +0200
@@ -1,7 +1,7 @@
 [global]
 name=akaneiro
-channel="http://127.0.0.1:5000/channel/spicyhorse/aka/"
-executable="mac/Akaneiro.app/Contents/MacOS/Akaneiro"
+channel="http://delivery.spicyhorse.com/channel/spicyhorse/aka/"
+executable="lin/Akaneiro.x86"
 play_on_error=True
 
 [gui]
