--- a/xen/drivers/passthrough/amd/iommu_acpi.c	2013-07-09 05:46:56.000000000 -0500
+++ b/xen/drivers/passthrough/amd/iommu_acpi.c	2013-08-31 01:26:18.614846438 -0500
@@ -651,9 +651,9 @@
         return 0;
     }
 
-    AMD_IOMMU_DEBUG("IVHD Special: %04x:%02x:%02x.%u variety %#x handle %#x\n",
+    AMD_IOMMU_DEBUG("IVHD Special: %04x:%02x:%02x.%u variety %#x handle %#x used_id %#x\n",
                     seg, PCI_BUS(bdf), PCI_SLOT(bdf), PCI_FUNC(bdf),
-                    special->variety, special->handle);
+                    special->variety, special->handle, special->used_id);
     add_ivrs_mapping_entry(bdf, bdf, special->header.data_setting, iommu);
 
     switch ( special->variety )
