diff -Naur xen/Config.mk xenb/Config.mk
--- xen/Config.mk	2013-03-09 10:53:08.509845278 -0700
+++ xenb/Config.mk	2013-03-09 10:56:26.796511981 -0700
@@ -174,7 +174,6 @@
 $(call cc-option-add,HOSTCFLAGS,HOSTCC,-Wdeclaration-after-statement)
 $(call cc-option-add,CFLAGS,CC,-Wdeclaration-after-statement)
 $(call cc-option-add,CFLAGS,CC,-Wno-unused-but-set-variable)
-$(call cc-option-add,CFLAGS,CC,-Wno-unused-local-typedefs)
 
 LDFLAGS += $(foreach i, $(EXTRA_LIB), -L$(i)) 
 CFLAGS += $(foreach i, $(EXTRA_INCLUDES), -I$(i))
diff -Naur xen/p xenb/p
--- xen/p	1969-12-31 17:00:00.000000000 -0700
+++ xenb/p	2013-03-09 10:57:51.199845278 -0700
@@ -0,0 +1,87 @@
+# HG changeset patch
+# User Michael Young <m.a.young@durham.ac.uk>
+# Date 1360774815 0
+# Node ID 742dde457258422a3d08e3ddbf9a7eae55c93acb
+# Parent  63594ce1708ffb1269cd124c3a864485eeba1736
+tools: Fix memset(&p,0,sizeof(p)) idiom in several places.
+
+gcc 4.8 identifies several places where code of the form memset(x, 0,
+sizeof(x)); is used incorrectly, meaning that less memory is set to
+zero than required.
+
+Signed-off-by: Michael Young <m.a.young@durham.ac.uk>
+Committed-by: Keir Fraser <keir@xen.org>
+
+diff -r 63594ce1708f -r 742dde457258 tools/blktap2/drivers/md5.c
+--- a/tools/blktap2/drivers/md5.c	Wed Feb 13 09:31:48 2013 +0100
++++ b/tools/blktap2/drivers/md5.c	Wed Feb 13 17:00:15 2013 +0000
+@@ -174,7 +174,7 @@ void MD5Final(unsigned char digest[16], 
+     MD5Transform(ctx->buf, (uint32_t *) ctx->in);
+     byteReverse((unsigned char *) ctx->buf, 4);
+     memcpy(digest, ctx->buf, 16);
+-    memset(ctx, 0, sizeof(ctx));     /* In case it's sensitive */
++    memset(ctx, 0, sizeof(*ctx));     /* In case it's sensitive */
+ }
+ 
+ /* The four core functions - F1 is optimized somewhat */
+diff -r 63594ce1708f -r 742dde457258 tools/debugger/kdd/kdd-xen.c
+--- a/tools/debugger/kdd/kdd-xen.c	Wed Feb 13 09:31:48 2013 +0100
++++ b/tools/debugger/kdd/kdd-xen.c	Wed Feb 13 17:00:15 2013 +0000
+@@ -333,7 +333,7 @@ int kdd_get_regs(kdd_guest *g, int cpuid
+     if (!cpu) 
+         return -1;
+ 
+-    memset(r, 0, sizeof(r));
++    memset(r, 0, sizeof(*r));
+     
+     if (w64)
+         kdd_get_regs_x86_64(cpu, &r->r64);
+diff -r 63594ce1708f -r 742dde457258 tools/libxc/xc_dom_boot.c
+--- a/tools/libxc/xc_dom_boot.c	Wed Feb 13 09:31:48 2013 +0100
++++ b/tools/libxc/xc_dom_boot.c	Wed Feb 13 17:00:15 2013 +0000
+@@ -266,7 +266,7 @@ int xc_dom_boot_image(struct xc_dom_imag
+         return rc;
+ 
+     /* let the vm run */
+-    memset(ctxt, 0, sizeof(ctxt));
++    memset(ctxt, 0, sizeof(*ctxt));
+     if ( (rc = dom->arch_hooks->vcpu(dom, ctxt)) != 0 )
+         return rc;
+     xc_dom_unmap_all(dom);
+diff -r 63594ce1708f -r 742dde457258 tools/libxl/libxl_qmp.c
+--- a/tools/libxl/libxl_qmp.c	Wed Feb 13 09:31:48 2013 +0100
++++ b/tools/libxl/libxl_qmp.c	Wed Feb 13 17:00:15 2013 +0000
+@@ -378,7 +378,7 @@ static int qmp_open(libxl__qmp_handler *
+     ret = libxl_fd_set_cloexec(qmp->ctx, qmp->qmp_fd, 1);
+     if (ret) return -1;
+ 
+-    memset(&qmp->addr, 0, sizeof (&qmp->addr));
++    memset(&qmp->addr, 0, sizeof (qmp->addr));
+     qmp->addr.sun_family = AF_UNIX;
+     strncpy(qmp->addr.sun_path, qmp_socket_path,
+             sizeof (qmp->addr.sun_path));
+diff -r 63594ce1708f -r 742dde457258 tools/python/xen/lowlevel/netlink/libnetlink.c
+--- a/tools/python/xen/lowlevel/netlink/libnetlink.c	Wed Feb 13 09:31:48 2013 +0100
++++ b/tools/python/xen/lowlevel/netlink/libnetlink.c	Wed Feb 13 17:00:15 2013 +0000
+@@ -37,7 +37,7 @@ int rtnl_open_byproto(struct rtnl_handle
+        int sndbuf = 32768;
+        int rcvbuf = 32768;
+ 
+-       memset(rth, 0, sizeof(rth));
++       memset(rth, 0, sizeof(*rth));
+ 
+        rth->fd = socket(AF_NETLINK, SOCK_RAW, protocol);
+        if (rth->fd < 0) {
+diff -r 63594ce1708f -r 742dde457258 tools/xenstat/libxenstat/src/xenstat_linux.c
+--- a/tools/xenstat/libxenstat/src/xenstat_linux.c	Wed Feb 13 09:31:48 2013 +0100
++++ b/tools/xenstat/libxenstat/src/xenstat_linux.c	Wed Feb 13 17:00:15 2013 +0000
+@@ -113,7 +113,7 @@ int parseNetDevLine(char *line, char *if
+ 
+ 	/* Initialize all variables called has passed as non-NULL to zeros */
+ 	if (iface != NULL)
+-		memset(iface, 0, sizeof(iface));
++		memset(iface, 0, sizeof(*iface));
+ 	if (rxBytes != NULL)
+ 		*rxBytes = 0;
+ 	if (rxPackets != NULL)
+
diff -Naur xen/tools/blktap2/drivers/md5.c xenb/tools/blktap2/drivers/md5.c
--- xen/tools/blktap2/drivers/md5.c	2013-03-09 10:39:52.266511944 -0700
+++ xenb/tools/blktap2/drivers/md5.c	2013-03-09 10:57:54.973178611 -0700
@@ -174,7 +174,7 @@
     MD5Transform(ctx->buf, (uint32_t *) ctx->in);
     byteReverse((unsigned char *) ctx->buf, 4);
     memcpy(digest, ctx->buf, 16);
-    memset(ctx, 0, sizeof(*ctx));     /* In case it's sensitive */
+    memset(ctx, 0, sizeof(ctx));     /* In case it's sensitive */
 }
 
 /* The four core functions - F1 is optimized somewhat */
diff -Naur xen/tools/debugger/kdd/kdd-xen.c xenb/tools/debugger/kdd/kdd-xen.c
--- xen/tools/debugger/kdd/kdd-xen.c	2013-03-09 10:39:52.286511944 -0700
+++ xenb/tools/debugger/kdd/kdd-xen.c	2013-03-09 10:57:54.973178611 -0700
@@ -333,7 +333,7 @@
     if (!cpu) 
         return -1;
 
-    memset(r, 0, sizeof(*r));
+    memset(r, 0, sizeof(r));
     
     if (w64)
         kdd_get_regs_x86_64(cpu, &r->r64);
diff -Naur xen/tools/libxc/xc_dom_boot.c xenb/tools/libxc/xc_dom_boot.c
--- xen/tools/libxc/xc_dom_boot.c	2013-03-09 10:39:52.366511945 -0700
+++ xenb/tools/libxc/xc_dom_boot.c	2013-03-09 10:57:54.973178611 -0700
@@ -266,7 +266,7 @@
         return rc;
 
     /* let the vm run */
-    memset(ctxt, 0, sizeof(*ctxt));
+    memset(ctxt, 0, sizeof(ctxt));
     if ( (rc = dom->arch_hooks->vcpu(dom, ctxt)) != 0 )
         return rc;
     xc_dom_unmap_all(dom);
diff -Naur xen/tools/libxl/libxl_qmp.c xenb/tools/libxl/libxl_qmp.c
--- xen/tools/libxl/libxl_qmp.c	2013-03-09 10:39:52.389845277 -0700
+++ xenb/tools/libxl/libxl_qmp.c	2013-03-09 10:57:54.976511944 -0700
@@ -378,7 +378,7 @@
     ret = libxl_fd_set_cloexec(qmp->ctx, qmp->qmp_fd, 1);
     if (ret) return -1;
 
-    memset(&qmp->addr, 0, sizeof (qmp->addr));
+    memset(&qmp->addr, 0, sizeof (&qmp->addr));
     qmp->addr.sun_family = AF_UNIX;
     strncpy(qmp->addr.sun_path, qmp_socket_path,
             sizeof (qmp->addr.sun_path));
diff -Naur xen/tools/python/xen/lowlevel/netlink/libnetlink.c xenb/tools/python/xen/lowlevel/netlink/libnetlink.c
--- xen/tools/python/xen/lowlevel/netlink/libnetlink.c	2013-03-09 10:39:52.443178611 -0700
+++ xenb/tools/python/xen/lowlevel/netlink/libnetlink.c	2013-03-09 10:57:54.976511944 -0700
@@ -37,7 +37,7 @@
        int sndbuf = 32768;
        int rcvbuf = 32768;
 
-       memset(rth, 0, sizeof(*rth));
+       memset(rth, 0, sizeof(rth));
 
        rth->fd = socket(AF_NETLINK, SOCK_RAW, protocol);
        if (rth->fd < 0) {
diff -Naur xen/tools/xenstat/libxenstat/src/xenstat_linux.c xenb/tools/xenstat/libxenstat/src/xenstat_linux.c
--- xen/tools/xenstat/libxenstat/src/xenstat_linux.c	2013-03-09 10:39:52.529845278 -0700
+++ xenb/tools/xenstat/libxenstat/src/xenstat_linux.c	2013-03-09 10:57:54.976511944 -0700
@@ -113,7 +113,7 @@
 
 	/* Initialize all variables called has passed as non-NULL to zeros */
 	if (iface != NULL)
-		memset(iface, 0, sizeof(*iface));
+		memset(iface, 0, sizeof(iface));
 	if (rxBytes != NULL)
 		*rxBytes = 0;
 	if (rxPackets != NULL)
