--- configure.old	2010-10-09 12:25:07.000000000 +0800
+++ configure	2010-10-09 12:44:07.000000000 +0800
@@ -197,7 +197,7 @@
  rm -f conf.*
  echo "YES"
  echo "HAVE_READLINE=-D HAVE_READLINE=1" >> Makefile
- echo "LINK_READLINE=-ltermcap -lreadline" >> Makefile
+ echo "LINK_READLINE=-lreadline" >> Makefile
 else
  rm -f conf.*
  echo "NO"
@@ -319,15 +319,6 @@
 # 19. CHECK TO SEE WHETHER THIS SYSTEM HAS KPATHSEA HEADERS
 
 echo $ECHO_N "Checking for libkpathsea-dev   ............. $ECHO_C"
-if printf '#include <stdlib.h>\n#include <stdio.h>\n#include <kpathsea/kpathsea.h>\nint main() { return 0; }\n' | gcc -x c - -o conf.out > conf.stdout 2> conf.stderr
-then
- rm -f conf.*
- echo "YES"
- echo "HAVE_KPATHSEA=-D HAVE_KPATHSEA=1" >> Makefile
- echo "LINK_KPATHSEA=-lkpathsea" >> Makefile
- echo "KPSE_COMMAND=/dev/null" >> Makefile
-else
- rm -f conf.*
  echo "NO"
  echo "HAVE_KPATHSEA=-D NOHAVE_KPATHSEA=1" >> Makefile
  echo "LINK_KPATHSEA=" >> Makefile
@@ -343,8 +334,37 @@
  else
   echo "NO" ; echo "ERROR: Neither the header files for the library 'libkpathsea-dev', nor the program kpsewhich could not be found." >&2 ; exit
  fi
+
+
+# +1 CHECK FOR PYTHON 2
+echo $ECHO_N "Checking for suitable Python   ............. $ECHO_C"
+whichout_python=`which python 2> conf.stderr`
+rm -f conf.*
+unset PY_MAJOR PY_MID PY_MIN
+eval `$whichout_python -V 2>&1 | awk '{printf("PY_MAJOR=%d\nPY_MID=%d\nPY_MIN=%d\n",substr($2,1,1),substr($2,3,1),substr($2,5,1));}'`
+if [ $PY_MAJOR -gt 2 ]
+then
+  whichout_python=`which python2 2> conf.stderr`
+  rm -f conf.*
+  unset PY_MAJOR PY_MID PY_MIN
+  eval `$whichout_python -V 2>&1 | awk '{printf("PY_MAJOR=%d\nPY_MID=%d\nPY_MIN=%d\n",substr($2,1,1),substr($2,3,1),substr($2,5,1));}'`
+  if [ $PY_MAJOR -eq 2 ]
+  then
+    echo "YES ($whichout_python)"
+    echo "PYTHON_COMMAND=${whichout_python}" >> Makefile
+    PYTHON_COMMAND="$whichout_python"
+  else
+    echo "NO"
+    echo "ERROR: Required python 2.x.x version could not be found." >&2
+    exit
+  fi
+else
+  echo "YES ($whichout_python)"
+  echo "PYTHON_COMMAND=${whichout_python}" >> Makefile
+  PYTHON_COMMAND="$whichout_python"
 fi
 
+
 # 20. OUTPUT MAKEFILE
 
 cat Makefile.skel >> Makefile
--- Makefile.skel.old	2010-10-09 12:46:17.000000000 +0800
+++ Makefile.skel	2010-10-09 12:48:34.000000000 +0800
@@ -94,7 +94,7 @@
 all: $(LOCAL_BINDIR)/pyxplot $(LOCAL_BINDIR)/pyxplot_watch $(LOCAL_BINDIR)/pyxplot_fitshelper $(LOCAL_BINDIR)/pyxplot_timehelper $(LOCAL_BINDIR)/debug/pyxplot $(LOCAL_BINDIR)/debug/pyxplot_watch $(LOCAL_BINDIR)/install/pyxplot $(LOCAL_BINDIR)/install/pyxplot_watch $(LOCAL_BINDIR)/install/pyxplot_fitshelper $(LOCAL_BINDIR)/install/pyxplot_timehelper pyxplot.1 pyxplot_watch.1
 
 src/EPSMaker/eps_colours.h: colourlist_generate.py
-	python colourlist_generate.py
+	$(PYTHON_COMMAND) colourlist_generate.py
 
 src/EPSMaker/eps_colours.c: src/EPSMaker/eps_colours.h
 
@@ -177,10 +177,10 @@
 #
 
 pyxplot.1: manpage_pyxplot.py
-	python manpage_pyxplot.py       ${DOCDIR}/pyxplot.pdf > pyxplot.1
+	$(PYTHON_COMMAND) manpage_pyxplot.py       ${DOCDIR}/pyxplot.pdf > pyxplot.1
 
 pyxplot_watch.1: manpage_pyxplot_watch.py
-	python manpage_pyxplot_watch.py ${DOCDIR}/pyxplot.pdf > pyxplot_watch.1
+	$(PYTHON_COMMAND) manpage_pyxplot_watch.py ${DOCDIR}/pyxplot.pdf > pyxplot_watch.1
 
 
 #
