--- Makefile.old	2012-08-30 11:07:45.515706893 +0800
+++ Makefile	2012-08-30 11:08:52.110530436 +0800
@@ -17,6 +17,7 @@
 HAVE_KPATHSEA=-D HAVE_KPATHSEA=1
 LINK_KPATHSEA=-lkpathsea
 KPSE_COMMAND=/dev/null
+PYTHON=python2
 # MAKEFILE.SKEL
 #
 # The code in this file is part of Pyxplot
@@ -113,12 +114,12 @@
 all: $(LOCAL_BINDIR)/pyxplot $(LOCAL_BINDIR)/pyxplot_watch $(LOCAL_BINDIR)/pyxplot_fitshelper $(LOCAL_BINDIR)/pyxplot_timehelper $(LOCAL_BINDIR)/debug/pyxplot $(LOCAL_BINDIR)/debug/pyxplot_watch $(LOCAL_BINDIR)/install/pyxplot $(LOCAL_BINDIR)/install/pyxplot_watch $(LOCAL_BINDIR)/install/pyxplot_fitshelper $(LOCAL_BINDIR)/install/pyxplot_timehelper $(LOCAL_OBJDIR)/pyxplot.1 $(LOCAL_OBJDIR)/pyxplot_watch.1
 
 src/settings/epsColors.h: buildScripts/colorlistGenerate.py
-	python buildScripts/colorlistGenerate.py
+	${PYTHON} buildScripts/colorlistGenerate.py
 
 src/settings/epsColors.c: src/settings/epsColors.h
 
 src/parser/cmdList.h: buildScripts/parser_data.py buildScripts/parser_data.dat
-	python buildScripts/parser_data.py
+	${PYTHON} buildScripts/parser_data.py
 
 src/parser/cmdList.c: src/parser/cmdList.h
 
@@ -202,11 +203,11 @@
 
 $(LOCAL_OBJDIR)/pyxplot.1: buildScripts/manpage_pyxplot.py
 	mkdir -p $(LOCAL_OBJDIR)
-	python buildScripts/manpage_pyxplot.py       ${DOCDIR}/pyxplot.pdf > $(LOCAL_OBJDIR)/pyxplot.1
+	${PYTHON} buildScripts/manpage_pyxplot.py       ${DOCDIR}/pyxplot.pdf > $(LOCAL_OBJDIR)/pyxplot.1
 
 $(LOCAL_OBJDIR)/pyxplot_watch.1: buildScripts/manpage_pyxplot_watch.py
 	mkdir -p $(LOCAL_OBJDIR)
-	python buildScripts/manpage_pyxplot_watch.py ${DOCDIR}/pyxplot.pdf > $(LOCAL_OBJDIR)/pyxplot_watch.1
+	${PYTHON} buildScripts/manpage_pyxplot_watch.py ${DOCDIR}/pyxplot.pdf > $(LOCAL_OBJDIR)/pyxplot_watch.1
 
 
 #
