--- src/archup.c	2011-03-18 14:08:02.306459380 -0300
+++ src/archup.c	2011-03-18 14:08:41.794459380 -0300
@@ -213,7 +213,7 @@
 			output_string[strlen(output_string)-1] = '\0';
 		}
 		/* Constructs the notification. */
-		my_notify = notify_notification_new("New updates for Archlinux available!",output_string,icon,NULL);
+		my_notify = notify_notification_new("New updates for Archlinux available!",output_string,icon);
 		/* Sets the timeout until the notification disappears. */
 		notify_notification_set_timeout(my_notify,timeout);
 		/* We set the category.*/
