diff -ru klatexformula-3.2.5/src/klflibview_p.h klatexformula-3.2.5-patch/src/klflibview_p.h
--- klatexformula-3.2.5/src/klflibview_p.h	2012-01-01 17:08:31.000000000 +0100
+++ klatexformula-3.2.5-patch/src/klflibview_p.h	2012-05-17 22:29:11.600455885 +0200
@@ -172,7 +172,7 @@
       }
       if (insertPos == QList<N>::size()) {
 	pContainsNonAllocated = false;
-	append(n);
+	this->append(n);
 	return insertPos;
       }
       QList<N>::operator[](insertPos) = n;
