diff -u -r 3.0-beta//iris/xmpp-im/xmpp_task.cpp 3.0-beta//iris/xmpp-im/xmpp_task.cpp
--- 3.0-beta//iris/xmpp-im/xmpp_task.cpp	2010-02-04 16:28:07.000000000 +0200
+++ 3.0-beta//iris/xmpp-im/xmpp_task.cpp	2010-03-08 10:38:40.618305274 +0200
@@ -20,6 +20,8 @@
 #include <QTimer>
 #include <qplatformdefs.h>
 
+#include <stdio.h>
+
 #include "safedelete.h"
 #include "xmpp_task.h"
 #include "xmpp_client.h"
@@ -237,7 +239,7 @@
 		buf = new char[size];
 		va_list ap;
 		va_start(ap, fmt);
-		r = QT_VSNPRINTF(buf, size, fmt, ap);
+		r = vsnprintf(buf, size, fmt, ap);
 		va_end(ap);
 
 		if(r != -1)
diff -u -r 3.0-beta//third-party/libjingle-0.4.0/talk/base/logging.cc 3.0-beta//third-party/libjingle-0.4.0/talk/base/logging.cc
--- 3.0-beta//third-party/libjingle-0.4.0/talk/base/logging.cc	2010-02-04 16:28:07.000000000 +0200
+++ 3.0-beta//third-party/libjingle-0.4.0/talk/base/logging.cc	2010-03-08 11:04:36.938149347 +0200
@@ -35,6 +35,8 @@
 #include <iostream>
 #include <iomanip>
 
+#include <stdio.h>
+
 #include "talk/base/common.h"
 #include "talk/base/logging.h"
 #include "talk/base/stream.h"
diff -u -r 3.0-beta//third-party/libjingle-0.4.0/talk/base/stream.h 3.0-beta//third-party/libjingle-0.4.0/talk/base/stream.h
--- 3.0-beta//third-party/libjingle-0.4.0/talk/base/stream.h	2010-02-04 16:28:07.000000000 +0200
+++ 3.0-beta//third-party/libjingle-0.4.0/talk/base/stream.h	2010-03-08 11:05:31.614812353 +0200
@@ -28,6 +28,8 @@
 #ifndef TALK_BASE_STREAM_H__
 #define TALK_BASE_STREAM_H__
 
+#include <stdio.h>
+
 #include "talk/base/basictypes.h"
 #include "talk/base/logging.h"
 #include "talk/base/scoped_ptr.h"
diff -u -r 3.0-beta//third-party/libjingle-0.4.0/talk/session/phone/phonesessionclient.h 3.0-beta//third-party/libjingle-0.4.0/talk/session/phone/phonesessionclient.h
--- 3.0-beta//third-party/libjingle-0.4.0/talk/session/phone/phonesessionclient.h	2010-02-04 16:28:07.000000000 +0200
+++ 3.0-beta//third-party/libjingle-0.4.0/talk/session/phone/phonesessionclient.h	2010-03-08 11:08:30.424791783 +0200
@@ -38,6 +38,7 @@
 #include "talk/p2p/base/sessionclient.h"
 #include "talk/p2p/base/sessiondescription.h"
 #include <map>
+#include <stdio.h>
 
 namespace cricket {
 
