From 3ba609d3154b3859724546038f4a5a5828a5d71b Mon Sep 17 00:00:00 2001
From: Rafael Reggiani Manzo <rr.manzo@gmail.com>
Date: Fri, 4 Jul 2014 13:00:55 -0300
Subject: [PATCH] Fixes keys on reference for ClangStaticAnalyzer

---
 lib/Analizo/Extractor/ClangStaticAnalyzer.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Analizo/Extractor/ClangStaticAnalyzer.pm b/lib/Analizo/Extractor/ClangStaticAnalyzer.pm
index cbb92fa..8c8c3bb 100644
--- a/lib/Analizo/Extractor/ClangStaticAnalyzer.pm
+++ b/lib/Analizo/Extractor/ClangStaticAnalyzer.pm
@@ -120,7 +120,7 @@ sub feed {
 
     $self->model->declare_module($module, $file_name);
 
-    foreach $bug (keys $bugs_hash) {
+    foreach $bug (keys %$bugs_hash) {
       my $value = $tree->{$file_name}->{$bug};
       $self->model->declare_security_metrics($bug, $module, $value);
     }
-- 
2.0.1

