diff -rupN mzx284c.orig/arch/unix/Makefile.in mzx284c/arch/unix/Makefile.in
--- mzx284c.orig/arch/unix/Makefile.in	2013-10-01 18:03:54.026450290 +0100
+++ mzx284c/arch/unix/Makefile.in	2013-10-01 18:05:20.595259874 +0100
@@ -1,4 +1,4 @@
-#
+
 # linux makefile generics
 #
 
@@ -42,14 +42,14 @@ ifeq (${BUILD_MODULAR},1)
 	 ${install} -m 0755 ${editor_target} ${DESTDIR}${LIBDIR}
 endif
 ifeq (${BUILD_MZXRUN},1)
-	@${install} -m 0755 -d ${DESTDIR}${GAMESDIR} && \
-	 ${install} -m 0755 ${mzxrun} ${DESTDIR}${GAMESDIR}/ && \
+	@${install} -m 0755 -d ${DESTDIR}${BINDIR} && \
+	 ${install} -m 0755 ${mzxrun} ${DESTDIR}${BINDIR}/ && \
 	 ${install} -m 0644 arch/unix/mzxrun.desktop \
 		${DESTDIR}${SHAREDIR}/applications/mzxrun.desktop
 endif
 ifeq (${BUILD_EDITOR},1)
-	@${install} -m 0755 -d ${DESTDIR}${GAMESDIR} && \
-	 ${install} -m 0755 ${mzx} ${DESTDIR}${GAMESDIR}/ && \
+	@${install} -m 0755 -d ${DESTDIR}${BINDIR} && \
+	 ${install} -m 0755 ${mzx} ${DESTDIR}${BINDIR}/ && \
 	 ${install} -m 0644 assets/ascii.chr assets/blank.chr \
 		assets/smzx.chr ${DESTDIR}${SHAREDIR}/megazeux/assets
 endif
