--- goo-0_155-any-dev/c/g2c/goo/boot.c	2003-11-19 16:49:58.000000000 -0300
+++ goo-0_155-any-dev-patched/c/g2c/goo/boot.c	2007-06-07 13:13:24.000000000 -0300
@@ -982,11 +982,13 @@
 
 /* FUNCTION CODES: */
 
+
+
 P YPallocate_stack(P num_) {
   P retF1;
   P T0,T1,T2,T3;
 DEFCREGS();
-  (P)YPdef_regs();
+  regs = REGSREF();
   T3 = (P)YPsp_reg();
   retF1 = T3;
   T1 = (P)YPsp_reg();
@@ -6761,7 +6763,7 @@
   P nF692;
   P T0,T1,T2,T3,T4,T5,T6,T7,T8,T9;
 DEFCREGS();
-  (P)YPdef_regs();
+  regs = REGSREF();
   T9 = (P)YPsp_elt((P)-2);
   nF692 = T9;
   T8 = (P)YPi_((P)-2,nF692);
@@ -6961,7 +6963,8 @@
   P T0,T1,T2,T3,T4,T5,T6,T7,T8,T9,T10,T11,T12,T13,T14,T15;
   P T16,T17,T18,T19,T20,T21,T22,T23,T24,T25,T26,T27,T28;
 DEFCREGS();
-  (P)YPset_regs(regs_);
+  // (P)YPset_regs(regs_);
+  regs = (REGS)(regs_);
   xF724 = fn_;
   T28 = (P)YPprop_elt(xF724,(P)1);
   xF725 = T28;
@@ -7061,7 +7064,8 @@
   P T0,T1,T2,T3,T4,T5,T6,T7,T8,T9,T10,T11,T12,T13,T14,T15;
   P T16,T17,T18,T19,T20,T21,T22,T23,T24,T25;
 DEFCREGS();
-  (P)YPset_regs(regs_);
+  //(P)YPset_regs(regs_);
+  regs = (REGS)(regs_);
   if (VARREF(YPclasses_readyQ) != YPfalse) {
     oF747 = res_;
     tF748 = type_;
@@ -7134,7 +7138,8 @@
 INLINE P YPstack_check(P regs_) {
   P T0,T1,T2,T3,T4,T5,T6,T7,T8,T9,T10,T11,T12,T13,T14,T15;
 DEFCREGS();
-  (P)YPset_regs(regs_);
+  //(P)YPset_regs(regs_);
+  regs = (REGS)(regs_);
   T4 = (P)YPmax_stack_len();
   T3 = (P)YPi_(T4,(P)1000);
   T6 = (P)YPsp_reg();
@@ -7202,7 +7207,8 @@
   P T16,T17,T18,T19,T20,T21,T22,T23,T24,T25,T26,T27,T28,T29,T30,T31;
   P T32,T33,T34,T35,T36,T37,T38,T39,T40,T41,T42,T43;
 DEFCREGS();
-  (P)YPset_regs(regs_);
+  //(P)YPset_regs(regs_);
+  regs = (REGS)(regs_);
   T43 = (P)YPsp_elt((P)-1);
   fnF767 = T43;
   xF768 = fnF767;
