diff -aur revit07-20070804.pristine/src/media_access/media_access_ewf.c revit07-20070804.new/src/media_access/media_access_ewf.c
--- revit07-20070804.pristine/src/media_access/media_access_ewf.c	2009-12-21 22:38:30.097739610 +0100
+++ revit07-20070804.new/src/media_access/media_access_ewf.c	2009-12-21 22:40:18.457741304 +0100
@@ -177,6 +177,7 @@
 off_t media_access_ewf_size( media_access_handle_t *handle )
 {
 	LIBEWF_HANDLE *ewf_handle = NULL;
+	size64_t      *media_size = NULL;
 
 	if( handle == NULL )
 	{
@@ -192,7 +193,7 @@
 	}
 	ewf_handle = (LIBEWF_HANDLE *) handle->handle;
 
-	return( (off_t) libewf_get_media_size( ewf_handle ) );
+	return( (off_t) libewf_get_media_size( ewf_handle, media_size ) );
 }
 
 #endif /* HAVE_LIBEWF */
