diff -Naur ./dockbarx_0.90.3.orig/GNOME_DockBarXApplet.server ./dockbarx_0.90.3/GNOME_DockBarXApplet.server
--- ./dockbarx_0.90.3.orig/GNOME_DockBarXApplet.server	2013-04-13 18:59:12.664371248 -0400
+++ ./dockbarx_0.90.3/GNOME_DockBarXApplet.server	1969-12-31 19:00:00.000000000 -0500
@@ -1,25 +0,0 @@
-<oaf_info>
-<oaf_server iid="OAFIID:GNOME_DockBarXApplet_Factory" type="exe" location="/usr/bin/dockbarx_factory">
-            
-    <oaf_attribute name="repo_ids" type="stringv">
-        <item value="IDL:Bonobo/GenericFactory:1.0"/>
-        <item value="IDL:Bonobo/Unknown:1.0"/>
-    </oaf_attribute>
-    <oaf_attribute name="name" type="string" value="DockBarX Applet"/>
-    <oaf_attribute name="description" type="string" value="DockBarX Applet"/>
-</oaf_server>
-                                                                                                                                                    
-<oaf_server iid="OAFIID:GNOME_DockBarXApplet"
-        type="factory" location="OAFIID:GNOME_DockBarXApplet_Factory">
-    <oaf_attribute name="repo_ids" type="stringv">
-        <item value="IDL:GNOME/Vertigo/PanelAppletShell:1.0"/>
-        <item value="IDL:Bonobo/Control:1.0"/>
-        <item value="IDL:Bonobo/Unknown:1.0"/>
-    </oaf_attribute>
-    <oaf_attribute name="name" type="string" value="DockBarX Applet"/>
-    <oaf_attribute name="description" type="string" value="DockBarX Applet"/>
-    <oaf_attribute name="panel:category" type="string" value="Utility"/>
-    <oaf_attribute name="panel:icon" type="string" value="dockbarx"/>
-</oaf_server>
-</oaf_info>
-                                                                                                                                                                                                                                                                                                                                                                        
diff -Naur ./dockbarx_0.90.3.orig/dockbarx/dockbar.py ./dockbarx_0.90.3/dockbarx/dockbar.py
--- ./dockbarx_0.90.3.orig/dockbarx/dockbar.py	2013-04-13 18:59:12.707682680 -0400
+++ ./dockbarx_0.90.3/dockbarx/dockbar.py	2013-04-13 19:49:47.669650988 -0400
@@ -143,14 +143,15 @@
 
         #--- Applet / Window container
         if self.applet is not None:
-            global gnomeapplet
-            import gnomeapplet
+            global mateapplet
+            import gi
+            from gi.repository import MatePanelApplet
             self.applet.set_applet_flags(
-                            gnomeapplet.HAS_HANDLE|gnomeapplet.EXPAND_MINOR)
-            orients = {gnomeapplet.ORIENT_DOWN: "down",
-                       gnomeapplet.ORIENT_UP: "up",
-                       gnomeapplet.ORIENT_LEFT: "left",
-                       gnomeapplet.ORIENT_RIGHT: "right"}
+                            mateapplet.HAS_HANDLE|mateapplet.EXPAND_MINOR)
+            orients = {mateapplet.ORIENT_DOWN: "down",
+                       mateapplet.ORIENT_UP: "up",
+                       mateapplet.ORIENT_LEFT: "left",
+                       mateapplet.ORIENT_RIGHT: "right"}
             self.orient = orients[self.applet.get_orient()]
             if self.orient in ("down", "up"):
                 self.container = gtk.HBox()
@@ -434,19 +435,19 @@
                         group.button.dockbar_moved()
 
     def __on_change_orient(self, arg1, data):
-        orients = {gnomeapplet.ORIENT_DOWN: "down",
-                   gnomeapplet.ORIENT_UP: "up",
-                   gnomeapplet.ORIENT_LEFT: "left",
-                   gnomeapplet.ORIENT_RIGHT: "right"}
+        orients = {mateapplet.ORIENT_DOWN: "down",
+                   mateapplet.ORIENT_UP: "up",
+                   mateapplet.ORIENT_LEFT: "left",
+                   mateapplet.ORIENT_RIGHT: "right"}
         self.orient = orients[self.applet.get_orient()]
 
     def __on_change_background(self, applet, type, color, pixmap):
         applet.set_style(None)
         rc_style = gtk.RcStyle()
         applet.modify_style(rc_style)
-        if type == gnomeapplet.COLOR_BACKGROUND:
+        if type == mateapplet.COLOR_BACKGROUND:
             applet.modify_bg(gtk.STATE_NORMAL, color)
-        elif type == gnomeapplet.PIXMAP_BACKGROUND:
+        elif type == mateapplet.PIXMAP_BACKGROUND:
             style = applet.style
             style.bg_pixmap[gtk.STATE_NORMAL] = pixmap
             applet.set_style(style)
@@ -980,7 +981,7 @@
                 os.system("cp %s %s"%(path, new_path))
         else:
             new_path = os.path.join(launcher_dir, "%s.desktop"%identifier)
-        process = subprocess.Popen(["gnome-desktop-item-edit", new_path],
+        process = subprocess.Popen(["mate-desktop-item-edit", new_path],
                                    env=os.environ)
         gobject.timeout_add(100, self.__wait_for_launcher_editor,
                             process, path, new_path, identifier)
diff -Naur ./dockbarx_0.90.3.orig/dockbarx_factory ./dockbarx_0.90.3/dockbarx_factory
--- ./dockbarx_0.90.3.orig/dockbarx_factory	2013-04-13 18:59:12.707682680 -0400
+++ ./dockbarx_0.90.3/dockbarx_factory	2013-04-13 19:37:44.905133498 -0400
@@ -27,7 +27,11 @@
 import pygtk
 pygtk.require("2.0")
 import gtk
-import gnomeapplet
+import gi
+#gi.require_version("Gtk", "2.0")
+#from gi.repository import Gtk
+
+from gi.repository import MatePanelApplet
 import dockbarx.dockbar
 
 class DockBarWindow():
@@ -65,6 +69,6 @@
     dockbarwin = DockBarWindow()
     dockbarwin.main()
 else:
-    gnomeapplet.bonobo_factory("OAFIID:GNOME_DockBarXApplet_Factory",
-                                     gnomeapplet.Applet.__gtype__,
-                                     "dockbar applet", "0", dockbar_factory)
\ No newline at end of file
+    MatePanelApplet.Applet.factory_main("DockBarXAppletFactory", True,
+                                    MatePanelApplet.Applet.__gtype__,
+                                    applet_factory, None)
diff -Naur ./dockbarx_0.90.3.orig/org.mate.panel.DockBarXApplet.mate-panel-applet ./dockbarx_0.90.3/org.mate.panel.DockBarXApplet.mate-panel-applet
--- ./dockbarx_0.90.3.orig/org.mate.panel.DockBarXApplet.mate-panel-applet	1969-12-31 19:00:00.000000000 -0500
+++ ./dockbarx_0.90.3/org.mate.panel.DockBarXApplet.mate-panel-applet	2013-04-13 19:19:24.411623366 -0400
@@ -0,0 +1,13 @@
+[Applet Factory]
+Id=DockBarXAppletFactory
+InProcess=false
+Location=/usr/bin/dockbarx_factory
+Name=DockBarX Applet Factory
+Description=Advanced Taskbar
+ 
+[DockBarXApplet]
+Name=dockBarX
+Description=Advanced Taskbar
+Icon=/usr/share/pixmaps/dockbarx.png
+MateComponentId=OAFIID:MATE_DockBarXApplet;
+
diff -Naur ./dockbarx_0.90.3.orig/org.mate.panel.applet.DockBarXAppletFactory.service ./dockbarx_0.90.3/org.mate.panel.applet.DockBarXAppletFactory.service
--- ./dockbarx_0.90.3.orig/org.mate.panel.applet.DockBarXAppletFactory.service	1969-12-31 19:00:00.000000000 -0500
+++ ./dockbarx_0.90.3/org.mate.panel.applet.DockBarXAppletFactory.service	2013-04-13 19:21:37.097860850 -0400
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=org.mate.panel.applet.DockBarXAppletFactory
+Exec=/usr/bin/dockbarx_factory
diff -Naur ./dockbarx_0.90.3.orig/setup.py ./dockbarx_0.90.3/setup.py
--- ./dockbarx_0.90.3.orig/setup.py	2013-04-13 18:59:12.711014329 -0400
+++ ./dockbarx_0.90.3/setup.py	2013-04-13 19:29:59.800324448 -0400
@@ -131,7 +131,8 @@
                                              "dockx_applets/namebar_window_title.applet",
                                              "dockx_applets/namebar_window_title.py"]),
             ("/usr/bin", ["dockbarx_factory", "dbx_preference", "dockx"]),
-            ("/usr/lib/bonobo/servers", ["GNOME_DockBarXApplet.server"]),
+            ("/usr/share/dbus-1/services", ["org.mate.panel.applet.DockBarXAppletFactory.service"]),
+            ("/usr/share/mate-panel/applets", ["org.mate.panel.DockBarXApplet.mate-panel-applet"]),
             ("/usr/share/applications/", ["dbx_preference.desktop"]),
             ("/usr/share/applications/", ["DockX.desktop"]),
          ]
