--- ../dummy/src/music.c	2011-11-23 21:00:22.255164854 -0200
+++ src/music.c	2011-11-23 21:01:06.274514383 -0200
@@ -275,7 +275,7 @@
 		if (sound->mutex == NULL) luaL_error(L, "out of memory");
 		sound->cond = SDL_CreateCond();
 		if (sound->cond == NULL) luaL_error(L, "out of memory");
-		sound->loaderThread = SDL_CreateThread(streamingLoader, sound);
+		sound->loaderThread = SDL_CreateThread(streamingLoader, "t-sound", sound);
 	}
 	else {
 		sound->static_source = 0;
