diff -ur vpython-wx-6.05-orig/setup.py vpython-wx-6.05/setup.py
--- vpython-wx-6.05-orig/setup.py	2014-01-31 15:04:17.000000000 +0100
+++ vpython-wx-6.05/setup.py	2014-01-31 15:04:59.000000000 +0100
@@ -75,7 +75,7 @@
     LINK_FLAGS="-Wl,--export-dynamic" 
      
     GTK_VIS_LIBS = get_libs()
-    GTK_VIS_LIBS.append('boost_python-mt-py' + versionString) 
+    GTK_VIS_LIBS.append('boost_python') 
     GTK_VIS_LIBS.append('boost_signals') 
      
     GTK_INCDIRS = get_includes() 
