--- setup.py.orig	2014-10-27 19:38:07.000000000 +0100
+++ setup.py	2014-10-27 19:38:28.000000000 +0100
@@ -1,14 +1,5 @@
 from __future__ import print_function
 
-try:
-    from distribute_setup import use_setuptools
-    use_setuptools()
-except ImportError:
-    #
-    # attempt to use the updated distribute version
-    #
-    pass
-
 import sys
 import os
 from glob import glob
@@ -80,7 +71,7 @@
     LINK_FLAGS="-Wl,--export-dynamic"
 
     GTK_VIS_LIBS = get_libs()
-    GTK_VIS_LIBS.append('boost_python-mt-py' + versionString)
+    GTK_VIS_LIBS.append('boost_python')
     GTK_VIS_LIBS.append('boost_signals')
 
     GTK_INCDIRS = get_includes()
