Index: evas/evas.c_evas_object.pxi
===================================================================
--- evas/evas.c_evas_object.pxi	(revision 55980)
+++ evas/evas.c_evas_object.pxi	(working copy)
@@ -217,7 +217,7 @@
        anti_alias_set, anti_alias_get, anti_alias, pointer_mode_set,
        pointer_mode_get, pointer_mode
     """
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self.obj = NULL
         self.evas = None
         self.data = dict()
Index: evas/evas.c_evas_canvas.pxi
===================================================================
--- evas/evas.c_evas_canvas.pxi	(revision 55980)
+++ evas/evas.c_evas_canvas.pxi	(working copy)
@@ -117,7 +117,7 @@
            obscured_clear, render_updates, render, norender, freeze_get,
            output_method_set, output_method_get, output_method
     """
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self.obj = NULL
         self._callbacks = [None] * evas_canvas_event_callbacks_len
 
Index: evas/evas.c_evas_object_smart.pxi
===================================================================
--- evas/evas.c_evas_object_smart.pxi	(revision 55980)
+++ evas/evas.c_evas_object_smart.pxi	(working copy)
@@ -359,7 +359,7 @@
 
     @see: L{ClippedSmartObject}
     """
-    def __new__(self, *a, **ka):
+    def __cinit__(self, *a, **ka):
         self._smart_callbacks = dict()
         cls = self.__class__
         self._m_delete = _smart_class_get_impl_method(cls, "delete")
