--- airoscript.conf.orig	2009-06-24 23:22:36.000000000 +0200
+++ airoscript.conf	2009-06-24 23:24:04.000000000 +0200
@@ -41,7 +41,7 @@
 	#export resonset="1" # this is also necesary
 
 #Your dhcp client utility
-	DHCPSOFT="dhclient"
+	DHCPSOFT="dhcpcd"
 
 # Debug:
 	# Uncomment next line to make airoscript not launch terminals, just print the command to be executed.
@@ -61,7 +61,7 @@
 	# For example, in systems where you have compiled aircrack-ng 
 	# or in slax it'll be /usr/local/bin or BINDIR and SBINDIR
 	# And, on some systems, mdk3 will be installed on /usr/local/bin/
-	PREFIX="/usr/local"
+	PREFIX="/usr"
 
 	BINDIR="$PREFIX/bin/"
 	SBINDIR="$PREFIX/sbin/"                     
@@ -77,7 +77,7 @@
 	MACCHANGER=$BINDIR"macchanger"
 
 	# Fix to be able to use screenrc defined screenrc's
-	[[ "$SCREENRC" == "" ]] && SCREENRC="$PREFIX/share/airoscript/screenrc"
+	[[ "$SCREENRC" == "" ]] && SCREENRC="/etc/airoscript-screenrc"
 	
 	FUNCTIONS="$PREFIX/share/airoscript/airoscfunc.sh"
 	SCREEN_FUNCTIONS="$PREFIX/share/airoscript/airoscfunc_screen.sh"
@@ -90,7 +90,7 @@
 	iwconfig="iwconfig"
 
 # Path to optional binaries
-	MDK3="/usr/local/sbin/mdk3"
+	MDK3="/usr/sbin/mdk3"
 
 	# As i know, those are just used for spanish routers...
 	JTD="jazzteldecrypter"
