Index: vdr-1.6.0/config.h
===================================================================
--- vdr-1.6.0.orig/config.h
+++ vdr-1.6.0/config.h
@@ -117,7 +117,7 @@ public:
                 if (!isempty(s)) {
                    T *l = new T;
                    if (l->Parse(s))
-                      Add(l);
+                      this->Add(l);
                    else {
                       esyslog("ERROR: error in %s, line %d", fileName, line);
                       delete l;

