diff --git a/tools/seafile-admin b/tools/seafile-admin
index 102cf9f..e2a0cca 100755
--- a/tools/seafile-admin
+++ b/tools/seafile-admin
@@ -54,7 +54,6 @@ CONF_SEAFILE_DIR = 'seafile_dir'
 CONF_SEAHUB_DIR = 'seafile_dir'
 CONF_CCNET_PORT = 'ccnet_port'
 CONF_SEAFILE_PORT = 'seafile_port'
-CONF_FILESERVER_PORT = 'fileserver_port'
 CONF_IP_OR_DOMAIN = 'ip_or_domain'
 
 CONF_SEAHUB_CONF = 'seahub_conf'
@@ -237,7 +236,6 @@ def check_seafile_install():
         'seaf-server-init',
         'seaf-server',
         'ccnet-server',
-        'fileserver',
         'seafile-controller',
     ]
 
@@ -359,22 +357,6 @@ def get_seafile_port():
                                            default=default,
                                            validate=validate)
 
-def get_fileserver_port():
-    def validate(s):
-        try:
-            port = int(s)
-        except ValueError:
-            return False
-
-        return port > 0 and port < 65536
-
-    question = 'Which port do you want to use for the seafile fileserver?'
-    key = 'seafile fileserver port'
-    default = '8082'
-    conf[CONF_FILESERVER_PORT] = ask_question(question,
-                                              key=key,
-                                              default=default,
-                                              validate=validate)
 def get_seafile_data_dir():
     question = 'Where do you want to put your seafile data?'
     key = 'seafile-data'
@@ -521,7 +503,6 @@ def config_ccnet_seafile():
     get_seafile_data_dir()
     if not conf[CONF_SEAFILE_CONF_EXISTS]:
         get_seafile_port()
-        get_fileserver_port()
 
     info('This is your configuration')
     info('------------------------------------------')
@@ -538,7 +519,6 @@ def config_ccnet_seafile():
     else:
         info('seafile data dir:         %s' % highlight(conf[CONF_SEAFILE_DIR]))
         info('seafile port:             %s' % highlight(conf[CONF_SEAFILE_PORT]))
-        info('seafile fileserver port:  %s' % highlight(conf[CONF_FILESERVER_PORT]))
 
     info('------------------------------------------')
     info('Press ENTER if the config is right, or anything else to re-config ')
@@ -570,7 +550,6 @@ def init_ccnet_seafile():
             'seaf-server-init',
             '--seafile-dir', conf[CONF_SEAFILE_DIR],
             '--port', conf[CONF_SEAFILE_PORT],
-            '--fileserver-port', conf[CONF_FILESERVER_PORT],
         ]
 
         if run_argv(argv) != 0:
@@ -767,7 +746,7 @@ def check_necessary_files():
             error('%s not found' % fpath)
 
 def start_seafile(args):
-    '''start ccnet/seafile/seahub/fileserver'''
+    '''start ccnet/seafile/seahub'''
     if is_running('seafile-controller'):
         error(highlight('NOTE: Seafile is already running'))
 
