--- a/tools/Makefile.am	2013-08-20 16:08:58.239195887 -0400
+++ b/tools/Makefile.am	2013-08-20 16:09:21.402115531 -0400
@@ -5,7 +5,7 @@
 
 seaf_server_init_SOURCES = seaf-server-init.c ../common/seaf-db.c
 
-seaf_server_init_LDADD = @GLIB2_LIBS@ @ZDB_LIBS@
+seaf_server_init_LDADD = -lpthread @GLIB2_LIBS@ @ZDB_LIBS@
 seaf_server_init_LDFLAGS = @STATIC_COMPILE@ @SERVER_PKG_RPATH@
 
 seaf_server_init_CPPFLAGS = @GLIB2_CFLAGS@ @ZDB_CFLAGS@ @MSVC_CFLAGS@
