diff --git a/dwarftherapist.pro b/dwarftherapist.pro
index befc9e2bf15d..7718cd931083 100644
--- a/dwarftherapist.pro
+++ b/dwarftherapist.pro
@@ -89,8 +89,8 @@ else:unix {
     target.path = /usr/bin
     INSTALLS += target
 
-    bin.path = /usr/bin; cp dist/dwarftherapist $(INSTALL_ROOT)/usr/bin/dwarftherapist
-    bin.files += bin/release/DwarfTherapist; chmod +x $(INSTALL_ROOT)/usr/bin/dwarftherapist
+    bin.path = /usr/bin
+    bin.files += bin/release/DwarfTherapist
     INSTALLS += bin
 
     application.path = /usr/share/applications
@@ -107,11 +107,11 @@ else:unix {
     INSTALLS += icon
 
     share.path = /usr/share/dwarftherapist
-    share.extra = mkdir -p $(INSTALL_ROOT)/usr/share/dwarftherapist/etc; \
-        mkdir -p $(INSTALL_ROOT)/usr/share/dwarftherapist/etc/memory_layouts; \
-        mkdir -p $(INSTALL_ROOT)/usr/share/dwarftherapist/etc/memory_layouts/linux; \
-        cp etc/game_data.ini $(INSTALL_ROOT)/usr/share/dwarftherapist/etc; \
-        cp etc/memory_layouts/linux/* $(INSTALL_ROOT)/usr/share/dwarftherapist/etc/memory_layouts/linux
+    share.extra = mkdir -p $(INSTALL_ROOT)/usr/share/dwarftherapist/; \
+	mkdir -p $(INSTALL_ROOT)/usr/share/dwarftherapist/memory_layouts; \
+	mkdir -p $(INSTALL_ROOT)/usr/share/dwarftherapist/memory_layouts/linux; \
+	cp etc/game_data.ini $(INSTALL_ROOT)/usr/share/dwarftherapist; \
+	cp etc/memory_layouts/linux/* $(INSTALL_ROOT)/usr/share/dwarftherapist/memory_layouts/linux
 
     INSTALLS += share
 }
diff --git a/src/dfinstance.cpp b/src/dfinstance.cpp
index a89972808e3b..a1113fb419c9 100644
--- a/src/dfinstance.cpp
+++ b/src/dfinstance.cpp
@@ -102,7 +102,7 @@ DFInstance::DFInstance(QObject* parent)
     search_paths << working_dir.path();
 
     QString subdir = LAYOUT_SUBDIR;
-    search_paths << QString("etc/memory_layouts/%1").arg(subdir);
+    search_paths << QString("/usr/share/dwarftherapist/memory_layouts/%1").arg(subdir);
 
     TRACE << "Searching for MemoryLayout ini files in the following directories";
     foreach(QString path, search_paths) {
diff --git a/src/dwarftherapist.cpp b/src/dwarftherapist.cpp
index 23129a7c906a..494bcb5497a6 100644
--- a/src/dwarftherapist.cpp
+++ b/src/dwarftherapist.cpp
@@ -128,7 +128,7 @@ void DwarfTherapist::setup_logging() {
 
     //setup logging
     m_log_mgr = new LogManager(this);
-    TruncatingFileLogger *log = m_log_mgr->add_logger("log/run.log");
+    TruncatingFileLogger *log = m_log_mgr->add_logger(QDir::tempPath() + "/dwarftherapist.log");
     if (log) {
         LogAppender *app = m_log_mgr->add_appender("core", log, LL_TRACE);
         if (app) {
diff --git a/src/gamedatareader.cpp b/src/gamedatareader.cpp
index 74561e7e6c16..a328ba6e44c9 100644
--- a/src/gamedatareader.cpp
+++ b/src/gamedatareader.cpp
@@ -44,7 +44,7 @@ GameDataReader::GameDataReader(QObject *parent)
     : QObject(parent)
 {
     QDir working_dir = QDir::current();
-    QString filename = working_dir.absoluteFilePath("etc/game_data.ini");
+    QString filename = "/usr/share/dwarftherapist/game_data.ini";
     m_data_settings = new QSettings(filename, QSettings::IniFormat);
 
     build_calendar();
