--- usr/local/Brother/Printer/HL2230/lpd/filterHL2230	2013-08-30 22:51:27.630629862 -0400
+++ usr/local/Brother/Printer/HL2230/lpd/filterHL2230.patched	2013-08-30 22:44:37.597287533 -0400
@@ -22,7 +22,7 @@
 PRINTER_TMP=$(echo $PRINTER | sed -e 's/ //g' -e 's/-//g')
 PCL=1
 
-BR_PRT_PATH=/usr/local/Brother/Printer/$PRINTER_TMP
+BR_PRT_PATH=/usr/share/brother
 RCFILE=`eval echo $BR_PRT_PATH/inf/brPRINTERrc | eval sed 's/PRINTER/"$PRINTER_TMP"/'`
 PAPER_INF=$BR_PRT_PATH/inf/paperinf
 PSCONV=$BR_PRT_PATH/lpd/psconvert2
@@ -117,14 +117,7 @@
 	cp $INPUT_TEMP1 $INPUT_TEMP
 fi
 	
-#PSTOPSFILTER=`which pstops`
-
-if [ -e '/usr/bin/pstops' ];then
-	PSTOPSFILTER='/usr/bin/pstops'
-else
-	PSTOPSFILTER=''
-fi
-
+PSTOPSFILTER=/usr/lib/cups/filter/pstops
 
 trap '' 2
 if [ $postfilter = cat ];then
--- usr/local/Brother/Printer/HL2230/cupswrapper/cupswrapperHL2230-2.0.4	2013-08-30 22:51:27.630629862 -0400
+++ usr/local/Brother/Printer/HL2230/cupswrapper/cupswrapperHL2230-2.0.4.patched	2013-08-30 22:44:01.713953409 -0400
@@ -18,6 +18,7 @@
 # Place, Suite 330, Boston, MA  02111-1307  USA
 #
 
+if /bin/false; then
 if [ "$1" = '-e' ]; then
   lpadmin -x HL2230
   rm -f /usr/share/cups/model/HL2230.ppd
@@ -65,7 +66,8 @@
    ppd_file_name=/usr/share/ppd/HL2230.ppd
 fi
 
-cat <<ENDOFPPDFILE >$ppd_file_name
+fi
+cat <<ENDOFPPDFILE > ppd_file
 *PPD-Adobe: "4.3"
 *%================================================
 *%	Copyright Brother Industries,Ltd 2006-2008
@@ -334,7 +336,7 @@
 
 
 ENDOFPPDFILE
-
+if /bin/false; then
 chmod 755  $ppd_file_name
 if [ -e '/usr/share/ppd' ];then
   if [ -e '/usr/share/cups/model' ];then
@@ -351,11 +353,11 @@
   fi
 fi
 
-
-cat <<!ENDOFWFILTER! >$brotherlpdwrapper
+fi
+cat <<!ENDOFWFILTER! > wrapper
 #! /bin/sh
 #
-# Brother Print filter  >>  $brotherlpdwrapper
+# Brother Print filter
 # Copyright (C) 2005 Brother. Industries, Ltd.
 #                                    Ver1.00
 
@@ -481,25 +483,25 @@
 fi
 
 
-if [ -e "/usr/local/Brother/Printer/\$PRINTER/lpd/filter\$PRINTER" ]; then
+if [ -e "/usr/share/brother/lpd/filter\$PRINTER" ]; then
        :
 else
-    echo "ERROR: /usr/local/Brother/Printer/\$PRINTER/lpd/filter\$PRINTER does not exist"   >>\$LOGFILE
+    echo "ERROR: /usr/share/brother/lpd/filter\$PRINTER does not exist"   >>\$LOGFILE
     errorcode=30
     exit $errorcode
 fi
 
-if [ -e "/usr/local/Brother/Printer/HL2230/cupswrapper/brcupsconfig4" ]; then
+if [ -e "/usr/share/brother/cupswrapper/brcupsconfig4" ]; then
   if [ \$DEBUG = 0 ]; then
-     /usr/local/Brother/Printer/HL2230/cupswrapper/brcupsconfig4  \$PRINTER  \$PPD 0 "\$options" >> /dev/null
+     /usr/share/brothercupswrapper/brcupsconfig4  \$PRINTER  \$PPD 0 "\$options" >> /dev/null
   else
-     /usr/local/Brother/Printer/HL2230/cupswrapper/brcupsconfig4  \$PRINTER  \$PPD \$LOGCLEVEL "\$options" >>\$LOGFILE
+     /usr/share/cupswrapper/brcupsconfig4  \$PRINTER  \$PPD \$LOGCLEVEL "\$options" >>\$LOGFILE
   fi
 fi
 
 
 if [ \$DEBUG -le 2 ]; then
-    cat    \$INPUT_TEMP_PS | /usr/local/Brother/Printer/\$PRINTER/lpd/filter\$PRINTER
+    cat    \$INPUT_TEMP_PS | /usr/share/brother/lpd/filter\$PRINTER
 fi
 
 if [ \$DEBUG -ge 2 ];  then
@@ -515,7 +517,7 @@
 
 !ENDOFWFILTER!
 
-
+if /bin/false; then
 chmod 755 $brotherlpdwrapper
 if [ -e /usr/lib/cups/filter ]; then
  if [ -e /usr/lib64/cups/filter ]; then
@@ -551,3 +553,4 @@
 fi
 lpadmin -p HL2230 -E -v $port -P $ppd_file_name
   exit 0 
+fi
--- usr/local/Brother/Printer/HL2230/inf/brHL2230func	2013-08-30 22:51:27.633963195 -0400
+++ usr/local/Brother/Printer/HL2230/inf/brHL2230func.patched	2013-08-30 22:44:01.713953409 -0400
@@ -24,4 +24,5 @@
 Sleep={PrinterDefault,"1-99"}
 Toner Save Mode={ON,OFF}
 
-
+[psconvert2]
+pstops=/usr/lib/cups/filter/pstops
