diff --git a/include/mr_rqt/MrRqt.h b/include/mr_rqt/MrRqt.h
index 0d5b45d..354723f 100644
--- a/include/mr_rqt/MrRqt.h
+++ b/include/mr_rqt/MrRqt.h
@@ -32,10 +32,12 @@
 #include <iostream>
 #include <set>
 
-#include <boost/foreach.hpp>
-#include <boost/date_time.hpp>
-#include <boost/thread.hpp>
-#include <boost/algorithm/string.hpp>
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
+# include <boost/foreach.hpp>
+# include <boost/date_time.hpp>
+# include <boost/thread.hpp>
+# include <boost/algorithm/string.hpp>
+#endif
 
 #include <ros/ros.h>
 #include <std_msgs/String.h>
