diff -rauN asus_acpi/asus_acpi.c asus_acpi_backlight_patch/asus_acpi.c
--- asus_acpi/asus_acpi.c	2010-09-09 18:11:45.000000000 +0200
+++ asus_acpi_backlight_patch/asus_acpi.c	2010-09-09 18:12:23.000000000 +0200
@@ -1495,6 +1495,7 @@
 static int __init asus_acpi_init(void)
 {
 	int result;
+	struct backlight_properties props;
 
 	if (acpi_disabled)
 		return -ENODEV;
@@ -1526,8 +1527,11 @@
 		return result;
 	}
 
+	memset(&props, 0, sizeof(struct backlight_properties));
+	props.max_brightness = 20 - 1;
+
 	asus_backlight_device = backlight_device_register("asus",NULL,NULL,
-							  &asus_backlight_data);
+							  &asus_backlight_data,&props);
         if (IS_ERR(asus_backlight_device)) {
 		printk(KERN_ERR "Could not register asus backlight device\n");
 		asus_backlight_device = NULL;
