--- lcnb_acpi-0.10-1mdv2008.1/Makefile	2009-02-19 20:19:58.000000000 +0100
+++ Makefile	2009-02-19 19:25:05.000000000 +0100
@@ -2,6 +2,8 @@
 # Makefile for the Linux ACPI interpreter
 #
 
+
+
 ACPI_CFLAGS	:= -Os
 
 ifdef CONFIG_ACPI_DEBUG
@@ -11,3 +13,9 @@
 EXTRA_CFLAGS	+= $(ACPI_CFLAGS)
 
 obj-m				+= lcnb_acpi.o
+
+KVERSION = $(shell uname -r)
+all:
+	        make -C /lib/modules/$(KVERSION)/build M=$(PWD) modules
+clean:
+	        make -C /lib/modules/$(KVERSION)/build M=$(PWD) clean
