--- asciijump-1.0.2~beta.orig/asciijump.desktop	2010-08-27 10:12:17.000000000 +0600
+++ asciijump-1.0.2~beta/asciijump.desktop	2012-02-29 10:02:15.034987892 +0600
@@ -2,10 +2,10 @@
 Name=asciijump
 Comment=A clone of the dsj game
 Comment[pl]=Klon gry dsj
-Icon=asciijump.xpm
+Icon=asciijump.png
 Exec=asciijump
 Terminal=true
 Type=Application
-Categories=GNOME;Application;Game;BoardGame;
+Categories=Application;Game;SportsGame;
 StartupNotify=true
 Encoding=UTF-8
--- asciijump-1.0.2~beta.orig/Makefile.in	2010-08-27 10:12:17.000000000 +0600
+++ asciijump-1.0.2~beta/Makefile.in	2012-02-23 16:49:29.989464305 +0600
@@ -51,7 +51,7 @@
 
 INCLUDES	= -I. -I${top_srcdir}/include -I${top_builddir}
 DEFINES		= -DVERSION=\"${VERSION}\" -DPACKAGE=\"${PACKAGE}\" \
-		-DDATADIR=\"${datadir}/${PACKAGE}\" -DVARDIR=\"${vardir}/games/asciijump/\"
+		-DDATADIR=\"${datadir}/${PACKAGE}\" -DVARDIR=\"${vardir}/lib/asciijump/\"
 ALL_CFLAGS	= ${INCLUDES} ${DEFINES} ${CFLAGS} -g
 
 LIBS		= @LIBS@
@@ -142,8 +142,8 @@
 	${INSTALL} -m 644 asciijump.desktop \
 		$(DESTDIR)$(prefix)/share/applications
 		
-	${INSTALL} -d $(DESTDIR)$(prefix)/X11R6/share/pixmaps
-	${INSTALL} -m 644 asciijump.png $(DESTDIR)$(prefix)/X11R6/share/pixmaps
+#	${INSTALL} -d $(DESTDIR)$(prefix)/X11R6/share/pixmaps
+#	${INSTALL} -m 644 asciijump.png $(DESTDIR)$(prefix)/X11R6/share/pixmaps
 		
 	${INSTALL} -d $(DESTDIR)$(prefix)/share/pixmaps
 	${INSTALL} -m 644 asciijump.png $(DESTDIR)$(prefix)/share/pixmaps
@@ -151,8 +151,8 @@
 	${INSTALL} -d $(DESTDIR)$/$(mandir)/man6
 	${INSTALL} -m 644 asciijump.6 $(DESTDIR)$/$(mandir)/man6
 
-	${INSTALL} -d $(DESTDIR)$/$(vardir)/games
-	${INSTALL} -d -m 777 $(DESTDIR)$/$(vardir)/games/asciijump
+	${INSTALL} -d $(DESTDIR)$/$(vardir)/lib
+	${INSTALL} -d -m 775 -g games $(DESTDIR)$/$(vardir)/lib/asciijump
 
 install-strip:
 	${MAKE} INSTALL_PROGRAM='${INSTALL_PROGRAM} -s' install
