--- src/bjfilter.c
+++ src/bjfilter.org.c
@@ -91,7 +91,7 @@
 
 #if DEBUGLOG
 
-#define DEBUGLOGFILE "/var/tmp/bjlogXXXXXX"
+#define DEBUGLOGFILE "/tmp/bjlogXXXXXX"
 
 char		log_path_name[256];
 int			log_fd;
@@ -1748,7 +1748,7 @@
     int fd;
 
 	strncpy(log_path_name, DEBUGLOGFILE, sizeof(log_path_name));
-	log_path_name[sizeof(log_path_name)-1] = '\0'
+	log_path_name[sizeof(log_path_name)-1] = '\0';
 	fd = mkstemp(log_path_name);
 	if (fd < 0) return;
 	close(fd);
--- src/Makefile.in
+++ src/Makefile.org.in
@@ -261,27 +261,6 @@
 	    || cp -p $$d/$$file $(distdir)/$$file || :; \
 	  fi; \
 	done
-bjferror.o: bjferror.c bjferror.h
-bjfilter.o: bjfilter.c ../include/cncl/cncl.h ../include/cncl/cpktypes.h \
-	../include/cncl/cncldef.h ../include/cncl/cnclucom.h \
-	../include/cncl/cncludef.h bjfimage.h bjfoption.h bjfrcaccess.h \
-	bjfpos.h bjipc.h bjfilter.h uitypes.h bjflist.h bjfpath.h \
-	../config.h
-bjfimage.o: bjfimage.c ../include/cncl/cpktypes.h ../include/cncl/cncl.h \
-	../include/cncl/cncldef.h bjfimage.h bjferror.h
-bjflist.o: bjflist.c bjflist.h
-bjfoption.o: bjfoption.c ../include/cncl/cncl.h \
-	../include/cncl/cpktypes.h ../include/cncl/cncldef.h \
-	../include/cncl/cnclucom.h ../include/cncl/cncludef.h \
-	bjfoption.h bjfrcaccess.h uitypes.h bjipc.h \
-	../include/misc/psize.h bjfversion.h bjfpath.h ../config.h
-bjfpos.o: bjfpos.c ../include/cncl/cncl.h ../include/cncl/cpktypes.h \
-	../include/cncl/cncldef.h ../include/cncl/cnclucom.h \
-	../include/cncl/cncludef.h bjfoption.h bjfpos.h bjipc.h
-bjfrcaccess.o: bjfrcaccess.c bjfrcaccess.h bjfpath.h ../config.h
-getipc.o: getipc.c ../include/cncl/cnclucom.h bjipc.h \
-	../include/cncl/cncldef.h
-
 info-am:
 info: info-am
 dvi-am:
--- configure
+++ configure.org
@@ -1322,6 +1322,16 @@
 
 
 
+cat >>confdefs.h <<\_ACEOF
+#define DEBUG 1
+_ACEOF
+
+
+cat >>confdefs.h <<\_ACEOF
+#define DEBUGLOG 1
+_ACEOF
+
+
 # Check whether --enable-binpath or --disable-binpath was given.
 if test "${enable_binpath+set}" = set; then
   enableval="$enable_binpath"
--- configure.in
+++ configure.org.in
@@ -19,6 +19,8 @@
 AC_SUBST(PIXMAIP1000)
 AC_SUBST(PIXMAIP1500)
 
+AC_DEFINE(DEBUG, 1, "temp")
+AC_DEFINE(DEBUGLOG, 1, "temp")
 
 AC_ARG_ENABLE(binpath)
 if test "${enable_binpath}" = ""; then
--- config.h.in
+++ config.h.org.in
@@ -1,5 +1,11 @@
 /* config.h.in.  Generated from configure.in by autoheader.  */
 
+/* "temp" */
+#undef DEBUG
+
+/* "temp" */
+#undef DEBUGLOG
+
 /* Define to 1 if you have the <inttypes.h> header file. */
 #undef HAVE_INTTYPES_H
 
