diff --git a/Makefile b/Makefile
index c8aa90f..31e27c8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 # Inform the location to intall the modules
-LUAPATH		?= /usr/share/lua/5.1
-LUACPATH	?= /usr/lib/lua/5.1
+LUAPATH		?= /usr/share/lua/5.2
+LUACPATH	?= /usr/lib/lua/5.2
 
 # Compile with build-in LuaSocket's help files.
 # Comment this lines if you will link with non-internal LuaSocket's help files
diff --git a/src/ssl.lua b/src/ssl.lua
index 1b062f6..1b63e70 100644
--- a/src/ssl.lua
+++ b/src/ssl.lua
@@ -4,6 +4,8 @@
 --
 ------------------------------------------------------------------------------
 
+local unpack  = unpack or table.unpack
+
 local core    = require("ssl.core")
 local context = require("ssl.context")
 local x509    = require("ssl.x509")
