diff -ur a/youtube2mp3.sh b/youtube2mp3.sh
--- a/youtube2mp3.sh	2012-04-01 11:27:53.000000000 +0300
+++ b/youtube2mp3.sh	2012-06-03 12:40:04.000000000 +0300
@@ -17,7 +17,8 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 address=$(zenity --width=600 --height=150 --entry --title="YouTube MP3 Extractor" --text "Enter Youtube address (then click OK and then select the destination directory for your MP3 file):")
-bitrate=$(zenity  --list  --text "YouTube MP3 Bitrate" --radiolist  --column "Pick" --column "Bitrate" TRUE 128 FALSE 192 FALSE 256 FALSE 320)
+# zenity's issue https://bugzilla.gnome.org/show_bug.cgi?id=673529
+#bitrate=$(zenity  --list  --text "YouTube MP3 Bitrate" --radiolist  --column "Pick" --column "Bitrate" TRUE 128 FALSE 192 FALSE 256 FALSE 320)
 return_code=$?
 regex='v=(.*)'
 dest_dir=$(zenity --file-selection --directory)
