diff -Naur mistelix-0.33.orig/configure.ac mistelix-0.33/configure.ac
--- mistelix-0.33.orig/configure.ac	2011-01-16 15:45:31.000000000 +0100
+++ mistelix-0.33/configure.ac	2012-03-07 22:38:53.562387550 +0100
@@ -72,7 +72,6 @@
 
 PKG_CHECK_MODULES(MISTELIX,
 	gtk-sharp-2.0 >= $GTKSHARP_REQUIRED
-	gnome-sharp-2.0 >= $GTKSHARP_REQUIRED
 	mono-addins >= 0.3
 	mono-addins-setup >= 0.3
 	mono-addins-gui >= 0.3)
diff -Naur mistelix-0.33.orig/src/mistelix.cs mistelix-0.33/src/mistelix.cs
--- mistelix-0.33.orig/src/mistelix.cs	2011-01-17 20:22:49.000000000 +0100
+++ mistelix-0.33/src/mistelix.cs	2012-03-07 22:34:09.359199021 +0100
@@ -24,7 +24,6 @@
 using System;
 using Gtk;
 using Gdk;
-using Gnome;
 using System.Runtime.InteropServices;
 using System.Collections.Generic;
 using System.Diagnostics;
@@ -44,7 +43,7 @@
 	//
 	// Mistelix main program class
 	//
-	class Mistelix : Program
+	class Mistelix
 	{
 		[GtkBeans.Builder.Object ("mistelix")] Gtk.Window app_window;
 		[GtkBeans.Builder.Object ("vertical_paned")] Gtk.HPaned vpaned;
@@ -99,8 +98,7 @@
 			}
 		}
 
-		public Mistelix (string [] args, params object [] props)
-		: base ("mistelix", Defines.VERSION, Modules.UI,  args, props)
+		public Mistelix (string [] args)
 		{
 			Gtk.Application.Init ();
 			Logger.Info ("Mistelix " + Defines.VERSION + " starting");
@@ -210,7 +208,7 @@
 				Logger.Error ("Mistelix.Main. Cannot set process name");
 			}
 
-			gui.Run ();
+			Gtk.Application.Run ();
 		}
 
 		void OnAddVideos (object sender, EventArgs args)
@@ -263,13 +261,13 @@
 		void OnQuit (object sender, EventArgs args)
 		{
 			SaveWindowPosition ();
-			Quit ();	
+			Gtk.Main.Quit ();	
 		}	
 
 		void OnDeleteWindow (object sender, DeleteEventArgs args)
 		{
 			SaveWindowPosition ();
-			Quit ();
+			Gtk.Main.Quit ();
 		}
 
 		void LoadWindowPosition ()
