--- 9wm.c.orig	2014-09-04 19:23:41.497084967 +0400
+++ 9wm.c	2014-09-04 19:26:07.670414437 +0400
@@ -288,7 +288,7 @@
 	  if (use_keys)
 	    {
 	      if (XLookupKeysym(&(ev.xkey),0) == XK_Tab && 
-		  (ev.xkey.state & ControlMask))
+		  (ev.xkey.state & Mod1Mask))
 		{
 		  if (ev.xkey.state & ShiftMask)
 		    activateprevious();

--- client.c.orig	2014-09-04 19:31:05.590406557 +0400
+++ client.c	2014-09-04 19:32:07.803738248 +0400
@@ -25,11 +25,11 @@
 	  {
 	    XGrabKey(dpy, 
 		     XKeysymToKeycode(dpy, XK_Tab),
-		     ControlMask, 
+		     Mod1Mask, 
 		     root, False, GrabModeAsync, GrabModeAsync);
 	    XGrabKey(dpy, 
 		     XKeysymToKeycode(dpy, XK_Tab),
-		     ControlMask|ShiftMask, 
+		     Mod1Mask|ShiftMask, 
 		     root, False, GrabModeAsync, GrabModeAsync);
 	  }
 
