--- manage.c.orig	2014-09-04 19:24:22.140417227 +0400
+++ manage.c	2014-09-04 01:14:47.585479065 +0400
@@ -91,7 +91,7 @@
 
     if (doreshape) {
         cmapfocus(0);
-        if (!(fixsize ? drag(c) : sweep(c)) && c->is9term) {
+        if (sweep(c) && c->is9term) {
             XDestroyWindow(dpy, c->window);
             rmclient(c);
             cmapfocus(current);
@@ -108,7 +108,7 @@
     XSelectInput(dpy, c->parent, SubstructureRedirectMask | SubstructureNotifyMask);
     if (mapped)
         c->reparenting = 1;
-    if (doreshape && !fixsize)
+    if (doreshape)
         XResizeWindow(dpy, c->window, c->dx, c->dy);
     XSetWindowBorderWidth(dpy, c->window, 0);
     XReparentWindow(dpy, c->window, c->parent, BORDER-1, BORDER-1);
