--- menu.c.orig	2014-09-04 19:24:50.937083133 +0400
+++ menu.c	2014-09-04 19:55:35.287034369 +0400
@@ -1,7 +1,7 @@
 /* Copyright (c) 1994 David Hogan, 2000 Benjamin Drieu, see README for licence details */
 #include <stdio.h>
 #include <unistd.h>
-#include <unistd.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <X11/X.h>
@@ -14,6 +14,7 @@
 
 int numhidden;
 int virtual = 0;
+extern int numvirtuals;
 
 Client * currents[NUMVIRTUALS] =
 {
@@ -135,6 +136,20 @@
 	else
             button3(e);
         break;
+    
+    case Button4:
+            /* scroll up changes to previous virtual screen */
+            if(!c && e->type == ButtonPress)
+                    if(numvirtuals > 1 && virtual > 0)
+                            switch_to(virtual - 1);
+            return;
+
+    case Button5:
+            /* scroll down changes to next virtual screen */
+            if(!c && e->type == ButtonPress)
+                    if(numvirtuals > 1 && virtual < numvirtuals - 1)
+                            switch_to(virtual + 1);
+            return;
     }
 }
 
