diff --git a/src/kimchi/vnc.py b/src/kimchi/vnc.py
index 089d64a..b31eda8 100644
--- a/src/kimchi/vnc.py
+++ b/src/kimchi/vnc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Project Kimchi
 #
diff --git a/src/kimchid.in b/src/kimchid.in
index 7865713..888a463 100644
--- a/src/kimchid.in
+++ b/src/kimchid.in
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Project Kimchi
 #
