--- Configure	2014-03-03 23:15:34.986680578 -0500
+++ Configure	2014-03-03 23:20:44.973347560 -0500
@@ -1576,6 +1576,8 @@ if ($strict_warnings)
 		}
 	}
 
+$shared_ldflag .= " -Wl,--version-script=openssl.ld";
+
 open(IN,'<Makefile.org') || die "unable to read Makefile.org:$!\n";
 unlink("$Makefile.new") || die "unable to remove old $Makefile.new:$!\n" if -e "$Makefile.new";
 open(OUT,">$Makefile.new") || die "unable to create $Makefile.new:$!\n";
--- engines/openssl.ld	1969-12-31 19:00:00.000000000 -0500
+++ engines/openssl.ld	2014-03-03 23:21:52.033347630 -0500
@@ -0,0 +1,4 @@
+OPENSSL_1.0.0 {
+    global:
+        *;
+};
--- Makefile	2014-03-03 23:15:34.960013911 -0500
+++ Makefile	2014-03-03 23:21:28.420014272 -0500
@@ -174,9 +174,9 @@ WDIRS=  windows
 LIBS=   libcrypto.a libssl.a
 SHARED_CRYPTO=libcrypto$(SHLIB_EXT)
 SHARED_SSL=libssl$(SHLIB_EXT)
-SHARED_LIBS=
-SHARED_LIBS_LINK_EXTS=
-SHARED_LDFLAGS=
+SHARED_LIBS=$(SHARED_CRYPTO) $(SHARED_SSL)
+SHARED_LIBS_LINK_EXTS=.so.$(SHLIB_MAJOR) .so
+SHARED_LDFLAGS=-m64 -Wl,--version-script=openssl.ld
 
 GENERAL=        Makefile
 BASENAME=       openssl
--- openssl.ld	1969-12-31 19:00:00.000000000 -0500
+++ openssl.ld	2014-03-03 23:21:59.190014303 -0500
@@ -0,0 +1,4 @@
+OPENSSL_1.0.0 {
+    global:
+        *;
+};
