--- MainPage.qml.orig	2013-06-25 08:18:38.718907000 +0200
+++ MainPage.qml	2013-06-25 08:20:02.248904189 +0200
@@ -244,7 +244,7 @@
                 text: "Help"
                 buttonImage: "qrc:///images/icon_help.png"
                 pressedButtonImage: "qrc:///images/icon_help_pressed.png"
-                onClicked: { mainMenu.hide(); openFile("HelpPage.qml") }
+                onClicked: { mainMenu.hide(); Qt.openUrlExternally("file:///usr/share/heebo/help.html") }
                 /* For testing levels... */
 //                onClicked: { mainMenu.hide(); Jewels.nextLevel() }
             }
@@ -252,7 +252,7 @@
                 text: "About"
                 buttonImage: "qrc:///images/icon_about.png"
                 pressedButtonImage: "qrc:///images/icon_about_pressed.png"
-                onClicked: { mainMenu.hide(); openFile("AboutPage.qml") }
+                onClicked: { mainMenu.hide(); Qt.openUrlExternally("file:///usr/share/heebo/about.html") }
                 /* For testing levels... */
 //                onClicked: { mainMenu.hide(); Jewels.prevLevel() }
                 /* For testing dialogs */
