--- Velocity/Velocity.pro	2014-05-04 12:53:36.564787263 +0200
+++ Velocity/Velocity.pro	2014-05-04 12:55:52.228124886 +0200
@@ -22,8 +22,8 @@
     INCLUDEPATH += C:/botan/include
 }
 macx|unix {
-    INCLUDEPATH += /usr/local/include/botan-1.10
-    LIBS += /usr/local/lib/libbotan-1.10.a
+    INCLUDEPATH += /usr/include/botan-1.10
+    LIBS += /usr/lib/libbotan-1.10.so
 }
 
 # phonon, icon
--- XboxInternals/XboxInternals.pro	2014-05-04 12:53:36.564787263 +0200
+++ XboxInternals/XboxInternals.pro	2014-05-04 12:56:32.148126150 +0200
@@ -23,8 +23,8 @@
     INCLUDEPATH += C:/botan/include
 }
 macx|unix {
-    INCLUDEPATH += /usr/local/include/botan-1.10
-    LIBS += /usr/local/lib/libbotan-1.10.a
+    INCLUDEPATH += /usr/include/botan-1.10
+    LIBS += /usr/lib/libbotan-1.10.so
 }
 
 SOURCES += \
