--- src/comun.py	2012-08-03 07:37:35.000000000 +0200
+++ src/comun.py	2012-08-03 17:06:51.075277793 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#! /usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 __author__="atareao"
@@ -82,26 +82,17 @@
 ICON = '/usr/share/pixmaps/my-weather-indicator.png'
 #########################################
 
-# check if running from source
-if is_package():
-    ROOTDIR = '/usr/share/'
-    LANGDIR = os.path.join(ROOTDIR, 'locale-langpack')
-    APPDIR = os.path.join(ROOTDIR, APP)
-    LOGOSDIR = os.path.join(APPDIR, 'logos')    
-else:
-    VERSION = VERSION + '-src'
-    ROOTDIR = os.path.dirname(__file__)
-    LANGDIR = os.path.normpath(os.path.join(ROOTDIR, '../template1'))
-    APPDIR = ROOTDIR
-    LOGOSDIR = os.path.normpath(os.path.join(ROOTDIR, '../data/logos'))
-
+ROOTDIR = '/usr/share/'
+LANGDIR = os.path.join(ROOTDIR, 'locale')
+APPDIR = os.path.join(ROOTDIR, APP)
+LOGOSDIR = os.path.join(APPDIR, 'logos')    
 GOOGLELOGO = os.path.join(LOGOSDIR, 'wgooglelogo.png')
 UNDERGROUNDLOGO = os.path.join(LOGOSDIR, 'wundergroundlogo.png')   
 YAHOOLOGO = os.path.join(LOGOSDIR, 'wyahoologo.png')   
 
 ####
-gettext.bindtextdomain(APP, LANGDIR)
-gettext.textdomain(APP)
-locale.setlocale(locale.LC_ALL, '')
-t = gettext.translation(APP, 'locale')
-_ = t.ugettext
+current_locale, encoding = locale.getdefaultlocale()
+locale_path = '/usr/share/locale/'
+language = gettext.translation ('my-weather-indicator', locale_path, [current_locale] )
+language.install()
+_ = language.ugettext
