diff -urN x48-0.6.3.orig/Makefile.am x48-0.6.3/Makefile.am
--- x48-0.6.3.orig/Makefile.am	2010-01-31 03:06:57.000000000 +0100
+++ x48-0.6.3/Makefile.am	2010-07-31 01:54:59.557055305 +0200
@@ -12,6 +12,3 @@
 	cp $< $@
 
 appdefault_DATA = X48
-
-DISTCHECK_CONFIGURE_FLAGS = --with-appdefaultdir='$${datadir}/X11/app-defaults'
-
diff -urN x48-0.6.3.orig/configure.ac x48-0.6.3/configure.ac
--- x48-0.6.3.orig/configure.ac	2010-02-01 15:11:49.000000000 +0100
+++ x48-0.6.3/configure.ac	2010-07-31 01:47:15.317157141 +0200
@@ -114,6 +114,7 @@
      [appdefaultdir="$withval"],
      [appdefaultdir=`$PKG_CONFIG --variable=appdefaultdir xt`])
 AC_SUBST(appdefaultdir)
+AC_DEFINE_UNQUOTED([CONFIG_APPDEFAULTDIR], ["$appdefaultdir"], ["Xt app-defaults directory"])
 
 AM_INIT_AUTOMAKE
 
diff -urN x48-0.6.3.orig/src/x48_x11.c x48-0.6.3/src/x48_x11.c
--- x48-0.6.3.orig/src/x48_x11.c	2010-02-01 17:56:42.000000000 +0100
+++ x48-0.6.3/src/x48_x11.c	2010-07-31 01:53:07.761232478 +0200
@@ -1101,7 +1101,7 @@
 
   /* 2. /usr/lib/X11/app-defaults/X48 */
 
-  merge_app_defaults("/usr/lib/X11/app-defaults", &rdb);
+  merge_app_defaults(CONFIG_APPDEFAULTDIR, &rdb);
 
   /* 3. Values in $XUSERFILESEARCHPATH/X48, or $XAPPLRESDIR/X48 */
 
