diff -u airpwn-1.4.orig/configure.ac airpwn-1.4/configure.ac
--- airpwn-1.4.orig/configure.ac	2009-05-27 09:15:23.000000000 -0700
+++ airpwn-1.4/configure.ac	2010-05-06 21:13:36.576175190 -0700
@@ -14,6 +14,7 @@
 AC_CHECK_LIB(pthread, pthread_create, [], AC_MSG_ERROR(libpthread required))
 AC_CHECK_LIB(orcon, tx80211_txpacket, [], AC_MSG_ERROR(LORCON required. Install LORCON before building airpwn.))
 AC_CHECK_LIB(ssl, MD5_Init, [], AC_MSG_NOTICE(openssl required for full WEP key generation))
+AC_CHECK_LIB(crypto, MD5_Init, [], AC_MSG_NOTICE(libcrypto required for full WEP key generation))
 AC_CHECK_LIB(python2.4, PyString_FromString, [], AC_MSG_NOTICE(Python2.4 required))
 
 # Checks for header files.
