--- /src/BatteryMonitor.py_orig	2011-01-16 03:18:11.000000000 +1000
+++ /src/BatteryMonitor.py	2011-07-01 22:01:08.407609000 +1100
@@ -81,13 +81,13 @@
             self.__lmenu = gtk.Menu()
             if self.__backend.can_suspend():
                 suspend_item = gtk.ImageMenuItem(_('Suspend'))
-                suspend_icon = gtk.image_new_from_icon_name('batti-suspend', gtk.ICON_SIZE_MENU)
+                suspend_icon = gtk.image_new_from_icon_name('system-suspend', gtk.ICON_SIZE_MENU)
                 suspend_item.set_image(suspend_icon)
                 suspend_item.connect('activate', self.__suspend)
                 self.__lmenu.append(suspend_item)
             if self.__backend.can_hibernate():
                 hibernate_item = gtk.ImageMenuItem(_('Hibernate'))
-                hibernate_icon = gtk.image_new_from_icon_name('batti-hibernate', gtk.ICON_SIZE_MENU)
+                hibernate_icon = gtk.image_new_from_icon_name('system-hibernate', gtk.ICON_SIZE_MENU)
                 hibernate_item.set_image(hibernate_icon)
                 hibernate_item.connect('activate', self.__hibernate)
                 self.__lmenu.append(hibernate_item)
@@ -126,11 +126,12 @@
 
     def about(self, button):
         about_dg = gtk.AboutDialog()
-        about_dg.set_name("Batti")
+        about_dg.set_name(Constants.NAME)
         about_dg.set_program_name(Constants.NAME)
         about_dg.set_version(Constants.VERSION)
         about_dg.set_comments(_("Battery Monitor"))
         about_dg.set_license(Constants.LICENSE_TEXT)
+	about_dg.set_logo_icon_name(Constants.NAME)
         about_dg.set_authors(['%s <%s>' % (Constants.AUTHOR, Constants.AUTHOR_EMAIL)])
         about_dg.connect("response", lambda d, r: d.destroy())
         about_dg.show()
