--- setup.py	2011-05-10 16:01:33.000000000 +0200
+++ setup.py	2011-08-09 08:00:13.000000000 +0200
@@ -27,12 +27,12 @@
 import sys
 import tempfile
 
-if not os.access('/usr/share/virtualbricks/', os.X_OK):
-	try:
-		os.mkdir('/usr/share/virtualbricks')
-	except:
-		print "Cannot create directory. Bye."
-		sys.exit(1)
+#if not os.access('/usr/share/virtualbricks/', os.X_OK):
+#	try:
+#		os.mkdir('/usr/share/virtualbricks')
+#	except:
+#		print "Cannot create directory. Bye."
+#		sys.exit(1)
 
 
 FILES = [
