--- setup.py
+++ setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # coding: utf-8
 
 """
@@ -33,9 +33,9 @@
		sys.prefix=arg.split('=')[1]
 
 glade = open('share/virtualbricks.template.glade','r').read()
-micro = open('.bzr/branch/last-revision','r').read().split(' ')[0]
-if micro == '':
-	micro = '000'
+"""micro = open('.bzr/branch/last-revision','r').read().split(' ')[0]
+if micro == '':"""
+micro = '000'
 
 virtualbricks_version=CURRENT_VERSION+'.'+micro
