diff -r 51d06f7938f3 dist/build.py
--- a/dist/build.py	Thu Oct 06 13:50:40 2011 +0200
+++ b/dist/build.py	Sun Jan 15 23:03:12 2012 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 ## -*- coding: utf-8 -*-
 
 import sys
@@ -12,7 +12,7 @@
 
 
 LAUNCHER_STUB = """\
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ## -*- coding: utf-8 -*-
 
 
diff -r 51d06f7938f3 spyrit.sh
--- a/spyrit.sh	Thu Oct 06 13:50:40 2011 +0200
+++ b/spyrit.sh	Sun Jan 15 23:03:12 2012 +0100
@@ -20,7 +20,7 @@
 
 if [[ $_BUILD_RESOURCES == "1" ]] ; then
   cd $_THIS_DIR/src/resources/
-  python mkresources.py
+  python2 mkresources.py
   cd $_OLD_DIR
 fi
 
@@ -28,7 +28,7 @@
 ## Launch Spyrit, but only check dependencies:
 
 cd $_THIS_DIR/src
-errmsg=$(python spyrit.py --check-dependencies-only)
+errmsg=$(python2 spyrit.py --check-dependencies-only)
 errstate=$?
 cd $_OLD_DIR
 
@@ -53,7 +53,7 @@
 ## Launch Spyrit for real.
 
 cd $_THIS_DIR/src
-python spyrit.py $@
+python2 spyrit.py $@
 errstate=$?
 cd $_OLD_DIR
 
diff -r 51d06f7938f3 src/resources/mkresources.py
--- a/src/resources/mkresources.py	Thu Oct 06 13:50:40 2011 +0200
+++ b/src/resources/mkresources.py	Sun Jan 15 23:03:12 2012 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 import os
 import subprocess
diff -r 51d06f7938f3 src/resources/spyrcc.py
--- a/src/resources/spyrcc.py	Thu Oct 06 13:50:40 2011 +0200
+++ b/src/resources/spyrcc.py	Sun Jan 15 23:03:12 2012 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 # -*- coding: utf-8 -*-
 
 ## Copyright (c) 2007-2011 Pascal Varet <p.varet@gmail.com>
