--- imgur.old	2014-02-02 12:26:50.086785984 +0100
+++ imgur	2014-02-02 12:27:48.111094239 +0100
@@ -59,8 +59,7 @@
 # parse the response and output our stuff
 url=$(echo $response | sed -r 's/.*<original_image>(.*)<\/original_image>.*/\1/')
 deleteurl=$(echo $response | sed -r 's/.*<delete_page>(.*)<\/delete_page>.*/\1/')
-qdbus org.kde.klipper /klipper org.kde.klipper.klipper.setClipboardContents "$url"
-kdialog --passivepopup "Done. Image link copied to the clipboard" 5
+kde-open "$url"
 echo >>$HOME/.imgur_history
 echo "Source image: $1" >>$HOME/.imgur_history
 echo "-------------" >>$HOME/.imgur_history
