--- /apmil/apmil.py
+++ /apmil/apmil.py
@@ -26,8 +26,8 @@
 else:
     ICON = wx.Icon(('/usr/share/pixmaps/ap_logo.png'if not \
         os.path.exists('ap_logo.png') else 'ap_logo.png'), wx.BITMAP_TYPE_PNG)
-    config_path = os.path.join(os.environ['XDG_CONFIG_HOME'], 'apmil')
-    data_path = os.path.join(os.environ['XDG_DATA_HOME'], 'apmil')
+    config_path = os.path.join(os.environ['HOME'], '.config', 'apmil')
+    data_path = os.path.join(os.environ['HOME'], '.local', 'share', 'apmil')
     if not os.path.exists(data_path): os.mkdir(data_path)
     if not os.path.exists(config_path): os.mkdir(config_path)
     CFG = wx.FileConfig('apmil', localFilename=config_path+os.sep+'config')
