--- omxplayer/omxplayer.cpp~	2012-12-14 16:56:22.983517121 +0000
+++ omxplayer/omxplayer.cpp	2012-12-16 15:06:19.932644711 +0000
@@ -66,7 +66,7 @@
 bool              m_HWDecode            = false;
 std::string       deviceString          = "omx:local";
 int               m_use_hw_audio        = false;
-std::string       m_font_path           = "/usr/share/fonts/truetype/freefont/FreeSans.ttf";
+std::string       m_font_path           = "/usr/share/fonts/TTF/FreeSans.ttf";
 float             m_font_size           = 0.055f;
 bool              m_centered            = false;
 bool              m_Pause               = false;
@@ -135,7 +135,7 @@
   printf("         -l / --pos                     start position (in seconds)\n");  
   printf("              --boost-on-downmix        boost volume when downmixing\n");
   printf("              --font path               subtitle font\n");
-  printf("                                        (default: /usr/share/fonts/truetype/freefont/FreeSans.ttf)\n");
+  printf("                                        (default: /usr/share/fonts/TTF/FreeSans.ttf)\n");
   printf("              --font-size size          font size as thousandths of screen height\n");
   printf("                                        (default: 55)\n");
   printf("              --align left/center       subtitle alignment (default: left)\n");
