--- omxplayer.orig/Makefile	2012-06-04 21:08:51.753177732 +0100
+++ omxplayer/Makefile	2012-06-04 21:29:54.303177671 +0100
@@ -2,8 +2,8 @@
 
 CFLAGS+=-DSTANDALONE -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -DTARGET_POSIX -D_LINUX -fPIC -DPIC -D_REENTRANT -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -DHAVE_CMAKE_CONFIG -D__VIDEOCORE4__ -U_FORTIFY_SOURCE -Wall -mfpu=vfp -mfloat-abi=softfp -mno-apcs-stack-check -DHAVE_OMXLIB -DUSE_EXTERNAL_FFMPEG  -DHAVE_LIBAVCODEC_AVCODEC_H -DHAVE_LIBAVUTIL_MEM_H -DHAVE_LIBAVUTIL_AVUTIL_H -DHAVE_LIBAVFORMAT_AVFORMAT_H -DHAVE_LIBAVFILTER_AVFILTER_H -DOMX -DOMX_SKIP64BIT -ftree-vectorize -pipe -DUSE_EXTERNAL_OMX -DHAVE_PLATFORM_RASPBERRY_PI -DUSE_EXTERNAL_LIBBCM_HOST -Wno-psabi -I$(SDKSTAGE)/opt/vc/include/ 
 
-LDFLAGS+=-L./ -lc -lWFC -lGLESv2 -lEGL -lbcm_host -lopenmaxil -Lffmpeg_compiled/usr/local/lib/
-INCLUDES+=-I./ -Ilinux -Iffmpeg_compiled/usr/local/include/
+LDFLAGS+=-L./ -lc -lWFC -lGLESv2 -lEGL -lbcm_host -lopenmaxil -lpthread -lrt
+INCLUDES+=-I./ -Ilinux
 
 SRC=linux/XMemUtils.cpp \
 		utils/log.cpp \
@@ -29,7 +29,7 @@
 
 OBJS+=$(filter %.o,$(SRC:.cpp=.o))
 
-all: omxplayer.bin
+all: omxplayer
 
 %.o: %.cpp
 	@rm -f $@ 
@@ -38,28 +38,21 @@
 list_test:
 	$(CXX) -O3 -o list_test list_test.cpp
 
-omxplayer.bin: $(OBJS)
-	$(CXX) $(LDFLAGS) -o omxplayer.bin -Wl,--whole-archive $(OBJS) -Wl,--no-whole-archive -rdynamic -lavutil -lavcodec -lavformat -lswscale -lpcre
-	#arm-unknown-linux-gnueabi-strip omxplayer.bin
+omxplayer: $(OBJS)
+	$(CXX) $(LDFLAGS) -o omxplayer -Wl,--whole-archive $(OBJS) -Wl,--no-whole-archive -rdynamic -lavutil -lavcodec -lavformat -lswscale -lpcre
 
 clean:
 	for i in $(OBJS); do (if test -e "$$i"; then ( rm $$i ); fi ); done
 	@rm -f omxplayer.old.log omxplayer.log
-	@rm -f omxplayer.bin
+	@rm -f omxplayer
 	@rm -rf omxplayer-dist
 	@rm -f omxplayer-dist.tar.gz
 	make -f Makefile.ffmpeg clean
 
-ffmpeg:
-	@rm -rf ffmpeg
-	make -f Makefile.ffmpeg
-	make -f Makefile.ffmpeg install
-
-dist: omxplayer.bin
-	mkdir -p omxplayer-dist/usr/lib/omxplayer
-	mkdir -p omxplayer-dist/usr/usr/bin
-	mkdir -p omxplayer-dist/usr/share/doc
-	cp omxplayer omxplayer.bin omxplayer-dist/usr/usr/bin
-	cp README COPYING omxplayer-dist/usr/share/doc/
-	cp -a ffmpeg_compiled/usr/local/lib/*.so* omxplayer-dist/usr/lib/omxplayer/
-	tar -czf omxplayer-dist.tar.gz omxplayer-dist
+install:
+	install -d $(DESTDIR)/usr/bin
+	install -d $(DESTDIR)/usr/share/doc/omxplayer
+	install omxplayer $(DESTDIR)/usr/bin/
+	install README COPYING $(DESTDIR)/usr/share/doc/omxplayer/
+
+	
--- omxplayer.orig/Makefile.include	2012-06-04 21:08:51.753177732 +0100
+++ omxplayer/Makefile.include	2012-06-06 17:21:23.106690043 +0100
@@ -1,37 +1,36 @@
-USE_BUILDROOT=1
+USE_BUILDROOT=0
 
 ifeq ($(USE_BUILDROOT), 1)
-BUILDROOT	:=/opt/xbmc-bcm/buildroot
+BUILDROOT	:=./
 SDKSTAGE	:=$(BUILDROOT)/output/staging
 TARGETFS	:=$(BUILDROOT)/output/target
 TOOLCHAIN	:=$(BUILDROOT)/output/host/usr/
-HOST			:=arm-unknown-linux-gnueabi
+HOST			:=
 SYSROOT		:=$(BUILDROOT)/output/host/usr/arm-unknown-linux-gnueabi/sysroot
 else
-BUILDROOT	:=/opt/bcm-rootfs
-SDKSTAGE	:=/opt/bcm-rootfs
-TARGETFS	:=/opt/bcm-rootfs
-TOOLCHAIN	:=/usr/local/bcm-gcc
-HOST			:=bcm2708
-SYSROOT		:=$(TOOLCHAIN)/arm-bcm2708-linux-gnueabi/sys-root
+BUILDROOT	:=
+SDKSTAGE	:=
+TARGETFS	:=
+TOOLCHAIN	:=gcc
+HOST		:=
+SYSROOT		:=
 endif
 
-JOBS=7
+JOBS=1
 
-CFLAGS			:= -isystem$(PREFIX)/include
+CFLAGS			:= 
 CXXFLAGS		:= $(CFLAGS)
 CPPFLAGS		:= $(CFLAGS)
-LDFLAGS			:= -L$(BUILDROOT)/lib
-LD					:= $(TOOLCHAIN)/bin/$(HOST)-ld --sysroot=$(SYSROOT)
-CC					:= $(TOOLCHAIN)/bin/$(HOST)-gcc --sysroot=$(SYSROOT)
-CXX         := $(TOOLCHAIN)/bin/$(HOST)-g++ --sysroot=$(SYSROOT)
-OBJDUMP			:= $(TOOLCHAIN)/bin/$(HOST)-objdump
-RANLIB			:= $(TOOLCHAIN)/bin/$(HOST)-ranlib
-STRIP				:= $(TOOLCHAIN)/bin/$(HOST)-strip
-AR 					:= $(TOOLCHAIN)/bin/$(HOST)-ar
+LDFLAGS			:= 
+LD			:= ld
+CC			:= gcc
+CXX         := g++
+OBJDUMP			:= objdump
+RANLIB			:= ranlib
+STRIP			:= strip
+AR 			:= ar
 CXXCP 			:= $(CXX) -E
-PATH 				:= $(PREFIX)/bin:$(BUILDROOT)/output/host/usr/bin:$(PATH)
 
 CFLAGS			+= -pipe -mfloat-abi=softfp -mcpu=arm1176jzf-s -fomit-frame-pointer -mabi=aapcs-linux -mtune=arm1176jzf-s -mfpu=vfp -Wno-psabi -mno-apcs-stack-check -O3 -mstructure-size-boundary=32 -mno-sched-prolog
 LDFLAGS			+= -L$(SDKSTAGE)/lib -L$(SDKSTAGE)/usr/lib -L$(SDKSTAGE)/opt/vc/lib/
-INCLUDES		+= -isystem$(SDKSTAGE)/staging/usr/include -isystem$(SDKSTAGE)/staging/opt/vc/include -isystem$(SYSROOT)/usr/include
+INCLUDES		+= -I/opt/vc/include 
