--- Makefile.orig	2010-10-23 17:54:17.929231534 +0200
+++ Makefile	2010-10-23 17:56:26.775898200 +0200
@@ -1,4 +1,6 @@
-export prefix=/usr/local
+DESTDIR ?= 
+prefix ?= $(DESTDIR)/usr
+export prefix
 export milena_dir=$(prefix)/share/milena
 
 all:
--- data/Makefile.orig	2010-10-23 17:56:35.022564867 +0200
+++ data/Makefile	2010-10-23 17:59:27.812564868 +0200
@@ -4,11 +4,10 @@
 	cat pl_phraser.dat.in pl_imiona.in >pl_phraser.dat
 
 install:
-	mkdir -p $(milena_dir)
+	install -dm 0755 $(milena_dir)
 	install -m 0644 *.dat $(milena_dir)
 clean:
 	rm -f *~ pl_phraser.dat
-	
 uninstall:
 	for i in pl_genpor_theme.dat pl_phraser.dat *.dat;do rm -f $(milena_dir)/"$$i";done
 	if [ -d $(milena_dir) ] ; then rmdir --ignore-fail-on-non-empty $(milena_dir);fi
