--- src/KXLvisual.c.old	2012-02-06 15:26:58.989193583 +0600
+++ src/KXLvisual.c	2012-02-06 15:27:36.942527131 +0600
@@ -429,9 +429,12 @@
                                  0, 0);
     KXL_Root->WinFont = XLoadQueryFont(KXL_Root->Display, str);
     if (KXL_Root->WinFont == (XFontStruct *)NULL) {
+      KXL_Root->WinFont = XLoadQueryFont(KXL_Root->Display, "fixed");
+    }
+    if (KXL_Root->WinFont == (XFontStruct *)NULL) {
       fprintf(stderr,
               "KXL error message\n"
-              "loading font error (%s)\n",
+              "loading font error (%s), fallback failed\n",
               str);
       exit(-1);
     }
