diff -ru typhon-svn/src/editor.cpp typhon-svn-build/src/editor.cpp
--- typhon-svn/src/editor.cpp	2011-08-28 19:16:00.000000000 +0200
+++ typhon-svn-build/src/editor.cpp	2011-08-28 19:41:16.000000000 +0200
@@ -145,7 +145,7 @@
 	}
 	
 #if defined(SFML_VERSION_MAJOR) && (SFML_VERSION_MAJOR>=2)
-	else if((ekey == sf::Keyboard::Delete || ekey == sf::Keyboard::Back) && searchlength
+	else if((ekey == sf::Keyboard::Delete || ekey == sf::Keyboard::Back) && searchlength)
 #else	
 	else if((ekey == sf::Key::Delete || ekey == sf::Key::Back) && searchlength)
 #endif
diff -ru typhon-svn/src/inputhandling.cpp typhon-svn-build/src/inputhandling.cpp
--- typhon-svn/src/inputhandling.cpp	2011-08-28 19:16:00.000000000 +0200
+++ typhon-svn-build/src/inputhandling.cpp	2011-08-28 19:47:48.000000000 +0200
@@ -751,7 +751,7 @@
 	if (strcasecmp(Key,"7") == 0)            return sf::Keyboard::Num7;
 	if (strcasecmp(Key,"8") == 0)            return sf::Keyboard::Num8;
 	if (strcasecmp(Key,"9") == 0)            return sf::Keyboard::Num9;
-	return sf::Keyboard::Code(0);
+	return (0);
 }
 
 #else
@@ -1343,7 +1343,7 @@
 			}
 
 #if defined(SFML_VERSION_MAJOR) && (SFML_VERSION_MAJOR>=2)
-			if(event.Type==event.JoystickMoved && usejoypad))
+			if(event.Type==event.JoystickMoved && usejoypad)
 			{
 				if (event.JoystickMove.Axis == sf::Joystick::X)
 				{
