--- dbconfig.py.orig	2013-03-24 20:28:59.718754044 +0100
+++ dbconfig.py	2013-03-24 20:29:29.922736655 +0100
@@ -35,7 +35,7 @@
 		else:
 			self.hostfile = None
 		# read schema version on init; if not 0 or 1, we should not try to guess
-		if self.schemaversion not in (0,1):
+		if self.schemaversion not in (0,1,2):
 			raise Exception('Dropbox schema version %s is not supported.' % self.schemaversion)
 
 	_configpath = None
