--- blockcontrol/blockcontrol_old.lib	2009-09-14 19:35:24.000000000 +0200
+++ blockcontrol/blockcontrol.lib	2009-11-05 20:29:50.000000000 +0100
@@ -1344,10 +1344,10 @@
 				DAEMON_OPTS="$CUSTOM_DAEMON_OPTS"
 			elif [ "$NAME" = moblock ] ; then
 				# The options MoBlock is started with
-				DAEMON_OPTS="-$BLOCKLIST_FORMAT $BLOCKLIST $NFQUEUE_NUMBER_OPT $LOG_SYSLOG_OPT $LOG_TIMESTAMP_OPT $REJECT_OPT $ACCEPT_OPT $DAEMON_LOG"
+				DAEMON_OPTS="-$BLOCKLIST_FORMAT $BLOCKLIST $NFQUEUE_NUMBER_OPT $LOG_SYSLOG_OPT $DAEMON_LOG"
 			elif [ "$NAME" = nfblockd ] ; then
 				# The options NFBlock is started with
-				DAEMON_OPTS="-d -f $BLOCKLIST -p $PIDFILE $NFQUEUE_NUMBER_OPT $REJECT_OPT $ACCEPT_OPT $LOG_SYSLOG_OPT"
+				DAEMON_OPTS="-d -f $BLOCKLIST -p $PIDFILE $NFQUEUE_NUMBER_OPT $LOG_SYSLOG_OPT"
 			fi
 
 			# Test NICE_LEVEL setting.
