diff -Naur batti-0.3.8-orig/src/Battery.py batti-0.3.8/src/Battery.py
--- batti-0.3.8-orig/src/Battery.py	2011-01-15 18:18:11.000000000 +0100
+++ batti-0.3.8/src/Battery.py	2012-10-22 14:30:50.703906043 +0200
@@ -92,14 +92,18 @@
         if info._state == STATE_CHARGING:
             self.__systray.set_blinking(False)
             self.__systray.set_tooltip(_('Charging battery\nBattery level: %s%%') % info._percentage)
-            if info._percentage > 80:
-                icon = 'battery-full-charging'
+            if info._percentage > 90:
+                icon = 'notification-gpm-battery-100-charging'
+            elif info._percentage > 80:
+                icon = 'notification-gpm-battery-080-charging'
+            elif info._percentage > 60:
+                icon = 'notification-gpm-battery-060-charging'
             elif info._percentage > 40:
-                icon = 'battery-good-charging'
+                icon = 'notification-gpm-battery-040-charging'
             elif info._percentage > 20:
-                icon = 'battery-low-charging'
+                icon = 'notification-gpm-battery-020-charging'
             else:
-                icon = 'battery-caution-charging'
+                icon = 'notification-gpm-battery-000-charging'
             self.__set_tray_icon(icon)
             
             self.__shown_bat_charged = False
@@ -112,19 +116,23 @@
                 
         elif info._state == STATE_DISCHARGING:
             self.__systray.set_tooltip(tooltip)
-            if info._percentage > 80:
-                icon = 'battery-full'
+            if info._percentage == 100:
+                icon = 'notification-gpm-battery-100'
+            elif info._percentage > 60:
+                icon = 'notification-gpm-battery-080'
+            elif info._percentage > 60:
+                icon = 'notification-gpm-battery-060'
             elif info._percentage > 40:
-                icon = 'battery-good'
+                icon = 'notification-gpm-battery-040'
             elif info._percentage > 20:
-                icon = 'battery-low'
+                icon = 'notification-gpm-battery-020'
             elif info._percentage > 5:
-                icon = 'battery-caution'
+                icon = 'notification-gpm-battery-020'
                 if not self.__shown_bat_low:
                     self._notify(True, icon, _('Low battery level'), notification)
                     self.__shown_bat_low = True
             else:
-                icon = 'battery-empty'
+                icon = 'notification-gpm-battery-000'
                 self.__systray.set_blinking(True)
                 if not self.__shown_bat_critical:
                     self._notify(True, icon, _('Critical battery level'), notification)
@@ -138,10 +146,10 @@
                     
         elif info._state == STATE_CHARGED:
             self.__systray.set_tooltip(_('Battery charged\n%s') % tooltip)
-            self.__set_tray_icon('battery-full')
+            self.__set_tray_icon('notification-gpm-battery-charged')
             self.__systray.set_blinking(False)
             if not self.__shown_bat_charged:
-                self._notify(False, 'battery-full-charging', _('Battery charged'), notification)
+                self._notify(False, 'notification-gpm-battery-charged', _('Battery charged'), notification)
                 self.__shown_bat_charged = True
                      
     
