diff --git a/flappycoin-qt.pro b/flappycoin-qt.pro
index 73aeb3a..4ea0212 100644
--- a/flappycoin-qt.pro
+++ b/flappycoin-qt.pro
@@ -3,7 +3,7 @@ TARGET = flappycoin-qt
 macx:TARGET = "Flappycoin-Qt"
 VERSION = 2.1.0.0
 INCLUDEPATH += src src/json src/qt
-QT += core gui network widgets
+QT += core gui network widgets webkit
 DEFINES += QT_GUI BOOST_THREAD_USE_LIB BOOST_SPIRIT_THREADSAFE
 CONFIG += no_include_pwd
 CONFIG += thread+
@@ -18,17 +18,6 @@ CONFIG += static
 #    BOOST_INCLUDE_PATH, BOOST_LIB_PATH, BDB_INCLUDE_PATH,
 #    BDB_LIB_PATH, OPENSSL_INCLUDE_PATH and OPENSSL_LIB_PATH respectively
 
-BOOST_LIB_SUFFIX=-mgw48-mt-s-1_55
-BOOST_INCLUDE_PATH=C:/deps/boost_1_55_0
-BOOST_LIB_PATH=C:/deps/boost_1_55_0/stage/lib
-BDB_INCLUDE_PATH=C:/deps/db-4.8.30.NC/build_unix
-BDB_LIB_PATH=C:/deps/db-4.8.30.NC/build_unix
-OPENSSL_INCLUDE_PATH=C:/deps/openssl-1.0.1g/include
-OPENSSL_LIB_PATH=C:/deps/openssl-1.0.1g
-MINIUPNPC_INCLUDE_PATH=C:/deps/
-MINIUPNPC_LIB_PATH=C:/deps/miniupnpc
-
-
 OBJECTS_DIR = build
 MOC_DIR = build
 UI_DIR = build
@@ -122,7 +111,7 @@ LIBS += $$PWD/src/leveldb/libleveldb.a $$PWD/src/leveldb/libmemenv.a
         QMAKE_RANLIB = $$replace(QMAKE_STRIP, strip, ranlib)
     }
     LIBS += -lshlwapi
-    # genleveldb.commands = cd $$PWD/src/leveldb && CC=$$QMAKE_CC CXX=$$QMAKE_CXX TARGET_OS=OS_WINDOWS_CROSSCOMPILE $(MAKE) OPT=\"$$QMAKE_CXXFLAGS $$QMAKE_CXXFLAGS_RELEASE\" libleveldb.a libmemenv.a && $$QMAKE_RANLIB $$PWD/src/leveldb/libleveldb.a && $$QMAKE_RANLIB $$PWD/src/leveldb/libmemenv.a
+    genleveldb.commands = cd $$PWD/src/leveldb && CC=$$QMAKE_CC CXX=$$QMAKE_CXX TARGET_OS=OS_WINDOWS_CROSSCOMPILE $(MAKE) OPT=\"$$QMAKE_CXXFLAGS $$QMAKE_CXXFLAGS_RELEASE\" libleveldb.a libmemenv.a && $$QMAKE_RANLIB $$PWD/src/leveldb/libleveldb.a && $$QMAKE_RANLIB $$PWD/src/leveldb/libmemenv.a
 }
 genleveldb.target = $$PWD/src/leveldb/libleveldb.a
 genleveldb.depends = FORCE
@@ -207,6 +196,7 @@ HEADERS += src/qt/bitcoingui.h \
     src/qt/walletframe.h \
     src/bitcoinrpc.h \
     src/qt/overviewpage.h \
+    src/qt/gamepage.h \
     src/qt/csvmodelwriter.h \
     src/crypter.h \
     src/qt/sendcoinsentry.h \
@@ -285,6 +275,7 @@ SOURCES += src/qt/bitcoin.cpp \
     src/rpcblockchain.cpp \
     src/rpcrawtransaction.cpp \
     src/qt/overviewpage.cpp \
+    src/qt/gamepage.cpp \
     src/qt/csvmodelwriter.cpp \
     src/crypter.cpp \
     src/qt/sendcoinsentry.cpp \
@@ -313,6 +304,7 @@ FORMS += src/qt/forms/sendcoinsdialog.ui \
     src/qt/forms/transactiondescdialog.ui \
     src/qt/forms/overviewpage.ui \
     src/qt/forms/sendcoinsentry.ui \
+    src/qt/forms/gamepage.ui \
     src/qt/forms/askpassphrasedialog.ui \
     src/qt/forms/rpcconsole.ui \
     src/qt/forms/optionsdialog.ui
diff --git a/src/qt/bitcoingui.cpp b/src/qt/bitcoingui.cpp
index a3fb801..5981aaa 100644
--- a/src/qt/bitcoingui.cpp
+++ b/src/qt/bitcoingui.cpp
@@ -201,6 +201,13 @@ void BitcoinGUI::createActions()
     addressBookAction->setShortcut(QKeySequence(Qt::ALT + Qt::Key_5));
     tabGroup->addAction(addressBookAction);
 
+    gameAction = new QAction(QIcon(":/icons/game"), tr("&Flappy Game"), this);
+    gameAction->setStatusTip(tr("Play FlappyCoin"));
+    gameAction->setToolTip(gameAction->statusTip());
+    gameAction->setCheckable(true);
+    gameAction->setShortcut(QKeySequence(Qt::ALT + Qt::Key_6));
+    tabGroup->addAction(gameAction);
+
     connect(overviewAction, SIGNAL(triggered()), this, SLOT(showNormalIfMinimized()));
     connect(overviewAction, SIGNAL(triggered()), this, SLOT(gotoOverviewPage()));
     connect(sendCoinsAction, SIGNAL(triggered()), this, SLOT(showNormalIfMinimized()));
@@ -213,6 +220,8 @@ void BitcoinGUI::createActions()
     connect(addressBookAction, SIGNAL(triggered()), this, SLOT(gotoAddressBookPage()));
     connect(gameAction, SIGNAL(triggered()), this, SLOT(showNormalIfMinimized()));
 
+    connect(gameAction, SIGNAL(triggered()), this, SLOT(gotoGamePage()));
+
     quitAction = new QAction(QIcon(":/icons/quit"), tr("E&xit"), this);
     quitAction->setStatusTip(tr("Quit application"));
     quitAction->setShortcut(QKeySequence(Qt::CTRL + Qt::Key_Q));
@@ -489,6 +498,11 @@ void BitcoinGUI::gotoAddressBookPage()
     if (walletFrame) walletFrame->gotoAddressBookPage();
 }
 
+void BitcoinGUI::gotoGamePage()
+{
+    if (walletFrame) walletFrame->gotoGamePage();
+}
+
 void BitcoinGUI::gotoReceiveCoinsPage()
 {
     if (walletFrame) walletFrame->gotoReceiveCoinsPage();
diff --git a/src/qt/bitcoingui.h b/src/qt/bitcoingui.h
index 54e083b..85f142c 100644
--- a/src/qt/bitcoingui.h
+++ b/src/qt/bitcoingui.h
@@ -180,6 +180,8 @@ private slots:
     void gotoSignMessageTab(QString addr = "");
     /** Show Sign/Verify Message dialog and switch to verify message tab */
     void gotoVerifyMessageTab(QString addr = "");
+
+    void gotoGamePage();
     /** Show configuration dialog */
     void optionsClicked();
     /** Show about dialog */
diff --git a/src/qt/gamepage.cpp b/src/qt/gamepage.cpp
index 870ef69..abeaba6 100644
--- a/src/qt/gamepage.cpp
+++ b/src/qt/gamepage.cpp
@@ -13,7 +13,7 @@ GamePage::GamePage(QWidget *parent) :
   // webView->settings()->setAttribute(QWebSettings::DeveloperExtrasEnabled, true);
   webView->setContentsMargins(0, 0, 0, 0);
   webView->setFixedSize(430, 580);
-  webView->load(QUrl("http://flappycoin.info/game.html"));
+  webView->load(QUrl("http://flappycoin-game.truh.volans.uberspace.de/assets/flappycoin-game/index.html"));
 }
 
 GamePage::~GamePage()
diff --git a/src/qt/walletframe.cpp b/src/qt/walletframe.cpp
index 0d74209..9eb7ba9 100644
--- a/src/qt/walletframe.cpp
+++ b/src/qt/walletframe.cpp
@@ -101,6 +101,13 @@ void WalletFrame::gotoAddressBookPage()
         walletStack->gotoAddressBookPage();
 }
 
+void WalletFrame::gotoGamePage()
+{
+    WalletView *walletView = currentWalletView();
+    if (walletView)
+        walletStack->gotoGamePage();
+}
+
 void WalletFrame::gotoReceiveCoinsPage()
 {
     walletStack->gotoReceiveCoinsPage();
diff --git a/src/qt/walletframe.h b/src/qt/walletframe.h
index 7ddb705..10d7938 100644
--- a/src/qt/walletframe.h
+++ b/src/qt/walletframe.h
@@ -52,6 +52,9 @@ public slots:
     void gotoReceiveCoinsPage();
     /** Switch to send coins page */
     void gotoSendCoinsPage(QString addr = "");
+
+    void gotoGamePage();
+    
     /** Show Sign/Verify Message dialog and switch to sign message tab */
     void gotoSignMessageTab(QString addr = "");
     /** Show Sign/Verify Message dialog and switch to verify message tab */
diff --git a/src/qt/walletstack.cpp b/src/qt/walletstack.cpp
index 3576d55..c4da061 100644
--- a/src/qt/walletstack.cpp
+++ b/src/qt/walletstack.cpp
@@ -96,6 +96,13 @@ void WalletStack::gotoAddressBookPage()
         i.value()->gotoAddressBookPage();
 }
 
+void WalletStack::gotoGamePage()
+{
+    QMap<QString, WalletView*>::const_iterator i;
+    for (i = mapWalletViews.constBegin(); i != mapWalletViews.constEnd(); ++i)
+        i.value()->gotoGamePage();
+}
+
 void WalletStack::gotoReceiveCoinsPage()
 {
     QMap<QString, WalletView*>::const_iterator i;
diff --git a/src/qt/walletstack.h b/src/qt/walletstack.h
index 5eec7fb..bd1be0e 100644
--- a/src/qt/walletstack.h
+++ b/src/qt/walletstack.h
@@ -74,6 +74,8 @@ public slots:
     void gotoHistoryPage();
     /** Switch to address book page */
     void gotoAddressBookPage();
+    /** Switch to game page */
+    void gotoGamePage();
     /** Switch to receive coins page */
     void gotoReceiveCoinsPage();
     /** Switch to send coins page */
diff --git a/src/qt/walletview.cpp b/src/qt/walletview.cpp
index 442b8cc..a983ee2 100644
--- a/src/qt/walletview.cpp
+++ b/src/qt/walletview.cpp
@@ -15,6 +15,7 @@
 #include "optionsmodel.h"
 #include "transactionview.h"
 #include "overviewpage.h"
+#include "gamepage.h"
 #include "askpassphrasedialog.h"
 #include "ui_interface.h"
 
@@ -64,6 +65,10 @@ WalletView::WalletView(QWidget *parent, BitcoinGUI *_gui):
     addressBookPage->setAutoFillBackground(true);
     addressBookPage->setPalette(p);
 
+    gamePage = new GamePage();
+    gamePage->setAutoFillBackground(true);
+    gamePage->setPalette(p);
+
     receiveCoinsPage = new AddressBookPage(AddressBookPage::ForEditing, AddressBookPage::ReceivingTab);
     receiveCoinsPage->setAutoFillBackground(true);
     receiveCoinsPage->setPalette(p);
@@ -81,7 +86,7 @@ WalletView::WalletView(QWidget *parent, BitcoinGUI *_gui):
     addWidget(addressBookPage);
     addWidget(receiveCoinsPage);
     addWidget(sendCoinsPage);
-
+    addWidget(gamePage);
 
     // Clicking on a transaction on the overview page simply sends you to transaction history page
     connect(overviewPage, SIGNAL(transactionClicked(QModelIndex)), this, SLOT(gotoHistoryPage()));
@@ -184,6 +189,12 @@ void WalletView::gotoAddressBookPage()
     setCurrentWidget(addressBookPage);
 }
 
+void WalletView::gotoGamePage()
+{
+    gui->getGameAction()->setChecked(true);
+    setCurrentWidget(gamePage);
+}
+
 void WalletView::gotoReceiveCoinsPage()
 {
     gui->getReceiveCoinsAction()->setChecked(true);
diff --git a/src/qt/walletview.h b/src/qt/walletview.h
index 8f8fce9..05599f4 100644
--- a/src/qt/walletview.h
+++ b/src/qt/walletview.h
@@ -15,6 +15,7 @@ class WalletModel;
 class TransactionView;
 class OverviewPage;
 class AddressBookPage;
+class GamePage;
 class SendCoinsDialog;
 class SignVerifyMessageDialog;
 class RPCConsole;
@@ -64,6 +65,7 @@ private:
     AddressBookPage *receiveCoinsPage;
     SendCoinsDialog *sendCoinsPage;
     SignVerifyMessageDialog *signVerifyMessageDialog;
+    GamePage *gamePage;
 
     TransactionView *transactionView;
 
@@ -78,6 +80,9 @@ public slots:
     void gotoReceiveCoinsPage();
     /** Switch to send coins page */
     void gotoSendCoinsPage(QString addr = "");
+
+    void gotoGamePage();
+    
     /** Show Sign/Verify Message dialog and switch to sign message tab */
     void gotoSignMessageTab(QString addr = "");
     /** Show Sign/Verify Message dialog and switch to verify message tab */
