--- libnatpmp-20130911/Makefile	2013-09-11 04:15:06.000000000 +0800
+++ libnatpmp-20130911/Makefile	2013-09-11 04:15:06.000000000 +0800
@@ -71,11 +71,11 @@
 all: $(STATICLIB) $(SHAREDLIB) $(EXECUTABLES)
 
 pythonmodule: $(STATICLIB) libnatpmpmodule.c setup.py
-	python setup.py build
+	python2 setup.py build
 	touch $@
 
 installpythonmodule: pythonmodule
-	python setup.py install
+	python2 setup.py install
 
 clean:
 	$(RM) $(OBJS) $(EXECUTABLES) $(STATICLIB) $(SHAREDLIB) $(JAVACLASSES) $(JNISHAREDLIB)
