diff -ur OpenCV-2.1.0/src/highgui/cvcap_dshow.cpp OpenCV-2.1.02/src/highgui/cvcap_dshow.cpp
--- OpenCV-2.1.0/src/highgui/cvcap_dshow.cpp	2009-11-18 13:33:53.000000000 +0100
+++ OpenCV-2.1.02/src/highgui/cvcap_dshow.cpp	2010-10-16 22:58:16.133010410 +0200
@@ -42,7 +42,7 @@
 #include "_highgui.h"
 
 #ifdef HAVE_VIDEOINPUT
-#include "videoinput.h"
+#include "videoInput.h"
 
 /********************* Capturing video from camera via VFW *********************/
 
