diff -ur grass-6.4.2/doc/python/example_ctypes.py grass-6.4.2-new/doc/python/example_ctypes.py
--- grass-6.4.2/doc/python/example_ctypes.py	2010-04-17 14:15:26.000000000 +0200
+++ grass-6.4.2-new/doc/python/example_ctypes.py	2012-08-16 09:33:01.480497320 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import os, sys, subprocess
 from ctypes import *
 grass = CDLL("libgrass_gis.so")
diff -ur grass-6.4.2/general/g.parser/test.py grass-6.4.2-new/general/g.parser/test.py
--- grass-6.4.2/general/g.parser/test.py	2009-07-23 23:08:41.000000000 +0200
+++ grass-6.4.2-new/general/g.parser/test.py	2012-08-16 09:33:01.513831191 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # g.parser demo script for python programing
 
diff -ur grass-6.4.2/gui/scripts/g.change.gui.py grass-6.4.2-new/gui/scripts/g.change.gui.py
--- grass-6.4.2/gui/scripts/g.change.gui.py	2009-05-30 07:36:32.000000000 +0200
+++ grass-6.4.2-new/gui/scripts/g.change.gui.py	2012-08-16 09:33:04.353875090 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ############################################################################
 #
diff -ur grass-6.4.2/gui/scripts/g.extension.py grass-6.4.2-new/gui/scripts/g.extension.py
--- grass-6.4.2/gui/scripts/g.extension.py	2012-02-18 18:28:29.000000000 +0100
+++ grass-6.4.2-new/gui/scripts/g.extension.py	2012-08-16 09:33:04.393875703 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ############################################################################
 #
diff -ur grass-6.4.2/gui/scripts/g.extension.rebuild.all.py grass-6.4.2-new/gui/scripts/g.extension.rebuild.all.py
--- grass-6.4.2/gui/scripts/g.extension.rebuild.all.py	2012-02-01 10:26:11.000000000 +0100
+++ grass-6.4.2-new/gui/scripts/g.extension.rebuild.all.py	2012-08-16 09:33:04.320541242 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 ############################################################################
 #
diff -ur grass-6.4.2/gui/scripts/v.type_wrapper.py grass-6.4.2-new/gui/scripts/v.type_wrapper.py
--- grass-6.4.2/gui/scripts/v.type_wrapper.py	2009-05-30 07:36:32.000000000 +0200
+++ grass-6.4.2-new/gui/scripts/v.type_wrapper.py	2012-08-16 09:33:04.287207393 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ############################################################################
 #
 # MODULE:       v.type_wrapper.py (v.type wrapper script)
diff -ur grass-6.4.2/gui/wxpython/gui_modules/menuform.py grass-6.4.2-new/gui/wxpython/gui_modules/menuform.py
--- grass-6.4.2/gui/wxpython/gui_modules/menuform.py	2012-01-04 10:55:50.000000000 +0100
+++ grass-6.4.2-new/gui/wxpython/gui_modules/menuform.py	2012-08-16 09:33:03.000520822 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 @brief Construct simple wx.Python GUI from a GRASS command interface
 description.
diff -ur grass-6.4.2/gui/wxpython/scripts/d.rast3d.py grass-6.4.2-new/gui/wxpython/scripts/d.rast3d.py
--- grass-6.4.2/gui/wxpython/scripts/d.rast3d.py	2010-10-17 19:15:32.000000000 +0200
+++ grass-6.4.2-new/gui/wxpython/scripts/d.rast3d.py	2012-08-16 09:33:04.083870925 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 ############################################################################
 #
diff -ur grass-6.4.2/gui/wxpython/scripts/p.cmd.py grass-6.4.2-new/gui/wxpython/scripts/p.cmd.py
--- grass-6.4.2/gui/wxpython/scripts/p.cmd.py	2010-10-17 19:15:32.000000000 +0200
+++ grass-6.4.2-new/gui/wxpython/scripts/p.cmd.py	2012-08-16 09:33:03.870534267 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ############################################################################
 #
 # MODULE:       p.cmd
diff -ur grass-6.4.2/gui/wxpython/scripts/p.db.py grass-6.4.2-new/gui/wxpython/scripts/p.db.py
--- grass-6.4.2/gui/wxpython/scripts/p.db.py	2010-10-17 19:15:32.000000000 +0200
+++ grass-6.4.2-new/gui/wxpython/scripts/p.db.py	2012-08-16 09:33:04.047203667 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ############################################################################
 #
 # MODULE:       p.db
diff -ur grass-6.4.2/gui/wxpython/scripts/p.mon.py grass-6.4.2-new/gui/wxpython/scripts/p.mon.py
--- grass-6.4.2/gui/wxpython/scripts/p.mon.py	2010-10-17 19:15:32.000000000 +0200
+++ grass-6.4.2-new/gui/wxpython/scripts/p.mon.py	2012-08-16 09:33:03.940535364 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ############################################################################
 #
 # MODULE:       p.mon
diff -ur grass-6.4.2/gui/wxpython/scripts/p.rast.py grass-6.4.2-new/gui/wxpython/scripts/p.rast.py
--- grass-6.4.2/gui/wxpython/scripts/p.rast.py	2010-10-17 19:15:32.000000000 +0200
+++ grass-6.4.2-new/gui/wxpython/scripts/p.rast.py	2012-08-16 09:33:04.013869835 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ############################################################################
 #
 # MODULE:       p.rast
diff -ur grass-6.4.2/gui/wxpython/scripts/p.vect.py grass-6.4.2-new/gui/wxpython/scripts/p.vect.py
--- grass-6.4.2/gui/wxpython/scripts/p.vect.py	2010-10-17 19:15:32.000000000 +0200
+++ grass-6.4.2-new/gui/wxpython/scripts/p.vect.py	2012-08-16 09:33:03.980535999 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 ############################################################################
 #
 # MODULE:       p.vect
diff -ur grass-6.4.2/imagery/i.atcorr/create_iwave.py grass-6.4.2-new/imagery/i.atcorr/create_iwave.py
--- grass-6.4.2/imagery/i.atcorr/create_iwave.py	2010-12-22 12:09:24.000000000 +0100
+++ grass-6.4.2-new/imagery/i.atcorr/create_iwave.py	2012-08-16 09:33:04.430542943 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 Created on Sat Mar 27 11:35:32 2010
 
diff -ur grass-6.4.2/include/Make/Platform.make.in grass-6.4.2-new/include/Make/Platform.make.in
--- grass-6.4.2/include/Make/Platform.make.in	2010-11-08 08:23:48.000000000 +0100
+++ grass-6.4.2-new/include/Make/Platform.make.in	2012-08-16 09:33:35.484354613 +0200
@@ -217,7 +217,7 @@
 USE_CAIRO           = @USE_CAIRO@
 
 #Python
-PYTHON              = python
+PYTHON              = python2
 PYTHONINC           = @PYTHONINC@
 PYTHONCFLAGS        = @PYTHONCFLAGS@
 PYTHONLDFLAGS       = @PYTHONLDFLAGS@
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/ctypedescs.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/ctypedescs.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/ctypedescs.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/ctypedescs.py	2012-08-16 09:33:04.987218247 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 ctypesgencore.ctypedescs contains classes to represent a C type. All of them
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/descriptions.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/descriptions.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/descriptions.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/descriptions.py	2012-08-16 09:33:04.883883291 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 ctypesgencore.descriptions contains classes to represent a description of a
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/expressions.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/expressions.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/expressions.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/expressions.py	2012-08-16 09:33:05.023885450 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 The expressions module contains classes to represent an expression. The main
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/__init__.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/__init__.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/__init__.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/__init__.py	2012-08-16 09:33:04.920550527 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 Ctypesgencore is the module that contains the main body of ctypesgen - in fact,
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/messages.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/messages.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/messages.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/messages.py	2012-08-16 09:33:04.503877428 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 ctypesgencore.messages contains functions to display status, error, or warning
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/options.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/options.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/options.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/options.py	2012-08-16 09:33:04.707213885 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 All of the components of ctypegencore require an argument called "options".
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/cdeclarations.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/cdeclarations.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/cdeclarations.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/cdeclarations.py	2012-08-16 09:33:05.393891162 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 This file contains classes that represent C declarations. cparser produces
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/cgrammar.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/cgrammar.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/cgrammar.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/cgrammar.py	2012-08-16 09:33:05.433891816 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''This is a yacc grammar for C.
 
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/cparser.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/cparser.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/cparser.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/cparser.py	2012-08-16 09:33:05.213888390 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 Parse a C source file.
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/ctypesparser.py	2012-08-16 09:33:05.093886533 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''
 ctypesgencore.parser.ctypesparser contains a class, CtypesParser, which is a
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/datacollectingparser.py	2012-08-16 09:33:05.473892425 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 DataCollectingParser subclasses ctypesparser.CtypesParser and builds Description
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/__init__.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/__init__.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/__init__.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/__init__.py	2012-08-16 09:33:05.283889499 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 This package parses C header files and generates lists of functions, typedefs,
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/pplexer.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/pplexer.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/pplexer.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/pplexer.py	2012-08-16 09:33:05.180554533 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''Preprocess a C source file using gcc and convert the result into
    a token stream
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/preprocessor.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/preprocessor.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/parser/preprocessor.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/parser/preprocessor.py	2012-08-16 09:33:05.250555640 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 '''Preprocess a C source file using gcc and convert the result into
    a token stream
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/printer/__init__.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/printer/__init__.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/printer/__init__.py	2011-04-13 14:57:45.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/printer/__init__.py	2012-08-16 09:33:04.640546189 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 This module is the backend to ctypesgen; it contains classes to
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/printer/printer.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/printer/printer.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/printer/printer.py	2011-04-13 14:57:45.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/printer/printer.py	2012-08-16 09:33:04.537211279 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import os, sys, time
 from ctypesgencore.descriptions import *
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/processor/dependencies.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/processor/dependencies.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/processor/dependencies.py	2011-04-13 14:57:45.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/processor/dependencies.py	2012-08-16 09:33:04.743881125 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 The dependencies module determines which descriptions depend on which other
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/processor/__init__.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/processor/__init__.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/processor/__init__.py	2011-04-13 14:57:45.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/processor/__init__.py	2012-08-16 09:33:04.810548840 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 This module contains functions to operate on the DeclarationCollection produced
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/processor/operations.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/processor/operations.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/processor/operations.py	2011-04-13 14:57:45.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/processor/operations.py	2012-08-16 09:33:04.847216053 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 """
 The operations module contains various functions to process the
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgencore/processor/pipeline.py grass-6.4.2-new/lib/python/ctypes/ctypesgencore/processor/pipeline.py
--- grass-6.4.2/lib/python/ctypes/ctypesgencore/processor/pipeline.py	2011-04-13 14:57:45.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgencore/processor/pipeline.py	2012-08-16 09:33:04.777214964 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import ctypes, re, os
 from ctypesgencore.processor.operations import *
diff -ur grass-6.4.2/lib/python/ctypes/ctypesgen.py grass-6.4.2-new/lib/python/ctypes/ctypesgen.py
--- grass-6.4.2/lib/python/ctypes/ctypesgen.py	2011-04-13 14:57:46.000000000 +0200
+++ grass-6.4.2-new/lib/python/ctypes/ctypesgen.py	2012-08-16 09:33:05.573893972 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 def find_names_in_modules(modules):
     names = set()
