diff -ur include/Make/Platform.make.in include/Make/Platform.make.in-new
--- include/Make/Platform.make.in	2010-11-08 08:23:48.000000000 +0100
+++ include/Make/Platform.make.in-new	2012-08-16 09:33:35.484354613 +0200
@@ -217,7 +217,7 @@
 USE_CAIRO           = @USE_CAIRO@
 
 #Python
-PYTHON              = python
+PYTHON              = python2
 PYTHONINC           = @PYTHONINC@
 PYTHONCFLAGS        = @PYTHONCFLAGS@
 PYTHONLDFLAGS       = @PYTHONLDFLAGS@
