--- a/src/gfxoutputdrv/gifdrv.c
+++ b/src/gfxoutputdrv/gifdrv.c
@@ -107,7 +107,7 @@ static int gifdrv_open(screenshot_t *screenshot, const char *filename)
 
     if (EGifPutScreenDesc(sdata->fd, screenshot->width, screenshot->height, 8, 0, gif_colors) == GIF_ERROR ||
         EGifPutImageDesc(sdata->fd, 0, 0, screenshot->width, screenshot->height, 0, NULL) == GIF_ERROR) {
-        EGifCloseFile(sdata->fd);
+        EGifCloseFile(sdata->fd, NULL);
         VICE_FreeMapObject(gif_colors);
         lib_free(sdata->data);
         lib_free(sdata->ext_filename);
@@ -139,7 +139,7 @@ static int gifdrv_close(screenshot_t *screenshot)
 
     sdata = screenshot->gfxoutputdrv_data;
 
-    EGifCloseFile(sdata->fd);
+    EGifCloseFile(sdata->fd, NULL);
     VICE_FreeMapObject(gif_colors);
 
     /* for some reason giflib will create a file with unexpected
@@ -179,7 +179,7 @@ static char *gifdrv_memmap_ext_filename;
 
 static int gifdrv_close_memmap(void)
 {
-    EGifCloseFile(gifdrv_memmap_fd);
+    EGifCloseFile(gifdrv_memmap_fd, NULL);
     VICE_FreeMapObject(gif_colors);
     lib_free(gifdrv_memmap_ext_filename);
 
@@ -226,7 +226,7 @@ static int gifdrv_open_memmap(const char *filename, int x_size, int y_size, BYTE
 
     if (EGifPutScreenDesc(gifdrv_memmap_fd, x_size, y_size, 8, 0, gif_colors) == GIF_ERROR ||
         EGifPutImageDesc(gifdrv_memmap_fd, 0, 0, x_size, y_size, 0, NULL) == GIF_ERROR) {
-        EGifCloseFile(gifdrv_memmap_fd);
+        EGifCloseFile(gifdrv_memmap_fd, NULL);
         VICE_FreeMapObject(gif_colors);
         lib_free(gifdrv_memmap_ext_filename);
         return -1;
