*** MakefileORIG	2012-11-04 13:45:31.588078774 +0100
--- Makefile686	2012-11-04 13:45:45.391340767 +0100
***************
*** 38,44 ****
  TARGET_PLATFORM := $(shell uname -m)
  ifeq ($(TARGET_PLATFORM),i686)
  # Generic x86 (tune for Pentium III, since that's more common these days)
!   OPT_CFLAGS += -mtune=i686 -march=i386
  endif
  ifeq ($(TARGET_PLATFORM),amd64)
    OPT_CFLAGS += -march=k8 -msse -msse2 -mfpmath=sse
--- 38,44 ----
  TARGET_PLATFORM := $(shell uname -m)
  ifeq ($(TARGET_PLATFORM),i686)
  # Generic x86 (tune for Pentium III, since that's more common these days)
!   #OPT_CFLAGS += -mtune=i686 -march=i386
  endif
  ifeq ($(TARGET_PLATFORM),amd64)
    OPT_CFLAGS += -march=k8 -msse -msse2 -mfpmath=sse
