--- renpy-6.13.9/launcher/jEdit.edit.py 2012-03-11 08:19:50.000000000 +1100
+++ renpy-6.13.9.new/launcher/jEdit.edit.py     2012-03-16 06:06:59.754291055 +1100
@@ -9,7 +9,7 @@
         args = [ ]
         
         mydir = os.path.dirname(__file__)
-        jar = os.path.join(mydir, "../jedit/jedit.jar")
+        jar = os.path.join("/usr/share/java/jedit/jedit.jar")
         
         # My Java does not like having non-ASCII characters in jar paths. 
         # Using relpath won't guarantee those characters won't exist - but it
