--- lib/mc.ext.in
+++ lib/mc.ext.in
@@ -115,7 +115,7 @@
 	# Open=%cd %p#utar
 	View=%view{ascii} bzip -dc %f 2>/dev/null | tar tvvf -
 
-regex/\.t(ar\.bz2|bz|b2)$
+regex/\.t(ar\.bz2|bz|b2|bz2)$
 	Open=%cd %p#utar
 	View=%view{ascii} bzip2 -dc %f 2>/dev/null | tar tvvf -
 
@@ -173,7 +173,7 @@
 	View=%view{ascii} cpio -itv <'%f' 2>/dev/null
 
 # ls-lR
-regex/(^|\.)ls-?lR(\.g?z|Z|bz2)?$
+regex/(^|\.)ls-?lR(\.(bz2|gz|Z))?$
 	Open=%cd %p#lslR
 
 # patch
@@ -339,7 +339,7 @@
 	View=sxpm %f
 
 include/image
-	Open=if [ "$DISPLAY" = "" ]; then zgv %f; else (gqview %f &); fi
+	Open=if [ "$DISPLAY" = "" ]; then asciiview %f; else (display %f >/dev/null 2>&1 &); fi
 	View=%view{ascii} identify %f
 	#View=%view{ascii} asciiview %f
 
@@ -357,11 +357,11 @@
        Open=vplay -s 22 %f
 
 regex/\.([mM][pP]3)$
-	Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else (xmms %f >/dev/null 2>&1 &); fi
+	Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else if [ -z "`which realplay`" ]; then (xmms -e %f 1>/dev/null 2>&1 &); else (realplay %f >/dev/null 2>&1 &); fi; fi
 	View=%view{ascii} mpg123 -vtn1 %f 2>&1 | sed -n '/^Title/,/^Comment/p;/^MPEG/,/^Audio/p'
 
 regex/\.([oO][gG][gG])$
-	Open=if [ "$DISPLAY" = "" ]; then ogg123 %f; else (xmms %f >/dev/null 2>&1 &); fi
+	Open=if [ "$DISPLAY" = "" ]; then ogg123 %f; else (xmms -e %f >/dev/null 2>&1 &); fi
 	View=%view{ascii} ogginfo %s
 
 regex/\.([mM][iI][dD][iI]?|[rR][mM][iI][dD]?)$
@@ -417,12 +417,12 @@
 
 # Postscript
 type/^PostScript
-	Open=(gv %f &)
+	Open=(gv %f >/dev/null 2>&1 &)
 	View=%view{ascii} ps2ascii %f
 
 # PDF
 type/^PDF
-	Open=(xpdf %f &)
+	Open=(xpdf %f >/dev/null 2>&1 &)
 	#Open=(acroread %f &)
 	#Open=(ghostview %f &)
 	View=%view{ascii} pdftotext %f -
@@ -437,54 +437,53 @@
 
 # StarOffice 5.2
 shell/.sdw
-	Open=(ooffice %f &)
+	Open=(ooffice %f >/dev/null 2>&1 &)
 
 # StarOffice 6 and OpenOffice.org formats
-regex/\.(odt|ott|sxw|stw|ods|ots|sxc|stc|odp|otp|sxi|sti|odg|otg|sxd|std|odb|odf|sxm|odm|sxg)$
-	Open=(ooffice %f &)
+regex/\.(odb|odc|odf|odg|odi|odm|odp|ods|odt|otg|oth|otp|ots|ott|sda|sdc|sdd|sdp|sds|sdw|sgl|smf|stc|std|sti|stw|sxc|sxd|sxg|sxi|sxm|sxw|vor)$
+	Open=(ooffice %f >/dev/null 2>&1 &)
 	View=%view{ascii} unzip -p %f content.xml | o3totxt
 
 # AbiWord
 shell/.abw
-	Open=(abiword %f &)
+	Open=if which abiword ; then (abiword %f >/dev/null 2>&1 &); else (ooffice %f >/dev/null 2>&1 &); fi >/dev/null
 
 # Microsoft Word Document
 regex/\.([Dd][oO][cCtT]|[Ww][rR][iI])$
-	Open=(abiword %f >/dev/null 2>&1 &)
+	Open=if which abiword ; then (abiword %f >/dev/null 2>&1 &); else (ooffice %f >/dev/null 2>&1 &); fi >/dev/null
 	View=%view{ascii} catdoc -w %f || word2x -f text %f - || strings %f
 type/^Microsoft\ Word
-	Open=(abiword %f >/dev/null 2>&1 &)
+	Open=if which abiword ; then (abiword %f >/dev/null 2>&1 &); else (ooffice %f >/dev/null 2>&1 &); fi >/dev/null
 	View=%view{ascii} catdoc -w %f || word2x -f text %f - || strings %f
 
 # RTF document
 regex/\.([rR][tT][fF])$
-	Open=(abiword %f >/dev/null 2>&1 &)
+	Open=if which abiword ; then (abiword %f >/dev/null 2>&1 &); else (ooffice %f >/dev/null 2>&1 &); fi >/dev/null
 
 # Microsoft Excel Worksheet
 regex/\.([xX][lL][sSwW])$
-	Open=(gnumeric %f >/dev/null 2>&1 &)
+	Open=if which gnumeric ; then (gnumeric %f >/dev/null 2>&1 &); else (ooffice %f >/dev/null 2>&1 &); fi >/dev/null
 	View=%view{ascii} xls2csv %f || strings %f
 type/^Microsoft\ Excel
-	Open=(gnumeric %f >/dev/null 2>&1 &)
+	Open=if which gnumeric ; then (gnumeric %f >/dev/null 2>&1 &); else (ooffice %f >/dev/null 2>&1 &); fi >/dev/null
 	View=%view{ascii} xls2csv %f || strings %f
 
-# Use OpenOffice.org to open any MS Office documents
-type/^Microsoft\ Office\ Document
-	Open=(ooffice %f &)
-
 # Framemaker
 type/^FrameMaker
 	Open=fmclient -f %f
 
 # DVI
 regex/\.([dD][vV][iI])$
-	Open=if [ x$DISPLAY = x ]; then dvisvga %f; else (xdvi %f &); fi
+	Open=if [ x$DISPLAY = x ]; then dvisvga %f; else (xdvi %f >/dev/null 2>&1 &); fi
 	View=%view{ascii} dvi2tty %f
 
 # TeX
 regex/\.([Tt][Ee][Xx])$
 	Open=%var{EDITOR:vi} %f
 
+# DJVU
+regex/\.(djvu|DJVU)$
+       Open=djview %f >/dev/null 2>&1 &
 
 ### Miscellaneous ###
 
