--- lib/mc.ext.in
+++ lib/mc.ext.in
@@ -449,12 +449,17 @@
 	Open=if which abiword ; then (abiword %f >/dev/null 2>&1 &); else (ooffice %f >/dev/null 2>&1 &); fi >/dev/null
 
 # Microsoft Word Document
-regex/\.([Dd][oO][cCtT]|[Ww][rR][iI])$
+type/^Microsoft\ Office\ Document
 	Open=if which abiword ; then (abiword %f >/dev/null 2>&1 &); else (ooffice %f >/dev/null 2>&1 &); fi >/dev/null
-	View=%view{ascii} catdoc -w %f || word2x -f text %f - || strings %f
+        View=%view{ascii} wvText %f - || strings %f
+
 type/^Microsoft\ Word
 	Open=if which abiword ; then (abiword %f >/dev/null 2>&1 &); else (ooffice %f >/dev/null 2>&1 &); fi >/dev/null
-	View=%view{ascii} catdoc -w %f || word2x -f text %f - || strings %f
+        View=%view{ascii} wvText %f - || strings %f
+
+regex/\.([Dd]ot|DOT|[Ww]ri|WRI)$
+	Open=if which abiword ; then (abiword %f >/dev/null 2>&1 &); else (OOo %f >/dev/null 2>&1 &); fi >/dev/null
+        View=%view{ascii} wvText %f - || strings %f
 
 # RTF document
 regex/\.([rR][tT][fF])$
