--- lib/mc.ext.in
+++ lib/mc.ext.in
@@ -432,8 +432,8 @@
 
 # html
 regex/\.([hH][tT][mM][lL]?)$
-	Open=(if test -n "@X11_WWW@" && test -n "$DISPLAY"; then (@X11_WWW@ file://%d/%p &) 1>&2; else links %f || lynx -force_html %f || ${PAGER:-more} %f; fi) 2>/dev/null
-	View=%view{ascii} lynx -dump -force_html %f
+        Open=/usr/share/mc/bin/x11_browser %f
+        View=%view{ascii} w3m -dump -T text/html %f;
 
 # StarOffice 5.2
 shell/.sdw
