diff --git a/old/node/Makefile b/new/node/Makefile
index d02a64a..6c5ced2 100644
--- a/old/node/Makefile
+++ b/new/node/Makefile
@@ -1,7 +1,7 @@
 -include config.mk
 
 BUILDTYPE ?= Release
-PYTHON ?= python
+PYTHON ?= python2
 DESTDIR ?=
 SIGN ?=
 
diff --git a/old/node/configure b/new/node/configure
index 7211682..6a1e3ea 100755
--- a/old/node/configure
+++ b/new/node/configure
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 import optparse
 import os
 import pprint
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/buildbot/buildbot_run.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/buildbot/buildbot_run.py
index e0fda36..0f9d11c 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/buildbot/buildbot_run.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/buildbot/buildbot_run.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/gyptest.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/gyptest.py
index d9c814f..c9f7d25 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/gyptest.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/gyptest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/MSVSSettings_test.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/MSVSSettings_test.py
index 49e8a1d..a34ca73 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/MSVSSettings_test.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/MSVSSettings_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/__init__.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/__init__.py
index 1402713..a52d4a3 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/__init__.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/__init__.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/easy_xml_test.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/easy_xml_test.py
index a2aa4f2..4ca5bcd 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/easy_xml_test.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/easy_xml_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/generator/msvs_test.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/generator/msvs_test.py
index 5a69c1c..9aec97d 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/generator/msvs_test.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/generator/msvs_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/generator/ninja_test.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/generator/ninja_test.py
index af2e6d3..fe2c661 100644
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/generator/ninja_test.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/generator/ninja_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/mac_tool.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/mac_tool.py
index 42dd6c9..b6f9adb 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/mac_tool.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/mac_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/sun_tool.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/sun_tool.py
index 90d59c8..5b1956f 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/sun_tool.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/sun_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/system_test.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/system_test.py
index 51c71e3..ea68283 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/system_test.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/pylib/gyp/system_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/setup.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/setup.py
index ed2b41a..3e34ac3 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/setup.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2009 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/graphviz.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/graphviz.py
index 326ae22..553a101 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/graphviz.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/graphviz.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_gyp.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_gyp.py
index 3749792..8582c60 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_gyp.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_gyp.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright (c) 2011 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_sln.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_sln.py
index 7013f2b..2319808 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_sln.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_sln.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2009 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_vcproj.py b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_vcproj.py
index ba65bf2..aacadfa 100755
--- a/old/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_vcproj.py
+++ b/new/node/deps/npm/node_modules/node-gyp/legacy/tools/gyp/tools/pretty_vcproj.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2009 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/deps/v8/tools/disasm.py b/new/node/deps/v8/tools/disasm.py
index 681b425..63c782c 100644
--- a/old/node/deps/v8/tools/disasm.py
+++ b/new/node/deps/v8/tools/disasm.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2011 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/gc-nvp-trace-processor.py b/new/node/deps/v8/tools/gc-nvp-trace-processor.py
index fe5a7f3..daff955 100755
--- a/old/node/deps/v8/tools/gc-nvp-trace-processor.py
+++ b/new/node/deps/v8/tools/gc-nvp-trace-processor.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2010 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/gen-postmortem-metadata.py b/new/node/deps/v8/tools/gen-postmortem-metadata.py
index f59cfd3..2c7acc5 100644
--- a/old/node/deps/v8/tools/gen-postmortem-metadata.py
+++ b/new/node/deps/v8/tools/gen-postmortem-metadata.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #
 # Copyright 2012 the V8 project authors. All rights reserved.
diff --git a/old/node/deps/v8/tools/grokdump.py b/new/node/deps/v8/tools/grokdump.py
index a9f0cb9..6340058 100755
--- a/old/node/deps/v8/tools/grokdump.py
+++ b/new/node/deps/v8/tools/grokdump.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2012 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/gyp/v8.gyp b/new/node/deps/v8/tools/gyp/v8.gyp
index ea82d31..9d87f26 100644
--- a/old/node/deps/v8/tools/gyp/v8.gyp
+++ b/new/node/deps/v8/tools/gyp/v8.gyp
@@ -783,7 +783,7 @@
                 '<(SHARED_INTERMEDIATE_DIR)/libraries.cc',
               ],
               'action': [
-                'python',
+                'python2',
                 '../../tools/js2c.py',
                 '<@(_outputs)',
                 'CORE',
@@ -801,7 +801,7 @@
                 '<(SHARED_INTERMEDIATE_DIR)/experimental-libraries.cc',
               ],
               'action': [
-                'python',
+                'python2',
                 '../../tools/js2c.py',
                 '<@(_outputs)',
                 'EXPERIMENTAL',
@@ -831,7 +831,7 @@
                   '<(SHARED_INTERMEDIATE_DIR)/debug-support.cc',
                 ],
                 'action': [
-                  'python',
+                  'python2',
                   '../../tools/gen-postmortem-metadata.py',
                   '<@(_outputs)',
                   '<@(heapobject_files)'
diff --git a/old/node/deps/v8/tools/js2c.py b/new/node/deps/v8/tools/js2c.py
index d06cbe4..fb36645 100644
--- a/old/node/deps/v8/tools/js2c.py
+++ b/new/node/deps/v8/tools/js2c.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2012 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/ll_prof.py b/new/node/deps/v8/tools/ll_prof.py
index 51ba672..1d31690 100755
--- a/old/node/deps/v8/tools/ll_prof.py
+++ b/new/node/deps/v8/tools/ll_prof.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2012 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/presubmit.py b/new/node/deps/v8/tools/presubmit.py
index a0b81e8..5137b56 100755
--- a/old/node/deps/v8/tools/presubmit.py
+++ b/new/node/deps/v8/tools/presubmit.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2012 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/process-heap-prof.py b/new/node/deps/v8/tools/process-heap-prof.py
index a26cbf1..c7a4e6e 100755
--- a/old/node/deps/v8/tools/process-heap-prof.py
+++ b/new/node/deps/v8/tools/process-heap-prof.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2009 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/run-valgrind.py b/new/node/deps/v8/tools/run-valgrind.py
index 49c1b70..067c326 100755
--- a/old/node/deps/v8/tools/run-valgrind.py
+++ b/new/node/deps/v8/tools/run-valgrind.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 #
 # Copyright 2009 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/stats-viewer.py b/new/node/deps/v8/tools/stats-viewer.py
index ab8e287..702151d 100755
--- a/old/node/deps/v8/tools/stats-viewer.py
+++ b/new/node/deps/v8/tools/stats-viewer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/test-wrapper-gypbuild.py b/new/node/deps/v8/tools/test-wrapper-gypbuild.py
index d99d055..071cee6 100755
--- a/old/node/deps/v8/tools/test-wrapper-gypbuild.py
+++ b/new/node/deps/v8/tools/test-wrapper-gypbuild.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2012 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/deps/v8/tools/test.py b/new/node/deps/v8/tools/test.py
index 5131ad7..9ce6648 100755
--- a/old/node/deps/v8/tools/test.py
+++ b/new/node/deps/v8/tools/test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2012 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
@@ -1432,7 +1432,7 @@ def Main():
   # command flag with a command that uses the run-valgrind.py script.
   if options.valgrind:
     run_valgrind = join(workspace, "tools", "run-valgrind.py")
-    options.special_command = "python -u " + run_valgrind + " @"
+    options.special_command = "python2 -u " + run_valgrind + " @"
 
   if options.build_system == 'gyp':
     SUFFIX['debug'] = ''
diff --git a/old/node/node.gyp b/new/node/node.gyp
index 82af2fe..9bdd8c7 100644
--- a/old/node/node.gyp
+++ b/new/node/node.gyp
@@ -285,14 +285,14 @@
           'conditions': [
             [ 'node_use_dtrace=="true" or node_use_etw=="true"', {
               'action': [
-                'python',
+                'python2',
                 'tools/js2c.py',
                 '<@(_outputs)',
                 '<@(_inputs)',
               ],
             }, { # No Dtrace
               'action': [
-                'python',
+                'python2',
                 'tools/js2c.py',
                 '<@(_outputs)',
                 '<@(_inputs)',
diff --git a/old/node/tools/closure_linter/closure_linter/__init__.py b/new/node/tools/closure_linter/closure_linter/__init__.py
index 4265cc3..18ff536 100755
--- a/old/node/tools/closure_linter/closure_linter/__init__.py
+++ b/new/node/tools/closure_linter/closure_linter/__init__.py
@@ -1 +1 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
diff --git a/old/node/tools/closure_linter/closure_linter/checker.py b/new/node/tools/closure_linter/closure_linter/checker.py
index 4cdac93..ba76df2 100755
--- a/old/node/tools/closure_linter/closure_linter/checker.py
+++ b/new/node/tools/closure_linter/closure_linter/checker.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/checkerbase.py b/new/node/tools/closure_linter/closure_linter/checkerbase.py
index 123cb72..f59f0bb 100755
--- a/old/node/tools/closure_linter/closure_linter/checkerbase.py
+++ b/new/node/tools/closure_linter/closure_linter/checkerbase.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/__init__.py b/new/node/tools/closure_linter/closure_linter/common/__init__.py
index 4265cc3..18ff536 100755
--- a/old/node/tools/closure_linter/closure_linter/common/__init__.py
+++ b/new/node/tools/closure_linter/closure_linter/common/__init__.py
@@ -1 +1 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
diff --git a/old/node/tools/closure_linter/closure_linter/common/error.py b/new/node/tools/closure_linter/closure_linter/common/error.py
index 0e3b476..83e9bcf 100755
--- a/old/node/tools/closure_linter/closure_linter/common/error.py
+++ b/new/node/tools/closure_linter/closure_linter/common/error.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/erroraccumulator.py b/new/node/tools/closure_linter/closure_linter/common/erroraccumulator.py
index 7bb0c97..3b1ec7a 100755
--- a/old/node/tools/closure_linter/closure_linter/common/erroraccumulator.py
+++ b/new/node/tools/closure_linter/closure_linter/common/erroraccumulator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/errorhandler.py b/new/node/tools/closure_linter/closure_linter/common/errorhandler.py
index 764d54d..8ee08fa 100755
--- a/old/node/tools/closure_linter/closure_linter/common/errorhandler.py
+++ b/new/node/tools/closure_linter/closure_linter/common/errorhandler.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/errorprinter.py b/new/node/tools/closure_linter/closure_linter/common/errorprinter.py
index c975406..ed8eaaf 100755
--- a/old/node/tools/closure_linter/closure_linter/common/errorprinter.py
+++ b/new/node/tools/closure_linter/closure_linter/common/errorprinter.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/filetestcase.py b/new/node/tools/closure_linter/closure_linter/common/filetestcase.py
index ae4b883..245436d 100755
--- a/old/node/tools/closure_linter/closure_linter/common/filetestcase.py
+++ b/new/node/tools/closure_linter/closure_linter/common/filetestcase.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/htmlutil.py b/new/node/tools/closure_linter/closure_linter/common/htmlutil.py
index 26d44c5..c8a215c 100755
--- a/old/node/tools/closure_linter/closure_linter/common/htmlutil.py
+++ b/new/node/tools/closure_linter/closure_linter/common/htmlutil.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/lintrunner.py b/new/node/tools/closure_linter/closure_linter/common/lintrunner.py
index 07842c7..eb40193 100755
--- a/old/node/tools/closure_linter/closure_linter/common/lintrunner.py
+++ b/new/node/tools/closure_linter/closure_linter/common/lintrunner.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/matcher.py b/new/node/tools/closure_linter/closure_linter/common/matcher.py
index 9b4402c..182a120 100755
--- a/old/node/tools/closure_linter/closure_linter/common/matcher.py
+++ b/new/node/tools/closure_linter/closure_linter/common/matcher.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/position.py b/new/node/tools/closure_linter/closure_linter/common/position.py
index cebf17e..fef9ca7 100755
--- a/old/node/tools/closure_linter/closure_linter/common/position.py
+++ b/new/node/tools/closure_linter/closure_linter/common/position.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/simplefileflags.py b/new/node/tools/closure_linter/closure_linter/common/simplefileflags.py
index 3402bef..844241c 100755
--- a/old/node/tools/closure_linter/closure_linter/common/simplefileflags.py
+++ b/new/node/tools/closure_linter/closure_linter/common/simplefileflags.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/tokenizer.py b/new/node/tools/closure_linter/closure_linter/common/tokenizer.py
index 0234720..223d22b 100755
--- a/old/node/tools/closure_linter/closure_linter/common/tokenizer.py
+++ b/new/node/tools/closure_linter/closure_linter/common/tokenizer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/common/tokens.py b/new/node/tools/closure_linter/closure_linter/common/tokens.py
index 5eaffa8..95ab42f 100755
--- a/old/node/tools/closure_linter/closure_linter/common/tokens.py
+++ b/new/node/tools/closure_linter/closure_linter/common/tokens.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/ecmalintrules.py b/new/node/tools/closure_linter/closure_linter/ecmalintrules.py
index a971b44..901ebe8 100755
--- a/old/node/tools/closure_linter/closure_linter/ecmalintrules.py
+++ b/new/node/tools/closure_linter/closure_linter/ecmalintrules.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/ecmametadatapass.py b/new/node/tools/closure_linter/closure_linter/ecmametadatapass.py
index 2c797b3..8fc06a5 100755
--- a/old/node/tools/closure_linter/closure_linter/ecmametadatapass.py
+++ b/new/node/tools/closure_linter/closure_linter/ecmametadatapass.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2010 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/error_fixer.py b/new/node/tools/closure_linter/closure_linter/error_fixer.py
index 904cf86..baeb278 100755
--- a/old/node/tools/closure_linter/closure_linter/error_fixer.py
+++ b/new/node/tools/closure_linter/closure_linter/error_fixer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/errorrules.py b/new/node/tools/closure_linter/closure_linter/errorrules.py
index afb6fa9..dad0aef 100755
--- a/old/node/tools/closure_linter/closure_linter/errorrules.py
+++ b/new/node/tools/closure_linter/closure_linter/errorrules.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2010 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/errors.py b/new/node/tools/closure_linter/closure_linter/errors.py
index 7c86941..cfadaa8 100755
--- a/old/node/tools/closure_linter/closure_linter/errors.py
+++ b/new/node/tools/closure_linter/closure_linter/errors.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/fixjsstyle.py b/new/node/tools/closure_linter/closure_linter/fixjsstyle.py
index 8782e64..ba89be2 100755
--- a/old/node/tools/closure_linter/closure_linter/fixjsstyle.py
+++ b/new/node/tools/closure_linter/closure_linter/fixjsstyle.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/fixjsstyle_test.py b/new/node/tools/closure_linter/closure_linter/fixjsstyle_test.py
index 42e9c59..2d1e9ff 100755
--- a/old/node/tools/closure_linter/closure_linter/fixjsstyle_test.py
+++ b/new/node/tools/closure_linter/closure_linter/fixjsstyle_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/full_test.py b/new/node/tools/closure_linter/closure_linter/full_test.py
index f11f235..68ef48a 100755
--- a/old/node/tools/closure_linter/closure_linter/full_test.py
+++ b/new/node/tools/closure_linter/closure_linter/full_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/gjslint.py b/new/node/tools/closure_linter/closure_linter/gjslint.py
index e33bddd..b807ea7 100755
--- a/old/node/tools/closure_linter/closure_linter/gjslint.py
+++ b/new/node/tools/closure_linter/closure_linter/gjslint.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/indentation.py b/new/node/tools/closure_linter/closure_linter/indentation.py
index d740607..d162ea7 100755
--- a/old/node/tools/closure_linter/closure_linter/indentation.py
+++ b/new/node/tools/closure_linter/closure_linter/indentation.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2010 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/javascriptlintrules.py b/new/node/tools/closure_linter/closure_linter/javascriptlintrules.py
index 6b9f1be..045b9c4 100755
--- a/old/node/tools/closure_linter/closure_linter/javascriptlintrules.py
+++ b/new/node/tools/closure_linter/closure_linter/javascriptlintrules.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/javascriptstatetracker.py b/new/node/tools/closure_linter/closure_linter/javascriptstatetracker.py
index 9cce376..bd39000 100755
--- a/old/node/tools/closure_linter/closure_linter/javascriptstatetracker.py
+++ b/new/node/tools/closure_linter/closure_linter/javascriptstatetracker.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/javascriptstatetracker_test.py b/new/node/tools/closure_linter/closure_linter/javascriptstatetracker_test.py
index e4288b7..300e0ac 100755
--- a/old/node/tools/closure_linter/closure_linter/javascriptstatetracker_test.py
+++ b/new/node/tools/closure_linter/closure_linter/javascriptstatetracker_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2010 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/javascripttokenizer.py b/new/node/tools/closure_linter/closure_linter/javascripttokenizer.py
index 097d3fd..ff0080e 100755
--- a/old/node/tools/closure_linter/closure_linter/javascripttokenizer.py
+++ b/new/node/tools/closure_linter/closure_linter/javascripttokenizer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/javascripttokens.py b/new/node/tools/closure_linter/closure_linter/javascripttokens.py
index f46d4e1..e7646a3 100755
--- a/old/node/tools/closure_linter/closure_linter/javascripttokens.py
+++ b/new/node/tools/closure_linter/closure_linter/javascripttokens.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/statetracker.py b/new/node/tools/closure_linter/closure_linter/statetracker.py
index 5630c17..2c3e1c5 100755
--- a/old/node/tools/closure_linter/closure_linter/statetracker.py
+++ b/new/node/tools/closure_linter/closure_linter/statetracker.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/closure_linter/tokenutil.py b/new/node/tools/closure_linter/closure_linter/tokenutil.py
index 6ed5f7f..5cce8b6 100755
--- a/old/node/tools/closure_linter/closure_linter/tokenutil.py
+++ b/new/node/tools/closure_linter/closure_linter/tokenutil.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2007 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/closure_linter/gflags.py b/new/node/tools/closure_linter/gflags.py
index 21aa88e..c970c0d 100644
--- a/old/node/tools/closure_linter/gflags.py
+++ b/new/node/tools/closure_linter/gflags.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2007, Google Inc.
 # All rights reserved.
diff --git a/old/node/tools/closure_linter/setup.py b/new/node/tools/closure_linter/setup.py
index 1d1764f..c829138 100755
--- a/old/node/tools/closure_linter/setup.py
+++ b/new/node/tools/closure_linter/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2010 The Closure Linter Authors. All Rights Reserved.
 #
diff --git a/old/node/tools/genv8constants.py b/new/node/tools/genv8constants.py
index efc6441..06a4216 100755
--- a/old/node/tools/genv8constants.py
+++ b/new/node/tools/genv8constants.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 #
 # genv8constants.py output_file libv8_base.a
diff --git a/old/node/tools/gyp/buildbot/buildbot_run.py b/new/node/tools/gyp/buildbot/buildbot_run.py
index 57fdb65..594125c 100755
--- a/old/node/tools/gyp/buildbot/buildbot_run.py
+++ b/new/node/tools/gyp/buildbot/buildbot_run.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/old/node/tools/gyp/gyptest.py b/new/node/tools/gyp/gyptest.py
index c84f3d3..5baf705 100755
--- a/old/node/tools/gyp/gyptest.py
+++ b/new/node/tools/gyp/gyptest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/pylib/gyp/MSVSSettings_test.py b/new/node/tools/gyp/pylib/gyp/MSVSSettings_test.py
index 4e06da3..abec190 100755
--- a/old/node/tools/gyp/pylib/gyp/MSVSSettings_test.py
+++ b/new/node/tools/gyp/pylib/gyp/MSVSSettings_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/pylib/gyp/__init__.py b/new/node/tools/gyp/pylib/gyp/__init__.py
index 54488e5..9f96223 100755
--- a/old/node/tools/gyp/pylib/gyp/__init__.py
+++ b/new/node/tools/gyp/pylib/gyp/__init__.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/pylib/gyp/common_test.py b/new/node/tools/gyp/pylib/gyp/common_test.py
index aabaf34..fd179ba 100755
--- a/old/node/tools/gyp/pylib/gyp/common_test.py
+++ b/new/node/tools/gyp/pylib/gyp/common_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/pylib/gyp/easy_xml_test.py b/new/node/tools/gyp/pylib/gyp/easy_xml_test.py
index df64354..c8b594f 100755
--- a/old/node/tools/gyp/pylib/gyp/easy_xml_test.py
+++ b/new/node/tools/gyp/pylib/gyp/easy_xml_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/pylib/gyp/generator/msvs_test.py b/new/node/tools/gyp/pylib/gyp/generator/msvs_test.py
index 5a69c1c..9aec97d 100755
--- a/old/node/tools/gyp/pylib/gyp/generator/msvs_test.py
+++ b/new/node/tools/gyp/pylib/gyp/generator/msvs_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/pylib/gyp/generator/ninja_test.py b/new/node/tools/gyp/pylib/gyp/generator/ninja_test.py
index 90dd153..1460eb9 100644
--- a/old/node/tools/gyp/pylib/gyp/generator/ninja_test.py
+++ b/new/node/tools/gyp/pylib/gyp/generator/ninja_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/pylib/gyp/mac_tool.py b/new/node/tools/gyp/pylib/gyp/mac_tool.py
index b918c58..07f9f25 100755
--- a/old/node/tools/gyp/pylib/gyp/mac_tool.py
+++ b/new/node/tools/gyp/pylib/gyp/mac_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/old/node/tools/gyp/pylib/gyp/sun_tool.py b/new/node/tools/gyp/pylib/gyp/sun_tool.py
index 90d59c8..5b1956f 100755
--- a/old/node/tools/gyp/pylib/gyp/sun_tool.py
+++ b/new/node/tools/gyp/pylib/gyp/sun_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/old/node/tools/gyp/pylib/gyp/system_test.py b/new/node/tools/gyp/pylib/gyp/system_test.py
index 51c71e3..ea68283 100755
--- a/old/node/tools/gyp/pylib/gyp/system_test.py
+++ b/new/node/tools/gyp/pylib/gyp/system_test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/pylib/gyp/win_tool.py b/new/node/tools/gyp/pylib/gyp/win_tool.py
index c9c6a7b..8854619 100644
--- a/old/node/tools/gyp/pylib/gyp/win_tool.py
+++ b/new/node/tools/gyp/pylib/gyp/win_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/setup.py b/new/node/tools/gyp/setup.py
index ed2b41a..3e34ac3 100755
--- a/old/node/tools/gyp/setup.py
+++ b/new/node/tools/gyp/setup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2009 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/tools/graphviz.py b/new/node/tools/gyp/tools/graphviz.py
index 326ae22..553a101 100755
--- a/old/node/tools/gyp/tools/graphviz.py
+++ b/new/node/tools/gyp/tools/graphviz.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2011 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/tools/pretty_gyp.py b/new/node/tools/gyp/tools/pretty_gyp.py
index c51d358..262a0b3 100755
--- a/old/node/tools/gyp/tools/pretty_gyp.py
+++ b/new/node/tools/gyp/tools/pretty_gyp.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/tools/pretty_sln.py b/new/node/tools/gyp/tools/pretty_sln.py
index 3195d85..e51c44f 100755
--- a/old/node/tools/gyp/tools/pretty_sln.py
+++ b/new/node/tools/gyp/tools/pretty_sln.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/gyp/tools/pretty_vcproj.py b/new/node/tools/gyp/tools/pretty_vcproj.py
index 6099bd7..41530a8 100755
--- a/old/node/tools/gyp/tools/pretty_vcproj.py
+++ b/new/node/tools/gyp/tools/pretty_vcproj.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 # Copyright (c) 2012 Google Inc. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
diff --git a/old/node/tools/install.py b/new/node/tools/install.py
index 028e5ca..b1a2311 100755
--- a/old/node/tools/install.py
+++ b/new/node/tools/install.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import errno
 import json
diff --git a/old/node/tools/js2c.py b/new/node/tools/js2c.py
index 69ec8a8..3c6b41f 100755
--- a/old/node/tools/js2c.py
+++ b/new/node/tools/js2c.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2006-2008 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/tools/run-valgrind.py b/old/node/tools/run-valgrind.py
deleted file mode 120000
index 07935ad..0000000
--- a/old/node/tools/run-valgrind.py
+++ /dev/null
@@ -1 +0,0 @@
-../deps/v8/tools/run-valgrind.py
\ No newline at end of file
diff --git a/new/node/tools/run-valgrind.py b/new/node/tools/run-valgrind.py
new file mode 100755
index 0000000..067c326
--- /dev/null
+++ b/new/node/tools/run-valgrind.py
@@ -0,0 +1,77 @@
+#!/usr/bin/python2
+#
+# Copyright 2009 the V8 project authors. All rights reserved.
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+#       copyright notice, this list of conditions and the following
+#       disclaimer in the documentation and/or other materials provided
+#       with the distribution.
+#     * Neither the name of Google Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# Simple wrapper for running valgrind and checking the output on
+# stderr for memory leaks.
+
+import subprocess
+import sys
+import re
+
+VALGRIND_ARGUMENTS = [
+  'valgrind',
+  '--error-exitcode=1',
+  '--leak-check=full',
+  '--smc-check=all'
+]
+
+# Compute the command line.
+command = VALGRIND_ARGUMENTS + sys.argv[1:]
+
+# Run valgrind.
+process = subprocess.Popen(command, stderr=subprocess.PIPE)
+code = process.wait();
+errors = process.stderr.readlines();
+
+# If valgrind produced an error, we report that to the user.
+if code != 0:
+  sys.stderr.writelines(errors)
+  sys.exit(code)
+
+# Look through the leak details and make sure that we don't
+# have any definitely, indirectly, and possibly lost bytes.
+LEAK_RE = r"(?:definitely|indirectly|possibly) lost: "
+LEAK_LINE_MATCHER = re.compile(LEAK_RE)
+LEAK_OKAY_MATCHER = re.compile(r"lost: 0 bytes in 0 blocks")
+leaks = []
+for line in errors:
+  if LEAK_LINE_MATCHER.search(line):
+    leaks.append(line)
+    if not LEAK_OKAY_MATCHER.search(line):
+      sys.stderr.writelines(errors)
+      sys.exit(1)
+
+# Make sure we found between 2 and 3 leak lines.
+if len(leaks) < 2 or len(leaks) > 3:
+  sys.stderr.writelines(errors)
+  sys.stderr.write('\n\n#### Malformed valgrind output.\n#### Exiting.\n')
+  sys.exit(1)
+
+# No leaks found.
+sys.exit(0)
diff --git a/old/node/tools/test.py b/new/node/tools/test.py
index d711f9c..f40c0c1 100755
--- a/old/node/tools/test.py
+++ b/new/node/tools/test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2008 the V8 project authors. All rights reserved.
 # Redistribution and use in source and binary forms, with or without
diff --git a/old/node/tools/waf-light b/new/node/tools/waf-light
index b4acc4b..d76057b 100755
--- a/old/node/tools/waf-light
+++ b/new/node/tools/waf-light
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: ISO8859-1
 # Thomas Nagy, 2005-2010
 
diff --git a/old/node/tools/wafadmin/Build.py b/new/node/tools/wafadmin/Build.py
index 675bca1..4256719 100644
--- a/old/node/tools/wafadmin/Build.py
+++ b/new/node/tools/wafadmin/Build.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005 (ita)
 
diff --git a/old/node/tools/wafadmin/Configure.py b/new/node/tools/wafadmin/Configure.py
index 846e185..4bfe955 100644
--- a/old/node/tools/wafadmin/Configure.py
+++ b/new/node/tools/wafadmin/Configure.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005-2008 (ita)
 
diff --git a/old/node/tools/wafadmin/Constants.py b/new/node/tools/wafadmin/Constants.py
index 2b531c2..f6d1e4b 100644
--- a/old/node/tools/wafadmin/Constants.py
+++ b/new/node/tools/wafadmin/Constants.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Yinon dot me gmail 2008
 
diff --git a/old/node/tools/wafadmin/Environment.py b/new/node/tools/wafadmin/Environment.py
index 0b6e72c..78a13cb 100644
--- a/old/node/tools/wafadmin/Environment.py
+++ b/new/node/tools/wafadmin/Environment.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005 (ita)
 
diff --git a/old/node/tools/wafadmin/Logs.py b/new/node/tools/wafadmin/Logs.py
index c160b37..53e7130 100644
--- a/old/node/tools/wafadmin/Logs.py
+++ b/new/node/tools/wafadmin/Logs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005 (ita)
 
diff --git a/old/node/tools/wafadmin/Node.py b/new/node/tools/wafadmin/Node.py
index 33d12f9..5d989f0 100644
--- a/old/node/tools/wafadmin/Node.py
+++ b/new/node/tools/wafadmin/Node.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005 (ita)
 
diff --git a/old/node/tools/wafadmin/Options.py b/new/node/tools/wafadmin/Options.py
index c90bb99..9c6db87 100644
--- a/old/node/tools/wafadmin/Options.py
+++ b/new/node/tools/wafadmin/Options.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Scott Newton, 2005 (scottn)
 # Thomas Nagy, 2006 (ita)
diff --git a/old/node/tools/wafadmin/Runner.py b/new/node/tools/wafadmin/Runner.py
index 5237c1a..ef19b52 100644
--- a/old/node/tools/wafadmin/Runner.py
+++ b/new/node/tools/wafadmin/Runner.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005-2008 (ita)
 
diff --git a/old/node/tools/wafadmin/Scripting.py b/new/node/tools/wafadmin/Scripting.py
index d975bd9..5c4590a 100644
--- a/old/node/tools/wafadmin/Scripting.py
+++ b/new/node/tools/wafadmin/Scripting.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005 (ita)
 
diff --git a/old/node/tools/wafadmin/Task.py b/new/node/tools/wafadmin/Task.py
index 24edf2c..e218dcd 100644
--- a/old/node/tools/wafadmin/Task.py
+++ b/new/node/tools/wafadmin/Task.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005-2008 (ita)
 
diff --git a/old/node/tools/wafadmin/TaskGen.py b/new/node/tools/wafadmin/TaskGen.py
index 260d999..1390081 100644
--- a/old/node/tools/wafadmin/TaskGen.py
+++ b/new/node/tools/wafadmin/TaskGen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005-2008 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/__init__.py b/new/node/tools/wafadmin/Tools/__init__.py
index bc6ca23..1dfc48c 100644
--- a/old/node/tools/wafadmin/Tools/__init__.py
+++ b/new/node/tools/wafadmin/Tools/__init__.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/ar.py b/new/node/tools/wafadmin/Tools/ar.py
index af9b17f..fc7f452 100644
--- a/old/node/tools/wafadmin/Tools/ar.py
+++ b/new/node/tools/wafadmin/Tools/ar.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006-2008 (ita)
 # Ralf Habacker, 2006 (rh)
diff --git a/old/node/tools/wafadmin/Tools/cc.py b/new/node/tools/wafadmin/Tools/cc.py
index 59edae4..0e27eb1 100644
--- a/old/node/tools/wafadmin/Tools/cc.py
+++ b/new/node/tools/wafadmin/Tools/cc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/ccroot.py b/new/node/tools/wafadmin/Tools/ccroot.py
index 5e7fcf3..6191dbd 100644
--- a/old/node/tools/wafadmin/Tools/ccroot.py
+++ b/new/node/tools/wafadmin/Tools/ccroot.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005-2008 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/compiler_cc.py b/new/node/tools/wafadmin/Tools/compiler_cc.py
index f3cb8cd..c952c6d 100644
--- a/old/node/tools/wafadmin/Tools/compiler_cc.py
+++ b/new/node/tools/wafadmin/Tools/compiler_cc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Matthias Jahn jahn dôt matthias ât freenet dôt de, 2007 (pmarat)
 
diff --git a/old/node/tools/wafadmin/Tools/compiler_cxx.py b/new/node/tools/wafadmin/Tools/compiler_cxx.py
index 6cad191..5a69036 100644
--- a/old/node/tools/wafadmin/Tools/compiler_cxx.py
+++ b/new/node/tools/wafadmin/Tools/compiler_cxx.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Matthias Jahn jahn dôt matthias ât freenet dôt de 2007 (pmarat)
 
diff --git a/old/node/tools/wafadmin/Tools/compiler_d.py b/new/node/tools/wafadmin/Tools/compiler_d.py
index 1ea5efa..46acc20 100644
--- a/old/node/tools/wafadmin/Tools/compiler_d.py
+++ b/new/node/tools/wafadmin/Tools/compiler_d.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Carlos Rafael Giani, 2007 (dv)
 
diff --git a/old/node/tools/wafadmin/Tools/config_c.py b/new/node/tools/wafadmin/Tools/config_c.py
index 21729d4..5047647 100644
--- a/old/node/tools/wafadmin/Tools/config_c.py
+++ b/new/node/tools/wafadmin/Tools/config_c.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005-2008 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/cxx.py b/new/node/tools/wafadmin/Tools/cxx.py
index 052bcef..04b7f5c 100644
--- a/old/node/tools/wafadmin/Tools/cxx.py
+++ b/new/node/tools/wafadmin/Tools/cxx.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/d.py b/new/node/tools/wafadmin/Tools/d.py
index 602f5af..d7f68b6 100644
--- a/old/node/tools/wafadmin/Tools/d.py
+++ b/new/node/tools/wafadmin/Tools/d.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Carlos Rafael Giani, 2007 (dv)
 # Thomas Nagy, 2007-2008 (ita)
diff --git a/old/node/tools/wafadmin/Tools/dmd.py b/new/node/tools/wafadmin/Tools/dmd.py
index 9c74908..d724c62 100644
--- a/old/node/tools/wafadmin/Tools/dmd.py
+++ b/new/node/tools/wafadmin/Tools/dmd.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Carlos Rafael Giani, 2007 (dv)
 # Thomas Nagy, 2008 (ita)
diff --git a/old/node/tools/wafadmin/Tools/gas.py b/new/node/tools/wafadmin/Tools/gas.py
index c983b0a..a9eff2a 100644
--- a/old/node/tools/wafadmin/Tools/gas.py
+++ b/new/node/tools/wafadmin/Tools/gas.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2008 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/gcc.py b/new/node/tools/wafadmin/Tools/gcc.py
index 40a788e..5bdefb8 100644
--- a/old/node/tools/wafadmin/Tools/gcc.py
+++ b/new/node/tools/wafadmin/Tools/gcc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006-2008 (ita)
 # Ralf Habacker, 2006 (rh)
diff --git a/old/node/tools/wafadmin/Tools/gdc.py b/new/node/tools/wafadmin/Tools/gdc.py
index 4d2a321..994f2ad 100644
--- a/old/node/tools/wafadmin/Tools/gdc.py
+++ b/new/node/tools/wafadmin/Tools/gdc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Carlos Rafael Giani, 2007 (dv)
 
diff --git a/old/node/tools/wafadmin/Tools/gnu_dirs.py b/new/node/tools/wafadmin/Tools/gnu_dirs.py
index 856e4a7..10e0916 100644
--- a/old/node/tools/wafadmin/Tools/gnu_dirs.py
+++ b/new/node/tools/wafadmin/Tools/gnu_dirs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Ali Sabil, 2007
 
diff --git a/old/node/tools/wafadmin/Tools/gob2.py b/new/node/tools/wafadmin/Tools/gob2.py
index 00aaa32..4459a79 100644
--- a/old/node/tools/wafadmin/Tools/gob2.py
+++ b/new/node/tools/wafadmin/Tools/gob2.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Ali Sabil, 2007
 
diff --git a/old/node/tools/wafadmin/Tools/gxx.py b/new/node/tools/wafadmin/Tools/gxx.py
index 133e116..fcca9db 100644
--- a/old/node/tools/wafadmin/Tools/gxx.py
+++ b/new/node/tools/wafadmin/Tools/gxx.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006 (ita)
 # Ralf Habacker, 2006 (rh)
diff --git a/old/node/tools/wafadmin/Tools/icc.py b/new/node/tools/wafadmin/Tools/icc.py
index 9c9a926..a389877 100644
--- a/old/node/tools/wafadmin/Tools/icc.py
+++ b/new/node/tools/wafadmin/Tools/icc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Stian Selnes, 2008
 # Thomas Nagy 2009
diff --git a/old/node/tools/wafadmin/Tools/icpc.py b/new/node/tools/wafadmin/Tools/icpc.py
index 726e3a4..a98b4e6 100644
--- a/old/node/tools/wafadmin/Tools/icpc.py
+++ b/new/node/tools/wafadmin/Tools/icpc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy 2009
 
diff --git a/old/node/tools/wafadmin/Tools/intltool.py b/new/node/tools/wafadmin/Tools/intltool.py
index deb8f4a..d1bb6af 100644
--- a/old/node/tools/wafadmin/Tools/intltool.py
+++ b/new/node/tools/wafadmin/Tools/intltool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/libtool.py b/new/node/tools/wafadmin/Tools/libtool.py
index a9b5689..a1c0112 100644
--- a/old/node/tools/wafadmin/Tools/libtool.py
+++ b/new/node/tools/wafadmin/Tools/libtool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Matthias Jahn, 2008, jahn matthias ath freenet punto de
 # Thomas Nagy, 2008 (ita)
diff --git a/old/node/tools/wafadmin/Tools/misc.py b/new/node/tools/wafadmin/Tools/misc.py
index 9903ee4..54a773e 100644
--- a/old/node/tools/wafadmin/Tools/misc.py
+++ b/new/node/tools/wafadmin/Tools/misc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/nasm.py b/new/node/tools/wafadmin/Tools/nasm.py
index b99c3c7..8251c9a 100644
--- a/old/node/tools/wafadmin/Tools/nasm.py
+++ b/new/node/tools/wafadmin/Tools/nasm.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2008
 
diff --git a/old/node/tools/wafadmin/Tools/osx.py b/new/node/tools/wafadmin/Tools/osx.py
index 7ca95f5..57a09ae 100644
--- a/old/node/tools/wafadmin/Tools/osx.py
+++ b/new/node/tools/wafadmin/Tools/osx.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy 2008
 
diff --git a/old/node/tools/wafadmin/Tools/preproc.py b/new/node/tools/wafadmin/Tools/preproc.py
index 1a9b0ff..03cefa8 100644
--- a/old/node/tools/wafadmin/Tools/preproc.py
+++ b/new/node/tools/wafadmin/Tools/preproc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006-2009 (ita)
 
diff --git a/old/node/tools/wafadmin/Tools/python.py b/new/node/tools/wafadmin/Tools/python.py
index e9f9048..b7e20d4 100644
--- a/old/node/tools/wafadmin/Tools/python.py
+++ b/new/node/tools/wafadmin/Tools/python.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2007 (ita)
 # Gustavo Carneiro (gjc), 2007
diff --git a/old/node/tools/wafadmin/Tools/suncc.py b/new/node/tools/wafadmin/Tools/suncc.py
index 1387229..0be7862 100644
--- a/old/node/tools/wafadmin/Tools/suncc.py
+++ b/new/node/tools/wafadmin/Tools/suncc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006 (ita)
 # Ralf Habacker, 2006 (rh)
diff --git a/old/node/tools/wafadmin/Tools/suncxx.py b/new/node/tools/wafadmin/Tools/suncxx.py
index b6e9ef8..b66029a 100644
--- a/old/node/tools/wafadmin/Tools/suncxx.py
+++ b/new/node/tools/wafadmin/Tools/suncxx.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006 (ita)
 # Ralf Habacker, 2006 (rh)
diff --git a/old/node/tools/wafadmin/Tools/unittestw.py b/new/node/tools/wafadmin/Tools/unittestw.py
index b85ccc0..951b138 100644
--- a/old/node/tools/wafadmin/Tools/unittestw.py
+++ b/new/node/tools/wafadmin/Tools/unittestw.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Carlos Rafael Giani, 2006
 
diff --git a/old/node/tools/wafadmin/Tools/winres.py b/new/node/tools/wafadmin/Tools/winres.py
index 2500d43..6f19a60 100644
--- a/old/node/tools/wafadmin/Tools/winres.py
+++ b/new/node/tools/wafadmin/Tools/winres.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Brant Young, 2007
 
diff --git a/old/node/tools/wafadmin/Tools/xlc.py b/new/node/tools/wafadmin/Tools/xlc.py
index e787f7d..aae10f5 100644
--- a/old/node/tools/wafadmin/Tools/xlc.py
+++ b/new/node/tools/wafadmin/Tools/xlc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006-2008 (ita)
 # Ralf Habacker, 2006 (rh)
diff --git a/old/node/tools/wafadmin/Tools/xlcxx.py b/new/node/tools/wafadmin/Tools/xlcxx.py
index 696fcae..1430ae0 100644
--- a/old/node/tools/wafadmin/Tools/xlcxx.py
+++ b/new/node/tools/wafadmin/Tools/xlcxx.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2006 (ita)
 # Ralf Habacker, 2006 (rh)
diff --git a/old/node/tools/wafadmin/Utils.py b/new/node/tools/wafadmin/Utils.py
index b86b76f..60ef2c0 100644
--- a/old/node/tools/wafadmin/Utils.py
+++ b/new/node/tools/wafadmin/Utils.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005 (ita)
 
diff --git a/old/node/tools/wafadmin/__init__.py b/new/node/tools/wafadmin/__init__.py
index 01273cf..f36d257 100644
--- a/old/node/tools/wafadmin/__init__.py
+++ b/new/node/tools/wafadmin/__init__.py
@@ -1,3 +1,3 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2005 (ita)
diff --git a/old/node/tools/wafadmin/py3kfixes.py b/new/node/tools/wafadmin/py3kfixes.py
index 6aeb26b..d5c6596 100644
--- a/old/node/tools/wafadmin/py3kfixes.py
+++ b/new/node/tools/wafadmin/py3kfixes.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # encoding: utf-8
 # Thomas Nagy, 2009 (ita)
 
