diff -Naur xbmc-11.0-old/xbmc/threads/platform/pthreads/ThreadImpl.h xbmc-11.0/xbmc/threads/platform/pthreads/ThreadImpl.h
--- xbmc-11.0-old/xbmc/threads/platform/pthreads/ThreadImpl.h	2012-03-24 23:14:41.353089828 +0100
+++ xbmc-11.0/xbmc/threads/platform/pthreads/ThreadImpl.h	2012-03-24 23:14:08.204626508 +0100
@@ -22,6 +22,7 @@
 #pragma once
 
 #include <pthread.h>
+#include <unistd.h>
 
 struct threadOpaque
 {
