--- ../mvriel-Docblox-0db98ef.orig/bin/docblox.php	2011-08-21 16:08:56.000000000 +0200
+++ bin/docblox.php	2011-08-22 17:24:43.366953620 +0200
@@ -25,9 +25,7 @@
 }
 
 // determine base include folder, if @php_bin@ contains @php_bin then we do not install via PEAR
-$base_include_folder = (strpos('@php_dir@', '@php_dir') === 0)
-  ? dirname(__FILE__) . '/../src'
-  : '@php_dir@/DocBlox/src';
+$base_include_folder = '/usr/share/pear/docblox/src';
 
 // set path to add lib folder, load the Zend Autoloader and include the symfony timer
 set_include_path($base_include_folder . PATH_SEPARATOR . get_include_path());
