diff -Naur old/Makefile new/Makefile
--- old/Makefile	2012-06-17 20:52:53.000000000 +0200
+++ new/Makefile	2013-03-19 17:44:43.000000000 +0100

@@ -62,7 +62,7 @@
 
 SHELL=/bin/sh
 
-DESTDIR=
+DESTDIR ?=
 
 # AMC components to build
 
@@ -203,10 +203,7 @@
 	install    -m 0644 $(USER_GROUP) $(GLADE) $(DESTDIR)/$(MODSDIR)/perl
 	install -d -m 0755 $(USER_GROUP) $(DESTDIR)/$(TEXDIR)
 	install    -m 0644 $(USER_GROUP) $(STY) $(DESTDIR)/$(TEXDIR)
-ifneq ($(TEXDOCDIR),)
-	install -d -m 0755 $(USER_GROUP) $(DESTDIR)/$(TEXDOCDIR)
-	install    -m 0644 $(USER_GROUP) doc/sty/*.pdf doc/sty/*.tex $(DESTDIR)/$(TEXDOCDIR)
-endif
+
 ifneq ($(DESKTOPDIR),)
 	install -d -m 0755 $(USER_GROUP) $(DESTDIR)/$(DESKTOPDIR)
 	install    -m 0644 $(USER_GROUP) -T auto-multiple-choice.desktop $(DESTDIR)/$(DESKTOPDIR)/auto-multiple-choice.desktop
@@ -234,6 +231,11 @@
 	install    -m 0644 $(USER_GROUP) AMC-perl/AMC/DataModule/*.pm $(DESTDIR)/$(PERLDIR)/AMC/DataModule
 	install    -m 0644 $(USER_GROUP) AMC-perl/AMC/Gui/*.pm $(DESTDIR)/$(PERLDIR)/AMC/Gui
 	install    -m 0644 $(USER_GROUP) AMC-perl/AMC/Gui/*.glade $(DESTDIR)/$(PERLDIR)/AMC/Gui
+install_doc:FORCE
+ifneq ($(TEXDOCDIR),)
+	install -d -m 0755 $(USER_GROUP) $(DESTDIR)/$(TEXDOCDIR)
+	install    -m 0644 $(USER_GROUP) doc/sty/*.pdf doc/sty/*.tex $(DESTDIR)/$(TEXDOCDIR)
+endif
 ifneq ($(SYSTEM_TYPE),deb) # with debian, done with dh_install{doc,man}
 	install -d -m 0755 $(USER_GROUP) $(DESTDIR)/$(DOCDIR)
 	install    -m 0644 $(USER_GROUP) $(wildcard doc/auto-multiple-choice.??.xml doc/auto-multiple-choice.??.pdf) $(DESTDIR)/$(DOCDIR)
 	
diff -Naur old/Makefile.conf new/Makefile.conf
--- old/Makefile.conf	2012-06-17 20:49:05.000000000 +0200
+++ new/Makefile.conf	2013-03-19 13:45:22.000000000 +0100

@@ -33,7 +33,7 @@
 # directory for executable
 BINDIR=/usr/bin
 # directory for perl modules
-PERLDIR=/usr/share/perl5
+PERLDIR=/usr/share/perl5/vendor_perl
 # directory for some AMC executables, not to be used by end-user
 MODSDIR=/usr/lib/AMC
 # directory for LaTeX style file
@@ -73,7 +73,7 @@
 
 # BUILD : options to compile with netpbm / opencv libraries
 
-GCC_NETPBM=-lnetpbm
+GCC_NETPBM=-I /usr/include/netpbm -lnetpbm
 
 GCC_OPENCV=-I /usr/include/opencv
 GCC_OPENCV_LIBS=auto
