Common subdirectories: nct6775/.git and nct6775-2/.git
diff -Nau nct6775/Makefile nct6775-2/Makefile
--- nct6775/Makefile	2012-05-20 20:41:54.841128770 +0200
+++ nct6775-2/Makefile	2012-05-20 20:43:53.396408837 +0200
@@ -5,7 +5,7 @@
 #TARGET		:= 2.6.33.5
 KERNEL_MODULES	:= /lib/modules/$(TARGET)
 # KERNEL_BUILD	:= $(KERNEL_MODULES)/build
-KERNEL_BUILD	:= /usr/src/linux-headers-$(TARGET)
+KERNEL_BUILD	:= /usr/src/linux-$(TARGET)
 
 #SYSTEM_MAP	:= $(KERNEL_BUILD)/System.map
 SYSTEM_MAP	:= /boot/System.map-$(TARGET)
