--- a/jetty.sh	2013-05-07 03:01:09.000000000 +0300
+++ b/jetty.sh	2013-06-03 20:14:08.219707383 +0300
@@ -340,6 +340,8 @@
 #####################################################
 # Find a pid and state file
 #####################################################
+mkdir -p $JETTY_RUN
+chown -R $JETTY_USER:$JETTY_USER $JETTY_RUN
 if [ -z "$JETTY_PID" ] 
 then
   JETTY_PID="$JETTY_RUN/jetty.pid"
@@ -347,7 +349,7 @@
 
 if [ -z "$JETTY_STATE" ] 
 then
-  JETTY_STATE=$JETTY_HOME/jetty.state
+  JETTY_STATE=$JETTY_RUN/jetty.state
 fi
 JAVA_OPTIONS+=("-Djetty.state=$JETTY_STATE")
 rm -f $JETTY_STATE
@@ -469,7 +471,7 @@
         touch "$JETTY_PID"
         chown "$JETTY_USER" "$JETTY_PID"
         # FIXME: Broken solution: wordsplitting, pathname expansion, arbitrary command execution, etc.
-        su - "$JETTY_USER" -c "
+        su -s /bin/sh - "$JETTY_USER" -c "
           exec ${RUN_CMD[*]} --daemon &
           disown \$!
           echo \$! > '$JETTY_PID'"
