diff -rub ../phpsh/setup.py ./setup.py
--- ../phpsh/setup.py	2011-01-18 09:17:24.630020120 +0100
+++ ./setup.py	2011-01-18 09:21:52.820020307 +0100
@@ -14,10 +14,7 @@
     os.waitpid(p.pid, 0)

 # something better than this?
-if os.getenv("USER") == "root":
-    config_dir = "/etc/phpsh"
-else:
-    config_dir = os.getenv("HOME") + "/.phpsh"
+config_dir = "/etc/phpsh"

 setup(
     name="phpsh",
Solo in .: setup.py.orig
