diff -Naur a/backends/globals.py b/backends/globals.py
--- a/backends/globals.py	2010-11-11 14:26:19.484237001 +0100
+++ b/backends/globals.py	2010-11-11 15:09:40.364237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/log.py b/backends/log.py
--- a/backends/log.py	2010-11-11 14:26:19.484237001 +0100
+++ b/backends/log.py	2010-11-11 15:09:36.384237003 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/backlight.py b/backends/session/backlight.py
--- a/backends/session/backlight.py	2010-11-11 14:26:19.474237001 +0100
+++ b/backends/session/backlight.py	2010-11-11 15:09:19.834237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/bluetooth.py b/backends/session/bluetooth.py
--- a/backends/session/bluetooth.py	2010-11-11 14:26:19.434237001 +0100
+++ b/backends/session/bluetooth.py	2010-11-11 15:09:06.894237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/cpu.py b/backends/session/cpu.py
--- a/backends/session/cpu.py	2010-11-11 14:26:19.474237001 +0100
+++ b/backends/session/cpu.py	2010-11-11 15:09:21.714237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/options.py b/backends/session/options.py
--- a/backends/session/options.py	2010-11-11 14:26:19.464237001 +0100
+++ b/backends/session/options.py	2010-11-11 15:09:17.814237000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/util/config.py b/backends/session/util/config.py
--- a/backends/session/util/config.py	2010-11-11 14:26:19.484237001 +0100
+++ b/backends/session/util/config.py	2010-11-11 15:09:31.794237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/util/hotkeys.py b/backends/session/util/hotkeys.py
--- a/backends/session/util/hotkeys.py	2010-11-11 14:26:19.484237001 +0100
+++ b/backends/session/util/hotkeys.py	2010-11-11 15:09:27.714237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/util/icons.py b/backends/session/util/icons.py
--- a/backends/session/util/icons.py	2010-11-11 14:26:19.484237001 +0100
+++ b/backends/session/util/icons.py	2010-11-11 15:09:29.784237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/util/locales.py b/backends/session/util/locales.py
--- a/backends/session/util/locales.py	2010-11-11 14:26:19.484237001 +0100
+++ b/backends/session/util/locales.py	2010-11-11 15:09:23.764237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/util/notifications.py b/backends/session/util/notifications.py
--- a/backends/session/util/notifications.py	2010-11-11 14:26:19.484237001 +0100
+++ b/backends/session/util/notifications.py	2010-11-11 15:09:25.804237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/webcam.py b/backends/session/webcam.py
--- a/backends/session/webcam.py	2010-11-11 14:26:19.454237001 +0100
+++ b/backends/session/webcam.py	2010-11-11 15:09:10.384237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/session/wireless.py b/backends/session/wireless.py
--- a/backends/session/wireless.py	2010-11-11 14:26:19.444237001 +0100
+++ b/backends/session/wireless.py	2010-11-11 15:09:13.784237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/system/backlight.py b/backends/system/backlight.py
--- a/backends/system/backlight.py	2010-11-11 14:26:19.424237001 +0100
+++ b/backends/system/backlight.py	2010-11-11 15:08:54.574237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/system/bluetooth.py b/backends/system/bluetooth.py
--- a/backends/system/bluetooth.py	2010-11-11 14:26:19.404237001 +0100
+++ b/backends/system/bluetooth.py	2010-11-11 15:08:44.404237000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/system/cpu.py b/backends/system/cpu.py
--- a/backends/system/cpu.py	2010-11-11 14:26:19.424237001 +0100
+++ b/backends/system/cpu.py	2010-11-11 15:08:56.714237003 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/system/fan.py b/backends/system/fan.py
--- a/backends/system/fan.py	2010-11-11 14:26:19.424237001 +0100
+++ b/backends/system/fan.py	2010-11-11 15:08:48.154237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/system/options.py b/backends/system/options.py
--- a/backends/system/options.py	2010-11-11 14:26:19.424237001 +0100
+++ b/backends/system/options.py	2010-11-11 15:08:52.264237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/system/powermanagement.py b/backends/system/powermanagement.py
--- a/backends/system/powermanagement.py	2010-11-11 14:26:19.424237001 +0100
+++ b/backends/system/powermanagement.py	2010-11-11 15:08:58.834237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/system/util/config.py b/backends/system/util/config.py
--- a/backends/system/util/config.py	2010-11-11 14:26:19.424237001 +0100
+++ b/backends/system/util/config.py	2010-11-11 15:09:01.124237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/system/webcam.py b/backends/system/webcam.py
--- a/backends/system/webcam.py	2010-11-11 14:26:19.424237001 +0100
+++ b/backends/system/webcam.py	2010-11-11 15:08:50.384237000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/backends/system/wireless.py b/backends/system/wireless.py
--- a/backends/system/wireless.py	2010-11-11 14:26:19.414237001 +0100
+++ b/backends/system/wireless.py	2010-11-11 15:08:45.984237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/samsung-tools-applet.py b/samsung-tools-applet.py
--- a/samsung-tools-applet.py	2010-11-11 14:26:19.484237001 +0100
+++ b/samsung-tools-applet.py	2010-11-11 15:09:44.694237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/samsung-tools-preferences.py b/samsung-tools-preferences.py
--- a/samsung-tools-preferences.py	2010-11-11 14:26:19.394237001 +0100
+++ b/samsung-tools-preferences.py	2010-11-11 15:07:03.814237002 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/samsung-tools.py b/samsung-tools.py
--- a/samsung-tools.py	2010-11-11 14:26:19.334237001 +0100
+++ b/samsung-tools.py	2010-11-11 15:06:54.424237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/session-service.py b/session-service.py
--- a/session-service.py	2010-11-11 14:26:19.484237001 +0100
+++ b/session-service.py	2010-11-11 15:09:47.164237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
diff -Naur a/system-service.py b/system-service.py
--- a/system-service.py	2010-11-11 14:26:19.334237001 +0100
+++ b/system-service.py	2010-11-11 15:06:43.334237001 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2.7
 # coding=UTF-8
 #
 # Samsung-Tools
