--- old/agg-plot/support_x11.cpp	2013-05-22 11:05:54.379104589 +0900
+++ new/agg-plot/support_x11.cpp	2013-05-22 11:09:20.804395519 +0900
@@ -7,8 +7,8 @@
 #define TTF_SYSTEM_DIR "/Library/Fonts/"
 #define CONSOLE_FONT_NAME "Monaco"
 #else
-const char *ttf_names[] = {"ubuntu-font-family/Ubuntu-R.ttf", "freefont/FreeSans.ttf", "ttf-dejavu/DejaVuSans.ttf", 0};
-#define TTF_SYSTEM_DIR "/usr/share/fonts/truetype/"
+const char *ttf_names[] = {"FreeSans.ttf", "DejaVuSans.ttf", 0};
+#define TTF_SYSTEM_DIR "/usr/share/fonts/TTF/"
 #define CONSOLE_FONT_NAME "monospace"
 #endif
 
