diff -aurN '--exclude=*~' rinse-1.9.1.orig/etc/fedora-15.packages rinse-1.9.1.fedora-15/etc/fedora-15.packages
--- rinse-1.9.1.orig/etc/fedora-15.packages	1969-12-31 16:00:00.000000000 -0800
+++ rinse-1.9.1.fedora-15/etc/fedora-15.packages	2011-10-31 00:47:15.268095933 -0700
@@ -0,0 +1,86 @@
+audit-libs
+basesystem
+bash
+bzip2-libs
+ca-certificates
+chkconfig
+coreutils
+coreutils-libs
+cracklib
+cracklib-dicts
+curl
+cyrus-sasl-lib
+db4
+db4-utils
+elfutils-libelf
+expat
+file-libs
+filesystem
+gamin
+gdbm
+glib2
+glibc
+glibc-common
+gmp
+gnupg2
+gpgme
+grep
+gzip
+info
+keyutils-libs
+krb5-libs
+libacl
+libassuan
+libattr
+libcap
+libcom_err
+libcurl
+libdb
+libffi
+libgcc
+libgcrypt
+libgpg-error
+libidn
+libselinux
+libsepol
+libssh2
+libstdc++
+libxml2
+lua
+ncurses
+ncurses-base
+ncurses-libs
+nspr
+nss
+nss-softokn
+nss-softokn-freebl
+nss-sysinit
+nss-util
+openldap
+openssl
+pam
+pcre
+pinentry
+pkgconfig
+popt
+pth
+pygpgme
+python
+python-iniparse
+python-libs
+python-pycurl
+python-urlgrabber
+readline
+rpm
+rpm-build-libs
+rpm-libs
+rpm-python
+sed
+setup
+shared-mime-info
+sqlite
+tzdata
+xz-libs
+yum
+yum-metadata-parser
+zlib
diff -aurN '--exclude=*~' rinse-1.9.1.orig/etc/rinse.conf rinse-1.9.1.fedora-15/etc/rinse.conf
--- rinse-1.9.1.orig/etc/rinse.conf	2011-09-01 06:30:02.000000000 -0700
+++ rinse-1.9.1.fedora-15/etc/rinse.conf	2011-10-31 01:19:52.434846570 -0700
@@ -73,6 +73,9 @@
 mirror       = http://ftp.heanet.ie/pub/fedora-archive/fedora/linux/releases/10/Fedora/i386/os/Packages/
 mirror.amd64 = http://ftp.heanet.ie/pub/fedora-archive/fedora/linux/releases/10/Fedora/x86_64/os/Packages/
 
+[fedora-15]
+mirror       = http://mirrors.kernel.org/fedora/releases/15/Fedora/i386/os/Packages/
+mirror.amd64 = http://mirrors.kernel.org/fedora/releases/15/Fedora/x86_64/os/Packages/
 
 [opensuse-10.1]
 mirror       = http://ftp.hosteurope.de/mirror/ftp.opensuse.org/discontinued/SL-10.1/inst-source/suse/i686/
@@ -98,3 +101,4 @@
 mirror       = http://download.opensuse.org/distribution/11.1/repo/oss/suse/i686/
 mirror.amd64 = http://download.opensuse.org/distribution/11.1/repo/oss/suse/x86_64/
 
+
diff -aurN '--exclude=*~' rinse-1.9.1.orig/scripts/fedora-15/post-install.sh rinse-1.9.1.fedora-15/scripts/fedora-15/post-install.sh
--- rinse-1.9.1.orig/scripts/fedora-15/post-install.sh	1969-12-31 16:00:00.000000000 -0800
+++ rinse-1.9.1.fedora-15/scripts/fedora-15/post-install.sh	2011-10-31 01:21:27.781517325 -0700
@@ -0,0 +1,97 @@
+#!/bin/sh
+#
+#  Customise the distribution post-install.
+#
+
+
+
+prefix=$1
+
+if [ ! -d "${prefix}" ]; then
+  echo "Serious error - the named directory doesn't exist."
+  exit
+fi
+
+
+#
+#  2.  Copy the cached .RPM files into the yum directory, so that
+#     yum doesn't need to make them again.
+#
+echo "  Setting up YUM cache"
+if [ ! -d ${prefix}/var/cache/yum/core/packages/ ]; then
+    mkdir -p ${prefix}/var/cache/yum/core/packages/
+fi
+if [ ! -d ${prefix}/var/cache/yum/updates-released/packages/ ]; then
+    mkdir -p ${prefix}/var/cache/yum/updates-released/packages/
+fi
+
+for i in ${prefix}/*.rpm ; do
+    cp $i ${prefix}/var/cache/yum/core/packages/
+    cp $i ${prefix}/var/cache/yum/updates-released/packages/
+done
+
+
+
+#
+#  3.  Ensure that Yum has a working configuration file.
+#
+arch=i386
+if [ $ARCH = "amd64" ] ; then
+    arch=x86_64
+fi
+
+echo "  Creating yum.conf"
+cat > ${prefix}/etc/yum.conf <<EOF
+[main]
+cachedir=/var/cache/yum
+debuglevel=1
+reposdir=/dev/null
+logfile=/var/log/yum.log
+retries=20
+obsoletes=1
+gpgcheck=0
+assumeyes=1
+
+# repos
+
+[core]
+name=core
+baseurl=$mirror
+
+EOF
+
+
+#
+#  4.  Run "yum install yum".
+#
+echo "  Priming the yum cache"
+if [ ! -d "${prefix}/var/cache/yum/core/packages/" ]; then
+    mkdir -p ${prefix}/var/cache/yum/core/packages
+fi
+cp /var/cache/rinse/fedora-15.$ARCH/* ${prefix}/var/cache/yum/core/packages/
+
+echo "  Bootstrapping yum"
+chroot ${prefix} /sbin/ldconfig
+chroot ${prefix} /usr/bin/yum -y install yum         2>/dev/null
+chroot ${prefix} /usr/bin/yum -y install vim-minimal 2>/dev/null
+chroot ${prefix} /usr/bin/yum -y install dhclient    2>/dev/null
+
+
+#
+#  5.  Clean up
+#
+echo "  Cleaning up"
+chroot ${prefix} /usr/bin/yum clean all
+umount ${prefix}/proc
+umount ${prefix}/sys
+
+
+#
+#  6.  Remove the .rpm files from the prefix root.
+#
+echo "  Final tidy..."
+for i in ${prefix}/*.rpm; do
+    rm -f $i
+done
+find ${prefix} -name '*.rpmorig' -delete
+find ${prefix} -name '*.rpmnew' -delete
