--- CMakeLists2.txt     2013-08-02 16:05:38.674154738 +0200
+++ CMakeLists.txt      2013-08-02 16:06:33.403746275 +0200
@@ -97,4 +97,5 @@
 set_target_properties( ibex PROPERTIES
 RESOURCE "${RESOURCES}" )
 
-install (DIRECTORY resources DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
+install (DIRECTORY resources DESTINATION "${CMAKE_INSTALL_PREFIX}/share/ibex")
+install (TARGETS ibex RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/share/ibex")
