--- sysconfig.linux.sh
+++ sysconfig.linux.sh
@@ -62,7 +62,7 @@
 ifn=`route -n|grep "^1.1.1.1"|awk '{print $8}'`
 echo "interface $ifn" >> "$base/forticlientsslvpn.log"
 
-addr=`ifconfig $ifn |grep "inet"|awk ' {print $2 }'| awk -F : '{ print $2}'`
+addr=`ifconfig $ifn |grep "inet"|awk ' {print $2 }'`
 echo "address $addr" >> "$base/forticlientsslvpn.log"
 
 echo "delete route 1.1.1.1" >> "$base/forticlientsslvpn.log"
