diff --git i/Makefile w/Makefile
index fc6bb43..6ae7cf0 100644
--- i/Makefile
+++ w/Makefile
@@ -12,13 +12,13 @@ CC?=gcc
 
 CFLAGS := -O2 -g -fstack-protector -D_FORTIFY_SOURCE=2 -Wall -W -Wstrict-prototypes -Wundef -fno-common -Werror-implicit-function-declaration -Wdeclaration-after-statement -Wformat -Wformat-security -Werror=format-security
 
-MY_CFLAGS := `pkg-config --cflags libnotify gtk+-2.0`
+MY_CFLAGS := `pkg-config --cflags libnotify gtk+-2.0 dbus-glib-1`
 #
 # pkg-config tends to make programs pull in a ton of libraries, not all 
 # are needed. -Wl,--as-needed tells the linker to just drop unused ones,
 # and that makes the applet load faster and use less memory.
 #
-LDF_A := -Wl,--as-needed `pkg-config --libs libnotify gtk+-2.0`
+LDF_A := -Wl,--as-needed `pkg-config --libs libnotify gtk+-2.0 dbus-glib-1`
 LDF_D := -Wl,--as-needed `pkg-config --libs glib-2.0 dbus-glib-1` `curl-config --libs` -Wl,"-z relro" -Wl,"-z now" 
 
 all:	kerneloops kerneloops-applet kerneloops.8.gz
diff --git i/kerneloops-applet.c w/kerneloops-applet.c
index 3f07f6c..39c8058 100644
--- i/kerneloops-applet.c
+++ w/kerneloops-applet.c
@@ -241,7 +241,7 @@ static void got_a_message(void)
 	close_notification();
 
 	notify = notify_notification_new(summary, message,
-				"/usr/share/kerneloops/icon.png", NULL);
+				"/usr/share/kerneloops/icon.png");
 
 	notify_notification_set_timeout(notify, 0);
 	notify_notification_set_urgency(notify, NOTIFY_URGENCY_CRITICAL);
@@ -307,7 +307,7 @@ static void sent_an_oops(void)
 	url_to_oops[0] = 0;
 
 	notify = notify_notification_new(summary, message,
-				"/usr/share/kerneloops/icon.png", NULL);
+				"/usr/share/kerneloops/icon.png");
 
 	notify_notification_set_timeout(notify, 5000);
 	notify_notification_set_urgency(notify, NOTIFY_URGENCY_LOW);
