diff -Naur libsx-2.04/freq/freq.c libsx-2.04_new/freq/freq.c
--- libsx-2.04/freq/freq.c	1999-07-04 06:07:13.000000000 +0200
+++ libsx-2.04_new/freq/freq.c	2007-05-27 18:48:13.709266055 +0200
@@ -56,7 +56,7 @@
  * in the butt.  Why on earth isn't strdup() in the POSIX standard
  * but something completely useless like mbstowcs() is?
  */
-char *strdup(const char *str);
+// char *strdup(const char *str);
 
 
 /*
diff -Naur libsx-2.04/libsx_defs libsx-2.04_new/libsx_defs
--- libsx-2.04/libsx_defs	1999-12-14 13:59:35.000000000 +0100
+++ libsx-2.04_new/libsx_defs	2007-05-27 18:44:07.699246742 +0200
@@ -44,7 +44,7 @@
 #  CFLAGS = -O3 -D_POSIX_SOURCE
 #
 # On Linux, use CC=cc, RANLIB=ranlib, and:
-CFLAGS = -Wall -O2 -D_POSIX_SOURCE
+CFLAGS = -Wall -O2 -fPIC
 #  
 # On a Motorola Delta/88K box, you can use (with gcc):
 #  CFLAGS = -O2 -DSYSV -DUSG -DMOTOROLA -DMOTOR32V2
diff -Naur libsx-2.04/src/colorsel.c libsx-2.04_new/src/colorsel.c
--- libsx-2.04/src/colorsel.c	2000-07-29 11:10:32.000000000 +0200
+++ libsx-2.04_new/src/colorsel.c	2007-05-27 18:47:41.523431888 +0200
@@ -63,7 +63,7 @@
 static void show_best_match(Widget w, CSelData *cdata);
 */
 
-int trunc(float v)
+int mytrunc(float v)
 {
   return (int)(0.5+v);
 }
@@ -201,25 +201,25 @@
         reset = True;
 	}
 
-      if (reset || trunc(cdata->r) != w_r)
+      if (reset || mytrunc(cdata->r) != w_r)
 	{
-        w_r = trunc(cdata->r);
+        w_r = mytrunc(cdata->r);
         sprintf(cvalue, "%d", w_r);
         SetScrollbar(cdata->red_scroll, (float)cdata->r, 255.0, 1.0);
         SetStringEntry(cdata->red_string, cvalue);
 	}
 
-      if (reset || trunc(cdata->g) != w_g)
+      if (reset || mytrunc(cdata->g) != w_g)
 	{
-        w_g = trunc(cdata->g);
+        w_g = mytrunc(cdata->g);
         sprintf(cvalue, "%d", w_g);
         SetScrollbar(cdata->green_scroll, (float)cdata->g, 255.0, 1.0);
         SetStringEntry(cdata->green_string, cvalue);
 	}
       
-      if (reset || trunc(cdata->b) != w_b)
+      if (reset || mytrunc(cdata->b) != w_b)
 	{
-        w_b = trunc(cdata->b);
+        w_b = mytrunc(cdata->b);
         sprintf(cvalue, "%d", w_b);
         SetScrollbar(cdata->blue_scroll, (float)cdata->b, 255.0, 1.0);
         SetStringEntry(cdata->blue_string, cvalue);
@@ -250,25 +250,25 @@
         reset = True;
 	}
 
-      if (reset || trunc(cdata->h) != w_h)
+      if (reset || mytrunc(cdata->h) != w_h)
 	{
-        w_h = trunc(cdata->h);
+        w_h = mytrunc(cdata->h);
         sprintf(cvalue, "%d", w_h);
         SetScrollbar(cdata->red_scroll, (float)cdata->h, 360.0, 1.0);
         SetStringEntry(cdata->red_string, cvalue);
 	}
       
-      if (reset || trunc(cdata->s) != w_s)
+      if (reset || mytrunc(cdata->s) != w_s)
 	{
-        w_s = trunc(cdata->s);
+        w_s = mytrunc(cdata->s);
         sprintf(cvalue, "%d", w_s);
         SetScrollbar(cdata->green_scroll, (float)cdata->s, 100.0, 1.0);
         SetStringEntry(cdata->green_string, cvalue);
 	}
       
-      if (reset || trunc(cdata->v) != w_v)
+      if (reset || mytrunc(cdata->v) != w_v)
 	{
-        w_v = trunc(cdata->v);
+        w_v = mytrunc(cdata->v);
         sprintf(cvalue, "%d", w_v);
         SetScrollbar(cdata->blue_scroll, (float)cdata->v, 255.0, 1.0);
         SetStringEntry(cdata->blue_string, cvalue);
@@ -300,33 +300,33 @@
         reset = True;
 	}
 
-      if (reset || trunc(cdata->c) != w_c)
+      if (reset || mytrunc(cdata->c) != w_c)
 	{
-        w_c = trunc(cdata->c);
+        w_c = mytrunc(cdata->c);
         sprintf(cvalue, "%d", w_c);
         SetScrollbar(cdata->red_scroll, (float)cdata->c, 100.0, 1.0);
         SetStringEntry(cdata->red_string, cvalue);
 	}
       
-      if (reset || trunc(cdata->m) != w_m)
+      if (reset || mytrunc(cdata->m) != w_m)
 	{
-        w_m = trunc(cdata->m);
+        w_m = mytrunc(cdata->m);
         sprintf(cvalue, "%d", w_m);
         SetScrollbar(cdata->green_scroll, (float)cdata->m, 100.0, 1.0);
         SetStringEntry(cdata->green_string, cvalue);
 	}
       
-      if (reset || trunc(cdata->y) != w_y)
+      if (reset || mytrunc(cdata->y) != w_y)
 	{
-        w_y = trunc(cdata->y);
+        w_y = mytrunc(cdata->y);
         sprintf(cvalue, "%d", w_y);
         SetScrollbar(cdata->blue_scroll, (float)cdata->y, 100.0, 1.0);
         SetStringEntry(cdata->blue_string, cvalue);
 	}
 
-      if (reset || trunc(cdata->k) != w_k)
+      if (reset || mytrunc(cdata->k) != w_k)
 	{
-        w_k = trunc(cdata->k);
+        w_k = mytrunc(cdata->k);
         sprintf(cvalue, "%d", w_k);
         SetScrollbar(cdata->black_scroll, (float)cdata->k, 100.0, 1.0);
         SetStringEntry(cdata->black_string, cvalue);
@@ -407,7 +407,7 @@
     }
 
   sprintf(cdata->match_list[0], "  %3d %3d %3d      %s",
-          trunc(fr), trunc(fg), trunc(fb), 
+          mytrunc(fr), mytrunc(fg), mytrunc(fb), 
           SX_Dialog[COLSEL_DIAL]);
     cdata->match_ptr[0] = cdata->match_list[0];
 
@@ -415,7 +415,7 @@
     {
     for(i=1; i<=num; i++)
       {
-      sprintf(format, "%%s%%%ds %%s%%2.2f %%%%", 
+      sprintf(format, "%%s%%%lds %%s%%2.2f %%%%", 
             50-strlen(cdata->rgb_ptr[rank[i-1]]));
       perc = my_sqrt((d[rank[i-1]]/19.5075));
       sprintf(cdata->match_list[i], format, 
@@ -446,7 +446,7 @@
     {
     numerical:
     sprintf(cdata->save, "#%02X%02X%02X",      
-          trunc(cdata->r), trunc(cdata->g), trunc(cdata->b));
+          mytrunc(cdata->r), mytrunc(cdata->g), mytrunc(cdata->b));
     }
   else
     {
@@ -454,9 +454,9 @@
     sscanf(cdata->match_ptr[0], "%d %d %d %s", &r, &g, &b, cdata->save);
     if (cdata->output==1)
       {
-      if (r!=trunc(cdata->r)
-             || g!=trunc(cdata->g)
-             || b!=trunc(cdata->b)) goto numerical;
+      if (r!=mytrunc(cdata->r)
+             || g!=mytrunc(cdata->g)
+             || b!=mytrunc(cdata->b)) goto numerical;
       }
     }
   if (mydepth<16)
@@ -818,10 +818,10 @@
     SetColor(cdata.color);
     }
 
-  SetThumbBitmap(cdata.red_scroll, black_bits, 8, 1);
-  SetThumbBitmap(cdata.green_scroll, black_bits, 8, 1);
-  SetThumbBitmap(cdata.blue_scroll, black_bits, 8, 1);
-  SetThumbBitmap(cdata.black_scroll, black_bits, 8, 1);
+  SetThumbBitmap(cdata.red_scroll, (char *) black_bits, 8, 1);
+  SetThumbBitmap(cdata.green_scroll, (char *) black_bits, 8, 1);
+  SetThumbBitmap(cdata.blue_scroll, (char *) black_bits, 8, 1);
+  SetThumbBitmap(cdata.black_scroll, (char *) black_bits, 8, 1);
 
   if (INPUTBG) 
     {
diff -Naur libsx-2.04/src/dialog.c libsx-2.04_new/src/dialog.c
--- libsx-2.04/src/dialog.c	2000-07-23 18:02:11.000000000 +0200
+++ libsx-2.04_new/src/dialog.c	2007-05-27 18:46:48.080386343 +0200
@@ -39,7 +39,7 @@
 
 static int selected;
 
-static int button_flags[] = { Yes, Abort, No, Okay, Cancel, Retry};
+static long button_flags[] = { Yes, Abort, No, Okay, Cancel, Retry};
 
 
 static void SetSelected(Widget w, XtPointer client_data, XtPointer call_data)
@@ -248,7 +248,6 @@
 void PopdownDialog(Dialog popup, char **answer)
 {
   char *tmp;
-  extern char *strdup(char *str);
   
     if (answer)
      {
diff -Naur libsx-2.04/src/dirlist.c libsx-2.04_new/src/dirlist.c
--- libsx-2.04/src/dirlist.c	1999-12-30 08:22:14.000000000 +0100
+++ libsx-2.04_new/src/dirlist.c	2007-05-27 18:47:05.893401450 +0200
@@ -32,7 +32,7 @@
 #include <sys/param.h>
 #include <sys/stat.h>
 
-extern char *strdup(char *str);
+// extern char *strdup(char *str);
 extern char _FreqFilter[84];
 extern int view_dir, view_pt;
 
diff -Naur libsx-2.04/src/draw.c libsx-2.04_new/src/draw.c
--- libsx-2.04/src/draw.c	1999-11-13 20:04:43.000000000 +0100
+++ libsx-2.04_new/src/draw.c	2007-05-27 18:43:44.005896536 +0200
@@ -547,7 +547,7 @@
 
 
 void 
-DrawImage(char *data, int x, int y, int width, int height)
+SX_DrawImage(char *data, int x, int y, int width, int height)
 {
   XImage *xi;
 
diff -Naur libsx-2.04/src/drawing.c libsx-2.04_new/src/drawing.c
--- libsx-2.04/src/drawing.c	1999-11-13 20:05:12.000000000 +0100
+++ libsx-2.04_new/src/drawing.c	2007-05-27 18:45:17.059199341 +0200
@@ -16,7 +16,7 @@
 #include "drawingP.h"
 
 #ifndef caddr_t
-extern char *caddr_t;
+// extern char *caddr_t;
 #endif
 
 static void	Initialize();
diff -Naur libsx-2.04/src/freq.c libsx-2.04_new/src/freq.c
--- libsx-2.04/src/freq.c	2000-07-29 11:19:31.000000000 +0200
+++ libsx-2.04_new/src/freq.c	2007-05-27 18:47:19.378169901 +0200
@@ -60,7 +60,7 @@
  * in the butt.  Why on earth isn't strdup() in the POSIX standard
  * but something completely useless like mbstowcs() is?
  */
-char *strdup(const char *str);
+// char *strdup(const char *str);
 
 
 /*
diff -Naur libsx-2.04/src/grabpix.c libsx-2.04_new/src/grabpix.c
--- libsx-2.04/src/grabpix.c	1999-06-22 21:28:38.000000000 +0200
+++ libsx-2.04_new/src/grabpix.c	2007-05-27 18:43:44.005896536 +0200
@@ -138,7 +138,7 @@
    0x8f, 0x07, 0x77, 0x07, 0xfe, 0x0f, 0xfc, 0x1f, 0xf8, 0x3f, 0x00, 0x7f,
    0x00, 0xfe, 0x00, 0x7c, 0x00, 0x38, 0x00, 0x10};
 
-static void
+void
 InitCursors()
 {
 Pixmap cursor,mask;
@@ -155,7 +155,7 @@
                    lens_x_hot,lens_y_hot);
 }
 
-static void 
+void 
 SetupGC()
 {
     selectGCV.function = GXxor;
@@ -165,7 +165,7 @@
 		       &selectGCV);
 }  
 
-static Window 
+Window 
 FindWindow(x, y)
      int x, y;	  
 {
@@ -187,14 +187,14 @@
   return findW;
 }
 
-static void
+void
 CreateRoot()
 {
   root = XtCreateWidget("root", rootWindowWidgetClass, toplevel, NULL, 0);
   XtRealizeWidget(root);
 }
 
-static void
+void
 StartRootPtrGrab()   
 {
   Window    rootR, childR, window;
diff -Naur libsx-2.04/src/libsx.h libsx-2.04_new/src/libsx.h
--- libsx-2.04/src/libsx.h	2002-03-13 10:37:43.000000000 +0100
+++ libsx-2.04_new/src/libsx.h	2007-05-27 18:43:44.005896536 +0200
@@ -151,7 +151,7 @@
 void   DrawText(char *string, int x, int y);
 void   DrawArc(int x, int y, int width, int height, int angle1, int angle2);
 void   DrawFilledArc(int x, int y, int w, int h, int angle1, int angle2);
-void   DrawImage(char *data, int x, int y, int width, int height);
+void   SX_DrawImage(char *data, int x, int y, int width, int height);
 void   DrawBitmap(char *data, int x, int y, int width, int height);
 void   GetImage(char *data, int x, int y, int width, int height);
 
diff -Naur libsx-2.04/src/scrollbar.c libsx-2.04_new/src/scrollbar.c
--- libsx-2.04/src/scrollbar.c	1999-11-26 15:31:53.000000000 +0100
+++ libsx-2.04_new/src/scrollbar.c	2007-05-27 18:44:51.025715777 +0200
@@ -89,11 +89,11 @@
 static void
 my_scroll_proc(Widget scroll_widget, XtPointer client_data, XtPointer position)
 {
-  int   pos;
+  XtPointer   pos;
   ScrollInfo *si = (ScrollInfo *)client_data;
   float old_val = si->val;
   
-  pos = (int)position;
+  pos = position;
   
   if (pos < 0)   /* button 3 pressed, go up */
    {
