--- a/encrypt_hook	2013-02-25 10:43:10.852608671 +0100
+++ b/encrypt_hook	2013-02-25 10:46:16.813369721 +0100
@@ -74,13 +74,15 @@
             fi
             # Ask for a passphrase
             if [ ${dopassphrase} -gt 0 ]; then
-                echo ""
+                echo
                 echo "A password is required to access the ${cryptname} volume:"
 
                 #loop until we get a real password
                 while ! eval cryptsetup open --type luks ${resolved} ${cryptname} ${cryptargs} ${CSQUIET}; do
                     sleep 2;
                 done
+                plymouth ask-for-password --prompt="Password for ${cryptname} volume" --dont-pause-progress --number-of-tries=5 --command="/sbin/cryptsetup luksOpen --key-file=- ${cryptdev} ${cryptname} ${CSQUIET}"
+		sleep 2
             fi
             if [ -e "/dev/mapper/${cryptname}" ]; then
                 if [ ${DEPRECATED_CRYPT} -eq 1 ]; then
