--- systemd-units/plymouth-start.service.orig	2012-07-10 16:19:35.000000000 +0200
+++ systemd-units/plymouth-start.service	2012-08-09 03:49:32.020563270 +0200
@@ -5,9 +5,9 @@
 After=systemd-vconsole-setup.service systemd-udev-settle.service
 Before=systemd-ask-password-plymouth.service
 ConditionKernelCommandLine=!plymouth.enable=0
-ConditionPathExists=!/run/plymouth/pid
+ConditionPathExists=!/run/plymouth.pid
 
 [Service]
-ExecStart=/sbin/plymouthd --mode=boot --pid-file=/run/plymouth/pid --attach-to-session
+ExecStart=/sbin/plymouthd --mode=boot --pid-file=/run/plymouth.pid --attach-to-session
 ExecStartPost=-/bin/udevadm settle --timeout=30 --exit-if-exists=/sys/class/drm/card0/dev;-/bin/udevadm settle --timeout=30 --exit-if-exists=/sys/class/graphics/fb0/dev; -/bin/plymouth show-splash
 Type=forking
--- systemd-units/plymouth-start.service.in.orig	2012-06-20 17:57:03.000000000 +0200
+++ systemd-units/plymouth-start.service.in	2012-08-09 03:50:10.850352910 +0200
@@ -5,9 +5,9 @@
 After=systemd-vconsole-setup.service systemd-udev-settle.service
 Before=systemd-ask-password-plymouth.service
 ConditionKernelCommandLine=!plymouth.enable=0
-ConditionPathExists=!@plymouthruntimedir@/pid
+ConditionPathExists=!/run/plymouth.pid
 
 [Service]
-ExecStart=@PLYMOUTH_DAEMON_DIR@/plymouthd --mode=boot --pid-file=@plymouthruntimedir@/pid --attach-to-session
+ExecStart=@PLYMOUTH_DAEMON_DIR@/plymouthd --mode=boot --pid-file=/run/plymouth.pid --attach-to-session
 ExecStartPost=-/bin/udevadm settle --timeout=30 --exit-if-exists=/sys/class/drm/card0/dev;-/bin/udevadm settle --timeout=30 --exit-if-exists=/sys/class/graphics/fb0/dev; -@PLYMOUTH_CLIENT_DIR@/plymouth show-splash
 Type=forking
--- systemd-units/systemd-ask-password-plymouth.path.orig	2012-07-10 16:19:35.000000000 +0200
+++ systemd-units/systemd-ask-password-plymouth.path	2012-08-09 03:50:31.420241474 +0200
@@ -6,7 +6,7 @@
 After=plymouth-start.service
 Before=basic.target shutdown.target
 ConditionKernelCommandLine=!plymouth.enable=0
-ConditionPathExists=/run/plymouth/pid
+ConditionPathExists=/run/plymouth.pid
 
 [Path]
 DirectoryNotEmpty=/run/systemd/ask-password
--- systemd-units/systemd-ask-password-plymouth.path.in.orig	2012-07-09 17:54:42.000000000 +0200
+++ systemd-units/systemd-ask-password-plymouth.path.in	2012-08-09 03:50:46.290160916 +0200
@@ -6,7 +6,7 @@
 After=plymouth-start.service
 Before=basic.target shutdown.target
 ConditionKernelCommandLine=!plymouth.enable=0
-ConditionPathExists=/run/plymouth/pid
+ConditionPathExists=/run/plymouth.pid
 
 [Path]
 DirectoryNotEmpty=/run/systemd/ask-password
--- systemd-units/systemd-ask-password-plymouth.service.orig	2012-07-10 16:19:35.000000000 +0200
+++ systemd-units/systemd-ask-password-plymouth.service	2012-08-09 03:51:01.083414108 +0200
@@ -6,7 +6,7 @@
 After=plymouth-start.service
 Before=shutdown.target
 ConditionKernelCommandLine=!plymouth.enable=0
-ConditionPathExists=/run/plymouth/pid
+ConditionPathExists=/run/plymouth.pid
 
 [Service]
 ExecStart=/bin/systemd-tty-ask-password-agent --watch --plymouth
--- systemd-units/systemd-ask-password-plymouth.service.in.orig	2012-07-09 17:55:56.000000000 +0200
+++ systemd-units/systemd-ask-password-plymouth.service.in	2012-08-09 03:51:10.883361017 +0200
@@ -6,7 +6,7 @@
 After=plymouth-start.service
 Before=shutdown.target
 ConditionKernelCommandLine=!plymouth.enable=0
-ConditionPathExists=/run/plymouth/pid
+ConditionPathExists=/run/plymouth.pid
 
 [Service]
 ExecStart=/bin/systemd-tty-ask-password-agent --watch --plymouth
