diff -u --new-file qmail-1.03-orig/qmail-control.9 qmail-1.03/qmail-control.9
--- qmail-1.03-orig/qmail-control.9	2006-03-21 16:13:11.000000000 -0300
+++ qmail-1.03/qmail-control.9	2006-03-21 16:15:58.000000000 -0300
@@ -40,11 +40,19 @@
 .ta 5c 10c
 control	default	used by
 
+.I aliasempty	\fR./Mailbox	\fRqmail init
 .I badmailfrom	\fR(none)	\fRqmail-smtpd, qmail-qmtpd
+.I bindroutes	\fR(none)	\fRqmail-remote
 .I bouncefrom	\fRMAILER-DAEMON	\fRqmail-send
 .I bouncehost	\fIme	\fRqmail-send
+.I checkpassword	\fRcheckpassword	\fRpop3d, spop3d init script
 .I concurrencylocal	\fR10	\fRqmail-send
+.I concurrencypop3d	\fR20	\fRpop3d init script
+.I concurrencyqmqpd	\fR20	\fRqmqpd init script
+.I concurrencyqmtpd	\fR20	\fRqmtpd init script
 .I concurrencyremote	\fR20	\fRqmail-send
+.I concurrencysmtpd	\fR20	\fRsmtpd init script
+.I concurrencyspop3d	\fR20	\fRspop3d init script
 .I defaultdomain	\fIme	\fRqmail-inject
 .I defaulthost	\fIme	\fRqmail-inject
 .I databytes	\fR0	\fRqmail-smtpd
@@ -55,6 +63,7 @@
 .I idhost	\fIme	\fRqmail-inject
 .I localiphost	\fIme	\fRqmail-smtpd
 .I locals	\fIme	\fRqmail-send
+.I logger	\fRsplogger	\fRall init scripts
 .I morercpthosts	\fR(none)	\fRqmail-smtpd
 .I percenthack	\fR(none)	\fRqmail-send
 .I plusdomain	\fIme	\fRqmail-inject
@@ -66,12 +75,16 @@
 .I timeoutconnect	\fR60	\fRqmail-remote
 .I timeoutremote	\fR1200	\fRqmail-remote
 .I timeoutsmtpd	\fR1200	\fRqmail-smtpd
+.I ulimitdata	\f$unlimited	\f$init scripts
 .I virtualdomains	\fR(none)	\fRqmail-send
 .fi
 .RE
 .SH "SEE ALSO"
+qmail-init(8),
 qmail-inject(8),
 qmail-qmqpc(8),
+qmail-qmqpd(8),
+qmail-qmtpd(8),
 qmail-remote(8),
 qmail-send(8),
 qmail-showctl(8),
diff -u --new-file qmail-1.03-orig/qmail-remote.8 qmail-1.03/qmail-remote.8
--- qmail-1.03-orig/qmail-remote.8	2006-03-21 16:13:11.000000000 -0300
+++ qmail-1.03/qmail-remote.8	2006-03-21 16:16:42.000000000 -0300
@@ -114,6 +114,18 @@
 always exits zero.
 .SH "CONTROL FILES"
 .TP 5
+.I bindroutes
+Artificial interface binding routes.
+Each route has the form
+.IR remote-address\fB:\fIlocal-address ,
+without any extra spaces.
+If the address of the host to which we are delivering matches the remote
+address, then bind the socket to the interface matching the local
+address, making it appear as if the connection had originated from the
+named local address instead of the default local address.
+.I bindroutes
+may contain suffix wildcards.
+.TP 5
 .I helohost
 Current host name,
 for use solely in saying hello to the remote SMTP server.
Subdirectorios comunes: qmail-1.03-orig/qmail-rhinit y qmail-1.03/qmail-rhinit
diff -u --new-file qmail-1.03-orig/qmail-showctl.c qmail-1.03/qmail-showctl.c
--- qmail-1.03-orig/qmail-showctl.c	2006-03-21 16:13:11.000000000 -0300
+++ qmail-1.03/qmail-showctl.c	2006-03-21 16:18:46.000000000 -0300
@@ -214,11 +214,19 @@
     _exit(111);
   }
 
+  do_lst("aliasempty","./Mailbox","Default delivery target contains: ","");
   do_lst("badmailfrom","Any MAIL FROM is allowed.",""," not accepted in MAIL FROM.");
+  do_lst("bindroutes","No binding routes.","Binding route: ","");
   do_str("bouncefrom",0,"MAILER-DAEMON","Bounce user name is ");
   do_str("bouncehost",1,"bouncehost","Bounce host name is ");
+  do_str("checkpassword",0,"checkpassword","Password checking program is ");
   do_int("concurrencylocal","10","Local concurrency is ","");
+  do_int("concurrencypop3d","20","POP-3 daemon concurrency is ","");
+  do_int("concurrencyqmqpd","20","QMQP daemon concurrency is ","");
+  do_int("concurrencyqmtpd","20","QMTP daemon concurrency is ","");
   do_int("concurrencyremote","20","Remote concurrency is ","");
+  do_int("concurrencysmtpd","20","SMTP daemon concurrency is ","");
+  do_int("concurrencyspop3d","20","SSL POP-3 daemon concurrency is ","");
   do_int("databytes","0","SMTP DATA limit is "," bytes");
   do_str("defaultdomain",1,"defaultdomain","Default domain name is ");
   do_str("defaulthost",1,"defaulthost","Default host name is ");
@@ -229,6 +237,7 @@
   do_str("idhost",1,"idhost","Message-ID host name is ");
   do_str("localiphost",1,"localiphost","Local IP address becomes ");
   do_lst("locals","Messages for me are delivered locally.","Messages for "," are delivered locally.");
+  do_str("logger",0,"multilog t /var/log/{}","Logging is done via: ");
   do_str("me",0,"undefined! Uh-oh","My name is ");
   do_lst("percenthack","The percent hack is not allowed.","The percent hack is allowed for user%host@",".");
   do_str("plusdomain",1,"plusdomain","Plus domain name is ");
@@ -260,6 +269,7 @@
   do_int("timeoutconnect","60","SMTP client connection timeout is "," seconds");
   do_int("timeoutremote","1200","SMTP client data timeout is "," seconds");
   do_int("timeoutsmtpd","1200","SMTP server data timeout is "," seconds");
+  do_str("ulimitdata",0,"1500000","Maximum process data size in bytes is ");
   do_lst("virtualdomains","No virtual domains.","Virtual domain: ","");
 
   while (d = readdir(dir)) {
@@ -296,6 +306,16 @@
     if (str_equal(d->d_name,"timeoutremote")) continue;
     if (str_equal(d->d_name,"timeoutsmtpd")) continue;
     if (str_equal(d->d_name,"virtualdomains")) continue;
+    if (str_equal(d->d_name,"aliasempty")) continue;
+    if (str_equal(d->d_name,"bindroutes")) continue;
+    if (str_equal(d->d_name,"checkpassword")) continue;
+    if (str_equal(d->d_name,"concurrencypop3d")) continue;
+    if (str_equal(d->d_name,"concurrencyqmqpd")) continue;
+    if (str_equal(d->d_name,"concurrencyqmtpd")) continue;
+    if (str_equal(d->d_name,"concurrencysmtpd")) continue;
+    if (str_equal(d->d_name,"concurrencyspop3d")) continue;
+    if (str_equal(d->d_name,"logger")) continue;
+    if (str_equal(d->d_name,"ulimitdata")) continue;
     substdio_puts(subfdout,"\n");
     substdio_puts(subfdout,d->d_name);
     substdio_puts(subfdout,": I have no idea what this file does.\n");
