diff -ur src/dejasmin.java src-fixed/dejasmin.java
--- src/dejasmin.java	2006-02-05 23:27:12.000000000 +0100
+++ src-fixed/dejasmin.java	2007-11-28 15:17:49.000000000 +0100
@@ -22,7 +22,7 @@
 import org.apache.bcel.classfile.JavaClass;
 
 public class dejasmin {
-    private final static String version = "dejasmin v0.3";
+    private final static String version = "dejasmin v0.4";
 
     private RandomAccessFile datain;
 
