--- configure.ac.orig	2011-10-04 21:56:56.000000000 +0200
+++ configure.ac	2011-10-04 21:58:08.000000000 +0200
@@ -15,13 +15,16 @@
 AM_GLIB_GNU_GETTEXT
 IT_PROG_INTLTOOL([0.35.0])
 
-AM_PATH_PYTHON(2.7)
-AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(Could not find Python headers)])
-
 AC_ENABLE_SHARED(yes)
 AC_ENABLE_STATIC(no)
 
-PYTHON_LIBS=`python-config --libs`
+PYTHON="python2"
+PYTHON_VERSION="2.7"
+PYTHON_PREFIX="/usr/lib/python2.7"
+PYTHON_EXEC_PREFIX="/usr/bin"
+PYTHON_PLATFORM="linux2"
+PYTHON_LIBS="-lpthread -ldl -lutil -lm -lpython2.7"
+
 AC_SUBST(PYTHON_LIBS)
 
 AC_ARG_VAR([PYREXC], [Pyrex compiler])
--- module/Makefile.am.orig	2011-10-04 21:58:43.000000000 +0200
+++ module/Makefile.am	2011-10-04 21:59:13.000000000 +0200
@@ -2,9 +2,9 @@
 
 bluemanlibdir = $(pyexecdir)
 bluemanlib_LTLIBRARIES = _blueman.la
-_blueman_la_CFLAGS = $(BLUEZ_CFLAGS) @PYTHON_INCLUDES@ -DSN_API_NOT_YET_FROZEN
+_blueman_la_CFLAGS = $(BLUEZ_CFLAGS) -I/usr/include/python2.7 -DSN_API_NOT_YET_FROZEN
 _blueman_la_LDFLAGS = -module -avoid-version -fPIC
-_blueman_la_LIBADD = $(BLUEZ_LIBS) @PYTHON_LIBS@
+_blueman_la_LIBADD = $(BLUEZ_LIBS) -lpthread -ldl -lutil -lm -lpython2.7
 _blueman_la_SOURCES = \
 	_blueman.c			\
 	libblueman.c	\

--- sendto/Makefile.am.orig	2011-10-04 22:09:31.000000000 +0200
+++ sendto/Makefile.am	2011-10-04 22:09:58.000000000 +0200
@@ -8,8 +8,8 @@
 
 libnstblueman_la_SOURCES = nautilus-bluetooth-sendto.c
 libnstblueman_la_LDFLAGS = -module -avoid-version
-libnstblueman_la_LIBADD = $(BLUEZ_LIBS) @PYTHON_LIBS@
-libnstblueman_la_CFLAGS = @PYTHON_INCLUDES@ $(BLUEZ_CFLAGS)
+libnstblueman_la_LIBADD = $(BLUEZ_LIBS) -lpthread -ldl -lutil -lm -lpython2.7
+libnstblueman_la_CFLAGS = -I/usr/include/python2.7 $(BLUEZ_CFLAGS)
 
 endif
 
