Index: src/Akai.cpp
===================================================================
--- src/Akai.cpp        (Revision 2580)
+++ src/Akai.cpp        (Arbeitskopie)
@@ -1877,7 +1877,8 @@
   {
   #if defined(_CARBON_) || defined(__APPLE__) || LINUX
   const uint bufferSize = 524288; // 512 kB
-  int fOut = open(path, O_WRONLY | O_NONBLOCK | O_CREAT | O_TRUNC);
+  mode_t mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH;
+  int fOut = open(path, O_WRONLY | O_NONBLOCK | O_CREAT | O_TRUNC, mode);
   if (mFile <= 0) {
     printf("Can't open output file %s\n", path);
     return false;

