diff -ur libgig-svn/akai.pc.in /home/moritz/tmp/libgig-svn/akai.pc.in
--- akai.pc.in	2014-05-28 23:50:43.203032933 +0200
+++ akai.pc.in	2014-05-28 23:49:53.363354734 +0200
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
+libdir=@libdir@/libgig
 includedir=@includedir@
 
 Name: akai
diff -ur libgig-svn/gig.pc.in /home/moritz/tmp/libgig-svn/gig.pc.in
--- gig.pc.in	2014-05-28 21:56:20.379678561 +0200
+++ gig.pc.in	2014-05-28 23:51:33.979371882 +0200
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
+libdir=@libdir@/libgig
 includedir=@includedir@
 
 Name: gig
