=== modified file 'libgwibber-gtk/stream-view-tile.vala'
--- old/libgwibber-gtk/stream-view-tile.vala	2011-10-22 02:27:58 +0000
+++ new/libgwibber-gtk/stream-view-tile.vala	2011-11-10 18:47:33 +0000
@@ -144,9 +144,11 @@
       time = new Gtk.Label ("");
       time.set_markup ("");
       time.set_alignment (0.0f, 1.0f);
+      time.set_size_request (16, 16);
       hbox.pack_end (time, false, false, 0);
 
       action_box = new GwibberGtk.ActionBox ();
+      action_box.set_size_request (16, 16);
       action_box.hide ();
       //action_box.set_no_show_all (true);
       hbox.pack_end (action_box, false, false, 0);

