diff -ur source/tint2-read-only/src//config.c Hack/src//config.c
--- source/tint2-read-only/src//config.c	2010-08-26 02:25:34.786473252 +0200
+++ Hack/src//config.c	2010-08-26 02:15:22.043215408 +0200
@@ -416,6 +416,7 @@
 	/* Taskbar */
 	else if (strcmp (key, "taskbar_mode") == 0) {
 		if (strcmp (value, "multi_desktop") == 0) panel_mode = MULTI_DESKTOP;
+		else if (strcmp (value, "viewports") == 0) panel_mode = VIEWPORTS;
 		else panel_mode = SINGLE_DESKTOP;
 	}
 	else if (strcmp (key, "taskbar_padding") == 0) {
Seulement dans Hack/src/: config.c.orig
diff -ur source/tint2-read-only/src//panel.h Hack/src//panel.h
--- source/tint2-read-only/src//panel.h	2010-08-26 02:25:34.786473252 +0200
+++ Hack/src//panel.h	2010-08-26 02:15:02.059865050 +0200
@@ -37,7 +37,7 @@
 extern int mouse_tilt_right;
 
 //panel mode
-enum { SINGLE_DESKTOP=0, MULTI_DESKTOP };
+enum { SINGLE_DESKTOP=0, MULTI_DESKTOP , VIEWPORTS};
 enum { BOTTOM_LAYER, NORMAL_LAYER, TOP_LAYER };
 extern int panel_mode;
 extern int wm_menu;
Seulement dans Hack/src/: panel.h.orig
diff -ur source/tint2-read-only/src//taskbar/taskbar.c Hack/src//taskbar/taskbar.c
--- source/tint2-read-only/src//taskbar/taskbar.c	2010-08-26 02:25:34.033991079 +0200
+++ Hack/src//taskbar/taskbar.c	2010-08-26 02:16:17.159875973 +0200
@@ -232,6 +232,29 @@
 		return 0;
 }
 
+int is_window_visible(Window win)
+{
+	Panel *panel = get_panel(win);
+	int x_pos, y_pos;
+	unsigned int width, height, others, monitor_width, monitor_height;
+	XGetGeometry(server.dsp, win, &win, &x_pos, &y_pos, &width, &height, &others, &others);
+	if(panel == 0)
+	{
+		monitor_height = server.monitor[0].height;
+		monitor_width = server.monitor[0].width; 
+	}
+	else
+	{
+		monitor_width = server.monitor[panel->monitor].width;
+		monitor_height = server.monitor[panel->monitor].height;
+	}
+	if(x_pos < (int)monitor_width && x_pos+(int)width > 0 && y_pos < (int)monitor_height && y_pos+(int)height > 0)
+		return 1;
+	else
+		return 0;
+	
+}
+
 
 void task_refresh_tasklist ()
 {
@@ -246,17 +269,20 @@
 	for (it=win_list; it; it=it->next) {
 		for (i = 0; i < num_results; i++)
 			if (*((Window*)it->data) == win[i])
-				break;
+			{
+				if( panel_mode == VIEWPORTS && !is_window_visible(win[i]) ) 
+					taskbar_remove_task(it->data, 0, 0);
+				else
+					break;
+			}
 		if (i == num_results)
 			taskbar_remove_task(it->data, 0, 0);
 	}
 	g_list_free(win_list);
-
 	// Add any new
 	for (i = 0; i < num_results; i++)
-		if (!task_get_task (win[i]))
+		if (!task_get_task (win[i]) && ( panel_mode != VIEWPORTS || is_window_visible(win[i])) )
 			add_task (win[i]);
-
 	XFree (win);
 }
 
Seulement dans Hack/src//taskbar: taskbar.c.orig
diff -ur source/tint2-read-only/src//taskbar/taskbar.h Hack/src//taskbar/taskbar.h
--- source/tint2-read-only/src//taskbar/taskbar.h	2010-08-26 02:25:34.033991079 +0200
+++ Hack/src//taskbar/taskbar.h	2010-08-26 02:16:06.459877783 +0200
@@ -51,6 +51,7 @@
 
 void resize_taskbar(void *obj);
 
+int is_window_visible(Window win); 
 
 #endif
 
diff -ur source/tint2-read-only/src//tint2conf/tintwizard.py Hack/src//tint2conf/tintwizard.py
--- source/tint2-read-only/src//tint2conf/tintwizard.py	2010-08-26 02:25:33.916454395 +0200
+++ Hack/src//tint2conf/tintwizard.py	2010-08-26 02:16:45.763206328 +0200
@@ -535,7 +535,7 @@
 		self.tableTaskbar.set_col_spacings(5)
 
 		createLabel(self.tableTaskbar, text="Taskbar Mode", gridX=0, gridY=0, xPadding=10)
-		self.taskbarMode = createComboBox(self.tableTaskbar, ["single_desktop", "multi_desktop"], gridX=1, gridY=0, handler=self.changeOccurred)
+		self.taskbarMode = createComboBox(self.tableTaskbar, ["single_desktop", "multi_desktop", "viewports"], gridX=1, gridY=0, handler=self.changeOccurred)
 		self.registerComponent("taskbar_mode", self.taskbarMode)
 
 		createLabel(self.tableTaskbar, text="Padding (x, y)", gridX=0, gridY=1, xPadding=10)
@@ -1662,7 +1662,7 @@
 			prop.activate()
 		elif eType == gtk.ComboBox:
 			# This allows us to select the correct combo-box value.
-			if string in ["bottom", "top", "left", "right", "center", "single_desktop", "multi_desktop", "single_monitor",
+			if string in ["bottom", "top", "left", "right", "center", "single_desktop", "multi_desktop", "viewports", "single_monitor",
 							"none", "close", "shade", "iconify", "toggle", "toggle_iconify", "maximize_restore",
 							"desktop_left", "desktop_right", "horizontal", "vertical", "ascending", "descending",
 							"left2right", "right2left", "next_task", "prev_task", "minimum", "follow_size", "normal"]:
@@ -1670,7 +1670,7 @@
 					i = 0
 				elif string in ["top", "right", "multi_desktop", "close", "vertical", "descending", "minimum"]:
 					i = 1
-				elif string in ["center", "single_monitor", "toggle", "left2right", "follow_size", "normal"]:
+				elif string in ["center","viewports" , "single_monitor", "toggle", "left2right", "follow_size", "normal"]:
 					i = 2
 				elif string in ["right2left"]:
 					i = 3
diff -ur source/tint2-read-only/src//tint.c Hack/src//tint.c
--- source/tint2-read-only/src//tint.c	2010-08-26 02:25:34.786473252 +0200
+++ Hack/src//tint.c	2010-08-26 02:15:45.916546968 +0200
@@ -553,6 +553,8 @@
 		// Change active
 		else if (at == server.atom._NET_ACTIVE_WINDOW) {
 			active_task();
+			if(panel_mode == VIEWPORTS) 
+				task_refresh_tasklist();
 			panel_refresh = 1;
 		}
 		else if (at == server.atom._XROOTPMAP_ID) {
