commit 1154a3d3e4c1d92aa06305557fb6e8dd7568df64
Author: Benjamin Chrétien <chretien@lirmm.fr>
Date:   Tue Mar 25 20:37:56 2014 +0100

    Do not generate ruby messages for rosgrah_msgs.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b967e54..4ede85a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,7 @@ catkin_add_env_hooks(${PROJECT_NAME} SHELLS sh DIRECTORY ${CMAKE_CURRENT_SOURCE_
 
 file(MAKE_DIRECTORY ${ROSRUBY_DEVEL_LIB_DESTINATION})
 
-rosruby_generate_messages(std_msgs rosgraph_msgs)
+rosruby_generate_messages(std_msgs)
 
 install(PROGRAMS
   bin/rubyroscore
