diff -ur anyterm-1.1.29.backup//libpbe/src/SmtpClient.cc anyterm-1.1.29/libpbe/src/SmtpClient.cc
--- anyterm-1.1.29.backup//libpbe/src/SmtpClient.cc	2010-04-24 22:10:22.163990187 -0400
+++ anyterm-1.1.29/libpbe/src/SmtpClient.cc	2010-04-24 22:11:36.664229627 -0400
@@ -22,6 +22,8 @@
 
 #include <syslog.h>
 #include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
 
 #ifdef __OpenBSD__
 // Is this really needed?
