diff -Naur noiz2sa-0.51a.orig/src/barragemanager.h noiz2sa-0.51a.new/src/barragemanager.h
--- noiz2sa-0.51a.orig/src/barragemanager.h	2003-08-10 13:09:46.000000000 -0700
+++ noiz2sa-0.51a.new/src/barragemanager.h	2008-12-30 15:11:00.000000000 -0800
@@ -12,9 +12,9 @@
 #ifndef BARRAGEMANAGER_H_
 #define BARRAGEMANAGER_H_
 
-#include "bulletml/bulletmlparser.h"
-#include "bulletml/bulletmlparser-tinyxml.h"
-#include "bulletml/bulletmlrunner.h"
+#include <bulletml/bulletmlparser.h>
+#include <bulletml/bulletmlparser-tinyxml.h>
+#include <bulletml/bulletmlrunner.h>
 
 #define BARRAGE_TYPE_NUM 3
 #define BARRAGE_MAX 16
