--- joy2key.c.orig	2005-01-27 23:35:01.000000000 +0300
+++ joy2key.c	2009-12-08 12:55:16.000000000 +0300
@@ -68,7 +68,7 @@
 
 #define DEFAULT_AUTOREPEAT             5
 #define DEFAULT_DEADZONE               50
-#define DEFAULT_DEVICE                 "/dev/js0"
+#define DEFAULT_DEVICE                 "/dev/input/js0"
 #define DEFAULT_RCFILE                 ".joy2keyrc" /* located in $(HOME) */
 #define PETERS_EMAIL                   "tetron@interreality.org"
 
