diff -Naur original/gtk/CMakeLists.txt modified/gtk/CMakeLists.txt
--- original/gtk/CMakeLists.txt	2014-11-07 19:52:59.000000000 +0000
+++ modified/gtk/CMakeLists.txt	2014-11-07 19:52:28.000000000 +0000
@@ -26,9 +26,11 @@
 
         if (USE_METACITY)
             pkg_check_modules (METACITY libmetacity-private>=3.12.0)
-            if (NOT METACITY_FOUND)
+            if (METACITY_FOUND)
+                compiz_pkg_check_modules (HAVE_METACITY_3_14_0 libmetacity-private>=3.14.0)
+            else (METACITY_FOUND)
                 compiz_set (USE_METACITY 0)
-            endif (NOT METACITY_FOUND)
+            endif (METACITY_FOUND)
         endif (USE_METACITY)
 
 	if (USE_GNOME)
diff -Naur original/gtk/config.h.gtk.in modified/gtk/config.h.gtk.in
--- original/gtk/config.h.gtk.in	2014-11-07 19:52:59.000000000 +0000
+++ modified/gtk/config.h.gtk.in	2014-11-07 19:52:28.000000000 +0000
@@ -1,4 +1,7 @@
 /* Define to 1 if Metacity support is enabled */
 #cmakedefine USE_METACITY 1
 
+/* Define to 1 if metacity version >= 3.14.0 */
+#cmakedefine HAVE_METACITY_3_14_0 1
+
 #define GETTEXT_PACKAGE "${GETTEXT_PACKAGE}"
diff -Naur original/gtk/window-decorator/metacity.c modified/gtk/window-decorator/metacity.c
--- original/gtk/window-decorator/metacity.c	2014-11-07 19:52:59.000000000 +0000
+++ modified/gtk/window-decorator/metacity.c	2014-11-07 19:52:28.000000000 +0000
@@ -451,6 +451,9 @@
 			      MetaFrameType	frame_type,
 			      GdkRectangle      *clip)
 {
+#ifdef HAVE_METACITY_3_14_0
+    MetaFrameBorders borders;
+#endif
     gint left_width, right_width, top_height, bottom_height;
 
     if (!(frame_type < META_FRAME_TYPE_LAST))
@@ -525,6 +528,17 @@
     if (d->state & WNCK_WINDOW_STATE_ABOVE)
 	*flags |= (MetaFrameFlags ) META_FRAME_ABOVE;
 
+#ifdef HAVE_METACITY_3_14_0
+    meta_theme_get_frame_borders (theme,
+				  frame_type,
+				  d->frame->text_height,
+				  *flags,
+				  &borders);
+    top_height = borders.visible.top;
+    bottom_height = borders.visible.bottom;
+    left_width = borders.visible.left;
+    right_width = borders.visible.right;
+#else
     meta_theme_get_frame_borders (theme,
 				  frame_type,
 				  d->frame->text_height,
@@ -533,6 +547,7 @@
 				  &bottom_height,
 				  &left_width,
 				  &right_width);
+#endif
 
     clip->x = d->context->left_space - left_width;
     clip->y = d->context->top_space - top_height;
@@ -1552,6 +1567,9 @@
 meta_update_border_extents (decor_frame_t *frame)
 {
     MetaTheme *theme = meta_theme_get_current ();
+#ifdef HAVE_METACITY_3_14_0
+    MetaFrameBorders borders;
+#endif
 
     gwd_decor_frame_ref (frame);
     MetaFrameType frame_type = meta_frame_type_from_string (frame->type);
@@ -1560,6 +1578,17 @@
     if (!(frame_type < META_FRAME_TYPE_LAST))
 	frame_type = META_FRAME_TYPE_NORMAL;
 
+#ifdef HAVE_METACITY_3_14_0
+    meta_theme_get_frame_borders (theme,
+				  frame_type,
+				  frame->text_height,
+				  0,
+				  &borders);
+    top_height = borders.visible.top;
+    bottom_height = borders.visible.bottom;
+    left_width = borders.visible.left;
+    right_width = borders.visible.right;
+#else
     meta_theme_get_frame_borders (theme,
 				  frame_type,
 				  frame->text_height,
@@ -1568,6 +1597,7 @@
 				  &bottom_height,
 				  &left_width,
 				  &right_width);
+#endif
 
     frame->win_extents.top    = frame->win_extents.top;
     frame->win_extents.bottom = bottom_height;
@@ -1576,6 +1606,17 @@
 
     frame->titlebar_height = top_height - frame->win_extents.top;
 
+#ifdef HAVE_METACITY_3_14_0
+    meta_theme_get_frame_borders (theme,
+				  frame_type,
+				  frame->text_height,
+				  META_FRAME_MAXIMIZED,
+				  &borders);
+    top_height = borders.visible.top;
+    bottom_height = borders.visible.bottom;
+    left_width = borders.visible.left;
+    right_width = borders.visible.right;
+#else
     meta_theme_get_frame_borders (theme,
 				  frame_type,
 				  frame->text_height,
@@ -1584,6 +1625,7 @@
 				  &bottom_height,
 				  &left_width,
 				  &right_width);
+#endif
 
     frame->max_win_extents.top    = frame->win_extents.top;
     frame->max_win_extents.bottom = bottom_height;
