diff -aur i3lock-2.4.1/i3lock.c i3lock-2.4.1.new/i3lock.c
--- i3lock-2.4.1/i3lock.c	2012-06-02 15:45:42.000000000 +0300
+++ i3lock-2.4.1.new/i3lock.c	2013-01-07 23:32:27.000000000 +0300
@@ -286,7 +286,8 @@
 
     /* Whether Caps Lock (all lowercase alphabetic keys will be replaced by
      * their uppercase variant) is active at the moment. */
-    bool capslock = (event->state & capslockmask);
+    //I don`t want to convert lowercase to uppercase
+    bool capslock = false;
 
     DEBUG("shift = %d, capslock = %d\n",
           shift, capslock);