diff -Naur freetype-2.5.0.1/include/freetype/config/ftoption.h freetype-2.5.0.1.new/include/freetype/config/ftoption.h
--- freetype-2.5.0.1/include/config/ftoption.h	2013-06-19 17:20:04.000000000 -0400
+++ freetype-2.5.0.1.new/include/config/ftoption.h	2013-09-15 12:29:22.332178007 -0400
@@ -673,7 +673,7 @@
   /*   http://www.microsoft.com/typography/otspec/glyf.htm                 */
   /*   http://fonts.apple.com/TTRefMan/RM06/Chap6glyf.html                 */
   /*                                                                       */
-#undef TT_CONFIG_OPTION_COMPONENT_OFFSET_SCALED
+#define TT_CONFIG_OPTION_COMPONENT_OFFSET_SCALED
 
 
   /*************************************************************************/
