diff -Naur freetype-2.5.0.1/include/freetype/config/ftoption.h freetype-2.5.0.1.new/include/freetype/config/ftoption.h
--- freetype-2.5.0.1/include/config/ftoption.h	2013-06-19 17:20:04.000000000 -0400
+++ freetype-2.5.0.1.new/include/config/ftoption.h	2013-09-15 12:38:16.386761852 -0400
@@ -92,7 +92,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
