diff -rup graphcalc/linux/Makefile graphcalc.new/linux/Makefile
--- graphcalc/linux/Makefile	2012-08-29 11:20:16.330264152 -0400
+++ graphcalc.new/linux/Makefile	2012-08-29 11:20:41.090794033 -0400
@@ -19,7 +19,7 @@ YACCFLAGS= -d
 INCPATH  = -I/opt/qt/mkspecs/default -I. -I/usr/kde/3.1/include -I$(QTDIR)/include -I.ui/ -I. -I.moc/
 LINK     = g++
 LFLAGS   = 
-LIBS     = $(SUBLIBS) -L$(QTDIR)/lib -L/usr/X11R6/lib64 -L/usr/lib -L/usr/kde/3.1/lib -lginac -lcln -lgmp -lqt -lXext -lX11 -lm
+LIBS     = $(SUBLIBS) -L$(QTDIR)/lib -L/usr/X11R6/lib64 -L/usr/lib -L/usr/kde/3.1/lib -lginac -lcln -lgmp -lqt-mt -lXext -lX11 -lm
 AR       = ar cqs
 RANLIB   = 
 MOC      = $(QTDIR)/bin/moc
