diff -urN orig/gscan2pdf-0.9.31//bin/gscan2pdf gscan2pdf-0.9.31//bin/gscan2pdf
--- orig/gscan2pdf-0.9.31//bin/gscan2pdf	2010-07-12 09:37:05.000000000 -0400
+++ gscan2pdf-0.9.31//bin/gscan2pdf	2010-08-02 15:48:35.234501330 -0400
@@ -128,8 +128,8 @@
 
 use warnings;
 use strict;
-use forks 0.33;
-use forks::shared;
+use threads;
+use threads::shared;
 use Thread::Semaphore;
 use Gscan2pdf;
 use Gtk2 -init;
