*** milkytracker-0.90.85/src/tracker/sdl/SDL_Main.cpp	2009-02-22 10:20:39.140388000 +0000
--- milkytracker-trunk/src/tracker/sdl/SDL_Main.cpp	2010-01-03 18:11:52.541377210 +0000
***************
*** 751,758 ****
  		case SDLUserEventLMouseRepeat:
  		{
  			PPPoint p;
! 			p.x = (pp_int32)event.data1;
! 			p.y = (pp_int32)event.data2;
  			PPEvent myEvent(eLMouseRepeat, &p, sizeof(PPPoint));		
  			RaiseEventSerialized(&myEvent);
  			break;
--- 751,758 ----
  		case SDLUserEventLMouseRepeat:
  		{
  			PPPoint p;
! 			p.x = reinterpret_cast<intptr_t> (event.data1);
! 			p.y = reinterpret_cast<intptr_t> (event.data2);
  			PPEvent myEvent(eLMouseRepeat, &p, sizeof(PPPoint));		
  			RaiseEventSerialized(&myEvent);
  			break;
***************
*** 761,768 ****
  		case SDLUserEventRMouseRepeat:
  		{
  			PPPoint p;
! 			p.x = (pp_int32)event.data1;
! 			p.y = (pp_int32)event.data2;
  			PPEvent myEvent(eRMouseRepeat, &p, sizeof(PPPoint));		
  			RaiseEventSerialized(&myEvent);
  			break;
--- 761,768 ----
  		case SDLUserEventRMouseRepeat:
  		{
  			PPPoint p;
! 			p.x = reinterpret_cast<intptr_t> (event.data1);
! 			p.y = reinterpret_cast<intptr_t> (event.data2);
  			PPEvent myEvent(eRMouseRepeat, &p, sizeof(PPPoint));		
  			RaiseEventSerialized(&myEvent);
  			break;
***************
*** 770,777 ****
  
  		case SDLUserEventMidiKeyDown:
  		{
! 			pp_int32 note = (pp_int32)event.data1;
! 			pp_int32 volume = (pp_int32)event.data2;
  			globalMutex->lock();
  			myTracker->sendNoteDown(note, volume);
  			globalMutex->unlock();
--- 770,777 ----
  
  		case SDLUserEventMidiKeyDown:
  		{
! 			pp_int32 note = reinterpret_cast<intptr_t> (event.data1);
! 			pp_int32 volume = reinterpret_cast<intptr_t> (event.data2);
  			globalMutex->lock();
  			myTracker->sendNoteDown(note, volume);
  			globalMutex->unlock();
***************
*** 780,786 ****
  
  		case SDLUserEventMidiKeyUp:
  		{
! 			pp_int32 note = (pp_int32)event.data1;
  			globalMutex->lock();
  			myTracker->sendNoteUp(note);
  			globalMutex->unlock();
--- 780,786 ----
  
  		case SDLUserEventMidiKeyUp:
  		{
! 			pp_int32 note = reinterpret_cast<intptr_t> (event.data1);
  			globalMutex->lock();
  			myTracker->sendNoteUp(note);
  			globalMutex->unlock();
