--- old/ocroinst	2012-06-05 17:03:42.757205525 +0200
+++ new/ocroinst	2012-06-07 17:51:06.530243089 +0200
@@ -5,7 +5,7 @@
 default_download=${OCRODOWNLOAD:-http://www-user.rhrk.uni-kl.de/~tmb}
 default_cmodels="uw3unlv-240-4-60-g2.cmodel uw3unlv-240-8-40-nogeo.cmodel space.model space2.model default.fst none.fst case.fst" 
 default_cmodel=uw3unlv-240-8-40-nogeo.cmodel
-datadir=/usr/local/share/ocropus
+datadir=$PKGDIR/usr/share/ocropus
 
 die() {
     echo ERROR "$*"
