*** bnbt.h	2006-07-27 17:27:30.000000000 +0200
--- newbnbt.h	2009-05-02 16:08:45.001310132 +0200
***************
*** 5,12 ****
  #ifndef BNBT_H
   #define BNBT_H
  
! #include <stdio.h>
! #include <time.h>
  
  #ifdef WIN32
   #define WIN32_LEAN_AND_MEAN
--- 5,13 ----
  #ifndef BNBT_H
   #define BNBT_H
  
! #include <cstdio>
! #include <ctime>
! #include <cstring>
  
  #ifdef WIN32
   #define WIN32_LEAN_AND_MEAN
