--- wlassistant-0.5.5/bksys/kde.py.BAD	2006-02-28 19:00:25.000000000 -0600
+++ wlassistant-0.5.5/bksys/kde.py	2006-02-28 19:01:14.000000000 -0600
@@ -25,10 +25,7 @@
 		lenv.Append( CPPFLAGS = '-DQT_NO_TRANSLATION' )
 	if 'rpath' in _flags:
 		## Use this to set rpath - this may cause trouble if folders are moved (chrpath)
-		kdelibpaths=[]
-		if lenv['KDELIBPATH'] == lenv['KDELIB']: kdelibpaths = [lenv['KDELIB']]
-		else: kdelibpaths = [lenv['KDELIBPATH'],  lenv['KDELIB']]
-		lenv.Append( RPATH = [lenv['QTLIBPATH'], lenv['KDEMODULE']]+kdelibpaths )
+		lenv.Append( RPATH = [lenv['QTLIBPATH'], lenv['KDEMODULE']] )
 	if 'thread' in _flags:
 		## Uncomment the following if you need threading support
 		lenv.KDEaddflags_cxx( ['-DQT_THREAD_SUPPORT', '-D_REENTRANT'] )
