--- Csound5.09.0/SConstruct.orig	2008-09-14 19:15:20.000000000 +0000
+++ Csound5.09.0/SConstruct	2008-12-16 22:04:42.000000000 +0000
@@ -361,11 +361,7 @@
     fltkFound = False
     portaudioFound = False
     oscFound = True
-    ##commonEnvironment['gcc4opt'] = '1'
-    commonEnvironment['gcc3opt'] = 'k6'
     commonEnvironment['useLrint'] = '1'
-    commonEnvironment['Word64'] = '0'
-    commonEnvironment['Lib64'] = '0'
     commonEnvironment['buildPDClass'] = '0'
     commonEnvironment['buildDSSI'] = '0'
     commonEnvironment['buildVirtual'] = '1'
--- Csound5.09.0/install-olpc.py.orig	2008-05-29 10:12:38.000000000 +0000
+++ Csound5.09.0/install-olpc.py	2008-12-16 22:36:58.000000000 +0000
@@ -58,6 +58,8 @@
             raise SystemExit(0)
         elif sys.argv[i][:9] == '--prefix=':
             prefix = sys.argv[i][9:]
+        elif sys.argv[i][:9] == '--libdir=':
+            libdir = sys.argv[i][9:]
         elif sys.argv[i][:10] == '--instdir=':
             instDir = sys.argv[i][10:]
         elif sys.argv[i][:17] == '--install-headers':
@@ -92,7 +94,10 @@
 # Csound API header files
 includeDir  = concatPath([prefix, '/include/csound'])
 # Csound API libraries
-libDir      = concatPath([prefix, '/lib'])
+if libdir:
+    libDir  = libdir
+else:
+    libDir  = concatPath([prefix, '/lib'])
 # single precision plugin libraries
 pluginDir32 = concatPath([libDir, '/csound/plugins'])
 # documentation
@@ -100,7 +105,7 @@
 # locale
 localeDir   = concatPath([prefix, '/share/locale'])
 # python module
-pythonDir   = concatPath([prefix, '/lib/python%s/site-packages' % (pyVersion)])
+pythonDir   = concatPath([libDir, '/python%s/site-packages' % (pyVersion)])
 
 def runCmd(args):
     return os.spawnvp(os.P_WAIT, args[0], args)
