diff --git a/SConstruct b/SConstruct
index 0b4755d..6f2eab2 100644
--- a/SConstruct
+++ b/SConstruct
@@ -32,7 +32,7 @@ if not env['verbose']:
 
 env.Append(CXXFLAGS="-std=c++0x",
            CFLAGS="-std=c99",
-           CPPPATH=Split("src chipmunk/include/chipmunk"))
+           CPPPATH=Split("src /usr/include/chipmunk/"))
 lib = env.Library(source=Glob("src/*.cpp") + Glob("src/chipmunkpp/*.cpp"),
                   target="chipmunk++")
 env.Append(LIBPATH=['chipmunk/src'], LIBS=[lib, 'chipmunk'])
