--- llvm/utils/llvm-build/llvm-build	2012-10-22 21:14:58.148087630 +0300
+++ llvm/utils/llvm-build/llvm-build	2012-10-22 21:15:01.888067419 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import llvmbuild
 
--- clang/tools/CMakeLists.txt	2012-10-25 07:04:10.215529005 +0300
+++ clang/tools/CMakeLists.txt	2012-10-25 07:04:33.148738133 +0300
@@ -14,6 +14,8 @@ if(CLANG_ENABLE_STATIC_ANALYZER)
   add_subdirectory(clang-check)
 endif()
 
+add_subdirectory(include-what-you-use)
+
 # We support checking out the clang-tools-extra repository into the 'extra'
 # subdirectory. It contains tools developed as part of the Clang/LLVM project
 # on top of the Clang tooling platform. We keep them in a separate repository
--- clang/tools/Makefile	2013-08-18 22:02:08.020429268 +0300
+++ clang/tools/Makefile	2013-08-18 22:05:02.080848540 +0300
@@ -27,6 +27,8 @@ ifeq ($(ENABLE_CLANG_ARCMT), 1)
   PARALLEL_DIRS += arcmt-test
 endif
 
+PARALLEL_DIRS += include-what-you-use
+
 # Recurse into the extra repository of tools if present.
 OPTIONAL_PARALLEL_DIRS := extra
 
--- include-what-you-use/fix_includes.py	(revision 468)
+++ include-what-you-use/fix_includes.py	(working copy)
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 ##===--- fix_includes.py - rewrite source files based on iwyu output ------===##
 #
