Index: llvm/utils/llvm-build/llvm-build
===================================================================
--- llvm/utils/llvm-build/llvm-build	(revision 215281)
+++ llvm/utils/llvm-build/llvm-build	(working copy)
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import llvmbuild
 
Index: clang/tools/CMakeLists.txt
===================================================================
--- clang/tools/CMakeLists.txt	(revision 215281)
+++ clang/tools/CMakeLists.txt	(working copy)
@@ -10,6 +10,7 @@ if(CLANG_ENABLE_ARCMT)
   add_subdirectory(arcmt-test)
   add_subdirectory(c-arcmt-test)
 endif()
+add_subdirectory(include-what-you-use)
 
 if(CLANG_ENABLE_STATIC_ANALYZER)
   add_subdirectory(clang-check)
Index: clang/tools/Makefile
===================================================================
--- clang/tools/Makefile	(revision 215281)
+++ clang/tools/Makefile	(working copy)
@@ -32,4 +32,6 @@ ifeq ($(BUILD_CLANG_ONLY),YES)
   OPTIONAL_PARALLEL_DIRS :=
 endif
 
+PARALLEL_DIRS += include-what-you-use
+
 include $(CLANG_LEVEL)/Makefile
Index: include-what-you-use/CMakeLists.txt
===================================================================
--- include-what-you-use/CMakeLists.txt	(revision 559)
+++ include-what-you-use/CMakeLists.txt	(working copy)
@@ -50,9 +50,10 @@ endif()
 
 # Pick up SVN revision so we can report it
 # in version information.
-include(FindSubversion)
-if( Subversion_FOUND AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.svn" )
-  Subversion_WC_INFO(${CMAKE_CURRENT_SOURCE_DIR} IWYU)
+find_package(Subversion)
+get_filename_component(SOURCE_DIR_REALPATH ${CMAKE_CURRENT_SOURCE_DIR} REALPATH)
+if( Subversion_FOUND AND EXISTS "${SOURCE_DIR_REALPATH}/.svn" )
+  Subversion_WC_INFO(${SOURCE_DIR_REALPATH} IWYU)
 endif()
 add_definitions(
   -DIWYU_SVN_REVISION="${IWYU_WC_REVISION}"
Index: include-what-you-use/fix_includes.py
===================================================================
--- include-what-you-use/fix_includes.py	(revision 559)
+++ include-what-you-use/fix_includes.py	(working copy)
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 ##===--- fix_includes.py - rewrite source files based on iwyu output ------===##
 #
