--- llvm/utils/llvm-build/llvm-build	2012-10-22 21:14:58.148087630 +0300
+++ llvm/utils/llvm-build/llvm-build	2012-10-22 21:15:01.888067419 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 import llvmbuild
 
--- clang/tools/CMakeLists.txt	2012-10-25 07:04:10.215529005 +0300
+++ clang/tools/CMakeLists.txt	2012-10-25 07:04:33.148738133 +0300
@@ -6,6 +6,7 @@
 add_subdirectory(driver)
 add_subdirectory(clang-check)
 add_subdirectory(clang-format)
+add_subdirectory(include-what-you-use)
 
 # We support checking out the clang-tools-extra repository into the 'extra'
 # subdirectory. It contains tools developed as part of the Clang/LLVM project
--- clang/tools/Makefile	2013-08-18 22:02:08.020429268 +0300
+++ clang/tools/Makefile	2013-08-18 22:05:02.080848540 +0300
@@ -12,7 +12,7 @@
 include $(CLANG_LEVEL)/../../Makefile.config
 
 DIRS := libclang c-index-test arcmt-test c-arcmt-test
-PARALLEL_DIRS := driver diagtool clang-format
+PARALLEL_DIRS := driver diagtool clang-format include-what-you-use
 
 ifeq ($(ENABLE_CLANG_STATIC_ANALYZER),1)
   PARALLEL_DIRS += clang-check
--- include-what-you-use/fix_includes.py	(revision 468)
+++ include-what-you-use/fix_includes.py	(working copy)
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 ##===--- fix_includes.py - rewrite source files based on iwyu output ------===##
 #
