diff -Nur ./ctags-5.8/c.c ./ctags-5.8.1/c.c
--- ./ctags-5.8/c.c	2009-07-09 18:03:58.000000000 +0400
+++ ./ctags-5.8.1/c.c	2010-11-09 15:11:45.828125000 +0300
@@ -2707,8 +2707,26 @@
 					/*  For an anonymous struct or union we use a unique ID
 					 *  a number, so that the members can be found.
 					 */
-					char buf [20];  /* length of "_anon" + digits  + null */
-					sprintf (buf, "__anon%d", ++AnonymousID);
+					/* PATHED: added filename to anon structs name */
+#define STRUCT_NAME_LENGTH 60
+					char buf [STRUCT_NAME_LENGTH];  /* length of "_anon" + digits  + null */
+					int i;
+					char * pCurChar = buf;
+					sprintf (buf, "__anon_%s_%d", getInputFileShortName (), getSourceLineNumber ());
+					for (i = 0; i < STRUCT_NAME_LENGTH; i++, pCurChar++){
+						if (
+								*pCurChar == '.' 
+								|| *pCurChar == ':' 
+								|| *pCurChar == '\\' 
+								|| *pCurChar == '/' 
+								|| *pCurChar == ' ' 
+						   ){
+							*pCurChar = '_';
+						} else if (*pCurChar == 0){
+							break;
+						}
+
+					}
 					vStringCopyS (st->blockName->name, buf);
 					st->blockName->type = TOKEN_NAME;
 					st->blockName->keyword = KEYWORD_NONE;
diff -Nur ./ctags-5.8/ctags.h ./ctags-5.8.1/ctags.h
--- ./ctags-5.8/ctags.h	2009-07-09 18:03:58.000000000 +0400
+++ ./ctags-5.8.1/ctags.h	2010-11-09 12:17:40.640625000 +0300
@@ -15,12 +15,12 @@
 *   MACROS
 */
 #ifndef PROGRAM_VERSION
-# define PROGRAM_VERSION "5.8"
+# define PROGRAM_VERSION "5.8.1"
 #endif
 #define PROGRAM_NAME      "Exuberant Ctags"
 #define PROGRAM_URL       "http://ctags.sourceforge.net"
 #define PROGRAM_COPYRIGHT "Copyright (C) 1996-2009"
-#define AUTHOR_NAME       "Darren Hiebert"
+#define AUTHOR_NAME       "Darren Hiebert (Pathed by Dmitry Frank, dimon.frank@gmail.com)"
 #define AUTHOR_EMAIL      "dhiebert@users.sourceforge.net"
 
 #endif	/* _CTAGS_H */
diff -Nur ./ctags-5.8/read.c ./ctags-5.8.1/read.c
--- ./ctags-5.8/read.c	2009-07-09 18:03:58.000000000 +0400
+++ ./ctags-5.8.1/read.c	2010-11-09 16:21:15.140625000 +0300
@@ -25,6 +25,7 @@
 #include "main.h"
 #include "routines.h"
 #include "options.h"
+#include "vstring.h"
 
 /*
 *   DATA DEFINITIONS
@@ -80,6 +81,14 @@
 	if (File.source.name != NULL)
 		vStringDelete (File.source.name);
 	File.source.name = fileName;
+    //-- generating short filename (for anonymous structures). Its length must not be more than FILENAME_MAX_LENGTH
+    if (fileName->length <=  FILENAME_MAX_LENGTH){
+        File.source.shortName = fileName;
+    } else {
+        File.source.shortName = vStringNew();
+        vStringResize(File.source.shortName, FILENAME_MAX_LENGTH);
+        memcpy (File.source.shortName->buffer, &File.source.name->buffer[ File.source.name->length - FILENAME_MAX_LENGTH + 1/*this is null-symbol at end of line*/], FILENAME_MAX_LENGTH);
+    }
 
 	if (File.source.tagPath != NULL)
 		eFree (File.source.tagPath);
diff -Nur ./ctags-5.8/read.h ./ctags-5.8.1/read.h
--- ./ctags-5.8/read.h	2009-07-09 18:03:58.000000000 +0400
+++ ./ctags-5.8.1/read.h	2010-11-09 15:10:57.140625000 +0300
@@ -11,6 +11,8 @@
 #ifndef _READ_H
 #define _READ_H
 
+#define FILENAME_MAX_LENGTH 40
+
 #if defined(FILE_WRITE) || defined(VAXC)
 # define CONST_FILE
 #else
@@ -32,6 +34,7 @@
 *   MACROS
 */
 #define getInputLineNumber()     File.lineNumber
+#define getInputFileShortName()  vStringValue (File.source.shortName)
 #define getInputFileName()       vStringValue (File.source.name)
 #define getInputFilePosition()   File.filePosition
 #define getSourceFileName()      vStringValue (File.source.name)
@@ -85,6 +88,7 @@
 	 */
 	struct sSource {
 		vString *name;           /* name to report for source file */
+        vString *shortName;      /* name that will not be longer that FILENAME_MAX_LENGTH */
 		char    *tagPath;        /* path of source file relative to tag file */
 		unsigned long lineNumber;/* line number in the source file */
 		boolean  isHeader;       /* is source file a header file? */
diff -Nur ./ctags-5.8/vstring.c ./ctags-5.8.1/vstring.c
--- ./ctags-5.8/vstring.c	2009-07-09 18:03:58.000000000 +0400
+++ ./ctags-5.8.1/vstring.c	2010-11-09 14:35:09.234375000 +0300
@@ -31,7 +31,7 @@
 *   FUNCTION DEFINITIONS
 */
 
-static void vStringResize (vString *const string, const size_t newSize)
+extern void vStringResize (vString *const string, const size_t newSize)
 {
 	char *const newBuffer = xRealloc (string->buffer, newSize, char);
 
diff -Nur ./ctags-5.8/vstring.h ./ctags-5.8.1/vstring.h
--- ./ctags-5.8/vstring.h	2009-07-09 18:03:58.000000000 +0400
+++ ./ctags-5.8.1/vstring.h	2010-11-09 14:34:44.156250000 +0300
@@ -61,6 +61,7 @@
 *   FUNCTION PROTOTYPES
 */
 extern boolean vStringAutoResize (vString *const string);
+extern void vStringResize (vString *const string, const size_t newSize);
 extern void vStringClear (vString *const string);
 extern vString *vStringNew (void);
 extern void vStringDelete (vString *const string);
