diff --git a/Makefile b/Makefile
index f8d5048..9b169b4 100644
--- a/Makefile
+++ b/Makefile
@@ -55,5 +55,5 @@ donothing: # To protect against an idle "make" to "see what happens"
 	@echo " "
 	@echo "See the README_ files in ./Portability for more information."
 
-install: checkroot LICENSE_43R4_ACCEPTED MAILHOME_43R4 makedirs cwpstuff plot sustuff tristuff tetrastuff compstuff reflstuff
+install: makedirs cwpstuff plot sustuff tristuff tetrastuff compstuff reflstuff
