diff -Naur gdesklets-0.36.3.orig/main/TrayIcon.py gdesklets-0.36.3/main/TrayIcon.py
--- gdesklets-0.36.3.orig/main/TrayIcon.py	2010-06-07 09:41:57.000000000 +0200
+++ gdesklets-0.36.3/main/TrayIcon.py	2012-01-10 02:06:46.185535280 +0100
@@ -2,6 +2,10 @@
 
 import gtk
 
+try:
+    import appindicator
+except ImportError:
+    appindicator = None
 
 class TrayIcon:
 
@@ -11,13 +15,20 @@
 
         self.__traymenu = gtk.Menu()
 
-#        self.__trayicon = gtk.status_icon_new_from_file(ICON)
-#        self.__trayicon.connect("popup-menu", self.__on_button)
-        self.__trayicon = gtk.StatusIcon()
-        self.__trayicon.set_from_file(ICON)
-        self.__trayicon.set_tooltip("gDesklets")
-        self.__trayicon.connect("popup-menu", self.__on_button, self.__traymenu)
-        self.__trayicon.set_visible(True)
+        if not appindicator:
+#            self.__trayicon = gtk.status_icon_new_from_file(ICON)
+#            self.__trayicon.connect("popup-menu", self.__on_button)
+            self.__trayicon = gtk.StatusIcon()
+            self.__trayicon.set_from_file(ICON)
+            self.__trayicon.set_tooltip("gDesklets")
+            self.__trayicon.connect("popup-menu", self.__on_button, self.__traymenu)
+            self.__trayicon.set_visible(True)
+        else:
+            self.__appindicator = appindicator.Indicator("gDesklets",
+                                      "gdesklets",
+                                      appindicator.CATEGORY_APPLICATION_STATUS)
+            self.__appindicator.set_status(appindicator.STATUS_ACTIVE)
+            self.__appindicator.set_icon("gdesklets", "gDesklets")
 
 
 
@@ -53,5 +64,8 @@
 
             item.show()
             self.__traymenu.append(item)
-        self.__trayicon.set_visible(True)
+        if not appindicator:
+            self.__trayicon.set_visible(True)
+        else:
+            self.__appindicator.set_menu(self.__traymenu)
 
