diff -r 737cf52cd031 config.def.h
--- a/config.def.h	Sat Jun 23 20:12:49 2012 +0200
+++ b/config.def.h	Fri Jul 06 14:46:50 2012 +0800
@@ -39,7 +39,7 @@
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY Mod4Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
