Index: trunk/retroshare-nogui/src/notifytxt.cc
===================================================================
--- trunk/retroshare-nogui/src/notifytxt.cc	(revision 5082)
+++ trunk/retroshare-nogui/src/notifytxt.cc	(working copy)
@@ -27,6 +27,7 @@
 #include "notifytxt.h"
 #include <retroshare/rspeers.h>
 #include <string.h>
+#include <unistd.h>
 
 #include <iostream>
 #include <sstream>
Index: trunk/retroshare-nogui/src/retroshare.cc
===================================================================
--- trunk/retroshare-nogui/src/retroshare.cc	(revision 5082)
+++ trunk/retroshare-nogui/src/retroshare.cc	(working copy)
@@ -30,6 +30,7 @@
 
 #include "notifytxt.h"
 
+#include <unistd.h>
 #include <iostream>
 #ifdef WINDOWS_SYS
 #include <winsock2.h>
Index: trunk/retroshare-gui/src/main.cpp
===================================================================
--- trunk/retroshare-gui/src/main.cpp	(revision 5082)
+++ trunk/retroshare-gui/src/main.cpp	(working copy)
@@ -54,6 +54,7 @@
 #include <retroshare/rsinit.h>
 #include <retroshare/rsiface.h>
 #include "gui/notifyqt.h"
+#include <unistd.h>
 
 int main(int argc, char *argv[])
 { 
Index: trunk/libbitdht/src/udp/udpbitdht.cc
===================================================================
--- trunk/libbitdht/src/udp/udpbitdht.cc	(revision 5082)
+++ trunk/libbitdht/src/udp/udpbitdht.cc	(working copy)
@@ -31,6 +31,7 @@
 #include "bitdht/bencode.h"
 
 #include <stdlib.h>
+#include <unistd.h>		/* for usleep() */
 #include <iostream>
 #include <sstream>
 #include <iomanip>
Index: trunk/libretroshare/src/ft/ftserver.cc
===================================================================
--- trunk/libretroshare/src/ft/ftserver.cc	(revision 5082)
+++ trunk/libretroshare/src/ft/ftserver.cc	(working copy)
@@ -23,6 +23,8 @@
  *
  */
 
+#include <unistd.h>		/* for (u)sleep() */
+#include <iostream>
 #include <sstream>
 #include "util/rsdebug.h"
 #include "util/rsdir.h"
@@ -48,9 +50,6 @@
 
 #include "serialiser/rsserviceids.h"
 
-#include <iostream>
-#include <sstream>
-
 /***
  * #define SERVER_DEBUG 1
  * #define DEBUG_TICK   1
Index: trunk/libretroshare/src/ft/ftextralist.cc
===================================================================
--- trunk/libretroshare/src/ft/ftextralist.cc	(revision 5082)
+++ trunk/libretroshare/src/ft/ftextralist.cc	(working copy)
@@ -31,6 +31,7 @@
 #include "serialiser/rsconfigitems.h"
 #include "util/rsdir.h"
 #include <stdio.h>
+#include <unistd.h>		/* for (u)sleep() */
 
 /******
  * #define DEBUG_ELIST	1
Index: trunk/libretroshare/src/ft/ftcontroller.cc
===================================================================
--- trunk/libretroshare/src/ft/ftcontroller.cc	(revision 5082)
+++ trunk/libretroshare/src/ft/ftcontroller.cc	(working copy)
@@ -59,6 +59,7 @@
 
 #include "serialiser/rsconfigitems.h"
 #include <stdio.h>
+#include <unistd.h>		/* for (u)sleep() */
 #include <sstream>
 
 /******
Index: trunk/libretroshare/src/turtle/p3turtle.cc
===================================================================
--- trunk/libretroshare/src/turtle/p3turtle.cc	(revision 5082)
+++ trunk/libretroshare/src/turtle/p3turtle.cc	(working copy)
@@ -25,6 +25,7 @@
 
 //#define P3TURTLE_DEBUG 
 
+#include <unistd.h>
 #include <stdexcept>
 #include <stdlib.h>
 #include <assert.h>
Index: trunk/libretroshare/src/pqi/authgpg.cc
===================================================================
--- trunk/libretroshare/src/pqi/authgpg.cc	(revision 5082)
+++ trunk/libretroshare/src/pqi/authgpg.cc	(working copy)
@@ -34,6 +34,7 @@
 
 #include <util/rsdir.h>
 #include <util/pgpkey.h>
+#include <unistd.h>		/* for (u)sleep() */
 #include <iostream>
 #include <sstream>
 #include <algorithm>
Index: trunk/libretroshare/src/pqi/pqissllistener.cc
===================================================================
--- trunk/libretroshare/src/pqi/pqissllistener.cc	(revision 5082)
+++ trunk/libretroshare/src/pqi/pqissllistener.cc	(working copy)
@@ -39,6 +39,7 @@
 
 #include "util/rsdebug.h"
 #include <sstream>
+#include <unistd.h>
 
 const int pqissllistenzone = 49787;
 
Index: trunk/libretroshare/src/pqi/p3dhtmgr.cc
===================================================================
--- trunk/libretroshare/src/pqi/p3dhtmgr.cc	(revision 5082)
+++ trunk/libretroshare/src/pqi/p3dhtmgr.cc	(working copy)
@@ -23,6 +23,7 @@
  *
  */
 
+#include <unistd.h>
 #include <sstream>
 #include <iomanip>
 #include <stdio.h>
Index: trunk/libretroshare/src/pqi/pqinetwork.cc
===================================================================
--- trunk/libretroshare/src/pqi/pqinetwork.cc	(revision 5082)
+++ trunk/libretroshare/src/pqi/pqinetwork.cc	(working copy)
@@ -32,6 +32,7 @@
 #include <errno.h>
 #include <iostream>
 #include <stdio.h>
+#include <unistd.h>
 
 #include "util/rsdebug.h"
 #include <sstream>
Index: trunk/libretroshare/src/distrib/p3distrib.cc
===================================================================
--- trunk/libretroshare/src/distrib/p3distrib.cc	(revision 5082)
+++ trunk/libretroshare/src/distrib/p3distrib.cc	(working copy)
@@ -28,6 +28,7 @@
 #endif
 
 #include <stdio.h>
+#include <unistd.h>
 
 #include <openssl/err.h>
 #include <openssl/ssl.h>
Index: trunk/libretroshare/src/rsserver/p3face-server.cc
===================================================================
--- trunk/libretroshare/src/rsserver/p3face-server.cc	(revision 5082)
+++ trunk/libretroshare/src/rsserver/p3face-server.cc	(working copy)
@@ -30,6 +30,7 @@
 
 #include "tcponudp/tou.h"
 #include <sstream>
+#include <unistd.h>
 
 #include "pqi/authssl.h"
 #include <sys/time.h>
