diff -aur deadbeef-0.4.4/junklib.c deadbeef-0.4.4.new/junklib.c
--- deadbeef-0.4.4/junklib.c	2010-11-14 23:38:28.000000000 +0800
+++ deadbeef-0.4.4.new/junklib.c	2011-03-08 13:01:45.720000002 +0800
@@ -380,9 +380,9 @@
         enc = UTF8;
     }
     else if (encoding == 0) {
-        // hack to add limited cp1251 recoding support
+        // hack to add limited gbk recoding support
         if (can_be_russian (str)) {
-            enc = "cp1251";
+            enc = "gbk";
         }
     }
     else if (encoding != 1 && !(version == 4 && encoding == 3)){
@@ -415,9 +415,9 @@
         }
     }
     else if (encoding == 0) {
-        // hack to add limited cp1251 recoding support
+        // hack to add limited gbk recoding support
         if (can_be_russian (str)) {
-            enc = "cp1251";
+            enc = "gbk";
         }
         else {
             enc = "iso8859-1";
@@ -469,7 +469,7 @@
     }
     const char *enc = "iso8859-1";
     if (can_be_russian (str)) {
-        enc = "cp1251";
+        enc = "gbk";
     }
 
     if (junk_iconv (str, sz, out, sizeof (out), enc, UTF8) > 0) {
@@ -3009,7 +3009,7 @@
     }
     // check if that could be non-latin1 (too many nonascii chars)
     if (can_be_russian (s)) {
-        return "cp1251";
+        return "gbk";
     }
     return "iso8859-1";
 }
