--- make-buildinfo.py	2011-12-19 20:04:44.929313016 +0800
+++ make-buildinfo.py.new	2011-12-19 20:11:46.813694138 +0800
@@ -71,8 +71,9 @@
 
    for line in open(options.inputFile, 'rb'):
       for substKey in substitutions.keys():
-         if line.startswith(substKey):
+         if line.startswith(bytes(substKey,'utf8')):
             line = substKey + '=' + substitutions[substKey] + "\n"
+            line = line.encode('utf-8')
             break
       lines.append(line)
 
