diff -aur libjson-rpc-cpp-0.4.2.pristine/src/test/main.cpp libjson-rpc-cpp-0.4.2.new/src/test/main.cpp
--- libjson-rpc-cpp-0.4.2.pristine/src/test/main.cpp	2015-01-21 18:01:30.623344403 +0100
+++ libjson-rpc-cpp-0.4.2.new/src/test/main.cpp	2015-01-21 18:01:49.173344932 +0100
@@ -8,6 +8,4 @@
  ************************************************************************/
 
 #define BOOST_TEST_MODULE "libjson-rpc-cpp test suite"
-#define BOOST_TEST_DYN_LINK
-#define BOOST_TEST_MAIN
 #include <boost/test/unit_test.hpp>
