diff -rup kgraphviewer-2.1.1/src/kgraphviewer.cpp kgraphviewer-2.1.1-new/src/kgraphviewer.cpp
--- kgraphviewer-2.1.1/src/kgraphviewer.cpp	2010-11-12 09:07:15.000000000 +0100
+++ kgraphviewer-2.1.1-new/src/kgraphviewer.cpp	2014-03-16 11:19:59.881485776 +0100
@@ -138,14 +138,15 @@ void KGraphViewerWindow::reloadPreviousF
 void KGraphViewerWindow::openUrl(const KUrl& url)
 {
   kDebug() << url;
-  KLibFactory *factory = KLibLoader::self()->factory("kgraphviewerpart");
+  KPluginFactory *factory = KPluginLoader("kgraphviewerpart").factory();
   if (factory)
   {
-    KParts::ReadOnlyPart* part = static_cast<KParts::ReadOnlyPart*>(factory->create(this, "kgraphviewerpart"));
+    KParts::ReadOnlyPart* part = factory->create<KParts::ReadOnlyPart>("kgraphviewerpart", this);
     KGraphViewer::KGraphViewerInterface* kgv = qobject_cast<KGraphViewer::KGraphViewerInterface*>( part );
     if( ! kgv )
     {
       // This should not happen
+      kError() << "Failed to get KPart" << endl;
       return;
     }
     kgv->setBackgroundColor(KGraphViewerSettings::backgroundColor());
Only in kgraphviewer-2.1.1-new/src: kgraphviewer.cpp.orig
diff -rup kgraphviewer-2.1.1/src/part/kgraphviewer_interface.h kgraphviewer-2.1.1-new/src/part/kgraphviewer_interface.h
--- kgraphviewer-2.1.1/src/part/kgraphviewer_interface.h	2010-11-12 09:07:15.000000000 +0100
+++ kgraphviewer-2.1.1-new/src/part/kgraphviewer_interface.h	2014-03-16 11:19:59.881485776 +0100
@@ -48,9 +48,9 @@ class DotGraph;
  * Use it like this:
  * \code
  *  // fetch the Part..
- *  KLibFactory *factory = KLibLoader::self()->factory("kgraphviewerpart");
+ *  KPluginFactory *factory = KPluginLoader("kgraphviewerpart").factory();
  *  if (factory) {
- *      part = factory->create<KParts::ReadOnlyPart>(this);
+ *      KParts::ReadOnlyPart* part = factory->create<KParts::ReadOnlyPart>("kgraphviewerpart", this);
  *
  *      // cast the part to the KGraphViewerInterface...
  *      KGraphViewerInterface* graph = qobject_cast<KGraphViewerInterface*>( part );
Only in kgraphviewer-2.1.1-new/src/part: kgraphviewer_interface.h.orig
diff -rup kgraphviewer-2.1.1/src/part/kgraphviewer_part.cpp kgraphviewer-2.1.1-new/src/part/kgraphviewer_part.cpp
--- kgraphviewer-2.1.1/src/part/kgraphviewer_part.cpp	2010-11-12 09:07:15.000000000 +0100
+++ kgraphviewer-2.1.1-new/src/part/kgraphviewer_part.cpp	2014-03-16 11:21:22.866992430 +0100
@@ -46,6 +46,16 @@
 namespace KGraphViewer
 {
 
+K_PLUGIN_FACTORY(KGraphViewerPartFactory, registerPlugin<KGraphViewerPart>("kgraphviewerpart");)
+static KAboutData createAboutData()
+{
+    return KAboutData( "kgraphviewerpart", 0, ki18n("KGraphViewerPart"),
+                       "1.0", ki18n( "GraphViz dot files viewer" ),
+                       KAboutData::License_GPL,
+                       ki18n("(c) 2005-2006, Gaël de Chalendar <kleag at free.fr>"));
+}
+K_EXPORT_PLUGIN(KGraphViewerPartFactory(createAboutData()))
+
 class KGraphViewerPartPrivate
 {
 public:
@@ -65,7 +75,7 @@ public:
   
 };
 
-KGraphViewerPart::KGraphViewerPart( QWidget *parentWidget, QObject *parent)
+KGraphViewerPart::KGraphViewerPart( QWidget *parentWidget, QObject *parent, const QVariantList & )
 : KParts::ReadOnlyPart(parent), d(new KGraphViewerPartPrivate())
 {
   kDebug() ;
@@ -427,69 +437,6 @@ void KGraphViewerPart::slotRenameNode(co
   d->m_widget->graph()->renameNode(oldNodeName,newNodeName);
 }
 
-extern "C"
-{
-  /**
-  * This function is the 'main' function of this part.  It takes
-  * the form 'void *init_lib<library name>()  It always returns a
-  * new factory object
-  */
-  KDE_EXPORT void *init_kgraphviewerpart()
-  {
-    return new KGraphViewerPartFactory();
-  }
-}
-
-// KComponentData KGraphViewerPartFactory::s_instance = 0L;
-KAboutData* KGraphViewerPartFactory::s_about = new KAboutData(
-"kgraphviewerpart", 0, ki18n("KGraphViewerPart"),
-"1.0", ki18n( "GraphViz dot files viewer" ),
-KAboutData::License_GPL,
-ki18n("(c) 2005-2006, Gaël de Chalendar &lt;kleag@free.fr&gt;"));
-
-KComponentData KGraphViewerPartFactory::s_instance(s_about);
-
-KGraphViewerPartFactory::KGraphViewerPartFactory(QObject* parent)
-: KParts::Factory(parent)
-{
-}
-
-KGraphViewerPartFactory::~KGraphViewerPartFactory()
-{
-gvFreeContext(gvContext());
-delete s_about;
-}
-
-KParts::Part* KGraphViewerPartFactory::createPartObject( QWidget *parentWidget,
-QObject *parent,
-const char *classname,
-const QStringList &args )
-{
-  Q_UNUSED(classname);
-  Q_UNUSED(args);
-  //     Create an instance of our Part
-  KGraphViewerPart* obj = new KGraphViewerPart( parentWidget, parent);
-
-  //     See if we are to be read-write or not
-  //     if (QCString(classname) == "KParts::ReadOnlyPart")
-  //         obj->setReadWrite(false);
-
-  return obj;
-}
-
-KComponentData KGraphViewerPartFactory::componentData()
-{
-  /*    if( !s_instance )
-  {
-    s_about = new KAboutData( "kgraphviewerpart", 0, ki18n("KGraphViewerPart"),
-                              "1.0", ki18n( "GraphViz dot files viewer" ),
-                              KAboutData::License_GPL,
-                              ki18n("(c) 2005-2006, Gaël de Chalendar <kleag@free.fr>"));
-                              s_instance(s_about);
-  }*/
-  return s_instance;
-}
-                                                                                                                       
 }
 
 #include "kgraphviewer_part.moc"
Only in kgraphviewer-2.1.1-new/src/part: kgraphviewer_part.cpp.orig
diff -rup kgraphviewer-2.1.1/src/part/kgraphviewer_part.h kgraphviewer-2.1.1-new/src/part/kgraphviewer_part.h
--- kgraphviewer-2.1.1/src/part/kgraphviewer_part.h	2010-11-12 09:07:15.000000000 +0100
+++ kgraphviewer-2.1.1-new/src/part/kgraphviewer_part.h	2014-03-16 11:19:59.881485776 +0100
@@ -69,7 +69,7 @@ public:
     /**
      * Default constructor
      */
-    KGraphViewerPart(QWidget *parentWidget, QObject *parent);
+    KGraphViewerPart(QWidget *parentWidget, QObject *parent, const QVariantList &);
 
     
     /**
@@ -155,21 +155,5 @@ private:
   
 };
 
-class KGraphViewerPartFactory : public KParts::Factory
-{
-    Q_OBJECT
-public:
-    KGraphViewerPartFactory(QObject* parent = 0);
-    ~KGraphViewerPartFactory();
-    virtual KParts::Part* createPartObject( QWidget *parentWidget,
-                                            QObject *parent,
-                                            const char *classname, const QStringList &args );
-    static KComponentData componentData();
-
-private:
-    static KComponentData s_instance;
-    static KAboutData* s_about;
-};
-
 }
 #endif // _KGRAPHVIEWERPART_H_
Only in kgraphviewer-2.1.1-new/src/part: kgraphviewer_part.h.orig
