--- kgraphviewer-2.1.1/src/part/DotGraphParsingHelper.cpp	2010-11-12 09:07:15.000000000 +0100
+++ kgraphviewer-2.1.1-new/src/part/DotGraphParsingHelper.cpp	2014-03-16 11:33:58.183083349 +0100
@@ -25,13 +25,6 @@
 #include "graphnode.h"
 #include "graphedge.h"
 
-#include <boost/throw_exception.hpp>
-#include <boost/spirit/core.hpp>
-#include <boost/spirit/utility/distinct.hpp>
-#include <boost/spirit/utility/loops.hpp>
-#include <boost/spirit/utility/confix.hpp>
-
-
 #include <iostream>
 
 #include <kdebug.h>
