--- a/src/conky.c	2010-03-25 23:27:32.000000000 +0300
+++ b/src/conky.c	2010-05-20 17:32:58.000000000 +0400
@@ -376,6 +376,7 @@
 /* border */
 static int draw_borders;
 static int draw_graph_borders;
+static int draw_bar_borders;
 static int stippled_borders;
 
 int get_stippled_borders(void)
@@ -2946,11 +2947,14 @@
 						w = 0;
 					}
 
+					if (draw_bar_borders) {
 					XSetLineAttributes(display, window.gc, 1, LineSolid,
 						CapButt, JoinMiter);
 
 					XDrawRectangle(display, window.drawable, window.gc, cur_x,
 						by, w, h);
+					}
+
 					XFillRectangle(display, window.drawable, window.gc, cur_x,
 						by, w * bar_usage / 255, h);
 					if (h > cur_y_add
@@ -4257,6 +4261,7 @@
 	draw_shades = 1;
 	draw_borders = 0;
 	draw_graph_borders = 1;
+	draw_bar_borders = 1;
 	draw_outline = 0;
 	set_first_font("6x10");
 	gap_x = 5;
@@ -4774,6 +4779,9 @@
 		CONF("draw_graph_borders") {
 			draw_graph_borders = string_to_bool(value);
 		}
+		CONF("draw_bar_borders") {
+			draw_bar_borders = string_to_bool(value);
+		}
 		CONF("draw_shades") {
 			draw_shades = string_to_bool(value);
 		}
--- a/src/defconfig.h	2010-03-25 23:31:30.000000000 +0300
+++ b/src/defconfig.h	2010-05-20 17:33:55.000000000 +0400
@@ -36,6 +36,7 @@
 "default_shade_color white\n", \
 "draw_borders no\n", \
 "draw_graph_borders yes\n", \
+"draw_bar_borders yes\n", \
 "draw_outline no\n", \
 "draw_shades no\n", \
 "use_xft yes\n", \
