--- blackboard_git/start/Linux_Start.sh	2013-08-10 20:48:02.163631131 +0200
+++ blackboard_git/start/Linux_Start_alt.sh	2013-08-10 20:26:18.000000000 +0200
@@ -6,4 +6,14 @@
   cd $RELATIVEDIR
 fi
 
-java -Dsun.java2d.opengl=true -jar BlackBoard.jar
+if [ ! -f ~/.BlackBoard ]; then
+   mkdir ~/.BlackBoard
+fi
+for LIBDIR in models parts simulators symbols
+do
+  if [ ! -f ~/.BlackBoard/$LIBDIR ]; then
+    cp -a ./$LIBDIR ~/.BlackBoard/
+  fi
+done
+
+java -Dsun.java2d.opengl=true -jar BlackBoard.jar
\ No newline at end of file
