--- src/libstatgrab/disk_stats.c_alt	2011-01-13 09:16:57.856666673 +0100
+++ src/libstatgrab/disk_stats.c	2011-01-13 09:17:13.836666673 +0100
@@ -48,7 +48,7 @@
 
 #ifdef LINUX
 #define VALID_FS_TYPES {"adfs", "affs", "befs", "bfs", "efs", "ext2", \
-			"ext3", "vxfs", "hfs", "hfsplus", "hpfs", "jffs", \
+			"ext3", "ext4", "vxfs", "hfs", "hfsplus", "hpfs", "jffs", \
 			"jffs2", "minix", "msdos", "ntfs", "qnx4", "ramfs", \
 			"rootfs", "reiserfs", "sysv", "v7", "udf", "ufs", \
 			"umsdos", "vfat", "xfs", "jfs", "nfs"}
