From 0aa2afc1b5ce24329f65c507111bfc1de5c53706 Mon Sep 17 00:00:00 2001
From: Alam Arias <Alam.GBC@gmail.com>
Date: Fri, 8 Mar 2013 13:33:42 -0500
Subject: [PATCH] added back support for gtkmm-2.4 builds

---
 src/devicewidget.cc | 5 +++++
 src/devicewidget.h  | 4 ++++
 2 files changed, 9 insertions(+)

diff --git a/src/devicewidget.cc b/src/devicewidget.cc
index 1c34ea6..1a148ee 100644
--- a/src/devicewidget.cc
+++ b/src/devicewidget.cc
@@ -63,8 +63,13 @@
     for (unsigned i = 0; i < PA_CHANNELS_MAX; i++)
         channelWidgets[i] = NULL;
 
+#ifdef HAVE_GTK3
     offsetAdjustment = Gtk::Adjustment::create(0.0, -2000.0, 2000.0, 10.0, 50.0, 0.0);
     offsetButton->configure(offsetAdjustment, 0, 2);
+#else
+    offsetAdjustment = new Gtk::Adjustment(0.0, -2000.0, 2000.0, 10.0, 50.0, 0.0);
+    offsetButton->configure(*offsetAdjustment, 0.0, 2);
+#endif  /* HAVE_GTK3 */
 }
 
 void DeviceWidget::init(MainWindow* mainWindow, Glib::ustring deviceType) {
diff --git a/src/devicewidget.h b/src/devicewidget.h
index d2dff51..586c2cf 100644
--- a/src/devicewidget.h
+++ b/src/devicewidget.h
@@ -98,7 +98,11 @@ class DeviceWidget : public MinimalStreamWidget {
     Gtk::HBox *portSelect, *offsetSelect;
     Gtk::ComboBox *portList;
     Glib::RefPtr<Gtk::ListStore> treeModel;
+#ifdef HAVE_GTK3
     Glib::RefPtr<Gtk::Adjustment> offsetAdjustment;
+#else
+    Gtk::Adjustment *offsetAdjustment;
+#endif /* HAVE_GTK3 */
 
 private:
     Glib::ustring mDeviceType;
-- 
1.8.1.6

