diff -Naur kcometen4-1.0.2/src/CMakeLists.txt kcometen4-1.0.2-new/src/CMakeLists.txt
--- kcometen4-1.0.2/src/CMakeLists.txt	2008-09-03 09:54:06.000000000 -0700
+++ kcometen4-1.0.2-new/src/CMakeLists.txt	2008-09-06 04:29:17.000000000 -0700
@@ -1,4 +1,6 @@
 include_directories( ${KDE4_INCLUDES} )
+include_directories(/usr/include/Qt/)
+find_package(KDE4 REQUIRED)
 
 set( kcometen4_SRCS bezier.cpp cometen3.cpp cometenmath.cpp cometenscene.cpp
      curvecomet.cpp decal.cpp explosion.cpp glow.cpp kcometen4.cpp
