--- scripts/fribidi-config	2014-02-11 05:37:17.328857447 +0100
+++ scripts/fribidi-config.new	2014-02-11 05:45:04.874629654 +0100
@@ -1,9 +1,16 @@
 #!/bin/sh
 set -e
 
-OPTIONS=""
-
 BUILD="$(pwd)"
+
+if test -f "$BUILD"/fribidi_options ; then
+    USER_OPTS=$(cat "$BUILD"/fribidi_options)
+fi
+
+OPTIONS=" $USER_OPTS"
+
+echo Using fribidi options: $OPTIONS
+
 cd "$BUILD"/fribidi
 ./bootstrap
 ./configure --prefix="$BUILD/build_libs" --libdir="$BUILD/build_libs/lib" --enable-static --disable-shared --without-glib $OPTIONS
