Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 7831)
+++ CMakeLists.txt	(working copy)
@@ -384,12 +384,12 @@
         COMPONENT COMPONENT_FREEORION
     )
 
-    install(
-        FILES
-        ${CMAKE_SOURCE_DIR}/OISInput.cfg
-        DESTINATION share/freeorion/default
-        COMPONENT COMPONENT_FREEORION
-    )
+    #install(
+        #FILES
+        #${CMAKE_SOURCE_DIR}/OISInput.cfg
+        #DESTINATION share/freeorion/default
+        #COMPONENT COMPONENT_FREEORION
+    #)
 
     install(
         FILES
