--- Makefile.orig	2008-02-03 20:01:14.000000000 +0100
+++ Makefile	2008-02-03 20:01:23.000000000 +0100
@@ -58,8 +58,8 @@
 YACC    = /usr/bin/yacc
 
 # select Tcl/Tk version (8.5 recommended due to modernized widget appearence)
-TCL_VER := $(shell echo 'puts [package require Tcl]' | tclsh)
-#TCL_VER = 8.5
+#TCL_VER := $(shell echo 'puts [package require Tcl]' | tclsh)
+TCL_VER = 8.5
 
 ifeq ($(shell test -d /usr/include/tcl$(TCL_VER) && echo YES),YES)
 INCS   += -I/usr/include/tcl$(TCL_VER)
