diff -cr old_src//freecad/src/3rdParty/salomesmesh/src/SMESH/SMESH_MeshEditor.cpp new_src//freecad/src/3rdParty/salomesmesh/src/SMESH/SMESH_MeshEditor.cpp
*** old_src//freecad/src/3rdParty/salomesmesh/src/SMESH/SMESH_MeshEditor.cpp   2011-03-13 10:21:14.832281625 +0100
--- freecad/src/3rdParty/salomesmesh/src/SMESH/SMESH_MeshEditor.cpp   2011-03-13 10:17:19.460595577 +0100
***************
*** 2205,2212 ****
    if ( projector.IsDone() ) {
      double u, v, minVal = DBL_MAX;
      for ( int i = projector.NbExt(); i > 0; i-- )
!       if ( projector.Value( i ) < minVal ) {
!         minVal = projector.Value( i );
          projector.Point( i ).Parameter( u, v );
        }
      result.SetCoord( u, v );
--- 2205,2212 ----
    if ( projector.IsDone() ) {
      double u, v, minVal = DBL_MAX;
      for ( int i = projector.NbExt(); i > 0; i-- )
!       if ( projector.SquareDistance( i ) < minVal ) {
!         minVal = projector.SquareDistance( i );
          projector.Point( i ).Parameter( u, v );
        }
      result.SetCoord( u, v );
diff -cr old_src//freecad/src/3rdParty/salomesmesh/src/SMESH/SMESH_Pattern.cpp new_src//freecad/src/3rdParty/salomesmesh/src/SMESH/SMESH_Pattern.cpp
*** old_src//freecad/src/3rdParty/salomesmesh/src/SMESH/SMESH_Pattern.cpp   2011-03-13 10:21:14.832281625 +0100
--- freecad/src/3rdParty/salomesmesh/src/SMESH/SMESH_Pattern.cpp   2011-03-13 11:05:38.320707860 +0100
***************
*** 436,443 ****
    }
    double u, v, minVal = DBL_MAX;
    for ( int i = theProjectorPS.NbExt(); i > 0; i-- )
!     if ( theProjectorPS.Value( i ) < minVal ) {
!       minVal = theProjectorPS.Value( i );
        theProjectorPS.Point( i ).Parameter( u, v );
      }
    return gp_XY( u, v );
--- 436,443 ----
    }
    double u, v, minVal = DBL_MAX;
    for ( int i = theProjectorPS.NbExt(); i > 0; i-- )
!     if ( theProjectorPS.SquareDistance( i ) < minVal ) {
!       minVal = theProjectorPS.SquareDistance( i );
        theProjectorPS.Point( i ).Parameter( u, v );
      }
    return gp_XY( u, v );

