--- src/Mod/Part/Gui/ViewProviderExt.cpp.orig	2011-04-07 13:55:54.936376868 +0200
+++ src/Mod/Part/Gui/ViewProviderExt.cpp	2011-04-07 13:55:46.165957768 +0200
@@ -458,7 +458,8 @@
         // count triangles and nodes in the mesh
         TopExp_Explorer Ex;
         for(Ex.Init(cShape,TopAbs_FACE);Ex.More();Ex.Next()) {
-               Handle (Poly_Triangulation) mesh = BRep_Tool::Triangulation(TopoDS::Face(Ex.Current()),TopLoc_Location());
+               TopLoc_Location temp;
+               Handle (Poly_Triangulation) mesh = BRep_Tool::Triangulation(TopoDS::Face(Ex.Current()),temp);
                if (mesh.IsNull()) continue;
                nbrTriangles += mesh->NbTriangles();
                nbrNodes     += mesh->NbNodes();
