--- notalon.py.orig	2009-08-10 02:08:08.000000000 +0300
+++ notalon.py	2009-08-28 05:40:29.000000000 +0300
@@ -17,13 +17,18 @@ import logging
 import os.path
 import sys
 import wx
+import shutil
 
 from notalonlib.window import NotalonWindow
 
 # The next line deals with a known path problem in wxPython
-__builtin__.__dict__['homepath'] = os.path.abspath(os.path.dirname(sys.argv[0]))
+__builtin__.__dict__['homepath'] = os.path.expanduser('~/.config/notalon')
 __builtin__.__dict__['_'] = lambda x: x # TODO: later, include locales
 
+# Create homepath directory if it doesn't exist
+if not os.path.exists(homepath):
+    shutil.copytree('/usr/share/notalon', homepath)
+
 # Set up loggers programmatically (we want to hard-code the file location based
 # on `homepath`, which we can't do with a configuration file)
 for name in ['application', 'root', 'window', 'pdf', 'gui', 'resource']:
@@ -78,4 +83,3 @@ if __name__ == "__main__": # if called a
     app.MainLoop()
 
     log.debug('Finished main loop -- exiting.')
-
