--- setup.py.orig	2009-08-15 18:34:20.000000000 +0300
+++ setup.py	2009-08-28 06:18:41.000000000 +0300
@@ -35,10 +35,8 @@ if __name__ == '__main__':
                    ['notalonlib.widgets.dialogs'],
         scripts =  ['notalon.py', 'notalon.pyw'],
         data_files = [
-            ('res','notalon.ico'),
-            ('', 'preferences.xml'),
-            ('res', glob.glob(os.path.join('res','*.neme'))),
-            (os.path.join("locale", "en_US", "LC_MESSAGES"), glob.glob(os.path.join("locale", "en_US", "LC_MESSAGES", "*.mo")))
+            ('share/notalon/res', ['notalon.ico'] + glob.glob(os.path.join('res', '*.neme'))),
+            ('share/notalon', ['preferences.xml']),
+            (os.path.join('share', 'locale', 'en_US', 'LC_MESSAGES'), glob.glob(os.path.join('locale', 'en_US', 'LC_MESSAGES', '*.mo')))
         ],
     )
-
