--- old/config.omake	2009-05-18 01:29:33.000000000 +0200
+++ new/config.omake	2012-06-07 18:36:41.322652222 +0200
@@ -21,7 +21,7 @@
 
 ruby-conf(key) = 
    ConfMsgChecking(for Ruby $(key))
-   res=$(shell ruby -rrbconfig -e "puts Config::CONFIG['$(key)']")
+   res=$(shell ruby -rrbconfig -e "puts RbConfig::CONFIG['$(key)']")
    ConfMsgResult($(res))
    return $(res)
 
@@ -46,7 +46,7 @@
 		export
 	else
 		OCAMLDEP_MODULES_ENABLED = $(OCAMLDEP_MODULES_AVAILABLE)
-		OCAMLDEP=$(OCAMLDEP_MODULES)
+		#OCAMLDEP=$(OCAMLDEP_MODULES)
 	ConfMsgChecking(for ocaml-getopt)
 	GETOPT_DIR=$(shell ocamlfind query getopt)
 	ConfMsgResult($(GETOPT_DIR))
@@ -81,10 +81,11 @@
 
         # paths retrieved from Ruby
         $(CheckProg ruby)
-        RSPEC = $(CheckProgVerbose spec, RSpec testing framework, skipping Ruby tests)
+        RSPEC = $(CheckProgVerbose rspec, RSpec testing framework, skipping Ruby tests)
 
         RUBYVERSION = $(ruby-conf ruby_version)
-        if $(test $(RUBYVERSION) != 1.8)
+        #if $(test $(RUBYVERSION) != 1.8)
+        if $(test $(RUBYVERSION) != 1.9.1)
           eprintln(Only Ruby version 1.8.x is supported at this time)
           exit 1
         RUBYSITEDIR = $(conf-dir RUBYSITEDIR, $(ruby-conf sitelibdir))
