--- ../apertium-mk-en-0.1.0/Makefile.am	2010-09-19 15:54:43.000000000 +0200
+++ apertium-mk-en-0.1.0/Makefile.am	2010-09-23 16:39:28.995632217 +0200
@@ -78,7 +78,7 @@
 clean-dicts:
 	touch $(BASENAME).$(LANG1).dix $(BASENAME).$(LANG2).dix $(BASENAME).$(PREFIX1).dix
 	rm $(BASENAME).$(LANG1).dix
-	
+
 apertium_mk_enmodesdir=$(prefix)/share/apertium/modes/
 apertium_mk_endir=$(prefix)/share/apertium/apertium-$(PREFIX1)/
 apertium_mk_en_DATA=$(PREFIX1).automorf.bin $(PREFIX2).automorf.bin \
@@ -97,12 +97,12 @@
 		    $(BASENAME).$(PREFIX1).t3x \
 		    $(PREFIX1).autopgen.bin \
 		    $(PREFIX1).mode 
-		  
+
 install-data-local:
 	apertium-gen-modes modes.xml apertium-$(PREFIX1)
-	$(INSTALL_DATA) $(PREFIX1).mode $(apertium_mk_enmodesdir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t1x $(apertium_mk_endir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t2x $(apertium_mk_endir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t3x $(apertium_mk_endir)
+	$(INSTALL_DATA) $(PREFIX1).mode $(DESTDIR)$(apertium_mk_enmodesdir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t1x $(DESTDIR)$(apertium_mk_endir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t2x $(DESTDIR)$(apertium_mk_endir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t3x $(DESTDIR)$(apertium_mk_endir)
 
 CLEANFILES = -r $(TARGETS_COMMON) 
