diff -aur tribler-5.3.7.old/Tribler/Core/DecentralizedTracking/pymdht/plugins/routing_nice_rtt64.py tribler-5.3.7.new/Tribler/Core/DecentralizedTracking/pymdht/plugins/routing_nice_rtt64.py
--- tribler-5.3.7.old/Tribler/Core/DecentralizedTracking/pymdht/plugins/routing_nice_rtt64.py	2011-02-04 03:03:12.616246001 +0300
+++ tribler-5.3.7.new/Tribler/Core/DecentralizedTracking/pymdht/plugins/routing_nice_rtt64.py	2011-02-04 03:03:55.153246001 +0300
@@ -268,7 +268,7 @@
                 # 50% of the rnode's RTT (ie. two times faster). After two
                 # hours, a rnode cannot be replaced becouse of better RTT.
 #                print 'RTT replacement: newRTT: %f, oldRTT: %f, age: %f' % (
-                    rtt, rnode.rtt, current_time - rnode.bucket_insertion_ts)
+#                    rtt, rnode.rtt, current_time - rnode.bucket_insertion_ts)
                 rnode_to_be_replaced = rnode
                 break
         if rnode_to_be_replaced:
diff -aur tribler-5.3.7.old/Tribler/Test/test_torrentcollecting.py tribler-5.3.7.new/Tribler/Test/test_torrentcollecting.py
--- tribler-5.3.7.old/Tribler/Test/test_torrentcollecting.py	2011-02-04 03:04:47.587246001 +0300
+++ tribler-5.3.7.new/Tribler/Test/test_torrentcollecting.py	2011-02-04 03:03:27.956246001 +0300
@@ -7,7 +7,7 @@
 from Tribler.Core.BuddyCast.TorrentCollecting import SimpleTorrentCollecting
 from bak_tribler_sdb import *
     
-CREATE_SQL_FILE = os.path.join('..',"schema_sdb_v"+str(CURRENT_MAIN_DB_VERSION)+".sql"))
+CREATE_SQL_FILE = os.path.join('..',"schema_sdb_v"+str(CURRENT_MAIN_DB_VERSION)+".sql")
 assert os.path.isfile(CREATE_SQL_FILE)
 
 def init():
