diff -aur teuf-sandbox/src/db-artwork-parser.c teuf-sandbox.new/src/db-artwork-parser.c
--- teuf-sandbox/src/db-artwork-parser.c	2010-04-15 12:01:50.000000000 +0300
+++ teuf-sandbox.new/src/db-artwork-parser.c	2010-05-15 18:32:26.398382398 +0300
@@ -99,7 +99,7 @@
 }
 
 struct ParsedMhodString {
-        enum MhodType mhod_type;
+        enum MhodArtworkType mhod_type;
         char *mhod_string;
 };
 
